Index: libgnomevfs/gnome-vfs-mime-magic.c
===================================================================
RCS file: /cvs/gnome/gnome-vfs/libgnomevfs/gnome-vfs-mime-magic.c,v
retrieving revision 1.43
diff -u -r1.43 gnome-vfs-mime-magic.c
--- libgnomevfs/gnome-vfs-mime-magic.c	19 Dec 2002 17:57:19 -0000	1.43
+++ libgnomevfs/gnome-vfs-mime-magic.c	20 Jan 2003 12:57:20 -0000
@@ -855,6 +855,13 @@
 		&& (sniff_buffer->buffer[7] < 0x80u)
 		&& (sniff_buffer->buffer[8] < 0x80u)
 		&& (sniff_buffer->buffer[9] < 0x80u)) {
+		/* checks for existance of vorbis identification header */
+		for (offset=10; offset < GNOME_VFS_MP3_SNIFF_LENGTH-7; offset++) {
+			if (strncmp ((char *) &sniff_buffer->buffer[offset], 
+				     "\x01vorbis", 7) == 0) {
+				return FALSE;
+			}
+		}
 		return TRUE;
 	}
 
