diff -ur bitbake-0.orig/setup.py bitbake-0/setup.py
--- bitbake-0.orig/setup.py	2005-01-20 22:16:24.484916216 -0500
+++ bitbake-0/setup.py	2005-01-20 22:16:33.175595032 -0500
@@ -20,10 +20,8 @@
 from distutils.core import setup
 import os, sys
 
-# bbdir = os.path.join(sys.prefix, 'share', 'bitbake')
-# docdir = os.path.join(sys.prefix, 'share', 'doc')
-bbdir = os.path.join('bitbake')
-docdir = os.path.join('doc')
+bbdir = os.path.join(sys.prefix, 'share', 'bitbake')
+docdir = os.path.join(sys.prefix, 'share', 'doc')
 
 def clean_doc(type):
     origpath = os.path.abspath(os.curdir)
