diff -Naur NVIDIA-Linux-x86_64-1.0-6629-pkg2.orig/usr/src/nv/Makefile.kbuild NVIDIA-Linux-x86_64-1.0-6629-pkg2/usr/src/nv/Makefile.kbuild
--- NVIDIA-Linux-x86_64-1.0-6629-pkg2.orig/usr/src/nv/Makefile.kbuild	2004-11-11 14:55:52.000000000 -0800
+++ NVIDIA-Linux-x86_64-1.0-6629-pkg2/usr/src/nv/Makefile.kbuild	2004-11-11 14:58:11.000000000 -0800
@@ -73,7 +73,7 @@
 #
 
 EXTRA_CFLAGS += -I$(src)
-EXTRA_CFLAGS += -Wall -Wimplicit -Wreturn-type -Wswitch -Wformat -Wchar-subscripts -Wparentheses -Wpointer-arith  -Wno-multichar  -Werror -O -fno-common -mno-red-zone -MD $(DEFINES) $(INCLUDES) -Wno-cast-qual -Wno-error
+EXTRA_CFLAGS += -Wall -Wimplicit -Wreturn-type -Wswitch -Wformat -Wchar-subscripts -Wparentheses -Wno-multichar  -Werror -O -fno-common -mno-red-zone -MD $(DEFINES) $(INCLUDES) -Wno-cast-qual -Wno-error
 
 #
 # We rely on these two definitions below; if they aren't set, we set them to
