diff -Naur oprofile-0.8.2/module/oprofile.h oprofile-0.8.2-fixed/module/oprofile.h
--- oprofile-0.8.2/module/oprofile.h	2003-11-01 19:56:53.000000000 +0100
+++ oprofile-0.8.2-fixed/module/oprofile.h	2005-04-26 20:10:16.000000000 +0200
@@ -141,7 +141,7 @@
  * performance counter */
 extern struct op_int_operations op_rtc_ops;
 
-void FASTCALL(op_do_profile(uint cpu, long eip, long irq_enabled, int ctr));
+void op_do_profile(uint cpu, long eip, long irq_enabled, int ctr);
 extern struct _oprof_data oprof_data[NR_CPUS];
 extern struct oprof_sysctl sysctl_parms;
 extern int lproc_dointvec(ctl_table *table, int write, struct file *filp, void *buffer, size_t *lenp);
