--- CMakeLists.txt.old	2010-04-13 17:15:39.000000000 +0200
+++ CMakeLists.txt	2010-04-13 19:11:45.000000000 +0200
@@ -106,6 +106,7 @@
 set (BUILD_SHARED_LIBS OFF)
 set (Boost_USE_STATIC_LIBS   ON)
 set (Boost_USE_MULTITHREADED ON)
+set (Boost_ADDITIONAL_VERSIONS "1.42" "1.42.0")
 find_package(Boost 1.35.0 REQUIRED)
 include_directories(${Boost_INCLUDE_DIR})
 target_link_libraries(widelands ${Boost_LIBRARIES})
@@ -345,7 +346,7 @@
     worlds
     pics
   DESTINATION ${WLBUILD_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;Gentoo
   COMPONENT CoreDataFiles
   PATTERN "*.svn*" EXCLUDE
   PATTERN "SCons*" EXCLUDE
@@ -380,7 +381,7 @@
   set (WL_CREATE_SYMLINK_COMMAND "create_symlink")
 endif (WIN32)
 
-if (${CMAKE_BUILD_TYPE} STREQUAL Release)
+if (${CMAKE_BUILD_TYPE} STREQUAL Gentoo)
   add_custom_target (
     lang ALL
     COMMAND cmake -E echo "Preparing directory po"
@@ -399,7 +400,7 @@
     COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/utils/buildlocale.py
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   )
-else (${CMAKE_BUILD_TYPE} STREQUAL Release)
+else (${CMAKE_BUILD_TYPE} STREQUAL Gentoo)
   add_custom_target (
     lang
     COMMAND cmake -E echo "Preparing directory po"
@@ -423,7 +424,7 @@
     COMMAND cmake -E make_directory ${CMAKE_CURRENT_BINARY_DIR}/locale
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   )
-endif (${CMAKE_BUILD_TYPE} STREQUAL Release)
+endif (${CMAKE_BUILD_TYPE} STREQUAL Gentoo)
 
 set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES locale)
 
@@ -431,7 +432,7 @@
   DIRECTORY
     ${CMAKE_CURRENT_BINARY_DIR}/locale/
   DESTINATION ${WLBUILD_INSTALL_LOCALEDIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;Gentoo
   COMPONENT CoreLanguageFiles
   PATTERN "*.svn*" EXCLUDE
   PATTERN "SCons*" EXCLUDE
