--- build/scons-tools/scons_configure.py.old	2009-10-27 09:39:30.000000000 +0100
+++ build/scons-tools/scons_configure.py	2009-10-27 09:43:53.000000000 +0100
@@ -505,11 +505,7 @@
 	do_configure_basic_compiling(conf, env)
 	do_configure_locale(conf, env)
 	do_configure_libraries(conf, env)
-	do_configure_debugtools(conf, env)
 
-	do_configure_compiler_warnings(conf, env)
-	do_configure_compiler_features(conf, env)
-	do_configure_linker_features(conf, env)
 
 ################################################################################
 
