--- Makefile
+++ Makefile
@@ -61,9 +61,6 @@
 #  release    optimized
 #  profile    optimized, debugging symbols, profiling
 #
-ifndef BUILD
-BUILD:=debug
-endif
 
 endif
 
@@ -124,11 +121,6 @@
 BUILD:=debug
 OPTIMIZE:=yes
 DEBUG:=yes
-else
-BUILD:=debug-no-parachute
-OPTIMIZE:=YES
-DEBUG:=YES
-ADD_CFLAGS:=-DNO_PARACHUTE
 endif
 endif
 endif
@@ -139,7 +131,7 @@
 endif
 
 ifdef DEBUG
-ADD_CFLAGS += -g -DDEBUG  -fmessage-length=0
+ADD_CFLAGS += -DDEBUG
 else
 ADD_CFLAGS += -DNDEBUG
 endif
@@ -160,9 +152,9 @@
 # Object files and directories, final compilation flags
 
 OBJECT_DIR:=src/$(TARGET)-$(BUILD)
-CFLAGS:=-Wall $(shell $(SDL_CONFIG) --cflags) $(ADD_CFLAGS)
+CFLAGS:=-Wall $(shell $(SDL_CONFIG) --cflags) $(ADD_CFLAGS) $(CFLAGS)
 CXXFLAGS:=$(CFLAGS)
-LDFLAGS:=$(shell $(SDL_CONFIG) --libs) $(ADD_LDFLAGS) -lz -lpng -lSDL_image -lSDL_mixer -lSDL_ttf -lSDL_net
+LDFLAGS:=$(shell $(SDL_CONFIG) --libs) $(ADD_LDFLAGS) -lz -lpng -lSDL_image -lSDL_mixer -lSDL_ttf -lSDL_net $(LDFLAGS)
 
 ##############################################################################
 # Building
@@ -220,7 +212,7 @@
 -include $(DEP)
 
 $(OBJECT_DIR)/%.o: src/%.cc
-	$(CXX) -pipe $(CXXFLAGS) -MMD -MP -MF $@.d -c -o $@ $<
+	$(CXX) $(CXXFLAGS) -MMD -MP -MF $@.d -c -o $@ $<
 	sed -e 's@^\(.*\)\.o:@\1.d \1.o:@' $@.d > $(OBJECT_DIR)/$*.d
 	rm $@.d
 
