Modified: pym/portage.py
===================================================================
--- pym/portage.py	2006-05-06 13:23:02 UTC (rev 3325)
+++ pym/portage.py	2006-05-06 18:37:18 UTC (rev 3326)
@@ -977,9 +977,6 @@
 					writemsg("!!! Error: %s='%s' is not a directory. Please correct this.\n" % (k, v))
 					raise portage_exception.DirectoryNotFound(v)
 
-			self.backupenv["PORTAGE_CONFIGROOT"] = config_root
-			self.backupenv["ROOT"] = target_root
-
 			self.depcachedir = DEPCACHE_PATH
 
 			if not config_profile_path:
@@ -1329,6 +1326,11 @@
 		self.backupenv["PORTAGE_BIN_PATH"] = PORTAGE_BIN_PATH
 		self.backupenv["PORTAGE_PYM_PATH"] = PORTAGE_PYM_PATH
 
+		self["PORTAGE_CONFIGROOT"] = config_root
+		self.backup_changes("PORTAGE_CONFIGROOT")
+		self["ROOT"] = target_root
+		self.backup_changes("ROOT")
+
 		self._init_dirs()
 
 	def _init_dirs(self):
