--- src/ustring.cpp.old	2009-09-23 01:05:50.750887617 +0400
+++ src/ustring.cpp	2009-09-23 01:08:17.317619492 +0400
@@ -78,9 +78,9 @@
 {
 #if defined(HAVE_FUNC_ISINF)
   return (isinf(d) == 1);
-#elif HAVE_FUNC_FINITE
+#elif defined(HAVE_FUNC_FINITE)
   return finite(d) == 0 && d == d; // ### can we distinguish between + and - ?
-#elif HAVE_FUNC__FINITE
+#elif defined(HAVE_FUNC__FINITE)
   return _finite(d) == 0 && d == d; // ###
 #else
   return false;
@@ -91,9 +91,9 @@
 {
 #if defined(HAVE_FUNC_ISINF)
   return (isinf(d) == -1);
-#elif HAVE_FUNC_FINITE
+#elif defined(HAVE_FUNC_FINITE)
   return finite(d) == 0 && d == d; // ###
-#elif HAVE_FUNC__FINITE
+#elif defined(HAVE_FUNC__FINITE)
   return _finite(d) == 0 && d == d; // ###
 #else
   return false;
