--- admin/acinclude.m4.in.old	2008-08-19 22:20:50.000000000 +0200
+++ admin/acinclude.m4.in	2009-05-08 16:04:24.000000000 +0200
@@ -4649,18 +4649,20 @@
 
 AC_DEFUN([KDE_CHECK_PYTHON],
 [
-  KDE_CHECK_PYTHON_INTERN("2.5",
-    [KDE_CHECK_PYTHON_INTERN("2.4",
-     [KDE_CHECK_PYTHON_INTERN("2.3", 
-       [KDE_CHECK_PYTHON_INTERN("2.2", 
-         [KDE_CHECK_PYTHON_INTERN("2.1", 
-           [KDE_CHECK_PYTHON_INTERN("2.0", 
-             [KDE_CHECK_PYTHON_INTERN($1, $2) ])
-           ])
-         ])
-       ])
-     ])
-  ])
+  KDE_CHECK_PYTHON_INTERN("2.6",
+    [KDE_CHECK_PYTHON_INTERN("2.5",
+      [KDE_CHECK_PYTHON_INTERN("2.4",
+        [KDE_CHECK_PYTHON_INTERN("2.3", 
+          [KDE_CHECK_PYTHON_INTERN("2.2", 
+            [KDE_CHECK_PYTHON_INTERN("2.1", 
+              [KDE_CHECK_PYTHON_INTERN("2.0", 
+                [KDE_CHECK_PYTHON_INTERN($1, $2) ])
+              ])
+            ])
+          ])
+        ])
+      ])
+    ])
 ])
 
 AC_DEFUN([KDE_CHECK_STL],
