=== modified file 'bin/dobin'
--- bin/dobin	
+++ bin/dobin	
@@ -12,16 +12,11 @@
 	install -d "${D}${DESTTREE}/bin" || exit 2
 fi
 
-case ${CHOST} in
-	*-freebsd*) group=wheel ;;
-	*)          group=root ;;
-esac
-
 ret=0
 
 for x in "$@" ; do
 	if [[ -e ${x} ]] ; then
-		install -m0755 -o root -g ${group} "${x}" "${D}${DESTTREE}/bin"
+		install -m0755 -o 0 -g 0 "${x}" "${D}${DESTTREE}/bin"
 	else
 		echo "!!! ${0##*/}: ${x} does not exist" 1>&2
 		false

=== modified file 'bin/dosbin'
--- bin/dosbin	
+++ bin/dosbin	
@@ -12,16 +12,11 @@
 	install -d "${D}${DESTTREE}/sbin" || exit 2
 fi
 
-case ${CHOST} in
-	*-freebsd*) group=wheel ;;
-	*)          group=root ;;
-esac
-
 ret=0
 
 for x in "$@" ; do
 	if [[ -e ${x} ]] ; then
-		install -m0755 -o root -g ${group} "${x}" "${D}${DESTTREE}/sbin"
+		install -m0755 -o 0 -g 0 "${x}" "${D}${DESTTREE}/sbin"
 	else
 		echo "!!! ${0##*/}: ${x} does not exist" 1>&2
 		false

