--- setup.py.orig	2006-05-19 08:46:36.000000000 +0200
+++ setup.py	2006-05-19 08:46:43.000000000 +0200
@@ -8,13 +8,8 @@
 changelog_text = ""
 version = open('version.txt').read().strip()
 
-try:
-    from setuptools import setup
-    from setuptools.extension import Extension
-    setup_args['zip_safe'] = False
-except ImportError:
-    from distutils.core import setup
-    from distutils.extension import Extension
+from distutils.core import setup
+from distutils.extension import Extension
 
 try:
     from Pyrex.Distutils import build_ext as build_pyx
