diff -rc qgis-1.5.0/cmake/FindSIP.cmake qgis-1.5.0-patched/cmake/FindSIP.cmake
*** qgis-1.5.0/cmake/FindSIP.cmake	Sat Jan 16 12:08:04 2010
--- qgis-1.5.0-patched/cmake/FindSIP.cmake	Wed Nov 17 16:46:34 2010
***************
*** 36,41 ****
--- 36,42 ----
    EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} ${_find_sip_py} OUTPUT_VARIABLE sip_config)
    IF(sip_config)
      STRING(REGEX REPLACE "^sip_version:([^\n]+).*$" "\\1" SIP_VERSION ${sip_config})
+     STRING(REGEX REPLACE ".*\nsip_version_num:([^\n]+).*$" "\\1" SIP_VERSION_NUM ${sip_config})
      STRING(REGEX REPLACE ".*\nsip_version_str:([^\n]+).*$" "\\1" SIP_VERSION_STR ${sip_config})
      STRING(REGEX REPLACE ".*\nsip_bin:([^\n]+).*$" "\\1" SIP_BINARY_PATH ${sip_config})
      STRING(REGEX REPLACE ".*\ndefault_sip_dir:([^\n]+).*$" "\\1" SIP_DEFAULT_SIP_DIR ${sip_config})
diff -rc qgis-1.5.0/cmake/FindSIP.py qgis-1.5.0-patched/cmake/FindSIP.py
*** qgis-1.5.0/cmake/FindSIP.py	Sat Jan 16 12:04:41 2010
--- qgis-1.5.0-patched/cmake/FindSIP.py	Wed Nov 17 16:46:57 2010
***************
*** 9,14 ****
--- 9,15 ----
  
  sipcfg = sipconfig.Configuration()
  print("sip_version:%06.0x" % sipcfg.sip_version)
+ print("sip_version_num:%d" % sipcfg.sip_version)
  print("sip_version_str:%s" % sipcfg.sip_version_str)
  print("sip_bin:%s" % sipcfg.sip_bin)
  print("default_sip_dir:%s" % sipcfg.default_sip_dir)
diff -rc qgis-1.5.0/python/CMakeLists.txt qgis-1.5.0-patched/python/CMakeLists.txt
*** qgis-1.5.0/python/CMakeLists.txt	Sat May 15 10:21:42 2010
--- qgis-1.5.0-patched/python/CMakeLists.txt	Wed Nov 17 16:47:57 2010
***************
*** 39,44 ****
--- 39,48 ----
    SET(SIP_DISABLE_FEATURES ${SIP_DISABLE_FEATURES} PROXY_FACTORY)
  ENDIF(PYQT4_VERSION_NUM LESS 263680)
  
+ IF(NOT PYQT4_VERSION_NUM LESS 263941)	# 0x040705
+   SET(SIP_DISABLE_FEATURES ${SIP_DISABLE_FEATURES} QSETINT_CONVERSION)
+ ENDIF(NOT PYQT4_VERSION_NUM LESS 263941)
+ 
  # core module
  FILE(GLOB sip_files_core core/*.sip)
  set(SIP_EXTRA_FILES_DEPEND ${sip_files_core})
diff -rc qgis-1.5.0/python/core/conversions.sip qgis-1.5.0-patched/python/core/conversions.sip
*** qgis-1.5.0/python/core/conversions.sip	Tue Apr 20 14:25:42 2010
--- qgis-1.5.0-patched/python/core/conversions.sip	Wed Nov 17 16:45:03 2010
***************
*** 14,19 ****
--- 14,21 ----
  - QMap<int, QgsOverlayObject*>*
  */
  
+ %Feature QSETINT_CONVERSION
+ 
  %ModuleHeaderCode
  // From Python 2.5, some functions use Py_ssize_t instead of int
  // thus this typedef is for maintaining backward compatibility
***************
*** 265,270 ****
--- 267,273 ----
  
  
  
+ %If (QSETINT_CONVERSION)
  %MappedType QSet<int>
  {
  %TypeHeaderCode
***************
*** 316,321 ****
--- 319,325 ----
  %End
  
  };
+ %End
  
  
  template <TYPE>
