diff -ur gnupg-1.4.3.orig/doc/Makefile.in gnupg-1.4.3/doc/Makefile.in
--- gnupg-1.4.3.orig/doc/Makefile.in	2006-04-03 11:45:06.000000000 +0200
+++ gnupg-1.4.3/doc/Makefile.in	2006-04-09 14:10:13.000000000 +0200
@@ -303,7 +303,7 @@
              HACKING OpenPGP README.W32 samplekeys.asc gnupg.7 \
              TRANSLATE gpg.ru.sgml gpg.ru.1 highlights-1.4.txt
 
-man_MANS = gpg.1 gpgv.1 gnupg.7 gpg.ru.1
+man_MANS = gpg.1 gpgv.1 gnupg.7
 info_TEXINFOS = gpg.texi gpgv.texi
 
 # Need this to avoid building of dvis with automake 1.4
