diff -ur gnupg-1.4.3.orig/checks/Makefile.in gnupg-1.4.3/checks/Makefile.in
--- gnupg-1.4.3.orig/checks/Makefile.in	2006-04-03 11:45:05.000000000 +0200
+++ gnupg-1.4.3/checks/Makefile.in	2006-04-09 14:29:21.000000000 +0200
@@ -540,9 +540,11 @@
 
 ./gpg_dearmor:
 	echo '#!/bin/sh' >./gpg_dearmor
-	echo "../g10/gpg --no-options --no-greeting \
+	echo "../g10/gpg --no-options --no-greeting --homedir . \
              --no-secmem-warning --batch --dearmor" >>./gpg_dearmor
 	chmod 755 ./gpg_dearmor
+	if test ! -f ./pubring.gpg; then touch -t 197001020000 ./pubring.gpg; fi
+	if test ! -f ./secring.gpg; then touch -t 197001020000 ./secring.gpg; fi
 
 ./pubring.gpg: $(srcdir)/pubring.asc $(srcdir)/pubdemo.asc ./gpg_dearmor
 	./gpg_dearmor > ./pubring.gpg < $(srcdir)/pubring.asc
diff -ur gnupg-1.4.3.orig/g10/openfile.c gnupg-1.4.3/g10/openfile.c
--- gnupg-1.4.3.orig/g10/openfile.c	2006-03-28 09:41:45.000000000 +0200
+++ gnupg-1.4.3/g10/openfile.c	2006-04-09 14:30:42.000000000 +0200
@@ -401,8 +401,10 @@
      * To cope with HOME, we do compare only the suffix if we see that
      * the default homedir does start with a tilde.
      */
+#ifndef ENABLE_SELINUX_HACKS
     if( opt.dry_run || opt.no_homedir_creation )
 	return;
+#endif
 
     if ( ( *defhome == '~'
            && ( strlen(fname) >= strlen (defhome+1)
