diff -urNp gnupg-1.9.94.org/common/Makefile.am gnupg-1.9.94/common/Makefile.am
--- gnupg-1.9.94.org/common/Makefile.am	2006-10-17 13:49:02.000000000 +0200
+++ gnupg-1.9.94/common/Makefile.am	2006-10-30 21:21:39.000000000 +0200
@@ -80,6 +80,6 @@ module_tests = t-convert
 t_common_ldadd = ../jnlib/libjnlib.a ../common/libcommon.a ../gl/libgnu.a \
                  $(LIBGCRYPT_LIBS) $(GPG_ERROR_LIBS)
 
-t_convert_DEPENDENCIES = convert.c
+t_convert_DEPENDENCIES = convert.c libcommon.a
 t_convert_LDADD = $(t_common_ldadd)
 
