Index: gnupg-1.9.20/configure.ac
===================================================================
--- gnupg-1.9.20.orig/configure.ac
+++ gnupg-1.9.20/configure.ac
@@ -737,7 +737,7 @@ fi
 
 # Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS(string.h unistd.h langinfo.h termio.h locale.h)
+AC_CHECK_HEADERS(string.h unistd.h langinfo.h termio.h locale.h pty.h)
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
Index: gnupg-1.9.20/tools/symcryptrun.c
===================================================================
--- gnupg-1.9.20.orig/tools/symcryptrun.c
+++ gnupg-1.9.20/tools/symcryptrun.c
@@ -69,9 +69,12 @@
 #include <errno.h>
 #include <assert.h>
 #include <sys/stat.h>
+#include <sys/signal.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#ifdef HAVE_PTY_H
 #include <pty.h>
+#endif
 #include <utmp.h>
 #include <ctype.h>
 #ifdef HAVE_LOCALE_H
