--- configure.in.orig	2006-10-23 02:23:15.000000000 +0200
+++ configure.in	2006-10-23 02:24:45.000000000 +0200
@@ -228,7 +228,11 @@
       AC_MSG_ERROR(X11 headers or libraries not available: X11 driver disabled)
    else
       DRIVERS="$DRIVERS XWindow.o"
-      DRVLIBS="$DRVLIBS -L$ac_x_libraries -lX11"
+      if test "x$ac_x_libraries" = "x"; then
+         DRVLIBS="$DRVLIBS -lX11"
+      else
+         DRVLIBS="$DRVLIBS -L$ac_x_libraries -lX11"
+      fi
       AC_DEFINE(WITH_X11,1,[junk])
    fi
 fi
