diff -ru kmms-0.8beta1.orig/kmms/kmmsdock.cpp kmms-0.8beta1-fix-memory-leak/kmms/kmmsdock.cpp
--- kmms/kmmsdock.cpp	2002-08-19 10:54:37.000000000 +0930
+++ kmms/kmmsdock.cpp	2002-11-29 18:21:26.000000000 +1030
@@ -313,7 +313,9 @@
   volume          = xmms_remote_get_main_volume(0); if (volume > 100 || volume < 0) volume = 0;
   currentTime     = xmms_remote_get_output_time(0);
   trackTime       = xmms_remote_get_playlist_time(0, playlistpos);
-  temptitle       = xmms_remote_get_playlist_title(0, playlistpos);
+  char * tempstring = xmms_remote_get_playlist_title(0, playlistpos);
+  temptitle       = tempstring;
+  free(tempstring);
 
 	// speedup scrolling when title has changed
   if( qstrcmp(temptitle, lasttitle) != 0 && welcomeDisplayed ) {
