Ripped from Fedora.

--- traceroute-1.4a5/traceroute.c
+++ traceroute-1.4a5/traceroute.c
@@ -503,6 +496,16 @@
 
 	default:
 		usage();
+	}
+
+	/* This checking was moved here by oh3mqu+rpm@vip.fi */
+	/* It was useless before packlen gets command line value */
+	if (packlen == 0)
+		packlen = minpacket;		/* minimum sized packet */
+	else if (minpacket > packlen || packlen > maxpacket) {
+		Fprintf(stderr, "%s: packet size must be %d <= s <= %d\n",
+		    prog, minpacket, maxpacket);
+		exit(1);
 	}
 
 #ifdef HAVE_SETLINEBUF
