--- ../traceroute-1.4a12/traceroute.c	Thu Dec 14 10:04:50 2000
+++ traceroute.c	Thu Oct 23 13:09:39 2003
@@ -764,8 +764,12 @@
 	}
 #endif
 
-	Fprintf(stderr, "%s to %s (%s)",
-	    prog, hostname, inet_ntoa(to->sin_addr));
+	Fprintf(stderr, "%s to ",prog);
+	if (nflag)
+		Fprintf(stderr,"%s",hostname);
+	else
+		Fprintf(stderr,"%s",inetname(to->sin_addr));
+	Fprintf(stderr," (%s)", inet_ntoa(to->sin_addr));
 	if (source)
 		Fprintf(stderr, " from %s", source);
 	Fprintf(stderr, ", %d hops max, %d byte packets\n", max_ttl, packlen);

