Only in ecl-0.9d: Makefile
diff --exclude '*~' -ur ecl-0.9d.orig/Makefile.in ecl-0.9d/Makefile.in
--- ecl-0.9d.orig/Makefile.in	2004-11-29 18:46:39.000000000 -0600
+++ ecl-0.9d/Makefile.in	2004-11-29 20:53:09.479364087 -0600
@@ -80,13 +80,6 @@
 uninstall:
 	cd build; $(MAKE) uninstall
 
-# ==================== Documentation ====================
-
-info:
-	(cd ${srcdir}/doc; $(MAKE) info)
-dvi:
-	(cd ${srcdir}/doc; $(MAKE) dvi)
-
 # ==================== Cleaning up and miscellanea ====================
 
 #   `clean'
Only in ecl-0.9d: build
diff --exclude '*~' -ur ecl-0.9d.orig/src/Makefile.in ecl-0.9d/src/Makefile.in
--- ecl-0.9d.orig/src/Makefile.in	2004-11-29 18:46:39.000000000 -0600
+++ ecl-0.9d/src/Makefile.in	2004-11-29 20:48:57.000000000 -0600
@@ -40,7 +40,7 @@
 LSP_LIBRARIES = @LSP_LIBRARIES@
 TARGETS = @TARGETS@
 
-all:	$(TARGETS) ecl-config doc
+all:	$(TARGETS) ecl-config
 .PHONY:	all
 
 %Makefile: $(srcdir)/%Makefile.in config.status
@@ -94,7 +94,6 @@
 	head -8 config.status | tail -6 >> $@
 install: BUILD-STAMP install-base
 	cd c; $(MAKE) prefix=$(prefix) install
-	cd doc; $(MAKE) prefix=$(prefix) install
 install-base:
 	$(mkinstalldirs) $(bindir) $(libdir)/h
 	for i in $(TARGETS); do \
@@ -108,15 +107,10 @@
 flatinstall: BUILD-STAMP
 	$(MAKE) bindir=$(prefix) libdir=$(prefix) install-base
 	cd c; $(MAKE) prefix=$(prefix) flatinstall
-	cd doc; $(MAKE) prefix=$(prefix) flatinstall
 
 uninstall:
 	for i in $(TARGETS) ecl-config; do rm -rf $(bindir)/$$i; done
 	rm -rf $(libdir)
-	cd doc; $(MAKE) uninstall
-
-doc: $(TARGETS)
-	cd doc; $(MAKE)
 
 clean: clean_lisp
 	for i in ${SUBDIR}; do (cd $$i; $(MAKE) clean); done
