diff -ur ecl-0.9g.orig/src/h/ecl.h ecl-0.9g/src/h/ecl.h
--- ecl-0.9g.orig/src/h/ecl.h	2005-08-12 11:26:42.000000000 -0500
+++ ecl-0.9g/src/h/ecl.h	2005-08-13 13:10:59.000000000 -0500
@@ -13,6 +13,9 @@
     See file '../Copyright' for full details.
 */
 
+#ifndef __ECL_H__
+#define __ECL_H__
+
 #ifndef _MSC_VER
 #include <sys/param.h>		/* includes <sys/signal.h> and <sys/types.h> */
 #else
@@ -69,3 +72,5 @@
 #endif
 
 typedef void (*ecl_init_function_t)(cl_object block);
+
+#endif
diff -ur ecl-0.9g.orig/src/h/object.h ecl-0.9g/src/h/object.h
--- ecl-0.9g.orig/src/h/object.h	2005-08-12 11:26:42.000000000 -0500
+++ ecl-0.9g/src/h/object.h	2005-08-13 13:18:03.000000000 -0500
@@ -22,8 +22,13 @@
 	Integer and boolean types (see config.h)
 */
 
+#ifndef TRUE
 #define	TRUE		1	/*  boolean true value  */
+#endif
+
+#ifndef FALSE
 #define	FALSE		0	/*  boolean false value  */
+#endif
 
 #define	CHAR_CODE_LIMIT	256	/*  ASCII character code limit  */
 
