diff -urN libindi-0.7.2.old/libindi.pc.cmake libindi-0.7.2/libindi.pc.cmake
--- libindi-0.7.2.old/libindi.pc.cmake	2011-01-13 13:56:50.769108903 +0100
+++ libindi-0.7.2/libindi.pc.cmake	2011-01-13 13:57:13.421108903 +0100
@@ -8,5 +8,5 @@
 URL=http://www.indilib.org/
 Version: @CMAKE_INDI_VERSION_STRING@
 Libs: -L@LIB_DESTINATION@ -lindi
-Cflags: -I@INCLUDE_INSTALL_DIR@ -@INCLUDE_INSTALL_DIR@/libindi
+Cflags: -I@INCLUDE_INSTALL_DIR@ -I@INCLUDE_INSTALL_DIR@/libindi
 
