diff -ur gaim-0.75/configure gaim-0.75-branded/configure
--- gaim-0.75/configure	2004-03-24 14:12:41.729288968 -0500
+++ gaim-0.75-branded/configure	2004-03-24 14:11:18.347964856 -0500
@@ -1743,7 +1743,7 @@
 
 # Define the identity of the package.
  PACKAGE=gaim
- VERSION=GAIM_VERSION
+ VERSION=GENTOO_EBUILD_VERSION
 
 
 cat >>confdefs.h <<_ACEOF
diff -ur gaim-0.75/configure.ac gaim-0.75-branded/configure.ac
--- gaim-0.75/configure.ac	2004-03-24 14:12:41.730288816 -0500
+++ gaim-0.75-branded/configure.ac	2004-03-24 14:11:18.350964400 -0500
@@ -1,7 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(src/main.c)
 AM_CONFIG_HEADER(config.h)
-AM_INIT_AUTOMAKE([gaim], [GAIM_VERSION])
+AM_INIT_AUTOMAKE([gaim], [GENTOO_EBUILD_VERSION])
 
 AC_PREREQ([2.50])
 
diff -ur gaim-0.75/src/about.c gaim-0.75-branded/src/about.c
--- gaim-0.75/src/about.c	2004-03-24 14:12:41.675297176 -0500
+++ gaim-0.75-branded/src/about.c	2004-03-24 14:23:09.779810776 -0500
@@ -97,6 +97,15 @@
 				  "Gtk+ and is licensed under the GPL.<BR><BR>"), GTK_IMHTML_NO_SCROLL);
 
 	gtk_imhtml_append_text(GTK_IMHTML(text),
+			  _("This version of gaim has been modified by the Gentoo "
+				  "developers. If you experience any problems with it, do "
+				  "<B>NOT</B> report them as bugs in the gaim bug tracker, "
+					"and certainly do <B><U>NOT</U></B> seek help in #gaim. "
+					"Report them in the gentoo bugzilla at "
+					"<A HREF=\"http://bugs.gentoo.org\">http://bugs.gentoo.org</A>"
+				  ".<BR><BR>"), GTK_IMHTML_NO_SCROLL);
+
+	gtk_imhtml_append_text(GTK_IMHTML(text),
 			"<FONT SIZE=\"3\">URL:</FONT> <A HREF=\"" GAIM_WEBSITE "\">"
 			GAIM_WEBSITE "</A><BR><BR>", GTK_IMHTML_NO_SCROLL);
 
diff -ur gaim-0.75/VERSION gaim-0.75-branded/VERSION
--- gaim-0.75/VERSION	2004-03-24 14:12:41.731288664 -0500
+++ gaim-0.75-branded/VERSION	2004-03-24 14:11:18.351964248 -0500
@@ -1 +1 @@
-GAIM_VERSION
+GENTOO_EBUILD_VERSION
