diff -ruN gaim-0.71/src/protocols/jabber/message.c gaim-0.71-r1/src/protocols/jabber/message.c
--- gaim-0.71/src/protocols/jabber/message.c	2003-10-09 16:01:01.000000000 -0500
+++ gaim-0.71-r1/src/protocols/jabber/message.c	2003-10-16 14:46:14.000000000 -0500
@@ -228,13 +228,8 @@
 		} else if(!strcmp(child->name, "body")) {
 			if(!jm->body)
 				jm->body = xmlnode_get_data(child);
-		} else if(!strcmp(child->name, "html") && child->child) {
-			/* check to see if the <html> actually contains anything,
-			 * otherwise we'll ignore it */
-			char *txt = xmlnode_get_data(child);
-			if(!jm->xhtml && txt)
+		} else if(!strcmp(child->name, "html")) {
 				jm->xhtml = xmlnode_to_str(child);
-			g_free(txt);
 		} else if(!strcmp(child->name, "error")) {
 			const char *code = xmlnode_get_attrib(child, "code");
 			char *code_txt = NULL;
