Index: src/gtkimhtml.c
===================================================================
RCS file: /cvsroot/gaim/gaim/src/gtkimhtml.c,v
retrieving revision 1.343
diff -u -p -r1.343 gtkimhtml.c
--- src/gtkimhtml.c	9 Jun 2004 04:34:25 -0000	1.343
+++ src/gtkimhtml.c	15 Jun 2004 02:37:22 -0000
@@ -2348,13 +2348,13 @@ void gtk_imhtml_insert_html_at_iter(GtkI
 							if (font->underline != oldfont->underline)
 							    gtk_imhtml_toggle_underline(imhtml);
 
-							if (!oldfont->face || strcmp(font->face, oldfont->face) != 0)
+							if (font->face && (!oldfont->face || strcmp(font->face, oldfont->face) != 0))
 							    gtk_imhtml_toggle_fontface(imhtml, oldfont->face);
 
-							if (!oldfont->fore || strcmp(font->fore, oldfont->fore) != 0)
+							if (font->fore && (!oldfont->fore || strcmp(font->fore, oldfont->fore) != 0))
 							    gtk_imhtml_toggle_forecolor(imhtml, oldfont->fore);
 
-							if (!oldfont->back || strcmp(font->back, oldfont->back) != 0)
+							if (font->back && (!oldfont->back || strcmp(font->back, oldfont->back) != 0))
 						      gtk_imhtml_toggle_backcolor(imhtml, oldfont->back);
 						}
 
