diff -ur ../groff-1.19.1.orig/src/preproc/eqn/box.h src/preproc/eqn/box.h
--- ../groff-1.19.1.orig/src/preproc/eqn/box.h	2004-04-17 07:23:30.000000000 +0200
+++ src/preproc/eqn/box.h	2006-03-15 23:23:28.000000000 +0100
@@ -82,6 +82,10 @@
   friend box *make_lineup_box(box *);
 };
 
+extern box *make_script_box(box *, box *, box *);
+extern box *make_mark_box(box *);
+extern box *make_lineup_box(box *);
+
 enum alignment { LEFT_ALIGN, RIGHT_ALIGN, CENTER_ALIGN };
 
 class column : public box_list {
diff -ur ../groff-1.19.1.orig/src/roff/troff/div.h src/roff/troff/div.h
--- ../groff-1.19.1.orig/src/roff/troff/div.h	2006-03-15 23:19:12.000000000 +0100
+++ src/roff/troff/div.h	2006-03-15 23:15:11.000000000 +0100
@@ -56,6 +56,8 @@
   virtual void copy_file(const char *filename) = 0;
 };
 
+extern void end_diversions();
+
 class macro;
 
 class macro_diversion : public diversion {
diff -ur ../groff-1.19.1.orig/src/roff/troff/env.h src/roff/troff/env.h
--- ../groff-1.19.1.orig/src/roff/troff/env.h	2006-03-15 23:19:04.000000000 +0100
+++ src/roff/troff/env.h	2006-03-15 23:15:40.000000000 +0100
@@ -353,6 +353,8 @@
   friend void do_divert(int append, int boxing);
 };
 	
+extern void title();
+
 extern environment *curenv;
 extern void pop_env();
 extern void push_env(int);
diff -ur ../groff-1.19.1.orig/src/roff/troff/token.h src/roff/troff/token.h
--- ../groff-1.19.1.orig/src/roff/troff/token.h	2006-03-15 23:19:19.000000000 +0100
+++ src/roff/troff/token.h	2006-03-15 23:16:02.000000000 +0100
@@ -102,6 +102,8 @@
   friend void process_input_stack();
 };
 
+extern void process_input_stack();
+
 extern token tok;		// the current token
 
 extern symbol get_name(int required = 0);
