--- docs/kdebase/khelpcenter/userguide/shortcuts.docbook	2002-10-26 17:03:36.000000000 -0400
+++ docs/kdebase/khelpcenter/userguide/shortcuts.docbook	2003-01-28 16:50:33.000000000 -0500
@@ -26,7 +26,7 @@
 >Tab</keycap
 > </keycombo
 > o <keycombo
->&Alt;&Shift;<keycap
+>&Alt;<keycap>&Shift;</keycap><keycap
 >Tab</keycap
 ></keycombo
 ></term
@@ -42,7 +42,7 @@
 >Tab</keycap
 ></keycombo
 > o <keycombo
->&Ctrl;&Shift;<keycap
+>&Ctrl;<keycap>&Shift;</keycap><keycap
 >Tab</keycap
 > </keycombo
 ></term
@@ -102,7 +102,7 @@
 > <varlistentry
 > <term
 ><keycombo
->&Ctrl;&Alt;<keycap
+>&Ctrl;<keycap>&Alt;</keycap><keycap
 >Esc</keycap
 > </keycombo
 ></term
@@ -114,7 +114,7 @@
 > <varlistentry
 > <term
 ><keycombo
->&Ctrl;&Alt;<keycap
+>&Ctrl;<keycap>&Alt;</keycap><keycap
 >Retroceso</keycap
 > </keycombo
 ></term
@@ -126,7 +126,7 @@
 > <varlistentry
 > <term
 ><keycombo
->&Ctrl;&Alt;<keycap
+>&Ctrl;<keycap>&Alt;</keycap><keycap
 >Teclado numérico +</keycap
 ></keycombo
 ></term
@@ -138,7 +138,7 @@
 > <varlistentry
 > <term
 ><keycombo
->&Ctrl;&Alt;<keycap
+>&Ctrl;<keycap>&Alt;</keycap><keycap
 >Teclado numérico -</keycap
 > </keycombo
 ></term
@@ -223,8 +223,7 @@
 > <varlistentry
 > <term
 ><keycombo
->&Alt; botón <mousebutton
->izquierdo</mousebutton
+>&Alt;<mousebutton>botón izquierdo</mousebutton
 ></keycombo
 ></term
 > <listitem
@@ -235,8 +234,7 @@
 > <varlistentry
 > <term
 ><keycombo
->&Alt; botón<mousebutton
->central</mousebutton
+>&Alt;<mousebutton>botón central</mousebutton
 > </keycombo
 ></term
 > <listitem
--- docs/kdebase/konqueror/commands.docbook	2002-12-10 16:48:47.000000000 -0500
+++ docs/kdebase/konqueror/commands.docbook	2003-01-28 17:09:40.000000000 -0500
@@ -461,7 +461,7 @@
 ><menuchoice
 > <shortcut
 > <keycombo action="simul"
->&Ctrl;&Shift;<keycap
+>&Ctrl;<keycap>&Shift;</keycap><keycap
 >Supr</keycap
 ></keycombo
 > </shortcut
@@ -1589,7 +1589,7 @@
 ><menuchoice
 > <shortcut
 > <keycombo action="simul"
->&Ctrl;Mayúsculas<keycap
+>&Ctrl;<keycap>Mayúsculas</keycap><keycap
 >i</keycap
 ></keycombo
 ></shortcut
@@ -1609,7 +1609,7 @@
 ><menuchoice
 > <shortcut
 > <keycombo action="simul"
->&Ctrl;Mayúsculas<keycap
+>&Ctrl;<keycap>Mayúsculas</keycap><keycap
 >s</keycap
 ></keycombo
 ></shortcut
@@ -1629,7 +1629,7 @@
 ><menuchoice
 > <shortcut
 > <keycombo action="simul"
->&Ctrl;Mayúsculas<keycap
+>&Ctrl;<keycap>Mayúsculas</keycap><keycap
 >E</keycap
 ></keycombo
 ></shortcut
@@ -1663,7 +1663,7 @@
 ><menuchoice
 > <shortcut
 > <keycombo action="simul"
->&Ctrl;&Shift;<keycap
+>&Ctrl;<keycap>&Shift;</keycap><keycap
 >F</keycap
 ></keycombo
 > </shortcut
@@ -1683,7 +1683,7 @@
 ><menuchoice
 > <shortcut
 > <keycombo action="simul"
->&Ctrl;&Shift;<keycap
+>&Ctrl;<keycap>&Shift;</keycap><keycap
 >D</keycap
 ></keycombo
 > </shortcut
@@ -1703,7 +1703,7 @@
 ><menuchoice
 > <shortcut
 > <keycombo action="simul"
->&Ctrl;&Shift;<keycap
+>&Ctrl;<keycap>&Shift;</keycap><keycap
 >B</keycap
 ></keycombo
 > </shortcut
@@ -1743,7 +1743,7 @@
 ><menuchoice
 > <shortcut
 > <keycombo action="simul"
->&Ctrl;Mayúsculas<keycap
+>&Ctrl;<keycap>Mayúsculas</keycap><keycap
 >F</keycap
 ></keycombo
 > </shortcut
