diff -ur manpages-ru-0.7.orig/Makefile manpages-ru-0.7/Makefile
--- manpages-ru-0.7.orig/Makefile	2002-10-04 14:33:20.000000000 +0300
+++ manpages-ru-0.7/Makefile	2002-10-04 14:24:10.000000000 +0300
@@ -44,18 +44,18 @@
 
 all:
 	for d in 1 2 3 4 5 6 7 8; do \
-		make -C man$$d all; \
+		if [ -f man$$d/[Mm]akefile ] ; then make -C man$$d all; fi \
 	done
 
 clean:
 	for d in 1 2 3 4 5 6 7 8; do \
-		make -C man$$d clean; \
+		if [ -f man$$d/[Mm]akefile ] ; then make -C man$$d clean; fi \
 	done
 
 install: install-pages install-aliases
 
 install-pages:
-	[ -d $(INSTALLPATH) ] || mkdir $(INSTALLPATH)
+	[ -d $(INSTALLPATH) ] || mkdir -p $(INSTALLPATH)
 	[ -d $(INSTALLPATH)/$(LANG_SUBDIR) ] || mkdir $(INSTALLPATH)/$(LANG_SUBDIR)
 	if [ x$(COMPRESS) != xgzip -a \
 	     x$(COMPRESS) != xbzip2 -a \

