--- include/mod_gnutls.h.in
+++ include/mod_gnutls.h.in
@@ -52,10 +52,8 @@
 
 /* Recent Versions of 2.1 renamed several hooks. This allows us to 
    compile on 2.0.xx  */
-#if AP_SERVER_MINORVERSION_NUMBER >= 1
-#if AP_SERVER_PATCHLEVEL_NUMBER >= 3
+#if (AP_SERVER_MINORVERSION_NUMBER == 1 && AP_SERVER_PATCHLEVEL_NUMBER >= 3) || AP_SERVER_MINORVERSION_NUMBER >= 2
 #define USING_2_1_RECENT 1
-#endif
 #endif 
 
 #ifndef USING_2_1_RECENT
--- src/gnutls_cache.c
+++ src/gnutls_cache.c
@@ -38,6 +38,10 @@
 #define MC_TAG_LEN sizeof(MC_TAG)
 #define STR_SESSION_LEN (GNUTLS_SESSION_ID_STRING_LEN + MC_TAG_LEN)
 
+#if MODULE_MAGIC_NUMBER_MAJOR < 20081201
+#define ap_unixd_config unixd_config
+#endif
+
 char *mgs_session_id2sz(unsigned char *id, int idlen,
                                char *str, int strsize)
 {
@@ -544,9 +548,9 @@
 #if !defined(OS2) && !defined(WIN32) && !defined(BEOS) && !defined(NETWARE)
     /* Running as Root */
     if (geteuid() == 0)  {
-        chown(path1, unixd_config.user_id, -1);
+        chown(path1, ap_unixd_config.user_id, -1);
         if (path2 != NULL) { 
-            chown(path2, unixd_config.user_id, -1);
+            chown(path2, ap_unixd_config.user_id, -1);
         }
     }
 #endif
