diff -Naur convertfs-20020318/contrib/convertfs convertfs/contrib/convertfs
--- convertfs-20020318/contrib/convertfs	2002-03-18 17:15:15.000000000 +0100
+++ convertfs/contrib/convertfs	2004-06-27 11:41:49.265225592 +0200
@@ -192,7 +192,7 @@
 echo "== Creating clone of \`$from_fs_type' filesystem that's on \`$dev'. =="
 umount $dev 2>/dev/null
 mount -t $from_fs_type $dev $fs1root || error
-./devclone $dev $fsimage || error
+devclone $dev $fsimage || error
 
 echo "===== Creating destination \`$to_fs_type' filesystem. ====="
 umount $loopdev 2>/dev/null
@@ -216,13 +216,13 @@
 losetup -d $loopdev 2>/dev/null
 
 echo "=== Preparing info for block relocation ==="
-./prepindex $fsimage $fsindex $fssuper || error
+prepindex $fsimage $fsindex $fssuper || error
 umount $dev || error
 dd if=$fssuper of=$dev || exit
 sync
 
 echo "============ Relocating blocks ============"
-./devremap $dev || exit
+devremap $dev || exit
 
 echo "=== Filesystem conversion accomplished! ==="
 rm -rf $tmp
