commit 436d5379abe83c53c652d8a251f7ab504a1523df
Author: Colin Guthrie <colin@mageia.org>
Date:   Thu Aug 4 16:55:47 2011 +0100

    pulse: Use qPow() as pow() is sometimes not found.

diff --git a/phonon/pulsestream.cpp b/phonon/pulsestream.cpp
index d1e6711..e129f1b 100644
--- a/phonon/pulsestream.cpp
+++ b/phonon/pulsestream.cpp
@@ -21,6 +21,7 @@
 */
 
 #include "pulsestream_p.h"
+#include <QtCore/qmath.h>
 
 QT_BEGIN_NAMESPACE
 
@@ -81,7 +82,7 @@ void PulseStream::setVolume(const pa_cvolume *volume)
         // AudioOutput expects the "backend" to supply values that have been
         // adjusted for Stephens' law, so we need to fudge them accordingly
         // so that the %ages match up in KMix/the application's own slider.
-        emit volumeChanged(pow(vol, VOLTAGE_TO_LOUDNESS_EXPONENT));
+        emit volumeChanged(qPow(vol, VOLTAGE_TO_LOUDNESS_EXPONENT));
     }
 }
 
