diff -ru linux-2.4.22-openmosix-r1/mm/mmap.c linux-2.4.22-openmosix-r1.fixed/mm/mmap.c
--- linux-2.4.22-openmosix-r1/mm/mmap.c	2003-12-02 11:45:58.000000000 -0600
+++ linux-2.4.22-openmosix-r1.fixed/mm/mmap.c	2003-12-02 11:53:34.868028290 -0600
@@ -1391,6 +1391,9 @@
 	if (!len)
 		return addr;
 
+	if ((addr + len) > TASK_SIZE || (addr + len) < addr)
+		return -EINVAL;
+
 #ifdef CONFIG_MOSIX
 	if(current->mosix.dflags & DDEPUTY)
 		return(mosix_deputy_brk(addr, len));
