--- packETH-1.6.3/Makefile.orig	2009-01-20 16:07:51.000000000 +0100
+++ packETH-1.6.3/Makefile	2009-11-01 23:22:52.000000000 +0100
@@ -22,17 +22,15 @@
 prefix = /usr
 includedir = /usr/src/include
 
+packages = gthread-2.0 gtk+-x11-2.0
+
 INSTALL = /usr/bin/install -c
 INSTALL_DATA = ${INSTALL} -m 644
 INSTALL_PREFIX = /usr
 
-CC = gcc
-CPPFLAGS =  -I/usr/include/gtk-2.0 -I/usr/lib/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include 
-LDFLAGS = 
-CFLAGS = -g -O2 -Wall -Wunused -Wmissing-prototypes -Wmissing-declarations 
-#LIBS = -L/usr/lib -L/usr/X11R6/lib -lgtk -lgdk -lglib -lXi -lXext -lX11 -lm -lpthread -lgthread
-#LIBS = -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgdk_pixbuf-2.0 -lm -lpangocairo-1.0 -lpango-1.0 -lcairo -lgobject-2.0 -lgmodule-2.0 -ldl -lglib-2.0 -lpthread -lgthread-2.0
-LIBS = -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgdk_pixbuf-2.0 -lm -lgobject-2.0 -lglib-2.0 -lpthread -lgthread-2.0
+CPPFLAGS = $(shell pkg-config --cflags $(packages))
+CFLAGS += -Wall -Wunused -Wmissing-prototypes -Wmissing-declarations 
+LIBS = $(shell pkg-config --libs $(packages))
 
 SOURCES = \
 	main.c \
