diff -urN mercury-extras-10.04.2.orig/posix/Mmakefile mercury-extras-10.04.1/posix/Mmakefile
--- mercury-extras-10.04.2.orig/posix/Mmakefile	2008-09-02 21:45:54.000000000 +1200
+++ mercury-extras-10.04.2/posix/Mmakefile	2010-09-06 19:40:40.000000000 +1200
@@ -34,7 +34,7 @@
 
 .PHONY: install
 install: libposix.install
-	cp $(ADDITIONAL_HDRS) $(INSTALL_PREFIX)/lib/mercury/inc
+	cp $(ADDITIONAL_HDRS) $(INSTALL_PREFIX)/@libdir@/mercury/inc
 
 .PHONY: clean
 clean:
