--- mercury-extras-0.13.1.orig/references/Mmakefile	2006-04-21 17:01:39.000000000 +1200
+++ mercury-extras-0.13.1/references/Mmakefile	2006-12-04 22:07:20.000000000 +1300
@@ -10,7 +10,9 @@
 
 # Install in an "extras" subdirectory of the main installation tree
 INSTALL_PREFIX := $(INSTALL_PREFIX)/extras
-LIBGRADES = asm_fast.gc.tr asm_fast.gc.tr.debug
+LIBGRADES := 	$(foreach grade,$(LIBGRADES), \
+		$(subst .tr.,$(grade),$(findstring .tr.,$(grade))) \
+		$(filter %.tr,$(grade)))
 
 MAIN_TARGET = libglobal
 
