--- scotch-5.1.6.dfsg/src/libscotch/Makefile~	2008-05-29 09:01:12.000000000 +0000
+++ scotch-5.1.6.dfsg/src/libscotch/Makefile	2008-07-21 14:34:37.000000000 +0000
@@ -55,8 +55,8 @@
 					$(MAKE) CC="$(CCS)" CCD="$(CCS)" SCOTCHLIB=scotch	\
 					scotch.h						\
 					scotchf.h						\
-					libscotch$(LIB)						\
 					libscotcherr$(LIB)					\
+					libscotch$(LIB)						\
 					libscotcherrexit$(LIB)
 
 ptscotch			:	clean
@@ -2275,6 +2275,8 @@
 					vmesh_separate_zr$(OBJ)			\
 					vmesh_separate_st$(OBJ)			\
 					vmesh_store$(OBJ)
+					$(CC) -shared -Wl,-soname,lib$(SCOTCHLIB)-5.1.so -L. -lscotcherr -lz -lpthread -lm -lrt -o lib$(SCOTCHLIB)-5.1.so $^
+					ln -s lib$(SCOTCHLIB)-5.1.so lib$(SCOTCHLIB).so 
 					$(AR) $(ARFLAGS) lib$(SCOTCHLIB)$(LIB) $(?)
 					-$(RANLIB) lib$(SCOTCHLIB)$(LIB)
 
@@ -2438,9 +2440,13 @@
 					-$(RANLIB) $(@)
 
 libscotcherr$(LIB)		:	library_error$(OBJ)
+					$(CC) -shared -Wl,-soname,libscotcherr-5.1.so -o libscotcherr-5.1.so $^
+					ln -s libscotcherr-5.1.so libscotcherr.so 
 					$(AR) $(ARFLAGS) $(@) $(?)
 					-$(RANLIB) $(@)
 
 libscotcherrexit$(LIB)		:	library_error_exit$(OBJ)
+					$(CC) -shared -Wl,-soname,libscotcherrexit-5.1.so -o libscotcherrexit-5.1.so $^
+					ln -s libscotcherrexit-5.1.so libscotcherrexit.so 
 					$(AR) $(ARFLAGS) $(@) $(?)
 					-$(RANLIB) $(@)
--- scotch-5.0.6.dfsg/src/libscotchmetis/Makefile~	2008-05-07 16:23:07.000000000 +0000
+++ scotch-5.0.6.dfsg/src/libscotchmetis/Makefile	2008-07-18 19:45:15.000000000 +0000
@@ -136,5 +136,7 @@
 					metis_graph_order_f$(OBJ)		\
 					metis_graph_part$(OBJ)			\
 					metis_graph_part_f$(OBJ)
+					$(CC) -shared -Wl,-soname,libscotchmetis-5.1.so -L../libscotch -lscotch -o libscotchmetis-5.1.so $^
+					ln -s libscotchmetis-5.1.so libscotchmetis.so 
 					$(AR) $(ARFLAGS) $(@) $(^)
 					-$(RANLIB) $(@)
