diff -urN discover-2.0.1/discover/discover.c discover-2.0.1-new/discover/discover.c
--- discover-2.0.1/discover/discover.c	2002-10-11 11:07:26.000000000 -0700
+++ discover-2.0.1-new/discover/discover.c	2003-09-16 03:43:08.000000000 -0700
@@ -247,38 +247,14 @@
 static void
 print_usage(void)
 {
-    puts("\
-usage: discover [--bus-summary] [OPTIONS] [BUS [...]]
-       discover --type-summary [OPTIONS] [TYPE [...]]
-       discover --data-path=PATH [--data-version=VERSION] TYPE [...]
-       discover --version
-       discover --help");
+    puts("\nusage: discover [--bus-summary] [OPTIONS] [BUS [...]] \ndiscover --type-summary [OPTIONS] [TYPE [...]] \ndiscover --data-path=PATH [--data-version=VERSION] TYPE [...] \ndiscover --version \ndiscover --help");
 }
 
 static void
 print_help(void)
 {
     print_usage();
-    puts("
-  -b, --bus-summary          Summarize by bus.
-  -t, --type-summary         Summarize by device type.
-  --data-path PATH           Print the data matching PATH.
-
-OPTIONS
-  -d, --disable-bus BUS      Disable the bus BUS.
-  -e, --enable-bus BUS       Enable the bus BUS.
-  --data-version VERSION     Print only data matching VERSION.
-  --insert-url URL           Insert URL in front of the list of data sources.
-  --append-url URL           Append URL at the end of the list of data sources.
-  --[no-]vendor              Do [not] print the vendor name.
-  --[no-]vendor-id           Do [not] print the bus-specific vendor ID.
-  --[no-]model               Do [not] print the model name.
-  --[no-]model-id            Do [not] print the bus-specific model ID.
-  --normalize-whitespace     Condense all white space into individual spaces
-                             and eliminate leading and trailing white space.
-  --format TEMPLATE          Format the query data output.
-  -v, --verbose              Print verbose output.
-");
+    puts("\n-b, --bus-summary          Summarize by bus.\n-t, --type-summary         Summarize by device type.\n--data-path PATH           Print the data matching PATH.\n\nOPTIONS\n-d, --disable-bus BUS      Disable the bus BUS.\n-e, --enable-bus BUS       Enable the bus BUS.\n--data-version VERSION     Print only data matching VERSION.\n--insert-url URL           Insert URL in front of the list of datasources.\n--append-url URL           Append URL at the end of the list of datasources.\n--[no-]vendor              Do [not] print the vendor name.\n--[no-]vendor-id           Do [not] print the bus-specific vendor ID.\n--[no-]model               Do [not] print the model name.\n--[no-]model-id            Do [not] print the bus-specific model ID.\n--normalize-whitespace     Condense all white space into individual spaces\nand eliminate leading and trailing white\nspace.\n--format TEMPLATE          Format the query data output.\n-v, --verbose              Print verbose output.\n");
     printf("Report bugs to <%s>.\n", PACKAGE_BUGREPORT);
 }
 
diff -urN discover-2.0.1/discover-xml/discover-xml.c discover-2.0.1-new/discover-xml/discover-xml.c
--- discover-2.0.1/discover-xml/discover-xml.c	2002-10-16 14:35:15.000000000 -0700
+++ discover-2.0.1-new/discover-xml/discover-xml.c	2003-09-16 03:44:10.000000000 -0700
@@ -77,25 +77,14 @@
 static void
 print_usage()
 {
-    puts("
-usage: discover-xml [OPTIONS]
-       discover-xml --type-summary [OPTIONS]
-       discover-xml --version
-       discover-xml --help");
+    puts("\nusage: discover-xml [OPTIONS]\ndiscover-xml --type-summary [OPTIONS]\ndiscover-xml --version\ndiscover-xml --help");
 }
 
 static void
 print_help()
 {
     print_usage();
-    puts("
-  -t, --type-summary        Summarize by device type.
-
-OPTIONS
-  -d, --disable-bus BUS     Disable the bus BUS.
-  -e, --enable-bus BUS      Enable the bus BUS.
-  -v, --verbose             Print verbose output.
-");
+    puts("\n-t, --type-summary        Summarize by device type.\n\nOPTIONS\n-d, --disable-bus BUS     Disable the bus BUS.\n-e, --enable-bus BUS      Enable the bus BUS.\n-v, --verbose             Print verbose output.\n");
     printf("Report bugs to <%s>.\n", PACKAGE_BUGREPORT);
 }
 
