diff -Naur bnc2.8.6.org/bncsetup bnc2.8.6/bncsetup
--- bnc2.8.6.org/bncsetup	2000-12-31 09:19:39.000000000 +0100
+++ bnc2.8.6/bncsetup	2004-06-22 00:31:00.242167464 +0200
@@ -63,15 +63,15 @@
 
 }
 
-if ! [ -e ./mkpasswd ]
+if ! [ -e /usr/bin/bncmkpasswd ]
 then
   echo "You need to Make the package before you can use this script"
-  echo "(required ./mkpasswd)"
+  echo "(required /usr/bin/bncmkpasswd)"
   bsclean 0
 fi
 
 echo "BAD\$" > $TMPFILE
-echo "Test" | ./mkpasswd -s >> $TMPFILE
+echo "Test" | /usr/bin/bncmkpasswd -s >> $TMPFILE
 CRCYS=$(cat $TMPFILE | grep '\$' | tail -1 )
 
 
@@ -201,13 +201,13 @@
 
 if [ x$ERRORLEVEL = x0 ]
 then 
-  if [ -x ./mkpasswd ]
+  if [ -x /usr/bin/bncmkpasswd ]
   then
-    SUPERPASS="+"$(echo "$SUPERPASS" | ./mkpasswd -s)
-    BNCPASS="+"$(echo "$BNCPASS" | ./mkpasswd -s)
+    SUPERPASS="+"$(echo "$SUPERPASS" | /usr/bin/bncmkpasswd -s)
+    BNCPASS="+"$(echo "$BNCPASS" | /usr/bin/bncmkpasswd -s)
   else
     clear
-    echo "You have not did ./configure and make so there is no ./mkpasswd"
+    echo "You have not did ./configure and make so there is no /usr/bin/bncmkpasswd"
     echo "This program is required to produce the encryption"
     bsclean 0
   fi
