--- uucp-1.06.1/policy.h	1995-08-20 19:36:06.000000000 +0000
+++ uucp-1.06.1.old/policy.h	2003-04-12 07:33:43.000000000 +0000
@@ -295,13 +295,14 @@
    these lock files will be placed in the named directory; otherwise
    they will be placed in the default spool directory.  On some HDB
    systems the lock files are placed in /etc/locks.  On some they are
-   placed in /usr/spool/locks.  On the NeXT they are placed in
-   /usr/spool/uucp/LCK.  */
-/* #define LOCKDIR "/usr/spool/uucp" */
+   placed in /var/spool/locks.  On the NeXT they are placed in
+   /var/spool/uucp/LCK.  */
+/* #define LOCKDIR "/var/spool/uucp" */
 /* #define LOCKDIR "/etc/locks" */
-/* #define LOCKDIR "/usr/spool/locks" */
-/* #define LOCKDIR "/usr/spool/uucp/LCK" */
+/* #define LOCKDIR "/var/spool/locks" */
+/* #define LOCKDIR "/var/spool/uucp/LCK" */
 /* #define LOCKDIR "/var/spool/lock" */
+# define LOCKDIR "/var/lock/subsys"
 
 /* You must also specify the format of the lock files by setting
    exactly one of the following macros to 1.  Check an existing lock
@@ -522,7 +523,7 @@
 
    If you define HAVE_HDB_LOGGING, each program will by default use a
    separate log file.  For uucico talking to uunet, for example, it
-   will be /usr/spool/uucp/.Log/uucico/uunet.  Each line will look
+   will be /var/spool/uucp/.Log/uucico/uunet.  Each line will look
    something like this:
 
    uucp uunet (12/10-09:04:22,16390,1) Receiving uunet/D./D.uunetSwJ72
@@ -571,16 +572,16 @@
 /* The name of the default spool directory.  If HAVE_TAYLOR_CONFIG is
    set to 1, this may be overridden by the ``spool'' command in the
    configuration file.  */
-#define SPOOLDIR "/usr/spool/uucp"
 /* #define SPOOLDIR "/var/spool/uucp" */
+ #define SPOOLDIR "/var/spool/uucp" 
 
 /* The name of the default public directory.  If HAVE_TAYLOR_CONFIG is
    set to 1, this may be overridden by the ``pubdir'' command in the
    configuration file.  Also, a particular system may be given a
    specific public directory by using the ``pubdir'' command in the
    system file.  */
-#define PUBDIR "/usr/spool/uucppublic"
 /* #define PUBDIR "/var/spool/uucppublic" */
+ #define PUBDIR "/var/spool/uucppublic" 
 
 /* The default command path.  This is a space separated list of
    directories.  Remote command executions requested by uux are looked
@@ -643,20 +644,20 @@
 /* The default log file when using HAVE_TAYLOR_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``logfile''
    command in the configuration file.  */
-#define LOGFILE "/usr/spool/uucp/Log"
-/* #define LOGFILE "/var/spool/uucp/Log" */
+/*#define LOGFILE "/var/log/uucp/Log" */
+ #define LOGFILE "/var/log/uucp/Log" 
 
 /* The default statistics file when using HAVE_TAYLOR_LOGGING.  When
    using HAVE_TAYLOR_CONFIG, this may be overridden by the
    ``statfile'' command in the configuration file.  */
-#define STATFILE "/usr/spool/uucp/Stats"
-/* #define STATFILE "/var/spool/uucp/Stats" */
+/*#define STATFILE "/var/log/uucp/Stats" */
+ #define STATFILE "/var/log/uucp/Stats" 
 
 /* The default debugging file when using HAVE_TAYLOR_LOGGING.  When
    using HAVE_TAYLOR_CONFIG, this may be overridden by the
    ``debugfile'' command in the configuration file.  */
-#define DEBUGFILE "/usr/spool/uucp/Debug"
-/* #define DEBUGFILE "/var/spool/uucp/Debug" */
+/*#define DEBUGFILE "/var/log/uucp/Debug" */
+#define DEBUGFILE "/var/log/uucp/Debug" 
 
 #endif /* HAVE_TAYLOR_LOGGING */
 
@@ -665,17 +666,19 @@
 /* The default log file when using HAVE_V2_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``logfile''
    command in the configuration file.  */
-#define LOGFILE "/usr/spool/uucp/LOGFILE"
+/*#define LOGFILE "/var/log/uucp/LOGFILE" */
+#define LOGFILE "/var/log/uucp/LOGFILE"
 
 /* The default statistics file when using HAVE_V2_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``statfile''
    command in the configuration file.  */
-#define STATFILE "/usr/spool/uucp/SYSLOG"
+/* #define STATFILE "/var/log/uucp/SYSLOG" */
+#define STATFILE "/var/log/uucp/SYSLOG"
 
 /* The default debugging file when using HAVE_V2_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``debugfile''
    command in the configuration file.  */
-#define DEBUGFILE "/usr/spool/uucp/DEBUG"
+#define DEBUGFILE "/var/log/uucp/DEBUG"
 
 #endif /* HAVE_V2_LOGGING */
 
@@ -688,16 +691,16 @@
    be replaced by the system name (if there is no appropriate system,
    "ANY" will be used).  No other '%' character may appear in the
    string.  */
-#define LOGFILE "/usr/spool/uucp/.Log/%s/%s"
+#define LOGFILE "/var/log/uucp/.Log/%s/%s"
 
 /* The default statistics file when using HAVE_HDB_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``statfile''
    command in the configuration file.  */
-#define STATFILE "/usr/spool/uucp/.Admin/xferstats"
+#define STATFILE "/var/log/uucp/.Admin/xferstats"
 
 /* The default debugging file when using HAVE_HDB_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``debugfile''
    command in the configuration file.  */
-#define DEBUGFILE "/usr/spool/uucp/.Admin/audit.local"
+#define DEBUGFILE "/var/log/uucp/.Admin/audit.local"
 
 #endif /* HAVE_HDB_LOGGING */
