diff -Naur uucp-1.07.old/policy.h uucp-1.07/policy.h
--- uucp-1.07.old/policy.h	2003-05-29 06:08:45.000000000 +0000
+++ uucp-1.07/policy.h	2003-09-14 17:09:13.000000000 +0000
@@ -303,7 +303,7 @@
 /* #define LOCKDIR "/usr/spool/uucp/LCK" */
 /* #define LOCKDIR "/var/spool/lock" */
 /* #define LOCKDIR "/var/lock" */
-
+#define LOCKDIR "/var/log/subsys"
 /* You must also specify the format of the lock files by setting
    exactly one of the following macros to 1.  Check an existing lock
    file to decide which of these choices is more appropriate.
@@ -572,16 +572,16 @@
 /* The name of the default spool directory.  If HAVE_TAYLOR_CONFIG is
    set to 1, this may be overridden by the ``spool'' command in the
    configuration file.  */
-#define SPOOLDIR "/usr/spool/uucp"
-/* #define SPOOLDIR "/var/spool/uucp" */
+/*#define SPOOLDIR "/usr/spool/uucp" */
+ #define SPOOLDIR "/var/spool/uucp"
 
 /* The name of the default public directory.  If HAVE_TAYLOR_CONFIG is
    set to 1, this may be overridden by the ``pubdir'' command in the
    configuration file.  Also, a particular system may be given a
    specific public directory by using the ``pubdir'' command in the
    system file.  */
-#define PUBDIR "/usr/spool/uucppublic"
-/* #define PUBDIR "/var/spool/uucppublic" */
+/*#define PUBDIR "/usr/spool/uucppublic" */
+ #define PUBDIR "/var/spool/uucppublic" 
 
 /* The default command path.  This is a space separated list of
    directories.  Remote command executions requested by uux are looked
@@ -644,23 +644,23 @@
 /* The default log file when using HAVE_TAYLOR_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``logfile''
    command in the configuration file.  */
-#define LOGFILE "/usr/spool/uucp/Log"
+/*#define LOGFILE "/usr/spool/uucp/Log" */
 /* #define LOGFILE "/var/spool/uucp/Log" */
-/* #define LOGFILE "/var/log/uucp/Log" */
+ #define LOGFILE "/var/log/uucp/Log" 
 
 /* The default statistics file when using HAVE_TAYLOR_LOGGING.  When
    using HAVE_TAYLOR_CONFIG, this may be overridden by the
    ``statfile'' command in the configuration file.  */
-#define STATFILE "/usr/spool/uucp/Stats"
+/*#define STATFILE "/usr/spool/uucp/Stats" */
 /* #define STATFILE "/var/spool/uucp/Stats" */
-/* #define STATFILE "/var/log/uucp/Stats" */
+ #define STATFILE "/var/log/uucp/Stats" 
 
 /* The default debugging file when using HAVE_TAYLOR_LOGGING.  When
    using HAVE_TAYLOR_CONFIG, this may be overridden by the
    ``debugfile'' command in the configuration file.  */
-#define DEBUGFILE "/usr/spool/uucp/Debug"
+/*#define DEBUGFILE "/usr/spool/uucp/Debug" */
 /* #define DEBUGFILE "/var/spool/uucp/Debug" */
-/* #define DEBUGFILE "/var/log/uucp/Debug" */
+#define DEBUGFILE "/var/log/uucp/Debug" 
 
 #endif /* HAVE_TAYLOR_LOGGING */
 
@@ -669,17 +669,20 @@
 /* The default log file when using HAVE_V2_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``logfile''
    command in the configuration file.  */
-#define LOGFILE "/usr/spool/uucp/LOGFILE"
+/*#define LOGFILE "/usr/spool/uucp/LOGFILE" */
+#define LOGFILE "/var/log/uucp/LOGFILE"
 
 /* The default statistics file when using HAVE_V2_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``statfile''
    command in the configuration file.  */
-#define STATFILE "/usr/spool/uucp/SYSLOG"
+/*#define STATFILE "/usr/spool/uucp/SYSLOG" */
+#define STATFILE "/var/log/uucp/SYSLOG"
 
 /* The default debugging file when using HAVE_V2_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``debugfile''
    command in the configuration file.  */
-#define DEBUGFILE "/usr/spool/uucp/DEBUG"
+/*#define DEBUGFILE "/usr/spool/uucp/DEBUG" */
+#define DEBUGFILE "/var/log/uucp/DEBUG"
 
 #endif /* HAVE_V2_LOGGING */
 
@@ -692,16 +695,19 @@
    be replaced by the system name (if there is no appropriate system,
    "ANY" will be used).  No other '%' character may appear in the
    string.  */
-#define LOGFILE "/usr/spool/uucp/.Log/%s/%s"
+/*#define LOGFILE "/usr/spool/uucp/.Log/%s/%s" */
+#define LOGFILE "/var/log/uucp/.Log/%s/%s"
 
 /* The default statistics file when using HAVE_HDB_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``statfile''
    command in the configuration file.  */
-#define STATFILE "/usr/spool/uucp/.Admin/xferstats"
+/*#define STATFILE "/usr/spool/uucp/.Admin/xferstats" */
+#define STATFILE "/var/log/uucp/.Admin/xferstats"
 
 /* The default debugging file when using HAVE_HDB_LOGGING.  When using
    HAVE_TAYLOR_CONFIG, this may be overridden by the ``debugfile''
    command in the configuration file.  */
-#define DEBUGFILE "/usr/spool/uucp/.Admin/audit.local"
+/*#define DEBUGFILE "/usr/spool/uucp/.Admin/audit.local" */
+#define DEBUGFILE "/var/log/uucp/.Admin/audit.local"
 
 #endif /* HAVE_HDB_LOGGING */
