--- src/mtoolsfm.h.org	2003-12-22 16:52:27.251534559 +0100
+++ src/mtoolsfm.h	2003-12-22 16:52:03.930968738 +0100
@@ -135,18 +135,18 @@
 
 #define MTOOLSFMOPTIONS_1 _("Options:\n\n\
 --nosysconf\n\
-      Don´t attempt to read system-wide\n\
+      Don't attempt to read system-wide\n\
       configuration in ")
 
 #ifdef DEFAULT_NOSAVE
 #define MTOOLSFMOPTIONS_2 _("\n\
 --nouserconf\n\
-      Don´t read user config file in ~/.mtoolsfm\n\
+      Don't read user config file in ~/.mtoolsfm\n\
 --save\n\
       Save configuration in ~/.mtoolsfm when\n\
       terminating\n\
 --nosave\n\
-      Don´t save configuration in ~/.mtoolsfm when\n\
+      Don't save configuration in ~/.mtoolsfm when\n\
       terminating (default)\n\
 --debug or -v\n\
       Output debugging messages to stdout\n\
@@ -155,12 +155,12 @@
 #else
 #define MTOOLSFMOPTIONS_2 _("\n\
 --nouserconf\n\
-      Don´t read user config file in ~/.mtoolsfm\n\
+      Don't read user config file in ~/.mtoolsfm\n\
 --save\n\
       Save configuration in ~/.mtoolsfm when\n\
       terminating (default)\n\
 --nosave\n\
-      Don´t save configuration in ~/.mtoolsfm when\n\
+      Don't save configuration in ~/.mtoolsfm when\n\
       terminating\n\
 --debug or -v\n\
       Output debugging messages to stdout\n\
@@ -189,7 +189,7 @@
 the copy command from the context-menu. You can also print files, make new\n\
 directories and delete files or directories. If you choose \"Text mode for copying\"\n\
 from the \"Options\" menu, files will be copied to / from floppy using appropriate\n\
-CR-LF translation for DOS text files. This is achieved using mcopy´s -t flag.\n\n\
+CR-LF translation for DOS text files. This is achieved using mcopy's -t flag.\n\n\
 Each of the two window halfs has got its own menu. If you have changed the\n\
 floppy in the drive, you can choose \"View -> Refresh\" to refresh the directory\n\
 listing. By clicking onto the column titles in the directory listing, you can\n\
