--- src/netstatus-iface.c.orig	2005-06-12 16:51:52.000000000 -0400
+++ src/netstatus-iface.c	2005-06-12 16:52:25.000000000 -0400
@@ -1218,7 +1218,7 @@
       struct ifreq *if_req = (struct ifreq *) p;
       gboolean      loopback = FALSE;
 
-      p += sizeof (if_req->ifr_name) + NETSTATUS_SA_LEN (&if_req->ifr_addr);
+      p += sizeof (struct ifreq);
 
       if (ioctl (fd, SIOCGIFFLAGS, if_req) < 0)
 	{
