diff -u -r motioneye-1.3/avi.c motioneye-1.3-new/avi.c
--- motioneye-1.3/avi.c	2002-11-16 16:57:01.000000000 +0100
+++ motioneye-1.3-new/avi.c	2010-09-14 22:36:25.000000000 +0200
@@ -81,7 +81,7 @@
 		++fps;
 
 	bzero(&ah, sizeof(ah));
-	strcpy(ah.avih, "avih");
+	strncpy(ah.avih, "avih",4);
 	ah.time = 1e6 / fps;
 	ah.numstreams = 1;
 	ah.scale = 1;
@@ -89,21 +89,21 @@
 	ah.length = fps * nframes;
 
 	bzero(&sh, sizeof(sh));
-	strcpy(sh.strh, "strh");
-	strcpy(sh.vids, "vids");
-	strcpy(sh.codec, "MJPG");
+	strncpy(sh.strh, "strh",4);
+	strncpy(sh.vids, "vids",4);
+	strncpy(sh.codec, "MJPG",4);
 	sh.scale = 1;
 	sh.rate = fps;
 	sh.length = fps * nframes;
 	sh.quality = -1;
 
 	bzero(&fh, sizeof(fh));
-	strcpy(fh.strf, "strf");
+	strncpy(fh.strf, "strf",4);
 	fh.width = width;
 	fh.height = height;
 	fh.planes = 1;
 	fh.bitcount = 24;
-	strcpy(fh.codec,"MJPG");
+	strncpy(fh.codec,"MJPG",4);
 	fh.unpackedsize = 3*width*height;
 
 	rh.size = sizeof(lh1)+sizeof(ah)+sizeof(lh2)+sizeof(sh)+
diff -u -r motioneye-1.3/Makefile motioneye-1.3-new/Makefile
--- motioneye-1.3/Makefile	2003-05-12 17:25:30.000000000 +0200
+++ motioneye-1.3-new/Makefile	2010-09-14 22:31:03.000000000 +0200
@@ -17,7 +17,7 @@
 all: motioneye motioneye.1
 
 motioneye: $(OBJ)
-	$(CC) -o motioneye $(OBJ) $(LIBS)
+	$(CC) -o motioneye $(LDFLAGS) $(OBJ) $(LIBS)
 
 motioneye.o: motioneye.c motioneye.h
 avi.o: avi.c avi.h motioneye.h
