--- port/master.mk.in.orig	1998-11-25 08:26:56.000000000 -0800
+++ port/master.mk.in	2003-11-02 17:47:46.000000000 -0800
@@ -351,7 +351,7 @@
 ############################################################################
 # `etc/' files:
 ############################################################################
-ETCDIR	= $(prefix)/etc
+ETCDIR	= /etc
 ETCFILE	= dummy-etcfile		# to silence GNU make(1)
 
 installed_etcfiles:
@@ -372,7 +372,7 @@
 installed_etcfile:	$(ETCDIR)/$(ETCFILE)
 
 $(ETCDIR)/dummy $(ETCDIR)/$(ETCFILE):	$(ETCDIR) $(ETCFILE)
-	cp $(ETCFILE) $@
+	cp $(ETCFILE) ${D}$@
 
 dummy_etcdir $(ETCDIR):
 	mkdir -p $@
@@ -382,7 +382,7 @@
 ############################################################################
 # Manual Pages:
 ############################################################################
-MANDIR		= $(prefix)/man
+MANDIR		= $(prefix)/share/man
 MANUAL		= dummy-manual	# to slience GNU make
 WHATIS		= @WHATIS@
 # The following macro should be empty on systems that don't
@@ -443,9 +443,6 @@
 		else \
 		    mkdir $(MANDIR)$$subchapter; \
 		fi; \
-		rm -f $(MANDIR)$$subchapter/$(MANUAL); \
-		ln -s ../`basename $(MANDIR)`/$(MANUAL) \
-		    $(MANDIR)$$subchapter/$(MANUAL); \
 		;; \
 	esac
 
