diff -u automake-1.5/tests/ccnoco.test automake-1.6.3/tests/ccnoco.test
--- automake-1.5/tests/ccnoco.test	2003-06-07 06:34:18.000000000 +0900
+++ automake-1.6.3/tests/ccnoco.test	2003-06-07 06:34:18.000000000 +0900
@@ -39,7 +39,6 @@
 }
 END
 
-$needs_autoconf
 gcc -v || exit 77
 
 cat > Mycomp << 'END'
@@ -51,7 +50,7 @@
     ;;
 esac
 
-exec gcc ${1+"$@"}
+exec gcc "$@"
 END
 
 chmod +x Mycomp
@@ -59,6 +58,10 @@
 set -e
 
 
+# Ignore user CFLAGS.
+CFLAGS=
+export CFLAGS
+
 for conf in configure.1 configure.3; do
    cp $conf configure.in
 
