--- Terminal/TerminalParser_Linux.h
+++ Terminal/TerminalParser_Linux.h
@@ -31,7 +31,7 @@
 	int utf_count;
 
 	unsigned char input_buf[16];
-	int input_buf_len;
+	size_t input_buf_len;
 
 #define TITLE_BUF_SIZE 255
 	char title_buf[TITLE_BUF_SIZE+1];
--- Terminal/TerminalParser_Linux.m
+++ Terminal/TerminalParser_Linux.m
@@ -1118,9 +1118,9 @@
 			screen_char_t ch;
 
 			char *inp;
-			int in_size;
+			size_t in_size;
 			char *outp;
-			int out_size;
+			size_t out_size;
 			int char_width;
 
 			int ret;
@@ -1211,10 +1211,10 @@
 	if (iconv_input_state)
 	{
 		unsigned int *inp;
-		int insize;
+		size_t insize;
 		char *outp;
 		char buf[16+1];
-		int outsize;
+		size_t outsize;
 		int ret;
 	
 		for (i=0;i<l;i++)
