diff -Nur esearch.orig/esearch.py esearch/esearch.py
--- esearch.orig/esearch.py	2008-12-28 22:48:35.000000000 -0800
+++ esearch/esearch.py	2008-12-28 22:55:45.000000000 -0800
@@ -13,7 +13,10 @@
 sys.path.insert(0, "/usr/lib/portage/pym")
 sys.path.insert(0, "/usr/lib/esearch")
 
-from output import bold, red, green, darkgreen, turquoise, nocolor
+try:
+    from portage.output import bold, red, green, darkgreen, turquoise, nocolor
+except ImportError:
+    from output import bold, red, green, darkgreen, turquoise, nocolor
 from os.path import exists
 import re
 
diff -Nur esearch.orig/esync.py esearch/esync.py
--- esearch.orig/esync.py	2008-12-28 22:48:35.000000000 -0800
+++ esearch/esync.py	2008-12-28 22:58:02.000000000 -0800
@@ -18,7 +18,10 @@
 sys.path.insert(0, "/usr/lib/portage/pym")
 
 import portage
-from output import red, green, bold, darkgreen, nocolor, xtermTitle
+try:
+    from portage.output import red, green, bold, darkgreen, nocolor, xtermTitle
+except ImportError:
+    from output import red, green, bold, darkgreen, nocolor, xtermTitle
 
 from common import needdbversion
 
diff -Nur esearch.orig/eupdatedb.py esearch/eupdatedb.py
--- esearch.orig/eupdatedb.py	2008-12-28 22:48:35.000000000 -0800
+++ esearch/eupdatedb.py	2008-12-28 22:57:05.000000000 -0800
@@ -20,7 +20,10 @@
 sys.path.insert(0, "/usr/lib/esearch")
 
 import portage
-from output import red, darkgreen, green, bold, nocolor
+try:
+    from portage.output import red, darkgreen, green, bold, nocolor
+except ImportError:
+    from output import red, darkgreen, green, bold, nocolor
 
 from common import needdbversion, version
 
