--- utils/keytable/Makefile
+++ utils/keytable/Makefile
@@ -54,8 +54,8 @@
 install: $(TARGETS)
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
 	install -m 755 $(TARGETS) $(DESTDIR)$(PREFIX)/bin
-	install -m 644 rc_maps.cfg.example /etc
-	install -m 755 -d /etc/rc_keymaps
-	install -m 644 rc_keymaps/* /etc/rc_keymaps
+	install -m 755 -d $(DESTDIR)/etc/rc_keymaps
+	install -m 644 rc_maps.cfg.example $(DESTDIR)/etc/
+	install -m 644 rc_keymaps/* $(DESTDIR)/etc/rc_keymaps
 
 include ../../Make.rules
--- utils/Makefile
+++ utils/Makefile
@@ -2,12 +2,12 @@
 
 all install: 
 	@for i in $(SUBDIRS); do \
-		make -C $$i $@; \
+		$(MAKE) -C $$i $@; \
 	done
 
 	# Test if libsysfs is installed
 	@-if [ -f /usr/include/sysfs/libsysfs.h ]; then \
-		make -C v4l2-sysfs-path $@; \
+		$(MAKE) -C v4l2-sysfs-path $@; \
 	fi
 
 	# Test whether qmake is installed, and whether it is for qt4.
@@ -24,16 +24,16 @@
 	fi
 
 sync-with-kernel:
-	make -C keytable $@
-	make -C v4l2-dbg $@
-	make -C xc3028-firmware $@
+	$(MAKE) -C keytable $@
+	$(MAKE) -C v4l2-dbg $@
+	$(MAKE) -C xc3028-firmware $@
 
 clean::
 	@for i in $(SUBDIRS); do \
-		make -C $$i $@; \
+		$(MAKE) -C $$i $@; \
 	done
-	make -C v4l2-sysfs-path $@
-	if [ -f qv4l2/Makefile ]; then make -C qv4l2 $@; fi
+	$(MAKE) -C v4l2-sysfs-path $@
+	if [ -f qv4l2/Makefile ]; then $(MAKE) -C qv4l2 $@; fi
 	rm -f qv4l2/Makefile qv4l2/qv4l2
 
 include ../Make.rules
