--- a/src/lib/GourmetThreads.py	2008-10-28 22:33:12.000000000 -0400
+++ b/src/lib/GourmetThreads.py	2009-06-25 23:24:02.000000000 -0400
@@ -23,10 +23,9 @@
         self.c = runnerClass
         self.pre_hooks=pre_hooks
         self.post_hooks=post_hooks
-        self.name = name
         self.completed = False
         debug("SuspendableThread starting thread.",2)
-        self.initialize_thread()
+        threading.Thread.__init__(self, target=self.target_func, name=name)
 
     def initialize_thread (self):
         threading.Thread.__init__(self, target=self.target_func, name=name)
