diff -aur apcupsd-3.10.5-orig/platforms/gentoo/apcupsd.in apcupsd-3.10.5/platforms/gentoo/apcupsd.in
--- apcupsd-3.10.5-orig/platforms/gentoo/apcupsd.in	2003-02-26 14:09:57.000000000 -0800
+++ apcupsd-3.10.5/platforms/gentoo/apcupsd.in	2003-02-26 14:14:06.000000000 -0800
@@ -22,3 +22,11 @@
 	start-stop-daemon --stop --quiet --pidfile $APCPID
 	eend $?
 } 
+
+restart() {
+	svc_stop
+	# need to sleep or new process may choke grabbing
+	#  the dying processes network socket
+	sleep 1
+	svc_start
+}
