--- kaffeine/player-parts/http.c.orig	2004-11-03 00:14:18.004567416 +0100
+++ kaffeine/player-parts/http.c	2004-11-03 00:15:08.342914824 +0100
@@ -521,10 +521,10 @@
 	    this->contentlength = contentlength;
 	  }
         }
-        if (sscanf(this->buf, "Content-Type: %s", mime_type) == 1) {
+        /*if (sscanf(this->buf, "Content-Type: %s", mime_type) == 1) {
 	  printf ("http: content type = '%s'\n", mime_type);
 	  strcpy(this->mime_type, mime_type);
-	}	
+	}*/	
 	if (!strncasecmp(this->buf, "Location: ", 10)) {
       	  printf ("http: Location redirection not implemented\n");
 	  free (this);
