--- kaffeine-0.7.1/kaffeine/kaffeineapp.cpp.orig	2005-08-28 16:19:27.000000000 +0100
+++ kaffeine-0.7.1/kaffeine/kaffeineapp.cpp	2006-01-03 15:29:02.436692994 +0000
@@ -95,7 +95,7 @@
   QString device = args->getOption("device");
   if (device == "default")
     device = QString::null;
-  else if (device.startsWith("media:/"))
+  else if (device.startsWith("media:/") || device.startsWith("system:/"))
   {
     KURL devicePath(device);
     DCOPRef mediamanager("kded","mediamanager");
