diff -Naur twin-0.3.8/MakePaths twin-0.3.8-new/MakePaths
--- twin-0.3.8/MakePaths	Mon Feb 26 04:40:45 2001
+++ twin-0.3.8-new/MakePaths	Tue Apr 24 23:03:24 2001
@@ -2,10 +2,11 @@
 # set installation directories
 
 sysconfdir = /etc
-prefix = /usr/local
+prefix = /usr
 bindir = $(prefix)/bin
 libdir = $(prefix)/lib
 datadir = $(prefix)/share
+docdir = $(datadir)/doc
 mandir = $(datadir)/man
 includedir = $(prefix)/include
 
diff -Naur twin-0.3.8/docs/Makefile twin-0.3.8-new/docs/Makefile
--- twin-0.3.8/docs/Makefile	Fri Feb 23 04:32:14 2001
+++ twin-0.3.8-new/docs/Makefile	Tue Apr 24 23:04:21 2001
@@ -42,6 +42,6 @@
 install:
 	$(INSTALL) -d $(DESTDIR)$(mandir)/man1
 	$(INSTALL) twin.1 $(DESTDIR)$(mandir)/man1
-	$(INSTALL) -d $(DESTDIR)$(datadir)/twin/docs
-	$(INSTALL) $(DOCS) $(DESTDIR)$(datadir)/twin/docs
-	cd $(TOPDIR) && $(INSTALL) $(TOPDOCS) $(DESTDIR)$(datadir)/twin
+	$(INSTALL) -d $(DESTDIR)$(docdir)/twin-0.3.8/docs
+	$(INSTALL) $(DOCS) $(DESTDIR)$(docdir)/twin-0.3.8/docs
+	cd $(TOPDIR) && $(INSTALL) $(TOPDOCS) $(DESTDIR)$(docdir)/twin-0.3.8
