diff -ruN quickplot-0.8.2/File.h quickplot-0.8.2-cryos/File.h
--- quickplot-0.8.2/File.h	2004-08-27 00:05:52.000000000 +0100
+++ quickplot-0.8.2-cryos/File.h	2005-01-04 04:44:58.000000000 +0000
@@ -1,8 +1,9 @@
 /* Copyright (c) 1998, 1999, 2003, 2004  Lance Arsenault, (GNU GPL (v2+))
  */
+#ifdef USE_LIBSNDFILE
+#include <sndfile.h>
+#endif
 
-
-struct ::SNDFILE;
 class Source;
 class FileReader;
 class FileList;
@@ -22,7 +23,7 @@
   void init(const FileList *fileList);
   
 #ifdef USE_LIBSNDFILE
-
+  
   // This will not work with stdin.
   bool readSndFile(const FileList *fileList, int fd=-1);
 
diff -ruN quickplot-0.8.2/File_readASCIIFile.cpp quickplot-0.8.2-cryos/File_readASCIIFile.cpp
--- quickplot-0.8.2/File_readASCIIFile.cpp	2004-08-27 00:05:52.000000000 +0100
+++ quickplot-0.8.2-cryos/File_readASCIIFile.cpp	2005-01-04 09:17:29.058633216 +0000
@@ -11,7 +11,7 @@
 
 #include <list>
 #include <iomanip>
-
+#include <errno.h>
 #include <gtkmm.h>
 
 
@@ -40,7 +40,6 @@
 #include "MainWindow.h"
 #include "App.h"
 
-
 #include "errorStr.h"
 #include "FileReader.h"
 
diff -ruN quickplot-0.8.2/ValueSlider.cpp quickplot-0.8.2-cryos/ValueSlider.cpp
--- quickplot-0.8.2/ValueSlider.cpp	2004-08-27 00:05:52.000000000 +0100
+++ quickplot-0.8.2-cryos/ValueSlider.cpp	2005-01-04 09:17:18.083301720 +0000
@@ -6,6 +6,7 @@
 #include <string.h>
 #include <stdio.h>
 #include <math.h>
+#include <errno.h>
 
 #include <gtkmm.h>
 #include <glibmm/ustring.h>
