--- gcc-3.2.1/gcc/system.h.orig	2002-12-23 12:13:00.000000000 +0200
+++ gcc-3.2.1/gcc/system.h	2002-12-23 12:13:29.000000000 +0200
@@ -26,7 +26,7 @@
 /* This is the location of the online document giving information how
    to report bugs. If you change this string, also check for strings
    not under control of the preprocessor.  */
-#define GCCBUGURL "<URL:http://bugzilla.redhat.com/bugzilla/>"
+#define GCCBUGURL "<URL:http://bugs.gentoo.org/>"
 
 /* We must include stdarg.h/varargs.h before stdio.h.  */
 #ifdef ANSI_PROTOTYPES
