--- gcc/config/alpha/linux-elf.h.jj	2001-12-31 12:12:33.000000000 +0100
+++ gcc/config/alpha/linux-elf.h	2002-12-09 13:49:25.000000000 +0100
@@ -39,5 +39,5 @@ Boston, MA 02111-1307, USA.  */
 #ifndef USE_GNULIBC_1
 #undef LIB_SPEC
 #define LIB_SPEC \
-"%{shared:-lc}%{!shared:%{pthread:-lpthread }%{profile:-lc_p}%{!profile:-lc}} "
+"%{pthread:-lpthread }%{shared:-lc}%{!shared:%{profile:-lc_p}%{!profile:-lc}} "
 #endif
--- gcc/config/alpha/linux.h.jj	2002-04-16 13:03:23.000000000 +0200
+++ gcc/config/alpha/linux.h	2002-12-09 13:49:56.000000000 +0100
@@ -34,9 +34,9 @@ SUB_CPP_PREDEFINES
 
 #undef LIB_SPEC
 #define LIB_SPEC \
-  "%{shared: -lc} \
-   %{!shared: %{pthread:-lpthread} \
-              %{profile:-lc_p} %{!profile: -lc}}"
+  "%{pthread:-lpthread} \
+   %{shared:-lc} \
+   %{!shared: %{profile:-lc_p}%{!profile:-lc}}"
 
 /* Show that we need a GP when profiling.  */
 #undef TARGET_PROFILING_NEEDS_GP
--- gcc/config/arm/linux-elf.h.jj	2002-04-16 13:03:38.000000000 +0200
+++ gcc/config/arm/linux-elf.h	2002-12-09 13:50:39.000000000 +0100
@@ -48,9 +48,9 @@ Boston, MA 02111-1307, USA.  */
 
 /* Now we define the strings used to build the spec file.  */
 #define LIB_SPEC \
-  "%{shared: -lc} \
-   %{!shared: %{pthread:-lpthread} \
-   %{profile:-lc_p} %{!profile: -lc}}"
+  "%{pthread:-lpthread} \
+   %{shared:-lc} \
+   %{!shared:%{profile:-lc_p}%{!profile:-lc}}"
 
 #define LIBGCC_SPEC "%{msoft-float:-lfloat} -lgcc"
 
--- gcc/config/pa/pa-linux.h.jj	2002-06-03 17:24:33.000000000 +0200
+++ gcc/config/pa/pa-linux.h	2002-12-09 13:54:55.000000000 +0100
@@ -33,9 +33,9 @@ Boston, MA 02111-1307, USA.  */
 
 #undef	LIB_SPEC
 #define LIB_SPEC \
-  "%{shared: -lgcc -lc} \
-   %{!shared: %{mieee-fp:-lieee} %{pthread:-lpthread} \
-	%{shared-libgcc: -lgcc} %{profile:-lc_p} %{!profile: -lc}}"
+  "%{pthread:-lpthread} \
+   %{shared:-lgcc -lc} \
+   %{!shared:%{mieee-fp:-lieee} %{shared-libgcc:-lgcc} %{profile:-lc_p}%{!profile:-lc}}"
 
 #undef ASM_SPEC
 #define ASM_SPEC \
--- gcc/config/sparc/linux.h.jj	2002-04-29 16:22:40.000000000 +0200
+++ gcc/config/sparc/linux.h	2002-12-09 13:56:39.000000000 +0100
@@ -123,9 +123,9 @@ Boston, MA 02111-1307, USA.  */
      %{!profile:%{!ggdb:-lc} %{ggdb:-lg}}}"
 #else
 #define LIB_SPEC \
-  "%{shared: -lc} \
-   %{!shared: %{mieee-fp:-lieee} %{pthread:-lpthread} \
-     %{profile:-lc_p} %{!profile: -lc}}"
+  "%{pthread:-lpthread} \
+   %{shared:-lc} \
+   %{!shared:%{mieee-fp:-lieee} %{profile:-lc_p}%{!profile:-lc}}"
 #endif
 #else
 #define LIB_SPEC \
--- gcc/config/sparc/linux64.h.jj	2002-10-08 15:29:41.000000000 +0200
+++ gcc/config/sparc/linux64.h	2002-12-09 13:56:25.000000000 +0100
@@ -126,9 +126,9 @@ Boston, MA 02111-1307, USA.  */
 
 #undef LIB_SPEC
 #define LIB_SPEC \
-  "%{shared: -lc} \
-   %{!shared: %{mieee-fp:-lieee} %{pthread:-lpthread} \
-     %{profile:-lc_p} %{!profile: -lc}}"
+  "%{pthread:-lpthread} \
+   %{shared:-lc} \
+   %{!shared: %{mieee-fp:-lieee} %{profile:-lc_p}%{!profile:-lc}}"
 
 /* Provide a LINK_SPEC appropriate for GNU/Linux.  Here we provide support
    for the special GCC options -static and -shared, which allow us to
--- gcc/config/linux.h.jj	2001-12-18 01:30:48.000000000 +0100
+++ gcc/config/linux.h	2002-12-09 13:48:20.000000000 +0100
@@ -93,9 +93,9 @@ Boston, MA 02111-1307, USA.  */
      %{!profile:%{!ggdb:-lc} %{ggdb:-lg}}}"
 #else
 #define LIB_SPEC \
-  "%{shared: -lc} \
-   %{!shared: %{mieee-fp:-lieee} %{pthread:-lpthread} \
-	%{profile:-lc_p} %{!profile: -lc}}"
+  "%{pthread:-lpthread} \
+   %{shared:-lc} \
+   %{!shared:%{mieee-fp:-lieee} %{profile:-lc_p}%{!profile:-lc}}"
 #endif
 #else
 #define LIB_SPEC \
