--- gcc-3.2/libjava/include/dwarf2-signal.h.libjava-x86_64-biarch-sigaction	2002-09-20 12:08:46.000000000 -0400
+++ gcc-3.2/libjava/include/dwarf2-signal.h	2002-09-20 13:30:07.000000000 -0400
@@ -209,7 +209,7 @@ do								\
 while (0)  
 #endif
 #elif !defined(__ia64__)
-#if defined(__x86_64__)
+#if defined(__x86_64__) || (defined(__i386__) && !defined(SYS_sigaction))
 #define SYS_sigaction SYS_rt_sigaction
 #endif
 #define INIT_SEGV						\
