--- gcc/testsuite/g++.dg/template/varmod1.C.jj	2002-12-08 21:43:53.000000000 +0100
+++ gcc/testsuite/g++.dg/template/varmod1.C	2002-12-09 14:01:33.000000000 +0100
@@ -6,5 +6,5 @@ void bar()
 {
   int i;
   int A[i][i]; 
-  foo(A); // { dg-error } 
+  foo(A); // { dg-error "" }
 }
--- gcc/testsuite/g++.dg/parse/parameter-declaration-1.C.jj	2002-12-08 21:43:51.000000000 +0100
+++ gcc/testsuite/g++.dg/parse/parameter-declaration-1.C	2002-12-09 14:04:07.000000000 +0100
@@ -3,4 +3,4 @@
 // { dg-do compile }
 
 struct {
-   a(void = 0; a(0), a(0)                 // { dg-error "" }
+   a(void = 0; a(0), a(0)	// { dg-error "" "" { target *-*-* } 7 }
