--- pl-5.6.4.orig/packages/Makefile.in	2005-11-13 03:15:23.000000000 +1300
+++ pl-5.6.4/packages/Makefile.in	2006-02-13 23:09:00.000000000 +1300
@@ -76,7 +76,7 @@
 
 install::
 	for p in $(PKG); do \
-	    if [ -r $$p/Makefile ]; then ( cd $$p && $(MAKE) $@ ); fi; \
+	    if [ -r $$p/Makefile ]; then $(MAKE) -C $$p $@ || exit 1; fi; \
         done
 	@echo "*** Forcing update of library index ***"
 	rm -f $(DESTDIR)$(PLLIB)/INDEX.pl
@@ -110,7 +110,7 @@
 	        echo "**********************"; \
 		echo "* Package $$p"; \
 		echo "**********************"; \
-	        ( cd $$p && $(CONFIG) ); \
+	        ( cd $$p && $(CONFIG) ) || exit 1; \
 	    fi; \
         done
 
@@ -127,7 +127,7 @@
 
 check::
 	for p in $(PKG); do \
-	    if [ -r $$p/Makefile ]; then ( cd $$p && $(MAKE) $@ ); fi; \
+	    if [ -r $$p/Makefile ]; then $(MAKE) -C $$p $@ || exit 1; fi; \
         done
 
 # test environment
--- pl-5.6.17.orig/packages/db/Makefile.in	2005-12-15 04:54:38.000000000 +1300
+++ pl-5.6.17/packages/db/Makefile.in	2006-08-27 10:42:25.000000000 +1200
@@ -61,6 +61,8 @@
 		(cd $(PLBASE)/library && rm -f $(LIBPL))
 		$(PL) -f none -g make -t halt
 
+check::
+
 ################################################################
 # Documentation
 ################################################################
