--- pl-5.6.17.orig/src/configure	2006-07-11 20:31:16.000000000 +1200
+++ pl-5.6.17/src/configure	2006-08-26 17:49:55.000000000 +1200
@@ -3807,7 +3807,7 @@
 		esac
 	fi
 else
-	COFLAGS=$CFLAGS
+	COFLAGS="$CFLAGS -fno-strict-aliasing"
 fi
 
 
@@ -14213,7 +14213,7 @@
   LNLIBS="$LIBS"
 fi
 
-echo "	CFLAGS=$CWFLAGS $COFLAGS $CIFLAGS $CMFLAGS"
+echo "	CFLAGS=$CWFLAGS $COFLAGS $CMFLAGS"
 echo "	LDFLAGS=$LDFLAGS"
 echo "	LIBS=$LIBS"
 
--- pl-5.6.17.orig/src/Makefile.in	2006-07-15 01:51:53.000000000 +1200
+++ pl-5.6.17/src/Makefile.in	2006-08-26 17:47:12.000000000 +1200
@@ -69,7 +69,7 @@
 CWFLAGS=@CWFLAGS@
 CIFLAGS=@CIFLAGS@
 CMFLAGS=@CMFLAGS@
-CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CIFLAGS) $(CMFLAGS)
+CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CMFLAGS)
 LDFLAGS=@LDFLAGS@
 
 PLFOREIGN=@PLFOREIGN@
--- pl-5.6.17.orig/src/rc/Makefile.in	2006-07-11 20:22:28.000000000 +1200
+++ pl-5.6.17/src/rc/Makefile.in	2006-08-26 17:47:52.000000000 +1200
@@ -30,7 +30,7 @@
 CIFLAGS=@CIFLAGS@
 CMFLAGS=@CMFLAGS@
 LDFLAGS=@LDFLAGS@
-CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CIFLAGS) $(CMFLAGS)
+CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CMFLAGS)
 
 INSTALL=@INSTALL@
 INSTALL_PROGRAM=@INSTALL_PROGRAM@
--- pl-5.6.17.orig/src/libtai/Makefile.in	2006-07-11 20:41:05.000000000 +1200
+++ pl-5.6.17/src/libtai/Makefile.in	2006-08-26 17:52:02.000000000 +1200
@@ -29,7 +29,7 @@
 CIFLAGS=@CIFLAGS@
 CMFLAGS=@CMFLAGS@
 LDFLAGS=@LDFLAGS@
-CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CIFLAGS) $(CMFLAGS)
+CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CMFLAGS)
 
 INSTALL=@INSTALL@
 INSTALL_PROGRAM=@INSTALL_PROGRAM@
--- pl-5.6.17.orig/packages/clib/configure	2006-01-11 09:41:55.000000000 +1300
+++ pl-5.6.17/packages/clib/configure	2006-08-26 16:23:26.000000000 +1200
@@ -3117,7 +3117,6 @@
 
 fi
 
-CFLAGS="$CMFLAGS"
 # The Ultrix 4.2 mips builtin alloca declared by alloca.h only works
 # for constant arguments.  Useless!
 echo "$as_me:$LINENO: checking for working alloca.h" >&5
@@ -3709,7 +3708,7 @@
 
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.17.orig/packages/clib/maildrop/rfc2045/configure	2005-07-25 02:37:57.000000000 +1200
+++ pl-5.6.17/packages/clib/maildrop/rfc2045/configure	2006-08-26 16:31:05.000000000 +1200
@@ -3155,7 +3155,7 @@
 
 if test "$GCC" = yes ; then
         CXXFLAGS="$CXXFLAGS -Wall"
-        CFLAGS="$CFLAGS -Wall"
+        CFLAGS="$CFLAGS -fpic -Wall"
 fi
 
 
pl-5.6.17/packages/clib/maildrop/rfc822/configure
--- pl-5.6.17.orig/packages/clib/maildrop/rfc822/configure	2005-07-25 02:37:57.000000000 +1200
+++ pl-5.6.17/packages/clib/maildrop/rfc822/configure	2006-08-26 16:32:11.000000000 +1200
@@ -3144,7 +3144,7 @@
 
 if test "$GCC" = yes ; then
         CXXFLAGS="$CXXFLAGS -Wall"
-        CFLAGS="$CFLAGS -Wall"
+        CFLAGS="$CFLAGS -fpic -Wall"
 fi
 
 
--- pl-5.6.17.orig/packages/cppproxy/configure	2006-01-23 22:29:29.000000000 +1300
+++ pl-5.6.17/packages/cppproxy/configure	2006-08-26 16:40:49.000000000 +1200
@@ -3111,7 +3111,6 @@
 
 fi
 
-CFLAGS="$CMFLAGS"
 # The Ultrix 4.2 mips builtin alloca declared by alloca.h only works
 # for constant arguments.  Useless!
 echo "$as_me:$LINENO: checking for working alloca.h" >&5
@@ -3703,7 +3702,7 @@
 
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.17.orig/packages/db/configure.in	2006-07-07 23:29:11.000000000 +1200
+++ pl-5.6.17/packages/db/configure.in	2006-08-26 18:35:22.000000000 +1200
@@ -93,13 +93,13 @@
 AC_ISC_POSIX
 AC_HEADER_STDC
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
     CWFLAGS=
 fi
-CFLAGS="$CMFLAGS $CIFLAGS"
+
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
--- pl-5.6.17.orig/packages/jpl/configure.in	2005-01-31 23:47:37.000000000 +1300
+++ pl-5.6.17/packages/jpl/configure.in	2006-08-26 18:50:11.000000000 +1200
@@ -70,10 +70,10 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
+
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS="-Wall -Wno-strict-aliasing"
 else
     COFLAGS=-O
--- pl-5.6.17.orig/packages/ltx2htm/configure	2004-09-22 02:28:17.000000000 +1200
+++ pl-5.6.17/packages/ltx2htm/configure	2006-08-26 15:58:58.000000000 +1200
@@ -1351,6 +1351,7 @@
 CC=$PLLD
 LD=$PLLD
 LDSOFLAGS=-shared
+CMFLAGS=-fpic
 
 SO="$PLSOEXT"
 
@@ -2925,7 +2926,6 @@
 
 fi
 
-CFLAGS="$CMFLAGS"
 # The Ultrix 4.2 mips builtin alloca declared by alloca.h only works
 # for constant arguments.  Useless!
 echo "$as_me:$LINENO: checking for working alloca.h" >&5
@@ -3455,7 +3455,7 @@
 
 
 if test ! -z "$GCC"; then
-    COFLAGS=-O2
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.17.orig/packages/nlp/configure	2005-11-26 23:54:48.000000000 +1300
+++ pl-5.6.17/packages/nlp/configure	2006-08-26 17:33:02.000000000 +1200
@@ -3111,7 +3111,6 @@
 
 fi
 
-CFLAGS="$CMFLAGS"
 # The Ultrix 4.2 mips builtin alloca declared by alloca.h only works
 # for constant arguments.  Useless!
 echo "$as_me:$LINENO: checking for working alloca.h" >&5
@@ -3703,7 +3702,7 @@
 
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.17.orig/packages/odbc/configure	2006-02-16 09:48:34.000000000 +1300
+++ pl-5.6.17/packages/odbc/configure	2006-08-26 17:34:38.000000000 +1200
@@ -3147,7 +3147,6 @@
 
 fi
 
-CFLAGS="$CMFLAGS"
 # The Ultrix 4.2 mips builtin alloca declared by alloca.h only works
 # for constant arguments.  Useless!
 echo "$as_me:$LINENO: checking for working alloca.h" >&5
@@ -3739,7 +3738,7 @@
 
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.17.orig/packages/semweb/configure	2006-03-08 03:43:04.000000000 +1300
+++ pl-5.6.17/packages/semweb/configure	2006-08-26 17:35:48.000000000 +1200
@@ -3145,7 +3145,6 @@
 
 fi
 
-CFLAGS="$CMFLAGS"
 # The Ultrix 4.2 mips builtin alloca declared by alloca.h only works
 # for constant arguments.  Useless!
 echo "$as_me:$LINENO: checking for working alloca.h" >&5
@@ -4294,7 +4293,7 @@
 
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O3 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.17.orig/packages/sgml/configure	2006-02-04 04:38:50.000000000 +1300
+++ pl-5.6.17/packages/sgml/configure	2006-08-26 17:37:53.000000000 +1200
@@ -3184,16 +3184,12 @@
 fi
 
 if test ! -z "$GCC"; then
-    if test $gmake = yes; then
-        COFLAGS='$(shell if test -r .coflags; then cat .coflags; else echo "-O2"; fi)'
-    else
-	COFLAGS="-O2 -fno-strict-aliasing"
-    fi
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS="$CWFLAGS -Wall"
 else
     COFLAGS=-O
 fi
-CFLAGS="$CMFLAGS"
+
 echo "$as_me:$LINENO: checking whether byte ordering is bigendian" >&5
 echo $ECHO_N "checking whether byte ordering is bigendian... $ECHO_C" >&6
 if test "${ac_cv_c_bigendian+set}" = set; then
--- pl-5.6.17.orig/packages/ssl/configure	2006-02-04 04:38:50.000000000 +1300
+++ pl-5.6.17/packages/ssl/configure	2006-08-26 17:40:02.000000000 +1200
@@ -3147,7 +3147,6 @@
 
 fi
 
-CFLAGS="$CMFLAGS"
 # The Ultrix 4.2 mips builtin alloca declared by alloca.h only works
 # for constant arguments.  Useless!
 echo "$as_me:$LINENO: checking for working alloca.h" >&5
@@ -3739,7 +3738,7 @@
 
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.17.orig/packages/table/configure	2005-01-31 23:47:37.000000000 +1300
+++ pl-5.6.17/packages/table/configure	2006-08-26 17:41:22.000000000 +1200
@@ -3112,7 +3112,6 @@
 
 fi
 
-CFLAGS="$CMFLAGS"
 # The Ultrix 4.2 mips builtin alloca declared by alloca.h only works
 # for constant arguments.  Useless!
 echo "$as_me:$LINENO: checking for working alloca.h" >&5
@@ -3704,7 +3703,7 @@
 
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.17.orig/packages/xpce/src/configure	2006-04-05 19:29:14.000000000 +1200
+++ pl-5.6.17/packages/xpce/src/configure	2006-08-26 17:43:18.000000000 +1200
@@ -4391,7 +4391,7 @@
 fi
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS -fno-strict-aliasing"
     if test "$ac_cv_header_stdc" = "yes" || test "$syslibh" = "yes"; then
         CWFLAGS=-Wall
     else
@@ -4401,7 +4401,7 @@
     COFLAGS=-O
     CWFLAGS=
 fi
-CFLAGS="$CMFLAGS"
+
 echo "$as_me:$LINENO: checking for inline" >&5
 echo $ECHO_N "checking for inline... $ECHO_C" >&6
 if test "${ac_cv_c_inline+set}" = set; then
