--- pl-5.6.18.orig/packages/Makefile.in	2006-08-30 20:35:12.000000000 +1200
+++ pl-5.6.18/packages/Makefile.in	2006-09-01 20:31:39.000000000 +1200
@@ -10,7 +10,7 @@
 # libXPM and libjpeg to compile.  This   variable  is normally overruled
 # from the configure script.  MUST be on one line
 
-PKG=	clib cpp odbc table xpce sgml sgml/RDF semweb http chr clpqr nlp ssl pldoc plunit jpl
+PKG=	clib cpp cppproxy odbc table xpce sgml sgml/RDF semweb http chr clpqr nlp ssl pldoc plunit jpl
 
 # Define additional config options.  If you wish to build the multi-threaded
 # version of SWI-Prolog and pl-mt to the same install-base, you must provide
