--- pl-5.6.4.orig/src/configure	2006-02-04 04:38:51.000000000 +1300
+++ pl-5.6.4/src/configure	2006-02-15 21:02:38.000000000 +1300
@@ -14094,7 +14095,7 @@
   LNLIBS="$LIBS"
 fi
 
-echo "	CFLAGS=$CWFLAGS $COFLAGS $CIFLAGS $CMFLAGS"
+echo "	CFLAGS=$CWFLAGS $COFLAGS $CMFLAGS"
 echo "	LDFLAGS=$LDFLAGS"
 echo "	LIBS=$LIBS"
 
--- pl-5.6.4.orig/src/Makefile.in	2006-02-04 04:38:51.000000000 +1300
+++ pl-5.6.4/src/Makefile.in	2006-02-15 20:08:21.000000000 +1300
@@ -69,7 +69,7 @@
 CWFLAGS=@CWFLAGS@
 CIFLAGS=@CIFLAGS@
 CMFLAGS=@CMFLAGS@
-CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CIFLAGS) $(CMFLAGS)
+CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CMFLAGS)
 LDFLAGS=@LDFLAGS@
 
 PLFOREIGN=@PLFOREIGN@
--- pl-5.6.4.orig/src/rc/Makefile.in	2005-09-24 00:02:05.000000000 +1200
+++ pl-5.6.4/src/rc/Makefile.in	2006-02-15 20:16:46.000000000 +1300
@@ -34,7 +34,7 @@
 CIFLAGS=@CIFLAGS@
 CMFLAGS=@CMFLAGS@
 LDFLAGS=@LDFLAGS@
-CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CIFLAGS) $(CMFLAGS)
+CFLAGS=	$(CWFLAGS) $(COFLAGS) $(CMFLAGS)
 
 INSTALL=@INSTALL@
 INSTALL_PROGRAM=@INSTALL_PROGRAM@
--- pl-5.6.4.orig/packages/clib/configure.in	2006-01-11 09:41:55.000000000 +1300
+++ pl-5.6.4/packages/clib/configure.in	2006-02-13 22:35:58.000000000 +1300
@@ -72,12 +72,11 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.4.orig/packages/clib/maildrop/rfc2045/Makefile.in	2005-07-25 02:37:57.000000000 +1200
+++ pl-5.6.4/packages/clib/maildrop/rfc2045/Makefile.in	2006-02-13 22:38:05.000000000 +1300
@@ -5,7 +5,7 @@
 ################################################################
 
 CC=@CC@
-CFLAGS=@CFLAGS@ @DEFS@
+CFLAGS=@CFLAGS@ -fPIC @DEFS@
 LIBS=@LIBS@
 AR=@AR@
 RANLIB=@RANLIB@
--- pl-5.6.4.orig/packages/clib/maildrop/rfc822/Makefile.in	2006-02-04 04:38:49.000000000 +1300
+++ pl-5.6.4/packages/clib/maildrop/rfc822/Makefile.in	2006-02-13 22:39:03.000000000 +1300
@@ -5,7 +5,7 @@
 ################################################################
 
 CC=@CC@
-CFLAGS=@CFLAGS@ @DEFS@
+CFLAGS=@CFLAGS@ -fPIC @DEFS@
 LIBS=@LIBS@
 AR=@AR@
 RANLIB=@RANLIB@
--- pl-5.6.4.orig/packages/cppproxy/configure.in	2006-01-23 02:56:38.000000000 +1300
+++ pl-5.6.4/packages/cppproxy/configure.in	2006-02-15 21:28:45.000000000 +1300
@@ -71,12 +71,11 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.4.orig/packages/db/configure.in	2005-12-15 05:12:24.000000000 +1300
+++ pl-5.6.4/packages/db/configure.in	2006-02-13 22:39:53.000000000 +1300
@@ -92,13 +92,12 @@
 AC_ISC_POSIX
 AC_HEADER_STDC
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
     CWFLAGS=
 fi
-CFLAGS="$CMFLAGS $CIFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
--- pl-5.6.4.orig/packages/db/Makefile.in	2005-12-15 04:54:38.000000000 +1300
+++ pl-5.6.4/packages/db/Makefile.in	2006-02-13 22:40:29.000000000 +1300
@@ -16,7 +16,7 @@
 COFLAGS=@COFLAGS@
 CWFLAGS=@CWFLAGS@
 CMFLAGS=@CMFLAGS@
-CIFLAGS=-I. -I$(PLBASE)/include @CIFLAGS@
+CIFLAGS=-I. -I../../include @CIFLAGS@
 CFLAGS=$(COFLAGS) $(CWFLAGS) $(CMFLAGS) $(CIFLAGS) @DEFS@
 LD_RUN_PATH=@LD_RUN_PATH@
 SO=@SO@
--- pl-5.6.4.orig/packages/jpl/configure.in	2005-01-31 23:47:37.000000000 +1300
+++ pl-5.6.4/packages/jpl/configure.in	2006-02-13 22:41:09.000000000 +1300
@@ -70,10 +70,9 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS="-Wall -Wno-strict-aliasing"
 else
     COFLAGS=-O
--- pl-5.6.0.orig/packages/ltx2htm/Makefile.in	2005-12-02 21:53:05.000000000 +1300
+++ pl-5.6.0/packages/ltx2htm/Makefile.in	2006-01-07 09:55:01.000000000 +1300
@@ -27,7 +29,7 @@
 LDFLAGS=@LDSOFLAGS@
 
 CDFLAGS=-D__SWI_PROLOG__
-CFLAGS=$(COFLAGS) $(CWFLAGS) $(CMFLAGS) $(CIFLAGS) @DEFS@ $(CDFLAGS)
+CFLAGS=$(COFLAGS) $(CWFLAGS) $(CMFLAGS) -fPIC $(CIFLAGS) @DEFS@ $(CDFLAGS)
 INSTALL=@INSTALL@
 INSTALL_PROGRAM=@INSTALL_PROGRAM@
 INSTALL_DATA=@INSTALL_DATA@
--- pl-5.6.4.orig/packages/ltx2htm/configure.in	2004-09-22 02:28:17.000000000 +1200
+++ pl-5.6.4/packages/ltx2htm/configure.in	2006-02-13 22:41:48.000000000 +1300
@@ -56,12 +56,11 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
 if test ! -z "$GCC"; then
-    COFLAGS=-O2
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.4.orig/packages/mp/configure.in	2004-09-22 02:28:17.000000000 +1200
+++ pl-5.6.4/packages/mp/configure.in	2006-02-13 22:42:32.000000000 +1300
@@ -85,13 +85,12 @@
 AC_ISC_POSIX
 AC_HEADER_STDC
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
     CWFLAGS=
 fi
-CFLAGS="$CIFLAGS $CMFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
--- pl-5.6.4.orig/packages/mp/Makefile.in	2006-02-04 04:38:50.000000000 +1300
+++ pl-5.6.4/packages/mp/Makefile.in	2006-02-13 22:42:56.000000000 +1300
@@ -12,7 +12,7 @@
 COFLAGS=@COFLAGS@
 CWFLAGS=@CWFLAGS@
 CMFLAGS=@CMFLAGS@
-CIFLAGS=@CIFLAGS@ -I. -I$(PLBASE)/include
+CIFLAGS=@CIFLAGS@ -I. -I../../include
 CFLAGS=$(COFLAGS) $(CWFLAGS) $(CMFLAGS) $(CIFLAGS) @DEFS@
 LOCALLIB=@LOCALLIB@
 LIBS=@LIBS@
--- pl-5.6.4.orig/packages/nlp/configure.in	2005-11-26 23:54:48.000000000 +1300
+++ pl-5.6.4/packages/nlp/configure.in	2006-02-13 22:43:37.000000000 +1300
@@ -71,12 +71,11 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.4.orig/packages/odbc/configure.in	2006-02-04 04:38:50.000000000 +1300
+++ pl-5.6.4/packages/odbc/configure.in	2006-02-13 22:44:24.000000000 +1300
@@ -67,12 +67,11 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.4.orig/packages/semweb/configure.in	2006-02-07 04:59:54.000000000 +1300
+++ pl-5.6.4/packages/semweb/configure.in	2006-02-13 22:44:55.000000000 +1300
@@ -65,14 +65,13 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 AC_C_INLINE
 AC_CHECK_SIZEOF(long)
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O3 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.4.orig/packages/sgml/configure.in	2006-02-04 04:38:50.000000000 +1300
+++ pl-5.6.4/packages/sgml/configure.in	2006-02-13 22:46:04.000000000 +1300
@@ -96,16 +96,11 @@
 AC_ISC_POSIX
 AC_HEADER_STDC
 if test ! -z "$GCC"; then
-    if test $gmake = yes; then
-        COFLAGS='$(shell if test -r .coflags; then cat .coflags; else echo "-O2"; fi)'
-    else
-	COFLAGS="-O2 -fno-strict-aliasing"
-    fi
+    COFLAGS="$CFLAGS"
     CWFLAGS="$CWFLAGS -Wall"
 else
     COFLAGS=-O
 fi
-CFLAGS="$CMFLAGS"
 AC_C_BIGENDIAN
 AC_C_INLINE
 AC_CHECK_SIZEOF(long, 4)
--- pl-5.6.4.orig/packages/ssl/configure.in	2006-02-04 04:38:50.000000000 +1300
+++ pl-5.6.4/packages/ssl/configure.in	2006-02-13 22:46:41.000000000 +1300
@@ -67,12 +67,11 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.4.orig/packages/table/configure.in	2005-01-31 23:47:37.000000000 +1300
+++ pl-5.6.4/packages/table/configure.in	2006-02-13 22:47:06.000000000 +1300
@@ -69,12 +69,11 @@
 AC_PROG_INSTALL
 AC_ISC_POSIX
 AC_HEADER_STDC
-CFLAGS="$CMFLAGS"
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
 
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     CWFLAGS=-Wall
 else
     COFLAGS=-O
--- pl-5.6.4.orig/packages/xpce/src/configure.in	2006-02-04 04:38:51.000000000 +1300
+++ pl-5.6.4/packages/xpce/src/configure.in	2006-02-13 22:47:59.000000000 +1300
@@ -539,7 +539,7 @@
 AC_ISC_POSIX
 AC_HEADER_STDC
 if test ! -z "$GCC"; then
-    COFLAGS="-O2 -fno-strict-aliasing"
+    COFLAGS="$CFLAGS"
     if test "$ac_cv_header_stdc" = "yes" || test "$syslibh" = "yes"; then
         CWFLAGS=-Wall
     else
@@ -549,7 +549,6 @@
     COFLAGS=-O
     CWFLAGS=
 fi
-CFLAGS="$CMFLAGS"
 AC_C_INLINE
 AC_FUNC_ALLOCA
 AC_C_BIGENDIAN
