--- pl-5.6.17.orig/packages/jpl/Makefile.in	2006-02-04 04:38:49.000000000 +1300
+++ pl-5.6.17/packages/jpl/Makefile.in	2006-08-26 14:24:40.000000000 +1200
@@ -34,6 +34,7 @@
 JAVA_HOME=@JAVA_HOME@
 JAVAC=@JAVAC@
 JAVALIBS=@JAVALIBS@
+JAVALDPATH=@JAVALDPATH@
 
 DOCTOTEX=$(PCEHOME)/bin/doc2tex
 PLTOTEX=$(PCEHOME)/bin/pl2tex
@@ -68,7 +69,7 @@
 ################################################################
 
 check::		TestJPL.class
-		$(PL) -q -f check.pl -g test,halt -t 'halt(1)'
+		LD_LIBRARY_PATH=$(JAVALDPATH) $(PL) -q -f check.pl -g test,halt -t 'halt(1)'
 
 TestJPL.class:	TestJPL.java
 		$(JAVAC) -classpath jpl.jar TestJPL.java
pl-5.6.17/packages/jpl/ac/ac_jni_include_dirs.m4
--- pl-5.6.17.orig/packages/jpl/ac/ac_jni_include_dirs.m4	2006-02-04 04:38:49.000000000 +1300
+++ pl-5.6.17/packages/jpl/ac/ac_jni_include_dirs.m4	2006-08-26 10:16:50.000000000 +1200
@@ -54,19 +54,32 @@
 done
 
 case "$host_cpu" in
-	i?86)	_JNI_LIBDIRS="i386"
+	i?86)
+		_JNI_LIBDIRS="lib/i386"
 		_JNI_LIBSUBDIR="client"
 		;;
-	x86_64)	_JNI_LIBDIRS="amd64"
+	x86_64)
+		_JNI_LIBDIRS="lib/amd64"
 		_JNI_LIBSUBDIR="server"
 		;;
-	*)	_JNI_LIBDIRS=""
+	powerpc)
+		case "$host_os" in
+		linux*)
+			_JNI_LIBDIRS="bin"
+			_JNI_LIBSUBDIR="classic"
+			;;
+		*)
+			_JNI_LIBDIRS=""
+		esac
+		;;
+	*)	
+		_JNI_LIBDIRS=""
 esac
 
 for d in $_JNI_LIBDIRS; do
-	echo "Trying $_JTOPDIR/jre/lib/$d/$_JNI_LIBSUBDIR"
-	if test -d $_JTOPDIR/jre/lib/$d/$_JNI_LIBSUBDIR; then
-		JNI_CLIENT_DIRS="$JNI_CLIENT_DIRS $_JTOPDIR/jre/lib/$d/$_JNI_LIBSUBDIR $_JTOPDIR/jre/lib/$d"
+	echo "Trying $_JTOPDIR/jre/$d/$_JNI_LIBSUBDIR"
+	if test -d $_JTOPDIR/jre/$d/$_JNI_LIBSUBDIR; then
+		JNI_CLIENT_DIRS="$JNI_CLIENT_DIRS $_JTOPDIR/jre/$d/$_JNI_LIBSUBDIR $_JTOPDIR/jre/$d"
  	fi
 done
 
pl-5.6.17/packages/jpl/configure.in
--- pl-5.6.17.orig/packages/jpl/configure.in	2005-01-31 23:47:37.000000000 +1300
+++ pl-5.6.17/packages/jpl/configure.in	2006-08-26 18:50:11.000000000 +1200
@@ -18,6 +18,7 @@
 AC_SUBST(LDSOFLAGS)
 AC_SUBST(JAVA_HOME)
 AC_SUBST(JAVALIBS)
+AC_SUBST(JAVALDPATH)
 
 AC_ARG_ENABLE(mt, [  --enable-mt		  Enable Multi-threading],
 	      [case "$enableval" in
@@ -84,6 +85,9 @@
    *darwin*)
 	JAVALIBS="-Wl,-framework,JavaVM"
 	;;
+   *powerpc-linux*)
+	JAVALIBS="-ljava -ljvm"
+	;;
    *)
 	JAVALIBS="-ljava -lverify -ljvm"
 	;;
@@ -128,6 +132,7 @@
 
 for d in $JNI_CLIENT_DIRS; do
   LDSOFLAGS="$LDSOFLAGS -L$d"
+  JAVALDPATH="$JAVALDPATH:$d"
 done
 
 if test "$java_abs_paths" = yes; then
