diff -ur pl-5.6.24.orig/packages/jpl/Makefile.in pl-5.6.24/packages/jpl/Makefile.in
--- pl-5.6.24.orig/packages/jpl/Makefile.in	2006-11-28 22:16:43.000000000 +1300
+++ pl-5.6.24/packages/jpl/Makefile.in	2006-11-29 20:51:24.000000000 +1300
@@ -36,6 +36,7 @@
 JAVA=@JAVA@
 JUNIT=@JUNIT@
 JAVALIBS=@JAVALIBS@
+JAVALDPATH=@JAVALDPATH@
 
 DOCTOTEX=$(PCEHOME)/bin/doc2tex
 PLTOTEX=$(PCEHOME)/bin/pl2tex
@@ -75,7 +76,7 @@
 check:		check_pl check_java
 
 check_pl::		
-		$(PL) -q -f test_jpl.pl -g run_tests,halt -t 'halt(1)'
+		LD_LIBRARY_PATH=$(JAVALDPATH) $(PL) -q -f test_jpl.pl -g run_tests,halt -t 'halt(1)'
 check_java::
 		CLASSPATH=$(JUNIT):jpl.jar:jpltest.jar $(JAVA) junit.textui.TestRunner jpl.test.TestJUnit
 
diff -ur pl-5.6.24.orig/packages/jpl/configure pl-5.6.24/packages/jpl/configure
--- pl-5.6.24.orig/packages/jpl/configure	2006-11-28 22:16:44.000000000 +1300
+++ pl-5.6.24/packages/jpl/configure	2006-11-29 20:50:13.000000000 +1300
@@ -3979,6 +3979,7 @@
 
 for d in $JNI_CLIENT_DIRS; do
   LDSOFLAGS="$LDSOFLAGS -L$d"
+  JAVALDPATH="$JAVALDPATH:$d"
 done
 
 if test "$java_abs_paths" = yes; then
@@ -5254,6 +5255,7 @@
 s,@LDSOFLAGS@,$LDSOFLAGS,;t t
 s,@JAVA_HOME@,$JAVA_HOME,;t t
 s,@JAVALIBS@,$JAVALIBS,;t t
+s,@JAVALDPATH@,$JAVALDPATH,;t t
 s,@JUNIT@,$JUNIT,;t t
 s,@PLLD@,$PLLD,;t t
 s,@MAKE@,$MAKE,;t t
