--- pl-5.6.17.orig/packages/ssl/Makefile.in	2006-02-04 04:38:50.000000000 +1300
+++ pl-5.6.17/packages/ssl/Makefile.in	2007-02-24 10:50:04.000000000 +1300
@@ -128,6 +128,9 @@
 ################################################################
 
 check::
+		(cd etc && \
+		openssl ca -batch -policy policy_anything -passin pass:apenoot -out server/server-cert.pem -infiles server/server-key.pem && \
+		openssl ca -batch -policy policy_anything -passin pass:apenoot -out client/client-cert.pem -infiles client/client-key.pem)
 		$(PL) -q -f ssl_test.pl -F none -g test,halt -t 'halt(1)'
 
 ################################################################
