--- pl-5.6.4.orig/packages/Makefile.in	2005-11-13 03:15:23.000000000 +1300
+++ pl-5.6.4/packages/Makefile.in	2006-02-13 23:09:00.000000000 +1300
@@ -76,7 +76,7 @@
 
 install::
 	for p in $(PKG); do \
-	    if [ -r $$p/Makefile ]; then ( cd $$p && $(MAKE) $@ ); fi; \
+	    if [ -r $$p/Makefile ]; then $(MAKE) -C $$p $@ || exit 1; fi; \
         done
 	@echo "*** Forcing update of library index ***"
 	rm -f $(DESTDIR)$(PLLIB)/INDEX.pl
@@ -110,7 +110,7 @@
 	        echo "**********************"; \
 		echo "* Package $$p"; \
 		echo "**********************"; \
-	        ( cd $$p && $(CONFIG) ); \
+	        ( cd $$p && $(CONFIG) ) || exit 1; \
 	    fi; \
         done
 
@@ -127,7 +127,7 @@
 
 check::
 	for p in $(PKG); do \
-	    if [ -r $$p/Makefile ]; then ( cd $$p && $(MAKE) $@ ); fi; \
+	    if [ -r $$p/Makefile ]; then $(MAKE) -C $$p $@ || exit 1; fi; \
         done
 
 # test environment
--- pl-5.6.4.orig/packages/db/Makefile.in	2005-12-15 04:54:38.000000000 +1300
+++ pl-5.6.4/packages/db/Makefile.in	2006-02-13 23:05:28.000000000 +1300
@@ -9,6 +9,7 @@
 PL=@PL@
 PLBASE=@PLBASE@
 PLARCH=@PLARCH@
+PKGDOC=$(PLBASE)/doc/packages
 PCEHOME=/staff/jan/src/xpce
 DESTDIR=
 
@@ -57,10 +58,16 @@
 		$(INSTALL) -m 644 $(LIBPL) $(DESTDIR)$(PLBASE)/library
 		$(PL) -f none -g make -t halt
 
+html-install::
+		mkdir -p $(DESTDIR)$(PKGDOC)
+		$(INSTALL_DATA) $(DOC).html $(DESTDIR)$(PKGDOC)
+
 uninstall:
 		(cd $(PLBASE)/library && rm -f $(LIBPL))
 		$(PL) -f none -g make -t halt
 
+check::
+
 ################################################################
 # Documentation
 ################################################################
--- pl-5.6.4.orig/packages/mp/Makefile.in	2006-02-04 04:38:50.000000000 +1300
+++ pl-5.6.4/packages/mp/Makefile.in	2006-02-13 22:53:53.000000000 +1300
@@ -7,6 +7,7 @@
 PL=@PL@
 PLBASE=@PLBASE@
 PLARCH=@PLARCH@
+PKGDOC=$(PLBASE)/doc/packages
 
 CC=@CC@
 COFLAGS=@COFLAGS@
@@ -17,6 +18,8 @@
 LOCALLIB=@LOCALLIB@
 LIBS=@LIBS@
 
+DOC=mp
+
 LD=@LD@
 LDFLAGS=@LDSOFLAGS@
 
@@ -50,6 +53,12 @@
 		done
 		$(PL) -f none -g make -t halt
 
+html-install:
+		mkdir -p $(DESTDIR)$(PKGDOC)
+		$(INSTALL_DATA) $(DOC).html $(DESTDIR)$(PKGDOC)
+
+check::
+
 uninstall:
 		(cd $(PLBASE)/lib/$(PLARCH) && rm -f $(TARGETS))
 		(cd $(PLBASE)/library && rm -f $(LIBPL))
--- pl-5.6.4.orig/packages/mp/configure.in	2004-09-22 02:28:17.000000000 +1200
+++ pl-5.6.4/packages/mp/configure.in	2006-02-13 23:02:37.000000000 +1300
@@ -68,6 +68,7 @@
 AC_MSG_RESULT("		PLSOEXT=$PLSOEXT")
 else
 PLLD=../plld.sh
+PL=../pl.sh
 fi
 
 CC=$PLLD
