diff -Nur nam-1.11.orig/netgraph.h nam-1.11/netgraph.h
--- nam-1.11.orig/netgraph.h	2000-05-18 14:06:32.000000000 -0400
+++ nam-1.11/netgraph.h	2006-06-17 02:36:36.000000000 -0400
@@ -68,7 +68,7 @@
 	virtual void handle(const TraceEvent&, double now, int direction);
 	virtual int command(int argc, const char *const *argv);
 	virtual void BoundingBox(BBox&);
-	virtual void NetGraph::render(GraphView* view);
+	virtual void render(GraphView* view);
     protected:
 	GraphView *views_;
 	float graphdata_[MAX_GRAPH];
diff -Nur nam-1.11.orig/node.h nam-1.11/node.h
--- nam-1.11.orig/node.h	2001-08-09 21:45:47.000000000 -0400
+++ nam-1.11/node.h	2006-06-17 02:17:40.000000000 -0400
@@ -223,7 +223,7 @@
 	Queue* queue() { return queue_; }
 	void add_sess_queue(unsigned int grp, Queue *q);
 	char * getTclScript();
-	char * Node::getTclScriptLabel();
+	char * getTclScriptLabel();
 	void setTclScript(const char * label, const char * script);
 
 	int command(int argc, const char * const * argv);
diff -Nur nam-1.11.orig/parser.h nam-1.11/parser.h
--- nam-1.11.orig/parser.h	2005-01-24 14:53:07.000000000 -0500
+++ nam-1.11/parser.h	2006-06-17 02:16:59.000000000 -0400
@@ -94,8 +94,8 @@
 class ParseTable {
 public:
   //<zheng: +++>
-  static bool ParseTable::nam4wpan;
-  static int ParseTable::wpan_bradius;
+  static bool nam4wpan;
+  static int wpan_bradius;
   //</zheng: +++>
   TraceSyntax * syntax_list;
 private:
