--- Makefile.old	2006-10-29 15:32:22.000000000 +0100
+++ Makefile	2006-10-29 15:36:27.000000000 +0100
@@ -58,7 +58,7 @@
 # low overhead version (internal parser)
 lo_xcalib: xcalib.c
 	$(CC) $(CFLAGS) -c xcalib.c -I$(XINCLUDEDIR) -DXCALIB_VERSION=\"$(XCALIB_VERSION)\"
-	$(CC) $(CFLAGS) -L$(XLIBDIR) -lm -o xcalib xcalib.o -lX11 -lXxf86vm -lXext
+	$(CC) $(CFLAGS) $(LDFLAGS) -L$(XLIBDIR) -o xcalib xcalib.o -lm -lX11 -lXxf86vm -lXext
 
 fglrx_xcalib: xcalib.c
 	$(CC) $(CFLAGS) -c xcalib.c -I$(XINCLUDEDIR) -DXCALIB_VERSION=\"$(XCALIB_VERSION)\" -I$(FGLRXINCLUDEDIR) -DFGLRX
