diff -urN mpeg4ip-1.5.0.1-orig/configure.in mpeg4ip-1.5.0.1/configure.in
--- mpeg4ip-1.5.0.1-orig/configure.in	2006-06-09 04:15:16.000000000 +0900
+++ mpeg4ip-1.5.0.1/configure.in	2008-06-21 13:44:38.000000000 +0900
@@ -495,8 +495,8 @@
         AC_DEFINE(HAVE_FFMPEG_INSTALLED, 1, [defined if ffmpeg is installed])
         FFMPEG_INC=
         FFMPEG_LIB="-lavcodec -lz $ffmpeg_add_libs"
-        AC_CHECK_TYPES([AVRational], , , [#include <ffmpeg/avcodec.h>])
-	AC_CHECK_MEMBERS(AVCodecContext.time_base, , , [#include <ffmpeg/avcodec.h>])
+        AC_CHECK_TYPES([AVRational], , , [#include <libavcodec/avcodec.h>])
+	AC_CHECK_MEMBERS(AVCodecContext.time_base, , , [#include <libavcodec/avcodec.h>])
         AC_DEFINE(HAVE_FFMPEG, 1, [enable ffmpeg for mp4live])
      fi
   fi
diff -urN mpeg4ip-1.5.0.1-orig/player/plugin/audio/ffmpeg/ffmpeg.h mpeg4ip-1.5.0.1/player/plugin/audio/ffmpeg/ffmpeg.h
--- mpeg4ip-1.5.0.1-orig/player/plugin/audio/ffmpeg/ffmpeg.h	2004-10-12 06:26:46.000000000 +0900
+++ mpeg4ip-1.5.0.1/player/plugin/audio/ffmpeg/ffmpeg.h	2008-06-21 13:45:35.000000000 +0900
@@ -25,7 +25,7 @@
 #define __MPEG4IP_FFMPEG_H__ 1
 #include "codec_plugin.h"
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -urN mpeg4ip-1.5.0.1-orig/player/plugin/video/ffmpeg/ffmpeg.h mpeg4ip-1.5.0.1/player/plugin/video/ffmpeg/ffmpeg.h
--- mpeg4ip-1.5.0.1-orig/player/plugin/video/ffmpeg/ffmpeg.h	2005-04-06 03:04:46.000000000 +0900
+++ mpeg4ip-1.5.0.1/player/plugin/video/ffmpeg/ffmpeg.h	2008-06-21 13:47:48.000000000 +0900
@@ -25,7 +25,7 @@
 #define __MPEG4IP_FFMPEG_H__ 1
 #include "codec_plugin.h"
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -urN mpeg4ip-1.5.0.1-orig/server/mp4live/audio_ffmpeg.h mpeg4ip-1.5.0.1/server/mp4live/audio_ffmpeg.h
--- mpeg4ip-1.5.0.1-orig/server/mp4live/audio_ffmpeg.h	2005-09-15 08:15:42.000000000 +0900
+++ mpeg4ip-1.5.0.1/server/mp4live/audio_ffmpeg.h	2008-06-21 13:48:21.000000000 +0900
@@ -24,7 +24,7 @@
 
 #include "audio_encoder.h"
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -urN mpeg4ip-1.5.0.1-orig/server/mp4live/video_encoder_class.cpp mpeg4ip-1.5.0.1/server/mp4live/video_encoder_class.cpp
--- mpeg4ip-1.5.0.1-orig/server/mp4live/video_encoder_class.cpp	2006-03-11 06:40:57.000000000 +0900
+++ mpeg4ip-1.5.0.1/server/mp4live/video_encoder_class.cpp	2008-06-21 13:48:31.000000000 +0900
@@ -26,7 +26,7 @@
 #include "video_util_filter.h"
 #ifdef HAVE_FFMPEG
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -urN mpeg4ip-1.5.0.1-orig/server/mp4live/video_ffmpeg.h mpeg4ip-1.5.0.1/server/mp4live/video_ffmpeg.h
--- mpeg4ip-1.5.0.1-orig/server/mp4live/video_ffmpeg.h	2005-09-09 02:40:36.000000000 +0900
+++ mpeg4ip-1.5.0.1/server/mp4live/video_ffmpeg.h	2008-06-21 13:48:13.000000000 +0900
@@ -24,7 +24,7 @@
 
 #include "video_encoder.h"
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
