diff -urp lm_sensors-2.10.0/Makefile lm_sensors-2.10.0-modules-only/Makefile
--- lm_sensors-2.10.0/Makefile	2006-02-15 02:46:46.000000000 +0100
+++ lm_sensors-2.10.0-modules-only/Makefile	2006-02-15 18:01:09.000000000 +0100
@@ -287,6 +287,7 @@ all::
 
 # Include all makefiles for sub-modules
 INCLUDEFILES := 
+SRCDIRS=kernel kernel/busses kernel/chips
 include $(patsubst %,%/Module.mk,$(SRCDIRS))
 ifneq ($(MAKECMDGOALS),clean)
 ifneq ($(MAKECMDGOALS),uninstall)
Only in lm_sensors-2.10.0-modules-only: Makefile.orig
