diff -urN abinit-5.3.4.orig/config/m4/init.m4 abinit-5.3.4/config/m4/init.m4
--- abinit-5.3.4.orig/config/m4/init.m4	2007-05-08 23:23:04.000000000 -0700
+++ abinit-5.3.4/config/m4/init.m4	2007-05-08 23:25:19.000000000 -0700
@@ -207,14 +207,14 @@
  fi
 
  dnl Set-up all directory names
- abinit_bindir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/bin"
- abinit_chkdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/tests"
+ abinit_bindir="${abinit_prefix}/bin"
+ abinit_chkdir="${abinit_prefix}/share/abinit/tests"
  abinit_datdir="${abinit_prefix}/abinit"
- abinit_docdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/doc"
- abinit_incdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/include"
- abinit_libdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/lib"
- abinit_mandir="${abinit_prefix}/abinit/man"
- abinit_rundir="${abinit_prefix}/abinit/bin"
+ abinit_docdir="${abinit_prefix}/share/doc/abinit-${ABINIT_VERSION_BASE}"
+ abinit_incdir="${abinit_prefix}/include"
+ abinit_libdir="${abinit_prefix}/lib"
+ abinit_mandir="${abinit_prefix}/share/man/man"
+ abinit_rundir="${abinit_prefix}/bin"
  abinit_wwwdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/www"
 
  dnl Substitute all variables
@@ -288,7 +288,7 @@
  fi
  ABINIT_VERSION_BUILD=`date '+%Y%m%d'`
 
- ABINIT_VERSION_BASE="${ABINIT_VERSION_MAJOR}.${ABINIT_VERSION_MINOR}"
+ ABINIT_VERSION_BASE="${ABINIT_VERSION_MAJOR}.${ABINIT_VERSION_MINOR}.${ABINIT_VERSION_MICRO}"
 
  dnl Make numbers available to source files
  AC_DEFINE_UNQUOTED(ABINIT_VERSION,"${ABINIT_VERSION}",
