diff --git a/config/m4/conf-init.m4 b/config/m4/conf-init.m4
index 3a20c30..8e84730 100644
--- a/config/m4/conf-init.m4
+++ b/config/m4/conf-init.m4
@@ -315,15 +315,15 @@ AC_DEFUN([ABI_INIT_INSTALL_DIRS],[
   fi
 
   dnl Set-up all directory names
-  abinit_bindir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/bin"
-  abinit_chkdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/tests"
-  abinit_datdir="${abinit_prefix}/abinit"
-  abinit_docdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/doc"
-  abinit_incdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/include"
-  abinit_libdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/lib"
-  abinit_mandir="${abinit_prefix}/abinit/man"
-  abinit_rundir="${abinit_prefix}/abinit/bin"
-  abinit_wwwdir="${abinit_prefix}/abinit/${ABINIT_VERSION_BASE}/www"
+  abinit_bindir="${abinit_prefix}/bin"
+  abinit_chkdir="${abinit_prefix}/share/abinit/tests"
+  abinit_datdir="${abinit_prefix}/share/abinit"
+  abinit_docdir="${abinit_prefix}/share/doc/abinit-${ABINIT_VERSION_BASE}"
+  abinit_incdir="${includedir}/abinit"
+  abinit_libdir="${libdir}"
+  abinit_mandir="${abinit_prefix}/share/man/man"
+  abinit_rundir="${abinit_prefix}/bin"
+  abinit_wwwdir="${abinit_prefix}/share/abinit/www"
 
   dnl Substitute all variables
   AC_SUBST(abinit_prefix)
