diff -urN abinit-5.0.1.orig/config/m4/init.m4 abinit-5.0.1/config/m4/init.m4
--- abinit-5.0.1.orig/config/m4/init.m4	2005-12-11 06:56:09.000000000 -0800
+++ abinit-5.0.1/config/m4/init.m4	2006-02-10 23:31:47.000000000 -0800
@@ -132,9 +132,9 @@
 
   debian)
    abinit_prefix="/usr"
-   abinit_bindir="/usr/lib/abinit/${ABINIT_VERSION_BASE}/bin"
+   abinit_bindir="/usr/bin"
    abinit_datdir="/usr/share/abinit"
-   abinit_docdir="/usr/share/doc/abinit/${ABINIT_VERSION_BASE}"
+   abinit_docdir="/usr/share/doc/abinit-${ABINIT_VERSION_BASE}"
    abinit_libdir="/usr/lib/abinit/${ABINIT_VERSION_BASE}/lib"
    abinit_incdir="/usr/include/abinit/${ABINIT_VERSION_BASE}"
    abinit_mandir="/usr/share/man"
@@ -212,7 +212,7 @@
  ABINIT_VERSION_MICRO=`echo "${ABINIT_VERSION}" | cut -d. -f3`
  ABINIT_VERSION_BUILD=`date '+%Y%m%d'`
 
- ABINIT_VERSION_BASE="${ABINIT_VERSION_MAJOR}.${ABINIT_VERSION_MINOR}"
+ ABINIT_VERSION_BASE="${ABINIT_VERSION_MAJOR}.${ABINIT_VERSION_MINOR}.${ABINIT_VERSION_MICRO}"
 
  dnl Make numbers available to source files
  AC_DEFINE_UNQUOTED(ABINIT_VERSION,"${ABINIT_VERSION}",
