diff -urN abinit-5.1.2.orig/config/m4/init.m4 abinit-5.1.2/config/m4/init.m4
--- abinit-5.1.2.orig/config/m4/init.m4	2006-05-23 23:09:03.000000000 -0700
+++ abinit-5.1.2/config/m4/init.m4	2006-05-23 23:07:31.000000000 -0700
@@ -137,7 +137,7 @@
   alpha*|powerpc*)
    abi_cpu_model="${target_cpu}"
    abi_cpu_64bits=`echo "${abi_cpu_model}" | grep 64`
-   if test abi_cpu_64bits = ""; then
+   if test "${abi_cpu_64bits}" = ""; then
     abi_cpu_64bits="no"
    else
     abi_cpu_64bits="yes"
