diff -urN vlc-0.6.1/Makefile.am vlc-0.6.1-modified/Makefile.am
--- vlc-0.6.1/Makefile.am	2003-07-28 17:26:11.000000000 -0500
+++ vlc-0.6.1-modified/Makefile.am	2003-07-31 23:37:00.000000000 -0500
@@ -7,7 +7,7 @@
 # SUBDIRS stores the directories where a "make" is required when building
 # something. DIST_SUBDIRS stores the directories where nothing is built but
 # which have makefiles with distribution information.
-SUBDIRS = intl modules po mozilla share m4 .
+SUBDIRS = intl modules po . mozilla share m4
 DIST_SUBDIRS = $(SUBDIRS) src debian doc ipkg lib autotools
 
 EXTRA_DIST = \
