diff -Naur kimera-2.11.orig/kimera.pro kimera-2.11/kimera.pro
--- kimera-2.11.orig/kimera.pro	2009-01-26 00:36:40.000000000 +0900
+++ kimera-2.11/kimera.pro	2009-02-11 09:12:00.000000000 +0900
@@ -1,5 +1,5 @@
 TEMPLATE        = app
-CONFIG         += qt x11 warn_on release
+CONFIG         += qt x11 warn_on release nostrip
 QT             += network
 TARGET          = kimera-bin
 VERSION         = 2.11
