diff --git a/makefile_refmac.inc1 b/makefile_refmac.inc1
index 97791e0..6c43c5d 100644
--- a/makefile_refmac.inc1
+++ b/makefile_refmac.inc1
@@ -83,9 +83,7 @@ messy_pdb.o                 \
 bridge.o                    \
 i0.o                        \
 i1.o                        \
-chbevl.o                    \
-$(LLIBCIF)                  \
-$(LLIBLAPACK)
+chbevl.o
 
 
 SUBL =                      \
@@ -118,9 +116,8 @@ cell_symm_subrs.o           \
 iheap_sort.o                \
 linalgebra.o                \
 math_refmac.o               \
-subnumrec.o                 \
-$(LLIBLAPACK)               \
-$(LLIBCCP)
+subnumrec.o
+
 
 #--------------------------------------------------------------------------------
 
@@ -188,7 +185,7 @@ linalgebra_f90.o: linalgebra_f90.f90
 	$(FC) $(FFLAGS) -c linalgebra_f90.f90
 
 refmac:	$(SUBR)
-	$(FC) $(FFLAGS) -o refmac${VERSION}   ${SUBR} $(LLIBOTHERS) $(LLIBCCP)
+	$(FC) $(FFLAGS) $(LDFLAGS) -o refmac${VERSION}   ${SUBR} $(LLIBOTHERS) $(LLIBCCP) $(LLIBCIF) $(LLIBLAPACK)
 
 mol2_file.o:	mol2_file.f odb_files.f90
 	$(FC) $(FFLAGS) -c odb_files.f90 mol2_file.f
@@ -209,7 +206,7 @@ libcheck.o:	libcheck.f elements_module.f90 pdb2libcheck.f90
 	$(FC) $(FFLAGS) -c elements_module.f90 pdb2libcheck.f90 libcheck.f
 
 libcheck:	$(SUBL)
-	$(FC) $(FFLAGS) -o libcheck${VERSION} $(SUBL)
+	$(FC) $(FFLAGS) $(LDFLAGS) -o libcheck${VERSION} $(SUBL) $(LLIBCCP) $(LLIBCIF) $(LLIBLAPACK)
 
 clean:
 	/bin/rm *.o
