--- panel-plugin/hddtemp.c	2007-12-11 23:14:02.000000000 +0200
+++ panel-plugin/hddtemp.c.new	2008-04-03 12:47:36.000000000 +0300
@@ -122,6 +122,8 @@
     while ( (dirname = g_dir_read_name (gdir))!=NULL ) {
         if ( strncmp (dirname, "ram", 3)!=0 &&
              strncmp (dirname, "loop", 4)!=0 &&
+	     strncmp (dirname, "md", 2)!=0 &&
+	     strncmp (dirname, "fd", 2)!=0 &&
              strncmp (dirname, "dm-", 3)!=0 ) {
             /* TODO: look, if /dev/dirname exists? */
             chipfeature = g_new0 (t_chipfeature, 1);
@@ -283,7 +285,7 @@
 
     TRACE ("enters get_hddtemp_value for %s", disk);
 
-    cmd_line = g_strdup_printf ( "%s -F -n -q %s", PATH_HDDTEMP, disk);
+    cmd_line = g_strdup_printf ( "%s -n -q %s", PATH_HDDTEMP, disk);
 
     msg_text = NULL;
 
