diff -Naur xfce4-sensors-plugin-0.10.99.3_a/panel-plugin/middlelayer.c xfce4-sensors-plugin-0.10.99.3_b/panel-plugin/middlelayer.c
--- xfce4-sensors-plugin-0.10.99.3_a/panel-plugin/middlelayer.c	2007-11-24 16:03:25.000000000 +0100
+++ xfce4-sensors-plugin-0.10.99.3_b/panel-plugin/middlelayer.c	2008-04-16 12:04:48.000000000 +0200
@@ -187,7 +187,7 @@
             feature = (t_chipfeature *) g_ptr_array_index (chip->chip_features, number);
             g_assert (feature!=NULL);
             /*  *value = get_acpi_zone_value (feature->name); */
-            /* refresh_acpi ((gpointer) feature);   */
+            refresh_acpi ((gpointer) feature, NULL);
             *value = feature->raw_value;
             return 0; /* HERE    I    AM,    I    WANNA    BE    FIXED    */
         #else
