From b3611b52578dcab5fca78f26ba37406b352a1419 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 23 Nov 2009 11:59:58 +0100
Subject: Makefile: add WITH_SWITCH_ROOT

---
 Makefile |   12 +++++++++++-
 1 files changed, 11 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index f6d162f..34cdc65 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,13 @@ mandir ?= ${prefix}/share/man
 modules.d/99base/switch_root: switch_root.c
 	gcc -D _GNU_SOURCE -D 'PACKAGE_STRING="dracut"' -std=gnu99 -fsigned-char -g -O2 -o modules.d/99base/switch_root switch_root.c	
 
-all: modules.d/99base/switch_root
+ifeq (1,${WITH_SWITCH_ROOT})
+targets = modules.d/99base/switch_root
+else
+targets = 
+endif
+
+all: $(targets)
 
 .PHONY: install clean archive rpm testimage test all check
 
@@ -24,12 +30,16 @@ install:
 	install -m 0755 dracut $(DESTDIR)$(sbindir)/dracut
 	install -m 0755 dracut-gencmdline $(DESTDIR)$(sbindir)/dracut-gencmdline
 	install -m 0755 dracut-catimages $(DESTDIR)$(sbindir)/dracut-catimages
+ifeq (1,${WITH_SWITCH_ROOT})
 	install -m 0755 modules.d/99base/switch_root $(DESTDIR)$(sbindir)/switch_root
+endif
 	install -m 0644 dracut.conf $(DESTDIR)$(sysconfdir)/dracut.conf
 	install -m 0755 dracut-functions $(DESTDIR)$(pkglibdir)/dracut-functions
 	cp -arx modules.d $(DESTDIR)$(pkglibdir)
 	install -m 0644 dracut.8 $(DESTDIR)$(mandir)/man8
+ifeq (1,${WITH_SWITCH_ROOT})
 	rm $(DESTDIR)$(pkglibdir)/modules.d/99base/switch_root
+endif
 
 clean:
 	rm -f *~
-- 
1.6.2.5

