From 2cc392f32c86b1a603d4b07ea20e25e209113e47 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20=C5=BBo=C5=82nowski?= <aidecoe@aidecoe.name>
Date: Wed, 16 Nov 2011 15:16:07 +0100
Subject: [PATCH] Makefile: Append target files names for man pages.

---
 Makefile |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 80173a3..454cd8e 100644
--- a/Makefile
+++ b/Makefile
@@ -40,11 +40,11 @@ install: doc
 	install -m 0755 dracut-functions $(DESTDIR)$(pkglibdir)/dracut-functions
 	install -m 0755 dracut-logger $(DESTDIR)$(pkglibdir)/dracut-logger
 	cp -arx modules.d $(DESTDIR)$(pkglibdir)
-	install -m 0644 dracut.8 $(DESTDIR)$(mandir)/man8
-	install -m 0644 dracut-catimages.8 $(DESTDIR)$(mandir)/man8
-	install -m 0644 dracut-gencmdline.8 $(DESTDIR)$(mandir)/man8
-	install -m 0644 dracut.conf.5 $(DESTDIR)$(mandir)/man5
-	install -m 0644 dracut.kernel.7 $(DESTDIR)$(mandir)/man7
+	install -m 0644 dracut.8 $(DESTDIR)$(mandir)/man8/dracut.8 
+	install -m 0644 dracut-catimages.8 $(DESTDIR)$(mandir)/man8/dracut-catimages.8 
+	install -m 0644 dracut-gencmdline.8 $(DESTDIR)$(mandir)/man8/dracut-gencmdline.8 
+	install -m 0644 dracut.conf.5 $(DESTDIR)$(mandir)/man5/dracut.conf.5 
+	install -m 0644 dracut.kernel.7 $(DESTDIR)$(mandir)/man7/dracut.kernel.7 
 
 clean:
 	$(RM) *~
-- 
1.7.8.rc1

