--- kportage/config/parsemakeconfs.cpp.orig	2002-11-02 16:40:24.000000000 +0100
+++ kportage/config/parsemakeconfs.cpp	2002-11-11 19:22:40.000000000 +0100
@@ -48,11 +48,11 @@
 					//kdDebug() << "multiline: " << foo << endl;
 					if ( !t.eof()) foo = t.readLine().stripWhiteSpace();
 					while ( !t.eof() && foo.contains('"')==0 ) {
-						foo = t.readLine().stripWhiteSpace();
 						//kdDebug() << foo << endl;
-						value += foo+" ";
+						value += foo.replace(QRegExp("\"|\'"),"")+" ";
+						foo = t.readLine().stripWhiteSpace();
 					}
-					value += foo+" ";
+					value += foo.replace(QRegExp("\"|\'"),"")+" ";
 					value.stripWhiteSpace();
 					//kdDebug() << "end of multiline, grabbed " << value << endl;
 				}
@@ -79,7 +79,7 @@
 bool ParseMakeConfs::incremental(QString foo) {
 	if (foo.contains(QRegExp("^FEATURES"))>0 || foo.contains(QRegExp("^ACCEPT_KEYWORDS"))>0 ||
 		foo.contains(QRegExp("^ACCEPT_LICENSE"))>0 || foo.contains(QRegExp("^CONFIG_PROTECT"))>0 ||
-		foo.contains(QRegExp("^CONFIG_PROTECT_MASK"))>0) //|| foo.contains(QRegExp("^USE"))>0)
+		foo.contains(QRegExp("^CONFIG_PROTECT_MASK"))>0 || foo.contains(QRegExp("^USE"))>0)
 		return true;
 	else return false;
 }
@@ -410,8 +410,11 @@
 		if (file.open(IO_WriteOnly)) {
 			QTextStream w( &file);
 			for (QStringList::Iterator it=makeConf.begin(); it!=makeConf.end(); ++it) {
-				w << *it << endl;
-				//kdDebug() << *it << endl;
+				if ((*it).contains('=')==0 && (*it).contains('#')==0)
+					w << "#" << endl;
+					//kdDebug() << "#" << endl;
+				else w << *it << endl;
+				//else kdDebug() << *it << endl;
 			}
 			file.close();
 		} else
