--- Makefile.orig	2008-11-27 18:16:03.000000000 +0000
+++ Makefile	2008-11-27 18:17:05.000000000 +0000
@@ -1,15 +1,13 @@
-CC=gcc
-CFLAGS=-Wall `pkg-config --cflags gtk+-2.0`
+MYCFLAGS=-Wall `pkg-config --cflags gtk+-2.0`
 LIBS=`pkg-config --libs gtk+-2.0`
 bindir ?= /usr/bin
 mandir ?= /usr/share/man
 
 .c.o:
-	$(CC) -c $(CFLAGS) $(CPPFLAGS) $<
+	$(CC) -c $(CFLAGS) $(CPPFLAGS) $(MYCFLAGS) $<
 
 g3data: main.o sort.o points.o drawing.o g3data.1.gz
-	$(CC) $(CFLAGS) -o g3data main.o sort.o points.o drawing.o $(LIBS)
-	strip g3data
+	$(CC) $(CFLAGS) $(MYCFLAGS) -o g3data main.o sort.o points.o drawing.o $(LIBS)
 
 main.o: main.c main.h strings.h vardefs.h
 
