diff -urN kdeui/kdockwidget.h kdeui/kdockwidget.h
--- kdeui/kdockwidget.h	2004-11-27 15:14:58.000000000 +0200
+++ kdeui/kdockwidget.h	2005-02-06 19:07:43.000000000 +0200
@@ -100,7 +100,7 @@
  *
  * @author Max Judin (documentation: Falk Brettschneider).
  */
-class KDockWidgetAbstractHeader : public QFrame
+class KDEUI_EXPORT KDockWidgetAbstractHeader : public QFrame
 {
   Q_OBJECT
 public:
diff -urN khtml/dom/css_stylesheet.h khtml/dom/css_stylesheet.h
--- khtml/dom/css_stylesheet.h	2003-10-19 19:58:55.000000000 +0200
+++ khtml/dom/css_stylesheet.h	2005-02-06 19:00:30.000000000 +0200
@@ -54,7 +54,7 @@
  * instruction </a> .
  *
  */
-class StyleSheet
+class KHTML_EXPORT StyleSheet
 {
 public:
     StyleSheet();
@@ -203,7 +203,7 @@
  * content type is "text/css".
  *
  */
-class CSSStyleSheet : public StyleSheet
+class KHTML_EXPORT CSSStyleSheet : public StyleSheet
 {
 public:
     CSSStyleSheet();
@@ -305,7 +305,7 @@
  * abstraction of an ordered collection of style sheets.
  *
  */
-class StyleSheetList
+class KHTML_EXPORT StyleSheetList
 {
 public:
     StyleSheetList();
@@ -359,7 +359,7 @@
  * strings.
  *
  */
-class MediaList
+class KHTML_EXPORT MediaList
 {
 public:
     MediaList();
@@ -443,7 +443,7 @@
 
 class LinkStyleImpl;
 
-class LinkStyle
+class KHTML_EXPORT LinkStyle
 {
 public:
     LinkStyle();
@@ -465,7 +465,7 @@
 
 class DocumentStyleImpl;
 
-class DocumentStyle
+class KHTML_EXPORT DocumentStyle
 {
 public:
     DocumentStyle();
diff -urN khtml/dom/dom2_traversal.h khtml/dom/dom2_traversal.h
--- khtml/dom/dom2_traversal.h	2005-01-07 14:43:08.000000000 +0200
+++ khtml/dom/dom2_traversal.h	2005-02-06 18:59:58.000000000 +0200
@@ -335,7 +335,7 @@
  * how deeply nested the structure of the original document.
  *
  */
-class TreeWalker
+class KHTML_EXPORT TreeWalker
 {
     friend class Document;
     friend class TreeWalkerImpl;
diff -urN khtml/dom/dom_xml.h khtml/dom/dom_xml.h
--- khtml/dom/dom_xml.h	2005-01-03 16:53:24.000000000 +0200
+++ khtml/dom/dom_xml.h	2005-02-06 17:08:22.000000000 +0200
@@ -121,7 +121,7 @@
  *  An \c Entity node does not have any parent.
  *
  */
-class Entity : public Node
+class KHTML_EXPORT Entity : public Node
 {
 public:
     Entity();
@@ -185,7 +185,7 @@
  * evaluation.
  *
  */
-class EntityReference : public Node
+class KHTML_EXPORT EntityReference : public Node
 {
     friend class Document;
 public:
@@ -219,7 +219,7 @@
  *  A \c Notation node does not have any parent.
  *
  */
-class Notation : public Node
+class KHTML_EXPORT Notation : public Node
 {
 public:
     Notation();
@@ -256,7 +256,7 @@
  * processor-specific information in the text of the document.
  *
  */
-class ProcessingInstruction : public Node
+class KHTML_EXPORT ProcessingInstruction : public Node
 {
     friend class Document;
 public:
diff -urN khtml/dom/html_head.h khtml/dom/html_head.h
--- khtml/dom/html_head.h	2005-01-03 16:53:24.000000000 +0200
+++ khtml/dom/html_head.h	2005-02-06 17:09:36.000000000 +0200
@@ -332,7 +332,7 @@
  * SCRIPT element definition </a> in HTML 4.0.
  *
  */
-class HTMLScriptElement : public HTMLElement
+class KHTML_EXPORT HTMLScriptElement : public HTMLElement
 {
 public:
     HTMLScriptElement();
@@ -446,7 +446,7 @@
  * STYLE element definition </a> in HTML 4.0.
  *
  */
-class HTMLStyleElement : public HTMLElement
+class KHTML_EXPORT HTMLStyleElement : public HTMLElement
 {
 public:
     HTMLStyleElement();
@@ -519,7 +519,7 @@
  * TITLE element definition </a> in HTML 4.0.
  *
  */
-class HTMLTitleElement : public HTMLElement
+class KHTML_EXPORT HTMLTitleElement : public HTMLElement
 {
 public:
     HTMLTitleElement();
diff -urN khtml/dom/html_image.h khtml/dom/html_image.h
--- khtml/dom/html_image.h	2003-08-30 12:24:18.000000000 +0300
+++ khtml/dom/html_image.h	2005-02-06 17:11:04.000000000 +0200
@@ -44,7 +44,7 @@
  * AREA element definition </a> in HTML 4.0.
  *
  */
-class HTMLAreaElement : public HTMLElement
+class KHTML_EXPORT HTMLAreaElement : public HTMLElement
 {
 public:
     HTMLAreaElement();
@@ -182,7 +182,7 @@
  * IMG element definition </a> in HTML 4.0.
  *
  */
-class HTMLImageElement : public HTMLElement
+class KHTML_EXPORT HTMLImageElement : public HTMLElement
 {
 public:
     HTMLImageElement();
@@ -390,7 +390,7 @@
  * MAP element definition </a> in HTML 4.0.
  *
  */
-class HTMLMapElement : public HTMLElement
+class KHTML_EXPORT HTMLMapElement : public HTMLElement
 {
 public:
     HTMLMapElement();
diff -urN khtml/dom/html_object.h khtml/dom/html_object.h
--- khtml/dom/html_object.h	2004-01-09 03:56:16.000000000 +0200
+++ khtml/dom/html_object.h	2005-02-06 17:11:13.000000000 +0200
@@ -47,7 +47,7 @@
  * deprecated in HTML 4.0.
  *
  */
-class HTMLAppletElement : public HTMLElement
+class KHTML_EXPORT HTMLAppletElement : public HTMLElement
 {
 public:
     HTMLAppletElement();
@@ -255,7 +255,7 @@
  * OBJECT element definition </a> in HTML 4.0.
  *
  */
-class HTMLObjectElement : public HTMLElement
+class KHTML_EXPORT HTMLObjectElement : public HTMLElement
 {
 public:
     HTMLObjectElement();
@@ -554,7 +554,7 @@
  * PARAM element definition </a> in HTML 4.0.
  *
  */
-class HTMLParamElement : public HTMLElement
+class KHTML_EXPORT HTMLParamElement : public HTMLElement
 {
 public:
     HTMLParamElement();
diff -urN kmdi/kmditaskbar.h kmdi/kmditaskbar.h
--- kmdi/kmditaskbar.h	2005-01-03 16:53:26.000000000 +0200
+++ kmdi/kmditaskbar.h	2005-02-06 19:06:40.000000000 +0200
@@ -48,7 +48,7 @@
   * It's a special kind of QPushButton catching mouse clicks.
   * And you have the ability to abbreviate the text that it fits in the button.
   */
-class KMdiTaskBarButton : public QPushButton
+class KMDI_EXPORT KMdiTaskBarButton : public QPushButton
 {
 	Q_OBJECT
 	// methods
