--- kio/bookmarks/kbookmarkmenu.cc~	2003-02-03 16:41:06.000000000 +0100
+++ kio/bookmarks/kbookmarkmenu.cc	2003-02-03 16:44:46.000000000 +0100
@@ -269,10 +269,10 @@
       else
       {
         // kdDebug(1203) << "Creating URL bookmark menu item for " << bm.text() << endl;
-        // create a normal URL item, with ID as a name
+        // create a normal URL item, with address as a name
         KAction * action = new KAction( text, bm.icon(), 0,
                                         this, SLOT( slotBookmarkSelected() ),
-                                        m_actionCollection, bm.url().url().utf8() );
+                                        m_actionCollection, bm.address().utf8() );
 
         action->setStatusText( bm.url().prettyURL() );
 
@@ -369,8 +369,11 @@
   if ( !m_pOwner ) return; // this view doesn't handle bookmarks...
   //kdDebug(1203) << sender()->name() << endl;
 
-  // The name of the action is the URL to open
-  m_pOwner->openBookmarkURL( QString::fromUtf8(sender()->name()) );
+  KBookmark bookmark = m_pManager->findByAddress( QString::fromUtf8(sender()->name()) );
+  Q_ASSERT(!bookmark.isNull());
+  Q_ASSERT(!bookmark.isGroup());
+
+  m_pOwner->openBookmarkURL( bookmark.url().url() );
 }
 
 // -----------------------------------------------------------------------------
