--- kio/kssl/ksslcertificate.cc~	2002-12-03 16:41:23.000000000 +0100
+++ kio/kssl/ksslcertificate.cc	2002-12-03 16:42:12.000000000 +0100
@@ -513,6 +513,7 @@
 int KSSLCertificate::purposeToOpenSSL(KSSLCertificate::KSSLPurpose p) const {
 int rc = 0;
 
+#ifdef HAVE_SSL
 	if (p == KSSLCertificate::SSLServer) {
 		rc = X509_PURPOSE_SSL_SERVER;
 	} else if (p == KSSLCertificate::SSLClient) {
@@ -524,6 +525,7 @@
 	} else if (p == KSSLCertificate::Any) {
 		rc = X509_PURPOSE_ANY;
 	}
+#endif
 
 return rc;	
 }
