--- khtml/html/html_formimpl.cpp	2005-02-23 13:57:04.000000000 +0200
+++ khtml/html/html_formimpl.cpp	2005-03-08 14:47:02.000000000 +0200
@@ -2463,15 +2463,15 @@
         if ( m_render && m_initialized ) {
             RenderTextArea* renderArea = static_cast<RenderTextArea*>( m_render );
             m_value = renderArea->text();
+            m_dirtyvalue = false; // before onChange (#100963) 
             onChange();
         } else {
             m_value = defaultValue().string();
             m_initialized = true;
-        }
-
+
         m_dirtyvalue = false;
+               }
     }
-
     if ( m_value.isNull() ) return "";
 
     return m_value;
