--- postgresql-7.1/src/interfaces/perl5/GNUmakefile.orig	Tue Apr 17 21:28:21 2001
+++ postgresql-7.1/src/interfaces/perl5/GNUmakefile	Tue Apr 17 21:28:27 2001
@@ -36,20 +36,12 @@
 
 install: Makefile
 	$(MAKE) -f Makefile clean
-	POSTGRES_LIB="$(libdir)" \
-	  POSTGRES_INCLUDE="$(includedir)" \
-	  $(PERL) $(srcdir)/Makefile.PL
-	$(MAKE) -f Makefile all
-	-@if [ -w "`$(MAKE) --quiet -f Makefile echo-installdir`" ]; then \
-		$(MAKE) -f Makefile install; \
-		$(MAKE) clean; \
-	else \
-		echo "*****" ;\
-		echo "* Skipping the installation of the Perl module for lack of permissions."; \
-		echo "* To install it, change to the directory "`pwd`","; \
-		echo "* become the appropriate user, and do \`$(MAKE) install'."; \
-		echo "*****"; \
-	fi
+	POSTGRES_LIB="$(DESTDIR)$(libdir)" \
+	POSTGRES_INCLUDE="$(DESTDIR)$(includedir)" \
+	$(PERL) $(srcdir)/Makefile.PL
+	$(MAKE) -f Makefile all; \
+        $(MAKE) PREFIX=$(DESTDIR)$(prefix) INSTALLMAN3DIR=$(DESTDIR)$(mandir)/man3 -f Makefile install; \
+        $(MAKE) clean;
 
 
 uninstall:
