--- Makefile.am.orig	2008-12-27 15:31:15.000000000 +0100
+++ Makefile.am	2008-12-27 15:31:47.000000000 +0100
@@ -10,11 +10,11 @@
 	README.betatester README.win32 hamlib.spec
 
 # BINDINGS_LIST subdirs are no longer built
-SUBDIRS = macros include lib libltdl src @BACKEND_LIST@ @ROT_BACKEND_LIST@ \
+SUBDIRS = macros include lib src @BACKEND_LIST@ @ROT_BACKEND_LIST@ \
 			@BINDINGS@ tests doc
 
 # tcl, perl, and kylix subdirs are no longer distributed
-DIST_SUBDIRS = macros include lib libltdl src c++ bindings tests doc \
+DIST_SUBDIRS = macros include lib src c++ bindings tests doc \
 	icom kenwood aor yaesu dummy pcr alinco uniden tentec kachina jrc \
 	rpcrig winradio easycomm fodtrack rpcrot gnuradio drake rotorez \
 	flexradio sartek lowe rft tapr kit skanti wj racal tuner \
--- configure.ac.orig	2008-12-27 12:59:03.000000000 +0100
+++ configure.ac	2008-12-27 12:59:48.000000000 +0100
@@ -174,7 +174,7 @@
 ## ------------------------ ##
 dnl Enable building of the convenience library
 dnl and set LIBLTDL accordingly
-AC_LIBLTDL_CONVENIENCE
+AC_LIBLTDL_INSTALLABLE
 dnl Substitute INCLTDL and LIBLTDL in the Makefiles
 AC_SUBST([INCLTDL])
 AC_SUBST([LIBLTDL])
@@ -187,7 +187,7 @@
 AC_SUBST([LIBTOOL_DEPS])
 AC_LIB_LTDL
 dnl Configure libltdl
-AC_CONFIG_SUBDIRS(libltdl)
+#AC_CONFIG_SUBDIRS(libltdl)
 
 case "$host_os" in
 darwin* | rhapsody*)
