--- src/userinfohelpers.cpp~	2007-04-02 11:01:38.000000000 +0000
+++ src/userinfohelpers.cpp	2007-04-02 11:01:45.000000000 +0000
@@ -23,6 +23,8 @@
 
 #include "sstream_fix.h"
 
+#include <cstdlib>
+
 using std::vector;
 using std::string;
 using std::ostringstream;


--- src/socket.cpp.orig	2008-06-08 17:55:26.000000000 +0200
+++ src/socket.cpp	2008-06-08 17:55:57.000000000 +0200
@@ -19,6 +19,7 @@
  *
  */
 
+#include <cstring>
 #include "sstream_fix.h"
 #include <algorithm>
 
 
--- src/Capabilities.cpp.orig	2008-06-08 18:04:34.000000000 +0200
+++ src/Capabilities.cpp	2008-06-08 18:05:15.000000000 +0200
@@ -19,6 +19,7 @@
  *
  */
 
+#include <cstring>
 #include <libicq2000/Capabilities.h>
 
 using std::set;
