Index: listing.php
===================================================================
--- listing.php	(Revision 1)
+++ listing.php	(Arbeitskopie)
@@ -232,7 +232,15 @@
 $vars["action"] = "";
 $vars["author"] = $log['author'];
 $vars["date"] = $log['date'];
-$vars["log"] = nl2br($bugtraq->replaceIDs(create_anchors($log['message'])));
+$log_mess= nl2br($bugtraq->replaceIDs(create_anchors($log['message'])));
+$log_mess=str_replace("?\\195?\\182","&ouml;",$log_mess);
+$log_mess=str_replace("?\\195?\\150","&ouml;",$log_mess);
+$log_mess=str_replace("?\\195?\\156","&Uuml;",$log_mess);
+$log_mess=str_replace("?\\195?\\188","&uuml;",$log_mess);
+$log_mess=str_replace("?\\195?\\164","&auml;",$log_mess);
+$log_mess=str_replace("?\\195?\\132","&Auml;",$log_mess);
+$log_mess=str_replace("?\\195?\\159","&szlig;",$log_mess);
+$vars["log"] = $log_mess;
 $vars["rev"] = $rev;
 $vars["lastchangedrev"] = $logrev;
 $vars["path"] = $ppath;
Index: log.php
===================================================================
--- log.php	(Revision 1)
+++ log.php	(Arbeitskopie)
@@ -162,6 +162,7 @@
       {
          // Turn all the HTML entities into real characters.  
          
+	 $log_mess=$log["message"];
          if (version_compare(phpversion(), "4.1.0", "<"))
             $msg = html_entity_decode($log["message"]);
          else
@@ -215,7 +216,15 @@
       
       $listing[$index]["revauthor"] = $log["author"];
       $listing[$index]["revage"] = $log["age"];
-      $listing[$index]["revlog"] = nl2br($bugtraq->replaceIDs(create_anchors($log["message"])));
+      $log_mess = nl2br($bugtraq->replaceIDs(create_anchors($log["message"])));
+	  $log_mess=str_replace("?\\195?\\182","&ouml;",$log_mess);
+	  $log_mess=str_replace("?\\195?\\150","&ouml;",$log_mess);
+	  $log_mess=str_replace("?\\195?\\156","&Uuml;",$log_mess);
+	  $log_mess=str_replace("?\\195?\\188","&uuml;",$log_mess);
+	  $log_mess=str_replace("?\\195?\\164","&auml;",$log_mess);
+	  $log_mess=str_replace("?\\195?\\132","&Auml;",$log_mess);
+	  $log_mess=str_replace("?\\195?\\159","&szlig;",$log_mess);
+      $listing[$index]["revlog"] = $log_mess;
       $listing[$index]["rowparity"] = "$row";
       
       $row = 1 - $row;
Index: rss.php
===================================================================
--- rss.php	(Revision 1)
+++ rss.php	(Arbeitskopie)
@@ -98,7 +98,15 @@
  
    $url = $config->getURL($rep, $parent, "dir");
    
-   $desc = $log["message"];
+   $log_mess = $log["message"];
+   $log_mess=str_replace("?\\195?\\182","&ouml;",$log_mess);
+   $log_mess=str_replace("?\\195?\\150","&ouml;",$log_mess);
+   $log_mess=str_replace("?\\195?\\156","&Uuml;",$log_mess);
+   $log_mess=str_replace("?\\195?\\188","&uuml;",$log_mess);
+   $log_mess=str_replace("?\\195?\\164","&auml;",$log_mess);
+   $log_mess=str_replace("?\\195?\\132","&Auml;",$log_mess);
+   $log_mess=str_replace("?\\195?\\159","&szlig;",$log_mess);
+   $desc=$log_mess;
    $item = new FeedItem();
    
    // For the title, we show the first 10 words of the description
