diff -Nuar hwinfo-16.0.orig/src/ids/Makefile hwinfo-16.0/src/ids/Makefile
--- hwinfo-16.0.orig/src/ids/Makefile	2009-05-12 03:10:34.000000000 -0700
+++ hwinfo-16.0/src/ids/Makefile	2009-07-03 13:55:42.522438453 -0700
@@ -55,8 +55,11 @@
 
 hd_ids.c: hd_ids.h hd_ids_tiny.h
 
-hd_ids.h hd.ids: check_hd $(IDFILES)
+hd_ids.h hd.ids: build_ids
+
+build_ids: check_hd $(IDFILES)
 	./check_hd --check --sort --cfile hd_ids.h $(IDFILES)
+	touch build_ids
 
 hd_ids_tiny.h: check_hd hd.ids
 	./check_hd --mini --cfile hd_ids_tiny.h --log=hd_tiny.log --out=hd_tiny.ids hd.ids
diff -Nuar hwinfo-16.0.orig/src/isdn/cdb/Makefile hwinfo-16.0/src/isdn/cdb/Makefile
--- hwinfo-16.0.orig/src/isdn/cdb/Makefile	2009-05-12 03:10:34.000000000 -0700
+++ hwinfo-16.0/src/isdn/cdb/Makefile	2009-07-03 14:13:54.336372459 -0700
@@ -12,8 +12,9 @@
 	fi
 
 isdn_cdb.h: isdn_cdb ISDN.CDB.txt
-	rm -f isdn_cdb.h
-	./isdn_cdb ISDN.CDB.txt isdn_cdb.h
+	rm -f $@
+	./isdn_cdb ISDN.CDB.txt $@.tmp
+	mv $@.tmp $@
 
 lex.yy.c: isdn_cdb.lex
 	lex isdn_cdb.lex
@@ -29,5 +30,8 @@
 mk_isdnhwdb.o: mk_isdnhwdb.c cdb_read.h lex.yy.c
 
 ISDN.CDB.hwdb: mk_isdnhwdb ISDN.CDB.txt
-	rm -f  ISDN.CDB.hwdb
-	./mk_isdnhwdb ISDN.CDB.txt ISDN.CDB.hwdb
+	rm -f $@
+	./mk_isdnhwdb ISDN.CDB.txt $@.tmp
+	mv $@.tmp $@
+
+OBJS += isdn_cdb.h ISDN.CDB.hwdb
