diff -urN hwinfo-13.0.orig/Makefile.common hwinfo-13.0/Makefile.common
--- hwinfo-13.0.orig/Makefile.common	2005-09-20 10:32:52.000000000 +0100
+++ hwinfo-13.0/Makefile.common	2006-07-22 15:00:38.000000000 +0100
@@ -20,7 +20,7 @@
 
 CC	= gcc
 LD	= ld
-CFLAGS	= -Wall -Wno-pointer-sign -O2 -pipe -g $(SHARED_FLAGS) $(EXTRA_FLAGS) -I$(TOPDIR)/src/hd
+CFLAGS	= -Wall -pipe -g $(SHARED_FLAGS) $(EXTRA_FLAGS) -I$(TOPDIR)/src/hd
 SHARED_FLAGS	= -fPIC
 
 LDFLAGS	= -Lsrc

