diff -ru fakeroot-0.4.4.orig/faked.cc fakeroot-0.4.4/faked.cc
--- fakeroot-0.4.4.orig/faked.cc	1999-11-13 13:52:38.000000000 -0600
+++ fakeroot-0.4.4/faked.cc	2002-09-29 22:44:13.000000000 -0500
@@ -85,9 +85,10 @@
 #include <string.h>
 #include <set>
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 #include <signal.h>
 
+using namespace std;
 
 #if HAVE_SEMUN_DEF == 0
   union semun {
