Originally based on Debian patch jove_4.16.0.70-3.1.diff.gz

--- jove4.16.0.73-orig/doc/intro.nr
+++ jove4.16.0.73/doc/intro.nr
@@ -56,9 +56,8 @@
 ..
 .
 .\" Change the extra vertical spacing around .DS/.DE
-.\" Does not work with groff's version of MS (GS is 1 iff groff MS)
-.if !\n(GS .if n .nr DD 0v
-.if !\n(GS .if t .nr DD \n(PD/2u
+.if n .nr DD 0v
+.if t .nr DD \n(PD/2u
 .
 .nr LL 6.5i
 .nr LT 6.5i
@@ -152,7 +151,7 @@
 }
 
 _	
-\fBJOVE\ (C OvrWt)\ \ \ [Main:1]\ \ "hello.c"\ \ \(**\ \ /home/foo\fP	<\ the Mode Line
+\fBJOVE\ (C OvrWt)\ \ \ [Main:1]\ \ "hello.c"\ \ \(**\ \ /home/foo\ \ \ \ 15:23\fP	<\ the Mode Line
 _	
 : write-file (default hello.c) aloha.c\^\(sq 	<\ the Message Line
 =	
@@ -1978,7 +1977,7 @@
 \ \ \ \ return 0;	<\ second Window
 }\^\(sq
 _	
-\fBJOVE\ (C OvrWt)\ \ \ [Main:1]\ \ "aloha.c"\ \ \-\-\ \ /home/foo\fP	<\ the Mode Line
+\fBJOVE\ (C OvrWt)\ \ \ [Main:1]\ \ "aloha.c"\ \ \-\-\ \ /home/foo\ \ \ \ 15:28\fP	<\ the Mode Line
 _	
 [Point pushed]	<\ the Message Line
 =	
--- jove4.16.0.73-orig/doc/xjove.nr
+++ jove4.16.0.73/doc/xjove.nr
@@ -1,6 +1,6 @@
 .TH XJOVE 1 "8 July 1993"
 .SH NAME
-Xjove, Jovetool \- run JOVE under X-windows or Sunview with function-key and mouse support.
+Xjove \- run JOVE under X-windows with function-key and mouse support.
 .SH SYNOPSIS
 .B xjove
 [
@@ -13,23 +13,10 @@
 [
 args
 \&.\|.\|. ]
-.br
-.B jovetool
-[
-.I window_args
-] [
-.B \-nf
-] [
-.BI \-rc \ run_command_path
-]\&.\|.\|.
-[
-.I args
-\&.\|.\|. ]
+
 .SH DESCRIPTION
 .I Xjove
-(or
-.IR Jovetool )
-creates an XView or SunView frame and a tty subwindow within which mouse events
+creates an XView frame and a tty subwindow within which mouse events
 and function keys are translated to ASCII sequences which JOVE can
 parse.  The translated input events are sent to the process running in
 the tty subwindow, which is typically JOVE.
