Index: trunk/avahi-daemon/static-hosts.c
===================================================================
--- trunk/avahi-daemon/static-hosts.c (revision 1167)
+++ trunk/avahi-daemon/static-hosts.c (revision 1285)
@@ -99,5 +99,6 @@
     AVAHI_LLIST_REMOVE(StaticHost, hosts, hosts, s);
 
-    avahi_s_entry_group_free (s->group);
+    if (s->group)
+        avahi_s_entry_group_free (s->group);
 
     avahi_free(s->host);
@@ -113,5 +114,8 @@
 
     if (!h->group)
-        h->group = avahi_s_entry_group_new (avahi_server, entry_group_callback, h);
+        if (!(h->group = avahi_s_entry_group_new (avahi_server, entry_group_callback, h))) {
+            avahi_log_error("avahi_s_entry_group_new() failed: %s", avahi_strerror(err));
+            return;
+        }
 
     if (!avahi_address_parse (h->ip, AVAHI_PROTO_UNSPEC, &a)) {
@@ -130,5 +134,6 @@
 static void remove_static_host_from_server(StaticHost *h)
 {
-    avahi_s_entry_group_reset (h->group);
+    if (h->group)
+        avahi_s_entry_group_reset (h->group);
 }
  
