Index: avahi-compat-libdns_sd/compat.c
===================================================================
--- avahi-compat-libdns_sd/compat.c	(revision 1373)
+++ avahi-compat-libdns_sd/compat.c	(revision 1374)
@@ -469,25 +469,26 @@
 }
 
 int DNSSD_API DNSServiceRefSockFD(DNSServiceRef sdref) {
+
+    AVAHI_WARN_LINKAGE;
+    
     if (!sdref || sdref->n_ref <= 0)
         return -1;
 
-    AVAHI_WARN_LINKAGE;
-    
     return sdref->main_fd;
 }
 
 DNSServiceErrorType DNSSD_API DNSServiceProcessResult(DNSServiceRef sdref) {
     DNSServiceErrorType ret = kDNSServiceErr_Unknown;
 
-    assert(sdref);
-    assert(sdref->n_ref >= 1);
-    
     AVAHI_WARN_LINKAGE;
 
+    if (!sdref || sdref->n_ref <= 0)
+        return kDNSServiceErr_BadParam;
+    
+    sdref_ref(sdref);
+
     ASSERT_SUCCESS(pthread_mutex_lock(&sdref->mutex));
-
-    sdref_ref(sdref);
     
     /* Cleanup notification socket */
     if (read_command(sdref->main_fd) != COMMAND_POLL_DONE)
@@ -512,10 +513,10 @@
     
 finish:
 
+    ASSERT_SUCCESS(pthread_mutex_unlock(&sdref->mutex));
+    
     sdref_unref(sdref);
 
-    ASSERT_SUCCESS(pthread_mutex_unlock(&sdref->mutex));
-    
     return ret;
 }
 
@@ -613,7 +614,6 @@
 
     if (!ret_sdref)
         return kDNSServiceErr_BadParam;
-
     *ret_sdref = NULL;
 
     assert(regtype);
@@ -739,7 +739,10 @@
 
     AVAHI_WARN_LINKAGE;
 
-    assert(ret_sdref);
+    if (!ret_sdref)
+        return kDNSServiceErr_BadParam;
+    *ret_sdref = NULL;
+
     assert(name);
     assert(regtype);
     assert(domain);
@@ -853,7 +856,10 @@
 
     AVAHI_WARN_LINKAGE;
 
-    assert(ret_sdref);
+    if (!ret_sdref)
+        return kDNSServiceErr_BadParam;
+    *ret_sdref = NULL;
+
     assert(callback);
 
     if (interface == kDNSServiceInterfaceIndexLocalOnly ||
@@ -1096,7 +1102,6 @@
 
     if (!ret_sdref)
         return kDNSServiceErr_BadParam;
-
     *ret_sdref = NULL;
     
     if (!regtype)
@@ -1210,10 +1215,12 @@
 
     int ret = kDNSServiceErr_Unknown;
     AvahiStringList *txt = NULL;
-    assert(sdref);
 
     AVAHI_WARN_LINKAGE;
 
+    if (!sdref || sdref->n_ref <= 0)
+        return kDNSServiceErr_BadParam;
+
     if (flags || rref) {
         AVAHI_WARN_UNSUPPORTED;
         return kDNSServiceErr_Unsupported;
