diff -Naurp pilot-link-0.11.5.orig/bindings/Makefile.in pilot-link-0.11.5/bindings/Makefile.in
--- pilot-link-0.11.5.orig/bindings/Makefile.in	2002-09-28 15:19:45.000000000 -0400
+++ pilot-link-0.11.5/bindings/Makefile.in	2002-11-17 11:41:24.000000000 -0500
@@ -368,11 +368,11 @@ python-clean: Python/setup.py
 Perl/Makefile: Perl/Makefile.PL
 	cd Perl && $(PERL) Makefile.PL INSTALLDIRS=vendor
 
-perl-build: Perl/Makefile
-	cd Perl && $(MAKE) PREFIX=$(DESTDIR)$(prefix)
+perl-build:
+#	cd Perl && $(MAKE) PREFIX=$(DESTDIR)$(prefix)
 
-perl-install: perl-build
-	cd Perl && $(MAKE) install 
+perl-install:
+#	cd Perl && $(MAKE) install 
 
 perl-uninstall: Perl/Makefile
 	cd Perl && $(MAKE) uninstall
