diff -urN icecream.orig/suse/sysconfig.icecream icecream/suse/sysconfig.icecream
--- icecream.orig/suse/sysconfig.icecream	2004-08-30 10:01:30.000000000 +0100
+++ icecream/suse/sysconfig.icecream	2006-08-12 16:05:09.000000000 +0100
@@ -39,6 +39,15 @@
 #
 ## Type: string
 ## Path: Applications/icecream
+## Defaut: -v
+#
+# icecream's level of verbosity ([-v[v[v]]])
+#
+ICECREAM_VERBOSITY="-v"
+
+#
+## Type: string
+## Path: Applications/icecream
 ## Defaut: ""
 #
 # Identification for the network the scheduler and daemon run on. 
