Fix compilation with GNU GCC 4.3.0 and up.

Index: k3guitune-1.0/k3guitune/soundinput.h
===================================================================
--- k3guitune-1.0.orig/k3guitune/soundinput.h
+++ k3guitune-1.0/k3guitune/soundinput.h
@@ -41,7 +41,7 @@ class SoundInputException : public std::
   */
  class SoundInput {
   public:
-    static SoundInput *autodetect( const char *alsa_devicename, const char *dsp_devicename, int &blksize, int &sampfreq, double &sampfreq );
+    static SoundInput *autodetect( const char *alsa_devicename, const char *dsp_devicename, int &blksize, int &sampfreq1, double &sampfreq2 );
     virtual ~SoundInput();
     virtual int getData( short int *buf, int blksize ) = 0;
  };
