diff -Naur openal-20040817.orig/linux/Makefile.in openal-20040817/linux/Makefile.in
--- openal-20040817.orig/linux/Makefile.in	2002-09-05 15:54:36.000000000 -0700
+++ openal-20040817/linux/Makefile.in	2004-10-06 19:02:21.785622393 -0700
@@ -1,8 +1,7 @@
 INSTALL=@INSTALL@
 LN_S=@LN_S@
-DESTDIR=@prefix@/
-DESTINC=$(DESTDIR)include
-DESTLIB=$(DESTDIR)lib
+DESTINC=$(DESTDIR)@prefix@/@includedir@
+DESTLIB=$(DESTDIR)@libdir@
 MAKE=@MAKE@
 MAJOR_VERSION=@MAJOR_VERSION@
 MINOR_VERSION=@MINOR_VERSION@
