diff -Naur bibus1.4.3/Makefile bibus1.4.3.new/Makefile
--- bibus1.4.3/Makefile	2008-05-20 16:16:41.000000000 -0400
+++ bibus1.4.3.new/Makefile	2008-05-22 09:08:56.000000000 -0400
@@ -42,23 +42,23 @@
 	chmod 755 $(datadir)/bibus/bibusStart.py
 
 	# basic doc files
-	install -m755 -d $(datadir)/doc/bibus
-	install -m644 Docs/*.txt Docs/CHANGELOG Docs/copying $(datadir)/doc/bibus
+	install -m755 -d $(datadir)/doc/GENTOO-PF
+	install -m644 Docs/*.txt Docs/CHANGELOG Docs/copying $(datadir)/doc/GENTOO-PF
 
 	# generating bibus.cfg file
 	echo '[PATH]' > $(datadir)/bibus/bibus.cfg
 	echo 'python = $(python)' >> $(datadir)/bibus/bibus.cfg
 	echo 'oopath = $(oopath)' >> $(datadir)/bibus/bibus.cfg
-	echo 'docdir = $(datadir)/doc/bibus/html' >> $(datadir)/bibus/bibus.cfg
-	echo 'licence = $(datadir)/doc/bibus/copying' >> $(datadir)/bibus/bibus.cfg
-	echo 'localedir = $(datadir)/locale' >> $(datadir)/bibus/bibus.cfg
-	echo 'systemconf = $(sysconfdir)/bibus.config' >> $(datadir)/bibus/bibus.cfg
+	echo 'docdir = /usr/share/doc/GENTOO-PF/html' >> $(datadir)/bibus/bibus.cfg
+	echo 'licence = /usr/share/doc/GENTOO-PF/copying' >> $(datadir)/bibus/bibus.cfg
+	echo 'localedir = /usr/share/locale' >> $(datadir)/bibus/bibus.cfg
+	echo 'systemconf = /etc/bibus.config' >> $(datadir)/bibus/bibus.cfg
 endef
 
 define install-doc-en
 	# copying docs in from Docs/html/en/ for bibus-doc-en
 	cd Docs;\
-	find html/en/* -type f -exec install -m644 '{}' -D $(datadir)/doc/bibus/'{}' ';'
+	find html/en/* -type f -exec install -m644 '{}' -D $(datadir)/doc/GENTOO-PF/'{}' ';'
 endef
 
 define compile
@@ -85,7 +85,7 @@
 		echo "find $(datadir)/locale -name bibus.mo -exec rm -f {} \;" >> $(datadir)/bibus/Setup/uninstall.sh
 		echo "rm $(sysconfdir)/bibus.config" >> $(datadir)/bibus/Setup/uninstall.sh
 		echo "rm $(man1dir)/bibus.1" >> $(datadir)/bibus/Setup/uninstall.sh
-		echo "rm -rf $(datadir)/doc/bibus" >> $(datadir)/bibus/Setup/uninstall.sh
+		echo "rm -rf $(datadir)/doc/GENTOO-PF" >> $(datadir)/bibus/Setup/uninstall.sh
 		echo "rm $(datadir)/applications/bibus.desktop" >> $(datadir)/bibus/Setup/uninstall.sh
 		echo "rm $(datadir)/icons/hicolor/48x48/apps/bibus.png" >> $(datadir)/bibus/Setup/uninstall.sh
 	chmod 744 $(datadir)/bibus/Setup/uninstall.sh
diff -Naur bibus1.4.3/Setup/bibus.desktop bibus1.4.3.new/Setup/bibus.desktop
--- bibus1.4.3/Setup/bibus.desktop	2008-05-20 16:16:41.000000000 -0400
+++ bibus1.4.3.new/Setup/bibus.desktop	2008-05-22 09:28:30.000000000 -0400
@@ -1,10 +1,23 @@
 [Desktop Entry]
 Version=1.0
 Encoding=UTF-8
-Name=Bibus bibliographic database
-Name[fr]=Base de donnée bibliographique Bibus
-Comment=Bibus bibliographic database
+Name=Bibus
+Name[en]=Bibus
+Name[de]=Bibus
+Name[fr]=Bibus
+Name[hu]=Bibus
+GenericName[en]=Bibus bibliographic database
+GenericName[de]=Bibliographische Datenbank Bibus
+GenericName[fr]=Base de donnée bibliographique Bibus
+GenericName[hu]=Bibus referencia kezelő
+Comment=manage references and citations in your publications
+Comment[en]=manage references and citations in your publications
+Comment[de]=verwalte Literaturverweise und Zitate in Publikationen 
+Comment[fr]=Logiciel de gestion bibliographique
+Comment[hu]=közlemények citációját és referenciáit kezeli
 Categories=GNOME;Application;WordProcessor;Office;
 Type=Application
-GenericName[fr_FR]=Bibliographic database
+Terminal=false
+Exec=bibus
+Icon=bibus.png
 StartupNotify=true
diff -Naur bibus1.4.3/Setup/bibus.sh bibus1.4.3.new/Setup/bibus.sh
--- bibus1.4.3/Setup/bibus.sh	2008-05-20 16:16:41.000000000 -0400
+++ bibus1.4.3.new/Setup/bibus.sh	2008-05-22 09:08:56.000000000 -0400
@@ -1,4 +1,4 @@
 #!/bin/sh
 export LD_LIBRARY_PATH=/usr/lib/openoffice/program
 export PYTHONPATH=/usr/lib/openoffice/program
-exec /usr/bin/python /usr/local/share/bibus/bibus.py
+exec /usr/bin/python /usr/share/bibus/bibus.py
