diff -Naur bibus-1.4.3.1/Makefile bibus-1.4.3.1.new/Makefile
--- bibus-1.4.3.1/Makefile	2008-10-28 13:46:10.000000000 -0400
+++ bibus-1.4.3.1.new/Makefile	2008-11-12 09:55:51.000000000 -0500
@@ -2,14 +2,14 @@
 DESTDIR = /usr/local
 python = /usr/bin/python
 oopath = /usr/lib/openoffice/program
-ooure = /usr/lib/openoffice.org/basis-link/ure-link/lib
+ooure = /usr/lib/openoffice/basis-link/ure-link/lib
 oobasis = /usr/lib/openoffice/basis-link/program
 #
-prefix = $(DESTDIR)
+prefix = $(DESTDIR)/usr
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
 datadir = $(prefix)/share
-sysconfdir = /etc
+sysconfdir = $(DESTDIR)/etc
 mandir = $(prefix)/share/man
 man1dir = $(mandir)/man1
 
@@ -44,8 +44,8 @@
 	chmod 755 $(datadir)/bibus/bibusStart.py
 
 	# basic doc files
-	install -m755 -d $(datadir)/doc/bibus
-	install -m644 Docs/*.txt Docs/CHANGELOG Docs/copying $(datadir)/doc/bibus
+	install -m755 -d $(datadir)/doc/${PF}
+	install -m644 Docs/*.txt Docs/CHANGELOG Docs/copying $(datadir)/doc/${PF}
 
 	# generating bibus.cfg file
 	echo '[PATH]' > $(datadir)/bibus/bibus.cfg
@@ -53,16 +53,16 @@
 	echo 'oopath = $(oopath)' >> $(datadir)/bibus/bibus.cfg
 	echo 'ooure = $(ooure)' >> $(datadir)/bibus/bibus.cfg
 	echo 'oobasis = $(oobasis)' >> $(datadir)/bibus/bibus.cfg
-	echo 'docdir = $(datadir)/doc/bibus/html' >> $(datadir)/bibus/bibus.cfg
-	echo 'licence = $(datadir)/doc/bibus/copying' >> $(datadir)/bibus/bibus.cfg
-	echo 'localedir = $(datadir)/locale' >> $(datadir)/bibus/bibus.cfg
-	echo 'systemconf = $(sysconfdir)/bibus.config' >> $(datadir)/bibus/bibus.cfg
+	echo 'docdir = /usr/share/doc/${PF}/html' >> $(datadir)/bibus/bibus.cfg
+	echo 'licence = /usr/share/doc/${PF}/copying' >> $(datadir)/bibus/bibus.cfg
+	echo 'localedir = /usr/share/locale' >> $(datadir)/bibus/bibus.cfg
+	echo 'systemconf = /etc/bibus.config' >> $(datadir)/bibus/bibus.cfg
 endef
 
 define install-doc-en
 	# copying docs in from Docs/html/en/ for bibus-doc-en
 	cd Docs;\
-	find html/en/* -type f -exec install -m644 '{}' -D $(datadir)/doc/bibus/'{}' ';'
+	find html/en/* -type f -exec install -m644 '{}' -D $(datadir)/doc/${PF}/'{}' ';'
 endef
 
 define compile
@@ -71,6 +71,8 @@
 	$(python) -O -c "import compileall ; compileall.compile_dir('$(datadir)/bibus')"
 endef
 
+all:
+
 install-files:
 	$(install-files)
 
@@ -84,13 +86,13 @@
 
 	# write uninstaller in $(datadir)/bibus/Setup/uninstall.sh
 		echo "#!/bin/sh" > $(datadir)/bibus/Setup/uninstall.sh
-		echo "rm -rf $(datadir)/bibus" >> $(datadir)/bibus/Setup/uninstall.sh
-		echo "rm $(bindir)/bibus" >> $(datadir)/bibus/Setup/uninstall.sh
-		echo "find $(datadir)/locale -name bibus.mo -exec rm -f {} \;" >> $(datadir)/bibus/Setup/uninstall.sh
-		echo "rm $(sysconfdir)/bibus.config" >> $(datadir)/bibus/Setup/uninstall.sh
-		echo "rm $(man1dir)/bibus.1" >> $(datadir)/bibus/Setup/uninstall.sh
-		echo "rm -rf $(datadir)/doc/bibus" >> $(datadir)/bibus/Setup/uninstall.sh
-		echo "rm $(datadir)/applications/bibus.desktop" >> $(datadir)/bibus/Setup/uninstall.sh
-		echo "rm $(datadir)/icons/hicolor/48x48/apps/bibus.png" >> $(datadir)/bibus/Setup/uninstall.sh
+		echo "rm -rf /usr/share/bibus" >> $(datadir)/bibus/Setup/uninstall.sh
+		echo "rm /usr/share/bibus" >> $(datadir)/bibus/Setup/uninstall.sh
+		echo "find /usr/share/locale -name bibus.mo -exec rm -f {} \;" >> $(datadir)/bibus/Setup/uninstall.sh
+		echo "rm /etc/bibus.config" >> $(datadir)/bibus/Setup/uninstall.sh
+		echo "rm /usr/share/man/man1/bibus.1" >> $(datadir)/bibus/Setup/uninstall.sh
+		echo "rm -rf /usr/share/doc/${PF}" >> $(datadir)/bibus/Setup/uninstall.sh
+		echo "rm /usr/share/applications/bibus.desktop" >> $(datadir)/bibus/Setup/uninstall.sh
+		echo "rm /usr/share/icons/hicolor/48x48/apps/bibus.png" >> $(datadir)/bibus/Setup/uninstall.sh
 	chmod 744 $(datadir)/bibus/Setup/uninstall.sh
 	# end uninstaller
diff -Naur bibus-1.4.3.1/Setup/bibus.cfg bibus-1.4.3.1.new/Setup/bibus.cfg
--- bibus-1.4.3.1/Setup/bibus.cfg	2005-10-01 09:10:41.000000000 -0400
+++ bibus-1.4.3.1.new/Setup/bibus.cfg	2008-11-12 09:15:08.000000000 -0500
@@ -1,7 +1,7 @@
 [PATH]
 python = /usr/bin/python
 oopath = /usr/lib/openoffice/program
-docdir = /usr/local/share/doc/bibus/html
-licence = /usr/local/share/doc/bibus/copying
-localedir = /usr/local/share/locale
-systemconf = /usr/local/etc/bibus.config
+docdir = /usr/share/doc/${P}/html
+licence = /usr/share/doc/${P}/copying
+localedir = /usr/share/locale
+systemconf = /etc/bibus.config
diff -Naur bibus-1.4.3.1/Setup/bibus.sh bibus-1.4.3.1.new/Setup/bibus.sh
--- bibus-1.4.3.1/Setup/bibus.sh	2005-11-30 18:21:40.000000000 -0500
+++ bibus-1.4.3.1.new/Setup/bibus.sh	2008-11-12 09:15:08.000000000 -0500
@@ -1,4 +1,4 @@
 #!/bin/sh
-export LD_LIBRARY_PATH=/usr/lib/openoffice/program
-export PYTHONPATH=/usr/lib/openoffice/program
-exec /usr/bin/python /usr/local/share/bibus/bibus.py
+export LD_LIBRARY_PATH=/usr/lib/openoffice/program:/usr/lib/openoffice/basis-link/program:/usr/lib/openoffice/basis-link/ure-link/lib
+export PYTHONPATH=/usr/lib/openoffice/program:/usr/lib/openoffice/basis-link/program:/usr/lib/openoffice/basis-link/ure-link/lib
+exec /usr/bin/python /usr/share/bibus/bibus.py
