--- poppler-0.4.3/utils/HtmlOutputDev.cc.orig	2006-01-01 15:15:05.000000000 +0000
+++ poppler-0.4.3/utils/HtmlOutputDev.cc	2006-01-01 15:15:30.000000000 +0000
@@ -27,7 +27,9 @@
 #include "config.h"
 #include "Error.h"
 #include "GfxState.h"
+#ifdef ENABLE_LIBJPEG
 #include "DCTStream.h"
+#endif
 #include "GlobalParams.h"
 #include "HtmlOutputDev.h"
 #include "HtmlFonts.h"
--- poppler-0.4.3/utils/ImageOutputDev.cc.orig	2006-01-01 15:15:10.000000000 +0000
+++ poppler-0.4.3/utils/ImageOutputDev.cc	2006-01-01 15:16:09.000000000 +0000
@@ -22,7 +22,9 @@
 #include "GfxState.h"
 #include "Object.h"
 #include "Stream.h"
+#ifdef ENABLE_LIBJPEG
 #include "DCTStream.h"
+#endif
 #include "ImageOutputDev.h"
 
 ImageOutputDev::ImageOutputDev(char *fileRootA, GBool dumpJPEGA) {
