--- ksensors-0.7.3/src/Makefile.am.orig	2009-02-10 03:01:36.000000000 +0100
+++ ksensors-0.7.3/src/Makefile.am	2009-02-10 03:01:54.000000000 +0100
@@ -46,7 +46,7 @@
 ksensors_LDADD = $(LIB_KDEUI) $(LIB_KDECORE) -lsensors -lm
 
 # this is where the desktop file will go 
-shelldesktopdir   = $(kde_appsdir)/Utilities
+shelldesktopdir   = $(xdg_appsdir)
 shelldesktop_DATA = ksensors.desktop
 
 # this is where the shell's XML-GUI resource file goes
--- ksensors-0.7.3/src/ksensors.desktop.orig	2009-02-10 03:02:15.000000000 +0100
+++ ksensors-0.7.3/src/ksensors.desktop	2009-02-10 03:03:16.000000000 +0100
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=KSensors
 Exec=ksensors
 Icon=ksensors
