--- ksensors/lmsensordockpanel.cpp.orig	2002-12-26 12:48:59.000000000 +1100
+++ ksensors/lmsensordockpanel.cpp	2002-12-26 12:53:06.000000000 +1100
@@ -28,6 +28,7 @@
 LMSensorDockPanel::LMSensorDockPanel(Sensor *newSensor, const char *name) : KSystemTray( 0, name )
 {
  strValue   = "";
+ setPixmap(BarIcon("ksensorsdocked"));
  if(newSensor) {
     sensor= newSensor;
     updateConfig();
@@ -35,7 +36,7 @@
     connect( newSensor, SIGNAL(configChanged())      , this, SLOT(updateConfig()) );
  } else {
     sensor= 0;
-    setPixmap(BarIcon("ksensorsdocked"));
+    //setPixmap(BarIcon("ksensorsdocked"));
  }
 }
 
