diff -ruN 111/gDeskCal-0.57.1/code/CalMediator.py gDeskCal-0.57.1/code/CalMediator.py
--- 111/gDeskCal-0.57.1/code/CalMediator.py	2004-03-15 23:46:05.000000000 +0100
+++ gDeskCal-0.57.1/code/CalMediator.py	2005-05-07 00:34:45.000000000 +0200
@@ -326,4 +326,4 @@
     #
     def __cmd_evolution(self):
 
-        os.system("evolution evolution:/local/Calendar &")
+        os.system("evolution evolution -c calendar &")
diff -ruN 111/gDeskCal-0.57.1/code/values.py gDeskCal-0.57.1/code/values.py
--- 111/gDeskCal-0.57.1/code/values.py	2004-03-15 23:46:05.000000000 +0100
+++ gDeskCal-0.57.1/code/values.py	2005-05-07 00:26:48.000000000 +0200
@@ -15,7 +15,7 @@
 # files
 SKINFILE = "skin.xml"
 CONFIGFILE = "config.xml"
-EVOLUTION_CALENDAR = "~/evolution/local/Calendar/calendar.ics"
+EVOLUTION_CALENDAR = "~/.evolution/calendar/local/system/calendar.ics"
 MY_CALENDAR = "calendar.ics"
 
 # config data files
