--- bin/catalina.sh.orig	2003-03-22 00:34:17.000000000 -0500
+++ bin/catalina.sh	2003-03-22 02:14:52.000000000 -0500
@@ -35,12 +35,6 @@
 # $Id: gentoo.diff,v 1.1 2003/03/22 10:42:02 absinthe Exp $
 # -----------------------------------------------------------------------------
 
-# OS specific support.  $var _must_ be set to either true or false.
-cygwin=false
-case "`uname`" in
-CYGWIN*) cygwin=true;;
-esac
-
 # resolve links - $0 may be a softlink
 PRG="$0"
 
@@ -56,20 +50,12 @@
 
 # Get standard environment variables
 PRGDIR=`dirname "$PRG"`
+source /etc/conf.d/tomcat
 CATALINA_HOME=`cd "$PRGDIR/.." ; pwd`
 if [ -r "$CATALINA_HOME"/bin/setenv.sh ]; then
   . "$CATALINA_HOME"/bin/setenv.sh
 fi
 
-# For Cygwin, ensure paths are in UNIX format before anything is touched
-if $cygwin; then
-  [ -n "$JAVA_HOME" ] && JAVA_HOME=`cygpath --unix "$JAVA_HOME"`
-  [ -n "$CATALINA_HOME" ] && CATALINA_HOME=`cygpath --unix "$CATALINA_HOME"`
-  [ -n "$CATALINA_BASE" ] && CATALINA_BASE=`cygpath --unix "$CATALINA_BASE"`
-  [ -n "$CLASSPATH" ] && CLASSPATH=`cygpath --path --unix "$CLASSPATH"`
-  [ -n "$JSSE_HOME" ] && JSSE_HOME=`cygpath --path --unix "$JSSE_HOME"`
-fi
-
 # Get standard Java environment variables
 if [ -r "$CATALINA_HOME"/bin/setclasspath.sh ]; then
   BASEDIR="$CATALINA_HOME"
@@ -95,16 +81,6 @@
   CATALINA_TMPDIR="$CATALINA_BASE"/temp
 fi
 
-# For Cygwin, switch paths to Windows format before running java
-if $cygwin; then
-  JAVA_HOME=`cygpath --path --windows "$JAVA_HOME"`
-  CATALINA_HOME=`cygpath --path --windows "$CATALINA_HOME"`
-  CATALINA_BASE=`cygpath --path --windows "$CATALINA_BASE"`
-  CATALINA_TMPDIR=`cygpath --path --windows "$CATALINA_TMPDIR"`
-  CLASSPATH=`cygpath --path --windows "$CLASSPATH"`
-  JSSE_HOME=`cygpath --path --windows "$JSSE_HOME"`
-fi
-
 # ----- Execute The Requested Command -----------------------------------------
 
 echo "Using CATALINA_BASE:   $CATALINA_BASE"
@@ -188,7 +164,7 @@
 elif [ "$1" = "start" ] ; then
 
   shift
-  touch "$CATALINA_BASE"/logs/catalina.out
+  touch "$CATALINA_OUT"
   if [ "$1" = "-security" ] ; then
     echo "Using Security Manager"
     shift
@@ -200,7 +176,7 @@
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
       org.apache.catalina.startup.Bootstrap "$@" start \
-      >> "$CATALINA_BASE"/logs/catalina.out 2>&1 &
+      >> "$CATALINA_OUT" 2>&1 &
   else
     "$_RUNJAVA" $JAVA_OPTS $CATALINA_OPTS \
       -Djava.endorsed.dirs="$JAVA_ENDORSED_DIRS" -classpath "$CLASSPATH" \
@@ -208,7 +184,7 @@
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
       org.apache.catalina.startup.Bootstrap "$@" start \
-      >> "$CATALINA_BASE"/logs/catalina.out 2>&1 &
+      >> "$CATALINA_OUT" 2>&1 &
   fi
 
 elif [ "$1" = "stop" ] ; then
--- bin/setclasspath.sh.orig	2003-03-22 00:42:16.000000000 -0500
+++ bin/setclasspath.sh	2003-03-22 05:14:10.000000000 -0500
@@ -6,13 +6,15 @@
 
 # Make sure prerequisite environment variables are set
 if [ -z "$JAVA_HOME" ]; then
-  echo "The JAVA_HOME environment variable is not defined"
-  echo "This environment variable is needed to run this program"
+  echo "The JAVA_HOME environment variable is not defined!"
+  echo "This environment variable is needed to run this program."
+  echo "See the /etc/conf.d/tomcat file for details."
   exit 1
 fi
 if [ ! -r "$JAVA_HOME"/bin/java -o ! -r "$JAVA_HOME"/bin/jdb -o ! -r "$JAVA_HOME"/bin/javac ]; then
-  echo "The JAVA_HOME environment variable is not defined correctly"
-  echo "This environment variable is needed to run this program"
+  echo "The JAVA_HOME environment variable is not defined correctly!"
+  echo "This environment variable is needed to run this program."
+  echo "See the /etc/conf.d/tomcat file for details."
   exit 1
 fi
 if [ -z "$BASEDIR" ]; then
@@ -27,7 +29,7 @@
 fi
 
 # Set the default -Djava.endorsed.dirs argument
-JAVA_ENDORSED_DIRS="$BASEDIR"/bin:"$BASEDIR"/common/endorsed
+JAVA_ENDORSED_DIRS="$TOMCAT_LIBDIR"
 
 # Set standard CLASSPATH
 CLASSPATH="$JAVA_HOME"/lib/tools.jar
