diff -ur a/macros/ffmpeg.m4 b/macros/ffmpeg.m4
--- a/macros/ffmpeg.m4	2011-04-09 17:26:26.817016067 +0200
+++ b/macros/ffmpeg.m4	2011-04-09 17:51:04.142016943 +0200
@@ -96,6 +96,9 @@
         if test -f ${ffmpeg_top_incl}/${i}/avcodec.h; then
           ac_cv_path_ffmpeg_incl="-I`(cd ${ffmpeg_top_incl}/${i}; pwd)`"
           avcodec_h=${ffmpeg_top_incl}/${i}/avcodec.h
+          if test -f ${ffmpeg_top_incl}/${i}/version.h; then
+            avcodec_version_h=${ffmpeg_top_incl}/${i}/version.h
+          fi
           break
         fi
       done
