From f05022aca2c14022a942d36a22764fae4b8f00c0 Mon Sep 17 00:00:00 2001
From: Brad Smith <brad@comstyle.com>
Date: Sun, 03 Apr 2011 10:56:27 +0000
Subject: Fix newer FFmpeg detection with pkg-config.

---
diff --git a/macros/ffmpeg.m4 b/macros/ffmpeg.m4
index 2a3d3fb..42c7555 100644
--- a/macros/ffmpeg.m4
+++ b/macros/ffmpeg.m4
@@ -70,6 +70,9 @@ AC_DEFUN([GNASH_PATH_FFMPEG],
           if test -f ${ffmpeg_top_incl}/${i}/avcodec.h; then
             ac_cv_path_ffmpeg_incl="-I`(cd ${ffmpeg_top_incl}; pwd)`"
             avcodec_h="${ffmpeg_top_incl}/${i}/avcodec.h"
+            if test -f ${ffmpeg_top_incl}/${i}/version.h; then
+              avcodec_version_h=${ffmpeg_top_incl}/${i}/version.h
+            fi
             break
           fi
         done
--
cgit v0.8.3.4
