diff -ur cljl.orig/jpeg.lisp cljl/jpeg.lisp
--- cljl.orig/jpeg.lisp	2005-03-28 12:59:29.000000000 -0600
+++ cljl/jpeg.lisp	2005-03-28 13:00:44.000000000 -0600
@@ -59,11 +59,13 @@
 ;;; to the Independent JPEG Group - colorspace conversion and DCT algorithms were adopted from their sources;
 ;;; to Jeff Dalton for his wise paper "Common Lisp Pitfalls".
 
-(defpackage #:jpeg (:use #:common-lisp))
-(in-package #:jpeg)
+(defpackage #:jpeg
+  (:use #:common-lisp)
+  (:export #:encode-image
+	   #:decode-image
+	   #:jpeg-to-bmp))
 
-(eval-when (compile)
-  (export '(encode-image decode-image jpeg-to-bmp)))
+(in-package #:jpeg)
 
 (declaim (inline csize write-stuffed quantize get-average zigzag encode-block llm-dct descale crunch colorspace-convert subsample
                  inverse-llm-dct dequantize upsample extend recieve decode-ac decode-dc decode-block izigzag write-bits))
Only in cljl: jpeg.lisp.~1.8.~
