diff -ur otp_src_R10B-6.orig/lib/tools/emacs/erlang.el otp_src_R10B-6/lib/tools/emacs/erlang.el
--- otp_src_R10B-6.orig/lib/tools/emacs/erlang.el	2004-09-30 02:05:50.000000000 -0500
+++ otp_src_R10B-6/lib/tools/emacs/erlang.el	2005-07-18 13:52:13.000000000 -0500
@@ -2518,7 +2518,7 @@
 
 (defun erlang-man-make-menu-item (file)
   "Create a menu item containing the name of the man page."
-  (and (string-match ".*/\\([^/]+\\)\\.[^.]$" file)
+  (and (string-match ".*/\\([^/]+\\)\\.[0-9]erl\\.gz$" file)
        (let ((page (substring file (match-beginning 1) (match-end 1))))
 	 (list (capitalize page)
 	       (list 'lambda '()
@@ -2529,7 +2529,7 @@
 
 (defun erlang-man-get-files (dir)
   "Return files in directory DIR."
-  (directory-files dir t ".*\\.[0-9]\\'"))
+  (directory-files dir t ".*\\.[0-9]erl\\.gz\\'"))
 
 
 (defun erlang-man-module (&optional module)
@@ -2690,7 +2690,7 @@
     (error nil))
   (if file
       (let ((process-environment (copy-sequence process-environment)))
-	(if (string-match "\\(.*\\)/man[^/]*/\\([^/]+\\)\\.[^.]$" file)
+	(if (string-match "\\(.*\\)/man[^/]*/\\([^/]+\\)\\.[0-9]erl.gz$" file)
 	    (let ((dir (substring file (match-beginning 1) (match-end 1)))
 		  (page (substring file (match-beginning 2) (match-end 2))))
 	      (if (fboundp 'setenv)
Only in otp_src_R10B-6/lib/tools/emacs: erlang.el~
