--- Source_Files/Misc/sdl_widgets.h.old	2006-05-05 22:25:11.000000000 +0200
+++ Source_Files/Misc/sdl_widgets.h	2006-05-05 22:25:55.000000000 +0200
@@ -944,7 +944,7 @@
 	virtual ~SelectorWidget () {}
 
 protected:
-	SelectorWidget::SelectorWidget (widget* in_widget)
+	SelectorWidget (widget* in_widget)
 		: SDLWidgetWidget (in_widget)
 		, m_callback (NULL)
 		{}
@@ -955,7 +955,7 @@
 class PopupSelectorWidget : public SelectorWidget
 {
 public:
-	PopupSelectorWidget::PopupSelectorWidget (w_select_popup* select_popup_w)
+	PopupSelectorWidget (w_select_popup* select_popup_w)
 		: SelectorWidget (select_popup_w)
 		, m_select_popup (select_popup_w)
 		{ select_popup_w->set_popup_callback (boost::bind(&PopupSelectorWidget::massage_callback, this, _1), NULL); }
@@ -974,7 +974,7 @@
 class SelectSelectorWidget : public SelectorWidget
 {
 public:
-	SelectSelectorWidget::SelectSelectorWidget (w_select* select_w)
+	SelectSelectorWidget (w_select* select_w)
 		: SelectorWidget (select_w)
 		, m_select (select_w)
 		{ m_select->set_selection_changed_callback (boost::bind(&SelectSelectorWidget::massage_callback, this, _1)); }
@@ -996,7 +996,7 @@
 class ColourSelectorWidget : public SelectSelectorWidget
 {
 public:
-	ColourSelectorWidget::ColourSelectorWidget (w_player_color* player_color_w)
+	ColourSelectorWidget (w_player_color* player_color_w)
 		: SelectSelectorWidget (player_color_w) {}
 
 	// We ignore the labels and use swatches of colour instead
@@ -1028,7 +1028,7 @@
 class StaticTextWidget : public SDLWidgetWidget
 {
 public:
-	StaticTextWidget::StaticTextWidget (w_static_text* static_text_w)
+	StaticTextWidget (w_static_text* static_text_w)
 		: SDLWidgetWidget (static_text_w)
 		, m_static_text (static_text_w)
 		{}
--- Source_Files/TCPMess/CommunicationsChannel.h.old	2006-05-05 22:34:47.000000000 +0200
+++ Source_Files/TCPMess/CommunicationsChannel.h	2006-05-05 22:34:58.000000000 +0200
@@ -95,7 +95,7 @@
 	// Calls back message handlers (if appropriate)
 	void		dispatchIncomingMessages();
 
-	bool		CommunicationsChannel::isMessageAvailable();
+	bool		isMessageAvailable();
 
 	// Call does not return unless (1) times out (NULL); (2) disconnected (NULL); or
 	// (3) some message received (pointer to inflated message object).
--- Source_Files/Network/network_dialogs_sdl.cpp.old	2006-05-05 22:38:31.000000000 +0200
+++ Source_Files/Network/network_dialogs_sdl.cpp	2006-05-05 22:39:01.000000000 +0200
@@ -838,7 +838,7 @@
 class SdlGatherDialog : public GatherDialog
 {
 public:
-	SdlGatherDialog::SdlGatherDialog()
+	SdlGatherDialog()
 	{
 		m_dialog.add(new w_static_text("GATHER NETWORK GAME", TITLE_FONT, TITLE_COLOR));
 	
@@ -921,7 +921,7 @@
 class SdlJoinDialog : public JoinDialog
 {
 public:
-	SdlJoinDialog::SdlJoinDialog()
+	SdlJoinDialog()
 	{
 		m_dialog.add(new w_static_text("JOIN NETWORK GAME", TITLE_FONT, TITLE_COLOR));
 		m_dialog.add(new w_spacer());
