--- configure~	2004-08-22 20:10:52.000000000 +0400
+++ configure	2004-08-22 20:11:10.000000000 +0400
@@ -6388,7 +6388,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lgpgme  $LIBS"
+LIBS="-lgpgme3  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -6438,7 +6438,7 @@
 echo "${ECHO_T}$ac_cv_lib_gpgme_gpgme_check_version" >&6
 if test $ac_cv_lib_gpgme_gpgme_check_version = yes; then
 
-                    LIBS="$LIBS -lgpgme"
+                    LIBS="$LIBS -lgpgme3"
 
 cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBGPGME 1
