--- blackbox-0.65.0/src/Window.cc	2002-09-17 11:23:35.000000000 -0400
+++ Window.cc	2003-10-09 13:09:22.993741976 -0400
@@ -55,6 +55,7 @@
 #include "Window.hh"
 #include "Windowmenu.hh"
 #include "Workspace.hh"
+#include <assert.h>
 
 
 /*
@@ -2050,7 +2051,7 @@
     if (decorations & Decor_Titlebar) {
       if (functions & Func_Close)   // close button is controlled by function
         decorations |= Decor_Close; // not decor type
-    } else { 
+    } else {
       if (flags.shaded) // we can not be shaded if we lack a titlebar
         shade();
     }
@@ -2986,7 +2987,7 @@
   // do not leave a shaded window as an icon unless it was an icon
   if (flags.shaded && ! flags.iconic)
     current_state = NormalState;
-  
+
   setState(current_state);
 
   restoreGravity(client.rect);
@@ -3099,7 +3100,7 @@
     if (decorations & Decor_Titlebar) {
       if (functions & Func_Close)   // close button is controlled by function
         decorations |= Decor_Close; // not decor type
-    } else { 
+    } else {
       if (flags.shaded) // we can not be shaded if we lack a titlebar
         shade();
     }
