--- a/m4/dpkg-arch.m4	2010-01-08 18:23:06.000000000 +0100
+++ b/m4/dpkg-arch.m4	2010-02-05 08:13:09.000000000 +0100
@@ -5,7 +5,7 @@
 AC_DEFUN([_DPKG_ARCHITECTURE], [
 AC_REQUIRE([AC_CANONICAL_HOST])dnl
 $2=$(cd $srcdir/scripts; \
-    PERL5LIB=$(pwd) ./dpkg-architecture.pl -t$host -q$1 2>/dev/null)
+    pkgdatadir=.. PERL5LIB=$(pwd) ./dpkg-architecture.pl -t$host -q$1 2>/dev/null)
 ])# _DPKG_ARCHITECURE
 
 # DPKG_CPU_TYPE
--- a/scripts/Dpkg/Arch.pm	2010-01-08 09:00:34.000000000 +0100
+++ b/scripts/Dpkg/Arch.pm	2010-02-05 08:49:42.000000000 +0100
@@ -47,8 +47,7 @@
 	return $build_arch if defined $build_arch;
 
 	my $build_arch = `dpkg --print-architecture`;
-	# FIXME: Handle bootstrapping
-	syserr("dpkg --print-architecture failed") if $? >> 8;
+	if ($? >> 8) { $build_arch = get_raw_host_arch(); }
 
 	chomp $build_arch;
 	return $build_arch;
