--- main/Makefile.in	2003-05-11 18:23:18.000000000 +0200
+++ main.new/Makefile.in	2003-05-11 18:23:42.000000000 +0200
@@ -1,10 +1,10 @@
 VPATH		= @srcdir@
 srcdir		= @srcdir@
-top_srcdir	= @top_srcdir@
 top_builddir	= ..
 
 default: all
 include ../Makefile.conf
+top_srcdir	= @top_srcdir@
 
 CXX_SOURCES	= basecmds.cc baselist.cc basetop.cc bindings.cc curkeys.cc \
 		  main.cc methkeys.cc methlist.cc method.cc methparse.cc \
--- dselect/curkeys.cc	2003-05-11 18:44:26.000000000 +0200
+++ dselect/curkeys.cc.1	2003-05-11 18:43:57.000000000 +0200
@@ -19,6 +19,8 @@
  * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#define NCURSES_WGETCH_EVENTS
+
 extern "C" {
 #include <config.h>
 #include <dpkg.h>
--- doc/Makefile.in	2003-05-11 18:47:18.000000000 +0200
+++ doc.new/Makefile.in	2003-05-11 18:47:31.000000000 +0200
@@ -1,10 +1,10 @@
 VPATH		= @srcdir@
 srcdir		= @srcdir@
-top_srcdir	= @top_srcdir@
 
 default: all
 
 include ../Makefile.conf
+top_srcdir	= @top_srcdir@
 
 ifeq (@USE_SGML_DOC@, true)
 all: internals.html/index.html
--- Makefile.in	2003-05-11 19:00:49.000000000 +0200
+++ Makefile.in.new	2003-05-11 19:00:40.000000000 +0200
@@ -10,7 +10,7 @@
 include Makefile.conf
 
 SUBDIRS		= include optlib lib dpkg-deb split scripts utils main \
-		  @DSELECTDIR@ po methods man doc
+		  @DSELECTDIR@ po methods man
 
 all: version.h all-recursive
 
--- Makefile.conf.in	2003-05-11 19:06:03.000000000 +0200
+++ Makefile.conf.in.new	2003-05-11 19:06:29.000000000 +0200
@@ -43,7 +43,7 @@
 subdir = $(patsubst $(top_srcdir)/%,%,$(srcdir))/
 endif
 PERL			= @PERL@
-mkinstalldirs		= $(SHELL) $(top_srcdir)/mkinstalldirs
+mkinstalldirs		= /usr/share/automake-1.5/mkinstalldirs
 LN			= @LN_S@
 INSTALL			= @INSTALL@
 INSTALL_PROGRAM		= @INSTALL_PROGRAM@
--- man/Makefile.in	2003-05-11 19:11:06.000000000 +0200
+++ man.new/Makefile.in	2003-05-11 19:11:18.000000000 +0200
@@ -1,11 +1,11 @@
 VPATH		= @srcdir@
 srcdir		= @srcdir@
-top_srcdir	= @top_srcdir@
 
 SUBDIRS		= de en fr ja pt_BR ru sv
 
 default: all
 include ../Makefile.conf
+top_srcdir	= @top_srcdir@
 
 clean: clean-recursive
 
