Index: configure.in
===================================================================
--- configure.in	(revision 55836)
+++ configure.in	(working copy)
@@ -49,7 +49,7 @@
 AC_SUBST(GTK_SHARP_CFLAGS)
 AC_SUBST(GTK_SHARP_LIBS)
 
-PKG_CHECK_MODULES(GSF, libgsf-1 >= 1.13 libgsf-1 < 1.14 libgsf-gnome-1 >= 1.13 libgsf-gnome-1 < 1.14)
+PKG_CHECK_MODULES(GSF, libgsf-1 >= 1.13 libgsf-1 < 1.14)
 AC_SUBST(GSF_CFLAGS)
 AC_SUBST(GSF_LIBS)
 gsf_prefix=`pkg-config --variable=prefix libgsf-1`
