diff -Naurp getstack.c.old getstack.c
--- getstack.c.old	2004-03-05 12:50:42.000000000 +0100
+++ getstack.c	2009-06-29 22:52:19.036289874 +0200
@@ -6,6 +6,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 
 int main( int argc, char *argv[] ){
 	char a;
diff -Naurp randbody.c.old randbody.c
--- randbody.c.old	2005-06-19 16:53:40.000000000 +0200
+++ randbody.c	2009-06-29 22:52:05.021047393 +0200
@@ -5,6 +5,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #define COUNT	(25)
 
diff -Naurp rettofunc1.c.old rettofunc1.c
--- rettofunc1.c.old	2005-05-29 18:02:41.000000000 +0200
+++ rettofunc1.c	2009-06-29 22:52:39.553037638 +0200
@@ -6,6 +6,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include "body.h"
 
 const char testname[] = "Return to function (strcpy)              ";
diff -Naurp rettofunc1x.c.old rettofunc1x.c
--- rettofunc1x.c.old	2005-05-29 18:02:47.000000000 +0200
+++ rettofunc1x.c	2009-06-29 22:52:45.869055602 +0200
@@ -6,6 +6,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include "body.h"
 
 const char testname[] = "Return to function (strcpy, RANDEXEC)    ";
