--- Makefile.am.orig	2008-12-28 22:57:11.000000000 -0500
+++ Makefile.am	2008-12-28 22:56:49.000000000 -0500
@@ -46,7 +46,7 @@
 	intltool-update
 
 installdir = ${coredir}
-install_SCRIPTS = ctrlinfo gdesklets gdesklets-daemon gdesklets-logview gdesklets-migration-tool gdesklets-shell
+install_SCRIPTS = ctrlinfo gdesklets gdesklets-daemon gdesklets-logview gdesklets-migration-tool gdesklets-shell test-control.py
 
 install-data-hook:
 	if test '!' -d $(DESTDIR)$(bindir); then \
