--- gDesklets-0.23/Sensors/FontSelector/__init__.py.orig	2003-10-26 14:19:55.885667016 +0800
+++ gDesklets-0.23/Sensors/FontSelector/__init__.py	2003-10-26 14:20:03.978436728 +0800
@@ -29,7 +29,7 @@
         Sensor.__init__(self)
         self._watch_config(self.config_watcher)
         # setup configuration entries
-        if (len(fontargs) == 1 and "  " in fontargs[0]):
+        if (len(fontargs) == 1 and " " in fontargs[0]):
             fontargs = \
                (fontargs[0].replace("   ", ",").replace("  ", ",")).split(",")
             print "Using spaces to separate font entries is deprecated."
