diff -Nru polipo-1.0.0.orig/Makefile polipo-1.0.0/Makefile
--- polipo-1.0.0.orig/Makefile	2007-03-06 21:29:05.000000000 +0200
+++ polipo-1.0.0/Makefile	2007-06-17 09:43:17.000000000 +0300
@@ -97,11 +97,6 @@
 	cp -f localindex.html $(TARGET)$(LOCAL_ROOT)/index.html
 
 install.man: all
-	mkdir -p $(TARGET)$(MANDIR)/man1
-	mkdir -p $(TARGET)$(INFODIR)
-	cp -f polipo.man $(TARGET)$(MANDIR)/man1/polipo.1
-	cp polipo.info $(TARGET)$(INFODIR)/
-	install-info --info-dir=$(INFODIR) polipo.info
 
 
 polipo.info: polipo.texi
diff -Nru polipo-1.0.0.orig/polipo.h polipo-1.0.0/polipo.h
--- polipo-1.0.0.orig/polipo.h	2007-03-06 21:29:05.000000000 +0200
+++ polipo-1.0.0/polipo.h	2007-06-17 09:43:17.000000000 +0300
@@ -65,7 +65,9 @@
 #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
 /* nothing */
 #elif defined(__GNUC__)
+#ifndef inline
 #define inline __inline
+#endif
 #if  (__GNUC__ >= 3)
 #define restrict __restrict
 #else
