Index: xl/covers.py
===================================================================
--- xl/covers.py	(revision 2372)
+++ xl/covers.py	(working copy)
@@ -42,7 +42,8 @@
 QUERY = "/onca/xml3?t=webservices-20&dev-t=%s&mode=music&type=lite&" % (KEY) + \
     "locale={locale}&page=1&f=xml&KeywordSearch="
 IMAGE_PATTERN = re.compile(
-    r"http://(images(?:-\w\w)?\.amazon\.com)(/images/.*?LZ+\.jpg)")
+    r"<ImageUrlMedium>http://(\w+\.images-amazon\.com)"
+    "(/images/.*?\.jpg)</ImageUrlMedium>", re.DOTALL)
 
 """
     Fetches album covers from Amazon.com
