diff -ur exaile-0.3.0.orig/Makefile exaile-0.3.0/Makefile
--- exaile-0.3.0.orig/Makefile	2009-08-27 22:24:25.000000000 +0300
+++ exaile-0.3.0/Makefile	2009-08-27 22:25:14.000000000 +0300
@@ -53,7 +53,7 @@
 	rm -f $(DESTDIR)$(PREFIX)/share/pixmaps/exaile.png
 	make uninstall -C plugins
 
-install: install-target locale install-locale
+install: install-target install-locale
 
 install_no_locale: install-target
 
