diff -urNp truecrypt-5.1a-source.org/Main/Main.make truecrypt-5.1a-source/Main/Main.make
--- truecrypt-5.1a-source.org/Main/Main.make	2008-03-11 20:48:42.000000000 +0200
+++ truecrypt-5.1a-source/Main/Main.make	2008-03-21 18:38:43.000000000 +0200
@@ -70,21 +70,25 @@ CXXFLAGS += -I$(BASE_DIR)/Main
 
 #------ wxWidgets configuration ------
 
+WX_CONFIG ?= $(WX_BUILD_DIR)/wx-config
+WX_CONFIG_EXTRA ?= --static
+
 ifdef TC_NO_GUI
 WX_CONFIG_LIBS := base
 else
 WX_CONFIG_LIBS := adv,core,base
 endif
+WX_CONFIG_LIBS =
 
 ifeq "$(TC_BUILD_CONFIG)" "Release"
 
-CXXFLAGS += $(shell $(WX_BUILD_DIR)/wx-config --unicode --static --cxxflags)
-WX_LIBS = $(shell $(WX_BUILD_DIR)/wx-config --unicode --static --libs $(WX_CONFIG_LIBS))
+CXXFLAGS += $(shell $(WX_CONFIG) --unicode $(WX_CONFIG_EXTRA) --cxxflags)
+WX_LIBS = $(shell $(WX_CONFIG) --unicode $(WX_CONFIG_EXTRA) --libs $(WX_CONFIG_LIBS))
 
 else
 
-CXXFLAGS += $(shell $(WX_BUILD_DIR)/wx-config --debug --unicode --static --cxxflags)
-WX_LIBS = $(shell $(WX_BUILD_DIR)/wx-config --debug --unicode --static --libs $(WX_CONFIG_LIBS))
+CXXFLAGS += $(shell $(WX_CONFIG) --debug --unicode $(WX_CONFIG_EXTRA) --cxxflags)
+WX_LIBS = $(shell $(WX_CONFIG) --debug --unicode $(WX_CONFIG_EXTRA) --libs $(WX_CONFIG_LIBS))
 
 endif
 
