diff -Nru vmailmgr-tools-0.2.orig/vcheckquota.c vmailmgr-tools-0.2/vcheckquota.c
--- vmailmgr-tools-0.2.orig/vcheckquota.c	2002-05-11 00:00:37.000000000 +0300
+++ vmailmgr-tools-0.2/vcheckquota.c	2005-12-26 13:33:02.084556000 +0200
@@ -33,9 +33,9 @@
 const char program[] = "vcheckquota";
 const int msg_show_pid = 0;
 const char cli_help_prefix[] = "vmailmgr quota enforcement program\n";
-const char cli_help_suffix[] = "
-Warning: the soft-message is linked into the users maildir once for each
-message that is received while the account is over its soft quota.  This may
+const char cli_help_suffix[] = "\
+Warning: the soft-message is linked into the users maildir once for each\
+message that is received while the account is over its soft quota.  This may\
 result in multiple warning messages.\n";
 const char cli_args_usage[] = "";
 const int cli_args_min = 0;
