=== modified file 'bin/repoman'
--- bin/repoman	
+++ bin/repoman	
@@ -1574,7 +1574,12 @@
 		if "PORTAGE_GPG_KEY" not in repoman_settings:
 			raise portage_exception.MissingParameter("PORTAGE_GPG_KEY is unset!")
 		if "PORTAGE_GPG_DIR" not in repoman_settings:
-			raise portage_exception.MissingParameter("PORTAGE_GPG_DIR is unset!")
+			if os.environ.has_key("HOME"):
+				repoman_settings["PORTAGE_GPG_DIR"] = os.path.join(os.environ["HOME"], ".gnupg")
+				if quiet < 1:
+					print "Automatically setting PORTAGE_GPG_DIR to",repoman_settings["PORTAGE_GPG_DIR"]
+			else:
+				raise portage_exception.MissingParameter("PORTAGE_GPG_DIR is unset!")
 		if not os.access(repoman_settings["PORTAGE_GPG_DIR"], os.X_OK):
 			raise portage_exception.InvalidLocation(
 				"Unable to access directory: PORTAGE_GPG_DIR='%s'" % \

