diff -ru beecrypt.orig/python/Makefile.in beecrypt/python/Makefile.in
--- beecrypt.orig/python/Makefile.in	2003-08-17 08:57:46.000000000 +0200
+++ beecrypt/python/Makefile.in	2004-01-24 21:32:32.000000000 +0100
@@ -163,11 +163,11 @@
 
 LINT = splint
 
-PYVER = 2.2
+PYVER = $(shell python -c 'import sys; print ".".join(sys.version.split()[0].split(".")[0:2])')
 
 SUBDIRS = test
 
-pylibdir = $(shell python -c 'import sys; print sys.path[1]')
+pythondir = $(shell python -c 'from distutils import sysconfig; print sysconfig.get_python_lib(1,0)')
 pyincdir = $(prefix)/include/python${PYVER}
 
 EXTRA_DIST = debug-py.c
@@ -183,7 +183,6 @@
 
 LDADD = 
 
-pythondir = $(pylibdir)/site-packages
 python_LTLIBRARIES = _bc.la
 
 _bc_la_SOURCES = _bc-py.c mpw-py.c rng-py.c
