Index: softplay-cvs/getFFmpegLibs.c
===================================================================
--- softplay-cvs.orig/getFFmpegLibs.c
+++ softplay-cvs/getFFmpegLibs.c
@@ -1,5 +1,7 @@
 #include <stdio.h>
+extern "C" {
 #include <avcodec.h>
+}
 
 int main() {
         if ( LIBAVCODEC_BUILD > 4795 )
Index: softplay-cvs/SoftHandles.h
===================================================================
--- softplay-cvs.orig/SoftHandles.h
+++ softplay-cvs/SoftHandles.h
@@ -12,7 +12,9 @@
 #ifndef __SOFTHANDLES_H
 #define __SOFTHANDLES_H
 
+extern "C" {
 #include <avformat.h>
+}
 
 #if VDRVERSNUM >= 10330
 #define SOFTDEVICE_VIDEO_STREAM  1
Index: softplay-cvs/SoftPlayer.h
===================================================================
--- softplay-cvs.orig/SoftPlayer.h
+++ softplay-cvs/SoftPlayer.h
@@ -15,7 +15,9 @@
 #include <vdr/device.h>
 #include <vdr/player.h>
 
+extern "C" {
 #include <avformat.h>
+}
 
 #include "SoftHandles.h"
 #include "PlayList.h"
