Index: freepv/trunk/src/libfreepv/Utils/signatures.cpp
===================================================================
--- freepv/trunk/src/libfreepv/Utils/signatures.cpp	(révision 162)
+++ freepv/trunk/src/libfreepv/Utils/signatures.cpp	(révision 163)
@@ -24,6 +24,7 @@
 
 #include "signatures.h"
 #include <iostream>
+#include <cstdio>
 
 namespace FPV{
 namespace Utils{
Index: freepv/trunk/src/libfreepv/QTVRDecoder.cpp
===================================================================
--- freepv/trunk/src/libfreepv/QTVRDecoder.cpp	(révision 162)
+++ freepv/trunk/src/libfreepv/QTVRDecoder.cpp	(révision 163)
@@ -42,6 +42,7 @@
 #include <errno.h>
 #include <vector>
 #include <cstring>
+#include <cstdio>
 
 #include <zlib.h>
 
Index: freepv/trunk/src/libfreepv/utils.cpp
===================================================================
--- freepv/trunk/src/libfreepv/utils.cpp	(révision 162)
+++ freepv/trunk/src/libfreepv/utils.cpp	(révision 163)
@@ -23,6 +23,7 @@
  */
 
 #include <algorithm>
+#include <cstdio>
 
 #include "utils.h"
 
Index: freepv/trunk/src/libfreepv/OpenGLRenderer.cpp
===================================================================
--- freepv/trunk/src/libfreepv/OpenGLRenderer.cpp	(révision 162)
+++ freepv/trunk/src/libfreepv/OpenGLRenderer.cpp	(révision 163)
@@ -26,7 +26,8 @@
 #include <config.h>
 
 #include <vector>
-#include <math.h>
+#include <cmath>
+#include <cstdio>
 
 #include "OpenGLRenderer.h"
 #include "Scene.h"
