diff --exclude='*~' --exclude='.*' -I '$Id:' -urN ZThread-2.3.1.orig/configure.ac ZThread-2.3.1/configure.ac
--- ZThread-2.3.1.orig/configure.ac	2005-11-29 17:56:05.000000000 -0500
+++ ZThread-2.3.1/configure.ac	2005-11-29 18:44:52.000000000 -0500
@@ -81,7 +81,7 @@
 case "$CXX" in
   *cl.exe)   # Microsoft Visual Studio
   CXXFLAGS="/nologo /MT /W3 /GX /O2 /YX /FD /c"
-  CXXFLAGS="$CXXFLAGS /D \"_MBCS\" /D \"_LIB" /D \"WIN32\"
+  CXXFLAGS="$CXXFLAGS /D \"_MBCS\" /D \"_LIB\" /D \"WIN32\""
   ;;
   *)         # GNU
   CXXFLAGS="$CXXFLAGS -Wall"
@@ -95,27 +95,27 @@
 
 dnl Check for debug-mode
 AC_ARG_ENABLE(debug,
-[  --enable-debug          Enable debug symbols [default=no]],
+    AC_HELP_STRING([--enable-debug],[Enable debug symbols [default=no]]),
+    [ENABLE_DEBUG=$enableval],[ENABLE_DEBUG=no])
 
-[
-dnl Enable debug info
-case "$CXX" in
-  *cl.exe)
-    ;;
-  *)
-    CXXFLAGS="$CXXFLAGS -g"
-esac
-],[
-dnl Disable debug info
-case "$CXX" in
-  *cl.exe)
-    CXXFLAGS="$CXXFLAGS /D NDEBUG"
-    ;;
-  *)
-
-    CXXFLAGS="$CXXFLAGS -DNDEBUG"
-esac
-])
+if test x"$ENABLE_DEBUG" = "xyes" ; then
+    case "$CXX" in
+	*cl.exe)
+	    ;;
+	*)
+	    CXXFLAGS="$CXXFLAGS -g"
+	    ;;
+    esac
+else
+    case "$CXX" in
+	*cl.exe)
+	    CXXFLAGS="$CXXFLAGS /D NDEBUG"
+	    ;;
+	*)
+	    CXXFLAGS="$CXXFLAGS -DNDEBUG"
+	    ;;
+    esac
+fi
 
 dnl Select implementation
 AC_MSG_CHECKING(for target implementation)
