Index: libarchive-2.0b12/libarchive/test/test_read_extract.c
===================================================================
--- libarchive-2.0b12.orig/libarchive/test/test_read_extract.c
+++ libarchive-2.0b12/libarchive/test/test_read_extract.c
@@ -126,7 +126,11 @@ DEFINE_TEST(test_read_extract)
 	assert(0 == stat("dir4/c", &st));
 	assert(st.st_mode == (S_IFDIR | 0711));
 	assert(0 == lstat("symlink", &st));
+#ifdef __linux__
+	assert(st.st_mode == (S_IFLNK | 0777));
+#else
 	assert(st.st_mode == (S_IFLNK | 0755));
+#endif
 	assert(0 == stat("symlink", &st));
 	assert(st.st_mode == (S_IFREG | 0755));
 
Index: libarchive-2.0b12/libarchive/test/test_write_disk_secure.c
===================================================================
--- libarchive-2.0b12.orig/libarchive/test/test_write_disk_secure.c
+++ libarchive-2.0b12/libarchive/test/test_write_disk_secure.c
@@ -88,7 +88,11 @@ DEFINE_TEST(test_write_disk_secure)
 
 	assert(0 == lstat("link_to_dir", &st));
 	failure("link_to_dir: st.st_mode=%o", st.st_mode);
+#ifdef __linux__
+	assert((st.st_mode & 07777) == 0777);
+#else
 	assert((st.st_mode & 07777) == 0755);
+#endif
 	failure("link_to_dir: st.st_mode=%o", st.st_mode);
 	assert(S_ISLNK(st.st_mode));
 
