--- src/mod-xslt.c.orig	2003-09-21 22:35:43.000000000 +0000
+++ src/mod-xslt.c	2003-09-21 22:37:15.000000000 +0000
@@ -104,7 +104,7 @@
 	temp = (char*)malloc((tmplen + 1) *sizeof(char));
 	va_start(args,msg);
 	tmplen = vsnprintf(temp,tmplen,msg,args);
-	va_end;
+	va_end(args);
 	temp[tmplen] = 0;
 	writeLog(APLOG_NOTICE,temp);
 	free(temp);
--- src/logging.c.orig	2003-09-21 22:35:43.000000000 +0000
+++ src/logging.c	2003-09-21 22:37:27.000000000 +0000
@@ -74,7 +74,7 @@
 	temp = (char*)malloc((tmplen + 2) *sizeof(char));
 	va_start(args,msg);
 	tmplen = vsnprintf(temp,tmplen + 1,msg,args);
-	va_end;
+	va_end(args);
 	temp[tmplen + 1] = 0;
 	if (request && (level < LOGLEVEL)) {
 #ifdef DEBUG
