--- balsa-2.0.3.orig/configure.in	2002-10-26 18:08:14.000000000 +0200
+++ balsa-2.0.3/configure.in	2002-10-30 01:18:02.000000000 +0100
@@ -89,9 +89,16 @@
 AC_CONFIG_SUBDIRS(libmutt)
 
 
-AC_CHECK_LIB(pspell, new_pspell_manager,,[
-	AC_MSG_ERROR([*** libpspell(-devel) not found.  Please check http://www.balsa.net for information on the required files.])
-], -lstdc++ -lpspell-modules)
+AC_ARG_WITH(aspell,
+   [  --with-aspell            Turn on aspell ],
+    [case "${withval}" in
+       yes) LIBS="-lstdc++ -laspell" ;
+               aspell=true ;;
+       no) LIBS="-lstdc++" ;
+               aspell=false ;;
+       *) AC_MSG_ERROR(bad value ${withval} for --with-aspell) ;;
+   esac],[LIBS=""])
+AM_CONDITIONAL(WITH_ASPELL, test, x$aspell = xtrue)
 
 dnl Reset
 # LDFLAGS="$balsa_save_LDFLAGS"
