diff -NurB nautilus-media-0.3.3.1-orig/media-info/media-info.c nautilus-media-0.3.3.1/media-info/media-info.c
--- nautilus-media-0.3.3.1-orig/media-info/media-info.c	2003-08-12 09:04:06.000000000 -0500
+++ nautilus-media-0.3.3.1/media-info/media-info.c	2003-09-15 08:27:05.000000000 -0500
@@ -258,10 +258,12 @@
 gboolean
 gst_media_info_read_idler (GstMediaInfo *info, GstMediaInfoStream **streamp)
 {
+  GstMediaInfoPriv *priv;
+  
   /* if it's NULL then we're sure something went wrong higher up) */
   if (info == NULL) return FALSE;
 
-  GstMediaInfoPriv *priv = info->priv;
+  priv = info->priv;
 
   g_assert (streamp != NULL);
   switch (priv->state)
