--- nautilus-media-0.2.2/audio-properties-view/audio-properties-view.c.orig	2003-09-12 00:11:22.366614368 +0200
+++ nautilus-media-0.2.2/audio-properties-view/audio-properties-view.c	2003-09-12 00:11:32.604058040 +0200
@@ -275,7 +275,7 @@
 
 	view = g_new0 (AudioPropertiesView, 1);
 
-	xml = glade_xml_new (DATADIR_UNINST
+	xml = glade_xml_new (DATADIR
 			     "/audio-properties-view/audio-properties-view.glade",
 			     "content", NULL);
 	if (!xml)
