Index: pvr350-2007-08-07/pvr350ivtv.h
===================================================================
--- pvr350-2007-08-07.orig/pvr350ivtv.h
+++ pvr350-2007-08-07/pvr350ivtv.h
@@ -22,7 +22,6 @@
 
 #ifndef _PVR350_IVTV_H
 #define _PVR350_IVTV_H
-#include <linux/i2c-id.h>
 #ifndef __KERNEL__
 #define __user
 #endif
Index: pvr350-2007-08-07/video.h
===================================================================
--- pvr350-2007-08-07.orig/video.h
+++ pvr350-2007-08-07/video.h
@@ -24,8 +24,6 @@
 #ifndef _DVBVIDEO_H_
 #define _DVBVIDEO_H_
 
-#include <linux/compiler.h>
-
 #ifdef __KERNEL__
 #include <linux/types.h>
 #else
