--- ./camstream/ftpput.cc	2002-05-01 01:23:20.000000000 +0100
+++ ./camstream/ftpput.cc.patched	2003-10-04 05:26:00.000000000 +0100
@@ -35,18 +35,19 @@
 
 void CFTPPut::Usage()
 {
-   qDebug("
-Not enough/wrong arguments. Syntax:
+   qDebug("\n"
+"Not enough/wrong arguments. Syntax:\n"
+"\n\t"
+"ftpput file [options] [file ...] user:password@server[/path]\n"
+"\n"
+"The file is uploaded to the server using the FTP protocol. Path can be\n"
+"relative to the user's directory or absolute (use a double /).\n"
+"\n"
+"Options:\n\t"
+"--passive         Use passive FTP to upload data\n"
+"\n"
 
-   ftpput file [options] [file ...] user:password@server[/path]
-   
-The file is uploaded to the server using the FTP protocol. Path can be
-relative to the user's directory or absolute (use a double /).
-
-Options:
-  --passive         Use passive FTP to upload data
-
-");
+);
 }
 
 /** 

