Index: amSynth-1.2.3/configure.in
===================================================================
--- amSynth-1.2.3.orig/configure.in
+++ amSynth-1.2.3/configure.in
@@ -5,7 +5,6 @@ AC_CONFIG_HEADERS([config.h])
 
 
 addCXXFLAGS="-D_REENTRANT"
-CXXFLAGS="$CXXFLAGS -Wall -W"
 
 AC_PROG_CXX
 AC_PROG_INSTALL
@@ -90,7 +89,6 @@ AC_ARG_ENABLE(debug,
 if test "${enable_debug}" = "yes" ; then
 	AC_MSG_RESULT([yes])
 	debugging=yes
-	CXXFLAGS="$CXXFLAGS -g -Wall -D_DEBUG"
 	COMPOPSTR="debugging $COMPOPSTR"
 	OPTION_DEBUG="yes"
 else
@@ -113,9 +111,6 @@ if test "$enable_profile" = "yes" ; then
                 CXXFLAGS="$CXXFLAGS -p"
         fi
 else
-	if test "$GCC" = "yes" ; then
-		CXXFLAGS="$CXXFLAGS -O3 -ffast-math -fno-exceptions -fomit-frame-pointer"
-	fi
 	AC_MSG_RESULT([no])
 	profiling=no
 fi
