--- Makefile.in
+++ Makefile.in
@@ -208,7 +208,7 @@
 
 rlocate_SOURCES = pidfile.h pidfile.c main.c link.c misc.c link.h \
 	misc.h rlocate.h rlocate.c $(am__append_1)
-SUBDIRS = doc rlocate-daemon rlocate-module rlocate-scripts
+SUBDIRS = doc rlocate-daemon rlocate-scripts
 EXTRA_DIST = rlocate.cron debian rlocate-scripts autogen.sh \
              contrib install-cron.sh.in ChangeLog.slocate
 
--- rlocate-daemon/Makefile.in
+++ rlocate-daemon/Makefile.in
@@ -172,7 +172,7 @@
 	      -DRLOCATE_GRP=\"$(rlocate_group)\"
 
 rlocated_SOURCES = ../misc.h ../misc.c ../pidfile.h ../pidfile.c rlocated.c
-AM_CFLAGS = -O2 -Wall
+AM_CFLAGS = -Wall
 all: all-am
 
 .SUFFIXES:
--- rlocate-scripts/Makefile.in
+++ rlocate-scripts/Makefile.in
@@ -344,8 +344,6 @@
 	rm $(srcdir)/rlocate-checkpoint.temp
 
 install-exec-hook:
-@FREEBSD_TRUE@	-/usr/sbin/pw groupadd $(rlocate_group)
-@FREEBSD_FALSE@	-/usr/sbin/groupadd $(rlocate_group)
 	chown root:$(rlocate_group) $(DESTDIR)$(bindir)/rlocate-checkpoint
 	chmod 2755 $(DESTDIR)$(bindir)/rlocate-checkpoint
 
