--- src/Makefile.in
+++ src/Makefile.in
@@ -190,9 +190,9 @@
 		  rlocate.h rlocate.c cmds.c cmds.h conf.c conf.h utils.c \
 	   	  utils.h
 
-SUBDIRS = rlocate-daemon rlocate-module rlocate-scripts
+SUBDIRS = rlocate-daemon rlocate-scripts
 EXTRA_DIST = rlocate.cron rlocate-scripts
-AM_CFLAGS = -O2 -Wall
+AM_CFLAGS = -Wall
 all: all-recursive
 
 .SUFFIXES:
--- src/rlocate-daemon/Makefile.in
+++ src/rlocate-daemon/Makefile.in
@@ -169,7 +169,7 @@
 	      -DRLOCATE_GRP=\"$(rlocate_group)\"
 
 rlocated_SOURCES = ../pidfile.h ../pidfile.c rlocated.c
-AM_CFLAGS = -O2 -Wall
+AM_CFLAGS = -Wall
 all: all-am
 
 .SUFFIXES:
--- src/rlocate-scripts/Makefile.in
+++ src/rlocate-scripts/Makefile.in
@@ -343,7 +343,6 @@
 	rm $(srcdir)/rlocate-checkpoint.temp
 
 install-exec-hook:
-	-/usr/sbin/groupadd $(rlocate_group)
 	chown $(root_uid):$(rlocate_group) $(DESTDIR)$(bindir)/rlocate-checkpoint
 	chmod 2755 $(DESTDIR)$(bindir)/rlocate-checkpoint
 
