diff -prN libstdc++-v3-orig/include/bits/locale_facets.h libstdc++-v3/include/bits/locale_facets.h
*** libstdc++-v3-orig/include/bits/locale_facets.h	2004-03-24 21:11:32.000000000 +0000
--- libstdc++-v3/include/bits/locale_facets.h	2004-04-23 15:45:34.000000000 +0100
*************** namespace std
*** 1161,1168 ****
  
  	_M_widen_ok = 1;
  	// Set _M_widen_ok to 2 if memcpy can't be used.
! 	for (size_t __i = 0; __i < sizeof(_M_widen); ++__i)
! 	  if (__tmp[__i] != _M_widen[__i])
  	    {
  	      _M_widen_ok = 2;
  	      break;
--- 1161,1168 ----
  
  	_M_widen_ok = 1;
  	// Set _M_widen_ok to 2 if memcpy can't be used.
! 	for (size_t __j = 0; __j < sizeof(_M_widen); ++__j)
! 	  if (__tmp[__j] != _M_widen[__j])
  	    {
  	      _M_widen_ok = 2;
  	      break;
*************** namespace std
*** 1182,1192 ****
  	// Check if any default values were created.  Do this by
  	// renarrowing with a different default value and comparing.
  	bool __consecutive = true;
! 	for (size_t __i = 0; __i < sizeof(_M_narrow); ++__i)
! 	  if (!_M_narrow[__i])
  	    {
  	      char __c;
! 	      do_narrow(__tmp + __i, __tmp + __i + 1, 1, &__c);
  	      if (__c == 1)
  		{
  		  __consecutive = false;
--- 1182,1192 ----
  	// Check if any default values were created.  Do this by
  	// renarrowing with a different default value and comparing.
  	bool __consecutive = true;
! 	for (size_t __j = 0; __j < sizeof(_M_narrow); ++__j)
! 	  if (!_M_narrow[__j])
  	    {
  	      char __c;
! 	      do_narrow(__tmp + __j, __tmp + __j + 1, 1, &__c);
  	      if (__c == 1)
  		{
  		  __consecutive = false;
