diff -urdN gcc-3.4.0.old/gcc/libgcc-std.ver gcc-3.4.0/gcc/libgcc-std.ver
--- gcc-3.4.0.old/gcc/libgcc-std.ver	2004-05-13 02:41:39.633595352 -0400
+++ gcc-3.4.0/gcc/libgcc-std.ver	2004-05-13 02:46:04.836278416 -0400
@@ -175,9 +175,11 @@
   _Unwind_SjLj_ForcedUnwind
   _Unwind_SjLj_Resume
 
+%if !defined(_LIBC_PROVIDES_SSP_)
   # stack smash handler symbols
   __guard
   __stack_smash_handler
+%endif
 }
 
 %inherit GCC_3.3 GCC_3.0
diff -urdN gcc-3.4.0.old/gcc/libgcc2.c gcc-3.4.0/gcc/libgcc2.c
--- gcc-3.4.0.old/gcc/libgcc2.c	2004-05-13 02:41:32.928614664 -0400
+++ gcc-3.4.0/gcc/libgcc2.c	2004-05-13 02:48:49.101306336 -0400
@@ -1669,6 +1669,7 @@
 
 
 #ifdef L_stack_smash_handler
+#ifndef _LIBC_PROVIDES_SSP_
 #include <stdio.h>
 #include <string.h>
 #include <fcntl.h>
@@ -1784,4 +1785,5 @@
 #endif
   _exit (127);
 }
+#endif /* _LIBC_PROVIDES_SSP_ */
 #endif /* L_stack_smash_handler */
