--- interact.c.orig	2003-02-27 14:31:08.000000000 +0100
+++ interact.c	2004-05-30 03:57:03.055194149 +0200
@@ -33,8 +33,7 @@
 unsigned int buf_a6,
 unsigned int buf_a7)
 {
-  fprintf(statfile, "  | %7d | %7d | %7d | %7d | %7d | %7d | %7d | %7d     | %7d | %7d | %11.0f 
-| %6d | %6d | %6d | %6d | %6d | %6d | %6d | %6d | %6d |\n",
+  fprintf(statfile, "  | %7d | %7d | %7d | %7d | %7d | %7d | %7d | %7d     | %7d | %7d | %11.0f | %6d | %6d | %6d | %6d | %6d | %6d | %6d | %6d | %6d |\n",
       nsectors_a, nsectors_a1, nsectors_a2, nsectors_a3, nsectors_a4, nsectors_a5, nsectors_a6, nsectors_a7,
  nsectors_v, nsectors_p, nbytes, buf_a, buf_a1, buf_a2, buf_a3, buf_a4, buf_a5, buf_a6, buf_a7, buf_v);
 }
