diff -Nruwp gwydion-dylan-2.4.0/d2c/runtime/garbage-collection/garbage-collection-exports.dylan gwydion-dylan-2.4.0.orig/d2c/runtime/garbage-collection/garbage-collection-exports.dylan
--- gwydion-dylan-2.4.0/d2c/runtime/garbage-collection/garbage-collection-exports.dylan	2002-01-25 11:38:07.000000000 +0000
+++ gwydion-dylan-2.4.0.orig/d2c/runtime/garbage-collection/garbage-collection-exports.dylan	2008-03-09 18:28:27.000000000 +0000
@@ -14,7 +14,6 @@ define module garbage-collection
 
   export GC-gc-no, 
     // GC-oom-fn, GC-oom-fn-setter,
-    GC-quiet, GC-quiet-setter, 
     GC-dont-gc, GC-dont-gc-setter,
     GC-dont-expand, GC-dont-expand-setter, 
     GC-full-freq, GC-full-freq-setter,
diff -Nruwp gwydion-dylan-2.4.0/d2c/runtime/garbage-collection/garbage-collection.dylan gwydion-dylan-2.4.0.orig/d2c/runtime/garbage-collection/garbage-collection.dylan
--- gwydion-dylan-2.4.0/d2c/runtime/garbage-collection/garbage-collection.dylan	2003-04-08 23:22:30.000000000 +0000
+++ gwydion-dylan-2.4.0.orig/d2c/runtime/garbage-collection/garbage-collection.dylan	2008-03-09 18:22:02.000000000 +0000
@@ -57,15 +57,6 @@ define sealed method GC-all-interior-poi
   value;
 end method GC-all-interior-pointers-setter;
 
-define sealed method GC-quiet () => (result :: <integer>);
-  as(<integer>, c-variable-ref(int: "&GC_quiet"));
-end method GC-quiet;
-
-define sealed method GC-quiet-setter (value :: <integer>) => (result :: <integer>);
-  c-variable-ref(int: "&GC_quiet") := value;
-  value;
-end method GC-quiet-setter;
-
 define sealed method GC-finalize-on-demand () => (result :: <integer>);
   as(<integer>, c-variable-ref(int: "&GC_finalize_on_demand"));
 end method GC-finalize-on-demand;
