--- DenyHosts-2.2/denyhosts.cfg-dist	2006-02-28 10:28:21.000000000 +0000
+++ DenyHosts-2.2-new/denyhosts.cfg-dist	2006-02-28 10:42:13.000000000 +0000
@@ -9,13 +9,16 @@
 # argument
 #
 # Redhat or Fedora Core:
-SECURE_LOG = /var/log/secure
+#SECURE_LOG = /var/log/secure
 #
 # Mandrake, FreeBSD or OpenBSD: 
 #SECURE_LOG = /var/log/auth.log
 #
-# SuSE:
-#SECURE_LOG = /var/log/messages
+# Gentoo/SuSE:
+SECURE_LOG = /var/log/messages
+#
+# Metalog:
+#SECURE_LOG = /var/log/sshd/current
 #
 ########################################################################
 
@@ -126,7 +129,7 @@
 # Note: it is recommended that you use an absolute pathname
 # for this value (eg. /home/foo/denyhosts/data)
 #
-WORK_DIR = /usr/share/denyhosts/data
+WORK_DIR = /var/lib/denyhosts
 #
 #######################################################################
 
@@ -168,11 +171,14 @@
 # running at a time.
 #
 # Redhat/Fedora:
-LOCK_FILE = /var/lock/subsys/denyhosts
+#LOCK_FILE = /var/lock/subsys/denyhosts
 #
 # Debian
 #LOCK_FILE = /var/run/denyhosts.pid
 #
+# Gentoo
+LOCK_FILE = /var/run/denyhosts.pid
+#
 # Misc
 #LOCK_FILE = /tmp/denyhosts.lock
 #
diff -urN DenyHosts-2.1/setup.py DenyHosts-2.1-new/setup.py
--- DenyHosts-2.1/setup.py	2006-02-10 01:03:39.000000000 +0000
+++ DenyHosts-2.1-new/setup.py	2006-02-13 11:03:08.000000000 +0000
@@ -20,17 +20,17 @@
       author="Phil Schwartz",
       author_email="phil_schwartz@users.sourceforge.net",
       url="http://denyhosts.sourceforge.net",
-      scripts=['denyhosts.py'],
+##      scripts=['denyhosts.py'],
       package_dir={'DenyHosts': 'DenyHosts'},
       packages=["DenyHosts"],
-      data_files=[(libpath, glob("denyhosts.cfg-dist")),
-                  (libpath, glob("setup.py")),
-                  (libpath, glob("daemon-control-dist")),
-                  (libpath, glob("CHANGELOG.txt")),
-                  (libpath, glob("README.txt")),
-                  (scriptspath, glob("scripts/*")),
-                  (pluginspath, glob("plugins/*")),
-                  (libpath, glob("LICENSE.txt"))],
+##      data_files=[(libpath, glob("denyhosts.cfg-dist")),
+##                  (libpath, glob("setup.py")),
+##                  (libpath, glob("daemon-control-dist")),
+##                  (libpath, glob("CHANGELOG.txt")),
+##                  (libpath, glob("README.txt")),
+##                  (scriptspath, glob("scripts/*")),
+##                  (pluginspath, glob("plugins/*")),
+##                  (libpath, glob("LICENSE.txt"))],
       license="GPL v2",
       ##extra_path='denyhosts',
       long_description="""
