--- linden/indra/llmessage/lltemplatemessagereader.cpp~	2009-05-20 15:26:15.000000000 -0600
+++ linden/indra/llmessage/lltemplatemessagereader.cpp	2009-05-25 21:56:50.000000000 -0600
@@ -676,8 +676,7 @@
 
 						// default to 0s.
 						U32 size = mvci.getSize();
-						std::vector<U8> data(size);
-						memset(&(data[0]), 0, size);
+						std::vector<U8> data(size, 0);
 						cur_data_block->addData(mvci.getName(), &(data[0]), 
 												size, mvci.getType());
 					}
