--- linden/indra/llwindow/llwindowsdl.cpp~	2009-05-20 15:26:16.000000000 -0600
+++ linden/indra/llwindow/llwindowsdl.cpp	2009-05-25 21:42:55.000000000 -0600
@@ -2195,6 +2195,7 @@
 		orig_color.red = guint16(65535 * *r);
 		orig_color.green= guint16(65535 * *g);
 		orig_color.blue = guint16(65535 * *b);
+		orig_color.pixel = 0;
 		color = orig_color;
 
 		gtk_color_selection_set_previous_color (colorsel, &color);
