--- linden/indra/newview/llappviewerlinux.cpp~	2009-05-20 15:26:19.000000000 -0600
+++ linden/indra/newview/llappviewerlinux.cpp	2009-05-25 23:24:31.000000000 -0600
@@ -188,7 +188,7 @@
 		for (i = 0; i < size; i++)
 		{
 			// the format of the StraceFile is very specific, to allow (kludgy) machine-parsing
-			fprintf(StraceFile, "%-3d ", i);
+                    fprintf(StraceFile, "%-3ld ", (long) i);
 			fprintf(StraceFile, "%-32s\t", "unknown");
 			fprintf(StraceFile, "%p ", stackarray[i]);
 			fprintf(StraceFile, "%s\n", strings[i]);
