--- src/log_pam.c.original	2009-05-18 14:05:58.409532683 +0200
+++ src/log_pam.c	2009-05-18 14:06:24.348921783 +0200
@@ -156,10 +156,10 @@
     result->auth_ok = 0;
     PAM_password = pass;
     PAM_username = user;
-    pam_error = pam_start("pure-ftpd", user, &PAM_conversation, &pamh);
+    pam_error = pam_start("ftp", user, &PAM_conversation, &pamh);
     PAM_BAIL;
 # ifdef PAM_TTY
-    (void) pam_set_item(pamh, PAM_TTY, "pure-ftpd");
+    (void) pam_set_item(pamh, PAM_TTY, "ftp");
 # endif
 # ifdef PAM_RUSER
     (void) pam_set_item(pamh, PAM_RUSER, user);
