--- ftpmount/Makefile.orig	2001-08-14 19:03:22.000000000 +0200
+++ ftpmount/Makefile	2003-08-25 11:34:44.000000000 +0200
@@ -2,7 +2,7 @@
 FTPMOUNT := /usr/bin/ftpmount
 
 AUTOMOUNTED_FTP_DIR := /mnt/ftpfs
-AUTOMOUNT_FILES := /etc
+AUTOMOUNT_FILES := /etc/autofs
 
 # Directories to search for header files
 
@@ -114,11 +114,9 @@
 	rm -f ${FTPMOUNT}
 	
 install_automount: automount
-	install -m755 --owner=0 --group=0 -b -D automount ${AUTOMOUNT_FILES}/auto.ftp
-	install -d -m755 --owner=0 --group=0 ${AUTOMOUNTED_FTP_DIR}
-	cp ${AUTOMOUNT_FILES}/auto.master ${AUTOMOUNT_FILES}/auto.master.ftpfs
-	echo "${AUTOMOUNTED_FTP_DIR} ${AUTOMOUNT_FILES}/auto.ftp" >> ${AUTOMOUNT_FILES}/auto.master
-	service autofs restart
+	install -m755 --owner=0 --group=0 -b -D automount ${AUTOMOUNT_FILES_PORTAGE}/auto.ftp
+	install -d -m755 --owner=0 --group=0 ${AUTOMOUNTED_FTP_DIR_PORTAGE}
+	touch ${AUTOMOUNTED_FTP_DIR_PORTAGE}/.keep
 
 uninstall_automount:
 	mv ${AUTOMOUNT_FILES}/auto.master.ftpfs ${AUTOMOUNT_FILES}/auto.master
