--- configure.org	2004-08-28 15:55:58.000000000 +0200
+++ configure	2004-08-28 15:56:21.000000000 +0200
@@ -13220,9 +13220,3 @@
         ffmpeg                  $BACKEND_FFMPEG
         divx4linux              $BACKEND_DIVX4LINUX
 "
-
-test "x$BACKEND_MP1E" = "xyes" -a -e $srcdir/mp1e/configure.in && echo "
-Note mp1e is now configured as rte backend. To build the mp1e command
-line program, configure and make in mp1e/ instead. You can build
-rte and mp1e by configuring from other directories too.
-"
