diff -urN kmymoney-4.5.3.orig/kmymoney/mymoney/storage/CMakeLists.txt kmymoney-4.5.3/kmymoney/mymoney/storage/CMakeLists.txt
--- kmymoney-4.5.3.orig/kmymoney/mymoney/storage/CMakeLists.txt	2011-02-12 10:05:48.000000000 +0100
+++ kmymoney-4.5.3/kmymoney/mymoney/storage/CMakeLists.txt	2011-08-06 22:12:57.501609127 +0200
@@ -33,7 +33,8 @@
   kde4_add_unit_test( mymoneydatabasemgrtest TESTNAME kmymoney-mymoneydatabasemgrtest ${mymoneydatabasemgrtest_SRCS} )
   ADD_DEPENDENCIES( mymoneydatabasemgrtest kmymoney_base kmymoney_common kmm_storage )
 
-  target_link_libraries( mymoneydatabasemgrtest kmymoney_base kmymoney_common kmm_storage ${QT_QTGUI_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTSQL_LIBRARY} ${QT_QTTEST_LIBRARY} )
+  target_link_libraries( mymoneydatabasemgrtest kmymoney_base kmymoney_common kmm_storage ${QT_QTGUI_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTSQL_LIBRARY} 
+    ${QT_QTXML_LIBRARY} ${QT_QTTEST_LIBRARY} )
 ENDIF(NOT WIN32)
 
   # MyMoneyMap
