diff -ruN kmymoney-4.6.0.orig/kmymoney/converter/mymoneygncreader.cpp kmymoney-4.6.0/kmymoney/converter/mymoneygncreader.cpp
--- kmymoney-4.6.0.orig/kmymoney/converter/mymoneygncreader.cpp	2011-08-09 07:48:04.000000000 +0200
+++ kmymoney-4.6.0/kmymoney/converter/mymoneygncreader.cpp	2011-10-22 00:40:05.731618915 +0200
@@ -1267,6 +1267,7 @@
   if (bAnonymize) setFileHideFactor();
   //m_defaultPayee = createPayee (i18n("Unknown payee"));
 
+  MyMoneyFile::instance()->attachStorage(m_storage);
   MyMoneyFileTransaction ft;
   m_xr = new XmlReader(this);
   try {
@@ -1277,6 +1278,7 @@
     KMessageBox::error(0, i18n("Import failed:\n\n%1", e->what()), PACKAGE);
     qWarning("%s", qPrintable(e->what()));
   } // end catch
+  MyMoneyFile::instance()->detachStorage(m_storage);
   signalProgress(0, 1, i18n("Import complete"));  // switch off progress bar
   delete m_xr;
   qDebug("Exiting gnucash importer");
