--- tcl/utils/sound.tcl.old	2006-09-08 00:26:02.000000000 +0200
+++ tcl/utils/sound.tcl	2006-09-08 00:26:19.000000000 +0200
@@ -38,7 +38,7 @@
   variable soundFolder
 
   ::splash::add "Setting up audio move announcement..."
-  if {[catch {package require snack 2.0}]} {
+  if {[catch {package require snack}]} {
     set hasSnackPackage 0
     ::splash::add "    Move speech disabled - Snack sound package not found"
     return
--- tcl/start.tcl.old	2007-08-30 11:45:25.000000000 +0200
+++ tcl/start.tcl	2007-08-30 11:51:26.000000000 +0200
@@ -16,7 +16,7 @@
 # The "\" at the end of the comment line below is necessary! It means
 #   that the "exec" line is a comment to Tcl/Tk, but not to /bin/sh.
 # The next line restarts using tkscid: \
-exec tkscid "$0" "$@"
+TCLLIBPATH="@snack_path@ $TCLLIBPATH" exec tkscid "$0" "$@"
 
 # For the above to work, tkscid must be in a directory in your PATH.
 # Alternatively, you can set the first line to start tkscid directly
@@ -117,17 +117,13 @@
 # scidConfigDir, scidDataDir, scidLogDir:
 #   Location of Scid configuration, data and log files.
 set scidConfigDir [file nativename [file join $scidUserDir "config"]]
-set scidDataDir [file nativename [file join $scidUserDir "data"]]
+set scidDataDir "@GENTOO_DATADIR@"
 set scidLogDir [file nativename [file join $scidUserDir "log"]]
 
 # ecoFile: the ECO file for opening classification. Scid will try to load
 # this first, and if that fails, it will try to load  "scid.eco" in the
 # current directory.
-if {$windowsOS} {
   set ecoFile [file join $scidDataDir "scid.eco"]
-} else {
-  set ecoFile "/usr/local/share/scid/scid.eco"
-}
 
 # boardSizes: a list of the available board sizes.
 set boardSizes [list 25 30 35 40 45 50 55 60 65 70]
@@ -597,18 +593,14 @@
 
 ### Audio move announcement options:
 
-set ::utils::sound::soundFolder [file nativename [file join $::scidExeDir sounds]]
+set ::utils::sound::soundFolder [file nativename [file join $::scidDataDir sounds]]
 set ::utils::sound::announceNew 0
 set ::utils::sound::announceForward 0
 set ::utils::sound::announceBack 0
 
 
 # Spell-checking file: default is "spelling.ssp".
-if {$windowsOS} {
   set spellCheckFile [file join $scidDataDir "spelling.ssp"]
-} else {
-  set spellCheckFile "/usr/local/share/scid/spelling.ssp"
-}
 
 # Engines list file: -- OLD NAMES, NO LONGER USED
 #set engines(file) [file join $scidUserDir "engines.lis"]
@@ -863,7 +855,7 @@
 }
 
 makeScidDir $scidConfigDir
-makeScidDir $scidDataDir
+#makeScidDir $scidDataDir
 makeScidDir $scidLogDir
 
 # Rename old email log file if necessary:
