diff -u serialmail-0.75-orig_1/maildirsmtp.sh serialmail-0.75-orig_2/maildirsmtp.sh
--- serialmail-0.75-orig_1/maildirsmtp.sh	2004-01-01 17:08:29.000000000 +0200
+++ serialmail-0.75-orig_2/maildirsmtp.sh	2004-01-01 17:35:20.000000000 +0200
@@ -1,5 +1,16 @@
 
+if [ -z "$6" ]; then
+
+exec \
+HOME/bin/maildirserial -b -t 1209600 -- "$1" "$2" \
+tcpclient -RHl0 -- "$3" 25 \
+HOME/bin/serialsmtp "$2" "$4"
+
+else
+
 exec \
 HOME/bin/maildirserial -b -t 1209600 -- "$1" "$2" \
 tcpclient -RHl0 -- "$3" 25 \
 HOME/bin/serialsmtp "$2" "$4" "$5" "$6"
+
+fi
