--- util/link
+++ util/link
@@ -21,7 +21,7 @@
 xtmp=/tmp/ld.$$.$USER.out
 
 echo LD "$($bin/cleanname ${BASE}$outfile)"
-$LD -o $outfile $ofiles $LDFLAGS $args >$xtmp 2>&1
+$LD -o $outfile $ofiles $LDFLAGS $args $LDLIBS $LIBS >$xtmp 2>&1
 status=$?
 
 sed 's/.*: In function `[^:]*: *//' $xtmp | egrep . | 
--- util/compile
+++ util/compile
@@ -8,7 +8,7 @@
 xtmp=/tmp/cc.$$.$USER.out
 
 echo CC $($bin/cleanname ${BASE}$outfile)
-$CC -o $outfile $CFLAGS $@ 2>$xtmp
+$CC -c -o $outfile $CFLAGS $@ 2>$xtmp
 status=$?
 
 base=$(echo $BASE | sed 's/,/\\,/g')
--- config.mk
+++ config.mk
@@ -11,10 +11,10 @@
 TASKINC = -I${HOME}/libtask
 
 # paths
-PREFIX = /usr/local
+PREFIX = $(DESTDIR)/usr
 BIN = ${PREFIX}/bin
 MAN = ${PREFIX}/share/man
-ETC = ${PREFIX}/etc
+ETC = /etc
 LIBDIR = ${PREFIX}/lib
 INCLUDE = ${PREFIX}/include
 
@@ -23,14 +23,10 @@
 LIBS = -L/usr/lib -lc
 
 # Flags
-include ${ROOT}/mk/gcc.mk
-CFLAGS = -g -Wall ${INCS} -DVERSION=\"${VERSION}\"
-LDFLAGS = -g ${LIBS}
+MKDEP = cpp -M
+SOCFLAGS += -fPIC
+EXTRACFLAGS = ${INCS} -DVERSION=\"${VERSION}\"
 
-# Compiler
-CC = cc -c
-# Linker (Under normal circumstances, this should *not* be 'ld')
-LD = cc
 # Other
 AR = ar crs
 #AR = sh -c 'ar cr "$$@" && ranlib "$$@"'
--- mk/dir.mk
+++ mk/dir.mk
@@ -9,15 +9,15 @@
 	done
 
 dall:
-	dirs="${DIRS}"; ${MKSUBDIR}
+	+dirs="${DIRS}"; ${MKSUBDIR}
 dclean:
-	dirs="${DIRS}"; ${MKSUBDIR}
+	+dirs="${DIRS}"; ${MKSUBDIR}
 dinstall:
-	dirs="${INSTDIRS}"; ${MKSUBDIR}
+	+dirs="${INSTDIRS}"; ${MKSUBDIR}
 duninstall:
-	dirs="${INSTDIRS}"; ${MKSUBDIR}
+	+dirs="${INSTDIRS}"; ${MKSUBDIR}
 ddepend:
-	dirs="${DIRS}"; ${MKSUBDIR}
+	+dirs="${DIRS}"; ${MKSUBDIR}
 
 all: dall
 clean: dclean
--- mk/hdr.mk
+++ mk/hdr.mk	2010-06-11 16:53:56.018621649 +0000
@@ -41,8 +41,9 @@
 
 .a.install:
 	echo INSTALL $$($(CLEANNAME) ${BASE}$<)
-	cp -f $< ${LIBDIR}/$<
-	chmod 0644 ${LIBDIR}/$<
+	file=$<; \
+	cp -f $< ${LIBDIR}/$${file##*/}; \
+	chmod 0644 ${LIBDIR}/$${file##*/}
 .a.uninstall:
 	echo UNINSTALL $$($(CLEANNAME) ${BASE}$<)
 	rm -f ${LIBDIR}/$<
@@ -76,8 +77,8 @@
 depend: cleandep
 
 FILTER = cat
-COMPILE= CC="${CC}" CFLAGS="${CFLAGS}" ${ROOT}/util/compile
-LINK= LD="${LD}" LDFLAGS="${LDFLAGS}" ${ROOT}/util/link
+COMPILE= CC="$(CC)" CFLAGS="$(CFLAGS) $(EXTRACFLAGS)" ${ROOT}/util/compile
+LINK= LD="$(LD)" LDFLAGS="$(LDFLAGS)" LIBS="$(LIBS)" LDLIBS="$(LDLIBS)" ${ROOT}/util/link
 CLEANNAME=${ROOT}/util/cleanname
 
 include ${ROOT}/config.mk
