 touchcal.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/touchcal.c b/touchcal.c
index 7341fd7..93b262c 100644
--- a/touchcal.c
+++ b/touchcal.c
@@ -138,8 +138,8 @@ static void init_mu(char dev[])
 	set_page[3]='1';
 	set_page[4]=0x0d;
 
-	/*define data string*/
-	strcpy(data,"00180769001D0793");
+    /* define data string */
+    strncpy(data, "00180769001D0793", 16);
 
 	bzero(&newtio, sizeof(newtio)); /* clear struct for new port settings */
 
