diff -ru 2.1.22-orig/src/package.c 2.1.22-r1/src/package.c
--- 2.1.22-orig/src/package.c   2007-03-09 11:48:31.000000000 -0800
+++ cfengine-2.1.22/src/package.c     2007-03-09 11:50:26.000000000 -0800
@@ -446,6 +446,7 @@
          {
          ReadLine(line,CF_BUFSIZE-1,pp);
          snprintf(OUTPUT,CF_BUFSIZE,"Package install: %s\n",line);
+         CfLog(cfinform,OUTPUT,""); 
          }
       
       if (cfpclose(pp) != 0)
