--- configure.ac
+++ configure.ac
@@ -81,21 +81,16 @@
 
 AC_CHECK_LIB(howl, sw_discovery_init,
 [
-	AC_CHECK_HEADERS([howl/howl_config.h],
-	[
 		AC_DEFINE(WITH_HOWL, 1, [Use howl for zeroconf mechanism])
-		HOWL_INCLUDES="-I/usr/include/howl"
+		HOWL_INCLUDES="-I/usr/include/howl -I/usr/include/avahi-compat-howl"
 		LIB_HOWL="-lhowl -lpthread"
 		AC_SUBST(HOWL_INCLUDES)
 		AC_SUBST(LIB_HOWL)
 		AC_C_BIGENDIAN
-	],
-	[
-		AC_MSG_WARN([Zeroconf diabled, libhowl (headers) not found.])
-	])
 ],
 [
 	AC_MSG_WARN([Zeroconf disabled, libhowl (libs) not found.])
+	exit 1
 ],
 [-lpthread])
 
