--- elfutils-0.94/src/strip.c.old	2003-09-26 08:01:49.000000000 +0200
+++ elfutils-0.94/src/strip.c	2004-02-22 22:20:42.000000000 +0100
@@ -265,8 +265,16 @@
 
       /* If we have to preserve the timestamp, we need it in the
 	 format utimes() understands.  */
-      TIMESPEC_TO_TIMEVAL (&tv[0], &pre_st.st_atim);
-      TIMESPEC_TO_TIMEVAL (&tv[1], &pre_st.st_mtim);
+
+#if defined ( __alpha ) || defined ( __mips)
+	tv[0].tv_sec = pre_st.st_atime;
+        tv[0].tv_usec = 0;
+        tv[1].tv_sec = pre_st.st_mtime;
+        tv[1].tv_usec = 0;
+#else
+	TIMESPEC_TO_TIMEVAL (&tv[0], &pre_st.st_atim);
+    	TIMESPEC_TO_TIMEVAL (&tv[1], &pre_st.st_mtim);
+#endif
     }
 
   /* Open the file.  */
