diff -ur kickpim-0.5.3.orig/src/dialogs/kickpimcard.cpp kickpim-0.5.3/src/dialogs/kickpimcard.cpp
--- kickpim-0.5.3.orig/src/dialogs/kickpimcard.cpp	2003-12-04 14:13:33.000000000 +0100
+++ kickpim-0.5.3/src/dialogs/kickpimcard.cpp	2004-10-17 23:29:55.751344904 +0200
@@ -309,7 +309,7 @@
   if (prefFound)      m_content[PHONE]->selectContent( i18n("Preferred Number") );
   else if (homeFound) m_content[PHONE]->selectContent( i18n("home phone") );
   else                m_content[PHONE]->selectContent( i18n(phoneMap.begin().key()) );
-};
+}
 
 
 void KickPimCard::setAddressContent()
@@ -352,7 +352,7 @@
   if (type && KABC::Address::Pref)   text+="Pref<br>";
 */
   return text;
-};
+}
 
 void KickPimCard::setBirthdayContent ()
 {
diff -ur kickpim-0.5.3.orig/src/kickpimcontactview.cpp kickpim-0.5.3/src/kickpimcontactview.cpp
--- kickpim-0.5.3.orig/src/kickpimcontactview.cpp	2003-11-30 14:40:02.000000000 +0100
+++ kickpim-0.5.3/src/kickpimcontactview.cpp	2004-10-17 23:32:48.107142840 +0200
@@ -111,7 +111,7 @@
   setSorting(0);
   sort();
   setSorting(-1);
-};
+}
 
 void KickPimContactView::onContactsDoubleClick(QListViewItem* qitem)
 {
@@ -129,7 +129,7 @@
   KickPimContactViewItem* item = 0;
   item = new KickPimContactViewItem( this, 0, KickPIM::rep()->displayName(contact) );
   item->setContact(contact);
-};
+}
 
 void KickPimContactView::addDistributionList(DistributionList* distList)
 {
@@ -137,7 +137,7 @@
   KickPimContactViewItem* item = 0;
   item = new KickPimContactViewItem( this, lastItem(), "["+distList->name()+"]" );
   item->setDistributionList(distList);
-};
+}
 
 int KickPimContactView::preferredHeight()
 {
diff -ur kickpim-0.5.3.orig/src/kpcontactevent.cpp kickpim-0.5.3/src/kpcontactevent.cpp
--- kickpim-0.5.3.orig/src/kpcontactevent.cpp	2003-09-06 13:42:07.000000000 +0200
+++ kickpim-0.5.3/src/kpcontactevent.cpp	2004-10-17 23:34:03.972609536 +0200
@@ -24,7 +24,7 @@
   m_text    = "";
   m_date    = QDate(25,61,61); // invalid
   m_contact = 0;
-};
+}
 
 KPContactEvent::~KPContactEvent()
 {
diff -ur kickpim-0.5.3.orig/src/kpkabcontact.cpp kickpim-0.5.3/src/kpkabcontact.cpp
--- kickpim-0.5.3.orig/src/kpkabcontact.cpp	2003-11-15 00:09:16.000000000 +0100
+++ kickpim-0.5.3/src/kpkabcontact.cpp	2004-10-17 23:34:59.549160616 +0200
@@ -240,7 +240,7 @@
   {
     m_addr->insertEmail(emailAddr, preferred);
   }
-};
+}
 
 
 
@@ -255,7 +255,7 @@
     ++itor;
   }
 
-};
+}
 
 void KPKabContact::addPhoneNumber(QString number, QString key)
 {
@@ -270,20 +270,20 @@
       }
     }
   }
-};
+}
 
 void KPKabContact::setBirthday    ( QDate date )
 {
   if (!m_addr) return;
   m_addr->setBirthday(date);
-};
+}
 
 void KPKabContact::setAnniversary ( QDate date )
 {
   if (!m_addr) return;
   QString dateStr = date.toString( Qt::ISODate );
   m_addr->insertCustom( "KADDRESSBOOK", "X-Anniversary" , dateStr );
-};
+}
 
 
 bool KPKabContact::edit()
@@ -322,4 +322,4 @@
     }
   }
   KABC::StdAddressBook::save();
-};
+}
