From 72412557f05822c9f670849300f8f47d91fa5963 Mon Sep 17 00:00:00 2001
From: Takao Fujiwara <tfujiwar@redhat.com>
Date: Tue, 18 May 2010 02:20:13 +0000
Subject: [yelp-toc] Fix TOC title I18N

https://bugzilla.gnome.org/show_bug.cgi?id=615141
---
diff --git a/src/yelp-toc.c b/src/yelp-toc.c
index 201cabb..085c47c 100644
--- a/src/yelp-toc.c
+++ b/src/yelp-toc.c
@@ -553,7 +553,7 @@ xml_trim_titles (xmlNodePtr node, xmlChar * nodetype)
 	if (!xmlStrcmp (cur->name, nodetype)) {
 	    xmlChar *cur_lang = NULL;
 	    int cur_pri = INT_MAX;
-	    cur_lang = xmlNodeGetLang (cur);
+	    cur_lang = xmlGetNsProp(cur, BAD_CAST "lang", XML_XML_NAMESPACE);
 	    if (cur_lang) {
 		for (j = 0; langs[j]; j++) {
 		    if (g_str_equal (cur_lang, langs[j])) {
--
cgit v0.8.3.1
