--- apel-10.3/APEL-CFG.org	Sat Jun 23 19:25:54 2001
+++ apel-10.3/APEL-CFG	Sat Jun 23 19:26:09 2001
@@ -74,8 +74,8 @@
 ;; XEmacs 21.0 and later.
 ;; (setq PACKAGEDIR "/usr/local/lib/xemacs/xemacs-packages")
 
-;; (setq APEL_PREFIX "apel")
-;; (setq EMU_PREFIX "emu")
+(setq APEL_PREFIX "apel")
+(setq EMU_PREFIX "apel")
 
 ;; If you want to install all of APEL modules to VERSION_SPECIFIC_LISPDIR,
 ;; uncomment and edit this.
