diff -u cl-regex-1.orig/macs.lisp cl-regex-1/macs.lisp
--- cl-regex-1.orig/macs.lisp	2002-10-06 13:30:14.000000000 -0500
+++ cl-regex-1/macs.lisp	2003-10-15 13:20:18.790073928 -0500
@@ -516,14 +516,11 @@
   (clrhash *pattern-cache*)
   nil)
 
-
-(defconstant +special-class-names+
-  '((":alpha:" alpha) (":upper:" upper) (":lower:" lower) (":digit:" digit)
-    (":alnum:" alnum) (":xdigit:" xdigit) (":odigit:" odigit) (":punct:" punct)
-    (":space:" space) (":word:" wordchar)))
-
-
-
+(eval-when (:compile-toplevel)
+  (defconstant +special-class-names+
+    '((":alpha:" alpha) (":upper:" upper) (":lower:" lower) (":digit:" digit)
+      (":alnum:" alnum) (":xdigit:" xdigit) (":odigit:" odigit) (":punct:" punct)
+      (":space:" space) (":word:" wordchar))))
 
 (defun expand-char-class (chars)
 "Expand an encoded char class into an explicit enumeration of all
