--- cdparanoia-III-alpha9.8/paranoia/Makefile.in.old	2007-01-06 19:54:23.632598436 +0100
+++ cdparanoia-III-alpha9.8/paranoia/Makefile.in	2007-01-06 19:55:15.038341210 +0100
@@ -22,7 +22,7 @@
 OFILES = paranoia.o p_block.o overlap.o gap.o isort.o
 #TFILES = isort.t gap.t p_block.t paranoia.t
 
-LIBS = ../interface/libcdda_interface.a -lm
+LIBS = ../interface/libcdda_interface.so -lm
 export VERSION
 
 all: lib slib
@@ -44,7 +44,7 @@
 	$(RANLIB) libcdda_paranoia.a
 
 libcdda_paranoia.so: 	$(OFILES)	
-	$(CC) $(LDFLAGS) $(LIBFLAGS) -shared -o libcdda_paranoia.so.0.$(VERSION) -Wl,-soname -Wl,libcdda_paranoia.so.0 $(OFILES)
+	$(CC) $(OFILES) -shared -o libcdda_paranoia.so.0.$(VERSION) -Wl,-soname -Wl,libcdda_paranoia.so.0 $(LDFLAGS) $(LIBFLAGS) $(LIBS)
 	[ -e libcdda_paranoia.so.0 ] || ln -s libcdda_paranoia.so.0.$(VERSION) libcdda_paranoia.so.0
 	[ -e libcdda_paranoia.so ] || ln -s libcdda_paranoia.so.0.$(VERSION) libcdda_paranoia.so
 
