--- ziproxy-2.7.2.orig/configure.in
+++ ziproxy-2.7.2/configure.in
@@ -24,7 +24,7 @@
 AC_CHECK_LIB(jpeg, jpeg_start_decompress,, AC_MSG_ERROR([libjpeg not found.]))
 
 USE_LIB(png)
-AC_CHECK_LIB(png, png_check_sig,, AC_MSG_ERROR([libpng not found.]), -lz -lm)
+AC_CHECK_LIB(png, png_sig_cmp,, AC_MSG_ERROR([libpng not found.]), -lz -lm)
 
 USE_LIB(pthread)
 AC_CHECK_LIB(pthread, pthread_create,, AC_MSG_ERROR([libpthread not found.]), -lpthread)
diff -ur ziproxy-2.7.2.orig/src/image.c ziproxy-2.7.2/src/image.c
--- ziproxy-2.7.2.orig/src/image.c	2009-09-28 18:07:39.000000000 +0300
+++ ziproxy-2.7.2/src/image.c	2010-03-08 22:12:55.000000000 +0200
@@ -457,7 +457,7 @@
 				color_type == PNG_COLOR_TYPE_GRAY_ALPHA)
 	{
 		grayscale=1;	
-        if(bit_depth < 8) png_set_gray_1_2_4_to_8(png_ptr);
+        if(bit_depth < 8) png_set_expand_gray_1_2_4_to_8(png_ptr);
 	}else grayscale=0;
 	
 	if (bit_depth == 16)
