--- konsole/konsole/konsole.cpp.orig	2002-11-22 15:05:57.000000000 +0100
+++ konsole/konsole/konsole.cpp	2002-11-22 15:06:12.000000000 +0100
@@ -357,7 +357,7 @@
    // Remove the empty separator Qt inserts if the menu is empty on popup,
    // not sure if this will be "fixed" in Qt, for now use this hack (malte)
    if(!(isRestored)) {
-     if (sender()->inherits("QPopupMenu") &&
+     if (sender() && sender()->inherits("QPopupMenu") &&
        static_cast<const QPopupMenu *>(sender())->count() == 1)
        const_cast<QPopupMenu *>(static_cast<const QPopupMenu *>(sender()))->removeItemAt(0);
        }
