diff -urN xdvik-22.40y-j1.17/texk/xdvik/vfontmap.sample xdvik-22.40y/texk/xdvik/vfontmap.sample
--- xdvik-22.40y-j1.17/texk/xdvik/vfontmap.sample	2003-08-18 22:26:35.000000000 +0900
+++ xdvik-22.40y/texk/xdvik/vfontmap.sample	2003-08-18 22:25:31.000000000 +0900
@@ -1,31 +1,34 @@
+#
+# Sample vfontmap entry
+#
 
-min	min     # for ASCII
-nmin	min     # for ASCII
-goth	goth    # for ASCII
-ngoth	goth    # for ASCII
-jis	min	# for JIS
-jisg	goth	# for JIS
-dm	min     # for NTT
-dg	goth    # for NTT
-
-mgoth	maru
-fmin	fmin
-fgoth	goth
+#min	min     # for ASCII
+#nmin	min     # for ASCII
+#goth	goth    # for ASCII
+#ngoth	goth    # for ASCII
+#jis	min	# for JIS
+#jisg	goth	# for JIS
+#dm	min     # for NTT
+#dg	goth    # for NTT
+#
+#mgoth	maru
+#fmin	fmin
+#fgoth	goth
 
 #  --with-vflib=vf2ft
-# min     /usr/local/share/texmf/dvipdfm/CIDFont/HiraMinPro-W3.otf
-# nmin    /usr/local/share/texmf/dvipdfm/CIDFont/HiraMinPro-W3.otf
-# goth    /usr/local/share/texmf/dvipdfm/CIDFont/HiraKakuPro-W6.otf
-# tmin    /usr/local/share/texmf/dvipdfm/CIDFont/HiraMinPro-W3.otf
-# tgoth   /usr/local/share/texmf/dvipdfm/CIDFont/HiraKakuPro-W6.otf
-# ngoth   /usr/local/share/texmf/dvipdfm/CIDFont/HiraKakuPro-W6.otf
-# jis     /usr/local/share/texmf/dvipdfm/CIDFont/HiraMinPro-W3.otf
-# jisg    /usr/local/share/texmf/dvipdfm/CIDFont/HiraKakuPro-W6.otf
-# dm      /usr/local/share/texmf/dvipdfm/CIDFont/HiraMinPro-W3.otf  # not checked
-# dg      /usr/local/share/texmf/dvipdfm/CIDFont/HiraKakuPro-W6.otf # not checked
-# mgoth   /usr/local/share/texmf/dvipdfm/CIDFont/HiraMaruPro-W4.otf
-# fmin    /usr/local/share/texmf/dvipdfm/CIDFont/HiraMinPro-W6.otf
-# fgoth   /usr/local/share/texmf/dvipdfm/CIDFont/HiraKakuPro-W8.otf
+min     /usr/X11R6/lib/X11/fonts/truetype/kochi-mincho-subst.ttf
+nmin    /usr/X11R6/lib/X11/fonts/truetype/kochi-mincho-subst.ttf
+goth    /usr/X11R6/lib/X11/fonts/truetype/kochi-gothic-subst.ttf
+tmin    /usr/X11R6/lib/X11/fonts/truetype/kochi-mincho-subst.ttf
+tgoth   /usr/X11R6/lib/X11/fonts/truetype/kochi-gothic-subst.ttf
+ngoth   /usr/X11R6/lib/X11/fonts/truetype/kochi-gothic-subst.ttf
+jis     /usr/X11R6/lib/X11/fonts/truetype/kochi-mincho-subst.ttf
+jisg    /usr/X11R6/lib/X11/fonts/truetype/kochi-gothic-subst.ttf
+dm      /usr/X11R6/lib/X11/fonts/truetype/kochi-mincho-subst.ttf
+dg      /usr/X11R6/lib/X11/fonts/truetype/kochi-gothic-subst.ttf
+mgoth   /usr/X11R6/lib/X11/fonts/truetype/kochi-gothic-subst.ttf
+fmin    /usr/X11R6/lib/X11/fonts/truetype/kochi-mincho-subst.ttf
+fgoth   /usr/X11R6/lib/X11/fonts/truetype/kochi-gothic-subst.ttf
 
 # UTF package (needed --with-vflib=vf2ft)
 #
