diff -Nru nsis-2.23-src.orig/SCons/Config/gnu nsis-2.23-src/SCons/Config/gnu
--- nsis-2.23-src.orig/SCons/Config/gnu	2006-11-11 15:04:17.000000000 +0200
+++ nsis-2.23-src/SCons/Config/gnu	2007-01-15 15:18:20.000000000 +0200
@@ -30,6 +30,8 @@
 defenv['ALIGN_FLAG'] = '-Wl,--file-alignment,512'
 defenv['CPP_REQUIRES_STDLIB'] = 1
 defenv['SUBSYS_CON'] = '-Wl,--subsystem,console'
+defenv.Append(CFLAGS = '-fno-strict-aliasing')
+defenv.Append(CPPFLAGS = '-fno-strict-aliasing')
 
 ### helper functions
 
@@ -68,6 +70,7 @@
 if not defenv['DEBUG']:
 	stub_env.Append(LINKFLAGS = '-s')               # strip
 stub_env.Append(LINKFLAGS = '-mwindows')          # build windows executables
+stub_env.Append(LINKFLAGS = '-L/usr/mingw32/usr/lib')
 stub_env.Append(LINKFLAGS = '$NODEFLIBS_FLAG')    # no standard libraries
 stub_env.Append(LINKFLAGS = '$ALIGN_FLAG')        # 512 bytes align
 stub_env.Append(LINKFLAGS = '-Wl,-e,_WinMain@16') # entry point
@@ -83,8 +86,6 @@
 
 conf = FlagsConfigure(makensis_env)
 conf.CheckLinkFlag('$MAP_FLAG')                   # generate map file
-if not defenv['DEBUG']:
-	TestStrip(conf)                                 # strip
 conf.Finish()
 
 ### plugin environment
@@ -112,8 +113,6 @@
 
 conf = FlagsConfigure(cp_util_env)
 conf.CheckLinkFlag('$MAP_FLAG')                   # generate map file
-if not defenv['DEBUG']:
-	TestStrip(conf)                                 # strip
 conf.Finish()
 
 ### util environment
