diff -Nru nsis-2.24-src.orig/SCons/Config/gnu nsis-2.24-src/SCons/Config/gnu
--- nsis-2.24-src.orig/SCons/Config/gnu	2007-03-11 10:52:57.000000000 +0200
+++ nsis-2.24-src/SCons/Config/gnu	2007-03-11 10:54:11.000000000 +0200
@@ -113,6 +113,7 @@
 if not defenv['DEBUG']:
 	cp_util_env.Append(CCFLAGS = '-O2')             # optimize
 cp_util_env.Append(CCFLAGS = '-Wall')             # all warnings
+cp_util_env.Append(CCFLAGS = '-fno-strict-aliasing') # not safe for strict aliasing
 
 conf = FlagsConfigure(cp_util_env)
 conf.CheckLinkFlag('$MAP_FLAG')                   # generate map file
