diff -Npaur libmxmlplus-0.9.2/src/mxml_attribute.cpp libmxmlplus-0.9.2-fixed/src/mxml_attribute.cpp
--- libmxmlplus-0.9.2/src/mxml_attribute.cpp	2004-01-11 14:32:45.000000000 +0100
+++ libmxmlplus-0.9.2-fixed/src/mxml_attribute.cpp	2004-08-12 04:36:40.696926344 +0200
@@ -32,7 +32,7 @@ Attribute::Attribute( std::istream &in, 
    m_value = "";
    m_name = "";
 
-   while ( iStatus < 6 && ( in >> chr ) ) {
+   while ( iStatus < 6 && ( in.get(chr) ) ) {
       //std::cout << "LINE: " << line() << "  POS: " << character() << std::endl;
       switch ( iStatus ) {
          // begin
