--- openmcu/main.cxx.orig	2003-10-20 23:38:28.000000000 +0200
+++ openmcu/main.cxx	2003-10-20 23:40:25.000000000 +0200
@@ -286,7 +286,7 @@
 
 ///////////////////////////////////////////////////////////////
 #ifdef LOGGING
-static PString DEFAULT_CALL_LOG = "c:\\mcu_log.txt"; 
+static PString DEFAULT_CALL_LOG = "/var/log/openmcu/call.log"; 
 static PMutex logMutex;
 static PTextFile logFile;
 static PFilePath logFilename = DEFAULT_CALL_LOG;
@@ -340,6 +340,8 @@
   : PProcess("OpenH323 Project", "OpenMCU",
              MAJOR_VERSION, MINOR_VERSION, BUILD_TYPE, BUILD_NUMBER)
 {
+  // set location of configuration file
+  SetConfigurationPath("/etc/openmcu.ini");
 }
 
 
