diff -ruN openbox-3.3-rc2.bak/openbox/client.c openbox-3.3-rc2/openbox/client.c
--- openbox-3.3-rc2.bak/openbox/client.c	2005-08-11 15:23:59.000000000 +0200
+++ openbox-3.3-rc2/openbox/client.c	2005-10-07 16:34:21.000000000 +0200
@@ -126,7 +126,7 @@
         windows = NULL;
 
     PROP_SETA32(RootWindow(ob_display, ob_screen),
-                net_client_list, window, (guint32*)windows, size);
+                net_client_list, window, (gulong*)windows, size);
 
     if (windows)
         g_free(windows);
@@ -1297,7 +1297,7 @@
 
 static void client_change_allowed_actions(ObClient *self)
 {
-    guint32 actions[9];
+    gulong actions[9];
     gint num = 0;
 
     /* desktop windows are kept on all desktops */
@@ -1712,8 +1712,8 @@
 
 static void client_change_state(ObClient *self)
 {
-    guint32 state[2];
-    guint32 netstate[11];
+    gulong state[2];
+    gulong netstate[11];
     guint num;
 
     state[0] = self->wmstate;
diff -ruN openbox-3.3-rc2.bak/openbox/frame.c openbox-3.3-rc2/openbox/frame.c
--- openbox-3.3-rc2.bak/openbox/frame.c	2005-07-14 15:44:33.000000000 +0200
+++ openbox-3.3-rc2/openbox/frame.c	2005-10-07 16:06:50.000000000 +0200
@@ -422,7 +422,7 @@
         }
 
         if (!STRUT_EQUAL(self->size, oldsize)) {
-            guint32 vals[4];
+            gulong vals[4];
             vals[0] = self->size.left;
             vals[1] = self->size.right;
             vals[2] = self->size.top;
Files openbox-3.3-rc2.bak/openbox/.libs/lt-openbox and openbox-3.3-rc2/openbox/.libs/lt-openbox differ
Files openbox-3.3-rc2.bak/openbox/.libs/openbox and openbox-3.3-rc2/openbox/.libs/openbox differ
diff -ruN openbox-3.3-rc2.bak/openbox/prop.c openbox-3.3-rc2/openbox/prop.c
--- openbox-3.3-rc2.bak/openbox/prop.c	2004-03-21 02:03:00.000000000 +0100
+++ openbox-3.3-rc2/openbox/prop.c	2005-10-07 16:58:45.000000000 +0200
@@ -206,7 +206,7 @@
                     data[i] = xdata[i];
                     break;
                 case 16:
-                    ((guint16*)data)[i] = ((guint16*)xdata)[i];
+                    ((gushort*)data)[i] = ((gushort*)xdata)[i];
                     break;
                 case 32:
                     ((guint32*)data)[i] = ((gulong*)xdata)[i];
@@ -245,7 +245,7 @@
                     (*data)[i] = xdata[i];
                     break;
                 case 16:
-                    ((guint16*)*data)[i] = ((guint16*)xdata)[i];
+                    ((gushort*)*data)[i] = ((gushort*)xdata)[i];
                     break;
                 case 32:
                     ((guint32*)*data)[i] = ((gulong*)xdata)[i];
@@ -383,13 +383,13 @@
     return FALSE;
 }
 
-void prop_set32(Window win, Atom prop, Atom type, guint32 val)
+void prop_set32(Window win, Atom prop, Atom type, gulong val)
 {
     XChangeProperty(ob_display, win, prop, type, 32, PropModeReplace,
                     (guchar*)&val, 1);
 }
 
-void prop_set_array32(Window win, Atom prop, Atom type, guint32 *val,
+void prop_set_array32(Window win, Atom prop, Atom type, gulong *val,
                       guint num)
 {
     XChangeProperty(ob_display, win, prop, type, 32, PropModeReplace,
diff -ruN openbox-3.3-rc2.bak/openbox/prop.h openbox-3.3-rc2/openbox/prop.h
--- openbox-3.3-rc2.bak/openbox/prop.h	2004-03-21 02:03:00.000000000 +0100
+++ openbox-3.3-rc2/openbox/prop.h	2005-10-07 16:15:38.000000000 +0200
@@ -177,8 +177,8 @@
 gboolean prop_get_strings_locale(Window win, Atom prop, gchar ***ret);
 gboolean prop_get_strings_utf8(Window win, Atom prop, gchar ***ret);
 
-void prop_set32(Window win, Atom prop, Atom type, guint32 val);
-void prop_set_array32(Window win, Atom prop, Atom type, guint32 *val,
+void prop_set32(Window win, Atom prop, Atom type, gulong val);
+void prop_set_array32(Window win, Atom prop, Atom type, gulong *val,
                       guint num);
 void prop_set_string_utf8(Window win, Atom prop, gchar *val);
 void prop_set_strings_utf8(Window win, Atom prop, gchar **strs);
diff -ruN openbox-3.3-rc2.bak/openbox/screen.c openbox-3.3-rc2/openbox/screen.c
--- openbox-3.3-rc2.bak/openbox/screen.c	2005-10-03 17:47:53.000000000 +0200
+++ openbox-3.3-rc2/openbox/screen.c	2005-10-07 16:29:58.000000000 +0200
@@ -156,7 +156,7 @@
     XSetWindowAttributes attrib;
     pid_t pid;
     gint i, num_support;
-    guint32 *supported;
+    gulong *supported;
 
     /* create the netwm support window */
     attrib.override_redirect = TRUE;
@@ -206,7 +206,7 @@
     /* set the _NET_SUPPORTED_ATOMS hint */
     num_support = 51;
     i = 0;
-    supported = g_new(guint32, num_support);
+    supported = g_new(gulong, num_support);
     supported[i++] = prop_atoms.net_current_desktop;
     supported[i++] = prop_atoms.net_number_of_desktops;
     supported[i++] = prop_atoms.net_desktop_geometry;
@@ -340,7 +340,7 @@
     static gint oldw = 0, oldh = 0;
     gint w, h;
     GList *it;
-    guint32 geometry[2];
+    gulong geometry[2];
 
     w = WidthOfScreen(ScreenOfDisplay(ob_display, ob_screen));
     h = HeightOfScreen(ScreenOfDisplay(ob_display, ob_screen));
@@ -368,7 +368,7 @@
 void screen_set_num_desktops(guint num)
 {
     guint i, old;
-    guint32 *viewport;
+    gulong *viewport;
     GList *it;
 
     g_assert(num > 0);
@@ -381,7 +381,7 @@
                net_number_of_desktops, cardinal, num);
 
     /* set the viewport hint */
-    viewport = g_new0(guint32, num * 2);
+    viewport = g_new0(gulong, num * 2);
     PROP_SETA32(RootWindow(ob_display, ob_screen),
                 net_desktop_viewport, cardinal, viewport, num * 2);
     g_free(viewport);
@@ -969,7 +969,7 @@
 void screen_update_areas()
 {
     guint i, x;
-    guint32 *dims;
+    long *dims;
     GList *it;
     gint o;
 
@@ -987,7 +987,7 @@
         area[i] = g_new0(Rect, screen_num_monitors + 1);
     area[i] = NULL;
      
-    dims = g_new(guint32, 4 * screen_num_desktops);
+    dims = g_new(gulong, 4 * screen_num_desktops);
 
     for (i = 0; i < screen_num_desktops + 1; ++i) {
         Strut *struts;

diff -ruN openbox-3.3-rc2.bak/openbox/stacking.c openbox-3.3-rc2/openbox/stacking.c
--- openbox-3.3-rc2.bak/openbox/stacking.c	2005-02-08 05:33:54.000000000 +0100
+++ openbox-3.3-rc2/openbox/stacking.c	2005-10-07 16:21:27.000000000 +0200
@@ -49,7 +49,7 @@
     }
 
     PROP_SETA32(RootWindow(ob_display, ob_screen),
-                net_client_list_stacking, window, (guint32*)windows, i);
+                net_client_list_stacking, window, (gulong*)windows, i);
 
     g_free(windows);
 }
