Index: openbox-3.3-rc2/Makefile.am
===================================================================
--- openbox-3.3-rc2.orig/Makefile.am
+++ openbox-3.3-rc2/Makefile.am
@@ -58,6 +58,7 @@ render_libobrender_la_CPPFLAGS = \
 render_libobrender_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 render_libobrender_la_LIBADD = \
+	parser/libobparser.la \
 	$(X_LIBS) \
 	$(XFT_LIBS) \
 	$(PANGO_LIBS) \
