--- cmz80/z80stb.h
+++ cmz80/z80stb.h
@@ -128,8 +128,6 @@
 /* inline */ BYTE Srl(BYTE b);
 /* inline */ BYTE Sla(BYTE b);
 /* inline */ BYTE Sra(BYTE b);
-/* inline */ BYTE Inc(BYTE b);
-/* inline */ BYTE Dec(BYTE b);
 
 /* inline */ WORD Add_2(WORD wArg1, WORD wArg2);
 /* inline */ WORD Adc_2(WORD wArg1, WORD wArg2);
--- cpu68k/def68k.c
+++ cpu68k/def68k.c
@@ -1318,7 +1318,7 @@
   } /* block */
 }
 
-int clocks_movetable[] = {
+static int clocks_movetable[] = {
    4,   4,  8,  8,  8, 12, 14, 12, 16,
    4,   4,  8,  8,  8, 12, 14, 12, 16,
    8,   8, 12, 12, 12, 16, 18, 16, 20,
--- raze/Makefile.in.old	2006-09-03 03:17:39.000000000 +0200
+++ raze/Makefile.in	2006-09-03 03:16:36.000000000 +0200
@@ -343,6 +343,10 @@
 raze:	raze.o
 raze.o:	raze.asm raze.reg
 	nasm -e raze.asm -o raze2.asm
+	echo "%ifidn __OUTPUT_FORMAT__,elf" >> raze2.asm
+	echo "section .note.GNU-stack noalloc noexec nowrite progbits" \
+		>> raze2.asm
+	echo "%endif" >> raze2.asm
 	nasm -f @ASMOUTPUT@ raze2.asm -o raze.o -praze.reg
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
