diff -ur m17n-lib-1.0.2/src/m17n-core.h m17n-lib-1.0.2.new/src/m17n-core.h
--- m17n-lib-1.0.2/src/m17n-core.h	2004-03-29 11:58:53.000000000 +0800
+++ m17n-lib-1.0.2.new/src/m17n-core.h	2004-06-14 12:38:07.809982344 +0800
@@ -103,7 +103,7 @@
 
     #MSymbol はシンボルオブジェクトの型へのポインタである。 */
 
-typedef struct MSymbol *MSymbol;
+typedef struct MSymbol_ *MSymbol;
 
 /*=*/
 
diff -ur m17n-lib-1.0.2/src/symbol.h m17n-lib-1.0.2.new/src/symbol.h
--- m17n-lib-1.0.2/src/symbol.h	2004-03-09 13:26:14.000000000 +0800
+++ m17n-lib-1.0.2.new/src/symbol.h	2004-06-14 12:37:58.491398984 +0800
@@ -25,7 +25,7 @@
 
 #include "plist.h"
 
-struct MSymbol
+struct MSymbol_
 {
   /** 1 iff a value of property (including text-property) whose key is
       the symbol is a managed object.  */
@@ -40,7 +40,7 @@
   /* Plist of the symbol.  */
   MPlist plist;
 
-  MSymbol next;
+  struct MSymbol_ * next;
 };
 
 #define MSYMBOL_NAME(sym) ((sym)->name)
