--- Makefile.orig	2004-10-01 18:00:34.108028240 -0400
+++ Makefile	2004-10-01 18:01:01.275898096 -0400
@@ -2,11 +2,10 @@
 #
 # Please uncomment the appropriate settings
 
-CC = gcc
-CFLAGS = -Wall -DHAVE_GETOPT_LONG
-LDFLAGS =
-INSTALLPATH = $(DESTDIR)/usr/local/bin
-MANPATH = $(DESTDIR)/usr/local/man/man1
+CC ?= gcc
+CFLAGS += -Wall -DHAVE_GETOPT_LONG
+INSTALLPATH = $(DESTDIR)/usr/bin
+MANPATH = $(DESTDIR)/usr/share/man/man1
 
 PROGNAME = proxytunnel
 OBJ = proxytunnel.o	\
@@ -25,5 +24,6 @@
 
 install:
 		mkdir -p $(INSTALLPATH)
+		mkdir -p $(MANPATH)
 		install -g root -m755 -o root $(PROGNAME) $(INSTALLPATH)/$(PROGNAME)
 		install -g root -m644 -o root debian/$(PROGNAME).1 $(MANPATH)/$(PROGNAME).1
