diff -ur libquicktime-1.0.0.orig/configure.ac libquicktime-1.0.0/configure.ac
--- libquicktime-1.0.0.orig/configure.ac	2007-04-17 20:43:19.000000000 +0300
+++ libquicktime-1.0.0/configure.ac	2007-09-23 14:17:36.000000000 +0300
@@ -365,7 +365,7 @@
 AC_SUBST(GTK_REQUIRED)
 
 dnl Always be in sync with the newest gtk
-GTK_CFLAGS="$GTK_CFLAGS -DGTK_DISABLE_DEPRECATED"
+GTK_CFLAGS="$GTK_CFLAGS"
 
 dnl
 dnl Check for libdv
