diff -Naur scipy-0.6.0/scipy/fftpack/src/zfftnd_fftw.c scipy-0.6.0-new/scipy/fftpack/src/zfftnd_fftw.c
--- scipy-0.6.0/scipy/fftpack/src/zfftnd_fftw.c	2007-09-22 03:56:38.000000000 -0400
+++ scipy-0.6.0-new/scipy/fftpack/src/zfftnd_fftw.c	2007-10-05 09:20:19.000000000 -0400
@@ -25,7 +25,7 @@
 	  free(caches_zfftnd_fftw[id].dims);, 10)
 
 
-extern void zfftnd_mkl(complex_double * inout, int rank,
+extern void zfftnd_fftw(complex_double * inout, int rank,
 		       int *dims, int direction, int howmany,
 		       int normalize)
 {
