--- Lib/special/setup.py.orig	2007-09-11 18:11:04.457232054 +0100
+++ Lib/special/setup.py	2007-09-11 18:09:23.103456231 +0100
@@ -20,9 +20,9 @@
     # Fortran libraries
     config.add_library('mach',sources=[join('mach','*.f')],
                        config_fc={'noopt':(__file__,1)})
-    config.add_library('toms',sources=[join('amos','*.f')])
-    config.add_library('amos',sources=[join('toms','*.f')])
-    config.add_library('cdf',sources=[join('cdflib','*.f')])
+    config.add_library('toms',sources=[join('toms','*.f')])
+    config.add_library('amos',sources=[join('amos','*.f')])
+    config.add_library('cdflib',sources=[join('cdflib','*.f')])
     config.add_library('specfun',sources=[join('specfun','*.f')])
 
     # Extension _cephes
@@ -30,7 +30,7 @@
                'toms_wrappers.c','cdf_wrappers.c','ufunc_extras.c']
     config.add_extension('_cephes', sources=sources,
                          libraries=['amos','toms','c_misc','cephes','mach',
-                                    'cdf', 'specfun'],
+                                    'cdflib', 'specfun'],
                          define_macros = define_macros
                          )
     # Extension specfun
