--- Lib/sandbox/montecarlo/tests/test_dictsampler.py.orig	2007-06-06 14:25:47.000000000 +0100
+++ Lib/sandbox/montecarlo/tests/test_dictsampler.py	2007-06-06 14:26:02.000000000 +0100
@@ -37,7 +37,7 @@
         #import pdb
         #pdb.set_trace()
         s = sampler.sample(n)
-        assert sum(s[i]=='b' for i in range(n),axis=0)*1./n > 0.75
+        assert sum((s[i]=='b' for i in range(n)),axis=0)*1./n > 0.75
 
         #lam = 10.0
         #n = 35
