diff -ruN libalkimia-4.3.0.orig/CMakeLists.txt libalkimia-4.3.0/CMakeLists.txt
--- libalkimia-4.3.0.orig/CMakeLists.txt	2011-01-12 11:41:50.000000000 +0100
+++ libalkimia-4.3.0/CMakeLists.txt	2011-04-21 01:59:24.079220359 +0200
@@ -72,6 +72,8 @@
 
 ########### documentation ###################
 
+if(BUILD_DOC)
+
 # check for Doxygen
 find_package(Doxygen)
 if(DOXYGEN_FOUND)
@@ -81,9 +83,20 @@
 
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libalkimia.doxygen.in ${CMAKE_CURRENT_BINARY_DIR}/libalkimia.doxygen IMMEDIATE)
 
-  add_custom_target(apidoc ${DOXYGEN} ${CMAKE_CURRENT_BINARY_DIR}/libalkimia.doxygen)
+  add_custom_target(apidoc ALL ${DOXYGEN} ${CMAKE_CURRENT_BINARY_DIR}/libalkimia.doxygen)
+else(DOXYGEN_FOUND)
+  MESSAGE(FATAL_ERROR "API documentation requested but doxygen not found")
 endif(DOXYGEN_FOUND)
 
+ELSE(BUILD_DOC)
+
+  MESSAGE(STATUS "")
+  MESSAGE(STATUS "Documentation will NOT be installed")
+  MESSAGE(STATUS "Because the option -DBUILD_DOC=ON wasn't passed to cmake")
+  MESSAGE(STATUS "")
+
+ENDIF(BUILD_DOC)
+
 ########### tests ###################
 
 set( alkvaluetest_SRCS alkvaluetest.cpp )
