http://bugs.gentoo.org/142765

Index: trunk/cracklib/lib/packlib.c
===================================================================
--- trunk/cracklib/lib/packlib.c	(revision 89)
+++ trunk/cracklib/lib/packlib.c	(revision 90)
@@ -450,7 +450,7 @@ GetPW(pwp, number)
     return (data[number % NUMWORDS]);
 }
 
-uint32_t
+unsigned int
 FindPW(pwp, string)
     PWDICT *pwp;
     char *string;
Index: trunk/cracklib/lib/packer.h
===================================================================
--- trunk/cracklib/lib/packer.h	(revision 89)
+++ trunk/cracklib/lib/packer.h	(revision 90)
@@ -88,7 +88,7 @@ typedef struct {
 
 extern PWDICT *PWOpen(const char *prefix, char *mode);
 extern int PWClose(PWDICT *pwp);
-extern uint32_t FindPW(PWDICT *pwp, char *string);
+extern unsigned int FindPW(PWDICT *pwp, char *string);
 extern int PutPW(PWDICT *pwp, char *string);
 extern int PMatch(char *control, char *string);
 extern char *Mangle(char *input, char *control);
