--- cracklib-2.7/cracklib/Makefile	2003-11-09 23:01:29.405532640 +0200
+++ cracklib-2.7.static/cracklib/Makefile	2003-11-09 23:17:57.983245984 +0200
@@ -6,20 +6,26 @@
 # and upwards.
 ###
 
-LIB	= libcrack.so
+LIB	= libcrack
 OBJ	= fascist.o packlib.o rules.o stringlib.o
 CFLAGS += -g -I../cracklib -DIN_CRACKLIB -fPIC
 
-$(LIB):	$(OBJ) Makefile
-	$(LD) -shared -soname $(LIB).$(MAJOR) -o $(LIB).$(VERSION) $(OBJ) -lc
-	rm -f $(LIB).$(MAJOR) $(LIB)
-	ln -s $(LIB).$(VERSION) $(LIB).$(MAJOR)
-	ln -s $(LIB).$(MAJOR) $(LIB)
+all: $(LIB).so $(LIB).a
+
+$(LIB).so:	$(OBJ) Makefile
+	$(LD) -shared -soname $(LIB).so.$(MAJOR) -o $(LIB).so.$(VERSION) $(OBJ) -lc
+	rm -f $(LIB).so.$(MAJOR) $(LIB).so
+	ln -s $(LIB).so.$(VERSION) $(LIB).so.$(MAJOR)
+	ln -s $(LIB).so.$(MAJOR) $(LIB).so
+
+$(LIB).a:	$(OBJ) Makefile
+	$(AR) cru $(LIB).a $(OBJ)
 
 clean:
-	-rm -f $(OBJ) $(LIB) $(LIB).$(VERSION) *~
+	-rm -f $(OBJ) $(LIB).so $(LIB).a $(LIB).so.$(VERSION) *~
 
-install: $(LIB) crack.h
-	install -m 755 $(LIB).$(VERSION) $(DESTDIR)/usr/lib
-	ln -sf $(LIB).$(VERSION) $(DESTDIR)/usr/lib/$(LIB)
+install: $(LIB).so $(LIB).a crack.h
+	install -m 755 $(LIB).so.$(VERSION) $(DESTDIR)/usr/lib
+	ln -sf $(LIB).so.$(VERSION) $(DESTDIR)/usr/lib/$(LIB).so
+	install -m 755 $(LIB).a $(DESTDIR)/usr/lib
 	install -m 644 crack.h $(DESTDIR)/usr/include
