--- gmyth.orig/m4/ac_doxygen.m4	2010-01-09 16:45:19.715489995 +0100
+++ gmyth/m4/ac_doxygen.m4	2010-01-09 16:46:11.841495719 +0100
@@ -78,7 +78,7 @@
 AC_PATH_TOOL([$1], [$2])
 if test "$DX_FLAG_[]DX_CURRENT_FEATURE$$1" = 1; then
     AC_MSG_WARN([$2 not found - will not DX_CURRENT_DESCRIPTION])
-    AC_SUBST([DX_FLAG_[]DX_CURRENT_FEATURE], 0)
+    AC_SUBST([DX_FLAG_]DX_CURRENT_FEATURE, 0)
 fi
 ])
 
@@ -101,7 +101,7 @@
 # ----------------------------------------------------------
 # Turn off the DX_CURRENT_FEATURE if the required feature is off.
 AC_DEFUN([DX_CLEAR_DEPEND], [
-test "$DX_FLAG_$1" = "$2" || AC_SUBST([DX_FLAG_[]DX_CURRENT_FEATURE], 0)
+test "$DX_FLAG_$1" = "$2" || AC_SUBST([DX_FLAG_]DX_CURRENT_FEATURE, 0)
 ])
 
 # DX_FEATURE_ARG(FEATURE, DESCRIPTION,
--- gmyth.orig/src/Makefile.am	2007-05-23 20:50:41.000000000 +0200
+++ gmyth/src/Makefile.am	2010-01-09 16:50:18.609492323 +0100
@@ -55,17 +55,18 @@
 	$(GSTBASE_CFLAGS)			\
 	$(GSTPLUGINSBASE_CFLAGS)	\
 	$(MYSQL_CFLAGS)				\
-	$(LIBXML_CFLAGS)
+	$(LIBXML_CFLAGS)			\
+	$(LIBCURL_CFLAGS)
 
 libgmyth_la_LDFLAGS = 			\
-	-export-dynamic 			\
-	$(GLIB_CFLAGS) 				\
-	$(GOBJECT_CFLAGS)			\
+	-export-dynamic 			
+libgmyth_la_LIBADD  =			\
 	$(MYSQL_LIBS) 				\
 	$(GST_LIBS) 				\
 	$(GSTBASE_LIBS)				\
 	$(GSTPLUGINS_LIBS)			\
-	$(LIBXML_LIBS)
+	$(LIBXML_LIBS)				\
+	$(LIBCURL_LIBS)				
 
 libgmyth_includedir = 			\
 	$(pkgincludedir)
