Index: sandal-prawn-c5c1ddba07efa45fd4b10085ab5ca255a43269be/lib/prawn.rb
===================================================================
--- sandal-prawn-c5c1ddba07efa45fd4b10085ab5ca255a43269be.orig/lib/prawn.rb
+++ sandal-prawn-c5c1ddba07efa45fd4b10085ab5ca255a43269be/lib/prawn.rb
@@ -18,7 +18,7 @@ module Prawn
   dir = File.dirname(file)
                           
   # The base source directory for Prawn as installed on the system
-  BASEDIR = File.expand_path(File.join(dir, '..'))
+  BASEDIR = "/usr/share/prawn"
   
   VERSION = "0.4.1"
   
