This patch is from Debian's enigmail-0.91 patch set

--- mozilla/mailnews/extensions/enigmail/ui/content/enigmailMessengerOverlay.xul.orig	2004-11-08 23:34:04.000000000 +0000
+++ mozilla/mailnews/extensions/enigmail/ui/content/enigmailMessengerOverlay.xul	2004-11-08 23:35:20.000000000 +0000
@@ -159,10 +159,12 @@
         </menu>
         
         <menuseparator/>
+	<!--
         <menuitem id="enigmail_upgrade"
              label="&enigmail.upgrade.label;"
              accesskey="&enigmail.upgrade.accesskey;"
              oncommand="EnigUpgrade();" />
+	-->
 
         <menuitem id="enigmail_help"
              label="&enigmail.help.label;"
