diff -ur rubygems-0.8.11/post-install.rb rubygems-0.8.11.new/post-install.rb
--- rubygems-0.8.11/post-install.rb	2005-07-10 06:18:40.000000000 +0700
+++ rubygems-0.8.11.new/post-install.rb	2007-01-26 22:05:18.000000000 +0700
@@ -61,7 +61,8 @@
 def install_sources
   $: << "lib"
   require 'rubygems'
-  Gem::manage_gems
+  require 'rubygems/builder'
+  require 'rubygems/installer'
   Dir.chdir("pkgs/sources") do
     load "sources.gemspec"
     spec = Gem.sources_spec
