--- am-utils-6.0.8/include/am_defs.h.sopwith	2002-12-11 16:18:36.000000000 -0500
+++ am-utils-6.0.8/include/am_defs.h	2002-12-11 16:20:14.000000000 -0500
@@ -327,19 +327,20 @@
  */
 #ifdef HAVE_NDBM_H
 # include <ndbm.h>
-# ifndef DATUM
+# define HAVE_NDBM
+#elif defined(HAVE_GDBM_NDBM_H)
+# include <gdbm/ndbm.h>
+# define HAVE_NDBM
+#elif defined(HAVE_DB1_NDBM_H)
+# include <db1/ndbm.h>
+# define HAVE_NDBM
+
+#endif
+
+#if !defined(DATUM) && defined(HAVE_NDBM)
 /* ensure that struct datum is not included again from <rpcsvc/yp_prot.h> */
 #  define DATUM
 # endif /* not DATUM */
-#else /* not HAVE_NDBM_H */
-# ifdef HAVE_DB1_NDBM_H
-#  include <db1/ndbm.h>
-#  ifndef DATUM
-/* ensure that struct datum is not included again from <rpcsvc/yp_prot.h> */
-#   define DATUM
-#  endif /* not DATUM */
-# endif /* HAVE_DB1_NDBM_H */
-#endif /* HAVE_NDBM_H */
 
 /*
  * Actions to take if <net/errno.h> exists.
--- am-utils-6.0.8/configure.in.sopwith	2002-12-11 16:15:35.000000000 -0500
+++ am-utils-6.0.8/configure.in	2002-12-11 16:18:09.000000000 -0500
@@ -231,10 +231,8 @@
 dnl some systems replaced old ndbm.h include with GNU ndbm.h, which requires
 dnl libgdbm for dbm_pagfno, so check for this.
 AC_MSG_CHECKING(for GNU ndbm.h)
-AC_TRY_LINK([#include <ndbm.h>], [
-DBM *db;
-dbm_pagfno(db);
-], AC_MSG_RESULT(no), AC_CHECK_LIB(gdbm, dbm_pagfno))
+AC_CHECK_HEADERS(ndbm.h gdbm/ndbm.h)
+AC_CHECK_LIB(gdbm, dbm_pagfno)
 dnl ======================================================================
 
 dnl ######################################################################
