diff -ru pyflag-0.80-orig/configure.in pyflag-0.80/configure.in
--- pyflag-0.80-orig/configure.in	2006-01-19 09:43:16.000000000 +1100
+++ pyflag-0.80/configure.in	2006-01-20 20:03:04.000000000 +1100
@@ -11,7 +11,14 @@
 AC_CHECK_HEADER(zlib.h,,AC_MSG_ERROR([You Must install zlib-dev to build pyflag]))
 AC_CHECK_HEADER(pcap.h,,AC_MSG_ERROR([You Must install libpcap-dev to build pyflag]))
 AC_CHECK_HEADER(magic.h,,AC_MSG_ERROR([You Must install libmagic-dev to build pyflag this may be part of file the package for some distros]))
+
+# if we have dbtool don't need sleuthkit
+AC_CHECK_PROG([HAVE_DBTOOL],dbtool,"yes", "no")
+if test "x$HAVE_DBTOOL" = xno
+then
 AC_CHECK_FILE(src/filesystems/sleuthkit-2.03.tar.gz,,AC_MSG_ERROR([You Must download sleuthkit-2.03.tar.gz and place in src/filesystems ]))
+fi
+AM_CONDITIONAL(HAVE_DBTOOL, test x$HAVE_DBTOOL = xyes)
 
 # python checks 
 # (requires autoconf 1.5+ and the macros in acinclude.m4)
diff -ru pyflag-0.80-orig/src/Makefile.am pyflag-0.80/src/Makefile.am
--- pyflag-0.80-orig/src/Makefile.am	2006-01-09 12:13:40.000000000 +1100
+++ pyflag-0.80/src/Makefile.am	2006-01-20 20:03:04.000000000 +1100
@@ -1,4 +1,10 @@
-SUBDIRS = lib mailtools indextools network regtools virustools filesystems include
+
+SUBDIRS = lib mailtools indextools network regtools virustools include
+
+if HAVE_DBTOOL
+else
+SUBDIRS+=filesystems
+endif
 
 # main pyflag (python) application
 # not sure how to do this recursively
diff -ru pyflag-0.80-slightlypatched/configure.in pyflag-0.80/configure.in
--- pyflag-0.80-slightlypatched/configure.in	2006-01-20 20:42:03.000000000 +1100
+++ pyflag-0.80/configure.in	2006-01-20 20:42:52.000000000 +1100
@@ -44,7 +44,7 @@
 fi
 
 # try to find magic files
-AC_ARG_WITH(magic, AC_HELP_STRING([--with-magic], [specify a colon-separated list of system magic files, default autodetect]),,[magic=$(file -v | tail -1 | awk '{print $4}')])
+AC_ARG_WITH(magic, AC_HELP_STRING([--with-magic], [specify a colon-separated list of system magic files, default autodetect]),,[magic=$(file -v | tail -n 1 | awk '{print $4}')])
 AC_SUBST(magic)
             
 
--- pyflag-0.80-old/src/plugins/Configuration.py	2006-01-19 09:43:16.000000000 +1100
+++ pyflag-0.80/src/plugins/Configuration.py	2006-01-22 13:00:26.000000000 +1100
@@ -3,6 +3,7 @@
 config=pyflag.conf.ConfObject()
 import os
 import DB
+import stat
 
 class Configure(Reports.report):
     """ Configures pyflag """
@@ -38,7 +39,8 @@
                 result.para("We do not seem to have enough privileges to access %s, or the path (%s) does not exist" %(k,config.__class__.__dict__[k]))
                 return
 
-        fd=open(os.environ['HOME'] + '/.pyflagrc', 'a+', S_IRWXU)
+        fd=open(os.environ['HOME'] + '/.pyflagrc', 'a+') #, os.S_IRWXU)
+	os.chmod(os.environ['HOME'] + '/.pyflagrc',  stat.S_IRWXU)
 	## TODO Think append is wrong?
         result.para("Writing new $HOME/.pyflagrc")
 
--- pyflag-0.80/Makefile.am	2006-01-19 09:00:50.000000000 +1100
+++ pyflag-0.80.new/Makefile.am	2006-01-22 13:04:54.000000000 +1100
@@ -10,7 +10,7 @@
 # perform subs in scripts and config files
 edit = sed \
 	-e 's,@datadir\@,$(pkgdatadir),g' \
-	-e 's,@sysconf\@,$(sysconfdir),g' \
+	-e 's,@sysconfdir\@,$(sysconfdir),g' \
 	-e 's,@pkgpyexec\@,$(pkgpyexecdir),g' \
 	-e 's,@pythondir\@,$(pythondir),g' \
 	-e 's,@prefix\@,$(prefix),g' \
diff -ru pyflag-0.80-orig/pyflag.in pyflag-0.80/pyflag.in
--- pyflag-0.80-orig/pyflag.in	2006-01-10 12:04:31.000000000 +1100
+++ pyflag-0.80/pyflag.in	2006-01-22 13:19:31.000000000 +1100
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-. @sysconf@/pyflagrc
+. @sysconfdir@/pyflagrc
 
 ## Try to load personalised configurations
 if [ -e ~/.pyflagrc ] ; then source ~/.pyflagrc; fi
diff -ru pyflag-0.80-orig/pyflag_launch.in pyflag-0.80/pyflag_launch.in
--- pyflag-0.80-orig/pyflag_launch.in	2006-01-10 12:04:59.000000000 +1100
+++ pyflag-0.80/pyflag_launch.in	2006-01-22 13:20:30.000000000 +1100
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-. @sysconf@/pyflagrc
+. @sysconfdir@/pyflagrc
 
 ## Try to load personalised configurations
 if [ -e ~/.pyflagrc ] ; then source ~/.pyflagrc; fi
diff -ru pyflag-0.80-orig/pyflash.in pyflag-0.80/pyflash.in
--- pyflag-0.80-orig/pyflash.in	2006-01-10 12:04:48.000000000 +1100
+++ pyflag-0.80/pyflash.in	2006-01-22 13:20:11.000000000 +1100
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-. @sysconf@/pyflagrc
+. @sysconfdir@/pyflagrc
 
 ## Try to load personalised configurations
 if [ -e ~/.pyflagrc ] ; then source ~/.pyflagrc; fi
