diff -ur ntop-3.3.8.orig/configure.in ntop-3.3.8/configure.in
--- ntop-3.3.8.orig/configure.in	2008-10-02 10:54:15.000000000 +0000
+++ ntop-3.3.8/configure.in	2008-10-05 10:39:47.000000000 +0000
@@ -504,13 +504,6 @@
     LOCALEDIR="/usr/lib/locale"
 fi
 
-dnl> Add /usr/local/
-CFLAGS="${CFLAGS} -I/usr/local/include -I/opt/local/include"
-CPPFLAGS="${CPPFLAGS} -I/usr/local/include -I/opt/local/include"
-LDFLAGS="${LDFLAGS} -L/usr/local/lib -L/opt/local/lib"
-
-
-
 echo
 echo "Test the compiler and setup CFLAGS..."
 echo
diff -ur ntop-3.3.8.orig/globals-core.c ntop-3.3.8/globals-core.c
--- ntop-3.3.8.orig/globals-core.c	2008-10-01 21:03:36.000000000 +0000
+++ ntop-3.3.8/globals-core.c	2008-10-05 10:39:47.000000000 +0000
@@ -424,7 +424,7 @@
   _mtuSize[DLT_ATM_RFC1483] = 9180                             /* LLC/SNAP encapsulated atm */;
   _headerSize[DLT_ATM_RFC1483] = 0;
 
-  /* _mtuSize[DLT_RAW] = ?                                        raw IP */
+  _mtuSize[DLT_RAW] = 1500                                     /*   raw IP */;
   _headerSize[DLT_RAW] = 0;
 
   /* Others defined in bpf.h at tcpdump.org as of the resync - it would be NICE
diff -ur ntop-3.3.8.orig/report.c ntop-3.3.8/report.c
--- ntop-3.3.8.orig/report.c	2008-08-17 21:06:33.000000000 +0000
+++ ntop-3.3.8/report.c	2008-10-05 10:39:47.000000000 +0000
@@ -2301,7 +2301,7 @@
   if(fetchPrefsValue("dot.path", buf, sizeof(buf)) != -1) {
     snprintf(dotPath, sizeof(dotPath), "%s", buf);
   } else {
-    snprintf(dotPath, sizeof(dotPath), "/usr/local/bin/dot");
+    snprintf(dotPath, sizeof(dotPath), "/usr/bin/dot");
     storePrefsValue("dot.path", dotPath); /* Set the default */
   }
 
