diff -NurB darkice-0.12/src/MultiThreadedConnector.h darkice-0.12-patched/src/MultiThreadedConnector.h
--- darkice-0.12/src/MultiThreadedConnector.h	2002-10-19 08:35:21.000000000 -0500
+++ darkice-0.12-patched/src/MultiThreadedConnector.h	2003-01-14 03:29:53.000000000 -0500
@@ -180,15 +180,6 @@
         void
         strip ( void )                              throw ( Exception );
 
-        /**
-         *  This is the function for each thread.
-         *  This function has to return fast
-         *
-         *  @param ixSink the index of the sink this thread works on.
-         */
-        void
-        sinkThread( int     ixSink );
-
     protected:
 
         /**
@@ -206,6 +197,15 @@
     public:
 
         /**
+         *  This is the function for each thread.
+         *  This function has to return fast
+         *
+         *  @param ixSink the index of the sink this thread works on.
+         */
+        void
+        sinkThread( int     ixSink );
+
+        /**
          *  Constructor based on a Source.
          *
          *  @param source the source to connect to the sinks.
