diff -ruN MPlayer-1.0pre5try2_old/libmpdemux/asf_mmst_streaming.c MPlayer-1.0pre5try2/libmpdemux/asf_mmst_streaming.c
--- MPlayer-1.0pre5try2_old/libmpdemux/asf_mmst_streaming.c	2004-12-15 22:18:31.000000000 +0100
+++ MPlayer-1.0pre5try2/libmpdemux/asf_mmst_streaming.c	2005-04-16 09:43:16.000000000 +0200
@@ -334,8 +334,12 @@
 
       printf ("stream object, stream id: %d\n", stream_id);
 
+      if (num_stream_ids < 20) {
       stream_ids[num_stream_ids] = stream_id;
       num_stream_ids++;
+      } else {
+        printf ("too many id, stream skipped");
+      }
       
     } else {
       printf ("unknown object\n");
