Index: lftp_ssl.cc
===================================================================
RCS file: /home/lav/cvsroot/lftp/src/lftp_ssl.cc,v
retrieving revision 1.30
diff -u -p -r1.30 lftp_ssl.cc
--- lftp_ssl.cc	6 Jul 2006 13:52:02 -0000	1.30
+++ lftp_ssl.cc	2 Aug 2006 14:23:08 -0000
@@ -311,7 +311,8 @@ lftp_ssl_gnutls::~lftp_ssl_gnutls()
    if(handshake_done)
       gnutls_bye(session,GNUTLS_SHUT_RDWR);  // FIXME - E_AGAIN
 #endif
-   gnutls_certificate_free_credentials(cred);
+   if(cred)
+      gnutls_certificate_free_credentials(cred);
    gnutls_deinit(session);
 }
 
