--- src/Makefile.am.orig	2006-08-02 13:50:55.000000000 +1000
+++ src/Makefile.am	2006-08-02 13:52:33.000000000 +1000
@@ -77,7 +77,7 @@
  StringPool.cc StringPool.h DirColors.cc DirColors.h IdNameCache.cc\
  IdNameCache.h PatternSet.cc PatternSet.h LocalDir.cc LocalDir.h\
  mbswidth.c mbswidth.h wcwidth.h
-liblftp_tasks_la_LIBADD = $(TASK_MODULES_STATIC) $(top_builddir)/lib/liblib.la $(LTLIBINTL) $(LTLIBICONV)
+liblftp_tasks_la_LIBADD = $(TASK_MODULES_STATIC) $(top_builddir)/lib/liblib.la $(LTLIBINTL) $(LTLIBICONV) @SOCKSLIBS@
 
 liblftp_jobs_la_SOURCES = Job.cc Job.h CmdExec.cc CmdExec.h\
  commands.cc mgetJob.h mgetJob.cc SysCmdJob.cc SysCmdJob.h rmJob.cc rmJob.h\
