--- cowloop-3.0/src/Makefile.orig	2006-12-19 17:12:10.000000000 +1100
+++ cowloop-3.0/src/Makefile	2006-12-19 17:13:33.000000000 +1100
@@ -23,28 +23,28 @@
 # make -C $(KERNDIR) SUBDIRS=$(THISDIR) -I. -Wall modules
 
 cowdev:		cowdev.c version.h cowloop.h
-		$(CC) -I. -Wall -o cowdev cowdev.c
+		$(CC) $(CFLAGS) -I. -Wall -o cowdev cowdev.c
 
 cowwatch:	cowwatch.c version.h cowloop.h
-		$(CC) -I. -Wall -o cowwatch cowwatch.c
+		$(CC) $(CFLAGS) -I. -Wall -o cowwatch cowwatch.c
 
 cowctl:		cowctl.c version.h cowloop.h
-		$(CC) -I. -Wall -o cowctl cowctl.c
+		$(CC) $(CFLAGS) -I. -Wall -o cowctl cowctl.c
 
 cowsync:	cowsync.c version.h cowloop.h
-		$(CC) -I. -Wall -o cowsync cowsync.c
+		$(CC) $(CFLAGS) -I. -Wall -o cowsync cowsync.c
 
 cowlist:	cowlist.c version.h cowloop.h
-		$(CC) -I. -Wall -o cowlist cowlist.c
+		$(CC) $(CFLAGS) -I. -Wall -o cowlist cowlist.c
 
 cowrepair:	cowrepair.c version.h cowloop.h
-		$(CC) -I. -Wall -o cowrepair cowrepair.c
+		$(CC) $(CFLAGS) -I. -Wall -o cowrepair cowrepair.c
 
 cowmerge:	cowmerge.c version.h cowloop.h
-		$(CC) -I. -Wall -o cowmerge cowmerge.c
+		$(CC) $(CFLAGS) -I. -Wall -o cowmerge cowmerge.c
 
 cowpack:	cowpack.c version.h cowloop.h
-		$(CC) -I. -Wall -o cowpack cowpack.c -lz
+		$(CC) $(CFLAGS) -I. -Wall -o cowpack cowpack.c -lz
 
 #--------------------------------------------------------------------
 
