Index: kernel/scsi_transport_iscsi.c
===================================================================
--- kernel/scsi_transport_iscsi.c	(revision 438)
+++ kernel/scsi_transport_iscsi.c	(working copy)
@@ -1234,8 +1234,7 @@
 	if (err)
 		goto unregister_session_class;
 
-	nls = netlink_kernel_create(NETLINK_ISCSI, 1, iscsi_if_rx,
-			THIS_MODULE);
+	nls = netlink_kernel_create(NETLINK_ISCSI, iscsi_if_rx);
 	if (!nls) {
 		err = -ENOBUFS;
 		goto unregister_notifier;
