--- 4stAttack-2.1.4/4stattack.py	2002-10-01 07:38:03.000000000 -0700
+++ new/4stattack.py	2003-07-24 22:11:34.000000000 -0700
@@ -28,6 +28,9 @@
 images	= None
 screen	= None
 
+f = os.path.expanduser('~/.4stattackrc')
+if not os.path.exists(f):
+    os.system('cp /usr/share/games/4stattack/settings.ini %s' % f)
 
 def playMusic(datadir):
 	if pygame.mixer.get_init():
@@ -49,11 +52,11 @@
 		screen = pygame.display.set_mode(resolution, FULLSCREEN)
 	else:
 		screen = pygame.display.set_mode(resolution)
-	
+
 	pygame.display.set_caption('4st Attack 2')
 	pygame.mouse.set_visible(1)
 	pygame.display.init()
-	
+
 	pygame.display.Info()
 	return screen
 
@@ -72,21 +75,21 @@
 	pygame.quit()
 	os._exit(0)
 
-def main():                    
+def main():
 	options = getOptions(sys.argv)
-	
+
 	if options.has_key('-ini'):
 		ini_settings = IniSettings(options['-ini'])
 	else:
-		ini_settings = IniSettings('settings.ini')
-	
+		ini_settings = IniSettings(os.path.expanduser('~/.4stattackrc'))
+
 	settings = ini_settings.settings
-	
+
 	datadir = settings['path']['data']
-	
+
 	if settings['sound']['music'] == 'yes':
 		playMusic(datadir)
-	
+
 	resolution = settings['video']['resolution']
 	res = string.split(resolution, 'x')
 	screen = setDisplay((int(res[0]), int(res[1])), settings['video']['fullscreen'])
