diff -Naur kazehakase-org/src/actions/kz-actions.c kazehakase/src/actions/kz-actions.c
--- kazehakase-org/src/actions/kz-actions.c	2005-03-24 05:45:25.587573560 +0900
+++ kazehakase/src/actions/kz-actions.c	2005-03-24 05:15:02.000000000 +0900
@@ -1286,7 +1286,7 @@
 						  dirname);
 
 	filename = g_strconcat(g_get_home_dir(),
-			       "/.kazehakase/",
+			       "/.", PACKAGE, "/",
 			       dirname,
 			       "/",
 			       htmlname,
@@ -1372,7 +1372,7 @@
 
 	/* reload from current session file */
 	current_session  = g_build_filename(g_get_home_dir(),
-					    ".kazehakase",
+					    "."PACKAGE,
 					    "current_session.xml",
 					     NULL);
 	if (!g_file_test(current_session, G_FILE_TEST_EXISTS))
diff -Naur kazehakase-org/src/kazehakase.h kazehakase/src/kazehakase.h
--- kazehakase-org/src/kazehakase.h	2005-03-24 05:45:25.606570672 +0900
+++ kazehakase/src/kazehakase.h	2005-03-24 05:45:01.554227184 +0900
@@ -36,8 +36,8 @@
 
 #define KAZEHAKASE_URI "http://kazehakase.sourceforge.jp/"
 
-#define HISTORY_INDEX "/.kazehakase/history_index "
-#define HISTORY_DIR   "/.kazehakase/history/"
+#define HISTORY_INDEX "/."PACKAGE"/history_index "
+#define HISTORY_DIR   "/."PACKAGE"/history/"
 
 
 typedef enum {
diff -Naur kazehakase-org/src/kz-favicon.c kazehakase/src/kz-favicon.c
--- kazehakase-org/src/kz-favicon.c	2005-03-24 05:45:25.617569000 +0900
+++ kazehakase/src/kz-favicon.c	2005-03-24 05:14:42.000000000 +0900
@@ -97,7 +97,7 @@
 	KzFaviconPrivate *priv = KZ_FAVICON_GET_PRIVATE (kzfav);
 
 	favicon_file = g_build_filename(g_get_home_dir(),
-				        ".kazehakase",
+				        "."PACKAGE,
 					"faviconrc",
 				   	NULL);
 	priv->uri_list = kz_profile_open(favicon_file, NULL);
@@ -120,7 +120,7 @@
 	KzFaviconPrivate *priv = KZ_FAVICON_GET_PRIVATE (kzfav);
 	
 	favicon_file = g_build_filename(g_get_home_dir(),
-				   	".kazehakase",
+				   	"."PACKAGE,
 					"faviconrc",
 				   	NULL);
 	
@@ -357,7 +357,7 @@
 
 	favicon_filename = create_filename_from_uri(uri);
 	favicon_file_location = g_build_filename(g_get_home_dir(),
-					    	 ".kazehakase",
+					    	 "."PACKAGE,
 						 "favicon",
 					    	 favicon_filename,
 					    	 NULL);
diff -Naur kazehakase-org/src/kz-popup-preview.c kazehakase/src/kz-popup-preview.c
--- kazehakase-org/src/kz-popup-preview.c	2005-03-24 05:45:25.547579640 +0900
+++ kazehakase/src/kz-popup-preview.c	2005-03-24 05:14:42.000000000 +0900
@@ -503,7 +503,7 @@
 	tmp_name = egg_str_get_md5_str(uri);
 	image_name = g_strconcat(tmp_name, ".png", NULL);
 	popup_file_name = g_build_filename(g_get_home_dir(),
-					   ".kazehakase",
+					   "."PACKAGE,
 					   "popup",
 					   image_name,
 					   NULL);
diff -Naur kazehakase-org/src/kz-window.c kazehakase/src/kz-window.c
--- kazehakase-org/src/kz-window.c	2005-03-24 05:45:25.557578120 +0900
+++ kazehakase/src/kz-window.c	2005-03-24 05:14:42.000000000 +0900
@@ -766,7 +766,7 @@
 	case KZ_UI_LEVEL_CUSTOM:
 		action_str = "UILevelCustom";
 		ui_file = g_strconcat(g_get_home_dir(),
-				      "/.kazehakase/kz-ui.xml",
+				      "/.", PACKAGE, "/kz-ui.xml",
 				      NULL);
 		break;
 	case KZ_UI_LEVEL_EXPERT:
diff -Naur kazehakase-org/src/main.c kazehakase/src/main.c
--- kazehakase-org/src/main.c	2005-03-24 05:45:25.585573864 +0900
+++ kazehakase/src/main.c	2005-03-24 05:14:42.000000000 +0900
@@ -573,14 +573,15 @@
 
 	/* load prefs */
 	sysconf_file = g_strdup(KZ_SYSCONFDIR "/kzrc");
-	conf_file = g_build_filename(g_get_home_dir(), ".kazehakase",
+	conf_file = g_build_filename(g_get_home_dir(),
+				     "."PACKAGE,
 				     "kzrc", NULL);
 	
 	kz_global_profile = kz_profile_open(conf_file, sysconf_file);
 
 	/* load bookmarks */
 	bookmark_file         = g_build_filename(g_get_home_dir(),
-					    	 ".kazehakase",
+					    	 "."PACKAGE,
 					    	 "bookmarks.xml",
 					    	 NULL);
 	sys_bookmark_file     = g_build_filename(KZ_SYSCONFDIR,
@@ -588,25 +589,25 @@
 					    	 NULL);
 
         bookmark_bar_file     = g_build_filename(g_get_home_dir(),
-					    	 ".kazehakase",
+					    	 "."PACKAGE,
 						 "bookmarkbar.xml",
 					    	 NULL);
 	sys_bookmark_bar_file = g_build_filename(KZ_SYSCONFDIR,
 					    	 "bookmarkbar.xml",
 					    	 NULL);
 	clip_file             = g_build_filename(g_get_home_dir(),
-					    	 ".kazehakase",
+					    	 "."PACKAGE,
 						 "clip.xml",
 					    	 NULL);
         smartbookmark_file    = g_build_filename(g_get_home_dir(),
-					    	 ".kazehakase",
+					    	 "."PACKAGE,
 					    	 "smartbookmarks.xml",
 					    	 NULL);
 	sys_smartbookmark_file= g_build_filename(KZ_SYSCONFDIR,
 					    	 "smartbookmarks.xml",
 					    	 NULL);
 	current_session_file  = g_build_filename(g_get_home_dir(),
-					    	 ".kazehakase",
+						 "."PACKAGE,
 						 "current_session.xml",
 					    	 NULL);
 	
@@ -624,14 +625,14 @@
 
 	/* Load other prefs... */
 	accel_prefs_file = g_build_filename (g_get_home_dir(),
-					     ".kazehakase",
+					     "."PACKAGE,
 					     "keyaccelrc",
 					     NULL);
 	gtk_accel_map_load (accel_prefs_file);
 
 	sysproxy_file = g_build_filename(KZ_SYSCONFDIR, "proxyrc", NULL);
 	proxy_file = g_build_filename (g_get_home_dir(),
-				       ".kazehakase",
+				       "."PACKAGE,
 				       "proxyrc", NULL);
 	kz_proxy = kz_profile_open(proxy_file, sysproxy_file);
 
@@ -674,7 +675,7 @@
 	gchar *accel_prefs_file;
 
 	accel_prefs_file = g_build_filename (g_get_home_dir(),
-					     ".kazehakase",
+					     "."PACKAGE,
 					     "keyaccelrc",
 					     NULL);
 
@@ -722,7 +723,7 @@
 
 	/* reload from current session file */
 	current_session  = g_build_filename(g_get_home_dir(),
-					    ".kazehakase",
+					    "."PACKAGE,
 					    "current_session.xml",
 					     NULL);
 	if (!g_file_test(current_session, G_FILE_TEST_EXISTS))
diff -Naur kazehakase-org/src/mozilla/kz-mozwrapper.cpp kazehakase/src/mozilla/kz-mozwrapper.cpp
--- kazehakase-org/src/mozilla/kz-mozwrapper.cpp	2005-03-24 05:45:25.565576904 +0900
+++ kazehakase/src/mozilla/kz-mozwrapper.cpp	2005-03-24 05:15:30.000000000 +0900
@@ -1260,7 +1260,7 @@
 		gchar *filename = create_filename_from_uri(uri);
 
 		dir = g_strconcat(g_get_home_dir(),
-				  "/.kazehakase/",
+				  "/.", PACKAGE, "/",
 				  storedir, "/",
 				  type, /* images or css */ 
 				  "/",
diff -Naur kazehakase-org/src/mozilla/mozilla-prefs.cpp kazehakase/src/mozilla/mozilla-prefs.cpp
--- kazehakase-org/src/mozilla/mozilla-prefs.cpp	2005-03-24 05:45:25.564577056 +0900
+++ kazehakase/src/mozilla/mozilla-prefs.cpp	2005-03-24 05:15:41.000000000 +0900
@@ -49,7 +49,7 @@
 	}
 
 	g_snprintf(full_path, G_N_ELEMENTS(full_path),
-		   "%s/%s", home_path, ".kazehakase/mozilla");
+		   "%s/%s", home_path, "."PACKAGE"/mozilla");
 	gtk_moz_embed_set_profile_path(full_path, "kazehakase");
 
 	return TRUE;
