--- configure.orig	Tue Feb 27 14:08:13 2001
+++ configure.in	Tue Feb 27 14:09:50 2001
@@ -101,6 +101,9 @@
 if test "$ac_cv_path_SENDMAIL" != "" ; then
 AC_DEFINE_UNQUOTED(SENDMAIL,"$ac_cv_path_SENDMAIL")
 MAIL_CMD="$ac_cv_path_SENDMAIL"
+else
+AC_DEFINE_UNQUOTED(SENDMAIL,"/usr/sbin/snedmail")
+MAIL_CMD="/usr/sbin/sendmail"
 fi
 
 AC_SUBST(MAIL_CMD)
@@ -152,7 +155,9 @@
   AC_DEFINE(SPOOLDIR, "/usr/spool/cron")
   AC_MSG_RESULT(/usr/spool/cron)
 else
-  AC_ERROR(Cannot determine value for spool directory)
+  sp=/var/spool/cron
+  AC_DEFINE(SPOOLDIR, "/var/spool/cron")
+  AC_MSG_RESULT(/var/spool/cron)
 fi
 
 AC_MSG_CHECKING(jobdir)
