Index: konversation-0.19/konversation/src/channel.cpp
===================================================================
--- konversation-0.19.orig/konversation/src/channel.cpp
+++ konversation-0.19/konversation/src/channel.cpp
@@ -625,6 +625,9 @@ void Channel::completeNick()
 
     if(mode == 'c')
     {
+        // FIXME: dirty workaround to avoid a crash, but won't work as expected anyway
+        if ( oldPos > pos ) return;
+
         line.remove(oldPos,pos-oldPos);
         pos = oldPos;
     }
