--- konversation-1.0.1/konversation/src/konversationapplication.cpp	2006-10-06 18:43:29.000000000 +0200
+++ konversation-new/konversation/src/konversationapplication.cpp	2007-10-12 18:38:23.000000000 +0200
@@ -126,7 +126,7 @@
         emit appearanceChanged();
         mainWindow->show();
 
-        if (Preferences::showServerList()) mainWindow->openServerList();
+        bool openServerList = Preferences::showServerList();
 
         // handle autoconnect on startup
         Konversation::ServerGroupList serverGroups = Preferences::serverGroupList();
@@ -135,12 +135,18 @@
         {
             for (Konversation::ServerGroupList::iterator it = serverGroups.begin(); it != serverGroups.end(); ++it)
             {
-                if ((*it)->autoConnectEnabled()) connectToServer((*it)->id());
+                if ((*it)->autoConnectEnabled())
+                {
+                    openServerList = false;
+                    connectToServer((*it)->id());
+                }
             }
         }
         else
             quickConnectToServer(m_hostName, m_port, m_channel, m_nick, m_password, m_useSSL);
 
+        if (openServerList) mainWindow->openServerList();
+
         // prepare dcop interface
         dcopObject = new KonvDCOP;
         kapp->dcopClient()->setDefaultObject(dcopObject->objId());
