Index: camlimages-3.0.1/examples/liv/liv.ml
===================================================================
--- camlimages-3.0.1.orig/examples/liv/liv.ml
+++ camlimages-3.0.1/examples/liv/liv.ml
@@ -12,6 +12,10 @@
 
 (* $Id: camlimages-3.0.1-lablgtk.patch,v 1.1 2009/07/24 20:51:30 aballier Exp $ *)
 
+
+module D = Display
+open D
+
 open Images;;
 open OImages;;
 
@@ -21,14 +25,13 @@ open GMain;;
 
 open Livmisc;;
 open Gui;;
-open Display;;
 open Tout;;
 
 open Gc;;
 
 exception Skipped;;
 
-let base_filters = ref ([] : Display.filter list);;
+let base_filters = ref ([] : D.filter list);;
 
 let r = Gc.get () in
 r.max_overhead <- 0; Gc.set r;
@@ -52,15 +55,15 @@ Arg.parse [
   "-root",
     Arg.String
       (function
-       | "center" -> Display.root_mode := `CENTER
-       | "random" -> Display.root_mode := `RANDOM
+       | "center" -> D.root_mode := `CENTER
+       | "random" -> D.root_mode := `RANDOM
        | _ -> raise (Failure "root mode")),
     ": on root [center|random]";
   "-transition",
     Arg.String
       (function
-       | "myst" -> Display.transition := `MYST
-       | "transparent" -> Display.transition := `TRANSPARENT
+       | "myst" -> D.transition := `MYST
+       | "transparent" -> D.transition := `TRANSPARENT
        | _ -> raise (Failure "transition")),
     ": transition [myst|transparent]";
   "-transparentborder",
