--- WinQuake/sys_linux.c
+++ WinQuake/sys_linux.c
@@ -86,7 +86,7 @@
 void Sys_Printf (char *fmt, ...)
 {
 	va_list		argptr;
-	char		text[1024];
+	char		text[4096];
 	unsigned char		*p;
 
 	va_start (argptr,fmt);
--- QW/client/sys_linux.c
+++ QW/client/sys_linux.c
@@ -141,7 +141,7 @@
 void Sys_Error (char *error, ...)
 { 
     va_list     argptr;
-    char        string[1024];
+    char        string[4096];
 
 // change stdin to non blocking
     fcntl (0, F_SETFL, fcntl (0, F_GETFL, 0) & ~FNDELAY);
