diff -Nur metakit-2.4.9.3-orig/python/PyRowRef.cpp metakit-2.4.9.3/python/PyRowRef.cpp
--- metakit-2.4.9.3-orig/python/PyRowRef.cpp	2004-01-20 17:46:06.000000000 -0800
+++ metakit-2.4.9.3/python/PyRowRef.cpp	2006-07-03 23:40:57.000000000 -0700
@@ -24,12 +24,12 @@
 }
 
 static int PyRowRef_print(PyRowRef *o, FILE *f, int) {
-  fprintf(f, "<PyRowRef object at %x>", (int)o);
+  fprintf(f, "<PyRowRef object at %p>", (long)o);
   return 0;
 }
 
 static int PyRORowRef_print(PyRowRef *o, FILE *f, int) {
-  fprintf(f, "<PyRORowRef object at %x>", (int)o);
+  fprintf(f, "<PyRORowRef object at %p>", (long)o);
   return 0;
 }
 
diff -Nur metakit-2.4.9.3-orig/python/PyView.cpp metakit-2.4.9.3/python/PyView.cpp
--- metakit-2.4.9.3-orig/python/PyView.cpp	2004-01-20 17:46:06.000000000 -0800
+++ metakit-2.4.9.3/python/PyView.cpp	2006-07-03 23:40:57.000000000 -0700
@@ -971,15 +971,15 @@
 }
 
 static int PyView_print(PyView *o, FILE *f, int) {
-  fprintf(f, "<PyView object at %x>", (int)o);
+  fprintf(f, "<PyView object at %p", (long)o);
   return 0;
 }
 static int PyViewer_print(PyView *o, FILE *f, int) {
-  fprintf(f, "<PyViewer object at %x>", (int)o);
+  fprintf(f, "<PyViewer object at %p>", (long)o);
   return 0;
 }
 static int PyROViewer_print(PyView *o, FILE *f, int) {
-  fprintf(f, "<PyROViewer object at %x>", (int)o);
+  fprintf(f, "<PyROViewer object at %p>", (long)o);
   return 0;
 }
 
