Index: packages/semweb/rdf_db.c
===================================================================
RCS file: /usr/local/cvspl/pl/packages/semweb/rdf_db.c,v
retrieving revision 1.87
diff -u -p -r1.87 rdf_db.c
--- packages/semweb/rdf_db.c	12 Nov 2005 15:03:05 -0000	1.87
+++ packages/semweb/rdf_db.c	1 Dec 2005 08:52:40 -0000
@@ -4842,7 +4842,7 @@ typedef struct agenda
 } agenda;
 
 #ifndef offsetof
-#define offsetof(structure, field) ((int) &(((structure *)NULL)->field))
+#define offsetof(structure, field) ((unsigned long) &(((structure *)NULL)->field))
 #endif
 #define CHUNK_SIZE(n) offsetof(chunk, nodes[n])
 
