Fix make dependencies so `make -j` works

--- src/Makefile.in
+++ src/Makefile.in
@@ -161,7 +161,7 @@
 		@echo "****************"
 
 
-$(PL):		$(RUNTIME) $(PLOBJ)
+$(PL):		$(RUNTIME) $(PLOBJ) $(RUNTIMEDIR)/$(SHARED)
 		$(LDEXE) $(LDFLAGS) -o $(PL) $(PLOBJ) -L../lib/$(ARCH) -lpl $(LNLIBS)
 
 $(STARTUPPATH):	$(PLINIT) $(PLSRC) $(PL)
@@ -177,7 +177,7 @@
 		$(AR) r $@ $(OBJ) $(RCOBJ) $(XOBJ)
 		$(RANLIB) $@
 
-$(RUNTIMEDIR)/$(SHAREDV):	$(RUNTIMEDIR) $(OBJ) $(RCOBJ) $(XOBJ)
+$(RUNTIMEDIR)/$(SHAREDV):	dirs $(OBJ) $(RCOBJ) $(XOBJ)
 		$(CC) -shared $(COFLAGS) -o $@ $(SONAMEOPT) \
 			$(OBJ) $(RCOBJ) $(XOBJ) $(LIBS)
 
@@ -188,7 +188,7 @@
 rc-lib:
 		@(cd rc && $(MAKE) COFLAGS="$(COFLAGS)")
 
-../library/INDEX.pl:	../library ../library/[a-z]*.pl
+../library/INDEX.pl:	../library ../library/[a-z]*.pl $(DEVPL) $(STARTUPPATH)
 		$(DEVPL) -x $(STARTUPPATH) \
 			-f none -F none \
 			-g make_library_index\(\'../library\'\) \
@@ -243,11 +243,7 @@
 		@echo "#define C_LDFLAGS    \"$(LDFLAGS)\""    >> $@%
 		@if cmp -s $@% $@; then rm $@%; else mv $@% $@; fi
 
-pl-main.o:	parms.h
-pl-setup.o:	parms.h
-pl-wic.o:	parms.h
-pl-funct.o:	$(srcdir)/pl-funct.ih
-pl-atom.o:	$(srcdir)/pl-funct.ih
+$(OBJ): headers parms.h $(srcdir)/pl-funct.ih
 pl-wam.o:	pl-alloc.c pl-index.c pl-fli.c
 
 $(srcdir)/pl-funct.ih:	$(srcdir)/ATOMS
@@ -262,7 +258,7 @@
 frame.o:
 		ar x `$(CC) -print-libgcc-file-name` $@
 
-plld:		$(srcdir)/plld.c
+plld:		$(srcdir)/plld.c parms.h
 		$(CC) -I. -I$(srcdir) $(CFLAGS) -o $@ $(srcdir)/plld.c
 
 tags:		$(srcdir)/TAGS
