diff -Naur acpitool-0.2.6.orig/configure.in acpitool-0.2.6/configure.in
--- acpitool-0.2.6.orig/configure.in	2004-11-28 18:19:14.000000000 +0100
+++ acpitool-0.2.6/configure.in	2005-05-07 11:47:40.935057296 +0200
@@ -11,7 +11,7 @@
 
 # Put the following line in comment and rerun autoconf if you really want
 # to compile with -g (=debugging on, seems to be default)
-CXXFLAGS=""
+#CXXFLAGS=""
 
 # Checks for libraries.
 
diff -Naur acpitool-0.2.6.orig/src/Makefile.am acpitool-0.2.6/src/Makefile.am
--- acpitool-0.2.6.orig/src/Makefile.am	2004-08-29 01:49:11.000000000 +0200
+++ acpitool-0.2.6/src/Makefile.am	2005-05-07 11:47:53.195193472 +0200
@@ -2,7 +2,7 @@
 
 acpitool_SOURCES = main.cpp acpitool.cpp toshiba.cpp
 
-AM_CXXFLAGS = -O2 -Wall -march=i686
+AM_CXXFLAGS = ${CFLAGS}
 
 EXTRA_DIST = acpitool.h toshiba.h
 
