diff -u ipw2100-1.2.1/Makefile ipw2100-1.2.1-double-parentheses/Makefile
--- ipw2100-1.2.1/Makefile	2006-02-21 19:46:16.000000000 +0100
+++ ipw2100-1.2.1-double-parentheses/Makefile	2006-07-04 19:14:17.000000000 +0200
@@ -150,7 +150,7 @@
 else # We've got a kernel with seperate output, copy the config, and use O=
 	mkdir -p $(TMP)
 	cp $(KSRC_OUTPUT)/.config $(TMP)
-	$(MAKE) -C $(KSRC) M=$(PWD) MODVERDIR=$(MODVERDIR)) O=$(PWD)/tmp modules
+	$(MAKE) -C $(KSRC) M=$(PWD) MODVERDIR=$(MODVERDIR) O=$(PWD)/tmp modules
 endif
 
 install: modules
