diff -urp ipw2100-0.55/ipw2100.c ipw2100-0.55-modparam/ipw2100.c
--- ipw2100-0.55/ipw2100.c	2004-09-28 01:29:17.000000000 +0800
+++ ipw2100-0.55-modparam/ipw2100.c	2004-09-28 13:07:17.989689056 +0800
@@ -250,12 +250,12 @@ MODULE_PARM(disable, "i");
 #else /* LINUX_VERSION_CODE < 2.6.0 */
 
 #include <linux/moduleparam.h>
-module_param(debug, int, -1);
-module_param(if_name, charp, 0);
-module_param(mode, int, 0);
-module_param(channel, int, 1);
-module_param(associate, int, 1);
-module_param(disable, int, 1);
+module_param(debug, int, 0444);
+module_param(if_name, charp, 0444);
+module_param(mode, int, 0444);
+module_param(channel, int, 0444);
+module_param(associate, int, 0444);
+module_param(disable, int, 0444);
 
 #endif /* LINUX_VERSION_CODE < 2.6.0 */
 
