Fix compile with ieee80211-1.0.3

Signed-off-by: Zhu Yi <yi.zhu@intel.com>
--
diff -urp ipw2100-1.1.2/ipw2100.h ipw2100-1.1.2-header_data/ipw2100.h
--- ipw2100-1.1.2/ipw2100.h	2005-07-14 02:59:37.000000000 +0800
+++ ipw2100-1.1.2-header_data/ipw2100.h	2005-08-08 14:03:57.000000000 +0800
@@ -786,7 +786,7 @@ struct ipw2100_priv {
 
 
 
-#define IPW_HEADER_802_11_SIZE		 sizeof(struct ieee80211_header_data)
+#define IPW_HEADER_802_11_SIZE		 sizeof(struct ieee80211_hdr_3addr)
 #define IPW_MAX_80211_PAYLOAD_SIZE              2304U
 #define IPW_MAX_802_11_PAYLOAD_LENGTH		2312
 #define IPW_MAX_ACCEPTABLE_TX_FRAME_LENGTH	1536
@@ -801,7 +801,7 @@ struct ipw2100_priv {
 		IPW_802_11_FCS_LENGTH)
 
 #define IPW_802_11_PAYLOAD_OFFSET \
-        (sizeof(struct ieee80211_header_data) + \
+        (sizeof(struct ieee80211_hdr_3addr) + \
          sizeof(struct ieee80211_snap_hdr))
 
 struct ipw2100_rx {
