diff -urp ipw2100-1.0.5/ieee80211.h ipw2100-1.0.5-is_multicast_ether_addr/ieee80211.h
--- ipw2100-1.0.5/ieee80211.h	2005-02-08 21:38:23.000000000 +0100
+++ ipw2100-1.0.5-is_multicast_ether_addr/ieee80211.h	2005-06-24 23:18:54.000000000 +0200
@@ -643,10 +643,12 @@ enum ieee80211_state {
 #define MAC_ARG(x) ((u8*)(x))[0],((u8*)(x))[1],((u8*)(x))[2],((u8*)(x))[3],((u8*)(x))[4],((u8*)(x))[5]
 
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,12)
 extern inline int is_multicast_ether_addr(const u8 *addr)
 {
 	return ((addr[0] != 0xff) && (0x01 & addr[0]));
 }
+#endif
 
 extern inline int is_broadcast_ether_addr(const u8 *addr)
 {
