diff -ur lisp2wish.orig/package.lisp lisp2wish/package.lisp
--- lisp2wish.orig/package.lisp	2004-01-30 17:09:07.000000000 -0600
+++ lisp2wish/package.lisp	2005-02-04 16:34:30.588988396 -0600
@@ -1,10 +1,8 @@
 
-(in-package #-:sbcl "USER" #+:sbcl "COMMON-LISP-USER")
+(in-package #:cl-user)
 
-#-:KCL
 (defpackage WISH
-  (:use #-:sbcl LISP #+:sbcl "COMMON-LISP" #+:ALLEGRO EXCL #+:LUCID LCL)
-  #-(or :cmu :sbcl :clisp) (:shadowing-import-from  USER  TEST-WISH)
+  (:use #:common-lisp)
   (:export "*WISH*"
 	   "*WISH-PROG*"
 	   "*WISH-ARGS*"
@@ -15,21 +13,3 @@
 	   "WITH-OUTPUT-TO-WISH"
 	   "LISTEN-TO-WISH"
 	   "DEFAULT-WISH-LISTENER"))
-
-#+:KCL
-(eval-when (load eval compile)
-  (unless (find-package "WISH")
-    'TEST-WISH
-    (make-package "WISH")
-    (in-package "WISH")
-    (shadowing-import '(USER::TEST-WISH))
-    (export (mapcar #'intern '("*WISH*"
-			       "*WISH-PROG*"
-			       "*WISH-ARGS*"
-			       "*WISH-EXIT-MSG*"
-			       "*WISH-LISTENER*"
-			       "SEND-TO-WISH"
-			       "WITH-WISH"
-			       "WITH-OUTPUT-TO-WISH"
-			       "LISTEN-TO-WISH"
-			       "DEFAULT-WISH-LISTENER")))))
