diff -uNr jaaa-0.1.2-orig/prbsgen.h jaaa-0.1.2/prbsgen.h
--- jaaa-0.1.2-orig/prbsgen.h	2006-05-26 12:20:34.000000000 +0100
+++ jaaa-0.1.2/prbsgen.h	2006-05-26 12:20:50.000000000 +0100
@@ -123,8 +123,8 @@
   void sync_forw (U32 bits);
   void sync_back (U32 bits);
   int  step (void);
-  void Prbsgen::crc_in (int b);
-  int  Prbsgen::crc_out (void);
+  void crc_in (int b);
+  int  crc_out (void);
 
   U32 stat (void) const;
   U32 poly (void) const;
