--- captive-1.1.7/src/client/cmdline/Makefile.in	2006-01-26 16:20:40.000000000 -0500
+++ captive-1.1.7/src/client/cmdline/Makefile.in	2006-09-12 10:44:22.110968408 -0400
@@ -459,7 +459,7 @@
 		utf8.h
 
 captive_cmdline_CFLAGS = $(GNOME_VFS_CFLAGS) $(LIBXML_CFLAGS) 
-captive_cmdline_LDADD = $(captive_library) $(GNOME_VFS_LIBS)   $(LIBXML_LIBS)   $(READLINE_LIBS) $(INTLLIBS)
+captive_cmdline_LDADD = $(captive_library) $(GNOME_VFS_LIBS)   $(LIBXML_LIBS)   $(READLINE_LIBS) $(INTLLIBS) $(POPT_LIBS)
 captive_cmdline_LDFLAGS = $(READLINE_LDFLAGS)
 @ENABLE_MAN_PAGES_TRUE@captive_cmdline_man_cond = captive-cmdline.1
 man_MANS = $(captive_cmdline_man_cond)
--- captive-1.1.7/src/client/sandbox-server/Makefile.in	2006-01-26 16:20:43.000000000 -0500
+++ captive-1.1.7/src/client/sandbox-server/Makefile.in	2006-09-12 10:50:15.509243704 -0400
@@ -401,7 +401,7 @@
 		main.c
 
 captive_sandbox_server_CFLAGS = $(ORBIT_CFLAGS) $(LINC_CFLAGS) $(GNOME_VFS_CFLAGS) $(LIBXML_CFLAGS) 
-captive_sandbox_server_LDADD = $(ORBIT_LIBS)   $(LINC_LIBS)   $(GNOME_VFS_LIBS)   $(LIBXML_LIBS)   $(captive_library) $(INTLLIBS)
+captive_sandbox_server_LDADD = $(ORBIT_LIBS)   $(LINC_LIBS)   $(GNOME_VFS_LIBS)   $(LIBXML_LIBS)   $(captive_library) $(INTLLIBS) $(POPT_LIBS)
 @ENABLE_MAN_PAGES_TRUE@captive_sandbox_server_man_cond = captive-sandbox-server.1
 man_MANS = $(captive_sandbox_server_man_cond)
 all: $(BUILT_SOURCES)
--- captive-1.1.7/src/client/fuse/Makefile.in	2006-01-26 16:20:41.000000000 -0500
+++ captive-1.1.7/src/client/fuse/Makefile.in	2006-09-12 10:55:57.854199360 -0400
@@ -505,7 +505,7 @@
 mount_captive_CFLAGS = $(GNOME_VFS_CFLAGS) $(LIBXML_CFLAGS) \
 	$(FUSE_CFLAGS) -DLIBEXECDIR="$(libexecdir)" \
 	-DVARLIBCAPTIVEDIR="$(localstatedir)/lib/$(PACKAGE)"
-mount_captive_LDADD = $(captive_library) $(GNOME_VFS_LIBS)   $(LIBXML_LIBS)   $(FUSE_LIBS)   $(INTLLIBS)
+mount_captive_LDADD = $(captive_library) $(GNOME_VFS_LIBS)   $(LIBXML_LIBS)   $(FUSE_LIBS)   $(INTLLIBS)   $(POPT_LIBS)
 mount_captive_LDFLAGS = $(READLINE_LDFLAGS)
 @ENABLE_FUSE_TRUE@@ENABLE_SBIN_MOUNT_TRUE@mount_captive_cond = mount.captive
 @ENABLE_FUSE_TRUE@@ENABLE_SBIN_MOUNT_TRUE@initd_SCRIPTS_cond = captive
