--- WordNet-2.1/include/wn.h	2005-07-06 14:48:44.000000000 -0400
+++ /var/tmp/portage/wordnet-2.1/work/WordNet-2.1/include/wn.h	2006-01-09 19:31:15.000000000 -0500
@@ -30,7 +30,7 @@
 #define CNTLISTFILE     "%s\\cntlist.rev"
 #else
 #define DICTDIR         "/dict"
-#define DEFAULTPATH	"/usr/local/WordNet-2.1/dict"
+#define DEFAULTPATH	"/usr/dict"
 #define DATAFILE	"%s/data.%s"
 #define INDEXFILE	"%s/index.%s"
 #define SENSEIDXFILE	"%s/index.sense"
