diff -rU5 IcePy-3.2.0-orig/Makefile IcePy-3.2.0/Makefile
--- IcePy-3.2.0-orig/Makefile	2007-03-02 08:11:25.000000000 +1300
+++ IcePy-3.2.0/Makefile	2007-07-20 22:54:53.000000000 +1200
@@ -38,10 +38,11 @@
 	    echo "making $@ in $$subdir"; \
 	    ( cd $$subdir && $(MAKE) $@ ) || exit 1; \
 	done
 
 install::
-	$(call installdata,ICE_LICENSE,$(prefix))
-	$(call installdata,LICENSE,$(prefix))
+	mkdir -p $(install_slicedir)
+	$(call installdata,ICE_LICENSE,$(install_slicedir))
+	$(call installdata,LICENSE,$(install_slicedir))
 
 test::
 	@python $(top_srcdir)/allTests.py
diff -rU5 IcePy-3.2.0-orig/config/Make.rules IcePy-3.2.0/config/Make.rules
--- IcePy-3.2.0-orig/config/Make.rules	2007-03-02 08:11:25.000000000 +1300
+++ IcePy-3.2.0/config/Make.rules	2007-07-20 22:46:28.000000000 +1200
@@ -16,17 +16,17 @@
 #
 # Select an installation base directory. The directory will be created
 # if it does not exist.
 #
 
-prefix			= /opt/IcePy-$(VERSION)
+prefix			= $(DESTDIR)/usr
 
 #
 # The "root directory" for runpath embedded in executables. Can be unset
 # to avoid adding a runpath to Ice executables.
 #
-embedded_runpath_prefix  ?= /opt/Ice-$(VERSION_MAJOR).$(VERSION_MINOR)
+#embedded_runpath_prefix  ?= /opt/Ice-$(VERSION_MAJOR).$(VERSION_MINOR)
 
 #
 # Define OPTIMIZE as yes if you want to build with optimization.
 # Otherwise the Ice extension is built with debug information.
 #
@@ -99,12 +99,12 @@
 VERSION_MINOR           = 2
 VERSION			= 3.2.0
 SOVERSION		= 32
 libdir			= $(top_srcdir)/python
 
-install_slicedir	= $(prefix)/slice
-install_pythondir	= $(prefix)/python
+install_slicedir	= $(prefix)/share/icepy-$(VERSION)/slice
+install_pythondir	= $(prefix)/lib/$(PYTHON_VERSION)/site-packages
 
 INSTALL			= cp -fp
 INSTALL_PROGRAM		= ${INSTALL}
 INSTALL_LIBRARY		= ${INSTALL}
 INSTALL_DATA		= ${INSTALL}
@@ -136,22 +136,22 @@
 endif
 
 ifneq ($(ICE_HOME),)
     slicedir		= $(ICE_HOME)/slice
 else
-    slicedir		= /usr/share/Ice-$(VERSION)/slice
+    slicedir		= /usr/share/Ice/slice
 endif
 
-install_libdir	        = $(prefix)/python
+install_libdir	        = $(prefix)/lib
 
-ifneq ($(embedded_runpath_prefix),)
-   ifeq ($(LP64),yes)
-       runpath_libdir	  := $(embedded_runpath_prefix)/lib$(lp64suffix)
-   else
-       runpath_libdir	  := $(embedded_runpath_prefix)/lib
-   endif
-endif
+#ifneq ($(embedded_runpath_prefix),)
+#   ifeq ($(LP64),yes)
+#       runpath_libdir	  := $(embedded_runpath_prefix)/lib$(lp64suffix)
+#   else
+#       runpath_libdir	  := $(embedded_runpath_prefix)/lib
+#   endif
+#endif
 
 CPPFLAGS		=
 ICECPPFLAGS		= -I$(slicedir)
 SLICE2PYFLAGS		= $(ICECPPFLAGS)
 
@@ -198,11 +198,11 @@
     installprogram	= $(INSTALL_PROGRAM) $(1) $(2); \
 			  chmod a+rx $(2)/$(notdir $(1))
 endif
 
 ifeq ($(mkdir),)
-    mkdir		= mkdir $(1) ; \
+    mkdir		= mkdir -p $(1) ; \
 			  chmod a+rx $(1)
 endif
 
 #
 # A Python extension library cannot have a "lib" prefix, so Python-specific
diff -rU5 IcePy-3.2.0-orig/modules/IcePy/Makefile IcePy-3.2.0/modules/IcePy/Makefile
--- IcePy-3.2.0-orig/modules/IcePy/Makefile	2007-01-09 08:25:09.000000000 +1300
+++ IcePy-3.2.0/modules/IcePy/Makefile	2007-07-20 23:03:25.000000000 +1200
@@ -51,7 +51,8 @@
 	rm -f $@
 	ln -s $(SONAME) $@
 
 install:: all
 	$(call installlib,$(install_libdir),$(libdir),$(LIBFILENAME),$(SONAME),$(LIBNAME))
+	ln -s $(install_libdir)/$(SONAME) $(install_pythondir)/$(LIBNAME)
 
 include .depend
