--- Makefile.org	2002-10-31 13:03:27.000000000 +0100
+++ Makefile	2002-10-31 13:03:51.000000000 +0100
@@ -62,13 +62,6 @@
 
 install: $(PROG)
 	@echo
-	@echo "Where should the iperf binary be installed? [/usr/local/bin]"
-	@read  INSTALL_DIR;                                                 \
-	export INSTALL_DIR;                                                 \
-	INSTALL_DIR=`echo $$INSTALL_DIR | perl -p -e 's/~/$$ENV{"HOME"}/'`; \
-	if test -z "$$INSTALL_DIR"; then                                    \
-	  INSTALL_DIR=/usr/local/bin;                                       \
-	fi;                                                                 \
 	echo ; (cd src  ; make install);                                    \
 	echo ; (cd jsrc ; make install)
 
