diff --exclude='*~' --exclude='.*' -I '$Id:' -urN dvxml-0.1.4.orig/doc/Makefile.in dvxml-0.1.4/doc/Makefile.in
--- dvxml-0.1.4.orig/doc/Makefile.in	2005-05-07 04:35:42.000000000 -0400
+++ dvxml-0.1.4/doc/Makefile.in	2005-05-07 04:19:14.000000000 -0400
@@ -310,7 +310,7 @@
 	uninstall-info-am uninstall-nobase_dist_pkgdataDATA
 
 install-exec-local: # ensures this is done before install-data
-	$(mkinstalldirs) $(pkgdatadir)/download
+	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)/download
 
 ############################################
 # edit -l options of installdox as needed
diff --exclude='*~' --exclude='.*' -I '$Id:' -urN dvxml-0.1.4.orig/Makefile.in dvxml-0.1.4/Makefile.in
--- dvxml-0.1.4.orig/Makefile.in	2005-05-07 04:07:22.000000000 -0400
+++ dvxml-0.1.4/Makefile.in	2005-05-07 04:39:00.000000000 -0400
@@ -135,7 +135,7 @@
 xmlwrapp_LIB_DIR = @xmlwrapp_LIB_DIR@
 
 # $Id: dvxml-0.1.4-gentoo.diff,v 1.1 2005/05/07 14:12:35 ka0ttic Exp $
-SUBDIRS = dvxml m4 doc
+SUBDIRS = dvxml m4
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
