From a468c8aeb1ef35ed3f314d4ea4f47fc04c403d96 Mon Sep 17 00:00:00 2001
From: Holger Hetterich <hhetter@novell.com>
Date: Tue, 7 Jun 2011 14:48:28 +0200
Subject: [PATCH] Change the order of the divisions


diff --git a/src/common/common.c b/src/common/common.c
index 4e6cb07..65aae46 100644
--- a/src/common/common.c
+++ b/src/common/common.c
@@ -36,30 +36,34 @@ char *common_make_human_readable( TALLOC_CTX *ctx, long long int kb )
 {
 	char kbstring[20];
 	char *output;
-	long long int result = 0;
+	long long int result = kb;
 	long long int rest = 0;
 	lldiv_t diff;
 	strcpy(kbstring,"Bytes");
-	if (kb >= 1024) { diff =  lldiv(kb, (long long) 1024); // kb
-			strcpy(kbstring,"KB");
-			result = diff.quot;
-			rest = diff.rem;
-			}
-	if (kb >= 1024) { diff = lldiv(kb,(long long) 1024*1024); // mb
-			strcpy(kbstring,"MB");
-			result = diff.quot;
-			rest = diff.rem;
-		}
-	if (kb >= 1024) {
-	                diff = lldiv(kb,(long long) 1024*1024*1024); // gb
-	                strcpy(kbstring,"GB");}
-			result = diff.quot;
-			rest = diff.rem;
-	if (kb >= 1024) {
-	                diff = lldiv(kb,(long long ) 1024*1024*1024*1024); // tb
-	                strcpy(kbstring,"TB");}
-			result = diff.quot;
-			rest = diff.rem;
+	if (kb >= (long long )1024*1024*1024*1024) {
+                diff = lldiv(kb,(long long ) 1024*1024*1024*1024); // tb
+                strcpy(kbstring,"TB");
+		result = diff.quot;
+		rest = diff.rem;
+	} else 
+	if (kb >= (long long )1024*1024*1024) {
+                diff = lldiv(kb,(long long) 1024*1024*1024); // gb
+                strcpy(kbstring,"GB");
+		result = diff.quot;
+		rest = diff.rem;
+	} else
+	if (kb >= (long long) 1024*1024) {
+		diff = lldiv(kb,(long long) 1024*1024); // mb
+		strcpy(kbstring,"MB");
+		result = diff.quot;
+		rest = diff.rem;
+	} else
+	if (kb >= 1024) { 
+		diff =  lldiv(kb, (long long) 1024); // kb
+		strcpy(kbstring,"KB");
+		result = diff.quot;
+		rest = diff.rem;
+	}
 	output = talloc_asprintf( ctx,"%lli.%lli %s",result,rest,kbstring);
 	return output;
 }
-- 
1.7.5.3

