diff -ur musicman-orig/musicman/Makefile.in musicman/musicman/Makefile.in
--- musicman-orig/musicman/Makefile.in	2004-07-07 23:44:30.716251088 +0200
+++ musicman/musicman/Makefile.in	2004-07-07 23:41:52.214347040 +0200
@@ -312,7 +312,7 @@
 #>- 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
 #>- 	$(AM_CXXFLAGS) $(CXXFLAGS)
 #>+ 3
-LTCXXCOMPILE = $(LIBTOOL) --mode=compile --tag=CXX $(CXX) $(DEFS) \
+LTCXXCOMPILE = $(LIBTOOL) --mode=compile --tag=CXX $(CXX) -fPIC $(DEFS) \
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
 	$(AM_CXXFLAGS) $(CXXFLAGS) $(KDE_CXXFLAGS)
 CXXLD = $(CXX)
diff -ur musicman-orig/musicman/kcmmusicman/Makefile.in musicman/musicman/kcmmusicman/Makefile.in
--- musicman-orig/musicman/kcmmusicman/Makefile.in	2004-07-07 23:44:31.811084648 +0200
+++ musicman/musicman/kcmmusicman/Makefile.in	2004-07-07 23:39:34.050351160 +0200
@@ -285,7 +285,7 @@
 #>- 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
 #>- 	$(AM_CXXFLAGS) $(CXXFLAGS)
 #>+ 3
-LTCXXCOMPILE = $(LIBTOOL) --mode=compile --tag=CXX $(CXX) $(DEFS) \
+LTCXXCOMPILE = $(LIBTOOL) --mode=compile --tag=CXX $(CXX) -fPIC $(DEFS) \
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
 	$(AM_CXXFLAGS) $(CXXFLAGS) $(KDE_CXXFLAGS)
 CXXLD = $(CXX)
--- musicman-orig/admin/acinclude.m4.in	2004-07-08 00:20:08.385276000 +0200
+++ musicman/admin/acinclude.m4.in	2004-07-08 00:41:30.770323840 +0200
@@ -2774,7 +2774,7 @@
      LDFLAGS=""
   fi
 
-  CXXFLAGS=" $CXXFLAGS"
+  CXXFLAGS=" $CXXFLAGS -fPIC"
 
   AC_PROG_CXX
 
