--- interceptor.c.orig	2003-10-27 14:59:18.474879600 -0500
+++ interceptor.c	2003-10-27 14:59:05.186899680 -0500
@@ -322,12 +322,12 @@
 
     dev_add_pack(&dummy_pt);
     /* this should be the original IP packet handler */
-    default_pt = dummy_pt.next;
+    default_pt = list_entry(dummy_pt.list.next, struct packet_type, list);
     /* there may be more than one other packet handler in our bucket,
      * so look through all the buckets */
     while (default_pt != NULL && default_pt->type != htons(ETH_P_IP))
     {
-        default_pt = default_pt->next;
+        default_pt = list_entry((*default_pt).list.next, struct packet_type, list);
     }
     if (!default_pt)
     {
