--- interceptor.c.orig	2004-02-24 17:34:43.492845856 -0500
+++ interceptor.c	2004-02-24 17:37:45.465181856 -0500
@@ -364,11 +364,6 @@
         error = VPNIFUP_FAILURE;
         goto error_exit;
     }
-    error = register_netdevice_notifier(&interceptor_notifier);
-    if (error)
-    {
-        goto error_exit;
-    }
 
     vpn_is_up = TRUE;
     return error;
@@ -388,8 +383,6 @@
 {
     int i;
 
-    unregister_netdevice_notifier(&interceptor_notifier);
-
     cleanup_frag_queue();
     /*restore IP packet handler */
     if (original_ip_handler.pt != NULL)
@@ -436,6 +429,9 @@
 {
     struct net_device *dev = (struct net_device *) val;
 
+    if (!vpn_is_up)
+        return 1;
+
     switch (event)
     {
     case NETDEV_REGISTER:
@@ -853,6 +849,8 @@
         CNICallbackTable = *PCNICallbackTable;
         CniPluginDeviceCreated();
 
+        register_netdevice_notifier(&interceptor_notifier);
+
         if ((status = register_netdev(&interceptor_dev)) != 0)
         {
             printk(KERN_INFO "%s: error %d registering device \"%s\".\n",
@@ -876,6 +874,9 @@
     CniPluginUnload();
 
     unregister_netdev(&interceptor_dev);
+
+    unregister_netdevice_notifier(&interceptor_notifier);
+
     return;
 }
 
