Index: noatun-plugins/configure.in.in
===================================================================
--- noatun-plugins/configure.in.in	(revision 428268)
+++ noatun-plugins/configure.in.in	(working copy)
@@ -173,14 +173,20 @@
 noatun_sdl=no
 dnl Check for SDL
 dnl require SDL 1.2 just because it's the new stable version, and it's what I have
-NOATUN_PATH_SDL(
-[1.2.0],
-[
-noatun_sdl=yes
-],
-[
-]
-)
+
+AC_ARG_WITH(sdl,
+  [AS_HELP_STRING(--with-sdl,
+    [enable SDL Noatun plugins @<:@default=check@:>@])],
+  [], with_sdl=check)
+
+if test "x$with_sdl" != xno; then
+  NOATUN_PATH_SDL([1.2.0], [noatun_sdl=yes], [])
+
+  if test "x$with_sdl" != xcheck && test "x$noatun_sdl" != xyes; then
+    AC_MSG_FAILURE([--with-sdl was given, but test for SDL failed])
+  fi
+fi
+
 AM_CONDITIONAL(include_SDL, test "$noatun_sdl" = yes)
 
 AC_CHECK_HEADERS(endian.h)
