diff -Naur slmodem-2.9.9b.old/modem/modem.c slmodem-2.9.9b/modem/modem.c
--- slmodem-2.9.9b.old/modem/modem.c	2004-06-29 21:13:07.000000000 +0800
+++ slmodem-2.9.9b/modem/modem.c	2005-04-30 14:16:57.000000000 +0800
@@ -1599,9 +1599,9 @@
 }
 
 
+static int sregs_init(unsigned char sregs[]);
 int modem_reset(struct modem *m)
 {
-	static int sregs_init(unsigned char sregs[]);
 	MODEM_DBG("modem reset...\n");
 	if(m->state != STATE_MODEM_IDLE)
 		modem_hup(m,1);
