diff -Nru slmodem-2.9.11-20051101.orig/drivers/amrmo_init.c slmodem-2.9.11-20051101/drivers/amrmo_init.c
--- slmodem-2.9.11-20051101.orig/drivers/amrmo_init.c	2005-10-10 00:14:13.000000000 +0300
+++ slmodem-2.9.11-20051101/drivers/amrmo_init.c	2006-09-09 18:13:47.219081000 +0300
@@ -53,7 +53,13 @@
 #include <linux/poll.h>
 #include <linux/fs.h>
 #include <asm/uaccess.h>
-#include <linux/devfs_fs_kernel.h>
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,18)
+#	include <linux/devfs_fs_kernel.h>
+#else
+#	define devfs_mk_cdev(dev, mode, fmt, i)
+#	define devfs_remove(fmt, i)
+#endif
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0)
 #define OLD_KERNEL 1
diff -Nru slmodem-2.9.11-20051101.orig/drivers/st7554.c slmodem-2.9.11-20051101/drivers/st7554.c
--- slmodem-2.9.11-20051101.orig/drivers/st7554.c	2005-11-01 17:31:45.000000000 +0200
+++ slmodem-2.9.11-20051101/drivers/st7554.c	2006-09-09 18:14:36.910186500 +0300
@@ -55,7 +55,13 @@
 #include <linux/poll.h>
 #include <linux/usb.h>
 #include <linux/device.h>
-#include <linux/devfs_fs_kernel.h>
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,18)
+#	include <linux/devfs_fs_kernel.h>
+#else
+#	define devfs_mk_cdev(dev, mode, fmt, i)
+#	define devfs_remove(fmt, i)
+#endif
 
 #include <modem_defs.h>
 
