diff -ur cdcover.orig/Makefile cdcover/Makefile
--- cdcover.orig/Makefile	2008-05-14 21:16:55.000000000 +0300
+++ cdcover/Makefile	2008-06-07 12:17:32.000000000 +0300
@@ -9,8 +9,8 @@
 IMAGES=./share/images
 TEMPLATES=./share/templates
 
-DOCFILES=BUGS COPYING INSTALL README VERSION
-IMAGEFILES=autonum.gif cddb.gif exit.gif gv.gif logo.gif nocddb.gif ps.gif
+DOCFILES=BUGS README VERSION
+IMAGEFILES=application-exit.gif cddb.gif document-print-preview.gif document-print-preview.png document-save.gif document-save.png logo.gif nocddb.gif view-sort-ascending.gif view-sort-ascending.png
 TEMPLATEFILES=CD-Classic.pst CD-Modern.pst CD-Tracktime.pst CD-TwoColumns.pst DVD-Classic.pst
 
 INSTALL=install -D
