diff --git a/src/Makefile.in b/src/Makefile.in
index 8f273eb..d005731 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -95,7 +95,7 @@ package:	$(tcldom_LIB_FILE)
 
 $(tcldom_LIB_FILE): $(tcldom_OBJECTS)
 	-rm -f $(tcldom_LIB_FILE)
-	$(SHLIB_LD) $(SHLIB_LDFLAGS) @SHLIB_LD_OUT@$@ \
+	$(SHLIB_LD) $(SHLIB_LDFLAGS) $(LDFLAGS) @SHLIB_LD_OUT@$@ \
 		$(tcldom_OBJECTS) $(tcldom_LIBRARIES) $(SHLIB_LD_LIBS)
 
 ##  Documentation section.
