--- luminance-hdr-2.2.0.orig/CMakeLists.txt
+++ luminance-hdr-2.2.0/CMakeLists.txt
@@ -27,30 +27,6 @@
 
 set(LIBS ${LIBS} ${QT_QTCORE_LIBRARIES} ${QT_QTGUI_LIBRARIES} ${QT_QTWEBKIT_LIBRARIES} ${QT_QTXML_LIBRARIES} ${QT_QTSQL_LIBRARIES})
 
-FIND_PACKAGE(Git)
-#FIND_PACKAGE(Git QUIET)
-IF(NOT GIT_FOUND)
-	message(STATUS "Git not found")
-    FIND_FILE(GIT_EXECUTABLE git.exe 
-        $ENV{CYGWIN_DIR}\\bin
-        $ENV{CYGWIN_DIR}
-    )
-    IF (EXISTS ${GIT_EXECUTABLE})
-        SET(GIT_FOUND 1)
-    ENDIF()
-    FIND_FILE(GIT_EXE git
-        $ENV{/usr/bin}
-    )
-    IF (EXISTS ${GIT_EXE})
-        SET(GIT_FOUND 1)
-    ENDIF()
-ENDIF()
-
-IF(GIT_FOUND)
-    include(GetGitRevisionDescription)
-    get_git_head_revision(GIT_REFSPEC GIT_SHA1)
-ENDIF()
-
 IF(WIN32)
 # Windows Section
 	SET(OPENEXR_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/../DEPs/include/OpenEXR")
@@ -318,7 +294,6 @@
 	src/UI/UMessageBox.h
 	src/Viewers/Histogram.h
 	src/Common/global.h
-	src/Common/GitSHA1.h
     src/Common/config.h
 
 # new viewer
@@ -408,11 +383,8 @@
         src/TonemappingEngine/TonemapOperator.cpp
 
 	src/main.cpp
-	${CMAKE_CURRENT_BINARY_DIR}/src/Common/GitSHA1.cpp     # for actual git version
 )
 
-CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/src/Common/GitSHA1.cpp.in" "${CMAKE_CURRENT_BINARY_DIR}/src/Common/GitSHA1.cpp" @ONLY)
-
 SET(LUMINANCE_HDR_MOC
 	src/BatchHDR/BatchHDRDialog.h
         src/BatchTM/BatchTMDialog.h
--- luminance-hdr-2.2.0.orig/src/UI/UMessageBox.cpp
+++ luminance-hdr-2.2.0/src/UI/UMessageBox.cpp
@@ -28,7 +28,6 @@
 
 #include "UI/UMessageBox.h"
 
-#include "Common/GitSHA1.h"
 #include "Common/config.h"
 #include "Common/archs.h"
 #include "ui_about.h"
@@ -78,7 +77,7 @@
     ui.authorsBox->setOpenExternalLinks(true);
     ui.thanksToBox->setOpenExternalLinks(true);
     ui.GPLbox->setTextInteractionFlags(Qt::TextSelectableByMouse);
-    ui.label_version->setText(ui.label_version->text().append(QString(LUMINANCEVERSION)).append(" [Build ").append(QString(g_GIT_SHA1).left(6)).append("]"));
+    ui.label_version->setText(ui.label_version->text().append(QString(LUMINANCEVERSION)));
 
     bool license_file_not_found=true;
     QString docDir = QCoreApplication::applicationDirPath();
