backported from upstream revision 393

diff -ru qtpfsgui-1.9.3.orig/src/MainWindow/mainWindow.cpp qtpfsgui-1.9.3/src/MainWindow/mainWindow.cpp
--- qtpfsgui-1.9.3.orig/src/MainWindow/mainWindow.cpp	2009-01-08 18:22:41.000000000 +0100
+++ qtpfsgui-1.9.3/src/MainWindow/mainWindow.cpp	2009-05-23 10:56:17.000000000 +0200
@@ -274,11 +274,13 @@
 			normalSizeAct->setEnabled(true);
 		}
 	} else {
-		currenthdr=NULL;
-		normalSizeAct->setEnabled(false);
-		zoomInAct->setEnabled(false);
-		zoomOutAct->setEnabled(false);
-		fitToWindowAct->setEnabled(false);
+		if (mdiArea->subWindowList().empty()) {
+			currenthdr=NULL;
+			normalSizeAct->setEnabled(false);
+			zoomInAct->setEnabled(false);
+			zoomOutAct->setEnabled(false);
+			fitToWindowAct->setEnabled(false);
+		}
 	}
 }
 
