diff -Nru squid-3.1.9.orig/helpers/negotiate_auth/squid_kerb_auth/configure.in squid-3.1.9/helpers/negotiate_auth/squid_kerb_auth/configure.in
--- squid-3.1.9.orig/helpers/negotiate_auth/squid_kerb_auth/configure.in	2010-11-27 13:44:10.000000000 +0100
+++ squid-3.1.9/helpers/negotiate_auth/squid_kerb_auth/configure.in	2010-11-27 13:44:46.000000000 +0100
@@ -94,7 +94,7 @@
                        else
                          ac_gssapi_libs=`krb5-config --libs gssapi 2>/dev/null`
                          if test "x$ac_gssapi_libs" != "x" ; then
-                           LDFLAGS="$LDFLAGS $ac_gssapi_libs"
+                           LIBS="$LIBS $ac_gssapi_libs"
                          else
                            for lib in $ac_gss_libs; do
                              AC_CHECK_LIB($lib,main)
@@ -118,7 +118,7 @@
                      fi
                      ac_gssapi_libs=`krb5-config --libs gssapi 2>/dev/null`
                      if test "x$ac_gssapi_libs" != "x" ; then
-                         LDFLAGS="$LDFLAGS $ac_gssapi_libs"
+                         LIBS="$LIBS $ac_gssapi_libs"
                      else
                          for lib in $ac_gss_libs; do
                             AC_CHECK_LIB($lib,main)
@@ -172,7 +172,7 @@
                              ac_libdir=`echo $ac_gssapi_libs | sed -e 's/.*-L//' | sed -e 's/ .*//'`  
                              LDFLAGS="$LDFLAGS $w_flag$ac_libdir$w_flag_2"
                            fi
-                           LDFLAGS="$LDFLAGS $ac_gssapi_libs"
+                           LIBS="$LIBS $ac_gssapi_libs"
                          else
                            for lib in $ac_gss_libs; do
                              AC_CHECK_LIB($lib,main)
@@ -201,7 +201,7 @@
                            ac_libdir=`echo $ac_gssapi_libs | sed -e 's/.*-L//' | sed -e 's/ .*//'`  
                            LDFLAGS="$LDFLAGS $w_flag$ac_libdir$w_flag_2"
                          fi
-                         LDFLAGS="$LDFLAGS $ac_gssapi_libs"
+                         LIBS="$LIBS $ac_gssapi_libs"
                      else
                          for lib in $ac_gss_libs; do
                             AC_CHECK_LIB($lib,main)
