diff -Nru squid-3.1.0.15.orig/helpers/negotiate_auth/squid_kerb_auth/configure.in squid-3.1.0.15/helpers/negotiate_auth/squid_kerb_auth/configure.in
--- squid-3.1.0.15.orig/helpers/negotiate_auth/squid_kerb_auth/configure.in	2009-11-28 10:51:49.000000000 +0100
+++ squid-3.1.0.15/helpers/negotiate_auth/squid_kerb_auth/configure.in	2009-11-28 10:52:03.000000000 +0100
@@ -94,7 +94,7 @@
                        else
                          ac_gssapi_libs=`krb5-config --libs gssapi 2>/dev/null`
                          if test "x$ac_gssapi_libs" != "x" ; then
-                           LDFLAGS="$LDFLAGS $ac_gssapi_libs"
+                           LIBS="$LIBS $ac_gssapi_libs"
                          else
                            for lib in $ac_gss_libs; do
                              AC_CHECK_LIB($lib,main)
@@ -118,7 +118,7 @@
                      fi
                      ac_gssapi_libs=`krb5-config --libs gssapi 2>/dev/null`
                      if test "x$ac_gssapi_libs" != "x" ; then
-                         LDFLAGS="$LDFLAGS $ac_gssapi_libs"
+                         LIBS="$LIBS $ac_gssapi_libs"
                      else
                          for lib in $ac_gss_libs; do
                             AC_CHECK_LIB($lib,main)
@@ -172,7 +172,7 @@
                              ac_libdir=`echo $ac_gssapi_libs | sed -e 's/.*-L//' | sed -e 's/ .*//'`  
                              LDFLAGS="$LDFLAGS $w_flag$ac_libdir$w_flag_2"
                            fi
-                           LDFLAGS="$LDFLAGS $ac_gssapi_libs"
+                           LIBS="$LIBS $ac_gssapi_libs"
                          else
                            for lib in $ac_gss_libs; do
                              AC_CHECK_LIB($lib,main)
@@ -201,7 +201,7 @@
                            ac_libdir=`echo $ac_gssapi_libs | sed -e 's/.*-L//' | sed -e 's/ .*//'`  
                            LDFLAGS="$LDFLAGS $w_flag$ac_libdir$w_flag_2"
                          fi
-                         LDFLAGS="$LDFLAGS $ac_gssapi_libs"
+                         LIBS="$LIBS $ac_gssapi_libs"
                      else
                          for lib in $ac_gss_libs; do
                             AC_CHECK_LIB($lib,main)
diff -Nru squid-3.1.0.15.orig/src/forward.cc squid-3.1.0.15/src/forward.cc
--- squid-3.1.0.15.orig/src/forward.cc	2009-11-23 04:10:09.000000000 +0100
+++ squid-3.1.0.15/src/forward.cc	2009-11-28 10:52:03.000000000 +0100
@@ -999,7 +999,12 @@
                         break;
 
                     if (o->cmsg_level == SOL_IP && o->cmsg_type == IP_TOS) {
-                        clientFde->upstreamTOS = (unsigned char)(*(int*)CMSG_DATA(o));
+                        union {
+                           unsigned char *pchar;
+                           int  *pint;
+                        } data;
+                        data.pchar = CMSG_DATA(o);
+                        clientFde->upstreamTOS = (unsigned char)*data.pint;
                         break;
                     }
                     p += CMSG_LEN(o->cmsg_len);
