diff -Nru squid-3.0.STABLE20.orig/configure.in squid-3.0.STABLE20/configure.in
--- squid-3.0.STABLE20.orig/configure.in	2009-11-28 10:11:48.000000000 +0100
+++ squid-3.0.STABLE20/configure.in	2009-11-28 10:12:23.000000000 +0100
@@ -1895,6 +1895,10 @@
 	;;
 esac
 
+dnl Define HOSTCXX
+HOSTCXX="$build-g++"
+AC_SUBST(HOSTCXX)
+
 dnl Check for programs
 AC_PROG_CPP
 AC_PROG_INSTALL
diff -Nru squid-3.0.STABLE20.orig/src/Makefile.am squid-3.0.STABLE20/src/Makefile.am
--- squid-3.0.STABLE20.orig/src/Makefile.am	2009-11-28 10:11:48.000000000 +0100
+++ squid-3.0.STABLE20/src/Makefile.am	2009-11-28 10:12:23.000000000 +0100
@@ -1011,6 +1011,8 @@
 
 DEFS = @DEFS@ -DDEFAULT_CONFIG_FILE=\"$(DEFAULT_CONFIG_FILE)\"
 
+HOSTCXX ?= @HOSTCXX@
+
 $(OBJS): $(top_srcdir)/include/version.h ../include/autoconf.h
 
 snmp_core.o snmp_agent.o: ../snmplib/libsnmp.a $(top_srcdir)/include/cache_snmp.h
@@ -1034,6 +1036,11 @@
 squid.conf.default: cf_parser.h
 	true
 
+cf_gen$(EXEEXT): $(cf_gen_SOURCES) $(cf_gen_DEPENDENCIES)
+	$(HOSTCXX) -o $@ $(srcdir)/cf_gen.cc \
+		$(top_srcdir)/lib/util.c $(top_srcdir)/lib/assert.c \
+		-DNDEBUG -DBUILD_HOST_TOOL ${INCLUDES}
+
 cf_parser.h: cf.data cf_gen$(EXEEXT)
 	./cf_gen cf.data $(srcdir)/cf.data.depend
 
