diff -ur squid-2.7.STABLE4.orig/src/cf.data.pre squid-2.7.STABLE4/src/cf.data.pre
--- squid-2.7.STABLE4.orig/src/cf.data.pre	2008-07-10 09:47:28.000000000 +0000
+++ squid-2.7.STABLE4/src/cf.data.pre	2008-08-15 06:00:39.000000000 +0000
@@ -3538,7 +3538,7 @@
 
 	When using the concurrency option, the protocol is changed by introducing
 	a query channel tag infront of the request/response. The query channel
-	tag is a number between 0 and concurrency−1.
+	tag is a number between 0 and concurrency-1.
 
 	FORMAT specifications:
 
@@ -3570,7 +3570,7 @@
 			present, the supplied value completely replaces
 			the cached value.
 
-	In the event of a helper−related error (e.g., overload), Squid
+	In the event of a helper-related error (e.g., overload), Squid
 	will always default to STALE.
 DOC_END
 
