--- ../drbd-cvs/drbd/scripts/Makefile	2003-07-14 10:43:25.000000000 +0200
+++ scripts/Makefile	2003-09-01 17:32:03.000000000 +0200
@@ -42,6 +42,9 @@
   INITD=$(PREFIX)etc/rc.d/init.d/
 endif
 
+ifeq ($(DIST),gentoo)
+  INITD=$(PREFIX)sbin/
+endif
 
 all:
 	ln -sf drbd datadisk
@@ -52,16 +55,16 @@
 	mkdir -p $(PREFIX)etc/ha.d/resource.d
 	install -d $(INITD)
 	install -m 755 drbd $(INITD)
-	@ if [ ! -e $(PREFIX)etc/drbd.conf ]; then \
-		install -d $(PREFIX)etc/; \
-		install -m 644 drbd.conf $(PREFIX)etc/; \
-	fi
+#	@ if [ ! -e $(PREFIX)etc/drbd.conf ]; then \
+#		install -d $(PREFIX)etc/; \
+#		install -m 644 drbd.conf $(PREFIX)etc/; \
+#	fi
 	ln -f $(INITD)drbd $(PREFIX)etc/ha.d/resource.d/datadisk
-ifeq ($(DIST),suse)
-	ln -sf $(INITD)drbd $(PREFIX)sbin/rcdrbd
-endif
-	@ echo
-	@ echo "Don't forget to run update-rc.d or chkconfig"
+#ifeq ($(DIST),suse)
+#	ln -sf $(INITD)drbd $(PREFIX)sbin/rcdrbd
+#endif
+#	@ echo
+#	@ echo "Don't forget to run update-rc.d or chkconfig"
 
 clean:
 	rm -f *~
