--- ext/fcgi/fcgi.c.orig	2003-08-27 20:28:35.000000000 +0900
+++ ext/fcgi/fcgi.c	2005-02-01 16:57:39.741372024 +0900
@@ -4,6 +4,7 @@
  * Copyright (C) 2002 MoonWolf <moonwolf@moonwolf.com>
  */
 
+#include <errno.h>
 #include <stdio.h>
 #include <sys/types.h>
 #include <unistd.h>
@@ -155,7 +156,8 @@
   int err = FCGX_GetError(stream);\
   if (err) {\
     if (err > 0) {\
-      rb_sys_fail(0);\
+      errno=err;\
+      rb_sys_fail(strerror(err));\
     }\
     else {\
       switch (err) {\
