diff -aur mcs-1.0.5-orig/class/corlib/System.IO/Path.cs mcs-1.0.5/class/corlib/System.IO/Path.cs
--- mcs-1.0.5-orig/class/corlib/System.IO/Path.cs	2004-12-07 17:29:31.000000000 -0500
+++ mcs-1.0.5/class/corlib/System.IO/Path.cs	2004-12-15 20:45:25.060090272 -0500
@@ -386,7 +386,7 @@
 			//if (root == path) return path;
 				
 			// STEP 3: split the directories, this gets rid of consecutative "/"'s
-			string [] dirs = path.Split (DirectorySeparatorChar, AltDirectorySeparatorChar);
+			string [] dirs = path.Split (DirectorySeparatorChar);
 			// STEP 4: Get rid of directories containing . and ..
 			int target = 0;
 			
