--- libgphoto2/gphoto2-camera.c	2006/05/30 18:45:48	1.1
+++ libgphoto2/gphoto2-camera.c	2006/05/30 18:45:58
@@ -543,7 +543,7 @@ gp_camera_ref (Camera *camera)
  * @camera: a #Camera
  *
  * Decrements the reference count of a @camera. If the reference count 
- * reaches %0, the @camera will be freed automatically.
+ * reaches 0, the @camera will be freed automatically.
  *
  * Return value: a gphoto2 error code
  **/
