diff -Naur qmbtagger-0.07.orig/src/qmbflacfile.cpp qmbtagger-0.07.fixed/src/qmbflacfile.cpp
--- qmbtagger-0.07.orig/src/qmbflacfile.cpp	2004-05-01 13:18:00.000000000 -0700
+++ qmbtagger-0.07.fixed/src/qmbflacfile.cpp	2004-10-10 10:51:59.000000000 -0700
@@ -30,6 +30,8 @@
 #include <musicbrainz/mb_c.h>
 #include <FLAC++/metadata.h>
 
+#include <errno.h>
+
 QMBFLACFile::QMBFLACFile()
 	: QMBFile(), FLAC::Decoder::File(), offset(0), decode_error(false)
 {
diff -Naur qmbtagger-0.07.orig/src/qmbmp3file.cpp qmbtagger-0.07.fixed/src/qmbmp3file.cpp
--- qmbtagger-0.07.orig/src/qmbmp3file.cpp	2004-05-01 13:18:00.000000000 -0700
+++ qmbtagger-0.07.fixed/src/qmbmp3file.cpp	2004-10-10 11:00:43.807278465 -0700
@@ -32,6 +32,8 @@
 #include <id3/misc_support.h>
 #include <fstream>
 
+#include <errno.h>
+
 QMBMP3File::QMBMP3File()
 	: QMBFile()
 {
