diff -u -r mrproject-0.4.0.orig/client/widgets/Makefile.am mrproject-0.4.0/client/widgets/Makefile.am
--- mrproject-0.4.0.orig/client/widgets/Makefile.am	Mon Jul 30 14:28:23 2001
+++ mrproject-0.4.0/client/widgets/Makefile.am	Sat Oct  6 13:42:04 2001
@@ -1,5 +1,6 @@
 INCLUDES = 						\
 	$(GNOMEUI_GAL_CFLAGS)				\
+	$(ORBIT_CFLAGS)					\
 	-I$(top_srcdir)					\
 	-I$(top_builddir)
 
diff -u -r mrproject-0.4.0.orig/libmrproject/Makefile.am mrproject-0.4.0/libmrproject/Makefile.am
--- mrproject-0.4.0.orig/libmrproject/Makefile.am	Wed Sep  5 21:28:55 2001
+++ mrproject-0.4.0/libmrproject/Makefile.am	Sat Oct  6 13:31:06 2001
@@ -3,7 +3,8 @@
 	-I$(top_builddir) 		\
 	-I$(top_srcdir)/libmrproject 	\
 	-I$(top_builddir)/libmrproject 	\
-	$(IDL_FLAGS)
+	$(IDL_FLAGS)			\
+	$(ORBIT_CFLAGS)
 
 LDFLAGS=				\
 	$(ORBIT_LIBS)
diff -u -r mrproject-0.4.0.orig/util/Makefile.am mrproject-0.4.0/util/Makefile.am
--- mrproject-0.4.0.orig/util/Makefile.am	Mon Sep 17 16:39:38 2001
+++ mrproject-0.4.0/util/Makefile.am	Sat Oct  6 13:30:58 2001
@@ -2,6 +2,7 @@
 	-I$(top_srcdir)				\
 	$(GNOME_CFLAGS)				\
 	$(VFS_CFLAGS)				\
+	$(ORBIT_CFLAGS)				\
 	-DMRPROJECT_IMAGEDIR=\"$(datadir)/images/mrproject/\" \
 	-DG_LOG_DOMAIN=\"MrProject\"
 
