--- b/acinclude.m4	2008-12-30 12:00:47.000000000 -0500
+++ a/acinclude.m4	2010-03-23 00:51:13.000000000 -0400
@@ -444,6 +439,14 @@
 ])
 
 AC_DEFUN([OGRE_CHECK_CEGUI], [
+AC_ARG_ENABLE(cegui,
+               AC_HELP_STRING([--disable-cegui],
+	                      [Don't build CEGUIRenderer, even if CEGUI is installed.]),
+	       [if test x$enableval = xyes; then
+	          build_cegui_sample=true
+	        fi],
+	       [build_cegui_sample=true])
+    if test x$build_cegui_sample = xtrue; then
     PKG_CHECK_MODULES(CEGUI, CEGUI >= 0.5.0, 
             [build_cegui_sample=true], [build_cegui_sample=false])
     if test x$build_cegui_sample = xtrue; then
@@ -459,6 +462,7 @@
     else
         AC_MSG_RESULT([CEGUI not available, Gui and FacialAnimation samples will not be built])
     fi
+    fi
     AM_CONDITIONAL([HAVE_CEGUI], [test x$build_cegui_sample = xtrue])
 ])
 
