--- binutils-2.13.90.0.10/ld/emulparams/elf_i386_glibc21.sh.glibc21	2002-11-15 15:12:42.000000000 +0100
+++ binutils-2.13.90.0.10/ld/emulparams/elf_i386_glibc21.sh	2002-11-15 15:12:42.000000000 +0100
@@ -0,0 +1,11 @@
+SCRIPT_NAME=elf
+OUTPUT_FORMAT="elf32-i386"
+TEXT_START_ADDR=0x08048000
+MAXPAGESIZE=0x1000
+NONPAGED_TEXT_START_ADDR=0x08048000
+ARCH=i386
+MACHINE=
+NOP=0x9090
+TEMPLATE_NAME=elf32
+GENERATE_SHLIB_SCRIPT=yes
+LIB_PATH=/usr/i386-glibc21-linux/lib:/usr/local/lib
--- binutils-2.13.90.0.10/ld/configure.tgt.glibc21	2002-10-10 06:09:41.000000000 +0200
+++ binutils-2.13.90.0.10/ld/configure.tgt	2002-11-15 15:14:38.000000000 +0100
@@ -147,6 +147,8 @@ i[3456]86-*-linux-gnu*)	targ_emul=elf_i3
 			if test x${want64} = xtrue; then
 			  targ_extra_emuls="$targ_extra_emuls elf_x86_64"
 			fi
+			targ_extra_emuls="i386linux elf_i386_glibc21"
+			targ_extra_libpath=elf_i386_glibc21
 			tdir_i386linux=${targ_alias}aout
 			;;
 x86_64-*-linux-gnu*)	targ_emul=elf_x86_64
--- binutils-2.13.90.0.10/ld/Makefile.am.glibc21	2002-10-10 06:09:41.000000000 +0200
+++ binutils-2.13.90.0.10/ld/Makefile.am	2002-11-15 15:12:42.000000000 +0100
@@ -185,6 +185,7 @@ ALL_EMULATIONS = \
 	eelf_i386_be.o \
 	eelf_i386_chaos.o \
 	eelf_i386_fbsd.o \
+	eelf_i386_glibc21.o \
 	eelf_i386_ldso.o \
 	eelf_s390.o \
 	egld960.o \
@@ -734,6 +735,9 @@ eelf_i386_be.c: $(srcdir)/emulparams/elf
 eelf_i386_chaos.c: $(srcdir)/emulparams/elf_i386_chaos.sh \
   $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf_chaos.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_chaos "$(tdir_elf_i386_chaos)"
+eelf_i386_glibc21.c: $(srcdir)/emulparams/elf_i386_glibc21.sh \
+  $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} elf_i386_glibc21 "$(tdir_elf_i386_glibc21)"
 eelf_i386_fbsd.c: $(srcdir)/emulparams/elf_i386_fbsd.sh \
   $(srcdir)/emulparams/elf_i386.sh \
   $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
--- binutils-2.13.90.0.10/ld/Makefile.in.glibc21	2002-10-10 06:09:41.000000000 +0200
+++ binutils-2.13.90.0.10/ld/Makefile.in	2002-11-15 15:12:42.000000000 +0100
@@ -295,6 +295,7 @@ ALL_EMULATIONS = \
 	eelf_i386.o \
 	eelf_i386_be.o \
 	eelf_i386_chaos.o \
+	eelf_i386_glibc21.o \
 	eelf_i386_fbsd.o \
 	eelf_i386_ldso.o \
 	eelf_s390.o \
@@ -1457,6 +1458,9 @@ eelf_i386_be.c: $(srcdir)/emulparams/elf
 eelf_i386_chaos.c: $(srcdir)/emulparams/elf_i386_chaos.sh \
   $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf_chaos.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_chaos "$(tdir_elf_i386_chaos)"
+eelf_i386_glibc21.c: $(srcdir)/emulparams/elf_i386_glibc21.sh \
+  $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} elf_i386_glibc21 "$(tdir_elf_i386_glibc21)"
 eelf_i386_fbsd.c: $(srcdir)/emulparams/elf_i386_fbsd.sh \
   $(srcdir)/emulparams/elf_i386.sh \
   $(srcdir)/emultempl/elf32.em $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
