--- bfd/elflink.h.jj	2003-02-22 15:46:41.000000000 -0500
+++ bfd/elflink.h	2003-02-24 09:27:06.000000000 -0500
@@ -8328,9 +8328,10 @@ elf_bfd_discard_info (output_bfd, info)
 	continue;
 
       eh = bfd_get_section_by_name (abfd, ".eh_frame");
-      if (eh != NULL
-	  && (eh->_raw_size == 0
-	      || bfd_is_abs_section (eh->output_section)))
+      if (info->relocateable
+	  || (eh != NULL
+	      && (eh->_raw_size == 0
+		  || bfd_is_abs_section (eh->output_section))))
 	eh = NULL;
 
       stab = bfd_get_section_by_name (abfd, ".stab");
@@ -8431,6 +8432,7 @@ elf_bfd_discard_info (output_bfd, info)
     }
 
   if (info->eh_frame_hdr
+      && !info->relocateable
       && _bfd_elf_discard_section_eh_frame_hdr (output_bfd, info))
     ret = TRUE;
 
