diff -urN text-format-0.64/install-pkg.rb text-format-0.64.new/install-pkg.rb
--- text-format-0.64/install-pkg.rb	2003-09-18 12:45:00.000000000 +0900
+++ text-format-0.64.new/install-pkg.rb	2004-04-11 01:51:32.000000000 +0900
@@ -310,7 +310,7 @@
 module FileOperations
 
   def mkdir_p( dname, prefix = nil )
-    dname = prefix + dname if prefix
+    #dname = prefix + dname if prefix
     $stderr.puts "mkdir -p #{dname}" if verbose?
     return if no_harm?
 
@@ -373,7 +373,8 @@
     $stderr.puts "install #{from} #{dest}" if verbose?
     return if no_harm?
 
-    realdest = prefix + dest if prefix
+    #realdest = prefix + dest if prefix
+    realdest = dest
     if dir? realdest then
       realdest += '/' + File.basename(from)
     end
