diff -urN miniChinput-0.1.9.orig/configure miniChinput-0.1.9/configure
--- miniChinput-0.1.9.orig/configure	2003-07-14 14:09:41.000000000 +0900
+++ miniChinput-0.1.9/configure	2006-11-06 02:11:37.000000000 +0900
@@ -3413,11 +3413,11 @@
 
 # Check Xft related headers and libs.
 
-tmpar=`find /usr/include /usr/X11R6/include -name Xft.h | sed 's?/X11/Xft/Xft.h??'`
+tmpar=`find /usr/include -name Xft.h | sed 's?/X11/Xft/Xft.h??'`
 TMP_XFT2=`echo "$tmpar"|grep Xft2`
 TMP_XFT=`echo "$tmpar"|grep -v Xft2`
 
-tmpar=`find /usr/include /usr/X11R6/include -name freetype.h | sed 's?/freetype/freetype.h??'`
+tmpar=`find /usr/include -name freetype.h | sed 's?/freetype/freetype.h??'`
 TMP_FREE2=`echo "$tmpar"|grep freetype2`
 TMP_FREE=`echo "$tmpar"|grep -v freetype2`
 if test "x$TMP_XFT2" != "x" -a "x$TMP_FREE2" != "x"; then
diff -urN miniChinput-0.1.9.orig/configure.in miniChinput-0.1.9/configure.in
--- miniChinput-0.1.9.orig/configure.in	2003-07-14 14:09:41.000000000 +0900
+++ miniChinput-0.1.9/configure.in	2006-11-06 02:11:54.000000000 +0900
@@ -45,11 +45,11 @@
 
 # Check Xft related headers and libs.
 
-tmpar=`find /usr/include /usr/X11R6/include -name Xft.h | sed 's?/X11/Xft/Xft.h??'`
+tmpar=`find /usr/include -name Xft.h | sed 's?/X11/Xft/Xft.h??'`
 TMP_XFT2=`echo "$tmpar"|grep Xft2`
 TMP_XFT=`echo "$tmpar"|grep -v Xft2`
 
-tmpar=`find /usr/include /usr/X11R6/include -name freetype.h | sed 's?/freetype/freetype.h??'`
+tmpar=`find /usr/include -name freetype.h | sed 's?/freetype/freetype.h??'`
 TMP_FREE2=`echo "$tmpar"|grep freetype2`
 TMP_FREE=`echo "$tmpar"|grep -v freetype2`
 if test "x$TMP_XFT2" != "x" -a "x$TMP_FREE2" != "x"; then 
