diff -ur typespeed-0.4.4.orig/file.c typespeed-0.4.4/file.c
--- typespeed-0.4.4.orig/file.c	2003-06-18 02:15:39.000000000 -0500
+++ typespeed-0.4.4/file.c	2005-01-15 21:56:50.386541848 -0600
@@ -14,6 +14,7 @@
 	wordcount = -1;
 	nakkimakkara = 0;
 
+chdir(WORDLIST_PATH);
 	if ((wordfile = fopen(stringi, "r")) == NULL) {
 		return 1;
 	}
@@ -89,6 +90,7 @@
 	k = 0;
 	a = 1;
 
+chdir(WORDLIST_PATH);
 	dcolor_set(5, NULL);
 	n = scandir(".", &namelist, 0, alphasort);
 	tot = n;
@@ -272,6 +274,7 @@
 	{"3l33t", "Pro", "VeryGood", "Good", "Average", "NoGood", "ADP-Pro", "Loser", "None", "None"};
 
 
+chdir(HIGHSCORE_PATH);
 	filetys = (char *) malloc(sizeof(char) * (strlen(sanafile) + strlen(".high") + 1));
 	strcpy(filetys, "");
 	strcat(filetys, "high.");
@@ -311,6 +314,7 @@
 		float typoratio;
 		char faim[21];
 	} high;
+chdir(HIGHSCORE_PATH);
 	filetys = (char *) malloc(sizeof(char) * (strlen(sanafile) + strlen(".high") + 1));
 	strcpy(filetys, "");
 	strcat(filetys, "high.");
@@ -386,6 +390,7 @@
 						strlen(".high")
 						+ 1));
 
+chdir(HIGHSCORE_PATH);
 	dcolor_set(5, NULL);
 	strcpy(filet, "");
 	strcat(filet, "high.");
@@ -526,6 +531,7 @@
 	struct dirent **namelist;
         int n;
 
+chdir(WORDLIST_PATH);
 	n = scandir(".", &namelist, 0, alphasort);
 	if (n < 0) {
 		printf("scandir failed.\n");
diff -ur typespeed-0.4.4.orig/typespeed.1 typespeed-0.4.4/typespeed.1
--- typespeed-0.4.4.orig/typespeed.1	2003-06-18 02:36:37.000000000 -0500
+++ typespeed-0.4.4/typespeed.1	2005-01-15 21:53:34.856266968 -0600
@@ -1,4 +1,4 @@
-.TH TYPESPEED 1
+.TH TYPESPEED 6
 .SH NAME
 typespeed \- Test your typing speed, and get your fingers' CPS
 .SH SYNOPSIS
diff -ur typespeed-0.4.4.orig/typespeed.h typespeed-0.4.4/typespeed.h
--- typespeed-0.4.4.orig/typespeed.h	2003-06-18 01:43:13.000000000 -0500
+++ typespeed-0.4.4/typespeed.h	2005-01-15 21:56:52.443229184 -0600
@@ -38,6 +38,9 @@
 #define DEFAULT_TRAININGSPEED 2
 #define DEFAULT_CHEAT 0
 
+#define HIGHSCORE_PATH "/var/games/typespeed"
+#define WORDLIST_PATH "/usr/share/games/typespeed"
+
 typedef struct
 {
 	int score;
