--- src/include/mwcollect/log.h.orig	2008-12-23 20:33:53.000000000 -0500
+++ src/include/mwcollect/log.h	2008-12-23 20:34:07.000000000 -0500
@@ -9,6 +9,7 @@
 
 #include <list>
 #include <stdio.h>
+#include <string.h>
 
 
 #define LT_LEVEL_LOW 1
--- src/core/config.cpp.orig	2008-12-23 20:34:50.000000000 -0500
+++ src/core/config.cpp	2008-12-23 20:35:22.000000000 -0500
@@ -10,6 +10,8 @@
 #include <mwcollect/config.h>
 #include <stdio.h>
 #include <ctype.h>
+#include <malloc.h>
+#include <stdlib.h>
 
 namespace mwccore
 {
--- src/core/vshell.cpp.orig	2008-12-23 20:36:13.000000000 -0500
+++ src/core/vshell.cpp	2008-12-23 20:36:27.000000000 -0500
@@ -9,6 +9,7 @@
 #include <mwcollect/core.h>
 #include <time.h>
 #include <string>
+#include <malloc.h>
 
 namespace mwccore
 {
--- src/net-posix/interface.cpp.orig	2008-12-23 20:37:59.000000000 -0500
+++ src/net-posix/interface.cpp	2008-12-23 20:38:23.000000000 -0500
@@ -23,6 +23,7 @@
 #include <errno.h>
 #include <string.h>
 #include <assert.h>
+#include <malloc.h>
 
 PosixInterface::PosixInterface(Configuration * pConfig)
 {
--- src/shell-basic/shell-basic.cpp.orig	2008-12-23 21:10:01.000000000 -0500
+++ src/shell-basic/shell-basic.cpp	2008-12-23 21:11:44.000000000 -0500
@@ -7,6 +7,7 @@
  
 #include "shell-basic.h"
 #include <string>
+#include <malloc.h>
 
 extern "C"
 {
--- src/log-irc/log-irc.cpp.orig	2008-12-23 21:12:32.000000000 -0500
+++ src/log-irc/log-irc.cpp	2008-12-23 21:13:07.000000000 -0500
@@ -11,6 +11,8 @@
 #include <fnmatch.h>
 #include <time.h>
 #include <unistd.h>
+#include <malloc.h>
+#include <stdlib.h>
 
 extern "C"
 {
--- src/download-tftp/download-tftp-module.cpp.orig	2008-12-23 21:14:33.000000000 -0500
+++ src/download-tftp/download-tftp-module.cpp	2008-12-23 21:14:51.000000000 -0500
@@ -8,6 +8,7 @@
  
 #include "download-tftp.h"
 #include <assert.h>
+#include <malloc.h>
 
 extern "C"
 {
--- src/download-tftp/download-tftp-subscription.cpp.orig	2008-12-23 21:17:28.000000000 -0500
+++ src/download-tftp/download-tftp-subscription.cpp	2008-12-23 21:18:06.000000000 -0500
@@ -8,6 +8,7 @@
  
 #include "download-tftp.h"
 #include <netinet/in.h>
+#include <malloc.h>
 
 void TftpSubscription::incomingData(unsigned char * pucData, unsigned int nLength)
 {
--- src/download-curl/download-curl-container.cpp.orig	2008-12-23 21:18:50.000000000 -0500
+++ src/download-curl/download-curl-container.cpp	2008-12-23 21:19:12.000000000 -0500
@@ -8,6 +8,7 @@
  
 #include "download-curl.h"
 #include <string.h>
+#include <malloc.h>
 #if defined(OBSD_FLAVOURED) /* select() */
 	#include <sys/types.h>
 	#include <sys/time.h>
--- src/vuln-ms05-51/vuln-ms0551-subscription.cpp.orig	2008-12-23 21:19:44.000000000 -0500
+++ src/vuln-ms05-51/vuln-ms0551-subscription.cpp	2008-12-23 21:20:44.000000000 -0500
@@ -9,6 +9,7 @@
  
 #include "vuln-ms0551.h"
 
+#include <stdlib.h>
 #include <string.h>
 
 // from http://www.frsirt.com/exploits/20051201.MS05-051msdtc.cpp.php
