--- src/rio_add_song.c	2002-04-15 14:09:13.000000000 +0200
+++ src/rio_add_song.c
@@ -325,6 +325,7 @@ main(int argc, char *argv[])
    send_command (rio_dev, 0x58, 0x0, card_number);
 
 try_next:
+	;
    } /* end of add file loop */
 
    /* Close device */
@@ -626,6 +626,7 @@
          break;

         default:                                       /* No Action */
+       ;
       }

     } else {
