--- em-4.0.18-orig/buffer.c	1994-05-21 12:40:56.000000000 +0200
+++ em-4.0.18/buffer.c	2007-06-11 16:59:35.000000000 +0200
@@ -448,7 +448,6 @@
         register BUFFER *bp;
 	register BUFFER *sb;	/* buffer to insert after */
         register LINE   *lp;
-	char *malloc();
 
         bp = bheadp;
         while (bp != NULL) {
--- em-4.0.18-orig/display.c	2005-02-22 12:03:13.000000000 +0100
+++ em-4.0.18/display.c	2007-06-11 16:59:46.000000000 +0200
@@ -56,7 +56,6 @@
 {
     register int i;
     register VIDEO *vp;
-    char *malloc();
 
     TTopen();		/* open the screen */
     TTkopen();		/* open the keyboard */
--- em-4.0.18-orig/edef.h	1994-05-21 12:40:57.000000000 +0200
+++ em-4.0.18/edef.h	2007-06-11 21:14:54.000000000 +0200
@@ -8,6 +8,9 @@
  *	modified by Petri Kutvonen
  */
 
+#include <stdlib.h>
+#include <string.h>
+
 /* some global fuction declarations */
 
 char *flook();
@@ -19,12 +22,13 @@
 char *gtusr();
 char *itoa();
 char *ltos();
-char *malloc();
+/*char *malloc();*/
 char *mklower();
 char *mkupper();
-char *strcat();
-char *strcpy();
-char *strncpy();
+/*char *strcat();*/
+/*char *strcpy();*/
+/*char *strncpy();*/
+char *uestrncpy();
 char *token();
 char *transbind();
 unsigned int getckey();
--- em-4.0.18-orig/epath.h	1994-05-21 12:41:01.000000000 +0200
+++ em-4.0.18/epath.h	2007-06-11 22:00:08.000000000 +0200
@@ -32,7 +32,7 @@
 	"/usr/local/lib/",
 #endif
 	"/usr/local/",
-	"/usr/lib/",
+	"/usr/share/uemacs-pk/",
 	""
 };
 #endif
--- em-4.0.18-orig/eval.c	2005-02-22 12:03:13.000000000 +0100
+++ em-4.0.18/eval.c	2007-06-11 21:04:58.000000000 +0200
@@ -554,7 +554,7 @@
 
 atoi(st)
 
-char *st;
+const char *st;
 
 {
 	int result;	/* resulting number */
--- em-4.0.18-orig/line.c	1996-05-02 14:22:12.000000000 +0200
+++ em-4.0.18/line.c	2007-06-11 17:03:53.000000000 +0200
@@ -33,7 +33,6 @@
 {
 	register LINE	*lp;
 	register int	size;
-	char *malloc();
 
 	size = (used+NBLOCK-1) & ~(NBLOCK-1);
 	if (size == 0)				/* Assume that an empty */
--- em-4.0.18-orig/main.c	2005-02-22 12:03:13.000000000 +0100
+++ em-4.0.18/main.c	2007-06-11 21:15:57.000000000 +0200
@@ -82,6 +82,7 @@
 
 #if	UNIX
 #include <signal.h>
+static void emergencyexit();
 #endif
 
 #if	CALLED
@@ -123,7 +124,6 @@
 #endif
 
 #if UNIX
-	static void emergencyexit();
 #ifdef SIGWINCH
 	extern void sizesignal();
 #endif
@@ -443,7 +443,6 @@
 {
         register BUFFER *bp;
         register WINDOW *wp;
-	char *malloc();
 
         bp = bfind(bname, TRUE, 0);             /* First buffer         */
         blistp = bfind("*List*", TRUE, BFINVS); /* Buffer list buffer   */
--- em-4.0.18-orig/search.c	2005-02-22 12:03:13.000000000 +0100
+++ em-4.0.18/search.c	2007-06-11 17:01:48.000000000 +0200
@@ -1557,8 +1557,6 @@
  */
 static	BITMAP clearbits()
 {
-	char		*malloc();
-
 	BITMAP		cclstart, cclmap;
 	register int	j;
 
--- em-4.0.18-orig/window.c	1994-05-21 12:41:05.000000000 +0200
+++ em-4.0.18/window.c	2007-06-11 17:02:20.000000000 +0200
@@ -332,7 +332,6 @@
         register int    ntrd;
         register WINDOW *wp1;
         register WINDOW *wp2;
-	char *malloc();
 
         if (curwp->w_ntrows < 3) {
                 mlwrite("Cannot split a %d line window", curwp->w_ntrows);
