diff -Naur ptetex3-20090610.orig/2extract-src.sh ptetex3-20090610/2extract-src.sh
--- ptetex3-20090610.orig/2extract-src.sh	2009-06-07 17:00:07.000000000 +0900
+++ ptetex3-20090610/2extract-src.sh	2009-11-12 00:23:18.635898734 +0900
@@ -147,25 +147,25 @@
     $MKDIR $SRC/texk/pdvipsk
     $CP $SRC/texk/dvipsk/* $SRC/texk/pdvipsk/ 2> /dev/null
     tarcp $SRC/texk/xdvik $SRC/texk/pxdvik
-    P=p
+    DVIPSK_P=p
 fi # "$JAPANESE" = "international"
 ## for dvipsk
-#tarx archive/dvipsk-jpatch-p*.tar.gz         $SRC/texk/${P}dvipsk
-#cpatch $SRC/texk/${P}dvipsk/dvipsk-*.patch 1 $SRC/texk/${P}dvipsk
-tarx archive/dvipsk-*-p*.tar.gz              $SRC/texk/${P}dvipsk
-cpatch $SRC/texk/${P}dvipsk/dvipsk-*.diff 1  $SRC/texk/${P}dvipsk
-perlprint 's/^kpse_include/ac_include/g'     $SRC/texk/${P}dvipsk/Makefile.in
+#tarx archive/dvipsk-jpatch-p*.tar.gz         $SRC/texk/${DVIPSK_P}dvipsk
+#cpatch $SRC/texk/${DVIPSK_P}dvipsk/dvipsk-*.patch 1 $SRC/texk/${DVIPSK_P}dvipsk
+tarx archive/dvipsk-*-p*.tar.gz              $SRC/texk/${DVIPSK_P}dvipsk
+cpatch $SRC/texk/${DVIPSK_P}dvipsk/dvipsk-*.diff 1  $SRC/texk/${DVIPSK_P}dvipsk
+perlprint 's/^kpse_include/ac_include/g'     $SRC/texk/${DVIPSK_P}dvipsk/Makefile.in
 
 ## for xdvik
-zpatch archive/xdvik-200?????-tetex-3.0.diff.gz 3 $SRC/texk/${P}xdvik
+zpatch archive/xdvik-200?????-tetex-3.0.diff.gz 3 $SRC/texk/${DVIPSK_P}xdvik
 # fix for FC5
 perlprint 'print "#define NeedWidePrototypes 1\n" if /limits/' \
-    $SRC/texk/${P}xdvik/xdvi.h
+    $SRC/texk/${DVIPSK_P}xdvik/xdvi.h
 # pTeX fontmap
 perlnoprint "s/^ptexmap .*/ptexmap kanjix.map/; print unless /IPA/" \
-    $SRC/texk/${P}xdvik/texmf/xdvi.cfg
+    $SRC/texk/${DVIPSK_P}xdvik/texmf/xdvi.cfg
 # adjust X resource
-cat <<EOF >> $SRC/texk/${P}xdvik/texmf/XDvi || exit 1
+cat <<EOF >> $SRC/texk/${DVIPSK_P}xdvik/texmf/XDvi || exit 1
 
 
 !!! ------- added by ptetex3 -------
diff -Naur ptetex3-20090610.orig/3extract-texmf.sh ptetex3-20090610/3extract-texmf.sh
--- ptetex3-20090610.orig/3extract-texmf.sh	2009-06-10 17:10:51.000000000 +0900
+++ ptetex3-20090610/3extract-texmf.sh	2009-11-12 01:28:53.475843907 +0900
@@ -85,8 +85,8 @@
 $CP $MAP/ptex-noEmbed.map $MAP/ptex-morisawa.map
 perlprint "s/min/\!Ryumin-Light/g;
 	  s/goth/\!GothicBBB-Medium/g;"   $MAP/ptex-noEmbed.map
-perlprint "s/min/ipam.ttf/g;
-	  s/goth/ipag.ttf/g;"             $MAP/ptex-ipa.map
+perlprint "s/min/ipam.otf/g;
+	  s/goth/ipag.otf/g;"             $MAP/ptex-ipa.map
 perlprint "s/min/HiraMinPro-W3/g;
 	  s/goth/HiraKakuPro-W6/g;"       $MAP/ptex-hiragino.map
 perlprint "s/min/KozMinPro-Regular-Acro.otf/g;
diff -Naur ptetex3-20090610.orig/4make-install.sh ptetex3-20090610/4make-install.sh
--- ptetex3-20090610.orig/4make-install.sh	2009-05-05 01:35:49.000000000 +0900
+++ ptetex3-20090610/4make-install.sh	2009-11-12 00:23:18.635898734 +0900
@@ -44,11 +44,9 @@
 
 echo "[--tetex--] configure"
 cd $SRC || exit
-$MKDIR $PREFIX/share/texmf 2> /dev/null # avoid warning of texmf tree
 ./configure --prefix=$PREFIX --datadir=$DATADIR \
     --disable-multiplatform --with-ptexenc=$KANJI_CODE \
     $CONF_OPTION || exit
-$RMDIR $PREFIX/share/texmf 2> /dev/null # avoid warning of texmf tree
 xdvi_check
 echo "[--tetex--] make"
 $MAKE $MAKE_OPTION || exit
diff -Naur ptetex3-20090610.orig/5macro.sh ptetex3-20090610/5macro.sh
--- ptetex3-20090610.orig/5macro.sh	2008-04-17 14:30:43.000000000 +0900
+++ ptetex3-20090610/5macro.sh	2009-11-12 01:29:22.365322180 +0900
@@ -104,8 +104,8 @@
 hkbv        V                 !GothicBBB-Medium
 EOF
 
-perl -pe 's/\!Ryumin-Light/ipam.ttf/g; 
-	  s/\!GothicBBB-Medium/ipag.ttf/g;' \
+perl -pe 's/\!Ryumin-Light/ipam.otf/g; 
+	  s/\!GothicBBB-Medium/ipag.otf/g;' \
 	      < $MAP/utf-noEmbed.map > $MAP/utf-ipa.map
 perl -pe 's/\!Ryumin-Light/HiraMinPro-W3/g; 
 	  s/\!GothicBBB-Medium/HiraKakuPro-W6/g;' \
@@ -197,8 +197,8 @@
 hmgothr-v	V			!Ryumin-Light
 EOF
 
-perl -pe 's/\!Ryumin-Light/ipam.ttf/g; 
-	  s/\!GothicBBB-Medium/ipag.ttf/g;' \
+perl -pe 's/\!Ryumin-Light/ipam.otf/g; 
+	  s/\!GothicBBB-Medium/ipag.otf/g;' \
 	      < $MAP/otf-noEmbed.map > $MAP/otf-ipa.map
 
 cat <<EOF > $MAP/otf-cktx.map || exit
diff -Naur ptetex3-20090610.orig/7font-search.sh ptetex3-20090610/7font-search.sh
--- ptetex3-20090610.orig/7font-search.sh	2008-06-16 13:06:07.000000000 +0900
+++ ptetex3-20090610/7font-search.sh	2009-11-12 00:23:18.635898734 +0900
@@ -58,6 +58,26 @@
     done
 }
 
+mksymlink_gentoo(){
+    $MKDIR $TRUETYPE
+    fonts_dir=/usr/share/fonts/
+    find $fonts_dir -maxdepth 1 -mindepth 1 -type d | while read font_path; do
+	if [ `find $font_path -maxdepth 1 -type f -iname '*.ttf' | wc -l` -gt 0 ]; then
+	    font_name=${font_path/$fonts_dir/}
+	    test -h ${TRUETYPE}/${font_name} && continue
+	    $LN "${font_path}" "${TRUETYPE}/${font_name}"
+	fi
+    done
+    $MKDIR $OPENTYPE
+    find $fonts_dir -maxdepth 1 -mindepth 1 -type d | while read font_path; do
+	if [ `find $font_path -maxdepth 1 -type f -iname '*.otf' | wc -l` -gt 0 ]; then
+	    font_name=${font_path/$fonts_dir/}
+	    test -h ${OPENTYPE}/${font_name} && continue
+	    $LN "${font_path}" "${OPENTYPE}/${font_name}"
+	fi
+    done
+
+}
 
 setrmlgbm() {
     if test -n "$2"; then
@@ -111,10 +131,10 @@
 
 mksymlink $CMAP <<EOF
 ### リンク名  検索ディレクトリ/ファイル
-Acrobat4    /usr{,/local,/sfw}{,/lib}/Acrobat4/Resource/CMap
-Acrobat5    /usr{,/local,/sfw}{,/lib}/Acrobat5/Resource/Font
-Acrobat7    /usr{,/local,/sfw}{,/lib}{,/Adobe}/Acrobat7.0/Resource/CMap
-Acrobat8    /usr{,/local,/sfw}{,/lib}{,/Adobe}/Acrobat8.0/Resource/CMap
+Acrobat4    {/opt,/usr{,/local,/sfw}{,/lib}}/Acrobat4/Resource/CMap
+Acrobat5    {/opt,/usr{,/local,/sfw}{,/lib}}/Acrobat5/Resource/Font
+Acrobat7    {/opt,/usr{,/local,/sfw}{,/lib}}{,/Adobe}/Acrobat7{,.0}/Resource/CMap
+Acrobat8    {/opt,/usr{,/local,/sfw}{,/lib}}{,/Adobe}/{Acrobat,Reader}8{,.0}/Resource/CMap
 ghostscript /usr{,/local,/sfw}/{share,lib}/ghostscript{,/*}/Resource/CMap
 openwinja   /usr/openwin/lib/locale/ja/X11/Resource/CMap
 openwinko   /usr/openwin/lib/locale/ko/X11/Resource/CMap
@@ -122,21 +142,7 @@
 openwintw   /usr/openwin/lib/locale/zh_TW.BIG5/X11/Resource/CMap
 EOF
 
-mksymlink $TRUETYPE <<EOF
-### リンク名  検索ディレクトリ/ファイル
-x11       /usr{,/local,/X11R6,/pkg}/lib/X11/fonts/truetype
-X11       /usr{,/local,/X11R6,/pkg}/lib/X11/fonts/TrueType
-X11TTF    /usr{,/local,/X11R6,/pkg}/lib/X11/fonts/TTF
-truetype  /usr{,/local}/share/fonts/truetype
-TrueType  /usr{,/local}/share/fonts/TrueType
-TTF	  /usr{,/local}/share/fonts/TTF
-ja        /usr{,/local}/share/fonts/ja/TrueType
-japanese  /usr{,/local}/share/fonts/japanese/TrueType
-openwin   /usr/openwin/lib/locale/ja/X11/fonts/TT
-QtPalmtop /opt/QtPalmtop/lib/X11/fonts/TrueType
-f8sazanami-min  /usr/share/fonts/sazanami-fonts-mincho
-f8sazanami-goth /usr/share/fonts/sazanami-fonts-gothic
-EOF
+mksymlink_gentoo
 
 mksymlink $OPENTYPE <<EOF
 ### リンク名  検索ディレクトリ/ファイル
@@ -222,6 +228,8 @@
 # IPA font http://ossipedia.ipa.go.jp/ipafont/
 	ipam.ttf
 	ipag.ttf
+	ipam.otf
+	ipag.otf
 # DynaLab
 	dfmimp3.ttc
 	dfgotp5.ttc
diff -Naur ptetex3-20090610.orig/common.sh ptetex3-20090610/common.sh
--- ptetex3-20090610.orig/common.sh	2009-05-04 18:32:26.000000000 +0900
+++ ptetex3-20090610/common.sh	2009-11-12 00:23:18.635898734 +0900
@@ -221,12 +221,12 @@
 	echo "You are root."
 	echo "It is RECOMMENDED to execute by a user."
 	echo "Executing by root may cause SERIOUS DAMAGE for your system."
-	echo -n "Are you sure to execute by root AT YOUR OWN RISK? (yes/NO) "
-	read REPLY
-	case $REPLY in
-	    yes) ;;
-	    *) exit 1;;
-	esac
+#	echo -n "Are you sure to execute by root AT YOUR OWN RISK? (yes/NO) "
+#	read REPLY
+#	case $REPLY in
+#	    yes) ;;
+#	    *) exit 1;;
+#	esac
     fi
 }
 
