--- ptetex3-20090610.orig/2extract-src.sh       2009-06-07 10:00:07.000000000 +0200
+++ ptetex3-20090610/2extract-src.sh    2009-09-16 08:05:52.000000000 +0200
@@ -147,25 +147,25 @@
     $MKDIR $SRC/texk/pdvipsk
     $CP $SRC/texk/dvipsk/* $SRC/texk/pdvipsk/ 2> /dev/null
     tarcp $SRC/texk/xdvik $SRC/texk/pxdvik
-    P=p
+    DVIPSK_P=p
 fi # "$JAPANESE" = "international"
 ## for dvipsk
-#tarx archive/dvipsk-jpatch-p*.tar.gz         $SRC/texk/${P}dvipsk
-#cpatch $SRC/texk/${P}dvipsk/dvipsk-*.patch 1 $SRC/texk/${P}dvipsk
-tarx archive/dvipsk-*-p*.tar.gz              $SRC/texk/${P}dvipsk
-cpatch $SRC/texk/${P}dvipsk/dvipsk-*.diff 1  $SRC/texk/${P}dvipsk
-perlprint 's/^kpse_include/ac_include/g'     $SRC/texk/${P}dvipsk/Makefile.in
+#tarx archive/dvipsk-jpatch-p*.tar.gz         $SRC/texk/${DVIPSK_P}dvipsk
+#cpatch $SRC/texk/${DVIPSK_P}dvipsk/dvipsk-*.patch 1 $SRC/texk/${DVIPSK_P}dvipsk
+tarx archive/dvipsk-*-p*.tar.gz              $SRC/texk/${DVIPSK_P}dvipsk
+cpatch $SRC/texk/${DVIPSK_P}dvipsk/dvipsk-*.diff 1  $SRC/texk/${DVIPSK_P}dvipsk
+perlprint 's/^kpse_include/ac_include/g'     $SRC/texk/${DVIPSK_P}dvipsk/Makefile.in

 ## for xdvik
-zpatch archive/xdvik-200?????-tetex-3.0.diff.gz 3 $SRC/texk/${P}xdvik
+zpatch archive/xdvik-200?????-tetex-3.0.diff.gz 3 $SRC/texk/${DVIPSK_P}xdvik
 # fix for FC5
 perlprint 'print "#define NeedWidePrototypes 1\n" if /limits/' \
-    $SRC/texk/${P}xdvik/xdvi.h
+    $SRC/texk/${DVIPSK_P}xdvik/xdvi.h
 # pTeX fontmap
 perlnoprint "s/^ptexmap .*/ptexmap kanjix.map/; print unless /IPA/" \
-    $SRC/texk/${P}xdvik/texmf/xdvi.cfg
+    $SRC/texk/${DVIPSK_P}xdvik/texmf/xdvi.cfg
 # adjust X resource
-cat <<EOF >> $SRC/texk/${P}xdvik/texmf/XDvi || exit 1
+cat <<EOF >> $SRC/texk/${DVIPSK_P}xdvik/texmf/XDvi || exit 1


 !!! ------- added by ptetex3 -------
--- ptetex3-20090610.orig/4make-install.sh      2009-05-04 18:35:49.000000000 +0200
+++ ptetex3-20090610/4make-install.sh   2009-09-16 08:40:47.000000000 +0200
@@ -44,11 +44,9 @@

 echo "[--tetex--] configure"
 cd $SRC || exit
-$MKDIR $PREFIX/share/texmf 2> /dev/null # avoid warning of texmf tree
 ./configure --prefix=$PREFIX --datadir=$DATADIR \
     --disable-multiplatform --with-ptexenc=$KANJI_CODE \
     $CONF_OPTION || exit
-$RMDIR $PREFIX/share/texmf 2> /dev/null # avoid warning of texmf tree
 xdvi_check
 echo "[--tetex--] make"
 $MAKE $MAKE_OPTION || exit
diff -Naur ptetex3-20090610.orig/7font-search.sh ptetex3-20090610/7font-search.sh
--- ptetex3-20090610.orig/7font-search.sh	2008-04-14 18:07:04.000000000 +0900
+++ ptetex3-20090610/7font-search.sh	2008-05-10 09:36:33.000000000 +0900
@@ -58,6 +58,17 @@
     done
 }
 
+mksymlink_gentoo(){
+    $MKDIR $TRUETYPE
+    fonts_dir=/usr/share/fonts/
+    find $fonts_dir -maxdepth 1 -mindepth 1 -type d | while read font_path; do
+	if [ `find $font_path -maxdepth 1 -type f -iname '*.ttf' | wc -l` -gt 0 ]; then
+	    font_name=${font_path/$fonts_dir/}
+	    test -h ${TRUETYPE}/${font_name} && continue
+	    $LN "${font_path}" "${TRUETYPE}/${font_name}"
+	fi
+    done
+}
 
 setrmlgbm() {
     if test -n "$2"; then
@@ -111,10 +122,10 @@
 
 mksymlink $CMAP <<EOF
 ### リンク名  検索ディレクトリ/ファイル
-Acrobat4    /usr{,/local,/sfw}{,/lib}/Acrobat4/Resource/CMap
-Acrobat5    /usr{,/local,/sfw}{,/lib}/Acrobat5/Resource/Font
-Acrobat7    /usr{,/local,/sfw}{,/lib}{,/Adobe}/Acrobat7.0/Resource/CMap
-Acrobat8    /usr{,/local,/sfw}{,/lib}{,/Adobe}/Acrobat8.0/Resource/CMap
+Acrobat4    {/opt,/usr{,/local,/sfw}{,/lib}}/Acrobat4/Resource/CMap
+Acrobat5    {/opt,/usr{,/local,/sfw}{,/lib}}/Acrobat5/Resource/Font
+Acrobat7    {/opt,/usr{,/local,/sfw}{,/lib}}{,/Adobe}/Acrobat7{,.0}/Resource/CMap
+Acrobat8    {/opt,/usr{,/local,/sfw}{,/lib}}{,/Adobe}/{Acrobat,Reader}8{,.0}/Resource/CMap
 ghostscript /usr{,/local,/sfw}/{share,lib}/ghostscript{,/*}/Resource/CMap
 openwinja   /usr/openwin/lib/locale/ja/X11/Resource/CMap
 openwinko   /usr/openwin/lib/locale/ko/X11/Resource/CMap
@@ -122,21 +133,7 @@
 openwintw   /usr/openwin/lib/locale/zh_TW.BIG5/X11/Resource/CMap
 EOF
 
-mksymlink $TRUETYPE <<EOF
-### リンク名  検索ディレクトリ/ファイル
-x11       /usr{,/local,/X11R6,/pkg}/lib/X11/fonts/truetype
-X11       /usr{,/local,/X11R6,/pkg}/lib/X11/fonts/TrueType
-X11TTF    /usr{,/local,/X11R6,/pkg}/lib/X11/fonts/TTF
-truetype  /usr{,/local}/share/fonts/truetype
-TrueType  /usr{,/local}/share/fonts/TrueType
-TTF	  /usr{,/local}/share/fonts/TTF
-ja        /usr{,/local}/share/fonts/ja/TrueType
-japanese  /usr{,/local}/share/fonts/japanese/TrueType
-openwin   /usr/openwin/lib/locale/ja/X11/fonts/TT
-QtPalmtop /opt/QtPalmtop/lib/X11/fonts/TrueType
-f8sazanami-min  /usr/share/fonts/sazanami-fonts-mincho
-f8sazanami-goth /usr/share/fonts/sazanami-fonts-gothic
-EOF
+mksymlink_gentoo
 
 mksymlink $OPENTYPE <<EOF
 ### リンク名  検索ディレクトリ/ファイル
diff -Naur ptetex3-20090610.orig/common.sh ptetex3-20090610/common.sh
--- ptetex3-20090610.orig/common.sh	2008-04-05 17:51:07.000000000 +0900
+++ ptetex3-20090610/common.sh	2008-05-10 09:47:00.000000000 +0900
@@ -220,12 +220,12 @@
 	echo "You are root."
 	echo "It is RECOMMENDED to execute by a user."
 	echo "Executing by root may cause SERIOUS DAMAGE for your system."
-	echo -n "Are you sure to execute by root AT YOUR OWN RISK? (yes/NO) "
-	read REPLY
-	case $REPLY in
-	    yes) ;;
-	    *) exit 1;;
-	esac
+#	echo -n "Are you sure to execute by root AT YOUR OWN RISK? (yes/NO) "
+#	read REPLY
+#	case $REPLY in
+#	    yes) ;;
+#	    *) exit 1;;
+#	esac
     fi
 }
 
