diff -ur seahorse-0.7.9/configure.in seahorse-0.7.9-patched/configure.in
--- seahorse-0.7.9/configure.in	2005-07-03 05:34:41.000000000 +0800
+++ seahorse-0.7.9-patched/configure.in	2005-09-18 20:06:37.000000000 +0800
@@ -251,9 +251,10 @@
 	echo "enabling gedit plugin..."
 	echo "checking for supported versions of gedit..."
 	
-	PKG_CHECK_MODULES(GEDIT, gedit-2.10 >= 2.9.0, , 
-    	[PKG_CHECK_MODULES(GEDIT, gedit-2.8 >= 2.8.0, , 
-	    	[PKG_CHECK_MODULES(GEDIT, gedit-2.6 >= 2.6.0)])])
+	PKG_CHECK_MODULES(GEDIT, gedit-2.12 >= 2.12.0, ,
+		[PKG_CHECK_MODULES(GEDIT, gedit-2.10 >= 2.9.0, , 
+    		[PKG_CHECK_MODULES(GEDIT, gedit-2.8 >= 2.8.0, , 
+	    		[PKG_CHECK_MODULES(GEDIT, gedit-2.6 >= 2.6.0)])])])
 	
 	AC_SUBST(GEDIT_CFLAGS)
 fi
