diff -urNp libx86-0.99.org/Makefile libx86-0.99/Makefile
--- libx86-0.99.org/Makefile	2006-10-30 02:14:18.000000000 +0200
+++ libx86-0.99/Makefile	2007-10-10 21:05:34.000000000 +0200
@@ -1,5 +1,8 @@
 OBJECTS = x86-common.o
-CFLAGS = -O2 -Wall -DDEBUG -g
+CFLAGS ?= -O2 -Wall -DDEBUG -g
+PREFIX ?= /usr
+LIBDIR ?= $(PREFIX)/lib
+INCDIR ?= $(PREFIX)/include
 
 ifeq ($(BACKEND), x86emu)
 	OBJECTS += thunk.o x86emu/decode.o x86emu/debug.o x86emu/fpu.o \
@@ -31,7 +34,7 @@ clean: objclean
 	rm -f *.so.1 *.a
 
 install: libx86.so.1
-	install -D libx86.so.1 $(DESTDIR)/usr/lib/libx86.so.1
-	install -D libx86.a $(DESTDIR)/usr/lib/libx86.a
-	ln -sf /usr/lib/libx86.so.1 $(DESTDIR)/usr/lib/libx86.so
-	install -D lrmi.h $(DESTDIR)/usr/include/libx86.h
+	install -D libx86.so.1 $(DESTDIR)/$(LIBDIR)/libx86.so.1
+	install -D libx86.a $(DESTDIR)/$(LIBDIR)/libx86.a
+	ln -sf libx86.so.1 $(DESTDIR)/$(LIBDIR)/libx86.so
+	install -D lrmi.h $(DESTDIR)/$(INCDIR)/libx86.h
