diff --git a/src/inclan/unix.for b/src/inclan/unix.for
index 8650129..f08eeef 100644
--- a/src/inclan/unix.for
+++ b/src/inclan/unix.for
@@ -206,7 +206,7 @@ c     ==================================================================
       end if
       if (iwbeg.lt.0) then
         iwbeg=igttim
-        iexpir=(julday(12,31,2010)-julday(1,1,1970))*24*3600
+        iexpir=(julday(12,31,2100)-julday(1,1,1970))*24*3600
         if (iwbeg.gt.iexpir) call fatmsg ('License expired.')
         j=(iexpir-iwbeg)/(24*3600)
         if (j.lt.1000)
