--- ./install.py.orig	2008-03-01 12:46:17.000000000 +0100
+++ ./install.py	2008-03-11 15:34:36.000000000 +0100
@@ -184,12 +184,10 @@
     global cfg, distDir
     
     pdir = os.path.join(cfg['mdir'], "eric4plugins")
-    if distDir:
-        pdir = distDir + pdir
     fname = os.path.join(pdir, "__init__.py")
     if not os.path.exists(fname):
         if not os.path.exists(pdir):
-            os.mkdir(pdir,  0755)
+            os.makedirs(pdir,  0755)
         f = open(fname, "wb")
         f.write(\
 '''# -*- coding: utf-8 -*-
@@ -208,11 +206,14 @@
     
     @return name of the directory containing the API files (string)
     """
-    global cfg
+    global cfg, distDir
     
     try:
         from PyQt4 import pyqtconfig
-        qtDataDir = pyqtconfig._pkg_config["qt_data_dir"]
+        if distDir:
+            qtDataDir = distDir + pyqtconfig._pkg_config["qt_data_dir"]
+        else:
+            qtDataDir = pyqtconfig._pkg_config["qt_data_dir"]
     except (AttributeError, ImportError):
         print
         print "Please enter the name of the Qt data directory."
@@ -474,7 +475,7 @@
        cfg['ericTemplatesDir'],
        cfg['ericCodeTemplatesDir'], cfg['ericOthersDir'],
        cfg['bindir'], cfg['mdir'], 
-       getAPIsPath(), apis)
+       getAPIsPath()[len(distDir):], apis)
     copyToFile(fn, config)
 
 def doDependancyChecks():    
