diff -ru ./Makefile.in /var/tmp/portage/net-irc/unrealircd-3.2.8/work/Unreal3.2/Makefile.in
--- Makefile.in	2006-06-05 19:16:34.000000000 -0400
+++ Makefile.in	2009-03-03 18:59:32.000000000 -0500
@@ -34,11 +34,11 @@
 #
 
 #XCFLAGS=-O -g -export-dynamic
-IRCDLIBS=@IRCDLIBS@ @TRELIBS@ @CARESLIBS@
+IRCDLIBS=@IRCDLIBS@ @TRE_LIBS@ @CARESLIBS@
 CRYPTOLIB=@CRYPTOLIB@
 OPENSSLINCLUDES=
 
-XCFLAGS=-I@TREINCDIR@ -I@CARESINCDIR@ @CARESLIBDIR@ @CFLAGS@
+XCFLAGS=@TRE_CFLAGS@ -I@CARESINCDIR@ @CARESLIBDIR@ @CFLAGS@
 #
 # use the following on MIPS:
 #CFLAGS= -systype bsd43 -DSYSTYPE_BSD43 -I$(INCLUDEDIR)
Only in /var/tmp/portage/net-irc/unrealircd-3.2.8/work/Unreal3.2: Makefile.in.~1.1.2.23.2.6.~
diff -ru ./configure.in /var/tmp/portage/net-irc/unrealircd-3.2.8/work/Unreal3.2/configure.in
--- autoconf/configure.in	2009-02-01 11:43:33.000000000 -0500
+++ autoconf/configure.in	2009-03-03 19:01:23.000000000 -0500
@@ -392,6 +392,7 @@
 	AC_DEFINE(DISABLE_USERMOD))
 AC_ARG_WITH(operoverride-verify, [AC_HELP_STRING([--with-operoverride-verify], [Require opers to invite themselves to +s/+p channels])],
 	AC_DEFINE(OPEROVERRIDE_VERIFY))
+AC_ARG_WITH(system-tre, [AC_HELP_STRING([--with-system-tre], [Use the system tre package instead of bundled, discovered using pkg-config])], [], [with_system_tre=no ])
 CHECK_SSL
 CHECK_ZLIB
 CHECK_LIBCURL
@@ -424,6 +425,7 @@
 ;;
 esac
 
+AS_IF([test "x$with_system_tre" = "xno"],[
 dnl REMEMBER TO CHANGE WITH A NEW TRE RELEASE!
 tre_version="0.7.5"
 AC_MSG_RESULT(extracting TRE regex library)
@@ -446,16 +448,20 @@
 $ac_cv_prog_MAKER || exit 1
 AC_MSG_RESULT(installing TRE regex library)
 $ac_cv_prog_MAKER install || exit 1
-TREINCDIR="$cur_dir/extras/regexp/include"
-AC_SUBST(TREINCDIR)
+TRE_CFLAGS="-I$cur_dir/extras/regexp/include"
+AC_SUBST(TRE_CFLAGS)
 if test "x$ac_cv_path_PKGCONFIG" = "x" ; then
-	TRELIBS="-L../extras/regexp/lib -ltre"
+	TRE_LIBS="-L../extras/regexp/lib -ltre"
 else
-	TRELIBS=`$ac_cv_path_PKGCONFIG --libs tre.pc`
+	TRE_LIBS=`$ac_cv_path_PKGCONFIG --libs tre.pc`
 fi
-AC_SUBST(TRELIBS)
+AC_SUBST(TRE_LIBS)
 cd $cur_dir
-
+],[
+dnl use pkgconfig for tre:
+PKG_PROG_PKG_CONFIG(0.17)
+PKG_CHECK_MODULES([TRE], tre >= 0.7.5)
+])
 
 dnl REMEMBER TO CHANGE WITH A NEW C-ARES RELEASE!
 cares_version="1.6.0"
