Index: libtorrent-rasterbar-0.14.12/include/libtorrent/udp_socket.hpp
===================================================================
--- libtorrent-rasterbar-0.14.12.orig/include/libtorrent/udp_socket.hpp
+++ libtorrent-rasterbar-0.14.12/include/libtorrent/udp_socket.hpp
@@ -37,7 +37,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/buffer.hpp"
 #include "libtorrent/session_settings.hpp"
 
-#include <vector>
+#include <list>
 #include <boost/function.hpp>
 #include <boost/thread/mutex.hpp>
 
