--- configure.in.orig	2010-02-19 08:03:57.000000000 +0100
+++ configure.in	2010-03-24 01:04:37.000000000 +0100
@@ -157,7 +157,6 @@
 
 dnl Apply boost config.
 CPPFLAGS="$CPPFLAGS $BOOST_CPPFLAGS"
-LDFLAGS="$LDFLAGS $BOOST_LDFLAGS"
 
 if [[ "x$BUILD_TESTCLIENT" != "xno" ]]; then
 	CLIENT_TEST_BIN=client_test;
@@ -519,6 +518,9 @@
 dnl Compile time options.
 AC_SUBST(COMPILETIME_OPTIONS)
 
+dnl Check for clock_gettime
+AC_SEARCH_LIBS([clock_gettime],[rt])
+
 AC_OUTPUT([Makefile
 	   src/Makefile
 	   include/Makefile
