diff -ur libtorrent-0.13.orig/include/libtorrent/disk_io_thread.hpp libtorrent-0.13/include/libtorrent/disk_io_thread.hpp
--- libtorrent-0.13.orig/include/libtorrent/disk_io_thread.hpp	2009-05-27 10:45:06.000000000 +0200
+++ libtorrent-0.13/include/libtorrent/disk_io_thread.hpp	2009-05-27 11:33:50.000000000 +0200
@@ -38,6 +38,7 @@
 #endif
 
 #include "libtorrent/storage.hpp"
+#include <boost/thread/condition.hpp>
 #include <boost/thread/thread.hpp>
 #include <boost/function.hpp>
 #include <boost/thread/mutex.hpp>
