diff -u -r1.6 -r1.7
--- man/Makefile.am	2005/12/06 22:48:42	1.6
+++ man/Makefile.am	2006/02/12 18:19:20	1.7
@@ -27,6 +27,8 @@
 
 libmandir = $(LIB_MAN_DIR)
 
+LIB_MAN_DIR_SUFFIX = $(LIB_MAN_DIR:@mandir@/man%=%)
+
 libman_PRE = XRes.man
 
 BUILT_SOURCES = shadows.DONE
@@ -69,5 +71,5 @@
 shadows.DONE:
 	-rm -f $(XRes_shadows:=.@LIB_MAN_SUFFIX@)
 	(for i in $(XRes_shadows:=.@LIB_MAN_SUFFIX@) ; do \
-	 echo .so man$(LIB_MAN_SUFFIX)/XRes.$(LIB_MAN_SUFFIX) > $$i; \
+	 echo .so man$(LIB_MAN_DIR_SUFFIX)/XRes.$(LIB_MAN_SUFFIX) > $$i; \
 	 done)
