diff -NurpP trac-webadmin-0.1.2/webadmin/templates/admin_basics.cs webadmin/webadmin/templates/admin_basics.cs
--- trac-webadmin-0.1.2/webadmin/templates/admin_basics.cs	2006-11-13 23:48:04.000000000 +0100
+++ webadmin/webadmin/templates/admin_basics.cs	2008-03-04 11:08:15.525564021 +0100
@@ -10,7 +10,7 @@
   </div>
   <div class="field">
    <label>URL:<br />
-    <input type="text" name="url" size="48 "value="<?cs
+    <input type="text" name="url" size="48" value="<?cs
       var:admin.project.url ?>" />
    </label>
   </div>
diff -NurpP trac-webadmin-0.1.2/webadmin/templates/admin_enum.cs webadmin/webadmin/templates/admin_enum.cs
--- trac-webadmin-0.1.2/webadmin/templates/admin_enum.cs	2006-11-13 23:48:05.000000000 +0100
+++ webadmin/webadmin/templates/admin_enum.cs	2008-03-04 11:08:15.525564021 +0100
@@ -5,7 +5,7 @@ if:admin.enum.name ?>
   <fieldset>
    <legend>Modify <?cs var:admin.enum.label_singular ?></legend>
    <div class="field">
-    <label>Name: <input type="text" name="name"value="<?cs
+    <label>Name: <input type="text" name="name" value="<?cs
       var:admin.enum.name ?>" /></label>
    </div>
    <div class="buttons">
diff -NurpP trac-webadmin-0.1.2/webadmin/templates/admin_log.cs webadmin/webadmin/templates/admin_log.cs
--- trac-webadmin-0.1.2/webadmin/templates/admin_log.cs	2006-11-13 23:48:06.000000000 +0100
+++ webadmin/webadmin/templates/admin_log.cs	2008-03-04 11:08:15.529564249 +0100
@@ -30,7 +30,7 @@
   var:admin.log.dir ?></code>).</p>
  </div>
  <script type="text/javascript">
-   var logType = document.getElementById("log_type");
+   var log_type = document.getElementById("log_type");
    var enableLevelAndFile = function() {
      enableControl("log_level", log_type.selectedIndex > 0);
      enableControl("log_file",
diff -NurpP trac-webadmin-0.1.2/webadmin/templates/admin_milestone.cs webadmin/webadmin/templates/admin_milestone.cs
--- trac-webadmin-0.1.2/webadmin/templates/admin_milestone.cs	2006-11-13 23:48:05.000000000 +0100
+++ webadmin/webadmin/templates/admin_milestone.cs	2008-03-04 11:08:15.529564249 +0100
@@ -5,7 +5,7 @@ if:admin.milestone.name ?>
   <fieldset>
    <legend>Modify Milestone:</legend>
    <div class="field">
-    <label>Name:<br /> <input type="text" name="name"value="<?cs
+    <label>Name:<br /> <input type="text" name="name" value="<?cs
       var:admin.milestone.name ?>" /></label>
    </div>
    <div class="field">
diff -NurpP trac-webadmin-0.1.2/webadmin/templates/admin_plugin.cs webadmin/webadmin/templates/admin_plugin.cs
--- trac-webadmin-0.1.2/webadmin/templates/admin_plugin.cs	2006-11-13 23:48:05.000000000 +0100
+++ webadmin/webadmin/templates/admin_plugin.cs	2008-03-04 11:08:15.525564021 +0100
@@ -4,7 +4,7 @@
  <fieldset>
   <legend>Install Plugin:</legend>
   <div class="field">
-   <label>File: <input type="file" name="egg_file"<?cs
+   <label>File: <input type="file" name="plugin_file"<?cs
      if:admin.readonly ?> disabled="disabled"<?cs /if ?> /></label>
   </div>
   <p class="help"><?cs
@@ -63,7 +63,7 @@
     if:!component.required ?><input type="hidden" name="component" value="<?cs
      var:component.module ?>.<?cs var:component.name ?>" /><?cs
     /if ?><input type="checkbox" name="enable" value="<?cs
-     var:component.module ?>.<?cs var:component.name ?>"<?cs 
+     var:component.module ?>.<?cs var:component.name ?>"<?cs
      if:component.enabled ?> checked="checked"<?cs
      /if ?><?cs
      if:component.required ?> disabled="disabled"<?cs
diff -NurpP trac-webadmin-0.1.2/webadmin/templates/admin_version.cs webadmin/webadmin/templates/admin_version.cs
--- trac-webadmin-0.1.2/webadmin/templates/admin_version.cs	2006-11-13 23:48:05.000000000 +0100
+++ webadmin/webadmin/templates/admin_version.cs	2008-03-04 11:08:15.525564021 +0100
@@ -5,7 +5,7 @@ if:admin.version.name ?>
   <fieldset>
    <legend>Modify Version:</legend>
    <div class="field">
-    <label>Name:<br /> <input type="text" name="name"value="<?cs
+    <label>Name:<br /> <input type="text" name="name" value="<?cs
       var:admin.version.name ?>" /></label>
    </div>
    <div class="field">
