Fix building on FreeBSD systems.
Index: domino-0.4/configure.in.in
===================================================================
--- domino-0.4.orig/configure.in.in
+++ domino-0.4/configure.in.in
@@ -1 +1,3 @@
 #MIN_CONFIG
+
+AC_CHECK_HEADERS([endian.h sys/endian.h])
Index: domino-0.4/domino/misc.cpp
===================================================================
--- domino-0.4.orig/domino/misc.cpp
+++ domino-0.4/domino/misc.cpp
@@ -22,19 +22,30 @@
 #include "misc.h"
 #include <qimage.h>
 #include <qpixmap.h>
-#include <endian.h>
-
-#if __BYTE_ORDER == __LITTLE_ENDIAN
-#undef WORDS_BIGENDIAN
-#define WORDS_LITTLEENDIAN 1
-#endif
-#if __BYTE_ORDER == __BIG_ENDIAN
-	#undef WORDS_LITTLEENDIAN
-	#define WORDS_BIGENDIAN 1
-#endif
-
 
+#include <config.h>
 
+#ifdef HAVE_ENDIAN_H
+# include <endian.h>
+# if __BYTE_ORDER == __LITTLE_ENDIAN
+#  undef WORDS_BIGENDIAN
+#  define WORDS_LITTLEENDIAN 1
+# endif
+# if __BYTE_ORDER == __BIG_ENDIAN
+#  undef WORDS_LITTLEENDIAN
+#  define WORDS_BIGENDIAN 1
+# endif
+#elif HAVE_SYS_ENDIAN_H /* *BSD in general */
+# include <sys/endian.h>
+# if _BYTE_ORDER == _LITTLE_ENDIAN
+#  undef WORDS_BIGENDIAN
+#  define WORDS_LITTLEENDIAN 1
+# endif
+# if _BYTE_ORDER == _BIG_ENDIAN
+#  undef WORDS_LITTLEENDIAN
+#  define WORDS_BIGENDIAN 1
+# endif
+#endif
 
 QColor alphaBlendColors(const QColor &bgColor, const QColor &fgColor, const int a)
 {
