diff -uNr toshutils-2.0.1.orig/configure.in toshutils-2.0.1/configure.in
--- toshutils-2.0.1.orig/configure.in	Sun Jan 27 08:30:17 2002
+++ toshutils-2.0.1/configure.in	Fri Feb  8 17:16:58 2002
@@ -4,7 +4,7 @@
 AC_REVISION([configure.in 1.2])
 AC_INIT(src/sci.c)
 
-PROGRAMS="fan svpw tpasswd dispswitch tdocked thotswap tbacklight ownerstring alarm"
+PROGRAMS="fan svpw tpasswd tdocked thotswap tbacklight ownerstring alarm"
 
 VERSION=2.0.1
 AC_SUBST(VERSION)
@@ -63,13 +63,10 @@
   EXTRAFLAGS=""
   PROGRAMS="$PROGRAMS hotkey wmtuxtime"       # programs to build if X11 found
   AC_CHECK_LIB(gtk, gtk_main,
-    PROGRAMS="$PROGRAMS tuxtime-conf", # progams to build only if X+GTK found
+    PROGRAMS="$PROGRAMS tuxtime-conf dispswitch", # progams to build only if X+GTK found
     AC_MSG_WARN(*** GTK+ libraries not found.  X code will not be built. ***),
     `gtk-config --libs`)
 else
-  AC_MSG_ERROR([
-*** X libraries or include files not found. Check 'config.log' for
-*** more details.])
   XLIB=""
   X_CFLAGS=""
   X_LIBS=""
