--- kdenetwork-3.3.0/kopete/protocols/oscar/oscarsocket/rtf.cc	2004-07-17 21:10:33.000000000 +0200
+++ rtf.cc	2004-08-23 12:17:00.405431600 +0200
@@ -1790,7 +1790,7 @@
         case TAG_FONT_COLOR:
             {
                // RTF colors are 1-based; colors[] is a 0-based array.
-               if (t.param > colors.size())
+               if (t.param > colors.size() || t.param == 0)
                    break;
                QColor &c = colors[t.param-1];
                PrintUnquoted("<span style=\"color:#%02X%02X%02X\">", c.red(), c.green(), c.blue());
