Index: kaddressbook/xxport/configure.in.in
===================================================================
--- kaddressbook/xxport/configure.in.in	(revision 433932)
+++ kaddressbook/xxport/configure.in.in	(working copy)
@@ -1,6 +1,2 @@
 # $Id: configure-fix-kdepim-gnokii-3.4.patch,v 1.1 2005/07/28 21:16:10 danarmak Exp $
-
-KDE_CHECK_BLUETOOTH
-KDE_CHECK_GNOKII
-
 AM_CONDITIONAL(compile_GNOKIIXXPORT, test -n "$LIB_GNOKII")
Index: kaddressbook/xxport/configure.in.bot
===================================================================
--- kaddressbook/xxport/configure.in.bot	(revision 433932)
+++ kaddressbook/xxport/configure.in.bot	(working copy)
@@ -1,4 +1,4 @@
-if test -z "$LIB_GNOKII"; then
+if test "x$with_gnokii" = xcheck && test -z "$LIB_GNOKII"; then
   echo ""
   echo "libgnokii (http://www.gnokii.org) is missing. The KDE Addressbook mobile phone import/export filter will not be available."
   echo ""
Index: configure.in.in
===================================================================
--- configure.in.in	(revision 433932)
+++ configure.in.in	(working copy)
@@ -46,7 +46,6 @@
 [
   KDE_CHECK_HEADERS(gnokii.h)
   AC_REQUIRE([AC_HAVE_XPM])
-  LIB_GNOKII=
   AC_CHECK_LIB(gnokii, gn_cfg_read,
        [ LIB_GNOKII="-lgnokii $LIB_BLUETOOTH" ],
        [
@@ -58,7 +57,22 @@
   AC_SUBST(LIB_GNOKII)
 ])
 
+AC_ARG_WITH(gnokii,
+  [AS_HELP_STRING(--with-gnokii,
+    [enable support for mobile phones through gnokii @<:@default=check@:>@])],
+  [], with_gnokii=check)
 
+LIB_GNOKII=
+if test "x$with_gnokii" != xno; then
+  # for kaddressbook/xxport/ and kmobile/devices/gnokii/
+  KDE_CHECK_BLUETOOTH
+  KDE_CHECK_GNOKII
+
+  if test "x$with_gnokii" != xcheck && test -z "$LIB_GNOKII"; then
+    AC_MSG_FAILURE([--with-gnokii was given, but test for gnokii failed])
+  fi
+fi
+
 dnl the below is copied from kdenetwork's configure as too many apps moved
 dnl and noone can say which tests are needed and which are obsolete for kdepim ;(
 
