diff -urbB picptk-0.5a.old/IfParPort.C picptk-0.5a/IfParPort.C
--- picptk-0.5a.old/IfParPort.C	1998-12-30 17:40:31.000000000 -0800
+++ picptk-0.5a/IfParPort.C	2003-10-20 02:42:14.000000000 -0700
@@ -27,6 +27,9 @@
  *    Byte wide transfers GetByte()
  *    Support bidiriectional pins 
  */
+
+using namespace std;
+
 #include <IfParPort.h>
 #include <stdio.h>
 #include <unistd.h>
@@ -72,7 +75,8 @@
     { out, 1, 2, 0x08 },	/* pin 17	(out)	!SI	*/
     {  nc, 0, 1, 0x00 },	/* pin 18	GND		*/
   };
-  cPinInfo = pinfo;
+  //cPinInfo = pinfo;
+  for(uint i = 0; i < kMaxPins; i++) cPinInfo[i] = pinfo[i];
   if(port >= kMaxPorts) 
     throw("Bad printer port number");
   cPort = ports[port];
diff -urbB picptk-0.5a.old/IfParPort.h picptk-0.5a/IfParPort.h
--- picptk-0.5a.old/IfParPort.h	1998-12-30 17:40:31.000000000 -0800
+++ picptk-0.5a/IfParPort.h	2003-10-20 02:42:14.000000000 -0700
@@ -32,6 +32,7 @@
 #ifndef _ParPort_h_
 #define _ParPort_h_
 
+using namespace std;
 #include <stdlib.h>		// for uint
 #include <UtEnvLite.h>
 
@@ -45,12 +46,14 @@
   uint cState[kPortSize];	// in bytes...
 
   typedef enum { nc, in, out, } Dir_t;
-  struct PinInfo_t {
+  typedef struct PinInfo_t {
     Dir_t dir;
     char sense;
     char offset;
     char mask;
-  } cPinInfo[kMaxPins];
+  };
+
+  PinInfo_t cPinInfo[kMaxPins];
 
 
   // Gain access to parallel port, return 0 on success...
diff -urbB picptk-0.5a.old/IoIntel.C picptk-0.5a/IoIntel.C
--- picptk-0.5a.old/IoIntel.C	1998-12-30 17:40:32.000000000 -0800
+++ picptk-0.5a/IoIntel.C	2003-10-20 02:42:14.000000000 -0700
@@ -22,6 +22,7 @@
  * Intel Hex file I/O formatting and such...
  *
  */
+using namespace std;
 #include <IoIntel.h>
 #include <stdio.h>		// for sscanf
 // ParseLine(...) - parse one line of Intel Hex8 format
@@ -65,7 +66,7 @@
 // If length is zero, writes end of file marker...
 // This should be smarter about supressing zeros, maybe?
 bool
-IntelHex::OutputLine(ostream &os, int len, uint addr = 0, char *buffer = 0) {
+IntelHex::OutputLine(ostream &os, int len, uint addr, char *buffer) {
   char line[200];
   int sum = 0;
   int type = (len) ? 0 : 1;
diff -urbB picptk-0.5a.old/IoIntel.h picptk-0.5a/IoIntel.h
--- picptk-0.5a.old/IoIntel.h	1998-12-30 17:40:32.000000000 -0800
+++ picptk-0.5a/IoIntel.h	2003-10-20 02:43:00.000000000 -0700
@@ -24,10 +24,11 @@
 
 #ifndef IntelHex_h
 #define IntelHex_h
+using namespace std;
 
 #include <stdlib.h>		// for uint
-#include <fstream.h>
-#include <iostream.h>
+#include <fstream>
+#include <iostream>
 
 // This probably should be more general and read multiple formats...
 class IntelHex {
diff -urbB picptk-0.5a.old/Makefile.am picptk-0.5a/Makefile.am
--- picptk-0.5a.old/Makefile.am	1998-12-30 18:23:05.000000000 -0800
+++ picptk-0.5a/Makefile.am	2003-10-20 02:42:14.000000000 -0700
@@ -24,7 +24,7 @@
 		.picprc          \
 		picptk.itk
 
-INCLUDES = -I . -I /usr/include/tcl8.0-int/generic/
+INCLUDES = -I. -I/usr/include/tcl8.0-int/generic/ -I/usr/lib/tcl8.3/include/generic -I/usr/lib/tcl8.4/include/generic 
 
 ##dnl info_TEXINFOS = picptk.texi
 
diff -urbB picptk-0.5a.old/PicCore.C picptk-0.5a/PicCore.C
--- picptk-0.5a.old/PicCore.C	1998-12-30 17:40:32.000000000 -0800
+++ picptk-0.5a/PicCore.C	2003-10-20 02:42:14.000000000 -0700
@@ -19,6 +19,7 @@
  *
  * $Id: picptk-0.5a-r1-headerfix.patch,v 1.1 2006/02/25 11:46:50 robbat2 Exp $
  */
+using namespace std;
 #include <PicCore.h>
 #include <IoIntel.h>
 #include <string.h>		// for strcat()
@@ -504,7 +505,7 @@
   SetPin(pin_do, 0);
 }
  // SetPin() - Hook to the driver's set pin...
-void Pic::SetPin(Pin_t pin, int val = 1) {
+void Pic::SetPin(Pin_t pin, int val) {
   if(!cDriver) throw (cDriver);	// Get the hint?
   cDriver->SetPin(pin, val);
 }
diff -urbB picptk-0.5a.old/PicCore.h picptk-0.5a/PicCore.h
--- picptk-0.5a.old/PicCore.h	1998-12-30 17:40:32.000000000 -0800
+++ picptk-0.5a/PicCore.h	2003-10-20 02:43:17.000000000 -0700
@@ -21,8 +21,9 @@
  */
 #ifndef Pic_h
 #define Pic_h
+using namespace std;
 
-#include <fstream.h>
+#include <fstream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <UtEnvLite.h>		// for Environment
@@ -130,7 +131,7 @@
   int VerifySegment(Segment &seg, ProgressCb_t cb=0, void *arg=0);
   int EraseSegment (Segment &seg, ProgressCb_t cb=0, void *arg=0);
   int InitSegment  (Segment &seg, ProgressCb_t cb=0, void *arg=0) { seg.Init(); return(0); }
-  int DumpSegment  (Segment &seg, ProgressCb_t cb=0, void *arg=0) { seg.Dump((ostream &)*arg); return(0); }
+  int DumpSegment  (Segment &seg, ProgressCb_t cb=0, void *arg=0) { seg.Dump((ostream &)arg); return(0); }
   int BlankCheckSegment (Segment &seg, ProgressCb_t cb=0, void *arg=0);
 
   // Set/Get based on Pin name
@@ -149,13 +150,13 @@
   const char *GetPref(const char *attr) { return(cEnv.Get(attr)); }
 
   // Make user functions apply to all segments...
-  int Read  (ProgressCb_t cb = 0, void *obj = 0) { return(EachSegment(&ReadSegment,   cb, obj)); }
-  int Write (ProgressCb_t cb = 0, void *obj = 0) { return(EachSegment(&WriteSegment,  cb, obj)); }
-  int Verify(ProgressCb_t cb = 0, void *obj = 0) { return(EachSegment(&VerifySegment, cb, obj)); }
-  int Erase (ProgressCb_t cb = 0, void *obj = 0) { return(EachSegment(&EraseSegment,  cb, obj)); }
-  int DumpRam (ostream &os)                        { return(EachSegment(&DumpSegment,    0, &os, false)); }
+  int Read  (ProgressCb_t cb = 0, void *obj = 0) { return(EachSegment(&Pic::ReadSegment,   cb, obj)); }
+  int Write (ProgressCb_t cb = 0, void *obj = 0) { return(EachSegment(&Pic::WriteSegment,  cb, obj)); }
+  int Verify(ProgressCb_t cb = 0, void *obj = 0) { return(EachSegment(&Pic::VerifySegment, cb, obj)); }
+  int Erase (ProgressCb_t cb = 0, void *obj = 0) { return(EachSegment(&Pic::EraseSegment,  cb, obj)); }
+  int DumpRam (ostream &os)                        { return(EachSegment(&Pic::DumpSegment,    0, &os, false)); }
   int LoadRam (const char *file);
-  int InitRam ()                                 { return(EachSegment(&InitSegment,    0, 0, false)); }
+  int InitRam ()                                 { return(EachSegment(&Pic::InitSegment,    0, 0, false)); }
 
   // Configuration functions...
   virtual const char **Chips();		// What chip types we support...
diff -urbB picptk-0.5a.old/PicDriver.h picptk-0.5a/PicDriver.h
--- picptk-0.5a.old/PicDriver.h	1998-12-30 17:40:32.000000000 -0800
+++ picptk-0.5a/PicDriver.h	2003-10-20 02:43:19.000000000 -0700
@@ -25,6 +25,7 @@
  */
 #ifndef _PicDriver_h_
 #define _PicDriver_h_
+using namespace std;
 
 #include <stdlib.h>		// for uint
 #include <termios.h>		// for serial port support
diff -urbB picptk-0.5a.old/PicMem.h picptk-0.5a/PicMem.h
--- picptk-0.5a.old/PicMem.h	1998-12-30 17:40:32.000000000 -0800
+++ picptk-0.5a/PicMem.h	2003-10-20 02:43:22.000000000 -0700
@@ -21,9 +21,10 @@
  */
 #ifndef Segment_h
 #define Segment_h
+using namespace std;
 
 #include <stdlib.h>		// for uint
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>		// for sprintf()
 #include <string.h>		// for strcat()
 
diff -urbB picptk-0.5a.old/PicParallel.C picptk-0.5a/PicParallel.C
--- picptk-0.5a.old/PicParallel.C	1998-12-30 17:40:32.000000000 -0800
+++ picptk-0.5a/PicParallel.C	2003-10-20 02:42:14.000000000 -0700
@@ -27,6 +27,7 @@
  * into actions...  
  */
 
+using namespace std;
 #include <PicParallel.h>
 #include <unistd.h>		// for open(), close(), ...
 
@@ -82,7 +83,7 @@
 }
 
 // Set specified pin...
-void ParallelDriver::SetPin(Pic::Pin_t pin, int val = 1) {
+void ParallelDriver::SetPin(Pic::Pin_t pin, int val) {
   int pinnum = cPinMap[pin];
   if(pinnum < 0) cPort->SetPin(-pinnum, val^1);
   else cPort->SetPin(pinnum, val);
diff -urbB picptk-0.5a.old/PicParallel.h picptk-0.5a/PicParallel.h
--- picptk-0.5a.old/PicParallel.h	1998-12-30 17:40:32.000000000 -0800
+++ picptk-0.5a/PicParallel.h	2003-10-20 02:43:24.000000000 -0700
@@ -26,6 +26,7 @@
  */
 #ifndef _PicParallel_h_
 #define _PicParallel_h_
+using namespace std;
 
 #include <PicDriver.h>
 #include <IfParPort.h>
diff -urbB picptk-0.5a.old/PicPersonality.C picptk-0.5a/PicPersonality.C
--- picptk-0.5a.old/PicPersonality.C	1998-12-30 17:40:33.000000000 -0800
+++ picptk-0.5a/PicPersonality.C	2003-10-20 02:42:14.000000000 -0700
@@ -24,6 +24,7 @@
  * ToDo:
  *    Place this information in a config file rather than in code...
  */
+using namespace std;
 #include <PicCore.h>
 
  // Pic::cAllPics - Description of all supported Pic members
diff -urbB picptk-0.5a.old/PicSerial.C picptk-0.5a/PicSerial.C
--- picptk-0.5a.old/PicSerial.C	1998-12-30 17:40:33.000000000 -0800
+++ picptk-0.5a/PicSerial.C	2003-10-20 02:42:14.000000000 -0700
@@ -34,6 +34,7 @@
  * and they need a non-standard tty driver to hold TX low indefinately.
  * The driver appears to be a seperate kernel patch...
  */
+using namespace std;
 
 #include <PicSerial.h>
 #include <unistd.h>		// for open(), close(), ...
@@ -54,18 +55,18 @@
 SerialDriver::SerialDriver(Environment &env) : Driver(env) {
   
 #ifndef TIOCCBRK
-  throw("No TIOCCBRK ioctl?!
-You probably can't use a serial port
-programmer unless you install a kernel
+  throw("No TIOCCBRK ioctl?!\n\
+You probably can't use a serial port\n\
+programmer unless you install a kernel\n\
 patch.");
 
   // That'll get user's attention, now let compile it anyway...
 #define TIOCCBRK 0
 #endif
 #ifndef TIOSCBRK
-  throw("No TIOCCBRK ioctl?!
-You probably can't use a serial port
-programmer unless you install a kernel
+  throw("No TIOCCBRK ioctl?!\n\
+You probably can't use a serial port\n\
+programmer unless you install a kernel\n\
 patch.");
 #define TIOSCBRK 0
 #endif // TIOSCBRK
@@ -94,7 +95,7 @@
 }
 
 // Set specified pin...
-void SerialDriver::SetPin(Pic::Pin_t pin, int val = 1) {
+void SerialDriver::SetPin(Pic::Pin_t pin, int val) {
   int tmp;
   val &= 1;
   switch(pin) {
diff -urbB picptk-0.5a.old/PicSerial.h picptk-0.5a/PicSerial.h
--- picptk-0.5a.old/PicSerial.h	1998-12-30 17:40:33.000000000 -0800
+++ picptk-0.5a/PicSerial.h	2003-10-20 02:43:25.000000000 -0700
@@ -26,6 +26,7 @@
  */
 #ifndef _PicSerial_h_
 #define _PicSerial_h_
+using namespace std;
 
 #include <PicDriver.h>
 /* Serial port programmer...
diff -urbB picptk-0.5a.old/UiItk.C picptk-0.5a/UiItk.C
--- picptk-0.5a.old/UiItk.C	1998-12-31 08:33:53.000000000 -0800
+++ picptk-0.5a/UiItk.C	2003-10-20 02:42:50.000000000 -0700
@@ -27,6 +27,7 @@
  * a the need...)
  *
  */
+using namespace std;
 
 #include <UiItk.h>
 
@@ -34,7 +35,7 @@
 #include <itk.h>
 #endif
 
-#include <strstream.h>		// for ostrstream
+#include <strstream>		// for ostrstream
 
 
 /* Gui dispatch table for TCL commands...
diff -urbB picptk-0.5a.old/UiItk.h picptk-0.5a/UiItk.h
--- picptk-0.5a.old/UiItk.h	1998-12-30 17:40:33.000000000 -0800
+++ picptk-0.5a/UiItk.h	2003-10-20 02:43:27.000000000 -0700
@@ -30,6 +30,7 @@
 
 // Even though the interface may be incrTk, 
 // <tk.h> contains what we need here...
+using namespace std;
 #include <tk.h>
 #include <PicCore.h>
 
diff -urbB picptk-0.5a.old/UtEnvLite.h picptk-0.5a/UtEnvLite.h
--- picptk-0.5a.old/UtEnvLite.h	1998-12-30 17:40:33.000000000 -0800
+++ picptk-0.5a/UtEnvLite.h	2003-10-20 02:43:31.000000000 -0700
@@ -27,8 +27,9 @@
  */
 #ifndef _Environment_h_
 #define _Environment_h_
+using namespace std;
 
-#include <fstream.h>
+#include <fstream>
 #include <stdio.h>
 #include <stdlib.h>
 
diff -urbB picptk-0.5a.old/UtRealTime.h picptk-0.5a/UtRealTime.h
--- picptk-0.5a.old/UtRealTime.h	1998-12-30 17:40:33.000000000 -0800
+++ picptk-0.5a/UtRealTime.h	2003-10-20 02:43:33.000000000 -0700
@@ -25,6 +25,7 @@
 
 #ifndef RealTime_h
 #define RealTime_h
+using namespace std;
 
 #include <time.h>		// for uint
 #include <sched.h>		// for sched_setscheduler()
diff -urbB picptk-0.5a.old/configure.in picptk-0.5a/configure.in
--- picptk-0.5a.old/configure.in	1998-12-31 08:46:54.000000000 -0800
+++ picptk-0.5a/configure.in	2003-10-20 02:42:14.000000000 -0700
@@ -7,12 +7,12 @@
 AC_PROG_INSTALL
 
 dnl Checks for libraries.
-AC_CHECK_LIB(itk3.0, Itk_Init)
-AC_CHECK_LIB(itk3.0, Itk_Init, cp picptk.itk picptk, cp picptk.tcl picptk)
+AC_CHECK_LIB(itk3.3, Itk_Init)
+AC_CHECK_LIB(itk3.3, Itk_Init, cp picptk.itk picptk, cp picptk.tcl picptk)
 
 dnl The following isn't totally true, but let's force the issue...
-AC_CHECK_LIB(itk3.0, Itk_Init,,AC_MSG_ERROR([
-**** picptk now requires itcl3.0
+AC_CHECK_LIB(itk3.3, Itk_Init,,AC_MSG_ERROR([
+**** picptk now requires itcl3.3
 **** See the README file for more info.
 ]))
 
