--- search.c.orig	2003-09-23 06:02:21.000000000 +0900
+++ search.c	2003-11-30 16:33:33.000000000 +0900
@@ -69,6 +69,12 @@
 	}
     }
     tmp = Str_conv_from_system(Strfgets(migemor));
+#ifdef USE_M17N
+    if (SearchConv && !WcOption.pre_conv &&
+	Currentbuf->document_charset != DisplayCharset)
+	tmp = Strnew_charp(wtf_conv_fit(
+	    tmp->ptr, Currentbuf->document_charset));
+#endif
     Strchop(tmp);
     if (tmp->length == 0)
 	goto err;
