diff -ruN NVIDIA_kernel-1.0-4496-2.5/nv-linux.h NVIDIA_kernel-1.0-4496-2.6-bk5/nv-linux.h
--- NVIDIA_kernel-1.0-4496-2.5/nv-linux.h	2003-07-29 00:02:15.000000000 +0200
+++ NVIDIA_kernel-1.0-4496-2.6-bk5/nv-linux.h	2003-08-17 21:24:19.000000000 +0200
@@ -361,7 +361,11 @@
 #  define NV_DEVICE_NUMBER(_minor)      ((kdev_val(_minor)) & 0x0f)
 #  define NV_IS_CONTROL_DEVICE(_minor)  (((kdev_val(_minor)) & 0xff) == 0xff)
 #  define NV_IS_SUSER()                 capable(CAP_SYS_ADMIN)
-#  define NV_PCI_DEVICE_NAME(x)         ((x)->dev.name)
+#  ifdef CONFIG_PCI_NAMES
+#   define NV_PCI_DEVICE_NAME(x)        ((x)->pretty_name)
+#  else
+#   define NV_PCI_DEVICE_NAME(x)        ""
+#  endif
 #  define NV_CLI()                      local_irq_disable()
 #  define NV_SAVE_FLAGS(x)              local_save_flags(x)
 #  define NV_RESTORE_FLAGS(x)           local_irq_restore(x)
