diff -Nur libcap-1.10-14/libcap/Makefile libcap-1.10-14.static/libcap/Makefile
--- libcap-1.10-14/libcap/Makefile	2004-09-16 02:45:13.000000000 +0200
+++ libcap-1.10-14.static/libcap/Makefile	2004-09-16 02:48:30.000000000 +0200
@@ -33,6 +33,7 @@
 # Library version
 #
 LIBNAME=libcap
+STATLIBNAME=$(LIBNAME).a
 #
 
 FILES=cap_alloc cap_proc cap_extint cap_flag cap_text cap_sys
@@ -46,7 +47,7 @@
 MAJLIBNAME=$(LIBNAME).so.$(VERSION)
 MINLIBNAME=$(MAJLIBNAME).$(MINOR)
 
-all: $(MINLIBNAME) $(LIBNAME).a
+all: $(MINLIBNAME) $(STATLIBNAME)
 
 _makenames: _makenames.c cap_names.sed
 	$(CC) $(CFLAGS) $(LDFLAGS) $< -o $@
@@ -58,8 +59,8 @@
 	@echo "=> making cap_names.c from <sys/capability.h>"
 	@sed -ne '/^#define[ \t]CAP[_A-Z]\+[ \t]\+[0-9]\+/{s/^#define \([^ \t]*\)[ \t]*\([^ \t]*\)/  \{ \2, \"\1\" \},/;y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/;p;}' < include/sys/capability.h | fgrep -v 0x > cap_names.sed #   @sed -ne '/^#define[ \t]CAP[_A-Z]\+[ \t]\+[0-9]\+/{s/^#define CAP_\([^ \t]*\)[ \t]*\([^ \t]*\)/  \{ \2, \"\1\" \},/;y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/;p;}' < /usr/include/linux/capability.h | fgrep -v 0x > cap_names.sed
 
-$(LIBNAME).a: $(OBJS)
-	ar cruv $(LIBNAME).a $(OBJS)
+$(STATLIBNAME): $(OBJS)
+	ar csruv $(STATLIBNAME) $(OBJS)
 
 $(MINLIBNAME): $(LOBJS)
 	$(CC) -shared -fPIC -Wl,-soname,$(MAJLIBNAME) -o $@ $(LOBJS)
@@ -77,7 +78,7 @@
 	mkdir -p -m 0755 $(INCDIR)/sys
 	install -m 0644 include/sys/capability.h $(INCDIR)/sys
 	mkdir -p -m 0755 $(LIBDIR)
-	install -m 0644 $(LIBNAME).a $(LIBDIR)
+	install -m 0644 $(STATLIBNAME) $(LIBDIR)
 	install -m 0644 $(MINLIBNAME) $(LIBDIR)/$(MINLIBNAME)
 	ln -sf $(MINLIBNAME) $(LIBDIR)/$(MAJLIBNAME)
 	ln -sf $(MAJLIBNAME) $(LIBDIR)/$(LIBNAME).so
@@ -85,7 +86,7 @@
 
 clean:
 	$(LOCALCLEAN)
-	rm -f $(OBJS) $(LOBJS) $(LIBNAME).a $(LIBNAME).so*
+	rm -f $(OBJS) $(LOBJS) $(STATLIBNAME) $(LIBNAME).so*
 	rm -f cap_names.h cap_names.sed _makenames
 	cd include/sys && $(LOCALCLEAN)
 
