diff -ru kerry-0.09/src/beaglesearch.cpp kerry-0.09-double/src/beaglesearch.cpp
--- kerry-0.09/src/beaglesearch.cpp	2006-02-14 05:07:44.000000000 -0600
+++ kerry-0.09-double/src/beaglesearch.cpp	2006-03-25 14:41:41.000000000 -0600
@@ -83,6 +83,11 @@
        { BeagleSearch::Website, "dc:title", 0 },
        { BeagleSearch::Website, "Title", 0 },
        { BeagleSearch::Note, "dc:title", 0 },
+// Gentoo ebuilds
+       { BeagleSearch::Ebuilds, "dc:title", I18N_NOOP("Title: %1")"<br>" },
+       { BeagleSearch::Ebuilds, "dc:description", I18N_NOOP("Description: %1")"<br>" },
+       { BeagleSearch::Ebuilds, "fixme:version", I18N_NOOP("Version: %1")"<br>" },
+       { BeagleSearch::Ebuilds, "dc:source", I18N_NOOP("Homepage: %1")"<br>" },
        { BeagleSearch::Unknown, "dc:author", I18N_NOOP("Author: %1")"<br>" },
        { BeagleSearch::Unknown, "dc:title", I18N_NOOP("Title: %1")"<br>" },
 // Calendar
@@ -148,6 +153,7 @@
        { BeagleSearch::Website, 0, "Google", 0 },
        { BeagleSearch::Website, 0, "WebHistory", 0 },
        { BeagleSearch::Website, 0, 0, "beagle/x-konq-cache" },
+       { BeagleSearch::Ebuilds, "*.ebuild", 0, 0 },
        { BeagleSearch::Unknown, "file://", "File", 0 },  // mimetype wildcard must be last
        { 0, 0, 0, 0}
 };
@@ -330,7 +336,7 @@
 #endif
 
 		for( uint i = 0; hitflavorinfos[i].tilegroup!=0; i++ ) {
-			if( (hitflavorinfos[i].uri == 0 || (*(result->uri)).startsWith(hitflavorinfos[i].uri)) &&
+			if( (hitflavorinfos[i].uri == 0 || (*(result->uri)).startsWith(hitflavorinfos[i].uri) || (hitflavorinfos[i].uri[0]=='*' && (*(result->uri)).endsWith(hitflavorinfos[i].uri+1))) &&
 			    (hitflavorinfos[i].type == 0 || hitflavorinfos[i].type==result->hit_type) &&
 			    (hitflavorinfos[i].mimetype == 0 || (*(result->mime_type)).startsWith(hitflavorinfos[i].mimetype))) {
 				result->tilegroup = (TileGroup)hitflavorinfos[i].tilegroup;
diff -ru kerry-0.09/src/beaglesearch.h kerry-0.09-double/src/beaglesearch.h
--- kerry-0.09/src/beaglesearch.h	2006-02-14 05:07:44.000000000 -0600
+++ kerry-0.09-double/src/beaglesearch.h	2006-03-25 14:41:41.000000000 -0600
@@ -51,7 +51,8 @@
 	Conversations,  // in combo box
 	Website,        // in combo box
 	Feed,
-	Note
+	Note,
+	Ebuilds
 };
 
 typedef QValueList<QString> PropertyList;
diff -ru kerry-0.09/src/searchdlg.cpp kerry-0.09-double/src/searchdlg.cpp
--- kerry-0.09/src/searchdlg.cpp	2006-03-13 04:40:58.000000000 -0600
+++ kerry-0.09-double/src/searchdlg.cpp	2006-03-25 14:54:21.000000000 -0600
@@ -669,6 +669,9 @@
           case Website:
 		show = ( result->tilegroup == BeagleSearch::Website);
 		break;
+          case Ebuilds:
+		show = ( result->tilegroup == BeagleSearch::Ebuilds);
+		break;
           default:
 		break;
         }
diff -ru kerry-0.09/src/searchdlg.h kerry-0.09-double/src/searchdlg.h
--- kerry-0.09/src/searchdlg.h	2006-01-10 10:13:29.000000000 -0600
+++ kerry-0.09-double/src/searchdlg.h	2006-03-25 14:53:41.000000000 -0600
@@ -42,7 +42,8 @@
 		Conversations,
 		Images,
 		Media,
-                Website
+                Website,
+		Ebuilds
 	};
 
 	enum SortType {
diff -ru kerry-0.09/src/searchdlg_layout.ui kerry-0.09-double/src/searchdlg_layout.ui
--- kerry-0.09/src/searchdlg_layout.ui	2006-03-25 14:37:22.000000000 -0600
+++ kerry-0.09-double/src/searchdlg_layout.ui	2006-03-25 14:54:05.000000000 -0600
@@ -250,6 +250,11 @@
                             <string>Web Pages</string>
                         </property>
 		    </item>
+                    <item>
+                        <property name="text">
+                            <string>Ebuilds</string>
+                        </property>
+		    </item>
                     <property name="name">
                         <cstring>comboShow</cstring>
                     </property>
