--- linux-2.4.26-sparc-r2.orig/mm/memory.c	2004-06-26 03:21:42.000000000 +0200
+++ linux-2.4.26-sparc-r2/mm/memory.c	2004-06-26 03:30:21.000000000 +0200
@@ -1595,10 +1595,15 @@
 		pte_t *new;
 
 		spin_unlock(&mm->page_table_lock);
-		new = pte_alloc_one(mm, address);
+		new = pte_alloc_one_fast(mm, address);
 		spin_lock(&mm->page_table_lock);
-		if (!new)
-			return NULL;
+		if (!new) {
+			spin_unlock(&mm->page_table_lock);
+			new = pte_alloc_one(mm, address);
+			spin_lock(&mm->page_table_lock);
+			if (!new)
+				return NULL;
+		}
 
 		/*
 		 * Because we dropped the lock, we should re-check the
