*** art/cursor/cursorthemegen/main.c.orig	Thu Feb 27 19:37:28 2003
--- art/cursor/cursorthemegen/main.c	Thu Feb 27 19:38:53 2003
***************
*** 236,247 ****
  {
    CursorImage *image = cursor_source_find_image (source, frame_index);
    int x, y, width, height;
! 
    cursor_frame_find_bounds (cursor, frame, source, frame_index,
  			    &x, &y, &width, &height);
  
!   GdkPixbuf *tmp_pixbuf = gdk_pixbuf_new_subpixbuf (image->image,
! 						    x, y, width, height);
  
    frame->hot_x -= x - cursor->column * source->gridsize;
    frame->hot_y -= y - cursor->row * source->gridsize;
--- 236,249 ----
  {
    CursorImage *image = cursor_source_find_image (source, frame_index);
    int x, y, width, height;
!   
!   GdkPixbuf *tmp_pixbuf;
!   
    cursor_frame_find_bounds (cursor, frame, source, frame_index,
  			    &x, &y, &width, &height);
  
!   tmp_pixbuf = gdk_pixbuf_new_subpixbuf (image->image,
! 				         x, y, width, height);
  
    frame->hot_x -= x - cursor->column * source->gridsize;
    frame->hot_y -= y - cursor->row * source->gridsize;
