--- xmame-0.71.1-orig/src/cpu/mips/mips3drc.c.orig	2003-08-02 20:49:23.023587861 -0500
+++ xmame-0.71.1/src/cpu/mips/mips3drc.c		2003-08-02 20:51:44.377905935 -0500
@@ -162,11 +162,11 @@
 **#################################################################################################*/
 
 /* PC and pointer pair */
-struct pc_ptr_pair
+/*struct pc_ptr_pair
 {
 	UINT32		pc;
 	UINT8 *		target;
-};
+};*/
 
 
 /* memory access function table */
@@ -358,7 +358,8 @@
 
 	/* fill in the config */
 	memset(&drconfig, 0, sizeof(drconfig));
-	drconfig.cachesize      = CACHE_SIZE;
+	drconfig.cache_size      = CACHE_SIZE;
+	drconfig.max_instructions = MAX_INSTRUCTIONS;
 	drconfig.address_bits   = 32;
 	drconfig.lsbs_to_ignore = 2;
 	drconfig.uses_fp        = 1;
