--- utils.c.orig	2006-07-18 00:13:19.000000000 -0400
+++ utils.c	2006-07-18 00:15:43.000000000 -0400
@@ -26,7 +26,7 @@
 /*
  * HSI color system utilities
  */
-static int trunc(double f)
+static int truncate(double f)
 {
 	int i;
 
@@ -44,9 +44,9 @@
 	Gv=1+S*sin(H);
 	Bv=1+S*sin(H+2*M_PI/3);
 	T=255.999*I/2;
-	*r=trunc(Rv*T);
-	*g=trunc(Gv*T);
-	*b=trunc(Bv*T);
+	*r=truncate(Rv*T);
+	*g=truncate(Gv*T);
+	*b=truncate(Bv*T);
 }
 
 /*
