--- silc-client-1.1.8/lib/Makefile.am
+++ silc-client-1.1.8/lib/Makefile.am
@@ -75,7 +75,7 @@
 
 install-exec-hook:
 if SILC_ENABLE_SHARED
-	-mkdir -p $(libdir)
+	-mkdir -p $(DESTDIR)$(libdir)
 	-$(LIBTOOL) --mode=install $(INSTALL) libsilc.la $(DESTDIR)$(libdir)/
 	-$(LIBTOOL) --mode=install $(INSTALL) libsilcclient.la $(DESTDIR)$(libdir)/
 	-rm -rf $(DESTDIR)$(libdir)/libsilc.a
--- silc-client-1.1.8/lib/Makefile.in
+++ silc-client-1.1.8/lib/Makefile.in
@@ -640,7 +640,7 @@
 	$(MAKE) remove $(SILCLIB) $(SILCSERVERLIB) $(SILCCLIENTLIB)
 
 install-exec-hook:
-@SILC_ENABLE_SHARED_TRUE@	-mkdir -p $(libdir)
+@SILC_ENABLE_SHARED_TRUE@	-mkdir -p $(DESTDIR)$(libdir)
 @SILC_ENABLE_SHARED_TRUE@	-$(LIBTOOL) --mode=install $(INSTALL) libsilc.la $(DESTDIR)$(libdir)/
 @SILC_ENABLE_SHARED_TRUE@	-$(LIBTOOL) --mode=install $(INSTALL) libsilcclient.la $(DESTDIR)$(libdir)/
 @SILC_ENABLE_SHARED_TRUE@	-rm -rf $(DESTDIR)$(libdir)/libsilc.a
