--- im-ja/im-ja/src/gtk/im-ja-gtk.c	2004/06/25 08:57:18	1.7
+++ im-ja/im-ja/src/gtk/im-ja-gtk.c	2004/12/09 13:07:09	1.8
@@ -37,8 +37,8 @@
 #include "../actionmenu.h"
 
 extern GType gtk_im_module_get_type(void);
-#define GTK_TYPE_IM_MODULE (gtk_im_module_get_type())
-#define GTK_IM_MODULE(im_module) (G_TYPE_CHECK_INSTANCE_CAST((im_module), GTK_TYPE_IM_MODULE, GtkIMModule))
+/*#define GTK_TYPE_IM_MODULE (gtk_im_module_get_type())*/
+/*#define GTK_IM_MODULE(im_module) (G_TYPE_CHECK_INSTANCE_CAST((im_module), GTK_TYPE_IM_MODULE, GtkIMModule))*/
 
 IMJAConfig cfg; 
 gint notify_id = 0;
@@ -429,7 +429,7 @@
 	GtkIMModule *im_module;
 	IM_JA_DEBUG("im_module_init()\n");
 
-	im_module = GTK_IM_MODULE(module);
+	im_module = module;
 	g_module_make_resident(im_module->library); /*FIXME: This is only a workaround for the GConf bug */
 
 	im_ja_gtk_register_type(module);
