Index: chat.c
===================================================================
RCS file: /cvsroot/gnocatan/gnocatan/client/gtk/chat.c,v
retrieving revision 1.2
diff -u -p -r1.2 chat.c
--- chat.c	1 Jan 2004 19:23:09 -0000	1.2
+++ chat.c	23 Feb 2004 23:04:29 -0000
@@ -19,15 +19,16 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#define GDK_DISABLE_DEPRECATED 1
+#define GTK_DISABLE_DEPRECATED 1
+#define GNOME_DISABLE_DEPRECATED 1
+ 
 #include "config.h"
-#include <string.h>
-
-#include "log.h"
 #include "frontend.h"
 
-#define MAX_CHAT 512		/* maximum chat message size */
+#define MAX_CHAT 512          /* maximum chat message size */
 
-static GtkWidget *chat_entry;	/* messages text widget */
+static GtkWidget *chat_entry; /* messages text widget */
 
 static void chat_cb(GtkEditable *editable, UNUSED(gpointer user_data))
 {
@@ -59,8 +60,9 @@ GtkWidget *chat_build_panel()
 	gtk_widget_show(frame);
 
 	chat_entry = gtk_entry_new();
-	gtk_signal_connect(GTK_OBJECT(chat_entry), "activate",
-			   GTK_SIGNAL_FUNC(chat_cb), NULL);
+	gtk_entry_set_max_length(GTK_ENTRY(chat_entry), MAX_CHAT);
+	g_signal_connect(G_OBJECT(chat_entry), "activate",
+			G_CALLBACK(chat_cb), NULL);
 	gtk_widget_show(chat_entry);
 	gtk_container_add(GTK_CONTAINER(frame), chat_entry);
 
@@ -69,5 +71,8 @@ GtkWidget *chat_build_panel()
 
 void chat_set_focus(void)
 {
-	gtk_widget_grab_focus(GTK_WIDGET(chat_entry));
+	if (!gtk_widget_is_focus(chat_entry)) {
+		gtk_widget_grab_focus(chat_entry);
+		gtk_editable_set_position(GTK_EDITABLE(chat_entry), -1);
+	}
 }
