Index: client/gtk/gui.c
===================================================================
RCS file: /cvsroot/gnocatan/gnocatan/client/gtk/gui.c,v
retrieving revision 1.4
diff -u -p -r1.4 gui.c
--- client/gtk/gui.c	2 Jan 2004 13:33:50 -0000	1.4
+++ client/gtk/gui.c	6 Feb 2004 18:25:41 -0000
@@ -296,7 +296,7 @@ void gui_highlight_chits(gint roll)
 static gint expose_map_cb(GtkWidget *area, GdkEventExpose *event,
 		UNUSED(gpointer user_data))
 {
-	if (area->window == NULL || map == NULL)
+	if (area->window == NULL || map == NULL || gmap == NULL)
 		return FALSE;
 
 	if (gmap->pixmap == NULL) {
