--- Image.cc	1999-08-15 09:43:55.000000000 -0500
+++ Image.cc.new	2002-12-12 12:18:35.000000000 -0500
@@ -225,12 +225,12 @@
     }
 
   if (control->doDither()) {
-    short er, eg, eb, *or, *og, *ob, *nor, *nog, *nob, *por, *pog, *pob;
+    short er, eg, eb, *gor, *og, *ob, *nor, *nog, *nob, *por, *pog, *pob;
     unsigned short *ort, *ogt, *obt;
     
-    control->getDitherBuffers(width + 2, &or, &og, &ob, &nor, &nog, &nob,
+    control->getDitherBuffers(width + 2, &gor, &og, &ob, &nor, &nog, &nob,
 			      &ort, &ogt, &obt);
-    if ((! or) || (! og) || (! ob) || (! nor) || (! nog) || (! nob) ||
+    if ((! gor)||(! og)||(! ob) || (! nor) || (! nog) || (! nob) ||
 	(! ort) || (! ogt) || (! obt)) {
       fprintf(stderr,
 	      "BImage::renderXImage: error getting dither information\n");
@@ -240,7 +240,7 @@
     
     x = width;
 
-    por = or;
+    por = gor;
     pog = og;
     pob = ob;
 
@@ -271,14 +271,14 @@
       }
       
       for (x = 0; x < width; x++) {
-	if (*(or + x) > 255) *(or + x) = 255;
-	else if (*(or + x) < 0) *(or + x) = 0;
+	if (*(gor + x) > 255) *(gor + x) = 255;
+	else if (*(gor + x) < 0) *(gor + x) = 0;
 	if (*(og + x) > 255) *(og + x) = 255;
 	else if (*(og + x) < 0) *(og + x) = 0;
 	if (*(ob + x) > 255) *(ob + x) = 255;
 	else if (*(ob + x) < 0) *(ob + x) = 0;
 	
-	r = *(tr + *(or + x));
+	r = *(tr + *(gor + x));
 	g = *(tg + *(og + x));
 	b = *(tb + *(ob + x));
 	
@@ -340,11 +340,11 @@
 	  return 0;
 	}
 
-	er = *(or + x) - *(ort + *(or + x));
+	er = *(gor + x) - *(ort + *(gor + x));
 	eg = *(og + x) - *(ogt + *(og + x));
 	eb = *(ob + x) - *(obt + *(ob + x));
 	
-	*(or + x + 1) += er;
+	*(gor + x + 1) += er;
 	*(og + x + 1) += eg;
 	*(ob + x + 1) += eb;
 	
@@ -360,7 +360,7 @@
       off += image->width;
       idata = (pd += image->bytes_per_line);
       
-      por = or; or = nor; nor = por;
+      por = gor; gor = nor; nor = por;
       pog = og; og = nog; nog = pog;
       pob = ob; ob = nob; nob = pob;
     }
