=== modified file 'PySilverCity/SilverCity/CSS.py'
--- PySilverCity/SilverCity/CSS.py	
+++ PySilverCity/SilverCity/CSS.py	
@@ -11,8 +11,9 @@
         self._properties = properties
         self._lexer = find_lexer_module_by_id(SCLEX_CSS)
         self._keyword_lists = [
-            WordList(Keywords.css_keywords),
-            WordList(Keywords.css_keywords_2),
+            WordList(Keywords.css_keywords), # keywords
+            WordList(Keywords.css_keywords_2), # pseudoClasses
+            WordList(), # keywords2
                                ]
             
 class CSSHandler(DispatchHandler):

=== modified file 'PySilverCity/SilverCity/Python.py'
--- PySilverCity/SilverCity/Python.py	
+++ PySilverCity/SilverCity/Python.py	
@@ -11,7 +11,8 @@
         self._properties = properties
         self._lexer = find_lexer_module_by_id(SCLEX_PYTHON)
         self._keyword_lists = [
-            WordList(Keywords.python_keywords)
+            WordList(Keywords.python_keywords),
+            WordList(),
                             ]
 
 class PythonHandler(DispatchHandler):

=== modified file 'PySilverCity/SilverCity/SQL.py'
--- PySilverCity/SilverCity/SQL.py	
+++ PySilverCity/SilverCity/SQL.py	
@@ -11,7 +11,14 @@
         self._properties = properties
         self._lexer = find_lexer_module_by_id(SCLEX_SQL)
         self._keyword_lists = [
-            WordList(Keywords.sql_keywords),
+            WordList(Keywords.sql_keywords), # keywords1
+            WordList(), # keywords2
+            WordList(), # kw_pldoc
+            WordList(), # kw_sqlplus
+            WordList(), # kw_user1
+            WordList(), # kw_user2
+            WordList(), # kw_user3
+            WordList(), # kw_user4
                                ]
             
 class SQLHandler(DispatchHandler):

