diff -Nru resin-3.1.1.vanilla/modules/c/src/apache2/Makefile.in resin-3.1.1/modules/c/src/apache2/Makefile.in
--- resin-3.1.1.vanilla/modules/c/src/apache2/Makefile.in	2006-12-26 02:44:02.000000000 +0100
+++ resin-3.1.1/modules/c/src/apache2/Makefile.in	2006-12-26 02:46:14.000000000 +0100
@@ -36,11 +36,8 @@
 	$(CC) -c $(INCLUDES) $(CFLAGS) $<
 
 install	: mod_caucho.la
-	$(LIBTOOL) --mode=install $(CP) mod_caucho.la $(apache_libexec)
-	- rm -f $(apache_libexec)/mod_caucho.la
-	- rm -f $(apache_libexec)/mod_caucho.a
-	sh install.sh -conf $(apache_conf) -libexec $(apache_libexec) \
-	   -resin_home $(resin_home)
+	mkdir -p $(DESTDIR)$(apache_libexec)
+	$(LIBTOOL) --mode=install $(CP) mod_caucho.la $(DESTDIR)$(apache_libexec)
 
 clean	:
 	- rm -r *.o *.lo *.la *.so .libs
