--- gtk-2.0/galaxy_theme_draw.c	2004-01-23 10:47:07.000000000 -0500
+++ gtk-2.0/galaxy_theme_draw.c.orig	2004-03-20 01:44:28.430956120 -0500
@@ -740,7 +740,7 @@
     }
 
     if (is_combo_button = (DETAIL("button") && 
-			   GTK_IS_COMBO(gtk_widget_get_parent(widget)))) {
+			   GTK_IS_COMBO_BOX(gtk_widget_get_parent(widget)))) {
       xthick = ythick = 1;
     }
 
@@ -1048,7 +1048,7 @@
     }
 
     if (shadow_type == GTK_SHADOW_IN && 
-	widget && (GTK_IS_SPIN_BUTTON (widget) || (GTK_IS_COMBO (gtk_widget_get_parent (widget)))) &&
+	widget && (GTK_IS_SPIN_BUTTON (widget) || (GTK_IS_COMBO_BOX (gtk_widget_get_parent (widget)))) &&
 	 DETAIL ("entry"))
     {
       GdkGC * gc1 = style->base_gc[GTK_WIDGET_STATE(widget)];
@@ -1323,7 +1323,7 @@
 	gdk_gc_set_clip_rectangle (galaxy_style->gray_gc[2], area);
       }
 
-      if (("button") && GTK_IS_COMBO(gtk_widget_get_parent(widget))) {
+      if (("button") && GTK_IS_COMBO_BOX(gtk_widget_get_parent(widget))) {
 	xthick = ythick = 1;
       }
 
@@ -2675,7 +2675,7 @@
     y += 1;
   }
   
-  if (GTK_IS_COMBO(gtk_widget_get_parent(gtk_widget_get_parent(widget)))) {
+  if (GTK_IS_COMBO_BOX(gtk_widget_get_parent(gtk_widget_get_parent(widget)))) {
     /* same design as KDE arrow, add a line */
     if (area) {
       gdk_gc_set_clip_rectangle (style->fg_gc[state], area);
