--- btcontent.cpp.orig	2007-01-19 00:12:24.000000000 +0100
+++ btcontent.cpp	2007-01-19 00:12:35.000000000 +0100
@@ -9,9 +9,9 @@
 #else
 #include <unistd.h>
 #include <sys/param.h>
-#if defined(HAVE_LIBCRYPT) || defined(HAVE_LIBMD) || defined(HAVE_LIBCRYPTO)
+#if defined(HAVE_LIBCRYPT) || defined(HAVE_LIBMD)
 #include <sha.h>
-#elif defined(HAVE_LIBSSL)
+#elif defined(HAVE_LIBSSL) || defined(HAVE_LIBCRYPTO)
 #include <openssl/sha.h>
 #endif
 #endif
--- btfiles.cpp.orig	2007-01-19 00:17:04.000000000 +0100
+++ btfiles.cpp	2007-01-19 00:17:18.000000000 +0100
@@ -8,9 +8,9 @@
 #include <unistd.h>
 #include <dirent.h>
 #include <sys/param.h>
-#if defined(HAVE_LIBCRYPT) || defined(HAVE_LIBMD) || defined(HAVE_LIBCRYPTO)
+#if defined(HAVE_LIBCRYPT) || defined(HAVE_LIBMD)
 #include <sha.h>
-#elif defined(HAVE_LIBSSL)
+#elif defined(HAVE_LIBSSL) || defined(HAVE_LIBCRYPTO)
 #include <openssl/sha.h>
 #endif
 #endif
