diff -urN xine-d5d-0.2.7.orig/src/input_dvd.c xine-d5d-0.2.7/src/input_dvd.c
--- xine-d5d-0.2.7.orig/src/input_dvd.c	Fri Sep 20 00:50:43 2002
+++ xine-d5d-0.2.7/src/input_dvd.c	Fri Sep 20 00:52:34 2002
@@ -1050,7 +1050,8 @@
   dvdnav_input_plugin_t *this;
   config_values_t *config = xine->config;
 
-  if (iface != 7) {
+/* Josh McKinney 20 Sept 2002 - added api ver 8 to support xine 0.9.13 */
+  if (iface != 7 && iface != 8) {
     printf("d5d input plugin doesn't support plugin API version %d.\n"
 	   "PLUGIN DISABLED.\n"
 	   "This means there's a version mismatch between xine and this input"

