--- interface.cpp.original	2009-08-08 20:46:34.000000000 -0600
+++ interface.cpp	2009-08-08 20:46:42.000000000 -0600
@@ -163,11 +163,11 @@
 	if (val != -1000000) return val;
 	if (0 == strcmp ("cur-fps"	, name)) return (cnt_fps>0) ? sum_fps / cnt_fps : -1;
 	return  -1000000;
 	}}}
 void Set_Resolution  (const char *res) {{{
-  	char *w;
+  	const char *w;
   	if (res == NULL) return;
   	w = strchr (res, 'x');
   	if (w == NULL) return;
 	v4l[v4l_akt]->Resolution(atoi (res), atoi (w + 1));
 //  	V4L_resolution_set (atoi (res), atoi (w + 1));
