From 29612b2d322ee5d04118e23a67fa2d2c9f0389a7 Mon Sep 17 00:00:00 2001
From: Sebastian Pölsterl <sebp@k-d-w.org>
Date: Thu, 14 Apr 2011 17:12:42 +0000
Subject: glib-2.0.c: Added annotation for g_variant_new_variant to mark it as constructor

https://bugzilla.gnome.org/show_bug.cgi?id=647796
---
diff --git a/gir/glib-2.0.c b/gir/glib-2.0.c
index 3782db3..f8f4234 100644
--- a/gir/glib-2.0.c
+++ b/gir/glib-2.0.c
@@ -211,6 +211,12 @@
  */
 
 /**
+ * g_variant_new_variant: (constructor)
+ * @value:
+ * @returns:
+ */
+
+/**
  * g_variant_get_strv:
  * @length: (allow-none):
  * @returns: (array length=length) (transfer container):
--
cgit v0.9
