diff -ruN cinelerra-1.1.7.orig/esound/esd.h cinelerra-1.1.7/esound/esd.h
--- cinelerra-1.1.7.orig/esound/esd.h	2003-09-26 13:18:40.000000000 +0200
+++ cinelerra-1.1.7/esound/esd.h	2003-09-26 13:32:03.000000000 +0200
@@ -8,7 +8,7 @@
 
 /* path and name of the default EsounD domain socket */
 #define ESD_UNIX_SOCKET_DIR	"/tmp/.esd"
-#define ESD_UNIX_SOCKET_NAME	ESD_UNIX_SOCKET_DIR ## "/" ## "socket"
+#define ESD_UNIX_SOCKET_NAME	ESD_UNIX_SOCKET_DIR  "/" "socket"
 
 /* length of the audio buffer size */
 #define ESD_BUF_SIZE (4 * 1024)
