diff -cwb etc.bak/changeme.in etc/changeme.in
*** etc.bak/changeme.in	Sun Mar 19 01:39:52 2006
--- etc/changeme.in	Sun Mar 19 01:41:31 2006
***************
*** 13,21 ****
  #
  (
     echo "Subject: $MSG"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL -s "$MSG" $SYSADMIN
  exit 0
--- 13,22 ----
  #
  (
     echo "Subject: $MSG"
+    echo "To: $SYSADMIN"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL $SYSADMIN
  exit 0
diff -cwb etc.bak/commfailure.in etc/commfailure.in
*** etc.bak/commfailure.in	Sun Mar 19 01:39:52 2006
--- etc/commfailure.in	Sun Mar 19 01:41:51 2006
***************
*** 13,21 ****
  #
  (
     echo "Subject: $MSG"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL -s "$MSG" $SYSADMIN
  exit 0
--- 13,22 ----
  #
  (
     echo "Subject: $MSG"
+    echo "To: $SYSADMIN"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL $SYSADMIN
  exit 0
diff -cwb etc.bak/commok.in etc/commok.in
*** etc.bak/commok.in	Sun Mar 19 01:39:52 2006
--- etc/commok.in	Sun Mar 19 01:42:05 2006
***************
*** 13,21 ****
  #
  (
     echo "Subject: $MSG"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL -s "$MSG" $SYSADMIN
  exit 0
--- 13,22 ----
  #
  (
     echo "Subject: $MSG"
+    echo "To: $SYSADMIN"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL $SYSADMIN
  exit 0
diff -cwb etc.bak/masterconnect.in etc/masterconnect.in
*** etc.bak/masterconnect.in	Sun Mar 19 01:39:52 2006
--- etc/masterconnect.in	Sun Mar 19 01:42:22 2006
***************
*** 13,21 ****
  #
  (
     echo "Subject: $MSG"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL -s "$MSG" $SYSADMIN
  exit 0
--- 13,22 ----
  #
  (
     echo "Subject: $MSG"
+    echo "To: $SYSADMIN"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL $SYSADMIN
  exit 0
diff -cwb etc.bak/mastertimeout.in etc/mastertimeout.in
*** etc.bak/mastertimeout.in	Sun Mar 19 01:39:52 2006
--- etc/mastertimeout.in	Sun Mar 19 01:42:35 2006
***************
*** 13,21 ****
  #
  (
     echo "Subject: $MSG"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL -s "$MSG" $SYSADMIN
  exit 0
--- 13,22 ----
  #
  (
     echo "Subject: $MSG"
+    echo "To: $SYSADMIN"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL $SYSADMIN
  exit 0
diff -cwb etc.bak/offbattery.in etc/offbattery.in
*** etc.bak/offbattery.in	Sun Mar 19 01:39:52 2006
--- etc/offbattery.in	Sun Mar 19 01:42:53 2006
***************
*** 13,21 ****
  #
  (
     echo "Subject: $MSG"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL -s "$MSG" $SYSADMIN
  exit 0
--- 13,22 ----
  #
  (
     echo "Subject: $MSG"
+    echo "To: $SYSADMIN"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL $SYSADMIN
  exit 0
diff -cwb etc.bak/onbattery.in etc/onbattery.in
*** etc.bak/onbattery.in	Sun Mar 19 01:39:52 2006
--- etc/onbattery.in	Sun Mar 19 01:43:10 2006
***************
*** 13,21 ****
  #
  (
     echo "Subject: $MSG"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL -s "$MSG" $SYSADMIN
  exit 0
--- 13,22 ----
  #
  (
     echo "Subject: $MSG"
+    echo "To: $SYSADMIN"
     echo " "
     echo "$MSG"
     echo " "
     @sbindir@/apcaccess status
! ) | $APCUPSD_MAIL $SYSADMIN
  exit 0
