diff -U2 -r ecore-0.0.2/src/Ecore.h ecore-0.0.2-lsd/src/Ecore.h
--- ecore-0.0.2/src/Ecore.h	Sun Apr  8 12:00:39 2001
+++ ecore-0.0.2-lsd/src/Ecore.h	Sat Apr 13 21:14:31 2002
@@ -21,4 +21,8 @@
 #endif
 
+#ifdef	__cplusplus
+extern "C" {
+#endif
+
 #define XEV_NONE              NoEventMask
 #define XEV_KEY               KeyPressMask | KeyReleaseMask
@@ -205,5 +209,5 @@
 void                e_window_set_name_class(Window win, char *name,
 
-					    char *class);
+					    char *e_class);
 void                e_window_set_min_size(Window win, int w, int h);
 void                e_window_set_max_size(Window win, int w, int h);
@@ -771,3 +775,7 @@
 double              e_get_time(void);
 
+#ifdef	__cplusplus
+}
+#endif
+
 #endif
diff -U2 -r ecore-0.0.2/src/e_x.c ecore-0.0.2-lsd/src/e_x.c
--- ecore-0.0.2/src/e_x.c	Sun Apr  8 12:00:26 2001
+++ ecore-0.0.2-lsd/src/e_x.c	Sat Apr 13 20:51:33 2002
@@ -2213,10 +2213,10 @@
 
 void
-e_window_set_name_class(Window win, char *name, char *class)
+e_window_set_name_class(Window win, char *name, char *e_class)
 {
    XClassHint          hint;
 
    hint.res_name = name;
-   hint.res_class = class;
+   hint.res_class = e_class;
    XSetClassHint(disp, win, &hint);
 }
