diff -rupN a/libhttpd.c b/libhttpd.c
--- a/libhttpd.c	2012-02-21 05:43:57.949988934 +0000
+++ b/libhttpd.c	2012-02-21 05:46:44.446655700 +0000
@@ -2549,7 +2549,7 @@ figure_mime( httpd_conn* hc )
     size_t ext_len, encodings_len;
     int i, top, bot, mid;
     int r;
-    char* default_type = "text/plain; charset=%s";
+    char* default_type = "application/octet-stream";
 
     /* Peel off encoding extensions until there aren't any more. */
     n_me_indexes = 0;
