Index: museeq/CMakeLists.txt
===================================================================
--- museeq/CMakeLists.txt	(revisión: 735)
+++ museeq/CMakeLists.txt	(copia de trabajo)
@@ -29,7 +29,7 @@
       SET(_moc ${CMAKE_CURRENT_BINARY_DIR}/${_basename}.moc.cpp)
 
       ADD_CUSTOM_COMMAND(OUTPUT ${_moc}
-         COMMAND moc #${QT_MOC_EXECUTABLE} 
+         COMMAND /usr/qt/3/bin/moc #${QT_MOC_EXECUTABLE} 
          ARGS ${_tmp_FILE} -o ${_moc}
          DEPENDS ${_tmp_FILE}
       )
@@ -90,6 +90,7 @@
     /usr/share/qt3/lib
     C:/Progra~1/qt/lib
     /usr/X11R6/lib
+    /opt/qt/lib
 )
 
 IF(PREFIX)
@@ -304,8 +305,12 @@
 
 
 add_executable(museeq ${MUSEEQ_SOURCES} ${MUSEEQ_MOC_HEADERS})
+set(LINK_LIBRARIES  Mucipher ${QT_LIBRARIES})
+if (QSA)
+  set(LINK_LIBRARIES ${LINK_LIBRARIES} ${QSA_LIBRARY})
+endif (QSA)
 
-target_link_libraries(museeq Mucipher ${QT_LIBRARIES} ${QSA_LIBRARY})
+target_link_libraries(museeq ${LINK_LIBRARIES})
 
 install(
   TARGETS museeq
@@ -331,13 +336,13 @@
 )
 
 file(GLOB_RECURSE THEME_DEFAULT default/*.png)
-install(FILES ${THEME_DEFAULT} DESTINATION ${DATADIR}/museek/museeq/default)
+install(FILES ${THEME_DEFAULT} DESTINATION ${DATADIR}/museek/museeq/icons/default)
 file(GLOB_RECURSE THEME_BLUEBOX bluebox/*.png)
-install(FILES ${THEME_BLUEBOX} DESTINATION ${DATADIR}/museek/museeq/bluebox)
+install(FILES ${THEME_BLUEBOX} DESTINATION ${DATADIR}/museek/museeq/icons/bluebox)
 file(GLOB_RECURSE THEME_SILK mikelabo-silk/*.png)
-install(FILES ${THEME_SILK} DESTINATION ${DATADIR}/museek/museeq/mikelabo-silk)
+install(FILES ${THEME_SILK} DESTINATION ${DATADIR}/museek/museeq/icons/mikelabo-silk)
 file(GLOB_RECURSE THEME_TRI mikelabo-tri/*.png)
-install(FILES ${THEME_TRI} DESTINATION ${DATADIR}/museek/museeq/mikelabo-tri)
+install(FILES ${THEME_TRI} DESTINATION ${DATADIR}/museek/museeq/icons/mikelabo-tri)
 
 set(ICONS
   museeq.png
@@ -369,4 +374,4 @@
   MESSAGE(" * without a Trayicon.. [TRAYICON=1]")
 else (NOT TRAYICON)
   MESSAGE(" * with a Trayicon.. [TRAYICON=0]")
-endif (NOT TRAYICON)
\ Sin fin-de-línea al final del archivo
+endif (NOT TRAYICON)
Index: museeq/mainwin.cpp
===================================================================
--- museeq/mainwin.cpp	(revisión: 735)
+++ museeq/mainwin.cpp	(copia de trabajo)
@@ -119,13 +119,8 @@
 	mMenuSettings->setItemEnabled(1, true);
 
 	mMenuSettings->setItemEnabled(3, false);
-	mMenuSettings->setItemChecked(3, museeq->mShowTickers);
-	mMenuSettings->setItemEnabled(4, false);
-	mMenuSettings->setItemChecked(4, museeq->mShowStatusLog);
-	mMenuSettings->setItemChecked(5, museeq->mShowTimestamps);
-	mMenuSettings->setItemEnabled(8, true);
-	mMenuSettings->setItemChecked(8, museeq->mUsetray);
 	
+	
 	menuBar()->insertItem(tr("&Settings"), mMenuSettings);
 	mMenuModes = new QPopupMenu(this);
 	mMenuModes->insertItem( IMG("chatroom-small"), tr("&Chat Rooms"), this, SLOT(changeCMode()), 0, 0);
@@ -263,12 +258,8 @@
 
 	connect(museeq->driver(), SIGNAL(userStatus(const QString&, uint)), SLOT(slotUserStatus(const QString&, uint)));
 	QSettings settings;
-	QString showStatusLog = settings.readEntry("/TheGraveyard.org/Museeq/showStatusLog");
-	if (! showStatusLog.isEmpty() and (showStatusLog == "true" || showStatusLog == true)) {
-		museeq->mShowStatusLog = true;
-	} else if (! showStatusLog.isEmpty() and (showStatusLog == "false" || showStatusLog == false)) {
-		museeq->mShowStatusLog = false;
-	}
+	museeq->mShowStatusLog = settings.readBoolEntry("/TheGraveyard.org/Museeq/showStatusLog", FALSE);
+
 	if ( ! museeq->mShowStatusLog)
 		mLog->hide();
 	QString exitdialog = settings.readEntry("/TheGraveyard.org/Museeq/ShowExitDialog");
@@ -326,9 +317,10 @@
 	museeq->mColorTrusted = settings.readEntry("/TheGraveyard.org/Museeq/colorTrusted");
 	museeq->mColorRemote = settings.readEntry("/TheGraveyard.org/Museeq/colorRemote");
 	museeq->mColorTime = settings.readEntry("/TheGraveyard.org/Museeq/colorTime");
-	QString showTimestamps = settings.readEntry("/TheGraveyard.org/Museeq/showTimestamps");
-	QString showIPinLog = settings.readEntry("/TheGraveyard.org/Museeq/showIPinLog");
-	QString showAlertsInLog = settings.readEntry("/TheGraveyard.org/Museeq/showAlertsInLog");
+	museeq->mShowTickers = settings.readBoolEntry("/TheGraveyard.org/Museeq/showTickers", FALSE);
+	museeq->mShowTimestamps = settings.readBoolEntry("/TheGraveyard.org/Museeq/showTimestamps", FALSE);
+	museeq->mIPLog = settings.readBoolEntry("/TheGraveyard.org/Museeq/showIPinLog", FALSE);
+	museeq->mOnlineAlert = settings.readBoolEntry("/TheGraveyard.org/Museeq/showAlertsInLog", FALSE);
 	
 	if (! museeq->mFontTime.isEmpty()) {
 		mSettingsDialog->STimeFont->setText(museeq->mFontTime);
@@ -357,28 +349,14 @@
 	if (! museeq->mColorTime.isEmpty()) {
 		mSettingsDialog->STimeText->setText(museeq->mColorTime);
 	}
-	if (! showTimestamps.isEmpty() and (showTimestamps == "true" || showTimestamps == true)) {
-		museeq->mShowTimestamps = true;
-		mMenuSettings->setItemChecked(5, true);
-	} else if (! showTimestamps.isEmpty() and (showTimestamps == "false" || showTimestamps == false)) {
-		museeq->mShowTimestamps = false;
-		mMenuSettings->setItemChecked(5, false);
-	}
-	if (! showIPinLog.isEmpty() and (showIPinLog == "true" || showIPinLog == true)) {
-		mSettingsDialog->SIPLog->setChecked(true);
-		museeq->mIPLog = true;
-		
-	} else if (! showIPinLog.isEmpty() and (showIPinLog == "false" || showIPinLog == false)){
-		mSettingsDialog->SIPLog->setChecked(false);
-		museeq->mIPLog = false;
-	}
-	if (! showAlertsInLog.isEmpty() and (showAlertsInLog == "true" || showAlertsInLog == true)) {
-		mSettingsDialog->SOnlineAlerts->setChecked(true);
-		museeq->mOnlineAlert = true;
-	} else if (! showAlertsInLog.isEmpty() and (showAlertsInLog == "false" || showAlertsInLog == false)){
-		mSettingsDialog->SOnlineAlerts->setChecked(false);
-		museeq->mOnlineAlert = false;
-	}
+
+	
+	mMenuSettings->setItemChecked(3, museeq->mShowTickers);
+	mMenuSettings->setItemChecked(4, museeq->mShowStatusLog);
+	mMenuSettings->setItemChecked(5, museeq->mShowTimestamps);
+	mMenuSettings->setItemChecked(8, museeq->mUsetray);
+	mSettingsDialog->SOnlineAlerts->setChecked(museeq->mOnlineAlert);
+	mSettingsDialog->SIPLog->setChecked(museeq->mIPLog);
 	box->setEnabled(false);
 	daemon = new QProcess(this);
 	connect( daemon, SIGNAL(readyReadStdout()), this,   SLOT(readFromStdout()) );
Index: museeq/system.h
===================================================================
--- museeq/system.h	(revisión: 741)
+++ museeq/system.h	(copia de trabajo)
@@ -66,7 +66,7 @@
 #endif
 
 
-#define HAVE_UINT
+/* #undef HAVE_UINT */
 #ifndef HAVE_UINT
  typedef unsigned int uint;
 #endif
Index: Tools/mp3.c
===================================================================
--- Tools/mp3.c	(revisión: 735)
+++ Tools/mp3.c	(copia de trabajo)
@@ -19,6 +19,7 @@
 
 #include <system.h>
 
+#include <stdlib.h>
 #include "mp3.h"
 
 #define ENDIAN(head) ((head >> 24) | ((head & 0x00ff0000) >> 8) | ((head & 0x0000ff00) << 8) | (head << 24))
