--- CMakeLists.txt	2007-09-23 13:10:33.000000000 +0200
+++ CMakeLists.txt	2007-09-23 13:14:32.000000000 +0200
@@ -261,7 +261,7 @@
 endif (NOT DIR_H)
 
 # Detect Vorbisfile library
-if (HAVE_VORBIS_H)
+if (HAVE_VORBIS_H AND VORBIS)
   find_library(VORBIS_LIBRARIES NAMES vorbisfile PATHS /usr/lib /usr/local/lib)
   if (VORBIS_LIBRARIES)
     set(HAVE_VORBIS 1)
@@ -270,9 +270,9 @@
     message(STATUS "Vorbisfile library wasn't found, ogg metadata will not be created")
     set(VORBIS_LIBRARIES "")
   endif (VORBIS_LIBRARIES)
-else (HAVE_VORBIS_H)
+else (HAVE_VORBIS_H AND VORBIS)
   set(VORBIS_LIBRARIES "")
-endif (HAVE_VORBIS_H)
+endif (HAVE_VORBIS_H AND VORBIS)
 
 set(CMAKE_REQUIRED_LIBRARIES)
 
@@ -340,4 +340,4 @@
   add_subdirectory(museeq)
 else (C_MUSEEQ)
   message("!!! museeq will NOT be installed [NO_MUSEEQ=0]")
-endif (C_MUSEEQ)
\ No hay ningún carácter de nueva línea al final del fichero
+endif (C_MUSEEQ)
--- museeq/CMakeLists.txt	2007-09-23 13:10:33.000000000 +0200
+++ museeq/CMakeLists.txt	2007-09-23 13:13:58.000000000 +0200
@@ -1,9 +1,7 @@
 project(museeq) # the name of your project
 
 MESSAGE("DECLARATIONS: PREFIX (/usr), MANDIR (PREFIX/man), DATADIR (PREFIX/share), BINRELOC, TRAYICON
-Prefix declarations with -D and set them with either 1, 0 or a path.
-QSA is detected automatically
-")
+Prefix declarations with -D and set them with either 1, 0 or a path.")
 
 
 cmake_minimum_required(VERSION 2.4.0)
@@ -175,13 +173,17 @@
   int main() {
      QSInterpreter::defaultInterpreter()->hasFunction(\"\");
   }
-" QSA)
+" HAVE_QSA)
 
 if (QSA)
-MESSAGE(STATUS "QSA available")
- add_definitions(-DHAVE_QSA=1 -DHAVE_QSA_DIALOG=1 )
+  if (HAVE_QSA)
+    MESSAGE(STATUS "QSA available")
+    add_definitions(-DHAVE_QSA=1 -DHAVE_QSA_DIALOG=1 )
+  else (HAVE_QSA)
+    MESSAGE(FATAL_ERROR "!!! QSA NOT available")
+  endif (HAVE_QSA)
 else (QSA)
-MESSAGE("!!! QSA NOT available")
+MESSAGE("!!! QSA disabled")
 endif (QSA)
 
 message(STATUS "moc binary:" ${QT_MOC_EXECUTABLE})
@@ -369,4 +371,4 @@
   MESSAGE(" * without a Trayicon.. [TRAYICON=1]")
 else (NOT TRAYICON)
   MESSAGE(" * with a Trayicon.. [TRAYICON=0]")
-endif (NOT TRAYICON)
\ No hay ningún carácter de nueva línea al final del fichero
+endif (NOT TRAYICON)
--- Tools/CMakeLists.txt	2007-09-23 13:10:33.000000000 +0200
+++ Tools/CMakeLists.txt	2007-09-23 13:11:55.000000000 +0200
@@ -42,7 +42,7 @@
 )
 MESSAGE("--> muscan (file scanner) will be installed..")
 # To build, or not build muscand
-if (FAM_LIBRARIES AND FAM_H)
+if (NOT NO_MUSCAND AND FAM_LIBRARIES AND FAM_H)
 
 add_executable(muscand ${MUSCAND_SOURCES})
 
@@ -68,9 +68,9 @@
   DESTINATION bin
 )
   MESSAGE("--> muscand (FAM-based file scanning daemon) will be installed..")
-else(FAM_LIBRARIES AND FAM_H)
+else(NOT NO_MUSCAND AND FAM_LIBRARIES AND FAM_H)
   MESSAGE("!!! muscand will NOT be installed..")
-endif(FAM_LIBRARIES AND FAM_H)
+endif(NOT NO_MUSCAND AND FAM_LIBRARIES AND FAM_H)
 
 target_link_libraries(
   muscan
