diff -ru icewm-1.0.9/src/wmclient.h icewm-1.0.9-edited/src/wmclient.h
--- icewm-1.0.9/src/wmclient.h  Tue Oct  9 16:54:03 2001
+++ icewm-1.0.9-edited/src/wmclient.h   Thu May 23 20:44:53 2002
@@ -11,10 +11,7 @@
 typedef int FrameState;

 #ifndef __YIMP_UTIL__
-//!!! remove these if possible
-typedef struct XWMHints;
-typedef struct XSizeHints;
-typedef struct XClassHint;
+#   include <X11/Xutil.h>
 #endif

 class ClientData {
