diff -Nur kdepim-3.4.2.orig/configure.in.in kdepim-3.4.2/configure.in.in
--- kdepim-3.4.2.orig/configure.in.in	2005-07-20 12:03:13.000000000 +0200
+++ kdepim-3.4.2/configure.in.in	2005-07-26 14:12:19.000000000 +0200
@@ -46,7 +46,6 @@
 [
   KDE_CHECK_HEADERS(gnokii.h)
   AC_REQUIRE([AC_HAVE_XPM])
-  LIB_GNOKII=
   AC_CHECK_LIB(gnokii, gn_cfg_read,
        [ LIB_GNOKII="-lgnokii $LIB_BLUETOOTH" ],
        [
@@ -58,6 +57,21 @@
   AC_SUBST(LIB_GNOKII)
 ])
 
+AC_ARG_WITH(gnokii,
+  [AC_HELP_STRING(--with-gnokii,
+    [enable support for mobile phones through gnokii @<:@default=check@:>@])],
+  [], with_gnokii=check)
+
+LIB_GNOKII=
+if test "x$with_gnokii" != xno; then
+  # for kaddressbook/xxport/ and kmobile/devices/gnokii/
+  KDE_CHECK_BLUETOOTH
+  KDE_CHECK_GNOKII
+
+  if test "x$with_gnokii" != xcheck && test -z "$LIB_GNOKII"; then
+    AC_MSG_ERROR([--with-gnokii was given, but test for gnokii failed])
+  fi
+fi
 
 dnl the below is copied from kdenetwork's configure as too many apps moved
 dnl and noone can say which tests are needed and which are obsolete for kdepim ;(
diff -Nur kdepim-3.4.2.orig/kaddressbook/xxport/configure.in.bot kdepim-3.4.2/kaddressbook/xxport/configure.in.bot
--- kdepim-3.4.2.orig/kaddressbook/xxport/configure.in.bot	2005-05-23 14:12:19.000000000 +0200
+++ kdepim-3.4.2/kaddressbook/xxport/configure.in.bot	2005-07-26 14:11:23.000000000 +0200
@@ -1,4 +1,4 @@
-if test -z "$LIB_GNOKII"; then
+if test "x$with_gnokii" = xcheck && test -z "$LIB_GNOKII"; then
   echo ""
   echo "libgnokii (http://www.gnokii.org) is missing. The KDE Addressbook mobile phone import/export filter will not be available."
   echo ""
diff -Nur kdepim-3.4.2.orig/kaddressbook/xxport/configure.in.in kdepim-3.4.2/kaddressbook/xxport/configure.in.in
--- kdepim-3.4.2.orig/kaddressbook/xxport/configure.in.in	2005-05-23 14:12:19.000000000 +0200
+++ kdepim-3.4.2/kaddressbook/xxport/configure.in.in	2005-07-26 14:11:23.000000000 +0200
@@ -1,6 +1,2 @@
 # $Id: kdepim-3.4-configure.patch,v 1.1 2005/07/28 12:58:14 greg_g Exp $
-
-KDE_CHECK_BLUETOOTH
-KDE_CHECK_GNOKII
-
 AM_CONDITIONAL(compile_GNOKIIXXPORT, test -n "$LIB_GNOKII")
diff -Nur kdepim-3.4.2.orig/kioslaves/configure.in.bot kdepim-3.4.2/kioslaves/configure.in.bot
--- kdepim-3.4.2.orig/kioslaves/configure.in.bot	2005-05-23 14:12:02.000000000 +0200
+++ kdepim-3.4.2/kioslaves/configure.in.bot	2005-07-26 14:11:34.000000000 +0200
@@ -1,4 +1,4 @@
-if test -z "$SASL2_LIBS"; then
+if test "x$with_sasl" = xcheck && test -z "$SASL2_LIBS"; then
   echo ""
   echo "cyrus-sasl 2 library is missing. The sieve ioslave will not be built, and imap4 will lack of a lot of authentication methods."
   echo ""
diff -Nur kdepim-3.4.2.orig/kioslaves/configure.in.in kdepim-3.4.2/kioslaves/configure.in.in
--- kdepim-3.4.2.orig/kioslaves/configure.in.in	2005-05-23 14:12:02.000000000 +0200
+++ kdepim-3.4.2/kioslaves/configure.in.in	2005-07-26 14:12:33.000000000 +0200
@@ -1,16 +1,27 @@
 KDE_CHECK_SSL
 
+AC_ARG_WITH(sasl,
+  [AC_HELP_STRING(--with-sasl,
+    [enable support for authentication through cyrus-sasl @<:@default=check@:>@])],
+  [], with_sasl=check)
+
 sasl2_header="no"
 SASL2_LIBS=""
+if test "x$with_sasl" != xno; then
+  KDE_CHECK_HEADERS(sasl/sasl.h, sasl2_header="yes")
+  if test "$sasl2_header" = "yes" ; then
+    KDE_CHECK_LIB(sasl2, sasl_client_init, SASL2_LIBS="-lsasl2")
+  fi
 
-KDE_CHECK_HEADERS(sasl/sasl.h, sasl2_header="yes")
-if test "$sasl2_header" = "yes" ; then
-  KDE_CHECK_LIB(sasl2, sasl_client_init, SASL2_LIBS="-lsasl2")
-fi
+  if test "x$SASL2_LIBS" != "x" ; then
+    AC_DEFINE_UNQUOTED(HAVE_LIBSASL2, 1, [Define if you have cyrus-sasl2 libraries])
+  fi
 
-if test "x$SASL2_LIBS" != "x" ; then
-  AC_DEFINE_UNQUOTED(HAVE_LIBSASL2, 1, [Define if you have cyrus-sasl2 libraries])
+  if test "x$with_sasl" != xcheck && test -z "$SASL2_LIBS"; then
+    AC_MSG_ERROR([--with-sasl was given, but test for cyrus-sasl failed])
+  fi
 fi
+
 AC_SUBST(SASL2_LIBS)
 
 AM_CONDITIONAL(compile_kio_sieve, test -n "$SASL2_LIBS")
diff -Nur kdepim-3.4.2.orig/kmobile/devices/gnokii/configure.in.in kdepim-3.4.2/kmobile/devices/gnokii/configure.in.in
--- kdepim-3.4.2.orig/kmobile/devices/gnokii/configure.in.in	2005-05-23 14:11:26.000000000 +0200
+++ kdepim-3.4.2/kmobile/devices/gnokii/configure.in.in	2005-07-26 14:11:23.000000000 +0200
@@ -1,7 +1,2 @@
 # $Id: kdepim-3.4-configure.patch,v 1.1 2005/07/28 12:58:14 greg_g Exp $
-
-# check for libgnokii (Nokia mobile phone library)
-KDE_CHECK_GNOKII
-
-AM_CONDITIONAL(include_gnokii_device, [test -n "$LIB_GNOKII"])
-
+AM_CONDITIONAL(include_gnokii_device, test -n "$LIB_GNOKII")
