diff -u -3 -p -r1.658.2.8 kmheaders.cpp
--- kmail/kmheaders.cpp	19 Nov 2004 11:46:04 -0000	1.658.2.8
+++ kmail/kmheaders.cpp	11 Dec 2004 18:47:49 -0000
@@ -1251,7 +1251,6 @@ void KMHeaders::msgAdded(int id)
   connect( this, SIGNAL(currentChanged(QListViewItem*)),
            this, SLOT(highlightMessage(QListViewItem*)));
 
-  emit messageListUpdated();
   END_TIMER(msgAdded);
   SHOW_TIMER(msgAdded);
 }
@@ -2188,7 +2187,7 @@ void KMHeaders::highlightMessage(QListVi
   if (mReaderWindowActive)
   {
     KMMessage *msg = mFolder->getMsg(idx);
-    if (!msg || msg->transferInProgress())
+    if (!msg )
     {
       emit selected( 0 );
       mPrevCurrent = 0;
