--- nagstamon.py.old	2011-02-12 03:33:35.154930846 +0100
+++ nagstamon.py	2011-02-12 03:34:08.100522965 +0100
@@ -32,13 +32,8 @@ conf = Config()
 # check for old settings when upgrading from a nagstamon version < 0.8 and convert them
 conf.Convert_Conf_to_Multiple_Servers()
 
-# try to get resources path if nagstamon got be installed by setup.py
-try:
-    import pkg_resources
-    Resources = pkg_resources.resource_filename("Nagstamon", "resources")
-except Exception, err:
-    # set resources to "" in case there are no resources available from setup.py
-    Resources = ""
+# set resources to "" in case there are no resources available from setup.py
+Resources = "/usr/share/nagstamon/resources"
 
 # Resources on MacOS X, environment variable _MEIPASS2 comes from pyinstaller packed executable
 # - does not really work at the moment
