--- nagstamon.py.orig	2009-05-15 18:07:31.873689350 +0200
+++ nagstamon.py	2009-05-15 18:08:32.130689362 +0200
@@ -497,9 +497,9 @@
         self.website = "http://nagstamon.sourceforge.net/"
         self.copyright = "©2009 Henri Wahl\nh.wahl@ifw-dresden.de"
         self.comments = "Nagios status monitor for your desktop"
-        
-        # get resources directory from current directory
-        self.resources = os.path.normcase(os.getcwd() + "/resources")
+
+        # Set path for resources/ directory
+        self.resources = "/usr/share/nagstamon/resources"
             
         # get and store local username for use with acknowledge + downtime actions
         if platform.system() == "Windows":
