--- mozilla-firefox-1.0.3.orig/browser/app/profile/firefox.js
+++ mozilla-firefox-1.0.3/browser/app/profile/firefox.js
@@ -38,6 +38,9 @@
 // SYNTAX HINTS:  dashes are delimiters.  Use underscores instead.
 //  The first character after a period must be alphabetic.
 
+// Disable default browser checking.
+pref("browser.shell.checkDefaultBrowser", false);
+
 // pref("startup.homepage_override_url","chrome://browser-region/locale/region.properties");
 pref("general.startup.browser", true);
 
@@ -62,8 +65,8 @@
 );
 
 // App-specific update preferences
-pref("app.update.enabled", true);               // Whether or not app updates are enabled
-pref("app.update.autoUpdateEnabled", true);     // Whether or not background app updates 
+pref("app.update.enabled", false);               // Whether or not app updates are enabled
+pref("app.update.autoUpdateEnabled", false);     // Whether or not background app updates 
                                                 // are enabled
 pref("app.update.url", "chrome://mozapps/locale/update/update.properties");
 pref("app.update.updatesAvailable", false);
@@ -81,7 +84,7 @@
 //  .. etc ..
 //
 pref("extensions.update.enabled", true);
-pref("extensions.update.autoUpdateEnabled", true);
+pref("extensions.update.autoUpdateEnabled", false);
 pref("extensions.update.url", "chrome://mozapps/locale/extensions/extensions.properties");
 pref("extensions.update.autoUpdate", false);    // Automatically download and install 
                                                 // updates to themes and extensions. 
@@ -237,6 +240,9 @@
 pref("network.cookie.cookieBehavior",       0); // cookies enabled
 pref("network.cookie.enableForCurrentSessionOnly", false);
 
+// Use LANG environment variable to choose locale
+pref("intl.locale.matchOS", true);
+
 // l12n and i18n
 pref("intl.accept_languages", "chrome://global/locale/intl.properties");
 // collationOption is only set on linux for japanese. see bug 18338 and 62015
@@ -327,3 +333,17 @@
 
 // plugin finder service
 pref("pfs.datasource.url", "chrome://mozapps/locale/plugins/plugins.properties");
+
+// enable negotiate extension
+pref("network.negotiate-auth.trusted-uris", "https://");
+
+// don't raise windows by default
+pref("mozilla.widget.raise-on-setfocus", false);
+
+// disable insecure ciphers
+pref("security.enable_ssl2", false);
+
+pref("security.ssl2.rc2_40", false);
+pref("security.ssl2.rc4_40", false);
+pref("security.ssl3.rsa_rc4_40_md5", false);
+pref("security.ssl3.rsa_rc2_40_md5", false);
--- mozilla-firefox-1.0.3.orig/browser/components/prefwindow/content/pref-advanced.xul
+++ mozilla-firefox-1.0.3/browser/components/prefwindow/content/pref-advanced.xul
@@ -72,9 +72,9 @@
               open="true" persist="open" clearhidden="true" align="left">
       <label>&softwareupdateinfo.label;</label>
       <vbox class="indent" align="left">
-        <checkbox id="enableSmartUpdate"
+        <!-- checkbox id="enableSmartUpdate"
                   label="&enableSmartUpdate.label;" accesskey="&enableSmartUpdate.accesskey;"
-                  prefstring="app.update.autoUpdateEnabled"/>
+                  prefstring="app.update.autoUpdateEnabled"/ -->
         <checkbox id="enableExtensionUpdate"
                   label="&enableExtensionUpdate.label;" accesskey="&enableExtensionUpdate.accesskey;"
                   prefstring="extensions.update.autoUpdateEnabled"/>      
