--- gtk+-1.2.7/gtk/gtkctree.c~	Wed Feb 16 06:18:07 2000
+++ gtk+-1.2.7/gtk/gtkctree.c	Fri May 19 23:05:15 2000
@@ -609,8 +609,8 @@
   ctree->tree_indent    = 20;
   ctree->tree_spacing   = 5;
   ctree->tree_column    = 0;
-  ctree->line_style     = GTK_CTREE_LINES_SOLID;
-  ctree->expander_style = GTK_CTREE_EXPANDER_SQUARE;
+  ctree->line_style     = GTK_CTREE_LINES_NONE;
+  ctree->expander_style = GTK_CTREE_EXPANDER_TRIANGLE;
   ctree->drag_compare   = NULL;
   ctree->show_stub      = TRUE;
 
