--- tcl/unix/configure.orig	2006-07-18 21:53:35.000000000 -0700
+++ tcl/unix/configure	2006-07-18 21:54:04.000000000 -0700
@@ -4966,7 +4966,7 @@
 	    # results, and the version is kept in special file).
 	
 	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-		system=MP-RAS-`awk '{print }' /etc/.relid'`
+		system=MP-RAS-`awk '{print }' /etc/.relid`
 	    fi
 	    if test "`uname -s`" = "AIX" ; then
 		system=AIX-`uname -v`.`uname -r`
--- tcl/unix/configure.orig	2006-07-18 21:58:59.000000000 -0700
+++ tcl/unix/configure	2006-07-18 21:59:47.000000000 -0700
@@ -6004,7 +6004,7 @@
 	    # results, and the version is kept in special file).
 	
 	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-		system=MP-RAS-`awk '{print }' /etc/.relid'`
+		system=MP-RAS-`awk '{print }' /etc/.relid`
 	    fi
 	    if test "`uname -s`" = "AIX" ; then
 		system=AIX-`uname -v`.`uname -r`
--- tk/unix/configure.orig	2006-07-18 22:32:21.000000000 -0700
+++ tk/unix/configure	2006-07-18 22:33:17.000000000 -0700
@@ -535,7 +535,7 @@
 
 ac_exeext=
 ac_objext=o
-if (echo "testing\c"; echo 1,2,3) | grep c >/dev/null; then
+if (echo "testing\\c"; echo 1,2,3) | grep c >/dev/null; then
   # Stardent Vistra SVR4 grep lacks -e, says ghazi@caip.rutgers.edu.
   if (echo -n testing; echo 1,2,3) | sed s/-n/xn/ | grep xn >/dev/null; then
     ac_n= ac_c='
@@ -1470,7 +1470,7 @@
 	    # results, and the version is kept in special file).
 	
 	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-		system=MP-RAS-`awk '{print }' /etc/.relid'`
+		system=MP-RAS-`awk '{print }' /etc/.relid`
 	    fi
 	    if test "`uname -s`" = "AIX" ; then
 		system=AIX-`uname -v`.`uname -r`
