diff -ru /tmp/sourcenav-5.1.1/Makefile.in ./Makefile.in
--- a/sourcenav-5.1.1/Makefile.in	2002-06-25 19:05:44.000000000 +0100
+++ b/sourcenav-5.1.1/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -24,6 +24,8 @@
 NOTPARALLEL = .NOTPARALLEL
 $(NOTPARALLEL):
 
+DESTDIR =
+
 srcdir = .
 
 prefix = /usr/local
@@ -418,7 +420,8 @@
 	"gxx_include_dir=$(gxx_include_dir)" \
 	"gcc_version=$(gcc_version)" \
 	"gcc_version_trigger=$(gcc_version_trigger)" \
-	"target_alias=$(target_alias)" 
+	"target_alias=$(target_alias)" \
+	"DESTDIR=$(DESTDIR)" 
 
 # For any flags above that may contain shell code that varies from one
 # target library to another.  When doing recursive invocations of the
diff -ru /tmp/sourcenav-5.1.1/itcl/itcl/Makefile.in ./itcl/itcl/Makefile.in
--- a/sourcenav-5.1.1/itcl/itcl/Makefile.in	2001-09-09 20:49:04.000000000 +0100
+++ b/sourcenav-5.1.1/itcl/itcl/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -272,16 +272,17 @@
 #========================================================================
 
 install-libraries: libraries
-	$(mkinstalldirs) $(includedir)
-	@echo "Installing header files in $(includedir)"
+	$(mkinstalldirs) $(DESTDIR)$(includedir)
+	@echo "Installing header files in $(DESTDIR)$(includedir)"
 	@for i in $(GENERIC_HDRS) ; do \
 	    echo "Installing $$i" ; \
-	    $(INSTALL_DATA) $$i $(includedir) ; \
+	    $(INSTALL_DATA) $$i $(DESTDIR)$(includedir) ; \
 	done;
-	@echo "Installing library files in $(ITCL_LIBRARY)"
+	$(mkinstalldirs) $(DESTDIR)$(ITCL_LIBRARY)
+	@echo "Installing library files in $(DESTDIR)$(ITCL_LIBRARY)"
 	@for i in $(srcdir)/library/*.tcl ; do \
 	    echo "Installing $$i" ; \
-	    $(INSTALL_DATA) $$i $(ITCL_LIBRARY) ; \
+	    $(INSTALL_DATA) $$i $(DESTDIR)$(ITCL_LIBRARY) ; \
 	done;
 
 #========================================================================
diff -ru /tmp/sourcenav-5.1.1/itcl/itk/Makefile.in ./itcl/itk/Makefile.in
--- a/sourcenav-5.1.1/itcl/itk/Makefile.in	2001-09-09 20:49:05.000000000 +0100
+++ b/sourcenav-5.1.1/itcl/itk/Makefile.in	2002-12-11 20:56:07.000000000 +0000
@@ -267,16 +267,17 @@
 #========================================================================
 
 install-libraries: libraries
-	$(mkinstalldirs) $(includedir)
-	@echo "Installing header files in $(includedir)"
+	$(mkinstalldirs) $(DESTDIR)$(includedir)
+	@echo "Installing header files in $(DESTDIR)$(includedir)"
 	@for i in $(GENERIC_HDRS) ; do \
 	    echo "Installing $$i" ; \
-	    $(INSTALL_DATA) $$i $(includedir) ; \
+	    $(INSTALL_DATA) $$i $(DESTDIR)$(includedir) ; \
 	done;
-	@echo "Installing library files in $(ITK_LIBRARY)"
+	$(mkinstalldirs) $(DESTDIR)$(ITK_LIBRARY)
+	@echo "Installing library files in $(DESTDIR)$(ITK_LIBRARY)"
 	@for i in $(srcdir)/library/*.* $(srcdir)/library/tclIndex ; do \
 	    echo "Installing $$i" ; \
-	    $(INSTALL_DATA) $$i $(ITK_LIBRARY) ; \
+	    $(INSTALL_DATA) $$i $(DESTDIR)$(ITK_LIBRARY) ; \
 	done;
 
 #========================================================================
diff -ru /tmp/sourcenav-5.1.1/itcl/iwidgets3.0.0/Makefile.in ./itcl/iwidgets3.0.0/Makefile.in
--- a/sourcenav-5.1.1/itcl/iwidgets3.0.0/Makefile.in	2002-01-11 20:18:03.000000000 +0000
+++ b/sourcenav-5.1.1/itcl/iwidgets3.0.0/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -1,3 +1,5 @@
+DESTDIR = 
+
 #
 # This file is a Makefile for [incr Widgets].  If it has the name
 # "Makefile.in" then it is a template for a Makefile; to generate
@@ -37,12 +39,12 @@
 # to be different than those used for actually reference files at
 # run-time.  INSTALL_ROOT is prepended to $prefix and $exec_prefix
 # when installing files.
-INSTALL_ROOT =
+INSTALL_ROOT = $(DESTDIR)
 
 # Path name to use when installing library scripts:
 
 # CYGNUS LOCAL
-SCRIPT_INSTALL_DIR = @datadir@/iwidgets$(VERSION)
+SCRIPT_INSTALL_DIR = $(DESTDIR)@datadir@/iwidgets$(VERSION)
 # END CYGNUS LOCAL
 
 # Directory in which to install the archive libtcl.a:
diff -ru /tmp/sourcenav-5.1.1/snavigator/Makefile.in ./snavigator/Makefile.in
--- a/sourcenav-5.1.1/snavigator/Makefile.in	2002-05-09 08:54:48.000000000 +0100
+++ b/sourcenav-5.1.1/snavigator/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -518,7 +518,7 @@
 
 
 install-exec-local:
-	$(INSTALL) $(top_srcdir)/../COPYING $(prefix)
+	$(INSTALL) $(top_srcdir)/../COPYING $(DESTDIR)$(prefix)
 	(cd ../tcl; $(MAKE) install-libraries)
 	(cd ../tk; $(MAKE) install-libraries)
 	(cd ../tix; $(MAKE) install-libraries)
diff -ru /tmp/sourcenav-5.1.1/snavigator/bitmaps/Makefile.in ./snavigator/bitmaps/Makefile.in
--- a/sourcenav-5.1.1/snavigator/bitmaps/Makefile.in	2002-04-26 08:14:27.000000000 +0100
+++ b/sourcenav-5.1.1/snavigator/bitmaps/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -333,8 +333,8 @@
 
 
 install-data-local:
-	$(INSTALL_DATA) $(srcdir)/@SPLASHFILE@ $(imagedir)/splash.gif
-	$(INSTALL_DATA) $(srcdir)/@X11ICON@ $(imagedir)/paf.xbm
+	$(INSTALL_DATA) $(srcdir)/@SPLASHFILE@ $(DESTDIR)$(imagedir)/splash.gif
+	$(INSTALL_DATA) $(srcdir)/@X11ICON@ $(DESTDIR)$(imagedir)/paf.xbm
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -ru /tmp/sourcenav-5.1.1/snavigator/db/Makefile.in ./snavigator/db/Makefile.in
--- a/sourcenav-5.1.1/snavigator/db/Makefile.in	2002-04-26 08:14:28.000000000 +0100
+++ b/sourcenav-5.1.1/snavigator/db/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -582,8 +582,8 @@
 
 
 install-data-local:
-	$(INSTALL_DATA) $(top_builddir)/../db/PORT/libdb.a $(sdkdir)/libpafdb.a
-	$(RANLIB) $(sdkdir)/libpafdb.a
+	$(INSTALL_DATA) $(top_builddir)/../db/PORT/libdb.a $(DESTDIR)$(sdkdir)/libpafdb.a
+	$(RANLIB) $(DESTDIR)$(sdkdir)/libpafdb.a
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -ru /tmp/sourcenav-5.1.1/snavigator/demo/Makefile.in ./snavigator/demo/Makefile.in
--- a/sourcenav-5.1.1/snavigator/demo/Makefile.in	2002-04-26 08:14:28.000000000 +0100
+++ b/sourcenav-5.1.1/snavigator/demo/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -288,16 +288,16 @@
 
 
 install-data-local:
-	@$(mkinstalldirs) $(demosdir)
+	@$(mkinstalldirs) $(DESTDIR)$(demosdir)
 	echo CVS > /tmp/sn-install-$$$$ ; \
 	for p in @C_DEMO@ @ASM_DEMO@ @FOR_DEMO@ @COB_DEMO@ @MONOP_DEMO@ @JAV_DEMO@ ; do \
-	  echo "copying $$p to $(demosdir)" ; \
+	  echo "copying $$p to $(DESTDIR)$(demosdir)" ; \
 	  ( \
 	  cd $(srcdir) ; \
 	  $(TAR) -ch$(OLDTARFLAG) -f /tmp/$$p.tar-$$$$ -X /tmp/sn-install-$$$$ $$p ; \
-	  cp /tmp/$$p.tar-$$$$ $(demosdir)/$$p.tar-$$$$ ; \
+	  cp /tmp/$$p.tar-$$$$ $(DESTDIR)$(demosdir)/$$p.tar-$$$$ ; \
 	  rm /tmp/$$p.tar-$$$$ ; \
-	  cd $(demosdir) ; \
+	  cd $(DESTDIR)$(demosdir) ; \
 	  $(TAR) -xf $$p.tar-$$$$ ; \
 	  rm $$p.tar-$$$$ ; \
           ) \
diff -ru /tmp/sourcenav-5.1.1/snavigator/doc/html/Makefile.in ./snavigator/doc/html/Makefile.in
--- a/sourcenav-5.1.1/snavigator/doc/html/Makefile.in	2002-04-26 08:14:28.000000000 +0100
+++ b/sourcenav-5.1.1/snavigator/doc/html/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -333,18 +333,18 @@
 
 
 install-data-local:
-	$(mkinstalldirs) $(htmldir)
+	$(mkinstalldirs) $(DESTDIR)$(htmldir)
 	for p in @SN_BOOKS@; do \
-		$(mkinstalldirs) $(htmldir)/$$p; \
-		rm -f $(htmldir)/$$p/*.html; \
-		rm -f $(htmldir)/$$p/*.gif; \
-		echo "copying $$p to $(htmldir)/$$p" ;\
+		$(mkinstalldirs) $(DESTDIR)$(htmldir)/$$p; \
+		rm -f $(DESTDIR)$(htmldir)/$$p/*.html; \
+		rm -f $(DESTDIR)$(htmldir)/$$p/*.gif; \
+		echo "copying $$p to $(DESTDIR)$(htmldir)/$$p" ;\
 		( \
 		cd $(srcdir)/@SN_DOCSET@/$$p ;\
 		$(TAR) -ch$(OLDTARFLAG) -f /tmp/$$p.tar-$$$$ *.html *.gif ;\
-		cp /tmp/$$p.tar-$$$$ $(htmldir)/$$p ;\
+		cp /tmp/$$p.tar-$$$$ $(DESTDIR)$(htmldir)/$$p ;\
 		rm /tmp/$$p.tar-$$$$ ;\
-		cd $(htmldir)/$$p ;\
+		cd $(DESTDIR)$(htmldir)/$$p ;\
 		$(TAR) -xf $$p.tar-$$$$ ;\
 		rm $$p.tar-$$$$ ;\
 		) \
diff -ru /tmp/sourcenav-5.1.1/snavigator/packages/Makefile.in ./snavigator/packages/Makefile.in
--- a/sourcenav-5.1.1/snavigator/packages/Makefile.in	2002-04-26 08:14:28.000000000 +0100
+++ b/sourcenav-5.1.1/snavigator/packages/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -312,8 +312,8 @@
 install-data-local:
 	for i in $(packages); \
 	    do \
-		mkdir -p $(progdir)/$$i ;\
-		cp -p $(srcdir)/$$i/*.tcl $(progdir)/$$i; \
+		mkdir -p $(DESTDIR)$(progdir)/$$i ;\
+		cp -p $(srcdir)/$$i/*.tcl $(DESTDIR)$(progdir)/$$i; \
 	    done;
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -ru /tmp/sourcenav-5.1.1/snavigator/sdk/Makefile.in ./snavigator/sdk/Makefile.in
--- a/sourcenav-5.1.1/snavigator/sdk/Makefile.in	2002-04-26 08:14:28.000000000 +0100
+++ b/sourcenav-5.1.1/snavigator/sdk/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -412,7 +412,7 @@
 
 
 install-exec-local:
-	$(INSTALL_DATA) @TCL_LIB_FULL_PATH@ @SN_SDK_EXEC_PREFIX@/lib
+	$(INSTALL_DATA) @TCL_LIB_FULL_PATH@ $(DESTDIR)@SN_SDK_EXEC_PREFIX@/lib
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -ru /tmp/sourcenav-5.1.1/snavigator/sdk/api/c/Makefile.in ./snavigator/sdk/api/c/Makefile.in
--- a/sourcenav-5.1.1/snavigator/sdk/api/c/Makefile.in	2002-04-26 08:14:28.000000000 +0100
+++ b/sourcenav-5.1.1/snavigator/sdk/api/c/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -415,9 +415,9 @@
 
 
 install-data-local:
-	$(mkinstalldirs) $(sdkincludedir)
+	$(mkinstalldirs) $(DESTDIR)$(sdkincludedir)
 	for h in $(sdkinclude_h) ; do \
-	    cp $(top_srcdir)/../db/include/$$h $(sdkincludedir) ;\
+	    cp $(top_srcdir)/../db/include/$$h $(DESTDIR)$(sdkincludedir) ;\
 	done
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -ru /tmp/sourcenav-5.1.1/snavigator/snavigator/unix/Makefile.in ./snavigator/snavigator/unix/Makefile.in
--- a/sourcenav-5.1.1/snavigator/snavigator/unix/Makefile.in	2002-04-26 08:14:29.000000000 +0100
+++ b/sourcenav-5.1.1/snavigator/snavigator/unix/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -339,7 +339,7 @@
 install-exec-local: $(bindir)/@SN_LAUNCH_SHELL@
 
 $(bindir)/@SN_LAUNCH_SHELL@: snavigator
-	$(INSTALL_SCRIPT) snavigator $(bindir)/@SN_LAUNCH_SHELL@
+	$(INSTALL_SCRIPT) snavigator $(DESTDIR)$(bindir)/@SN_LAUNCH_SHELL@
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -ru /tmp/sourcenav-5.1.1/tcl/unix/Makefile.in ./tcl/unix/Makefile.in
--- a/sourcenav-5.1.1/tcl/unix/Makefile.in	2002-01-10 20:17:57.000000000 +0000
+++ b/sourcenav-5.1.1/tcl/unix/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -1,3 +1,5 @@
+DESTDIR = 
+
 #
 # This file is a Makefile for Tcl.  If it has the name "Makefile.in"
 # then it is a template for a Makefile;  to generate the actual Makefile,
@@ -35,7 +37,7 @@
 # to be different than those used for actually reference files at
 # run-time.  INSTALL_ROOT is prepended to $prefix and $exec_prefix
 # when installing files.
-INSTALL_ROOT		=
+INSTALL_ROOT		= $(DESTDIR)
 
 TCL_LIBRARY =	@datadir@/tcl$(VERSION)
 
diff -ru /tmp/sourcenav-5.1.1/tcl/win/Makefile.in ./tcl/win/Makefile.in
--- a/sourcenav-5.1.1/tcl/win/Makefile.in	2001-09-10 00:56:10.000000000 +0100
+++ b/sourcenav-5.1.1/tcl/win/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -1,3 +1,5 @@
+DESTDIR = 
+
 #
 # This file is a Makefile for Tcl.  If it has the name "Makefile.in"
 # then it is a template for a Makefile;  to generate the actual Makefile,
@@ -34,7 +36,7 @@
 # to be different than those used for actually reference files at
 # run-time.  INSTALL_ROOT is prepended to $prefix and $exec_prefix
 # when installing files.
-INSTALL_ROOT	=
+INSTALL_ROOT	= $(DESTDIR)
 
 # Directory from which applications will reference the libary of Tcl
 # scripts (note: you can set the TCL_LIBRARY environment variable at
diff -ru /tmp/sourcenav-5.1.1/tix/unix/Makefile.in ./tix/unix/Makefile.in
--- a/sourcenav-5.1.1/tix/unix/Makefile.in	2002-01-07 09:52:42.000000000 +0000
+++ b/sourcenav-5.1.1/tix/unix/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -1,3 +1,5 @@
+DESTDIR = 
+
 # This file is a Makefile for Tix.  If it has the name "Makefile.in"
 # then it is a template for a Makefile;  to generate the actual Makefile,
 # run "./configure", which is a configuration script generated by the
@@ -28,30 +30,30 @@
 # Directory in which to install the library of Tix scripts and demos
 # (note: you can set the TIX_LIBRARY environment variable at run-time to
 # override the compiled-in location):
-TIX_LIBRARY =	$(prefix)/share/tix$(TIX_VERSION)
+TIX_LIBRARY =	$(DESTDIR)$(prefix)/share/tix$(TIX_VERSION)
 
 # Directory in which to install the archive libtix.a:
-LIB_DIR =	$(exec_prefix)/lib
+LIB_DIR =	$(DESTDIR)$(exec_prefix)/lib
 
 # Directory in which to install the program wish:
-BIN_DIR =	$(exec_prefix)/bin
+BIN_DIR =	$(DESTDIR)$(exec_prefix)/bin
 
 # Directory in which to install the include file tix.h:
-INCLUDE_DIR =	$(prefix)/include
+INCLUDE_DIR =	$(DESTDIR)$(prefix)/include
 
 # Top-level directory for manual entries:
-MAN_DIR =	$(prefix)/man
+MAN_DIR =	$(DESTDIR)$(prefix)/man
 
 # Directory in which to install manual entry for wish:
-MAN1_DIR =	$(MAN_DIR)/man1
+MAN1_DIR =	$(DESTDIR)$(MAN_DIR)/man1
 
 # Directory in which to install manual entries for Tix's C library
 # procedures:
-MAN3_DIR =	$(MAN_DIR)/man3
+MAN3_DIR =	$(DESTDIR)$(MAN_DIR)/man3
 
 # Directory in which to install manual entries for the built-in
 # Tcl commands implemented by Tix:
-MANN_DIR =	$(MAN_DIR)/mann
+MANN_DIR =	$(DESTDIR)$(MAN_DIR)/mann
 
 # Some versions of make, like SGI's, use the following variable to
 # determine which shell to use for executing commands:
diff -ru /tmp/sourcenav-5.1.1/tix/unix/tk8.0/Makefile.in ./tix/unix/tk8.0/Makefile.in
--- a/sourcenav-5.1.1/tix/unix/tk8.0/Makefile.in	2001-09-09 00:25:55.000000000 +0100
+++ b/sourcenav-5.1.1/tix/unix/tk8.0/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -1,3 +1,5 @@
+DESTDIR = 
+
 # Makefile --
 #
 # This file is a Makefile to compile Tix with Tk version
@@ -33,15 +35,15 @@
 # Directory in which to install the library of Tix scripts and demos
 # (note: you can set the TIX_LIBRARY environment variable at run-time to
 # override the compiled-in location):
-TIX_LIBRARY =	$(prefix)/share/tix$(TIX_VERSION)
+TIX_LIBRARY =	$(DESTDIR)$(prefix)/share/tix$(TIX_VERSION)
 
 # Directory in which to install the archive libtix.a:
-LIB_DIR =	$(exec_prefix)/lib
+LIB_DIR =	$(DESTDIR)$(exec_prefix)/lib
 LIB_INSTALL_DIR = $(LIB_DIR)
 LIB_RUNTIME_DIR = $(LIB_DIR)
 
 # Directory in which to install the program tixwish:
-BIN_DIR =	$(exec_prefix)/bin
+BIN_DIR =	$(DESTDIR)$(exec_prefix)/bin
 
 # To change the compiler switches, for example to change from -O
 # to -g, change the following line:
diff -ru /tmp/sourcenav-5.1.1/tix/unix/tk8.1/Makefile.in ./tix/unix/tk8.1/Makefile.in
--- a/sourcenav-5.1.1/tix/unix/tk8.1/Makefile.in	2001-09-09 00:25:55.000000000 +0100
+++ b/sourcenav-5.1.1/tix/unix/tk8.1/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -1,3 +1,5 @@
+DESTDIR = 
+
 # Makefile --
 #
 # This file is a Makefile to compile Tix with Tk version
@@ -33,15 +35,15 @@
 # Directory in which to install the library of Tix scripts and demos
 # (note: you can set the TIX_LIBRARY environment variable at run-time to
 # override the compiled-in location):
-TIX_LIBRARY =	$(prefix)/share/tix$(TIX_VERSION)
+TIX_LIBRARY =	$(DESTDIR)$(prefix)/share/tix$(TIX_VERSION)
 
 # Directory in which to install the archive libtix.a:
-LIB_DIR =	$(exec_prefix)/lib
+LIB_DIR =	$(DESTDIR)$(exec_prefix)/lib
 LIB_INSTALL_DIR = $(LIB_DIR)
 LIB_RUNTIME_DIR = $(LIB_DIR)
 
 # Directory in which to install the program tixwish:
-BIN_DIR =	$(exec_prefix)/bin
+BIN_DIR =	$(DESTDIR)$(exec_prefix)/bin
 
 # To change the compiler switches, for example to change from -O
 # to -g, change the following line:
diff -ru /tmp/sourcenav-5.1.1/tix/unix/tk8.3/Makefile.in ./tix/unix/tk8.3/Makefile.in
--- a/sourcenav-5.1.1/tix/unix/tk8.3/Makefile.in	2002-01-11 23:43:38.000000000 +0000
+++ b/sourcenav-5.1.1/tix/unix/tk8.3/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -1,3 +1,5 @@
+DESTDIR = 
+
 # Makefile --
 #
 # This file is a Makefile to compile Tix with Tk version
@@ -33,15 +35,15 @@
 # Directory in which to install the library of Tix scripts and demos
 # (note: you can set the TIX_LIBRARY environment variable at run-time to
 # override the compiled-in location):
-TIX_LIBRARY =	$(prefix)/share/tix$(TIX_VERSION)
+TIX_LIBRARY =	$(DESTDIR)$(prefix)/share/tix$(TIX_VERSION)
 
 # Directory in which to install the archive libtix.a:
-LIB_DIR =	$(exec_prefix)/lib
+LIB_DIR =	$(DESTDIR)$(exec_prefix)/lib
 LIB_INSTALL_DIR = $(LIB_DIR)
 LIB_RUNTIME_DIR = $(LIB_DIR)
 
 # Directory in which to install the program tixwish:
-BIN_DIR =	$(exec_prefix)/bin
+BIN_DIR =	$(DESTDIR)$(exec_prefix)/bin
 
 # To change the compiler switches, for example to change from -O
 # to -g, change the following line:
diff -ru /tmp/sourcenav-5.1.1/tix/win/Makefile.in ./tix/win/Makefile.in
--- a/sourcenav-5.1.1/tix/win/Makefile.in	2001-09-13 01:43:23.000000000 +0100
+++ b/sourcenav-5.1.1/tix/win/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -469,27 +469,27 @@
 # CYGNUS LOCAL: dj - change to autoconf'd directories
 
 # Directory in which to install the archive libtix.a:
-LIB_DIR =	$(libdir)
+LIB_DIR =	$(DESTDIR)$(libdir)
 
 # Directory in which to install the program wish:
-BIN_DIR =	$(bindir)
+BIN_DIR =	$(DESTDIR)$(bindir)
 
 # Directory in which to install the include file tix.h:
-INCLUDE_DIR =	$(includedir)
+INCLUDE_DIR =	$(DESTDIR)$(includedir)
 
 # Top-level directory for manual entries:
-MAN_DIR =	$(mandir)
+MAN_DIR =	$(DESTDIR)$(mandir)
 
 # Directory in which to install manual entry for wish:
-MAN1_DIR =	$(MAN_DIR)/man1
+MAN1_DIR =	$(DESTDIR)$(MAN_DIR)/man1
 
 # Directory in which to install manual entries for Tix's C library
 # procedures:
-MAN3_DIR =	$(MAN_DIR)/man3
+MAN3_DIR =	$(DESTDIR)$(MAN_DIR)/man3
 
 # Directory in which to install manual entries for the built-in
 # Tcl commands implemented by Tix:
-MANN_DIR =	$(MAN_DIR)/mann
+MANN_DIR =	$(DESTDIR)$(MAN_DIR)/mann
 
 INSTALL		=	@SRC_DIR@/install.sh -c
 INSTALL_PROGRAM =	@INSTALL_PROGRAM@
diff -ru /tmp/sourcenav-5.1.1/tk/unix/Makefile.in ./tk/unix/Makefile.in
--- a/sourcenav-5.1.1/tk/unix/Makefile.in	2002-01-10 20:21:26.000000000 +0000
+++ b/sourcenav-5.1.1/tk/unix/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -1,3 +1,5 @@
+DESTDIR =
+
 #
 # This file is a Makefile for Tk.  If it has the name "Makefile.in"
 # then it is a template for a Makefile;  to generate the actual Makefile,
@@ -37,7 +39,7 @@
 # to be different than those used for actually reference files at
 # run-time.  INSTALL_ROOT is prepended to $prefix and $exec_prefix
 # when installing files.
-INSTALL_ROOT		=
+INSTALL_ROOT		= $(DESTDIR)
 
 # Directory from which applications will reference the library of Tcl
 # scripts (note: you can set the TK_LIBRARY environment variable at
diff -ru /tmp/sourcenav-5.1.1/tk/win/Makefile.in ./tk/win/Makefile.in
--- a/sourcenav-5.1.1/tk/win/Makefile.in	2001-09-10 02:16:55.000000000 +0100
+++ b/sourcenav-5.1.1/tk/win/Makefile.in	2002-12-11 20:55:29.000000000 +0000
@@ -1,3 +1,5 @@
+DESTDIR = 
+
 # This file is a Makefile for Tk.  If it has the name "Makefile.in"
 # then it is a template for a Makefile;  to generate the actual Makefile,
 # run "./configure", which is a configuration script generated by the
@@ -34,7 +36,7 @@
 # to be different than those used for actually reference files at
 # run-time.  INSTALL_ROOT is prepended to $prefix and $exec_prefix
 # when installing files.
-INSTALL_ROOT		=
+INSTALL_ROOT		= $(DESTDIR)
 
 # Directory from which applications will reference the library of Tk
 # scripts (note: you can set the TK_LIBRARY environment variable at
