Index: quark-3.21/strange-quark/Makefile.am
===================================================================
--- quark-3.21.orig/strange-quark/Makefile.am
+++ quark-3.21/strange-quark/Makefile.am
@@ -6,7 +6,6 @@ AM_CPPFLAGS = \
   $(GTK_CFLAGS) \
   $(GCONF_CFLAGS) \
   $(GDK_PIXBUF_CFLAGS) \
-  -DGTK_DISABLE_DEPRECATED \
   -DG_DISABLE_DEPRECATED \
   -DG_LOG_DOMAIN=\"strange-quark\" \
   -DPIXMAPDIR=\"$(pixmapdir)\" \
Index: quark-3.21/strange-quark/Makefile.in
===================================================================
--- quark-3.21.orig/strange-quark/Makefile.in
+++ quark-3.21/strange-quark/Makefile.in
@@ -155,7 +155,6 @@ AM_CPPFLAGS = \
   $(GTK_CFLAGS) \
   $(GCONF_CFLAGS) \
   $(GDK_PIXBUF_CFLAGS) \
-  -DGTK_DISABLE_DEPRECATED \
   -DG_DISABLE_DEPRECATED \
   -DG_LOG_DOMAIN=\"strange-quark\" \
   -DPIXMAPDIR=\"$(pixmapdir)\" \
