diff -Nru squidclamav-3.5.orig/squidclamav.conf.dist squidclamav-3.5/squidclamav.conf.dist
--- squidclamav-3.5.orig/squidclamav.conf.dist	2008-03-28 10:51:18.000000000 +0200
+++ squidclamav-3.5/squidclamav.conf.dist	2008-05-15 01:05:55.000000000 +0300
@@ -29,7 +29,7 @@
 #
 #	stat 0|1
 #
-#	squidguard /usr/local/squidGuard/bin/squidGuard
+#	squidguard /usr/bin/squidGuard
 #
 #	whitelist this.trustdomain.com
 #	whitelist .*\.domain\.*
@@ -39,12 +39,12 @@
 #proxy http://127.0.0.1:3128/
 #logfile /var/log/squidclamav.log
 #redirect http://proxy.domain.com/cgi-bin/clwarn.cgi
-#squidguard /usr/local/squidGuard/bin/squidGuard
+#squidguard /usr/bin/squidGuard
 #debug 0
 #force 1
 #stat 1
-#clamd_local /tmp/clamd
-#clamd_ip 192.168.1.5
+#clamd_local /var/run/clamav/clamd.sock
+#clamd_ip 127.0.0.1
 #clamd_port 3310
 #timeout 60
 #abort ^.*\.gz$
