https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=3314

=== modified file 'epan/dissectors/packet-sflow.c'
--- epan/dissectors/packet-sflow.c	2009-03-08 10:18:22 +0000
+++ epan/dissectors/packet-sflow.c	2009-03-08 10:22:32 +0000
@@ -563,13 +563,13 @@
 {
 	guint32 	header_proto, frame_length;
 	volatile 	guint32 	header_length;
-	tvbuff_t 	*next_tvb;
-	proto_tree 	*sflow_header_tree;
+	tvbuff_t 	*volatile next_tvb;
+	proto_tree 	*volatile sflow_header_tree;
 	proto_item 	*ti;
 	/* stuff for saving column state before calling other dissectors.
 	 * Thanks to Guy Harris for the tip. */
-	gboolean save_writable;
-	gboolean save_in_error_pkt;
+	volatile gboolean save_writable;
+	volatile gboolean save_in_error_pkt;
 	volatile address 	save_dl_src;
 	volatile address 	save_dl_dst;
 	volatile address 	save_net_src;

