diff --git a/graf3d/gl/src/TGLFontManager.cxx b/graf3d/gl/src/TGLFontManager.cxx
index 5334023..8f35ccc 100644
--- a/graf3d/gl/src/TGLFontManager.cxx
+++ b/graf3d/gl/src/TGLFontManager.cxx
@@ -6,7 +6,7 @@
 #include "TObjString.h"
 #include "TGLUtil.h"
 
-#include <FTGL/ftgl.h>
+#include <FTGL/FTGL.h>
 #include "FTGLExtrdFont.h"
 #include "FTGLOutlineFont.h"
 #include "FTGLPolygonFont.h"
diff --git a/graf3d/gl/src/TGLText.cxx b/graf3d/gl/src/TGLText.cxx
index a60c532..bb0dc7c 100644
--- a/graf3d/gl/src/TGLText.cxx
+++ b/graf3d/gl/src/TGLText.cxx
@@ -18,7 +18,7 @@
 #include "TSystem.h"
 #include "TEnv.h"
 
-#include <FTGL/ftgl.h>
+#include <FTGL/FTGL.h>
 #include "FTGLOutlineFont.h"
 #include "FTGLPolygonFont.h"
 #include "FTGLTextureFont.h"
