--- configure.orig	2008-01-17 17:11:16.765239211 +0000
+++ configure	2008-01-17 17:38:57.088901480 +0000
@@ -1539,7 +1539,7 @@
       if pkg-config --exists ftgl ; then
 	  result "ok"
 	  ftglincdir=`pkg-config --cflags-only-I ftgl | sed 's/-I//g'`
-	  ftgllibs=`pkg-config --libs-only-l ftgl | sed s'/-lftgl/-lftgl_pic/'`
+	  ftgllibs=`pkg-config --libs-only-l ftgl`
 	  ftgllibdir=`pkg-config --libs-only-L ftgl | sed 's/-L//g'`
 	  enable_builtin_ftgl=no
       else
@@ -3868,8 +3868,8 @@
     gslskip=
     hasmathmore="define"
     # by default use now static GSL library everywhere
-    #enable_gsl_shared=$enable_shared"
-    enable_gsl_shared="no"
+    enable_gsl_shared="$enable_shared"
+    #enable_gsl_shared="no"
 
     check_header "gsl/gsl_version.h" "$gslincdir" $GSL $GSL/include \
 	$GSL/api /usr/local/include /usr/include /opt/gsl/include \
@@ -3955,10 +3955,10 @@
 	        # and linker flags
 		result "ok"
 	        gslflags=`gsl-config --cflags`
-	        #gsllibs=`gsl-config --libs`
+	        gsllibs=`gsl-config --libs`
 		#use static libs
 		gslprefix=`gsl-config --prefix`
-		gsllibs="$gslprefix/lib/libgsl.a $gslprefix/lib/libgslcblas.a"
+		#gsllibs="$gslprefix/lib/libgsl.a $gslprefix/lib/libgslcblas.a"
 		gsllibdir=""
                 gslversion=$gsl_version
             fi
