diff -uar kdebindings-4.4.5-orig/csharp/kimono/CMakeLists.txt kdebindings-4.4.5/csharp/kimono/CMakeLists.txt
--- kdebindings-4.4.5-orig/csharp/kimono/CMakeLists.txt	2010-10-24 16:11:27.000000000 +0200
+++ kdebindings-4.4.5/csharp/kimono/CMakeLists.txt	2010-10-24 16:12:17.000000000 +0200
@@ -22,7 +22,7 @@
                       ${KDE4_KDECORE_LIBS} ${KDE4_KIO_LIBS})
 
 # write FindPackage modules for these
-pkg_search_module(MONO mono)
+pkg_search_module(MONO mono-2 mono)
 pkg_search_module(GLIB2 glib-2.0)
 
 # if libmono-dev is not installed, then MONO_INCLUDES is not defined and compiling
diff -uar kdebindings-4.4.5-orig/csharp/kimono/src/kimonopluginfactory.cpp kdebindings-4.4.5/csharp/kimono/src/kimonopluginfactory.cpp
--- kdebindings-4.4.5-orig/csharp/kimono/src/kimonopluginfactory.cpp	2010-10-24 16:11:27.000000000 +0200
+++ kdebindings-4.4.5/csharp/kimono/src/kimonopluginfactory.cpp	2010-10-24 16:12:40.000000000 +0200
@@ -44,6 +44,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#include <glib.h>
 
 class QWidget;
 
