--- Makefile.orig	2003-08-17 17:03:30.643555488 -0400
+++ Makefile	2003-08-17 17:03:50.415549688 -0400
@@ -195,8 +195,8 @@
 ENGINEDIR = .
 ASMFLAGS = -f $(ASMOBJFMT) $(ASMDEFS)
 LINKER = gcc
-CFLAGS += $(USE_ASM) -funsigned-char -O2 -DPLATFORM_UNIX -g -Wall $(SDL_CFLAGS) -fno-omit-frame-pointer
-LDFLAGS += -g $(SDL_LDFLAGS)
+CFLAGS += $(USE_ASM) $(OPTFLAGS) -funsigned-char -DPLATFORM_UNIX -Wall $(SDL_CFLAGS) -fno-omit-frame-pointer
+LDFLAGS += $(SDL_LDFLAGS)
 
 # Rules for turning source files into .o files
 %.o: %.c
