diff -urN wmget/Makefile wmget-modified/Makefile
--- wmget/Makefile	2002-06-19 17:22:57.000000000 -0500
+++ wmget-modified/Makefile	2002-10-03 10:51:38.000000000 -0500
@@ -46,7 +46,7 @@
 
 # To specify a different prefix, you can override this on the command line
 #	make PREFIX=/opt/dockapps install
-PREFIX=		/usr/local
+PREFIX=		/usr
 
 INSTALLDIR=	install -d -m 755
 INSTALLBIN=	install -m 555
@@ -86,12 +86,12 @@
 			echo $(PREFIX) > install.prefix ; \
 			$(INSTALLDIR) $(PREFIX)/bin ; \
 			$(INSTALLBIN) wmget $(PREFIX)/bin/wmget ; \
-			$(INSTALLDIR) $(PREFIX)/man/man1 ; \
-			$(INSTALLMAN) wmget.1 $(PREFIX)/man/man1/wmget.1 ; \
+			$(INSTALLDIR) $(PREFIX)/share/man/man1 ; \
+			$(INSTALLMAN) wmget.1 $(PREFIX)/share/man/man1/wmget.1 ; \
 		fi
 
 uninstall:
-		cd `cat install.prefix` && rm -f bin/wmget man/man1/wmget.1
+		cd `cat install.prefix` && rm -f bin/wmget share/man/man1/wmget.1
 		rm -f install.prefix
 
 wmget:		dockapplib $(OBJS)
