--- usr/lib/vdr/rcscript/pre-start-40-config-files.sh	(Revision 165)
+++ usr/lib/vdr/rcscript/pre-start-40-config-files.sh	(Revision 166)
@@ -13,11 +13,11 @@
 
 	if [[ -f /etc/vdr/setup.conf ]]; then
 		if [[ -n "${STARTUP_VOLUME}" ]]; then
-			/usr/bin/sed -i /etc/vdr/setup.conf -e "s/^CurrentVolume =.*\$/CurrentVolume = ${STARTUP_VOLUME}/"
+			/bin/sed -i /etc/vdr/setup.conf -e "s/^CurrentVolume =.*\$/CurrentVolume = ${STARTUP_VOLUME}/"
 		fi
 
 		if [[ -n "${STARTUP_CHANNEL}" ]]; then
-			/usr/bin/sed -i /etc/vdr/setup.conf -e "s/^CurrentChannel =.*\$/CurrentChannel = ${STARTUP_CHANNEL}/"
+			/bin/sed -i /etc/vdr/setup.conf -e "s/^CurrentChannel =.*\$/CurrentChannel = ${STARTUP_CHANNEL}/"
 		fi
 	fi
 
