diff -ru gentoo-vdr-scripts-0.2.2-orig/etc/conf.d/vdr gentoo-vdr-scripts-0.2.2/etc/conf.d/vdr
--- gentoo-vdr-scripts-0.2.2-orig/etc/conf.d/vdr	2006-02-07 20:28:29.967073000 +0100
+++ gentoo-vdr-scripts-0.2.2/etc/conf.d/vdr	2006-02-07 20:29:48.135958250 +0100
@@ -110,14 +110,16 @@
 # compile time
 # (Usable at vdr version starting from 1.3.28)
 #   allowed values: lirc rcu
-#   default: use what was selected at compile time
+#   default: use no remote
+#            (up to vdr-1.3.36-r2: use what was selected by use-flag)
 #IR_CTRL="lirc"
 
 # Change the filename-format to be vfat and samba compatible. Only necessary if
 # not set at compile time.
 # (Usable at vdr version starting from 1.3.32)
 #   allowed values: yes no
-#   default: use what was selected at compile time
+#   default: no
+#            (up to vdr-1.3.36-r2: use what was selected by use-flag)
 #VFAT_FILENAMES="yes"
 
 # use PORT for SVDRP (default: 2001)
