--- gentoo-vdr-scripts-0.2.2/etc/conf.d/vdr
+++ gentoo-vdr-scripts-0.2.2/etc/conf.d/vdr
@@ -97,10 +97,10 @@
 #EPGFILE="/var/vdr/epg.data"
 
 # activate the (internal to vdr) watchdog timer with
-# a timeout of SEC seconds (default: disabled)
+# a timeout of SEC seconds (recommended by Klaus Schmidinger)
 # '0' disables the watchdog
 #   allowed values: 0 1 2 ...
-#   default: 60
+#   default: enabled / 60 seconds
 #INTERNAL_WATCHDOG="60"
 
 # If you seek the settings for the extern watchdog look
