diff -Naur hplip-3.9.12-old/installer/core_install.py hplip-3.9.12/installer/core_install.py
--- hplip-3.9.12-old/installer/core_install.py	2009-12-15 22:22:13.000000000 +0100
+++ hplip-3.9.12/installer/core_install.py	2009-12-20 02:11:47.000000000 +0100
@@ -29,6 +29,7 @@
 import grp
 import pwd
 import tarfile
+import commands # TODO: Replace with subprocess (commands is deprecated in Python 3.0)
 
 try:
     import hashlib # new in 2.5
@@ -1036,9 +1037,8 @@
     def check_cupsddk(self):
         log.debug("Checking for cups-ddk...")
         # TODO: Compute these paths some way or another...
-        #return check_tool("/usr/lib/cups/driver/drv list") and os.path.exists("/usr/share/cupsddk/include/media.defs")
-        return (check_file('drv', "/usr/lib/cups/driver") or check_file('drv', "/usr/lib64/cups/driver")) and \
-            check_file('media.defs', "/usr/share/cupsddk/include")
+        #return check_tool(commands.getoutput("cups-config --serverbin") + "/driver/drv list") and os.path.exists("/usr/share/cupsddk/include/media.defs")
+        return check_file('drv', commands.getoutput("cups-config --serverbin") + "/driver") and check_file('media.defs', "/usr/share/cupsddk/include")
 
 
     def check_policykit(self):
