--- Makefile.am.old	2009-11-06 22:56:26.000000000 +0100
+++ Makefile.am	2009-11-14 16:50:39.000000000 +0100
@@ -325,6 +325,7 @@
 hppgsz_CFLAGS = $(libapdk_la_CFLAGS)
 hppgsz_LDADD = libapdk.la -ljpeg -ldl
 
+if GUI_BUILD
 # hplip.desktop
 hplip_desktopdir = $(icondir)
 hplip_desktop_DATA = hplip.desktop
@@ -332,6 +333,7 @@
 # hplip-systray.desktop
 hplip_systraydir = $(systraydir)
 hplip_systray_DATA = hplip-systray.desktop
+endif
 
 # hpipjs (CUPS filter)
 hplipjsdir = $(cupsfilterdir)
