diff -Naur hplip-3.9.10-old/configure.in hplip-3.9.10/configure.in
--- hplip-3.9.10-old/configure.in	2009-11-15 18:24:03.000000000 +0100
+++ hplip-3.9.10/configure.in	2009-11-15 18:30:14.000000000 +0100
@@ -319,6 +319,9 @@
 AC_ARG_WITH(docdir, AC_HELP_STRING([--with-docdir=DIR], [set hplip documentation directory [default=datadir/doc]]),
    hpdocdir=$withval, hpdocdir="$datadir/doc/hplip-$VERSION")
 
+AC_ARG_WITH(htmldir, AC_HELP_STRING([--with-htmldir=DIR], [set hplip html documentation directory [default=datadir/doc]]),
+   hphtmldir=$withval, hphtmldir="$datadir/doc/hplip-$VERSION")
+
 AC_MSG_CHECKING([for foomatic ppd install])
 AC_ARG_ENABLE(foomatic_ppd_install,
   [  --enable-foomatic-ppd-install    enable foomatic static ppd install (default=no), uses hpppddir],
@@ -543,6 +546,7 @@
 AC_DEFINE_DIR([abs_sbindir], [sbindir])
 AC_DEFINE_DIR([abs_hpppddir], [hpppddir])
 AC_DEFINE_DIR([abs_docdir], [hpdocdir])
+AC_DEFINE_DIR([abs_htmldir], [hphtmldir])
 abs_ppddir=${abs_hpppddir%/*}
 AC_DEFINE_DIR([abs_drvdir], [drvdir])
 
@@ -550,6 +554,7 @@
 AC_SUBST(abs_sbindir) 
 AC_SUBST(abs_hpppddir) 
 AC_SUBST(abs_docdir) 
+AC_SUBST(abs_htmldir)
 AC_SUBST(abs_ppddir) 
 AC_SUBST(abs_drvdir) 
 AC_SUBST(icondir)
@@ -558,6 +563,7 @@
 AC_SUBST(cupsfilterdir)
 AC_SUBST(hpppddir)
 AC_SUBST(hpdocdir)
+AC_SUBST(hphtmldir)
 AC_SUBST(drvdir)
 AC_SUBST(network_build)
 AC_SUBST(pp_build)
diff -Naur hplip-3.9.10-old/hplip.conf.in hplip-3.9.10/hplip.conf.in
--- hplip-3.9.10-old/hplip.conf.in	2009-11-15 18:24:03.000000000 +0100
+++ hplip-3.9.10/hplip.conf.in	2009-11-15 18:25:01.000000000 +0100
@@ -9,6 +9,7 @@
 ppd=@abs_hpppddir@
 ppdbase=@abs_ppddir@
 doc=@abs_docdir@
+html=@abs_htmldir@
 icon=@icondir@
 cupsbackend=@cupsbackenddir@
 cupsfilter=@cupsfilterdir@
@@ -42,4 +43,4 @@
 lite-build=@lite_build@
 udev-acl-rules=@udev_acl_rules@
 hpcups-only-build=@hpcups_only_build@
-hpijs-only-build=@hpijs_only_build@
\ Kein Zeilenumbruch am Dateiende.
+hpijs-only-build=@hpijs_only_build@
diff -Naur hplip-3.9.10-old/hplip.list.in hplip-3.9.10/hplip.list.in
--- hplip-3.9.10-old/hplip.list.in	2009-11-15 18:24:03.000000000 +0100
+++ hplip-3.9.10/hplip.list.in	2009-11-15 18:25:01.000000000 +0100
@@ -49,6 +49,7 @@
 $icondir=@icondir@
 $systraydir=@systraydir@
 $docdir=@hpdocdir@
+$htmldir=@hphtmldirdir@
 $version=@VERSION@
 $internal_tag=@PACKAGE_BUGREPORT@
 
@@ -252,11 +253,11 @@
 @epm_full@f 644 root root $home/pcard/photocard.py pcard/photocard.py
 @epm_full@f 755 root root $home/info.py info.py
 # docs
-@epm_full@f 644 root root $docdir doc/*
-@epm_full@d 775 root root $docdir/images -
-@epm_full@f 644 root root $docdir/images doc/images/*
-@epm_full@d 775 root root $docdir/styles -
-@epm_full@f 644 root root $docdir/styles doc/styles/*
+@epm_full@f 644 root root $htmldir doc/*
+@epm_full@d 775 root root $htmldir/images -
+@epm_full@f 644 root root $htmldir/images doc/images/*
+@epm_full@d 775 root root $htmldir/styles -
+@epm_full@f 644 root root $htmldir/styles doc/styles/*
 # python commands
 @epm_full@l 755 root root $bindir/hp-toolbox ../share/hplip/toolbox.py
 @epm_full@l 755 root root $bindir/hp-colorcal ../share/hplip/colorcal.py
diff -Naur hplip-3.9.10-old/Makefile.am hplip-3.9.10/Makefile.am
--- hplip-3.9.10-old/Makefile.am	2009-11-15 18:24:03.000000000 +0100
+++ hplip-3.9.10/Makefile.am	2009-11-15 18:27:20.000000000 +0100
@@ -9,6 +9,7 @@
 hplipdir = $(datadir)/hplip
 
 docdir = $(hpdocdir)
+htmldir = $(hphtmldir)
 doc_DATA = COPYING copyright prnt/hpijs/README_LIBJPG
 EXTRA_DIST = prnt/hpijs/gdevijs-krgb-1.5-gs8.61.patch prnt/hpijs/README_LIBJPG copyright
 
@@ -23,11 +24,11 @@
 
 if DOC_BUILD
 wwwsrc = doc
-www0dir = $(docdir)
+www0dir = $(htmldir)
 dist_www0_DATA = $(wwwsrc)/index.html $(wwwsrc)/commandline.html $(wwwsrc)/copying.html $(wwwsrc)/devicemanager.html $(wwwsrc)/faxtrouble.html $(wwwsrc)/gettinghelp.html $(wwwsrc)/hpscan.html $(wwwsrc)/mainttask.html $(wwwsrc)/plugins.html $(wwwsrc)/print.html $(wwwsrc)/printing.html $(wwwsrc)/printoptions.html $(wwwsrc)/printtroubleshooting.html $(wwwsrc)/scanning.html $(wwwsrc)/scantrouble.html $(wwwsrc)/sendfax.html $(wwwsrc)/setup.html $(wwwsrc)/systray.html $(wwwsrc)/troubleshooting.html $(wwwsrc)/uninstalling.html $(wwwsrc)/upgrading.html
-www3dir = $(docdir)/styles
+www3dir = $(htmldir)/styles
 dist_www3_DATA = $(wwwsrc)/styles/*
-www4dir = $(docdir)/images
+www4dir = $(htmldir)/images
 dist_www4_DATA = $(wwwsrc)/images/*
 endif
 
