Index: installer/core_install.py
===================================================================
--- installer/core_install.py	(revision 13658)
+++ installer/core_install.py	(working copy)
@@ -1990,7 +1990,11 @@
         if mode == GUI_MODE:
             return os.system("sh %s --nox11 -- -u" % plugin_file) == 0
         else:
-            return os.system("sh %s --nox11 -- -i" % plugin_file) == 0
+            if os.system("sh %s --nox11 -- -i" % plugin_file) == 0:
+                return True
+            else:
+                log.error("Python gobject/dbus may be not installed")
+                return False
 
 
     def delete_plugin(self):
