--- terminatorX-3.80/gnome-support/Makefile.in.orig	2003-06-25 11:10:16.000000000 +0200
+++ terminatorX-3.80/gnome-support/Makefile.in	2003-06-25 11:12:32.000000000 +0200
@@ -116,9 +116,9 @@
 build_alias = @build_alias@
 datadir = @datadir@
 exec_prefix = @exec_prefix@
 gnomeconfig = @gnomeconfig@
-gnomedatadir = @gnomedatadir@
+gnomedatadir = $(DESTDIR)@gnomedatadir@
 gnomepresent = @gnomepresent@
 host_alias = @host_alias@
 includedir = @includedir@
 infodir = @infodir@
diff -ur terminatorX-3.80/doc/terminatorX-manual/C/Makefile.in terminatorX-3.80.orig/doc/terminatorX-manual/C/Makefile.in
--- terminatorX-3.80/doc/terminatorX-manual/C/Makefile.in	2003-06-25 12:06:28.000000000 +0200
+++ terminatorX-3.80.orig/doc/terminatorX-manual/C/Makefile.in	2003-06-25 11:58:51.000000000 +0200
@@ -186,7 +186,7 @@
 install_sh = @install_sh@
 libdir = @libdir@
 libexecdir = @libexecdir@
-localstatedir = @localstatedir@
+localstatedir = $(DESTDIR)@localstatedir@
 mandir = @mandir@
 oldincludedir = @oldincludedir@
 prefix = @prefix@
