diff -NrU5 nant-0.86-beta1.orig/src/NAnt.NUnit/NAnt.NUnit.build nant-0.86-beta1/src/NAnt.NUnit/NAnt.NUnit.build
--- nant-0.86-beta1.orig/src/NAnt.NUnit/NAnt.NUnit.build	2009-05-03 17:04:46.000000000 +0200
+++ nant-0.86-beta1/src/NAnt.NUnit/NAnt.NUnit.build	2009-05-03 17:05:03.000000000 +0200
@@ -15,12 +15,14 @@
                 <!-- common assembly-level attributes -->
                 <include name="../CommonAssemblyInfo.cs" />
             </sources>
             <references>
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
+	    <pkg-references>
+	        <package name="mono-nunit"/>
+	    </pkg-references>
         </csc>
         <csc target="library" define="${current.build.defines}" warnaserror="${build.warnaserror}" debug="${build.debug}" output="${build.dir}/bin/NAnt.NUnit2Tasks.dll" doc="${build.dir}/bin/NAnt.NUnit2Tasks.xml">
             <nowarn>
                 <!-- do not report deprecation warnings -->
                 <warning number="0618" />
@@ -36,12 +38,12 @@
                 <include name="${lib.framework.dir}/NDoc.Core.dll" />
                 <!-- end workaround -->
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
                 <include name="${build.dir}/bin/NAnt.NUnit.dll" />
                 <include name="${build.dir}/bin/NAnt.DotNetTasks.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
-                <include name="${lib.framework.dir}/nunit.core.dll" />
-                <include name="${lib.framework.dir}/nunit.util.dll" />
             </references>
+	    <pkg-references>
+	        <package name="mono-nunit"/>
+	    </pkg-references>
         </csc>
     </target>
 </project>
