diff -NrU5 nant-0.86-beta1.orig/src/NAnt.NUnit/NAnt.NUnit.build nant-0.86-beta1/src/NAnt.NUnit/NAnt.NUnit.build
--- nant-0.86-beta1.orig/src/NAnt.NUnit/NAnt.NUnit.build	2009-05-03 16:38:14.000000000 +0200
+++ nant-0.86-beta1/src/NAnt.NUnit/NAnt.NUnit.build	2009-05-03 16:39:16.000000000 +0200
@@ -18,34 +18,10 @@
             <references>
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
                 <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
         </csc>
-        <csc target="library" define="${current.build.defines}" warnaserror="${build.warnaserror}" debug="${build.debug}" output="${build.dir}/bin/NAnt.NUnit1Tasks.dll" doc="${build.dir}/bin/NAnt.NUnit1Tasks.xml">
-            <nowarn>
-                <!-- do not report warnings for missing XML comments -->
-                <warning number="1591" />
-                 <!-- workaround for Mono bug #61902: do not report deprecation warnings -->
-                <warning number="0618" if="${framework::get-family(framework::get-target-framework()) == 'mono'}" />
-            </nowarn>
-            <sources>
-                <include name="NUnit1/**/*.cs"/>
-                <!-- common assembly-level attributes -->
-                <include name="../CommonAssemblyInfo.cs" />
-            </sources>
-            <references>
-                <!-- temporary workaround for Mono Runtime bug #57602 -->
-                <include name="${lib.framework.dir}/log4net.dll" />
-                <!-- end workaround -->
-                <include name="${build.dir}/bin/NAnt.Core.dll" />
-                <include name="${build.dir}/bin/NAnt.NUnit.dll" />
-                <include name="${lib.dir}/common/neutral/NUnitCore.dll" />
-            </references>
-            <resources>
-                <include name="*.resx"/>
-            </resources>
-        </csc>
         <csc target="library" define="${current.build.defines}" warnaserror="${build.warnaserror}" debug="${build.debug}" output="${build.dir}/bin/NAnt.NUnit2Tasks.dll" doc="${build.dir}/bin/NAnt.NUnit2Tasks.xml">
             <nowarn>
                 <!-- do not report deprecation warnings -->
                 <warning number="0618" />
             </nowarn>
