diff -ur libwefts-0.99c/Makefile.am libwefts-0.99c-order/Makefile.am
--- libwefts-0.99c/Makefile.am	2004-04-10 22:14:07.000000000 +0200
+++ libwefts-0.99c-order/Makefile.am	2005-05-28 13:57:17.152117664 +0200
@@ -4,7 +4,7 @@
 # Toplevel Makefile.am
 # $Id: wefts-order.patch,v 1.1 2005/05/28 12:34:30 flameeyes Exp $
 
-SUBDIRS = include ltlib testsuite tests src doc
+SUBDIRS = include ltlib src testsuite tests doc
 
 EXTRA_DIST = \
    AUTHORS README RELNOTES \
diff -ur libwefts-0.99c/tests/Makefile.am libwefts-0.99c-order/tests/Makefile.am
--- libwefts-0.99c/tests/Makefile.am	2004-03-28 10:28:10.000000000 +0200
+++ libwefts-0.99c-order/tests/Makefile.am	2005-05-28 14:07:42.447058384 +0200
@@ -43,92 +43,92 @@
    testxmutex
 
 test_SOURCES = test.cpp
-test_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+test_LDADD = $(top_builddir)/src/libwefts.a
 test_LDFLAGS=-lpthread
 
 teststop_SOURCES = teststop.cpp
-teststop_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+teststop_LDADD = $(top_builddir)/src/libwefts.a
 teststop_LDFLAGS=-lpthread
 
 testbarrier_SOURCES = testbarrier.cpp
-testbarrier_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testbarrier_LDADD = $(top_builddir)/src/libwefts.a
 testbarrier_LDFLAGS=-lpthread
 
 testcritical_SOURCES = testcritical.cpp
-testcritical_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testcritical_LDADD = $(top_builddir)/src/libwefts.a
 testcritical_LDFLAGS=-lpthread
 
 
 testopenbar_SOURCES = testopenbar.cpp
-testopenbar_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testopenbar_LDADD = $(top_builddir)/src/libwefts.a
 testopenbar_LDFLAGS=-lpthread
 
 testsubscr_SOURCES = testsubscr.cpp
-testsubscr_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testsubscr_LDADD = $(top_builddir)/src/libwefts.a
 testsubscr_LDFLAGS=-lpthread
 
 testtimesub_SOURCES = testtimesub.cpp
-testtimesub_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testtimesub_LDADD = $(top_builddir)/src/libwefts.a
 testtimesub_LDFLAGS=-lpthread
 
 testrlock_SOURCES = testrlock.cpp
-testrlock_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testrlock_LDADD = $(top_builddir)/src/libwefts.a
 testrlock_LDFLAGS=-lpthread
 
 testrwlock_SOURCES = testrwlock.cpp
-testrwlock_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testrwlock_LDADD = $(top_builddir)/src/libwefts.a
 testrwlock_LDFLAGS=-lpthread
 
 testfence_SOURCES = testfence.cpp
-testfence_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testfence_LDADD = $(top_builddir)/src/libwefts.a
 testfence_LDFLAGS=-lpthread
 
 testmoan_SOURCES = testmoan.cpp
-testmoan_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testmoan_LDADD = $(top_builddir)/src/libwefts.a
 testmoan_LDFLAGS=-lpthread
 
 testpromote_SOURCES = testpromote.cpp
-testpromote_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testpromote_LDADD = $(top_builddir)/src/libwefts.a
 testpromote_LDFLAGS=-lpthread
 
 testcleanup_SOURCES = testcleanup.cpp
-testcleanup_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testcleanup_LDADD = $(top_builddir)/src/libwefts.a
 testcleanup_LDFLAGS=-lpthread
 
 testcoffee_SOURCES = testcoffee.cpp
-testcoffee_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a ../lib/libwefts.a
+testcoffee_LDADD = $(top_builddir)/src/libwefts.a
 testcoffee_LDFLAGS=-lpthread
 
 testcoffee2_SOURCES = testcoffee2.cpp
-testcoffee2_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testcoffee2_LDADD = $(top_builddir)/src/libwefts.a
 testcoffee2_LDFLAGS=-lpthread
 
 testcond_SOURCES = testcond.cpp
-testcond_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testcond_LDADD = $(top_builddir)/src/libwefts.a
 testcond_LDFLAGS=-lpthread
 
 testspecific_SOURCES = testspecific.cpp
-testspecific_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testspecific_LDADD = $(top_builddir)/src/libwefts.a
 testspecific_LDFLAGS=-lpthread
 
 testqueue_SOURCES = testqueue.cpp
-testqueue_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testqueue_LDADD = $(top_builddir)/src/libwefts.a
 testqueue_LDFLAGS=-lpthread
 
 testring_SOURCES = testring.cpp
-testring_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testring_LDADD = $(top_builddir)/src/libwefts.a
 testring_LDFLAGS=-lpthread
 
 testsemaphore_SOURCES = testsemaphore.cpp
-testsemaphore_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testsemaphore_LDADD = $(top_builddir)/src/libwefts.a
 testsemaphore_LDFLAGS=-lpthread
 
 teststack_SOURCES = teststack.cpp
-teststack_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+teststack_LDADD = $(top_builddir)/src/libwefts.a
 teststack_LDFLAGS=-lpthread
 
 testxmutex_SOURCES = testxmutex.cpp
-testxmutex_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testxmutex_LDADD = $(top_builddir)/src/libwefts.a
 testxmutex_LDFLAGS=-lpthread
 
 endif
diff -ur libwefts-0.99c/testsuite/Makefile.am libwefts-0.99c-order/testsuite/Makefile.am
--- libwefts-0.99c/testsuite/Makefile.am	2004-03-28 23:37:18.000000000 +0200
+++ libwefts-0.99c-order/testsuite/Makefile.am	2005-05-28 14:05:45.987762888 +0200
@@ -18,7 +18,7 @@
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/testsuite
 noinst_HEADERS =  testsuite.h
 testsuite_SOURCES = main.cpp testsuite.cpp ts_cancel.cpp ts_cond.cpp ts_moaning.cpp ts_rmutex.cpp ts_thread.cpp ts_process.cpp
-testsuite_LDADD = $(top_builddir)/src/libwefts.a $(top_builddir)/lib/libwefts.a
+testsuite_LDADD = $(top_builddir)/src/libwefts.a
 testsuite_LDFLAGS = -lpthread
 METASOURCES = AUTO
 
