diff -ur libwefts-0.99c/include/wefts_ringbuffer.h libwefts-0.99c-m_ret/include/wefts_ringbuffer.h
--- libwefts-0.99c/include/wefts_ringbuffer.h	2004-03-28 23:45:40.000000000 +0200
+++ libwefts-0.99c-m_ret/include/wefts_ringbuffer.h	2005-05-28 13:55:15.527607400 +0200
@@ -226,9 +226,9 @@
 
       m_mutex.lock();
       if ( m_wPos > m_rPos )
-         m_ret = m_size -m_wPos + m_rPos + 1;
+         ret = m_size -m_wPos + m_rPos + 1;
       else
-         m_ret = m_rPos - m_wPos;
+         ret = m_rPos - m_wPos;
       m_mutex.unlock();
 
       return ret;
