diff -ur kde-i18n-sv-3.1/docs/kdeutils/kedit/index.docbook kde-i18n-3.1.new/sv/docs/kdeutils/kedit/index.docbook
--- kde-i18n-3.1/sv/docs/kdeutils/kedit/index.docbook	2002-10-26 17:18:48.000000000 -0400
+++ kde-i18n-3.1.new/sv/docs/kdeutils/kedit/index.docbook	2003-01-29 16:27:45.000000000 -0500
@@ -386,7 +386,7 @@
 > <row
 > <entry
 ><keycombo action="simul"
->&Shift;-<keycap
+>&Shift;<keycap
 >vänsterpil</keycap
 ></keycombo
 ></entry
@@ -398,7 +398,7 @@
 > <row
 > <entry
 ><keycombo action="simul"
->&Shift;-<keycap
+>&Shift;<keycap
 >högerpil</keycap
 ></keycombo
 ></entry
--- kde-i18n-3.1/sv/docs/kdenetwork/kit/index.docbook~	2003-02-28 01:47:32.000000000 +0100
+++ kde-i18n-3.1.new/sv/docs/kdenetwork/kit/index.docbook	2003-02-28 01:48:00.000000000 +0100
@@ -1,10 +1,10 @@
 <?xml version="1.0" ?>
 <!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kappname "&kit;">
-  <!ENTITY AOL "<trademark class='registered'
-><acronym
->AOL</acronym
-></trademark
+  <!ENTITY AOL "<acronym
+><trademark class='registered'
+>AOL</trademark
+></acronym
 >">
   <!ENTITY package "kdenetwork">
   <!ENTITY % addindex "IGNORE">
