--- gui2/gui_recode.c.orig	2008-09-22 17:29:52.000000000 +0100
+++ gui2/gui_recode.c	2008-10-08 21:19:22.000000000 +0100
@@ -285,6 +285,10 @@
     return ret;
 }
 
+#else 
+gchar *my_filename_from_utf8 (char *fname) { return fname; }
+gchar *my_filename_to_utf8 (const char *fname) { return g_strdup(fname); }
+int maybe_recode_gp_file_to_utf8 (const char *fname) { return 1; }
 #endif /* ENABLE_NLS */
 
 
