--- cli/Makefile.in.orig	2009-07-06 05:26:32.000000000 +0100
+++ cli/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -3,6 +3,7 @@
 
 CC = @CC@
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
 MAKE = @MAKE@
 INSTALL = @INSTALL@
 
@@ -51,7 +52,7 @@
 	$(CC) $(CFLAGS) -MM $< > .deps/$*.d 
 
 $(CLI): .deps $(OBJS)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJS) $(GRETLLIB) \
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $(OBJS) $(GRETLLIB) \
 	$(READLINE_LIBS)
 
 .deps:
--- doc/Makefile.in.orig	2008-09-21 21:22:59.000000000 +0100
+++ doc/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -8,12 +8,12 @@
 subdirs: $(SUBDIRS)
 
 $(SUBDIRS):
-	make -C $@
+	$(MAKE) -C $@
 
 clean:
 	for d in $(SUBDIRS) ; do \
-	make -C $$d clean ; done
+	$(MAKE) -C $$d clean ; done
 
 install: 
 	for d in $(SUBDIRS) ; do \
-	make -C $$d install ; done
+	$(MAKE) -C $$d install ; done
--- doc/tex/Makefile.in.orig	2009-06-13 16:42:16.000000000 +0100
+++ doc/tex/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -93,16 +93,16 @@
 	if [ ! -f $$f ] ; then ln -sf $(docsrc)/tex/$$f . ; fi ; done
 
 cmdtex:
-	make -C ../commands cmdtex
+	$(MAKE) -C ../commands cmdtex
 
 topiclist:
-	make -C ../commands topiclist_en
+	$(MAKE) -C ../commands topiclist_en
 
 textables:
-	make -C ../commands textables
+	$(MAKE) -C ../commands textables
 
 functex:
-	make -C ../commands functex
+	$(MAKE) -C ../commands functex
 
 date.tex:
 	date "+%B, %Y" > $@
--- doc/tex_pt/Makefile.in.orig	2009-03-28 13:37:17.000000000 +0000
+++ doc/tex_pt/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -69,16 +69,16 @@
 	if [ ! -f $$f ] ; then ln -sf $(docsrc)/tex_pt/$$f . ; fi ; done
 
 cmdtex:
-	make -C ../commands cmdtex_pt
+	$(MAKE) -C ../commands cmdtex_pt
 
 topiclist:
-	make -C ../commands topiclist_pt
+	$(MAKE) -C ../commands topiclist_pt
 
 textables_pt:
-	make -C ../commands textables_pt
+	$(MAKE) -C ../commands textables_pt
 
 functex:
-	make -C ../commands functex_pt
+	$(MAKE) -C ../commands functex_pt
 
 date.tex:
 	LANG=pt_PT date "+%B %Y" | sed 's/^[a-z]/\u&/' > $@
--- extra/Makefile.in.orig	2003-04-08 19:57:13.000000000 +0100
+++ extra/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -5,11 +5,11 @@
 VPATH = $(topsrc)/extra
 
 ifeq ($(have_gtk),2.0)
-  CFLAGS = -g -O2 -Wall `pkg-config --cflags gretl`
-  LDFLAGS = `pkg-config --libs gretl`
+  CFLAGS :=  $(CFLAGS) `pkg-config --cflags gretl`
+  LDFLAGS := $(LDFLAGS) `pkg-config --libs gretl`
 else  
-  CFLAGS = -g -O2 -Wall `gretl-config --cflags`
-  LDFLAGS = `gretl-config --libs`
+  CFLAGS :=  $(CFLAGS) `gretl-config --cflags`
+  LDFLAGS := $(LDFLAGS) `gretl-config --libs`
 endif
 
 all: simple_client
--- gui2/Makefile.in.orig	2009-06-29 15:28:31.000000000 +0100
+++ gui2/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -7,6 +7,7 @@
 libdir = @libdir@
 CC = @CC@
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
 MAKE = @MAKE@
 INSTALL = @INSTALL@
 LN = @LN_S@
@@ -68,7 +69,7 @@
 endif
 
 ifeq ($(have_gnome),2.0)
-  query_install_gnome = install-gnome
+  queryOA_install_gnome = install-gnome
   GNOMEDEF = -DDATADIR=\"$(gnome_prefix)/share\"
 endif
 
@@ -161,13 +162,13 @@
 	$(CC) $(CFLAGS) -MM $< > .deps/$*.d 
 
 $(PROG): .deps $(OBJS) $(HACK_OBJ) $(GTKSOURCEVIEW_LITE) 
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJS) $(HACK_OBJ) $(LIBS) $(INETLIB) $(CARBONLIB)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $(OBJS) $(HACK_OBJ) $(LIBS) $(INETLIB) $(CARBONLIB)
 
 $(HACK_OBJ): $(HACK_SRC) $(HACK_HDR)
 	$(CC) $(CFLAGS) $(GTK_CFLAGS) -c $<
 
 mklang: mklang.o
-	$(LIBTOOL) --mode=link $(CC) -o $@ $< ../lib/libgretl-1.0.la $(XML_LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $< ../lib/libgretl-1.0.la $(XML_LIBS)
 
 mklang.o: mklang.c $(topsrc)/lib/src/options.c $(topsrc)/lib/src/genlex.c $(topsrc)/lib/src/gretl_commands.c
 	$(CC) -c -I.. -I$(libsrc) $(XML_CFLAGS) $<
@@ -176,7 +177,7 @@
 	./mklang $(LANGOPT) > $@
 
 $(GTKSOURCEVIEW_LITE):
-	make -C gtksourceview
+	$(MAKE) -C gtksourceview
 
 .deps:
 	mkdir $@
@@ -220,7 +221,7 @@
 clean:
 	rm -f *.o $(PROG) mklang gretl.lang gretl.lang.2
 	rm -rf .libs .deps
-	make -C gtksourceview clean
+	$(MAKE) -C gtksourceview clean
 
 distclean: clean
 	rm -f Makefile debug
--- lib/Makefile.in.orig	2009-07-06 05:24:20.000000000 +0100
+++ lib/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -6,6 +6,7 @@
 bindir = @bindir@
 CC = @CC@
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
 MAKE = @MAKE@
 INSTALL = @INSTALL@
 LN = @LN_S@
@@ -236,7 +237,7 @@
 	$(CC) $(CFLAGS) -MM -MT $*.lo $< > .deps/$*.d
 
 $(LIBGRETL): .deps $(LOBJS) $(AUX_LOBJ)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(LOBJS) $(AUX_LOBJ) \
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $(LOBJS) $(AUX_LOBJ) \
 	-rpath $(prefix)/lib $(LIBS) \
 	-version-info $(CURRENT):$(REVISION):$(AGE)
 
--- Makefile.in.orig	2009-05-02 21:55:41.000000000 +0100
+++ Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -6,6 +6,7 @@
 libdir = @libdir@
 CC = @CC@
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
 MAKE = @MAKE@
 INSTALL = @INSTALL@
 LN = @LN_S@
@@ -85,16 +86,16 @@
 	$(tooldir)/makedist
 
 distclean: clean
-	make -C win32 clean
+	$(MAKE) -C win32 clean
 	rm -f config.log config.cache config.status config.h gretl.pc
 	rm -f gretl_sh gretl-config libtool Makefile
 	for d in $(ALLSUBDIRS) ; do $(MAKE) -C $$d distclean; done
 
 check:
-	make -C tests check
+	$(MAKE) -C tests check
 
 osx-dist:
-	make -C osx postinst
+	$(MAKE) -C osx postinst
 
 # dependency for parallel builds
 cli: lib
--- plugin/Makefile.in.orig	2009-06-22 15:41:31.000000000 +0100
+++ plugin/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -7,6 +7,7 @@
 
 CC = @CC@
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
 MAKE = @MAKE@
 INSTALL = @INSTALL@
 LN = @LN_S@
@@ -47,7 +48,7 @@
 
 LIBTOOL = ../libtool
 COMPILE = $(LIBTOOL) --mode=compile $(CC) -c $(CFLAGS) 
-LINK = $(LIBTOOL) --mode=link $(CC) -module -avoid-version -rpath $(plugindir)
+LINK = $(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -module -avoid-version -rpath $(plugindir)
 
 GRETLINC = -I$(topsrc)/lib/src
 GRETLLIB = ../lib/libgretl-1.0.la
--- share/Makefile.in.orig	2008-09-21 21:23:15.000000000 +0100
+++ share/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -41,14 +41,14 @@
 all: $(query_make_help)
 
 help:
-	make -C ../doc/commands help-all
+	$(MAKE) -C ../doc/commands help-all
 
 .PHONY :
 
 install: installdirs install_datafiles install_scripts install_functions \
 	install_fonts install_help
 	$(INSTALL_DATA) $(sharesrc)/gretl.stamp $(DESTDIR)$(gretldir)/gretl.stamp
-	make -C bcih install
+	$(MAKE) -C bcih install
 
 install-strip: install
 
@@ -108,7 +108,7 @@
 	$(tooldir)/mkinstalldirs $(DESTDIR)$(gretldir)/fonts
 
 clean:
-	make -C bcih clean
+	$(MAKE) -C bcih clean
 
 textclean: 
 	rm -f *.txt
@@ -118,4 +118,4 @@
 
 distclean: clean
 	rm -f Makefile
-	make -C bcih distclean
+	$(MAKE) -C bcih distclean
--- tests/Makefile.in.orig	2006-06-28 20:14:50.000000000 +0100
+++ tests/Makefile.in	2009-08-07 16:31:37.000000000 +0100
@@ -5,6 +5,7 @@
 
 CC = @CC@
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
 have_gtk = @have_gtk@
 GMP_LIBS = @GMP_LIBS@
 FLIB = @FLIB@
@@ -35,7 +36,7 @@
 all: nistcheck
 
 nistcheck: nistcheck.o $(LIBGRETL)
-	../libtool --mode=link $(CC) -o $@ $< $(LIBGRETL) $(XML_LIBS)
+	../libtool --mode=link $(CC) $(LDFLAGS) -o $@ $< $(LIBGRETL) $(XML_LIBS)
 
 nistcheck.o: nistcheck.c
 	$(CC) $(CFLAGS) $(XML_CFLAGS) -c $<
@@ -47,10 +48,10 @@
 	./mtrand && ./diehard
 
 diehard: diehard.c
-	$(CC) -I$(topsrc)/lib/src -o $@ $< $(FLIB) -lm
+	$(CC) -I$(topsrc)/lib/src $(LDFLAGS) -o $@ $< $(FLIB) -lm
 
 mtrand: mtrand.o
-	../libtool --mode=link $(CC) -o $@ $< $(RAND_LIBS)
+	../libtool --mode=link $(CC) $(LDFLAGS) -o $@ $< $(RAND_LIBS)
 
 .PHONY :
 
