diff -Naur less-382.orig/multi.c less-382/multi.c
--- less-382.orig/multi.c	2008-11-02 20:17:16.000000000 +0900
+++ less-382/multi.c	2008-11-02 20:20:07.000000000 +0900
@@ -279,7 +279,6 @@
 	case 0x70: return 4;	/* or more bytes */
 	}
     }
-    assert(0);
     return (0);
 }
 
@@ -1337,8 +1336,6 @@
 	default:   goto wrongone;
 	}
 	break;
-    default:
-	assert(0);
     }
     if (mp->eseq == NOESC) {
 	fix_status_for_escape_sequence(mp);
@@ -1857,8 +1854,6 @@
 		strcpy(buf, "96N( )");
 		buf[4] = CS2FT(mp->icharset);
 		break;
-	default:
-		assert(0);
 	}
 	if (CS2IRR(mp->icharset) > 0)
 	{
@@ -2078,7 +2073,6 @@
 		return (nullcvbuffer);
 	} else
 	{
-		assert(0);
 		cvindex = 0;
 	}
 
@@ -2171,7 +2165,6 @@
 		cvindex = 0;
 		return (nullcvbuffer);
 	}
-	assert(0);
 	cvindex = 0;
 	return (cvbuffer);
 }
@@ -2270,7 +2263,6 @@
 		cvindex = 0;
 		return (nullcvbuffer);
 	}
-	assert(0);
 	cvindex = 0;
 	return (cvbuffer);
 }
@@ -2296,7 +2288,6 @@
 
 	cs = CS2CHARSET(cs);
 
-	assert(0);
 	if (cs == ASCII || cs == JISX0201ROMAN)
 	{
 		assert(cvindex == 1);
@@ -2324,7 +2315,6 @@
 		cvindex = 0;
 		return (nullcvbuffer);
 	}
-	assert(0);
 	cvindex = 0;
 	return (cvbuffer);
 }
@@ -2390,7 +2380,6 @@
 	case TYPE_96N_CHARSET:
 		return (2);
 	default:
-		assert(0);
 		return (0);
 	}
 }
@@ -2410,7 +2399,7 @@
 	case 4: p = "iso8"; mp->io.inputr = ESISO8; break;
 	case 5: p = "noconv"; mp->io.inputr = ESNOCONV; break;
 	case 6: p = "none"; mp->io.inputr = ESNONE; break;
-	default: assert(0); break;
+	default: break;
 	}
 	init_priority(mp);
 	return (p);
