diff -burN aqualung-0.9beta9.1-orig/configure.ac aqualung-0.9beta9.1/configure.ac
--- aqualung-0.9beta9.1-orig/configure.ac	2008-02-10 13:14:31.000000000 +0100
+++ aqualung-0.9beta9.1/configure.ac	2008-04-15 03:16:16.000000000 +0200
@@ -562,6 +562,8 @@
 	if test "$lavc" = "detect"; then
 		lavc=$avc_lib
 	fi
+
+	AC_CHECK_HEADERS(libavcodec/avcodec.h)
 fi
 
 
diff -burN aqualung-0.9beta9.1-orig/src/decoder/dec_lavc.h aqualung-0.9beta9.1/src/decoder/dec_lavc.h
--- aqualung-0.9beta9.1-orig/src/decoder/dec_lavc.h	2007-04-21 09:57:14.000000000 +0200
+++ aqualung-0.9beta9.1/src/decoder/dec_lavc.h	2008-04-15 04:02:41.000000000 +0200
@@ -23,8 +23,13 @@
 #define _DEC_LAVC_H
 
 #ifdef HAVE_LAVC
+#if HAVE_LIBAVCODEC_AVCODEC_H
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
+#else
 #include <ffmpeg/avcodec.h>
 #include <ffmpeg/avformat.h>
+#endif
 #endif /* HAVE_LAVC */
 
 #include "file_decoder.h"
