diff --git a/quazip/ioapi.h b/quazip/ioapi.h
index 716dd4b..a114915 100644
--- a/quazip/ioapi.h
+++ b/quazip/ioapi.h
@@ -37,13 +37,13 @@
 extern "C" {
 #endif
 
-typedef voidpf (ZCALLBACK *open_file_func) OF((voidpf opaque, voidpf file, int mode));
-typedef uLong  (ZCALLBACK *read_file_func) OF((voidpf opaque, voidpf stream, void* buf, uLong size));
-typedef uLong  (ZCALLBACK *write_file_func) OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
-typedef uLong   (ZCALLBACK *tell_file_func) OF((voidpf opaque, voidpf stream));
-typedef int   (ZCALLBACK *seek_file_func) OF((voidpf opaque, voidpf stream, uLong offset, int origin));
-typedef int    (ZCALLBACK *close_file_func) OF((voidpf opaque, voidpf stream));
-typedef int    (ZCALLBACK *testerror_file_func) OF((voidpf opaque, voidpf stream));
+typedef voidpf (ZCALLBACK *open_file_func) _Z_OF((voidpf opaque, voidpf file, int mode));
+typedef uLong  (ZCALLBACK *read_file_func) _Z_OF((voidpf opaque, voidpf stream, void* buf, uLong size));
+typedef uLong  (ZCALLBACK *write_file_func) _Z_OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
+typedef uLong   (ZCALLBACK *tell_file_func) _Z_OF((voidpf opaque, voidpf stream));
+typedef int   (ZCALLBACK *seek_file_func) _Z_OF((voidpf opaque, voidpf stream, uLong offset, int origin));
+typedef int    (ZCALLBACK *close_file_func) _Z_OF((voidpf opaque, voidpf stream));
+typedef int    (ZCALLBACK *testerror_file_func) _Z_OF((voidpf opaque, voidpf stream));
 
 typedef struct zlib_filefunc_def_s
 {
@@ -59,7 +59,7 @@ typedef struct zlib_filefunc_def_s
 
 
 
-void fill_qiodevice_filefunc OF((zlib_filefunc_def* pzlib_filefunc_def));
+void fill_qiodevice_filefunc _Z_OF((zlib_filefunc_def* pzlib_filefunc_def));
 
 #define ZREAD(filefunc,filestream,buf,size) ((*((filefunc).zread_file))((filefunc).opaque,filestream,buf,size))
 #define ZWRITE(filefunc,filestream,buf,size) ((*((filefunc).zwrite_file))((filefunc).opaque,filestream,buf,size))
