diff -uNr linux-2.6.14.3-pa0.orig/arch/parisc/mm/init.c linux-2.6.14.3-pa0/arch/parisc/mm/init.c
--- linux-2.6.14.3-pa0.orig/arch/parisc/mm/init.c	2005-12-05 13:07:38.000000000 +0100
+++ linux-2.6.14.3-pa0/arch/parisc/mm/init.c	2005-12-05 13:08:55.000000000 +0100
@@ -300,6 +300,13 @@
 			max_pfn = start_pfn + npages;
 	}
 
+	/* IOMMU is always used to access "high mem" on those boxes
+	 * that can support enough mem that a PCI device couldn't
+	 * directly DMA to any physical addresses.
+	 * ISA DMA support will need to revisit this.
+	 */
+	max_low_pfn = max_pfn;
+
 	if ((bootmap_pfn - bootmap_start_pfn) != bootmap_pages) {
 		printk(KERN_WARNING "WARNING! bootmap sizing is messed up!\n");
 		BUG();
