diff -u autossh.c autossh.c-jedi 
--- autossh.c  2003-02-14 00:47:46.000000000 +0100 
+++ autossh.c-jedi     2003-03-01 14:42:03.000000000 +0100 
@@ -955,6 +955,7 @@ 
 { 
       int sock; 
       struct addrinfo *res; 
+      int on = 1; 
 
       /* 
        * Unlike conn_remote, we don't need to cache the 
@@ -967,6 +968,10 @@ 
           res->ai_protocol)) == -1) 
               xerrlog(LOG_ERR, "socket: %s", strerror(errno)); 
 
+      if (setsockopt(sock, SOL_SOCKET, SO_REUSEADDR, 
+                       (char *) &on, sizeof on) != 0) { 
+              xerrlog(LOG_ERR, "socket: %s", strerror(errno)); 
+       } 
       if (bind(sock, (struct sockaddr *)res->ai_addr, 
           res->ai_addrlen) == -1) 
               xerrlog(LOG_ERR, "bind: %s", strerror(errno)); 
