--- cmn/game.cpp.old	2006-10-15 23:42:45.000000000 +0200
+++ cmn/game.cpp	2006-10-15 23:43:05.000000000 +0200
@@ -794,7 +794,7 @@
 
   // Just for tracing in the debugger.
   char* version = Utils::get_OS_info();
-  delete version;
+  delete [] version;
 
   noUi = False;
   noNewLevel = False;
@@ -1057,7 +1057,7 @@
 
 Game::~Game() {
   if (oneItem) {
-    delete oneItem;
+    delete [] oneItem;
   }
 
   delete ui;
@@ -2252,7 +2252,6 @@
               if (keysym != NoSymbol)
                 right[n][which] = keysym;
             }
-            delete option;
 	      }
 	      
 	      { // Left Keys.
@@ -2271,7 +2271,6 @@
               if (keysym != NoSymbol)
                 left[n][which] = keysym;
             }
-            delete option;
 	      }
 	    }
 
--- x11/ui.cpp.old	2006-10-16 00:15:14.000000000 +0200
+++ x11/ui.cpp	2006-10-16 00:15:21.000000000 +0200
@@ -867,7 +867,7 @@
       char *msg;
       if (msg = locator->message_deq()) {
         set_message(msg);
-        delete msg;
+        delete [] msg;
         messageTimer.set(MESSAGE_TIME);
       }
     }
