diff -Naur ccmath-2.2.1.orig/makelibs.sh ccmath-2.2.1/makelibs.sh
--- ccmath-2.2.1.orig/makelibs.sh	2001-11-08 16:57:07.000000000 +0100
+++ ccmath-2.2.1/makelibs.sh	2004-10-19 23:33:36.745735608 +0200
@@ -16,23 +16,23 @@
   do
     cd $MDR/$dr
     echo `pwd`
-    cc -c -O3 *.c
+    cc -fPIC -c -O3 *.c
     mv *.o $LSOD
   done
 
 # Compile corrected svd QR support without optimizing
 #     The GNU optimizer destroys the fix!
 cd $MDR/matrix
-cc -c qrb*.c
+cc -fPIC -c qrb*.c
 mv *.o $LSOD
 cd $MDR
 
 if [ $F = "y" ]
   then cd $MDR/matrix
-    cc -c -O3 solv.s
+    cc -fPIC -c -O3 solv.s
     mv *.o $LSOD
     cd $MDR/simu
-    cc -c -O3 *.s
+    cc -fPIC -c -O3 *.s
     mv *.o $LSOD
 fi
 cd $LSOD
