--- main.cc	2003-07-20 08:38:56.000000000 +0900
+++ main.cc.modified	2003-07-20 08:39:05.000000000 +0900
@@ -1019,8 +1019,8 @@
 	window = glade_xml_get_widget (xml, "MainWindow");
 	
 	gnome_config_push_prefix("=.sbconf=/General/");
-	emupath = gnome_config_get_string("emupath=/usr/local/bin/");
-	dsppath = gnome_config_get_string("effectsdir=/usr/local/share/emu10k1/");
+	emupath = gnome_config_get_string("emupath=/usr/bin/");
+	dsppath = gnome_config_get_string("effectsdir=/usr/share/emu10k1/");
 
 	// configue the about box
 	gchar** command=new char*[3];

