=== modified file 'util/grub.d/10_linux.in'
--- util/grub.d/10_linux.in	2012-03-10 14:21:25 +0000
+++ util/grub.d/10_linux.in	2012-04-11 19:26:45 +0000
@@ -150,7 +150,8 @@
 EOF
 }
 
-case x`uname -m` in
+machine=`uname -m`
+case "x$machine" in
     xi?86 | xx86_64)
 	list=`for i in /boot/vmlinuz-* /vmlinuz-* /boot/kernel-* ; do
                   if grub_file_is_not_garbage "$i" ; then echo -n "$i " ; fi
@@ -161,6 +162,15 @@
 	     done` ;;
 esac
 
+if test "x${GENKERNEL_ARCH-unset}" = xunset; then
+  case "$machine" in
+    i?86) GENKERNEL_ARCH="x86" ;;
+    mips|mips64) GENKERNEL_ARCH="mips" ;;
+    arm*) GENKERNEL_ARCH="arm" ;;
+    *) GENKERNEL_ARCH="$machine" ;;
+  esac
+fi
+
 prepare_boot_cache=
 prepare_root_cache=
 boot_device_id=
@@ -186,8 +196,8 @@
 	   "initrd-${version}" "initramfs-${version}.img" \
 	   "initrd.img-${alt_version}" "initrd-${alt_version}.img" \
 	   "initrd-${alt_version}" "initramfs-${alt_version}.img" \
-	   "initramfs-genkernel-${version}" \
-	   "initramfs-genkernel-${alt_version}"; do
+	   "initramfs-genkernel-${GENKERNEL_ARCH}-${version}" \
+	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}"; do
     if test -e "${dirname}/${i}" ; then
       initrd="$i"
       break

