diff -Naur grub-0.95.20040823.old/grub/Makefile.am grub-0.95.20040823/grub/Makefile.am
--- grub-0.95.20040823.old/grub/Makefile.am	2004-08-24 00:21:31.000000000 +0100
+++ grub-0.95.20040823/grub/Makefile.am	2004-11-21 22:05:06.921259824 +0000
@@ -13,7 +13,7 @@
 	$(SERIAL_FLAGS) -I$(top_srcdir)/stage2 \
 	-I$(top_srcdir)/stage1 -I$(top_srcdir)/lib
 
-AM_CFLAGS = $(GRUB_CFLAGS) -fwritable-strings
+AM_CFLAGS = $(GRUB_CFLAGS)
 
 grub_SOURCES = main.c asmstub.c
 grub_LDADD = ../stage2/libgrub.a  ../lib/libcommon.a $(GRUB_LIBS)
diff -Naur grub-0.95.20040823.old/grub/Makefile.in grub-0.95.20040823/grub/Makefile.in
--- grub-0.95.20040823.old/grub/Makefile.in	2004-08-24 00:21:31.000000000 +0100
+++ grub-0.95.20040823/grub/Makefile.in	2004-11-21 22:05:14.233148248 +0000
@@ -185,7 +185,7 @@
 	$(SERIAL_FLAGS) -I$(top_srcdir)/stage2 \
 	-I$(top_srcdir)/stage1 -I$(top_srcdir)/lib
 
-AM_CFLAGS = $(GRUB_CFLAGS) -fwritable-strings
+AM_CFLAGS = $(GRUB_CFLAGS)
 grub_SOURCES = main.c asmstub.c
 grub_LDADD = ../stage2/libgrub.a  ../lib/libcommon.a $(GRUB_LIBS)
 all: all-am
diff -Naur grub-0.95.20040823.old/stage2/Makefile.am grub-0.95.20040823/stage2/Makefile.am
--- grub-0.95.20040823.old/stage2/Makefile.am	2004-11-21 22:03:48.225223000 +0000
+++ grub-0.95.20040823/stage2/Makefile.am	2004-11-21 22:05:28.105039400 +0000
@@ -24,8 +24,7 @@
 	-DGRUB_UTIL=1 -DFSYS_EXT2FS=1 -DFSYS_FAT=1 -DFSYS_FFS=1 \
 	-DFSYS_ISO9660=1 -DFSYS_JFS=1 -DFSYS_MINIX=1 -DFSYS_REISERFS=1 \
 	-DFSYS_UFS2=1 -DFSYS_VSTAFS=1 -DFSYS_XFS=1 \
-	-DUSE_MD5_PASSWORDS=1 -DSUPPORT_SERIAL=1 -DSUPPORT_HERCULES=1 \
-	-fwritable-strings
+	-DUSE_MD5_PASSWORDS=1 -DSUPPORT_SERIAL=1 -DSUPPORT_HERCULES=1
 
 # Stage 2 and Stage 1.5's.
 pkglibdir = $(libdir)/$(PACKAGE)/$(host_cpu)-$(host_vendor)
diff -Naur grub-0.95.20040823.old/stage2/Makefile.in grub-0.95.20040823/stage2/Makefile.in
--- grub-0.95.20040823.old/stage2/Makefile.in	2004-08-24 00:21:32.000000000 +0100
+++ grub-0.95.20040823/stage2/Makefile.in	2004-11-21 22:05:35.523911560 +0000
@@ -579,8 +579,7 @@
 	-DGRUB_UTIL=1 -DFSYS_EXT2FS=1 -DFSYS_FAT=1 -DFSYS_FFS=1 \
 	-DFSYS_ISO9660=1 -DFSYS_JFS=1 -DFSYS_MINIX=1 -DFSYS_REISERFS=1 \
 	-DFSYS_UFS2=1 -DFSYS_VSTAFS=1 -DFSYS_XFS=1 \
-	-DUSE_MD5_PASSWORDS=1 -DSUPPORT_SERIAL=1 -DSUPPORT_HERCULES=1 \
-	-fwritable-strings
+	-DUSE_MD5_PASSWORDS=1 -DSUPPORT_SERIAL=1 -DSUPPORT_HERCULES=1
 
 @DISKLESS_SUPPORT_FALSE@pkglib_DATA = stage2 stage2_eltorito e2fs_stage1_5 fat_stage1_5 \
 @DISKLESS_SUPPORT_FALSE@	ffs_stage1_5 iso9660_stage1_5 jfs_stage1_5 minix_stage1_5 \
