--- util/grub.d/10_linux.in
+++ util/grub.d/10_linux.in
@@ -102,7 +102,7 @@
 EOF
 }
 
-list=`for i in /boot/vmlinu[zx]-* /vmlinu[zx]-* ; do
+list=`for i in /boot/kernel-* /boot/vmlinu[zx]-* /vmlinu[zx]-* ; do
         if grub_file_is_not_garbage "$i" ; then echo -n "$i " ; fi
       done`
 prepare_boot_cache=
@@ -120,6 +120,7 @@
   initrd=
   for i in "initrd.img-${version}" "initrd-${version}.img" \
 	   "initrd-${version}" "initramfs-${version}.img" \
+	   "initramfs-genkernel-${version}" "initramfs-genkernel-${alt_version}" \
 	   "initrd.img-${alt_version}" "initrd-${alt_version}.img" \
 	   "initrd-${alt_version}" "initramfs-${alt_version}.img"; do
     if test -e "${dirname}/${i}" ; then
