diff -up rhpxl-0.51/rhpxl/xserver.py.jx rhpxl-0.51/rhpxl/xserver.py
--- rhpxl-0.51/rhpxl/xserver.py.jx	2008-03-10 15:41:00.000000000 -0400
+++ rhpxl-0.51/rhpxl/xserver.py	2008-03-14 12:11:11.000000000 -0400
@@ -52,7 +52,7 @@ class XServer:
         self.resolution = "800x600"
         self.root = "/"
         self.serverflags = ["vt6", "-config", "/tmp/XConfig.test",
-                            "-extension", "Composite", "-s", "1440", "-dpms",
+                            "-s", "1440", "-dpms",
                             "-v", "-ac", "-nolisten", "tcp"]
         self.videohw = None
 
