Index: mod_auth_kerb-5.0-rc6/Makefile.in
===================================================================
--- mod_auth_kerb-5.0-rc6.orig/Makefile.in
+++ mod_auth_kerb-5.0-rc6/Makefile.in
@@ -7,7 +7,7 @@ LIB_resolv = @LIB_resolv@
 SPNEGO_SRCS = @SPNEGO_SRCS@
 
 CPPFLAGS = -I. -Ispnegokrb5 $(KRB5_CPPFLAGS) $(KRB4_CPPFLAGS) $(DEFS)
-LDFLAGS = $(KRB5_LDFLAGS) $(KRB4_LDFLAGS) $(LIB_resolv)
+LDFLAGS = $(KRB5_LDFLAGS) $(KRB4_LDFLAGS) $(LIB_resolv) -Wl,-export-symbols-regex -Wl,auth_kerb_module
 CFLAGS =
 
 all: src/mod_auth_kerb.so
