--- src/modules/kerberos/mod_auth_kerb.c.orig	2002-09-05 08:21:45.000000000 -0400
+++ src/modules/kerberos/mod_auth_kerb.c	2002-09-05 08:24:25.000000000 -0400
@@ -1932,11 +1932,18 @@
     return rec;
 }
 
+static int kerbauth_init_handler(apr_pool_t *p, apr_pool_t *plog, 
+                                 apr_pool_t *ptemp, server_rec *s)
+{
+    ap_add_version_component(p, "auth_kerberos/4.11");
+    return OK;
+}
 
 /* register the functions in the correct places */
 #ifdef APXS2
 static void register_hooks(apr_pool_t *p)
 {
+        ap_hook_post_config(kerbauth_init_handler, NULL, NULL, APR_HOOK_MIDDLE);
 	ap_hook_check_user_id(kerb_authenticate_user, NULL, NULL, APR_HOOK_MIDDLE);
 #ifdef HESIOD
 	ap_hook_access_checker(check_user_access, NULL, NULL, APR_HOOK_MIDDLE);
