diff -Nrup cvs-1.11.18.orig/src/login.c cvs-1.11.18/src/login.c
--- cvs-1.11.18.orig/src/login.c	2004-03-29 16:56:44.000000000 +0000
+++ cvs-1.11.18/src/login.c	2005-03-24 17:29:13.000000000 +0000
@@ -114,7 +114,7 @@ password_entry_parseline (cvsroot_canoni
 
 	if (isspace(*(linebuf + 1)))
 	    /* special case since strtoul ignores leading white space */
-	    entry_version = 0;
+	    q = linebuf + 1;
 	else
 	    entry_version = strtoul (linebuf + 1, &q, 10);
 
diff -Nrup cvs-1.11.18.orig/src/patch.c cvs-1.11.18/src/patch.c
--- cvs-1.11.18.orig/src/patch.c	2004-04-02 19:25:32.000000000 +0000
+++ cvs-1.11.18/src/patch.c	2005-03-24 17:30:42.000000000 +0000
@@ -385,6 +385,7 @@ patch_fileproc (callerdat, finfo)
     struct utimbuf t;
     char *vers_tag, *vers_head;
     char *rcs = NULL;
+    char *rcs_orig = NULL;
     RCSNode *rcsfile;
     FILE *fp1, *fp2, *fp3;
     int ret = 0;
@@ -415,7 +416,7 @@ patch_fileproc (callerdat, finfo)
     if ((rcsfile->flags & VALID) && (rcsfile->flags & INATTIC))
 	isattic = 1;
 
-    rcs = xmalloc (strlen (finfo->file) + sizeof (RCSEXT) + 5);
+    rcs_orig = rcs = xmalloc (strlen (finfo->file) + sizeof (RCSEXT) + 5);
     (void) sprintf (rcs, "%s%s", finfo->file, RCSEXT);
 
     /* if vers_head is NULL, may have been removed from the release */
@@ -757,8 +758,8 @@ failed to read diff file header %s for %
 	free (vers_tag);
     if (vers_head != NULL)
 	free (vers_head);
-    if (rcs != NULL)
-	free (rcs);
+    if (rcs_orig)
+	free (rcs_orig);
     return ret;
 }
 
diff -Nrup cvs-1.11.18.orig/src/rcs.c cvs-1.11.18/src/rcs.c
--- cvs-1.11.18.orig/src/rcs.c	2004-10-29 19:55:14.000000000 +0000
+++ cvs-1.11.18/src/rcs.c	2005-03-24 17:29:13.000000000 +0000
@@ -3024,8 +3024,7 @@ RCS_getdate (rcs, date, force_tag_match)
     if (retval != NULL)
 	return (retval);
 
-    if (!force_tag_match ||
-	(vers != NULL && RCS_datecmp (vers->date, date) <= 0))
+    if (vers && (!force_tag_match || RCS_datecmp (vers->date, date) <= 0))
 	return xstrdup (vers->version);
     else
 	return NULL;
@@ -4112,7 +4111,7 @@ RCS_checkout (rcs, workfile, rev, nameta
     size_t len;
     int free_value = 0;
     char *log = NULL;
-    size_t loglen;
+    size_t loglen = 0;
     Node *vp = NULL;
 #ifdef PRESERVE_PERMISSIONS_SUPPORT
     uid_t rcs_owner = (uid_t) -1;
@@ -7437,7 +7436,7 @@ RCS_deltas (rcs, fp, rcsbuf, version, op
 
 		for (ln = 0; ln < headlines.nlines; ++ln)
 		{
-		    char buf[80];
+		    char *buf;
 		    /* Period which separates year from month in date.  */
 		    char *ym;
 		    /* Period which separates month from day in date.  */
@@ -7448,10 +7447,12 @@ RCS_deltas (rcs, fp, rcsbuf, version, op
 		    if (prvers == NULL)
 			prvers = vers;
 
+		    buf = xmalloc (strlen (prvers->version) + 24);
 		    sprintf (buf, "%-12s (%-8.8s ",
 			     prvers->version,
 			     prvers->author);
 		    cvs_output (buf, 0);
+		    free (buf);
 
 		    /* Now output the date.  */
 		    ym = strchr (prvers->date, '.');
