--- rioutil-1.5.0b/librioutil/mp3.c	2005-03-23 11:33:43.000000000 +1100
+++ rioutil-1.5.0b/librioutil/mp3.c.good	2005-04-22 17:07:38.000000000 +1000
@@ -249,7 +249,7 @@
     return -errno;
 
   mp3->file_size = mp3->data_size = statinfo.st_size;
-  mp3->mod_date  = statinfo.st_mtimespec.tv_sec;
+  mp3->mod_date  = statinfo.st_mtime;
 
   mp3->fh = fopen (file_name, "r");
   if (mp3->fh == NULL) 
--- rioutil-1.5.0b/aclocal.m4.orig	2005-04-22 18:36:39.000000000 +1000
+++ rioutil-1.5.0b/aclocal.m4	2005-04-22 18:39:26.000000000 +1000
@@ -175,20 +175,20 @@
 test -z "$ac_objext" && ac_objext=o
 
 # Determine commands to create old-style static archives.
-old_archive_cmds='$AR $AR_FLAGS $oldlib$oldobjs$old_deplibs'
+old_archive_cmds='$AR $AR_FLAGS $oldlib $oldobjs $old_deplibs'
 old_postinstall_cmds='chmod 644 $oldlib'
 old_postuninstall_cmds=
 
 if test -n "$RANLIB"; then
   case $host_os in
   openbsd*)
-    old_postinstall_cmds="\$RANLIB -t \$oldlib~$old_postinstall_cmds"
+    old_postinstall_cmds="\$RANLIB -t \$oldlib; $old_postinstall_cmds"
     ;;
   *)
-    old_postinstall_cmds="\$RANLIB \$oldlib~$old_postinstall_cmds"
+    old_postinstall_cmds="\$RANLIB \$oldlib; $old_postinstall_cmds"
     ;;
   esac
-  old_archive_cmds="$old_archive_cmds~\$RANLIB \$oldlib"
+  old_archive_cmds="$old_archive_cmds;$RANLIB \$oldlib"
 fi
 
 cc_basename=`$echo X"$compiler" | $Xsed -e 's%^.*/%%'`
