diff -ruN TiMidity++-2.12.0-pre1.orig/configure.in TiMidity++-2.12.0-pre1/configure.in
--- TiMidity++-2.12.0-pre1.orig/configure.in	2002-01-17 18:51:32.000000000 -0500
+++ TiMidity++-2.12.0-pre1/configure.in	2003-11-16 17:43:19.354413704 -0500
@@ -1268,6 +1268,7 @@
   ])
 
 dnl UMP configuration
+AM_CONDITIONAL(ENABLE_PLUGIN, false)
 CONFIG_INTERFACE(ump, PLUGIN, p,
   [  --enable-ump            UMP configuration               (default is no)],
   [ if test ! -d ump; then
