diff -ur libprojectM-1.01.orig/CMakeLists.txt libprojectM-1.01/CMakeLists.txt
--- libprojectM-1.01.orig/CMakeLists.txt	2007-10-10 17:47:14.000000000 +0300
+++ libprojectM-1.01/CMakeLists.txt	2007-11-02 16:05:18.000000000 +0200
@@ -6,6 +6,7 @@
 BuiltinFuncs.cpp BuiltinParams.cpp Renderer.cpp 
 PresetLoader.cpp  PresetChooser.cpp PresetFrameIO.cpp PresetMerge.cpp
 ConfigFile.cpp  IdlePreset.cpp TextureManager.cpp MoodBar.cpp image_DXT.c image_helper.c SOIL.c stb_image.c  )
+SET_TARGET_PROPERTIES(projectM PROPERTIES VERSION 1.0.0 SOVERSION 1)
 
 OPTION(USE_FTGL "Use FTGL for on-screen fonts" ON)
 
