Index: bin/repoman
===================================================================
--- bin/repoman	(revision 3970)
+++ bin/repoman	(revision 3971)
@@ -843,7 +843,8 @@
 
 	if mymode == "commit":
 		repoman_settings["O"] = checkdir
-		portage.digestgen([], repoman_settings, manifestonly=1, myportdb=portage.portdb)
+		portage.digestgen(
+			[], repoman_settings, manifestonly=1, myportdb=portdb)
 
 	mf = Manifest(checkdir, repoman_settings["DISTDIR"])
 	mydigests=mf.getTypeDigests("DIST")
