Index: bin/emerge
===================================================================
--- bin/emerge	(revision 5278)
+++ bin/emerge	(revision 5279)
@@ -961,7 +961,7 @@
 			self.digraph.addnode(jbigkey, myparent,
 				priority=priority)
 
-		if addme:
+		if mybigkey[-1] == "merge":
 			self.mydbapi[myroot].cpv_inject(mykey,
 				metadata=dict(izip(self._mydbapi_keys,
 				mydbapi.aux_get(mykey, self._mydbapi_keys))))
