Index: bin/emerge
===================================================================
--- bin/emerge	(revision 3672)
+++ bin/emerge	(revision 3674)
@@ -2816,7 +2816,7 @@
 	mybestpv=portage.portdb.xmatch("bestmatch-visible","sys-apps/portage")
 	mypvs=portage.best(portage.db[portage.root]["vartree"].dbapi.match("sys-apps/portage"))
 
-	chk_updated_cfg_files()
+	chk_updated_cfg_files("/", settings.get("CONFIG_PROTECT","").split())
 
 	if myaction != "metadata":
 		if os.access(portage.USER_CONFIG_PATH + "/bin/post_sync", os.X_OK):
@@ -3089,7 +3089,7 @@
 			if "cvs_id_string" in dir(module):
 				print "%s: %s" % (str(x), str(module.cvs_id_string))
 
-def action_search(myopts, myfiles, spinner):
+def action_search(settings, vartree, myopts, myfiles, spinner):
 	if not myfiles:
 		print "emerge: no search terms provided."
 	else:
@@ -3795,7 +3795,8 @@
 	
 	# SEARCH action
 	elif "search"==myaction:
-		action_search(myopts, myfiles)
+		action_search(settings, trees["/"]["vartree"],
+			myopts, myfiles, spinner)
 	elif "unmerge"==myaction or "prune"==myaction or "clean"==myaction:
 		vartree = trees[settings["ROOT"]]["vartree"]
 		if 1 == unmerge(settings, myopts, vartree, myaction, myfiles,
