diff -uNr portage-2.1_pre3/pym/output.py portage-2.1_pre3-r1/pym/output.py
--- portage-2.1_pre3/pym/output.py	2005-12-31 15:23:23.000000000 +0900
+++ portage-2.1_pre3-r1/pym/output.py	2006-01-01 14:44:53.000000000 +0900
@@ -102,7 +102,7 @@
 				sys.stderr.flush()
 				break
 
-prompt_command = os.getenv("PROMPT_COMMAND", 'echo -ne "\033]0;${USER}@${HOSTNAME%%.*}:${PWD/$HOME/~}\007"')
+prompt_command = os.getenv("PROMPT_COMMAND", 'echo -ne "${USER}@${HOSTNAME%%.*}:${PWD/$HOME/~}"')
 default_xterm_title = commands.getoutput(prompt_command)
 del prompt_command
 
diff -uNr portage-2.1_pre3/pym/portage.py portage-2.1_pre3-r1/pym/portage.py
--- portage-2.1_pre3/pym/portage.py	2005-12-31 15:24:24.000000000 +0900
+++ portage-2.1_pre3-r1/pym/portage.py	2006-01-01 14:45:01.000000000 +0900
@@ -4,7 +4,7 @@
 # $Id: /var/cvsroot/gentoo-src/portage/pym/portage.py,v 1.524.2.76 2005/05/29 12:40:08 jstubbs Exp $
 
 
-VERSION="2.1_pre3"
+VERSION="2.1_pre3-r1"
 
 # ===========================================================================
 # START OF IMPORTS -- START OF IMPORTS -- START OF IMPORTS -- START OF IMPORT
