diff -Nur mail-notification-4.1.dfsg.1/src/mn-evolution-folder-tree-control.gob mail-notification-4.1.dfsg.1.new/src/mn-evolution-folder-tree-control.gob
--- mail-notification-4.1.dfsg.1/src/mn-evolution-folder-tree-control.gob	2007-06-26 13:36:57.000000000 +0200
+++ mail-notification-4.1.dfsg.1.new/src/mn-evolution-folder-tree-control.gob	2007-08-15 16:54:28.000000000 +0200
@@ -126,7 +126,7 @@
     switch (arg_id)
       {
       case PROP_SELECTED_URI:
-	em_folder_tree_set_selected(tree, BONOBO_ARG_GET_STRING(arg));
+	em_folder_tree_set_selected(tree, BONOBO_ARG_GET_STRING(arg), TRUE);
 	break;
 
       default:
