--- Makefile.orig	2007-07-29 15:19:05.000000000 +0300
+++ Makefile	2007-07-29 15:23:51.000000000 +0300
@@ -33,7 +33,7 @@
 POST_UNINSTALL = :
 build_triplet = i686-pc-linux-gnu
 host_triplet = i686-pc-linux-gnu
-bin_PROGRAMS = sfxload$(EXEEXT) asfxload$(EXEEXT) aweset$(EXEEXT) \
+bin_PROGRAMS = sfxload$(EXEEXT) aweset$(EXEEXT) \
 	gusload$(EXEEXT) setfx$(EXEEXT) sf2text$(EXEEXT) \
 	text2sf$(EXEEXT) sfxtest$(EXEEXT)
 subdir = .
@@ -53,9 +53,6 @@
 am__installdirs = "$(DESTDIR)$(bindir)" "$(DESTDIR)$(man1dir)"
 binPROGRAMS_INSTALL = $(INSTALL_PROGRAM)
 PROGRAMS = $(bin_PROGRAMS)
-am_asfxload_OBJECTS = asfxload.$(OBJEXT) alsa.$(OBJEXT)
-asfxload_OBJECTS = $(am_asfxload_OBJECTS)
-asfxload_DEPENDENCIES = awelib/libawe.a
 am_aweset_OBJECTS = aweset.$(OBJEXT) seq.$(OBJEXT)
 aweset_OBJECTS = $(am_aweset_OBJECTS)
 aweset_LDADD = $(LDADD)
@@ -96,10 +93,10 @@
 LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
 	$(LDFLAGS) -o $@
-SOURCES = $(asfxload_SOURCES) $(aweset_SOURCES) $(gusload_SOURCES) \
+SOURCES = $(aweset_SOURCES) $(gusload_SOURCES) \
 	$(setfx_SOURCES) $(sf2text_SOURCES) $(sfxload_SOURCES) \
 	$(sfxtest_SOURCES) $(text2sf_SOURCES)
-DIST_SOURCES = $(asfxload_SOURCES) $(aweset_SOURCES) \
+DIST_SOURCES = $(aweset_SOURCES) \
 	$(gusload_SOURCES) $(setfx_SOURCES) $(sf2text_SOURCES) \
 	$(sfxload_SOURCES) $(sfxtest_SOURCES) $(text2sf_SOURCES)
 RECURSIVE_TARGETS = all-recursive check-recursive dvi-recursive \
@@ -166,7 +163,7 @@
 INSTALL_STRIP_PROGRAM = $(install_sh) -c -s
 LDFLAGS = -Wl,-O1,--as-needed
 LIBOBJS = 
-LIBS = 
+LIBS = -lm -ldl -lpthread
 LIBTOOL = $(SHELL) $(top_builddir)/libtool
 LN_S = ln -s
 LTLIBOBJS = 
@@ -240,8 +237,6 @@
 SUBDIRS = awelib include samples etc
 LDADD = awelib/libawe.a
 INCLUDES = -Iinclude
-asfxload_SOURCES = asfxload.c alsa.c
-asfxload_LDADD = awelib/libawe.a 
 sfxload_SOURCES = sfxload.c seq.c
 aweset_SOURCES = aweset.c seq.c
 gusload_SOURCES = gusload.c seq.c
@@ -317,9 +312,6 @@
 	  echo " rm -f $$p $$f"; \
 	  rm -f $$p $$f ; \
 	done
-asfxload$(EXEEXT): $(asfxload_OBJECTS) $(asfxload_DEPENDENCIES) 
-	@rm -f asfxload$(EXEEXT)
-	$(LINK) $(asfxload_OBJECTS) $(asfxload_LDADD) $(LIBS)
 aweset$(EXEEXT): $(aweset_OBJECTS) $(aweset_DEPENDENCIES) 
 	@rm -f aweset$(EXEEXT)
 	$(LINK) $(aweset_OBJECTS) $(aweset_LDADD) $(LIBS)
@@ -348,8 +340,6 @@
 distclean-compile:
 	-rm -f *.tab.c
 
-include ./$(DEPDIR)/alsa.Po
-include ./$(DEPDIR)/asfxload.Po
 include ./$(DEPDIR)/aweset.Po
 include ./$(DEPDIR)/gusload.Po
 include ./$(DEPDIR)/seq.Po
@@ -828,8 +818,8 @@
 
 
 install-data-hook:
-	rm -f $(DESTDIR)$(mandir)/man1/asfxload.1
-	(cd $(DESTDIR)$(mandir)/man1 && $(LN_S) sfxload.1 asfxload.1)
+#	rm -f $(DESTDIR)$(mandir)/man1/asfxload.1
+#	(cd $(DESTDIR)$(mandir)/man1 && $(LN_S) sfxload.1 asfxload.1)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
