diff -urN libdc1394-2.0.0-pre7.orig/libdc1394/dc1394_vendor_avt.c libdc1394-2.0.0-pre7/libdc1394/dc1394_vendor_avt.c
--- libdc1394-2.0.0-pre7.orig/libdc1394/dc1394_vendor_avt.c	2006-06-19 13:19:06.522104015 -0400
+++ libdc1394-2.0.0-pre7/libdc1394/dc1394_vendor_avt.c	2006-06-19 13:19:20.670991015 -0400
@@ -22,6 +22,7 @@
  */
 
 #include "dc1394_vendor_avt.h"
+#include "config.h"
 
 /********************************************************/
 /* Configuration Register Offsets for Advances features */
diff -urN libdc1394-2.0.0-pre7.orig/libdc1394/dc1394_vendor_avt.h libdc1394-2.0.0-pre7/libdc1394/dc1394_vendor_avt.h
--- libdc1394-2.0.0-pre7.orig/libdc1394/dc1394_vendor_avt.h	2006-06-19 13:18:55.533415118 -0400
+++ libdc1394-2.0.0-pre7/libdc1394/dc1394_vendor_avt.h	2006-06-19 13:19:14.342594287 -0400
@@ -26,7 +26,6 @@
 #include <libraw1394/raw1394.h>
 #include "dc1394_control.h"
 #include "dc1394_register.h"
-#include "config.h"
 
 typedef struct __dc1394_avt_adv_feature_info_struct 
 {
