--- khexedit/configure.in.in	2003-11-03 21:11:09.000000000 +0200
+++ khexedit/configure.in.in	2004-10-16 18:40:54.523771976 +0200
@@ -1,8 +1,4 @@
-# kossebau: copied from kdelibs/configure.in.in for khexedit
 AC_LANG_SAVE
 AC_LANG_C
-dnl AC_C_BIGENDIAN has a bug (one of its tests uses "main()" instead of
-dnl "int main()") so C++ compilers would break. Thats why we switch languages
 AC_C_BIGENDIAN
 AC_LANG_RESTORE
-# kossebau end
\ No newline at end of file
