--- configure.ac	2004-08-12 19:35:56.695660963 +0000
+++ configure.ac	2004-08-12 19:36:26.232540924 +0000
@@ -202,9 +202,9 @@
 # tcl
 if test "$with_tcl" != "no"; then
     CPPFLAGS_save="$CPPFLAGS"
-    WV_APPEND(CPPFLAGS, -I/usr/include/tcl8.3)
+    WV_APPEND(CPPFLAGS, -I/usr/include/tcl8.4)
     AC_CHECK_HEADERS(tcl.h,, [with_tcl=no])
-    AC_CHECK_LIB(tcl8.3, TclInterpInit,, [with_tcl=no])
+    AC_CHECK_LIB(tcl8.4, TclInterpInit,, [with_tcl=no])
     if test "$with_tcl" != "no"; then
         CPPFLAGS="$CPPFLAGS_save"
     fi
--- vars.mk	2004-08-12 21:32:32.627250401 +0000
+++ vars.mk	2004-08-12 21:32:39.932489430 +0000
@@ -178,6 +178,6 @@
 libwvgtk.a libwvgtk.so: $(call objects,gtk)
 libwvgtk.so: -lgtk -lgdk libwvstreams.so libwvutils.so
 
-libuniconf_tcl.so: bindings/uniconf_tcl.o -ltcl8.3 -luniconf
+libuniconf_tcl.so: bindings/uniconf_tcl.o -ltcl8.4 -luniconf
 
 uniconf/daemon/uniconfdaemon: libuniconf.so libwvutils.so -ldb
