diff -Nru wvstreams-4.4.1.orig/utils/wvcrash.cc wvstreams-4.4.1/utils/wvcrash.cc
--- wvstreams-4.4.1.orig/utils/wvcrash.cc	2007-08-17 08:15:30.000000000 +0300
+++ wvstreams-4.4.1/utils/wvcrash.cc	2008-01-20 13:07:57.000000000 +0200
@@ -388,7 +388,6 @@
     act.sa_handler = wvcrash;
     sigfillset(&act.sa_mask);
     act.sa_flags = SA_ONSTACK | SA_RESTART;
-    act.sa_restorer = NULL;
     
     if (sigaction(sig, &act, NULL))
         fprintf(stderr, "Failed to setup wvcrash handler for signal %d: %s\n",
