diff -Nru wvstreams-4.3.orig/linuxstreams/wvinterface.cc wvstreams-4.3/linuxstreams/wvinterface.cc
--- wvstreams-4.3.orig/linuxstreams/wvinterface.cc	2007-02-07 21:06:03.000000000 +0200
+++ wvstreams-4.3/linuxstreams/wvinterface.cc	2007-03-12 13:55:29.000000000 +0200
@@ -21,6 +21,7 @@
 #include <errno.h>
 #include <linux/sockios.h>
 
+#include <sys/sysctl.h> /* Needed for __user definition */
 #define _LINUX_IF_H /* Hack to prevent loading linux/if.h */
 #include <linux/wireless.h>
 
