diff -Nru wvstreams-4.2.1.orig/linuxstreams/wvinterface.cc wvstreams-4.2.1/linuxstreams/wvinterface.cc
--- wvstreams-4.2.1.orig/linuxstreams/wvinterface.cc	2005-11-17 22:45:05.000000000 +0200
+++ wvstreams-4.2.1/linuxstreams/wvinterface.cc	2005-12-24 10:14:34.966632750 +0200
@@ -21,6 +21,7 @@
 #include <errno.h>
 #include <linux/sockios.h>
 
+#include <linux/compiler.h>
 #define _LINUX_IF_H /* Hack to prevent loading linux/if.h */
 #include <linux/wireless.h>
 
