diff -Nru wvstreams-4.2.2.orig/Makefile wvstreams-4.2.2/Makefile
--- wvstreams-4.2.2.orig/Makefile	2006-01-10 23:53:31.000000000 +0200
+++ wvstreams-4.2.2/Makefile	2006-05-06 10:13:33.211497500 +0300
@@ -182,8 +182,8 @@
 	$(INSTALL_PROGRAM) uniconf/tests/uni $(DESTDIR)$(bindir)/
 	$(INSTALL) -d $(DESTDIR)$(sbindir)
 	$(INSTALL_PROGRAM) uniconf/daemon/uniconfd $(DESTDIR)$(sbindir)/
-	$(INSTALL) -d $(DESTDIR)$(localstatedir)/lib/uniconf
-	touch $(DESTDIR)$(localstatedir)/lib/uniconf/uniconfd.ini
+	$(INSTALL) -d $(DESTDIR)$(localstatedir)/uniconf
+	$(INSTALL_DATA) uniconf/daemon/uniconfd.ini $(DESTDIR)$(localstatedir)/uniconf
 	$(INSTALL) -d $(DESTDIR)$(mandir)/man8
 	$(INSTALL_DATA) uniconf/daemon/uniconfd.8 $(DESTDIR)$(mandir)/man8
 	$(INSTALL_DATA) uniconf/tests/uni.8 $(DESTDIR)$(mandir)/man8
