diff -Nru wvstreams-4.4.1.orig/configure.ac wvstreams-4.4.1/configure.ac
--- wvstreams-4.4.1.orig/configure.ac	2008-01-20 14:04:54.000000000 +0200
+++ wvstreams-4.4.1/configure.ac	2008-01-20 14:06:45.000000000 +0200
@@ -488,10 +488,9 @@
 
 # xplc
 if test "$with_xplc" != "no"; then
-    if test "$with_xplc" = ""; then
-	if pkg-config --modversion xplc-${xplc_version}; then
-            WV_APPEND(CPPFLAGS, [`pkg-config --cflags xplc-${xplc_version}`])
-	    WV_APPEND(LDFLAGS, [`pkg-config --libs xplc-${xplc_version}`])
+	if pkg-config --atleast-version=${xplc_version} xplc; then
+	    WV_APPEND(CPPFLAGS, [`pkg-config --cflags xplc`])
+	    WV_APPEND(LDFLAGS, [`pkg-config --libs xplc`])
 	    AC_CHECK_HEADERS(xplc/core.h,, [with_xplc=no], [#define UNSTABLE])
 	    LIBS_save="$LIBS"
 	    AC_CHECK_LIB(xplc, XPLC_getServiceManager,, [with_xplc=no])
@@ -501,24 +500,6 @@
 	else
 	    with_xplc='no'
 	fi
-    else
-        WV_APPEND(CPPFLAGS, [-I$with_xplc/include])
-	WV_APPEND(LDFLAGS, [-L$with_xplc])
-	AC_CHECK_HEADERS(xplc/core.h,, [with_xplc=no], [#define UNSTABLE])
-	LIBS_save="$LIBS"
-	AC_CHECK_LIB(xplc, XPLC_getServiceManager,, [with_xplc=no])
-	LIBS="$LIBS_save"
-	xplc_pc="xplc-${xplc_version}"
-    fi
-    # What if we can't find it?
-    if test "$with_xplc" = "no" && test -d xplc; then
-        WV_APPEND(CPPFLAGS, [-I\$(WVSTREAMS)/xplc/include])
-        WV_APPEND(LDFLAGS, [-L\$(WVSTREAMS)/xplc])
-        AC_CONFIG_SUBDIRS(xplc)
-        build_xplc=yes
-        with_xplc='$(WVSTREAMS)/xplc'
-	xplc_pc='wvxplc'
-    fi
 fi
 
 # zlib
