diff -ruN wvstreams-4.4/utils/wvtask.cc wvstreams-4.4.patched/utils/wvtask.cc
--- wvstreams-4.4/utils/wvtask.cc	2007-07-19 23:22:55.000000000 +0200
+++ wvstreams-4.4.patched/utils/wvtask.cc	2008-04-02 20:20:54.000000000 +0200
@@ -37,12 +37,12 @@
 #include <unistd.h>
 #include <sys/resource.h>
 
-#ifdef HAVE_VALGRIND_MEMCHECK_H
-#include <valgrind/memcheck.h>
-#else
+//#ifdef HAVE_VALGRIND_MEMCHECK_H
+//#include <valgrind/memcheck.h>
+//#else
 #define VALGRIND_MAKE_READABLE(x, y)
 #define RUNNING_ON_VALGRIND 0
-#endif
+//#endif
 
 #define TASK_DEBUG 0
 #if TASK_DEBUG
