diff -Nru wvstreams-4.4.1.orig/Makefile wvstreams-4.4.1/Makefile
--- wvstreams-4.4.1.orig/Makefile	2007-08-30 05:48:09.000000000 +0300
+++ wvstreams-4.4.1/Makefile	2008-01-20 12:53:19.000000000 +0200
@@ -189,8 +189,8 @@
 	$(INSTALL_PROGRAM) uniconf/tests/uni $(DESTDIR)$(bindir)/
 	$(INSTALL) -d $(DESTDIR)$(sbindir)
 	$(INSTALL_PROGRAM) uniconf/daemon/uniconfd $(DESTDIR)$(sbindir)/
-	$(INSTALL) -d $(DESTDIR)$(localstatedir)/lib/uniconf
-	touch $(DESTDIR)$(localstatedir)/lib/uniconf/uniconfd.ini
+	$(INSTALL) -d $(DESTDIR)$(localstatedir)/uniconf
+	$(INSTALL_DATA) uniconf/daemon/uniconfd.ini $(DESTDIR)$(localstatedir)/uniconf
 	$(INSTALL) -d $(DESTDIR)$(mandir)/man8
 	$(INSTALL_DATA) uniconf/daemon/uniconfd.8 $(DESTDIR)$(mandir)/man8
 	$(INSTALL_DATA) uniconf/tests/uni.8 $(DESTDIR)$(mandir)/man8
