diff -Nru wvstreams-4.2.1.orig/wvrules.mk wvstreams-4.2.1/wvrules.mk
--- wvstreams-4.2.1.orig/wvrules.mk	2005-11-17 22:45:09.000000000 +0200
+++ wvstreams-4.2.1/wvrules.mk	2005-12-24 10:31:41.346777500 +0200
@@ -296,7 +296,7 @@
 %.E: %.cc;	$(call wvcxx,$@,$<,$*,,-E)
 %.E: %.cpp;	$(call wvcxx,$@,$<,$*,,-E)
 
-%.moc: %.h;	moc -o $@ $<
+%.moc: %.h;	$(MOC) -o $@ $<
 
 %: %.o;		$(call wvlink,$@,$^) 
 %.t: %.t.o;	$(call wvlink,$@,$(call reverse,$(filter %.o,$^)) $(filter-out %.o,$^) $(LIBWVTEST))
