--- glibc-2.2.5/linuxthreads/sysdeps/alpha/pt-machine.h	Mon Jul 23 17:52:42 2001
+++ glibc-2.2.5-alpha-gcc3-fix/linuxthreads/sysdeps/alpha/pt-machine.h	Sat Sep  7 14:58:23 2002
@@ -71,7 +71,7 @@
 #define THREAD_SELF \
 ({									      \
   register pthread_descr __self __asm__("$0");				      \
-  __asm__ ("call_pal %1" : "=r"(__self) : "i"(PAL_rduniq) : "$0");	      \
+  __asm__ ("call_pal %1" : "=r"(__self) : "i"(PAL_rduniq));		      \
   __self;								      \
 })
 
