diff -ur src.orig/game.cc src/game.cc
--- src.orig/game.cc	2003-09-14 00:31:24.735219680 -0400
+++ src/game.cc	2003-09-14 00:30:40.688915736 -0400
@@ -18,7 +18,8 @@
 
 #include <config.h>
 
-#include <strstream.h>
+#include <strstream>
+using namespace std;
 #include <time.h>
 #include <unistd.h>
 
diff -ur src.orig/highscore_tab.cc src/highscore_tab.cc
--- src.orig/highscore_tab.cc	2003-09-14 00:31:24.744218312 -0400
+++ src/highscore_tab.cc	2003-09-14 00:30:56.376530856 -0400
@@ -16,7 +16,8 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
-#include <strstream.h>
+#include <strstream>
+using namespace std;
 
 #include "highscore_tab.h"
 #include "intl.h"
diff -ur src.orig/solved_dialog.cc src/solved_dialog.cc
--- src.orig/solved_dialog.cc	2003-09-14 00:31:24.743218464 -0400
+++ src/solved_dialog.cc	2003-09-14 00:30:48.212771936 -0400
@@ -16,7 +16,8 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
-#include <strstream.h>
+#include <strstream>
+using namespace std;
 
 #include "intl.h"
 #include "solved_dialog.h"
