--- configure.in.orig	2006-10-16 12:42:26.000000000 -0500
+++ configure.in	2006-10-16 12:46:13.000000000 -0500
@@ -42,21 +42,32 @@
 AC_SUBST(GTK_SHARP_LIBS)
 
 
-PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-2.0, enable_gtkhtml=yes, enable_gtkhtml=no)
-AC_SUBST(GTKHTML_SHARP_LIBS)
+AC_ARG_ENABLE([gtkhtml],
+              AC_HELP_STRING([--enable-gtkhtml], [Use GTKHTML Browser]),,
+              [enable_gtkhtml=yes])
+
+if test "x$enable_gtkhtml" = "xyes" ; then
+   PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-2.0)
+   AC_SUBST(GTKHTML_SHARP_LIBS)
+fi
+
 AM_CONDITIONAL(ENABLE_GTKHTML, test "x$enable_gtkhtml" = "xyes")
 
 
-PKG_CHECK_MODULES(GECKO_SHARP, gecko-sharp-2.0, enable_gecko=yes, enable_gecko=no)
-AC_SUBST(GECKO_SHARP_LIBS)
-AM_CONDITIONAL(ENABLE_GECKO, test "x$enable_gecko" = "xyes")
+AC_ARG_ENABLE([mozilla],
+              AC_HELP_STRING([--enable-mozilla], [Use Gecko Browser]),,
+              [enable_gecko=yes])
+
+if test "x$enable_gecko" = "xyes" ; then
+   PKG_CHECK_MODULES(GECKO_SHARP, gecko-sharp-2.0)
+   AC_SUBST(GECKO_SHARP_LIBS)
 
-if test "x$enable_gecko" = "xyes"; then
-   # get the mozilla home directory
    MOZILLA_HOME="`$PKG_CONFIG --variable=libdir mozilla-gtkmozembed`"
    AC_SUBST(MOZILLA_HOME)
 fi
 
+AM_CONDITIONAL(ENABLE_GECKO, test "x$enable_gecko" = "xyes")
+
 
 if test "x$enable_gtkhtml$enable_gecko" = "xnono" ; then
    AC_MSG_ERROR([no HTML display found. You need either gtkhtml or gecko])
