diff -urN log4cxx-0.9.7.orig/src/msxml.cpp log4cxx-0.9.7/src/msxml.cpp
--- log4cxx-0.9.7.orig/src/msxml.cpp	2004-05-10 08:37:45.000000000 -0400
+++ log4cxx-0.9.7/src/msxml.cpp	2005-01-07 01:29:54.013941856 -0500
@@ -289,4 +289,5 @@
 	}
 }
 
-#endif
\ No newline at end of file
+#endif
+
diff -urN log4cxx-0.9.7.orig/src/smtpappender.cpp log4cxx-0.9.7/src/smtpappender.cpp
--- log4cxx-0.9.7.orig/src/smtpappender.cpp	2004-05-10 08:38:01.000000000 -0400
+++ log4cxx-0.9.7/src/smtpappender.cpp	2005-01-07 01:29:33.730025480 -0500
@@ -134,7 +134,7 @@
 	for (i = recipients.begin(); i != recipients.end(); i++)
 	{
 		if (::libsmtp_add_recipient(LIBSMTP_REC_TO,
-			(TCHAR *)T2A(i->c_str()),
+			T2A((TCHAR *)i->c_str()),
 			(libsmtp_session_struct *)session) != 0)
 		{
 			LogLog::error(_T("Could not add recipient ")+*i+_T("."));
@@ -316,6 +316,8 @@
 */
 void SMTPAppender::sendBuffer()
 {
+	USES_CONVERSION;
+
 	// Note: this code already owns the monitor for this
 	// appender. This frees us from needing to synchronize on 'cb'.
 	try
