diff -uNr dahdi-tools-2.2.0.ORIG/Makefile dahdi-tools-2.2.0/Makefile
--- dahdi-tools-2.2.0.ORIG/Makefile	2009-12-22 12:20:03.000000000 +0000
+++ dahdi-tools-2.2.0/Makefile	2009-12-22 12:20:46.000000000 +0000
@@ -293,18 +293,6 @@
 	$(ADD_INITD)
 endif
 	@echo "DAHDI has been configured."
-	@echo ""
-	@echo "List of detected DAHDI devices:"
-	@echo ""
-	@if [ `xpp/dahdi_hardware | tee /dev/stderr | wc -l` -eq 0 ]; then \
-		echo "No hardware found"; \
-	else \
-		echo ""; \
-		echo "run 'dahdi_genconf modules' to load support for only " ;\
-		echo "the DAHDI hardware installed in this system.  By "; \
-		echo "default support for all DAHDI hardware is loaded at "; \
-		echo "DAHDI start. "; \
-	fi
 
 update:
 	@if [ -d .svn ]; then \
