Index: ekeyd-1.1.3/daemon/Makefile
===================================================================
--- ekeyd-1.1.3.orig/daemon/Makefile
+++ ekeyd-1.1.3/daemon/Makefile
@@ -118,7 +118,7 @@ daemonise_ulusbd.o: daemonise.c
 	$(COMPILE.c) $(OUTPUT_OPTION) $(PTHFLAGS) $^
 
 egd-linux: egd-linux.o daemonise.o
-	$(CC) $(CFLAGS) -o $@ $^
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 
 ekeyd: ekeyd.o daemonise.o lstate.o connection.o stream.o frame.o packet.o keydb.o util.o fds.o krnlop.o foldback.o stats.o nonce.o ../device/frames/pem.o ../device/skeinwrap.o ../device/skein/skein.o ../device/skein/skein_block.o
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
