diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdsl/driver.c usr/src/kernel-modules/fcdsl/src/src.fcdsl/driver.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdsl/driver.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdsl/driver.c
@@ -122,7 +122,7 @@
 static void			rx_task (unsigned long data);
 static void			tx_handler (card_p cp);
 static void			rx_handler (card_p cp);
-static irqreturn_t		irq_handler (int, void *, struct pt_regs *);
+static irqreturn_t		irq_handler (int, void *);
 static void			enable_thread (void);
 static void			disable_thread (void);
 static int			make_thread (void);
@@ -1549,7 +1549,7 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-static irqreturn_t irq_handler (int irq, void * args, struct pt_regs * regs) {
+static irqreturn_t irq_handler (int irq, void * args) {
 	unsigned long	flags;
 	int		tx_flag	= 0;
 	int		rx_flag	= 0;
@@ -1557,7 +1557,6 @@
 	card_p		cp;
 	
 	UNUSED_ARG (irq);
-	UNUSED_ARG (regs);
 	if (capi_card != (card_p) args) {
 		return IRQ_NONE;
 	}
@@ -1797,19 +1796,19 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-int driver_init (void) {
+int avm_driver_init (void) {
 
 	return (NULL != (capi_lib = link_library (&capi_card)));
-} /* driver_init */
+} /* avm_driver_init */
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-void driver_exit (void) {
+void avm_driver_exit (void) {
 
 	assert (capi_lib != NULL);
 	free_library ();
 	capi_lib = NULL;
-} /* driver_exit */
+} /* avm_driver_exit */
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdsl/driver.h usr/src/kernel-modules/fcdsl/src/src.fcdsl/driver.h
--- usr/src/kernel-modules/fcdsl/src/src.fcdsl/driver.h
+++ usr/src/kernel-modules/fcdsl/src/src.fcdsl/driver.h
@@ -23,7 +23,7 @@
 #ifndef __have_driver_h__
 #define __have_driver_h__
 
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/pci.h>
 #include <linux/list.h>
 #include <linux/skbuff.h>
@@ -139,8 +139,8 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-extern int driver_init (void);
-extern void driver_exit (void);
+extern int avm_driver_init (void);
+extern void avm_driver_exit (void);
 
 #endif
 
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdsl/main.c usr/src/kernel-modules/fcdsl/src/src.fcdsl/main.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdsl/main.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdsl/main.c
@@ -23,7 +23,7 @@
 #include <stdarg.h>
 #include <asm/uaccess.h>
 #include <linux/pci.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
@@ -115,14 +115,14 @@
 		return -ENODEV;
 	}
 	NOTE( "Loading...\n");
-	if (!driver_init ()) {
+	if (!avm_driver_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		ERROR("Not loaded.\n");
 		return -ENOSYS;
 	}
 	if (0 != (res = add_card (dev))) {
 		ERROR("Not loaded.\n");
-		driver_exit ();
+		avm_driver_exit ();
 		return res;
 	}
 	NOTE("Loaded.\n");
@@ -143,7 +143,7 @@
 	NOTE("Removing...\n");
 	remove_ctrls (card);
 	NOTE("Removed.\n");
-	driver_exit ();
+	avm_driver_exit ();
 #ifndef NDEBUG
 	if (hallocated() != 0) {
 		ERROR( "%u bytes leaked.\n", hallocated());
@@ -196,7 +196,7 @@
 		return -ENOSYS;
 	}
 
-	if (0 == (err = pci_module_init (&fcdsl_driver))) {
+	if (0 == (err = pci_register_driver (&fcdsl_driver))) {
 		LOG("PCI driver registered.\n");
 		register_capi_driver (&fcdsl_capi_driver);
 		LOG("CAPI driver registered.\n");
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdsl2/devif.c usr/src/kernel-modules/fcdsl/src/src.fcdsl2/devif.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdsl2/devif.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdsl2/devif.c
@@ -747,12 +747,11 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-irqreturn_t device_interrupt (int irq, void * args, struct pt_regs * regs) {
+irqreturn_t device_interrupt (int irq, void * args) {
 	unsigned long	intpins;
 	card_p		cp	= (card_p) args;
 
 	UNUSED_ARG (irq);
-	UNUSED_ARG (regs);
 	assert (capi_card == cp);
 
 	intpins = PINL (cp->mmio_base + C6205_PCI_HSR_OFFSET);
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdsl2/devif.h usr/src/kernel-modules/fcdsl/src/src.fcdsl2/devif.h
--- usr/src/kernel-modules/fcdsl/src/src.fcdsl2/devif.h
+++ usr/src/kernel-modules/fcdsl/src/src.fcdsl2/devif.h
@@ -80,7 +80,7 @@
 extern void set_interrupt_callback (irq_callback_t, void *);
 extern void clear_interrupt_callback (void);
 
-extern irqreturn_t device_interrupt (int, void *, struct pt_regs *);
+extern irqreturn_t device_interrupt (int, void *);
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdsl2/driver.c usr/src/kernel-modules/fcdsl/src/src.fcdsl2/driver.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdsl2/driver.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdsl2/driver.c
@@ -1563,19 +1563,19 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-int driver_init (void) {
+int avm_driver_init (void) {
 
 	return (NULL != (capi_lib = link_library (&capi_card)));
-} /* driver_init */
+} /* avm_driver_init */
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-void driver_exit (void) {
+void avm_driver_exit (void) {
 
 	assert (capi_lib);
 	free_library ();
 	capi_lib = NULL;
-} /* driver_exit */
+} /* avm_driver_exit */
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdsl2/driver.h usr/src/kernel-modules/fcdsl/src/src.fcdsl2/driver.h
--- usr/src/kernel-modules/fcdsl/src/src.fcdsl2/driver.h
+++ usr/src/kernel-modules/fcdsl/src/src.fcdsl2/driver.h
@@ -24,7 +24,7 @@
 #define __have_driver_h__
 
 #include <asm/atomic.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/skbuff.h>
 #include <linux/pci.h>
 #include <linux/spinlock.h>
@@ -161,8 +161,8 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-extern int driver_init (void);
-extern void driver_exit (void);
+extern int avm_driver_init (void);
+extern void avm_driver_exit (void);
 
 #endif
 
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdsl2/main.c usr/src/kernel-modules/fcdsl/src/src.fcdsl2/main.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdsl2/main.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdsl2/main.c
@@ -23,7 +23,7 @@
 #include <stdarg.h>
 #include <asm/uaccess.h>
 #include <linux/pci.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
@@ -116,14 +116,14 @@
 		return -ENODEV;
 	}
 	NOTE("Loading...\n");
-	if (!driver_init ()) {
+	if (!avm_driver_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		NOTE("Not loaded.\n");
 		return -ENOSYS;
 	}
 	if (0 != (res = add_card (dev))) {
 		NOTE("Not loaded.\n");
-		driver_exit ();
+		avm_driver_exit ();
 		return res;
 	}
 	NOTE("Loaded.\n");
@@ -144,7 +144,7 @@
 	NOTE("Removing...\n");
 	remove_ctrls (cp);
 	NOTE("Removed.\n");
-	driver_exit ();
+	avm_driver_exit ();
 #ifndef NDEBUG
 	if (hallocated() != 0) {
 		ERROR("%u bytes leaked.\n", hallocated());
@@ -195,7 +195,7 @@
 	NOTE("-- 32 bit CAPI driver --\n");
 #endif
 
-	if (0 == (err = pci_module_init (&fcdsl2_driver))) {
+	if (0 == (err = pci_register_driver (&fcdsl2_driver))) {
 		LOG("PCI driver registered.\n");
 		register_capi_driver (&fcdsl2_capi_driver);
 		LOG("CAPI driver registered.\n");
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslsl/devif.c usr/src/kernel-modules/fcdsl/src/src.fcdslsl/devif.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdslsl/devif.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslsl/devif.c
@@ -747,12 +747,11 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-irqreturn_t device_interrupt (int irq, void * args, struct pt_regs * regs) {
+irqreturn_t device_interrupt (int irq, void * args) {
 	unsigned long	intpins;
 	card_p		cp	= (card_p) args;
 
 	UNUSED_ARG (irq);
-	UNUSED_ARG (regs);
 	assert (capi_card == cp);
 
 	intpins = PINL (cp->mmio_base + C6205_PCI_HSR_OFFSET);
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslsl/devif.h usr/src/kernel-modules/fcdsl/src/src.fcdslsl/devif.h
--- usr/src/kernel-modules/fcdsl/src/src.fcdslsl/devif.h
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslsl/devif.h
@@ -80,7 +80,7 @@
 extern void set_interrupt_callback (irq_callback_t, void *);
 extern void clear_interrupt_callback (void);
 
-extern irqreturn_t device_interrupt (int, void *, struct pt_regs *);
+extern irqreturn_t device_interrupt (int, void *);
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslsl/driver.c usr/src/kernel-modules/fcdsl/src/src.fcdslsl/driver.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdslsl/driver.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslsl/driver.c
@@ -1563,19 +1563,19 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-int driver_init (void) {
+int avm_driver_init (void) {
 
 	return (NULL != (capi_lib = link_library (&capi_card)));
-} /* driver_init */
+} /* avm_driver_init */
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-void driver_exit (void) {
+void avm_driver_exit (void) {
 
 	assert (capi_lib);
 	free_library ();
 	capi_lib = NULL;
-} /* driver_exit */
+} /* avm_driver_exit */
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslsl/driver.h usr/src/kernel-modules/fcdsl/src/src.fcdslsl/driver.h
--- usr/src/kernel-modules/fcdsl/src/src.fcdslsl/driver.h
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslsl/driver.h
@@ -24,7 +24,7 @@
 #define __have_driver_h__
 
 #include <asm/atomic.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/skbuff.h>
 #include <linux/pci.h>
 #include <linux/spinlock.h>
@@ -161,8 +161,8 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-extern int driver_init (void);
-extern void driver_exit (void);
+extern int avm_driver_init (void);
+extern void avm_driver_exit (void);
 
 #endif
 
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslsl/main.c usr/src/kernel-modules/fcdsl/src/src.fcdslsl/main.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdslsl/main.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslsl/main.c
@@ -23,7 +23,7 @@
 #include <stdarg.h>
 #include <asm/uaccess.h>
 #include <linux/pci.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
@@ -116,14 +116,14 @@
 		return -ENODEV;
 	}
 	NOTE("Loading...\n");
-	if (!driver_init ()) {
+	if (!avm_driver_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		NOTE("Not loaded.\n");
 		return -ENOSYS;
 	}
 	if (0 != (res = add_card (dev))) {
 		NOTE("Not loaded.\n");
-		driver_exit ();
+		avm_driver_exit ();
 		return res;
 	}
 	NOTE("Loaded.\n");
@@ -144,7 +144,7 @@
 	NOTE("Removing...\n");
 	remove_ctrls (cp);
 	NOTE("Removed.\n");
-	driver_exit ();
+	avm_driver_exit ();
 #ifndef NDEBUG
 	if (hallocated() != 0) {
 		ERROR("%u bytes leaked.\n", hallocated());
@@ -195,7 +195,7 @@
 	NOTE("-- 32 bit CAPI driver --\n");
 #endif
 
-	if (0 == (err = pci_module_init (&fcdsl2_driver))) {
+	if (0 == (err = pci_register_driver (&fcdsl2_driver))) {
 		LOG("PCI driver registered.\n");
 		register_capi_driver (&fcdsl2_capi_driver);
 		LOG("CAPI driver registered.\n");
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslslusb/driver.h usr/src/kernel-modules/fcdsl/src/src.fcdslslusb/driver.h
--- usr/src/kernel-modules/fcdsl/src/src.fcdslslusb/driver.h
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslslusb/driver.h
@@ -24,7 +24,7 @@
 #define __have_driver_h__
 
 #include <asm/atomic.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/skbuff.h>
 #include <linux/spinlock.h>
 #include <linux/usb.h>
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslslusb/main.c usr/src/kernel-modules/fcdsl/src/src.fcdslslusb/main.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdslslusb/main.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslslusb/main.c
@@ -27,7 +27,7 @@
 #include <stdarg.h>
 #include <asm/uaccess.h>
 #include <linux/usb.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslusb/driver.h usr/src/kernel-modules/fcdsl/src/src.fcdslusb/driver.h
--- usr/src/kernel-modules/fcdsl/src/src.fcdslusb/driver.h
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslusb/driver.h
@@ -25,7 +25,7 @@
 
 #include <asm/semaphore.h>
 #include <asm/atomic.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/skbuff.h>
 #include <linux/capi.h>
 #include <linux/isdn/capilli.h>
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslusb2/driver.h usr/src/kernel-modules/fcdsl/src/src.fcdslusb2/driver.h
--- usr/src/kernel-modules/fcdsl/src/src.fcdslusb2/driver.h
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslusb2/driver.h
@@ -24,7 +24,7 @@
 #define __have_driver_h__
 
 #include <asm/atomic.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/skbuff.h>
 #include <linux/spinlock.h>
 #include <linux/usb.h>
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslusb2/main.c usr/src/kernel-modules/fcdsl/src/src.fcdslusb2/main.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdslusb2/main.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslusb2/main.c
@@ -27,7 +27,7 @@
 #include <stdarg.h>
 #include <asm/uaccess.h>
 #include <linux/usb.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslusba/driver.h usr/src/kernel-modules/fcdsl/src/src.fcdslusba/driver.h
--- usr/src/kernel-modules/fcdsl/src/src.fcdslusba/driver.h
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslusba/driver.h
@@ -24,7 +24,7 @@
 #define __have_driver_h__
 
 #include <asm/atomic.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/skbuff.h>
 #include <linux/spinlock.h>
 #include <linux/usb.h>
diff -Nur usr/src/kernel-modules/fcdsl/src/src.fcdslusba/main.c usr/src/kernel-modules/fcdsl/src/src.fcdslusba/main.c
--- usr/src/kernel-modules/fcdsl/src/src.fcdslusba/main.c
+++ usr/src/kernel-modules/fcdsl/src/src.fcdslusba/main.c
@@ -27,7 +27,7 @@
 #include <stdarg.h>
 #include <asm/uaccess.h>
 #include <linux/usb.h>
-#include <linux/config.h>
+#include <linux/autoconf.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
