diff -ur ams-1.8.8-rc2.orig/spectrumscreen.cpp ams-1.8.8-rc2/spectrumscreen.cpp
--- ams-1.8.8-rc2.orig/spectrumscreen.cpp	2003-10-22 20:12:11.000000000 +0300
+++ ams-1.8.8-rc2/spectrumscreen.cpp	2008-04-10 13:21:55.000000000 +0300
@@ -9,7 +9,7 @@
 #include <qevent.h>
 #include "synthdata.h"
 #include "spectrumscreen.h"
-#include <rfftw.h>
+#include <srfftw.h>
 #include <math.h>
 
 SpectrumScreen::SpectrumScreen(QWidget* parent, const char *name, SynthData *p_synthdata) : QWidget (parent, name)
diff -ur ams-1.8.8-rc2.orig/spectrumscreen.h ams-1.8.8-rc2/spectrumscreen.h
--- ams-1.8.8-rc2.orig/spectrumscreen.h	2007-05-05 12:59:10.000000000 +0300
+++ ams-1.8.8-rc2/spectrumscreen.h	2008-04-10 13:23:25.000000000 +0300
@@ -1,4 +1,4 @@
-#ifdef OUTDATED_CODE
+//#ifdef OUTDATED_CODE
 
 #ifndef KSPECTRUMSCREEN_H
 #define KSPECTRUMSCREEN_H
@@ -14,7 +14,7 @@
 #include <qpixmap.h>
 #include <qevent.h>
 #include "synthdata.h"
-#include <rfftw.h>
+#include <srfftw.h>
 
 #define MINIMUM_WIDTH                 100
 #define MINIMUM_HEIGHT                 50
@@ -115,4 +115,4 @@
   
 #endif
 
-#endif
+//#endif
