--- Ted-2.17/appFrame/aclocal.m4
+++ Ted-2.17/appFrame/aclocal.m4
@@ -210,6 +210,7 @@
     then
 	# Includes
 	for ac_dir in				\
+	    /usr/include			\
 	    /usr/X11R6/include			\
 	    /usr/local/include			\
 	    /usr/dt/include			\
--- Ted-2.17/Ted/aclocal.m4
+++ Ted-2.17/Ted/aclocal.m4
@@ -279,6 +279,7 @@
     then
 	# Includes
 	for ac_dir in				\
+	    /usr/include			\
 	    /usr/X11R6/include			\
 	    /usr/local/include			\
 	    /usr/dt/include			\
