--- libdbi-drivers-0.8.1/drivers/oracle/dbd_oracle.c.orig	2005-08-31 23:44:53.000000000 +0200
+++ libdbi-drivers-0.8.1/drivers/oracle/dbd_oracle.c	2006-05-30 13:10:48.000000000 +0200
@@ -523,7 +523,7 @@
 	unsigned int sizeattrib;
 	dbi_data_t *data;
 	char *ptr, *cols[result->numfields];
-	dword status;
+	sword status;
 
 	/* 
 	 * Prefetch all cols as char *'s 
@@ -700,7 +700,7 @@
 }
 
 /* taken from sqlite3 driver */
-size_t dbd_quote_binary (dbi_conn_t *conn, const unsigned char *orig, size_t from_length, char **ptr_dest ) {
+size_t dbd_quote_binary (dbi_conn_t *conn, const unsigned char *orig, size_t from_length, unsigned char **ptr_dest ) {
   unsigned char *temp;
   size_t len;
 
