diff -ur TwistedWeb2-0.1-orig/twisted/web2/test/test_stream.py TwistedWeb2-0.1/twisted/web2/test/test_stream.py
--- TwistedWeb2-0.1-orig/twisted/web2/test/test_stream.py	2005-10-10 03:14:19.000000000 +0200
+++ TwistedWeb2-0.1/twisted/web2/test/test_stream.py	2005-10-10 03:15:14.000000000 +0200
@@ -436,7 +436,7 @@
     """
 
 
-__doctests__ = ['twisted.web2.test.test_stream.CompoundStreamTest', 'twisted.web2.stream.generatorToStream']
+__doctests__ = ['twisted.web2.test.test_stream', 'twisted.web2.stream']
 # TODO: 
 # CompoundStreamTest
 # more tests for ProducerStreamTest
