--- hddtemp.c	Wed Jan 16 11:13:45 2002
+++ hddtemp.c.new	Mon Jun 10 16:25:06 2002
@@ -44,11 +44,11 @@
 #define F_to_C(val) (int)(((double)(val)-32.0)/1.8)
 #define C_to_F(val) (int)(((double)(val)*1.8)+32)
 
 #define HDDTEMP_VERSION        "0.2"
 #define DEFAULT_ATTRIBUTE_ID   194
-#define DEFAULT_DATABASE_PATH  "/usr/share/hddtemp.db"
+#define DEFAULT_DATABASE_PATH  "/usr/share/hddtemp/hddtemp.db"
 #define MAX_LINE_LEN           1024
 
 struct harddrive_entry   *supported_drives = NULL;
 struct harddrive_entry   **last_entry = &supported_drives;
 struct harddrive_entry   *drive_entry;
