Index: advi-1.7.3/src/Makefile.am
===================================================================
--- advi-1.7.3.orig/src/Makefile.am
+++ advi-1.7.3/src/Makefile.am
@@ -84,13 +84,13 @@ OCAMLLIBS_BYT	= $(CAMLIMAGES_LIBS_BYTE) 
 
 advi$(EXEEXT): $(COBJS) $(OCAMLOPTOBJS)
 	$(OCAMLOPT) -o advi$(EXEEXT) \
-		$(OCAMLCFLAGS) $(AM_OCAMLCFLAGS) $(AM_LINKFLAGS) \
-		$(COBJS) $(OCAMLLIBS_OPT) $(OCAMLOPTOBJS)
+		$(OCAMLCFLAGS) $(AM_OCAMLCFLAGS) \
+		$(COBJS) $(OCAMLLIBS_OPT) $(OCAMLOPTOBJS) $(AM_LINKFLAGS)
 
 advi.byt$(EXEEXT): $(COBJS) $(OCAMLBYTOBJS)
 	$(OCAMLC) -o advi.byt$(EXEEXT) \
-		$(OCAMLCFLAGS) $(AM_OCAMLCFLAGS) $(AM_LINKFLAGS) -custom \
-		$(COBJS) $(OCAMLLIBS_BYT) $(OCAMLBYTOBJS)
+		$(OCAMLCFLAGS) $(AM_OCAMLCFLAGS) -custom \
+		$(COBJS) $(OCAMLLIBS_BYT) $(OCAMLBYTOBJS) $(AM_LINKFLAGS)
 
 install_hook: $(INSTALLEXEC) installdata
 	- texhash
