diff -ru xsnap-1.5.1.vanilla/po/Makefile xsnap-1.5.1/po/Makefile
--- xsnap-1.5.1.vanilla/po/Makefile	2006-11-07 07:58:59.000000000 +0100
+++ xsnap-1.5.1/po/Makefile	2006-11-08 15:01:17.000000000 +0100
@@ -7,7 +7,8 @@
 install:
 	for cat in $(CATALOGS) ; do msgfmt -o $$cat.mo -v $$cat.po ; done
 	for cat in $(CATALOGS) ; do \
-		cp -f $$cat.mo $(LOCALEDIR)/$$cat/LC_MESSAGES/xsnap.mo ; done	
+		mkdir -p $(DESTDIR)/$(LOCALEDIR)/$$cat/LC_MESSAGES; \
+		cp -f $$cat.mo $(DESTDIR)/$(LOCALEDIR)/$$cat/LC_MESSAGES/xsnap.mo ; done	
 
 clean:
 	rm -f core *~ *.mo
