--- oo_cvs/sw/source/core/txtnode/thints.cxx	Thu Nov  8 09:35:13 2001
+++ oo_stable1_cvs20020409_src/sw/source/core/txtnode/thints.cxx	Thu Apr 11 11:45:43 2002
@@ -1851,7 +1851,7 @@
 			( SFX_ITEM_SET == rNode.GetpSwAttrSet()->GetItemState( nWhich,
 				FALSE, &pParaItem ) ) && ( pParaItem == &pHint->GetAttr() );
 		BOOL bReplace = !( SETATTR_DONTREPLACE & nMode );
-		SwpHtStart_SAR *pTmpHints = 0;
+		::SwpHtStart_SAR *pTmpHints = 0;
 
 		USHORT i;
 		// Wir wollen zwar von nHtStart bis nMaxEnd, muessen aber ggf.
@@ -2027,7 +2027,7 @@
 									// Beim Einfuegen spaltet es sich selbst.
 										if( pHistory ) pHistory->Add( pOther );
 										if( !pTmpHints )
-											pTmpHints = new SwpHtStart_SAR();
+											pTmpHints = new ::SwpHtStart_SAR();
 										pTmpHints->C40_INSERT( SwTxtAttr, pOther,
 											pTmpHints->Count() );
 										Cut( i );
@@ -2076,7 +2076,7 @@
 									// Beim Einfuegen spaltet es sich selbst.
 										if( pHistory ) pHistory->Add( pOther );
 										if( !pTmpHints )
-											pTmpHints = new SwpHtStart_SAR();
+											pTmpHints = new ::SwpHtStart_SAR();
 										pTmpHints->C40_INSERT( SwTxtAttr, pOther,
 											pTmpHints->Count() );
 										Cut( i );
