diff -ur oo_cvs/config_office/configure.in oo_stable1_cvs20020411_src/config_office/configure.in
--- oo_cvs/config_office/configure.in	Mon Mar 25 10:07:39 2002
+++ oo_stable1_cvs20020411_src/config_office/configure.in	Fri Apr 12 19:55:09 2002
@@ -241,18 +241,18 @@
 if test "$_os" != "WINNT" && test -z "$with_gcc_home"; then
    AC_PROG_CC
 fi
-if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" -o \( \( "$_os" = "IRIX" -o "$_os" = "IRIX64" \) -a "$CC" = "gcc" \) ; then
+if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" -o \( \( "$_os" = "IRIX" -o "$_os" = "IRIX64" \) -a "$GCC" = "yes" \) ; then
    dnl ******************************************
    dnl Testing for GNU compiler and version...
    dnl ******************************************
    dnl Check whether the gnu gcc compiler is used.
    if test -z "$with_gcc_home"; then
-      if test "$CC" = "gcc"; then
-         AC_PATH_PROG(GCC, gcc)
-         COMPATH=`echo $GCC | $SED -n "s/\/gcc//p"`
-         _gcc_include_start=/usr/lib/gcc-lib
-         _gxx_include_start=/usr/include
-      fi
+
+      AC_PROG_CC
+      COMPATH=`echo $CC | $SED -n "s,/[^/]\+$,,p"`
+      _gcc_include_start=/usr/lib/gcc-lib
+      _gxx_include_start=/usr/include
+
    else
       dnl Compiler has been installed to its own directory and it's home is arg with_gcc_home
       _gcc_path="$with_gcc_home/bin/gcc"
@@ -1177,12 +1177,14 @@
 echo "\$MINGWIN32="$WITH_MINGWIN
 echo "\$with_use_shell="$with_use_shell
 echo "\$USE_GCC3="$USE_GCC3
+echo "\$CC="$CC
+echo "\$CXX="$CXX
 echo
 fi
 
 # Executing the set_soenv script to setup the environment variables.
 if test -z "$enable_check_only"; then
-   './set_soenv' $COMPATH $_gcc_include_path $_gxx_include_path $JAVA_HOME $TCSH $PERL $x_libraries $x_includes $_LOCAL_SOLENV $_LOCAL_SOLVER $upd $CYGWIN $STLPORT4 $enable_xprint $with_lang $with_asm_home $with_unzip_home $WITH_MINGWIN $with_use_shell $USE_GCC3
+   './set_soenv' $COMPATH $_gcc_include_path $_gxx_include_path $JAVA_HOME $TCSH $PERL $x_libraries $x_includes $_LOCAL_SOLENV $_LOCAL_SOLVER $upd $CYGWIN $STLPORT4 $enable_xprint $with_lang $with_asm_home $with_unzip_home $WITH_MINGWIN $with_use_shell $USE_GCC3 $CC $CXX
 else
    echo
    echo Test Complete
diff -ur oo_cvs/config_office/set_soenv.1 oo_stable1_cvs20020411_src/config_office/set_soenv.1
--- oo_cvs/config_office/set_soenv.1	Wed Mar 20 19:10:58 2002
+++ oo_stable1_cvs20020411_src/config_office/set_soenv.1	Fri Apr 12 16:45:11 2002
@@ -40,7 +40,7 @@
 #
 # Set this value equal to the corresponding amount of
 # command line arguments. 
-my $numArgs = 20;
+my $numArgs = 22;
 # Determining the amount of arguments in the call.
 my $numEnter = $#ARGV + 1;
 if ( $numEnter != $numArgs ) 
@@ -96,7 +96,7 @@
      $SOLARINC, $LOCALINI, $MAC_LIB, $PATHEXTRA, $FRAMEWORKSHOME, $COMEX, $MULTITHREAD_OBJ, $PERL, 
      $COMP_ENV, $IENV, $ILIB, $WIN_INCLUDE, $JAVAHOME, $WIN_LIB, $WIN_HOME, 
      $BISON_HAIRY, $BISON_SIMPLE, $TEMP, $COMMON_BUILD_TOOLS, $WIN_GREP, $WIN_FIND, $WIN_LS,
-     $WIN_GNUCOPY, $WIN_TOUCH, $STLPORT4, $USE_GCC3 );
+     $WIN_GNUCOPY, $WIN_TOUCH, $STLPORT4, $USE_GCC3, $CC, $CXX );
 #
 #-------------------------------------------
 # IId. Declaring the aliases.
@@ -146,6 +146,8 @@
 $MINGWIN32	= $ARGV [ 17 ];     # use MINGWIN32
 $use_shell	= $ARGV [ 18 ];     # preferred shell
 $USE_GCC3	= $ARGV [ 19 ];     # use gcc3 (only for unxlngi4 at the moment)
+$CC		= $ARGV [ 20 ];     # C compiler
+$CXX		= $ARGV [ 21 ];     # C++ compiler
 #
 #---------------------------------------------------------------
 # IIIb. Initialising the variables for the system commands, etc.
@@ -1464,6 +1466,8 @@
 ToFile( "SOLARINC",          $SOLARINC,          "e" );
 ToFile( "COMP_ENV",          $COMP_ENV,          "e" );
 ToFile( "JAVAHOME",          $JAVAHOME,          "e" );
+ToFile( "CCCOMP",            $CC,                "e" );
+ToFile( "CXXCOMP",           $CXX,               "e" );
 ToFile( "BISON_HAIRY",       $BISON_HAIRY,       "e" );
 ToFile( "BISON_SIMPLE",      $BISON_SIMPLE,      "e" );
 ToFile( "COMMON_BUILD_TOOLS",$COMMON_BUILD_TOOLS,"e" );
