diff -Nurd oo_cvs/config_office/configure.in oo_stable1_cvs20020325_src/config_office/configure.in
--- oo_cvs/config_office/configure.in	Mon Mar 25 10:07:39 2002
+++ oo_stable1_cvs20020325_src/config_office/configure.in	Tue Mar 26 17:39:20 2002
@@ -277,13 +277,13 @@
 
    if test $GCC; then
       AC_MSG_CHECKING([the GNU gcc compiler version])
-      _gcc_releasetype=`$GCC --version | $AWK -F"-" '{ print \$1 }'`
+      _gcc_releasetype=`$CC --version | $AWK -F"-" '{ print \$1 }'`
       if test "$_gcc_releasetype" = "egcs"; then
          AC_MSG_WARN([found egcs- version,  use (stable=gcc-) version of the compiler])
          AC_MSG_WARN([found egcs- version,  use (stable=gcc-) version of the compiler]) >> warn
-         _gcc_version=`$GCC --version | $AWK -F"-" '{ print \$2 }'`
+         _gcc_version=`$CC --version | $AWK -F"-" '{ print \$2 }'`
       else
-         _gcc_version=`$GCC --version`
+         _gcc_version=`$CC --version`
       fi
       _gcc_major=`echo $_gcc_version | $AWK -F. '{ print \$1 }'` 
       if test -n "$enable_gcc3"; then 
@@ -296,13 +296,13 @@
            AC_MSG_WARN([found version "$_gcc_version", use version 2.95.2+ of the gcc compiler]) >> warn
            COMPATH="NO_GCC"
         else
-           _gcc_middle=`$GCC --version | $AWK -F. '{ if ($2 != 95) print "false"; else print "true" }'` 
+           _gcc_middle=`$CC --version | $AWK -F. '{ if ($2 != 95) print "false"; else print "true" }'` 
            if test "$_gcc_middle" = "false"; then
               AC_MSG_WARN([found version "$_gcc_version", use version 2.95.2+ of the gcc compiler])
               AC_MSG_WARN([found version "$_gcc_version", use version 2.95.2+ of the gcc compiler]) >> warn
               COMPATH="NO_GCC"
            else
-              _gcc_minor=`$GCC --version | $AWK -F. '{ if ($3 < 2) print "false"; else print "true" }'` 
+              _gcc_minor=`$CC --version | $AWK -F. '{ if ($3 < 2) print "false"; else print "true" }'` 
               if test "$_gcc_minor" = "false"; then
                  AC_MSG_WARN([found version "$_gcc_version", use version 2.95.2+ of the gcc compiler])
                  AC_MSG_WARN([found version "$_gcc_version", use version 2.95.2+ of the gcc compiler]) >> warn
