--- oo-src/product/inc/scp/udk_basefiles.scp	Fri Mar 15 12:58:28 2002
+++ oo_stable1_cvs20020414_src/product/inc/scp/udk_basefiles.scp	Mon Apr 15 08:25:49 2002
@@ -10,6 +10,7 @@
     Styles		= (PACKED, SETUPZIP);
 End
 
+#ifdef NOTHING
 File gid_File_Lib_Stdc
     BIN_FILE_BODY;
     Name		= STRING(libstdc++.so.3.0.1);
@@ -23,6 +24,7 @@
     Name        = STRING(libstdc++.so.3);
     Styles      = (NETWORK,RELATIVE);
 End
+#endif
 #endif
 
 File gid_File_Lib_Sal
