--- oo_641c_src/tools/source/datetime/ttime.cxx.gcc3	Mon Jun 11 19:17:46 2001
+++ oo_641c_src/tools/source/datetime/ttime.cxx	Thu Feb 21 19:44:52 2002
@@ -73,7 +73,7 @@
 #include <dos.h>
 #elif defined UNX
 #include <limits.h>
-#ifdef IRIX
+#if defined( IRIX ) || defined( UNX )
 #include <unistd.h>
 #endif
 #include <sys/times.h>
@@ -493,7 +493,13 @@
 
 	if ( !nImplTicksPerSecond )
 	{
+#if defined( CLK_TCK )
 		nImplTicksPerSecond = CLK_TCK;
+#elif defined( _SC_CLK_TCK )
+		nImplTicksPerSecond = sysconf(_SC_CLK_TCK);
+#else
+#error "I don't know how to get CLK_TCK, you lose."
+#endif
 		dImplTicksPerSecond = nImplTicksPerSecond;
 		dImplTicksULONGMAX	= (double)(ULONG)ULONG_MAX;
 	}
