diff -Nru gnokii-9999.orig/po/POTFILES.in gnokii-9999/po/POTFILES.in
--- gnokii-9999.orig/po/POTFILES.in	2009-08-16 11:48:52.000000000 +0200
+++ gnokii-9999/po/POTFILES.in	2009-08-16 11:56:02.000000000 +0200
@@ -40,6 +40,7 @@
 gnokii/gnokii-dial.c
 gnokii/gnokii-file.c
 gnokii/gnokii-logo.c
+gnokii/gnokii-mms.c
 gnokii/gnokii-monitor.c
 gnokii/gnokii-other.c
 gnokii/gnokii-phonebook.c
@@ -58,6 +59,7 @@
 smsd/mysql.c
 smsd/pq.c
 smsd/smsd.c
+utils/gnokii-configure.glade
 utils/mgnokiidev.c
 xgnokii/xgnokii.desktop.in
 xgnokii/xgnokii.c
