--- src/kernel/qapplication_x11.cpp.sav	2003-09-28 20:59:02.000000000 +0200
+++ src/kernel/qapplication_x11.cpp	2003-10-05 22:47:42.000000000 +0200
@@ -3148,9 +3148,12 @@ int QApplication::x11ClientMessage(QWidg
 		QWhatsThis::enterWhatsThisMode();
 #endif // QT_NO_WHATSTHIS
 	    } else if ( a == qt_net_wm_ping ) {
-		event->xclient.window = QPaintDevice::x11AppRootWindow( w->x11Screen() );
-		XSendEvent( event->xclient.display, event->xclient.window,
-			    False, SubstructureNotifyMask|SubstructureRedirectMask, event );
+                // avoid receiving replies
+                if( event->xclient.window != QPaintDevice::x11AppRootWindow( w->x11Screen())) {
+		    event->xclient.window = QPaintDevice::x11AppRootWindow( w->x11Screen() );
+		    XSendEvent( event->xclient.display, event->xclient.window,
+			        False, SubstructureNotifyMask|SubstructureRedirectMask, event );
+                }
 	    }
 	} else if ( event->xclient.message_type == qt_qt_scrolldone ) {
 	    widget->translateScrollDoneEvent(event);
