--- wepdecrypt-0.7/src/Makefile.orig	2006-05-25 23:27:29.000000000 +1000
+++ wepdecrypt-0.7/src/Makefile	2006-05-25 23:27:38.000000000 +1000
@@ -4,7 +4,7 @@
 
 
 wepdecrypt: 	wepdecrypt.o rc4.o wepfilter.o log.o modes.o misc.o verify.o keygen.o localkeygen.o wepserver.o wepclient.o md5_digest.o
-	$(LD) $(LDFLAGS) -o $@ wepdecrypt.o rc4.o wepfilter.o log.o\
+	$(LD) $(LDFLAGS) -o $@ wepdecrypt.o rc4.o wepfilter.o log.o \
 	modes.o misc.o md5_digest.o  wepserver.o wepclient.o verify.o keygen.o localkeygen.o $(LIBS)
 	strip wepdecrypt
 
