diff -urN LVM.orig/1.0.7/tools/Makefile.in LVM/1.0.7/tools/Makefile.in
--- LVM.orig/1.0.7/tools/Makefile.in	2003-02-06 09:05:29.000000000 -0600
+++ LVM/1.0.7/tools/Makefile.in	2003-10-25 17:33:23.151387547 -0500
@@ -53,7 +53,7 @@
 
 $(ALL_OBJECTS):	Makefile lib/liblvm.h lvm_user.h
 
-all_this:	$(ALL_OBJECTS) $(ALL) $(ALL_SOURCES) $(ALL_HEADERS) $(OTHER_DEP) $(LIBS)
+all_this:	$(ALL_OBJECTS) $(ALL) $(ALL_SOURCES) $(ALL_HEADERS) $(OTHER_DEP) $(LIBS) vgscan.static vgchange.static
 		@echo ""
 		@echo "Done Tools parts"
 		@echo ""
@@ -70,6 +70,12 @@
 
 l:	list
 
+vgscan.static: vgscan.o
+	$(CC) vgscan.o $(LDFLAGS) $(LOADLIBES) $(LDLIBS) -static -o vgscan.static
+                                                                                                       
+vgchange.static: vgchange.o
+	$(CC) vgchange.o $(LDFLAGS) $(LOADLIBES) $(LDLIBS) -static -o vgchange.static
+
 install-bin:	$(ALL_TARGETS) $(ALL_SCRIPTS)
 	@echo "*** Installing $(ALL_TARGETS) $(ALL_SCRIPTS) in $(DESTDIR)$(sbindir) ***"
 	@for f in $(ALL_TARGETS); \
