--- Makefile.orig	2003-11-23 20:15:23.000000000 -0500
+++ Makefile	2003-11-23 20:16:46.000000000 -0500
@@ -176,6 +176,15 @@
 			-finline-functions -ffast-math' \
 		mpg123-make
 
+linux-generic:
+	$(MAKE) CC=gcc LDFLAGS= \
+		OBJECTS='decode.o dct64.o audio_oss.o' \
+		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -Wall \
+			-DOSS \
+			-fomit-frame-pointer -funroll-all-loops \
+			-finline-functions -ffast-math' \
+		mpg123-make
+
 linux-pentium:
 	$(MAKE) CC=gcc LDFLAGS= \
 		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
