--- source/screen.c.orig	2004-06-20 03:56:13.636821920 +0200
+++ source/screen.c	2004-06-20 03:56:17.696204800 +0200
@@ -1819,7 +1819,7 @@
 				 * Otherwise, we just use /set continued_line, 
 				 * whatever it is.
 				 */
-				else if ((!cont || !*cont) && *cont_ptr)
+				else if ((!cont || !*cont) && cont_ptr)
 					cont = LOCAL_COPY(cont_ptr);
 
 				cont_free = cont = normalize_string(cont, 0);
