diff -urN usbmuxd-1.0.0-rc2.old/CMakeLists.txt usbmuxd-1.0.0-rc2/CMakeLists.txt
--- usbmuxd-1.0.0-rc2.old/CMakeLists.txt	2009-10-31 02:30:59.000000000 +0100
+++ usbmuxd-1.0.0-rc2/CMakeLists.txt	2009-12-07 01:13:46.000000000 +0100
@@ -16,13 +16,15 @@
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/Modules/")
 
+set( CMAKE_C_FLAGS "-Wall -O2 -g" )
+
 add_subdirectory (libusbmuxd)
 add_subdirectory (daemon)
 add_subdirectory (tools)
 add_subdirectory (udev)
 
 # pkg-config
-CONFIGURE_FILE ("${CMAKE_SOURCE_DIR}/libusbmuxd.pc.in" "${CMAKE_CURRENT_BINARY_DIR}/libusbmuxd.pc")
+CONFIGURE_FILE ("${CMAKE_CURRENT_SOURCE_DIR}/libusbmuxd.pc.in" "${CMAKE_CURRENT_BINARY_DIR}/libusbmuxd.pc")
 # install pkg-config file
 INSTALL (FILES "${CMAKE_CURRENT_BINARY_DIR}/libusbmuxd.pc" DESTINATION lib${LIB_SUFFIX}/pkgconfig/)
 
diff -urN usbmuxd-1.0.0-rc2.old/daemon/CMakeLists.txt usbmuxd-1.0.0-rc2/daemon/CMakeLists.txt
--- usbmuxd-1.0.0-rc2.old/daemon/CMakeLists.txt	2009-10-31 02:30:59.000000000 +0100
+++ usbmuxd-1.0.0-rc2/daemon/CMakeLists.txt	2009-12-07 01:13:34.000000000 +0100
@@ -6,7 +6,8 @@
 include_directories (${CMAKE_SOURCE_DIR}/daemon)
 include_directories (${CMAKE_SOURCE_DIR}/libusbmuxd)
 
-add_definitions(-Wall -O2 -g -DUSBMUXD_DAEMON -DUSBMUXD_VERSION="${USBMUXD_VERSION}")
+set( CMAKE_C_FLAGS "-Wall -O2 -g" )
+add_definitions(-DUSBMUXD_DAEMON -DUSBMUXD_VERSION="${USBMUXD_VERSION}")
 add_executable(usbmuxd main.c usb-linux.c log.c ../common/utils.c device.c client.c)
 target_link_libraries(usbmuxd ${LIBS})
 
diff -urN usbmuxd-1.0.0-rc2.old/libusbmuxd/CMakeLists.txt usbmuxd-1.0.0-rc2/libusbmuxd/CMakeLists.txt
--- usbmuxd-1.0.0-rc2.old/libusbmuxd/CMakeLists.txt	2009-10-31 02:30:59.000000000 +0100
+++ usbmuxd-1.0.0-rc2/libusbmuxd/CMakeLists.txt	2009-12-07 01:31:06.000000000 +0100
@@ -1,8 +1,8 @@
 include_directories (${CMAKE_SOURCE_DIR}/common)
+find_package(Threads)
 
-add_library (libusbmuxd SHARED libusbmuxd.c sock_stuff.c ../common/utils.c)
-find_library (PTHREAD pthread)
-target_link_libraries (libusbmuxd ${PTHREAD})
+add_library (libusbmuxd SHARED libusbmuxd.c sock_stuff.c ${CMAKE_SOURCE_DIR}/common/utils.c)
+target_link_libraries (libusbmuxd ${CMAKE_THREAD_LIBS_INIT})
 # 'lib' is a UNIXism, the proper CMake target is usbmuxd
 # But we can't use that due to the conflict with the usbmuxd daemon,
 # so instead change the library output base name to usbmuxd here
diff -urN usbmuxd-1.0.0-rc2.old/tools/CMakeLists.txt usbmuxd-1.0.0-rc2/tools/CMakeLists.txt
--- usbmuxd-1.0.0-rc2.old/tools/CMakeLists.txt	2009-10-31 02:30:59.000000000 +0100
+++ usbmuxd-1.0.0-rc2/tools/CMakeLists.txt	2009-12-07 01:24:27.000000000 +0100
@@ -1,8 +1,7 @@
 include_directories (${CMAKE_SOURCE_DIR}/libusbmuxd)
 link_directories (${CMAKE_BINARY_DIR}/libusbmuxd)
 
-add_definitions(-Wall -O2 -g)
 add_executable(iproxy iproxy.c)
 target_link_libraries(iproxy libusbmuxd pthread)
 
-install(TARGETS iproxy RUNTIME DESTINATION bin)
\ Chybí znak konce řádku na konci souboru
+install(TARGETS iproxy RUNTIME DESTINATION bin)
