diff -Nur file-browser-applet-0.6.5/CMakeLists.txt file-browser-applet-0.6.5_patched/CMakeLists.txt
--- file-browser-applet-0.6.5/CMakeLists.txt	2009-12-21 19:42:30.000000000 +0200
+++ file-browser-applet-0.6.5_patched/CMakeLists.txt	2010-02-13 17:40:46.000000000 +0200
@@ -3,7 +3,7 @@
 PROJECT (FILE_BROWSER_APPLET C)
 
 #compiler flags
-SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -g -Wall -ansi")
+SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -ansi")
 SET (EXEC file-browser-applet)
 
 #install path options
