--- gcc-3.3.5/gcc/f/com.h.orig	2004-12-30 18:43:49.520799009 -0800
+++ gcc-3.3.5/gcc/f/com.h	2004-12-30 18:44:17.719152427 -0800
@@ -233,7 +233,7 @@ void ffecom_finish_decl (tree decl, tree
 void ffecom_finish_progunit (void);
 tree ffecom_get_invented_identifier (const char *pattern, ...)
   ATTRIBUTE_PRINTF_1;
-ffeinfoKindtype ffecom_gfrt_basictype (ffecomGfrt ix);
+ffeinfoBasictype ffecom_gfrt_basictype (ffecomGfrt ix);
 ffeinfoKindtype ffecom_gfrt_kindtype (ffecomGfrt ix);
 void ffecom_init_0 (void);
 void ffecom_init_2 (void);
