###MY_VER_RANGE  [5.0.6_beta,)
--- mysql.old/mysys/default.c	2005-05-22 02:35:25.000000000 +0200
+++ mysql/mysys/default.c	2005-05-22 02:36:17.000000000 +0200
@@ -823,19 +823,19 @@
     5. ""
 
   On Novell NetWare, this is:
-    1. sys:/etc/
+    1. sys:/etc/mysql/
     2. getenv(DEFAULT_HOME_ENV)
     3. ""
 
   On OS/2, this is:
     1. getenv(ETC)
-    2. /etc/
+    2. /etc/mysql/
     3. getenv(DEFAULT_HOME_ENV)
     4. ""
     5. "~/"
 
   Everywhere else, this is:
-    1. /etc/
+    1. /etc/mysql/
     2. getenv(DEFAULT_HOME_ENV)
     3. ""
     4. "~/"
@@ -860,13 +860,13 @@
 #endif
 
 #elif defined(__NETWARE__)
-  *ptr++= "sys:/etc/";
+  *ptr++= "sys:/etc/mysql/";
 #else
 #if defined(__EMX__) || defined(OS2)
   if ((env= getenv("ETC")))
     *ptr++= env;
 #endif
-  *ptr++= "/etc/";
+  *ptr++= "/etc/mysql/";
 #endif
   if ((env= getenv(STRINGIFY_ARG(DEFAULT_HOME_ENV))))
     *ptr++= env;
