--- mysql-4.0.21/scripts/mysql_install_db.sh.orig	
+++ mysql-4.0.21/scripts/mysql_install_db.sh	
@@ -371,6 +371,7 @@
   echo "To do so, start the server, then issue the following commands:"
-  echo "$bindir/mysqladmin -u root password 'new-password'"
+  echo "/etc/init.d/mysql start (you have to start the server first!)"
   echo "$bindir/mysqladmin -u root -h $hostname password 'new-password'"
+  echo "$bindir/mysqladmin -u root password 'new-password'"
   echo "See the manual for more instructions."
   #
   # Print message about upgrading unless we have created a new db table.
