diff -ur mysql-4.0.18.orig/scripts/mysqlbug.sh mysql-4.0.18/scripts/mysqlbug.sh
--- mysql-4.0.18.orig/scripts/mysqlbug.sh	2004-02-10 10:15:54.000000000 -0800
+++ mysql-4.0.18/scripts/mysqlbug.sh	2004-04-13 09:00:53.000000000 -0700
@@ -254,9 +254,9 @@
 if cmp -s $TEMP $TEMP.x
 then
   echo "File not changed, no bug report submitted."
-  cp $TEMP /tmp/failed-mysql-bugreport
+  mv -f $TEMP /tmp/failed-mysql-bugreport
   echo "The raw bug report exists in /tmp/failed-mysql-bugreport"
-  echo "If you use this remember that the first lines of the report now is a lie.."
+  echo "If you use this remember that the first lines of the report are now a lie.."
   exit 1
 fi
 
