diff -ruN mysql-4.0.17/configure.in mysql-4.0.17-nptl/configure.in
--- mysql-4.0.17/configure.in	2003-12-14 16:23:49.000000000 -0500
+++ mysql-4.0.17-nptl/configure.in	2004-02-07 12:00:00.715528784 -0500
@@ -1191,7 +1191,10 @@
 then
   # Look for LinuxThreads.
   AC_MSG_CHECKING("LinuxThreads")
-    res=`grep Linuxthreads /usr/include/pthread.h 2>/dev/null | wc -l`
+   # res=`grep Linuxthreads /usr/include/pthread.h 2>/dev/null | wc -l`
+   # All gentoo systems support this stuff, and this is an improper detection
+   # for LinuxThreads anyhow, always set the result to 1
+   res=1
     if test "$res" -gt 0
     then
       AC_MSG_RESULT("Found")
