diff -uNr mysql-3.23.51.orig/sql/mysqld.cc mysql-3.23.51/sql/mysqld.cc
--- mysql-3.23.51.orig/sql/mysqld.cc	Mon Jun  3 06:39:02 2002
+++ mysql-3.23.51/sql/mysqld.cc	Thu Jun 20 16:11:47 2002
@@ -91,9 +91,9 @@
 int deny_severity = LOG_WARNING;
 
 #ifdef __linux__
-#define my_fromhost(A)	   fromhost()
-#define my_hosts_access(A) hosts_access()
-#define my_eval_client(A)  eval_client()
+#define my_fromhost(A)	   fromhost(A)
+#define my_hosts_access(A) hosts_access(A)
+#define my_eval_client(A)  eval_client(A)
 #else
 #define my_fromhost(A)	   fromhost(A)
 #define my_hosts_access(A) hosts_access(A)
