--- mysql-4.0.23/scripts/mysql_install_db.sh.orig	2004-12-23 02:28:09.755109232 -0800
+++ mysql-4.0.23/scripts/mysql_install_db.sh	2004-12-23 02:31:32.204332272 -0800
@@ -368,10 +368,13 @@
     echo
   fi
   echo "PLEASE REMEMBER TO SET A PASSWORD FOR THE MySQL root USER !"
-  echo "To do so, start the server, then issue the following commands:"
-  echo "$bindir/mysqladmin -u root password 'new-password'"
-  echo "$bindir/mysqladmin -u root -h $hostname password 'new-password'"
-  echo "See the manual for more instructions."
+  echo "To do so, issue the following commands to start the server"
+  echo "and change the applicable passwords:"
+  echo "  /etc/init.d/mysql start"
+  echo "  $bindir/mysqladmin -u root -h $hostname password 'new-password'"
+  echo "  $bindir/mysqladmin -u root password 'new-password'"
+  echo "Depending on your configuration, a -p option may be needed"
+  echo "in the last command. See the manual for more details."
   #
   # Print message about upgrading unless we have created a new db table.
   if test -z "$c_d"
