diff -u -r mysql-4.0.17.orig/libmysql_r/Makefile.in mysql-4.0.17/libmysql_r/Makefile.in
--- mysql-4.0.17.orig/libmysql_r/Makefile.in	2003-12-14 13:24:02.000000000 -0800
+++ mysql-4.0.17/libmysql_r/Makefile.in	2003-12-25 13:39:53.000000000 -0800
@@ -209,8 +209,8 @@
 CHARSET_OBJS = @CHARSET_OBJS@
 
 target = libmysqlclient_r.la
-target_defs = -DDONT_USE_RAID @LIB_EXTRA_CCFLAGS@
-LIBS = @LIBS@ @openssl_libs@
+target_defs = -DDONT_USE_RAID @LIB_EXTRA_CCFLAGS@ -DMYSQL_CLIENT
+LIBS = @CLIENT_LIBS@ @openssl_libs@
 
 INCLUDES = @MT_INCLUDES@ -I$(srcdir)/../include -I../include \
 		-I$(srcdir)/.. -I$(top_srcdir) -I.. $(openssl_includes)
