diff -ur mysql-4.0.16.orig/scripts/mysqld_safe.sh mysql-4.0.16/scripts/mysqld_safe.sh
--- mysql-4.0.16.orig/scripts/mysqld_safe.sh	2003-10-16 22:29:17.000000000 -0700
+++ mysql-4.0.16/scripts/mysqld_safe.sh	2003-10-26 15:22:17.000000000 -0800
@@ -88,43 +88,15 @@
   done
 }
 
-
-MY_PWD=`pwd`
-# Check if we are starting this relative (for the binary release)
-if test -d $MY_PWD/data/mysql -a -f ./share/mysql/english/errmsg.sys -a \
- -x ./bin/mysqld
-then
-  MY_BASEDIR_VERSION=$MY_PWD		# Where bin, share and data are
-  ledir=$MY_BASEDIR_VERSION/bin		# Where mysqld is
-  DATADIR=$MY_BASEDIR_VERSION/data
-  if test -z "$defaults"
-  then
-    defaults="--defaults-extra-file=$MY_BASEDIR_VERSION/data/my.cnf"
-  fi
-# Check if this is a 'moved install directory'
-elif test -f ./var/mysql/db.frm -a -f ./share/mysql/english/errmsg.sys -a \
- -x ./libexec/mysqld
-then
-  MY_BASEDIR_VERSION=$MY_PWD		# Where libexec, share and var are
-  ledir=$MY_BASEDIR_VERSION/libexec	# Where mysqld is
-  DATADIR=$MY_BASEDIR_VERSION/var
-else
-  MY_BASEDIR_VERSION=@prefix@
-  DATADIR=@localstatedir@
-  ledir=@libexecdir@
-fi
+MY_BASEDIR_VERSION=@prefix@
+DATADIR=@localstatedir@
+ledir=@libexecdir@
 
 safe_mysql_unix_port=${mysql_unix_port:-${MYSQL_UNIX_PORT:-@MYSQL_UNIX_ADDR@}}
 user=@MYSQLD_USER@
 niceness=0
 
-# Use the mysqld-max binary by default if the user doesn't specify a binary
-if test -x $ledir/mysqld-max
-then
-  MYSQLD=mysqld-max
-else
-  MYSQLD=mysqld
-fi
+MYSQLD=mysqld
 
 # these rely on $DATADIR by default, so we'll set them later on
 pid_file=
