gcc 4 fixes from upstream

--- busybox-1.00/archival/gzip.c
+++ busybox-1.00/archival/gzip.c
@@ -2162,7 +2162,7 @@
  * Construct the Huffman tree for the bit lengths and return the index in
  * bl_order of the last bit length code to send.
  */
-static const int build_bl_tree()
+static int build_bl_tree()
 {
 	int max_blindex;	/* index of last bit length code of non zero freq */
 
--- busybox-1.00/networking/ifupdown.c
+++ busybox-1.00/networking/ifupdown.c
@@ -150,7 +150,9 @@
 
 static char no_act = 0;
 static char verbose = 0;
+#ifndef __USE_GNU
 static char **environ = NULL;
+#endif
 
 #ifdef CONFIG_FEATURE_IFUPDOWN_IP
 
