Index: insmod.c
===================================================================
RCS file: /var/cvs/busybox/modutils/insmod.c,v
retrieving revision 1.125
retrieving revision 1.126
diff -u -r1.125 -r1.126
--- busybox/modutils/insmod.c	2 Sep 2004 23:03:25 -0000	1.125
+++ busybox/modutils/insmod.c	26 Dec 2004 09:13:32 -0000	1.126
@@ -109,6 +109,14 @@
 #endif
 
 
+/* Alpha */
+#if defined(__alpha__)   
+#define MATCH_MACHINE(x) (x == EM_ALPHA)
+#define SHT_RELM       SHT_RELA
+#define Elf64_RelM     Elf64_Rela
+#define ELFCLASSM      ELFCLASS64
+#endif
+
 /* ARM support */
 #if defined(__arm__)
 #define MATCH_MACHINE(x) (x == EM_ARM)
@@ -135,6 +143,19 @@
 #endif
 #endif
 
+/* PA-RISC / HP-PA */
+#if defined(__hppa__)
+#define MATCH_MACHINE(x) (x == EM_PARISC)
+#define SHT_RELM       SHT_RELA
+#if defined(__LP64__)
+#define Elf64_RelM     Elf64_Rela
+#define ELFCLASSM      ELFCLASS64
+#else
+#define Elf32_RelM     Elf32_Rela
+#define ELFCLASSM      ELFCLASS32
+#endif
+#endif
+
 /* H8/300 */
 #if defined(__H8300H__) || defined(__H8300S__)
 #define MATCH_MACHINE(x) (x == EM_H8_300)
