diff -ru lgeneral-1.2.orig/configure.in lgeneral-1.2/configure.in
--- lgeneral-1.2.orig/configure.in	2010-09-28 13:44:39.000000000 -0400
+++ lgeneral-1.2/configure.in	2010-11-30 22:37:30.245177173 -0500
@@ -106,13 +106,8 @@
 
 dnl find out whether we can determine the path of the running executable
 ac_paths_determ=generic
-AC_MSG_CHECKING(if executable's path can be determined at runtime)
-readlink /proc/$$/exe >/dev/null 2>&1 && ac_paths_determ=linux
-ac_can_determ_paths=yes
-if test x$ac_paths_determ = xgeneric ; then
   ac_can_determ_paths=no
   AC_DEFINE_UNQUOTED(PATHS_GENERIC, 1, [Defined if detecting executable's path name is not supported at runtime.])
-fi
 AC_MSG_RESULT($ac_can_determ_paths)
 AC_MSG_RESULT(executable's path determination strategy... $ac_paths_determ)
 AM_CONDITIONAL(compile_paths_linux, test x$ac_paths_determ = xlinux)
@@ -150,7 +145,7 @@
 AC_SUBST(dl_lib_flag)
 
 dnl installation path
-inst_dir=$datadir/games/lgeneral
+inst_dir=$datadir/lgeneral
 inst_flag="-DINSTALLDIR=\\\"$inst_dir\\\""
 
 dis_flag=""
