--- src/kmplayer.desktop.o	2009-02-15 01:45:40.000000000 +0100
+++ src/kmplayer.desktop	2009-02-15 01:45:51.000000000 +0100
@@ -1,10 +1,9 @@
 # KDE Config File
 [Desktop Entry]
-Encoding=UTF-8
 Type=Application
-Exec=kmplayer -caption "%c" %i %m %U
-Icon=kmplayer.png
-DocPath=kmplayer/index.html
+Exec=kmplayer -caption "%c" %i %U
+Icon=kmplayer
+X-DocPath=kmplayer/index.html
 Comment=KDE interface for MPlayer
 Comment[af]=KDE koppelvlak na MPlayer
 Comment[ar]=واجهة KDE لِــ MPlayer
