diff -aur monodevelop-0.12-orig/configure.in monodevelop-0.12/configure.in
--- monodevelop-0.12-orig/configure.in	2006-09-05 12:36:06.000000000 -0700
+++ monodevelop-0.12/configure.in	2006-10-28 07:42:43.000000000 -0700
@@ -120,7 +120,7 @@
 AC_ARG_ENABLE(java,
 	AC_HELP_STRING([--enable-java],
 		[enable support for Java via ikvm [default=no]]),
-		[PKG_CHECK_MODULES(IKVM, ikvm >= $IKVM_REQUIRED_VERSION, enable_java=yes, enable_java=no)],
+		[PKG_CHECK_MODULES(IKVM, ikvm >= $IKVM_REQUIRED_VERSION, enable_java="$enableval", enable_java=no)],
 		enable_java=no)
 AM_CONDITIONAL(ENABLE_JAVA, test x$enable_java = xyes)
 
@@ -128,7 +128,7 @@
 AC_ARG_ENABLE(boo,
 	AC_HELP_STRING([--enable-boo],
 		[enable support for boo [default=no]]),
-		[PKG_CHECK_MODULES(BOO, boo >= $BOO_REQUIRED_VERSION, enable_boo=yes, enable_boo=no)],
+		[PKG_CHECK_MODULES(BOO, boo >= $BOO_REQUIRED_VERSION, enable_boo="$enableval", enable_boo=no)],
 		enable_boo=no)
 
 AM_CONDITIONAL(ENABLE_BOO, test x$enable_boo = xyes)
diff -aur monodevelop-0.12-orig/monodevelop.in monodevelop-0.12/monodevelop.in
--- monodevelop-0.12-orig/monodevelop.in	2006-09-05 12:36:06.000000000 -0700
+++ monodevelop-0.12/monodevelop.in	2006-10-28 07:42:03.000000000 -0700
@@ -24,8 +24,10 @@
 	MOZILLA_HOME=$(grep -h GRE_PATH= /etc/gre.d/*.conf | cut -d '"' -f 2 -d = | head -n 1)
 elif [ $(which mozilla) ] && grep MOZILLA_FIVE_HOME= "$(which mozilla)" > /dev/null ; then
 	MOZILLA_HOME=$(grep MOZILLA_FIVE_HOME= $(which mozilla) | cut -d '"' -f 2)
-elif [ $(which firefox) ] && grep MOZILLA_FIVE_HOME= "$(which firefox)" > /dev/null ; then
-	MOZILLA_HOME=$(grep MOZILLA_FIVE_HOME= $(which firefox) | cut -d '"' -f 2)
+elif [ $(which firefox) ] && grep MOZILLA_LIBDIR= "$(which firefox)" > /dev/null ; then
+	MOZILLA_HOME=$(grep MOZILLA_LIBDIR= $(which firefox) | cut -d '=' -f 2)
+elif [ $(which seamonkey) ] && grep MOZILLA_LIBDIR= "$(which seamonkey)" > /dev/null ; then
+	MOZILLA_HOME=$(grep MOZILLA_LIBDIR= $(which seamonkey) | cut -d '=' -f 2)
 else
 	echo "Cannot find mozilla installation directory. Please set MOZILLA_FIVE_HOME to your mozilla directory"
 	exit 1
