diff -aur monodevelop-0.8-orig/Extras/VersionControl/VersionControl/Subversion.cs monodevelop-0.8/Extras/VersionControl/VersionControl/Subversion.cs
--- monodevelop-0.8-orig/Extras/VersionControl/VersionControl/Subversion.cs	2005-09-23 13:11:15.000000000 -0700
+++ monodevelop-0.8/Extras/VersionControl/VersionControl/Subversion.cs	2005-11-12 22:54:54.000000000 -0800
@@ -474,7 +474,7 @@
 			ArrayList statuses;
 			public StatusCollector(ArrayList statuses) { this.statuses = statuses; }
 			public void Func(IntPtr baton, IntPtr path, ref svn_wc_status_t status) {
-				if (status.to__svn_wc_entry_t == IntPtr.Zero)
+				if (status.to_svn_wc_entry_t == IntPtr.Zero)
 					return;				
 				string pathstr = Marshal.PtrToStringAnsi(path);
 				statuses.Add(new StatusEnt(status, pathstr));
@@ -586,18 +586,18 @@
 
 			internal StatusEnt(svn_wc_status_t status, string localpath) {
 				LocalFilePath = localpath;
-				TextStatus = (NodeStatus)status.svn_wc_status_kind__text;
-				PropsStatus = (NodeStatus)status.svn_wc_status_kind__props;
+				TextStatus = (NodeStatus)status.svn_wc_status_kind_text;
+				PropsStatus = (NodeStatus)status.svn_wc_status_kind_props;
 				Locked = status.locked != 0;
 				Copied = status.copied != 0;
 				Switched = status.switched != 0;
-				RemoteTextStatus = (NodeStatus)status.svn_wc_status_kind__text__repo;
-				RemotePropsStatus = (NodeStatus)status.svn_wc_status_kind__props__repo;
+				RemoteTextStatus = (NodeStatus)status.svn_wc_status_kind_text_repo;
+				RemotePropsStatus = (NodeStatus)status.svn_wc_status_kind_props_repo;
 				
-				if (status.to__svn_wc_entry_t == IntPtr.Zero)
+				if (status.to_svn_wc_entry_t == IntPtr.Zero)
 					return;
 					
-				svn_wc_entry_t ent = (svn_wc_entry_t)Marshal.PtrToStructure(status.to__svn_wc_entry_t, typeof(svn_wc_entry_t));
+				svn_wc_entry_t ent = (svn_wc_entry_t)Marshal.PtrToStructure(status.to_svn_wc_entry_t, typeof(svn_wc_entry_t));
 				Name = ent.name;
 				Revision = ent.revision;
 	  			Url = ent.url;
@@ -695,7 +695,7 @@
 		private struct svn_error_t {
 			public int apr_err;
 			public string message;
-			public IntPtr svn_error_t__child;
+			public IntPtr svn_error_t_child;
 			public IntPtr pool;
 		}
 		
@@ -774,14 +774,14 @@
   		}
   				
 		internal struct svn_wc_status_t {
-			public IntPtr to__svn_wc_entry_t;
-			public int svn_wc_status_kind__text;
-			public int svn_wc_status_kind__props;
+			public IntPtr to_svn_wc_entry_t;
+			public int svn_wc_status_kind_text;
+			public int svn_wc_status_kind_props;
 			public int locked;
 			public int copied;
 			public int switched;
-			public int svn_wc_status_kind__text__repo;
-			public int svn_wc_status_kind__props__repo;
+			public int svn_wc_status_kind_text_repo;
+			public int svn_wc_status_kind_props_repo;
 		}
 		
 		public struct Rev {
@@ -877,7 +877,7 @@
 			IntPtr ctx, IntPtr pool);
 			
 		[DllImport(svnclientlib)] static extern IntPtr svn_client_log (
-			IntPtr apr_array_header_t__targets,
+			IntPtr apr_array_header_t_targets,
 			ref Rev rev_start, ref Rev rev_end,
 			int discover_changed_paths,
             int strict_node_history,
@@ -911,8 +911,8 @@
 			int recurse, IntPtr ctx, IntPtr pool);
 			
 		[DllImport(svnclientlib)] static extern IntPtr svn_client_commit (
-			ref IntPtr svn_client_commit_info_t__commit_info,
-			IntPtr apr_array_header_t__targets, int nonrecursive,
+			ref IntPtr svn_client_commit_info_t_commit_info,
+			IntPtr apr_array_header_t_targets, int nonrecursive,
 			IntPtr ctx, IntPtr pool);
 	}
 
