--- cpufreqd-1.1.1/sys_check.c	2004-01-27 12:22:13.000000000 +0100
+++ sys_check.c	2004-03-02 23:13:13.158613312 +0100
@@ -40,14 +40,6 @@
   struct stat sb;
   int rc;
   
-  /* try fisrt with the 2.4 series */
-  rc = stat(CPUFREQ_PROC_INTERFACE, &sb);
-  if (rc == 0) {
-    cp_log(LOG_DEBUG, "find_cpufreq_interface(): found %s interface.\n", CPUFREQ_PROC_INTERFACE);
-    configuration->has_sysfs = 0;
-    return 0;
-  }
-
   /* try with the 2.5 series */
   rc = stat(CPUFREQ_SYSFS_INTERFACE, &sb);
   if (rc == 0) {
@@ -56,6 +48,14 @@
     return 0;
   }
   
+  /* try fisrt with the 2.4 series */
+  rc = stat(CPUFREQ_PROC_INTERFACE, &sb);
+  if (rc == 0) {
+    cp_log(LOG_DEBUG, "find_cpufreq_interface(): found %s interface.\n", CPUFREQ_PROC_INTERFACE);
+    configuration->has_sysfs = 0;
+    return 0;
+  }
+
   cp_log(LOG_ERR, "find_cpufreq_interface(): no cpufreq interface found.\n");
   return -1;
     
