add DESTDIR support

--- INSTALL
+++ INSTALL
@@ -1,18 +1,22 @@
 #!/bin/bash
 
-BINDIR=/usr/bin
-DOCDIR=/usr/share/doc/collectl
-SHRDIR=/usr/share/collectl
+#DESTDIR=
+BINDIR=$DESTDIR/usr/bin
+DOCDIR=$DESTDIR/usr/share/doc/collectl
+SHRDIR=$DESTDIR/usr/share/collectl
 
+mkdir -p $BINDIR
 mkdir -p $DOCDIR
 mkdir -p $SHRDIR
 mkdir -p $SHRDIR/util
-mkdir -p /var/log/collectl
+mkdir -p $DESTDIR/etc/init.d
+mkdir -p $DESTDIR/usr/share/man/man1
+mkdir -p $DESTDIR/var/log/collectl
 
 cp collectl.pl           $BINDIR/collectl
-cp collectl.conf         /etc
-cp initd/*               /etc/init.d
-cp man1/*                /usr/share/man/man1
+cp collectl.conf         $DESTDIR/etc
+cp initd/*               $DESTDIR/etc/init.d
+cp man1/*                $DESTDIR/usr/share/man/man1
 
 cp docs/*                $DOCDIR
 cp GPL ARTISTIC COPYING  $DOCDIR
@@ -28,15 +32,17 @@
 cp col2tlviz.pl          $SHRDIR/util
 
 # Force in case redoing the install and files already zipped
-gzip -f /usr/share/man/man1/collectl*
+gzip -f $DESTDIR/usr/share/man/man1/collectl*
 
-chmod 755 /etc/init.d/collectl*
-chmod 444 /etc/collectl.conf
+chmod 755 $DESTDIR/etc/init.d/collectl*
+chmod 444 $DESTDIR/etc/collectl.conf
 chmod 755 $BINDIR/collectl
 chmod 444 $DOCDIR/ARTISTIC $DOCDIR/COPYING $DOCDIR/GPL
 chmod 755 $SHRDIR/util/*
 chmod 444 $SHRDIR/formatit.ph $SHRDIR/*expr.ph $SHRDIR/vmstat.ph
 
+[[ -n $DESTDIR ]] && exit 0
+
 # remove any stale versions in case the names/numbers used have changed.
 # on new ROCKS installion 'rm' isn't there yet!  [thanks roy]
 if [ -x /bin/rm ] ; then
