--- build.py.orig	2003-04-09 07:17:13.000000000 +0200
+++ build.py	2003-04-13 19:52:32.000000000 +0200
@@ -651,7 +651,7 @@
 
 
 def checkThreading ():
-        qtmodlib = modDir + '/libqtcmodule.so'
+        qtmodlib = sipModuleDir + '/libqtcmodule.so'
         os.system ('ldd ' + qtmodlib + ' > lddtmp')
         f = open ('lddtmp', 'r')
         buff = f.read ()
@@ -843,7 +843,7 @@
     if sipModuleDir is None:
         error("The SIP module/library could not be found. Use the -s argument to set the correct directory")
 
-    proPatches["LIBS"] = [re.compile("@BL_SIPMODDIR@",re.M), sipModuleDir]
+    proPatches["LIBS"] = [re.compile("@BL_SIPMODDIR@",re.M), (sipModuleDir + " -L" + modDir)]
 
     inform("%s contains the SIP module." % (sipModuleDir))
 
