diff -ur grub/stage2/stage2.c grub-0.90/stage2/stage2.c
--- grub/stage2/stage2.c	Thu Jul 19 01:43:41 2001
+++ grub-0.90/stage2/stage2.c	Thu Jul 19 01:50:23 2001
@@ -503,19 +503,6 @@
 
 	  c = translate_keycode (getkey ());
 
-	  if (grub_timeout >= 0)
-	    {
-	      if (terminal & TERMINAL_DUMB)
-		grub_putchar ('\r');
-	      else
-		gotoxy (view_x0 + 3, 22);
-	      printf ("                                                                    ");
-	      grub_timeout = -1;
-	      fallback_entry = -1;
-	      if (! (terminal & TERMINAL_DUMB))
-		gotoxy (view_x1 - 6, view_x0 + 4 + entryno);
-	    }
-
 	  /* We told them above (at least in SUPPORT_SERIAL) to use
 	     '^' or 'v' so accept these keys.  */
 	  if (c == 16 || c == '^')
@@ -792,6 +779,12 @@
 		}
 #endif
 	    }
+	  if (grub_timeout >= 0)
+	    {
+	      grub_timeout = -1;
+	      fallback_entry = -1;
+	      goto restart;
+	    }
 	}
     }
 
