diff -uprN jxplorer.orig/src/com/ca/commons/cbutil/CBOpenConWin.java jxplorer/src/com/ca/commons/cbutil/CBOpenConWin.java
--- jxplorer.orig/src/com/ca/commons/cbutil/CBOpenConWin.java	2004-12-28 19:30:42.000000000 +0100
+++ jxplorer/src/com/ca/commons/cbutil/CBOpenConWin.java	2005-01-02 17:24:15.765367432 +0100
@@ -501,4 +501,4 @@ public abstract class CBOpenConWin exten
      */
 
     public abstract void connect(ConnectionData connectData);
-}
\ No newline at end of file
+}
diff -uprN jxplorer.orig/src/com/ca/commons/cbutil/CBUtility.java jxplorer/src/com/ca/commons/cbutil/CBUtility.java
--- jxplorer.orig/src/com/ca/commons/cbutil/CBUtility.java	2004-12-28 19:30:42.000000000 +0100
+++ jxplorer/src/com/ca/commons/cbutil/CBUtility.java	2005-01-02 17:10:12.559554200 +0100
@@ -1399,7 +1399,7 @@ if (debug) System.out.println("\n\n Star
             String pathToConfigFile;
 
             // the potential path to a config file in the user home directory
-            String userHomeConfigDir = System.getProperty("user.home") + File.separator + "jxplorer" ;
+            String userHomeConfigDir = System.getProperty("user.home") + File.separator + ".jxplorer" ;
             String userHomeConfigFilePath = userHomeConfigDir + File.separator + configFileName;
 
             // the default config location is to the directory JXplorer is run from.
@@ -1440,4 +1440,4 @@ if (debug) System.out.println("\n\n Star
             // the directory JXplorer is run from.
             return defaultConfigFilePath;
         }
-}
\ No newline at end of file
+}
diff -uprN jxplorer.orig/src/com/ca/directory/jxplorer/editor/PostalAddressEditor.java jxplorer/src/com/ca/directory/jxplorer/editor/PostalAddressEditor.java
--- jxplorer.orig/src/com/ca/directory/jxplorer/editor/PostalAddressEditor.java	2004-12-28 19:30:42.000000000 +0100
+++ jxplorer/src/com/ca/directory/jxplorer/editor/PostalAddressEditor.java	2004-12-28 19:57:10.000000000 +0100
@@ -240,4 +240,4 @@ public class PostalAddressEditor extends
         setVisible(false);
         dispose();
     }
-}
\ No newline at end of file
+}
diff -uprN jxplorer.orig/src/com/ca/directory/jxplorer/JXplorer.java jxplorer/src/com/ca/directory/jxplorer/JXplorer.java
--- jxplorer.orig/src/com/ca/directory/jxplorer/JXplorer.java	2004-12-28 19:30:42.000000000 +0100
+++ jxplorer/src/com/ca/directory/jxplorer/JXplorer.java	2005-01-02 17:09:17.747886840 +0100
@@ -1874,4 +1874,4 @@ public class JXplorer extends JFrame    
     {
         return buttonRegister;
     }
-}
\ No newline at end of file
+}
