diff -Naur gtkmathview-bonobo-0.0.4.orig/src/aux.cc gtkmathview-bonobo-0.0.4/src/aux.cc
--- gtkmathview-bonobo-0.0.4.orig/src/aux.cc	2003-10-05 16:44:09.000000000 +0900
+++ gtkmathview-bonobo-0.0.4/src/aux.cc	2008-11-24 19:04:15.000000000 +0900
@@ -24,6 +24,7 @@
 #include <config.h>
 
 #include <cassert>
+#include <cstdlib>
 
 #include <sstream>
 
@@ -34,8 +35,6 @@
 
 #include "aux.h"
 
-#define MATHML_NS_URI "http://www.w3.org/1998/Math/MathML"
-
 namespace DOM = GdomeSmartDOM;
 
 static DOM::Element
diff -Naur gtkmathview-bonobo-0.0.4.orig/src/aux.h gtkmathview-bonobo-0.0.4/src/aux.h
--- gtkmathview-bonobo-0.0.4.orig/src/aux.h	2003-09-14 00:17:10.000000000 +0900
+++ gtkmathview-bonobo-0.0.4/src/aux.h	2008-11-24 19:03:55.000000000 +0900
@@ -26,6 +26,8 @@
 
 #include <gdome.h>
 
+#define MATHML_NS_URI "http://www.w3.org/1998/Math/MathML"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
diff -Naur gtkmathview-bonobo-0.0.4.orig/src/control-factory.c gtkmathview-bonobo-0.0.4/src/control-factory.c
--- gtkmathview-bonobo-0.0.4.orig/src/control-factory.c	2004-05-22 01:06:26.000000000 +0900
+++ gtkmathview-bonobo-0.0.4/src/control-factory.c	2008-11-24 19:04:39.000000000 +0900
@@ -33,6 +33,7 @@
 #include "persist-stream.h"
 #include "handlers.h"
 #include "view.h"
+#include "aux.h"
 
 enum
   {
