diff -urN balsa-1.4.0.old/configure.in balsa-1.4.0/configure.in
--- balsa-1.4.0.old/configure.in	Mon Aug 19 11:04:50 2002
+++ balsa-1.4.0/configure.in	Tue Sep 10 06:19:57 2002
@@ -49,9 +49,16 @@
 dnl this should come after `AC_PROG_CC'
 GNOME_X_CHECKS
 
-AC_CHECK_LIB(pspell, new_pspell_manager,,[
-	AC_MSG_ERROR([*** libpspell(-devel) not found.  Please check http://balsa.gnome.org/ for information on the required files.])
-], -lstdc++ -lpspell-modules)
+AC_ARG_WITH(aspell, 
+	[  --with-aspell	Turn on aspell ],
+	 [case "${withval}" in
+	   yes) LIBS="-lstdc++ -laspell" ;
+	   		aspell=true ;;
+	   no) LIBS="-lstdc++" ;
+	   		aspell=false ;;
+		*) AC_MSG_ERROR(bad value ${withval} for --with-aspell) ;;
+	esac],[LIBS=""])
+AM_CONDITIONAL(WITH_ASPELL, test, x$aspell = xtrue)
 
 dnl Reset
 # LDFLAGS="$balsa_save_LDFLAGS"
