--- configure.in.orig	2004-08-08 01:03:56.919735072 +0930
+++ configure.in	2004-08-08 01:06:51.330220648 +0930
@@ -102,7 +102,7 @@
 		CFLAGS="$CFLAGS -I$have_gpgme/include"
 	fi
 	AC_MSG_CHECKING([gpgme library version])
-	AC_TRY_RUN([#include <gpgme.h>
+	AC_TRY_RUN([#include <gpgme3.h>
 		    #include <stdio.h>
 		    int main() {
 			char gv[] = GPGME_VERSION;
@@ -115,9 +115,9 @@
 		AC_DEFINE(HAVE_GPGME,1,[Defined when gpgme is available.])
 		if test x"$have_gpgme" != xyes ; then
 			BALSA_CFLAGS="$BALSA_CFLAGS -I$have_gpgme/include"
-			BALSA_LIBS="$BALSA_LIBS -L$have_gpgme/lib -lgpgme"
+			BALSA_LIBS="$BALSA_LIBS -L$have_gpgme/lib -lgpgme3"
 		else
-			BALSA_LIBS="$BALSA_LIBS -lgpgme"
+			BALSA_LIBS="$BALSA_LIBS -lgpgme3"
 		fi
 		have_gpgme=yes
 		AC_PATH_PROG(gpg_path, gpg)
