diff -ruN balsa-2.0.11.orig/libbalsa/Makefile.in balsa-2.0.11/libbalsa/Makefile.in
--- balsa-2.0.11.orig/libbalsa/Makefile.in	2004-05-06 15:43:57.050305474 -0700
+++ balsa-2.0.11/libbalsa/Makefile.in	2004-05-06 15:44:53.376879127 -0700
@@ -210,11 +210,7 @@
 
 
 DEFS = -DLIBMUTT                \
-       -DG_DISABLE_DEPRECATED   \
-       -DGTK_DISABLE_DEPRECATED \
-       -DGNOME_DISABLE_DEPRECATED
-
-
+       
 #Signal marshallers
 marshal_prefix = libbalsa
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
diff -ruN balsa-2.0.11.orig/libinit_balsa/Makefile.in balsa-2.0.11/libinit_balsa/Makefile.in
--- balsa-2.0.11.orig/libinit_balsa/Makefile.in	2004-05-06 15:43:57.043307515 -0700
+++ balsa-2.0.11/libinit_balsa/Makefile.in	2004-05-06 15:45:06.391083960 -0700
@@ -146,10 +146,7 @@
         $(BALSA_CFLAGS)
 
 
-DEFS = -DG_DISABLE_DEPRECATED   \
-       -DGTK_DISABLE_DEPRECATED \
-       -DGNOME_DISABLE_DEPRECATED
-
+DEFS = 
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES = 
diff -ruN balsa-2.0.11.orig/src/Makefile.in balsa-2.0.11/src/Makefile.in
--- balsa-2.0.11.orig/src/Makefile.in	2004-05-06 15:43:57.080296727 -0700
+++ balsa-2.0.11/src/Makefile.in	2004-05-06 15:44:37.439526797 -0700
@@ -223,10 +223,7 @@
 
 BALSA_ORB_SRCLIST = orbit-glue.c 
 
-DEFS = -DG_DISABLE_DEPRECATED   \
-       -DGTK_DISABLE_DEPRECATED \
-       -DGNOME_DISABLE_DEPRECATED
-
+DEFS = 
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES = 
