--- src/dh-gecko-utils.cpp	2006-03-05 12:16:10.000000000 -0600
+++ src/dh-gecko-utils.cpp	2006-03-05 12:25:58.000000000 -0600
@@ -21,33 +21,34 @@
  */
 
 #include <config.h>
-#include <gtkmozembed.h>
-#include <gtkmozembed_internal.h>
+#include <gtkembedmoz/gtkmozembed.h>
+#include <gtkembedmoz/gtkmozembed_internal.h>
 
-#include <nsCOMPtr.h>
-#include <nsMemory.h>
-#include <nsEmbedString.h>
-#include <nsIPrefService.h>
-#include <nsICommandManager.h>
-#include <nsIInterfaceRequestorUtils.h>
+#include <xpcom/nsCOMPtr.h>
+#include <xpcom/nsMemory.h>
+#include <string/nsEmbedString.h>
+#include <pref/nsIPrefService.h>
+#include <commandhandler/nsICommandManager.h>
+#include <xpcom/nsIInterfaceRequestorUtils.h>
 #define MOZILLA_INTERNAL_API
-#include <nsIServiceManager.h>
+#include <xpcom/nsIServiceManager.h>
 #undef MOZILLA_INTERNAL_API
-#include <nsISupportsPrimitives.h>
-#include <nsILocalFile.h>
-#include <nsIDOMMouseEvent.h>
+#include <xpcom/nsISupportsPrimitives.h>
+#include <xpcom/nsILocalFile.h>
+#include <dom/nsIDOMMouseEvent.h>
+#include <necko/nsIURL.h>
 
 #include <stdlib.h>
 
 #if defined (HAVE_CHROME_NSICHROMEREGISTRYSEA_H)
 #include <chrome/nsIChromeRegistrySea.h>
 #elif defined(MOZ_NSIXULCHROMEREGISTRY_SELECTSKIN)
-#include <nsIChromeRegistry.h>
+#include <content/nsIChromeRegistry.h>
 #endif
 
 #ifdef ALLOW_PRIVATE_API
 // FIXME: For setting the locale. hopefully gtkmozembed will do itself soon
-#include <nsILocaleService.h>
+#include <locale/nsILocaleService.h>
 #endif
 
 #include "dh-util.h"
