diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2011-05-05 00:28:39.000000000 +0200
+++ b/CMakeLists.txt	2011-05-16 17:21:31.562259992 +0200
@@ -122,6 +122,7 @@
 SET(COMPILE_LIGHTSPARK TRUE CACHE BOOL "Compile Lightspark?")
 SET(COMPILE_TIGHTSPARK TRUE CACHE BOOL "Compile Tightspark?")
 SET(COMPILE_PLUGIN TRUE CACHE BOOL "Compile the browser plugin?")
+SET(AUDIO_BACKEND "pulse" CACHE STRING "Choose audio backend: none pulse alsa, default is pulse")
 SET(ENABLE_CURL TRUE CACHE BOOL "Enable CURL? (Required for Downloader functionality)")
 SET(ENABLE_LIBAVCODEC TRUE CACHE BOOL "Enable libavcodec and dependent functionality?")
 SET(ENABLE_RTMP FALSE CACHE BOOL "Enable librtmp and dependent functionality?")
@@ -412,7 +413,9 @@
 GETTEXT_CREATE_TRANSLATIONS(i18n/lightspark.pot ALL i18n/pl.po i18n/fr.po
         i18n/zh_CN.po)
 
-ADD_SUBDIRECTORY(backends/interfaces/audio/pulse)
+IF(${AUDIO_BACKEND} MATCHES "pulse")
+  ADD_SUBDIRECTORY(backends/interfaces/audio/pulse)
+ENDIF(${AUDIO_BACKEND} MATCHES "pulse")
 #ADD_SUBDIRECTORY(backends/interfaces/audio/alsa)
 #ADD_SUBDIRECTORY(backends/interfaces/audio/openal)
 
