--- qp7.2.orig/src/xqp/interact.cc	2004-05-25 16:31:33.000000000 +1200
+++ qp7.2/src/xqp/interact.cc	2006-05-27 15:04:32.000000000 +1200
@@ -26,13 +26,17 @@
 #include "term.h"
 #include "xqpqueries.h"
 #include <qmessagebox.h>
-#include <qfiledialog.h>
+#include <q3filedialog.h>
 #include <qfile.h>
+#include <QMouseEvent>
+#include <QTextStream>
+#include <QKeyEvent>
+#include <QEvent>
 
 using namespace std;
 
 Interact::Interact( QWidget *box )
-  : QTextEdit(box)
+  : Q3TextEdit(box)
 {
   parent = box;
   setTextFormat(Qt::PlainText);
@@ -98,13 +102,13 @@
   if ((p < para) || ((p == para) && (i < indent)))
     {
       if (e->button() != Qt::MidButton)
-	QTextEdit::contentsMouseReleaseEvent(e);
+	Q3TextEdit::contentsMouseReleaseEvent(e);
       setCursorPosition(currpara,currindent);
       readonly = true;
     }
   else
     {
-      QTextEdit::contentsMouseReleaseEvent(e);
+      Q3TextEdit::contentsMouseReleaseEvent(e);
       readonly = false;
     }
 }
@@ -116,21 +120,21 @@
     {
       QMouseEvent lbe(QEvent::MouseButtonPress, e->pos(), 
 		      Qt::LeftButton, Qt::LeftButton);
-      QTextEdit::contentsMousePressEvent(&lbe);
-      QTextEdit::contentsMouseReleaseEvent(&lbe);
+      Q3TextEdit::contentsMousePressEvent(&lbe);
+      Q3TextEdit::contentsMouseReleaseEvent(&lbe);
     }
-  QTextEdit::contentsMousePressEvent(e);
+  Q3TextEdit::contentsMousePressEvent(e);
 }
 
 void Interact::cut()
 {
-  if (readonly) QTextEdit::copy();
-  else QTextEdit::cut();
+  if (readonly) Q3TextEdit::copy();
+  else Q3TextEdit::cut();
 }
 
 void Interact::paste()
 {
-  if (!readonly) QTextEdit::paste();
+  if (!readonly) Q3TextEdit::paste();
 }
 
 void Interact::keyPressEvent(QKeyEvent *k)
@@ -138,10 +142,10 @@
   int key_pressed = k->key();
   if (key_pressed == Qt::Key_Control)
     {
-      QTextEdit::keyPressEvent(k);
+      Q3TextEdit::keyPressEvent(k);
       return;
     }
-  if (k->state() == ControlButton)
+  if (k->state() == Qt::ControlModifier)
     {
       in_history = false;
       if (key_pressed == 'D')
@@ -151,7 +155,7 @@
         } 
       if (key_pressed == 'C')
 	{
-	  QTextEdit::keyPressEvent(k);
+	  Q3TextEdit::keyPressEvent(k);
 	  return;
 	}
     }
@@ -220,7 +224,7 @@
   if (key_pressed == Qt::Key_Home)
     {
       in_history = false;
-      if (k->state() == ControlButton)
+      if (k->state() == Qt::ControlModifier)
 	{
 	  setCursorPosition(para,indent);
 	  return;
@@ -244,7 +248,7 @@
     }
 
   in_history = false;
-  QTextEdit::keyPressEvent(k);
+  Q3TextEdit::keyPressEvent(k);
 }
 
 void Interact::addHistoryItem(QString* s)
@@ -271,11 +275,11 @@
 
 void Interact::openQueryFile()
 {
-  QString fileName = QFileDialog::getOpenFileName(QString::null, "*", this);
+  QString fileName = Q3FileDialog::getOpenFileName(QString::null, "*", this);
   if (fileName != QString::null)
     {
       QFile f(fileName);
-      if (f.open(IO_ReadOnly))
+      if (f.open(QIODevice::ReadOnly))
 	{
 	  QTextStream t(&f);
 	  XQPQueries* xqp_queries = new XQPQueries(parent, fileName, t.read());
@@ -289,11 +293,11 @@
 
 void Interact::saveHistory()
 {  
-  QString fileName = QFileDialog::getSaveFileName(QString::null, "*", this);
+  QString fileName = Q3FileDialog::getSaveFileName(QString::null, "*", this);
   if (fileName != QString::null)
     {
       QFile f(fileName);
-      if (f.open(IO_WriteOnly))
+      if (f.open(QIODevice::WriteOnly))
 	{
 	  QString* item = firstHistoryItem();
 	  while (item != NULL)
@@ -309,11 +313,11 @@
 
 void Interact::saveSession()
 {
-  QString fileName = QFileDialog::getSaveFileName(QString::null, "*", this);
+  QString fileName = Q3FileDialog::getSaveFileName(QString::null, "*", this);
   if (fileName != QString::null)
     {
       QFile f(fileName);
-      if (f.open(IO_WriteOnly))
+      if (f.open(QIODevice::WriteOnly))
 	{
 	  QString s = text();
 	  f.writeBlock(s, s.length());
--- qp7.2.orig/src/xqp/interact.h	2006-05-27 15:42:00.000000000 +1200
+++ qp7.2/src/xqp/interact.h	2006-05-27 15:07:05.000000000 +1200
@@ -21,17 +21,20 @@
 #define INTERACT_H
 
 #include <qwidget.h>
-#include <qtextedit.h>
+#include <q3textedit.h>
 #include <qstring.h>
+#include <QMouseEvent>
+#include <QKeyEvent>
+#include <Q3PtrList>
 #include <fstream>
-#include <qaccel.h>
+#include <q3accel.h>
 #include <qregexp.h>
 #include <qsocketnotifier.h>
 #include <qstringlist.h>
 
 
 
-class Interact : public QTextEdit  {
+class Interact : public Q3TextEdit  {
   Q_OBJECT
     public: 
   Interact(QWidget *box);
@@ -58,7 +61,7 @@
   bool readonly;
   bool in_query;
   bool in_history;
-  QPtrList<QString> history;
+  Q3PtrList<QString> history;
   QString* firstHistoryItem(void);
   QString* nextHistoryItem(void);
   QString* previousHistoryItem(void);
--- qp7.2.orig/src/xqp/qthelp.cc	2004-05-25 16:31:33.000000000 +1200
+++ qp7.2/src/xqp/qthelp.cc	2006-05-27 14:50:15.000000000 +1200
@@ -73,9 +73,9 @@
 
 
 QTHelp::QTHelp(QWidget *p)
-  : QMainWindow(p)
+  : Q3MainWindow(p)
 {
-  help = new QTextEdit(this);
+  help = new Q3TextEdit(this);
   setCentralWidget(help);
   parent = p;
   resize(600,400);
--- qp7.2.orig/src/xqp/qthelp.h	2004-05-10 11:51:09.000000000 +1200
+++ qp7.2/src/xqp/qthelp.h	2006-05-27 14:50:15.000000000 +1200
@@ -21,11 +21,11 @@
 #define QTHELP_H
 
 #include <qwidget.h>
-#include <qmainwindow.h>
-#include <qtextedit.h>
+#include <q3mainwindow.h>
+#include <q3textedit.h>
 
 
-class QTHelp : public QMainWindow
+class QTHelp : public Q3MainWindow
 {
   Q_OBJECT
     public:
@@ -37,7 +37,7 @@
  signals:
 
  private:
-  QTextEdit* help;
+  Q3TextEdit* help;
   QWidget* parent;
   
  protected:
--- qp7.2.orig/src/xqp/term.cc	2004-05-10 11:51:10.000000000 +1200
+++ qp7.2/src/xqp/term.cc	2006-05-27 15:00:38.000000000 +1200
@@ -38,7 +38,7 @@
 
 bool graphic_char(QChar c)
 {
-    switch (c)
+    switch (c.unicode())
     {
        case '-':
        case '/':
--- qp7.2.orig/src/xqp/xqp.cc	2004-05-16 16:27:55.000000000 +1200
+++ qp7.2/src/xqp/xqp.cc	2006-05-27 15:06:15.000000000 +1200
@@ -26,13 +26,14 @@
 #include <qpushbutton.h>
 #include <qlabel.h>
 #include <qapplication.h>
-#include <qpopupmenu.h>
-#include <qmainwindow.h>
+#include <q3popupmenu.h>
+#include <q3mainwindow.h>
 #include <qmenubar.h>
-#include <qvbox.h>
+#include <q3vbox.h>
 #include <qmessagebox.h>
 #include <qinputdialog.h>
-#include <qlistbox.h>
+#include <q3listbox.h>
+#include <QCloseEvent>
 
 #include "xqp.h"
 #include "interact.h"
@@ -63,7 +64,7 @@
      cancel->setGeometry(190,200,120,40);
      font->setGeometry(10,10,150,40);
      colour->setGeometry(170,10,150,40);
-     browser = new QTextBrowser(this);
+     browser = new Q3TextBrowser(this);
      browser->setGeometry(10,80,310,100);
      browser->setText("AaBb :- \n| ?-");
      browser->setFont(f);
@@ -105,7 +106,7 @@
 
 
 Xqp::Xqp(int e_stdout, int e_stdin, pid_t c_pid)
-  : QMainWindow()
+  : Q3MainWindow()
 {
   config = new QPConfig(this);
   qp_stdin = e_stdin;
@@ -124,20 +125,20 @@
   qpint->setReadOnly(false);  
   // Set up menu items
   // File menu
-  QPopupMenu* file = new QPopupMenu(this);
+  Q3PopupMenu* file = new Q3PopupMenu(this);
   file->insertItem("Open Query File", qpint, SLOT(openQueryFile()));
   file->insertItem("Save History", qpint, SLOT(saveHistory()));
   file->insertItem("Save Session", qpint, SLOT(saveSession()));
   file->insertItem("Configure", this, SLOT(configure_int()));
   menuBar()->insertItem("File", file);
   
-  QPopupMenu* sigs = new QPopupMenu(this);
+  Q3PopupMenu* sigs = new Q3PopupMenu(this);
   sigs->insertItem("SIGINT", this, SLOT(process_CTRL_C()));
   menuBar()->insertItem("Signals", sigs);
   
   
   // Help menu
-  QPopupMenu* help = new QPopupMenu(this);
+  Q3PopupMenu* help = new Q3PopupMenu(this);
   help->insertItem("Help", this, SLOT(showHelp()));
   help->insertItem("About", this, SLOT(showAbout()));
   menuBar()->insertItem("Help", help);
@@ -170,7 +171,7 @@
   
 void Xqp::closeEvent(QCloseEvent *e)
 {
-  QMainWindow::closeEvent(e);
+  Q3MainWindow::closeEvent(e);
 }
 
 
--- qp7.2.orig/src/xqp/xqp.h	2004-05-16 16:27:55.000000000 +1200
+++ qp7.2/src/xqp/xqp.h	2006-05-27 15:06:25.000000000 +1200
@@ -26,13 +26,15 @@
 
 #include <qapplication.h>
 #include <qwidget.h>
-#include <qmainwindow.h>
+#include <q3mainwindow.h>
+#include <Q3PopupMenu>
+#include <QCloseEvent>
 #include <sys/types.h>
 #include <qstring.h>
-#include <qmultilineedit.h>
+#include <q3multilineedit.h>
 #include <qdialog.h>
-#include <qmainwindow.h>
-#include <qtextbrowser.h>
+#include <q3mainwindow.h>
+#include <q3textbrowser.h>
 #include <qcolordialog.h>
 
 #include "interact.h"
@@ -54,12 +56,12 @@
   Xqp* p;
   QFont f;
   QColor c;
-  QTextBrowser *browser;
+  Q3TextBrowser *browser;
 };
 
 
 /** Xqp is the base class of the project */
-class Xqp : public QMainWindow
+class Xqp : public Q3MainWindow
 {
   Q_OBJECT 
     public:
@@ -93,7 +95,7 @@
   static const int buff_size = 10000;
   char* read_buff;
 
-  QPopupMenu* view_menu;
+  Q3PopupMenu* view_menu;
   QTHelp* qthelp;
 
 public:
--- qp7.2.orig/src/xqp/xqp.pro	2004-05-11 15:23:59.000000000 +1200
+++ qp7.2/src/xqp/xqp.pro	2006-05-27 15:09:32.000000000 +1200
@@ -3,6 +3,7 @@
 ######################################################################
 
 TEMPLATE = app
+QT += qt3support  
 INCLUDEPATH += .
 
 # Input
--- qp7.2.orig/src/xqp/xqpqueries.cc	2004-05-10 11:51:11.000000000 +1200
+++ qp7.2/src/xqp/xqpqueries.cc	2006-05-27 15:06:34.000000000 +1200
@@ -19,11 +19,12 @@
 
 #include "xqpqueries.h"
 #include "term.h"
+#include <QKeyEvent>
 
 XQPQueries::XQPQueries(QWidget *p, QString& name, QString text)
-  : QMainWindow(p, name)
+  : Q3MainWindow(p, name)
 {
-  queries = new QTextEdit(this);
+  queries = new Q3TextEdit(this);
   setCentralWidget(queries);
   parent = p;
   resize(400,300);
@@ -88,6 +89,6 @@
     }
   else
     {
-      QMainWindow::keyPressEvent(k);
+      Q3MainWindow::keyPressEvent(k);
     }
 }
--- qp7.2.orig/src/xqp/xqpqueries.h	2004-05-10 11:51:11.000000000 +1200
+++ qp7.2/src/xqp/xqpqueries.h	2006-05-27 15:06:42.000000000 +1200
@@ -21,11 +21,12 @@
 #define XQPQUERIES_H
 
 #include <qwidget.h>
-#include <qmainwindow.h>
-#include <qtextedit.h>
+#include <q3mainwindow.h>
+#include <q3textedit.h>
+#include <QKeyEvent>
 
 
-class XQPQueries : public QMainWindow
+class XQPQueries : public Q3MainWindow
 {
   Q_OBJECT
     public:
@@ -38,7 +39,7 @@
  void process_text(QString);
  void process_return();
  private:
-  QTextEdit* queries;
+  Q3TextEdit* queries;
   QWidget* parent;
   
  protected:
