diff -ur elmo-1.3.2.orig/src/stats.c elmo-1.3.2/src/stats.c
--- elmo-1.3.2.orig/src/stats.c	2004-04-29 10:06:10.000000000 +0200
+++ elmo-1.3.2/src/stats.c	2005-07-21 16:19:27.000000000 +0200
@@ -24,6 +24,7 @@
  ****************************************************************************/
 
 #include <errno.h>
+#include <stdlib.h>
 
 #include "stats.h"
 #include "wrapbox.h"
@@ -181,7 +182,8 @@
 void
 stats_dump (void)
 {
-        int            i;
+        int            i, fd;
+        char          template[] = "/tmp/elmostatsXXXXXX";
         FILE          *fp;
         mail_t        *mail;
         struct estats  stats;
@@ -200,9 +202,10 @@
                         process_mail (& stats, mail);
         }
 
-        fp = fopen ("/tmp/elmostats", "w");
+        fd = mkstemp(template);
+        fp = fdopen (fd, "w");
         if (fp == NULL){
-                error_ (errno, _("couldn't open %s"), "/tmp/elmostats");
+                error_ (errno, _("couldn't open %s"), template);
                 destroy_stats (& stats);
                 return;
         }
