--- src/Core/pluginmanager.h
+++ src/Core/pluginmanager.h
@@ -42,6 +42,8 @@
 {
 
 class GejengelCore;
+class GejengelPlugin;
+class Track;
 
 class PluginManager
 {
--- src/ui/pluginview.cpp
+++ src/ui/pluginview.cpp
@@ -19,6 +19,7 @@
 #include "Utils/log.h"
 #include "Utils/numericoperations.h"
 #include "Core/pluginmanager.h"
+#include "Core/gejengelplugin.h"
 #include "pluginwidget.h"
 
 #include <glibmm/i18n.h>
