--- configure.in.orig	Thu Feb 22 15:04:29 2001
+++ configure.in	Thu Feb 22 15:04:35 2001
@@ -101,16 +101,21 @@
 	fi
 ])
 AC_ARG_WITH(gpm-mouse, 
-	[  --with-gpm-mouse=base-dir  Specifies the base gpm directory],
+	[  --with-gpm-mouse=base-dir/no  Specifies the base gpm directory],
 	if test x$withval = xyes
 	then 
-		AC_MSG_WARN(Usage is: --with-gpm-mouse=base-dir)
+		AC_MSG_WARN(Usage is: --with-gpm-mouse=base-dir/no )
 	else
+	    if test x$withval = xno
+	    then
+		gpm_mousedriver_test=no
+	    else
 		LIBS="$LIBS -L$withval/lib -lgpm"
                 AC_DEFINE(GPM_MOUSEDRIVER)
 	        gpm_mousedriver=true
 		CFLAGS="$CFLAGS -I$withval/include"
 		AC_MSG_RESULT(defining use_gpm)
+	    fi
 	fi
 )
 
