Index: scim-1.4.4/configure.ac
===================================================================
--- scim-1.4.4.orig/configure.ac
+++ scim-1.4.4/configure.ac
@@ -80,8 +80,12 @@ AC_PROG_CC
 AC_LANG(C++)
 
 # Checks if doxygen is avaiable
+AC_ARG_WITH([doxygen],
+  AS_HELP_STRING([--without-doxygen], [Don't build API documentation]) )
 
-AC_PATH_PROG(DOXYGEN, doxygen, no)
+if test "x$with_doxygen" != "xno"; then
+  AC_PATH_PROG(DOXYGEN, doxygen, no)
+fi
 AM_CONDITIONAL(HAVE_DOXYGEN, test x$DOXYGEN != xno)
 AC_SUBST(DOXYGEN)
 
