--- kazehakase-0.1.9/src/mozilla/kz-mozembed.cpp	2004-08-27 10:05:13.000000000 +0900
+++ kazehakase-0.1.9_fix/src/mozilla/kz-mozembed.cpp	2004-09-11 21:16:58.000000000 +0900
@@ -633,7 +633,7 @@
 	gchar *cache = g_strconcat("file://", g_get_home_dir(),
 			    	   HISTORY_DIR, NULL);
 	const gchar *location = kz_moz_embed_get_location(kzembed);
-	if (location && create_thumbnail &&
+	if (location && 
 	    (g_str_has_prefix(location, "http:") || 
 	     g_str_has_prefix(location, "https:") || 
 	     g_str_has_prefix(location, "history-search:") || 
@@ -2060,7 +2060,7 @@
 	g_return_val_if_fail (wrapper != NULL, NULL);
 
 	PRUint32 persistFlags = 0;
-	
+
 	nsCOMPtr<nsIWebBrowserPersist> bpersist = 
 		do_CreateInstance(NS_WEBBROWSERPERSIST_CONTRACTID);
 	if (!bpersist) return NULL;
