diff -Naru kazehakase-0.1.9.org/configure kazehakase-0.1.9/configure
--- kazehakase-0.1.9.org/configure	2004-08-28 15:40:43.000000000 +0100
+++ kazehakase-0.1.9/configure	2004-09-22 13:17:19.340900485 +0100
@@ -19392,6 +19392,7 @@
   1.7) MOZILLA_SNAPSHOT=15 ;;
   1.7.1) MOZILLA_SNAPSHOT=15 ;;
   1.7.2) MOZILLA_SNAPSHOT=15 ;;
+  1.7.3) MOZILLA_SNAPSHOT=15 ;;
  1.8a) MOZILLA_SNAPSHOT=16 ;;
  1.8a2) MOZILLA_SNAPSHOT=16 ;;
  1.8a3) MOZILLA_SNAPSHOT=16 ;;
