--- src/kz-tab-label.c.orig	2004-04-06 23:58:19.413616920 -0400
+++ src/kz-tab-label.c	2004-04-06 23:58:24.880785784 -0400
@@ -1170,6 +1170,7 @@
 		  gint state, gint status,
 		  KzTabLabel *kztab)
 {
+	GdkGC *gc;
 	gdouble progress;
 	GtkWidget *widget;	       
 
@@ -1186,7 +1187,7 @@
 
 		mask = gdk_pixmap_new(widget->window,
 				      width, height, 1);
-		GdkGC *gc = gdk_gc_new(mask);
+		gc = gdk_gc_new(mask);
 		gdk_draw_rectangle(mask,
 				   gc,
 				   TRUE,
