--- powersoftplus-0.1.8b/src/cshmem.cpp	2009-01-19 18:16:12.000000000 -0400
+++ powersoftplus-0.1.8/src/cshmem.cpp	2009-01-19 18:19:37.000000000 -0400
@@ -17,6 +17,7 @@
  *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
  ***************************************************************************/
 #include "cshmem.h"
+#include <cstring>
 
 CShMem::CShMem()
 {
--- powersoftplus-0.1.8b/src/glfunc.cpp	2009-01-19 18:16:12.000000000 -0400
+++ powersoftplus-0.1.8/src/glfunc.cpp	2009-01-19 18:19:57.000000000 -0400
@@ -22,6 +22,7 @@
 #endif
 #include <iostream>
 #include <cstdlib>
+#include <cstring>
 #include <math.h>
 #include "glfunc.h"
 #include "strings_en.h"
