diff -ur cl-pdf-110.orig/cl-pdf.asd cl-pdf-110/cl-pdf.asd
--- cl-pdf-110.orig/cl-pdf.asd	2005-12-30 17:54:52.000000000 -0600
+++ cl-pdf-110/cl-pdf.asd	2005-12-30 18:19:59.000000000 -0600
@@ -4,45 +4,21 @@
 ;;; You can reach me at marc.battyani@fractalconcept.com or marc@battyani.net
 ;;; The homepage of cl-pdf is here: http://www.fractalconcept.com/asp/html/cl-pdf.html
 
-(in-package asdf)
-
-;;;Choose the zlib implementation you want to use (only one!)
-(pushnew :use-salza-zlib cl:*features*)
-;(pushnew :use-uffi-zlib cl:*features*)
-;(pushnew :use-abcl-zlib cl:*features*)
-;(pushnew :use-no-zlib cl:*features*)
-
-#-(or use-uffi-zlib use-salza-zlib use-abcl-zlib use-no-zlib)
-(Error "You must choose which zlib version you want to use")
-
-#-(or uffi (not use-uffi-zlib))
-(ignore-errors
-  (print "Trying to load UFFI:")
-  (asdf:operate 'asdf:load-op :uffi)
-  (pushnew :uffi cl:*features*)
-  (print "UFFI loaded."))
-
-(load (merge-pathnames "iterate/iterate.asd" *load-truename*))
-
-#+use-salza-zlib
-(load (merge-pathnames "salza/salza.asd" *load-truename*))
+(defpackage #:cl-pdf-system (:use #:common-lisp #:asdf))
+(in-package #:cl-pdf-system)
 
 #+clisp (setf *warn-on-floating-point-contagion* nil)
 
-(defsystem :cl-pdf
+(defsystem #:cl-pdf
   :name "cl-pdf"
   :author "Marc Battyani <marc.battyani@fractalconcept.com>"
-  :version "2.0"
   :maintainer "Marc Battyani <marc.battyani@fractalconcept.com>"
   :licence "BSD like licence"
   :description "Common Lisp PDF Generation Library"
   :long-description "The cl-pdf package provides a stand-alone Common Lisp library to generate PDF files."
-  :perform (load-op :after (op cl-pdf)
-		    (pushnew :cl-pdf cl:*features*))
   :components ((:file "defpackage")
 	       (:file "config" :depends-on ("defpackage"))
-	       #+use-uffi-zlib (:file "init" :depends-on ("config"))
-	       (:file "zlib" :depends-on ("defpackage" #+use-uffi-zlib "init"))
+	       (:file "zlib" :depends-on ("defpackage"))
 	       (:file "font-metrics"  :depends-on ("config"))
 	       (:file "encodings"  :depends-on ("defpackage"))
 	       (:file "t1-font" :depends-on ("font-metrics" "encodings"))
@@ -56,4 +32,4 @@
 	       (:file "text" :depends-on ("pdf-base"))
 	       (:file "bar-codes" :depends-on ("pdf-geom"))
 	       (:file "chart" :depends-on ("text" "pdf-geom")))
-  :depends-on (:iterate #+use-salza-zlib :salza))
+  :depends-on (#:iterate #:salza))
diff -ur cl-pdf-110.orig/cl-pdf-parser.asd cl-pdf-110/cl-pdf-parser.asd
--- cl-pdf-110.orig/cl-pdf-parser.asd	2005-12-30 17:54:52.000000000 -0600
+++ cl-pdf-110/cl-pdf-parser.asd	2005-12-30 18:24:04.000000000 -0600
@@ -1,8 +1,9 @@
 ;;;; -*- Mode: LISP; Syntax: ANSI-Common-Lisp; Base: 10 -*-
 
-(in-package asdf)
+(defpackage #:cl-pdf-parser-system (:use #:common-lisp #:asdf))
+(in-package #:cl-pdf-parser-system)
 
-(defsystem :cl-pdf-parser
+(defsystem #:cl-pdf-parser
   :name "cl-pdf-parser"
   :author "Marc Battyani <marc.battyani@fractalconcept.com>"
   :maintainer "Marc Battyani <marc.battyani@fractalconcept.com>"
@@ -10,4 +11,4 @@
   :long-description "PDF parser"
   :components ((:file "pdf-parser" :depends-on ())
                (:file "pdf-template" :depends-on ("pdf-parser")))
-  :depends-on (:cl-pdf))
+  :depends-on (#:cl-pdf))
diff -ur cl-pdf-110.orig/config.lisp cl-pdf-110/config.lisp
--- cl-pdf-110.orig/config.lisp	2005-12-30 17:54:52.000000000 -0600
+++ cl-pdf-110/config.lisp	2005-12-30 18:26:52.000000000 -0600
@@ -32,7 +32,7 @@
 ;; The *afm-files-directories* is only for the 14 predefined fonts.
 ;; other fonts must have their afm files read only when they are loaded
 (defparameter *afm-files-directories*
-  (list (merge-pathnames #P"afm/*.afm" *cl-pdf-base-directory*))
+  '(#p"/usr/share/fonts/afm/")
   "The directory containing the Adobe Font Metrics files for the 14 predefined fonts")
 
 ;; define the :pdf-binary feature if your Lisp implementation accepts
diff -ur cl-pdf-110.orig/zlib.lisp cl-pdf-110/zlib.lisp
--- cl-pdf-110.orig/zlib.lisp	2005-12-30 17:54:52.000000000 -0600
+++ cl-pdf-110/zlib.lisp	2005-12-30 18:21:53.000000000 -0600
@@ -79,11 +79,9 @@
 
 ;;; salza zlib
 
-#+use-salza-zlib
 (defun load-zlib (&optional force)
   (setf *compress-streams* t))
 
-#+use-salza-zlib
 (defun compress-string (string)
   (let* ((input (if (stringp string)
 		    (deflate::string-to-octets string 0 (length string))
