diff -ur abcde-1.9.9.orig/abcde abcde-1.9.9/abcde
--- abcde-1.9.9.orig/abcde	Tue Mar 13 10:49:24 2001
+++ abcde-1.9.9/abcde	Mon Apr 16 21:14:26 2001
@@ -263,6 +263,7 @@
 	ALBUMFILE=$(mungefilename "$DALBUM")
 	ARTISTFILE=$(mungefilename "$TRACKARTIST")
 	TRACKFILE=$(mungefilename "$TRACKNAME")
+	TRACKNUM=$(printf %0.${TRACKNUMPADDING}d ${UTRACKNUM})
 	# Supported variables for OUTPUTFORMAT are ALBUMFILE, ARTISTFILE, TRACKFILE, and TRACKNUM.
 	if [ "$VARIOUSARTISTS" = "y" ]; then
 		OUTPUTFILE=$(eval echo $VAOUTPUTFORMAT)
@@ -307,7 +308,11 @@
 		TRACKFILE=$(mungefilename "$TRACKNAME")
 		ARTISTFILE=$(mungefilename "$TRACKARTIST")
 		TRACKNUM=$(printf %0.${TRACKNUMPADDING}d ${UTRACKNUM})
-		OUTPUTFILE=$(eval echo $OUTPUTFORMAT)
+		if [ "$VARIOUSARTISTS" = "y" ]; then
+			OUTPUTFILE=$(eval echo $VAOUTPUTFORMAT)
+		else
+			OUTPUTFILE=$(eval echo $OUTPUTFORMAT)
+		fi
 		if [ "$PLAYLISTDATAPREFIX" != "" ]; then
 			echo -n $PLAYLISTDATAPREFIX >> "$OUTPUTDIR/$PLAYLISTFILE"
 		fi
@@ -873,8 +878,8 @@
 fi
 
 # Load system defaults
-if [ -r /etc/abcde.conf ]; then
-	. /etc/abcde.conf
+if [ -r /etc/abcde/abcde.conf ]; then
+	. /etc/abcde/abcde.conf
 fi
 # Load user preference defaults
 if [ -r $HOME/.abcde.conf ]; then
--- abcde-1.9.9/Makefile.orig	Sun May 27 14:36:26 2001
+++ abcde-1.9.9/Makefile	Sun May 27 14:36:38 2001
@@ -1,12 +1,12 @@
 VERSION = 1.9.9
-INSTALL = /usr/bin/install -c
+INSTALL = /usr/bin/install -c
 
 # Installation directories
 prefix = ${DESTDIR}/usr
 exec_prefix = ${prefix}
 mandir = ${prefix}/share/man/man1
 bindir = ${exec_prefix}/bin
-etcdir = ${DESTDIR}/etc
+etcdir = ${DESTDIR}/etc/abcde
 
 all:
 
