diff -u -r abcde-2.1.9/Makefile abcde-2.1.9.new/Makefile
--- abcde-2.1.9/Makefile	2003-11-05 11:08:44.000000000 -0500
+++ abcde-2.1.9.new/Makefile	2003-11-27 13:14:57.000000000 -0500
@@ -1,12 +1,12 @@
 VERSION = 2.1.9
-INSTALL = /usr/bin/install -c
+INSTALL = /usr/bin/install -c
 
 # Installation directories
 prefix = ${DESTDIR}/usr
 exec_prefix = ${prefix}
 mandir = ${prefix}/share/man/man1
 bindir = ${exec_prefix}/bin
-etcdir = ${DESTDIR}/etc
+etcdir = ${DESTDIR}/etc/abcde
 
 all:
 
diff -u -r abcde-2.1.9/abcde abcde-2.1.9.new/abcde
--- abcde-2.1.9/abcde	2003-11-05 11:08:37.000000000 -0500
+++ abcde-2.1.9.new/abcde	2003-11-27 13:14:04.000000000 -0500
@@ -1281,8 +1281,8 @@
 fi
 
 # Load system defaults
-if [ -r /etc/abcde.conf ]; then
-	. /etc/abcde.conf
+if [ -r /etc/abcde/abcde.conf ]; then
+	. /etc/abcde/abcde.conf
 fi
 # Load user preference defaults
 if [ -r $HOME/.abcde.conf ]; then
