Index: libtunepimp-0.4.2/configure.in
===================================================================
--- libtunepimp-0.4.2.orig/configure.in
+++ libtunepimp-0.4.2/configure.in
@@ -64,7 +64,10 @@ AC_CHECK_LIB(z, compress,,
 	     ],)
 
 dnl Check for readline. This is optional.
-AC_CHECK_LIB(readline, readline,
+AC_ARG_WITH([readline], AC_HELP_STRING([--without-readline], [Disable readline support (default: check)]))
+
+if test "x$with_readline" != "xno"; then
+  AC_CHECK_LIB(readline, readline,
              [READLINE_LIBS="-lreadline -lncurses"
              AC_DEFINE(HAVE_LIBREADLINE,1,[Readline Support])],
              [echo "*"
@@ -73,22 +76,31 @@ AC_CHECK_LIB(readline, readline,
              echo "*  have readline support."
              echo "*"
 	     ],-ltermcap)
+fi
 
 AC_CHECK_HEADERS(iconv.h)
 
 dnl Check for TagLib 1.4
-AC_CHECK_TAGLIB(1.4.0, 
+AC_ARG_WITH([taglib], AC_HELP_STRING([--with-taglib], [Enable taglib support (default: check)]))
+
+if test "x$with_taglib" != "xno"; then
+  AC_CHECK_TAGLIB(1.4.0, 
                 [TP_PLUGINS="$TP_PLUGINS mpc wma"
                 AC_DEFINE(HAVE_TAGLIB,1,[TagLib Support])],
 		[AC_MSG_RESULT([no])
 		echo "*** NOT compiling TagLib support"
 		])
+fi
 
 dnl Check for mp4v2
-AC_CHECK_LIB(mp4v2, MP4Read, 
+AC_ARG_WITH([mp4v2], AC_HELP_STRING([--with-mp4v2], [Enable MP4 support (default: check)]))
+
+if test "x$with_mp4v2" != "xno"; then
+  AC_CHECK_LIB(mp4v2, MP4Read, 
              [TP_PLUGINS="$TP_PLUGINS mp4"
 	     AC_DEFINE(HAVE_MP4V2,1,[MP4 Support])],
 	     [echo "*** NOT compiling MP4 support"])
+fi
 
 dnl Find iconv. It may be in libiconv and may be iconv() or libiconv()
 if test "x$ac_cv_header_iconv_h" = "xyes"; then
@@ -126,8 +138,12 @@ if test "x$use_lgpl" = "xno"; then
                  ,-lm)
 fi
 
-AC_CHECK_LIB(ogg, ogg_sync_init, have_ogg="yes",)
-AC_CHECK_LIB(vorbisfile, ov_info, have_vorbis="yes",, -lvorbis -logg)
+AC_ARG_WITH([vorbis], AC_HELP_STRING([--without-vorbis], [Disable Vorbis support (default: check)]))
+
+if test "x$with_vorbis" != "xno"; then
+  AC_CHECK_LIB(ogg, ogg_sync_init, have_ogg="yes",)
+  AC_CHECK_LIB(vorbisfile, ov_info, have_vorbis="yes",, -lvorbis -logg)
+fi
 
 dnl Check for Ogg/Vorbis
 if test "x$have_ogg" = "xyes"; then
@@ -139,7 +155,11 @@ if test "x$have_ogg" = "xyes"; then
   fi
 fi
 
-AC_CHECK_LIB(FLAC, FLAC__metadata_object_new, have_flac="yes",, -lFLAC -lm)
+AC_ARG_WITH([flac], AC_HELP_STRING([--without-flac], [Disable FLAC support (default: check)]))
+
+if test "x$with_flac" != "xno"; then
+  AC_CHECK_LIB(FLAC, FLAC__metadata_object_new, have_flac="yes",, -lFLAC -lm)
+fi
 
 dnl Check for libFLAC
 if test "x$have_flac" = "xyes"; then
