diff -ur sound-juicer-0.1/configure.in sound-juicer-0.1.patched/configure.in
--- sound-juicer-0.1/configure.in	2003-04-10 18:54:32.000000000 +0200
+++ sound-juicer-0.1.patched/configure.in	2003-04-22 00:28:23.000000000 +0200
@@ -55,8 +55,8 @@
 fi
 AM_GCONF_SOURCE_2
 
-AM_GST_ELEMENT_CHECK(cdparanoia,,AC_MSG_ERROR([The 'cdparanoa' element was not found. This will cause Sound Juicer to fail at runtime.]))
-AM_GST_ELEMENT_CHECK(vorbisenc,,AC_MSG_ERROR([The 'vorbisenc' element was not found. This will cause Sound Juicer to fail at runtime.]))
+AM_GST_ELEMENT_CHECK(cdparanoia,,AC_MSG_WARN([The 'cdparanoia' element was not found. This will cause Sound Juicer to fail at runtime.]))
+AM_GST_ELEMENT_CHECK(vorbisenc,,AC_MSG_WARN([The 'vorbisenc' element was not found. This will cause Sound Juicer to fail at runtime.]))
 AM_GST_ELEMENT_CHECK(filesink,,AC_MSG_WARN([The 'filesink' element was not found. This will cause Sound Juicer to fail at runtime.]))
 
 AC_OUTPUT([
