--- gcc-3.4.1/gcc/config/i386/t-cygming.jj	2004-07-21 19:00:22.495332312 +0200
+++ gcc-3.4.1/gcc/config/i386/t-cygming	2004-07-21 19:01:08.511336816 +0200
@@ -8,7 +8,8 @@
 # If we are building next to winsup, this will let us find the real
 # limits.h when building libgcc2.  Otherwise, winsup must be installed
 # first.
-LIBGCC2_INCLUDES = -I$(srcdir)/../winsup/w32api/include
+LIBGCC2_INCLUDES = -I$(srcdir)/../winsup/w32api/include \
+	-I$(srcdir)/../winsup/cygwin/include
 
 LIB2FUNCS_EXTRA = $(srcdir)/config/i386/w32-shared-ptr.c
 
@@ -19,3 +20,5 @@
 	$(srcdir)/config/i386/winnt.c
 
 STMP_FIXINC=stmp-fixinc
+
+CRTSTUFF_CFLAGS += -isystem $(srcdir)/../winsup/cygwin/include
