diff --git a/kioslave/CMakeLists.txt b/kioslave/CMakeLists.txt
index 379ea2d..f3d5b00 100644
--- a/kioslave/CMakeLists.txt
+++ b/kioslave/CMakeLists.txt
@@ -43,7 +43,11 @@ if(NOT WIN32)
    add_subdirectory( floppy )
    add_subdirectory( finger )
    add_subdirectory( man )
-   add_subdirectory( nfs )  
+   check_include_files(rpc/rpc.h HAVE_RPC_RPC_H)
+   macro_log_feature(HAVE_RPC_RPC_H "rpc library" "library routines for remote procedure calls" "" FALSE "" "Needed to build the NFS kioslave")
+   if(HAVE_RPC_RPC_H)
+     add_subdirectory( nfs )  
+   endif(HAVE_RPC_RPC_H)
 endif(NOT WIN32)
 
 if(SAMBA_FOUND OR WIN32 AND NOT WINCE)
