diff -aurp old/netfilter/iptables_display.sh new/netfilter/iptables_display.sh
--- old/netfilter/iptables_display.sh	2008-10-27 20:53:09.000000000 +0000
+++ new/netfilter/iptables_display.sh	2008-10-27 20:53:38.000000000 +0000
@@ -1,6 +1,6 @@
 #! /bin/sh
 # $Id: iptables_display.sh,v 1.3 2006/11/23 12:32:56 nanard Exp $
-IPTABLES=iptables
+IPTABLES=/sbin/iptables
 
 #display all chains relative to miniupnpd
 $IPTABLES -v -n -t nat -L PREROUTING
diff -aurp old/netfilter/iptables_flush.sh new/netfilter/iptables_flush.sh
--- old/netfilter/iptables_flush.sh	2008-10-27 20:53:09.000000000 +0000
+++ new/netfilter/iptables_flush.sh	2008-10-27 20:53:46.000000000 +0000
@@ -1,6 +1,6 @@
 #! /bin/sh
 # $Id: iptables_flush.sh,v 1.2 2006/11/23 12:32:57 nanard Exp $
-IPTABLES=iptables
+IPTABLES=/sbin/iptables
 
 #flush all rules owned by miniupnpd
 $IPTABLES -t nat -F MINIUPNPD
diff -aurp old/netfilter/iptables_init.sh new/netfilter/iptables_init.sh
--- old/netfilter/iptables_init.sh	2008-10-27 20:53:09.000000000 +0000
+++ new/netfilter/iptables_init.sh	2008-10-27 20:54:39.000000000 +0000
@@ -1,6 +1,6 @@
 #! /bin/sh
 # $Id: iptables_init.sh,v 1.4 2008/04/25 18:15:08 nanard Exp $
-IPTABLES=iptables
+IPTABLES=/sbin/iptables
 
 #change this parameters :
 EXTIF=eth0
diff -aurp old/netfilter/iptables_removeall.sh new/netfilter/iptables_removeall.sh
--- old/netfilter/iptables_removeall.sh	2008-10-27 20:53:09.000000000 +0000
+++ new/netfilter/iptables_removeall.sh	2008-10-27 20:54:48.000000000 +0000
@@ -1,6 +1,6 @@
 #! /bin/sh
 # $Id: iptables_removeall.sh,v 1.4 2008/04/25 18:15:09 nanard Exp $
-IPTABLES=iptables
+IPTABLES=/sbin/iptables
 
 #change this parameters :
 EXTIF=eth0
