diff -dupr miniupnpd-old/netfilter/iptcrdr.c miniupnpd-20080427/netfilter/iptcrdr.c
--- miniupnpd-old/netfilter/iptcrdr.c	2008-09-07 18:24:13.108005833 +0200
+++ miniupnpd-20080427/netfilter/iptcrdr.c	2008-09-07 18:41:11.956007153 +0200
@@ -20,11 +20,13 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
 #include <linux/netfilter_ipv4/ip_nat.h>
 #else
-#include <linux/netfilter/nf_nat.h>
-/*
+#include <stdbool.h> /* bool */
+	/* below is in <kerneldir>/include/linux/kernel.h, but not in /usr/include/linux/kernel.h,
+	 * which gcc uses */
+#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
+#include <net/netfilter/nf_nat.h>
 #define ip_nat_multi_range	nf_nat_multi_range
 #define ip_nat_range		nf_nat_range
-*/
 #endif
 
 #include "iptcrdr.h"
