Index: simple-scan-2.31.90.2/configure.ac
===================================================================
--- simple-scan-2.31.90.2.orig/configure.ac	2010-08-30 08:49:27.000000000 +1000
+++ simple-scan-2.31.90.2/configure.ac	2010-08-30 08:49:41.147411001 +1000
@@ -30,7 +30,6 @@
     gconf-2.0
     gmodule-export-2.0
     gthread-2.0
-    zlib
     cairo
     gdk-pixbuf-2.0
     cairo-pdf
@@ -41,6 +40,7 @@
 
 AC_CHECK_HEADERS([sane/sane.h],[],[AC_MSG_ERROR([SANE not found])])
 AC_CHECK_HEADERS([sane/saneopts.h],[],[AC_MSG_ERROR([SANE not found])])
+AC_CHECK_HEADERS([zlib.h],[],[AC_MSG_ERROR([ZLib not found])])
 AC_CHECK_HEADERS([jpeglib.h],[],[AC_MSG_ERROR([libjpeg not found])])
 
 dnl ###########################################################################
Index: simple-scan-2.31.90.2/src/Makefile.am
===================================================================
--- simple-scan-2.31.90.2.orig/src/Makefile.am	2010-08-30 08:49:24.000000000 +1000
+++ simple-scan-2.31.90.2/src/Makefile.am	2010-08-30 08:49:29.997411002 +1000
@@ -30,6 +30,7 @@
 	$(SIMPLE_SCAN_LIBS) \
 	-lsane \
 	-ljpeg \
+	-lz \
 	-lm
 
 DISTCLEANFILES = \
