diff -uNr newt-0.52.7.orig/Makefile.in newt-0.52.7/Makefile.in
--- newt-0.52.7.orig/Makefile.in	2008-01-15 17:04:32.000000000 +0100
+++ newt-0.52.7/Makefile.in	2008-01-15 17:04:58.000000000 +0100
@@ -107,25 +107,25 @@
 	$(CC) $(SHCFLAGS) -c $(CFLAGS) $(CPPFLAGS) -o $@ $<
 
 install: $(LIBNEWT) install-sh whiptail
-	[ -d $(instroot)/$(bindir) ] || install -m 755 -d $(instroot)/$(bindir)
-	[ -d $(instroot)/$(libdir) ] || install -m 755 -d $(instroot)/$(libdir)
-	[ -d $(instroot)/$(includedir) ] || install -m 755 -d $(instroot)/$(includedir)
-	[ -d $(instroot)/$(man1dir) ] || install -m 755 -d $(instroot)/$(man1dir)
-	install -m 644 newt.h $(instroot)/$(includedir)
-	install -m 644 $(LIBNEWT) $(instroot)/$(libdir)
-	install -m 755 whiptail $(instroot)/$(bindir)
-	install -m 644 whiptail.1 $(instroot)/$(man1dir)
-	make -C po datadir=$(instroot)/$(datadir) install
+	[ -d $(DESTDIR)/$(bindir) ] || install -m 755 -d $(DESTDIR)/$(bindir)
+	[ -d $(DESTDIR)/$(libdir) ] || install -m 755 -d $(DESTDIR)/$(libdir)
+	[ -d $(DESTDIR)/$(includedir) ] || install -m 755 -d $(DESTDIR)/$(includedir)
+	[ -d $(DESTDIR)/$(man1dir) ] || install -m 755 -d $(DESTDIR)/$(man1dir)
+	install -m 644 newt.h $(DESTDIR)/$(includedir)
+	install -m 644 $(LIBNEWT) $(DESTDIR)/$(libdir)
+	install -m 755 whiptail $(DESTDIR)/$(bindir)
+	install -m 644 whiptail.1 $(DESTDIR)/$(man1dir)
+	make -C po datadir=$(DESTDIR)/$(datadir) install
 
 install-sh: sharedlib _snackmodule.so
-	[ -d $(instroot)/$(libdir) ] || install -m 755 -d $(instroot)/$(libdir)
-	install -m 755 $(LIBNEWTSH) $(instroot)/$(libdir)
-	ln -sf $(LIBNEWTSONAME) $(instroot)/$(libdir)/libnewt.so
-	ln -sf $(LIBNEWTSH) $(instroot)/$(libdir)/$(LIBNEWTSONAME)
+	[ -d $(DESTDIR)/$(libdir) ] || install -m 755 -d $(DESTDIR)/$(libdir)
+	install -m 755 $(LIBNEWTSH) $(DESTDIR)/$(libdir)
+	ln -sf $(LIBNEWTSONAME) $(DESTDIR)/$(libdir)/libnewt.so
+	ln -sf $(LIBNEWTSH) $(DESTDIR)/$(libdir)/$(LIBNEWTSONAME)
 	for ver in $(PYTHONVERS) ; do \
-	   [ -d $(instroot)/$(libdir)/$$ver/site-packages ] || install -m 755 -d $(instroot)/$(libdir)/$$ver/site-packages ;\
-	   install -m 755 $$ver/_snackmodule.so $(instroot)/$(libdir)/$$ver/site-packages ;\
-	   install -m 644 snack.py $(instroot)/$(libdir)/$$ver/site-packages ;\
+	   [ -d $(DESTDIR)/$(libdir)/$$ver/site-packages ] || install -m 755 -d $(DESTDIR)/$(libdir)/$$ver/site-packages ;\
+	   install -m 755 $$ver/_snackmodule.so $(DESTDIR)/$(libdir)/$$ver/site-packages ;\
+	   install -m 644 snack.py $(DESTDIR)/$(libdir)/$$ver/site-packages ;\
 	done
 
 configure:  configure.ac newt.spec
