diff -Nur kshutdown-0.6.0.orig/kshutdown/Makefile.am kshutdown-0.6.0/kshutdown/Makefile.am
--- kshutdown-0.6.0.orig/kshutdown/Makefile.am	2005-01-09 22:46:13.000000000 +0100
+++ kshutdown-0.6.0/kshutdown/Makefile.am	2005-03-10 16:16:40.000000000 +0100
@@ -10,6 +10,7 @@
 
 install-data-local:
 	$(mkinstalldirs) $(kde_appsdir)/Utilities/
+	$(mkinstalldirs) $(kde_datadir)/kconf_update/
 	$(INSTALL_DATA) $(srcdir)/kshutdown.desktop $(kde_appsdir)/Utilities/kshutdown.desktop
 	$(INSTALL_DATA) $(srcdir)/kshutdownwizard.desktop $(kde_appsdir)/Utilities/kshutdownwizard.desktop
 	$(INSTALL_DATA) $(srcdir)/kshutdown.upd $(kde_datadir)/kconf_update/kshutdown.upd
