Index: pulseaudio-0.9.2/src/daemon/esdcompat.in
===================================================================
--- pulseaudio-0.9.2.orig/src/daemon/esdcompat.in
+++ pulseaudio-0.9.2/src/daemon/esdcompat.in
@@ -28,12 +28,8 @@ fail() {
 
 ARGS=" --log-target=syslog"
 
-for N in $(seq $#) ; do
-
+while [ -n "$1" ]; do
     case "$1" in
-        "")
-            ;;
-
         -v|--version)
             echo "$VERSION_STRING"
             exit 0
@@ -70,20 +66,23 @@ EOF
         -spawnpid)
             shift
             ARGS="$ARGS '-Lmodule-esound-compat-spawnpid pid=$1'"
+	    shift
             ;;
 
         -spawnfd)
             shift
             ARGS="$ARGS '-Lmodule-esound-compat-spawnfd fd=$1'"
+	    shift
             ;;
         
         -unix|-b|-public|-terminate|-nobeeps|-trust|-tcp|-promiscuous)  
             # Ignore these commands
+	    shift
             ;; 
 
         -d|-r|-as|-port|-bind)
             # Ignore these commands and their arguments
-            shift
+            shift; shift
 
             ;;
 
