From 05f56bf49ee0de8979c29c11f61a01e30e964444 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Tue, 20 Jul 2010 17:27:50 -0400
Subject: [PATCH] fix parallel build between ubi-utils and mkfs.ubifs

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Artem Bityutskiy <Artem.Bityutskiy@nokia.com>
---
 Makefile |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/Makefile b/Makefile
index b558705..d315f39 100644
--- a/Makefile
+++ b/Makefile
@@ -25,6 +25,9 @@ LDDEPS = $(BUILDDIR)/lib/libmtd.a
 
 include common.mk
 
+# mkfs.ubifs needs -lubi which is in ubi-utils/
+subdirs_mkfs.ubifs_all: subdirs_ubi-utils_all
+
 clean::
 	-rm -f $(SYMLINKS)
 ifneq ($(BUILDDIR)/.git,)
-- 
1.7.1.1

