--- configure.in	2003-08-10 01:17:22.000000000 +0000
+++ configure.in	2004-02-06 14:15:57.889662976 +0000
@@ -179,7 +188,7 @@
 	AC_DEFINE(LINUX)
 
 	AC_MSG_CHECKING(for a dumb udphdr declaration)
-	AC_TRY_COMPILE(,
+	AC_TRY_COMPILE(
 #ifndef __FAVOR_BSD
 #define __FAVOR_BSD
 #endif
@@ -189,7 +198,7 @@
 #endif
 
 #include <netinet/udp.h>
-
+,
 struct udphdr foo;
 unsigned short bar = foo.uh_sport;
 ,
