diff -urNp quintuple-agent-1.0.3.old/configure quintuple-agent-1.0.3/configure
--- quintuple-agent-1.0.3.old/configure	2001-07-04 00:25:32.000000000 +0300
+++ quintuple-agent-1.0.3/configure	2006-09-20 20:31:25.000000000 +0300
@@ -4062,6 +4062,7 @@ else
 #line 4063 "configure"
 #include "confdefs.h"
 #include <sys/types.h>
+#include <unistd.h>
 #if STDC_HEADERS
 #include <stdlib.h>
 #include <stddef.h>
