diff -ru usr/src/nv/nv.c usr/src/nv.1435131/nv.c
--- usr/src/nv/nv.c	2005-11-23 03:58:01.000000000 +0100
+++ usr/src/nv.1435131/nv.c	2005-12-15 19:32:42.816768750 +0100
@@ -1084,9 +1084,12 @@
 
         // broken kernels may get confused after splitting the page and
         // restore the page before returning to us. detect that case.
-        if ( (pte_val(*kpte) == kpte_val) &&
-             (pte_val(*kpte) & _PAGE_PSE))
+        if (((pte_val(*kpte) & ~_PAGE_NX) == kpte_val) &&
+            (pte_val(*kpte) & _PAGE_PSE))
         {
+            if ((pte_val(*kpte) & _PAGE_NX) &&
+                    (__nv_supported_pte_mask & _PAGE_NX) == 0)
+                clear_bit(_PAGE_BIT_NX, kpte);
             spin_unlock(&init_mm.page_table_lock);
             // don't change the page back, as it's already been reverted
             put_page(kpte_page);
