Index: Makefile.am
===================================================================
RCS file: /cvs/gnome/gnumeric/plugins/corba/Makefile.am,v
retrieving revision 1.5
retrieving revision 1.6
diff -u -r1.5 -r1.6
--- Makefile.am	24 Sep 2004 01:33:18 -0000	1.5
+++ Makefile.am	22 Feb 2005 17:07:22 -0000	1.6
@@ -7,23 +7,26 @@
 xmldir = $(gnumeric_plugin_corbadir)
 gnumeric_plugin_corba_LTLIBRARIES = corba.la
 corba_la_LDFLAGS = -module $(GNUMERIC_PLUGIN_LDFLAGS)
-corba_la_SOURCES = $(GNUMERIC_GENERATED)	\
+corba_la_SOURCES =				\
 	corba-application.c 			\
 	corba-workbook.c 			\
 	corba-workbook.h 			\
 	corba-sheet.c 				\
 	corba-sheet.h
 
-GNUMERIC_GENERATED = 		\
+nodist_corba_la_SOURCES =		\
 	GNOME_Gnumeric-common.c		\
 	GNOME_Gnumeric-skels.c		\
 	GNOME_Gnumeric-stubs.c		\
 	GNOME_Gnumeric.h
 
-stamp_Gnumeric_idl : $(srcdir)/GNOME_Gnumeric.idl 
+# A hint is needed to build the header first:
+BUILT_SOURCES = GNOME_Gnumeric.h
+
+stamp_Gnumeric_idl: $(srcdir)/GNOME_Gnumeric.idl 
 	$(ORBIT_IDL) -I$(srcdir) $< && touch $@
 
-$(GNUMERIC_GENERATED): stamp_Gnumeric_idl
+$(nodist_corba_la_SOURCES): stamp_Gnumeric_idl
 
 idldir   = $(gnumeric_datadir)/idl
 idl_DATA = GNOME_Gnumeric.idl 
@@ -33,11 +36,5 @@
 
 @INTLTOOL_XML_RULE@
 
-# Do not use built sources BUILT_SOURCES
-# it attempts to build the source even if it is not used.
-DISTCLEANFILES	= $(GNUMERIC_GENERATED) stamp_Gnumeric_idl $(xml_DATA)
+CLEANFILES	= $(nodist_corba_la_SOURCES) stamp_Gnumeric_idl $(xml_DATA)
 EXTRA_DIST	= $(idl_DATA) $(xml_in_files)
-
-dist-hook :
-	cd $(distdir) && rm -f $(DISTCLEANFILES)
-
