diff -ur meld-0.8.2.orig/meld meld-0.8.2/meld
--- meld-0.8.2.orig/meld	2003-07-01 20:06:46.000000000 +0000
+++ meld-0.8.2/meld	2003-07-01 20:08:33.000000000 +0000
@@ -52,6 +52,12 @@
     print "Due to incompatible API changes some functions may not operate as expected."
 
 #
+# Add the library path to the system load path
+#
+import os
+sys.path = [os.curdir, '/usr/lib/meld'] + sys.path
+
+#
 # main
 #
 import meldapp
diff -ur meld-0.8.2.orig/meld.desktop meld-0.8.2/meld.desktop
--- meld-0.8.2.orig/meld.desktop	2003-07-01 20:06:46.000000000 +0000
+++ meld-0.8.2/meld.desktop	2003-07-01 20:09:17.000000000 +0000
@@ -6,6 +6,6 @@
 Comment=Compare and merge your files.
 TryExec=meld
 Exec=meld
-Path=./meld
 Terminal=false
 Categories=GNOME;Application;Development;
+Icon=/usr/share/meld/glade2/pixmaps/icon.png
diff -ur meld-0.8.2.orig/meldapp.py meld-0.8.2/meldapp.py
--- meld-0.8.2.orig/meldapp.py	2003-07-01 20:06:46.000000000 +0000
+++ meld-0.8.2/meldapp.py	2003-07-01 20:10:27.000000000 +0000
@@ -453,7 +453,7 @@
         gnome.url_show("http://meld.sourceforge.net")
 
     def on_menu_users_manual_activate(self, button):
-        gnome.url_show("file:///"+os.path.abspath(misc.appdir("manual/index.html") ) )
+	gnome.url_show("file:///usr/share/doc/meld-0.8.2/html/index.html")
 
     def on_menu_about_activate(self, *extra):
         about = gtk.glade.XML(misc.appdir("glade2/meld-app.glade"),"about").get_widget("about")
diff -ur meld-0.8.2.orig/misc.py meld-0.8.2/misc.py
--- meld-0.8.2.orig/misc.py	2003-07-01 20:06:46.000000000 +0000
+++ meld-0.8.2/misc.py	2003-07-01 20:16:42.000000000 +0000
@@ -49,7 +49,7 @@
 def appdir(pathin):
     """Return where the application is installed.
     """
-    where = os.path.dirname(sys.argv[0])
+    where = /usr/share/meld
     pathout = os.path.join( where, pathin )
     if not os.path.exists(pathout):
         run_dialog("Cannot find '%s'\nI looked in '%s'\n(%s)" % (pathin,where,pathout), gtk.MESSAGE_ERROR)
