--- Makefile.am.old	2005-12-08 20:02:54.000000000 -0700
+++ Makefile.am	2005-12-18 02:49:59.000000000 -0700
@@ -20,7 +20,11 @@
 #
 #  Process this file with autoconf to create configure.
 
+if NONMAN
 SUBDIRS = hardcopy man sgml
+else
+SUBDIRS = man
+endif
 
 EXTRA_DIST = \
 	misc/xlogo.epsi \
--- configure.ac.old	2005-12-14 17:24:13.000000000 -0700
+++ configure.ac	2005-12-18 02:49:59.000000000 -0700
@@ -34,6 +34,9 @@
 AC_SUBST([X11DOCDIR])
 AC_SUBST([HARDCOPYDIR])
 
+AC_ARG_ENABLE(non-man-docs,            AS_HELP_STRING([--enable-non-man-docs], [Install extra (non-man) documentation (default: no)]), [NONMAN=$enableval], [NONMAN=no])
+AM_CONDITIONAL(NONMAN, [test "x$NONMAN" = xyes])
+
 XORG_MANPAGE_SECTIONS
 XORG_RELEASE_VERSION
 
