diff -ruwN libao-0.8.4-old/configure.in libao-0.8.4/configure.in
--- libao-0.8.4-old/configure.in	2003-10-04 17:34:10.000000000 +0200
+++ libao-0.8.4/configure.in	2003-12-28 16:44:40.518164170 +0100
@@ -174,8 +174,8 @@
 
 if test "$BUILD_ESD" = "yes"; then
   AM_PATH_ESD(0.2.8, have_esd=yes, have_esd=no)
-  AM_CONDITIONAL(HAVE_ESD,test "x$have_esd" = xyes)
 fi
+AM_CONDITIONAL(HAVE_ESD,test "x$have_esd" = xyes)
 
 
 dnl Check for OSS
