--- src/plugins/alsa09/ao_alsa09.c	2001-12-19 07:39:24.000000000 +0900
+++ ao_alsa09.c	2003-03-23 17:30:40.000000000 +0900
@@ -192,18 +192,6 @@
 	if (err < 0)
 		goto error;
 
-	cmd = "snd_pcm_hw_params_set_period_size";
-	err = snd_pcm_hw_params_set_period_size(internal->pcm_handle, hwparams,
-			internal->buf_size / internal->sample_size, 0);
-	if (err < 0)
-		goto error;
-
-	cmd = "snd_pcm_hw_params_set_periods";
-	err = snd_pcm_hw_params_set_periods(internal->pcm_handle, hwparams,
-			internal->periods * format->channels, 0);
-	if (err < 0)
-		goto error;
-
 	cmd = "snd_pcm_hw_params";
 	err = snd_pcm_hw_params(internal->pcm_handle, hwparams);
 	if (err < 0)
