--- usr/bin/x2gomountdirs.org	2010-01-18 22:18:16.000000000 +0100
+++ usr/bin/x2gomountdirs	2010-01-18 22:16:33.000000000 +0100
@@ -144,34 +144,34 @@
             if(system("sshfs  -o Cipher=blowfish,IdentityFile=$key,UserKnownHostsFile=$key.ident $user\@$host:\"@dirs[$i]\" \"$mntpath\" -p $port 2>>~/mounts.log")==0)
             {   
         	print "mount @dirs[$i] ok\n";
-		if(! $printspool)
-		{
-        	    my $fname="$ENV{'HOME'}/Desktop";
-		    my $p=@dirs[$i];
-        	    $p=~s/\/cygdrive\///g;
-            	    $p=~s/\//_/g;
-		    $fname="$fname/$p";
-		    if($type eq "dir")
-		    {
-	    		$fname="$fname(sshfs-disk)";
-		    }
-		    else
-		    {
-			if($fname =~ m/CDROM/)
-			{
-	    	    	    $fname="$fname(sshfs-cdrom)";
-			}
-			else
-			{
-	    	    	    $fname="$fname(sshfs-removable)";
-			}
-			$fname=~s/_ramdrive_mnt_//;
-		    }
-		    print "fname: $fname\n";
-		    open(F,">$fname");
-		    print F "$mntpath\n$session\n\n\0";
-		    close(F);
-		}
+#		if(! $printspool)
+#		{
+#        	    my $fname="$ENV{'HOME'}/Desktop";
+#		    my $p=@dirs[$i];
+#        	    $p=~s/\/cygdrive\///g;
+#            	    $p=~s/\//_/g;
+#		    $fname="$fname/$p";
+#		    if($type eq "dir")
+#		    {
+#	    		$fname="$fname(sshfs-disk)";
+#		    }
+#		    else
+#		    {
+#			if($fname =~ m/CDROM/)
+#			{
+#	    	    	    $fname="$fname(sshfs-cdrom)";
+#			}
+#			else
+#			{
+#	    	    	    $fname="$fname(sshfs-removable)";
+#			}
+#			$fname=~s/_ramdrive_mnt_//;
+#		    }
+#		    print "fname: $fname\n";
+#		    open(F,">$fname");
+#		    print F "$mntpath\n$session\n\n\0";
+#		    close(F);
+#		}
     	    }
     	    else
     	    {
