Index: libcdio-0.76/configure.ac
===================================================================
--- libcdio-0.76.orig/configure.ac
+++ libcdio-0.76/configure.ac
@@ -436,7 +436,7 @@ int has_timeout=sizeof(test.timeout);],
  	LIBS="$LIBS -lwinmm -mwindows"
   	cd_drivers="${cd_drivers}, MinGW "
 	;;
-     freebsd4.*|freebsd5.*|freebsd6*)
+     freebsd4.*|freebsd5.*|freebsd6*|dragonfly*)
      	AC_DEFINE([HAVE_FREEBSD_CDROM], [1], 
                   [Define 1 if you have FreeBSD CD-ROM support])
  	LIBS="$LIBS -lcam"
