Index: tork-0.07/configure.in.in
===================================================================
--- tork-0.07.orig/configure.in.in
+++ tork-0.07/configure.in.in
@@ -209,22 +209,3 @@ if test "$ac_cv_openssl_linker_option" !
    LDFLAGS="$ac_cv_openssl_linker_option $LDFLAGS"
 fi
 
-
-# - GeoIP
-
-GEOVERSION=1.4.0
-PACKAGE=GeoIP
-GEOIP_VERSION_INFO=`echo $GEOVERSION | awk -F. '{ printf "%d:%d:%d", $1+$2, $3, $2 }'`
-AC_SUBST(GEOIP_VERSION_INFO)
-
-AC_ARG_WITH(city,
-[  --with-city=\"no\"   Enable/disable support for GeoIP City Edition. (yes)])
-if test "$with_city" = "no" ; then
-  echo Disabling GeoIP City Edition support
-  WITH_CITY=1
-else
-  echo Enabling GeoIP City Edition support
-  WITH_CITY=0
-fi
-AC_SUBST(WITH_CITY)
-AM_CONDITIONAL(WITH_CITY, test "$with_city" != "no")
Index: tork-0.07/src/Makefile.am
===================================================================
--- tork-0.07.orig/src/Makefile.am
+++ tork-0.07/src/Makefile.am
@@ -4,7 +4,7 @@
 # of the other Makefile.am variables
 bin_PROGRAMS = tork
 
-SUBDIRS=icons about arkollon GeoIP-1.4.0 .
+SUBDIRS=icons about arkollon .
 
 # set the include path for X, qt and KDE
 INCLUDES         = $(all_includes)
@@ -16,7 +16,7 @@ METASOURCES = AUTO
 tork_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 # the libraries to link against.
-tork_LDADD   = $(LIB_KFILE) $(LIB_KHTML) -L GeoIP-1.4.0/libGeoIP/libGeoIP.la
+tork_LDADD   = $(LIB_KFILE) $(LIB_KHTML) -lGeoIP
 
 # which sources should be compiled for tork
 tork_SOURCES = torkconfig.kcfgc main.cpp tork.cpp torkview.cpp torkview_base.ui \
Index: tork-0.07/src/torkview.cpp
===================================================================
--- tork-0.07.orig/src/torkview.cpp
+++ tork-0.07/src/torkview.cpp
@@ -30,7 +30,7 @@
 #include "portsandicons.h"
 #include "torkconfig.h"
 
-#include "GeoIP-1.4.0/libGeoIP/GeoIP.h"
+#include "GeoIP.h"
 
 
 #include "kdebug.h"
