Use system libpcre and make iselect not blow up

--- wml_backend/p2_mp4h/Makefile.am
+++ wml_backend/p2_mp4h/Makefile.am
@@ -1,15 +1,15 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-DIST_SUBDIRS = intl pcre lib src po doc modules tests
+DIST_SUBDIRS = intl lib src po doc modules tests
 
 if LOADABLE_MODULES
-SUBDIRS = intl pcre lib src po doc modules tests
+SUBDIRS = intl lib src po doc modules tests
 else
-SUBDIRS = intl pcre lib src po doc tests
+SUBDIRS = intl lib src po doc tests
 endif
 
-EXTRA_DIST = CREDITS README README.pcre bootstrap.sh shtool
+EXTRA_DIST = CREDITS README bootstrap.sh shtool
 
 deb:
 	if test -f ../mp4h_$(VERSION).orig.tar.gz; then \
--- wml_backend/p2_mp4h/configure.ac
+++ wml_backend/p2_mp4h/configure.ac
@@ -179,7 +179,6 @@ AC_SUBST(CFLAGS)
 AC_CONFIG_FILES([
    Makefile
    intl/Makefile
-   pcre/Makefile
    lib/Makefile
    src/Makefile
    po/Makefile.in
--- wml_backend/p2_mp4h/src/Makefile.am
+++ wml_backend/p2_mp4h/src/Makefile.am
@@ -1,5 +1,5 @@
 
-INCLUDES = -I$(top_srcdir)/pcre -I$(top_srcdir)/lib @INTLINCL@
+INCLUDES = -I$(top_srcdir)/lib @INTLINCL@
 
 bin_PROGRAMS = mp4h
 
@@ -16,7 +16,7 @@ mp4h_SOURCES = builtin.c debug.c devel.c
 if LOADABLE_MODULES
 mp4h_LDFLAGS = -export-dynamic
 endif
-mp4h_LDADD   = -lm $(top_builddir)/lib/libmp4h.a $(top_builddir)/pcre/libpcre.a @INTLLIBS@
+mp4h_LDADD   = -lm $(top_builddir)/lib/libmp4h.a -lpcre @INTLLIBS@
 
 include_HEADERS = mp4h.h
 noinst_HEADERS  = builtin.h ltdl.h
--- wml_aux/iselect/configure.in
+++ wml_aux/iselect/configure.in
@@ -3,6 +3,7 @@
 dnl ##  Copyright (c) Ralf S. Engelschall, <rse@engelschall.com>
 dnl ##
 
+AC_INIT(README)
 AC_PREREQ(2.10)dnl
 AC_REVISION($Revision: 1.3 $)
 SHTOOL="./etc/shtool"
@@ -11,7 +12,6 @@
 echo "${T_MD}Configuring for iSelect `./etc/shtool version -l c -d long iselect_version.c`${T_ME}"
 echo "Copyright (c) 1996-1999 Ralf S. Engelschall, All Rights Reserved."
 
-AC_INIT(README)
 AC_CONFIG_HEADER(config_ac.h)dnl
 AC_PREFIX_DEFAULT(/usr/local)
 
--- wml_aux/iselect/aclocal.m4
+++ wml_aux/iselect/aclocal.m4
@@ -186,24 +186,6 @@
 dnl ##
 dnl ##########################################################
 dnl
-AC_DEFUN(AC_C_PROTOTYPES,
-[AC_MSG_CHECKING([for function prototypes])
-AC_CACHE_VAL(ac_cv_c_prototypes,
-[AC_TRY_LINK([#ifndef __STDC__
-Syntax Error
-#endif], [extern int test (int i, double x);
-struct s1 {int (*f) (int a);};
-struct s2 {int (*f) (double a);};],
-  ac_cv_c_prototypes=yes, ac_cv_c_prototypes=no)])dnl
-AC_MSG_RESULT([$ac_cv_c_prototypes])
-if test $ac_cv_c_prototypes = yes; then
-  AC_DEFINE(HAVE_PROTOTYPES)
-  ANSI_CC=yes
-else
-  ANSI_CC=no
-fi
-AC_SUBST(ANSI_CC)
-])dnl
 dnl
 dnl
 dnl ##########################################################
