*** Makefile.orig	Thu Sep 13 00:57:10 2001
--- Makefile	Thu Sep 13 00:57:48 2001
***************
*** 39,45 ****
  	rm -f $(PROG) *.o *~
   
   install:
! 	install -c -m 755 -o root -g mail -s $(PROG) $(DESTDIR)/usr/bin/
! 	install -c -m 644 $(MFILES) $(DESTDIR)/usr/man/man1/
  	cd misc && install -c -m 644 $(EFILES) $(DESTDIR)/etc/
  	cd misc && install -c -m 644 $(SFILES) $(DESTDIR)/usr/lib/
--- 39,45 ----
  	rm -f $(PROG) *.o *~
   
   install:
! 	install -c -m 755 -o root -g mail -s $(PROG) $(DESTDIR)/bin/
! 	install -c -m 644 $(MFILES) $(DESTDIR)/usr/share/man/man1/
  	cd misc && install -c -m 644 $(EFILES) $(DESTDIR)/etc/
  	cd misc && install -c -m 644 $(SFILES) $(DESTDIR)/usr/lib/
