--- src/bigeye.c.orig	2004-02-15 19:12:19.270277021 +0000
+++ src/bigeye.c	2004-02-15 19:12:37.477869417 +0000
@@ -1132,7 +1132,7 @@
  lt = time(NULL);
  ptr = localtime(&lt);
 
- fprintf(fp, "IP address: %s - time: %sPort: %s - Protocol: 
+ fprintf(fp, "IP address: %s - time: %sPort: %s - Protocol: \
 %s\n",connection,asctime(ptr),port,protocol);
  if(emulate == 1 && signature != NULL)  fprintf(fp,"server response: %s\n",signature);
  if(emulate == 0 && signature != NULL)	fprintf(fp,"server response: %s\n",signature);
