Index: baselayout-1.13.0_alpha1/Makefile
===================================================================
--- baselayout-1.13.0_alpha1.orig/Makefile
+++ baselayout-1.13.0_alpha1/Makefile
@@ -18,13 +18,13 @@ ARCH = x86
 OS = Linux
 
 BASE_DIRS = /$(LIB)/rcscripts/init.d /$(LIB)/rcscripts/tmp
-KEEP_DIRS = /boot /dev /proc /home \
+KEEP_DIRS = /boot /proc /home \
 	/mnt/cdrom /mnt/floppy \
 	/usr/local/bin /usr/local/sbin /usr/local/share/doc /usr/local/share/man \
 	/var/run
 
 ifeq ($(OS),Linux)
-	KEEP_DIRS += sys
+	KEEP_DIRS += sys /dev
 	NET_LO = net.lo
 endif
 ifneq ($(OS),Linux)
