Index: net-scripts/net/pppd.sh
===================================================================
--- lib/rcscripts/net/pppd.sh	(revision 2437)
+++ lib/rcscripts/net/pppd.sh	(working copy)
@@ -110,8 +110,9 @@
 	opts="linkname ${iface} ${opts}"
 
 	# Setup auth info
-	[[ -n ${!username} ]] && opts="user '"${!username}"' ${opts}"
-	opts="remotename ${iface} ${opts}"
+	if [[ -n ${!username} ]] ; then
+		opts="user '"${!username}"' remotename ${iface} ${opts}"
+	fi
 
 	# Load a custom interface configuration file if it exists
 	[[ -f "/etc/ppp/options.${iface}" ]] \
Index: sbin/functions.sh
===================================================================
--- sbin/functions.sh	(revision 2437)
+++ sbin/functions.sh	(working copy)
@@ -712,11 +712,8 @@
 	shift
 
 	for x in "$@" ; do
-		if [[ -d ${x} ]] ; then
-			is_older_than "${ref}" "${x}"/* && return 0
-		elif [[ ${x} -nt ${ref} ]] ; then
-			return 0
-		fi
+		[[ ${x} -nt ${ref} ]] && return 0
+		[[ -d ${x} ]] && is_older_than "${ref}" "${x}"/* && return 0
 	done
 
 	return 1
Index: sbin/rc
===================================================================
--- sbin/rc	(revision 2437)
+++ sbin/rc	(working copy)
@@ -832,7 +832,7 @@
 
 if [[ ${SOFTLEVEL} == "single" ]] ; then
 	rm -rf "${svcdir}/softscripts.new"
-	/sbin/sulogin ${CONSOLE}
+	[[ ${RUNLEVEL} == "S" ]] && /sbin/sulogin ${CONSOLE}
 	exit 0
 fi
 
Index: sbin/depscan.sh
===================================================================
--- sbin/depscan.sh	(revision 2437)
+++ sbin/depscan.sh	(working copy)
@@ -89,7 +89,7 @@
 	fi
 
 	touch "${mtime_test}"
-	for config in /etc/conf.d/* /etc/init.d/* /etc/rc.conf
+	for config in /etc/conf.d /etc/init.d /etc/rc.conf
 	do
 		! ${update} \
 			&& is_older_than "${mysvcdir}/depcache" "${config}" \
