diff -u net.modules.d/iwconfig /local/rc-scripts-stable/net-scripts/net.modules.d/iwconfig
--- net.modules.d/iwconfig	2005-04-08 19:16:13.000000000 +0100
+++ lib/rcscripts/net.modules.d/iwconfig	2005-04-26 22:41:40.000000000 +0100
@@ -442,6 +442,10 @@
 	x=${x:-auto}
 	[[ ${mode} != ${x} ]] && iwconfig ${iface} mode ${x} &>/dev/null
 
+	# HUP the interface as some drivers/cards need it
+	interface_down ${iface}
+	interface_up ${iface}			
+
 	if [[ -z ${scan} ]]; then
 		ewarn "${iface} does not support scanning"
 		eval x=\"\$\{adhoc_essid_${ifvar}\}\"
diff -u net.modules.d/wpa_supplicant /local/rc-scripts-stable/net-scripts/net.modules.d/wpa_supplicant
--- net.modules.d/wpa_supplicant	2005-04-14 18:52:31.000000000 +0100
+++ lib/rcscripts/net.modules.d/wpa_supplicant	2005-04-26 22:41:40.000000000 +0100
@@ -102,7 +102,7 @@
 	local -a status=( "$( wpa_cli -i${1} status | awk -F= '/^key_mgmt|^wpa_state|^EAP state/ { print "\""$2"\"" }' )" )
 
 	case ${status[0]} in
-		"NONE"	)		[[ ${status[1]} == "ASSOCIATED" ]] ;;
+		"NONE"	)		[[ ${status[1]} == "ASSOCIATED" || ${status[1]} == "COMPLETED" ]] ;;
 		"IEEE 802.1X (no WPA)")	[[ ${status[2]} == "SUCCESS" ]] ;;
 		*)			[[ ${status[1]} == "COMPLETED" ]] ;;
 	esac
