Index: init.d/bootmisc
===================================================================
--- init.d/bootmisc	(revision 2180)
+++ init.d/bootmisc	(working copy)
@@ -111,8 +111,11 @@
 
 		(
 			# Make sure our X11 stuff have the correct permissions
+			# Omit the chown as bootmisc is run before network is up
+			# and users may be using lame LDAP auth #139411
+			rm -rf /tmp/.{ICE,X11}-unix
 			mkdir -p /tmp/.{ICE,X11}-unix
-			chown 0:0 /tmp/.{ICE,X11}-unix
+			#chown 0:0 /tmp/.{ICE,X11}-unix
 			chmod 1777 /tmp/.{ICE,X11}-unix
 			[[ -x /sbin/restorecon ]] && restorecon /tmp/.{ICE,X11}-unix
 		) &> /dev/null
Index: init.d/clock
===================================================================
--- init.d/clock	(revision 2180)
+++ init.d/clock	(working copy)
@@ -91,12 +91,14 @@
 		else
 			ret=0
 		fi
-		errstr="Failed to set system clock to hardware clock"
+		errstr="Failed to set clock"
 	else
 		ret=1
 		errstr="/sbin/hwclock not found"
 	fi
-	eend ${ret} "${errstr}"
+	eend ${ret} "${errstr}" "You will need to set the clock yourself"
+
+	return 0
 }
 
 stop() {
