Index: sbin/runscript.sh
===================================================================
--- sbin/runscript.sh	(revision 1834)
+++ sbin/runscript.sh	(working copy)
@@ -16,6 +16,12 @@
 	exit 1
 fi
 
+# Stop init scripts from working until sysinit completes
+if [[ -e /dev/.rcsysinit ]] ; then
+    eerror "ERROR:  cannot run ${myservice} until sysinit completes"
+    exit 1
+fi
+		
 # State variables
 svcpause="no"
 svcrestart="no"
Index: sbin/rc
===================================================================
--- sbin/rc	(revision 1834)
+++ sbin/rc	(working copy)
@@ -86,7 +86,7 @@
 			CRITICAL_SERVICES="${CRITICAL_SERVICES} ${x##*/}"
 		done
 	else
-		CRITICAL_SERVICES="checkroot hostname modules checkfs localmount clock"
+		CRITICAL_SERVICES="checkroot hostname modules checkfs localmount clock bootmisc"
 	fi
 
 	export CRITICAL_SERVICES
@@ -450,6 +450,9 @@
 		ewarn "Skipping /var/run/utmp initialization (ro root?)"
 	fi
 
+	# sysinit is now done, so allow init scripts to run normally
+	[[ -e /dev/.rcsysinit ]] && rm -f /dev/.rcsysinit
+
 	# All done logging
 	bootlog quit
 
