Index: lib/rcscripts/net/dhcpcd.sh
===================================================================
--- lib/rcscripts/net/dhcpcd.sh	(revision 2184)
+++ lib/rcscripts/net/dhcpcd.sh	(working copy)
@@ -23,15 +23,7 @@
 #
 # Returns 1 if dhcpcd is installed, otherwise 0
 dhcpcd_check_installed() {
-	if [[ -x /sbin/dhcpcd ]]; then
-		if dhcpcd -h 2>&1 | grep -q "etcDir" ; then
-			return 0
-		else
-			${1:-false} && eerror "We require dhcpcd-2.0.0 or newer"
-			return 1
-		fi
-	fi
-
+	[[ -x /sbin/dhcpcd ]] && return 0
 	${1:-false} && eerror "For DHCP (dhcpcd) support, emerge net-misc/dhcpcd"
 	return 1
 }
@@ -93,9 +85,6 @@
 			&& opts="-h \"${hname}\" ${opts}"
 	fi
 
-	# Stop dhcpcd from bringing the interface down when we exit
-	opts="${opts} -o"
-
 	# Add our route metric
 	metric="metric_${ifvar}"
 	[[ -n ${!metric} && ${!metric} != "0" ]] && opts="${opts} -m ${!metric}"
Index: lib/rcscripts/net/dhclient.sh
===================================================================
--- lib/rcscripts/net/dhclient.sh	(revision 2184)
+++ lib/rcscripts/net/dhclient.sh	(working copy)
@@ -106,7 +106,7 @@
 	# Bring up DHCP for this interface (or alias)
 	ebegin "Running dhclient"
 	echo -e "${dhconf}" | start-stop-daemon --start --exec /sbin/dhclient \
-		--pidfile "${pidfile}" -- ${opts} -q -1 -pf "${pidfile}"
+		--pidfile "${pidfile}" -- ${opts} -q -1 -pf "${pidfile}" "${iface}"
 	eend $? || return 1 
 
 	# DHCP succeeded, show address retrieved
