diff --git a/work/mpdscribble-0.19/Makefile.am b/work/mpdscribble-0.19/Makefile.am
index 626fde8..40862cb 100644
--- a/work/mpdscribble-0.19/Makefile.am
+++ b/work/mpdscribble-0.19/Makefile.am
@@ -61,12 +61,12 @@ install-dist_sysconfDATA:
 	test -z "$(sysconfdir)" || $(mkdir_p) "$(DESTDIR)$(sysconfdir)"
 	@list='$(dist_sysconf_DATA)'; \
 	for p in $$list; do \
-	    dest=$(am__strip_dir) \
-	    if test -f $(DESTDIR)$(sysconfdir)/$$dest; then \
-	        echo "$@ will not overwrite existing $(DESTDIR)$(sysconfdir)/$$dest"; \
+	    $(am__strip_dir) \
+	    if test -f $(DESTDIR)$(sysconfdir)/$$f; then \
+	        echo "$@ will not overwrite existing $(DESTDIR)$(sysconfdir)/$$f"; \
 	    else \
 	        if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	        echo " $(INSTALL_SECURE) $$d$$p $(DESTDIR)$(sysconfdir)/$$dest"; \
-	        $(INSTALL_SECURE) $$d$$p $(DESTDIR)$(sysconfdir)/$$dest; \
+	        echo " $(INSTALL_SECURE) $$d$$p $(DESTDIR)$(sysconfdir)/$$f"; \
+	        $(INSTALL_SECURE) $$d$$p $(DESTDIR)$(sysconfdir)/$$f; \
 	    fi; \
 	done
