diff -NrU5 gmfsk-0.6.original/src/mfsk/mfskrx.c gmfsk-0.6/src/mfsk/mfskrx.c
--- gmfsk-0.6.original/src/mfsk/mfskrx.c	2009-07-20 15:49:08.000000000 +0000
+++ gmfsk-0.6/src/mfsk/mfskrx.c	2009-07-20 15:56:08.000000000 +0000
@@ -54,11 +54,11 @@
 
 static void recvpic(struct trx *trx, complex z)
 {
 	struct mfsk *m = (struct mfsk *) trx->modem;
 
-	m->picf += carg(ccor(m->prevz, z)) * SampleRate / (2.0 * M_PI);
+	m->picf += c_arg(ccor(m->prevz, z)) * SampleRate / (2.0 * M_PI);
 	m->prevz = z;
 
 	if ((m->counter % SAMPLES_PER_PIXEL) == 0) {
 		m->picf = 256 * (m->picf / SAMPLES_PER_PIXEL - 1000) / trx->bandwidth;
 
@@ -309,11 +309,11 @@
 
 	if (m->currsymbol != m->prev1symbol)
 		return;
 
 	z = ccor(m->prev1vector, m->currvector);
-	x = carg(z) / m->symlen / (2.0 * M_PI / SampleRate);
+	x = c_arg(z) / m->symlen / (2.0 * M_PI / SampleRate);
 
 	if (x > -m->tonespacing / 2.0 &&  x < m->tonespacing / 2.0)
 		trx_set_freq(trx->frequency + (x / 8.0));
 }
 
diff -NrU5 gmfsk-0.6.original/src/misc/cmplx.c gmfsk-0.6/src/misc/cmplx.c
--- gmfsk-0.6.original/src/misc/cmplx.c	2009-07-20 15:49:08.000000000 +0000
+++ gmfsk-0.6/src/misc/cmplx.c	2009-07-20 15:55:23.000000000 +0000
@@ -133,19 +133,19 @@
 }
 
 /*
  * Argument of a complex number.
  */
-double carg(complex x)
+double c_arg(complex x)
 {
 	return atan2(c_im(x), c_re(x));
 }
 
 /*
  * Complex square root.
  */
-complex csqrt(complex x)
+complex c_sqrt(complex x)
 {
 	complex z;
 
 	c_re(z) = sqrt(cmod(x) + c_re(x)) / M_SQRT2;
 	c_im(z) = c_im(x) / c_re(z) / 2;
diff -NrU5 gmfsk-0.6.original/src/misc/cmplx.h gmfsk-0.6/src/misc/cmplx.h
--- gmfsk-0.6.original/src/misc/cmplx.h	2009-07-20 15:49:08.000000000 +0000
+++ gmfsk-0.6/src/misc/cmplx.h	2009-07-20 15:54:30.000000000 +0000
@@ -147,19 +147,19 @@
 }
 
 /*
  * Argument of a complex number.
  */
-extern __inline__ double carg(complex x)
+extern __inline__ double c_arg(complex x)
 {
 	return atan2(c_im(x), c_re(x));
 }
 
 /*
  * Complex square root.
  */
-extern __inline__ complex csqrt(complex x)
+extern __inline__ complex c_sqrt(complex x)
 {
 	complex z;
 
 	c_re(z) = sqrt(cmod(x) + c_re(x)) / M_SQRT2;
 	c_im(z) = c_im(x) / c_re(z) / 2;
diff -NrU5 gmfsk-0.6.original/src/misc/misc.c gmfsk-0.6/src/misc/misc.c
--- gmfsk-0.6.original/src/misc/misc.c	2009-07-20 15:49:08.000000000 +0000
+++ gmfsk-0.6/src/misc/misc.c	2009-07-20 16:17:35.000000000 +0000
@@ -103,11 +103,11 @@
 /* ---------------------------------------------------------------------- */
 
 /*
  * Integer base-2 logarithm
  */
-int log2(unsigned int x)
+int log_2(unsigned int x)
 {
 	int y = 0;
 
 	x >>= 1;
 
diff -NrU5 gmfsk-0.6.original/src/misc/misc.h gmfsk-0.6/src/misc/misc.h
--- gmfsk-0.6.original/src/misc/misc.h	2009-07-20 15:49:08.000000000 +0000
+++ gmfsk-0.6/src/misc/misc.h	2009-07-20 16:17:21.000000000 +0000
@@ -104,11 +104,11 @@
 /* ---------------------------------------------------------------------- */
 
 /*
  * Integer base-2 logarithm
  */
-extern inline int log2(unsigned int x)
+extern inline int log_2(unsigned int x)
 {
 	int y = 0;
 
 	x >>= 1;
 
diff -NrU5 gmfsk-0.6.original/src/psk31/psk31rx.c gmfsk-0.6/src/psk31/psk31rx.c
--- gmfsk-0.6.original/src/psk31/psk31rx.c	2009-07-20 15:49:08.000000000 +0000
+++ gmfsk-0.6/src/psk31/psk31rx.c	2009-07-20 15:56:42.000000000 +0000
@@ -76,11 +76,11 @@
 {
 	struct psk31 *s = (struct psk31 *) trx->modem;
 	double phase, error;
 	int bits, n;
 
-	if ((phase = carg(ccor(s->prevsymbol, symbol))) < 0)
+	if ((phase = c_arg(ccor(s->prevsymbol, symbol))) < 0)
 		phase += 2 * M_PI;
 
 	if (s->qpsk) {
 		bits = ((int) (phase / M_PI_2 + 0.5)) & 3;
 		n = 4;
diff -NrU5 gmfsk-0.6.original/src/rtty/rttyrx.c gmfsk-0.6/src/rtty/rttyrx.c
--- gmfsk-0.6.original/src/rtty/rttyrx.c	2009-07-20 15:49:08.000000000 +0000
+++ gmfsk-0.6/src/rtty/rttyrx.c	2009-07-20 15:55:38.000000000 +0000
@@ -209,11 +209,11 @@
 		n = fftfilt_run(s->fftfilt, z, &zp);
 
 		for (i = 0; i < n; i++) {
 			static complex prev;
 
-			f = carg(ccor(prev, zp[i])) * SampleRate / (2 * M_PI);
+			f = c_arg(ccor(prev, zp[i])) * SampleRate / (2 * M_PI);
 			prev = zp[i];
 
 			f = bbfilt(s, f);
 			s->pipe[s->pipeptr] = f;
 			s->pipeptr = (s->pipeptr + 1) % s->symbollen;
diff -NrU5 gmfsk-0.6.original/src/throb/throbrx.c gmfsk-0.6/src/throb/throbrx.c
--- gmfsk-0.6.original/src/throb/throbrx.c	2009-07-20 15:49:08.000000000 +0000
+++ gmfsk-0.6/src/throb/throbrx.c	2009-07-20 15:56:26.000000000 +0000
@@ -161,11 +161,11 @@
 		double f;
 
 		z1 = rxword[maxtone];
 		z2 = cmac(s->rxtone[maxtone], s->symbol, s->symptr + 2, s->rxsymlen);
 
-		f = carg(ccor(z1, z2)) / (2 * DownSample * M_PI / SampleRate);
+		f = c_arg(ccor(z1, z2)) / (2 * DownSample * M_PI / SampleRate);
 		f -= s->freqs[maxtone];
 
 		trx_set_freq(trx->frequency + f / 8.0);
 	}
 
