Fixes bug #222209.
--- moreutils.orig/ifdata.docbook	2008-05-28 19:23:58.000000000 +0200
+++ moreutils/ifdata.docbook	2008-05-28 19:24:21.000000000 +0200
@@ -21,7 +21,7 @@
 -->
 
 <!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.4//EN"
-"file:///usr/share/xml/docbook/schema/dtd/4.4/docbookx.dtd"
+"file:///usr/share/sgml/docbook/xml-dtd-4.4/docbookx.dtd"
 []>
 
 <refentry>
diff -ur moreutils.orig/ifne.docbook moreutils/ifne.docbook
--- moreutils.orig/ifne.docbook	2008-05-28 19:23:58.000000000 +0200
+++ moreutils/ifne.docbook	2008-05-28 19:24:21.000000000 +0200
@@ -21,7 +21,7 @@
 -->
 
 <!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.4//EN"
-"file:///usr/share/xml/docbook/schema/dtd/4.4/docbookx.dtd"
+"file:///usr/share/sgml/docbook/xml-dtd-4.4/docbookx.dtd"
 []>
 
 <refentry>
diff -ur moreutils.orig/isutf8.docbook moreutils/isutf8.docbook
--- moreutils.orig/isutf8.docbook	2008-05-28 19:23:58.000000000 +0200
+++ moreutils/isutf8.docbook	2008-05-28 19:24:21.000000000 +0200
@@ -21,7 +21,7 @@
 -->
 
 <!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.4//EN"
-"file:///usr/share/xml/docbook/schema/dtd/4.4/docbookx.dtd"
+"file:///usr/share/sgml/docbook/xml-dtd-4.4/docbookx.dtd"
 []>
 
 <refentry>
diff -ur moreutils.orig/lckdo.docbook moreutils/lckdo.docbook
--- moreutils.orig/lckdo.docbook	2008-05-28 19:23:58.000000000 +0200
+++ moreutils/lckdo.docbook	2008-05-28 19:24:21.000000000 +0200
@@ -8,7 +8,7 @@
 -->
 
 <!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.4//EN"
-"file:///usr/share/xml/docbook/schema/dtd/4.4/docbookx.dtd"
+"file:///usr/share/sgml/docbook/xml-dtd-4.4/docbookx.dtd"
 []>
 
 <refentry>
diff -ur moreutils.orig/mispipe.docbook moreutils/mispipe.docbook
--- moreutils.orig/mispipe.docbook	2008-05-28 19:23:58.000000000 +0200
+++ moreutils/mispipe.docbook	2008-05-28 19:24:21.000000000 +0200
@@ -21,7 +21,7 @@
 -->
 
 <!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.4//EN"
-"file:///usr/share/xml/docbook/schema/dtd/4.4/docbookx.dtd"
+"file:///usr/share/sgml/docbook/xml-dtd-4.4/docbookx.dtd"
 []>
 
 <refentry>
diff -ur moreutils.orig/pee.docbook moreutils/pee.docbook
--- moreutils.orig/pee.docbook	2008-05-28 19:23:58.000000000 +0200
+++ moreutils/pee.docbook	2008-05-28 19:24:21.000000000 +0200
@@ -21,7 +21,7 @@
 -->
 
 <!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.4//EN"
-"file:///usr/share/xml/docbook/schema/dtd/4.4/docbookx.dtd"
+"file:///usr/share/sgml/docbook/xml-dtd-4.4/docbookx.dtd"
 []>
 
 <refentry>
diff -ur moreutils.orig/sponge.docbook moreutils/sponge.docbook
--- moreutils.orig/sponge.docbook	2008-05-28 19:23:58.000000000 +0200
+++ moreutils/sponge.docbook	2008-05-28 19:24:21.000000000 +0200
@@ -21,7 +21,7 @@
 -->
 
 <!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook V4.4//EN"
-"file:///usr/share/xml/docbook/schema/dtd/4.4/docbookx.dtd"
+"file:///usr/share/sgml/docbook/xml-dtd-4.4/docbookx.dtd"
 []>
 
 <refentry>
