diff -uar ies4linux-2.0.5-orig/ies4linux ies4linux-2.0.5/ies4linux
--- ies4linux-2.0.5-orig/ies4linux	2006-10-15 01:46:12.000000000 +0200
+++ ies4linux-2.0.5/ies4linux	2007-06-20 23:58:55.000000000 +0200
@@ -7,10 +7,17 @@
 # Released under the GNU GPL. See LICENSE for more information
 
 # Discover ies4linux installation folder
-IES4LINUX=`dirname "$0"`
+IES4LINUX_RELPATH="$(test -L $0 && echo $(readlink $0) || echo $0)"
+IES4LINUX="$(dirname $IES4LINUX_RELPATH)"
 cd "$IES4LINUX"
 export IES4LINUX=`pwd`
 
+# Changes to the user's home directory, in case "$IES4LINUX" is not writable,
+# because executing `wine --version` (executed, notably, in "lib/messages.sh"),
+# while in a directory where the user does not have write permission,
+# leads to (unimportant?) warnings.
+cd "$HOME"
+
 # Create log file
 # mkdir -p "$HOME"/.ies4linux
 # touch "$HOME"/.ies4linux/log
@@ -41,9 +48,9 @@
 INSTALLIE2=0
 
 # Basic imports
-source lib/functions.sh
-source lib/terminal.sh
-source lib/messages.sh
+source "$IES4LINUX"/lib/functions.sh
+source "$IES4LINUX"/lib/terminal.sh
+source "$IES4LINUX"/lib/messages.sh
 
 # Get some command-line options.
 # If an option is specified on the command line, the user won't be asked for it later on.
diff -uar ies4linux-2.0.5-orig/lib/functions.sh ies4linux-2.0.5/lib/functions.sh
--- ies4linux-2.0.5-orig/lib/functions.sh	2006-10-15 08:08:50.000000000 +0200
+++ ies4linux-2.0.5/lib/functions.sh	2007-06-21 00:05:49.000000000 +0200
@@ -66,12 +66,12 @@
 		tmp="$tmp \"$(eval echo \${$num})\""
 		num=$((num+1))
 	done
-	eval $tmp &> "$IES4LINUX"/cabextract.log || {
-		cat "$IES4LINUX"/cabextract.log
-		rm "$IES4LINUX"/cabextract.log
+	eval $tmp &> "$BASEDIR"/tmp/cabextract.log || {
+		cat "$BASEDIR"/tmp/cabextract.log
+		rm "$BASEDIR"/tmp/cabextract.log
 		error $MSG_ERROR_CABEXTRACTING
 	}
-	rm "$IES4LINUX"/cabextract.log
+	rm "$BASEDIR"/tmp/cabextract.log
 }
 
 # Post install
@@ -93,4 +93,5 @@
 		local l=$BINDIR/ie$1
 		echo " ${l//\/\//\/}"
 	fi
-}
\ Brak znaku nowej linii na koñcu pliku
+}
+
diff -uar ies4linux-2.0.5-orig/lib/install.sh ies4linux-2.0.5/lib/install.sh
--- ies4linux-2.0.5-orig/lib/install.sh	2006-10-15 08:13:18.000000000 +0200
+++ ies4linux-2.0.5/lib/install.sh	2007-06-21 00:03:29.000000000 +0200
@@ -91,7 +91,7 @@
 
 	subsection $MSG_INSTALLING_REGISTRY
 		add_registry "$IES4LINUX"/winereg/ie6.reg
-		add_registry "$IES4LINUX"/winereg/homepage.reg
+		add_registry "$BASEDIR"/tmp/homepage.reg
 	
 	subsection $MSG_FINALIZING
 		wineboot &> /dev/null
@@ -239,3 +239,4 @@
 	
 	ok
 }
+
diff -uar ies4linux-2.0.5-orig/lib/messages.sh ies4linux-2.0.5/lib/messages.sh
--- ies4linux-2.0.5-orig/lib/messages.sh	2006-10-15 01:56:33.000000000 +0200
+++ ies4linux-2.0.5/lib/messages.sh	2007-06-21 00:04:31.000000000 +0200
@@ -101,7 +101,7 @@
     source "$msg_file"
 
     # Try to convert to user system encoding
-    iconv_tmpfile="$IES4LINUX"/lang/lang.sh
+    iconv_tmpfile="$BASEDIR"/tmp/lang.sh
     if iconv --version &> /dev/null; then
         iconv -f $TRANSLATION_ENCODING "$msg_file" > "$iconv_tmpfile" && source "$iconv_tmpfile"
 	rm -f "$iconv_tmpfile"
@@ -115,7 +115,7 @@
 exec 0<&6
 
 # generate homepage registry file
-cat <<FIM > "$IES4LINUX"/winereg/homepage.reg
+cat <<FIM > "$BASEDIR"/tmp/homepage.reg
 [HKEY_CURRENT_USER\Software\Microsoft\Internet Explorer\Main]
 "Start Page"="${OK_URL}"
 FIM
