diff -u -r ies4linux-2.5beta6-orig/ies4linux ies4linux-2.5beta6/ies4linux
--- ies4linux-2.5beta6-orig/ies4linux	2007-02-22 10:04:16.000000000 +0100
+++ ies4linux-2.5beta6/ies4linux	2007-06-22 01:38:10.000000000 +0200
@@ -7,7 +7,8 @@
 # Released under the GNU GPL. See LICENSE for more information
 
 # Discover ies4linux installation folder
-IES4LINUX=`dirname "$0"`
+IES4LINUX_RELPATH="$(test -L $0 && echo $(readlink $0) || echo $0)"
+IES4LINUX="$(dirname $IES4LINUX_RELPATH)"
 cd "$IES4LINUX"
 export IES4LINUX=`pwd`
 
