--- Libraries/MiKTeX/Core/texmfroot.cpp	2006-07-25 16:45:02.000000000 +1000
+++ Libraries/MiKTeX/Core/texmfroot.cpp	2007-03-10 14:21:51.000000000 +1000
@@ -229,11 +229,13 @@
 	  UNEXPECTED_CONDITION
 	    (T_("SessionImpl::InitializeRootDirectories"));
 	}
+	/* AVOIDING TEST FOR MULTIPLE ROOT DIRECTORIES IN $TEXMF AS IT FAILS ON GENTOO
       if (find(vec.begin(), vec.end(), root.GetCurrent()) != vec.end())
 	{
 	  UNEXPECTED_CONDITION
 	    (T_("SessionImpl::InitializeRootDirectories"));
 	}
+	*/
       if (startupConfig.commonConfigRoot == root.GetCurrent())
 	{
 	  if ((GetPolicyFlags() & PolicyFlags::DataRootHighestPriority) != 0)
