Index: ptabtools-0.3/Makefile
===================================================================
--- ptabtools-0.3.orig/Makefile
+++ ptabtools-0.3/Makefile
@@ -28,7 +28,7 @@ ptb2xml.o: ptb2xml.c
 	$(CC) -DPTB_VERSION=\"$(PTB_VERSION)\" $(CFLAGS) -fPIC -o $@ -c $< `pkg-config --cflags glib-2.0`
 
 libptb-$(PTB_VERSION).so: $(PTBSO_OBJS)
-	$(CC) -shared -fPIC $(LDFLAGS) -o $@ $(PTBSO_OBJS) `pkg-config --libs glib-2.0`
+	$(CC) -shared -fPIC $(LDFLAGS) -Wl,-soname -Wl,$@ -o $@ $(PTBSO_OBJS) `pkg-config --libs glib-2.0`
 
 ptb2xml: $(PTB2XML_OBJS)
 	$(CC) $(LDFLAGS) -o $@ $(PTB2XML_OBJS) `pkg-config --libs glib-2.0 libxml-2.0 libxslt` -lpopt
