Index: ptabtools-0.3/Makefile
===================================================================
--- ptabtools-0.3.orig/Makefile
+++ ptabtools-0.3/Makefile
@@ -9,7 +9,6 @@ PROGS = ptb2ly ptb2ascii ptbinfo $(shell
 LIBS = libptb-$(PTB_VERSION).so 
 PROGS_MANPAGES = $(patsubst %,%.1,$(PROGS))
 INSTALL = install
-CFLAGS = -g -Wall -DPTB_VERSION=\"$(PTB_VERSION)\" 
 
 PTB2LY_OBJS = ptb2ly.o ptb.o
 PTBINFO_OBJS = ptbinfo.o ptb.o
@@ -20,28 +19,28 @@ PTBSO_OBJS = ptb.lo
 all: $(PROGS) $(LIBS)
 
 ptb2xml.o: ptb2xml.c
-	$(CC) $(CFLAGS) -c $< `pkg-config --cflags glib-2.0 libxml-2.0 libxslt`
+	$(CC) -DPTB_VERSION=\"$(PTB_VERSION)\" $(CFLAGS) -c $< `pkg-config --cflags glib-2.0 libxml-2.0 libxslt`
 
 %.o: %.c
-	$(CC) $(CFLAGS) -c $< `pkg-config --cflags glib-2.0`
+	$(CC) -DPTB_VERSION=\"$(PTB_VERSION)\" $(CFLAGS) -c $< `pkg-config --cflags glib-2.0`
 
 %.lo: %.c
-	$(CC) $(CFLAGS) -fPIC -o $@ -c $< `pkg-config --cflags glib-2.0`
+	$(CC) -DPTB_VERSION=\"$(PTB_VERSION)\" $(CFLAGS) -fPIC -o $@ -c $< `pkg-config --cflags glib-2.0`
 
 libptb-$(PTB_VERSION).so: $(PTBSO_OBJS)
-	$(CC) -shared -fPIC $(CFLAGS) -o $@ $(PTBSO_OBJS) `pkg-config --libs glib-2.0`
+	$(CC) -shared -fPIC $(LDFLAGS) -o $@ $(PTBSO_OBJS) `pkg-config --libs glib-2.0`
 
 ptb2xml: $(PTB2XML_OBJS)
-	$(CC) $(CFLAGS) -o $@ $(PTB2XML_OBJS) `pkg-config --libs glib-2.0 libxml-2.0 libxslt` -lpopt
+	$(CC) $(LDFLAGS) -o $@ $(PTB2XML_OBJS) `pkg-config --libs glib-2.0 libxml-2.0 libxslt` -lpopt
 	
 ptb2ascii: $(PTB2ASCII_OBJS)
-	$(CC) $(CFLAGS) -o $@ $(PTB2ASCII_OBJS) `pkg-config --libs glib-2.0` -lpopt
+	$(CC) $(LDFLAGS) -o $@ $(PTB2ASCII_OBJS) `pkg-config --libs glib-2.0` -lpopt
 
 ptb2ly: $(PTB2LY_OBJS)
-	$(CC) $(CFLAGS) -o $@ $(PTB2LY_OBJS) `pkg-config --libs glib-2.0` -lpopt
+	$(CC) $(LDFLAGS) -o $@ $(PTB2LY_OBJS) `pkg-config --libs glib-2.0` -lpopt
 
 ptbinfo: $(PTBINFO_OBJS)
-	$(CC) $(CFLAGS) -o $@ $(PTBINFO_OBJS) `pkg-config --libs glib-2.0` -lpopt
+	$(CC) $(LDFLAGS) -o $@ $(PTBINFO_OBJS) `pkg-config --libs glib-2.0` -lpopt
 
 install: all
 	$(INSTALL) $(PROGS) $(DESTDIR)$(bindir)
