Index: dvbcut-0.5.4/src/buffer.cpp
===================================================================
--- dvbcut-0.5.4.orig/src/buffer.cpp
+++ dvbcut-0.5.4/src/buffer.cpp
@@ -392,7 +392,7 @@ outbuffer::~outbuffer()
 int outbuffer::open(const char* filename)
   {
   close=true;
-  return fd=::open(filename,O_WRONLY|O_CREAT|O_TRUNC|O_BINARY);
+  return fd=::open(filename,O_WRONLY|O_CREAT|O_TRUNC|O_BINARY,0644);
   }
 
 int outbuffer::putdata(const void *data, unsigned int len, bool autoresize)
