--- src/modules/linux/linux-urandom.c	2008-02-25 22:13:55.000000000 +0000
+++ src/modules/linux/linux-urandom.c	2009-03-10 22:54:50.000000000 +0000
@@ -59,7 +59,7 @@
 
 	int from_fd = open(from, O_RDONLY);
 	if (from_fd) {
-		int to_fd = open(to, O_WRONLY | O_CREAT);
+		int to_fd = open(to, O_WRONLY | O_CREAT, 0644);
 		if (to_fd) {
 			char buffer[s];
                         memset (buffer, 0, s);
