--- udev-103/extras/rule_generator/75-persistent-net-generator.rules.orig	2007-01-18 17:42:00.000000000 -0800
+++ udev-103/extras/rule_generator/75-persistent-net-generator.rules	2007-01-18 17:42:07.000000000 -0800
@@ -1,6 +1,7 @@
 # these rules generate rules for persistent network device naming
 
-ACTION=="add", SUBSYSTEM=="net", KERNEL=="eth*|ath*|wlan*|ra*|sta*" \
+ACTION=="add", SUBSYSTEM=="net", SUBSYSTEMS!="xen", \
+	KERNEL=="eth*|ath*|wlan*|ra*|sta*", \
 	NAME!="?*", DRIVERS=="?*", GOTO="persistent_net_generator_do"
 
 GOTO="persistent_net_generator_end"
@@ -10,7 +11,6 @@
 SUBSYSTEMS=="pci", ENV{COMMENT}="PCI device $attr{vendor}:$attr{device} ($attr{driver})"
 SUBSYSTEMS=="usb", ENV{COMMENT}="USB device 0x$attr{idVendor}:0x$attr{idProduct} ($attr{driver})"
 SUBSYSTEMS=="ieee1394", ENV{COMMENT}="Firewire device $attr{host_id})"
-SUBSYSTEMS=="xen", ENV{COMMENT}="Xen virtual device"
 ENV{COMMENT}=="", ENV{COMMENT}="$env{SUBSYSTEM} device ($attr{driver})"
 
 IMPORT{program}="write_net_rules $attr{address}"
