diff -r -ru udev-108-orig/etc/udev/gentoo/05-udev-early.rules udev-108/etc/udev/gentoo/05-udev-early.rules
--- udev-108-orig/etc/udev/gentoo/05-udev-early.rules	2007-04-11 09:32:13.007423206 +0200
+++ udev-108/etc/udev/gentoo/05-udev-early.rules	2007-04-11 09:32:30.587306470 +0200
@@ -7,7 +7,3 @@
 ACTION=="add", KERNEL=="[0-9]*:[0-9]*", SUBSYSTEM=="scsi", WAIT_FOR_SYSFS="ioerr_cnt"
 ACTION=="add", SUBSYSTEM=="net", WAIT_FOR_SYSFS="address"
 ACTION=="add", SUBSYSTEM=="net", ENV{PHYSDEVDRIVER}=="?*", WAIT_FOR_SYSFS="device/driver"
-
-# Set IN_HOTPLUG for compatibility with init-scripts called from udev
-ENV{IN_HOTPLUG}="1"
-
