diff -Naur info2html-1.4/README info2html/README
--- info2html-1.4/README	2001-11-21 19:31:28.000000000 +0000
+++ info2html/README	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-For the most recent version, documentation and installation
-instructions, see info2html's home page:
-http://info2html.sourceforge.net/
-
-info2html's original home page (now very out-of-date):
-http://www.iam.unibe.ch/work/docs/info2html/
diff -Naur info2html-1.4/info2html info2html/info2html
--- info2html-1.4/info2html	2001-11-21 19:31:28.000000000 +0000
+++ info2html/info2html	2003-09-28 01:58:16.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 #---------------------------------------------------------
 #                      info2html
 #---------------------------------------------------------
diff -Naur info2html-1.4/info2html.conf info2html/info2html.conf
--- info2html-1.4/info2html.conf	2001-11-21 19:31:28.000000000 +0000
+++ info2html/info2html.conf	2003-09-27 17:47:59.000000000 +0100
@@ -36,13 +36,8 @@
 
 #-- location of info files.
 @INFODIR       = (
-		'/usr/local/GNU/info',
-		'/usr/local/GNU/emacs18.58/info',
-		'/usr/local/info',
-		'/opt/FSFgzip/info',
-		'/opt/FSFgmake/info',
-		'/opt/GCC2721/info',
-		'/usr/local/lib/bash-doc-2.01/lib/readline/doc/'
+		'/usr/share/info',
+		'/usr/local/share/info'
 );
 
 #-- URL for documentation of info2html
diff -Naur info2html-1.4/infocat info2html/infocat
--- info2html-1.4/infocat	2001-11-21 19:31:28.000000000 +0000
+++ info2html/infocat	2003-09-28 01:58:23.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 #---------------------------------------------------------
 #                      infocat
 #---------------------------------------------------------
