--- extern/SConscript
+++ extern/SConscript
@@ -2,8 +2,6 @@
 
 Import('env')
 
-SConscript(['glew/SConscript'])
-
 if env['WITH_BF_BULLET']:
     SConscript(['bullet2/src/SConscript'])
 
--- source/blender/editors/include/BIF_gl.h
+++ source/blender/editors/include/BIF_gl.h
@@ -31,7 +31,7 @@
 #ifndef BIF_GL_H
 #define BIF_GL_H
 
-#include "GL/glew.h"
+#include <GL/glew.h>
 
 	/*
 	 * these should be phased out. cpack should be replaced in
