Index: include/config.h
===================================================================
--- include/config.h	(revision 911)
+++ include/config.h	(working copy)
@@ -48,7 +48,7 @@
  */
 
 /* dirs */
-#define DPATH   IRCD_PREFIX
+#define DPATH   "/var/lib/charybdis"
 #define BINPATH IRCD_PREFIX "/bin/"
 #define MODPATH MODULE_DIR
 #define AUTOMODPATH MODULE_DIR "/autoload/"
@@ -58,18 +58,18 @@
 #define HPATH  HELP_DIR "/opers"
 
 /* files */
-#define SPATH    BINPATH "/ircd"	/* ircd executable */
-#define SLPATH   BINPATH "/servlink"	/* servlink executable */
-#define CPATH    ETCPATH "/ircd.conf"	/* ircd.conf file */
-#define KPATH    ETCPATH "/kline.conf"	/* kline file */
-#define DLPATH   ETCPATH "/dline.conf"	/* dline file */
-#define XPATH	 ETCPATH "/xline.conf"	/* xline file */
-#define RESVPATH ETCPATH "/resv.conf"	/* resv file */
-#define RPATH    ETCPATH "/ircd.rsa"	/* ircd rsa private keyfile */
-#define MPATH    ETCPATH "/ircd.motd"	/* MOTD file */
-#define LPATH    LOGPATH "/ircd.log"	/* ircd logfile */
-#define PPATH    ETCPATH "/ircd.pid"	/* pid file */
-#define OPATH    ETCPATH "/opers.motd"	/* oper MOTD file */
+#define SPATH    BINPATH "/ircd"                   /* ircd executable */
+#define SLPATH   BINPATH "/servlink"               /* servlink executable */
+#define CPATH    ETCPATH "/ircd.conf"              /* ircd.conf file */
+#define KPATH    "/var/lib/charybdis/kline.conf"   /* kline file */
+#define DLPATH   "/var/lib/charybdis/dline.conf"   /* dline file */
+#define XPATH    "/var/lib/charybdis/xline.conf"   /* xline file */
+#define RESVPATH "/var/lib/charybdis/resv.conf"    /* resv file */
+#define RPATH    ETCPATH "/ircd.rsa"               /* ircd rsa private keyfile */
+#define MPATH    ETCPATH "/ircd.motd"              /* MOTD file */
+#define LPATH    LOGPATH "/ircd.log"               /* ircd logfile */
+#define PPATH    "/var/run/charybdis/ircd.pid"     /* pid file */
+#define OPATH    ETCPATH "/opers.motd"             /* oper MOTD file */
 
 /* IGNORE_BOGUS_TS
  * Ignore bogus timestamps from other servers. Yes this will desync
