--- htags/Makefile.am.orig	2010-08-30 14:19:04.000000000 +0200
+++ htags/Makefile.am	2010-09-02 19:00:48.000000000 +0200
@@ -33,7 +33,7 @@
 EXTRA_DIST = $(man_MANS) $(gtags_DATA) manual.in global.cgi.tmpl.in completion.cgi.tmpl.in \
 		const.h c.l cpp.l java.l php.l asm.l
 install-data-hook:
-	dir="$(localstatedir)/gtags/sitekeys"; \
+	dir="$(DESTDIR)$(localstatedir)/gtags/sitekeys"; \
 	if [ ! -d $$dir ]; then \
 		$(MKDIR_P) $$dir; \
 		chmod 755 $$dir; \
