--- Makefile.orig	Thu Jun 21 08:26:05 2001
+++ Makefile	Thu Jun 21 08:27:20 2001
@@ -4,18 +4,21 @@
 
 CC = gcc
 CFLAGS = -O2 -fPIC
-#LDFLAGS = -shared
-LDFLAGS = -G
+LDFLAGS = -shared
+#LDFLAGS = -G
 
 DESTDIR = /usr/lib/security
 MANDIR = /opt/local/man/man5
 
-OSLIBS = -lpam -lnsl -lsocket
-KRB5LIBS = -L/opt/local/lib -lkrb5 -lk5crypto -lcom_err
+#OSLIBS = -lpam -lnsl -lsocket
+OSLIBS = -lpam 
+#KRB5LIBS = -L/opt/local/lib -lkrb5 -lk5crypto -lcom_err
+KRB5LIBS = -L/usr/lib -lkrb5 -lk5crypto -lcom_err
 
 LIBS = $(OSLIBS) $(KRB5LIBS)
 
-INC = -I/opt/local/include
+#INC = -I/opt/local/include
+INC = -I/usr/include
 
 
 ####################################################################
