--- libtool-1.5.2/ltmain.in.orig	2004-11-04 04:20:50.608823576 +0000
+++ libtool-1.5.2/ltmain.in	2004-11-04 04:23:25.151329544 +0000
@@ -560,8 +560,12 @@
 	  if grep "^# ### BEGIN LIBTOOL TAG CONFIG: $z$" < "$0" > /dev/null; then
 	    # Evaluate the configuration.
 	    eval "`${SED} -n -e '/^# ### BEGIN LIBTOOL TAG CONFIG: '$z'$/,/^# ### END LIBTOOL TAG CONFIG: '$z'$/p' < $0`"
+	    # user sometimes does CC=<HOST>-gcc so we need to match that to 'gcc'
+	    trimedcc="${CC/${host}-}"
 	    case "$base_compile " in
-	    "$CC "* | " $CC "* | "`$echo $CC` "* | " `$echo $CC` "*)
+	    "gcc "* | " gcc "* | "${host}-gcc "* | " ${host}-gcc "*) tagname=CC ; break ;;
+	    "$CC "* | " $CC "* | "`$echo $CC` "* | " `$echo $CC` "*|\
+	    "$trimedcc "* | " $trimedcc "* | "`$echo $trimedcc` "* | " `$echo $trimedcc` "*)
 	      # The compiler in the base compile command matches
 	      # the one in the tagged configuration.
 	      # Assume this is the tagged configuration we want.
@@ -947,8 +950,12 @@
 	  if grep "^# ### BEGIN LIBTOOL TAG CONFIG: $z$" < "$0" > /dev/null; then
 	    # Evaluate the configuration.
 	    eval "`${SED} -n -e '/^# ### BEGIN LIBTOOL TAG CONFIG: '$z'$/,/^# ### END LIBTOOL TAG CONFIG: '$z'$/p' < $0`"
+	    # user sometimes does CC=<HOST>-gcc so we need to match that to 'gcc'
+	    trimedcc="${CC/${host}-}"
 	    case $base_compile in
-	    "$CC "* | " $CC "* | "`$echo $CC` "* | " `$echo $CC` "*)
+	    "gcc "* | " gcc "* | "${host}-gcc "* | " ${host}-gcc "*) tagname=CC ; break ;;
+	    "$CC "* | " $CC "* | "`$echo $CC` "* | " `$echo $CC` "*|\
+	    "$trimedcc "* | " $trimedcc "* | "`$echo $trimedcc` "* | " `$echo $trimedcc` "*)
 	      # The compiler in $compile_command matches
 	      # the one in the tagged configuration.
 	      # Assume this is the tagged configuration we want.
