--- NVIDIA-Linux-x86-1.0-5328-pkg1/usr/src/nv/os-interface.c.orig	2003-12-17 17:54:59.000000000 -0500
+++ NVIDIA-Linux-x86-1.0-5328-pkg1/usr/src/nv/os-interface.c	2003-12-25 15:19:29.000000000 -0500
@@ -618,12 +618,28 @@
     return value;
 }
 
+struct pci_dev* nvos_find_agp_by_class(unsigned int);
+
 U032 os_pci_read_dword(
     VOID *handle,
     U008 offset
 ) 
 {
     U032 value;
+    if (!handle) {
+        /*
+         * XXX: This hack works around a bug in the NVIDIA AGP
+         * GART driver, which calls us with an invalid handle
+         * on newer Via chipsets. What we need here instead is
+         * a handle for the AGP graphics card.
+         */
+        handle = nvos_find_agp_by_class(PCI_CLASS_DISPLAY_VGA);
+
+        if (!handle) {
+            printk("NVRM: os_pci_read_dword(): bad handle!\n");
+            return 0;
+        }
+    }
     pci_read_config_dword( (struct pci_dev *) handle, offset, (u32 *) &value);
     return value;
 }
