diff -urN policycoreutils-1.16.orig/scripts/genhomedircon policycoreutils-1.16/scripts/genhomedircon
--- policycoreutils-1.16.orig/scripts/genhomedircon	2004-09-06 02:19:00.927379664 -0400
+++ policycoreutils-1.16/scripts/genhomedircon	2004-09-06 02:28:25.143605760 -0400
@@ -87,7 +87,13 @@
 	return rc
 
 try:
-	if len(sys.argv) != 3:
+	if len(sys.argv) == 2:
+		FILECONTEXTDIR="."
+		FILECONTEXTS=sys.argv[1]
+	elif len(sys.argv) == 3:
+		FILECONTEXTDIR=sys.argv[1]
+		FILECONTEXTS=sys.argv[2]
+	else:
 		print len(sys.argv)
 		usage("Incorrect parameters")
 
@@ -97,13 +103,6 @@
 	else:
 		STARTING_UID=500
 
-	rc=commands.getstatusoutput("grep -h '^UID_MIN' /etc/login.defs | sed -e 's/^UID_MIN[^0-9]*//'")
-	if rc[0] == 0:
-		STARTING_UID=rc[1]
-	else:
-		STARTING_UID=500
-
-	FILECONTEXTDIR=sys.argv[1]
 	prefixes = getPrefixes()
 
 	rc=commands.getstatusoutput("grep -h '^HOME' /etc/default/useradd")
@@ -124,7 +123,7 @@
 	sed_dir = "h; s|^HOME_DIR|%s/[^/]+|; s|ROLE_|user_|" % (string.join(prefixes.keys(), "/[^/]+|; s|ROLE_|user_|; p; g; s|^HOME_DIR|"),)
 
 	# Fill in HOME_ROOT, HOME_DIR, and ROLE for users not explicitly defined in /etc/security/selinux/src/policy/users
-	rc=commands.getstatusoutput("sed -e \"/^HOME_ROOT/{%s}\" -e \"/^HOME_DIR/{%s}\" %s" % (sed_root, sed_dir, sys.argv[2]))
+	rc=commands.getstatusoutput("sed -e \"/^HOME_ROOT/{%s}\" -e \"/^HOME_DIR/{%s}\" %s" % (sed_root, sed_dir, FILECONTEXTS))
 	if rc[0] == 0:
 		print rc[1]
 	else:
@@ -135,7 +134,7 @@
 
 	# Fill in HOME and ROLE for users that are defined
 	for u in users.keys():
-		update(sys.argv[2], u, users[u]) 
+		update(FILECONTEXTS, u, users[u]) 
 except ValueError, error:
 	errorExit(string.join("ValueError ", error))
 except IndexError, error:
