diff -Naur qt4-qtruby-1.4.9-orig/smoke/qt/CMakeLists.txt qt4-qtruby-1.4.9/smoke/qt/CMakeLists.txt
--- qt4-qtruby-1.4.9-orig/smoke/qt/CMakeLists.txt	2007-06-20 07:14:26.000000000 -0400
+++ qt4-qtruby-1.4.9/smoke/qt/CMakeLists.txt	2007-12-30 20:52:13.000000000 -0500
@@ -125,7 +125,7 @@
 
 set_target_properties(smokeqt PROPERTIES VERSION 2.0.0 SOVERSION 2 )
 
-install(TARGETS smokeqt DESTINATION ${CMAKE_INSTALL_PREFIX}/lib )
+install(TARGETS smokeqt DESTINATION ${LIB_INSTALL_DIR} )
 
 configure_file(qtguess.pl.cmake ${CMAKE_CURRENT_BINARY_DIR}/qtguess.pl @ONLY)
 configure_file(generate.pl.cmake ${CMAKE_CURRENT_BINARY_DIR}/generate.pl @ONLY )

