diff -urN interbase_old/builds/original/build_kit interbase/builds/original/build_kit
--- interbase_old/builds/original/build_kit	2001-08-06 18:34:57.000000000 +0300
+++ interbase/builds/original/build_kit	2003-06-10 16:56:46.000000000 +0300
@@ -67,7 +67,7 @@
 
 #
 
-LD_LIBRARY_PATH=$CURDIR/jrd:$CURDIR/interbase/lib
+LD_LIBRARY_PATH=$CURDIR/jrd:$CURDIR/interbase/lib:$INTERBASE/lib
 SHLIB_PATH=$CURDIR/jrd
 
 export LD_LIBRARY_PATH
diff -urN interbase_old/builds/original/prefix.linux interbase/builds/original/prefix.linux
--- interbase_old/builds/original/prefix.linux	2003-05-30 12:56:17.000000000 +0300
+++ interbase/builds/original/prefix.linux	2003-06-10 16:58:09.000000000 +0300
@@ -33,7 +33,7 @@
 PROD_DEBUG_OBJECTS=	nodebug.o
 PROD_SHRLIB_DIR=	-Lsource/jrd -Lsource/interbase/lib
 PROD_VERSION_FLAG=	-DPROD_BUILD
-PROD_CFLAGS=            -O3 -march=i486 -fpic -DFLINTSTONE
+PROD_CFLAGS=           $(GENTOO_CFLAGS) -fpic -DFLINTSTONE
 
 
 DEV_DEBUG_OBJECTS=	grammar.o dbg.o dbt.o dmp.o
diff -urN interbase_old/jrd/val.c interbase/jrd/val.c
--- interbase_old/jrd/val.c.orig	2003-10-29 12:41:56.173403352 +0000
+++ interbase/jrd/val.c	2003-10-29 12:42:19.547849896 +0000
@@ -100,8 +100,8 @@
    If other attachments are already made locally or through the same multi-
    client server, validate gives up with the message:
 
-   "Lock timeout during wait transaction
-   -- Object "database_filename.gdb" is in use"
+   Lock timeout during wait transaction
+   -- Object "database_filename.gdb" is in use
 
    If other processes or servers are attached to the database, validate 
    waits for the exclusive lock on the database (i.e. waits for every 
