diff -Naur Firebird-2.0.4.13130-1_orig/builds/posix/Makefile.in.firebird Firebird-2.0.4.13130-1/builds/posix/Makefile.in.firebird
--- Firebird-2.0.4.13130-1_orig/builds/posix/Makefile.in.firebird	2008-07-23 17:52:13.000000000 -0400
+++ Firebird-2.0.4.13130-1/builds/posix/Makefile.in.firebird	2008-07-23 17:57:05.000000000 -0400
@@ -185,10 +185,10 @@
 # 
 
 external_libraries:
-	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.icu
-ifeq ($(EDITLINE_FLG),Y)
-	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.editline
-endif
+#	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.icu
+#ifeq ($(EDITLINE_FLG),Y)
+#	$(MAKE) -f $(GEN_ROOT)/Makefile.extern.editline
+#endif
 
 #---------------------------------------------------------------------------
 # Phase1: make a gpre_static from scratch so we can parse .e and .epp files
diff -Naur Firebird-2.0.4.13130-1_orig/builds/posix/make.rules Firebird-2.0.4.13130-1/builds/posix/make.rules
--- Firebird-2.0.4.13130-1_orig/builds/posix/make.rules	2008-07-23 17:52:13.000000000 -0400
+++ Firebird-2.0.4.13130-1/builds/posix/make.rules	2008-07-23 17:59:22.000000000 -0400
@@ -33,7 +33,7 @@
 
 
 # Please don't use compiler/platform specific flags here - nmcc 02-Nov-2002
-CFLAGS:= $(CFLAGS) -I$(SRC_ROOT)/include/gen -I$(SRC_ROOT)/include -I$(SRC_ROOT)/vulcan -I$(ROOT)/extern/icu/source/common -I$(ROOT)/extern/icu/source/i18n -DNAMESPACE=Vulcan
+CFLAGS:= $(CFLAGS) -I$(SRC_ROOT)/include/gen -I$(SRC_ROOT)/include -I$(SRC_ROOT)/vulcan -DNAMESPACE=Vulcan
 
 ifeq ($(IsProdTypeBuild),Y)
   CFLAGS:= $(CFLAGS) $(PROD_FLAGS) -DPROD_BUILD
@@ -88,19 +88,19 @@
 
 .SUFFIXES: .c .e .epp .cpp
 
-.e.c:
+.e.c: $(GPRE_CURRENT)
 	$(GPRE_CURRENT) $(GPRE_FLAGS) $< $@
 
 
-.epp.cpp:
+.epp.cpp: $(GPRE_CURRENT)
 	$(GPRE_CURRENT) $(GPRE_FLAGS) $< $@
 
 
-$(SRC_ROOT)/jrd/%.cpp:: $(SRC_ROOT)/jrd/%.epp 
+$(SRC_ROOT)/jrd/%.cpp:: $(SRC_ROOT)/jrd/%.epp $(GPRE_CURRENT)
 	$(GPRE_CURRENT) $(JRD_GPRE_FLAGS) $< $@
 
 
-$(SRC_ROOT)/dsql/%.cpp:: $(SRC_ROOT)/dsql/%.epp 
+$(SRC_ROOT)/dsql/%.cpp:: $(SRC_ROOT)/dsql/%.epp $(GPRE_CURRENT)
 	$(GPRE_CURRENT) $(DSQL_GPRE_FLAGS) $< $@
 
 
diff -Naur Firebird-2.0.4.13130-1_orig/builds/posix/make.shared.targets Firebird-2.0.4.13130-1/builds/posix/make.shared.targets
--- Firebird-2.0.4.13130-1_orig/builds/posix/make.shared.targets	2008-07-23 17:52:13.000000000 -0400
+++ Firebird-2.0.4.13130-1/builds/posix/make.shared.targets	2008-07-23 17:54:26.000000000 -0400
@@ -57,7 +57,7 @@
 
 # gpre_meta needs a special boot build since there is no database.
 
-$(SRC_ROOT)/gpre/gpre_meta.cpp:  $(SRC_ROOT)/gpre/gpre_meta.epp
+$(SRC_ROOT)/gpre/gpre_meta.cpp:  $(SRC_ROOT)/gpre/gpre_meta.epp $(GPRE_BOOT)
 	$(GPRE_BOOT) -lang_internal $(GPRE_FLAGS) $< $@
 
 
diff -Naur Firebird-2.0.4.13130-1_orig/builds/posix/prefix.linux Firebird-2.0.4.13130-1/builds/posix/prefix.linux
--- Firebird-2.0.4.13130-1_orig/builds/posix/prefix.linux	2008-07-23 17:52:13.000000000 -0400
+++ Firebird-2.0.4.13130-1/builds/posix/prefix.linux	2008-07-23 17:53:32.000000000 -0400
@@ -20,7 +20,7 @@
 
 # -fno-builtin is used because GCC 3.0-3.2.2 had bug with builtins expansion
 # you may remove it if engine is getting compiled with any other GCC version
-PROD_FLAGS=-ggdb -O3 -march=i586 -mcpu=i686 -fno-omit-frame-pointer -fno-builtin -DNDEBUG -DLINUX -pipe -MMD -fPIC -fmessage-length=0
+PROD_FLAGS=-ggdb -DNDEBUG -DLINUX -pipe -MMD -fPIC -fmessage-length=0
 DEV_FLAGS=-DUSE_VALGRIND -ggdb -DLINUX -pipe -MMD -p -fPIC -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0
 #DEV_FLAGS=-ggdb -DLINUX -pipe -MMD -p -fPIC -Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -fmessage-length=0
 
diff -Naur Firebird-2.0.4.13130-1_orig/configure.in Firebird-2.0.4.13130-1/configure.in
--- Firebird-2.0.4.13130-1_orig/configure.in	2008-07-23 17:52:12.000000000 -0400
+++ Firebird-2.0.4.13130-1/configure.in	2008-07-23 17:57:56.000000000 -0400
@@ -465,7 +465,7 @@
 			AC_CHECK_LIB(termcap, tgetent, TERMLIB=termcap, \
 				AC_CHECK_LIB(tinfo, tgetent, TERMLIB=tinfo, \
 					AC_MSG_ERROR(termcap support not found)))))  
-	AC_SUBST(EDITLINE_A, -leditline)
+	AC_SUBST(EDITLINE_A, -ledit)
 	AC_SUBST(TERMLIB)
 	XE_APPEND(-l$TERMLIB, LIBS)
 fi
