--- src/drivers/snmp/snmp.c 2003-04-05 14:58:00.000000000 +0200
+++ src/drivers/snmp/snmp.c 2003-08-21 14:25:11.000000000 +0200
@@ -245,7 +245,10 @@
     return ret;
 }

-int snmp_ups_program_eeprom(UPSINFO *ups, int command, char *data)
+/*int snmp_ups_program_eeprom(UPSINFO *ups, int command, char *data)
+ * Modified for debian
+*/
+int snmp_ups_program_eeprom(UPSINFO *ups, int command, void *data)
 {
     return 0;
 }
