diff -urp ieee80211-1.0.3/net/ieee80211.h ieee80211-1.0.3-broadcast/net/ieee80211.h
--- ieee80211-1.0.3/net/ieee80211.h	2005-07-15 01:53:35.000000000 +0200
+++ ieee80211-1.0.3-broadcast/net/ieee80211.h	2005-09-02 19:15:03.000000000 +0200
@@ -720,6 +720,11 @@ static inline int is_multicast_ether_add
 	return addr[0] & 0x01;
 }
 #endif
+static inline int is_broadcast_ether_addr(const u8 *addr)
+{
+    return (addr[0] & 0xff && addr[1] & 0xff && addr[2] & 0xff &&
+        addr[3] & 0xff && addr[4] & 0xff && addr[5] & 0xff);
+}
 
 #define CFG_IEEE80211_RESERVE_FCS (1<<0)
 #define CFG_IEEE80211_COMPUTE_FCS (1<<1)
