--- cxfe.c	2004-11-27 03:07:59.000000000 +0100
+++ cxfe-max-osd.c	2004-12-11 18:22:29.171711112 +0100
@@ -125,11 +125,22 @@
 //  if(fbxine.osd.enabled) {
     va_list   args;
     char     *buf;
-    int       n, size = 100;
+    int       n, size = 47;
     
     if((buf = xine_xmalloc(size)) == NULL) 
       return;
     
+    va_start(args, info);
+    n = vsnprintf(buf, size, info, args);
+    va_end(args);
+
+    buf = realloc(buf, size+3);
+    buf[size-1] = '.';
+    buf[size] = '.';
+    buf[size+1] = '.';
+    buf[size+2] = '\0';
+
+#if 0
     while(1) {
       
       va_start(args, info);
@@ -147,6 +158,7 @@
       if((buf = realloc(buf, size)) == NULL)
 	return;
     }
+#endif
 
     xine_osd_clear(osd);
 
