diff -Nur urlgfe-1.0.2.orig/configure.in urlgfe-1.0.2/configure.in
--- urlgfe-1.0.2.orig/configure.in	2007-07-25 00:44:30.000000000 +0000
+++ urlgfe-1.0.2/configure.in	2007-07-31 00:12:33.605934138 +0000
@@ -41,7 +41,7 @@
 dnl use "install-sh -d" instead "makeinstalldirs"
 dnl MKINSTALLDIRS="\$(top_srcdir)/install-sh -d"
 MKINSTALLDIRS="install-sh -d"
-
+AC_SUBST(MKINSTALLDIRS)
 
 dnl checks for requirements.  --- start ---
 PKG_CHECK_MODULES(GTK, gtk+-2.0)
