--- rhide-1.5pre-20020825/librhgdb/regs.c.orig	Mon Aug 26 02:20:10 2002
+++ rhide-1.5pre-20020825/librhgdb/regs.c	Tue Aug 27 18:37:38 2002
@@ -4,6 +4,10 @@
 #include <librhgdb.h>
 #include <rhgdbint.h>
 
+#ifndef FP_REGNUM_P
+# define FP_REGNUM_P(n)	(FP0_REGNUM <= (n) && (n) < (FP0_REGNUM + 8))
+#endif
+
 int
 register_count()
 {
@@ -13,7 +17,7 @@
 int
 is_float_reg(int num)
 {
-  return IS_FP_REGNUM(num);
+  return FP_REGNUM_P(num);
 }
 
 const char *
