--- tvision/include/compatlayer.h
+++ tvision/include/compatlayer.h
@@ -812,10 +812,6 @@
    #define _GNU_SOURCE
    #define CLY_getline getline
   #endif
-  // Avoid including libintl.h, we have the prototypes
-  // This is a temporal workaround.
-  #undef  _LIBINTL_H
-  #define _LIBINTL_H 1
  #endif
 
  /* Solaris using gcc but not glibc */
--- setedit/infview/include/inf.h
+++ setedit/infview/include/inf.h
@@ -310,7 +310,7 @@
 
  virtual TPalette& getPalette() const;
  virtual void close();
- virtual TInfWindow::~TInfWindow();
+ virtual ~TInfWindow();
  TInfViewer *viewer;
  Boolean isTheOne;
  static void SetMagicEditorInfoTranslation() { TInfViewer::TranslateName=1; };
--- setedit/setedit/runprog.cc
+++ setedit/setedit/runprog.cc
@@ -89,17 +89,15 @@
  if (!Command[0])
     strcpy(Command,"make");
 
- #pragma pack(1)
  struct
  {
-  char   ComAux[maxCommand]  __attribute__((packed));
+  char   ComAux[maxCommand];
-  uint32 Options             __attribute__((packed));
+  uint32 Options;
-  uint32 OpsScroll           __attribute__((packed));
+  uint32 OpsScroll;
-  uint32 OpsScrHz            __attribute__((packed));
+  uint32 OpsScrHz;
-  char   Lines[4]            __attribute__((packed));
+  char   Lines[4];
-  TListBoxRec tl             __attribute__((packed));
+  TListBoxRec tl;
  } box;
- #pragma pack()
  strcpy(box.ComAux,Command);
  box.Options=Options & (~edsmScrollMask);
  box.OpsScroll=(Options & edsmScrollMask)/edsmScrollShifter;
--- rhide-1.5.20050823/rhgdb/include/rhgdb.h
+++ rhide-1.5.20050823/rhgdb/include/rhgdb.h
@@ -46,8 +46,8 @@
 extern int use_dual_display;
 class TDirList;
 extern TDirList *src_dirs;
-class TDisassemblerWindow;
+class TWindow;
-extern TDisassemblerWindow *dis_win;
+extern TWindow *dis_win;
 class TDataWindow;
 
 void STEP(int _switch_to_user = 1);
--- rhide-1.5.20050823/rhgdb/main.cc
+++ rhide-1.5.20050823/rhgdb/main.cc
@@ -119,7 +119,7 @@
 int tabsize = 8;
 int use_dual_display = 0;
 
-TDisassemblerWindow *dis_win = NULL;
+TWindow *dis_win = NULL;
 
 static void ReadOptions(char *);
 static void SaveOptions(char *);
@@ -426,7 +426,7 @@
     TRect r(1, 1, 79, 20);
 
     dis_win = new TDisassemblerWindow(r, _("Disassembler window"));
-    AddWindow(dis_win, (TWindow **) & dis_win);
+    AddWindow(dis_win, & dis_win);
   }
 }
 
@@ -443,18 +443,18 @@
   if (TProgram::deskTop->execView(insp) == cmOK)
   {
     insp->getData(buf);
-    TInspector *w = new TInspector(TProgram::deskTop->getExtent(), buf);
+    TWindow *w = new TInspector(TProgram::deskTop->getExtent(), buf);
 
     w->helpCtx = hcInspector;
     w->options |= ofCentered;
-    w->update(buf);
+    ((TInspector *) w)->update(buf);
-    AddWindow(w, (TWindow **) & w);
+    AddWindow(w, & w);
   }
   TObject::destroy(insp);
 }
 
 
-static TDataWindow *stack_win = NULL;
+static TWindow *stack_win = NULL;
 
 void
 RHGDBApp::handleEvent(TEvent & event)
@@ -498,7 +498,7 @@
         case cmDisWindow:
           OpenDisWin();
           if (debugger_started())
-            dis_win->update(stop_pc);
+            ((TDisassemblerWindow *) dis_win)->update(stop_pc);
           clearEvent(event);
           break;
         case cmDataWindow:
@@ -511,7 +511,7 @@
           if (!stack_win)
           {
             if ((stack_win = TDataWindow::stackWindow()))
-              AddWindow(stack_win, (TWindow **) & stack_win);
+              AddWindow(stack_win, & stack_win);
           }
           else
             stack_win->select();
@@ -676,7 +676,7 @@
   UpdateCallStackWindow();
   TDataWindow::updateAll();
   if (dis_win)
-    dis_win->update(stop_pc);
+    ((TDisassemblerWindow *) dis_win)->update(stop_pc);
 }
 
 static void InitDebuggerInterface();
@@ -1060,7 +1060,7 @@
     else if (main_line != 0)
     {
       OpenDisWin();
-      dis_win->update(main_line);
+      ((TDisassemblerWindow *) dis_win)->update(main_line);
     }
   }
   app->run();
@@ -1227,7 +1227,7 @@
   {
     extern unsigned long stop_pc;
 
-    dis_win->update(stop_pc);
+    ((TDisassemblerWindow *) dis_win)->update(stop_pc);
   }
   if (select_dis_win)
     dis_win->select();
