diff -uN --recursive setedit/include/ced_clas.h setedit.new/include/ced_clas.h
--- setedit/include/ced_clas.h	Sat Oct 30 15:58:58 1999
+++ setedit.new/include/ced_clas.h	Thu Oct 25 20:37:20 2001
@@ -261,9 +261,9 @@
     static unsigned editorFlags;
     static unsigned fromFlags;
     static unsigned whereFlags;
-    static char near findStr[maxFindStrLen];
-    static char near replaceStr[maxReplaceStrLen];
-    static TCEditor * near clipboard;
+    static char findStr[maxFindStrLen];
+    static char replaceStr[maxReplaceStrLen];
+    static TCEditor * clipboard;
     static TCommandSet cmdsAux;     // This is used to quickly enable/disable all the
                                     // editor commands when it get/releases the focus
     uchar lockCount;
@@ -590,8 +590,8 @@
 
 private:
 
-    static const char * near clipboardTitle;
-    static const char * near untitled;
+    static const char * clipboardTitle;
+    static const char * untitled;
 
     SetDefStreamMembersNoConst(TCEditWindow)
 };
diff -uN --recursive setedit/include/ceditint.h setedit.new/include/ceditint.h
--- setedit/include/ceditint.h	Sun Sep 26 22:27:24 1999
+++ setedit.new/include/ceditint.h	Thu Oct 25 21:31:02 2001
@@ -6,6 +6,8 @@
 // International support from TV
 #include <intl.h>
 
+#include <ucdefs.h>
+
 #ifndef HAVE_INTL_SUPPORT
 #define NO_INTL_SUP
 #endif
@@ -44,7 +46,7 @@
 
 // PCRE support
 #if defined(HAVE_PCRE_LIB)
-#define SUP_PCRE
+#define SUP_PCRE
 #endif
 
 #ifdef HAVE_PCRE206
diff -uN --recursive setedit/infview/inf.cc setedit.new/infview/inf.cc
--- setedit/infview/inf.cc	Mon Oct 18 00:34:49 1999
+++ setedit.new/infview/inf.cc	Thu Oct 25 20:37:20 2001
@@ -206,6 +206,7 @@
 #define Uses_TSOSSortedListBox
 #include <inf.h>
 
+#include <ucdefs.h>
 #include <ctype.h>
 
 #ifndef Linux
diff -uN --recursive setedit/infview/infbase.cc setedit.new/infview/infbase.cc
--- setedit/infview/infbase.cc	Sun Dec 19 16:35:11 1999
+++ setedit.new/infview/infbase.cc	Thu Oct 25 21:36:10 2001
@@ -1596,15 +1596,13 @@
 }
 
 #ifndef __DJGPP__
-extern "C" long __filelength(int fhandle);
-/* Allready defined in bc.cc of TVision for non-djgpp platforms
 long filelength(int fhandle)
 {
  long p=lseek (fhandle,0,SEEK_CUR); // They doesn't have tell!!!
  long ret=lseek(fhandle,0,SEEK_END);
  lseek(fhandle,p,SEEK_SET);
  return ret;
-}*/
+}
 #endif
 
 FILE *TInfFile::fOpen(char *Nombre)
diff -uN --recursive setedit/infview/manview.cc setedit.new/infview/manview.cc
--- setedit/infview/manview.cc	Wed Sep 29 00:50:25 1999
+++ setedit.new/infview/manview.cc	Thu Oct 25 20:37:20 2001
@@ -14,6 +14,7 @@
 #define Uses_TEvent
 
 #include <tv.h>
+#include <ucdefs.h>
 
 #define Uses_TEnhancedText
 #define Uses_TManPageView
diff -uN --recursive setedit/mainsrc/ceditor.cc setedit.new/mainsrc/ceditor.cc
--- setedit/mainsrc/ceditor.cc	Thu Dec 30 01:13:26 1999
+++ setedit.new/mainsrc/ceditor.cc	Thu Oct 25 21:39:24 2001
@@ -11580,6 +11580,9 @@
 }
 #endif
 
+#ifndef __DJGPP__
+long filelength(int fhandle);
+#endif
 
 static
 FILE *ExpandToTempIfNeeded(FILE *f, char *&temp, char *name)
diff -uN --recursive setedit/mainsrc/loaddefl.cc setedit.new/mainsrc/loaddefl.cc
--- setedit/mainsrc/loaddefl.cc	Sat Sep 25 17:49:06 1999
+++ setedit.new/mainsrc/loaddefl.cc	Thu Oct 25 21:19:04 2001
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #include <string.h>
+#include <ucdefs.h>
 
 #define Uses_TSOSStringCollection
 #define Uses_TCEditor_External
diff -uN --recursive setedit/mainsrc/loadfunc.cc setedit.new/mainsrc/loadfunc.cc
--- setedit/mainsrc/loadfunc.cc	Sun Feb 14 22:59:22 1999
+++ setedit.new/mainsrc/loadfunc.cc	Thu Oct 25 21:19:28 2001
@@ -6,6 +6,8 @@
 #include <stdio.h>
 #include <tv.h>
 
+#include <ucdefs.h>
+
 #include <pathtool.h>
 
 int isValidForFile(int c);
diff -uN --recursive setedit/mainsrc/pmacros.cc setedit.new/mainsrc/pmacros.cc
--- setedit/mainsrc/pmacros.cc	Wed Sep  8 01:01:40 1999
+++ setedit.new/mainsrc/pmacros.cc	Thu Oct 25 21:20:33 2001
@@ -3,6 +3,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <ctype.h>
+#include <ucdefs.h>
 
 #define Uses_TPMCollection
 #define Uses_MsgBox
diff -uN --recursive setedit/setedit/edprj.cc setedit.new/setedit/edprj.cc
--- setedit/setedit/edprj.cc	Sun Jan  2 21:35:30 2000
+++ setedit.new/setedit/edprj.cc	Thu Oct 25 21:39:40 2001
@@ -649,7 +649,10 @@
     LoadEditorDesktop(0);
 }
 
-
+#ifndef __DJGPP__
+long filelength(int fhandle);
+#endif
+    
 // That's the interface with the SDG module.
 // These routines must provide the buffers with sources from the project
 
diff -uN --recursive setedit/setedit/loadcle.cc setedit.new/setedit/loadcle.cc
--- setedit/setedit/loadcle.cc	Sun Nov 21 23:26:09 1999
+++ setedit.new/setedit/loadcle.cc	Thu Oct 25 21:18:36 2001
@@ -2,15 +2,16 @@
    see copyrigh file for details */
 #include <ceditint.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
 
 #define Uses_TStringCollection
 #include <tv.h>
 
-#ifdef SUP_PCRE
+#ifdef SUP_PCRE
 #include <pcre.h>
-#endif
+#endif
 #include <dyncat.h>
 #include <loadcle.h>
 
@@ -337,7 +338,7 @@
  buf[len]=0;
 }
 #else
-static pcre *PCRECompileRegEx(char *) {return 0;}
+static pcre *PCRECompileRegEx(char *, int &) {return 0;}
 int CLEDoSearch(char *, int , pcre *) {return 0;}
 void CLEGetMatch(int , char *, int ) {}
 #endif
diff -uN --recursive setedit/setedit/menuload.cc setedit.new/setedit/menuload.cc
--- setedit/setedit/menuload.cc	Tue Jul 27 01:51:28 1999
+++ setedit.new/setedit/menuload.cc	Thu Oct 25 21:19:56 2001
@@ -10,6 +10,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <ucdefs.h>
 
 #define Uses_TMenu
 #define Uses_TMenuItem
diff -uN --recursive setedit/settvuti/tnocastc.cc setedit.new/settvuti/tnocastc.cc
--- setedit/settvuti/tnocastc.cc	Fri Dec  3 01:52:56 1999
+++ setedit.new/settvuti/tnocastc.cc	Thu Oct 25 20:37:20 2001
@@ -14,6 +14,7 @@
 #include <settvuti.h>
 
 #include <ctype.h>
+#include <ucdefs.h>
 
 #define shiftKeys (char)(event.keyDown.shiftState & 0xFF)

diff -uN --recursive setedit/sdg/txhgen.cc setedit.new/sdg/txhgen.cc
--- setedit/sdg/txhgen.cc.orig      Thu Oct 25 22:02:32 2001
+++ setedit.new/sdg/txhgen.cc   Thu Oct 25 22:02:59 2001
@@ -30,6 +30,7 @@
 #endif
 #include <stdlib.h>
 #include <unistd.h>
+#include <ucdefs.h>
 
 #define Uses_TNoCaseSOSStringCollection
 #define Uses_TNCSAssociative

diff -uN --recursive setedit/mainsrc/loadshl.cc setedit.new/mainsrc/loadshl.cc
--- setedit/mainsrc/loadshl.cc	Fri Dec  3 01:52:38 1999
+++ setedit.new/mainsrc/loadshl.cc	Fri Oct 26 01:47:32 2001
@@ -1047,8 +1047,8 @@
 // Dummies
 static void PCREInitCompiler() {}
 static void PCREStopCompiler() {}
-static pcre *PCRECompileRegEx(char *) { return 0; }
-static int PCREDoSearch(char *, int , pcre *) {}
+static pcre *PCRECompileRegEx(char *) { return 0; }
+static int PCREDoSearch(char *, int , pcre *) { return 0; }
 #endif
 /********************** End Regular expressions file matching stuff *****************/
 
