--- Makefile.orig	2007-04-24 09:07:22.000000000 -0700
+++ Makefile	2007-04-24 09:07:43.000000000 -0700
@@ -310,10 +310,10 @@
 
 $(MPPDEC_TARGET): $(MPPDEC_OBJ)
 ifndef BLD_STATIC
-	$(CC)         $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET)        $(LDADD)
+	$(CC)         $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET)        $(LDADD) -lesd
 #	-$(STRIP)     $(STRIPOPT)                $(MPPDEC_TARGET)
 else
-	$(CC) -static $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET)-static $(LDADD)
+	$(CC) -static $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET)-static $(LDADD) -lesd
 #	-$(STRIP)     $(STRIPOPT)                $(MPPDEC_TARGET)-static
 endif
 
