--- Makefile.orig	2004-06-18 23:27:10.227221536 +0300
+++ Makefile	2004-06-18 23:29:28.279234440 +0300
@@ -20,8 +20,7 @@
 # If unset Intel 386 with optimization for Pentium is selected.
 
 ifdef ARCH
-else
-ARCH     = -march=i586 -mcpu=i386
+ARCH     = #-march=i586 -mcpu=i386
 endif
 
 
@@ -80,7 +79,7 @@
 # Name of libraries you need for linking
 
 LDADD    = -lm
-LDADD   += -lesd
+LDADD   += `esd-config --libs`
 #LDADD  += -lossaudio
 #LDADD  += -lrt
 #LDADD  += -lsocket -lnsl
@@ -100,7 +99,7 @@
 # Warning options, unset if the compiler makes trouble because of unknown
 # options
 
-WARN  = \
+WARN  = #\
 	-Wall                   \
 	-pedantic               \
 	-W                      \
@@ -189,7 +188,7 @@
 
 # Merge all options together for CFLAGS and CFLAG_SIZE
 
-CFLAGS        = $(ARCH) $(FEATURE) $(WARN) $(PROFILE) $(DEBUG) -DMAKE_$(BITS)BIT -DMPPDEC_VERSION=\"$(MPPDEC_VERSION)\" -DMPPENC_VERSION=\"$(MPPENC_VERSION)\"
+CFLAGS       += $(ARCH) $(FEATURE) $(WARN) $(PROFILE) $(DEBUG) -DMAKE_$(BITS)BIT -DMPPDEC_VERSION=\"$(MPPDEC_VERSION)\" -DMPPENC_VERSION=\"$(MPPENC_VERSION)\"
 CFLAGS       += -I$(XINCLDIR) -L$(XLIBDIR)
 #CFLAGS      += $(ASSEM)
 CFLAGS_SIZE   = $(CFLAGS) $(OPTIM_SIZE)
@@ -275,10 +274,13 @@
 
 
 $(MPPDEC_TARGET): $(MPPDEC_OBJ)
+ifndef BLDSTATIC
 	$(CC)         $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET)        $(LDADD)
-	-$(STRIP)     $(STRIPOPT)                $(MPPDEC_TARGET)
-	$(CC) -static $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET)-static $(LDADD)
-	-$(STRIP)     $(STRIPOPT)                $(MPPDEC_TARGET)-static
+#	-$(STRIP)     $(STRIPOPT)                $(MPPDEC_TARGET)
+else
+	$(CC) -static $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET) $(LDADD)
+#	-$(STRIP)     $(STRIPOPT)                $(MPPDEC_TARGET)
+endif
 
 
 $(MPPDEC_TARGET)16: $(MPPDEC_OBJ)
@@ -303,10 +305,13 @@
 
 
 $(MPPENC_TARGET): $(MPPENC_OBJ)
+ifndef BLDSTATIC
 	$(CC)         $(CFLAGS) $(MPPENC_OBJ) -o $(MPPENC_TARGET)        $(LDADD)
-	-$(STRIP)     $(STRIPOPT)                $(MPPENC_TARGET)
-	$(CC) -static $(CFLAGS) $(MPPENC_OBJ) -o $(MPPENC_TARGET)-static $(LDADD)
-	-$(STRIP)     $(STRIPOPT)                $(MPPENC_TARGET)-static
+#	-$(STRIP)     $(STRIPOPT)                $(MPPENC_TARGET)
+else
+	$(CC) -static $(CFLAGS) $(MPPENC_OBJ) -o $(MPPENC_TARGET) $(LDADD)
+#	-$(STRIP)     $(STRIPOPT)                $(MPPENC_TARGET)
+endif
 
 
 $(MPPDEC_TARGET)-profiling: $(MPPDEC_OBJ)
@@ -336,12 +341,12 @@
 
 $(REPLAY_TARGET): $(REPLAY_OBJ) mpp.h config.c
 	$(CC) $(CFLAGS) $(REPLAY_OBJ) -o $(REPLAY_TARGET)      $(LDADD)
-	-$(STRIP)     $(STRIPOPT)           $(REPLAY_TARGET)
+#	-$(STRIP)     $(STRIPOPT)           $(REPLAY_TARGET)
 
 
 $(TAGGER_TARGET):
 	$(CC) -DMPP_DECODER $(CFLAGS) -o $(TAGGER_TARGET) $(TAGGER_TARGET).c
-	-$(STRIP)     $(STRIPOPT)           $(TAGGER_TARGET)
+#	-$(STRIP)     $(STRIPOPT)           $(TAGGER_TARGET)
 
 
 $(CLIPSTAT_TARGET):
