--- perl-5.8.0/Makefile.SH.orig	2002-12-28 07:46:27.000000000 +0200
+++ perl-5.8.0/Makefile.SH	2002-06-09 19:04:06.000000000 +0200
@@ -480,7 +480,7 @@
 	case "$useshrplib" in
 	true)
 		$spitshell >>Makefile <<'!NO!SUBS!'
-	$(LD) -o $@ $(SHRPLDFLAGS) perl$(OBJ_EXT) $(obj)
+	$(LD) -o $@ $(SHRPLDFLAGS) perl$(OBJ_EXT) $(obj) -Wl,-soname -Wl,libperl.so.`echo $(LIBPERL) | cut -d. -f3`
 !NO!SUBS!
 		case "$osname" in
 		aix)

