diff -ur ipcad-3.7.3.orig/main.c ipcad-3.7.3/main.c
--- ipcad-3.7.3.orig/main.c	2007-04-22 11:08:45.000000000 +0300
+++ ipcad-3.7.3/main.c	2011-03-31 20:19:12.000000000 +0300
@@ -26,6 +26,8 @@
  * $Id: ipcad-3.7.3-signal_h.patch,v 1.1 2011/03/31 17:27:08 ssuominen Exp $
  */
 
+#include <signal.h> /* signal */
+
 #include "ipcad.h"
 #include "cfgvar.h"
 #include "servers.h"
diff -ur ipcad-3.7.3.orig/pps.c ipcad-3.7.3/pps.c
--- ipcad-3.7.3.orig/pps.c	2007-04-22 11:08:45.000000000 +0300
+++ ipcad-3.7.3/pps.c	2011-03-31 20:18:45.000000000 +0300
@@ -26,6 +26,7 @@
  * $Id: ipcad-3.7.3-signal_h.patch,v 1.1 2011/03/31 17:27:08 ssuominen Exp $
  */
 
+#include <signal.h> /* sigprocmask */
 
 #include "ipcad.h"
 #include "opt.h"
diff -ur ipcad-3.7.3.orig/servers.c ipcad-3.7.3/servers.c
--- ipcad-3.7.3.orig/servers.c	2007-04-22 11:08:45.000000000 +0300
+++ ipcad-3.7.3/servers.c	2011-03-31 20:25:10.000000000 +0300
@@ -26,6 +26,8 @@
  * $Id: ipcad-3.7.3-signal_h.patch,v 1.1 2011/03/31 17:27:08 ssuominen Exp $
  */
 
+#include <signal.h> /* pthread_kill and sigprocmask */
+
 #include "ipcad.h"
 #include "servers.h"
 #include "opt.h"
