diff -upr opencryptoki-2.2.4.1.orig/usr/lib/pkcs11/tpm_stdll/tpm_util.c opencryptoki-2.2.4.1/usr/lib/pkcs11/tpm_stdll/tpm_util.c
--- opencryptoki-2.2.4.1.orig/usr/lib/pkcs11/tpm_stdll/tpm_util.c	2007-02-03 00:19:41.000000000 +0200
+++ opencryptoki-2.2.4.1/usr/lib/pkcs11/tpm_stdll/tpm_util.c	2007-03-25 20:42:38.926868516 +0300
@@ -51,6 +51,10 @@
 
 #include "tpm_specific.h"
 
+#ifndef UINT64
+#define UINT64 unsigned long long
+#endif
+
 extern TSS_HCONTEXT tspContext;
 
 UINT32
