--- ../vkeybd.orig/Makefile	2001-05-25 16:57:22.000000000 +0200
+++ Makefile	2002-12-16 03:36:46.000000000 +0100
@@ -8,7 +8,9 @@
 #
 # installation directory
 #
+ifndef $(PREFIX)
 PREFIX = /usr/local
+endif
 # binary and Tcl script are put there
 INSTALL_DIR = $(PREFIX)/bin
 # man page
@@ -24,9 +26,15 @@
 # device selections -- multiple avaialble
 # to disable the device, set value 0 (do not comment out!)
 #
+ifndef $(USE_AWE)
 USE_AWE = 1
+endif
+ifndef $(USE_MIDI)
 USE_MIDI = 1
+endif
+ifndef $(USE_ALSA)
 USE_ALSA = 1
+endif
 
 #
 # Tcl/Tk library -- depends on your distribution
