--- ../vkeybd.orig/Makefile	2002-12-16 07:15:57.000000000 +0100
+++ Makefile	2002-12-16 07:22:42.000000000 +0100
@@ -93,17 +93,17 @@
 	$(CC) -o $@ $^ -lm
 
 install: $(TARGETS) vkeybd.tcl vkeybd.list
-	mkdir -p $(INSTALL_DIR)
-	install -c -s vkeybd $(INSTALL_DIR)
-	install -c -s sftovkb $(INSTALL_DIR)
-	rm -f $(INSTALL_DIR)/vkeybd.tcl
-	mkdir -p $(VKBLIB_DIR)
-	install -c -m 444 vkeybd.tcl $(VKBLIB_DIR)
-	install -c -m 444 vkeybd.list $(VKBLIB_DIR)
+	mkdir -p ${DESTDIR}$(INSTALL_DIR)
+	install -c -s vkeybd ${DESTDIR}$(INSTALL_DIR)
+	install -c -s sftovkb ${DESTDIR}$(INSTALL_DIR)
+	rm -f ${DESTDIR}$(INSTALL_DIR)/vkeybd.tcl
+	mkdir -p ${DESTDIR}$(VKBLIB_DIR)
+	install -c -m 444 vkeybd.tcl ${DESTDIR}$(VKBLIB_DIR)
+	install -c -m 444 vkeybd.list ${DESTDIR}$(VKBLIB_DIR)
 
 install-man:
-	mkdir -p $(MAN_DIR)
-	install -c -m 444 vkeybd.man $(MAN_DIR)/vkeybd.$(MAN_SUFFIX)
+	mkdir -p ${DESTDIR}$(MAN_DIR)
+	install -c -m 444 vkeybd.man ${DESTDIR}$(MAN_DIR)/vkeybd.$(MAN_SUFFIX)
 
 install-all: install install-man
 
