--- hibernate-script-1.12/scriptlets.d/misclaunch	2005-11-26 02:39:05.000000000 +0200
+++ hibernate-script-1.12.new/scriptlets.d/misclaunch	2005-11-26 02:39:20.000000000 +0200
@@ -19,7 +19,7 @@ MiscLaunchAuxFunc${MISC_LAUNCH_COUNT}() 
 Executing $@...
 EOT2
     $@
-    [ $? -ne 0 ] && return 1
+    [ \$? -ne 0 ] && return 1
     return 0
 }
 EOT
@@ -39,7 +39,7 @@ MiscLaunchAuxFunc${MISC_LAUNCH_COUNT}() 
 Executing $@...
 EOT2
     $@
-    [ $? -ne 0 ] && return 1
+    [ \$? -ne 0 ] && return 1
     return 0
 }
 EOT
