diff -urN mkinitrd-3.5.7.orig/mkinitrd mkinitrd-3.5.7/mkinitrd
--- mkinitrd-3.5.7.orig/mkinitrd	2003-07-09 18:26:02.000000000 -0500
+++ mkinitrd-3.5.7/mkinitrd	2003-10-25 11:56:32.083678110 -0500
@@ -534,9 +534,8 @@
 	cp $verbose --parents -a $pv $MNTIMAGE/
     done
 
-    inst /sbin/vgwrapper "$MNTIMAGE/bin/vgwrapper"
-    ln "$MNTIMAGE/bin/vgwrapper" "$MNTIMAGE/bin/vgscan"
-    ln "$MNTIMAGE/bin/vgwrapper" "$MNTIMAGE/bin/vgchange"
+    inst /sbin/vgscan.static "$MNTIMAGE/bin/vgscan"
+    inst /sbin/vgchange.static "$MNTIMAGE/bin/vgchange"
 
     mknod $MNTIMAGE/dev/lvm b 109 0
 fi
