--- src/client/callbacks.c.old	2003-02-22 00:24:10.000000000 -0500
+++ src/client/callbacks.c	2003-02-22 00:25:54.000000000 -0500
@@ -70,15 +70,15 @@
       {
          /* Handle the failsafe session */
          if (!strcmp(e->xsession->path, "failsafe"))
-            snprintf(buf, PATH_MAX, "/etc/X11/Xsession failsafe");
+            snprintf(buf, PATH_MAX, "/etc/X11/Sessions/Xsession failsafe");
          else
             snprintf(buf, PATH_MAX, "%s", e->xsession->path);
       }
       else
-         snprintf(buf, PATH_MAX, "/etc/X11/Xsession %s", e->xsession->name);
+         snprintf(buf, PATH_MAX, "/etc/X11/Sessions/%s", e->xsession->name);
    }
    else
-      snprintf(buf, PATH_MAX, "/etc/X11/Xsession");
+      snprintf(buf, PATH_MAX, "/etc/X11/Sessions/Xsession");
    /* Default Session */
 
 #ifdef X_TESTING
