diff -ur swftools-0.9.0.orig/lib/pdf/GFXOutputDev.cc swftools-0.9.0/lib/pdf/GFXOutputDev.cc
--- swftools-0.9.0.orig/lib/pdf/GFXOutputDev.cc	2009-04-13 02:25:32.000000000 +0300
+++ swftools-0.9.0/lib/pdf/GFXOutputDev.cc	2009-08-09 17:33:23.000000000 +0300
@@ -2504,8 +2504,8 @@
     memset(f, 0, sizeof(fontfile_t));
     f->filename = filename;
     int len = strlen(filename);
-    char*r1 = strrchr(filename, '/');
-    char*r2 = strrchr(filename, '\\');
+    char*r1 = const_cast<char*> (strrchr(filename, '/'));
+    char*r2 = const_cast<char*> (strrchr(filename, '\\'));
     if(r2>r1)
         r1 = r2;
     if(r1) {
diff -ur swftools-0.9.0.orig/lib/pdf/pdf.cc swftools-0.9.0/lib/pdf/pdf.cc
--- swftools-0.9.0.orig/lib/pdf/pdf.cc	2009-04-13 02:25:32.000000000 +0300
+++ swftools-0.9.0/lib/pdf/pdf.cc	2009-08-09 17:36:46.000000000 +0300
@@ -458,7 +458,7 @@
     i->filename = strdup(filename);
 
     char*x = 0;
-    if((x = strchr(filename, '|'))) {
+    if((x = const_cast<char*> (strchr(filename, '|')))) {
 	*x = 0;
 	userPassword = x+1;
     }
