--- org/lcr.ui.h	2008-01-04 16:15:50.000000000 +0100
+++ src/lcr.ui.h	2008-01-04 16:31:17.000000000 +0100
@@ -43,7 +43,8 @@
 static QString debugfile=confdir+"debug.log";
 static QString connectionfile=confdir+"connections.xml";
 static QString connectionoutputfile=confdir+"connections.html";
-static QString providerfileurl="http://zippedprovider.webhop.org";
+//static QString providerfileurl="http://zippedprovider.webhop.org";
+static QString providerfileurl="http://home.arcor.de/cubical/provider.xml.bz2";
 
 static QString unknown="???";
 
@@ -313,7 +314,7 @@
     QRegExp wv("(Username|Password|Phone|Dialer LCR|Dialer Defaults|Modem|Init1|Init2|Init3|Baud|Dial Prefix|Dial Command)");
     
     QStringList lines;
-    //lines+="[Dialer Defaults]";
+    lines+="[Dialer Defaults]";
     lines+="Modem = "+ModemDevice->text();
     lines+="Baud = "+ModemBaud->text();
     lines+="Dial Command = "+DialString->text();
@@ -365,7 +366,7 @@
  if ( conffile.open( IO_WriteOnly ) ) {
      QTextStream confstream( &conffile );
      for ( QStringList::Iterator it = lines.begin(); it != lines.end(); ++it )
-  confstream << *it << "\n";
+	  confstream << *it << "\n";
      conffile.close();
  }
  
