Index: ccrtp-1.3.5/src/queue.cpp
===================================================================
--- ccrtp-1.3.5.orig/src/queue.cpp
+++ ccrtp-1.3.5/src/queue.cpp
@@ -458,7 +458,7 @@ static uint32 MD5BasedRandom32()
 		uint8 array[1];
 		struct {
 			timeval time;
-			uint32 address;
+			void *address;
 			uint8 cname[10];
 		} data;
 	} message;
@@ -473,7 +473,7 @@ static uint32 MD5BasedRandom32()
 	message.array[0] =
 		static_cast<uint8>(message.data.time.tv_sec * 
 				   message.data.time.tv_usec);
-	message.data.address = (uint32)(&message);
+	message.data.address = (&message);
 	memcpy(message.data.cname,
 	       defaultApplication().getSDESItem(SDESItemTypeCNAME).c_str(),10);
 
