diff -u cl-rsm-fuzzy-1.2.orig/check.lisp cl-rsm-fuzzy-1.2/check.lisp
--- cl-rsm-fuzzy-1.2.orig/check.lisp	2003-09-11 12:29:29.000000000 -0500
+++ cl-rsm-fuzzy-1.2/check.lisp	2004-08-06 11:27:33.632557472 -0500
@@ -31,7 +31,7 @@
           (unless (and (>= y 0.0)
                        (<= y 1.0))
             (return (values nil (format nil "~s has a y coordinate that is 
-not in the range [0,1]."))))))
+not in the range [0,1]." func-spec))))))
       pairs)))
 
 (defun check-special-func (func-spec)
Common subdirectories: cl-rsm-fuzzy-1.2.orig/debian and cl-rsm-fuzzy-1.2/debian
