diff -uNrx debian bitlbee-0.82/account.c bitlbee-0.83/account.c
--- bitlbee-0.82/account.c	Sat Nov  1 00:21:36 2003
+++ bitlbee-0.83/account.c	Sat Nov  1 00:32:37 2003
@@ -106,7 +106,7 @@
 		return;
 	}
 	
-	if( proto_prpl[a->protocol] == NULL )
+	if( proto_prpl[a->protocol]->login == NULL )
 	{
 		irc_usermsg( irc, "Support for protocol %s is not included in this BitlBee", proto_name[a->protocol] );
 		return;
