diff -ru bitlbee-1.2.2~/tests/check_set.c bitlbee-1.2.2/tests/check_set.c
--- bitlbee-1.2.2~/tests/check_set.c	2008-08-30 17:53:39.000000000 +0200
+++ bitlbee-1.2.2/tests/check_set.c	2008-08-30 17:53:58.000000000 +0200
@@ -98,6 +98,7 @@
 START_TEST(test_setstr_implicit)
 	void *data = "data";
 	set_t *s = NULL, *t;
+	t = set_add(&s, "name", "foo", NULL, data);
 	set_setstr(&s, "name", "bloe");
 	fail_unless(set_find(&s, "name") != NULL);
 END_TEST
