diff -ur orbital_eunuchs_sniper-1.29.orig/src/snipe2d.cpp orbital_eunuchs_sniper-1.29/src/snipe2d.cpp
--- orbital_eunuchs_sniper-1.29.orig/src/snipe2d.cpp	2003-12-31 16:05:07.079304568 -0500
+++ orbital_eunuchs_sniper-1.29/src/snipe2d.cpp	2003-12-31 16:06:52.962207920 -0500
@@ -1632,10 +1632,11 @@
 
     srand(SDL_GetTicks());
    
-    Game.mediaPath = (char*)calloc(strlen(argv[0]), sizeof(char));
+    Game.mediaPath = (char*)calloc(strlen("GENTOO_DIR\0")+1, sizeof(char));
 //   invocation = (char*)malloc(strlen(argv[0]) + 1); memcpy(invocation, argv[0], strlen(argv[0]));
     invocation = strdup(argv[0]);
-    sprintf(Game.mediaPath, "%s", dirname(invocation));
+    sprintf(Game.mediaPath, "GENTOO_DIR\0");
+    chdir(Game.mediaPath);
     free(invocation);
     invocation = argv[0];
 
