Index: rednotebook-0.9.2/rednotebook/gui/browser.py
===================================================================
--- rednotebook-0.9.2.orig/rednotebook/gui/browser.py
+++ rednotebook-0.9.2/rednotebook/gui/browser.py
@@ -28,6 +28,8 @@ import warnings
 import gtk
 import gobject
 
+webkit = None
+
 # Testing
 if __name__ == '__main__':
 	sys.path.insert(0, '../../')
@@ -35,13 +37,6 @@ if __name__ == '__main__':
 	#gtk.gdk.threads_init() # only initializes threading in the glib/gobject module
 	gobject.threads_init() # also initializes the gdk threads
 	
-
-try:
-	import webkit
-except ImportError:
-	logging.info('Importing webkit failed')
-	webkit = None
-	
 from rednotebook.util import filesystem
 	
 	
