Index: rednotebook-1.1.0/rednotebook/gui/browser.py
===================================================================
--- rednotebook-1.1.0.orig/rednotebook/gui/browser.py
+++ rednotebook-1.1.0/rednotebook/gui/browser.py
@@ -67,12 +67,6 @@ def windows_webkit_import():
 
 if sys.platform == 'win32':
 	windows_webkit_import()
-else:
-	try:
-		import webkit
-	except ImportError:
-		logging.info('webkit not found. For a nicer preview install python-webkit or pywebkitgtk')
-	
 	
 	
 def can_print_pdf():
