diff -Naur psi-0.10-test3-orig/psi.desktop psi-0.10-test3/psi.desktop
--- psi-0.10-test3-orig/psi.desktop	2005-11-07 00:39:22.000000000 +0000
+++ psi-0.10-test3/psi.desktop	2005-11-07 01:16:22.000000000 +0000
@@ -1,8 +1,12 @@
 Encoding=UTF-8
 Name=Psi
 GenericName=Jabber Client
+GenericName[pl]=Klient Jabbera
+GenericName[fr]=Client Jabber
 Comment=Communicate over the Jabber network
-Icon=psi
+Comment[pl]=Komunikator sieci Jabber
+Comment[fr]=Communiquer sur le réseau Jabber
+Icon=../icons/hicolor/48x48/apps/psi.png
 Exec=psi
 Terminal=false
 Categories=Qt;Network;InstantMessaging;
