diff -ur fam-oss-2.6.4.orig/fam/Listener.c++ fam-oss-2.6.4/fam/Listener.c++
--- fam-oss-2.6.4.orig/fam/Listener.c++	Tue Aug  7 13:03:44 2001
+++ fam-oss-2.6.4/fam/Listener.c++	Tue Aug  7 14:52:08 2001
@@ -213,6 +213,7 @@
     strcpy(sun.sun_path, tmpfile);
     free(tmpfile);
 
+    Cred::SuperUser.become_user();
     int client_sock = socket(PF_UNIX, SOCK_STREAM, 0);
     if (client_sock < 0)
     {   Log::perror("localclient socket(PF_UNIX, SOCK_STREAM, 0)");
