--- Makefile.orig	2004-04-17 01:01:36.000000000 -0700
+++ Makefile	2004-04-17 01:03:20.000000000 -0700
@@ -277,36 +277,36 @@
 $(MPPDEC_TARGET): $(MPPDEC_OBJ)
 	$(CC)         $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET)        $(LDADD)
 	-$(STRIP)     $(STRIPOPT)                $(MPPDEC_TARGET)
-	$(CC) -static $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET)-static $(LDADD)
-	-$(STRIP)     $(STRIPOPT)                $(MPPDEC_TARGET)-static
+#	$(CC) -static $(CFLAGS) $(MPPDEC_OBJ) -o $(MPPDEC_TARGET)-static $(LDADD)
+#	-$(STRIP)     $(STRIPOPT)                $(MPPDEC_TARGET)-static
 
 
 $(MPPDEC_TARGET)16: $(MPPDEC_OBJ)
 	make clean
 	BITS=16 make mppdec
 	$(MV_F) mppdec mppdec16
-	$(MV_F) mppdec-static mppdec16-static
+#	$(MV_F) mppdec-static mppdec16-static
 
 
 $(MPPDEC_TARGET)24: $(MPPDEC_OBJ)
 	make clean
 	BITS=24 make mppdec
 	$(MV_F) mppdec mppdec24
-	$(MV_F) mppdec-static mppdec24-static
+#	$(MV_F) mppdec-static mppdec24-static
 
 
 $(MPPDEC_TARGET)32: $(MPPDEC_OBJ)
 	make clean
 	BITS=32 make mppdec
 	$(MV_F) mppdec mppdec32
-	$(MV_F) mppdec-static mppdec32-static
+#	$(MV_F) mppdec-static mppdec32-static
 
 
 $(MPPENC_TARGET): $(MPPENC_OBJ)
 	$(CC)         $(CFLAGS) $(MPPENC_OBJ) -o $(MPPENC_TARGET)        $(LDADD)
 	-$(STRIP)     $(STRIPOPT)                $(MPPENC_TARGET)
-	$(CC) -static $(CFLAGS) $(MPPENC_OBJ) -o $(MPPENC_TARGET)-static $(LDADD)
-	-$(STRIP)     $(STRIPOPT)                $(MPPENC_TARGET)-static
+#	$(CC) -static $(CFLAGS) $(MPPENC_OBJ) -o $(MPPENC_TARGET)-static $(LDADD)
+#	-$(STRIP)     $(STRIPOPT)                $(MPPENC_TARGET)-static
 
 
 $(MPPDEC_TARGET)-profiling: $(MPPDEC_OBJ)
@@ -316,7 +316,7 @@
 	make speed2
 	$(CC3) -c      $(CFLAGS) $(CSRC)                            -fprofile-arcs         $(LDADD)
 	$(CC3)         $(CFLAGS) $(OBJ)  -o $(MPPDEC_TARGET)        -fbranch-probabilities $(LDADD)
-	$(CC3) -static $(CFLAGS) $(OBJ)  -o $(MPPDEC_TARGET)-static -fbranch-probabilities $(LDADD)
+#	$(CC3) -static $(CFLAGS) $(OBJ)  -o $(MPPDEC_TARGET)-static -fbranch-probabilities $(LDADD)
 
 
 $(MPPDEC_TARGET)-diet:   $(MPPDEC_ASO) config.h
