diff -NurB epiphany-1.0-orig/lib/ephy-node.c epiphany-1.0/lib/ephy-node.c
--- epiphany-1.0-orig/lib/ephy-node.c	2003-10-04 11:16:56.000000000 -0500
+++ epiphany-1.0/lib/ephy-node.c	2003-10-04 11:35:41.000000000 -0500
@@ -1109,8 +1109,7 @@
 
 	g_ptr_array_sort (newkids, compare_func);
 
-	new_order = g_new (int, newkids->len);
-	memset (new_order, -1, sizeof (int) * newkids->len);
+	new_order = g_new0 (int, newkids->len);
 
 	for (i = 0; i < newkids->len; i++)
 	{
