diff -NurB epiphany-1.0-orig/configure epiphany-1.0/configure
--- epiphany-1.0-orig/configure	2003-10-04 11:16:56.000000000 -0500
+++ epiphany-1.0/configure	2003-10-05 00:22:23.000000000 -0500
@@ -20222,9 +20222,9 @@
 		unset SAVE_CFLAGS
 	done
 	unset option
-	if test "$enable_compile_warnings" = "error" ; then
-	    warning_flags="$warning_flags -Werror"
-	fi
+#	if test "$enable_compile_warnings" = "error" ; then
+#	    warning_flags="$warning_flags -Werror"
+#	fi
 	;;
     *)
 	{ { echo "$as_me:$LINENO: error: Unknown argument '$enable_compile_warnings' to --enable-compile-warnings" >&5
