--- configure.orig	2004-08-17 17:22:14.243530601 +0100
+++ configure	2004-08-17 17:23:27.836122375 +0100
@@ -21923,6 +21923,7 @@
 1.7rc2) MOZILLA_SNAPSHOT=16 ;;
 1.7rc3) MOZILLA_SNAPSHOT=17 ;;
   1.7) MOZILLA_SNAPSHOT=17 ;;
+  1.7.2) MOZILLA_SNAPSHOT=17 ;;
 branch1.7) MOZILLA_SNAPSHOT=17 ;;
  1.8a) MOZILLA_SNAPSHOT=18 ;;
 1.8a1) MOZILLA_SNAPSHOT=18 ;;
@@ -21930,7 +21931,7 @@
 trunk) MOZILLA_SNAPSHOT=19 ;;
     *) { { echo "$as_me:$LINENO: error:
 *****************************************************************************
-Epiphany can be built using Mozilla 1.4, 1.4.1, 1.4.2, 1.5, 1.6 or 1.7,
+Epiphany can be built using Mozilla 1.4, 1.4.1, 1.4.2, 1.5, 1.6, 1.7 or 1.7.2,
 but the version available is $MOZILLA_VERSION.
 *****************************************************************************" >&5
 echo "$as_me: error:
