diff -NurB rhythmbox-0.6.5-orig/shell/rb-shell.c rhythmbox-0.6.5/shell/rb-shell.c
--- rhythmbox-0.6.5-orig/shell/rb-shell.c	2004-02-12 11:34:12.000000000 -0500
+++ rhythmbox-0.6.5/shell/rb-shell.c	2004-02-12 11:34:45.000000000 -0500
@@ -608,12 +608,13 @@
 			    const CORBA_char *uri,
 			    CORBA_Environment *ev)
 {
+	GnomeVFSURI *vfsuri;
 	RBShell *shell = RB_SHELL (bonobo_object (_servant));
 	RBPlaylist *parser;
 
 	rb_debug ("handling uri: %s", uri);
 
-	GnomeVFSURI *vfsuri = gnome_vfs_uri_new (uri);
+	vfsuri = gnome_vfs_uri_new (uri);
 	if (!vfsuri) {
 		rb_error_dialog (_("Unable to parse URI \"%s\"\n"), uri);
 		return;
