diff -uNr rhythmbox-0.8.7/metadata/rb-metadata-gst.c rhythmbox-0.8.7.patched/metadata/rb-metadata-gst.c
--- rhythmbox-0.8.7/metadata/rb-metadata-gst.c	2004-05-17 21:15:12.000000000 +0200
+++ rhythmbox-0.8.7.patched/metadata/rb-metadata-gst.c	2004-10-02 13:06:37.144459616 +0200
@@ -127,6 +127,7 @@
 	add_supported_type (md, "application/x-ape", NULL, "MonkeysAudio");
 	add_supported_type (md, "audio/x-mod", NULL, "MOD");
 	add_supported_type (md, "audio/mpeg", NULL, "MP3");
+	add_supported_type (md, "audio/x-shorten", NULL, "SHN");
 	
 	if ((elt = gst_element_factory_make ("faad", "faad")) != NULL) {
 		add_supported_type (md, "audio/x-m4a", NULL, "MPEG-4");
