diff -ur freecnc++.orig/src/Makefile freecnc++/src/Makefile
--- freecnc++.orig/src/Makefile	2002-07-29 08:59:44.000000000 -0400
+++ freecnc++/src/Makefile	2003-07-31 20:43:21.000000000 -0400
@@ -26,12 +26,11 @@
 
 #EFENCE = -lefence
 
-CFLAGS = $(DEBUG_FLAGS) -Werror -Wall -I./include `sdl-config --cflags`
+CFLAGS = -Wall -I./include `sdl-config --cflags` $(EXTRACFLAGS)
 LIBS = -lSDL_net
 
 # comment this out if your compiler isn't ANSI
 # needed on OSX and some gcc systems.
-CFLAGS += -ansi -pedantic
 
 # uncomment these lines if compiling on OSX
 #CFLAGS += -DDARWIN -D_WITHOUT_STRCASECMP
diff -ur freecnc++.orig/src/vfs/vfs_mix/Makefile freecnc++/src/vfs/vfs_mix/Makefile
--- freecnc++.orig/src/vfs/vfs_mix/Makefile	2002-07-29 08:59:44.000000000 -0400
+++ freecnc++/src/vfs/vfs_mix/Makefile	2003-07-31 20:42:58.000000000 -0400
@@ -8,8 +8,7 @@
 #DEBUG_FLAGS += -D_DEBUG -DSTRICT_DEBUG
 #LDFLAGS     += ../../misc/memtrack.o
 
-CFLAGS = $(DEBUG_FLAGS) -Werror -Wall -I.. -I../../include `sdl-config --cflags`
-CFLAGS += -ansi -pedantic 
+CFLAGS = -Wall -I.. -I../../include `sdl-config --cflags` $(EXTRACFLAGS)
 # uncomment these lines if compiling on OSX
 #CFLAGS += -D_WITHOUT_STRCASECMP
 #LDFLAGS += -bundle -flat_namespace -undefined suppress
diff -ur freecnc++.orig/src/vfs/vfs_tgz/Makefile freecnc++/src/vfs/vfs_tgz/Makefile
--- freecnc++.orig/src/vfs/vfs_tgz/Makefile	2002-07-29 08:59:45.000000000 -0400
+++ freecnc++/src/vfs/vfs_tgz/Makefile	2003-07-31 20:43:08.000000000 -0400
@@ -8,8 +8,7 @@
 #DEBUG_FLAGS += -D_DEBUG -DSTRICT_DEBUG
 #LDFLAGS     += ../../misc/memtrack.o
 
-CFLAGS = $(DEBUG_FLAGS) -Werror -Wall -I.. -I../../include `sdl-config --cflags`
-CFLAGS += -ansi -pedantic 
+CFLAGS = -Wall -I.. -I../../include `sdl-config --cflags` $(EXTRACFLAGS)
 # uncomment these lines if compiling on OSX
 #CFLAGS += -D_WITHOUT_STRCASECMP
 #LDFLAGS += -bundle -flat_namespace -undefined suppress
diff -ur freecnc++.orig/tools/audplay/Makefile freecnc++/tools/audplay/Makefile
--- freecnc++.orig/tools/audplay/Makefile	2002-07-20 09:20:15.000000000 -0400
+++ freecnc++/tools/audplay/Makefile	2003-07-31 20:42:05.000000000 -0400
@@ -9,10 +9,9 @@
 LOCALOBJS = $(LOCALSRC:.cpp=.o)
 OBJECTS = $(SRC:.cpp=.o)
 
-CFLAGS = -g -Werror -Wall -I. -I../../src/include `sdl-config --cflags`
+CFLAGS = -Wall -I. -I../../src/include `sdl-config --cflags` $(EXTRACFLAGS)
 
 #if your compiler is really broken and doesn't understand ANSI, comment this out
-CFLAGS += -ansi -pedantic
 
 #if your compiler is partially broken and allows the use of str(n)casecmp
 #even when compiling with -ansi -pedantic (it shouldn't), uncomment this:
diff -ur freecnc++.orig/tools/mixshell/Makefile freecnc++/tools/mixshell/Makefile
--- freecnc++.orig/tools/mixshell/Makefile	2002-07-20 09:20:15.000000000 -0400
+++ freecnc++/tools/mixshell/Makefile	2003-07-31 20:42:18.000000000 -0400
@@ -10,4 +10,4 @@
 
 .SUFFIXES: .cpp .o
 .cpp.o:
-	g++ -Wall -c $< `sdl-config --cflags`
+	g++ -Wall -c $< `sdl-config --cflags` $(EXTRACFLAGS)
diff -ur freecnc++.orig/tools/shpview/Makefile freecnc++/tools/shpview/Makefile
--- freecnc++.orig/tools/shpview/Makefile	2002-07-20 09:20:15.000000000 -0400
+++ freecnc++/tools/shpview/Makefile	2003-07-31 20:42:30.000000000 -0400
@@ -9,10 +9,9 @@
 LOCALOBJS = $(LOCALSRC:.cpp=.o)
 OBJECTS = $(SRC:.cpp=.o)
 
-CFLAGS = -g -Werror -Wall -I. -I../../src/include `sdl-config --cflags`
+CFLAGS = -Wall -I. -I../../src/include `sdl-config --cflags` $(EXTRACFLAGS)
 
 #if your compiler is really broken and doesn't understand ANSI, comment this out
-CFLAGS += -ansi -pedantic
 
 #if your compiler is partially broken and allows the use of str(n)casecmp
 #even when compiling with -ansi -pedantic (it shouldn't), uncomment this:
diff -ur freecnc++.orig/tools/tmpinied/Makefile freecnc++/tools/tmpinied/Makefile
--- freecnc++.orig/tools/tmpinied/Makefile	2002-07-20 09:20:15.000000000 -0400
+++ freecnc++/tools/tmpinied/Makefile	2003-07-31 20:42:41.000000000 -0400
@@ -7,7 +7,7 @@
 
 OBJECTS = $(SRC:.cpp=.o)
 
-CFLAGS = -g -Wall -I../../src/include `sdl-config --cflags`
+CFLAGS = -Wall -I../../src/include `sdl-config --cflags` $(EXTRACFLAGS)
 
 all: $(OBJECTS)
 	g++ -o tmpinied -g `sdl-config --libs` $(OBJECTS)
