--- firehol.sh	2004-05-03 00:51:38.000000000 +0200
+++ firehol.sh.new	2004-08-01 23:16:42.390810920 +0200
@@ -1788,7 +1788,7 @@
 	local prface="${work_inface}"
 	
 	local pre="pr"
-	unset reverse
+	local reverse=
 	if [ "${1}" = "reverse" ]
 	then
 		local reverse="reverse"	# needed to recursion
@@ -2691,7 +2691,7 @@
 				action="${2}"
 				shift 2
 				
-				unset action_param
+				local -a action_param=()
 				local action_is_chain=0
 				case "${action}" in
 					accept|ACCEPT)
@@ -3287,7 +3287,7 @@
 			local pr=
 			for pr in ${proto}
 			do
-				unset proto_arg
+				local -a proto_arg=()
 				
 				case ${pr} in
 					any|ANY)
@@ -3299,7 +3299,7 @@
 				esac
 				
 				rule_action_param "${negative_action}" "${pr}" "${action_param[@]}" -- ${table} -A "${negative_chain}" "${proto_arg[@]}"
-				unset action_param
+				local -a action_param=()
 			done
 		fi
 	fi
@@ -3311,8 +3311,8 @@
 	local tuid=
 	for tuid in ${uid}
 	do
-		unset uid_arg
-		unset owner_arg
+		local -a uid_arg=()
+		local -a owner_arg=()
 		
 		case ${tuid} in
 			any|ANY)
@@ -3327,7 +3327,7 @@
 		local tgid=
 		for tgid in ${gid}
 		do
-			unset gid_arg
+			local -a gid_arg=()
 			
 			case ${tgid} in
 				any|ANY)
@@ -3342,7 +3342,7 @@
 			local tpid=
 			for tpid in ${pid}
 			do
-				unset pid_arg
+				local -a pid_arg=()
 				
 				case ${tpid} in
 					any|ANY)
@@ -3357,7 +3357,7 @@
 				local tsid=
 				for tsid in ${sid}
 				do
-					unset sid_arg
+					local -a sid_arg=()
 					
 					case ${tsid} in
 						any|ANY)
@@ -3372,7 +3372,7 @@
 					local tcmd=
 					for tcmd in ${cmd}
 					do
-						unset cmd_arg
+						local -a cmd_arg=()
 						
 						case ${tcmd} in
 							any|ANY)
@@ -3387,7 +3387,7 @@
 						local pr=
 						for pr in ${proto}
 						do
-							unset proto_arg
+							local -a proto_arg=()
 							
 							case ${pr} in
 								any|ANY)
@@ -3401,7 +3401,7 @@
 							local inf=
 							for inf in ${inface}
 							do
-								unset inf_arg
+								local -a inf_arg=()
 								case ${inf} in
 									any|ANY)
 										;;
@@ -3414,7 +3414,7 @@
 								local outf=
 								for outf in ${outface}
 								do
-									unset outf_arg
+									local -a outf_arg=()
 									case ${outf} in
 										any|ANY)
 											;;
@@ -3427,7 +3427,7 @@
 									local inph=
 									for inph in ${physin}
 									do
-										unset inph_arg
+										local -a inph_arg=()
 										case ${inph} in
 											any|ANY)
 												;;
@@ -3441,7 +3441,7 @@
 										local outph=
 										for outph in ${physout}
 										do
-											unset outph_arg
+											local -a outph_arg=()
 											case ${outph} in
 												any|ANY)
 													;;
@@ -3455,7 +3455,7 @@
 											local sp=
 											for sp in ${sport}
 											do
-												unset sp_arg
+												local -a sp_arg=()
 												case ${sp} in
 													any|ANY)
 													;;
@@ -3468,7 +3468,7 @@
 												local dp=
 												for dp in ${dport}
 												do
-													unset dp_arg
+													local -a dp_arg=()
 													case ${dp} in
 														any|ANY)
 															;;
@@ -3481,7 +3481,7 @@
 													local mc=
 													for mc in ${mac}
 													do
-														unset mc_arg
+														local -a mc_arg=()
 														case ${mc} in
 															any|ANY)
 																;;
@@ -3494,7 +3494,7 @@
 														local s=
 														for s in ${src}
 														do
-															unset s_arg
+															local -a s_arg=()
 															case ${s} in
 																any|ANY)
 																	;;
@@ -3507,7 +3507,7 @@
 															local d=
 															for d in ${dst}
 															do
-																unset d_arg
+																local -a d_arg=()
 																case ${d} in
 																	any|ANY)
 																		;;
@@ -3517,19 +3517,19 @@
 																		;;
 																esac
 																
-																unset state_arg
+																local -a state_arg=()
 																if [ ! -z "${state}" ]
 																then
 																	local -a state_arg=("-m" "state" "${statenot}" "--state" "${state}")
 																fi
 																
-																unset limit_arg
+																local -a limit_arg=()
 																if [ ! -z "${limit}" ]
 																then
 																	local -a limit_arg=("-m" "limit" "--limit" "${limit}" "--limit-burst" "${burst}")
 																fi
 																
-																unset iplimit_arg
+																local -a iplimit_arg=()
 																if [ ! -z "${iplimit}" ]
 																then
 																	local -a iplimit_arg=("-m" "iplimit" "--iplimit-above" "${iplimit}" "--iplimit-mask" "${iplimit_mask}")
@@ -3537,7 +3537,7 @@
 																
 																declare -a basecmd=("${inf_arg[@]}" "${outf_arg[@]}" "${physdev_arg[@]}" "${inph_arg[@]}" "${outph_arg[@]}" "${limit_arg[@]}" "${iplimit_arg[@]}" "${proto_arg[@]}" "${s_arg[@]}" "${sp_arg[@]}" "${d_arg[@]}" "${dp_arg[@]}" "${owner_arg[@]}" "${uid_arg[@]}" "${gid_arg[@]}" "${pid_arg[@]}" "${sid_arg[@]}" "${cmd_arg[@]}" "${state_arg[@]}" "${mc_arg[@]}")
 																
-																unset logopts_arg
+																local -a logopts_arg=()
 																if [ "${FIREHOL_LOG_MODE}" = "ULOG" ]
 																then
 																	local -a logopts_arg=("--ulog-prefix='${logtxt}:'")
@@ -4822,10 +4822,8 @@
 			# find all the networks this IP can access directly
 			# or through its peer
 			netcount=0
-			unset ifnets
-			unset ofnets
-			set -a ifnets=
-			set -a ofnets=
+			ifnets=
+			ofnets=
 			for net in ${nets}
 			do
 				test "${net}" = "default" && continue
