--- Makefile.am.orig	2008-06-19 20:26:58.000000000 -0400
+++ Makefile.am	2008-06-19 20:28:10.000000000 -0400
@@ -26,12 +26,11 @@
 if USE_SONAME
 install-exec-hook:
 	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-	$(libdir)/libode.so
+	$(DESTDIR)$(libdir)/libode.so
 	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-	$(libdir)/@ODE_SONAME@
+	$(DESTDIR)$(libdir)/@ODE_SONAME@
 	ln -s $(libdir)/@ODE_SONAME@.@ODE_REVISION@.@ODE_AGE@ \
-	$(libdir)/@ODE_SONAME@.@ODE_REVISION@
-	/sbin/ldconfig
+	$(DESTDIR)$(libdir)/@ODE_SONAME@.@ODE_REVISION@
 else
 install-exec-hook:
 endif
