--- ode/src/Makefile.am.orig    2007-02-03 20:09:21.000000000 +0100
+++ ode/src/Makefile.am 2007-02-03 20:09:31.000000000 +0100
@@ -11,7 +11,7 @@
 # Fake an executable in order to get a shared library
 # Note the elegant and cunning way to trick Autotools to install a program
 # in a lib directory. --Rodrigo
-traplibdir=$(prefix)/lib
+traplibdir=$(prefix)/$(libdir)
 EXEEXT=@so_ext@
 traplib_PROGRAMS=libode
 libode_SOURCES=
