--- transkode-0.6.orig/src/plugins/wvunpack/wvunpackdecoderplugin.h	2006-09-10 01:02:44.000000000 +0200
+++ transkode-0.6/src/plugins/wvunpack/wvunpackdecoderplugin.h	2007-07-17 22:02:28.921011061 +0200
@@ -39,7 +39,7 @@ class WvUnpackDecoderPlugin : public Tra
 			proc.clearArguments();
 			bool pMode = proc.getShellParsingMode();
 			proc.setShellParsingMode( false );
-			proc << programPath() << "-d" << src << dst;
+			proc << programPath() << src << "-o" << dst;
 			proc.setShellParsingMode( pMode );
 		}
 
