diff -Naur xfe-0.70/src/FilePanel.cc xfe-0.70.new/src/FilePanel.cc
--- xfe-0.70/src/FilePanel.cc	2004-07-27 14:02:54.000000000 +0200
+++ xfe-0.70/src/FilePanel.cc	2004-09-02 14:04:24.032188440 +0200
@@ -1039,7 +1039,7 @@
                     {
                         // Update existing association
                         FXString oldfileassoc = getApp()->reg().readStringEntry("FILETYPES", ext.text(),"");
-						oldfileassoc.remove(0, oldfileassoc.section(';',0).section(',',0).length()-1);
+						oldfileassoc.remove(0, oldfileassoc.section(';',0).section(',',0).length());
                         oldfileassoc.prepend(dialog->getText());
                         getApp()->reg().writeStringEntry("FILETYPES", ext.text(), oldfileassoc.text());
 
