--- elfloader.c	2003-09-21 15:08:36.000000000 -0500
+++ elfloader-ia64	2003-09-21 15:12:07.000000000 -0500
@@ -2180,6 +2180,7 @@
 	    ia64_flush_cache(dest64);
 	    break;
 
+#if 0
 	case R_IA64_PCREL64LSB:
 	    dest64=(unsigned long *)(secp+rel->r_offset);
 #ifdef ELFDEBUG
@@ -2191,6 +2192,8 @@
 #endif
 	    *dest64 = symval + rel->r_addend - (unsigned long)dest64;
 	    break;
+
+#endif
 	
 	case R_IA64_PCREL64LSB:
 	    dest64=(unsigned long *)(secp+rel->r_offset);
