--- startkde.orig	Fri Jun 28 09:34:00 2002
+++ startkde	Fri Jun 28 09:35:45 2002
@@ -1,8 +1,20 @@
-#!/bin/sh
+#!/bin/sh --login
 #
-#  DEFAULT KDE STARTUP SCRIPT ( KDE-3.0.2 )
+#  Gentoo KDE Startup Script ( KDE-3.0.2 )
 #
 
+# Gentoo part:
+export PATH="/usr/kde/3/bin:${PATH}"
+export KDEDIR="/usr/kde/3"
+export KDEDIRS="/usr/kde/3:${KDEDIRS}"
+export KDEMAINDIR="/usr/kde/3"
+rm -rf ~/.kde
+ln -fs ~/.kde3 ~/.kde
+[ -e "~/.kde3" ] || mkdir ~/.kde3
+
+
+# Original KDE part:
+
 # Boot sequence:
 #
 # kdeinit is used to fork off processes which improves memory usage
@@ -30,14 +42,14 @@
 # people's heads. We use colours from the standard KDE palette for those with
 # palettised displays.
 
-test "$XDM_MANAGED" || bkg="-solid #C0C0C0"
+test "$XDM_MANAGED" || bkg="-solid #5477A0"
 xsetroot -cursor_name left_ptr $bkg
 
 # The user's personal KDE directory is usually ~/.kde, but this setting
 # may be overridden by setting KDEHOME.
 
 kdehome=$HOME/.kde
-test -n "$KDEHOME" && kdehome=`echo "$KDEHOME"|sed "s,^\~/,$HOME/,"`
+test -n "$KDEHOME" && eval kdehome=$KDEHOME
 
 # Activate the kde font directories.
 #
