--- alsa-kernel/include/emu10k1.h.old	2005-10-24 10:50:25.000000000 +0200
+++ alsa-kernel/include/emu10k1.h	2005-10-24 10:49:06.000000000 +0200
@@ -48,7 +48,7 @@
 
 /* FIXME? - according to the OSS driver the EMU10K1 needs a 29 bit DMA mask */
 #define EMU10K1_DMA_MASK	0x7fffffffUL	/* 31bit */
-#define AUDIGY_DMA_MASK		0xffffffffUL	/* 32bit */
+#define AUDIGY_DMA_MASK		0x7fffffffUL	/* 32bit */
 
 #define TMEMSIZE        256*1024
 #define TMEMSIZEREG     4
