Index: libjingle-0.3.12/talk/base/Makefile.am
===================================================================
--- libjingle-0.3.12.orig/talk/base/Makefile.am
+++ libjingle-0.3.12/talk/base/Makefile.am
@@ -31,7 +31,7 @@ libjinglebase_la_SOURCES =  socketaddres
                             signalthread.cc \
                             socketpool.cc
 
-libjinglebase_la_LIBADD = -lrt
+libjinglebase_la_LIBADD = -lrt -lssl -lcrypto
 
 libjinglebase_includedir = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/base
 libjinglebase_include_HEADERS = asyncfile.h \
Index: libjingle-0.3.12/talk/p2p/base/Makefile.am
===================================================================
--- libjingle-0.3.12.orig/talk/p2p/base/Makefile.am
+++ libjingle-0.3.12/talk/p2p/base/Makefile.am
@@ -38,7 +38,7 @@ libjinglep2pbase_include_HEADERS = candi
 								   sessionresponsemessage.h \
 								   sessionmessagefactory.h
 
-libjinglep2pbase_la_LIBADD = -lssl -lcrypto
+libjinglep2pbase_la_LIBADD =
 
 AM_CPPFLAGS = -DPOSIX
 
