diff -ur xfce4-settings-4.6.3.orig/configure.ac xfce4-settings-4.6.3/configure.ac
--- xfce4-settings-4.6.3.orig/configure.ac	2009-09-29 01:50:13.000000000 +0300
+++ xfce4-settings-4.6.3/configure.ac	2009-10-24 19:25:21.000000000 +0300
@@ -77,7 +77,8 @@
 dnl ***********************************
 dnl *** Check for required packages ***
 dnl ***********************************
-XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.100])
+XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.100], [],
+  [XDT_CHECK_PACKAGE([EXO], [exo-1], [0.5.0])])
 XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.10.0])
 XDT_CHECK_PACKAGE([GLIB], [glib-2.0], [2.12.0])
 XDT_CHECK_PACKAGE([GTHREAD], [gthread-2.0], [2.12.0])
