Index: bossogg-0.13.6/boss3/bossao/flac.c
===================================================================
--- bossogg-0.13.6.orig/boss3/bossao/flac.c
+++ bossogg-0.13.6/boss3/bossao/flac.c
@@ -93,7 +93,7 @@ static void error_callback (const FLAC__
 void *prepare_flac (song_s *song, char *filename)
 {
   //song->songlib->flac = malloc (sizeof (FLAC__FileDecoder));
-  (FLAC__FileDecoder *)song->songlib->flac = FLAC__file_decoder_new ();
+  song->songlib->flac = FLAC__file_decoder_new ();
   FLAC__FileDecoder *decoder = (FLAC__FileDecoder *)song->songlib->flac;
   
   /* callbacks here? */
@@ -108,7 +108,7 @@ void *prepare_flac (song_s *song, char *
   FLAC__SeekableStreamDecoderState state = FLAC__file_decoder_init (decoder);
   //printf ("inited\n");
   if (state != FLAC__FILE_DECODER_OK) {
-    printf ("Problem initizlizing FLAC file decoder: ", state);
+    printf ("Problem initizlizing FLAC file decoder: %d", state);
     if (state == FLAC__FILE_DECODER_ALREADY_INITIALIZED)
       printf ("already inited\n");
     if (state == FLAC__FILE_DECODER_SEEKABLE_STREAM_DECODER_ERROR)
