--- w32api-3.1/lib/directx/Makefile.in.jj	2004-10-25 09:19:50.005482368 -0700
+++ w32api-3.1/lib/directx/Makefile.in	2004-10-25 09:26:41.038995776 -0700
@@ -69,10 +69,10 @@
 EXTRA_INCLUDES = -I$(srcdir)/../../include -I$(srcdir)/../../../newlib/libc/include -I$(srcdir)/../../../newlib/libc/sys/cygwin
 endif
 ifeq ($(BUILDENV), mingw)
-EXTRA_INCLUDES = -I$(srcdir)/../../mingw/include
+EXTRA_INCLUDES = -I$(srcdir)/../../../mingw/include
 endif
 
-INCLUDES = -I$(srcdir)/../include -I$(srcdir)/../include/directx $(EXTRA_INCLUDES)
+INCLUDES = -I$(srcdir)/../../include -I$(srcdir)/../../include/directx $(EXTRA_INCLUDES)
 
 CFLAGS = @CFLAGS@
 ALL_CFLAGS = $(CFLAGS) $(INCLUDES)
