--- ../CherryPy-2.0.0b.orig/setup.py	2004-12-30 11:13:56.000000000 +0100
+++ ./setup.py	2005-02-08 16:06:46.007214608 +0100
@@ -22,7 +22,7 @@
       author_email="team@cherrypy.org",
       url="http://www.cherrypy.org",
       license="BSD",
-      packages=["cherrypy", "cherrypy.lib", "cherrypy.lib.filter", "cherrypy.tutorial", "cherrypy.test"],
+      packages=["cherrypy", "cherrypy.lib", "cherrypy.lib.filter"],
       download_url="http://trac.cherrypy.org/cgi-bin/trac.cgi/wiki/CherryPyDownload"
 )
 
