--- app_iconv-0.9.2/Makefile.orig	2006-04-15 23:02:21.000000000 +0000
+++ app_iconv-0.9.2/Makefile	2006-04-15 23:03:43.000000000 +0000
@@ -8,9 +8,9 @@
 
 NAME=app_iconv
 
-CFLAGS=-g -Wall -D_REENTRANT -D_GNU_SOURCE -fPIC
+CFLAGS+=-g -Wall -D_REENTRANT -D_GNU_SOURCE -fPIC
 LIBS=
-RES=$(shell if [ -f /usr/include/asterisk/channel.h ]; then echo "$(NAME).so"; fi)
+RES=$(NAME).so
 
 ifeq (${OSARCH},Darwin)
 SOLINK=-dynamic -bundle -undefined suppress -force_flat_namespace
@@ -29,9 +29,7 @@
 	rm -f $(NAME).o $(NAME).so
 
 install: all
-	if [ -f $(NAME).so ]; then \
-		install -m 755 $(NAME).so $(MODULES_DIR) ; \
-	fi
+	install -D -m 755 $(NAME).so $(DESTDIR)$(MODULES_DIR)/$(NAME).so ; \
 
 reload: install
 	asterisk -rx "unload ${RES}"
