--- wmdiscotux.c.old	2000-11-16 09:51:52.000000000 -0500
+++ wmdiscotux.c	2003-09-15 16:03:54.000000000 -0400
@@ -359,16 +359,7 @@
         gtk_container_border_width(GTK_CONTAINER(dialog), 5);
 
 
-        label = gtk_label_new ("
-wmDiscoTux\n\
-v " VERSION "\n\
-
-by Antti Takala <fragment@nic.fi>\n\
-
-get the newest version from:\n\
-http://wmdiscotux.stc.cx/\n\
-
-");
+        label = gtk_label_new ("wmDiscoTux\nv " VERSION "\nby Antti Takala <fragment@nic.fi>\nget the newest version from:\nhttp://wmdiscotux.stc.cx/\n");
 
         gtk_box_pack_start(GTK_BOX(GTK_DIALOG(dialog)->vbox), label, FALSE, FALSE, 0);
         gtk_widget_show(label);
