--- Makefile.old	2000-11-15 13:33:43.000000000 -0500
+++ Makefile	2003-08-16 22:16:19.000000000 -0400
@@ -1,14 +1,17 @@
+
 CC = gcc
-INCLUDEDIR = -I/usr/include -I/usr/X11R6/include -I/usr/local/include
-CFLAGS =   -O2 -Wall $(INCLUDEDIR) -fPIC `glib-config --cflags`
-LFLAGS =   -shared -fPIC -lpthread -L/usr/lib -ldl -lm -L/usr/X11R6/lib -lXpm -lX11
+OPT = -O2
+INCLUDEDIR = -I/usr/include -I/usr/X11R6/include -I/usr/local/include `gtk-config --cflags` `glib-config --cflags`
+CFLAGS = $(OPT) -Wall $(INCLUDEDIR) -fPIC
+LFLAGS = -shared -fPIC -lpthread -L/usr/lib -ldl -lm -L/usr/X11R6/lib -lXpm -lX11
+INSTALL-DIR = `xmms-config --visualization-plugin-dir`
 
 libwmdiscotux.so: wmdiscotux.o wmgeneral.o
-	$(CC)  -o libwmdiscotux.so wmdiscotux.o wmgeneral.o $(LFLAGS)
+	$(CC) -o libwmdiscotux.so wmdiscotux.o wmgeneral.o $(LFLAGS)
 
 clean:
-	rm -f *.o core *.so* 
+	rm -f *.o core *.so*
 
 install: libwmdiscotux.so
-	cp libwmdiscotux.so `xmms-config --visualization-plugin-dir`
-
+	mkdir -p $(DESTDIR)/$(INSTALL-DIR)
+	install libwmdiscotux.so $(DESTDIR)/$(INSTALL-DIR)
