diff -ur memtest86-3.0/config.c memtest86-3.0.m/config.c
--- memtest86-3.0/config.c	Tue May 21 13:27:11 2002
+++ memtest86-3.0.m/config.c	Fri Jul 26 00:59:07 2002
@@ -10,7 +10,6 @@
 
 extern int bail;
 extern struct tseq tseq[];
-extern struct vars *v;
 extern short e820_nr;
 extern char memsz_mode;
 
diff -ur memtest86-3.0/init.c memtest86-3.0.m/init.c
--- memtest86-3.0/init.c	Tue May 21 17:52:37 2002
+++ memtest86-3.0.m/init.c	Fri Jul 26 00:58:42 2002
@@ -11,7 +11,6 @@
 #include "pci.h"
 #include "io.h"
 
-extern struct vars *v;
 extern short memsz_mode;
 extern short firmware;
 
diff -ur memtest86-3.0/lib.c memtest86-3.0.m/lib.c
--- memtest86-3.0/lib.c	Tue May 21 18:12:07 2002
+++ memtest86-3.0.m/lib.c	Fri Jul 26 00:58:49 2002
@@ -9,7 +9,6 @@
 #include "config.h"
 #include "screen_buffer.h"
 
-extern struct vars *v;
 
 int slock = 0, lsr = 0;
 short serial_cons = SERIAL_CONSOLE_DEFAULT;
diff -ur memtest86-3.0/patn.c memtest86-3.0.m/patn.c
--- memtest86-3.0/patn.c	Wed Oct 17 18:30:32 2001
+++ memtest86-3.0.m/patn.c	Fri Jul 26 00:58:58 2002
@@ -18,7 +18,6 @@
 #define DEFAULT_MASK ((~0L) << 2)
 
 
-extern struct vars *v;
 
 
 /* What it does:
diff -ur memtest86-3.0/test.c memtest86-3.0.m/test.c
--- memtest86-3.0/test.c	Tue May 21 18:13:32 2002
+++ memtest86-3.0.m/test.c	Fri Jul 26 00:58:33 2002
@@ -9,7 +9,6 @@
 extern int segs, bail;
 extern volatile ulong *p;
 extern ulong p1, p2;
-extern struct vars *v;
 extern int test_ticks, nticks;
 extern struct tseq tseq[];
 void poll_errors();
