diff -Nru freeradius-server-2.0.5.orig/src/main/event.c freeradius-server-2.0.5/src/main/event.c
--- freeradius-server-2.0.5.orig/src/main/event.c	2008-06-06 15:10:27.000000000 +0200
+++ freeradius-server-2.0.5/src/main/event.c	2008-07-04 20:31:18.000000000 +0200
@@ -982,7 +982,7 @@
 			break;
 		}
 
-#if defined(HAVE_PTHREAD_H) || defined(WITH_PROXY)
+#if defined(HAVE_PTHREAD_H)
 		/*
 		 *	A child thread MAY still be running on the
 		 *	request.  Ask the thread to stop working on
