diff -Nru freeradius-server-2.0.5.orig/src/lib/Makefile freeradius-server-2.0.5/src/lib/Makefile
--- freeradius-server-2.0.5.orig/src/lib/Makefile	2008-04-20 17:18:39.000000000 +0200
+++ freeradius-server-2.0.5/src/lib/Makefile	2008-07-01 21:17:25.000000000 +0200
@@ -53,5 +53,3 @@
 	$(INSTALL) -d -m 755	$(R)$(libdir)
 	$(LIBTOOL) --mode=install $(INSTALL) -c $(TARGET).la \
 		$(R)$(libdir)/$(TARGET).la
-	rm -f $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la;
-	ln -s $(TARGET).la $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la
diff -Nru freeradius-server-2.0.5.orig/src/modules/rlm_eap/libeap/Makefile freeradius-server-2.0.5/src/modules/rlm_eap/libeap/Makefile
--- freeradius-server-2.0.5.orig/src/modules/rlm_eap/libeap/Makefile	2007-11-25 14:56:10.000000000 +0100
+++ freeradius-server-2.0.5/src/modules/rlm_eap/libeap/Makefile	2008-07-01 21:17:25.000000000 +0200
@@ -40,5 +40,3 @@
 install: all
 	$(LIBTOOL) --mode=install $(INSTALL) -c $(TARGET).la \
 		$(R)$(libdir)/$(TARGET).la
-	rm -f $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la;
-	ln -s $(TARGET).la $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la
diff -Nru freeradius-server-2.0.5.orig/src/modules/rules.mak freeradius-server-2.0.5/src/modules/rules.mak
--- freeradius-server-2.0.5.orig/src/modules/rules.mak	2008-01-10 11:12:01.000000000 +0100
+++ freeradius-server-2.0.5/src/modules/rules.mak	2008-07-01 21:17:25.000000000 +0200
@@ -163,6 +163,4 @@
 	if [ "x$(TARGET)" != "x" ]; then \
 	    $(LIBTOOL) --mode=install $(INSTALL) -c \
 		$(TARGET).la $(R)$(libdir)/$(TARGET).la || exit $$?; \
-	    rm -f $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la; \
-	    ln -s $(TARGET).la $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la || exit $$?; \
 	fi
