diff -Nru freeradius-server-2.1.3.orig/configure.in freeradius-server-2.1.3/configure.in
--- freeradius-server-2.1.3.orig/configure.in	2008-12-05 16:37:56.000000000 +0000
+++ freeradius-server-2.1.3/configure.in	2009-03-05 22:25:53.000000000 +0000
@@ -544,7 +544,19 @@
 	],
 	[ AC_MSG_WARN([pcap library not found, silently disabling the RADIUS sniffer.]) ])
 
-AC_LIB_READLINE
+AC_CHECK_LIB(readline, readline,
+	[ LIBREADLINE="-lreadline"
+	AC_DEFINE(HAVE_LIBREADLINE, 1,
+		[Define to 1 if you have a readline compatible library.])
+	AC_DEFINE(HAVE_READLINE_READLINE_H, 1,
+		[Define to 1 if you have the <readline/readline.h> header file.])
+	AC_DEFINE(HAVE_READLINE_HISTORY, 1,
+		[Define if your readline library has \`add_history'])
+	AC_DEFINE(HAVE_READLINE_HISTORY_H, 1,
+		[Define to 1 if you have the <readline/history.h> header file.])
+	],
+	[ LIBREADLINE="" ])
+AC_SUBST(LIBREADLINE)
 
 dnl #############################################################
 dnl #
diff -Nru freeradius-server-2.1.3.orig/src/lib/Makefile freeradius-server-2.1.3/src/lib/Makefile
--- freeradius-server-2.1.3.orig/src/lib/Makefile	2009-03-05 22:24:10.000000000 +0000
+++ freeradius-server-2.1.3/src/lib/Makefile	2009-03-05 22:25:53.000000000 +0000
@@ -41,7 +41,7 @@
 
 $(TARGET).la: $(LT_OBJS)
 	$(LIBTOOL) --mode=link $(CC) -release $(RADIUSD_VERSION) \
-	$(LDFLAGS) $(LINK_MODE) -o $@ -rpath $(libdir) $^
+	$(LDFLAGS) $(LINK_MODE) -o $@ -rpath $(libdir) $^ -lpthread
 
 $(LT_OBJS): $(INCLUDES)
 
diff -Nru freeradius-server-2.1.3.orig/src/main/listen.c freeradius-server-2.1.3/src/main/listen.c
--- freeradius-server-2.1.3.orig/src/main/listen.c	2008-12-05 16:37:56.000000000 +0000
+++ freeradius-server-2.1.3/src/main/listen.c	2009-03-05 22:25:53.000000000 +0000
@@ -45,6 +45,9 @@
 #include <fcntl.h>
 #endif
 
+#ifdef WITH_UDPFROMTO
+#include <freeradius-devel/udpfromto.h>
+#endif
 
 /*
  *	We'll use this below.
