diff -Nru freeradius-1.1.6.orig/configure.in freeradius-1.1.6/configure.in
--- freeradius-1.1.6.orig/configure.in	2007-04-04 12:23:57.000000000 +0300
+++ freeradius-1.1.6/configure.in	2007-04-12 21:33:37.000000000 +0300
@@ -803,7 +803,7 @@
 else
   devflags=""
   CFLAGS="$CFLAGS -DNDEBUG"
-  INSTALLSTRIP="-s"
+  INSTALLSTRIP=""
   AC_MSG_RESULT(no.)
 fi
 
diff -Nru freeradius-1.1.6.orig/src/main/mainconfig.c freeradius-1.1.6/src/main/mainconfig.c
--- freeradius-1.1.6.orig/src/main/mainconfig.c	2007-04-08 01:06:08.000000000 +0300
+++ freeradius-1.1.6/src/main/mainconfig.c	2007-04-12 21:33:44.000000000 +0300
@@ -23,6 +23,7 @@
 
 #include "autoconf.h"
 #include "libradius.h"
+#include "ltdl.h"
 
 #include <stdlib.h>
 #include <string.h>
@@ -1501,6 +1502,10 @@
 /*
  *	Free the configuration.
  */
+extern void rl_free(void);
+extern void paircompare_builtin_free(void);
+extern void xlat_free(void);
+
 int free_mainconfig(void)
 {
 	/*
diff -Nru freeradius-1.1.6.orig/src/modules/rlm_ldap/rlm_ldap.c freeradius-1.1.6/src/modules/rlm_ldap/rlm_ldap.c
--- freeradius-1.1.6.orig/src/modules/rlm_ldap/rlm_ldap.c	2007-04-08 10:14:45.000000000 +0300
+++ freeradius-1.1.6/src/modules/rlm_ldap/rlm_ldap.c	2007-04-12 21:33:44.000000000 +0300
@@ -175,6 +175,8 @@
 #include	<string.h>
 
 #include	<lber.h>
+#define LDAP_DEPRECATED 1
+#undef HAVE_LDAP_INT_TLS_CONFIG
 #include        <ldap.h>
 
 #include	<errno.h>
diff -Nru freeradius-1.1.6.orig/src/modules/rlm_smb/smblib.c freeradius-1.1.6/src/modules/rlm_smb/smblib.c
--- freeradius-1.1.6.orig/src/modules/rlm_smb/smblib.c	2004-02-26 21:04:35.000000000 +0200
+++ freeradius-1.1.6/src/modules/rlm_smb/smblib.c	2007-04-12 21:33:44.000000000 +0300
@@ -23,6 +23,8 @@
    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
 
+#include <string.h>
+
 int SMBlib_errno;
 int SMBlib_SMB_Error;
 #define SMBLIB_ERRNO
diff -Nru freeradius-1.1.6.orig/src/modules/rlm_smb/smblib-util.c freeradius-1.1.6/src/modules/rlm_smb/smblib-util.c
--- freeradius-1.1.6.orig/src/modules/rlm_smb/smblib-util.c	2004-02-26 21:04:35.000000000 +0200
+++ freeradius-1.1.6/src/modules/rlm_smb/smblib-util.c	2007-04-12 21:33:44.000000000 +0300
@@ -23,6 +23,8 @@
    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
 
+#include <string.h>
+
 #include "smblib-priv.h"
 
 #include "rfcnb.h"
