diff -Nru freeradius-1.0.5.orig/aclocal.m4 freeradius-1.0.5/aclocal.m4
--- freeradius-1.0.5.orig/aclocal.m4	2005-08-19 20:29:04.000000000 +0300
+++ freeradius-1.0.5/aclocal.m4	2005-09-24 20:20:47.733693672 +0300
@@ -1421,7 +1421,7 @@
     *)
       # ancient GNU ld didn't support --whole-archive et. al.
       if $LD --help 2>&1 | egrep 'no-whole-archive' > /dev/null; then
-	whole_archive_flag_spec="$wlarc"'--whole-archive$convenience '"$wlarc"'--no-whole-archive'
+	whole_archive_flag_spec="$wlarc"'--no-whole-archive'
       else
 	whole_archive_flag_spec=
       fi
diff -Nru freeradius-1.0.5.orig/configure freeradius-1.0.5/configure
--- freeradius-1.0.5.orig/configure	2005-09-24 20:20:08.000000000 +0300
+++ freeradius-1.0.5/configure	2005-09-24 20:20:47.737693064 +0300
@@ -3511,7 +3511,7 @@
     *)
       # ancient GNU ld didn't support --whole-archive et. al.
       if $LD --help 2>&1 | egrep 'no-whole-archive' > /dev/null; then
-	whole_archive_flag_spec="$wlarc"'--whole-archive$convenience '"$wlarc"'--no-whole-archive'
+	whole_archive_flag_spec="$wlarc"'--no-whole-archive'
       else
 	whole_archive_flag_spec=
       fi
diff -Nru freeradius-1.0.5.orig/ltconfig freeradius-1.0.5/ltconfig
--- freeradius-1.0.5.orig/ltconfig	2003-06-24 14:03:39.000000000 +0300
+++ freeradius-1.0.5/ltconfig	2005-09-24 20:20:47.739692760 +0300
@@ -1246,7 +1246,7 @@
     *)
       # ancient GNU ld didn't support --whole-archive et. al.
       if $LD --help 2>&1 | egrep 'no-whole-archive' > /dev/null; then
-        whole_archive_flag_spec="$wlarc"'--whole-archive$convenience '"$wlarc"'--no-whole-archive'
+        whole_archive_flag_spec="$wlarc"'--no-whole-archive'
       else
         whole_archive_flag_spec=
       fi
