--- trunk/configure.ac	2008/03/21 16:15:50	167
+++ trunk/configure.ac	2008/03/26 08:08:52	168
@@ -50,6 +50,13 @@
 fi
 AM_CONDITIONAL(EXTERNAL_DBUS, test "x$have_external_dbus" = "xyes")
 
+PKG_CHECK_MODULES(NOTIFY_SHARP, notify-sharp, have_external_notify_sharp="yes", have_external_notify_sharp="no")
+if test "x$have_external_notify_sharp" != "xyes"; then
+	AC_MSG_RESULT([using internal copy])
+fi
+AM_CONDITIONAL(EXTERNAL_NOTIFY_SHARP, test "x$have_external_notify_sharp" = "xyes")
+
+
 AC_PATH_PROG(MCS, gmcs)
 if test x$MCS = x; then
 	AC_MSG_ERROR(You need mcs)
--- trunk/src/Makefile.am	2008/03/21 16:15:50	167
+++ trunk/src/Makefile.am	2008/03/26 08:08:52	168
@@ -72,8 +72,12 @@
 XMLRPC_ASSEMBLIES = 				\
 	-r:${DIR_XMLRPC}/XmlRpcCS.dll
 
+if EXTERNAL_NOTIFY_SHARP
+NOTIFY_ASSEMBLIES = $(NOTIFY_SHARP_LIBS)
+else
 NOTIFY_ASSEMBLIES =				\
 	-r:$(DIR_NOTIFY)/notify-sharp.dll
+endif
 
 LASTEXIT_RESOURCES =				\
 	-resource:$(top_srcdir)/data/glade/FindStation.glade	\
--- trunk/notify-sharp/Makefile.am	2008/03/21 16:15:50	167
+++ trunk/notify-sharp/Makefile.am	2008/03/26 08:08:52	168
@@ -1,13 +1,13 @@
 NOTIFY_ASSEMBLY = notify-sharp.dll
+NOTIFY_SRC = \
+		$(srcdir)/Global.cs \
+		$(srcdir)/Notification.cs
 
+if !EXTERNAL_NOTIFY_SHARP
 notifydir = $(pkglibdir)
 notify_SCRIPTS = \
 	$(NOTIFY_ASSEMBLY)
 
-NOTIFY_SRC = 				\
-		$(srcdir)/Global.cs \
-		$(srcdir)/Notification.cs
-
 if EXTERNAL_DBUS
 DBUS_ASSEMBLIES = $(DBUS_LIBS)
 else
@@ -21,6 +21,8 @@
 $(NOTIFY_ASSEMBLY): $(NOTIFY_SRC)
 	$(MCS) -target:library -out:$@ $(DBUS_ASSEMBLIES) $(GTKSHARP_LIBS) $(NOTIFY_SRC)
 
+endif
+
 EXTRA_DIST = $(NOTIFY_SRC)
 DISTCLEANFILES = $(NOTIFY_ASSEMBLY) $(NOTIFY_ASSEMBLY).mdb
 MAINTAINERCLEANFILES = Makefile.in
