--- symtab.c.orig	2002-11-07 17:32:05.000000000 +0100
+++ tools/sources/symtab.c	2002-11-07 17:33:56.000000000 +0100
@@ -23,6 +23,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <math.h>             /* for MAXFLOAT */
+#include <values.h>
 
 #include "symtab.ph"
 #include "arglist.h"
--- ui.h.orig	2002-11-07 17:37:06.000000000 +0100
+++ xgui/sources/ui.h	2002-11-07 17:37:44.000000000 +0100
@@ -21,8 +21,8 @@
 
 ******************************************************************************/
 
-#ifndef __ui
-#define __ui
+#ifndef snns_ui
+#define snns_ui
 
 /* X11 */
 
--- ui_config.c.orig	2002-11-07 18:35:43.000000000 +0100
+++ xgui/sources/ui_config.c	2002-11-07 18:37:34.000000000 +0100
@@ -492,7 +492,7 @@
 	err = fscanf(filePtr,"%*[^:]: %d\n", &noParams);
 	chkretline(1);
 	lines++;
-	for (i=0; (i<UI_NO_LEARN_PARAMS+UI_NO_OF_CASCADE_PARAMS); i++){
+	for (i=0; (i<UI_NO_LEARN_PARAMS+UI_NO_OF_CASCADE_PARAMS) && (i<noParams); i++){
 	    err = fscanf(filePtr,"%*[^:]: %f\n", &ui_learnParameters[i]);
 	    chkretline(1);
 	    lines++;
