diff -Nru weatherng-0.0.5/i18n.c weatherng-0.0.5-neu/i18n.c
--- weatherng-0.0.5/i18n.c	2005-06-18 14:58:43.000000000 +0200
+++ weatherng-0.0.5-neu/i18n.c	2005-10-11 19:14:03.000000000 +0200
@@ -1198,7 +1198,7 @@
     "¿àØÒÕâ!",
   },
   { "Partly Cloudy",
-    "",
+    "teilweise bewölkt",
     "",// TODO
     "",// TODO
     "",// TODO
diff -Nru weatherng-0.0.5/img/bitmap.c weatherng-0.0.5-neu/img/bitmap.c
--- weatherng-0.0.5/img/bitmap.c	2005-06-07 04:56:21.000000000 +0200
+++ weatherng-0.0.5-neu/img/bitmap.c	2005-10-11 19:14:03.000000000 +0200
@@ -1,8 +1,9 @@
 #include <string>
 #include <vector>
-#include <Magick++.h>
-#include <vdr/tools.h>
 #include <vdr/osd.h>
+#include <vdr/tools.h>
+#include <Magick++.h>
+
 using namespace Magick;
 
 cBitmap* LoadMagick(const char *Filename, int height, int width, int colors, bool Quiet) {
diff -Nru weatherng-0.0.5/OsdWeather.c weatherng-0.0.5-neu/OsdWeather.c
--- weatherng-0.0.5/OsdWeather.c	2005-06-18 14:58:43.000000000 +0200
+++ weatherng-0.0.5-neu/OsdWeather.c	2005-10-11 19:14:30.000000000 +0200
@@ -217,7 +217,7 @@
 	sat_file = sat_file + "/daten.dat";
 	BOOL ex = if_exist_file(sat_file.c_str());
 	
-		int MAXHOEHE = 480;
+		int MAXHOEHE = 430;
 	 	int MAXBREITE = 624;
 		int HALBEHOEHE= (Setup.OSDHeight/2)+1;
 		int HALBEBREITE= (Setup.OSDWidth/2)+1;	
@@ -273,19 +273,11 @@
 			printf("Information about: %s\n",parser.ort.c_str());
 
 			  ausgabe = tr("Weather for: ");
-			  ausgabe = ausgabe + parser.ort.c_str();
+			  ausgabe = ausgabe + parser.ort.c_str() + "  /  ";
+			  ausgabe = ausgabe + tr(parser.dayname.c_str());
+			  ausgabe = ausgabe + "  (" + parser.date + ")";
 			osd->DrawText(8,HALBEHOEHE+4,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
 
-
-//			  ausgabe = replaceDay(parser.dayname.c_str()) + " ("+  parser.date.c_str() +")";
-			  ausgabe = parser.dayname;
-//			  ausgabe = tr(ausgabe);
-			osd->DrawText(HALBEBREITE+70,HALBEHOEHE+4,tr(ausgabe.c_str()), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
-
-		          ausgabe = "(" + parser.date + ")";
-    			osd->DrawText(HALBEBREITE+144,HALBEHOEHE+4,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
-
-
 			osd->DrawText(8,HALBEHOEHE+40,tr("High temperature: "), wetterTheme[wetterSetup.w_theme].clrFgHiColor,clrTransparent,font);
 		          ausgabe = parser.hi.c_str();
 		          ausgabe= ausgabe + tr("°C");
@@ -313,22 +305,22 @@
 			//right Side
 			osd->DrawText(HALBEBREITE+15,HALBEHOEHE+40,tr("Wind comes from: "), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
 			  ausgabe = parser.winddir.c_str();
-			osd->DrawText(HALBEBREITE+144,HALBEHOEHE+40,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
+			osd->DrawText(Setup.OSDWidth - CELLWIDTH -6 ,HALBEHOEHE+40,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
 
 			osd->DrawText(HALBEBREITE+15,HALBEHOEHE+70,tr("Wind speed: "), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
 			  ausgabe = parser.windspeed.c_str();
 			  ausgabe= ausgabe + tr(" km/h");
-			osd->DrawText(HALBEBREITE+144,HALBEHOEHE+70,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
+			osd->DrawText(Setup.OSDWidth - CELLWIDTH -6 ,HALBEHOEHE+70,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
 
 			osd->DrawText(HALBEBREITE+15,HALBEHOEHE+100,tr("Humitidy: "), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
 			  ausgabe = parser.humidity.c_str();
 			  ausgabe= ausgabe + tr(" %");
-			osd->DrawText(HALBEBREITE+144,HALBEHOEHE+100,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
+			osd->DrawText(Setup.OSDWidth - CELLWIDTH -6 ,HALBEHOEHE+100,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
 
 			osd->DrawText(HALBEBREITE+15,HALBEHOEHE+130,tr("Precipitation: "), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
 			  ausgabe = parser.raindown.c_str();
 			  ausgabe= ausgabe + tr(" %");
-			osd->DrawText(HALBEBREITE+144,HALBEHOEHE+130,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
+			osd->DrawText(Setup.OSDWidth - CELLWIDTH -6 ,HALBEHOEHE+130,ausgabe.c_str(), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font,CELLWIDTH,20,taRight);
 
 
                      // BORDER
@@ -360,7 +352,7 @@
 			}
 
 			if (day<10){
-				osd->DrawText(HALBEBREITE+ 150,MAXHOEHE-29,tr("Day forward -->"), wetterTheme[wetterSetup.w_theme].clrBackground,clrTransparent,font,CELLWIDTH,20,taRight);
+				osd->DrawText(Setup.OSDWidth - CELLWIDTH -6 ,MAXHOEHE-29,tr("Day forward -->"), wetterTheme[wetterSetup.w_theme].clrBackground,clrTransparent,font,CELLWIDTH,20,taRight);
 //				osd->DrawText((Setup.OSDWidth)-5,MAXHOEHE-29,tr("Day forward -->"), wetterTheme[wetterSetup.w_theme].clrBackground,clrTransparent,font,CELLWIDTH,20,taRight);
 			}
 
@@ -373,7 +365,7 @@
 		else
 	         {
 			osd->DrawText(20,HALBEHOEHE+10,tr("ERROR : NO DATA"), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
-			osd->DrawText(20,HALBEHOEHE+50,tr("Check path and that you are in offlinemode"), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
+			osd->DrawText(20,HALBEHOEHE+50,tr("Check path and setup ,that you are not in offlinemode"), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
 			osd->DrawText(20,HALBEHOEHE+80,tr("Your path to 'daten.dat' is:"), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
 			osd->DrawText(20,HALBEHOEHE+110,tr(DataDir), wetterTheme[wetterSetup.w_theme].clrFgHiColor,clrTransparent,font);
 			osd->DrawText(20,HALBEHOEHE+140,tr("Restart VDR with '--loglevel 3' and watch stdout"), wetterTheme[wetterSetup.w_theme].clrFgText,clrTransparent,font);
diff -Nru weatherng-0.0.5/parsing.c weatherng-0.0.5-neu/parsing.c
--- weatherng-0.0.5/parsing.c	2005-06-18 22:04:29.000000000 +0200
+++ weatherng-0.0.5-neu/parsing.c	2005-10-11 19:14:03.000000000 +0200
@@ -12,19 +12,19 @@
 
 
 #define TAG_ERR_REMAIN(_context) do { \
-                esyslog("ERROR: Weatherng: Unexpected tag %s within %s", \
+                esyslog("ERROR: Text2Skin: Unexpected tag %s within %s", \
                                 name.c_str(), _context); \
                 return false; \
         } while (0)
 
 #define TAG_ERR_CHILD(_context) do { \
-                esyslog("ERROR:  Weatherng: No child tag %s expected within %s", \
+                esyslog("ERROR: Text2Skin: No child tag %s expected within %s", \
                                 name.c_str(), _context); \
                 return false; \
         } while (0)
 
 #define TAG_ERR_END(_context) do { \
-                esyslog("ERROR:  Weatherng: Unexpected closing tag for %s within %s", \
+                esyslog("ERROR: Text2Skin: Unexpected closing tag for %s within %s", \
                                 name.c_str(), _context); \
                 return false; \
         } while (0)
@@ -36,7 +36,7 @@
 #define ATTRIB_MAN_STRING(_attr,_target) \
         ATTRIB_OPT_STRING(_attr,_target) \
         else { \
-                esyslog("ERROR:  Weatherng: Mandatory attribute %s missing in tag %s", \
+                esyslog("ERROR: Text2Skin: Mandatory attribute %s missing in tag %s", \
                                 _attr, name.c_str()); \
                 return false; \
         }
@@ -44,7 +44,7 @@
 #define ATTRIB_MAN_NUMBER(_attr,_target) \
         ATTRIB_OPT_NUMBER(_attr,_target) \
         else { \
-                esyslog("ERROR:  Weatherng: Mandatory attribute %s missing in tag %s", \
+                esyslog("ERROR: Text2Skin: Mandatory attribute %s missing in tag %s", \
                                 _attr, name.c_str()); \
                 return false; \
         }
@@ -52,7 +52,7 @@
 #define ATTRIB_OPT_FUNC(_attr,_func) \
         if (attrs.find(_attr) != attrs.end()) { \
                 if (!_func(attrs[_attr])) { \
-                        esyslog("ERROR:  Weatherng: Unexpected value %s for attribute %s", \
+                        esyslog("ERROR: Text2Skin: Unexpected value %s for attribute %s", \
                                         attrs[_attr].c_str(), _attr); \
                         return false; \
                 } \
@@ -61,7 +61,7 @@
 #define ATTRIB_MAN_FUNC(_attr,_func) \
         ATTRIB_OPT_FUNC(_attr,_func) \
         else { \
-                esyslog("ERROR: Weatherng: Mandatory attribute %s missing in tag %s", \
+                esyslog("ERROR: Text2Skin: Mandatory attribute %s missing in tag %s", \
                                 _attr, name.c_str()); \
                 return false; \
         }
diff -Nru weatherng-0.0.5/README.DE weatherng-0.0.5-neu/README.DE
--- weatherng-0.0.5/README.DE	2005-06-18 15:47:42.000000000 +0200
+++ weatherng-0.0.5-neu/README.DE	2005-10-11 19:14:03.000000000 +0200
@@ -98,7 +98,8 @@
 somit nicht mehr in die GPL (siehe permissions.txt im images Verzeichnis).
 
 
-Im Setup Menü müsst ihr lediglich die ID eingeben, die kann man wie folg herausfinden.
+Dann muss man seine Einstellungen im Setup machen.
+Die eigene ID kann man folgendermassen herausfinden:
 
 Ruft die Seite http://www.weather.com auf. 
 
@@ -106,14 +107,15 @@
 
 Nach dem Suchen bekommt Ihr ein Link zur eurer Stadt, in der URL steht ein Code,
 der wie folgt aufgebaut ist 
-
 GMXX0056 (Angabe für den Ort Herne)
-
 wobei 0056 ausgetauscht wird, mit der Nummer die für den jeweiligen Ort angegeben wurde.
 
 Im Offlinemodus wird nicht aufs Internet zugegriffen.
-somit ist es moeglich die Daten auf anderen Rechner zu besorgen und
+Somit ist es moeglich die Daten auf anderen Rechner zu besorgen und
 auf einen internetfreien VDR zu kopieren.
+Standardmaessig ist der Offlinemodus aktiviert und es werden keine Daten heruntergeladen.
+Also sollten keine Daten angezeigt werden , zuerst diesen Punkt ueberpruefen.
+Der Fehler macht sich bemerkbar durch eine fehlende Datei "daten.dat".
 
 
 ____________________________________________
diff -Nru weatherng-0.0.5/setup.c weatherng-0.0.5-neu/setup.c
--- weatherng-0.0.5/setup.c	2005-06-18 05:45:04.000000000 +0200
+++ weatherng-0.0.5-neu/setup.c	2005-10-11 19:14:03.000000000 +0200
@@ -6,7 +6,7 @@
 
 cWetterSetup::cWetterSetup(void)
 {
-	strcpy(stationId, "GMXX0049");
+	strcpy(stationId, "GMXX0056");
 	w_left       = 150;
 	w_top        = 120;
 	w_width      = 420;
@@ -50,9 +50,9 @@
   },    
   { // Enigma OK
     0xB8BFC9E6, //clrBackground
-    0xFFC4C400, //clrFgHiColor
-    0xFF4158BC, //clrFgLowColor
+    0xFFFF4040, //clrFgHiColor
+    0xFF000000, //clrFgLowColor
 //    0xB84158BC, //clrBorder
-    0xFFFFFFFF, //clrFgText + Border
+    0xB84E528F, //clrFgText + Border
   },    
 }; 
diff -Nru weatherng-0.0.5/Tools/SatDownload weatherng-0.0.5-neu/Tools/SatDownload
--- weatherng-0.0.5/Tools/SatDownload	2005-06-13 15:49:48.000000000 +0200
+++ weatherng-0.0.5-neu/Tools/SatDownload	2005-10-11 19:15:09.000000000 +0200
@@ -34,7 +34,7 @@
 # DANN FOLGENDES IN commands.conf EINTRAGEN:
 #   Wetter aktualisieren : at -f "/dein_pfad_zu_SatDownload/SatDownload" now
 
-cd /VDR/etc/plugins/weatherng;
+cd /var/vdr/weatherng;
 
           wget --tries=4 --timeout=20 -O overall.wet "$URL"
           wget --tries=4 --timeout=20 -O pic1.wet $URL1
diff -Nru weatherng-0.0.5/weatherng.c weatherng-0.0.5-neu/weatherng.c
--- weatherng-0.0.5/weatherng.c	2005-06-14 21:57:51.000000000 +0200
+++ weatherng-0.0.5-neu/weatherng.c	2005-10-11 19:14:03.000000000 +0200
@@ -111,33 +111,23 @@
 {
   themes[eWetterThemeClassic]     = tr("Classic");
   themes[eWetterThemeenElchi]     = tr("enElchi");
-  themes[eWetterThemeMoronimo]     = tr("Moronimo");
-  themes[eWetterThemeDeepBlue]     = tr("DeepBlue");
-  themes[eWetterThemeEnigma]     = tr("Enigma");
+  themes[eWetterThemeMoronimo]    = tr("Moronimo");
+  themes[eWetterThemeDeepBlue]    = tr("DeepBlue");
+  themes[eWetterThemeEnigma]      = tr("Enigma");
+
+  Add(new cMenuEditStrItem(tr("Station ID"),                  wetterSetup.stationId, 9, ALLOWED_STATION_CHARS));
+  Add(new cMenuEditIntItem(tr("Radar left"),                  &wetterSetup.w_left, 1, 200));
+  Add(new cMenuEditIntItem(tr("Radar top"),                   &wetterSetup.w_top, 1, 200));
+  Add(new cMenuEditIntItem(tr("Radar width"),                 &wetterSetup.w_width, 100, 450));
+  Add(new cMenuEditIntItem(tr("Radar height"),                &wetterSetup.w_height, 100, 400));
+  Add(new cMenuEditBoolItem(tr("256 Colordepth (radarmaps)"), &wetterSetup.w_rcolor));
+  Add(new cMenuEditBoolItem(tr("Use small fonts"),            &wetterSetup.w_fontsize));
+  Add(new cMenuEditBoolItem(tr("Use offlinemode"),            &wetterSetup.w_offline));
+  Add(new cMenuEditStraItem(tr("Theme"),                      &wetterSetup.w_theme, eWetterThemeMaxNumber, themes));
 
-  Setup();
 }    
 
 
-void cMenuWetterSetup::Setup(void)
-{
-    int current = Current();
-
-        Clear();
-	Add(new cMenuEditStrItem(tr("Station ID"),                   wetterSetup.stationId, 9, ALLOWED_STATION_CHARS));
-	Add(new cMenuEditIntItem(tr("Radar left"),                  &wetterSetup.w_left, 1, 200));
-	Add(new cMenuEditIntItem(tr("Radar top"),                   &wetterSetup.w_top, 1, 200));
-	Add(new cMenuEditIntItem(tr("Radar width"),                 &wetterSetup.w_width, 100, 450));
-	Add(new cMenuEditIntItem(tr("Radar height"),                &wetterSetup.w_height, 100, 400));
-	Add(new cMenuEditBoolItem(tr("256 Colordepth (radarmaps)"), &wetterSetup.w_rcolor));
-	Add(new cMenuEditBoolItem(tr("Use small fonts"),            &wetterSetup.w_fontsize));
-	Add(new cMenuEditBoolItem(tr("Use offlinemode"),            &wetterSetup.w_offline));
-	Add(new cMenuEditStraItem(tr("Theme"),                      &wetterSetup.w_theme, eWetterThemeMaxNumber, themes));
-
-  SetCurrent(Get(current));
-  Display();	
-}
-
 void cMenuWetterSetup::Store(void)
 {
 	for(char *ptr = wetterSetup.stationId; *ptr; ++ptr) {
@@ -157,13 +147,5 @@
 	SetupStore("Theme",           wetterSetup.w_theme);
 	}
 
-eOSState cMenuWetterSetup::ProcessKey(eKeys Key)
-{
- eOSState state = cMenuSetupPage::ProcessKey(Key);
- if (Key != kNone) {
-       Setup();
-       }
- return state;
-}
         
 VDRPLUGINCREATOR(cPluginWetter); // Don't touch this!
diff -Nru weatherng-0.0.5/weatherng.h weatherng-0.0.5-neu/weatherng.h
--- weatherng-0.0.5/weatherng.h	2005-06-14 22:04:14.000000000 +0200
+++ weatherng-0.0.5-neu/weatherng.h	2005-10-11 19:14:03.000000000 +0200
@@ -14,8 +14,7 @@
 static const char *DESCRIPTION    = "Weather forecast";
 static const char *MAINMENUENTRY  = "Weather-NG";
 
-static const char *ALLOWED_STATION_CHARS 	= "abcdefghijklmnopqrstuvwxyz0123456789";
-//static const char hex[] = { "0123456789abcdef" };
+static const char *ALLOWED_STATION_CHARS = tr("abcdefghijklmnopqrstuvwxyz0123456789");
 
 class cPluginWetter : public cPlugin {
 private:
@@ -41,9 +40,7 @@
 class cMenuWetterSetup : public cMenuSetupPage {
 	private:
                 const char *themes[eWetterThemeMaxNumber];
-		virtual void Setup(void);
 	protected:
-	        virtual eOSState ProcessKey(eKeys Key);
 		virtual void Store(void);
 	public:
   		cMenuWetterSetup(void);
