diff -Naur kita2-1.90.4.orig/config.rb kita2-1.90.4/config.rb
--- kita2-1.90.4.orig/config.rb	2008-09-22 23:38:14.000000000 +0900
+++ kita2-1.90.4/config.rb	2009-01-01 01:40:20.000000000 +0900
@@ -21,7 +21,7 @@
 
 require 'singleton'
 
-class Config
+class KitaConfig
   include Singleton
 
   def initialize
diff -Naur kita2-1.90.4.orig/mainwindow.rb kita2-1.90.4/mainwindow.rb
--- kita2-1.90.4.orig/mainwindow.rb	2008-04-24 22:52:46.000000000 +0900
+++ kita2-1.90.4/mainwindow.rb	2009-01-01 01:40:20.000000000 +0900
@@ -53,7 +53,7 @@
 
     statusBar.show
 
-    if Config.instance.auto_login
+    if KitaConfig.instance.auto_login
       login
     end
   end
@@ -90,7 +90,7 @@
   end
 
   def login
-    config = Config.instance
+    config = KitaConfig.instance
     user_id = config.user_id
     password = config.password
 
diff -Naur kita2-1.90.4.orig/preferencedialog.rb kita2-1.90.4/preferencedialog.rb
--- kita2-1.90.4.orig/preferencedialog.rb	2008-09-22 23:38:14.000000000 +0900
+++ kita2-1.90.4/preferencedialog.rb	2009-01-01 01:40:20.000000000 +0900
@@ -42,7 +42,7 @@
   end
 
   def load_config
-    config = Config.instance
+    config = KitaConfig.instance
     @face_page.thread_font_requester.setFont(config.thread_font)
     @account_page.id_line.text = config.user_id
     @account_page.password_line.text = config.password
@@ -61,7 +61,7 @@
     kotehan = @write_page.kotehan_edit.text.split("\n").grep(/^.+$/)
     thread_font = @face_page.thread_font_requester.font
 
-    config = Config.instance
+    config = KitaConfig.instance
     config.thread_font = thread_font
     config.user_id = id
     config.password = password
diff -Naur kita2-1.90.4.orig/threadview.rb kita2-1.90.4/threadview.rb
--- kita2-1.90.4.orig/threadview.rb	2008-09-22 23:38:14.000000000 +0900
+++ kita2-1.90.4/threadview.rb	2009-01-01 01:40:20.000000000 +0900
@@ -92,7 +92,7 @@
   end
 
   def write(dat)
-    config = Config.instance
+    config = KitaConfig.instance
     @html_part.begin()
     @html_part.write("<html><head><style type=\"text/css\">")
     @html_part.write("body { font-size: #{config.thread_font.pointSize()}pt;" +
diff -Naur kita2-1.90.4.orig/writedialog.rb kita2-1.90.4/writedialog.rb
--- kita2-1.90.4.orig/writedialog.rb	2008-09-17 22:59:42.000000000 +0900
+++ kita2-1.90.4/writedialog.rb	2009-01-01 01:40:20.000000000 +0900
@@ -43,14 +43,14 @@
     index_path = Util.dat_url_to_index_path(@dat_url)
     config = KDE::Config.new(index_path)
     names = config.readListEntry("Names")
-    kotehan = Config.instance.kotehan
+    kotehan = KitaConfig.instance.kotehan
 
     @name_combo.clear
-    @name_combo.insertItem(Config.instance.default_name)
+    @name_combo.insertItem(KitaConfig.instance.default_name)
     @name_combo.insertStringList(kotehan)
     @name_combo.insertStringList(names)
 
-    @mail_line.text = Config.instance.default_mail
+    @mail_line.text = KitaConfig.instance.default_mail
     @body_edit.text = ""
     @sage_checkbox.setChecked(false)
     @ok_button.setEnabled(false)
