--- libxslt-1.0.7/libxslt/xslt.h.orig	Fri Nov 16 13:31:46 2001
+++ libxslt-1.0.7/libxslt/xslt.h	Fri Nov 16 13:31:51 2001
@@ -44,6 +44,15 @@
 #define XSLT_NAMESPACE ((xmlChar *) "http://www.w3.org/1999/XSL/Transform")
 
 /**
+ * LIBXSLT_PUBLIC:
+ *
+ * Macro used on Windows to tag public identifiers from shared libraries
+**/
+#ifndef LIBXSLT_PUBLIC
+#define LIBXSLT_PUBLIC
+#endif
+
+/**
  * xsltMaxDepth:
  *
  * This value is used to detect templates loops
