diff -rc mysql++-1.7.9/Makefile.am mysql++-1.7.9-patched/Makefile.am
*** mysql++-1.7.9/Makefile.am	Tue Dec  3 18:29:44 2002
--- mysql++-1.7.9-patched/Makefile.am	Tue Dec  3 18:29:54 2002
***************
*** 1,4 ****
  EXTRA_DIST = .version Configure doc devel
  
  #SUBDIRS = sqlplusint examples doc
! SUBDIRS = sqlplusint examples
--- 1,4 ----
  EXTRA_DIST = .version Configure doc devel
  
  #SUBDIRS = sqlplusint examples doc
! SUBDIRS = sqlplusint
diff -rc mysql++-1.7.9/Makefile.in mysql++-1.7.9-patched/Makefile.in
*** mysql++-1.7.9/Makefile.in	Tue Dec  3 18:29:44 2002
--- mysql++-1.7.9-patched/Makefile.in	Tue Dec  3 18:30:06 2002
***************
*** 79,86 ****
  
  EXTRA_DIST = .version Configure doc devel
  
! #SUBDIRS = sqlplusint examples doc
! SUBDIRS = sqlplusint examples
  ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
  mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
  CONFIG_HEADER = config.h ./sqlplusint/exceptions.hh
--- 79,86 ----
  
  EXTRA_DIST = .version Configure doc devel
  
! #SUBDIRS = sqlplusint doc
! SUBDIRS = sqlplusint
  ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
  mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
  CONFIG_HEADER = config.h ./sqlplusint/exceptions.hh
