*** mysql++-1.7.9/sqlplusint/type_info1.hh	Sat May 19 20:44:48 2001
--- mysql++-1.7.9-patched/sqlplusint/type_info1.hh	Sat May  1 16:08:18 2004
***************
*** 28,34 ****
    const bool           _default;
    mysql_ti_sql_type_info (const mysql_ti_sql_type_info &b);           // can't do
    mysql_ti_sql_type_info& operator=(const mysql_ti_sql_type_info &b); //  "    "
!   mysql_ti_sql_type_info () {} 
    // OEP - didn't init _base_type and _default mysql_ti_sql_type_info () {} 
    // all private, only mysql_type_info can
    // create because there *must* be only one copy
--- 28,34 ----
    const bool           _default;
    mysql_ti_sql_type_info (const mysql_ti_sql_type_info &b);           // can't do
    mysql_ti_sql_type_info& operator=(const mysql_ti_sql_type_info &b); //  "    "
!   mysql_ti_sql_type_info () : _base_type(0), _default(false) {} 
    // OEP - didn't init _base_type and _default mysql_ti_sql_type_info () {} 
    // all private, only mysql_type_info can
    // create because there *must* be only one copy
