diff -NurdB libgda-2.99.5/gda-sharp/Gda.metadata libgda-2.99.5-patched/gda-sharp/Gda.metadata
--- libgda-2.99.5/gda-sharp/Gda.metadata	2007-02-12 06:01:51.000000000 -0500
+++ libgda-2.99.5-patched/gda-sharp/Gda.metadata	2007-02-13 01:09:42.000000000 -0500
@@ -4,6 +4,7 @@
   <attr path="/api/namespace/class[@cname='GType_']" name="hidden">1</attr>
   <attr path="/api/namespace/class[@name='Value']" name="hidden">1</attr>
   <attr path="/api/namespace/interface[@cname='GdaDataModel']" name="name">IGdaDataModel</attr>
+  <attr path="/api/namespace/interface[@cname='GdaDataModel']/signal[@name='Reset']" name="name">DataReset</attr>
   <attr path="/api/namespace/interface[@cname='GdaEntity']" name="name">IEntity</attr>
   <attr path="/api/namespace/interface[@cname='GdaXmlStorage']" name="name">IXmlStorage</attr>
   <attr path="/api/namespace/interface[@cname='GdaDataHandler']" name="name">IDataHandler</attr>
