diff -NurdB libgda-1.2.2/libgda/gda-log.c libgda-1.2.2-patched/libgda/gda-log.c
--- libgda-1.2.2/libgda/gda-log.c	2005-10-26 04:08:14.000000000 -0500
+++ libgda-1.2.2-patched/libgda/gda-log.c	2005-10-26 04:07:15.000000000 -0500
@@ -111,7 +111,7 @@
 #ifdef LIBGDA_WIN32
 	g_log ("Gda", G_LOG_LEVEL_INFO, "%s", msg);
 #else
-	syslog (LOG_USER | LOG_INFO, msg);
+	syslog (LOG_USER | LOG_INFO, "%s", msg);
 #endif
 	g_free (msg);
 }
@@ -144,7 +144,7 @@
 #ifdef LIBGDA_WIN32
 	g_log ("Gda", G_LOG_LEVEL_ERROR, "%s", msg);
 #else
-	syslog (LOG_USER | LOG_ERR, msg);
+	syslog (LOG_USER | LOG_ERR, "%s", msg);
 #endif
  	g_free (msg);
 }
