From 308a341ae545d991e0ad8f18e08050df393edbfa Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Wed, 29 Sep 2010 22:39:19 +0200
Subject: [PATCH] Fix json/libcrypto automagic

---
 configure.ac |   41 +++++++++++++++++++++++++++++------------
 1 files changed, 29 insertions(+), 12 deletions(-)

diff --git a/configure.ac b/configure.ac
index df31138..5abaa31 100644
--- a/configure.ac
+++ b/configure.ac
@@ -403,13 +403,21 @@ dnl ******************************
 dnl Checks for json-glib
 dnl ******************************
 
+AC_ARG_ENABLE([json],
+	AS_HELP_STRING([--disable-json], [Enable support for JSON]),
+	,
+	[enable_json=yes])
+
 JSON_GLIB_MODULES="json-glib-1.0"
-PKG_CHECK_MODULES(JSON_GLIB, $JSON_GLIB_MODULES, have_json=yes, have_json=no)
-AM_CONDITIONAL(HAVE_JSON_GLIB, test x"$have_json" = "xyes")
-if test x"$have_json" = "xyes"
-then
-        JSON_GLIB_CFLAGS="$JSON_GLIB_CFLAGS -DHAVE_JSON_GLIB"
-fi
+PKG_CHECK_MODULES(JSON_GLIB, $JSON_GLIB_MODULES,
+	AC_DEFINE(HAVE_JSON_GLIB, [], [Defined if JSON support is enabled])
+	have_json=yes,
+	if test "$enable_json" = "yes"; then
+		AC_MSG_ERROR([JSON support requested but not found])
+	fi
+	have_json=no
+)
+AM_CONDITIONAL(HAVE_JSON_GLIB, [test "$have_json" = "yes"])
 AC_SUBST(JSON_GLIB_CFLAGS)
 AC_SUBST(JSON_GLIB_LIBS)
 
@@ -422,13 +430,21 @@ dnl ******************************
 dnl Checks for libcrypto
 dnl ******************************
 
+AC_ARG_ENABLE([crypto],
+	AS_HELP_STRING([--disable-crypto], [Enable cryptographic support for SQL]),
+	,
+	[enable_crypto=yes])
+
 LIBCRYPTO_MODULES="libcrypto"
-PKG_CHECK_MODULES(LIBCRYPTO, $LIBCRYPTO_MODULES, have_crypto=yes, have_crypto=no)
-AM_CONDITIONAL(HAVE_LIBCRYPTO, test x"$have_crypto" = "xyes")
-if test x"$have_crypto" = "xyes"
-then
-        LIBCRYPTO_CFLAGS="$LIBCRYPTO_CFLAGS -DHAVE_LIBCRYPTO"
-fi
+PKG_CHECK_MODULES(LIBCRYPTO, $LIBCRYPTO_MODULES,
+	AC_DEFINE(HAVE_LIBCRYPTO, [], [Defined if cryptographic support is enabled])
+	have_crypto=yes,
+	if test "$enable_crypto" = "yes"; then
+		AC_MSG_ERROR([libcrypto support requested but not found])
+	fi
+	have_crypto=no
+)
+AM_CONDITIONAL(HAVE_LIBCRYPTO, [test "$have_crypto" = "yes"])
 AC_SUBST(LIBCRYPTO_CFLAGS)
 AC_SUBST(LIBCRYPTO_LIBS)
 
@@ -1832,6 +1848,7 @@ dnl echo "      mSQL = `if test x$msqldir != x; then echo yes; else echo no; fi`
 dnl echo "      ODBC = `if test x$odbcdir != x; then echo yes; else echo no; fi`"
 echo "      Oracle = `if test x$oracledir != x; then echo yes; else echo no; fi`"
 echo "      PostgreSQL = `if test x$postgresdir != x; then echo yes; else echo no; fi`"
+echo "      JSON = `if test x$have_json != xyes; then echo no; else echo yes; fi`"
 echo "      SQLite = yes `if test x$have_sqlite = xyes; then echo '(from system installation)'; else echo '(embedded)'; fi`"
 echo "      SQLCipher = `if test x$have_crypto != xyes; then echo no; else echo yes; fi`"
 dnl echo "      Sybase = `if test x$sybasedir != x; then echo yes; else echo no; fi`"
-- 
1.7.3

