Index: tools/gda-config.5
===================================================================
--- tools/gda-config.5	(revision 2809)
+++ tools/gda-config.5	(working copy)
@@ -4,7 +4,7 @@
 .\" Author:
 .\"   Gonzalo Paniagua Javier (gonzalo@gnome-db.org)
 .\"
-.TH gda-config "5" "January 2003" "libgda" "File Formats"
+.TH gda-config-3.0 "5" "January 2003" "libgda" "File Formats"
 .SH NAME
 $(sysconfdir)/libgda/config \- libgda global configuration file.
 $(HOME)/.libgda/config \- libgda user configuration file.
@@ -43,10 +43,10 @@
 .fi
 .sp
 .RE
-You can also use \fBgda-config-tool\fR or \fBmergeant\fR (recommended) to modify
+You can also use \fBgda-config-tool-3.0\fR or \fBmergeant\fR (recommended) to modify
 this file.
 .SH SEE ALSO
 .BR mergeant (1),
-.BR gda-config-tool (1)
+.BR gda-config-tool-3.0 (1)
 
 
Index: tools/gda-config-tool.1
===================================================================
--- tools/gda-config-tool.1	(revision 2809)
+++ tools/gda-config-tool.1	(working copy)
@@ -4,14 +4,14 @@
 .\" Author:
 .\"   Gonzalo Paniagua Javier (gonzalo@gnome-db.org)
 .\"
-.TH gda-config-tool "1" "October 2002" "gda-config-tool (libgda)" "User Commands"
+.TH gda-config-tool-3.0 "1" "October 2002" "gda-config-tool-3.0 (libgda)" "User Commands"
 .SH NAME
-gda-config-tool \- Tool for managing libgda configuration file.
+gda-config-tool-3.0 \- Tool for managing libgda configuration file.
 .SH SYNOPSIS
 .PP
-.B gda-config-tool [OPTIONS]
+.B gda-config-tool-3.0 [OPTIONS]
 .SH DESCRIPTION
-The \fIgda-config-tool\fP program can be used create/edit/remove a libgda
+The \fIgda-config-tool-3.0\fP program can be used create/edit/remove a libgda
 provider. It can be used both as a command line program and an interactive
 program.
 .SH OPTIONS
@@ -58,6 +58,6 @@
 User settings. Override global settings.
 .SH SEE ALSO
 .BR mergeant(1),
-.BR gda-config(5)
+.BR gda-config-3.0(5)
 
 
Index: tools/Makefile.am
===================================================================
--- tools/Makefile.am	(revision 2809)
+++ tools/Makefile.am	(working copy)
@@ -1,18 +1,24 @@
 if BUILD_CONFIG_TOOL
-CONFIG_TOOL=gda-config-tool
-man_MANS = gda-config-tool.1 \
-           gda-config.5
+CONFIG_TOOL=gda-config-tool-3.0
+man_MANS = gda-config-tool-3.0.1 \
+           gda-config-3.0.5
 else
 CONFIG_TOOL=
 endif
 
 if BUILD_RUN_TOOL
-RUN_TOOL=gda-run
+RUN_TOOL=gda-run-3.0
 else
 RUN_TOOL=
 endif
 
-bin_PROGRAMS = $(RUN_TOOL) $(CONFIG_TOOL) gda-diagnose gda-list-config gda-inspect-dict-file gda-author-dict-file
+bin_PROGRAMS = \
+	$(RUN_TOOL) \
+	$(CONFIG_TOOL) \
+	gda-diagnose-3.0 \
+	gda-list-config-3.0 \
+	gda-inspect-dict-file-3.0 \
+	gda-author-dict-file-3.0
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir) \
@@ -20,16 +26,16 @@
 	-I. \
 	$(LIBGDA_CFLAGS)
 
-gda_run_SOURCES = gda-run.c
-gda_run_LDADD = \
+gda_run_3_0_SOURCES = gda-run.c
+gda_run_3_0_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
 	$(top_builddir)/libgda/graph/libgda_graph-3.0.la \
         $(top_builddir)/libgda/handlers/libgda_handlers-3.0.la \
         $(top_builddir)/libgda/sql-delimiter/libgda_sql_delimiter-3.0.la \
 	$(LIBGDA_LIBS)
 
-gda_config_tool_SOURCES = gda-config-tool.c
-gda_config_tool_LDADD = \
+gda_config_tool_3_0_SOURCES = gda-config-tool.c
+gda_config_tool_3_0_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
 	$(top_builddir)/libgda/graph/libgda_graph-3.0.la \
 	$(top_builddir)/libgda/handlers/libgda_handlers-3.0.la \
@@ -39,46 +45,46 @@
 	$(POPT_LIB) \
 	$(LIBGDA_LIBS)
 
-gda_diagnose_SOURCES = \
+gda_diagnose_3_0_SOURCES = \
         html.h \
         html.c \
         gda-diagnose.c
 
-gda_diagnose_LDADD = \
+gda_diagnose_3_0_LDADD = \
         $(top_builddir)/libgda/libgda-3.0.la \
         $(top_builddir)/libgda/handlers/libgda_handlers-3.0.la \
         $(top_builddir)/libgda/graph/libgda_graph-3.0.la \
         $(top_builddir)/libgda/sql-delimiter/libgda_sql_delimiter-3.0.la \
         $(LIBGDA_LIBS)
 
-gda_inspect_dict_file_SOURCES = \
+gda_inspect_dict_file_3_0_SOURCES = \
         gda-inspect-dict-file.c
 
-gda_inspect_dict_file_LDADD = \
+gda_inspect_dict_file_3_0_LDADD = \
         $(top_builddir)/libgda/libgda-3.0.la \
         $(top_builddir)/libgda/handlers/libgda_handlers-3.0.la \
         $(top_builddir)/libgda/graph/libgda_graph-3.0.la \
         $(top_builddir)/libgda/sql-delimiter/libgda_sql_delimiter-3.0.la \
         $(LIBGDA_LIBS)
 
-gda_author_dict_file_SOURCES = \
+gda_author_dict_file_3_0_SOURCES = \
         gda-author-dict-file.c
 
-gda_author_dict_file_LDADD = \
+gda_author_dict_file_3_0_LDADD = \
         $(top_builddir)/libgda/libgda-3.0.la \
         $(top_builddir)/libgda/handlers/libgda_handlers-3.0.la \
         $(top_builddir)/libgda/graph/libgda_graph-3.0.la \
         $(top_builddir)/libgda/sql-delimiter/libgda_sql_delimiter-3.0.la \
         $(LIBGDA_LIBS)
 
-gda_list_config_SOURCES = \
+gda_list_config_3_0_SOURCES = \
         gda-list-config.c
 
-gda_list_config_LDADD = \
+gda_list_config_3_0_LDADD = \
         $(top_builddir)/libgda/libgda-3.0.la \
         $(top_builddir)/libgda/handlers/libgda_handlers-3.0.la \
         $(top_builddir)/libgda/graph/libgda_graph-3.0.la \
         $(top_builddir)/libgda/sql-delimiter/libgda_sql_delimiter-3.0.la \
         $(LIBGDA_LIBS)
 
-EXTRA_DIST = gda-config-tool.1 gda-config.5
+EXTRA_DIST = $(man_MANS)
Index: tools/gda-config-tool.c
===================================================================
--- tools/gda-config-tool.c	(revision 2809)
+++ tools/gda-config-tool.c	(working copy)
@@ -1476,6 +1476,8 @@
 main (int argc, char *argv [])
 {
 	options (argc, (const char **) argv);
+	gda_init ("Gda configuration manager", PACKAGE_VERSION, argc, argv);
+
 	if (cmdArgs.actions == ACTION_NONE) {
 		g_print ("Using configuration file from %s\n", cmdArgs.config_file);
 		initialize_readline ();
Index: doc/C/installation.xml
===================================================================
--- doc/C/installation.xml	(revision 2809)
+++ doc/C/installation.xml	(working copy)
@@ -176,11 +176,11 @@
     <title>Compiling with the library</title>
     <para>
       To compile you will need to set the C flags and to link the library, so we recommend
-      to use the <emphasis>pkg-config</emphasis> command.
+      to use the <command>pkg-config</command> command.
     </para>
     <programlisting>
-      <systemitem class="prompt">$</systemitem> <userinput> gcc -c full_example.c `pkg-config --cflags libgda`</userinput>
-      <systemitem class="prompt">$</systemitem> <userinput> gcc -o full_example `pkg-config --libs libgda` full_example.o</userinput>
+<systemitem class="prompt">$</systemitem> <userinput> gcc -c full_example.c `pkg-config --cflags libgda-3.0`</userinput>
+<systemitem class="prompt">$</systemitem> <userinput> gcc -o full_example `pkg-config --libs libgda-3.0` full_example.o</userinput>
     </programlisting>
     <para>
       Doing this will add to the C compiler command line all needed header files
Index: doc/C/libgda-docs.sgml
===================================================================
--- doc/C/libgda-docs.sgml	(revision 2809)
+++ doc/C/libgda-docs.sgml	(working copy)
@@ -29,6 +29,11 @@
 <!ENTITY GDADATAMODEL    "<link linkend='GdaDataModel'>GdaDataModel</link>">
 <!ENTITY GDADATAMODELARRAY "<link linkend='GdaDataModelArray'>GdaDataModelArray</link>">
 <!ENTITY GDADATAMODELHASH "<link linkend='GdaDataModelHash'>GdaDataModelHash</link>">
+<!ENTITY gda-author-dict-file "<command>gda-author-dict-file-3.0</command>">
+<!ENTITY gda-diagnose "<command>gda-diagnose-3.0</command>">
+<!ENTITY gda-inspect-dict-file "<command>gda-inspect-dict-file-3.0</command>">
+<!ENTITY gda-list-config "<command>gda-list-config-3.0</command>">
+<!ENTITY gda-test-connection "<command>gda-test-connection-3.0</command>">
 <!ENTITY fullexample SYSTEM "examples/full_example.c">
 <!ENTITY install SYSTEM "installation.xml">
 <!ENTITY examples SYSTEM "gettingstarted.xml">
@@ -130,7 +135,7 @@
 
 <book id="index">
   <bookinfo>
-    <title>GNOME Data Access manual</title>
+    <title>GNOME Data Access 3.0 manual</title>
     <authorgroup>
       <author>
         <firstname>Michael</firstname>
@@ -197,7 +202,7 @@
     </authorgroup>
     <date>1999 February</date>
     <copyright>
-      <year>1999 - 2006</year>
+      <year>1999 - 2007</year>
       <holder>The GNOME Foundation</holder>
     </copyright>
     <abstract>
@@ -638,15 +643,15 @@
     </chapter>
 
     <chapter id="libgda-tools-list-config">
-      <title>gda-list-config</title>
+      <title>&gda-list-config;</title>
       <para>
-	The gda-list-config tool simply lists all the declared data sources, and all the installed
+	The &gda-list-config; tool simply lists all the declared data sources, and all the installed
 	providers, giving some other usefull information as well for each.
       </para>
     </chapter>
 
     <chapter id="libgda-tools-test-connection">
-      <title>gda-test-connection</title>
+      <title>&gda-test-connection;</title>
       <para>
 	This tool is a 'ping' like tool to test connectivity to databases. Connections can either
 	be declared as DSN, or described in the command line.
@@ -654,9 +659,9 @@
     </chapter>
 
     <chapter id="libgda-tools-diagnose">
-      <title>gda-diagnose</title>
+      <title>&gda-diagnose;</title>
       <para>
-	The gda-diagnose tool is a quite exhaustive and verbose testing program which output 
+	The &gda-diagnose; tool is a quite exhaustive and verbose testing program which output 
 	several HTML files
 	in a specified directory (DIAGNOSE_OUTPUT by default). It can either test a single data source
 	or all the data sources if none is specified. Running this tool can be time consuming...
@@ -675,9 +680,9 @@
     </chapter>
 
     <chapter id="libgda-tools-author-dict">
-      <title>gda-author-dict-file</title>
+      <title>&gda-author-dict-file;</title>
       <para>
-	The gda-author-dict-file tool allows to create a XML dictionary file which can be loaded into a 
+	The &gda-author-dict-file; tool allows to create a XML dictionary file which can be loaded into a 
 	#GdaDict object, specifying a data source to use. Specifically it does (depending on command line
 	options):
 	<itemizedlist>
@@ -693,9 +698,9 @@
     </chapter>
 
     <chapter id="libgda-tools-inspect-dict">
-      <title>gda-inspect-dict-file</title>
+      <title>&gda-inspect-dict-file;</title>
       <para>
-	The gda-inspect-dict-file tool tests an existing XML dictionary file and (depending on command line
+	The &gda-inspect-dict-file; tool tests an existing XML dictionary file and (depending on command line
 	options):
 	<itemizedlist>
 	  <listitem>
Index: doc/C/Makefile.am
===================================================================
--- doc/C/Makefile.am	(revision 2809)
+++ doc/C/Makefile.am	(working copy)
@@ -3,7 +3,7 @@
 AUTOMAKE_OPTIONS = 1.6
 
 # The name of the module.
-DOC_MODULE=libgda
+DOC_MODULE=libgda-3.0
 
 # The top-level SGML file.
 DOC_MAIN_SGML_FILE=$(DOC_MODULE)-docs.sgml
@@ -37,7 +37,7 @@
         $(LIBGDA_LIBS)
 
 # Extra options to supply to gtkdoc-mkdb
-MKDB_OPTIONS=--sgml-mode --output-format=xml
+MKDB_OPTIONS=--sgml-mode --output-format=xml --ignore-files="lexer.c"
 
 # Extra SGML files that are included by $(DOC_MAIN_SGML_FILE)
 content_files = fdl-appendix.sgml 
Index: report/testing/Makefile.am
===================================================================
--- report/testing/Makefile.am	(revision 2809)
+++ report/testing/Makefile.am	(working copy)
@@ -1,19 +1,19 @@
 AM_CPPFLAGS = \
-        -I$(top_srcdir) \
-        -I$(top_builddir) \
+	-I$(top_srcdir) \
+	-I$(top_builddir) \
 	-I$(top_srcdir)/report \
-        -I$(top_builddir)/report \
-        $(LIBGDA_CFLAGS)
+	-I$(top_builddir)/report \
+	$(LIBGDA_CFLAGS)
 
-bin_PROGRAMS = gda-report-test 
+bin_PROGRAMS = gda-report-test-3.0
 
-gda_report_test_SOURCES = \
+gda_report_test_3_0_SOURCES = \
         gda-report-test.c
 
-gda_report_test_LDADD = \
-        $(top_builddir)/libgda/libgda-3.0.la \
-        $(top_builddir)/libgda/graph/libgda_graph-3.0.la \
+gda_report_test_3_0_LDADD = \
+	$(top_builddir)/libgda/libgda-3.0.la \
+	$(top_builddir)/libgda/graph/libgda_graph-3.0.la \
 	$(top_builddir)/libgda/handlers/libgda_handlers-3.0.la \
 	$(top_builddir)/libgda/sql-delimiter/libgda_sql_delimiter-3.0.la \
-        $(top_builddir)/report/libgda-report/libgda-report-3.0.la \
-        $(LIBGDA_LIBS)
+	$(top_builddir)/report/libgda-report/libgda-report-3.0.la \
+	$(LIBGDA_LIBS)
Index: testing/Makefile.am
===================================================================
--- testing/Makefile.am	(revision 2809)
+++ testing/Makefile.am	(working copy)
@@ -3,7 +3,7 @@
 	-I$(top_builddir) \
 	$(LIBGDA_CFLAGS)
 
-bin_PROGRAMS = gda-test-connection
+bin_PROGRAMS = gda-test-connection-3.0
 noinst_PROGRAMS = gda-test-sql gda-test-models gda-test-server-operation gda-test-model-query gda-test-blob
 
 gda_test_sql_SOURCES = \
@@ -30,10 +30,10 @@
         $(top_builddir)/libgda/sql-delimiter/libgda_sql_delimiter-3.0.la \
         $(LIBGDA_LIBS)
 
-gda_test_connection_SOURCES = \
+gda_test_connection_3_0_SOURCES = \
         gda-test-connection.c
 
-gda_test_connection_LDADD = \
+gda_test_connection_3_0_LDADD = \
         $(top_builddir)/libgda/libgda-3.0.la \
         $(top_builddir)/libgda/handlers/libgda_handlers-3.0.la \
         $(top_builddir)/libgda/graph/libgda_graph-3.0.la \
Index: configure.in
===================================================================
--- configure.in	(revision 2809)
+++ configure.in	(working copy)
@@ -988,7 +988,7 @@
 dnl
 dnl where to install DTD files
 dnl
-LIBGDA_DTDDIR=${datadir}/${PACKAGE}/dtd
+LIBGDA_DTDDIR=${datadir}/libgda-3.0/dtd
 AC_SUBST(LIBGDA_DTDDIR)
 
 
