--- oo_1.0.1_src/unoil/com/sun/star/document/makefile.mk~	2000-09-18 17:28:26.000000000 +0200
+++ oo_1.0.1_src/unoil/com/sun/star/document/makefile.mk	2002-09-15 23:13:13.000000000 +0200
@@ -64,6 +64,7 @@
 PRJ		= ..$/..$/..$/..
 TARGET	= unoil_document
 PACKAGE	= com$/sun$/star$/document
+GENJAVAFILES += $(GENJAVADIR)$/com/sun/star/task/XInteractionContinuation.java
 
 # --- Settings -----------------------------------------------------
 
--- oo_1.0.1_src/unoil/makefile.pmk~	2001-08-31 18:04:43.000000000 +0200
+++ oo_1.0.1_src/unoil/makefile.pmk	2002-09-15 23:21:53.000000000 +0200
@@ -74,7 +74,7 @@
 JAVAFILESLIST	:= $(shell +cd $(GENJAVADIR)$/$(PACKAGE) $(COMMANDSEPARATOR) ls *.java)
 .ENDIF          # "$(L10N_framework)"==""
 
-GENJAVAFILES	:= $(foreach,i,$(JAVAFILESLIST) $(GENJAVADIR)$/$(PACKAGE)$/$i)
+GENJAVAFILES	+= $(foreach,i,$(JAVAFILESLIST) $(GENJAVADIR)$/$(PACKAGE)$/$i)
 GENCLASSFILES	:= $(foreach,i,$(JAVAFILESLIST:b) $(GENCLASSDIR)$/$(PACKAGE)$/$i.class)
 
 # --- Targets ------------------------------------------------------

