--- xmltv-0.5.15/grab/na/ZapListings.pm.orig	2003-07-06 12:13:03.000000000 -0700
+++ xmltv-0.5.15/grab/na/ZapListings.pm	2003-08-11 19:47:27.000000000 -0700
@@ -913,7 +913,7 @@
     }
 
     my $res=&doRequest($self->{ua}, $req, $self->{Debug});
-    if ( !$res->is_success || $res->content()=~m/your session has timed out/i ) {
+    if ( !($res->is_success || $res->code eq '500') || $res->content()=~m/your session has timed out/i ) {
 	# again.
 	$res=&doRequest($self->{ua}, $req, $self->{Debug});
 
@@ -926,7 +926,7 @@
 	}
     }
 
-    if ( !$res->is_success ) {
+    if ( !($res->is_success || $res->code eq '500') ) {
 	main::errorMessage("zap2it failed to give us a page: ".$res->code().":".
 			 HTTP::Status::status_message($res->code())."\n");
 	main::errorMessage("check postal/zip code or www site (maybe they're down)\n");
@@ -955,7 +955,7 @@
     $req = $self->Form2Request($self->{GridForm});
 
     $res=&doRequest($self->{ua}, $req, $self->{Debug});
-    if ( !$res->is_success || $res->content()=~m/your session has timed out/i ) {
+    if ( !($res->is_success || $res->code eq '500') || $res->content()=~m/your session has timed out/i ) {
 	# again.
 	$res=&doRequest($self->{ua}, $req, $self->{Debug});
 
@@ -968,7 +968,7 @@
 	}
     }
 
-    if ( !$res->is_success ) {
+    if ( !($res->is_success || $res->code eq '500') ) {
 	main::errorMessage("zap2it failed to give us a page: ".$res->code().":".
 			 HTTP::Status::status_message($res->code())."\n");
 	main::errorMessage("check postal/zip code or www site (maybe they're down)\n");
@@ -2050,12 +2050,12 @@
 	# looks like some requests require two identical calls since
 	# the zap2it server gives us a cookie that works with the second
 	# attempt after the first fails
-	if ( !$res->is_success || $res->content()=~m/your session has timed out/i ) {
+	if ( !($res->is_success || $res->code eq '500') || $res->content()=~m/your session has timed out/i ) {
 	    # again.
 	    $res=&doRequest($self->{ua}, $req, $self->{Debug});
 	}
 
-	if ( !$res->is_success ) {
+	if ( !($res->is_success || $res->code eq '500') ) {
 	    main::errorMessage("zap2it failed to give us a page: ".$res->code().":".
 			     HTTP::Status::status_message($res->code())."\n");
 	    main::errorMessage("check postal/zip code or www site (maybe they're down)\n");
