add missing inlcudes for gcc-4.3.x
bug 227167

diff -ruN vdr2jpeg-0.0.12_orig/vdr2jpeg.cpp vdr2jpeg-0.0.12/vdr2jpeg.cpp
--- vdr2jpeg-0.0.12_orig/vdr2jpeg.cpp	2007-08-29 18:43:09.000000000 +0200
+++ vdr2jpeg-0.0.12/vdr2jpeg.cpp	2008-06-15 14:51:05.000000000 +0200
@@ -27,6 +27,7 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <errno.h>
 #include <time.h>
 #include <iostream>

