--- audacious-0.2.orig/audacious/skin.c	2006-01-16 19:32:57.000000000 +0100
+++ audacious-0.2/audacious/skin.c	2006-02-18 21:27:13.000000000 +0100
@@ -330,7 +330,7 @@ skin_load_pixmap_id(Skin * skin, SkinPix
 
     memset(&basenames, 0, sizeof(basenames));
 
-    for (i = 0, y = 0; ext_targets[i]; i++, y++)
+    for (i = 0, y = 0; i < EXTENSION_TARGETS; i++, y++)
     {
         basenames[y] = g_strdup_printf("%s.%s", pixmap_id_mapping->name,
 			ext_targets[i]);
