--- gnome-iconedit-1.2.0/src/layers.c~	Sat Jul  8 14:47:56 2000
+++ gnome-iconedit-1.2.0/src/layers.c	Mon Jul 10 17:16:44 2000
@@ -304,7 +304,7 @@
 void
 ie_primitive_clear_all (IECanvasLayer *layer)
 {
-  memset (layer->pixbuf, 0x00, layer->rowstride * layer->height);
+  memset (layer->pixbuf, 0x00, (size_t)(layer->rowstride * layer->height));
 }
 
 void
--- gnome-iconedit-1.2.0/src/Makefile.am.orig	Sat Jul  8 00:51:52 2000
+++ gnome-iconedit-1.2.0/src/Makefile.am	Wed Aug 30 10:34:15 2000
@@ -95,6 +95,7 @@
 	-lgdk_pixbuf -lz -lpng -lXpm -lxml	\
 	$(OAF_LIBS)		\
 	-lbonobo-print		\
+	-lbonobox		\
 	-lbonobo		\
 	$(print_libs)
 endif	
--- gnome-iconedit-1.2.0/src/menus.c.orig	Thu Jun 22 13:31:59 2000
+++ gnome-iconedit-1.2.0/src/menus.c	Mon Dec  4 11:36:40 2000
@@ -36,7 +36,8 @@
 #include <libgnomeprint/gnome-print-preview.h>
 #include <libgnomeprint/gnome-print-pixbuf.h>
 #include <libgnomeprint/gnome-font.h>
-#include <libgnomeprint/gnome-printer-profile.h>
+/* not in gnome-print (0.25) anymore */
+/*#include <libgnomeprint/gnome-printer-profile.h>*/
 #include <libgnomeprint/gnome-printer-dialog.h>
 #include <libgnomeprint/gnome-print-master.h>
 #include <libgnomeprint/gnome-print-master-preview.h>
--- gnome-iconedit-1.2.0/configure.in.orig	Sat Jul  8 15:33:31 2000
+++ gnome-iconedit-1.2.0/configure.in	Mon Dec  4 11:55:20 2000
@@ -93,11 +93,11 @@
 dnl Whether to use OAF
 
 AC_MSG_CHECKING(if Bonobo uses OAF)
-if ( gnome-config --libs bonobo | grep oaf ) > /dev/null 2>&1; then
-	enable_oaf="yes"
-else
+dnl if ( gnome-config --libs bonobo | grep oaf ) > /dev/null 2>&1; then
+dnl	enable_oaf="yes"
+dnl else
 	enable_oaf="no"
-fi
+dnl fi
 
 AC_MSG_RESULT("$enable_oaf")
 
--- gnome-iconedit-1.2.0/src/Iconedit.h.orig	Wed Oct  3 23:32:48 2001
+++ gnome-iconedit-1.2.0/src/Iconedit.h	Wed Oct  3 23:32:58 2001
@@ -13,7 +13,7 @@
 #endif				/* __cplusplus */
 
 /** typedefs **/
-#include <liboaf/gnome-factory.h>
+#include <libgnorba/gnome-factory.h>
 #if !defined(ORBIT_DECL_GNOME_Iconedit_Iconedit) && !defined(_GNOME_Iconedit_Iconedit_defined)
 #define ORBIT_DECL_GNOME_Iconedit_Iconedit 1
 #define _GNOME_Iconedit_Iconedit_defined 1
