diff -r -u ibpp-1-0-5-2_orig/database.cpp ibpp-1-0-5-2/database.cpp
--- ibpp-1-0-5-2_orig/database.cpp	2001-11-10 13:52:58.000000000 +0100
+++ ibpp-1-0-5-2/database.cpp	2002-03-21 09:52:02.000000000 +0100
@@ -354,9 +354,7 @@
 		throw ExceptionImpl("Database::Create", "Database is already connected.");
 	if (_DatabaseName == NULL)
 		throw ExceptionImpl("Database::Create", "Unspecified database name.");
-	if (_UserName == NULL)
-		throw ExceptionImpl("Database::Create", "Unspecified user name.");
-	if (_UserPassword == NULL)
+	if (_UserName != NULL && _UserPassword == NULL)
 		throw ExceptionImpl("Database::Create", "Unspecified user password.");
 
     // Create a new database
@@ -374,8 +372,11 @@
 	else connect[0] = '\0';
 	strcat(connect, _DatabaseName);
 
-	sprintf(create, "CREATE DATABASE \"%s\" USER \"%s\" PASSWORD \"%s\" ",
-		connect, _UserName, _UserPassword);
+	if (_UserName != NULL)
+		sprintf(create, "CREATE DATABASE \"%s\" USER \"%s\" PASSWORD \"%s\" ",
+			connect, _UserName, _UserPassword);
+	else
+		sprintf(create, "CREATE DATABASE \"%s\" ", connect);
 	if (_CreateParams != NULL)
 	{
 		strcat(create, " ");
@@ -395,9 +396,7 @@
 		throw ExceptionImpl("Database::Connect", "Database is already connected.");
 	if (_DatabaseName == NULL)
 		throw ExceptionImpl("Database::Connect", "Unspecified database name.");
-	if (_UserName == NULL)
-		throw ExceptionImpl("Database::Connect", "Unspecified user name.");
-	if (_UserPassword == NULL)
+	if (_UserName != NULL && _UserPassword == NULL)
 		throw ExceptionImpl("Database::Connect", "Unspecified user password.");
 
 	// Attach to the database
@@ -406,8 +405,8 @@
 	char connect[1024];
 
     // Build a DPB based on the properties
-    dpb.Insert(isc_dpb_user_name, _UserName);
-    dpb.Insert(isc_dpb_password, _UserPassword);
+    if (_UserName != NULL) dpb.Insert(isc_dpb_user_name, _UserName);
+    if (_UserPassword != NULL) dpb.Insert(isc_dpb_password, _UserPassword);
     dpb.Insert(isc_dpb_force_write, (short)0);
     if (_RoleName != NULL) dpb.Insert(isc_dpb_sql_role_name, _RoleName);
 
diff -r -u ibpp-1-0-5-2_orig/exception.cpp ibpp-1-0-5-2/exception.cpp
--- ibpp-1-0-5-2_orig/exception.cpp	2001-11-10 13:52:58.000000000 +0100
+++ ibpp-1-0-5-2/exception.cpp	2002-03-21 09:52:28.000000000 +0100
@@ -47,6 +47,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdarg.h>
 
 using namespace ibpp_internals;
 
diff -r -u ibpp-1-0-5-2_orig/tests/Makefile ibpp-1-0-5-2/tests/Makefile
--- ibpp-1-0-5-2_orig/tests/Makefile	2001-06-08 18:26:34.000000000 +0200
+++ ibpp-1-0-5-2/tests/Makefile	2002-03-21 09:52:02.000000000 +0100
@@ -52,7 +52,7 @@
 	#LIBS+=$(IBPPDIR)/$(TARGETDIR)/$(PLATFORM)/libibpp_core.a
 	#LIBS+=$(IBPPDIR)/$(TARGETDIR)/$(PLATFORM)/libibpp_helper.a
 
-	LIBS+=-lm -lcrypt -lgds
+	LIBS+=-lm -lcrypt -lgds -ldl
 
 	ifeq ($(TARGETDIR),release)
 		CXXFLAGS+= -O2
