--- Configure.orig	2004-11-05 18:43:59.396409176 -0500
+++ Configure	2004-11-05 18:47:23.086443584 -0500
@@ -2177,3 +2177,3 @@
     fi	# }
-    LSOF_CFGF="-DLINUXV=$LSOF_VERS"
+    LSOF_CFGF="${CFLAGS} -DLINUXV=$LSOF_VERS"
     if test $LSOF_VERS -lt 21072	# {
@@ -4083,3 +4083,3 @@
   echo "" >> $LSOF_MKFC
-  echo "CFGL=	$LSOF_FCFGL -L./$LSOF_LIB -llsof $LSOF_CFGL" >> $LSOF_MKFC
+  echo "CFGL=	$LSOF_FCFGL -L./$LSOF_LIB -llsof $LSOF_CFGL ${LDFLAGS}" >> $LSOF_MKFC
 fi	# }
--- Customize.orig	2004-11-05 19:01:48.905819032 -0500
+++ Customize	2004-11-05 19:05:33.349698376 -0500
@@ -100,7 +100,7 @@
 do
   echo ""
   echo $EO "Do you want to customize (y|n) [y]? $EC"
-  read ANS EXCESS
+  ANS=y
   if test "X$ANS" = "Xn" -o "X$ANS" = "XN"
   then
     exit 0
@@ -204,7 +204,7 @@
   else
     echo $EO "Enable HASSECURITY (y|n) [n]? $EC"
   fi
-  read ANS EXCESS
+  ANS=y
   if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
   then
     echo ""
@@ -267,7 +267,7 @@
     else
       echo $EO "Enable HASNOSOCKSECURITY (y|n) [n]? $EC"
     fi
-    read ANS EXCESS
+    ANS=y
     if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
     then
       echo ""
@@ -329,7 +329,7 @@
   else
     echo $EO "Disable WARNINGSTATE? (y|n) [n]? $EC"
   fi
-  read ANS EXCESS
+  ANS=n
   if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
   then
     echo ""
@@ -918,7 +918,7 @@
       else
         echo $EO "Enable HASKERNIDCK (y|n) [n]? $EC"
       fi
-      read ANS EXCESS
+      ANS=y
       if test "X$ANS" = "Xy" -o "X$ANS" = "XY"
       then
         echo ""
@@ -1115,7 +1115,7 @@
   echo ""
   echo "Do you want to rename $OLD to ${OLD}.old and replace it with"
   echo $EO "$NEW (y|n) [y]? $EC"
-  read ANS EXCESS
+  ANS=y
   if test "X$ANS" = "Xn" -o "X$ANS" = "XN"
   then
     END=1
