diff -ur xfce4-cpugraph-plugin-0.3.0.orig/panel-plugin/Makefile.am xfce4-cpugraph-plugin-0.3.0/panel-plugin/Makefile.am
--- xfce4-cpugraph-plugin-0.3.0.orig/panel-plugin/Makefile.am	2005-09-30 14:12:14.000000000 +0300
+++ xfce4-cpugraph-plugin-0.3.0/panel-plugin/Makefile.am	2007-01-21 23:31:59.000000000 +0200
@@ -5,7 +5,7 @@
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"			\
 	@LIBXFCE4PANEL_CFLAGS@
 
-xfce4_cpugraph_plugin_LDFLAGS =					\
+xfce4_cpugraph_plugin_LDADD =					\
 	@LIBXFCE4PANEL_LIBS@
 
 xfce4_cpugraph_plugin_SOURCES =					\
