From 8467497542c0b41e5361a9741d914a0ea2f42c39 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Mon, 18 Jan 2010 02:38:45 +0100
Subject: [PATCH] No longer treat sync warnings like errors

---
 CHANGES          |    2 ++
 layman/action.py |   10 ++++++++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/CHANGES b/CHANGES
index 1a9473c..b4f24b9 100644
--- a/CHANGES
+++ b/CHANGES
@@ -7,6 +7,8 @@ Version TODO
  - Syncing failed for overlays that no longer exist in the remote
    lists with no need to (bug #301174)
 
+ - No longer treat sync warnings like errors (bug #301327)
+
  - Fix faults introduced at refactoring (bug #301253)
 
 
diff --git a/layman/action.py b/layman/action.py
index 02ba1ce..d058e9c 100644
--- a/layman/action.py
+++ b/layman/action.py
@@ -108,6 +108,7 @@ class Sync:
 
         OUT.debug('Updating selected overlays', 6)
 
+        fatals = []
         warnings = []
         success  = []
         for i in self.selection:
@@ -151,7 +152,7 @@ class Sync:
                 self.db.sync(i, self.quiet)
                 success.append('Successfully synchronized overlay "' + i + '".')
             except Exception, error:
-                warnings.append(
+                fatals.append(
                     'Failed to sync overlay "' + i + '".\nError was: '
                     + str(error))
 
@@ -161,9 +162,14 @@ class Sync:
                 OUT.info(i, 3)
                 
         if warnings:
-            OUT.warn('\nErrors:\n------\n', 2)
+            OUT.warn('\nWarnings:\n------\n', 2)
             for i in warnings:
                 OUT.warn(i + '\n', 2)
+
+        if fatals:
+            OUT.error('\nErrors:\n------\n')
+            for i in fatals:
+                OUT.error(i + '\n')
             return 1
 
         return 0
-- 
1.6.6

