--- tor-orig/src/config/torrc.sample.in	2005-08-13 19:07:33.969978500 +0200
+++ tor-0.1.0.14/src/config/torrc.sample.in	2005-08-13 19:08:47.902599000 +0200
@@ -14,6 +14,9 @@
 # With the default Mac OS X installer, Tor will look in ~/.tor/torrc or
 # /Library/Tor/torrc
 
+## Default username and group the server will run as
+User tor
+Group tor
 
 ## Replace this with "SocksPort 0" if you plan to run Tor only as a
 ## server, and not make any local application connections yourself.
@@ -46,6 +49,7 @@
 #Log notice syslog
 ## To send all messages to stderr:
 #Log debug stderr
+Log notice file /var/log/tor/tor.log
 
 ## Uncomment this to start the process in the background... or use
 ## --runasdaemon 1 on the command line.
@@ -63,6 +67,7 @@
 ## The directory for keeping all the keys/etc. By default, we store
 ## things in $HOME/.tor on Unix, and in Application Data\tor on Windows.
 #DataDirectory @LOCALSTATEDIR@/lib/tor
+DataDirectory   /var/lib/tor/data
 
 ## The port on which Tor will listen for local connections from Tor controller
 ## applications, as documented in control-spec.txt.  NB: this feature is
