Index: Mesa-5.0/configure.in
===================================================================
--- Mesa-5.0.orig/configure.in
+++ Mesa-5.0/configure.in
@@ -48,42 +48,6 @@ AM_MAINTAINER_MODE
 AC_CONFIG_FILES(
   Makefile
   include/Makefile
-  include/GL/Makefile
-  src/Makefile
-  src/array_cache/Makefile
-  src/math/Makefile
-  src/swrast/Makefile
-  src/swrast_setup/Makefile
-  src/tnl/Makefile
-  src/Trace/Makefile
-  src/FX/Makefile
-  src/FX/X86/Makefile
-  src/GGI/Makefile
-  src/GGI/include/Makefile
-  src/GGI/include/ggi/Makefile
-  src/GGI/include/ggi/mesa/Makefile
-  src/GGI/ggimesa.conf
-  src/GGI/default/Makefile
-  src/GGI/default/genkgi.conf
-  src/GGI/display/Makefile
-  src/GGI/display/fbdev.conf
-  src/OSmesa/Makefile
-  src/SVGA/Makefile
-  src/X/Makefile
-  src/X86/Makefile
-  src/SPARC/Makefile
-  util/Makefile
-  ggi/demos/Makefile
-  ggi/ggiglut/Makefile
-  src-glu/Makefile
-  si-glu/Makefile
-  si-glu/include/Makefile
-  si-glu/libnurbs/Makefile
-  si-glu/libnurbs/interface/Makefile
-  si-glu/libnurbs/internals/Makefile
-  si-glu/libnurbs/nurbtess/Makefile
-  si-glu/libtess/Makefile
-  si-glu/libutil/Makefile
 )
 
 dnl ------------------------------------------
@@ -420,6 +384,9 @@ if test x$with_ggi != xno; then
   fi
 fi
 
+build_ggi_fbdev_target=no
+build_ggi_genkgi_driver=no
+
 AC_MSG_CHECKING(whether to build the GGI driver)
 AC_MSG_RESULT($with_ggi)
 if test "x$with_ggi" = xyes; then
@@ -460,9 +427,11 @@ if test "x$with_ggi" = xyes; then
     AC_MSG_RESULT(yes)
   fi
     
-  AM_CONDITIONAL(HAVE_GGI_FBDEV, test $build_ggi_fbdev_target = yes)
-  AM_CONDITIONAL(HAVE_GGI_GENKGI, test $build_ggi_genkgi_driver = yes)
 fi
+
+AM_CONDITIONAL(HAVE_GGI_FBDEV, test $build_ggi_fbdev_target = yes)
+AM_CONDITIONAL(HAVE_GGI_GENKGI, test $build_ggi_genkgi_driver = yes)
+
 ggi_confdir=$prefix/etc
 ggi_libdir=$prefix/lib
 AC_SUBST(ggi_confdir)
Index: Mesa-5.0/Makefile.am
===================================================================
--- Mesa-5.0.orig/Makefile.am
+++ Mesa-5.0/Makefile.am
@@ -28,8 +28,8 @@ exec:
 	  (cd $$subdir && $(MAKE) $(AM_MAKEFLAGS) exec); \
 	done;
 
-DIST_SUBDIRS = include src si-glu src-glu @glut_dir@ $(SUB_DEMOS) util
-SUBDIRS = include src $(sel_sub_glu) $(sel_sub_glut) $(sel_sub_demos) util
+DIST_SUBDIRS = include @glut_dir@
+SUBDIRS = include $(sel_sub_glut)
 
 EXTRA_DIST = \
 	3Dfx \
Index: Mesa-5.0/demos/Makefile.am
===================================================================
--- Mesa-5.0.orig/demos/Makefile.am
+++ Mesa-5.0/demos/Makefile.am
@@ -44,7 +44,7 @@ check_PROGRAMS = bounce \
 	tunnel2 \
 	winpos
 
-#	osdemo \
+#	osdemo
 
 
 CLEANFILES = test.ppm
