--- SPQR.orig/Source/spqr_factorize.cpp	2009-03-09 15:12:47.091289398 +0000
+++ SPQR/Source/spqr_factorize.cpp	2009-03-14 09:30:18.570989288 +0000
@@ -30,6 +30,7 @@
 // === macros ==================================================================
 // =============================================================================
 
+#include <cstring>
 #include "spqr.hpp"
 
 #define FCHUNK 32        // FUTURE: make a parameter; Householder block size
