diff -Naur libpq++-4.0.orig/Makefile libpq++-4.0/Makefile
--- libpq++-4.0.orig/Makefile	2002-11-26 20:46:08.000000000 -0800
+++ libpq++-4.0/Makefile	2004-07-12 22:42:41.293567000 -0700
@@ -20,7 +20,7 @@
 OBJS= pgconnection.o pgdatabase.o pgtransdb.o pgcursordb.o pglobject.o
 LIBPGXXHEADERS = pgconnection.h pgdatabase.h pgtransdb.h pgcursordb.h pglobject.h
 
-SHARED= -fpic -shared -Wl,-x,-soname,$(TARGET).so.$(SO_MAJOR_VERSION) -Wl,-rpath,$(POSTGRES_HOME)/lib
+SHARED= -fPIC -shared -Wl,-x,-soname,$(TARGET).so.$(SO_MAJOR_VERSION) -Wl,-rpath,$(POSTGRES_HOME)/lib
 soname=$(TARGET).so.$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
 
 all:  $(soname)
@@ -29,7 +29,7 @@
 	$(CXX) $(CXXFLAGS) $(OBJS) -L$(POSTGRES_HOME)/lib -l$(LIBS) $(SHARED) -o $(soname)
 
 %.o	: %.cc
-	$(CXX) -c $(CXXFLAGS) $(CXXOPTS) -I$(POSTGRES_HOME)/include $<
+	$(CXX) -c $(CXXFLAGS) $(CXXOPTS) -I$(POSTGRES_HOME)/include $< -fPIC -DPIC
 
 clean:
 	rm -f $(OBJS) $(soname)
