Index: amarok/src/Options7.ui
===================================================================
--- amarok/src/Options7.ui	(revision 460630)
+++ amarok/src/Options7.ui	(revision 460631)
@@ -74,7 +74,4 @@
     </image>
 </images>
 <layoutdefaults spacing="6" margin="11"/>
-<includehints>
-    <includehint>dbsetup.h</includehint>
-</includehints>
 </UI>
Index: amarok/src/podcastsettingsbase.ui
===================================================================
--- amarok/src/podcastsettingsbase.ui	(revision 460630)
+++ amarok/src/podcastsettingsbase.ui	(revision 460631)
@@ -141,7 +141,7 @@
                         <cstring>pushButton_ok</cstring>
                     </property>
                     <property name="text">
-                        <string>O&amp;K</string>
+                        <string>&amp;OK</string>
                     </property>
                     <property name="autoDefault">
                         <bool>true</bool>
Index: amarok/src/firstrunwizard.ui
===================================================================
--- amarok/src/firstrunwizard.ui	(revision 460630)
+++ amarok/src/firstrunwizard.ui	(revision 460631)
@@ -490,10 +490,5 @@
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
     <includehint>kactivelabel.h</includehint>
-    <includehint>kactivelabel.h</includehint>
-    <includehint>kactivelabel.h</includehint>
-    <includehint>kactivelabel.h</includehint>
-    <includehint>dbsetup.h</includehint>
-    <includehint>kactivelabel.h</includehint>
 </includehints>
 </UI>
