Index: /home/user/workspace/amarok/src/metabundle.cpp
===================================================================
--- /home/user/workspace/amarok/src/metabundle.cpp	(revision 541169)
+++ /home/user/workspace/amarok/src/metabundle.cpp	(working copy)
@@ -293,7 +293,7 @@
 bool
 MetaBundle::checkExists()
 {
-    const bool exists = isStream() || ( url().protocol() == "file" && QFile::exists( url().path() ) );
+    const bool exists = isStream() || url().protocol() == "cdda" || ( url().protocol() == "file" && QFile::exists( url().path() ) );
 
     if( exists )
         m_flags |= Flags::Exists;
