diff -Naur Pixie.orig/Makefile.am Pixie.Makefile/Makefile.am
--- Pixie.orig/Makefile.am	2004-06-13 20:00:09.000000000 -0700
+++ Pixie.Makefile/Makefile.am	2004-06-14 18:33:41.169944272 -0700
@@ -11,8 +11,8 @@
 	cp -rf $(srcdir)/shaders $(distdir)
 
 install-data-local:
-	cp -rf $(srcdir)/doc $(prefix)
-	cp -rf $(srcdir)/shaders $(prefix)
+	cp -rf $(srcdir)/doc $(DESTDIR)$(prefix)
+	cp -rf $(srcdir)/shaders $(DESTDIR)$(prefix)
 
 
 
