--- Pixie/src/ri/shader-orig.h	2006-10-30 19:58:59.000000000 +0000
+++ Pixie/src/ri/shader.h	2006-10-30 20:00:24.000000000 +0000
@@ -374,6 +374,9 @@ public:
 // Description			:	This class encapsulates a shader
 // Comments				:
 // Date last edited		:	3/13/2001
+
+CShader                 *parseShader(const char *,const char *);
+
 class	CShader	: public CFileResource {
 public:
 								CShader(const char *);
