diff -u -ur hunspell-1.1.3.orig/src/tools/Makefile.am hunspell-1.1.3/src/tools/Makefile.am
--- hunspell-1.1.3.orig/src/tools/Makefile.am	2006-01-27 19:32:19.000000000 +0100
+++ hunspell-1.1.3/src/tools/Makefile.am	2006-01-27 19:33:05.000000000 +0100
@@ -1,13 +1,13 @@
-bin_PROGRAMS=munch unmunch example hunspell hunmorph hunstem
+bin_PROGRAMS=hunspell-munch hunspell-unmunch hunspell-example hunspell hunmorph hunstem
 
 INCLUDES=-I${top_srcdir}/src/hunspell -I${top_srcdir}/src/parsers
 
-munch_SOURCES=munch.c
-unmunch_SOURCES=unmunch.c
+hunspell_munch_SOURCES=munch.c
+hunspell_unmunch_SOURCES=unmunch.c
 include_HEADERS=munch.h unmunch.h
 
-example_SOURCES=example.cxx
-example_LDADD = ../hunspell/libhunspell.a
+hunspell_example_SOURCES=example.cxx
+hunspell_example_LDADD = ../hunspell/libhunspell.a
 
 hunspell_SOURCES=hunspell.cxx
 hunspell_LDADD = @LIBINTL@ ../hunspell/libhunspell.a \
diff -u -ur hunspell-1.1.3.orig/src/tools/example.cxx hunspell-1.1.3/src/tools/example.cxx
--- hunspell-1.1.3.orig/src/tools/example.cxx	2006-01-27 19:32:19.000000000 +0100
+++ hunspell-1.1.3/src/tools/example.cxx	2006-01-27 19:33:31.000000000 +0100
@@ -24,21 +24,21 @@
        af = mystrdup(argv[1]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"example affix_file dictionary_file file_of_words_to_check\n");
+    fprintf(stderr,"hunspell-example affix_file dictionary_file file_of_words_to_check\n");
     exit(1);
   }
   if (argv[2]) {
        df = mystrdup(argv[2]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"example affix_file dictionary_file file_of_words_to_check\n");
+    fprintf(stderr,"hunspell-example affix_file dictionary_file file_of_words_to_check\n");
     exit(1);
   }
   if (argv[3]) {
        wtc = mystrdup(argv[3]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"example affix_file dictionary_file file_of_words_to_check\n");
+    fprintf(stderr,"hunspell-example affix_file dictionary_file file_of_words_to_check\n");
     exit(1);
   }
 
diff -u -ur hunspell-1.1.3.orig/src/tools/hunmorph.cxx hunspell-1.1.3/src/tools/hunmorph.cxx
--- hunspell-1.1.3.orig/src/tools/hunmorph.cxx	2006-01-27 19:32:19.000000000 +0100
+++ hunspell-1.1.3/src/tools/hunmorph.cxx	2006-01-27 19:35:18.000000000 +0100
@@ -26,7 +26,7 @@
 
     for (i = 1; i < 3; i++)
 	if (!argv[i]) {
-	    fprintf(stderr, "correct syntax is:\nexample affix_file");
+	    fprintf(stderr, "correct syntax is:\nhunmorph affix_file");
 	    fprintf(stderr, " dictionary_file file_of_words_to_check\n");
 	    exit(1);
 	}
diff -u -ur hunspell-1.1.3.orig/src/tools/hunstem.cxx hunspell-1.1.3/src/tools/hunstem.cxx
--- hunspell-1.1.3.orig/src/tools/hunstem.cxx	2006-01-27 19:32:19.000000000 +0100
+++ hunspell-1.1.3/src/tools/hunstem.cxx	2006-01-27 19:35:28.000000000 +0100
@@ -25,7 +25,7 @@
 
     for (i = 1; i < 3; i++)
 	if (!argv[i]) {
-	    fprintf(stderr, "correct syntax is:\nexample affix_file");
+	    fprintf(stderr, "correct syntax is:\nhunstem affix_file");
 	    fprintf(stderr, " dictionary_file file_of_words_to_check\n");
 	    exit(1);
 	}
diff -u -ur hunspell-1.1.3.orig/src/tools/munch.c hunspell-1.1.3/src/tools/munch.c
--- hunspell-1.1.3.orig/src/tools/munch.c	2006-01-27 19:32:19.000000000 +0100
+++ hunspell-1.1.3/src/tools/munch.c	2006-01-27 19:33:56.000000000 +0100
@@ -42,14 +42,14 @@
        wf = mystrdup(argv[1]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"munch word_list_file affix_file\n");
+    fprintf(stderr,"hunspell-munch word_list_file affix_file\n");
     exit(1);
   }
   if (argv[2]) {
        af = mystrdup(argv[2]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"munch word_list_file affix_file\n");
+    fprintf(stderr,"hunspell-munch word_list_file affix_file\n");
     exit(1);
   }
 
diff -u -ur hunspell-1.1.3.orig/src/tools/unmunch.c hunspell-1.1.3/src/tools/unmunch.c
--- hunspell-1.1.3.orig/src/tools/unmunch.c	2006-01-27 19:32:19.000000000 +0100
+++ hunspell-1.1.3/src/tools/unmunch.c	2006-01-27 19:34:10.000000000 +0100
@@ -39,14 +39,14 @@
        wf = mystrdup(argv[1]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"unmunch dic_file affix_file\n");
+    fprintf(stderr,"hunspell-unmunch dic_file affix_file\n");
     exit(1);
   }
   if (argv[2]) {
        af = mystrdup(argv[2]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"unmunch dic_file affix_file\n");
+    fprintf(stderr,"hunspell-unmunch dic_file affix_file\n");
     exit(1);
   }
 
