--- Makefile
+++ Makefile
@@ -12,10 +12,8 @@
 # BREAK YOUR LOCAL DIFFS! show-diff and anything using it will likely randomly
 # break unless your underlying filesystem supports those sub-second times
 # (my ext3 doesn't).
-CFLAGS=-g -O2 -Wall
-
 # Should be changed to /usr/local
-prefix=$(HOME)
+prefix=/usr/
 
 bindir=$(prefix)/bin
 
@@ -74,7 +72,7 @@
 	@echo Generating gitversion.sh...
 	@rm -f $@
 	@echo "#!/bin/sh" > $@
-	@echo "echo \"$(shell cat $(VERSION)) ($(shell commit-id))\"" >> $@
+	@echo "echo \"$(shell cat $(VERSION)) ($(shell ./commit-id))\"" >> $@
 	@chmod +x $@
 
 
