Index: compiz-plugins-extra-0.8.6/configure.ac
===================================================================
--- compiz-plugins-extra-0.8.6.orig/configure.ac
+++ compiz-plugins-extra-0.8.6/configure.ac
@@ -136,9 +136,12 @@ AM_CONDITIONAL(THREED_PLUGIN, test "x$ha
 PKG_CHECK_MODULES(COMPIZMOUSEPOLL, compiz-mousepoll, [have_compiz_mousepoll=yes], [have_compiz_mousepoll=no])
 AM_CONDITIONAL(SHOWMOUSE_PLUGIN, test "x$have_compiz_mousepoll" = "xyes")
 
-PKG_CHECK_MODULES(LIBNOTIFY, libnotify, [have_libnotify=yes], [have_libnotify=no])
+AC_ARG_WITH([libnotify], [AS_HELP_STRING([--without-libnotify], [Build without libnotify support @<:@default=auto@:>@])])
+AS_IF([test "x$with_libnotify" != "xno"], [
+  PKG_CHECK_MODULES(LIBNOTIFY, libnotify, [have_libnotify=yes], [have_libnotify=no])
+])
 AM_CONDITIONAL(NOTIFICATION_PLUGIN, test "x$have_libnotify" = "xyes")
-if test "$have_libnotify" = yes; then
+if test "x$have_libnotify" = "xyes"; then
   AC_DEFINE(USE_NOTIFICATION, 1, [Build notification plugin])
 fi
 
