--- ion-devel-20030814/ioncore/manage.c	2003-07-27 02:00:49.000000000 +0200
+++ ion-devel/ioncore/manage.c	2003-11-07 16:46:12.000000000 +0100
@@ -24,8 +24,7 @@
 /*{{{ Add */
 
 
-static WScreen *find_suitable_screen(WClientWin *cwin, 
-									 const WManageParams *param)
+WScreen *find_suitable_screen(WClientWin *cwin, const WManageParams *param)
 {
 	WScreen *scr=NULL, *found=NULL;
 	bool respectpos=(param->tfor!=NULL || param->userpos);
--- ion-devel-20030814/ioncore/manage.h	2003-07-08 20:08:26.000000000 +0200
+++ ion-devel/ioncore/manage.h	2003-11-07 16:46:35.000000000 +0100
@@ -33,6 +33,8 @@
 } WManageParams;
 
 
+extern WScreen *find_suitable_screen(WClientWin *cwin,
+									 const WManageParams *param);
 
 extern bool add_clientwin_default(WClientWin *cwin,
 								  const WManageParams *param);
