diff -ur plasma-applet-daisy-0.0.4.22-orig/applet/src/daisy.cpp plasma-applet-daisy-0.0.4.22/applet/src/daisy.cpp
--- plasma-applet-daisy-0.0.4.22-orig/applet/src/daisy.cpp	2009-12-21 03:18:08.000000000 +0200
+++ plasma-applet-daisy-0.0.4.22/applet/src/daisy.cpp	2010-06-03 16:09:19.163818236 +0300
@@ -870,7 +870,7 @@
 	  if (m_task)
 	  {
 	    TaskManager::BasicMenu *taskMenu;TaskManager::TaskItem *taskitem;
-	    taskitem = new TaskManager::TaskItem::TaskItem(this, m_task);
+	    taskitem = new TaskManager::TaskItem(this, m_task);
 	    taskMenu = new TaskManager::BasicMenu( qobject_cast<QWidget*>(this), taskitem, m_groupManager );
 	    taskMenu->popup( event->screenPos() );
 	  }
@@ -896,7 +896,7 @@
 	  {
 	    QList<QAction*> actionList;actionList << m_act_newinstance << m_act_editlauncher;
 	    TaskManager::BasicMenu *taskMenu;TaskManager::TaskItem *taskitem;
-	    taskitem = new TaskManager::TaskItem::TaskItem(this, m_task);
+	    taskitem = new TaskManager::TaskItem(this, m_task);
 	    taskMenu = new TaskManager::BasicMenu( qobject_cast<QWidget*>(this), taskitem, m_groupManager, actionList );
 	    taskMenu->popup( event->screenPos() );
 	  }
