diff -ur aacplusenc-0.17.1.orig/configure aacplusenc-0.17.1/configure
--- aacplusenc-0.17.1.orig/configure	2008-02-14 18:28:24.000000000 +0200
+++ aacplusenc-0.17.1/configure	2009-07-22 22:52:16.000000000 +0300
@@ -8,7 +8,7 @@
 
 echo -n 'Checking for libfftw3f...'
 
-gcc -o test-fft -Wall -x c -lfftw3f -I/usr/local/include -L/usr/local/lib - <<'EOF'
+cat >test-fft.c <<'EOF'
 #include <fftw3.h>
 
 #define N 128
@@ -29,6 +29,8 @@
 }
 EOF
 
+${CC} ${LDFLAGS} ${CFLAGS} -Wall -x c test-fft.c -lfftw3f -o test-fft -I/usr/local/include -L/usr/local/lib
+
 if ./test-fft
 then
 	echo OK
@@ -38,7 +40,7 @@
 
 echo -n 'Detecting endianness...'
 
-gcc -o test-endian -Wall -x c - <<'EOF'
+${CC} ${LDFLAGS} ${CFLAGS} -o test-endian -Wall -x c - <<'EOF'
 #include <stdint.h>
 #include <stdio.h>
 
@@ -76,4 +78,4 @@
 
 echo '#endif' >> config.h
 
-echo 'Now run make'
\ No newline at end of file
+echo 'Now run make'
