diff --git a/Makefile b/Makefile
index c0a2200..b058a68 100644
--- a/Makefile
+++ b/Makefile
@@ -167,10 +167,10 @@ libtclstub.a::
 		rm -f $@; test -f $$LIBRARY && ln -s $$LIBRARY $@ || ln -s $(TCLLIBDIR)/$@ $@
 
 packlibs/libfilesystem.so.1.1: packlibs/filesystem.c libtclstub.a
-	$(CC) -shared -fPIC -DUSE_TCL_STUBS $(CFLAGS) $(INCLUDES) -o $@ $?
+	$(CC) -shared $(LDFLAGS) -fPIC -DUSE_TCL_STUBS $(CFLAGS) $(INCLUDES) -o $@ $?
 
 packlibs/libnetwork.so.1.23: packlibs/network.c libtclstub.a
-	$(CC) -shared -fPIC -DUSE_TCL_STUBS $(CFLAGS) $(INCLUDES) -o $@ $? $(EXTRALIBS)
+	$(CC) -shared $(LDFLAGS) -fPIC -DUSE_TCL_STUBS $(CFLAGS) $(INCLUDES) -o $@ $? $(EXTRALIBS)
 
 packlibs/liblogging.so.1.0: packlibs/logging.c libtclstub.a
-	$(CC) -shared -fPIC -DUSE_TCL_STUBS $(CFLAGS) $(INCLUDES) -o $@ $?
+	$(CC) -shared $(LDFLAGS) -fPIC -DUSE_TCL_STUBS $(CFLAGS) $(INCLUDES) -o $@ $?
