--- script/confhc.old	2006-02-04 16:24:05.000000000 +0100
+++ script/confhc	2006-02-04 16:33:39.000000000 +0100
@@ -76,10 +76,7 @@
   
 # Now look for GHC.  Determining the version number here is due to Simon Marlow.
 ghcsym () {
-  echo __GLASGOW_HASKELL__ >ghcsym.hs;
-  $1 -E -cpp -optP-P ghcsym.hs -o ghcsym.out;
-  grep -v '^#' ghcsym.out > $2;
-  rm -f ghcsym.hs ghcsym.out;
+  $1 --numeric-version | cut -d'.' -s -f 1,2 |sed 's/\([0-9]\)\.\([0-9]\)/\10\2/' | sed 's/\([0-9]\)0\([0-9][0-9]\)/\1\2/'i > $2;
 }
 echo -n "  Looking for ghc...   "
 if which ghc >/dev/null 2>&1
