--- ghostscript-6.53/src/unixlink.mak_orig	2002-03-07 13:40:11.000000000 +0100
+++ ghostscript-6.53/src/unixlink.mak	2002-03-07 13:40:22.000000000 +0100
@@ -67,6 +67,9 @@
 	$(ECHOGS_XE) -w $(ldt_tr) -n - $(CCLD) $(LDFLAGS) -o $(GS_XE)
 	$(ECHOGS_XE) -a $(ldt_tr) -n -s $(PSOBJ)gs.$(OBJ) -s
 	cat $(ld_tr) >>$(ldt_tr)
+	cp $(ldt_tr) $(ldt_tr)_orig
+	sed -e "s#-lXt#-L${BUILD_PREFIX}/ghostscript-6.53-r2/work/gimp-print-4.2.1/src/main/.libs/ -lgimpprint -lXt#g" \
+		$(ldt_tr)_orig > $(ldt_tr)
 	$(ECHOGS_XE) -a $(ldt_tr) -s - $(EXTRALIBS) $(STDLIBS)
 	if [ x$(XLIBDIR) != x ]; then LD_RUN_PATH=$(XLIBDIR); export LD_RUN_PATH; fi; \
 	XCFLAGS= XINCLUDE= XLDFLAGS= XLIBDIRS= XLIBS= \
