diff -Nur gs7.00.orig/lib/dvipdf gs7.00/lib/dvipdf
--- gs7.00.orig/lib/dvipdf	Thu Mar  9 09:40:39 2000
+++ gs7.00/lib/dvipdf	Sun Apr 15 23:31:36 2001
@@ -22,7 +22,7 @@
 	exit 1
 fi
 
-infile=$1;
+infile="$1";
 
 if [ $# -eq 1 ]
 then
@@ -33,7 +33,7 @@
 	esac
 	outfile="${base}.pdf"
 else
-	outfile=$2
+	outfile="$2"
 fi
 
 # Doing an initial 'save' helps keep fonts from being flushed between pages.
diff -Nur gs7.00.orig/lib/pdf2dsc gs7.00/lib/pdf2dsc
--- gs7.00.orig/lib/pdf2dsc	Fri Dec 29 06:03:49 2000
+++ gs7.00/lib/pdf2dsc	Sun Apr 15 23:32:43 2001
@@ -20,7 +20,7 @@
 
 pdffile=$1
 dscfile=$2
-: ${dscfile:=`echo $pdffile | sed 's,\.[^/.]*,,'`.dsc}
+: ${dscfile:=`echo "$pdffile" | sed 's,\.[^/.]*,,'`.dsc}
 
 exec gs -q -dNODISPLAY -dSAFER -dDELAYSAFER\
     -sPDFname="$pdffile" -sDSCname="$dscfile" pdf2dsc.ps -c quit
diff -Nur gs7.00.orig/lib/pdf2ps gs7.00/lib/pdf2ps
--- gs7.00.orig/lib/pdf2ps	Fri May 19 08:38:25 2000
+++ gs7.00/lib/pdf2ps	Sun Apr 15 23:33:20 2001
@@ -14,7 +14,7 @@
 
 if [ $# -eq 2 ] 
 then
-    outfile=$2
+    outfile="$2"
 elif [ $# -eq 1 ]
 then
     outfile=`basename "$1" \.pdf`.ps
diff -Nur gs7.00.orig/lib/pfbtopfa gs7.00/lib/pfbtopfa
--- gs7.00.orig/lib/pfbtopfa	Wed Apr 12 19:41:41 2000
+++ gs7.00/lib/pfbtopfa	Sun Apr 15 23:33:52 2001
@@ -4,7 +4,7 @@
 
 if [ $# -eq 2 ] 
 then
-    outfile=$2
+    outfile="$2"
 elif [ $# -eq 1 ]
 then
     outfile=`basename "$1" \.pfb`.pfa
diff -Nur gs7.00.orig/lib/ps2epsi gs7.00/lib/ps2epsi
--- gs7.00.orig/lib/ps2epsi	Wed Apr  4 23:05:00 2001
+++ gs7.00/lib/ps2epsi	Sun Apr 15 23:36:15 2001
@@ -1,7 +1,7 @@
 #!/bin/sh
 # $Id: ghostscript-6.53-missquotes.patch,v 1.1 2002/04/28 18:40:09 jnelson Exp $
 
-tmpfile=/tmp/ps2epsi$$
+tmpfile=`mktemp /tmp/$0.XXXXXX` || exit 1
 
 export outfile
 
@@ -10,7 +10,7 @@
 	exit 1
 fi
 
-infile=$1;
+infile="$1";
 
 if [ $# -eq 1 ]
 then
@@ -22,9 +22,9 @@
 	  *.epsf)	base=`basename "${infile}" .epsf` ;;
 	  *)		base=`basename "${infile}"` ;;
 	esac
-	outfile=${base}.epsi
+	outfile="${base}.epsi"
 else
-	outfile=$2
+	outfile="$2"
 fi
 
 ls -l "${infile}" |
