--- ghostscript-7.07.1-r8-orig/work/espgs-7.07.1/ijs/Makefile.in	Tue Apr 23 07:58:32 2002
+++ ghostscript-7.07.1-r8/work/espgs-7.07.1/ijs/Makefile.in	Thu Mar 24 22:59:28 2005
@@ -11,6 +11,12 @@
 SHARED_OBJ=.so
 #SHARED_LDFLAGS=-dylib
 #SHARED_OBJ=.dylib
+
+CPU = $(shell uname)
+ifeq ($(CPU),Darwin)
+SHARED_LDFLAGS=-dynamiclib
+SHARED_OBJ=.dylib
+endif
 
 FE=-o 
 IJS_EXEC_SERVER=ijs_exec_unix$(OBJ)
@@ -60,8 +64,13 @@
 	$(RM) *~ gmon.out core ijs_spec.log ijs_spec.tex ijs_spec.aux libijs.a libijs$(SHARED_OBJ) config.cache config.log config.status ijs-config
 
 install:	all
+ifeq ($(CPU),Darwin)
+	$(INSTALL) ijs_client_example$(EXE) $(DESTDIR)$(bindir)/ijs_client_example$(EXE)
+	$(INSTALL) ijs-config $(DESTDIR)$(bindir)/ijs-config
+else
 	$(INSTALL) ijs_client_example$(EXE) -c $(DESTDIR)$(bindir)/ijs_client_example$(EXE)
 	$(INSTALL) ijs-config -c $(DESTDIR)$(bindir)/ijs-config
+endif
 	$(INSTALL) libijs.a $(DESTDIR)$(libdir)/libijs.a
 	$(INSTALL) libijs$(SHARED_OBJ) $(DESTDIR)$(libdir)/libijs$(SHARED_OBJ)
 	-mkdir $(DESTDIR)$(pkgincludedir)
diff -Naur ghostscript-7.07.1-r8-orig/work/espgs-7.07.1/ijs/ijs_server_example.c ghostscript-7.07.1-r8/work/espgs-7.07.1/ijs/ijs_server_example.c
--- ghostscript-7.07.1-r8-orig/work/espgs-7.07.1/ijs/ijs_server_example.c	Tue Apr 23 10:01:17 2002
+++ ghostscript-7.07.1-r8/work/espgs-7.07.1/ijs/ijs_server_example.c	Thu Mar 24 22:59:46 2005
@@ -25,7 +25,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <malloc.h>
+#include <malloc/malloc.h>
 #include "ijs.h"
 #include "ijs_server.h"
