--- espgs-7.05.6/src/gdevpx.c	2002-04-23 07:58:43.000000000 -0400
+++ espgs-7.05.6/src/gdevpx.c.new	2003-11-09 15:25:16.000000000 -0500
@@ -851,8 +851,7 @@
 	    px_put_s(s, (uint)pattern[i]);
 	px_put_a(s, pxaLineDashStyle);
 	if (offset != 0) {
-	    px_put_rl(s, offset);
-	    px_put_a(s, pxaDashOffset);
+	    px_put_usa(s, (uint)offset, pxaDashOffset);
 	}
     }
     spputc(s, pxtSetLineDash);
