diff -ur bmp-0.9.6.1.orig/Input/vorbis/fileinfo.c bmp-0.9.6.1/Input/vorbis/fileinfo.c
--- bmp-0.9.6.1.orig/Input/vorbis/fileinfo.c	2003-12-24 07:57:29.000000000 -0600
+++ bmp-0.9.6.1/Input/vorbis/fileinfo.c	2004-07-02 13:00:42.349657696 -0500
@@ -54,6 +54,8 @@
 static gint init_files(vcedit_state * state);
 static gint close_files(vcedit_state * state);
 
+GtkWidget *xmms_show_message_test(gchar *, gchar *, gchar *, gboolean, GtkSignalFunc, gpointer);
+
 extern GMutex *vf_mutex;
 static GtkWidget *window = NULL;
 static GList *genre_list = NULL;
