Index: klamav-0.32-source/klamav-0.32/src/arkollon/uninstallwizard.cpp
===================================================================
--- klamav-0.32-source.orig/klamav-0.32/src/arkollon/uninstallwizard.cpp
+++ klamav-0.32-source/klamav-0.32/src/arkollon/uninstallwizard.cpp
@@ -318,5 +318,5 @@ void UninstallWizard::finished()
 	nextButton->setEnabled(true);
 }
 
-
+#include "uninstallwizard.moc"
 
Index: klamav-0.32-source/klamav-0.32/src/arkollon/wizard.cpp
===================================================================
--- klamav-0.32-source.orig/klamav-0.32/src/arkollon/wizard.cpp
+++ klamav-0.32-source/klamav-0.32/src/arkollon/wizard.cpp
@@ -913,5 +913,5 @@ void Wizard::logPressed()
 	logDialog->show();
 }
 
-
+#include "wizard.moc"
 
Index: klamav-0.32-source/klamav-0.32/Makefile.am
===================================================================
--- klamav-0.32-source.orig/klamav-0.32/Makefile.am
+++ klamav-0.32-source/klamav-0.32/Makefile.am
@@ -20,4 +20,3 @@ EXTRA_DIST = admin COPYING configure.in.
 dist-hook:
 	cd $(top_distdir) && perl admin/am_edit -padmin
 	cd $(top_distdir) && $(MAKE) -f admin/Makefile.common subdirs
-_SOURCES = dcopklamscan.h
