--- c-posix.c.orig	2002-10-24 11:03:51.000000000 +0300
+++ c-posix.c	2005-01-12 20:19:24.000000000 +0200
@@ -6344,8 +6344,8 @@
   g_struct_msghdr();
   gcmnt("local socket address");
   { struct sockaddr_un DUMMY;
-    ifprintf(fp,"   subtype sun_path_string is
-      POSIX_String (1 .. %d);\n", sizeof (DUMMY.sun_path));
+    ifprintf(fp,"   subtype sun_path_string is\n"
+      "      POSIX_String (1 .. %d);\n", sizeof (DUMMY.sun_path));
   }
   g_struct_sockaddr_un();
   gcmnt("internet socket address");
