diff -ur Build-j2sdk-1.4.0.orig/j2se/make/jpda/back/Makefile Build-j2sdk-1.4.0/j2se/make/jpda/back/Makefile
--- Build-j2sdk-1.4.0.orig/j2se/make/jpda/back/Makefile	2002-02-12 07:55:28.000000000 -0600
+++ Build-j2sdk-1.4.0/j2se/make/jpda/back/Makefile	2002-09-09 15:35:39.000000000 -0500
@@ -35,6 +35,9 @@
 OTHER_LDLIBS = -ldl -lsocket
 endif # PLATFORM
 
+# Link to libjvm since it contains fork1
+# Added by Tushar <tush@yahoo.com>
+OTHER_LDLIBS += $(JVMLIB)
 
 #
 # Files to compile.
