--- stlport/stl/config/_gcc.h.orig	2006-12-24 02:38:24.000000000 +0100
+++ stlport/stl/config/_gcc.h	2006-12-24 02:40:20.000000000 +0100
@@ -9,6 +9,7 @@
 #  ifndef _STLP_USE_GLIBC
 #    define _STLP_USE_GLIBC 1
 #  endif
+#  define _STLP_NO_VENDOR_STDLIB_L
 #  if defined (__UCLIBC__) && !defined (_STLP_USE_UCLIBC)
 #    define _STLP_USE_UCLIBC 1
 #  endif
@@ -21,7 +22,9 @@
 #endif
 
 #if (__GNUC__ < 3)
-#  define _STLP_NO_VENDOR_STDLIB_L
+#  ifndef _STLP_NO_VENDOR_STDLIB_L
+#    define _STLP_NO_VENDOR_STDLIB_L
+#  endif
 #endif
 
 /* We guess if we are using the cygwin distrib that has a special include schema.
