diff -ur nicotine+-1.2.9.orig/pynicotine/gtkgui/frame.py nicotine+-1.2.9/pynicotine/gtkgui/frame.py
--- nicotine+-1.2.9.orig/pynicotine/gtkgui/frame.py	2007-12-04 10:33:57.000000000 +0100
+++ nicotine+-1.2.9/pynicotine/gtkgui/frame.py	2007-12-04 10:40:31.000000000 +0100
@@ -947,20 +947,8 @@
 		
 		for widget in self.BuddiesComboEntries:
 			gobject.idle_add(widget.Fill)
-			
-	def getTabPosition(self, string):
-		if string == "top":
-			position = gtk.POS_TOP
-		elif string == "bottom":
-			position = gtk.POS_BOTTOM
-		elif string == "left":
-			position = gtk.POS_LEFT
-		elif string == "right":
-			position = gtk.POS_RIGHT
-		else:
-			position = gtk.POS_TOP
-		return position
-		
+
+
 	def OnAutoAway(self):
 		if not self.away:
 			self.autoaway = True
@@ -1675,7 +1663,20 @@
 				message = message.replace(word, filler * len(word))
 				
 		return message
-			
+	
+	def getTabPosition(self, string):
+		if string in ("Top", "top", _("Top")):
+			position = gtk.POS_TOP
+		elif string in ("Bottom", "bottom", _("Bottom")):
+			position = gtk.POS_BOTTOM
+		elif string in ("Left", "left", _("Left")):
+			position = gtk.POS_LEFT
+		elif string in ("Right", "right", _("Right")):
+			position = gtk.POS_RIGHT
+		else:
+			position = gtk.POS_TOP
+		return position		
+
 	def SetTabPositions(self):
 		ui = self.np.config.sections["ui"]
 		self.ChatNotebook.set_tab_pos(self.getTabPosition(ui["tabrooms"]))
diff -ur nicotine+-1.2.9.orig/pynicotine/gtkgui/settingswindow.py nicotine+-1.2.9/pynicotine/gtkgui/settingswindow.py
--- nicotine+-1.2.9.orig/pynicotine/gtkgui/settingswindow.py	2007-12-04 10:33:57.000000000 +0100
+++ nicotine+-1.2.9/pynicotine/gtkgui/settingswindow.py	2007-12-04 10:37:06.000000000 +0100
@@ -1206,12 +1206,12 @@
 	def GetSettings(self):
 		return {
 			"ui": {
-				"tabmain": self.MainPosition.get_active_text().lower(),
-				"tabrooms": self.ChatRoomsPosition.get_active_text().lower(),
-				"tabprivate": self.PrivateChatPosition.get_active_text().lower(),
-				"tabinfo": self.UserInfoPosition.get_active_text().lower(),
-				"tabbrowse": self.UserBrowsePosition.get_active_text().lower(),
-				"tabsearch": self.SearchPosition.get_active_text().lower(),
+				"tabmain": self.MainPosition.get_active_text(),
+				"tabrooms": self.ChatRoomsPosition.get_active_text(),
+				"tabprivate": self.PrivateChatPosition.get_active_text(),
+				"tabinfo": self.UserInfoPosition.get_active_text(),
+				"tabbrowse": self.UserBrowsePosition.get_active_text(),
+				"tabsearch": self.SearchPosition.get_active_text(),
 				"labelmain": self.MainAngleSpin.get_value_as_int(),
 				"labelrooms": self.ChatRoomsAngleSpin.get_value_as_int(),
 				"labelprivate": self.PrivateChatAngleSpin.get_value_as_int(),
