--- configure.ac
+++ configure.ac
@@ -64,6 +64,8 @@
 dnl -- Static Libaries --
 AC_PROG_RANLIB
 
+CPPFLAGS="$CPPFLAGS `clanlib0.6-config --cflags`"
+LIBS="$LIBS `clanlib0.6-config --libs`"
 AC_CHECK_HEADER(ClanLib/core.h,, [echo "Please get ClanLib from http://www.clanlib.org"; exit])
 
 dnl FIXME: we don't need pthread, but some system might need it to succeed at the final link
@@ -274,11 +276,11 @@
 AC_CHECK_FUNCS(select strdup)
 
 dnl Checking for the install prefix and then use it to define PINGUS_DATADIR
-PINGUS_EXPAND_DIR(pingus_datadir, "$datadir/games/$PACKAGE")
+PINGUS_EXPAND_DIR(pingus_datadir, "$datadir/$PACKAGE")
 AC_DEFINE_UNQUOTED(PINGUS_DATADIR, "$pingus_datadir", [Location of Pingus data files])
 
 dnl -- Define the localedir, needed for gettext --
-PINGUS_EXPAND_DIR(localedir, "$datadir/locale")
+PINGUS_EXPAND_DIR(localedir, "/usr/share/locale")
 AC_DEFINE_UNQUOTED(LOCALEDIR, "$localedir", [Location of gettext files])
 
 # po/Makefile.in intl/Makefile <- this must be in one line, if its not
--- data/controller/Makefile.am
+++ data/controller/Makefile.am
@@ -15,7 +15,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/controller
+pkgdatadir = $(datadir)/@PACKAGE@/controller
 EXTRA_DIST = default.xml gamepad.xml
 pkgdata_DATA = $(EXTRA_DIST)
 
--- data/data/Makefile.am
+++ data/data/Makefile.am
@@ -15,7 +15,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/data
+pkgdatadir = $(datadir)/@PACKAGE@/data
 
 EXTRA_DIST  = \
 pingus.scr    \
--- data/demos/Makefile.am
+++ data/demos/Makefile.am
@@ -15,7 +15,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/demos
+pkgdatadir = $(datadir)/@PACKAGE@/demos
 
 pkgdata_DATA = level1.pde
 
--- data/images/Makefile.am
+++ data/images/Makefile.am
@@ -15,7 +15,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/images
+pkgdatadir = $(datadir)/@PACKAGE@/images
 
 nobase_pkgdata_DATA = backgrounds/large_star.png \
 backgrounds/middle_star.png \
--- data/levels/Makefile.am
+++ data/levels/Makefile.am
@@ -18,7 +18,7 @@
 #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 ## Override automake so that "make install" puts these in proper place:
-pkgdatadir = $(datadir)/games/@PACKAGE@/levels
+pkgdatadir = $(datadir)/@PACKAGE@/levels
 
 nobase_pkgdata_DATA = \
 	$(wildcard tutorial/*.xml) \
--- data/Makefile.am
+++ data/Makefile.am
@@ -18,9 +18,8 @@
 SUBDIRS = controller images levels demos data music sounds worldmaps themes prefabs
 
 # Tweaking the `datadir' to install the files in the correct location
-pkgdatadir = $(datadir)/games/@PACKAGE@
+pkgdatadir = $(datadir)/@PACKAGE@
 
 EXTRA_DIST = README
-pkgdata_DATA = $(EXTRA_DIST)
 
 # EOF #
--- data/music/Makefile.am
+++ data/music/Makefile.am
@@ -15,7 +15,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/music
+pkgdatadir = $(datadir)/@PACKAGE@/music
 EXTRA_DIST = gd-giirm.s3m   pingus-5.it  rough_journey.it \
 gd-ite.it	    pingus-1.it  pingus-6.it  success_1.it \
 README	     gd-matth.it    pingus-2.it  pingus-7.it  success_2.it \
@@ -23,4 +23,4 @@
 gd-cancn.it  goin_march.it  pingus-4.it  pingus-9.it
 pkgdata_DATA = $(EXTRA_DIST)
 
-# EOF #
\ No newline at end of file
+# EOF #
--- data/prefabs/Makefile.am
+++ data/prefabs/Makefile.am
@@ -16,7 +16,7 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 ## Override automake so that "make install" puts these in proper place:
-pkgdatadir = $(datadir)/games/@PACKAGE@/prefabs
+pkgdatadir = $(datadir)/@PACKAGE@/prefabs
 
 EXTRA_DIST = water.xml snow-entrance.xml
 pkgdata_DATA = $(EXTRA_DIST)
--- data/sounds/Makefile.am
+++ data/sounds/Makefile.am
@@ -15,9 +15,9 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/sounds
+pkgdatadir = $(datadir)/@PACKAGE@/sounds
 EXTRA_DIST = chink.wav   goodidea.wav  plop.wav   splash.wav  ting.wav \
              digger.wav  ohno.wav	  plop2.wav  tick.wav	 yipee.wav
 pkgdata_DATA = $(EXTRA_DIST)
 
-# EOF #
\ No newline at end of file
+# EOF #
--- data/themes/Makefile.am
+++ data/themes/Makefile.am
@@ -18,9 +18,9 @@
 #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 # Tweaking the `datadir' to install the files in the correct location
-pkgdatadir = $(datadir)/games/@PACKAGE@/themes
+pkgdatadir = $(datadir)/@PACKAGE@/themes
 
 EXTRA_DIST = $(wildcard *.xml)
 pkgdata_DATA = $(EXTRA_DIST)
 
-# EOF #
\ No newline at end of file
+# EOF #
--- data/worldmaps/Makefile.am
+++ data/worldmaps/Makefile.am
@@ -16,9 +16,9 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 # Tweaking the `datadir' to install the files in the correct location
-pkgdatadir = $(datadir)/games/@PACKAGE@/worldmaps
+pkgdatadir = $(datadir)/@PACKAGE@/worldmaps
 
 EXTRA_DIST = volcano.xml pacman.xml tutorial.xml
 pkgdata_DATA = $(EXTRA_DIST)
 
-# EOF #
\ No newline at end of file
+# EOF #
--- po/Makefile.in.in
+++ po/Makefile.in.in
@@ -21,7 +21,7 @@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
-localedir = $(datadir)/locale
+localedir = /usr/share/locale
 gettextsrcdir = $(datadir)/gettext/po
 
 INSTALL = @INSTALL@
--- src/Makefile.am
+++ src/Makefile.am
@@ -17,10 +17,9 @@
 #  along with this program; if not, write to the Free Software
 #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-bindir=$(prefix)/games
 
 # The next lines are for gettext
-localedir = $(datadir)/locale
+localedir = /usr/share/locale
 # DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 # INCLUDES = -I$(srcdir) -I.. -I$(top_srcdir)/lib -I../intl -I$(top_srcdir)/intl
 
