=== autotools/libmozjs.m4
==================================================================
--- autotools/libmozjs.m4	(revision 1354)
+++ autotools/libmozjs.m4	(revision 1355)
@@ -32,9 +32,19 @@
 
   AC_MSG_CHECKING([for libmozjs.so version >= $ggl_check_libmozjs_min_version])
 
+  AC_CHECK_HEADER([jsversion.h], [has_jsversion_h=yes], [has_jsversion_h=no])
+  if test x$has_jsversion_h = xyes; then
+    LIBMOZJS_CFLAGS="$LIBMOZJS_CFLAGS -DHAVE_JSVERSION_H"
+    CPPFLAGS=$LIBMOZJS_CFLAGS
+  fi
+
   AC_LINK_IFELSE([[
     #include<jsapi.h>
+    #ifdef HAVE_JSVERSION_H
+    #include<jsversion.h>
+    #else
     #include<jsconfig.h>
+    #endif
 
     #if JS_VERSION < $ggl_check_libmozjs_min_version
     #error "libmozjs.so version is too low."
@@ -68,7 +78,11 @@
       // MOZILLA_1_8_BRANCH macro is not defined but the library was compiled
       // with the flag, or vise versa.
       #include <jsapi.h>
+      #ifdef HAVE_JSVERSION_H
+      #include<jsversion.h>
+      #else
       #include<jsconfig.h>
+      #endif
 
       static JSBool f(JSContext *c, JSObject *o, uintN ac, jsval *av, jsval *r) {
         return JS_TRUE;
=== extensions/gtkmoz_browser_element/Makefile.am
==================================================================
--- extensions/gtkmoz_browser_element/Makefile.am	(revision 1354)
+++ extensions/gtkmoz_browser_element/Makefile.am	(revision 1355)
@@ -59,8 +59,10 @@
 
 gtkmoz_browser_child_CPPFLAGS	= $(GTK2_CFLAGS) \
 				  $(GTKMOZEMBED_CFLAGS) \
+				  $(LIBMOZJS_CFLAGS) \
 				  $(PREDEFINED_MACROS)
 
+# Don't link to libmozjs explicitly.
 gtkmoz_browser_child_LDADD	= $(GTK2_LIBS) \
 				  $(GTKMOZEMBED_LIBS) \
 				  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la \
=== extensions/gtkmoz_browser_element/browser_child.cc
==================================================================
--- extensions/gtkmoz_browser_element/browser_child.cc	(revision 1354)
+++ extensions/gtkmoz_browser_element/browser_child.cc	(revision 1355)
@@ -37,7 +37,12 @@
 #include <gtkmozembed.h>
 #include <gtkmozembed_internal.h>
 #include <jsapi.h>
+
+#ifdef HAVE_JSVERSION_H
+#include <jsversion.h>
+#else
 #include <jsconfig.h>
+#endif
 
 #include <nsComponentManagerUtils.h>
 #include <nsCOMPtr.h>
=== configure.ac
==================================================================
--- configure.ac	(revision 1354)
+++ configure.ac	(revision 1355)
@@ -905,9 +905,6 @@
     build_gtkmoz_browser_element=no
   fi
 else
-  # Add smjs related cflags, we don't want to link to smjs explicitly,
-  # so don't use its libs.
-  GTKMOZEMBED_CFLAGS="$GTKMOZEMBED_CFLAGS $LIBMOZJS_CFLAGS"
   # Add additional include dirs for mozilla 1.8 and pre 1.9
   EMBED_INCDIR=`$PKG_CONFIG --variable=includedir $has_gtkmozembed`
   GTKMOZEMBED_CFLAGS="$GTKMOZEMBED_CFLAGS -I$EMBED_INCDIR"
