diff -Naur brltty-3.10-orig/common.mk brltty-3.10/common.mk
--- brltty-3.10-orig/common.mk	2008-07-16 16:19:54.000000000 -0600
+++ brltty-3.10/common.mk	2008-09-22 21:53:19.000000000 -0600
@@ -73,8 +73,9 @@
 install-apihdr-directory:
 	$(INSTALL_DIRECTORY) $(INSTALL_APIHDR_DIRECTORY)
 
+INSTALL_API_SOCKET_DIRECTORY = $(INSTALL_ROOT)$(API_SOCKET_DIRECTORY)
 install-apisoc-directory:
-	-$(INSTALL_DIRECTORY) -m 1777 $(API_SOCKET_DIRECTORY)
+	-$(INSTALL_DIRECTORY) -m 1777 $(INSTALL_API_SOCKET_DIRECTORY)
 
 clean::
 	-rm -f *.$O *.auto.h *.auto.c core implib.a
