Index: jack-rack-1.4.6/src/Makefile.am
===================================================================
--- jack-rack-1.4.6.orig/src/Makefile.am
+++ jack-rack-1.4.6/src/Makefile.am
@@ -54,13 +54,7 @@ jack_rack_CFLAGS = \
 	$(XML_CFLAGS) \
 	-DPKGDATADIR=\"$(pkgdatadir)\" \
 	-DPIXMAPDIR=\"$(pixmapdir)\" \
-	-DDTDDIR=\"$(dtddir)\" \
-	-DG_DISABLE_DEPRECATED=1 \
-	-DGTK_DISABLE_DEPRECATED=1 \
-	-DGDK_DISABLE_DEPRECATED=1 \
-	-DGDK_PIXBUF_DISABLE_DEPRECATED=1 \
-	-DGNOME_DISABLE_DEPRECATED=1
-
+	-DDTDDIR=\"$(dtddir)\"
 
 jack_rack_LDADD = \
 	$(JACK_LIBS) \
