Index: src/fileimporter.h
===================================================================
--- src/fileimporter.h	(Revision 307)
+++ src/fileimporter.h	(Arbeitskopie)
@@ -39,7 +39,7 @@
         ~FileImporter();
 
         File* load ( const QString& text );
-        virtual File* load( QIODevice *iodevice ) = NULL;
+        virtual File* load( QIODevice *iodevice ) = 0;
 
         static bool guessCanDecode( const QString & ) { return FALSE; };
 
