diff -u install.orig/Makefile.am install/Makefile.am
--- install.orig/Makefile.am	Sat Nov  3 16:34:09 2001
+++ install/Makefile.am	Fri Jan 25 16:22:30 2002
@@ -4,17 +4,17 @@
 
 if USE_KDE
 install-kde:
-	mkdir -p $(KDE_DATA_DIR)/icons
-	mkdir -p $(KDE_DATA_DIR)/mimelnk/application
-	mkdir -p $(KDE_DATA_DIR)/applnk/Development
+	mkdir -p $(DESTDIR)/$(KDE_DATA_DIR)/icons
+	mkdir -p $(DESTDIR)/$(KDE_DATA_DIR)/mimelnk/application
+	mkdir -p $(DESTDIR)/$(KDE_DATA_DIR)/applnk/Development
 	if test -w $(KDE_DATA_DIR)/icons ; then \
-	  install -m 644 $(top_srcdir)/src/appicon.xpm $(KDE_DATA_DIR)/icons/poedit.xpm; \
+	  install -m 644 $(top_srcdir)/src/appicon.xpm $(DESTDIR)/$(KDE_DATA_DIR)/icons/poedit.xpm; \
 	fi
 	if test -w $(KDE_DATA_DIR)/mimelnk/application ; then \
-	  install -m 644 $(srcdir)/x-po.kdelnk $(KDE_DATA_DIR)/mimelnk/application; \
+	  install -m 644 $(srcdir)/x-po.kdelnk $(DESTDIR)/$(KDE_DATA_DIR)/mimelnk/application; \
 	fi
 	if test -w $(KDE_DATA_DIR)/applnk/Development ; then \
-	  install -m 644 $(srcdir)/poedit.kdelnk $(KDE_DATA_DIR)/applnk/Development; \
+	  install -m 644 $(srcdir)/poedit.kdelnk $(DESTDIR)/$(KDE_DATA_DIR)/applnk/Development; \
 	fi
 
 else
diff -u install.orig/Makefile.in install/Makefile.in
--- install.orig/Makefile.in	Mon Nov  5 16:11:56 2001
+++ install/Makefile.in	Fri Jan 25 16:23:09 2002
@@ -189,17 +189,17 @@
 install-data-local: install-gnome install-kde
 
 @USE_KDE_TRUE@install-kde:
-@USE_KDE_TRUE@	mkdir -p $(KDE_DATA_DIR)/icons
-@USE_KDE_TRUE@	mkdir -p $(KDE_DATA_DIR)/mimelnk/application
-@USE_KDE_TRUE@	mkdir -p $(KDE_DATA_DIR)/applnk/Development
+@USE_KDE_TRUE@	mkdir -p $(DESTDIR)/$(KDE_DATA_DIR)/icons
+@USE_KDE_TRUE@	mkdir -p $(DESTDIR)/$(KDE_DATA_DIR)/mimelnk/application
+@USE_KDE_TRUE@	mkdir -p $(DESTDIR)/$(KDE_DATA_DIR)/applnk/Development
 @USE_KDE_TRUE@	if test -w $(KDE_DATA_DIR)/icons ; then \
-@USE_KDE_TRUE@	  install -m 644 $(top_srcdir)/src/appicon.xpm $(KDE_DATA_DIR)/icons/poedit.xpm; \
+@USE_KDE_TRUE@	  install -m 644 $(top_srcdir)/src/appicon.xpm $(DESTDIR)/$(KDE_DATA_DIR)/icons/poedit.xpm; \
 @USE_KDE_TRUE@	fi
 @USE_KDE_TRUE@	if test -w $(KDE_DATA_DIR)/mimelnk/application ; then \
-@USE_KDE_TRUE@	  install -m 644 $(srcdir)/x-po.kdelnk $(KDE_DATA_DIR)/mimelnk/application; \
+@USE_KDE_TRUE@	  install -m 644 $(srcdir)/x-po.kdelnk $(DESTDIR)/$(KDE_DATA_DIR)/mimelnk/application; \
 @USE_KDE_TRUE@	fi
 @USE_KDE_TRUE@	if test -w $(KDE_DATA_DIR)/applnk/Development ; then \
-@USE_KDE_TRUE@	  install -m 644 $(srcdir)/poedit.kdelnk $(KDE_DATA_DIR)/applnk/Development; \
+@USE_KDE_TRUE@	  install -m 644 $(srcdir)/poedit.kdelnk $(DESTDIR)/$(KDE_DATA_DIR)/applnk/Development; \
 @USE_KDE_TRUE@	fi
 
 @USE_KDE_FALSE@install-kde:
