--- sys/i386/include/intr_machdep.h.orig	2006-07-08 12:27:19 -0300
+++ sys/i386/include/intr_machdep.h	2006-07-08 12:26:49 -0300
@@ -55,7 +55,7 @@
  * - 1 counter for each CPU for lapic timer.
  * - 7 counters for each CPU for IPI counters for SMP.
  */
-#ifdef SMP
+#ifndef SMP
 #define	INTRCNT_COUNT	(1 + NUM_IO_INTS * 2 + 1)
 #else
 #define	INTRCNT_COUNT	(1 + NUM_IO_INTS * 2 + (1 + 7) * MAXCPU)
--- sys/amd64/include/intr_machdep.h.orig	2006-07-08 12:27:56 -0300
+++ sys/amd64/include/intr_machdep.h	2006-07-08 12:27:34 -0300
@@ -55,7 +55,7 @@
  * - 1 counter for each CPU for lapic timer.
  * - 7 counters for each CPU for IPI counters for SMP.
  */
-#ifdef SMP
+#ifndef SMP
 #define	INTRCNT_COUNT	(1 + NUM_IO_INTS * 2 + 1)
 #else
 #define	INTRCNT_COUNT	(1 + NUM_IO_INTS * 2 + (1 + 7) * MAXCPU)
