Index: fbsd-6.0/sys/conf/newvers.sh
===================================================================
--- fbsd-6.0.orig/sys/conf/newvers.sh
+++ fbsd-6.0/sys/conf/newvers.sh
@@ -31,13 +31,13 @@
 # $FreeBSD: src/sys/conf/newvers.sh,v 1.69.2.8.2.1 2005/11/02 02:00:46 scottl Exp $
 
 TYPE="FreeBSD"
-REVISION="6.0"
-BRANCH="RELEASE"
+REVISION="%GENTOOPVR%"
+BRANCH="Gentoo"
 if [ "X${BRANCH_OVERRIDE}" != "X" ]; then
 	BRANCH=${BRANCH_OVERRIDE}
 fi
 RELEASE=6.0-RELEASE
-VERSION="${TYPE} ${RELEASE}"
+VERSION="${TYPE} ${BRANCH} ${REVISION}"
 
 if [ "X${PARAMFILE}" != "X" ]; then
 	RELDATE=$(awk '/__FreeBSD_version.*propagated to newvers/ {print $3}' \
