diff -ru hydrogen-0.8.2-orig/lib/DiskWriterDriver.cpp hydrogen-0.8.2/lib/DiskWriterDriver.cpp
--- hydrogen-0.8.2-orig/lib/DiskWriterDriver.cpp	2004-06-18 20:19:14.803571648 +0200
+++ hydrogen-0.8.2/lib/DiskWriterDriver.cpp	2004-06-18 20:21:58.474689856 +0200
@@ -75,9 +75,9 @@
 
 	afFreeFileSetup(outputSetup);
 
-#if defined(i386) || defined(alpha)
+#if __BYTE_ORDER == __LITTLE_ENDIAN
 	afSetVirtualByteOrder(outputFile, AF_DEFAULT_TRACK, AF_BYTEORDER_LITTLEENDIAN);
-#else
+#elif __BYTE_ORDER == __BIG_ENDIAN
 	afSetVirtualByteOrder(outputFile, AF_DEFAULT_TRACK, AF_BYTEORDER_BIGENDIAN);
 #endif
 
Only in hydrogen-0.8.2/lib: DiskWriterDriver.cpp~
diff -ru hydrogen-0.8.2-orig/lib/Sample.cpp hydrogen-0.8.2/lib/Sample.cpp
--- hydrogen-0.8.2-orig/lib/Sample.cpp	2004-06-18 20:19:14.806571192 +0200
+++ hydrogen-0.8.2/lib/Sample.cpp	2004-06-18 20:22:58.517561952 +0200
@@ -143,9 +143,9 @@
 	int samplewidth;
 	afGetSampleFormat(samplefile, AF_DEFAULT_TRACK, &sampleformat, &samplewidth);
 
-#if defined(i386) || defined(alpha)
+#if __BYTE_ORDER == __LITTLE_ENDIAN
 	afSetVirtualByteOrder(samplefile, AF_DEFAULT_TRACK, AF_BYTEORDER_LITTLEENDIAN);
-#else
+#elif __BYTE_ORDER == __BIG_ENDIAN
 	afSetVirtualByteOrder(samplefile, AF_DEFAULT_TRACK, AF_BYTEORDER_BIGENDIAN);
 #endif
 
Only in hydrogen-0.8.2/lib: Sample.cpp~
