Index: hydrogen-0.9.3/configure.in
===================================================================
--- hydrogen-0.9.3.orig/configure.in
+++ hydrogen-0.9.3/configure.in
@@ -368,6 +368,8 @@ echo ""
 echo "--- Checking for PortAudio -------------------------------------"
 
 portaudio_support=false
+AC_ARG_ENABLE(portaudio, [  --disable-portaudio   Compile without portaudio support], [want_portaudio=$enableval])
+if test "x$want_portaudio" != "xno"; then
 AC_MSG_CHECKING([whether PORTAUDIOPATH environment variable is set])
 if test "x$PORTAUDIOPATH" = "x"; then
 	AC_MSG_RESULT(PORTAUDIOPATH is not set. No PortAudio support.)
@@ -390,6 +392,7 @@ fi
 echo "-----------------------------------------------------------------"
 echo ""
 
+fi
 
 
 
