--- tools/Makefile.in.orig	2006-05-07 03:14:53.000000000 +0300
+++ tools/Makefile.in	2006-05-07 03:15:51.000000000 +0300
@@ -249,7 +249,7 @@
 x_libraries = @x_libraries@
 INCLUDES = -I../lib $(all_includes)
 hydrogenPlayer_SOURCES = hydrogenPlayer.cpp
-hydrogenPlayer_LDADD = ../lib/libhydrogen.a ../lib/xml/libtinyxml.a ../lib/drivers/libdrivers.a ../lib/fx/libfx.a ../lib/smf/libsmf.a
+hydrogenPlayer_LDADD = ../lib/libhydrogen.a ../lib/xml/libtinyxml.a ../lib/drivers/libdrivers.a ../lib/fx/libfx.a ../lib/smf/libsmf.a $(USER_LDFLAGS)
 hydrogenPlayer_LDFLAGS = -ldl $(all_libraries)
 #>- all: all-am
 #>+ 1
--- hydrogen_gui/Makefile.in.orig	2006-05-07 03:15:59.000000000 +0300
+++ hydrogen_gui/Makefile.in	2006-05-07 03:16:12.000000000 +0300
@@ -368,7 +368,7 @@
 	../lib/drivers/libdrivers.a \
 	../lib/fx/libfx.a \
 	../lib/smf/libsmf.a \
-	$(LIB_QPE) $(LIB_QT) $(LIBSOCKET)
+	$(LIB_QPE) $(LIB_QT) $(LIBSOCKET) $(USER_LDFLAGS)
 
 EXTRA_DIST = \
 	AudioEngineInfoForm_UI.ui AudioEngineInfoForm.cpp AudioEngineInfoForm.h \
