diff -ru hydrogen-0.8.2-orig/lib/Hydrogen.cpp hydrogen-0.8.2/lib/Hydrogen.cpp
--- hydrogen-0.8.2-orig/lib/Hydrogen.cpp	2004-07-01 10:11:27.644571768 +0200
+++ hydrogen-0.8.2/lib/Hydrogen.cpp	2004-07-01 10:13:53.642376728 +0200
@@ -2502,7 +2502,7 @@
 
 	float fBPM = 60000.0 / fInterval;
 
-	if ( abs(fOldBpm1 - fBPM) > 20 ) {	// troppa differenza, niente media
+	if ( fabsf(fOldBpm1 - fBPM) > 20 ) {	// troppa differenza, niente media
 		fOldBpm1 = fBPM;
 		fOldBpm2 = fBPM;
 		fOldBpm3 = fBPM;
