http://bugs.digium.com/view.php?id=13277

diff --git a/kernel/zaptel-base.c b/kernel/zaptel-base.c
index 100ef6e..1940e40 100644
--- a/kernel/zaptel-base.c
+++ b/kernel/zaptel-base.c
@@ -163,7 +163,10 @@ static devfs_handle_t timer;
 /* udev necessary data structures.  Yeah! */
 #ifdef CONFIG_ZAP_UDEV
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,27)
+#define CLASS_DEV_CREATE(class, devt, device, name) \
+	device_create(class, device, devt, NULL, "%s", name)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
 #define CLASS_DEV_CREATE(class, devt, device, name) \
 	device_create(class, device, devt, name)
 #elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,15)
