--- verbiste-0.1.14/src/gnome/Makefile.am.orig	2006-05-01 12:47:30.000000000 +0200
+++ verbiste-0.1.14/src/gnome/Makefile.am	2006-05-01 12:48:34.000000000 +0200
@@ -41,14 +41,14 @@
 bonoboserver_DATA = $(PACKAGE).server
 
 $(PACKAGE).server: $(PACKAGE).server.pre
-	sed 's|%{libexecdir}|$(libexecdir)|g; \
-		s|%{PACKAGE}|$(PACKAGE)|g; \
-		s|%{APPLET_FULL_NAME}|$(APPLET_FULL_NAME)|g; \
-		s|%{APPLET_FULL_NAME_FR}|$(APPLET_FULL_NAME_FR)|g; \
-		s|%{APPLET_DESCRIPTION}|$(APPLET_DESCRIPTION)|g; \
-		s|%{APPLET_DESCRIPTION_FR}|$(APPLET_DESCRIPTION_FR)|g; \
-		s|%{APPLET_CATEGORY}|$(APPLET_CATEGORY)|g; \
-		s|%{APPLET_CATEGORY_FR}|$(APPLET_CATEGORY_FR)|g' \
+	sed -e 's|%{libexecdir}|$(libexecdir)|g' \
+		-e 's|%{PACKAGE}|$(PACKAGE)|g' \
+		-e 's|%{APPLET_FULL_NAME}|$(APPLET_FULL_NAME)|g' \
+		-e 's|%{APPLET_FULL_NAME_FR}|$(APPLET_FULL_NAME_FR)|g' \
+		-e 's|%{APPLET_DESCRIPTION}|$(APPLET_DESCRIPTION)|g' \
+		-e 's|%{APPLET_DESCRIPTION_FR}|$(APPLET_DESCRIPTION_FR)|g' \
+		-e 's|%{APPLET_CATEGORY}|$(APPLET_CATEGORY)|g' \
+		-e 's|%{APPLET_CATEGORY_FR}|$(APPLET_CATEGORY_FR)|g' \
 			< $< | grep -v '^#' > $@.tmp
 	mv -f $@.tmp $@
 
