diff -Naur ipmiutil-2.1.0.orig/lib/lanplus/Makefile.am ipmiutil-2.1.0/lib/lanplus/Makefile.am
--- ipmiutil-2.1.0.orig/lib/lanplus/Makefile.am	2008-04-22 16:24:25.000000000 +0200
+++ ipmiutil-2.1.0/lib/lanplus/Makefile.am	2008-04-22 16:38:40.000000000 +0200
@@ -1,7 +1,7 @@
 #  Makefile.am for libipmi_lanplus.so and libipmi_lanplus.a
 
 ODIR 	= obj
-CFLAGS_STATIC  = -DHAVE_CONFIG_H -g -O2 -fno-strict-aliasing -fPIC
+CFLAGS_STATIC  = -DHAVE_CONFIG_H -fno-strict-aliasing -fPIC $(CFLAGS)
 
 MAINTAINERCLEANFILES    = Makefile.in
  
@@ -32,12 +32,12 @@
 	rm -rf $(ODIR)
 
 $(lanplus_liba):	$(am_libipmi_lanplus_a_OBJECTS)
-	ar cru $(lanplus_liba) $(am_libipmi_lanplus_a_OBJECTS)
-	ranlib $(lanplus_liba)
+	$(AR) cru $(lanplus_liba) $(am_libipmi_lanplus_a_OBJECTS)
+	$(RANLIB) $(lanplus_liba)
 		
 $(ODIR):
 	mkdir -p $(ODIR)
 
 $(ODIR)/%.o:    %.c $(ODIR)
-	gcc -c $(CFLAGS_STATIC) $(INCLUDES) -o $@ $<
+	$(CC) -c $(CFLAGS_STATIC) $(INCLUDES) -o $@ $<
 
diff -Naur ipmiutil-2.1.0.orig/util/ipmilanplus.c ipmiutil-2.1.0/util/ipmilanplus.c
--- ipmiutil-2.1.0.orig/util/ipmilanplus.c	2008-04-22 16:24:25.000000000 +0200
+++ ipmiutil-2.1.0/util/ipmilanplus.c	2008-04-22 16:25:15.000000000 +0200
@@ -8,6 +8,7 @@
  */
 #include <stdio.h>
 #include <stdarg.h>
+#include <string.h>
 
 // #define DEBUG  1
 #ifndef HAVE_LANPLUS
diff -Naur ipmiutil-2.1.0.orig/util/ipmi_port.c ipmiutil-2.1.0/util/ipmi_port.c
--- ipmiutil-2.1.0.orig/util/ipmi_port.c	2008-04-22 16:24:25.000000000 +0200
+++ ipmiutil-2.1.0/util/ipmi_port.c	2008-04-22 16:26:05.000000000 +0200
@@ -12,6 +12,7 @@
  */
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <netinet/in.h>
diff -Naur ipmiutil-2.1.0.orig/util/Makefile.am ipmiutil-2.1.0/util/Makefile.am
--- ipmiutil-2.1.0.orig/util/Makefile.am	2008-04-22 16:24:25.000000000 +0200
+++ ipmiutil-2.1.0/util/Makefile.am	2008-04-22 16:35:55.000000000 +0200
@@ -54,26 +54,26 @@
 
 # To build an imb api library (libimbapi.a):
 libimbapi.a:	imbapi.c
-	gcc -DHAVE_CONFIG_H -I. -I.. -DLINUX -fPIC -g -O2 -c imbapi.c
-	ar cru libimbapi.a imbapi.o
-	ranlib libimbapi.a
+	$(CC) -DHAVE_CONFIG_H -I. -I.. -DLINUX -fPIC $(CFLAGS) -c imbapi.c
+	$(AR) cru libimbapi.a imbapi.o
+	$(RANLIB) libimbapi.a
 	rm -f imbapi.o
  
 # To build a lib to support all of the IPMI drivers (libipmiutil.a)
 libipmiutil.a:	$(CMDMOD:.c=.o) @LANPLUS_LIB@
-	ar cru libipmiutil.a $(CMDMOD:.c=.o) @LANPLUS_LIB@
-	ranlib libipmiutil.a
+	$(AR) cru libipmiutil.a $(CMDMOD:.c=.o) @LANPLUS_LIB@
+	$(RANLIB) libipmiutil.a
 
 # To build ipmiutil, need to use METACFLAGS for each .c/.o
 
 idiscover:	idiscover.c
-	gcc $(CFLAGS) -o idiscover idiscover.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o idiscover idiscover.c
 
 events:         events.c
-	gcc $(CFLAGS) -DTEST -o events events.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -DTEST -o events events.c
 
 ipmi_port:      ipmi_port.c
-	gcc $(CFLAGS) -o ipmi_port ipmi_port.c 
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ipmi_port ipmi_port.c 
 
 ipmiutil_SOURCES  = $(METASOURCE) 
 
diff -Naur ipmiutil-2.1.0.orig/util/Makefile.am-all ipmiutil-2.1.0/util/Makefile.am-all
--- ipmiutil-2.1.0.orig/util/Makefile.am-all	2008-04-22 16:24:25.000000000 +0200
+++ ipmiutil-2.1.0/util/Makefile.am-all	2008-04-22 16:39:04.000000000 +0200
@@ -42,7 +42,7 @@
 AM_CPPFLAGS += @LANPLUS_CFLAGS@ 
 LDADD += @LANPLUS_LDADD@ 
 
-METACFLAGS = -I. -I.. -DMETACOMMAND $(AM_CPPFLAGS)
+METACFLAGS = -I. -I.. -DMETACOMMAND $(AM_CPPFLAGS) $(CFLAGS)
 METASOURCE = ipmiutil.c alarms.c bmchealth.c events.c fruconfig.c getevent.c hwreset.c icmd.c pefconfig.c sensor.c showsel.c tmconfig.c wdt.c isolconsole.c idiscover.c $(CMDMOD) 
 METALD = @LANDESK_LDADD@ @LANPLUS_LIB@ @LANPLUS_CRYPTO@ -lpthread
 LDADD += -lpthread
@@ -52,35 +52,35 @@
 
 # To build an imb api library (libimbapi.a):
 libimbapi.a:	imbapi.c
-	gcc -DHAVE_CONFIG_H -I. -I.. -DLINUX -fPIC -g -O2 -c imbapi.c
-	ar cru libimbapi.a imbapi.o
-	ranlib libimbapi.a
+	$(CC) -DHAVE_CONFIG_H -I. -I.. -DLINUX -fPIC $(CFLAGS) -c imbapi.c
+	$(AR) cru libimbapi.a imbapi.o
+	$(RANLIB) libimbapi.a
 	rm -f imbapi.o
  
 # To build a lib to support all of the IPMI drivers (libipmiutil.a)
 libipmiutil.a:	$(CMDMOD:.c=.o) @LANPLUS_LIB@
-	ar cru libipmiutil.a $(CMDMOD:.c=.o) @LANPLUS_LIB@
-	ranlib libipmiutil.a
+	$(AR) cru libipmiutil.a $(CMDMOD:.c=.o) @LANPLUS_LIB@
+	$(RANLIB) libipmiutil.a
 
 # To build ipmiutil, need to use METACFLAGS for each .c/.o
 ipmiutil:	$(METASOURCE) 
 	@for f in $(METASOURCE); do \
-	    echo "gcc $(METACFLAGS) -c $$f"; \
-	    gcc $(METACFLAGS) -c $$f; done
+	    echo "$(CC) $(METACFLAGS) -c $$f"; \
+	    $(CC) $(METACFLAGS) -c $$f; done
 	@METAOBJ=`echo $(METASOURCE) | sed -e 's/ipmilan2.c//' | \
 			sed -e 's/.c$$/.o/g' -e 's/.c /.o /g'`; \
-	echo "gcc -g -O2 -o ipmiutil $$METAOBJ $(METALD)"; \
-	gcc -g -O2 -o ipmiutil $$METAOBJ $(METALD); \
+	echo "$(CC) -o ipmiutil $$METAOBJ $(METALD)"; \
+	$(CC) -o ipmiutil $$METAOBJ $(METALD); \
 	rm -f $$METAOBJ 
 
 idiscover:	idiscover.c
-	gcc $(CFLAGS) -o idiscover idiscover.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o idiscover idiscover.c
 
 events:         events.c
-	gcc $(CFLAGS) -DTEST -o events events.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -DTEST -o events events.c
 
 ipmi_port:      ipmi_port.c
-	gcc $(CFLAGS) -o ipmi_port ipmi_port.c 
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ipmi_port ipmi_port.c 
 
 ipmiutil_SOURCES  = 
 
