diff -ur wmalms-1.1.1.orig/configure.in wmalms-1.1.1/configure.in
--- wmalms-1.1.1.orig/configure.in	2003-11-01 11:32:16.000000000 +0200
+++ wmalms-1.1.1/configure.in	2008-01-13 06:44:19.000000000 +0200
@@ -18,7 +18,7 @@
 
 dnl Checking CFLAGS
 dnl if (test "${default_flags}" = yes); then
-CFLAGS="${CFLAGS} -m486 -malign-loops=2 -malign-jumps=2 -malign-functions=2 -Wall"
+CFLAGS="${CFLAGS} -Wall -fno-strict-aliasing"
 dnl fi
 
 dnl Checks for header files.
@@ -73,36 +73,6 @@
 	exit 1
 fi
 
-AC_MSG_CHECKING(if /proc/bus/i2c exists)
-if !(test -e /proc/bus/i2c); then
-	AC_MSG_RESULT(no)
-	echo 1>&5
-	echo "i2c driver does not appear to be running." 
-	echo "Install anyway ? (y, n)" 
-	read
-	if test "${REPLY}" != "y" && test "${REPLY}" != "Y"; then
-           exit 1
-        fi
-else
-	AC_MSG_RESULT(yes)
-	echo "Bus: " `cat /proc/bus/i2c`
-fi
-
-AC_MSG_CHECKING(if /proc/sys/dev/sensors/chips exists)
-if !(test -e /proc/sys/dev/sensors/chips); then
-	AC_MSG_RESULT(no)
-	echo 1>&5
-	echo "It appears to be no supported sensor chips."
-	echo "Install anyway ? (y, n)"
-	read
-	if test "${REPLY}" != "y" && test "${REPLY}" != "Y"; then
-           exit 1
-        fi
-else
-	AC_MSG_RESULT(yes)
-	echo "Sensor(-s): " `cat /proc/sys/dev/sensors/chips`
-fi
-
 x_parms="${x_lib_parms} ${x_inc_parms}"
 
 LIBS=""
