diff -ur avifile-0.7-0.7.43/m4/sdl.m4 avifile-0.7-0.7.43-fixed/m4/sdl.m4
--- avifile-0.7-0.7.43/m4/sdl.m4	2004-02-18 21:54:39.000000000 +0100
+++ avifile-0.7-0.7.43-fixed/m4/sdl.m4	2005-04-18 01:04:22.831940176 +0200
@@ -50,7 +50,7 @@
     no_sdl=yes
   else
     SDL_CFLAGS=`$SDL_CONFIG $sdlconf_args --cflags`
-    SDL_LIBS=`$SDL_CONFIG $sdlconf_args --libs | sed -e 's!-L/usr/lib[[^/]]!!g'`
+    SDL_LIBS=`$SDL_CONFIG $sdlconf_args --libs`
 
     sdl_major_version=`$SDL_CONFIG $sdl_args --version | \
            sed 's/\([[0-9]]*\).\([[0-9]]*\).\([[0-9]]*\)/\1/'`
