diff -ur a/lib/common/mmx.cpp avifile-0.7-0.7.43/lib/common/mmx.cpp
--- a/lib/common/mmx.cpp	2005-02-15 15:33:39.000000000 +0300
+++ avifile-0.7-0.7.43/lib/common/mmx.cpp	2005-03-23 22:09:48.730535320 +0300
@@ -122,12 +122,12 @@
 		 //   x_accum-=x_min;
 		 "subl   %6, %%edx	\n\t"
 		 //   if(x_accum<=0) {
-		 "jnc    x2		\n\t"
+		 "jnc    2f		\n\t"
 		 //      xaccum+=dst_w;
 		 "addl   %7, %%edx	\n\t"
 		 //         src++;
 		 "addl   $2, %%eax	\n\t"
-		 "x2:			\n\t"
+		 "2:			\n\t"
 		 "decl   %8		\n\t"
 		 "jnz    1b		\n\t"
 		 "popl   %%ebx		\n\t"
