--- src/java/test/org/jaxen/test/SimpleVariableContextTest.java	2006-10-27 02:24:14.000000000 +0200
+++ src/java/test/org/jaxen/test/SimpleVariableContextTest.java	2006-10-27 02:22:23.000000000 +0200
@@ -99,7 +99,7 @@
         
         // test the result
         assertEquals("String Value", copy.getVariableValue("", "", "s"));
-        assertEquals(Double.valueOf(3.1415292), copy.getVariableValue("", "", "x"));
+        assertEquals(new Double(3.1415292), copy.getVariableValue("", "", "x"));
         assertEquals(Boolean.TRUE, copy.getVariableValue("", "", "b"));
         assertEquals("", "");
         
@@ -116,7 +116,7 @@
         
         // test the result
         assertEquals("String Value", context.getVariableValue("", "", "s"));
-        assertEquals(Double.valueOf(3.1415292), context.getVariableValue("", "", "x"));
+        assertEquals(new Double(3.1415292), context.getVariableValue("", "", "x"));
         assertEquals(Boolean.TRUE, context.getVariableValue("", "", "b"));
         assertEquals("", "");
         
