diff -N -u -r freelords-0.3.7-old/freelordsrc.in freelords-0.3.7/freelordsrc.in
--- freelords-0.3.7-old/freelordsrc.in	2006-05-16 23:21:48.709485176 +0200
+++ freelords-0.3.7/freelordsrc.in	2006-05-16 23:22:10.712637169 +0200
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <freelordsrc version="@VERSION@">
     <d_datapath>@FREELORDS_DATADIR@</d_datapath>
-    <d_savepath>@HOME@/.freelords</d_savepath>
+
     <d_lang>C</d_lang>
     <d_resolution>@SCREEN_RESOLUTION@</d_resolution>
     <d_fullscreen>@USE_FULLSCREEN@</d_fullscreen>
