diff -urNp bcrypt.org/mod/Makefile.alg bcrypt/mod/Makefile.alg
--- bcrypt.org/mod/Makefile.alg	2006-04-12 08:32:39.000000000 +0300
+++ bcrypt/mod/Makefile.alg	2006-10-09 09:12:34.000000000 +0200
@@ -24,7 +24,7 @@ LD = ld
 LN_S = ln -s
 LDFLAGS = -d -r
 ifeq ($(K2616), yes)
-KBDEF += -DKBUILD_MODNAME=\"$(MODNAME)\" -DKBUILD_BASENAME=\"BestCrypt\"
+KBDEF += -DKBUILD_MODNAME=\"$(MODNAME)\" -DKBUILD_BASENAME=\"BestCrypt\" -D_KBUILD_QUOTES
 else
 KBDEF += -DKBUILD_MODNAME=$(MODNAME) -DKBUILD_BASENAME=BestCrypt
 endif
