diff -ru /tmp/bcrypt/kgsha/Makefile bcrypt/kgsha/Makefile
--- /tmp/bcrypt/kgsha/Makefile	2005-05-14 16:50:36.000000000 +1000
+++ bcrypt/kgsha/Makefile	2005-05-23 22:30:00.000000000 +1000
@@ -20,6 +20,7 @@
 
 CPP=g++
 CPPFLAGS = -Wall -Werror -fno-strength-reduce -I../include 
+CPPFLAGS += $(EXTRA_CXXFLAGS)
 
 TARGET = libkgsha.so
 HEADERS = kg_defs.h kgsha.h kblock.h sha1.h random.h
diff -ru /tmp/bcrypt/mod/Makefile bcrypt/mod/Makefile
--- /tmp/bcrypt/mod/Makefile	2005-05-23 22:26:09.000000000 +1000
+++ bcrypt/mod/Makefile	2005-05-23 22:28:52.000000000 +1000
@@ -101,15 +101,7 @@
 
 CFLAGS += -I$(CUR_DIR)/../include/ -I$(KERNEL_DIR)/include/
 
-ifeq ($(ARCH), alpha)
-CFLAGS += -O2 -pipe -fno-strict-aliasing -fno-common -mno-fp-regs -ffixed-8 -mcpu=ev5
-else
-ifeq ($(ARCH), x86_64)
-CFLAGS += -O2 -fno-strength-reduce -fno-strict-aliasing -mcmodel=kernel
-else
-CFLAGS += -O6 -fno-strength-reduce -fno-strict-aliasing 
-endif
-endif
+CFLAGS += $(EXTRA_CFLAGS)
 
 LDFLAGS = -d -r
 
diff -ru /tmp/bcrypt/src/Makefile bcrypt/src/Makefile
--- /tmp/bcrypt/src/Makefile	2005-05-14 16:53:06.000000000 +1000
+++ bcrypt/src/Makefile	2005-05-23 22:29:31.000000000 +1000
@@ -24,6 +24,7 @@
 CFLAGS += -L../lib -I../include/ 
 CFLAGS += -Wall -fno-strict-aliasing -g # -O2
 #CFLAGS += -g -static -O6
+CFLAGS += $(EXTRA_CFLAGS)
 LDFLAGS = -ldl -lstdc++
 
 TARGET = bctool
