diff -Naur Ultimate2.8.7.old/src/Makefile Ultimate2.8.7/src/Makefile
--- Ultimate2.8.7.old/src/Makefile	2003-10-21 18:13:19.000000000 +0000
+++ Ultimate2.8.7/src/Makefile	2003-10-21 18:13:58.000000000 +0000
@@ -61,7 +61,7 @@
 	$(CC) $(CFLAGS) -c version.c
 
 ircd: $(OBJS)
-	$(CC) $(CFLAGS) $(OBJS) $(LDFLAGS) $(IRCDLIBS) -o ircd
+	$(CC) $(CFLAGS) $(OBJS) $(LDFLAGS) -lresolv -lcrypt -o ircd
 	chmod $(IRCDMODE) ircd
 
 saber: $(SRC)
