--- ark/tar.cpp	2004/03/30 00:20:33	1.24.2.1
+++ ark/tar.cpp	2004/04/16 20:30:28	1.24.2.2
@@ -81,7 +81,7 @@ TarArch::TarArch( ArkSettings *_settings
 
     m_fileMimeType = _openAsMimeType;
     if ( m_fileMimeType.isNull() )
-        m_fileMimeType = KMimeType::findByFileContent( _filename )->name();
+        m_fileMimeType = KMimeType::findByPath( _filename )->name();
 
     kdDebug(1601) << "TarArch::TarArch:  mimetype is " << m_fileMimeType << endl;
 
