diff -Npaur kdeutils-3.3.91/ksim/monitors/snmp/snmplib.h kdeutils-3.3.91-fixed/ksim/monitors/snmp/snmplib.h
--- kdeutils-3.3.91/ksim/monitors/snmp/snmplib.h	2003-08-29 11:56:50.000000000 +0200
+++ kdeutils-3.3.91-fixed/ksim/monitors/snmp/snmplib.h	2005-01-15 12:03:30.497268728 +0100
@@ -21,6 +21,7 @@
 #define SNMPLIB_H
 
 // ### REMOVEME: move snmplib somewhere else
+#define HAVE_STRLCPY 1
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 
diff -Npaur kdeutils-3.3.91/ksim/monitors/snmp/value_p.h kdeutils-3.3.91-fixed/ksim/monitors/snmp/value_p.h
--- kdeutils-3.3.91/ksim/monitors/snmp/value_p.h	2003-08-27 15:51:50.000000000 +0200
+++ kdeutils-3.3.91-fixed/ksim/monitors/snmp/value_p.h	2005-01-15 12:03:44.323983848 +0100
@@ -20,6 +20,7 @@
 #ifndef VALUE_P_H
 #define VALUE_P_H
 
+#define HAVE_STRLCPY 1
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 
