--- bin/dlzbdb/Makefile.in	2006-03-25 22:11:15.840755988 +0000
+++ bin/dlzbdb/Makefile.in	2006-03-25 22:33:43.873082259 +0000
@@ -26,6 +26,9 @@
 DLZINCLUDES =	@DLZ_BDB_INC@
 
 CINCLUDES =	-I${srcdir}/include -I${srcdir}/unix/include \
+		-I${top_srcdir}/lib/isc/include \
+		-I${top_srcdir}/lib/isc/unix/include \
+		-I${top_srcdir}/lib/isc/@ISC_THREAD_DIR@/include \
                 ${ISC_INCLUDES} ${DLZINCLUDES}
                 
 CDEFINES =      @USE_DLZ_BDB@
