--- a/Makefile	2010-08-20 20:27:26.000000000 +0200
+++ b/Makefile	2010-09-18 04:12:21.000000000 +0200
@@ -64,8 +64,8 @@
 include snmp.mk
 
 ## Common
-CFLAGS        = ${MCAST_INCLUDE} ${SNMPDEF} ${RSRRDEF} $(INCLUDES) $(DEFS) $(USERCOMPILE)
-CFLAGS       += -O2 -W -Wall -Werror
+CFLAGS       += ${MCAST_INCLUDE} ${SNMPDEF} ${RSRRDEF} $(INCLUDES) $(DEFS) $(USERCOMPILE) \
+					-W -Wall
 #CFLAGS       += -O -g
 LDLIBS        = ${SNMPLIBDIR} ${SNMPLIBS} ${LIB2}
 OBJS          = ${IGMP_OBJS} ${ROUTER_OBJS} ${MAPPER_OBJS} ${MRINFO_OBJS} \
