diff -ru ./Makefile.in ./Makefile.in
--- ./Makefile.in	2006-11-06 00:54:07.000000000 +0100
+++ ./Makefile.in	2006-11-24 22:01:04.000000000 +0100
@@ -36,7 +36,8 @@
 GBCFLAGS=$(EXTRA_CFLAGS) $(DEBUGGING) -I. -I@srcdir@/coldsync \
 	$(OPTIMIZATION) @CFLAGS@
 LDFLAGS=$(EXTRA_LDFLAGS) @LDFLAGS@
-DESTDIR=/usr/local/
+PREFIX=@prefix@
+INSTALL_DIR=$(DESTDIR)/$(PREFIX)
 
 # OTHER_ROOT=/opt/local	# For DarwinPorts on OSX
 # OTHER_ROOT=/sw		# Uncomment For Fink on OS X.
@@ -136,10 +137,12 @@
 # This will only work on UNIX-like substances.
 #
 install: @INSTALL_DEBUG@
-	install gpsbabel  $(DESTDIR)/bin
+	@mkdir -p $(INSTALL_DIR)/bin
+	install gpsbabel  $(INSTALL_DIR)/bin/
 
 install-debug:
-	install gpsbabel-debug  $(DESTDIR)/bin
+	@mkdir -p $(INSTALL_DIR)/bin
+	install gpsbabel-debug  $(INSTALL_DIR)/bin/
 
 # Nerdy release stuff that needs to work only on Linux.
 
