--- pvm3/lib/pvmgetarch.orig	Thu Sep 27 17:24:04 2001
+++ pvm3/lib/pvmgetarch	Wed Feb 19 19:44:26 2003
@@ -66,13 +66,16 @@
 	*,88k )                 ARCH=E88K ;;
 	*,mips )                ARCH=MIPS ;;
 	*,CRAY-2 )              ARCH=CRAY2 ;;
-	Linux,i[3456]86 )       ARCH=LINUX ;;
-	Linux,ia64 )            ARCH=LINUX64 ;;
+	Linux,i[3456]86 )       ARCH=LINUXI386 ;;
+	Linux,x86_64 )		ARCH=LINUXX86_64 ;;
+	Linux,ia64 )            ARCH=LINUXIA64 ;;
 	Linux,alpha )           ARCH=LINUXALPHA ;;
 	Linux,arm* )            ARCH=LINUXARM ;;
 	Linux,sparc* )          ARCH=LINUXSPARC ;;
 	Linux,hp_pa )           ARCH=LINUXHPPA ;;
 	Linux,ppc )             ARCH=LINUXPPC ;;
+	Linux,s390 )            ARCH=LINUXS390 ;;
+	Linux,s390x )           ARCH=LINUXS390X ;;
 	Linux,m68k )            ARCH=LINUXATARI ;;
 	BSD/OS,i[3456]86 )      ARCH=BSD386 ;;
 	FreeBSD,i[3456]86 )     ARCH=FREEBSD ;;
