diff -uNr pvm3/conf/LINUXAMD64.def ../../pvm-3.4.4-r1.old/work/pvm3/conf/LINUXAMD64.def
--- pvm3/conf/LINUXAMD64.def	1969-12-31 16:00:00.000000000 -0800
+++ pvm3/conf/LINUXAMD64.def	2004-09-28 11:34:03.557014346 -0700
@@ -0,0 +1,24 @@
+#
+# Notes on readline Usage:
+#
+#	If your Linux system has readline, then you may append the
+#	following flags to the defines below:
+#
+#		ARCHCFLAGS = . . .  -DHASREADLINE
+#		ARCHLIB = . . .  -lreadline
+#
+#	and recompile PVM.
+#
+#	On some Linux systems, readline requires ncurses, so you
+#	must also add "-lncurses" to the ARCHLIB define.
+#
+ARCHCFLAGS	=	-DSYSVSIGNAL -DNOWAIT3 -DRSHCOMMAND=\"/usr/bin/rsh\" \
+				-DNEEDENDIAN -DFDSETNOTSTRUCT -DHASERRORVARS \
+				-DCTIMEISTIMET -DSYSERRISCONST -DNOTMPNAM
+ARCHDLIB	=
+ARCHDOBJ	=
+ARCHLIB		=
+HASRANLIB	=	t
+AR			=	ar
+PVM_ARCH	=	LINUX
+MAKE		=	make
diff -uNr pvm3/conf/LINUXAMD64.m4 ../../pvm-3.4.4-r1.old/work/pvm3/conf/LINUXAMD64.m4
--- pvm3/conf/LINUXAMD64.m4	1969-12-31 16:00:00.000000000 -0800
+++ pvm3/conf/LINUXAMD64.m4	2004-09-28 11:34:14.110914614 -0700
@@ -0,0 +1,19 @@
+divert(-1)
+undefine(`len')
+#
+# append an underscore to FORTRAN function names
+#
+define(`FUNCTION',`$1_')
+#
+# FORTRAN character strings are passed as follows:
+# a pointer to the base of the string is passed in the normal
+# argument list, and the length is passed by value as an extra
+# argument, after all of the other arguments.
+#
+define(`ARGS',`($1`'undivert(1))')
+define(`SAVE',`divert(1)$1`'divert(0)')
+define(`STRING_ARG',`$1_ptr`'SAVE(`, $1_len')')
+define(`STRING_ARG_DECL',`char * $1_ptr; int $1_len')
+define(`STRING_LEN',`$1_len')
+define(`STRING_PTR',`$1_ptr')
+divert(0)
diff -uNr pvm3/lib/pvmgetarch ../../pvm-3.4.4-r1.old/work/pvm3/lib/pvmgetarch
--- pvm3/lib/pvmgetarch	2001-09-27 14:24:04.000000000 -0700
+++ pvm3/lib/pvmgetarch	2004-09-28 11:32:18.944915087 -0700
@@ -67,6 +67,7 @@
 	*,mips )                ARCH=MIPS ;;
 	*,CRAY-2 )              ARCH=CRAY2 ;;
 	Linux,i[3456]86 )       ARCH=LINUX ;;
+	Linux,x86_64 )          ARCH=LINUXAMD64 ;;
 	Linux,ia64 )            ARCH=LINUX64 ;;
 	Linux,alpha )           ARCH=LINUXALPHA ;;
 	Linux,arm* )            ARCH=LINUXARM ;;
