diff -Nurb Welcome2L-3.04/ChangeLog Welcome2L-3.04-patched/ChangeLog
--- Welcome2L-3.04/ChangeLog	Mon Jun 28 22:56:27 1999
+++ Welcome2L-3.04-patched/ChangeLog	Wed Oct 30 14:11:33 2002
@@ -1,6 +1,13 @@
 WELCOME2L - CHANGES HISTORY
 ===========================
 
+v3.04-gentoo 30-OKT-02
+----------------------
+
+- Added a lot of patches for the Gentoo Linux ebuild. Most
+  patches are taken from the Debian package.
+
+
 v3.04 28-JUN-99
 ---------------
 
diff -Nurb Welcome2L-3.04/Makefile Welcome2L-3.04-patched/Makefile
--- Welcome2L-3.04/Makefile	Mon Jun 28 22:56:27 1999
+++ Welcome2L-3.04-patched/Makefile	Wed Oct 30 14:13:56 2002
@@ -1,7 +1,6 @@
 SHELL=/bin/sh
 CC = gcc
-CFLAGS = -Wall -O2 -fno-strength-reduce #-pedantic 
-DESTDIR = /usr/local
+DESTDIR = /usr
 INSTALL_BINPATH = $(DESTDIR)/bin
 INSTALL_MANPATH = $(DESTDIR)/man
 PROGNAME = Welcome2L
@@ -12,13 +11,12 @@
 RPM_BUILDPATH = /usr/src/redhat
 #RPM_ICONNAME = 
 
-
 $(PROGNAME): main.o ansi.o 	     
 	$(CC) $(CFLAGS) -v *.o -o $(PROGNAME)
 	strip ./$(PROGNAME)	
 
 install : 
-	/usr/bin/install -s -m 755 $(PROGNAME) $(INSTALL_BINPATH)
+	/usr/bin/install -m 755 $(PROGNAME) $(INSTALL_BINPATH)
 	/usr/bin/install -m 644 $(PROGNAME).1 $(INSTALL_MANPATH)/man1
 	ln -fs $(PROGNAME) $(INSTALL_BINPATH)/$(PROGNAME_LNK)
 	ln -fs $(PROGNAME).1 $(INSTALL_MANPATH)/man1/$(PROGNAME_LNK).1
diff -Nurb Welcome2L-3.04/Welcome2L.1 Welcome2L-3.04-patched/Welcome2L.1
--- Welcome2L-3.04/Welcome2L.1	Mon Jun 28 22:56:27 1999
+++ Welcome2L-3.04-patched/Welcome2L.1	Wed Oct 30 14:31:59 2002
@@ -194,7 +194,7 @@
 Welcome2l \- Linux ANSI boot logo.
 .SH "SYNOPSIS"
 .IX Header "SYNOPSIS"
-\fBWelcome2L\fR [ \fB\-scr\fR ] [ \fB\-getty\fR ] [ \fB\-msg\fR ] [ \fB\-cpu\fR ] [ \fB\-lcol\fR ]  [ \fB\-nolf\fR ] [ \fB\-private\fR ]
+\fBWelcome2L\fR [ \fB\-scr\fR ] [ \fB\-getty\fR ] [ \fB\-gettps\fR] [ \fB\-msg\fR ] [ \fB\-cpu\fR ]  [ \fB-scrrand\fR ] [ \fB-xmas\fR ]  [ \fB\-xmasauto\fR ] [ \fB\-lcol\fR ] [ \fB\-nolf\fR ] [ \fB\-private\fR ] [\fB\-notime\fR \]
 .SH "DESCRIPTION"
 .IX Header "DESCRIPTION"
 \fBWelcome2L\fR is a little program that may run at login time to produce a BBS
@@ -209,6 +209,9 @@
 .Ip "\fB\-getty\fR" 8
 .IX Item "\fB\-getty\fR"
 Will display more accurate information when launched at boot time.
+.Ip "\fB\-gettps\fR" 8
+.IX Item "\fB\-gettps\fR"
+Like \fB\-getty\fR, but works with \fIgettyps\fR(8) program.
 .Ip "\fB\-nolf\fR" 8
 .IX Item "\fB\-nolf\fR"
 Will not add an empty line to the ansi screen produced.
@@ -224,6 +227,12 @@
 .Ip "\fB\-scrrand\fR" 8
 .IX Item "\fB\-scrrand\fR"
 Will display a random screen. 
+.Ip "\fB\-xmas\fR" 8
+.IX Item "\fB\-xmasauto\fR"
+Will display an Xmas screen.
+.Ip "\fB\-xmasauto\fR" 8
+.IX Item "\fB\-xmasauto\fR"
+Will automatically display an Xmas screen in December and January.
 .Ip "\fB\-lcol\fRxx" 8
 .IX Item "\fB\-lcol\fRxx"
 Will change the prompt color to the color number xx. Type Welcome2L \-help to see
@@ -231,107 +240,16 @@
 .Ip "\fB\-private\fR" 8
 .IX Item "\fB\-private\fR"
 Display an \*(L"Unauthorized access is prohibited\*(R" line at the bottom 
-of the \s-1ANSI\s0 screen
+of the \s-1ANSI\s0 screen.
+.Ip "\fB\-notime\fR" 8
+.IX Item "\fB\-notime\fR"
+When dispalying  date information, do not display time.
 .Ip "\fB\-help\fR" 8
 .IX Item "\fB\-help\fR"
 Will display usage information. 
 .SH "EXAMPLE"
 .IX Header "EXAMPLE"
-\&./Welcome2L \-getty \-msg"RedHat Linux\*(R" \-lcol14
- 
-.SH "INSTALLATION"
-.IX Header "INSTALLATION"
-.Sh "On RedHat systems "
-.IX Subsection "On RedHat systems "
-cd to /etc/rc.d/
-edit rc.local, near the end find these lines (may be not exactly the same) :
-.PP
-.Vb 3
-\&    echo "" > /etc/issue
-\&    echo "Red Hat Linux $R" >> /etc/issue
-\&    echo "Kernel $(uname r) on $a $(uname m)" >> /etc/issue
-.Ve
-comment them out and add this one (change the version numbers to yours) :
-.PP
-.Vb 1
-\&    /usr/local/bin/Welcome2L -msg"Red Hat 5.0 (Hurricane)" > /etc/issue
-.Ve
-.Sh "On Debian systems "
-.IX Subsection "On Debian systems "
-[ From: Gerd Bavendiek <bav@rw.sni.de> ]
-   I put a script in
-   lulu:/etc/rc.boot> ls
-   0setserial  1netenv     2hdparm  3Welcome2L
-    
-    It's contents is just:    
-   #!/bin/sh
-    /usr/local/bin/Welcome2L \-getty \-msg"Debian \s-1GNU/\s0Linux 2.0\*(R" > /etc/issue
-.Sh "On Slackware systems"
-.IX Subsection "On Slackware systems"
-[ From: Joe X <obliteration@hotmail.com> ]
-  add these lines in /etc/rc.d/rc.S
-  /usr/local/bin/Welcome2L > /etc/issue
-  /usr/local/bin/Welcome2L > /etc/issue.net
-.Sh "On \s-1SUSE\s0 5.3 system:"
-.IX Subsection "On \s-1SUSE\s0 5.3 system:"
-[ From: Ronny Ziegler <ronny.ziegler@usa.net> ]
-.PP
-.Vb 1
-\& 1) Put the following script called logo into /sbin/init.d
-.Ve
-.Vb 21
-\& #! /bin/sh
-\& #
-\& #Author: Ronny Ziegler <ronny.ziegler@usa.net>, 1998
-\& #
-\& # /sbin/init.d/logo
-\& #
-\& case "$1" in
-\&     start)
-\&         echo "Starting  logo"
-\&         /usr/local/bin/Welcome2L
-\&         /bin/sleep 1
-\&         ;;
-\&     stop)
-\&         ;;
-\&     reload|restart)
-\&         ;;
-\&     *)
-\&         echo "Usage: $0 {start|stop|reload|restart}"
-\&         exit 1
-\& esac
-\& exit 0
-.Ve
-.Vb 7
-\& 2) cd /sbin/init.d/rc2.d
-\& 3) mv S99zzreached S98zzreached
-\& 4) ln -s ../logo S99logo
-\&    Now the logo will be executed as last before login-prompt
-\& 5) You don't need the old Login-Text any more:
-\&     mv /etc/issue /etc/issue.orig
-\&     touch /etc/issue
-.Ve
-.Sh "On \s-1SUSE\s0 6.0 system:"
-.IX Subsection "On \s-1SUSE\s0 6.0 system:"
-[ From: Ronny Ziegler <ronny.ziegler@usa.net> ]
-.PP
-You should have an error-free bootprocess before installing Welcome2L.
- 
- 1) Add the following in your /sbin/init.d/rc file as 
-    the line before last (that´s the one with exit 0)  
-        /usr/local/bin/Welcome2L
-    where this is the path to your Welcome2L\-executeable
-.PP
-.Vb 4
-\& 1b)Optional: If you dont´t like Welcome2L to overwrite your 
-\&    boot-messages, add 
-\&        echo -e \e\ev\e\ev\e\ev\e\ev\e\ev\e\ev
-\&    (or some more \e\ev ´s) in you rc before starting Welcome2L.  
-.Ve
-.Vb 3
-\& 2) Now you don´t need the old Login-Text any more:
-\&    mv /etc/issue /etc/issue.orig
-\&    touch /etc/issue
+\&/usr/bin/Welcome2L \-getty \-msg"Gentoo Linux\*(R" \-lcol14
 .Ve
 .SH "RESTRICTIONS"
 .IX Header "RESTRICTIONS"
diff -Nurb Welcome2L-3.04/ansi/welcome1.h Welcome2L-3.04-patched/ansi/welcome1.h
--- Welcome2L-3.04/ansi/welcome1.h	Mon Jun 28 22:56:27 1999
+++ Welcome2L-3.04-patched/ansi/welcome1.h	Wed Oct 30 13:54:48 2002
@@ -37,7 +37,7 @@
   'Ü', 0x10, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
   ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
   ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
-  ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, 'þ', 0x08, 'Ü', 0x08,
+  ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x08, 'Ü', 0x08,
   'Ü', 0x08, ' ', 0x08, ' ', 0x08, ' ', 0x08, 'ß', 0x01, 'Û', 0x01,
   'Û', 0x01, 'Û', 0x01, 'ß', 0x10, 'Û', 0x10, 'Û', 0x10, 'Û', 0x10,
   'ß', 0x10, ' ', 0x10, ' ', 0x10, ' ', 0x10, 'Û', 0x11, 'Û', 0x11,
@@ -101,7 +101,7 @@
   'Û', 0x10, 'Ü', 0x10, ' ', 0x10, 'Ü', 0x10, 'Û', 0x10, 'Û', 0x10,
   'Û', 0x10, ' ', 0x10, 'Û', 0x10, 'Û', 0x10, 'Û', 0x10, 'Ü', 0x10,
   ' ', 0x10, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
-  ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, 'Û', 0x00, 'þ', 0x08, 'Ü', 0x08,
+  ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, 'Û', 0x00, 'ß', 0x08, 'Ü', 0x08,
   'Ü', 0x08, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
   ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
   ' ', 0x0F, 'ß', 0x08, 'Ü', 0x08, 'Ü', 0x08, ' ', 0x0F, ' ', 0x0F,
@@ -118,7 +118,7 @@
   'ß', 0x78, 'Ü', 0x08, 'ß', 0x08, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
   ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, 'Ü', 0x07, 'ß', 0x78, 'Û', 0x07,
   'Û', 0x07, 'Û', 0x07, 'Û', 0x07, 'ß', 0x78, 'Ü', 0x07, 'ß', 0x08,
-  'þ', 0x08, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
+  ' ', 0x08, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
   ' ', 0x0F, ' ', 0x0F, ' ', 0x18, 'Ü', 0x18, 'Ü', 0x18, 'Ü', 0x18,
   'Ü', 0x18, 'Ü', 0x18, 'Ü', 0x18, 'Ü', 0x18, 'Ü', 0x18, 'Ü', 0x18,
   'Ü', 0x18, 'Ü', 0x18, 'Ü', 0x18, 'Ü', 0x18, 'Ü', 0x18, 'Ü', 0x18,
@@ -154,10 +154,10 @@
   ' ', 0x08, ' ', 0x08, ' ', 0x08, ' ', 0x08, ' ', 0x08, ' ', 0x08,
   'Û', 0x18, 'Û', 0x10, ' ', 0x10, ' ', 0x10, ' ', 0x10, ' ', 0x10,
   ' ', 0x10, ' ', 0x10, ' ', 0x10, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
-  'Ý', 0x78, 'Ý', 0x7F, ' ', 0x0F, ' ', 0x0F, 'þ', 0x0F, 'ß', 0x07,
+  'Ý', 0x78, 'Ý', 0x7F, ' ', 0x0F, ' ', 0x0F, 'ß', 0x0F, ' ', 0x07,
   ' ', 0x07, 'Þ', 0x0F, '°', 0x7F, 'Ý', 0x08, 'ß', 0x08, 'ß', 0x08,
-  'þ', 0x08, 'Ý', 0x78, 'Þ', 0x7F, 'Ý', 0x7F, ' ', 0x0F, ' ', 0x0F,
-  ' ', 0x0F, 'þ', 0x0F, 'ß', 0x07, ' ', 0x07, 'Þ', 0x7F, 'Û', 0x0F,
+  ' ', 0x08, 'Ý', 0x78, 'Þ', 0x7F, 'Ý', 0x7F, ' ', 0x0F, ' ', 0x0F,
+  ' ', 0x0F, 'ß', 0x0F, ' ', 0x07, ' ', 0x07, 'Þ', 0x7F, 'Û', 0x0F,
   'Þ', 0x78, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
   ' ', 0x0F, ' ', 0x0F, ' ', 0x18, 'Û', 0x08, ' ', 0x08, ':', 0x08,
   ' ', 0x08, ' ', 0x08, 'W', 0x08, 'e', 0x08, 'l', 0x08, 'c', 0x08,
diff -Nurb Welcome2L-3.04/ansi/welcome2.h Welcome2L-3.04-patched/ansi/welcome2.h
--- Welcome2L-3.04/ansi/welcome2.h	Mon Jun 28 22:56:27 1999
+++ Welcome2L-3.04-patched/ansi/welcome2.h	Wed Oct 30 13:54:48 2002
@@ -64,7 +64,7 @@
     'Û', 0x10,  'Û', 0x01,  'Û', 0x10,  '±', 0x09,  'Û', 0x10,  'Û', 0x01,
     'Ü', 0x01,  ' ', 0x00,  ' ', 0x07,  'Û', 0x00,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
-    ' ', 0x07,  ' ', 0x07,  'þ', 0x08,  'ß', 0x08,  'ß', 0x08,  'ß', 0x08,
+    ' ', 0x07,  ' ', 0x07,  'ß', 0x08,  'ß', 0x08,  'ß', 0x08,  'ß', 0x08,
     'ß', 0x08,  'Ü', 0x08,  'Ü', 0x08,  'Ü', 0x08,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  'ß', 0x08,  'ß', 0x08,  'Ü', 0x18,  'Ü', 0x18,
     'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,
@@ -77,9 +77,9 @@
     'ß', 0x10,  'Û', 0x10,  'Û', 0x10,  'Ü', 0x10,  'Û', 0x00,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
-    ' ', 0x07,  ' ', 0x07,  'þ', 0x08,  ' ', 0x07,  ' ', 0x07,  'Ü', 0x08,
+    ' ', 0x07,  ' ', 0x07,  'ß', 0x08,  ' ', 0x07,  ' ', 0x07,  'Ü', 0x08,
     'ß', 0x78,  'Ü', 0x08,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  'ß', 0x08,
-    'ß', 0x08,  'þ', 0x08,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
+    'ß', 0x08,  'ß', 0x08,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  'ß', 0x08,  'Ü', 0x18,  'Û', 0x01,  'Û', 0x01,
     'Û', 0x01,  'Û', 0x01,  ' ', 0x10,  'Û', 0x10,  '°', 0x09,  'Û', 0x10,
     ' ', 0x10,  ' ', 0x10,  ' ', 0x10,  'Ü', 0x10,  ' ', 0x10,  'Þ', 0x4D,
@@ -101,7 +101,7 @@
     'Ü', 0x10,  'Û', 0x10,  'Û', 0x10,  'Ü', 0x10,  ' ', 0x10,  'ß', 0x10,
     'Û', 0x10,  'Û', 0x10,  'Ü', 0x10,  'Ü', 0x10,  'Û', 0x10,  'Û', 0x10,
     'Û', 0x10,  'Ü', 0x10,  ' ', 0x10,  'Ü', 0x10,  'Û', 0x10,  'Û', 0x10,
-    'Û', 0x10,  'þ', 0x10,  'Û', 0x00,  'Û', 0x10,  'Û', 0x00,  ' ', 0x07,
+    'Û', 0x10,  'ß', 0x10,  'Û', 0x00,  'Û', 0x10,  'Û', 0x00,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  'Ü', 0x07,  'Ü', 0x7F,
     'Û', 0x0F,  'Û', 0x0F,  'Û', 0x0F,  'Û', 0x0F,  'ß', 0x0F,  'ß', 0x0F,
@@ -115,7 +115,7 @@
     'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,
     'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,
     'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  'Û', 0x01,  '°', 0x01,
-    ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x08,  'ß', 0x08,  'þ', 0x08,
+    ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x08,  'ß', 0x08,  'ß', 0x08,
     ' ', 0x08,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  'Ü', 0x07,  'Ü', 0x7F,
     'Û', 0x0F,  'Û', 0x0F,  'Û', 0x0F,  'Û', 0x7F,  'ß', 0x0F,  'ß', 0x18,
     '±', 0x19,  '±', 0x19,  '±', 0x19,  '°', 0x19,  '±', 0x19,  'ß', 0x18,
@@ -129,7 +129,7 @@
     'Ü', 0x18,  'Ü', 0x18,  'Ü', 0x18,  'Ü', 0x18,  'Ü', 0x18,  'Ü', 0x18,
     'Ü', 0x18,  'Ü', 0x18,  'Ü', 0x18,  'Ü', 0x18,  'Û', 0x01,  'Û', 0x01,
     '°', 0x01,  ' ', 0x07,  'Ü', 0x08,  'ß', 0x78,  'Ü', 0x08,  ' ', 0x07,
-    ' ', 0x08,  ' ', 0x08,  'þ', 0x08,  ' ', 0x07,  ' ', 0x07,  'Þ', 0x07,
+    ' ', 0x08,  ' ', 0x08,  'ß', 0x08,  ' ', 0x07,  ' ', 0x07,  'Þ', 0x07,
     'Û', 0x0F,  'Û', 0x0F,  'Û', 0x0F,  'Û', 0x0F,  'ß', 0x0F,  'ß', 0x10,
     'ß', 0x01,  'Ü', 0x10,  'Ü', 0x10,  'Ü', 0x1F,  'Û', 0x0F,  'Û', 0x0F,
     'Û', 0x0F,  'Ü', 0x1F,  'Û', 0x11,  '²', 0x01,  'Ý', 0x01,  'Û', 0x0F,
@@ -144,7 +144,7 @@
     'Û', 0x10,  'Û', 0x61,  ' ', 0x07,  '²', 0x08,  '±', 0x7F,  'ß', 0x0F,
     '±', 0x01,  '°', 0x01,  'Û', 0x00,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
     'Þ', 0x08,  'Þ', 0x7F,  'Û', 0x0F,  'Û', 0x0F,  'Û', 0x0F,  'Û', 0x0F,
-    'Ý', 0x18,  'Ü', 0x19,  'Ý', 0x08,  'ß', 0x08,  'þ', 0x0F,  'Û', 0x00,
+    'Ý', 0x18,  'Ü', 0x19,  'Ý', 0x08,  'ß', 0x08,  'ß', 0x0F,  'Û', 0x00,
     'ß', 0x08,  'ß', 0x0F,  'ß', 0x7F,  'Û', 0x0F,  '²', 0x19,  '²', 0x01,
     'Ý', 0x01,  'Þ', 0x0F,  'Û', 0x0F,  'Û', 0x0F,  'Þ', 0x78,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x00,
@@ -159,7 +159,7 @@
     ' ', 0x07,  ' ', 0x07,  'Þ', 0x08,  'Þ', 0x7F,  'Û', 0x0F,  'Û', 0x0F,
     'Û', 0x0F,  'Û', 0x0F,  'Ý', 0x18,  'Û', 0x19,  'Û', 0x19,  'Ü', 0x01,
     'Ü', 0x07,  'Ü', 0x0F,  'Ü', 0x08,  'Û', 0x00,  ' ', 0x07,  ' ', 0x07,
-    'þ', 0x08,  '±', 0x01,  'Û', 0x00,  'Û', 0x0F,  'Û', 0x0F,  'Û', 0x0F,
+    'ß', 0x08,  '±', 0x01,  'Û', 0x00,  'Û', 0x0F,  'Û', 0x0F,  'Û', 0x0F,
     'Þ', 0x78,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x00,  ' ', 0x18,  'Û', 0x08,  ' ', 0x08,  ':', 0x08,
     ' ', 0x08,  ' ', 0x08,  'W', 0x09,  'e', 0x09,  'l', 0x09,  'c', 0x09,
@@ -291,7 +291,7 @@
     '°', 0x6E,  '²', 0x6E,  '°', 0x6E,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,
     'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,
     'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x66,
-    'Ü', 0x60,  'ß', 0x06,  'þ', 0x60,  'ß', 0x68,  'ß', 0x60,  'ß', 0x68,
+    'Ü', 0x60,  'ß', 0x06,  'ß', 0x60,  'ß', 0x68,  'ß', 0x60,  'ß', 0x68,
     'Û', 0x66,  'Û', 0x66,  'ß', 0x06,  'ß', 0x06,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x0F,  ' ', 0x0F,  ' ', 0x18,  'ß', 0x18,
@@ -304,7 +304,7 @@
     'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x66,  'Û', 0x06,  'Û', 0x06,
     'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,
     'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Û', 0x06,  'Ü', 0x60,
-    'þ', 0x60,  'ß', 0x60,  'ß', 0x68,  'Û', 0x06,  'Û', 0x06,  'Ü', 0x60,
+    'ß', 0x60,  'ß', 0x60,  'ß', 0x68,  'Û', 0x06,  'Û', 0x06,  'Ü', 0x60,
     'ß', 0x06,  'ß', 0x06,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,
     ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x07,  ' ', 0x0F,  ' ', 0x0F,
diff -Nurb Welcome2L-3.04/ansi/xmas.h Welcome2L-3.04-patched/ansi/xmas.h
--- Welcome2L-3.04/ansi/xmas.h	Mon Jun 28 22:56:27 1999
+++ Welcome2L-3.04-patched/ansi/xmas.h	Wed Oct 30 13:54:48 2002
@@ -62,7 +62,7 @@
   'Ü', 0x7F, 'Ü', 0x7F, 'Û', 0x0F, 'Û', 0x7F, 'Û', 0x0F, 'Û', 0x0F,
   'Û', 0x0F, 'Û', 0x7F, 'Ü', 0x7F, 'Ü', 0x7F, 'Ü', 0x7F, 'Ü', 0x7F,
   'Ü', 0x7F, 'Ü', 0x7F, 'Ü', 0x07, 'ß', 0x04, 'ß', 0x04, 'ß', 0x04,
-  'Û', 0x04, 'Û', 0x04, 'Û', 0x04, 'Û', 0x04, 'Û', 0x04, 'þ', 0x4C,
+  'Û', 0x04, 'Û', 0x04, 'Û', 0x04, 'Û', 0x04, 'Û', 0x04, 'ß', 0x4C,
   'Û', 0x04, 'Ü', 0x04, 'Ü', 0x04, 'Ü', 0x10, ' ', 0x10, ' ', 0x10,
   ' ', 0x10, ' ', 0x10, ' ', 0x10, ' ', 0x10, '*', 0x19, ' ', 0x10,
   ' ', 0x10, ' ', 0x10, 'Û', 0x10, '±', 0x09, 'Û', 0x10, ' ', 0x10,
@@ -205,10 +205,10 @@
   ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F,
   ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ' ', 0x0F, ',', 0x0A, '³', 0x08,
   'Û', 0x0F, 'Û', 0x10, ' ', 0x10, ' ', 0x10, ' ', 0x8E, ' ', 0x8E,
-  ' ', 0x8E, 'Ý', 0x78, 'Ý', 0x7F, ' ', 0x8E, ' ', 0x8E, 'þ', 0x0F,
-  'ß', 0x07, ' ', 0x07, 'Þ', 0x0F, '°', 0x7F, 'Ý', 0x08, 'ß', 0x08,
-  'ß', 0x08, 'þ', 0x08, 'Ý', 0x78, 'Þ', 0x7F, 'Ý', 0x7F, ' ', 0x8E,
-  ' ', 0x8E, ' ', 0x8E, 'þ', 0x0F, 'ß', 0x07, ' ', 0x07, 'Þ', 0x7F,
+  ' ', 0x8E, 'Ý', 0x78, 'Ý', 0x7F, ' ', 0x8E, ' ', 0x8E, 'ß', 0x0F,
+  ' ', 0x07, ' ', 0x07, 'Þ', 0x0F, '°', 0x7F, 'Ý', 0x08, 'ß', 0x08,
+  'ß', 0x08, 'ß', 0x08, 'Ý', 0x78, 'Þ', 0x7F, 'Ý', 0x7F, ' ', 0x8E,
+  ' ', 0x8E, ' ', 0x8E, 'ß', 0x0F, ' ', 0x07, ' ', 0x07, 'Þ', 0x7F,
   'Û', 0x0F, 'Þ', 0x78, ' ', 0x8E, ' ', 0x8E, ' ', 0x8E, ' ', 0x8E,
   ' ', 0x8E, ' ', 0x8E, 'Þ', 0x07, 'ß', 0x70, 'ß', 0x70, 'Ü', 0x07,
   'Ü', 0x10, ' ', 0x10, ' ', 0x10, '*', 0x19, ' ', 0x10, 'Û', 0x07,
diff -Nurb Welcome2L-3.04/distrib/Welcome2L.pod Welcome2L-3.04-patched/distrib/Welcome2L.pod
--- Welcome2L-3.04/distrib/Welcome2L.pod	Mon Jun 28 22:56:27 1999
+++ Welcome2L-3.04-patched/distrib/Welcome2L.pod	Wed Oct 30 13:58:09 2002
@@ -73,104 +73,7 @@
 
 =head1 EXAMPLE
 
-./Welcome2L -getty -msg"RedHat Linux" -lcol14
- 
-
-
-=head1 INSTALLATION
-
-
-=head2 On RedHat systems 
-
-cd to /etc/rc.d/
-edit rc.local, near the end find these lines (may be not exactly the same) :
-
-    echo "" > /etc/issue
-    echo "Red Hat Linux $R" >> /etc/issue
-    echo "Kernel $(uname r) on $a $(uname m)" >> /etc/issue
-
-comment them out and add this one (change the version numbers to yours) :
-
-    /usr/local/bin/Welcome2L -msg"Red Hat 5.0 (Hurricane)" > /etc/issue
-
-
-
-=head2 On Debian systems 
-
-[ From: Gerd Bavendiek <bav@rw.sni.de> ]
-   I put a script in
-   lulu:/etc/rc.boot> ls
-   0setserial  1netenv     2hdparm  3Welcome2L
-    
-    It's contents is just:    
-   #!/bin/sh
-    /usr/local/bin/Welcome2L -getty -msg"Debian GNU/Linux 2.0" > /etc/issue
-
-
-=head2 On Slackware systems
-
-[ From: Joe X <obliteration@hotmail.com> ]
-  add these lines in /etc/rc.d/rc.S
-  /usr/local/bin/Welcome2L > /etc/issue
-  /usr/local/bin/Welcome2L > /etc/issue.net
-
-
-=head2 On SUSE 5.3 system:
-
-[ From: Ronny Ziegler <ronny.ziegler@usa.net> ]
-
- 1) Put the following script called logo into /sbin/init.d
-
- #! /bin/sh
- #
- #Author: Ronny Ziegler <ronny.ziegler@usa.net>, 1998
- #
- # /sbin/init.d/logo
- #
- case "$1" in
-     start)
-         echo "Starting  logo"
-         /usr/local/bin/Welcome2L
-         /bin/sleep 1
-         ;;
-     stop)
-         ;;
-     reload|restart)
-         ;;
-     *)
-         echo "Usage: $0 {start|stop|reload|restart}"
-         exit 1
- esac
- exit 0
-
- 2) cd /sbin/init.d/rc2.d
- 3) mv S99zzreached S98zzreached
- 4) ln -s ../logo S99logo
-    Now the logo will be executed as last before login-prompt
- 5) You don't need the old Login-Text any more:
-     mv /etc/issue /etc/issue.orig
-     touch /etc/issue
-
-
-=head2 On SUSE 6.0 system:
-
-[ From: Ronny Ziegler <ronny.ziegler@usa.net> ]
-
-You should have an error-free bootprocess before installing Welcome2L.
- 
- 1) Add the following in your /sbin/init.d/rc file as 
-    the line before last (that´s the one with exit 0)  
-        /usr/local/bin/Welcome2L
-    where this is the path to your Welcome2L-executeable
-
- 1b)Optional: If you dont´t like Welcome2L to overwrite your 
-    boot-messages, add 
-        echo -e \\v\\v\\v\\v\\v\\v
-    (or some more \\v ´s) in you rc before starting Welcome2L.  
-
- 2) Now you don´t need the old Login-Text any more:
-    mv /etc/issue /etc/issue.orig
-    touch /etc/issue
+./Welcome2L -getty -msg"Gentoo Linux" -lcol14
 
 
 =head1 RESTRICTIONS
diff -Nurb Welcome2L-3.04/distrib/Welcome2L.pod~ Welcome2L-3.04-patched/distrib/Welcome2L.pod~
--- Welcome2L-3.04/distrib/Welcome2L.pod~	Mon Jun 28 22:56:27 1999
+++ Welcome2L-3.04-patched/distrib/Welcome2L.pod~	Thu Jan  1 01:00:00 1970
@@ -1,203 +0,0 @@
-=head1 NAME
-
-Welcome2l - Linux ANSI boot logo.
-
-
-=head1 SYNOPSIS
-
-B<Welcome2L> [ B<-scr> ] [ B<-getty> ] [ B<-msg> ] [ B<-cpu> ] [ B<-lcol> ]  [ B<-nolf> ] [ B<-private> ]
-
-
-=head1 DESCRIPTION
-
-B<Welcome2L> is a little program that may run at login time to produce a BBS
-like ANSI login logo. It's very similar to Linux_Logo. But where Linux_Logo
-intends to be portable, Welcome2L intends to produce the best looking
-ANSI screens by making full usage of PC graphic characters. Therefore
-an architecture able to display those characters (i386, Alpha with TGA adapter,
-... ) is required to use it. And, even if it will work on larger screens,
-it will only produce 80 column ANSI screens. 
-
-
-
-=head1 OPTIONS
-
-=over 8
-
-=item B<-getty>
-
-Will display more accurate information when launched at boot time.
-
-
-=item B<-nolf>
-
-Will not add an empty line to the ansi screen produced.
-
-
-=item B<-msg>
-
--msg"text of message" will show the message given as the distribution info.
-
-=item B<-cpu>
-
--cpu"name" will show the  name given as the CPU name.
-
-
-=item B<-scr>xx
-
-Will display screen number xx. 
-
-
-=item B<-lcol>xx
-
-Will change the prompt color to the color number xx. Type Welcome2L -help to see
-what number is attributed to a color.
-
-
-=item B<-private>
-
-Display an "Unauthorized access is prohibited" line at the bottom 
-of the ANSI screen
-
-=item B<-help>
-
-Will display usage information. 
-
-=back
-
-
-=head1 EXAMPLE
-
-./Welcome2L -getty -msg"RedHat Linux" -lcol14
- 
-
-
-=head1 INSTALLATION
-
-
-=head2 On RedHat systems 
-
-cd to /etc/rc.d/
-edit rc.local, near the end find these lines (may be not exactly the same) :
-
-    echo "" > /etc/issue
-    echo "Red Hat Linux $R" >> /etc/issue
-    echo "Kernel $(uname r) on $a $(uname m)" >> /etc/issue
-
-comment them out and add this one (change the version numbers to yours) :
-
-    /usr/local/bin/Welcome2L -msg"Red Hat 5.0 (Hurricane)" > /etc/issue
-
-
-
-=head2 On Debian systems 
-
-[ From: Gerd Bavendiek <bav@rw.sni.de> ]
-   I put a script in
-   lulu:/etc/rc.boot> ls
-   0setserial  1netenv     2hdparm  3Welcome2L
-    
-    It's contents is just:    
-   #!/bin/sh
-    /usr/local/bin/Welcome2L -getty -msg"Debian GNU/Linux 2.0" > /etc/issue
-
-
-=head2 On Slackware systems
-
-[ From: Joe X <obliteration@hotmail.com> ]
-  add these lines in /etc/rc.d/rc.S
-  /usr/local/bin/Welcome2L > /etc/issue
-  /usr/local/bin/Welcome2L > /etc/issue.net
-
-
-=head2 On SUSE 5.3 system:
-
-[ From: Ronny Ziegler <ronny.ziegler@usa.net> ]
-
- 1) Put the following script called logo into /sbin/init.d
-
- #! /bin/sh
- #
- #Author: Ronny Ziegler <ronny.ziegler@usa.net>, 1998
- #
- # /sbin/init.d/logo
- #
- case "$1" in
-     start)
-         echo "Starting  logo"
-         /usr/local/bin/Welcome2L
-         /bin/sleep 1
-         ;;
-     stop)
-         ;;
-     reload|restart)
-         ;;
-     *)
-         echo "Usage: $0 {start|stop|reload|restart}"
-         exit 1
- esac
- exit 0
-
- 2) cd /sbin/init.d/rc2.d
- 3) mv S99zzreached S98zzreached
- 4) ln -s ../logo S99logo
-    Now the logo will be executed as last before login-prompt
- 5) You don't need the old Login-Text any more:
-     mv /etc/issue /etc/issue.orig
-     touch /etc/issue
-
-
-=head2 On SUSE 6.0 system:
-
-[ From: Ronny Ziegler <ronny.ziegler@usa.net> ]
-
-You should have an error-free bootprocess before installing Welcome2L.
- 
- 1) Add the following in your /sbin/init.d/rc file as 
-    the line before last (that´s the one with exit 0)  
-        /usr/local/bin/Welcome2L
-    where this is the path to your Welcome2L-executeable
-
- 1b)Optional: If you dont´t like Welcome2L to overwrite your 
-    boot-messages, add 
-        echo -e \\v\\v\\v\\v\\v\\v
-    (or some more \\v ´s) in you rc before starting Welcome2L.  
-
- 2) Now you don´t need the old Login-Text any more:
-    mv /etc/issue /etc/issue.orig
-    touch /etc/issue
-
-
-=head1 RESTRICTIONS
-
-Welcome2L is not intended to run in a xterm.
-
-=head1 CAVEAT
-
-Because "A 'welcome' page has been proven to be an invitation 
-to come your system, thus making it legal to hack into your box." 
-you may want to use the "-private" arg when running Welcome2L
-to display an "Unauthorized access is prohibited" line at the bottom 
-of the ANSI screen. 
-
-=head1 BUGS
-
-Probably some.
-
-
-=head1 COPYRIGHT
-
-B<Welcome2L Copyright (C)1998-1999 Little Igloo Org> is freely distributable 
-under the B<GNU Public License>, a copy of which you should have received 
-with this software (in a file called F<COPYING>).
-
-
-=head1 AVAILABILITY
-
-You can get the last Welcome2L version, sources, binaries distribution
-or RedHat Packages, at : F<http://www.LittleIgloo.org>
-
-
-=head1 AUTHOR
-
-Jean-Marc Jacquet <jm@littleigloo.org>
diff -Nurb Welcome2L-3.04/main.c Welcome2L-3.04-patched/main.c
--- Welcome2L-3.04/main.c	Mon Jun 28 22:56:27 1999
+++ Welcome2L-3.04-patched/main.c	Wed Oct 30 13:56:30 2002
@@ -17,7 +17,13 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
-
+/*
+ * September 1999, Robert Luberda's <robert@pingu.ii.uj.edu.pl> patch:
+ * - added screens 4 and 5 (modified versions of 1 and 2);
+ * - changed default screen to 4;
+ * - added notime option;
+ * - some other small changes.
+ */
 
 #include <stdio.h>
 #include <string.h>
@@ -58,16 +64,39 @@
 static char CPUID[15];
 static char CPU_INFO = FALSE;
 static char want_getty_infos = FALSE;
-static char THIS_SCREEN = 1;
+static char THIS_SCREEN = 4;	/* default screen */
 static char last_color = FALSE;
 static char WANT_NEWLINE = TRUE;
 static int NEW_YEAR = FALSE;
 static char XMAS_AUTO = FALSE;
 static char IS_WELCOME = TRUE;
+static char NO_TIME = FALSE;
 char WSCREEN[10000];
 char XMAS_SCREEN = 3;
-#define MAX_RAND_SCREEN 2.0
+#define MAX_RAND_SCREEN 4.0
+
+
+/* 
+ * This function, added by Robert Luberda <robert@pingu.ii.uj.edu.pl>,
+ * changes color of  'Welcome to:', 'CPU:', etc. texts.
+ * It works only with screen 1 or screen 2. 
+ */
+static void
+change_text_colors (unsigned char *ansi_screen, int ansi_screen_length, 
+		     char color)
+{
+  register int i, f;
+  int 	tab_val[4] = {1609, 2409, 2569, 2729};  
+		
+  for (i = 0; i < 4; i ++)
+  {
+	if (tab_val[i] + 60 > ansi_screen_length) /* this should not happen */
+		return; 			 
 
+  	for (f = 0; f < 58; f += 2)
+		ansi_screen[tab_val[i] + f] = color;
+  }
+}
 
 static void
 draw_ansi_screen (unsigned char *ansi_screen, int ansi_screen_length, int ansi_screen_width)
@@ -113,6 +142,7 @@
 	{
 	  ANSI_printxy (xtty, ytty, "Login on ");
 
+          ANSI_do_color (color1);
 	  if (want_getty_infos)
 	    ANSI_print (want_getty_infos == 1 ? "\\l" : "@L");
 	  else
@@ -121,7 +151,7 @@
     }
 
 
-  ANSI_do_color (color3);
+  ANSI_do_color (color2);
   if (xwmsg + ywmsg)
     {
       ANSI_printxy (xwmsg, ywmsg, WMESG);
@@ -132,7 +162,14 @@
       if (xtime + ytime)
 	{
 	  ANSI_do_color (color2);
-	  ANSI_printxy (xtime, ytime, want_getty_infos == 1 ? "\\d - \\t" : "@D - @T");
+	  if (NO_TIME == FALSE)
+	  {
+	     ANSI_printxy (xtime, ytime, want_getty_infos == 1 ? " \\d - \\t" : " @D - @T");
+	  }
+	  else
+	  {
+	     ANSI_printxy (xtime, ytime, want_getty_infos == 1 ? " \\d" : " @D");
+	  }
 	}
       if (xrel + yrel)
 	{
@@ -178,7 +215,6 @@
 
   switch (THIS_SCREEN)
     {
-    default:
     case 1:
       {
 	char buf[80];
@@ -205,6 +241,35 @@
       }
       break;
 
+    default:
+    case 4:	/* R.L. */
+      {
+	char buf[80];
+	change_text_colors(WELCOME1, WELCOME1_LENGTH, 0x2);
+	draw_welcome_screen (WELCOME1, WELCOME1_LENGTH, WELCOME1_WIDTH,
+			     0x0E, 0x0C, 0x09,
+			     18, 11, 7, 12, 7, 13,
+			     7, 15, 16, 16, 16, 17, 16, 18);
+	ANSI_do_color (0x19);
+	sprintf (buf, "%s V%s (c)LittleIgloo.Org", PROG_NAME, PROG_VERSION);
+	ANSI_printxy (0, last_line = 22, buf);
+      }
+      break;
+
+    case 5:	/* R.L. */
+      {
+	char buf[80];
+	change_text_colors(WELCOME2, WELCOME2_LENGTH, 0x2);
+	draw_welcome_screen (WELCOME2, WELCOME2_LENGTH, WELCOME2_WIDTH,
+			     0x0E, 0x0C, 0x09,
+			     18, 11, 7, 12, 7, 13,
+			     7, 15, 16, 16, 16, 17, 16, 18);
+	ANSI_do_color (0x19);
+	sprintf (buf, "%s %s (c)LittleIgloo.Org", PROG_NAME, PROG_VERSION);
+	ANSI_printxy (0, last_line = 22, buf);
+      }
+      break;
+
 
     case 3:
       draw_welcome_screen (XMAS, XMAS_LENGTH, XMAS_WIDTH,
@@ -293,8 +358,9 @@
   printf ("\033[0;37m            COLOR NUMBERS ARE: %s\n", "\033[0;34m1 \033[32m2 \033[36m3 \033[31m4 \033[35m5 \033[33m6 \033[37m7 \033[1;30m8 \033[34m9 \033[32m10 \033[36m11 \033[31m12 \033[35m13 \033[33m14 \033[37m15");
   print_option ("cpu", "\"string\"", "Specifies an alternate CPU description");
   print_option ("private", "", "Add an \"Unauthorized access is PROHIBITED\" line");
+  print_option ("notime", "", "Do not dispaly current time\n");
   printf ("\n\033[1;36mEXAMPLE:\n\033[0;36m");
-  printf ("  ./Welcome2L -getty -msg\"RedHat Linux\" -lcol14");
+  printf ("  ./Welcome2L -getty -msg\"Gentoo Linux\" -lcol14");
 
 
   printf ("\033[m\n\n");
@@ -368,8 +434,8 @@
 
       if (strstr (argv[argc], "-xmas") && !XMAS_AUTO)
 	{
-	  printf ("here\n");
-	  THIS_SCREEN = 3;
+/*	  printf ("here\n"); */
+	  THIS_SCREEN = XMAS_SCREEN;
 	  SEE_ARG = TRUE;
 	}
 
@@ -386,6 +452,11 @@
 	  SEE_ARG = TRUE;
 	}
 
+      if (strstr (argv[argc], "-notime"))
+	{
+	  NO_TIME = TRUE;
+	  SEE_ARG = TRUE;
+	}
     }
   if (SEE_ARG == NOTSEEN)
     help_usage ();
@@ -400,19 +471,21 @@
 
   timep = time (NULL);
   timeptr = localtime (&timep);
-  strftime (TIME, sizeof (TIME), "%a %d %b %Y - %X", timeptr);
-
+  if (NO_TIME == FALSE)
+  	strftime (TIME, sizeof (TIME), " %a %d %b %Y - %X", timeptr);
+  else
+  	strftime (TIME, sizeof (TIME), " %a %d %b %Y", timeptr);
 
 
   if ((timeptr->tm_mon < 2) || (timeptr->tm_mon == 11 && timeptr->tm_mday > 25))
     NEW_YEAR = timeptr->tm_year + 1900 + (timeptr->tm_mon == 11 && timeptr->tm_mday > 25);
 
   if ((XMAS_AUTO) && ((timeptr->tm_mon == 0) || (timeptr->tm_mon == 11)))
-    THIS_SCREEN = 3;		/* Ross Younger's patch (revised) - no Xmas decorations outside of Jan/Dec. (enable it with -xmasauto) */
+    THIS_SCREEN = XMAS_SCREEN;		/* Ross Younger's patch (revised) - no Xmas decorations outside of Jan/Dec. (enable it with -xmasauto) */
 
 
   stat ("/proc/kcore", &stat_buf);
-  sprintf (MEMORY, "%ld Kb", stat_buf.st_size / 1024);
+  sprintf (MEMORY, "%ld kB", stat_buf.st_size >> 10);
 
   draw_this_screen ();
 
