diff -Naur ViennaRNA-1.7/Cluster/AS_main.c ViennaRNA-1.7.new/Cluster/AS_main.c
--- ViennaRNA-1.7/Cluster/AS_main.c	2001-04-05 04:36:41.000000000 -0400
+++ ViennaRNA-1.7.new/Cluster/AS_main.c	2008-08-28 10:53:11.000000000 -0400
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include "distance_matrix.h"
 #include "statgeom.h"
 #include "split.h"
diff -Naur ViennaRNA-1.7/Cluster/statgeom.c ViennaRNA-1.7.new/Cluster/statgeom.c
--- ViennaRNA-1.7/Cluster/statgeom.c	2001-04-05 04:36:41.000000000 -0400
+++ ViennaRNA-1.7.new/Cluster/statgeom.c	2008-08-28 10:53:11.000000000 -0400
@@ -7,6 +7,7 @@
 #include <stdio.h>
 #include <strings.h>
 #include <ctype.h>
+#include <stdlib.h>
 #include "utils.h"
 #include "PS3D.h"
 #include "distance_matrix.h"
diff -Naur ViennaRNA-1.7/Cluster/treeplot.c ViennaRNA-1.7.new/Cluster/treeplot.c
--- ViennaRNA-1.7/Cluster/treeplot.c	2001-04-05 04:36:41.000000000 -0400
+++ ViennaRNA-1.7.new/Cluster/treeplot.c	2008-08-28 10:53:11.000000000 -0400
@@ -1,6 +1,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <math.h>
+#include <stdlib.h>
 #include "distance_matrix.h"
 #include "utils.h"
 
diff -Naur ViennaRNA-1.7/RNAforester/src/main.cpp ViennaRNA-1.7.new/RNAforester/src/main.cpp
--- ViennaRNA-1.7/RNAforester/src/main.cpp	2005-08-05 10:14:10.000000000 -0400
+++ ViennaRNA-1.7.new/RNAforester/src/main.cpp	2008-08-28 10:53:11.000000000 -0400
@@ -19,6 +19,8 @@
 #include <list>
 #include <sstream>
 #include <string>
+#include <cstring>
+#include <climits>
 #include <map>
 
 //#include <sys/timeb.h>
diff -Naur ViennaRNA-1.7/RNAforester/src/ppforestbase.h ViennaRNA-1.7.new/RNAforester/src/ppforestbase.h
--- ViennaRNA-1.7/RNAforester/src/ppforestbase.h	2006-05-08 07:49:33.000000000 -0400
+++ ViennaRNA-1.7.new/RNAforester/src/ppforestbase.h	2008-08-28 10:53:11.000000000 -0400
@@ -16,6 +16,7 @@
 
 #include <algorithm>
 #include <cassert>
+#include <cstring>
 
 #include "misc.h"
 #include "types.h"
diff -Naur ViennaRNA-1.7/RNAforester/src/rna_algebra.h ViennaRNA-1.7.new/RNAforester/src/rna_algebra.h
--- ViennaRNA-1.7/RNAforester/src/rna_algebra.h	2004-09-01 17:55:22.000000000 -0400
+++ ViennaRNA-1.7.new/RNAforester/src/rna_algebra.h	2008-08-28 10:53:11.000000000 -0400
@@ -3,6 +3,8 @@
 
 #include <assert.h>
 #include <algorithm>
+#include <cstring>
+#include <climits>
 
 #include "algebra.h"
 #include "debug.h"
diff -Naur ViennaRNA-1.7/Readseq/ureadseq.c ViennaRNA-1.7.new/Readseq/ureadseq.c
--- ViennaRNA-1.7/Readseq/ureadseq.c	2002-11-27 06:23:56.000000000 -0500
+++ ViennaRNA-1.7.new/Readseq/ureadseq.c	2008-08-28 10:53:11.000000000 -0400
@@ -16,7 +16,7 @@
  *
  */
 
-
+#include <stdlib.h>
 #include <stdio.h>
 #include <ctype.h>
 #include <string.h>
