--- doc/Makefile.old	2004-10-08 23:37:58.858574088 +0200
+++ doc/Makefile	2004-10-08 23:38:20.896223856 +0200
@@ -300,7 +300,7 @@
 
 
 install-exec-local: # ensures this is done before install-data
-	$(mkinstalldirs) -m 0755 $(pkgdatadir)/download
+	$(mkinstalldirs) -m 0755 $(DESTDIR)$(pkgdatadir)/download
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
