Index: StepMania-3.9-src/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp
===================================================================
--- StepMania-3.9-src.orig/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp
+++ StepMania-3.9-src/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp
@@ -574,7 +574,7 @@ int URLRageFile_read( avcodec::URLContex
 	return f->Read( buf, size );
 }
 
-avcodec::offset_t URLRageFile_seek( avcodec::URLContext *h, avcodec::offset_t pos, int whence )
+int64_t URLRageFile_seek( avcodec::URLContext *h, int64_t pos, int whence )
 {
 	RageFile *f = (RageFile *) h->priv_data;
 	return f->Seek( (int) pos, whence );
