--- encodep.py~	2002-09-28 21:41:47.000000000 -0500
+++ encodep.py	2003-09-25 10:03:44.797532368 -0500
@@ -31,22 +31,22 @@
 
 # new stuff (ymsg 9 packets)
 def verify_server():
-	return struct.pack("!4slhhll", "YMSG", 0x9000000, 0, 0x4c, 0, 0)
+	return struct.pack("!4slhhll", "YMSG", 0x000b0000, 0, 0x4c, 0, 0)
 
 def request_key(user):
 	data = "1\xC0\x80%s\xC0\x80" % user
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0x57, 0, 0, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0x57, 0, 0, data)
 
 def m9_login(user, key1, key2):
 	data = "0\xC0\x80%s\xC0\x80%s\xC0\x80%s\xC0\x80%s\xC0\x80%s\xC0\x80%s\xC0\x80%s\xC0\x80%s\xC0\x80%s\xC0\x80" % (user, '6', key1, '96', key2, '2', user, '1', user)
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0x54, 0x5a55aa55, 0, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0x54, 0x5a55aa55, 0, data)
 
 def m9_ping(user, sessionID):
 	data = "109\xC0\x80%s\xC0\x80" % user
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0xa1, 0, sessionID, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0xa1, 0, sessionID, data)
 
 def m9_comment(user, room, text, sessionID):
 	data = '1%s%s%s104%s%s%s117%s%s%s124%s%s%s' % (
@@ -55,7 +55,7 @@
 			'\xC0\x80', text, '\xC0\x80',
 			'\xC0\x80', '1', '\xC0\x80')
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0xa8, 0, sessionID, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0xa8, 0, sessionID, data)
 
 def m9_emote(user, room, text, sessionID):
 	data = '1%s%s%s104%s%s%s117%s%s%s124%s%s%s' % (
@@ -64,7 +64,7 @@
 			'\xC0\x80', text, '\xC0\x80',
 			'\xC0\x80', '2', '\xC0\x80')
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0xa8, 0, sessionID, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0xa8, 0, sessionID, data)
 
 def m9_thought(user, room, text, sessionID):
 	data = '1%s%s%s104%s%s%s117%s. o O ( %s )%s124%s%s%s' % (
@@ -73,7 +73,7 @@
 			'\xC0\x80', text, '\xC0\x80',
 			'\xC0\x80', '3', '\xC0\x80')
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0xa8, 0, sessionID, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0xa8, 0, sessionID, data)
 
 def m9_addbuddy(user, friend, sessionID):
 	data = '1%s%s%s7%s%s%s65%s%s%s14%s%s%s' % (
@@ -91,7 +91,7 @@
 			'\xC0\x80', room, '\xC0\x80',
 			'\xC0\x80', "0", '\xC0\x80')
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0x98, 0, sessionID, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0x98, 0, sessionID, data)
 
 def m9_online(user,  sessionID):
 	data = '109%s%s%s1%s%s%s6%s%s%s' % (
@@ -99,7 +99,7 @@
 			'\xC0\x80', user, '\xC0\x80',
 			'\xC0\x80', "abcde", '\xC0\x80')
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0x96, 0, sessionID, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0x96, 0, sessionID, data)
 
 def m9_goAway(text, sessionID):
 	data = '%s\xC0\x80%s\xC0\x80%s\xC0\x80%s\xC0\x80%s\xC0\x80%s' % ("10", "99", "19", text, "47", "0")
@@ -119,7 +119,7 @@
 		'%s%s%s%s' % ("63", DELIM, ";0", DELIM),
 		'%s%s%s%s' % ("64", DELIM, "0", DELIM)])
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0x06, 0x5a55aa55, sessionID, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0x06, 0x5a55aa55, sessionID, data)
 
 def m9_goto(user, otheruser, sessionID):
 	data = ''.join([
@@ -127,12 +127,12 @@
 		'%s%s%s%s' % ("1", DELIM, user, DELIM),
 		'%s%s%s%s' % ("62", DELIM, "2", DELIM)])
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0x97, 0, sessionID, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0x97, 0, sessionID, data)
 
 def m9_logout(user, sessionID):
 	data = "%s%s%s%s" % ("1", DELIM, user, DELIM)
 	fmt = "!4slhhll%ss" % len(data)
-	return struct.pack(fmt, "YMSG", 0x9000000, len(data), 0xa0, 0, sessionID, data)
+	return struct.pack(fmt, "YMSG", 0x000b0000, len(data), 0xa0, 0, sessionID, data)
 
 # end of new stuff (ymsg 9 packets)
 def do(d):
