--- configure.ac
+++ configure.ac
@@ -63,7 +63,8 @@
 dnl *** Check for required packages ***
 dnl ***********************************
 XDT_CHECK_PACKAGE([GIO], [gio-2.0], [2.16.0])
-XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.0])
+XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.0], [],
+  [XDT_CHECK_PACKAGE([EXO], [exo-1], [0.5.0])])
 XDT_CHECK_PACKAGE([GLIB], [glib-2.0], [2.16.0])
 XDT_CHECK_PACKAGE([GDKX], [gdk-x11-2.0], [2.10.0])
 XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.10.0])
