--- modules/wb.validation/Makefile.am.~1~	2010-04-02 03:26:24.000000000 +0200
+++ modules/wb.validation/Makefile.am	2010-04-09 16:18:16.703343478 +0200
@@ -19,6 +19,7 @@
 
 
 wb_validation_grt_la_LDFLAGS=-module
+wb_validation_grt_la_LIBADD=@GRT_LIBS@ @GLIB_LIBS@ @SIGC_LIBS@
 #wb_mysql_import_grt_la_LIBADD=$(top_srcdir)/backend/grtdb/libgrtdbbe.la 
 
 wb_validation_grt_la_SOURCES=src/register_plugin.cpp\
--- modules/wb.mysql.validation/Makefile.am.~1~	2010-04-02 03:26:24.000000000 +0200
+++ modules/wb.mysql.validation/Makefile.am	2010-04-09 16:17:53.991717921 +0200
@@ -21,6 +21,7 @@
 
 
 wb_mysql_validation_grt_la_LDFLAGS=-module
+wb_mysql_validation_grt_la_LIBADD=@GRT_LIBS@ @GLIB_LIBS@ @SIGC_LIBS@
 #wb_mysql_import_grt_la_LIBADD=$(top_srcdir)
 
 wb_mysql_validation_grt_la_SOURCES=src/register_plugin.cpp\
