--- mysql-workbench-oss-5.2.16.orig/library/base/src/Makefile.am	2010-02-16 20:29:55.000000000 +0100
+++ mysql-workbench-oss-5.2.16/library/base/src/Makefile.am	2010-02-24 17:34:07.401777509 +0100
@@ -7,6 +7,9 @@ pkglib_LTLIBRARIES = libwbbase.la
 INCLUDES= -Wall -Wsign-compare\
         @MYSQL_INCLUDE@ @GLIB_CFLAGS@ @PCRE_CFLAGS@ @SIGC_CFLAGS@
 
+libwbbase_la_LIBADD=@GLIB_LIBS@\
+					@PCRE_LIBS@
+
 libwbbase_la_SOURCES=util.c\
 	util_functions.c\
 	xml_util_functions.c\
--- mysql-workbench-oss-5.2.16.orig/backend/wbpublic/Makefile.am	2010-02-16 20:29:46.000000000 +0100
+++ mysql-workbench-oss-5.2.16/backend/wbpublic/Makefile.am	2010-02-24 17:47:05.623513994 +0100
@@ -145,4 +145,4 @@ libwbpublic_la_SOURCES=\
 	grtui/checkbox_list_control.cpp\
   grtui/file_charset_dialog.cpp
 
-libwbpublic_la_LIBADD=../../library/canvas/src/libmdcanvas.la ../../library/forms/libmforms.la ../../library/dbc/libcdbc.la ../../library/vsqlite++/src/sqlite/libvsqlitepp.la
+libwbpublic_la_LIBADD=../../library/canvas/src/libmdcanvas.la ../../library/forms/libmforms.la ../../library/dbc/libcdbc.la ../../library/vsqlite++/src/sqlite/libvsqlitepp.la @CTEMPLATE_LIBS@
--- mysql-workbench-oss-5.2.16.orig/library/grt/src/Makefile.am	2010-02-16 20:29:56.000000000 +0100
+++ mysql-workbench-oss-5.2.16/library/grt/src/Makefile.am	2010-02-24 17:33:32.341520830 +0100
@@ -13,9 +13,11 @@ INCLUDES=-I. -I.. @GLIB_CFLAGS@ @SIGC_CF
 	@GRT_CFLAGS@
 
 libgrt_la_LIBADD=\
-        @LUA_LIBS@\
-        $(top_srcdir)/library/base/src/libwbbase.la
-
+		@LUA_LIBS@\
+		@GLIB_LIBS@\
+		@SIGC_LIBS@\
+		$(top_srcdir)/library/base/src/libwbbase.la
+        
 libgrt_la_SOURCES=\
 	grtpp_grt.cpp\
 	grtpp_helper.cpp\
