Index: /trunk/mapserver/HISTORY.TXT
===================================================================
--- /trunk/mapserver/HISTORY.TXT	(revision 11712)
+++ /trunk/mapserver/HISTORY.TXT	(revision 11713)
@@ -15,4 +15,6 @@
 Current Version (SVN trunk, 6.1-dev, future 6.2): 
 ------------------------------------------------- 
+
+- Fixed the build problem in mapunion.c (#3877)
 
 - Implement to get all shapes with the clustered layer (#3873)
Index: /trunk/mapserver/mapunion.c
===================================================================
--- /trunk/mapserver/mapunion.c	(revision 11710)
+++ /trunk/mapserver/mapunion.c	(revision 11713)
@@ -316,5 +316,5 @@
             /* get items requested by the union layer plus the required items */
             msLayerSetProcessingKey(srclayer, "ITEMS", itemlist);
-            if (msLayerWhichItems(srclayer, TRUE, NULL) != MS_SUCCESS)
+            if (msLayerWhichItems(srclayer, MS_TRUE, NULL) != MS_SUCCESS)
             {
                 msFree(itemlist);
@@ -325,5 +325,5 @@
         {
             /* get only the required items */
-            if (msLayerWhichItems(srclayer, FALSE, NULL) != MS_SUCCESS)
+            if (msLayerWhichItems(srclayer, MS_FALSE, NULL) != MS_SUCCESS)
                 return MS_FAILURE;
         }
@@ -354,5 +354,5 @@
             
             /* get only the required items */
-            if (msLayerWhichItems(srclayer, FALSE, NULL) != MS_SUCCESS)
+            if (msLayerWhichItems(srclayer, MS_FALSE, NULL) != MS_SUCCESS)
                 return MS_FAILURE;
         }
