--- libtheora-1.0alpha4/configure.ac	2004-12-15 21:06:04.000000000 +0100
+++ libtheora-1.0alpha4-works/configure.ac	2005-06-22 16:23:12.000000000 +0200
@@ -83,7 +83,7 @@
 ac_enable_float=yes
 AC_ARG_ENABLE(float,
      [  --disable-float         disable use of floating point code ],
-     [ ac_enable_float=no ], [ ac_enable_float=yes] )
+     [ ac_enable_float=$enableval ], [ ac_enable_float=yes] )
 
 if test "x${ac_enable_float}" = xyes ; then
     AC_DEFINE(THEORA_SUPPORT_FLOAT, [1], [Build floating point code])
@@ -97,7 +97,7 @@
 ac_enable_encode=yes
 AC_ARG_ENABLE(encode,
      [  --disable-encode        disable encoding support ],
-     [ ac_enable_encode=no ], [ ac_enable_encode=yes] )
+     [ ac_enable_encode=$enableval ], [ ac_enable_encode=yes] )
 
 if test "x${ac_enable_encode}" = xyes ; then
     BUILDABLE_EXAMPLES="$BUILDABLE_EXAMPLES encoder_example"
