--- Makefile.PL.orig	2003-10-30 16:38:27.000000000 -0800
+++ Makefile.PL	2003-10-30 16:38:37.000000000 -0800
@@ -464,8 +464,8 @@
 
 EOF
 
-    $string .= Apache::Test::install::nuke_Apache__test_target()
-        if APACHE_TEST_INSTALL();
+#    $string .= Apache::Test::install::nuke_Apache__test_target()
+#        if APACHE_TEST_INSTALL();
 
     $string;
 }
@@ -476,8 +476,8 @@
     ModPerl::MM::add_dep(\$string, pure_install => 'modperl_lib_install');
     ModPerl::MM::add_dep(\$string, pure_install => 'modperl_xs_h_install');
 
-    ModPerl::MM::add_dep(\$string, pure_install => 'nuke_Apache__test')
-          if APACHE_TEST_INSTALL();
+#    ModPerl::MM::add_dep(\$string, pure_install => 'nuke_Apache__test')
+#          if APACHE_TEST_INSTALL();
 
     $string;
 }
