diff -uNr gnome-cups-manager-0.17/gnome-cups-add/Makefile.am gnome-cups-manager-0.17.patched/gnome-cups-add/Makefile.am
--- gnome-cups-manager-0.17/gnome-cups-add/Makefile.am	2003-05-02 12:44:21.000000000 +0200
+++ gnome-cups-manager-0.17.patched/gnome-cups-add/Makefile.am	2004-03-22 00:29:34.260671832 +0100
@@ -9,9 +9,7 @@
 	-DDATADIR=\""$(datadir)"\" \
 	-DGNOME_CUPS_MANAGER_DATADIR=\""$(datadir)/gnome-cups-manager"\" \
 	-DLIBDIR=\""$(libdir)"\" \
-	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
-	-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED -DGNOME_DISABLE_DEPRECATED
+	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	
 
 cupsmanagerdatadir=$(datadir)/gnome-cups-manager
 cupsmanagerdata_DATA=gnome-cups-add.glade
@@ -39,4 +37,4 @@
 gnome_cups_add_LDFLAGS= $(GNOME_LIBS) \
 	snmpkit/libsnmpkit.a ../libgnomecups/libgnomecupsui-1.0.la
 
-EXTRA_DIST=gnome-cups-add.glade
\ No newline at end of file
+EXTRA_DIST=gnome-cups-add.glade
diff -uNr gnome-cups-manager-0.17/gnome-cups-add/Makefile.in gnome-cups-manager-0.17.patched/gnome-cups-add/Makefile.in
--- gnome-cups-manager-0.17/gnome-cups-add/Makefile.in	2003-07-01 17:41:06.000000000 +0200
+++ gnome-cups-manager-0.17.patched/gnome-cups-add/Makefile.in	2004-03-22 00:30:07.027690488 +0100
@@ -129,10 +129,7 @@
 	-DDATADIR=\""$(datadir)"\" \
 	-DGNOME_CUPS_MANAGER_DATADIR=\""$(datadir)/gnome-cups-manager"\" \
 	-DLIBDIR=\""$(libdir)"\" \
-	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
-	-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED -DGNOME_DISABLE_DEPRECATED
-
+	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"
 
 cupsmanagerdatadir = $(datadir)/gnome-cups-manager
 cupsmanagerdata_DATA = gnome-cups-add.glade
diff -uNr gnome-cups-manager-0.17/gnome-cups-manager/Makefile.am gnome-cups-manager-0.17.patched/gnome-cups-manager/Makefile.am
--- gnome-cups-manager-0.17/gnome-cups-manager/Makefile.am	2003-07-01 17:16:48.000000000 +0200
+++ gnome-cups-manager-0.17.patched/gnome-cups-manager/Makefile.am	2004-03-22 00:29:58.571975952 +0100
@@ -8,9 +8,7 @@
 	-DGNOME_CUPS_MANAGER_DATADIR=\""$(datadir)/gnome-cups-manager"\" \
 	-DGNOME_CUPS_MANAGER_PIXMAPDIR=\""$(datadir)/pixmaps/gnome-cups-manager"\" \
 	-DLIBDIR=\""$(libdir)"\" \
-	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
-	-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED -DGNOME_DISABLE_DEPRECATED
+	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"
 
 cupsmanagerdatadir=$(datadir)/gnome-cups-manager
 cupsmanagerdata_DATA=gnome-cups-manager.glade
diff -uNr gnome-cups-manager-0.17/gnome-cups-manager/Makefile.in gnome-cups-manager-0.17.patched/gnome-cups-manager/Makefile.in
--- gnome-cups-manager-0.17/gnome-cups-manager/Makefile.in	2003-07-01 17:41:07.000000000 +0200
+++ gnome-cups-manager-0.17.patched/gnome-cups-manager/Makefile.in	2004-03-22 00:29:48.958437432 +0100
@@ -128,9 +128,7 @@
 	-DGNOME_CUPS_MANAGER_DATADIR=\""$(datadir)/gnome-cups-manager"\" \
 	-DGNOME_CUPS_MANAGER_PIXMAPDIR=\""$(datadir)/pixmaps/gnome-cups-manager"\" \
 	-DLIBDIR=\""$(libdir)"\" \
-	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
-	-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED -DGNOME_DISABLE_DEPRECATED
+	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"
 
 
 cupsmanagerdatadir = $(datadir)/gnome-cups-manager
