--- gamess/comp	2005-11-14 15:44:06.000000000 +0000
+++ gamess-patched/comp	2005-12-07 03:08:59.000000000 +0000
@@ -11,8 +11,8 @@
 #      linux-ia64,linux-pc,mac32,macG5,necsx,sgi32,sgi64,sun32,sun64
 #  2. Choose your GAMESS directory tree.
 #
-set TARGET=ibm64
-chdir /u1/mike/gamess
+set TARGET=linux-pc
+#chdir /u1/mike/gamess
 #
 #    ---- and now the script begins...
 #
@@ -1015,11 +1015,12 @@
 #   Pentium II fractionally).  None of these optimizations are used below.
 #
 if ($TARGET == linux-pc) then
-   set OPT = '-O2'
+   set gentoo-OPT = '-O2'
    if ($MODULE == ormas1) set OPT='-O2 -fno-move-all-movables' # RH9, g77 3.3
 #
 #   g77 does not support FORTRAN logical units above 99, or 128 bit math.
 #
+   if ($MODULE == zheev) set OPT='-O1' 
    if (($MODULE == qeigen) || ($MODULE == int2c)) then
       mv $MODULE.f $MODULE.junk
       sed -e "s/Q-/D-/g" \
@@ -1027,13 +1028,8 @@
           -e "s/REAL\*16/DOUBLE PRECISION/" $MODULE.junk > $MODULE.f
       rm -f $MODULE.junk
    endif
-   if (($MODULE == iolib) || ($MODULE == inputa)) then
-      mv $MODULE.f $MODULE.junk
-      sed -e "s/MXUNIT=299/MXUNIT=99/g" $MODULE.junk > $MODULE.f
-      rm -f $MODULE.junk
-   endif
    set echo
-   g77 -c $OPT -malign-double -fautomatic \
+   gentoo-g77 -c $OPT -malign-double -fautomatic \
            -Wno-globals -fno-globals $MODULE.f
    unset echo
 endif
