--- configure.in.orig	2005-05-21 12:18:13.000000000 +0200
+++ configure.in	2005-05-21 12:41:26.000000000 +0200
@@ -292,14 +292,14 @@
 # ######################################################################
 # Hotplug stuff
 # ######################################################################
-AC_ARG_WITH(hotplug-script-dir, [  --with-hotplug-script-dir=PATH Where to install hotplug scripts [PREFIX/etc/hotplug/usb]])
+AC_ARG_WITH(hotplug-script-dir, [  --with-hotplug-script-dir=PATH Where to install hotplug scripts [/etc/hotplug/usb]])
 
 if test "x${with_hotplug_script_dir}" != "x"
 then # given as parameter
     hotplugscriptdir="${with_hotplug_script_dir}"
     AC_MSG_RESULT([${hotplugscriptdir}])
 else # start at docdir
-    hotplugscriptdir="${prefix}/etc/hotplug/usb"
+    hotplugscriptdir="/etc/hotplug/usb"
     AC_MSG_RESULT([${hotplugscriptdir} (default)])
 fi
 AC_SUBST(hotplugscriptdir)
