diff -Nur saods9.orig/ds9/ds9.C saods9/ds9/ds9.C
--- saods9.orig/ds9/ds9.C	2009-02-16 17:06:50.000000000 +0000
+++ saods9/ds9/ds9.C	2009-02-16 17:07:42.000000000 +0000
@@ -11,8 +11,8 @@
 
   void TclSetStartupScriptFileName(const char*);
 
-  int Zvfs_Init(Tcl_Interp*);
-  int Zvfs_Mount(Tcl_Interp*, char*, char *);
+/*  int Zvfs_Init(Tcl_Interp*);
+  int Zvfs_Mount(Tcl_Interp*, char*, char *);*/
 
   int Blt_Init(Tcl_Interp*);
   int Tktable_Init(Tcl_Interp*);
@@ -64,15 +64,15 @@
   appname = dupstr(argv[0]);
 
   // set default encoding dir
-  Tcl_SetDefaultEncodingDir("./zvfsmntpt/tcl8.4");
+  //Tcl_SetDefaultEncodingDir("./zvfsmntpt/tcl8.4");
 
   // so that tcl and tk know where to find their libs
   // we do it here before InitLibraryPath is called
-  putenv((char*)"TCL_LIBRARY=./zvfsmntpt/tcl8.4");
-  putenv((char*)"TK_LIBRARY=./zvfsmntpt/tk8.4");
+  //putenv((char*)"TCL_LIBRARY=./zvfsmntpt/tcl8.4");
+  //putenv((char*)"TK_LIBRARY=./zvfsmntpt/tk8.4");
 
   // invoke startup script
-  TclSetStartupScriptFileName("./zvfsmntpt/src/ds9.tcl");
+  TclSetStartupScriptFileName("/usr/share/ds9/src/ds9.tcl");
 }
 
 Tcl_Interp *global_interp;
@@ -91,13 +91,13 @@
   // We have to initialize the virtual filesystem before calling
   // Tcl_Init().  Otherwise, Tcl_Init() will not be able to find
   // its startup script files.
-  if (Zvfs_Init(interp) == TCL_ERROR)
+/*  if (Zvfs_Init(interp) == TCL_ERROR)
     return TCL_ERROR;
   Tcl_StaticPackage (interp, "zvfs", Zvfs_Init, 
 		     (Tcl_PackageInitProc*)NULL);
-
+*/
   // find current working directory, and set as mount point
-  {
+/*  {
     Tcl_DString pwd;
     Tcl_DStringInit(&pwd);
     Tcl_GetCwd(interp, &pwd);
@@ -113,7 +113,7 @@
 #endif
     Tcl_DStringFree(&pwd);
   }
-
+*/
   // Initialize Tcl and Tk
   if (Tcl_Init(interp))
     return TCL_ERROR;
@@ -121,7 +121,7 @@
   // Tk
   if (Tk_Init(interp))
     return TCL_ERROR;
-  Tcl_StaticPackage(interp,"Tk", Tk_Init, Tk_SafeInit);
+  //Tcl_StaticPackage(interp,"Tk", Tk_Init, Tk_SafeInit);
 
   {
     Tcl_DString pwd;
@@ -131,7 +131,7 @@
   }
 
   // Blt
-  if (Blt_Init(interp) == TCL_ERROR)
+  /*if (Blt_Init(interp) == TCL_ERROR)
     return TCL_ERROR;
   Tcl_StaticPackage (interp, "BLT", Blt_Init, 
 		     (Tcl_PackageInitProc*)NULL);
@@ -140,7 +140,7 @@
   if (Tktable_Init(interp) == TCL_ERROR)
     return TCL_ERROR;
   Tcl_StaticPackage (interp, "Tktable", Tktable_Init, 
-		     (Tcl_PackageInitProc*)NULL);
+		     (Tcl_PackageInitProc*)NULL);  */
 
   // Checkdns
   if (Checkdns_Init(interp) == TCL_ERROR)
@@ -161,10 +161,10 @@
 		     (Tcl_PackageInitProc*)NULL);
 
   // Tclxpa
-  if (Tclxpa_Init(interp) == TCL_ERROR)
+/*  if (Tclxpa_Init(interp) == TCL_ERROR)
     return TCL_ERROR;
   Tcl_StaticPackage (interp, "Tclxpa", Tclxpa_Init,
-		     (Tcl_PackageInitProc*)NULL);
+		     (Tcl_PackageInitProc*)NULL);*/
 
   // IIS
   if (Iis_Init(interp) == TCL_ERROR)
@@ -179,7 +179,7 @@
 		     (Tcl_PackageInitProc*)NULL);
 
   // Tkimg
-  if (Tkimg_Init(interp) == TCL_ERROR)
+/*  if (Tkimg_Init(interp) == TCL_ERROR)
     return TCL_ERROR;
   Tcl_StaticPackage (interp, "img", 
 		     Tkimg_Init,(Tcl_PackageInitProc*)NULL);
@@ -254,7 +254,7 @@
   if (Tkimgwindow_Init(interp) == TCL_ERROR)
     return TCL_ERROR;
   Tcl_StaticPackage (interp, "window", Tkimgwindow_Init,
-		     (Tcl_PackageInitProc*)NULL);
+		     (Tcl_PackageInitProc*)NULL);*/
 
 #ifdef _MACOSX
   if (Tkmacosx_Init(interp) == TCL_ERROR)
@@ -271,11 +271,11 @@
 #endif
 
   // Variables
-  Tcl_SetVar(interp, "auto_path", "./zvfsmntpt/tcl8.4 ./zvfsmntpt/tk8.4 ./zvfsmntpt/blt2.4 ./zvfsmntpt/tcllib1.6 ./zvfsmntpt/src", TCL_GLOBAL_ONLY); 
+/*  Tcl_SetVar(interp, "auto_path", "./zvfsmntpt/tcl8.4 ./zvfsmntpt/tk8.4 ./zvfsmntpt/blt2.4 ./zvfsmntpt/tcllib1.6 ./zvfsmntpt/src", TCL_GLOBAL_ONLY); 
   Tcl_SetVar(interp, "tcl_libPath", "./zvfsmntpt/tcl8.4", TCL_GLOBAL_ONLY);
   Tcl_SetVar(interp, "blt_library", "./zvfsmntpt/blt2.4", TCL_GLOBAL_ONLY);
   Tcl_SetVar(interp, "blt_libPath", "./zvfsmntpt/blt2.4", TCL_GLOBAL_ONLY);
-  Tcl_SetVar2(interp, "env", "TK_TABLE_LIBRARY", "", TCL_GLOBAL_ONLY);
+  Tcl_SetVar2(interp, "env", "TK_TABLE_LIBRARY", "", TCL_GLOBAL_ONLY);*/
 
   //Tcl_SetVar(interp, "tcl_rcFileName", "~/.wishrc", TCL_GLOBAL_ONLY);
 
diff -Nur saods9.orig/ds9/Makefile saods9/ds9/Makefile
--- saods9.orig/ds9/Makefile	2009-02-16 17:06:50.000000000 +0000
+++ saods9/ds9/Makefile	2009-02-16 17:33:47.000000000 +0000
@@ -4,13 +4,7 @@
 #--------------------------defines
 
 ZDIR	= zipdir/zvfsmntpt
-FILES	= $(ZDIR)/$(TCLVER) \
-	$(ZDIR)/$(TKVER) \
-	$(ZDIR)/$(BLTVER) \
-	$(ZDIR)/$(TCLLIBVER) \
-	$(ZDIR)/$(TKCONVER) \
-	$(ZDIR)/$(XMLRPCVER) \
-	$(ZDIR)/src \
+FILES	= $(ZDIR)/src \
 	$(ZDIR)/msgs \
 	$(ZDIR)/doc \
 	$(ZDIR)/template
@@ -23,37 +17,13 @@
 	../lib/libsaotk.a \
 	../lib/libtkhtml.a \
 	../lib/libtkmpeg.a \
-	../lib/$(TKIMGVER)/libtkimgwindow1.3.a \
-	../lib/$(TKIMGVER)/libtkimgjpeg1.3.a \
-	../lib/$(TKIMGVER)/libjpegtcl1.0.a \
-	../lib/$(TKIMGVER)/libtkimgtiff1.3.a \
-	../lib/$(TKIMGVER)/libtifftcl3.6.1.a \
-	../lib/$(TKIMGVER)/libtkimgpng1.3.a \
-	../lib/$(TKIMGVER)/libpngtcl1.2.6.a \
-	../lib/$(TKIMGVER)/libtkimgppm1.3.a \
-	../lib/$(TKIMGVER)/libtkimggif1.3.a \
-	../lib/$(TKIMGVER)/libtkimgbmp1.3.a \
-	../lib/$(TKIMGVER)/libtkimgxbm1.3.a \
-	../lib/$(TKIMGVER)/libzlibtcl1.2.1.a \
-	../lib/$(TKIMGVER)/libtkimg1.3.a \
-	../lib/$(TKTABLEVER)/libTktable2.9.a \
-	../lib/libfuntools.a \
 	../lib/librice.a \
 	../lib/libhcomp.a \
 	../lib/libplio.a \
-	../lib/libast.a \
 	../lib/libsaotk.a \
-	../lib/libwcs.a \
 	../lib/libsla.a \
-	../lib/libzvfs.a \
-	../lib/libzip.a \
-	../lib/libz.a \
-	../lib/libxpa.a \
 	../lib/libiis.a \
-	../lib/libcheckdns.a \
-	../lib/libBLT24.a \
-	../lib/libtk8.4.a \
-	../lib/libtcl8.4.a
+	../lib/libcheckdns.a
 endif
 
 ifeq ($(OS),windows)
@@ -168,14 +138,9 @@
 ds9.zip	: $(FILES)
 	cd zipdir; zip -r9 ../ds9.zip *	
 else
-ds9	: null.zip ds9Base $(FILES)
+ds9	: ds9Base $(FILES)
 	$(RM) $@
-	cp ds9Base ds9.zip
-	$(STRIP) ds9.zip
-	cat null.zip >> ds9.zip
-	cd zipdir; zip -rA9 ../ds9.zip *
-	mv ds9.zip ds9
-	cp ds9 ../bin/.
+	cp ds9Base ../bin/ds9
 
 debug	: null.zip ds9Base $(FILES)
 	$(RM) $@
@@ -382,7 +347,18 @@
 	/usr/lib/amd64/libdl.so
 	rm -f libstdc++.a
 endif
-
+# For gentoo                                                               
+ifeq ($(ARCH),gentoo)
+ds9Base : $(OBJS) $(LIBS)
+	$(RM) $@                                                                
+	$(CXX) ${OPTS} \
+	-o $@ $(OBJS) $(LIBS) \
+	-L$(X11LIB) -lX11 \
+	-ldl -lpthread \
+	-ltk -ltcl -lBLT24 \
+	-lfuntools `pkg-config --libs wcstools` -lz \
+	`ast_link`                                                              
+endif                 
 #--------------------------linux
 
 ifneq (,$(findstring linux,$(ARCH)))
