--- saods9.orig/src/ds9.tcl	2009-02-16 17:06:47.000000000 +0000
+++ saods9/src/ds9.tcl	2009-02-16 18:37:18.000000000 +0000
@@ -224,25 +224,29 @@
 # initialize, depending on how we are running: choices are zvfs, or wish
 # for now, its not possible to change the current work directory since
 # all of the paths are relative
-if {![catch {package present zvfs}]} {
-    set ds9(root) "zvfsmntpt"
+if {![catch {package present checkdns}]} {
+    set ds9(root) "/usr/share/ds9"
 
-    source $ds9(root)/tcl8.4/msgcat1.3/msgcat.tcl
-    source $ds9(root)/tcl8.4/http2.5/http.tcl
-
-    source $ds9(root)/tk8.4/tearoff.tcl
-
-    source $ds9(root)/tcllib1.10/base64/base64.tcl
-    source $ds9(root)/tcllib1.10/log/log.tcl
-    source $ds9(root)/tcllib1.10/ftp/ftp.tcl
-    source $ds9(root)/tcllib1.10/textutil/repeat.tcl
-    source $ds9(root)/tcllib1.10/textutil/tabify.tcl
-
-    source $ds9(root)/tkcon2.5/tkcon.tcl
-    source $ds9(root)/xmlrpc0.3/xmlrpc.tcl
-
-    source $ds9(root)/blt2.4/graph.tcl
-    source $ds9(root)/blt2.4/tabset.tcl
+    package require msgcat
+    package require http
+    package require base64
+    package require log
+    package require ftp
+    package require textutil::repeat
+    package require textutil::tabify
+    package require tkcon
+    package require BLT
+    package require Tktable
+    package require tclxpa
+    package require img::base
+    package require img::jpeg
+    package require img::tiff
+    package require img::png
+    package require img::gif
+    package require img::ppm
+    package require img::bmp
+    package require img::xbm
+    package require img::window
 
     source $ds9(root)/src/source.tcl
 
