diff -Nur saods9.orig/ds9/ds9.C saods9/ds9/ds9.C
--- saods9.orig/ds9/ds9.C	2008-10-08 20:36:37.000000000 +0100
+++ saods9/ds9/ds9.C	2008-11-04 11:46:44.000000000 +0000
@@ -1,5 +1,7 @@
 #include <iostream>
 #include <sstream>
+#include <cstring>
+#include <cstdlib>
 using namespace std;
 
 #include <tcl.h>
diff -Nur saods9.orig/iis/iistcl.C saods9/iis/iistcl.C
--- saods9.orig/iis/iistcl.C	2008-06-02 22:27:50.000000000 +0100
+++ saods9/iis/iistcl.C	2008-11-04 11:45:29.000000000 +0000
@@ -7,6 +7,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <cstdlib>
 using namespace std;
 
 #include <tcl.h>
diff -Nur saods9.orig/saotk/fitsy++/card.C saods9/saotk/fitsy++/card.C
--- saods9.orig/saotk/fitsy++/card.C	2008-01-04 17:29:24.000000000 +0000
+++ saods9/saotk/fitsy++/card.C	2008-11-04 11:45:29.000000000 +0000
@@ -3,6 +3,7 @@
 // For conditions of distribution and use, see copyright notice in "copyright"
 
 #include "card.h"
+#include <string.h>
 
 FitsCard::FitsCard()
 {
diff -Nur saods9.orig/saotk/fitsy++/compress.C saods9/saotk/fitsy++/compress.C
--- saods9.orig/saotk/fitsy++/compress.C	2008-06-05 21:37:51.000000000 +0100
+++ saods9/saotk/fitsy++/compress.C	2008-11-04 11:45:29.000000000 +0000
@@ -6,6 +6,7 @@
 #include "head.h"
 #include "util.h"
 #include "NaN.h"
+#include <string.h>
 
 FitsCompress::FitsCompress(FitsFile* fits)
 {
diff -Nur saods9.orig/saotk/fitsy++/file.C saods9/saotk/fitsy++/file.C
--- saods9.orig/saotk/fitsy++/file.C	2008-06-03 15:46:25.000000000 +0100
+++ saods9/saotk/fitsy++/file.C	2008-11-04 11:45:29.000000000 +0000
@@ -6,6 +6,7 @@
 #include <sstream>
 #include <iomanip>
 using namespace std;
+#include <string.h>
 
 #include "file.h"
 #include "util.h"
diff -Nur saods9.orig/saotk/fitsy++/iis.C saods9/saotk/fitsy++/iis.C
--- saods9.orig/saotk/fitsy++/iis.C	2008-01-04 17:29:24.000000000 +0000
+++ saods9/saotk/fitsy++/iis.C	2008-11-04 11:45:29.000000000 +0000
@@ -3,6 +3,7 @@
 // For conditions of distribution and use, see copyright notice in "copyright"
 
 #include "iis.h"
+#include <string.h>
 
 FitsIIS::FitsIIS(int width, int height)
 {
diff -Nur saods9.orig/saotk/frame/frscale.C saods9/saotk/frame/frscale.C
--- saods9.orig/saotk/frame/frscale.C	2008-01-04 17:29:26.000000000 +0000
+++ saods9/saotk/frame/frscale.C	2008-11-04 11:45:37.000000000 +0000
@@ -3,6 +3,7 @@
 // For conditions of distribution and use, see copyright notice in "copyright"
 
 #include <string.h>
+#include <stdlib.h>
 
 #include "frscale.h"
 
diff -Nur saods9.orig/saotk/widget/ps.C saods9/saotk/widget/ps.C
--- saods9.orig/saotk/widget/ps.C	2008-06-03 15:46:25.000000000 +0100
+++ saods9/saotk/widget/ps.C	2008-11-04 11:52:06.000000000 +0000
@@ -4,7 +4,8 @@
 
 #include "ps.h"
 #include "util.h"
-#include "string.h"
+#include <cstring>
+using std::memcpy;
 
 Filter::Filter()
 {
--- saods9.orig/saotk/colorbar/colorbarbase.C	2008-09-26 22:54:36.000000000 +0100
+++ saods9/saotk/colorbar/colorbarbase.C	2008-11-04 18:10:13.000000000 +0000
@@ -10,6 +10,7 @@
 
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
+#include <cstdlib>
 
 #include "parser.H"
 
