--- iis/iistcl.C.orig	2008-04-16 13:53:34.315972056 +0000
+++ iis/iistcl.C	2008-04-16 13:54:57.634895375 +0000
@@ -4,7 +4,7 @@
 
 #include <ctype.h>
 #include <string.h>
-
+#include <cstdlib>
 #include <iostream>
 #include <sstream>
 using namespace std;
--- saotk/widget/ps.C.orig	2008-04-16 14:09:20.865993878 +0000
+++ saotk/widget/ps.C	2008-04-16 14:10:04.020453111 +0000
@@ -4,6 +4,7 @@
 
 #include "ps.h"
 #include "util.h"
+#include <cstring>
 
 Filter::Filter()
 {
--- saotk/frame/callback.C.orig	2008-04-16 14:23:03.661659706 +0000
+++ saotk/frame/callback.C	2008-04-16 14:36:52.264879112 +0000
@@ -2,8 +2,8 @@
 // Smithsonian Astrophysical Observatory, Cambridge, MA, USA
 // For conditions of distribution and use, see copyright notice in "copyright"
 
-#include <string.h>
-#include <iostream.h>
+#include <cstring>
+#include <iostream>
 
 #include "callback.h"
 
--- saotk/frame/ciaoparser.C.orig	2008-04-16 15:12:42.535454290 +0000
+++ saotk/frame/ciaoparser.C	2008-04-16 15:13:18.023725663 +0000
@@ -128,9 +128,9 @@
 
 #define FITSPTR (fr->findFits(1))
 
-#include <math.h>
-#include <string.h>
-#include <iostream.h>
+#include <cmath>
+#include <cstring>
+#include <iostream>
 
 #include "framebase.h"
 #include "fitsimage.h"
--- saotk/frame/ds9parser.C.orig	2008-04-16 15:12:42.535454290 +0000
+++ saotk/frame/ds9parser.C	2008-04-16 15:13:40.601718953 +0000
@@ -333,9 +333,9 @@
 #define FITSPTR (fr->findFits(globalTile))
 #define DISCARD_(x) {yyclearin; mkDiscard(x);}
 
-#include <math.h>
-#include <string.h>
-#include <iostream.h>
+#include <cmath>
+#include <cstring>
+#include <iostream>
 
 #include "framebase.h"
 #include "fitsimage.h"
--- saotk/frame/fitsdata.C.orig	2008-04-16 15:09:33.746979438 +0000
+++ saotk/frame/fitsdata.C	2008-04-16 15:11:28.432297279 +0000
@@ -2,11 +2,11 @@
 // Smithsonian Astrophysical Observatory, Cambridge, MA, USA
 // For conditions of distribution and use, see copyright notice in "copyright"
 
-#include <stdlib.h>
-#include <limits.h>
-#include <float.h>
-#include <iostream.h>
-#include <math.h>
+#include <cstdlib>
+#include <climits>
+#include <cfloat>
+#include <iostream>
+#include <cmath>
 
 #include "fitsdata.h"
 #include "colorscale.h"
--- saotk/frame/inversescale.C.orig	2008-04-16 15:12:42.539452518 +0000
+++ saotk/frame/inversescale.C	2008-04-16 15:13:56.113594982 +0000
@@ -2,8 +2,8 @@
 // Smithsonian Astrophysical Observatory, Cambridge, MA, USA
 // For conditions of distribution and use, see copyright notice in "copyright"
 
-#include <iostream.h>
-#include <math.h>
+#include <iostream>
+#include <cmath>
 
 #include "inversescale.h"
 #include "framebase.h"
--- saotk/frame/parser.C.orig	2008-04-16 15:12:42.539452518 +0000
+++ saotk/frame/parser.C	2008-04-16 15:14:20.254970721 +0000
@@ -708,9 +708,9 @@
 
 #define FITSPTR (fr->findFits(1))
 
-#include <math.h>
-#include <string.h>
-#include <iostream.h>
+#include <cmath>
+#include <cstring>
+#include <iostream>
 
 #include "framebase.h"
 #include "frame.h"
--- saotk/frame/prosparser.C.orig	2008-04-16 15:12:42.543450746 +0000
+++ saotk/frame/prosparser.C	2008-04-16 15:14:40.360116448 +0000
@@ -155,9 +155,9 @@
 #define FITSPTR (fr->findFits(1))
 #define DISCARD_(x) {yyclearin; prosDiscard(x);}
 
-#include <math.h>
-#include <string.h>
-#include <iostream.h>
+#include <cmath>
+#include <cstring>
+#include <iostream>
 
 #include "framebase.h"
 #include "fitsimage.h"
--- saotk/frame/saoparser.C.orig	2008-04-16 15:12:42.547448974 +0000
+++ saotk/frame/saoparser.C	2008-04-16 15:14:56.133015294 +0000
@@ -121,9 +121,9 @@
 #define FITSPTR (fr->findFits(1))
 #define DISCARD_(x) {yyclearin; saoDiscard(x);}
 
-#include <math.h>
-#include <string.h>
-#include <iostream.h>
+#include <cmath>
+#include <cstring>
+#include <iostream>
 
 #include "framebase.h"
 #include "fitsimage.h"
--- saotk/frame/tngparser.C.orig	2008-04-16 15:12:42.547448974 +0000
+++ saotk/frame/tngparser.C	2008-04-16 15:15:11.549893852 +0000
@@ -167,9 +167,9 @@
 #define FITSPTR (fr->findFits(1))
 #define DISCARD_(x) {yyclearin; tngDiscard(x);}
 
-#include <math.h>
-#include <string.h>
-#include <iostream.h>
+#include <cmath>
+#include <cstring>
+#include <iostream>
 
 #include "framebase.h"
 #include "fitsimage.h"
--- saotk/frame/xyparser.C.orig	2008-04-16 15:12:42.551447202 +0000
+++ saotk/frame/xyparser.C	2008-04-16 15:15:28.970886617 +0000
@@ -200,9 +200,9 @@
 
 #define FITSPTR (fr->findFits(1))
 
-#include <math.h>
-#include <string.h>
-#include <iostream.h>
+#include <cmath>
+#include <cstring>
+#include <iostream>
 
 #include "framebase.h"
 #include "fitsimage.h"
--- saotk/colorbar/colorbarbase.C.orig	2008-04-16 15:41:10.236325709 +0000
+++ saotk/colorbar/colorbarbase.C	2008-04-16 15:33:33.482296780 +0000
@@ -11,6 +11,7 @@
 #include <X11/Xutil.h>
 
 #include "parser.H"
+#include <cstdlib>
 
 // Parser Stuff
 #undef yyFlexLexer
--- saotk/colorbar/colorbar.C.orig	2008-04-16 15:27:48.890467306 +0000
+++ saotk/colorbar/colorbar.C	2008-04-16 15:28:09.291629902 +0000
@@ -2,8 +2,8 @@
 // Smithsonian Astrophysical Observatory, Cambridge, MA, USA
 // For conditions of distribution and use, see copyright notice in "copyright"
 
-#include <stdlib.h>
-#include <string.h>
+#include <cstdlib>
+#include <cstring>
 
 #include "colorbar.h"
 #include "util.h"
--- saotk/util/conputs.C.orig	2008-04-16 15:41:48.354497940 +0000
+++ saotk/util/conputs.C	2008-04-16 15:42:01.775262745 +0000
@@ -1,3 +1,4 @@
+#include <cstring>
 #include <iostream>
 using namespace std;
 
--- ds9/ds9.C.orig	2008-04-16 15:45:04.281663192 +0000
+++ ds9/ds9.C	2008-04-16 15:45:17.622423439 +0000
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <iostream>
 using namespace std;
 
