--- curl-helper.c~	2003-12-03 19:48:31.000000000 +0100
+++ curl-helper.c	2004-05-04 21:23:07.917507520 +0200
@@ -2249,13 +2249,13 @@
      case 0: /* TIMECOND_IFMODSINCE */
           result = curl_easy_setopt(conn->connection,
                                     CURLOPT_TIMECONDITION,
-                                    TIMECOND_IFMODSINCE);
+                                    CURL_TIMECOND_IFMODSINCE);
           break;
 
      case 1: /* TIMECOND_IFUNMODSINCE */
           result = curl_easy_setopt(conn->connection,
                                     CURLOPT_TIMECONDITION,
-                                    TIMECOND_IFUNMODSINCE);
+                                    CURL_TIMECOND_IFUNMODSINCE);
           break;
 
      default:
