Index: net-snmp-5.2.2/snmplib/Makefile.in
===================================================================
--- net-snmp-5.2.2.orig/snmplib/Makefile.in
+++ net-snmp-5.2.2/snmplib/Makefile.in
@@ -185,11 +185,11 @@ all: standardall
 
 # how to build the libraries.
 libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    ${TOBJS}
-	$(LIB_LD_CMD) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS}
+	$(LIB_LD_CMD) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS} @LIBCRYPTO@
 	$(RANLIB) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
 
 libsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    ${TOBJS}
-	$(LIB_LD_CMD) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS}
+	$(LIB_LD_CMD) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS} @LIBCRYPTO@
 	$(RANLIB) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
 
 #
Index: net-snmp-5.2.2/configure.in
===================================================================
--- net-snmp-5.2.2.orig/configure.in
+++ net-snmp-5.2.2/configure.in
@@ -2362,11 +2362,12 @@ if test $? -eq 0 ; then
         *)
             AC_CHECK_HEADER(sensors/sensors.h,,
                 [AC_MSG_ERROR([asked to use lm_sensors but I couldn't find sensors/sensors.h])])
-            AC_CHECK_LIB(sensors, sensors_get_detected_chips, LIBS="${LIBS} -lsensors",
+            AC_CHECK_LIB(sensors, sensors_get_detected_chips, SENSORLIBS="-lsensors",
                 [AC_MSG_ERROR([asked to use lm_sensors but I couldn't find -lsensors])])
             ;;
         esac
 fi
+AC_SUBST(SENSORLIBS)
 
 #
 # Install the various perl modules
@@ -2691,6 +2692,7 @@ if test "x$useopenssl" != "xno" ; then
         encrmodes="[disabled]"
     fi
     AC_DEFINE(USE_OPENSSL)
+    LIBCRYPTO="-lcrypto"
     AC_MSG_RESULT(OpenSSL Support)
 elif test "x$usepkcs" != "xno" ; then
     authmodes="MD5 SHA1"
@@ -2710,6 +2712,7 @@ fi
 if test "x$enable_md5" = "xno"; then
     authmodes=`echo $authmodes | sed 's/MD5 *//;'`
 fi
+AC_SUBST(LIBCRYPTO)
 
 AC_MSG_CACHE_ADD(Authentication support:     $authmodes)
 AC_MSG_CACHE_ADD(Encryption support:         $encrmodes)
Index: net-snmp-5.2.2/agent/Makefile.in
===================================================================
--- net-snmp-5.2.2.orig/agent/Makefile.in
+++ net-snmp-5.2.2/agent/Makefile.in
@@ -130,11 +130,11 @@ snmpd$(EXEEXT):	${LAGENTOBJS} $(USELIBS)
 
 
 libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION):    ${LLIBAGENTOBJS} libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION)
-	$(LIB_LD_CMD) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION) ${LLIBAGENTOBJS}  $(LIB_LD_LIBS)
+	$(LIB_LD_CMD) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION) ${LLIBAGENTOBJS}  $(LIB_LD_LIBS) helpers/libnetsnmphelpers.la
 	$(RANLIB) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION)
 
 libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION):    ${LMIBOBJS}
-	$(LIB_LD_CMD) libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION) ${LMIBOBJS}  $(LIB_LD_LIBS)
+	$(LIB_LD_CMD) libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION) ${LMIBOBJS}  $(LIB_LD_LIBS) @SENSORLIBS@
 	$(RANLIB) libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION)
 
 embedperlinstall:
