diff -ruN gcc-20040130/gcc/version.c gcc-20040130-branding/gcc/version.c
--- gcc-20040130/gcc/version.c	2004-01-27 17:22:04.000000000 -0600
+++ gcc-20040130-branding/gcc/version.c	2004-02-09 10:31:50.945973456 -0600
@@ -6,7 +6,7 @@
    please modify this string to indicate that, e.g. by putting your
    organization's name in parentheses at the end of the string.  */
 
-const char version_string[] = "3.3.3-hammer 20040127 (prerelease)";
+const char version_string[] = "3.3.3  (Gentoo Linux 3.3.3_pre20040130)";
 
 /* This is the location of the online document giving instructions for
    reporting bugs.  If you distribute a modified version of GCC,
@@ -15,4 +15,4 @@
    forward us bugs reported to you, if you determine that they are
    not bugs in your modifications.)  */
 
-const char bug_report_url[] = "<URL:http://gcc.gnu.org/bugs.html>";
+const char bug_report_url[] = "<URL:http://bugs.gentoo.org/>";
