--- a/lib/wiper/wiperPosix.c
+++ b/lib/wiper/wiperPosix.c
@@ -998,8 +998,9 @@
                   break;
                }
             }
-         } else {
-            fgets(deviceName, sizeof deviceName, fp);
+         } else if (fgets(deviceName, sizeof deviceName, fp) != deviceName) {
+            Warning("%s: Error reading device name from /proc/devices.", __func__);
+            break;
          }
       }
 
