diff -ur src.old/PodSleuth/PodSleuth/ScsiReader.cs src/PodSleuth/PodSleuth/ScsiReader.cs
--- src.old/PodSleuth/PodSleuth/ScsiReader.cs	2007-09-17 04:51:40.000000000 -0300
+++ src/PodSleuth/PodSleuth/ScsiReader.cs	2008-12-14 02:22:14.000000000 -0400
@@ -50,7 +50,7 @@
             }
         }
         
-        [DllImport("libsgutils.so.1")]
+        [DllImport("libsgutils.so")]
         private static extern int sg_ll_inquiry(int sg_fd, int cmddt, int evpd, int pg_op,
             IntPtr resp, int mx_resp_len, int noisy, int verbose);
             
@@ -74,4 +74,4 @@
             }
         }
     }
-}
\ No newline at end of file
+}
diff -ur src.old/PodSleuth/PodSleuth.dll.config.in src/PodSleuth/PodSleuth.dll.config.in
--- src.old/PodSleuth/PodSleuth.dll.config.in	2007-09-17 04:51:40.000000000 -0300
+++ src/PodSleuth/PodSleuth.dll.config.in	2008-12-14 02:22:14.000000000 -0400
@@ -1,4 +1,4 @@
 <configuration>
-	<dllmap dll="libsgutils.so.1" target="libsgutils.so.1" />
+	<dllmap dll="libsgutils.so" target="libsgutils.so" />
 </configuration>
 
