--- GDALmake.opt.in.orig	2005-09-29 21:52:58.000000000 -0700
+++ GDALmake.opt.in	2005-10-24 20:49:49.000000000 -0700
@@ -53,10 +53,10 @@
 exec_prefix	=	@exec_prefix@
 INST_PREFIX	=	@exec_prefix@
 INST_INCLUDE	=	@includedir@
-INST_DATA 	=	@datadir@
+INST_DATA 	=	@datadir@/gdal
 INST_LIB	=	@libdir@
 INST_BIN	=	@bindir@
-INST_PYMOD      =       @pymoddir@
+INST_PYMOD      =	@pymoddir@
 INST_DOCS	=	@exec_prefix@/doc
 INST_MAN	=	@mandir@
 
--- GNUmakefile.orig	2005-09-16 08:47:29.000000000 -0700
+++ GNUmakefile	2005-10-24 20:38:33.000000000 -0700
@@ -132,7 +132,7 @@
 
 install-docs:
 	(cd ogr; $(MAKE) install-docs)
-	$(INSTALL_DIR) $(INST_DOCS)/gdal
+	$(INSTALL_DIR) $(DESTDIR)$(INST_DOCS)/gdal
 	cp html/*.* $(INST_DOCS)/gdal
 
 web-update:	docs
@@ -141,9 +141,9 @@
 install:	default install-actions
 
 install-actions: install-lib
-	$(INSTALL_DIR) $(INST_BIN)
-	$(INSTALL_DIR) $(INST_DATA)
-	$(INSTALL_DIR) $(INST_INCLUDE)
+	$(INSTALL_DIR) $(DESTDIR)$(INST_BIN)
+	$(INSTALL_DIR) $(DESTDIR)$(INST_DATA)
+	$(INSTALL_DIR) $(DESTDIR)$(INST_INCLUDE)
 	(cd port; $(MAKE) install)
 	(cd gcore; $(MAKE) install)
 	(cd frmts; $(MAKE) install)
@@ -155,14 +155,14 @@
 	(cd pymod; $(MAKE) install)
 endif
 	(cd swig; $(MAKE) install)
-	for f in data/*.* ; do $(INSTALL_DATA) $$f $(INST_DATA) ; done
+	for f in data/*.* ; do $(INSTALL_DATA) $$f $(DESTDIR)$(INST_DATA) ; done
 	$(LIBTOOL_FINISH) $(INST_LIB)
 
 ifeq ($(HAVE_LIBTOOL),yes)
 
 install-lib:
-	$(INSTALL_DIR) $(INST_LIB)
-	for f in $(LIBGDAL-yes) ; do $(INSTALL_LIB) $$f $(INST_LIB) ; done
+	$(INSTALL_DIR) $(DESTDIR)$(INST_LIB)
+	for f in $(LIBGDAL-yes) ; do $(INSTALL_LIB) $$f $(DESTDIR)$(INST_LIB) ; done
 
 else
 
@@ -172,11 +172,11 @@
 GDAL_SLIB_B	=	$(notdir $(GDAL_SLIB))
 
 install-lib:
-	$(INSTALL_DIR) $(INST_LIB)
+	$(INSTALL_DIR) $(DESTDIR)$(INST_LIB)
 	rm -f $(INST_LIB)/$(GDAL_SLIB_B)
 	rm -f $(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VER_MAJOR)
 	rm -f $(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VER)
-	$(INSTALL_LIB) $(GDAL_SLIB) $(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VER)
+	$(INSTALL_LIB) $(GDAL_SLIB) $(DESTDIR)$(INST_LIB)/$(GDAL_SLIB_B).$(GDAL_VER)
 	(cd $(INST_LIB) ; \
 	 ln -s $(GDAL_SLIB_B).$(GDAL_VER_MAJOR) $(GDAL_SLIB_B))
 	(cd $(INST_LIB) ; \
@@ -185,8 +185,8 @@
 else
 
 install-lib:
-	$(INSTALL_DIR) $(INST_LIB)
-	$(INSTALL_LIB) $(GDAL_LIB) $(INST_LIB)
+	$(INSTALL_DIR) $(DESTDIR)$(INST_LIB)
+	$(INSTALL_LIB) $(GDAL_LIB) $(DESTDIR)$(INST_LIB)
 
 endif # HAVE_LD_SHARED=no 
 
--- alg/GNUmakefile.orig	2004-12-26 13:26:15.000000000 -0800
+++ alg/GNUmakefile	2005-10-24 20:39:25.000000000 -0700
@@ -23,7 +23,7 @@
 	(cd ..; $(MAKE) docs)
 
 install:
-	for f in *.h ; do $(INSTALL_DATA) $$f $(INST_INCLUDE) ; done
+	for f in *.h ; do $(INSTALL_DATA) $$f $(DESTDIR)$(INST_INCLUDE) ; done
 
 lib:	$(OBJ)
 	(cd .. ; $(MAKE) force-lib)
--- apps/GNUmakefile.orig	2005-05-12 00:10:00.000000000 -0700
+++ apps/GNUmakefile	2005-10-24 20:40:28.000000000 -0700
@@ -89,6 +89,6 @@
 
 
 install: default
-	for f in $(BIN_LIST) ; do $(INSTALL) $$f $(INST_BIN) ; done
-	$(INSTALL) gdal-config-inst $(INST_BIN)/gdal-config
+	for f in $(BIN_LIST) ; do $(INSTALL) $$f $(DESTDIR)$(INST_BIN) ; done
+	$(INSTALL) gdal-config-inst $(DESTDIR)$(INST_BIN)/gdal-config
 
--- frmts/mem/GNUmakefile.orig	2005-01-24 10:09:57.000000000 -0800
+++ frmts/mem/GNUmakefile	2005-10-24 20:41:33.000000000 -0700
@@ -15,4 +15,4 @@
 install-obj:	$(O_OBJ)
 
 install:
-	$(INSTALL_DATA) memdataset.h $(INST_INCLUDE)
+	$(INSTALL_DATA) memdataset.h $(DESTDIR)$(INST_INCLUDE)
--- frmts/raw/GNUmakefile.orig	2005-01-24 10:09:57.000000000 -0800
+++ frmts/raw/GNUmakefile	2005-10-24 20:42:10.000000000 -0700
@@ -21,4 +21,4 @@
 $(OBJ) $(O_OBJ):	rawdataset.h
 
 install:
-	$(INSTALL_DATA) rawdataset.h $(INST_INCLUDE)
+	$(INSTALL_DATA) rawdataset.h $(DESTDIR)$(INST_INCLUDE)
--- frmts/vrt/GNUmakefile.orig	2005-01-24 10:09:57.000000000 -0800
+++ frmts/vrt/GNUmakefile	2005-10-24 20:42:48.000000000 -0700
@@ -19,4 +19,4 @@
 $(OBJ) $(O_OBJ):	vrtdataset.h
 
 install:
-	$(INSTALL_DATA) vrtdataset.h $(INST_INCLUDE)
+	$(INSTALL_DATA) vrtdataset.h $(DESTDIR)$(INST_INCLUDE)
--- gcore/GNUmakefile.orig	2005-09-24 12:03:11.000000000 -0700
+++ gcore/GNUmakefile	2005-10-24 20:43:25.000000000 -0700
@@ -33,4 +33,4 @@
 		$*.cpp -o $*.o
 
 install:
-	for f in *.h ; do $(INSTALL_DATA) $$f $(INST_INCLUDE) ; done
+	for f in *.h ; do $(INSTALL_DATA) $$f $(DESTDIR)$(INST_INCLUDE) ; done
--- man/GNUmakefile.orig	2004-02-20 09:26:13.000000000 -0800
+++ man/GNUmakefile	2005-10-24 20:44:21.000000000 -0700
@@ -5,5 +5,5 @@
 include ../GDALmake.opt
 
 install:
-	$(INSTALL_DIR) $(INST_MAN)/man1
-	for f in $(MAN1_FILES) ; do $(INSTALL_DATA) $$f $(INST_MAN)/man1 ; done
+	$(INSTALL_DIR) $(DESTDIR)$(INST_MAN)/man1
+	for f in $(MAN1_FILES) ; do $(INSTALL_DATA) $$f $(DESTDIR)$(INST_MAN)/man1 ; done
--- ogr/GNUmakefile.orig	2005-03-11 06:28:05.000000000 -0800
+++ ogr/GNUmakefile	2005-10-24 20:46:02.000000000 -0700
@@ -79,7 +79,7 @@
 	cp ogr_feature_style.html html
 
 install-docs:
-	$(INSTALL_DIR) $(INST_DOCS)/ogr
+	$(INSTALL_DIR) $(DESTDIR)$(INST_DOCS)/ogr
 	cp html/* $(INST_DOCS)/ogr
 
 gdalso:	$(GDAL_SLIB)
@@ -120,14 +120,14 @@
 
 ifeq ($(OGR_ENABLED),yes)
 install-full:	apps
-	$(INSTALL) ogr2ogr $(INST_BIN)
-	$(INSTALL) ogrinfo $(INST_BIN)
-	$(INSTALL) ogrtindex $(INST_BIN)
+	$(INSTALL) ogr2ogr $(DESTDIR)$(INST_BIN)
+	$(INSTALL) ogrinfo $(DESTDIR)$(INST_BIN)
+	$(INSTALL) ogrtindex $(DESTDIR)$(INST_BIN)
 else
 install-full:
 endif
 
 install:	install-full
 	for f in $(INST_H_FILES) ; \
-	    do $(INSTALL_DATA) $$f $(INST_INCLUDE) ; \
+	    do $(INSTALL_DATA) $$f $(DESTDIR)$(INST_INCLUDE) ; \
 	done
--- port/GNUmakefile.orig	2005-09-11 11:01:48.000000000 -0700
+++ port/GNUmakefile	2005-10-24 20:46:38.000000000 -0700
@@ -25,7 +25,7 @@
 	$(RM) *.o $(LIB)
 
 install:
-	for f in *.h ; do $(INSTALL_DATA) $$f $(INST_INCLUDE) ; done
+	for f in *.h ; do $(INSTALL_DATA) $$f $(DESTDIR)$(INST_INCLUDE) ; done
 
 xmlreformat:	xmlreformat.o 
 	$(CXX) $(CXXFLAGS) xmlreformat.o $(CONFIG_LIBS) -o xmlreformat
--- pymod/GNUmakefile.orig	2005-01-31 14:07:02.000000000 -0800
+++ pymod/GNUmakefile	2005-10-24 20:47:51.000000000 -0700
@@ -45,9 +45,9 @@
 
 # special care is taken so that python scripts are installed as executables
 install:	$(GDALMODULE_SO)
-	$(INSTALL_DIR) $(INST_PYMOD)
-	$(INSTALL_LIB) $(GDALMODULE_SO) $(INST_PYMOD)
-	for f in $(PY_COMMANDS) ; do $(INSTALL) $$f $(INST_BIN) ; done
-	for f in $(PY_MODULES) ; do $(INSTALL_DATA) $$f $(INST_PYMOD) ; done
+	$(INSTALL_DIR) $(DESTDIR)$(INST_PYMOD)
+	$(INSTALL_LIB) $(GDALMODULE_SO) $(DESTDIR)$(INST_PYMOD)
+	for f in $(PY_COMMANDS) ; do $(INSTALL) $$f $(DESTDIR)$(INST_BIN) ; done
+	for f in $(PY_MODULES) ; do $(INSTALL_DATA) $$f $(DESTDIR)$(INST_PYMOD) ; done
 	# ugh! spurius relinking leaves a file owned by root into .libs/.
 	rm -f .libs/*.soT
