--- Makefile
+++ Makefile
@@ -1,5 +1,5 @@
 INSTALL=install
-prefix?=/opt/diet
+prefix=/usr/diet
 # Set the following to install to a different root
 #DESTDIR=/tmp/fefix
 # Use "make DEBUG=1" to compile a debug version.
@@ -95,7 +95,6 @@
 
 profiling: $(OBJDIR)/libgmon.a $(OBJDIR)/pstart.o
 
-CFLAGS=-pipe -nostdinc
 CROSS=
 
 CC=gcc
@@ -126,10 +125,6 @@
 
 LIBMATHOBJ=$(patsubst %,$(OBJDIR)/%,$(LIBMATH))
 
-ifeq ($(CFLAGS),-pipe -nostdinc)
-CFLAGS+=-O -fomit-frame-pointer
-endif
-
 ifneq ($(DEBUG),)
 CFLAGS = -g
 COMMENT = :
