--- diet.c
+++ diet.c
@@ -26,8 +26,13 @@
 
 static const char* Os[] = {
   "i386","-Os","-mpreferred-stack-boundary=2",
+#if __GNUC__ >= 3
+	 "-falign-functions=1","-falign-jumps=1",
+	 "-falign-loops=1","-fomit-frame-pointer",0,
+#else
 	 "-malign-functions=1","-malign-jumps=1",
 	 "-malign-loops=1","-fomit-frame-pointer",0,
+#endif
   "x86_64","-Os","-fno-omit-frame-pointer",0,
   "sparc","-Os","-mcpu=supersparc",0,
   "sparc64","-Os","-m64",0,
