diff -Nru pulsar-4.0.24.orig/makefile.2.6 pulsar-4.0.24/makefile.2.6
--- pulsar-4.0.24.orig/makefile.2.6	2007-06-26 09:49:40.000000000 +0300
+++ pulsar-4.0.24/makefile.2.6	2008-01-29 13:42:52.000000000 +0200
@@ -5,7 +5,7 @@
 
 ifneq ($(KERNELRELEASE),)
 
-GCC	= $(shell ${CC} --version | head -n 1 | sed -e 's/[^0-9. ]//g;s/^ *//;s\[.].*\\g')
+GCC	= $(shell ${CC} --version | head -n 1 | sed -e 's/[[:alnum:]-]*//;s/[^0-9. ]//g;s/^ *//;s\[.].*\\g')
 
 obj-m		:= pulsar_atm.o
 pulsar_atm-lib	:= libpulsar_fw$(GCC).a
diff -Nru pulsar-4.0.24.orig/pulsar.c pulsar-4.0.24/pulsar.c
--- pulsar-4.0.24.orig/pulsar.c	2007-06-29 02:59:21.000000000 +0300
+++ pulsar-4.0.24/pulsar.c	2008-01-29 13:42:18.000000000 +0200
@@ -1015,7 +1015,11 @@
 
 	/* Register interrupt */
 	adsl_irq_disable(pmodem);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24))
 	err = request_irq(pci_dev->irq, isr, SA_SHIRQ, name, pmodem);
+#else
+	err = request_irq(pci_dev->irq, isr, IRQF_SHARED, name, pmodem);
+#endif
 	if (err) {
 		printk(KERN_ERR "%s: IRQ request failure\n", name);
 		goto pulsar_init_exit_irq;
