--- kmatplot/widgets/qscontour.cpp.orig	2002-03-18 17:06:56.000000000 +0100
+++ kmatplot/widgets/qscontour.cpp	2002-04-11 18:04:26.000000000 +0200
@@ -427,7 +427,7 @@
 
 //-------------------------------------------------------------//
 
-inline int QSContour::find_level_greater_than( double value )
+int QSContour::find_level_greater_than( double value )
 // find first level > value or return level.number if all levels are lower then value
  {
   double *first_level      = &d->level.value[0];
--- kmatplot/widgets/qscontour.h.orig	2002-03-12 00:59:49.000000000 +0100
+++ kmatplot/widgets/qscontour.h	2002-04-11 18:04:51.000000000 +0200
@@ -184,7 +184,7 @@
       				      QSPt3f *under_pts, int *under_npts,
       				      QSPt3f *cross_line = NULL, int *cross_npts = NULL );
 	inline double distance( const QSPt2f& p1, const QSPt2f& p2 );
- 	inline int find_level_greater_than( double value );
+ 	int find_level_greater_than( double value );
 	void flush_rectangle_buffer();
 	bool draw_rectangle( const QSPt3f pts[4] );
    };
