--- kmatplot/octave/kadd.cpp
+++ kmatplot/octave/kadd.cpp
@@ -46,6 +46,6 @@
  	dataset_number = plot_add_dataset( socket_fd, axesNumber(), (PlotType )type );
  	plot_disconnect( socket_fd );
      }
- return octave_value_list( (double )dataset_number );
+ return octave_value_list(1, octave_value((double )dataset_number) );
 }
 
--- kmatplot/octave/kplot.cpp
+++ kmatplot/octave/kplot.cpp
@@ -48,7 +48,7 @@
 " See also: ksetapp, ksetaxes, kadd, kremove, kremoveall, kaddaxes, kremoveaxes, kimage, kcontour, kmesh, ksetmatrix"
 )
 {
-  octave_value_list result(-1.0);
+  octave_value_list result(1, octave_value(-1.0));
   int nargin = args.length();
 
   int socket_fd = plot_connect( appNumber(), NULL, NULL );
@@ -113,7 +113,7 @@
    // Disconnect
    //		
    plot_disconnect( socket_fd );
-   return octave_value_list( (double )dataset_number );			
+   return octave_value_list( 1, octave_value((double )dataset_number) );			
  }
 
 
--- kmatplot/octave/ksetaxes.cpp
+++ kmatplot/octave/ksetaxes.cpp
@@ -34,6 +34,6 @@
      setAxes( new_plot_num );
      }
 
-  return octave_value_list( (double )plot_num );
+  return octave_value_list(1, octave_value((double )plot_num) );
 }
 
--- kmatplot/octave/kmesh.cpp
+++ kmatplot/octave/kmesh.cpp
@@ -26,7 +26,7 @@
 "\n"
 "See also: ksetapp, ksetaxes, kadd, kremove, kremoveall, kaddaxes, kremoveaxes, kplot, kimage, kcontour, ksetmatrix" )
 {
-  octave_value_list result(-1.0);
+  octave_value_list result(1, octave_value(-1.0));
 
   int nargin = args.length();
   if ( nargin != 1 && nargin != 3  ) {
@@ -105,7 +105,7 @@
     }
 
   plot_disconnect( socket_fd );
-  return octave_value_list( (double )dataset_number );
+  return octave_value_list(1, octave_value((double )dataset_number) );
 }
 
 
--- kmatplot/octave/kimage.cpp
+++ kmatplot/octave/kimage.cpp
@@ -28,7 +28,7 @@
 "\n"
 "See also: kadd, kremove, kremoveall, kaddaxes, kremoveaxes, ksetapp, ksetaxes, kplot, kcontour, kmesh, ksetmatrix" )
 {
-  octave_value_list result(-1.0);
+  octave_value_list result(1, octave_value(-1.0));
   int nargin = args.length ();
 
   if ( nargin != 1 &&
@@ -132,7 +132,7 @@
    }
 
   plot_disconnect( socket_fd );
-  return octave_value_list( (double )dataset_number );
+  return octave_value_list(1, octave_value((double )dataset_number) );
 }
 
 
--- kmatplot/octave/kaddaxes.cpp
+++ kmatplot/octave/kaddaxes.cpp
@@ -45,6 +45,6 @@
  	if ( axes_number > 0 ) setAxes( axes_number );
  	plot_disconnect( socket_fd );
      }
- return octave_value_list( (double )axes_number );
+ return octave_value_list(1, octave_value((double )axes_number) );
 }
 
--- kmatplot/octave/ksetapp.cpp
+++ kmatplot/octave/ksetapp.cpp
@@ -44,6 +44,6 @@
 		 setenv( ENV_APP_NUMBER, buff, 1 );
 		}
 
-  return octave_value_list( (double )app_num );
+  return octave_value_list(1, octave_value((double )app_num) );
 }
 
--- kmatplot/octave/kcontour.cpp
+++ kmatplot/octave/kcontour.cpp
@@ -26,7 +26,7 @@
 "\n"
 "See also: ksetapp, ksetaxes, kadd, kremove, kremoveall, kaddaxes, kremoveaxes, kplot, kimage, kmesh, ksetmatrix" )
 {
-  octave_value_list result( -1.0 );
+  octave_value_list result(1, octave_value( -1.0) );
 
   int nargin = args.length();
   if ( nargin != 1 && nargin != 3  ) {
@@ -94,6 +94,6 @@
     }
 
   plot_disconnect( socket_fd );
-  return octave_value_list( (double )dataset_number );
+  return octave_value_list(1, octave_value((double )dataset_number) );
 }
 
