diff -Naru DBBalancer.org/src/config/DBBalancerConfig.hh DBBalancer/src/config/DBBalancerConfig.hh
--- DBBalancer.org/src/config/DBBalancerConfig.hh	2002-07-20 21:03:37.000000000 +0100
+++ DBBalancer/src/config/DBBalancerConfig.hh	2005-07-23 00:07:00.000000000 +0100
@@ -1,5 +1,12 @@
 #include <string>
 #include <vector>
+#include<sys/types.h> 
+#include<sys/socket.h>
+#include <netinet/in.h> 
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <sys/un.h>
+#include <iostream.h>
 
 // Interface for the configuration management. Every configuration reader and holder should implement this.
 
diff -Naru DBBalancer.org/src/DBBalancerDaemon.hh DBBalancer/src/DBBalancerDaemon.hh
--- DBBalancer.org/src/DBBalancerDaemon.hh	2002-07-21 19:01:41.000000000 +0100
+++ DBBalancer/src/DBBalancerDaemon.hh	2005-07-23 00:07:19.000000000 +0100
@@ -1,5 +1,6 @@
 #include <string>
 #include <iostream>
+#include <sys/un.h>
 
 using std::string;
 
diff -Naru DBBalancer.org/src/tests/postgres_cc/pgtest.cc DBBalancer/src/tests/postgres_cc/pgtest.cc
--- DBBalancer.org/src/tests/postgres_cc/pgtest.cc	2001-08-27 20:27:42.000000000 +0100
+++ DBBalancer/src/tests/postgres_cc/pgtest.cc	2005-07-23 00:07:49.000000000 +0100
@@ -1,7 +1,9 @@
 #include <sys/time.h>
 #include <iostream>
+#include <string>
 #include <libpq++.h>
 
+using namespace std;
 
 int main(int argc, char* argv[]) {
 
