diff -x '*~' -ur flac-1.1.2/configure.in flac-1.1.2-noogg/configure.in
--- flac-1.1.2/configure.in	2005-02-03 06:03:25.000000000 +0100
+++ flac-1.1.2-noogg/configure.in	2005-05-07 02:37:21.504620688 +0200
@@ -151,7 +151,14 @@
 fi
 
 dnl check for ogg library
-XIPH_PATH_OGG(have_ogg=yes, AC_MSG_WARN([*** Ogg development enviroment not installed - Ogg support will not be built]))
+AC_ARG_ENABLE([ogg],
+	AC_HELP_STRING([--disable-ogg], [Disable ogg support (default:test)]),
+	[ want_ogg=$enableval ], [ want_ogg=yes ] )
+
+if test "x$want_ogg" != "xno"; then
+	XIPH_PATH_OGG(have_ogg=yes, AC_MSG_WARN([*** Ogg development enviroment not installed - Ogg support will not be built]))
+fi
+
 AM_CONDITIONAL(FLaC__HAS_OGG, [test x$have_ogg = xyes])
 if test x$have_ogg = xyes ; then
 AC_DEFINE(FLAC__HAS_OGG)
