? p
Index: src/mozilla/kz-mozfilepicker.cpp
===================================================================
RCS file: /cvsroot/kazehakase/kazehakase/src/mozilla/kz-mozfilepicker.cpp,v
retrieving revision 1.11
diff -u -p -r1.11 kz-mozfilepicker.cpp
--- src/mozilla/kz-mozfilepicker.cpp	20 Mar 2005 00:50:53 -0000	1.11
+++ src/mozilla/kz-mozfilepicker.cpp	30 Mar 2005 23:13:08 -0000
@@ -120,7 +120,7 @@ NS_IMETHODIMP KzFilePicker::Init(nsIDOMW
 #ifdef MOZ_NSIFILEPICKER_NSASTRING_
 	NS_UTF16ToCString(aTitle, NS_CSTRING_ENCODING_UTF8, mTitle);
 #else
-	mTitle = nsEmbedCString(aTitle);
+	NS_UTF16ToCString(nsEmbedString(aTitle), NS_CSTRING_ENCODING_UTF8, mTitle);
 #endif
 	mMode = aMode;
 	
@@ -151,7 +151,7 @@ NS_IMETHODIMP KzFilePicker::AppendFilter
 #ifdef MOZ_NSIFILEPICKER_NSASTRING_
 	NS_UTF16ToCString(aFilter, NS_CSTRING_ENCODING_UTF8, mFilter);
 #else
-	mFilter = nsEmbedCString(aFilter);
+	NS_UTF16ToCString(nsEmbedString(aFilter), NS_CSTRING_ENCODING_UTF8, mFilter);
 #endif
 	return NS_OK;
 }
@@ -176,7 +176,7 @@ NS_IMETHODIMP KzFilePicker::GetDefaultSt
 #ifdef MOZ_NSIFILEPICKER_NSASTRING_
 	aDefaultString = mDefaultString;
 #else
-	*aDefaultString = mDefaultString.get();
+	*aDefaultString = NS_StringCloneData(mDefaultString);
 #endif
 	return NS_OK;
 }
Index: src/mozilla/kz-mozlauncher.cpp
===================================================================
RCS file: /cvsroot/kazehakase/kazehakase/src/mozilla/kz-mozlauncher.cpp,v
retrieving revision 1.13
diff -u -p -r1.13 kz-mozlauncher.cpp
--- src/mozilla/kz-mozlauncher.cpp	15 Mar 2005 07:44:26 -0000	1.13
+++ src/mozilla/kz-mozlauncher.cpp	30 Mar 2005 23:13:08 -0000
@@ -124,7 +124,7 @@ NS_IMETHODIMP KzContentHandler::PromptFo
 	filePicker->SetDefaultString(defaultFile);
 #else
 	filePicker->Init(windowInternal,
-			 PromiseFlatString(title).get(), 
+			 title.get(), 
 			 nsIFilePicker::modeSave);
 	filePicker->SetDefaultString(aDefaultFile);
 #endif
