diff -Naur imms-1.1.orig/rules.mk imms-1.1/rules.mk
--- imms-1.1.orig/rules.mk	2004-04-15 13:15:24.000000000 -0700
+++ imms-1.1/rules.mk	2004-04-19 23:52:20.000000000 -0700
@@ -14,12 +14,12 @@
 	$(AR) $(ARFLAGS) $@ $(CORE_OBJ)
 
 %.o: %.cc vars.mk
-	@$(CXX) $(CXXFLAGS) -M -E $< > .$*.d
-	$(CXX) $(CXXFLAGS) -c $< -o $@
+	@$(CXX) $(CPPFLAGS) $(CXXFLAGS) -M -E $< > .$*.d
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $< -o $@
 
 %.o: %.c vars.mk
-	@$(CXX) $(CPPFLAGS) -M -E $< > .$*.d
-	$(CC) $(CPPFLAGS) -c $< -o $@
+	@$(CC) $(CPPFLAGS) $(CFLAGS) -M -E $< > .$*.d
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
 
 .PHONY: clean distclean
 
@@ -48,26 +48,8 @@
 	tar -C /tmp/ -cz imms-$(VERSION)/ -f imms-$(VERSION).tar.gz
 	rm -rf /tmp/imms-$(VERSION)/
 
-.PHONY: install install-user install-system user-message system-message
+.PHONY: install
 
-ifeq ($(shell id -u), 0)
-    install: system-message install-system
-else
-    install: user-message install-user
-endif
-
-system-message:
-	$(warning Defaulting to installing for all users.)
-	$(warning Use 'make install-user' to install for the current user only.)
-
-install-system: libimms.so
-	${INSTALL_PROGRAM} libimms.so ${DESTDIR}`xmms-config --visualization-plugin-dir`
-
-user-message:
-	$(warning Defaulting to installing for current user only.)
-	$(warning Use 'make install-system' to install for all users.)
-
-install-user: libimms.so
-	mkdir -p ${HOME}/.xmms/Plugins/Visualization/
-	rm -f ${HOME}/.xmms/Plugins/General/libimms.so
-	${INSTALL_PROGRAM} libimms.so ${HOME}/.xmms/Plugins/Visualization/
+install: all
+	mkdir -p $(DESTDIR)`xmms-config --visualization-plugin-dir`
+	$(INSTALL_PROGRAM) libimms.so $(DESTDIR)`xmms-config --visualization-plugin-dir`
diff -Naur imms-1.1.orig/vars.mk.in imms-1.1/vars.mk.in
--- imms-1.1.orig/vars.mk.in	2004-04-15 13:15:24.000000000 -0700
+++ imms-1.1/vars.mk.in	2004-04-19 23:54:58.000000000 -0700
@@ -2,7 +2,6 @@
 CXX = @CXX@
 VERSION = @PACKAGE_VERSION@
 INSTALL_PROGRAM = @INSTALL@
-DESTDIR = @prefix@
 
 SHELL = bash
 XMMS_OBJ = @PLUGIN_OBJ@ interface.o player.o libimmscore.a 
@@ -10,8 +9,9 @@
 	   fetcher.o picker.o spectrum.o songinfo.o comm.o server.o \
 	   regexx.o utils.o strmanip.o levenshtein.o xidle.o md5.o
 
-CPPFLAGS = `xmms-config --cflags` @XCPPFLAGS@ -Wall -fPIC -D_REENTRANT
-CXXFLAGS = ${CPPFLAGS} -fno-rtti
+CPPFLAGS = `xmms-config --cflags` @XCPPFLAGS@ @CPPFLAGS@ -D_REENTRANT
+CFLAGS = -fPIC @CFLAGS@ 
+CXXFLAGS = -fPIC -fno-rtti @CXXFLAGS@
 LDFLAGS = `xmms-config --libs` @LIBS@
 ARFLAGS = rs
 
