diff -ru gtkam-0.1.16.1.orig/configure.ac gtkam-0.1.16.1/configure.ac
--- gtkam-0.1.16.1.orig/configure.ac	2008-12-21 11:19:14.000000000 +0100
+++ gtkam-0.1.16.1/configure.ac	2008-12-21 11:19:28.000000000 +0100
@@ -61,12 +61,7 @@
 dnl ---------------------------------------------------------------------------
 dnl i18n support
 dnl ---------------------------------------------------------------------------
-GP_GETTEXT_HACK([],[Lutz Müller and others],[${MAIL_GPHOTO_TRANSLATION}])
 ALL_LINGUAS="be cs da de es fr hu it ja nb nl pa pl pt_BR pt ru sk sl sv uk vi zh_CN"
-AM_GNU_GETTEXT_VERSION([0.14.1])
-AM_GNU_GETTEXT([external])
-AM_ICONV()
-GP_GETTEXT_FLAGS()
 
 dnl We cannot use AC_DEFINE_UNQUOTED() for these definitions, as
 dnl we require make to do insert the proper $(datadir) value
