--- a/examples/raconvert/raconvert.c
+++ b/examples/raconvert/raconvert.c
@@ -299,7 +299,7 @@
          switch (ind) {
             case RASCIIDEBUG: {
                for (x = 0, ind = -1; x < RASCII_MAXDEBUG; x++) {
-                  if (!(strncasecmp (mode->mode, RaConvertDaemonModes[x], 3))) {
+                  if (!(strncasecmp (mode->mode, RaConvertDaemonModes[x], 2))) {
                      ArgusDebugMode |= 0x01 << x;
                      switch (ind) {
                         case RASCII_DEBUGTASKS:
