diff -u lynx2-8-5/src/HTTP.c lynx2-8-5/src/HTTP.c
--- lynx2-8-5/src/HTML.c	2004/10/21 17:04:47	1.5
+++ lynx2-8-5/src/HTML.c	2005/11/27 21:45:19	1.6
@@ -2104,7 +2104,8 @@
 	    int enval = 2;
 
 	    column = HText_getCurrentColumn(me->text);
-	    if (present[HTML_TAB_TO]) {
+	    if (present[HTML_TAB_TO] &&
+			value[HTML_TAB_TO] && *value[HTML_TAB_TO]) {
 		/*
 		 *  TO has priority over INDENT if both are present. - FM
 		 */
