diff -p -r zh-stardict-1.3/src/image.h zh-stardict-1.3.fix/src/image.h
*** zh-stardict-1.3/src/image.h	2000-05-26 09:33:16.000000000 +0800
--- zh-stardict-1.3.fix/src/image.h	2002-10-12 06:34:04.000000000 +0800
***************
*** 17,23 ****
  #ifndef IMAGE_H
  #define IMAGE_H
  
! #include <stdiostream.h>
  #include <stdlib.h>
  #include <memory.h>
  #include <string.h>
--- 17,24 ----
  #ifndef IMAGE_H
  #define IMAGE_H
  
! //#include <stdiostream.h>
! #include <iostream.h>
  #include <stdlib.h>
  #include <memory.h>
  #include <string.h>
