diff -urN pvfs-2.7.1/Makefile.in pvfs-2.7.1_patched/Makefile.in
--- pvfs-2.7.1/Makefile.in	2008-04-14 11:37:32.000000000 -0400
+++ pvfs-2.7.1_patched/Makefile.in	2008-04-16 22:00:55.354355245 -0400
@@ -984,10 +984,13 @@
 	install -d $(KMOD_DIR)
 	install -m 755 src/kernel/linux-2.6/pvfs2.ko $(KMOD_DIR)
 
-.PHONY: kmod_install
-kmod_install: kmod kernapps just_kmod_install
+.PHONY: kernapps_install
+kernapps_install: kernapps
 	install -d $(sbindir)
 	install -m 755 $(KERNAPPS) $(KERNAPPSTHR) $(sbindir)
+
+.PHONY: kmod_install
+kmod_install: kmod kernapps_install just_kmod_install
 endif
 
 ifneq (,$(LINUX24_KERNEL_SRC))
