diff -Naur gpsdrive-2.09-orig/configure.ac gpsdrive-2.09/configure.ac
--- gpsdrive-2.09-orig/configure.ac	2006-10-16 16:00:44.000000000 -0600
+++ gpsdrive-2.09/configure.ac	2006-10-16 16:01:35.000000000 -0600
@@ -128,7 +128,7 @@
 
 AC_PROG_INSTALL
 
-CFLAGS="$CFLAGS -g -Wall -Wno-format-y2k -pipe $OPT_CFLAGS"
+CFLAGS="$CFLAGS -fno-strict-aliasing -Wall -Wno-format-y2k -pipe $OPT_CFLAGS"
 
 dnl ************************
 dnl Check for standard headers
@@ -153,6 +153,14 @@
 	CFLAGS="$CFLAGS `$PCRE_CONFIG --cflags`"
 fi
 
+dnl --------------------------
+dnl Check for package libglib2.0-0
+dnl 26/03/2006 DP  Added
+dnl 06/06/2006 DP  Reduced from version 2.8.6 to 2.8.5
+dnl --------------------------
+dnl AM_PATH_GLIB_2_0([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND [, MODULES]]]])
+dnl MODULES  gmodule, gobject or gthread
+AM_PATH_GLIB_2_0(2.8.5, , AC_ERROR(need libglib2.0 at least version 2.8.5))
 
 localedir='${prefix}/share/locale'
 AC_SUBST(localedir)
