--- src/util/CMakeLists.txt.orig	2008-09-21 20:15:48.379309613 -0700
+++ src/util/CMakeLists.txt	2008-09-21 20:17:36.012310364 -0700
@@ -25,13 +25,14 @@
   install(TARGETS worldgen DESTINATION ${BIN_INSTALL_DIR})
 endif (GDAL_FOUND)
 
-if (GDAL_FOUND AND GTK2_FOUND)
-  add_executable(gmapview ${gmapview_SRCS})
-  target_link_libraries(gmapview
-    ${MAP_LIBRARY}
-    ${GDAL_LIBRARIES}
-    ${GTK2_LIBRARIES}
-  )
-  install(TARGETS gmapview DESTINATION ${BIN_INSTALL_DIR})
-endif (GDAL_FOUND AND GTK2_FOUND)
+## this bombs because there's no implementation of drawmarkers
+#if (GDAL_FOUND AND GTK2_FOUND)
+#  add_executable(gmapview ${gmapview_SRCS})
+#  target_link_libraries(gmapview
+#    ${MAP_LIBRARY}
+#    ${GDAL_LIBRARIES}
+#    ${GTK2_LIBRARIES}
+#  )
+#  install(TARGETS gmapview DESTINATION ${BIN_INSTALL_DIR})
+#endif (GDAL_FOUND AND GTK2_FOUND)
 
