diff -Nru libast-0.5.old/include/libast.h libast-0.5/include/libast.h
--- libast-0.5.old/include/libast.h	2004-07-08 13:39:36.837764880 -0400
+++ libast-0.5/include/libast.h	2004-07-08 13:40:21.287007568 -0400
@@ -337,6 +337,7 @@
         /* fill l with c. */ \
         l = (c) | (c)<<8; \
         l |= l<<16; \
+        l |= l<<32; \
         MEMSET_LONG(); \
  \
         /* fill in 1-byte chunks until boundary of long is reached. */ \
