--- install-xetex	2006-10-30 13:09:09.795732198 +0800
+++ install-xetex.gentoo	2006-10-30 13:09:27.064330965 +0800
@@ -30,9 +30,10 @@
 if [ -L ${texbin} ]; then
 	texbin=`readlink ${texbin}`
 fi
-texbindir=`dirname ${texbin}`
+texbindir=${D}`dirname ${texbin}`
 
 # copy the xetex binary to the same dir as standard tex
+mkdir -p ${texbindir}
 cp -pf Work/texk/web2c/xetex ${texbindir}/xetex
 echo ${texbindir}/xetex >> ${filelist}
 if [ -e Work/texk/xdv2pdf/xdv2pdf ]; then
@@ -48,7 +49,7 @@
 fi
 
 # copy our texmf additions into the local texmf tree
-texmflocal=`kpsewhich --var-value TEXMFLOCAL`
+texmflocal=${D}usr/share/texmf
 if [ "x${texmflocal}" == "x" ]; then
 	# if --var-value didn't work, try to find the definition in the texmf.cnf file
 	CNF=`kpsewhich texmf.cnf`
@@ -74,8 +75,8 @@
 echo ${texmflocal}/web2c/xetex.pool >> ${filelist}
 
 # update kpathsearch databases because we've installed a bunch of stuff
-texhash ${texmflocal}
+#texhash ${texmflocal}
 
 # this will build the format files on the system where this script is run;
 # for packaged binary installations, we must repeat this on the target
-sh ./rebuild-formats
+#sh ./rebuild-formats
