diff -Naur happy-1.14.orig/happy/src/Main.lhs happy-1.14/happy/src/Main.lhs
--- happy-1.14.orig/happy/src/Main.lhs	2004-10-21 17:03:27.923088968 +0200
+++ happy-1.14/happy/src/Main.lhs	2004-10-21 17:03:55.192943320 +0200
@@ -417,20 +417,11 @@
 
 CPP is turned on for -fglasogw-exts, so we can use conditional compilation:
 
-> import_glaexts = "#if __GLASGOW_HASKELL__ >= 503\n\ 
-> 		   \import GHC.Exts\n\ 
->		   \#else\n\ 
->		   \import GlaExts\n\ 
->		   \#endif\n"
-
-> import_debug = "#if __GLASGOW_HASKELL__ >= 503\n\ 
-> 		   \import System.IO\n\ 
-> 		   \import System.IO.Unsafe\n\ 
-> 		   \import Debug.Trace\n\ 
->		   \#else\n\ 
->		   \import IO\n\ 
->		   \import IOExts\n\ 
->		   \#endif\n"
+> import_glaexts = 
+>   "#if __GLASGOW_HASKELL__ >= 503\nimport GHC.Exts\n#else\nimport GlaExts\n#endif\n"
+
+> import_debug = 
+>   "#if __GLASGOW_HASKELL__ >= 503\nimport System.IO\nimport System.IO.Unsafe\nimport Debug.Trace\n#else\nimport IO\nimport IOExts\n#endif\n"
 
 ------------------------------------------------------------------------------
 Extract various command-line options.
@@ -468,8 +459,8 @@
 >	= if OptUseCoercions `elem` cli 
 >	     then if OptGhcTarget `elem` cli
 >			then return True
->			else dieHappy "-c/--coerce may only be used \ 
->				      \in conjunction with -g/--ghc\n"
+>			else dieHappy 
+>                              "-c/--coerce may only be used in conjunction with -g/--ghc\n"
 >	     else return False
 
 > getGhc cli = return (OptGhcTarget `elem` cli)
diff -Naur happy-1.14.orig/happy/src/Version.hs happy-1.14/happy/src/Version.hs
--- happy-1.14.orig/happy/src/Version.hs	2004-10-21 17:03:27.929088056 +0200
+++ happy-1.14/happy/src/Version.hs	2004-10-21 17:03:44.131624896 +0200
@@ -1,4 +1,3 @@
 module Version (version) where
 
-version = tail "\ 
-  \ HAPPY_VERSION"
+version = "1.14"
