--- mysql-connector-net-1.0.4-old/mysqlclient/AssemblyInfo.cs	2005-01-20 10:30:52.000000000 -0600
+++ mysql-connector-net-1.0.4/mysqlclient/AssemblyInfo.cs	2005-05-03 17:11:11.000000000 -0500
@@ -74,7 +74,5 @@
 //       documentation for more information on this.
 //
 [assembly: AssemblyDelaySign(false)]
-#if FINAL
 [assembly: AssemblyKeyFile(@"d:\work\connector-net\cnet.snk")]
-#endif
 [assembly: AssemblyKeyName("")]
