diff -aurN mysql-connector-net-1.0.4-old/Client.build mysql-connector-net-1.0.4/Client.build
--- mysql-connector-net-1.0.4-old/Client.build	2005-01-20 10:56:06.000000000 -0600
+++ mysql-connector-net-1.0.4/Client.build	2005-04-29 08:38:20.000000000 -0500
@@ -32,11 +32,8 @@
 
 	<!-- builds the client on all supported frameworks -->
 	<target name="allframeworks">
-		<property name="framework" value="net-1.0"/>
-		<call target="client"/>
-		<property name="framework" value="net-1.1"/>
+		<property name="framework" value="mono-1.0"/>
 		<call target="client"/>
-		<call target="Docs"/>
 	</target>
 
 	<!-- core managed driver target -->	
@@ -44,14 +41,18 @@
 		<property name="nant.settings.currentframework" value="${framework}"/>
 		<mkdir dir="bin/${framework}/${buildType}" failonerror="false"/>
 		<csc define="${defines}" output="bin/${framework}/${buildType}/MySql.Data.dll" target="library" 
-			debug="${debug}" optimize="${opt}" 
-			doc="bin/${framework}/${buildType}/MySql.Data.xml">
+			debug="${debug}" optimize="${opt}" >
 			<arg value="/nowarn:0679"/>
 			<references>
 				<include name="ICSharpCode.SharpZipLib.dll"/>
+				<include name="System.Data.dll"/>
+				<include name="System.dll"/>
+				<include name="Mono.Posix.dll"/>
+				<include name="System.Drawing.dll"/>
 			</references>
 			<sources refid="Source"/>
 		</csc>
+		<!--
 		<csc output="bin/${framework}/${buildType}/MySql.Data.Tests.dll" target="library" debug="${debug}" optimize="${opt}">
 			<arg value="/nowarn:0679"/>
 			<sources>
@@ -61,7 +62,7 @@
 				<include name="bin/${framework}/${buildType}/MySql.Data.dll"/>
 				<include name="${nunit}/nunit.framework.dll"/>
 			</references>
-		</csc>
+		    </csc> -->
 	</target>
 
 	<!-- core managed driver target 1.0 -->	
@@ -75,10 +76,14 @@
 		<property name="framework" value="net-1.1"/>
 		<call target="client"/>
 	</target>
-	
+
+        <target name="mono-1.0" description="1.0 Mono target">
+                <property name="framework" value="mono-1.0"/>
+                <call target="client"/>
+        </target>
 
 	<!-- testing targets -->
-	<target name="alltests" depends="test-net-1.1, test-net-1.0">
+	<target name="alltests" depends="test-net-1.1, test-net-1.0, test-mono-1.0">
 	</target>
 	
 	<target name="test-net-1.1" depends="net-1.1">
@@ -90,6 +95,12 @@
 		<property name="framework" value="net-1.0"/>
 		<call target="testclient"/>
 	</target>
+	
+	<target name="test-mono-1.0" depends="mono-1.0">
+		<property name="framework" value="mono-1.0"/>
+		<call target="testclient"/>
+	</target>
+
 
     <!-- target to run the test suite -->
 	<target name="testclient" description="Target to run NUnit tests">
diff -aurN mysql-connector-net-1.0.4-old/mysqlclient/SharedMemoryStream.cs mysql-connector-net-1.0.4/mysqlclient/SharedMemoryStream.cs
--- mysql-connector-net-1.0.4-old/mysqlclient/SharedMemoryStream.cs	2005-01-14 09:32:08.000000000 -0600
+++ mysql-connector-net-1.0.4/mysqlclient/SharedMemoryStream.cs	2005-04-28 14:52:45.000000000 -0500
@@ -43,7 +43,8 @@
 
 		private uint	EVENT_ALL_ACCESS = 0x001F0003;
 		private uint	FILE_MAP_WRITE = 0x2;
-		private int		BUFFERLENGTH = 16004;
+		private int	BUFFERLENGTH = 16004;
+		private uint	BUFFERLENGTH_UINT = 16004;
 
 		public SharedMemoryStream(string memName)
 		{
@@ -91,7 +92,7 @@
 			string dataMemoryName = memoryName + "_" + connectNumber;
 			dataMap = OpenFileMapping( FILE_MAP_WRITE, false, 
 				dataMemoryName + "_DATA" );
-			dataView = MapViewOfFile( dataMap, FILE_MAP_WRITE, 0, 0, (UIntPtr)BUFFERLENGTH );
+			dataView = MapViewOfFile( dataMap, FILE_MAP_WRITE, 0, 0, (UIntPtr)BUFFERLENGTH_UINT );
 
 			serverWrote = new AutoResetEvent(false);
 			serverWrote.Handle = OpenEvent( EVENT_ALL_ACCESS, false, 
@@ -151,7 +152,7 @@
 		{
 			try 
 			{
-				dataView = MapViewOfFile( dataMap, FILE_MAP_WRITE, 0, 0, (UIntPtr)BUFFERLENGTH );
+				dataView = MapViewOfFile( dataMap, FILE_MAP_WRITE, 0, 0, (UIntPtr)BUFFERLENGTH_UINT );
 				if (dataView == IntPtr.Zero) return true;
 				return false;
 			}
diff -aurN mysql-connector-net-1.0.4-old/mysqlclient/common/StreamCreator.cs mysql-connector-net-1.0.4/mysqlclient/common/StreamCreator.cs
--- mysql-connector-net-1.0.4-old/mysqlclient/common/StreamCreator.cs	2005-01-12 17:09:34.000000000 -0600
+++ mysql-connector-net-1.0.4/mysqlclient/common/StreamCreator.cs	2005-04-29 11:45:43.000000000 -0500
@@ -103,7 +103,7 @@
 
 			try
 			{
-				UnixEndPoint endPoint = new UnixEndPoint (hostList[0]);
+				UnixEndPoint endPoint = new UnixEndPoint (hostList);
 				socket.Connect (endPoint);
 				return new NetworkStream (socket, true);
 			}
