diff -urN xmlrpc-c-1.14.07.old/Makefile.config.in xmlrpc-c-1.14.07/Makefile.config.in
--- xmlrpc-c-1.14.07.old/Makefile.config.in	2008-03-27 14:17:42.000000000 -0400
+++ xmlrpc-c-1.14.07/Makefile.config.in	2008-07-27 23:52:02.241422830 -0400
@@ -292,7 +292,7 @@
 # PREFIX is designed to be overridden at make time if the user decides
 # he doesn't like the default specified at 'configure' time.
 
-LIBINST_DIR = $(PREFIX)/lib
+LIBINST_DIR = $(PREFIX)/@libdir@
 HEADERINST_DIR = $(PREFIX)/include
 PROGRAMINST_DIR = $(PREFIX)/bin
 
