Index: xmlrpc-c-1.05/include/Makefile
===================================================================
--- xmlrpc-c-1.05.orig/include/Makefile
+++ xmlrpc-c-1.05/include/Makefile
@@ -45,6 +45,7 @@ HEADERINST_PREFIX = /xmlrpc-c
 ifeq ($(MUST_BUILD_CLIENT),yes)
   HEADERS_TO_INSTALL += \
     xmlrpc-c/client.h \
+    xmlrpc-c/client_global.h \
     xmlrpc-c/transport.h \
 
   ifeq ($(ENABLE_CPLUSPLUS),yes)
