diff -NrU5 xmlrpc-c-1.16.04.orig/common.mk xmlrpc-c-1.16.04/common.mk
--- xmlrpc-c-1.16.04.orig/common.mk	2008-11-29 17:31:12.000000000 +0100
+++ xmlrpc-c-1.16.04/common.mk	2008-11-29 17:35:25.000000000 +0100
@@ -134,14 +134,14 @@
   TARGET_SHARED_LE_LIBS =
 endif
 
 #------ the actual rules ----------------------------------------------------
 $(TARGET_SHARED_LIBRARIES) dummyshlib:
-	$(CCLD) $(LDFLAGS_SHLIB) $(LIBDEP) -o $@ $(LIBOBJECTS) $(LADD)
+	$(CCLD) $(LADD) $(LDFLAGS_SHLIB) $(LIBOBJECTS)  $(LIBDEP) -o $@  
 
 $(TARGET_SHARED_LIBS_PP) dummyshlibpp:
-	$(CXXLD) $(LDFLAGS_SHLIB) $(LIBDEP) -o $@ $(LIBOBJECTS) $(LADD)
+	$(CXXLD) $(LADD) $(LDFLAGS_SHLIB) $(LIBOBJECTS)  $(LIBDEP) -o $@  
 #----------------------------------------------------------------------------
 
 LIBXMLRPC_UTIL_DIR = $(BLDDIR)/lib/libutil
 
 ifneq ($(OMIT_LIBXMLRPC_UTIL_RULE),Y)
diff -NrU5 xmlrpc-c-1.16.04.orig/dylib-common.make xmlrpc-c-1.16.04/dylib-common.make
--- xmlrpc-c-1.16.04.orig/dylib-common.make	2008-11-29 17:31:12.000000000 +0100
+++ xmlrpc-c-1.16.04/dylib-common.make	2008-11-29 17:35:25.000000000 +0100
@@ -3,13 +3,13 @@
 # See unix-common.make for an explanation of this file.  This file is
 # analogous to unix-common.make, but is for an Irix system.
 
 SONAME = $(@:%.$(MIN)=%)
 
-SHLIB_CMD = $(CCLD) $(LDFLAGS_SHLIB) -o $@ $^ $(LADD)
+SHLIB_CMD = $(CCLD) $(LADD) $(LDFLAGS_SHLIB) -o $@ $^ 
 
-SHLIBPP_CMD = $(CXXLD) $(LDFLAGS_SHLIB) -o $@ $^ $(LADD)
+SHLIBPP_CMD = $(CXXLD) $(LADD) $(LDFLAGS_SHLIB) -o $@ $^ 
 
 SHLIB_LE_TARGETS = $(call shliblefn, $(SHARED_LIBS_TO_BUILD))
 
 $(SHLIB_LE_TARGETS):%.$(SHLIB_SUFFIX):%.$(MAJ).$(MIN).$(SHLIB_SUFFIX)
 	rm -f $@
diff -NrU5 xmlrpc-c-1.16.04.orig/irix-common.make xmlrpc-c-1.16.04/irix-common.make
--- xmlrpc-c-1.16.04.orig/irix-common.make	2008-11-29 17:31:12.000000000 +0100
+++ xmlrpc-c-1.16.04/irix-common.make	2008-11-29 17:35:25.000000000 +0100
@@ -3,11 +3,11 @@
 # See unix-common.make for an explanation of this file.  This file is
 # analogous to unix-common.make, but is for an Irix system.
 
 SONAME = $(@:%.$(MAJ):%)
 
-SHLIB_CMD = $(CCLD) $(LDFLAGS_SHLIB) -o $@ $^ $(LADD)
+SHLIB_CMD = $(CCLD) $(LADD) $(LDFLAGS_SHLIB) -o $@ $^ 
 
 SHLIB_LE_TARGETS = $(call shliblefn, $(SHARED_LIBS_TO_BUILD))
 
 $(SHLIB_LE_TARGETS):%:%.$(MAJ)
 	rm -f $@
diff -NrU5 xmlrpc-c-1.16.04.orig/unix-common.make xmlrpc-c-1.16.04/unix-common.make
--- xmlrpc-c-1.16.04.orig/unix-common.make	2008-11-29 17:31:12.000000000 +0100
+++ xmlrpc-c-1.16.04/unix-common.make	2008-11-29 17:35:25.000000000 +0100
@@ -49,11 +49,11 @@
 # SONAME is the name of the library file being built, with the minor
 #   version number cut off.  E.g. if we're building libfoo.so.1.2, SONAME
 #   is libfoo.so.1 .
 SONAME = $(@:%.$(MIN)=%)
 
-SHLIB_CMD = $(CCLD) $(LDFLAGS_SHLIB) -o $@ $^ $(LADD)
+SHLIB_CMD = $(CCLD) $(LADD) $(LDFLAGS_SHLIB) -o $@ $^ 
 
 SHLIB_LE_TARGETS = $(call shliblefn, $(SHARED_LIBS_TO_BUILD))
 
 $(SHLIB_LE_TARGETS):%:%.$(MAJ).$(MIN)
 	rm -f $@
