diff -ur color-theme-6.5.5.orig/color-theme.el color-theme-6.5.5/color-theme.el
--- color-theme-6.5.5.orig/color-theme.el	2006-01-16 11:45:51.000000000 -0600
+++ color-theme-6.5.5/color-theme.el	2006-01-16 11:57:53.000000000 -0600
@@ -586,9 +586,10 @@
   (color-theme-mode))
 
 (require 'easymenu)
-(easy-menu-add-item nil '("Tools") "--")
-(easy-menu-add-item  nil '("Tools")
-  ["Color Themes" color-theme-select t])
+(let ((item (if (< emacs-major-version 22) "tools" "Tools")))
+  (easy-menu-add-item nil `(,item) "--")
+  (easy-menu-add-item  nil `(,item)
+    ["Color Themes" color-theme-select t]))
 
 (defun color-theme-mode ()
   "Major mode to select and install color themes.
