From 9783f0a3cf1449e2566de295b2ffa1ad17519fdc Mon Sep 17 00:00:00 2001
From: Guillaume Desmottes <guillaume.desmottes@collabora.co.uk>
Date: Tue, 15 Jun 2010 11:17:09 +0000
Subject: init GError so 'make check' will be happy (#621409)

---
diff --git a/libempathy/empathy-dispatcher.c b/libempathy/empathy-dispatcher.c
index c5908a8..f616a99 100644
--- a/libempathy/empathy-dispatcher.c
+++ b/libempathy/empathy-dispatcher.c
@@ -344,7 +344,7 @@ dispatcher_connection_invalidated_cb (TpConnection *connection,
   if (connection_data != NULL)
     {
       GList *l;
-      GError *error;
+      GError *error = NULL;
 
       error = g_error_new_literal (domain, code, message);
 
--
cgit v0.8.3.1
