--- netcdf-4.0.1.orig/configure.ac	2009-03-26 13:10:02.000000000 +0100
+++ netcdf-4.0.1/configure.ac	2010-01-28 19:01:47.156238559 +0100
@@ -801,7 +801,7 @@
              *)
                 # If it's a gnu compiler, guess f2c.
                 if test "x$ac_cv_fc_compiler_gnu" = xyes; then
-                   AC_DEFINE(f2cFortran, [1], [Turned on by netCDF configure.])
+                   AC_DEFINE(gFortran, [1], [Turned on by netCDF configure.])
                 fi
                 ;;
          esac

--- netcdf-4.0.1.orig/fortran/ncfortran.h	2009-02-19 13:50:25.000000000 +0100
+++ netcdf-4.0.1/fortran/ncfortran.h	2009-09-24 21:10:08.000000000 +0200
@@ -660,7 +660,7 @@
  * The following is for f2c-support only.
  */
 
-#if defined(f2cFortran) && !defined(pgiFortran)
+#if defined(f2cFortran) && !defined(pgiFortran) && !defined(gFortran)
 
 /*
  * The f2c(1) utility on BSD/OS and Linux systems adds an additional
--- netcdf-4.0.1.orig/nf_test/fortlib.c	2009-09-24 21:11:12.000000000 +0200
+++ netcdf-4.0.1/nf_test/fortlib.c	2009-09-24 21:11:36.000000000 +0200
@@ -14,7 +14,7 @@
 #include "../fortran/ncfortran.h"
 
 
-#if defined(f2cFortran) && !defined(pgiFortran)
+#if defined(f2cFortran) && !defined(pgiFortran) && !defined(gFortran)
 /*
  * The f2c(1) utility on BSD/OS and Linux systems adds an additional
  * underscore suffix (besides the usual one) to global names that have
