diff -ur fiveam_1.2.3.orig/src/check.lisp fiveam_1.2.3/src/check.lisp
--- fiveam_1.2.3.orig/src/check.lisp	2004-11-15 07:31:52.000000000 -0600
+++ fiveam_1.2.3/src/check.lisp	2005-05-23 14:37:14.000000000 -0500
@@ -10,7 +10,7 @@
      ,@body))
 
 (eval-when (:compile-toplevel :load-toplevel :execute)
-  (def-special-enviroment run-state ()
+  (def-special-environment run-state ()
     result-list
     current-test))
 
diff -ur fiveam_1.2.3.orig/src/run.lisp fiveam_1.2.3/src/run.lisp
--- fiveam_1.2.3.orig/src/run.lisp	2004-11-15 07:31:52.000000000 -0600
+++ fiveam_1.2.3/src/run.lisp	2005-05-23 14:37:14.000000000 -0500
@@ -96,7 +96,7 @@
 
 (defun return-result-list (test-lambda)
   "Run the test function TEST-LAMBDA and return a list of all
-  test results generated, does not modify the special enviroment
+  test results generated, does not modify the special environment
   variable RESULT-LIST."
   (bind-run-state ((result-list '())) 
     (funcall test-lambda)
