--- gwget-0.94/configure.in	2005-03-13 15:17:08.000000000 -0500
+++ gwget-0.94.new/configure.in	2005-04-21 22:28:27.148842056 -0400
@@ -116,12 +116,26 @@
                   $EPIPHANY_PKGCONFIG >= $EPIPHANY_REQUIRED)
 AC_SUBST(EPIPHANY_DEPENDENCY_CFLAGS)
 AC_SUBST(EPIPHANY_DEPENDENCY_LIBS)
+
+if test $EPHY_VER = "1.2"; then
+ 	 EPIPHANY_EXTENSIONS_DIR="`$PKG_CONFIG --variable=libdir epiphany-1.2`/epiphany/extensions"
+  	 else
+  	 EPIPHANY_EXTENSIONS_DIR="`$PKG_CONFIG --variable=extensionsdir epiphany-$EPHY_VER`"
+ 	 fi
+ 	 AC_SUBST([EPIPHANY_EXTENSIONS_DIR])
 else
 	enable_epiphany_extension=no
 fi
 
 AM_CONDITIONAL(ENABLE_EPIPHANY_EXTENSION, test "x$enable_epiphany_extension" = "xyes")
 
+ AC_ARG_WITH([extension-prefix],
+   	     [  --with-extension-prefix=path              Epiphany extension prefix],
+      	     [extension_prefix=$withval],
+      	     [extension_prefix=/]
+      	 )
+  	 EXTENSION_PREFIX="$extension_prefix"
+  	 AC_SUBST([EXTENSION_PREFIX])
 
 dnl Stuff for the CORBA interface
 AC_PATH_PROG(GLIB_GENMARSHAL, glib-genmarshal)
--- gwget-0.94/epiphany-extension/Makefile.am	2004-10-14 17:21:28.000000000 -0400
+++ gwget-0.94.new/epiphany-extension/Makefile.am	2005-04-22 21:50:34.017730728 -0400
@@ -1,7 +1,7 @@
 # The extension needs to installed into Epiphany's prefix,
 # not gwget's prefix.  
 # ...but that causes problems w/ make distcheck.
-extensiondir = $(libdir)/epiphany/extensions
+extensiondir = $(EXTENSION_PREFIX)/$(EPIPHANY_EXTENSIONS_DIR)
 
 INCLUDES =				\
  	$(EPIPHANY_DEPENDENCY_CFLAGS)	\
