diff -r -u mdbtools-0.4/configure.in mdbtools-0.4pbd/configure.in
--- mdbtools-0.4/configure.in	2002-04-13 18:22:26.000000000 -0500
+++ mdbtools-0.4pbd/configure.in	2002-10-03 08:52:04.000000000 -0500
@@ -114,7 +114,7 @@
 	],[])
 
 AC_CHECK_LIB(readline, add_history, [
-	READLINE_LIBS="$READLINE_LIBS -lreadline"
+	READLINE_LIBS="$READLINE_LIBS -lreadline -ltermcap"
 	AC_DEFINE(HAVE_READLINE,1,[ ]) 
 	],[])
 
