Index: ardour-2.0.3/libs/surfaces/powermate/SConscript
===================================================================
--- ardour-2.0.3.orig/libs/surfaces/powermate/SConscript
+++ ardour-2.0.3/libs/surfaces/powermate/SConscript
@@ -49,7 +49,7 @@ Default(libardour_powermate)
 if env['NLS']:
 	i18n (powermate, powermate_files, env)
 	      
-env.Alias('install', env.Install(os.path.join(install_prefix, 'lib/ardour2/surfaces'), libardour_powermate))
+env.Alias('install', env.Install(os.path.join(install_prefix, env['LIBDIR'], 'ardour2', 'surfaces'), libardour_powermate))
 
 env.Alias('tarball', env.Distribute (env['DISTTREE'],
 				     [ 'SConscript' ] +
