diff -Naur xpdf-3.00.old/splash/SplashT1Font.h xpdf-3.00/splash/SplashT1Font.h
--- xpdf-3.00.old/splash/SplashT1Font.h	2004-01-22 09:26:44.000000000 +0800
+++ xpdf-3.00/splash/SplashT1Font.h	2005-06-08 10:09:56.000000000 +0800
@@ -16,6 +16,7 @@
 #endif
 
 #include "SplashFont.h"
+#include "SplashT1FontFile.h"
 
 //------------------------------------------------------------------------
 // SplashT1Font
diff -Naur xpdf-3.00.old/xpdf/TextOutputDev.h xpdf-3.00/xpdf/TextOutputDev.h
--- xpdf-3.00.old/xpdf/TextOutputDev.h	2004-01-22 09:26:45.000000000 +0800
+++ xpdf-3.00/xpdf/TextOutputDev.h	2005-06-08 10:15:45.000000000 +0800
@@ -26,6 +26,10 @@
 class GfxState;
 class UnicodeMap;
 
+class TextBlock;
+class TextPage;
+class TextLineFrag;
+
 //------------------------------------------------------------------------
 
 typedef void (*TextOutputFunc)(void *stream, char *text, int len);
