diff -urN xpdf-3.01/Makefile.in xpdf-3.01-pdftoppm-no_X-patch/Makefile.in
--- xpdf-3.01/Makefile.in	2005-08-17 07:34:30.000000000 +0200
+++ xpdf-3.01-pdftoppm-no_X-patch/Makefile.in	2005-09-12 23:12:25.272986000 +0200
@@ -82,7 +82,7 @@
 	$(INSTALL_PROGRAM) xpdf/pdftotext$(EXE) $(DESTDIR)@bindir@/pdftotext$(EXE)
 	$(INSTALL_PROGRAM) xpdf/pdfinfo$(EXE) $(DESTDIR)@bindir@/pdfinfo$(EXE)
 	$(INSTALL_PROGRAM) xpdf/pdffonts$(EXE) $(DESTDIR)@bindir@/pdffonts$(EXE)
-@X@	$(INSTALL_PROGRAM) xpdf/pdftoppm$(EXE) $(DESTDIR)@bindir@/pdftoppm$(EXE)
+	$(INSTALL_PROGRAM) xpdf/pdftoppm$(EXE) $(DESTDIR)@bindir@/pdftoppm$(EXE)
 	$(INSTALL_PROGRAM) xpdf/pdfimages$(EXE) $(DESTDIR)@bindir@/pdfimages$(EXE)
 	-mkdir -p $(DESTDIR)@mandir@/man1
 @X@	$(INSTALL_DATA) $(srcdir)/doc/xpdf.1 $(DESTDIR)@mandir@/man1/xpdf.1
@@ -90,7 +90,7 @@
 	$(INSTALL_DATA) $(srcdir)/doc/pdftotext.1 $(DESTDIR)@mandir@/man1/pdftotext.1
 	$(INSTALL_DATA) $(srcdir)/doc/pdfinfo.1 $(DESTDIR)@mandir@/man1/pdfinfo.1
 	$(INSTALL_DATA) $(srcdir)/doc/pdffonts.1 $(DESTDIR)@mandir@/man1/pdffonts.1
-@X@	$(INSTALL_DATA) $(srcdir)/doc/pdftoppm.1 $(DESTDIR)@mandir@/man1/pdftoppm.1
+	$(INSTALL_DATA) $(srcdir)/doc/pdftoppm.1 $(DESTDIR)@mandir@/man1/pdftoppm.1
 	$(INSTALL_DATA) $(srcdir)/doc/pdfimages.1 $(DESTDIR)@mandir@/man1/pdfimages.1
 	-mkdir -p $(DESTDIR)@mandir@/man5
 	$(INSTALL_DATA) $(srcdir)/doc/xpdfrc.5 $(DESTDIR)@mandir@/man5/xpdfrc.5
diff -urN xpdf-3.01/xpdf/Makefile.in xpdf-3.01-pdftoppm-no_X-patch/xpdf/Makefile.in
--- xpdf-3.01/xpdf/Makefile.in	2005-08-17 07:34:31.000000000 +0200
+++ xpdf-3.01-pdftoppm-no_X-patch/xpdf/Makefile.in	2005-09-12 23:27:53.903705325 +0200
@@ -109,7 +109,7 @@
 	pdffonts$(EXE) pdftoppm$(EXE) pdfimages$(EXE)
 
 all-no-x: pdftops$(EXE) pdftotext$(EXE) pdfinfo$(EXE) pdffonts$(EXE) \
-	pdfimages$(EXE)
+	pdftoppm$(EXE) pdfimages$(EXE)
 
 #------------------------------------------------------------------------
 
