--- configure.ac.old	2009-04-26 11:08:53.000000000 -0400
+++ configure.ac	2009-04-26 11:49:27.000000000 -0400
@@ -49,8 +49,8 @@
 		wall_werror=yes
 	else
 		wall_werror=no
-	fi
-]
+	fi  
+]    
 )
 
 CFLAGS="-DORTP_INET6 $CFLAGS "
@@ -152,6 +152,15 @@
 
 found_sound=no
 
+AC_ARG_ENABLE(oss,
+      [  --enable-oss    Disable oss support],
+      [case "${enableval}" in
+        yes) oss=true ;;
+        no)  oss=false ;;
+        *) AC_MSG_ERROR(bad value ${enableval} for --disable-oss) ;;
+      esac],[oss=true])
+
+if "$oss" = "true"; then
 AC_CHECK_HEADERS(soundcard.h sys/soundcard.h machine/soundcard.h sys/audio.h)
 if test "${ac_cv_header_sys_soundcard_h}" = "yes" || \
 	test "${ac_cv_header_soundcard_h}" = "yes" || \
@@ -159,6 +168,7 @@
 	test "${ac_cv_header_machine_soundcard_h}" = "yes"; then
 	found_sound=yes
 fi
+fi
 
 AM_CONDITIONAL(BUILD_OSS, test x$found_sound = xyes)
 
@@ -280,6 +290,16 @@
 
 AM_CONDITIONAL(BUILD_MACAQSND, test x$macaqsnd_enabled = xtrue)
 
+AC_ARG_ENABLE(jack,
+      [  --disable-jack    Disable jack support],
+      [case "${enableval}" in
+        yes) jack=true ;;
+        no)  jack=false ;;
+        *) AC_MSG_ERROR(bad value ${enableval} for --disable-jack) ;;
+      esac],[jack=true])
+
+if test x$jack = xtrue; then
+
 dnl Check for samplerate libraries
 dnl Check for jack libraries (sound output plugin)
 PKG_CHECK_MODULES(JACK,jack >= 0.15.0, 
@@ -306,6 +326,8 @@
 AC_SUBST(JACK_CFLAGS)
 AC_SUBST(JACK_LIBS) 
 
+fi
+
 if test "$found_sound" = "no"; then
 	AC_MSG_ERROR([Could not find a support sound driver API])
 fi
@@ -315,6 +337,16 @@
 dnl	check for various codecs libraries
 dnl	*************************************
 
+AC_ARG_ENABLE(speex,
+      [  --disable-speex    Disable speex support],
+      [case "${enableval}" in
+        yes) speex=true ;;
+        no)  speex=false ;;
+        *) AC_MSG_ERROR(bad value ${enableval} for --disable-speex) ;;
+      esac],[speex=true])
+
+if test x$speex = xtrue; then
+
 dnl check for installed version of speex
 PKG_CHECK_MODULES(SPEEX, speex >= 1.2beta3,
 	[ AC_DEFINE(HAVE_SPEEX_NOISE,1,[tells whether the noise arg of speex_echo_cancel can be used]) ],
@@ -329,11 +361,26 @@
     [AC_MSG_NOTICE([No speexdsp library found.])
 ]
 )
+else
+SPEEX_CFLAGS="-DDISABLE_SPEEX"
+fi
+
 AC_SUBST(SPEEX_CFLAGS)
 AC_SUBST(SPEEX_LIBS)
+
 AM_CONDITIONAL(BUILD_SPEEX, test x$build_speex = xyes )
 AM_CONDITIONAL(BUILD_RESAMPLE, test x$build_resample = xyes )
 
+AC_ARG_ENABLE(gsm,
+      [  --disable-gsm    Disable gsm support],
+      [case "${enableval}" in
+        yes) gsm=true ;;
+        no)  gsm=false ;;
+        *) AC_MSG_ERROR(bad value ${enableval} for --disable-gsm) ;;
+      esac],[gsm=true])
+
+if test x$gsm = xtrue; then
+
 dnl check for gsm
 build_gsm=no
 AC_ARG_WITH( gsm,
@@ -350,6 +397,9 @@
 else
 	build_gsm=no
 fi
+
+fi
+
 AM_CONDITIONAL(BUILD_GSM, test x$build_gsm = xyes )
 
 MS_CHECK_VIDEO
@@ -415,7 +465,17 @@
 dnl # Check for doxygen
 dnl ##################################################
 
+AC_ARG_ENABLE(doc,
+			[  --disable-doc    Disable documentation build/install],
+      [case "${enableval}" in
+			  yes) enable_doc=true ;;
+        no)  enable_doc=false ;;
+				*) AC_MSG_ERROR(bad value ${enableval} for --disable-doc) ;;
+      esac],[enable_doc=true])
+
+if test "$enable_doc" = "true"; then
 AC_PATH_PROG(DOXYGEN,doxygen,false)
+fi
 AM_CONDITIONAL(HAVE_DOXYGEN, test $DOXYGEN != false)
 
 
--- acinclude.m4.old	2009-04-26 11:28:59.000000000 -0400
+++ acinclude.m4	2009-04-26 11:13:06.000000000 -0400
@@ -108,9 +108,30 @@
 			fi
 		fi
 
+		AC_ARG_ENABLE(theora,
+		  [  --disable-theora    Disable theora support],
+		  [case "${enableval}" in
+			yes) theora=true ;;
+			no)  theora=false ;;
+			*) AC_MSG_ERROR(bad value ${enableval} for --disable-theora) ;;
+		  esac],[theora=true])
+
+		if test x$theora = xtrue; then
 		PKG_CHECK_MODULES(THEORA, [theora >= 1.0alpha7 ], [have_theora=yes],
 					[have_theora=no])
+		fi
+
+		AC_ARG_ENABLE(x11,
+		  [  --disable-x11    Disable X11 support],
+		  [case "${enableval}" in
+			yes) enable_x11=true ;;
+			no)  enable_x11=false ;;
+			*) AC_MSG_ERROR(bad value ${enableval} for --disable-x11) ;;
+		  esac],[enable_x11=true])
+
+		if test "$enable_x11" = "true"; then
 		AC_CHECK_HEADERS(X11/Xlib.h)
+		fi
 		
 		VIDEO_CFLAGS=" $FFMPEG_CFLAGS -DVIDEO_ENABLED"
 		VIDEO_LIBS=" $FFMPEG_LIBS $SWSCALE_LIBS"
