diff -ur ./admin/acinclude.m4.in ../kdebindings-3.5.8/admin/acinclude.m4.in
--- ./admin/acinclude.m4.in	2008-02-12 01:42:21.000000000 +0200
+++ ../kdebindings-3.5.8/admin/acinclude.m4.in	2008-02-17 21:01:17.000000000 +0200
@@ -5263,8 +5263,6 @@
   kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libgcj.so
   kde_java_libgcjdir=`find $kde_java_bindir/.. -name libgcj.so | sed 's,libgcj.so,,'|head -n 1`
-  dnl Look for libhpi.so and avoid green threads
-  kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | head -n 1`
 
   dnl Now check everything's fine under there
   dnl the include dir is our flag for having the JDK
@@ -5315,12 +5313,6 @@
 
   dnl not needed for gcj compile
 
-  if test "x$kde_java_libgcjdir" = "x"; then 
-      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
-      fi
-  fi
-
   if test -n "$jni_includes"; then
     dnl Check for JNI version
     AC_LANG_SAVE
