--- xfc-4.3.2/examples/howto/selection/selection.cc.orig	2008-09-22 00:26:26.000000000 +0200
+++ xfc-4.3.2/examples/howto/selection/selection.cc	2008-09-22 00:26:50.000000000 +0200
@@ -3,6 +3,7 @@
 #include <xfc/gtk/buttonbox.hh>
 #include <xfc/gtk/label.hh>
 #include <iostream>
+#include <stdint.h>
 
 std::vector<Gtk::TargetEntry> SupplySelectionButton::target_entries;
 
@@ -47,7 +48,7 @@
 	cout << " * type() = " << selection_data.get_type() << endl;
 	cout << " * format() = " << selection_data.format() << endl;
 	cout.setf(ios_base::hex, ios_base::basefield);
-	cout << " * data() = 0x" << reinterpret_cast<unsigned int>(selection_data.data()) << endl;
+	cout << " * data() = 0x" << reinterpret_cast<unsigned uintptr_t>(selection_data.data()) << endl;
 	cout.setf(ios_base::dec, ios_base::basefield);
 	cout << " * length() = " << selection_data.length() << endl << endl;
 	
