diff -cr --new-file interpreter.orig/Makefile.in interpreter/Makefile.in
*** interpreter.orig/Makefile.in	Tue Nov 25 18:49:30 2003
--- interpreter/Makefile.in	Tue Nov 25 18:49:44 2003
***************
*** 462,467 ****
--- 462,468 ----
  
  init.tcl: init.tcl.pl
  	./init.tcl.pl $(libdir)/tcl $(lardlibdir) > init.tcl
+ 	./con.py
  
  main.c: init.tcl
  # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -cr --new-file interpreter.orig/con.py interpreter/con.py
*** interpreter.orig/con.py	Thu Jan  1 01:00:00 1970
--- interpreter/con.py	Tue Nov 25 18:49:59 2003
***************
*** 0 ****
--- 1,16 ----
+ #!/usr/bin/python
+ 
+ f=open('init.tcl')
+ s=f.read()
+ f.close()
+ s=s[s.find('"'):s.rfind('"')+1]
+ newstr=''
+ for c in s:
+     if c=='\n':
+         newstr=newstr+'\\n'
+     else:
+         newstr=newstr+c
+ 
+ f=open('init.tcl','w+')
+ f.write(newstr)
+ f.close()
