diff -u lard-2.0.14.orig/configure lard-2.0.14/configure
--- lard-2.0.14.orig/configure	2001-11-20 05:29:29.000000000 -0800
+++ lard-2.0.14/configure	2002-10-29 22:59:01.000000000 -0800
@@ -2633,7 +2633,7 @@
 
 
 
-	lardbindir=${bindir}/${host_alias}
+	lardbindir=${bindir}/
 	
 	echo "$ac_t""lardbindir set to ${lardbindir}" 1>&6
 	
@@ -2648,17 +2648,17 @@
 	echo "$ac_t""lardlibdir set to ${lardlibdir}" 1>&6
 	
 
-	lardcommondir=${bindir}/common
+	lardcommondir=${lardbindir}
 	
 	echo "$ac_t""lardcommondir set to ${lardcommondir}" 1>&6
 	
 
-	lardelispdir=${libdir}/elisp
+	lardelispdir=/usr/share/emacs/site-lisp
 	
 	echo "$ac_t""lardelispdir set to ${lardelispdir}" 1>&6
 	
 
-docdir=${prefix}/doc
+docdir=${prefix}/share/doc/lard-2.0.14
 
 tcllibdir=${prefix}/lib/tcl
 
