--- lard-20030109.orig/configure	2003-01-09 14:09:29.000000000 -0700
+++ lard-20030109/configure	2003-01-09 14:13:44.000000000 -0700
@@ -5750,7 +5750,7 @@
 
 
 
-	lardbindir=${bindir}/${host}
+	lardbindir=${bindir}/
 
 	echo "$as_me:$LINENO: result: lardbindir set to ${lardbindir}" >&5
 echo "${ECHO_T}lardbindir set to ${lardbindir}" >&6
@@ -5767,19 +5767,19 @@
 echo "${ECHO_T}lardlibdir set to ${lardlibdir}" >&6
 
 
-	lardcommondir=${bindir}/common
+	lardcommondir=${lardbindir}
 
 	echo "$as_me:$LINENO: result: lardcommondir set to ${lardcommondir}" >&5
 echo "${ECHO_T}lardcommondir set to ${lardcommondir}" >&6
 
 
-	lardelispdir=${libdir}/elisp
+	lardelispdir=/usr/share/emacs/site-lisp
 
 	echo "$as_me:$LINENO: result: lardelispdir set to ${lardelispdir}" >&5
 echo "${ECHO_T}lardelispdir set to ${lardelispdir}" >&6
 
 
-docdir=${prefix}/doc
+docdir=${prefix}/share/doc/lard-2.0.14
 
 tcllibdir=${prefix}/lib/tcl
 
