diff -urN e2fsprogs-1.34.orig/config.rpath e2fsprogs-1.34/config.rpath
--- e2fsprogs-1.34.orig/config.rpath	2003-07-23 05:57:24.000000000 -0700
+++ e2fsprogs-1.34/config.rpath	2003-12-30 01:24:38.000000000 -0800
@@ -478,7 +478,7 @@
 escaped_sys_lib_search_path_spec=`echo "X$sys_lib_search_path_spec" | sed -e 's/^X//' -e "$sed_quote_subst"`
 escaped_sys_lib_dlsearch_path_spec=`echo "X$sys_lib_dlsearch_path_spec" | sed -e 's/^X//' -e "$sed_quote_subst"`
 
-sed -e 's/^\([a-zA-Z0-9_]*\)=/acl_cv_\1=/' <<EOF
+sed -e 's/^\([[:alnum:]_]*\)=/acl_cv_\1=/' <<EOF
 
 # How to pass a linker flag through the compiler.
 wl="$escaped_wl"
diff -urN e2fsprogs-1.34.orig/configure e2fsprogs-1.34/configure
--- e2fsprogs-1.34.orig/configure	2003-07-23 06:07:31.000000000 -0700
+++ e2fsprogs-1.34/configure	2003-12-30 01:21:32.000000000 -0800
@@ -133,7 +133,7 @@
   fi
 
   case "$ac_option" in
-  -*=*) ac_optarg=`echo "$ac_option" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
+  -*=*) ac_optarg=`echo "$ac_option" | sed 's/[-_[:alnum:]]*=//'` ;;
   *) ac_optarg= ;;
   esac
 
@@ -167,7 +167,7 @@
   -disable-* | --disable-*)
     ac_feature=`echo $ac_option|sed -e 's/-*disable-//'`
     # Reject names that are not valid shell variable names.
-    if test -n "`echo $ac_feature| sed 's/[-a-zA-Z0-9_]//g'`"; then
+    if test -n "`echo $ac_feature| sed 's/[-[:alnum:]_]//g'`"; then
       { echo "configure: error: $ac_feature: invalid feature name" 1>&2; exit 1; }
     fi
     ac_feature=`echo $ac_feature| sed 's/-/_/g'`
@@ -176,7 +176,7 @@
   -enable-* | --enable-*)
     ac_feature=`echo $ac_option|sed -e 's/-*enable-//' -e 's/=.*//'`
     # Reject names that are not valid shell variable names.
-    if test -n "`echo $ac_feature| sed 's/[-_a-zA-Z0-9]//g'`"; then
+    if test -n "`echo $ac_feature| sed 's/[-_[:alnum:]]//g'`"; then
       { echo "configure: error: $ac_feature: invalid feature name" 1>&2; exit 1; }
     fi
     ac_feature=`echo $ac_feature| sed 's/-/_/g'`
@@ -407,7 +407,7 @@
   -with-* | --with-*)
     ac_package=`echo $ac_option|sed -e 's/-*with-//' -e 's/=.*//'`
     # Reject names that are not valid shell variable names.
-    if test -n "`echo $ac_package| sed 's/[-_a-zA-Z0-9]//g'`"; then
+    if test -n "`echo $ac_package| sed 's/[-_[:alnum:]]//g'`"; then
       { echo "configure: error: $ac_package: invalid package name" 1>&2; exit 1; }
     fi
     ac_package=`echo $ac_package| sed 's/-/_/g'`
@@ -420,7 +420,7 @@
   -without-* | --without-*)
     ac_package=`echo $ac_option|sed -e 's/-*without-//'`
     # Reject names that are not valid shell variable names.
-    if test -n "`echo $ac_package| sed 's/[-a-zA-Z0-9_]//g'`"; then
+    if test -n "`echo $ac_package| sed 's/[-[:alnum:]_]//g'`"; then
       { echo "configure: error: $ac_package: invalid package name" 1>&2; exit 1; }
     fi
     ac_package=`echo $ac_package| sed 's/-/_/g'`
@@ -448,7 +448,7 @@
     ;;
 
   *)
-    if test -n "`echo $ac_option| sed 's/[-a-z0-9.]//g'`"; then
+    if test -n "`echo $ac_option| sed 's/[-[:lower:][:digit:].]//g'`"; then
       echo "configure: warning: $ac_option: invalid host type" 1>&2
     fi
     if test "x$nonopt" != xNONE; then
@@ -610,7 +610,7 @@
   esac
   case $ac_prog in
     # Accept absolute paths.
-    [\\/]* | [A-Za-z]:[\\/]*)
+    [\\/]* | [[:alpha:]]:[\\/]*)
       re_direlt='/[^/][^/]*/\.\./'
       # Canonicalize the path of ld
       ac_prog=`echo $ac_prog| sed 's%\\\\%/%g'`
@@ -2198,7 +2198,7 @@
 #endif
 EOF
 if (eval "$ac_cpp conftest.$ac_ext") 2>&5 |
-  egrep "(^|[^a-zA-Z_0-9])off_t[^a-zA-Z_0-9]" >/dev/null 2>&1; then
+  egrep "(^|[^[:alpha:]_[:digit:]])off_t[^[:alpha:]_[:digit:]]" >/dev/null 2>&1; then
   rm -rf conftest*
   ac_cv_type_off_t=yes
 else
@@ -2231,7 +2231,7 @@
 #endif
 EOF
 if (eval "$ac_cpp conftest.$ac_ext") 2>&5 |
-  egrep "(^|[^a-zA-Z_0-9])size_t[^a-zA-Z_0-9]" >/dev/null 2>&1; then
+  egrep "(^|[^[:alpha:]_[:digit:]])size_t[^[:alpha:]_[:digit:]]" >/dev/null 2>&1; then
   rm -rf conftest*
   ac_cv_type_size_t=yes
 else
@@ -6204,11 +6204,11 @@
     # turns \\\\ into \\, and sed turns \\ into \).
     sed -n \
       -e "s/'/'\\\\''/g" \
-      -e "s/^\\([a-zA-Z0-9_]*_cv_[a-zA-Z0-9_]*\\)=\\(.*\\)/\\1=\${\\1='\\2'}/p"
+      -e "s/^\\([[:alnum:]_]*_cv_[[:alnum:]_]*\\)=\\(.*\\)/\\1=\${\\1='\\2'}/p"
     ;;
   *)
     # `set' quotes correctly as required by POSIX, so do not add quotes.
-    sed -n -e 's/^\([a-zA-Z0-9_]*_cv_[a-zA-Z0-9_]*\)=\(.*\)/\1=${\1=\2}/p'
+    sed -n -e 's/^\([[:alnum:]_]*_cv_[[:alnum:]_]*\)=\(.*\)/\1=${\1=\2}/p'
     ;;
   esac >> confcache
 if cmp -s $cache_file confcache; then
@@ -6242,7 +6242,7 @@
 # Protect against shell expansion while executing Makefile rules.
 # Protect against Makefile macro expansion.
 cat > conftest.defs <<\EOF
-s%#define \([A-Za-z_][A-Za-z0-9_]*\) *\(.*\)%-D\1=\2%g
+s%#define \([[:alpha:]_][[:alnum:]_]*\) *\(.*\)%-D\1=\2%g
 s%[ 	`~#$^&*(){}\\|;'"<>?]%\\&%g
 s%\[%\\&%g
 s%\]%\\&%g
diff -urN e2fsprogs-1.34.orig/lib/ss/ct_c.sed e2fsprogs-1.34/lib/ss/ct_c.sed
--- e2fsprogs-1.34.orig/lib/ss/ct_c.sed	2002-02-02 22:03:43.000000000 -0800
+++ e2fsprogs-1.34/lib/ss/ct_c.sed	2003-12-30 01:23:55.000000000 -0800
@@ -44,7 +44,7 @@
 # Now we take care of some syntatic sugar.....
 #
 /^unimplemented/ {
-	s/^unimplemented [A-Za-z_0-9]*/request ss_unimplemented/
+	s/^unimplemented [[:alpha:]_[:digit:]]*/request ss_unimplemented/
 	s/;/, (dont_list, dont_summarize);/
 }
 /^unknown/ {
@@ -93,9 +93,9 @@
 	x
 	b EMITHLP
 }
-s/[^a-zA-Z0-9].*//
+s/[^[:alnum:]].*//
 x
-s/[a-zA-Z0-9]*, *//
+s/[[:alnum:]]*, *//
 x
 :EMITHLP
 s/^/hlp: /
@@ -114,9 +114,9 @@
 	x
 	b EMITREQ
 }
-s/[^A-Za-z_0-9].*//
+s/[^[:alpha:]_[:digit:]].*//
 x
-s/[A-Za-z_0-9]*//
+s/[[:alpha:]_[:digit:]]*//
 s/, *//
 x
 :EMITREQ
@@ -132,14 +132,14 @@
 h
 : OPTLIST
 /^)/ b EOR
-/^[^A-Za-z_0-9]/ {
+/^[^[:alpha:]_[:digit:]]/ {
 	=
 	c\
 ERROR: parse error in options list
 }
-s/[^A-Za-z_0-9].*//
+s/[^[:alpha:]_[:digit:]].*//
 x
-s/[A-Za-z_0-9]*//
+s/[[:alpha:]_[:digit:]]*//
 s/, *//
 x
 s/^/opt: /
