--- plugins/Makefile.in.orig	2004-10-13 22:35:19.000000000 +0200
+++ plugins/Makefile.in	2004-10-13 22:36:15.000000000 +0200
@@ -371,7 +371,7 @@
 	  dir=`echo "$$file" | sed -e 's,/[^/]*$$,,'`; \
 	  if test "$$dir" != "$$file" && test "$$dir" != "."; then \
 	    dir="/$$dir"; \
-	    $(mkdir_p) "$(distdir)$$dir"; \
+	    $(mkdir_p) "$(DESTDIR)/$(distdir)$$dir"; \
 	  else \
 	    dir=''; \
 	  fi; \
--- plugins/ofx/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/ofx/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -220,7 +220,7 @@
 target_alias = @target_alias@
 @OFX_TRUE@noinst_HEADERS = ofxplugin.h
 @OFX_TRUE@lib_LTLIBRARIES = libofxio.la
-@OFX_TRUE@pidir = $(prefix)/plugins
+@OFX_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @OFX_TRUE@dbdir = $(pidir)/db
 @OFX_TRUE@importdir = $(pidir)/import
 @OFX_TRUE@exportdir = $(pidir)/export
--- plugins/qif/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/qif/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -219,7 +219,7 @@
 target_alias = @target_alias@
 @QIF_TRUE@noinst_HEADERS = qifplugin.h
 @QIF_TRUE@lib_LTLIBRARIES = libqifio.la
-@QIF_TRUE@pidir = $(prefix)/plugins
+@QIF_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @QIF_TRUE@dbdir = $(pidir)/db
 @QIF_TRUE@importdir = $(pidir)/import
 @QIF_TRUE@exportdir = $(pidir)/export
--- plugins/xml/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/xml/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -227,7 +227,7 @@
 target_alias = @target_alias@
 @XML_TRUE@noinst_HEADERS = xmlplugin.h
 @XML_TRUE@lib_LTLIBRARIES = libxmlio.la
-@XML_TRUE@pidir = $(prefix)/plugins
+@XML_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @XML_TRUE@dbdir = $(pidir)/db
 @XML_TRUE@importdir = $(pidir)/import
 @XML_TRUE@exportdir = $(pidir)/export
--- plugins/psql/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/psql/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -227,7 +227,7 @@
 target_alias = @target_alias@
 @PSQL_TRUE@noinst_HEADERS = psqlplugin.h
 @PSQL_TRUE@lib_LTLIBRARIES = libpsqlio.la
-@PSQL_TRUE@pidir = $(prefix)/plugins
+@PSQL_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @PSQL_TRUE@dbdir = $(pidir)/db
 @PSQL_TRUE@importdir = $(pidir)/import
 @PSQL_TRUE@exportdir = $(pidir)/export
--- plugins/reports/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/reports/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -300,7 +300,7 @@
 @PGPH_TRUE@									libaves.la libbudget.la libmbudget.la libjournal.la \
 @PGPH_TRUE@									libtrans.la libacct.la libtbal.la
 
-@PGPH_TRUE@pidir = $(prefix)/plugins
+@PGPH_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @PGPH_TRUE@reportdir = $(pidir)/report
 @PGPH_TRUE@noinst_LIBRARIES = libtemp.a
 @PGPH_TRUE@libtemp_a_SOURCES = reporter.cpp
--- plugins/mysql/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/mysql/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -227,7 +227,7 @@
 target_alias = @target_alias@
 @MYSQL_TRUE@noinst_HEADERS = mysqlplugin.h
 @MYSQL_TRUE@lib_LTLIBRARIES = libmysqlio.la
-@MYSQL_TRUE@pidir = $(prefix)/plugins
+@MYSQL_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @MYSQL_TRUE@dbdir = $(pidir)/db
 @MYSQL_TRUE@importdir = $(pidir)/import
 @MYSQL_TRUE@exportdir = $(pidir)/export
--- plugins/extprg/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/extprg/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -227,7 +227,7 @@
 target_alias = @target_alias@
 @PRG_TRUE@noinst_HEADERS = extprgplugin.h
 @PRG_TRUE@lib_LTLIBRARIES = libextprgio.la
-@PRG_TRUE@pidir = $(prefix)/plugins
+@PRG_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @PRG_TRUE@dbdir = $(pidir)/db
 @PRG_TRUE@importdir = $(pidir)/import
 @PRG_TRUE@exportdir = $(pidir)/export
--- plugins/graphs/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/graphs/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -267,7 +267,7 @@
 @PGPH_TRUE@									libdoublelinegraph.la libsinglebargraph.la \
 @PGPH_TRUE@									libdoublebargraph.la libpiegraph.la
 
-@PGPH_TRUE@pidir = $(prefix)/plugins
+@PGPH_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @PGPH_TRUE@graphdir = $(pidir)/graph
 @PGPH_TRUE@libgraphbase_la_SOURCES = grapher.cpp
 @PGPH_TRUE@libmonthlygraph_la_SOURCES = monthlygraph.cpp
--- plugins/archiver/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/archiver/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -219,7 +219,7 @@
 target_alias = @target_alias@
 @ARC_TRUE@noinst_HEADERS = arcplugin.h
 @ARC_TRUE@lib_LTLIBRARIES = libarcio.la
-@ARC_TRUE@pidir = $(prefix)/plugins
+@ARC_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @ARC_TRUE@importdir = $(pidir)/import
 @ARC_TRUE@exportdir = $(pidir)/export
 @ARC_TRUE@libarcio_la_SOURCES = arcplugin.cpp
--- plugins/upgrade/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/upgrade/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -227,7 +227,7 @@
 target_alias = @target_alias@
 @UPG_TRUE@noinst_HEADERS = upgplugin.h
 @UPG_TRUE@lib_LTLIBRARIES = libupgradeio.la
-@UPG_TRUE@pidir = $(prefix)/plugins
+@UPG_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @UPG_TRUE@dbdir = $(pidir)/db
 @UPG_TRUE@libupgradeio_la_SOURCES = upgplugin.cpp upgplugin.h
 @UPG_TRUE@RELEASE = 2:0:0
--- plugins/sqlite/Makefile.in.orig	2004-10-14 01:28:15.000000000 +0200
+++ plugins/sqlite/Makefile.in	2004-10-14 01:30:07.000000000 +0200
@@ -234,7 +234,7 @@
 target_alias = @target_alias@
 @SQLL_TRUE@noinst_HEADERS = sqliteplugin.h lfsqliteplugin.h
 @SQLL_TRUE@lib_LTLIBRARIES = libsqliteio.la liblfsqliteio.la
-@SQLL_TRUE@pidir = $(prefix)/plugins
+@SQLL_TRUE@pidir = $(DESTDIR)/$(prefix)/plugins
 @SQLL_TRUE@dbdir = $(pidir)/db
 @SQLL_TRUE@importdir = $(pidir)/import
 @SQLL_TRUE@exportdir = $(pidir)/export
--- plugins/cliimp/Makefile.in.orig	2004-10-14 01:48:52.000000000 +0200
+++ plugins/cliimp/Makefile.in	2004-10-14 01:52:29.000000000 +0200
@@ -520,8 +520,8 @@
 @CLIIMP_TRUE@	rm -f *.o *.core core *~ .*~ *.la *.lo
 
 @CLIIMP_TRUE@install-data-local:
-@CLIIMP_TRUE@	mkdir -p $(impdir)
-@CLIIMP_TRUE@	if test ! -L $(impdir)/libcliimpio.so ; then cd $(impdir) && ln -s $(libdir)/libcliimpio.so $(impdir) ; fi
+@CLIIMP_TRUE@	mkdir -p $(DESTDIR)/$(impdir)
+@CLIIMP_TRUE@	if test ! -L $(DESTDIR)/$(impdir)/libcliimpio.so ; then cd $(DESTDIR)/$(impdir) && ln -s $(DESTDIR)/$(libdir)/libcliimpio.so $(DESTDIR)/$(impdir) ; fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
--- plugins/ofx/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/ofx/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -267,7 +267,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
--- plugins/qif/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/qif/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -266,7 +266,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
--- plugins/xml/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/xml/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -274,7 +274,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
--- plugins/psql/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/psql/Makefile.in	2004-10-20 19:13:54.000000000 +0200
@@ -274,7 +274,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
@@ -524,9 +524,9 @@
 @PSQL_TRUE@	mkdir -p $(dbdir)
 @PSQL_TRUE@	if test ! -L $(dbdir)/libpsqlio.so ; then cd $(dbdir); ln -s $(libdir)/libpsqlio.so $(dbdir) ; fi
 @PSQL_TRUE@	mkdir -p $(importdir)
-@PSQL_TRUE@	if test ! -L $(importdir)/libpsqlio.so ; then cd $(importdir); ln -s ../../lib/libpsqlio.so $(importdir) ; fi
+@PSQL_TRUE@	if test ! -L $(importdir)/libpsqlio.so ; then cd $(importdir); ln -s $(libdir)/libpsqlio.so $(importdir) ; fi
 @PSQL_TRUE@	mkdir -p $(exportdir)
-@PSQL_TRUE@	if test ! -L $(exportdir)/libpsqlio.so ; then cd $(exportdir); ln -s ../../lib/libpsqlio.so $(exportdir) ; fi
+@PSQL_TRUE@	if test ! -L $(exportdir)/libpsqlio.so ; then cd $(exportdir); ln -s $(libdir)/libpsqlio.so $(exportdir) ; fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
--- plugins/reports/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/reports/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -385,7 +385,7 @@
 	$(RANLIB) libtemp.a
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
+++ plugins/mysql/Makefile.in	2004-10-20 19:13:32.000000000 +0200
--- plugins/mysql/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
@@ -274,7 +274,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
@@ -524,9 +524,9 @@
 @MYSQL_TRUE@	mkdir -p $(dbdir)
 @MYSQL_TRUE@	if test ! -L $(dbdir)/libmysqlio.so ; then cd $(dbdir); ln -s $(libdir)/libmysqlio.so $(dbdir) ; fi
 @MYSQL_TRUE@	mkdir -p $(importdir)
-@MYSQL_TRUE@	if test ! -L $(importdir)/libmysqlio.so ; then cd $(importdir); ln -s ../../lib/libmysqlio.so $(importdir) ; fi
+@MYSQL_TRUE@	if test ! -L $(importdir)/libmysqlio.so ; then cd $(importdir); ln -s $(libdir)/libmysqlio.so $(importdir) ; fi
 @MYSQL_TRUE@	mkdir -p $(exportdir)
-@MYSQL_TRUE@	if test ! -L $(exportdir)/libmysqlio.so ; then cd $(exportdir); ln -s ../../lib/libmysqlio.so $(exportdir) ; fi
+@MYSQL_TRUE@	if test ! -L $(exportdir)/libmysqlio.so ; then cd $(exportdir); ln -s $(libdir)/libmysqlio.so $(exportdir) ; fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
--- plugins/cliimp/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/cliimp/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -265,7 +265,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
--- plugins/extprg/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/extprg/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -274,7 +274,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
--- plugins/graphs/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/graphs/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -331,7 +331,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
--- plugins/generator/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/generator/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -271,7 +271,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
--- plugins/archiver/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/archiver/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -265,7 +265,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
--- plugins/upgrade/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/upgrade/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -272,7 +272,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
--- plugins/sqlite/Makefile.in.orig	2004-10-20 19:11:02.000000000 +0200
+++ plugins/sqlite/Makefile.in	2004-10-20 19:12:54.000000000 +0200
@@ -284,7 +284,7 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
+	test -z "$(DESTDIR)/$(libdir)" || $(mkdir_p) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
