diff -Nru gateway-1.4.0.orig/checks/check_list.c gateway-1.4.0/checks/check_list.c
--- gateway-1.4.0.orig/checks/check_list.c	2004-01-22 16:08:24.000000000 +0200
+++ gateway-1.4.0/checks/check_list.c	2005-12-06 23:52:08.521615000 +0200
@@ -155,7 +155,7 @@
 
 	while (list_len(list) > 0) {
 		item = list_get(list, 0);
-		list_delete(list, 0, 1);
+		list_delete_at(list, 0, 1);
 		warning(0, "main: %ld %ld %ld", (long) item->producer, 
 				item->num, item->index);
 	}
diff -Nru gateway-1.4.0.orig/gw/bb_smscconn.c gateway-1.4.0/gw/bb_smscconn.c
--- gateway-1.4.0.orig/gw/bb_smscconn.c	2004-11-18 18:31:32.000000000 +0200
+++ gateway-1.4.0/gw/bb_smscconn.c	2005-12-06 23:51:42.752004500 +0200
@@ -631,7 +631,7 @@
         }
         
         /* drop old connection from the active smsc list */
-        list_delete(smsc_list, i, 1);
+        list_delete_at(smsc_list, i, 1);
         /* destroy the connection */
         smscconn_destroy(conn);
         list_insert(smsc_list, i, new_conn);
diff -Nru gateway-1.4.0.orig/gw/dlr_mem.c gateway-1.4.0/gw/dlr_mem.c
--- gateway-1.4.0.orig/gw/dlr_mem.c	2004-08-08 22:50:18.000000000 +0300
+++ gateway-1.4.0/gw/dlr_mem.c	2005-12-06 23:51:54.400732500 +0200
@@ -102,7 +102,7 @@
     gw_rwlock_wrlock(&rwlock);
     len = list_len(dlr_waiting_list);
     for (i=0; i < len; i++)
-        list_delete(dlr_waiting_list, i, 1);
+        list_delete_at(dlr_waiting_list, i, 1);
     gw_rwlock_unlock(&rwlock);
 }
 
@@ -171,7 +171,7 @@
         dlr = list_get(dlr_waiting_list, i);
 
         if (dlr_mem_entry_match(dlr, smsc, ts, dst) == 0) {
-            list_delete(dlr_waiting_list, i, 1);
+            list_delete_at(dlr_waiting_list, i, 1);
             dlr_entry_destroy(dlr);
             break;
         }
diff -Nru gateway-1.4.0.orig/gw/smsc/smsc_soap.c gateway-1.4.0/gw/smsc/smsc_soap.c
--- gateway-1.4.0.orig/gw/smsc/smsc_soap.c	2004-08-28 21:09:19.000000000 +0300
+++ gateway-1.4.0/gw/smsc/smsc_soap.c	2005-12-06 23:51:59.533053250 +0200
@@ -1955,7 +1955,7 @@
             debug("bb.soap.init_query",0,"SOAP[%s]: init_query getting a client",octstr_get_cstr(privdata->name));
 
             /* client is not busy - get it */
-            list_delete(privdata->soap_client, index, 1);
+            list_delete_at(privdata->soap_client, index, 1);
             break;
         }
         cur_client = NULL;
diff -Nru gateway-1.4.0.orig/gwlib/dbpool.c gateway-1.4.0/gwlib/dbpool.c
--- gateway-1.4.0.orig/gwlib/dbpool.c	2004-06-08 13:25:16.000000000 +0300
+++ gateway-1.4.0/gwlib/dbpool.c	2005-12-06 23:46:43.589308000 +0200
@@ -311,7 +311,7 @@
         pconn = list_get(p->pool, i);
         if (p->db_ops->check(pconn->conn) != 0) {
             /* something was wrong, reinitialize the connection */
-            list_delete(p->pool, i, 1);
+            list_delete_at(p->pool, i, 1);
             dbpool_conn_destroy(pconn);
             p->curr_size--;
             reinit++;
diff -Nru gateway-1.4.0.orig/gwlib/http.c gateway-1.4.0/gwlib/http.c
--- gateway-1.4.0.orig/gwlib/http.c	2004-08-11 19:41:29.000000000 +0300
+++ gateway-1.4.0/gwlib/http.c	2005-12-06 23:47:02.922516250 +0200
@@ -2707,7 +2707,7 @@
 		    /* Delete old header */
 		    header = list_get(headers, i);
 		    octstr_destroy(header);
-                    list_delete(headers, i, 1);
+                    list_delete_at(headers, i, 1);
 
 		    /* Adds comma and new value to old header value */
                     octstr_append(value, octstr_imm(", "));
@@ -2722,7 +2722,7 @@
 		    /* Delete this header */
 		    header = list_get(headers, j);
 		    octstr_destroy(header);
-                    list_delete(headers, j, 1);
+                    list_delete_at(headers, j, 1);
                     j--;
                 }
             }
@@ -2833,7 +2833,7 @@
     while (i < list_len(headers)) {
 	h = list_get(headers, i);
 	if (header_is_called(h, name)) {
-	    list_delete(headers, i, 1);
+	    list_delete_at(headers, i, 1);
 	    octstr_destroy(h);
 	    count++;
 	} else
@@ -3087,7 +3087,7 @@
         if (octstr_get_char(element, pos) == '=') {
             octstr_append_char(auth_scheme, ';');
             octstr_append(auth_scheme, element);
-            list_delete(result, i, 1);
+            list_delete_at(result, i, 1);
             octstr_destroy(element);
         } else {
             unsigned char semicolon = ';';
diff -Nru gateway-1.4.0.orig/gwlib/list.c gateway-1.4.0/gwlib/list.c
--- gateway-1.4.0.orig/gwlib/list.c	2004-08-11 19:41:29.000000000 +0300
+++ gateway-1.4.0/gwlib/list.c	2005-12-06 23:47:16.827385250 +0200
@@ -226,7 +226,7 @@
 }
 
 
-void list_delete(List *list, long pos, long count)
+void list_delete_at(List *list, long pos, long count)
 {
     lock(list);
     delete_items_from_list(list, pos, count);
diff -Nru gateway-1.4.0.orig/gwlib/list.h gateway-1.4.0/gwlib/list.h
--- gateway-1.4.0.orig/gwlib/list.h	2004-06-28 12:53:38.000000000 +0300
+++ gateway-1.4.0/gwlib/list.h	2005-12-06 23:45:40.361356500 +0200
@@ -176,7 +176,7 @@
  * Delete items from the list. Note that this does _not_ free the memory
  * for the items, they are just dropped from the list.
  */
-void list_delete(List *list, long pos, long count);
+void list_delete_at(List *list, long pos, long count);
 
 
 /*
diff -Nru gateway-1.4.0.orig/wap/cookies.c gateway-1.4.0/wap/cookies.c
--- gateway-1.4.0.orig/wap/cookies.c	2004-03-11 20:02:39.000000000 +0200
+++ gateway-1.4.0/wap/cookies.c	2005-12-06 23:46:05.638936250 +0200
@@ -393,7 +393,7 @@
 			
             /* We have a match according to 4.3.3 - discard the old one */
             cookie_destroy(value);
-            list_delete(cookies, pos, 1);
+            list_delete_at(cookies, pos, 1);
 
             /* Discard the new cookie also if max-age is 0 - set if expiry date is up */
             if (cookie->max_age == 0) {
@@ -445,7 +445,7 @@
 					debug("wap.wsp.http", 0, "expire_cookies: Expired cookie (%s)",
 						  octstr_get_cstr(value->name));
 					cookie_destroy(value);
-					list_delete(cookies, pos, 1);
+					list_delete_at(cookies, pos, 1);
 				}
 			}
 		}
diff -Nru gateway-1.4.0.orig/wap/wsp_headers.c gateway-1.4.0/wap/wsp_headers.c
--- gateway-1.4.0.orig/wap/wsp_headers.c	2004-08-08 23:39:56.000000000 +0300
+++ gateway-1.4.0/wap/wsp_headers.c	2005-12-06 23:46:13.303415250 +0200
@@ -2041,7 +2041,7 @@
             Parameter *parm = list_get(parms, i);
             if (octstr_case_compare(realm, parm->key) == 0) {
                 realmparm = parm;
-                list_delete(parms, i, 1);
+                list_delete_at(parms, i, 1);
                 break;
             }
         }
diff -Nru gateway-1.4.0.orig/wap/wsp_session.c gateway-1.4.0/wap/wsp_session.c
--- gateway-1.4.0.orig/wap/wsp_session.c	2004-01-22 16:08:25.000000000 +0200
+++ gateway-1.4.0/wap/wsp_session.c	2005-12-06 23:46:30.436486000 +0200
@@ -816,7 +816,7 @@
 	bad_cap:
 		error(0, "WSP: Found illegal value in capabilities reply.");
 		wsp_cap_dump(cap);
-		list_delete(caps, i, 1);
+		list_delete_at(caps, i, 1);
 		i--;
 		wsp_cap_destroy(cap);
 		continue;
@@ -964,7 +964,7 @@
 
 		count = wsp_cap_count(req, cap->id, cap->name);
 		if (count == 0 && is_default_cap(cap)) {
-			list_delete(caps, i, 1);
+			list_delete_at(caps, i, 1);
 			wsp_cap_destroy(cap);
 		} else {
 			i++;
