diff -NurB gedit-2.4.0-orig/gedit/gedit2.c gedit-2.4.0/gedit/gedit2.c
--- gedit-2.4.0-orig/gedit/gedit2.c	2003-08-22 05:44:01.000000000 -0500
+++ gedit-2.4.0/gedit/gedit2.c	2003-09-14 10:26:06.000000000 -0500
@@ -32,6 +32,10 @@
 #include <config.h>
 #endif
 
+#ifdef HAVE_LOCALE_H
+#include <locale.h>
+#endif
+
 #include <string.h>
 
 #include <libgnome/libgnome.h>
@@ -347,11 +351,13 @@
 int
 main (int argc, char **argv)
 {
-    	GnomeProgram *program;
+	GnomeProgram *program;
 	gboolean restored = FALSE;
 	CORBA_Object factory;
 
+#ifdef HAVE_LOCALE_H
 	setlocale (LC_ALL, "");
+#endif
 
 	bindtextdomain (GETTEXT_PACKAGE, GEDIT_LOCALEDIR);
 	textdomain (GETTEXT_PACKAGE);
