--- warsow-0.32/warsow_0.32/source/qcommon/files.c	2007-08-07 17:47:31.000000000 +0200
+++ warsow-0.32.fixed/warsow_0.32/source/qcommon/files.c	2007-09-27 04:42:09.000000000 +0200
@@ -2277,7 +2277,7 @@
 	if( !fullname )
 		return qfalse;
 
-	pakfile = FS_LoadPK3File( filename, qtrue );
+	pakfile = FS_LoadPK3File( fullname, qtrue );
 	if( pakfile ) {		// unlock and free, we don't need this file anyway
 		if( pakfile->sysHandle )
 			Sys_FS_UnlockFile( pakfile->sysHandle );
