diff -Naurp vte-0.12.2.orig/src/iso2022.c vte-0.12.2/src/iso2022.c
--- vte-0.12.2.orig/src/iso2022.c	2006-03-15 11:02:59.000000000 +0000
+++ vte-0.12.2/src/iso2022.c	2006-06-07 15:53:48.000000000 +0000
@@ -35,7 +35,13 @@
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif
+
+#ifdef ENABLE_NLS
 #include <glib/gi18n-lib.h>
+#else
+#define _(String) String
+#define bindtextdomain(package,dir)
+#endif
 
 /* Maps which jive with XTerm's ESC ()*+ ? sequences, RFC 1468.  Add the
  * PC437 map because despite knowing that XTerm doesn't support it, certain
diff -Naurp vte-0.12.2.orig/src/reaper.c vte-0.12.2/src/reaper.c
--- vte-0.12.2.orig/src/reaper.c	2006-06-07 15:47:11.000000000 +0000
+++ vte-0.12.2/src/reaper.c	2006-06-07 15:52:35.000000000 +0000
@@ -34,7 +34,13 @@
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif
+
+#ifdef ENABLE_NLS
 #include <glib/gi18n-lib.h>
+#else
+#define _(String) String
+#define bindtextdomain(package,dir)
+#endif
 
 static VteReaper *singleton_reaper = NULL;
 struct reaper_info {
diff -Naurp vte-0.12.2.orig/src/vte.c vte-0.12.2/src/vte.c
--- vte-0.12.2.orig/src/vte.c	2006-06-07 15:47:11.000000000 +0000
+++ vte-0.12.2/src/vte.c	2006-06-07 15:52:35.000000000 +0000
@@ -47,7 +47,13 @@
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif
+
+#ifdef ENABLE_NLS
 #include <glib/gi18n-lib.h>
+#else
+#define _(String) String
+#define bindtextdomain(package,dir)
+#endif
 
 #ifndef HAVE_WINT_T
 typedef gunichar wint_t;
@@ -9557,7 +9563,7 @@ vte_terminal_class_init(VteTerminalClass
 #endif
 
 	bindtextdomain(GETTEXT_PACKAGE, LOCALEDIR);
-#ifdef HAVE_DECL_BIND_TEXTDOMAIN_CODESET
+#if HAVE_DECL_BIND_TEXTDOMAIN_CODESET
 	bind_textdomain_codeset(PACKAGE, "UTF-8");
 #endif
 
diff -Naurp vte-0.12.2.orig/src/vteaccess.c vte-0.12.2/src/vteaccess.c
--- vte-0.12.2.orig/src/vteaccess.c	2006-03-15 11:02:59.000000000 +0000
+++ vte-0.12.2/src/vteaccess.c	2006-06-07 15:55:28.000000000 +0000
@@ -34,7 +34,13 @@
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif
+
+#ifdef ENABLE_NLS
 #include <glib/gi18n-lib.h>
+#else
+#define _(String) String
+#define bindtextdomain(package,dir)
+#endif
 
 #define VTE_TERMINAL_ACCESSIBLE_PRIVATE_DATA "VteTerminalAccessiblePrivateData"
 
diff -Naurp vte-0.12.2.orig/src/vteapp.c vte-0.12.2/src/vteapp.c
--- vte-0.12.2.orig/src/vteapp.c	2006-03-15 11:02:59.000000000 +0000
+++ vte-0.12.2/src/vteapp.c	2006-06-07 15:54:19.000000000 +0000
@@ -34,7 +34,12 @@
 #include "debug.h"
 #include "vte.h"
 
+#ifdef ENABLE_NLS
 #include <glib/gi18n-lib.h>
+#else
+#define _(String) String
+#define bindtextdomain(package,dir)
+#endif
 
 #define DINGUS1 "(((news|telnet|nttp|file|http|ftp|https)://)|(www|ftp)[-A-Za-z0-9]*\\.)[-A-Za-z0-9\\.]+(:[0-9]*)?"
 #define DINGUS2 "(((news|telnet|nttp|file|http|ftp|https)://)|(www|ftp)[-A-Za-z0-9]*\\.)[-A-Za-z0-9\\.]+(:[0-9]*)?/[-A-Za-z0-9_\\$\\.\\+\\!\\*\\(\\),;:@&=\\?/~\\#\\%]*[^]'\\.}>\\) ,\\\"]"
diff -Naurp vte-0.12.2.orig/src/vtebg.c vte-0.12.2/src/vtebg.c
--- vte-0.12.2.orig/src/vtebg.c	2006-06-07 15:47:11.000000000 +0000
+++ vte-0.12.2/src/vtebg.c	2006-06-07 15:52:35.000000000 +0000
@@ -25,7 +25,12 @@
 #include "marshal.h"
 #include "vtebg.h"
 
+#ifdef ENABLE_NLS
 #include <glib/gi18n-lib.h>
+#else
+#define _(String) String
+#define bindtextdomain(package,dir)
+#endif
 
 struct VteBgPrivate {
 	GList *cache;
diff -Naurp vte-0.12.2.orig/src/vtexft.c vte-0.12.2/src/vtexft.c
--- vte-0.12.2.orig/src/vtexft.c	2006-03-15 11:02:59.000000000 +0000
+++ vte-0.12.2/src/vtexft.c	2006-06-07 15:54:58.000000000 +0000
@@ -37,7 +37,12 @@
 #include "vtexft.h"
 #include "vtetree.h"
 
+#ifdef ENABLE_NLS
 #include <glib/gi18n-lib.h>
+#else
+#define _(String) String
+#define bindtextdomain(package,dir)
+#endif
 
 #define FONT_INDEX_FUDGE 10
 #define CHAR_WIDTH_FUDGE 10
