diff -ur a/fsck/xchkdsk.c b/fsck/xchkdsk.c
--- a/fsck/xchkdsk.c	2006-06-05 20:31:40.000000000 +0100
+++ b/fsck/xchkdsk.c	2007-05-02 15:31:24.000000000 +0100
@@ -3028,8 +3028,13 @@
 		}
 		break;
 	default:
+		/* This is disabled as it's silly, we should be able to force
+		 * a read only check of the disk.
 		fsck_send_msg(fsck_PRMMUTEXCLLEVELS);
 		return (FSCK_FAILED);
+		*/
+		agg_recptr->parm_options[UFS_CHKDSK_LEVEL2] = 0;
+		break;
 	}
 	/*
 	 * the parms are valid.  record the messaging level they imply.
