--- Makefile.in.old	Sat Mar  2 08:18:24 2002
+++ Makefile.in	Sat Mar  2 08:18:44 2002
@@ -603,7 +603,7 @@
 
 
 install-data-local:
-	-xmlcatalog --noout --add "system" "http://glade.gnome.org/glade-2.0.dtd" $(xmldir)/glade-2.0.dtd $(XMLCATALOG)
+#	-xmlcatalog --noout --add "system" "http://glade.gnome.org/glade-2.0.dtd" $(xmldir)/glade-2.0.dtd $(XMLCATALOG)
 uninstall-local:
 	-xmlcatalog --noout --del $(xmldir)/glade-2.0.dtd $(XMLCATALOG)
 
