diff -ur gazpacho-0.6.2-orig/setup.py gazpacho-0.6.2/setup.py
--- gazpacho-0.6.2-orig/setup.py	2005-11-24 15:37:31.000000000 +0100
+++ gazpacho-0.6.2/setup.py	2005-11-24 15:45:17.000000000 +0100
@@ -9,7 +9,6 @@
 import sys
 import shutil
 
-from gazpacho import application
 
 def listfiles(*dirs):
     dir, pattern = os.path.split(os.path.join(*dirs))
@@ -58,7 +57,7 @@
 catalogs_dir = os.path.join(prefix, 'share', 'gazpacho', 'catalogs')
 resources_dir = os.path.join(prefix, 'share', 'gazpacho', 'resources')
 glade_dir = os.path.join(prefix, 'share', 'gazpacho', 'glade')
-docs_dir = os.path.join(prefix, 'share', 'doc', 'gazpacho')
+docs_dir = os.path.join(prefix, 'share', 'doc', 'EBUILD_PF_HERE')
 '''
         filename = os.path.join(self.install_dir, 'gazpacho', 
                                 '__installed__.py')
@@ -70,9 +69,9 @@
 	return install_lib.install(self) + [filename]
 
 data_files = [
-    ('share/doc/gazpacho', ('AUTHORS', 'ChangeLog', 'CONTRIBUTORS',
+    ('share/doc/EBUILD_PF_HERE', ('AUTHORS', 'ChangeLog', 'CONTRIBUTORS',
                             'COPYING', 'README', 'NEWS')),
-    ('share/doc/gazpacho/examples', listfiles('examples', '*')),
+    ('share/doc/EBUILD_PF_HERE/examples', listfiles('examples', '*')),
     ('share/gazpacho/catalogs', listfiles('catalogs', 'base.xml')),
     ('share/gazpacho/pixmaps', listfiles('pixmaps', '*.png')),
     ('share/gazpacho/resources/base', listfiles('resources', 'base', '*.png')),
@@ -95,7 +94,7 @@
                        ['gazpacho.desktop']))
 
 setup(name='gazpacho',
-      version=application.__version__,
+      version='EBUILD_PV_HERE',
       description='GTK+ GUI Designer',
       author='SICEm S.L.',
       author_email='lgs@sicem.biz',
