Index: sbin/depscan.sh
===================================================================
--- sbin/depscan.sh	(revision 211)
+++ sbin/depscan.sh	(revision 276)
@@ -3,68 +3,83 @@
 # Distributed under the terms of the GNU General Public License v2
 
 source /sbin/functions.sh
+mysvcdir="${svcdir}"
+update=false
 
-if [[ $1 == "--debug" ]] ; then
+while [[ -n $1 ]] ; do
+	case "$1" in
+		--debug|-d)
+			set -x
+			;;
+		--svcdir|-s)
+			if [[ -z $2 || $2 == -* ]] ; then
+				eerror "No svcdir specified"
+			else
+				shift
+				mysvcdir="$1"
+			fi
+			;;
+		--update|-u)
+			update=true
+			;;
+	esac
 	shift
-	set -x
-fi
+done
 
-if [[ ! -d ${svcdir} ]] ; then
-	if ! mkdir -p -m 0755 "${svcdir}" 2>/dev/null ; then
-		eerror "Could not create needed directory '${svcdir}'!"
+if [[ ! -d ${mysvcdir} ]] ; then
+	if ! mkdir -p -m 0755 "${mysvcdir}" 2>/dev/null ; then
+		eerror "Could not create needed directory '${mysvcdir}'!"
 	fi
 fi
 
 for x in softscripts snapshot options daemons \
 	started starting inactive wasinactive stopping failed \
 	exclusive exitcodes scheduled ; do
-	if [[ ! -d "${svcdir}/${x}" ]] ; then
-		if ! mkdir -p -m 0755 "${svcdir}/${x}" 2>/dev/null ; then
-			eerror "Could not create needed directory '${svcdir}/${x}'!"
+	if [[ ! -d "${mysvcdir}/${x}" ]] ; then
+		if ! mkdir -p -m 0755 "${mysvcdir}/${x}" 2>/dev/null ; then
+			eerror "Could not create needed directory '${mysvcdir}/${x}'!"
 		fi
 	fi
 done
 
 # Only update if files have actually changed
-update=1
+if ! ${update} ; then
+	clock_screw=false
+	mtime_test="${mysvcdir}/mtime-test.$$"
 
-if [[ $1 == "-u" ]] ; then
-	update=0
-	clock_screw=0
-	mtime_test="${svcdir}/mtime-test.$$"
-
 	# If its not there, we have to update, and make sure its present
 	# for next mtime testing
-	if [[ ! -e "${svcdir}/depcache" ]] ; then
-			update=1
-			touch "${svcdir}/depcache"
+	if [[ ! -e "${mysvcdir}/depcache" ]] ; then
+			update=true
+			touch "${mysvcdir}/depcache"
 	fi
 
 	touch "${mtime_test}"
 	for config in /etc/conf.d /etc/init.d /etc/rc.conf
 	do
-		[[ ${update} == 0 ]] && \
-			is_older_than "${svcdir}/depcache" "${config}" && update=1
+		! ${update} \
+			&& is_older_than "${mysvcdir}/depcache" "${config}" \
+			&& update=true
 		
-		is_older_than "${mtime_test}" "${config}" && clock_screw=1
+		is_older_than "${mtime_test}" "${config}" && clock_screw=true
 	done
 	rm -f "${mtime_test}"
 
-	[[ ${clock_screw} == 1 ]] && \
+	${clock_screw} && \
 		ewarn "Some file in '/etc/{conf.d,init.d}' have Modification time in the future!"
 
 	shift
 fi
 
-[[ ${update} == 0 && -e "${svcdir}/deptree" ]] && exit 0
+! ${update} && [[ -e "${mysvcdir}/deptree" ]] && exit 0
 
 ebegin "Caching service dependencies"
 
 # Clean out the non volitile directories ...
-rm -rf "${svcdir}"/dep{cache,tree} "${svcdir}"/{broken,snapshot}/*
+rm -rf "${mysvcdir}"/dep{cache,tree} "${mysvcdir}"/{broken,snapshot}/*
 
 retval=0
-SVCDIR="${svcdir}"
+SVCDIR="${mysvcdir}"
 DEPTYPES="${deptypes}"
 ORDTYPES="${ordtypes}"
 
@@ -77,14 +92,14 @@
 	-f /lib/rcscripts/awk/cachedepends.awk || \
 	retval=1
 
-bash "${svcdir}/depcache" | \
+bash "${mysvcdir}/depcache" | \
 /bin/gawk \
 	-f /lib/rcscripts/awk/functions.awk \
 	-f /lib/rcscripts/awk/gendepends.awk || \
 	retval=1
 
-touch "${svcdir}"/dep{cache,tree}
-chmod 0644 "${svcdir}"/dep{cache,tree}
+touch "${mysvcdir}"/dep{cache,tree}
+chmod 0644 "${mysvcdir}"/dep{cache,tree}
 
 eend ${retval} "Failed to cache service dependencies"
 
Index: sbin/rc-services.sh
===================================================================
--- sbin/rc-services.sh	(revision 234)
+++ sbin/rc-services.sh	(revision 276)
@@ -9,7 +9,7 @@
 
 if [[ ${RC_GOT_DEPTREE_INFO} != "yes" ]] ; then
 	# Only try and update if we are root
-	if [[ ${EUID} == "0" ]] && ! /sbin/depscan.sh -u ; then
+	if [[ ${EUID} == "0" ]] && ! /sbin/depscan.sh ; then
 		echo
 		eerror "Error running '/sbin/depscan.sh'!"
 		eerror "Please correct any problems above."
Index: sbin/rc
===================================================================
--- sbin/rc	(revision 227)
+++ sbin/rc	(revision 276)
@@ -163,7 +163,7 @@
 	check_critical_services
 
 	# Update the dependency cache
-	/sbin/depscan.sh -u
+	/sbin/depscan.sh
 
 	# Now that the dependency cache are up to date, make sure these
 	# are marked as started ...
