Index: net-scripts/net.modules.d/essidnet
===================================================================
--- lib/rcscripts/net.modules.d/essidnet	(revision 1828)
+++ net-scripts/net.modules.d/essidnet	(working copy)
@@ -83,7 +83,8 @@
 	done
 
 	# Nameserver setup for the essid if required
-	local -a nameservers=( \"\$\{mac_dns_servers_${mac}\[@\]\}\" )
+	local -a nameservers
+	eval nameservers=( \"\$\{mac_dns_servers_${mac}\[@\]\}\" )
 	[[ -z ${nameservers} ]] && eval nameservers=( \"\$\{dns_servers_${essid}\[@\]\}\" )
 	if [[ -n ${nameservers} ]]; then
 		# Make a backup of the origional file if necessary
