diff -dur libEMF-1.0.orig/include/wine/winnt.h libEMF-1.0/include/wine/winnt.h
--- libEMF-1.0.orig/include/wine/winnt.h	2002-01-29 05:31:53.000000000 +0100
+++ libEMF-1.0/include/wine/winnt.h	2004-01-22 14:22:35.382239593 +0100
@@ -45,6 +45,10 @@
 # undef WORDS_BIGENDIAN
 # undef BITFIELDS_BIGENDIAN
 # undef ALLOW_UNALIGNED_ACCESS
+#elif defined(__x86_64__)
+# undef  WORDS_BIGENDIAN
+# undef  BITFIELDS_BIGENDIAN
+# define ALLOW_UNALIGNED_ACCESS
 #elif !defined(RC_INVOKED)
 # error Unknown CPU architecture!
 #endif
@@ -211,7 +215,7 @@
 typedef unsigned short  WORD,       *PWORD,    *LPWORD;
 typedef int             INT,        *PINT,     *LPINT;
 typedef unsigned int    UINT,       *PUINT,    *LPUINT;
-#if !defined(__alpha__)
+#if !defined(__alpha__) && !defined(__x86_64__)
 typedef unsigned long   DWORD,      *PDWORD,   *LPDWORD;
 typedef unsigned long   ULONG,      *PULONG,   *LPULONG;
 #else /* defined( __alpha__ ) */
@@ -257,7 +261,7 @@
 typedef BYTE            BOOLEAN,    *PBOOLEAN;
 typedef char            CHAR,       *PCHAR;
 typedef short           SHORT,      *PSHORT;
-#if !defined(__alpha__)
+#if !defined(__alpha__) && !defined(__x86_64__)
 typedef long            LONG,       *PLONG,    *LPLONG;
 #else /* defined( __alpha__ ) */
 typedef int             LONG,       *PLONG,    *LPLONG;
@@ -1045,6 +1049,11 @@
 
 #endif  /* __sparc__ */
 
+#if defined(__x86_64__)
+#define CONTEXT_FULL 1
+typedef struct _CONTEXT CONTEXT;
+#endif
+
 #if !defined(CONTEXT_FULL) && !defined(RC_INVOKED)
 #error You need to define a CONTEXT for your CPU
 #endif
diff -dur libEMF-1.0.orig/libemf/libemf.h libEMF-1.0/libemf/libemf.h
--- libEMF-1.0.orig/libemf/libemf.h	2002-01-29 05:54:27.000000000 +0100
+++ libEMF-1.0/libemf/libemf.h	2004-01-22 14:23:14.315813724 +0100
@@ -332,7 +332,7 @@
 	fread( &dword, sizeof(DWORD), 1, fp_ );
       return *this;
     }
-#if !defined(__alpha__)
+#if !defined(__alpha__) && !defined(__x86_64__)
     /*!
      * Output a long int to the stream (swabbed).
      * \param long long int to output.
@@ -402,7 +402,7 @@
 	fread( &int_, sizeof(INT), 1, fp_ );
       return *this;
     }
-#if !defined(__alpha__)
+#if !defined(__alpha__) && !defined(__x86_64__)
     /*!
      * Output a (long) unsigned int to the stream (swabbed).
      * \param uint (long) unsigned int to output.
Only in libEMF-1.0/libemf: libemf.h~
