--- ./scene.cpp.gege	2002-10-10 14:57:58.000000000 +0200
+++ ./scene.cpp	2003-04-25 12:30:01.000000000 +0200
@@ -51,6 +51,7 @@
 extern bool bKeyDown;    
 
 
+/*
 PFNGLCOMBINERPARAMETERINVPROC glCombinerParameteriNV;
 PFNGLCOMBINERINPUTNVPROC glCombinerInputNV;
 PFNGLCOMBINEROUTPUTNVPROC glCombinerOutputNV;
@@ -67,6 +68,7 @@
 PFNGLACTIVETEXTUREARBPROC glActiveTextureARB;
 PFNGLMULTITEXCOORD2FARBPROC glMultiTexCoord2fARB; 
 #endif
+*/
 
 GLenum glError;
 
@@ -1511,6 +1511,7 @@
 		glTrackMatrixNV=(PFNGLTRACKMATRIXNVPROC)wglGetProcAddress("glTrackMatrixNV");
 		glProgramParameter4fNV=(PFNGLPROGRAMPARAMETER4FNVPROC)wglGetProcAddress("glProgramParameter4fNV");
 #else 
+/*
 		glCombinerParameteriNV=(PFNGLCOMBINERPARAMETERINVPROC)glXGetProcAddressARB((const GLubyte *)"glCombinerParameteriNV");
 		glCombinerInputNV=(PFNGLCOMBINERINPUTNVPROC)glXGetProcAddressARB((const GLubyte *)"glCombinerInputNV");
 		glCombinerOutputNV=(PFNGLCOMBINEROUTPUTNVPROC)glXGetProcAddressARB((const GLubyte *)"glCombinerOutputNV");
@@ -1520,6 +1521,7 @@
 		glBindProgramNV=(PFNGLBINDPROGRAMNVPROC)glXGetProcAddressARB((const GLubyte *)"glBindProgramNV");
 		glTrackMatrixNV=(PFNGLTRACKMATRIXNVPROC)glXGetProcAddressARB((const GLubyte *)"glTrackMatrixNV");
 		glProgramParameter4fNV=(PFNGLPROGRAMPARAMETER4FNVPROC)glXGetProcAddressARB((const GLubyte *)"glProgramParameter4fNV");
+*/
 #endif
 		// enabled in when required
 		glDisable(GL_VERTEX_PROGRAM_NV);
--- ./myship.cpp.gege	2002-10-12 23:16:16.000000000 +0200
+++ ./myship.cpp	2003-04-25 12:26:58.000000000 +0200
@@ -19,6 +19,7 @@
 
 
 
+#define GL_GLEXT_PROTOTYPES 1
 #include <GL/glut.h>
 #include <GL/glext.h>
 
@@ -35,10 +36,10 @@
 #include "modeles/protection.c"
 
 
-extern PFNGLCOMBINERPARAMETERINVPROC glCombinerParameteriNV;
-extern PFNGLCOMBINERINPUTNVPROC glCombinerInputNV;
-extern PFNGLCOMBINEROUTPUTNVPROC glCombinerOutputNV;
-extern PFNGLFINALCOMBINERINPUTNVPROC glFinalCombinerInputNV;
+//extern PFNGLCOMBINERPARAMETERINVPROC glCombinerParameteriNV;
+//extern PFNGLCOMBINERINPUTNVPROC glCombinerInputNV;
+//extern PFNGLCOMBINEROUTPUTNVPROC glCombinerOutputNV;
+//extern PFNGLFINALCOMBINERINPUTNVPROC glFinalCombinerInputNV;
 #ifdef WIN32
 extern PFNGLACTIVETEXTUREARBPROC glActiveTextureARB;
 #endif
