diff -Naur qmpdclient-1.0.8.old/src/qmpdclient.cpp qmpdclient-1.0.8/src/qmpdclient.cpp
--- qmpdclient-1.0.8.old/src/qmpdclient.cpp	2007-07-05 03:03:11.000000000 +0200
+++ qmpdclient-1.0.8/src/qmpdclient.cpp	2007-07-05 03:03:50.000000000 +0200
@@ -34,7 +34,7 @@
 #include <QTranslator>
 #include <QWheelEvent>
 
-QMPDClient::QMPDClient(int argc, char **argv) : QApplication(argc, argv), m_translator(0), m_qtTranslator(0) {
+QMPDClient::QMPDClient(int &argc, char **argv) : QApplication(argc, argv), m_translator(0), m_qtTranslator(0) {
 	setObjectName("qmpdclient");
 	setApplicationName("QMPDClient");
 	setOrganizationName("QMPDClient");
diff -Naur qmpdclient-1.0.8.old/src/qmpdclient.h qmpdclient-1.0.8/src/qmpdclient.h
--- qmpdclient-1.0.8.old/src/qmpdclient.h	2007-07-05 03:03:11.000000000 +0200
+++ qmpdclient-1.0.8/src/qmpdclient.h	2007-07-05 03:03:35.000000000 +0200
@@ -32,7 +32,7 @@
 class QMPDClient : public QApplication {
 	Q_OBJECT
 public:
-	QMPDClient(int, char **);
+	QMPDClient(int &, char **);
 	~QMPDClient();
 	bool eventFilter(QObject *, QEvent *);
 #ifdef Q_WS_X11
