diff --git a/CMakeLists.txt b/CMakeLists.txt
index 28fd3b4..4197a69 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 
-find_package(Qt4 4.4.0 COMPONENTS QtCore QtGui QtNetwork QtXml QtXmlPatterns REQUIRED QtDBus X11)
+find_package(Qt4 4.4.0 COMPONENTS QtCore QtGui QtNetwork QtXml QtXmlPatterns REQUIRED QtDBus)
+find_package(X11 REQUIRED)
 
 SET (QT_USE_QTNETWORK TRUE)
 SET (QT_USE_QTXMLPATTERNS TRUE)
@@ -279,6 +280,7 @@ add_definitions(-DNAMEVER="${NAMEVER}" -DPREFIX="${CMAKE_INSTALL_PREFIX}" -DVERS
 add_executable(qmpdclient ${QMPDClient_srcs} ${QMPDClient_ppd_hdrs} ${QMPDClient_ppd_uis} ${QMPDClient_ppd_res} ${COMPILED_TRANSLATIONS})
 
 target_link_libraries(qmpdclient ${QT_LIBRARIES})
+target_link_libraries(qmpdclient ${X11_LIBRARIES})
 
 install(TARGETS qmpdclient DESTINATION bin)
 if(UNIX)
