--- src/desktopfiles/gwenview.desktop.o	2009-02-13 23:50:30.000000000 +0100
+++ src/desktopfiles/gwenview.desktop	2009-02-13 23:50:54.000000000 +0100
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Gwenview
 Name[hi]=ग्वेनव्यू
 Name[ta]=gwen பார்வை
@@ -98,10 +97,10 @@
 Comment[zh_CN]=一个简单的图像查看器
 Comment[zh_TW]=一個簡易的影像檢視器
 Comment[zu]=Umbukisi wesithombe esilula
-Exec=gwenview %u -caption "%c" %i %m
+Exec=gwenview %u -caption "%c" %i
 Terminal=false
 Icon=gwenview
 Type=Application
-Categories=Qt;KDE;Application;Graphics;
+Categories=Qt;KDE;Graphics;
 MimeType=image/gif;image/x-xpm;image/x-xbm;image/jpeg;image/x-pcx;image/x-bmp;image/png;image/x-ico;image/x-portable-bitmap;image/x-portable-pixmap;image/x-portable-greymap;image/tiff;image/x-targa;
-DocPath=gwenview/index.html
+X-DocPath=gwenview/index.html
