diff -Nuar nut-2.2.2.orig/common/Makefile.am nut-2.2.2/common/Makefile.am
--- nut-2.2.2.orig/common/Makefile.am	2007-05-27 07:07:56.000000000 -0700
+++ nut-2.2.2/common/Makefile.am	2008-07-04 20:09:14.062814944 -0700
@@ -10,8 +10,8 @@
 
 # Define a dummy library so that Automake builds rules for the
 # corresponding object files.  This library is not actually built,
-EXTRA_LIBRARIES = libdummy.a
-libdummy_a_SOURCES = parseconf.c state.c upsconf.c
+#EXTRA_LIBRARIES = libdummy.a
+#libdummy_a_SOURCES = parseconf.c state.c upsconf.c
 
 ../include/nut_version.h: FORCE
 	(cd ../include/ && $(MAKE) $(AM_MAKEFLAGS) nut_version.h)
diff -Nuar nut-2.2.2.orig/drivers/Makefile.am nut-2.2.2/drivers/Makefile.am
--- nut-2.2.2.orig/drivers/Makefile.am	2008-07-04 20:12:38.201211819 -0700
+++ nut-2.2.2/drivers/Makefile.am	2008-07-04 20:12:59.829462787 -0700
@@ -211,10 +211,12 @@
  upscode2.h victronups.h powerpanel.h powerp-bin.h powerp-txt.h		\
  upshandler.h netxml-ups.h mge-xml.h
 
+EXTRA_LIBRARIES =
+
 # Define a dummy library so that Automake builds rules for the
 # corresponding object files.  This library is not actually built,
-EXTRA_LIBRARIES = libdummy.a
-libdummy_a_SOURCES = main.c dstate.c serial.c
+#EXTRA_LIBRARIES += libdummy.a
+#libdummy_a_SOURCES = main.c dstate.c serial.c
 
 # the nuthalmain library combines the code for main-hal.c and
 # dstate-hal.c. It is necessary for Automake-technical reasons,
