diff -uNp l1/include/asm-ia64/system.h.orig l1/include/asm-ia64/system.h
--- l1/include/asm-ia64/system.h.orig	2004-02-18 04:57:31.000000000 +0100
+++ l1/include/asm-ia64/system.h	2004-07-01 17:01:52.185515640 +0200
@@ -233,7 +233,8 @@ extern void ia64_load_extra (struct task
  * the latest fph state from another CPU.  In other words: eager save, lazy restore.
  */
 # define switch_to(prev,next,last) do {						\
-	if (ia64_psr(ia64_task_regs(prev))->mfh) {				\
+	if (ia64_psr(ia64_task_regs(prev))->mfh					\
+		&& ia64_is_local_fpu_owner(prev)) {				\
 		ia64_psr(ia64_task_regs(prev))->mfh = 0;			\
 		(prev)->thread.flags |= IA64_THREAD_FPH_VALID;			\
 		__ia64_save_fpu((prev)->thread.fph);				\
