--- linux-2.4.26/arch/ia64/ia32/sys_ia32.c	Fri Feb 20 10:07:21 2004
+++ linux/arch/ia64/ia32/sys_ia32.c	Sat Apr 17 02:13:55 2004
@@ -111,6 +111,8 @@
 			*ap++ = (char *) A(addr);
 		arg += sizeof(unsigned int);
 		n++;
+		if (n >= (MAX_ARG_PAGES * PAGE_SIZE) / sizeof(char *))
+			return -E2BIG;
 	} while (addr);
 	return n - 1;
 }
diff -urPX nopatch linux-2.4.26/arch/ia64/kernel/unwind.c linux/arch/ia64/kernel/unwind.c
--- linux-2.4.26/arch/ia64/kernel/unwind.c	Sat Apr 17 02:10:25 2004
+++ linux/arch/ia64/kernel/unwind.c	Sun Jun 13 10:40:37 2004
@@ -1750,7 +1750,7 @@
 			if (!state->pri_unat_loc)
 				state->pri_unat_loc = &state->sw->ar_unat;
 			/* register off. is a multiple of 8, so the least 3 bits (type) are 0 */
-			s[dst+1] = (*state->pri_unat_loc - s[dst]) | UNW_NAT_MEMSTK;
+			s[dst+1] = ((unsigned long) state->pri_unat_loc - s[dst]) | UNW_NAT_MEMSTK;
 			break;
 
 		      case UNW_INSN_SETNAT_TYPE:
