diff -Naur gift-gnutella-0.0.9.2.vanilla/src/gt_node.c gift-gnutella-0.0.9.2/src/gt_node.c
--- gift-gnutella-0.0.9.2.vanilla/src/gt_node.c	2003-12-11 01:43:14.000000000 -0600
+++ gift-gnutella-0.0.9.2/src/gt_node.c	2004-07-22 21:21:03.933389912 -0500
@@ -310,7 +310,7 @@
 	node->tx_deflated = FALSE;
 
 	/* close the connection for this node, if any */
-	tcp_close_null (&GT_CONN(node));
+	tcp_close_null (&node->c);
 
 	/*pong_cache_waiter_remove (c);*/
 
