--- Makefile.orig	2003-08-06 10:36:18.000000000 +0200
+++ Makefile	2003-08-06 10:37:06.000000000 +0200
@@ -113,6 +113,7 @@
 install:
 		install -d -m 755 $(PREFIX)/include/$(SOBASE)/;
 		install -m 644 ezV24.h $(PREFIX)/include/$(SOBASE)/
+		install -d -m 755 $(PREFIX)/lib
 		install -m 644 -s $(LIBNAME) $(PREFIX)/lib/$(LIBNAME)
 		install -m 755 -s $(NAME) $(PREFIX)/lib/$(NAME)
 		rm -f $(PREFIX)/lib/$(SONAME) $(PREFIX)/lib/$(PLAINNAME)
