--- partysip-2.2.3/configure.in.orig	2005-07-17 03:16:39.000000000 +0200
+++ partysip-2.2.3/configure.in	2005-07-17 03:17:16.000000000 +0200
@@ -284,16 +284,16 @@
 				[PPL_LIB="$PPL_LIB -ldb"])])
     ;;
     susedb3)
-	AC_CHECK_HEADERS(db3/db.h,[AC_CHECK_LIB(db,__db_ndbm_open,
-				[PPL_LIB="$PPL_LIB -ldb"])])
+	AC_CHECK_HEADERS(db3/db.h,[AC_CHECK_LIB(db-3,__db_ndbm_open,
+				[PPL_LIB="$PPL_LIB -ldb-3"])])
     ;;
     susedb2)
-	AC_CHECK_HEADERS(db2/db.h,[AC_CHECK_LIB(db,__db_ndbm_open,
-				[PPL_LIB="$PPL_LIB -ldb"])])
+	AC_CHECK_HEADERS(db2/db.h,[AC_CHECK_LIB(db-2,__db_ndbm_open,
+				[PPL_LIB="$PPL_LIB -ldb-2"])])
     ;;
     susedb1)
-	AC_CHECK_HEADERS(db1/db.h,[AC_CHECK_LIB(db,__db_ndbm_open,
-				[PPL_LIB="$PPL_LIB -ldb"])])
+	AC_CHECK_HEADERS(db1/db.h,[AC_CHECK_LIB(db-1,__db_ndbm_open,
+				[PPL_LIB="$PPL_LIB -ldb-1"])])
     ;;
     dbm)
 	AC_CHECK_HEADERS(dbm.h,[AC_CHECK_LIB(dbm,dbm_open,
@@ -301,7 +301,7 @@
     ;;
     gdbm)
 	AC_CHECK_HEADERS(gdbm-ndbm.h)
-dnl	AC_CHECK_HEADERS(gdbm.h)
+	AC_CHECK_HEADERS(gdbm.h)
 	AC_CHECK_LIB(gdbm,dbm_open,[PPL_LIB="$PPL_LIB -lgdbm"])
     ;;
     ndbm)
