--- siproxd-0.5.11/configure.in.orig	2005-05-07 16:43:13.333858336 +0200
+++ siproxd-0.5.11/configure.in	2005-05-07 16:52:22.980299376 +0200
@@ -72,10 +72,21 @@
 AC_PROG_RANLIB
 AC_PROG_INSTALL
 AC_PROG_LN_S
-AC_CHECK_PROG(docbook2pdf,docbook2pdf,yes,)
-AM_CONDITIONAL(have_docbook2pdf,test "$docbook2pdf" = "yes")
-AC_CHECK_PROG(docbook2html,docbook2html,yes,)
-AM_CONDITIONAL(have_docbook2html,test "$docbook2html" = "yes")
+
+AC_MSG_CHECKING(build statically linked  executable)
+AC_ARG_ENABLE(docs,
+   [  --enable-docs               build pdf/html documentation],
+   build_docs="$enableval";
+   AC_MSG_RESULT(yes), AC_MSG_RESULT(no))
+if test "x$build_docs" = "xyes"; then
+   AC_CHECK_PROG(docbook2pdf,docbook2pdf,yes,)
+   AM_CONDITIONAL(have_docbook2pdf,test "$docbook2pdf" = "yes")
+   AC_CHECK_PROG(docbook2html,docbook2html,yes,)
+   AM_CONDITIONAL(have_docbook2html,test "$docbook2html" = "yes")
+else
+   AM_CONDITIONAL(have_docbook2pdf,false)
+   AM_CONDITIONAL(have_docbook2html,false)
+fi
 
 dnl
 dnl add
