diff -burNp gsynaptics-0.9.13/configure.ac gsynaptics-0.9.13/configure.ac
--- gsynaptics-0.9.13/configure.ac	2007-11-28 16:12:02.000000000 -0800
+++ gsynaptics-0.9.13/configure.ac	2007-12-23 21:26:00.000000000 -0800
@@ -18,14 +18,14 @@ AC_PROG_CC
 # Checks for libraries.
 AM_PATH_GTK_2_0(2.6.0)
 
-PKG_CHECK_MODULES(GNOME, libgnomeui-2.0)
-AC_SUBST(GNOME_CFLAGS)
-AC_SUBST(GNOME_LIBS)
-
 PKG_CHECK_MODULES(GLADE2, libglade-2.0)
 AC_SUBST(GLADE2_CFLAGS)
 AC_SUBST(GLADE2_LIBS)
 
+PKG_CHECK_MODULES(GCONF2, gconf-2.0)
+AC_SUBST(GCONF2_CFLAGS)
+AC_SUBST(GCONF2_LIBS)
+
 GETTEXT_PACKAGE=gsynaptics
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE], ["$GETTEXT_PACKAGE"], [Gettext package])
diff -burNp gsynaptics-0.9.13/src/init.c gsynaptics-0.9.13/src/init.c
--- gsynaptics-0.9.13/src/init.c	2007-10-29 00:37:23.000000000 -0700
+++ gsynaptics-0.9.13/src/init.c	2007-12-23 21:31:30.000000000 -0800
@@ -22,7 +22,7 @@
 #  include <config.h>
 #endif
 
-#include <gnome.h>
+#include <glib/gi18n.h>
 #include <gconf/gconf-client.h>
 
 #include "gsynaptics.h"
@@ -91,10 +91,7 @@ main (int argc, char **argv)
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
 
-	gnome_program_init ("gsynaptics-init", VERSION,
-			LIBGNOMEUI_MODULE, argc, argv,
-			GNOME_PARAM_APP_DATADIR, NULL,
-			NULL);
+	gtk_init (&argc, &argv);
 
 	synaptics = g_synaptics_new ();
 
diff -burNp gsynaptics-0.9.13/src/main.c gsynaptics-0.9.13/src/main.c
--- gsynaptics-0.9.13/src/main.c	2007-11-28 16:09:08.000000000 -0800
+++ gsynaptics-0.9.13/src/main.c	2007-12-23 21:28:36.000000000 -0800
@@ -22,7 +22,7 @@
 #  include <config.h>
 #endif
 
-#include <gnome.h>
+#include <glib/gi18n.h>
 #include <glade/glade.h>
 #include <gconf/gconf-client.h>
 
@@ -683,7 +683,7 @@ setup_dialog (GladeXML *dialog)
 
 	gtk_window_set_default_size (GTK_WINDOW (widget), 400, -1);
 
-	path = gnome_pixmap_file ("touchpad.png");
+	path = g_build_filename (DATADIR "/pixmaps", "touchpad.png");
 	if (path)
 	{
 		icon_pixbuf = gdk_pixbuf_new_from_file (path, NULL);
@@ -712,10 +712,7 @@ main (int argc, char **argv)
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
 
-	gnome_program_init ("gsynaptics", VERSION,
-			LIBGNOMEUI_MODULE, argc, argv,
-			GNOME_PARAM_APP_DATADIR, DATADIR,
-			NULL);
+	gtk_init (&argc, &argv);
 
 	synaptics = g_synaptics_new ();
 	if (!g_synaptics_is_available(synaptics))
diff -burNp gsynaptics-0.9.13/src/Makefile.am gsynaptics-0.9.13/src/Makefile.am
--- gsynaptics-0.9.13/src/Makefile.am	2007-10-29 00:37:23.000000000 -0700
+++ gsynaptics-0.9.13/src/Makefile.am	2007-12-23 21:31:53.000000000 -0800
@@ -3,8 +3,8 @@ gsynaptics_CFLAGS =				\
 	-DDATADIR=\""$(datadir)/$(PACKAGE)"\" 	\
 	-DHELPDATADIR=\""$(datadir)"\"	 	\
 	$(GTK_CFLAGS)				\
-	$(GNOME_CFLAGS)				\
 	$(GLADE2_CFLAGS)			\
+	$(GCONF2_CFLAGS)			\
 	$(AM_CFLAGS)
 
 gsynaptics_SOURCES =			\
@@ -15,8 +15,8 @@ gsynaptics_SOURCES =			\
 
 gsynaptics_LDADD =	\
 	$(GTK_LIBS)	\
-	$(GNOME_LIBS)	\
-	$(GLADE2_LIBS)
+	$(GLADE2_LIBS)	\
+	$(GCONF2_LIBS)
 
 gsynaptics_init_SOURCES =		\
 	gsynaptics-conf.h		\
@@ -25,11 +25,14 @@ gsynaptics_init_SOURCES =		\
 	gsynaptics.c gsynaptics.h
 
 gsynaptics_init_CFLAGS =				\
+	$(GTK_CFLAGS)				\
 	$(GLIB_CFLAGS)				\
 	$(GNOME_CFLAGS)				\
+	$(GCONF2_CFLAGS)			\
 	$(AM_CFLAGS)
 
 gsynaptics_init_LDADD =	\
+	$(GTK_LIBS)	\
 	$(GLIB_LIBS)	\
-	$(GNOME_LIBS)
-
+	$(GNOME_LIBS)	\
+	$(GCONF2_LIBS)
