--- zmq/tests/test_message.py
+++ zmq/tests/test_message.py
@@ -256,7 +256,7 @@
             import numpy
         except ImportError:
             raise SkipTest("NumPy unavailable")
-        shapes = map(numpy.random.randint, [2]*5,[16]*5)
+        shapes = list(map(numpy.random.randint, [2]*5,[16]*5))
         for i in range(1,len(shapes)+1):
             shape = shapes[:i]
             A = numpy.random.random(shape)
