diff -Nru openswan-2.4.15.orig/programs/pluto/Makefile openswan-2.4.15/programs/pluto/Makefile
--- openswan-2.4.15.orig/programs/pluto/Makefile	2009-09-11 18:32:38.000000000 +0200
+++ openswan-2.4.15/programs/pluto/Makefile	2009-09-11 19:35:33.000000000 +0200
@@ -244,6 +244,7 @@
 	-DGCC_LINT \
 	-DUSE_AES -DUSE_3DES \
 	-DIKE_ALG -DKERNEL_ALG -DDB_CONTEXT \
+	-DALLOW_MICROSOFT_BAD_PROPOSAL \
 	${AGGRESSIVE_DEFS} \
 	${XAUTH_DEFS} ${XAUTHPAM_DEFS} \
 	${NAT_DEFS} ${CURL_DEFS}\
