diff -ru tvbrowser-2.5-orig/src/util/ui/Localizer.java tvbrowser-2.5/src/util/ui/Localizer.java
--- tvbrowser-2.5-orig/src/util/ui/Localizer.java	2007-03-11 17:03:17.662679355 +0100
+++ tvbrowser-2.5/src/util/ui/Localizer.java	2007-03-11 17:03:39.741495254 +0100
@@ -348,7 +348,7 @@
       langArray.add(Locale.ENGLISH);
       
       // First Step: look into tvbrowser.jar
-      JarFile file = new JarFile(new File("tvbrowser.jar"));
+      JarFile file = new JarFile(new File("lib/tvbrowser.jar"));
       
       Enumeration<JarEntry> entries = file.entries();
       
