diff -Naurp gmp-4.1.4.orig/configure.in gmp-4.1.4/configure.in
--- gmp-4.1.4.orig/configure.in	2004-09-21 06:02:43.000000000 -0700
+++ gmp-4.1.4/configure.in	2005-01-27 12:50:33.627680514 -0800
@@ -291,7 +291,7 @@ esac],
 # hugely bloat the output.
 
 
-AC_ARG_VAR(ABI, [desired ABI (for processors supporting more than one ABI)])
+AC_ARG_VAR(GMPABI, [desired ABI (for processors supporting more than one ABI)])
 
 # abilist needs to be non-empty, "standard" is just a generic name here
 abilist="standard"
@@ -1081,7 +1081,7 @@ CPPFLAGS_or_unset=${CPPFLAGS-'(unset)'}
 
 cat >&AC_FD_CC <<EOF
 configure:__line__: User:
-ABI=$ABI
+GMPABI=$GMPABI
 CC=$CC
 CFLAGS=$CFLAGS_or_unset
 CPPFLAGS=$CPPFLAGS_or_unset
@@ -1102,15 +1102,15 @@ done
 # If the user specifies an ABI then it must be in $abilist, after that
 # $abilist is restricted to just that choice.
 #
-if test -n "$ABI"; then
+if test -n "$GMPABI"; then
   found=no
   for abi in $abilist; do
-    if test $abi = "$ABI"; then found=yes; break; fi
+    if test $abi = "$GMPABI"; then found=yes; break; fi
   done
   if test $found = no; then
-    AC_MSG_ERROR([ABI=$ABI is not among the following valid choices: $abilist])
+    AC_MSG_ERROR([GMPABI=$GMPABI is not among the following valid choices: $abilist])
   fi
-  abilist="$ABI"
+  abilist="$GMPABI"
 fi
 
 found_compiler=no
@@ -1311,14 +1311,14 @@ if test $found_compiler = yes; then
     done
   fi
 
-  ABI="$abi"
+  GMPABI="$abi"
   CC="$cc"
   CFLAGS="$cflags"
   CPPFLAGS="$cppflags"
 
 
   # Could easily have this in config.h too, if desired.
-  ABI_nodots=`echo $ABI | sed 's/\./_/'`
+  ABI_nodots=`echo $GMPABI | sed 's/\./_/'`
   GMP_DEFINE_RAW("define_not_for_expansion(\`HAVE_ABI_$ABI_nodots')", POST)
 
 
@@ -1480,7 +1480,7 @@ fi
 
 cat >&AC_FD_CC <<EOF
 configure:__line__: Decided:
-ABI=$ABI
+GMPABI=$GMPABI
 CC=$CC
 CFLAGS=$CFLAGS
 CPPFLAGS=$CPPFLAGS
@@ -1489,7 +1489,7 @@ CXX=$CXX
 CXXFLAGS=$CXXFLAGS
 path=$path
 EOF
-echo "using ABI=\"$ABI\""
+echo "using GMPABI=\"$GMPABI\""
 echo "      CC=\"$CC\""
 echo "      CFLAGS=\"$CFLAGS\""
 echo "      CPPFLAGS=\"$CPPFLAGS\""
@@ -2012,13 +2012,13 @@ if test "$gmp_asm_syntax_testing" != no;
       GMP_INCLUDE_MPN(powerpc32/powerpc-defs.m4)
       case $host in
         powerpc*-*-aix*)
-	  case $ABI in
+	  case $GMPABI in
 	    64 | aix64) GMP_INCLUDE_MPN(powerpc64/aix.m4) ;;
             *)          GMP_INCLUDE_MPN(powerpc32/aix.m4) ;;
           esac
           ;;
         powerpc*-*-linux*)
-	  case $ABI in
+	  case $GMPABI in
 	    64) GMP_INCLUDE_MPN(powerpc64/linux64.m4) ;;
           esac
           ;;
@@ -2028,7 +2028,7 @@ if test "$gmp_asm_syntax_testing" != no;
       GMP_INCLUDE_MPN(powerpc32/aix.m4)
       ;;
     sparcv9*-*-* | ultrasparc*-*-* | sparc64-*-*)
-      case $ABI in
+      case $GMPABI in
         64)
           GMP_ASM_SPARC_REGISTER
           ;;
