Index: include/asm-mips/page.h
===================================================================
RCS file: /home/cvs/linux/include/asm-mips/page.h,v
retrieving revision 1.41
diff -u -r1.41 page.h
--- include/asm-mips/page.h	18 Mar 2004 22:54:01 -0000	1.41
+++ include/asm-mips/page.h	7 Jun 2004 04:19:34 -0000
@@ -10,10 +10,11 @@
 #define _ASM_PAGE_H
 
 #include <linux/config.h>
-#include <spaces.h>
 
 #ifdef __KERNEL__
 
+#include <spaces.h>
+
 /*
  * PAGE_SHIFT determines the page size
  */
