diff -ur ghostscript-8.54.orig/src/all-arch.mak ghostscript-8.54/src/all-arch.mak
--- ghostscript-8.54.orig/src/all-arch.mak	2006-05-23 19:59:45.000000000 +0200
+++ ghostscript-8.54/src/all-arch.mak	2006-05-23 20:32:44.000000000 +0200
@@ -295,24 +295,24 @@
 # Remove the old gs binary first, so as to preserve the previous
 # gs-x.yy version, if any.
 install-binary:
-	-$(RM) $(BINDIR)/$(GS)
+	-$(RM) $(DESTDIR)$(BINDIR)/$(GS)
 	@$(MAKE) $(ARGS) install GS=$(GS) XLIBDIRS=$(XLIBDIRSALL) ; \
 	d=`pwd` ; \
 	d=`basename $$d` ; \
 	d=`echo $$d | sed -e s/gs/gs-/` ; \
-	$(RM) $(BINDIR)/$$d ; \
-	ln $(BINDIR)/$(GS) $(BINDIR)/$$d ; \
-	ls -l $(BINDIR)/$(GS) $(BINDIR)/$$d
+	$(RM) $(DESTDIR)$(BINDIR)/$$d ; \
+	ln $(DESTDIR)$(BINDIR)/$(GS) $(DESTDIR)$(BINDIR)/$$d ; \
+	ls -l $(DESTDIR)$(BINDIR)/$(GS) $(DESTDIR)$(BINDIR)/$$d
 
 install-binary-gnu-readline:
-	-$(RM) $(BINDIR)/$(GS)
+	-$(RM) $(DESTDIR)$(BINDIR)/$(GS)
 	@$(MAKE) $(ARGS) install GS=$(GS)  XLIBDIRS=$(XLIBDIRSALL) $(GNU_READLINE_ARGS) ; \
 	d=`pwd` ; \
 	d=`basename $$d` ; \
 	d=`echo $$d | sed -e s/gs/gs-/` ; \
-	$(RM) $(BINDIR)/$$d ; \
-	ln $(BINDIR)/$(GS) $(BINDIR)/$$d ; \
-	ls -l $(BINDIR)/$(GS) $(BINDIR)/$$d
+	$(RM) $(DESTDIR)$(BINDIR)/$$d ; \
+	ln $(DESTDIR)$(BINDIR)/$(GS) $(DESTDIR)$(BINDIR)/$$d ; \
+	ls -l $(DESTDIR)$(BINDIR)/$(GS) $(DESTDIR)$(BINDIR)/$$d
 
 install-fontmap:
 	@d=`pwd` ; \
@@ -320,14 +320,14 @@
 	d=`echo $$d | sed -e s/gs//` ; \
 	if test -f $(GS_SRC_DIR)/Fontmap.new ; \
 	then \
-		if test -f $(GS_SHARE_DIR)/$$d/lib/Fontmap.org ; \
+		if test -f $(DESTDIR)$(GS_SHARE_DIR)/$$d/lib/Fontmap.org ; \
 		then \
 			true ; \
 		else \
-			mv $(GS_SHARE_DIR)/$$d/lib/Fontmap $(GS_SHARE_DIR)/$$d/lib/Fontmap.org ; \
+			mv $(DESTDIR)$(GS_SHARE_DIR)/$$d/lib/Fontmap $(DESTDIR)$(GS_SHARE_DIR)/$$d/lib/Fontmap.org ; \
 		fi ; \
-		$(CP) $(GS_SRC_DIR)/Fontmap.new $(GS_SHARE_DIR)/$$d/lib/Fontmap ; \
-		ls -l $(GS_SHARE_DIR)/$$d/lib/Fontmap* ; \
+		$(CP) $(GS_SRC_DIR)/Fontmap.new $(DESTDIR)$(GS_SHARE_DIR)/$$d/lib/Fontmap ; \
+		ls -l $(DESTDIR)$(GS_SHARE_DIR)/$$d/lib/Fontmap* ; \
 	fi
 
 install-pdfsec:
@@ -336,9 +336,9 @@
 	d=`echo $$d | sed -e s/gs//` ; \
 	if test -f $(GS_SRC_DIR)/lib/pdf_sec.ps ; \
 	then \
-		$(MV) $(GS_SHARE_DIR)/$$d/lib/pdf_sec.ps $(GS_SHARE_DIR)/$$d/lib/pdf_sec.ps.org ; \
-		$(CP) lib/pdf_sec.ps $(GS_SHARE_DIR)/$$d/lib/pdf_sec.ps ; \
-		$(CHMOD) 664 $(GS_SHARE_DIR)/$$d/lib/pdf_sec.ps ; \
+		$(MV) $(DESTDIR)$(GS_SHARE_DIR)/$$d/lib/pdf_sec.ps $(DESTDIR)$(GS_SHARE_DIR)/$$d/lib/pdf_sec.ps.org ; \
+		$(CP) lib/pdf_sec.ps $(DESTDIR)$(GS_SHARE_DIR)/$$d/lib/pdf_sec.ps ; \
+		$(CHMOD) 664 $(DESTDIR)$(GS_SHARE_DIR)/$$d/lib/pdf_sec.ps ; \
 	fi
 
 #=======================================================================
Only in ghostscript-8.54/src: all-arch.mak.orig
diff -ur ghostscript-8.54.orig/src/macos-fw.mak ghostscript-8.54/src/macos-fw.mak
--- ghostscript-8.54.orig/src/macos-fw.mak	2006-05-23 19:59:45.000000000 +0200
+++ ghostscript-8.54/src/macos-fw.mak	2006-05-23 20:32:44.000000000 +0200
@@ -100,19 +100,19 @@
 	$(MAKE) $(SODEFS) GENOPT='-DDEBUG' CFLAGS='$(CFLAGS_DEBUG) $(CFLAGS_SO) $(GCFLAGS) $(XCFLAGS)' $(GSSOC) $(GSSOX)
 
 install-so: so
-	-mkdir $(prefix)
-	-mkdir $(datadir)
-	-mkdir $(gsdir)
-	-mkdir $(gsdatadir)
-	-mkdir $(bindir)
-	-mkdir $(libdir)
-	$(INSTALL_PROGRAM) $(GSSOC) $(bindir)/$(GSSOC_XENAME)
-	$(INSTALL_PROGRAM) $(GSSOX) $(bindir)/$(GSSOX_XENAME)
-	$(INSTALL_PROGRAM) $(BINDIR)/$(SOBINRELDIR)/$(GS_SONAME_MAJOR_MINOR) $(libdir)/$(GS_SONAME_MAJOR_MINOR)
-	$(RM_) $(libdir)/$(GS_SONAME)
-	ln -s $(GS_SONAME_MAJOR_MINOR) $(libdir)/$(GS_SONAME)
-	$(RM_) $(libdir)/$(GS_SONAME_MAJOR)
-	ln -s $(GS_SONAME_MAJOR_MINOR) $(libdir)/$(GS_SONAME_MAJOR)
+	-mkdir $(DESTDIR)$(prefix)
+	-mkdir $(DESTDIR)$(datadir)
+	-mkdir $(DESTDIR)$(gsdir)
+	-mkdir $(DESTDIR)$(gsdatadir)
+	-mkdir $(DESTDIR)$(bindir)
+	-mkdir $(DESTDIR)$(libdir)
+	$(INSTALL_PROGRAM) $(GSSOC) $(DESTDIR)$(bindir)/$(GSSOC_XENAME)
+	$(INSTALL_PROGRAM) $(GSSOX) $(DESTDIR)$(bindir)/$(GSSOX_XENAME)
+	$(INSTALL_PROGRAM) $(BINDIR)/$(SOBINRELDIR)/$(GS_SONAME_MAJOR_MINOR) $(DESTDIR)$(libdir)/$(GS_SONAME_MAJOR_MINOR)
+	$(RM_) $(DESTDIR)$(libdir)/$(GS_SONAME)
+	ln -s $(GS_SONAME_MAJOR_MINOR) $(DESTDIR)$(libdir)/$(GS_SONAME)
+	$(RM_) $(DESTDIR)$(libdir)/$(GS_SONAME_MAJOR)
+	ln -s $(GS_SONAME_MAJOR_MINOR) $(DESTDIR)$(libdir)/$(GS_SONAME_MAJOR)
 
 soinstall: install-so install-scripts install-data
 
@@ -141,8 +141,8 @@
 	cp -r doc $(GS_FRAMEWORK)/Versions/Current
 
 framework_install : framework
-	rm -rf $(prefix)
-	cp -r $(GS_FRAMEWORK) $(prefix)
+	rm -rf $(DESTDIR)$(prefix)
+	cp -r $(GS_FRAMEWORK) $(DESTDIR)$(prefix)
 
 # Make the build directories
 SODIRS: STDDIRS
Only in ghostscript-8.54/src: macos-fw.mak.orig
diff -ur ghostscript-8.54.orig/src/unix-dll.mak ghostscript-8.54/src/unix-dll.mak
--- ghostscript-8.54.orig/src/unix-dll.mak	2006-05-23 19:59:45.000000000 +0200
+++ ghostscript-8.54/src/unix-dll.mak	2006-05-23 20:32:49.000000000 +0200
@@ -93,19 +93,19 @@
 	$(MAKE) $(SODEFS) GENOPT='-DDEBUG' CFLAGS='$(CFLAGS_DEBUG) $(CFLAGS_SO) $(GCFLAGS) $(XCFLAGS)' $(GSSOC) $(GSSOX)
 
 install-so: so
-	-mkdir -p $(prefix)
-	-mkdir -p $(datadir)
-	-mkdir -p $(gsdir)
-	-mkdir -p $(gsdatadir)
-	-mkdir -p $(bindir)
-	-mkdir -p $(libdir)
-	$(INSTALL_PROGRAM) $(GSSOC) $(bindir)/$(GSSOC_XENAME)
-	$(INSTALL_PROGRAM) $(GSSOX) $(bindir)/$(GSSOX_XENAME)
-	$(INSTALL_PROGRAM) $(BINDIR)/$(SOBINRELDIR)/$(GS_SONAME_MAJOR_MINOR) $(libdir)/$(GS_SONAME_MAJOR_MINOR)
-	$(RM_) $(libdir)/$(GS_SONAME)
-	ln -s $(GS_SONAME_MAJOR_MINOR) $(libdir)/$(GS_SONAME)
-	$(RM_) $(libdir)/$(GS_SONAME_MAJOR)
-	ln -s $(GS_SONAME_MAJOR_MINOR) $(libdir)/$(GS_SONAME_MAJOR)
+	-mkdir -p $(DESTDIR)$(prefix)
+	-mkdir -p $(DESTDIR)$(datadir)
+	-mkdir -p $(DESTDIR)$(gsdir)
+	-mkdir -p $(DESTDIR)$(gsdatadir)
+	-mkdir -p $(DESTDIR)$(bindir)
+	-mkdir -p $(DESTDIR)$(libdir)
+	$(INSTALL_PROGRAM) $(GSSOC) $(DESTDIR)$(bindir)/$(GSSOC_XENAME)
+	$(INSTALL_PROGRAM) $(GSSOX) $(DESTDIR)$(bindir)/$(GSSOX_XENAME)
+	$(INSTALL_PROGRAM) $(BINDIR)/$(SOBINRELDIR)/$(GS_SONAME_MAJOR_MINOR) $(DESTDIR)$(libdir)/$(GS_SONAME_MAJOR_MINOR)
+	$(RM_) $(DESTDIR)$(libdir)/$(GS_SONAME)
+	ln -s $(GS_SONAME_MAJOR_MINOR) $(DESTDIR)$(libdir)/$(GS_SONAME)
+	$(RM_) $(DESTDIR)$(libdir)/$(GS_SONAME_MAJOR)
+	ln -s $(GS_SONAME_MAJOR_MINOR) $(DESTDIR)$(libdir)/$(GS_SONAME_MAJOR)
 
 soinstall: install-so install-scripts install-data
 
Only in ghostscript-8.54/src: unix-dll.mak.orig
diff -ur ghostscript-8.54.orig/src/unixinst.mak ghostscript-8.54/src/unixinst.mak
--- ghostscript-8.54.orig/src/unixinst.mak	2006-05-23 19:59:45.000000000 +0200
+++ ghostscript-8.54/src/unixinst.mak	2006-05-23 20:39:35.000000000 +0200
@@ -25,17 +25,17 @@
 # rules, just in case bindir or scriptdir is a subdirectory of any of these.
 
 install-exec: STDDIRS $(GS_XE)
-	-mkdir -p $(datadir)
-	-mkdir -p $(gsdir)
-	-mkdir -p $(gsdatadir)
-	-mkdir -p $(bindir)
-	$(INSTALL_PROGRAM) $(GS_XE) $(bindir)/$(GS)
+	-mkdir -p $(DESTDIR)$(datadir)
+	-mkdir -p $(DESTDIR)$(gsdir)
+	-mkdir -p $(DESTDIR)$(gsdatadir)
+	-mkdir -p $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) $(GS_XE) $(DESTDIR)$(bindir)/$(GS)
 
 install-scripts: $(PSLIBDIR)/gsnd
-	-mkdir -p $(datadir)
-	-mkdir -p $(gsdir)
-	-mkdir -p $(gsdatadir)
-	-mkdir -p $(scriptdir)
+	-mkdir -p $(DESTDIR)$(datadir)
+	-mkdir -p $(DESTDIR)$(gsdir)
+	-mkdir -p $(DESTDIR)$(gsdatadir)
+	-mkdir -p $(DESTDIR)$(scriptdir)
 	$(SH) -c 'for f in \
 gsbj gsdj gsdj500 gslj gslp gsnd \
 bdftops dumphint dvipdf eps2eps font2c \
@@ -44,7 +44,7 @@
 fixmswrd.pl lprsetup.sh pj-gs.sh pv.sh sysvlp.sh unix-lpr.sh ;\
 	do if ( test -f $(PSLIBDIR)/$$f ); then \
 	  (cat $(PSLIBDIR)/$$f | sed -e "s/GS_EXECUTABLE=[^ \t]*/GS_EXECUTABLE=$(GS)/" > $(PSOBJDIR)/$$f); \
-	  $(INSTALL_PROGRAM) $(PSOBJDIR)/$$f $(scriptdir)/$$f; \
+	  $(INSTALL_PROGRAM) $(PSOBJDIR)/$$f $(DESTDIR)$(scriptdir)/$$f; \
 	fi;\
 	done'
 
@@ -59,10 +59,10 @@
 # one file from each subdirectory just as a sanity check.
 
 install-libdata: 
-	-mkdir -p $(datadir)
-	-mkdir -p $(gsdir)
-	-mkdir -p $(gsdatadir)
-	-mkdir -p $(gsdatadir)/lib
+	-mkdir -p $(DESTDIR)$(datadir)
+	-mkdir -p $(DESTDIR)$(gsdir)
+	-mkdir -p $(DESTDIR)$(gsdatadir)
+	-mkdir -p $(DESTDIR)$(gsdatadir)/lib
 	$(SH) -c 'for f in \
 $(EXTRA_INIT_FILES) Fontmap.GS \
 ht_ccsto.ps \
@@ -78,24 +78,24 @@
 viewpcx.ps viewpbm.ps viewps2a.ps \
 winmaps.ps wftopfa.ps wrfont.ps zeroline.ps \
 pdf2dsc.ps pdfopt.ps ;\
-	do if ( test -f $(PSLIBDIR)/$$f ); then $(INSTALL_DATA) $(PSLIBDIR)/$$f $(gsdatadir)/lib; fi;\
+	do if ( test -f $(PSLIBDIR)/$$f ); then $(INSTALL_DATA) $(PSLIBDIR)/$$f $(DESTDIR)$(gsdatadir)/lib; fi;\
 	done'
 	$(SH) -c 'for f in $(PSLIBDIR)/gs_*.ps $(PSLIBDIR)/pdf*.ps;\
-	do $(INSTALL_DATA) $$f $(gsdatadir)/lib ;\
+	do $(INSTALL_DATA) $$f $(DESTDIR)$(gsdatadir)/lib ;\
 	done'
 	$(SH) -c 'for f in $(PSLIBDIR)/*.ppd $(PSLIBDIR)/*.rpd $(PSLIBDIR)/*.upp $(PSLIBDIR)/*.xbm $(PSLIBDIR)/*.xpm;\
-	do $(INSTALL_DATA) $$f $(gsdatadir)/lib ;\
+	do $(INSTALL_DATA) $$f $(DESTDIR)$(gsdatadir)/lib ;\
 	done'
 
 # install the default resource files
 # copy in every category (directory) but CVS
 RES_CATEGORIES=`ls $(PSRESDIR) | grep -v CVS` 
 install-resdata: $(PSRESDIR)/Decoding/Unicode
-	-mkdir -p $(datadir)
-	-mkdir -p $(gsdir)
-	-mkdir -p $(gsdatadir)/Resource
+	-mkdir -p $(DESTDIR)$(datadir)
+	-mkdir -p $(DESTDIR)$(gsdir)
+	-mkdir -p $(DESTDIR)$(gsdatadir)/Resource
 	$(SH) -c 'for dir in $(RES_CATEGORIES); do \
-	  rdir=$(gsdatadir)/Resource/$$dir ; \
+	  rdir=$(DESTDIR)$(gsdatadir)/Resource/$$dir ; \
 	  test -d $$rdir || mkdir -p $$rdir ; \
 	  for file in $(PSRESDIR)/$$dir/*; do \
 	    if test -f $$file; then $(INSTALL_DATA) $$file $$rdir ; fi \
@@ -119,9 +119,9 @@
 	   Source.htm Testing.htm Unix-lpr.htm \
 	   Use.htm Xfonts.htm
 install-doc: $(PSDOCDIR)/News.htm
-	-mkdir -p $(docdir)
+	-mkdir -p $(DESTDIR)$(docdir)
 	$(SH) -c 'for f in $(DOC_PAGES) ;\
-	do if ( test -f $(PSDOCDIR)/$$f ); then $(INSTALL_DATA) $(PSDOCDIR)/$$f $(docdir); fi;\
+	do if ( test -f $(PSDOCDIR)/$$f ); then $(INSTALL_DATA) $(PSDOCDIR)/$$f $(DESTDIR)$(docdir); fi;\
 	done'
 
 # install the man pages for each locale
@@ -130,9 +130,9 @@
 MAN1_LINKS_PS2PDF=ps2pdf12 ps2pdf13
 MAN1_LINKS_GSLP=gsbj gsdj gsdj500 gslj
 install-man: $(PSMANDIR)/gs.1
-	$(SH) -c 'test -d $(mandir) || mkdir -p $(mandir)'
+	$(SH) -c 'test -d $(DESTDIR)$(mandir) || mkdir -p $(DESTDIR)$(mandir)'
 	$(SH) -c 'for d in $(MAN_LCDIRS) ;\
-	do man1dir=$(mandir)/$$d/man$(man1ext) ;\
+	do man1dir=$(DESTDIR)$(mandir)/$$d/man$(man1ext) ;\
 	  ( test -d $$man1dir || mkdir -p $$man1dir ) ;\
 	  for f in $(PSMANDIR)/$$d/*.1 ;\
 	    do $(INSTALL_DATA) $$f $$man1dir ;\
@@ -159,10 +159,10 @@
 
 # install the example files
 install-examples:
-	-mkdir -p $(exdir)
+	-mkdir -p $(DESTDIR)$(exdir)
 	for f in \
 alphabet.ps annots.pdf chess.ps colorcir.ps doretree.ps escher.ps \
 golfer.eps grayalph.ps snowflak.ps tiger.eps vasarely.ps waterfal.ps \
 ridt91.eps ;\
-	do $(INSTALL_DATA) $(PSEXDIR)/$$f $(exdir) ;\
+	do $(INSTALL_DATA) $(PSEXDIR)/$$f $(DESTDIR)$(exdir) ;\
 	done
Only in ghostscript-8.54/src: unixinst.mak.orig
Only in ghostscript-8.54/src: unixinst.mak.rej
