--- ijs/Makefile.in	2003-05-29 18:38:04.000000000 +0200
+++ ijs/Makefile.in	2003-12-06 16:52:37.000000000 +0100
@@ -7,7 +7,7 @@
 
 # todo: this needs to be .dylib on OS X - write a test
 SHARED_OBJ=.so
-SHARED_OBJ=.dylib
+#SHARED_OBJ=.dylib
 
 EXE=
 FE=-o 
@@ -43,8 +43,8 @@
 # makes sense to separate them, but they're small enough to make this probably
 # not worthwhile.
 libijs$(SHARED_OBJ):	$(LIB_OBJS)
-	#$(CC) -shared $^ -o $@
-	$(CC) -dynamic $^ -o $@
+	$(CC) -shared $^ -o $@
+	#$(CC) -dynamic $^ -o $@
 
 ijs_client_example$(EXE):	ijs_client_example$(OBJ) ijs_client$(OBJ) $(IJS_COMMON_OBJ) $(IJS_EXEC_SERVER)
 	$(CC) $(CFLAGS) $(FE)ijs_client_example$(EXE) ijs_client_example$(OBJ) ijs_client$(OBJ) $(IJS_COMMON_OBJ) $(IJS_EXEC_SERVER) $(LDLIBS)
