diff -ur schroedinger-0.9.0.orig/schroedinger/Makefile.in schroedinger-0.9.0/schroedinger/Makefile.in
--- schroedinger-0.9.0.orig/schroedinger/Makefile.in	2007-11-07 23:53:14.000000000 +0200
+++ schroedinger-0.9.0/schroedinger/Makefile.in	2008-01-26 10:28:13.000000000 +0200
@@ -339,7 +339,7 @@
 	$(SCHRO_CFLAGS) $(LIBOIL_CFLAGS) $(ERROR_CFLAGS) \
 	-DSCHRO_ENABLE_UNSTABLE_API
 
-libschroedinger_@SCHRO_MAJORMINOR@_la_LIBADD = $(LIBOIL_LIBS)
+libschroedinger_@SCHRO_MAJORMINOR@_la_LIBADD = -lpthread $(LIBOIL_LIBS)
 libschroedinger_@SCHRO_MAJORMINOR@_la_LDFLAGS = \
 	-no-undefined \
 	-version-info $(SCHRO_LIBVERSION) \
