diff -Nur gimageview-0.2.27-bak/src/gimv_thumb_win.c gimageview-0.2.27/src/gimv_thumb_win.c
--- gimageview-0.2.27-bak/src/gimv_thumb_win.c	2004-12-21 03:18:30.000000000 +0900
+++ gimageview-0.2.27/src/gimv_thumb_win.c	2005-01-07 13:46:40.805458176 +0900
@@ -2542,7 +2542,7 @@
 {
    if (tw && tw->priv)
       tw->priv->slideshow_selected_only
-         = gtk_check_menu_item_get_active (GTK_CHECK_MENU_ITEM (widget));
+         = GTK_CHECK_MENU_ITEM (widget)->active;
 }
 
 
