diff -ur gimageview-0.2.25/work/gimageview-0.2.25/src/gimv_thumb_win.c gimageview-0.2.25-fixed/work/gimageview-0.2.25/src/gimv_thumb_win.c
--- gimageview-0.2.25/work/gimageview-0.2.25/src/gimv_thumb_win.c	2003-07-06 09:46:21.000000000 -0700
+++ gimageview-0.2.25-fixed/work/gimageview-0.2.25/src/gimv_thumb_win.c	2004-06-11 22:35:00.516094744 -0700
@@ -202,7 +202,7 @@
                                      guint          action,
                                      GtkWidget     *widget);
 static void cb_move_tab             (GimvThumbWin  *tw,
-                                     GimvSortItem   sortitem,
+                                     MoveTabItem    item,
                                      GtkWidget     *widget);
 static void cb_cut_out_tab          (GimvThumbWin  *tw,
                                      guint          action,
