--- asterisk-oh323-0.6.6/Makefile.orig	2005-06-17 12:14:01.000000000 +0200
+++ asterisk-oh323-0.6.6/Makefile	2005-07-08 03:20:13.000000000 +0200
@@ -51,7 +51,7 @@
 # compile asterisk-oh323. Do not use the installed library of your
 # distribution. It won't work.
 #
-PWLIBDIR=/root/src/oh323/pwlib
+PWLIBDIR=/usr/share/pwlib
 
 #
 # Set OPENH323LIBDIR variable to the directory containing the sources of
@@ -61,13 +61,13 @@
 # compile asterisk-oh323. Do not use the installed library of your
 # distribution. It won't work.
 #
-OPENH323DIR=/root/src/oh323/openh323
+OPENH323DIR=/usr/share/openh323
 
 #
 # Set ASTERISKINCDIR variable to the directory containing the include files of
 # Asterisk PBX.
 #
-ASTERISKINCDIR=/root/src/asterisk/include
+ASTERISKINCDIR=/usr/include
 
 #
 # Set ASTERISKMODDIR variable to the directory where ASTERISK's modules reside.
@@ -93,7 +93,7 @@
 #
 # Note: This directory is created under $DESTDIR.
 #
-OH323WRAPLIBDIR=/usr/local/lib
+OH323WRAPLIBDIR=/usr/lib
 
 #
 # Set WRAPTRACING to 1 to enable tracing information from the OpenH323Wrap
@@ -120,7 +120,7 @@
 # With this flag on the channel driver is huge, in size, but Asterisk boots
 # faster!
 #
-OH323STAT=1
+OH323STAT=0
 
 #
 # Define options that were used during Asterisk compilation. It is
@@ -142,7 +142,7 @@
 #NOVIDEO=1
 #NO_SPEEX=1
 #NOAUDIOCODECS=1
-#NOTRACE=1
+NOTRACE=1
 NO_IPv6=1
 #P_PTHREADS=1
 
--- asterisk-oh323-0.6.6/rules.mak.orig	2005-07-08 03:19:20.000000000 +0200
+++ asterisk-oh323-0.6.6/rules.mak	2005-07-08 03:21:34.000000000 +0200
@@ -50,11 +50,9 @@
 AR = ar
 STRIP = strip
 STRIPFLAGS = 
-OH323WRAPINCLUDE = \
-				 -I$(PWLIBDIR)/include \
-				 -I$(OPENH323DIR)/include \
-				 -I$(OPENH323DIR)/include/openh323 \
-				 -I../asterisk-driver
+OH323WRAPINCLUDE=-I$(SSLINCDIR) \
+		 -I/usr/include/openh323 \
+		 -I../asterisk-driver
 ASTERISKINCLUDE = -I$(ASTERISKINCDIR) -I../wrapper
 
 ifdef	NOTRACE
--- asterisk-oh323-0.6.6/asterisk-driver/Makefile.orig	2005-06-17 12:14:02.000000000 +0200
+++ asterisk-oh323-0.6.6/asterisk-driver/Makefile	2005-07-08 03:27:38.000000000 +0200
@@ -34,23 +34,23 @@
 ifndef SKIP_SUFFIX
 	ifeq ($(OH323STAT),1)
 		ifeq ($(OH323BUILDTYPE),debug) # Static, debug
-			LIBPTPATH=$(word 1,$(shell ls $(PWLIBDIR)/lib/libpt*_d_s.a))
+			LIBPTPATH=$(word 1,$(shell ls /usr/lib/libpt*_d_s.a))
 		else                           # Static, opt
-			LIBPTPATH=$(word 1,$(shell ls $(PWLIBDIR)/lib/libpt*_r_s.a))
+			LIBPTPATH=$(word 1,$(shell ls /usr/lib/libpt*_r_s.a))
 		endif
 	else
 		ifeq ($(OH323BUILDTYPE),debug) # Shared, debug
-			LIBPTPATH=$(word 1,$(shell ls $(PWLIBDIR)/lib/libpt*d.so))
+			LIBPTPATH=$(word 1,$(shell ls /usr/lib/libpt*d.so))
 		else                           # Shared, opt
-			LIBPTPATH=$(word 1,$(shell ls $(PWLIBDIR)/lib/libpt*r.so))
+			LIBPTPATH=$(word 1,$(shell ls /usr/lib/libpt*r.so))
 		endif
 	endif
 	LIBPTNAME=$(subst lib,,$(basename $(notdir $(LIBPTPATH))))
 else
 	ifeq ($(OH323STAT),1)
-		LIBPTPATH=$(word 1,$(shell ls $(PWLIBDIR)/lib/$(LIBPT_BASE).a))
+		LIBPTPATH=$(word 1,$(shell ls /usr/lib/$(LIBPT_BASE).a))
 	else
-		LIBPTPATH=$(word 1,$(shell ls $(PWLIBDIR)/lib/$(LIBPT_BASE).so))
+		LIBPTPATH=$(word 1,$(shell ls /usr/lib/$(LIBPT_BASE).so))
 	endif
 	LIBPTNAME=$(subst lib,,$(basename $(notdir $(LIBPTPATH))))
 endif
@@ -59,12 +59,12 @@
 ifndef SKIP_SUFFIX
 	ifeq ($(OH323STAT),1)
 		ifeq ($(OH323BUILDTYPE),debug)   # Static, debug
-			LIBH323PATH=$(word 1,$(shell ls $(OPENH323DIR)/lib/libh323*_d_s.a))
+			LIBH323PATH=$(word 1,$(shell ls /usr/lib/libh323*_d_s.a))
 		else
 			ifeq ($(OH323BUILDTYPE),opt) # Static, opt
-				LIBH323PATH=$(word 1,$(shell ls $(OPENH323DIR)/lib/libh323*_r_s.a))
+				LIBH323PATH=$(word 1,$(shell ls /usr/lib/libh323*_r_s.a))
 			else                         # Static, opt, no trace
-				LIBH323PATH=$(word 1,$(shell ls $(OPENH323DIR)/lib/libh323*_n_s.a))
+				LIBH323PATH=$(word 1,$(shell ls /usr/lib/libh323*_n_s.a))
 			endif
 		endif
 	else
@@ -72,18 +72,18 @@
 			LIBH323PATH=$(word 1,$(shell ls $(OPENH323DIR)/lib/libh323*d.so))
 		else
 			ifeq ($(OH323BUILDTYPE),opt) # Shared, opt
-				LIBH323PATH=$(word 1,$(shell ls $(OPENH323DIR)/lib/libh323*r.so))
+				LIBH323PATH=$(word 1,$(shell ls /usr/lib/libh323*r.so))
 			else                         # Shared, opt, no trace
-				LIBH323PATH=$(word 1,$(shell ls $(OPENH323DIR)/lib/libh323*n.so))
+				LIBH323PATH=$(word 1,$(shell ls /usr/lib/libh323*n.so))
 			endif
 		endif
 	endif
 	LIBH323NAME=$(subst lib,,$(basename $(notdir $(LIBH323PATH))))
 else
 	ifeq ($(OH323STAT),1)
-		LIBH323PATH=$(word 1,$(shell ls $(OPENH323DIR)/lib/$(LIBH323_BASE).a))
+		LIBH323PATH=$(word 1,$(shell ls /usr/lib/$(LIBH323_BASE).a))
 	else
-		LIBH323PATH=$(word 1,$(shell ls $(OPENH323DIR)/lib/$(LIBH323_BASE).so))
+		LIBH323PATH=$(word 1,$(shell ls /usr/lib/$(LIBH323_BASE).so))
 	endif
 	LIBH323NAME=$(subst lib,,$(basename $(notdir $(LIBH323PATH))))
 endif
@@ -153,14 +153,12 @@
 ifeq ($(OH323STAT),1)
 	$(CC) -shared -Xlinker -x -g -o $@ $(TARGET_OBJ) \
 	-L../wrapper -loh323wrap_s \
-	-L$(OPENH323DIR)/lib -l$(LIBH323) \
-	-L$(PWLIBDIR)/lib -l$(LIBPT) \
+	-l$(LIBH323) -l$(LIBPT) \
 	-lstdc++ $(EXTLIBS)
 else
 	$(CC) -shared -Xlinker -x -g -o $@ $(TARGET_OBJ) \
 	-L../wrapper -loh323wrap \
-	-L$(OPENH323DIR)/lib -l$(LIBH323) \
-	-L$(PWLIBDIR)/lib -l$(LIBPT) \
+	-l$(LIBH323) -l$(LIBPT) \
 	-lstdc++ $(EXTLIBS)
 endif
 
