--- linux/include/linux/evms/evms_common.h	Wed Sep  4 23:42:32 2002
+++ linux/include/linux/evms/evms_common.h.new	Sun Sep  8 11:56:53 2002
@@ -34,8 +34,8 @@
 #define EVMS_MINOR_VERSION              1
 #define EVMS_PATCHLEVEL_VERSION         0
 
-/* We are using the O(1) Scheduler */
-#define O1_SCHEDULER
+/* We are not using the O(1) Scheduler */
+/* #define O1_SCHEDULER */
 
 #define MAX_EVMS_VOLUMES                256	/* There are 256 minors */
 #define EVMS_VOLUME_NAME_SIZE           127
