Index: branches/KDE/4.3/kdeutils/okteta/program/libkakao/core/system/kdocumentsyncmanager.cpp
===================================================================
--- branches/KDE/4.3/kdeutils/okteta/program/libkakao/core/system/kdocumentsyncmanager.cpp	(revision 1046928)
+++ branches/KDE/4.3/kdeutils/okteta/program/libkakao/core/system/kdocumentsyncmanager.cpp	(revision 1046929)
@@ -217,9 +217,9 @@
                 if( synchronizer )
                 {
                     AbstractSyncToRemoteJob *syncJob = synchronizer->startSyncToRemote();
-                    const bool syncFailed = JobManager::executeJob( syncJob, mWidget );
+                    const bool isSynced = JobManager::executeJob( syncJob, mWidget );
 
-                    canClose = !syncFailed;
+                    canClose = isSynced;
                 }
                 else
                     canClose = setSynchronizer( document );
