--- Makefile.in.orig	2009-02-17 21:45:45.469037983 +0000
+++ Makefile.in	2009-02-17 22:01:42.960921967 +0000
@@ -59,7 +59,6 @@
 DEBUG = 
 C_OPT = -O
 SYS = 
-CFLAGS = $(C_OPT) $(DEBUG) $(SYS) -I.
 
 # FILES		--------------------------------
 VERFILE = $(PREFIX)/versions
@@ -91,7 +90,7 @@
 # RULES		--------------------------------
 .SUFFIXES: .o .c 
 .c.o:
-	$(CC) $(CFLAGS) -c $<
+	$(CC) $(CFLAGS) -I. -c $<
 
 # DEPENDENCIES	--------------------------------
 
@@ -110,17 +109,15 @@
 		rm -f $$f; ln finducac2 $$f; done
 
 aclient: aclient.o $(OBJ)
-	$(CC) $@.o $(OBJ) $(LIBSYS) -o $@
-	$(STRIP) $@
+	$(CC) $(LDFLAGS) $@.o $(OBJ) $(LIBSYS) -o $@
 
 wwwget: wwwget.c
-	$(CC) wwwget.c $(LIBSYS) -o $@
-	$(STRIP) $@
+	$(CC) $(LDFLAGS) wwwget.c $(LIBSYS) -o $@
 
 doc: sk.tex man.tex aclient.tex
 	latex man.tex
 
-install: $(EXPORT_INSTALL) install_shs install_info
+install: $(EXPORT_INSTALL) install_shs
 	rm -rf *.cache
 
 export_install: install_bin install_man 
