diff -ur src/globalreplacedialog.cpp src/globalreplacedialog.cpp
--- src/globalreplacedialog.cpp	2008-01-20 20:14:24.000000000 +0000
+++ src/globalreplacedialog.cpp	2008-07-04 15:03:47.000000000 +0100
@@ -17,6 +17,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include <memory>
 #include "globalreplacedialog.h"
 #include "wrapregex.h"
 
diff -ur src/housestyle.cpp src/housestyle.cpp
--- src/housestyle.cpp	2008-01-20 20:14:24.000000000 +0000
+++ src/housestyle.cpp	2008-07-04 15:03:47.000000000 +0100
@@ -17,6 +17,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include <cstring>
 #include "housestyle.h"
 #include "readfile.h"
 
diff -ur src/housestylereader.h src/housestylereader.h
--- src/housestylereader.h	2008-01-20 20:14:30.000000000 +0000
+++ src/housestylereader.h	2008-07-04 15:03:47.000000000 +0100
@@ -24,6 +24,7 @@
 #include <map>
 #include <set>
 #include <vector>
+#include <memory>
 #include <expat.h>
 #include "wrapexpat.h"
 
diff -ur src/housestylewriter.h src/housestylewriter.h
--- src/housestylewriter.h	2008-01-20 20:14:30.000000000 +0000
+++ src/housestylewriter.h	2008-07-04 15:03:47.000000000 +0100
@@ -22,6 +22,7 @@
 
 #include <set>
 #include <string>
+#include <memory>
 #include <expat.h>
 #include "wrapexpat.h"
 #include "wrapregex.h"
diff -ur src/stringset.h src/stringset.h
--- src/stringset.h	2008-01-20 20:14:30.000000000 +0000
+++ src/stringset.h	2008-07-04 15:03:47.000000000 +0100
@@ -24,6 +24,7 @@
 #include <stdexcept>
 #include <cmath>
 #include <climits>
+#include <cstring>
 #ifdef __WXMSW__
 #include <mem.h>
 #endif
diff -ur src/wrapregex.cpp src/wrapregex.cpp
--- src/wrapregex.cpp	2008-01-20 20:14:24.000000000 +0000
+++ src/wrapregex.cpp	2008-07-04 15:03:47.000000000 +0100
@@ -21,6 +21,7 @@
 #include <string>
 #include <ctype.h>
 #include <stdexcept>
+#include <cstring>
 #include "wrapregex.h"
 #include "contexthandler.h"
 
diff -ur src/xmlassociatedtd.h src/xmlassociatedtd.h
--- src/xmlassociatedtd.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlassociatedtd.h	2008-07-04 15:03:47.000000000 +0100
@@ -23,6 +23,7 @@
 #include <vector>
 #include <expat.h>
 #include <string>
+#include <memory>
 #include "wrapexpat.h"
 
 struct DtdData : public ParserData
diff -ur src/xmlassociatexsd.cpp src/xmlassociatexsd.cpp
--- src/xmlassociatexsd.cpp	2008-01-20 20:14:24.000000000 +0000
+++ src/xmlassociatexsd.cpp	2008-07-04 15:03:47.000000000 +0100
@@ -21,6 +21,7 @@
 #include <vector>
 #include <stdexcept>
 #include <expat.h>
+#include <cstring>
 #include "xmlassociatexsd.h"
 #include "xmlparseschemans.h"
 #include "readfile.h"
diff -ur src/xmlassociatexsd.h src/xmlassociatexsd.h
--- src/xmlassociatexsd.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlassociatexsd.h	2008-07-04 15:03:47.000000000 +0100
@@ -24,6 +24,7 @@
 #include <expat.h>
 #include <string>
 #include <map>
+#include <memory>
 #include "wrapexpat.h"
 
 struct AssociateXsdData : public ParserData
diff -ur src/xmlassociatexsl.cpp src/xmlassociatexsl.cpp
--- src/xmlassociatexsl.cpp	2008-01-20 20:14:24.000000000 +0000
+++ src/xmlassociatexsl.cpp	2008-07-04 15:03:47.000000000 +0100
@@ -21,6 +21,7 @@
 #include <vector>
 #include <stdexcept>
 #include <expat.h>
+#include <cstring>
 #include "xmlassociatexsl.h"
 
 XmlAssociateXsl::XmlAssociateXsl ( const std::string& path, size_t size ) :
diff -ur src/xmlassociatexsl.h src/xmlassociatexsl.h
--- src/xmlassociatexsl.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlassociatexsl.h	2008-07-04 15:03:47.000000000 +0100
@@ -23,6 +23,7 @@
 #include <vector>
 #include <expat.h>
 #include <string>
+#include <memory>
 #include "wrapexpat.h"
 
 struct XslData : public ParserData
diff -ur src/xmlencodinghandler.cpp src/xmlencodinghandler.cpp
--- src/xmlencodinghandler.cpp	2008-01-20 20:14:24.000000000 +0000
+++ src/xmlencodinghandler.cpp	2008-07-04 15:03:47.000000000 +0100
@@ -17,6 +17,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include <cstring>
 #include "xmlencodinghandler.h"
 
 bool XmlEncodingHandler::hasDeclaration ( const std::string& utf8 )
diff -ur src/xmlencodingspy.h src/xmlencodingspy.h
--- src/xmlencodingspy.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlencodingspy.h	2008-07-04 15:03:47.000000000 +0100
@@ -22,6 +22,7 @@
 
 #include <expat.h>
 #include <string>
+#include <memory>
 #include "wrapexpat.h"
 
 struct EncodingData : public ParserData
diff -ur src/xmlfilterreader.cpp src/xmlfilterreader.cpp
--- src/xmlfilterreader.cpp	2008-01-20 20:14:24.000000000 +0000
+++ src/xmlfilterreader.cpp	2008-07-04 15:03:47.000000000 +0100
@@ -23,6 +23,7 @@
 #include <expat.h>
 #include <map>
 #include <set>
+#include <cstring>
 #include "xmlfilterreader.h"
 
 using namespace std;
diff -ur src/xmlfilterreader.h src/xmlfilterreader.h
--- src/xmlfilterreader.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlfilterreader.h	2008-07-04 15:03:47.000000000 +0100
@@ -24,6 +24,7 @@
 #include <vector>
 #include <map>
 #include <set>
+#include <memory>
 #include <expat.h>
 #include "wrapexpat.h"
 
diff -ur src/xmlparseschemans.h src/xmlparseschemans.h
--- src/xmlparseschemans.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlparseschemans.h	2008-07-04 15:03:47.000000000 +0100
@@ -23,6 +23,7 @@
 #include <vector>
 #include <expat.h>
 #include <string>
+#include <memory>
 #include "wrapexpat.h"
 
 struct ParseSchemaNsData : public ParserData
diff -ur src/xmlpromptgenerator.h src/xmlpromptgenerator.h
--- src/xmlpromptgenerator.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlpromptgenerator.h	2008-07-04 15:03:47.000000000 +0100
@@ -23,6 +23,7 @@
 #include <expat.h>
 #include <map>
 #include <set>
+#include <memory>
 #include "wrapexpat.h"
 #include "parserdata.h"
 
diff -ur src/xmlschemalocator.cpp src/xmlschemalocator.cpp
--- src/xmlschemalocator.cpp	2008-01-20 20:14:24.000000000 +0000
+++ src/xmlschemalocator.cpp	2008-07-04 15:03:47.000000000 +0100
@@ -20,6 +20,7 @@
 #include <string>
 #include <vector>
 #include <stdexcept>
+#include <cstring>
 #include <expat.h>
 #include "xmlschemalocator.h"
 
diff -ur src/xmlschemalocator.h src/xmlschemalocator.h
--- src/xmlschemalocator.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlschemalocator.h	2008-07-04 15:03:47.000000000 +0100
@@ -22,6 +22,7 @@
 
 #include <expat.h>
 #include <string>
+#include <memory>
 #include "wrapexpat.h"
 
 struct SchemaLocatorData
diff -ur src/xmlshallowvalidator.h src/xmlshallowvalidator.h
--- src/xmlshallowvalidator.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlshallowvalidator.h	2008-07-04 15:03:47.000000000 +0100
@@ -25,6 +25,7 @@
 #include <set>
 #include <vector>
 #include <utility>
+#include <memory>
 #include <expat.h>
 #include "wrapexpat.h"
 
diff -ur src/xmlutf8reader.h src/xmlutf8reader.h
--- src/xmlutf8reader.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlutf8reader.h	2008-07-04 15:03:47.000000000 +0100
@@ -23,6 +23,7 @@
 #include <vector>
 #include <expat.h>
 #include <string>
+#include <memory>
 #include "wrapexpat.h"
 
 struct UtfData : public ParserData
diff -ur src/xmlwordcount.h src/xmlwordcount.h
--- src/xmlwordcount.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xmlwordcount.h	2008-07-04 15:03:47.000000000 +0100
@@ -21,6 +21,7 @@
 #define XML_WORD_COUNT_H
 
 #include <string>
+#include <memory>
 #include "wrapexpat.h"
 
 struct WordCountData : public ParserData
diff -ur src/xsllocator.cpp src/xsllocator.cpp
--- src/xsllocator.cpp	2008-01-20 20:14:24.000000000 +0000
+++ src/xsllocator.cpp	2008-07-04 15:03:47.000000000 +0100
@@ -20,6 +20,7 @@
 #include <string>
 #include <vector>
 #include <stdexcept>
+#include <cstring>
 #include <expat.h>
 #include "xsllocator.h"
 
diff -ur src/xsllocator.h src/xsllocator.h
--- src/xsllocator.h	2008-01-20 20:14:30.000000000 +0000
+++ src/xsllocator.h	2008-07-04 15:03:47.000000000 +0100
@@ -22,6 +22,7 @@
 
 #include <expat.h>
 #include <string>
+#include <memory>
 #include "wrapexpat.h"
 
 struct XslLocatorData : public ParserData
