http://bugs.gentoo.org/320731

--- man/Makefile.am
+++ man/Makefile.am
@@ -5,9 +5,7 @@
 	pm-powersave.8		\
 	pm-action.8
 
-if HAVE_XMLTO
 man_MANS = $(pm_mans)
-endif
 
 EXTRA_DIST =			\
 	on_ac_power.xml		\
@@ -20,11 +18,13 @@
 MAINTAINERCLEANFILES =		\
 	$(pm_mans)
 
+if HAVE_XMLTO
 .xml.1:
 	$(XMLTO) man $<
 
 .xml.8:
 	$(XMLTO) man $<
+endif
 
 install-data-hook: install-man
 	@if test -f $(DESTDIR)$(mandir)/man8/pm-action.8; then \
