diff -ur -x'config*' Gauche-0.8.7.orig/ext/dbm/dbm.ac Gauche-0.8.7/ext/dbm/dbm.ac
--- Gauche-0.8.7.orig/ext/dbm/dbm.ac	2003-07-20 19:06:48.000000000 +0900
+++ Gauche-0.8.7/ext/dbm/dbm.ac	2006-06-06 05:54:22.000000000 +0900
@@ -20,27 +20,23 @@
   NDBM_HEADER=$ac_header
   DBM_ARCHFILES="ndbm.$SHLIB_SO_SUFFIX $DBM_ARCHFILES"
   DBM_SCMFILES="ndbm.scm $DBM_SCMFILES"
+  break
 ])
 AC_CHECK_HEADERS(dbm.h gdbm/dbm.h, [
   ODBM_HEADER=$ac_header
   DBM_ARCHFILES="odbm.$SHLIB_SO_SUFFIX $DBM_ARCHFILES"
   DBM_SCMFILES="odbm.scm $DBM_SCMFILES"
+  break
 ])
 
 AC_CHECK_LIB(gdbm, gdbm_open, [ GDBMLIB="-lgdbm" ])
 LIBSAVE="$LIBS"
-if test "$NDBM_HEADER" = "gdbm/ndbm.h"; then
-  NDBMLIB="-lgdbm"  # use emulation by gdbm
-else
-  AC_SEARCH_LIBS(dbm_open, ndbm gdbm gdbm_compat,
+LIBS=""
+  AC_SEARCH_LIBS(dbm_open, ndbm gdbm "gdbm_compat -lgdbm",
     [ NDBMLIB="$LIBS"; LIBS="$LIBSAVE" ])
-fi
-if test "$ODBM_HEADER" = "gdbm/dbm.h"; then
-  ODBMLIB="-lgdbm"  # use emulation by gdbm
-else
-  AC_SEARCH_LIBS(dbminit,  dbm gdbm gdbm_compat,
+  AC_SEARCH_LIBS(dbminit,  dbm gdbm "gdbm_compat -lgdbm",
     [ ODBMLIB="$LIBS"; LIBS="$LIBSAVE" ])
-fi
+LIBS="$LIBSAVE"
 
 AC_SUBST(DBM_ARCHFILES)
 AC_SUBST(DBM_SCMFILES)
