Subject: [PATCH] Fix the forced debugging mode even if --disable-debug is given on the command line
--- configure.in~	2010-06-05 12:00:40.000000000 +0200
+++ configure.in	2010-06-13 16:56:42.000000000 +0200
@@ -84,7 +84,9 @@
 dnl Enable debugging mode
 AC_ARG_ENABLE(debug,
   AC_HELP_STRING([--enable-debug],[Enable debug messages]),
-  AM_CFLAGS="$AM_CFLAGS -DDEBUG"
+  [if test "x$enableval" = "xyes"; then
+      AM_CFLAGS="$AM_CFLAGS -DDEBUG"
+  fi]
   AM_CXXFLAGS="$AM_CXXFLAGS -DDEBUG")
 AC_SUBST(AM_CFLAGS)
 AC_SUBST(AM_CXXFLAGS)
