--- scintilla/ScintillaGTK.cxx
+++ scintilla/ScintillaGTK.cxx
@@ -2120,8 +2120,8 @@
 extern void Platform_Initialise();
 extern void Platform_Finalise();
 
-guint scintilla_get_type() {
-	static guint scintilla_type = 0;
+GType scintilla_get_type() {
+	static GType scintilla_type = 0;
 
 	if (!scintilla_type) {
 		Platform_Initialise();
--- scintilla/include/ScintillaWidget.h
+++ scintilla/include/ScintillaWidget.h
@@ -34,7 +34,7 @@
 	void (* notify) (ScintillaObject *ttt);
 };
 
-guint		scintilla_get_type	(void);
+GType		scintilla_get_type	(void);
 GtkWidget*	scintilla_new		(void);
 void		scintilla_set_id	(ScintillaObject *sci,int id);
 sptr_t		scintilla_send_message	(ScintillaObject *sci,unsigned int iMessage, uptr_t wParam, sptr_t lParam);
--- src/goto_line.c
+++ src/goto_line.c
@@ -48,10 +48,10 @@
 static void on_go_to_line_response (GtkDialog* dialog, gint response,
                                     gpointer user_data);
 
-guint
+GType
 gotoline_get_type ()
 {
-	static guint obj_type = 0;
+	static GType obj_type = 0;
 	
 	if (!obj_type)
 	{
--- src/goto_line.h
+++ src/goto_line.h
@@ -52,7 +52,7 @@
   };
 
   /** GTK widget implementation function */
-  guint gotoline_get_type (void);
+  GType gotoline_get_type (void);
   
   /** Create a new instance of this gotoline dialog */
   GtkWidget *gotoline_new (void);
--- src/launcher.c
+++ src/launcher.c
@@ -158,10 +158,10 @@
 	obj->priv->callback_data = NULL;
 }
 
-guint
+GType
 anjuta_launcher_get_type ()
 {
-	static guint obj_type = 0;
+	static GType obj_type = 0;
 	
 	if (!obj_type)
 	{
--- src/launcher.h
+++ src/launcher.h
@@ -66,7 +66,7 @@
 	void (*busy_signal) (AnjutaLauncher *launcher, gboolean busy_flag);
 };
 
-guint anjuta_launcher_get_type (void);
+GType anjuta_launcher_get_type (void);
 GObject *anjuta_launcher_new (void);
 gboolean anjuta_launcher_is_busy (AnjutaLauncher *launcher);
 gboolean anjuta_launcher_execute (AnjutaLauncher *launcher,
--- src/message-manager.cc
+++ src/message-manager.cc
@@ -80,7 +80,7 @@
 	return amm;
 }
 
-guint
+GType
 an_message_manager_get_type (void)
 {
 	static GType type = 0;
--- src/message-manager.h
+++ src/message-manager.h
@@ -68,7 +68,7 @@
 
 // Public functions
 GtkWidget* an_message_manager_new();
-guint an_message_manager_get_type();
+GType an_message_manager_get_type();
 
 gboolean an_message_manager_add_type(AnMessageManager* amm, const gint type_name, const gchar* pixmap);
 gboolean an_message_manager_append(AnMessageManager* amm, const gchar* msg_string, gint type_name);
--- src/preferences.c
+++ src/preferences.c
@@ -35,6 +35,8 @@
 
 #include <glade/glade-parser.h>
 
+#include "anjuta.h"
+
 struct _AnjutaProperty
 {
 	GtkWidget                *object;
--- src/compiler_options.c	2004-04-08 07:21:41.000000000 -0500
+++ src/compiler_options.c	2004-08-24 04:04:46.201104024 -0500
@@ -352,7 +352,7 @@
 	gchar *pkg_cmd;
	FILE  *pkg_fd;
	gchar *line;
-	gint   length;
+	size_t   length;
	GtkTreeIter iter;

	int i = 0;
			
