--- CDROD-1.6.3/install.sh	2003-11-11 17:29:05.000000000 -0800
+++ CDROD-1.6.3-gentoo/install.sh	2003-11-14 15:44:22.000000000 -0800
@@ -182,6 +182,14 @@
 	I_DOCDIR=$I_PREFIX/share/doc/$DROD_VERSION
 }
 
+function set_gentoo {
+	I_RESDIR=/opt/drod/bin
+	I_DATDIR=/opt/drod/bin/Data
+	I_LIBDIR=/opt/drod/lib
+	I_DOCDIR=/opt/drod/doc
+	I_BINDIR=/opt/drod/bin
+}
+
 function read_otherdir {
 	eval olddir=\$$2
 	read_dir $1 $2
@@ -612,6 +620,7 @@
 	-d)	shift; get_dir_opt I_DATDIR $1;;
 	-s)	shift; get_dir_opt I_BINDIR $1; set_prefix; set_single;;
 	-u)	shift; get_dir_opt I_BINDIR $1; set_prefix; set_usrstyle;;
+	-g)	set_gentoo;;
 	-al)	set_all_libs;;
 	-nl)	set_no_libs;;
 	-rl)	set_req_libs;;
