--- trunk/extragear/graphics/kipi-plugins/picasawebexport/picasawebtalker.cpp	2009/03/21 20:09:51	942469
+++ trunk/extragear/graphics/kipi-plugins/picasawebexport/picasawebtalker.cpp	2009/03/21 20:11:49	942470
@@ -758,10 +758,7 @@
 
                     if(detailsNode.nodeName() == "title")
                     {
-                        album_title = "Not fetched";
-
-                        if(detailsNode.toElement().attribute("type")=="text")
-                            album_title = detailsNode.toElement().text();
+                        album_title = detailsNode.toElement().text();
 
                         //this is what is obtained from data.
                         fps.title = album_title;
