diff --exclude-from=/home/dang/.diffrc -u -ruN evolution-2.8.0.orig/mail/em-folder-view.c evolution-2.8.0/mail/em-folder-view.c
--- evolution-2.8.0.orig/mail/em-folder-view.c	2006-09-02 02:43:04.000000000 -0400
+++ evolution-2.8.0/mail/em-folder-view.c	2006-09-06 12:24:49.000000000 -0400
@@ -1486,7 +1486,7 @@
 
 	if (gtk_html_command(((EMFormatHTML *)emfv->preview)->html, "is-selection-active")
 	    && (html = gtk_html_get_selection_html (((EMFormatHTML *)emfv->preview)->html, &len))
-	    && len) {
+	    && len && html[0]) {
 		CamelMimeMessage *msg, *src;
 		struct _camel_header_raw *header;
 		
