--- linux-2.4.25_pre7-gss-r6/include/asm-i386/mki.h.orig	2004-06-29 21:24:45.720584416 +0100
+++ linux-2.4.25_pre7-gss-r6/include/asm-i386/mki.h	2004-06-29 21:24:30.119956072 +0100
@@ -37,6 +37,7 @@ extern void (*mhi_hooks[])(void *, void 
 /* Definitions for MKI_HOOK_SWAP */
 #define MKI_HARD_PRIORITY       1
 #define MKI_NORMAL_PRIORITY     6
+#define MM_CONTEXT_HAS_LDT_FIELD        1
 
 #endif /* INCLUDE_MKI_H */
 #endif /* CONFIG_MKI */
