--- lufs-0.9.7-orig/filesystems/sshfs/sshfs.cpp.orig	2003-11-26 03:03:02.000000000 +0100
+++ lufs-0.9.7/filesystems/sshfs/sshfs.cpp	2003-11-26 03:04:08.000000000 +0100
@@ -345,7 +345,8 @@
     if(!handle.size())
 	return -1;
 
-    handles.push_back((struct atbl){string(file), handle, time(NULL), mode});
+	struct atbl ni = {string(file), handle, time(NULL), mode};
+    handles.push_back(ni);
 
     TRACE(handles.size() << " files still opened.");
 
