diff -NrU5 lxsession-edit-0.1.1.original/configure.in lxsession-edit-0.1.1/configure.in
--- lxsession-edit-0.1.1.original/configure.in	2009-08-04 22:53:54.000000000 -0600
+++ lxsession-edit-0.1.1/configure.in	2009-08-04 22:55:13.000000000 -0600
@@ -2,10 +2,11 @@
 
 AC_INIT(configure.in)
 AM_INIT_AUTOMAKE(lxsession-edit, 0.1.1)
 AM_CONFIG_HEADER(config.h)
 AM_MAINTAINER_MODE
+IT_PROG_INTLTOOL
 
 AC_ISC_POSIX
 AC_PROG_CC
 AM_PROG_CC_STDC
 AM_PROG_CC_C_O
