diff -ur kload-0.9.4.old/kload.cpp kload-0.9.4/kload.cpp
--- kload-0.9.4.old/kload.cpp	2009-08-06 03:15:59.000000000 +0300
+++ kload-0.9.4/kload.cpp	2009-08-06 03:17:06.000000000 +0300
@@ -87,7 +87,7 @@
     for (;it != sv.end();++it)
     {
         const char *line = it->c_str();
-        char *colon = strchr(line,':');
+        const char *colon = strchr(line,':');
         if (NULL != colon)  // we believe that this is a "real line"
         {
             // determine actual device name
diff -ur kload-0.9.4.old/netpanel.cpp kload-0.9.4/netpanel.cpp
--- kload-0.9.4.old/netpanel.cpp	2009-08-06 03:15:59.000000000 +0300
+++ kload-0.9.4/netpanel.cpp	2009-08-06 03:16:42.000000000 +0300
@@ -114,7 +114,7 @@
     {
         const char *line = it->c_str();
         
-        char *colon = strchr(line,':');
+        const char *colon = strchr(line,':');
         if (NULL != colon)  // we believe that this is a "real line"
         {
             // determine actual device name
