diff -Naur kload-0.9.4.orig/cpupanel.cpp kload-0.9.4/cpupanel.cpp
--- kload-0.9.4.orig/cpupanel.cpp	2004-10-18 21:49:19.000000000 +0200
+++ kload-0.9.4/cpupanel.cpp	2008-06-29 23:23:28.000000000 +0200
@@ -9,6 +9,7 @@
 #include "cpupanel.h"
 #include "KloadClt.h"
 
+#include <stdlib.h>
 #include <cstdio>
 
 using namespace std;
diff -Naur kload-0.9.4.orig/KloadClt.cpp kload-0.9.4/KloadClt.cpp
--- kload-0.9.4.orig/KloadClt.cpp	2004-10-28 15:30:29.000000000 +0200
+++ kload-0.9.4/KloadClt.cpp	2008-06-29 23:23:28.000000000 +0200
@@ -12,6 +12,7 @@
 
 #include <sys/socket.h>
 #include <netdb.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <errno.h>
 
diff -Naur kload-0.9.4.orig/kload.cpp kload-0.9.4/kload.cpp
--- kload-0.9.4.orig/kload.cpp	2004-10-28 15:03:05.000000000 +0200
+++ kload-0.9.4/kload.cpp	2008-06-29 23:23:28.000000000 +0200
@@ -15,6 +15,7 @@
 #include <cstdio>
 #include <cstring>
 #include <cctype>
+#include <stdlib.h>
 #include <errno.h>
 #include <time.h>
 
diff -Naur kload-0.9.4.orig/lpanel.cpp kload-0.9.4/lpanel.cpp
--- kload-0.9.4.orig/lpanel.cpp	2004-10-28 14:54:48.000000000 +0200
+++ kload-0.9.4/lpanel.cpp	2008-06-29 23:23:28.000000000 +0200
@@ -9,6 +9,7 @@
 
 #include <math.h>
 #include <unistd.h>
+#include <stdlib.h>
 
 using namespace std;
 
diff -Naur kload-0.9.4.orig/netpanel.cpp kload-0.9.4/netpanel.cpp
--- kload-0.9.4.orig/netpanel.cpp	2004-10-28 14:54:17.000000000 +0200
+++ kload-0.9.4/netpanel.cpp	2008-06-29 23:24:06.000000000 +0200
@@ -7,6 +7,7 @@
 #include <assert.h>
 #include <ctype.h>
 #include <qregexp.h>
+#include <stdlib.h>
 
 #include "netpanel.h"
 #include "KloadClt.h"
