diff -Naurp ffmpeg-0.4.9-p20050226.orig/libavcodec/Makefile ffmpeg-0.4.9-p20050226/libavcodec/Makefile
--- ffmpeg-0.4.9-p20050226.orig/libavcodec/Makefile	2005-04-25 17:31:43.000000000 -0700
+++ ffmpeg-0.4.9-p20050226/libavcodec/Makefile	2005-04-25 17:34:46.000000000 -0700
@@ -73,8 +73,9 @@ OBJS+= a52dec.o
 
 # using builtin liba52 or runtime linked liba52.so.0
 ifneq ($(CONFIG_A52BIN),yes)
-OBJS+= liba52/bit_allocate.o liba52/bitstream.o liba52/downmix.o \
-	liba52/imdct.o  liba52/parse.o liba52/crc.o liba52/resample.o
+#OBJS+= liba52/bit_allocate.o liba52/bitstream.o liba52/downmix.o \
+#	liba52/imdct.o  liba52/parse.o liba52/crc.o liba52/resample.o
+EXTRALIBS += -la52
 endif
 endif
 
diff -Naurp ffmpeg-0.4.9-p20050226.orig/libavcodec/a52dec.c ffmpeg-0.4.9-p20050226/libavcodec/a52dec.c
--- ffmpeg-0.4.9-p20050226.orig/libavcodec/a52dec.c	2004-05-21 07:37:16.000000000 -0700
+++ ffmpeg-0.4.9-p20050226/libavcodec/a52dec.c	2005-04-25 17:37:06.000000000 -0700
@@ -23,7 +23,7 @@
  */
 
 #include "avcodec.h"
-#include "liba52/a52.h"
+#include <a52dec/a52.h>
 
 #ifdef CONFIG_A52BIN
 #include <dlfcn.h>
