diff -ur ffmpeg-0.4.9-p20050226/configure ffmpeg-0.4.9-p20050226-configure/configure
--- ffmpeg-0.4.9-p20050226/configure	2005-02-24 16:18:02.000000000 +0100
+++ ffmpeg-0.4.9-p20050226-configure/configure	2005-04-24 18:13:19.206706664 +0200
@@ -404,7 +404,7 @@
   ;;
   --enable-a52) a52="yes"
   ;;
-  --enable-a52bin) a52bin="yes" ; extralibs="$ldl $extralibs"
+  --enable-a52bin) a52bin="yes" ; extralibs="$ldl $extralibs -la52"
   ;;
   --enable-dts) dts="yes" ; extralibs="$extralibs -ldts"
   ;;
@@ -428,7 +428,7 @@
   ;;
   --enable-xvid) xvid="yes"
   ;;
-  --enable-dc1394) dc1394="yes"
+  --enable-dc1394) dc1394="yes" ; extralibs="$extralibs -ldc1394_control -lraw1394"
   ;;
   --disable-vhook) vhook="no"
   ;;
