--- Makefile.old	2005-02-24 10:18:02.000000000 -0500
+++ Makefile		2005-03-18 17:48:45.000000000 -0500
@@ -73,6 +73,10 @@
 EXTRALIBS+=-ldc1394_control -lraw1394
 endif
 
+ifeq ($(CONFIG_A52BIN),yes)
+EXTRALIBS+=-la52
+endif
+
 ifeq ($(BUILD_VHOOK),yes)
 VHOOK=videohook
 INSTALLVHOOK=install-vhook
--- libavcodec/a52dec.c.old	2005-02-28 10:23:54.002233360 +0900
+++ libavcodec/a52dec.c	2005-02-28 10:35:36.910375080 +0900
@@ -73,7 +73,7 @@
 {
     void* f = dlsym(handle, symbol);
     if (!f)
-	fprintf(stderr, "A52 Decoder - function '%s' can't be resolved\n", symbol);
+	av_log(stderr, "A52 Decoder - function '%s' can't be resolved\n", symbol);
     return f;
 }
 #endif
@@ -86,7 +86,7 @@
     s->handle = dlopen(liba52name, RTLD_LAZY);
     if (!s->handle)
     {
-	fprintf(stderr, "A52 library %s could not be opened! \n%s\n", liba52name, dlerror());
+	av_log(stderr, "A52 library %s could not be opened! \n%s\n", liba52name, dlerror());
         return -1;
     }
     s->a52_init = (a52_state_t* (*)(uint32_t)) dlsymm(s->handle, "a52_init");
