diff -Naur mbrowse-0.3.1.vanilla/configure mbrowse-0.3.1/configure
--- mbrowse-0.3.1.vanilla/configure	2003-03-12 10:37:41.000000000 -0600
+++ mbrowse-0.3.1/configure	2004-09-04 18:29:43.771221872 -0500
@@ -4699,7 +4699,7 @@
           }
 
           #ifdef HAVE_NETSNMP
-	    version2 = strdup( NetSnmpVersionInfo );
+	    version2 = strdup( netsnmp_get_version() );
           #else
             version2 = strdup( VersionInfo);
           #endif
diff -Naur mbrowse-0.3.1.vanilla/src/snmpstuff.c mbrowse-0.3.1/src/snmpstuff.c
--- mbrowse-0.3.1.vanilla/src/snmpstuff.c	2002-06-21 06:45:18.000000000 -0500
+++ mbrowse-0.3.1/src/snmpstuff.c	2004-09-04 18:30:35.136413176 -0500
@@ -360,6 +360,7 @@
       case TYPE_NETADDR:
       case TYPE_OTHER:
       default:
+      	{}
     }
   }
   if (type == 0) {
