--- gnome-vfs-1.0.5/libgnomevfs/gnome-vfs-mime-magic.c.orig	2003-01-02 01:59:52.000000000 +0200
+++ gnome-vfs-1.0.5/libgnomevfs/gnome-vfs-mime-magic.c	2003-01-02 02:03:24.000000000 +0200
@@ -196,6 +196,11 @@
 		fmttype = 'u';
 	}
 
+	printf ("offset: %p, bsize: %i, result: %p\n", offset, bsize, result);
+
+	if ((offset <= 0) || (bsize <= 0) || (result <= 0))
+		return FALSE;
+
 	switch (bsize) {
 	case 1:
 		fmtstr[0] = '%'; 
