--- gnome-vfs-1.0.5/modules/http-method.c.proxy	2001-09-05 22:06:14.000000000 +0200
+++ gnome-vfs-1.0.5/modules/http-method.c	2002-09-12 18:35:28.000000000 +0200
@@ -125,18 +125,18 @@
 #define DEFAULT_HTTP_PROXY_PORT 8080
 
 /* GConf paths and keys */
-#define PATH_GCONF_GNOME_VFS "/system/gnome-vfs"
-#define ITEM_GCONF_HTTP_PROXY_PORT "http-proxy-port"
-#define ITEM_GCONF_HTTP_PROXY_HOST "http-proxy-host"
+#define PATH_GCONF_GNOME_VFS "/system/http_proxy"
+#define ITEM_GCONF_HTTP_PROXY_PORT "port"
+#define ITEM_GCONF_HTTP_PROXY_HOST "host"
 #define KEY_GCONF_HTTP_PROXY_PORT (PATH_GCONF_GNOME_VFS "/" ITEM_GCONF_HTTP_PROXY_PORT)
 #define KEY_GCONF_HTTP_PROXY_HOST (PATH_GCONF_GNOME_VFS "/" ITEM_GCONF_HTTP_PROXY_HOST)
 
-#define ITEM_GCONF_USE_HTTP_PROXY "use-http-proxy"
+#define ITEM_GCONF_USE_HTTP_PROXY "use_http_proxy"
 #define KEY_GCONF_USE_HTTP_PROXY (PATH_GCONF_GNOME_VFS "/" ITEM_GCONF_USE_HTTP_PROXY)
 
-#define KEY_GCONF_HTTP_AUTH_USER (PATH_GCONF_GNOME_VFS "/" "http-proxy-authorization-user")
-#define KEY_GCONF_HTTP_AUTH_PW (PATH_GCONF_GNOME_VFS "/" "http-proxy-authorization-password")
-#define KEY_GCONF_HTTP_USE_AUTH (PATH_GCONF_GNOME_VFS "/" "use-http-proxy-authorization")
+#define KEY_GCONF_HTTP_AUTH_USER (PATH_GCONF_GNOME_VFS "/" "authentication_user")
+#define KEY_GCONF_HTTP_AUTH_PW (PATH_GCONF_GNOME_VFS "/" "authentication_password")
+#define KEY_GCONF_HTTP_USE_AUTH (PATH_GCONF_GNOME_VFS "/" "use_authentication")
 
 
 /* Some status code validation macros.  */
--- gnome-vfs-1.0.5/modules/ftp-method.c.proxy	2001-08-06 23:02:04.000000000 +0200
+++ gnome-vfs-1.0.5/modules/ftp-method.c	2002-09-12 18:27:20.000000000 +0200
@@ -76,8 +76,8 @@
 #define IS_400(X) ((X) >= 400 && (X) < 500)
 #define IS_500(X) ((X) >= 500 && (X) < 600)
 
-static const char PROXY_KEY[] = "/system/gnome-vfs/http-proxy";
-static const char USE_PROXY_KEY[] = "/system/gnome-vfs/use-http-proxy";
+static const char PROXY_KEY[] = "/system/http_proxy/host";
+static const char USE_PROXY_KEY[] = "/system/http_proxy/use_http_proxy";
 
 
 static GnomeVFSResult do_open	         (GnomeVFSMethod               *method,
--- gnome-vfs-1.0.5/modules/cdda-method.c.proxy	2001-07-16 20:22:27.000000000 +0200
+++ gnome-vfs-1.0.5/modules/cdda-method.c	2002-09-12 18:27:20.000000000 +0200
@@ -99,9 +99,9 @@
 						 long 				bytes);
 
 
-static const char PROXY_HOST_KEY[] = "/apps/nautilus/preferences/http-proxy-host";
-static const char PROXY_PORT_KEY[] = "/apps/nautilus/preferenceshttp-proxy-port";
-static const char USE_PROXY_KEY[] = "/apps/nautilus/preferencesuse-http-proxy";
+static const char PROXY_HOST_KEY[] = "/system/http_proxy/host";
+static const char PROXY_PORT_KEY[] = "/system/http_proxy/port";
+static const char USE_PROXY_KEY[] = "/system/http_proxy/use_http_proxy";
 
 static CDDAContext *
 cdda_context_new (cdrom_drive *drive, GnomeVFSURI *uri)
