diff -NurB gnome-vfs-1.0.5-orig/doc/about.sgml gnome-vfs-1.0.5/doc/about.sgml
--- gnome-vfs-1.0.5-orig/doc/about.sgml	2004-04-01 02:12:47.000000000 -0500
+++ gnome-vfs-1.0.5/doc/about.sgml	2004-04-01 02:13:19.000000000 -0500
@@ -83,7 +83,7 @@
     <para>
       The GnomeVFS equivalent is very similar, but you will notice slightly different data types. The
       consistent returning of a GnomeVFSResult also necessitated moving the return value of read into
-      a pass-back-value pointer <variable>bytes_read</variable>:
+      a pass-back-value pointer <varname>bytes_read</varname>:
       <programlisting>
 	GnomeVFSResult gnome_vfs_read (GnomeVFSHandle *handle,
 	                               gpointer buffer,
diff -NurB gnome-vfs-1.0.5-orig/doc/gnome-vfs-docs.sgml gnome-vfs-1.0.5/doc/gnome-vfs-docs.sgml
--- gnome-vfs-1.0.5-orig/doc/gnome-vfs-docs.sgml	2004-04-01 02:12:47.000000000 -0500
+++ gnome-vfs-1.0.5/doc/gnome-vfs-docs.sgml	2004-04-01 02:40:00.000000000 -0500
@@ -18,7 +18,6 @@
 <!entity gnome-vfs-inet-connection SYSTEM "sgml/inet-connection.sgml">
 <!entity gnome-vfs-init SYSTEM "sgml/init.sgml">
 <!entity gnome-vfs-iobuf SYSTEM "sgml/iobuf.sgml">
-<!entity gnome-vfs-list-sort SYSTEM "sgml/list-sort.sgml">
 <!entity gnome-vfs-method SYSTEM "sgml/method.sgml">
 <!entity gnome-vfs-mime-handlers SYSTEM "sgml/mime-handlers.sgml">
 <!entity gnome-vfs-mime-info SYSTEM "sgml/mime-info.sgml">
@@ -94,7 +93,7 @@
 	without the use of threads.
       </para>
       &gnome-vfs-async-ops;
-    <--FIXME: should we display this?gnome-vfs-cancellable-ops-->
+    <!--FIXME: should we display this?gnome-vfs-cancellable-ops-->
       &gnome-vfs-ops;
       &gnome-vfs-directory;
       &gnome-vfs-xfer;
@@ -136,12 +135,12 @@
       read-write, seeking, etc). Modules are dynamically loaded based upon the URI
       scheme passed into the high-level GnomeVFS APIs.
     </para>
-      &writing-modules;
       &gnome-vfs-method;
       &gnome-vfs-module-shared;
       &gnome-vfs-module;
       &gnome-vfs-module-callback-module-api;
   </chapter>
+  &writing-modules;
 
   <chapter id="everything-else">
     <title>Other APIs</title>
@@ -152,11 +151,10 @@
       &gnome-vfs-directory-filter;
       &gnome-vfs-file-size;
       &gnome-vfs-find-directory;
-      <--FIXME: I think this should be private: gnome-vfs-handle-->
+      <!--FIXME: I think this should be private: gnome-vfs-handle-->
       &gnome-vfs-inet-connection;
       &gnome-vfs-init;
       &gnome-vfs-iobuf;
-      &gnome-vfs-list-sort;
       &gnome-vfs-module-callback;
       &gnome-vfs-standard-callbacks;
       &gnome-vfs-parse-ls;
diff -NurB gnome-vfs-1.0.5-orig/doc/gnome-vfs-sections.txt gnome-vfs-1.0.5/doc/gnome-vfs-sections.txt
--- gnome-vfs-1.0.5-orig/doc/gnome-vfs-sections.txt	2004-04-01 02:12:47.000000000 -0500
+++ gnome-vfs-1.0.5/doc/gnome-vfs-sections.txt	2004-04-01 03:16:07.000000000 -0500
@@ -194,7 +194,7 @@
 
 <SECTION>
 <FILE>find-directory</FILE>
-<TITLE>Find Directory</TITLE>
+<TITLE>Find Directory Section</TITLE>
 gnome_vfs_find_directory
 </SECTION>
 
@@ -224,6 +224,7 @@
 
 <SECTION>
 <FILE>init</FILE>
+<TITLE>Init Section</TITLE>
 gnome_vfs_init
 gnome_vfs_initialized
 gnome_vfs_shutdown
diff -NurB gnome-vfs-1.0.5-orig/doc/tmpl/list-sort.sgml gnome-vfs-1.0.5/doc/tmpl/list-sort.sgml
--- gnome-vfs-1.0.5-orig/doc/tmpl/list-sort.sgml	2004-04-01 02:12:47.000000000 -0500
+++ gnome-vfs-1.0.5/doc/tmpl/list-sort.sgml	1969-12-31 19:00:00.000000000 -0500
@@ -1,38 +0,0 @@
-<!-- ##### SECTION Title ##### -->
-list-sort
-
-<!-- ##### SECTION Short_Description ##### -->
-
-
-<!-- ##### SECTION Long_Description ##### -->
-<para>
-
-</para>
-
-<!-- ##### SECTION See_Also ##### -->
-<para>
-
-</para>
-
-<!-- ##### USER_FUNCTION GnomeVFSListCompareFunc ##### -->
-<para>
-
-</para>
-
-@a: 
-@b: 
-@data: 
-@Returns: 
-
-
-<!-- ##### FUNCTION gnome_vfs_list_sort ##### -->
-<para>
-
-</para>
-
-@list: 
-@compare_func: 
-@data: 
-@Returns: 
-
-
diff -NurB gnome-vfs-1.0.5-orig/doc/tmpl/standard-callbacks.sgml gnome-vfs-1.0.5/doc/tmpl/standard-callbacks.sgml
--- gnome-vfs-1.0.5-orig/doc/tmpl/standard-callbacks.sgml	2004-04-01 02:12:47.000000000 -0500
+++ gnome-vfs-1.0.5/doc/tmpl/standard-callbacks.sgml	2004-04-01 02:32:34.000000000 -0500
@@ -24,7 +24,7 @@
 </para>
 
 
-</para><informaltable pgwide=1 frame="none" role="params">
+<informaltable pgwide=1 frame="none" role="params">
 <tgroup cols="2">
 <colspec colwidth="2*">
 <colspec colwidth="8*">
diff -NurB gnome-vfs-1.0.5-orig/doc/writing-modules.sgml gnome-vfs-1.0.5/doc/writing-modules.sgml
--- gnome-vfs-1.0.5-orig/doc/writing-modules.sgml	2004-04-01 02:12:47.000000000 -0500
+++ gnome-vfs-1.0.5/doc/writing-modules.sgml	2004-04-01 02:21:20.000000000 -0500
@@ -6,7 +6,6 @@
       GNOME Virtual File System.
     </para>
   </abstract>
- </artheader>
 
   <sect1 id="Introduction">
     <title>Introduction</title>
