--- catspeech.orig	2005-07-31 20:04:41.000000000 -0700
+++ catspeech	2005-07-31 20:05:17.000000000 -0700
@@ -24,12 +24,9 @@ $fifo = '/dev/speech';
 
 open FIFO, ">>$fifo" or die "Error opening $fifo for appending ($!)\n";
 
-while (1)
-{
- while($line = <STDIN> ) {
-   print FIFO $line, "\n";
-   autoflush FIFO 1;
- }
+while($line = <STDIN> ) {
+	print FIFO $line, "\n";
+	autoflush FIFO 1;
 }
 
 close FIFO;
