diff -ur oo_1.0.1_src.orig/unoil/com/sun/star/document/makefile.mk oo_1.0.1_src/unoil/com/sun/star/document/makefile.mk
--- oo_1.0.1_src.orig/unoil/com/sun/star/document/makefile.mk	2000-09-18 10:28:26.000000000 -0500
+++ oo_1.0.1_src/unoil/com/sun/star/document/makefile.mk	2002-09-12 18:05:27.000000000 -0500
@@ -64,7 +64,9 @@
 PRJ		= ..$/..$/..$/..
 TARGET	= unoil_document
 PACKAGE	= com$/sun$/star$/document
-
+KELL_MARKER	= 1
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  $(PRJ)$/makefile.pmk
+
+KELL_MARKER	= 0
diff -ur oo_1.0.1_src.orig/unoil/makefile.pmk oo_1.0.1_src/unoil/makefile.pmk
--- oo_1.0.1_src.orig/unoil/makefile.pmk	2001-08-31 11:04:43.000000000 -0500
+++ oo_1.0.1_src/unoil/makefile.pmk	2002-09-12 18:05:27.000000000 -0500
@@ -74,7 +74,15 @@
 JAVAFILESLIST	:= $(shell +cd $(GENJAVADIR)$/$(PACKAGE) $(COMMANDSEPARATOR) ls *.java)
 .ENDIF          # "$(L10N_framework)"==""
 
+# WARNING!!!  WARNING!!!  THIS IS A HACK!!!  Well...so is just about everything
+# else in this package.
+# Do a little something special for project com.sun.star.document.
+.IF "$(KELL_MARKER)"=="1"
+GENJAVAFILES	:= $(foreach,i,$(JAVAFILESLIST) $(GENJAVADIR)$/$(PACKAGE)$/$i) $(GENJAVADIR)$/com$/sun$/star$/task$/XInteractionContinuation.java
+.ELSE
 GENJAVAFILES	:= $(foreach,i,$(JAVAFILESLIST) $(GENJAVADIR)$/$(PACKAGE)$/$i)
+.ENDIF
+
 GENCLASSFILES	:= $(foreach,i,$(JAVAFILESLIST:b) $(GENCLASSDIR)$/$(PACKAGE)$/$i.class)
