--- /tmp/cscope	2007-01-06 19:11:10.000000000 +0100
+++ cscope	2007-01-06 19:15:16.000000000 +0100
@@ -1,4 +1,4 @@
-#!/bin/perl
+#!/usr/bin/perl
 
 # $Id: cscope-158831-warning_webscope.patch,v 1.1 2007/01/06 18:30:51 opfer Exp $
 #
@@ -375,6 +375,7 @@
 	print &HtmlTop ("Web-CScope");
 
 	print <<ENDOFTEXTA;
+<p style="color:red">Be aware that this webfrontend is insecure and allows viewing ALL apache readable files, including your configuration!</p>   
 Select an operation below and enter a symbol, function or text to search in
 the database.  The active version is $version.  Input is case-sensitive,
 so if your search returns no results, check the case and the symbol name.<hr>
