diff -ur openssh-3.7.1p2.orig/configure.ac openssh-3.7.1p2/configure.ac
--- openssh-3.7.1p2.orig/configure.ac	2003-09-23 10:24:21.000000000 +0100
+++ openssh-3.7.1p2/configure.ac	2003-10-04 12:51:15.000000000 +0100
@@ -1992,7 +1992,7 @@
                         else
                                 KRB5ROOT=${withval}
                         fi
-			CPPFLAGS="$CPPFLAGS -I${KRB5ROOT}/include"
+			CPPFLAGS="$CPPFLAGS -I${KRB5ROOT}/include -I/usr/include/heimdal -I/usr/include/gssapi"
                         LDFLAGS="$LDFLAGS -L${KRB5ROOT}/lib"
                         AC_DEFINE(KRB5)
 			KRB5_MSG="yes"
@@ -2001,7 +2001,7 @@
                                        [ char *tmp = heimdal_version; ],
                                        [ AC_MSG_RESULT(yes)
                                          AC_DEFINE(HEIMDAL)
-                                         K5LIBS="-lkrb5 -ldes -lcom_err -lasn1 -lroken"
+                                         K5LIBS="-lkrb5 -lcom_err -lasn1 -lroken -lresolv"
                                        ],
                                        [ AC_MSG_RESULT(no)
                                          K5LIBS="-lkrb5 -lk5crypto -lcom_err"
