diff -urN TeX.orig/texk/web2c/Makefile.in TeX/texk/web2c/Makefile.in
--- TeX.orig/texk/web2c/Makefile.in	2005-08-03 11:08:02.000000000 +0200
+++ TeX/texk/web2c/Makefile.in	2006-05-27 23:45:48.000000000 +0200
@@ -562,13 +562,13 @@
 # Common
 programs = bibtex ctangle ctie cweave dvicopy dvitomp dvitype gftodvi gftopk \
  gftype mft $(odvicopy) $(odvitype) $(otangle) patgen pktogf pktype pltotf \
- pooltype tangle tftopl tie $(ttf2afm) vftovp vptovf weave
+ pooltype tangle tftopl tie $(ttf2afm) vftovp vptovf weave pdftosrc
 engines = $(tex) $(etex) $(mf) $(mfn) $(mpost) \
           $(omega) $(aleph) \
           $(pdftex) $(pdfetex)
 
 programs: $(programs) $(engines) $(mpware_programs) \
-          $(omegafonts_programs) $(otps_programs) $(pdftosrc)
+          $(omegafonts_programs) $(otps_programs)
 
 # Additional dependencies for relinking.
 # Note that each program and engine already depends on $(web2c_programs).
