diff -ur usr.orig/src/kernel-modules/fritzcapi/fritz.classic/src/main.c usr/src/kernel-modules/fritzcapi/fritz.classic/src/main.c
--- usr.orig/src/kernel-modules/fritzcapi/fritz.classic/src/main.c	2007-07-11 08:09:35.000000000 +0200
+++ usr/src/kernel-modules/fritzcapi/fritz.classic/src/main.c	2007-07-11 08:18:31.000000000 +0200
@@ -374,7 +374,7 @@
 	}
 
 #if defined (__fcpci__)	
-	if (0 == (err = pci_module_init (&fcpci_driver))) {
+	if (0 == (err = pci_register_driver (&fcpci_driver))) {
 		LOG("PCI driver registered.\n");
 		register_capi_driver (&fritz_capi_driver);
 		LOG("CAPI driver registered.\n");
diff -ur usr.orig/src/kernel-modules/fritzcapi/fritz.pci/src/main.c usr/src/kernel-modules/fritzcapi/fritz.pci/src/main.c
--- usr.orig/src/kernel-modules/fritzcapi/fritz.pci/src/main.c	2007-07-11 08:09:35.000000000 +0200
+++ usr/src/kernel-modules/fritzcapi/fritz.pci/src/main.c	2007-07-11 08:18:31.000000000 +0200
@@ -364,7 +364,7 @@
 #endif
 
 #if defined (__fcpci__)	
-	if (0 == (err = pci_module_init (&fcpci_driver))) {
+	if (0 == (err = pci_register_driver (&fcpci_driver))) {
 		LOG("PCI driver registered.\n");
 		register_capi_driver (&fritz_capi_driver);
 		LOG("CAPI driver registered.\n");
diff -ur usr.orig/src/kernel-modules/fritzcapi/fritz.pcmcia/src/main.c usr/src/kernel-modules/fritzcapi/fritz.pcmcia/src/main.c
--- usr.orig/src/kernel-modules/fritzcapi/fritz.pcmcia/src/main.c	2007-07-11 08:09:35.000000000 +0200
+++ usr/src/kernel-modules/fritzcapi/fritz.pcmcia/src/main.c	2007-07-11 08:18:31.000000000 +0200
@@ -363,7 +363,7 @@
 #endif
 
 #if defined (__fcpci__)	
-	if (0 == (err = pci_module_init (&fcpci_driver))) {
+	if (0 == (err = pci_register_driver (&fcpci_driver))) {
 		LOG("PCI driver registered.\n");
 		register_capi_driver (&fritz_capi_driver);
 		LOG("CAPI driver registered.\n");
diff -ur usr.orig/src/kernel-modules/fritzcapi/fritz.pnp/src/main.c usr/src/kernel-modules/fritzcapi/fritz.pnp/src/main.c
--- usr.orig/src/kernel-modules/fritzcapi/fritz.pnp/src/main.c	2007-07-11 08:09:35.000000000 +0200
+++ usr/src/kernel-modules/fritzcapi/fritz.pnp/src/main.c	2007-07-11 08:18:31.000000000 +0200
@@ -375,7 +375,7 @@
 	}
 
 #if defined (__fcpci__)	
-	if (0 == (err = pci_module_init (&fcpci_driver))) {
+	if (0 == (err = pci_register_driver (&fcpci_driver))) {
 		LOG("PCI driver registered.\n");
 		register_capi_driver (&fritz_capi_driver);
 		LOG("CAPI driver registered.\n");
