--- cgdb/src/cgdbrc.c
+++ cgdb/src/cgdbrc.c
@@ -24,6 +24,7 @@
 #include "cgdb.h"
 #include "sys_util.h"
 #include "std_list.h"
+#include "kui.h"
 
 extern struct tgdb *tgdb;
 
@@ -570,7 +571,7 @@ command_parse_map (void)
   value = yylex ();
   if (value != IDENTIFIER)
   {
-    xfree (key_token);
+    free (key_token);
     return -1;
   }
 
--- various/util/src/fork_util.c
+++ various/util/src/fork_util.c
@@ -39,6 +39,7 @@
 #include "fs_util.h"
 #include "pseudo.h"
 #include "logger.h"
+#include "terminal.h"
 
 struct pty_pair {
   int masterfd;
