diff -r 9450ca28545a spellerdialog.h
--- a/spellerdialog.h	Fri Jun 27 20:29:16 2008 +0200
+++ b/spellerdialog.h	Fri Jun 27 20:39:29 2008 +0200
@@ -13,7 +13,7 @@
 
 #include "ui_spellerdialog.h"
 #include "latexeditor.h"
-#include "hunspell/hunspell.hxx"
+#include <hunspell.hxx>
 
 #include <QTextCursor>
 
diff -r 9450ca28545a texmaker.pro
--- a/texmaker.pro	Fri Jun 27 20:29:16 2008 +0200
+++ b/texmaker.pro	Fri Jun 27 20:39:29 2008 +0200
@@ -1,7 +1,8 @@
 TEMPLATE	= app
 LANGUAGE	= C++
 TARGET	 = texmaker
-CONFIG	+= qt warn_off release
+CONFIG	+= qt warn_off release link_pkgconfig
+PKGCONFIG = hunspell
 ###############################
 HEADERS	+= texmaker.h \
 	symbollistwidget.h \
@@ -32,22 +33,7 @@
 	webpublishdialog.h \
 	spellerdialog.h \
 	pstrickslistwidget.h \
-	blockdata.h \
-	hunspell/affentry.hxx \
-	hunspell/affixmgr.hxx \
-	hunspell/atypes.hxx \
-	hunspell/baseaffix.hxx \
-	hunspell/csutil.hxx \
-	hunspell/dictmgr.hxx \
-	hunspell/hashmgr.hxx \
-	hunspell/htypes.hxx \
-	hunspell/hunspell.hxx \
-	hunspell/hunspell.h \
-	hunspell/langnum.hxx \
-	hunspell/license.hunspell \
-	hunspell/phonet.hxx \
-	hunspell/suggestmgr.hxx \
-	hunspell/license.myspell 
+	blockdata.h 
 SOURCES	+= main.cpp \
 	texmaker.cpp \
 	symbollistwidget.cpp \
@@ -77,16 +63,7 @@
 	aboutdialog.cpp \
 	webpublishdialog.cpp \
 	spellerdialog.cpp \
-	pstrickslistwidget.cpp \
-	hunspell/affentry.cxx \
-	hunspell/affixmgr.cxx \
-	hunspell/csutil.cxx \
-	hunspell/dictmgr.cxx \
-	hunspell/hashmgr.cxx \
-	hunspell/hunspell.cxx \
-	hunspell/phonet.cxx \
-	hunspell/suggestmgr.cxx \
-	hunspell/utf_info.cxx
+	pstrickslistwidget.cpp 
 RESOURCES += texmaker.qrc
 FORMS   += findwidget.ui\
 	gotolinedialog.ui \
