diff -Naur busybox-1.00-rc3.orig/modutils/insmod.c busybox-1.00-rc3/modutils/insmod.c
--- busybox-1.00-rc3.orig/modutils/insmod.c	2004-07-20 08:43:57.000000000 -0700
+++ busybox-1.00-rc3/modutils/insmod.c	2004-09-20 07:47:17.818776178 -0700
@@ -148,6 +148,18 @@
 #define ELFCLASSM	ELFCLASS32
 #endif
 
+#if defined(__x86_64__)
+#define BB_USE_GOT_ENTRIES
+#define BB_GOT_ENTRY_SIZE 8
+#define BB_USE_SINGLE
+
+#define MATCH_MACHINE(x) (x == EM_X86_64)
+
+#define SHT_RELM       SHT_RELA
+#define Elf64_RelM     Elf64_Rela
+#define ELFCLASSM      ELFCLASS64
+#endif
+
 #if defined(__mc68000__)
 #define CONFIG_USE_GOT_ENTRIES
 #define CONFIG_GOT_ENTRY_SIZE 4
