--- msh.c	Wed Dec 19 10:40:39 2001
+++ msh.c	Wed Dec 19 10:42:34 2001
@@ -3093,11 +3093,19 @@
 int key;
 {
 	if (*wp != NULL) {
-		for (; *wp != NULL; wp++)
+		for (; *wp != NULL; wp++) {
+			if (isassign(*wp)) {
+				char *cp;
+				assign(*wp, COPYV);
+				for (cp = *wp; *cp != '='; cp++)
+					;
+				*cp='\0';
+			}
 			if (checkname(*wp))
 				(*f) (lookup(*wp));
 			else
 				badid(*wp);
+		}
 	} else
 		putvlist(key, 1);
 }
