--- configure	2003-07-24 18:39:36.000000000 -0600
+++ configure	2003-07-24 18:38:58.000000000 -0600
@@ -12347,7 +12347,7 @@
 
 
 
-for ac_header in FL/Fl.H
+for ac_header in fltk-1.1/FL/Fl.H
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if eval "test \"\${$as_ac_Header+set}\" = set"; then
@@ -14258,7 +14258,7 @@
 fi
 
 
-CXXFLAGS="-Wall -ggdb -D_REENTRANT"
+CXXFLAGS="-Wall -ggdb -D_REENTRANT $CXXFLAGS"
                                                   ac_config_files="$ac_config_files Makefile src/Makefile man/Makefile intl/Makefile po/Makefile.in"
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
