--- /Makefile.am	2010-03-11 15:44:49.000000000 +0100
+++ /Makefile.am.new	2010-03-11 15:45:00.000000000 +0100
@@ -8,21 +8,6 @@
 pkgconfigdir=$(libdir)/pkgconfig
 pkgconfig_DATA = libwbxml2.pc
 
-wbxmldocdir = ${prefix}/share/doc/wbxml2-${VERSION}
-wbxmldoc_DATA = \
-	AUTHORS\
-	bootstrap\
-	BUGS\
-	ChangeLog\
-	COPYING\
-	GNU-LGPL\
-	INSTALL\
-	NEWS\
-	README\
-	References\
-	THANKS\
-	TODO
-
 dist-bz2: distdir
 	sed -e "s/tar.gz/tar.bz2/g" $(distdir)/wbxml2.spec > $(distdir)/wbxml2.spec.aux
 	mv $(distdir)/wbxml2.spec.aux $(distdir)/wbxml2.spec
