--- ./src/part/radialMap/Makefile.am.orig	2004-11-30 19:05:15.000000000 +0100
+++ ./src/part/radialMap/Makefile.am	2004-11-30 19:05:28.000000000 +0100
@@ -1,4 +1,4 @@
 INCLUDES = $(all_includes) -I$(top_srcdir)/src/part
 METASOURCES = AUTO
-noinst_LIBRARIES = libradialmap.a
-libradialmap_a_SOURCES = widget.cpp builder.cpp map.cpp widgetEvents.cpp labels.cpp segmentTip.cpp
+noinst_LTLIBRARIES = libradialmap.la
+libradialmap_la_SOURCES = widget.cpp builder.cpp map.cpp widgetEvents.cpp labels.cpp segmentTip.cpp
--- ./src/part/Makefile.am.orig	2004-11-30 19:04:45.000000000 +0100
+++ ./src/part/Makefile.am	2004-11-30 19:04:59.000000000 +0100
@@ -4,7 +4,7 @@
 
 #Part
 kde_module_LTLIBRARIES  = libfilelight.la
-libfilelight_la_LIBADD  = ./radialMap/libradialmap.a $(LIB_KFILE) $(LIB_KPARTS) $(LIB_KDEUI) $(LIB_QT)
+libfilelight_la_LIBADD  = ./radialMap/libradialmap.la $(LIB_KFILE) $(LIB_KPARTS) $(LIB_KDEUI) $(LIB_QT)
 libfilelight_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
 libfilelight_la_SOURCES = \
    dialog.ui \
