Index: sudo-1.7.2p1/configure.in
===================================================================
--- sudo-1.7.2p1.orig/configure.in
+++ sudo-1.7.2p1/configure.in
@@ -985,18 +985,22 @@ if test "$insults" = "on"; then
     AC_MSG_RESULT([$i])
 fi
 
+SECURE_PATH_SET="not set"
 AC_MSG_CHECKING(whether to override the user's path)
 AC_ARG_WITH(secure-path, [AS_HELP_STRING([--with-secure-path], [override the user's path with a built-in one])],
 [case $with_secure_path in
     yes)	AC_DEFINE_UNQUOTED(SECURE_PATH, "/bin:/usr/ucb:/usr/bin:/usr/sbin:/sbin:/usr/etc:/etc")
 		AC_MSG_RESULT([:/usr/ucb:/usr/bin:/usr/sbin:/sbin:/usr/etc:/etc])
+		SECURE_PATH_SET="set"
 		;;
     no)		AC_MSG_RESULT(no)
 		;;
     *)		AC_DEFINE_UNQUOTED(SECURE_PATH, "$with_secure_path")
 		AC_MSG_RESULT([$with_secure_path])
+		SECURE_PATH_SET="set"
 		;;
 esac], AC_MSG_RESULT(no))
+AC_SUBST([SECURE_PATH_SET])
 
 AC_MSG_CHECKING(whether to get ip addresses from the network interfaces)
 AC_ARG_WITH(interfaces, [AS_HELP_STRING([--without-interfaces], [don't try to read the ip addr of ether interfaces])],
Index: sudo-1.7.2p1/sudoers.man.in
===================================================================
--- sudo-1.7.2p1.orig/sudoers.man.in
+++ sudo-1.7.2p1/sudoers.man.in
@@ -1216,7 +1216,7 @@ people running \fBsudo\fR to have a sane
 want to use this.  Another use is if you want to have the \*(L"root path\*(R"
 be separate from the \*(L"user path.\*(R"  Users in the group specified by the
 \&\fIexempt_group\fR option are not affected by \fIsecure_path\fR.
-This is not set by default.
+This is @SECURE_PATH_SET@ by default.
 .IP "syslog" 12
 .IX Item "syslog"
 Syslog facility if syslog is being used for logging (negate to
