Index: sudo-1.8.1rc1/Makefile.in
===================================================================
--- sudo-1.8.1rc1.orig/Makefile.in
+++ sudo-1.8.1rc1/Makefile.in
@@ -51,20 +51,17 @@ LIBTOOL_DEPS = @LIBTOOL_DEPS@
 
 SHELL = @SHELL@
 
-all: config.status
-	for d in $(SUBDIRS) $(SAMPLES); do (cd $$d && $(MAKE) $@) || break; done
-
-check: config.status
-	for d in $(SUBDIRS) $(SAMPLES); do (cd $$d && $(MAKE) $@) || break; done
+all check: config.status
+	for d in $(SUBDIRS) $(SAMPLES); do $(MAKE) -C $$d $@ && continue; exit 1; done
 
 pre-install:
-	for d in $(SUBDIRS); do (cd $$d && $(MAKE) $@) || break; done
+	for d in $(SUBDIRS) $(SAMPLES); do $(MAKE) -C $$d $@ && continue; exit 1; done
 
 install-dirs install-binaries install-includes install-plugin uninstall: config.status pre-install
-	for d in $(SUBDIRS); do (cd $$d && $(MAKE) $@) || break; done
+	for d in $(SUBDIRS) $(SAMPLES); do $(MAKE) -C $$d $@ && continue; exit 1; done
 
 install install-doc: config.status ChangeLog pre-install
-	for d in $(SUBDIRS); do (cd $$d && $(MAKE) $@) || break; done
+	for d in $(SUBDIRS) $(SAMPLES); do $(MAKE) -C $$d $@ && continue; exit 1; done
 
 autoconf:
 	autoconf -I m4
