Index: configure.ac
===================================================================
--- configure.ac
+++ configure.ac	2006-07-26 23:26:38.637089048 +0200
@@ -36,7 +36,7 @@
 dnl System type
 AC_CANONICAL_HOST
 case $host_cpu in
-i686|x86|x86_64|x86pc|k5|k6|k6-2|k6-3|pentium*|athlon*)
+i686|x86|x86pc|k5|k6|k6-2|k6-3|pentium*|athlon*)
 AC_DEFINE(HAVE_ARCH_X86,,[Whether this computer has an x86 CPU])
 ;;
 esac
Index: src/defines.h
===================================================================
--- src/defines.h
+++ src/defines.h	2006-07-26 23:27:48.781503880 +0200
@@ -24,7 +24,7 @@
 /* #undef SIZEOF_SIZE_T */
 
 /* Whether this computer has an x86 CPU */
-#define HAVE_ARCH_X86 
+/* #undef HAVE_ARCH_X86 */
 
 /* The number of CPUs in this machine */
 /* #undef CPU_COUNT */
Index: src/sample.c
===================================================================
--- src/sample.c
+++ src/sample.c	2006-07-26 23:32:15.611081901 +0200
@@ -616,6 +616,7 @@
     
     cpu_caps = arbiter_get_cpu_caps();
 
+#ifdef HAVE_ARCH_X86
 #if USE_MMX_MINMAX == 3
     DEBUG("using static 3DNow! minmax\n");
 #elif USE_MMX_MINMAX == 2
@@ -641,6 +642,9 @@
 #else
     DEBUG("Using plain C minmax\n");
 #endif /* USE_MMX_MINMAX */
+#else
+    DEBUG("Using plain C minmax\n");
+#endif
 
     return 0;
 }
Index: src/sample.h
===================================================================
--- src/sample.h
+++ src/sample.h	2006-07-26 23:34:42.999953554 +0200
@@ -25,6 +25,7 @@
 #include <config.h>
 #include <audiofile.h>
 
+#ifdef HAVE_ARCH_X86
 #if USE_MMX_MINMAX == 2
 #define PEAK_MIN_FUNC min_int8_mmx2
 #define PEAK_MAX_FUNC max_int8_mmx2
@@ -47,6 +48,14 @@
 #define MIN16_FUNC peak_funcs.min16_func
 #define MAX16_FUNC peak_funcs.max16_func
 #endif
+#else
+#define PEAK_MIN_FUNC peak_funcs.peak_min_func
+#define PEAK_MAX_FUNC peak_funcs.peak_max_func
+#define MIN8_FUNC peak_funcs.min8_func
+#define MAX8_FUNC peak_funcs.max8_func
+#define MIN16_FUNC peak_funcs.min16_func
+#define MAX16_FUNC peak_funcs.max16_func
+#endif
 
 struct peak_func_table {
     graph_bits_unit_t (*peak_min_func)(const peak_unit_t *a, int count);
