--- bash_completion.orig/bash_completion	2005-01-20 06:29:03.982356288 -0500
+++ bash_completion/bash_completion	2005-01-20 06:44:08.957779112 -0500
@@ -105,7 +105,8 @@
 complete -f -X '!*.@(ogg|OGG|m3u|flac|spx)' ogg123
 complete -f -X '!*.@(mp3|MP3|ogg|OGG|pls|m3u)' gqmpeg freeamp
 complete -f -X '!*.fig' xfig
-complete -f -X '!*.@(mid?(i))' timidity playmidi
+complete -f -X '!*.@(mid?(i))' playmidi
+complete -f -X '!*.@(mid?(i)|rmi|RMI)' timidity
 complete -f -X '*.@(o|so|so.!(conf)|a|tar?(.@(gz|bz2))|tgz|tbz2|rpm|zip|ZIP|gif|GIF|jp?(e)g|JP?(E)G|mp3|MP3|mp?(e)g|MPG|avi|AVI|asf|ASF|ogg|OGG|class|CLASS)' vi vim gvim rvim view rview rgvim rgview gview
 complete -f -X '*.@(o|so|so.!(conf)|a|rpm|gif|GIF|jp?(e)g|JP?(E)G|mp3|MP3|mp?(e)g|MPG|avi|AVI|asf|ASF|ogg|OGG|class|CLASS)' emacs
 complete -f -X '!*.@(exe|EXE|com|COM|scr|SCR)' wine
