diff -ruN roxterm-1.2.3.orig/src/rtdbus.c roxterm-1.2.3/src/rtdbus.c
--- roxterm-1.2.3.orig/src/rtdbus.c	2006-07-28 15:55:31.000000000 -0500
+++ roxterm-1.2.3/src/rtdbus.c	2006-07-28 15:56:24.000000000 -0500
@@ -41,7 +41,7 @@
 	{
 		if (disconnect)
 		{
-			dbus_connection_disconnect(rtdbus_connection);
+			dbus_connection_close(rtdbus_connection);
 		}
 		UNREF_LOG(dbus_connection_unref(rtdbus_connection));
 		rtdbus_connection = NULL;
