Index: kile-1.9rc1/src/kile/kiledocumentinfo.h
===================================================================
--- kile-1.9rc1.orig/src/kile/kiledocumentinfo.h
+++ kile-1.9rc1/src/kile/kiledocumentinfo.h
@@ -99,7 +99,7 @@ public:
 	static KURL repairInvalidCharacters(const KURL&);
 	static KURL repairExtension(const KURL&);
 	static KURL makeValidTeXURL(const KURL & url);
-	static KURL Info::renameIfExist(const KURL& url);
+	static KURL renameIfExist(const KURL& url);
 
 public:
 	Info(Kate::Document *doc, LatexCommands *commands);
