--- acinclude.m4.orig	2006-10-22 00:07:01.000000000 +0200
+++ acinclude.m4	2006-10-22 00:07:20.000000000 +0200
@@ -73,7 +73,7 @@
 [AC_MSG_CHECKING([for Boost unit-test framework])
  AC_REQUIRE([AC_PROG_CC])
  ql_original_LIBS=$LIBS
- for boost_lib in boost_unit_test_framework-$CC boost_unit_test_framework ; do
+ for boost_lib in boost_unit_test_framework-$CC boost_unit_test_framework boost_unit_test_framework-mt ; do
      LIBS="$ql_original_LIBS -l$boost_lib"
      boost_unit_found=no
      AC_LINK_IFELSE(
