Index: src/hat/runtime/getconstr.c
===================================================================
RCS file: /usr/src/master/nhc/src/hat/runtime/getconstr.c,v
retrieving revision 1.13
diff -u -r1.13 getconstr.c
--- src/hat/runtime/getconstr.c	2001/11/02 14:59:08	1.13
+++ src/hat/runtime/getconstr.c	2002/04/17 15:04:07
@@ -16,7 +16,7 @@
   NodePtr np;
   char *sp;
 
-  for (sp = s; *sp != '\0'; sp++);
+  for (sp = s; *sp++ != '\0';);
   np = mkNil();
   for (; --sp >= s;)
     np = mkCons(mkChar(*sp), np);
