--- xmms-1.2.4/xmms-arts-0.4/Makefile.am.build	Mon Jan  8 15:52:12 2001
+++ xmms-1.2.4/xmms-arts-0.4/Makefile.am	Mon Jan  8 15:52:26 2001
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = libartsout.la
 CFLAGS = @CFLAGS@ @GTK_CFLAGS@ `artsc-config --cflags`
 LIBS = @LIBS@ @GTK_LIBS@ @PTHREAD_LIBS@
-libartsout_la_LDFLAGS = `artsc-config --libs` -lcmallocc
+libartsout_la_LDFLAGS = `artsc-config --libs`
 libartsout_la_SOURCES = about.c artsout.c artsout.h audio.c configure.c init.c mixer.c
 libdir = $(xmms_output_plugin_dir)
 XMMSPATH = $(XMMS_PATH)
--- xmms-1.2.4/xmms-arts-0.4/Makefile.in.build	Mon Jan  8 15:52:17 2001
+++ xmms-1.2.4/xmms-arts-0.4/Makefile.in	Mon Jan  8 15:52:33 2001
@@ -83,7 +83,7 @@
 lib_LTLIBRARIES = libartsout.la
 CFLAGS = @CFLAGS@ @GTK_CFLAGS@ `artsc-config --cflags`
 LIBS = @LIBS@ @GTK_LIBS@ @PTHREAD_LIBS@
-libartsout_la_LDFLAGS = `artsc-config --libs` -lcmallocc
+libartsout_la_LDFLAGS = `artsc-config --libs`
 libartsout_la_SOURCES = about.c artsout.c artsout.h audio.c configure.c init.c mixer.c
 libdir = $(xmms_output_plugin_dir)
 XMMSPATH = $(XMMS_PATH)
