Index: src/tests/t_pkg_use_file.cc
===================================================================
--- src/tests/t_pkg_use_file.cc	(revision 584)
+++ src/tests/t_pkg_use_file.cc	(working copy)
@@ -56,7 +56,7 @@
         verifyUseSet("=x11-libs/gtk+-2.10.14", "jpeg tiff");
         verifyUseSet("=sys-libs/ncurses-5.4-r6", "");
         verifyUseSet("=sys-libs/ncurses-5.5-r3", "nocxx minimal");
-        verifyUseSet("=sys-kernel/gentoo-sources-2.6.19-r5", "");
+        verifyUseSet("=sys-kernel/gentoo-sources-2.6.24-r8", "");
         verifyUseSet("=sys-kernel/gentoo-sources-2.6.23-r9", "symlink");
         verifyUseSet("=dev-java/sun-jdk-1.4.2.17", "");
         verifyUseSet("=dev-java/sun-jdk-1.6.0.06", "examples doc jce");
