diff -ru ../gnomebaker-0.5.0.orig/src/exec.c ./src/exec.c
--- ../gnomebaker-0.5.0.orig/src/exec.c	2005-10-05 17:45:05.000000000 -0300
+++ ./src/exec.c	2005-10-17 17:44:00.000000000 -0200
@@ -85,7 +85,7 @@
         static const gint BUFF_SIZE = 1024;
         gchar buffer[BUFF_SIZE];
         memset(buffer, 0x0, BUFF_SIZE * sizeof(gchar));
-        guint bytes = 0;
+        gsize bytes = 0;
         const GIOStatus status = g_io_channel_read_chars(channel, buffer, (BUFF_SIZE - 1) * sizeof(gchar), &bytes, NULL);  
         if (status == G_IO_STATUS_ERROR || status == G_IO_STATUS_AGAIN) /* need to check what to do for again */
         {
diff -ru ../gnomebaker-0.5.0.orig/src/gbcommon.h ./src/gbcommon.h
--- ../gnomebaker-0.5.0.orig/src/gbcommon.h	2005-09-30 21:06:58.000000000 -0300
+++ ./src/gbcommon.h	2005-10-17 17:43:41.000000000 -0200
@@ -74,7 +74,7 @@
 
 
 #define GB_LOG_FUNC											\
-	if(showtrace) g_print("[%x] [%s] [%s] [%d]\n", (guint)g_thread_self(), __FUNCTION__, __FILE__, __LINE__);	\
+	if(showtrace) g_print("[%p] [%s] [%s] [%d]\n", (gpointer)g_thread_self(), __FUNCTION__, __FILE__, __LINE__);	\
 		
 #define GB_TRACE			\
 	if(showtrace) g_message	\
