diff -ur antlr-2.7.2a2~/lib/cpp/antlr/CharScanner.hpp antlr-2.7.2a2/lib/cpp/antlr/CharScanner.hpp
--- antlr-2.7.2a2~/lib/cpp/antlr/CharScanner.hpp	2002-01-03 20:03:43.000000000 -0600
+++ antlr-2.7.2a2/lib/cpp/antlr/CharScanner.hpp	2003-01-12 11:41:51.000000000 -0600
@@ -44,6 +44,8 @@
 #include <antlr/LexerSharedInputState.hpp>
 #include <map>
 
+#include <ctype.h>
+
 #ifdef ANTLR_CXX_SUPPORTS_NAMESPACE
 namespace antlr {
 #endif
