--- Makerules
+++ Makerules
@@ -7,10 +7,10 @@
 CFLAGS ?=
 LDFLAGS ?=
 
-CFLAGS += -Ifitz -Imupdf -Wall
+CFLAGS += -Ifitz -Imupdf
 
 ifeq "$(build)" "debug"
-CFLAGS += -pipe -g
+CFLAGS += 
 endif
 
 ifeq "$(build)" "profile"
@@ -28,7 +28,7 @@
 
 ifeq "$(OS)" "Linux"
 SYS_FREETYPE_INC := `pkg-config --cflags freetype2`
-X11LIBS := -lX11 -lXext
+X11LIBS := `pkg-config --libs x11 xext`
 PDFVIEW_EXE = $(X11VIEW_EXE)
 endif
 
