Michael Weber <xmw at gentoo dot org> 09/09/2010

--- apps/pdfapp.c.orig	2010-09-09 01:59:16.820000053 +0200
+++ apps/pdfapp.c	2010-09-09 02:02:32.400000054 +0200
@@ -602,6 +602,20 @@
 		pdfapp_showpage(app, 0, 1);
 		break;
 
+	case '_': // zoom to default resolution
+		app->resolution = 72;
+		pdfapp_showpage(app, 0, 1);
+		break;
+
+	case 'z': // zoom to fit window
+	case 'Z': // zoom to fill window
+		if ( ( c == 'Z' ) ^ ( app->image->w * app->winh > app->winw * app->image->h ) ) 
+			app->resolution = ( app->resolution * app->winw ) / app->image->w;
+		else
+			app->resolution = ( app->resolution * app->winh ) / app->image->h;
+		pdfapp_showpage(app, 0, 1);
+		break; 
+
 	case 'L':
 		app->rotate -= 90;
 		pdfapp_showpage(app, 0, 1);
@@ -725,6 +739,7 @@
 		break;
 
 	case 'b':
+	case '\b': //backspace
 		panto = DONT_PAN;
 		if (app->numberlen > 0)
 			app->pageno -= atoi(app->number);
@@ -733,6 +748,7 @@
 		break;
 
 	case ' ':
+	case 'f':
 		panto = DONT_PAN;
 		if (app->numberlen > 0)
 			app->pageno += atoi(app->number);
--- debian/mupdf.1.orig	2010-09-09 02:05:07.573000045 +0200
+++ debian/mupdf.1	2010-09-09 02:14:25.744000054 +0200
@@ -1,4 +1,4 @@
-.TH MUPDF 1 "March 15, 2010"
+.TH MUPDF 1 "September 9, 2010"
 .\" Please adjust this date whenever revising the manpage.
 .SH NAME
 mupdf \- MuPDF is a lightweight PDF viewer written in portable C
@@ -96,6 +96,15 @@
 .B \-
 Zoom out.
 .TP
+.B _
+Reset zoom to default (72dpi).
+.TP
+.B z
+Zoom page size to fit into the window frame.
+.TP
+.B Z
+Zoom page size to fill the window frame.
+.TP
 .B <
 Rotate page counter-clockwise or left by 90\(de.
 .TP
