--- src/key.c.org	2004-11-17 14:37:22.887315224 +0100
+++ src/key.c	2004-11-17 14:37:57.656029576 +0100
@@ -262,7 +262,7 @@
     define_sequences (mc_default_keys);
     
     /* Terminfo on irix does not have some keys */
-    if ((!strncmp (term, "iris-ansi", 9)) || (!strncmp (term, "xterm", 5)))
+    if (term && (!strncmp (term, "iris-ansi", 9)) || (!strncmp (term, "xterm", 5)))
 	define_sequences (xterm_key_defines);
     
     define_sequences (mc_bindings);
@@ -271,7 +271,7 @@
     load_xtra_key_defines();
     
 #ifdef __QNX__
-    if (strncmp(term, "qnx", 3) == 0){
+    if (term && strncmp(term, "qnx", 3) == 0){
 	/* Modify the default value of use_8th_bit_as_meta: we would
 	 * like to provide a working mc for a newbie who knows nothing
 	 * about [Options|Display bits|Full 8 bits input]...
