--- Gato/Gato.py	2008-01-27 14:10:49.000000000 +0100
+++ Gato-patched/Gato.py	2008-01-27 14:12:37.000000000 +0100
@@ -1,4 +1,5 @@
-#!/usr/bin/env python2.3
+#!/usr/bin/env python
+
 ################################################################################
 #
 #       This file is part of Gato (Graph Animation Toolbox) 
--- Gato/Gred.py	2008-01-27 14:17:02.000000000 +0100
+++ Gato-patched/Gred.py	2008-01-27 14:17:25.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.3
+#!/usr/bin/env python
 ################################################################################
 #
 #       This file is part of Gato (Graph Animation Toolbox) 
