diff -bur maildrop-1.7.0.20040907-orig/maildrop/main.C maildrop-1.7.0.20040907/maildrop/main.C
--- maildrop-1.7.0.20040907-orig/maildrop/main.C	2004-06-26 17:20:14.000000000 +0200
+++ maildrop-1.7.0.20040907/maildrop/main.C	2004-12-28 00:50:09.164726264 +0100
@@ -326,7 +326,9 @@
 
 	// Read the config
 	mysqlcfg = (mdmysqlconfig*)get_mysql_config(MAILDROPMYSQLCONFIG);
-	if ( mysqlcfg )
+	if ( !mysqlcfg )
+	  return 0;
+	else
 	  if ( !(mysqlcfg->hostname && mysqlcfg->database && mysqlcfg->dbtable) )
 	      return 0;
 
