--- module-init-tools-0.9.7/moduleops_core.c.chmou	2002-12-26 07:04:42.000000000 +0100
+++ module-init-tools-0.9.7/moduleops_core.c	2003-01-14 21:24:17.000000000 +0100
@@ -30,6 +30,9 @@
 	ksyms = PERBIT(load_section)(module->mmap, "__ksymtab", &size);
 	for (i = 0; i < size / sizeof(struct PERBIT(kernel_symbol)); i++)
 		add_symbol(ksyms[i].name, module);
+ 	ksyms = PERBIT(load_section)(module->mmap, "__gpl_ksymtab", &size);
+ 	for (i = 0; i < size / sizeof(struct PERBIT(kernel_symbol)); i++)
+	  add_symbol(ksyms[i].name, module);
 }
 
 /* Calculate the dependencies for this module */
