--- module-init-tools-0.9.7/generate-modprobe.conf.orig	2002-12-29 11:46:07.000000000 +0200
+++ module-init-tools-0.9.7/generate-modprobe.conf	2002-12-29 11:46:17.000000000 +0200
@@ -2,7 +2,7 @@
 
 # Naive shell script to translate modules.conf + defaults to modprobe.conf
 # Normal caveats apply, as with any machine translation.
-if [ $# -gt 1 -o x"$1" = x--help ]; then
+if [ $# -gt 2 -o x"$1" = x--help ]; then
     echo "Usage: $0 [modprobe.conf]"
     echo "   Converts your current modules setup to modprobe.conf."
     echo "   Don't trust it too much."
