diff -urN Gauche-gtk-0.3.orig/gtkgl/Makefile.in Gauche-gtk-0.3/gtkgl/Makefile.in
--- Gauche-gtk-0.3.orig/gtkgl/Makefile.in	2003-09-27 20:08:48.000000000 +0900
+++ Gauche-gtk-0.3/gtkgl/Makefile.in	2003-09-27 20:11:33.000000000 +0900
@@ -25,9 +25,9 @@
 GENERATED = gdkgllib.c gtkgllib.c
 
 INSTALL_TYPE = @INSTALL_TYPE@
-HEADER_INSTALL_DIR  = `$(GAUCHE_CONFIG) --$(INSTALL_TYPE)incdir`
-SCM_INSTALL_DIR     = `$(GAUCHE_CONFIG) --$(INSTALL_TYPE)libdir`
-ARCH_INSTALL_DIR    = `$(GAUCHE_CONFIG) --$(INSTALL_TYPE)archdir`
+HEADER_INSTALL_DIR  = ${DESTDIR}`$(GAUCHE_CONFIG) --$(INSTALL_TYPE)incdir`
+SCM_INSTALL_DIR     = ${DESTDIR}`$(GAUCHE_CONFIG) --$(INSTALL_TYPE)libdir`
+ARCH_INSTALL_DIR    = ${DESTDIR}`$(GAUCHE_CONFIG) --$(INSTALL_TYPE)archdir`
 
 all : $(TARGET)
 
diff -urN Gauche-gtk-0.3.orig/lib/Makefile.in Gauche-gtk-0.3/lib/Makefile.in
--- Gauche-gtk-0.3.orig/lib/Makefile.in	2003-09-27 20:08:48.000000000 +0900
+++ Gauche-gtk-0.3/lib/Makefile.in	2003-09-27 20:10:44.000000000 +0900
@@ -4,7 +4,7 @@
 GAUCHE_CONFIG = @GAUCHE_CONFIG@
 INSTALL = @INSTALL@
 INSTALL_TYPE = @INSTALL_TYPE@
-SCM_INSTALL_DIR = `$(GAUCHE_CONFIG) --$(INSTALL_TYPE)libdir`
+SCM_INSTALL_DIR = ${DESTDIR}`$(GAUCHE_CONFIG) --$(INSTALL_TYPE)libdir`
 
 SCMFILES = gtk.scm gtk/gtkaux.scm gtk/gdkkeysyms.scm gtk/listener.scm \
            gtk/error-dialog.scm @GTKGL_SCM@
diff -urN Gauche-gtk-0.3.orig/src/Makefile.in Gauche-gtk-0.3/src/Makefile.in
--- Gauche-gtk-0.3.orig/src/Makefile.in	2003-09-27 20:08:48.000000000 +0900
+++ Gauche-gtk-0.3/src/Makefile.in	2003-09-27 20:10:14.000000000 +0900
@@ -40,9 +40,9 @@
 TESTFILES = test-native test-native.o
 
 INSTALL_TYPE = @INSTALL_TYPE@
-HEADER_INSTALL_DIR  = `$(GAUCHE_CONFIG) --$(INSTALL_TYPE)incdir`
-SCM_INSTALL_DIR     = `$(GAUCHE_CONFIG) --$(INSTALL_TYPE)libdir`
-ARCH_INSTALL_DIR    = `$(GAUCHE_CONFIG) --$(INSTALL_TYPE)archdir`
+HEADER_INSTALL_DIR  = ${DESTDIR}`$(GAUCHE_CONFIG) --$(INSTALL_TYPE)incdir`
+SCM_INSTALL_DIR     = ${DESTDIR}`$(GAUCHE_CONFIG) --$(INSTALL_TYPE)libdir`
+ARCH_INSTALL_DIR    = ${DESTDIR}`$(GAUCHE_CONFIG) --$(INSTALL_TYPE)archdir`
 
 all : $(TARGET)
 
