Index: python/deskbar-handler/tracker-handler.py
===================================================================
--- python/deskbar-handler/tracker-handler.py	(revision 461)
+++ python/deskbar-handler/tracker-handler.py	(revision 462)
@@ -1,4 +1,5 @@
-
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
 #    This handler was originaly created by Mikkel Kamstrup (c) 2006 and updated by Eugenio Cutolo (eulin)
 #
 #    This program can be distributed under the terms of the GNU GPL version 2 or later.
