diff -burN tracker-0.5.3/data/Makefile.am tracker-0.5.3-clean/data/Makefile.am
--- tracker-0.5.3/data/Makefile.am	2006-11-29 18:55:12.000000000 -0600
+++ tracker-0.5.3/data/Makefile.am	2007-01-09 23:47:04.000000000 -0600
@@ -3,11 +3,8 @@
 configdir = $(datadir)/tracker
 
 config_DATA =   tracker-introspect.xml \
-		mysql-tracker.sql \
 		sqlite-tracker.sql \
-		mysql-stored-procs.sql \
-		sqlite-stored-procs.sql \
-		mysql-system.sql
+		sqlite-stored-procs.sql
 
 servicedir       = $(prefix)/share/dbus-1/services
 service_in_files = tracker.service.in
diff -burN tracker-0.5.3/src/trackerd/Makefile.am tracker-0.5.3-clean/src/trackerd/Makefile.am
--- tracker-0.5.3/src/trackerd/Makefile.am	2006-12-05 17:28:34.000000000 -0600
+++ tracker-0.5.3/src/trackerd/Makefile.am	2007-01-09 23:48:52.000000000 -0600
@@ -139,7 +139,6 @@
 trackerd_LDADD = $(GLIB2_LIBS) 	\
 		 $(FAM_LIBS)	\
 		 $(DBUS_LIBS)	\
-		 $(MYSQL_LIBS)	\
 		 $(PANGO_LIBS)	\
 		 $(GMIME_LIBS)	\
 		 $(additional_libs) \
diff -burN tracker-0.5.3/src/trackerd/tracker-db.h tracker-0.5.3-clean/src/trackerd/tracker-db.h
--- tracker-0.5.3/src/trackerd/tracker-db.h	2006-12-23 17:26:08.000000000 -0600
+++ tracker-0.5.3/src/trackerd/tracker-db.h	2007-01-09 23:48:39.000000000 -0600
@@ -31,8 +31,6 @@
 
 #ifdef USING_SQLITE
 #include "tracker-db-sqlite.h"
-#else
-#include "tracker-db-mysql.h"
 #endif
 
 gboolean	tracker_db_is_file_up_to_date 	(DBConnection *db_con, const char *uri, guint32 *id);
diff -burN tracker-0.5.3/src/trackerd/tracker-mbox-evolution.h tracker-0.5.3-clean/src/trackerd/tracker-mbox-evolution.h
--- tracker-0.5.3/src/trackerd/tracker-mbox-evolution.h	2006-12-23 17:26:07.000000000 -0600
+++ tracker-0.5.3/src/trackerd/tracker-mbox-evolution.h	2007-01-09 23:48:16.000000000 -0600
@@ -32,8 +32,6 @@
 
 #ifdef USING_SQLITE
 #   include "tracker-db-sqlite.h"
-#else
-#   include "tracker-db-mysql.h"
 #endif
 
 
diff -burN tracker-0.5.3/src/trackerd/tracker-mbox.h tracker-0.5.3-clean/src/trackerd/tracker-mbox.h
--- tracker-0.5.3/src/trackerd/tracker-mbox.h	2006-12-23 17:26:07.000000000 -0600
+++ tracker-0.5.3/src/trackerd/tracker-mbox.h	2007-01-09 23:49:11.000000000 -0600
@@ -32,8 +32,6 @@
 
 #ifdef USING_SQLITE
 #   include "tracker-db-sqlite.h"
-#else
-#   include "tracker-db-mysql.h"
 #endif
 
 
diff -burN tracker-0.5.3/src/trackerd/tracker-mbox-kmail.h tracker-0.5.3-clean/src/trackerd/tracker-mbox-kmail.h
--- tracker-0.5.3/src/trackerd/tracker-mbox-kmail.h	2006-12-23 17:26:07.000000000 -0600
+++ tracker-0.5.3/src/trackerd/tracker-mbox-kmail.h	2007-01-09 23:48:02.000000000 -0600
@@ -31,8 +31,6 @@
 
 #ifdef USING_SQLITE
 #   include "tracker-db-sqlite.h"
-#else
-#   include "tracker-db-mysql.h"
 #endif
 
 
diff -burN tracker-0.5.3/src/trackerd/tracker-mbox-thunderbird.h tracker-0.5.3-clean/src/trackerd/tracker-mbox-thunderbird.h
--- tracker-0.5.3/src/trackerd/tracker-mbox-thunderbird.h	2006-12-23 17:26:07.000000000 -0600
+++ tracker-0.5.3/src/trackerd/tracker-mbox-thunderbird.h	2007-01-09 23:49:24.000000000 -0600
@@ -32,8 +32,6 @@
 
 #ifdef USING_SQLITE
 #   include "tracker-db-sqlite.h"
-#else
-#   include "tracker-db-mysql.h"
 #endif
 
 
