--- radeontool-1.5/radeontool.c		2005-07-26 20:59:16.000000000 +0200
+++ radeontool-1.5-ati-vga/radeontool.c	2005-07-26 20:59:48.000000000 +0200
@@ -293,7 +293,7 @@
        if(fgets(line,sizeof(line),fp) == NULL) {  /* if end of file */
           fatal("Radeon hardware not found in lspci output.\n");
        }
-       if(strstr(line,"Radeon") || strstr(line,"ATI Tech")) {  /* if line contains a "radeon" string */
+       if(strstr(line,"Radeon") || strstr(line,"VGA compatible controller: ATI Tech")) {  /* if line contains a "radeon" string */
           if(skip-- < 1) {
              break;
           }
