diff -urp radeontool-1.5/radeontool.c radeontool-1.5-mmap/radeontool.c
--- radeontool-1.5/radeontool.c	2004-02-11 04:50:27.000000000 +0100
+++ radeontool-1.5-mmap/radeontool.c	2005-01-20 13:12:50.000000000 +0100
@@ -107,7 +107,7 @@ static unsigned char * map_devince_memor
         mem_fd, 
         base
     );
-    if ((long)device_mem < 0) {
+    if (device_mem == MAP_FAILED) {
         if(debug)
             fprintf(stderr,"mmap returned %d\n",(int)device_mem);
         fatal("mmap error \n");
