diff -ur pornview-0.2.0pre1.orig/pornview.desktop pornview-0.2.0pre1/pornview.desktop
--- pornview-0.2.0pre1.orig/pornview.desktop	2002-12-11 21:28:51.000000000 +0200
+++ pornview-0.2.0pre1/pornview.desktop	2008-04-21 16:49:47.000000000 +0300
@@ -2,6 +2,7 @@
 Name=PornView
 Comment=Photo/Movie collection manager
 Exec=pornview
-Icon=pornview.png
-Terminal=0
+Icon=pornview
+Terminal=false
 Type=Application
+Categories=GTK;Graphics;Viewer;
