diff -ur mozart-1.3.2.20060615.orig/share/bin/Makefile.in mozart-1.3.2.20060615/share/bin/Makefile.in
--- mozart-1.3.2.20060615.orig/share/bin/Makefile.in	2001-08-04 22:24:41.000000000 +0000
+++ mozart-1.3.2.20060615/share/bin/Makefile.in	2009-08-10 21:24:55.000000000 +0000
@@ -35,7 +35,7 @@
 INSTALL_BIN=	$(INSTALL) -m 555
 INSTALL_FILE=	$(INSTALL) -m 444
 
-SCRIPTS=	oz ozplatform
+SCRIPTS=	oz
 LINKS=		ozengine text2pickle pickle2text oldpickle2text oztool
 LIB_SCRIPTS=	$(addprefix $(BINDIR)/,$(SCRIPTS))
 LIB_LINKS=	$(addprefix $(BINDIR)/,$(LINKS))
diff -ur mozart-1.3.2.20060615.orig/share/bin/oz.in mozart-1.3.2.20060615/share/bin/oz.in
--- mozart-1.3.2.20060615.orig/share/bin/oz.in	2006-06-13 22:25:01.000000000 +0000
+++ mozart-1.3.2.20060615/share/bin/oz.in	2009-08-10 21:24:55.000000000 +0000
@@ -4,7 +4,7 @@
 # to draw links from say /usr/local/bin to OZHOME/bin
 # see chapter "installation" in the users manual for more information
 
-# OZHOME=/usr/local/mozart
+OZHOME=@prefix@/lib/mozart
 
 
 
@@ -24,7 +24,7 @@
 fi
 export OZHOME
 
-: ${OZPLATFORM=`"$OZHOME/bin/ozplatform"`}
+: ${OZPLATFORM="@PLATFORM@"}
 : ${OZEMULATOR="$OZHOME/platform/$OZPLATFORM/emulator.exe"}
 : ${OZVERSION="@OZVERSION@"}
 : ${OZ_DOTOZ="$HOME/.oz/$OZVERSION"}
diff -ur mozart-1.3.2.20060615.orig/share/bin/ozplatform mozart-1.3.2.20060615/share/bin/ozplatform
--- mozart-1.3.2.20060615.orig/share/bin/ozplatform	2006-05-03 12:17:13.000000000 +0000
+++ mozart-1.3.2.20060615/share/bin/ozplatform	2009-08-10 21:26:08.000000000 +0000
@@ -40,7 +40,7 @@
    ;;
    m68k\ Linux\ *)      OZARCH=linux-m68k
    ;;
-   sparc\ Linux\ *)     OZARCH=linux-sparc
+   sparc*\ Linux\ *)     OZARCH=linux-sparc
    ;;
    9000/7*HP-UX*)	OZARCH=hpux-700
    ;;
@@ -78,6 +78,8 @@
    ;;
    *i*86*Darwin*)       OZARCH=i486-darwin
    ;;
+   x86_64\ Linux*)	OZARCH=linux-x86_64_32
+   ;;
    *)  	                OZARCH=unknown-unknown
    ;;
 esac
