diff -ur mozart-1.3.2.20060615.orig/Makefile.in mozart-1.3.2.20060615/Makefile.in
--- mozart-1.3.2.20060615.orig/Makefile.in	2004-05-19 10:56:54.000000000 +1200
+++ mozart-1.3.2.20060615/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -111,7 +111,7 @@
 ./config.status: ./configure
 	./config.status --recheck
 
-install:: $(ALLFILES)
+install::
 
 $(ALLFILES): $(PREFIX)/% : % $(PREFIX)
 	$(INSTALL_FILE) $< $@
diff -ur mozart-1.3.2.20060615.orig/platform/emulator/Makefile.in mozart-1.3.2.20060615/platform/emulator/Makefile.in
--- mozart-1.3.2.20060615.orig/platform/emulator/Makefile.in	2004-01-12 23:03:44.000000000 +1300
+++ mozart-1.3.2.20060615/platform/emulator/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -226,10 +226,15 @@
 .PHONY:	all first install install-bin install-inc \
 	install-dir depend tags clean cleaner distclean veryclean
 
-all: $(TARGETS) $(DYNAMIC_MODULETARGETS)
+all: $(TARGETS) dynamic_modules
 
 bootstrap:: all
 
+dynamic_modules: $(TARGETS)
+	for dynamic_module in $(DYNAMIC_MODULETARGETS); do \
+	  $(MAKE) $$dynamic_module; \
+	done
+
 $(MODINTERS): bidecl.perl
 
 .PHONY: libDPB.a DPB.so-$(PLATFORM)
@@ -400,7 +405,7 @@
 $(NEWCACHEDIR):
 	$(INSTALL_DIR) $@
 
-$(NEWCACHEDIR)/% : %
+$(NEWCACHEDIR)/% : % $(NEWCACHEDIR)
 	$(INSTALL_FILE) $< $@
 
 install-inc: $(INCDIR)
diff -ur mozart-1.3.2.20060615.orig/platform/emulator/libdp/Makefile.in mozart-1.3.2.20060615/platform/emulator/libdp/Makefile.in
--- mozart-1.3.2.20060615.orig/platform/emulator/libdp/Makefile.in	2002-08-03 08:20:44.000000000 +1200
+++ mozart-1.3.2.20060615/platform/emulator/libdp/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -88,7 +88,7 @@
 	$(AR) -rc $@ $< && $(RANLIB) $@
 
 #
-dpMiscModule.lo: ../modDPMisc-if.cc
+dpMiscModule.lo: ../modDPMisc-if.cc $(CREATEDFILES)
 
 ../DPMisc.so-$(PLATFORM): dpMiscModule.lo 
 	$(DYNLD) -o $@ $<
@@ -146,6 +146,8 @@
 msgContainer_marshal.cc: msgContainer_marshal.m4cc msgFormat.m4
 msgContainer.o: msgContainer_marshal.m4cc msgFormat.m4
 
+$(REALCORESRCS): $(CREATEDCOREFILES)
+
 BUILDDIR=$(shell pwd)
 tags:	$(CREATEDFILES)
 	etags \
diff -ur mozart-1.3.2.20060615.orig/platform/tools/gump/Makefile.in mozart-1.3.2.20060615/platform/tools/gump/Makefile.in
--- mozart-1.3.2.20060615.orig/platform/tools/gump/Makefile.in	2003-01-08 02:40:10.000000000 +1300
+++ mozart-1.3.2.20060615/platform/tools/gump/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -66,15 +66,15 @@
 $(SHAREDIR) $(PLATFORMDIR) $(INCDIR) $(NEWCACHEDIR):
 	$(MKINSTALLDIRS) $@
 
-$(NEWCACHEDIR)/GumpScanner.so-$(PLATFORM): GumpScanner.so-$(PLATFORM)
+$(NEWCACHEDIR)/GumpScanner.so-$(PLATFORM): GumpScanner.so-$(PLATFORM) $(NEWCACHEDIR)
 	$(INSTALLDLL) $< $@
-$(NEWCACHEDIR)/Bison.so-$(PLATFORM): ozbison/Bison.so-$(PLATFORM)
+$(NEWCACHEDIR)/Bison.so-$(PLATFORM): ozbison/Bison.so-$(PLATFORM) $(NEWCACHEDIR)
 	$(INSTALLDLL) $< $@
 
-$(PLATFORMDIR)/flex.exe: ozflex/flex.exe
+$(PLATFORMDIR)/flex.exe: ozflex/flex.exe $(PLATFORMDIR)
 	$(INSTALLPRG) $< $@
 
-$(INCDIR)/%: %
+$(INCDIR)/%: % $(INCDIR)
 	$(INSTALLFILE) $< $@
 
 clean veryclean:
diff -ur mozart-1.3.2.20060615.orig/share/bin/Makefile.in mozart-1.3.2.20060615/share/bin/Makefile.in
--- mozart-1.3.2.20060615.orig/share/bin/Makefile.in	2001-08-05 10:24:41.000000000 +1200
+++ mozart-1.3.2.20060615/share/bin/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -55,13 +55,13 @@
 install: $(BINDIR) $(LIB_SCRIPTS) $(LIB_LINKS) $(WRAPPERS)
 
 
-$(LIB_SCRIPTS): $(BINDIR)/% : %
+$(LIB_SCRIPTS): $(BINDIR)/% : % $(BINDIR)
 	$(INSTALL_BIN) $< $(BINDIR)
 
-$(LIB_LINKS): $(BINDIR)/oz
+$(LIB_LINKS): $(BINDIR)/oz $(BINDIR)
 	cd $(BINDIR); rm -f $@; $(LN_S) oz $@
 
-$(WRAPPERS): $(BINDIR)/%: $(SRCTOP)/platform/mswindows/%
+$(WRAPPERS): $(BINDIR)/%: $(SRCTOP)/platform/mswindows/% $(BINDIR)
 	$(INSTALL_FILE) $< $@
 
 $(BINDIR):
diff -ur mozart-1.3.2.20060615.orig/share/demo/Makefile.in mozart-1.3.2.20060615/share/demo/Makefile.in
--- mozart-1.3.2.20060615.orig/share/demo/Makefile.in	2001-10-16 23:10:03.000000000 +1300
+++ mozart-1.3.2.20060615/share/demo/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -83,16 +83,19 @@
 install:: $(OZDEMO) $(DEMOAPPLETS) \
 	$(SYSTEMDOC) $(SYSTEMOZF) $(SYSTEMOZF)/MT10.ozf $(SYSTEMDOC)/MT10.oz
 
-$(OZDEMO) $(SYSTEMDOC) $(SYSTEMOZF):
+$(OZDEMO) $(SYSTEMOZF):
 	$(INSTALL_DIR) $@
 
-$(OZDEMO)/%.oza: %.oza
+$(SYSTEMDOC): $(OZDEMO)
+	$(INSTALL_DIR) $@
+
+$(OZDEMO)/%.oza: %.oza $(OZDEMO)
 	$(INSTALL_FILE) $< $@
 
-$(SYSTEMOZF)/%: %
+$(SYSTEMOZF)/%: % $(SYSTEMOZF)
 	$(INSTALL_FILE) $< $@
 
-$(SYSTEMDOC)/%: %
+$(SYSTEMDOC)/%: % $(SYSTEMDOC)
 	$(INSTALL_FILE) $< $@
 
 #
@@ -279,19 +282,21 @@
 
 install:: $(OZIMGDIRS) $(OZIMAGES)
 
-$(OZIMGDIRS):
+$(OZIMG): $(SYSTEMOZF)
+	$(INSTALL_DIR) $@
+$(OZIMGDIRS): $(OZIMG)
 	$(INSTALL_DIR) $@
-$(OZIMG)/animated-queens/%.xbm: images/animated-queens/%.xbm
+$(OZIMG)/animated-queens/%.xbm: images/animated-queens/%.xbm $(OZIMGDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZIMG)/lift/%.xbm: images/lift/%.xbm
+$(OZIMG)/lift/%.xbm: images/lift/%.xbm $(OZIMGDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZIMG)/transport/%.xbm: images/transport/%.xbm
+$(OZIMG)/transport/%.xbm: images/transport/%.xbm $(OZIMGDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZIMG)/trucks/%.ppm: images/trucks/%.ppm
+$(OZIMG)/trucks/%.ppm: images/trucks/%.ppm $(OZIMGDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZIMG)/college/%.xbm: images/college/%.xbm
+$(OZIMG)/college/%.xbm: images/college/%.xbm $(OZIMGDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZIMG)/dict-client/%: images/dict-client/%
+$(OZIMG)/dict-client/%: images/dict-client/% $(OZIMGDIRS)
 	$(INSTALL_FILE) $< $@
 
 
diff -ur mozart-1.3.2.20060615.orig/share/elisp/Makefile.in mozart-1.3.2.20060615/share/elisp/Makefile.in
--- mozart-1.3.2.20060615.orig/share/elisp/Makefile.in	2002-04-02 00:31:22.000000000 +1200
+++ mozart-1.3.2.20060615/share/elisp/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -75,10 +75,10 @@
 $(ELISPDIR):
 	$(INSTALL_DIR) $@
 
-$(ELISPDIR)/%.el: %.el
+$(ELISPDIR)/%.el: %.el $(ELISPDIR)
 	$(INSTALL_FILE) $< $@
 
-$(ELISPDIR)/%.elc: %.elc
+$(ELISPDIR)/%.elc: %.elc $(ELISPDIR)
 	$(INSTALL_FILE) $< $@
 
 clean:
diff -ur mozart-1.3.2.20060615.orig/share/examples/Makefile.in mozart-1.3.2.20060615/share/examples/Makefile.in
--- mozart-1.3.2.20060615.orig/share/examples/Makefile.in	2003-04-23 21:49:55.000000000 +1200
+++ mozart-1.3.2.20060615/share/examples/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -63,15 +63,17 @@
 	$(FUNCTORS:%=$(OZEX)/fd/graphics/%)
 
 EXDIRS = \
-	sampler sampler/constraints \
-	fd grammar grammar/shieber \
+	sampler/constraints \
+	grammar/shieber \
 	fd/graphics
 
 
 OZEXDIRS = $(EXDIRS:%=$(OZEX)/%)
 
 
-$(OZEX) $(OZEXDIRS):
+$(OZEX):
+	$(INSTALL_DIR) $@
+$(OZEXDIRS): $(OZEX)
 	$(INSTALL_DIR) $@
 
 FD = \
@@ -96,19 +98,19 @@
 	constraints/scheduling-compiler constraints oz
 SAMPLER = $(SAMPLER0:%=$(OZEX)/sampler/%.oz)
 
-$(OZEX)/fd/%.oz: fd/%.oz
+$(OZEX)/fd/%.oz: fd/%.oz $(OZEXDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZEX)/fd/graphics/%.ozf: %.ozf
+$(OZEX)/fd/graphics/%.ozf: %.ozf $(OZEXDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZEX)/fd/graphics/%.oz: fd/graphics/%.oz
+$(OZEX)/fd/graphics/%.oz: fd/graphics/%.oz $(OZEXDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZEX)/grammar/%.oz: grammar/%.oz
+$(OZEX)/grammar/%.oz: grammar/%.oz $(OZEXDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZEX)/grammar/shieber/%.oz: grammar/shieber/%.oz
+$(OZEX)/grammar/shieber/%.oz: grammar/shieber/%.oz $(OZEXDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZEX)/sampler/%.oz: sampler/%.oz
+$(OZEX)/sampler/%.oz: sampler/%.oz $(OZEXDIRS)
 	$(INSTALL_FILE) $< $@
-$(OZEX)/sampler/constraints/%.oz: sampler/constraints/%.oz
+$(OZEX)/sampler/constraints/%.oz: sampler/constraints/%.oz $(OZEXDIRS)
 	$(INSTALL_FILE) $< $@
 
 
diff -ur mozart-1.3.2.20060615.orig/share/lib/Makefile.in mozart-1.3.2.20060615/share/lib/Makefile.in
--- mozart-1.3.2.20060615.orig/share/lib/Makefile.in	2003-12-05 10:55:34.000000000 +1300
+++ mozart-1.3.2.20060615/share/lib/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -11,7 +11,7 @@
 PREFIX		= @prefix@
 OZPREFIX	= @prefix@
 
-OZBIN		= $(PREFIX)/bin
+BINDIR		= $(PREFIX)/bin
 OZLIB		= $(PREFIX)/share
 IMAGESDIR	= $(OZLIB)/images
 SHELL		= /bin/sh
@@ -290,7 +290,7 @@
 Builtins.ozf: compiler/Builtins.oz GroundZip.ozf compiler-Builtins.oz
 
 Unnester.ozf: compiler/Unnester.oz compiler/TupleSyntax.oz \
-	compiler/BindingAnalysis.oz compiler/UnnestFD.oz
+	compiler/BindingAnalysis.oz compiler/UnnestFD.oz Builtins.ozf
 
 Annotate.ozf: compiler/Annotate.oz
 
@@ -343,14 +343,14 @@
 	$(FUNCTORS:%=$(OZLIB)/%)
 
 BINFILES = \
-	$(SYSLETS:%=$(OZBIN)/%)
+	$(SYSLETS:%=$(BINDIR)/%)
 
 BOOTINITTXT	= boot-init.ozt
 BOOTOZCTXT	= boot-ozc.ozt
 
 install: install-main install-images install-cache
 
-install-main: all $(OZBIN) $(BINFILES)
+install-main: all $(BINDIR) $(BINFILES)
 
 NEWCACHEDIR = $(PREFIX)/cache/x-oz/system
 IMAGES1 = mini-dec.xbm mini-inc.xbm
@@ -376,7 +376,7 @@
 $(NEWCACHEDIR) $(NEWIMAGESDIR):
 	$(INSTALL_DIR) $@
 
-$(NEWCACHEDIR)/%: %
+$(NEWCACHEDIR)/%: % $(NEWCACHEDIR)
 	$(INSTALL_FILE) $< $@
 
 $(CACHEDIR):
@@ -384,19 +384,19 @@
 	-rm -rf $(CACHEDIR)
 	$(LN_S) ../../.. $(CACHEDIR)
 
-$(OZLIB) $(IMAGESDIR) $(OZBIN):
+$(OZLIB) $(IMAGESDIR) $(BINDIR):
 	$(INSTALL_DIR) $@
 
-$(OZLIB)/%.ozf: %.ozf
+$(OZLIB)/%.ozf: %.ozf $(OZLIB)
 	$(INSTALL_FILE) $< $@
 
-$(IMAGESDIR)/%: images/%
+$(IMAGESDIR)/%: images/% $(IMAGESDIR)
 	$(INSTALL_FILE) $< $@
 
-$(OZBIN)/%: %
+$(BINDIR)/%: % $(BINDIR)
 	$(INSTALL_BIN) $< $@
 
-$(OZLIB)/boot-%: boot-%
+$(OZLIB)/boot-%: boot-% $(OZLIB)
 	$(INSTALL_FILE) $< $@
 
 #---------------------------------------------------------------------
@@ -426,10 +426,10 @@
 .PHONY: bootstrap check
 
 # use the boot compiler to create the functors
-bootstrap:
-	$(MAKE) boot-unzip boot-init boot-ozc boot-all
+bootstrap: boot-unzip
+	$(MAKE) boot-init boot-ozc boot-all
 
-boot-all:
+boot-all: boot-init boot-ozc
 	$(MAKE) all \
 		OZC=$(SRCDIR)/boot-ozc.sh \
 		OZENGINE=$(SRCDIR)/oze.sh \
diff -ur mozart-1.3.2.20060615.orig/share/tools/Makefile.in mozart-1.3.2.20060615/share/tools/Makefile.in
--- mozart-1.3.2.20060615.orig/share/tools/Makefile.in	2002-01-30 01:39:23.000000000 +1300
+++ mozart-1.3.2.20060615/share/tools/Makefile.in	2008-06-02 10:04:05.000000000 +1200
@@ -7,7 +7,7 @@
 PREFIX		= @prefix@
 PLATFORM	= @PLATFORM@
 
-OZBIN		= $(PREFIX)/bin
+BINDIR		= $(PREFIX)/bin
 OZEX		= $(PREFIX)/examples
 OZEXGUMP	= $(OZEX)/gump
 OZLOADSEP	= @OZLOADSEP@
@@ -262,21 +262,21 @@
 #---------------------------------------------------------------------
 
 BINFILES = \
-	$(SYSLETS:%=$(OZBIN)/%)
+	$(SYSLETS:%=$(BINDIR)/%)
 
 install: install-main install-cache
 
 install-main: all \
-	$(OZBIN) $(BINFILES) \
+	$(BINDIR) $(BINFILES) \
 	$(OZEX) $(OZEXGUMP) $(EXAMPLEFILES)
 
-$(OZBIN) $(OZEX) $(OZEXGUMP):
+$(BINDIR) $(OZEX) $(OZEXGUMP):
 	$(INSTALL_DIR) $@
 
-$(OZBIN)/%: %
+$(BINDIR)/%: % $(BINDIR)
 	$(INSTALL_BIN) $< $@
 
-$(OZEXGUMP)/%: gump/examples/%
+$(OZEXGUMP)/%: gump/examples/% $(OZEXGUMP)
 	$(INSTALL_FILE) $< $@
 
 install-cache: \
@@ -288,13 +288,13 @@
 $(CACHEDIR) $(IMAGESDIR) $(OZCARIMAGESDIR) $(INSPECTORIMAGESDIR):
 	$(INSTALL_DIR) $@
 
-$(CACHEDIR)/%: %
+$(CACHEDIR)/%: % $(CACHEDIR)
 	$(INSTALL_FILE) $< $@
 
-$(OZCARIMAGESDIR)/%: ozcar/images/%
+$(OZCARIMAGESDIR)/%: ozcar/images/% $(OZCARIMAGESDIR)
 	$(INSTALL_FILE) $< $@
 
-$(INSPECTORIMAGESDIR)/%: inspector/images/%
+$(INSPECTORIMAGESDIR)/%: inspector/images/% $(INSPECTORIMAGESDIR)
 	$(INSTALL_FILE) $< $@
 
 #---------------------------------------------------------------------
