diff -ur mozart-1.3.2.20060615.orig/contrib/ap/Makefile.in mozart-1.3.2.20060615/contrib/ap/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/ap/Makefile.in	2001-08-05 10:24:30.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/ap/Makefile.in	2007-02-13 21:02:43.000000000 +1300
@@ -21,9 +21,9 @@
 install: $(NEWCACHEDIR) $(addprefix $(NEWCACHEDIR)/,$(FUNCTORS))
 $(LIB_DIR) $(NEWCACHEDIR):
 	$(INSTALL_DIR) $@
-$(LIB_DIR)/%.ozf: %.ozf
+$(LIB_DIR)/%.ozf: %.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
-$(NEWCACHEDIR)/% : %
+$(NEWCACHEDIR)/% : % $(NEWCACHEDIR)
 	$(INSTALL_LIB) $< $@
 
 clean veryclean:
diff -ur mozart-1.3.2.20060615.orig/contrib/compat/Makefile.in mozart-1.3.2.20060615/contrib/compat/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/compat/Makefile.in	2003-11-26 03:53:51.000000000 +1300
+++ mozart-1.3.2.20060615/contrib/compat/Makefile.in	2007-02-14 19:52:50.000000000 +1300
@@ -5,7 +5,7 @@
 VPATH		= @srcdir@
 PREFIX		= @prefix@
 PLATFORM	= @PLATFORM@
-BIN_DIR		= $(PREFIX)/bin
+BINDIR		= $(PREFIX)/bin
 CACHEDIR	= $(PREFIX)/cache/x-oz/contrib/compat
 HOMEURL         = @HOMEURL@
 
@@ -24,7 +24,7 @@
 CYGPATH		= $(SRCTOP)/share/lib/cygpath.sh
 
 TARGETS		= TextPickle.ozf convertTextPickle$(EXE_EXT)
-BIN_TARGETS	= $(BIN_DIR)/convertTextPickle$(EXE_EXT)
+BIN_TARGETS	= $(BINDIR)/convertTextPickle$(EXE_EXT)
 CACHE_TARGETS	= $(CACHEDIR)/TextPickle.ozf \
 		  $(CACHEDIR)/TextPickleScanner.so-$(PLATFORM)
 
@@ -36,18 +36,18 @@
 
 all: $(TARGETS)
 
-install: $(BIN_DIR) $(BIN_TARGETS) $(CACHEDIR) $(CACHE_TARGETS)
+install: $(BINDIR) $(BIN_TARGETS) $(CACHEDIR) $(CACHE_TARGETS)
 
-$(BIN_DIR) $(CACHEDIR):
+$(BINDIR) $(CACHEDIR):
 	$(INSTALL_DIR) $@
 
-$(BIN_DIR)/%: %
+$(BINDIR)/%: % $(BINDIR)
 	$(INSTALL_BIN) $< $@
 
-$(CACHEDIR)/%.ozf: %.ozf
+$(CACHEDIR)/%.ozf: %.ozf $(CACHEDIR)
 	$(INSTALL_FILE) $< $@
 
-$(CACHEDIR)/%.so-$(PLATFORM): %.so-$(PLATFORM)
+$(CACHEDIR)/%.so-$(PLATFORM): %.so-$(PLATFORM) $(CACHEDIR)
 	$(INSTALL_DLL) $< $@
 
 TextPickle.so-$(PLATFORM): TextPickle.ozf
diff -ur mozart-1.3.2.20060615.orig/contrib/davinci/Makefile.in mozart-1.3.2.20060615/contrib/davinci/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/davinci/Makefile.in	2001-08-05 10:24:30.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/davinci/Makefile.in	2007-02-13 21:04:12.000000000 +1300
@@ -59,7 +59,7 @@
 DaVinci.ozf: $(OZFNCTS)
 	$(OZL) -z 9 DaVinciObject.ozf -o $@
 
-$(LIB_DIR)/DaVinci.ozf: DaVinci.ozf
+$(LIB_DIR)/DaVinci.ozf: DaVinci.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 	for f in *.so-$(PLATFORM) ; do \
 	$(INSTALL_DLL) $$f $(LIB_DIR)/$$f ; \
diff -ur mozart-1.3.2.20060615.orig/contrib/directory/Makefile.in mozart-1.3.2.20060615/contrib/directory/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/directory/Makefile.in	2001-08-05 10:24:31.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/directory/Makefile.in	2007-02-14 19:58:11.000000000 +1300
@@ -59,13 +59,13 @@
 $(LIB_DIR) $(NEWCACHEDIR):
 	$(INSTALL_DIR) $@
 
-$(LIB_DIR)/%.ozf: %.ozf
+$(LIB_DIR)/%.ozf: %.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(LIB_DIR)/%.oz: %.oz
+$(LIB_DIR)/%.oz: %.oz $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(NEWCACHEDIR)/%: %
+$(NEWCACHEDIR)/%: % $(NEWCACHEDIR)
 	$(INSTALL_LIB) $< $@
 
 #----------------------------------------------------------------------
diff -ur mozart-1.3.2.20060615.orig/contrib/doc/code/Makefile.in mozart-1.3.2.20060615/contrib/doc/code/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/doc/code/Makefile.in	2002-04-02 00:28:57.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/doc/code/Makefile.in	2007-02-13 21:06:46.000000000 +1300
@@ -57,13 +57,13 @@
 $(LIB_DIR) $(ELIB_DIR) $(NEWCACHEDIR):
 	$(INSTALL_DIR) $@
 
-$(LIB_TARGETS): $(LIB_DIR)/% : %
+$(LIB_TARGETS): $(LIB_DIR)/% : % $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(NEW_TARGETS): $(NEWCACHEDIR)/% : %
+$(NEW_TARGETS): $(NEWCACHEDIR)/% : % $(NEWCACHEDIR)
 	$(INSTALL_LIB) $< $@
 
-$(ELIB_TARGETS): $(ELIB_DIR)/% : %
+$(ELIB_TARGETS): $(ELIB_DIR)/% : % $(ELIB_DIR)
 	$(INSTALL_LIB) $< $@
 
 %.ozf: %.oz
diff -ur mozart-1.3.2.20060615.orig/contrib/doc/sgml/Makefile.in mozart-1.3.2.20060615/contrib/doc/sgml/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/doc/sgml/Makefile.in	2001-08-05 10:24:31.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/doc/sgml/Makefile.in	2007-02-13 21:05:58.000000000 +1300
@@ -25,10 +25,10 @@
 $(LIB_DIR) $(NEWCACHEDIR):
 	$(INSTALL_DIR) $@
 
-$(LIB_TARGETS): $(LIB_DIR)/% : %
+$(LIB_TARGETS): $(LIB_DIR)/% : % $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(NEW_TARGETS): $(NEWCACHEDIR)/% : %
+$(NEW_TARGETS): $(NEWCACHEDIR)/% : % $(NEWCACHEDIR)
 	$(INSTALL_LIB) $< $@
 
 %.ozf: %.oz
diff -ur mozart-1.3.2.20060615.orig/contrib/fcp/Makefile.in mozart-1.3.2.20060615/contrib/fcp/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/fcp/Makefile.in	2001-08-05 10:24:32.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/fcp/Makefile.in	2007-02-13 21:07:28.000000000 +1300
@@ -77,10 +77,10 @@
 $(LIB_DIR):
 	$(INSTALL_DIR) $@
 
-$(LIB_DIR)/%.ozf: %.ozf
+$(LIB_DIR)/%.ozf: %.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(LIB_DIR)/%.so-$(PLATFORM): %.so-$(PLATFORM)
+$(LIB_DIR)/%.so-$(PLATFORM): %.so-$(PLATFORM) $(LIB_DIR)
 	$(INSTALL_DLL) $< $@
 
 #----------------------------------------------------------------------
diff -ur mozart-1.3.2.20060615.orig/contrib/gdbm/Makefile.in mozart-1.3.2.20060615/contrib/gdbm/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/gdbm/Makefile.in	2003-01-08 02:40:09.000000000 +1300
+++ mozart-1.3.2.20060615/contrib/gdbm/Makefile.in	2007-02-13 21:09:17.000000000 +1300
@@ -39,16 +39,16 @@
 $(LIB_DIR) $(NEWCACHEDIR):
 	$(INSTALL_DIR) $@
 
-$(LIB_DIR)/gdbm.ozf: gdbm.ozf
+$(LIB_DIR)/gdbm.ozf: gdbm.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(LIB_DIR)/gdbm.so-$(PLATFORM): gdbm.so-$(PLATFORM)
+$(LIB_DIR)/gdbm.so-$(PLATFORM): gdbm.so-$(PLATFORM) $(LIB_DIR)
 	$(INSTALL_DLL) $< $@
 
-$(NEWCACHEDIR)/gdbm.ozf: gdbm.ozf
+$(NEWCACHEDIR)/gdbm.ozf: gdbm.ozf $(NEWCACHEDIR)
 	$(INSTALL_LIB) $< $@
 
-$(NEWCACHEDIR)/gdbm.so-$(PLATFORM): gdbm.so-$(PLATFORM)
+$(NEWCACHEDIR)/gdbm.so-$(PLATFORM): gdbm.so-$(PLATFORM) $(NEWCACHEDIR)
 	$(INSTALL_DLL) $< $@
 
 gdbm.ozf: gdbm.oz
diff -ur mozart-1.3.2.20060615.orig/contrib/investigator/Makefile.in mozart-1.3.2.20060615/contrib/investigator/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/investigator/Makefile.in	2001-08-05 10:24:33.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/investigator/Makefile.in	2007-02-13 21:09:53.000000000 +1300
@@ -70,7 +70,7 @@
 Investigator.ozf: $(OZFNCTS)
 	$(OZL) -z 9 Main.ozf -o $@
 
-$(LIB_DIR)/Investigator.ozf: Investigator.ozf
+$(LIB_DIR)/Investigator.ozf: Investigator.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
 #----------------------------------------------------------------------
diff -ur mozart-1.3.2.20060615.orig/contrib/micq/Makefile.in mozart-1.3.2.20060615/contrib/micq/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/micq/Makefile.in	2001-08-05 10:24:34.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/micq/Makefile.in	2007-02-13 21:10:50.000000000 +1300
@@ -28,14 +28,14 @@
 install:: $(DEMO_DIR) $(DEMO_DIR)/MIM.oza \
 	$(NEWCACHEDIR) $(addprefix $(NEWCACHEDIR)/,$(APPLETS))
 
-$(NEWCACHEDIR)/% : %
+$(NEWCACHEDIR)/% : % $(NEWCACHEDIR)
 	$(INSTALL_LIB) $< $@
 
 $(LIB_DIR) $(DEMO_DIR) $(NEWCACHEDIR):
 	$(INSTALL_DIR) $@
-$(LIB_DIR)/%.oza: %.oza
+$(LIB_DIR)/%.oza: %.oza $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
-$(DEMO_DIR)/MIM.oza: client.oza
+$(DEMO_DIR)/MIM.oza: client.oza $(DEMO_DIR)
 	$(INSTALL_LIB) $< $@
 BMETH = $(BUILDTOP)/contrib/micq/methods.ozf
 SMETH = $(SRCDIR)/methods.oz
diff -ur mozart-1.3.2.20060615.orig/contrib/os/Makefile.in mozart-1.3.2.20060615/contrib/os/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/os/Makefile.in	2001-09-29 01:45:43.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/os/Makefile.in	2007-02-13 21:11:44.000000000 +1300
@@ -52,16 +52,16 @@
 $(LIB_DIR) $(NEWCACHEDIR):
 	$(INSTALL_DIR) $@
 
-$(LIB_FUNCTORS): $(LIB_DIR)/% : %
+$(LIB_FUNCTORS): $(LIB_DIR)/% : % $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(LIB_NATIVES): $(LIB_DIR)/%.so-$(PLATFORM): %.so-$(PLATFORM)
+$(LIB_NATIVES): $(LIB_DIR)/%.so-$(PLATFORM): %.so-$(PLATFORM) $(LIB_DIR)
 	$(INSTALL_DLL) $< $@
 
-$(NEW_FUNCTORS): $(NEWCACHEDIR)/% : %
+$(NEW_FUNCTORS): $(NEWCACHEDIR)/% : % $(NEWCACHEDIR)
 	$(INSTALL_LIB) $< $@
 
-$(NEW_NATIVES): $(NEWCACHEDIR)/%.so-$(PLATFORM): %.so-$(PLATFORM)
+$(NEW_NATIVES): $(NEWCACHEDIR)/%.so-$(PLATFORM): %.so-$(PLATFORM) $(NEWCACHEDIR)
 	$(INSTALL_DLL) $< $@
 
 %.ozf: %.oz
diff -ur mozart-1.3.2.20060615.orig/contrib/reflect/Makefile.in mozart-1.3.2.20060615/contrib/reflect/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/reflect/Makefile.in	2001-08-05 10:24:34.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/reflect/Makefile.in	2007-02-13 21:12:13.000000000 +1300
@@ -67,10 +67,10 @@
 $(LIB_DIR):
 	$(INSTALL_DIR) $@
 
-$(LIB_DIR)/%.ozf: %.ozf
+$(LIB_DIR)/%.ozf: %.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(LIB_DIR)/%.so-$(PLATFORM): %.so-$(PLATFORM)
+$(LIB_DIR)/%.so-$(PLATFORM): %.so-$(PLATFORM) $(LIB_DIR)
 	$(INSTALL_DLL) $< $@
 
 #----------------------------------------------------------------------
diff -ur mozart-1.3.2.20060615.orig/contrib/regex/Makefile.in mozart-1.3.2.20060615/contrib/regex/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/regex/Makefile.in	2003-01-08 02:40:10.000000000 +1300
+++ mozart-1.3.2.20060615/contrib/regex/Makefile.in	2007-02-13 21:13:05.000000000 +1300
@@ -38,16 +38,16 @@
 $(LIB_DIR):
 	$(INSTALL_DIR) $@
 
-$(LIB_DIR)/regex.ozf: regex.ozf
+$(LIB_DIR)/regex.ozf: regex.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(LIB_DIR)/regex.so-$(PLATFORM): regex.so-$(PLATFORM)
+$(LIB_DIR)/regex.so-$(PLATFORM): regex.so-$(PLATFORM) $(LIB_DIR)
 	$(INSTALL_DLL) $< $@
 
-$(NEWCACHEDIR)/regex.ozf: regex.ozf
+$(NEWCACHEDIR)/regex.ozf: regex.ozf $(NEWCACHEDIR)
 	$(INSTALL_LIB) $< $@
 
-$(NEWCACHEDIR)/regex.so-$(PLATFORM): regex.so-$(PLATFORM)
+$(NEWCACHEDIR)/regex.so-$(PLATFORM): regex.so-$(PLATFORM) $(NEWCACHEDIR)
 	$(INSTALL_DLL) $< $@
 
 regex.ozf: regex.oz
diff -ur mozart-1.3.2.20060615.orig/contrib/ri/Makefile.in mozart-1.3.2.20060615/contrib/ri/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/ri/Makefile.in	2002-07-07 01:05:36.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/ri/Makefile.in	2007-02-13 21:13:39.000000000 +1300
@@ -81,10 +81,10 @@
 $(LIB_DIR):
 	$(INSTALL_DIR) $@
 
-$(LIB_DIR)/%.ozf: %.ozf
+$(LIB_DIR)/%.ozf: %.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
 
-$(LIB_DIR)/%.so-$(PLATFORM): %.so-$(PLATFORM)
+$(LIB_DIR)/%.so-$(PLATFORM): %.so-$(PLATFORM) $(LIB_DIR)
 	$(INSTALL_DLL) $< $@
 
 
diff -ur mozart-1.3.2.20060615.orig/contrib/tk/Makefile.in mozart-1.3.2.20060615/contrib/tk/Makefile.in
--- mozart-1.3.2.20060615.orig/contrib/tk/Makefile.in	2001-08-05 10:24:35.000000000 +1200
+++ mozart-1.3.2.20060615/contrib/tk/Makefile.in	2007-02-13 21:14:06.000000000 +1300
@@ -21,9 +21,9 @@
 install:	$(NEWCACHEDIR) $(addprefix $(NEWCACHEDIR)/,$(FUNCTORS))
 $(LIB_DIR) $(NEWCACHEDIR):
 	$(INSTALL_DIR) $@
-$(LIB_DIR)/%.ozf: %.ozf
+$(LIB_DIR)/%.ozf: %.ozf $(LIB_DIR)
 	$(INSTALL_LIB) $< $@
-$(NEWCACHEDIR)/%.ozf: %.ozf
+$(NEWCACHEDIR)/%.ozf: %.ozf $(NEWCACHEDIR)
 	$(INSTALL_LIB) $< $@
 
 clean veryclean:
