diff -ruN rekonq-0.6.1.orig/src/tests/protocolhandler_test.cpp rekonq-0.6.1/src/tests/protocolhandler_test.cpp
--- rekonq-0.6.1.orig/src/tests/protocolhandler_test.cpp	2010-10-05 16:03:58.000000000 +0200
+++ rekonq-0.6.1/src/tests/protocolhandler_test.cpp	2011-01-14 23:37:27.000000000 +0100
@@ -23,6 +23,8 @@
 
 #include "protocolhandler.h"
 
+#include <QWebView>
+#include <QNetworkRequest>
 
 class ProtocolhandlerTest : public QObject
 {
