--- configure.in.in	2004-08-05 13:53:14.000000000 -0500
+++ configure.in.in	2004-08-26 14:15:51.901070600 -0500
@@ -697,6 +697,12 @@
 AC_SUBST(GLIB_LIBADD)
 AC_SUBST(GLIB_LDFLAGS)
 
+dnl Check jack
+AC_ARG_ENABLE(jack,
+[  --disable-jack             disable support for Jack Audio Connection Kit],
+[arts_with_jack="$enableval"],[arts_with_jack=yes])
+
+if test "x$arts_with_jack" = "xyes"; then
 AC_MSG_CHECKING(for Jack Audio Connection Kit)
 if $PKG_CONFIG --atleast-version 0.90 jack >/dev/null 2>&1 ; then
   JACK_CFLAGS="`$PKG_CONFIG --cflags jack`"
@@ -708,6 +714,8 @@
 else
   AC_MSG_RESULT(not installed)
 fi
+fi
+
 AC_SUBST(JACK_CFLAGS)
 AC_SUBST(JACK_LIBADD)
 AC_SUBST(JACK_LDFLAGS)
