--- arts/configure.in.in	2003/08/28 17:28:59	1.94
+++ arts/configure.in.in	2003/11/24 20:13:09	1.95
@@ -549,6 +549,26 @@ AC_DEFUN([AC_CHECK_LIBASOUND],
       #include <alsa/asoundlib.h>
       #endif
     ],[
+      #if (SND_LIB_MAJOR == 1) && (SND_LIB_MINOR == 0)
+        /* we have ALSA 1.x */
+      #else
+        #error not ALSA 1.x
+      #endif
+    ],
+    kde_has_alsa_1_0=yes,
+    kde_has_alsa_1_0=no)
+  fi
+
+  if test "x$kde_has_asoundlib" = "xyes"; then
+    AC_TRY_COMPILE([
+      #include "confdefs.h"
+      #ifdef HAVE_SYS_ASOUNDLIB_H
+      #include <sys/asoundlib.h>
+      #endif
+      #ifdef HAVE_ALSA_ASOUNDLIB_H
+      #include <alsa/asoundlib.h>
+      #endif
+    ],[
     #if (SND_LIB_MAJOR == 0) && (SND_LIB_MINOR == 5)
       /* we have ALSA 0.5.x */
     #else
@@ -575,6 +595,18 @@ AC_DEFUN([AC_CHECK_LIBASOUND],
           AC_DEFINE(HAVE_SND_PCM_RESUME, 1,
             [Define if libasound has snd_pcm_resume()])])
       fi
+      if test "x$kde_has_alsa_1_0" = "xyes"; then
+        LIBASOUND="-lasound"
+        AC_DEFINE(HAVE_LIBASOUND2, 1,
+          [Define if you have libasound.so.2 (required for ALSA 0.9.x/1.x support)])
+	AC_DEFINE(ALSA_PCM_OLD_SW_PARAMS_API, 1,
+          [Define if you have alsa 1.x])
+	AC_DEFINE(ALSA_PCM_OLD_HW_PARAMS_API, 1,
+          [Define if you have alsa 1.x])
+        AC_CHECK_LIB(asound,snd_pcm_resume,[
+          AC_DEFINE(HAVE_SND_PCM_RESUME, 1,
+            [Define if libasound has snd_pcm_resume()])])
+      fi
     ])
   fi
   AC_SUBST(LIBASOUND)
