diff -ruN dhcdbd-1.14.orig/dhcdbd.c dhcdbd-1.14/dhcdbd.c
--- dhcdbd-1.14.orig/dhcdbd.c	2006-05-05 11:25:35.000000000 -0500
+++ dhcdbd-1.14/dhcdbd.c	2006-05-05 23:40:18.000000000 -0500
@@ -259,7 +259,7 @@
 static void state_trans(DHC_IF* d_if, uint32_t next_state)
 {
     /* Uncomment to debug dhclient process state changes */
-/*    
+    
     char cs[16], ns[16];
     if( d_if->state == next_state)
 	return;
@@ -315,7 +315,7 @@
                  	           ? "RELEASER"	       
 	                           : ns
 	);	   
-*/
+
 }
 
 void dhcdbd_monitor( const void *p, const VISIT which, const int level )
