Index: softdevice-cvs/setup-softdevice.h
===================================================================
--- softdevice-cvs.orig/setup-softdevice.h
+++ softdevice-cvs/setup-softdevice.h
@@ -175,6 +175,8 @@ struct cSetupStore {
           setSourceRectangleLocked;
     char  alsaDevice [ALSA_DEVICE_NAME_LENGTH];
     char  alsaAC3Device [ALSA_DEVICE_NAME_LENGTH];
+
+    int   setupStoreShmid;
 };
 
 #define OSDMODE_PSEUDO    0
Index: softdevice-cvs/softdevice.c
===================================================================
--- softdevice-cvs.orig/softdevice.c
+++ softdevice-cvs/softdevice.c
@@ -850,6 +850,7 @@ cPluginSoftDevice::cPluginSoftDevice(voi
           ctl=NULL;
   };
   setupStore->InitSetupStore();
+  setupStore->setupStoreShmid = setupStoreShmId;
 #endif
 }
 
Index: softdevice-cvs/video-shm.c
===================================================================
--- softdevice-cvs.orig/video-shm.c
+++ softdevice-cvs/video-shm.c
@@ -91,6 +91,7 @@ cShmVideoOut::cShmVideoOut(cSetupStore *
                 ctl->pict_shmid=-1;
                 ctl->osd_shmid=-1;
                 ctl->attached = 0;
+                ctl->setup_shmid = setupStore->setupStoreShmid;
         };
 
         if ( ctl->semid == -1 ) {
