--- src/Makefile.am.orig	2010-02-10 15:29:43.000000000 +0100
+++ src/Makefile.am	2010-02-16 16:47:22.000000000 +0100
@@ -1,7 +1,7 @@
 #bin_PROGRAMS = tucnak soundwrapper stucnak
-bin_PROGRAMS = tucnak soundwrapper 
+bin_PROGRAMS = tucnak2 soundwrapper 
 
-tucnak_SOURCES = adif.c alsa.c bfu.c cabrillo.c charsets.c \
+tucnak2_SOURCES = adif.c alsa.c bfu.c cabrillo.c charsets.c \
     codepage.h codepage.inc \
 	control.c cor.c cor.h cordata.c cordata.h cordata.inc \
 	cwdaemon.c cwdb.c cwwindow.c davac4.c dsp.c dwdb.c dxc.c ebw.c edi.c \
@@ -17,7 +17,7 @@
 	ssbd.c stats.c subwin.c terminal.c threadpipe.c titlpage.c trig.c \
 	ttys.c txts.c txts.h txts.inc uni_7b.inc winkey.c wizz.c zstring.c
 
-tucnak_LDADD = @TUCNAK_LDADD@
+tucnak2_LDADD = @TUCNAK_LDADD@
 
 soundwrapper_SOURCES = soundwrapper.c
 
