--- rpm-4.4.6/rpmio/url.c.orig	2007-01-20 16:29:35.471962977 +0100
+++ rpm-4.4.6/rpmio/url.c	2007-01-20 16:30:34.438967191 +0100
@@ -28,6 +28,8 @@
 #define	IPPORT_PGPKEYSERVER	11371
 #endif

+#include <neon/ne_session.h>
+
 /**
  */
 /*@unchecked@*/

--- rpm-4.4.6/lib/rpmfi.c.orig	2007-01-20 18:11:30.186124993 +0100
+++ rpm-4.4.6/lib/rpmfi.c	2007-01-20 18:13:26.938313715 +0100
@@ -24,6 +24,7 @@

 #include "misc.h"	/* XXX stripTrailingChar */
 #include "rpmmacro.h"	/* XXX rpmCleanPath */
+#include "legacy.h" /* XXX dodigest */

 #include "debug.h"
