--- rexml.old/bin/install.rb	Sun Jun 16 22:43:05 2002
+++ rexml/bin/install.rb	Sun Jun 16 22:44:39 2002
@@ -9,8 +9,13 @@
 
 SRC = 'rexml'
 
-INSTDIR = File.join Config::CONFIG['sitedir'], 
+if ENV["PORTAGETMP"]
+  INSTDIR = File.join ENV["PORTAGETMP"], Config::CONFIG['sitedir'],
 	"#{Config::CONFIG['MAJOR']}.#{Config::CONFIG['MINOR']}"
+else
+  INSTDIR = File.join Config::CONFIG['sitedir'],
+    "#{Config::CONFIG['MAJOR']}.#{Config::CONFIG['MINOR']}"
+end
 DESTDIR = File.join INSTDIR, SRC
 
 opts = GetoptLong.new( [ "--uninstall",	"-u",		GetoptLong::NO_ARGUMENT ] )
