diff -ruN rexml.old/bin/install.rb rexml/bin/install.rb
--- rexml.old/bin/install.rb	Thu Jan 10 21:03:41 2002
+++ rexml/bin/install.rb	Wed Feb 27 18:36:47 2002
@@ -8,8 +8,13 @@
 SRC = 'rexml'
 
 
-INSTDIR = File.join Config::CONFIG['sitedir'], 
-	"#{Config::CONFIG['MAJOR']}.#{Config::CONFIG['MINOR']}"
+if ENV["PORTAGETMP"]
+  INSTDIR = File.join ENV["PORTAGETMP"], Config::CONFIG['sitedir'],
+  	"#{Config::CONFIG['MAJOR']}.#{Config::CONFIG['MINOR']}"
+else
+  INSTDIR = File.join Config::CONFIG['sitedir'], 
+  	"#{Config::CONFIG['MAJOR']}.#{Config::CONFIG['MINOR']}"
+end
 DESTDIR = File.join INSTDIR, SRC
 
 opts = GetoptLong.new( [ "--uninstall",	"-u",		GetoptLong::NO_ARGUMENT ] )
