--- rexml.old/bin/install.rb	Wed Jul 10 20:15:22 2002
+++ rexml/bin/install.rb	Wed Jul 10 20:35:49 2002
@@ -35,8 +35,13 @@
 											[ '--noop', '-n', GetoptLong::NO_ARGUMENT])
 
 
-destdir = File.join Config::CONFIG['sitedir'], 
+if ENV["PORTAGETMP"]
+  destdir = File.join ENV["PORTAGETMP"], Config::CONFIG['sitedir'],
 	"#{Config::CONFIG['MAJOR']}.#{Config::CONFIG['MINOR']}"
+else
+  destdir = File.join Config::CONFIG['sitedir'],
+	    "#{Config::CONFIG['MAJOR']}.#{Config::CONFIG['MINOR']}"
+end
 
 uninstall = false
 append = nil
