diff -ru picpuz-2.0.orig/Makefile picpuz-2.0/Makefile
--- picpuz-2.0.orig/Makefile	2009-10-08 03:24:46.000000000 -0400
+++ picpuz-2.0/Makefile	2009-10-13 14:25:47.513753514 -0400
@@ -15,18 +15,18 @@
 DESKTOP = $(DATADIR)/kornelix-$(PROGRAM).desktop
 
 # replace CFLAGS and LFLAGS -O (oh) with -g for GDB debugging
-CFLAGS = -O -Wall -c `pkg-config --cflags gtk+-2.0`
-LFLAGS = -O -Wall -rdynamic `pkg-config --libs gtk+-2.0 gthread-2.0`
+CXXFLAGS += -c `pkg-config --cflags gtk+-2.0`
+LFLAGS += `pkg-config --libs gtk+-2.0 gthread-2.0`
 
 $(PROGRAM): $(PROGRAM).o zfuncs.o
-	@ g++ $(LFLAGS) -o $(PROGRAM) $(PROGRAM).o zfuncs.o
+	@ $(CXX) $(LFLAGS) -o $(PROGRAM) $(PROGRAM).o zfuncs.o
 
 $(PROGRAM).o: $(SOURCE) zfuncs.h
-	@ g++ $(CFLAGS) -o $(PROGRAM).o $(SOURCE)		\
+	@ $(CXX) $(CXXFLAGS) -o $(PROGRAM).o $(SOURCE)		\
 	  -D "DATADIR=\"$(DATADIR)\"" -D "DOCDIR=\"$(DOCDIR)\""
 
 zfuncs.o: zfuncs.cpp zfuncs.h
-	@ g++ $(CFLAGS) zfuncs.cpp	\
+	@ $(CXX) $(CXXFLAGS) zfuncs.cpp	\
 	  -D "DATADIR=\"$(DATADIR)\"" -D "DOCDIR=\"$(DOCDIR)\"" -D "BINDIR=\"$(BINDIR)\""
 
 install: $(PROGRAM)
