diff -urN alsamixergui-0.9.0rc1-2/configure alsamixergui-0.9.0rc1-2-modified/configure
--- alsamixergui-0.9.0rc1-2/configure	2002-05-03 04:40:38.000000000 -0500
+++ alsamixergui-0.9.0rc1-2-modified/configure	2003-01-02 19:35:55.000000000 -0600
@@ -1468,10 +1468,10 @@
   echo "$ac_t""no" 1>&6
 fi
 
-
-echo $ac_n "checking for numericsort in -lfltk""... $ac_c" 1>&6
-echo "configure:1474: checking for numericsort in -lfltk" >&5
-ac_lib_var=`echo fltk'_'numericsort | sed 'y%./+-%__p_%'`
+ac_link='${CXX-g++} -o conftest${ac_exeext} $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
+echo $ac_n "checking for main in -lfltk""... $ac_c" 1>&6
+echo "configure:1474: checking for main in -lfltk" >&5
+ac_lib_var=`echo fltk'_'main | sed 'y%./+-%__p_%'`
 if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
@@ -1481,12 +1481,12 @@
 #line 1482 "configure"
 #include "confdefs.h"
 /* Override any gcc2 internal prototype to avoid an error.  */
-/* We use char because int might match the return type of a gcc2
-    builtin and then its argument prototype would still apply.  */
-char numericsort();
+/* We use char because int might match the return type of a gcc2 */
+/* builtin and then its argument prototype would still apply.  */
+/* char main(); */
 
 int main() {
-numericsort()
+main()
 ; return 0; }
 EOF
 if { (eval echo configure:1493: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
@@ -1656,10 +1656,10 @@
 ac_link='${CC-cc} -o conftest${ac_exeext} $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
 cross_compiling=$ac_cv_prog_cc_cross
 
-
-echo $ac_n "checking for snd_seq_create_event in -lasound""... $ac_c" 1>&6
-echo "configure:1662: checking for snd_seq_create_event in -lasound" >&5
-ac_lib_var=`echo asound'_'snd_seq_create_event | sed 'y%./+-%__p_%'`
+ac_link='${CXX-g++} -o conftest${ac_exeext} $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
+echo $ac_n "checking for main in -lasound""... $ac_c" 1>&6
+echo "configure:1662: checking for main in -lasound" >&5
+ac_lib_var=`echo asound'_'main | sed 'y%./+-%__p_%'`
 if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
@@ -1669,12 +1669,12 @@
 #line 1670 "configure"
 #include "confdefs.h"
 /* Override any gcc2 internal prototype to avoid an error.  */
-/* We use char because int might match the return type of a gcc2
-    builtin and then its argument prototype would still apply.  */
-char snd_seq_create_event();
+/* We use char because int might match the return type of a gcc2 */
+/* builtin and then its argument prototype would still apply.  */
+/* char main(); */
 
 int main() {
-snd_seq_create_event()
+main()
 ; return 0; }
 EOF
 if { (eval echo configure:1681: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
