diff -ru osdpip-0.0.8-orig/receiver.h osdpip-0.0.8/receiver.h
--- osdpip-0.0.8-orig/receiver.h	2006-07-06 15:34:18.711196142 +0200
+++ osdpip-0.0.8/receiver.h	2006-07-06 15:35:25.601662314 +0200
@@ -28,7 +28,7 @@
 	virtual void Action(void);
 
 public:
-	cOsdPipReceiver::cOsdPipReceiver(const cChannel *Channel, 
+	cOsdPipReceiver(const cChannel *Channel, 
 			cRingBufferFrame *ESBuffer);
 	virtual ~cOsdPipReceiver();
 };
