# fix gcc-4.3 missing headers 

Index: qtiplot-0.9.7.4/3rdparty/liborigin/OriginObj.h
===================================================================
--- qtiplot-0.9.7.4.orig/3rdparty/liborigin/OriginObj.h
+++ qtiplot-0.9.7.4/3rdparty/liborigin/OriginObj.h
@@ -33,6 +33,7 @@
 
 #include <string>
 #include <vector>
+#include <cstring>
 #include "boost/variant.hpp"
 #include "boost/date_time/posix_time/ptime.hpp"
 
