=== modified file 'qtiplot/qtiplot.pro'
--- qtiplot/qtiplot.pro	2010-01-06 10:51:53 +0000
+++ qtiplot/qtiplot.pro	2010-01-06 18:37:25 +0000
@@ -12,7 +12,6 @@
 INCLUDEPATH       += ../3rdparty/qwtplot3d/include
 INCLUDEPATH       += ../3rdparty/liborigin
 INCLUDEPATH       += ../3rdparty/zlib
-INCLUDEPATH 	  += ../3rdparty/QTeXEngine/src
 
 # configurable
 INCLUDEPATH       += $$MUPARSER_INCLUDEPATH
@@ -144,14 +143,6 @@
 include(src/table/table.pri)
 include(src/scripting/scripting.pri)
 
-###############################################################
-##################### TeX export: QTeXEngine ##################
-###############################################################
-
-INCLUDEPATH += ../3rdparty/QTeXEngine/src
-HEADERS 	+= ../3rdparty/QTeXEngine/src/QTeXEngine.h
-SOURCES     += ../3rdparty/QTeXEngine/src/QTeXPaintEngine.cpp
-SOURCES     += ../3rdparty/QTeXEngine/src/QTeXPaintDevice.cpp
 
 ###############################################################
 ##################### Scripting: PYTHON + SIP + PyQT ##########

