--- qtiplot-0.7.6.pro.orig	2006-01-16 22:43:32.000000000 +0000
+++ qtiplot-0.7.6.pro	2006-01-16 22:48:18.000000000 +0000
@@ -15,16 +15,14 @@
 
 win32:INCLUDEPATH += C:/Qt/WinGsl
 
-INCLUDEPATH += ../3rdparty/qwt/include
-INCLUDEPATH += ../3rdparty/qwtplot3d/include
+INCLUDEPATH += /usr/include/qwt
+INCLUDEPATH += /usr/include/qwtplot3d
 
 win32:LIBS     += ../3rdparty/qwt/lib/qwt.lib 
 win32:LIBS     += ../3rdparty/qwtplot3d/lib/qwtplot3d.lib 
 win32:LIBS     += C:/Qt/WinGsl/Lib/WinGsl.lib 
 
-unix:LIBS += -L../3rdparty/qwt/lib -lqwt
-unix:LIBS += -L../3rdparty/qwtplot3d/lib -lqwtplot3d
-unix:LIBS += -L /usr/lib -lgsl -lgslcblas
+unix:LIBS += -L_LIBDIR_ -lgsl -lgslcblas -lqwt -lqwtplot3d
 
 unix:target.path=/usr/bin
 unix:INSTALLS += target
@@ -194,4 +192,4 @@
 
 #Origin Import (liborigin)
 HEADERS+=../3rdparty/liborigin/OPJFile.h   
-SOURCES+=../3rdparty/liborigin/OPJFile.cpp
\ No newline at end of file
+SOURCES+=../3rdparty/liborigin/OPJFile.cpp
