Index: qtiplot-0.9.7.4/qtiplot/src/scripting/qti.sip
===================================================================
--- qtiplot-0.9.7.4.orig/qtiplot/src/scripting/qti.sip
+++ qtiplot-0.9.7.4/qtiplot/src/scripting/qti.sip
@@ -74,8 +74,6 @@ private:
   MdiSubWindow(const MdiSubWindow&);
 };
 
-typedef QList<MdiSubWindow*> MDIWindowList;
-
 class Table: MdiSubWindow
 {
 %TypeHeaderCode
@@ -1072,7 +1070,7 @@ public:
     Graph3D* plotParametricSurface(const QString&, const QString&, const QString&,
             double, double, double, double, int=40, int=40, bool=true, bool=true) /PyName=plot3D/;
 
-  MDIWindowList windowsList() /PyName=windows/;
+  QList<MdiSubWindow*> windowsList() /PyName=windows/;
 
   // folders
   Folder *activeFolder() /NoDerived/;
@@ -1867,7 +1865,7 @@ class Folder : QObject
 #include "src/core/Folder.h"
 %End
 public:
-  MDIWindowList windowsList() /PyName=windows/;
+  QList<MdiSubWindow*> windowsList() /PyName=windows/;
 // TODO: implement signal Folder::nameChanged and make it update the project explorer; adjust renaming from GUI accordingly
 //  void setFolderName(const QString&) /PyName=setName/;
   QString name();
