Index: kopete/protocols/oscar/liboscar/oscartypes.h
===================================================================
--- kopete/protocols/oscar/liboscar/oscartypes.h	(revision 560989)
+++ kopete/protocols/oscar/liboscar/oscartypes.h	(revision 560990)
@@ -237,9 +237,9 @@
 	const char ICQ_CLIENTSTRING[] 		= "ICQBasic";
 	const WORD ICQ_CLIENTID 		= 0x010A;
 	const WORD ICQ_MAJOR 			= 0x0014;
-	const WORD ICQ_MINOR 			= 0x0022;
+	const WORD ICQ_MINOR 			= 0x0034;
 	const WORD ICQ_POINT 			= 0x0000;
-	const WORD ICQ_BUILD 			= 0x0911;
+	const WORD ICQ_BUILD 			= 0x0A0F;
 	const char ICQ_OTHER[] 			= { 0x00, 0x00, 0x04, 0x3d };
 	const char ICQ_COUNTRY[] 		= "us";
 	const char ICQ_LANG[] 			= "en";
