--- kopete/protocols/icq_new/icqprotocol.cpp.orig	2002-11-04 12:49:11.000000000 +0100
+++ kopete/protocols/icq_new/icqprotocol.cpp	2002-11-04 12:52:52.000000000 +0100
@@ -86,17 +86,17 @@
 
 	engine->owner->Uin = mPrefs->uin().toULong();
 	engine->storePassword( mPrefs->passwd() );
-	engine->ServerHost = mPrefs->serverAdress();
+	engine->ServerHost = mPrefs->serverAdress().latin1();
 	engine->ServerPort = mPrefs->serverPort();
 
 	if ( mPrefs->useProxy() )
 	{
 		engine->ProxyType = mPrefs->proxyType();
-		engine->ProxyHost = mPrefs->proxyServerAdress();
+		engine->ProxyHost = mPrefs->proxyServerAdress().latin1();
 		engine->ProxyPort = mPrefs->proxyPort();
 		engine->ProxyAuth = mPrefs->proxyAuth();
-		engine->ProxyUser = mPrefs->proxyAuthUser();
-		engine->ProxyPasswd = mPrefs->proxyAuthPassword();
+		engine->ProxyUser = mPrefs->proxyAuthUser().latin1();
+		engine->ProxyPasswd = mPrefs->proxyAuthPassword().latin1();
 	}
 	else
 	{
@@ -118,7 +118,7 @@
 
 	ICQUser *mySelf = static_cast<ICQUser*>(engine->owner); //new ICQUser();
 	mySelf->Uin = mPrefs->uin().toULong();
-	mySelf->Nick = mPrefs->nickName();
+	mySelf->Nick = mPrefs->nickName().latin1();
 	engine->setInfo(mySelf);
 	KopeteMetaContact *meta = new KopeteMetaContact();
 	QString protocolId = id();
@@ -266,8 +266,8 @@
 		}
 */
 
-		user->Alias = alias;
-		user->Nick = alias; // initial nickname setting
+		user->Alias = alias.latin1();
+		user->Nick = alias.latin1(); // initial nickname setting
 
 		ICQContact *contact = new ICQContact(protocolId, metaContact, user, this);
 		KopeteContact *c = contact;
@@ -395,7 +395,7 @@
 	}
 
 	if ( !alias.isEmpty() && QString(user->Alias.c_str()).isEmpty() )
-		user->Alias = QString(alias);
+		user->Alias = QString(alias).latin1();
 
 	QString protocolId = this->id();
 	ICQContact *contact = new ICQContact( protocolId, meta, user, this);
@@ -424,7 +424,7 @@
 		kdDebug() << "[ICQProtocol] findGroup(), creating group " << groupName << "." << endl;
 		ICQGroup *tmpG = new ICQGroup();
 		ICQGroup *group = engine->getGroup(engine->contacts.getGroupId(tmpG), true);
-		group->Name = groupName;
+		group->Name = groupName.latin1();
 		delete tmpG;
 		return group;
 	}
@@ -1006,7 +1006,7 @@
 	}
 	if (ICQChangeStatus::getStatusMessage(this, ICQ_STATUS_NA) != "QDialog::Rejected")
 	{
-		engine->owner->AutoResponseNA = mPrefs->naMessage();
+		engine->owner->AutoResponseNA = mPrefs->naMessage().latin1();
 		if ( engine->m_state != ICQClient::Logged ) connectAttempt();
 		engine->setStatus(ICQ_STATUS_NA);
 	}
@@ -1043,7 +1043,7 @@
 
 	if (ICQChangeStatus::getStatusMessage(this, ICQ_STATUS_OCCUPIED) != "QDialog::Rejected")
 	{
-		engine->owner->AutoResponseOccupied = mPrefs->occMessage();
+		engine->owner->AutoResponseOccupied = mPrefs->occMessage().latin1();
 		if ( engine->m_state != ICQClient::Logged ) connectAttempt();
 		engine->setStatus(ICQ_STATUS_OCCUPIED);
 	}
@@ -1069,7 +1069,7 @@
 	}
 	if (ICQChangeStatus::getStatusMessage(this, ICQ_STATUS_DND) != "QDialog::Rejected")
 	{
-		engine->owner->AutoResponseDND = mPrefs->dndMessage();
+		engine->owner->AutoResponseDND = mPrefs->dndMessage().latin1();
 		if ( engine->m_state != ICQClient::Logged ) connectAttempt();
 		engine->setStatus(ICQ_STATUS_DND);
 //		statusBarIcon->setPixmap(dndIcon);
@@ -1078,7 +1078,7 @@
 
 void ICQProtocol::setAway(void)
 {
-	engine->owner->AutoResponseAway = KopeteAway::message();
+	engine->owner->AutoResponseAway = KopeteAway::message().latin1();
 	if ( engine->m_state != ICQClient::Logged ) connectAttempt();
 	engine->setStatus ( ICQ_STATUS_AWAY );
 }
@@ -1106,18 +1106,18 @@
 	
 	ICQUser *mySelf = new ICQUser();
 	mySelf->Uin = mPrefs->uin().toULong();
-	mySelf->Nick = mPrefs->nickName();
+	mySelf->Nick = mPrefs->nickName().latin1();
 	engine->setInfo(mySelf);
 	engine->owner->Uin = mPrefs->uin().toULong();
 
 	if ( mPrefs->useProxy() )
 	{
 		engine->ProxyType = mPrefs->proxyType();
-		engine->ProxyHost = mPrefs->proxyServerAdress();
+		engine->ProxyHost = mPrefs->proxyServerAdress().latin1();
 		engine->ProxyPort = mPrefs->proxyPort();
 		engine->ProxyAuth = mPrefs->proxyAuth();
-		engine->ProxyUser = mPrefs->proxyAuthUser();
-		engine->ProxyPasswd = mPrefs->proxyAuthPassword();
+		engine->ProxyUser = mPrefs->proxyAuthUser().latin1();
+		engine->ProxyPasswd = mPrefs->proxyAuthPassword().latin1();
 	}
 	else
 	{
