http://bugs.gentoo.org/148186

--- src/video/SDL_blit_A.c
+++ src/video/SDL_blit_A.c
@@ -1621,7 +1621,7 @@ BlitRGBtoRGBPixelAlpha(SDL_BlitInfo * in
 
 #if GCC_ASMBLIT
 /* fast (as in MMX with prefetch) ARGB888->(A)RGB888 blending with pixel alpha */
-inline static void BlitRGBtoRGBPixelAlphaMMX3DNOW(SDL_BlitInfo *info)
+static void BlitRGBtoRGBPixelAlphaMMX3DNOW(SDL_BlitInfo *info)
 {
 	int width = info->d_width;
 	int height = info->d_height;
