Index: k3d-0.5.0.34/application/main.cpp
===================================================================
--- k3d-0.5.0.34.orig/application/main.cpp
+++ k3d-0.5.0.34/application/main.cpp
@@ -688,10 +688,12 @@ int main(int argc, char* argv[])
 {
 	const std::string program_name = boost::filesystem::path(argv[0], boost::filesystem::native).leaf();
 
+#ifdef ENABLE_NLS
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, K3D_LOCALE);
 	bind_textdomain_codeset(PACKAGE, "UTF-8");
 	textdomain(PACKAGE);
+#endif
 
 	logbufs_t logbufs;
 
Index: k3d-0.5.0.34/bug-buddy/main.cpp
===================================================================
--- k3d-0.5.0.34.orig/bug-buddy/main.cpp
+++ k3d-0.5.0.34/bug-buddy/main.cpp
@@ -533,10 +533,12 @@ int main(int argc, char* argv[])
 {
 	const std::string program_name = boost::filesystem::path(argv[0], boost::filesystem::native).leaf();
 
+#ifdef ENABLE_NLS
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, K3D_LOCALE);
 	bind_textdomain_codeset(PACKAGE, "UTF-8");
 	textdomain(PACKAGE);
+#endif
 
 	Gtk::Main main(&argc, &argv);
 
Index: k3d-0.5.0.34/k3dsdk/i18n.h
===================================================================
--- k3d-0.5.0.34.orig/k3dsdk/i18n.h
+++ k3d-0.5.0.34/k3dsdk/i18n.h
@@ -23,12 +23,6 @@
 
 	#define _(String) (String)
 	#define N_(String) (String)
-	#define textdomain(String) 
-	#define gettext(String) (String)
-	#define dgettext(Domain, String) (String)
-	#define dcgettext(Domain, String, Type) (String)
-	#define bindtextdomain(Domain, Directory) 
-	#define bind_textdomain_codeset(Domain, Codeset) 
 
 #endif // !ENABLE_NLS
