diff --exclude='*~' -Naur k3d-0.5.0.37.orig/application/main.cpp k3d-0.5.0.37/application/main.cpp
--- k3d-0.5.0.37.orig/application/main.cpp	2006-02-02 17:22:31.000000000 -0200
+++ k3d-0.5.0.37/application/main.cpp	2006-02-02 17:23:20.000000000 -0200
@@ -685,10 +685,12 @@
 {
 	const std::string program_name = boost::filesystem::path(argv[0], boost::filesystem::native).leaf();
 
+#ifdef ENABLE_NLS
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, K3D_LOCALE);
 	bind_textdomain_codeset(PACKAGE, "UTF-8");
 	textdomain(PACKAGE);
+#endif
 
 	logbufs_t logbufs;
 
diff --exclude='*~' -Naur k3d-0.5.0.37.orig/bug-buddy/main.cpp k3d-0.5.0.37/bug-buddy/main.cpp
--- k3d-0.5.0.37.orig/bug-buddy/main.cpp	2006-02-02 17:22:34.000000000 -0200
+++ k3d-0.5.0.37/bug-buddy/main.cpp	2006-02-02 17:23:58.000000000 -0200
@@ -532,10 +532,12 @@
 {
 	const std::string program_name = boost::filesystem::path(argv[0], boost::filesystem::native).leaf();
 
+#ifdef ENABLE_NLS
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, K3D_LOCALE);
 	bind_textdomain_codeset(PACKAGE, "UTF-8");
 	textdomain(PACKAGE);
+#endif
 
 	Gtk::Main main(&argc, &argv);
 
diff --exclude='*~' -Naur k3d-0.5.0.37.orig/k3dsdk/i18n.h k3d-0.5.0.37/k3dsdk/i18n.h
--- k3d-0.5.0.37.orig/k3dsdk/i18n.h	2006-02-02 17:22:33.000000000 -0200
+++ k3d-0.5.0.37/k3dsdk/i18n.h	2006-02-02 17:24:50.000000000 -0200
@@ -23,12 +23,6 @@
 
 	#define _(String) (String)
 	#define N_(String) (String)
-	#define textdomain(String) 
-	#define gettext(String) (String)
-	#define dgettext(Domain, String) (String)
-	#define dcgettext(Domain, String, Type) (String)
-	#define bindtextdomain(Domain, Directory) 
-	#define bind_textdomain_codeset(Domain, Codeset) 
 
 #endif // !ENABLE_NLS 
 
