--- eject.c	2003-02-26 17:07:15.000000000 +0000
+++ eject.c	2003-02-26 17:11:55.000000000 +0000
@@ -273,15 +273,12 @@
 /* Return 1 if file/device exists, 0 otherwise. */
 static int FileExists(const char *name)
 {
-	int fd;
-
-	fd = open(name, O_RDONLY|O_NONBLOCK);
-	if (fd == -1) {
-		return 0;
-	} else {
-		close(fd);
+	if (access (name, F_OK) == 0) {
 		return 1;
 	}
+	else {
+		return 0;
+	}
 }
 
 
