diff -Naur glitz-0.2.3/src/Makefile.am glitz/src/Makefile.am
--- glitz-0.2.3/src/Makefile.am	2004-09-03 16:27:58.000000000 +0200
+++ glitz/src/Makefile.am	2004-12-21 13:39:31.655976048 +0100
@@ -63,7 +63,7 @@
 	glitz_glxint.h
 libglitz_glx_la_LDFLAGS = -version-info @VERSION_INFO@ -no-undefined
 libglitz_glx_la_CFLAGS = $(GLX_CFLAGS)
-libglitz_glx_la_LIBADD = -lglitz $(GLX_LIBS)
+libglitz_glx_la_LIBADD = libglitz.la $(GLX_LIBS)
 endif
 
 if GLITZ_BUILD_AGL_BACKEND
@@ -81,5 +81,5 @@
 	glitz_aglint.h
 libglitz_agl_la_LDFLAGS = -version-info @VERSION_INFO@
 libglitz_agl_la_CFLAGS = $(AGL_CFLAGS)
-libglitz_agl_la_LIBADD = -lglitz $(AGL_LIBS)
+libglitz_agl_la_LIBADD = libglitz.la $(AGL_LIBS)
 endif
