--- /usr/portage/media-gfx/gnuplot/gnuplot-3.7.1-r1.ebuild	Wed Oct 24 16:53:08 2001
+++ ./gnuplot-3.7.1-r1.ebuild	Thu Nov  8 02:42:33 2001
@@ -9,6 +9,7 @@
 HOMEPAGE="http://www.gnuplot.org"
 
 DEPEND="media-libs/libpng
+		media-libs/plotutils
 	readline? ( sys-libs/readline )
 	X? ( virtual/x11 )
 	svga? ( media-libs/svgalib )"
@@ -17,7 +18,9 @@
 	local myvar
 	#--with-lasergnu flag seems to be broken and I'm too lazy to fix now
 	#myvar=" --with-png --without-gd --with-lasergnu"
-	myvar=" --with-png --without-gd"
+	#--with-plot enables the Gnu plotutils library
+	#need to specify path to differentiate from Unix plot
+	myvar=" --with-png --without-gd --with-plot=/usr/lib"
 	if [ -z "`use X`" ]
 	then
 		myvar="${myvar} --without-x"
