--- postfix-20010228.2-r6.ebuild	Mon Sep 17 19:50:42 2001
+++ postfix-20010228.5.ebuild	Fri Oct  5 02:24:21 2001
@@ -3,7 +3,7 @@
 # Author Jerry Alexandratos <jerry@gentoo.org>
 # /home/cvsroot/gentoo-x86/net-mail/postfix/postfix-20010228.2-r6.ebuild,v 1.1 2001/09/18 00:50:42 agriffis Exp
 
-P=${PN}-20010228-pl02
+P=${PN}-20010228-pl05
 A=${P}.tar.gz
 S=${WORKDIR}/${P}
 DESCRIPTION="A fast and secure drop-in replacement for sendmail"
