--- bin/check_perms.org	2005-05-25 00:04:15.000000000 +0200
+++ bin/check_perms	2005-05-25 00:04:35.000000000 +0200
@@ -331,6 +331,8 @@
         print _('checking permissions on list data')
     # BAW: This needs to be converted to the Site module abstraction
     for dir in os.listdir(mm_cfg.LIST_DATA_DIR):
+        if not S_ISDIR(statmode(os.path.join(mm_cfg.LIST_DATA_DIR,dir))):
+            continue
         for file in checkfiles:
             path = os.path.join(mm_cfg.LIST_DATA_DIR, dir, file)
             if STATE.VERBOSE:
--- bin/update.org	2005-05-25 00:04:30.000000000 +0200
+++ bin/update	2005-05-25 00:04:35.000000000 +0200
@@ -34,6 +34,7 @@
 """
 
 import os
+import stat
 import md5
 import sys
 import time
@@ -425,7 +426,7 @@
     # Now update for the Mailman 2.1.5 qfile format.  For every filebase in
     # the qfiles/* directories that has both a .pck and a .db file, pull the
     # data out and re-queue them.
-    for dirname in os.listdir(mm_cfg.QUEUE_DIR):
+    for dirname in [x for x in os.listdir(mm_cfg.QUEUE_DIR) if stat.S_ISDIR(os.stat(os.path.join(mm_cfg.QUEUE_DIR,x)).st_mode)]:
         dirpath = os.path.join(mm_cfg.QUEUE_DIR, dirname)
         if dirpath == mm_cfg.BADQUEUE_DIR:
             # The files in qfiles/bad can't possibly be pickles
