--- llvm-2.6/tools/clang/lib/CodeGen/CGObjCGNU.cpp.bak	2010-04-27 15:07:20.000000000 +0300
+++ llvm-2.6/tools/clang/lib/CodeGen/CGObjCGNU.cpp	2010-04-27 15:08:26.000000000 +0300
@@ -1170,7 +1170,7 @@
     Elements.push_back(llvm::ConstantInt::get(LongTy, RuntimeVersion));
   }
   // sizeof(ModuleTy)
-  llvm::TargetData td = llvm::TargetData::TargetData(&TheModule);
+  llvm::TargetData td = llvm::TargetData(&TheModule);
   Elements.push_back(llvm::ConstantInt::get(LongTy,
                      td.getTypeSizeInBits(ModuleTy)/8));
   //FIXME: Should be the path to the file where this module was declared
@@ -1320,7 +1320,7 @@
   CGF.EmitBlock(TryHandler);
 
   // Get the correct versions of the exception handling intrinsics
-  llvm::TargetData td = llvm::TargetData::TargetData(&TheModule);
+  llvm::TargetData td = llvm::TargetData(&TheModule);
   int PointerWidth = td.getTypeSizeInBits(PtrTy);
   assert((PointerWidth == 32 || PointerWidth == 64) &&
     "Can't yet handle exceptions if pointers are not 32 or 64 bits");
