diff -Naur xmms-wma-1.0.3.orig/Makefile xmms-wma-1.0.3/Makefile
--- xmms-wma-1.0.3.orig/Makefile	2004-03-03 06:47:58.000000000 -0800
+++ xmms-wma-1.0.3/Makefile	2004-07-09 16:24:31.276059000 -0700
@@ -11,7 +11,7 @@
 export DEPS := $(SOURCES:%.c=$(DEPDIR)/%.d)
 
 all:
-	@cd ffmpeg-strip-wma; make; cd ..
+#	@cd ffmpeg-strip-wma; make; cd ..
 	@$(MAKE) --no-print-directory -f Makefile.inc
 	@strip $(PLUGIN_FILE)
 	@echo "The plug-in has been compiled. Run one of the following:"
diff -Naur xmms-wma-1.0.3.orig/Makefile.inc xmms-wma-1.0.3/Makefile.inc
--- xmms-wma-1.0.3.orig/Makefile.inc	2004-03-03 12:02:39.000000000 -0800
+++ xmms-wma-1.0.3/Makefile.inc	2004-07-09 16:31:44.366257000 -0700
@@ -1,18 +1,20 @@
 CC := gcc
-CFLAGS := -O2 -ffast-math -fomit-frame-pointer
-#CFLAGS := -Wall -g -O3
-CFLAGS += -DHAVE_AV_CONFIG_H -I.. -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE -DX86
-CFLAGS += -I ffmpeg-strip-wma
-CFLAGS += `xmms-config --cflags`
-LIBS := `xmms-config --libs` -L ffmpeg-strip-wma -lffwma
+MYCFLAGS := $(CFLAGS) -fPIC
+#MYCFLAGS := -Wall -g -O3
+MYCFLAGS += -DHAVE_AV_CONFIG_H -I.. -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE -fPIC
+#MYCFLAGS += -I ffmpeg-strip-wma
+MYCFLAGS += `xmms-config --cflags`
+MYCFLAGS += -I/usr/include/ffmpeg
+#LIBS := `xmms-config --libs` -L ffmpeg-strip-wma -lffwma
+LIBS := `xmms-config --libs` -lavcodec -lavformat
 
 $(PLUGIN_FILE): $(OBJECTS)
 	$(CC) -o $@ $^ $(LIBS) -shared
 
 $(OBJDIR)/%.o: %.c
-	$(CC) -c -o $@ $(CFLAGS) $<
+	$(CC) -c -o $@ $(MYCFLAGS) $<
 
 $(DEPDIR)/%.d: %.c
-	$(CC) -MM $(CFLAGS) $< | sed "s/^$*\.o:/$(OBJDIR)\/$*\.o $(DEPDIR)\/$*\.d:/" > $@
+	$(CC) -MM $(MYCFLAGS) $< | sed "s/^$*\.o:/$(OBJDIR)\/$*\.o $(DEPDIR)\/$*\.d:/" > $@
 
 -include $(DEPS)
diff -Naur xmms-wma-1.0.3.orig/wma123_examples/Makefile xmms-wma-1.0.3/wma123_examples/Makefile
--- xmms-wma-1.0.3.orig/wma123_examples/Makefile	2004-03-26 09:01:39.000000000 -0800
+++ xmms-wma-1.0.3/wma123_examples/Makefile	2004-07-09 16:25:08.197936000 -0700
@@ -1,7 +1,7 @@
-include ../ffmpeg-strip-wma/config.mak
+#include ../ffmpeg-strip-wma/config.mak
 
-INC=-I../ffmpeg-strip-wma
-LIB=-L../ffmpeg-strip-wma
+#INC=-I../ffmpeg-strip-wma
+#LIB=-L../ffmpeg-strip-wma
 CFLAGS=$(OPTFLAGS) -DHAVE_AV_CONFIG_H -I.. -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
 
 all: artswma alsawma osswma
