diff --exclude='*~' -Naur SoGtk.orig/src/Inventor/Gtk/SoGtkComponent.cpp SoGtk/src/Inventor/Gtk/SoGtkComponent.cpp
--- SoGtk.orig/src/Inventor/Gtk/SoGtkComponent.cpp	2005-07-29 17:02:46.000000000 -0300
+++ SoGtk/src/Inventor/Gtk/SoGtkComponent.cpp	2005-07-29 17:04:12.000000000 -0300
@@ -17,6 +17,8 @@
  *
  **************************************************************************/
 
+#include <string.h>
+
 #if SOGTK_DEBUG
 static const char rcsid[] =
   "$Id: SoGtk-20010601-string.patch,v 1.1 2005/07/29 20:19:14 vanquirius Exp $";
diff --exclude='*~' -Naur SoGtk.orig/src/Inventor/Gtk/SoGtk.cpp SoGtk/src/Inventor/Gtk/SoGtk.cpp
--- SoGtk.orig/src/Inventor/Gtk/SoGtk.cpp	2005-07-29 17:02:46.000000000 -0300
+++ SoGtk/src/Inventor/Gtk/SoGtk.cpp	2005-07-29 17:04:21.000000000 -0300
@@ -17,6 +17,8 @@
  *
  **************************************************************************/
 
+#include <string.h>
+
 #if SOGTK_DEBUG
 static const char rcsid[] =
   "$Id: SoGtk-20010601-string.patch,v 1.1 2005/07/29 20:19:14 vanquirius Exp $";
diff --exclude='*~' -Naur SoGtk.orig/src/Inventor/Gtk/widgets/SoGtkPopupMenu.cpp SoGtk/src/Inventor/Gtk/widgets/SoGtkPopupMenu.cpp
--- SoGtk.orig/src/Inventor/Gtk/widgets/SoGtkPopupMenu.cpp	2005-07-29 17:02:46.000000000 -0300
+++ SoGtk/src/Inventor/Gtk/widgets/SoGtkPopupMenu.cpp	2005-07-29 17:04:00.000000000 -0300
@@ -17,6 +17,8 @@
  *
  **************************************************************************/
 
+#include <string.h>
+
 #if SOGTK_DEBUG
 static const char rcsid[] =
   "$Id: SoGtk-20010601-string.patch,v 1.1 2005/07/29 20:19:14 vanquirius Exp $";
