diff -NrU5 qtparted-0.4.5.original/src/qp_libparted.cpp qtparted-0.4.5/src/qp_libparted.cpp
--- qtparted-0.4.5.original/src/qp_libparted.cpp	2009-07-29 15:42:54.000000000 +0000
+++ qtparted-0.4.5/src/qp_libparted.cpp	2009-07-29 15:49:16.000000000 +0000
@@ -1746,15 +1746,15 @@
 		if (ped_exception_throw(PED_EXCEPTION_ERROR,
 								(PedExceptionOption)PED_EXCEPTION_IGNORE_CANCEL,
 								label.latin1(),
 								path) != PED_EXCEPTION_IGNORE) goto error_free_path;
 	}
-	ped_free(path);
+	free(path);
 	return true;
 
 error_free_path:
-	ped_free(path);
+	free(path);
 
 //error:
 	return false;
 }
 
