--- pftpfxp-mew/src/main.cc.orig	2006-07-13 13:04:55.000000000 +1000
+++ pftpfxp-mew/src/main.cc	2006-07-13 13:07:35.000000000 +1000
@@ -547,7 +547,7 @@
                 if ((*label != '\0') && (strlen(value) > 0)) {
                     if (!strcasecmp(label, "DEVICE")) {
                         if (!DetermineOwnIP(value)) {
-                            printf("unknown network device '%s', sorry.\n",
+                            fprintf(stderr,"unknown network device '%s', sorry.\n",
                                    value);
                             fclose(in_file);
                             return (FALSE);
@@ -705,7 +705,7 @@
                         strcpy(sectionlabels[21], value);
                         //debuglog("section22: %s", sectionlabels[21]);
                     } else {
-                        printf("unknown label '%s' in configfile.\n", label);
+                        fprintf(stderr,"unknown label '%s' in configfile.\n", label);
                         fclose(in_file);
                         return (FALSE);
                     }
@@ -1096,9 +1096,26 @@
 }
 
 int main(int argc, char **argv) {
-    char msg[256], config_file[] = {".pftp/config"}, keymap_file[] = {".pftp/keymap"};
+    char msg[256];
+    char *home_directory;
+    char *config_file;
+    char *keymap_file;
     int n;
 
+    home_directory = getenv("HOME");
+	n = strlen(home_directory) + sizeof("/.pftp/config") + 1;
+    if ((config_file=(char*)malloc(n))==NULL) {
+        fprintf(stderr,"error allocating memory\n");
+        exit( -1);
+    }
+    if ((keymap_file=(char*)malloc(n))==NULL) {
+        free(config_file);
+        fprintf(stderr,"error allocating memory\n");
+        exit( -1);
+    }
+    sprintf(config_file, "%s/.pftp/config", home_directory);
+    sprintf(keymap_file, "%s/.pftp/keymap", home_directory);
+    
     pthread_mutex_init(&syscall_lock, NULL);
     pthread_mutex_init(&sigwinch_lock, NULL);
 
@@ -1137,35 +1154,39 @@
     getcwd(startcwd, SERVER_WORKINGDIR_SIZE);
 
     if (!ReadConfig(config_file)) {
-        printf("error reading/parsing configfile '%s', bailing out.\n", config_file);
+        fprintf(stderr,"error reading/parsing configfile '%s', bailing out.\n", config_file);
+        free(config_file);
         exit( -1);
     }
+    free(config_file);
 
     unlinklog();
 
     if (!ReadKeymap(keymap_file)) {
-        printf("error reading/parsing keymapfile '%s', bailing out.\n", keymap_file);
+        fprintf(stderr,"error reading/parsing keymapfile '%s', bailing out.\n", keymap_file);
+        free(keymap_file);
         exit( -1);
     }
+    free(keymap_file);
     debuglog("after keymap");
 
     if (display->ProbeBookmarkRC() == 1) {
-        printf("unknown or invalid bookmark file found, delete it\n");
+        fprintf(stderr,"unknown or invalid bookmark file found, delete it\n");
         exit( -1);
     }
 
     if (chdir(localdir) != 0) {
-        printf("please specify a valid dir for the LOCALDIR label or comment it out\n");
+        fprintf(stderr,"please specify a valid dir for the LOCALDIR label or comment it out\n");
         exit ( -1);
     }
 
     if (!use_own_ip) {
-        printf("you need to specify a network-device in the configfile.\n");
+        fprintf(stderr,"you need to specify a network-device in the configfile.\n");
         exit( -1);
     }
 
     if (!use_okay_dir) {
-        printf("you need to specify a dir for the .okay and .error files in the configfile.\n");
+        fprintf(stderr,"you need to specify a dir for the .okay and .error files in the configfile.\n");
         exit( -1);
     }
 
