diff -uNr -r fail2ban-0.7.6.orig/files/gentoo-initd fail2ban-0.7.6/files/gentoo-initd
--- fail2ban-0.7.6.orig/files/gentoo-initd	2006-12-17 23:32:53.000000000 +0100
+++ fail2ban-0.7.6/files/gentoo-initd	2007-03-03 22:46:09.191746854 +0100
@@ -21,7 +21,7 @@
 
 opts="start stop restart reload showlog"
 
-FAIL2BAN="/usr/bin/fail2ban-client"
+FAIL2BAN="/usr/bin/fail2ban-client ${FAIL2BAN_OPTIONS}"
 
 depend() {
 	need net
