diff -ru kvdr-0.64-gentoo/kvdr/kvdr.desktop kvdr-0.64-gentoo-desktop-file/kvdr/kvdr.desktop
--- kvdr-0.64-gentoo/kvdr/kvdr.desktop	2001-11-03 17:06:56.000000000 +0100
+++ kvdr-0.64-gentoo-desktop-file/kvdr/kvdr.desktop	2007-10-16 21:57:14.000000000 +0200
@@ -1,9 +1,10 @@
 # KDE Config File
 [KDE Desktop Entry]
+Categories=Qt;KDE;Application;AudioVideo;Player;
 Type=Application
 Exec=kvdr
-Icon=kvdr.png
-MiniIcon=mini-kvdr.png
+Icon=kvdr
+MiniIcon=mini-kvdr
 DocPath=kvdr/index.html
 Comment=Digital TV viewer and recorder
 Comment[de]=Digital-TV Viewer and Recorder
