--- kvdr/kvdr.desktop.o	2009-02-14 19:50:00.000000000 +0100
+++ kvdr/kvdr.desktop	2009-02-14 19:00:46.000000000 +0100
@@ -1,12 +1,12 @@
 # KDE Config File
-[KDE Desktop Entry]
+[Desktop Entry]
+Categories=Qt;KDE;AudioVideo;Player;
 Type=Application
 Exec=kvdr
-Icon=kvdr.png
-MiniIcon=mini-kvdr.png
-DocPath=kvdr/index.html
+Icon=kvdr
+X-DocPath=kvdr/index.html
 Comment=Digital TV viewer and recorder
 Comment[de]=Digital-TV Viewer and Recorder
-Terminal=0
+Terminal=false
 Name=Kvdr
 Name[de]=Kvdr
--- kvdr/Makefile.am.o	2009-02-14 18:58:57.000000000 +0100
+++ kvdr/Makefile.am	2009-02-14 19:00:00.000000000 +0100
@@ -8,8 +8,8 @@
 EXTRA_DIST = main.cpp kvdr.cpp kvdr.h kvdr.desktop lo32-app-kvdr.png lo16-app-kvdr.png ckvdrcfg.cpp ckvdrcfg.h ckvdrcfgdata.cpp kvdr.xpm kvdr16.png kvdr16h.png kvdr32.png kvdr32h.png kvdr48.png kvdr48h.png kvdr_xv.cpp kvdr_xv.h kvdrcfg.kdevdlg mini-kvdr.xpm svdrpc.cpp svdrpc.h 
 
 install-data-local:
-	$(mkinstalldirs) $(kde_appsdir)/Multimedia/
-	$(INSTALL_DATA) $(srcdir)/kvdr.desktop $(kde_appsdir)/Multimedia/kvdr.desktop
+	$(mkinstalldirs) $(xdg_appsdir)
+	$(INSTALL_DATA) $(srcdir)/kvdr.desktop $(xdg_appsdir)/kvdr.desktop
 	$(mkinstalldirs) $(kde_icondir)/locolor/32x32/apps/
 	$(INSTALL_DATA) $(srcdir)/lo32-app-kvdr.png $(kde_icondir)/locolor/32x32/apps/kvdr.png
 	$(mkinstalldirs) $(kde_icondir)/locolor/16x16/apps/
