diff -Nrup chillispot-0.94.orig/src/Makefile.am chillispot-0.94/src/Makefile.am
--- chillispot-0.94.orig/src/Makefile.am	2004-04-02 18:55:11.000000000 +0000
+++ chillispot-0.94/src/Makefile.am	2004-07-06 16:33:23.000000000 +0000
@@ -1,9 +1,9 @@
 sbin_PROGRAMS = chilli
 
-AM_CFLAGS = -O2 -D_GNU_SOURCE -fno-builtin -Wall -ansi -DSBINDIR='"$(sbindir)"' -ggdb -lcrypt
+AM_CFLAGS = -D_GNU_SOURCE -fno-builtin -ansi -DSBINDIR='"$(sbindir)"'
 
 chilli_SOURCES = chilli.c defs.h tun.c tun.h cmdline.c cmdline.h ippool.c ippool.h radius.h radius.c md5.c md5.h redir.h redir.c dhcp.c dhcp.h syserr.c syserr.h
-
+chilli_LDFLAGS = -lcrypt
 
 
 
diff -Nrup chillispot-0.94.orig/src/Makefile.in chillispot-0.94/src/Makefile.in
--- chillispot-0.94.orig/src/Makefile.in	2004-06-21 20:28:06.000000000 +0000
+++ chillispot-0.94/src/Makefile.in	2004-07-06 16:34:05.000000000 +0000
@@ -66,18 +66,26 @@ EXEEXT = @EXEEXT@
 OBJEXT = @OBJEXT@
 PATH_SEPARATOR = @PATH_SEPARATOR@
 AMTAR = @AMTAR@
+AR = @AR@
 AS = @AS@
 AWK = @AWK@
 CC = @CC@
+CXX = @CXX@
+CXXCPP = @CXXCPP@
 DEPDIR = @DEPDIR@
 DLLTOOL = @DLLTOOL@
 ECHO = @ECHO@
+EGREP = @EGREP@
+F77 = @F77@
+GCJ = @GCJ@
+GCJFLAGS = @GCJFLAGS@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 LIBTOOL = @LIBTOOL@
 LN_S = @LN_S@
 OBJDUMP = @OBJDUMP@
 PACKAGE = @PACKAGE@
 RANLIB = @RANLIB@
+RC = @RC@
 STRIP = @STRIP@
 VERSION = @VERSION@
 am__include = @am__include@
@@ -85,9 +93,10 @@ am__quote = @am__quote@
 install_sh = @install_sh@
 sbin_PROGRAMS = chilli
 
-AM_CFLAGS = -O2 -D_GNU_SOURCE -fno-builtin -Wall -ansi -DSBINDIR='"$(sbindir)"' -ggdb -lcrypt
+AM_CFLAGS = -D_GNU_SOURCE -fno-builtin -ansi -DSBINDIR='"$(sbindir)"'
 
 chilli_SOURCES = chilli.c defs.h tun.c tun.h cmdline.c cmdline.h ippool.c ippool.h radius.h radius.c md5.c md5.h redir.h redir.c dhcp.c dhcp.h syserr.c syserr.h
+chilli_LDFLAGS = -lcrypt
 subdir = src
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
@@ -101,7 +110,6 @@ am_chilli_OBJECTS = chilli.$(OBJEXT) tun
 chilli_OBJECTS = $(am_chilli_OBJECTS)
 chilli_LDADD = $(LDADD)
 chilli_DEPENDENCIES =
-chilli_LDFLAGS =
 
 DEFS = @DEFS@
 DEFAULT_INCLUDES =  -I. -I$(srcdir) -I$(top_builddir)
