diff -ru powder113_src.orig/make/makerules powder113_src/make/makerules
--- powder113_src.orig/make/makerules	2009-02-05 20:11:41.000000000 -0500
+++ powder113_src/make/makerules	2010-05-03 20:38:32.579540189 -0400
@@ -8,7 +8,7 @@
 all: $(TARGET)
 
 $(TARGET): $(OFILES)
-	$(CXX) -o $(TARGET) $(OFILES) $(CXXFLAGS)
+	$(CXX) -o $(TARGET) $(OFILES) $(CXXFLAGS) $(LDFLAGS)
 
 clean:
 	$(RM) $(OFILES)
diff -ru powder113_src.orig/make/makerules.LINUX powder113_src/make/makerules.LINUX
--- powder113_src.orig/make/makerules.LINUX	2009-01-18 19:54:09.000000000 -0500
+++ powder113_src/make/makerules.LINUX	2010-05-03 20:39:11.022462202 -0400
@@ -13,8 +13,8 @@
 	# and is to try and ensure we end up with a static link
 	# of stdc++.
 	$(RM) libstdc++.a
-	ln -s `g++ -print-file-name=libstdc++.a`
-	$(CXX) -static-libgcc -L. -o $(TARGET) $(OFILES) `sdl-config --libs` $(CXXFLAGS)
+	ln -s `$(CXX) -print-file-name=libstdc++.a`
+	$(CXX) -static-libgcc -L. -o $(TARGET) $(OFILES) `sdl-config --libs` $(CXXFLAGS) $(LDFLAGS)
 
 clean:
 	$(RM) $(OFILES)
diff -ru powder113_src.orig/port/linux/Makefile powder113_src/port/linux/Makefile
--- powder113_src.orig/port/linux/Makefile	2009-05-13 19:32:42.000000000 -0400
+++ powder113_src/port/linux/Makefile	2010-05-03 20:40:57.629463531 -0400
@@ -48,7 +48,7 @@
 all: $(TARGET)
 
 powder: premake $(OFILES)
-	$(CXX) -o $(TARGET) $(OFILES) `sdl-config --libs` $(CXXFLAGS)
+	$(CXX) -o $(TARGET) $(OFILES) `sdl-config --libs` $(CXXFLAGS) $(LDFLAGS)
 
 .cpp.o:
 	$(CXX) -DLINUX -I ../../port/sdl `sdl-config --cflags` -c $< -o $@ $(CXXFLAGS)
@@ -60,7 +60,7 @@
 	# of stdc++.
 	$(RM) libstdc++.a
 	ln -s `g++ -print-file-name=libstdc++.a`
-	$(CXX) -static-libgcc -L. -o $(TARGET) $(OFILES) `sdl-config --libs` $(CXXFLAGS)
+	$(CXX) -static-libgcc -L. -o $(TARGET) $(OFILES) `sdl-config --libs` $(CXXFLAGS) $(LDFLAGS)
 
 clean:
 	$(RM) $(OFILES)
