diff -Naur Python-2.4.3.orig/Include/Python.h Python-2.4.3/Include/Python.h
--- Python-2.4.3.orig/Include/Python.h	2006-05-04 14:20:30.000000000 +0000
+++ Python-2.4.3/Include/Python.h	2006-05-04 14:21:26.000000000 +0000
@@ -2,6 +2,11 @@
 #define Py_PYTHON_H
 /* Since this is a "meta-include" file, no #ifdef __cplusplus / extern "C" { */
 
+/* Required on Gentoo/OpenBSD */
+#if defined(__OpenBSD__)
+#include <sys/types.h>
+#endif
+
 /* Include nearly all Python header files */
 
 #include "patchlevel.h"
diff -Naur Python-2.4.3.orig/configure.in Python-2.4.3/configure.in
--- Python-2.4.3.orig/configure.in	2006-05-04 14:20:30.000000000 +0000
+++ Python-2.4.3/configure.in	2006-05-04 14:23:22.000000000 +0000
@@ -1528,9 +1528,16 @@
 
 # only check for sem_ini if thread support is requested
 if test "$with_threads" = "yes" -o -z "$with_threads"; then
+  case "$ac_sys_system" in
+  OpenBSD*)
+    LIBS="-pthread ${LIBS}"
+    ;;
+  *)
     AC_SEARCH_LIBS(sem_init, pthread rt posix4) # 'Real Time' functions on Solaris
 						# posix4 on Solaris 2.6
 						# pthread (first!) on Linux
+    ;;
+  esac
 fi
 
 # check if we need libintl for locale functions
