diff -ur downman-0.0.5.orig/clients/cmdline/Makefile.am downman-0.0.5/clients/cmdline/Makefile.am
--- downman-0.0.5.orig/clients/cmdline/Makefile.am	2003-10-05 18:54:18.000000000 +0300
+++ downman-0.0.5/clients/cmdline/Makefile.am	2007-11-04 15:55:19.000000000 +0200
@@ -4,9 +4,7 @@
 	main.c
 
 downman_LDADD =	\
-	$(top_srcdir)/utillib/libutil.a
-
-downman_LDFLAGS = \
+	$(top_srcdir)/utillib/libutil.a			\
 	$(DOWNMAN_LIBS)
 
 localedir = $(datadir)/locale
diff -ur downman-0.0.5.orig/clients/gnome/gdownman/Makefile.am downman-0.0.5/clients/gnome/gdownman/Makefile.am
--- downman-0.0.5.orig/clients/gnome/gdownman/Makefile.am	2003-08-28 17:48:58.000000000 +0300
+++ downman-0.0.5/clients/gnome/gdownman/Makefile.am	2007-11-04 15:58:18.000000000 +0200
@@ -42,11 +42,9 @@
 	servers.h		\
 	$(BUILT_SOURCES)
 
-gdownman_LDFLAGS =				\
-	$(GDOWNMAN_LIBS)
-
 gdownman_LDADD = 				\
-	$(top_srcdir)/utillib/libutil.a
+	$(top_srcdir)/utillib/libutil.a		\
+	$(GDOWNMAN_LIBS)
 
 localedir = $(datadir)/locale
 
diff -ur downman-0.0.5.orig/clients/gnome/monitor/Makefile.am downman-0.0.5/clients/gnome/monitor/Makefile.am
--- downman-0.0.5.orig/clients/gnome/monitor/Makefile.am	2003-05-30 22:08:01.000000000 +0300
+++ downman-0.0.5/clients/gnome/monitor/Makefile.am	2007-11-04 16:00:01.000000000 +0200
@@ -5,8 +5,8 @@
 	dmgmonitor.c dmgmonitor.h	\
 	dmgmonitor_ui.h
 
-downman_gmonitor_LDFLAGS = `pkg-config --libs gtk+-2.0 libgnome-2.0 libgnomeui-2.0`
-downman_gmonitor_LDADD = $(top_srcdir)/utillib/libutil.a
+downman_gmonitor_LDADD = $(top_srcdir)/utillib/libutil.a	\
+	$(GDOWNMAN_LIBS)
 
 localedir = $(datadir)/locale
 
diff -ur downman-0.0.5.orig/daemon/Makefile.am downman-0.0.5/daemon/Makefile.am
--- downman-0.0.5.orig/daemon/Makefile.am	2003-09-30 23:29:17.000000000 +0300
+++ downman-0.0.5/daemon/Makefile.am	2007-11-04 15:51:16.000000000 +0200
@@ -22,9 +22,7 @@
 	servers.c servers.h	\
 	$(BUILT_SOURCES)
 
-downmand_LDADD = $(top_srcdir)/utillib/libutil.a
-
-downmand_LDFLAGS = 			\
+downmand_LDADD = $(top_srcdir)/utillib/libutil.a	\
 	$(DOWNMAN_LIBS)			\
 	$(DOWNMAND_LIBS)
 
