Index: uim/uim-notify.c
===================================================================
--- uim/uim-notify.c	(revision 5997)
+++ uim/uim-notify.c	(working copy)
@@ -111,6 +111,8 @@
 
   if (strcmp(agent->desc()->name, name) == 0) {
     return UIM_TRUE;
+  } if (getenv("UIM_DISABLE_NOTIFY") != NULL) {
+    return UIM_TRUE;
   } else if (strcmp(name, "stderr") == 0) {
     agent->quit();
     if (notify_dlhandle)
@@ -238,6 +240,10 @@
 		    MAKE_STR(desc->name),
 		    MAKE_STR(desc->desc)),
 	      uim_scm_null());
+
+  if (getenv("UIM_DISABLE_NOTIFY") != NULL)
+    return uim_scm_callf("reverse", "o", ret_);
+
   dirp = opendir(NOTIFY_PLUGIN_PATH);
   if (dirp) {
     while ((dp = readdir(dirp)) != NULL) {
Index: scm/Makefile.am
===================================================================
--- scm/Makefile.am	(revision 5997)
+++ scm/Makefile.am	(working copy)
@@ -97,7 +97,8 @@
 UIM_MODULE_MANAGER_ENV = \
         LIBUIM_SYSTEM_SCM_FILES=$(abs_top_srcdir)/sigscheme/lib \
         LIBUIM_SCM_FILES=$(abs_srcdir) \
-        LIBUIM_PLUGIN_LIB_DIR=$(abs_top_builddir)/uim/.libs
+        LIBUIM_PLUGIN_LIB_DIR=$(abs_top_builddir)/uim/.libs \
+        UIM_DISABLE_NOTIFY=1
 
 installed-modules.scm loader.scm: $(SCM_FILES)
 	$(MAKE) $(AM_MAKEFLAGS) -C $(top_builddir)/uim uim-module-manager && \
--- scm/Makefile.in.orig	Tue Sep 15 02:01:18 2009
+++ scm/Makefile.in	Tue Sep 15 02:01:36 2009
@@ -363,7 +363,8 @@
 UIM_MODULE_MANAGER_ENV = \
         LIBUIM_SYSTEM_SCM_FILES=$(abs_top_srcdir)/sigscheme/lib \
         LIBUIM_SCM_FILES=$(abs_srcdir) \
-        LIBUIM_PLUGIN_LIB_DIR=$(abs_top_builddir)/uim/.libs
+        LIBUIM_PLUGIN_LIB_DIR=$(abs_top_builddir)/uim/.libs \
+        UIM_DISABLE_NOTIFY=1
 
 all: all-am
 
