--- src/extern.h~	Sun Aug  6 08:51:22 2000
+++ src/extern.h	Thu Nov 15 13:47:21 2001
@@ -147,7 +147,7 @@
 void	get(int, char **);
 struct cmd *getcmd(const char *);
 int	getit(int, char **, int, const char *);
-struct option *getoption(const char *);
+struct option2 *getoption(const char *);
 char   *getoptionvalue(const char *);
 void	getremoteinfo(void);
 int	getreply(int);
--- src/cmds.c~	Wed Aug  2 12:52:51 2000
+++ src/cmds.c	Thu Nov 15 13:44:52 2001
@@ -2556,7 +2556,7 @@
 void
 setoption(int argc, char *argv[])
 {
-	struct option *o;
+	struct option2 *o;
 
 	code = -1;
 	if (argc == 0 || (argc != 1 && argc != 3)) {
@@ -2591,7 +2591,7 @@
 void
 unsetoption(int argc, char *argv[])
 {
-	struct option *o;
+	struct option2 *o;
 
 	code = -1;
 	if (argc == 0 || argc != 2) {
--- src/ftp_var.h~	Wed Aug  2 12:52:51 2000
+++ src/ftp_var.h	Thu Nov 15 13:43:53 2001
@@ -147,7 +147,7 @@
 /*
  * Format of option table
  */
-struct option {
+struct option2 {
 	char	*name;
 	char	*value;
 };
--- src/main.c~	Thu Nov 15 13:36:15 2001
+++ src/main.c	Thu Nov 15 13:46:13 2001
@@ -516,7 +516,7 @@
 	static char	  buf[MAXPATHLEN];
 
 	if (prompt == NULL) {
-		struct option *o;
+		struct option2 *o;
 
 		o = getoption("prompt");
 		if (o == NULL)
@@ -537,7 +537,7 @@
 	static char	  buf[MAXPATHLEN];
 
 	if (rprompt == NULL) {
-		struct option *o;
+		struct option2 *o;
 
 		o = getoption("rprompt");
 		if (o == NULL)
@@ -913,11 +913,11 @@
 	}
 }
 
-struct option *
+struct option2 *
 getoption(const char *name)
 {
 	const char *p;
-	struct option *c;
+	struct option2 *c;
 
 	if (name == NULL)
 		return (NULL);
@@ -931,7 +931,7 @@
 char *
 getoptionvalue(const char *name)
 {
-	struct option *c;
+	struct option2 *c;
 
 	if (name == NULL)
 		errx(1, "getoptionvalue() invoked with NULL name");
