--- src/scripts/charmc.old      2009-05-31 23:09:12.285687725 -0500
+++ src/scripts/charmc  2009-05-31 23:08:49.782155308 -0500
@@ -335,19 +335,9 @@
        PROG_EXT=".exe"
 fi
 
-CHARMLIB="$CHARMBIN/../lib"
-CHARMINC="$CHARMBIN/../include"
-CHARMLIBSO=
-if test -d "$CHARMBIN/../lib_so" 
-then
-  CHARMLIBSO=`cd $CHARMBIN/../lib_so 2>/dev/null && pwd`
-  #getting absolute path is harder than thought because of symbolic links and ..
-  #ksh needs cd -P to resolve werid symbolic links, however -P is not portable
-  #csh is ok too if it exists 
-  test -z "$CHARMLIBSO" && CHARMLIBSO=`cd -P $CHARMBIN/../lib_so 2>/dev/null && pwd`
-  test -z "$CHARMLIBSO" && CHARMLIBSO=`csh -c "cd $CHARMBIN/../lib_so >& /dev/null && pwd"`
-  test -z "$CHARMLIBSO" && echo "$CHARMBIN/../lib_so: not found" && exit 1
-fi
+CHARMLIB="/usr/gentoo-libdir"
+CHARMINC="/usr/include/gentoo-include"
+CHARMLIBSO="/usr/gentoo-libdir"
 
 ##############################################################################
 #

