diff -Naur minissdpd-1.1.20120410.orig//Makefile minissdpd-1.1.20120410/Makefile
--- minissdpd-1.1.20120410.orig//Makefile	2012-04-09 18:39:42.000000000 -0400
+++ minissdpd-1.1.20120410/Makefile	2012-04-10 08:04:00.000000000 -0400
@@ -12,8 +12,8 @@
 #
 # install target is made for linux... sorry BSD users...
 #CFLAGS = -Wall -g -D_GNU_SOURCE -Wstrict-prototypes -ansi
-CFLAGS = -Wall -Os -D_GNU_SOURCE -fno-strict-aliasing -Wstrict-prototypes -ansi
-CC = gcc
+CFLAGS := -Wall -Os -D_GNU_SOURCE -fno-strict-aliasing -Wstrict-prototypes -ansi $(CFLAGS)
+CC ?= gcc
 RM = rm -f
 INSTALL = install
 OS = $(shell uname -s)
