Only in OpenEXR-1.0.7/: .cvsignore
diff -r -u OpenEXR-1.0.7/IlmImfExamples/Makefile.in OpenEXR-1.0.7-gentoo/IlmImfExamples/Makefile.in
--- OpenEXR-1.0.7/IlmImfExamples/Makefile.in	2003-12-02 08:41:17.000000000 +0200
+++ OpenEXR-1.0.7-gentoo/IlmImfExamples/Makefile.in	2003-12-02 21:40:33.000000000 +0200
@@ -104,7 +104,7 @@
 		      rgbaInterfaceExamples.h generalInterfaceExamples.h
 
 
-examplesdir = $(datadir)/doc/OpenEXR-@OPENEXR_VERSION@/examples
+examplesdir = $(datadir)/doc/$(P)/examples
 examples_DATA = $(imfexamples_SOURCES)
 
 imfexamplesdir = $(examplesdir)
diff -r -u OpenEXR-1.0.7/acinclude.m4 OpenEXR-1.0.7-gentoo/acinclude.m4
--- OpenEXR-1.0.7/acinclude.m4	2003-10-07 01:42:30.000000000 +0200
+++ OpenEXR-1.0.7-gentoo/acinclude.m4	2003-12-02 21:41:05.000000000 +0200
@@ -12,7 +12,7 @@
   if test x$nvsdk_prefix != x ; then
     NVSDK_CXXFLAGS="-DUNIX -I$nvsdk_prefix/inc"
   else
-    NVSDK_CXXFLAGS=""
+    NVSDK_CXXFLAGS="-DUNIX"
   fi
 
   AC_MSG_CHECKING(for Nvidia SDK)
