--- amavisd	2005-10-24 11:10:20.000000000 +0200
+++ amavisd.new	2005-10-24 11:11:08.000000000 +0200
@@ -4718,7 +4718,7 @@
   $via =~ /^bsmtp:(.*)\z/si or die "Bad fwd method: $via";
   my($bsmtp_file_final) = $1; my($mbxname);
   my($s) = $msginfo->sender;  # defanged sender name for use in filename
-  $s =~ tr/a-zA-Z0-9@._+-]/=/c;
+  $s =~ tr/a-zA-Z0-9@._+-/=/c;
   $s = substr($s,0,100)."..."  if length($s) > 100+3;
   $s =~ s/\@/_at_/g; $s =~ s/^(\.{0,2})\z/_$1/g;
   $bsmtp_file_final =~ s{%(.)}
@@ -6281,7 +6281,7 @@
     'sender-quarantine' =>
       sub { my($s) = $MSGINFO->sender;
             $s = substr($s,0,100)."..."  if length($s) > 100+3;
-            $s =~ tr/a-zA-Z0-9@._+-]/=/c; $s =~ s/\@/_at_/g;
+            $s =~ tr/a-zA-Z0-9@._+-/=/c; $s =~ s/\@/_at_/g;
             $s = untaint($s)  if $s =~ /^(?:[a-zA-Z0-9%=._+-]+)\z/;  # untaint
             ($QUARANTINEDIR, "sender-$s-%m.gz");   # suggested file name
           },
