Index: tooracleconnection.cpp
===================================================================
RCS file: /cvsroot/tora/tora/tooracleconnection.cpp,v
retrieving revision 1.80
diff -u -r1.80 tooracleconnection.cpp
--- tooracleconnection.cpp	9 Aug 2003 13:36:17 -0000	1.80
+++ tooracleconnection.cpp	10 Aug 2003 14:30:56 -0000
@@ -202,11 +202,13 @@
     }
     virtual ~oracleQuery()
     {
+      if(Query!=0) {
 #ifdef OTL_STREAM_POOLING_ON
-      if (!SaveInPool)
+        if (!SaveInPool)
 	Query->close(false);
 #endif
-      delete Query;
+        delete Query;
+      }
     }
     virtual void execute(void);
 
