diff -urN sysrqd-9.orig/sysrqd.h sysrqd-9/sysrqd.h
--- sysrqd-9.orig/sysrqd.h	2007-01-19 15:33:26.000000000 +0100
+++ sysrqd-9/sysrqd.h	2007-03-05 21:22:44.000000000 +0100
@@ -4,8 +4,8 @@
 #define BIND_MAX_LEN 16
 #define PROMPT "sysrq> "
 #define SYSRQ_TRIGGER_PATH "/proc/sysrq-trigger"
-#define AUTH_FILE "/etc/sysrqd.secret"
-#define BINDIP_FILE "/etc/sysrqd.bind"
+#define AUTH_FILE "/etc/sysrqd/sysrqd.secret"
+#define BINDIP_FILE "/etc/sysrqd/sysrqd.bind"
 #define PID_FILE "/var/run/sysrqd.pid"
 #define SYSRQD_PRIO -19
 #define SYSRQD_LISTEN_PORT 4094
