? configurable-aliasdir.patch
Index: configure.ac
===================================================================
RCS file: /cvs/xorg/font/alias/configure.ac,v
retrieving revision 1.2
diff -u -b -B -r1.2 configure.ac
--- configure.ac	29 Jul 2005 21:22:47 -0000	1.2
+++ configure.ac	8 Dec 2005 20:18:22 -0000
@@ -28,6 +28,13 @@
 
 AC_PROG_INSTALL
 
+DEFAULT_ALIASDIR=${libdir}/X11/fonts
+AC_ARG_WITH(aliasdir,
+	AS_HELP_STRING([--with-top-fontdir=ALIASDIR], [Path to install font encodings]),
+	[ALIASDIR="$withval"],
+	[ALIASDIR="$DEFAULT_ALIASDIR"])
+AC_SUBST(ALIASDIR)
+
 XORG_RELEASE_VERSION
 
 AC_OUTPUT([Makefile
Index: 100dpi/Makefile.am
===================================================================
RCS file: /cvs/xorg/font/alias/100dpi/Makefile.am,v
retrieving revision 1.1
diff -u -b -B -r1.1 Makefile.am
--- 100dpi/Makefile.am	12 Jul 2005 04:23:08 -0000	1.1
+++ 100dpi/Makefile.am	8 Dec 2005 20:18:22 -0000
@@ -19,7 +19,7 @@
 #  NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 #  CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-aliasdir = $(libdir)/X11/fonts/100dpi
+aliasdir = @ALIASDIR@/100dpi
 alias_DATA = fonts.alias
 
 EXTRA_DIST = $(alias_DATA)
Index: 75dpi/Makefile.am
===================================================================
RCS file: /cvs/xorg/font/alias/75dpi/Makefile.am,v
retrieving revision 1.1
diff -u -b -B -r1.1 Makefile.am
--- 75dpi/Makefile.am	12 Jul 2005 04:23:08 -0000	1.1
+++ 75dpi/Makefile.am	8 Dec 2005 20:18:22 -0000
@@ -19,7 +19,7 @@
 #  NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 #  CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-aliasdir = $(libdir)/X11/fonts/75dpi
+aliasdir = @ALIASDIR@/75dpi
 alias_DATA = fonts.alias
 
 EXTRA_DIST = $(alias_DATA)
Index: cyrillic/Makefile.am
===================================================================
RCS file: /cvs/xorg/font/alias/cyrillic/Makefile.am,v
retrieving revision 1.1
diff -u -b -B -r1.1 Makefile.am
--- cyrillic/Makefile.am	12 Jul 2005 04:23:08 -0000	1.1
+++ cyrillic/Makefile.am	8 Dec 2005 20:18:22 -0000
@@ -19,7 +19,7 @@
 #  NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 #  CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-aliasdir = $(libdir)/X11/fonts/cyrillic
+aliasdir = @ALIASDIR@/cyrillic
 alias_DATA = fonts.alias
 
 EXTRA_DIST = $(alias_DATA)
Index: misc/Makefile.am
===================================================================
RCS file: /cvs/xorg/font/alias/misc/Makefile.am,v
retrieving revision 1.1
diff -u -b -B -r1.1 Makefile.am
--- misc/Makefile.am	12 Jul 2005 04:23:08 -0000	1.1
+++ misc/Makefile.am	8 Dec 2005 20:18:22 -0000
@@ -19,7 +19,7 @@
 #  NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 #  CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-aliasdir = $(libdir)/X11/fonts/misc
+aliasdir = @ALIASDIR@/misc
 alias_DATA = fonts.alias
 
 EXTRA_DIST = $(alias_DATA)
