--- cf/db.m4	2004-03-30 09:00:23.315562886 +0000
+++ cf/db.m4	2004-03-30 09:00:47.278672201 +0000
@@ -14,6 +14,8 @@
 if test "$enable_berkeley_db" != no; then
 
   AC_CHECK_HEADERS([				\
+	db4.2/db.h				\
+	db4.1/db.h				\
 	db4/db.h				\
 	db3/db.h				\
 	db.h					\
@@ -24,6 +26,10 @@
 
   AC_FIND_FUNC_NO_LIBS(db_create, db4 db3 db, [
   #include <stdio.h>
+  #ifdef HAVE_DB4_2_DB_H
+  #include <db4.2/db.h>
+  #ifdef HAVE_DB4_1_DB_H
+  #include <db4.1/db.h>
   #ifdef HAVE_DB4_DB_H
   #include <db4/db.h>
   #elif defined(HAVE_DB3_DB_H)
