Modified: pym/portage.py
===================================================================
--- pym/portage.py	2006-03-11 19:16:46 UTC (rev 2853)
+++ pym/portage.py	2006-03-11 19:24:32 UTC (rev 2854)
@@ -2556,7 +2556,7 @@
 		# Should be ok again to set $T, as sandbox does not depend on it
 		# XXX Bug.  no way in hell this is valid for clean handling.
 		mysettings["T"]=mysettings["PORTAGE_BUILDDIR"]+"/temp"
-		if cleanup or mydo=="clean":
+		if cleanup:
 			if os.path.exists(mysettings["T"]):
 				shutil.rmtree(mysettings["T"])
 		if not os.path.exists(mysettings["T"]):
@@ -6099,7 +6099,7 @@
 		# Process ebuild logfiles
 		elog_process(self.mycpv, self.settings)
 		if "noclean" not in features:
-			doebuild(myebuild, "clean", root, self.settings, cleanup=cleanup, tree=self.treetype)
+			doebuild(myebuild, "clean", root, self.settings, tree=self.treetype)
 		return 0
 
 	def mergeme(self,srcroot,destroot,outfile,secondhand,stufftomerge,cfgfiledict,thismtime):
