Index: pym/portage.py
===================================================================
--- pym/portage.py	(revision 2961)
+++ pym/portage.py	(working copy)
@@ -1641,15 +1641,17 @@
 
 	if free:
 		keywords["opt_name"] += " bash"
+		spawn_func = portage_exec.spawn_bash
 	else:
 		keywords["opt_name"] += " sandbox"
+		spawn_func = portage_exec.spawn_sandbox
 
 	if sesandbox:
 		con = selinux.getcontext()
 		con = string.replace(con, mysettings["PORTAGE_T"], mysettings["PORTAGE_SANDBOX_T"])
 		selinux.setexec(con)
 
-	retval = portage_exec.spawn_bash(mystring, env=env, **keywords)
+	retval = spawn_func(mystring, env=env, **keywords)
 
 	if sesandbox:
 		selinux.setexec(None)
