=== modified file 'bin/ebuild.sh'
--- bin/ebuild.sh	
+++ bin/ebuild.sh	
@@ -42,7 +42,15 @@
 OCXX="$CXX"
 source /etc/profile.env &>/dev/null
 if [ -f "${PORTAGE_BASHRC}" ]; then
-	source "${PORTAGE_BASHRC}"
+	# If $- contains x, then tracing has already enabled elsewhere for some
+	# reason.  We preserve it's state so as not to interfere.
+	if [ "$PORTAGE_DEBUG" != "1" ] || [ "${-/x/}" != "$-" ]; then
+		source "${PORTAGE_BASHRC}"
+	else
+		set -x
+		source "${PORTAGE_BASHRC}"
+		set +x
+	fi
 fi
 [ ! -z "$OCC" ] && export CC="$OCC"
 [ ! -z "$OCXX" ] && export CXX="$OCXX"

=== modified file 'pym/portage.py'
--- pym/portage.py	
+++ pym/portage.py	
@@ -2389,10 +2389,13 @@
 	if phase_retval == os.EX_OK:
 		if mydo == "install":
 			mycommand = " ".join([MISC_SH_BINARY, "install_qa_check"])
-			return spawn(mycommand, mysettings, debug=debug,
+			qa_retval = spawn(mycommand, mysettings, debug=debug,
 				droppriv=actionmap[mydo]["args"][0],
 				free=actionmap[mydo]["args"][1],
 				sesandbox=actionmap[mydo]["args"][2], logfile=logfile)
+			if qa_retval:
+				writemsg("!!! install_qa_check failed; exiting.\n")
+			return qa_retval
 	return phase_retval
 
 # chunked out deps for each phase, so that ebuild binary can use it 
@@ -2769,7 +2772,9 @@
 			# Post phase logic and tasks that have been factored out of ebuild.sh.
 			myargs = [MISC_SH_BINARY, "preinst_mask", "preinst_sfperms",
 				"preinst_selinux_labels", "preinst_suid_scan"]
-			spawn(" ".join(myargs), mysettings, debug=debug, free=1, logfile=logfile)
+			phase_retval = spawn(" ".join(myargs), mysettings, debug=debug, free=1, logfile=logfile)
+			if phase_retval != os.EX_OK:
+				writemsg("!!! post preinst failed; exiting.\n")
 		del mysettings["IMAGE"]
 		return phase_retval
 	elif mydo in ["prerm","postrm","postinst","config"]:

