--- style/baghira.cpp.orig	2004-10-07 11:00:54.027113928 +0200
+++ style/baghira.cpp	2004-10-07 11:01:11.712425352 +0200
@@ -2033,7 +2033,7 @@
 				QPen oldPen( p->pen() );
 				if ( controls & SC_ComboBoxArrow ) {
                                     if ( controls & SC_ComboBoxEditField && ( ( const QComboBox * ) widget ) ->editable()) { // editable?
-                                        if (widget->parentWidget() && ::qt_cast<QToolBar*>(widget->parentWidget()) || widget->parentWidget()->inherits("KPanelApplet")){ // toolbar combo?
+                                        if (widget->parentWidget() && ( ::qt_cast<QToolBar*>(widget->parentWidget()) || widget->parentWidget()->inherits("KPanelApplet") ) ){ // toolbar combo?
                                             drawRoundFrame( p, r, cg, (flags & Style_HasFocus) || sunken , widget->backgroundOffset());
                                             p->setPen(cg.mid());
                                             p->drawPixmap(r.right() - 16, r.height()/2 - 2, *sbDown);
