diff -uNr libgtop-2.0.6/configure libgtop-2.0.6.patched/configure
--- libgtop-2.0.6/configure	2003-10-22 17:05:07.000000000 +0200
+++ libgtop-2.0.6.patched/configure	2003-10-24 00:54:46.000000000 +0200
@@ -1853,8 +1853,8 @@
 fi
 
 
-LIBGTOP_INTERFACE_AGE=5
-LIBGTOP_BINARY_AGE=5
+LIBGTOP_INTERFACE_AGE=6
+LIBGTOP_BINARY_AGE=6
 
 # you can set this to `-snap' for instance to create
 # a `libgtop-1.x.y-snap.tar.gz' tarball.
