--- enchant-1.0.0/src/ispell/lookup.cpp.aron	2003-08-26 09:20:40.000000000 -0400
+++ enchant-1.0.0/src/ispell/lookup.cpp	2003-11-02 00:08:56.000000000 -0500
@@ -381,22 +381,22 @@
 			if (dp->word == (char *) -1)
 				dp->word = NULL;
 			else
-				dp->word = &m_hashstrings [ reinterpret_cast<int>(dp->word) ];
+				dp->word = &m_hashstrings [ reinterpret_cast<unsigned long>(dp->word) ];
 			if (dp->next == (struct dent *) -1)
 				dp->next = NULL;
 			else
-				dp->next = &m_hashtbl [ reinterpret_cast<int>(dp->next) ];
+				dp->next = &m_hashtbl [ reinterpret_cast<unsigned long>(dp->next) ];
 	    }
 	}
 
     for (i = m_numsflags + m_numpflags, entry = m_sflaglist; --i >= 0; entry++)
 	{
 		if (entry->stripl)
-			entry->strip = reinterpret_cast<ichar_t *>(&m_hashstrings[reinterpret_cast<int>(entry->strip)]);
+			entry->strip = reinterpret_cast<ichar_t *>(&m_hashstrings[reinterpret_cast<unsigned long>(entry->strip)]);
 		else
 			entry->strip = NULL;
 		if (entry->affl)
-			entry->affix = reinterpret_cast<ichar_t *>(&m_hashstrings[reinterpret_cast<int>(entry->affix)]);
+			entry->affix = reinterpret_cast<ichar_t *>(&m_hashstrings[reinterpret_cast<unsigned long>(entry->affix)]);
 		else
 			entry->affix = NULL;
 	}
