diff -ur ganeti-2.1.1/Makefile.in ganeti-2.1.1.new/Makefile.in
--- ganeti-2.1.1/Makefile.in	2010-03-12 03:04:33.000000000 -0800
+++ ganeti-2.1.1.new/Makefile.in	2010-03-12 11:07:47.000000000 -0800
@@ -1320,11 +1320,11 @@
 
 man/%.7: man/%.7.in $(REPLACE_VARS_SED)
 	sed -f $(REPLACE_VARS_SED) < $< > $@
-	$(CHECK_MAN) $@
+	#$(CHECK_MAN) $@
 
 man/%.8: man/%.8.in $(REPLACE_VARS_SED)
 	sed -f $(REPLACE_VARS_SED) < $< > $@
-	$(CHECK_MAN) $@
+	#$(CHECK_MAN) $@
 
 man/%.html: man/%.html.in $(REPLACE_VARS_SED)
 	sed -f $(REPLACE_VARS_SED) < $< > $@
