diff -ru config_office.orig/configure.in config_office/configure.in
--- config_office/configure.in	2005-09-09 08:23:03.000000000 +0100
+++ config_office/configure.in	2005-09-09 08:24:46.000000000 +0100
@@ -226,9 +226,9 @@
 AC_ARG_WITH(system-python,
 [  --with-system-python    Use python already on system
 ],,)
-#AC_ARG_WITH(system-icu,
-#[  --with-system-icu       Use icu already on system
-#],,)
+AC_ARG_WITH(system-icu,
+[  --with-system-icu       Use icu already on system
+],,)
 AC_ARG_WITH(system-db,
 [  --with-system-db        Use berkeley db already on system
 ],,)
@@ -2965,32 +2965,32 @@
 fi
 AC_SUBST(SYSTEM_SANE_HEADER)
 
-# DISABLED; INCOMPLETE
 dnl ===================================================================
 dnl Check for system icu
 dnl ===================================================================
-#AC_MSG_CHECKING([which icu to use])
-#if test -n "$with_system_icu" -o -n "$with_system_libs" && \
-#	test "$with_system_icu" != "no"; then
-#   AC_MSG_RESULT([external])
-#   SYSTEM_ICU=YES
-#   AC_MSG_CHECKING([for unicode/rbbi.h])
-#   AC_TRY_CPP(unicode/rbbi.h, AC_MSG_RESULT([checked.]), AC_MSG_ERROR([icu headers not found.]))
-#   AC_PATH_PROG(SYSTEM_GENBRK, genbrk)
-#   if test -z "$SYSTEM_GENBRK"; then
-#      AC_MSG_ERROR([\"genbrk\" not found in \$PATH, install the icu development tool \"genbrk"\])
-#   fi
-#   AC_PATH_PROG(SYSTEM_GENCCODE, genccode)
-#   if test -z "$SYSTEM_GENCCODE"; then
-#      AC_MSG_ERROR([\"genccode\" not found in \$PATH, install the icu development tool \"genccode"\])
-#   fi
-#else
-#    AC_MSG_RESULT([internal])
-#    SYSTEM_ICU=NO
-#fi
-#AC_SUBST(SYSTEM_ICU)
-#AC_SUBST(SYSTEM_GENBRK)
-#AC_SUBST(SYSTEM_GENCCODE)
+AC_MSG_CHECKING([which icu to use])
+if test -n "$with_system_icu" -o -n "$with_system_libs" && \
+	test "$with_system_icu" != "no"; then
+   AC_MSG_RESULT([external])
+   SYSTEM_ICU=YES
+   AC_MSG_CHECKING([for unicode/rbbi.h])
+   AC_TRY_CPP(unicode/rbbi.h, AC_MSG_RESULT([checked.]), AC_MSG_ERROR([icu headers not found.]))
+   AC_PATH_PROG(SYSTEM_GENBRK, genbrk)
+   if test -z "$SYSTEM_GENBRK"; then
+      AC_MSG_ERROR([\"genbrk\" not found in \$PATH, install the icu development tool \"genbrk"\])
+   fi
+   AC_PATH_PROG(SYSTEM_GENCCODE, genccode)
+   if test -z "$SYSTEM_GENCCODE"; then
+      AC_MSG_ERROR([\"genccode\" not found in \$PATH, install the icu development tool \"genccode"\])
+   fi
+else
+    AC_MSG_RESULT([internal])
+    SYSTEM_ICU=NO
+    BUILD_TYPE="$BUILD_TYPE ICU"
+fi
+AC_SUBST(SYSTEM_ICU)
+AC_SUBST(SYSTEM_GENBRK)
+AC_SUBST(SYSTEM_GENCCODE)
 
 dnl ===================================================================
 dnl Checks for libraries.
diff -ru config_office.orig/set_soenv.in config_office/set_soenv.in
--- config_office/set_soenv.in	2005-09-09 08:23:03.000000000 +0100
+++ config_office/set_soenv.in	2005-09-09 08:25:28.000000000 +0100
@@ -1731,9 +1731,9 @@
 ToFile( "SYSTEM_PYTHON",     "@SYSTEM_PYTHON@",    "e" );
 ToFile( "PYTHON_CFLAGS",     "@PYTHON_CFLAGS@",    "e" );
 ToFile( "PYTHON_LIBS",       "@PYTHON_LIBS@",      "e" );
-#ToFile( "SYSTEM_ICU",        "@SYSTEM_ICU@",       "e" );
-#ToFile( "SYSTEM_GENBRK",     "@SYSTEM_GENBRK@",    "e" );
-#ToFile( "SYSTEM_GENCCODE",   "@SYSTEM_GENCCODE@",  "e" );
+ToFile( "SYSTEM_ICU",        "@SYSTEM_ICU@",       "e" );
+ToFile( "SYSTEM_GENBRK",     "@SYSTEM_GENBRK@",    "e" );
+ToFile( "SYSTEM_GENCCODE",   "@SYSTEM_GENCCODE@",  "e" );
 ToFile( "SYSTEM_JPEG",       "@SYSTEM_JPEG@",      "e" );
 ToFile( "SYSTEM_FREETYPE",   "@SYSTEM_FREETYPE@",  "e" );
 ToFile( "FREETYPE_CFLAGS",   "@FREETYPE_CFLAGS@",  "e" );
Index: inc/collator_unicode.hxx
===================================================================
RCS file: /cvs/l10n/i18npool/inc/collator_unicode.hxx,v
retrieving revision 1.7
diff -u -p -u -r1.7 collator_unicode.hxx
--- i18npool/inc/collator_unicode.hxx	21 Jul 2005 14:26:51 -0000	1.7
+++ i18npool/inc/collator_unicode.hxx	13 Sep 2005 13:49:44 -0000
@@ -110,6 +110,7 @@ protected:
     const sal_uInt8 *rulesImage;
 private:
 	RuleBasedCollator *collator;
+	RuleBasedCollator *ucacollator;
 };
 
 #define COLLATOR( algorithm ) \
Index: source/breakiterator/makefile.mk
===================================================================
RCS file: /cvs/l10n/i18npool/source/breakiterator/makefile.mk,v
retrieving revision 1.7
diff -u -p -u -r1.7 makefile.mk
--- i18npool/source/breakiterator/makefile.mk	23 Mar 2005 10:51:00 -0000	1.7
+++ i18npool/source/breakiterator/makefile.mk	13 Sep 2005 13:49:44 -0000
@@ -79,6 +79,8 @@ MY_BRK_TXTFILES:=$(shell ls data/*.txt)
 # insert "OpenOffice" as icu package name in front of the  name of each rule file for searching on application provided data
 MY_BRK_BRKFILES:=$(subst,data/,$(MISC)$/OpenOffice_ $(MY_BRK_TXTFILES:s/.txt/.brk/))
 
+MY_BRK_NAMEDBRKFILES:=$(subst,data/,OpenOffice_ $(MY_BRK_TXTFILES:s/.txt/.brk/))
+
 # OpenOffice_icu_dat.c is a generated file from the rule file list by gencmn
 MY_MISC_CXXFILES := \
         $(MISC)$/OpenOffice_icu_dat.c \
@@ -107,12 +109,12 @@ APP1STDLIBS = $(SALLIB) \
 # so the output (OpenOffice_icu_dat.c) is changed here to include a pragma to disable the warnings.
 # Output of gencmn is redirected to OpenOffice_icu_tmp.c with the -t switch.
 $(MISC)$/OpenOffice_icu_dat.c :  $(MY_BRK_BRKFILES)
-    +$(WRAPCMD) $(SOLARBINDIR)$/gencmn -e OpenOffice_icu -n OpenOffice_icu -t tmp -S -d $(MISC) O $(mktmp $(MY_BRK_BRKFILES:t"\n"))
+    +$(WRAPCMD) gencmn -e OpenOffice_icu -n OpenOffice_icu -t tmp -S -d $(MISC) O $(mktmp $(MY_BRK_NAMEDBRKFILES:t"\n"))
     +echo $(USQ)#ifdef _MSC_VER$(USQ) > $@
     +echo $(USQ)#pragma warning( disable : 4229 4668 )$(USQ) >> $@
     +echo $(USQ)#endif$(USQ) >> $@
     +$(TYPE) $(@:s/_dat/_tmp/) >> $@
 $(MISC)$/OpenOffice_%.brk : data/%.txt
-    +$(WRAPCMD) $(SOLARBINDIR)$/genbrk -r $< -o $(MISC)$/OpenOffice_$*.brk
+    +$(WRAPCMD) genbrk -r $< -o $(MISC)$/OpenOffice_$*.brk
 $(MISC)$/%_brk.c : $(MISC)$/%.brk
-    +$(WRAPCMD) $(SOLARBINDIR)$/genccode -d $(MISC)$ $(MISC)$/$*.brk
+    +$(WRAPCMD) genccode -e OpenOffice_icu -n OpenOffice_icu -d $(MISC)$ $(MISC)$/$*.brk
Index: source/collator/collator_unicode.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/collator/collator_unicode.cxx,v
retrieving revision 1.7
diff -u -p -u -r1.7 collator_unicode.cxx
--- i18npool/source/collator/collator_unicode.cxx	30 Jul 2004 14:39:21 -0000	1.7
+++ i18npool/source/collator/collator_unicode.cxx	13 Sep 2005 13:49:44 -0000
@@ -50,11 +50,13 @@
 {
 	implementationName = "com.sun.star.i18n.Collator_Unicode";
 	collator = NULL;
+	ucacollator = NULL;
 }
 
 Collator_Unicode::~Collator_Unicode()
 {
 	if (collator) delete collator;
+	if (ucacollator) delete ucacollator;
 }
 
 sal_Int32 SAL_CALL
@@ -110,7 +112,14 @@
                 }
                 if (func) {
                     const sal_uInt8* ruleImage=func();
+#ifdef SYSTEM_ICU
+                    ucacollator = (RuleBasedCollator*) icu::Collator::createInstance("", status);
+                    if (! U_SUCCESS(status))
+                        throw RuntimeException();
+                    collator = new RuleBasedCollator(reinterpret_cast<const uint8_t*>(ruleImage), -1, ucacollator, status);
+#else
                     collator = new RuleBasedCollator(reinterpret_cast<const uint8_t*>(ruleImage), status);
+#endif
                     if (! U_SUCCESS(status))
                         throw RuntimeException();
                 }
Index: source/hhconvdic.cxx
===================================================================
RCS file: /cvs/sw/linguistic/source/hhconvdic.cxx,v
retrieving revision 1.4
diff -u -p -u -r1.4 hhconvdic.cxx
--- linguistic/source/hhconvdic.cxx	26 Nov 2004 14:29:09 -0000	1.4
+++ linguistic/source/hhconvdic.cxx	10 Sep 2005 07:47:25 -0000
@@ -60,7 +60,7 @@
  ************************************************************************/
 
 #ifndef USCRIPT_H
-#include <external/unicode/uscript.h>
+#include <unicode/uscript.h>
 #endif
 
 #ifndef _LANG_HXX //autogen wg. LANGUAGE_ENGLISH_US
diff -ru scp2.orig/prj/build.lst scp2/prj/build.lst
--- scp2/prj/build.lst	2005-09-09 10:00:30.000000000 +0100
+++ scp2/prj/build.lst	2005-09-09 08:37:48.000000000 +0100
@@ -1,4 +1,4 @@
-cp    scp2    :    setup_native transex3 PYTHON:python icu NULL
+cp    scp2    :    setup_native transex3 PYTHON:python ICU:icu NULL
 cp    scp2                        usr1     -    all    cp_mkout NULL
 cp    scp2\macros                 nmake    -    all    cp_langmacros NULL
 cp    scp2\source\activex         nmake    -    all    cp_activex cp_langmacros NULL
diff -ru scp2.orig/source/ooo/file_library_ooo.scp scp2/source/ooo/file_library_ooo.scp
--- scp2/source/ooo/file_library_ooo.scp	2005-09-09 10:00:18.000000000 +0100
+++ scp2/source/ooo/file_library_ooo.scp	2005-09-09 10:04:07.000000000 +0100
@@ -801,6 +801,8 @@
     Styles = (PACKED,PATCH);
 End
 
+#ifndef SYSTEM_ICU
+
 File gid_File_Lib_Icudata
     TXT_FILE_BODY;
   #ifdef UNX
@@ -845,6 +847,8 @@
     Styles = (PACKED);
 End
 
+#endif
+
 File gid_File_Lib_Iiopbrdg
     TXT_FILE_BODY;
     Styles = (PACKED,UNO_COMPONENT);
diff -ru scp2.orig/source/ooo/makefile.mk scp2/source/ooo/makefile.mk
--- scp2/source/ooo/makefile.mk	2005-09-09 10:00:18.000000000 +0100
+++ scp2/source/ooo/makefile.mk	2005-09-09 13:00:17.000000000 +0100
@@ -68,7 +68,6 @@
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  settings.mk
-.INCLUDE :  icuversion.mk
 .INCLUDE :  i18npool/version.mk
 
 .IF "$(ENABLE_CRASHDUMP)"!=""
@@ -177,11 +176,17 @@
 SCPDEFS+=-DENABLE_CAIRO
 .ENDIF
 
+.IF "$(SYSTEM_ICU)" == "YES"
+SCPDEFS+=-DSYSTEM_ICU
+.ELSE
+.INCLUDE :  icuversion.mk
 SCPDEFS+=\
     -DICU_MAJOR=$(ICU_MAJOR) \
     -DICU_MINOR=$(ICU_MINOR) \
-    -DICU_MICRO=$(ICU_MICRO) \
-	-DISOLANG_MAJOR=$(ISOLANG_MAJOR)
+    -DICU_MICRO=$(ICU_MICRO)
+.ENDIF
+
+SCPDEFS+=-DISOLANG_MAJOR=$(ISOLANG_MAJOR)
 
 .IF "$(DISABLE_NEON)" == "TRUE"
 SCPDEFS+=-DDISABLE_NEON
diff -ru scp2.orig/source/ooo/shortcut_ooo.scp scp2/source/ooo/shortcut_ooo.scp
--- scp2/source/ooo/shortcut_ooo.scp	2005-09-09 10:00:18.000000000 +0100
+++ scp2/source/ooo/shortcut_ooo.scp	2005-09-09 10:05:21.000000000 +0100
@@ -193,6 +193,7 @@
 #endif
 
 #ifdef UNX
+#ifndef SYSTEM_ICU
 
 Shortcut gid_Shortcut_Lib_Icudata_0
     FileID = gid_File_Lib_Icudata;
@@ -201,10 +202,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icudata_1
     FileID = gid_File_Lib_Icudata;
     Dir = gid_Dir_Program;
@@ -212,10 +209,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icui18n_0
     FileID = gid_File_Lib_Icui18n;
     Dir = gid_Dir_Program;
@@ -223,10 +216,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icui18n_1
     FileID = gid_File_Lib_Icui18n;
     Dir = gid_Dir_Program;
@@ -234,10 +223,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icule_0
     FileID = gid_File_Lib_Icule;
     Dir = gid_Dir_Program;
@@ -245,10 +230,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icule_1
     FileID = gid_File_Lib_Icule;
     Dir = gid_Dir_Program;
@@ -256,10 +237,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icuuc_0
     FileID = gid_File_Lib_Icuuc;
     Dir = gid_Dir_Program;
@@ -267,10 +244,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icuuc_1
     FileID = gid_File_Lib_Icuuc;
     Dir = gid_Dir_Program;
@@ -279,6 +252,7 @@
 End
 
 #endif
+#endif
 
 #ifdef UNX
 
--- vcl/prj/build.lst	2005-09-13 08:51:51.000000000 +0100
+++ vcl/prj/build.lst	2005-09-09 08:37:54.000000000 +0100
@@ -1,4 +1,4 @@
-vc	vcl	:	BOOST:boost NAS:nas FREETYPE:freetype psprint rsc sot ucbhelper unotools icu i18npool unoil ridljar X11_EXTENSIONS:x11_extensions offuh basegfx SNDFILE:sndfile PORTAUDIO:portaudio transex3 SO:officenames NULL
+vc	vcl	:	BOOST:boost NAS:nas FREETYPE:freetype psprint rsc sot ucbhelper unotools ICU:icu i18npool unoil ridljar X11_EXTENSIONS:x11_extensions offuh basegfx SNDFILE:sndfile PORTAUDIO:portaudio transex3 SO:officenames NULL
 vc	vcl										usr1	-	all	vc_mkout NULL
 vc	vcl\source\unotypes						nmake	-	all	vc_unot NULL
 vc	vcl\source\glyphs						nmake	-	all	vc_glyphs vc_unot NULL
Index: prj/build.lst
===================================================================
RCS file: /cvs/l10n/i18npool/prj/build.lst,v
retrieving revision 1.18
diff -u -r1.18 build.lst
--- i18npool/prj/build.lst	1 Nov 2005 14:51:56 -0000	1.18
+++ i18npool/prj/build.lst	5 Nov 2005 16:13:03 -0000
@@ -1,4 +1,4 @@
-inp  i18npool    :   bridges sax stoc comphelper icu i18nutil regexp NULL
+inp  i18npool    :   bridges sax stoc comphelper ICU:icu i18nutil regexp NULL
 inp  i18npool                                   usr1    -   all inp_mkout NULL
 inp  i18npool\source\unotypes                   nmake   -   all inp_utypes NULL
 inp  i18npool\source\registerservices           nmake   -   all inp_rserv inp_utypes NULL
Index: source/breakiterator/breakiterator_unicode.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/breakiterator/breakiterator_unicode.cxx,v
retrieving revision 1.20
diff -u -r1.20 breakiterator_unicode.cxx
--- i18npool/source/breakiterator/breakiterator_unicode.cxx	17 Oct 2005 15:42:30 -0000	1.20
+++ i18npool/source/breakiterator/breakiterator_unicode.cxx	5 Nov 2005 16:14:56 -0000
@@ -42,7 +42,9 @@
 extern const char OpenOffice_icu_dat[];
 U_CDECL_END
 
-using namespace ::com::sun::star::uno;
+using ::com::sun::star::uno::RuntimeException;
+using ::com::sun::star::uno::Reference;
+using ::com::sun::star::uno::Sequence;
 using namespace ::com::sun::star::lang;
 using namespace ::rtl;
 
@@ -103,13 +105,21 @@
             if ( !U_SUCCESS(status) ) throw ERROR;
 
             status = U_ZERO_ERROR;
-            icuBI->aBreakIterator = new RuleBasedBreakIterator(udata_open("OpenOffice", "brk", 
+            UDataMemory *pData;
+            if (!rLocale.Language.getLength())
+                pData=0;
+            else
+                pData = udata_open("OpenOffice_icu/OpenOffice", "brk", 
                     OUStringToOString(OUString::createFromAscii(rule)+OUString::createFromAscii("_")+rLocale.Language,
-                    RTL_TEXTENCODING_ASCII_US).getStr(), &status), status);
-            if (!U_SUCCESS(status) ) {
+                    RTL_TEXTENCODING_ASCII_US).getStr(), &status);
+
+            icuBI->aBreakIterator = (pData && U_SUCCESS(status)) ? new RuleBasedBreakIterator(pData, status) : 0;
+
+            if ((!U_SUCCESS(status) || !icuBI->aBreakIterator)) {
                 status = U_ZERO_ERROR;
-                icuBI->aBreakIterator = new RuleBasedBreakIterator(udata_open("OpenOffice", "brk", rule, &status), status);
-                if (!U_SUCCESS(status) ) icuBI->aBreakIterator=NULL;
+                pData = udata_open("OpenOffice_icu/OpenOffice", "brk", rule, &status);
+                icuBI->aBreakIterator = (pData && U_SUCCESS(status)) ? new RuleBasedBreakIterator(pData, status) : 0;
+                if (!U_SUCCESS(status) ) icuBI->aBreakIterator=NULL;
             }
         }
 
--- i18npool/source/collator/makefile.mk.orig	2005-12-07 21:29:09.591358096 +0100
+++ i18npool/source/collator/makefile.mk	2005-12-07 21:29:33.881665408 +0100
@@ -88,6 +88,10 @@
 		$(ICUINLIB) \
 		$(ICUUCLIB)
 
+.IF "$(SYSTEM_ICU)" == "YES"
+CFLAGS += -DSYSTEM_ICU
+.ENDIF
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :	target.mk
