From 9316037f4392249a4f9ef4c9aea89e424b519468 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <benjamin@sipsolutions.net>
Date: Sat, 22 May 2010 10:31:56 +0000
Subject: Fix evolution table header workaround for new evo versions.

---
diff --git a/gtk-themes/ClearlooksClassic/gtkrc b/gtk-themes/ClearlooksClassic/gtkrc
index af9f716..96d7316 100644
--- a/gtk-themes/ClearlooksClassic/gtkrc
+++ b/gtk-themes/ClearlooksClassic/gtkrc
@@ -376,9 +376,9 @@ widget "gtk-tooltip*" style "tooltips"
 ##########################################################################
 
 # Workaround for the evolution ETable (bug #527532)
-widget_class "*.ETable.ECanvas" style "treeview_header"
+widget_class "*.<ETable>.<ECanvas>" style "treeview_header"
 # Workaround for the evolution ETree
-widget_class "*.ETree.ECanvas" style "treeview_header"
+widget_class "*.<ETree>.<ECanvas>" style "treeview_header"
 
 # Special case the nautilus-extra-view-widget
 # ToDo: A more generic approach for all applications that have a widget like this.
diff --git a/gtk-themes/ClearlooksTest/gtkrc b/gtk-themes/ClearlooksTest/gtkrc
index 8a2c4ee..ede08cc 100644
--- a/gtk-themes/ClearlooksTest/gtkrc
+++ b/gtk-themes/ClearlooksTest/gtkrc
@@ -398,9 +398,9 @@ widget "gtk-tooltip*" style "clearlooks-tooltips"
 ##########################################################################
 
 # Workaround for the evolution ETable (bug #527532)
-widget_class "*.ETable.ECanvas" style "clearlooks-treeview_header"
+widget_class "*.<ETable>.<ECanvas>" style "clearlooks-treeview_header"
 # Workaround for the evolution ETree
-widget_class "*.ETree.ECanvas" style "clearlooks-treeview_header"
+widget_class "*.<ETree>.<ECanvas>" style "clearlooks-treeview_header"
 
 # Special case the nautilus-extra-view-widget
 # ToDo: A more generic approach for all applications that have a widget like this.
diff --git a/gtk-themes/Glossy/gtkrc b/gtk-themes/Glossy/gtkrc
index a2e7baa..b15dd6d 100644
--- a/gtk-themes/Glossy/gtkrc
+++ b/gtk-themes/Glossy/gtkrc
@@ -377,9 +377,9 @@ widget "gtk-tooltip*" style "tooltips"
 ##########################################################################
 
 # Workaround for the evolution ETable (bug #527532)
-widget_class "*.ETable.ECanvas" style "treeview_header"
+widget_class "*.<ETable>.<ECanvas>" style "treeview_header"
 # Workaround for the evolution ETree
-widget_class "*.ETree.ECanvas" style "treeview_header"
+widget_class "*.<ETree>.<ECanvas>" style "treeview_header"
 
 # Special case the nautilus-extra-view-widget
 # ToDo: A more generic approach for all applications that have a widget like this.
diff --git a/gtk-themes/Inverted/gtkrc b/gtk-themes/Inverted/gtkrc
index fd3cccf..9991204 100644
--- a/gtk-themes/Inverted/gtkrc
+++ b/gtk-themes/Inverted/gtkrc
@@ -375,9 +375,9 @@ widget "gtk-tooltip*" style "tooltips"
 ##########################################################################
 
 # Workaround for the evolution ETable (bug #527532)
-widget_class "*.ETable.ECanvas" style "treeview_header"
+widget_class "*.<ETable>.<ECanvas>" style "treeview_header"
 # Workaround for the evolution ETree
-widget_class "*.ETree.ECanvas" style "treeview_header"
+widget_class "*.<ETree>.<ECanvas>" style "treeview_header"
 
 # Special case the nautilus-extra-view-widget
 # ToDo: A more generic approach for all applications that have a widget like this.
--
cgit v0.8.3.1
