diff -Naur tmp-old/src/xdrv/wcmCommon.c tmp-new/src/xdrv/wcmCommon.c
--- src/xdrv/wcmCommon.c	2007-09-16 19:57:42.000000000 +0000
+++ src/xdrv/wcmCommon.c	2007-09-16 20:39:26.000000000 +0000
@@ -1096,7 +1096,7 @@
 #if defined WCM_XFREE86 || GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 0
 	if (!miPointerCurrentScreen())
 #else
-	if (!miPointerGetScreen(pDev->dev))
+	if (!miPointerGetScreen(common->wcmDevices->local->dev))
 #endif
 	{
 		DBG(1, common->debugLevel, ErrorF("xf86WcmEvent: "
