--- hotplug-20030805/etc/hotplug/hotplug.functions.orig	2003-09-24 09:35:09.000000000 -0400
+++ hotplug-20030805/etc/hotplug/hotplug.functions	2003-09-24 15:59:20.000000000 -0400
@@ -139,7 +139,9 @@
     do
 	# maybe driver modules need loading
         LOADED=false
-	if ! lsmod | grep -q "^$MODULE " > /dev/null 2>&1; then
+	if lsmod | grep -q "^$MODULE "; then
+	    LOADED=true
+	else
 	    if grep -q "^$MODULE\$" $HOTPLUG_DIR/blacklist \
 		    >/dev/null 2>&1; then
 		debug_mesg "... blacklisted module:  $MODULE"
