--- pci.rc.orig	Fri May 10 18:31:51 2002
+++ pci.rc	Fri May 10 19:02:00 2002
@@ -9,27 +9,14 @@
 
 cd /etc/hotplug
 
-# source function library
-if [ -f /etc/init.d/functions ]; then
-	. /etc/init.d/functions
-elif [ -f /etc/rc.d/init.d/functions ]; then
-	. /etc/rc.d/init.d/functions
-fi
-
 . hotplug.functions
 
-
-# # override any of the defaults?
-# if [ -f /etc/sysconfig/pci ]; then
-#     . /etc/sysconfig/pci
-# fi
-
 pci_boot_events ()
 {
     LISTER=`type -p pcimodules`
     if [ "$LISTER" = "" -o ! -f /proc/bus/pci/devices -o ! -x pci.agent ]; then
 	    echo $"** can't synthesize pci hotplug events"
-	    return
+	    return 1
     fi
 
     # make sure the pci agent will run
@@ -61,7 +48,6 @@
 	pci_boot_events
         ;;
   stop)
-	echo $"pci stop -- ignored"
         ;;
   status)
 	echo $"PCI Status for kernel: "  `uname -srm`
