diff -Naur ftpcube-0.5.1-orig/libftpcube/__init__.py ftpcube-0.5.1/libftpcube/__init__.py
--- ftpcube-0.5.1-orig/libftpcube/__init__.py	2007-01-07 16:10:35.000000000 -0600
+++ ftpcube-0.5.1/libftpcube/__init__.py	2007-12-04 20:07:55.000000000 -0600
@@ -30,6 +30,11 @@
 
 wx_failure = None
 try:
+    import wxversion
+    wxversion.select("2.6")
+except Exception, strerror:
+    wx_failure = strerror
+try:
     import wx
 except ImportError, strerror:
     wx_failure = strerror
@@ -87,4 +92,4 @@
     'url',
     'utils',
     'version',
-]
\ No newline at end of file
+]
