--- src/doveadm/Makefile.in.orig	2010-08-04 17:34:24.000000000 +0000
+++ src/doveadm/Makefile.in	2010-08-05 07:38:30.000000000 +0000
@@ -293,8 +293,9 @@
 	$(unused_objects)
 
 doveadm_LDADD = \
-	$(libs) $(AUTH_LIBS) \
+	$(libs) \
 	$(cmd_pw_libs) \
+	$(AUTH_LIBS) \
 	$(LIBDOVECOT) \
 	$(MODULE_LIBS)
 
--- src/doveadm/Makefile.am.orig	2010-07-30 14:04:48.000000000 +0000
+++ src/doveadm/Makefile.am	2010-08-05 07:38:30.000000000 +0000
@@ -38,8 +38,9 @@
 	$(unused_objects)
 
 doveadm_LDADD = \
-	$(libs) $(AUTH_LIBS) \
+	$(libs) \
 	$(cmd_pw_libs) \
+	$(AUTH_LIBS) \
 	$(LIBDOVECOT) \
 	$(MODULE_LIBS)
 doveadm_DEPENDENCIES = \
