diff -ur dovecot-1.0.rc6.orig/src/login-common/ssl-proxy-openssl.c dovecot-1.0.rc6/src/login-common/ssl-proxy-openssl.c
--- dovecot-1.0.rc6.orig/src/login-common/ssl-proxy-openssl.c	2006-08-08 10:39:34.194002880 +0100
+++ dovecot-1.0.rc6/src/login-common/ssl-proxy-openssl.c	2006-08-08 10:40:37.000000000 +0100
@@ -654,7 +654,7 @@
 
 unsigned int ssl_proxy_get_count(void)
 {
-	return hash_size(ssl_proxies);
+	return ssl_proxies == NULL ? 0 : hash_size(ssl_proxies);
 }
 
 void ssl_proxy_init(void)
