diff -ur dovecot-1.0.rc29/src/auth/db-ldap.h dovecot-1.0.rc29/src/auth/db-ldap.h
--- dovecot-1.0.rc29/src/auth/db-ldap.h	2007-01-19 15:15:13.000000000 +0000
+++ dovecot-1.0.rc29/src/auth/db-ldap.h	2007-04-12 09:29:06.000000000 +0100
@@ -1,6 +1,10 @@
 #ifndef __DB_LDAP_H
 #define __DB_LDAP_H
 
+/* Functions like ldap_bind have been deprecated in openldap 2.3
+ * This define enables them until the code here can be refactored */
+#define LDAP_DEPRECATED 1
+
 #include <ldap.h>
 
 struct auth_request;
