diff -uNr asterisk-1.6.0.6.ORIG/autoconf/ast_check_gnu_make.m4 asterisk-1.6.0.6/autoconf/ast_check_gnu_make.m4
--- asterisk-1.6.0.6.ORIG/autoconf/ast_check_gnu_make.m4	2009-03-10 22:45:49.000000000 +0000
+++ asterisk-1.6.0.6/autoconf/ast_check_gnu_make.m4	2009-03-10 22:43:07.000000000 +0000
@@ -1,20 +1,20 @@
-AC_DEFUN([AST_CHECK_GNU_MAKE], [AC_CACHE_CHECK(for GNU make, GNU_MAKE,
-   GNU_MAKE='Not Found' ;
-   GNU_MAKE_VERSION_MAJOR=0 ;
-   GNU_MAKE_VERSION_MINOR=0 ;
+AC_DEFUN([AST_CHECK_GNU_MAKE], [AC_CACHE_CHECK([for GNU make], [ac_cv_GNU_MAKE],
+   ac_cv_GNU_MAKE='Not Found' ;
+   ac_cv_GNU_MAKE_VERSION_MAJOR=0 ;
+   ac_cv_GNU_MAKE_VERSION_MINOR=0 ;
    for a in make gmake gnumake ; do
       if test -z "$a" ; then continue ; fi ;
       if ( sh -c "$a --version" 2> /dev/null | grep GNU  2>&1 > /dev/null ) ;  then
-         GNU_MAKE=$a ;
-         GNU_MAKE_VERSION_MAJOR=`$GNU_MAKE --version | grep "GNU Make" | cut -f3 -d' ' | cut -f1 -d'.'`
-         GNU_MAKE_VERSION_MINOR=`$GNU_MAKE --version | grep "GNU Make" | cut -f2 -d'.' | cut -c1-2`
+         ac_cv_GNU_MAKE=$a ;
+         ac_cv_GNU_MAKE_VERSION_MAJOR=`$ac_cv_GNU_MAKE --version | grep "GNU Make" | cut -f3 -d' ' | cut -f1 -d'.'`
+         ac_cv_GNU_MAKE_VERSION_MINOR=`$ac_cv_GNU_MAKE --version | grep "GNU Make" | cut -f2 -d'.' | cut -c1-2`
          break;
       fi
    done ;
 ) ;
-if test  "x$GNU_MAKE" = "xNot Found"  ; then
+if test  "x$ac_cv_GNU_MAKE" = "xNot Found"  ; then
    AC_MSG_ERROR( *** Please install GNU make.  It is required to build Asterisk!)
    exit 1
 fi
-AC_SUBST([GNU_MAKE])
+AC_SUBST([GNU_MAKE], [$ac_cv_GNU_MAKE])
 ])
diff -uNr asterisk-1.6.0.6.ORIG/autoconf/ast_prog_sed.m4 asterisk-1.6.0.6/autoconf/ast_prog_sed.m4
--- asterisk-1.6.0.6.ORIG/autoconf/ast_prog_sed.m4	2009-03-10 22:45:49.000000000 +0000
+++ asterisk-1.6.0.6/autoconf/ast_prog_sed.m4	2009-03-10 22:44:17.000000000 +0000
@@ -12,7 +12,7 @@
      done
      echo "$ac_script" | sed 99q >conftest.sed
      $as_unset ac_script || ac_script=
-     _AC_PATH_PROG_FEATURE_CHECK(SED, [sed gsed],
+     ifdef([_AC_PATH_PROGS_FEATURE_CHECK], [_AC_PATH_PROGS_FEATURE_CHECK], [_AC_PATH_PROG_FEATURE_CHECK])(SED, [sed gsed],
 	[_AC_FEATURE_CHECK_LENGTH([ac_path_SED], [ac_cv_path_SED],
 		["$ac_path_SED" -f conftest.sed])])])
  SED="$ac_cv_path_SED"
diff -uNr asterisk-1.6.0.6.ORIG/configure.ac asterisk-1.6.0.6/configure.ac
--- asterisk-1.6.0.6.ORIG/configure.ac	2009-03-10 22:45:52.000000000 +0000
+++ asterisk-1.6.0.6/configure.ac	2009-03-10 22:43:07.000000000 +0000
@@ -23,7 +23,6 @@
 AC_COPYRIGHT("Asterisk")
 AC_REVISION($Revision: 1.1 $)
 
-AC_GNU_SOURCE
 AC_USE_SYSTEM_EXTENSIONS	# note- does not work on FreeBSD
 
 case "${host_os}" in
