--- ext/ffmpeg/gstffmpegcodecmap.c.old	2008-05-12 17:24:59.000000000 +0200
+++ ext/ffmpeg/gstffmpegcodecmap.c	2008-05-12 17:26:20.000000000 +0200
@@ -1051,6 +1051,9 @@
             "samplesize", G_TYPE_INT, context->bits_per_sample, NULL);
       }
       break;
+    case CODEC_ID_VORBIS:
+      caps = gst_ff_aud_caps_new (context, codec_id, "audio/x-vorbis", NULL);
+      break;
     default:
       g_warning ("Unknown codec ID %d, please add here", codec_id);
       break;
