diff -ruN xdotool-20091231.02.orig/Makefile xdotool-20091231.02/Makefile
--- xdotool-20091231.02.orig/Makefile	2010-01-01 04:28:54.000000000 +0100
+++ xdotool-20091231.02/Makefile	2010-01-14 05:53:51.000000000 +0100
@@ -29,19 +29,17 @@
 install: installlib installprog installman installheader
 
 installprog: xdotool
-	install -m 755 xdotool $(INSTALLBIN)/
+	install -D -m 755 xdotool $(INSTALLBIN)/xdotool
 
 installlib: libxdo.so
-	install libxdo.so $(INSTALLLIB)/libxdo.so.$(MINOR)
+	install -D libxdo.so $(INSTALLLIB)/libxdo.so.$(MINOR)
 	ln -sf libxdo.so.$(MINOR) $(INSTALLLIB)/libxdo.so
 
 installheader: xdo.h
-	install xdo.h $(INSTALLINCLUDE)/xdo.h
+	install -D xdo.h $(INSTALLINCLUDE)/xdo.h
 
 installman: xdotool.1
-	[ -d $(INSTALLMAN) ] || mkdir $(INSTALLMAN)
-	[ -d $(INSTALLMAN)/man1 ] || mkdir $(INSTALLMAN)/man1
-	install -m 644 xdotool.1 $(INSTALLMAN)/man1/
+	install -D -m 644 xdotool.1 $(INSTALLMAN)/man1/xdotool.1
 
 deinstall: uninstall
 uninstall: 
