? a.diff
Index: configure.ac
===================================================================
RCS file: /usr/local/lyx/cvsroot/lyx-devel/config/configure.ac,v
retrieving revision 1.23
diff -u -r1.23 configure.ac
--- configure.ac	30 Jan 2003 10:05:18 -0000	1.23
+++ configure.ac	7 Feb 2003 10:39:25 -0000
@@ -238,7 +238,7 @@
 AC_TYPE_SIZE_T
 AC_TYPE_UID_T
 
-AC_CHECK_FUNCS(snprintf vsnprintf)
+AC_CHECK_FUNCS(snprintf vsnprintf strerror)
 LYX_CHECK_DECL(snprintf, stdio.h)
 LYX_CHECK_DECL(vsnprintf, stdio.h)
 LYX_CHECK_DECL(istreambuf_iterator, iterator)
Index: configure.in
===================================================================
RCS file: /usr/local/lyx/cvsroot/lyx-devel/config/configure.in,v
retrieving revision 1.16
