--- lyx-1.3.2/lib/configure	2003-05-07 00:01:50.000000000 +1200
+++ lyx-1.3.2-gentoo/lib/configure	2003-09-01 03:56:32.000000000 +1200
@@ -1151,7 +1151,7 @@
     esac
   done | sed -e 's%^.*layouts/\(.*\)\.layout$%\\TestDocClass{\1}%'\
 	     > chklayouts.tex
-  eval ${LATEX} wrap_chkconfig.ltx 2>/dev/null | grep '^\+'
+  eval ${LATEX} --no-mktex=tfm wrap_chkconfig.ltx 2>/dev/null | grep '^\+'
   eval `cat chkconfig.vars | sed 's/-/_/g'`
 fi
 

