diff -urN lyx-1.4.0.orig/lib/configure lyx-1.4.0/lib/configure
--- lyx-1.4.0.orig/lib/configure	2006-03-07 00:49:49.000000000 +0100
+++ lyx-1.4.0/lib/configure	2006-03-29 18:07:01.000000000 +0200
@@ -997,7 +997,7 @@
 echo $ac_n "checking for a DVI to PS converter""... $ac_c"
 echo "$ac_t"""
 dvi_to_ps_command=
-for ac_prog in "dvips -o \$\$o \$\$i"
+for ac_prog in "dvips -R0 -o \$\$o \$\$i"
 do
 # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog ; ac_word=$2
diff -urN lyx-1.4.0.orig/lib/configure.m4 lyx-1.4.0/lib/configure.m4
--- lyx-1.4.0.orig/lib/configure.m4	2006-02-03 22:47:25.000000000 +0100
+++ lyx-1.4.0/lib/configure.m4	2006-03-29 18:07:21.000000000 +0200
@@ -302,7 +302,7 @@
 SEARCH_PROG([for a PS to PDF converter],ps_to_pdf_command, "ps2pdf13 \$\$i")
 
 # Search for a program to convert dvi to ps
-SEARCH_PROG([for a DVI to PS converter],dvi_to_ps_command, "dvips -o \$\$o \$\$i")
+SEARCH_PROG([for a DVI to PS converter],dvi_to_ps_command, "dvips -R0 -o \$\$o \$\$i")
 
 # Search for a program to convert dvi to pdf
 SEARCH_PROG([for a DVI to PDF converter],dvi_to_pdf_command, "dvipdfm \$\$i")
diff -urN lyx-1.4.0.orig/lib/lyxrc.defaults lyx-1.4.0/lib/lyxrc.defaults
--- lyx-1.4.0.orig/lib/lyxrc.defaults	2006-03-07 00:50:44.000000000 +0100
+++ lyx-1.4.0/lib/lyxrc.defaults	2006-03-29 18:05:20.000000000 +0200
@@ -53,7 +53,7 @@
 \converter docbook    dvi        "db2dvi $$i"	""
 \converter docbook    html       "db2html $$i"	""
 \converter dvi        pdf3       "dvipdfm $$i"	""
-\converter dvi        ps         "dvips -o $$o $$i"	""
+\converter dvi        ps         "dvips -R0 -o $$o $$i"	""
 \converter fen        asciichess "python $$s/scripts/fen2ascii.py $$i $$o"	""
 \converter fig        pdftex     "sh $$s/scripts/fig2pdftex.sh $$i $$o"	""
 \converter fig        pstex      "sh $$s/scripts/fig2pstex.sh $$i $$o"	""
diff -urN lyx-1.4.0.orig/lib/lyxrc.example lyx-1.4.0/lib/lyxrc.example
--- lyx-1.4.0.orig/lib/lyxrc.example	2005-09-08 11:20:16.000000000 +0200
+++ lyx-1.4.0/lib/lyxrc.example	2006-03-29 18:06:12.000000000 +0200
@@ -348,7 +348,7 @@
 # The converter command has 4 arguments: The source format, the target format,
 # the command, and additional flags.
 # For example, the following defines a DVI->Postscript converter:
-#\converter dvi ps "dvips -o $$o $$i" ""
+#\converter dvi ps "dvips -R0 -o $$o $$i" ""
 # The variable name $$i is replaced with the name of the source file,
 # and $$o is replaced with the name of the target file.
 # The flags argument is a list of comma separated flags.
