diff -Naur lyx-1.5.0.orig/lib/configure.py lyx-1.5.0/lib/configure.py
--- lyx-1.5.0.orig/lib/configure.py	2007-07-25 06:08:25.000000000 +0900
+++ lyx-1.5.0/lib/configure.py	2007-07-27 08:05:53.000000000 +0900
@@ -214,7 +214,7 @@
 \\@@end
 ''')
         # run latex on chklatex.ltx and check result
-        if cmdOutput(LATEX + ' chklatex.ltx').find('ThisIsLaTeX2e') != -1:
+        if cmdOutput(LATEX + ' --no-mktex=tfm chklatex.ltx').find('ThisIsLaTeX2e') != -1:
             # valid latex2e
             return LATEX
         else:
@@ -386,7 +386,7 @@
     checkProg('a DVI to TXT converter', ['catdvi $$i > $$o'],
         rc_entry = [ r'\converter dvi        text4      "%%"	""' ])
     #
-    checkProg('a DVI to PS converter', ['dvips -o $$o $$i'],
+    checkProg('a DVI to PS converter', ['dvips -R0 -o $$o $$i'],
         rc_entry = [ r'\converter dvi        ps         "%%"	""' ])
     #
     checkProg('a DVI to PDF converter', ['dvipdfmx -o $$o $$i', 'dvipdfm -o $$o $$i'],
@@ -650,7 +650,7 @@
         cl.close()
         #
         # we have chklayouts.tex, then process it
-        fout = os.popen(LATEX + ' wrap_chkconfig.ltx')
+        fout = os.popen(LATEX + ' --no-mktex=tfm wrap_chkconfig.ltx')
         while True:
             line = fout.readline()
             if not line:
