diff -urN lyx-1.3.6/lib/configure lyx-1.3.6-gentoo/lib/configure
--- lyx-1.3.6/lib/configure	2005-07-17 01:07:13.000000000 +0900
+++ lyx-1.3.6-gentoo/lib/configure	2005-07-31 18:17:06.000000000 +0900
@@ -779,9 +779,9 @@
 
 # Search for a spellchecker
 echo $ac_n "checking for a spell-checker""... $ac_c"
-echo "$ac_t""(ispell)"
+echo "$ac_t""(aspell)"
 prog=
-for ac_prog in ispell
+for ac_prog in aspell
 do
 # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog ; ac_word=$2
@@ -1221,7 +1221,7 @@
 	 fi ;;
     esac
   done > chklayouts.tex
-  ${LATEX} wrap_chkconfig.ltx 2>/dev/null | grep '^\+'
+  ${LATEX} --no-mktex=tfm wrap_chkconfig.ltx 2>/dev/null | grep '^\+'
   eval `cat chkconfig.vars | sed 's/-/_/g'`
   test -n "${rmcopy}" && rm -f chkconfig.ltx
 fi
@@ -1351,6 +1351,11 @@
 
 $rc_entries
 \\font_encoding "$chk_fontenc"
+
+# Gentoo systems have TrueType fonts available, use them by default
+\\screen_font_roman "Times New Roman"
+\\screen_font_sans "Verdana"
+\\screen_font_typewriter "Courier New"
 \\tex_allows_spaces $tex_allows_spaces
 EOF
 
