diff -uNr cluster-1.03.00/gnbd-kernel/src/gnbd.c cluster-1.04.00/gnbd-kernel/src/gnbd.c
--- cluster-1.03.00/gnbd-kernel/src/gnbd.c	2006-07-12 19:34:29.000000000 +0200
+++ cluster-1.04.00/gnbd-kernel/src/gnbd.c	2007-02-07 16:25:05.000000000 +0100
@@ -578,10 +577,10 @@
 		struct gnbd_device *dev;
 
 		blkdev_dequeue_request(req);
-		dprintk(DBG_BLKDEV, "%s: request %p: dequeued (flags=%lx)\n",
-				req->rq_disk->disk_name, req, req->flags);
+		dprintk(DBG_BLKDEV, "%s: request %p: dequeued (flags=%x)\n",
+				req->rq_disk->disk_name, req, req->cmd_type);
 
-		if (!(req->flags & REQ_CMD))
+		if (!blk_fs_request(req))
 			goto error_out;
 		
 		dev = req->rq_disk->private_data;
@@ -890,12 +888,12 @@
 		printk(KERN_CRIT "gnbd: sizeof gnbd_request needs to be 28 in order to work!\n" );
 		return -EIO;
 	}
-	shutdown_req.flags = REQ_SPECIAL;
+	shutdown_req.cmd_type = REQ_TYPE_SPECIAL;
 	gnbd_cmd(&shutdown_req) = GNBD_CMD_DISC;
 	shutdown_req.sector = 0;
 	shutdown_req.nr_sectors = 0;
 
-	ping_req.flags = REQ_SPECIAL;
+	ping_req.cmd_type = REQ_TYPE_SPECIAL;
 	gnbd_cmd(&ping_req) = GNBD_CMD_PING;
 	ping_req.sector = 0;
 	ping_req.nr_sectors = 0;
