diff -uNr cluster-1.03.00.orig/gnbd-kernel/src/gnbd.c cluster-1.03.00/gnbd-kernel/src/gnbd.c
--- cluster-1.03.00.orig/gnbd-kernel/src/gnbd.c	2006-07-12 19:34:29.000000000 +0200
+++ cluster-1.03.00/gnbd-kernel/src/gnbd.c	2007-03-03 01:18:40.000000000 +0100
@@ -30,7 +30,6 @@
 #include <linux/buffer_head.h>
 #include <linux/miscdevice.h>
 #include <linux/moduleparam.h>
-#include <linux/devfs_fs_kernel.h>
 
 #include <asm/uaccess.h>
 #include <asm/types.h>
@@ -845,7 +844,6 @@
 {
         .minor = MISC_DYNAMIC_MINOR,
         .name  = "gnbd_ctl",
-        .devfs_name = "gnbd_ctl",
         .fops = &_gnbd_ctl_fops
 };
 
@@ -926,7 +924,6 @@
 	printk(KERN_INFO "gnbd: registered device at major %d\n", major_nr);
 	dprintk(DBG_INIT, "gnbd: debugflags=0x%x\n", debugflags);
 
-	devfs_mk_dir("gnbd_minor");
 	err = class_register(&gnbd_class);
 	if (err)
 		goto out_unregister;
@@ -982,7 +979,6 @@
 		disk->fops = &gnbd_fops;
 		disk->private_data = &gnbd_dev[i];
 		sprintf(disk->disk_name, "gnbd%d", i);
-		sprintf(disk->devfs_name, "gnbd_minor/%d", i);
 		set_capacity(disk, 0);
 		add_disk(disk);
 		if(sysfs_create_link(&gnbd_dev[i].class_dev.kobj,
@@ -1036,7 +1032,6 @@
 			kfree(gnbd_dev[i].server_name);
 	}
 	class_unregister(&gnbd_class);
-	devfs_remove("gnbd");
 	unregister_blkdev(major_nr, "gnbd");
 	printk(KERN_INFO "gnbd: unregistered device at major %d\n", major_nr);
 }
