--- fcpackage.2_1/fontconfig/setfontdirs.orig	2002-12-23 11:38:24.000000000 +0200
+++ fcpackage.2_1/fontconfig/setfontdirs	2002-12-23 11:40:02.000000000 +0200
@@ -6,7 +6,7 @@
 export LANG
 FONTDIRS=fontdirs$$
 trap "rm $FONTDIRS" 0
-sh ./findfonts ${1+"$@"} > $FONTDIRS
+sh ./findfonts ${1+"$@"} | awk '!/\/usr\/X11R6\/lib\/X11\/fonts/ { print }' > $FONTDIRS
 cp fonts.conf.in fonts.conf
 chmod +w fonts.conf
 EDITOR=ed
@@ -21,10 +21,10 @@
 fi
 $EDITOR fonts.conf << EOF
 /FONTPATH_END/a
-<!-- Font directory list configured on `date` -->
 .
 +r $FONTDIRS
 a
+	<dir>/usr/X11R6/lib/X11/fonts</dir>
 	<dir>~/.fonts</dir>
 
 .
