diff -uNr gnome-system-monitor-2.10.1/src/prettytable.c gnome-system-monitor-2.10.1.patched/src/prettytable.c
--- gnome-system-monitor-2.10.1/src/prettytable.c	2005-04-05 14:59:18.000000000 +0200
+++ gnome-system-monitor-2.10.1.patched/src/prettytable.c	2005-04-29 00:25:46.000000000 +0200
@@ -157,6 +157,9 @@
 {
 	GError *error = NULL;
 	GdkPixbuf *scaled;
+	
+	if (!g_file_test (iconpath, G_FILE_TEST_EXISTS))
+		return NULL;
 
 	scaled = gdk_pixbuf_new_from_file_at_scale(iconpath,
 						   16, 16,
