--- ../../examples/Demo/Makefile.old	2006-10-14 13:06:42.000000000 +0200
+++ ../../examples/Demo/Makefile	2006-10-14 13:08:32.000000000 +0200
@@ -2,9 +2,9 @@
 # Handwritten by me, made with Kate. So it is far from being perfect. :)
 
 CXX = g++
-CPPFLAGS = -I"../../include" -I"/usr/X11R6/include"
+CPPFLAGS = -I/usr/include/irrlicht
 CXXFLAGS = -O3 -ffast-math
-LDFLAGS = -L"/usr/X11R6/lib" -L"../../lib/Linux" -lIrrlicht -lGL -lGLU -lXxf86vm -lXext -lX11
+LDFLAGS = -lIrrlicht -lGL -lGLU -lXxf86vm -lXext -lX11 -lpng -ljpeg
 # if you enable sound add the proper library for linking
 # -laudiere
 # -lSDL_mixer -lSDL
