let portage handle the packaging details

--- Makefile.PL
+++ Makefile.PL
@@ -259,9 +259,7 @@
     my $self = shift;
 
     ## INSTALL_DIR = dir where data go
-    my $INSTALL_DIRS = '';
-    if (exists $ENV{XDG_DATA_DIRS}) { $INSTALL_DIRS = '$(XDG_DATA_DIRS)' }
-    else { $INSTALL_DIRS = '$(DESTDIR)$(PREFIX)/share' }
+    my $INSTALL_DIRS = '$(DESTDIR)$(PREFIX)/share';
 
     ###########################################
     ## MO files
@@ -270,10 +268,7 @@
     my $update_mo = "update_mo :\n";
     $update_mo .= "\tmake -C po update-mo\n";
     # install mo files
-    my $mopath = '';
-    if (exists $ENV{LOCALE_PREFIX}) { 
-	$mopath = "$ENV{LOCALE_PREFIX}/share/locale" 
-    } else { $mopath = '$(DESTDIR)$(PREFIX)/share/locale' }
+    my $mopath = '$(DESTDIR)$(PREFIX)/share/locale';
     my $mo_install = '';
     my $mo_uninstall = '';
     foreach my $po (<po/*.po>) {
@@ -327,8 +322,6 @@
     # update Scrollkeeper database
     my $pkglocalstatedir = `scrollkeeper-config --pkglocalstatedir`;
     chomp $pkglocalstatedir;
-    $doc_install .= "\tscrollkeeper-update -p \$(DESTDIR)$pkglocalstatedir -o \$(DESTDIR)$omfdir/gwine\n" unless $::deb_pkg;
-    $doc_uninstall .= "\tscrollkeeper-update -p \$(DESTDIR)$pkglocalstatedir\n" unless $::deb_pkg;
     # uninstall 
     foreach (@doc_install) {
 	next if m/^\s*install -d/;
@@ -368,27 +361,21 @@
     ## Install gwine.desktop
     my $desktop_install = "\tinstall -d $INSTALL_DIRS/applications/\n";
     $desktop_install .= "\tinstall -m 644 gwine.desktop $INSTALL_DIRS/applications/gwine.desktop\n";
-    $desktop_install .= "\tupdate-desktop-database\n" unless $::deb_pkg;
     my $desktop_uninstall = "\trm -f $INSTALL_DIRS/applications/gwine.desktop\n";
 
     ## Install gwine.xml
     my $mime_install = "\tinstall -d $INSTALL_DIRS/mime/packages/\n";
     $mime_install .= "\tinstall -m 644 gwine.xml $INSTALL_DIRS/mime/packages/gwine.xml\n";
-    $mime_install .= "\tupdate-mime-database $INSTALL_DIRS/mime\n" unless $::deb_pkg;
     my $mime_uninstall = "\trm -f $INSTALL_DIRS/mime/packages/gwine.xml\n";
 
     ###########################################
     ## Install gconf schemas
     my $gconf_install = '';
     my $gconf_uninstall = '';
-    if ($::install_gconf_schema) {
-	$gconf_install = "\tgconftool --install-schema-file=gwine.schemas\n"
-    } elsif ($::deb_pkg) {
 	# Place schema in /etc/gconf/schemas (it seems dh_gconf searches here)
 	$gconf_install = "\tinstall -d \$(DESTDIR)/etc/gconf/schemas\n";
 	$gconf_install .= "\tinstall -m 644 gwine.schemas \$(DESTDIR)/etc/gconf/schemas\n";
-	$gconf_uninstall = "\trm \$(DESTDIR)/etc/gconf/schemas\n"
+	$gconf_uninstall = "\trm \$(DESTDIR)/etc/gconf/schemas\n";
-    }
 
 <<EOT;
 $update_mo
