http://code.google.com/p/gmpy/source/detail?r=251

--- test3/gmpy_test_mpf.py
+++ test3/gmpy_test_mpf.py
@@ -419,6 +419,7 @@
 
     sav = sys.stdout
     class _Dummy:
+        encoding = None
         def write(self,*whatever):
             pass
     try:
--- test3/gmpy_test_mpq.py
+++ test3/gmpy_test_mpq.py
@@ -359,6 +359,7 @@
 
     sav = sys.stdout
     class _Dummy:
+        encoding = None
         def write(self,*whatever):
             pass
     try:
--- test3/gmpy_test_mpz.py
+++ test3/gmpy_test_mpz.py
@@ -967,6 +967,7 @@
 
     sav = sys.stdout
     class _Dummy:
+        encoding = None
         def write(self,*whatever):
             pass
     try:
