diff -urNp --exclude CVS --exclude BitKeeper x/kernel/ksyms.c x-new/kernel/ksyms.c
--- x/kernel/ksyms.c	2003-09-02 16:43:44.000000000 +0200
+++ x-new/kernel/ksyms.c	2003-09-02 17:06:03.000000000 +0200
@@ -625,6 +625,7 @@ EXPORT_SYMBOL(remove_bh);
 EXPORT_SYMBOL(tasklet_init);
 EXPORT_SYMBOL(tasklet_kill);
 EXPORT_SYMBOL(__run_task_queue);
+EXPORT_SYMBOL(open_softirq);
 EXPORT_SYMBOL(do_softirq);
 EXPORT_SYMBOL(raise_softirq);
 EXPORT_SYMBOL(cpu_raise_softirq);
