--- kdump/Makefile
+++ kdump/Makefile
@@ -24,7 +24,7 @@
 
 $(KDUMP): $(KDUMP_OBJS)
 	mkdir -p $(@D)
-	$(CC) $(CFLAGS) -o $@ $(KDUMP_OBJS)
+	$(CC) $(LDFLAGS) -o $@ $(KDUMP_OBJS)
 
 $(KDUMP_MANPAGE): kdump/kdump.8
 	$(MKDIR) -p     $(MANDIR)/man8
--- kexec/Makefile
+++ kexec/Makefile
@@ -50,7 +50,7 @@
 
 $(KEXEC): $(KEXEC_OBJS) $(UTIL_LIB)
 	mkdir -p $(@D)
-	$(CC) $(KCFLAGS) -o $@ $(KEXEC_OBJS) $(UTIL_LIB) $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $(KEXEC_OBJS) $(UTIL_LIB) $(LIBS)
 
 $(KEXEC_MANPAGE): kexec/kexec.8
 	 $(MKDIR) -p     $(MANDIR)/man8
