Index: pam_login-3.17/src/faillog.c
===================================================================
--- pam_login-3.17.orig/src/faillog.c
+++ pam_login-3.17/src/faillog.c
@@ -41,14 +41,20 @@
 #include <unistd.h>
 #include <getopt.h>
 #include <locale.h>
-#include <libintl.h>
 
-#include "faillog.h"
+#ifdef ENABLE_NLS
+#include <libintl.h>
 
 #ifndef _
 #define _(s) gettext(s)
 #endif
 
+#else
+#define _(s) s
+#endif
+
+#include "faillog.h"
+
 #define memzero(ptr, size) memset((void *)(ptr), 0, (size))
 
 static const char *Prog = "faillog";   /* program name */
@@ -92,9 +98,11 @@ int main (int argc, char **argv)
 	int c, anyflag = 0;
 	struct passwd *pwent;
 
+#ifdef ENABLE_NLS
 	setlocale (LC_ALL, "");
 	bindtextdomain (PACKAGE, LOCALEDIR);
 	textdomain (PACKAGE);
+#endif
 
 	/* try to open for read/write, if that fails - read only */
 
Index: pam_login-3.17/src/lastlog.c
===================================================================
--- pam_login-3.17.orig/src/lastlog.c
+++ pam_login-3.17/src/lastlog.c
@@ -41,12 +41,18 @@
 #include <stdlib.h>
 #include <getopt.h>
 #include <locale.h>
+
+#ifdef ENABLE_NLS
 #include <libintl.h>
 
 #ifndef _
 #define _(s) gettext(s)
 #endif
 
+#else
+#define _(s) s
+#endif
+
 #define DAY (24L*3600L)
 
 static FILE *lastlogfile;	/* lastlog file stream */
@@ -77,9 +83,11 @@ int main (int argc, char **argv)
 {
 	int c;
 
+#ifdef ENABLE_NLS
 	setlocale (LC_ALL, "");
 	bindtextdomain (PACKAGE, LOCALEDIR);
 	textdomain (PACKAGE);
+#endif
 
 	if ((lastlogfile = fopen (_PATH_LASTLOG, "r")) == (FILE *) 0) {
 		perror (_PATH_LASTLOG);
Index: pam_login-3.17/src/login.c
===================================================================
--- pam_login-3.17.orig/src/login.c
+++ pam_login-3.17/src/login.c
@@ -63,7 +63,7 @@
 #include <sys/sysmacros.h>
 #include <netdb.h>
 #include <locale.h>
-#include <libintl.h>
+
 #include <sys/sysmacros.h>
 #include <linux/major.h>
 #include <security/pam_appl.h>
@@ -92,10 +92,17 @@
 #include "setproctitle.h"
 #include "getdef.h"
 
+#ifdef ENABLE_NLS
+#include <libintl.h>
+
 #ifndef _
 #define _(s) gettext(s)
 #endif
 
+#else
+#define _(s) s
+#endif
+
 #ifdef WITH_SELINUX
 int selinux_enabled = -1;
 security_context_t user_context=NULL;
@@ -109,7 +116,7 @@ static inline void
 print_version (void)
 {
   fprintf (stdout, "login (%s) %s\n", PACKAGE, VERSION);
-  fprintf (stdout, gettext ("\
+  fprintf (stdout, _ ("\
 Copyright (C) 1980, 1987, 1988 The Regents of the University of California.\n\
 Copyright (C) 1999, 2000, 2001, 2002 SuSE GmbH Nuernberg, Germany.\n\
 This is free software; see the source for copying conditions.  There is NO\n\
@@ -534,9 +541,11 @@ main (int argc, char **argv)
 
   init_sighandler ();
 
+#ifdef ENABLE_NLS
   setlocale (LC_ALL, "");
   bindtextdomain ("pam_login", LOCALEDIR);
   textdomain ("pam_login");
+#endif
 
   setpriority (PRIO_PROCESS, 0, 0);
 #ifdef HAVE_QUOTA
