--- linux-2.6.10/net/sunrpc/svc.c.orig	2005-01-18 03:18:08.235362992 +0000
+++ linux-2.6.10/net/sunrpc/svc.c	2005-01-18 03:18:28.532277392 +0000
@@ -446,7 +446,7 @@ err_bad_auth:
 err_bad_prog:
 #ifdef RPC_PARANOIA
 	if (prog != 100227 || progp->pg_prog != 100003)
-		printk("svc: unknown program %d (me %d)\n", prog, progp->pg_prog);
+		printk("svc: unknown program %d (me %d)\n", prog, serv->sv_program->pg_prog);
 	/* else it is just a Solaris client seeing if ACLs are supported */
 #endif
 	serv->sv_stats->rpcbadfmt++;
