diff -NrU5 denemo-0.8.0.orig/Makefile.am denemo-0.8.0/Makefile.am
--- denemo-0.8.0.orig/Makefile.am	2008-11-13 22:41:49.000000000 +0100
+++ denemo-0.8.0/Makefile.am	2008-11-13 22:42:25.000000000 +0100
@@ -14,11 +14,11 @@
 	missing install-sh config.sub \
 	acinclude.m4 config.log \
 	depcomp
 
 install-data-hook:
-	cp -r actions  $(datadir)/${PACKAGE}/
-	chmod -R 755 $(datadir)/${PACKAGE}/actions
+	cp -r actions  $(DESTDIR)/$(datadir)/${PACKAGE}/
+	chmod -R 755 $(DESTDIR)/$(datadir)/${PACKAGE}/actions
 dist-hook: @PACKAGE@.spec
 	cp $(srcdir)/@PACKAGE@.spec $(distdir)
 
 ACLOCAL_AMFLAGS = -I m4
