diff -NurpP passenger-1.0.5.orig/Rakefile passenger-1.0.5/Rakefile
--- passenger-1.0.5.orig/Rakefile	2008-05-07 18:27:24.000000000 +0000
+++ passenger-1.0.5/Rakefile	2008-06-01 10:57:04.860048239 +0000
@@ -37,7 +37,7 @@ APR_FLAGS.nil? and raise "Could not find
 
 CXX = "g++"
 THREADING_FLAGS = "-D_REENTRANT"
-CXXFLAGS = "#{THREADING_FLAGS} -Wall -g -I/usr/local/include " << MULTI_ARCH_FLAGS
+CXXFLAGS = "#{ENV['CXXFLAGS']} #{THREADING_FLAGS} -Wall" << MULTI_ARCH_FLAGS
 LDFLAGS = ""
 
 
@@ -87,8 +87,7 @@ subdir 'ext/boost/src' do
 		# processes, sometimes pthread errors will occur. These errors are harmless
 		# and should be ignored. Defining NDEBUG guarantees that boost::thread() will
 		# not abort if such an error occured.
-		flags = "-O2 -fPIC -I../.. #{THREADING_FLAGS} -DNDEBUG #{MULTI_ARCH_FLAGS}"
-		compile_cxx "*.cpp", flags
+		compile_cxx "*.cpp", "#{CXXFLAGS} -fPIC -I../.. -DNDEBUG"
 		create_static_library "libboost_thread.a", "*.o"
 	end
 	
@@ -101,7 +100,7 @@ end
 ##### Apache module
 
 class APACHE2
-	CXXFLAGS = "-I.. -fPIC -g -DPASSENGER_DEBUG #{APR_FLAGS} #{APXS2_FLAGS} #{CXXFLAGS}"
+	CXXFLAGS = "-I.. -fPIC #{APR_FLAGS} #{APXS2_FLAGS} #{CXXFLAGS}"
 	OBJECTS = {
 		'Configuration.o' => %w(Configuration.cpp Configuration.h),
 		'Hooks.o' => %w(Hooks.cpp Hooks.h
diff -NurpP passenger-1.0.5.orig/bin/passenger-spawn-server passenger-1.0.5/bin/passenger-spawn-server
--- passenger-1.0.5.orig/bin/passenger-spawn-server	2008-04-03 15:49:06.000000000 +0000
+++ passenger-1.0.5/bin/passenger-spawn-server	2008-06-01 10:55:57.585311422 +0000
@@ -16,7 +16,6 @@
 #  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
 require 'pathname'
-$LOAD_PATH.unshift(File.expand_path("#{File.dirname(__FILE__)}/../lib"))
 require 'passenger/spawn_manager'
 require 'passenger/utils'
 
diff -NurpP passenger-1.0.5.orig/lib/passenger/platform_info.rb passenger-1.0.5/lib/passenger/platform_info.rb
--- passenger-1.0.5.orig/lib/passenger/platform_info.rb	2008-04-28 21:34:04.000000000 +0000
+++ passenger-1.0.5/lib/passenger/platform_info.rb	2008-06-01 10:58:27.772399440 +0000
@@ -115,7 +115,6 @@ private
 		else
 			flags = `#{APXS2} -q CFLAGS`.strip << " -I" << `#{APXS2} -q INCLUDEDIR`
 			flags.strip!
-			flags.gsub!(/-O\d? /, '')
 			return flags
 		end
 	end
@@ -151,7 +150,6 @@ private
 		else
 			flags = `#{APR_CONFIG} --cppflags --includes`.strip
 			libs = `#{APR_CONFIG} --link-ld`.strip
-			flags.gsub!(/-O\d? /, '')
 			return [flags, libs]
 		end
 	end
diff -NurpP passenger-1.0.5.orig/lib/passenger/request_handler.rb passenger-1.0.5/lib/passenger/request_handler.rb
--- passenger-1.0.5.orig/lib/passenger/request_handler.rb	2008-05-07 14:05:55.000000000 +0000
+++ passenger-1.0.5/lib/passenger/request_handler.rb	2008-06-01 10:55:57.585311422 +0000
@@ -102,7 +102,6 @@ class RequestHandler
 	NINJA_PATCHING_LOCK = Mutex.new
 	@@ninja_patched_action_controller = false
 	
-	File.read("#{File.dirname(__FILE__)}/../../Rakefile") =~ /^PACKAGE_VERSION = "(.*)"$/
 	PASSENGER_VERSION = $1
 	PASSENGER_HEADER = "Phusion Passenger (mod_rails) #{PASSENGER_VERSION}"
 	if File.exist?("#{File.dirname(__FILE__)}/../../enterprisey.txt")
diff -NurpP passenger-1.0.5.orig/lib/passenger/utils.rb passenger-1.0.5/lib/passenger/utils.rb
--- passenger-1.0.5.orig/lib/passenger/utils.rb	2008-05-07 18:26:59.000000000 +0000
+++ passenger-1.0.5/lib/passenger/utils.rb	2008-06-01 10:55:57.586311270 +0000
@@ -20,7 +20,7 @@ require 'etc'
 require 'thread'
 require 'fastthread'
 require 'timeout'
-require File.expand_path("#{File.dirname(__FILE__)}/../../ext/passenger/native_support.so")
+require 'native_support'
 
 module Passenger
 
