--- Rakefile.orig	2009-03-15 15:15:44.000000000 +0100
+++ Rakefile	2009-03-15 15:24:51.000000000 +0100
@@ -39,12 +39,8 @@
 CXX = "g++"
 LIBEXT = PlatformInfo.library_extension
 # _GLIBCPP__PTHREADS is for fixing Boost compilation on OpenBSD.
-if OPTIMIZE
-	OPTIMIZATION_FLAGS = "#{PlatformInfo.debugging_cflags} -O2 -DBOOST_DISABLE_ASSERTS"
-else
-	OPTIMIZATION_FLAGS = "#{PlatformInfo.debugging_cflags} -DPASSENGER_DEBUG -DBOOST_DISABLE_ASSERTS"
-end
-CXXFLAGS = "-Wall #{OPTIMIZATION_FLAGS}"
+OPTIMIZATION_FLAGS = "#{ENV['CXXFLAGS']} -DNDEBUG"
+CXXFLAGS = "#{OPTIMIZATION_FLAGS} -Wall"
 EXTRA_LDFLAGS = ""
 
 #### Default tasks
@@ -540,15 +536,15 @@
 task :fakeroot => [:apache2, :native_support, :doc] do
 	require 'rbconfig'
 	include Config
-	fakeroot = "pkg/fakeroot"
+	fakeroot = ENV['DISTDIR']
 
 	# We don't use CONFIG['archdir'] and the like because we want
 	# the files to be installed to /usr, and the Ruby interpreter
 	# on the packaging machine might be in /usr/local.
-	libdir = "#{fakeroot}/usr/lib/ruby/#{CONFIG['ruby_version']}"
+	libdir = "#{fakeroot}/#{CONFIG['sitedir']}/#{CONFIG['ruby_version']}"
 	extdir = "#{libdir}/#{CONFIG['arch']}"
 	bindir = "#{fakeroot}/usr/bin"
-	docdir = "#{fakeroot}/usr/share/doc/phusion_passenger"
+	docdir = "#{fakeroot}/usr/share/doc/passenger-#{PACKAGE_VERSION}"
 	libexecdir = "#{fakeroot}/usr/lib/phusion_passenger"
 	
 	sh "rm -rf #{fakeroot}"
@@ -567,7 +563,6 @@
 	sh "cp bin/* #{bindir}/"
 	
 	sh "mkdir -p #{libexecdir}"
-	sh "cp ext/apache2/mod_passenger.so #{libexecdir}/"
 	sh "mv #{fakeroot}/usr/bin/passenger-spawn-server #{libexecdir}/"
 	sh "cp ext/apache2/ApplicationPoolServerExecutable #{libexecdir}/"
 	
