--- riceplugin/Profiler.cpp.old	2005-02-05 12:38:52.721617392 -0800
+++ riceplugin/Profiler.cpp	2005-02-05 12:39:16.467082794 -0800
@@ -14,7 +14,7 @@
 
 
 
-HRESULT CUnique< CProfiler >::Create()
+template <> HRESULT CUnique< CProfiler >::Create()
 {
 	_ASSERTE(m_pInstance == NULL);
 	
--- riceplugin/DaedalusRender.h.old	2006-03-12 14:38:00.000000000 +0100
+++ riceplugin/DaedalusRender.h	2006-03-12 14:39:06.000000000 +0100
@@ -60,7 +60,7 @@
 	
 	inline bool IsTexel0Enable() {return m_pColorCombiner->m_bTex0Enabled;}
 	inline bool IsTexel1Enable() {return m_pColorCombiner->m_bTex1Enabled;}
-	inline bool CDaedalusRender::IsTextureEnabled() { return (m_pColorCombiner->m_bTex0Enabled||m_pColorCombiner->m_bTex1Enabled); }
+	inline bool IsTextureEnabled() { return (m_pColorCombiner->m_bTex0Enabled||m_pColorCombiner->m_bTex1Enabled); }
 
 	inline DaedalusRenderTexture& GetCurrentTexture() { return g_textures[gRSP.curTile]; }
 	inline DaedalusRenderTexture& GetTexture(DWORD dwTile) { return g_textures[dwTile]; }
