diff -r -u freetype-2.0.5.orig/include/freetype/config/ftoption.h freetype-2.0.5/include/freetype/config/ftoption.h
--- freetype-2.0.5.orig/include/freetype/config/ftoption.h	Tue Aug 21 10:50:45 2001
+++ freetype-2.0.5/include/freetype/config/ftoption.h	Thu Dec 13 18:15:16 2001
@@ -338,7 +338,7 @@
   /* By undefining this, you will only compile the code necessary to load  */
   /* TrueType glyphs without hinting.                                      */
   /*                                                                       */
-#undef  TT_CONFIG_OPTION_BYTECODE_INTERPRETER
+#define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
