--- Makefile.in.orig	2003-08-13 00:44:11.999413120 -0400
+++ Makefile.in	2003-08-13 00:44:57.784452736 -0400
@@ -74,8 +74,6 @@
         CHEATS.O cheats2.o data.o unix/unix.o unix/config.o GLOBALS.O \
 	$(SPC7110OBJ) $(OBC1OBJ) $(SETAOBJ) $(KREEDOBJ)
 
-OBJECTS += ../zlib/libz.a
-
 ifdef GUI
 GUIOBJS = unix/snes9x_gui.o unix/moc_snes9x_gui.o
 GUILIBS = -L$(QTDIR)/lib -lqt
@@ -132,7 +130,7 @@
 CC = @CC@
 NASM = @NASM@
 
-INCLUDES = -I../zlib @XINCLUDES@
+INCLUDES = @XINCLUDES@
 
 OPTIMISE = @OPTIMIZE@
 
@@ -170,7 +168,7 @@
 CFLAGS=$(CCFLAGS)
 
 .SUFFIXES: .o .cpp .c .cc .h .m .i .S .asm .obj .O .CPP .C .ASM
-LDLIBS = -L/usr/X11R6/lib -L../zlib
+LDLIBS = -L/usr/X11R6/lib
 
 ifdef GLIDE
 all: offsets gsnes9x
@@ -206,22 +204,19 @@
 	$(CCC) $(INCLUDES) -o $@ offsets.o
 	./offsets >i386/offsets.h
 
-../zlib/libz.a:
-	cd ../zlib && sh ./configure && make
-
-snes9x: $(OBJECTS) ../zlib/libz.a unix/x11.o $(AIDOOBJS) $(GUIOBJS)
+snes9x: $(OBJECTS)  unix/x11.o $(AIDOOBJS) $(GUIOBJS)
 	$(CCC) $(INCLUDES) -o $@ $(OBJECTS) $(AIDOOBJS) $(GLIDEOBJS) $(OPENGLOBJS) unix/x11.o $(GUIOBJS) $(LDLIBS) $(GLIDELIBS) $(OPENGLLIBS) $(GUILIBS) @SYSLIBS@ -lXext -lX11 $(EXTRALIBS) -lz -lm
 
-ssnes9x: $(OBJECTS) ../zlib/libz.a unix/svga.o 
+ssnes9x: $(OBJECTS)  unix/svga.o 
 	$(CCC) $(INCLUDES) -o $@ $(OBJECTS) $(GLIDEOBJS) unix/svga.o $(LDLIBS) $(GLIDELIBS) -lvga -lvgagl -lz $(EXTRALIBS) -lm
 
-gsnes9x: $(OBJECTS) ../zlib/libz.a $(GLIDEOBJS)
+gsnes9x: $(OBJECTS)  $(GLIDEOBJS)
 	$(CCC) $(INCLUDES) -o $@ $(OBJECTS) $(GLIDEOBJS) $(LDLIBS) -lglide -lz $(EXTRALIBS) -lm
 
-ggisnes9x: $(OBJECTS) ../zlib/libz.a unix/ggi.o
+ggisnes9x: $(OBJECTS)  unix/ggi.o
 	$(CCC) $(INCLUDES) -o $@ $(OBJECTS) unix/ggi.o $(LDLIBS) -lggi -lz $(EXTRALIBS) -lm
 
-osnes9x: $(OBJECTS) ../zlib/libz.a unix/x11.o $(OPENGLOBJS)
+osnes9x: $(OBJECTS)  unix/x11.o $(OPENGLOBJS)
 	$(CCC) $(INCLUDES) -o $@ $(OBJECTS) unix/x11.o $(OPENGLOBJS) $(LDLIBS) $(OPENGLLIBS) @SYSLIBS@ -lXext -lX11 $(EXTRALIBS) -lz -lm
 
 s9xserver: $(SERVER_OBJECTS)
