--- unix/unix.cpp.orig	2004-04-13 03:40:13.000000000 -0700
+++ unix/unix.cpp	2004-04-13 04:00:08.000000000 -0700
@@ -233,10 +233,12 @@
 
 void S9xParseArg (char **argv, int &i, int argc)
 {
-#ifdef JOYSTICK_SUPPORT
     if (strcmp (argv [i], "-j") == 0 ||
-	     strcasecmp (argv [i], "-nojoy") == 0)
+	     strcasecmp (argv [i], "-nojoy") == 0) {
 	Settings.JoystickEnabled = FALSE;
+	return;
+    }
+#ifdef JOYSTICK_SUPPORT
     else if (strcasecmp (argv [i], "-joydev1") == 0)
     {
 	if (i + 1 < argc)
