--- snes9x-1.43-src/snes9x/unix/x11.cpp.old	2005-02-14 13:18:59.000000000 +0100
+++ snes9x-1.43-src/snes9x/unix/x11.cpp	2005-02-14 13:20:10.000000000 +0100
@@ -1213,26 +1213,26 @@
         // re-enable all sound channels
         Define(TOGGLE_SOUND, 8,         Alt(XK_F12), Ctrl(XK_F12));
         // numbered quicksaves
-        Define(WRITE_SAVE_NUM, 0,       XK_F1);
-        Define(WRITE_SAVE_NUM, 1,       XK_F2);
-        Define(WRITE_SAVE_NUM, 2,       XK_F3);
-        Define(WRITE_SAVE_NUM, 3,       XK_F4);
-        Define(WRITE_SAVE_NUM, 4,       XK_F5);
-        Define(WRITE_SAVE_NUM, 5,       XK_F6);
-        Define(WRITE_SAVE_NUM, 6,       XK_F7);
-        Define(WRITE_SAVE_NUM, 7,       XK_F8);
-        Define(WRITE_SAVE_NUM, 8,       XK_F9);
-        Define(WRITE_SAVE_NUM, 9,       XK_F10);
-        Define(LOAD_SAVE_NUM, 0,       Shift(XK_F1));
-        Define(LOAD_SAVE_NUM, 1,       Shift(XK_F2));
-        Define(LOAD_SAVE_NUM, 2,       Shift(XK_F3));
-        Define(LOAD_SAVE_NUM, 3,       Shift(XK_F4));
-        Define(LOAD_SAVE_NUM, 4,       Shift(XK_F5));
-        Define(LOAD_SAVE_NUM, 5,       Shift(XK_F6));
-        Define(LOAD_SAVE_NUM, 6,       Shift(XK_F7));
-        Define(LOAD_SAVE_NUM, 7,       Shift(XK_F8));
-        Define(LOAD_SAVE_NUM, 8,       Shift(XK_F9));
-        Define(LOAD_SAVE_NUM, 9,       Shift(XK_F10));
+        Define(LOAD_SAVE_NUM, 0,       XK_F1);
+        Define(LOAD_SAVE_NUM, 1,       XK_F2);
+        Define(LOAD_SAVE_NUM, 2,       XK_F3);
+        Define(LOAD_SAVE_NUM, 3,       XK_F4);
+        Define(LOAD_SAVE_NUM, 4,       XK_F5);
+        Define(LOAD_SAVE_NUM, 5,       XK_F6);
+        Define(LOAD_SAVE_NUM, 6,       XK_F7);
+        Define(LOAD_SAVE_NUM, 7,       XK_F8);
+        Define(LOAD_SAVE_NUM, 8,       XK_F9);
+        Define(LOAD_SAVE_NUM, 9,       XK_F10);
+        Define(WRITE_SAVE_NUM, 0,       Shift(XK_F1));
+        Define(WRITE_SAVE_NUM, 1,       Shift(XK_F2));
+        Define(WRITE_SAVE_NUM, 2,       Shift(XK_F3));
+        Define(WRITE_SAVE_NUM, 3,       Shift(XK_F4));
+        Define(WRITE_SAVE_NUM, 4,       Shift(XK_F5));
+        Define(WRITE_SAVE_NUM, 5,       Shift(XK_F6));
+        Define(WRITE_SAVE_NUM, 6,       Shift(XK_F7));
+        Define(WRITE_SAVE_NUM, 7,       Shift(XK_F8));
+        Define(WRITE_SAVE_NUM, 8,       Shift(XK_F9));
+        Define(WRITE_SAVE_NUM, 9,       Shift(XK_F10));
         // layer toggles
         Define(TOGGLE_LAYER,  0,       XK_1);
         Define(TOGGLE_LAYER,  1,       XK_2);
