--- unix/unix.cpp.old	2009-06-05 08:52:45.000000000 +0200
+++ unix/unix.cpp	2009-06-05 08:54:31.000000000 +0200
@@ -1131,11 +1131,11 @@
 {
     *drive = 0;
 
-    char *slash = strrchr (path, '/');
+    const char *slash = strrchr (path, '/');
     if (!slash)
 	slash = strrchr (path, '\\');
 
-    char *dot = strrchr (path, '.');
+    const char *dot = strrchr (path, '.');
 
     if (dot && slash && dot < slash)
 	dot = NULL;
