Index: gpsim-0.21.11/cli/Makefile.am
===================================================================
--- gpsim-0.21.11.orig/cli/Makefile.am
+++ gpsim-0.21.11/cli/Makefile.am
@@ -46,6 +46,7 @@ libgpsimcli_la_SOURCES = \
 	cmd_version.h misc.h cmd_disasm.h  cmd_list.h  cmd_quit.h \
 	cmd_stimulus.h cmd_x.h symbol_t.h cmd_frequency.h cmd_icd.h \
 	cmd_macro.h ui_gpsim.h
+libgpsimcli_la_LIBADD = $(LIBREADLINE) ../eXdbm/libgpsim_eXdbm.la
 
 
 AM_YFLAGS = -dvt
Index: gpsim-0.21.11/gui/Makefile.am
===================================================================
--- gpsim-0.21.11.orig/gui/Makefile.am
+++ gpsim-0.21.11/gui/Makefile.am
@@ -17,6 +17,7 @@ libgpsimgui_la_SOURCES =  gui_break.cc g
 	preferences.cc preferences.h \
 	settings.h settings_exdbm.h
 
+libgpsimgui_la_LIBADD = ../eXdbm/libgpsim_eXdbm.la
 libgpsimgui_la_LDFLAGS = @X_LDFLAGS@ @Y_LDFLAGS@
 
 pixmapdir = $(datadir)/gpsim
Index: gpsim-0.21.11/gpsim/Makefile.am
===================================================================
--- gpsim-0.21.11.orig/gpsim/Makefile.am
+++ gpsim-0.21.11/gpsim/Makefile.am
@@ -9,6 +9,6 @@ bin_PROGRAMS = gpsim
 gpsim_SOURCES = main.cc
 
 gpsim_LDFLAGS =  $(shell gtk-config --cflags)
-gpsim_LDADD = ../src/libgpsim.la ../cli/libgpsimcli.la ../gui/libgpsimgui.la \
-  ../eXdbm/libgpsim_eXdbm.la @GTK@ @GDK@ @GLIB@ -lstdc++ -lpopt \
+gpsim_LDADD = ../src/libgpsim.la ../eXdbm/libgpsim_eXdbm.la ../cli/libgpsimcli.la ../gui/libgpsimgui.la \
+  @GTK@ @GDK@ @GLIB@ -lstdc++ -lpopt \
   @X_LDFLAGS@ @Y_LDFLAGS@ @LIBREADLINE@
Index: gpsim-0.21.11/Makefile.am
===================================================================
--- gpsim-0.21.11.orig/Makefile.am
+++ gpsim-0.21.11/Makefile.am
@@ -1,5 +1,5 @@
 
-SUBDIRS = cli xpms eXdbm gui src gpsim modules examples doc
+SUBDIRS = eXdbm cli xpms gui src gpsim modules examples doc
 
 dist-hook:
 	cp gpsim.spec $(distdir)     
