diff -NurB gpsim-0.20.14-orig/cli/input.cc gpsim-0.20.14/cli/input.cc
--- gpsim-0.20.14-orig/cli/input.cc	2003-10-21 01:32:05.000000000 -0500
+++ gpsim-0.20.14/cli/input.cc	2003-10-21 02:00:20.000000000 -0500
@@ -34,8 +34,8 @@
 extern void simulation_cleanup(void);
 extern int use_gui;
 
-//#define RALF
-#define SCOTT
+#define RALF
+//#define SCOTT
 
 //#if 0        //tsd removed in 0.20.4
 extern "C" {
