--- configure.ac.orig	2006-08-15 15:46:22.000000000 +0200
+++ configure.ac	2006-08-15 15:47:30.000000000 +0200
@@ -33,71 +33,21 @@
 	AC_DEFINE([HAVE_GUI],[],[True if GUI is being used])
 fi
 
-dnl --enable-gtk1 : link with GTK+ 1.x
-dnl    The default is to link with GTK+ 2.x
-
-AC_ARG_ENABLE(gtk1,
-     [  --enable-gtk1           Use GTK+ 1.x for the gui (deprecated)],
-     [case "${enableval}" in
-       yes) use_gtk1=yes ;;
-       no)  use_gtk1=no ;;
-       *) AC_MSG_ERROR(bad value ${enableval} for --enable-gtk1) ;;
-     esac],[use_gtk1=no])
-
-if test "$use_gtk1" = "no"; then
-        echo linking with GTK+ 2.x
-else
-        echo linking with GTK+ 1.x
-        [
-         use_gtk1=yes
-        ]
-fi
-
 GTK=
 GDK=
 GLIB=
 
+AC_PATH_PROG(PKGCONFIG, pkg-config, no, ${PATH})
+if test $PKGCONFIG = "no"; then
+  AC_MSG_ERROR(Cannot find pkg-config)
+fi
 if test "$use_gui" = "no"; then
-  dnl glib checks
-  AC_PATH_PROG(GLIBCONFIG, glib-config, no, ${PATH})
-
-  if test $GLIBCONFIG = "no"; then
-    AC_MSG_ERROR(Cannot find glib-config: glib is needed even for non-gui version.)
-  fi
-
-  X_LDFLAGS=`$GLIBCONFIG --libs`
-  X_CFLAGS=`$GLIBCONFIG --cflags`
+  dnl glib2 checks
+  X_LDFLAGS=`$PKGCONFIG --libs glib-2.0 gthread-2.0 gmodule-2.0`
+  X_CFLAGS=`$PKGCONFIG --cflags glib-2.0`
   Y_LDFLAGS=
   Y_CFLAGS=
 else
-  if test "$use_gtk1" = "yes"; then
-    dnl gtk checks
-    dnl note that the gtk12-config is the name of gtk-config on BSD
-    AC_PATH_PROG(GTKCONFIG, gtk-config gtk12-config, no, ${PATH})
-    dnl  AC_PATH_PROGS(GTKCONFIG, gtk-config gtk12-config, no, ${PATH})
-
-    if test $GTKCONFIG = "no"; then
-      AC_MSG_ERROR(Cannot find gtk-config)
-    fi
-
-    AC_PATH_PROG(GTKEXTRACONFIG, gtkextra-config, no, ${PATH})
-
-    if test $GTKEXTRACONFIG = "no"; then
-      AC_MSG_ERROR(Cannot find gtkextra-config)
-    fi
-
-    X_LDFLAGS=`$GTKCONFIG --libs gthread`
-    X_CFLAGS=`$GTKCONFIG --cflags`
-    Y_LDFLAGS=`$GTKEXTRACONFIG --libs`
-    Y_CFLAGS=`$GTKEXTRACONFIG --cflags`
-  else
-    dnl gtk2 checks
-    AC_PATH_PROG(PKGCONFIG, pkg-config, no, ${PATH})
-
-    if test $PKGCONFIG = "no"; then
-      AC_MSG_ERROR(Cannot find pkg-config)
-    fi
-
     $PKGCONFIG --exists gtkextra-2.0
     if test $? != 0; then
       AC_MSG_ERROR(Cannot find gtkextra-2.0 package)
@@ -110,8 +60,6 @@
     GTK_VERSION_T=`$PKGCONFIG --modversion gtk+-2.0`
     echo linking with gtk-$GTK_VERSION_T
     AC_DEFINE_UNQUOTED([GTK_VERSION],"$GTK_VERSION_T",[gtk version])
-  fi
-
 fi
 
 wi_LIB_READLINE
@@ -124,7 +72,6 @@
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
 AC_PROG_YACC
 AM_PROG_LEX
 AM_PROG_LIBTOOL
@@ -227,6 +174,5 @@
   C++ compiler:         $CXX $CPPFLAGS $CXXFLAGS
 
   gui:                  $use_gui
-  GTK+ 1.x              $use_gtk1
 
 ])
