diff -ur sb-readline-0.1.orig/sb-readline.lisp sb-readline-0.1/sb-readline.lisp
--- sb-readline-0.1.orig/sb-readline.lisp       2003-06-09 10:41:47.000000000 -0500
+++ sb-readline-0.1/sb-readline.lisp    2005-09-20 10:24:58.000000000 -0500
@@ -8,8 +8,8 @@
 (in-package :readline)

 (eval-when (:compile-toplevel :load-toplevel :execute)
-  (load-1-foreign "/lib/libncurses.so")
-  (load-1-foreign "/lib/libreadline.so"))
+  (load-shared-object "/lib/libncurses.so")
+  (load-shared-object "/lib/libreadline.so"))

 (define-alien-routine "add_history" void (command c-string))
 (define-alien-routine "readline" c-string (prompt c-string))
