--- xxdiff.pro.orig	2007-01-22 19:42:25.000000000 +0100
+++ xxdiff.pro	2007-01-22 19:42:40.000000000 +0100
@@ -35,9 +35,9 @@
 
 
 # To enable KDE support, uncomment the following lines:
-#QMAKE_CFLAGS += -DXX_KDE -I/usr/include/kde
-#QMAKE_CXXFLAGS += -DXX_KDE -I/usr/include/kde
-#QMAKE_LIBS += -lkdeui -lkio -lqt-mt
+QMAKE_CFLAGS += -DXX_KDE -I/usr/include/kde
+QMAKE_CXXFLAGS += -DXX_KDE -I/usr/include/kde
+QMAKE_LIBS += -lkdeui -lkio -lqt-mt
 
 
 # Enable debugging modes within xxdiff.
