--- ggz-gtk/ggzclient.c.orig	2004-12-26 17:57:46.578440888 -0500
+++ ggz-gtk/ggzclient.c	2004-12-26 17:58:35.865948048 -0500
@@ -67,7 +67,6 @@
 static GGZHookReturn ggz_entered_fail(GGZServerEvent id, void* event_data, void* user_data);
 static GGZHookReturn ggz_logout(GGZServerEvent id, void* event_data, void* user_data);
 static GGZHookReturn ggz_motd_loaded(GGZServerEvent id, void* event_data, void* user_data);
-static GGZHookReturn ggz_table_left(GGZServerEvent id, void* event_data, void* user_data);
 static GGZHookReturn ggz_server_error(GGZServerEvent id, void* event_data, void* user_data);
 static GGZHookReturn ggz_net_error(GGZServerEvent id, void* event_data, void* user_data);
 
