--- Makefile.PL.old	2006-10-18 23:23:43.000000000 +0000
+++ Makefile.PL	2006-10-18 23:24:16.000000000 +0000
@@ -449,7 +449,7 @@
         my $icuflags = parrot_config($base, $parrot_config, 'icu_shared');
         my $include_path = parrot_config($base, $parrot_config, 'prefix') . parrot_config($base, $parrot_config, 'inc');
         my $rpath_blib = parrot_config($base, $parrot_config, 'rpath_blib');
-        my $build_dir = parrot_config($base, $parrot_config, 'top_builddir', 'build_dir');
+        my $build_dir = $ENV{PARROT_PATH};
         my $is_shared = parrot_config($base, $parrot_config, 'parrot_is_shared');
         my $parrot_libdir = parrot_config($base, $parrot_config, 'lib_dir');
 
