--- Makefile.in.orig	Sun Apr  8 18:30:01 2001
+++ Makefile.in	Sun Apr  8 18:31:35 2001
@@ -44,34 +44,34 @@
 	cd doc ; make man
 
 install: 
-	$(INSTALL) -d $(scripts_dest) $(lib_dest) $(config_dest) $(doc_dest)
-	cd scripts ; $(INSTALL) -o root -m 655 $(SCRIPTS) $(scripts_dest)
+	$(INSTALL) -d $(DESTDIR)/$(scripts_dest) $(DESTDIR)/$(lib_dest) $(DESTDIR)/$(config_dest) $(DESTDIR)/$(doc_dest)
+	cd scripts ; $(INSTALL) -o root -m 655 $(SCRIPTS) $(DESTDIR)/$(scripts_dest)
 	cd extras ; make install
 	cd doc    ; make install
-	$(INSTALL_DATA) -o root yard_utils.pl yardconfig.pm            $(lib_dest)
+	$(INSTALL_DATA) -o root yard_utils.pl yardconfig.pm            $(DESTDIR)/$(lib_dest)
 	$(INSTALL_DATA) -o root Bootdisk_Contents \
-		Bootdisk_Contents.{minimal,sample,Pilotbackup}         $(lib_dest)
-	$(INSTALL_DATA) -o root Config.pl                              $(lib_dest)
+		Bootdisk_Contents.{minimal,sample,Pilotbackup}         $(DESTDIR)/$(lib_dest)
+	$(INSTALL_DATA) -o root Config.pl			       $(DESTDIR)/$(lib_dest)
 	#   Install create_replacements as a lib script.
 	#   We need access to it but it shouldn't be visible to the user.
-	cd scripts; $(INSTALL) -o root -m 655 create_replacements      $(lib_dest)
-	$(CP) --parents `cat Replacements/ALL_REPLACEMENT_FILES`       $(lib_dest)
+	cd scripts; $(INSTALL) -o root -m 655 create_replacements      $(DESTDIR)/$(lib_dest)
+	$(CP) --parents `cat Replacements/ALL_REPLACEMENT_FILES`       $(DESTDIR)/$(lib_dest)
 	@echo Done.  You should now type \"make customize\" to create
 	@echo customizable files, if you don\'t already have them.
 
 customize:
-	$(INSTALL) -d $(config_dest)/Replacements/etc
-	$(INSTALL) -d $(config_dest)/Replacements/root
+	$(INSTALL) -d $(DESTDIR)/$(config_dest)/Replacements/etc
+	$(INSTALL) -d $(DESTDIR)/$(config_dest)/Replacements/root
 	#    create_replacements puts files into config_dest
-	$(lib_dest)/create_replacements
+	$(DESTDIR)/$(lib_dest)/create_replacements
 	@echo !
 	@echo !    Now installing configuration files, which may already exist.
 	@echo !    Answer \'no\' if you want to preserve your
 	@echo !    existing configuration.
 	@echo !
-	$(CP) -ir $(lib_dest)/Replacements      $(config_dest)
-	$(CP) -ir $(lib_dest)/Bootdisk_Contents $(config_dest)
-	$(CP) -ir $(lib_dest)/Config.pl         $(config_dest)
+	$(CP) -ir $(DESTDIR)/$(lib_dest)/Replacements      $(DESTDIR)/$(config_dest)
+	$(CP) -ir $(DESTDIR)/$(lib_dest)/Bootdisk_Contents $(DESTDIR)/$(config_dest)
+	$(CP) -ir $(DESTDIR)/$(lib_dest)/Config.pl         $(DESTDIR)/$(config_dest)
 	
 test:
 	$(CP) -r tests/Bootdisk_Contents.errors1 Bootdisk_Contents
