fix from upstream for building with alsa disabled

Index: mediastreamer2/src/mssndcard.c
===================================================================
RCS file: /sources/linphone/linphone/mediastreamer2/src/mssndcard.c,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -p -r1.7 -r1.8
--- mediastreamer2/src/mssndcard.c	12 Oct 2007 13:31:54 -0000	1.7
+++ mediastreamer2/src/mssndcard.c	10 Dec 2007 17:12:40 -0000	1.8
@@ -152,3 +152,10 @@ void ms_snd_card_destroy(MSSndCard *obj)
 	if (obj->id!=NULL)	ms_free(obj->id);
 	ms_free(obj);
 }
+
+#ifndef __ALSA_ENABLED__
+MSSndCard * ms_alsa_card_new_custom(const char *pcmdev, const char *mixdev){
+	ms_warning("Alsa support not available in this build of mediastreamer2");
+	return NULL;
+}
+#endif
