--- linux/kernel/context.c	2002-10-28 21:14:53.000000000 -0600
+++ linux/kernel/context.c.new	2002-10-28 21:34:31.000000000 -0600
@@ -16,7 +16,7 @@
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/sched.h>
-#include <linux/sched_runqueue.h>
+/* #include <linux/sched_runqueue.h> */
 #include <linux/init.h>
 #include <linux/unistd.h>
 #include <linux/signal.h>
--- linux/include/linux/sched.h	2002-10-28 21:29:32.000000000 -0600
+++ linux/include/linux/sched.h.new	2002-10-28 21:59:50.000000000 -0600
@@ -464,7 +464,7 @@
 #define PF_MEMDIE	0x00001000	/* Killed for out-of-memory */
 #define PF_FREE_PAGES	0x00002000	/* per process page freeing */
 #define PF_NOIO		0x00004000	/* avoid generating further I/O */
-
+#define PF_FSTRANS	0x00080000	/* inside a filesystem transaction */
 #define PF_USEDFPU	0x00100000	/* task used FPU this quantum (SMP) */
 
 /*
