--- perllib/FVWM/Module/Tk.pm	2003-11-12 08:55:58.000000000 +0000
+++ perllib/FVWM/Module/Tk.pm	2003-11-12 04:13:25.000000000 +0000
@@ -51,6 +51,14 @@
 	$top->fileevent($self->{istream},
 		readable => sub {
 			unless ($self->processPacket($self->readPacket)) {
+				if ($self->{disconnected}) {
+					# Seems like something does not want to exit - force it.
+					# For example, a new Tk window is launched on ON_EXIT.
+					$top->destroy if defined $top && defined $top->{Configure};
+					$self->debug("Forced to exit to escape eventLoop, fix the module", 0);
+					exit 1;
+				}
+				$self->eventLoopFinished(@params);
 				$top->destroy;
 			} else {
 				$self->eventLoopPrepared(@params);
@@ -58,7 +66,6 @@
 		}
 	);
 	MainLoop;
-	$self->eventLoopFinished(@params);
 }
 
 sub showError ($$;$) {
@@ -73,7 +80,7 @@
 		-bitmap => 'error',
 		-default_button => 'Close',
 		-text => $error,
-		-buttons => ['Close', 'Close All Errors', 'Exit Module']
+		-buttons => ['Close', 'Close All Errors', 'Exit Module'],
 	);
 	my $btn = $dialog->Show;
 
@@ -90,7 +97,7 @@
 		-icon => 'info',
 		-type => 'ok',
 		-title => $title,
-		-message => $msg
+		-message => $msg,
 	);
 }
 
