diff -ur srecord-1.42.orig/lib/arglex.cc srecord-1.42/lib/arglex.cc
--- srecord-1.42.orig/lib/arglex.cc	2008-06-02 01:38:46.000000000 +0300
+++ srecord-1.42/lib/arglex.cc	2009-07-25 21:58:48.000000000 +0300
@@ -16,6 +16,7 @@
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 //
 
+#include <cstdio>
 #include <cctype>
 #include <cstring>
 #include <cstdlib>
diff -ur srecord-1.42.orig/lib/interval.cc srecord-1.42/lib/interval.cc
--- srecord-1.42.orig/lib/interval.cc	2008-06-02 01:38:46.000000000 +0300
+++ srecord-1.42/lib/interval.cc	2009-07-25 22:01:00.000000000 +0300
@@ -18,6 +18,8 @@
 //
 
 
+#include <cstdio>
+
 #include <lib/interval.h>
 #include <iostream>
 using namespace std;
diff -ur srecord-1.42.orig/lib/quit/normal.cc srecord-1.42/lib/quit/normal.cc
--- srecord-1.42.orig/lib/quit/normal.cc	2008-06-02 01:38:46.000000000 +0300
+++ srecord-1.42/lib/quit/normal.cc	2009-07-25 22:01:57.000000000 +0300
@@ -17,6 +17,7 @@
 //      <http://www.gnu.org/licenses/>.
 //
 
+#include <cstdio>
 #include <cstdarg>
 #include <cstdlib>
 #include <cstring>
diff -ur srecord-1.42.orig/lib/srec/input/file.cc srecord-1.42/lib/srec/input/file.cc
--- srecord-1.42.orig/lib/srec/input/file.cc	2008-06-02 01:38:46.000000000 +0300
+++ srecord-1.42/lib/srec/input/file.cc	2009-07-25 22:03:08.000000000 +0300
@@ -17,6 +17,7 @@
 //      <http://www.gnu.org/licenses/>.
 //
 
+#include <cstdio>
 
 #include <errno.h>
 #include <iostream>
diff -ur srecord-1.42.orig/lib/srec/input/generator/constant.cc srecord-1.42/lib/srec/input/generator/constant.cc
--- srecord-1.42.orig/lib/srec/input/generator/constant.cc	2008-06-02 01:38:46.000000000 +0300
+++ srecord-1.42/lib/srec/input/generator/constant.cc	2009-07-25 22:04:11.000000000 +0300
@@ -17,6 +17,8 @@
 //      <http://www.gnu.org/licenses/>.
 //
 
+#include <cstdio>
+
 #include <lib/srec/input/generator/constant.h>
 
 
diff -ur srecord-1.42.orig/lib/srec/output/file/asm.cc srecord-1.42/lib/srec/output/file/asm.cc
--- srecord-1.42.orig/lib/srec/output/file/asm.cc	2008-06-02 01:38:46.000000000 +0300
+++ srecord-1.42/lib/srec/output/file/asm.cc	2009-07-25 22:08:49.000000000 +0300
@@ -17,6 +17,7 @@
 //      <http://www.gnu.org/licenses/>.
 //
 
+#include <cstdio>
 #include <cstring>
 
 #include <lib/interval.h>
diff -ur srecord-1.42.orig/lib/srec/output/file/basic.cc srecord-1.42/lib/srec/output/file/basic.cc
--- srecord-1.42.orig/lib/srec/output/file/basic.cc	2008-06-02 01:38:46.000000000 +0300
+++ srecord-1.42/lib/srec/output/file/basic.cc	2009-07-25 22:14:36.000000000 +0300
@@ -17,6 +17,7 @@
 //      <http://www.gnu.org/licenses/>.
 //
 
+#include <cstdio>
 #include <cstring>
 
 #include <lib/interval.h>
diff -ur srecord-1.42.orig/lib/srec/output/file.cc srecord-1.42/lib/srec/output/file.cc
--- srecord-1.42.orig/lib/srec/output/file.cc	2008-06-02 01:38:46.000000000 +0300
+++ srecord-1.42/lib/srec/output/file.cc	2009-07-25 22:07:17.000000000 +0300
@@ -17,6 +17,7 @@
 //      <http://www.gnu.org/licenses/>.
 //
 
+#include <cstdio>
 
 #include <errno.h>
 #include <iostream>
diff -ur srecord-1.42.orig/lib/srec/output.cc srecord-1.42/lib/srec/output.cc
--- srecord-1.42.orig/lib/srec/output.cc	2008-06-02 01:38:46.000000000 +0300
+++ srecord-1.42/lib/srec/output.cc	2009-07-25 22:05:56.000000000 +0300
@@ -17,6 +17,7 @@
 //      <http://www.gnu.org/licenses/>.
 //
 
+#include <cstdio>
 #include <cerrno>
 #include <cstring>
 
