--- help-browser/Makefile.in.orig	Tue Mar  6 06:10:46 2001
+++ help-browser/Makefile.in	Tue Mar  6 06:10:53 2001
@@ -153,7 +153,7 @@
 
 INCLUDES =  	-I$(includedir)							-I$(top_srcdir)/intl -I$(top_builddir)/intl			$(GNOME_INCLUDEDIR)						-DGNOME_PREFIX=\""$(prefix)\"" 		-DGNOMELOCALEDIR=\""$(datadir)/locale\""		$(htmlcflags)
 
-@HELP_USE_GTKHTML_TRUE@htmllibs = $(GNOME_XML_LIB) @PIXBUF_LIBS@ 	$(GTK_HTML_PREFIX)/lib/libgtkhtml.a		$(GNOME_PRINT_PREFIX)/lib/libgnomeprint.a
+@HELP_USE_GTKHTML_TRUE@htmllibs = $(GNOME_XML_LIB) @PIXBUF_LIBS@ $(GTK_HTML_PREFIX)/lib/libgtkhtml.a $(GTK_HTML_PREFIX)/lib/libgnomecanvaspixbuf.a $(GTK_HTML_PREFIX)/lib/libglade.a $(GTK_HTML_PREFIX)/lib/libglade-gnome.a $(GTK_HTML_PREFIX)/lib/libgal.a $(GTK_HTML_PREFIX)/lib/libunicode.a -lgnomeprint
 @HELP_USE_GTKHTML_FALSE@htmllibs = $(GTKXMHTML_LIBS)
 
 LDADD =  	$(GNOMEUI_LIBS)			$(GNORBA_LIBS)			$(GNOME_LIBDIR)			$(ORBIT_LIBS)		        $(INTLLIBS)
