diff -ur gnome-core-1.4.2-orig/applets/desk-guide/deskguide_applet.c gnome-core-1.4.2/applets/desk-guide/deskguide_applet.c
--- gnome-core-1.4.2-orig/applets/desk-guide/deskguide_applet.c	2005-02-22 03:57:46.304302821 -0500
+++ gnome-core-1.4.2/applets/desk-guide/deskguide_applet.c	2005-02-22 03:59:51.348022944 -0500
@@ -343,7 +343,7 @@
 	return tmp_value ? item->tmp_value : item->value;
     }
   
-  g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to find config value for <%s>", path);
+  g_warning ("%s(): unable to find config value for <%s>", G_GNUC_PRETTY_FUNCTION, path);
   
   return NULL;
 }
@@ -470,10 +470,10 @@
       if (GTK_BIN (gp_container)->child)
 	gtk_widget_destroy (GTK_BIN (gp_container)->child);
       else
-	g_warning (G_GNUC_PRETTY_FUNCTION "(): missing container child");
+	g_warning ("%s(): missing container child", G_GNUC_PRETTY_FUNCTION);
     }
   else
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): missing container");
+    g_warning ("%s(): missing container", G_GNUC_PRETTY_FUNCTION );
 }
 
 static void
diff -ur gnome-core-1.4.2-orig/applets/desk-guide/gstc.c gnome-core-1.4.2/applets/desk-guide/gstc.c
--- gnome-core-1.4.2-orig/applets/desk-guide/gstc.c	2005-02-22 03:58:35.568768375 -0500
+++ gnome-core-1.4.2/applets/desk-guide/gstc.c	2005-02-22 03:53:21.793436099 -0500
@@ -172,7 +172,7 @@
     }
   if (ci >= 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't add known window %ld to %ld",
+      g_warning ("%s(): can't add known window %ld to %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild,
 		 GSTC_PARENT_XWINDOW (sparent));
       return;
@@ -207,7 +207,7 @@
   /* sanity checks */
   if (ci < 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't remove unknown window %ld",
+      g_warning ("%s(): can't remove unknown window %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild);
       return;
     }
@@ -249,13 +249,13 @@
   /* sanity checks */
   if (ci < 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't restack unknown window %ld",
+      g_warning ("%s(): can't restack unknown window %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild);
       return;
     }
   if (xbelow && bi < 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't restack window %ld on top of unknown window %ld",
+      g_warning ("%s(): can't restack window %ld on top of unknown window %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild,
 		 xbelow);
       return;
@@ -297,10 +297,10 @@
     {
     case CreateNotify:
       if (xevent->xcreatewindow.parent != xparent)
-	g_warning (G_GNUC_PRETTY_FUNCTION "(): now what is THIS? "
+	g_warning ("%s(): now what is THIS? "
 		   "i receive a SubstructureNotify XCreateWindowEvent "
 		   "for a *foreign* child (%ld)??? X is on drugs!",
-		   xevent->xcreatewindow.window);
+		   G_GNUC_PRETTY_FUNCTION, xevent->xcreatewindow.window);
       else
 	gstc_parent_add_child (sparent, xevent->xcreatewindow.window);
       break;
diff -ur gnome-core-1.4.2-orig/applets/desk-guide/gwmh.c gnome-core-1.4.2/applets/desk-guide/gwmh.c
--- gnome-core-1.4.2-orig/applets/desk-guide/gwmh.c	2005-02-22 03:58:35.570768434 -0500
+++ gnome-core-1.4.2/applets/desk-guide/gwmh.c	2005-02-22 03:53:21.797436218 -0500
@@ -230,7 +230,7 @@
       /* setup the root window event monitor */
       window = gdk_window_ref_from_xid (GDK_ROOT_WINDOW ());
       if (!window)
-	g_error (G_GNUC_PRETTY_FUNCTION "(): root window id %ld invalid? bad bad...",
+	g_error ("%s(): root window id %ld invalid? bad bad...", G_GNUC_PRETTY_FUNCTION,
 		 GDK_ROOT_WINDOW ());
       gdk_window_add_filter (window,
 			     root_event_monitor,
@@ -390,13 +390,13 @@
     }
   if (!abort && bytes_after_return)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): Eeek, property has more than %u bytes, stored on harddisk?",
+      g_warning ("%s(): Eeek, property has more than %u bytes, stored on harddisk?", G_GNUC_PRETTY_FUNCTION,
 		 prop_buffer_lengh);
       abort++;
     }
   if (!abort && expected_format && expected_format != format_returned)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): Expected format (%u) unmatched (%d), programmer was drunk?",
+      g_warning ("%s(): Expected format (%u) unmatched (%d), programmer was drunk?", G_GNUC_PRETTY_FUNCTION,
 		 expected_format, format_returned);
       abort++;
     }
@@ -1586,7 +1586,7 @@
   g_return_if_fail (id > 0);
 
   if (!g_hook_destroy (&gwmh_desk_hook_list, id))
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier (%d)",
+    g_warning ("%s(): unable to remove notifier (%d)", G_GNUC_PRETTY_FUNCTION,
 	       id);
 }
 
@@ -1596,7 +1596,7 @@
   g_return_if_fail (id > 0);
 
   if (!g_hook_destroy (&gwmh_task_hook_list, id))
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier (%d)",
+    g_warning ("%s(): unable to remove notifier (%d)", G_GNUC_PRETTY_FUNCTION,
 	       id);
 }
 
@@ -1624,7 +1624,7 @@
   if (hook)
     g_hook_destroy_link (&gwmh_desk_hook_list, hook);
   else
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier <%p> (%p)",
+    g_warning ("%s(): unable to remove notifier <%p> (%p)",G_GNUC_PRETTY_FUNCTION,
 	       func, func_data);
 }
 
@@ -1643,7 +1643,7 @@
   if (hook)
     g_hook_destroy_link (&gwmh_task_hook_list, hook);
   else
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier <%p> (%p)",
+    g_warning ("%s(): unable to remove notifier <%p> (%p)", G_GNUC_PRETTY_FUNCTION,
 	       func, func_data);
 }
 
diff -ur gnome-core-1.4.2-orig/applets/tasklist/gstc.c gnome-core-1.4.2/applets/tasklist/gstc.c
--- gnome-core-1.4.2-orig/applets/tasklist/gstc.c	2005-02-22 03:57:46.401305707 -0500
+++ gnome-core-1.4.2/applets/tasklist/gstc.c	2005-02-22 04:01:57.858787533 -0500
@@ -172,7 +172,7 @@
     }
   if (ci >= 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't add known window %ld to %ld",
+      g_warning ("%s(): can't add known window %ld to %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild,
 		 GSTC_PARENT_XWINDOW (sparent));
       return;
@@ -207,7 +207,7 @@
   /* sanity checks */
   if (ci < 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't remove unknown window %ld",
+      g_warning ("%s(): can't remove unknown window %ld", G_GNUC_PRETTY_FUNCTION ,
 		 xchild);
       return;
     }
@@ -249,13 +249,13 @@
   /* sanity checks */
   if (ci < 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't restack unknown window %ld",
+      g_warning ("%s(): can't restack unknown window %ld", G_GNUC_PRETTY_FUNCTION ,
 		 xchild);
       return;
     }
   if (xbelow && bi < 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't restack window %ld on top of unknown window %ld",
+      g_warning ("%s(): can't restack window %ld on top of unknown window %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild,
 		 xbelow);
       return;
@@ -297,10 +297,10 @@
     {
     case CreateNotify:
       if (xevent->xcreatewindow.parent != xparent)
-	g_warning (G_GNUC_PRETTY_FUNCTION "(): now what is THIS? "
+	g_warning ("%s(): now what is THIS? "
 		   "i receive a SubstructureNotify XCreateWindowEvent "
 		   "for a *foreign* child (%ld)??? X is on drugs!",
-		   xevent->xcreatewindow.window);
+		   G_GNUC_PRETTY_FUNCTION, xevent->xcreatewindow.window);
       else
 	gstc_parent_add_child (sparent, xevent->xcreatewindow.window);
       break;
diff -ur gnome-core-1.4.2-orig/applets/tasklist/gwmh.c gnome-core-1.4.2/applets/tasklist/gwmh.c
--- gnome-core-1.4.2-orig/applets/tasklist/gwmh.c	2005-02-22 03:57:46.402305736 -0500
+++ gnome-core-1.4.2/applets/tasklist/gwmh.c	2005-02-22 04:03:17.760165559 -0500
@@ -230,7 +230,7 @@
       /* setup the root window event monitor */
       window = gdk_window_ref_from_xid (GDK_ROOT_WINDOW ());
       if (!window)
-	g_error (G_GNUC_PRETTY_FUNCTION "(): root window id %ld invalid? bad bad...",
+	g_error ("%s(): root window id %ld invalid? bad bad...", G_GNUC_PRETTY_FUNCTION,
 		 GDK_ROOT_WINDOW ());
       gdk_window_add_filter (window,
 			     root_event_monitor,
@@ -390,13 +390,13 @@
     }
   if (!abort && bytes_after_return)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): Eeek, property has more than %u bytes, stored on harddisk?",
+      g_warning ("%s(): Eeek, property has more than %u bytes, stored on harddisk?", G_GNUC_PRETTY_FUNCTION,
 		 prop_buffer_lengh);
       abort++;
     }
   if (!abort && expected_format && expected_format != format_returned)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): Expected format (%u) unmatched (%d), programmer was drunk?",
+      g_warning ("%s(): Expected format (%u) unmatched (%d), programmer was drunk?", G_GNUC_PRETTY_FUNCTION,
 		 expected_format, format_returned);
       abort++;
     }
@@ -1586,7 +1586,7 @@
   g_return_if_fail (id > 0);
 
   if (!g_hook_destroy (&gwmh_desk_hook_list, id))
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier (%d)",
+    g_warning ("%s(): unable to remove notifier (%d)", G_GNUC_PRETTY_FUNCTION,
 	       id);
 }
 
@@ -1596,7 +1596,7 @@
   g_return_if_fail (id > 0);
 
   if (!g_hook_destroy (&gwmh_task_hook_list, id))
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier (%d)",
+    g_warning ("%s(): unable to remove notifier (%d)", G_GNUC_PRETTY_FUNCTION, 
 	       id);
 }
 
@@ -1624,7 +1624,7 @@
   if (hook)
     g_hook_destroy_link (&gwmh_desk_hook_list, hook);
   else
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier <%p> (%p)",
+    g_warning ("%s(): unable to remove notifier <%p> (%p)", G_GNUC_PRETTY_FUNCTION,
 	       func, func_data);
 }
 
@@ -1643,7 +1643,7 @@
   if (hook)
     g_hook_destroy_link (&gwmh_task_hook_list, hook);
   else
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier <%p> (%p)",
+    g_warning ("%s(): unable to remove notifier <%p> (%p)", G_GNUC_PRETTY_FUNCTION,
 	       func, func_data);
 }
 
diff -ur gnome-core-1.4.2-orig/applets/tasklist/tasklist_icon.c gnome-core-1.4.2/applets/tasklist/tasklist_icon.c
--- gnome-core-1.4.2-orig/applets/tasklist/tasklist_icon.c	2005-02-22 03:57:46.401305707 -0500
+++ gnome-core-1.4.2/applets/tasklist/tasklist_icon.c	2005-02-22 04:04:04.824568729 -0500
@@ -81,16 +81,16 @@
 	if (!abort &&
 	    requested_type != AnyPropertyType &&
 	    requested_type != type_returned) {
-		g_warning (G_GNUC_PRETTY_FUNCTION "(): Property has wrong type, probably on crack");
+		g_warning ("%s(): Property has wrong type, probably on crack", G_GNUC_PRETTY_FUNCTION );
 		abort++;
 	}
 	if (!abort && bytes_after_return) {
-			g_warning (G_GNUC_PRETTY_FUNCTION "(): Eeek, property has more than %u bytes, stored on harddisk?",
+			g_warning ("%s(): Eeek, property has more than %u bytes, stored on harddisk?", G_GNUC_PRETTY_FUNCTION,
 				   prop_buffer_lengh);
 			abort++;
 	}
 	if (!abort && expected_format && expected_format != format_returned) {
-		g_warning (G_GNUC_PRETTY_FUNCTION "(): Expected format (%u) unmatched (%d), programmer was drunk?",
+		g_warning ("%s(): Expected format (%u) unmatched (%d), programmer was drunk?", G_GNUC_PRETTY_FUNCTION,
 			   expected_format, format_returned);
 		abort++;
 	}
diff -ur gnome-core-1.4.2-orig/panel/gstc.c gnome-core-1.4.2/panel/gstc.c
--- gnome-core-1.4.2-orig/panel/gstc.c	2005-02-22 03:58:35.571768464 -0500
+++ gnome-core-1.4.2/panel/gstc.c	2005-02-22 03:53:21.799436277 -0500
@@ -172,7 +172,7 @@
     }
   if (ci >= 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't add known window %ld to %ld",
+      g_warning ("%s(): can't add known window %ld to %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild,
 		 GSTC_PARENT_XWINDOW (sparent));
       return;
@@ -207,7 +207,7 @@
   /* sanity checks */
   if (ci < 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't remove unknown window %ld",
+      g_warning ("%s(): can't remove unknown window %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild);
       return;
     }
@@ -249,13 +249,13 @@
   /* sanity checks */
   if (ci < 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't restack unknown window %ld",
+      g_warning ("%s(): can't restack unknown window %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild);
       return;
     }
   if (xbelow && bi < 0)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): can't restack window %ld on top of unknown window %ld",
+      g_warning ("%s(): can't restack window %ld on top of unknown window %ld", G_GNUC_PRETTY_FUNCTION,
 		 xchild,
 		 xbelow);
       return;
@@ -297,10 +297,10 @@
     {
     case CreateNotify:
       if (xevent->xcreatewindow.parent != xparent)
-	g_warning (G_GNUC_PRETTY_FUNCTION "(): now what is THIS? "
+	g_warning ("%s(): now what is THIS? "
 		   "i receive a SubstructureNotify XCreateWindowEvent "
 		   "for a *foreign* child (%ld)??? X is on drugs!",
-		   xevent->xcreatewindow.window);
+		   G_GNUC_PRETTY_FUNCTION, xevent->xcreatewindow.window);
       else
 	gstc_parent_add_child (sparent, xevent->xcreatewindow.window);
       break;
diff -ur gnome-core-1.4.2-orig/panel/gwmh.c gnome-core-1.4.2/panel/gwmh.c
--- gnome-core-1.4.2-orig/panel/gwmh.c	2005-02-22 03:58:35.573768524 -0500
+++ gnome-core-1.4.2/panel/gwmh.c	2005-02-22 03:53:21.803436396 -0500
@@ -230,7 +230,7 @@
       /* setup the root window event monitor */
       window = gdk_window_ref_from_xid (GDK_ROOT_WINDOW ());
       if (!window)
-	g_error (G_GNUC_PRETTY_FUNCTION "(): root window id %ld invalid? bad bad...",
+	g_error ("%s(): root window id %ld invalid? bad bad...", G_GNUC_PRETTY_FUNCTION,
 		 GDK_ROOT_WINDOW ());
       gdk_window_add_filter (window,
 			     root_event_monitor,
@@ -390,13 +390,13 @@
     }
   if (!abort && bytes_after_return)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): Eeek, property has more than %u bytes, stored on harddisk?",
+      g_warning ("%s(): Eeek, property has more than %u bytes, stored on harddisk?", G_GNUC_PRETTY_FUNCTION,
 		 prop_buffer_lengh);
       abort++;
     }
   if (!abort && expected_format && expected_format != format_returned)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): Expected format (%u) unmatched (%d), programmer was drunk?",
+      g_warning ("%s(): Expected format (%u) unmatched (%d), programmer was drunk?", G_GNUC_PRETTY_FUNCTION,
 		 expected_format, format_returned);
       abort++;
     }
@@ -1586,7 +1586,7 @@
   g_return_if_fail (id > 0);
 
   if (!g_hook_destroy (&gwmh_desk_hook_list, id))
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier (%d)",
+    g_warning ("%s(): unable to remove notifier (%d)", G_GNUC_PRETTY_FUNCTION,
 	       id);
 }
 
@@ -1596,7 +1596,7 @@
   g_return_if_fail (id > 0);
 
   if (!g_hook_destroy (&gwmh_task_hook_list, id))
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier (%d)",
+    g_warning ("%s(): unable to remove notifier (%d)", G_GNUC_PRETTY_FUNCTION,
 	       id);
 }
 
@@ -1624,7 +1624,7 @@
   if (hook)
     g_hook_destroy_link (&gwmh_desk_hook_list, hook);
   else
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier <%p> (%p)",
+    g_warning ("%s(): unable to remove notifier <%p> (%p)",G_GNUC_PRETTY_FUNCTION,
 	       func, func_data);
 }
 
@@ -1643,7 +1643,7 @@
   if (hook)
     g_hook_destroy_link (&gwmh_task_hook_list, hook);
   else
-    g_warning (G_GNUC_PRETTY_FUNCTION "(): unable to remove notifier <%p> (%p)",
+    g_warning ("%s(): unable to remove notifier <%p> (%p)", G_GNUC_PRETTY_FUNCTION,
 	       func, func_data);
 }
 
diff -ur gnome-core-1.4.2-orig/panel/xstuff.c gnome-core-1.4.2/panel/xstuff.c
--- gnome-core-1.4.2-orig/panel/xstuff.c	2005-02-22 03:58:35.575768583 -0500
+++ gnome-core-1.4.2/panel/xstuff.c	2005-02-22 03:53:21.805436456 -0500
@@ -322,19 +322,19 @@
       requested_type != AnyPropertyType &&
       requested_type != type_returned)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): Property has wrong type, probably on crack");
+      g_warning ("%s(): Property has wrong type, probably on crack", G_GNUC_PRETTY_FUNCTION);
       abort++;
     }
   if (!abort && bytes_after_return)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): Eeek, property has more than %u bytes, stored on harddisk?",
+      g_warning ("%s(): Eeek, property has more than %u bytes, stored on harddisk?", G_GNUC_PRETTY_FUNCTION,
 		 prop_buffer_lengh);
       abort++;
     }
   if (!abort && expected_format && expected_format != format_returned)
     {
-      g_warning (G_GNUC_PRETTY_FUNCTION "(): Expected format (%u) unmatched (%d), programmer was drunk?",
-		 expected_format, format_returned);
+      g_warning ("%s(): Expected format (%u) unmatched (%d), programmer was drunk?",
+		 G_GNUC_PRETTY_FUNCTION, expected_format, format_returned);
       abort++;
     }
   if (!abort && prop_data && nitems_return && format_returned)
