diff -rc5 gnome-core-1.4.0.6.orig/help-browser/help/es/Makefile.am gnome-core-1.4.0.6/help-browser/help/es/Makefile.am
*** gnome-core-1.4.0.6.orig/help-browser/help/es/Makefile.am	Fri Oct 26 11:13:30 2001
--- gnome-core-1.4.0.6/help-browser/help/es/Makefile.am	Sun Feb 24 19:33:33 2002
***************
*** 2,12 ****
  
  EXTRA_DIST = gnome-help-browser-help.html \
  	     gnome-help-browser-topic.dat \
  	     default-page.html
  
! help_browser_helpdir = $(datadir)/gnome/help/help-browser/C
  install-data-local:
  	$(mkinstalldirs) $(DESTDIR)$(help_browser_helpdir)
  	$(INSTALL_DATA)  $(srcdir)/gnome-help-browser-help.html \
  			 $(DESTDIR)$(help_browser_helpdir)/help-browser.html
  	$(INSTALL_DATA)  $(srcdir)/gnome-help-browser-topic.dat \
--- 2,12 ----
  
  EXTRA_DIST = gnome-help-browser-help.html \
  	     gnome-help-browser-topic.dat \
  	     default-page.html
  
! help_browser_helpdir = $(datadir)/gnome/help/help-browser/es
  install-data-local:
  	$(mkinstalldirs) $(DESTDIR)$(help_browser_helpdir)
  	$(INSTALL_DATA)  $(srcdir)/gnome-help-browser-help.html \
  			 $(DESTDIR)$(help_browser_helpdir)/help-browser.html
  	$(INSTALL_DATA)  $(srcdir)/gnome-help-browser-topic.dat \
