diff -ur clsql-2.9.2.orig/clsql-mysql.asd clsql-2.9.2/clsql-mysql.asd
--- clsql-2.9.2.orig/clsql-mysql.asd	2004-04-20 22:31:27.000000000 -0500
+++ clsql-2.9.2/clsql-mysql.asd	2004-04-21 18:21:08.239569448 -0500
@@ -86,8 +86,7 @@
   ((:module :db-mysql
 	    :components
 	    ((:file "mysql-package")
-	     (:clsql-mysql-source-file "mysql" :depends-on ("mysql-package"))
-	     (:file "mysql-loader" :depends-on ("mysql-package" "mysql"))
+	     (:file "mysql-loader" :depends-on ("mysql-package"))
 	     (:file "mysql-client-info" :depends-on ("mysql-loader"))
 	     (:file "mysql-api" :depends-on ("mysql-client-info"))
 	     (:file "mysql-sql" :depends-on ("mysql-api"))))))
Only in clsql-2.9.2: clsql-mysql.asd~
diff -ur clsql-2.9.2.orig/clsql-uffi.asd clsql-2.9.2/clsql-uffi.asd
--- clsql-2.9.2.orig/clsql-uffi.asd	2004-04-20 22:31:27.000000000 -0500
+++ clsql-2.9.2/clsql-uffi.asd	2004-04-21 18:21:23.809202504 -0500
@@ -91,6 +91,5 @@
   ((:module :uffi
 	    :components
 	    ((:file "clsql-uffi-package")
-	     (:clsql-uffi-source-file "uffi" :depends-on ("clsql-uffi-package"))
-	     (:file "clsql-uffi-loader" :depends-on ("clsql-uffi-package" "uffi"))
+	     (:file "clsql-uffi-loader" :depends-on ("clsql-uffi-package"))
 	     (:file "clsql-uffi" :depends-on ("clsql-uffi-loader"))))))
Only in clsql-2.9.2: clsql-uffi.asd~
