diff -ur clsql-3.0.8.orig/clsql-mysql.asd clsql-3.0.8/clsql-mysql.asd
--- clsql-3.0.8.orig/clsql-mysql.asd	2004-10-01 12:02:22.000000000 -0500
+++ clsql-3.0.8/clsql-mysql.asd	2004-10-13 10:58:41.159182560 -0500
@@ -85,8 +85,8 @@
   ((:module :db-mysql
 	    :components
 	    ((:file "mysql-package")
-	     (:clsql-mysql-source-file "mysql" :depends-on ("mysql-package"))
-	     (:file "mysql-loader" :depends-on ("mysql-package" "mysql"))
+	     #+nil (:clsql-mysql-source-file "mysql" :depends-on ("mysql-package"))
+	     (:file "mysql-loader" :depends-on ("mysql-package" #+nil "mysql"))
 	     (:file "mysql-client-info" :depends-on ("mysql-loader"))
 	     (:file "mysql-api" :depends-on ("mysql-client-info"))
 	     (:file "mysql-sql" :depends-on ("mysql-api"))
Only in clsql-3.0.8: clsql-mysql.asd~
diff -ur clsql-3.0.8.orig/clsql-uffi.asd clsql-3.0.8/clsql-uffi.asd
--- clsql-3.0.8.orig/clsql-uffi.asd	2004-10-01 12:02:22.000000000 -0500
+++ clsql-3.0.8/clsql-uffi.asd	2004-10-13 10:58:14.357257072 -0500
@@ -90,6 +90,6 @@
   ((:module :uffi
 	    :components
 	    ((:file "clsql-uffi-package")
-	     (:clsql-uffi-source-file "uffi" :depends-on ("clsql-uffi-package"))
-	     (:file "clsql-uffi-loader" :depends-on ("clsql-uffi-package" "uffi"))
+	     #+nil (:clsql-uffi-source-file "uffi" :depends-on ("clsql-uffi-package"))
+	     (:file "clsql-uffi-loader" :depends-on ("clsql-uffi-package" #+nil "uffi"))
 	     (:file "clsql-uffi" :depends-on ("clsql-uffi-loader"))))))
Only in clsql-3.0.8: clsql-uffi.asd~
Only in clsql-3.0.8/uffi: uffi.so
