--- kdebluetooth/kbluelock/kbluelock.cpp	2007-09-17 18:52:33.000000000 +0200
+++ kdebluetooth/kbluelock/kbluelock.cpp	2008-06-11 22:57:51.000000000 +0200
@@ -147 +147,18 @@
-	if (adapter->getRemoteMajorClass(addr) == "phone") {
+	if (
+		adapter->getRemoteMajorClass(addr) == "phone" ||
+		adapter->getRemoteMajorClass(addr) == "wearable" ||
+		adapter->getRemoteMajorClass(addr) == "computer" && (
+			adapter->getRemoteMinorClass(addr) == "palm" ||
+			adapter->getRemoteMinorClass(addr) == "handheld" ||
+			adapter->getRemoteMinorClass(addr) == "wearable"
+		) ||
+		adapter->getRemoteMajorClass(addr) == "imaging" && (
+			adapter->getRemoteMinorClass(addr) == "camera"
+		) ||
+		adapter->getRemoteMajorClass(addr) == "audio_video" && (
+			adapter->getRemoteMinorClass(addr) == "headset" ||
+			adapter->getRemoteMinorClass(addr) == "video camera" ||
+			adapter->getRemoteMinorClass(addr) == "camcorder"
+		)
+	)
+	{
