Index: live-0.1.0/Makefile
===================================================================
--- live-0.1.0.orig/Makefile
+++ live-0.1.0/Makefile
@@ -26,7 +26,7 @@ LDFLAGS  ?= -fPIC -g
 ECPPC    ?= ecppc
 CXXFLAGS += `tntnet-config --cxxflags`
 
-LDFLAGS  += `tntnet-config --libs`
+LIBS  += $(shell tntnet-config --libs)
 
 ### The directory environment:
 
@@ -71,7 +71,7 @@ WEBLIBS    = pages/libpages.a css/libcss
 
 .PHONY: all dist clean SUBDIRS
 
-all: SUBDIRS libvdr-$(PLUGIN).so
+all: libvdr-$(PLUGIN).so
 
 ### Implicit rules:
 
@@ -94,7 +94,7 @@ SUBDIRS:
 		make -C $$dir CXX="$(CXX)" CXXFLAGS="$(CXXFLAGS)" || exit 1; \
 	done
 
-libvdr-$(PLUGIN).so: $(PLUGINOBJS) $(LIBS) $(WEBLIBS)
+libvdr-$(PLUGIN).so: SUBDIRS $(PLUGINOBJS)
 	$(CXX) $(LDFLAGS) -shared -o $@  $(PLUGINOBJS) -Wl,--whole-archive $(WEBLIBS) -Wl,--no-whole-archive $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
