--- phlib-1.20/Makefile.in~	Mon Oct 20 21:08:46 2003
+++ phlib-1.20/Makefile.in	Fri Dec  5 18:02:38 2003
@@ -108,7 +108,9 @@
 $(OURLIBSO): $(OBJS) $(HEAD)
 	@echo Creating shared library $(OURLIBSO)
 	@rm -f $@
-	$(SHLD) -o $@ $(OBJS)
+	$(CC) -shared -o$@.$(VER_MAJOR).$(VER_MINOR) -Wl,-soname=$@.$(VER_MAJOR) $(OBJS)
+	ln -sf $@.$(VER_MAJOR).$(VER_MINOR) $@.$(VER_MAJOR)
+	ln -sf $@.$(VER_MAJOR).$(VER_MINOR) $@
 
 $(OURLIBDB): $(OBJS) $(MDBOBJS) $(HEAD)
 	@echo Creating library $@
@@ -118,15 +120,17 @@
 $(OURLIBDBSO): $(OBJS) $(MDBOBJS) $(HEAD)
 	@echo Creating shared library $@
 	@rm -f $@
-	$(SHLD) -o $@ $(OBJS) $(MDBOBJS)
+	$(CC) -shared -o$@.$(VER_MAJOR).$(VER_MINOR) -Wl,-soname=$@.$(VER_MAJOR) $(OBJS)
+	ln -sf $@.$(VER_MAJOR).$(VER_MINOR) $@.$(VER_MAJOR)
+	ln -sf $@.$(VER_MAJOR).$(VER_MINOR) $@
 
 $(OBJ)/%.o: %.c $(HEAD)
 	$(CC) $(CFLAGS) -D_XL_LIB_BUILD -o $@	$<
 
 install: $(OURLIB)
 	@./mkinstalldirs $(i_libdir) $(MINC)
-	@./install-lib $(OURLIB) $(OURLIB) $(i_libdir) $(VER_MAJOR) $(VER_MINOR)
-	@./install-lib $(OURLIBSO) $(OURLIBSO) $(i_libdir) $(VER_MAJOR) $(VER_MINOR)
+	@install $(OURLIB) $(i_libdir)/$(OURLIB)
+	@./install-lib $(OURLIBSO).$(VER_MAJOR).$(VER_MINOR) $(OURLIBSO) $(i_libdir) $(VER_MAJOR) $(VER_MINOR)
 	@./install-head $(SINC) $(MINC) $(HEADF)
 
 clean:
