diff -urN libnasl/nasl/Makefile libnasl-modified/nasl/Makefile
--- libnasl/nasl/Makefile	Mon Oct 15 12:58:07 2001
+++ libnasl-modified/nasl/Makefile	Fri Jul 26 12:56:36 2002
@@ -41,6 +41,7 @@
 
 install : libnasl.la nasl
 	test -d ${libdir} || $(INSTALL_DIR) -m 755 ${libdir}
+	test -d ${bindir} || $(INSTALL_DIR) -m 755 ${bindir}
 	$(LIBTOOL) --mode install $(INSTALL) libnasl.la ${libdir}/libnasl.la
 	$(LIBTOOL) --finish ${libdir}
 	$(INSTALL) nasl ${bindir}/
