diff -Naur anyterm-1.1.8/browser/anyterm.js anyterm/browser/anyterm.js
--- anyterm-1.1.8/browser/anyterm.js	2005-11-24 19:54:15.000000000 +0100
+++ anyterm/browser/anyterm.js	2006-01-25 16:52:46.000000000 +0100
@@ -26,16 +26,21 @@
 var open=false;
 var session;
 
-//var post_method="POST";
-var post_method="GET";
+#USE=opera#//var post_method="POST";
+#USE=opera#var post_method="GET";
+#USE=-opera#var post_method="POST";
+#USE=-opera#//var post_method="GET";
 
 // Random sequence numbers are needed to prevent Opera from caching
 // replies
 
 var is_opera = navigator.userAgent.toLowerCase().indexOf("opera") != -1;
-if (is_opera) {
-  post_method="GET";
-}
+#USE=opera#if (is_opera) {
+#USE=opera#  post_method="GET";
+#USE=opera#}
+#USE=-opera#//if (is_opera) {
+#USE=-opera#//  post_method="GET";
+#USE=-opera#//}
 
 var seqnum_val=Math.round(Math.random()*100000);
 function cachebust() {
diff -Naur anyterm-1.1.8/browser/.htaccess anyterm/browser/.htaccess
--- anyterm-1.1.8/browser/.htaccess	2005-09-05 00:49:44.000000000 +0200
+++ anyterm/browser/.htaccess	2006-01-25 17:03:29.000000000 +0100
@@ -6,6 +6,8 @@
 # will be ignored if the anyterm module has not been loaded.
 
 <IfModule anyterm>
+#USE=ssl#<IfModule mod_ssl.c>
+#USE=pam#<IfModule mod_auth_pam.c>
 
 # Use an anyterm_command directive to specify the command to run
 # inside the terminal:
@@ -24,6 +26,16 @@
 # Example:
 # anyterm_command '/path/to/anygetty --remotehost "Anyterm: %h" --autologin=%u'
 
+# twp: Use ssh to avoid problems with Gentoo's /bin/login.
+anyterm_command '/usr/bin/ssh %u@localhost'
+
+#USE=ssl#SSLRequireSSL
+#USE=ssl#
+#USE=pam#AuthPAM_Enabled on
+#USE=pam#AuthType Basic
+#USE=pam#AuthName "Anyterm"
+#USE=pam#Require valid-user
+#USE=pam#
 <Files anyterm-module>
   SetHandler anyterm
 
@@ -34,7 +46,10 @@
   # CustomLog /path/to/logfile combined env=!DONTLOG
   # See the Apache documentation for details.  Note "=!" not "!=" !
 
-  # SetEnv DONTLOG
+#USE=opera#  SetEnv DONTLOG
+#USE=-opera#  # SetEnv DONTLOG
 </Files>
 
+#USE=pam#</IfModule>
+#USE=ssl#</IfModule>
 </IfModule>
