diff -aur tomboy-0.3.2-orig/Tomboy/panelapplet/AppletFactory.cs tomboy-0.3.2/Tomboy/panelapplet/AppletFactory.cs
--- tomboy-0.3.2-orig/Tomboy/panelapplet/AppletFactory.cs	2004-10-23 14:21:58.000000000 -0400
+++ tomboy-0.3.2/Tomboy/panelapplet/AppletFactory.cs	2005-06-14 20:50:11.000000000 -0400
@@ -4,9 +4,9 @@
 
 namespace PanelApplet
 {
-	public class AppletFactory
+	public class AppletFactory : GLib.Object
 	{
-		private AppletFactory () { }
+		private AppletFactory () : base (IntPtr.Zero) { }
 
 		private static string _IID;
 		private static string _factoryIID;
@@ -19,7 +19,7 @@
 			cb_wrapper = new PanelAppletSharp.FactoryCallbackWrapper (new FactoryCallback (Creation), null);
 			_IID = applet.IID;
 			_factoryIID = applet.FactoryIID;
-			panel_applet_factory_main(_factoryIID, GLib.Object.LookupGType (applet_type).Val, cb_wrapper.NativeDelegate, IntPtr.Zero);
+			panel_applet_factory_main(_factoryIID, GLib.Object.LookupGType(applet_type).Val, cb_wrapper.NativeDelegate, IntPtr.Zero);
 		}
 
 		private static bool Creation (PanelApplet applet, string iid)
