diff -aur tomboy-0.3.2-orig/Tomboy/Trie.cs tomboy-0.3.2/Tomboy/Trie.cs
--- tomboy-0.3.2-orig/Tomboy/Trie.cs	2005-01-18 02:33:54.000000000 -0500
+++ tomboy-0.3.2/Tomboy/Trie.cs	2005-05-05 13:26:48.000000000 -0400
@@ -186,7 +186,7 @@
 					 MatchHandler match_handler)
 		{
 			TrieState q = root;
-			TrieMatch m;
+			TrieMatch m = null;
 			int idx = 0, start_idx = 0, last_idx = 0;
 
 			while (idx < haystack.Length) {
