diff -ur kdebase-4.3.1-orig/apps/konsole/src/SessionController.cpp kdebase-4.3.1/apps/konsole/src/SessionController.cpp
--- kdebase-4.3.1-orig/apps/konsole/src/SessionController.cpp	2009-04-28 16:46:08.000000000 +0300
+++ kdebase-4.3.1/apps/konsole/src/SessionController.cpp	2009-09-02 20:16:47.393341281 +0300
@@ -95,12 +95,10 @@
     Q_ASSERT( view );
 
     // handle user interface related to session (menus etc.)
-
-#ifdef KONSOLE_PART
-    setXMLFile("konsole/partui.rc");
-#else
-    setXMLFile("konsole/sessionui.rc");
-#endif
+    if (isKonsolePart())
+        setXMLFile("konsole/partui.rc");
+    else
+        setXMLFile("konsole/sessionui.rc");
 
     setupActions();
     actionCollection()->addAssociatedWidget(view);
@@ -1041,6 +1039,15 @@
     }
 }
 
+bool SessionController::isKonsolePart() const
+{
+    // Check to see if we are being called from Konsole or a KPart
+    if (QString(kapp->metaObject()->className()) == "Konsole::Application")
+        return false;
+    else
+        return true;
+}
+
 SessionTask::SessionTask(QObject* parent)
     :  QObject(parent)
     ,  _autoDelete(false)
diff -ur kdebase-4.3.1-orig/apps/konsole/src/SessionController.h kdebase-4.3.1/apps/konsole/src/SessionController.h
--- kdebase-4.3.1-orig/apps/konsole/src/SessionController.h	2009-04-15 13:25:25.000000000 +0300
+++ kdebase-4.3.1/apps/konsole/src/SessionController.h	2009-09-02 20:16:50.539589477 +0300
@@ -209,6 +209,9 @@
 
     void updateSearchFilter();
 
+    /* Returns true if called within a KPart; false if called within Konsole. */
+    bool isKonsolePart() const;
+
 private:
     // begins the search
     // text - pattern to search for