--- numarray-1.5.2/Src/libnumarray.ch~	2007-05-24 14:21:43 -0600
+++ numarray-1.5.2/Src/libnumarray.ch	2007-05-24 14:22:58 -0600
@@ -1920,9 +1920,9 @@
 	return retstatus;
 }
 
-#elif defined(linux) || defined(darwin) || defined(__CYGWIN__)
+#elif defined(linux) || defined(__FreeBSD__) || defined(darwin) || defined(__CYGWIN__)
 
-#if defined(__GLIBC__) || defined(darwin) || defined(__MINGW32__)
+#if defined(__GLIBC__) || defined(__FreeBSD__) || defined(darwin) || defined(__MINGW32__)
 #include <fenv.h>
 #elif defined(__CYGWIN__)
 #include <mingw/fenv.h>
@@ -3886,7 +3886,7 @@
 }
 
 /* Likewise for Integer overflows */
-#if defined(linux)
+#if defined(linux) || defined(__FreeBSD__)
 static int int_overflow_error(Float64 value) { /* For x86_64 */
 	feraiseexcept(FE_OVERFLOW);
 	return (int) value;
