diff -Nur Include.orig/numarray/arraybase.h Include/numarray/arraybase.h
--- Include.orig/numarray/arraybase.h	2007-02-08 16:08:06.000000000 +0000
+++ Include/numarray/arraybase.h	2007-02-08 16:07:30.000000000 +0000
@@ -1,6 +1,8 @@
 #if !defined(_arraybase_h)
 #define _arraybase_h 1
 
+#include <Python.h>
+
 #if defined(SIZEOF_VOID_P)
 #if SIZEOF_VOID_P == 8
 #define LP64 1
@@ -17,6 +19,8 @@
 #define HAS_UINT64 1
 #endif
 
+#include "numconfig.h"
+
 #define REFCOUNT(obj) (((PyObject *)(obj))->ob_refcnt)
 #define MAX_ELSIZE 16
 
diff -Nur Include.orig/numarray/cfunc.h Include/numarray/cfunc.h
--- Include.orig/numarray/cfunc.h	2007-02-08 16:08:06.000000000 +0000
+++ Include/numarray/cfunc.h	2007-02-08 16:07:30.000000000 +0000
@@ -1,6 +1,8 @@
 #if !defined(__cfunc__)
 #define __cfunc__ 1
 
+#include <Python.h>
+
 typedef PyObject *(*CFUNCasPyValue)(void *);
 typedef int (*UFUNC)(long, long, long, void **, long*);
 /* typedef void (*CFUNC_2ARG)(long, void *, void *); */
diff -Nur Include.orig/numarray/nummacro.h Include/numarray/nummacro.h
--- Include.orig/numarray/nummacro.h	2007-02-08 16:08:06.000000000 +0000
+++ Include/numarray/nummacro.h	2007-02-08 16:07:30.000000000 +0000
@@ -1,6 +1,8 @@
 #if !defined(_ndarraymacro)
 #define _ndarraymacro
 
+#include <Python.h>
+
 /* The structs defined here are private implementation details of numarray
 which are subject to change w/o notice.
 */
diff -Nur Include.orig/numarray/tc.h Include/numarray/tc.h
--- Include.orig/numarray/tc.h	2007-02-08 16:08:06.000000000 +0000
+++ Include/numarray/tc.h	2007-02-08 16:07:30.000000000 +0000
@@ -1,5 +1,7 @@
 #if defined(MEASURE_TIMING)
 
+#include <Python.h>
+
 /* ----------------------------------------------------------------------- */
 
 typedef struct 
