--- pcap_interface.c.orig	2004-07-10 07:58:10.000000000 +0300
+++ pcap_interface.c	2007-07-09 13:40:18.000000000 +0300
@@ -513,9 +513,7 @@
     break;
 #endif
   default:
-    fprintf(stderr, "at line %d\n", __LINE__);
-    throw_exception(-1, "unsupported address family");
-    return NULL;
+    return PyString_FromFormat("<AF %d>", sa->sa_family);
   case AF_UNSPEC:
     Py_INCREF(Py_None);
     return Py_None;
