--- a/ptracemodule.c	2001-11-26 04:24:30.000000000 +0000
+++ b/ptracemodule.c	2003-10-02 15:28:17.173323016 +0100
@@ -292,9 +292,10 @@
 #ifdef PTRACE_SETOPTIONS
 
 static char ptrace_settracesysgood__doc__[] =
-"settracesysgood(pid) -> None\n\
-Set the TRACESYSGOOD flag for this child.  This causes the high (0x80) bit to
- be set on system call stops.";
+"settracesysgood(pid) -> None\n" \
+"Set the TRACESYSGOOD flag for this child." \
+"This causes the high (0x80) bit to" \
+" be set on system call stops.";
 
 static PyObject *
 ptrace_settracesysgood(PyObject *self, PyObject *args)
