Index: c_icap-0.1.3/Makefile.am
===================================================================
--- c_icap-0.1.3.orig/Makefile.am
+++ c_icap-0.1.3/Makefile.am
@@ -37,8 +37,8 @@ c_icap_SOURCES = aserver.c request.c cfg
 # libicapapi ......
 libicapapi_la_CFLAGS= -Iinclude/ @ZLIB_ADD_FLAG@ -DCI_BUILD_LIB
 
-libicapapi_la_LIBADD = @ZLIB_ADD_LDFLAG@
-libicapapi_la_LDFLAGS= -shared -version-info @CICAPLIB_VERSION@
+libicapapi_la_LIBADD = @ZLIB_ADD_LDFLAG@ @THREADS_LDADD@ @DL_ADD_FLAG@
+libicapapi_la_LDFLAGS= -shared -version-info @CICAPLIB_VERSION@ @THREADS_LDFLAGS@
 
 
 #c_icap the main server 
@@ -47,7 +47,7 @@ c_icap_CFLAGS= -Iinclude/ -DCONFDIR=\"$(
                           -DSERVDIR=\"$(SERVICESDIR)\" -DLOGDIR=\"$(LOGDIR)\"  \
                           -DDATADIR=\"$(DATADIR)\"
 
-c_icap_LDADD = @THREADS_LDADD@ libicapapi.la @DL_ADD_FLAG@
+c_icap_LDADD = libicapapi.la
 c_icap_LDFLAGS = -rdynamic -rpath @libdir@ @THREADS_LDFLAGS@
 
 
