diff -urN uim.orig/configure.ac uim/configure.ac
--- uim.orig/configure.ac	2005-03-16 14:20:53.000000000 +0900
+++ uim/configure.ac	2005-04-24 09:39:47.129795192 +0900
@@ -232,7 +232,7 @@
       ;;
       yes|*)
         PKG_CHECK_MODULES(GTK2, gtk+-2.0 >= 2.2.0, use_gtk2="yes",use_gtk2="no")
-        PKG_CHECK_MODULES(GTK2_4, gtk+-2.0 >= 2.4.0, use_gtk2.4="yes",use_gtk2.4="no")
+        PKG_CHECK_MODULES(GTK2_4, gtk+-2.0 >= 2.4.0, use_gtk2_4="yes",use_gtk2_4="no")
       ;;
     esac
   ],
