Allows to use arbitrary annotations with the cyradm tool.

diff -r a07884e259c3 doc/man/cyradm.1.html
--- a/doc/man/cyradm.1.html	Thu Oct 25 08:24:39 2007 +0200
+++ b/doc/man/cyradm.1.html	Thu Oct 25 08:29:40 2007 +0200
@@ -241,6 +241,13 @@ The currently supported attributes are:<
 <dd>
 <p>Sets an email address to which messages injected into the server via NNTP 
 will be sent.</p>
+</dd>
+</li>
+<dt><strong><a NAME="item__2fexplicit_2fannotation"><code>/explicit/annotation</code></a></strong>
+
+<dd>
+<p>Sets the annotation <em>/explicit/annotation</em>
+on <em>mailbox</em> to <em>value</em>.</p>
 </dd>
 </li>
 <dt><strong><a name="item_sharedseen"><code>sharedseen</code></a></strong>
diff -r a07884e259c3 perl/imap/IMAP/Admin.pm
--- a/perl/imap/IMAP/Admin.pm	Thu Oct 25 08:24:39 2007 +0200
+++ b/perl/imap/IMAP/Admin.pm	Thu Oct 25 08:29:40 2007 +0200
@@ -796,11 +796,11 @@ sub mboxconfig {
     return undef;
   }
 
-  if(!exists($values{$entry})) {
-    $self->{error} = "Unknown parameter $entry";
-  }
-
-  $entry = $values{$entry};
+  if(exists($values{$entry})) {
+    $entry = $values{$entry};    
+  } else {
+    $self->{error} = "Unknown parameter $entry" unless substr($entry,0,1) eq "/";
+  }
 
   my ($rc, $msg);
 
diff -r a07884e259c3 perl/imap/IMAP/Shell.pm
--- a/perl/imap/IMAP/Shell.pm	Thu Oct 25 08:24:39 2007 +0200
+++ b/perl/imap/IMAP/Shell.pm	Thu Oct 25 08:29:40 2007 +0200
@@ -126,7 +126,7 @@ my %builtins = (exit =>
 		  [\&_sc_info, '[mailbox]',
 		   'display mailbox/server metadata'],
 		mboxcfg =>
-		  [\&_sc_mboxcfg, 'mailbox [comment|condstore|expire|news2mail|sharedseen|sieve|squat] value',
+		  [\&_sc_mboxcfg, 'mailbox [comment|condstore|news2mail|expire|sieve|squat|/<explicit annotation>] value',
 		   'configure mailbox'],
 		mboxconfig => 'mboxcfg',
 		reconstruct =>
@@ -1436,7 +1436,7 @@ sub _sc_mboxcfg {
   while (defined ($opt = shift(@argv))) {
     last if $opt eq '--';
     if ($opt =~ /^-/) {
-      die "usage: mboxconfig mailbox [comment|condstore|expire|news2mail|sharedseen|sieve|squat] value\n";
+      die "usage: mboxconfig mailbox [comment|condstore|expire|news2mail|sharedseen|sieve|squat|/<explicit annotation>] value\n";
     }
     else {
       push(@nargv, $opt);
@@ -1445,7 +1445,7 @@ sub _sc_mboxcfg {
   }
   push(@nargv, @argv);
   if (@nargv < 2) {
-    die "usage: mboxconfig mailbox [comment|condstore|expire|news2mail|sharedseen|sieve|squat] value\n";
+    die "usage: mboxconfig mailbox [comment|condstore|expire|news2mail|sharedseen|sieve|squat|/<explicit annotation>] value\n";
   }
   if (!$cyrref || !$$cyrref) {
     die "mboxconfig: no connection to server\n";
diff -r a07884e259c3 perl/imap/cyradm.sh
--- a/perl/imap/cyradm.sh	Thu Oct 25 08:24:39 2007 +0200
+++ b/perl/imap/cyradm.sh	Thu Oct 25 08:29:40 2007 +0200
@@ -241,6 +241,10 @@ mailboxes).
 
 Indicates that the mailbox should have a squat index created for it.
 
+=item C</explicit/annotation>
+
+Sets the annotation I</explicit/annotation> on I<mailbox> to I<value>. 
+
 =back 
 
 =item C<renamemailbox> [C<--partition> I<partition>] I<oldname> I<newname>
