--- exiv2-0.21.orig/src/xmp.cpp	2010-11-06 13:53:15.000000000 +0000
+++ exiv2-0.21/src/xmp.cpp	2010-12-20 12:35:34.931520681 +0000
@@ -420,11 +420,10 @@ namespace Exiv2 {
         }
     } // XmpParser::registerNs
 #else
-    bool XmpParser::registerNs(const std::string& /*ns*/,
+    void XmpParser::registerNs(const std::string& /*ns*/,
                                const std::string& /*prefix*/)
     {
         initialize();
-        return true;
     } // XmpParser::registerNs
 #endif
 
