--- src/jpgimage.cpp.orig	2006-06-02 17:29:31.000000000 +0200
+++ src/jpgimage.cpp	2006-08-19 00:59:00.000000000 +0200
@@ -314,8 +314,9 @@
                 // Find actual Iptc data within the APP13 segment
                 if (!Photoshop::locateIptcIrb(psData.pData_, psData.size_,
                                           &record, &sizeHdr, &sizeIptc)) {
-                    assert(sizeIptc);
-                    if (iptcData_.load(record + sizeHdr, sizeIptc)) throw Error(36, "IPTC");
+                    if (sizeIptc) {
+                        if (iptcData_.load(record + sizeHdr, sizeIptc)) throw Error(36, "IPTC");
+                    }
                 }
                 --search;
             }
