--- cairo-0.4.0/src/Makefile.in.old	2005-03-28 02:57:44.924552656 -0500
+++ cairo-0.4.0/src/Makefile.in	2005-03-28 02:59:13.385104608 -0500
@@ -653,7 +653,7 @@
 
 info-am:
 
-install-data-am: install-cairoincludeHEADERS install-data-local
+install-data-am: install-cairoincludeHEADERS
 
 install-exec-am: install-libLTLIBRARIES
 
@@ -689,7 +689,7 @@
 	distclean-compile distclean-generic distclean-libtool \
 	distclean-tags distdir dvi dvi-am html html-am info info-am \
 	install install-am install-cairoincludeHEADERS install-data \
-	install-data-am install-data-local install-exec \
+	install-data-am install-exec \
 	install-exec-am install-info install-info-am \
 	install-libLTLIBRARIES install-man install-strip installcheck \
 	installcheck-am installdirs maintainer-clean \
@@ -698,16 +698,6 @@
 	tags uninstall uninstall-am uninstall-cairoincludeHEADERS \
 	uninstall-info-am uninstall-libLTLIBRARIES
 
-
-install-data-local:
-	@if test -f $(includedir)/cairo.h || test -f $(includedir)/cairo-features.h ; then \
-	  echo "****************************************************************" ;	\
-	  echo "*** Error: Old headers found. You should remove the following" ;	\
-	  echo "***        files and then type 'make install' again." ;			\
-	  ls $(includedir)/cairo*.h ;							\
-	  echo "****************************************************************" ;	\
-	  false ;									\
-        fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
