diff -ruN isdn4k-utils.orig/isdnlog/Makefile.in isdn4k-utils/isdnlog/Makefile.in
--- isdn4k-utils.orig/isdnlog/Makefile.in	2002-07-19 21:03:55.000000000 +0200
+++ isdn4k-utils/isdnlog/Makefile.in	2003-04-26 12:06:28.000000000 +0200
@@ -1794,6 +1794,7 @@
 				$(INSTALL_DATA) samples/$$f.conf.$(NATION) $(DESTDIR)$(I4LCONFDIR)/$$f.conf.new ; \
 			fi ; \
 		done
+		@[ -f "$(DESTDIR)$(I4LCONFDIR)/isdn.conf" ] || $(INSTALL_DATA) samples/isdn.conf $(DESTDIR)$(I4LCONFDIR)
 		@for f in callerid.conf isdnlog.isdnctrl0.options isdnlog.users ; do \
 			if [ ! -f "$(DESTDIR)$(I4LCONFDIR)/$$f" ]; then \
 				$(INSTALL_DATA) samples/$$f $(DESTDIR)$(I4LCONFDIR)/$$f ; \
@@ -1836,8 +1837,8 @@
 		$(INSTALL_DATA) tools/isdnrate.1     $(DESTDIR)$(MAN1DIR)/isdnrate$(MAN1EXT)
 		$(INSTALL_DATA) tools/rate-files.5     $(DESTDIR)$(MAN5DIR)/rate-files$(MAN5EXT)
 		$(INSTALL_DATA) country-de.dat	$(DESTDIR)$(DATADIR)/country.dat
-		$(INSTALL_DATA) rate-$(NATION).dat      $(DESTDIR)$(DATADIR)
-		$(INSTALL_DATA) holiday-$(NATION).dat   $(DESTDIR)$(DATADIR)
+		@[ ! -f rate-${NATION}.dat ] || $(INSTALL_DATA) rate-$(NATION).dat      $(DESTDIR)$(DATADIR)
+		@[ ! -f holiday-${NATION}.dat ] || $(INSTALL_DATA) holiday-$(NATION).dat   $(DESTDIR)$(DATADIR)
 		@for f in `ls zone-$(NATION)-*.gdbm zone-$(NATION)-*.db zone-$(NATION)-*.cdb tools/dest/dest.gdbm tools/dest/dest.db tools/dest/dest.cdb 2>/dev/null` ; do \
 		  $(INSTALL_DATA) $$f $(DESTDIR)$(DATADIR); \
 		  echo Installing $(DESTDIR)$(DATADIR)/$$f; \
diff -ruN isdn4k-utils.orig/isdnlog/samples/isdn.conf isdn4k-utils/isdnlog/samples/isdn.conf
--- isdn4k-utils.orig/isdnlog/samples/isdn.conf	2002-01-29 17:39:20.000000000 +0100
+++ isdn4k-utils/isdnlog/samples/isdn.conf	2003-04-26 12:05:01.000000000 +0200
@@ -20,13 +20,13 @@
 CHARGEMAX       = 50.00
 CURRENCY = 0.062,EUR
 
-COUNTRYFILE = /usr/lib/isdn/country.dat
-DESTFILE= /usr/lib/isdn/dest.cdb
+COUNTRYFILE = /usr/share/isdn/country.dat
+DESTFILE= /usr/share/isdn/dest.cdb
 RATECONF= /etc/isdn/rate.conf
 # replace the xx in the next 3 lines with your country's letters!
-RATEFILE= /usr/lib/isdn/rate-xx.dat
-HOLIDAYS= /usr/lib/isdn/holiday-xx.dat
-ZONEFILE= /usr/lib/isdn/zone-xx-%s.cdb
+RATEFILE= /usr/share/isdn/rate-xx.dat
+HOLIDAYS= /usr/share/isdn/holiday-xx.dat
+ZONEFILE= /usr/share/isdn/zone-xx-%s.cdb
 
 # providerselect
 VBN = 10	# change this, depends on your country
