--- gtkhtml-3.2.5/src/gtkhtml.c.orig	2005-03-13 16:04:23.000000000 +0000
+++ gtkhtml-3.2.5/src/gtkhtml.c	2005-03-13 16:07:42.000000000 +0000
@@ -27,7 +27,7 @@
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <gtk/gtk.h>
 #include <string.h>
-
+#include <glib/gi18n-lib.h>
 #include <gnome.h>
 
 #include "../a11y/factory.h"
--- gtkhtml-3.2.5/src/htmlbutton.c.orig	2005-03-13 16:03:52.000000000 +0000
+++ gtkhtml-3.2.5/src/htmlbutton.c	2005-03-13 16:08:00.000000000 +0000
@@ -26,6 +26,7 @@
 #include "htmlform.h"
 #include <string.h>
 #include <libgnome/gnome-i18n.h>
+#include <glib/gi18n-lib.h>
 
 HTMLButtonClass html_button_class;
 
--- gtkhtml-3.2.5/src/htmlengine-print.c.orig	2005-03-13 16:06:27.000000000 +0000
+++ gtkhtml-3.2.5/src/htmlengine-print.c	2005-03-13 16:09:36.000000000 +0000
@@ -23,6 +23,8 @@
 #include <gtk/gtk.h>
 #include <libgnome/gnome-i18n.h>
 #include <libgnomeui/gnome-dialog-util.h>
+#include <glib/gi18n-lib.h>
+
 #include "gtkhtml.h"
 #include "gtkhtml-private.h"
 #include "gtkhtml-properties.h"
--- gtkhtml-3.2.5/src/test.c.orig	2005-03-13 16:10:20.000000000 +0000
+++ gtkhtml-3.2.5/src/test.c	2005-03-13 16:12:28.000000000 +0000
@@ -36,8 +36,10 @@
 #include <gtk/gtkbutton.h>
 #include <gtk/gtksignal.h>
 
+#include <config.h>
 #include <libgnome/gnome-i18n.h>
 #include <libgnomeui/gnome-ui-init.h>
+#include <glib/gi18n-lib.h>
 
 #include "gtkhtml.h"
 #include "gtkhtmldebug.h"
