--- mixxx-1.10.0/plugins/soundsourcem4a/m4a/mp4.c
+++ mixxx-1.10.0/plugins/soundsourcem4a/m4a/mp4.c
@@ -114,7 +114,11 @@
 	faacDecSetConfiguration(priv->decoder, neaac_cfg);
 
 	/* open mpeg-4 file */
+#ifdef MP4_CLOSE_DO_NOT_COMPUTE_BITRATE /* r479 fix */
+	priv->mp4.handle = MP4Read(ip_data->filename);
+#else
 	priv->mp4.handle = MP4Read(ip_data->filename, 0);
+#endif
 	if (!priv->mp4.handle) {
 		d_print("MP4Read failed\n");
 		goto out;
--- mixxx-1.10.0/plugins/soundsourcem4a/m4a/mp4-mixxx.cpp
+++ mixxx-1.10.0/plugins/soundsourcem4a/m4a/mp4-mixxx.cpp
@@ -164,7 +164,11 @@
     faacDecSetConfiguration(priv->decoder, neaac_cfg);
 
     /* open mpeg-4 file */
+#ifdef MP4_CLOSE_DO_NOT_COMPUTE_BITRATE /* r479 fix */
+    priv->mp4.handle = MP4Read(ip_data->filename);
+#else
     priv->mp4.handle = MP4Read(ip_data->filename, 0);
+#endif
     if (!priv->mp4.handle) {
         qDebug() << "MP4Read failed";
         goto out;
