--- Makefile.old	2010-10-01 09:57:28.000000000 +0200
+++ Makefile	2010-10-01 09:57:52.000000000 +0200
@@ -36,7 +36,7 @@
 	rm -f $(MAIN)/gb_debug.txt $(SYSDIR)/gb_debug.txt
 
 $(PROGNAME): $(OBJS)
-	$(CC) $(CFLAGS) $(OBJS) -o $(PROGNAME) $(LIBS)
+	$(CC) $(LDFLAGS) $(CFLAGS) $(OBJS) -o $(PROGNAME) $(LIBS)
 
 bin: all
 	strip --strip-unneeded $(PROGNAME)
--- GUI_unx.mak.old	2010-10-01 10:00:08.000000000 +0200
+++ GUI_unx.mak	2010-10-01 10:00:40.000000000 +0200
@@ -61,7 +61,7 @@
 	rm -f $(PROGNAME) $(SYSDIR)/*.o $(SYSDIR)/core
 
 $(PROGNAME): $(OBJS)
-	$(CXX) $(CFLAGS) $(OBJS) -o $(PROGNAME) $(LDFLAGS) $(LIBS)
+	$(CXX) $(LDFLAGS) $(CFLAGS) $(OBJS) -o $(PROGNAME) $(LDFLAGS) $(LIBS)
 
 bin: all
 	strip --strip-unneeded $(PROGNAME)
