--- devtools/Site/site.config.m4.orig	2007-10-13 13:44:57.000000000 +1000
+++ devtools/Site/site.config.m4	2007-10-13 13:45:45.000000000 +1000
@@ -67,9 +67,9 @@
 dnl to have the "tre" library and its header files installed.  If necessary,
 dnl enable these lines by deleting "dnl" from the front of them and edit
 dnl paths as needed.
-dnl APPENDDEF(`confINCDIRS', `-I/usr/local/include/tre ')
+APPENDDEF(`confINCDIRS', `-I/usr/include/tre ')
 dnl APPENDDEF(`confLIBDIRS', `-L/usr/local/lib/tre ')
-dnl APPENDDEF(`confLIBS', `-ltre ')
+APPENDDEF(`confLIBS', `-ltre ')
 
 dnl Code For Future Release (FFRs):
 dnl
