--- 0.5/Makefile	2008-10-15 20:58:07.000000000 +0200
+++ 0.5-1/Makefile	2008-10-15 20:57:51.000000000 +0200
@@ -33,9 +33,9 @@
 PNGINC = ${shell libpng-config --cflags}
 PNGLIB = ${shell libpng-config --ldflags}
 
-CFLAGS = -O3 -g -Wall -pedantic -I. -D VERSION=\"$(VERSION)\" $(PNGINC) -funroll-loops -fomit-frame-pointer 
+CFLAGS += -O3 -g -Wall -pedantic -I. -D VERSION=\"$(VERSION)\" $(PNGINC) -funroll-loops -fomit-frame-pointer 
 
-LDFLAGS = $(PNGLIB)
+LDFLAGS += $(PNGLIB)
 
 OBJS = pngnq.o rwpng.o neuquant32.o freegetopt/getopt.o
 #############################################
