--- scrotwm-0.9.28/linux/Makefile
+++ scrotwm-0.9.28/linux/Makefile
@@ -1,7 +1,7 @@
 # $scrotwm: Makefile,v 1.5 2010/09/16 02:00:51 marco Exp $ 
 
-CFLAGS+= -Wall -ggdb -D_GNU_SOURCE -I.
-CFLAGS+= -DSWM_LIB=\"$(LIBDIR)/libswmhack.so.$(LVERS)\"
+CFLAGS+= -D_GNU_SOURCE -I.
+CFLAGS+= -DSWM_LIB=\"libswmhack.so.$(LVERS)\"
 LDADD+= -lX11 -lXrandr
 
 PREFIX?= /usr/local
@@ -9,7 +9,7 @@
 LIBDIR?= $(PREFIX)/lib
 MANDIR?= $(PREFIX)/share/man
 
-CC= gcc
+CC?= gcc
 
 LVERS= $(shell . ../lib/shlib_version; echo $$major.$$minor)
 
@@ -25,10 +25,10 @@
 	$(CC) $(LDFLAGS) -o $@ $+ $(LDADD)
 
 %.so: %.c
-	$(CC) $(CFLAGS) -c -fpic -DPIC $+ -o $@
+	$(CC) $(LDFLAGS) $(CFLAGS) -c -fpic $+ -o $@
 
 libswmhack.so.$(LVERS): swm_hack.so
-	$(CC) -shared -fpic -o libswmhack.so.$(LVERS) swm_hack.so $(LDADD)
+	$(CC) $(LDFLAGS) -Wl,-soname,$@ -shared -fpic -o libswmhack.so.$(LVERS) swm_hack.so $(LDADD)
 
 install: all
 	install -m 755 -d $(DESTDIR)$(BINDIR)
