--- ecb-help.el	2002-08-12 14:53:10.000000000 -0500
+++ /tmp/ecb-help.el	2002-09-01 20:19:32.000000000 -0500
@@ -57,19 +57,18 @@
                                            '("html" "info"))))
              ecb-show-help-format)))
     (if (equal f 'info)
-        (if (file-exists-p (concat ecb-ecb-dir "ecb.info"))
-            (info (concat ecb-ecb-dir "ecb.info"))
-          (ecb-error "File %s does not exist" (concat ecb-ecb-dir "ecb.info")))
+        (if (file-exists-p "@ECBINFOFILE@")
+	    (info "@ECBINFOFILE@")
+          (ecb-error "File %s does not exist" "@ECBINFOFILE@"))
       (message "Opening ECB online-help in a web-browser...")
-      (if (file-exists-p (concat ecb-ecb-dir "ecb.html"))
+      (if (file-exists-p "@ECBHTMLFILE@")
           (progn
-            (browse-url (concat "file://"
-                                (ecb-fix-filename ecb-ecb-dir "ecb.html"))
+            (browse-url (concat "file://" "@ECBHTMLFILE@")
                         (if (boundp 'browse-url-new-window-flag)
                             browse-url-new-window-flag
                           browse-url-new-window-p))
             (message "Opening ECB online-help in a web-browser...done"))
-        (ecb-error "File %s does not exist" (concat ecb-ecb-dir "ecb.html"))))))
+        (ecb-error "File %s does not exist" "@ECBHTMLFILE@")))))
 
 ;;
 ;; Problem reporting functions stolen from JDE
