--- include/gbfwordjs.h.orig	2008-06-09 00:45:08.000000000 +0200
+++ include/gbfwordjs.h	2008-06-09 00:45:49.000000000 +0200
@@ -21,6 +21,8 @@
 #ifndef GBFWORDSJS_H
 #define GBFWORDSJS_H
 
+
+#include <cstdio>
 #include <swoptfilter.h>
 
 SWORD_NAMESPACE_START

--- include/flatapi.h.orig	2008-06-09 00:51:23.000000000 +0200
+++ include/flatapi.h	2008-06-09 00:51:43.000000000 +0200
@@ -29,6 +29,8 @@
 
 extern "C" {
 
+
+#include <stdint.h>
 #define SWHANDLE intptr_t
 
 

--- utilities/diatheke/corediatheke.h.orig	2008-06-09 00:57:06.000000000 +0200
+++ utilities/diatheke/corediatheke.h	2008-06-09 00:57:57.000000000 +0200
@@ -3,7 +3,8 @@
 // Licensed under GNU General Public License (GPL)
 // see accompanying LICENSE file for license details
 
-#include <stdio.h>
+#include <cstdio>
+#include <fstream>
 
 #include "diathekemgr.h"
 #include <localemgr.h>

--- utilities/diatheke/osiscgi.h.orig	2008-06-09 01:01:14.000000000 +0200
+++ utilities/diatheke/osiscgi.h	2008-06-09 01:01:45.000000000 +0200
@@ -19,6 +19,7 @@
 
 #include <swbasicfilter.h>
 
+#include <cctype>
 #include <defs.h>
 
 SWORD_NAMESPACE_START

