--- include/multimapwdef.h.old	2004-04-29 04:22:07.516715976 +0100
+++ include/multimapwdef.h	2004-04-29 04:22:20.051810352 +0100
@@ -12,14 +12,14 @@
 public:
 	typedef std::pair<const Key, T> value_type;
 	T& getWithDefault(const Key& k, const T& defaultValue) {
-		if (find(k) == end()) {
+		if (find(k) == this->end()) {
 			insert(value_type(k, defaultValue));
 		}
 		return (*(find(k))).second;
 	}
 
 	T& operator[](const Key& k) {
-		if (find(k) == end()) {
+		if (find(k) == this->end()) {
 			insert(value_type(k, T()));
 		}
 		return (*(find(k))).second;
