--- libtool-1.4.2/libtool.m4.s390	Tue Sep 18 10:47:23 2001
+++ libtool-1.4.2/libtool.m4	Tue Sep 18 10:47:49 2001
@@ -3372,7 +3372,7 @@
 # This must be Linux ELF.
 linux-gnu*)
   case $host_cpu in
-  alpha* | hppa* | i*86 | mips | mipsel | powerpc* | sparc* | ia64*)
+  alpha* | hppa* | i*86 | mips | mipsel | powerpc* | sparc* | ia64* | s390* | x86_64*)
     lt_cv_deplibs_check_method=pass_all ;;
   *)
     # glibc up to 2.1.1 does not perform some relocations on ARM
