Index: fftw-2.1.5/rfftw/Makefile.am
===================================================================
--- fftw-2.1.5.orig/rfftw/Makefile.am
+++ fftw-2.1.5/rfftw/Makefile.am
@@ -18,6 +18,7 @@ lib@FFTW_PREFIX@rfftw_la_SOURCES = $(COD
 					 rfftw.h                   
 
 lib@FFTW_PREFIX@rfftw_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
+lib@FFTW_PREFIX@rfftw_la_LIBADD = ../fftw/lib@FFTW_PREFIX@fftw.la
 MAINTAINERCLEANFILES = $(CODELETS) rconfig.c
 DISTCLEANFILES = srfftw.h drfftw.h
 
