--- Makefile
+++ Makefile
@@ -34,10 +34,10 @@
 VERSION := $(shell sh mk-version)
 RELEASE_VERSION := $(shell cat VERSION)
 
-CC=gcc
-CFLAGS += -W -Wall -O3 -Wmissing-declarations -Wwrite-strings -g
-CPPFLAGS = -DVERSION=\"$(VERSION)\"
-LDFLAGS = -g $(shell libgcrypt-config --libs)
+CC?=gcc
+CFLAGS += -W -Wall -Wmissing-declarations -Wwrite-strings
+CPPFLAGS += -DVERSION=\"$(VERSION)\"
+LDFLAGS += $(shell libgcrypt-config --libs)
 CFLAGS +=  $(shell libgcrypt-config --cflags)
 
 ifeq ($(shell uname -s), SunOS)
