--- Makefile.PL.bak     2005-11-21 13:24:06.000000000 -0500
+++ Makefile.PL 2005-11-21 13:28:04.000000000 -0500
@@ -16,7 +16,11 @@ my $DEBUG = delete $config{DEBUG};
 eval {
     print "running gdome-config... ";
     my $ver = backtick('gdome-config --version');
-    my ($major, $minor, $point) = $ver =~ /(\d+).(\d+)\.(\d+)/g;
+       my $point =0;
+       my ($major, $minor) = $ver =~ /(\d+)\.(\d+)/g;
+       if ($minor =~ m/\./) {
+               ($minor,$point) = $minor =~ /(\d+)\.(\d+)/g;
+               }
     die "VERSION" unless $major == 0 && (($minor == 7 && $point >= 2)
                                         || ($minor >= 8));
     $config{LIBS} ||= backtick('gdome-config --libs');
