From 039ca32b6787bf06a52e12d51952c4453e1c6bbe Mon Sep 17 00:00:00 2001
From: Gabriel Burt <gabriel.burt@gmail.com>
Date: Thu, 11 Mar 2010 09:07:24 -0800
Subject: [PATCH] [Magnatune] Replace unicode char with \u expansion

---
 .../gtk-gui/Magnatune.Configuration.cs             |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Magnatune/Banshee.Magnatune/gtk-gui/Magnatune.Configuration.cs b/src/Magnatune/Banshee.Magnatune/gtk-gui/Magnatune.Configuration.cs
index ecb54b2..a5bf178 100644
--- a/src/Magnatune/Banshee.Magnatune/gtk-gui/Magnatune.Configuration.cs
+++ b/src/Magnatune/Banshee.Magnatune/gtk-gui/Magnatune.Configuration.cs
@@ -66,7 +66,7 @@ namespace Banshee.Magnatune {
             this.username.CanFocus = true;
             this.username.Name = "username";
             this.username.IsEditable = true;
-            this.username.InvisibleChar = '●';
+            this.username.InvisibleChar = '\u25cf';
             this.hbox2.Add(this.username);
             Gtk.Box.BoxChild w3 = ((Gtk.Box.BoxChild)(this.hbox2[this.username]));
             w3.Position = 1;
@@ -94,7 +94,7 @@ namespace Banshee.Magnatune {
             this.password.Name = "password";
             this.password.IsEditable = true;
             this.password.Visibility = false;
-            this.password.InvisibleChar = '●';
+            this.password.InvisibleChar = '\u25cf';
             this.hbox1.Add(this.password);
             Gtk.Box.BoxChild w6 = ((Gtk.Box.BoxChild)(this.hbox1[this.password]));
             w6.Position = 1;
-- 
1.7.0.3

