--- info/emodules.info.original	Fri May 10 17:29:36 2002
+++ info/emodules.info	Fri May 10 17:29:17 2002
@@ -1,6 +1,11 @@
 This is ../info/emodules.info, produced by makeinfo version 4.1 from
 emodules.texi.
 
+INFO-DIR-SECTION XEmacs Editor
+START-INFO-DIR-ENTRY
+* emodules: (emodules).                XEmacs module loading technology.
+END-INFO-DIR-ENTRY
+
    This file documents the module loading technology of XEmacs.
 
    Copyright (C) 1998 J. Kean Johnston.
