--- configure.in.orig	Sun Jun 23 13:25:00 2002
+++ configure.in	Sun Jun 23 13:25:20 2002
@@ -2538,7 +2538,7 @@
   AC_MSG_CHECKING(for \"-z nocombreloc\" linker flag)
   case "`ld --help 2>&1`" in
     *-z\ nocombreloc* ) AC_MSG_RESULT(yes)
-      XE_PREPEND(-z nocombreloc, ld_switch_site) ;;
+      XE_PREPEND(-Wl,-z,nocombreloc, ld_switch_site) ;;
     *) AC_MSG_RESULT(no) ;;
   esac
 fi
--- configure.orig	Sun Jun 23 13:25:04 2002
+++ configure	Sun Jun 23 13:25:56 2002
@@ -5010,7 +5010,7 @@
 echo "configure:5011: checking for \"-z nocombreloc\" linker flag" >&5
   case "`ld --help 2>&1`" in
     *-z\ nocombreloc* ) echo "$ac_t""yes" 1>&6
-      ld_switch_site="-z nocombreloc $ld_switch_site" &&  if test "$extra_verbose" = "yes"; then echo "    Prepending \"-z nocombreloc\" to \$ld_switch_site"; fi ;;
+      ld_switch_site="-Wl,-z,nocombreloc $ld_switch_site" &&  if test "$extra_verbose" = "yes"; then echo "    Prepending \"-Wl,-z,nocombreloc\" to \$ld_switch_site"; fi ;;
     *) echo "$ac_t""no" 1>&6 ;;
   esac
 fi
