diff -ur src-new/Makefile src/Makefile
--- Makefile	Wed Mar 28 15:40:00 2001
+++ Makefile	Thu Mar 21 12:22:47 2002
@@ -3,12 +3,12 @@
 #
 # Adjust these two lines for your ANSI C compiler
 CC=gcc -ansi -pedantic
-CFLAGS=
+#CFLAGS=
 # If you have Icon, you should use the Icon versions of the noweb pipeline.
 # Set LIBSRC=icon
 LIBSRC=awk
 # If you have no Icon compiler, but do have icont, make ICONC=icont
-ICONC=iconc
+ICONC=icont
 ICONT=icont
 
 # BIN is where the commands (notangle, noweave, nountangle, noroots) land
@@ -18,12 +18,12 @@
 # MAN7EXT is the extension for the nowebstyle man page (usually 7)
 # TEXINPUTS is the directory for TeX macro files
 # ELISP is the directory for emacs lisp files, or /dev/null not to install
-BIN=/usr/local/noweb
-LIB=/usr/local/noweb/lib
-MAN=/usr/local/noweb/man
+BIN=/usr/bin
+LIB=/usr/lib
+MAN=/usr/man
 MANEXT=1
 MAN7EXT=7
-TEXINPUTS=/usr/local/tex/inputs
+TEXINPUTS=/usr/share/texmf/tex/inputs
 ELISP=/dev/null
 
 # change WEAVE if you want a different version of noweave to be installed
@@ -53,79 +53,79 @@
 	/bin/cp $(HOME)/www/noweb/FAQ.html FAQ.html
 	chmod -w FAQ.html
 
-install: install-code install-man install-tex install-elisp
+install: install-code install-gzipped-man install-tex install-elisp
 
 install-shell:
-	-mkdir $(BIN) $(LIB) 2>/dev/null
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/noweb > $(BIN)/noweb
-	chmod +x $(BIN)/noweb
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/notangle > $(BIN)/notangle
-	chmod +x $(BIN)/notangle
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/$(WEAVE) > $(BIN)/$(WEAVE)
-	chmod +x $(BIN)/$(WEAVE)
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/nountangle > $(BIN)/nountangle
-	chmod +x $(BIN)/nountangle
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/nodefs > $(BIN)/nodefs
-	chmod +x $(BIN)/nodefs
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/noroots > $(BIN)/noroots
-	chmod +x $(BIN)/noroots
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/nuweb2noweb > $(BIN)/nuweb2noweb
-	chmod +x $(BIN)/nuweb2noweb
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/cpif > $(BIN)/cpif
-	chmod +x $(BIN)/cpif
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/htmltoc > $(BIN)/htmltoc
-	chmod +x $(BIN)/htmltoc
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/noroff > $(BIN)/noroff
-	chmod +x $(BIN)/noroff
-	sed "s@|LIBDIR|@$(LIBNAME)@" shell/toroff > $(LIB)/toroff
-	chmod +x $(LIB)/toroff
-	cp shell/tmac.w $(LIB)
+	-mkdir -p ${DESTDIR}/$(BIN) ${DESTDIR}/$(LIB) 2>/dev/null
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/noweb > ${DESTDIR}/$(BIN)/noweb
+	chmod +x ${DESTDIR}/$(BIN)/noweb
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/notangle > ${DESTDIR}/$(BIN)/notangle
+	chmod +x ${DESTDIR}/$(BIN)/notangle
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/$(WEAVE) > ${DESTDIR}/$(BIN)/$(WEAVE)
+	chmod +x ${DESTDIR}/$(BIN)/$(WEAVE)
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/nountangle > ${DESTDIR}/$(BIN)/nountangle
+	chmod +x ${DESTDIR}/$(BIN)/nountangle
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/nodefs > ${DESTDIR}/$(BIN)/nodefs
+	chmod +x ${DESTDIR}/$(BIN)/nodefs
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/noroots > ${DESTDIR}/$(BIN)/noroots
+	chmod +x ${DESTDIR}/$(BIN)/noroots
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/nuweb2noweb > ${DESTDIR}/$(BIN)/nuweb2noweb
+	chmod +x ${DESTDIR}/$(BIN)/nuweb2noweb
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/cpif > ${DESTDIR}/$(BIN)/cpif
+	chmod +x ${DESTDIR}/$(BIN)/cpif
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/htmltoc > ${DESTDIR}/$(BIN)/htmltoc
+	chmod +x ${DESTDIR}/$(BIN)/htmltoc
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/noroff > ${DESTDIR}/$(BIN)/noroff
+	chmod +x ${DESTDIR}/$(BIN)/noroff
+	sed "s@|LIBDIR|@$(LIBNAME)@" shell/toroff > ${DESTDIR}/$(LIB)/toroff
+	chmod +x ${DESTDIR}/$(LIB)/toroff
+	cp shell/tmac.w ${DESTDIR}/$(LIB)
 
 install-code: install-shell
-	-mkdir $(BIN) $(LIB) 2>/dev/null
+	-mkdir -p ${DESTDIR}/$(BIN) ${DESTDIR}/$(LIB) 2>/dev/null
 	strip c/nt c/markup c/mnt c/finduses
-	cp c/nt c/markup c/mnt c/finduses $(LIB)
-	cd $(LIBSRC); make ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) install
-	cd lib; make LIB=$(LIB) install
+	cp c/nt c/markup c/mnt c/finduses ${DESTDIR}/$(LIB)
+	cd $(LIBSRC); make ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) DESTDIR=$(DESTDIR) install
+	cd lib; make LIB=$(LIB) DESTDIR=$(DESTDIR) install
 install-man:
-	-mkdir $(MAN) $(MANDIR) $(MAN7DIR) 2>/dev/null
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/cpif.1 > $(MANDIR)/cpif.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nodefs.1 > $(MANDIR)/nodefs.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noroots.1 > $(MANDIR)/noroots.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noweb.1 > $(MANDIR)/noweb.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noindex.1 > $(MANDIR)/noindex.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nuweb2noweb.1 > $(MANDIR)/nuweb2noweb.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/notangle.1 > $(MANDIR)/notangle.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noroff.1 > $(MANDIR)/noroff.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/sl2h.1 > $(MANDIR)/sl2h.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/htmltoc.1 > $(MANDIR)/htmltoc.$(MANEXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nowebstyle.7 > $(MAN7DIR)/nowebstyle.$(MAN7EXT)
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nowebfilters.7 > $(MAN7DIR)/nowebfilters.$(MAN7EXT)
-	rm -f $(MANDIR)/noweave.$(MANEXT)
-	(cd $(MANDIR); ln notangle.$(MANEXT) noweave.$(MANEXT))
-	rm -f $(MANDIR)/nountangle.$(MANEXT)
-	(cd $(MANDIR); ln notangle.$(MANEXT) nountangle.$(MANEXT))
+	-mkdir -p ${DESTDIR}/$(MAN) ${DESTDIR}/$(MANDIR) ${DESTDIR}/$(MAN7DIR) 2>/dev/null
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/cpif.1 > ${DESTDIR}/$(MANDIR)/cpif.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nodefs.1 > ${DESTDIR}/$(MANDIR)/nodefs.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noroots.1 > ${DESTDIR}/$(MANDIR)/noroots.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noweb.1 > ${DESTDIR}/$(MANDIR)/noweb.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noindex.1 > ${DESTDIR}/$(MANDIR)/noindex.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nuweb2noweb.1 > ${DESTDIR}/$(MANDIR)/nuweb2noweb.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/notangle.1 > ${DESTDIR}/$(MANDIR)/notangle.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noroff.1 > ${DESTDIR}/$(MANDIR)/noroff.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/sl2h.1 > ${DESTDIR}/$(MANDIR)/sl2h.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/htmltoc.1 > ${DESTDIR}/$(MANDIR)/htmltoc.$(MANEXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nowebstyle.7 > ${DESTDIR}/$(MAN7DIR)/nowebstyle.$(MAN7EXT)
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nowebfilters.7 > ${DESTDIR}/$(MAN7DIR)/nowebfilters.$(MAN7EXT)
+	rm -f ${DESTDIR}/$(MANDIR)/noweave.$(MANEXT)
+	(cd ${DESTDIR}/$(MANDIR); ln notangle.$(MANEXT) noweave.$(MANEXT))
+	rm -f ${DESTDIR}/$(MANDIR)/nountangle.$(MANEXT)
+	(cd ${DESTDIR}/$(MANDIR); ln notangle.$(MANEXT) nountangle.$(MANEXT))
 install-gzipped-man:
-	-mkdir $(MAN) $(MANDIR) $(MAN7DIR) 2>/dev/null
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/cpif.1  | gzip -9 > $(MANDIR)/cpif.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nodefs.1  | gzip -9 > $(MANDIR)/nodefs.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noroots.1  | gzip -9 > $(MANDIR)/noroots.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noweb.1  | gzip -9 > $(MANDIR)/noweb.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noindex.1  | gzip -9 > $(MANDIR)/noindex.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nuweb2noweb.1  | gzip -9 > $(MANDIR)/nuweb2noweb.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/notangle.1  | gzip -9 > $(MANDIR)/notangle.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noroff.1  | gzip -9 > $(MANDIR)/noroff.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/sl2h.1  | gzip -9 > $(MANDIR)/sl2h.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/htmltoc.1  | gzip -9 > $(MANDIR)/htmltoc.$(MANEXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nowebstyle.7  | gzip -9 > $(MAN7DIR)/nowebstyle.$(MAN7EXT).gz
-	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nowebfilters.7  | gzip -9 > $(MAN7DIR)/nowebfilters.$(MAN7EXT).gz
-	rm -f $(MANDIR)/noweave.$(MANEXT).gz
-	(cd $(MANDIR); ln notangle.$(MANEXT).gz noweave.$(MANEXT).gz)
-	rm -f $(MANDIR)/nountangle.$(MANEXT).gz
-	(cd $(MANDIR); ln notangle.$(MANEXT).gz nountangle.$(MANEXT).gz)
+	-mkdir -p ${DESTDIR}/$(MAN) ${DESTDIR}/$(MANDIR) ${DESTDIR}/$(MAN7DIR) 2>/dev/null
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/cpif.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/cpif.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nodefs.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/nodefs.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noroots.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/noroots.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noweb.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/noweb.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noindex.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/noindex.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nuweb2noweb.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/nuweb2noweb.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/notangle.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/notangle.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noroff.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/noroff.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/sl2h.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/sl2h.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/htmltoc.1  | gzip -9 > ${DESTDIR}/$(MANDIR)/htmltoc.$(MANEXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nowebstyle.7  | gzip -9 > ${DESTDIR}/$(MAN7DIR)/nowebstyle.$(MAN7EXT).gz
+	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nowebfilters.7  | gzip -9 > ${DESTDIR}/$(MAN7DIR)/nowebfilters.$(MAN7EXT).gz
+	rm -f ${DESTDIR}/$(MANDIR)/noweave.$(MANEXT).gz
+	(cd ${DESTDIR}/$(MANDIR); ln notangle.$(MANEXT).gz noweave.$(MANEXT).gz)
+	rm -f ${DESTDIR}/$(MANDIR)/nountangle.$(MANEXT).gz
+	(cd ${DESTDIR}/$(MANDIR); ln notangle.$(MANEXT).gz nountangle.$(MANEXT).gz)
 install-preformat-man:
 	-echo "Warning: install-preformat-man is obsolete, even on Slackware systems" 1>&2
-	-mkdir $(MAN) $(CATDIR) $(CAT7DIR) 2>/dev/null
+	-mkdir -p $(MAN) $(CATDIR) $(CAT7DIR) 2>/dev/null
 	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/cpif.txt  | gzip > $(CATDIR)/cpif.$(MANEXT).gz
 	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/nodefs.txt  | gzip > $(CATDIR)/nodefs.$(MANEXT).gz
 	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/noroots.txt  | gzip > $(CATDIR)/noroots.$(MANEXT).gz
@@ -143,13 +143,14 @@
 	rm -f $(CATDIR)/nountangle.$(MANEXT).gz
 	(cd $(CATDIR); ln notangle.$(MANEXT).gz nountangle.$(MANEXT).gz)
 install-tex:
-	-mkdir $(TEXINPUTS) 2>/dev/null
-	cp tex/nwmac.tex tex/noweb.sty $(TEXINPUTS)
-	-texhash || echo "Program texhash not found or failed"
+	-mkdir -p ${DESTDIR}/$(TEXINPUTS) 2>/dev/null
+	cp tex/nwmac.tex tex/noweb.sty ${DESTDIR}/$(TEXINPUTS)
+	#-texhash || echo "Program texhash not found or failed"
 
 install-elisp:
-	-mkdir $(ELISP) 2>/dev/null
-	cp elisp/noweb-mode.el $(ELISP)
+	# right now $ELISP is /dev/null so don't mess with it
+	#-mkdir -p $(DESTDIR)/$(ELISP) 2>/dev/null
+	#cp elisp/noweb-mode.el $(DESTDIR)$(ELISP)
 
 checkin:
 	for i in lib tex xdoc; do (cd $$i; make "CINAME=$(CINAME)" "CIMSG=$(CIMSG)" checkin); done
diff -ur src-new/awk/Makefile src/awk/Makefile
--- awk/Makefile	Mon Mar 27 02:00:00 2000
+++ awk/Makefile	Thu Mar 21 12:09:35 2002
@@ -16,8 +16,8 @@
 	touch $(EXECS)
 
 install: all
-	cp $(LIBEXECS) $(LIB)
-	cp $(BINEXECS) $(BIN)
+	cp $(LIBEXECS) $(DESTDIR)/$(LIB)
+	cp $(BINEXECS) $(DESTDIR)/$(BIN)
 
 source: $(EXECS)
 
diff -ur src-new/icon/Makefile src/icon/Makefile
--- icon/Makefile	Thu Sep 14 03:00:00 2000
+++ icon/Makefile	Thu Mar 21 12:08:58 2002
@@ -32,9 +32,9 @@
 	touch $(SRCS)
 
 install: $(EXECS)
-	cp $(LIBEXECS) $(LIB)
-	cp $(BINEXECS) $(BIN)
-	cp $(LIBSPECIAL) $(LIB)/autodefs.c
+	cp $(LIBEXECS) ${DESTDIR}/$(LIB)
+	cp $(BINEXECS) ${DESTDIR}/$(BIN)
+	cp $(LIBSPECIAL) ${DESTDIR}/$(LIB)/autodefs.c
 
 clean:
 	/bin/rm -f *.tex *.dvi *.aux *.log *.blg *.bbl *~  *.toc *.html *.u1 *.u2 
diff -ur src-new/lib/Makefile src/lib/Makefile
--- lib/Makefile	Fri Jun 23 13:56:00 2000
+++ lib/Makefile	Thu Mar 21 12:08:58 2002
@@ -11,9 +11,9 @@
 	chmod +x unmarkup emptydefn toascii nwmtime pipedocs h2a btdefn
 
 install: all
-	cp unmarkup emptydefn toascii nwmtime h2a btdefn $(LIB)
-	sed 's@|LIBDIR|@$(LIB)@g' pipedocs > $(LIB)/pipedocs
-	chmod +x $(LIB)/pipedocs
+	cp unmarkup emptydefn toascii nwmtime h2a btdefn ${DESTDIR}/$(LIB)
+	sed 's@|LIBDIR|@$(LIB)@g' pipedocs > ${DESTDIR}/$(LIB)/pipedocs
+	chmod +x ${DESTDIR}/$(LIB)/pipedocs
 
 checkin:
 	ci -l $(CINAME) $(CIMSG) $(RCSFILES)
