diff -r 1f4ec950e186 -r 408f51a850f4 tools/ioemu/hw/vga.c
--- a/tools/ioemu/hw/vga.c      Mon Apr 10 16:14:15 2006
+++ b/tools/ioemu/hw/vga.c      Mon Apr 10 16:14:36 2006
@@ -1369,10 +1369,10 @@
 {
     unsigned int eax, edx;
 
-    __asm__("cpuid"
+    __asm__("pushl %%ebx; cpuid; popl %%ebx"
             : "=a" (eax), "=d" (edx)
             : "0" (op)
-            : "bx", "cx");
+            : "cx");
 
     return edx;
 }
