--- cpuspeedy.py.old	2004-03-29 12:50:25.843441768 +0930
+++ cpuspeedy.py	2004-03-29 12:50:58.150530344 +0930
@@ -526,7 +526,7 @@
         cpufreq=pycpufreq()
     except ERROR_NO_INTERFACE,ERROR_INVALID_INTERFACE:
         # FIXME: hum this is dumb.
-        sys,stderr.write(PACKAGE+": "+"ERROR_NO_INTERFACE, ERROR_INVALID_INTERFACE"+"\n")
+        sys.stderr.write(PACKAGE+": "+"ERROR_NO_INTERFACE, ERROR_INVALID_INTERFACE"+"\n")
         sys.exit(EXIT_FAILURE)
         
     cpufreq.status()
