diff -ur ssrc-1.30.orig/Makefile ssrc-1.30/Makefile
--- ssrc-1.30.orig/Makefile	2005-06-28 11:10:07.000000000 +0300
+++ ssrc-1.30/Makefile	2009-06-19 17:03:01.000000000 +0300
@@ -1,14 +1,13 @@
-CC = gcc
-CFLAGS = -O2 -lm 
+CC ?= gcc
 #CFLAGS += -DBIGENDIAN
 
 all : ssrc ssrc_hp
 
 ssrc : ssrc.c fftsg_ld.c dbesi0.c
-	$(CC) $(CFLAGS) ssrc.c fftsg_ld.c dbesi0.c -o ssrc
+	$(CC) $(LDFLAGS) $(CFLAGS) ssrc.c fftsg_ld.c dbesi0.c -o ssrc -lm
 
 ssrc_hp : ssrc.c fftsg_ld.c dbesi0.c
-	$(CC) $(CFLAGS) -DHIGHPREC ssrc.c fftsg_ld.c dbesi0.c -o ssrc_hp
+	$(CC) $(LDFLAGS) $(CFLAGS) -DHIGHPREC ssrc.c fftsg_ld.c dbesi0.c -o ssrc_hp -lm
 
 clean :
-	rm -f ssrc ssrc_hp *.o *~
\ No newline at end of file
+	rm -f ssrc ssrc_hp *.o *~
diff -ur ssrc-1.30.orig/ssrc.c ssrc-1.30/ssrc.c
--- ssrc-1.30.orig/ssrc.c	2005-06-28 02:51:11.000000000 +0300
+++ ssrc-1.30/ssrc.c	2009-06-19 17:04:29.000000000 +0300
@@ -3,6 +3,7 @@
 #include <math.h>
 #include <assert.h>
 #include <time.h>
+#include <string.h>
 
 #define NDEBUG
 
