--- NVIDIA-Linux-x86-1.0-5336-pkg1/usr/include/GL/gl.h.old	2004-05-13 02:41:58.554622248 +1000
+++ NVIDIA-Linux-x86-1.0-5336-pkg1/usr/include/GL/gl.h	2004-05-13 02:42:17.148795504 +1000
@@ -1728,7 +1728,7 @@
 void GLAPI glFogCoordf (GLfloat fog);
 void GLAPI glFogCoordfv (const GLfloat *fog);
 void GLAPI glFogCoordPointer (GLenum type, GLsizei stride, const GLvoid *pointer);
-void GLAPI glMultiDrawArrays (GLenum mode, const GLint *first, const GLsizei *count, GLsizei primcount);
+void GLAPI glMultiDrawArrays (GLenum mode, GLint *first, GLsizei *count, GLsizei primcount);
 void GLAPI glMultiDrawElements (GLenum mode, const GLsizei *count, GLenum type, const GLvoid* *indices, GLsizei primcount);
 void GLAPI glSecondaryColor3b (GLbyte red, GLbyte green, GLbyte blue);
 void GLAPI glSecondaryColor3bv (const GLbyte *v);
