--- Makefile	2003-05-27 03:46:06.000000000 +0900
+++ Makefile.new	2003-05-27 03:48:27.000000000 +0900
@@ -154,12 +154,12 @@
 # from  top Makefile
   BOOTSTRAPCFLAGS =
 
-               CC = gcc -m32
+               CC = g++ -m32
                AS = gcc -m32 -c -x assembler
 
 .SUFFIXES: .cc
 
-              CXX = c++ -m32
+              CXX = g++ -m32
 
           CXXFILT = c++filt
            CXXLIB =
@@ -958,13 +958,6 @@
 	fi
 	$(INSTALL) $(INSTALLFLAGS) $(INSTPGMFLAGS)  wmsmixer $(DESTDIR)$(BINDIR)/wmsmixer
 
-all:: wmsmixer.$(MANNEWSUFFIX)
-
-wmsmixer.$(MANNEWSUFFIX): wmsmixer.$(MANSRCSUFFIX)
-	$(RM) $@
-	cd `dirname wmsmixer` && \
-	$(LN) `basename wmsmixer.$(MANSRCSUFFIX)` `basename $@`
-
 cleandir::
 	$(RM) wmsmixer.$(MANNEWSUFFIX)
