diff -urN cyrus-sasl-1.5.27.old/configure.in cyrus-sasl-1.5.27/configure.in
--- cyrus-sasl-1.5.27.old/configure.in	2002-12-06 05:13:39.000000000 -0600
+++ cyrus-sasl-1.5.27/configure.in	2002-12-06 05:24:43.000000000 -0600
@@ -507,7 +507,7 @@
   if test "$gss_impl" = mit; then
      GSSAPIBASE_LIBS="$GSSAPIBASE_LIBS/lib -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err"
   elif test "$gss_impl" = "heimdal"; then
-     GSSAPIBASE_LIBS="$GSSAPIBASE_LIBS/lib -lgssapi -lkrb5 -ldes -lasn1 -lroken ${LIB_CRYPT} -lcom_err"
+     GSSAPIBASE_LIBS="$GSSAPIBASE_LIBS/lib -lgssapi"
   else
      gssapi="no"
      AC_WARN(Disabling GSSAPI)
