--- mingw-runtime-3.3/mingwex/Makefile.in.jj	2004-09-09 17:35:05.120782584 -0700
+++ mingw-runtime-3.3/mingwex/Makefile.in	2004-09-09 17:35:31.553764160 -0700
@@ -85,8 +85,8 @@
 INCLUDES = -I$(srcdir) -I$(srcdir)/../include \
   -nostdinc -nostdinc++ \
   -iwithprefixbefore include
-ALL_CFLAGS = $(CFLAGS) $(OPTFLAGS) $(W32API_INCLUDE) $(INCLUDES) $(MNO_CYGWIN)
-ALL_CXXFLAGS = $(CXXFLAGS) $(OPTFLAGS) $(W32API_INCLUDE) $(INCLUDES) $(MNO_CYGWIN)
+ALL_CFLAGS = $(CFLAGS) $(OPTFLAGS) $(INCLUDES) $(W32API_INCLUDE) $(MNO_CYGWIN)
+ALL_CXXFLAGS = $(CXXFLAGS) $(OPTFLAGS) $(INCLUDES) $(W32API_INCLUDE) $(MNO_CYGWIN)
 
 AS = @AS@
 AR = @AR@
--- mingw-runtime-3.3/profile/Makefile.in.jj	2004-09-09 17:35:36.718978928 -0700
+++ mingw-runtime-3.3/profile/Makefile.in	2004-09-09 17:36:02.584046840 -0700
@@ -51,8 +51,8 @@
 INCLUDES = -I$(srcdir) -I$(srcdir)/../include \
   -nostdinc -nostdinc++ \
   -iwithprefixbefore include
-ALL_CFLAGS = $(CFLAGS) $(W32API_INCLUDE) $(INCLUDES) $(MNO_CYGWIN)
-ALL_CXXFLAGS = $(CXXFLAGS) $(W32API_INCLUDE) $(INCLUDES) $(MNO_CYGWIN)
+ALL_CFLAGS = $(CFLAGS) $(INCLUDES) $(W32API_INCLUDE) $(MNO_CYGWIN)
+ALL_CXXFLAGS = $(CXXFLAGS) $(INCLUDES) $(W32API_INCLUDE) $(MNO_CYGWIN)
 
 AS = @AS@
 AR = @AR@
