--- Source/plugin-ui.cpp.bak	2008-09-18 22:32:44.000000000 -0400
+++ Source/plugin-ui.cpp	2008-09-18 22:32:51.000000000 -0400
@@ -283,11 +283,6 @@
     int term, chop;
     int PercentRedraw;
 
-    if (0) {
-	printf("Widget: %p\n", w);
-	printf("Message: %s\n", message);
-    }
-
     if (DEBUG > 1)
 	printf("state = %i, height = %i width = %i\n", instance->state,
 	       instance->movie_height, instance->movie_width);
