--- galeon-2.0.4/po/POTFILES.skip.1	2007-12-18 19:06:03.000000000 +0100
+++ galeon-2.0.4/po/POTFILES.skip	2007-12-18 19:06:26.000000000 +0100
@@ -5,3 +5,16 @@
 libegg/dock/egg-dock-placeholder.c
 libegg/dock/egg-dock-tablabel.c
 libegg/dock/egg-dock.c
+bookmarks-editor-small.glade
+bookmarks-editor.glade
+bookmarks.glade
+certificate-dialogs.glade
+galeon-bookmarks-editor-ui.xml
+galeon-bookmarks-editor-ui.xml.in
+galeon.glade
+nautilus-galeon-view-ui.xml
+nautilus-galeon-view-ui.xml.in
+prefs-dialog.glade
+print.glade
+ui/galeon-bookmarks-editor-ui.xml
+ui/nautilus-galeon-view-ui.xml
