diff -urb src-orig/metadata/ape/taglib_monkeysaudiofiletyperesolver.cpp src/metadata/ape/taglib_monkeysaudiofiletyperesolver.cpp
--- src-orig/metadata/ape/taglib_monkeysaudiofiletyperesolver.cpp	2008-04-23 19:46:24.000000000 +0200
+++ src/metadata/ape/taglib_monkeysaudiofiletyperesolver.cpp	2008-04-23 19:46:35.000000000 +0200
@@ -4,6 +4,7 @@
 
 #include "taglib_monkeysaudiofiletyperesolver.h"
 #include <taglib/mpcfile.h>
+#include <cstring>
 
 TagLib::File *MonkeysAudioFileTypeResolver::createFile(const char *fileName,
         bool readProperties,
diff -urb src-orig/metadata/optimfrog/taglib_optimfrogfiletyperesolver.cpp src/metadata/optimfrog/taglib_optimfrogfiletyperesolver.cpp
--- src-orig/metadata/optimfrog/taglib_optimfrogfiletyperesolver.cpp	2008-04-23 19:46:24.000000000 +0200
+++ src/metadata/optimfrog/taglib_optimfrogfiletyperesolver.cpp	2008-04-23 19:47:08.000000000 +0200
@@ -4,6 +4,7 @@
 
 #include "taglib_optimfrogfiletyperesolver.h"
 #include <taglib/mpcfile.h>
+#include <cstring>
 
 TagLib::File *OptimFrogFileTypeResolver::createFile(const char *fileName,
         bool readProperties,
