diff -ur soundkonverter-0.3.9.orig/src/soundkonverter.desktop soundkonverter-0.3.9/src/soundkonverter.desktop
--- soundkonverter-0.3.9.orig/src/soundkonverter.desktop	2008-03-02 12:34:32.000000000 +0200
+++ soundkonverter-0.3.9/src/soundkonverter.desktop	2009-05-13 21:26:17.000000000 +0300
@@ -1,6 +1,5 @@
 [Desktop Entry]
-Categories=Qt;KDE;AudioVideo;AudioVideoEditing
-Encoding=UTF-8
+Categories=Qt;KDE;AudioVideo;AudioVideoEditing;
 Name=soundKonverter
 Exec=soundkonverter
 Icon=soundkonverter
