diff -ru netboot-0.10.1/bootrom/Makefile netboot-0.10.1-installfixed/bootrom/Makefile
--- netboot-0.10.1/bootrom/Makefile	2006-01-09 00:24:34.000000000 +1100
+++ netboot-0.10.1-installfixed/bootrom/Makefile	2007-01-29 13:56:16.000000000 +1100
@@ -56,21 +56,21 @@
 # Install all files in place
 instbin:
 	@echo Make all necessary directories
-	$(INSTALL) -d $(nbbindir)
-	$(INSTALL) -d $(nbmiscdir)
+	$(INSTALL) -d $(DESTDIR)$(nbbindir)
+	$(INSTALL) -d $(DESTDIR)$(nbmiscdir)
 	@echo Remove files of old versions
-	@if test -d $(nblibdir)/binaries; then			\
-	  rm -f $(nblibdir)/binaries/*86.bin;			\
-	  rm -f $(nblibdir)/binaries/kernelm.bin;		\
+	@if test -d $(DESTDIR)$(nblibdir)/binaries; then			\
+	  rm -f $(DESTDIR)$(nblibdir)/binaries/*86.bin;			\
+	  rm -f $(DESTDIR)$(nblibdir)/binaries/kernelm.bin;		\
 	fi
 	@echo Install new binaries
 	@for i in $(bindestdir)/*.bin; do			\
-	  echo $(INSTALL_DATA) $$i $(nbbindir);			\
-	  $(INSTALL_DATA) $$i $(nbbindir);			\
+	  echo $(INSTALL_DATA) $$i $(DESTDIR)$(nbbindir);			\
+	  $(INSTALL_DATA) $$i $(DESTDIR)$(nbbindir);			\
 	done
 	@for i in $(bindestdir)/*.com; do			\
-	  echo $(INSTALL_DATA) $$i $(nbmiscdir);		\
-	  $(INSTALL_DATA) $$i $(nbmiscdir);			\
+	  echo $(INSTALL_DATA) $$i $(DESTDIR)$(nbmiscdir);		\
+	  $(INSTALL_DATA) $$i $(DESTDIR)$(nbmiscdir);			\
 	done
 
 install:	install-recursive instbin
diff -ru netboot-0.10.1/bootrom/netdrvr/Makefile netboot-0.10.1-installfixed/bootrom/netdrvr/Makefile
--- netboot-0.10.1/bootrom/netdrvr/Makefile	2006-07-07 06:54:57.000000000 +1000
+++ netboot-0.10.1-installfixed/bootrom/netdrvr/Makefile	2007-01-29 13:20:51.000000000 +1100
@@ -55,9 +55,10 @@
 netdrvr-nognu:
 
 netdrvr-inst:	$(TARGETS)
+		$(INSTALL) -d $(DESTDIR)$(bindestdir)
 		@for i in $^; do				\
 		  j=`basename $$i`;				\
-		  $(CPATTR) $$i $(bindestdir)/$$j;		\
+		  $(CPATTR) $$i $(DESTDIR)$(bindestdir)/$$j;		\
 		done
 
 
diff -ru netboot-0.10.1/bootrom/netdrvr/ndis2/drivers/Makefile netboot-0.10.1-installfixed/bootrom/netdrvr/ndis2/drivers/Makefile
--- netboot-0.10.1/bootrom/netdrvr/ndis2/drivers/Makefile	2006-01-09 00:24:38.000000000 +1100
+++ netboot-0.10.1-installfixed/bootrom/netdrvr/ndis2/drivers/Makefile	2007-01-29 13:53:39.000000000 +1100
@@ -53,7 +53,7 @@
 $(MD5SUMS):	$(DRIVERS) | $(INSTDRV) $(MD5PROG)
 		@MD5PROG="$(MD5PROGEXEC)" INSTALL="$(INSTALL)" $(INSTDRV) -u \
 			-t $(MODSUBDIR) -s drivers -v \
-			-n $(ROMDIR)/netdrvr -m $(MD5SUMS) \
+			-n $(DESTDIR)$(ROMDIR)/netdrvr -m $(MD5SUMS) \
 			$(foreach fname,$(DRIVERS),-d $(fname))
 
 
@@ -64,12 +64,12 @@
 
 # Installation
 install:	$(DRIVERS) | $(INSTDRV) $(MD5PROG)
-	$(INSTALL) -d $(nbnetdrvdir)
-	$(INSTALL) -d $(nbnetdrvdir)/$(MODSUBDIR)
+	$(INSTALL) -d $(DESTDIR)$(nbnetdrvdir)
+	$(INSTALL) -d $(DESTDIR)$(nbnetdrvdir)/$(MODSUBDIR)
 	@echo Installing network drivers
 	@MD5PROG="$(MD5PROGEXEC)" INSTALL="$(INSTALL)" $(INSTDRV) \
 			-t $(MODSUBDIR) -v \
-			-n $(nbnetdrvdir) -m $(nbnetdrvdir)/MD5SUMS \
+			-n $(DESTDIR)$(nbnetdrvdir) -m $(nbnetdrvdir)/MD5SUMS \
 			$(foreach fname,$(DRIVERS),-d $(fname))
 
 
diff -ru netboot-0.10.1/bootrom/netdrvr/pktdrvr/drivers/Makefile netboot-0.10.1-installfixed/bootrom/netdrvr/pktdrvr/drivers/Makefile
--- netboot-0.10.1/bootrom/netdrvr/pktdrvr/drivers/Makefile	2006-01-09 00:24:38.000000000 +1100
+++ netboot-0.10.1-installfixed/bootrom/netdrvr/pktdrvr/drivers/Makefile	2007-01-29 13:48:10.000000000 +1100
@@ -53,7 +53,7 @@
 $(MD5SUMS):	$(DRIVERS) | $(INSTDRV) $(MD5PROG)
 		@MD5PROG="$(MD5PROGEXEC)" INSTALL="$(INSTALL)" $(INSTDRV) -u \
 			-t $(MODSUBDIR) -s drivers -v \
-			-n $(ROMDIR)/netdrvr -m $(MD5SUMS) \
+			-n $(DESTDIR)$(ROMDIR)/netdrvr -m $(MD5SUMS) \
 			$(foreach fname,$(DRIVERS),-d $(fname))
 
 
@@ -64,12 +64,12 @@
 
 # Installation
 install:	$(DRIVERS) | $(INSTDRV) $(MD5PROG)
-	$(INSTALL) -d $(nbnetdrvdir)
-	$(INSTALL) -d $(nbnetdrvdir)/$(MODSUBDIR)
+	$(INSTALL) -d $(DESTDIR)$(nbnetdrvdir)
+	$(INSTALL) -d $(DESTDIR)$(nbnetdrvdir)/$(MODSUBDIR)
 	@echo Installing network drivers
 	@MD5PROG="$(MD5PROGEXEC)" INSTALL="$(INSTALL)" $(INSTDRV) \
 			-t $(MODSUBDIR) -v \
-			-n $(nbnetdrvdir) -m $(nbnetdrvdir)/MD5SUMS \
+			-n $(DESTDIR)$(nbnetdrvdir) -m $(DESTDIR)$(nbnetdrvdir)/MD5SUMS \
 			$(foreach fname,$(DRIVERS),-d $(fname))
 
 
diff -ru netboot-0.10.1/bootrom/netdrvr/undi/drivers/Makefile netboot-0.10.1-installfixed/bootrom/netdrvr/undi/drivers/Makefile
--- netboot-0.10.1/bootrom/netdrvr/undi/drivers/Makefile	2006-01-09 00:24:39.000000000 +1100
+++ netboot-0.10.1-installfixed/bootrom/netdrvr/undi/drivers/Makefile	2007-01-29 13:54:34.000000000 +1100
@@ -53,7 +53,7 @@
 $(MD5SUMS):	$(DRIVERS) | $(INSTDRV) $(MD5PROG)
 		@MD5PROG="$(MD5PROGEXEC)" INSTALL="$(INSTALL)" $(INSTDRV) -u \
 			-t $(MODSUBDIR) -s drivers -v \
-			-n $(ROMDIR)/netdrvr -m $(MD5SUMS) \
+			-n $(DESTDIR)$(ROMDIR)/netdrvr -m $(MD5SUMS) \
 			$(foreach fname,$(DRIVERS),-d $(fname))
 
 
@@ -64,12 +64,12 @@
 
 # Installation
 install:	$(DRIVERS) | $(INSTDRV) $(MD5PROG)
-	$(INSTALL) -d $(nbnetdrvdir)
-	$(INSTALL) -d $(nbnetdrvdir)/$(MODSUBDIR)
+	$(INSTALL) -d $(DESTDIR)$(nbnetdrvdir)
+	$(INSTALL) -d $(DESTDIR)$(nbnetdrvdir)/$(MODSUBDIR)
 	@echo Installing network drivers
 	@MD5PROG="$(MD5PROGEXEC)" INSTALL="$(INSTALL)" $(INSTDRV) \
 			-t $(MODSUBDIR) -v \
-			-n $(nbnetdrvdir) -m $(nbnetdrvdir)/MD5SUMS \
+			-n $(DESTDIR)$(nbnetdrvdir) -m $(nbnetdrvdir)/MD5SUMS \
 			$(foreach fname,$(DRIVERS),-d $(fname))
 
 
diff -ru netboot-0.10.1/include/Makefile netboot-0.10.1-installfixed/include/Makefile
--- netboot-0.10.1/include/Makefile	2006-01-09 00:24:40.000000000 +1100
+++ netboot-0.10.1-installfixed/include/Makefile	2007-01-29 13:20:51.000000000 +1100
@@ -30,8 +30,8 @@
 
 
 install:	nblib.h
-		$(INSTALL) -d $(includedir)
-		$(INSTALL_DATA) nblib.h $(includedir)/nblib.h
+		$(INSTALL) -d $(DESTDIR)$(includedir)
+		$(INSTALL_DATA) nblib.h $(DESTDIR)$(includedir)/nblib.h
 
 
 clean:
diff -ru netboot-0.10.1/makerom/Makefile netboot-0.10.1-installfixed/makerom/Makefile
--- netboot-0.10.1/makerom/Makefile	2006-07-07 06:54:57.000000000 +1000
+++ netboot-0.10.1-installfixed/makerom/Makefile	2007-01-29 13:20:51.000000000 +1100
@@ -104,13 +104,13 @@
 distrib:	stamp-flash-gnu
 
 install:	$(LIBTOOL) all
-		$(INSTALL) -d $(bindir)
-		$(INSTALL) -d $(mandir)/man$(MANEXT)
-		$(INSTALL) -d $(nbdatadir)
+		$(INSTALL) -d $(DESTDIR)$(bindir)
+		$(INSTALL) -d $(DESTDIR)$(mandir)/man$(MANEXT)
+		$(INSTALL) -d $(DESTDIR)$(nbdatadir)
 		@$(LIBTOOL) --mode=install \
-		  $(INSTALL_PROGRAM) makerom $(bindir)/$(TARGET)
-		$(INSTALL_DATA) makerom.man $(mandir)/man$(MANEXT)/$(TARGET).$(MANEXT)
-		$(INSTALL_DATA) makerom.help $(nbdatadir)/makerom.help
+		  $(INSTALL_PROGRAM) makerom $(DESTDIR)$(bindir)/$(TARGET)
+		$(INSTALL_DATA) makerom.man $(DESTDIR)$(mandir)/man$(MANEXT)/$(TARGET).$(MANEXT)
+		$(INSTALL_DATA) makerom.help $(DESTDIR)$(nbdatadir)/makerom.help
 
 
 clean:		clean-local clean-recursive
diff -ru netboot-0.10.1/misc/Makefile netboot-0.10.1-installfixed/misc/Makefile
--- netboot-0.10.1/misc/Makefile	2006-01-09 00:24:48.000000000 +1100
+++ netboot-0.10.1-installfixed/misc/Makefile	2007-01-29 13:20:50.000000000 +1100
@@ -130,27 +130,27 @@
 
 
 install-config:		$(nbconfig)
-	$(INSTALL) -d $(sysconfdir)
-	$(INSTALL_DATA) $(nbconfig) $(sysconfdir)/$(nbconfig)
+	$(INSTALL) -d $(DESTDIR)$(sysconfdir)
+	$(INSTALL_DATA) $(nbconfig) $(DESTDIR)$(sysconfdir)/$(nbconfig)
 
 
 install-drivers:	$(drivers)
-	$(INSTALL) -d $(nblibdir)
-	$(INSTALL_DATA) $(drivers) $(nblibdir)/$(drivers)
+	$(INSTALL) -d $(DESTDIR)$(nblibdir)
+	$(INSTALL_DATA) $(drivers) $(DESTDIR)$(nblibdir)/$(drivers)
 
 
 install-man:		netboot.man $(nbconfig).man $(drivers).man $(db).man
-	$(INSTALL) -d $(mandir)/man$(MANEXT)
-	$(INSTALL_DATA) netboot.man $(mandir)/man$(MANEXT)/netboot.$(MANEXT)
-	$(INSTALL_DATA) $(nbconfig).man $(mandir)/man$(MANEXT)/$(nbconfig).$(MANEXT)
-	$(INSTALL_DATA) $(drivers).man $(mandir)/man$(MANEXT)/$(drivers).$(MANEXT)
-	$(INSTALL_DATA) $(db).man $(mandir)/man$(MANEXT)/$(db).$(MANEXT)
+	$(INSTALL) -d $(DESTDIR)$(mandir)/man$(MANEXT)
+	$(INSTALL_DATA) netboot.man $(DESTDIR)$(mandir)/man$(MANEXT)/netboot.$(MANEXT)
+	$(INSTALL_DATA) $(nbconfig).man $(DESTDIR)$(mandir)/man$(MANEXT)/$(nbconfig).$(MANEXT)
+	$(INSTALL_DATA) $(drivers).man $(DESTDIR)$(mandir)/man$(MANEXT)/$(drivers).$(MANEXT)
+	$(INSTALL_DATA) $(db).man $(DESTDIR)$(mandir)/man$(MANEXT)/$(db).$(MANEXT)
 
 
 install-utils:		instdrv nbmd5
-	$(INSTALL) -d $(nbutildir)
-	$(INSTALL_SCRIPT) instdrv $(nbutildir)/instdrv
-	@$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) nbmd5 $(nbutildir)/nbmd5
+	$(INSTALL) -d $(DESTDIR)$(nbutildir)
+	$(INSTALL_SCRIPT) instdrv $(DESTDIR)$(nbutildir)/instdrv
+	@$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) nbmd5 $(DESTDIR)$(nbutildir)/nbmd5
 
 
 clean:	clean-general clean-recursive
diff -ru netboot-0.10.1/misc/instdrv.in netboot-0.10.1-installfixed/misc/instdrv.in
--- netboot-0.10.1/misc/instdrv.in	2005-02-28 07:27:54.000000000 +1100
+++ netboot-0.10.1-installfixed/misc/instdrv.in	2007-01-29 13:49:45.000000000 +1100
@@ -497,21 +497,21 @@
 	# its final destination.
 
 	if [ "$DOINSTALL" = "TRUE" ]; then
-		DESTDIR="${NETDRVDIR}/${dname}"
+		DDIR="${NETDRVDIR}/${dname}"
 		if [ "$VERBOSE" -gt 0 ]; then
-			echo "Installing $sname into $DESTDIR"
+			echo "Installing $sname into $DDIR"
 		fi
-		if [ ! -d "$DESTDIR" ]; then
+		if [ ! -d "$DDIR" ]; then
 			if [ "$DEBUG" = "TRUE" ]; then
-				echo "$INSTPROG -d $DESTDIR"
+				echo "$INSTPROG -d $DDIR"
 			else
-				$INSTPROG -d $DESTDIR
+				$INSTPROG -d $DDIR
 			fi
 		fi
 		if [ "$DEBUG" = "TRUE" ]; then
-			echo "$INSTPROG -m 644 $sname $DESTDIR/$fname"
+			echo "$INSTPROG -m 644 $sname $DDIR/$fname"
 		else
-			$INSTPROG -m 644 $sname $DESTDIR/$fname
+			$INSTPROG -m 644 $sname $DDIR/$fname
 		fi
 	fi
 done <$TMPSRC 5>>$TMPMD5
diff -ru netboot-0.10.1/misc/romcheck/Makefile netboot-0.10.1-installfixed/misc/romcheck/Makefile
--- netboot-0.10.1/misc/romcheck/Makefile	2006-01-09 00:24:49.000000000 +1100
+++ netboot-0.10.1-installfixed/misc/romcheck/Makefile	2007-01-29 13:20:50.000000000 +1100
@@ -70,8 +70,8 @@
 
 # Install
 install:	romcheck-$(GNU16BIT)
-		$(INSTALL) -d $(nbmiscdir)
-		$(INSTALL_DATA) romcheck.bin $(nbmiscdir)/romcheck.bin
+		$(INSTALL) -d $(DESTDIR)$(nbmiscdir)
+		$(INSTALL_DATA) romcheck.bin $(DESTDIR)$(nbmiscdir)/romcheck.bin
 
 
 # Clean up
diff -ru netboot-0.10.1/mknbi-dos/Makefile netboot-0.10.1-installfixed/mknbi-dos/Makefile
--- netboot-0.10.1/mknbi-dos/Makefile	2006-07-07 06:54:57.000000000 +1000
+++ netboot-0.10.1-installfixed/mknbi-dos/Makefile	2007-01-29 13:20:51.000000000 +1100
@@ -133,11 +133,11 @@
 
 
 install:	$(LIBTOOL) all
-		$(INSTALL) -d $(bindir)
-		$(INSTALL) -d $(mandir)/man$(MANEXT)
+		$(INSTALL) -d $(DESTDIR)$(bindir)
+		$(INSTALL) -d $(DESTDIR)$(mandir)/man$(MANEXT)
 		@$(LIBTOOL) --mode=install \
-		  $(INSTALL_PROGRAM) mknbi $(bindir)/$(TARGET)
-		$(INSTALL_DATA) mknbi.man $(mandir)/man$(MANEXT)/$(TARGET).$(MANEXT)
+		  $(INSTALL_PROGRAM) mknbi $(DESTDIR)$(bindir)/$(TARGET)
+		$(INSTALL_DATA) mknbi.man $(DESTDIR)$(mandir)/man$(MANEXT)/$(TARGET).$(MANEXT)
 		@echo making install in utils
 		@(cd utils && $(MAKE) install) || exit 1
 
diff -ru netboot-0.10.1/mknbi-dos/utils/Makefile netboot-0.10.1-installfixed/mknbi-dos/utils/Makefile
--- netboot-0.10.1/mknbi-dos/utils/Makefile	2006-01-09 00:24:51.000000000 +1100
+++ netboot-0.10.1-installfixed/mknbi-dos/utils/Makefile	2007-01-29 13:20:51.000000000 +1100
@@ -79,11 +79,11 @@
 
 # Install
 install:	tools
-		$(INSTALL) -d $(nbmiscdir)
-		$(INSTALL) -d $(nbutildir)
-		$(INSTALL_DATA) rmrd.com $(nbmiscdir)/rmrd.com
-		$(INSTALL_DATA) nbramdrv.sys $(nbmiscdir)/nbramdrv.sys
-		$(INSTALL_SCRIPT) mntnbi.pl $(nbutildir)/mntnbi.pl
+		$(INSTALL) -d $(DESTDIR)$(nbmiscdir)
+		$(INSTALL) -d $(DESTDIR)$(nbutildir)
+		$(INSTALL_DATA) rmrd.com $(DESTDIR)$(nbmiscdir)/rmrd.com
+		$(INSTALL_DATA) nbramdrv.sys $(DESTDIR)$(nbmiscdir)/nbramdrv.sys
+		$(INSTALL_SCRIPT) mntnbi.pl $(DESTDIR)$(nbutildir)/mntnbi.pl
 
 
 # Clean up
diff -ru netboot-0.10.1/mknbi-linux/Makefile netboot-0.10.1-installfixed/mknbi-linux/Makefile
--- netboot-0.10.1/mknbi-linux/Makefile	2006-07-07 06:54:58.000000000 +1000
+++ netboot-0.10.1-installfixed/mknbi-linux/Makefile	2007-01-29 13:20:51.000000000 +1100
@@ -104,11 +104,11 @@
 distrib:	stamp-first-gnu stamp-firstd-gnu
 
 install:	$(LIBTOOL) all
-		$(INSTALL) -d $(bindir)
-		$(INSTALL) -d $(mandir)/man$(MANEXT)
+		$(INSTALL) -d $(DESTDIR)$(bindir)
+		$(INSTALL) -d $(DESTDIR)$(mandir)/man$(MANEXT)
 		@$(LIBTOOL) --mode=install \
-		  $(INSTALL_PROGRAM) mknbi $(bindir)/$(TARGET)
-		$(INSTALL_DATA) mknbi.man $(mandir)/man$(MANEXT)/$(TARGET).$(MANEXT)
+		  $(INSTALL_PROGRAM) mknbi $(DESTDIR)$(bindir)/$(TARGET)
+		$(INSTALL_DATA) mknbi.man $(DESTDIR)$(mandir)/man$(MANEXT)/$(TARGET).$(MANEXT)
 
 
 clean:			clean-local clean-recursive
diff -ru netboot-0.10.1/mknbi-mgl/Makefile netboot-0.10.1-installfixed/mknbi-mgl/Makefile
--- netboot-0.10.1/mknbi-mgl/Makefile	2006-07-07 06:54:58.000000000 +1000
+++ netboot-0.10.1-installfixed/mknbi-mgl/Makefile	2007-01-29 13:20:51.000000000 +1100
@@ -197,13 +197,13 @@
 		$(DISTSRCS)
 
 install:	$(LIBTOOL) all
-		$(INSTALL) -d $(bindir)
-		$(INSTALL) -d $(mandir)/man$(MANEXT)
+		$(INSTALL) -d $(DESTDIR)$(bindir)
+		$(INSTALL) -d $(DESTDIR)$(mandir)/man$(MANEXT)
 		@$(LIBTOOL) --mode=install \
-		  $(INSTALL_PROGRAM) mknbi $(bindir)/$(TARGET)
-		$(INSTALL_DATA) mknbi.man $(mandir)/man$(MANEXT)/$(TARGET).$(MANEXT)
-		$(INSTALL_DATA) misc/mgl.vim $(nbmiscdir)/mgl.vim
-		$(INSTALL_DATA) test/menu.mgl $(nbmiscdir)/menu.mgl
+		  $(INSTALL_PROGRAM) mknbi $(DESTDIR)$(bindir)/$(TARGET)
+		$(INSTALL_DATA) mknbi.man $(DESTDIR)$(mandir)/man$(MANEXT)/$(TARGET).$(MANEXT)
+		$(INSTALL_DATA) misc/mgl.vim $(DESTDIR)$(nbmiscdir)/mgl.vim
+		$(INSTALL_DATA) test/menu.mgl $(DESTDIR)$(nbmiscdir)/menu.mgl
 
 
 test:		mknbi
diff -ru netboot-0.10.1/nblib/Makefile netboot-0.10.1-installfixed/nblib/Makefile
--- netboot-0.10.1/nblib/Makefile	2006-04-18 00:06:57.000000000 +1000
+++ netboot-0.10.1-installfixed/nblib/Makefile	2007-01-29 13:20:51.000000000 +1100
@@ -88,9 +88,9 @@
 
 # Install library
 install:	$(LIBTOOL) $(TARGET)
-	$(INSTALL) -d $(libdir)
+	$(INSTALL) -d $(DESTDIR)$(libdir)
 	@$(LIBTOOL) --mode=install \
-	  $(INSTALL_PROGRAM) $(TARGET) $(libdir)/$(TARGET)
+	  $(INSTALL_PROGRAM) $(TARGET) $(DESTDIR)$(libdir)/$(TARGET)
 
 
 # Cleanup
