Index: muse-0.9pre2/muse/Makefile.in
===================================================================
--- muse-0.9pre2.orig/muse/Makefile.in
+++ muse-0.9pre2/muse/Makefile.in
@@ -390,11 +390,12 @@ muse_LDADD = \
 	cliplist/libcliplist.a \
       instruments/libinstruments.a \
 	widgets/libwidgets.a \
-      ../synti/libsynti/libsynti.la
-
-muse_LDFLAGS = -fno_exceptions $(X_LIBS) $(QT_LIBS) \
+      ../synti/libsynti/libsynti.la \
+	$(X_LIBS) $(QT_LIBS) \
 	-lasound $(SNDFILE_LIBS) $(JACK_LIBS) $(LASH_LIBS) $(FST_LIBS)
 
+muse_LDFLAGS = -fno_exceptions 
+
 CLEANFILES = stamp-chown
 DISTCLEANFILES = .muse.bin.bak
 MAINTAINERCLEANFILES = depcomp config.guess \
