diff -ruN multi-gnome-terminal-1.6.2.orig/gnome-terminal/vt.c multi-gnome-terminal-1.6.2/gnome-terminal/vt.c
--- multi-gnome-terminal-1.6.2.orig/gnome-terminal/vt.c	2004-09-03 08:44:36.521631688 +0200
+++ multi-gnome-terminal-1.6.2/gnome-terminal/vt.c	2004-09-03 08:52:39.949139536 +0200
@@ -782,6 +782,7 @@
       vt_up(vt);    
       break;
     default:
+      break;
     }
 
 }
@@ -806,6 +807,7 @@
       vt_down(vt);
       break;
     default:
+      break;
     }
 }
 
@@ -870,6 +872,7 @@
 	}
       break; 
     default:
+      break;
     }
 }
 
@@ -956,6 +959,7 @@
 	} 
       break;
     default:
+      break;
     }
     /*vt->cursorx = (vt->cursorx-1) & (~7);*/
 }
@@ -980,6 +984,7 @@
 	} 
      break; 
     default:
+     break;
     } 
 }
 
@@ -2073,9 +2078,11 @@
 */	      
 	break;
 	default:
+	break;
 	}
       break;
 	default:
+	break;
     } 
   /* do nothing!*/
 }
@@ -2102,6 +2109,7 @@
     break;
   default:
     vt_reset_terminal(vt, 0);
+    break;
   }
 }
 
@@ -2134,6 +2142,7 @@
       } else {
 	d(printf("Unknown function %d\n", i));
       }
+      break;
     }
   }
 }
@@ -2284,6 +2293,7 @@
     } else {
       d(printf("Unknown dtterm (CSI %d [; Ps [; Ps ]]t) sequence\n", cmd));
     }
+    break;
   }
 }
 
