--- asterisk-addons-1.2.3/asterisk-ooh323c/Makefile.am.orig	2006-04-30 17:08:32.000000000 +0200
+++ asterisk-addons-1.2.3/asterisk-ooh323c/Makefile.am	2006-06-05 15:19:59.000000000 +0200
@@ -22,10 +22,10 @@
 	$(MAKE) "CFLAGS = -g -DGNU -D_GNU_SOURCE -D_REENTRANT $(DEBUG_THREADS)" all
 
 install: 
-	cp .libs/libchan_h323.so.1.0.1 $(DESTDIR)/usr/lib/asterisk/modules/chan_ooh323.so
+	$(INSTALL) -D -m755 .libs/libchan_h323.so.1.0.1 $(DESTDIR)/usr/lib/asterisk/modules/chan_ooh323.so
 
 sample:
-	cp h323.conf.sample $(DESTDIR)/etc/asterisk/ooh323.conf
+	$(INSTALL) -D -m644  h323.conf.sample $(DESTDIR)/etc/asterisk/ooh323.conf
 
 
 EXTRA_DIST = extensions.conf h323.conf README COPYING
--- asterisk-addons-1.2.3/asterisk-ooh323c/Makefile.in.orig	2006-05-31 18:50:32.000000000 +0200
+++ asterisk-addons-1.2.3/asterisk-ooh323c/Makefile.in	2006-06-05 15:20:52.000000000 +0200
@@ -1239,10 +1239,10 @@
 	$(MAKE) "CFLAGS = -g -DGNU -D_GNU_SOURCE -D_REENTRANT $(DEBUG_THREADS)" all
 
 install: 
-	cp .libs/libchan_h323.so.1.0.1 $(DESTDIR)/usr/lib/asterisk/modules/chan_ooh323.so
+	$(INSTALL) -D -m755 .libs/libchan_h323.so.1.0.1 $(DESTDIR)/usr/lib/asterisk/modules/chan_ooh323.so
 
 sample:
-	cp h323.conf.sample $(DESTDIR)/etc/asterisk/ooh323.conf
+	$(INSTALL) -D -m644 h323.conf.sample $(DESTDIR)/etc/asterisk/ooh323.conf
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
