--- old/mailfilter.crm	2004-06-07 11:56:04.701302534 -0700
+++ new/mailfilter.crm	2004-06-07 11:57:49.645986881 -0700
@@ -108,6 +108,13 @@
         {
             match [:config:] /.+/
             input [:*:config:] (:option_txt:)
+            {
+                match [:config:] /^\/.+/
+            }
+            alius
+            {
+                input [:*:fileprefix::*:config:] (:option_txt:)
+            }
         }
         alius
         {
