diff -urN cyrus-imapd-2.1.11/configure cyrus-imapd-2.1.11-modified/configure
--- cyrus-imapd-2.1.11/configure	2002-12-04 10:18:28.000000000 -0600
+++ cyrus-imapd-2.1.11-modified/configure	2002-12-04 16:49:23.000000000 -0600
@@ -1493,10 +1493,10 @@
     CFLAGS="-Wall ${CFLAGS}"
   GCC_VERSION=`$CC -v 2>&1 | sed -n -e '/version/s/.*version \([0-9]\)\..*/\1/p'`
   if test "$GCC_VERSION" != "3"; then
-    CPPFLAGS="${CPPFLAGS} -I/usr/local/include"
+    CPPFLAGS="${CPPFLAGS}"
   fi
 else
-  CPPFLAGS="${CPPFLAGS} -I/usr/local/include"
+  CPPFLAGS="${CPPFLAGS}"
 fi 
 
   # this is CMU ADD LIBPATH
@@ -1962,7 +1962,7 @@
   if test -d $withval/include/db3; then
     CPPFLAGS="-I$withval/include/db3 $CPPFLAGS"
   else
-    CPPFLAGS="-I$withval/include $CPPFLAGS"
+    CPPFLAGS="$CPPFLAGS"
   fi
 else
   # otherwise, check for ../include/db3
@@ -3571,7 +3571,7 @@
 if test "${with_afs+set}" = set; then
   withval="$with_afs"
   with_afs="${withval}"
-	CFLAGS="${CFLAGS} -I${withval}/include"
+	CFLAGS="${CFLAGS}"
 else
   with_afs="/usr/local"
 fi
@@ -3783,14 +3783,14 @@
 	{ echo "configure: error: --enable-statickrb specified but --with-krb did not specify a valid directory" 1>&2; exit 1; }
   fi 
 
-  echo $ac_n "checking for des_ecb_encrypt in -ldes""... $ac_c" 1>&6
-echo "configure:3788: checking for des_ecb_encrypt in -ldes" >&5
+  echo $ac_n "checking for des_ecb_encrypt in -ldes425""... $ac_c" 1>&6
+echo "configure:3788: checking for des_ecb_encrypt in -ldes425" >&5
 ac_lib_var=`echo des'_'des_ecb_encrypt | sed 'y%./+-%__p_%'`
 if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-ldes  $LIBS"
+LIBS="-ldes425  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 3796 "configure"
 #include "confdefs.h"
@@ -3821,7 +3821,7 @@
   if test "$with_statickrb" = "yes"; then
 	    KRB_LIBS="$with_krb/lib/libdes.a"
 	else
-	    KRB_LIBS="-ldes"
+	    KRB_LIBS="-ldes425"
 	fi
 else
   echo "$ac_t""no" 1>&6
@@ -4091,8 +4091,8 @@
 
 		
 		;;
-	*)  OPENSSL_INC="-I${with_openssl}/include"
-	    OPENSSL_LIB="-L${with_openssl}/lib"
+	*)  OPENSSL_INC=""
+	    OPENSSL_LIB=""
 	    CPPFLAGS="${CPPFLAGS} ${OPENSSL_INC}"
 	    LDFLAGS="${LDFLAGS} ${OPENSSL_LIB}"
 	    LIBS="${LIBS} -lssl -lcrypto";;
@@ -4187,7 +4187,7 @@
        *)  if test -d ${with_zephyr}/include/zephyr; then
                ZEPHYR_CPPFLAGS="-I${with_zephyr}/include/zephyr"
            else
-               ZEPHYR_CPPFLAGS="-I${with_zephyr}/include"
+               ZEPHYR_CPPFLAGS=""
            fi
            ZEPHYR_LIBS="-lzephyr";;
 esac
@@ -4428,10 +4428,10 @@
   *) # use whatever they told us, or whatever we found
      COMPILE_ET="${with_com_err}/bin/compile_et"
      COM_ERR_LIBS="${with_com_err}/lib/libcom_err.a"
-     COM_ERR_CPPFLAGS="-I${with_com_err}/include"
+     COM_ERR_CPPFLAGS=""
      # Ever get the feeling people hide this stuff on purpose?
      if test -d "${with_com_err}/include/et" ; then
-       COM_ERR_CPPFLAGS="-I${with_com_err}/include"
+       COM_ERR_CPPFLAGS=""
      fi
      COMPILE_ET="${with_com_err}/bin/compile_et"
 esac
@@ -5035,7 +5035,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lgssapi $GSSAPIBASE_LIBS -lgssapi -lkrb5 -ldes -lasn1 -lroken ${LIB_CRYPT} -lcom_err $LIBS"
+LIBS="-lgssapi $GSSAPIBASE_LIBS -lgssapi -lkrb5 -ldes425 -lasn1 -lroken ${LIB_CRYPT} -lcom_err $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 5041 "configure"
 #include "confdefs.h"
@@ -5073,7 +5073,7 @@
      GSSAPIBASE_LIBS="$GSSAPIBASE_LIBS -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err"
      GSSAPIBASE_STATIC_LIBS="$GSSAPIBASE_LIBS $gssapi_dir/libgssapi_krb5.a $gssapi_dir/libkrb5.a $gssapi_dir/libk5crypto.a $gssapi_dir/libcom_err.a"
   elif test "$gss_impl" = "heimdal"; then
-     GSSAPIBASE_LIBS="$GSSAPIBASE_LIBS -lgssapi -lkrb5 -ldes -lasn1 -lroken ${LIB_CRYPT} -lcom_err"
+     GSSAPIBASE_LIBS="$GSSAPIBASE_LIBS -lgssapi -lkrb5 -ldes425 -lasn1 -lroken ${LIB_CRYPT} -lcom_err"
      GSSAPIBASE_STATIC_LIBS="$GSSAPIBASE_STATIC_LIBS $gssapi_dir/libgssapi.a $gssapi_dir/libkrb5.a $gssapi_dir/libdes.a $gssapi_dir/libasn1.a $gssapi_dir/libroken.a $gssapi_dir/libcom_err.a ${LIB_CRYPT}"
   else
      gssapi="no"
@@ -5235,7 +5235,7 @@
             ac_cv_sasl_where_lib=${with_sasl}/lib
             ac_cv_sasl_where_inc=${with_sasl}/include
 
-	    DYNSASLFLAGS="-I$ac_cv_sasl_where_inc"
+	    DYNSASLFLAGS=""
 	    if test "$ac_cv_sasl_where_lib" != ""; then
 		
   # this is CMU ADD LIBPATH TO
@@ -5247,7 +5247,7 @@
 
 	    fi
 	    LIB_DYN_SASL="$LIB_DYN_SASL -lsasl2"
-	    CPPFLAGS="${cmu_saved_CPPFLAGS} -I${ac_cv_sasl_where_inc}"
+	    CPPFLAGS="${cmu_saved_CPPFLAGS}"
 	    LDFLAGS="${cmu_saved_LDFLAGS} -L${ac_cv_sasl_where_lib}"
 	fi
 
@@ -5909,7 +5909,7 @@
 
   if test "$with_libwrap" != no; then
     if test -d "$with_libwrap"; then
-      CPPFLAGS="$CPPFLAGS -I${with_libwrap}/include"
+      CPPFLAGS="$CPPFLAGS"
       LDFLAGS="$LDFLAGS -L${with_libwrap}/lib"
     fi
     cmu_save_LIBS="$LIBS"
@@ -6055,7 +6055,7 @@
 
   if test "$with_ucdsnmp" != no; then
     if test -d "$with_ucdsnmp"; then
-      CPPFLAGS="$CPPFLAGS -I${with_ucdsnmp}/include"
+      CPPFLAGS="$CPPFLAGS"
       LDFLAGS="$LDFLAGS -L${with_ucdsnmp}/lib"
     fi
     cmu_save_LIBS="$LIBS"
@@ -6153,7 +6153,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lrpm -lpopt $LIBS"
+LIBS="-lrpm -lpopt -lrpmdb -lrpmio $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 6159 "configure"
 #include "confdefs.h"
@@ -6181,7 +6181,7 @@
 fi
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  LIB_UCDSNMP="${LIB_UCDSNMP} -lrpm -lpopt"
+  LIB_UCDSNMP="${LIB_UCDSNMP} -lrpm -lrpmdb -lrpmio -lpopt"
 else
   echo "$ac_t""no" 1>&6
 fi
diff -urN cyrus-imapd-2.1.11/master/master.c cyrus-imapd-2.1.11-modified/master/master.c
--- cyrus-imapd-2.1.11/master/master.c	2002-11-01 10:44:33.000000000 -0600
+++ cyrus-imapd-2.1.11-modified/master/master.c	2002-12-07 18:21:41.000000000 -0600
@@ -103,6 +103,10 @@
 static int verbose = 0;
 static int listen_queue_backlog = 32;
 
+int allow_severity = LOG_DEBUG;
+int deny_severity = LOG_ERR;
+int hosts_ctl;
+
 struct service *Services = NULL;
 int allocservices = 0;
 int nservices = 0;
