--- ./osasnmpd/Makefile.orig	2003-10-16 08:14:25.000000000 +0000
+++ ./osasnmpd/Makefile	2004-04-07 15:18:26.000000000 +0000
@@ -2,7 +2,7 @@
 include ../common.mak
 
 ifeq ($(strip $(shell which net-snmp-config 2>/dev/null)),)
-LIBS = -lucdagent -lucdmibs -lsnmp -lrpmdb -lrpm -lpopt -lrpmio -lwrap -lnsl -ldl -lcrypto
+LIBS = -lwrap -lcrypto -lucdagent -lucdmibs -lsnmp
 CFLAGS = -g -Wall
 else
 LIBS = `net-snmp-config --agent-libs` 
