Index: xorg/hw/kdrive/ephyr/Makefile.am
===================================================================
RCS file: /cvs/xorg/xserver/xorg/hw/kdrive/ephyr/Makefile.am,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -b -B -r1.7 -r1.8
--- xorg/hw/kdrive/ephyr/Makefile.am	30 Mar 2006 21:21:59 -0000	1.7
+++ xorg/hw/kdrive/ephyr/Makefile.am	17 Apr 2006 07:10:31 -0000	1.8
@@ -31,11 +31,11 @@
 Xephyr_LDADD = 						\
 	libxephyr.a					\
 	libxephyr-hostx.a			        \
+	../../../exa/libexa.la				\
 	@KDRIVE_LIBS@					\
 	@KDRIVE_LIBS@					\
         $(TSLIB_LIBS)                                   \
-        @XEPHYR_LIBS@					\
-	../../../exa/libexa.la
+        @XEPHYR_LIBS@
 
 Xephyr_DEPENDENCIES =					\
 	libxephyr.a					\
