Index: xorg/configure.ac
===================================================================
RCS file: /cvs/xorg/xserver/xorg/configure.ac,v
retrieving revision 1.30
diff -u -b -B -r1.30 configure.ac
--- xorg/configure.ac	27 Jul 2005 08:11:26 -0000	1.30
+++ xorg/configure.ac	28 Jul 2005 20:49:35 -0000
@@ -541,7 +547,7 @@
 if test x$XF86VIDMODE = xyes; then
   AC_DEFINE(XF86VIDMODE,1,[Support XFree86 Video Mode extension])
 fi
-if test x$XF86MIS = xyes; then
+if test x$XF86MISC = xyes; then
   AC_DEFINE(XF86MISC,1,[Support XFree86 miscellaneous extensions])
 fi
 # must come first as it uses stuff in Shape -d
