--- chemtool-1.5/Makefile.in	2002-04-11 14:41:37.000000000 -0500
+++ chemtool-1.5-gentoo/Makefile.in	2002-11-13 01:35:30.000000000 -0500
@@ -2,10 +2,13 @@
 exec_prefix=@exec_prefix@
 datadir=@datadir@
 localedir=@localedir@
+uselocales=@uselocales@
 mandir=@mandir@
 kdedir=@kdemimedir@
+usekde=@usekde@
 gnomedir=@gnomemimedir@
-CFLAGS=-O2 -Wall -Wunused -Wuninitialized -I. `gtk-config --cflags` @DEFS@ -DLOCALEDIR=\"${localedir}\"
+usegnome=@usegnome@
+CFLAGS=@CFLAGS@ -I. `gtk-config --cflags` @DEFS@
 
 CC=@CC@
 CP=/bin/cp
@@ -26,7 +29,7 @@
 	${CC} ${OBJS} ${SYS_LIBRARIES} -o chemtool
 
 cht:	src-cht/cht-2.1.c
-	cd src-cht;${MAKE}
+	cd src-cht && ${MAKE}
 
 
 install: chemtool src-cht/cht
@@ -34,21 +37,37 @@
 	${INSTALL} chemtool src-cht/cht ${prefix}/bin
 	$(INSTALL) chemtool.1 ${mandir}/man1
 	$(INSTALL) cht.1 ${mandir}/man1
-ifdef localedir
-	for L in `ls locales`; do ${INSTALL} -d ${localedir}/$$L/LC_MESSAGES; ${INSTALL} -m664 locales/$$L/chemtool.mo ${localedir}/$$L/LC_MESSAGES;done
+ifdef uselocales
+	for L in `ls locales` ; do \
+		${INSTALL} -d ${localedir}/$$L/LC_MESSAGES ; \
+		${INSTALL} -m664 locales/$$L/chemtool.mo \
+			${localedir}/$$L/LC_MESSAGES ; \
+	done
 endif
-ifdef kdedir
-	${INSTALL} kde/applnk/Graphics/chemtool.desktop ${kdedir}/share/applnk/Graphics 
-	${INSTALL} kde/mimelnk/application/x-chemtool.desktop ${kdedir}/share/mimelnk/application
-	${INSTALL} kde/icons/hicolor/32x32/mimetypes/chemtool.png ${kdedir}/share/icons/hicolor/32x32/mimetypes
+ifdef usekde
+	${INSTALL} -d ${kdedir}/share/applnk/Graphics \
+		${kdedir}/share/mimelnk/application \
+		${kdedir}/share/icons/hicolor/32x32/mimetypes
+	${INSTALL} kde/applnk/Graphics/chemtool.desktop \
+		${kdedir}/share/applnk/Graphics
+	${INSTALL} kde/mimelnk/application/x-chemtool.desktop \
+		${kdedir}/share/mimelnk/application
+	${INSTALL} kde/icons/hicolor/32x32/mimetypes/chemtool.png \
+		${kdedir}/share/icons/hicolor/32x32/mimetypes
+endif
+ifdef usegnome
+	${INSTALL} -d ${gnomedir}/share/mime-types \
+		${gnomedir}/share/pixmaps/mc
+	${INSTALL} gnome/mime-types/chemtool.keys \
+		gnome/mime-types/chemtool.mime \
+		${gnomedir}/share/mime-types
+	${INSTALL} gnome/gnome-application-chemtool.png \
+		${gnomedir}/share/pixmaps/mc
 endif
 
-ifdef gnomedir
-	${INSTALL} gnome/mime-types/chemtool.keys gnome/mime-types/chemtool.mime ${gnomedir}/share/mime-types
-	${INSTALL} gnome/gnome-application-chemtool.png ${gnomedir}/share/pixmaps/mc
-endif 
 clean:
 	-${RM} *.o *~ src-cht/*.o src-cht/*~ chemtool src-cht/cht
 
 distclean: clean
-	-${RM} Makefile src-cht/Makefile config.log config.cache config.status config.h
+	-${RM} Makefile src-cht/Makefile config.log config.cache \
+		config.status config.h
