diff -Nur src.orig/Makefile.am src/Makefile.am
--- src.orig/Makefile.am	2007-12-29 15:38:01.000000000 +0100
+++ src/Makefile.am	2007-12-29 15:38:27.000000000 +0100
@@ -5,13 +5,13 @@
 # extractor
 ift_extract_SOURCES = extract.c
 ift_extract_CFLAGS = $(IFT_CFLAGS) $(AM_CFLAGS)
-ift_extract_LDFLAGS = $(IFT_LIBS) -lgcrypt
+ift_extract_LDADD = $(IFT_LIBS) -lgcrypt
 
 
 # export
 ift_export_SOURCES = export.c
 ift_export_CFLAGS = $(IFT_CFLAGS) $(AM_CFLAGS)
-ift_export_LDFLAGS = $(IFT_LIBS)
+ift_export_LDADD = $(IFT_LIBS)
 
 libudevdir = ${libdir}/udev
 libudev_PROGRAMS = ift-load
@@ -19,4 +19,4 @@
 # loader
 ift_load_SOURCES = load.c
 ift_load_CFLAGS = $(IFT_CFLAGS) $(LOAD_CFLAGS) $(AM_CFLAGS)
-ift_load_LDFLAGS = $(IFT_LIBS) $(LOAD_LIBS)
+ift_load_LDADD = $(IFT_LIBS) $(LOAD_LIBS)
diff -Nur src.orig/Makefile.in src/Makefile.in
--- src.orig/Makefile.in	2007-12-29 15:38:01.000000000 +0100
+++ src/Makefile.in	2007-12-29 15:38:27.000000000 +0100
@@ -215,18 +215,18 @@
 # extractor
 ift_extract_SOURCES = extract.c
 ift_extract_CFLAGS = $(IFT_CFLAGS) $(AM_CFLAGS)
-ift_extract_LDFLAGS = $(IFT_LIBS) -lgcrypt
+ift_extract_LDADD = $(IFT_LIBS) -lgcrypt
 
 # export
 ift_export_SOURCES = export.c
 ift_export_CFLAGS = $(IFT_CFLAGS) $(AM_CFLAGS)
-ift_export_LDFLAGS = $(IFT_LIBS)
+ift_export_LDADD = $(IFT_LIBS)
 libudevdir = ${libdir}/udev
 
 # loader
 ift_load_SOURCES = load.c
 ift_load_CFLAGS = $(IFT_CFLAGS) $(LOAD_CFLAGS) $(AM_CFLAGS)
-ift_load_LDFLAGS = $(IFT_LIBS) $(LOAD_LIBS)
+ift_load_LDADD = $(IFT_LIBS) $(LOAD_LIBS)
 all: all-am
 
 .SUFFIXES:
