--- ./src/kbddef.c.xml2 Sun Jun  2 22:37:33 2002
+++ ./src/kbddef.c      Sun Jun  2 22:37:43 2002
@@ -38,7 +38,7 @@
 #include "XF86keysym.h"

 #include <xmlmemory.h>
-#include <parser.h>
+#include <libxml2/libxml/parser.h>

 #include "hotkeys.h"
 #include "kbddef.h"

--- /tmp/kbddef.c       Wed Sep 11 18:51:19 2002
+++ src/kbddef.c        Wed Sep 11 18:51:33 2002
@@ -37,7 +37,7 @@
  */
 #include "XF86keysym.h"

-#include <xmlmemory.h>
+#include <libxml2/libxml/xmlmemory.h>
 #include <libxml2/libxml/parser.h>

 #include "hotkeys.h"

