--- slirpvde/slirpvde.c	2006-12-18 16:24:27.000000000 +0100
+++ slirpvde/slirpvde.c	2008-07-10 15:44:39.000000000 +0200
@@ -32,8 +32,8 @@
 
 #include "misc.h"
 
+#include <limits.h>
 #ifdef VDE_DARWIN
-#	include <limits.h>
 #	if defined HAVE_SYSLIMITS_H
 #		include <syslimits.h>
 #	elif defined HAVE_SYS_SYSLIMITS_H
--- vde_plug2tap/vde_plug2tap.c	2006-12-18 14:49:57.000000000 +0100
+++ vde_plug2tap/vde_plug2tap.c	2008-07-10 15:45:12.000000000 +0200
@@ -30,9 +30,9 @@
 #include <linux/if_tun.h>
 #endif
 
+#include <limits.h>
 #ifdef VDE_DARWIN
 #	define TAP_PREFIX "/dev/"
-#	include <limits.h>
 #	if defined HAVE_SYSLIMITS_H
 #		include <syslimits.h>
 #	elif defined HAVE_SYS_SYSLIMITS_H
--- vdetaplib/libvdetap.c	2006-12-18 14:36:05.000000000 +0100
+++ vdetaplib/libvdetap.c	2008-07-10 15:47:24.000000000 +0200
@@ -20,6 +20,7 @@
 #include <linux/ioctl.h>
 #include <linux/if.h>
 #include <linux/if_tun.h>
+#include <limits.h>
 
 #define TUNTAPPATH "/dev/net/tun"
 #define VDETAPEXEC LIBEXECDIR "/vdetap"
