diff -ur vde-1.5.8-orig/qemu/Makefile.am vde-1.5.8/qemu/Makefile.am
--- vde-1.5.8-orig/qemu/Makefile.am	2006-04-03 14:01:35.000000000 -0400
+++ vde-1.5.8/qemu/Makefile.am	2006-04-03 14:01:57.000000000 -0400
@@ -6,8 +6,8 @@
 EXTRA_DIST = README
 
 install-exec-hook:
-	rm -f $(bindir)/vdeqemu
-	ln -s $(bindir)/vdeq $(bindir)/vdeqemu
+	rm -f $(DESTDIR)$(bindir)/vdeqemu
+	ln -s $(DESTDIR)$(bindir)/vdeq $(DESTDIR)$(bindir)/vdeqemu
 
 uninstall-local:
 	rm -f $(bindir)/vdeqemu
diff -ur vde-1.5.8-orig/slirpvde/Makefile.am vde-1.5.8/slirpvde/Makefile.am
--- vde-1.5.8-orig/slirpvde/Makefile.am	2006-04-03 14:01:35.000000000 -0400
+++ vde-1.5.8/slirpvde/Makefile.am	2006-04-03 14:02:06.000000000 -0400
@@ -40,6 +40,6 @@
 		   bootp.h \
 		   slirpvde.c
 
-INCLUDES = -DVDE -D"u_int=unsigned int"
+INCLUDES = -DVDE
 
 EXTRA_DIST = README
