From 22b1a9dd0f68e3307cdcf3e6f8fcaeda34acd483 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Michel=20D=C3=A4nzer?= <michel@tungstengraphics.com>
Date: Thu, 18 Oct 2007 17:44:48 +0200
Subject: [PATCH] EXA: Skip empty glyphs. (cherry picked from commit ce50bfd3369686cfecee5a138bd84ef1107a249d)

---
 exa/exa_render.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/exa/exa_render.c b/exa/exa_render.c
index 2dd3fc1..ad1c02b 100644
--- a/exa/exa_render.c
+++ b/exa/exa_render.c
@@ -1184,6 +1184,7 @@ exaGlyphs (CARD8	op,
 	    y1 = y - glyph->info.y;
 
 	    if (x1 >= pCmpDrw->width || y1 >= pCmpDrw->height ||
+		glyph->info.width == 0 || glyph->info.height == 0 ||
 		(x1 + glyph->info.width) <= 0 || (y1 + glyph->info.height) <= 0)
 		goto nextglyph;
 
-- 
1.5.5.1

