--- gnomad-2.9.1.orig/src/jukebox.c	2008/03/25 23:52:29	1.122
+++ gnomad-2.9.1/src/jukebox.c	2008/06/26 23:52:41	1.123
@@ -2177,11 +2177,14 @@
 	trackmeta->duration = length * 1000;
 	trackmeta->filesize = get_64bit_file_size(tmpfname);
 	trackmeta->filetype = filetype_d->filetype;
+	// Hardcode parent (folder) to 0 (== root directory)
+	trackmeta->parent_id = 0;
+	// Hardcode storage to primary
+	trackmeta->storage_id = 0;
 	// TODO: Discarding return value as for now
 	// g_print("Transferring MTP track...\n");
-	// Hardcode parent (folder) to 0 (== root directory)
 	ret = LIBMTP_Send_Track_From_File(mtpdevice, tmpfname, trackmeta,
-					  mtp_progress, NULL, 0);
+					  mtp_progress, NULL);
 	if (ret != 0) {
 	  g_print("Error sending file \"%s\" to MTP device!\n", tmpfname);
 	  LIBMTP_Dump_Errorstack(mtpdevice);
@@ -2348,7 +2351,7 @@
 	filemeta->filesize = (uint64_t) hdmeta->size;
 	filemeta->filetype = filetype_d->filetype;
 	filemeta->parent_id = mtp_current_filter_id;
-	if (LIBMTP_Send_File_From_File(mtpdevice, tmpfname, filemeta, mtp_progress, NULL, mtp_current_filter_id) != 0) {
+	if (LIBMTP_Send_File_From_File(mtpdevice, tmpfname, filemeta, mtp_progress, NULL) != 0) {
 	  LIBMTP_destroy_file_t(filemeta);
 	  g_free(tmpfname);
 	  g_print("Error sending file \"%s\" to MTP device!\n", tmpfname);
@@ -2746,8 +2749,10 @@
 
     pl = LIBMTP_new_playlist_t();
     pl->name = g_strdup(plname);
+    pl->parent_id = 0;
+    pl->storage_id = 0;
     // This will create the new playlist in the default folder
-    if (LIBMTP_Create_New_Playlist(mtpdevice, pl, 0) != 0) {
+    if (LIBMTP_Create_New_Playlist(mtpdevice, pl) != 0) {
       create_error_dialog(_("Could not create playlist"));
       return 0;
     }
@@ -3609,7 +3614,7 @@
 
 #ifdef HAVE_LIBMTP
   if (use_mtp) {
-    int ret=LIBMTP_Create_Folder(mtpdevice,foldername,mtp_current_filter_id);
+    int ret=LIBMTP_Create_Folder(mtpdevice,foldername,mtp_current_filter_id,0);
     mtp_initialize_folders();
     rebuild_datafile_list(folder);
     return;
