--- input/xineplug_inp_cd.c	2003-10-17 02:15:44.000000000 +0200
+++ input/xineplug_inp_cd.c.new	2004-05-12 13:10:33.574749582 +0200
@@ -1255,17 +1255,6 @@
 }
 
 
-static void 
-meta_info_assign(int field, xine_stream_t *stream, const char * info) 
-{
-  if (NULL != info) {
-    if (stream->meta_info[field])
-      free(stream->meta_info[field]);
-    dbg_print(INPUT_DBG_META, "meta[%d]: %s\n", field, info);
-    stream->meta_info[field] = strdup(info);
-  }
-}
-
 /* Update the xine player title text. */
 static void 
 cdda_update_title(cdda_input_plugin_t *this) 
--- input/xineplug_inp_vcd.c	2003-10-08 04:51:38.000000000 +0200
+++ input/xineplug_inp_vcd.c.new	2004-05-12 13:16:09.319051269 +0200
@@ -164,37 +164,6 @@
   return true;
 }
 
-
-static void 
-meta_info_assign(int field, xine_stream_t *stream, const char * info) 
-{
-  if (NULL != info) {
-    if (stream->meta_info[field])
-      free(stream->meta_info[field]);
-    dbg_print(INPUT_DBG_META, "meta[%d]: %s\n", field, info);
-    stream->meta_info[field] = strdup(info);
-  }
-}
-
-#define stream_info_assign(field, stream, info) \
-  stream->stream_info[field] = info;
-
-/* Set stream information. */
-static void
-vcd_set_meta_info (vcd_input_plugin_t *xine_vcd) 
-{
-  vcdinfo_obj_t *obj= xine_vcd->player.vcd;
-  meta_info_assign(XINE_META_INFO_ALBUM, xine_vcd->stream, 
-                   vcdinfo_get_album_id(obj));
-  meta_info_assign(XINE_META_INFO_ARTIST,  xine_vcd->stream,
-                   vcdinfo_get_preparer_id(obj));
-  meta_info_assign(XINE_META_INFO_COMMENT, xine_vcd->stream,
-                   vcdplayer_format_str(&xine_vcd->player,
-                                   xine_vcd->v_config.comment_format));
-  meta_info_assign(XINE_META_INFO_GENRE, xine_vcd->stream,
-                   vcdinfo_get_format_version_str(obj));
-}
-
 static void
 vcd_force_redisplay (void) 
 {
@@ -1355,13 +1324,6 @@
   title_str = vcdplayer_format_str(&my_vcd.player, 
                                    my_vcd.v_config.title_format);
 
-  meta_info_assign(XINE_META_INFO_TITLE, my_vcd.stream, title_str);
-  meta_info_assign(XINE_META_INFO_COMMENT, my_vcd.stream,
-                   vcdplayer_format_str(&my_vcd.player, 
-                                        my_vcd.v_config.comment_format));
-  stream_info_assign(XINE_STREAM_INFO_VIDEO_HAS_STILL, my_vcd.stream, 
-                     my_vcd.player.in_still);
-
   /* Set_str title/chapter display */
   dbg_print((INPUT_DBG_MRL|INPUT_DBG_CALL),
             "Changing title to read '%s'\n", title_str);
@@ -1469,7 +1431,6 @@
   dbg_print(INPUT_DBG_PBC, "Jumping to NUM >%i<, type >%i<\n", 
             itemid.num, itemid.type);
 
-  vcd_set_meta_info(&my_vcd);
   vcdplayer_play(&my_vcd.player, itemid);
   
 
