--- Makefile.orig	2003-11-03 21:31:31.154535784 -0500
+++ Makefile	2003-11-03 21:32:32.385227304 -0500
@@ -138,9 +138,8 @@
 
 
 OUTPUT = $(OBJDIR)/libmpeg3.a
-UTILS = $(OBJDIR)/mpeg3dump $(OBJDIR)/mpeg3toc  $(OBJDIR)/mpeg3cat
-
-#$(OBJDIR)/mpeg3split
+SHAREDOUTPUT = $(OBJDIR)/libmpeg3.so
+UTILS = $(OBJDIR)/mpeg3dump $(OBJDIR)/mpeg3toc  $(OBJDIR)/mpeg3cat $(OBJDIR)/mpeg3split
 
 
 LIBS = -lm -lpthread
@@ -152,13 +151,14 @@
 $(shell echo $(OBJS) $(ASMOBJS) $(A52OBJS) $(NASMOBJS) > $(OBJDIR)/objs)
 $(shell mkdir -p $(DIRS) )
 
-all: $(OUTPUT) $(UTILS)
+all: $(OUTPUT) $(SHAREDOUTPUT) $(UTILS)
 
 
 $(OUTPUT): $(OBJS) $(ASMOBJS) $(NASMOBJS) $(A52OBJS)
 	ar rcs $(OUTPUT) `cat $(OBJDIR)/objs`
 
-
+$(SHAREDOUTPUT): $(OBJS) $(ASMOBJS) $(NASMOBJS)
+	gcc -shared -o $(SHAREDOUTPUT) $(OBJS) $(ASMOBJS) $(NASMOBJS) $(LIBS)
 
 $(OBJDIR)/mpeg3dump: $(OUTPUT) mpeg3dump.c
 	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3dump mpeg3dump.c $(OUTPUT) $(LIBS)
@@ -169,8 +169,8 @@
 $(OBJDIR)/mpeg3cat: $(OUTPUT) mpeg3cat.c
 	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3cat mpeg3cat.c $(OUTPUT) $(LIBS)
 
-#$(OBJDIR)/mpeg3split: $(OUTPUT)
-#	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3split mpeg3split.c $(OUTPUT) $(LIBS)
+$(OBJDIR)/mpeg3split: $(OUTPUT)
+	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3split mpeg3split.c $(OUTPUT) $(LIBS)
 
 $(OBJDIR)/mpeg2qt: $(OUTPUT)
 	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg2qt mpeg2qt.c \
@@ -213,7 +213,7 @@
 $(OBJDIR)/mpeg3demux.o: 			    mpeg3demux.c
 $(OBJDIR)/mpeg3ifo.o: 				    mpeg3ifo.c
 $(OBJDIR)/mpeg3io.o: 				    mpeg3io.c
-#$(OBJDIR)/mpeg3split.o: 			    mpeg3split.c
+$(OBJDIR)/mpeg3split.o: 			    mpeg3split.c
 $(OBJDIR)/mpeg3title.o: 			    mpeg3title.c
 $(OBJDIR)/mpeg3toc.o: 				    mpeg3toc.c
 $(OBJDIR)/mpeg3toc3.o:  			    mpeg3toc3.c
--- mpeg3split.c.orig	2003-11-03 21:33:17.923304464 -0500
+++ mpeg3split.c	2003-11-03 21:33:36.419492616 -0500
@@ -2,6 +2,7 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include "mpeg3private.h"
 #include "mpeg3private.inc"
 
 void copy_data(FILE *out, FILE *in, long bytes)
