diff -r 531b73491ac2 virtinst/DistroManager.py
--- a/virtinst/DistroManager.py	Wed Oct 10 14:24:48 2007 -0400
+++ b/virtinst/DistroManager.py	Thu Oct 11 11:39:58 2007 -0400
@@ -161,7 +161,7 @@ class MountedImageFetcher(ImageFetcher):
             if tmpfile is not None:
                 os.unlink(tmpfile)
             return True
-        except RuntimeError, e:
+        except Exception, e:
             logging.debug("Cannot find file %s" % filename)
             return False
 
