--- partimage-0.6.2/Makefile.am.orig	2002-11-25 06:08:36.000000000 -0500
+++ partimage-0.6.2/Makefile.am	2002-11-25 06:08:58.000000000 -0500
@@ -23,8 +23,6 @@
 	rm ${sysconfdir}/partimaged/partimaged.csr ;\
 	chmod 600 ${sysconfdir}/partimaged/partimaged.key ;\
 	chmod 600 ${sysconfdir}/partimaged/partimaged.cert ;\
-	chown partimag.root ${sysconfdir}/partimaged/partimaged.key ;\
-	chown partimag.root ${sysconfdir}/partimaged/partimaged.cert ;\
 	else \
 	echo "SSL disabled, no certificate will be generated." ;\
 	fi
@@ -57,7 +55,6 @@
 	echo "#sample # user 'sample' is allowed to connect partimaged" >> \
            ${sysconfdir}/partimaged/partimagedusers ;\
 	chmod 600 ${sysconfdir}/partimaged/partimagedusers ;\
-	chown partimag.root ${sysconfdir}/partimaged/partimagedusers ;\
 	fi 
 
 	$(mkinstalldirs) ${infodir}
--- partimage-0.6.2/Makefile.in.orig	2002-11-25 06:15:31.000000000 -0500
+++ partimage-0.6.2/Makefile.in	2002-11-25 06:15:58.000000000 -0500
@@ -651,8 +651,6 @@
 	rm ${sysconfdir}/partimaged/partimaged.csr ;\
 	chmod 600 ${sysconfdir}/partimaged/partimaged.key ;\
 	chmod 600 ${sysconfdir}/partimaged/partimaged.cert ;\
-	chown partimag.root ${sysconfdir}/partimaged/partimaged.key ;\
-	chown partimag.root ${sysconfdir}/partimaged/partimaged.cert ;\
 	else \
 	echo "SSL disabled, no certificate will be generated." ;\
 	fi
@@ -684,7 +682,6 @@
 	echo "#sample # user 'sample' is allowed to connect partimaged" >> \
            ${sysconfdir}/partimaged/partimagedusers ;\
 	chmod 600 ${sysconfdir}/partimaged/partimagedusers ;\
-	chown partimag.root ${sysconfdir}/partimaged/partimagedusers ;\
 	fi 
 
 	$(mkinstalldirs) ${infodir}
