diff -urN Canna36p3.ORIG/Canna.conf Canna36p3/Canna.conf
--- Canna36p3.ORIG/Canna.conf	2003-01-25 17:47:47.000000000 +0900
+++ Canna36p3/Canna.conf	2003-05-05 19:48:58.000000000 +0900
@@ -133,7 +133,7 @@
 /* rootにならずにインストールする時はこれを定義して下さい */
 /* #define InstallAsUser */
 /* 従来と互換のディレクトリ構造にする場合はこれを定義してください */
-#define ObsoleteInstallDir
+/* #define ObsoleteInstallDir */
 #ifdef InstallAsUser
 # define DefCannaPrefix $(HOME)/canna
 # define DefErrDir $(cannaPrefix)/log
@@ -144,7 +144,7 @@
 #  define DefErrDir /usr/spool/canna
 #  define DefLibCannaDir /usr/lib
 # else
-#  define DefErrDir /var/log
+#  define DefErrDir /var/log/canna
 #  define DefLibCannaDir $(cannaExecPrefix)/lib
 # endif
 #endif
@@ -156,16 +156,18 @@
 # define DefCannaLibDir $(cannaPrefix)/share/canna
 #endif
 
-cannaPrefix = DefCannaPrefix
+cannaPrefix = /usr
 cannaExecPrefix = $(cannaPrefix)
 cannaBinDir = $(cannaExecPrefix)/bin
 cannaSrvDir = DefCannaSrvDir
 XCOMM cannaLibDir = /var/lib/canna
 XCOMM cannaLibDir = /var/db/canna
 cannaLibDir = DefCannaLibDir
-XCOMM cannaManDir = $(cannaPrefix)/share/man
-cannaManDir = $(cannaPrefix)/man
+cannaManDir = $(cannaPrefix)/share/man
+XCOMM cannaManDir = $(cannaPrefix)/man
 cannaIncDir = $(cannaPrefix)/include/canna
+cannaDocDir = $(cannaPrefix)/share/doc/canna-3.6_p3
+cannaShareDir = $(cannaPrefix)/share/canna
 
 libCannaDir = DefLibCannaDir
 ErrDir  = DefErrDir
@@ -248,7 +250,7 @@
 /* #define UseInstalledLibCanna YES */
 #define UseInstalledLibCanna NO
 
-DicDir   = $(cannaLibDir)/dic
+DicDir   = /var/lib/canna/dic
 
 #ifdef NeXT
 #define NO_X_ENV
@@ -365,6 +367,7 @@
 	 echo "s%@(UnixSockDir)%$(UnixSockDir)%g"; \			@@\
 	 echo "s%@(UnixSockName)%$(UnixSockName)%g"; \			@@\
 	 echo "s%@(AccessFile)%$(AccessFile)%g"; \			@@\
+	 echo "s%@(DicDir)%$(DicDir)%g"; \				@@\
 	 echo "s%@(ErrDir)%$(ErrDir)%g") > $(MANUALSED)			@@\
 clean::									@@\
 	$(RM) $(MANUALSED)
@@ -381,6 +384,7 @@
 	echo "s%@(UnixSockDir)%$(UnixSockDir)%g" >> $(MANUALSED)	@@\
 	echo "s%@(UnixSockName)%$(UnixSockName)%g" >> $(MANUALSED)	@@\
 	echo "s%@(AccessFile)%$(AccessFile)%g" >> $(MANUALSED)		@@\
+	echo "s%@(DicDir)%$(DicDir)%g" >> $(MANUALSED)			@@\
 	echo "s%@(ErrDir)%$(ErrDir)%g" >> $(MANUALSED)			@@\
 clean::									@@\
 	$(RM) $(MANUALSED)
@@ -561,6 +565,7 @@
 	echo "#define CANNAHOSTFILE '$(cannaLibDir)/cannahost'" >> junk	@@\
 	echo "/* for lib/canna */" >> junk				@@\
 	echo "#define CANNALIBDIR '$(cannaLibDir)'" >> junk		@@\
+	echo "#define CANNASHAREDIR '$(cannaShareDir)'" >> junk		@@\
 	echo "/* for scripts */" >> junk				@@\
 	echo "#define CANNABINDIR $(cannaBinDir)" >> junk		@@\
 	echo "/* others */" >> junk					@@\
@@ -587,6 +592,7 @@
 	 echo "#define CANNAHOSTFILE \"$(cannaLibDir)/cannahost\"";\	@@\
 	 echo "/* for lib/canna */";\					@@\
 	 echo "#define CANNALIBDIR \"$(cannaLibDir)\""; \		@@\
+	 echo "#define CANNASHAREDIR \"$(cannaShareDir)\""; \		@@\
 	 echo "/* for scripts */";\					@@\
 	 echo "#define CANNABINDIR $(cannaBinDir)"; \			@@\
 	 echo "/* others */";\					@@\
diff -urN Canna36p3.ORIG/cmd/catdic/cannakill.man Canna36p3/cmd/catdic/cannakill.man
--- Canna36p3.ORIG/cmd/catdic/cannakill.man	2002-10-23 00:19:13.000000000 +0900
+++ Canna36p3/cmd/catdic/cannakill.man	2003-05-05 18:41:06.000000000 +0900
@@ -20,7 +20,7 @@
 .\" PERFORMANCE OF THIS SOFTWARE.
 .\"
 .\" $Id: cannakill.man,v 1.2 2002/10/22 15:19:13 aida_s Exp $
-.TH CANNAKILL 1M
+.TH CANNAKILL 1
 .SH "NAME"
 cannakill \- Terminate operation of the Kana-Kanji conversion server
 cannaserver
@@ -29,14 +29,14 @@
 .SH "DESCRIPTION"
 .PP
 cannakill terminates operation of Kana-Kanji conversion server
-cannaserver(1M).  This causes the server to delete the socket and
+cannaserver(1).  This causes the server to delete the socket and
 close all open dictionaries before terminating its processing.
 .PP
-The user who has started cannaserver(1M) or the super user can only
-use cannakill(1M) to terminate operation of canncannaer.
+The user who has started cannaserver(1) or the super user can only
+use cannakill(1) to terminate operation of canncannaer.
 .SH "FILES"
 .IP "@(UnixSockDir)/@(UnixSockName)[:num]" 36
 Socket
 .SH "SEE ALSO"
 .PP
-cannaserver(1M), cannastat(1), cannacheck(1)
+cannaserver(1), cannastat(1), cannacheck(1)
diff -urN Canna36p3.ORIG/cmd/catdic/catdic.man Canna36p3/cmd/catdic/catdic.man
--- Canna36p3.ORIG/cmd/catdic/catdic.man	2002-10-19 17:27:33.000000000 +0900
+++ Canna36p3/cmd/catdic/catdic.man	2003-05-05 18:41:39.000000000 +0900
@@ -30,7 +30,7 @@
 catdic downloads a dictionary file to the current directory. 
 The file downloaded is a dictionary file the dictionary name of which is
 \fIremote-dic\fP.  It is in the user dictionary directory of the machine 
-on which \fIcannaserver\fP(1M) is in operation.  (Hereafter, this machine 
+on which \fIcannaserver\fP(1) is in operation.  (Hereafter, this machine 
 is called the remote host.)
 .SH "OPTIONS"
 .IP "\-cs \fIcanna-server\fP (or \-cannaserver \fIcanna-server\fP)" 12
@@ -61,4 +61,4 @@
 file ``test2.t''.
 .SH "SEE ALSO"
 .PP
-cannaserver(1M), lsdic(1), mkdic(1)
+cannaserver(1), lsdic(1), mkdic(1)
diff -urN Canna36p3.ORIG/cmd/catdic/mkdic.man Canna36p3/cmd/catdic/mkdic.man
--- Canna36p3.ORIG/cmd/catdic/mkdic.man	2002-10-19 17:27:33.000000000 +0900
+++ Canna36p3/cmd/catdic/mkdic.man	2003-05-05 18:42:01.000000000 +0900
@@ -28,7 +28,7 @@
 .SH "DESCRIPTION"
 .PP
 mkdic creates user dictionary named \fIremote-dic\fP in the user dictionary 
-directory of the remote host, on which \fIcannaserver\fP(1M) is in operation. 
+directory of the remote host, on which \fIcannaserver\fP(1) is in operation. 
 Dictionary file name \fIremote-dic\fP is assigned to this file at this time.
 .PP
 If the \- and \-l option isn't specified, mkdic creates an empty dictionary.
@@ -96,4 +96,4 @@
 user1.t is a dictionary file name generated automatically by the system.
 .SH "SEE ALSO"
 .PP
-catdic(1), cannaserver(1M), lsdic(1)
+catdic(1), cannaserver(1), lsdic(1)
diff -urN Canna36p3.ORIG/cmd/cshost/cshost.man Canna36p3/cmd/cshost/cshost.man
--- Canna36p3.ORIG/cmd/cshost/cshost.man	2002-10-19 17:27:34.000000000 +0900
+++ Canna36p3/cmd/cshost/cshost.man	2003-05-05 18:42:29.000000000 +0900
@@ -6,9 +6,9 @@
 .SH "DESCRIPTION"
 .PP
 the \fIcshost\fP(1) prints the list of machines that are allowed to make
-connections to \fIcannaserver\fP(1M).
+connections to \fIcannaserver\fP(1).
 .PP
-\fIcannaserver\fP(1M) allows network connections only from programs
+\fIcannaserver\fP(1) allows network connections only from programs
 running on the same machine or machines listed in the file
 /etc/hosts.canna.  If /etc/hosts.canna does not exist or the file
 is empty, access is granted to everyone.
diff -urN Canna36p3.ORIG/dic/phono/Imakefile Canna36p3/dic/phono/Imakefile
--- Canna36p3.ORIG/dic/phono/Imakefile	2002-10-19 17:27:40.000000000 +0900
+++ Canna36p3/dic/phono/Imakefile	2003-05-05 20:35:00.000000000 +0900
@@ -27,9 +27,10 @@
 
        ROMAJI_DIC_DEF = -DSHIFT
 
-               DICDIR = $(cannaLibDir)/dic
-            SAMPLEDIR = $(cannaLibDir)/sample
-         SAMPLESRCDIR = $(cannaLibDir)/sample/src
+               DICDIR = $(DicDir)
+        CANNASHAREDIR = $(cannaShareDir)
+            SAMPLEDIR = $(cannaDocDir)/sample
+         SAMPLESRCDIR = $(cannaDocDir)/sample/src
 
     KPDEF = default.kpdef jdaemon.kpdef just.kpdef kaisoku.kpdef kana.kpdef \
             lan5.kpdef matsu.kpdef newjis.kpdef romaji.kpdef sokuon.kpdef \
@@ -69,7 +70,7 @@
 MakeDirectories(install,$(DICDIR))
 MakeDirectories(install,$(SAMPLESRCDIR))
 
-InstallMultiple($(ROMKANA_TABLES),$(DICDIR))
+InstallMultiple($(ROMKANA_TABLES),$(CANNASHAREDIR))
 
 #ifdef USE_OBSOLETE_STYLE_FILENAME
 InstallNonExec(default.kpdef,$(SAMPLESRCDIR))
diff -urN Canna36p3.ORIG/lib/canna/Imakefile Canna36p3/lib/canna/Imakefile
--- Canna36p3.ORIG/lib/canna/Imakefile	2002-10-20 17:20:33.000000000 +0900
+++ Canna36p3/lib/canna/Imakefile	2003-05-05 18:45:37.000000000 +0900
@@ -116,8 +116,8 @@
 #endif /* ProjectX */
 #endif /* DoSharedLib */
 
-sglobal.h: GSYM.awk globnames
-	awk -f GSYM.awk < globnames > $@
+includes::
+	awk -f GSYM.awk < globnames > sglobal.h
 
 clean::
 	$(RM) y.tab.h y.tab.c lex.yy.c gram.h gram.c lex.c sglobal.h
diff -urN Canna36p3.ORIG/lib/canna/romaji.c Canna36p3/lib/canna/romaji.c
--- Canna36p3.ORIG/lib/canna/romaji.c	2003-01-17 17:48:19.000000000 +0900
+++ Canna36p3/lib/canna/romaji.c	2003-05-05 19:17:15.000000000 +0900
@@ -438,7 +438,7 @@
 	}
       }
       
-#if 0 /* currently CANNASHAREDDIR is not defined */
+#if 1 /* currently CANNASHAREDDIR is defined */
       if (retval == (struct RkRxDic *)NULL) { /* added for Debian by ISHIKAWA Mutsumi <ishikawa@linux.or.jp> */
         extern jrUserInfoStruct *uinfo;
 	
diff -urN Canna36p3.ORIG/misc/Imakefile Canna36p3/misc/Imakefile
--- Canna36p3.ORIG/misc/Imakefile	2003-01-13 14:34:30.000000000 +0900
+++ Canna36p3/misc/Imakefile	2003-05-05 19:19:27.000000000 +0900
@@ -34,14 +34,14 @@
 MakeLintLibSubdirs($(SUBDIRS))
 MakeLintSubdirs($(SUBDIRS),install.ln,install.ln)
 
-MakeDirectories(install,$(cannaLibDir)/sample/src)
+MakeDirectories(install,$(cannaDocDir)/sample/src)
 #if defined(nec_ews_svr4) || defined(pcux_r32)
-InstallNamedNonExec(rMakefile,Makefile,$(cannaLibDir)/sample/src)
+InstallNamedNonExec(rMakefile,Makefile,$(cannaDocDir)/sample/src)
 #else
-InstallNamedNonExec(cMakefile,Makefile,$(cannaLibDir)/sample/src)
+InstallNamedNonExec(cMakefile,Makefile,$(cannaDocDir)/sample/src)
 #endif /* nec_ews_svr4 */
 
-InstallNonExec(is.c,$(cannaLibDir)/sample/src)
+InstallNonExec(is.c,$(cannaDocDir)/sample/src)
 
 clean::
 	$(RM) $(PROGRAMS)
diff -urN Canna36p3.ORIG/misc/initfiles/Imakefile Canna36p3/misc/initfiles/Imakefile
--- Canna36p3.ORIG/misc/initfiles/Imakefile	2003-01-21 19:57:03.000000000 +0900
+++ Canna36p3/misc/initfiles/Imakefile	2003-05-05 19:21:20.000000000 +0900
@@ -6,7 +6,7 @@
 #endif
 
             CANNAROOT = ../..
-            SAMPLEDIR = $(cannaLibDir)/sample
+            SAMPLEDIR = $(cannaShareDir)
 
 all::
 
@@ -22,7 +22,7 @@
 Install: 1.1.cob 1.2.cob fixer.cob jdemon.cob just.cob lan5.cob modes.cob \
          matsu.cob sample.cob skk.cob tut.cob unix.cob vje.cob wx2+.cob \
          default.cob
-InstallNamedNonExec(default.cob,default.canna,$(cannaLibDir))
+InstallNamedNonExec(default.cob,default.canna,$(cannaShareDir))
 InstallNamedNonExec(1.1.cob,1.1.canna,$(SAMPLEDIR))
 InstallNamedNonExec(1.2.cob,1.2.canna,$(SAMPLEDIR))
 InstallNamedNonExec(fixer.cob,fixer.canna,$(SAMPLEDIR))
@@ -38,7 +38,7 @@
 InstallNamedNonExec(vje.cob,vje.canna,$(SAMPLEDIR))
 InstallNamedNonExec(wx2+.cob,wx2+.canna,$(SAMPLEDIR))
 #else
-InstallNamedNonExec(default.ca,default.canna,$(cannaLibDir))
+InstallNamedNonExec(default.ca,default.canna,$(cannaShareDir))
 InstallNonExec(1.1.canna,$(SAMPLEDIR))
 InstallNonExec(1.2.canna,$(SAMPLEDIR))
 InstallNonExec(fixer.canna,$(SAMPLEDIR))
diff -urN Canna36p3.ORIG/server/server.man Canna36p3/server/server.man
--- Canna36p3.ORIG/server/server.man	2002-11-01 17:32:06.000000000 +0900
+++ Canna36p3/server/server.man	2003-05-05 19:21:54.000000000 +0900
@@ -20,7 +20,7 @@
 .\" PERFORMANCE OF THIS SOFTWARE. 
 .\"
 .\" $Id: server.man,v 1.6 2002/11/01 08:32:06 zu2 Exp $
-.TH CANNASERVER 1M
+.TH CANNASERVER 1
 .SH "NAME"
 cannaserver \- Kana-Kanji conversion server
 .SH "SYNOPSIS"
@@ -28,19 +28,19 @@
 .SH "DESCRIPTION"
 .PP
 cannaserver provides the Kana-Kanji conversion service.  Usually all
-users on the network can access cannaserver(1M).  The scope of
+users on the network can access cannaserver(1).  The scope of
 accessing sources (hosts, users) can be restricted using the
 /etc/hosts.canna file.
 .PP
-After starting, cannaserver(1M) immediately forks and enters the
+After starting, cannaserver(1) immediately forks and enters the
 background process.  Thus, you do not need to use '&' to set the
 background process explicitly.
 .PP
-When starting, cannaserver(1M) creates the UNIX domain socket
+When starting, cannaserver(1) creates the UNIX domain socket
 @(UnixSockDir)/@(UnixSockName) to communicate clients.  The
-socket is deleted automatically at the end of cannaserver(1M) if no
-trouble occurs.  The socket may exist although cannaserver(1M) is
-not in operation (because cannaserver(1M) terminates abnormally, for
+socket is deleted automatically at the end of cannaserver(1) if no
+trouble occurs.  The socket may exist although cannaserver(1) is
+not in operation (because cannaserver(1) terminates abnormally, for
 example).  In this case, delete the file manually.
 .PP
 Specification of the dictionary to use is determined by referencing
@@ -50,25 +50,25 @@
 .IP "\fIddpath\fP" 12
 Specifies the directory that contains the dictionary.
 .IP "\-inet" 12
-The remote host can connect with the cannaserver(1M) using
+The remote host can connect with the cannaserver(1) using
 the inet domain socket.
-By the default, the cannaserver(1M) uses only UNIX domain socket.
+By the default, the cannaserver(1) uses only UNIX domain socket.
 i.e. Only clients in your local machine can connect to the server.
 .IP "\-l \fIn\fP" 12
-Outputs the cannaserver(1M) log to /tmp/canna.log.  n (1 to 5)
+Outputs the cannaserver(1) log to /tmp/canna.log.  n (1 to 5)
 specifies the log level.
 .IP "\-d" 12
-Starts cannaserver(1M) in debug mode.  It is executed as the
+Starts cannaserver(1) in debug mode.  It is executed as the
 foreground process.  Log data at log level 5 is output to the standard
 output.
 .IP "\-u \fIuserid\fP" 12
-Specifies which user cannaserver(1M) runs as.  By default, cannaserver(1M)
-runs as the user who started cannaserver(1M).
+Specifies which user cannaserver(1) runs as.  By default, cannaserver(1)
+runs as the user who started cannaserver(1).
 .IP "\-syslog" 12
-Outputs the cannaserver(1M) log to syslogd(8).
+Outputs the cannaserver(1) log to syslogd(8).
 .IP "\-p \fInum\fP" 12
-If specified, cannaserver(1M) will use the port number 5680 + num.  By
-default, cannaserver(1M) uses the port number 5680.
+If specified, cannaserver(1) will use the port number 5680 + num.  By
+default, cannaserver(1) uses the port number 5680.
 .SH "FILES"
 .IP "@(UnixSockDir)/@(UnixSockName)[:num]" 36
 Socket
@@ -78,4 +78,4 @@
 Access control file
 .SH "SEE ALSO"
 .PP
-cannakill(1M), cannastat(1), cannacheck(1)
+cannakill(1), cannastat(1), cannacheck(1)
