--- configure.in.orig	2006-05-04 00:49:06.000000000 -0500
+++ configure.in	2006-05-04 00:50:20.000000000 -0500
@@ -109,12 +109,13 @@
 AC_SUBST(CAPPLET_CFLAGS)
 AC_SUBST(CAPPLET_LIBS)
 
-PKG_CHECK_MODULES([NOTIFY], [libnotify >= 0.3.0], [enable_notify=yes],
-			    [enable_notify=no])
-if test "x$enable_notify" != "xno"; then
-   AC_DEFINE_UNQUOTED([ENABLE_NOTIFY], 1,
-   [Enable notifications with libnotify])
-fi
+dnl Hard disable libnotify on Gentoo for the 2.14 release
+dnl PKG_CHECK_MODULES([NOTIFY], [libnotify >= 0.3.0], [enable_notify=yes],
+dnl 			    [enable_notify=no])
+dnl if test "x$enable_notify" != "xno"; then
+dnl   AC_DEFINE_UNQUOTED([ENABLE_NOTIFY], 1,
+dnl    [Enable notifications with libnotify])
+dnl fi
 
 AC_PATH_PROG(GCONFTOOL, gconftool-2)
 AM_GCONF_SOURCE_2
