--- ./src/tcs/Makefile.am.orig	2005-01-28 08:19:35.000000000 +0930
+++ ./src/tcs/Makefile.am	2005-02-06 17:28:55.191846984 +0930
@@ -1,7 +1,7 @@
 noinst_LIBRARIES=libtcs.a
 
 AM_OBJS=atmel.o cache.o cxt.o tcscm.o tcsem.o imaem.o tcskcm.o tcspbg.o tcs_utils.o log.o ps/ps_utils.o ps/tcsps.o req_mgr.o auth_mgr.o
-AM_CFLAGS=-DAPPID=\"TCS\" -DVAR_PREFIX=\"@localstatedir@\" -DETC_PREFIX=\"@sysconfdir@\"
+AM_CFLAGS=-DAPPID=\"TCS\" -DVAR_PREFIX=\"@localstatedir@/state\" -DETC_PREFIX=\"@sysconfdir@\"
 
 .c.o:
 	$(CC) $(CFLAGS) $(AM_CFLAGS) -c -o $@ $<
--- ./src/tcsd/Makefile.am.orig	2004-12-11 01:07:06.000000000 +0930
+++ ./src/tcsd/Makefile.am	2005-02-06 17:28:55.191846984 +0930
@@ -1,5 +1,5 @@
 sbin_PROGRAMS=tcsd
 
 tcsd_SOURCES=svrside.c tcsd_wrap.c tcsd_conf.c tcsd_threads.c
-tcsd_CFLAGS=-DAPPID=\"TCSD\" -DVAR_PREFIX=\"@localstatedir@\" -DETC_PREFIX=\"@sysconfdir@\"
+tcsd_CFLAGS=-DAPPID=\"TCSD\" -DVAR_PREFIX=\"@localstatedir@/state\" -DETC_PREFIX=\"@sysconfdir@\"
 tcsd_LDFLAGS=../tcs/libtcs.a ../tddl/libtddl.a -lpthread
--- ./src/tspi/Makefile.am.orig	2005-01-28 08:17:36.000000000 +0930
+++ ./src/tspi/Makefile.am	2005-02-06 17:28:55.191846984 +0930
@@ -1,6 +1,6 @@
 all: libtspi.so
 
-AM_CFLAGS=-I../include -DAPPID=\"TSPI\" -DVAR_PREFIX=\"@localstatedir@\" -DETC_PREFIX=\"@sysconfdir@\" @GTK_CFLAGS@
+AM_CFLAGS=-I../include -DAPPID=\"TSPI\" -DVAR_PREFIX=\"@localstatedir@/state\" -DETC_PREFIX=\"@sysconfdir@\" @GTK_CFLAGS@
 AM_LDFLAGS=@GTK_LIBS@
 
 AM_OBJS=ConvertLib.o obj.o spi_context.o spi_getset.o spi_key.o spi_policy.o spi_utils.o keyreg.o memmgr.o secrets.o spi_data.o spi_hash.o spi_pcr.o spi_tpm.o ../tcsd_api/calltcsapi.o ../tcsd_api/tcstp.o ../tcsd_api/hosttable.o ../tcsd_api/clntside.o gtk/main.o gtk/support.o gtk/interface.o gtk/callbacks.o crypto/@CRYPTO_PACKAGE@/crypto.o ps/tspps.o ps/ps_utils.o log.o
--- ./dist/tcsd.conf.in.orig	2005-02-06 17:32:11.449871920 +0930
+++ ./dist/tcsd.conf.in	2005-02-06 17:28:55.192845832 +0930
@@ -26,7 +26,7 @@
 # Values: Any absolute directory path
 # Description: Path where the tcsd creates its persistent storage file.
 #
-# system_ps_file = @localstatedir@/tpm/system.data
+# system_ps_file = @localstatedir@/state/tpm/system.data
 #
 
 # Option: firmware_log_file
--- ./dist/Makefile.am.orig	2005-01-21 07:08:52.000000000 +0930
+++ ./dist/Makefile.am	2005-02-06 17:28:55.192845832 +0930
@@ -12,9 +12,9 @@
 	/usr/sbin/groupadd tss || true
 	/usr/sbin/useradd -r tss -g tss || true
 	/bin/chown tss:tss ${DESTDIR}/@sbindir@/tcsd
-	/bin/sh -c 'if [ ! -e ${DESTDIR}/@localstatedir@/tpm ];then mkdir -p ${DESTDIR}/@localstatedir@/tpm; fi'
-	/bin/chown tss:tss ${DESTDIR}/@localstatedir@/tpm
-	/bin/chmod 1777 ${DESTDIR}/@localstatedir@/tpm
+	/bin/sh -c 'if [ ! -e ${DESTDIR}/@localstatedir@/state/tpm ];then mkdir -p ${DESTDIR}/@localstatedir@/tpm; fi'
+	/bin/chown tss:tss ${DESTDIR}/@localstatedir@/state/tpm
+	/bin/chmod 1777 ${DESTDIR}/@localstatedir@/state/tpm
 
 uninstall-hook:
 	/usr/sbin/groupdel tss
