diff -ur tclap-1.1.0.orig/include/tclap/MultiArg.h tclap-1.1.0/include/tclap/MultiArg.h
--- tclap-1.1.0.orig/include/tclap/MultiArg.h	2005-09-11 02:25:43.000000000 +0300
+++ tclap-1.1.0/include/tclap/MultiArg.h	2009-07-24 21:20:59.000000000 +0300
@@ -23,6 +23,8 @@
 #ifndef TCLAP_MULTIPLE_ARGUMENT_H
 #define TCLAP_MULTIPLE_ARGUMENT_H
 
+#include <cstdio>
+
 #include <string>
 #include <vector>
 
diff -ur tclap-1.1.0.orig/include/tclap/ValueArg.h tclap-1.1.0/include/tclap/ValueArg.h
--- tclap-1.1.0.orig/include/tclap/ValueArg.h	2005-09-11 02:25:43.000000000 +0300
+++ tclap-1.1.0/include/tclap/ValueArg.h	2009-07-24 21:20:46.000000000 +0300
@@ -23,6 +23,8 @@
 #ifndef TCLAP_VALUE_ARGUMENT_H
 #define TCLAP_VALUE_ARGUMENT_H
 
+#include <cstdio>
+
 #include <string>
 #include <vector>
 
