diff -urNd ncurses-5.2/c++/cursesw.cc ncurses-5.2/c++/cursesw.cc.new
--- ncurses-5.2/c++/cursesw.cc	Sat May 25 12:23:26 2002
+++ ncurses-5.2/c++/cursesw.cc.new	Sat May 25 12:22:51 2002
@@ -51,8 +51,7 @@
     char buf[BUFSIZ];
     int result = wgetstr(w, buf);
     if (result == OK) {
-	strstreambuf ss(buf, sizeof(buf));
-	result = ss.vscan(fmt, (_IO_va_list)args);
+	result = vsscanf(buf, fmt, args);
     }
     va_end(args);
     return result;
@@ -73,8 +72,7 @@
     if (result == OK) {
 	result = wgetstr(w, buf);
 	if (result == OK) {
-	    strstreambuf ss(buf, sizeof(buf));
-	    result = ss.vscan(fmt, (_IO_va_list)args);
+	    result = vsscanf(buf, fmt, args);
 	}
     }
     va_end(args);
