diff -urN pd-0.35-0/extra/fiddle~/fiddle~.c pd-0.35-0-modified/extra/fiddle~/fiddle~.c
--- pd-0.35-0/extra/fiddle~/fiddle~.c	Mon Jun 10 21:10:24 2002
+++ pd-0.35-0-modified/extra/fiddle~/fiddle~.c	Wed Sep 25 13:33:42 2002
@@ -693,11 +693,11 @@
 
     for (npitch = 0; npitch < x->x_npitch; npitch++)
     {
-	int index;
+	int index_gentoo;
 	float best;
 	if (npitch)
 	{
-	    for (best = 0, index = -1, j=1; j < maxbin-1; j++)
+	    for (best = 0, index_gentoo = -1, j=1; j < maxbin-1; j++)
 	    {
 		if (histogram[j] > best && histogram[j] > histogram[j-1] &&
 		    histogram[j] > histogram[j+1])
@@ -717,7 +717,7 @@
 			if (histogram[j + sigfiddle_intpartialonset[k]]
 			    > histogram[j]) goto peaknogood;
 		    }
-		    index = j;
+		    index_gentoo = j;
 		    best = histogram[j];
 		}
 	    peaknogood: ;
@@ -725,13 +725,13 @@
 	}
 	else
 	{
-	    for (best = 0, index = -1, j=0; j < maxbin; j++)
+	    for (best = 0, index_gentoo = -1, j=0; j < maxbin; j++)
 		if (histogram[j] > best)
-		    index = j,  best = histogram[j];
+		    index_gentoo = j,  best = histogram[j];
 	}
-	if (index < 0) break;
+	if (index_gentoo < 0) break;
 	histvec[npitch].h_value = best;
-	histvec[npitch].h_index = index;
+	histvec[npitch].h_index = index_gentoo;
     }
 #if 1
     if (x->x_nprint)
@@ -1632,8 +1632,7 @@
 {
      if (sp[0]->s_n > x->x_hop) {
     	x->x_downsample = sp[0]->s_n / x->x_hop;
-    	post("* warning: fiddle~: will downsample input by
-%ld",x->x_downsample);
+    	post("* warning: fiddle~: will downsample input by %ld",x->x_downsample);
     	x->x_sr = sp[0]->s_sr / x->x_downsample;
     } else {
     	x->x_downsample = 1;
diff -urN pd-0.35-0/src/makefile.in pd-0.35-0-modified/src/makefile.in
--- pd-0.35-0/src/makefile.in	Fri May 24 12:08:40 2002
+++ pd-0.35-0-modified/src/makefile.in	Wed Sep 25 14:03:37 2002
@@ -12,6 +12,8 @@
 INSTALL_PREFIX = @prefix@
 GFLAGS = -DINSTALL_PREFIX=\"$(INSTALL_PREFIX)\"
 
+MANDIR = @mandir@
+
 # ALSA compilation
 
 SOUND_ALSA = @alsa@
@@ -164,13 +166,14 @@
 	cd ../extra/pique;make @EXTERNTARGET@
 
 INSTDIR = $(DESTDIR)/$(INSTALL_PREFIX)
+MANINSTDIR = $(DESTDIR)/$(MANDIR)
 install: 
 	install -d $(INSTDIR)/lib/pd/bin
 	install $(BIN_DIR)/$(GUINAME) $(INSTDIR)/lib/pd/bin/$(GUINAME)
 	install $(BIN_DIR)/pd-watchdog $(INSTDIR)/lib/pd/bin/pd-watchdog
-	install -m644 $(BIN_DIR)/pd.tk $(INSTDIR)/lib/pd/bin/pd.tk
+	install -m 644 $(BIN_DIR)/pd.tk $(INSTDIR)/lib/pd/bin/pd.tk
 	install -d $(INSTDIR)/bin
-	install -m755 $(PDEXEC) $(INSTDIR)/bin/pd
+	install -m 755 $(PDEXEC) $(INSTDIR)/bin/pd
 	install -m 755 $(BIN_DIR)/pdsend $(INSTDIR)/bin/pdsend
 	install -m 755 $(BIN_DIR)/pdreceive $(INSTDIR)/bin/pdreceive 
 	install -d $(INSTDIR)/lib/pd/extra
@@ -178,13 +181,14 @@
 	install -m 644 $(EXTERNS) $(INSTDIR)/lib/pd/extra
 	cp -r ../doc $(INSTDIR)/lib/pd/
 	install -d $(INSTDIR)/include
-	install -m644 m_pd.h $(INSTDIR)/include/m_pd.h
-	gzip < ../man/pd.1 >  $(INSTDIR)/man/man1/pd.1.gz
-	chmod 644 $(INSTDIR)/man/man1/pd.1.gz
-	gzip < ../man/pdsend.1 >  $(INSTDIR)/man/man1/pdsend.1.gz
-	chmod 644 $(INSTDIR)/man/man1/pdsend.1.gz
-	gzip < ../man/pdreceive.1 >  $(INSTDIR)/man/man1/pdreceive.1.gz
-	chmod 644 $(INSTDIR)/man/man1/pdreceive.1.gz
+	install -m 644 m_pd.h $(INSTDIR)/include/m_pd.h
+	install -d $(MANINSTDIR)/man1
+	gzip < ../man/pd.1 >  $(MANINSTDIR)/man1/pd.1.gz
+	chmod 644 $(MANINSTDIR)/man1/pd.1.gz
+	gzip < ../man/pdsend.1 >  $(MANINSTDIR)/man1/pdsend.1.gz
+	chmod 644 $(MANINSTDIR)/man1/pdsend.1.gz
+	gzip < ../man/pdreceive.1 >  $(MANINSTDIR)/man1/pdreceive.1.gz
+	chmod 644 $(MANINSTDIR)/man1/pdreceive.1.gz
 
 local-clean:
 	-rm -f ../obj/* $(BIN_DIR)/pd $(BIN_DIR)/$(GUINAME) $(BIN_DIR)/pdsend \
