diff -Naur vmd-1.8.7.old/configure vmd-1.8.7/configure
--- vmd-1.8.7.old/configure	2009-08-01 03:19:34.000000000 -0400
+++ vmd-1.8.7/configure	2009-08-05 09:30:33.000000000 -0400
@@ -2804,8 +2804,7 @@
 	-\$(COPY) ../data/.vmdrc ../data/.vmdsensors ../data/vmd_completion.dat "$install_library_dir"
 	\$(CD) $vmd_bin_dir ; \\
 	if [ -f run_vmd_tmp ]; then \$(DELETE) run_vmd_tmp; fi ; \\
-	if [ ! -x "/bin/csh" ]; then \\
-		\$(ECHO) "Info: /bin/csh shell not found, installing Bourne shell startup script instead" ; \\
+	if [ -x "/bin/bash" ]; then \\
 		\$(ECHO) '#!/bin/sh' >> run_vmd_tmp ; \\
 		\$(ECHO) 'defaultvmddir="$install_library_dir"' >> run_vmd_tmp ; \\
 		\$(ECHO) 'vmdbasename=vmd' >> run_vmd_tmp ; \\
