The \".*\" case is duplicated, remove the useless version ;).

--- pcalc-000/pcalcl.l
+++ pcalc-000/pcalcl.l
@@ -287,16 +287,6 @@
                                 #endif
                                 }
 
-\".*\"                          {
-                                #ifdef TEST
-                                printf(" String:  [ %s ]\n", yytext);
-                                #endif
-
-                                #ifdef RET_EACH
-                                //return(STR);
-                                #endif
-                                }
-
 ";"                             {
                                 count();
 
