--- sqlgui/main.cpp	Sat Jul 13 17:21:37 2002
+++ sqlgui/main.cpp	Wed Aug 21 17:31:37 2002
@@ -16,7 +16,9 @@
  ***************************************************************************/
 
 
+using namespace std;
 
+#include <iostream>
 
 #include <qdir.h>
 
--- sqlgui/sqlgui.cpp	Sat Jul 13 17:20:58 2002
+++ sqlgui/sqlgui.cpp	Wed Aug 21 18:17:06 2002
@@ -14,6 +14,8 @@
  *   (at your option) any later version.                                   *
  *                                                                         *
  ***************************************************************************/
+using namespace std;
+ 
 #include <iostream>
 #include <iomanip>
 
