From 167364225f23fd7261c50f3b0e5324e511ddc556 Mon Sep 17 00:00:00 2001
From: Michael Krelin <hacker@klever.net>
Date: Fri, 22 Jun 2007 13:41:07 +0200
Subject: [PATCH] make automake happy

---
 dotconf.m4 |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/dotconf.m4 b/dotconf.m4
index 08c46b8..0f14a35 100644
--- a/dotconf.m4
+++ b/dotconf.m4
@@ -2,7 +2,7 @@
 dnl ## AM_PATH_DOTCONF([MINIMUM_VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT_FOUND]]])
 dnl ##
 dnl ##
-AC_DEFUN(AM_PATH_DOTCONF,
+AC_DEFUN([AM_PATH_DOTCONF],
 [dnl
 dnl Get the cflags and libs from the dotconf-config script
 dnl
-- 
1.5.2.2

