--- Makefile.in.old     2009-02-11 09:30:12.000000000 +0100
+++ Makefile.in 2009-02-11 09:30:28.000000000 +0100
@@ -120,7 +120,7 @@
 
 XPP_LIBS = $(NET_LIBS) pgame/libxpp_game.a pai/libxpp_ai.a
 XPP_NET_LIBS = pnet/libxpp_net.a	# This must be after interface
-XPP_SOUND_LIBS = $(LIBMIKMOD_LIBS) psound/libxpp_sound.a
+XPP_SOUND_LIBS = psound/libxpp_sound.a $(LIBMIKMOD_LIBS)
 
 @GTK_TRUE@xpuyopuyo_LDADD = $(XPP_LIBS) pgtk/libxpp_interface.a $(XPP_NET_LIBS) $(XPP_SOUND_LIBS) $(GTK_LIBS) pgame/libxpp_game.a putil/libxpp_util.a
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
