--- configure.ac.orig	2011-05-16 12:27:31.759991392 +1000
+++ configure.ac	2011-05-16 12:27:56.026801099 +1000
@@ -1,6 +1,6 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
-define([QD_PATCH_VERSION], 7)
+define([QD_PATCH_VERSION], 11)
 AC_PREREQ(2.60)
 AC_INIT(qd, 2.3.QD_PATCH_VERSION, yozo@cs.berkeley.edu)
 AC_CONFIG_SRCDIR([src/qd_real.cpp])
