--- src/etats_latex.c.orig	2005-01-16 17:15:46.349575352 +0100
+++ src/etats_latex.c	2005-01-16 17:16:19.340559960 +0100
@@ -245,7 +245,7 @@
 	     "\\documentclass{article}\n\n"
 	     "\\special{! TeXDict begin /landplus90{true}store end }\n"
 	     "\\usepackage{a4}\n"
-	     "\\usepackage[utf8]{inputenc}\n"
+	     "\\usepackage[utf8x]{inputenc}\n"
 	     "\\usepackage{eurosym}\n"
 	     "\\usepackage{boxedminipage}\n"
 	     "\\usepackage{longtable}\n"
