--- Yap-5.1.1.orig/Makefile.in	2006-04-13 23:34:33.000000000 +1200
+++ Yap-5.1.1/Makefile.in	2007-01-21 10:52:46.000000000 +1300
@@ -527,7 +527,7 @@
 	$(RANLIB) libYap.a
 
 @IN_UNIX@@DO_SECOND_LD@libYap@SHLIB_SUFFIX@: $(LIB_OBJECTS)
-@IN_UNIX@@DO_SECOND_LD@	@SHLIB_LD@ -o libYap@SHLIB_SUFFIX@ $(LIB_OBJECTS) $(LIBS)
+@IN_UNIX@@DO_SECOND_LD@	@SHLIB_LD@ -soname=$@ -o libYap@SHLIB_SUFFIX@ $(LIB_OBJECTS) $(LIBS)
 
 install: install_bin install_data
 
