diff -ur yap-6.orig/packages/PLStream/Makefile.in yap-6/packages/PLStream/Makefile.in
--- yap-6.orig/packages/PLStream/Makefile.in	2010-10-19 20:40:59.000000000 +1300
+++ yap-6/packages/PLStream/Makefile.in	2010-10-19 20:40:04.000000000 +1300
@@ -20,7 +20,7 @@
 #
 #
 CC=@CC@
-CFLAGS= @SHLIB_CFLAGS@ $(YAP_EXTRAS) $(DEFS)  -D_YAP_NOT_INSTALLED_=1 -I$(srcdir) -I../.. -I$(srcdir)/../../include  @CPPFLAGS@
+CFLAGS= @SHLIB_CFLAGS@ $(YAP_EXTRAS) $(DEFS) -DHAVE_CONFIG_H=1 -D_YAP_NOT_INSTALLED_=1 -I$(srcdir) -I../.. -I$(srcdir)/../../include  @CPPFLAGS@
 LDFLAGS=@LDFLAGS@
 #
 #
diff -ur yap-6.orig/packages/semweb/Makefile.in yap-6/packages/semweb/Makefile.in
--- yap-6.orig/packages/semweb/Makefile.in	2010-10-19 20:40:59.000000000 +1300
+++ yap-6/packages/semweb/Makefile.in	2010-10-19 20:40:24.000000000 +1300
@@ -37,7 +37,7 @@
 #
 #
 CC=@CC@
-CFLAGS= @SHLIB_CFLAGS@ $(YAP_EXTRAS) $(DEFS) -D_YAP_NOT_INSTALLED_=1 -I$(srcdir) -I../.. -I$(srcdir)/../../include @CPPFLAGS@
+CFLAGS= @SHLIB_CFLAGS@ $(YAP_EXTRAS) $(DEFS) -DHAVE_CONFIG_H=1 -D_YAP_NOT_INSTALLED_=1 -I$(srcdir) -I../.. -I$(srcdir)/../../include @CPPFLAGS@
 MKINDEX=true
 
 
diff -ur yap-6.orig/packages/sgml/Makefile.in yap-6/packages/sgml/Makefile.in
--- yap-6.orig/packages/sgml/Makefile.in	2010-10-19 20:40:59.000000000 +1300
+++ yap-6/packages/sgml/Makefile.in	2010-10-19 20:40:45.000000000 +1300
@@ -29,7 +29,7 @@
 #
 #
 CC=@CC@
-CFLAGS= @SHLIB_CFLAGS@ $(YAP_EXTRAS) $(DEFS) -D_YAP_NOT_INSTALLED_=1 -I$(srcdir) -I../.. -I$(srcdir)/../../include @CPPFLAGS@
+CFLAGS= @SHLIB_CFLAGS@ $(YAP_EXTRAS) $(DEFS) -DHAVE_CONFIG_H=1 -D_YAP_NOT_INSTALLED_=1 -I$(srcdir) -I../.. -I$(srcdir)/../../include @CPPFLAGS@
 LDFLAGS=@LDFLAGS@
 MKINDEX=true
 
