--- yap-6.orig/library/tries/itries.c	2010-08-05 05:36:28.000000000 +1200
+++ yap-6/library/tries/itries.c	2010-09-12 17:20:09.000000000 +1200
@@ -620,7 +620,7 @@
   /* check args */
   if (!YAP_IsVarTerm(arg_itrie)) 
     return FALSE;
-  if (!(file = (FILE*) Yap_FileDescriptorFromStream(arg_stream)))
+  if (!(file = (FILE*) YAP_FileDescriptorFromStream(arg_stream)))
     return FALSE;
 
   /* load itrie */
