diff -Naur ircd-hybrid-7.0.orig/help/Makefile.in ircd-hybrid-7.0/help/Makefile.in
--- ircd-hybrid-7.0.orig/help/Makefile.in	2002-02-02 16:01:13.000000000 +0000
+++ ircd-hybrid-7.0/help/Makefile.in	2003-10-10 20:46:34.000000000 +0000
@@ -80,10 +80,6 @@
 		$(INSTALL_DATA) $$help $(uhelpdir); \
 	fi \
 	done
-	@for link in $(SYMLINKS); do \
-		rm -f $(uhelpdir)/$$link; \
-		ln -s $(ohelpdir)/$$link $(uhelpdir); \
-	done
 
 
 distclean:
diff -Naur ircd-hybrid-7.0.orig/include/Makefile.in ircd-hybrid-7.0/include/Makefile.in
--- ircd-hybrid-7.0.orig/include/Makefile.in	2002-04-29 22:30:24.000000000 +0000
+++ ircd-hybrid-7.0/include/Makefile.in	2003-10-10 20:46:34.000000000 +0000
@@ -10,6 +10,7 @@
 exec_prefix=	@exec_prefix@
 bindir=		@bindir@
 sysconfdir=	@sysconfdir@
+includedir=	@includedir@
 
 all:
 build:
@@ -19,13 +20,12 @@
 
 install:
 	@echo installing header files...
-	-@if test ! -d $(prefix)/include; then \
-		echo "mkdir $(prefix)/include"; \
-		mkdir $(prefix)/include; \
-		mkdir $(prefix)/include/ircd-hybrid-7; \
+	-@if test ! -d $(includedir); then \
+		echo "mkdir -p $(includedir)"; \
+		mkdir -p $(includedir); \
 	fi
 	@for header in *.h; do \
-		${INSTALL_DATA} $$header $(prefix)/include/ircd-hybrid-7; \
+		${INSTALL_DATA} $$header $(includedir); \
 	done
 
 distclean:
diff -Naur ircd-hybrid-7.0.orig/src/Makefile.in ircd-hybrid-7.0/src/Makefile.in
--- ircd-hybrid-7.0.orig/src/Makefile.in	2003-05-29 03:02:15.000000000 +0000
+++ ircd-hybrid-7.0/src/Makefile.in	2003-10-10 20:46:34.000000000 +0000
@@ -140,10 +140,6 @@
 		echo "mkdir $(sysconfdir)"; \
 		mkdir $(sysconfdir); \
 	fi
-	-@if test ! -d $(localstatedir)/logs; then \
-		echo "mkdir $(localstatedir)/logs"; \
-		mkdir $(localstatedir)/logs; \
-	fi
 
 install: install-mkdirs build
 	@for i in $(PROGS); do \
diff -Naur ircd-hybrid-7.0.orig/tools/Makefile.in ircd-hybrid-7.0/tools/Makefile.in
--- ircd-hybrid-7.0.orig/tools/Makefile.in	2002-07-11 23:40:27.000000000 +0000
+++ ircd-hybrid-7.0/tools/Makefile.in	2003-10-10 20:46:34.000000000 +0000
@@ -129,10 +129,6 @@
 		echo "mkdir $(sysconfdir)"; \
 		mkdir $(sysconfdir); \
 	fi
-	-@if test ! -d $(localstatedir)/logs; then \
-		echo "mkdir $(localstatedir)/logs"; \
-		mkdir $(localstatedir)/logs; \
-	fi
 
 install: install-mkdirs build
 	@for i in $(PROGS); do \
@@ -144,6 +140,4 @@
 		$(INSTALL_BIN) $$i $(bindir); \
 	done; \
         $(RM) -f $(bindir)/vimotd $(bindir)/viklines; \
-        $(LN) $(bindir)/viconf $(bindir)/vimotd; \
-        $(LN) $(bindir)/viconf $(bindir)/viklines
 
