--- src/ExportDialog.cc.orig	2008-06-19 12:01:25.000000000 +0100
+++ src/ExportDialog.cc	2008-06-19 12:02:44.000000000 +0100
@@ -28,7 +28,9 @@
 #ifdef HAVE_NETCDF
 #include <netcdf.h>
 #endif
+#ifdef HAVE_AUDIOFILE
 #include <audiofile.h>
+#endif
 #ifdef HAVE_MAGICK
 #include <Magick++.h>
 using namespace Magick;
--- src/LabPlotDialog.cc.orig	2007-10-27 15:18:59.000000000 +0100
+++ src/LabPlotDialog.cc	2008-06-19 12:08:11.000000000 +0100
@@ -14,7 +14,9 @@
 #include <jasper/jas_config.h>
 #endif
 
+#ifdef HAVE_AUDIOFILE
 #include <audiofile.h>
+#endif
 
 #ifdef HAVE_GSL
 #include <gsl/gsl_version.h>
@@ -186,11 +188,15 @@
 	// audiofile
 	hb = new QHBox(vbox);
 	new QLabel(i18n("audiofile library : "),hb);
+#ifdef HAVE_AUDIOFILE
 	st = QString::number(LIBAUDIOFILE_MAJOR_VERSION)+'.'+QString::number(LIBAUDIOFILE_MINOR_VERSION);
 #ifdef LIBAUDIOFILE_MICRO_VERSION
 	st += '.'+QString::number(LIBAUDIOFILE_MICRO_VERSION);
 #endif
 	tmp = new QLabel(st,hb);
+#else
+	tmp = new QLabel("no",hb);
+#endif
 	tmp->setAlignment(Qt::AlignRight);
 
 	// R
