diff -uNr asterisk-1.0.5.orig/codecs/gsm/Makefile asterisk-1.0.5/codecs/gsm/Makefile
--- asterisk-1.0.5.orig/codecs/gsm/Makefile	2005-02-19 21:49:10.060313408 +0100
+++ asterisk-1.0.5/codecs/gsm/Makefile	2005-02-19 21:49:46.176822864 +0100
@@ -217,12 +217,16 @@
 ifneq ($(shell uname -m),ppc)
 ifneq ($(shell uname -m),alpha)
 ifneq ($(shell uname -m),sparc64)
+ifneq ($(shell uname -m),parisc)
+ifneq ($(shell uname -m),parisc64)
 GSM_SOURCES+= $(SRC)/k6opt.s
 endif
 endif
 endif
 endif
 endif
+endif
+endif
 
 TOAST_SOURCES = $(SRC)/toast.c 		\
 		$(SRC)/toast_lin.c	\
@@ -273,12 +277,16 @@
 ifneq ($(shell uname -m), ppc)
 ifneq ($(shell uname -m), alpha)
 ifneq ($(shell uname -m), sparc64)
+ifneq ($(shell uname -m), parisc)
+ifneq ($(shell uname -m), parisc64)
 GSM_OBJECTS+= $(SRC)/k6opt.o
 endif
 endif
 endif
 endif
 endif
+endif
+endif
 
 TOAST_OBJECTS =	$(SRC)/toast.o 		\
 		$(SRC)/toast_lin.o	\
