--- asterisk-1.0.7/contrib/scripts/addmailbox.orig	2005-05-11 23:17:10.000000000 +0200
+++ asterisk-1.0.7/contrib/scripts/addmailbox	2005-05-11 23:21:57.000000000 +0200
@@ -21,14 +21,14 @@
 	echo
 	echo "New Voicemail context.."
 	echo "Creating Voicemail context directory..."
-	mkdir -p ${VMHOME}/${context}
+	mkdir -m 750 -p ${VMHOME}/${context}
 fi
 
 echo
 echo "Creating Voicemail directory..." 
-mkdir -p ${VMHOME}/${context}/${mailbox}
+mkdir -m 750 -p ${VMHOME}/${context}/${mailbox}
 echo "Creating INBOX..."
-mkdir -p ${VMHOME}/${context}/${mailbox}/INBOX
+mkdir -m 750 -p ${VMHOME}/${context}/${mailbox}/INBOX
 echo "Creating Default greetings..."
 cat ${SNDHOME}/vm-theperson.gsm > ${VMHOME}/${context}/${mailbox}/unavail.gsm
 cat ${SNDHOME}/vm-theperson.gsm > ${VMHOME}/${context}/${mailbox}/busy.gsm
@@ -41,5 +41,8 @@
 done
 cat ${SNDHOME}/vm-isunavail.gsm >> ${VMHOME}/${context}/${mailbox}/unavail.gsm
 cat ${SNDHOME}/vm-isonphone.gsm >> ${VMHOME}/${context}/${mailbox}/busy.gsm
+echo "Changing permissions..."
+chmod -R asterisk:asterisk ${VMHOME}/${context}
+chmod    u=rwX,g=rX,o=     ${VMHOME}/${context}/${mailbox}/*.gsm
 echo "Complete."
 
--- asterisk-1.0.7/contrib/scripts/astgenkey.orig	2005-05-11 23:28:37.000000000 +0200
+++ asterisk-1.0.7/contrib/scripts/astgenkey	2005-05-11 23:30:12.000000000 +0200
@@ -55,7 +55,11 @@
 		echo "Key creation successful."
 		echo "Public key:  ${KEY}.pub"
 		echo "Private key: ${KEY}.key"
+		echo "Changing permissions..."
 	fi
+	chown root:asterisk ${KEY}.pub ${KEY}.key
+	chmod u=rw,g=r,o=   ${KEY}.pub ${KEY}.key
 else
 	echo "Unknown error creating keys."
 fi
+
