--- asterisk-1.0.8/codecs/gsm/Makefile.orig	2005-06-25 09:30:45.146479088 +0200
+++ asterisk-1.0.8/codecs/gsm/Makefile	2005-06-25 09:32:05.008338240 +0200
@@ -46,6 +46,8 @@
 ifneq (${PROC},sparc64)
 ifneq (${PROC},ppc)
 ifneq (${PROC},ppc64)
+ifneq ($(PROC),parisc)
+ifneq ($(PROC),parisc64)
 OPTIMIZE+=-march=$(PROC)
 endif
 endif
@@ -56,6 +58,8 @@
 endif
 endif
 endif
+endif
+endif
 
 #The problem with sparc is the best stuff is in newer versions of gcc (post 3.0) only.
 #This works for even old (2.96) versions of gcc and provides a small boost either way.
@@ -224,6 +228,8 @@
 ifneq ($(shell uname -m),alpha)
 ifneq ($(shell uname -m),armv4l)
 ifneq ($(shell uname -m),sparc64)
+ifneq ($(shell uname -m),parisc)
+ifneq ($(shell uname -m),parisc64)
 GSM_SOURCES+= $(SRC)/k6opt.s
 endif
 endif
@@ -231,6 +237,8 @@
 endif
 endif
 endif
+endif
+endif
 
 TOAST_SOURCES = $(SRC)/toast.c 		\
 		$(SRC)/toast_lin.c	\
@@ -282,6 +290,8 @@
 ifneq ($(shell uname -m), alpha)
 ifneq ($(shell uname -m), armv4l)
 ifneq ($(shell uname -m), sparc64)
+ifneq ($(shell uname -m), parisc)
+ifneq ($(shell uname -m), parisc64)
 GSM_OBJECTS+= $(SRC)/k6opt.o
 endif
 endif
@@ -289,6 +299,8 @@
 endif
 endif
 endif
+endif
+endif
 
 TOAST_OBJECTS =	$(SRC)/toast.o 		\
 		$(SRC)/toast_lin.o	\
