--- bsdsfv.c.orig	2003-04-16 16:05:16.000000000 -0400
+++ bsdsfv.c	2003-04-16 16:32:35.000000000 -0400
@@ -36,3 +36,2 @@
 
-
 #define FNAMELEN		250
@@ -42,2 +41,11 @@
 #define BADTAG 			".bad"
+#define OPENBRACKET             "\e[34;01m[\e[0m"
+#define CLOSEBRACKET            "\e[34;01m]\e[0m"
+#define OKCOLOR                 "\e[32;01m"
+#define OKMSG                   "OK"
+#define BADCOLOR                "\e[31;01m"
+#define BADMSG                  "BAD"
+#define MISSINGCOLOR            "\e[33;01m"
+#define MISSINGMSG              "MISSING"
+
 
@@ -525,3 +533,3 @@
 				{
-				  printf ("OK\n");
+				  printf ("%s %s%s %s\n", OPENBRACKET, OKCOLOR, OKMSG, CLOSEBRACKET);
 				  if (glftpdmode)
@@ -537,3 +545,3 @@
 				{
-				  printf ("BAD\n");
+				  printf ("%s %s%s %s\n", OPENBRACKET, BADCOLOR, BADMSG, CLOSEBRACKET);
 				  badfiles++;
@@ -551,3 +559,3 @@
 			{
-			  printf ("MISSING\n");
+			  printf ("%s %s%s %s\n", OPENBRACKET, MISSINGCOLOR, MISSINGMSG, CLOSEBRACKET);
 			  missingfiles++;
@@ -558,5 +566,7 @@
 
-	  printf ("\n%d file(s) tested - %d OK - %d bad - %d missing...\n",
-			  numfiles, numfiles - badfiles - missingfiles, badfiles,
-			  missingfiles);
+	  printf ("\n%d file(s) tested - %s %s%d %s %s - %s %s%d %s %s - %s %s%d %s %s ...\n\n",
+	  	numfiles,
+	  	OPENBRACKET, OKCOLOR, numfiles - (badfiles + missingfiles), OKMSG, CLOSEBRACKET,
+	  	OPENBRACKET, BADCOLOR, badfiles, BADMSG, CLOSEBRACKET,
+	  	OPENBRACKET, MISSINGCOLOR, missingfiles, MISSINGMSG, CLOSEBRACKET);
 
@@ -637,3 +647,3 @@
 					{
-					  printf ("OK\n");
+					  printf ("%s %s%s %s\n", OPENBRACKET, OKCOLOR, OKMSG, CLOSEBRACKET);
 					}
@@ -643,3 +653,3 @@
 						{
-						  printf ("MISSING\n");
+						  printf ("%s %s%s %s\n", OPENBRACKET, MISSINGCOLOR, MISSINGMSG, CLOSEBRACKET);
 						  missingfiles++;
@@ -648,3 +658,3 @@
 						{
-						  printf ("BAD\n");
+						  printf ("%s %s%s %s\n", OPENBRACKET, BADCOLOR, BADMSG, CLOSEBRACKET);
 						  badfiles++;
@@ -670,4 +680,5 @@
 			{
-			  printf ("Testing %s ... listed = 0x%08X ... Local = MISSING",
-						  sfvTable[cnt].filename, sfvTable[cnt].crc);
+			  printf ("Testing %s ... listed = 0x%08X ... local =            ... %s %s%s %s\n",
+						  sfvTable[cnt].filename, sfvTable[cnt].crc,
+						  OPENBRACKET, MISSINGCOLOR, MISSINGMSG, CLOSEBRACKET);
 			  missingfiles++;
@@ -716,6 +727,8 @@
 		}
-
-	  printf ("\n%d file(s) tested - %d OK - %d bad - %d missing ...\n\n",
-			  numfiles, numfiles - (badfiles + missingfiles), badfiles,
-			  missingfiles);
+			  
+	printf ("\n%d file(s) tested - %s %s%d %s %s - %s %s%d %s %s - %s %s%d %s %s ...\n\n",
+			numfiles,
+			OPENBRACKET, OKCOLOR, numfiles - (badfiles + missingfiles), OKMSG, CLOSEBRACKET,
+			OPENBRACKET, BADCOLOR, badfiles, BADMSG, CLOSEBRACKET,
+			OPENBRACKET, MISSINGCOLOR, missingfiles, MISSINGMSG, CLOSEBRACKET);
 
