http://www.riverbankcomputing.co.uk/hg/sip/rev/f9f4b161c940

--- siplib/siplib.c.in
+++ siplib/siplib.c.in
@@ -7604,9 +7604,10 @@
 #endif
 
         /*
-         * Check any possible reimplementation is not the wrapped C++ method.
+         * Check any possible reimplementation is not the wrapped C++ method or
+         * a default special method implementation..
          */
-        if (cls_dict != NULL && (cls_attr = PyDict_GetItem(cls_dict, mname_obj)) != NULL && Py_TYPE(cls_attr) != &sipMethodDescr_Type)
+        if (cls_dict != NULL && (cls_attr = PyDict_GetItem(cls_dict, mname_obj)) != NULL && Py_TYPE(cls_attr) != &sipMethodDescr_Type && Py_TYPE(cls_attr) != &PyWrapperDescr_Type)
         {
             reimp = cls_attr;
             break;
