--- Imakefile.in.old	Sat Sep  1 17:32:38 2001
+++ Imakefile.in	Sat Sep  1 17:32:57 2001
@@ -160,7 +160,7 @@
 	fi
 	$(LN) $(NAME).$(MANSUFFIX) $(DESTDIR)$(MANDIR)/ssh-askpass.$(MANSUFFIX)
 
-.PHONY: depends changelog realclean distclean spotless tar dist
+.PHONY: depends changelog realclean  spotless tar dist
 
 all:: $(MAN_OBJ)
 
@@ -197,11 +197,7 @@
 	-$(RM) $(CLASS).ad
 	-$(RM) *.LOG *.log
 
-distclean: realclean
-	-$(RM) Makefile Makefile.bak $(IMAKEFILE_OBJ) $(CHANGELOG)
 
-spotless: distclean
-	-$(RM) $(DIST)
 
 tar: dist
 dist: $(DIST)
