--- /dev/null	1970-01-01 01:00:00.000000000 +0100
+++ Makefile	2005-09-05 23:59:22.000000000 +0200
@@ -0,0 +1,27 @@
+.EXPORT_ALL_VARIABLES:
+
+CC=cc
+INSTALL=install
+AST_CFG=/usr/bin/asterisk-config
+
+AST_INCLUDE_DIR=/usr/include/asterisk
+AST_CONFIG_DIR=/etc/asterisk
+AST_MODULES_DIR=/usr/lib/asterisk/modules
+
+TARGETS:=app_rxfax.so app_txfax.so
+
+CFLAGS=$(shell $(AST_CFG) --cflags)
+SOLINK=$(shell $(AST_CFG) --solink)
+LIBS=-lspandsp -ltiff
+
+SUBDIR=$(shell [ -f $(AST_INCLUDE_DIR)/channel_pvt.h ] && echo "asterisk-1.0.x" || echo "asterisk-1.1.x")
+
+.PHONY: all install $(SUBDIR)
+
+all: $(SUBDIR)
+
+$(SUBDIR):
+	$(MAKE) -C $(SUBDIR) all
+
+install:
+	$(MAKE) -C $(SUBDIR) install
--- /dev/null	1970-01-01 01:00:00.000000000 +0100
+++ asterisk-1.0.x/Makefile	2005-09-06 00:00:05.000000000 +0200
@@ -0,0 +1,10 @@
+all: $(TARGETS)
+
+$(TARGETS): $(patsubst %.so,%.c,$(TARGETS))
+	$(CC) $(CFLAGS) $(SOLINK) -fPIC -o app_rxfax.so app_rxfax.c $(LIBS)
+	$(CC) $(CFLAGS) $(SOLINK) -fPIC -o app_txfax.so app_txfax.c $(LIBS)
+
+install:
+	for x in $(TARGETS); do \
+		$(INSTALL) -D -m 755 $$x $(DESTDIR)$(AST_MODULES_DIR)/$$x; \
+	done
--- /dev/null	1970-01-01 01:00:00.000000000 +0100
+++ asterisk-1.1.x/Makefile	2005-09-06 00:00:37.000000000 +0200
@@ -0,0 +1,10 @@
+all: $(TARGETS)
+
+$(TARGETS): $(patsubst %.so,%.c,$(TARGETS))
+	$(CC) $(CFLAGS) $(SOLINK) -fPIC -o app_rxfax.so app_rxfax.c $(LIBS)
+	$(CC) $(CFLAGS) $(SOLINK) -fPIC -o app_txfax.so app_txfax.c $(LIBS)
+
+install:
+	for x in $(TARGETS); do \
+		$(INSTALL) -D -m 755 $$x $(DESTDIR)$(AST_MODULES_DIR)/$$x; \
+	done
