diff --exclude='*.orig' --exclude='*.rej' -ruN poppler-0.5.1.orig/glib/Makefile.am poppler-0.5.1/glib/Makefile.am
--- poppler-0.5.1.orig/glib/Makefile.am	2006-03-03 19:37:27.000000000 +0100
+++ poppler-0.5.1/glib/Makefile.am	2006-03-03 19:38:15.000000000 +0100
@@ -79,7 +79,7 @@
 	poppler-private.h
 
 libpoppler_glib_la_LIBADD =				\
-	$(top_builddir)/poppler/libpoppler.la		\
+	-lpoppler					\
 	$(POPPLER_GLIB_LIBS)				\
 	$(FREETYPE_LIBS)				\
 	$(cairo_libs)
@@ -92,7 +92,7 @@
        test-poppler-glib.c
 
 test_poppler_glib_LDADD =			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-glib.la			\
 	$(POPPLER_GLIB_LIBS)			\
 	$(FREETYPE_LIBS)			\
diff --exclude='*.orig' --exclude='*.rej' -ruN poppler-0.5.1.orig/Makefile.am poppler-0.5.1/Makefile.am
--- poppler-0.5.1.orig/Makefile.am	2006-03-03 19:37:27.000000000 +0100
+++ poppler-0.5.1/Makefile.am	2006-03-03 19:37:33.000000000 +0100
@@ -29,7 +29,7 @@
 utils_subdir = utils
 endif
 
-SUBDIRS = goo fofi $(splash_subdir) poppler $(utils_subdir) $(glib_subdir) $(qt_subdir) test $(qt4_subdir)
+SUBDIRS = $(glib_subdir) $(qt_subdir) $(qt4_subdir)
 
 EXTRA_DIST =					\
 	README-XPDF				\
@@ -40,9 +40,7 @@
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA =				\
-	poppler.pc				\
 	$(cairo_pc_file)			\
-	$(splash_pc_file)			\
 	$(glib_pc_file)				\
 	$(qt_pc_file)                           \
 	$(qt4_pc_file)
diff --exclude='*.orig' --exclude='*.rej' -ruN poppler-0.5.1.orig/qt/Makefile.am poppler-0.5.1/qt/Makefile.am
--- poppler-0.5.1.orig/qt/Makefile.am	2006-03-03 19:37:27.000000000 +0100
+++ poppler-0.5.1/qt/Makefile.am	2006-03-03 19:37:33.000000000 +0100
@@ -22,7 +22,7 @@
 	poppler-private.h
 
 libpoppler_qt_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)
 
@@ -38,7 +38,7 @@
        test-poppler-qt.cpp
 
 test_poppler_qt_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-qt.la			\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)
diff --exclude='*.orig' --exclude='*.rej' -ruN poppler-0.5.1.orig/qt4/src/Makefile.am poppler-0.5.1/qt4/src/Makefile.am
--- poppler-0.5.1.orig/qt4/src/Makefile.am	2006-03-03 19:37:27.000000000 +0100
+++ poppler-0.5.1/qt4/src/Makefile.am	2006-03-03 19:37:33.000000000 +0100
@@ -23,7 +23,7 @@
 	poppler-private.h
 
 libpoppler_qt4_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(POPPLER_QT4_LIBS)
 
 libpoppler_qt4_la_LDFLAGS = -version-info 1:0:0
diff --exclude='*.orig' --exclude='*.rej' -ruN poppler-0.5.1.orig/qt4/tests/Makefile.am poppler-0.5.1/qt4/tests/Makefile.am
--- poppler-0.5.1.orig/qt4/tests/Makefile.am	2006-03-03 19:37:27.000000000 +0100
+++ poppler-0.5.1/qt4/tests/Makefile.am	2006-03-03 19:37:33.000000000 +0100
@@ -7,7 +7,7 @@
 	$(POPPLER_QTTEST_CXXFLAGS)
 
 LDADDS =					\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(POPPLER_QT4_LIBS)
 
