diff -Naur fritz.orig/src.drv/tools.c fritz/src.drv/tools.c
--- fritz.orig/src.drv/tools.c  2002-06-07 00:00:00.000000000 +0200
+++ fritz/src.drv/tools.c       2002-07-17 23:42:18.000000000 +0200
@@ -24,11 +24,11 @@
 #include <linux/vmalloc.h>
 #include <linux/kernel.h>
 #include <linux/version.h>
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,4,18)
+//#if LINUX_VERSION_CODE < KERNEL_VERSION(2,4,18)
 #include <linux/slab.h>
-#else
-#include <linux/malloc.h>
-#endif
+//#else
+//#include <linux/malloc.h>
+//#endif
 
 #include "defs.h"
 #include "tools.h"

