--- src/ksubtile.desktop.o	2009-02-15 02:00:09.000000000 +0100
+++ src/ksubtile.desktop	2009-02-15 02:00:18.000000000 +0100
@@ -1,12 +1,11 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=KSubtile
-MimeType=application/srt
-DocPath=ksubtile/index.html
+MimeType=application/srt;
+X-DocPath=ksubtile/index.html
 Comment=A video subtitle editor for KDE
 GenericName=KDE Subtitle Editor
-Exec=ksubtile %i %m -caption "%c"
+Exec=ksubtile %i -caption "%c"
 Icon=ksubtile
 Type=Application
 Terminal=false
-Categories=Qt;KDE;AudioVideo;;
+Categories=Qt;KDE;AudioVideo;
