diff -Naur scilab-4.1.2/pvm3/pvmgs/pvmgsu_aux.c scilab-4.1.2.new/pvm3/pvmgs/pvmgsu_aux.c
--- scilab-4.1.2/pvm3/pvmgs/pvmgsu_aux.c	2002-10-14 10:28:41.000000000 -0400
+++ scilab-4.1.2.new/pvm3/pvmgs/pvmgsu_aux.c	2008-01-19 07:19:28.000000000 -0500
@@ -53,6 +53,8 @@
 #include "pvmgsd.h"
 #include "pvmproto.h"
 
+#include <strings.h>
+
 extern int pvm_errno;
 extern int pvmmytid;
 extern int pvmtoplvl;
diff -Naur scilab-4.1.2/pvm3/src/hoster.c scilab-4.1.2.new/pvm3/src/hoster.c
--- scilab-4.1.2/pvm3/src/hoster.c	2005-11-23 05:12:27.000000000 -0500
+++ scilab-4.1.2.new/pvm3/src/hoster.c	2008-01-19 07:19:28.000000000 -0500
@@ -263,6 +263,7 @@
 #include "bfunc.h"
 #include <pvmtev.h>
 #include "global.h"
+#include <unistd.h>
 
 #ifdef IMA_BEOSCYLD
 #include <string.h>
diff -Naur scilab-4.1.2/pvm3/src/lpvm.c scilab-4.1.2.new/pvm3/src/lpvm.c
--- scilab-4.1.2/pvm3/src/lpvm.c	2002-10-15 03:40:21.000000000 -0400
+++ scilab-4.1.2.new/pvm3/src/lpvm.c	2008-01-19 07:19:28.000000000 -0500
@@ -607,6 +607,7 @@
 #include "waitc.h"
 #include "global.h"
 #include "lmsg.h"
+#include "unistd.h"
 
 #ifdef IMA_MPP
 #include "pvmmimd.h"
diff -Naur scilab-4.1.2/pvm3/tracer/cmd.c scilab-4.1.2.new/pvm3/tracer/cmd.c
--- scilab-4.1.2/pvm3/tracer/cmd.c	2002-10-14 11:00:41.000000000 -0400
+++ scilab-4.1.2.new/pvm3/tracer/cmd.c	2008-01-19 07:19:28.000000000 -0500
@@ -31,6 +31,8 @@
 
 #include "trcglob.h"
 
+#include <stdlib.h>
+
 #ifdef  SYSVSTR
 #include <string.h>
 #else
diff -Naur scilab-4.1.2/pvm3/tracer/tracer.c scilab-4.1.2.new/pvm3/tracer/tracer.c
--- scilab-4.1.2/pvm3/tracer/tracer.c	2002-10-14 11:00:41.000000000 -0400
+++ scilab-4.1.2.new/pvm3/tracer/tracer.c	2008-01-19 07:19:28.000000000 -0500
@@ -31,6 +31,8 @@
 
 #include "trcglob.h"
 
+#include <stdlib.h>
+
 #ifdef NEEDSSELECTH 
 #include <sys/select.h>
 #endif
diff -Naur scilab-4.1.2/pvm3/tracer/trccompat.c scilab-4.1.2.new/pvm3/tracer/trccompat.c
--- scilab-4.1.2/pvm3/tracer/trccompat.c	2002-10-14 11:00:41.000000000 -0400
+++ scilab-4.1.2.new/pvm3/tracer/trccompat.c	2008-01-19 07:19:28.000000000 -0500
@@ -28,6 +28,7 @@
 /* Tracer Headers */
 
 #include "trclib.h"
+#include <stdlib.h>
 
 #include "trccompat.h"
 
diff -Naur scilab-4.1.2/pvm3/tracer/trcsort.c scilab-4.1.2.new/pvm3/tracer/trcsort.c
--- scilab-4.1.2/pvm3/tracer/trcsort.c	2002-10-15 08:09:48.000000000 -0400
+++ scilab-4.1.2.new/pvm3/tracer/trcsort.c	2008-01-19 07:19:28.000000000 -0500
@@ -28,6 +28,7 @@
 /* Tracer Headers */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 
diff -Naur scilab-4.1.2/pvm3/tracer/trcutil.c scilab-4.1.2.new/pvm3/tracer/trcutil.c
--- scilab-4.1.2/pvm3/tracer/trcutil.c	2002-10-14 11:00:41.000000000 -0400
+++ scilab-4.1.2.new/pvm3/tracer/trcutil.c	2008-01-19 07:19:28.000000000 -0500
@@ -26,6 +26,7 @@
 
 
 #include "trclib.h"
+#include <stdlib.h>
 
 #ifdef WIN32
 #include <winsock.h>
diff -Naur scilab-4.1.2/routines/graphics/Events.c scilab-4.1.2.new/routines/graphics/Events.c
--- scilab-4.1.2/routines/graphics/Events.c	2007-09-10 03:52:29.000000000 -0400
+++ scilab-4.1.2.new/routines/graphics/Events.c	2008-01-19 07:19:28.000000000 -0500
@@ -3,6 +3,7 @@
  *    jpc@cermics.enpc.fr 
  --------------------------------------------------------------------------*/
 #include <stdio.h>
+#include <string.h>
 #include "bcg.h"
 
 typedef struct but
diff -Naur scilab-4.1.2/routines/gzvt2/vte.c scilab-4.1.2.new/routines/gzvt2/vte.c
--- scilab-4.1.2/routines/gzvt2/vte.c	2004-10-05 17:07:34.000000000 -0400
+++ scilab-4.1.2.new/routines/gzvt2/vte.c	2008-01-19 07:19:28.000000000 -0500
@@ -23,6 +23,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <string.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkx.h>
diff -Naur scilab-4.1.2/routines/scicos/scoWindowScope.c scilab-4.1.2.new/routines/scicos/scoWindowScope.c
--- scilab-4.1.2/routines/scicos/scoWindowScope.c	2007-10-15 04:45:30.000000000 -0400
+++ scilab-4.1.2.new/routines/scicos/scoWindowScope.c	2008-01-19 07:20:15.000000000 -0500
@@ -13,6 +13,7 @@
 #include "scoGetProperty.h"
 #include "scoSetProperty.h"
 #include <stdio.h>
+#include <string.h>
 
 void scoSetWindowIDInUserData(ScopeMemory * pScopeMemory,int block_number)
 {
