--- scilab-4.0/configure.in	2006-01-27 10:13:17.000000000 -0600
+++ scilab-4.0-new/configure.in	2006-04-26 19:05:17.000000000 -0500
@@ -894,13 +894,19 @@
       [LIEEELIB=-lieee])
     GCC=yes
     CC=gcc
-    CC_OPTIONS='-O -DNDEBUG -D_GNU_SOURCE -Dlinux -fwritable-strings'
+    CC_OPTIONS='-O -DNDEBUG -D_GNU_SOURCE -Dlinux'
     CC_LDFLAGS=$LIEEELIB
     CC_PICFLAGS=-fPIC
     # find Fortran compiler
     # g77 already asked and found: good
     if test "$WITH_G77" = yes; then
       FC=g77
+    elif test "$WITH_GFORTRAN" = yes;then
+      FC=gfortran
+      FC_OPTIONS="-O -ff2c"
+      FC_OPTIONS_O0="-ff2c"
+      FC_PICFLAGS=-fPIC
+      AC_DEFINE(G95_FORTRAN)
     # option with-f2c
     elif test "$with_f2c" = yes; then
       AC_CHECK_PROG(F2C,f2c,yes,no)
@@ -932,7 +938,7 @@
     LD=ld
     LD_LDFLAGS=$LIEEELIB
     if test "$enable_debug" = yes; then 
-      CC_OPTIONS='-g -Dlinux -D_GNU_SOURCE -fwritable-strings'
+      CC_OPTIONS='-g -Dlinux -D_GNU_SOURCE'
       FC_OPTIONS=-g
     fi
     MAKEFILE_TARGET=Makefile.linux
