diff -Naur scilab-4.0/routines/Javasci/Makefile.in scilab-4.0-new/routines/Javasci/Makefile.in
--- scilab-4.0/routines/Javasci/Makefile.in	2006-01-28 10:27:50.000000000 -0500
+++ scilab-4.0-new/routines/Javasci/Makefile.in	2006-11-21 21:27:59.000000000 -0500
@@ -13,7 +13,7 @@
 OBJSC=  javasci_SciStringArray.o javasci_globals.o javasci_Scilab.o javasci_SciDoubleArray.o @GCCFIX@
 OBJSF =
 OBJS = $(OBJSC)
-CFLAGS = $(CC_OPTIONS) $(JAVAINC)
+CFLAGS = $(CC_OPTIONS) -fPIC $(JAVAINC)
 FLIBS=@FLIBS@
 
 include ../../Makefile.incl
diff -Naur scilab-4.0/routines/Javasci/Makefile.linux scilab-4.0-new/routines/Javasci/Makefile.linux
--- scilab-4.0/routines/Javasci/Makefile.linux	2005-05-23 07:22:17.000000000 -0400
+++ scilab-4.0-new/routines/Javasci/Makefile.linux	2006-11-21 21:27:14.000000000 -0500
@@ -4,4 +4,4 @@
 	@echo Creation of $(LIBRARY)
 	@$(RM) $@;
 	@echo "linking";
-	@$(FC) -Wl,-shared -g -o $@ $(OBJS) $(DEFAULTS) $(LIBR) $(TCLTK_LIBS) $(XLIBS) $(TERMCAPLIB);
+	@$(CC) -shared -o $@ $(OBJS) $(TCLTK_LIBS) $(XLIBS) $(TERMCAPLIB);
