--- tuprolog.orig/src/alice/tuprologx/pj/test/AnotherList.java	2010-01-21 21:29:48.000000000 +1300
+++ tuprolog/src/alice/tuprologx/pj/test/AnotherList.java	2010-11-06 12:39:28.000000000 +1300
@@ -47,7 +47,7 @@
       AnotherList lists = PJ.newInstance(AnotherList.class);
       List<Int> l=lists.nil();          // Creating an empty list []
       for (int i=10;i>0;i--){
-          l=lists.add(new Int(i),l);    // Adding elements...[1,2,3,4,5,6,7,8,9,10]
+          l=lists.<Int,List<Int>,List<Int>>add(new Int(i),l);    // Adding elements...[1,2,3,4,5,6,7,8,9,10]
       }
       Var<List<Int>> l1 = new Var<List<Int>>("X");
       Var<List<Int>> l2 = new Var<List<Int>>("Y");
