--- boinc-6.6.40.orig/client/file_names.h	2008-10-07 14:45:06.000000000 +0200
+++ boinc-6.6.40/client/file_names.h	2009-09-30 14:57:03.147882778 +0200
@@ -73,7 +73,6 @@
 #define REMOTEHOST_FILE_NAME        "remote_hosts.cfg"
 #define ACCT_MGR_REQUEST_FILENAME   "acct_mgr_request.xml"
 #define ACCT_MGR_REPLY_FILENAME     "acct_mgr_reply.xml"
-#define GUI_RPC_PASSWD_FILE         "gui_rpc_auth.cfg"
 #define PROJECT_INIT_FILENAME       "project_init.xml"
 #define ACCT_MGR_URL_FILENAME       "acct_mgr_url.xml"
 #define ACCT_MGR_LOGIN_FILENAME     "acct_mgr_login.xml"
--- boinc-6.6.40.orig/lib/Makefile.am	2009-09-30 14:55:04.717196558 +0200
+++ boinc-6.6.40/lib/Makefile.am	2009-09-30 15:01:23.272107407 +0200
@@ -16,10 +16,25 @@
 
 if ENABLE_LIBRARIES
 
+libfcgi_sources = \ 
+    boinc_fcgi.cpp \ 
+    coproc.cpp \ 
+    crypt.cpp \ 
+    filesys.cpp \ 
+    hostinfo.cpp \ 
+    md5.c \ 
+    md5_file.cpp \ 
+    miofile.cpp \ 
+    msg_log.cpp \ 
+    parse.cpp \ 
+    shmem.cpp \ 
+    str_util.cpp \ 
+    synch.cpp \ 
+    util.cpp 
+
 generic_sources = \
     app_ipc.cpp \
     base64.cpp \
-    boinc_fcgi.cpp \
     cert_sig.cpp \
     coproc.cpp \
     crypt.cpp \
@@ -43,9 +58,9 @@
     proxy_info.cpp \
     shmem.cpp \
     str_util.cpp \
-    util.cpp	\
     unix_util.cpp \
-    synch.cpp 
+    synch.cpp \
+    util.cpp
 
 if OS_DARWIN
 mac_sources = \
@@ -107,7 +122,7 @@
 
 if ENABLE_FCGI
 lib_LTLIBRARIES += libboinc_fcgi.la
-libboinc_fcgi_la_SOURCES = $(generic_sources) $(mac_sources)
+libboinc_fcgi_la_SOURCES = $(libfcgi_sources) $(mac_sources)
 libboinc_fcgi_la_CFLAGS = -D_USING_FCGI_ $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_fcgi_la_CXXFLAGS = -D_USING_FCGI_ $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_fcgi_la_LDFLAGS = -L$(libdir) -rpath $(libdir) -version-number $(LIBBOINC_VERSION)
