diff -Naur boinc-5.10.45/client/client_msgs.C boinc-5.10.45.new/client/client_msgs.C
--- boinc-5.10.45/client/client_msgs.C	2008-04-15 13:20:37.000000000 -0400
+++ boinc-5.10.45.new/client/client_msgs.C	2008-04-17 14:26:48.000000000 -0400
@@ -27,6 +27,7 @@
 
 using std::deque;
 
+#include <cstring>
 #include "log_flags.h"
 #include "client_types.h"
 #include "client_msgs.h"
diff -Naur boinc-5.10.45/client/cs_account.C boinc-5.10.45.new/client/cs_account.C
--- boinc-5.10.45/client/cs_account.C	2008-04-15 13:20:36.000000000 -0400
+++ boinc-5.10.45.new/client/cs_account.C	2008-04-17 14:28:19.000000000 -0400
@@ -30,6 +30,7 @@
 #endif
 #endif
 
+#include <algorithm>
 #include "filesys.h"
 #include "parse.h"
 #include "str_util.h"
diff -Naur boinc-5.10.45/lib/crypt.C boinc-5.10.45.new/lib/crypt.C
--- boinc-5.10.45/lib/crypt.C	2008-04-15 13:19:19.000000000 -0400
+++ boinc-5.10.45.new/lib/crypt.C	2008-04-17 14:24:26.000000000 -0400
@@ -28,6 +28,7 @@
 #include <cstdlib>
 #endif
 
+#include <cstring>
 #include "md5_file.h"
 #include "error_numbers.h"
 
diff -Naur boinc-5.10.45/lib/parse.h boinc-5.10.45.new/lib/parse.h
--- boinc-5.10.45/lib/parse.h	2008-04-15 13:19:19.000000000 -0400
+++ boinc-5.10.45.new/lib/parse.h	2008-04-17 14:23:30.000000000 -0400
@@ -27,6 +27,7 @@
 #include <cstdlib>
 #include <string>
 #include <math.h>
+#include <cstring>
 #ifdef solaris
 #include <ieeefp.h>
 #endif
diff -Naur boinc-5.10.45/lib/procinfo_unix.C boinc-5.10.45.new/lib/procinfo_unix.C
--- boinc-5.10.45/lib/procinfo_unix.C	2008-04-15 13:19:19.000000000 -0400
+++ boinc-5.10.45.new/lib/procinfo_unix.C	2008-04-17 14:25:30.000000000 -0400
@@ -29,6 +29,7 @@
 
 #include <stdio.h>
 
+#include <cstring>
 #include <ctype.h>
 #include <sys/types.h>
 #include <dirent.h>
