--- libkarma-0.1.0-orig/src/Makefile	2007-05-15 12:29:02.000000000 -0400
+++ libkarma-0.1.0/src/Makefile	2008-05-28 19:15:24.000000000 -0400
@@ -6,7 +6,7 @@
 	util.c wav.c
 INST_HEADERS=lkarma.h utf8.h
 
-LIBDIR=../lib
+LIBDIR=../$(_LIBDIR)
 
 CC?=gcc
 AR?=ar
@@ -37,15 +37,15 @@
 default: all
 
 install: shared-lib-stamp static-lib-stamp
-	../install-sh -m 0755 -d $(DEST)/lib
+	../install-sh -m 0755 -d $(DEST)/$(_LIBDIR)
 	../install-sh -m 0755 -d $(DEST)/include
 	../install-sh -m 0755 -d $(DEST)/include/libkarma
-	../install-sh -m 0644 -t $(DEST)/lib/ $(LIBDIR)/$(TARGET).a
-	../install-sh -m 0755 -t $(PREFIX)/lib/ $(LIBDIR)/$(OUT_VERSION)
-	$(RM) $(PREFIX)/lib/$(OUT_MAJOR)
-	$(RM) $(PREFIX)/lib/$(OUT_BASE)
-	$(LN_S) $(OUT_VERSION) $(PREFIX)/lib/$(OUT_MAJOR)
-	$(LN_S) $(OUT_VERSION) $(PREFIX)/lib/$(OUT_BASE)
+	../install-sh -m 0644 -t $(DEST)/$(_LIBDIR)/ $(LIBDIR)/$(TARGET).a
+	../install-sh -m 0755 -t $(PREFIX)/$(_LIBDIR)/ $(LIBDIR)/$(OUT_VERSION)
+	$(RM) $(PREFIX)/$(_LIBDIR)/$(OUT_MAJOR)
+	$(RM) $(PREFIX)/$(_LIBDIR)/$(OUT_BASE)
+	$(LN_S) $(OUT_VERSION) $(PREFIX)/$(_LIBDIR)/$(OUT_MAJOR)
+	$(LN_S) $(OUT_VERSION) $(PREFIX)/$(_LIBDIR)/$(OUT_BASE)
 	../install-sh -m 0644 -t $(DEST)/include/libkarma/ $(INST_HEADERS)
 
 uninstall:
--- libkarma-0.1.0-orig/tools/Makefile	2007-05-15 12:29:02.000000000 -0400
+++ libkarma-0.1.0/tools/Makefile	2008-05-28 20:14:23.000000000 -0400
@@ -1,5 +1,5 @@
 INCLUDES=-I../src
-LIBDIR=../lib
+LIBDIR=../$(_LIBDIR)
 LIBKARMA=$(LIBDIR)/libkarma.a
 ifeq ($(MAKECMDGOALS),static)
   LIBKARMA=$(LIBDIR)/libkarma.a
