Index: alevt-1.6.1/Makefile
===================================================================
--- alevt-1.6.1.orig/Makefile
+++ alevt-1.6.1/Makefile
@@ -1,12 +1,11 @@
 VER=1.6.1
-OPT=-O2 -s
 #OPT=-O -g
 DEFS=-DWITH_PNG
 FONT=vtxt
 # a smaller and thinner font
 #FONT=neep9
 
-CFLAGS=$(OPT) -DVERSION=\"$(VER)\" $(DEFS) -I/usr/X11R6/include -I/usr/local/include -W
+CFLAGS+=-DVERSION=\"$(VER)\" $(DEFS)
 
 EXPOBJS=export.o exp-txt.o exp-html.o exp-gfx.o font.o
 OBJS=main.o ui.o xio.o fdset.o vbi.o cache.o help.o edline.o search.o edit.o misc.o hamm.o lang.o $(EXPOBJS)
@@ -21,13 +20,13 @@ all: alevt alevt-date alevt-cap alevt.1x
 	sync
 
 alevt: $(OBJS)
-	$(CC) $(OPT) $(OBJS) -o alevt -L/usr/X11R6/lib -lX11 $(EXPLIBS)
+	$(CC) $(LDFLAGS) $(OPT) $(OBJS) -o alevt -L/usr/X11R6/lib -lX11 $(EXPLIBS)
 
 alevt-date: $(TOBJS)
-	$(CC) $(OPT) $(TOBJS) -o alevt-date
+	$(CC) $(LDFLAGS) $(OPT) $(TOBJS) -o alevt-date
 
 alevt-cap: $(COBJS)
-	$(CC) $(OPT) $(COBJS) -o alevt-cap $(EXPLIBS)
+	$(CC) $(LDFLAGS) $(OPT) $(COBJS) -o alevt-cap $(EXPLIBS)
 
 font.o: font1.xbm font2.xbm
 fontsize.h: font1.xbm font2.xbm
@@ -40,7 +39,7 @@ font2.xbm: bdf2xbm $(FONT)-latin-2.bdf
 	./bdf2xbm font2 <$(FONT)-latin-2.bdf >font2.xbm
 
 bdf2xbm: bdf2xbm.c
-	$(CC) -O -s bdf2xbm.c -o bdf2xbm
+	$(CC) $(CFLAGS) $(LDFLAGS) bdf2xbm.c -o bdf2xbm
 
 alevt.1x: alevt.1x.in
 	sed s/VERSION/$(VER)/g <alevt.1x.in >alevt.1x
