diff -ru alevt-1.6.0.orig/Makefile alevt-1.6.0/Makefile
--- alevt-1.6.0.orig/Makefile	Wed Oct 11 20:08:29 2000
+++ alevt-1.6.0/Makefile	Thu Apr 11 19:56:50 2002
@@ -126,3 +126,5 @@
 ui.o: search.h export.h ui.h
 vbi.o: os.h vt.h misc.h vbi.h dllist.h cache.h lang.h fdset.h hamm.h
 xio.o: vt.h misc.h dllist.h xio.h fdset.h lang.h icon.xbm font.h
+font.o: font1.xbm font2.xbm
+
diff -ru alevt-1.6.0.orig/alevt-date.c alevt-1.6.0/alevt-date.c
--- alevt-1.6.0.orig/alevt-date.c	Sat Sep 16 23:34:42 2000
+++ alevt-1.6.0/alevt-date.c	Wed Apr 10 21:48:09 2002
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #include <sys/time.h>
+#include <time.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include <signal.h>
diff -ru alevt-1.6.0.orig/main.c alevt-1.6.0/main.c
--- alevt-1.6.0.orig/main.c	Sat Sep 16 23:34:18 2000
+++ alevt-1.6.0/main.c	Wed Apr 10 23:09:40 2002
@@ -15,7 +15,7 @@
 /* current args for next start() */
 static char *geometry;
 static char *dpy_name;
-static char *vbi_name = "/dev/vbi";
+static char *vbi_name = "/dev/v4l/vbi0";
 static struct xio *xio;
 static struct vbi *vbi;
 static int fine_tune = 1; // auto = 999;
@@ -32,7 +32,7 @@
 	    "  Valid options:\t\tDefault:\n"
 	    "    --help\n"
 	    "    --version\n"
-	    "    -vbi <vbidev>\t\t/dev/vbi\n"
+	    "    -vbi <vbidev>\t\t/dev/v4l/vbi0\n"
 	    "    -display <dpy>\t\t$DISPLAY\n"
 	    "    -geometry <geo>\t\t41x25\n"
 	    "    -finetune <-4..4|auto>\t0\n"
