--- a/lib/MIME/Lite.pm
+++ b/lib/MIME/Lite.pm
@@ -401,13 +401,13 @@ if ( $^O =~ /win32/i ) {
         }
     }
     unless (-x $SENDMAIL) {
-        Carp::croak "can't find an executable sendmail"
+        undef $SENDMAIL;
     }
 }
 
 ### Our sending facilities:
 my %SenderArgs = (
-  sendmail  => ["$SENDMAIL -t -oi -oem"],
+  sendmail  => [$SENDMAIL ? "$SENDMAIL -t -oi -oem" : undef],
   smtp      => [],
   sub       => [],
 );
@@ -2666,6 +2666,7 @@ sub send_by_sendmail {
     if ( @_ == 1 and !ref $_[0] ) {
         ### Use the given command...
         my $sendmailcmd = shift @_;
+        Carp::croak "No sendmail command available" unless $sendmailcmd;
 
         ### Do it:
         local *SENDMAIL;
