From d437f3107635a85486faae15e48a7237b6d69e42 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?R=C3=A9mi=20Cardona?= <remi.cardona@lri.fr>
Date: Mon, 20 Oct 2008 13:17:32 +0200
Subject: [PATCH 3/4] use standard automake macros for handling symlinks

---
 configure.ac    |    1 +
 src/Makefile.am |    6 +++---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index a53b225..36852aa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -40,6 +40,7 @@ AC_PROG_LIBTOOL
 DOLT
 AC_PROG_CC
 AM_PROG_CC_C_O
+AC_PROG_LN_S
 
 AC_CHECK_PROG(gen4asm, [intel-gen4asm], yes, no)
 AM_CONDITIONAL(HAVE_GEN4ASM, test x$gen4asm = xyes)
diff --git a/src/Makefile.am b/src/Makefile.am
index 9b32215..b16ba6c 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -223,8 +223,8 @@ intel_drv_la_SOURCES += \
 	$(INTEL_XVMC_SRCS)
 endif
 
-install-data-local: install-intel_drv_laLTLIBRARIES
-	(cd $(DESTDIR)$(intel_drv_ladir) && rm -f i810_drv.so && ln -s intel_drv.so i810_drv.so)
+install-exec-hook:
+	(cd $(DESTDIR)$(intel_drv_ladir) && rm -f i810_drv.so && $(LN_S) intel_drv.so i810_drv.so)
 
-uninstall-local:
+uninstall-hook:
 	(cd $(DESTDIR)$(intel_drv_ladir) && rm -f i810_drv.so)
-- 
1.6.0.2

