From 7b847b328e314ac24cdad56dc9846a06b9d44e23 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?R=C3=A9mi=20Cardona?= <remi.cardona@lri.fr>
Date: Mon, 20 Oct 2008 11:27:36 +0200
Subject: [PATCH 2/4] clean up man pages generation and installation

---
 man/Makefile.am |   21 +++++----------------
 1 files changed, 5 insertions(+), 16 deletions(-)

diff --git a/man/Makefile.am b/man/Makefile.am
index 7432d5a..2a1cc81 100644
--- a/man/Makefile.am
+++ b/man/Makefile.am
@@ -25,21 +25,14 @@
 # 
 
 drivermandir = $(DRIVER_MAN_DIR)
-
-DRIVER_MAN_DIR_SUFFIX = $(DRIVER_MAN_DIR:@mandir@/man%=%)
-
-driverman_PRE = @DRIVER_NAME@.man
-
 driverman_DATA = \
-	$(driverman_PRE:man=@DRIVER_MAN_SUFFIX@) \
-	i810.@DRIVER_MAN_SUFFIX@
+	$(DRIVER_NAME).$(DRIVER_MAN_SUFFIX) \
+	i810.$(DRIVER_MAN_SUFFIX)
 
-EXTRA_DIST = @DRIVER_NAME@.man
+EXTRA_DIST = $(DRIVER_NAME).man
 
 CLEANFILES = $(driverman_DATA)
 
-SED = sed
-
 # Strings to replace in man pages
 XORGRELSTRING = @PACKAGE_STRING@
   XORGMANNAME = X Version 11
@@ -61,9 +54,5 @@ SUFFIXES = .$(DRIVER_MAN_SUFFIX) .man
 .man.$(DRIVER_MAN_SUFFIX):
 	sed $(MAN_SUBSTS) < $< > $@
 
-BUILT_SOURCES = shadows.DONE
-
-shadows.DONE:
-	-rm -f i810.@DRIVER_MAN_SUFFIX@
-	echo .so man$(DRIVER_MAN_DIR_SUFFIX)/intel.$(DRIVER_MAN_SUFFIX) > \
-		i810.@DRIVER_MAN_SUFFIX@
+i810.$(DRIVER_MAN_SUFFIX):
+	echo .so man$(DRIVER_MAN_SUFFIX)/$(DRIVER_NAME).$(DRIVER_MAN_SUFFIX) > $@
-- 
1.6.0.2

