diff -Naur gnat-3.15p-i686-pc-linux-gnu-bin.orig/Makefile gnat-3.15p-i686-pc-linux-gnu-bin/Makefile
--- gnat-3.15p-i686-pc-linux-gnu-bin.orig/Makefile	2002-10-23 10:54:23.000000000 +0200
+++ gnat-3.15p-i686-pc-linux-gnu-bin/Makefile	2003-10-04 10:46:40.000000000 +0200
@@ -53,7 +53,7 @@
 	  fi \
 	done
 	cp -p -f gnat1$(ext)     $(libsubdir)
-	tar cf - rts* | (cd $(libsubdir); tar xf -)
+	tar cf - rts* | (cd $(libsubdir); tar xf -) || true
 	case `uname` in \
 	   *_NT*) cp -p -f *.dll $(user_bin); cp -p -f *.a $(libsubdir); \
 	          mv -f $(libsubdir)/rts/adainclude $(libsubdir)/adainclude; \
