diff -uNr nipper-0.9.4/IOS/process-banner.c nipper-0.9.4.orig/IOS/process-banner.c
--- nipper-0.9.4/IOS/process-banner.c	2007-05-01 15:37:19.000000000 +0100
+++ nipper-0.9.4.orig/IOS/process-banner.c	2007-04-29 12:40:45.000000000 +0100
@@ -53,7 +53,7 @@
 		readLine(input, line, lineSize);
 
 		// Get all parts of the banner message...
+		while ((strstr(line, bannerTerminator) == 0))
-		while ((strstr(line, bannerTerminator) == 0) && (line[0] != '!'))
 		{
 			if (debugMode == true)
 			{
