diff -Nur pgpool-II-1.0.2/pgpool.conf.sample pgpool-II-1.0.2.new/pgpool.conf.sample
--- pgpool-II-1.0.2/pgpool.conf.sample	2006-11-29 22:54:26.000000000 +0100
+++ pgpool-II-1.0.2.new/pgpool.conf.sample	2007-02-28 15:38:36.000000000 +0100
@@ -7,21 +7,21 @@
 listen_addresses = 'localhost'
 
 # Port number for pgpool
-port = 9999
+port = 5433
 
 # Port number for pgpool communication manager
 pcp_port = 9898
 
 # Unix domain socket path.  (The Debian package defaults to
 # /var/run/postgresql.)
-socket_dir = '/tmp'
+socket_dir = '/var/run/postgresql'
 
 # Unix domain socket path for pgpool communication manager.
 # (Debian package defaults to /var/run/postgresql)
-pcp_socket_dir = '/tmp'
+pcp_socket_dir = '/var/run/postgresql'
 
 # Unix domain socket path for the backend. Debian package defaults to /var/run/postgresql!
-backend_socket_dir = '/tmp'
+backend_socket_dir = '/var/run/postgresql'
 
 # pgpool communication manager timeout. 0 means no timeout, but strongly not recommended!
 pcp_timeout = 10
@@ -43,8 +43,8 @@
 # 0 means no exit.
 child_max_connections = 0
 
-# Logging directory
-logdir = '/tmp'
+# Logging directory (more accurately, the directory for the PID file)
+logdir = '/var/run/postgresql'
 
 # Replication mode
 replication_mode = false
diff -Nur pgpool-II-1.0.2/pool.h pgpool-II-1.0.2.new/pool.h
--- pgpool-II-1.0.2/pool.h	2007-02-13 04:19:04.000000000 +0100
+++ pgpool-II-1.0.2.new/pool.h	2007-02-28 15:39:07.000000000 +0100
@@ -46,10 +46,10 @@
 #define PCP_PASSWD_FILE_NAME "pcp.conf"
 
 /* pid file directory */
-#define DEFAULT_LOGDIR "/tmp"
+#define DEFAULT_LOGDIR "/var/run/postgresql"
 
 /* Unix domain socket directory */
-#define DEFAULT_SOCKET_DIR "/tmp"
+#define DEFAULT_SOCKET_DIR "/var/run/postgresql"
 
 /* pid file name */
 #define PID_FILE_NAME "pgpool.pid"
