diff -ru pommed-1.10~/pommed/Makefile pommed-1.10/pommed/Makefile
--- pommed-1.10~/pommed/Makefile	2007-10-21 14:04:18.000000000 +0200
+++ pommed-1.10/pommed/Makefile	2007-10-21 14:05:16.000000000 +0200
@@ -14,7 +14,7 @@
 CONFUSE_CFLAGS = $(shell pkg-config libconfuse --cflags)
 CONFUSE_LIBS = $(shell pkg-config libconfuse --libs)
 
-CFLAGS = -g -O2 -Wall $(DBUS_CFLAGS) $(ALSA_CFLAGS) $(AUDIOFILE_CFLAGS) $(CONFUSE_CFLAGS)
+CFLAGS += -g -Wall $(DBUS_CFLAGS) $(ALSA_CFLAGS) $(AUDIOFILE_CFLAGS) $(CONFUSE_CFLAGS)
 
 ifeq ($(ARCH), ppc)
 OFLIB ?=
@@ -39,7 +39,7 @@
 
 else
 
-LDFLAGS = -lz -lsmbios $(DBUS_LIBS) $(ALSA_LIBS) $(AUDIOFILE_LIBS) $(CONFUSE_LIBS)
+LDFLAGS = -lz -lsmbios -lstdc++ $(DBUS_LIBS) $(ALSA_LIBS) $(AUDIOFILE_LIBS) $(CONFUSE_LIBS)
 
 SOURCES = pommed.c cd_eject.c evdev.c conffile.c audio.c dbus.c power.c beep.c \
 		mactel/x1600_backlight.c mactel/gma950_backlight.c \
