diff -Nru xscreensaver-4.23.orig/configure.in xscreensaver-4.23/configure.in
--- xscreensaver-4.23.orig/configure.in	2005-10-17 03:05:49.000000000 +0200
+++ xscreensaver-4.23/configure.in	2005-10-31 22:16:53.000000000 +0100
@@ -715,6 +715,8 @@
   # Look for the directory under a standard set of common directories.
   # Check X11 before X11Rn because it's often a symlink to the current release.
   for ac_dir in                                 \
+    /usr/lib/X11/app-defaults                   \
+                                                \
     /usr/X11/lib/app-defaults                   \
     /usr/X11R6/lib/app-defaults                 \
     /usr/X11R6/lib/X11/app-defaults             \
@@ -745,7 +747,6 @@
     /usr/x386/lib/X11/app-defaults              \
     /usr/XFree86/lib/X11/app-defaults           \
                                                 \
-    /usr/lib/X11/app-defaults                   \
     /usr/local/lib/X11/app-defaults             \
     /usr/unsupported/lib/X11/app-defaults       \
     /usr/athena/lib/X11/app-defaults            \
