--- driver/xscreensaver.c	2005-07-27 15:27:57.000000000 -0400
+++ driver/xscreensaver.c	2005-07-27 15:30:08.000000000 -0400
@@ -1641,8 +1641,11 @@
       if (hint.res_class) XFree (hint.res_class);
     }
 
+  /*
   fprintf (stderr, "%s: %d: unrecognised ClientMessage \"%s\" received\n",
            blurb(), screen, (str ? str : "(null)"));
+  */
+
   fprintf (stderr, "%s: %d: for window 0x%lx (%s)\n",
            blurb(), screen, (unsigned long) w, wdesc);
   if (str) XFree (str);
