--- lib-src/soundtouch/src/STTypes.h.orig	2004-05-06 09:47:37.808353000 -0700
+++ lib-src/soundtouch/src/STTypes.h	2004-05-06 09:48:18.408353000 -0700
@@ -49,6 +49,10 @@
 	virtual const char *what() const throw() { return "soundtouch library st_assertion failure"; }
 };
 
+#if defined(__GNUC__) && __GNUC__ >= 3  /* GCC 3.x */
+#define st_assert(expr) ((expr) ? 0 : throw soundtouch_exception())
+#else 
 #define st_assert(expr) ((expr) ? (void) 0 : throw soundtouch_exception())
+#endif
 
 #endif
