=== modified file 'pym/portage.py'
--- pym/portage.py	
+++ pym/portage.py	
@@ -2745,8 +2745,6 @@
 
 	# if any of these are being called, handle them -- running them out of the sandbox -- and stop now.
 	if mydo in ["clean","cleanrm"]:
-		if "noclean" in features:
-			return 0
 		return spawn(EBUILD_SH_BINARY+" clean",mysettings,debug=debug,free=1,logfile=None)
 	elif mydo in ["help","setup"]:
 		return spawn(EBUILD_SH_BINARY+" "+mydo,mysettings,debug=debug,free=1,logfile=logfile)
@@ -6089,7 +6087,8 @@
 
 		# Process ebuild logfiles
 		elog_process(self.mycpv, self.settings)
-		doebuild(myebuild, "clean", root, self.settings, tree=self.treetype)
+		if "noclean" not in self.settings.features:
+			doebuild(myebuild, "clean", root, self.settings, tree=self.treetype)
 		return 0
 
 	def mergeme(self,srcroot,destroot,outfile,secondhand,stufftomerge,cfgfiledict,thismtime):

