--- gobby-0.4.5/src/preferences.cpp	2008-06-28 18:56:10.000000000 +0200
+++ gobby-0.4.5/src/preferences.cpp	2008-06-28 18:57:54.000000000 +0200
@@ -16,6 +16,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#include <cstring>
 #include "features.hpp"
 
 #ifdef WITH_GTKSOURCEVIEW2
--- gobby-0.4.5/src/preferencesdialog.cpp	2008-06-28 18:56:20.000000000 +0200
+++ gobby-0.4.5/src/preferencesdialog.cpp	2008-06-28 18:58:12.000000000 +0200
@@ -17,6 +17,7 @@
  */
 
 #include <stdexcept>
+#include <cstring>
 #include <gtkmm/stock.h>
 #include <gtkmm/messagedialog.h>
 
--- gobby-0.4.5/src/header.cpp	2008-06-28 18:55:44.000000000 +0200
+++ gobby-0.4.5/src/header.cpp	2008-06-28 18:56:55.000000000 +0200
@@ -16,6 +16,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#include <cstring>
 #include <gtkmm/stock.h>
 #include <obby/format_string.hpp>
 
--- gobby-0.4.5/src/unix.cpp	2008-06-28 18:55:33.000000000 +0200
+++ gobby-0.4.5/src/unix.cpp	2008-06-28 18:57:37.000000000 +0200
@@ -17,6 +17,7 @@
  */
 
 #include <stdexcept>
+#include <cstring>
 #include "unix.hpp"
 
 namespace
