diff -ur libtorrent-0.6.2/configure.ac libtorrent-0.6.2-fbsd/configure.ac
--- libtorrent-0.6.2/configure.ac	2005-04-27 20:38:53.000000000 +0200
+++ libtorrent-0.6.2-fbsd/configure.ac	2005-05-25 13:41:39.581095808 +0200
@@ -71,6 +71,8 @@
 
 AC_DEFINE(HAVE_CONFIG_H, 1, true if config.h was included)
 
+AC_CHECK_HEADERS([malloc.h])
+
 AC_OUTPUT([
 	libtorrent.pc
 	Makefile
Only in libtorrent-0.6.2-fbsd/: configure.ac~
diff -ur libtorrent-0.6.2/src/net/socket_fd.cc libtorrent-0.6.2-fbsd/src/net/socket_fd.cc
--- libtorrent-0.6.2/src/net/socket_fd.cc	2005-04-22 00:51:43.000000000 +0200
+++ libtorrent-0.6.2-fbsd/src/net/socket_fd.cc	2005-05-25 13:21:56.969880040 +0200
@@ -28,6 +28,7 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <netinet/in.h>
+#include <netinet/in_systm.h>
 #include <netinet/ip.h>
 
 #include "torrent/exceptions.h"
diff -ur libtorrent-0.6.2/src/utils/sha_fast.cc libtorrent-0.6.2-fbsd/src/utils/sha_fast.cc
--- libtorrent-0.6.2/src/utils/sha_fast.cc	2005-04-14 14:47:42.000000000 +0200
+++ libtorrent-0.6.2-fbsd/src/utils/sha_fast.cc	2005-05-25 13:39:46.289318776 +0200
@@ -35,7 +35,10 @@
 
 #include "config.h"
 
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
+
 #include <memory.h>
 #include "sha_fast.h"
 
Only in libtorrent-0.6.2-fbsd/src/utils: sha_fast.cc~
