diff -Nrup pax-utils-0.1.10/porting.h pax-utils/porting.h
--- pax-utils-0.1.10/porting.h	2006-02-18 11:43:13.000000000 -0500
+++ pax-utils/porting.h	2006-02-22 09:58:22.000000000 -0500
@@ -32,8 +32,16 @@
 #ifndef HATE_GLOB
 # undef HAVE_GLOB
 # define HAVE_GLOB
+
 # include <glob.h>
-#endif
+
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__OpenBSD__) || defined(__NetBSD__)
+# define glob64_t glob_t
+# define globfree64 globfree
+# define glob64 glob
+#endif /* BSD */
+
+#endif /* !HATE_GLOB */
 
 #include <sys/mman.h>
 #include "elf.h"
diff -Nrup pax-utils-0.1.10/scanelf.c pax-utils/scanelf.c
--- pax-utils-0.1.10/scanelf.c	2006-02-18 11:43:13.000000000 -0500
+++ pax-utils/scanelf.c	2006-02-22 09:58:22.000000000 -0500
@@ -204,7 +204,7 @@ static char *scanelf_file_phdr(elfobj *e
 			flags = EGET(phdr[i].p_flags); \
 			if (be_quiet && ((flags & check_flags) != check_flags)) \
 				continue; \
-			if (fix_elf && ((flags & PF_X) != flags)) { \
+			if ((EGET(phdr[i].p_type) != PT_LOAD) && (fix_elf && ((flags & PF_X) != flags))) { \
 				ESET(phdr[i].p_flags, flags & (PF_X ^ (size_t)-1)); \
 				ret[3] = ret[7] = '!'; \
 				flags = EGET(phdr[i].p_flags); \
