 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98dbc70..b323e10 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -430,3 +430,7 @@ install(FILES html/qhull.man         DESTINATION ${MAN_INSTALL_DIR} RENAME qhull
 install(FILES html/rbox.man          DESTINATION ${MAN_INSTALL_DIR} RENAME rbox.1)
 install(FILES ${doc_FILES}           DESTINATION ${DOC_INSTALL_DIR})
 install(DIRECTORY html/              DESTINATION ${DOC_INSTALL_DIR})
+
+
+configure_file("qhull.pc.in" "qhull.pc" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/qhull.pc" DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig")
