--- chrome/chrome.gyp.orig	2010-06-10 10:25:49.000000000 +0200
+++ chrome/chrome.gyp	2010-06-10 10:26:52.000000000 +0200
@@ -462,6 +462,7 @@
         ['OS=="linux" or OS=="freebsd" or OS=="openbsd" or OS=="solaris"', {
           'dependencies': [
             '../build/linux/system.gyp:gtk',
+            '../third_party/libevent/libevent.gyp:libevent',
           ],
         }],
       ],
--- net/net.gyp.orig	2010-06-10 11:42:35.000000000 +0200
+++ net/net.gyp	2010-06-10 11:43:21.000000000 +0200
@@ -179,6 +179,7 @@
               '../build/linux/system.gyp:gconf',
               '../build/linux/system.gyp:gdk',
               '../build/linux/system.gyp:nss',
+              '../third_party/libevent/libevent.gyp:libevent',
             ],
           },
           {  # else: OS is not in the above list
@@ -560,6 +561,7 @@
               '../build/linux/system.gyp:gconf',
               '../build/linux/system.gyp:gdk',
               '../build/linux/system.gyp:nss',
+              '../third_party/libevent/libevent.gyp:libevent',
             ],
           },
           {  # else: OS is not in the above list
