diff -Naur chromium-6.0.495.0_old/ipc/ipc_message_utils.h chromium-6.0.495.0/ipc/ipc_message_utils.h
--- chromium-6.0.495.0_old/ipc/ipc_message_utils.h	2010-08-15 11:41:25.000000000 +0300
+++ chromium-6.0.495.0/ipc/ipc_message_utils.h	2010-08-18 21:38:48.000000000 +0300
@@ -27,7 +27,7 @@
 // ipc_message_utils_impl.h in those files) and exported, instead of expanded
 // at every call site. Special note: GCC happily accepts the attribute before
 // the method declaration, but only acts on it if it is after.
-#define IPC_MSG_NOINLINE  __attribute__((noinline));
+#define IPC_MSG_NOINLINE  __attribute__((noinline, noclone));
 #elif defined(COMPILER_MSVC)
 // MSVC++ doesn't do this.
 #define IPC_MSG_NOINLINE
