--- chrome/chrome.gyp.orig	2010-06-16 21:19:14.000000000 +0200
+++ chrome/chrome.gyp	2010-06-16 21:26:33.000000000 +0200
@@ -1098,6 +1098,9 @@
       'include_dirs': [
         '..',
       ],
+      'variables': {
+      	'use_cups%': 1,
+      },
       'conditions': [
         ['OS=="win"', {
           'defines': [
@@ -1114,7 +1117,7 @@
             '../build/linux/system.gyp:gtk',
           ],
         }],
-        ['OS=="linux" and chromeos==0 and target_arch!="arm"', {
+        ['OS=="linux" and chromeos==0 and target_arch!="arm" and use_cups==1', {
           'link_settings': {
             'libraries': [
               '-lcups',
