--- layer0/MyPNG.c	2004-01-30 00:40:33.000000000 +0100
+++ layer0/MyPNG.c.new	2004-04-20 16:05:35.399498139 +0200
@@ -14,17 +14,8 @@
 Z* -------------------------------------------------------------------
 */
 
-#include"os_predef.h"
-#include"os_std.h"
-
-#include"Base.h"
-#include "MyPNG.h"
-#include"MemoryDebug.h"
-#include "Setting.h"
-
 #ifdef _HAVE_LIBPNG
-#include<png.h>
-
+#include <png.h>
  /* The png_jmpbuf() macro, used in error handling, became available in
   * libpng version 1.0.6.  If you want to be able to run your code with older
   * versions of libpng, you must define the macro yourself (but only if it
@@ -37,6 +28,14 @@
 
 #endif
 
+#include"os_predef.h"
+#include"os_std.h"
+
+#include"Base.h"
+#include "MyPNG.h"
+#include"MemoryDebug.h"
+#include "Setting.h"
+
 int MyPNGWrite(char *file_name,unsigned char *p,unsigned int width,unsigned int height)
 {
 #ifdef _HAVE_LIBPNG
