--- plex86-20021117.orig/kernel/host-linux.c	2005-08-20 12:52:20.000000000 -0300
+++ plex86-20021117/kernel/host-linux.c	2005-08-20 14:05:50.000000000 -0300
@@ -30,10 +30,7 @@
 #include <linux/fs.h>
 #include <linux/mm.h>
 #include <linux/proc_fs.h>
-#include <linux/wrapper.h>
 #include <linux/version.h>
-#include <asm/irq.h>
-
 
 #ifndef VERSION_CODE
 #  define VERSION_CODE(vers,rel,seq) ( ((vers)<<16) | ((rel)<<8) | (seq) )
@@ -45,7 +42,10 @@
 
 #include <asm/io.h>
 
-
+/* Vanquirius's hacks */
+typedef struct devfs_entry *devfs_handle_t;
+#define DEVFS_FL_NONE 0x000
+#define DEVFS_FL_DEFAULT DEVFS_FL_NONE
 
 /************************************************************************/
 /* Compatibility macros for older kernels                               */
@@ -423,7 +423,8 @@
         /* Do *not* free pages that are still mapped to user space! */
 #if LINUX_VERSION_CODE >= VERSION_CODE(2,3,99)
         /* Not sure when this changed.  If you know, email us. */
-        if (inode->i_data.i_mmap != NULL)
+	/* DANGEROUS hack */
+        /* if (inode->i_data.i_mmap != NULL) */
 #else
         if (inode->i_mmap != NULL)
 #endif
@@ -902,13 +903,10 @@
      * hence we rely on the fact that Linux lets at least one page of 
      * virtual address space unused after the end of the module.
      */
-#ifdef THIS_MODULE
-    void *start_addr = THIS_MODULE;
-    unsigned size    = THIS_MODULE->size;
-#else
-    void *start_addr = &mod_use_count_;
+
+    /* Below you will find a DANGEROUS hack */
+    void *start_addr; // = &mod_use_count_;
     unsigned size    = 0x10000000;  /* Actual size determined below */
-#endif
 
     int n_pages;
 
@@ -1040,7 +1038,8 @@
   unsigned
 host_idle(void)
 {
-  if (NEED_RESCHED)
+/* Another DANGEROUS hack */
+// if (NEED_RESCHED)
     schedule();
 
   /* return !current_got_fatal_signal(); */
