--- Singular/fereadl.c.orig	2003-07-26 01:28:47.000000000 +0200
+++ Singular/fereadl.c	2003-07-26 01:27:47.000000000 +0200
@@ -287,15 +287,15 @@
       extern char PC;
       #endif
       /* OB: why this ??? */
-      /* char *t_buf=(char *)omAlloc(128); */
-      char t_buf[128];
+      char *t_buf=(char *)omAlloc(128); 
+      //char t_buf[128];
       char *temp;
 
       /* Extract information that termcap functions use.  */
-      temp = tgetstr ("pc", (char **)&t_buf);
+      temp = tgetstr ("pc", t_buf);
       PC = (temp!=NULL) ? *temp : '\0';
-      BC=tgetstr("le",(char **)&t_buf);
-      UP=tgetstr("up",(char **)&t_buf);
+      BC=tgetstr("le",t_buf);
+      UP=tgetstr("up",t_buf);
 
       /* Extract information we will use */
       colmax=tgetnum("co");
@@ -303,7 +303,7 @@
       fe_cursor_line=pagelength-1;
 
       /* init screen */
-      temp = tgetstr ("ti", (char **)&t_buf);
+      temp = tgetstr ("ti", t_buf);
       #if 0
       if (temp!=NULL) tputs(temp,1,fe_out_char);
       #endif
--- Singular/feResource.cc.orig	2003-08-06 15:19:27.000000000 +0200
+++ Singular/feResource.cc	2003-08-06 15:21:09.000000000 +0200
@@ -85,7 +85,7 @@
   {"SearchPath",    's',     feResPath,  NULL,
    "$SINGULARPATH;"
    "%b/LIB;"
-   "%r/LIB;"
+   "%r/share/singular/LIB;"
    "%r/../LIB;"
    "%d/LIB;"
    "%d/../LIB"
--- Singular/mod_raw.cc.orig	2003-08-06 16:52:02.000000000 +0200
+++ Singular/mod_raw.cc	2003-08-06 16:55:53.000000000 +0200
@@ -40,11 +40,11 @@
 void* dynl_open_binary_warn(char* binary_name, const char* msg)
 {
   void* handle = NULL;
-  const char* bin_dir = feGetResource('b');
+  const char* bin_dir = feGetResource('r');
   if (bin_dir != NULL)
   {
     char path_name[MAXPATHLEN];
-    sprintf(path_name, "%s%s%s.%s", bin_dir, DIR_SEPP, binary_name, 
+    sprintf(path_name, "%s%s%s%s%s.%s", bin_dir, DIR_SEPP, "lib/singular", DIR_SEPP, binary_name, 
             DL_TAIL);
     handle = dynl_open(path_name);
     if (handle == NULL && ! warn_handle)
--- Singular/mod_raw.cc.orig	2003-08-08 00:08:28.000000000 +0200
+++ Singular/mod_raw.cc	2003-08-08 00:09:44.000000000 +0200
@@ -84,7 +84,7 @@
 /*****************************************************************************
  * SECTION ix86-linux / alpha-linux / IA64-linux                             *
  *****************************************************************************/
-#if defined(ix86_Linux) || defined(DecAlpha_Linux) || defined(ix86_Linux_libc5) || defined(IA64_Linux)
+#if defined(ix86_Linux) || defined(DecAlpha_Linux) || defined(ix86_Linux_libc5) || defined(IA64_Linux) || defined(ppc_Linux)
 #include <dlfcn.h>
 
 static void* kernel_handle = NULL;
--- singuname.sh	2003-08-07 12:48:09.000000000 +0200
+++ singuname.sh.new	2003-08-07 12:50:14.000000000 +0200
@@ -180,6 +180,18 @@
         echo ${prefix}-Unknown
         exit 1
     fi
+# PowerPC Linux ########################################################
+elif (echo $uname_a | $egrep " ppc " > $devnull)
+then
+    prefix=ppc
+    if (echo $uname_a | $egrep "Linux" > $devnull)
+    then
+        echo ${prefix}-Linux
+        exit 0
+    else
+        echo ${prefix}-Unknown
+        exit 1
+    fi
 else # Unknown ########################################################
     echo Unknown
     exit 2
