--- pciutils-2.1.11.orig/lib/configure	2004-08-30 14:37:52.623313920 -0500
+++ pciutils-2.1.11/lib/configure	2004-08-30 14:38:58.500299096 -0500
@@ -98,6 +98,6 @@
 if [ -z "$ok" ] ; then
 	echo "WARNING: No real configuration access method is available."
 fi
-echo >>$c "#define PATH_PCI_IDS \"$sharedir/pci.ids\""
+echo >>$c "#define PATH_PCI_IDS \"$sharedir/misc/pci.ids\""
 echo >>$c "#define PCILIB_VERSION \"$version\""
 sed '/^#define [^ ]*$/!d;s/^#define \(.*\)/\1=1/' <$c >config.mk
--- pciutils-2.1.11.orig/Makefile	2004-08-30 14:37:52.623313920 -0500
+++ pciutils-2.1.11/Makefile	2004-08-30 14:38:58.500299096 -0500
@@ -52,1 +52,1 @@
-	sed <$< >$@ "s@^DEST=.*@DEST=$(SHAREDIR)/pci.ids@"
+	sed <$< >$@ "s@^DEST=.*@DEST=$(SHAREDIR)/misc/pci.ids@"
--- pciutils-2.1.11.orig/lspci.man	2004-08-30 14:37:52.623313920 -0500
+++ pciutils-2.1.11/lspci.man	2004-08-30 14:38:58.500299096 -0500
@@ -70,7 +70,7 @@
 Use
 .B
 <file>
-as PCI ID database instead of /usr/share/pci.ids.
+as PCI ID database instead of /usr/share/misc/pci.ids.
 .TP
 .B -p <dir>
 Use
@@ -131,7 +131,7 @@
 
 .SH FILES
 .TP
-.B @SHAREDIR@/pci.ids
+.B @SHAREDIR@/misc/pci.ids
 A list of all known PCI ID's (vendors, devices, classes and subclasses).
 .TP
 .B /proc/bus/pci
--- pciutils-2.1.11.orig/update-pciids.man	2004-08-30 14:37:52.634312248 -0500
+++ pciutils-2.1.11/update-pciids.man	2004-08-30 14:39:53.749899880 -0500
@@ -17,7 +17,7 @@
 
 .SH FILES
 .TP
-.B @SHAREDIR@/pci.ids
+.B @SHAREDIR@/misc/pci.ids
 Here we install the new list.
 
 .SH SEE ALSO
