--- pciutils-2.1.11/lspci.man
+++ pciutils-2.1.11/lspci.man
@@ -70,7 +70,7 @@
 Use
 .B
 <file>
-as PCI ID database instead of /usr/share/pci.ids.
+as PCI ID database instead of @SHAREDIR@/pci.ids.
 .TP
 .B -p <dir>
 Use
--- pciutils-2.1.11/Makefile
+++ pciutils-2.1.11/Makefile
@@ -12,5 +12,5 @@
 
-PREFIX=/usr/local
+PREFIX=/usr
 SBINDIR=$(PREFIX)/sbin
-SHAREDIR=$(PREFIX)/share
+SHAREDIR=$(PREFIX)/share/misc
 MANDIR=$(shell if [ -d $(PREFIX)/share/man ] ; then echo $(PREFIX)/share/man ; else echo $(PREFIX)/man ; fi)
