diff -NrU5 src.original/pefile.cpp src/pefile.cpp
--- src.original/pefile.cpp	2009-08-02 23:54:36.000000000 -0600
+++ src/pefile.cpp	2009-08-02 23:54:18.000000000 -0600
@@ -1347,12 +1347,12 @@
     };
 
     // FIXME this comparison is not too exact
     while (1)
     {
-        char *delim1 = strchr(keep, '/');
-        char *delim2 = strchr(keep, ',');
+        const char *delim1 = strchr(keep, '/');
+        const char *delim2 = strchr(keep, ',');
         if (helper::match(itype, ntype, keep))
         {
             if (!delim1)
                 return true;
             if (delim2 && delim2 < delim1)
