--- tcsh-6.13.00/sh.sem.c.fcntl	2004-08-23 12:55:48.300002488 +0200
+++ tcsh-6.13.00/sh.sem.c	2004-08-23 13:00:59.745655552 +0200
@@ -863,7 +863,7 @@
 		stderror(ERR_SYSTEM, tmp, strerror(errno));
 	    /* allow input files larger than 2Gb  */
 #ifndef WINNT_NATIVE
-	    (void) fcntl(fd, O_LARGEFILE, 0);
+	    (void) fcntl(fd, F_SETFL, fcntl(fd, F_GETFL) | O_LARGEFILE);
 #endif /*!WINNT_NATIVE*/
 	    (void) dmove(fd, 0);
 	}
@@ -923,7 +923,7 @@
 		stderror(ERR_SYSTEM, tmp, strerror(errno));
 	    /* allow input files larger than 2Gb  */
 #ifndef WINNT_NATIVE
-	    (void) fcntl(fd, O_LARGEFILE, 0);
+	    (void) fcntl(fd, F_SETFL, fcntl(fd, F_GETFL) | O_LARGEFILE);
 #endif /*!WINNT_NATIVE*/
 	}
 	(void) dmove(fd, 1);
