Index: xfsprogs-2.10.1/include/buildmacros
===================================================================
--- xfsprogs-2.10.1.orig/include/buildmacros
+++ xfsprogs-2.10.1/include/buildmacros
@@ -124,7 +124,7 @@ INSTALL_LINGUAS = \
 endif
 
 SUBDIRS_MAKERULE = \
-	@for d in $(SUBDIRS) ""; do \
+	+@for d in $(SUBDIRS) ""; do \
 		if test -d "$$d" -a ! -z "$$d"; then \
 			$(ECHO) === $$d ===; \
 			$(MAKEF) -C $$d $@ || exit $$?; \
@@ -139,10 +139,10 @@ MAN_MAKERULE = \
 	done
 
 DIST_MAKERULE = \
-	$(MAKEF) -C build dist
+	+$(MAKEF) -C build dist
 
 SOURCE_MAKERULE = \
-	@test -z "$$DIR" && DIR="."; \
+	+@test -z "$$DIR" && DIR="."; \
 	for f in $(SRCFILES) ""; do \
 	    test -z "$$f" && break; \
 	    test -L "$$f" || $(ECHO) $$DIR/$$f; \
