--- src/out_lame.c.orig	2004-11-20 18:55:37.610089432 +0000
+++ src/out_lame.c	2004-11-20 18:55:51.539971768 +0000
@@ -23,10 +23,10 @@
 #include <string.h>
 
 #include <bmp/plugin.h>
-#include <beep/beepctrl.h>
-#include <beep/dirbrowser.h>
-#include <beep/configfile.h>
-#include <beep/util.h>
+#include <bmp/beepctrl.h>
+#include <bmp/dirbrowser.h>
+#include <bmp/configfile.h>
+#include <bmp/util.h>
 
 #include <lame/lame.h>
 
