--- src/avl/Makefile.in	2005-01-17 20:47:31.000000000 +0100
+++ src/avl/Makefile.in	2005-01-17 20:46:50.000000000 +0100
@@ -204,21 +204,24 @@
 	$(LTCOMPILE) -c -o $@ `test -f '$<' || echo '$(srcdir)/'`$<
 
 libiceavl_la-avl.o: avl.c
-@AMDEP_TRUE@	source='avl.c' object='libiceavl_la-avl.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libiceavl_la-avl.Po' tmpdepfile='$(DEPDIR)/libiceavl_la-avl.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='avl.c' object='libiceavl_la-avl.o' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libiceavl_la-avl.Po' tmpdepfile='$(DEPDIR)/libiceavl_la-avl.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libiceavl_la_CFLAGS) $(CFLAGS) -c -o libiceavl_la-avl.o `test -f 'avl.c' || echo '$(srcdir)/'`avl.c
 
 libiceavl_la-avl.obj: avl.c
-@AMDEP_TRUE@	source='avl.c' object='libiceavl_la-avl.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libiceavl_la-avl.Po' tmpdepfile='$(DEPDIR)/libiceavl_la-avl.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='avl.c' object='libiceavl_la-avl.obj' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libiceavl_la-avl.Po' tmpdepfile='$(DEPDIR)/libiceavl_la-avl.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libiceavl_la_CFLAGS) $(CFLAGS) -c -o libiceavl_la-avl.obj `cygpath -w avl.c`
 
 libiceavl_la-avl.lo: avl.c
-@AMDEP_TRUE@	source='avl.c' object='libiceavl_la-avl.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libiceavl_la-avl.Plo' tmpdepfile='$(DEPDIR)/libiceavl_la-avl.TPlo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='avl.c' object='libiceavl_la-avl.lo' libtool=yes @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libiceavl_la-avl.Plo' tmpdepfile='$(DEPDIR)/libiceavl_la-avl.TPlo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libiceavl_la_CFLAGS) $(CFLAGS) -c -o libiceavl_la-avl.lo `test -f 'avl.c' || echo '$(srcdir)/'`avl.c
 CCDEPMODE = @CCDEPMODE@
 
--- src/httpp/Makefile.in	2005-01-17 20:48:07.000000000 +0100
+++ src/httpp/Makefile.in	2005-01-17 20:48:24.000000000 +0100
@@ -205,21 +205,24 @@
 	$(LTCOMPILE) -c -o $@ `test -f '$<' || echo '$(srcdir)/'`$<
 
 libicehttpp_la-httpp.o: httpp.c
-@AMDEP_TRUE@	source='httpp.c' object='libicehttpp_la-httpp.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicehttpp_la-httpp.Po' tmpdepfile='$(DEPDIR)/libicehttpp_la-httpp.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='httpp.c' object='libicehttpp_la-httpp.o' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicehttpp_la-httpp.Po' tmpdepfile='$(DEPDIR)/libicehttpp_la-httpp.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicehttpp_la_CFLAGS) $(CFLAGS) -c -o libicehttpp_la-httpp.o `test -f 'httpp.c' || echo '$(srcdir)/'`httpp.c
 
 libicehttpp_la-httpp.obj: httpp.c
-@AMDEP_TRUE@	source='httpp.c' object='libicehttpp_la-httpp.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicehttpp_la-httpp.Po' tmpdepfile='$(DEPDIR)/libicehttpp_la-httpp.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='httpp.c' object='libicehttpp_la-httpp.obj' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicehttpp_la-httpp.Po' tmpdepfile='$(DEPDIR)/libicehttpp_la-httpp.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicehttpp_la_CFLAGS) $(CFLAGS) -c -o libicehttpp_la-httpp.obj `cygpath -w httpp.c`
 
 libicehttpp_la-httpp.lo: httpp.c
-@AMDEP_TRUE@	source='httpp.c' object='libicehttpp_la-httpp.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicehttpp_la-httpp.Plo' tmpdepfile='$(DEPDIR)/libicehttpp_la-httpp.TPlo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='httpp.c' object='libicehttpp_la-httpp.lo' libtool=yes @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicehttpp_la-httpp.Plo' tmpdepfile='$(DEPDIR)/libicehttpp_la-httpp.TPlo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicehttpp_la_CFLAGS) $(CFLAGS) -c -o libicehttpp_la-httpp.lo `test -f 'httpp.c' || echo '$(srcdir)/'`httpp.c
 CCDEPMODE = @CCDEPMODE@
 
--- src/net/Makefile.in	2005-01-17 20:52:34.000000000 +0100
+++ src/net/Makefile.in	2005-01-17 20:51:26.000000000 +0100
@@ -207,39 +207,45 @@
 	$(LTCOMPILE) -c -o $@ `test -f '$<' || echo '$(srcdir)/'`$<
 
 libicenet_la-sock.o: sock.c
-@AMDEP_TRUE@	source='sock.c' object='libicenet_la-sock.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-sock.Po' tmpdepfile='$(DEPDIR)/libicenet_la-sock.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='sock.c' object='libicenet_la-sock.o' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-sock.Po' tmpdepfile='$(DEPDIR)/libicenet_la-sock.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicenet_la_CFLAGS) $(CFLAGS) -c -o libicenet_la-sock.o `test -f 'sock.c' || echo '$(srcdir)/'`sock.c
 
 libicenet_la-sock.obj: sock.c
-@AMDEP_TRUE@	source='sock.c' object='libicenet_la-sock.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-sock.Po' tmpdepfile='$(DEPDIR)/libicenet_la-sock.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='sock.c' object='libicenet_la-sock.obj' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-sock.Po' tmpdepfile='$(DEPDIR)/libicenet_la-sock.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicenet_la_CFLAGS) $(CFLAGS) -c -o libicenet_la-sock.obj `cygpath -w sock.c`
 
 libicenet_la-sock.lo: sock.c
-@AMDEP_TRUE@	source='sock.c' object='libicenet_la-sock.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-sock.Plo' tmpdepfile='$(DEPDIR)/libicenet_la-sock.TPlo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='sock.c' object='libicenet_la-sock.lo' libtool=yes @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-sock.Plo' tmpdepfile='$(DEPDIR)/libicenet_la-sock.TPlo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicenet_la_CFLAGS) $(CFLAGS) -c -o libicenet_la-sock.lo `test -f 'sock.c' || echo '$(srcdir)/'`sock.c
 
 libicenet_la-resolver.o: resolver.c
-@AMDEP_TRUE@	source='resolver.c' object='libicenet_la-resolver.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-resolver.Po' tmpdepfile='$(DEPDIR)/libicenet_la-resolver.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='resolver.c' object='libicenet_la-resolver.o' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-resolver.Po' tmpdepfile='$(DEPDIR)/libicenet_la-resolver.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicenet_la_CFLAGS) $(CFLAGS) -c -o libicenet_la-resolver.o `test -f 'resolver.c' || echo '$(srcdir)/'`resolver.c
 
 libicenet_la-resolver.obj: resolver.c
-@AMDEP_TRUE@	source='resolver.c' object='libicenet_la-resolver.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-resolver.Po' tmpdepfile='$(DEPDIR)/libicenet_la-resolver.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='resolver.c' object='libicenet_la-resolver.obj' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-resolver.Po' tmpdepfile='$(DEPDIR)/libicenet_la-resolver.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicenet_la_CFLAGS) $(CFLAGS) -c -o libicenet_la-resolver.obj `cygpath -w resolver.c`
 
 libicenet_la-resolver.lo: resolver.c
-@AMDEP_TRUE@	source='resolver.c' object='libicenet_la-resolver.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-resolver.Plo' tmpdepfile='$(DEPDIR)/libicenet_la-resolver.TPlo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='resolver.c' object='libicenet_la-resolver.lo' libtool=yes @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicenet_la-resolver.Plo' tmpdepfile='$(DEPDIR)/libicenet_la-resolver.TPlo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicenet_la_CFLAGS) $(CFLAGS) -c -o libicenet_la-resolver.lo `test -f 'resolver.c' || echo '$(srcdir)/'`resolver.c
 CCDEPMODE = @CCDEPMODE@
 
--- src/thread/Makefile.in	2005-01-17 20:53:48.000000000 +0100
+++ src/thread/Makefile.in	2005-01-17 20:54:08.000000000 +0100
@@ -204,21 +204,24 @@
 	$(LTCOMPILE) -c -o $@ `test -f '$<' || echo '$(srcdir)/'`$<
 
 libicethread_la-thread.o: thread.c
-@AMDEP_TRUE@	source='thread.c' object='libicethread_la-thread.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicethread_la-thread.Po' tmpdepfile='$(DEPDIR)/libicethread_la-thread.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='thread.c' object='libicethread_la-thread.o' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicethread_la-thread.Po' tmpdepfile='$(DEPDIR)/libicethread_la-thread.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicethread_la_CFLAGS) $(CFLAGS) -c -o libicethread_la-thread.o `test -f 'thread.c' || echo '$(srcdir)/'`thread.c
 
 libicethread_la-thread.obj: thread.c
-@AMDEP_TRUE@	source='thread.c' object='libicethread_la-thread.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicethread_la-thread.Po' tmpdepfile='$(DEPDIR)/libicethread_la-thread.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='thread.c' object='libicethread_la-thread.obj' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicethread_la-thread.Po' tmpdepfile='$(DEPDIR)/libicethread_la-thread.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicethread_la_CFLAGS) $(CFLAGS) -c -o libicethread_la-thread.obj `cygpath -w thread.c`
 
 libicethread_la-thread.lo: thread.c
-@AMDEP_TRUE@	source='thread.c' object='libicethread_la-thread.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicethread_la-thread.Plo' tmpdepfile='$(DEPDIR)/libicethread_la-thread.TPlo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='thread.c' object='libicethread_la-thread.lo' libtool=yes @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicethread_la-thread.Plo' tmpdepfile='$(DEPDIR)/libicethread_la-thread.TPlo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicethread_la_CFLAGS) $(CFLAGS) -c -o libicethread_la-thread.lo `test -f 'thread.c' || echo '$(srcdir)/'`thread.c
 CCDEPMODE = @CCDEPMODE@
 
--- src/timing/Makefile.in	2005-01-17 20:56:22.000000000 +0100
+++ src/timing/Makefile.in	2005-01-17 20:56:35.000000000 +0100
@@ -202,21 +202,24 @@
 	$(LTCOMPILE) -c -o $@ `test -f '$<' || echo '$(srcdir)/'`$<
 
 libicetiming_la-timing.o: timing.c
-@AMDEP_TRUE@	source='timing.c' object='libicetiming_la-timing.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicetiming_la-timing.Po' tmpdepfile='$(DEPDIR)/libicetiming_la-timing.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='timing.c' object='libicetiming_la-timing.o' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicetiming_la-timing.Po' tmpdepfile='$(DEPDIR)/libicetiming_la-timing.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicetiming_la_CFLAGS) $(CFLAGS) -c -o libicetiming_la-timing.o `test -f 'timing.c' || echo '$(srcdir)/'`timing.c
 
 libicetiming_la-timing.obj: timing.c
-@AMDEP_TRUE@	source='timing.c' object='libicetiming_la-timing.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicetiming_la-timing.Po' tmpdepfile='$(DEPDIR)/libicetiming_la-timing.TPo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='timing.c' object='libicetiming_la-timing.obj' libtool=no @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicetiming_la-timing.Po' tmpdepfile='$(DEPDIR)/libicetiming_la-timing.TPo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicetiming_la_CFLAGS) $(CFLAGS) -c -o libicetiming_la-timing.obj `cygpath -w timing.c`
 
 libicetiming_la-timing.lo: timing.c
-@AMDEP_TRUE@	source='timing.c' object='libicetiming_la-timing.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@	depfile='$(DEPDIR)/libicetiming_la-timing.Plo' tmpdepfile='$(DEPDIR)/libicetiming_la-timing.TPlo' @AMDEPBACKSLASH@
-@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	source='timing.c' object='libicetiming_la-timing.lo' libtool=yes @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	depfile='$(DEPDIR)/libicetiming_la-timing.Plo' tmpdepfile='$(DEPDIR)/libicetiming_la-timing.TPlo' @AMDEPBACKSLASH@
+#@AMDEP_TRUE@	$(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libicetiming_la_CFLAGS) $(CFLAGS) -c -o libicetiming_la-timing.lo `test -f 'timing.c' || echo '$(srcdir)/'`timing.c
 CCDEPMODE = @CCDEPMODE@
 
