Don't force crappy flags.

--- SConstruct.orig	2008-08-04 11:10:58.000000000 -0400
+++ SConstruct	2008-08-04 12:37:10.000000000 -0400
@@ -90,18 +90,8 @@
 		#env.Append(CPPDEFINES = ['NDEBUG']) 
 
 else:
-	if env['gcc_visibility']: 
-		env.Append(CCFLAGS=['-fvisibility=hidden']);
-		env.Append(CXXFLAGS=['-fvisibility-inlines-hidden', '-fvisibility=hidden']);
-	if debug:
-		env.Append(CCFLAGS=['-ggdb'])
-		env.Append(CPPFLAGS=['-ggdb'])
-	else: 
-		env.Append(CCFLAGS=['-O3'])
-		env.Append(CPPFLAGS=['-O3'])
-		
-	env.Append(CPPFLAGS=['-Wall', '-pedantic', '-Wno-long-long', '-pipe', '-pthread'])
-	env.Append(CCFLAGS=['-Wall', '-pedantic', '-Wno-long-long', '-pipe', '-pthread'])
+	env.Append(CPPFLAGS=['-Wall', '-pedantic', '-Wno-long-long', '-pthread'])
+	env.Append(CCFLAGS=['-Wall', '-pedantic', '-Wno-long-long', '-pthread'])
 
 
 conf_env = env.Clone()
