--- src/callbacks.c	2002-06-05 13:47:11.000000000 +0200
+++ src/callbacks.c.new	2004-06-15 09:08:54.324083974 +0200
@@ -806,6 +806,9 @@
     update = cfg.auto_update_clist;
     cfg.auto_update_clist = FALSE;
     
+    if(!GTK_CLIST(mwin_clist)->selection) {
+    	return;
+    } 
     row = (int) GTK_CLIST (mwin_clist)->selection->data;
     gtk_clist_get_text (GTK_CLIST (mwin_clist), row, 1, &txt);
     start_frame = atoi (txt);
--- src/cfg.c   2003-09-30 09:14:32.000000000 +0200
+++ src/cfg.c.new       2003-09-30 09:15:00.000000000 +0200
@@ -241,7 +241,7 @@
     gtk_clist_set_selection_mode (GTK_CLIST (tab3.clist_lang), GTK_SELECTION_BROWSE);
     
     i = 0;
-    while (cfg.lang_list[i])
+    if(cfg.lang_list) while (cfg.lang_list[i])
     {
        gtk_clist_append (GTK_CLIST (tab3.clist_lang), (gchar **) &(cfg.lang_list[i]));
        i += 2;
