Index: gsubedit-0.4pre1/src/fileio.c
===================================================================
--- gsubedit-0.4pre1.orig/src/fileio.c
+++ gsubedit-0.4pre1/src/fileio.c
@@ -7,7 +7,7 @@ GList *error_lines = NULL;
 int gse_fl_write_file (char *filename, gint filetype, gint max_rows)
 {
     FILE *fl;
-    gchar line[BUF_LEN], msg[100], *tmp[4], converted[4][20];
+    gchar line[BUF_LEN], msg[100], *tmp[4], converted[4][200];
     GtkWidget *msgbox;
     gint i, col, hr, min, sec, msec;
     gdouble tot_sec;
