diff -ru vtun.orig/configure vtun/configure
--- vtun.orig/configure	2002-12-21 04:22:15.000000000 +1100
+++ vtun/configure	2005-11-13 16:57:22.000000000 +1100
@@ -2001,14 +2001,14 @@
 	  CPPFLAGS="$CPPFLAGS -I$p"
 	fi
         echo "$ac_t""($dir) yes " 1>&6 
-	echo $ac_n "checking for lzo1x_decompress in -llzo""... $ac_c" 1>&6
-echo "configure:2006: checking for lzo1x_decompress in -llzo" >&5
-ac_lib_var=`echo lzo'_'lzo1x_decompress | sed 'y%./+-%__p_%'`
+	echo $ac_n "checking for lzo1x_decompress in -llzo2""... $ac_c" 1>&6
+echo "configure:2006: checking for lzo1x_decompress in -llzo2" >&5
+ac_lib_var=`echo lzo2'_'lzo1x_decompress | sed 'y%./+-%__p_%'`
 if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-llzo  $LIBS"
+LIBS="-llzo2  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 2014 "configure"
 #include "confdefs.h"
@@ -2037,7 +2037,7 @@
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
   
-	       LIBS="$LIBS -llzo"
+	       LIBS="$LIBS -llzo2"
 	       cat >> confdefs.h <<\EOF
 #define HAVE_LZO 1
 EOF
@@ -2437,7 +2437,7 @@
 echo "$ac_t""" 1>&6
 
 
-REL=`echo 'REL-2_6' | tr -d '$: \-' | sed 's/^[A-Za-z]*//' | sed 's/\_/\./'`
+REL=`echo '$Name:  $' | tr -d '$: \-' | sed 's/^[A-Za-z]*//' | sed 's/\_/\./'`
 
 
 cat >> confdefs.h <<EOF
diff -ru vtun.orig/configure.in vtun/configure.in
--- vtun.orig/configure.in	2002-12-21 04:13:03.000000000 +1100
+++ vtun/configure.in	2005-11-13 16:56:59.000000000 +1100
@@ -119,9 +119,9 @@
    AC_CHECKING( for LZO Library and Header files ... )
    AC_SEARCH_HEADERS(lzo1x.h, 
 	$LZO_HDR_DIR /usr/include/lzo "" /usr/local/include,
-        AC_CHECK_LIB(lzo, lzo1x_decompress,
+        AC_CHECK_LIB(lzo2, lzo1x_decompress,
 	    [
-	       LIBS="$LIBS -llzo"
+	       LIBS="$LIBS -llzo2"
 	       AC_DEFINE(HAVE_LZO)
 	    ],
 	    AC_MSG_ERROR( LZO library not found. )
diff -ru vtun.orig/lfd_lzo.c vtun/lfd_lzo.c
--- vtun.orig/lfd_lzo.c	2002-04-25 19:19:50.000000000 +1000
+++ vtun/lfd_lzo.c	2005-11-13 17:01:46.000000000 +1100
@@ -74,7 +74,7 @@
 	vtun_syslog(LOG_ERR,"Can't allocate buffer for the compressor");
 	return 1;
      }	
-     if( !(wmem = lzo_malloc(mem)) ){
+     if( !(wmem = malloc(mem)) ){
 	vtun_syslog(LOG_ERR,"Can't allocate buffer for the compressor");
 	return 1;
      }	
@@ -92,7 +92,7 @@
 int free_lzo()
 {
      lfd_free(zbuf); zbuf = NULL;
-     lzo_free(wmem); wmem = NULL;
+     free(wmem); wmem = NULL;
      return 0;
 }
 
