--- email-2.3.1/Makefile.in	2004-05-27 21:00:03.000000000 +0200
+++ email-2.3.1-fixed/Makefile.in	2005-01-15 01:58:30.791131848 +0100
@@ -19,9 +19,9 @@
 	cd $(SRCDIR) && $(MAKE)
 
 install:
-	./install.sh --bindir $(bindir) --sysconfdir $(sysconfdir) \
-		--mandir $(mandir) --binext '$(bin_suffix)' --version "$(VERSION)" \
-        --docdir $(docdir)
+	./install.sh --bindir $(DESTDIR)usr$(bindir) --sysconfdir $(DESTDIR)$(sysconfdir) \
+		--mandir $(DESTDIR)$(mandir) --binext '$(bin_suffix)' --version "$(VERSION)" \
+        --docdir $(DESTDIR)$(docdir)
 
 distclean:
 	cd $(SRCDIR) && $(MAKE) clean-all
