--- canorus.desktop.orig	2009-08-29 14:12:20.000000000 +0300
+++ canorus.desktop	2009-08-29 14:12:54.000000000 +0300
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Canorus
 GenericName=Music score editor
 GenericName[sl]=Program za pisanje not
@@ -9,6 +8,6 @@
 Icon=canorus
 Terminal=false
 Type=Application
-Categories=Application;AudioVideo;
+Categories=AudioVideo;
 StartupNotify=true
-MimeType=application/x-notes
+MimeType=application/x-notes;
