--- chan_capi-cm-0.6/Makefile.orig	2005-09-11 17:46:46.000000000 +0200
+++ chan_capi-cm-0.6/Makefile	2005-09-21 10:02:21.000000000 +0200
@@ -41,14 +41,14 @@
 PROC=$(shell uname -m)
 
 DEBUG=-g #-pg
+ifneq ($(ASTERISK_HEADER_DIR),/usr/include)
 INCLUDE=-I$(ASTERISK_HEADER_DIR)
+endif
 ifeq (${OSNAME},FreeBSD)
 INCLUDE+=$(shell [ -d /usr/include/i4b/include ] && \
 	echo -n -I/usr/include/i4b/include)
 endif
-CFLAGS=-pipe -fPIC -Wall -Wmissing-prototypes -Wmissing-declarations $(DEBUG) $(INCLUDE) -D_REENTRANT -D_GNU_SOURCE
-CFLAGS+=-O6
-CFLAGS+=$(shell if $(CC) -march=$(PROC) -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-march=$(PROC)"; fi)
+CFLAGS+=-pipe -fPIC -Wall -Wmissing-prototypes -Wmissing-declarations $(DEBUG) $(INCLUDE) -D_REENTRANT -D_GNU_SOURCE
 CFLAGS+=$(shell if uname -m | grep -q ppc; then echo "-fsigned-char"; fi)
 
 CFLAGS+=-DASTERISKVERSION=\"$(ASTERISKVERSION)\"
@@ -78,9 +78,11 @@
 	$(CC) -shared -Xlinker -x -o $@ $^ -lcapi20
 
 install: all
+	if [ ! -d $(MODULES_DIR) ]; then mkdir -p $(MODULES_DIR); fi
 	for x in $(SHAREDOS); do $(INSTALL) -m 755 $$x $(MODULES_DIR) ; done
 
 config: all
+	if [ ! -d $(INSTALL_PREFIX)/etc/asterisk ]; then mkdir -p $(INSTALL_PREFIX)/etc/asterisk; fi
 	cp capi.conf $(INSTALL_PREFIX)/etc/asterisk/
 
 samples: config
