--- load_lists.c
+++ load_lists.c
@@ -50,7 +50,7 @@
     }
   }
   
-  g_string_assign (info->conf_file, "gamepick.conf");
+  g_string_assign (info->conf_file, "CONFDIR/gamepick.conf");
   if (file_readable (info->conf_file->str))
     return NULL;
 
--- gamepick.h
+++ gamepick.h
@@ -6,9 +6,9 @@
 
 /* configure these to your own systems */
 /* or overide them with -c <conf file> */
-static const gchar default_cmd_dir[] = ".";
-static const gchar default_icon_dir[] = ".";
-static const gchar default_stage2[] = "./gamepick-stage2";
+static const gchar default_cmd_dir[] = "GAMES_BINDIR";
+static const gchar default_icon_dir[] = "GAMES_DATADIR";
+static const gchar default_stage2[] = "GAMES_BINDIR/gamepick-stage2";
 
 enum browse_targets {bt_cmd, bt_icon};
 
