diff -ruNp 9011-processor.h.patch-old/include/asm-i386/processor.h 9011-processor.h.patch-new/include/asm-i386/processor.h
--- 9011-processor.h.patch-old/include/asm-i386/processor.h	2005-10-28 02:02:08.000000000 +0200
+++ 9011-processor.h.patch-new/include/asm-i386/processor.h	2005-12-29 21:40:37.000000000 +0100
@@ -718,4 +718,10 @@ extern void mtrr_bp_init(void);
 #define mtrr_bp_init() do {} while (0)
 #endif
 
+#ifdef CONFIG_X86_MCE
+extern void mcheck_init(struct cpuinfo_x86 *c);
+#else
+#define mcheck_init(c) do {} while(0)
+#endif
+
 #endif /* __ASM_I386_PROCESSOR_H */
