diff -burN Src.orig/s_mus.c Src/s_mus.c
--- Src.orig/s_mus.c	2006-05-31 11:30:58.000000000 -0400
+++ Src/s_mus.c	2006-05-31 11:32:04.000000000 -0400
@@ -122,7 +122,7 @@
 #endif
 #ifdef UNIX
     // The available interfaces have already been loaded.
-    if(musd_loaded.Init())
+    if(musd_loaded.Init && musd_loaded.Init())
     {
         imus = (musd_loaded_imus.gen.Init ? &musd_loaded_imus : 0);
         iext = (musd_loaded_iext.gen.Init ? &musd_loaded_iext : 0);
@@ -174,7 +174,7 @@
     musd_win.Shutdown();
 #endif
 #ifdef UNIX
-    musd_loaded.Shutdown();
+    if (musd_loaded.Shutdown) musd_loaded.Shutdown();
 #endif
 
     // No more interfaces.
