diff -Nadurp gnucap-2006-05-06/modelgen/Make2.g++ gnucap-2006-05-06-gentoo/modelgen/Make2.g++
--- gnucap-2006-05-06/modelgen/Make2.g++	2006-05-06 07:02:51.000000000 +0200
+++ gnucap-2006-05-06-gentoo/modelgen/Make2.g++	2006-05-21 17:45:07.000000000 +0200
@@ -25,9 +25,8 @@ VPATH = .:..
 
 CCC = g++
 CCFLAGS = \
--DUNIX -O2 -DNDEBUG -I.. -I. -fexceptions -W
+-DUNIX -O2 -DNDEBUG -I.. -I. -fexceptions -W $(CXXFLAGS)
 LIBS = -lm
-LDFLAGS =
 
 .SUFFIXES : .o .cc
 .cc.o:; $(CCC) $(CCFLAGS) -c $<
diff -Nadurp gnucap-2006-05-06/src/configure gnucap-2006-05-06-gentoo/src/configure
--- gnucap-2006-05-06/src/configure	2005-09-06 21:08:39.000000000 +0200
+++ gnucap-2006-05-06-gentoo/src/configure	2006-05-21 17:50:10.000000000 +0200
@@ -19,7 +19,7 @@ else
 fi
 rm -f a.out
 
-echo "-DUNIX -O2 -DNDEBUG -I.. -I. -W" >>Make.ccflags
+echo "-DUNIX -O2 -DNDEBUG -I.. -I. -W ${CXXFLAGS}" >>Make.ccflags
 echo "" >>Make.libs
 echo "" >>Make.libs
 
@@ -28,7 +28,6 @@ cat <<CAT_EOF >Make.sys
 VPATH = .:..
 CCC = g++
 
-LDFLAGS =
 
 .SUFFIXES : .o .cc
 .cc.o:; \$(CCC) \$(CCFLAGS) -c \$<
