diff -ur aewm++_goodies/ae_fspanel-1.0/Makefile aewm++_goodies~/ae_fspanel-1.0/Makefile
--- aewm++_goodies/ae_fspanel-1.0/Makefile	2002-11-11 18:30:27.000000000 -0600
+++ aewm++_goodies~/ae_fspanel-1.0/Makefile	2002-11-20 14:00:07.000000000 -0600
@@ -1,7 +1,7 @@
 CC = gcc
 prefix=/usr
-C_FLAGS = -DNOSTDLIB -DHAVE_XPM -Wall -I/usr/X11R6/include -m386 -malign-loops=2 -malign-jumps=2 -malign-functions=2
-L_FLAGS = -O1 -Xlinker -s -L/usr/X11R6/lib -lX11 -lXpm
+C_FLAGS = -DNOSTDLIB -DHAVE_XPM -Wall -I/usr/X11R6/include -falign-loops=2 -falign-jumps=2 -falign-functions=2 $(CFLAGS)
+L_FLAGS = -Xlinker -s -L/usr/X11R6/lib -lX11 -lXpm
 PROGNAME = ae_fspanel
 
 $(PROGNAME): Makefile ae_fspanel.c ae_fspanel.h icon.xpm
diff -ur aewm++_goodies/appbar-1.0/Makefile aewm++_goodies~/appbar-1.0/Makefile
--- aewm++_goodies/appbar-1.0/Makefile	2002-11-11 18:36:35.000000000 -0600
+++ aewm++_goodies~/appbar-1.0/Makefile	2002-11-20 14:01:49.000000000 -0600
@@ -1,7 +1,7 @@
 CC        = g++
 CFLAGS    = -g -O2 -Wall
 prefix     = /usr
-INCLUDES  = -I$(prefix)/include
+INCLUDES  =
 LIBS      = -lX11 -lXext -lXpm 
 LDPATH    = -L/usr/X11R6/lib
 
diff -ur aewm++_goodies/appbar-1.0/icon.cc aewm++_goodies~/appbar-1.0/icon.cc
--- aewm++_goodies/appbar-1.0/icon.cc	2002-01-26 18:10:17.000000000 -0600
+++ aewm++_goodies~/appbar-1.0/icon.cc	2002-11-20 14:00:55.000000000 -0600
@@ -35,6 +35,8 @@
 
 #include "icon.hh"
 
+using namespace std;
+
 char * default_icon_xpm[] = {
 "16 16 16 1",
 " 	c None",
diff -ur aewm++_goodies/appbar-1.0/scanner.hh aewm++_goodies~/appbar-1.0/scanner.hh
--- aewm++_goodies/appbar-1.0/scanner.hh	2002-11-09 12:48:07.000000000 -0600
+++ aewm++_goodies~/appbar-1.0/scanner.hh	2002-11-20 14:06:21.000000000 -0600
@@ -25,13 +25,15 @@
 #ifndef _SCANNER_HH_
 #define _SCANNER_HH_
 
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <cstdlib>
 #include <string>
 
 #define MAX_TOKEN_BUFFER 256
 
+using namespace std;
+
 class Scanner {
 private:
 
diff -ur aewm++_goodies/setrootimage-1.0/Makefile aewm++_goodies~/setrootimage-1.0/Makefile
--- aewm++_goodies/setrootimage-1.0/Makefile	2002-11-11 18:29:35.000000000 -0600
+++ aewm++_goodies~/setrootimage-1.0/Makefile	2002-11-20 14:07:16.000000000 -0600
@@ -1,7 +1,7 @@
 CC        = g++
 CFLAGS    = -g -O2 -Wall
 prefix     = /usr
-INCLUDES  = -I$(prefix)/include
+INCLUDES  =
 LIBS      = -lX11 -lXext
 LDPATH    = -L/usr/X11R6/lib
 
diff -ur aewm++_goodies/setrootimage-1.0/setrootimage.cc aewm++_goodies~/setrootimage-1.0/setrootimage.cc
--- aewm++_goodies/setrootimage-1.0/setrootimage.cc	2002-02-02 08:14:22.000000000 -0600
+++ aewm++_goodies~/setrootimage-1.0/setrootimage.cc	2002-11-20 14:07:32.000000000 -0600
@@ -38,6 +38,8 @@
 #include <iostream>
 #include "image.hh"
 
+using namespace std;
+
 BColor parseCommandlineColor(char* the_color)
 {
 	BColor color;
