--- configure.orig	2005-03-12 14:05:28.427700360 +0100
+++ configure	2005-03-12 14:05:33.591915280 +0100
@@ -23596,12 +23596,10 @@
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
-
-
 if test "x$flavour" = "xmozilla"; then
-	as_ac_File=`echo "ac_cv_file_$MOZILLA_INCLUDE_ROOT/nsIPassword.h" | $as_tr_sh`
-echo "$as_me:$LINENO: checking for $MOZILLA_INCLUDE_ROOT/nsIPassword.h" >&5
-echo $ECHO_N "checking for $MOZILLA_INCLUDE_ROOT/nsIPassword.h... $ECHO_C" >&6
+	as_ac_File=`echo "ac_cv_file_$MOZILLA_INCLUDE_ROOT/wallet/nsIPassword.h" | $as_tr_sh`
+echo "$as_me:$LINENO: checking for $MOZILLA_INCLUDE_ROOT/wallet/nsIPassword.h" >&5
+echo $ECHO_N "checking for $MOZILLA_INCLUDE_ROOT/wallet/nsIPassword.h... $ECHO_C" >&6
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -23609,7 +23607,7 @@
   { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
-if test -r "$MOZILLA_INCLUDE_ROOT/nsIPassword.h"; then
+if test -r "$MOZILLA_INCLUDE_ROOT/wallet/nsIPassword.h"; then
   eval "$as_ac_File=yes"
 else
   eval "$as_ac_File=no"

