diff -uNr epiphany-1.2.9/configure epiphany-1.2.9.patched/configure
--- epiphany-1.2.9/configure	2004-09-11 16:32:47.000000000 +0200
+++ epiphany-1.2.9.patched/configure	2004-09-16 16:01:43.000000000 +0200
@@ -22061,7 +22061,7 @@
   1.7) MOZILLA_SNAPSHOT=17 ;;
 1.7.1) MOZILLA_SNAPSHOT=17 ;;
 1.7.2) MOZILLA_SNAPSHOT=17 ;;
-1.7.3) MOZILLA_SNAPSHOT=18 ;;
+1.7.3) MOZILLA_SNAPSHOT=17 ;;
 1.7.4) MOZILLA_SNAPSHOT=18 ;;
 1.7.5) MOZILLA_SNAPSHOT=18 ;;
 branch1.7) MOZILLA_SNAPSHOT=18 ;;
diff -uNr epiphany-1.2.9/configure.in epiphany-1.2.9.patched/configure.in
--- epiphany-1.2.9/configure.in	2004-09-05 01:02:32.000000000 +0200
+++ epiphany-1.2.9.patched/configure.in	2004-09-16 16:01:19.000000000 +0200
@@ -205,7 +205,7 @@
   1.7) MOZILLA_SNAPSHOT=17 ;;
 1.7.1) MOZILLA_SNAPSHOT=17 ;;
 1.7.2) MOZILLA_SNAPSHOT=17 ;;
-1.7.3) MOZILLA_SNAPSHOT=18 ;;
+1.7.3) MOZILLA_SNAPSHOT=17 ;;
 1.7.4) MOZILLA_SNAPSHOT=18 ;;
 1.7.5) MOZILLA_SNAPSHOT=18 ;;
 branch1.7) MOZILLA_SNAPSHOT=18 ;;
