diff -ur epiphany-1.4.6/configure epiphany-1.4.6-patched/configure
--- epiphany-1.4.6/configure	2004-11-29 03:54:24.000000000 +0800
+++ epiphany-1.4.6-patched/configure	2004-12-20 16:15:06.608361048 +0800
@@ -23030,8 +23030,8 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <nsIURI.h>
-		  #include <nsIGlobalHistory.h>
+#include <necko/nsIURI.h>
+#include <docshell/nsIGlobalHistory.h>
 		  #include <docshell/nsIGlobalHistory2.h>
 		  #include <history/nsIBrowserHistory.h>
 int
diff -ur epiphany-1.4.6/configure.in epiphany-1.4.6-patched/configure.in
--- epiphany-1.4.6/configure.in	2004-11-25 05:14:25.000000000 +0800
+++ epiphany-1.4.6-patched/configure.in	2004-12-20 16:14:43.770832880 +0800
@@ -259,8 +259,8 @@
 
 AC_COMPILE_IFELSE(
 	[AC_LANG_PROGRAM(
-		[[#include <nsIURI.h>
-		  #include <nsIGlobalHistory.h>
+		[[#include <necko/nsIURI.h>
+		  #include <docshell/nsIGlobalHistory.h>
 		  #include <docshell/nsIGlobalHistory2.h>
 		  #include <history/nsIBrowserHistory.h>]],
 		[[nsIBrowserHistory* p;
