3cc902d2fd203127a98e027f42175db35dafa579
diff --git a/embed/downloader-view.c b/embed/downloader-view.c
index 66de70b..68ce1af 100644
--- a/embed/downloader-view.c
+++ b/embed/downloader-view.c
@@ -141,11 +141,6 @@ downloader_view_class_init (DownloaderViewClass *klass)
 	object_class->finalize = downloader_view_finalize;
 
 	g_type_class_add_private (object_class, sizeof(DownloaderViewPrivate));
-
-#ifdef HAVE_LIBNOTIFY
-	notify_init (PACKAGE);
-#endif
-
 }
 
 static void
@@ -289,13 +284,6 @@ downloader_view_finalize (GObject *object)
 
 	G_OBJECT_CLASS (downloader_view_parent_class)->finalize (object);
 
-#ifdef HAVE_LIBNOTIFY	
-	if (notify_is_initted ())
-	{
-		notify_uninit ();
-	}
-#endif
-
 	if (idle_unref)
 	{
 		ephy_object_idle_unref (embed_shell);
diff --git a/src/ephy-main.c b/src/ephy-main.c
index 13c8f9e..4e92d42 100644
--- a/src/ephy-main.c
+++ b/src/ephy-main.c
@@ -50,6 +50,10 @@
 #include <errno.h>
 #include <string.h>
 
+#ifdef HAVE_LIBNOTIFY
+#include <libnotify/notify.h>
+#endif
+
 static GQuark startup_error_quark = 0;
 #define STARTUP_ERROR_QUARK	(startup_error_quark)
 
@@ -737,9 +741,18 @@ main (int argc,
 	g_object_weak_ref (G_OBJECT (ephy_shell), shell_weak_notify, NULL);
 	ephy_object_idle_unref (ephy_shell);
 
+#ifdef HAVE_LIBNOTIFY	
+	/* Init notifications for the download manager */
+	notify_init (PACKAGE);
+#endif
+
 	gtk_main ();
 
 	/* Shutdown */
+#ifdef HAVE_LIBNOTIFY	
+	if (notify_is_initted ())
+		notify_uninit ();
+#endif
 	eel_gconf_monitor_remove ("/apps/epiphany/general");
 	gnome_accelerators_sync ();
 	ephy_state_save ();
