--- src/configure.ac
+++ src/configure.ac
@@ -187,6 +187,10 @@
   # Need libdwfl-capable recent elfutils from Fedora
   save_LIBS="$LIBS"
   AS_IF([test "x$enable_staticdw" != xyes],[
+    AC_CHECK_LIB(dl, dlopen,,[
+      AC_MSG_ERROR([missing libdl])])
+    AC_CHECK_LIB(elf, gelf_getehdr,,[
+      AC_MSG_ERROR([missing elfutils development headers/libraries (dw 0.123+)])])
     AC_CHECK_LIB(dw, dwfl_module_getsym,,[
       AC_MSG_ERROR([missing elfutils development headers/libraries (dw 0.123+)])])
     AC_CHECK_LIB(ebl, ebl_openbackend,,[
