--- config_office/configure.in.orig	2007-01-13 14:18:26.000000000 +0100
+++ config_office/configure.in	2007-01-13 14:20:11.000000000 +0100
@@ -58,6 +58,10 @@
 [  --with-firefox          Use Firefox instead of Mozilla Suite for building.
                           Only honoured with --with-system-mozilla
 ],,)
+AC_ARG_WITH(seamonkey,
+[  --with-seamonkey        Use Seamonkey instead of Mozilla Suite for building.
+                          Only honoured with --with-system-mozilla
+],,)
 AC_ARG_WITH(xulrunner,
 [  --with-xulrunner        Use XULRunner instead of Mozilla Suite for building.
                           Only honoured with --with-system-mozilla
@@ -3714,6 +3718,8 @@
     AC_MSG_CHECKING([which Mozilla flavour to use])
     if test -n "$with_firefox" && test "$with_firefox" != "no"; then
       MOZ_FLAVOUR=firefox
+    elif test -n "$with_seamonkey" && test "$with_seamonkey" != "no"; then
+      MOZ_FLAVOUR=seamonkey
     elif test -n "$with_xulrunner" && test "$with_xulrunner" != "no"; then
       MOZ_FLAVOUR=xulrunner
     else
