--- bin/package-ooo 
+++ bin/package-ooo 
@@ -57,6 +57,7 @@
 #!/bin/sh
 export OOO_EXTRA_ARG='$2'
+unset PYTHONPATH
 $OOINSTBASE/program/ooqstart $3 "\$@"
 EOT
     chmod 755 $1
 }
--- patches/src680/apply 
+++ patches/src680/apply 
@@ -1630,6 +1630,8 @@
 
 [ GentooExperimental ]
 disable-parser-jaxp.diff, hmth
+
+[ GentooOnly ]
 # db.jar unused with CXXHelpLinker
 no-db.jar.diff, i#70155, hmth
 
