diff -ur hearts-1.98.orig/common/communication/protocol.cpp hearts-1.98/common/communication/protocol.cpp
--- hearts-1.98.orig/common/communication/protocol.cpp	2004-02-03 01:57:42.000000000 +0200
+++ hearts-1.98/common/communication/protocol.cpp	2009-08-12 03:16:25.000000000 +0300
@@ -1,5 +1,6 @@
 #include "protocol.h"
 #include "general/helper.h"
+#include <cstdio>
 #include <cassert>
 
 using namespace std;
