diff -Nur a/docs/inspircd.conf.example b/docs/inspircd.conf.example
--- a/docs/inspircd.conf.example	2011-06-24 03:21:01.000000000 +0200
+++ b/docs/inspircd.conf.example	2012-04-03 19:09:59.649826121 +0200
@@ -401,11 +401,11 @@
 
 # This file has all the information about oper classes, types and o:lines.
 # You *MUST* edit it.
-<include file="conf/opers.conf.example">
+<include file="/etc/inspircd/opers.conf.example">
 
 # This file has all the information about server links and ulined servers.
 # You *MUST* edit it if you intend to link servers.
-<include file="conf/links.conf.example">
+<include file="/etc/inspircd/links.conf.example">
 
 #-#-#-#-#-#-#-#-#-#-  MISCELLANEOUS CONFIGURATION  -#-#-#-#-#-#-#-#-#-#
 #                                                                     #
@@ -415,7 +415,8 @@
 #   motd - displayed on connect and when a user executes /MOTD
 #   rules - displayed when the user executes /RULES
 # Modules can also define their own files
-<files motd="conf/inspircd.motd.example" rules="conf/inspircd.rules.example">
+<files motd="/etc/inspircd/inspircd.motd.example"
+       rules="/etc/inspircd/inspircd.rules.example">
 
 # Example of an executable file include. Note this will be read on rehash,
 # not when the command is run.
@@ -461,7 +462,7 @@
 # the default of 'inspircd.pid' is used.                              #
 #                                                                     #
 
-#<pid file="/path/to/inspircd.pid">
+<pid file="/var/run/inspircd/inspircd.pid">
 
 #-#-#-#-#-#-#-#-#-#-#-#-#- BANLIST LIMITS #-#-#-#-#-#-#-#-#-#-#-#-#-#-#
 #                                                                     #
@@ -629,7 +630,8 @@
 
              # nouserdns: If enabled, no DNS lookups will be performed on
              # connecting users. This can save a lot of resources on very busy servers.
-             nouserdns="no">
+	     # Set nouserdns to yes as a workaround for CVE-2012-1836.
+             nouserdns="yes">
 
 #-#-#-#-#-#-#-#-#-#-#-# SECURITY CONFIGURATION  #-#-#-#-#-#-#-#-#-#-#-#
 #                                                                     #
@@ -907,7 +909,7 @@
 # provide almost all the features of InspIRCd. :)                     #
 #                                                                     #
 # The default does nothing -- we include it for simplicity for you.   #
-<include file="conf/modules.conf.example">
+<include file="/etc/inspircd/modules.conf.example">
 
 # Here are some pre-built modules.conf files that closely match the
 # default configurations of some popular IRCd's. You still may want to
@@ -919,10 +921,10 @@
 # recommended that you make your own modules file based on modules.conf.example.
 
 # Settings similar to UnrealIRCd defaults.
-#<include file="conf/modules/modules.conf.unreal">
+#<include file="/etc/inspircd/modules/modules.conf.unreal">
 
 # Settings similar to Charybdis IRCd defaults.
-#<include file="conf/modules/modules.conf.charybdis">
+#<include file="/etc/inspircd/modules/modules.conf.charybdis">
 
 
 #########################################################################
diff -Nur a/make/template/main.mk b/make/template/main.mk
--- a/make/template/main.mk	2011-06-24 03:21:01.000000000 +0200
+++ b/make/template/main.mk	2012-04-03 19:09:59.649826121 +0200
@@ -54,22 +54,22 @@
   CXXFLAGS += -D_ALL_SOURCE -I/usr/local/include
 @ENDIF
 
-@IFNDEF D
-  D=0
+@IFNDEF DEBUGLEVEL
+  DEBUGLEVEL=0
 @ENDIF
 
 DBGOK=0
-@IFEQ $(D) 0
+@IFEQ $(DEBUGLEVEL) 0
   CXXFLAGS += -O2 -g1
   HEADER = std-header
   DBGOK=1
 @ENDIF
-@IFEQ $(D) 1
+@IFEQ $(DEBUGLEVEL) 1
   CXXFLAGS += -O0 -g3 -Werror
   HEADER = debug-header
   DBGOK=1
 @ENDIF
-@IFEQ $(D) 2
+@IFEQ $(DEBUGLEVEL) 2
   CXXFLAGS += -O2 -g3
   HEADER = debug-header
   DBGOK=1
@@ -111,10 +111,10 @@
     @GNU_ONLY TARGET = modules/$(M:.so=).so
 @ENDIF
 
-@IFDEF T
+@IFDEF TGT
     HEADER =
     FOOTER = target
-    TARGET = $(T)
+    TARGET = $(TGT)
 @ENDIF
 
 @IFEQ $(DBGOK) 0
diff -Nur a/src/modules/extra/m_ssl_gnutls.cpp b/src/modules/extra/m_ssl_gnutls.cpp
--- a/src/modules/extra/m_ssl_gnutls.cpp	2011-06-24 03:21:01.000000000 +0200
+++ b/src/modules/extra/m_ssl_gnutls.cpp	2012-04-03 19:09:59.649826121 +0200
@@ -20,7 +20,7 @@
 
 /* $ModDesc: Provides SSL support for clients */
 /* $CompileFlags: pkgconfincludes("gnutls","/gnutls/gnutls.h","") */
-/* $LinkerFlags: rpath("pkg-config --libs gnutls") pkgconflibs("gnutls","/libgnutls.so","-lgnutls") */
+/* $LinkerFlags: rpath("pkg-config --libs gnutls") pkgconflibs("gnutls","/libgnutls.so","-lgnutls") -lgcrypt */
 
 enum issl_status { ISSL_NONE, ISSL_HANDSHAKING_READ, ISSL_HANDSHAKING_WRITE, ISSL_HANDSHAKEN, ISSL_CLOSING, ISSL_CLOSED };
 
diff -Nur a/src/modules/m_xline_db.cpp b/src/modules/m_xline_db.cpp
--- a/src/modules/m_xline_db.cpp	2011-06-24 03:21:01.000000000 +0200
+++ b/src/modules/m_xline_db.cpp	2012-04-03 19:11:38.389825566 +0200
@@ -92,7 +92,7 @@
 		 *		-- w00t
 		 */
 		ServerInstance->Logs->Log("m_xline_db",DEBUG, "xlinedb: Opening temporary database");
-		f = fopen("data/xline.db.new", "w");
+		f = fopen("/var/lib/inspircd/data/xline.db.new", "w");
 		if (!f)
 		{
 			ServerInstance->Logs->Log("m_xline_db",DEBUG, "xlinedb: Cannot create database! %s (%d)", strerror(errno), errno);
@@ -133,7 +133,7 @@
 		}
 
 		// Use rename to move temporary to new db - this is guarenteed not to fuck up, even in case of a crash.
-		if (rename("data/xline.db.new", "data/xline.db") < 0)
+		if (rename("/var/lib/inspircd/data/xline.db.new", "/var/lib/inspircd/data/xline.db") < 0)
 		{
 			ServerInstance->Logs->Log("m_xline_db",DEBUG, "xlinedb: Cannot move new to old database! %s (%d)", strerror(errno), errno);
 			ServerInstance->SNO->WriteToSnoMask('a', "database: cannot replace old with new db: %s (%d)", strerror(errno), errno);
@@ -149,7 +149,7 @@
 		char linebuf[MAXBUF];
 		unsigned int lineno = 0;
 
-		f = fopen("data/xline.db", "r");
+		f = fopen("/var/lib/inspircd/data/xline.db", "r");
 		if (!f)
 		{
 			if (errno == ENOENT)
