--- errors.c.orig	2008-07-17 22:43:28.791326638 -0400
+++ errors.c	2008-07-17 22:44:37.258432153 -0400
@@ -68,12 +68,12 @@
 	time(&c_time);
 	l_time = localtime(&c_time);
 	strftime(logmsg, sizeof(logmsg), "[%H:%M:%S] ", l_time);
-	strcat(logmsg, errmsg);
+	strncat(logmsg, errmsg, sizeof(logmsg)-strlen(logmsg)-2);
 
 	if(message[strlen(message)-1] != '\n') {
 		strcat(logmsg, "\n");
 	}
-	fprintf(err_file, logmsg);
+	fputs(logmsg, err_file);
   	fflush (err_file);
 }
 
