--- Slim/bootstrap.pm.old	2010-07-06 08:30:49.267542493 +0100
+++ Slim/bootstrap.pm	2010-07-06 08:30:49.255502095 +0100
@@ -192,27 +192,6 @@
 			print "To run from source on Windows, please install ActivePerl 5.10.  ActivePerl 5.8.8 is no longer supported.\n";
 			print "http://www.activestate.com/activeperl/\n\n";
 		}
-		else {
-			print qq{
-*******
-
-NOTE:
-
-If you're running some unsupported Linux/Unix platform, please use the buildme.sh 
-script located here:
-
-http://svn.slimdevices.com/repos/slim/7.5/trunk/vendor/CPAN/
-
-You should never need to do this if you're on Windows or Mac OSX. If the installers
-don't work for you, ask for help and/or report a bug.
-
-If 7.5 is outdated by the time you read this, Replace "7.5" with the major version
-of Squeezebox Server you are running.
-
-*******
-
-			\n};
-		}
 		
 		print "Exiting..\n";
 
