diff -Naur --exclude-from=/home/rathnor/fluxbox/work/diff-exclude fluxbox-0.1.14.orig/src/Window.cc fluxbox-0.1.14/src/Window.cc
--- fluxbox-0.1.14.orig/src/Window.cc	Sun Dec  8 07:04:24 2002
+++ fluxbox-0.1.14/src/Window.cc	Tue Dec 10 01:00:53 2002
@@ -71,7 +71,7 @@
 display(0),
 lastButtonPressTime(0),
 m_windowmenu(0),
-m_layer(LAYER_NORMAL),
+m_layer(LAYER_NORMAL), old_decoration(DECOR_NORMAL),
 tab(0) {
 	
 	lastFocusTime.tv_sec = lastFocusTime.tv_usec = 0;
@@ -3178,7 +3179,11 @@
 		setDecoration(DECOR_NONE); 
 		decorations.enabled = false;
 	} else { //revert back to old decoration
-		setDecoration(old_decoration);
+		if (old_decoration == DECOR_NONE) { // make sure something happens
+			setDecoration(DECOR_NORMAL);
+		} else {
+			setDecoration(old_decoration);
+		}
 		decorations.enabled = true;
 	}
 }
