diff -U1 -r1.59 Window.cc
--- Window.cc	4 Jun 2002 09:50:58 -0000	1.59
+++ Window.cc	9 Jun 2002 12:31:01 -0000
@@ -3703,5 +3703,10 @@
 				
-			while (tr->client.transient) 
+			while (tr->client.transient) {
 				tr = tr->client.transient;
-				
+				if (tr == tr->client.transient) { //ops! something is wrong with transient
+					tr->client.transient = 0;
+				}
+			}
+			
+
 			client.transient_for = tr;
@@ -3713,4 +3718,8 @@
 					
-				while (tr->client.transient) 
+				while (tr->client.transient) {
 					tr = tr->client.transient;
+					if (tr == tr->client.transient) { //ops! somehtin is wrong with transient
+						tr->client.transient = 0;
+					}
+				}
 					
