--- tests/test___init__.py.old	2006-11-04 22:14:02.000000000 +0100
+++ tests/test___init__.py	2006-11-04 22:17:38.000000000 +0100
@@ -146,6 +146,6 @@
         from mutagen import version
         from os.path import dirname, basename, abspath
         trunk = basename(dirname(dirname(dirname(abspath(__file__)))))
-        if trunk != "trunk":
+        if trunk == "trunk":
             self.failIf(version[-1] < 0, "mutagen.version is a prerelease.")
 add(TMutagen)
