diff -ruN coda-6.0.10/coda-src/scripts/venus-setup coda-6.0.10-fixed/coda-src/scripts/venus-setup
--- coda-6.0.10/coda-src/scripts/venus-setup	2005-04-21 22:03:47.000000000 +0200
+++ coda-6.0.10-fixed/coda-src/scripts/venus-setup	2005-05-04 21:26:49.354219336 +0200
@@ -190,7 +190,7 @@
   fi
 
   # make the psdev, on udev and devfs systems the device should exist by now
-  if [ ! -c /dev/cfs0 -o ! -c /dev/coda/0 ] ; then
+  if [ ! -c /dev/cfs0 -a ! -c /dev/coda/0 ] ; then
     if [ ! -x /dev/MAKEDEV ] ; then
       echo 'You need a character device for the Coda kernel module'
       echo 'On *BSD systems you probably have to run "mknod /dev/cfs0 c 93 0"'
