diff -urN clearsilver-0.10.1.orig/cgi/Makefile clearsilver-0.10.1/cgi/Makefile
--- clearsilver-0.10.1.orig/cgi/Makefile	2005-08-28 21:20:03.154672096 +0000
+++ clearsilver-0.10.1/cgi/Makefile	2005-08-28 21:22:22.077552608 +0000
@@ -15,6 +15,7 @@
 STATIC_OBJ = $(STATIC_SRC:%.c=%.o)
 STATIC_CSO = $(STATIC_EXE:%.cgi=%.cso)
 
+CFLAGS += -fPIC
 DLIBS += -lneo_cgi -lneo_cs -lneo_utl # -lefence
 
 TARGETS = $(CGI_LIB) $(STATIC_EXE) $(STATIC_CSO)
diff -urN clearsilver-0.10.1.orig/cs/Makefile clearsilver-0.10.1/cs/Makefile
--- clearsilver-0.10.1.orig/cs/Makefile	2005-08-28 21:20:03.141674072 +0000
+++ clearsilver-0.10.1/cs/Makefile	2005-08-28 21:22:42.327474152 +0000
@@ -22,7 +22,7 @@
 CSDUMP_SRC = csdump.c
 CSDUMP_OBJ = $(CSDUMP_SRC:%.c=%.o)
 
-CFLAGS += -I$(NEOTONIC_ROOT)
+CFLAGS += -I$(NEOTONIC_ROOT) -fPIC
 LIBS += -L$(LIB_DIR) -lneo_cs -lneo_utl # -lefence
 
 TARGETS = $(CS_LIB) $(CSTEST_EXE) $(CSR_EXE) test
diff -urN clearsilver-0.10.1.orig/java-jni/Makefile clearsilver-0.10.1/java-jni/Makefile
--- clearsilver-0.10.1.orig/java-jni/Makefile	2005-08-28 21:20:03.170669664 +0000
+++ clearsilver-0.10.1/java-jni/Makefile	2005-08-28 21:23:03.941188368 +0000
@@ -16,7 +16,7 @@
 NEO_UTIL_OBJ = $(NEO_UTIL_SRC:%.c=%.o)
 
 
-CFLAGS += -I$(NEOTONIC_ROOT) -I$(JAVA_PATH)/include -I$(JAVA_PATH)/include/linux
+CFLAGS += -I$(NEOTONIC_ROOT) -I$(JAVA_PATH)/include -I$(JAVA_PATH)/include/linux -fPIC
 DLIBS += -lneo_cgi -lneo_cs -lneo_utl 
 LIBS += -L$(LIB_DIR) $(DLIBS)
 
diff -urN clearsilver-0.10.1.orig/python/Makefile clearsilver-0.10.1/python/Makefile
--- clearsilver-0.10.1.orig/python/Makefile	2005-08-28 21:20:03.156671792 +0000
+++ clearsilver-0.10.1/python/Makefile	2005-08-28 21:23:24.182111280 +0000
@@ -11,7 +11,7 @@
 NEO_UTIL_SRC = neo_cgi.c neo_util.c neo_cs.c
 NEO_UTIL_OBJ = $(NEO_UTIL_SRC:%.c=%.o)
 
-CFLAGS += -I$(NEOTONIC_ROOT) $(PYTHON_INC)
+CFLAGS += -I$(NEOTONIC_ROOT) $(PYTHON_INC) -fPIC
 DLIBS += -lneo_cgi -lneo_cs -lneo_utl 
 LIBS += -L$(LIB_DIR) $(DLIBS) $(DB2_LIB)
 
diff -urN clearsilver-0.10.1.orig/util/Makefile clearsilver-0.10.1/util/Makefile
--- clearsilver-0.10.1.orig/util/Makefile	2005-08-28 21:20:03.128676048 +0000
+++ clearsilver-0.10.1/util/Makefile	2005-08-28 21:25:26.469520768 +0000
@@ -13,6 +13,8 @@
 UTL_OBJ = $(UTL_SRC:%.c=%.o) $(EXTRA_UTL_OBJS)
 UTL_HDR = $(UTL_SRC:%.c=%.h)
 
+CFLAGS += -fPIC
+
 TARGETS = $(UTL_LIB)
 
 all: $(TARGETS)
