=== modified file 'twisted/test/test_process.py'
--- twisted/test/test_process.py	
+++ twisted/test/test_process.py	
@@ -76,7 +76,7 @@
 
 class EchoProtocol(protocol.ProcessProtocol):
 
-    s = "1234567" * 1001
+    s = "1234567" * 11
     finished = 0
 
     def connectionMade(self):
@@ -86,7 +86,7 @@
 
     def outReceived(self, data):
         self.buffer += data
-        if len(self.buffer) == 70070:
+        if len(self.buffer) == 770:
             self.transport.closeStdin()
 
     def processEnded(self, reason):

