diff -uNr partimage-0.6.4-1.orig/src/client/Makefile.am partimage-0.6.4-1/src/client/Makefile.am
--- partimage-0.6.4-1.orig/src/client/Makefile.am	2004-02-04 01:57:36.000000000 +0100
+++ partimage-0.6.4-1/src/client/Makefile.am	2006-10-15 00:41:34.000000000 +0200
@@ -33,7 +33,7 @@
     mbr_backup.cpp 
 
 # all libs are added in LIBS, we needn't add them to partimage_LDADD 
-partimage_LDADD = -L/usr/lib -lslang fs/libfs.a ../shared/libshared.a
+partimage_LDADD = -L$(libdir) -lslang fs/libfs.a ../shared/libshared.a
 partimage_LDFLAGS = @PARTIMAGE_LDFLAGS@
 
 DEFS=@DEFS@ -DLOCALEDIR=\"${localedir}\" -D_REENTRANT -D_FILE_OFFSET_BITS=64
diff -uNr partimage-0.6.4-1.orig/src/client/Makefile.in partimage-0.6.4-1/src/client/Makefile.in
--- partimage-0.6.4-1.orig/src/client/Makefile.in	2004-04-29 22:08:53.000000000 +0200
+++ partimage-0.6.4-1/src/client/Makefile.in	2006-10-15 00:42:11.000000000 +0200
@@ -271,7 +271,7 @@
 
 
 # all libs are added in LIBS, we needn't add them to partimage_LDADD 
-partimage_LDADD = -L/usr/lib -lslang fs/libfs.a ../shared/libshared.a
+partimage_LDADD = -L$(libdir) -lslang fs/libfs.a ../shared/libshared.a
 partimage_LDFLAGS = @PARTIMAGE_LDFLAGS@
 
 DEFS = @DEFS@ -DLOCALEDIR=\"${localedir}\" -D_REENTRANT -D_FILE_OFFSET_BITS=64
diff -uNr partimage-0.6.4-1.orig/src/server/Makefile.am partimage-0.6.4-1/src/server/Makefile.am
--- partimage-0.6.4-1.orig/src/server/Makefile.am	2004-02-04 01:57:36.000000000 +0100
+++ partimage-0.6.4-1/src/server/Makefile.am	2006-10-15 00:42:52.000000000 +0200
@@ -21,7 +21,7 @@
                  partimaged-gui_dummy.h partimaged-gui_newt.h partimaged.h
 
 #partimaged_LDADD = -L/usr/lib -lz -lbz2 -lnewt -lpthread -lcrypt -lslang -lcom_err -lext2fs ../shared/libshared.a
-partimaged_LDADD = -L/usr/lib -lslang ../shared/libshared.a
+partimaged_LDADD = -L$(libdir) -lslang ../shared/libshared.a
 
 partimaged_LDFLAGS = @PARTIMAGE_LDFLAGS@
 
diff -uNr partimage-0.6.4-1.orig/src/server/Makefile.in partimage-0.6.4-1/src/server/Makefile.in
--- partimage-0.6.4-1.orig/src/server/Makefile.in	2004-04-29 22:08:53.000000000 +0200
+++ partimage-0.6.4-1/src/server/Makefile.in	2006-10-15 00:43:38.000000000 +0200
@@ -269,7 +269,7 @@
 
 
 #partimaged_LDADD = -L/usr/lib -lz -lbz2 -lnewt -lpthread -lcrypt -lslang -lcom_err -lext2fs ../shared/libshared.a
-partimaged_LDADD = -L/usr/lib -lslang ../shared/libshared.a
+partimaged_LDADD = -L$(libdir) -lslang ../shared/libshared.a
 
 partimaged_LDFLAGS = @PARTIMAGE_LDFLAGS@
 
