--- partimage-0.6.6/Makefile.in.orig	2007-08-11 12:36:57.000000000 +0200
+++ partimage-0.6.6/Makefile.in	2008-01-14 00:39:42.000000000 +0100
@@ -688,34 +688,34 @@
 	@echo "/etc/pam.d/partimaged"
 
 install-data-local:
-	$(mkinstalldirs) ${sysconfdir}
-	$(mkinstalldirs) ${sysconfdir}/partimaged
+	$(mkinstalldirs) ${DESTDIR}${sysconfdir}
+	$(mkinstalldirs) ${DESTDIR}${sysconfdir}/partimaged
 
-	if test ! -f ${sysconfdir}/partimaged/partimagedusers ; then \
+	if test ! -f ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ; then \
 	echo "# note: '#' introduces comments" > \
-          ${sysconfdir}/partimaged/partimagedusers ;\
+          ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
 	echo -n "#add only users allowed to " >>  \
-           ${sysconfdir}/partimaged/partimagedusers ;\
-	echo "connect partimaged" >> ${sysconfdir}/partimaged/partimagedusers ;\
+           ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
+	echo "connect partimaged" >> ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
 	echo "# (only one login per line)" >> \
-           ${sysconfdir}/partimaged/partimagedusers ;\
-	echo "" >> ${sysconfdir}/partimaged/partimagedusers ;\
+           ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
+	echo "" >> ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
 	echo "#sample # user 'sample' is allowed to connect partimaged" >> \
-           ${sysconfdir}/partimaged/partimagedusers ;\
-	chmod 600 ${sysconfdir}/partimaged/partimagedusers ;\
-	chown partimag:root ${sysconfdir}/partimaged/partimagedusers ;\
+           ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
+	chmod 600 ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
+	chown partimag:root ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
 	fi 
 
-	$(mkinstalldirs) ${infodir}
-	$(INSTALL_DATA) AUTHORS ${infodir}/AUTHORS
-	$(INSTALL_DATA) BUGS ${infodir}/BUGS
-	$(INSTALL_DATA) COPYING ${infodir}/COPYING
-	$(INSTALL_DATA) ChangeLog ${infodir}/ChangeLog
-	$(INSTALL_DATA) INSTALL ${infodir}/INSTALL
-	$(INSTALL_DATA) README ${infodir}/README
-	$(INSTALL_DATA) README.partimaged ${infodir}/README.partimaged
-	$(INSTALL_DATA) TODO ${infodir}/TODO
-	$(INSTALL_DATA) partimage.lsm ${infodir}/partimage.lsm
+	$(mkinstalldirs) ${DESTDIR}${infodir}
+	$(INSTALL_DATA) AUTHORS ${DESTDIR}${infodir}/AUTHORS
+	$(INSTALL_DATA) BUGS ${DESTDIR}${infodir}/BUGS
+	$(INSTALL_DATA) COPYING ${DESTDIR}${infodir}/COPYING
+	$(INSTALL_DATA) ChangeLog ${DESTDIR}${infodir}/ChangeLog
+	$(INSTALL_DATA) INSTALL ${DESTDIR}${infodir}/INSTALL
+	$(INSTALL_DATA) README ${DESTDIR}${infodir}/README
+	$(INSTALL_DATA) README.partimaged ${DESTDIR}${infodir}/README.partimaged
+	$(INSTALL_DATA) TODO ${DESTDIR}${infodir}/TODO
+	$(INSTALL_DATA) partimage.lsm ${DESTDIR}${infodir}/partimage.lsm
 
 uninstall-local:
 	-rm -f ${infodir}/AUTHORS
