--- partimage-0.6.4/Makefile.in.chris	2005-01-14 01:32:01.000000000 +0100
+++ partimage-0.6.4/Makefile.in	2005-01-14 01:33:01.483820264 +0100
@@ -685,16 +685,16 @@
 	chmod 600 ${sysconfdir}/partimaged/partimagedusers ;\
 	fi 
 
-	$(mkinstalldirs) ${infodir}
-	$(INSTALL_DATA) AUTHORS ${infodir}/AUTHORS
-	$(INSTALL_DATA) BUGS ${infodir}/BUGS
-	$(INSTALL_DATA) COPYING ${infodir}/COPYING
-	$(INSTALL_DATA) ChangeLog ${infodir}/ChangeLog
-	$(INSTALL_DATA) INSTALL ${infodir}/INSTALL
-	$(INSTALL_DATA) README ${infodir}/README
-	$(INSTALL_DATA) README.partimaged ${infodir}/README.partimaged
-	$(INSTALL_DATA) TODO ${infodir}/TODO
-	$(INSTALL_DATA) partimage.lsm ${infodir}/partimage.lsm
+	#$(mkinstalldirs) ${infodir}
+	#$(INSTALL_DATA) AUTHORS ${infodir}/AUTHORS
+	#$(INSTALL_DATA) BUGS ${infodir}/BUGS
+	#$(INSTALL_DATA) COPYING ${infodir}/COPYING
+	#$(INSTALL_DATA) ChangeLog ${infodir}/ChangeLog
+	#$(INSTALL_DATA) INSTALL ${infodir}/INSTALL
+	#$(INSTALL_DATA) README ${infodir}/README
+	#$(INSTALL_DATA) README.partimaged ${infodir}/README.partimaged
+	#$(INSTALL_DATA) TODO ${infodir}/TODO
+	#$(INSTALL_DATA) partimage.lsm ${infodir}/partimage.lsm
 
 uninstall-local:
 	-rm -f ${infodir}/AUTHORS
