diff -uNr partimage-0.6.6.orig/Makefile.am partimage-0.6.6/Makefile.am
--- partimage-0.6.6.orig/Makefile.am	2006-04-22 16:59:22.000000000 +0200
+++ partimage-0.6.6/Makefile.am	2008-01-14 23:11:00.000000000 +0100
@@ -23,8 +23,6 @@
 	rm ${sysconfdir}/partimaged/partimaged.csr ;\
 	chmod 600 ${sysconfdir}/partimaged/partimaged.key ;\
 	chmod 600 ${sysconfdir}/partimaged/partimaged.cert ;\
-	chown partimag:root ${sysconfdir}/partimaged/partimaged.key ;\
-	chown partimag:root ${sysconfdir}/partimaged/partimaged.cert ;\
 	else \
 	echo "SSL disabled, no certificate will be generated." ;\
 	fi
@@ -57,7 +55,6 @@
 	echo "#sample # user 'sample' is allowed to connect partimaged" >> \
            ${sysconfdir}/partimaged/partimagedusers ;\
 	chmod 600 ${sysconfdir}/partimaged/partimagedusers ;\
-	chown partimag:root ${sysconfdir}/partimaged/partimagedusers ;\
 	fi 
 
 	$(mkinstalldirs) ${infodir}
diff -uNr partimage-0.6.6.orig/Makefile.in partimage-0.6.6/Makefile.in
--- partimage-0.6.6.orig/Makefile.in	2008-01-14 23:09:15.000000000 +0100
+++ partimage-0.6.6/Makefile.in	2008-01-14 23:11:07.000000000 +0100
@@ -670,8 +670,6 @@
 	rm ${sysconfdir}/partimaged/partimaged.csr ;\
 	chmod 600 ${sysconfdir}/partimaged/partimaged.key ;\
 	chmod 600 ${sysconfdir}/partimaged/partimaged.cert ;\
-	chown partimag:root ${sysconfdir}/partimaged/partimaged.key ;\
-	chown partimag:root ${sysconfdir}/partimaged/partimaged.cert ;\
 	else \
 	echo "SSL disabled, no certificate will be generated." ;\
 	fi
@@ -703,7 +701,6 @@
 	echo "#sample # user 'sample' is allowed to connect partimaged" >> \
            ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
 	chmod 600 ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
-	chown partimag:root ${DESTDIR}${sysconfdir}/partimaged/partimagedusers ;\
 	fi 
 
 	$(mkinstalldirs) ${DESTDIR}${infodir}
