Index: src/client/main.cpp
===================================================================
--- src/client/main.cpp	(Revision 76)
+++ src/client/main.cpp	(Arbeitskopie)
@@ -842,7 +842,8 @@
   printf("===============================================================================\n");
   printf(i18n("Partition Image (http://www.partimage.org/) version %s [%s]\n"
 	      "---- distributed under the GPL 2 license (GNU General Public License) ----\n\n"
-	      "Supported file systems:....ext2fs, reiserfs, fat16, fat32, ntfs(exp), hpfs\n\n"
+	      "Supported file systems: Ext2/3, Reiser3, FAT16/32, HPFS, JFS, XFS, \n"
+	      "                        UFS(beta), HFS(beta), NTFS(experimental)\n\n"
 	      "usage: partimage [options] <action> <device> <image_file>\n"
 	      "       partimage <imginfo/restmbr> <image_file>\n\n"
 	      "ex: partimage -z1 -o -d save /dev/hda12 /mnt/backup/redhat-6.2.partimg.gz\n"
@@ -858,39 +859,42 @@
 	      "  - restmbr: restore a MBR of the image file to an hard disk\n"
 	      "  - imginfo: show informations about the image file\n"
 	      "* <device>: partition to save/restore (example: /dev/hda1)\n"
-	      "* <image_file>: file where data will be read/written. can be very big\n\n"
+	      "* <image_file>: file where data will be read/written. Can be very big.\n"
+	      "                For restore, <image_file> can have the value 'stdin'. This allows\n"
+              "                for providing image files through a pipe.\n\n"
 	      "Options:\n"
-	      "* -z,  --compress     (image file compression level):\n"
-	      "  -z0, --compress=0   don't compress: very fast but very big image file\n"
-	      "  -z1, --compress=1   compress using gzip: fast and small image file (default)\n"
-	      "  -z2, --compress=2   (compress using bzip2: very slow and very small image file):\n"
-	      "* -c,  --nocheck      don't check the partition before saving\n"
-	      "* -o,  --overwrite    overwrite the existing image file without confirmation\n"
-	      "* -d,  --nodesc       don't ask any description for the image file\n"
-	      "* -V,  --volume       (split image into multiple volumes files)\n"
-	      "  -VX, --volume=X     create volumes with a size of X KB\n"
-	      "* -w,  --waitvol      wait for a confirmation after each volume change\n"
-	      "* -e,  --erase        erase empty blocks on restore with zero bytes\n"
-	      "* -m,  --allowmnt     don't fail if the partition is mounted. Dangerous !\n"
-	      "* -M,  --nombr        don't create a backup of the MBR (Mast Boot Record) in the image file\n"
-	      "* -h,  --help         show help\n"
-	      "* -v,  --version      show version\n"
-	      "* -i,  --compilinfo   show compilation options used\n"
-	      "* -f,  --finish       (action to do if finished successfully):\n"
-	      "  -f0, --finish=0     wait: don't make anything\n"
-	      "  -f1, --finish=1     halt (power off) the computer\n"
-	      "  -f2, --finish=2     reboot (restart the computer):\n"
-	      "  -f3, --finish=3     quit\n"
-	      "* -b,  --batch        batch mode: the GUI won't wait for an user action\n"
-	      "* -y,  --nosync       don't synchronize the disks at the end of the operation (dangerous)\n"
-	      "* -sX, --server=X     give partimaged server's ip address\n"
-	      "* -pX, --port=X       give partimaged server's listening port\n"
-	      "* -g,  --debug=X      set the debug level to X (default: 1):\n"
-	      "* -n,  --nossl        disable SSL in network mode\n"
-	      "* -S,  --simulate     simulation of restoration mode\n"
-	      "* -aX, --automnt=X    Automatic mount with X options. Read the doc for more details\n"
-		  "* -UX  --username=X   Username to authenticate to server\n"
-		  "* -PX  --Password=X   Password for authentication of user to server\n"),
+	      "* -z,  --compress      (image file compression level):\n"
+	      "  -z0, --compress=0    don't compress: very fast but very big image file\n"
+	      "  -z1, --compress=1    compress using gzip: fast and small image file (default)\n"
+	      "  -z2, --compress=2    (compress using bzip2: very slow and very small image file):\n"
+	      "* -c,  --nocheck       don't check the partition before saving\n"
+	      "* -o,  --overwrite     overwrite the existing image file without confirmation\n"
+	      "* -d,  --nodesc        don't ask any description for the image file\n"
+	      "* -V,  --volume        (split image into multiple volumes files)\n"
+	      "  -VX, --volume=X      create volumes with a size of X MB\n"
+	      "* -w,  --waitvol       wait for a confirmation after each volume change\n"
+	      "* -e,  --erase         erase empty blocks on restore with zero bytes\n"
+	      "* -m,  --allowmnt      don't fail if the partition is mounted. Dangerous !\n"
+	      "* -M,  --nombr         don't create a backup of the MBR (Mast Boot Record) in the image file\n"
+	      "* -h,  --help          show help\n"
+	      "* -v,  --version       show version\n"
+	      "* -i,  --compilinfo    show compilation options used\n"
+	      "* -f,  --finish        (action to do if finished successfully):\n"
+	      "  -f0, --finish=0      wait: don't make anything\n"
+	      "  -f1, --finish=1      halt (power off) the computer\n"
+	      "  -f2, --finish=2      reboot (restart the computer):\n"
+	      "  -f3, --finish=3      quit\n"
+	      "* -b,  --batch         batch mode: the GUI won't wait for an user action\n"
+              "* -BX, --fully-batch=X batch mode without GUI, X is a challenge response string\n"
+	      "* -y,  --nosync        don't synchronize the disks at the end of the operation (dangerous)\n"
+	      "* -sX, --server=X      give partimaged server's ip address\n"
+	      "* -pX, --port=X        give partimaged server's listening port\n"
+	      "* -g,  --debug=X       set the debug level to X (default: 1):\n"
+	      "* -n,  --nossl         disable SSL in network mode\n"
+	      "* -S,  --simulate      simulation of restoration mode\n"
+	      "* -aX, --automnt=X     automatic mount with X options. Read the doc for more details\n"
+	      "* -UX  --username=X    username to authenticate to server\n"
+	      "* -PX  --password=X    password for authentication of user to server\n"),
 	 PACKAGE_VERSION, isVersionStable(PACKAGE_VERSION) ? i18n("stable") : i18n("unstable")); 
   printf("===============================================================================\n");
 }
Index: README
===================================================================
--- README	(Revision 73)
+++ README	(Arbeitskopie)
@@ -12,8 +12,17 @@
 the Free Software Foundation; either version 2 of the License, or
 at your option) any later version.
 
-Partition Image is a Linux utility which saves partitions in the ext2fs, reiserfs, fat16, and fat32
-formats an image file. Only used blocks are copied to save the space and to increase the speed.
+Partition Image is a partition imaging utility which has support for the following file systems:
+  * Ext2/3, the linux standard
+  * Reiser3, a journalised and powerful file system
+  * FAT16/32, DOS and Windows file systems
+  * HPFS, IBM OS/2 File System
+  * JFS, journalised File System, from IBM, used on AIX
+  * XFS, another journalised and efficient File System, from sgi, used on Irix
+  * UFS (beta), Unix File System
+  * HFS (beta), MacOS File System
+  * NTFS (experimental), Windows NT, 2000 and XP
+Only used blocks are copied to save the space and to increase the speed.
 The image file can be compressed, in gzip, or bzip2 formats, and can be splitted into
 small files to be copied on floppies (1,44 MB, ZIP, ...).
 Then, the partition can be restored if there is a problem on the partition (file system error,
