--- src/common/files.c.old	2010-03-01 09:07:30.000000000 +0100
+++ src/common/files.c	2010-03-01 09:08:18.000000000 +0100
@@ -894,6 +894,8 @@
 	/* start up with base by default */
 	FS_AddGameDirectory(va("%s/" BASEDIRNAME, fs_basedir->string));
 
+	FS_AddGameDirectory("@GAMES_LIBDIR@");
+
 	/* then add a '.ufoai/base' directory in home directory by default */
 	if (fs_usehomedir->integer)
 		FS_AddHomeAsGameDirectory(BASEDIRNAME);
