diff -ru java-config-1.3.5-orig/java_config/jc_util.py java-config-1.3.5/java_config/jc_util.py
--- java-config-1.3.5-orig/java_config/jc_util.py	2006-09-06 08:52:24.000000000 -0400
+++ java-config-1.3.5/java_config/jc_util.py	2006-09-06 08:56:53.000000000 -0400
@@ -104,10 +104,10 @@
 	can't be found
 	"""
 
-	jh = JAVA_HOME
-
 	if java_home is not None:
 		jh = java_home
+	else:
+		jh = get_java_home()
 
 	exec_0 = jh+'/bin/'+exe
 	exec_1 = jh+'/jre/bin/'+exe
