Index: src/eselect/java-nsplugin.eselect
===================================================================
--- src/eselect/java-nsplugin.eselect	(revision 2466)
+++ src/eselect/java-nsplugin.eselect	(working copy)
@@ -96,7 +96,7 @@
 ## }}}
 
 get_targets() {
-	for plugin in $(ls ${PLUGINS_HOME}/*-javaplugin.so);
+	for plugin in $(ls ${PLUGINS_HOME}/*-javaplugin.so 2>/dev/null);
 	do
 		echo $(plugin_to_vm ${plugin})
 	done
