diff -NrU5 ming-0.3.0.orig/src/Makefile ming-0.3.0/src/Makefile
--- ming-0.3.0.orig/src/Makefile	2008-06-09 02:00:14.000000000 +0200
+++ ming-0.3.0/src/Makefile	2008-06-09 02:00:30.000000000 +0200
@@ -30,11 +30,11 @@
 dynamic: $(SHAREDLIB)
 
 $(SHAREDLIB): $(OFILES)
 	cd blocks && $(MAKE) CFLAGS="$(CFLAGS) -I.."
 	cd actioncompiler && $(MAKE) CFLAGS="$(CFLAGS) -I.."
-	$(CC) $(OFILES) $(COMPILEROFILES) $(LDFLAGS) $(EXTRA_LIBS) blocks/*.o $(SHLIBLDFLAGS) -o $(SHAREDLIB)
+	$(CC) $(OFILES) $(COMPILEROFILES) $(LDFLAGS) blocks/*.o $(SHLIBLDFLAGS) -o $(SHAREDLIB) $(EXTRA_LIBS)
 	cp $(SHAREDLIB) ..
 	ln -fs $(SHAREDLIB) ../libming$(NOVAR_SHLIBEXT)
 	ln -fs $(SHAREDLIB) ../libming$(SHORT_SHLIBEXT)
 
 static: libming.a
diff -NrU5 ming-0.3.0.orig/util/Makefile ming-0.3.0/util/Makefile
--- ming-0.3.0.orig/util/Makefile	2008-06-09 02:00:15.000000000 +0200
+++ ming-0.3.0/util/Makefile	2008-06-09 02:07:22.000000000 +0200
@@ -1,9 +1,10 @@
 include ../Makefile.config
 
 CFLAGS += -I.. -I../src 
-LDFLAGS += -L.. -lming $(LIBS)
+LDFLAGS += -L..  -L/usr/lib/
+LIBS := -lming $(LIBS)
 
 UTILOBJ = action.o blocktypes.o decompile.o main.o parser.o read.o
 
 PAGER ?= less
 
@@ -26,62 +27,62 @@
 
 ../libming.a:
 	cd ..; $(MAKE) static
 
 listswf_d: $(UTILOBJ) outputtxt.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) $(UTILOBJ) outputtxt.c -o listswf_d
+	$(CC) $(CFLAGS) $(LDFLAGS) $(UTILOBJ) outputtxt.c -o listswf_d $(LIBS)
 
 listswf: $(UTILOBJ) outputtxt.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) -DNODECOMPILE $(UTILOBJ) outputtxt.c -o listswf
+	$(CC) $(CFLAGS) $(LDFLAGS) -DNODECOMPILE $(UTILOBJ) outputtxt.c -o listswf $(LIBS)
 
 listaction_d: $(UTILOBJ) outputtxt.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) -DACTIONONLY $(LDFLAGS) $(UTILOBJ) outputtxt.c -o listaction_d
+	$(CC) $(CFLAGS) -DACTIONONLY $(LDFLAGS) $(UTILOBJ) outputtxt.c -o listaction_d $(LIBS)
 
 listaction: $(UTILOBJ) outputtxt.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) -DACTIONONLY $(LDFLAGS) -DNODECOMPILE $(UTILOBJ) outputtxt.c -o listaction
+	$(CC) $(CFLAGS) -DACTIONONLY $(LDFLAGS) -DNODECOMPILE $(UTILOBJ) outputtxt.c -o listaction $(LIBS)
 
 swftoperl: $(UTILOBJ) outputscript.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) -DSWFPERL $(CFLAGS) $(LDFLAGS) outputscript.c $(UTILOBJ) -o swftoperl
+	$(CC) -DSWFPERL $(CFLAGS) $(LDFLAGS) outputscript.c $(UTILOBJ) -o swftoperl $(LIBS)
 
 swftophp: $(UTILOBJ) outputscript.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) -DSWFPHP $(CFLAGS) $(LDFLAGS) outputscript.c $(UTILOBJ) -o swftophp
+	$(CC) -DSWFPHP $(CFLAGS) $(LDFLAGS) outputscript.c $(UTILOBJ) -o swftophp $(LIBS)
 
 swftopython: $(UTILOBJ) outputscript.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) -DSWFPYTHON $(CFLAGS) $(LDFLAGS) outputscript.c $(UTILOBJ) -o swftopython
+	$(CC) -DSWFPYTHON $(CFLAGS) $(LDFLAGS) outputscript.c $(UTILOBJ) -o swftopython $(LIBS)
 
 parser.o: parser.c
-	$(CC) $(CFLAGS) -c parser.c
+	$(CC) $(CFLAGS) -c parser.c 
 
 decompile.o: decompile.c
 	$(CC) $(CFLAGS) -c decompile.c -o decompile.o
 
 ttftofft: ttftofft.c outputswf.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) `freetype-config --cflags` $(CFLAGS) -DNODECOMPILE $< outputswf.c `freetype-config --libs` $(LDFLAGS) -o $@
+	$(CC) `freetype-config --cflags` $(CFLAGS) $(LDFLAGS) -DNODECOMPILE $< outputswf.c -o $@ $(LIBS) `freetype-config --libs`
 
 makefdb: blocktypes.o makefdb.c read.o ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) makefdb.c read.o -o makefdb
+	$(CC) $(CFLAGS) $(LDFLAGS) makefdb.c read.o -o makefdb $(LIBS)
 
 gif2mask: gif2mask.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) gif2mask.c -o gif2mask 
+	$(CC) $(CFLAGS) $(LDFLAGS) gif2mask.c -o gif2mask $(LIBS)
 
 png2dbl: png2dbl.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) png2dbl.c -o png2dbl
+	$(CC) $(CFLAGS) $(LDFLAGS) png2dbl.c -o png2dbl $(LIBS)
 
 dbl2png: dbl2png.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) dbl2png.c -o dbl2png
+	$(CC) $(CFLAGS) $(LDFLAGS) dbl2png.c -o dbl2png $(LIBS)
 
 gif2dbl: gif2dbl.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) gif2dbl.c -o gif2dbl 
+	$(CC) $(CFLAGS) $(LDFLAGS) gif2dbl.c -o gif2dbl $(LIBS)
 
 raw2adpcm: raw2adpcm.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) raw2adpcm.c -o raw2adpcm 
+	$(CC) $(CFLAGS) $(LDFLAGS) raw2adpcm.c -o raw2adpcm  $(LIBS)
 
 png2swf: png2swf.c ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) png2swf.c -o png2swf 
+	$(CC) $(CFLAGS) $(LDFLAGS) png2swf.c -o png2swf $(LIBS)
 
 makeswf: makeswf.c makeswf.h ../libming$(NOVAR_SHLIBEXT)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< 
+	$(CC) $(CFLAGS) -o $@ $<  $(LDFLAGS) $(LIBS)
 
 install: $(BINARIES)
 	install -d $(bindir)
 	install -m 0755 $(BINARIES) $(bindir)
 	install -m 0755 ming-config $(bindir)
diff -NrU5 ming-0.3.0.orig/util/Makefile.orig ming-0.3.0/util/Makefile.orig
--- ming-0.3.0.orig/util/Makefile.orig	1970-01-01 01:00:00.000000000 +0100
+++ ming-0.3.0/util/Makefile.orig	2008-06-09 01:59:38.000000000 +0200
@@ -0,0 +1,87 @@
+include ../Makefile.config
+
+CFLAGS += -I.. -I../src 
+LDFLAGS += -L.. -lming $(LIBS)
+
+UTILOBJ = action.o blocktypes.o decompile.o main.o parser.o read.o
+
+PAGER ?= less
+
+BINARIES = listswf listswf_d swftoperl swftophp swftopython \
+	makeswf listaction listaction_d listfdb makefdb listjpeg raw2adpcm \
+	listmp3 dbl2png ttftofft
+
+BINARIES += $(EXTRA_BINS)
+
+all: $(BINARIES)
+
+help:
+	$(PAGER) README
+
+clean:
+	rm -f *.o core $(BINARIES)
+
+../libming$(NOVAR_SHLIBEXT):
+	cd ..; $(MAKE) dynamic
+
+../libming.a:
+	cd ..; $(MAKE) static
+
+listswf_d: $(UTILOBJ) outputtxt.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(UTILOBJ) outputtxt.c -o listswf_d
+
+listswf: $(UTILOBJ) outputtxt.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) -DNODECOMPILE $(UTILOBJ) outputtxt.c -o listswf
+
+listaction_d: $(UTILOBJ) outputtxt.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) -DACTIONONLY $(LDFLAGS) $(UTILOBJ) outputtxt.c -o listaction_d
+
+listaction: $(UTILOBJ) outputtxt.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) -DACTIONONLY $(LDFLAGS) -DNODECOMPILE $(UTILOBJ) outputtxt.c -o listaction
+
+swftoperl: $(UTILOBJ) outputscript.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) -DSWFPERL $(CFLAGS) $(LDFLAGS) outputscript.c $(UTILOBJ) -o swftoperl
+
+swftophp: $(UTILOBJ) outputscript.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) -DSWFPHP $(CFLAGS) $(LDFLAGS) outputscript.c $(UTILOBJ) -o swftophp
+
+swftopython: $(UTILOBJ) outputscript.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) -DSWFPYTHON $(CFLAGS) $(LDFLAGS) outputscript.c $(UTILOBJ) -o swftopython
+
+parser.o: parser.c
+	$(CC) $(CFLAGS) -c parser.c
+
+decompile.o: decompile.c
+	$(CC) $(CFLAGS) -c decompile.c -o decompile.o
+
+ttftofft: ttftofft.c outputswf.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) `freetype-config --cflags` $(CFLAGS) -DNODECOMPILE $< outputswf.c `freetype-config --libs` $(LDFLAGS) -o $@
+
+makefdb: blocktypes.o makefdb.c read.o ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) makefdb.c read.o -o makefdb
+
+gif2mask: gif2mask.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) gif2mask.c -o gif2mask 
+
+png2dbl: png2dbl.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) png2dbl.c -o png2dbl
+
+dbl2png: dbl2png.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) dbl2png.c -o dbl2png
+
+gif2dbl: gif2dbl.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) gif2dbl.c -o gif2dbl 
+
+raw2adpcm: raw2adpcm.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) raw2adpcm.c -o raw2adpcm 
+
+png2swf: png2swf.c ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) png2swf.c -o png2swf 
+
+makeswf: makeswf.c makeswf.h ../libming$(NOVAR_SHLIBEXT)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< 
+
+install: $(BINARIES)
+	install -d $(bindir)
+	install -m 0755 $(BINARIES) $(bindir)
+	install -m 0755 ming-config $(bindir)
