diff -Nuar --exclude '*~' ming-0.2a.orig/py_ext/Makefile ming-0.2a/py_ext/Makefile
--- ming-0.2a.orig/py_ext/Makefile	2006-06-11 03:55:18.102054000 -0700
+++ ming-0.2a/py_ext/Makefile	2006-06-11 03:59:15.500918176 -0700
@@ -7,7 +7,7 @@
 all:
 
 mingcmodule.so: ming_wrap.o
-	$(CC) $(CFLAGS) -shared -o mingcmodule.so ming_wrap.o -lming
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -o $@ $^ -L.. -lming 
 
 mingc.pyd: ming_wrap.o
 	dllwrap --dllname mingc.pyd --driver-name gcc --def mingc.def \
diff -Nuar --exclude '*~' ming-0.2a.orig/src/Makefile ming-0.2a/src/Makefile
--- ming-0.2a.orig/src/Makefile	2006-06-11 03:55:18.076055000 -0700
+++ ming-0.2a/src/Makefile	2006-06-11 04:04:20.289175057 -0700
@@ -24,7 +24,7 @@
 dynamic: ${OFILES}
 	cd blocks && make
 	cd actioncompiler && make
-	${CC} ${CFLAGS} ${OFILES} ${COMPILEROFILES} blocks/*.o -shared \
+	${CC} ${CFLAGS} ${LDFLAGS} ${OFILES} ${COMPILEROFILES} blocks/*.o -shared \
 	  -o libming.so
 	cp libming.so ..
 
@@ -45,4 +45,4 @@
 #	${CC} ${FLAGS} -c $<
 
 test: static test.o
-	${CC} -o test test.o libming.a -lm
+	${CC} ${LDFLAGS} -o test test.o libming.a -lm
