--- desktop
+++ desktop
@@ -3,7 +3,7 @@
 Categories=Game
 Type=Application
 Terminal=false
-Exec=/usr/local/bin/galaxy
-Icon=/usr/local/share/galaxy/icons/galaxy.png
+Exec=/usr/bin/galaxy
+Icon=/usr/share/galaxy/icons/galaxy.png
 GenericName=Stellar Simulation
 Comment=Stars steered by gravity, galaxy evolution
--- Makefile
+++ Makefile
@@ -8,38 +8,39 @@
 # replace -O3 with -g for GDB debugging
 CXXFLAGS ?= -O3 -Wall 
 LDFLAGS ?= -O3 -Wall -rdynamic
-PREFIX ?= /usr/local
+PREFIX = /usr
 
-CFLAGS = $(CXXFLAGS) -c `pkg-config --cflags gtk+-2.0`
-LFLAGS = $(LDFLAGS) `pkg-config --libs gtk+-2.0 gthread-2.0`
+GTK_CFLAGS = `pkg-config --cflags gtk+-2.0`
+GTK_LIBS   = `pkg-config --libs gtk+-2.0 gthread-2.0`
 
 BINDIR = $(PREFIX)/bin
 DATADIR = $(PREFIX)/share/$(PROGRAM)
 ICONDIR = $(DATADIR)/icons
-DESKTOP = $(DATADIR)/kornelix-$(PROGRAM).desktop
-DOCDIR = $(PREFIX)/share/doc/$(PROGRAM)
+DESKDIR = $(PREFIX)/share/applications
+DESKTOP = $(DESKDIR)/$(PROGRAM).desktop
+DOCDIR = $(PREFIX)/share/doc/$(PROGRAM)-$(VERSION)
 
 $(PROGRAM): $(PROGRAM).o zfuncs.o
-	$(CXX) $(LFLAGS) -o $(PROGRAM) $(PROGRAM).o zfuncs.o
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o $(PROGRAM) $(PROGRAM).o zfuncs.o $(GTK_LIBS)
 
 $(PROGRAM).o: $(SOURCE)
-	$(CXX) $(CFLAGS) -o $(PROGRAM).o $(SOURCE)		\
+	$(CXX) $(CXXFLAGS) $(GTK_CFLAGS) -o $(PROGRAM).o -c $(SOURCE) \
 	  -D "DATADIR=\"$(DATADIR)\"" -D "DOCDIR=\"$(DOCDIR)\""
 
 zfuncs.o: zfuncs.cc
-	$(CXX) $(CFLAGS) zfuncs.cc	\
+	$(CXX) $(CXXFLAGS) $(GTK_CFLAGS) -c zfuncs.cc \
 	  -D "DATADIR=\"$(DATADIR)\"" -D "DOCDIR=\"$(DOCDIR)\"" -D "BINDIR=\"$(BINDIR)\""
 
 install: $(PROGRAM)
 	mkdir -p $(DESTDIR)$(DOCDIR) 
 	mkdir -p $(DESTDIR)$(BINDIR)
-	mkdir -p $(DESTDIR)$(DATADIR)
 	mkdir -p $(DESTDIR)$(ICONDIR) 
-	cp -R doc/* $(DESTDIR)$(DOCDIR)
+	mkdir -p $(DESTDIR)$(DESKDIR)
+	cp -R doc/{images,userguide-en.html} $(DESTDIR)$(DOCDIR)
 	cp -R data/* $(DESTDIR)$(DATADIR)
 	cp icons/* $(DESTDIR)$(ICONDIR)
+	cp -f desktop $(DESTDIR)$(DESKTOP)
 	cp $(PROGRAM) $(DESTDIR)$(BINDIR)
-	@ echo use make menu to install desktop menu
 
 menu:
 	cp -f desktop $(DESKTOP)
