--- reallyslick/cpp_src/hyperspace.cpp	2005-04-24 01:56:24.000000000 +0200
+++ reallyslick/cpp_src/hyperspace.cpp.new	2005-04-24 01:56:12.000000000 +0200
@@ -26,7 +26,7 @@
 
 
 #include "config.h"
-
+#include <cassert>
 
 #include <stdio.h>
 #include <math.h>
--- reallyslick/cpp_src/skyrocket.cpp	2005-04-10 01:00:43.000000000 +0200
+++ reallyslick/cpp_src/skyrocket.cpp.new	2005-04-24 01:26:34.000000000 +0200
@@ -22,7 +22,7 @@
 #include <stdio.h>
 #include <GL/gl.h>
 #include <GL/glu.h>
-
+#include <cassert>
 #include <list>
 
 #include "driver.h"
