diff -NrU5 zphoto-1.2.original/image.cpp zphoto-1.2/image.cpp
--- zphoto-1.2.original/image.cpp	2009-07-20 17:13:30.000000000 +0000
+++ zphoto-1.2/image.cpp	2009-07-20 17:18:38.000000000 +0000
@@ -552,12 +552,12 @@
 }
 
 static int
 convert_needed_p (const char *src, const char *dest)
 {
-    char *suffix1 = strrchr(src,  '.');
-    char *suffix2 = strrchr(dest, '.');
+    char *suffix1 = const_cast<char*>(strrchr(src,  '.'));
+    char *suffix2 = const_cast<char*>(strrchr(dest, '.'));
 
     return strcmp(suffix1, suffix2) != 0;
 }
 
 extern "C" void
