--- Makefile	2008-02-05 17:02:37.000000000 +1300
+++ Makefile.install	2008-02-05 17:03:27.000000000 +1300
@@ -1,6 +1,7 @@
 program := beanstalkd
 export CFLAGS := $(LDFLAGS) -Wall -Werror
 export LDFLAGS := $(LDFLAGS) -levent
+export BINDIR := $(DESTDIR)/usr/bin
 
 sources := $(shell ls *.c | fgrep -v $(program))
 objects := $(sources:.c=.o)
@@ -9,6 +10,10 @@
 all: export CFLAGS := $(CFLAGS) -O2
 all: $(program)
 
+install: all
+	mkdir -p "$(BINDIR)"
+	install $(program) "$(BINDIR)"
+
 debug: export CFLAGS := $(CFLAGS) -g -pg -DDEBUG
 debug: export LDFLAGS := $(LDFLAGS) -pg
 debug: $(program)
