diff -urN ori/dvmysql/mysqldb.C dvmysql-0.4.7/dvmysql/mysqldb.C
--- ori/dvmysql/mysqldb.C	2003-03-13 23:52:13.000000000 +0100
+++ dvmysql-0.4.7/dvmysql/mysqldb.C	2003-03-13 23:52:32.000000000 +0100
@@ -262,7 +262,7 @@
 Command_::check_error() {
 static const char*	OK = "ok";
 
-char*	errmsg = mysql_error(static_cast<MYSQL*>(db_.mysql_));
+const char*	errmsg = mysql_error(static_cast<MYSQL*>(db_.mysql_));
 
 if (errmsg&&(*errmsg!=0)) {
   ok(false);
