--- src/adesklets.c.old	2006-02-14 01:49:01.000000000 -0500
+++ src/adesklets.c	2006-02-14 01:49:54.000000000 -0500
@@ -2910,6 +2910,7 @@
 #ifdef CONTROL_ON_CONTEXT_MENU
 	      } else {
+		XUngrabPointer(adesklets.display, ev.xbutton.time);
 		ev.xbutton.window = adesklets.root ;
 		XSendEvent(adesklets.display, ev.xbutton.window, 
 			   False, ButtonPressMask, &ev);
 		debug("menu call redirected to root\n");
