diff -Naur ipp2p-0.7.orig/Makefile ipp2p-0.7/Makefile
--- ipp2p-0.7.orig/Makefile	2004-12-23 03:52:39.000000000 -0800
+++ ipp2p-0.7/Makefile	2004-12-26 11:50:11.064315453 -0800
@@ -15,16 +15,11 @@
 SED = sed
 IPTABLES = /sbin/iptables
 IPTABLES_VERSION = \
-	$(shell iptables --version | sed -e 's/^iptables v//')
+	$(shell $(IPTABLES) --version | sed -e 's/^iptables v//')
 IPTABLES_OPTION = -DIPTABLES_VERSION=\"$(IPTABLES_VERSION)\"
 
 
 KERNEL_SRC = /lib/modules/$(shell uname -r)/build
-IUSER = -I/usr/src/iptables-1.2.9/include
-CC = gcc
-CFLAGS = -O3 -Wall
-
-
 
 all: modules libipt_ipp2p.so
 
@@ -34,11 +29,11 @@
 	$(CC) $(CFLAGS) -I$(KERNEL_SRC)/include -c ipt_ipp2p.c -D__KERNEL__ -DMODULE $(MODVERSIONS)
 
 ipt_ipp2p.ko:
-	$(MAKE) -C $(KERNEL_SRC) SUBDIRS=$(PWD) modules
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) modules
 
 
 libipt_ipp2p.so:   libipt_ipp2p.c ipt_ipp2p.h
-		$(CC) $(CFLAGS) $(IPTABLES_OPTION) $(IUSER) -fPIC -c libipt_ipp2p.c
+		$(CC) $(CFLAGS) $(IPTABLES_OPTION) -fPIC -c libipt_ipp2p.c
 		ld -shared -o libipt_ipp2p.so libipt_ipp2p.o
 
 clean:
