diff -NurB gok-0.9.3-orig/gok/gok-spy.c gok-0.9.3/gok/gok-spy.c
--- gok-0.9.3-orig/gok/gok-spy.c	2004-02-11 12:19:33.000000000 -0500
+++ gok-0.9.3/gok/gok-spy.c	2004-02-11 12:21:51.000000000 -0500
@@ -1246,8 +1246,9 @@
 				if ( len != 0 )  {
 					if (len > 18) 
 					{
+						gchar *tmp;
 						gok_log("shortening description");
-						gchar *tmp = g_strndup (s, 15);
+						tmp = g_strndup (s, 15);
 						pName = g_strconcat (tmp, "...", NULL);
 						g_free (tmp);
 					}
@@ -1273,7 +1274,6 @@
 		}
 		*/
 		else if (Accessible_isText (pAccessible)) {
-			gok_log("still no name, but there is text...");
 			/* 
 			* sadly, many text objects are still nameless - yet we want 
 			* to allow the user to get to them.
@@ -1284,6 +1284,8 @@
 			AccessibleText *text = Accessible_getText (pAccessible);
 			gint j = 0, len;
 			gchar *word, *gs = NULL;
+
+			gok_log("still no name, but there is text...");
 			if (text) {
 				long int start, end = 0;
 				do {
