diff -urN xen-3.0.2/tools/pygrub/src/fsys/reiser/reisermodule.c xen-3.0.2-b/tools/pygrub/src/fsys/reiser/reisermodule.c
--- xen-3.0.2/tools/pygrub/src/fsys/reiser/reisermodule.c	2006-04-09 18:05:53.000000000 -0400
+++ xen-3.0.2-b/tools/pygrub/src/fsys/reiser/reisermodule.c	2006-08-18 12:51:42.000000000 -0400
@@ -17,7 +17,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-#include <dal/file_dal.h>
+#include <dal/file.h>
 #include <reiserfs/reiserfs.h>
 
 #if (PYTHON_API_VERSION >= 1011)
@@ -46,8 +46,7 @@
 
 	if (!dal) return;
 
-	close((int)(unsigned long)dal->dev);
-	dal_free(dal);
+	dal_close(dal);
 }
 
 /* reiser file object */
@@ -195,7 +194,7 @@
     if (fs->fs != NULL)
     {
         reiserfs_fs_close(fs->fs);
-        file_dal_close(fs->dal);
+        file_close(fs->dal);
 		fs->fs = NULL;
     }
     Py_INCREF(Py_None);
@@ -218,13 +217,13 @@
         return NULL;
     }
 
-    if (!(dal = file_dal_open(name, block_size, O_RDONLY))) {
+    if (!(dal = file_open(name, block_size, O_RDONLY))) {
         PyErr_SetString(PyExc_ValueError, "Couldn't create device abstraction");
         return NULL;    
     }
     
     if (!(rfs = reiserfs_fs_open_fast(dal, dal))) {
-		file_dal_close(dal);
+		file_close(dal);
         PyErr_SetString(PyExc_ValueError, "unable to open file");
         return NULL;
     }
@@ -265,7 +264,7 @@
     if (fs->fs != NULL)
 	{
         reiserfs_fs_close(fs->fs);
-		file_dal_close(fs->dal);
+		file_close(fs->dal);
 		fs->fs = NULL;
 	}
 	PyObject_DEL(fs);
