=== modified file 'setup.py'
--- setup.py	2006-09-04 14:46:59 +0000
+++ setup.py	2006-09-04 14:47:28 +0000
@@ -207,7 +207,7 @@
 else:
     # std setup
     ARGS = {'scripts': ['bzr'],
-            'data_files': [('man/man1', ['bzr.1'])],
+            'data_files': [('share/man/man1', ['bzr.1'])],
             'cmdclass': {'build': bzr_build,
                          'install_scripts': my_install_scripts,
                         },

