--- programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c	2004/05/21 19:54:16	1.1
+++ programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c	2004/05/21 19:58:29
@@ -99,6 +99,14 @@ GetKbdLeds(InputInfoPtr pInfo)
 #include <asm/kbio.h>
 #endif
 
+/* Deal with spurious kernel header change */
+#include <linux/version.h>
+#if defined(LINUX_VERSION_CODE) && defined(KERNEL_VERSION)
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,42)
+#  define rate period
+# endif
+#endif
+
 static int
 KDKBDREP_ioctl_ok(int rate, int delay) {
 #if defined(KDKBDREP) && !defined(__sparc__)
@@ -160,6 +168,14 @@ KIOCSRATE_ioctl_ok(int rate, int delay) 
 #endif /* KIOCSRATE */
 }
 
+/* Deal with spurious kernel header change */
+#include <linux/version.h>
+#if defined(LINUX_VERSION_CODE) && defined(KERNEL_VERSION)
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,42)
+#  undef rate
+# endif
+#endif
+
 static void
 SetKbdRepeat(InputInfoPtr pInfo, char rad)
 {
--- programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c	2004/05/21 19:53:03	1.1
+++ programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c	2004/05/21 19:55:24
@@ -73,6 +73,7 @@ xf86GetKbdLeds()
 #endif
 
 /* Deal with spurious kernel header change */
+#include <linux/version.h>
 #if defined(LINUX_VERSION_CODE) && defined(KERNEL_VERSION)
 # if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,42)
 #  define rate period

