--- vnc-3.3.3r2.orig/Xvnc/programs/Xserver/include/servermd.h	Mon Jul 28 10:17:34 1997
+++ vnc-3.3.3r2/Xvnc/programs/Xserver/include/servermd.h	Tue Dec 12 13:40:57 2000
@@ -380,6 +380,27 @@
 #define GETLEFTBITS_ALIGNMENT  1
 
 #endif /* linux/m68k */
+
+#if defined (linux) && defined (PPC)
+
+#define IMAGE_BYTE_ORDER       MSBFirst
+#define BITMAP_BIT_ORDER       MSBFirst
+#define GLYPHPADBYTES          4
+#define GETLEFTBITS_ALIGNMENT  1
+
+#define LARGE_INSTRUCTION_CACHE
+#define FAST_CONSTANT_OFFSET_MODE
+
+#endif /* linux/PPC */
+
+#if defined (linux) && defined (__sparc__)
+
+#define IMAGE_BYTE_ORDER       MSBFirst
+#define BITMAP_BIT_ORDER       MSBFirst
+#define GLYPHPADBYTES          4
+#define GETLEFTBITS_ALIGNMENT  1
+
+#endif /* linux/__sparc__ */
 
 #ifdef sgi
 
