--- capplets/common/gconf-property-editor.c.old	2002-07-16 18:07:40.000000000 -0400
+++ capplets/common/gconf-property-editor.c	2003-02-18 16:55:14.000000000 -0500
@@ -101,6 +101,7 @@
 					       GConfChangeSet        *changeset,
 					       GObject               *ui_control,
 					       const gchar           *first_prop_name,
+					       int                    alpha_hack,
 					       va_list                var_args,
 					       const gchar	     *first_custom,
 					       ...);
@@ -341,6 +342,7 @@
 		   GConfChangeSet        *changeset,
 		   GObject               *ui_control,
 		   const gchar           *first_prop_name,
+		   int                    alpha_hack,
 		   va_list                var_args,
 		   const gchar		 *first_custom,
 		   ...) 
@@ -452,7 +454,7 @@
 		 changeset,
 		 G_OBJECT (checkbox),
 		 first_property_name,
-		 var_args,
+		 0, var_args, 
 		 NULL);
 
 	va_end (var_args);
@@ -526,7 +528,7 @@
 		 changeset,
 		 G_OBJECT (entry),
 		 first_property_name,
-		 var_args, NULL);
+		 0, var_args,  NULL);
 
 	g_signal_connect_swapped (G_OBJECT (entry), "changed",
 				  (GCallback) peditor_integer_widget_changed, peditor);
@@ -618,7 +620,7 @@
 		 changeset,
 		 G_OBJECT (entry),
 		 first_property_name,
-		 var_args, NULL);
+		 0, var_args,  NULL);
 
 	g_signal_connect_swapped (G_OBJECT (entry), "changed",
 				  (GCallback) peditor_string_widget_changed, peditor);
@@ -748,7 +750,7 @@
 		 changeset,
 		 G_OBJECT (cp),
 		 first_property_name,
-		 var_args, NULL);
+		 0, var_args,  NULL);
 
 	va_end (var_args);
 
@@ -902,7 +904,7 @@
 		 changeset,
 		 G_OBJECT (option_menu),
 		 first_property_name,
-		 var_args, NULL);
+		 0, var_args,  NULL);
 
 	va_end (var_args);
 
@@ -943,7 +945,7 @@
 		 changeset,
 		 G_OBJECT (option_menu),
 		 first_property_name,
-		 var_args,
+		 0, var_args, 
 		 "conv-to-widget-cb",
 		 peditor_enum_conv_to_widget,
 		 "conv-from-widget-cb",
@@ -1041,7 +1043,7 @@
 		 changeset,
 		 G_OBJECT (first_button),
 		 first_property_name,
-		 var_args, NULL);
+		 0, var_args,  NULL);
 
 	va_end (var_args);
 
@@ -1116,7 +1118,7 @@
 		 changeset,
 		 G_OBJECT (adjustment),
 		 first_property_name,
-		 var_args, NULL);
+		 0, var_args,  NULL);
 
 	va_end (var_args);
 
@@ -1351,7 +1353,7 @@
 		 changeset,
 		 G_OBJECT (font_picker),
 		 first_property_name,
-		 var_args,
+		 0, var_args, 
 		 "data", GUINT_TO_POINTER (font_type),
 		 NULL);
 
@@ -1445,7 +1447,7 @@
 		 changeset,
 		 G_OBJECT (checkbox),
 		 first_property_name,
-		 var_args,
+		 0, var_args, 
 		 "conv-to-widget-cb",
 		 peditor_enum_toggle_conv_to_widget,
 		 "conv-from-widget-cb",
@@ -1670,7 +1672,7 @@
 		 changeset,
 		 G_OBJECT (button),
 		 first_property_name,
-		 var_args, NULL);
+		 0, var_args,  NULL);
 
 	va_end (var_args);
 
@@ -1715,7 +1717,7 @@
 		 changeset,
 		 G_OBJECT (first_button),
 		 first_property_name,
-		 var_args,
+		 0, var_args,
 		 "conv-to-widget-cb",
 		 peditor_enum_conv_to_widget,
 		 "conv-from-widget-cb",
