diff -ur control-center-2.4.0.orig/capplets/font/main.c control-center-2.4.0/capplets/font/main.c
--- control-center-2.4.0.orig/capplets/font/main.c	2004-01-12 20:03:32.860516011 +0100
+++ control-center-2.4.0/capplets/font/main.c	2004-01-12 20:06:09.399260795 +0100
@@ -85,7 +85,7 @@
 
 static void
 sample_expose (GtkWidget      *darea,
-	       GdkEventExpose  expose)
+               GdkEventExpose *expose)
 {
 	GdkPixbuf *pixbuf = g_object_get_data (G_OBJECT (darea), "sample-pixbuf");
 	int width = gdk_pixbuf_get_width (pixbuf);
diff -ur control-center-2.4.0.orig/gnome-settings-daemon/gnome-settings-gtk1theme.c control-center-2.4.0/gnome-settings-daemon/gnome-settings-gtk1theme.c
--- control-center-2.4.0.orig/gnome-settings-daemon/gnome-settings-gtk1theme.c	2004-01-12 20:03:17.865618679 +0100
+++ control-center-2.4.0/gnome-settings-daemon/gnome-settings-gtk1theme.c	2004-01-12 20:04:36.964003364 +0100
@@ -151,7 +151,7 @@
 	gchar *theme_filename;
 	gchar *rc_filename;
 	gchar *current_contents;
-	gint current_length;
+	gsize current_length;
 	gchar *new_contents;
 	GError *err = NULL;
 
diff -ur control-center-2.4.0.orig/vfs-methods/themus/theme-method.c control-center-2.4.0/vfs-methods/themus/theme-method.c
--- control-center-2.4.0.orig/vfs-methods/themus/theme-method.c	2004-01-12 20:03:19.535607241 +0100
+++ control-center-2.4.0/vfs-methods/themus/theme-method.c	2004-01-12 20:06:54.772898040 +0100
@@ -21,7 +21,7 @@
 #include <errno.h>
 #include <sys/stat.h>
 #include <sys/types.h>
-
+#include <libintl.h>
 #include <libgnomevfs/gnome-vfs.h>
 #include <libgnomevfs/gnome-vfs-cancellable-ops.h>
 #include <libgnomevfs/gnome-vfs-module.h>
