diff -ur control-center-2.2.0/capplets/common/gnome-theme-test.c control-center-2.2.0.patched/capplets/common/gnome-theme-test.c
--- control-center-2.2.0/capplets/common/gnome-theme-test.c	2003-01-19 08:14:34.000000000 +0100
+++ control-center-2.2.0.patched/capplets/common/gnome-theme-test.c	2003-02-02 00:32:33.000000000 +0100
@@ -10,11 +10,12 @@
 int
 main (int argc, char *argv[])
 {
-  gtk_init (&argc, &argv);
-  gnome_vfs_init ();
   gboolean monitor_not_added = FALSE;
   GList *themes, *list;
 
+  gtk_init (&argc, &argv);
+  gnome_vfs_init ();
+
   gnome_theme_init (&monitor_not_added);
   
   themes = gnome_theme_meta_info_find_all ();
