diff -NurdB control-center-2.11.92/capplets/mouse/gnome-mouse-properties.c control-center-2.11.92-patched/capplets/mouse/gnome-mouse-properties.c
--- control-center-2.11.92/capplets/mouse/gnome-mouse-properties.c	2005-08-01 10:59:43.000000000 -0500
+++ control-center-2.11.92-patched/capplets/mouse/gnome-mouse-properties.c	2005-08-24 22:17:28.000000000 -0500
@@ -690,7 +690,7 @@
 			
 			for (i = 0; sizes[i] != 0; i++) {
 				cursor = XcursorLibraryLoadImage("left_ptr", name, sizes[i]);
-				if (cursor->size != sizes[i]) {
+				if (cursor != NULL && cursor->size != sizes[i]) {
 					XcursorImageDestroy (cursor);
 					cursor = NULL;
 				}
