--- configure.in	2004-03-14 02:12:35.406882224 +0100
+++ configure.in.patched	2004-03-14 02:12:45.294379096 +0100
@@ -329,7 +329,7 @@
 
 if test "x$ENABLE_GSTREAMER" = "xyes"; then
    have_gstreamer=no
-   PKG_CHECK_MODULES(GST, gstreamer-interfaces-0.7 >= 0.7.3.1,
+   PKG_CHECK_MODULES(GST, gstreamer-interfaces-0.8 >= 0.7.3.1,
 		   have_gstreamer=yes,
 		   AC_MSG_RESULT(*** All of GStreamer dependent parts will be disabled ***))
 else
