--- control-center/capplet-manager.c.orig	Thu Jan 24 21:57:56 2002
+++ control-center/capplet-manager.c	Thu Jan 24 21:58:01 2002
@@ -323,7 +323,10 @@
                 close_capplet (TRUE, nd);
                 nd->state = CAPPLET_INACTIVE;
                 style = gtk_widget_get_style (GTK_WIDGET (nd->ctree));
-                gtk_ctree_node_set_foreground (nd->ctree, nd->node, &style->fg[GTK_STATE_NORMAL]);
+
+                if (style) {
+                        gtk_ctree_node_set_foreground (nd->ctree, nd->node, &style->fg[GTK_STATE_NORMAL]);
+                }
         }
 }
 static void
