--- xvjpeg.c.orig	2005-01-04 13:49:48.074362760 +0000
+++ xvjpeg.c	2005-01-04 13:51:04.286776712 +0000
@@ -761,6 +761,7 @@
   jpeg_set_defaults(&cinfo);
   jpeg_set_quality(&cinfo, (int)qDial.val, TRUE);
   cinfo.smoothing_factor = (int)smDial.val;
+  cinfo.optimize_coding = TRUE;
 
 
   jpeg_start_compress(&cinfo, TRUE);
--- jpeg/jconfig.cfg.orig	1994-12-22 22:35:00.000000000 +0000
+++ jpeg/jconfig.cfg	2005-01-04 13:51:04.285776864 +0000
@@ -42,3 +42,5 @@
 #undef PROGRESS_REPORT
 
 #endif /* JPEG_CJPEG_DJPEG */
+
+#define JDCT_DEFAULT JDCT_FLOAT
