--- xv-3.10a/xvps.c	Thu Dec 22 14:12:17 1994
+++ xv-3.10a/xvps.c	Sun Nov 30 22:55:21 1997
@@ -891,7 +891,7 @@
   int   i, j, q, err, rpix, gpix, bpix, nc, ptype;
   int   iw, ih, ox, oy, slen, lwidth, bits, colorps, w, h, pfree;
   double iwf, ihf;
-  byte *inpix, *rmap, *gmap, *bmap;
+  byte *save_inpix, *inpix, *rmap, *gmap, *bmap;
 
   slen = bits = colorps = 0;
 
@@ -901,7 +901,8 @@
 
   WaitCursor();
   
-  inpix = GenSavePic(&ptype, &w, &h, &pfree, &nc, &rmap, &gmap, &bmap);
+  save_inpix = inpix =
+	GenSavePic(&ptype, &w, &h, &pfree, &nc, &rmap, &gmap, &bmap);
 
     
   /* printed image will have size iw,ih (in picas) */
@@ -1123,7 +1124,7 @@
   fprintf(fp,"origstate restore\n\n");
   fprintf(fp,"%%%%Trailer\n");
 
-  if (pfree) free(inpix);
+  if (pfree) free(save_inpix);
 
   if (CloseOutFile(fp, filename, (err==EOF)) == 0) {
     DirBox(0);
