--- source/Makefile	Wed Aug  8 15:52:11 2001
+++ source/Makefile	Wed Aug  8 15:53:59 2001
@@ -1003,10 +1003,9 @@
 	else (set -x; $(MKDIRHIER) $(DESTDIR)$(GV_BINDIR)); fi
 	$(INSTALL) $(INSTALLFLAGS) $(INSTPGMFLAGS)  gv $(DESTDIR)$(GV_BINDIR)/gv
 
-install.man:: gv.man
+install.man:: 
 	@if [ -d $(DESTDIR)$(MANDIR) ]; then set +x; \
 	else (set -x; $(MKDIRHIER) $(DESTDIR)$(MANDIR)); fi
-	$(INSTALL) $(INSTALLFLAGS) $(INSTMANFLAGS) gv.man $(DESTDIR)$(MANDIR)/gv.$(MANSUFFIX)
 
 depend::
 	$(DEPEND) $(DEPENDFLAGS) -- $(ALLDEFINES) $(DEPEND_DEFINES) -- $(SRCS)
