diff -ur kdesdk-3.0.6/doc/cervisia/index.docbook kdesdk/doc/cervisia/index.docbook
--- doc/cervisia/index.docbook	Mon Jul  1 21:59:41 2002
+++ doc/cervisia/index.docbook	Thu Jul  4 14:36:03 2002
@@ -1466,9 +1466,9 @@
 <screen><command>cvs admin -l <replaceable>filenames</replaceable></command></screen>
 </para>
 
-The reverse effect is achieved by using
+<para>The reverse effect is achieved by using
 <menuchoice><guimenu>Advanced</guimenu><guimenuitem>Unlock</guimenuitem></menuchoice>.
-This runs the command
+This runs the command</para>
 
 <para>
 <screen><command>cvs admin -u <replaceable>filenames</replaceable></command></screen>
