--- phpwebsite-0.10.0-full/lib/pear/XML/RPC.php.old	2004-09-21 21:44:25.000000000 +0200
+++ phpwebsite-0.10.0-full/lib/pear/XML/RPC.php	2005-07-04 22:28:08.000000000 +0200
@@ -158,7 +158,7 @@
         break;
 
     case "NAME":
-        $XML_RPC_xh[$parser]['st'] .= "'";
+        $XML_RPC_xh[$parser]['st'] .= '"';
         $XML_RPC_xh[$parser]['ac'] = "";
         break;
 
@@ -238,7 +238,7 @@
         break;
 
     case "NAME":
-        $XML_RPC_xh[$parser]['st'] .= $XML_RPC_xh[$parser]['ac'] . "' => ";
+        $XML_RPC_xh[$parser]['st'] .= $XML_RPC_xh[$parser]['ac'] . '" => ';
         break;
 
     case "BOOLEAN":
@@ -263,7 +263,7 @@
             // we use double quotes rather than single so backslashification works OK
             $XML_RPC_xh[$parser]['st'] .= "\"" . $XML_RPC_xh[$parser]['ac'] . "\"";
         } elseif ($XML_RPC_xh[$parser]['qt'] == 2) {
-            $XML_RPC_xh[$parser]['st'] .= "base64_decode('" . $XML_RPC_xh[$parser]['ac'] . "')";
+	    $XML_RPC_xh[$parser]['st'] .= 'base64_decode("' . $XML_RPC_xh[$parser]['ac'] . '")';
         } elseif ($name=="BOOLEAN") {
             $XML_RPC_xh[$parser]['st'] .= $XML_RPC_xh[$parser]['ac'];
         } else {
