--- gb.sdl/configure.ac.orig	2008-04-16
+++ gb.sdl/configure.ac	2008-04-16
@@ -7,7 +7,7 @@
 GB_CHECK_XWINDOW
 
 dnl TODO: add a version test in GB_COMPONENT!
-GB_PATH_SDL(1.2.8, [], [touch DISABLED])
+dnl GB_PATH_SDL(1.2.8, [], [touch DISABLED])
 
 dnl check Xcursor support
 AC_CHECK_LIB(Xcursor, XcursorLibraryLoadCursor,
--- gb.sdl.sound/configure.ac.orig	2008-04-16
+++ gb.sdl.sound/configure.ac	2008-04-16
@@ -5,7 +5,7 @@
 AC_PROG_LIBTOOL
 
 dnl TODO: add a version test in GB_COMPONENT!
-GB_PATH_SDL(1.2.8, [] , [touch DISABLED])
+dnl GB_PATH_SDL(1.2.8, [] , [touch DISABLED])
 
 dnl check for SDL libs
 GB_COMPONENT(
