--- app/Makefile.am.orig	2008-09-01
+++ app/Makefile.am	2008-09-01
@@ -4,17 +4,35 @@
 	@echo "Installing the development environment..."
 	@(cd $(srcdir)/src; d=`pwd`; \
 	  for p in `cat INSTALL`; do \
-	    echo "Compiling $$p..."; cd $$d/$$p; \
-	    $(DESTDIR)$(bindir)/gbc$(GAMBAS_VERSION) -ag -r $(DESTDIR)$(prefix); \
+	    echo "Compiling $$p..."; \
+	    cd $$d/$$p; \
+	    { \
+	    $(DESTDIR)$(bindir)/gbc$(GAMBAS_VERSION) -ag -r $(DESTDIR)$(prefix) && \
 	    $(DESTDIR)$(bindir)/gba$(GAMBAS_VERSION); \
+	    } && { \
 	    rm -rf .gambas; \
+	    echo "Installing $$p..."; \
 	    $(INSTALL) $$p.gambas $(DESTDIR)$(bindir); \
+	    } || echo "Failed"; \
 	  done)
 	##@if test "$(bindir)" != "$(ROOT)/usr/bin" && test "$(bindir)" != "$(ROOT)/usr/bin/"; then
-	@ln -s gambas$(GAMBAS_VERSION).gambas $(DESTDIR)$(bindir)/gambas$(GAMBAS_VERSION) || true
+	@[[ -e $(DESTDIR)$(bindir)/gambas$(GAMBAS_VERSION).gambas ]] && \
+	{ \
+	echo "Installing the IDE..."; \
+	ln -s gambas$(GAMBAS_VERSION).gambas $(DESTDIR)$(bindir)/gambas$(GAMBAS_VERSION); \
+	} || true
 	
-	@echo "Installing the scripter..."
-	@ln -s gbs$(GAMBAS_VERSION).gambas $(DESTDIR)$(bindir)/gbs$(GAMBAS_VERSION) || true
+	@[[ -e $(DESTDIR)$(bindir)/gambas$(GAMBAS_VERSION)-database-manager.gambas ]] && \
+	{ \
+	echo "Installing the Database Manager..."; \
+	ln -s gambas$(GAMBAS_VERSION)-database-manager.gambas $(DESTDIR)$(bindir)/gambas$(GAMBAS_VERSION)-database-manager; \
+	} || true
+	
+	@[[ -e $(DESTDIR)$(bindir)/gbs$(GAMBAS_VERSION).gambas ]] && \
+	{ \
+	echo "Installing the scripter..."; \
+	ln -s gbs$(GAMBAS_VERSION).gambas $(DESTDIR)$(bindir)/gbs$(GAMBAS_VERSION); \
+	} || true
 
 uninstall-local:
 	@rm -f $(DESTDIR)$(bindir)/gambas$(GAMBAS_VERSION)
