diff -Naur octave-forge-2006.03.17/extra/graceplot/__grcmd__.cc octave-forge-2006.03.17.new/extra/graceplot/__grcmd__.cc
--- octave-forge-2006.03.17/extra/graceplot/__grcmd__.cc	2005-05-24 23:43:38.000000000 -0400
+++ octave-forge-2006.03.17.new/extra/graceplot/__grcmd__.cc	2008-04-27 10:43:59.000000000 -0400
@@ -36,6 +36,7 @@
 #include <errno.h>
 #include <limits.h>
 #include <string>
+#include <cstring>
 #include <map>
 #ifndef OPEN_MAX
 #  define OPEN_MAX 256
diff -Naur octave-forge-2006.03.17/extra/mex/mex.cc octave-forge-2006.03.17.new/extra/mex/mex.cc
--- octave-forge-2006.03.17/extra/mex/mex.cc	2004-07-08 00:56:03.000000000 -0400
+++ octave-forge-2006.03.17.new/extra/mex/mex.cc	2008-04-27 10:43:59.000000000 -0400
@@ -26,6 +26,7 @@
 #include <iomanip>
 #include <set>
 #include <string>
+#include <cstring>
 typedef void * Pix;
 typedef std::set<Pix> MemSet;
 
diff -Naur octave-forge-2006.03.17/extra/pdb/creadpdb.cc octave-forge-2006.03.17.new/extra/pdb/creadpdb.cc
--- octave-forge-2006.03.17/extra/pdb/creadpdb.cc	2006-01-08 00:47:57.000000000 -0500
+++ octave-forge-2006.03.17.new/extra/pdb/creadpdb.cc	2008-04-27 10:43:59.000000000 -0400
@@ -20,6 +20,7 @@
 #include <sys/mman.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <cstring>
 #include <fcntl.h>
 
 #define BUFLEN 80
diff -Naur octave-forge-2006.03.17/FIXES/oct-sort.cc octave-forge-2006.03.17.new/FIXES/oct-sort.cc
--- octave-forge-2006.03.17/FIXES/oct-sort.cc	2004-01-25 10:17:25.000000000 -0500
+++ octave-forge-2006.03.17.new/FIXES/oct-sort.cc	2008-04-27 10:43:59.000000000 -0400
@@ -85,6 +85,7 @@
 #include <octave/lo-mappers.h>
 #include <octave/config.h>
 #include <octave/quit.h>
+#include <cstring>
 #include "oct-sort.h"
 
 #define IFLT(a,b)  if (compare == NULL ? ((a) < (b)) : compare ((a), (b)))
diff -Naur octave-forge-2006.03.17/FIXES/sort.cc octave-forge-2006.03.17.new/FIXES/sort.cc
--- octave-forge-2006.03.17/FIXES/sort.cc	2004-04-28 04:55:14.000000000 -0400
+++ octave-forge-2006.03.17.new/FIXES/sort.cc	2008-04-27 10:43:59.000000000 -0400
@@ -27,6 +27,7 @@
 #include <octave/lo-ieee.h>
 #include <octave/data-conv.h>
 #include <octave/ov-cx-mat.h>
+#include <cstring>
 #include "oct-sort.cc"
 
 // ======= Cruft to support ancient versions of Octave =========
diff -Naur octave-forge-2006.03.17/main/comm/gf.cc octave-forge-2006.03.17.new/main/comm/gf.cc
--- octave-forge-2006.03.17/main/comm/gf.cc	2005-10-12 14:47:48.000000000 -0400
+++ octave-forge-2006.03.17.new/main/comm/gf.cc	2008-04-27 10:43:59.000000000 -0400
@@ -49,6 +49,7 @@
 #include "galois.h"
 #include "ov-galois.h"
 #include <octave/utils.h>
+#include <cstring>
 #include <octave/variables.h>
 
 #ifndef OCTAVE_LOCAL_BUFFER
diff -Naur octave-forge-2006.03.17/main/fixed/fixedComplex.cc octave-forge-2006.03.17.new/main/fixed/fixedComplex.cc
--- octave-forge-2006.03.17/main/fixed/fixedComplex.cc	2004-07-06 16:55:23.000000000 -0400
+++ octave-forge-2006.03.17.new/main/fixed/fixedComplex.cc	2008-04-27 10:43:59.000000000 -0400
@@ -25,6 +25,7 @@
 
 #include <complex>
 #include <cmath>
+#include <cstdlib>
 #include "int/fixed.h"
 #include "fixedComplex.h"
 
diff -Naur octave-forge-2006.03.17/main/miscellaneous/listen.cc octave-forge-2006.03.17.new/main/miscellaneous/listen.cc
--- octave-forge-2006.03.17/main/miscellaneous/listen.cc	2005-11-16 16:41:07.000000000 -0500
+++ octave-forge-2006.03.17.new/main/miscellaneous/listen.cc	2008-04-27 10:43:59.000000000 -0400
@@ -11,6 +11,7 @@
 #include <cstdlib>
 #include <unistd.h>
 #include <cerrno>
+#include <cstring>
 // #include <string.h>
 #include <sys/types.h>
 #include <sys/socket.h>
diff -Naur octave-forge-2006.03.17/main/miscellaneous/mfilename.cc octave-forge-2006.03.17.new/main/miscellaneous/mfilename.cc
--- octave-forge-2006.03.17/main/miscellaneous/mfilename.cc	2005-12-21 15:42:40.000000000 -0500
+++ octave-forge-2006.03.17.new/main/miscellaneous/mfilename.cc	2008-04-27 10:43:59.000000000 -0400
@@ -22,6 +22,7 @@
 //#ifdef HAVE_CONFIG_H
 #include <config.h>
 //#endif
+#include <algorithm>
 
 #include "defun-dld.h"
 #include "error.h"
diff -Naur octave-forge-2006.03.17/main/plot/graphics.cc octave-forge-2006.03.17.new/main/plot/graphics.cc
--- octave-forge-2006.03.17/main/plot/graphics.cc	2004-01-12 02:39:09.000000000 -0500
+++ octave-forge-2006.03.17.new/main/plot/graphics.cc	2008-04-27 10:43:59.000000000 -0400
@@ -11,6 +11,7 @@
  */
 
 #include <string>
+#include <cstring>
 
 #include <octave/oct.h>
 #include <octave/toplev.h>
