diff -ur mscore-0.9.4.orig/mscore/awl/CMakeLists.txt mscore-0.9.4/mscore/awl/CMakeLists.txt
--- mscore-0.9.4.orig/mscore/awl/CMakeLists.txt	2009-02-06 12:32:19.000000000 +0200
+++ mscore-0.9.4/mscore/awl/CMakeLists.txt	2009-05-10 16:34:42.000000000 +0300
@@ -83,6 +83,8 @@
       colorlabel.h
       )
 
+if (BUILD_AWL_PLUGIN)
+
 add_library (
       awlplugin SHARED
       ${PROJECT_BINARY_DIR}/all.h
@@ -107,4 +109,6 @@
 
 install( TARGETS awlplugin DESTINATION ${QT_PLUGINS_DIR}/designer/ )
 
+endif (BUILD_AWL_PLUGIN)
+
 endif (NOT MINGW AND NOT APPLE)
diff -ur mscore-0.9.4.orig/mscore/CMakeLists.txt mscore-0.9.4/mscore/CMakeLists.txt
--- mscore-0.9.4.orig/mscore/CMakeLists.txt	2009-02-06 12:37:31.000000000 +0200
+++ mscore-0.9.4/mscore/CMakeLists.txt	2009-05-10 16:34:42.000000000 +0300
@@ -28,8 +28,10 @@
 
 set(USE_GLOBAL_FLUID FALSE)               # use internal fluid lib
 
-set(BUILD_SCRIPT_INTERFACE TRUE)          # this requires qt >= 4.4 && cmake 2.6
-set(BUILD_SCRIPT_DEBUG TRUE)
+set(BUILD_AWL_PLUGIN FALSE)
+
+set(BUILD_SCRIPT_INTERFACE FALSE)          # this requires qt >= 4.4 && cmake 2.6
+set(BUILD_SCRIPT_DEBUG FALSE)
 set(STATIC_SCRIPT_BINDINGS FALSE)         # this requires qt 4.5
 
 if (MINGW)
