--- openais-0.80.3/Makefile.inc.orig	2008-02-17 18:39:07.000000000 +0100
+++ openais-0.80.3/Makefile.inc	2008-02-17 18:39:36.000000000 +0100
@@ -61,14 +61,14 @@
 
 # default CFLAGS, LDFLAGS
 #
-CFLAGS = 
-LDFLAGS =
-DYFLAGS =
+CFLAGS ?= 
+LDFLAGS ?=
+DYFLAGS ?=
 
 # build CFLAGS, LDFLAGS
 #
 ifeq (${OPENAIS_BUILD}, RELEASE) 
-	CFLAGS += -O3 -Wall
+	CFLAGS += -Wall
 # -Wstrict-aliasing=2 TODO sameday fix all of these
 ifndef OPENAIS_PROFILE
 	CFLAGS += -fomit-frame-pointer
