--- bin/package-ooo 
+++ bin/package-ooo 
@@ -33,6 +33,7 @@
 {
     cat <<EOT >$1
 #!/bin/sh
+unset PYTHONPATH
 $OOINSTBASE/program/soffice $2 "\$@"
 EOT
     chmod 755 $1
--- configure.in 
+++ configure.in 
@@ -1077,7 +1077,7 @@
    UPSTREAM_VERSION=2.3.0
 elif test "$with_tag" = "OOG680_m9" ; then
    UPSTREAM_VERSION=2.3.1
-elif test "$with_tag" = "OOH680_m11" ; then
+elif test "$with_tag" = "OOH680_m12" ; then
    UPSTREAM_VERSION=2.4.0
 fi
 if test -n "$UPSTREAM_VERSION"; then

