diff --git a/src/xfdesktop-icon-view.c b/src/xfdesktop-icon-view.c
index 0d7ba89..c6717ff 100644
--- a/src/xfdesktop-icon-view.c
+++ b/src/xfdesktop-icon-view.c
@@ -2591,12 +2591,14 @@ xfdesktop_icon_view_paint_icon(XfdesktopIconView *icon_view,
     GdkRectangle pix_area, text_area, intersection, adj_area;
     const gchar *label;
     guint16 row, col;
+    gboolean ret = FALSE;
     gchar x_offset = 0, y_offset = 0;
     GdkColor *sh_text_col = NULL;
     
     /*TRACE("entering (%s)", xfdesktop_icon_peek_label(icon));*/
     
-    g_return_if_fail(xfdesktop_icon_get_position(icon, &row, &col));
+    ret = xfdesktop_icon_get_position(icon, &row, &col);
+    g_return_if_fail(ret);
     
     if(g_list_find(icon_view->priv->selected_icons, icon)) {
         if(GTK_WIDGET_FLAGS(widget) & GTK_HAS_FOCUS)
