--- muse-0.6.2-orig/midithread.cpp	2004-07-11 02:36:40.102556216 +0200
+++ muse-0.6.2/midithread.cpp	2004-07-11 03:54:11.612419232 +0200
@@ -1015,7 +1015,7 @@
                                     if (track->type() == Track::DRUM)  { //Map drum-notes to the drum-map values
                                           int instr = drumInmap[ev->pitch()];
                                           pitch = drumMap[instr].anote;
-                                          port = drumMap[instr].port;
+                                          //port = drumMap[instr].port;	// respect the specified output port!
                                           channel = drumMap[instr].channel;
                                           }
                                     else
