diff -burN /var/tmp/portage/museseq-0.7.0-r1/work/muse-0.7.0/configure muse-0.7.0/configure
--- /var/tmp/portage/museseq-0.7.0-r1/work/muse-0.7.0/configure	2004-07-18 12:37:36.000000000 -0500
+++ muse-0.7.0/configure	2004-09-24 10:06:02.862960672 -0500
@@ -11922,7 +11922,8 @@
 ### suid-install ###
 ####################
 
-SUIDINSTALL="yes"
+#SUIDINSTALL="yes"
+SUIDINSTALL="no"
 
 
 ##################
diff -burN /var/tmp/portage/museseq-0.7.0-r1/work/muse-0.7.0/configure.ac muse-0.7.0/configure.ac
--- /var/tmp/portage/museseq-0.7.0-r1/work/muse-0.7.0/configure.ac	2004-07-18 12:34:31.000000000 -0500
+++ muse-0.7.0/configure.ac	2004-09-24 10:08:38.932756136 -0500
@@ -229,7 +229,8 @@
 ### suid-install ###
 ####################
 
-SUIDINSTALL="yes"
+#SUIDINSTALL="yes"
+SUIDINSTALL="no"
 AC_SUBST(SUIDINSTALL)
 
 ##################
