--- widgets/Makefile.in	2004-04-16 13:38:18.525257991 +0000
+++ widgets/Makefile.in	2004-04-16 13:39:12.334069714 +0000
@@ -765,7 +765,7 @@
 	-rm -rf ./$(DEPDIR)
 
 .cpp.o:
-@am__fastdepCXX_TRUE@	if $(CXXCOMPILE) -MT $@ -MD -MP -MF "$(DEPDIR)/$*.Tpo" \
+@am__fastdepCXX_TRUE@	if $(CXXCOMPILE) -fPIC -MT $@ -MD -MP -MF "$(DEPDIR)/$*.Tpo" \
 @am__fastdepCXX_TRUE@	  -c -o $@ `test -f '$<' || echo '$(srcdir)/'`$<; \
 @am__fastdepCXX_TRUE@	then mv "$(DEPDIR)/$*.Tpo" "$(DEPDIR)/$*.Po"; \
 @am__fastdepCXX_TRUE@	else rm -f "$(DEPDIR)/$*.Tpo"; exit 1; \
@@ -798,7 +798,7 @@
 @am__fastdepCXX_FALSE@	$(LTCXXCOMPILE) -c -o $@ `test -f '$<' || echo '$(srcdir)/'`$<
 
 xml.o: $(top_srcdir)/xml.cpp
-@am__fastdepCXX_TRUE@	if $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -MT xml.o -MD -MP -MF "$(DEPDIR)/xml.Tpo" \
+@am__fastdepCXX_TRUE@	if $(CXX) -fPIC $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -MT xml.o -MD -MP -MF "$(DEPDIR)/xml.Tpo" \
 @am__fastdepCXX_TRUE@	  -c -o xml.o `test -f '$(top_srcdir)/xml.cpp' || echo '$(srcdir)/'`$(top_srcdir)/xml.cpp; \
 @am__fastdepCXX_TRUE@	then mv "$(DEPDIR)/xml.Tpo" "$(DEPDIR)/xml.Po"; \
 @am__fastdepCXX_TRUE@	else rm -f "$(DEPDIR)/xml.Tpo"; exit 1; \
@@ -831,7 +831,7 @@
 @am__fastdepCXX_FALSE@	$(LIBTOOL) --mode=compile $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -c -o xml.lo `test -f '$(top_srcdir)/xml.cpp' || echo '$(srcdir)/'`$(top_srcdir)/xml.cpp
 
 tempo.o: $(top_srcdir)/tempo.cpp
-@am__fastdepCXX_TRUE@	if $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -MT tempo.o -MD -MP -MF "$(DEPDIR)/tempo.Tpo" \
+@am__fastdepCXX_TRUE@	if $(CXX) -fPIC $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -MT tempo.o -MD -MP -MF "$(DEPDIR)/tempo.Tpo" \
 @am__fastdepCXX_TRUE@	  -c -o tempo.o `test -f '$(top_srcdir)/tempo.cpp' || echo '$(srcdir)/'`$(top_srcdir)/tempo.cpp; \
 @am__fastdepCXX_TRUE@	then mv "$(DEPDIR)/tempo.Tpo" "$(DEPDIR)/tempo.Po"; \
 @am__fastdepCXX_TRUE@	else rm -f "$(DEPDIR)/tempo.Tpo"; exit 1; \
@@ -864,7 +864,7 @@
 @am__fastdepCXX_FALSE@	$(LIBTOOL) --mode=compile $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -c -o tempo.lo `test -f '$(top_srcdir)/tempo.cpp' || echo '$(srcdir)/'`$(top_srcdir)/tempo.cpp
 
 sig.o: $(top_srcdir)/sig.cpp
-@am__fastdepCXX_TRUE@	if $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -MT sig.o -MD -MP -MF "$(DEPDIR)/sig.Tpo" \
+@am__fastdepCXX_TRUE@	if $(CXX) -fPIC $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -MT sig.o -MD -MP -MF "$(DEPDIR)/sig.Tpo" \
 @am__fastdepCXX_TRUE@	  -c -o sig.o `test -f '$(top_srcdir)/sig.cpp' || echo '$(srcdir)/'`$(top_srcdir)/sig.cpp; \
 @am__fastdepCXX_TRUE@	then mv "$(DEPDIR)/sig.Tpo" "$(DEPDIR)/sig.Po"; \
 @am__fastdepCXX_TRUE@	else rm -f "$(DEPDIR)/sig.Tpo"; exit 1; \
@@ -897,7 +897,7 @@
 @am__fastdepCXX_FALSE@	$(LIBTOOL) --mode=compile $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -c -o sig.lo `test -f '$(top_srcdir)/sig.cpp' || echo '$(srcdir)/'`$(top_srcdir)/sig.cpp
 
 pos.o: $(top_srcdir)/pos.cpp
-@am__fastdepCXX_TRUE@	if $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -MT pos.o -MD -MP -MF "$(DEPDIR)/pos.Tpo" \
+@am__fastdepCXX_TRUE@	if $(CXX) -fPIC $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) -MT pos.o -MD -MP -MF "$(DEPDIR)/pos.Tpo" \
 @am__fastdepCXX_TRUE@	  -c -o pos.o `test -f '$(top_srcdir)/pos.cpp' || echo '$(srcdir)/'`$(top_srcdir)/pos.cpp; \
 @am__fastdepCXX_TRUE@	then mv "$(DEPDIR)/pos.Tpo" "$(DEPDIR)/pos.Po"; \
 @am__fastdepCXX_TRUE@	else rm -f "$(DEPDIR)/pos.Tpo"; exit 1; \
