diff -Nur muse-0.6.2-orig/app.cpp muse-0.6.2/app.cpp
--- muse-0.6.2-orig/app.cpp	2003-11-01 19:49:06.000000000 +0000
+++ muse-0.6.2/app.cpp	2004-06-29 00:05:35.712525381 +0000
@@ -1190,7 +1190,7 @@
             song->setName(project.baseName());
       setCaption(QString("MusE: Song: ") + project.baseName());
       arranger->reset();
-      arranger->setMode(int(song->mtype()));
+      arranger->setMode(long(song->mtype()));
       newSongFlag = true;
       song->dirty = false;
 
@@ -1257,8 +1257,8 @@
       {
       PianoRoll* pianoroll = new PianoRoll(pl, this);
       pianoroll->show();
-      toplevels.push_back(Toplevel(Toplevel::PIANO_ROLL, int(pianoroll), pianoroll));
-      connect(pianoroll, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+      toplevels.push_back(Toplevel( Toplevel::PIANO_ROLL, pianoroll, pianoroll));
+      connect(pianoroll, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
       }
 
 //---------------------------------------------------------
@@ -1277,8 +1277,8 @@
       {
       Score* scoreEditor = new Score(pl);
       scoreEditor->show();
-      toplevels.push_back(Toplevel(Toplevel::NOTEN, int(scoreEditor), scoreEditor));
-      connect(scoreEditor, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+      toplevels.push_back(Toplevel(Toplevel::NOTEN, scoreEditor, scoreEditor));
+      connect(scoreEditor, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
       }
 
 //---------------------------------------------------------
@@ -1297,8 +1297,8 @@
       {
       ListEdit* listEditor = new ListEdit(pl);
       listEditor->show();
-      toplevels.push_back(Toplevel(Toplevel::LISTE, int(listEditor), listEditor));
-      connect(listEditor, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+      toplevels.push_back(Toplevel(Toplevel::LISTE, listEditor, listEditor));
+      connect(listEditor, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
       }
 
 //---------------------------------------------------------
@@ -1309,8 +1309,8 @@
       {
       MasterEdit* masterEditor = new MasterEdit();
       masterEditor->show();
-      toplevels.push_back(Toplevel(Toplevel::MASTER, int(masterEditor), masterEditor));
-      connect(masterEditor, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+      toplevels.push_back(Toplevel(Toplevel::MASTER, masterEditor, masterEditor));
+      connect(masterEditor, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
       }
 
 //---------------------------------------------------------
@@ -1321,8 +1321,8 @@
       {
       LMaster* lmaster = new LMaster();
       lmaster->show();
-      toplevels.push_back(Toplevel(Toplevel::LMASTER, int(lmaster), lmaster));
-      connect(lmaster, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+      toplevels.push_back(Toplevel(Toplevel::LMASTER, lmaster, lmaster));
+      connect(lmaster, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
       }
 
 //---------------------------------------------------------
@@ -1341,8 +1341,8 @@
       {
       DrumEdit* drumEditor = new DrumEdit(pl, this);
       drumEditor->show();
-      toplevels.push_back(Toplevel(Toplevel::DRUM, int(drumEditor), drumEditor));
-      connect(drumEditor, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+      toplevels.push_back(Toplevel(Toplevel::DRUM, drumEditor, drumEditor));
+      connect(drumEditor, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
       }
 
 //---------------------------------------------------------
@@ -1363,8 +1363,8 @@
       {
       WaveEdit* waveEditor = new WaveEdit(pl);
       waveEditor->show();
-      toplevels.push_back(Toplevel(Toplevel::WAVE, int(waveEditor), waveEditor));
-      connect(waveEditor, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+      toplevels.push_back(Toplevel(Toplevel::WAVE, waveEditor, waveEditor));
+      connect(waveEditor, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
       }
 
 //---------------------------------------------------------
@@ -1375,10 +1375,10 @@
       {
       if (midiMixer == 0) {
             midiMixer = new Mixer();
-            toplevels.push_back(Toplevel(Toplevel::M_MIXER, int(midiMixer), midiMixer));
+            toplevels.push_back(Toplevel(Toplevel::M_MIXER, midiMixer, midiMixer));
             connect(midiMixer, SIGNAL(ctrlChanged(int,int,int,int)), song, SLOT(ctrlChanged(int,int,int,int)));
             connect(midiMixer, SIGNAL(masterVolChanged(int)), song, SLOT(masterVolChanged(int)));
-            connect(midiMixer, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+            connect(midiMixer, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
             }
       midiMixer->show();
       menu_functions->setItemChecked(midid1, true);
@@ -1401,8 +1401,8 @@
       {
       if (markerView == 0) {
             markerView = new MarkerView(this);
-            toplevels.push_back(Toplevel(Toplevel::MARKER, int(markerView), markerView));
-//            connect(markerView, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+            toplevels.push_back(Toplevel(Toplevel::MARKER, markerView, markerView));
+//            connect(markerView, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
             }
       markerView->show();
       }
@@ -1415,8 +1415,8 @@
       {
       if (audioMixer == 0) {
             audioMixer = new AudioMixerApp();
-            toplevels.push_back(Toplevel(Toplevel::A_MIXER, int(audioMixer), audioMixer));
-            connect(audioMixer, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+            toplevels.push_back(Toplevel(Toplevel::A_MIXER, audioMixer, audioMixer));
+            connect(audioMixer, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
             }
       audioMixer->show();
       menu_audio->setItemChecked(aid1, true);
@@ -1430,8 +1430,8 @@
       {
       if (clipListEdit == 0) {
             clipListEdit = new ClipListEdit();
-            toplevels.push_back(Toplevel(Toplevel::CLIPLIST, int(clipListEdit), clipListEdit));
-            connect(clipListEdit, SIGNAL(deleted(int)), SLOT(toplevelDeleted(int)));
+            toplevels.push_back(Toplevel(Toplevel::CLIPLIST, clipListEdit, clipListEdit));
+            connect(clipListEdit, SIGNAL(deleted(void*)), SLOT(toplevelDeleted(long)));
             }
       clipListEdit->show();
       menu_audio->setItemChecked(aid2, true);
@@ -1476,10 +1476,10 @@
 //   toplevelDeleted
 //---------------------------------------------------------
 
-void MusE::toplevelDeleted(int tl)
+void MusE::toplevelDeleted(void* tl)
       {
       for (iToplevel i = toplevels.begin(); i != toplevels.end(); ++i) {
-            if (i->object() == tl) {
+            if ( i->object() == tl ) {
                   switch(i->type()) {
                         case Toplevel::M_MIXER:
                               menu_functions->setItemChecked(midid1, false);
@@ -2156,7 +2156,7 @@
             Toplevel tl = *i;
             if (tl.type() == Toplevel::WAVE) {
                   TopWin* obj = tl.cobject();
-                  ((WaveEdit*)(obj))->setBg(WaveEdit::configBg());
+                  reinterpret_cast<WaveEdit*>(obj)->setBg(WaveEdit::configBg());
                   }
             }
       }
@@ -2593,7 +2593,7 @@
 again:
       for (iToplevel i = toplevels.begin(); i != toplevels.end(); ++i) {
             Toplevel tl = *i;
-            int obj = tl.object();
+            void* obj = tl.object();
             switch (tl.type()) {
                   case Toplevel::A_MIXER:
                         delete audioMixer;
@@ -2611,7 +2611,7 @@
                   case Toplevel::WAVE:
                   case Toplevel::LMASTER:
                   case Toplevel::MIXDOWN:
-                        ((QWidget*)(obj))->close(true);
+                        reinterpret_cast<QWidget*>(obj)->close(true);
                         goto again;
                   }
             }
diff -Nur muse-0.6.2-orig/app.h muse-0.6.2/app.h
--- muse-0.6.2-orig/app.h	2003-10-30 12:11:50.000000000 +0000
+++ muse-0.6.2/app.h	2004-06-28 23:52:54.358355806 +0000
@@ -227,7 +227,7 @@
       void ctrlChanged();
       void loadProjectFile(const QString&);
       void startAudioMixer();
-      void toplevelDeleted(int tl);
+      void toplevelDeleted(void* tl);
       void loadTheme(QString);
       void setTransportHandleColor(QColor);
       void setBigTimeForegroundColor(QColor);
diff -Nur muse-0.6.2-orig/cliplist/cliplist.cpp muse-0.6.2/cliplist/cliplist.cpp
--- muse-0.6.2-orig/cliplist/cliplist.cpp	2003-10-29 10:06:37.000000000 +0000
+++ muse-0.6.2/cliplist/cliplist.cpp	2004-06-29 00:13:59.767149620 +0000
@@ -184,7 +184,7 @@
 
 void ClipListEdit::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/cliplist/cliplist.h muse-0.6.2/cliplist/cliplist.h
--- muse-0.6.2-orig/cliplist/cliplist.h	2003-10-29 10:06:36.000000000 +0000
+++ muse-0.6.2/cliplist/cliplist.h	2004-06-29 00:14:11.080311133 +0000
@@ -37,7 +37,7 @@
       void clicked(QListViewItem*);
 
    signals:
-      void deleted(int);
+      void deleted(void*);
 
    public:
       ClipListEdit();
diff -Nur muse-0.6.2-orig/cobject.h muse-0.6.2/cobject.h
--- muse-0.6.2-orig/cobject.h	2003-10-29 10:05:10.000000000 +0000
+++ muse-0.6.2/cobject.h	2004-06-29 00:02:30.187403540 +0000
@@ -44,18 +44,16 @@
          , M_PATCHBAY
 #endif /* PATCHBAY */
          };
-      Toplevel(ToplevelType t, int obj, TopWin* cobj) {
-            _type = t;
-            _object = obj;
-            _cobject = cobj;
-            }
+	  // Rethink that : saving void*_object ( formerly int) and TopWin* _cobject makes no sense!
+      Toplevel(ToplevelType t, void * obj, TopWin* cobj)
+	  : _type(t), _object(obj), _cobject(cobj){ }
       ToplevelType type() const { return _type; }
-      int object()        const { return _object; }
+      void* object()        const { return _object; }
       TopWin* cobject()   const { return _cobject; }
 
    private:
       ToplevelType _type;
-      int _object;
+      void* _object;
       TopWin* _cobject;
       };
 
diff -Nur muse-0.6.2-orig/liste/listedit.cpp muse-0.6.2/liste/listedit.cpp
--- muse-0.6.2-orig/liste/listedit.cpp	2003-10-29 10:05:22.000000000 +0000
+++ muse-0.6.2/liste/listedit.cpp	2004-06-28 23:37:13.000000000 +0000
@@ -139,7 +139,7 @@
 
 void ListEdit::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/liste/listedit.h muse-0.6.2/liste/listedit.h
--- muse-0.6.2-orig/liste/listedit.h	2003-10-29 10:05:22.000000000 +0000
+++ muse-0.6.2/liste/listedit.h	2004-06-28 23:37:23.000000000 +0000
@@ -58,7 +58,7 @@
       void songChanged(int);
 
    signals:
-      void deleted(int);
+      void deleted(void*);
 
    public:
       ListEdit(PartList*);
diff -Nur muse-0.6.2-orig/marker/markerview.cpp muse-0.6.2/marker/markerview.cpp
--- muse-0.6.2-orig/marker/markerview.cpp	2003-10-29 10:06:13.000000000 +0000
+++ muse-0.6.2/marker/markerview.cpp	2004-06-28 23:40:45.776186926 +0000
@@ -136,7 +136,7 @@
 
 void MarkerView::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/marker/markerview.h muse-0.6.2/marker/markerview.h
--- muse-0.6.2-orig/marker/markerview.h	2003-10-29 10:06:13.000000000 +0000
+++ muse-0.6.2/marker/markerview.h	2004-06-28 23:44:51.775914379 +0000
@@ -67,7 +67,7 @@
       void songChanged(int);
 
    signals:
-      void deleted(int);
+      void deleted(void*);
 
    public:
       MarkerView(QWidget* parent);
diff -Nur muse-0.6.2-orig/master/lmaster.cpp muse-0.6.2/master/lmaster.cpp
--- muse-0.6.2-orig/master/lmaster.cpp	2003-10-29 10:06:13.000000000 +0000
+++ muse-0.6.2/master/lmaster.cpp	2004-06-29 00:12:14.582240878 +0000
@@ -26,7 +26,7 @@
 
 void LMaster::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/master/lmaster.h muse-0.6.2/master/lmaster.h
--- muse-0.6.2-orig/master/lmaster.h	2003-10-29 10:06:13.000000000 +0000
+++ muse-0.6.2/master/lmaster.h	2004-06-29 00:12:32.127391543 +0000
@@ -41,7 +41,7 @@
       void songChanged(int);
 
    signals:
-      void deleted(int);
+      void deleted(void*);
 
    public:
       LMaster();
diff -Nur muse-0.6.2-orig/master/masteredit.cpp muse-0.6.2/master/masteredit.cpp
--- muse-0.6.2-orig/master/masteredit.cpp	2003-10-29 10:06:14.000000000 +0000
+++ muse-0.6.2/master/masteredit.cpp	2004-06-29 00:12:42.339732011 +0000
@@ -49,7 +49,7 @@
 
 void MasterEdit::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/master/masteredit.h muse-0.6.2/master/masteredit.h
--- muse-0.6.2-orig/master/masteredit.h	2003-10-29 10:06:13.000000000 +0000
+++ muse-0.6.2/master/masteredit.h	2004-06-29 00:12:59.382962426 +0000
@@ -59,7 +59,7 @@
 //      void tempoChanged(double);
 
    signals:
-      void deleted(int);
+      void deleted(void*);
 
    public:
       MasterEdit();
diff -Nur muse-0.6.2-orig/memory.h muse-0.6.2/memory.h
--- muse-0.6.2-orig/memory.h	2003-10-29 10:05:18.000000000 +0000
+++ muse-0.6.2/memory.h	2004-06-28 23:30:01.906496190 +0000
@@ -50,7 +50,7 @@
       {
       int idx = ((n + sizeof(int) - 1) / sizeof(int)) - 1;
       if (idx >= dimension) {
-            printf("panic: alloc %d\n", n);
+            printf("panic: alloc %lu\n", n);
             exit(-1);
             }
       if (head[idx] == 0)
@@ -68,7 +68,7 @@
       {
       int idx = ((n + sizeof(int) - 1) / sizeof(int)) - 1;
       if (idx >= dimension) {
-            printf("panic: alloc %d\n", n);
+            printf("panic: alloc %lu\n", n);
             exit(-1);
             }
       Verweis* p = static_cast<Verweis*>(b);
diff -Nur muse-0.6.2-orig/midi.cpp muse-0.6.2/midi.cpp
--- muse-0.6.2-orig/midi.cpp	2003-11-01 18:12:52.000000000 +0000
+++ muse-0.6.2/midi.cpp	2004-06-28 23:48:57.449826762 +0000
@@ -522,9 +522,9 @@
             MidiEvent* event = readEvent(track);
             if (event == 0)
                   break;
-            if (int(event) == -1)
+            if (long(event) == -1)
                   continue;
-            if (int(event) == -2)         // error
+            if (long(event) == -2)         // error
                   return true;
             if (lastchan == -1) {
                   el->add(event);         // kann kein channel event sein
diff -Nur muse-0.6.2-orig/midiedit/drumedit.cpp muse-0.6.2/midiedit/drumedit.cpp
--- muse-0.6.2-orig/midiedit/drumedit.cpp	2003-10-29 10:06:08.000000000 +0000
+++ muse-0.6.2/midiedit/drumedit.cpp	2004-06-28 23:33:26.000000000 +0000
@@ -133,7 +133,7 @@
       _dlistWidthInit = *it; //There are only 2 values stored in the sizelist, size of dlist widget and dcanvas widget 
       it++; 
       _dcanvasWidthInit = *it; 
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/midiedit/drumedit.h muse-0.6.2/midiedit/drumedit.h
--- muse-0.6.2-orig/midiedit/drumedit.h	2003-10-29 10:06:10.000000000 +0000
+++ muse-0.6.2/midiedit/drumedit.h	2004-06-28 23:33:39.000000000 +0000
@@ -120,7 +120,7 @@
       void dlistSelectionChanged(DrumMap*); // called by dlist
 
    signals:
-      void deleted(int);
+      void deleted(void*);
       void keyFilterChanged(int);
 
    public:
diff -Nur muse-0.6.2-orig/midiedit/pianoroll.cpp muse-0.6.2/midiedit/pianoroll.cpp
--- muse-0.6.2-orig/midiedit/pianoroll.cpp	2003-10-29 10:06:09.000000000 +0000
+++ muse-0.6.2/midiedit/pianoroll.cpp	2004-06-28 23:30:56.037570909 +0000
@@ -446,7 +446,7 @@
 
 void PianoRoll::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/midiedit/pianoroll.h muse-0.6.2/midiedit/pianoroll.h
--- muse-0.6.2-orig/midiedit/pianoroll.h	2003-10-29 10:06:07.000000000 +0000
+++ muse-0.6.2/midiedit/pianoroll.h	2004-06-28 23:32:34.737301042 +0000
@@ -108,7 +108,7 @@
       void songChanged1(int);
 
    signals:
-      void deleted(int);
+      void deleted(void *);
 
    public:
       PianoRoll(PartList*, QWidget* parent = 0, const char* name = 0);
diff -Nur muse-0.6.2-orig/mixer/amixer.cpp muse-0.6.2/mixer/amixer.cpp
--- muse-0.6.2-orig/mixer/amixer.cpp	2003-10-29 10:05:23.000000000 +0000
+++ muse-0.6.2/mixer/amixer.cpp	2004-06-28 23:54:59.000000000 +0000
@@ -753,7 +753,7 @@
 
 void AudioMixerApp::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/mixer/amixer.h muse-0.6.2/mixer/amixer.h
--- muse-0.6.2-orig/mixer/amixer.h	2003-10-29 10:05:24.000000000 +0000
+++ muse-0.6.2/mixer/amixer.h	2004-06-28 23:55:11.316020930 +0000
@@ -94,7 +94,7 @@
       void updateOutputRoute();
 
    signals:
-      void deleted(int);
+      void deleted(void*);
 
    private slots:
       void heartBeat();
diff -Nur muse-0.6.2-orig/mixer/mixer.cpp muse-0.6.2/mixer/mixer.cpp
--- muse-0.6.2-orig/mixer/mixer.cpp	2003-10-29 10:05:24.000000000 +0000
+++ muse-0.6.2/mixer/mixer.cpp	2004-06-29 00:13:40.588266335 +0000
@@ -226,7 +226,7 @@
 
 void Mixer::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/mixer/mixer.h muse-0.6.2/mixer/mixer.h
--- muse-0.6.2-orig/mixer/mixer.h	2003-10-29 10:05:23.000000000 +0000
+++ muse-0.6.2/mixer/mixer.h	2004-06-29 00:13:52.681301128 +0000
@@ -60,7 +60,7 @@
       void masterChanged(int val);
 
    signals:
-      void deleted(int);
+      void deleted(void*);
       void ctrlChanged(int port, int chan, int ctrl, int val);
       void masterVolChanged(int val);
 
diff -Nur muse-0.6.2-orig/score/score.cpp muse-0.6.2/score/score.cpp
--- muse-0.6.2-orig/score/score.cpp	2003-10-29 10:05:26.000000000 +0000
+++ muse-0.6.2/score/score.cpp	2004-06-28 23:40:32.568460346 +0000
@@ -360,7 +360,7 @@
 
 void Score::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/score/score.h muse-0.6.2/score/score.h
--- muse-0.6.2-orig/score/score.h	2003-10-29 10:05:24.000000000 +0000
+++ muse-0.6.2/score/score.h	2004-06-28 23:41:15.037151699 +0000
@@ -91,7 +91,7 @@
    signals:
       void rasterChanged(int);
       void quantChanged(int);
-      void deleted(int);
+      void deleted(void*);
 
    public:
       Score(PartList*);
diff -Nur muse-0.6.2-orig/synti/synth.cpp muse-0.6.2/synti/synth.cpp
--- muse-0.6.2-orig/synti/synth.cpp	2003-10-29 10:06:00.000000000 +0000
+++ muse-0.6.2/synti/synth.cpp	2004-06-28 23:26:35.000000000 +0000
@@ -334,7 +334,7 @@
             nsynthis = list->count();
             if (debugMsg)
                   printf("%d soft synth found\n", nsynthis);
-            synthis  = new (Synth*)[nsynthis];
+            synthis  = new Synth*[nsynthis];
             QFileInfoListIterator it(*list);
             QFileInfo* fi;
             int i = 0;
diff -Nur muse-0.6.2-orig/track.cpp muse-0.6.2/track.cpp
--- muse-0.6.2-orig/track.cpp	2003-10-29 22:14:12.000000000 +0000
+++ muse-0.6.2/track.cpp	2004-06-28 23:49:48.000000000 +0000
@@ -95,7 +95,7 @@
 
 void Track::dump() const
       {
-      printf("Track <%s>: typ %d, parts %d sel %d\n",
+      printf("Track <%s>: typ %d, parts %lu sel %d\n",
          _name.latin1(), _type, _parts.size(), _selected);
       }
 
diff -Nur muse-0.6.2-orig/waveedit/waveedit.cpp muse-0.6.2/waveedit/waveedit.cpp
--- muse-0.6.2-orig/waveedit/waveedit.cpp	2003-10-29 10:06:14.000000000 +0000
+++ muse-0.6.2/waveedit/waveedit.cpp	2004-06-29 00:13:17.236061212 +0000
@@ -57,7 +57,7 @@
 
 void WaveEdit::closeEvent(QCloseEvent* e)
       {
-      emit deleted((int)this);
+      emit deleted(this);
       e->accept();
       }
 
diff -Nur muse-0.6.2-orig/waveedit/waveedit.h muse-0.6.2/waveedit/waveedit.h
--- muse-0.6.2-orig/waveedit/waveedit.h	2003-10-29 10:06:14.000000000 +0000
+++ muse-0.6.2/waveedit/waveedit.h	2004-06-29 00:13:30.436915997 +0000
@@ -48,7 +48,7 @@
       void soloChanged(bool flag);
 
    signals:
-      void deleted(int);
+      void deleted(void*);
 
    public:
       WaveEdit(PartList*);

