diff -Nru colortail-0.3.0/CfgFileParser.cc colortailnew/CfgFileParser.cc
--- colortail-0.3.0/CfgFileParser.cc	Tue Aug  3 14:40:44 1999
+++ colortailnew/CfgFileParser.cc	Sun Jul 14 22:08:05 2002
@@ -19,12 +19,13 @@
 
 #include "CfgFileParser.h"
 
-#include <stdio.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <assert.h>
-#include <string.h>
+#include <string>
 #include <malloc.h>
 
+using namespace std;
 
 // ## class SearchData ##
 
diff -Nru colortail-0.3.0/Colorizer.cc colortailnew/Colorizer.cc
--- colortail-0.3.0/Colorizer.cc	Tue Aug  3 14:41:17 1999
+++ colortailnew/Colorizer.cc	Sun Jul 14 22:10:42 2002
@@ -22,9 +22,10 @@
 #include "TailFile.h"
 
 #include <assert.h>
-#include <stdio.h>
-#include <strstream.h>
+#include <iostream>
+#include <string>
 
+using namespace std;
 
 Colorizer::Colorizer()
 {
