--- src/sounds/Makefile.in.bad	2007-06-02 11:24:44.000000000 +0200
+++ src/sounds/Makefile.in	2007-06-02 11:25:08.000000000 +0200
@@ -373,7 +353,7 @@
 
 info-am:
 
-install-data-am: install-SoundsDATA install-data-local
+install-data-am: install-SoundsDATA
 
 install-exec-am:
 
--- src/translations/Makefile.in.bad	2007-06-02 11:24:48.000000000 +0200
+++ src/translations/Makefile.in	2007-06-02 11:25:08.000000000 +0200
@@ -374,7 +353,7 @@
 
 info-am:
 
-install-data-am: install-TranslationsDATA install-data-local
+install-data-am: install-TranslationsDATA
 
 install-exec-am:
 
