--- library/text.tcl	Mon Nov 11 17:39:09 2002
+++ library/text.tcl	Mon Nov 11 17:39:28 2002
@@ -338,14 +338,6 @@
     }
 }
 
-if {[string compare $tcl_platform(platform) "windows"]} {
-bind Text <Control-v> {
-    if {!$tk_strictMotif} {
-	tkTextScrollPages %W 1
-    }
-}
-}
-
 bind Text <Meta-b> {
     if {!$tk_strictMotif} {
 	tkTextSetCursor %W [tkTextPrevPos %W insert tcl_startOfPreviousWord]
