--- setup.py.orig	2009-02-27 20:17:52.004477424 +0000
+++ setup.py	2009-02-27 20:18:42.009572270 +0000
@@ -135,12 +135,6 @@
 execfile(os.path.join('pygraphviz','release.py'))
 
 packages = ["pygraphviz","pygraphviz.tests"]
-docdirbase  = 'share/doc/pygraphviz-%s' % version
-data = [(docdirbase, glob("*.txt")),
-        (os.path.join(docdirbase, 'examples'),glob("examples/*.py")),
-        (os.path.join(docdirbase, 'examples'),glob("examples/*.dat")),
-        (os.path.join(docdirbase, 'examples'),glob("examples/*.dat.gz")),
-        ]
 extension = [Extension("pygraphviz._graphviz",
                       ["pygraphviz/graphviz_wrap.c"],
                       include_dirs=include_dirs,
@@ -165,7 +159,6 @@
       url              = url,      
       download_url     = download_url,
       packages         = packages,
-      data_files       = data,
       classifiers      = classifiers,
       ext_modules      = extension,
       package_data     = package_data,
--- setup_egg.py.orig	2009-02-27 20:19:32.994136036 +0000
+++ setup_egg.py	2009-02-27 20:20:14.253100416 +0000
@@ -34,7 +34,6 @@
       url              = url,      
       download_url     = download_url,
       packages         = packages,
-      data_files       = data,
       classifiers      = classifiers,
       ext_modules      = extension,
       package_data     = package_data,
