--- transcode-0.6.12/bbmpeg/mplex.h.old	2004-09-22 17:26:14.722188082 -0700
+++ transcode-0.6.12/bbmpeg/mplex.h	2004-09-22 17:26:48.601657790 -0700
@@ -364,7 +364,9 @@
   unsigned int     write_seq_end,
   unsigned int     write_seq_hdr,
   unsigned int     sh_length,
-  unsigned char    *seq_hdr);
+  unsigned char    *seq_hdr,
+  unsigned int     nsyncwords,
+  unsigned int     firstsync);
 
 void create_sys_header	  (	/* erstellt einen System Header		*/
    Sys_header_struc *sys_header,
--- transcode-0.6.12/bbmpeg/tcmplex/multplex.c.old	2004-09-22 17:27:04.443007097 -0700
+++ transcode-0.6.12/bbmpeg/tcmplex/multplex.c	2004-09-22 17:32:41.781855873 -0700
@@ -1787,7 +1787,7 @@
                 istream_v, VIDEO_STR_0, 0, 1, video_buffer_size/1024,
                 video_PSTD, &PTSsave, &DTSsave, timestamps, which_streams, 0, i,
                 do_broken_link, (start_new_file || stop_output) && write_end_codes,
-                do_sequence_header, sh_length, seq_hdr))
+                do_sequence_header, sh_length, seq_hdr, 0, 0))
     return FALSE;
 
 
@@ -1899,6 +1899,7 @@
   Timecode_struc PTSsave;
   int eos, tcfound;
   char tmpStr[100];
+  unsigned int nsyncwords, firstsync;  /* firstsync is 1-based */
 
   if (marker_pack)
   {
@@ -1938,6 +1939,8 @@
 
   tcfound = 0;
   eos = 0;
+  nsyncwords = 0;
+  firstsync = 0;
   /* Let's generate packet					*/
 
   /* does a audio frame start in this packet?			*/
@@ -1945,6 +1948,8 @@
   /* CASE: packet starts with new access unit			*/
   if (*audio_frame_start)
   {
+    nsyncwords++;
+    firstsync = 1;
     timestamps = (mplex_type > MPEG_VCD) ? MPEG2_TIMESTAMPS_PTS : MPEG1_TIMESTAMPS_PTS;
     PTSsave = audio_au->PTS;
     tcfound = 1;
@@ -1979,6 +1984,9 @@
         *audio_counter = *audio_counter + 1;
         if ((!restart_output && !stop_output) || (get_timecode(&audio_au->PTS) < start_video_PTS))
         {
+         nsyncwords++;
+         if (!firstsync)
+             firstsync = temp + 1;
           add_to_timecode(SCR_delay, &audio_au->PTS);
           if (temp + audio_au->length > packet_data_size)
           {
@@ -2023,7 +2031,8 @@
   if (!create_sector(&sector, pack_ptr, sys_header_ptr,
 		packet_data_size, temp,
 		istream_a, aid, audio_subid, abuffer_scale, abuffer_size,
-		audio_PSTD, &PTSsave, NULL, timestamps, which_streams, k, 0, 0, 0, 0, 0, NULL))
+                audio_PSTD, &PTSsave, NULL, timestamps, which_streams, k, 0, 0, 0, 0, 0, NULL,
+                nsyncwords, firstsync))
      return FALSE;
 
   *audio_frame_start = FALSE;
@@ -2186,7 +2195,7 @@
 		i, i,
 		NULL, PADDING_STR, 0, 0, 0,
 		FALSE, NULL, NULL,
-		TIMESTAMPS_NO, j, k, 0, 0, 0, 0, 0, NULL);
+                TIMESTAMPS_NO, j, k, 0, 0, 0, 0, 0, NULL, 0, 0);
 
   if (output_on)
   {
--- transcode-0.6.12/bbmpeg/tcmplex/systems.c.old	2004-09-22 17:32:58.548108803 -0700
+++ transcode-0.6.12/bbmpeg/tcmplex/systems.c	2004-09-22 17:35:56.562559425 -0700
@@ -32,10 +32,12 @@
 unsigned int     write_seq_end,
 unsigned int     write_seq_hdr,
 unsigned int     sh_length,
-unsigned char    *seq_hdr)
+unsigned char    *seq_hdr,
+unsigned int     nsyncwords,
+unsigned int     firstsync)
 {
-  int i, j, data_bytes, numFound, syncFound, firstOffset, pad_size;
-  unsigned char *index, *tmpPtr=NULL;
+  int i, j, data_bytes, pad_size;
+  unsigned char *index;
   unsigned int data_size, stuff_size, k;
   unsigned int l;
 
@@ -329,41 +331,14 @@
     if (type == PRIVATE_STREAM1)
     {
       *(index++) = subtype;
-      tmpPtr = index; // remember for AC3 data
-      *(index++) = 0; // num of AC3 syncwords in packet
-      *(index++) = 0; // hi offset of first AC3 syncword
-      *(index++) = 0; // lo offset of first AC3 syncword (starting at 1)
+      *(index++) = nsyncwords;        // num of AC3 syncwords in packet
+      *(index++) = firstsync >> 8;    // hi offset of first AC3 syncword
+                                      //   (starting at 1)
+      *(index++) = firstsync & 0xFF;  // lo offset of first AC3 syncword
       data_bytes -= 4;
     }
 
     memcpy(index, data_buffer, data_bytes);
-    if (type == PRIVATE_STREAM1)
-    {
-      numFound = 0;
-      syncFound = 0;
-      firstOffset = 0;
-      for (j = 0; j < data_bytes; j++)
-      {
-        switch (syncFound)
-        {
-          case 0:
-            if (index[j] == 0x0b)
-              syncFound = 1;
-            break;
-          case 1:
-            if (index[j] == 0x77)
-            {
-              numFound++;
-              if (numFound == 1)
-                firstOffset = j;
-            }
-            syncFound = 0;
-        }
-      }
-      tmpPtr[0] = (unsigned char) numFound;
-      tmpPtr[1] = (unsigned char) (firstOffset >> 8);
-      tmpPtr[2] = (unsigned char) (firstOffset & 0xff);
-    }
     index += data_bytes;
 
     if (type == PRIVATE_STREAM1)
