diff -ur transcode-0.6.14.orig/libvo/yuv2rgb.c transcode-0.6.14/libvo/yuv2rgb.c
--- transcode-0.6.14.orig/libvo/yuv2rgb.c	2004-11-03 04:45:39.000000000 +0100
+++ transcode-0.6.14/libvo/yuv2rgb.c	2005-04-13 01:37:20.095054432 +0200
@@ -71,7 +71,8 @@
 void yuv2rgb_init (int bpp, int mode) 
 {
     yuv2rgb = NULL;
-#ifdef ARCH_X86
+#ifdef HAVE_MMX
+# if defined(ARCH_X86) || defined(ARCH_X86_64)
     if ((yuv2rgb == NULL) && (vo_mm_accel & MM_ACCEL_X86_MMXEXT)) {
 	yuv2rgb = yuv2rgb_init_mmxext (bpp, mode);
 	if (yuv2rgb != NULL)
@@ -82,6 +83,7 @@
 	if (yuv2rgb != NULL)
 	    fprintf (stderr, "Using MMX for colorspace transform\n");
     }
+# endif
 #endif
 #ifdef LIBVO_MLIB
     if ((yuv2rgb == NULL) && (vo_mm_accel & MM_ACCEL_MLIB)) {
