--- testlib.py
+++ testlib.py
@@ -1022,8 +1022,12 @@
 
     def __init__(self, methodName='runTest'):
         super(TestCase, self).__init__(methodName)
+        # internal API changed in python2.7
+        if sys.version_info >= (2, 7):
+            self.__exc_info = sys.exc_info
+            self.__testMethodName = self._testMethodName
         # internal API changed in python2.5
-        if sys.version_info >= (2, 5):
+        elif sys.version_info >= (2, 5):
             self.__exc_info = self._exc_info
             self.__testMethodName = self._testMethodName
         else:
