diff -ur rigs-0.2.2.20050714/Ruby/GNUmakefile rigs-0.2.2.20050714-test/Ruby/GNUmakefile
--- rigs-0.2.2.20050714/Ruby/GNUmakefile	2005-06-19 18:21:43.000000000 -0400
+++ rigs-0.2.2.20050714-test/Ruby/GNUmakefile	2005-07-14 20:45:37.000000000 -0400
@@ -28,13 +28,13 @@
 
 # Dynamically determine where the things must be installed
 # in Ruby
-RUBY_SITE_LIB_DIR := $(shell ruby -rrbconfig -e 'print Config::CONFIG["sitelibdir"]')
+#RUBY_SITE_LIB_DIR := $(shell ruby -rrbconfig -e 'print Config::CONFIG["sitelibdir"]')
 
 
-after-install::
-	@cp -f rigs.rb Foundation.rb AppKit.rb $(RUBY_SITE_LIB_DIR)
-	@rm -rf $(RUBY_SITE_LIB_DIR)/rigs/*
-	@cp -rf rigs $(RUBY_SITE_LIB_DIR)
+#after-install::
+#	@cp -f rigs.rb Foundation.rb AppKit.rb $(RUBY_SITE_LIB_DIR)
+#	@rm -rf $(RUBY_SITE_LIB_DIR)/rigs/*
+#	@cp -rf rigs $(RUBY_SITE_LIB_DIR)
 
 # Is there something more appropriate than tool.make here ?
-include $(GNUSTEP_MAKEFILES)/tool.make
\ No newline at end of file
+include $(GNUSTEP_MAKEFILES)/tool.make
diff -ur rigs-0.2.2.20050714/Source/GNUmakefile.postamble rigs-0.2.2.20050714-test/Source/GNUmakefile.postamble
--- rigs-0.2.2.20050714/Source/GNUmakefile.postamble	2005-06-19 18:21:43.000000000 -0400
+++ rigs-0.2.2.20050714-test/Source/GNUmakefile.postamble	2005-07-14 21:02:06.000000000 -0400
@@ -41,17 +41,17 @@
 # Things to do after installing
 # install the librigs.so in Ruby dir. Name it librigs.so even if it is
 # a debug version (omit the _d suffix)
-ifeq ($(debug),)
-
-after-install::
-	@cp -f $(GNUSTEP_OBJ_DIR)/$(LIBRARY_NAME).so $(RUBY_SITE_ARCH_DIR)
-
-else
-
-after-install::
-	@cp -f $(GNUSTEP_OBJ_DIR)/$(LIBRARY_NAME)_d.so $(RUBY_SITE_ARCH_DIR)/$(LIBRARY_NAME).so
-
-endif
+#ifeq ($(debug),)
+#
+#after-install::
+#	@cp -f $(GNUSTEP_OBJ_DIR)/$(LIBRARY_NAME).so $(RUBY_SITE_ARCH_DIR)
+#
+#else
+#
+#after-install::
+#	@cp -f $(GNUSTEP_OBJ_DIR)/$(LIBRARY_NAME)_d.so $(RUBY_SITE_ARCH_DIR)/$(LIBRARY_NAME).so
+#
+#endif
 
 # Things to do before uninstalling
 #before-uninstall::
