diff -uNr shorewall-4.4.10.ORIG//install.sh shorewall-4.4.10/install.sh
--- shorewall-4.4.10.ORIG//install.sh	2010-06-16 15:32:46.000000000 +0100
+++ shorewall-4.4.10/install.sh	2010-06-16 15:33:29.000000000 +0100
@@ -745,7 +745,7 @@
 #
 # Install the  Makefiles
 #
-install-file Makefile-lite ${DESTDIR}/usr/share/shorewall/configfiles/Makefile 0644
+install_file Makefile-lite ${DESTDIR}/usr/share/shorewall/configfiles/Makefile 0644
 
 if [ -z "$SPARSE" ]; then
     run_install $OWNERSHIP -m 0600 Makefile ${DESTDIR}/etc/shorewall
