--- openafs-1.2.6/src/pinstall/install.c	Fri Oct 12 23:21:46 2001
+++ src/pinstall/install.c	Tue Oct  1 18:11:55 2002
@@ -169,6 +169,10 @@
 		"cp", 0, 0, 0,
 	};
 	
+	static char *empty[] = {
+	  0,
+	};
+	
 	/*
 	 * first, copy the `iname' to the `oname'
 	 */
@@ -180,7 +184,7 @@
 	    case 0:			/* child	*/
 		copy[1] = iname;
 		copy[2] = oname;
-		execve("/bin/cp", copy, (char **)0);
+		execve("/bin/cp", copy, empty);
 		perror("/bin/cp");
 		exit(1);
 
@@ -277,7 +281,11 @@
 	static char *copy[] = {
 		"cp", 0, 0, 0,
 	};
-
+	
+static char *empty[] = {
+	  0,
+	};
+	
 	/*
 	 * first, copy the `iname' to the `oname'
 	 */
@@ -289,7 +297,7 @@
 	    case 0:			/* child	*/
 		copy[1] = iname;
 		copy[2] = oname;
-		execve("/bin/cp", copy, 0);
+		execve("/bin/cp", copy, empty);
 		perror("/bin/cp");
 		exit(1);
 
