*** mythtv-0.15/libs/libmythtv/videoout_directfb.cpp~	Wed May 26 22:50:27 2004
--- mythtv-0.15/libs/libmythtv/videoout_directfb.cpp	Wed May 26 22:50:27 2004
***************
*** 337,343 ****
      desc.flags = DSDESC_CAPS;
      desc.caps = DSCAPS_PRIMARY;
      if(data->cardCapabilities.acceleration_mask & DFXL_BLIT)
!         desc.caps = (DFBSurfaceCapabilities)(desc.caps | DSCAPS_DOUBLE);
      DFBCHECKFAIL(data->dfb->CreateSurface(data->dfb, &desc, &(data->primarySurface)), false);
      DFBCHECKFAIL(data->primarySurface->Clear(data->primarySurface, 0, 0, 0, 0xff), false);
      DFBCHECKFAIL(data->primarySurface->Flip(data->primarySurface, 0, DSFLIP_ONSYNC), false);
--- 337,343 ----
      desc.flags = DSDESC_CAPS;
      desc.caps = DSCAPS_PRIMARY;
      if(data->cardCapabilities.acceleration_mask & DFXL_BLIT)
!         desc.caps = (DFBSurfaceCapabilities)(desc.caps | DSCAPS_FLIPPING);
      DFBCHECKFAIL(data->dfb->CreateSurface(data->dfb, &desc, &(data->primarySurface)), false);
      DFBCHECKFAIL(data->primarySurface->Clear(data->primarySurface, 0, 0, 0, 0xff), false);
      DFBCHECKFAIL(data->primarySurface->Flip(data->primarySurface, 0, DSFLIP_ONSYNC), false);
