Index: libs/libmythtv/channelutil.cpp
===================================================================
--- libs/libmythtv/channelutil.cpp     (revision 8554)
+++ libs/libmythtv/channelutil.cpp     (working copy)
@@ -11,11 +11,11 @@
                                  // DVB specific
                                  int transport_id,      int network_id,
                                  bool set_odfm_info,
-                                 int symbol_rate,       char bandwidth,
-                                 char polarity,         char inversion,
-                                 char trans_mode,
+                                 int symbol_rate,       signed char bandwidth,
+                                 signed char polarity,         signed char inversion,
+                                 signed char trans_mode,
                                  QString inner_FEC,     QString constellation,
-                                 char hierarchy,        QString hp_code_rate,
+                                 signed char hierarchy,        QString hp_code_rate,
                                  QString lp_code_rate,  QString guard_interval)
 {
     MSqlQuery query(MSqlQuery::InitCon());

