--- chan_sccp-20050725/Makefile.orig	2005-07-29 04:55:59.000000000 +0000
+++ chan_sccp-20050725/Makefile	2005-07-29 04:56:21.000000000 +0000
@@ -1,24 +1,16 @@
 INSTALL_PREFIX=
-ASTERISK_HEADER_DIR=$(INSTALL_PREFIX)/usr/include
+ASTERISK_HEADER_DIR=/usr/include
 
 # Location asterisk modules install to
-AST_MODULES_DIR=/usr/lib/asterisk/modules/
+AST_MODULES_DIR=/usr/lib/asterisk/modules
 
 PROC=$(shell uname -m)
 
 DEBUG=-ggdb
 
-DEFINE=-Iinclude -I$(ASTERISK_SOURCE)/include -D_REENTRANT -D_GNU_SOURCE  -O  -DCRYPTO -fPIC -DASTERISK_VERSION_$(ASTERISK_VERSION)
+DEFINE=-Iinclude -D_REENTRANT -D_GNU_SOURCE -DCRYPTO -fPIC
 CFLAG= -pipe -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations
-CFLAG+=$(shell if $(CC) -march=$(PROC) -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo " -march=$(PROC)"; fi)
 CFLAG+=$(shell if uname -m | grep -q ppc; then echo " -fsigned-char"; fi)
-#CFLAG+= -pedantic
-#CFLAG+= -W
-#CFLAG+= -Wpointer-arith
-#CFLAG+= -Wcast-qual
-#CFLAG+= -Wwrite-strings
-#CFLAG+= -Wconversion
-#CFLAG+= -Wredundant-decls # Useless as too many false positives from asterisk source
 CFLAG+= -Wnested-externs 
 CFLAG+= -Wlong-long
 CFLAG+= $(DEBUG)
@@ -41,26 +33,31 @@
 FLS+=sccp_socket
 FLS+=sccp_indicate
 
-CFLAGS+=$(DEFINE) $(CFLAG) -I$(ASTERISK_SOURCE)
+CFLAGS+=$(DEFINE) $(CFLAG)
 HEADERS=$(shell for i in $(FLS) ; do echo $$i.h ; done)
 OBJECTS=$(shell for i in $(FLS) ; do echo .tmp/$$i.o ; done)
 SOURCES=$(shell for i in $(FLS) ; do echo $$i.c ; done)
 
+all: chan_sccp.so
+
 chan_sccp.so: .config .tmp $(OBJECTS) $(HEADERS) $(SOURCES)
 	@echo "Linking chan_sccp.so"
 	@$(CC) -shared -Xlinker -x -o chan_sccp.so $(OBJECTS)
 
 .tmp/%.o: $(HEADERS)
 	@printf "Now compiling .... %-15s\t%s lines \n" $*.c "`wc -l <$*.c`"
-	@$(CC) -I/usr/include/ $(CFLAGS) -I$(ASTERISK_SOURCE) $(DEFINE) -c $*.c -o .tmp/$*.o
+	@$(CC) $(CFLAGS) -c $*.c -o .tmp/$*.o
 
-install: chan_sccp.so
+install:
 	@echo "Now Installing chan_sccp.so"
-	@install -m 755 chan_sccp.so $(AST_MODULES_DIR);
-	@if ! [ -f /etc/asterisk/sccp.conf ]; then \
-		echo "Installing config file /etc/asterisk/sccp.conf"; \
-		cp conf/sccp.conf /etc/asterisk/; \
+	@if ! [ -d $(INSTALL_PREFIX)$(AST_MODULES_DIR) ]; then \
+		mkdir -p -m775 $(INSTALL_PREFIX)$(AST_MODULES_DIR); \
+	fi
+	@install -m 755 chan_sccp.so $(INSTALL_PREFIX)$(AST_MODULES_DIR);
+	@if ! [ -d $(INSTALL_PREFIX)/etc/asterisk ]; then \
+		mkdir -p -m750 $(INSTALL_PREFIX)/etc/asterisk; \
 	fi
+	@install -m 640 conf/sccp.conf $(INSTALL_PREFIX)/etc/asterisk
 	@echo "Chan_sccp is now installed"
 	@echo "Remember to disable chan_skinny by adding the following"
 	@echo "line to /etc/asterisk/modules.conf:"
