Index: lib/myghty/importer.py
===================================================================
--- lib/myghty/importer.py	(revision 2153)
+++ lib/myghty/importer.py	(revision 2155)
@@ -47,11 +47,20 @@
 
 # override __import__ to look in our own local module dict first
 builtin_importer = __builtin__.__import__
-def import_module(name, globals = None, locals = None, fromlist = None):
-    try:
-        return modules[name].module
-    except KeyError:
-        return builtin_importer(name, globals, locals, fromlist)
+if sys.version_info >= (2, 5):
+    def import_module(name, globals = None, locals = None, fromlist = None, level = -1):
+        if level == -1:
+            try:
+                return modules[name].module
+            except KeyError:
+                pass
+        return builtin_importer(name, globals, locals, fromlist, level)
+else:
+    def import_module(name, globals = None, locals = None, fromlist = None):
+        try:
+            return modules[name].module
+        except KeyError:
+            return builtin_importer(name, globals, locals, fromlist)
 
 __builtin__.__import__ = import_module
 
