diff -ur GConf-1.0.7/backends.orig/xml-backend.c GConf-1.0.7/backends/xml-backend.c
--- GConf-1.0.7/backends.orig/xml-backend.c	Fri Oct 12 21:30:26 2001
+++ GConf-1.0.7/backends/xml-backend.c	Sat Nov 24 15:26:48 2001
@@ -26,8 +26,8 @@
 #include "xml-cache.h"
 
 
-#include <libxml/tree.h>
-#include <libxml/parser.h>
+#include <gnome-xml/tree.h>
+#include <gnome-xml/parser.h>
 
 #include <stdio.h>
 #include <time.h>
diff -ur GConf-1.0.7/backends.orig/xml-cache.h GConf-1.0.7/backends/xml-cache.h
--- GConf-1.0.7/backends.orig/xml-cache.h	Fri Oct 12 21:01:20 2001
+++ GConf-1.0.7/backends/xml-cache.h	Sat Nov 24 15:27:04 2001
@@ -21,7 +21,7 @@
 #define GCONF_XML_CACHE_H
 
 #include <gconf/gconf.h>
-#include <libxml/tree.h>
+#include <gnome-xml/tree.h>
 #include "xml-dir.h"
 
 typedef struct _Cache Cache;
diff -ur GConf-1.0.7/backends.orig/xml-dir.c GConf-1.0.7/backends/xml-dir.c
--- GConf-1.0.7/backends.orig/xml-dir.c	Fri Oct 12 21:01:25 2001
+++ GConf-1.0.7/backends/xml-dir.c	Sat Nov 24 15:27:19 2001
@@ -20,7 +20,7 @@
 #include "xml-dir.h"
 #include "xml-entry.h"
 
-#include <libxml/parser.h>
+#include <gnome-xml/parser.h>
 
 #include <stdio.h>
 #include <time.h>
diff -ur GConf-1.0.7/backends.orig/xml-dir.h GConf-1.0.7/backends/xml-dir.h
--- GConf-1.0.7/backends.orig/xml-dir.h	Fri Oct 12 21:01:30 2001
+++ GConf-1.0.7/backends/xml-dir.h	Sat Nov 24 15:27:27 2001
@@ -21,7 +21,7 @@
 #define GCONF_XML_DIR_H
 
 #include <gconf/gconf.h>
-#include <libxml/tree.h>
+#include <gnome-xml/tree.h>
 
 /* Dir stores the information about a given directory */
 
diff -ur GConf-1.0.7/backends.orig/xml-entry.c GConf-1.0.7/backends/xml-entry.c
--- GConf-1.0.7/backends.orig/xml-entry.c	Fri Oct 12 21:30:58 2001
+++ GConf-1.0.7/backends/xml-entry.c	Sat Nov 24 15:27:44 2001
@@ -20,8 +20,8 @@
 #include "xml-entry.h"
 #include <gconf/gconf-internals.h>
 #include <stdlib.h>
-#include <libxml/entities.h>
-#include <libxml/xmlmemory.h>
+#include <gnome-xml/entities.h>
+#include <gnome-xml/xmlmemory.h>
 
 static void
 entry_sync_if_needed(Entry* e, GConfValue* val);
