Index: /branches/lighttpd-1.4.x/src/mod_cgi.c
===================================================================
--- /branches/lighttpd-1.4.x/src/mod_cgi.c (revision 2065)
+++ /branches/lighttpd-1.4.x/src/mod_cgi.c (revision 2107)
@@ -1005,4 +1005,5 @@
 		/* error */
 		log_error_write(srv, __FILE__, __LINE__, "ss", "fork failed:", strerror(errno));
+		return -1;
 		break;
 	default: {
Index: /branches/lighttpd-1.4.x/NEWS
===================================================================
--- /branches/lighttpd-1.4.x/NEWS (revision 2106)
+++ /branches/lighttpd-1.4.x/NEWS (revision 2107)
@@ -46,4 +46,5 @@
   * fixed body handling of status 300 requests 
   * spawn-fcgi: only try to connect to unix socket (not tcp) before spawning (#1575)
+  * fix sending source of cgi script instead of 500 error if fork fails
 
 - 1.4.18 - 2007-09-09
