diff -ur gnome-libs-1.4.2-orig/configure.in gnome-libs-1.4.2/configure.in
--- gnome-libs-1.4.2-orig/configure.in	2006-01-16 03:30:34.000000000 +0000
+++ gnome-libs-1.4.2/configure.in	2006-01-16 03:34:06.000000000 +0000
@@ -304,7 +304,6 @@
 AC_CHECK_FUNCS(strdup)
 AC_CHECK_HEADERS(utime.h)
 AC_FUNC_MMAP
-AC_CHECK_TYPE(umode_t, int)
 
 AC_ARG_ENABLE(prefer-db1, [  --enable-prefer-db1     Prefer Berkeley DB 1.x] [default=no]],[prefer_db1="$enableval"],[prefer_db1=no])
 dnl
