--- configure.in.old	2006-08-18 14:05:40.000000000 +0200
+++ configure.in	2006-08-18 14:05:17.000000000 +0200
@@ -52,10 +52,10 @@
 dnl
 ALL_LINGUAS="ja zh_CN zh_TW"
 AM_GNU_GETTEXT([external])
-LDFLAGS="$LDFLAGS $INTLLIBS"
+LIBS="$LIBS $INTLLIBS"
 AC_CHECK_LIB(iconv, iconv, haslibiconv=yes, haslibiconv=no)
 if test "x$haslibiconv" = "xyes"; then
-    LDFLAGS="$LDFLAGS $LIBICONV"
+    LIBS="$LIBS $LIBICONV"
 fi
 localedir='${prefix}/share/locale'
 AC_SUBST(localedir)
@@ -311,10 +311,10 @@
 
   *)
     AC_MSG_CHECKING(whether threads work with -pthread)
-    LDSAVEFLAGS=$LDFLAGS
-    LDFLAGS="$LDFLAGS -pthread"
+    LIBSSAVE=$LIBS
+    LIBS="$LIBS -pthread"
     AC_TRY_LINK_FUNC(pthread_create, pthread_ok=yes, pthread_ok=no)
-    LDFLAGS=$LDSAVEFLAGS
+    LIBS=$LIBSSAVE
 
     if test "x$pthread_ok" = "xyes"; then
       AC_MSG_RESULT(yes)
@@ -349,20 +349,20 @@
 case $target in
   *-*-solaris*)
     AC_MSG_CHECKING(whether SunOS has -lrt )
-    LDSAVEFLAGS="$LDFLAGS"
-    LDFLAGS="$LDFLAGS -lrt"
+    LIBSSAVE="$LIBS"
+    LIBS="$LIBS -lrt"
     AC_TRY_LINK_FUNC(nanosleep, rt_ok=yes, rt_ok=no)
     if test "x$rt_ok" = "xyes"; then
       AC_MSG_RESULT(yes)
     else
       AC_MSG_RESULT(no)
       AC_MSG_CHECKING(whether SunOS has -lposix4 )
-      LDFLAGS="$LDSAVEFLAGS -lposix4"
+      LIBS="$LIBSSAVE -lposix4"
       AC_TRY_LINK_FUNC(nanosleep, posix_ok=yes, posix_ok=no)
       if test "x$posix_ok" = "xyes"; then
         AC_MSG_RESULT(yes)
       else
-        LDFLAGS=$LDSAVEFLAGS
+        LIBS=$LIBSSAVE
         AC_MSG_RESULT(no)
         AC_MSG_WARN([*** Try setting LIBS or LDFLAGS manually to point to the library with nanosleep()***])
       fi
@@ -406,7 +406,7 @@
     fi
   else
     CFLAGS="$CFLAGS `$XFT_CONFIG --cflags`"
-    LDFLAGS="$LDFLAGS `$XFT_CONFIG --libs`"
+    LIBS="$LIBS `$XFT_CONFIG --libs`"
   fi
 fi
 if test "x$enable_cookies" = "xno" ; then
