diff -ur ffmpegthumbnailer-1.2.3.orig/src/libffmpegthumbnailer/jpegwriter.cpp ffmpegthumbnailer-1.2.3/src/libffmpegthumbnailer/jpegwriter.cpp
--- ffmpegthumbnailer-1.2.3.orig/src/libffmpegthumbnailer/jpegwriter.cpp	2008-04-08 21:27:03.000000000 +0300
+++ ffmpegthumbnailer-1.2.3/src/libffmpegthumbnailer/jpegwriter.cpp	2008-04-15 22:30:03.000000000 +0300
@@ -16,6 +16,7 @@
 
 #include "jpegwriter.h"
 #include <stdexcept>
+#include <cstring>
 #include <assert.h>
 
 using namespace std;
diff -ur ffmpegthumbnailer-1.2.3.orig/src/libffmpegthumbnailer/moviedecoder.cpp ffmpegthumbnailer-1.2.3/src/libffmpegthumbnailer/moviedecoder.cpp
--- ffmpegthumbnailer-1.2.3.orig/src/libffmpegthumbnailer/moviedecoder.cpp	2008-04-08 21:27:13.000000000 +0300
+++ ffmpegthumbnailer-1.2.3/src/libffmpegthumbnailer/moviedecoder.cpp	2008-04-15 22:29:57.000000000 +0300
@@ -19,6 +19,7 @@
 
 #include <assert.h>
 #include <iostream>
+#include <cstring>
 
 extern "C" {
 #if HAVE_LIBSWSCALE_SWSCALE_H
diff -ur ffmpegthumbnailer-1.2.3.orig/src/main.cpp ffmpegthumbnailer-1.2.3/src/main.cpp
--- ffmpegthumbnailer-1.2.3.orig/src/main.cpp	2008-04-08 21:27:03.000000000 +0300
+++ ffmpegthumbnailer-1.2.3/src/main.cpp	2008-04-15 22:34:38.000000000 +0300
@@ -17,6 +17,7 @@
 #include <iostream>
 #include <unistd.h>
 #include <stdexcept>
+#include <stdlib.h>
 
 #include "libffmpegthumbnailer/videothumbnailer.h"
 #include "libffmpegthumbnailer/stringoperations.h"
