diff -ur rcairo-1.2.0/packages/cairo/ext/rb_cairo_constants.c rcairo-1.2.0.new/packages/cairo/ext/rb_cairo_constants.c
--- rcairo-1.2.0/packages/cairo/ext/rb_cairo_constants.c	2006-07-01 21:45:55.000000000 +0700
+++ rcairo-1.2.0.new/packages/cairo/ext/rb_cairo_constants.c	2006-12-31 11:56:41.000000000 +0700
@@ -97,7 +97,9 @@
 DEFINE_RVAL2ENUM(format, FORMAT)
 DEFINE_RVAL2ENUM(extend, EXTEND)
 DEFINE_RVAL2ENUM(filter, FILTER)
+#if CAIRO_HAS_SVG_SURFACE
 DEFINE_RVAL2ENUM(svg_version, SVG_VERSION)
+#endif
 
 
 void
