--- test/Makefile.orig	2007-09-25 10:47:24.000000000 -0400
+++ test/Makefile	2007-09-25 10:47:30.000000000 -0400
@@ -12,13 +12,7 @@
 include $(top_srcdir)/config/Make.rules
 
 SUBDIRS		= IceUtil \
-		  Ice \
-                  IceSSL \
-		  IceStorm \
-		  Freeze \
-		  FreezeScript \
-		  Glacier2 \
-		  IceGrid
+		  Ice
 
 $(EVERYTHING)::
 	@for subdir in $(SUBDIRS); \
--- src/Makefile.orig	2007-09-25 10:31:16.000000000 -0400
+++ src/Makefile	2007-09-25 10:52:38.000000000 -0400
@@ -15,28 +15,7 @@
 		  icecpp \
 		  Slice \
 		  slice2cpp \
-		  slice2cs \
-		  slice2vb \
-		  slice2freeze \
-		  slice2freezej \
-		  slice2docbook \
-		  slice2java \
-		  slice2py \
-		  slice2cppe \
-		  slice2javae \
-		  slice2rb \
-		  slice2html \
-		  Ice \
-		  IceXML \
-		  IceSSL \
-		  Freeze \
-		  FreezeScript \
-		  IceBox \
-		  IceStorm \
-		  Glacier2 \
-		  IcePatch2 \
-		  IceGrid \
-		  ca
+		  Ice
 
 $(EVERYTHING)::
 	@for subdir in $(SUBDIRS); \
