diff --exclude-from=/home/dang/.diffrc -u -ruN sabayon-2.12.4.orig/lib/Makefile.in sabayon-2.12.4/lib/Makefile.in
--- sabayon-2.12.4.orig/lib/Makefile.in	2006-06-22 11:12:08.000000000 -0400
+++ sabayon-2.12.4/lib/Makefile.in	2006-08-08 17:41:10.000000000 -0400
@@ -53,7 +53,7 @@
     *) f=$$p;; \
   esac;
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
-am__installdirs = "$(DESTDIR)$(sabayondir)" "$(DESTDIR)$(sabayondir)"
+am__installdirs = "$(DESTDIR)$(sabayondir)" "$(DESTDIR)$(sabayonexecdir)"
 sabayonLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(sabayon_LTLIBRARIES)
 am__DEPENDENCIES_1 =
@@ -296,6 +296,7 @@
 	config.py.in
 
 sabayondir = $(pythondir)/sabayon
+sabayonexecdir = $(pyexecdir)/sabayon
 sabayon_PYTHON = $(PYTHON_FILES)
 DISTCLEANFILES = config.py
 BUILT_SOURCES = config.py
@@ -335,12 +336,12 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-sabayonLTLIBRARIES: $(sabayon_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
-	test -z "$(sabayondir)" || $(mkdir_p) "$(DESTDIR)$(sabayondir)"
+	test -z "$(sabayonexecdir)" || $(mkdir_p) "$(DESTDIR)$(sabayonexecdir)"
 	@list='$(sabayon_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
-	    echo " $(LIBTOOL) --mode=install $(sabayonLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(sabayondir)/$$f'"; \
-	    $(LIBTOOL) --mode=install $(sabayonLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(sabayondir)/$$f"; \
+	    echo " $(LIBTOOL) --mode=install $(sabayonLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(sabayonexecdir)/$$f'"; \
+	    $(LIBTOOL) --mode=install $(sabayonLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(sabayonexecdir)/$$f"; \
 	  else :; fi; \
 	done
 
@@ -348,8 +349,8 @@
 	@$(NORMAL_UNINSTALL)
 	@set -x; list='$(sabayon_LTLIBRARIES)'; for p in $$list; do \
 	  p=$(am__strip_dir) \
-	  echo " $(LIBTOOL) --mode=uninstall rm -f '$(DESTDIR)$(sabayondir)/$$p'"; \
-	  $(LIBTOOL) --mode=uninstall rm -f "$(DESTDIR)$(sabayondir)/$$p"; \
+	  echo " $(LIBTOOL) --mode=uninstall rm -f '$(DESTDIR)$(sabayonexecdir)/$$p'"; \
+	  $(LIBTOOL) --mode=uninstall rm -f "$(DESTDIR)$(sabayonexecdir)/$$p"; \
 	done
 
 clean-sabayonLTLIBRARIES:
@@ -361,7 +362,7 @@
 	  rm -f "$${dir}/so_locations"; \
 	done
 xlib.la: $(xlib_la_OBJECTS) $(xlib_la_DEPENDENCIES) 
-	$(LINK) -rpath $(sabayondir) $(xlib_la_LDFLAGS) $(xlib_la_OBJECTS) $(xlib_la_LIBADD) $(LIBS)
+	$(LINK) -rpath $(sabayonexecdir) $(xlib_la_LDFLAGS) $(xlib_la_OBJECTS) $(xlib_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
@@ -610,7 +611,7 @@
 all-am: Makefile $(LTLIBRARIES)
 installdirs: installdirs-recursive
 installdirs-am:
-	for dir in "$(DESTDIR)$(sabayondir)" "$(DESTDIR)$(sabayondir)"; do \
+	for dir in "$(DESTDIR)$(sabayondir)" "$(DESTDIR)$(sabayonexecdir)"; do \
 	  test -z "$$dir" || $(mkdir_p) "$$dir"; \
 	done
 install: $(BUILT_SOURCES)
