--- ktoblzcheck-1.7/src/include/iban.h	2005-05-24 10:04:21.000000000 +0200
+++ ktoblzcheck-1.7-gcc41/src/include/iban.h	2005-11-28 23:54:52.107511750 +0100
@@ -219,8 +219,8 @@
   bool readSpecTable(std::istream &fin, const std::string& stopcomment);
   bool readCountryTable(std::istream &fin);
   static int to_number(char c) { return c - 'A' + 10; }
-  static std::string IbanCheck::iban2number(const std::string& iban);
-  static int IbanCheck::modulo97(const std::string& number);
+  static std::string iban2number(const std::string& iban);
+  static int modulo97(const std::string& number);
 
   specmap m_IbanSpec;
   countrymap m_CountryMap;
