diff -Naur uade-0.91-orig/plugindir/beepmp/uade.c uade-0.91/plugindir/beepmp/uade.c
--- uade-0.91-orig/plugindir/beepmp/uade.c	2004-07-27 16:22:37.000000000 +0200
+++ uade-0.91/plugindir/beepmp/uade.c	2004-09-09 15:37:24.609571008 +0200
@@ -305,9 +305,12 @@
 /* xmms calls this function to check song */
 static int is_our_file(char *filename) {
   int ours;
+  const char *prefix = "file://";
   if(!uade_is_operational) {
     return FALSE;
   }
+  if (strncasecmp(filename, prefix, strlen(prefix)) == 0)
+    filename += strlen(prefix);
   ours = check_my_file(filename, 0, 0);
   if (ours) {
     FILE *f = fopen(filename, "rb");
diff -Naur uade-0.91-orig/src/uade.c uade-0.91/src/uade.c
--- uade-0.91-orig/src/uade.c	2004-07-27 16:22:37.000000000 +0200
+++ uade-0.91/src/uade.c	2004-09-09 15:38:58.252335144 +0200
@@ -443,6 +443,9 @@
 {
   int bytesread, status;
   FILE *file;
+  const char *prefix = "file://";
+  if (strncasecmp(name, prefix, strlen(prefix)) == 0)
+    name += strlen(prefix);
   file = fopen(name, "rb");
   if (!file) {
     fprintf(stderr,"uade: couldn't load %s %s\n", expl, name);
