diff -Naurp uade-2.12-orig/src/frontends/uadefs/Makefile.in uade-2.12/src/frontends/uadefs/Makefile.in
--- uade-2.12-orig/src/frontends/uadefs/Makefile.in	2008-11-30 20:23:27.000000000 +0100
+++ uade-2.12/src/frontends/uadefs/Makefile.in	2008-11-30 20:24:01.000000000 +0100
@@ -16,7 +16,7 @@ MODULES = uadefs.o $(COMMONMODULES)
 	$(CC) $(CFLAGS) -c $<
 
 uadefs:	$(MODULES)
-	$(CC) $(CFLAGS) $(CLIBS) -o $@ $(MODULES)
+	$(CC) $(CFLAGS) -o $@ $(MODULES) $(CLIBS)
 
 uadefs.o:	uadefs.c
 
