Index: kdebase-3.5.0/ksplashml/themeengine/simple/configure.in.in
===================================================================
--- kdebase-3.5.0/ksplashml/themeengine/simple/configure.in.in	(revision 497108)
+++ kdebase-3.5.0/ksplashml/themeengine/simple/configure.in.in	(working copy)
@@ -1,9 +1,13 @@
-xinerama_save_ldflags="$LDFLAGS"
-LDFLAGS="$X_LDFLAGS"
-
 LIB_XINERAMA=
 
-KDE_CHECK_HEADER(X11/extensions/Xinerama.h,
+AC_ARG_WITH([xinerama],
+  AS_HELP_STRING([--without-xinerama], [Disable xinerama extension support (default: check)]) )
+
+if test "x$with_xinerama" != "xno"; then
+  xinerama_save_ldflags="$LDFLAGS"
+  LDFLAGS="$X_LDFLAGS"
+
+  KDE_CHECK_HEADER(X11/extensions/Xinerama.h,
     [
     AC_CHECK_LIB(Xinerama,XineramaQueryExtension,
 	[
@@ -17,5 +21,7 @@
      #include <X11/Xlib.h>
      ] )
 
-AC_SUBST(LIB_XINERAMA)
-LDFLAGS="$xinerama_save_ldflags"
+  AC_SUBST(LIB_XINERAMA)
+  LDFLAGS="$xinerama_save_ldflags"
+fi
+
