diff -urp pcmcia-cs-3.2.8-orinoco/wireless/orinoco.c pcmcia-cs-3.2.8-orinoco-prototype/wireless/orinoco.c
--- pcmcia-cs-3.2.8-orinoco/wireless/orinoco.c	2005-04-17 15:37:56.000000000 +0200
+++ pcmcia-cs-3.2.8-orinoco-prototype/wireless/orinoco.c	2005-04-17 15:39:32.000000000 +0200
@@ -569,6 +569,10 @@ static void orinoco_stat_gather(struct n
 static struct net_device_stats *orinoco_get_stats(struct net_device *dev);
 static struct iw_statistics *orinoco_get_wireless_stats(struct net_device *dev);
 
+void orinoco_int_rxmonitor(struct net_device *dev, u16 rxfid,
+			   int len, struct hermes_rx_descriptor *rxdesc,
+			   struct ieee802_11_hdr *hdr);
+
 /* Hardware control routines */
 
 static int __orinoco_program_rids(struct net_device *dev);
