--- aircrack.c.orig	2005-04-11 16:59:28.000000000 +0200
+++ aircrack.c	2005-04-11 17:19:14.000000000 +0200
@@ -121,7 +121,12 @@
     unsigned long cnt1;
     unsigned long cnt2;
     unsigned char *h80211;
-    struct pcap_pkthdr pkh;
+    struct {
+        struct {
+	    int tv_sec, tv_usec;
+        } ts;
+	int caplen, len;
+    } pkh;
     struct pcap_file_header pfh;
 
     /* open the file and check the pcap header */
