diff -ur led-0.0.3.old/led.cc led-0.0.3/led.cc
--- led-0.0.3.old/led.cc	2000-07-28 19:35:53.000000000 -0700
+++ led-0.0.3/led.cc	2003-10-20 20:30:36.000000000 -0700
@@ -64,7 +64,7 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <string>
-#include <iostream.h>
+#include <iostream>
 
 #include <gtk/gtk.h>
 
@@ -82,7 +82,7 @@
 
 }
 
-Led_Port::Led_Port (unsigned int _num_iopins=8) : IOPORT(_num_iopins)
+Led_Port::Led_Port (unsigned int _num_iopins) : IOPORT(_num_iopins)
 {
 
 }
