Index: src/fe-text/fe-text.c
===================================================================
RCS file: /cvsroot/xchat/xchat2/src/fe-text/fe-text.c,v
retrieving revision 1.19
retrieving revision 1.20
diff -u -b -B -r1.19 -r1.20
--- src/fe-text/fe-text.c	8 Jul 2004 05:03:25 -0000	1.19
+++ src/fe-text/fe-text.c	23 Nov 2004 02:58:34 -0000	1.20
@@ -815,3 +815,13 @@
 void fe_confirm (const char *message, void (*yesproc)(void *), void (*noproc)(void *), void *ud)
 {
 }
+char *fe_get_inputbox_contents (struct session *sess)
+{
+	return NULL;
+}
+void fe_set_inputbox_contents (struct session *sess, char *text)
+{
+}
+void fe_set_inputbox_cursor (struct session *sess, int delta, int pos)
+{
+}
