--- xchat-1.8.11/src/fe-gtk/xtext.c	2002-12-03 18:08:45.000000000 +1100
+++ xchat-1.8.11p1/src/fe-gtk/xtext.c	2003-01-05 20:56:01.000000000 +1100
@@ -505,7 +505,11 @@
 	int height_only = FALSE;
 
 	if (allocation->width == widget->allocation.width)
+	{
+		if (allocation->height == widget->allocation.height)
+			return;
 		height_only = TRUE;
+	}
 
 	widget->allocation = *allocation;
 	if (GTK_WIDGET_REALIZED (widget))
