diff -ru xchat-1.8.5/src/common/xchat.c xchat-1.8.5p1/src/common/xchat.c
--- xchat-1.8.5/src/common/xchat.c	Mon Oct 15 00:46:36 2001
+++ xchat-1.8.5p1/src/common/xchat.c	Sun Oct 28 20:37:48 2001
@@ -69,6 +69,7 @@
 GSList *urlhandler_list = 0;
 static GSList *away_list = 0;
 
+static int in_xchat_exit = FALSE;
 int xchat_is_quitting = FALSE;
 int auto_connect = TRUE;
 
@@ -730,7 +731,7 @@
 	free (killsess->current_modes);
 	free (killsess);
 
-	if (!sess_list && !xchat_is_quitting)
+	if (!sess_list && !in_xchat_exit)
 		xchat_exit ();						/* sess_list is empty, quit! */
 
 	list = sess_list;
@@ -1051,6 +1052,7 @@
 xchat_exit (void)
 {
 	xchat_is_quitting = TRUE;
+	in_xchat_exit = TRUE;
 #ifdef USE_PERL
 	perl_end ();
 #endif
diff -ru xchat-1.8.5/src/fe-gtk/maingui.c xchat-1.8.5p1/src/fe-gtk/maingui.c
--- xchat-1.8.5/src/fe-gtk/maingui.c	Thu Oct 25 17:19:29 2001
+++ xchat-1.8.5p1/src/fe-gtk/maingui.c	Sun Oct 28 20:23:18 2001
@@ -923,7 +923,6 @@
 static void
 gui_main_window_kill (gpointer userdata)
 {
-#if 0
 	GSList *list;
 	session *sess;
 
@@ -941,7 +940,6 @@
 		}
 		list = list->next;
 	}
-#endif
 
 	main_window = 0;
 	current_tab = 0;
