diff -Nur xchat-2.6.6.orig/plugins/dbus/xchat-remote.c xchat-2.6.6/plugins/dbus/xchat-remote.c
--- xchat-2.6.6.orig/plugins/dbus/xchat-remote.c	2006-07-15 08:27:21.000000000 +0000
+++ xchat-2.6.6/plugins/dbus/xchat-remote.c	2006-10-13 14:16:57.801263851 +0000
@@ -110,7 +110,7 @@
         if (fork() == 0)
         {
           gchar *url = g_strdup_printf ("--url=%s", opt_open_url);
-          execl (PREFIX"/bin/"PACKAGE_TARNAME, PACKAGE_TARNAME, "-a", url, NULL);
+          execl (PREFIX"/bin/"PACKAGE_TARNAME"-2", PACKAGE_TARNAME"-2", "-a", url, NULL);
         }
       } else
         write_error (_("Failed to complete command"), error);
