diff -ru --exclude=po --exclude=Makefile.in xchat-1.9.3/src/common/util.c xchat-1.9.4/src/common/util.c
--- xchat-1.9.3/src/common/util.c	Wed Aug 21 19:47:38 2002
+++ xchat-1.9.4/src/common/util.c	Sat Oct  5 02:18:36 2002
@@ -266,7 +266,7 @@
 
 	if (len == 0)
 		return (char *)s;
-	while (tolower(*s) != tolower(*wanted) || strncasecmp (s, wanted, len))
+	while (rfc_tolower(*s) != rfc_tolower(*wanted) || strncasecmp (s, wanted, len))
 		if (*s++ == '\0')
 			return (char *)NULL;
 	return (char *)s;
@@ -574,7 +574,7 @@
 	if (*m == '?' || *m == '*')
 	  ch = *m++;
       default:
-	if (tolower(*s) != tolower(ch))
+	if (rfc_tolower(*s) != rfc_tolower(ch))
 	  return 0;
       case '?':
 	if (!*s++)
@@ -604,8 +604,8 @@
 break_while:
   if (!ch)
     return 1;			/* mask ends with '*', we got it */
-  ch = tolower(ch);
-  while (tolower(*s++) != ch)
+  ch = rfc_tolower(ch);
+  while (rfc_tolower(*s++) != ch)
     if (!*s)
       return 0;
   bs = s;			/* Next try start from here */
@@ -621,7 +621,7 @@
 	if (*m == '?' || *m == '*')
 	  ch = *m++;
       default:
-	if (tolower(*s) != tolower(ch))
+	if (rfc_tolower(*s) != rfc_tolower(ch))
 	{
 	  m = bm;
 	  s = bs;
@@ -676,7 +676,7 @@
 {
 	while (*str)
 	{
-		*str = tolower (*str);
+		*str = rfc_tolower (*str);
 		str++;
 	}
 }
@@ -1122,7 +1122,7 @@
 	register unsigned char *str2 = (unsigned char *) s2;
 	register int res;
 
-	while ((res = tolower (*str1) - tolower (*str2)) == 0)
+	while ((res = rfc_tolower (*str1) - rfc_tolower (*str2)) == 0)
 	{
 		if (*str1 == '\0')
 			return 0;
@@ -1139,7 +1139,7 @@
 	register unsigned char *s2 = (unsigned char *) str2;
 	register int res;
 
-	while ((res = tolower (*s1) - tolower (*s2)) == 0)
+	while ((res = rfc_tolower (*s1) - rfc_tolower (*s2)) == 0)
 	{
 		s1++;
 		s2++;
@@ -1150,7 +1150,7 @@
 	return (res);
 }
 
-const unsigned char tolowertab[] =
+const unsigned char rfc_tolowertab[] =
 	{ 0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xa,
 	0xb, 0xc, 0xd, 0xe, 0xf, 0x10, 0x11, 0x12, 0x13, 0x14,
 	0x15, 0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d,
diff -ru --exclude=po --exclude=Makefile.in xchat-1.9.3/src/common/util.h xchat-1.9.4/src/common/util.h
--- xchat-1.9.3/src/common/util.h	Wed Aug 21 19:47:44 2002
+++ xchat-1.9.4/src/common/util.h	Sat Oct  5 02:17:39 2002
@@ -11,10 +11,9 @@
  *        --+ Dagmar d'Surreal
  */
 
-#undef tolower
-#define tolower(c) (tolowertab[(unsigned char)(c)])
+#define rfc_tolower(c) (rfc_tolowertab[(unsigned char)(c)])
 
-extern const unsigned char tolowertab[];
+extern const unsigned char rfc_tolowertab[];
 
 int my_poptParseArgvString(const char * s, int * argcPtr, char *** argvPtr);
 char *expand_homedir (char *file);
diff -ru --exclude=po --exclude=Makefile.in xchat-1.9.3/src/fe-gtk/fkeys.c xchat-1.9.4/src/fe-gtk/fkeys.c
--- xchat-1.9.3/src/fe-gtk/fkeys.c	Wed Jun 26 01:01:42 2002
+++ xchat-1.9.4/src/fe-gtk/fkeys.c	Sat Oct  5 02:19:13 2002
@@ -1808,7 +1808,7 @@
 				if (match_char == 0xff)
 				{
 					match_char = current_nick[match_pos];
-				} else if (tolower (current_nick[match_pos]) != tolower (match_char))
+				} else if (rfc_tolower (current_nick[match_pos]) != rfc_tolower (match_char))
 				{
 					match_text = malloc (match_pos + 1);
 					memcpy (match_text, current_nick, match_pos);
