--- mod_lisp2.c.orig	2005-12-24 23:04:20.000000000 +0100
+++ mod_lisp2.c	2005-12-24 23:05:09.000000000 +0100
@@ -58,13 +58,17 @@
 /* 
   Change log:
 
-  Fixed a declaration for some versions of gcc
-  -- Marc Battyani
-     2005-08-26
-
-  Set r->mtime directly
-  -- Dr. Edmund Weitz <edi@agharta.de>
-     2005-06-07
+   Handle Lisp data correctly if there's no Content-Length header
+   -- Dr. Edmund Weitz <edi@agharta.de>
+      2005-12-24
+
+   Fixed a declaration for some versions of gcc
+   -- Marc Battyani
+      2005-08-26
+
+   Set r->mtime directly
+   -- Dr. Edmund Weitz <edi@agharta.de>
+      2005-06-07
 
    Read data from Lisp even if it's a HEAD request
    Added "Lisp-Content-Length" header (send after "Content-Length" header to overwrite its value)
@@ -88,7 +92,7 @@
       2003-12-02
 */
 
-#define VERSION_STRING "1.2"
+#define VERSION_STRING "1.3"
 #define READ_TIMEOUT 60000000
 
 #include "httpd.h"
@@ -673,7 +677,6 @@
     }
 
   /* Copy the reply entity from Lisp to the client...  */
-  if (content_length > 0)
   {
     unsigned int n_read = 0;
     input_buffer_t * buffer;
