--- bin/setup	2004-06-27 06:25:06.000000000 +0200
+++ bin/setup	2004-07-29 11:41:01.000000000 +0200
@@ -38,7 +38,8 @@
 # Use the default remote config by default
 if ( ${?LGCONFIG} == 0 ) then
 #    setenv LGCONFIG file://${scriptdir}/lgconfig_x_remote.xml
-    setenv LGCONFIG 
+    setenv LGCONFIG file://${scriptdir}/../etc/lg3d/lgconfig_1p_nox.xml
+    #setenv LGCONFIG 
 endif
 
 setenv CLASSPATH ${lgdir}/lib/ext/lg3d-core.jar:${lgdir}/ext/escher-0.2.2.jar:${lgdir}/
