Index: gen_compile.sh
===================================================================
RCS file: /var/cvsroot/gentoo/src/genkernel_bradmssw/gen_compile.sh,v
retrieving revision 1.38
retrieving revision 1.39
diff -u -b -B -r1.38 -r1.39
--- gen_compile.sh	15 Jun 2005 19:05:31 -0000	1.38
+++ gen_compile.sh	16 Jun 2005 21:23:13 -0000	1.39
@@ -689,7 +689,13 @@
 		print_info 1 'udev: >> Compiling...'
 
 		ln -snf "${KERNEL_DIR}" klibc/linux || gen_die "Could not link to ${KERNEL_DIR}"
-		compile_generic "ARCH=${ARCH} KERNEL_DIR=$KERNEL_DIR USE_KLIBC=true USE_LOG=false DEBUG=false udevdir=/dev all etc/udev/udev.conf" utils
+		if [ "${ARCH}" = 'um' ]
+		then
+			compile_generic "ARCH=um KERNEL_DIR=$KERNEL_DIR USE_KLIBC=true USE_LOG=false DEBUG=false udevdir=/dev all etc/udev/udev.conf" utils
+		else
+			compile_generic "KERNEL_DIR=$KERNEL_DIR USE_KLIBC=true USE_LOG=false DEBUG=false udevdir=/dev all etc/udev/udev.conf" utils
+		fi
+
 
 		strip udev || gen_die 'Failed to strip the udev binary!'
 
