diff --git a/setup.py b/setup.py
index 8db801c..80aa3bf 100755
--- a/setup.py
+++ b/setup.py
@@ -482,20 +482,6 @@ def DoSourcePackage():
     else:
         from distutils.core import setup
 
-    # Try to remove possibly conflicting files from an old install
-    try:
-        import Editra
-        path = Editra.__file__
-        if '__init__' in path:
-            path = os.path.dirname(path)
-            path = os.path.join(path, 'src')
-            del sys.modules['Editra']
-            shutil.rmtree(path)
-    except (ImportError, OSError):
-        pass
-    except:
-        sys.stderr.write("[ERROR] Failed to remove old source files")
-
     # Make sure to delete any existing MANIFEST file beforehand to
     # prevent stale file lists
     if os.path.exists('MANIFEST'):
