diff -Nur kopete-otr-0.2.orig/src/Makefile.am kopete-otr-0.2/src/Makefile.am
--- kopete-otr-0.2.orig/src/Makefile.am	2007-04-27 11:38:28.000000000 +0200
+++ kopete-otr-0.2/src/Makefile.am	2007-04-27 11:39:02.000000000 +0200
@@ -38,6 +38,7 @@
 lib_LTLIBRARIES = libkotr.la
 libkotr_la_LDFLAGS = $(all_libraries)
 libkotr_la_SOURCES = otrlchatinterface.cpp otrlconfinterface.cpp
+libkotr_la_LIBADD = -lotr
 
 SUBDIRS = pics
 
