diff -Naur konq-plugins-4.4.0/adblock/adblock.cpp konq-plugins-4.4.0-new/adblock/adblock.cpp
--- konq-plugins-4.4.0/adblock/adblock.cpp	2010-02-12 00:12:41.000000000 +0200
+++ konq-plugins-4.4.0-new/adblock/adblock.cpp	2010-10-07 00:29:41.808999905 +0300
@@ -300,7 +300,7 @@
 // ----------------------------------------------------------------------------
 
 AdElement::AdElement() :
-  m_url(0), m_category(0), m_type(0), m_blocked(false) {}
+  m_blocked(false) {}
 
 AdElement::AdElement(const QString &url, const QString &category,
 		     const QString &type, bool blocked, const DOM::Node&node) :
