--- gprolog-1.2.19.orig/src/Ma2Asm/ix86_any.c	2005-06-14 03:40:15.000000000 +1200
+++ gprolog-1.2.19/src/Ma2Asm/ix86_any.c	2006-03-05 09:26:21.000000000 +1300
@@ -164,6 +164,9 @@
 void
 Asm_Stop(void)
 {
+  Label_Printf("#ifdef __ELF__");
+  Label_Printf(".section .note.GNU-stack,\"\",@progbits");
+  Label_Printf("#endif");
 }
 
 
--- gprolog-1.2.19.orig/src/Ma2Asm/powerpc_any.c	2005-06-14 03:40:22.000000000 +1200
+++ gprolog-1.2.19/src/Ma2Asm/powerpc_any.c	2006-03-05 09:40:34.000000000 +1300
@@ -161,6 +161,9 @@
 void
 Asm_Stop(void)
 {
+  Label_Printf("#ifdef __ELF__");
+  Label_Printf(".section .note.GNU-stack,\"\",@progbits");
+  Label_Printf("#endif");
 }
 
 
