diff -Naur src-orig/BipsPl/stream_supp.h src/BipsPl/stream_supp.h
--- src-orig/BipsPl/stream_supp.h	2005-10-30 22:04:53.000000000 +0100
+++ src/BipsPl/stream_supp.h	2005-10-30 22:20:42.000000000 +0100
@@ -276,9 +276,9 @@
 extern int atom_debugger_input;
 extern int atom_debugger_output;
 
-extern int atom_read;
-extern int atom_write;
-extern int atom_append;
+static int atom_read;
+static int atom_write;
+static int atom_append;
 
 extern int atom_reposition;
 
@@ -287,7 +287,7 @@
 extern int atom_text;
 extern int atom_binary;
 
-extern int atom_error;
+static int atom_error;
 extern int atom_eof_code;
 extern int atom_reset;
 
diff -Naur src-orig/EnginePl/obj_chain.h src/EnginePl/obj_chain.h
--- src-orig/EnginePl/obj_chain.h	2005-10-30 22:04:53.000000000 +0100
+++ src/EnginePl/obj_chain.h	2005-10-30 22:17:31.000000000 +0100
@@ -76,7 +76,7 @@
 
 #ifndef _MSC_VER
 
-extern ObjChain *obj_chain_stop;
+static ObjChain *obj_chain_stop;
 
 static ObjChain obj_chain_start =
   { OBJ_CHAIN_MAGIC_1, OBJ_CHAIN_MAGIC_2, &obj_chain_stop, OBJ_INIT };
