--- fetchmail-6.2.5.orig/configure.in	2004-02-22 10:23:16.711353561 +0000
+++ fetchmail-6.2.5/configure.in	2004-02-22 10:26:37.916800464 +0000
@@ -465,15 +465,13 @@
         else
           AC_MSG_RESULT([found])
           ac_krblibs="-lcom_err"
-           AC_CHECK_LIB(crypto,
-             krb5_des_string_to_key,
-             libk5crypto=-lcrypto,
-                AC_CHECK_LIB(k5crypto,
-                   krb5_des_string_to_key,
-                   libk5crypto=-lk5crypto,
-                      AC_ERROR([Kerberos 5 DES libraries not found]),
-                   ${LDEFLAGS} ${ac_krblibs}),
-                 ${LDEFLAGS} ${ac_krblibs})
+           AC_CHECK_LIB(crypto, krb5_des_string_to_key, libk5crypto=-lcrypto,
+             AC_CHECK_LIB(k5crypto, krb5_des_string_to_key, libk5crypto=-lk5crypto,
+               AC_CHECK_LIB(k5crypto, krb5int_des_string_to_key, libk5crypto=-lk5crypto,
+                 AC_ERROR([Kerberos 5 DES libraries not found]),
+               ${LDEFLAGS} ${ac_krblibs}),
+             ${LDEFLAGS} ${ac_krblibs}),
+           ${LDEFLAGS} ${ac_krblibs})
         fi
         AC_CHECK_LIB(krb5, krb5_init_context,[],
            continue,
