Index: kasumi-2.0/KasumiException.cxx
===================================================================
--- kasumi-2.0.orig/KasumiException.cxx
+++ kasumi-2.0/KasumiException.cxx
@@ -47,7 +47,7 @@ void handleException(KasumiException e){
                                     (GtkDialogFlags)0,
                                     GTK_MESSAGE_ERROR,
                                     GTK_BUTTONS_CLOSE,
-                                    e.getMessage().c_str());
+                                    "%s", e.getMessage().c_str());
     gtk_dialog_run(GTK_DIALOG (dialog));
     gtk_widget_destroy(dialog);
     break;
@@ -56,7 +56,7 @@ void handleException(KasumiException e){
                                     (GtkDialogFlags)0,
                                     GTK_MESSAGE_WARNING,
                                     GTK_BUTTONS_OK,
-                                    e.getMessage().c_str());
+                                    "%s", e.getMessage().c_str());
     gtk_dialog_run(GTK_DIALOG (dialog));
     gtk_widget_destroy(dialog);
     break;
