--- configure.in.orig	2004-04-20 16:28:13.000000000 -0700
+++ configure.in	2004-05-04 20:21:45.418353000 -0700
@@ -149,8 +149,7 @@
 
 JACK_CORE_CFLAGS="-I\$(top_builddir)/config -I\$(top_srcdir) \
 -I\$(top_builddir) -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -Wall"
-JACK_OPT_CFLAGS="$JACK_CORE_CFLAGS -g -march=pentium2 -mcpu=pentium4 -O3 \
--ffast-math -funroll-loops -fprefetch-loop-arrays"
+JACK_OPT_CFLAGS="$JACK_CORE_CFLAGS -g -O3 -ffast-math -funroll-loops -fprefetch-loop-arrays"
 JACK_CFLAGS="$JACK_CORE_CFLAGS $CFLAGS"
 
 dnl
@@ -514,10 +513,26 @@
 fi
 AC_SUBST(READLINE_DEPS)
 
-# you need doxygen to make dist.
-AC_CHECK_PROG(HAVE_DOXYGEN, doxygen, true, false)
-if test $HAVE_DOXYGEN = "false"; then
-             AC_MSG_WARN([*** doxygen not found, docs will not be built])
+AC_ARG_ENABLE(html-docs,
+	[  --enable-html-docs    build the html documentation (default yes)],
+        [ if test "x$enable_html_docs" != "xno" ; then
+                ENABLE_HTML_DOCS="yes"
+		AC_MSG_RESULT([JACK will build html docs])
+	else
+		AC_MSG_RESULT([JACK will not build html docs])
+	fi;
+        ], 
+	[ ENABLE_HTML_DOCS="yes"; ]
+)
+
+if test "$ENABLE_HTML_DOCS" == "yes" ; then
+	# you need doxygen to make dist.
+	AC_CHECK_PROG(HAVE_DOXYGEN, doxygen, true, false)
+	if test $HAVE_DOXYGEN == "false"; then
+		AC_MSG_ERROR([*** doxygen not found, unable to build docs.])
+	fi
+else
+	HAVE_DOXYGEN="false"
 fi
 
 AM_CONDITIONAL(HAVE_SNDFILE, $HAVE_SNDFILE)
