diff -Nru3p xmms-1.2.8.orig/xmms/dock.c xmms-1.2.8/xmms/dock.c
--- xmms-1.2.8.orig/xmms/dock.c	2001-03-08 14:33:30.000000000 +0000
+++ xmms-1.2.8/xmms/dock.c	2001-03-08 14:33:30.000000000 +0000
@@ -363,8 +363,8 @@ void dock_shade(GList *window_list, GtkW
 	}
 	g_list_free(slist);
 	free_docked_list(docked_list);
-	dock_set_uposition(widget, x, y + off_y - (new_h - h));
 	dock_window_resize(widget, w, new_h, w, h);
+	dock_set_uposition(widget, x, y + off_y - (new_h - h));
 }
 
 static GList *resize_move_list(GList *list, GtkWidget *widget, gint offset_x, gint offset_y)
@@ -523,8 +523,8 @@ void dock_resize(GList *window_list, Gtk
 	}
 		   
 	
-	dock_set_uposition(widget, x + off_x, y + off_y);
 	dock_window_resize(widget, new_w, new_h, w, h);
+	dock_set_uposition(widget, x + off_x, y + off_y);
 }
 	
 void dock_move_press(GList *window_list, GtkWidget *w, GdkEventButton *event, gboolean move_list)
diff -Nru3p xmms-1.2.8.orig/xmms/playlistwin.c xmms-1.2.8/xmms/playlistwin.c
--- xmms-1.2.8.orig/xmms/playlistwin.c	2003-04-04 17:42:51.000000000 +0000
+++ xmms-1.2.8/xmms/playlistwin.c	2003-04-04 17:42:51.000000000 +0000
@@ -340,10 +340,9 @@ void playlistwin_create_mask(void)
 void playlistwin_set_shade_menu_cb(gboolean shaded)
 {
 	cfg.playlist_shaded = shaded;
+	gtk_window_set_default_size(GTK_WINDOW(playlistwin), cfg.playlist_width, PLAYLIST_HEIGHT);
 	if (!cfg.show_wm_decorations)
 		dock_shade(dock_window_list, playlistwin, PLAYLIST_HEIGHT);
-	else 
-		gtk_window_set_default_size(GTK_WINDOW(playlistwin), cfg.playlist_width, PLAYLIST_HEIGHT);
 
 	if (shaded)
 	{
@@ -1283,7 +1282,7 @@ static gboolean playlistwin_configure(Gt
 	if (cfg.show_wm_decorations || hint_move_resize_available())
 	{
 		if (event->width != cfg.playlist_width ||
-		    event->height != cfg.playlist_height)
+		    event->height != PLAYLIST_HEIGHT)
 			playlistwin_queue_resize(event->width, event->height);
 	}
 	
