Index: sys/boot/alpha/common/Makefile.common
===================================================================
--- sys/boot/alpha/common/Makefile.common.orig
+++ sys/boot/alpha/common/Makefile.common
@@ -34,7 +34,7 @@ CLEANFILES+=	vers.c ${BOOT} ${BOOT}.sym
 
 CFLAGS+=	-Wall
 
-CFLAGS+=	-I${LIBSTANDDIR}
+CFLAGS+=	-I/usr/include/libstand
 CFLAGS+=	-I${.CURDIR}/..
 
 vers.c: ${.CURDIR}/../../common/newvers.sh ${.CURDIR}/version
Index: sys/boot/alpha/libalpha/Makefile
===================================================================
--- sys/boot/alpha/libalpha/Makefile.orig
+++ sys/boot/alpha/libalpha/Makefile
@@ -5,7 +5,7 @@ INTERNALLIB=
 
 #CFLAGS+=	-DDISK_DEBUG
 
-CFLAGS+=	-I${LIBSTANDDIR}
+CFLAGS+=	-I/usr/include/libstand
 
 # Pick up the bootstrap header for some interface items
 CFLAGS+=	-I${.CURDIR}/../../common
Index: sys/boot/arc/lib/Makefile
===================================================================
--- sys/boot/arc/lib/Makefile.orig
+++ sys/boot/arc/lib/Makefile
@@ -7,7 +7,7 @@ CFLAGS+=	-ffreestanding
 .PATH:		${.CURDIR}/arch/${MACHINE_ARCH}
 # XXX hack to pick up stand.h
 LIBSTANDDIR=	${.CURDIR}/../../../../lib/libstand
-CFLAGS+=	-I${LIBSTANDDIR}
+CFLAGS+=	-I/usr/include/libstand
 CFLAGS+=	-DDEBUG
 
 # Pick up the bootstrap header for some interface items
Index: sys/boot/arc/loader/Makefile
===================================================================
--- sys/boot/arc/loader/Makefile.orig
+++ sys/boot/arc/loader/Makefile
@@ -26,7 +26,7 @@ CLEANFILES+=	machine
 
 CFLAGS+=	-Wall
 
-CFLAGS+=	-I${LIBSTANDDIR}
+CFLAGS+=	-I/usr/include/libstand
 CFLAGS+=	-I${.CURDIR}/../include
 CRT=		start.o
 STRIP=
Index: sys/boot/efi/libefi/Makefile
===================================================================
--- sys/boot/efi/libefi/Makefile.orig
+++ sys/boot/efi/libefi/Makefile
@@ -14,7 +14,7 @@ SRCS+=	efifpswa.c pal.S
 
 CFLAGS+= -I${.CURDIR}/../include
 CFLAGS+= -I${.CURDIR}/../include/${MACHINE_ARCH}
-CFLAGS+= -I${.CURDIR}/../../../../lib/libstand
+CFLAGS+= -I/usr/include/libstand
 
 # Pick up the bootstrap header for some interface items
 CFLAGS+= -I${.CURDIR}/../../common
Index: sys/boot/i386/boot2/Makefile
===================================================================
--- sys/boot/i386/boot2/Makefile.orig
+++ sys/boot/i386/boot2/Makefile
@@ -24,7 +24,7 @@ CFLAGS=	-Os \
 	-fno-guess-branch-probability \
 	-fomit-frame-pointer \
 	-fno-unit-at-a-time \
-	-mno-align-long-strings \
+	-march=i386 \
 	-mrtd \
 	-mno-mmx -mno-3dnow -mno-sse -mno-sse2 \
 	-D${BOOT2_UFS} \
Index: sys/boot/i386/libi386/Makefile
===================================================================
--- sys/boot/i386/libi386/Makefile.orig
+++ sys/boot/i386/libi386/Makefile
@@ -28,7 +28,7 @@ CFLAGS+=	-I${.CURDIR}/../../common -I${.
 		-I${.CURDIR}/../../../contrib/dev/acpica \
 		-I${.CURDIR}/../../.. -I.
 # the location of libstand
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand/
+CFLAGS+=	-I/usr/include/libstand/
 
 .if ${MACHINE_ARCH} == "amd64"
 CLEANFILES+=	machine
Index: sys/boot/ofw/libofw/Makefile
===================================================================
--- sys/boot/ofw/libofw/Makefile.orig
+++ sys/boot/ofw/libofw/Makefile
@@ -7,7 +7,7 @@ SRCS=	devicename.c elf_freebsd.c ofw_con
 	ofw_memory.c ofw_module.c ofw_net.c ofw_reboot.c \
 	ofw_time.c openfirm.c
 
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand/
+CFLAGS+=	-I/usr/include/libstand/
 
 # Pick up the bootstrap header for some interface items
 CFLAGS+=	-I${.CURDIR}/../../common -I${.CURDIR}/../../.. -I.
Index: sys/boot/pc98/libpc98/Makefile
===================================================================
--- sys/boot/pc98/libpc98/Makefile.orig
+++ sys/boot/pc98/libpc98/Makefile
@@ -28,6 +28,6 @@ CFLAGS+=	-I${.CURDIR}/../../common -I${.
 		-I${.CURDIR}/../../i386/libi386 \
 		-I${.CURDIR}/../../.. -I.
 # the location of libstand
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand/
+CFLAGS+=	-I/usr/include/libstand/
 
 .include <bsd.lib.mk>
Index: sys/boot/sparc64/loader/Makefile
===================================================================
--- sys/boot/sparc64/loader/Makefile.orig
+++ sys/boot/sparc64/loader/Makefile
@@ -63,7 +63,7 @@ LIBOFW=		${.OBJDIR}/../../ofw/libofw/lib
 CFLAGS+=	-I${.CURDIR}/../../ofw/libofw/
 
 # where to get libstand from
-CFLAGS+=	-I${.CURDIR}/../../../../lib/libstand/
+CFLAGS+=	-I/usr/include/libstand/
 
 DPADD=		${LIBFICL} ${LIBOFW} ${LIBSTAND}
 LDADD=		${LIBFICL} ${LIBOFW} -lstand
