diff -Naur kdenetwork-3.5.9.orig/kopete/protocols/oscar/oscarversionupdater.cpp kdenetwork-3.5.9/kopete/protocols/oscar/oscarversionupdater.cpp
--- kdenetwork-3.5.9.orig/kopete/protocols/oscar/oscarversionupdater.cpp	2008-02-14 01:11:52.000000000 +0100
+++ kdenetwork-3.5.9/kopete/protocols/oscar/oscarversionupdater.cpp	2008-07-01 20:06:00.000000000 +0200
@@ -89,13 +89,13 @@
 	kdDebug(OSCAR_RAW_DEBUG) << k_funcinfo << endl;
 	KConfigGroup config( KGlobal::config(), "ICQVersion" );
 	
-	mICQVersion.clientString = config.readEntry( "ClientString", "ICQBasic" );
+	mICQVersion.clientString = config.readEntry( "ClientString", "ICQ Client" );
 	mICQVersion.clientId = config.readEntry( "ClientId", "0x010A" ).toUShort( 0, 0 );
-	mICQVersion.major = config.readEntry( "Major", "0x0014" ).toUShort( 0, 0 );
-	mICQVersion.minor = config.readEntry( "Minor", "0x0034" ).toUShort( 0, 0 );
+	mICQVersion.major = config.readEntry( "Major", "0x0006" ).toUShort( 0, 0 );
+	mICQVersion.minor = config.readEntry( "Minor", "0x0000" ).toUShort( 0, 0 );
 	mICQVersion.point = config.readEntry( "Point", "0x0000" ).toUShort( 0, 0 );
-	mICQVersion.build = config.readEntry( "Build", "0x0BB8" ).toUShort( 0, 0 );
-	mICQVersion.other = config.readEntry( "Other", "0x0000043D" ).toUInt( 0, 0 );
+	mICQVersion.build = config.readEntry( "Build", "0x17AB" ).toUShort( 0, 0 );
+	mICQVersion.other = config.readEntry( "Other", "0x00007535" ).toUInt( 0, 0 );
 	mICQVersion.country = config.readEntry( "Country", "us" );
 	mICQVersion.lang = config.readEntry( "Lang", "en" );
 }
