--- kopete/kopete/chatwindow/kopetechatwindow.cpp	2007-11-12 04:11:29.000000000 +0100
+++ kopete/kopete/chatwindow/kopetechatwindow.cpp	2007-11-12 04:14:24.000000000 +0100
@@ -442,7 +442,11 @@
 void KopeteChatWindow::slotStopAnimation( ChatView* view )
 {
 	if( view == m_activeView )
+	{
 		anim->setPixmap( normalIcon );
+		if( !animIcon.isNull() )
+			animIcon.pause();
+	}
 }
 
 void KopeteChatWindow::slotUpdateSendEnabled()
