--- Makefile
+++ Makefile
@@ -12,11 +12,9 @@
 CXX      = g++
 LEX      = flex
 YACC     = yacc
-CFLAGS   = -pipe -Wall -W -O2 -fomit-frame-pointer -pipe -march=i586 -mcpu=pentiumpro -D_REENTRANT  -DQT_NO_DEBUG -DQT_THREAD_SUPPORT
-CXXFLAGS = -pipe -Wall -W -O2 -fomit-frame-pointer -pipe -march=i586 -mcpu=pentiumpro -D_REENTRANT  -DQT_NO_DEBUG -DQT_THREAD_SUPPORT
 LEXFLAGS = 
 YACCFLAGS= -d
-INCPATH  = -I$(QTDIR)/mkspecs/default -I. -I$(QTDIR)/include
+INCPATH  = -I$(QTDIR)/mkspecs/linux-g++ -I. -I$(QTDIR)/include
 LINK     = g++
 LFLAGS   = 
 LIBS     = $(SUBLIBS) -L$(QTDIR)/lib -L/usr/X11R6/lib -lqt-mt -lXext -lX11 -lm -lpthread
@@ -115,10 +113,7 @@
 
 mocables: $(SRCMOC)
 
-$(MOC): 
-	( cd $(QTDIR)/src/moc ; $(MAKE) )
-
-Makefile: qtvba.pro  $(QTDIR)/mkspecs/default/qmake.conf 
+Makefile: qtvba.pro  $(QTDIR)/mkspecs/linux-g++/qmake.conf 
 	$(QMAKE) -o Makefile qtvba.pro
 qmake: 
 	@$(QMAKE) -o Makefile qtvba.pro
