diff --exclude-from=/home/dang/.diffrc -up -ruN linux-2.6.18-usermode-r1.orig/arch/i386/Kconfig.cpu linux-2.6.18-usermode-r1/arch/i386/Kconfig.cpu
--- linux-2.6.18-usermode-r1.orig/arch/i386/Kconfig.cpu	2007-01-02 21:03:00.000000000 -0500
+++ linux-2.6.18-usermode-r1/arch/i386/Kconfig.cpu	2007-01-02 21:54:14.000000000 -0500
@@ -7,7 +7,6 @@ choice
 
 config M386
 	bool "386"
-	depends on !UML
 	---help---
 	  This is the processor type of your CPU. This information is used for
 	  optimizing purposes. In order to compile a kernel that can run on
@@ -302,7 +301,7 @@ config X86_USE_PPRO_CHECKSUM
 
 config X86_USE_3DNOW
 	bool
-	depends on (MCYRIXIII || MK7 || MGEODE_LX) && !UML
+	depends on MCYRIXIII || MK7 || MGEODE_LX
 	default y
 
 config X86_OOSTORE
diff --exclude-from=/home/dang/.diffrc -up -ruN linux-2.6.18-usermode-r1.orig/arch/um/include/kern_util.h linux-2.6.18-usermode-r1/arch/um/include/kern_util.h
--- linux-2.6.18-usermode-r1.orig/arch/um/include/kern_util.h	2007-01-02 21:03:00.000000000 -0500
+++ linux-2.6.18-usermode-r1/arch/um/include/kern_util.h	2007-01-02 21:54:14.000000000 -0500
@@ -6,6 +6,7 @@
 #ifndef __KERN_UTIL_H__
 #define __KERN_UTIL_H__
 
+#include "linux/threads.h"
 #include "sysdep/ptrace.h"
 #include "sysdep/faultinfo.h"
 
diff --exclude-from=/home/dang/.diffrc -up -ruN linux-2.6.18-usermode-r1.orig/arch/um/os-Linux/sys-i386/tls.c linux-2.6.18-usermode-r1/arch/um/os-Linux/sys-i386/tls.c
--- linux-2.6.18-usermode-r1.orig/arch/um/os-Linux/sys-i386/tls.c	2007-01-02 21:03:00.000000000 -0500
+++ linux-2.6.18-usermode-r1/arch/um/os-Linux/sys-i386/tls.c	2007-01-02 21:54:14.000000000 -0500
@@ -1,6 +1,4 @@
 #include <errno.h>
-#include <sys/syscall.h>
-#include <unistd.h>
 #include <linux/unistd.h>
 #include "sysdep/tls.h"
 #include "user_util.h"
diff --exclude-from=/home/dang/.diffrc -up -ruN linux-2.6.18-usermode-r1.orig/arch/um/os-Linux/tls.c linux-2.6.18-usermode-r1/arch/um/os-Linux/tls.c
--- linux-2.6.18-usermode-r1.orig/arch/um/os-Linux/tls.c	2007-01-02 21:03:00.000000000 -0500
+++ linux-2.6.18-usermode-r1/arch/um/os-Linux/tls.c	2007-01-02 21:54:14.000000000 -0500
@@ -1,8 +1,6 @@
 #include <errno.h>
 #include <sys/ptrace.h>
-#include <sys/syscall.h>
 #include <asm/ldt.h>
-#include <unistd.h>
 #include "sysdep/tls.h"
 #include "uml-config.h"
 
diff --exclude-from=/home/dang/.diffrc -up -ruN linux-2.6.18-usermode-r1.orig/arch/um/sys-x86_64/stub_segv.c linux-2.6.18-usermode-r1/arch/um/sys-x86_64/stub_segv.c
--- linux-2.6.18-usermode-r1.orig/arch/um/sys-x86_64/stub_segv.c	2007-01-02 21:03:00.000000000 -0500
+++ linux-2.6.18-usermode-r1/arch/um/sys-x86_64/stub_segv.c	2007-01-02 21:54:14.000000000 -0500
@@ -5,6 +5,7 @@
 
 #include <stddef.h>
 #include <signal.h>
+#include <linux/compiler.h>
 #include <asm/unistd.h>
 #include "uml-config.h"
 #include "sysdep/sigcontext.h"
