diff -uNr asterisk-1.6.1.20.ORIG//main/channel.c asterisk-1.6.1.20/main/channel.c
--- asterisk-1.6.1.20.ORIG//main/channel.c	2010-06-07 12:13:50.000000000 +0100
+++ asterisk-1.6.1.20/main/channel.c	2010-06-07 12:14:25.000000000 +0100
@@ -1987,6 +1987,8 @@
 			chan->generator->release(chan, chan->generatordata);
 		chan->generatordata = NULL;
 	}
+
+	ast_prod(chan);
 	if (gen->alloc && !(chan->generatordata = gen->alloc(chan, params))) {
 		res = -1;
 	}
@@ -1996,8 +1998,6 @@
 	}
 	ast_channel_unlock(chan);
 
-	ast_prod(chan);
-
 	return res;
 }
 
