diff -uNr libgnomeprintui-1.116.0/libgnomeprintui/gnome-print-dialog.h libgnomeprintui-1.116.0.patched/libgnomeprintui/gnome-print-dialog.h
--- libgnomeprintui-1.116.0/libgnomeprintui/gnome-print-dialog.h	2002-05-25 20:47:08.000000000 +0200
+++ libgnomeprintui-1.116.0.patched/libgnomeprintui/gnome-print-dialog.h	2003-09-21 20:22:41.000000000 +0200
@@ -60,7 +60,7 @@
 	GNOME_PRINT_RANGE_ALL = (1 << 1),
 	GNOME_PRINT_RANGE_RANGE = (1 << 2),
 	GNOME_PRINT_RANGE_SELECTION = (1 << 3),
-	GNOME_PRINT_RANGE_SELECTION_UNSENSITIVE  = (1 << 4),
+	GNOME_PRINT_RANGE_SELECTION_UNSENSITIVE  = (1 << 4)
 };
 
 /*
