diff -ruN mol-0.9.69_pre6.orig/configure.in mol-0.9.69_pre6/configure.in
--- mol-0.9.69_pre6.orig/configure.in	2003-11-18 22:35:10.000000000 +0100
+++ mol-0.9.69_pre6/configure.in	2003-11-18 23:02:40.000000000 +0100
@@ -96,30 +96,30 @@
 case "$host_cpu-$host_os" in 
      i?86-*linux*)
 	Arch=Linux
-        ARCH=i386-linux ;;
+        ARCH_=i386-linux ;;
      powerpc-*linux* | ppc-*linux*)
 	Arch=Linux
-        ARCH=ppc-linux
+        ARCH_=ppc-linux
         AC_DEFINE(NATIVE_PPC, 1, [Define if native PPC] ) ;;
      powerpc-*darwin*)
 	Arch=Darwin
 	AC_DEFINE(__darwin__, 1, [Define if running Darwin] ) 
-        ARCH=darwin ;;
+        ARCH_=darwin ;;
      *)
         AC_MSG_ERROR([Unsupported architecture $host_cpu-$host_os ]) ;;
 esac
 
 case "$host_os" in *linux*) OS=linux ;; esac
 
-AM_CONDITIONAL(ARCH_386_LINUX, test $ARCH = i386-linux )
-AM_CONDITIONAL(ARCH_PPC_LINUX, test $ARCH = ppc-linux )
-AM_CONDITIONAL(ARCH_DARWIN, test $ARCH = darwin )
+AM_CONDITIONAL(ARCH_386_LINUX, test $ARCH_ = i386-linux )
+AM_CONDITIONAL(ARCH_PPC_LINUX, test $ARCH_ = ppc-linux )
+AM_CONDITIONAL(ARCH_DARWIN, test $ARCH_ = darwin )
 AM_CONDITIONAL(CPU_386, test $CPU = i386 )
 AM_CONDITIONAL(CPU_PPC, test $CPU = ppc )
 AM_CONDITIONAL(LINUX, test "$OS" = linux )
 
 AC_SUBST(Arch)
-AC_SUBST(ARCH)
+AC_SUBST(ARCH_)
 AC_SUBST(CPU)
 
 
@@ -206,7 +206,7 @@
 ################################################################################
 
 AC_ARG_ENABLE(oldworld, [  --disable-oldworld      don not include oldworld support], [], [])
-test $ARCH = darwin && enable_oldworld=no
+test $ARCH_ = darwin && enable_oldworld=no
 
 AM_CONDITIONAL(OLDWORLD_SUPPORT, test "$enable_oldworld" != no )
 test "$enable_oldworld" != no && {
@@ -231,7 +231,7 @@
 #}
 
 AC_ARG_ENABLE(esound, [  --disable-alsa          disable ALSA sound driver support], [], [])
-test $ARCH = darwin && enable_alsa=no
+test $ARCH_ = darwin && enable_alsa=no
 
 test "$enable_alsa" != no && {
      AC_CHECK_LIB(asound, snd_pcm_open, [LIB_ALSA=-lasound], [enable_alsa=no])
@@ -510,7 +510,7 @@
 
 CFLAGS="$CFLAGS -Wall -Wmissing-prototypes "'$(UNIMPORTANT_CFLAGS)'
 #CFLAGS="$CFLAGS -fno-strict-aliasing"
-test $ARCH = darwin && CFLAGS="$CFLAGS -traditional-cpp"
+test $ARCH_ = darwin && CFLAGS="$CFLAGS -traditional-cpp"
 
 ASMFLAGS=
 AC_SUBST(ASMFLAGS)
