--- Makefile.in
+++ Makefile.in
@@ -58,6 +58,10 @@
 inst_libdir:=$(libdir)
 inst_docdir:=$(prefix)/doc/mingw-runtime
 endif
+inst_bindir:=$(DESTDIR)$(inst_bindir)
+inst_includedir:=$(DESTDIR)$(inst_includedir)
+inst_libdir:=$(DESTDIR)$(inst_libdir)
+inst_docdir:=$(DESTDIR)$(inst_docdir)
 
 # The Mingw headers are installed under a subdirectory of 
 # $(tooldir)/include when configuring in Cygwin.
@@ -199,6 +199,7 @@
 all_dlls_host: $(DLLS)
 
 install_dlls_host:
+	$(mkinstalldirs) $(inst_bindir)
 	for i in $(DLLS); do \
 		$(INSTALL_PROGRAM) $$i $(inst_bindir)/$$i ; \
 	done
@@ -401,12 +401,14 @@
 	$(mkinstalldirs) $(inst_docdir)
 
 install: all install-dirs $(install_dlls_host)
+	$(mkinstalldirs) $(inst_libdir)
 	for i in $(LIBS); do \
 		$(INSTALL_DATA) $$i $(inst_libdir)/$$i ; \
 	done
 	for i in $(CRT0S); do \
 		$(INSTALL_DATA) $$i $(inst_libdir)/$$i ; \
 	done
+	$(mkinstalldirs) $(inst_docdir)
 	for i in $(INSTDOCS); do \
 		$(INSTALL_DATA) $(srcdir)/$$i $(inst_docdir)/$$i ; \
 	done
