diff --git a/src/OSspecific/Unix/signals/sigFpe.C b/src/OSspecific/Unix/signals/sigFpe.C
index e3bcf15..759f8c8 100644
--- a/src/OSspecific/Unix/signals/sigFpe.C
+++ b/src/OSspecific/Unix/signals/sigFpe.C
@@ -39,6 +39,7 @@ License
 
 #   include <fenv.h>
 #   include <malloc.h>
+#   include <stdint.h>
 
 #elif defined(sgiN32) || defined(sgiN32Gcc)
 
diff --git a/src/meshTools/triSurface/surfaceFeatures/surfaceFeatures.C b/src/meshTools/triSurface/surfaceFeatures/surfaceFeatures.C
index 9f858b9..25f2641 100644
--- a/src/meshTools/triSurface/surfaceFeatures/surfaceFeatures.C
+++ b/src/meshTools/triSurface/surfaceFeatures/surfaceFeatures.C
@@ -763,10 +763,11 @@ Foam::Map<Foam::label> Foam::surfaceFeatures::nearestSamples
     const scalarField& maxDist
 ) const
 {
+    treeBoundBox tbb(samples);
     // Build tree out of all samples.
     octree<octreeDataPoint> ppTree
     (
-        treeBoundBox(samples),      // overall search domain
+        tbb,      // overall search domain
         octreeDataPoint(samples),   // all information needed to do checks
         1,          // min levels
         20.0,       // maximum ratio of cubes v.s. cells
@@ -864,10 +865,11 @@ Foam::Map<Foam::label> Foam::surfaceFeatures::nearestSamples
     scalar maxSearch = max(maxDist);
     vector span(maxSearch, maxSearch, maxSearch);
 
+    treeBoundBox tbb(samples);
     // octree.shapes holds reference!
     octree<octreeDataPoint> ppTree
     (
-        treeBoundBox(samples),      // overall search domain
+        tbb,      // overall search domain
         octreeDataPoint(samples),   // all information needed to do checks
         1,          // min levels
         20.0,       // maximum ratio of cubes v.s. cells
