diff -ur XSB.orig/packages/dbdrivers/mysql/cc/mysql_driver_defs.h XSB/packages/dbdrivers/mysql/cc/mysql_driver_defs.h
--- XSB.orig/packages/dbdrivers/mysql/cc/mysql_driver_defs.h	2006-02-12 18:21:03.000000000 +1300
+++ XSB/packages/dbdrivers/mysql/cc/mysql_driver_defs.h	2007-01-28 12:01:01.000000000 +1300
@@ -35,7 +35,7 @@
 #include "driver_manager_defs.h"
 
 #define MAX_HANDLES 25
-#define MAX_QUERIES 25
+#define MAX_QUERIES 100
 
 /****** data structures for handles *****/
 
diff -ur XSB.orig/packages/dbdrivers/mysql/configure XSB/packages/dbdrivers/mysql/configure
--- XSB.orig/packages/dbdrivers/mysql/configure	2003-09-29 21:40:02.000000000 +1200
+++ XSB/packages/dbdrivers/mysql/configure	2007-01-28 12:01:01.000000000 +1300
@@ -564,10 +564,10 @@
 
 
 if test -z "$with_config"; then
-echo '
-- You did not tell me what kind of host system you want to configure.
-- I will attempt to guess the kind of system this is.
-' 1>&6
+#echo '
+#- You did not tell me what kind of host system you want to configure.
+#- I will attempt to guess the kind of system this is.
+#' 1>&6
     
 # Make sure we can run config.sub.
 if ${CONFIG_SHELL-/bin/sh} $ac_config_sub sun4 >/dev/null 2>&1; then :
@@ -626,8 +626,8 @@
    LDFLAGS="-L${libdir} ${LDFLAGS}"
 fi
 
-echo $ac_n "checking for  in -lmysqlclient""... $ac_c" 1>&6
-echo "configure:631: checking for  in -lmysqlclient" >&5
+echo $ac_n "checking for mysql_init in -lmysqlclient""... $ac_c" 1>&6
+echo "configure:631: checking for mysql_init in -lmysqlclient" >&5
 ac_lib_var=`echo mysqlclient'_' | sed 'y%./+-%__p_%'`
 if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
@@ -640,10 +640,10 @@
 /* Override any gcc2 internal prototype to avoid an error.  */
 /* We use char because int might match the return type of a gcc2
     builtin and then its argument prototype would still apply.  */
-char ();
+char mysql_init ();
 
 int main() {
-()
+return mysql_init();
 ; return 0; }
 EOF
 if { (eval echo configure:650: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
@@ -661,9 +661,9 @@
 fi
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  mysql_error=yes
 else
   echo "$ac_t""no" 1>&6
+  mysql_error=yes
 fi
 
 # Here we provide additional directories to check headers in
@@ -808,10 +808,8 @@
 support=yes
 fi
 
-
-
-
-
+CFLAGS=`mysql_config --cflags`
+LDFLAGS=`mysql_config --libs`
 
 
 
diff -ur XSB.orig/packages/dbdrivers/mysql/mysql_driver_config.P XSB/packages/dbdrivers/mysql/mysql_driver_config.P
--- XSB.orig/packages/dbdrivers/mysql/mysql_driver_config.P	2006-02-12 18:21:04.000000000 +1300
+++ XSB/packages/dbdrivers/mysql/mysql_driver_config.P	2007-01-28 21:23:55.000000000 +1300
@@ -43,7 +43,7 @@
 	%% Creates something like -Wl,-rpath,<libdir>
 	runtime_loader_flag(XCC, MysqlLibdir, MysqlRunpathFlag),
 	fmt_write_string(LD_flags,
-			 ' %s %s -lmysqlclient -ldl %s%sdriver_manager.so %s%smysql_driver.so', f(MysqlRunpathFlag, MysqlLibdir_flag, ConfigLibdir, Slash, ConfigLibdir, Slash)),
+			 ' %s %s -lmysqlclient -ldl driver_manager.so', f(MysqlRunpathFlag, MysqlLibdir_flag)),
 	package_configuration(dir(mysql_driver), MySQLDriverDir),
 	fmt_write_string(LD_directives_file,
 			 '%s%scc%smysql_driver.H', f(MySQLDriverDir, Slash, Slash)),
