diff -ur XSB.orig/packages/xpath/cc/xpathparser.c XSB/packages/xpath/cc/xpathparser.c
--- XSB.orig/packages/xpath/cc/xpathparser.c	2005-03-04 07:37:19.000000000 +1300
+++ XSB/packages/xpath/cc/xpathparser.c	2007-01-27 12:03:02.000000000 +1300
@@ -157,7 +157,7 @@
       namespace = p2c_string( ns_term);
     }
   /*This is the function which evaluates the xpath expression on xml input*/
-  ret = execute_xpath_expression( source, xpath_expr, namespace, output_term, flag);
+  ret = execute_xpath_expression( source, (xmlChar*)xpath_expr, (xmlChar*)namespace, output_term, flag);
   if( ret == FALSE){
     return xpath_error(ERR_MISC, "xpath", "Unable to parse the xpath expression");
   }
@@ -230,7 +230,7 @@
   for( i = 0; i < size; i++){
     buf[i]=xmlBufferCreate();
     xmlNodeDump( buf[i], doc, xpathObj->nodesetval->nodeTab[i],0,0);
-    bufsize+=strlen(buf[i]->content); 
+    bufsize+=strlen((char*)buf[i]->content); 
   }
 
   output_buffer = malloc( bufsize);
@@ -240,8 +240,8 @@
 
   ptr = output_buffer;
   for(j=0;j<i;j++){
-    strcpy( ptr, buf[j]->content);
-    ptr+=strlen(buf[j]->content);
+    strcpy( ptr, (char*)buf[j]->content);
+    ptr+=strlen((char*)buf[j]->content);
   }
   *ptr='\0';
 
diff -ur XSB.orig/packages/xpath/configure XSB/packages/xpath/configure
--- XSB.orig/packages/xpath/configure	2005-03-06 13:29:02.000000000 +1300
+++ XSB/packages/xpath/configure	2007-01-27 11:34:56.000000000 +1300
@@ -550,10 +550,10 @@
 ac_configure=$ac_aux_dir/configure # This should be Cygnus configure.
 
 
-echo '
-- You did not tell me what kind of host system you want to configure.
-- I will attempt to guess the kind of system this is.
-' 1>&6
+#echo '
+#- You did not tell me what kind of host system you want to configure.
+#- I will attempt to guess the kind of system this is.
+#' 1>&6
 
 # Make sure we can run config.sub.
 if ${CONFIG_SHELL-/bin/sh} $ac_config_sub sun4 >/dev/null 2>&1; then :
@@ -592,6 +592,7 @@
 ccflags=
 ldflags=
 
+if test false ; then
 case "$canonical" in
      *-cygwin*)
         pkg_src_dir=$build_dir/cc
@@ -614,6 +615,10 @@
 	ldflags="`sh $xpath_config_prog $exec_prefix_option $prefix_option --libs`"
 	;;
 esac
+fi
+
+xmllib_ccflags=`xml2-config --cflags`
+ldflags=`xml2-config --libs`
 
 ccflags="$xmllib_ccflags"	
 if test  "$xmllib_ccflags" != "ERROR" && test "$ldflags" != "ERROR"  ; then
