Index: qt/pinentrydialog.cpp
===================================================================
--- qt/pinentrydialog.cpp	(revision 233)
+++ qt/pinentrydialog.cpp	(working copy)
@@ -33,7 +33,7 @@
 
 PinEntryDialog::PinEntryDialog( QWidget* parent, const char* name, 
                                 bool modal, bool enable_quality_bar )
-  : QDialog( parent, name, modal ), _grabbed( false )
+  : QDialog( parent, name, modal, Qt::WStyle_StaysOnTop ), _grabbed( false )
 {
   QBoxLayout* top = new QVBoxLayout( this, 6 );
   QBoxLayout* upperLayout = new QHBoxLayout( top );
Index: qt4/pinentrydialog.cpp
===================================================================
--- qt4/pinentrydialog.cpp	(revision 233)
+++ qt4/pinentrydialog.cpp	(working copy)
@@ -83,7 +83,7 @@
 
 PinEntryDialog::PinEntryDialog( QWidget* parent, const char* name, bool modal,
                                 bool enable_quality_bar )
-  : QDialog( parent ), _grabbed( false )
+  : QDialog( parent, Qt::WindowStaysOnTopHint ), _grabbed( false )
 {
   setWindowFlags( windowFlags() & ~Qt::WindowContextHelpButtonHint );
 
