Index: motion-3.2.10.1/configure.in
===================================================================
--- motion-3.2.10.1.orig/configure.in
+++ motion-3.2.10.1/configure.in
@@ -303,7 +303,7 @@ if test x$MJPEG_HEADER1_SUPPORT != xyes 
    LDFLAGS=$TEMP_LDFLAGS  
 	
    AC_CHECK_LIB(lavjpeg, decode_jpeg_raw,[ 
-		TEMP_LDFLAGS="$TEMP_LDFLAGS -lmjpegutils -llavjpeg"
+		TEMP_LIBS="$TEMP_LIBS -lmjpegutils -llavjpeg"
 		MJPEG_SUPPORT="yes"
 		TEMP_CFLAGS="$TEMP_CFLAGS -DMJPEGT"
                  ], , -lmjpegutils)
