diff -urN xine-lib-0.9.10/configure xine-lib-0.9.10_fix/configure
--- xine-lib-0.9.10/configure	Sat May 25 16:55:58 2002
+++ xine-lib-0.9.10_fix/configure	Sun Jun 16 21:49:37 2002
@@ -12693,7 +12693,7 @@
 
 
 
-INCLUDES='-I$(top_srcdir) -I$(top_builddir) -I$(top_srcdir)/src -I$(top_builddir)/src -I$(top_srcdir)/src/xine-engine -I$(top_builddir)/src/xine-engine -I$(top_srcdir)/src/xine-utils -I$(top_builddir)/src/xine-utils $(INTLDIR)'
+INCLUDES='-I /usr/include/directfb -I$(top_srcdir) -I$(top_builddir) -I$(top_srcdir)/src -I$(top_builddir)/src -I$(top_srcdir)/src/xine-engine -I$(top_builddir)/src/xine-engine -I$(top_srcdir)/src/xine-utils -I$(top_builddir)/src/xine-utils $(INTLDIR)'
 
 
 
