diff -ur src.orig/libw32dll/w32codec.c src/libw32dll/w32codec.c
--- src.orig/libw32dll/w32codec.c	2004-12-16 15:59:07.000000000 +0200
+++ src/libw32dll/w32codec.c	2005-04-23 01:53:07.000000000 +0300
@@ -1089,6 +1089,12 @@
     _x_meta_info_set_utf8(this->stream, XINE_META_INFO_AUDIOCODEC, 
       "Windows Media Audio v3 (win32)");
     return "wma9dmod.dll";
+  case BUF_AUDIO_WMALL:
+    this->driver_type = DRIVER_DMO;
+    this->guid=&wma3_clsid;
+    _x_meta_info_set_utf8(this->stream, XINE_META_INFO_AUDIOCODEC, 
+      "Windows Media Audio Lossless (win32)");
+    return "wma9dmod.dll";
   case BUF_AUDIO_WMAV:
     this->driver_type = DRIVER_DMO;
     this->guid=&wmav_clsid;
@@ -1703,6 +1709,7 @@
   BUF_AUDIO_WMAV1, BUF_AUDIO_WMAV2, BUF_AUDIO_WMAV3, BUF_AUDIO_MSADPCM, 
   BUF_AUDIO_MSIMAADPCM, BUF_AUDIO_MSGSM, BUF_AUDIO_IMC, BUF_AUDIO_LH,
   BUF_AUDIO_VOXWARE, BUF_AUDIO_ACELPNET, BUF_AUDIO_VIVOG723, BUF_AUDIO_WMAV,
+  BUF_AUDIO_WMALL,
   0
  };
 
diff -ur src.orig/xine-engine/buffer.h src/xine-engine/buffer.h
--- src.orig/xine-engine/buffer.h	2004-12-13 00:01:30.000000000 +0200
+++ src/xine-engine/buffer.h	2005-04-23 02:07:40.000000000 +0300
@@ -226,6 +226,7 @@
 #define BUF_AUDIO_VMD		0x03320000
 #define BUF_AUDIO_XAN_DPCM	0x03330000
 #define BUF_AUDIO_ALAC		0x03340000
+#define BUF_AUDIO_WMALL		0x03350000
 
 /* spu buffer types:    */
 
diff -ur src.orig/xine-engine/buffer_types.c src/xine-engine/buffer_types.c
--- src.orig/xine-engine/buffer_types.c	2004-11-30 02:40:59.000000000 +0200
+++ src/xine-engine/buffer_types.c	2005-04-23 01:43:43.000000000 +0300
@@ -709,6 +709,13 @@
 },
 {
   {
+    0x163, 0
+  },
+  BUF_AUDIO_WMALL,
+  "Windows Media Audio Lossless"
+},
+{
+  {
     0xA, 0
   },
   BUF_AUDIO_WMAV,
