diff -Nru xine-lib-0.9.12/src/video_out/Makefile.in xine-lib-0.9.12.new/src/video_out/Makefile.in
--- xine-lib-0.9.12/src/video_out/Makefile.in	Sun Jun 23 04:51:37 2002
+++ xine-lib-0.9.12.new/src/video_out/Makefile.in	Sun Jun 23 11:25:49 2002
@@ -84,8 +84,6 @@
 DATADIRNAME = @DATADIRNAME@
 DEPCOMP = @DEPCOMP@
 DEPMOD = @DEPMOD@
-DIRECTFB_CFLAGS = @DIRECTFB_CFLAGS@
-DIRECTFB_LIBS = @DIRECTFB_LIBS@
 DIVX4_LIBS = @DIVX4_LIBS@
 DLLTOOL = @DLLTOOL@
 DOXYGEN = @DOXYGEN@
@@ -206,8 +204,6 @@
 
 @HAVE_FB_TRUE@fb_module = xineplug_vo_out_fb.la
 
-@HAVE_DIRECTFB_TRUE@directfb_module = xineplug_vo_out_directfb.la
-
 @HAVE_SDL_TRUE@sdl_module = xineplug_vo_out_sdl.la
 
 # important:
@@ -215,7 +211,7 @@
 # all xine video out plugins should be named  
 # "xineplug_vo_out_*"
 
-lib_LTLIBRARIES = $(xshm_module) $(xv_module) $(directfb_module) $(aa_module)                   $(syncfb_module) $(fb_module) $(opengl_module) $(sdl_module) $(vidix_module)
+lib_LTLIBRARIES = $(xshm_module) $(xv_module) $(aa_module)                   $(syncfb_module) $(fb_module) $(opengl_module) $(sdl_module) $(vidix_module)
 
 
 xineplug_vo_out_xv_la_SOURCES = deinterlace.c alphablend.c video_out_xv.c
@@ -247,10 +243,6 @@
 xineplug_vo_out_fb_la_LIBADD = $(top_builddir)/src/xine-utils/libxineutils.la
 xineplug_vo_out_fb_la_LDFLAGS = -avoid-version -module
 
-xineplug_vo_out_directfb_la_SOURCES = alphablend.c video_out_directfb.c
-xineplug_vo_out_directfb_la_LIBADD = $(DIRECTFB_LIBS)
-xineplug_vo_out_directfb_la_LDFLAGS = -avoid-version -module
-
 xineplug_vo_out_sdl_la_SOURCES = alphablend.c video_out_sdl.c
 xineplug_vo_out_sdl_la_LIBADD = $(SDL_LIBS)
 xineplug_vo_out_sdl_la_LDFLAGS = -avoid-version -module
@@ -284,9 +276,6 @@
 $(top_builddir)/src/xine-utils/libxineutils.la
 xineplug_vo_out_xv_la_OBJECTS =  deinterlace.lo alphablend.lo \
 video_out_xv.lo
-xineplug_vo_out_directfb_la_DEPENDENCIES = 
-xineplug_vo_out_directfb_la_OBJECTS =  alphablend.lo \
-video_out_directfb.lo
 xineplug_vo_out_aa_la_DEPENDENCIES = 
 xineplug_vo_out_aa_la_OBJECTS =  video_out_aa.lo
 xineplug_vo_out_syncfb_la_DEPENDENCIES =  \
@@ -317,8 +306,8 @@
 
 TAR = tar
 GZIP_ENV = --best
-SOURCES = $(xineplug_vo_out_xshm_la_SOURCES) $(xineplug_vo_out_xv_la_SOURCES) $(xineplug_vo_out_directfb_la_SOURCES) $(xineplug_vo_out_aa_la_SOURCES) $(xineplug_vo_out_syncfb_la_SOURCES) $(xineplug_vo_out_fb_la_SOURCES) $(xineplug_vo_out_opengl_la_SOURCES) $(xineplug_vo_out_sdl_la_SOURCES) $(xineplug_vo_out_vidix_la_SOURCES)
-OBJECTS = $(xineplug_vo_out_xshm_la_OBJECTS) $(xineplug_vo_out_xv_la_OBJECTS) $(xineplug_vo_out_directfb_la_OBJECTS) $(xineplug_vo_out_aa_la_OBJECTS) $(xineplug_vo_out_syncfb_la_OBJECTS) $(xineplug_vo_out_fb_la_OBJECTS) $(xineplug_vo_out_opengl_la_OBJECTS) $(xineplug_vo_out_sdl_la_OBJECTS) $(xineplug_vo_out_vidix_la_OBJECTS)
+SOURCES = $(xineplug_vo_out_xshm_la_SOURCES) $(xineplug_vo_out_xv_la_SOURCES) $(xineplug_vo_out_aa_la_SOURCES) $(xineplug_vo_out_syncfb_la_SOURCES) $(xineplug_vo_out_fb_la_SOURCES) $(xineplug_vo_out_opengl_la_SOURCES) $(xineplug_vo_out_sdl_la_SOURCES) $(xineplug_vo_out_vidix_la_SOURCES)
+OBJECTS = $(xineplug_vo_out_xshm_la_OBJECTS) $(xineplug_vo_out_xv_la_OBJECTS) $(xineplug_vo_out_aa_la_OBJECTS) $(xineplug_vo_out_syncfb_la_OBJECTS) $(xineplug_vo_out_fb_la_OBJECTS) $(xineplug_vo_out_opengl_la_OBJECTS) $(xineplug_vo_out_sdl_la_OBJECTS) $(xineplug_vo_out_vidix_la_OBJECTS)
 
 all: all-redirect
 .SUFFIXES:
@@ -406,9 +395,6 @@
 xineplug_vo_out_xv.la: $(xineplug_vo_out_xv_la_OBJECTS) $(xineplug_vo_out_xv_la_DEPENDENCIES)
 	$(LINK) -rpath $(libdir) $(xineplug_vo_out_xv_la_LDFLAGS) $(xineplug_vo_out_xv_la_OBJECTS) $(xineplug_vo_out_xv_la_LIBADD) $(LIBS)
 
-xineplug_vo_out_directfb.la: $(xineplug_vo_out_directfb_la_OBJECTS) $(xineplug_vo_out_directfb_la_DEPENDENCIES)
-	$(LINK) -rpath $(libdir) $(xineplug_vo_out_directfb_la_LDFLAGS) $(xineplug_vo_out_directfb_la_OBJECTS) $(xineplug_vo_out_directfb_la_LIBADD) $(LIBS)
-
 xineplug_vo_out_aa.la: $(xineplug_vo_out_aa_la_OBJECTS) $(xineplug_vo_out_aa_la_DEPENDENCIES)
 	$(LINK) -rpath $(libdir) $(xineplug_vo_out_aa_la_LDFLAGS) $(xineplug_vo_out_aa_la_OBJECTS) $(xineplug_vo_out_aa_la_LIBADD) $(LIBS)
 
