https://bugs.freedesktop.org/show_bug.cgi?id=4149

diff -Nur libXScrnSaver-0.99.0.orig/src/Makefile.am libXScrnSaver-0.99.0/src/Makefile.am
--- libXScrnSaver-0.99.0.orig/src/Makefile.am	2005-08-01 22:15:26.000000000 +0200
+++ libXScrnSaver-0.99.0/src/Makefile.am	2005-08-21 03:35:14.000000000 +0200
@@ -1,11 +1,11 @@
 AM_CFLAGS = $(DEP_CFLAGS)
-lib_LTLIBRARIES = libXScrnSaver.la
+lib_LTLIBRARIES = libXss.la
 
-libXScrnSaver_la_SOURCES = \
+libXss_la_SOURCES = \
          XScrnSaver.c
 
-libXScrnSaver_la_LIBADD = $(DEP_LIBS)
+libXss_la_LIBADD = $(DEP_LIBS)
 
-libXScrnSaver_la_LDFLAGS = -version-info 7:0:0 -no-undefined
+libXss_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 
 
diff -Nur libXScrnSaver-0.99.0.orig/xscrnsaver.pc.in libXScrnSaver-0.99.0/xscrnsaver.pc.in
--- libXScrnSaver-0.99.0.orig/xscrnsaver.pc.in	2005-08-01 22:15:26.000000000 +0200
+++ libXScrnSaver-0.99.0/xscrnsaver.pc.in	2005-08-21 03:35:27.000000000 +0200
@@ -7,5 +7,5 @@
 Description: The XScrnSaver Library
 Version: @PACKAGE_VERSION@
 Cflags: -I${includedir} @DEP_CFLAGS@
-Libs: -L${libdir} -lXScrnSaver @DEP_LIBS@
+Libs: -L${libdir} -lXss @DEP_LIBS@
 
