diff -urN xmame-0.80.1/makefile.unix xmame-0.80.2/makefile.unix
--- xmame-0.80.1/makefile.unix	2004-03-16 07:10:18.615979992 -0600
+++ xmame-0.80.2/makefile.unix	2004-03-16 07:10:36.664236240 -0600
@@ -28,6 +28,12 @@
 ###########################################################################
 
 
+##############################################################################
+# Makefile Identification
+##############################################################################
+CURMAKEFILE := $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST))
+
+
 ###########################################################################
 # Xmame or xmess or...?
 ###########################################################################
diff -urN xmame-0.80.1/src/unix/unix.mak xmame-0.80.2/src/unix/unix.mak
--- xmame-0.80.1/src/unix/unix.mak	2004-03-16 07:10:27.393645584 -0600
+++ xmame-0.80.2/src/unix/unix.mak	2004-03-16 07:19:43.053172496 -0600
@@ -147,7 +147,8 @@
 ZLIB = src/unix/contrib/cutzlib-1.2.1/libz.a
 endif
 
-all: objdirs osdepend-objdirs $(NAME).$(DISPLAY_METHOD)
+all: objdirs osdepend-objdirs 
+	$(MAKE) -f $(CURMAKEFILE) $(MAKEFLAGS) $(NAME).$(DISPLAY_METHOD)
 
 # CPU core include paths
 VPATH = src $(wildcard src/cpu/*)
