Index: trunk/ath/if_ath.c
===================================================================
--- trunk/ath/if_ath.c (revision 1720)
+++ trunk/ath/if_ath.c (revision 1721)
@@ -45,5 +45,7 @@
 #include "opt_ah.h"
 
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/ath/if_ath_ahb.c
===================================================================
--- trunk/ath/if_ath_ahb.c (revision 1714)
+++ trunk/ath/if_ath_ahb.c (revision 1721)
@@ -11,5 +11,7 @@
 #endif
 
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/ath/if_ath_pci.c
===================================================================
--- trunk/ath/if_ath_pci.c (revision 1714)
+++ trunk/ath/if_ath_pci.c (revision 1721)
@@ -43,5 +43,7 @@
 #endif
 
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/ath_rate/amrr/amrr.c
===================================================================
--- trunk/ath_rate/amrr/amrr.c (revision 1667)
+++ trunk/ath_rate/amrr/amrr.c (revision 1721)
@@ -44,5 +44,7 @@
  *    Mathieu Lacage, Hossein Manshaei, Thierry Turletti
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/ath_rate/onoe/onoe.c
===================================================================
--- trunk/ath_rate/onoe/onoe.c (revision 1667)
+++ trunk/ath_rate/onoe/onoe.c (revision 1721)
@@ -40,5 +40,7 @@
  * Atsushi Onoe's rate control algorithm.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/ath_rate/sample/sample.c
===================================================================
--- trunk/ath_rate/sample/sample.c (revision 1667)
+++ trunk/ath_rate/sample/sample.c (revision 1721)
@@ -42,5 +42,7 @@
  */
 
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/hal/linux/ah_osdep.c
===================================================================
--- trunk/hal/linux/ah_osdep.c (revision 1711)
+++ trunk/hal/linux/ah_osdep.c (revision 1721)
@@ -42,5 +42,7 @@
 #endif
 
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211.c
===================================================================
--- trunk/net80211/ieee80211.c (revision 1666)
+++ trunk/net80211/ieee80211.c (revision 1721)
@@ -39,5 +39,7 @@
  * IEEE 802.11 generic handler
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_acl.c
===================================================================
--- trunk/net80211/ieee80211_acl.c (revision 1614)
+++ trunk/net80211/ieee80211_acl.c (revision 1721)
@@ -46,5 +46,7 @@
  * or rejected (ACL_POLICY_DENT).
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_beacon.c
===================================================================
--- trunk/net80211/ieee80211_beacon.c (revision 1575)
+++ trunk/net80211/ieee80211_beacon.c (revision 1721)
@@ -39,5 +39,7 @@
  * IEEE 802.11 beacon handling routines
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_crypto.c
===================================================================
--- trunk/net80211/ieee80211_crypto.c (revision 1426)
+++ trunk/net80211/ieee80211_crypto.c (revision 1721)
@@ -39,5 +39,7 @@
  * IEEE 802.11 generic crypto support.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_crypto_ccmp.c
===================================================================
--- trunk/net80211/ieee80211_crypto_ccmp.c (revision 1643)
+++ trunk/net80211/ieee80211_crypto_ccmp.c (revision 1721)
@@ -39,5 +39,7 @@
  * it's license is included below.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_crypto_none.c
===================================================================
--- trunk/net80211/ieee80211_crypto_none.c (revision 1426)
+++ trunk/net80211/ieee80211_crypto_none.c (revision 1721)
@@ -35,5 +35,7 @@
  * IEEE 802.11 NULL crypto support.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_crypto_tkip.c
===================================================================
--- trunk/net80211/ieee80211_crypto_tkip.c (revision 1426)
+++ trunk/net80211/ieee80211_crypto_tkip.c (revision 1721)
@@ -39,5 +39,7 @@
  * it's license is included below.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_crypto_wep.c
===================================================================
--- trunk/net80211/ieee80211_crypto_wep.c (revision 1443)
+++ trunk/net80211/ieee80211_crypto_wep.c (revision 1721)
@@ -35,5 +35,7 @@
  * IEEE 802.11 WEP crypto support.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_input.c
===================================================================
--- trunk/net80211/ieee80211_input.c (revision 1713)
+++ trunk/net80211/ieee80211_input.c (revision 1721)
@@ -39,5 +39,7 @@
  * IEEE 802.11 input handling.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_linux.c
===================================================================
--- trunk/net80211/ieee80211_linux.c (revision 1690)
+++ trunk/net80211/ieee80211_linux.c (revision 1721)
@@ -34,5 +34,7 @@
  * IEEE 802.11 support (Linux-specific code)
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_monitor.c
===================================================================
--- trunk/net80211/ieee80211_monitor.c (revision 1720)
+++ trunk/net80211/ieee80211_monitor.c (revision 1721)
@@ -35,5 +35,7 @@
  * IEEE 802.11 monitor mode 
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_node.c
===================================================================
--- trunk/net80211/ieee80211_node.c (revision 1640)
+++ trunk/net80211/ieee80211_node.c (revision 1721)
@@ -39,5 +39,7 @@
  * IEEE 802.11 node handling support.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_output.c
===================================================================
--- trunk/net80211/ieee80211_output.c (revision 1637)
+++ trunk/net80211/ieee80211_output.c (revision 1721)
@@ -39,5 +39,7 @@
  * IEEE 802.11 output handling.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_power.c
===================================================================
--- trunk/net80211/ieee80211_power.c (revision 1637)
+++ trunk/net80211/ieee80211_power.c (revision 1721)
@@ -39,5 +39,7 @@
  * IEEE 802.11 power save support.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_proto.c
===================================================================
--- trunk/net80211/ieee80211_proto.c (revision 1699)
+++ trunk/net80211/ieee80211_proto.c (revision 1721)
@@ -39,5 +39,7 @@
  * IEEE 802.11 protocol support.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/kmod.h>
Index: trunk/net80211/ieee80211_scan.c
===================================================================
--- trunk/net80211/ieee80211_scan.c (revision 1698)
+++ trunk/net80211/ieee80211_scan.c (revision 1721)
@@ -38,5 +38,7 @@
  * IEEE 802.11 scanning support.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_scan_ap.c
===================================================================
--- trunk/net80211/ieee80211_scan_ap.c (revision 1698)
+++ trunk/net80211/ieee80211_scan_ap.c (revision 1721)
@@ -38,5 +38,7 @@
  * IEEE 802.11 ap scanning support.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_scan_sta.c
===================================================================
--- trunk/net80211/ieee80211_scan_sta.c (revision 1698)
+++ trunk/net80211/ieee80211_scan_sta.c (revision 1721)
@@ -38,5 +38,7 @@
  * IEEE 802.11 station scanning support.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/ieee80211_wireless.c
===================================================================
--- trunk/net80211/ieee80211_wireless.c (revision 1717)
+++ trunk/net80211/ieee80211_wireless.c (revision 1721)
@@ -40,5 +40,7 @@
  * Wireless extensions support for 802.11 common code.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 
 #ifdef CONFIG_NET_WIRELESS
Index: trunk/net80211/ieee80211_xauth.c
===================================================================
--- trunk/net80211/ieee80211_xauth.c (revision 1426)
+++ trunk/net80211/ieee80211_xauth.c (revision 1721)
@@ -47,5 +47,7 @@
  * entirely from messages about stations joining and leaving.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/net80211/if_media.c
===================================================================
--- trunk/net80211/if_media.c (revision 1614)
+++ trunk/net80211/if_media.c (revision 1721)
@@ -50,5 +50,7 @@
 #endif
 
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/regression/ccmp/test_ccmp.c
===================================================================
--- trunk/regression/ccmp/test_ccmp.c (revision 1426)
+++ trunk/regression/ccmp/test_ccmp.c (revision 1721)
@@ -45,5 +45,7 @@
  * 1, 2, and 3.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/regression/tkip/test_tkip.c
===================================================================
--- trunk/regression/tkip/test_tkip.c (revision 1426)
+++ trunk/regression/tkip/test_tkip.c (revision 1721)
@@ -35,5 +35,7 @@
  * TKIP test module.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
Index: trunk/regression/wep/test_wep.c
===================================================================
--- trunk/regression/wep/test_wep.c (revision 1426)
+++ trunk/regression/wep/test_wep.c (revision 1721)
@@ -45,5 +45,7 @@
  * 1, 2, and 3.
  */
+#ifndef AUTOCONF_INCLUDED
 #include <linux/config.h>
+#endif
 #include <linux/version.h>
 #include <linux/module.h>
