--- freenet/build.xml	2009-08-04 20:54:09.267989544 +0200
+++ freenet/build.xml.new	2009-08-04 20:57:01.295527065 +0200
@@ -68,19 +68,9 @@
 		</exec>
 	</target>
 
-	<target name="get-extjar" depends="env" unless="freenet-ext.present"
-		description="Download some external libraries which Freenet relies on">
-		<mkdir dir="${lib}"/>
-		<get src="http://checksums.freenetproject.org/cc/freenet-ext.jar"
-			dest="${freenet-ext.location}"
-			verbose="true"
-			usetimestamp="true"/>
-		<property name="freenet-ext.present" value="true"/>
-	</target>
-
 	<!-- ================================================== -->
 
-	<target name="compile" depends="get-extjar">
+	<target name="compile" depends="env">
 		<!-- Create the time stamp -->
 		<tstamp/>

diff -ur freenet-old/src/freenet/node/NodeStarter.java freenet/src/freenet/node/NodeStarter.java
--- freenet-old/src/freenet/node/NodeStarter.java	2008-10-28 21:40:50.000000000 +0100
+++ freenet/src/freenet/node/NodeStarter.java	2008-10-28 21:39:14.000000000 +0100
@@ -162,37 +162,8 @@
 	}
 
 	private void getExtBuild() {
-		try {
-			extBuildNumber = ExtVersion.buildNumber;
-			extRevisionNumber = ExtVersion.cvsRevision;
-			String builtWithMessage = "freenet.jar built with freenet-ext.jar Build #" + extBuildNumber + " r" + extRevisionNumber;
-			Logger.normal(this, builtWithMessage);
-			System.out.println(builtWithMessage);
-			extBuildNumber = ExtVersion.buildNumber();
-			if(extBuildNumber == -42) {
-				extBuildNumber = ExtVersion.extBuildNumber();
-				extRevisionNumber = ExtVersion.extRevisionNumber();
-			}
-			if(extBuildNumber == 0) {
-				String buildMessage = "extBuildNumber is 0; perhaps your freenet-ext.jar file is corrupted?";
-				Logger.error(this, buildMessage);
-				System.err.println(buildMessage);
-				extBuildNumber = -1;
-			}
-			if(extRevisionNumber == null) {
-				String revisionMessage = "extRevisionNumber is null; perhaps your freenet-ext.jar file is corrupted?";
-				Logger.error(this, revisionMessage);
-				System.err.println(revisionMessage);
-				extRevisionNumber = "INVALID";
-			}
-		} catch(Throwable t) {
-			// Compatibility code ... will be removed
-			Logger.error(this, "Unable to get the version of your freenet-ext file : it's probably corrupted!");
-			System.err.println("Unable to get the version of your freenet-ext file : it's probably corrupted!");
-			System.err.println(t.getMessage());
-			extRevisionNumber = "INVALID";
-			extBuildNumber = -1;
-		}
+               extRevisionNumber = "";
+               extBuildNumber = RECOMMENDED_EXT_BUILD_NUMBER;	
 	}
 
 	/**
