Only in smalltalk-2.1.8.orig/: gnu-smalltalk.spec
diff -ur smalltalk-2.1.8.orig/gst-package.in smalltalk-2.1.8/gst-package.in
--- smalltalk-2.1.8.orig/gst-package.in	2003-05-27 12:41:10.000000000 -0500
+++ smalltalk-2.1.8/gst-package.in	2004-07-05 16:11:46.795594856 -0500
@@ -138,8 +138,13 @@
 
 echo "Merging $files and $IMAGE_PATH/packages.xml..."
 $dry_run || {
-  sed -e '$i\' -e '</packages>' -e '/<.packages>/,/<packages>/d' $files > $IMAGE_PATH/packages.tmp
-  mv $IMAGE_PATH/packages.tmp $IMAGE_PATH/packages.xml
+  tmpfile=$(mktemp)
+  sed -e '$i\' -e '</packages>' -e '/<.packages>/,/<packages>/d' $files > $tmpfile
+  if [ -n ${EBUILD_PHASE} ]; then
+    mv $tmpfile ${D}/usr/share/smalltalk/packages.xml
+  else
+    mv $tmpfile $IMAGE_PATH/packages.xml
+  fi
 }
 
 
