--- ./platforms/unix/plugins/MIDIPlugin/sqUnixMIDIALSA.inc.orig	Wed Jun 24 17:39:12 2009 -0400
+++ ./platforms/unix/plugins/MIDIPlugin/sqUnixMIDIALSA.inc	Wed Jun 24 18:01:19 2009 -0400
@@ -116,6 +116,12 @@
 
 #include <als./asoundlib.h>
 
+#if (DEBUG)
+# define dprintf(...) printf(__VA_ARGS__)
+#else
+# define dprintf(...)
+#endif
+
 static snd_seq_t *seq      =  0;
 static int	  queue    =  0;
 static int	  in_port  = -1;
--- ./platforms/unix/vm-display-fbdev/sqUnixFBDev.c.orig	Wed Jun 24 17:39:12 2009 -0400
+++ ./platforms/unix/vm-display-fbdev/sqUnixFBDev.c	Wed Jun 24 18:01:19 2009 -0400
@@ -49,7 +49,13 @@
 # undef ioMSecs
 #endif
 
-#include <stdio.h>
+// we need stdio.h to override glibc's dprintf function
+// since new glibc has this function with different interface
+#include <features.h>
+#ifdef __USE_XOPEN2K8
+# include <stdio.h>
+#endif
+
 #include <time.h>
 #include <sys/time.h>
 #include <sys/types.h>
@@ -68,15 +74,11 @@
 #endif
 
 
-static void dprintf(const char *fmt, ...)
-{
 #if (DEBUG)
-  va_list ap;
-  va_start(ap, fmt);
-  vprintf(fmt, ap);
-  va_end(ap);
+# define dprintf(...) printf(__VA_ARGS__)
+#else
+# define dprintf(...)
 #endif
-}
 
 static void fatalError(const char *who)
 {
--- ./platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c.orig	Wed Jun 24 17:39:12 2009 -0400
+++ ./platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c	Wed Jun 24 18:01:19 2009 -0400
@@ -33,13 +33,12 @@
 #include <audio/audiolib.h>
 #include <assert.h>
 
-#ifdef DEBUG
-# define dprintf printf
+#if (DEBUG)
+# define dprintf(...) printf(__VA_ARGS__)
 #else
-  static void dprintf(char *fmt, ...) {}
+# define dprintf(...)
 #endif
 
-
 #ifdef WORDS_BIGENDIAN
 # define AU_FORMAT	AuFormatLinearSigned16MSB
 #else
