diff -urN libwmfun-0.0.4/libwmfun/Makefile.in libwmfun-0.0.4-modified/libwmfun/Makefile.in
--- libwmfun-0.0.4/libwmfun/Makefile.in	Wed Dec  6 17:54:04 2000
+++ libwmfun-0.0.4-modified/libwmfun/Makefile.in	Tue Jul 16 22:08:46 2002
@@ -82,7 +82,7 @@
 LTLIBRARIES =  $(lib_LTLIBRARIES)
 
 
-DEFS = @DEFS@ -I. -I$(srcdir) -I.
+DEFS = @DEFS@ -I. -I$(srcdir) -I. -I/usr/include/freetype2
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
diff -urN libwmfun-0.0.4/libwmfun/drawstring.c libwmfun-0.0.4-modified/libwmfun/drawstring.c
--- libwmfun-0.0.4/libwmfun/drawstring.c	Sun Jul 29 14:18:48 2001
+++ libwmfun-0.0.4-modified/libwmfun/drawstring.c	Tue Jul 16 20:45:39 2002
@@ -1,4 +1,4 @@
-#include <proplist.h>
+#include <WINGs/proplist-compat.h>
 #include <WINGs/WINGs.h>
 #include <WINGs/WINGsP.h>
 #include "generic.h"
