--- svninfo/Makefile.old	2006-02-11 23:57:52.000000000 +0100
+++ svninfo/Makefile	2006-02-11 23:58:10.000000000 +0100
@@ -74,7 +74,7 @@
 	$(PDFLATEX) $*.tex
 
 %.ps:%.dvi
-	$(DVIPS) $*.dvi
+	$(DVIPS) -o $@ $<
 
 html:	svninfo.perl svninfo2html.tex
 	rm -fr $(HTML_DIR)
--- svninfo/svninfo.dtx.old	2006-02-12 00:05:39.000000000 +0100
+++ svninfo/svninfo.dtx	2006-02-12 00:05:45.000000000 +0100
@@ -833,13 +833,6 @@
 \IfFileExists{html.sty}{\usepackage{html}}{}
 \usepackage{svninfo}
 
-\IfFileExists{html.sty}{%
-% the svninfo.perl stuff needs that the .aux file has been read!
-\begin{htmlonly}%
-\input{svninfo2html.aux}%
-\end{htmlonly}%
-}{}
-
 \newcommand{\LatexToHtml}{\LaTeX 2\texttt{HTML}}
 \newcommand{\Var}[1]{\texttt{\$svninfo::#1}}
 \newcommand{\SVN}{\emph{SVN}}
