diff -u gsl-1.2-r1-orig/ode-initval/rk8pd.c gsl-1.2-r1/ode-initval/rk8pd.c
--- gsl-1.2-r1-orig/ode-initval/rk8pd.c	2002-10-10 22:22:37.000000000 -0500
+++ gsl-1.2-r1/ode-initval/rk8pd.c	2002-10-11 02:09:26.000000000 -0500
@@ -29,6 +29,10 @@
 
 #include "odeiv_util.h"
 
+/* There is a variable here named k6, make sure it is not
+   defined as processor type */
+#undef k6
+
 /* Prince-Dormand constants */
 
 static const double Abar[] = {
diff -u gsl-1.2-r1-orig/ode-initval/rkck.c gsl-1.2-r1/ode-initval/rkck.c
--- gsl-1.2-r1-orig/ode-initval/rkck.c	2002-10-10 22:21:51.000000000 -0500
+++ gsl-1.2-r1/ode-initval/rkck.c	2002-10-11 02:09:42.000000000 -0500
@@ -29,6 +29,10 @@
 
 #include "odeiv_util.h"
 
+/* There is a variable here named k6, make sure it is not
+   defined as processor type */
+#undef k6
+
 /* Cash-Karp constants */
 static const double ah[] = { 1.0 / 5.0, 0.3, 3.0 / 5.0, 1.0, 7.0 / 8.0 };
 static const double b21 = 1.0 / 5.0;
diff -u gsl-1.2-r1-orig/ode-initval/rkf45.c gsl-1.2-r1/ode-initval/rkf45.c
--- gsl-1.2-r1-orig/ode-initval/rkf45.c	2002-10-10 22:22:41.000000000 -0500
+++ gsl-1.2-r1/ode-initval/rkf45.c	2002-10-11 02:07:27.000000000 -0500
@@ -27,6 +27,11 @@
 
 #include "odeiv_util.h"
 
+/* There is a variable here named k6, make sure it is not
+   defined as processor type */
+#undef k6
+
+
 /* Runge-Kutta-Fehlberg constants */
 static const double ah[] = { 1.0/4.0, 3.0/8.0, 12.0/13.0, 1.0, 1.0/2.0 };
 
