Index: parser.cpp
diff -c parser.cpp:1.63 parser.cpp:1.63.2.1
*** parser.cpp:1.63	Mon Sep  2 12:06:48 2002
--- parser.cpp	Mon Sep  9 12:39:21 2002
***************
*** 428,436 ****
          if (!(firstTime && rawBuffer[0] == (char) 0xEF && 
              rawBuffer[1] == (char) 0xBB && rawBuffer[2] == (char) 0xBF))
              S.setCurrLine(XML_GetCurrentLineNumber(parser));
!         int code = XML_GetErrorCode(parser); 
          Str eCodeStr, eNameStr;
!         eCodeStr = code;
          eNameStr = (char*) XML_ErrorString(code);
          // XML_ParserFree(parser); -- done later
          Err2T(this_, S, E_XML, eCodeStr, eNameStr);
--- 428,436 ----
          if (!(firstTime && rawBuffer[0] == (char) 0xEF && 
              rawBuffer[1] == (char) 0xBB && rawBuffer[2] == (char) 0xBF))
              S.setCurrLine(XML_GetCurrentLineNumber(parser));
!         XML_Error code = XML_GetErrorCode(parser); 
          Str eCodeStr, eNameStr;
!         eCodeStr = (int)code;
          eNameStr = (char*) XML_ErrorString(code);
          // XML_ParserFree(parser); -- done later
          Err2T(this_, S, E_XML, eCodeStr, eNameStr);
Index: situa.cpp
diff -c situa.cpp:1.23 situa.cpp:1.23.6.1
*** situa.cpp:1.23	Thu Jul  4 16:12:34 2002
--- situa.cpp	Mon Sep  9 13:12:12 2002
***************
*** 262,268 ****
      {
        if (l[i] > size / __numargs)
  	{
! 	  b[i] = new char[size / __numargs];
  	  strcpy(b[i], "...");
  	  strcpy(b[i] + 3, a[i] + (l[i] - size / __numargs + 3));
  	}
--- 262,268 ----
      {
        if (l[i] > size / __numargs)
  	{
! 	  b[i] = new char[size / __numargs + 1];
  	  strcpy(b[i], "...");
  	  strcpy(b[i] + 3, a[i] + (l[i] - size / __numargs + 3));
  	}
