--- gamin-0.0.6/server/gam_server.c	2004-08-24 19:20:13.397795664 +0200
+++ gamin-0.0.6.az/server/gam_server.c	2004-08-24 19:20:23.935193736 +0200
@@ -33,7 +33,7 @@
 #include "gam_channel.h"
 #include "gam_subscription.h"
 #include "gam_poll.h"
-#ifdef USE_INOTIFY
+#ifdef ENABLE_INOTIFY
 #include "gam_inotify.h"
 #endif
 #ifdef linux
@@ -68,7 +68,7 @@
 
     gam_backend = 0;
 
-#ifdef USE_INOTIFY
+#ifdef ENABLE_INOTIFY
     if ((!gam_backend) && (ret = gam_inotify_init())) {
 	gam_backend = BACKEND_INOTIFY;
 	gam_debug(DEBUG_INFO, "Using INotify as backend\n");
@@ -138,7 +138,7 @@
     }
  ***/
     switch (gam_backend) {
-#ifdef USE_INOTIFY
+#ifdef ENABLE_INOTIFY
     case BACKEND_INOTIFY:
 	return (gam_inotify_add_subscription(sub));
 	break;
@@ -165,7 +165,7 @@
 gam_remove_subscription(GamSubscription * sub)
 {
     switch (gam_backend) {
-#ifdef USE_INOTIFY
+#ifdef ENABLE_INOTIFY
     case BACKEND_INOTIFY:
 	return (gam_inotify_remove_subscription(sub));
 	break;
