diff -urN Guppi-0.40.3.old/libguppitank/guppi-object-barchart.c Guppi-0.40.3/libguppitank/guppi-object-barchart.c
--- Guppi-0.40.3.old/libguppitank/guppi-object-barchart.c	2002-01-08 01:28:53.000000000 -0500
+++ Guppi-0.40.3/libguppitank/guppi-object-barchart.c	2004-08-30 14:26:15.617630328 -0400
@@ -699,8 +699,6 @@
     if (obar->bar_callback3)
       obar->bar_callback3 (r, c, obar->bar_callback3_data);
     break;
-
-  default:
   }
 }
 
@@ -731,8 +729,6 @@
 			      (obar->data_cols - 1 - box) : box, 
 			      obar->legend_callback3_data);
     break;
-
-  default:
   }
 }
 
diff -urN Guppi-0.40.3.old/src/demo.c Guppi-0.40.3/src/demo.c
--- Guppi-0.40.3.old/src/demo.c	2002-01-08 01:28:57.000000000 -0500
+++ Guppi-0.40.3/src/demo.c	2004-08-30 14:23:18.144610328 -0400
@@ -450,7 +450,6 @@
   case GUPPI_EAST:
     compass_pos = GUPPI_NORTH;
     break;
-  default:
   }
 
   guppi_element_state_set (state,
diff -urN Guppi-0.40.3.old/src/libguppiplot/guppi-layout-constraint.c Guppi-0.40.3/src/libguppiplot/guppi-layout-constraint.c
--- Guppi-0.40.3.old/src/libguppiplot/guppi-layout-constraint.c	2002-01-18 21:45:21.000000000 -0500
+++ Guppi-0.40.3/src/libguppiplot/guppi-layout-constraint.c	2004-08-30 14:20:44.331993400 -0400
@@ -162,9 +162,6 @@
     guppi_layout_constraint_add_term (glc, GLC_REGION_TOP,     factor, geom);
     guppi_layout_constraint_add_term (glc, GLC_REGION_BOTTOM, -factor, geom);
     return;
-
-  default:
-    /* Fall through, do nothing. */
   }
 
   ti = term_info_new (type, factor, geom);
diff -urN Guppi-0.40.3.old/src/libguppiplot/guppi-root-group-item.c Guppi-0.40.3/src/libguppiplot/guppi-root-group-item.c
--- Guppi-0.40.3.old/src/libguppiplot/guppi-root-group-item.c	2002-01-08 01:31:08.000000000 -0500
+++ Guppi-0.40.3/src/libguppiplot/guppi-root-group-item.c	2004-08-30 14:22:24.517762848 -0400
@@ -546,8 +546,6 @@
 
   case GDK_MOTION_NOTIFY:
     return motion_notify_event (root, &(ev->motion));
-
-  default:
   }
 
   return FALSE;
@@ -772,11 +770,6 @@
     }
 
     break;
-      
-
-  case ROOT_GROUP_RESIZE_NONE:
-  default:
-    /* insure we do nothing */
   }
 
   if (please_resize_canvas)
diff -urN Guppi-0.40.3.old/src/libguppiplot/guppi-text-block.c Guppi-0.40.3/src/libguppiplot/guppi-text-block.c
--- Guppi-0.40.3.old/src/libguppiplot/guppi-text-block.c	2001-11-26 15:00:17.000000000 -0500
+++ Guppi-0.40.3/src/libguppiplot/guppi-text-block.c	2004-08-30 14:20:05.056964112 -0400
@@ -897,9 +897,6 @@
       /* Each hard break gets a corresponding soft break --- so do
 	 nothing. */
       break;
-
-    default:
-      /* Do nothing. */
     }
 
     render_stack_evolve (stack, tt);
diff -urN Guppi-0.40.3.old/src/libguppiplot/guppi-view-interval.c Guppi-0.40.3/src/libguppiplot/guppi-view-interval.c
--- Guppi-0.40.3.old/src/libguppiplot/guppi-view-interval.c	2002-01-18 21:45:21.000000000 -0500
+++ Guppi-0.40.3/src/libguppiplot/guppi-view-interval.c	2004-08-30 14:17:27.552908368 -0400
@@ -226,7 +226,6 @@
     return x != 0;
 #endif
 
-  default:
   }
 
   return TRUE;
