diff --git a/Makefile.am b/Makefile.am
index 508d13e..2cd704e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -40,16 +40,3 @@ MAINTAINERCLEANFILES = \
 	*.log core FILE *~ Makefile.in configure config.h.in \
 	config.status aclocal.m4 config.guess config.sub \
 	depcomp install-sh mkinstalldirs missing
-
-
-# Run ldconfig after installing the library:
-#
-#   '-ldconfig' expands into 'PATH="$PATH:/sbin" ldconfig -n /usr/local/bin'
-#
-# Somehow, this doen't work.
-#
-install-exec-hook:
-	-ldconfig
-
-uninstall-hook:
-	-ldconfig
