diff --git a/utils/Makefile.am b/utils/Makefile.am
index 82c2345..6550015 100644
--- a/utils/Makefile.am
+++ b/utils/Makefile.am
@@ -15,8 +15,8 @@ noinst_PROGRAMS = bytereader
 ASSADIR = ${top_srcdir}/assa
 INCLUDES = -I$(top_srcdir)
 
-LDFLAGS += -static
-LDADD = $(top_builddir)/assa/libassa-@ASSA_RELEASE@.la @WIN32_EXTRA_LIBADD@
+LDFLAGS += -L$(top_builddir)/assa/.libs
+LIBS = -lassa-3.5
 
 bytereader_SOURCES   = bytereader.cpp
 
