--- configure.in.orig	2004-02-03 23:57:29.000000000 +0100
+++ configure.in	2004-12-02 03:27:39.649492528 +0100
@@ -44,6 +44,10 @@
   [  --with-dv-yv12,                   libDV PAL YV12 read-support])
 AC_ARG_ENABLE(zalpha,
   [  --enable-zalpha,                  include MPEG Z/Alpha support in mplex (www.geofront.se/products_software_mza.php)])
+AC_ARG_WITH(v4l,
+  [ --without-v4l		       disable video4linux support])
+AC_ARG_WITH(gtk,
+  [--without-gtk		       disable gtk+ support])
 
 AC_GNU_SOURCE
 
@@ -155,12 +159,16 @@
 os_type=`uname`
 have_video4linux=false
 if test x$os_type = x"Linux"; then
-  AC_CHECK_HEADER(linux/videodev.h,
-    have_video4linux=true
-    AC_DEFINE(HAVE_V4L,1,
-		      [Building for Linux - using the video4linux API]),
-    AC_MSG_ERROR([videodev.h not found - please install the linux kernel headers])
-    )
+  if test x$with_v4l != xno; then
+    AC_CHECK_HEADER(linux/videodev.h,
+      have_video4linux=true
+      AC_DEFINE(HAVE_V4L,1,
+  		      [Building for Linux - using the video4linux API]),
+      AC_MSG_ERROR([videodev.h not found - please install the linux kernel headers])
+      )
+  else
+    AC_MSG_WARN([video4linux support disapled])
+  fi
 else
   AC_MSG_WARN([Did not find linux platform - video4linux functionality will be disabled])
   AC_MSG_WARN([If you are on MacOS X, please note that only the classic libjpeg works !])
@@ -294,6 +302,7 @@
 fi
 
 have_dv=false
+if test x$with_dv != xno ; then
 EXTRA_LIBS=" $LIBM_LIBS $GLIB_LIBS  $PTHREAD_LIBS"
 AC_CHECK_LIB(dv, dv_decoder_free,
   [ DV_LIBS="-L${with_dv}/lib -ldv ${EXTRA_LIBS}"
@@ -302,6 +311,7 @@
 			  [libdv is present])
     have_dv=true],, 
     -L${with_dv}/lib -ldv ${EXTRA_LIBS})
+fi
 
 dnl *********************************************************************
 dnl Check for libDV YV12 read-support
@@ -364,11 +374,13 @@
 dnl Check for Gtk+/glib (for glav)
 dnl ********************************************************************
 have_gtk=false
-ifdef([AM_PATH_GTK],
-      [ AM_PATH_GTK(1.2.0, [ have_gtk=true
-                           AC_DEFINE(HAVE_GTK,1,
-									 [Gtk library present]
-							)],,) ])
+if test x$with_gtk != xno; then
+ ifdef([AM_PATH_GTK],
+       [ AM_PATH_GTK(1.2.0, [ have_gtk=true
+                            AC_DEFINE(HAVE_GTK,1,
+                                                                        [Gtk library present]
+                                                       )],,) ])
+fi
 
 dnl ********************************************************************
 dnl Check for the SDL library (for software playback)
