--- configure.in	2004-02-03 23:57:29.000000000 +0100
+++ configure.in.nogtk	2004-07-01 13:05:28.279752280 +0200
@@ -364,11 +364,6 @@
 dnl Check for Gtk+/glib (for glav)
 dnl ********************************************************************
 have_gtk=false
-ifdef([AM_PATH_GTK],
-      [ AM_PATH_GTK(1.2.0, [ have_gtk=true
-                           AC_DEFINE(HAVE_GTK,1,
-									 [Gtk library present]
-							)],,) ])
 
 dnl ********************************************************************
 dnl Check for the SDL library (for software playback)
