diff -ru mjpegtools-1.8.0-orig/configure.ac mjpegtools-1.8.0/configure.ac
--- mjpegtools-1.8.0-orig/configure.ac	2006-08-20 20:39:40.719351212 +0200
+++ mjpegtools-1.8.0/configure.ac	2006-08-20 20:20:43.652651808 +0200
@@ -233,6 +233,8 @@
       have_jpeg=true ],,)
 fi
 
+PKG_PROG_PKG_CONFIG
+
 have_libquicktime=false
 if test x$with_libquicktime != xno ; then
   PKG_CHECK_MODULES(LIBQUICKTIME, [libquicktime >= 0.9.4],
