--- mjpegtools-1.6.1.91/lavtools/Makefile.in.old	2003-12-31 01:04:06.000000000 +0100
+++ mjpegtools-1.6.1.91/lavtools/Makefile.in	2003-12-31 01:04:24.000000000 +0100
@@ -334,7 +334,7 @@
 multiblend_flt_SOURCES = multiblend.flt.c
 multiblend_flt_LDADD = $(top_builddir)/utils/libmjpegutils.a
 
-@HAVE_X_TRUE@LAVPLAY_X_LIBS = $(X_LIBS) -lX11
+@HAVE_X_TRUE@LAVPLAY_X_LIBS = $(X_LIBS)
 
 lavplay_SOURCES = lavplay.c
 lavplay_LDADD = $(top_builddir)/utils/libmjpegutils.a \
