--- libssh-0.5.0/src/messages.c.orig
+++ libssh-0.5.0/src/messages.c
@@ -41,7 +41,7 @@
 #include "libssh/keys.h"
 #include "libssh/dh.h"
 #include "libssh/messages.h"
-#if WITH_SERVER
+#ifdef WITH_SERVER
 #include "libssh/server.h"
 #endif
 
@@ -69,6 +69,26 @@
   return msg;
 }
 
+#ifndef WITH_SERVER
+
+/* Reduced version of the reply default that only reply with
+ * SSH_MSG_UNIMPLEMENTED
+ */
+static int ssh_message_reply_default(ssh_message msg) {
+  ssh_log(msg->session, SSH_LOG_FUNCTIONS, "Reporting unknown packet");
+
+  if (buffer_add_u8(msg->session->out_buffer, SSH2_MSG_UNIMPLEMENTED) < 0)
+    goto error;
+  if (buffer_add_u32(msg->session->out_buffer,
+      htonl(msg->session->recv_seq-1)) < 0)
+    goto error;
+  return packet_send(msg->session);
+  error:
+  return SSH_ERROR;
+}
+
+#endif
+
 static int ssh_execute_message_callback(ssh_session session, ssh_message msg) {
     int ret;
     if(session->ssh_message_callback != NULL) {
