 src/grib_ieeefloat.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/grib_ieeefloat.c b/src/grib_ieeefloat.c
index 2190f56..36122a5 100644
--- a/src/grib_ieeefloat.c
+++ b/src/grib_ieeefloat.c
@@ -12,6 +12,7 @@
  *                                                                         *
  ***************************************************************************/
 #include "grib_api_internal.h"
+#include <stdint.h>
 
 #if 1
 
@@ -332,7 +333,7 @@ unsigned long grib_ieee_to_long(double x) {
 #ifdef IEEE
 
 unsigned long grib_ieee64_to_long(double x) {
-  unsigned long lval;
+  uint64_t lval;
 #if IEEE_LE
   unsigned char s[8]={0,};
   unsigned char* buf=(unsigned char*)&x;
