--- src/appl/telnet/configure.in.old	2006-06-28 17:42:55.000000000 +0200
+++ src/appl/telnet/configure.in	2006-06-28 17:43:03.000000000 +0200
@@ -50,11 +50,9 @@
 #endif
 ])
 dnl
-AC_CHECK_LIB(termcap,main,AC_DEFINE(TERMCAP)
-LIBS="$LIBS -ltermcap",
   AC_CHECK_LIB(curses,setupterm,LIBS="$LIBS -lcurses",
   AC_CHECK_LIB(ncurses,setupterm,LIBS="$LIBS -lncurses")
-))
+)
 KRB5_AC_INET6
 AC_CHECK_FUNCS(setupterm)
 AC_CHECK_HEADER(termios.h,AC_DEFINE(USE_TERMIO) ac_termio=1)
@@ -96,11 +94,9 @@
 dnl from old telnetd/configure.in
 dnl
 dnl AC_PROG_INSTALL
-AC_CHECK_LIB(termcap,main,AC_DEFINE(TERMCAP)
-LIBS="$LIBS -ltermcap",
   AC_CHECK_LIB(curses,setupterm,LIBS="$LIBS -lcurses",
   AC_CHECK_LIB(ncurses,setupterm,LIBS="$LIBS -lncurses")
-))
+)
 dnl ... whole termios.h/termio.h/NO_CC_T thing again ...
 AC_HEADER_TIME
 dnl KRB5_AC_INET6
