--- glew/Makefile.orig	2006-08-20 16:13:12.000000000 +0200
+++ glew/Makefile	2006-08-20 16:13:39.000000000 +0200
@@ -94,10 +94,10 @@
 	$(LN) $(LIB.SHARED) lib/$(LIB.DEVLNK)
 endif
 
-bin/$(GLEWINFO.BIN): $(GLEWINFO.BIN.SRCS)
+bin/$(GLEWINFO.BIN): $(GLEWINFO.BIN.SRCS) lib/$(LIB.SHARED)
 	$(CC) $(CFLAGS) -o $@ $^ $(BIN.LIBS)
 
-bin/$(VISUALINFO.BIN): $(VISUALINFO.BIN.SRCS)
+bin/$(VISUALINFO.BIN): $(VISUALINFO.BIN.SRCS) lib/$(LIB.SHARED)
 	$(CC) $(CFLAGS) -o $@ $^ $(BIN.LIBS)
 
 %.o: %.c
