diff -ru zile-1.7-b3/etc/Makefile.in zile-1.7-b3.new/etc/Makefile.in
--- zile-1.7-b3/etc/Makefile.in	2004-02-18 21:22:35.000000000 +0100
+++ zile-1.7-b3.new/etc/Makefile.in	2004-06-24 21:27:18.203145272 +0200
@@ -436,7 +436,7 @@
 	uninstall-zileDATA
 
 
-all-local: HELP AUTODOC refcard.ps
+all-local: HELP AUTODOC
 
 HELP: $(top_srcdir)/doc/zile.texi
 	echo "**********************************************************" >$@
