diff -Naur ParaView3.2.1/VTK/Parallel/CMakeLists.txt ParaView3.2.1.new/VTK/Parallel/CMakeLists.txt
--- ParaView3.2.1/VTK/Parallel/CMakeLists.txt	2007-11-07 15:51:57.000000000 -0500
+++ ParaView3.2.1.new/VTK/Parallel/CMakeLists.txt	2008-02-20 08:51:14.000000000 -0500
@@ -104,6 +104,7 @@
 
 IF (VTK_USE_MPI)
   INCLUDE (${CMAKE_ROOT}/Modules/FindMPI.cmake)
+  INCLUDE(${VTK_SOURCE_DIR}/Parallel/FindOpenmpi.cmake)
   SET ( Kit_SRCS
     ${Kit_SRCS}
     vtkMPICommunicator.cxx
@@ -138,6 +139,12 @@
   IF (MPI_EXTRA_LIBRARY)   
     SET(KIT_LIBS ${KIT_LIBS} "${MPI_EXTRA_LIBRARY}")   
   ENDIF (MPI_EXTRA_LIBRARY) 
+  IF (OPENMPI_LIBRARY) 
+    SET(KIT_LIBS ${KIT_LIBS} "${OPENMPI_LIBRARY}")   
+  ENDIF (OPENMPI_LIBRARY)   
+  IF (OPENMPI_EXTRA_LIBRARY)   
+      SET(KIT_LIBS ${KIT_LIBS} "${OPENMPI_EXTRA_LIBRARY}")   
+  ENDIF (OPENMPI_EXTRA_LIBRARY)
 ENDIF (VTK_USE_MPI)
 
 SET(Kit_EXTRA_SRCS)
diff -Naur ParaView3.2.1/VTK/Parallel/FindOpenmpi.cmake ParaView3.2.1.new/VTK/Parallel/FindOpenmpi.cmake
--- ParaView3.2.1/VTK/Parallel/FindOpenmpi.cmake	1969-12-31 19:00:00.000000000 -0500
+++ ParaView3.2.1.new/VTK/Parallel/FindOpenmpi.cmake	2008-02-20 09:35:06.000000000 -0500
@@ -0,0 +1,18 @@
+# - Find openmpi
+FIND_PATH(MPI_INCLUDE_PATH mpi.h 
+          /usr/include 
+          /usr/include/openmpi
+)
+
+FIND_LIBRARY(OPENMPI_LIBRARY 
+             NAMES mpi
+             PATHS /usr/lib /usr/lib64 
+)
+
+FIND_LIBRARY(OPENMPI_EXTRA_LIBRARY 
+             NAMES mpi_cxx
+             PATHS /usr/lib /usr/lib64 
+             DOC "If a second mpi library is necessary, specify it here.")
+
+MARK_AS_ADVANCED(OPENMPI_LIBRARY OPENMPI_EXTRA_LIBRARY)
+
