diff --git a/TESTING/cdrive.c b/TESTING/cdrive.c
index 0ff08e1..6c5cd36 100644
--- a/TESTING/cdrive.c
+++ b/TESTING/cdrive.c
@@ -78,7 +78,7 @@ main(int argc, char *argv[])
     trans_t        trans;
     SuperLUStat_t  stat;
     static char    matrix_type[8];
-    static char    equed[1], path[3], sym[1], dist[1];
+    static char    equed[1], path[4], sym[1], dist[1];
 
     /* Fixed set of parameters */
     int            iseed[]  = {1988, 1989, 1990, 1991};
diff --git a/TESTING/ddrive.c b/TESTING/ddrive.c
index 587dd59..d2b12f3 100644
--- a/TESTING/ddrive.c
+++ b/TESTING/ddrive.c
@@ -78,7 +78,7 @@ main(int argc, char *argv[])
     trans_t        trans;
     SuperLUStat_t  stat;
     static char    matrix_type[8];
-    static char    equed[1], path[3], sym[1], dist[1];
+    static char    equed[1], path[4], sym[1], dist[1];
 
     /* Fixed set of parameters */
     int            iseed[]  = {1988, 1989, 1990, 1991};
diff --git a/TESTING/sdrive.c b/TESTING/sdrive.c
index b209ae6..d44ff4b 100644
--- a/TESTING/sdrive.c
+++ b/TESTING/sdrive.c
@@ -78,7 +78,7 @@ main(int argc, char *argv[])
     trans_t        trans;
     SuperLUStat_t  stat;
     static char    matrix_type[8];
-    static char    equed[1], path[3], sym[1], dist[1];
+    static char    equed[1], path[4], sym[1], dist[1];
 
     /* Fixed set of parameters */
     int            iseed[]  = {1988, 1989, 1990, 1991};
diff --git a/TESTING/zdrive.c b/TESTING/zdrive.c
index 5844f6a..33bd22e 100644
--- a/TESTING/zdrive.c
+++ b/TESTING/zdrive.c
@@ -78,7 +78,7 @@ main(int argc, char *argv[])
     trans_t        trans;
     SuperLUStat_t  stat;
     static char    matrix_type[8];
-    static char    equed[1], path[3], sym[1], dist[1];
+    static char    equed[1], path[4], sym[1], dist[1];
 
     /* Fixed set of parameters */
     int            iseed[]  = {1988, 1989, 1990, 1991};
