--- octave-2.1.73/configure.in	2006-03-20 16:10:52.000000000 -0500
+++ octave-2.1.73-new/configure.in	2006-10-31 21:12:46.000000000 -0500
@@ -564,7 +564,7 @@
     CONFLIB_ARG="-L. -lconflib"
   fi
   rm -f conftest*
-  AC_CHECK_LIB(f2c, f_open, FLIBS=-lf2c, FLIBS=, $CONFLIB_ARG)
+  AC_CHECK_LIB(f2c, f_open, FLIBS="/usr/lib/libf2c.a", FLIBS=, $CONFLIB_ARG -static)
   rm -f $oct_conflib
 
   if test -z "$FLIBS"; then
