--- VirtualBox-3.0.8_OSE/src/VBox/Main/linux/HostHardwareLinux.cpp
+++ VirtualBox-3.0.8_OSE/src/VBox/Main/linux/HostHardwareLinux.cpp
@@ -212,10 +212,8 @@
             && (!success || testing()))
             rc = getDriveInfoFromHal(&mFloppyList, false /* isDVD */, &success);
 #endif /* VBOX_WITH_DBUS defined */
-        if (   RT_SUCCESS(rc)
-            && RT_SUCCESS(VBoxLoadDBusLib())
-            && (!success || testing()))
-            rc = getDriveInfoFromHal(&mFloppyList, false /* isDVD */, &success);
+	if (  RT_SUCCESS(rc) && (!success || testing())) 
+	    rc = getDriveInfoFromSysfs(&mFloppyList, false /* isDVD */, &success);
         /* As with the CDROMs, on Linux we have to take a multi-level approach
          * involving parsing the mount tables. As this is not bulletproof, we
          * give the user the chance to override the detection using an
@@ -1086,6 +1084,7 @@
     LogFlowFunc (("rc=%Rrc, halSuccess=%d\n", rc, halSuccess));
     return rc;
 }
+#endif  /* RT_OS_LINUX && VBOX_WITH_DBUS */
 
 /**
  * Send an SCSI INQUIRY command to a device and return selected information.
@@ -1422,6 +1421,8 @@
     return rc;
 }
 
+#if defined(RT_OS_LINUX) && defined(VBOX_WITH_DBUS)
+
 /**
  * Helper function to query the hal subsystem for information about drives
  * attached to the system.
