diff -Naur alsaplayer-0.99.77-orig/app/AlsaNode.cpp alsaplayer-0.99.77/app/AlsaNode.cpp
--- alsaplayer-0.99.77-orig/app/AlsaNode.cpp	2004-02-15 12:42:13.000000000 -0600
+++ alsaplayer-0.99.77/app/AlsaNode.cpp	2007-02-13 00:14:29.000000000 -0600
@@ -488,7 +488,7 @@
 	looping = false;
 	pthread_mutex_lock(&thread_mutex);
 	if (thread_running) {
-		if (pthread_join(looper_thread, NULL)) {
+		if (looper_thread != NULL && pthread_join(looper_thread, NULL)) {
 			// Hmmm
 		}
 		thread_running = false;
