--- alsaplayer-0.99.70.original/interface/gtk/PlaylistWindow.cpp	Mon May 13 08:31:56 2002
+++ alsaplayer-0.99.70/interface/gtk/PlaylistWindow.cpp	Wed May 22 16:37:52 2002
@@ -149,7 +149,7 @@
 	gtk_clist_freeze(GTK_CLIST(gtkpl->playlist_list));
 	if (item.title.size()) {
 		sprintf(tmp, "%s %s", item.title.c_str(),
-			item.artist.size() ? (string("- ") + item.artist).c_str() : "");
+			item.artist.size() ? (std::string("- ") + item.artist).c_str() : "");
 		gtk_clist_set_text(GTK_CLIST(gtkpl->playlist_list), position,
 				1, g_strdup(tmp));
 	}
@@ -438,7 +438,7 @@
 			filename = (gchar *)g_strdup(new_path);
 	if (item->title.size()) {
 		sprintf(pt, "%s %s", item->title.c_str(), 
-			item->artist.size() ? (string("- ") + item->artist).c_str() :
+			item->artist.size() ? (std::string("- ") + item->artist).c_str() :
 				"");
 	} else {
 		sprintf(pt, "%s", filename);
