--- input/flac/FlacStream.h-dist	2007-04-25 14:44:52.000000000 +0200
+++ input/flac/FlacStream.h	2007-04-25 14:45:11.000000000 +0200
@@ -191,7 +191,7 @@ class FlacStream
 
     FLAC__StreamDecoderReadStatus 
                  realReadCallBack (FLAC__byte buffer[],
-				   unsigned * bytes);
+				   size_t * bytes);
 
 
  protected:
@@ -233,7 +233,7 @@ class FlacStream
     static FLAC__StreamDecoderReadStatus
 	readCallBack (const FLAC__StreamDecoder * decoder,
 		      FLAC__byte buffer[],
-		      unsigned * bytes,
+		      size_t * bytes,
 		      void * client_data);
 
     static void errCallBack (const FLAC__StreamDecoder * decoder,
--- input/flac/FlacStream.cpp-dist	2007-04-25 14:44:10.000000000 +0200
+++ input/flac/FlacStream.cpp	2007-04-25 14:44:49.000000000 +0200
@@ -321,7 +321,7 @@
 
 
 FLAC__StreamDecoderReadStatus
-FlacStream::realReadCallBack (FLAC__byte buffer[], unsigned * bytes)
+FlacStream::realReadCallBack (FLAC__byte buffer[], size_t * bytes)
 {
     *bytes = reader_read (buffer, *bytes, _datasource);
     return *bytes > 0 ? FLAC__STREAM_DECODER_READ_STATUS_CONTINUE :
@@ -336,7 +336,7 @@
 FLAC__StreamDecoderReadStatus
 FlacStream::readCallBack (const FLAC__StreamDecoder *,
 			  FLAC__byte buffer[],
-			  unsigned * bytes,
+			  size_t * bytes,
 			  void * client_data)
 {
     if (!client_data)
--- input/flac/FlacSeekableStream.cpp-dist	2007-04-25 14:45:29.000000000 +0200
+++ input/flac/FlacSeekableStream.cpp	2007-04-25 14:45:43.000000000 +0200
@@ -255,7 +255,7 @@
 FlacSeekableStream::readCallBack (const FLAC__StreamDecoder * /*decoder*/,
 #endif
 				  FLAC__byte buffer[],
-				  unsigned * bytes,
+				  size_t * bytes,
 				  void * client_data)
 {
     if (!client_data)
--- input/flac/FlacSeekableStream.h-dist	2007-04-25 14:45:32.000000000 +0200
+++ input/flac/FlacSeekableStream.h	2007-04-25 14:45:53.000000000 +0200
@@ -114,7 +114,7 @@ class FlacSeekableStream : public FlacSt
 	readCallBack (const FLAC__StreamDecoder * decoder,
 #endif
 		      FLAC__byte buffer[],
-		      unsigned * bytes,
+		      size_t * bytes,
 		      void * client_data);
 
 #ifdef LEGACY_FLAC
