diff -ur /var/tmp/portage/kbarcode-1.7.2/work/kbarcode-1.7.2/kbarcode/smalldialogs.cpp /tmp/kbarcode-1.7.2/kbarcode/smalldialogs.cpp
--- /var/tmp/portage/kbarcode-1.7.2/work/kbarcode-1.7.2/kbarcode/smalldialogs.cpp	2004-01-17 11:28:48.000000000 +0100
+++ /tmp/kbarcode-1.7.2/kbarcode/smalldialogs.cpp	2004-03-07 09:34:28.018627744 +0100
@@ -64,7 +64,7 @@
     init();
 }
 
-AddItemsDialog::AddItemsDialog( const QString & a, const QString & g, int c, QWidget* parent = 0, const char* name = 0 )
+AddItemsDialog::AddItemsDialog( const QString & a, const QString & g, int c, QWidget* parent /* = 0*/, const char* name /*= 0*/ )
     : KDialogBase( KDialogBase::Plain, i18n("Edit Item"),
       KDialogBase::Ok| KDialogBase::Close, KDialogBase::Ok, parent,name)
 {
