diff -Nur ../wcstools-3.6.7.orig/getdate.c ./getdate.c
--- ../wcstools-3.6.7.orig/getdate.c	2007-01-10 19:59:34.000000000 +0000
+++ ./getdate.c	2007-02-21 15:33:41.000000000 +0000
@@ -412,7 +412,7 @@
     char *fitsdate, *newfdate, *stdate;
     char temp[64];
     char fyear[16];
-    char ts0[8];
+    char ts0[9];
     char *tchar;
     int its, its1;
     time_t lts;
diff -Nur ../wcstools-3.6.7.orig/libwcs/binread.c ./libwcs/binread.c
--- ../wcstools-3.6.7.orig/libwcs/binread.c	2007-01-10 18:28:19.000000000 +0000
+++ ./libwcs/binread.c	2007-02-21 15:34:40.000000000 +0000
@@ -1372,7 +1372,7 @@
 	for (i = 0; i < nmag; i++) {
 	    moveb (sc->catline, (char *) st->mag, 2, sc->entmag[0]+(i*2), i*2);
 	    if (sc->byteswapped)
-		binswap2 (&st->mag[i], 2);
+		binswap2 ((char*)&st->mag[i], 2);
 	    st->xmag[i] = 0.01 * (double) st->mag[i];
 	    }
 	}
diff -Nur ../wcstools-3.6.7.orig/libwcs/catutil.c ./libwcs/catutil.c
--- ../wcstools-3.6.7.orig/libwcs/catutil.c	2007-01-10 17:33:31.000000000 +0000
+++ ./libwcs/catutil.c	2007-02-21 15:35:46.000000000 +0000
@@ -726,7 +726,7 @@
 	    lname = 0;
 	else
 	    lname = strlen (refcatname);
-	catname = (char *)calloc (lname + 16, 1);
+	catname = (char *)calloc (lname + 37, 1);
 	if (lname > 0)
 	    sprintf (catname, "%s sources", refcatname);
 	else
