--- source/linux/Makefile
+++ source/linux/Makefile
@@ -25,6 +25,7 @@
 endif
 endif
 NOARCH=noarch
+ARCH=
 
 MOUNT_DIR=..
 
@@ -50,7 +51,7 @@
 _LIB := lib
 endif
 
-BASE_CFLAGS=-Dstricmp=strcasecmp -D_stricmp=strcasecmp -D__linux__
+BASE_CFLAGS=$(CFLAGS) -Dstricmp=strcasecmp -D_stricmp=strcasecmp -D__linux__
 
 ifeq ($(ARCH),axp)
 RELEASE_CFLAGS=$(BASE_CFLAGS) -ffast-math -funroll-loops \
@@ -79,8 +80,8 @@
 
 ARENA_CFLAGS=-DARENA
 
-DEBUG_CFLAGS=$(BASE_CFLAGS) -g
-LDFLAGS=-ldl -lm
+DEBUG_CFLAGS=$(BASE_CFLAGS) -fno-strict-aliasing
+LDFLAGS+=-ldl -lm
 SVGALDFLAGS=-lvga -lm
 XLDFLAGS=-L/usr/X11R6/$(_LIB) -lX11 -lXext -lXxf86dga
 XCFLAGS=
@@ -120,10 +121,13 @@
 	$(BUILDDIR)/game$(ARCH).$(SHLIBEXT) \
 	$(BUILDDIR)/arena/game$(ARCH).$(SHLIBEXT)
 else
-TARGETS=$(BUILDDIR)/crded.$(ARCH) \
-	$(BUILDDIR)/crx.$(ARCH) \
-	$(BUILDDIR)/game$(ARCH).$(SHLIBEXT) \
-	$(BUILDDIR)/arena/game$(ARCH).$(SHLIBEXT)
+TARGETS=$(BUILDDIR)/game$(ARCH).$(SHLIBEXT)
+ifeq ($(BUILD_ARENA),1)
+TARGETS+=$(BUILDDIR)/crx.$(ARCH)
+endif
+ifeq ($(BUILD_DED),1)
+TARGETS+=$(BUILDDIR)/crded.$(ARCH)
+endif
 endif
 
 ifeq ($(strip $(SDLSOUND)),1)
