--- source/configure.in.old	2005-08-19 19:16:29.000000000 +0200
+++ source/configure.in	2005-08-31 16:19:41.000000000 +0200
@@ -1911,8 +1911,8 @@
 dnl library directory is there and check the ABI directory first (which
 dnl should be harmless on other systems.
 dnl For IA64 HPUX systems, the libs are located in lib/hpux32 instead of lib.
-    for l in "lib32" "lib" "lib/hpux32"; do
-        if test -d "$i/$l" ; then
+    for l in "lib" "lib/hpux32"; do
+        if test -d "$i/$l" -o -L "$i/$l"; then
                 LDFLAGS="$save_LDFLAGS -L$i/$l"
                 LIBS=
                 export LDFLAGS LIBS CPPFLAGS
