diff -r -u samba-3.0.2a/examples/printing/smbprint samba-3.0.2a-patched/examples/printing/smbprint
--- samba-3.0.2a/examples/printing/smbprint	2004-04-09 13:33:17.304613696 -0700
+++ samba-3.0.2a-patched/examples/printing/smbprint	2004-04-09 14:01:05.817961112 -0700
@@ -109,6 +109,9 @@
         if [ $debug -eq 0 ] ; then
 		debugfile=/dev/null debugargs=
 	else
+        if [ -h "$debugfile" -o ! -f "$debugfile" ]; then
+            debugfile="/dev/null"
+        fi
 		set -x; exec >>$debugfile 2>&1
 		debugargs="$debugfile."
 		#[ "x$debugsmb" == "x" ] || debugargs="$debugargs -d $debugsmb"
diff -r -u samba-3.0.2a/examples/printing/smbprint-new.sh samba-3.0.2a-patched/examples/printing/smbprint-new.sh
--- samba-3.0.2a/examples/printing/smbprint-new.sh	2004-04-09 13:33:24.086582680 -0700
+++ samba-3.0.2a-patched/examples/printing/smbprint-new.sh	2004-04-09 14:01:32.791860456 -0700
@@ -109,6 +109,9 @@
         if [ $debug -eq 0 ] ; then
 		debugfile=/dev/null debugargs=
 	else
+        if [ -h "$debugfile" -o ! -f "$debugfile" ]; then
+            debugfile="/dev/null"
+        fi
 		set -x; exec >>$debugfile 2>&1
 		debugargs="$debugfile."
 		#[ "x$debugsmb" == "x" ] || debugargs="$debugargs -d $debugsmb"
