--- samba-3.0.0beta3/source/python/setup.py     2003-06-07 13:57:36.000000000 -0400
+++ samba_3_0_20030720/source/python/setup.py   2003-08-10 15:52:53.000000000 -0400
@@ -50,13 +50,22 @@
 libraries = []
 library_dirs = []

+next_is_path = 0
 for lib in string.split(samba_libs):
-    if lib[0:2] == "-l":
+    if next_is_path <> 0:
+       library_dirs.append(lib);
+       next_is_path = 0;
+       continue
+    if lib == "-Wl,-rpath":
+       next_is_path = 1;
+       continue
+    elif lib[0:2] == "-l":
         libraries.append(lib[2:])
         continue
-    if lib[0:2] == "-L":
+    elif lib[0:2] == "-L":
         library_dirs.append(lib[2:])
         continue
+
     print "Unknown entry '%s' in $LIBS variable passed to setup.py" % lib
     sys.exit(1)
