diff -u -r1.115.2.150 srv_spoolss_nt.c
--- samba-2.2.3a/source/rpc_server/srv_spoolss_nt.c	12 Feb 2002 15:21:04 -0000	1.115.2.150
+++ samba-2.2.3a/source/rpc_server/srv_spoolss_nt.c	12 Feb 2002 19:10:47 -0000
@@ -4800,7 +4814,7 @@
 		 * lookup previously saved driver initialization info, which is then
 		 * bound to the printer, simulating what happens in the Windows arch.
 		 */
-		if (strequal(printer->info_2->drivername, old_printer->info_2->drivername))
+		if (!strequal(printer->info_2->drivername, old_printer->info_2->drivername))
 			set_driver_init(printer, 2);
 	}
 
