diff -uNr samba-vscan-0.3.5.orig/clamav/vscan-clamav.c samba-vscan-0.3.5/clamav/vscan-clamav.c
--- samba-vscan-0.3.5.orig/clamav/vscan-clamav.c	2004-03-02 08:56:06.000000000 +1100
+++ samba-vscan-0.3.5/clamav/vscan-clamav.c	2004-10-19 12:51:06.290962012 +1000
@@ -567,7 +567,11 @@
 	}
 
 
+#ifdef LIBCLAMAV
+	sockfd = 1;
+#else
 	sockfd = vscan_clamav_init();
+#endif /* LIBCLAMAV */
         if ( sockfd >= 0 ) {
 		safe_strcpy(client_ip, fsp->conn->client_address, CLIENT_IP_SIZE -1);
 		/* scan only file, do nothing */
