diff -ur alsa-utils-1.0.21.orig/alsactl/init/default alsa-utils-1.0.21/alsactl/init/default
--- alsa-utils-1.0.21.orig/alsactl/init/default	2009-05-06 10:07:24.000000000 +0300
+++ alsa-utils-1.0.21/alsactl/init/default	2009-08-17 22:29:46.000000000 +0300
@@ -49,7 +49,7 @@
 CTL{name}="Headphone Playback Volume",PROGRAM=="__ctl_search",GOTO="headphone0_end"
 # if master volume control is present, turn headphone volume to max
 ENV{has_pmaster_vol}=="true",CTL{values}="0dB",RESULT=="0",GOTO="headphone0_end"
-ENV{has_pmaster_vol}=="true",CTL{values)="100%",GOTO="headphone0_end"
+ENV{has_pmaster_vol}=="true",CTL{values}="100%",GOTO="headphone0_end"
 CTL{values}="$env{pvolume}",RESULT!="0",CTL{values}="$env{ppercent}"
 LABEL="headphone0_end"
 CTL{name}="Headphone Playback Switch",PROGRAM=="__ctl_search", \
