diff -ur alsa-utils-1.0.21.orig/alsactl/init/default alsa-utils-1.0.21/alsactl/init/default
--- alsa-utils-1.0.21.orig/alsactl/init/default	2009-05-06 10:07:24.000000000 +0300
+++ alsa-utils-1.0.21/alsactl/init/default	2009-08-17 22:29:46.000000000 +0300
@@ -39,7 +39,7 @@
 CTL{name}="Front Playback Volume",PROGRAM!="__ctl_search",GOTO="front0_end"
 # if master volume control is present, turn front volume to max
 ENV{has_pmaster_vol}=="true",CTL{values}="0dB",RESULT=="0",GOTO="front0_end"
-ENV{has_pmaster_vol}=="true",CTL{values)="100%",GOTO="front0_end"
+ENV{has_pmaster_vol}=="true",CTL{values}="100%",GOTO="front0_end"
 CTL{values}="$env{pvolume}",RESULT!="0",CTL{values}="$env{ppercent}"
 LABEL="front0_end"
 CTL{name}="Front Playback Switch",PROGRAM=="__ctl_search", \
@@ -49,7 +49,7 @@
 CTL{name}="Headphone Playback Volume",PROGRAM!="__ctl_search",GOTO="headphone0_end"
 # if master volume control is present, turn headphone volume to max
 ENV{has_pmaster_vol}=="true",CTL{values}="0dB",RESULT=="0",GOTO="headphone0_end"
-ENV{has_pmaster_vol}=="true",CTL{values)="100%",GOTO="headphone0_end"
+ENV{has_pmaster_vol}=="true",CTL{values}="100%",GOTO="headphone0_end"
 CTL{values}="$env{pvolume}",RESULT!="0",CTL{values}="$env{ppercent}"
 LABEL="headphone0_end"
 CTL{name}="Headphone Playback Switch",PROGRAM=="__ctl_search", \
@@ -72,7 +72,7 @@
  CTL{name}="PCM Volume",PROGRAM!="__ctl_search", GOTO="pcm0_end"
 # if master volume control is present, turn PCM volume to max
 ENV{has_pmaster_vol}=="true",CTL{values}="0dB",RESULT=="0",GOTO="pcm0_end"
-ENV{has_pmaster_vol}=="true",CTL{values)="100%",GOTO="pcm0_end"
+ENV{has_pmaster_vol}=="true",CTL{values}="100%",GOTO="pcm0_end"
 # exception - some HDA codecs have shifted dB range
 CTL{dBmin}=="-34.50dB",CTL{dBmax}=="12.00dB",CTL{values}="0dB",GOTO="pcm0_end"
 CTL{dBmin}=="-30.00dB",CTL{dBmax}=="0dB",CTL{values}="0dB",GOTO="pcm0_end"
@@ -86,7 +86,7 @@
   CTL{name}="PCM Volume",PROGRAM!="__ctl_search",GOTO="pcm1_end"
 # if master volume control is present, turn PCM volume to max
 ENV{has_pmaster_vol}=="true",CTL{values}="0dB",RESULT=="0",GOTO="pcm1_end"
-ENV{has_pmaster_vol}=="true",CTL{values)="100%",GOTO="pcm1_end"
+ENV{has_pmaster_vol}=="true",CTL{values}="100%",GOTO="pcm1_end"
 # exception - some HDA codecs have shifted dB range
 CTL{dBmin}=="-34.50dB",CTL{dBmax}=="12.00dB",CTL{values}="0dB",GOTO="pcm1_end"
 CTL{dBmin}=="-30.00dB",CTL{dBmax}=="0dB",CTL{values}="0dB",GOTO="pcm1_end"
@@ -125,7 +125,7 @@
 CTL{name}="CD Playback Volume",PROGRAM!="__ctl_search", GOTO="cd0_end"
 # if master volume control is present, turn CD volume to max
 ENV{has_pmaster_vol}=="true",CTL{values}="0dB",RESULT=="0",GOTO="cd0_end"
-ENV{has_pmaster_vol}=="true",CTL{values)="100%",GOTO="cd0_end"
+ENV{has_pmaster_vol}=="true",CTL{values}="100%",GOTO="cd0_end"
 # exception - some HDA codecs have shifted dB range
 CTL{dBmin}=="-34.50dB",CTL{dBmax}=="12.00dB",CTL{values}="0dB",GOTO="cd0_end"
 CTL{dBmin}=="-30.00dB",CTL{dBmax}=="0dB",CTL{values}="0dB",GOTO="cd0_end"
