--- version-0.79/vutil/vutil.h	2010-01-10 14:08:42.000000000 +0000
+++ version-0.79/vutil/vutil.h	2010-01-11 14:00:25.700941525 +0000
@@ -41,7 +41,7 @@
 	(a != Perl_prescan_version(aTHX_ a, b, NULL, NULL, NULL, NULL))
 
 #define PERL_ARGS_ASSERT_PRESCAN_VERSION	\
-	assert(s); assert(strict); assert(sqv); assert(ssaw_period);\
+	assert(s); assert(sqv); assert(ssaw_decimal);\
 	assert(swidth); assert(salpha);
 #define PERL_ARGS_ASSERT_SCAN_VERSION	\
 	assert(s); assert(rv)
