diff -ur armagetronad-0.2.7.1-orig/src/engine/eGrid.h armagetronad-0.2.7.1/src/engine/eGrid.h
--- armagetronad-0.2.7.1-orig/src/engine/eGrid.h	2005-02-03 17:26:03.000000000 +0100
+++ armagetronad-0.2.7.1/src/engine/eGrid.h	2006-06-04 20:06:17.000000000 +0200
@@ -39,6 +39,7 @@
 class ePoint;
 class eFace;
 class eHalfEdge;
+class eGameObject;
 class eWall;
 class eGrid;
 class eWallView;
diff -ur armagetronad-0.2.7.1-orig/src/engine/eVoter.cpp armagetronad-0.2.7.1/src/engine/eVoter.cpp
--- armagetronad-0.2.7.1-orig/src/engine/eVoter.cpp	2005-03-02 23:25:18.000000000 +0100
+++ armagetronad-0.2.7.1/src/engine/eVoter.cpp	2006-06-04 20:13:54.000000000 +0200
@@ -39,6 +39,8 @@
 #include "ePlayer.h"
 #include "eGrid.h"
 
+class eMenuItemVote;
+
 static unsigned short se_votingItemID = 0;
 static float se_votingTimeout = 300.0f;
 static nSettingItem< float > se_vt( "VOTING_TIMEOUT", se_votingTimeout );
diff -ur armagetronad-0.2.7.1-orig/src/engine/eWall.h armagetronad-0.2.7.1/src/engine/eWall.h
--- armagetronad-0.2.7.1-orig/src/engine/eWall.h	2005-02-01 17:54:13.000000000 +0100
+++ armagetronad-0.2.7.1/src/engine/eWall.h	2006-06-04 19:49:36.000000000 +0200
@@ -53,6 +53,8 @@
 
 
 class eWall;
+class eWallView;
+class eWallHolder;
 class eGameObject;
 
 class eWallView:public tHeapElement{
diff -ur armagetronad-0.2.7.1-orig/src/network/nPriorizing.h armagetronad-0.2.7.1/src/network/nPriorizing.h
--- armagetronad-0.2.7.1-orig/src/network/nPriorizing.h	2004-11-29 11:26:42.000000000 +0100
+++ armagetronad-0.2.7.1/src/network/nPriorizing.h	2006-06-04 20:16:45.000000000 +0200
@@ -36,6 +36,7 @@
 class nSendBuffer;
 class nBandwidthControl;
 class nBandwidthTask;
+class nBandwidthTaskPriorizer;
 
 tDECLARE_REFOBJ( nBandwidthTask );
 tDECLARE_REFOBJ( nBandwidthArbitrator );
diff -ur armagetronad-0.2.7.1-orig/src/tron/gCycle.h armagetronad-0.2.7.1/src/tron/gCycle.h
--- armagetronad-0.2.7.1-orig/src/tron/gCycle.h	2005-03-06 08:26:03.000000000 +0100
+++ armagetronad-0.2.7.1/src/tron/gCycle.h	2006-06-04 19:59:52.000000000 +0200
@@ -40,6 +40,10 @@
 class gTextureCycle;
 class eSoundPlayer;
 class gSensor;
+class gCycle;
+class gPlayerWall;
+class gNetPlayerWall;
+class gDestination;
 
 // minimum time between two cycle turns
 extern REAL sg_delayCycle;
