--- xspin.tcl.orig	2010-06-11 16:29:55.000000000 +0200
+++ xspin.tcl	2010-06-22 19:43:37.000000000 +0200
@@ -82,7 +82,7 @@
 
 #-- GUI configuration options - by Leszek Holenderski <lhol@win.tue.nl>
 #-- basic text size:
-	set HelvBig	-Adobe-Helvetica-Medium-R-Normal--*-120-*-*-*-*-*-*
+	set HelvBig	-Adobe-Helvetica-Medium-R-Normal--*-80-*-*-*-*-*-*
 # mscs:
 	if {$NT} {	;# on a windows nt machine
 	set SmallFont	"-*-Courier-Bold-R-Normal--*-110-*"
@@ -157,7 +157,7 @@
 #--- fonts
 	# fonts for the input/log area (default is system dependent,
 	# usually -*-Courier-Medium-R-Normal--*-120-*)
-option add *Input*Text.font	-*-Helvetica-Medium-R-Normal--*-120-*	startupFile
+option add *Input*Text.font	-*-Helvetica-Medium-R-Normal--*-80-*	startupFile
 #option add *Input*Text.font	-schumacher-clean-medium-r-normal--*-120-*-60-*	startupFile
 #option add *Log*Text.font	-schumacher-clean-medium-r-normal--*-120-*-60-*	startupFile
 
@@ -168,7 +168,7 @@
 	option add *SimpleText.Text.background white
 
 	# sections (decorated frames for grouping related buttons)
-	option add *Section*Title.font		-*-Helvetica-Bold-R-Normal--*-100-*	startupFile
+	option add *Section*Title.font		-*-Helvetica-Bold-R-Normal--*-60-*	startupFile
 
 #option add *Section*Checkbutton.font	-*-Helvetica-Medium-R-Normal--*-100-*	startupFile
 #option add *Section*Radiobutton.font	-*-Helvetica-Medium-R-Normal--*-100-*	startupFile
@@ -480,7 +480,7 @@
 	.menu.edit.m add command -label "Paste" \
 		-command {tk_textPaste .inp.t}
 
-set FSz 110
+set FSz 80
 
 menu .menu.view.m
 	.menu.view.m add command -label "Larger" \
@@ -491,7 +491,7 @@
 		}
 	.menu.view.m add command -label "Default text size" \
 		-command {
-			set FSz 110
+			set FSz 80
 			set HelvBig "-Adobe-Helvetica-Medium-R-Normal--*-$FSz-*-*-*-*-*-*"
 			.inp.t configure -font $HelvBig
 		}
