diff -Nurpa mapson-3.0.orig/extract-addresses.cc mapson-3.0/extract-addresses.cc
--- mapson-3.0.orig/extract-addresses.cc	2008-12-17 03:59:36.000000000 +0300
+++ mapson-3.0/extract-addresses.cc	2008-12-17 04:02:19.000000000 +0300
@@ -12,6 +12,7 @@
 #include "extract-addresses.hh"
 #include "log.hh"
 #include "config.hh"
+#include <cstring>
 
 using namespace std;
 
diff -Nurpa mapson-3.0.orig/libvarexp/search-and-replace.cc mapson-3.0/libvarexp/search-and-replace.cc
--- mapson-3.0.orig/libvarexp/search-and-replace.cc	2008-12-17 03:59:36.000000000 +0300
+++ mapson-3.0/libvarexp/search-and-replace.cc	2008-12-17 04:00:16.000000000 +0300
@@ -1,6 +1,7 @@
 #include <sys/types.h>
 #include <regex.h>
 #include "internal.hh"
+#include <cstring>
 
 namespace varexp
     {
diff -Nurpa mapson-3.0.orig/RegExp/RegExp.hh mapson-3.0/RegExp/RegExp.hh
--- mapson-3.0.orig/RegExp/RegExp.hh	2008-12-17 03:59:36.000000000 +0300
+++ mapson-3.0/RegExp/RegExp.hh	2008-12-17 04:02:46.000000000 +0300
@@ -8,7 +8,7 @@
 
 #include <stdexcept>
 #include <vector>
-#include <string>
+#include <cstring>
 #include <memory>
 #include <sys/types.h>
 #include <regex.h>
diff -Nurpa mapson-3.0.orig/system-error/system-error.hh mapson-3.0/system-error/system-error.hh
--- mapson-3.0.orig/system-error/system-error.hh	2008-12-17 03:59:36.000000000 +0300
+++ mapson-3.0/system-error/system-error.hh	2008-12-17 04:01:28.000000000 +0300
@@ -9,7 +9,7 @@
 // ISO C++ headers.
 #include <stdexcept>
 #include <cerrno>
-#include <string>
+#include <cstring>
 
 struct system_error : public std::runtime_error
     {
