diff -urN quirc-0.9.83/server.cc quirc-0.9.83-modified/server.cc
--- quirc-0.9.83/server.cc	2002-10-02 01:19:22.000000000 -0500
+++ quirc-0.9.83-modified/server.cc	2002-10-21 23:54:37.000000000 -0500
@@ -11,7 +11,8 @@
 #include <ctype.h>
 #include <netdb.h> 
 #include <stdarg.h>
-#include <iostream.h>
+
+using namespace std;
 
 #include "config.h"
 
diff -urN quirc-0.9.83/servertcl.cc quirc-0.9.83-modified/servertcl.cc
--- quirc-0.9.83/servertcl.cc	2002-10-02 01:18:59.000000000 -0500
+++ quirc-0.9.83-modified/servertcl.cc	2002-10-21 23:53:04.000000000 -0500
@@ -1,6 +1,7 @@
 #include <sys/types.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
+#include <ctype.h>
 
 #ifdef DMALLOC
 #include <dmalloc.h>
diff -urN quirc-0.9.83/tcl.cc quirc-0.9.83-modified/tcl.cc
--- quirc-0.9.83/tcl.cc	2002-10-01 03:32:32.000000000 -0500
+++ quirc-0.9.83-modified/tcl.cc	2002-10-21 23:52:06.000000000 -0500
@@ -1,6 +1,7 @@
 #include <sys/time.h>
 #include <unistd.h>
 #include <sys/socket.h>
+#include <ctype.h>
 
 #ifdef DMALLOC
 #include <dmalloc.h>
