--- vdradmind.pl.orig	2006-10-19 13:59:19.000000000 +0200
+++ vdradmind.pl	2006-10-19 13:59:54.000000000 +0200
@@ -343,6 +343,7 @@
     if (/--config|-c/) {
         ReadConfig() if (-e $CONFFILE);
         LoadTranslation();
+	STDOUT->autoflush(1);
         $CONFIG{VDR_HOST}   = Question(gettext("What's your VDR hostname (e.g video.intra.net)?"),               $CONFIG{VDR_HOST});
         $CONFIG{VDR_PORT}   = Question(gettext("On which port does VDR listen to SVDRP queries?"),               $CONFIG{VDR_PORT});
         $CONFIG{SERVERHOST} = Question(gettext("On which address should VDRAdmin-AM listen (0.0.0.0 for any)?"), $CONFIG{SERVERHOST});
