--- Makefile.orig	2005-11-12 19:03:18.713964552 +0000
+++ Makefile	2005-11-12 19:11:19.461879760 +0000
@@ -3,8 +3,8 @@
 PACKAGE=\"libspeex\"
 
 INCDIRS=`xmms-config --cflags`
-CFLAGS = -Wall
-OBJS = http.o config.o fileinfo.o speexutil.o # utf8.o
+CFLAGS = -Wall -fPIC
+OBJS = http.o config.o fileinfo.o speexutil.o utf8.o
 GUIOBJS = gui/interface.o gui/support.o gui/callbacks.o
 GUISRCS = gui/interface.c gui/interface.h gui/support.c gui/support.h gui/callbacks.c gui/callbacks.h
 
@@ -16,8 +16,8 @@
 libspeex.so: libspeex.c libspeex.h $(OBJS) $(GUIOBJS)
 	$(CC) -shared -logg -lspeex -lgtk -lpthread -lxmms $(INCDIRS) $(CFLAGS) -DVERSION=$(VERSION) -DPACKAGE=\"libspeex\" -o libspeex.so libspeex.c $(OBJS) $(GUIOBJS)
 
-#utf8.o: utf8.c utf8.h
-#	$(CC) -c $(INCDIRS) $(CFLAGS) -DHAVE_ICONV -DICONV_CONST= -o utf8.o utf8.c
+utf8.o: utf8.c utf8.h
+	$(CC) -c $(INCDIRS) $(CFLAGS) -DHAVE_ICONV -DICONV_CONST= -o utf8.o utf8.c
 
 http.o: http.c http.h
 	$(CC) -c $(INCDIRS) $(CFLAGS) -DVERSION=$(VERSION) -DPACKAGE=$(PACKAGE) -o http.o http.c
@@ -33,7 +33,8 @@
 
 install: libspeex.so
 	install libspeex.so `xmms-config --input-plugin-dir`
-
+	install -d $(DESTDIR)`xmms-config --input-plugin-dir`
+	install libspeex.so $(DESTDIR)`xmms-config --input-plugin-dir`
 clean:
 	$(MAKE) -C gui/ clean
 	rm -f *.so *.o
