diff -Naur pspp-0.6.0/Makefile.in pspp-0.6.0.new/Makefile.in
--- pspp-0.6.0/Makefile.in	2008-06-05 00:28:32.000000000 -0400
+++ pspp-0.6.0.new/Makefile.in	2008-06-29 15:53:51.000000000 -0400
@@ -126,7 +126,7 @@
 	$(top_srcdir)/gl/m4/iconv_h.m4 \
 	$(top_srcdir)/gl/m4/iconv_open.m4 \
 	$(top_srcdir)/gl/m4/include_next.m4 \
-	$(top_srcdir)/gl/m4/inline.m4 $(top_srcdir)/gl/m4/intmax_t.m4 \
+	$(top_srcdir)/gl/m4/intmax_t.m4 \
 	$(top_srcdir)/gl/m4/inttostr.m4 \
 	$(top_srcdir)/gl/m4/isfinite.m4 $(top_srcdir)/gl/m4/isnand.m4 \
 	$(top_srcdir)/gl/m4/isnanf.m4 $(top_srcdir)/gl/m4/isnanl.m4 \
diff -Naur pspp-0.6.0/aclocal.m4 pspp-0.6.0.new/aclocal.m4
--- pspp-0.6.0/aclocal.m4	2008-06-05 00:28:16.000000000 -0400
+++ pspp-0.6.0.new/aclocal.m4	2008-06-29 15:53:51.000000000 -0400
@@ -7746,7 +7746,6 @@
 m4_include([gl/m4/iconv_h.m4])
 m4_include([gl/m4/iconv_open.m4])
 m4_include([gl/m4/include_next.m4])
-m4_include([gl/m4/inline.m4])
 m4_include([gl/m4/intmax_t.m4])
 m4_include([gl/m4/inttostr.m4])
 m4_include([gl/m4/isfinite.m4])
diff -Naur pspp-0.6.0/gl/Makefile.in pspp-0.6.0.new/gl/Makefile.in
--- pspp-0.6.0/gl/Makefile.in	2008-06-05 00:28:30.000000000 -0400
+++ pspp-0.6.0.new/gl/Makefile.in	2008-06-29 15:53:51.000000000 -0400
@@ -79,7 +79,7 @@
 	$(top_srcdir)/gl/m4/iconv_h.m4 \
 	$(top_srcdir)/gl/m4/iconv_open.m4 \
 	$(top_srcdir)/gl/m4/include_next.m4 \
-	$(top_srcdir)/gl/m4/inline.m4 $(top_srcdir)/gl/m4/intmax_t.m4 \
+	$(top_srcdir)/gl/m4/intmax_t.m4 \
 	$(top_srcdir)/gl/m4/inttostr.m4 \
 	$(top_srcdir)/gl/m4/isfinite.m4 $(top_srcdir)/gl/m4/isnand.m4 \
 	$(top_srcdir)/gl/m4/isnanf.m4 $(top_srcdir)/gl/m4/isnanl.m4 \
diff -Naur pspp-0.6.0/gl/m4/gnulib-comp.m4 pspp-0.6.0.new/gl/m4/gnulib-comp.m4
--- pspp-0.6.0/gl/m4/gnulib-comp.m4	2008-06-05 00:24:02.000000000 -0400
+++ pspp-0.6.0.new/gl/m4/gnulib-comp.m4	2008-06-29 16:01:31.000000000 -0400
@@ -95,7 +95,6 @@
   AM_ICONV
   gl_ICONV_H
   gl_FUNC_ICONV_OPEN
-  gl_INLINE
   gl_INTTOSTR
   gl_ISFINITE
   gl_MATH_MODULE_INDICATOR([isfinite])
@@ -625,7 +624,6 @@
   m4/iconv_h.m4
   m4/iconv_open.m4
   m4/include_next.m4
-  m4/inline.m4
   m4/intmax_t.m4
   m4/inttostr.m4
   m4/inttypes_h.m4
diff -Naur pspp-0.6.0/gl/m4/inline.m4 pspp-0.6.0.new/gl/m4/inline.m4
--- pspp-0.6.0/gl/m4/inline.m4	2008-05-13 12:20:04.000000000 -0400
+++ pspp-0.6.0.new/gl/m4/inline.m4	1969-12-31 19:00:00.000000000 -0500
@@ -1,40 +0,0 @@
-# inline.m4 serial 3
-dnl Copyright (C) 2006 Free Software Foundation, Inc.
-dnl This file is free software; the Free Software Foundation
-dnl gives unlimited permission to copy and/or distribute it,
-dnl with or without modifications, as long as this notice is preserved.
-
-dnl Test for the 'inline' keyword or equivalent.
-dnl Define 'inline' to a supported equivalent, or to nothing if not supported,
-dnl like AC_C_INLINE does.  Also, define HAVE_INLINE if 'inline' or an
-dnl equivalent is effectively supported, i.e. if the compiler is likely to
-dnl drop unused 'static inline' functions.
-AC_DEFUN([gl_INLINE],
-[
-  AC_REQUIRE([AC_C_INLINE])
-  AC_CACHE_CHECK([whether the compiler generally respects inline],
-    [gl_cv_c_inline_effective],
-    [if test $ac_cv_c_inline = no; then
-       gl_cv_c_inline_effective=no
-     else
-       dnl GCC defines __NO_INLINE__ if not optimizing or if -fno-inline is
-       dnl specified.
-       dnl Use AC_COMPILE_IFELSE here, not AC_EGREP_CPP, because the result
-       dnl depends on optimization flags, which can be in CFLAGS.
-       dnl (AC_EGREP_CPP looks only at the CPPFLAGS.)
-       AC_COMPILE_IFELSE(
-         [AC_LANG_PROGRAM([[]],
-           [[#ifdef __NO_INLINE__
-               #error "inline is not effective"
-             #endif]])],
-         [gl_cv_c_inline_effective=yes],
-         [gl_cv_c_inline_effective=no])
-     fi
-    ])
-  if test $gl_cv_c_inline_effective = yes; then
-    AC_DEFINE([HAVE_INLINE], 1,
-      [Define to 1 if the compiler supports one of the keywords
-       'inline', '__inline__', '__inline' and effectively inlines
-       functions marked as such.])
-  fi
-])
diff -Naur pspp-0.6.0/gl/m4/xalloc.m4 pspp-0.6.0.new/gl/m4/xalloc.m4
--- pspp-0.6.0/gl/m4/xalloc.m4	2008-05-13 12:20:04.000000000 -0400
+++ pspp-0.6.0.new/gl/m4/xalloc.m4	2008-06-29 16:01:17.000000000 -0400
@@ -14,7 +14,6 @@
 
 # Prerequisites of lib/xalloc.h.
 AC_DEFUN([gl_PREREQ_XALLOC], [
-  AC_REQUIRE([gl_INLINE])
   :
 ])
 
