--- src/lib/MainWindow.py.old	2002-11-02 17:12:40.000000000 +0100
+++ src/lib/MainWindow.py	2002-11-02 17:13:00.000000000 +0100
@@ -630,10 +630,10 @@
         self._item_view.display_item(item)
 
     def display_status_message(self, m):
-        cid = self._status_view.get_context_id("asdf")
-        self._status_view.pop(cid)
-        if len(m):
-            self._status_view.push(cid, m)
+        #cid = self._status_view.get_context_id("asdf")
+        #self._status_view.pop(cid)
+        #if len(m):
+        #    self._status_view.push(cid, m)
         return
 
     def number_of_new(self, feed):
