--- icd-1.0.0/icd_bridge.c.orig	2005-06-15 20:44:32.000000000 +0200
+++ icd-1.0.0/icd_bridge.c	2005-06-15 20:44:58.000000000 +0200
@@ -605,7 +605,7 @@
                *data = '\0';
                data++;
                /* BCA - Is ULAW the only format we support? */
-               chan = ast_request(type, AST_FORMAT_ULAW, data);
+               chan = ast_request(type, AST_FORMAT_ULAW, data, NULL);
           }
           ICD_STD_FREE(type);
      }
