diff -urN agent/Makefile.in agent.parallel/Makefile.in
--- agent/Makefile.in	2001-10-02 02:38:03.000000000 +0200
+++ agent.parallel/Makefile.in	2003-09-23 23:34:29.002308952 +0200
@@ -62,15 +62,20 @@
 getmibstat.o: mibgroup/kernel_sunos5.c
 	$(CC) $(CFLAGS) -o $@ -D_GETMIBSTAT_TEST -DDODEBUG -c $? 
 
-snmpd$(EXEEXT):	${LAGENTOBJS} $(USELIBS) $(LIBTARG)
+snmpd.fake: ${LIBAGENTOBJS} ${AGENTOBJS}
+
+snmpd$(EXEEXT):	snmpd.fake ${LAGENTOBJS} $(USELIBS) $(LIBTARG)
 	$(LINK) $(CFLAGS) -o $@ ${LAGENTOBJS} $(LOCAL_LIBS) ${LDFLAGS} ${AGENT_LIBS}  
 
+libucdagent.fake: ${LIBAGENTOBJS}
 
-libucdagent.$(LIB_EXTENSION)$(LIB_VERSION):    ${LLIBAGENTOBJS} libucdmibs.$(LIB_EXTENSION)$(LIB_VERSION)
+libucdagent.$(LIB_EXTENSION)$(LIB_VERSION):    libucdagent.fake ${LLIBAGENTOBJS} libucdmibs.$(LIB_EXTENSION)$(LIB_VERSION)
 	$(LIB_LD_CMD) libucdagent.$(LIB_EXTENSION)$(LIB_VERSION) ${LLIBAGENTOBJS}  $(LIB_LD_LIBS)
 	$(RANLIB) libucdmibs.$(LIB_EXTENSION)$(LIB_VERSION)
 
-libucdmibs.$(LIB_EXTENSION)$(LIB_VERSION):    ${LMIBOBJS}
+libucdmibs.fake: ${MIBOBJS}
+
+libucdmibs.$(LIB_EXTENSION)$(LIB_VERSION):    libucdmibs.fake ${LMIBOBJS}
 	$(LIB_LD_CMD) libucdmibs.$(LIB_EXTENSION)$(LIB_VERSION) ${LMIBOBJS}  $(LIB_LD_LIBS)
 	$(RANLIB) libucdmibs.$(LIB_EXTENSION)$(LIB_VERSION)
 
diff -urN agent/mibgroup/Makefile.in agent.parallel/mibgroup/Makefile.in
--- agent/mibgroup/Makefile.in	2001-10-02 02:38:03.000000000 +0200
+++ agent.parallel/mibgroup/Makefile.in	2003-09-23 23:34:39.344736664 +0200
@@ -13,7 +13,9 @@
 LOBJS = @module_list_lo@
 SRCS = @module_list_c@
 
-all: $(LOBJS)
+all.fake: $(OBJS)
+
+all: all.fake $(LOBJS)
 
 TAGS: $(SRCS)
 	etags *.c *.h
