--- powertweak-0.99.4.orig/src/libpowertweak/profile.c
+++ powertweak-0.99.4/src/libpowertweak/profile.c
@@ -11,7 +11,7 @@
 #include <stdlib.h>
 #include <assert.h>
 #include <string.h>
-#include <parser.h>		/* libxml */
+#include <libxml/parser.h>		/* libxml */
 
 
 #include <powertweak.h>

--- powertweak-0.99.4.orig/src/backends/PCI/xml.c
+++ powertweak-0.99.4/src/backends/PCI/xml.c
@@ -13,7 +13,7 @@
 #include <assert.h>
 #include <powertweak.h>
 #include <string.h>
-#include <parser.h>
+#include <libxml/parser.h>
 
 #include "pci.h"

--- powertweak-0.99.4.orig/src/backends/CPU/x86/xml.c
+++ powertweak-0.99.4/src/backends/CPU/x86/xml.c
@@ -17,7 +17,7 @@
 #include <assert.h>

 #include <powertweak.h>
-#include <parser.h>            /* libxml */
+#include <libxml/parser.h>             /* libxml */
 #include <libxmlcompat.h>

 #include <sys/types.h>         /* for readdir() */

--- powertweak-0.99.4.orig/src/backends/PCI/pcitweaks.c
+++ powertweak-0.99.4/src/backends/PCI/pcitweaks.c
@@ -21,7 +21,7 @@
 #include "pci.h"
 #include "PCILIB/pciutils.h"
 
-#include <tree.h>
+#include <libxml/tree.h>
 
 extern int xmlDoValidityCheckingDefaultValue;
 
--- powertweak-0.99.4.orig/src/backends/proc/xml.c
+++ powertweak-0.99.4/src/backends/proc/xml.c
@@ -20,7 +20,7 @@
 #include <dirent.h>		/* for readdir() */
 
 #include <powertweak.h>
-#include <parser.h>		/* libxml */
+#include <libxml/parser.h>	/* libxml */
 #include <libxmlcompat.h>
 
 #include "proc.h"
--- powertweak-0.99.4.orig/src/backends/proc/proctweaks.c
+++ powertweak-0.99.4/src/backends/proc/proctweaks.c
@@ -18,7 +18,7 @@
 #include <config.h>
 #include <string.h>
 #include "proc.h"
-#include <tree.h>
+#include <libxml/tree.h>
 #include <libxmlcompat.h>


--- powertweak-0.99.4.orig/src/include/libxmlcompat.h
+++ powertweak-0.99.4/src/include/libxmlcompat.h
@@ -12,7 +12,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
-#include <tree.h> /* libxml */
+#include <libxml/tree.h> /* libxml */
 
 /* Check if this is 1.8 or 2.x */
 #ifdef LIBXML_VERSION_NUMBER

--- powertweak-0.99.4.orig/src/backends/CPU/x86/cputweaks.c    Tue Aug  6 20:39:01 2002
+++ powertweak-0.99.4/src/backends/CPU/x86cputweaks.c Tue Aug  6 20:39:09 2002
@@ -13,7 +13,7 @@
 #include <powertweak.h>
 #include <config.h>
 #include <string.h>
-#include <tree.h>
+#include <libxml/tree.h>
 #include <libxmlcompat.h>

 /* for readdir/stat and friends */

