--- crrcsim-0.9.11/configure.ac
+++ crrcsim-0.9.11/configure.ac
@@ -79,8 +79,12 @@
 dnl Default is: no portaudio present
 portaudio=0
 
+AC_ARG_WITH([portaudio], AS_HELP_STRING([--without-portaudio], [Build without portaudio library (default: test)]))
+
+AS_IF([test "x$with_portaudio" != "xno"], [
 dnl Check for the Portaudio header file
 AC_CHECK_HEADER([portaudio.h])
+])
 
 if  (test "x$ac_cv_header_portaudio_h" = "xyes"); then 
   dnl Check for Portaudio 18
@@ -116,9 +120,11 @@
 fi
 AC_SUBST(PA_LIBS)
 
+AC_ARG_WITH([sdl], AS_HELP_STRING([--without-sdl], [Build without sdl library (default: test)]))
 dnl Search for SDL 1.2.5 or newer
 SDL_VERSION=1.2.5
 sdl_mousewheel="yes  (SDL >= 1.2.5)"
+AS_IF([test "x$with_foo" != "xno"], [
 AM_PATH_SDL($SDL_VERSION,:,AC_MSG_NOTICE([*** SDL version $SDL_VERSION or newer not found! See if at least 1.2.0 is present...]))
 if test "x$no_sdl" = "xyes"
 then
@@ -129,6 +135,10 @@
   AC_DEFINE([SDL_WITHOUT_MOUSEWHEEL], [1], [SDL < 1.2.5 did not feature mousewheel button macros])
   sdl_mousewheel="no   (SDL < 1.2.5)"
 fi
+])
+AS_IF([test "x$with_foo" != "xyes"], [
+  AC_DEFINE([SDL_WITHOUT_MOUSEWHEEL], [1], [SDL disabled])
+])
 AC_SUBST(SDL_LIBS)
 AC_SUBST(SDL_CFLAGS)
 
