http://bugs.gentoo.org/216804

--- src/gcursor.c
+++ src/gcursor.c
@@ -16,6 +16,9 @@
 #include <gconf/gconf-client.h>
 #include <glade/glade.h>
 
+#define CONNECT_SIGNAL(name) glade_xml_signal_connect(main_window, #name, \
+		G_CALLBACK(name))
+
 void reload_tree(GtkWidget *tree, GtkListStore *store);
 void size_changed();
 GladeXML *main_window = NULL;
@@ -429,6 +432,10 @@
 	client  = gconf_client_get_default();
 
 	main_window = glade_xml_new(GLADE_PATH"/gcursor.glade", "select_dialog", NULL);
+	CONNECT_SIGNAL(extract_theme);
+	CONNECT_SIGNAL(open_theme_dir);
+	CONNECT_SIGNAL(entry_selected);
+	CONNECT_SIGNAL(size_changed);
 	store       = gtk_list_store_new(2, GDK_TYPE_PIXBUF, GTK_TYPE_STRING);
 	tree        = glade_xml_get_widget(main_window, "cursor_tree");
 	gtk_tree_view_set_model(GTK_TREE_VIEW(tree), GTK_TREE_MODEL(store));
