Index: src/gaim-xmms-remote.c
===================================================================
RCS file: /var/lib/cvs/gaim-xmms-remote/src/gaim-xmms-remote.c,v
retrieving revision 1.20
retrieving revision 1.21
diff -u -p -r1.20 -r1.21
--- src/gaim-xmms-remote.c	6 Jul 2004 00:40:41 -0000	1.20
+++ src/gaim-xmms-remote.c	26 Jul 2004 14:53:59 -0000	1.21
@@ -551,7 +551,7 @@ gxr_conv_destroyed_cb(GaimConversation *
 }
 
 static GaimCmdRet
-gxr_cmd_cb(GaimConversation *c, const gchar *cmd, gchar **args, gchar **error) {
+gxr_cmd_cb(GaimConversation *c, const gchar *cmd, gchar **args, gchar **error, void *data) {
 	gchar *lower;
 	gint session = GXR_SESSION;
 
@@ -780,7 +780,7 @@ gxr_load(GaimPlugin *plugin) {
 	/* register our command */
 	gxr_cmd = gaim_cmd_register("gxr", "w", GAIM_CMD_P_PLUGIN,
 								GAIM_CMD_FLAG_IM | GAIM_CMD_FLAG_CHAT, NULL,
-								gxr_cmd_cb, help);
+								gxr_cmd_cb, help, NULL);
 
 	/* hijack the popup menus in the imhtmls */
 	gxr_hook_popups();
