diff -ru freeride-0.6.0/freeride.rb work/freeride.rb
--- freeride-0.6.0/freeride.rb	2003-11-06 20:44:23.000000000 +0100
+++ work/freeride.rb	2004-05-14 23:31:19.000000000 +0200
@@ -77,7 +77,7 @@
     end
     
     #This method will not return until FreeRIDE is closed (shut down)
-    FreeBASE::Core.startup("#{dir}/properties.yaml","#{dir}/default.yaml") do |core|
+    FreeBASE::Core.startup("properties.yaml","#{dir}/default.yaml") do |core|
       #This method will not return until FreeRIDE is closed (shut down)
       #trap("INT") do
       #  core.bus["/system/shutdown"].call(2)
