 src/chat_control.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/chat_control.py b/src/chat_control.py
index e59a871..c34f86a 100644
--- a/src/chat_control.py
+++ b/src/chat_control.py
@@ -1912,7 +1912,7 @@ class ChatControl(ChatControlBase):
 
         if status_escaped:
             status_text = self.urlfinder.sub(self.make_href, status_escaped)
-            status_text = '<span %s>%s</span>' % (font_attrs_small, status_escaped)
+            status_text = '<span %s>%s</span>' % (font_attrs_small, status_text)
             self.banner_status_label.set_tooltip_text(status)
             self.banner_status_label.set_no_show_all(False)
             self.banner_status_label.show()
