Index: mhwaveedit-1.4.16/configure.in
===================================================================
--- mhwaveedit-1.4.16.orig/configure.in
+++ mhwaveedit-1.4.16/configure.in
@@ -384,10 +384,6 @@ if test "x$FOUND_GTK" = "xno"; then
 	fi
 fi
 
-if test "x$FOUND_GTK" = "xno"; then
-AM_PATH_GTK(1.2.5,FOUND_GTK="yes",AC_MSG_ERROR(mhWaveEdit requires GTK+ version 1.2.5 or higher...))
-fi
-
 dnl Check for gthread if we're using GTK+ 2
 dnl Not needed by the app itself, but needed for some aRts versions.
 GTHREAD_CFLAGS=""
