diff -Nru linux-atm-2.4.1.orig/src/arpd/io.c linux-atm-2.4.1/src/arpd/io.c
--- linux-atm-2.4.1.orig/src/arpd/io.c	2007-04-13 23:50:25.000000000 +0300
+++ linux-atm-2.4.1/src/arpd/io.c	2007-04-13 23:30:28.000000000 +0300
@@ -16,7 +16,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <net/if.h>
+#include <linux/if.h>
 #include <netinet/in.h>
 #include <atm.h>
 #include <linux/atmclip.h> /* for CLIP_DEFAULT_IDLETIMER */
