diff -Naurp eb-4.2.2-orig/eb/Makefile.in eb-4.2.2/eb/Makefile.in
--- eb-4.2.2-orig/eb/Makefile.in	2005-11-25 21:02:13.000000000 -0600
+++ eb-4.2.2/eb/Makefile.in	2008-06-21 07:37:51.000000000 -0600
@@ -318,7 +318,7 @@ clean-libLTLIBRARIES:
 	  rm -f "$${dir}/so_locations"; \
 	done
 libeb.la: $(libeb_la_OBJECTS) $(libeb_la_DEPENDENCIES) 
-	$(LINK) -rpath $(libdir) $(libeb_la_LDFLAGS) $(libeb_la_OBJECTS) $(libeb_la_LIBADD) $(LIBS)
+	$(LINK) -rpath $(libdir) $(libeb_la_LDFLAGS) $(libeb_la_OBJECTS) $(libeb_la_LIBADD) $(LIBS) $(ZLIBLIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
