--- mkvtoolnix-1.8.0/configure.old	2006-11-19 11:58:59.000000000 +0100
+++ mkvtoolnix-1.8.0/configure	2006-11-19 11:59:31.000000000 +0100
@@ -10284,13 +10284,13 @@
 
 
   wxw_ver_req_major=2
-  wxw_ver_req_minor=4
+  wxw_ver_req_minor=6
   wxw_ver_req_micro=0
   wxw_min_ver=$wxw_ver_req_major.$wxw_ver_req_minor.$wxw_ver_req_micro
 
   if test x"$enable_gui" = x"yes" -o x"$enable_gui" = "x"; then
     # Extract the first word of "wx-config", so it can be a program name with args.
-set dummy wx-config; ac_word=$2
+set dummy wx-config-2.6; ac_word=$2
 { echo "$as_me:$LINENO: checking for $ac_word" >&5
 echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6; }
 if test "${ac_cv_path_WX_CONFIG+set}" = set; then
