--- mkvtoolnix-1.4.2/configure.in	2005-04-09 19:36:20.000000000 +0200
+++ mkvtoolnix-1.4.2-configure/configure.in	2005-04-18 19:10:35.679161416 +0200
@@ -460,11 +460,19 @@
 dnl
 dnl Check for libFLAC
 dnl
-  AC_CHECK_LIB(FLAC, FLAC__seekable_stream_decoder_new,
-               [ FLAC_LIBS="-lFLAC -lm"
-                 flac_found=yes ],
-               [ flac_found=no ],
-               "-lm")
+  AC_ARG_WITH([flac],
+              AC_HELP_STRING([--without-flac], [do not build with flac support]),
+              [ with_flac=${withval} ], [ with_flac=yes ])
+
+  if test "$with_flac" != "no"; then
+    AC_CHECK_LIB(FLAC, FLAC__seekable_stream_decoder_new,
+                 [ FLAC_LIBS="-lFLAC -lm"
+                   flac_found=yes ],
+                 [ flac_found=no ],
+                 "-lm")
+  else
+    flac_found=no
+  fi
   if test "$flac_found" = "yes"; then
     AC_CHECK_MEMBER(FLAC__StreamMetadata_StreamInfo.sample_rate, ,
                     [ flac_found=no ],
@@ -488,7 +496,6 @@
     opt_features_no="$opt_features_no; FLAC audio"
   fi
 
-
 AC_SUBST(FLAC_LIBS)
 
 
