diff -urN OpenPBS_2_3_16.orig/src/lib/Libpbs/Makefile.in OpenPBS_2_3_16/src/lib/Libpbs/Makefile.in
--- OpenPBS_2_3_16.orig/src/lib/Libpbs/Makefile.in	2006-09-26 00:14:41.000000000 -0700
+++ OpenPBS_2_3_16/src/lib/Libpbs/Makefile.in	2006-09-26 00:15:26.000000000 -0700
@@ -95,7 +95,7 @@
 INCS = -I@top_builddir@/src/include -I$(top_srcdir)/src/include \
        -I$(srcdir)/../Libdis -I$(srcdir)/../Libifl
 
-CFLAGS = @CFLAGS@ $(INCS) @DEFS@ -DIFF_PATH=\"$(IFF_PATH)\"
+CFLAGS = @CFLAGS@ -fPIC $(INCS) @DEFS@ -DIFF_PATH=\"$(IFF_PATH)\"
 
 TARGET = libpbs.a
 
