--- src/objectp.cpp.wrong	2003-09-26 02:43:48.380421352 -0500
+++ src/objectp.cpp	2003-09-26 02:42:29.977340432 -0500
@@ -451,16 +451,16 @@
 //				if (height)
 					height--;
 
-				uint64 data = (p0 & 0xFFFFF80000000000) >> 40;
+				uint64 data = (p0 & 0xFFFFF80000000000LL) >> 40;
 				uint64 dwidth = (p1 & 0xFFC0000) >> 15;
 				data += dwidth;
 
-				p0 &= ~0xFFFFF80000FFC000;			// Mask out old data...
+				p0 &= ~0xFFFFF80000FFC000LL;			// Mask out old data...
 				p0 |= (uint64)height << 14;
 				p0 |= data << 40;
 				OPStorePhrase(oldOPP, p0);
 			}
-			op_pointer = (p0 & 0x000007FFFF000000) >> 21;
+			op_pointer = (p0 & 0x000007FFFF000000LL) >> 21;
 			break;
 		}
 		case OBJECT_TYPE_SCALE:
@@ -504,7 +504,7 @@
 //				if (remainder & 0x80)				// I.e., it's negative
 				if ((remainder & 0x80) || remainder == 0)	// I.e., it's <= 0
 				{
-					uint64 data = (p0 & 0xFFFFF80000000000) >> 40;
+					uint64 data = (p0 & 0xFFFFF80000000000LL) >> 40;
 					uint64 dwidth = (p1 & 0xFFC0000) >> 15;
 
 //					while (remainder & 0x80)
@@ -516,21 +516,21 @@
 
 						data += dwidth;
 					}
-					p0 &= ~0xFFFFF80000FFC000;		// Mask out old data...
+					p0 &= ~0xFFFFF80000FFC000LL;		// Mask out old data...
 					p0 |= (uint64)height << 14;
 					p0 |= data << 40;
 					OPStorePhrase(oldOPP, p0);
 				}
 
 //WriteLog(" [%08X%08X -> ", (uint32)(p2>>32), (uint32)(p2&0xFFFFFFFF));
-				p2 &= ~0x0000000000FF0000;
+				p2 &= ~0x0000000000FF0000LL;
 				p2 |= (uint64)remainder << 16;
 //WriteLog("%08X%08X]\n", (uint32)(p2>>32), (uint32)(p2&0xFFFFFFFF));
 				OPStorePhrase(oldOPP+16, p2);
 //remainder = (uint8)(p2 >> 16), vscale = (uint8)(p2 >> 8);
 //WriteLog(" [after]: rem=%02X, vscale=%02X\n", remainder, vscale);
 			}
-			op_pointer = (p0 & 0x000007FFFF000000) >> 21;
+			op_pointer = (p0 & 0x000007FFFF000000LL) >> 21;
 			break;
 		}
 		case OBJECT_TYPE_GPU:
