--- ../parrot-0.2.3/config/gen/makefiles/root.in	2005-07-29 12:07:33.000000000 +0200
+++ config/gen/makefiles/root.in	2005-08-07 17:40:26.000000000 +0200
@@ -656,10 +656,8 @@
 runtime/parrot/include/config.fpmc : myconfig config_lib.pasm $(MINIPARROT)
 	@echo Invoking Parrot to generate runtime/parrot/include/config.fpmc --cross your fingers
 	$(MINIPARROT) config_lib.pasm > $@
-	@echo If the next line prints $(VERSION), it did help.
-	$(MINIPARROT) parrot-config.imc VERSION DEVEL
 
-parrot.def :
+parrot.def : build_tools\parrotdef.pl
 	perl build_tools/parrotdef.pl
 
 $(PARROT) : $(IMCC_DIR)/main$(O) $(GEN_HEADERS) $(LIBPARROT) \
@@ -1029,6 +1027,8 @@
 $(OPS_DIR)/core_ops_switch.c $(INC_DIR)/oplib/core_ops_switch.h : $(OPS_FILES) $(BUILD_TOOLS_DIR)/ops2c.pl lib/Parrot/OpsFile.pm lib/Parrot/Op.pm $(INC_DIR)/config.h lib/Parrot/OpLib/core.pm
 	$(PERL) $(BUILD_TOOLS_DIR)/ops2c.pl CSwitch --core
 
+build_tools\parrotdef.pl : $(CLASS_PMC_FILES)
+
 ${TEMP_cg_c}
 
 ${TEMP_gc_c}
