Index: redir.c
===================================================================
--- redir.c	(revision 366)
+++ redir.c	(working copy)
@@ -103,6 +103,9 @@
 
 static inline void rfs_truncate_inode_pages(struct inode *inode)
 {
+	return;
+
+	/*
 	if (!inode)
 		return;
 
@@ -111,6 +114,7 @@
 		truncate_inode_pages(&inode->i_data, 0);
 		mutex_unlock(&inode->i_mutex);
 	}
+	*/
 }
 
 int rfs_replace_ops(struct rpath *path_old, struct rpath *path_new, struct filter *flt)
