diff -ur clawk.orig/clawk.lisp clawk/clawk.lisp
--- clawk.orig/clawk.lisp	2006-03-11 11:55:18.000000000 -0600
+++ clawk/clawk.lisp	2006-03-12 13:37:03.000000000 -0600
@@ -32,7 +32,7 @@
                        (setq symname (concatenate 'string symname (string c) (string c2))))
                       (t (setq symname (concatenate 'string symname (string c2)))))))
              ((or (char= c #\Return) (char= c #\Newline))
-              (error "Unterminated regular expression: " symname))
+              (error "Unterminated regular expression: ~S" symname))
              (t (setq symname (concatenate 'string symname (string c)))))))))
 
 (defun install-regex-syntax ()
@@ -56,7 +56,7 @@
                       (t (princ c cmdname-strm)
                          (princ c2 cmdname-strm)))))
              ((or (char= c #\Return) (char= c #\Newline))
-              (error "Unterminated shell command: " (get-output-stream-string cmdname-strm)))
+              (error "Unterminated shell command: ~S" (get-output-stream-string cmdname-strm)))
              (t (princ c cmdname-strm)))))))
 
 (defun install-cmd-syntax ()
@@ -112,7 +112,7 @@
 (defvar *CURFILE*)
 (defvar *CURLINE* "")
 (defvar *FS-IS-WS* t)
-(defconstant +WS-FIELDSEP-PAT+ "[ \\t]+")
+(cl+:defconst +WS-FIELDSEP-PAT+ "[ \\t]+")
 (defvar *FS* +WS-FIELDSEP-PAT+)
 (defvar *RSTART* 0)
 (defvar *RLENGTH* 0)
Only in clawk: clawk.lisp~
