diff --git a/src/i830_dri.h b/src/i830_dri.h
index f97c1dd..44be353 100644
--- a/src/i830_dri.h
+++ b/src/i830_dri.h
@@ -3,7 +3,6 @@
 #ifndef _I830_DRI_H
 #define _I830_DRI_H
 
-#include "xf86dri.h"
 #include "xf86drm.h"
 #include "i830_common.h"
 
@@ -19,6 +18,21 @@ typedef struct _I830DRIRec {
    drm_handle_t regs;
    drmSize regsSize;
 
+   drmSize unused1; /* backbufferSize */
+   drm_handle_t unused2; /* backbuffer */
+
+   drmSize unused3; /* depthbufferSize */
+   drm_handle_t unused4; /* depthbuffer */
+
+   drmSize unused5; /* rotatedSize /*/
+   drm_handle_t unused6; /* rotatedbuffer */
+
+   drm_handle_t unused7; /* textures */
+   int unused8; /* textureSize */
+
+   drm_handle_t unused9; /* agp_buffers */
+   drmSize unused10; /* agp_buf_size */
+
    int deviceID;
    int width;
    int height;
@@ -26,6 +40,11 @@ typedef struct _I830DRIRec {
    int cpp;
    int bitsPerPixel;
 
+   int unused11[8]; /* was front/back/depth/rotated offset/pitch */
+
+   int unused12; /* logTextureGranularity */
+   int unused13; /* textureOffset */
+
    int irq;
    int sarea_priv_offset;
 } I830DRIRec, *I830DRIPtr;
