--- sounds/raw2h.c.orig	2001-09-21 21:37:37.000000000 +0200
+++ sounds/raw2h.c	2004-02-17 17:18:07.678921384 +0200
@@ -28,20 +28,19 @@
 	res = read(i, buf, sizeof(buf));
 	assert(res >= 0);
 	assert(res < sizeof(buf));
-	fprintf(o, 
-"/*
-  * Signed 16-bit audio data
-  *
-  * Source: %s
-  *
-  * Copyright (C) 1999, Mark Spencer and Linux Support Services
-  *
-  * Distributed under the terms of the GNU General Public License
-  *
-  */
-
-static signed short %s[] = {
-",	argv[1], n);
+	fprintf(o, "/*\n"
+                   " * Signed 16-bit audio data\n"
+                   " *\n"
+                   " * Source: %s\n"
+                   " *\n"
+                   " * Copyright (C) 1999, Mark Spencer and Linux Support Services\n"
+                   " *\n"
+                   " * Distributed under the terms of the GNU General Public License\n"
+                   " *\n"
+                   " */\n"
+                   "\n"
+                   "static signed short %s[] = {\n",
+                argv[1], n);
 	for (x=0;x<res/2 - 1;x++) {
 		fprintf(o, "%#06hx, ", buf[x]);
 		if ((x % 10) == 9)
