diff -urNp banshee-1-1.7.4.orig/build/m4/shamrock/gnome-doc.m4 banshee-1-1.7.4/build/m4/shamrock/gnome-doc.m4
--- banshee-1-1.7.4.orig/build/m4/shamrock/gnome-doc.m4	2010-08-29 02:24:28.598617103 +0530
+++ banshee-1-1.7.4/build/m4/shamrock/gnome-doc.m4	2010-08-29 11:07:22.709617074 +0530
@@ -14,6 +14,8 @@ AC_DEFUN([SHAMROCK_CHECK_GNOME_DOC_UTILS
 
 	if test "x$enable_user_help" = "xyes"; then
         GNOME_DOC_INIT([$1], enable_user_help=yes, enable_user_help=no)
+    else
+        AM_CONDITIONAL(ENABLE_SK, false)
     fi
 
 	AM_CONDITIONAL(HAVE_GNOME_DOC_UTILS, test "x$enable_user_help" = "xyes")
