diff -ru NVIDIA_kernel-1.0-4191-2.5-tl/Makefile NVIDIA_kernel-1.0-4191-2.5.54/Makefile
--- NVIDIA_kernel-1.0-4191-2.5-tl/Makefile	Fri Dec 27 05:36:52 2002
+++ NVIDIA_kernel-1.0-4191-2.5.54/Makefile	Fri Jan  3 16:35:20 2003
@@ -22,10 +22,8 @@
 # this is slightly more brain-dead, but works..
 ifeq ($(shell if test -d $(KERNDIR)/build; then echo yes; fi),yes)
 KERNINC=$(KERNDIR)/build/include
-MACHINC=$(KERNDIR)/build/arch/i386/mach-generic
 else
 KERNINC=/usr/src/linux/include
-MACHINC=/usr/src/linux/arch/i386/mach-generic
 endif
 
 ifeq ($(shell if test -d $(KERNDIR)/kernel; then echo yes; fi),yes)
@@ -60,10 +58,10 @@
 # allow specification of alternate include file tree on command line and extra defines
 ifdef SYSINCLUDE
 INCLUDES += -I$(SYSINCLUDE)
-INCLUDES += -I$(SYSINCLUDE)/../arch/i386/mach-generic
+INCLUDES += -I$(SYSINCLUDE)/asm/mach-default
 else
 INCLUDES += -I$(KERNINC)
-INCLUDES += -I$(MACHINC)
+INCLUDES += -I$(KERNINC)/asm/mach-default
 endif
 
 DEFINES+=$(EXTRA_DEFINES)
@@ -139,7 +137,7 @@
 	fi
 
 nvidia.o: gcc-check $(LINUX_MODULE) $(RESMAN_KERNEL_MODULE) 
-	ld -r -o $@ $(LINUX_MODULE) $(RESMAN_KERNEL_MODULE)
+	ld -d -r -o $@ $(LINUX_MODULE) $(RESMAN_KERNEL_MODULE)
 	size $@
 
 $(VERSION_HDR): 
