--- kdetv/libkdetv/lirc.cpp.orig	2005-01-30 16:52:51.000000000 +0100
+++ kdetv/libkdetv/lirc.cpp	2005-01-30 16:53:09.000000000 +0100
@@ -82,7 +82,7 @@
             while (lirc_code2char(_config,code,&cmd)==0 && cmd != NULL) {
                 kdDebug() << "lirc: cmd '" << cmd << "'" << endl;
                 if (0 == strcasecmp(cmd, "default")) {
-                    QMap<QString, QString>::const_iterator val = _keymap.constFind(event);
+                    QMap<QString, QString>::const_iterator val = _keymap.find(event);
                     if ( val != _keymap.constEnd())
                         emit command(*val, repeat);
                     else
@@ -93,7 +93,7 @@
             }
         } else {
             /* standalone mode */
-            QMap<QString, QString>::const_iterator val = _keymap.constFind(event);
+            QMap<QString, QString>::const_iterator val = _keymap.find(event);
             if ( val != _keymap.constEnd())
                 emit command(*val, repeat);
             else
