Index: src/trackerlist.h
===================================================================
--- src/trackerlist.h	(revision 4105)
+++ src/trackerlist.h	(working copy)
@@ -61,10 +61,8 @@
 public slots:
   void setRowColor(int row, QColor color);
 
-#if LIBTORRENT_VERSION_MINOR < 15
   void moveSelectionUp();
   void moveSelectionDown();
-#endif
 
   void clear();
   void loadStickyItems(const QTorrentHandle &h);
Index: src/trackerlist.cpp
===================================================================
--- src/trackerlist.cpp	(revision 4105)
+++ src/trackerlist.cpp	(working copy)
@@ -94,8 +94,8 @@
   }
 }
 
+void TrackerList::moveSelectionUp() {
 #if LIBTORRENT_VERSION_MINOR < 15
-void TrackerList::moveSelectionUp() {
   QTorrentHandle h = properties->getCurrentTorrent();
   if(!h.is_valid()) {
     clear();
@@ -129,9 +129,11 @@
   h.replace_trackers(trackers);
   // Reannounce
   h.force_reannounce();
+#endif
 }
 
 void TrackerList::moveSelectionDown() {
+#if LIBTORRENT_VERSION_MINOR < 15
   QTorrentHandle h = properties->getCurrentTorrent();
   if(!h.is_valid()) {
     clear();
@@ -165,8 +167,8 @@
   h.replace_trackers(trackers);
   // Reannounce
   h.force_reannounce();
+#endif
 }
-#endif
 
 void TrackerList::clear() {
   qDeleteAll(tracker_items.values());
