--- Makefile.in.orig	2007-12-17 18:56:40.000000000 +0100
+++ Makefile.in	2007-12-17 18:58:25.000000000 +0100
@@ -102,36 +102,36 @@
 	${INSTALL} -d ${DESTDIR}${mandir}/man5
 	${INSTALL} -d ${DESTDIR}${sysconfdir}
 	${INSTALL} -d ${DESTDIR}${datadir}/detox
-	${INSTALL} detox ${DESTDIR}${bindir}
-	${INSTALL} detox.1 ${DESTDIR}${mandir}/man1
-	${INSTALL} detoxrc.5 detox.tbl.5 ${DESTDIR}${mandir}/man5
+	${INSTALL} -m 644 detox ${DESTDIR}${bindir}
+	${INSTALL} -m 644 detox.1 ${DESTDIR}${mandir}/man1
+	${INSTALL} -m 644 detoxrc.5 detox.tbl.5 ${DESTDIR}${mandir}/man5
 
 install-safe-config:
 	@if [ ! -f ${DESTDIR}${sysconfdir}/detoxrc ]; then \
-		${INSTALL} detoxrc ${DESTDIR}${sysconfdir}; \
+		${INSTALL} -m 644 detoxrc ${DESTDIR}${sysconfdir}; \
 	else \
 		echo "${DESTDIR}${sysconfdir}/detoxrc exists, skipping"; \
 	fi
 	@if [ ! -f ${DESTDIR}${datadir}/detox/iso8859_1.tbl ]; then \
-		${INSTALL} iso8859_1.tbl ${DESTDIR}${datadir}/detox; \
+		${INSTALL} -m 644 iso8859_1.tbl ${DESTDIR}${datadir}/detox; \
 	else \
 		echo "${DESTDIR}${datadir}/detox/iso8859_1.tbl exists, skipping"; \
 	fi
 	@if [ ! -f ${DESTDIR}${datadir}/detox/unicode.tbl ]; then \
-		${INSTALL} unicode.tbl ${DESTDIR}${datadir}/detox; \
+		${INSTALL} -m 644 unicode.tbl ${DESTDIR}${datadir}/detox; \
 	else \
 		echo "${DESTDIR}${datadir}/detox/unicode.tbl exists, skipping"; \
 	fi
 
 install-unsafe-config:
-	${INSTALL} detoxrc ${DESTDIR}${sysconfdir}
-	${INSTALL} iso8859_1.tbl ${DESTDIR}${datadir}/detox
-	${INSTALL} unicode.tbl ${DESTDIR}${datadir}/detox
+	${INSTALL} -m 644 detoxrc ${DESTDIR}${sysconfdir}
+	${INSTALL} -m 644 iso8859_1.tbl ${DESTDIR}${datadir}/detox
+	${INSTALL} -m 644 unicode.tbl ${DESTDIR}${datadir}/detox
 
 install-sample-config:
-	${INSTALL} detoxrc ${DESTDIR}${sysconfdir}/detoxrc.sample
-	${INSTALL} iso8859_1.tbl ${DESTDIR}${datadir}/detox/iso8859_1.tbl.sample
-	${INSTALL} unicode.tbl ${DESTDIR}${datadir}/detox/unicode.tbl.sample
+	${INSTALL} -m 644 detoxrc ${DESTDIR}${sysconfdir}/detoxrc.sample
+	${INSTALL} -m 644 iso8859_1.tbl ${DESTDIR}${datadir}/detox/iso8859_1.tbl.sample
+	${INSTALL} -m 644 unicode.tbl ${DESTDIR}${datadir}/detox/unicode.tbl.sample
 
 clean:
 	rm -f *.o core *~ a.out *.core
