diff -u -r1.87 configure.ac
--- configure.ac        30 Sep 2004 01:32:17 -0000      1.87
+++ configure.ac        30 Sep 2004 21:52:25 -0000
@@ -330,7 +330,7 @@
   AC_DEFINE_UNQUOTED(GPGSM_PATH, "$GPGSM", [Path to the GPGSM binary.])
   AC_SUBST(GPGSM)
 fi
-AM_CONDITIONAL(HAVE_GPGSM, [test -n "$GPGSM"])
+AM_CONDITIONAL(HAVE_GPGSM, ["$GPGSM" != "no"])
 dnl Check for GPGSM version requirement.
 GPGSM_VERSION=unknown
 ok=maybe