--- net/ipv4/netfilter/ip_tables.c.orig	2004-06-30 18:19:22.957763360 -0400
+++ net/ipv4/netfilter/ip_tables.c	2004-06-30 18:19:51.516421784 -0400
@@ -1458,7 +1458,7 @@ tcp_find_option(u_int8_t option,
 		int *hotdrop)
 {
 	/* tcp.doff is only 4 bits, ie. max 15 * 4 bytes */
-	char opt[60 - sizeof(struct tcphdr)];
+	u_int8_t opt[60 - sizeof(struct tcphdr)];
 	unsigned int i;
 
 	duprintf("tcp_match: finding option\n");
