--- certmanager/lib/backends/qgpgme/qgpgmebackend.cpp
+++ certmanager/lib/backends/qgpgme/qgpgmebackend.cpp
@@ -54,7 +54,7 @@
     mOpenPGPProtocol( 0 ),
     mSMIMEProtocol( 0 )
 {
-
+  GpgME::initializeLibrary();
 }
 
 Kleo::QGpgMEBackend::~QGpgMEBackend() {
--- libkdenetwork/gpgmepp/context.h
+++ libkdenetwork/gpgmepp/context.h
@@ -287,6 +287,8 @@
   //
   //
 
+  KDE_EXPORT void initializeLibrary();
+
   KDE_EXPORT GpgME::Error setDefaultLocale( int category, const char * value );
 
   KDE_EXPORT Context * wait( GpgME::Error & e, bool hang=true );
