diff -uNr xf86-input-synaptics-0.15.0.ORIG/src/synaptics.c xf86-input-synaptics-0.15.0/src/synaptics.c
--- xf86-input-synaptics-0.15.0.ORIG/src/synaptics.c	2008-08-23 22:19:55.000000000 +0100
+++ xf86-input-synaptics-0.15.0/src/synaptics.c	2008-08-23 22:21:16.000000000 +0100
@@ -136,8 +136,8 @@
     MODULEVENDORSTRING,
     MODINFOSTRING1,
     MODINFOSTRING2,
-    XF86_VERSION_CURRENT,
-    1, 0, 0,
+    XORG_VERSION_CURRENT,
+    PACKAGE_VERSION_MAJOR, PACKAGE_VERSION_MINOR, PACKAGE_VERSION_PATCHLEVEL,
     ABI_CLASS_XINPUT,
     ABI_XINPUT_VERSION,
     MOD_CLASS_XINPUT,
