From 7c0361d4ec6b1f1325cb6551d0ee2e7f5cfae15b Mon Sep 17 00:00:00 2001
From: Daniel Stone <daniel@fooishbar.org>
Date: Thu, 09 Jun 2011 19:03:05 +0000
Subject: Adjust acceleration scheme for input ABI v14

v14 wants doubles, rather than floats, from acceleration schemes.

Signed-off-by: Daniel Stone <daniel@fooishbar.org>
Reviewed-by: Peter Hutterer <peter.hutterer@who-t.net>
---
diff --git a/src/synaptics.c b/src/synaptics.c
index c44d1f7..e05147e 100644
--- a/src/synaptics.c
+++ b/src/synaptics.c
@@ -593,11 +593,22 @@ static void set_default_parameters(InputInfoPtr pInfo)
     }
 }
 
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 14
+static double SynapticsAccelerationProfile(DeviceIntPtr dev,
+                                           DeviceVelocityPtr vel,
+                                           double velocity,
+                                           double thr,
+                                           double acc) {
+#else
 static float SynapticsAccelerationProfile(DeviceIntPtr dev,
                                           DeviceVelocityPtr vel,
-                                          float velocity,
-                                          float thr,
-                                          float acc) {
+                                          float velocity_f,
+                                          float thr_f,
+                                          float acc_f) {
+    double velocity = velocity_f;
+    double thr = thr_f;
+    double acc = acc_f;
+#endif
     InputInfoPtr pInfo = dev->public.devicePrivate;
     SynapticsPrivate *priv = (SynapticsPrivate *) (pInfo->private);
     SynapticsParameters* para = &priv->synpara;
--
cgit v0.9.0.2-2-gbebe
