diff -burN AppConfig-1.56.orig/lib/AppConfig/File.pm AppConfig-1.56/lib/AppConfig/File.pm
--- AppConfig-1.56.orig/lib/AppConfig/File.pm	2004-02-04 02:28:28.000000000 -0800
+++ AppConfig-1.56/lib/AppConfig/File.pm	2004-05-26 15:54:17.854304993 -0700
@@ -170,7 +170,7 @@
 	    s/\s+$//;
 
 	    # look for a [block] to set $prefix
-	    if (/^\[([^\]]+)\]$/) {
+	    if (/^\[\s*([^\]\s]+)\s*\]$/) {
 		$prefix = $1;
 		print STDERR "Entering [$prefix] block\n" if $debug;
 		next;
