diff -Naur scons-local-0.96.1.orig/SCons/Tool/gas.py scons-local-0.96.1/SCons/Tool/gas.py
--- scons-local-0.96.1.orig/SCons/Tool/gas.py	2005-04-22 06:18:50.000000000 +0900
+++ scons-local-0.96.1/SCons/Tool/gas.py	2009-02-14 09:27:20.000000000 +0900
@@ -1,7 +1,7 @@
-import as
+as_module = __import__('as', globals(), locals(), [])
 assemblers=['as', 'gas']
 def generate(env):
-    as.generate(env)
+    as_module.generate(env)
     env['AS']       =env.Detect(assemblers) or 'as'
 def exists(env):
     return env.Detect(assemblers)
