--- skim-1.4.5.orig/scons-local-0.96.1/SCons/Util.py	2009-04-29 12:49:05.000000000 +1000
+++ skim-1.4.5/scons-local-0.96.1/SCons/Util.py	2009-04-29 12:49:41.000000000 +1000
@@ -696,6 +696,10 @@
 class CLVar(UserList.UserList):
     def __init__(self, seq=[]):
         UserList.UserList.__init__(self, Split(seq))
+    def __add__(self, other):
+        return UserList.UserList.__add__(self, CLVar(other))
+    def __radd__(self, other):
+        return UserList.UserList.__radd__(self, CLVar(other))
     def __coerce__(self, other):
         return (self, CLVar(other))
     def __str__(self):
