Index: bmake/makefile.boot.in
===================================================================
--- bmake.orig/makefile.boot.in
+++ bmake/makefile.boot.in
@@ -31,13 +31,15 @@ BMAKE_ENV= CC="$(CC)" LIBC= MAKEFLAGS= M
 bootstrap:	bmake.boot .mk.done
 	${BMAKE_ENV} ./bmake.boot -f Makefile 
 	${BMAKE_ENV} ./bmake.boot -f Makefile bmake.1
-	${BMAKE_ENV} `pwd`/bmake -f Makefile test
-	
+
 bmake:	bmake.boot
 	@echo you might want to try: 
 	@echo ${MAKE} -f makefile.boot bootstrap
 	cp bmake.boot $@
 
+check:
+	${BMAKE_ENV} `pwd`/bmake -f Makefile test
+	
 bmake.boot: ${OBJ}
 	(cd lst.lib; $(MAKE) -f makefile.boot CC="$(CC)" CFLAGS="-I.. -I../${srcdir} -I${srcdir} ${CFLAGS}" )
 	${CC} *.o lst.lib/*.o -o $@ @LDFLAGS@ @LIBS@
