diff -ur sbcl-0.8.8.original/contrib/sb-posix/posix-tests.lisp sbcl-0.8.8/contrib/sb-posix/posix-tests.lisp
--- sbcl-0.8.8.original/contrib/sb-posix/posix-tests.lisp	2004-02-07 00:53:29.000000000 -0600
+++ sbcl-0.8.8/contrib/sb-posix/posix-tests.lisp	2004-02-25 14:22:07.426306400 -0600
@@ -95,7 +95,7 @@
       (sb-posix:syscall-errno c)))
   #.sb-posix::eexist)
 
-(deftest mkdir.error.3
+#+nil (deftest mkdir.error.3
   (handler-case
       (sb-posix:mkdir "/almost-certainly-does-not-exist" 0)
     (sb-posix:syscall-error (c)
@@ -129,7 +129,7 @@
       (sb-posix:syscall-errno c)))
   #.sb-posix::enotdir)
 
-#-sunos ; Apparently gives EINVAL on SunOS 8, which doesn't make sense
+#+nil
 (deftest rmdir.error.3
   (handler-case
       (sb-posix:rmdir "/")
Only in sbcl-0.8.8/contrib/sb-posix: posix-tests.lisp.~1.11.~
diff -ur sbcl-0.8.8.original/src/runtime/GNUmakefile sbcl-0.8.8/src/runtime/GNUmakefile
--- sbcl-0.8.8.original/src/runtime/GNUmakefile	2004-02-20 01:51:03.000000000 -0600
+++ sbcl-0.8.8/src/runtime/GNUmakefile	2004-02-24 23:18:58.000000000 -0600
@@ -34,6 +34,7 @@
 # symlink to Config-foo.
 include Config
 
+CFLAGS += -DSBCL_HOME='"/usr/lib/sbcl"'
 
 C_SRCS = alloc.c backtrace.c breakpoint.c coreparse.c \
 	dynbind.c gc-common.c globals.c interr.c interrupt.c \
Only in sbcl-0.8.8/src/runtime: GNUmakefile.~1.18.~
