diff -urp hostap-driver-0.3.5/driver/modules/hostap_config.h hostap-driver-0.3.5-firmware/driver/modules/hostap_config.h
--- hostap-driver-0.3.5/driver/modules/hostap_config.h	2005-01-24 05:43:17.000000000 +0100
+++ hostap-driver-0.3.5-firmware/driver/modules/hostap_config.h	2005-01-24 15:46:25.000000000 +0100
@@ -59,7 +59,7 @@
  * In addition, please note that it is possible to kill your card with
  * non-volatile download if you are using incorrect image. This feature has not
  * been fully tested, so please be careful with it. */
-/* #define PRISM2_NON_VOLATILE_DOWNLOAD */
+#define PRISM2_NON_VOLATILE_DOWNLOAD
 #endif /* PRISM2_DOWNLOAD_SUPPORT */
 
 /* Include wireless extensions sub-ioctl support even if wireless extensions
