Only in glademm-0.6.4b/: glademm.spec
diff -ru glademm-0.6.4b/src/Cxx_Fs_Signal.cc /usr/src/glademm-0.6.4b/src/Cxx_Fs_Signal.cc
--- glademm-0.6.4b/src/Cxx_Fs_Signal.cc	2001-11-22 10:44:00.000000000 +0100
+++ /usr/src/glademm-0.6.4b/src/Cxx_Fs_Signal.cc	2002-06-25 02:22:33.000000000 +0200
@@ -99,7 +99,7 @@
  * in the Widget it is derived from, we should make this more explicit
  */
 
-void Cxx_Fileset::DeclareSignalHandler(const WriterBase &wr, const Widget &w, const Widget &top, bool container=false)
+void Cxx_Fileset::DeclareSignalHandler(const WriterBase &wr, const Widget &w, const Widget &top, bool container)
 {  // signals are part of the widget declaration
    if (w.getBool(CXX_SEPERATE_CLASS) && !container) return;
    
diff -ru glademm-0.6.4b/src/SourceWriter/CxxFile.cc /usr/src/glademm-0.6.4b/src/SourceWriter/CxxFile.cc
--- glademm-0.6.4b/src/SourceWriter/CxxFile.cc	2002-03-01 22:50:24.000000000 +0100
+++ /usr/src/glademm-0.6.4b/src/SourceWriter/CxxFile.cc	2002-06-25 01:57:34.000000000 +0200
@@ -503,7 +503,7 @@
    }
 }
 
-void CxxFile::push(CxxFile_Context c,int indent_delta=0)
+void CxxFile::push(CxxFile_Context c,int indent_delta)
 {  pushed.push_back(state);
 #ifdef DEBUG_STATE_MACHINE
    std::cout << '+' << state.context << "->" << c << " | " << state.global << '\n';
diff -ru glademm-0.6.4b/src/SourceWriter/SystemFile.cc /usr/src/glademm-0.6.4b/src/SourceWriter/SystemFile.cc
--- glademm-0.6.4b/src/SourceWriter/SystemFile.cc	2002-01-21 18:22:13.000000000 +0100
+++ /usr/src/glademm-0.6.4b/src/SourceWriter/SystemFile.cc	2002-06-25 01:58:50.000000000 +0200
@@ -106,7 +106,7 @@
    }
 }
 
-SystemFile::SystemFile(const std::string &name,std::ios::openmode mode=std::ios::out, int prot=0664)
+SystemFile::SystemFile(const std::string &name,std::ios::openmode mode, int prot)
    : filename(name), is_stringstream(false), deallocate(true), 
    	remembered(0), mystage(0)
 {  streamptr.of=new std::ofstream(name.c_str(),mode/*,prot*/);
diff -ru glademm-0.6.4b/src/writers/menubar.cc /usr/src/glademm-0.6.4b/src/writers/menubar.cc
--- glademm-0.6.4b/src/writers/menubar.cc	2002-01-21 18:19:39.000000000 +0100
+++ /usr/src/glademm-0.6.4b/src/writers/menubar.cc	2002-06-25 02:09:32.000000000 +0200
@@ -82,7 +82,7 @@
    return true;
 }
 
-void Gtk_MenuBar::AdditionalMemberVars(const Widget &w,CxxFile &f,bool container=false) const
+void Gtk_MenuBar::AdditionalMemberVars(const Widget &w,CxxFile &f,bool container) const
 {  Parent::AdditionalMemberVars(w,f,container);
    if (!Configuration.gnome_support) return;
    // all are internal!
diff -ru glademm-0.6.4b/src/writers/optionmenu.cc /usr/src/glademm-0.6.4b/src/writers/optionmenu.cc
--- glademm-0.6.4b/src/writers/optionmenu.cc	2002-01-21 18:19:40.000000000 +0100
+++ /usr/src/glademm-0.6.4b/src/writers/optionmenu.cc	2002-06-25 02:13:26.000000000 +0200
@@ -63,7 +63,7 @@
    Parent::GCInclude(w,f);
 }
 
-void Gtk_OptionMenu::AdditionalMemberVars(const Widget &w,CxxFile &f,bool container=false) const
+void Gtk_OptionMenu::AdditionalMemberVars(const Widget &w,CxxFile &f,bool container) const
 {  Parent::AdditionalMemberVars(w,f,container);
    if (w.hasTag("items") && Configuration.optionmenu_create_enum)
    {  const std::string name(Configuration.InstanceName(w.Name()));
diff -ru glademm-0.6.4b/src/writers/toolbar.cc /usr/src/glademm-0.6.4b/src/writers/toolbar.cc
--- glademm-0.6.4b/src/writers/toolbar.cc	2002-04-02 17:35:13.000000000 +0200
+++ /usr/src/glademm-0.6.4b/src/writers/toolbar.cc	2002-06-25 02:17:42.000000000 +0200
@@ -181,7 +181,7 @@
       f.Statement() << instance << "set_space_style(" << space_style << ')';
 }
 
-void Gtk_Toolbar::AdditionalMemberVars(const Widget &w,CxxFile &f,bool container=false) const
+void Gtk_Toolbar::AdditionalMemberVars(const Widget &w,CxxFile &f,bool container) const
 {  Parent::AdditionalMemberVars(w,f,container);
                      
    for (Widget::const_iterator i=w.begin();i!=w.end();++i)
diff -ru glademm-0.6.4b/src/writers/window.cc /usr/src/glademm-0.6.4b/src/writers/window.cc
--- glademm-0.6.4b/src/writers/window.cc	2002-01-21 18:22:47.000000000 +0100
+++ /usr/src/glademm-0.6.4b/src/writers/window.cc	2002-06-25 02:19:43.000000000 +0200
@@ -124,7 +124,7 @@
    assert(never_reached);
 }
 
-void Gtk_Window::AdditionalMemberVars(const Widget &w,CxxFile &f,bool container=false) const
+void Gtk_Window::AdditionalMemberVars(const Widget &w,CxxFile &f,bool container) const
 {  if (Configuration.has_accelerators)
    {  Widget w2(get_accelgroup(w));
       f.Private();
