--- SConstruct.orig	2006-10-18 00:57:28.000000000 +0200
+++ SConstruct	2006-10-18 00:58:26.000000000 +0200
@@ -20,16 +20,17 @@
     IM_LIB_PATH='/usr/lib/'
     GD_LIBS=['Magick','Magick++','Wand']
     PYTHON_INCLUDE='/usr/include/python2.4'
-    BOOST_PYTHON_LIB=['libboost_python-gcc']
-    CPP_FLAGS=['-DBOOST_PYTHON_STATIC_LIB','-O2']
+    BOOST_PYTHON_LIB=['libboost_python']
+    CPP_FLAGS=['-DBOOST_PYTHON_STATIC_LIB']
 
 #   setup the environment
 env=Environment(
     LIBPATH=['./',BOOSTLIBPATH, IM_LIB_PATH],
     CPPPATH=[BOOST, PYTHON_INCLUDE] + IM_INCLUDE_PATH,
-    RPATH=['./',BOOSTLIBPATH],
+    RPATH=[BOOSTLIBPATH],
     CPPFLAGS=CPP_FLAGS
 )
+env.SConsignFile()
 
 def fix_includes(target,source,env):
     for t in target:
