diff -udBbr kdeedu-3.2.2/work/kdeedu-3.2.2/kstars/kstars/indi/indidevapi.h kdeedu-3.2.2-fix1/work/kdeedu-3.2.2/kstars/kstars/indi/indidevapi.h
--- kdeedu-3.2.2/work/kdeedu-3.2.2/kstars/kstars/indi/indidevapi.h	2004-04-04 11:47:53.000000000 +0400
+++ kdeedu-3.2.2-fix1/work/kdeedu-3.2.2/kstars/kstars/indi/indidevapi.h	2004-04-24 09:04:04.812573952 +0400
@@ -155,9 +155,9 @@
 /* function to set all property switches off */
 extern void IUResetSwitches(const ISwitchVectorProperty *svp);
 
-extern int IUUpdateSwitches(const ISwitchVectorProperty *svp, ISState *states, char *names[], int n);
+extern int IUUpdateSwitches(ISwitchVectorProperty *svp, ISState *states, char *names[], int n);
 
-extern int IUUpdateNumbers(const INumberVectorProperty *nvp, double values[], char *names[], int n);
+extern int IUUpdateNumbers(INumberVectorProperty *nvp, double values[], char *names[], int n);
 
 /* function to reliably save new text in a IText */
 extern void IUSaveText (IText *tp, const char *newtext);
diff -udBbr kdeedu-3.2.2/work/kdeedu-3.2.2/kstars/kstars/indi/indidrivermain.c kdeedu-3.2.2-fix1/work/kdeedu-3.2.2/kstars/kstars/indi/indidrivermain.c
--- kdeedu-3.2.2/work/kdeedu-3.2.2/kstars/kstars/indi/indidrivermain.c	2004-04-04 11:47:53.000000000 +0400
+++ kdeedu-3.2.2-fix1/work/kdeedu-3.2.2/kstars/kstars/indi/indidrivermain.c	2004-04-24 09:07:50.076328656 +0400
@@ -46,11 +46,11 @@
 static void clientMsgCB(int fd, void *arg);
 static int dispatch (XMLEle *root, char msg[]);
 static int crackDN (XMLEle *root, char **dev, char **name, char msg[]);
-static char *pstateStr(IPState s);
-static char *sstateStr(ISState s);
-static char *ruleStr(ISRule r);
-static char *permStr(IPerm p);
-static char *timestamp (void);
+static const char *pstateStr(IPState s);
+static const char *sstateStr(ISState s);
+static const char *ruleStr(ISRule r);
+static const char *permStr(IPerm p);
+static const char *timestamp (void);
 
 static int verbose;			/* chatty */
 char *me;				/* a.out name */
@@ -546,7 +546,7 @@
 
 /* Update property switches in accord with states and names. */
 int 
-IUUpdateSwitches(const ISwitchVectorProperty *svp, ISState *states, char *names[], int n)
+IUUpdateSwitches(ISwitchVectorProperty *svp, ISState *states, char *names[], int n)
 {
  int i=0;
  
@@ -571,7 +571,7 @@
 }
 
 /* Update property numbers in accord with values and names */
-int IUUpdateNumbers(const INumberVectorProperty *nvp, double values[], char *names[], int n)
+int IUUpdateNumbers(INumberVectorProperty *nvp, double values[], char *names[], int n)
 {
   int i=0;
   
@@ -724,7 +724,7 @@
 	    static char **names;
 	    static int maxn;
 	    char *dev, *name;
-	    XMLEle *ep;
+	    /* XMLEle *ep; */
 
 	    /* pull out device and name */
 	    if (crackDN (root, &dev, &name, msg) < 0)
@@ -856,7 +856,7 @@
 }
 
 /* return static string corresponding to the given property or light state */
-static char *
+static const char *
 pstateStr (IPState s)
 {
 	switch (s) {
@@ -871,7 +871,7 @@
 }
 
 /* return static string corresponding to the given switch state */
-static char *
+static const char *
 sstateStr (ISState s)
 {
 	switch (s) {
@@ -884,7 +884,7 @@
 }
 
 /* return static string corresponding to the given Rule */
-static char *
+static const char *
 ruleStr (ISRule r)
 {
 	switch (r) {
@@ -898,7 +898,7 @@
 }
 
 /* return static string corresponding to the given IPerm */
-static char *
+static const char *
 permStr (IPerm p)
 {
 	switch (p) {
@@ -912,7 +912,7 @@
 }
 
 /* return current system time in message format */
-static char *
+static const char *
 timestamp()
 {
 	static char ts[32];
