--- Makefile.am.orig	2005-06-11 15:49:35.000000000 +1000
+++ Makefile.am	2005-06-11 15:48:22.000000000 +1000
@@ -105,11 +105,11 @@
 	cp pcap-namedb.h $(DESTDIR)$(includedir)
 	rm -f $(DESTDIR)$(libdir)/libpcap.la
 	rm -f $(DESTDIR)$(libdir)/libpcap.so*
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so)
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so.0.$(RMINOR).$(RREVISION))
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so.0.$(RMINOR))
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so.0.7)
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so.0)
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).${VAGE) libpcap.so)
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).${VAGE) libpcap.so.0.$(RMINOR).$(RREVISION))
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).${VAGE) libpcap.so.0.$(RMINOR))
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).${VAGE) libpcap.so.0.7)
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).${VAGE) libpcap.so.0)
 
 # generated source
 
--- Makefile.in.orig	2005-06-11 15:42:42.000000000 +1000
+++ Makefile.in	2005-06-11 15:50:23.000000000 +1000
@@ -514,11 +514,11 @@
 	cp pcap-namedb.h $(DESTDIR)$(includedir)
 	rm -f $(DESTDIR)$(libdir)/libpcap.la
 	rm -f $(DESTDIR)$(libdir)/libpcap.so*
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so)
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so.0.$(RMINOR).$(RREVISION))
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so.0.$(RMINOR))
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so.0.7)
-	(cd $(DESTDIR)$(libdir); ln -s libpcap-0.$(RMINOR).$(RREVISION).so libpcap.so.0)
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).$(VAGE) libpcap.so)
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).$(VAGE) libpcap.so.0.$(RMINOR).$(RREVISION))
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).$(VAGE) libpcap.so.0.$(RMINOR))
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).$(VAGE) libpcap.so.0.7)
+	(cd $(DESTDIR)$(libdir); ln -s libpcap.so.$(VCURRENT).$(VREVISION).$(VAGE) libpcap.so.0)
 
 # generated source
 
