--- mozilla/security/coreconf/Linux.mk.orig     2004-08-26 00:47:42.000000000 -0600
+++ mozilla/security/coreconf/Linux.mk  2004-08-26 00:48:25.000000000 -0600
@@ -90,6 +90,14 @@
        OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
        CPU_ARCH        = s390x
 else
+ifeq ($(OS_TEST),mips)
+       OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+       CPU_ARCH        = mips
+else
+ifeq ($(OS_TEST),mips64)
+       OS_REL_CFLAGS   = -DLINUX1_2 -D_XOPEN_SOURCE
+       CPU_ARCH        = mips
+else
        OS_REL_CFLAGS   = -DLINUX1_2 -Di386 -D_XOPEN_SOURCE
        CPU_ARCH        = x86
 endif
@@ -102,6 +110,8 @@
 endif
 endif
 endif
+endif
+endif
 
 
 LIBC_TAG               = _glibc
