Index: mozilla/security/coreconf/Linux.mk
===================================================================
RCS file: /cvsroot/mozilla/security/coreconf/Linux.mk,v
retrieving revision 1.27
diff -u -r1.27 Linux.mk
--- mozilla/security/coreconf/Linux.mk	16 Sep 2005 17:09:23 -0000	1.27
+++ mozilla/security/coreconf/Linux.mk	15 Dec 2005 21:25:00 -0000
@@ -165,7 +165,7 @@
 ARCH			= linux
 
 DSO_CFLAGS		= -fPIC
-DSO_LDOPTS		= -shared $(ARCHFLAG)
+DSO_LDOPTS		= -shared $(ARCHFLAG) -Wl,-z,defs
 DSO_LDFLAGS		=
 LDFLAGS			+= $(ARCHFLAG)
 
