--- gst-plugins-0.4.2/sys/oss/gstosssink.c.orig	2002-12-27 18:37:40.000000000 +0200
+++ gst-plugins-0.4.2/sys/oss/gstosssink.c	2002-12-27 18:38:56.000000000 +0200
@@ -886,11 +886,23 @@
   GST_INFO (GST_CAT_PLUGIN_INFO, "osssink: attempting to open sound device");
 
   /* first try to open the sound card */
+  /* FIXME: This code is dubious, why do we need to open and close this ?
+   *        For linux at least this causes the second open to never return
+   *        if the device was already in use .. */
   sink->fd = open (sink->device, O_WRONLY | O_NONBLOCK);
   if (errno == EBUSY) {
     g_warning ("osssink: unable to open the sound device (in use ?)\n");
     return FALSE;
   }
+#ifdef __linux__
+  else if (sink->fd >= 0) {
+    /* remove the non-blocking flag */
+    if(fcntl (sink->fd, F_SETFL, 0) < 0) {
+      g_warning ("osssink: unable to make filedescriptor blocking\n");
+      return FALSE;
+    }
+  }
+#else
 
   /* re-open the sound device in blocking mode */
   close (sink->fd);
@@ -900,6 +912,7 @@
     g_warning ("osssink: unable to open the sound device (errno=%d)\n", errno); 
     return FALSE;
   }
+#endif
 	
   /* we have it, set the default parameters and go have fun */
   /* set card state */
