--- gst/playondemand/Makefile.in.orig	2002-12-06 18:25:37.000000000 -0600
+++ gst/playondemand/Makefile.in	2002-12-06 18:25:53.000000000 -0600
@@ -239,7 +239,7 @@
 am_demo_mp3_OBJECTS = demo_mp3-demo-mp3.$(OBJEXT)
 demo_mp3_OBJECTS = $(am_demo_mp3_OBJECTS)
 demo_mp3_LDADD = $(LDADD)
-demo_mp3_DEPENDENCIES =
+demo_mp3_DEPENDENCIES = ./libgstplayondemand.la
 
 DEFS = @DEFS@
 DEFAULT_INCLUDES =  -I. -I$(srcdir) -I$(top_builddir)
