--- mh/Makefile.in.orig	2003-06-07 05:37:31.000000000 -0400
+++ mh/Makefile.in	2004-06-30 12:14:46.979325522 -0400
@@ -952,11 +952,8 @@
 install-exec-hook: @BUILD_MH_EXEC_HOOK@
 
 mh_finish_install:
-	@here=`pwd`; \
-	cd $(DESTDIR); \
-	rm -f $(bindir)/folders; \
-	$(LN_S) $(bindir)/folder $(bindir)/folders; \
-	cd $$here
+	rm -f $(DESTDIR)/$(bindir)/folders; \
+	$(LN_S) folder $(DESTDIR)/$(bindir)/folders
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
