--- gtkextra/gtksheet.c.old	2006-09-25 11:04:06.000000000 -0500
+++ gtkextra/gtksheet.c	2006-09-25 11:04:33.000000000 -0500
@@ -4149,7 +4149,7 @@
 			   &pixmap_width, &pixmap_height);
       if ((pixmap_width != width) || (pixmap_height != height))
 	{
-          g_free(sheet->pixmap);
+          g_object_unref(sheet->pixmap);
 	  sheet->pixmap = gdk_pixmap_new (sheet->sheet_window,
 					       width, height,
 					       -1);
