diff -ur gtk+extra/gtkextra/gtksheet.c gtk+extra-2.1.1/gtkextra/gtksheet.c
--- gtk+extra/gtkextra/gtksheet.c	2008-02-01 01:04:09.000000000 +0100
+++ gtk+extra-2.1.1/gtkextra/gtksheet.c	2008-02-01 01:08:52.000000000 +0100
@@ -7956,11 +7956,10 @@
     }
 
     tbl->maxallocrow-=MIN(nrows,tbl->maxallocrow-row+1);
-    tbl->maxallocrow = MIN(tbl->maxallocrow, tbl->maxrow);
-
   }
 
   tbl->maxrow-=nrows;
+  tbl->maxallocrow = MIN(tbl->maxallocrow, tbl->maxrow);
   gtk_sheet_recalc_top_ypixels(tbl, 0);
   return TRUE;
 } 
