--- fenris/dress.c	2002-08-05 16:08:56.000000000 +0100
+++ fenris.new/dress.c	2004-01-08 23:34:03.536699488 +0000
@@ -141,7 +141,7 @@
 int main(int argc,char* argv[]) {
   bfd* b;
   char opt;
-  struct sec* ss;
+  asection* ss;
   int fi;
   int i;
 
@@ -263,7 +263,7 @@
 
 void copier(char* src,char* dst,char* secname) {
 
-  struct sec* s;
+  asection* s;
   bfd *ibfd,*obfd;
 
   bfd_init();
@@ -288,7 +288,7 @@
   debug("[+] Setting up sections: ");
 
   while (s) {
-    struct sec* os;
+    asection* os;
 
     os=bfd_make_section_anyway(obfd,bfd_section_name(ibfd,s));
     if (s->name[0]=='.') debug("%s ",bfd_section_name(ibfd,s));
