--- libcaca-0.99.beta12-new/src/cacaserver.c	2007-11-06 13:19:34 +0000
+++ libcaca-0.99.beta12/src/cacaserver.c	2007-11-06 13:19:50 +0000
@@ -25,13 +25,14 @@
 /* for sockaddr_in (not included from arpa/inet.h in 10.3.9 SDK) */
 #       include <netinet/in.h>
 #   endif
-#elif defined(HAVE_NETINET_IN_H)
-#   include <netinet/in.h>
 #elif defined(HAVE_WINSOCK2_H)
 #   include <winsock2.h>
 #   include <ws2tcpip.h>
 #   define USE_WINSOCK 1
 #endif
+#if defined(HAVE_NETINET_IN_H)
+#   include <netinet/in.h>
+#endif
 #if defined(HAVE_UNISTD_H)
 #   include <unistd.h>
 #endif
