Patch adopted from debians patches

diff --git a/src/hostmon.c b/src/hostmon.c
index 14df2c8..7e50575 100644
--- a/src/hostmon.c
+++ b/src/hostmon.c
@@ -711,7 +711,7 @@ void hostmon(const struct OPTIONS *options, int facilitytime, char *ifptr,
     int is_ip;
     int ch;
 
-    char ifname[10];
+    char ifname[18];
 
     struct timeval tv;
     unsigned long starttime;
diff --git a/src/ifstats.c b/src/ifstats.c
index fac4db5..583279a 100644
--- a/src/ifstats.c
+++ b/src/ifstats.c
@@ -129,7 +129,7 @@ void initiflist(struct iflist **list)
 {
     FILE *fd;
     char buf[161];
-    char ifname[10];
+    char ifname[18];
     struct iflist *itmp = NULL;
     struct iflist *tail = NULL;
     unsigned int index = 0;
@@ -433,7 +433,7 @@ void ifstats(const struct OPTIONS *options, struct filterstate *ofilter,
     FILE *logfile = NULL;
 
     int br;
-    char ifname[10];
+    char ifname[18];
 
     int ch;
 
@@ -760,7 +760,7 @@ void detstats(char *iface, const struct OPTIONS *options, int facilitytime,
     char *tpacket;
     unsigned int iphlen;
 
-    char ifname[10];
+    char ifname[18];
     struct sockaddr_ll fromaddr;
     unsigned short linktype;
 
diff --git a/src/ifstats.h b/src/ifstats.h
index 350dd13..abcde1e 100644
--- a/src/ifstats.h
+++ b/src/ifstats.h
@@ -6,7 +6,7 @@ ifstats.h - structure definitions for interface counts
 ***/
 
 struct iflist {
-    char ifname[8];
+    char ifname[18];
     unsigned int encap;
     unsigned long long iptotal;
     unsigned long badtotal;
diff --git a/src/itrafmon.c b/src/itrafmon.c
index a54dcc2..044bae1 100644
--- a/src/itrafmon.c
+++ b/src/itrafmon.c
@@ -604,7 +604,7 @@ void ipmon(struct OPTIONS *options,
     int curwin = 0;
 
     int readlen;
-    char ifname[10];
+    char ifname[18];
 
     unsigned long long total_pkts = 0;
 
diff --git a/src/othptab.h b/src/othptab.h
index e412562..ab0d5be 100644
--- a/src/othptab.h
+++ b/src/othptab.h
@@ -42,7 +42,7 @@ struct othptabent {
     int s_fstat;
     int d_fstat;
     unsigned int protocol;
-    char iface[8];
+    char iface[18];
     unsigned int pkt_length;
 
     union {
diff --git a/src/promisc.c b/src/promisc.c
index c3ed691..d126243 100644
--- a/src/promisc.c
+++ b/src/promisc.c
@@ -49,7 +49,7 @@ void init_promisc_list(struct promisc_states **list)
 {
     FILE *fd;
     int ifd;
-    char buf[8];
+    char buf[18];
     struct promisc_states *ptmp;
     struct promisc_states *tail = NULL;
     struct ifreq ifr;
@@ -81,6 +81,7 @@ void init_promisc_list(struct promisc_states **list)
              */
 
             if ((strncmp(buf, "eth", 3) == 0) ||
+                (strncmp(buf, "ath", 3) == 0) ||
                 (strncmp(buf, "fddi", 4) == 0) ||
                 (strncmp(buf, "tr", 2) == 0) ||
                 (strncmp(ptmp->params.ifname, "wvlan", 4) == 0) ||
diff --git a/src/promisc.h b/src/promisc.h
index c3f07a1..a06471c 100644
--- a/src/promisc.h
+++ b/src/promisc.h
@@ -9,7 +9,7 @@
  */
 
 struct promisc_params {
-    char ifname[8];
+    char ifname[18];
     int saved_state;
     int state_valid;
 };
diff --git a/src/serv.c b/src/serv.c
index c8309ad..f7e7daf 100644
--- a/src/serv.c
+++ b/src/serv.c
@@ -739,7 +739,7 @@ void servmon(char *ifname, struct porttab *ports,
     unsigned short linktype;
     int br;
 
-    char iface[8];
+    char iface[18];
     unsigned int idx = 1;
 
     unsigned int sport = 0;
diff --git a/src/tcptable.h b/src/tcptable.h
index 3e17793..c029c73 100644
--- a/src/tcptable.h
+++ b/src/tcptable.h
@@ -64,7 +64,7 @@ struct tcptableent {
     unsigned long finack;
     int partial;
     int finsent;
-    char ifname[8];
+    char ifname[18];
     unsigned int index;
     int reused;
     int timedout;
