diff -Naur universalindentgui-1.1.0.orig/src/IndentHandler.cpp universalindentgui-1.1.0/src/IndentHandler.cpp
--- universalindentgui-1.1.0.orig/src/IndentHandler.cpp	2009-07-16 13:52:38.000000000 +0200
+++ universalindentgui-1.1.0/src/IndentHandler.cpp	2010-06-26 20:30:41.000000000 +0200
@@ -514,6 +514,10 @@
     //indentProcess.setReadChannelMode(QProcess::MergedChannels);
 
     // Set the directory for the indenter execution
+    QStringList env = indentProcess.environment();
+    env << "PWD=" + QFileInfo(tempDirctoryStr).absoluteFilePath();
+    indentProcess.setEnvironment( env );
+    
     indentProcess.setWorkingDirectory( QFileInfo(tempDirctoryStr).absoluteFilePath() );
 
     qDebug() << __LINE__ << " " << __FUNCTION__ << ": Will call the indenter in the directory " << indentProcess.workingDirectory() << " using this commandline call: " << indenterCompleteCallString;
@@ -558,6 +562,7 @@
         QApplication::restoreOverrideCursor();
         errorMessageDialog->showMessage(tr("Error calling Indenter"), processReturnString);
     }
+    
 
     // If the indenter returned an error code != 0 show its output.
     if ( indentProcess.exitCode() != 0 ) {
