--- Src/Linux/Makefile.orig	2003-08-14 13:11:45.664823032 -0400
+++ Src/Linux/Makefile	2003-08-14 13:13:07.997306592 -0400
@@ -5,13 +5,17 @@
 VERSION = 0
 BUILD = 5
 PLUGIN = libGSsoftx-${VERSION}.${BUILD}.so
-CFLAGS+= -fPIC -Wall -O3 -fomit-frame-pointer -I. -I..
+CFLAGS+= -fPIC -Wall $(OPTFLAGS) -I. -I..
 OBJS = ../GS.o ../Prim.o ../Soft.o ../Draw.o ../Texts.o ../Color.o
 OBJS+= X11.o Conf.o interface.o support.o DrawString.o
 DEPS:= $(OBJS:.o=.d)
 LIBS = $(shell gtk-config --libs) -lX11
 CFLAGS+= $(shell gtk-config --cflags) -D__LINUX__ -DVERSION=${VERSION} -DBUILD=${BUILD}
 
+ifeq (${USE_SDL},1)
+	CFLAGS+= $(shell sdl-config --cflags) -I../../nsx2/ -I../../nsx2/iniapi
+	LIBS+= $(shell sdl-config --libs)
+endif
 ifeq (${USE_DGA2}, TRUE)
 	CFLAGS+= -DUSE_DGA2
 	LIBS+= -lXxf86dga
