diff -ur emu10k1-v0.20a/utils/as10k1/as10k1.c emu10k1-v0.20a-gentoo/utils/as10k1/as10k1.c
--- emu10k1-v0.20a/utils/as10k1/as10k1.c	2003-09-16 04:01:01.000000000 +1200
+++ emu10k1-v0.20a-gentoo/utils/as10k1/as10k1.c	2003-09-16 04:25:18.000000000 +1200
@@ -39,33 +39,32 @@
 
 char version[]="As10k1 assembler version 1.0";
 
-char help[]="
-Usage: as10k1 [option...] asmfile
-
-The as10k1 assembler is for the emu10k1 dsp processor
-found in Creative Lab's sblive series of sound cards.
-
-Author: Daniel Bertrand <d.bertrand@ieee.org>
-BUGS:   emu10k1-devel mailing list <emu10k1-devel@lists.sourceforge.net>
-URLs:   members.home.com/d-bertrand, opensource.creative.com
-
-Options:\n
-     -l [listfile]         Specify a listing file, default is none.
-     -o [binfile]          Specify an output file, default is based input,
-                           Substituting .asm fo .bin
-     -d [dbg_options...]   Turn on debug messages.
-            s              prints all symbols                
-            g              prints defined gprs
-            t              prints tram usage
-            i              prints instructions
-
-     -n                    no gpr declaration check 
-     -h                    Prints this message  
-     -v                    Prints version number. 
-
-This program is distributed under the GPL.
-
-";
+char help[]=
+"Usage: as10k1 [option...] asmfile\n"
+"\n"
+"The as10k1 assembler is for the emu10k1 dsp processor\n"
+"found in Creative Lab's sblive series of sound cards.\n"
+"\n"
+"Author: Daniel Bertrand <d.bertrand@ieee.org>\n"
+"BUGS:   emu10k1-devel mailing list <emu10k1-devel@lists.sourceforge.net>\n"
+"URLs:   members.home.com/d-bertrand, opensource.creative.com\n"
+"\n"
+"Options:\n\n"
+"     -l [listfile]         Specify a listing file, default is none.\n"
+"     -o [binfile]          Specify an output file, default is based input,\n"
+"                           Substituting .asm fo .bin\n"
+"     -d [dbg_options...]   Turn on debug messages.\n"
+"            s              prints all symbols        \n"
+"            g              prints defined gprs\n"
+"            t              prints tram usage\n"
+"            i              prints instructions\n"
+"\n"
+"     -n                    no gpr declaration check \n"
+"     -h                    Prints this message  \n"
+"     -v                    Prints version number. \n"
+"\n"
+"This program is distributed under the GPL.\n"
+"\n";
 
 //  cli==Command-Line Interface && !Creative Labs Inc.
 
diff -ur emu10k1-v0.20a/utils/mixer/lib/debug.c emu10k1-v0.20a-gentoo/utils/mixer/lib/debug.c
--- emu10k1-v0.20a/utils/mixer/lib/debug.c	2003-09-16 04:05:59.000000000 +1200
+++ emu10k1-v0.20a-gentoo/utils/mixer/lib/debug.c	2003-09-16 04:30:02.000000000 +1200
@@ -10,7 +10,7 @@
  * 
  *     This program is distributed in the hope that it will be useful, 
  *     but WITHOUT ANY WARRANTY; without even the implied warranty of 
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  *     GNU General Public License for more details. 
  * 
  *     You should have received a copy of the GNU General Public 
@@ -360,20 +360,19 @@
 		else
 			k=i;
 	}
-	printf("\
-Usage Summary:
---------
-Instructions : %d
-GPRS:          %d (last used is %d)
-
-Tram
- Internal
-   buffers:    %d
-   space  :    0x%06x
- External
-   buffers:    %d
-   space  :    0x%06x
-",
+	printf(
+"Usage Summary:\n"
+"--------\n"
+"Instructions : %d\n"
+"GPRS:          %d (last used is %d)\n"
+"\n"
+"Tram\n"
+" Internal\n"
+"   buffers:    %d\n"
+"   space  :    0x%06x\n"
+" External\n"
+"   buffers:    %d\n"
+"   space  :    0x%06x\n",
 	       mgr->code_free_start/2, j,k,
 	       mgr->traml_ifree_start,  mgr->tramb_ifree_start,
 	       mgr->traml_efree_start,  mgr->tramb_efree_start);
diff -ur emu10k1-v0.20a/utils/mixer/lib/dsp.c emu10k1-v0.20a-gentoo/utils/mixer/lib/dsp.c
--- emu10k1-v0.20a/utils/mixer/lib/dsp.c	2003-09-16 04:05:59.000000000 +1200
+++ emu10k1-v0.20a-gentoo/utils/mixer/lib/dsp.c	2003-09-16 04:29:44.000000000 +1200
@@ -10,7 +10,7 @@
  * 
  *     This program is distributed in the hope that it will be useful, 
  *     but WITHOUT ANY WARRANTY; without even the implied warranty of 
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  *     GNU General Public License for more details. 
  * 
  *     You should have received a copy of the GNU General Public 
@@ -1403,12 +1403,12 @@
 int dsp_print_in_name(struct dsp_patch_manager *mgr){
 	int i;
 
-	printf("
-Available Inputs
-
-Mono Name       Stereo Name      Address
----------       -----------      -------
-");
+	printf(
+"Available Inputs\n"
+"\n"
+"Mono Name       Stereo Name      Address\n"
+"---------       -----------      -------\n"
+);
 	if(mgr->card_info.card_type==AUDIGY_CARD)
 		for(i=0;i<A_DSP_NUM_INPUTS;i++){
 			if(dsp_audigy_in_name[i][0]!='\0')
@@ -1425,12 +1425,12 @@
 int dsp_print_out_name(struct dsp_patch_manager *mgr){
 	int i;
 
-	printf("
-Available Outputs
-
-Mono Name       Stereo Name      Address
----------       -----------      -------
-");
+	printf(
+"Available Outputs\n"
+"\n"
+"Mono Name       Stereo Name      Address\n"
+"---------       -----------      -------\n"
+);
 	if(mgr->card_info.card_type==AUDIGY_CARD)
 		for(i=0;i<A_DSP_NUM_OUTPUTS;i++){
 			if(dsp_audigy_out_name[i][0]!='\0')
