diff -ur gyach-0.9.4.orig/src/yahoochat.c gyach-0.9.4/src/yahoochat.c
--- gyach-0.9.4.orig/src/yahoochat.c	2003-06-07 23:00:45.000000000 -0500
+++ gyach-0.9.4/src/yahoochat.c	2003-09-25 10:36:29.421864072 -0500
@@ -209,7 +209,7 @@
 				*((u_long *)ptr) = htonl( 0x06000000 );	ptr += 4;  
 				break;
 		default:
-				*((u_long *)ptr) = htonl( 0x09000000 );	ptr += 4;  
+				*((u_long *)ptr) = htonl( 0x000b0000 );	ptr += 4;  
 				break;
 	}
 
