=== modified file 'pkgcore_checks/glsa_scan.py'
--- pkgcore_checks/glsa_scan.py	2006-09-05 07:43:33 +0000
+++ pkgcore_checks/glsa_scan.py	2006-09-30 23:59:10 +0000
@@ -33,7 +33,7 @@
             if not os.path.isdir(glsa_loc):
                 # form of 'optional' limiting; if they are using -c, force the
                 # error, else disable
-                if options.check_to_run:
+                if options.checks_to_run:
                     raise optparse.OptionValueError("--glsa-dir must be "
                         "specified, couldn't identify glsa src from %r" %
                             options.src_repo)

