=== modified file 'pkgcore_checks/unported_mod_x.py'
--- pkgcore_checks/unported_mod_x.py	2006-10-13 00:50:12 +0000
+++ pkgcore_checks/unported_mod_x.py	2006-11-03 06:04:12 +0000
@@ -182,11 +182,14 @@
                         continue
                     elif h in cache:
                         break
-                    elif a not in query_cache:
+                    elif h not in query_cache:
                         query_cache[h] = caching_iter(repo.itermatch(a))
                     if any(True for pkg in query_cache[h] if
                         vfilter.match(pkg)):
                         # one is visible.
+                        cache.add(h)
                         break
+                    else:
+                        insoluable.add(h)
                 else:
                     failed.update(modx_candidates)

