diff -urP linux-2.4.4-xfs-20010523-gentoo/Makefile linux-2.4.4-xfs-20010523-cleanup-gentoo/Makefile
--- linux-2.4.4-xfs-20010523-gentoo/Makefile	Wed May 23 21:40:04 2001
+++ linux-2.4.4-xfs-20010523-cleanup-gentoo/Makefile	Wed May 23 21:40:40 2001
@@ -105,7 +105,7 @@
 CPPFLAGS := -D__KERNEL__ -I$(HPATH)
 CPPFLAGS += $(patsubst %,-I%,$(CROSS_COMPILE_INC))
 
-CFLAGS := $(CPPFLAGS) -Wformat -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -fno-strict-aliasing
+CFLAGS := $(CPPFLAGS) -Wformat -Wall -Wstrict-prototypes -O2 -fno-strict-aliasing
 ifndef CONFIG_FRAME_POINTER
 CFLAGS += -fomit-frame-pointer
 endif
