diff -ruN valgrind-3.0.0/none/tests/x86/Makefile.in valgrind-3.0.0-fixed/none/tests/x86/Makefile.in
--- valgrind-3.0.0/none/tests/x86/Makefile.in	2005-08-03 22:43:14.000000000 +0200
+++ valgrind-3.0.0-fixed/none/tests/x86/Makefile.in	2005-08-04 22:52:05.563599512 +0200
@@ -151,7 +151,7 @@
 CCAS = @CCAS@
 CCASFLAGS = @CCASFLAGS@
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -fno-pie
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CXX = @CXX@
diff -ruN valgrind-3.0.0/tests/Makefile.in valgrind-3.0.0-fixed/tests/Makefile.in
--- valgrind-3.0.0/tests/Makefile.in	2005-08-03 22:43:14.000000000 +0200
+++ valgrind-3.0.0-fixed/tests/Makefile.in	2005-08-04 22:52:15.937022512 +0200
@@ -93,7 +93,7 @@
 CCAS = @CCAS@
 CCASFLAGS = @CCASFLAGS@
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -fno-pie
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CXX = @CXX@
