diff -ruN valgrind-2.2.0/coregrind/vg_syscalls.c valgrind-2.2.0-fixed/coregrind/vg_syscalls.c
--- valgrind-2.2.0/coregrind/vg_syscalls.c	2004-08-31 02:08:37.000000000 +0200
+++ valgrind-2.2.0-fixed/coregrind/vg_syscalls.c	2005-02-22 20:20:38.904577288 +0100
@@ -900,7 +900,7 @@
    Addr ret = VG_(brk_limit);
    static const Bool debug = False;
    Segment *seg;
-   Addr current, newaddr;
+   Addr currentaddr, newaddr;
 
 
    if (debug)
@@ -924,20 +924,20 @@
    if (seg != NULL && newbrk > seg->addr)
       return VG_(brk_limit);
 
-   current = PGROUNDUP(VG_(brk_limit));
+   currentaddr = PGROUNDUP(VG_(brk_limit));
    newaddr = PGROUNDUP(newbrk);
-   if (newaddr != current) {
+   if (newaddr != currentaddr) {
 
       /* new brk in a new page - fix the mappings */
       if (newbrk > VG_(brk_limit)) {
 	 
 	 if (debug)
 	    VG_(printf)("  extending brk: current=%p newaddr=%p delta=%d\n",
-			current, newaddr, newaddr-current);
+			currentaddr, newaddr, newaddr-currentaddr);
 
-	 if (newaddr == current) {
+	 if (newaddr == currentaddr) {
 	    ret = newbrk;
-         } else if ((void*)-1 != VG_(mmap)((void*)current, newaddr-current,
+         } else if ((void*)-1 != VG_(mmap)((void*)currentaddr, newaddr-currentaddr,
                VKI_PROT_READ|VKI_PROT_WRITE|VKI_PROT_EXEC,
                VKI_MAP_PRIVATE|VKI_MAP_ANONYMOUS|VKI_MAP_FIXED|VKI_MAP_CLIENT,
                SF_FIXED|SF_BRK, -1, 0)) 
@@ -949,10 +949,10 @@
 
 	 if (debug)
 	    VG_(printf)("  shrinking brk: current=%p newaddr=%p delta=%d\n",
-			current, newaddr, current-newaddr);
+			currentaddr, newaddr, currentaddr-newaddr);
 
-	 if (newaddr != current) {
-	    int res = VG_(munmap)((void *)newaddr, current - newaddr);
+	 if (newaddr != currentaddr) {
+	    int res = VG_(munmap)((void *)newaddr, currentaddr - newaddr);
             vg_assert(0 == res);
 	 }
 	 ret = newbrk;
