diff -ruN valgrind-2.4.1/configure valgrind-2.4.1-fixed/configure
--- valgrind-2.4.1/configure	2005-08-03 22:12:16.789460240 +0200
+++ valgrind-2.4.1-fixed/configure	2005-08-03 22:14:38.055984464 +0200
@@ -4735,6 +4735,7 @@
 if test "$handles_mmmx" = "yes"; then
   USE_MMX_REGISTER="-mmmx"
 fi
+CFLAGS=$safe_CFLAGS
 
 
 
@@ -4804,6 +4805,7 @@
 if test "$handles_msse" = "yes"; then
   USE_SSE_REGISTER="-msse"
 fi
+CFLAGS=$safe_CFLAGS
 
 
 
