Index: fbsd-6.1/usr.bin/rpcgen/rpc_hout.c
===================================================================
--- fbsd-6.1.orig/usr.bin/rpcgen/rpc_hout.c
+++ fbsd-6.1/usr.bin/rpcgen/rpc_hout.c
@@ -344,8 +344,7 @@ pprogramdef(def)
 
 		if(!Cflag){
 			ext = "extern  ";
-			f_print(fout, "%s", ext);
-			pdispatch(def->def_name, vers->vers_num, 2);
+			/* pdispatch(def->def_name, vers->vers_num, 2); */
 			for (proc = vers->procs; proc != NULL;
 			     proc = proc->next) {
 				if (!define_printed(proc,
@@ -373,8 +372,7 @@ pprogramdef(def)
 					ext = "extern  ";
 				}
 
-				f_print(fout, "%s", ext);
-				pdispatch(def->def_name, vers->vers_num, i);
+				/* pdispatch(def->def_name, vers->vers_num, i); */
 				for (proc = vers->procs; proc != NULL;
 				     proc = proc->next) {
 					if (!define_printed(proc,
