--- kivio/plugins/kivioconnectortool/straight_connector/Makefile.am	2002/05/25 16:08:19	1.8
+++ kivio/plugins/kivioconnectortool/straight_connector/Makefile.am	2002/09/13 08:30:39	1.8.2.1
@@ -2,7 +2,7 @@
 ####   Makefile.am generated automatically by KDEStudio   ####
 #### WARNING! All changes made in this part will be lost! ####
 ##############################################################
-
+libdir = $(kde_datadir)/kivio/autoloadStencils
 lib_LTLIBRARIES =libstraight_connector.la
 
 
