--- src/nodepath.cpp	2004-11-27 21:09:15.000000000
-0600
+++ src/nodepath.cpp	2004-12-13 04:06:14.000000000 -0600
@@ -2217,9 +2217,9 @@
 
     // remember the origin of the control
     if (n->p.knot == knot) {
-        n->p.origin = Radial(n->p.pos - n->pos);
+        n->p.origin = n->p.pos - n->pos;
     } else if (n->n.knot == knot) {
-        n->n.origin = Radial(n->n.pos - n->pos);
+        n->n.origin = n->n.pos - n->pos;
     } else {
         g_assert_not_reached();
     }

