--- inkscape-0.38.1/src/libnr/nr-maybe.h	2004/04/22 11:17:19	1.1
+++ inkscape-0.38.1/src/libnr/nr-maybe.h	2004/04/22 11:17:52
@@ -24,7 +24,7 @@
 /** An exception class for run-time type errors */
 template <typename T>
 class IsNot : public std::domain_error {
-    IsNot() : domain_error(string("Is not ") + typeid(T).name()) {}
+    IsNot() : domain_error(std::string("Is not ") + typeid(T).name()) {}
 };
 
 /** A type with only one value, which (in principle) is only equal to itself.
