--- unetbootin.cpp.orig	2009-02-03 23:02:03.000000000 +0100
+++ unetbootin.cpp	2009-02-14 18:58:44.000000000 +0100
@@ -2610,7 +2610,7 @@
 		QFile usbmbrF(rawtargetDev);
 		QFile mbrbinF(":/mbr.bin");
 		#ifdef NOSTATIC
-		mbrbinF.setFileName("/usr/lib/syslinux/mbr.bin");
+		mbrbinF.setFileName(QFile::exists("/usr/share/syslinux/mbr.bin") ? "/usr/share/syslinux/mbr.bin" : "/usr/lib/syslinux/mbr.bin");
 		#endif
 		usbmbrF.open(QIODevice::WriteOnly);
 		mbrbinF.open(QIODevice::ReadOnly);
