diff -ru PyOpenGL-2.0.0.44.orig/config/linux.cfg PyOpenGL-2.0.0.44/config/linux.cfg
--- PyOpenGL-2.0.0.44.orig/config/linux.cfg	2001-08-22 14:05:44.000000000 -0700
+++ PyOpenGL-2.0.0.44/config/linux.cfg	2003-04-30 15:26:32.000000000 -0700
@@ -11,8 +11,8 @@
 [General]
 build_togl=1
 gl_platform=GLX
-include_dirs=/usr/include:/usr/local/include:/usr/X11/include
-library_dirs=/usr/lib:/usr/local/lib:/usr/X11/lib
+include_dirs=/usr/include:/usr/local/include:/usr/X11R6/include:/usr/lib/tk8.4/include
+library_dirs=/usr/lib:/usr/local/lib:/usr/X11R6/lib
 
 ; a sys.pathsep separated list of the libs needed when linking GL
 [GL]
diff -ru PyOpenGL-2.0.0.44.orig/src/Togl-1.5/togl.c PyOpenGL-2.0.0.44/src/Togl-1.5/togl.c
--- PyOpenGL-2.0.0.44.orig/src/Togl-1.5/togl.c	2001-05-14 15:55:19.000000000 -0700
+++ PyOpenGL-2.0.0.44/src/Togl-1.5/togl.c	2003-04-30 15:23:14.000000000 -0700
@@ -137,6 +137,8 @@
 #  include "tkInt8.0p2.h"
 #elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==3
 #  include "tkInt8.3.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4
+#  include <generic/tkInt.h>
 #else
 #error Sorry, you will have to edit togl.c to include the right tkInt.h file
 #endif
