diff -urN cyrus-imapd-2.1.11/master/master.c cyrus-imapd-2.1.11-modified/master/master.c
--- cyrus-imapd-2.1.11/master/master.c	2002-11-01 10:44:33.000000000 -0600
+++ cyrus-imapd-2.1.11-modified/master/master.c	2002-12-07 18:21:41.000000000 -0600
@@ -103,6 +103,10 @@
 static int verbose = 0;
 static int listen_queue_backlog = 32;
 
+int allow_severity = LOG_DEBUG;
+int deny_severity = LOG_ERR;
+int hosts_ctl;
+
 struct service *Services = NULL;
 int allocservices = 0;
 int nservices = 0;
