--- eboard-0.9.5/board.cc.orig	2005-11-11 02:18:41.000000000 +0100
+++ eboard-0.9.5/board.cc	2005-11-11 02:19:20.000000000 +0100
@@ -1059,8 +1059,8 @@
   cerr.setf(ios::hex,ios::basefield);
   cerr.setf(ios::showbase);
 
-  cerr << "[board " << ((unsigned int)this) << "] ";
-  cerr << "game=[" << ((unsigned int)mygame) << "] ";
+  cerr << "[board " << ((unsigned long)this) << "] ";
+  cerr << "game=[" << ((unsigned long)mygame) << "] ";
 
   cerr.setf(ios::dec,ios::basefield);
   cerr << "paneid=" << getPageId() << endl;
--- eboard-0.9.5/chess.cc.orig	2005-11-11 02:19:45.000000000 +0100
+++ eboard-0.9.5/chess.cc	2005-11-11 02:20:02.000000000 +0100
@@ -641,7 +641,7 @@
   cerr.setf(ios::hex,ios::basefield);
   cerr.setf(ios::showbase);
 
-  cerr << "[game " << ((unsigned int)this) << "] ";
+  cerr << "[game " << ((unsigned long)this) << "] ";
 
   cerr.setf(ios::dec,ios::basefield);
 
@@ -655,7 +655,7 @@
   cerr.setf(ios::hex,ios::basefield);
   cerr.setf(ios::showbase);
 
-  cerr << "board=[" << ((unsigned int)myboard) << "]" << endl;
+  cerr << "board=[" << ((unsigned long)myboard) << "]" << endl;
 }
 
 char * ChessGame::getPlayerString(int index) {
--- eboard-0.9.5/notebook.cc.orig	2005-11-11 02:20:41.000000000 +0100
+++ eboard-0.9.5/notebook.cc	2005-11-11 02:20:54.000000000 +0100
@@ -127,7 +127,7 @@
 void Page::dump() {
   cerr.setf(ios::hex,ios::basefield);
   cerr.setf(ios::showbase);
-  cerr << "[page " << ((unsigned int)this) << "] ";
+  cerr << "[page " << ((unsigned long)this) << "] ";
   cerr.setf(ios::dec,ios::basefield);
   cerr << "pageid=" << number << " ";
   cerr << "title=" << title << " ";
