--- app_conference/conference.c.orig	2005-05-06 16:07:11.284165472 +0200
+++ app_conference/conference.c	2005-05-06 16:09:43.245063904 +0200
@@ -611,7 +611,7 @@
 	// acquire conference mutexes
 	ast_mutex_lock( &conf->lock ) ;
 	
-	if ( pthread_create( &conf->conference_thread, NULL, (void*)conference_exec, conf ) == 0 ) 
+	if ( ast_pthread_create( &conf->conference_thread, NULL, (void*)conference_exec, conf ) == 0 ) 
 	{
 		// detach the thread so it doesn't leak
 		pthread_detach( conf->conference_thread ) ;
--- app_conference/member.c.orig	2005-05-06 16:12:26.710213424 +0200
+++ app_conference/member.c	2005-05-06 16:14:36.121539912 +0200
@@ -71,10 +71,14 @@
 	//
 	// setup asterisk read/write formats
 	//
-	
+
+#ifdef ASTERISK_UNSTABLE_CVS
 	ast_log( AST_CONF_DEBUG, "CHANNEL INFO, CHANNEL => %s, DNID => %s, CALLER_ID => %s, ANI => %s\n", 
 		chan->name, chan->cid.cid_dnid, chan->cid.cid_num, chan->cid.cid_ani ) ;
-
+#else
+	ast_log( AST_CONF_DEBUG, "CHANNEL INFO, CHANNEL => %s, DNID => %s, CALLER_ID => %s, ANI => %s\n", 
+		chan->name, chan->dnid, chan->callerid, chan->ani ) ;
+#endif
 	ast_log( AST_CONF_DEBUG, "CHANNEL CODECS, CHANNEL => %s, NATIVE => %d, READ => %d, WRITE => %d\n", 
 		chan->name, chan->nativeformats, member->read_format, member->write_format ) ;
 
--- app_conference/app_conference.h.orig	2005-05-06 16:10:08.133280320 +0200
+++ app_conference/app_conference.h	2005-05-06 16:10:36.249006080 +0200
@@ -31,6 +31,7 @@
 #include <asterisk/channel.h>
 #include <asterisk/file.h>
 #include <asterisk/cli.h>
+#include <asterisk/utils.h>
 
 /* standard includes */
 #include <stdlib.h>
@@ -40,8 +41,6 @@
 #include <math.h>
 #include <stdio.h>
 
-#include <pthread.h>
-
 #if (SILDET == 2)
 #include "libspeex/speex_preprocess.h"
 #endif
