Index: branches/KDE/4.3/kdepim/ktimetracker/mainwindow.cpp
===================================================================
--- branches/KDE/4.3/kdepim/ktimetracker/mainwindow.cpp	(revision 1035758)
+++ branches/KDE/4.3/kdepim/ktimetracker/mainwindow.cpp	(revision 1035759)
@@ -73,6 +73,7 @@
       ((TimetrackerWidget) (m_part->widget())).openFile(icsfile);
       connect(configureAction, SIGNAL(triggered(bool)),
         m_part->widget(), SLOT(showSettingsDialog()));
+      ((TimetrackerWidget *) (m_part->widget()))->setupActions(actionCollection());
       m_part->openFile(icsfile);
       setupGUI();
     }
