--- src/coord.h	2003-09-29 14:06:14.000000000 +0200
+++ ../../../ssc-0.8/work/ssc-0.8/src/coord.h	2006-01-19 01:03:02.000000000 +0100
@@ -34,7 +34,7 @@
 	}
 
 	// copy constructor
-	Coord2(Self &other)
+	Coord2(const Self &other)
 		: x(other.x), y(other.y)
 	{
 	}
--- src/coord.cc	2003-09-28 19:34:11.000000000 +0200
+++ ../../../ssc-0.8/work/ssc-0.8/src/coord.cc	2006-01-19 07:46:47.000000000 +0100
@@ -1,7 +1,7 @@
 #include "coord.h"
 
-const Coord2<double> Coord2<double>::zero(0, 0);
-const Coord3<double> Coord3<double>::zero(0, 0, 0);
-const Coord3<double> Coord3<double>::up(0, 1, 0);
-const Coord3<double> Coord3<double>::forward(0, 0, 1);
-const Coord3<double> Coord3<double>::side(-1, 0, 0);
+template<> const Coord2<double> Coord2<double>::zero(0,0);
+template<> const Coord3<double> Coord3<double>::zero(0, 0, 0);
+template<> const Coord3<double> Coord3<double>::up(0, 1, 0);
+template<> const Coord3<double> Coord3<double>::forward(0, 0, 1);
+template<> const Coord3<double> Coord3<double>::side(-1, 0, 0);
--- src/font.cc	2003-09-27 17:15:07.000000000 +0200
+++ ../../../ssc-0.8/work/ssc-0.8/src/font.cc	2006-01-19 07:48:20.000000000 +0100
@@ -22,6 +22,7 @@
 
 #include <iostream>
 #include <iomanip>
+#include <math.h>
 #include "font.h"
 
 namespace OGLFT {
--- src/model.cc	2006-01-19 08:09:15.000000000 +0100
+++ src/model.cc.new	2006-01-19 08:09:00.000000000 +0100
@@ -111,7 +111,6 @@
 
 extern void DrawHeading();
 
-inline
 void Model::draw(double dt, bool doMove)
 {
 	draw::clearScreen();
