--- test/HttpHeaderTest.cc.orig	2008-03-15 22:53:10.000000000 +0100
+++ test/HttpHeaderTest.cc	2008-03-15 22:53:36.000000000 +0100
@@ -37,9 +37,9 @@
     
     SharedHandle<Range> range = httpHeader.getRange();
     
-    CPPUNIT_ASSERT_EQUAL(9223372036854775800LL, range->getStartByte());
-    CPPUNIT_ASSERT_EQUAL(9223372036854775801LL, range->getEndByte());
-    CPPUNIT_ASSERT_EQUAL(9223372036854775807LL, range->getEntityLength());
+    CPPUNIT_ASSERT_EQUAL((int64_t)9223372036854775800LL, range->getStartByte());
+    CPPUNIT_ASSERT_EQUAL((int64_t)9223372036854775801LL, range->getEndByte());
+    CPPUNIT_ASSERT_EQUAL((int64_t)9223372036854775807LL, range->getEntityLength());
   }
 }
 
