diff -ur xfce4-panel-4.6.2.orig/configure.in xfce4-panel-4.6.2/configure.in
--- xfce4-panel-4.6.2.orig/configure.in	2009-10-16 16:18:49.000000000 +0300
+++ xfce4-panel-4.6.2/configure.in	2009-10-24 19:36:32.000000000 +0300
@@ -117,7 +117,8 @@
 XDT_CHECK_PACKAGE([GTHREAD], [gthread-2.0], [2.12.0])
 XDT_CHECK_PACKAGE([GMODULE], [gmodule-2.0], [2.12.0])
 XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.6.0])
-XDT_CHECK_PACKAGE([LIBEXO], [exo-0.3], [0.3.100])
+XDT_CHECK_PACKAGE([LIBEXO], [exo-0.3], [0.3.100], [],
+  [XDT_CHECK_PACKAGE([LIBEXO], [exo-1], [0.5.0])])
 XDT_CHECK_PACKAGE([CAIRO], [cairo], [1.0.0])
 
 dnl **********************
