From f2abad09584e72c49f2064926d742e8cfab52698 Mon Sep 17 00:00:00 2001
From: Nick Schermer <nick@xfce.org>
Date: Sun, 27 Feb 2011 21:25:14 +0000
Subject: Tasklist: add button release event to proxy item (bug #7362).

(cherry picked from commit 8bb6d5f273053bc75d3a17cfdae26f7798e714d6)
---
diff --git a/plugins/tasklist/tasklist-widget.c b/plugins/tasklist/tasklist-widget.c
index 0ff7db6..783618a 100644
--- a/plugins/tasklist/tasklist-widget.c
+++ b/plugins/tasklist/tasklist-widget.c
@@ -2664,6 +2664,8 @@ xfce_tasklist_button_proxy_menu_item (XfceTasklistChild *child,
 
   g_signal_connect (G_OBJECT (mi), "button-press-event",
       G_CALLBACK (xfce_tasklist_button_button_press_event), child);
+  g_signal_connect (G_OBJECT (mi), "button-release-event",
+      G_CALLBACK (xfce_tasklist_button_button_release_event), child);
 
   /* TODO bold labels for urgent windows */
   /* TODO item dnd */
--
cgit v0.8.3.4
