--- libvirt-0.4.4/configure.in	2008-06-25 01:37:56.000000000 -0700
+++ libvirt-0.4.4-new/configure.in	2008-07-14 04:51:49.000000000 -0700
@@ -661,7 +661,41 @@
 fi
 AM_CONDITIONAL([WITH_STORAGE_FS], [test "$with_storage_fs" = "yes"])
 
-AC_PATH_PROG([QEMU_IMG], [qemu-img], [], [$PATH:/sbin:/usr/sbin:/bin:/usr/bin])
+AC_MSG_CHECKING([path the of the binary for qemu])
+AC_ARG_WITH([qemu-path], [AC_HELP_STRING([--with-qemu-path=[path of binary|/usr/bin/qemu]], [Path of the qemu binary])])
+if test "x$with_qemu_path" == "x" ; then
+    QEMU_BIN="/usr/bin/qemu"
+else
+    QEMU_BIN="$with_qemu_path"
+fi
+AC_SUBST([QEMU_BIN])
+AC_MSG_RESULT($QEMU_BIN)
+AC_DEFINE_UNQUOTED([QEMU_BIN],["$QEMU_BIN"],
+      [Path of the qemu program])
+
+AC_MSG_CHECKING([path the of the binary for qemu-kvm])
+AC_ARG_WITH([qemu-kvm-path], [AC_HELP_STRING([--with-qemu-kvm-path=[path of binary|/usr/bin/qemu-kvm]], [Path of the qemu-kvm binary])])
+if test "x$with_qemu_kvm_path" == "x" ; then
+    KVM_BIN="/usr/bin/qemu-kvm"
+else
+    KVM_BIN="$with_qemu_kvm_path"
+fi
+AC_SUBST([KVM_BIN])
+AC_MSG_RESULT($KVM_BIN)
+AC_DEFINE_UNQUOTED([KVM_BIN],["$KVM_BIN"],
+      [Name of the qemu-kvm program])
+
+AC_MSG_CHECKING([name the disk image binary for qemu is])
+AC_ARG_WITH([qemu-img-name], [AC_HELP_STRING([--with-qemu-img-name=[name of binary|qemu-img]], [Name of the qemu-img binary])])
+if test "x$with_qemu_img_name" == "x" ; then
+    QEMU_IMG_BIN="qemu-img"
+else
+    QEMU_IMG_BIN="$with_qemu_img_name"
+fi
+AC_SUBST([QEMU_IMG_BIN])
+AC_MSG_RESULT($QEMU_IMG_BIN)
+
+AC_PATH_PROG([QEMU_IMG], [$QEMU_IMG_BIN], [], [$PATH:/sbin:/usr/sbin:/bin:/usr/bin])
 if test -n "$QEMU_IMG" ; then
   AC_DEFINE_UNQUOTED([HAVE_QEMU_IMG], 1, [whether qemu-img is available for non-raw files])
   AC_DEFINE_UNQUOTED([QEMU_IMG],["$QEMU_IMG"],
--- libvirt-0.4.4/src/qemu_conf.c	2008-06-19 06:15:02.000000000 -0700
+++ libvirt-0.4.4-new/src/qemu_conf.c	2008-07-14 04:50:55.000000000 -0700
@@ -372,7 +372,7 @@
             if (access("/dev/kvm", F_OK) == 0 &&
                 virCapabilitiesAddGuestDomain(guest,
                                               "kvm",
-                                              "/usr/bin/qemu-kvm",
+                                              KVM_BIN,
                                               NULL,
                                               0,
                                               NULL) == NULL)
--- libvirt-0.4.4/src/qemu_driver.c	2008-06-12 07:52:53.000000000 -0700
+++ libvirt-0.4.4-new/src/qemu_driver.c	2008-07-14 04:50:27.000000000 -0700
@@ -1682,7 +1682,9 @@
 {
     if ((virFileExists("/usr/bin/qemu")) ||
         (virFileExists("/usr/bin/qemu-kvm")) ||
-        (virFileExists("/usr/bin/xenner"))) {
+        (virFileExists("/usr/bin/xenner")) ||
+        (virFileExists(QEMU_BIN)) ||
+        (virFileExists(KVM_BIN))) {
         if (getuid() == 0) {
             return("qemu:///system");
         } else {
