diff -urN libvirt-0.4.4/src/qemu_conf.c libvirt-0.4.4-new/src/qemu_conf.c
--- libvirt-0.4.4/src/qemu_conf.c	2008-06-19 13:15:02.000000000 +0000
+++ libvirt-0.4.4-new/src/qemu_conf.c	2008-07-02 20:47:51.000000000 +0000
@@ -2612,11 +2612,18 @@
                 break;
             }
 
-            snprintf(opt, PATH_MAX, "file=%s,if=%s,%sindex=%d%s",
-                     disk->src, qemudBusIdToName(disk->bus, 1),
-                     media ? media : "",
-                     idx,
-                     bootable ? ",boot=on" : "");
+            if (disk->device == QEMUD_DISK_CDROM) {
+                snprintf(opt, PATH_MAX, "file=%s,if=%s,%sindex=%d",
+                         disk->src, qemudBusIdToName(disk->bus, 1),
+                         media ? media : "",
+                         idx);
+            } else { 
+                snprintf(opt, PATH_MAX, "file=%s,if=%s,%sindex=%d%s",
+                         disk->src, qemudBusIdToName(disk->bus, 1),
+                         media ? media : "",
+                         idx,
+                         bootable ? ",boot=on" : "");
+            }
 
             ADD_ARG_LIT("-drive");
             ADD_ARG_LIT(opt);
