diff -Nur libvirt-0.6.4/src/storage_backend_fs.c libvirt-0.6.4-qemu-img/src/storage_backend_fs.c
--- libvirt-0.6.4/src/storage_backend_fs.c	2009-05-20 02:02:59.000000000 -0500
+++ libvirt-0.6.4-qemu-img/src/storage_backend_fs.c	2009-06-08 14:38:36.000000000 -0500
@@ -1255,7 +1255,7 @@
          * backing store, not really sure what use it serves though, and it
          * may cause issues with lvm. Untested essentially.
          */
-        if (!inputBackingPath ||
+        if (inputBackingPath &&
             !STREQ(inputBackingPath, vol->backingStore.path)) {
             virStorageReportError(conn, VIR_ERR_INTERNAL_ERROR,
                                   "%s", _("a different backing store can not "
