diff -Naur libdsk-1.1.10.orig/lib/Makefile.am libdsk-1.1.10/lib/Makefile.am
--- libdsk-1.1.10.orig/lib/Makefile.am	2006-04-16 00:48:10.000000000 +1200
+++ libdsk-1.1.10/lib/Makefile.am	2007-04-29 12:45:10.000000000 +1200
@@ -5,6 +5,7 @@
 JAVAH=@JAVAH@
 JAVA=@JAVA@
 JAVADOC=@JAVADOC@
+JAVACFLAGS=@JAVACFLAGS@
 JAR=@JAR@
 CLASSCPRE=uk.co.demon.seasip.libdsk
 CLASSDPRE=uk/co/demon/seasip/libdsk
@@ -83,10 +84,10 @@
 	$(JAVADOC) -d $(top_srcdir)/doc/jni $(CLASSCPRE)
 
 $(CLASSDPRE)/%.class:   $(srcdir)/$(CLASSDPRE)/%.java
-	here=`pwd` && cd $(srcdir) && $(JAVAC) -d $$here $<
+	here=`pwd` && cd $(srcdir) && $(JAVAC) $(JAVACFLAGS) -d $$here $<
 
 %.class:        $(srcdir)/%.java
-	here=`pwd` && cd $(srcdir) && $(JAVAC) -d $$here $<
+	here=`pwd` && cd $(srcdir) && $(JAVAC) $(JAVACFLAGS) -d $$here $<
 
 
 $(HEADERPRE)_%.h:         $(CLASSDPRE)/%.class
diff -Naur libdsk-1.1.10.orig/lib/Makefile.in libdsk-1.1.10/lib/Makefile.in
--- libdsk-1.1.10.orig/lib/Makefile.in	2006-04-20 09:12:10.000000000 +1200
+++ libdsk-1.1.10/lib/Makefile.in	2007-04-29 12:50:50.000000000 +1200
@@ -488,10 +488,10 @@
 	$(JAVADOC) -d $(top_srcdir)/doc/jni $(CLASSCPRE)
 
 $(CLASSDPRE)/%.class:   $(srcdir)/$(CLASSDPRE)/%.java
-	here=`pwd` && cd $(srcdir) && $(JAVAC) -d $$here $<
+	here=`pwd` && cd $(srcdir) && $(JAVAC) $(JAVACFLAGS) -d $$here $<
 
 %.class:        $(srcdir)/%.java
-	here=`pwd` && cd $(srcdir) && $(JAVAC) -d $$here $<
+	here=`pwd` && cd $(srcdir) && $(JAVAC) $(JAVACFLAGS) -d $$here $<
 
 $(HEADERPRE)_%.h:         $(CLASSDPRE)/%.class
 	$(JAVAH) -jni $(CLASSCPRE).`basename $< .class`
diff -Naur libdsk-1.1.10.orig/tools/Makefile.am libdsk-1.1.10/tools/Makefile.am
--- libdsk-1.1.10.orig/tools/Makefile.am	2006-04-12 10:05:09.000000000 +1200
+++ libdsk-1.1.10/tools/Makefile.am	2007-04-29 12:23:32.000000000 +1200
@@ -32,5 +32,5 @@
 CLEANFILES=*.class
 
 %.class:        $(srcdir)/%.java
-	here=`pwd` && cd $(srcdir) && $(JAVAC) -classpath $(CLASSPATH):$$here/../lib/libdsk.jar -d $$here $<
+	here=`pwd` && cd $(srcdir) && $(JAVAC) $(JAVACFLAGS) -classpath $(CLASSPATH):$$here/../lib/libdsk.jar -d $$here $<
 
