--- lib/Makefile.in.orig	2005-06-26 20:48:09.000000000 +1000
+++ lib/Makefile.in	2005-06-26 20:48:50.000000000 +1000
@@ -484,7 +484,7 @@
 
 install-exec-local:
 	$(mkinstalldirs) $(libdir)
-	if test x@LIBDSKJAR@ = x; then true; else $(INSTALL_DATA) @LIBDSKJAR@ $(libdir); fi
+	if test x@LIBDSKJAR@ = x; then true; else $(INSTALL_DATA) @LIBDSKJAR@ $(DESTDIR)/$(libdir); fi
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
