--- include/config.h.orig	2003-06-21 21:03:14.000000000 -0400
+++ include/config.h	2003-06-21 21:03:20.000000000 -0400
@@ -47 +47 @@
-/* #define QT_GRAPHICS */	/* Qt interface */
+#define QT_GRAPHICS
@@ -91 +90,0 @@
-# define USE_XPM		/* Use XPM format for images (required) */
--- src/Makefile.orig	2003-06-21 21:04:41.000000000 -0400
+++ src/Makefile	2003-06-21 21:04:58.000000000 -0400
@@ -189 +189 @@
-WINQTOBJ = qt_win.o qt_clust.o qttableview.o tile.o
+WINQTOBJ = qt_win.o qt_clust.o qttableview.o
@@ -217,2 +217,2 @@
-WINSRC = $(WINTTYSRC) $(WINX11SRC)
-WINOBJ = $(WINTTYOBJ) $(WINX11OBJ)
+WINSRC = $(WINTTYSRC) $(WINX11SRC) $(WINQTSRC)
+WINOBJ = $(WINTTYOBJ) $(WINX11OBJ) $(WINQTOBJ)
@@ -258 +258 @@
-WINLIB = $(WINTTYLIB) $(WINX11LIB)
+WINLIB = $(WINTTYLIB) $(WINX11LIB) $(WINQTLIB)
