diff -ur root-portal-0.4.11.orig/src/config/configuration.h root-portal-0.4.11/src/config/configuration.h
--- root-portal-0.4.11.orig/src/config/configuration.h	2003-10-15 16:00:20.512226040 -0400
+++ root-portal-0.4.11/src/config/configuration.h	2003-10-15 16:10:12.625211184 -0400
@@ -14,6 +14,7 @@
 #endif
 
 #include <vector>
+#include <assert.h>
 using namespace std;
 
 class ModuleModifier;
diff -ur root-portal-0.4.11.orig/src/dataset.h root-portal-0.4.11/src/dataset.h
--- root-portal-0.4.11.orig/src/dataset.h	2003-10-15 16:00:20.466233032 -0400
+++ root-portal-0.4.11/src/dataset.h	2003-10-15 16:08:13.625301920 -0400
@@ -11,6 +11,7 @@
 #include <string>
 #include <cstdlib>
 #include <cstdio>
+#include <assert.h>
 using namespace std;
 
 class BadDataType {};
diff -ur root-portal-0.4.11.orig/src/modules/filetail/filetail.h root-portal-0.4.11/src/modules/filetail/filetail.h
--- root-portal-0.4.11.orig/src/modules/filetail/filetail.h	2003-10-15 16:00:20.483230448 -0400
+++ root-portal-0.4.11/src/modules/filetail/filetail.h	2003-10-15 16:05:47.017589680 -0400
@@ -2,6 +2,7 @@
 #define FILETAIL_H
 
 #include <cstdio>
+#include <assert.h>
 using namespace std;
 
 #include "../../module.h"
diff -ur root-portal-0.4.11.orig/src/modules/process/process.h root-portal-0.4.11/src/modules/process/process.h
--- root-portal-0.4.11.orig/src/modules/process/process.h	2003-10-15 16:00:20.484230296 -0400
+++ root-portal-0.4.11/src/modules/process/process.h	2003-10-15 16:06:51.646764552 -0400
@@ -6,6 +6,7 @@
 #include <cstdlib>
 #include <cstdio>
 #include <vector>
+#include <assert.h>
 using namespace std;
 
 #include "../../module.h"
diff -ur root-portal-0.4.11.orig/src/modules/roottext/roottext.h root-portal-0.4.11/src/modules/roottext/roottext.h
--- root-portal-0.4.11.orig/src/modules/roottext/roottext.h	2003-10-15 16:00:20.486229992 -0400
+++ root-portal-0.4.11/src/modules/roottext/roottext.h	2003-10-15 16:07:23.406936272 -0400
@@ -9,6 +9,7 @@
 #include "../../rootwindow.h"
 #include "../../drawing_area.h"
 #include <vector>
+#include <assert.h>
 using namespace std;
 
 typedef unsigned long Pixel;
diff -ur root-portal-0.4.11.orig/src/path.h root-portal-0.4.11/src/path.h
--- root-portal-0.4.11.orig/src/path.h	2003-10-15 16:00:20.467232880 -0400
+++ root-portal-0.4.11/src/path.h	2003-10-15 16:11:46.954870896 -0400
@@ -3,6 +3,7 @@
 
 #include <vector>
 #include <string>
+#include <assert.h>
 using namespace std;
 
 class Path
