diff -urN ATLAS.orig/config.c ATLAS/config.c
--- ATLAS.orig/config.c	2006-01-16 19:08:16.000000000 -0800
+++ ATLAS/config.c	2006-01-16 19:27:22.000000000 -0800
@@ -712,11 +712,6 @@
                gcc, major, minor, patch);
    if (major == -1)
       iret = 0;
-   else if ( MachIsX86(mach) ) /* don't use gcc 2.96 - 3.0 */
-   {
-      if ((major == 2 && minor < 96) || (major == 3 && minor > 1))
-         iret = 1;
-   }
    else if (mach == IA64Itan || mach == IA64Itan2 || MachIsUS(mach) ||
             mach == Dec21164 || mach == Dec21264)
    {
