--- src/utils/downloader.c.old	2005-07-28 19:57:30.000000000 +0200
+++ src/utils/downloader.c	2005-11-21 17:09:08.000000000 +0100
@@ -118,6 +118,8 @@
 
 };
 
+static void gf_dm_connect(GF_DownloadSession *sess);
+
 #ifdef GPAC_HAS_SSL
 
 static void init_prng (void)
@@ -400,7 +402,6 @@
 #define GF_WAIT_REPLY_SLEEP	20
 static u32 gf_dm_session_thread(void *par)
 {
-	static void gf_dm_connect(GF_DownloadSession *sess);
 	GF_DownloadSession *sess = par;
 
 	sess->flags &= ~GF_DOWNLOAD_SESSION_THREAD_DEAD;
