--- configure.orig	2006-07-12 13:25:26.000000000 -0400
+++ configure	2006-08-01 11:32:47.000000000 -0400
@@ -143,6 +143,9 @@
 echo "  --use-mad=OPT            force MAD OPT=[no,local]"
 echo "  --use-xvid=OPT           force XVID OPT=[no,local]"
 echo "  --use-ffmpeg=OPT         force FFMPEG OPT=[no,local]"
+echo "  --use-ogg=OPT            force OGG OPT=[no,system,local]"
+echo "  --use-vorbis=OPT         force vorbis OPT=[no,system,local]"
+echo "  --use-theora=OPT         force theora OPT=[no,system,local]"
 echo ""
 echo "NOTE: The object files are build at the place where configure is launched"
 exit 1
@@ -786,6 +789,12 @@
   ;;
   --use-ffmpeg=*) has_ffmpeg=${opt#--use-ffmpeg=}
   ;;
+  --use-ogg=*) has_ogg=${opt#--use-ogg=}
+  ;;
+  --use-vorbis=*) has_vorbis=${opt#--use-vorbis=}
+  ;;
+  --use-theora=*) has_theora=${opt#--use-theora=}
+  ;;
   --use-png=*) has_png=${opt#--use-png=}
   ;;
   esac
