diff -Naurp gtkspell-2.0.11.orig/gtkspell/gtkspell.c gtkspell-2.0.11/gtkspell/gtkspell.c
--- gtkspell-2.0.11.orig/gtkspell/gtkspell.c	2006-05-26 20:59:21 -0500
+++ gtkspell-2.0.11/gtkspell/gtkspell.c	2006-05-26 21:03:04 -0500
@@ -5,12 +5,16 @@
 /* vim: set ts=4 sw=4 wm=5 : */
 
 #include <gtk/gtk.h>
-#include <libintl.h>
 #include <locale.h>
 #include "../config.h"
 #include "gtkspell.h"
 
+#ifdef ENABLE_NLS
+#include <libintl.h>
 #define _(String) dgettext (PACKAGE, String)
+#else
+#define _(String) String
+#endif
 
 #define GTKSPELL_MISSPELLED_TAG "gtkspell-misspelled"
 
