diff -r -U1 execnet-1.0.6.orig/testing/test_xspec.py execnet-1.0.6/testing/test_xspec.py
--- execnet-1.0.6.orig/testing/test_xspec.py	2010-04-21 16:19:48.000000000 +0200
+++ execnet-1.0.6/testing/test_xspec.py	2010-05-03 22:12:56.000000000 +0200
@@ -72,2 +72,8 @@
     def test_popen_nice(self):
+        import os
+        localnice = os.nice(0)
+        MAXNICE = 19
+        nice = localnice+5
+        if nice > MAXNICE:
+            nice = MAXNICE
         gw = execnet.makegateway("popen//nice=5")
@@ -81,3 +87,3 @@
         if remotenice is not None:
-            assert remotenice == 5
+            assert remotenice == nice
 
