--- gnomevfs-mount/src/Makefile.am
+++ gnomevfs-mount/src/Makefile.am
@@ -4,10 +4,9 @@
 
 gnomevfs_mount_SOURCES = gnomevfs-mount.c gnomevfs-mount.h
 
-gnomevfs_mount_LDFLAGS = @GNOME_VFS_LIBS@ @GNOME_KEYRING_LIBS@ -lfuse
-gnomevfs_mount_LDADD = -lfuse
+gnomevfs_mount_LDADD = -lfuse @GLIB_LIBS@ @GNOME_VFS_LIBS@ @GNOME_KEYRING_LIBS@
 gnomevfs_mount_CFLAGS =  @GNOME_VFS_CFLAGS@ @GNOME_KEYRING_CFLAGS@
 
-gnomevfs_umount_LDFLAGS = @GLIB_LIBS@
+gnomevfs_umount_LDADD = @GLIB_LIBS@
 gnomevfs_umount_CFLAGS = @GLIB_CFLAGS@
 
