diff --git a/xdsi b/xdsi
index 2fcc543..2ce8262 100755
--- a/xdsi
+++ b/xdsi
@@ -20,8 +20,8 @@
 ####################################################################
 # These are the only two things you may change#
 
-set Templates "/mydir/templates"
-set initial "/mydir/SLS-2009"
+set Templates "GENTOOTEMPLATE"
+set initial "./"
 
 
 ####################################################################
@@ -122,7 +122,7 @@ return 1
 }
 
 
-CheckExecutables "kpdf gnuplot xds_par xdsstat xdsviewer convert pointless ipmosflm VIEW"
+CheckExecutables "xpdf gnuplot xds_par xdsviewer convert pointless ipmosflm"
 
 wm resizable . 0 0
 
