--- oo_1.0.2_src/vcl/source/gdi/outdev3.cxx.default-fonts	2002-10-31 21:27:17.000000000 +0100
+++ oo_1.0.2_src/vcl/source/gdi/outdev3.cxx	2003-03-08 16:53:25.000000000 +0100
@@ -1041,10 +1041,10 @@ static void ImplGetMapName( const String
 
 // =======================================================================
 
-static char const aImplSubsSansUnicode[] = "andalesansui;arialunicodems;lucidaunicode";
-static char const aImplSubsSans[] = "albany;arial;helvetica;lucidasans;lucida;geneva;helmet;sansserif;nimbussansl;nimbussans";
-static char const aImplSubsSerif[] = "thorndale;timesnewroman;times;timesroman;newyork;timmons;serif;lucidaserif;lucidabright;roman;nimbusromanno9;nimbusromanno9l;bookman;itcbookman;garamond;garamondmt;palatino";
-static char const aImplSubsFixed[] = "cumberland;couriernew;courier;lucidatypewriter;lucidasanstypewriter;monaco;monospaced;nimbusmono;nimbusmonol";
+static char const aImplSubsSansUnicode[] = "andalesansui;luxisans;arialunicodems;lucidaunicode";
+static char const aImplSubsSans[] = "luxisans;albany;arial;helvetica;lucidasans;lucida;geneva;helmet;sansserif;nimbussansl;nimbussans";
+static char const aImplSubsSerif[] = "nimbusromanno9l;luxiserif;thorndale;timesnewroman;times;timesroman;newyork;timmons;serif;lucidaserif;lucidabright;roman;nimbusromanno9;bookman;itcbookman;garamond;garamondmt;palatino";
+static char const aImplSubsFixed[] = "luximono;cumberland;couriernew;courier;lucidatypewriter;lucidasanstypewriter;monaco;monospaced;nimbusmono;nimbusmonol";
 static char const aImplSubsStarSymbol[] = "starsymbol;opensymbol;starbats;wingdings;zapfdingbats;itczapfdingbats;monotypesorts;dingbats;lucidadingbats;lucidasansdingbats;webdings;symbol;standardsymbols;standardsymbolsl";
 static char const aImplSubsDingBats[] = "starsymbol;zapfdingbats;itczapfdingbats;monotypesorts;dingbats;opensymbol";
 static char const aImplSubsSymbol[] = "starsymbol;symbol;standardsymbols;standardsymbolsl;mtsymbol;opensymbol";
@@ -1123,7 +1123,7 @@ struct ImplFontNameAttr
 static ImplFontNameAttr const aImplFontNameList[] =
 {
 {   "albany",               aImplSubsSans, aImplSubsSansUnicode, NULL, NULL, aImplMSSubsArial, aImplPSSubsHelvetica, aImplHTMLSubsSansSerif,
-                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SANSSERIF | IMPL_FONT_ATTR_STANDARD | IMPL_FONT_ATTR_DEFAULT },
+                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SANSSERIF | IMPL_FONT_ATTR_STANDARD },
 {   "algerian",             aImplSubsShadow, aImplSubsImprintShadow, aImplSubsOutline, NULL, NULL, NULL, NULL,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_OUTLINE | IMPL_FONT_ATTR_SHADOW | IMPL_FONT_ATTR_DECORATIVE | IMPL_FONT_ATTR_SPECIAL | IMPL_FONT_ATTR_TITLING },
 {   "almanac",              aImplSubsStarSymbol, aImplSubsSansUnicode, NULL, NULL, NULL, NULL, NULL,
@@ -1215,7 +1215,7 @@ static ImplFontNameAttr const aImplFontN
 {   "couriernew",           aImplSubsFixed, NULL, NULL, NULL, NULL, aImplPSSubsCourier, aImplHTMLSubsMonospace,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_TYPEWRITER | IMPL_FONT_ATTR_FIXED | IMPL_FONT_ATTR_STANDARD },
 {   "cumberland",           aImplSubsFixed, NULL, NULL, NULL, aImplMSSubsCourierNew, aImplPSSubsCourier, aImplHTMLSubsMonospace,
-                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_TYPEWRITER | IMPL_FONT_ATTR_FIXED | IMPL_FONT_ATTR_STANDARD | IMPL_FONT_ATTR_DEFAULT },
+                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_TYPEWRITER | IMPL_FONT_ATTR_FIXED | IMPL_FONT_ATTR_STANDARD },
 {   "curlz",                aImplSubsComic, aImplSubsZapfChancery, aImplSubsPalaceScript, NULL, NULL, NULL, aImplHTMLSubsCursive,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_SCRIPT | IMPL_FONT_ATTR_ITALIC | IMPL_FONT_ATTR_SPECIAL },
 {   "cursive",              aImplSubsComic, aImplSubsZapfChancery, aImplSubsPalaceScript, NULL, NULL, NULL, aImplHTMLSubsCursive,
@@ -1364,6 +1364,12 @@ static ImplFontNameAttr const aImplFontN
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SANSSERIF },
 {   "lucidatypewriter",     aImplSubsFixed, NULL, NULL, NULL, NULL, NULL, aImplHTMLSubsMonospace,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_TYPEWRITER | IMPL_FONT_ATTR_FIXED },
+{   "luximono",             aImplSubsFixed, NULL, NULL, NULL, NULL, NULL, aImplHTMLSubsMonospace,
+                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_TYPEWRITER | IMPL_FONT_ATTR_FIXED | IMPL_FONT_ATTR_STANDARD | IMPL_FONT_ATTR_DEFAULT },
+{   "luxisans",             aImplSubsSans, aImplSubsSansUnicode, NULL, NULL, NULL, NULL, aImplHTMLSubsSansSerif,
+                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SANSSERIF | IMPL_FONT_ATTR_STANDARD | IMPL_FONT_ATTR_DEFAULT },
+{   "luxiserif",            aImplSubsSerif, NULL, NULL, NULL, NULL, NULL, aImplHTMLSubsSerif,
+                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SERIF | IMPL_FONT_ATTR_STANDARD },
 {   "marlett",              aImplSubsStarSymbol, aImplSubsSansUnicode, NULL, NULL, NULL, NULL, NULL,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_SPECIAL | IMPL_FONT_ATTR_SYMBOL },
 {   "mhei",                 aImplSubsSCHei, aImplSubsSCSun, aImplSubsSansUnicode, NULL, NULL, NULL, NULL,
@@ -1389,7 +1395,7 @@ static ImplFontNameAttr const aImplFontN
 {   "mono",                 aImplSubsFixed, NULL, NULL, NULL, NULL, NULL, NULL,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_FIXED },
 {   "monol",                aImplSubsFixed, NULL, NULL, NULL, NULL, NULL, NULL,
-                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_FIXED },
+                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_FIXED | IMPL_FONT_ATTR_STANDARD },
 {   "monospace",            aImplSubsFixed, NULL, NULL, NULL, NULL, NULL, NULL,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_FIXED },
 {   "monospaced",           aImplSubsFixed, NULL, NULL, NULL, NULL, NULL, aImplHTMLSubsMonospace,
@@ -1467,7 +1473,7 @@ static ImplFontNameAttr const aImplFontN
 {   "romanno9",             aImplSubsSerif, NULL, NULL, NULL, NULL, NULL, aImplHTMLSubsSerif,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SERIF },
 {   "romanno9l",            aImplSubsSerif, NULL, NULL, NULL, NULL, NULL, aImplHTMLSubsSerif,
-                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SERIF },
+                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SERIF | IMPL_FONT_ATTR_STANDARD | IMPL_FONT_ATTR_DEFAULT },
 {   "romanps",              aImplSubsFixed, NULL, NULL, NULL, NULL, NULL, aImplHTMLSubsMonospace,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_FIXED },
 {   "roundgothic",          aImplSubsKRGulim, aImplSubsKRBatang, aImplSubsSansUnicode, NULL, NULL, NULL, NULL,
@@ -1477,7 +1483,7 @@ static ImplFontNameAttr const aImplFontN
 {   "sanscondensed",        aImplSubsSansNarrow, aImplSubsSans, aImplSubsSansUnicode, NULL, NULL, NULL, NULL,
                             WEIGHT_NORMAL, WIDTH_CONDENSED, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SANSSERIF },
 {   "sansl",                aImplSubsSans, aImplSubsSansUnicode, NULL, NULL, NULL, NULL, NULL,
-                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SANSSERIF },
+                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SANSSERIF | IMPL_FONT_ATTR_STANDARD },
 {   "sanslcondensed",       aImplSubsSansNarrow, aImplSubsSans, aImplSubsSansUnicode, NULL, NULL, NULL, NULL,
                             WEIGHT_NORMAL, WIDTH_CONDENSED, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SANSSERIF },
 {   "sansserif",            aImplSubsSans, aImplSubsSansUnicode, NULL, NULL, NULL, NULL, NULL,
@@ -1535,7 +1541,7 @@ static ImplFontNameAttr const aImplFontN
 {   "tahoma",               aImplSubsSansUnicode, aImplSubsSans, NULL, NULL, NULL, NULL, NULL,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SANSSERIF },
 {   "thorndale",            aImplSubsSerif, NULL, NULL, NULL, aImplMSSubsTimesNewRoman, aImplPSSubsTimes, aImplHTMLSubsSerif,
-                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SERIF | IMPL_FONT_ATTR_STANDARD | IMPL_FONT_ATTR_DEFAULT },
+                            WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SERIF | IMPL_FONT_ATTR_STANDARD },
 {   "times",                aImplSubsSerif, NULL, NULL, NULL, NULL, NULL, aImplHTMLSubsSerif,
                             WEIGHT_NORMAL, WIDTH_NORMAL, IMPL_FONT_ATTR_NORMAL | IMPL_FONT_ATTR_SERIF | IMPL_FONT_ATTR_STANDARD },
 {   "timesnewroman",        aImplSubsSerif, NULL, NULL, NULL, NULL, aImplPSSubsTimes, aImplHTMLSubsSerif,
@@ -1812,23 +1818,26 @@ static BOOL ImplFontSubstitute( XubStrin
 
 // =======================================================================
 
-static char const aImplDefSansUnicode[] = "Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma";
-static char const aImplDefSansUI[] = "Interface User;WarpSans;Geneva;Tahoma;MS Sans Serif;Helv;Dialog;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Helmet;Interface System;Sans Serif";
-static char const aImplDefSans[] = "Albany;Arial;Helvetica;Lucida;Geneva;Helmet;SansSerif";
-static char const aImplDefSerif[] = "Thorndale;Times New Roman;Times;Lucida Serif;Lucida Bright;Timmons;New York;Serif";
-static char const aImplDefFixed[] = "Cumberland;Courier New;Courier;Lucida Sans Typewriter;Lucida Typewriter;Monaco;Monospaced";
+static char const aImplDefSansUnicode[] = "Andale Sans UI;Luxi Sans;Arial Unicode MS;Lucida Sans Unicode;Tahoma";
+static char const aImplDefSansUI[] = "Interface User;Luxi Sans;WarpSans;Geneva;Tahoma;MS Sans Serif;Helv;Dialog;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Helmet;Interface System;Sans Serif";
+static char const aImplDefSans[] = "Luxi Sans;Albany;Arial;Helvetica;Lucida;Geneva;Helmet;SansSerif";
+static char const aImplDefSerif[] = "Nimbus Roman No9 L;Luxi Serif;Thorndale;Times New Roman;Times;Lucida Serif;Lucida Bright;Timmons;New York;Serif";
+static char const aImplDefSerif_Greek[] = "Kerkis;Nimbus Roman No9 L;Luxi Serif;Thorndale;Times New Roman;Times;Lucida Serif;Lucida Bright;Timmons;New York;Serif";
+static char const aImplDefFixed[] = "Luxi Mono;Cumberland;Courier New;Courier;Lucida Sans Typewriter;Lucida Typewriter;Monaco;Monospaced";
 static char const aImplDefSymbol[] = "StarSymbol;OpenSymbol;Andale Sans UI;Arial Unicode MS;StarBats;Zapf Dingbats;WingDings;Symbol";
-static char const aImplDef_CJK_JP_Mincho[] = "HG Mincho Light J;MS Mincho;HG Mincho J;HG Mincho L;HG Mincho;Mincho;MS PMincho";
-static char const aImplDef_CJK_JP_Gothic[] = "HG Mincho Light J;MS Gothic;HG Gothic J;HG Gothic B;HG Gothic;Gothic;MS PGothic";
-static char const aImplDef_CJK_JP_UIGothic[] = "Andale Sans UI;MS Gothic;HG Gothic J;HG Gothic B;HG Gothic;Gothic;MS PGothic";
-static char const aImplDef_CJK_SC[] = "MSung Light SC;SimSun;Song;FZSongYi;FZShuSong;NSimSun";
-static char const aImplDef_CJK_TC[] = "MSung Light TC;MingLiU;Ming;PMingLiU";
-static char const aImplDef_CJK_KR_Batang[] = "HY MyeongJo Light K;Batang;Myeongjo";
-static char const aImplDef_CJK_KR_Gulim[] = "Andale Sans UI;Gulim;Roundgothic";
-static char const aImplDef_CJK_JP_Fixed[] = "HG Mincho Light J";
-static char const aImplDef_CJK_SC_Fixed[] = "MSung Light SC";
-static char const aImplDef_CJK_TC_Fixed[] = "MSung Light TC";
+static char const aImplDef_CJK_JP_Mincho[] = "Kochi Mincho;HG Mincho Light J;MS Mincho;HG Mincho J;HG Mincho L;HG Mincho;Mincho;MS PMincho";
+static char const aImplDef_CJK_JP_Gothic[] = "Kochi Gothic;HG Mincho Light J;MS Gothic;HG Gothic J;HG Gothic B;HG Gothic;Gothic;MS PGothic";
+static char const aImplDef_CJK_JP_UIGothic[] = "Andale Sans UI;Kochi Gothic;MS Gothic;HG Gothic J;HG Gothic B;HG Gothic;Gothic;MS PGothic";
+static char const aImplDef_CJK_SC[] = "AR PL SungtiL GB;MSung Light SC;SimSun;Song;FZSongYi;FZShuSong;NSimSun";
+static char const aImplDef_CJK_TC[] = "AR PL Mingti2L Big5;MSung Light TC;MingLiU;Ming;PMingLiU";
+static char const aImplDef_CJK_KR_Batang[] = "Baekmuk Batang;HY MyeongJo Light K;Batang;Myeongjo";
+static char const aImplDef_CJK_KR_Gulim[] = "Andale Sans UI;Baekmuk Gulim;Gulim;Roundgothic";
+static char const aImplDef_CJK_JP_Fixed[] = "Kochi Gothic;HG Mincho Light J";
+static char const aImplDef_CJK_SC_Fixed[] = "AR PL SungtiL GB;MSung Light SC";
+static char const aImplDef_CJK_TC_Fixed[] = "AR PL Mingti2L Big5;MSung Light TC";
 static char const aImplDef_CJK_KR_Fixed[] = "HY MyeongJo Light K";
+static char const aImplDef_CTL_AR[] = "KacstBook;Arial Unicode MS";
+static char const aImplDef_CTL_TH[] = "Norasi;Arial Unicode MS";
 
 // -----------------------------------------------------------------------
 
@@ -1956,7 +1965,10 @@ Font OutputDevice::GetDefaultFont( USHOR
         case DEFAULTFONT_SERIF:
         case DEFAULTFONT_LATIN_TEXT:
         case DEFAULTFONT_LATIN_PRESENTATION:
-            pSearch1 = aImplDefSerif;
+            if ( eLang == LANGUAGE_GREEK )
+                pSearch1 = aImplDefSerif_Greek;
+            else
+                pSearch1 = aImplDefSerif;
             aFont.SetFamily( FAMILY_ROMAN );
             break;
 
@@ -2041,7 +2053,34 @@ Font OutputDevice::GetDefaultFont( USHOR
         case DEFAULTFONT_CTL_SPREADSHEET:
         case DEFAULTFONT_CTL_HEADING:
         case DEFAULTFONT_CTL_DISPLAY:
-            pSearch1 = "Arial Unicode MS";
+            switch ( eLang )
+            {
+                case LANGUAGE_ARABIC:
+                case LANGUAGE_ARABIC_SAUDI_ARABIA:
+                case LANGUAGE_ARABIC_IRAQ:
+                case LANGUAGE_ARABIC_EGYPT:
+                case LANGUAGE_ARABIC_LIBYA:
+                case LANGUAGE_ARABIC_ALGERIA:
+                case LANGUAGE_ARABIC_MOROCCO:
+                case LANGUAGE_ARABIC_TUNISIA:
+                case LANGUAGE_ARABIC_OMAN:
+                case LANGUAGE_ARABIC_YEMEN:
+                case LANGUAGE_ARABIC_SYRIA:
+                case LANGUAGE_ARABIC_JORDAN:
+                case LANGUAGE_ARABIC_LEBANON:
+                case LANGUAGE_ARABIC_KUWAIT:
+                case LANGUAGE_ARABIC_UAE:
+                case LANGUAGE_ARABIC_BAHRAIN:
+                case LANGUAGE_ARABIC_QATAR:
+                    pSearch1 = aImplDef_CTL_AR;
+                    break;
+                case LANGUAGE_THAI:
+                    pSearch1 = aImplDef_CTL_TH;
+                    break;
+                default:
+                    pSearch1 = "Arial Unicode MS";
+                    break;
+            }
             break;
     }
 
