--- XmlSearch/src/com/sun/xmlsearch/util/ExtensibleURLStreamHandlerFactory.java.orig	2002-12-02 22:59:26.000000000 -0500
+++ XmlSearch/src/com/sun/xmlsearch/util/ExtensibleURLStreamHandlerFactory.java	2002-12-02 23:01:49.000000000 -0500
@@ -79,7 +79,7 @@
 		Class.forName(className != null
 			      ? className
 			      : "sun.net.www.protocol." + protocol + ".Handler");
-	    return (URLStreamHandler)handlerClass.newInstance();;
+	    return (URLStreamHandler)handlerClass.newInstance();
 	}
 	catch (Exception e) {
 	    System.err.println(e);

--- sj2/stardiv/app/AppletViewer.java.orig	2002-11-25 15:30:49.000000000 -0500
+++ sj2/stardiv/app/AppletViewer.java	2002-11-25 15:29:15.000000000 -0500
@@ -196,7 +196,7 @@
     public AppletViewer(int x, int y, URL doc, Hashtable atts, PrintStream statusMsgStream) {
 //  		resourceViewer = new stardiv.util.ResourceViewer();
 //  		resourceViewer.show();
-		System.err.println("#*#*#*:" + sun.awt.ScreenUpdater.updater);
+//		System.err.println("#*#*#*:" + sun.awt.ScreenUpdater.updater);
 		this.statusMsgStream = statusMsgStream;
 		this.atts = atts;
 
--- sj2/stardiv/applet/DocumentProxy.java.orig	2002-11-25 22:11:41.000000000 -0500
+++ sj2/stardiv/applet/DocumentProxy.java	2002-11-25 22:16:25.000000000 -0500
@@ -73,6 +73,8 @@
 import java.util.Hashtable;
 import java.util.Observer;
 import java.util.Observable;
+import java.util.Iterator;
+import java.io.InputStream;
 
 import java.net.URL;
 
@@ -96,6 +98,22 @@
         return documentProxy;
     }
 
+	// AppletContext. This method is new since 1.4. We insert it so as to
+	// have the project buildable
+	public void setStream( String key,InputStream stream)
+		throws java.io.IOException {
+	}
+	// AppletContext. This method is new since 1.4. We insert it so as to
+	// have the project buildable
+	public InputStream getStream( String key) {
+		return null;
+	}
+	// AppletContext. This method is new since 1.4. We insert it so as to
+	// have the project buildable
+	public Iterator getStreamKeys() {
+		return null;
+	}
+		
 
     /*
     ** interface cachable methods
