--- oo_1.0.2_src/vcl/source/glyphs/gcach_ftyp.cxx.ft-antialias-advice	2003-03-08 15:24:01.000000000 +0100
+++ oo_1.0.2_src/vcl/source/glyphs/gcach_ftyp.cxx	2003-03-08 16:17:23.000000000 +0100
@@ -900,10 +900,7 @@ bool FreetypeServerFont::GetAntialiasAdv
     if( GetFontSelData().mbNonAntialiased )
         return false;
     // TODO: also use GASP & EBDT tables
-    bool bAdviseAA = (mnLoadFlags & FT_LOAD_NO_HINTING) != 0;
-    int nHeight = GetFontSelData().mnHeight;
-    bAdviseAA |= (nHeight > 12);
-    bAdviseAA |= (nHeight < 8);
+    bool bAdviseAA = true;
     return bAdviseAA;
 }
 
