--- config_office/set_soenv.in.orig	2003-06-25 10:27:37.000000000 -0400
+++ config_office/set_soenv.in	2003-06-25 10:28:17.000000000 -0400
@@ -349,7 +349,7 @@
       $JRETOOLKITDIR  = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds."i386".$ds."client";
       $JRETHREADDIR   = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds."i386".$ds."native_threads";
    }
-   elsif ($machine_type eq "sparc")
+   elsif (($machine_type eq "sparc") or ($machine_type eq "sparc64"))
    {  print "Setting Linux Sparc specific values... ";
       $outfile        = "LinuxSparcEnv.Set"; 
       $CPU            = "S";
