diff -Nru MIDAS-2.2f.orig/MIDASa/MIDASa.c MIDAS-2.2f/MIDASa/MIDASa.c
--- MIDAS-2.2f.orig/MIDASa/MIDASa.c	2004-03-01 03:19:31.000000000 +0900
+++ MIDAS-2.2f/MIDASa/MIDASa.c	2005-08-09 08:22:22.000000000 +0900
@@ -78,7 +78,7 @@
 	(void) signal(SIGKILL, CatchSignal);
 	(void) signal(SIGTERM, CatchSignal);
 		
-	CFGPATH = dsprintf(CFGPATH, "%s/etc/MIDASa.cf", APPROOT);
+	CFGPATH = dsprintf(CFGPATH, "%s/MIDASa.cf", SYSCONFDIR);
 	ConfigXMLTree = FileToXMLTree(CFGPATH);
 		
 	if(!ConfigXMLTree)
diff -Nru MIDAS-2.2f.orig/MIDASb/MIDASb.c MIDAS-2.2f/MIDASb/MIDASb.c
--- MIDAS-2.2f.orig/MIDASb/MIDASb.c	2004-03-01 03:19:34.000000000 +0900
+++ MIDAS-2.2f/MIDASb/MIDASb.c	2005-08-09 08:22:22.000000000 +0900
@@ -81,7 +81,7 @@
 	(void) signal(SIGKILL, CatchSignal);
 	(void) signal(SIGTERM, CatchSignal);
 	
-	CFGPATH = dsprintf(CFGPATH, "%s/etc/MIDASb.cf", APPROOT);
+	CFGPATH = dsprintf(CFGPATH, "%s/MIDASb.cf", SYSCONFDIR);
 	ConfigXMLTree = FileToXMLTree(CFGPATH);
 	
 	if(!ConfigXMLTree)
diff -Nru MIDAS-2.2f.orig/MIDASc/MIDASc.c MIDAS-2.2f/MIDASc/MIDASc.c
--- MIDAS-2.2f.orig/MIDASc/MIDASc.c	2004-03-01 03:19:37.000000000 +0900
+++ MIDAS-2.2f/MIDASc/MIDASc.c	2005-08-09 08:22:22.000000000 +0900
@@ -93,7 +93,7 @@
 	(void) signal(SIGKILL, CatchSignal);
 	(void) signal(SIGTERM, CatchSignal);
 	                                
-	CFGPATH = dsprintf(CFGPATH, "%s/etc/MIDASc.cf", APPROOT);
+	CFGPATH = dsprintf(CFGPATH, "%s/MIDASc.cf", SYSCONFDIR);
 	ConfigXMLTree = FileToXMLTree(CFGPATH);
 
 	if(!ConfigXMLTree)
diff -Nru MIDAS-2.2f.orig/MIDASd/MIDASd.c MIDAS-2.2f/MIDASd/MIDASd.c
--- MIDAS-2.2f.orig/MIDASd/MIDASd.c	2004-03-01 03:37:21.000000000 +0900
+++ MIDAS-2.2f/MIDASd/MIDASd.c	2005-08-09 08:22:22.000000000 +0900
@@ -91,7 +91,7 @@
 	(void) signal(SIGKILL, CatchSignal);
 	(void) signal(SIGTERM, CatchSignal);
 	
-	CFGPATH = dsprintf(CFGPATH, "%s/etc/MIDASd.cf", APPROOT);
+	CFGPATH = dsprintf(CFGPATH, "%s/MIDASd.cf", SYSCONFDIR);
 	ConfigXMLTree = FileToXMLTree(CFGPATH);
 	if(!ConfigXMLTree)
 	{
diff -Nru MIDAS-2.2f.orig/MIDASn/MIDASn.c MIDAS-2.2f/MIDASn/MIDASn.c
--- MIDAS-2.2f.orig/MIDASn/MIDASn.c	2004-03-01 03:19:44.000000000 +0900
+++ MIDAS-2.2f/MIDASn/MIDASn.c	2005-08-09 08:22:23.000000000 +0900
@@ -88,7 +88,7 @@
 	(void) signal(SIGHUP, CatchSignal);
 	(void) signal(SIGTERM, CatchSignal);
 
-	CFGPATH = dsprintf(CFGPATH, "%s/etc/MIDASn.cf", APPROOT);		
+	CFGPATH = dsprintf(CFGPATH, "%s/MIDASn.cf", SYSCONFDIR);		
 	ConfigXMLTree = FileToXMLTree(CFGPATH);
 	
 	if(!ConfigXMLTree)
diff -Nru MIDAS-2.2f.orig/MIDASs/MIDASs.c MIDAS-2.2f/MIDASs/MIDASs.c
--- MIDAS-2.2f.orig/MIDASs/MIDASs.c	2004-03-01 03:19:48.000000000 +0900
+++ MIDAS-2.2f/MIDASs/MIDASs.c	2005-08-09 08:22:23.000000000 +0900
@@ -73,7 +73,7 @@
 	(void) signal(SIGKILL, CatchSignal);
 	(void) signal(SIGTERM, CatchSignal);
 	
-	CFGPATH = dsprintf(CFGPATH, "%s/etc/MIDASs.cf", APPROOT);
+	CFGPATH = dsprintf(CFGPATH, "%s/MIDASs.cf", SYSCONFDIR);
 	ConfigXMLTree = FileToXMLTree(CFGPATH);
 		
 	if(!ConfigXMLTree)
diff -Nru MIDAS-2.2f.orig/Makefile MIDAS-2.2f/Makefile
--- MIDAS-2.2f.orig/Makefile	2004-03-13 07:43:46.000000000 +0900
+++ MIDAS-2.2f/Makefile	2005-08-09 08:22:23.000000000 +0900
@@ -24,6 +24,7 @@
 #		value is /usr/local/MIDAS
 #
 APPROOT = -DAPPROOT=\"/usr/local/MIDAS\"
+SYSCONFDIR = -DSYSCONFDIR=\"/etc/midas\"
 
 #
 #	CC	
@@ -149,7 +150,7 @@
 #
 #	CFLAG declare
 #
-CFLAGS = -Wall $(DMALLOC) $(MIDAS_VER) $(GETLOADAVG) $(UPTIMECMD) $(NEEDATOLL) $(NEEDSTRSEP) $(HAVEMYSQL) $(HAVEPCAP) $(NEEDGETTIMEOFDAY) $(NEEDGETLOADAVG) $(APPROOT) $(NODEBUG) -o $@
+CFLAGS = -Wall $(DMALLOC) $(MIDAS_VER) $(GETLOADAVG) $(UPTIMECMD) $(NEEDATOLL) $(NEEDSTRSEP) $(HAVEMYSQL) $(HAVEPCAP) $(NEEDGETTIMEOFDAY) $(NEEDGETLOADAVG) $(APPROOT) $(SYSCONFDIR) $(NODEBUG) -o $@
 LDFLAGS = -O3 $(SOLARIS_LDFLAGS)
 
 #
diff -Nru MIDAS-2.2f.orig/Makefile.in MIDAS-2.2f/Makefile.in
--- MIDAS-2.2f.orig/Makefile.in	2004-03-13 07:37:45.000000000 +0900
+++ MIDAS-2.2f/Makefile.in	2005-08-09 08:22:23.000000000 +0900
@@ -24,6 +24,7 @@
 #		value is /usr/local/MIDAS
 #
 APPROOT = -DAPPROOT=\"@prefix@\"
+SYSCONFDIR = -DSYSCONFDIR=\"@sysconfdir@\"
 
 #
 #	CC	
@@ -149,7 +150,7 @@
 #
 #	CFLAG declare
 #
-CFLAGS = -Wall $(DMALLOC) $(MIDAS_VER) $(GETLOADAVG) $(UPTIMECMD) $(NEEDATOLL) $(NEEDSTRSEP) $(HAVEMYSQL) $(HAVEPCAP) $(NEEDGETTIMEOFDAY) $(NEEDGETLOADAVG) $(APPROOT) $(NODEBUG) @CFLAG_OUTPUT@
+CFLAGS = -Wall $(DMALLOC) $(MIDAS_VER) $(GETLOADAVG) $(UPTIMECMD) $(NEEDATOLL) $(NEEDSTRSEP) $(HAVEMYSQL) $(HAVEPCAP) $(NEEDGETTIMEOFDAY) $(NEEDGETLOADAVG) $(APPROOT) $(SYSCONFDIR) $(NODEBUG) @CFLAG_OUTPUT@
 LDFLAGS = -O3 $(SOLARIS_LDFLAGS)
 
 #
