Index: frontends/ovdb_stat.c
--- frontends/ovdb_stat.c.orig	2006-03-20 05:14:57 +0100
+++ frontends/ovdb_stat.c	2006-09-29 07:17:19 +0200
@@ -266,13 +266,17 @@
  { INT32, F(st_nobjects),      -1, -1,           "Current objects" },
  { INT32, F(st_maxnobjects),   -1, -1,           "Maximum objects" },
 #endif
+#if !(DB_VERSION_MAJOR >= 4 && DB_VERSION_MINOR >= 4)
  { INT32, F(st_nconflicts),    -1, -1,           "Lock conflicts" },
+#endif
  { INT32, F(st_nrequests),     -1, -1,           "Lock requests" },
  { INT32, F(st_nreleases),     -1, -1,           "Lock releases" },
  { DIFF32, F(st_nrequests), F(st_nreleases), F(st_ndeadlocks), "Outstanding locks" },
 #if DB_VERSION_MAJOR >= 4 || (DB_VERSION_MAJOR == 3 && DB_VERSION_MINOR > 0)
+#if !(DB_VERSION_MAJOR >= 4 && DB_VERSION_MINOR >= 4)
  { INT32, F(st_nnowaits),      -1, -1,           "Lock requests that would have waited" },
 #endif
+#endif
  { INT32, F(st_ndeadlocks),    -1, -1,           "Deadlocks" },
 #if DB_VERSION_MAJOR >= 4
  { INT32, F(st_nlocktimeouts), -1, -1,           "Lock timeouts" },
Index: scripts/innwatch.in
--- scripts/innwatch.in.orig	2006-03-20 05:14:57 +0100
+++ scripts/innwatch.in	2006-09-29 07:17:19 +0200
@@ -126,7 +126,12 @@
 NEXTSLEEP=1
 HASEXITED=false
 
-while { sleep ${NEXTSLEEP} & wait; } ; : ; do
+SLEEPTINY=4
+while { sleep ${SLEEPTINY} & wait; } ; : ; do
+    NEXTSLEEP=`expr ${NEXTSLEEP} - ${SLEEPTINY}`
+    if [ $NEXTSLEEP -gt 0 ]; then
+        continue;
+    fi
     NEXTSLEEP=${INNWATCHSLEEPTIME}
 
     ##  If news.daily is running, idle:  we don't want to change the
Index: site/Makefile
--- site/Makefile.orig	2006-03-20 05:14:57 +0100
+++ site/Makefile	2006-09-29 07:17:19 +0200
@@ -137,7 +137,6 @@
 	chmod $(FILEMODE) $@
 $D$(PATH_HISTORY):
 	touch $@
-	$(PATHBIN)/makedbz -i -o
 
 ##  Remove files that are unchanged from the release version.
 clean:
Index: storage/ovdb/ovdb.c
--- storage/ovdb/ovdb.c.orig	2006-03-20 05:14:57 +0100
+++ storage/ovdb/ovdb.c	2006-09-29 07:55:00 +0200
@@ -1526,7 +1526,9 @@
 
     OVDBenv->set_errcall(OVDBenv, OVDBerror);
     OVDBenv->set_cachesize(OVDBenv, 0, ovdb_conf.cachesize, 1);
+#if DB_VERSION_MAJOR < 4 || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR < 5)
     OVDBenv->set_lk_max(OVDBenv, ovdb_conf.maxlocks);
+#endif
 
 #if DB_VERSION_MAJOR >= 4 || (DB_VERSION_MAJOR == 3 && DB_VERSION_MINOR >= 2)
     if(ovdb_conf.txn_nosync)
