diff -Nur inn-2.4.3.orig/configure.in inn-2.4.3/configure.in
--- inn-2.4.3.orig/configure.in	2009-05-07 13:49:50.000000000 +0200
+++ inn-2.4.3/configure.in	2009-05-07 14:44:30.000000000 +0200
@@ -804,19 +818,23 @@
 
 dnl Check for Kerberos libraries for auth_krb5, and if found define KRB5_AUTH
 dnl to the relevant object file, which will enable compilation of it.
-INN_SEARCH_AUX_LIBS(krb5_parse_name, krb5, KRB5_LIB,
-    [KRB5_AUTH="auth_krb5"
-     KRB5_LIB="$KRB5_LDFLAGS $KRB5_LIB -lk5crypto -lcom_err"
-     AC_SUBST(KRB5_AUTH)
-     AC_SUBST(KRB5_INC)
-     AC_CHECK_HEADERS([et/com_err.h])], , [$LIBS -lk5crypto -lcom_err])
+if test x"$KRB5_INC" != x ; then
+    INN_SEARCH_AUX_LIBS(krb5_parse_name, krb5, KRB5_LIB,
+        [KRB5_AUTH="auth_krb5"
+         KRB5_LIB="$KRB5_LDFLAGS $KRB5_LIB -lk5crypto -lcom_err"
+         AC_SUBST(KRB5_AUTH)
+         AC_SUBST(KRB5_INC)
+         AC_CHECK_HEADERS([et/com_err.h])], , [$LIBS -lk5crypto -lcom_err])
+fi
 
 dnl Check for necessity of krb5_init_ets
 dnl OSX does not require this function
-inn_save_LIBS=$LIBS
-LIBS=$KRB5_LIB
-AC_CHECK_FUNCS(krb5_init_ets)
-LIBS=$inn_save_LIBS
+if test x"$KRB5_INC" != x ; then
+    inn_save_LIBS=$LIBS
+    LIBS=$KRB5_LIB
+    AC_CHECK_FUNCS(krb5_init_ets)
+    LIBS=$inn_save_LIBS
+fi
 
 dnl Libraries and flags for embedded Perl.  Some distributions of Linux have
 dnl Perl linked with gdbm but don't normally have gdbm installed, so on that
diff -Nur inn-2.4.3.orig/include/config.h.in inn-2.4.3/include/config.h.in
--- inn-2.4.3.orig/include/config.h.in	2006-03-20 05:14:57.000000000 +0100
+++ inn-2.4.3/include/config.h.in	2008-06-29 19:56:57.000000000 +0200
@@ -312,6 +312,9 @@
 /* Define if you have the <dlfcn.h> header file.  */
 #undef HAVE_DLFCN_H
 
+/* Define if you have the <et/com_err.h> header file.  */
+#undef HAVE_ET_COM_ERR_H
+
 /* Define if you have the <gdbm-ndbm.h> header file.  */
 #undef HAVE_GDBM_NDBM_H
 
