--- sancho-0.9.4-58-linux-gtk.sh.old	2006-11-01 12:12:25.000000000 +0100
+++ sancho-0.9.4-58-linux-gtk.sh	2006-11-01 12:13:00.000000000 +0100
@@ -365,15 +365,15 @@
 echo
 
 cd "$tmpdir"
-if test x"/home/three" != x; then
-	sanchodir=$HOME/.sancho
-	mkdir -p "$sanchodir" || {
-		echo "Could not create $sanchodir" >&2
-		exit 1
-	}
-	echo "Copying language files to $sanchodir ..." 
-	cp distrib/*.properties "$sanchodir"
-fi
+#if test x"/home/three" != x; then
+#	sanchodir=$HOME/.sancho
+#	mkdir -p "$sanchodir" || {
+#		echo "Could not create $sanchodir" >&2
+#		exit 1
+#	}
+#	echo "Copying language files to $sanchodir ..." 
+#	cp distrib/*.properties "$sanchodir"
+#fi
 
