--- Config.kmk.orig	2007-02-26 23:37:44.000000000 +0100
+++ Config.kmk	2007-02-26 23:38:56.000000000 +0100
@@ -160,8 +160,6 @@
 VBOX_WITH_USB = 1
 # Enable the ISCSI feature.
 VBOX_WITH_ISCSI = 1
-# Disable ALSA support for Linux (currently broken).
-VBOX_WITH_ALSA = 1
 # Enable the kchmviewer
 VBOX_WITH_KCHMVIEWER = 1
 # Build the testsuite.
--- configure.orig	2007-02-26 23:37:49.000000000 +0100
+++ configure	2007-02-26 23:38:40.000000000 +0100
@@ -1196,7 +1196,7 @@
 (($OSE==0)) && check_pam
 [ "$OS" != "darwin" ] && check_sdl
 (($OSE==0)) && check_sdl_ttf
-[ "$OS" != "darwin" ] && check_alsa
+[ "$OS" != "darwin" ]
 [ "$OS" != "darwin" ] && check_x
 [ "$OS" != "darwin" ] && check_xcursor
 (($XPCOM==1)) && check_qt
