--- mercury-compiler-0.13.1.orig/library/exception.m	2006-07-31 18:01:49.000000000 +1200
+++ mercury-compiler-0.13.1/library/exception.m	2006-12-30 14:46:21.000000000 +1300
@@ -1970,7 +1970,7 @@
 
 MR_proc_static_user_no_site(exception, builtin_throw, 1, 0,
     ""exception.m"", MR_DUMMY_LINE, MR_TRUE);
-MR_STATIC_USER_PROC_STATIC_PROC_LAYOUT(
+MR_EXTERN_USER_PROC_STATIC_PROC_LAYOUT(
         MR_DETISM_DET, 1, MR_LONG_LVAL_STACKVAR(1),
         MR_PREDICATE, exception, builtin_throw, 1, 0);
 MR_MAKE_USER_INTERNAL_LAYOUT(exception, builtin_throw, 1, 0, 1);
