--- mercury-compiler-0.12.2.orig/Makefile	2004-11-01 17:46:18.000000000 +1300
+++ mercury-compiler-0.12.2/Makefile	2006-09-11 20:50:59.000000000 +1200
@@ -15,7 +15,7 @@
 #
 #-----------------------------------------------------------------------------#
 
-
+PARALLEL=	$(MAKEOPTS)
 SHELL=		/bin/sh
 MMAKE=		MMAKE_DIR=`pwd`/scripts scripts/mmake 
 MMAKE_SUBDIR=	MMAKE_DIR=`pwd`/../scripts ../scripts/mmake $(PARALLEL)
@@ -69,6 +69,6 @@
 	$(MMAKE) uninstall
 
 .DEFAULT:
-	$(MMAKE) $@
+	$(MMAKE) MMAKEFLAGS=$(PARALLEL) $@
 
 #-----------------------------------------------------------------------------#
