diff -ur mercury-compiler-10.04.1.orig/boehm_gc/Makefile.direct mercury-compiler-10.04.1/boehm_gc/Makefile.direct
--- mercury-compiler-10.04.1.orig/boehm_gc/Makefile.direct	2010-02-24 20:04:32.000000000 +1300
+++ mercury-compiler-10.04.1/boehm_gc/Makefile.direct	2010-09-05 17:42:33.000000000 +1200
@@ -351,7 +351,7 @@
 		-ldl `./threadlibs`
 	./if_not_there on_sparc_sunos5_so $(MAKE) dyn_load.o
 	./if_not_there on_sparc_sunos5_so \
-	$(LINK_SHARED_OBJ) -o lib$(GC_GRADE).so $(OBJS) dyn_load.o -lc
+	$(LINK_SHARED_OBJ) -o lib$(GC_GRADE).so $(OBJS) dyn_load.o -lc $(EXTRA_LDFLAGS)
 
 # Darwin shared library version of the gc.
 lib$(GC_GRADE).dylib: $(OBJS) $(UTILS)
diff -ur mercury-compiler-10.04.1.orig/library/Mmakefile mercury-compiler-10.04.1/library/Mmakefile
--- mercury-compiler-10.04.1.orig/library/Mmakefile	2010-09-05 11:44:16.000000000 +1200
+++ mercury-compiler-10.04.1/library/Mmakefile	2010-05-20 21:29:30.000000000 +1200
@@ -268,8 +268,8 @@
 	[ -d jmercury/runtime ] || cp -r ../java/runtime jmercury
 
 install_mer_rt: $(RT_LIB_NAME).jar
-	mkdir -p $(INSTALL_LIBDIR)/lib/$(GRADE)
-	cp $(RT_LIB_NAME).jar $(INSTALL_LIBDIR)/lib/$(GRADE)
+	mkdir -p $(INSTALL_PREFIX)/lib/mercury/lib/$(GRADE)
+	cp $(RT_LIB_NAME).jar $(INSTALL_PREFIX)/lib/mercury/lib/$(GRADE)
 
 else
 
diff -ur mercury-compiler-10.04.1.orig/runtime/Mmakefile mercury-compiler-10.04.1/runtime/Mmakefile
--- mercury-compiler-10.04.1.orig/runtime/Mmakefile	2010-07-16 03:24:52.000000000 +1200
+++ mercury-compiler-10.04.1/runtime/Mmakefile	2010-09-05 17:42:33.000000000 +1200
@@ -387,7 +387,7 @@
 	$(LINK_SHARED_OBJ) $(ERROR_UNDEFINED)				\
 		-o lib$(RT_LIB_NAME).so $(PIC_OBJS)			\
 		$(SHLIB_RPATH_OPT)$(FINAL_INSTALL_MERC_GC_LIB_DIR)	\
-		$(LDFLAGS) $(LDLIBS) $(THREADLIBS)			\
+		$(ALL_LDFLAGS) $(LDLIBS) $(THREADLIBS)			\
 		$(SHARED_LIBS)
 
 # For Darwin we should pass the -install_name option.
diff -ur mercury-compiler-10.04.1.orig/trace/Mmakefile mercury-compiler-10.04.1/trace/Mmakefile
--- mercury-compiler-10.04.1.orig/trace/Mmakefile	2008-07-30 05:58:58.000000000 +1200
+++ mercury-compiler-10.04.1/trace/Mmakefile	2010-09-05 17:42:33.000000000 +1200
@@ -273,6 +273,7 @@
 	$(LINK_SHARED_OBJ) $(ERROR_UNDEFINED)				\
 		-o lib$(TRACE_LIB_NAME).so $(TRACE_PIC_OBJS)		\
 		$(RPATH_1)$(RPATH_2)					\
+		$(ALL_LDFLAGS)                                          \
 		$(TRACE_LDFLAGS) $(TRACE_LDLIBS) $(THREADLIBS)		\
 		$(SHARED_LIBS)
 
@@ -280,6 +281,7 @@
 	$(LINK_SHARED_OBJ) $(ERROR_UNDEFINED)				\
 		-o lib$(EVENTSPEC_LIB_NAME).so $(EVENTSPEC_PIC_OBJS)	\
 		$(RPATH_1)$(RPATH_2)					\
+		$(ALL_LDFLAGS)                                          \
 		$(EVENTSPEC_LDFLAGS) $(EVENTSPEC_LDLIBS) $(THREADLIBS)	\
 		$(SHARED_LIBS)
 
diff -ur mercury-compiler-10.04.1.orig/util/Mmakefile mercury-compiler-10.04.1/util/Mmakefile
--- mercury-compiler-10.04.1.orig/util/Mmakefile	2010-07-12 04:31:03.000000000 +1200
+++ mercury-compiler-10.04.1/util/Mmakefile	2010-09-05 17:42:33.000000000 +1200
@@ -41,16 +41,16 @@
 
 .c:
 	$(MGNUC) --no-mercury-stdlib-dir \
-		$(GRADEFLAGS) $(ALL_MGNUCFLAGS) -o $@ $< $(GETOPT_SRC)
+		$(GRADEFLAGS) $(ALL_MGNUCFLAGS) $(ALL_LDFLAGS) -o $@ $< $(GETOPT_SRC)
 
 mkinit: mkinit.c mkinit_common.c mkinit_common.h
 	$(MGNUC) --no-mercury-stdlib-dir \
-		$(GRADEFLAGS) $(ALL_MGNUCFLAGS) -o $@ \
+		$(GRADEFLAGS) $(ALL_MGNUCFLAGS) $(ALL_LDFLAGS) -o $@ \
 			mkinit.c mkinit_common.c $(GETOPT_SRC)
 
 mkinit_erl: mkinit_erl.c mkinit_common.c mkinit_common.h
 	$(MGNUC) --no-mercury-stdlib-dir \
-		$(GRADEFLAGS) $(ALL_MGNUCFLAGS) -o $@ \
+		$(GRADEFLAGS) $(ALL_MGNUCFLAGS) $(ALL_LDFLAGS) -o $@ \
 			mkinit_erl.c mkinit_common.c $(GETOPT_SRC)
 
 tags:
