--- mercury-compiler-10.04.1.orig/Mmakefile	2009-10-20 17:07:24.000000000 +1300
+++ mercury-compiler-10.04.1/Mmakefile	2010-10-10 10:05:55.000000000 +1300
@@ -167,6 +167,13 @@
 		library/$(deps_subdir)$(STD_LIB_NAME).dep
 	+cd deep_profiler && $(SUBDIR_MMAKE) depend
 
+.PHONY: bootstrap_depend
+ifeq ("$(BOOTSTRAP_MC)","")
+bootstrap_depend:
+else
+bootstrap_depend: depend
+endif
+
 # depend_library MUST be done before depend_compiler and depend_profiler
 
 .PHONY: depend
--- mercury-compiler-10.04.1.orig/Mmake.common.in	2009-10-10 23:44:13.000000000 +1300
+++ mercury-compiler-10.04.1/Mmake.common.in	2010-10-10 10:21:52.000000000 +1300
@@ -30,6 +30,7 @@
 
 # Specify the Mercury compiler to use for bootstrapping.
 MC		= @BOOTSTRAP_MC_COMPILER@
+BOOTSTRAP_MC    = @BOOTSTRAP_MC@
 
 # Specify the compilation model to use for compiling the compiler.
 GRADE		= @GRADE@
--- mercury-compiler-10.04.1.orig/configure.in	2010-08-23 23:04:53.000000000 +1200
+++ mercury-compiler-10.04.1/configure.in	2010-10-10 11:42:43.000000000 +1300
@@ -4426,11 +4426,11 @@
 #-----------------------------------------------------------------------------#
 
 if test "$BOOTSTRAP_MC" = ""; then
-    BOOTSTRAP_MC=mmc
+    BOOTSTRAP_MC_COMPILER=mmc
+else
+    BOOTSTRAP_MC_COMPILER="$BOOTSTRAP_MC"
 fi
-BOOTSTRAP_MC_COMPILER="$BOOTSTRAP_MC"
 BOOTSTRAP_MC_ARGS="$HAVE_BOXED_FLOATS --conf-low-tag-bits $LOW_TAG_BITS --bits-per-word $BITS_PER_WORD --bytes-per-word $BYTES_PER_WORD"
-BOOTSTRAP_MC="$BOOTSTRAP_MC_COMPILER $BOOTSTRAP_MC_ARGS"
 AC_SUBST(BOOTSTRAP_MC_COMPILER)
 AC_SUBST(BOOTSTRAP_MC_ARGS)
 AC_SUBST(BOOTSTRAP_MC)
