--- mercury-compiler-0.13.0-beta-2006-06-01.orig/configure	2006-06-01 22:35:57.000000000 +1200
+++ mercury-compiler-0.13.0-beta-2006-06-01/configure	2006-06-09 19:14:47.000000000 +1200
@@ -1609,7 +1609,7 @@
 
 
 PREFIX="`$CYGPATH $PREFIX`"
-LIBDIR="`$CYGPATH $PREFIX/lib/mercury`"
+LIBDIR="`$CYGPATH $PREFIX/lib/mercury-$PACKAGE_VERSION`"
 NONSHARED_LIB_DIR=${MERCURY_NONSHARED_LIB_DIR=$PREFIX/lib/nonshared}
 
 
--- mercury-compiler-0.13.0-beta-2006-06-01.orig/scripts/Mmake.vars.in	2006-05-23 18:46:54.000000000 +1200
+++ mercury-compiler-0.13.0-beta-2006-06-01/scripts/Mmake.vars.in	2006-06-09 19:14:47.000000000 +1200
@@ -717,18 +717,18 @@
 INSTALL_PREFIX		= @prefix@
 FINAL_INSTALL_PREFIX	= @prefix@
 INSTALL_BINDIR		= $(INSTALL_PREFIX)/bin
-INSTALL_LIBDIR		= $(INSTALL_PREFIX)/lib/mercury
-FINAL_INSTALL_LIBDIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury
-INSTALL_JAVA_LIBRARY_DIR= $(INSTALL_PREFIX)/lib/mercury/lib/java
+INSTALL_LIBDIR		= $(INSTALL_PREFIX)/lib/mercury-@PACKAGE_VERSION@
+FINAL_INSTALL_LIBDIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury-@PACKAGE_VERSION@
+INSTALL_JAVA_LIBRARY_DIR= $(INSTALL_LIBDIR)/lib/java
 INSTALL_INFO_DIR	= $(INSTALL_PREFIX)/info
-INSTALL_DVI_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_TEXT_DIR	= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PS_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PDF_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
+INSTALL_DVI_DIR		= $(INSTALL_LIBDIR)/doc
+INSTALL_TEXT_DIR	= $(INSTALL_LIBDIR)/doc
+INSTALL_PS_DIR		= $(INSTALL_LIBDIR)/doc
+INSTALL_PDF_DIR		= $(INSTALL_LIBDIR)/doc
 INSTALL_MAN_DIR		= $(INSTALL_PREFIX)/man
-INSTALL_HTML_DIR	= $(INSTALL_PREFIX)/lib/mercury/html
-INSTALL_MDB_DOC_DIR	= $(INSTALL_PREFIX)/lib/mercury/mdb
-INSTALL_ELISP_DIR	= $(INSTALL_PREFIX)/lib/mercury/elisp
+INSTALL_HTML_DIR	= $(INSTALL_LIBDIR)/html
+INSTALL_MDB_DOC_DIR	= $(INSTALL_LIBDIR)/mdb
+INSTALL_ELISP_DIR	= $(INSTALL_LIBDIR)/elisp
 INSTALL_CGI_DIR		= @CGIDIR@
 
 # You should not need to override anything below here
--- mercury-compiler-0.13.0-beta-2006-06-01.orig/scripts/mercury_config.in	2005-10-27 14:48:24.000000000 +1300
+++ mercury-compiler-0.13.0-beta-2006-06-01/scripts/mercury_config.in	2006-06-09 19:14:47.000000000 +1200
@@ -90,8 +90,8 @@
 	shift
 done
 
-input_libdir="$input_prefix/lib/mercury"
-output_libdir="$output_prefix/lib/mercury"
+input_libdir="$input_prefix/lib/mercury-@PACKAGE_VERSION@"
+output_libdir="$output_prefix/lib/mercury-@PACKAGE_VERSION@"
 
 # mercury_config regenerates itself, so we can't just execute it in place.
 case $recursive in
