--- mercury-compiler-10.04-beta.orig/library/Mmakefile	2009-11-05 14:59:45.000000000 +1300
+++ mercury-compiler-10.04-beta/library/Mmakefile	2010-04-25 19:27:49.000000000 +1200
@@ -37,6 +37,9 @@
 # Module-specific options should go in Mercury.options so they
 # can be found by `mmc --make'.
 include Mercury.options
+ifeq ("$(ARCH)","ppc")
+MGNUCFLAGS-lexer += -O0
+endif
 
 MERCURY_MAIN_MODULES = mer_std
 
