--- mercury-compiler-10.04.orig/configure	2010-07-18 20:40:26.000000000 +1200
+++ mercury-compiler-10.04/configure	2010-07-24 19:26:57.000000000 +1200
@@ -10804,7 +10804,7 @@
 # The asm_fast and reg grades also do not currently work on Linux/PPC.
 #
 case "$host" in
-    i*86*apple*darwin*|x86_64*apple*darwin*|powerpc*-linux-gnu)
+    i*86*apple*darwin*|x86_64*apple*darwin*|powerpc*-linux-gnu|sparc*-linux-gnu)
         BEST_LLDS_BASE_GRADE=none
         ;;
 esac
