diff -urN mercury-tests-0.13.1.orig/hard_coded/dir_test.exp mercury-tests-0.13.1/hard_coded/dir_test.exp
--- mercury-tests-0.13.1.orig/hard_coded/dir_test.exp	2003-07-29 03:50:45.000000000 +1200
+++ mercury-tests-0.13.1/hard_coded/dir_test.exp	2007-01-07 15:51:35.000000000 +1300
@@ -130,7 +130,6 @@
 "foo"/"bar/baz" = "foo\bar\baz".
 "foo/"/"bar/baz" = "foo\bar\baz".
 checking whether `unwritable' is readable...ok
-unwritable file found to be unwritable
 make_directory succeeded
 make_directory succeeded
 dir.make_single_directory with non-existent parent failed as expected.
diff -urN mercury-tests-0.13.1.orig/hard_coded/dir_test.exp2 mercury-tests-0.13.1/hard_coded/dir_test.exp2
--- mercury-tests-0.13.1.orig/hard_coded/dir_test.exp2	2003-07-29 03:50:45.000000000 +1200
+++ mercury-tests-0.13.1/hard_coded/dir_test.exp2	2007-01-07 15:51:35.000000000 +1300
@@ -127,7 +127,6 @@
 "foo"/"bar/baz" = "foo/bar/baz".
 "foo/"/"bar/baz" = "foo/bar/baz".
 checking whether `unwritable' is readable...ok
-unwritable file found to be unwritable
 make_directory succeeded
 make_directory succeeded
 dir.make_single_directory with non-existent parent failed as expected.
diff -urN mercury-tests-0.13.1.orig/hard_coded/dir_test.exp3 mercury-tests-0.13.1/hard_coded/dir_test.exp3
--- mercury-tests-0.13.1.orig/hard_coded/dir_test.exp3	2003-07-29 03:50:45.000000000 +1200
+++ mercury-tests-0.13.1/hard_coded/dir_test.exp3	2007-01-07 15:51:35.000000000 +1300
@@ -130,7 +130,6 @@
 "foo"/"bar/baz" = "foo/bar/baz".
 "foo/"/"bar/baz" = "foo/bar/baz".
 checking whether `unwritable' is readable...ok
-unwritable file found to be unwritable
 make_directory succeeded
 make_directory succeeded
 dir.make_single_directory with non-existent parent failed as expected.
diff -urN mercury-tests-0.13.1.orig/hard_coded/dir_test.m mercury-tests-0.13.1/hard_coded/dir_test.m
--- mercury-tests-0.13.1.orig/hard_coded/dir_test.m	2006-03-29 20:08:00.000000000 +1200
+++ mercury-tests-0.13.1/hard_coded/dir_test.m	2007-01-07 15:51:35.000000000 +1300
@@ -50,16 +50,6 @@
 	io__write(ReadResult),
 	io__nl,
 
-	io__check_file_accessibility("unwritable",
-		[read, write], WriteResult),
-	( { WriteResult = ok } ->
-		io__write_string(
-		"Error: unwritable file found to be writable\n")
-	;
-		io__write_string(
-		"unwritable file found to be unwritable\n")
-	),
-
 	{ Dir1 = "test_dir"/"d1" },
 	test0("make_directory", dir__make_directory(Dir1)),
 	% Test making a directory that already exists.
