--- mercury-tests-10.04-beta.orig/Mmake.common	2009-08-14 15:21:53.000000000 +1200
+++ mercury-tests-10.04-beta/Mmake.common	2010-02-15 19:09:51.000000000 +1300
@@ -379,12 +379,7 @@
 
 .PHONY: $(RUNTESTS_IN_SUBDIRS)
 $(RUNTESTS_IN_SUBDIRS): runtests_in_%:
-	+if (cd $* && mmake runtests) ; then \
-		true ; \
-	else \
-		echo Could not run tests in directory $* ; \
-		echo Could not run tests in directory $* >> NOMAKE_DIRS ; \
-	fi
+	+cd $* && mmake runtests
 
 realclean_subdirs:
 	+succeeded=true; \
