diff -uNr asterisk-1.8.0.ORIG//apps/app_dahdiras.c asterisk-1.8.0/apps/app_dahdiras.c
--- asterisk-1.8.0.ORIG//apps/app_dahdiras.c	2010-10-27 00:41:40.000000000 +0100
+++ asterisk-1.8.0/apps/app_dahdiras.c	2010-10-27 00:50:00.000000000 +0100
@@ -119,8 +119,6 @@
 		c = strsep(&stringp, ",");
 	}
 
-	argv[argc++] = "plugin";
-	argv[argc++] = "dahdi.so";
 	argv[argc++] = "stdin";
 
 	/* Finally launch PPP */
