--- ext/src/core/Exceptions.c	(revision 2857)
+++ ext/src/core/Exceptions.c	(working copy)
@@ -28,7 +28,11 @@
 PHPAPI zend_class_entry *onphp_ce_WrongArgumentException;
 PHPAPI zend_class_entry *onphp_ce_WrongStateException;
 
+#if (PHP_MAJOR_VERSION == 5) && (PHP_MINOR_VERSION < 2)
 #define onphp_ce_Exception zend_exception_get_default()
+#else
+#define onphp_ce_Exception zend_exception_get_default(TSRMLS_C)
+#endif
 
 PHP_MINIT_FUNCTION(Exceptions)
 {
--- ext/src/core/DB/ImaginaryDialect.c	(revision 2857)
+++ ext/src/core/DB/ImaginaryDialect.c	(working copy)
@@ -23,6 +23,12 @@
 #include "core/OSQL/DBValue.h"
 #include "core/OSQL/DialectString.h"
 
+#if (PHP_MAJOR_VERSION == 5) && (PHP_MINOR_VERSION < 2)
+#define onphp_implode(glue, words, copy) php_implode(glue, words, copy)
+#else
+#define onphp_implode(glue, words, copy) php_implode(glue, words, copy TSRMLS_CC)
+#endif
+
 PHPAPI zend_class_entry *onphp_ce_ImaginaryDialect;
 
 ONPHP_METHOD(ImaginaryDialect, me)
@@ -197,7 +203,7 @@
 		ZVAL_STRING(glue, " | ", 1);
 	} 
 	
-	php_implode(glue, words, copy);
+	onphp_implode(glue, words, copy);
 	
 	smart_str_appends(&out, "(\"");
 	
@@ -268,7 +274,7 @@
 		ZVAL_STRING(glue, " | ", 1);
 	} 
 	
-	php_implode(glue, words, copy);
+	onphp_implode(glue, words, copy);
 	
 	smart_str_appends(&out, "(RANK BY \"");
 	
