diff -ur xc-4.3.2.orig/Makefile xc-4.3.2/Makefile
--- xc-4.3.2.orig/Makefile	1996-11-24 06:56:14.000000000 +0000
+++ xc-4.3.2/Makefile	2008-10-18 09:06:12.000000000 +0000
@@ -5,21 +5,23 @@
 
 SHELL	= /bin/sh
 
+DESTDIR =
+
 #WARN 	= -Wall -ansi -pedantic -Wshadow -Wmissing-prototypes
 
 #machine = -m486
-CC	= gcc 
-GCCOPT	= -pipe -O2 -fno-strength-reduce -fomit-frame-pointer $(machine) 
+#CC	= gcc 
+GCCOPT	= -fno-strength-reduce 
 #GCCOPT	= -O -g
 CDEFS	= -D_POSIX_SOURCE=1
-CFLAGS	= $(WARN) $(CDEFS) $(GCCOPT)
+CFLAGS	+= $(WARN) $(CDEFS) $(GCCOPT)
 
-prefix	= /usr/local
+prefix	= /usr
 bindir	= $(prefix)/bin
 libdir	= $(prefix)/lib/xc 
-mandir	= /usr/man/man1
+mandir	= /usr/share/man
 catdir	= /var/catman/cat1
-manown	= -o root -g man
+manown	= -o root -g root
 binown	= -o root -g root 
 
 export CC CFLAGS manown binown
@@ -37,7 +39,7 @@
 # 	gs -sDEVICE=ljet4 -dNOPAUSE -sOutputFile=$@ $<
 
 INSTALL	= install
-LDFLAGS	= -ltermcap editline/libedit.a
+LIBS	= -lncurses editline/libedit.a
 NROFF 	= groff -mandoc
 
 # defines for use with cextract under Linux
@@ -56,7 +58,7 @@
 	@echo "Now you may execute 'make install'"
 
 xc:	$(OBJS) eline
-	$(CC) $(CFLAGS) $(OBJS) -o xc $(LDFLAGS) 
+	$(CC) $(LDFLAGS) $(OBJS) -o xc $(LIBS) 
 
 uninstall:
 	rm -f $(bindir)/xc $(mandir)/xc.1 $(catdir)/xc.1.*
@@ -64,10 +66,10 @@
 
 install: ./bin/xc ./bin/crc xc.1 crc.1
 	@echo " "
-	$(INSTALL) $(binown) -m 755 -s ./bin/xc $(bindir) 
-	$(INSTALL) $(binown) -m 755 -s ./bin/crc $(bindir) 
-	$(INSTALL) $(manown) -m 644 xc.1 $(mandir)
-	$(INSTALL) $(manown) -m 644 crc.1 $(mandir)
+	$(INSTALL) $(binown) -m 755 ./bin/xc $(DESTDIR)$(bindir) 
+	$(INSTALL) $(binown) -m 755 ./bin/crc $(DESTDIR)$(bindir) 
+	$(INSTALL) $(manown) -m 644 xc.1 $(DESTDIR)$(mandir)/man1
+	$(INSTALL) $(manown) -m 644 crc.1 $(DESTDIR)$(mandir)/man1
 	@echo " "
 	@echo " "
 	@echo "You will want to manually install 'phonelist' and"
@@ -113,7 +115,6 @@
 bin: xc crc
 	cp  ./xc ./bin
 	cp  ./crc ./bin
-	strip ./bin/xc ./bin/crc
 
 #########################################
 ##  mostly unused stuff below here ... ##
diff -ur xc-4.3.2.orig/xcsubs.c xc-4.3.2/xcsubs.c
--- xc-4.3.2.orig/xcsubs.c	1996-09-21 08:22:17.000000000 +0000
+++ xc-4.3.2/xcsubs.c	2008-10-18 09:04:32.000000000 +0000
@@ -23,7 +23,7 @@
 	 *tgetstr(), *tgoto();
 int	LI,	/* One less than screen length in termcap entry */
 	CO;	/* Screen width */
-speed_t ospeed; /* Used by termcap lib */
+//speed_t ospeed; /* Used by termcap lib */
 static char tc[LG_BUFF];	/* termcap buffer */
 static char tbuf[LG_BUFF], *CD, *CF, *CL, *CM, *CN, *AE, *SE, *SO, *ME;
 char *CE, PC; /* used by termcap -- padding character */
@@ -410,7 +410,7 @@
 {
 	struct stat statbuf;
 
-	if (stat(pathname,&statbuf) || (statbuf.st_mode & S_IFMT) != S_IFREG)
+	if (stat(pathname,&statbuf) || (statbuf.st_mode & __S_IFMT) != __S_IFREG)
 		return NIL(FILE);
 	return fopen(pathname, "r");
 }
