--- Makefile.orig	2007-10-09 16:34:53.000000000 -0700
+++ Makefile	2007-10-09 16:54:28.000000000 -0700
@@ -326,8 +326,8 @@
   else
 
 $(DUMASO): duma_config.h $(SO_OBJECTS)
-	$(CXX) -g -shared -Wl,-soname,$(DUMASO) -o $(DUMASO) $(SO_OBJECTS) -lpthread -lc
-	$(CXX) -g -shared -o $(DUMASO) $(SO_OBJECTS) -lpthread -lc
+	$(CXX) -g -shared -Wl,-soname -Wl,$(DUMASO) -o $(DUMASO) $(SO_OBJECTS) -lpthread -lc
+#	$(CXX) -g -shared -o $(DUMASO) $(SO_OBJECTS) -lpthread -lc
 
   endif
 
