diff -urN Makefile Makefile
--- Makefile	2007-01-07 14:38:21.000000000 +0100
+++ Makefile	2007-01-07 14:38:39.000000000 +0100
@@ -109,7 +109,7 @@
 prefix=/usr
 BIN_INSTALL_DIR= $(prefix)/bin
 LIB_INSTALL_DIR= $(prefix)/lib
-MAN_INSTALL_DIR= $(prefix)/man/man3
+MAN_INSTALL_DIR= $(prefix)/share/man/man3
 
 
 
@@ -123,7 +123,9 @@
 
 SO_OBJECTS = dumapp_so.o duma_so.o sem_inc_so.o print_so.o
 
-all:	libduma.a $(DUMASO) tstheap dumatest dumatestpp testoperators tstheap_so dumatestpp_so
+all:	libduma.a $(DUMASO) 
+
+check:	tstheap dumatest dumatestpp testoperators tstheap_so dumatestpp_so
 	@ echo "Testing DUMA."
 	@ echo "After the last test, it should print that the test has PASSED."
 	$(CURPATH)dumatest
@@ -134,13 +136,16 @@
 	@ echo "DUMA confidence test PASSED."
 
 install: libduma.a duma.3 $(DUMASO)
+	${INSTALL} -d ${BIN_INSTALL_DIR}
 	$(INSTALL) -m 755 duma.sh $(BIN_INSTALL_DIR)/duma
+	${INSTALL} -d ${LIB_INSTALL_DIR}
 	$(INSTALL) -m 644 libduma.a $(LIB_INSTALL_DIR)
 	$(INSTALL) -m 755 $(DUMASO) $(LIB_INSTALL_DIR)
 	- rm -f $(LIB_INSTALL_DIR)/$(DUMASO_LINK1)
 	ln -s $(DUMASO) $(LIB_INSTALL_DIR)/$(DUMASO_LINK1)
 	- rm -f $(LIB_INSTALL_DIR)/$(DUMASO_LINK2)
 	ln -s $(DUMASO) $(LIB_INSTALL_DIR)/$(DUMASO_LINK2)
+	${INSTALL} -d ${MAN_INSTALL_DIR}
 	$(INSTALL) -m 644 duma.3 $(MAN_INSTALL_DIR)/duma.3
 
 clean:
@@ -201,8 +206,8 @@
 ifneq ($(OS), Windows_NT)
 
 $(DUMASO): duma_config.h $(SO_OBJECTS)
-	$(CXX) -g -shared -Wl,-soname,$(DUMASO) -o $(DUMASO) $(SO_OBJECTS) -lpthread -lc
-	$(CXX) -g -shared -o $(DUMASO) $(SO_OBJECTS) -lpthread -lc
+	$(CXX) -g -shared -Wl,-soname,$(DUMASO) -o $(DUMASO) $(SO_OBJECTS) -lpthread
+#	$(CXX) -g -shared -o $(DUMASO) $(SO_OBJECTS) -lpthread
 
 endif
 
