diff -Naur wxPython-src-2.6.4.0-orig/contrib/src/Makefile.in wxPython-src-2.6.4.0/contrib/src/Makefile.in
--- wxPython-src-2.6.4.0-orig/contrib/src/Makefile.in	2004-09-24 10:57:49.000000000 -0600
+++ wxPython-src-2.6.4.0/contrib/src/Makefile.in	2008-06-28 22:00:31.000000000 -0600
@@ -1,6 +1,6 @@
 # $Id: wxGTK-2.6.4-mmedia.patch,v 1.1 2008/06/29 04:31:59 dirtyepic Exp $
 
-CONTRIB_SUBDIRS=fl gizmos mmedia ogl plot stc svg deprecated animate #applet
+CONTRIB_SUBDIRS=fl gizmos ogl plot stc svg deprecated animate #applet mmedia
 
 all:
 	@for d in $(CONTRIB_SUBDIRS); do (cd $$d && $(MAKE)); done
