--- wxWidgets-2.6.2/src/common/uri.cpp	2005-08-13 02:24:13.000000000 +0200
+++ wxWidgets-2.6.2-gcc41/src/common/uri.cpp	2005-12-04 22:49:30.996859750 +0100
@@ -886,8 +886,8 @@
                     op += 3;
             }
 
-            m_path = base.m_path.substr(0, bp - base.m_path.c_str()) +
-                    m_path.substr((op - m_path.c_str()), m_path.Length());
+            m_path = (wxString)base.m_path.substr(0, bp - base.m_path.c_str()) +
+                    (wxString)m_path.substr((op - m_path.c_str()), m_path.Length());
         }
     }
 
