Index: jruby-1.4.0/bin/jruby
===================================================================
--- jruby-1.4.0.orig/bin/jruby
+++ jruby-1.4.0/bin/jruby
@@ -308,12 +308,7 @@ if $cygwin; then
 fi
 
 if [ "$nailgun_client" != "" ]; then
-  if [ -f $JRUBY_HOME/tool/nailgun/ng ]; then
-    exec $JRUBY_HOME/tool/nailgun/ng org.jruby.util.NailMain $JRUBY_OPTS "$@"
-  else
-    echo "error: ng executable not found; run 'make' in ${JRUBY_HOME}/tool/nailgun"
-    exit 1
-  fi
+  exec /usr/bin/ng org.jruby.util.NailMain $JRUBY_OPTS "$@"
 else
 if [ "$VERIFY_JRUBY" != "" ]; then
   if [ "$PROFILE_ARGS" != "" ]; then
