diff -ur gentoolkit-0.2.0-r2/makedefs.mak gentoolkit-0.2.0-r3/makedefs.mak
--- gentoolkit-0.2.0-r2/makedefs.mak	2005-10-12 11:30:42.000000000 -0500
+++ gentoolkit-0.2.0-r3/makedefs.mak	2005-11-10 14:52:44.000000000 -0600
@@ -7,7 +7,7 @@
 # Override this on command line when making a release, ie 'dist'
 
 VERSION=0.2.0
-RELEASE_TAG=-r2
+RELEASE_TAG=-r3
 PYVERSION="`python-config | sed 's/-l//' | sed 's/ -lm.*//'`"
 DESTDIR=
 
diff -ur gentoolkit-0.2.0-r2/src/revdep-rebuild/revdep-rebuild gentoolkit-0.2.0-r3/src/revdep-rebuild/revdep-rebuild
--- gentoolkit-0.2.0-r2/src/revdep-rebuild/revdep-rebuild	2005-10-12 11:02:29.000000000 -0500
+++ gentoolkit-0.2.0-r3/src/revdep-rebuild/revdep-rebuild	2005-11-10 14:53:23.000000000 -0600
@@ -87,13 +87,13 @@
 		SEARCH_BROKEN=false
 		shift 2
 		;;
-	--soname-regexp=* )
+	--soname-regexp=* | --library=* )
 		SONAME="${1#*=}"
 		SONAME_GREP=grep
 		SEARCH_BROKEN=false
 		shift
 		;;
-	--soname-regexp )
+	--soname-regexp | --library )
 		SONAME="$2"
 		SONAME_GREP=grep
 		SEARCH_BROKEN=false
