--- ldso/ldso/dl-elf.c.orig	2004-08-09 19:01:02 +0000
+++ ldso/ldso/dl-elf.c	2004-08-09 19:00:40 +0000
@@ -329,6 +329,7 @@
 #endif
 		for (i = 0; i < header->nlibs; i++) {
 			if ((libent[i].flags == LIB_ELF ||
+						libent[i].flags == LIB_ELF_LIBC0 ||
 						libent[i].flags == LIB_ELF_LIBC5) &&
 					_dl_strcmp(libname, strs + libent[i].sooffset) == 0 &&
 					(tpnt1 = _dl_load_elf_shared_library(secure,
