diff -rupN dc-qt-0.1.2b/src/hubwdgt.cc dc-qt-0.1.2/src/hubwdgt.cc
--- dc-qt-0.1.2b/src/hubwdgt.cc	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/hubwdgt.cc	2009-01-26 02:35:14.000000000 -0400
@@ -30,6 +30,7 @@
 #include <qdir.h>
 #include <zlib.h>
 #include <qsplitter.h>
+#include <cstdlib>
 
 #include "hubwdgt.h"
 #include "debugdlg.h"
diff -rupN dc-qt-0.1.2b/src/icons/back.xpm dc-qt-0.1.2/src/icons/back.xpm
--- dc-qt-0.1.2b/src/icons/back.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/back.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * back_xpm[] = {
+static const char * back_xpm[] = {
 "24 25 227 2",
 "  	c None",
 ". 	c #C3E1FF",
diff -rupN dc-qt-0.1.2b/src/icons/close.xpm dc-qt-0.1.2/src/icons/close.xpm
--- dc-qt-0.1.2b/src/icons/close.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/close.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * close_xpm[] = {
+static const char * close_xpm[] = {
 "22 22 101 2",
 "  	c None",
 ". 	c #3B3B5A",
diff -rupN dc-qt-0.1.2b/src/icons/closetab.xpm dc-qt-0.1.2/src/icons/closetab.xpm
--- dc-qt-0.1.2b/src/icons/closetab.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/closetab.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * closetab_xpm[] = {
+static const char * closetab_xpm[] = {
 "14 14 105 2",
 "  	c None",
 ". 	c #DF7F7F",
diff -rupN dc-qt-0.1.2b/src/icons/common_prof.xpm dc-qt-0.1.2/src/icons/common_prof.xpm
--- dc-qt-0.1.2b/src/icons/common_prof.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/common_prof.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * common_prof_xpm[] = {
+static const char * common_prof_xpm[] = {
 "32 32 438 2",
 "  	c None",
 ". 	c #CCCFD0",
diff -rupN dc-qt-0.1.2b/src/icons/configure.xpm dc-qt-0.1.2/src/icons/configure.xpm
--- dc-qt-0.1.2b/src/icons/configure.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/configure.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * configure_xpm[] = {
+static const char * configure_xpm[] = {
 "22 22 162 2",
 "  	c None",
 ". 	c #B8B8C7",
diff -rupN dc-qt-0.1.2b/src/icons/connect.xpm dc-qt-0.1.2/src/icons/connect.xpm
--- dc-qt-0.1.2b/src/icons/connect.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/connect.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * connect_xpm[] = {
+static const char * connect_xpm[] = {
 "22 22 16 1",
 " 	c None",
 ".	c #000000",
diff -rupN dc-qt-0.1.2b/src/icons/dcqt_icon16x16.xpm dc-qt-0.1.2/src/icons/dcqt_icon16x16.xpm
--- dc-qt-0.1.2b/src/icons/dcqt_icon16x16.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/dcqt_icon16x16.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * dcqt_icon16x16_xpm[] = {
+static const char * dcqt_icon16x16_xpm[] = {
 "16 16 194 2",
 "  	c None",
 ". 	c #B9D3D1",
diff -rupN dc-qt-0.1.2b/src/icons/default_prof.xpm dc-qt-0.1.2/src/icons/default_prof.xpm
--- dc-qt-0.1.2b/src/icons/default_prof.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/default_prof.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * default_prof_xpm[] = {
+static const char * default_prof_xpm[] = {
 "32 32 324 2",
 "  	c None",
 ". 	c #7C8A8A",
diff -rupN dc-qt-0.1.2b/src/icons/disconnect.xpm dc-qt-0.1.2/src/icons/disconnect.xpm
--- dc-qt-0.1.2b/src/icons/disconnect.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/disconnect.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * disconnect_xpm[] = {
+static const char * disconnect_xpm[] = {
 "22 22 18 1",
 " 	c None",
 ".	c #FFFF00",
diff -rupN dc-qt-0.1.2b/src/icons/exit.xpm dc-qt-0.1.2/src/icons/exit.xpm
--- dc-qt-0.1.2b/src/icons/exit.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/exit.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * exit_xpm[] = {
+static const char * exit_xpm[] = {
 "22 22 299 2",
 "  	c None",
 ". 	c #80170A",
diff -rupN dc-qt-0.1.2b/src/icons/ffwd.xpm dc-qt-0.1.2/src/icons/ffwd.xpm
--- dc-qt-0.1.2b/src/icons/ffwd.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ffwd.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ffwd_xpm[] = {
+static const char * ffwd_xpm[] = {
 "22 22 198 2",
 "  	c None",
 ". 	c #8E8EC2",
diff -rupN dc-qt-0.1.2b/src/icons/folder.xpm dc-qt-0.1.2/src/icons/folder.xpm
--- dc-qt-0.1.2b/src/icons/folder.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/folder.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * folder_xpm[] = {
+static const char * folder_xpm[] = {
 "16 16 94 2",
 "  	c None",
 ". 	c #A6A6A6",
diff -rupN dc-qt-0.1.2b/src/icons/forward.xpm dc-qt-0.1.2/src/icons/forward.xpm
--- dc-qt-0.1.2b/src/icons/forward.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/forward.xpm	2009-01-26 02:39:22.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * forward_xpm[] = {
+static const char * forward_xpm[] = {
 "24 25 231 2",
 "  	c None",
 ". 	c #C3E1FF",
diff -rupN dc-qt-0.1.2b/src/icons/fullscreen.xpm dc-qt-0.1.2/src/icons/fullscreen.xpm
--- dc-qt-0.1.2b/src/icons/fullscreen.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/fullscreen.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * fullscreen_xpm[] = {
+static const char * fullscreen_xpm[] = {
 "22 22 208 2",
 "  	c None",
 ". 	c #000000",
diff -rupN dc-qt-0.1.2b/src/icons/home.xpm dc-qt-0.1.2/src/icons/home.xpm
--- dc-qt-0.1.2b/src/icons/home.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/home.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * home_xpm[] = {
+static const char * home_xpm[] = {
 "22 22 227 2",
 "  	c None",
 ". 	c #5B8EC3",
diff -rupN dc-qt-0.1.2b/src/icons/ihq_closehub.xpm dc-qt-0.1.2/src/icons/ihq_closehub.xpm
--- dc-qt-0.1.2b/src/icons/ihq_closehub.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ihq_closehub.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ihq_closehub_xpm[] = {
+static const char * ihq_closehub_xpm[] = {
 "22 22 217 2",
 "  	c None",
 ". 	c #27276E",
diff -rupN dc-qt-0.1.2b/src/icons/ihq_connect.xpm dc-qt-0.1.2/src/icons/ihq_connect.xpm
--- dc-qt-0.1.2b/src/icons/ihq_connect.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ihq_connect.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ihq_connect_xpm[] = {
+static const char * ihq_connect_xpm[] = {
 "22 22 71 1",
 " 	c None",
 ".	c #3F3F3F",
diff -rupN dc-qt-0.1.2b/src/icons/ihq_disconnect.xpm dc-qt-0.1.2/src/icons/ihq_disconnect.xpm
--- dc-qt-0.1.2b/src/icons/ihq_disconnect.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ihq_disconnect.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ihq_disconnect_xpm[] = {
+static const char * ihq_disconnect_xpm[] = {
 "22 22 67 1",
 " 	c None",
 ".	c #3F3F3F",
diff -rupN dc-qt-0.1.2b/src/icons/ihq_exit.xpm dc-qt-0.1.2/src/icons/ihq_exit.xpm
--- dc-qt-0.1.2b/src/icons/ihq_exit.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ihq_exit.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ihq_exit_xpm[] = {
+static const char * ihq_exit_xpm[] = {
 "22 22 296 2",
 "  	c None",
 ". 	c #463232",
diff -rupN dc-qt-0.1.2b/src/icons/ihq_iconmode.xpm dc-qt-0.1.2/src/icons/ihq_iconmode.xpm
--- dc-qt-0.1.2b/src/icons/ihq_iconmode.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ihq_iconmode.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ihq_iconmode_xpm[] = {
+static const char * ihq_iconmode_xpm[] = {
 "22 22 169 2",
 "  	c None",
 ". 	c #2A2E47",
diff -rupN dc-qt-0.1.2b/src/icons/ihq_newhub.xpm dc-qt-0.1.2/src/icons/ihq_newhub.xpm
--- dc-qt-0.1.2b/src/icons/ihq_newhub.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ihq_newhub.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ihq_newhub_xpm[] = {
+static const char * ihq_newhub_xpm[] = {
 "22 22 309 2",
 "  	c None",
 ". 	c #27276E",
diff -rupN dc-qt-0.1.2b/src/icons/ihq_preferences.xpm dc-qt-0.1.2/src/icons/ihq_preferences.xpm
--- dc-qt-0.1.2b/src/icons/ihq_preferences.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ihq_preferences.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ihq_preferences_xpm[] = {
+static const char * ihq_preferences_xpm[] = {
 "22 22 298 2",
 "  	c None",
 ". 	c #3232E5",
diff -rupN dc-qt-0.1.2b/src/icons/ihq_refresh.xpm dc-qt-0.1.2/src/icons/ihq_refresh.xpm
--- dc-qt-0.1.2b/src/icons/ihq_refresh.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ihq_refresh.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ihq_refresh_xpm[] = {
+static const char * ihq_refresh_xpm[] = {
 "22 22 212 2",
 "  	c None",
 ". 	c #3131B2",
diff -rupN dc-qt-0.1.2b/src/icons/ihq_search.xpm dc-qt-0.1.2/src/icons/ihq_search.xpm
--- dc-qt-0.1.2b/src/icons/ihq_search.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/ihq_search.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * ihq_search_xpm[] = {
+static const char * ihq_search_xpm[] = {
 "22 22 160 2",
 "  	c None",
 ". 	c #047DAB",
diff -rupN dc-qt-0.1.2b/src/icons/image.xpm dc-qt-0.1.2/src/icons/image.xpm
--- dc-qt-0.1.2b/src/icons/image.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/image.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * image_xpm[] = {
+static const char * image_xpm[] = {
 "16 16 96 2",
 "  	c None",
 ". 	c #919295",
diff -rupN dc-qt-0.1.2b/src/icons/miniicon.xpm dc-qt-0.1.2/src/icons/miniicon.xpm
--- dc-qt-0.1.2b/src/icons/miniicon.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/miniicon.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * miniicon_xpm[] = {
+static const char * miniicon_xpm[] = {
 "22 22 245 2",
 "  	c None",
 ". 	c #E4E1E0",
diff -rupN dc-qt-0.1.2b/src/icons/minimode.xpm dc-qt-0.1.2/src/icons/minimode.xpm
--- dc-qt-0.1.2b/src/icons/minimode.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/minimode.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * minimode_xpm[] = {
+static const char * minimode_xpm[] = {
 "75 75 539 2",
 "  	c None",
 ". 	c #E5E3E3",
diff -rupN dc-qt-0.1.2b/src/icons/minimode_alpha.xpm dc-qt-0.1.2/src/icons/minimode_alpha.xpm
--- dc-qt-0.1.2b/src/icons/minimode_alpha.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/minimode_alpha.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * minimode_alpha_xpm[] = {
+static const char * minimode_alpha_xpm[] = {
 "75 75 3 1",
 " 	c None",
 ".	c #FFFFFF",
diff -rupN dc-qt-0.1.2b/src/icons/minimode_dl.xpm dc-qt-0.1.2/src/icons/minimode_dl.xpm
--- dc-qt-0.1.2b/src/icons/minimode_dl.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/minimode_dl.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * minimode_dl_xpm[] = {
+static const char * minimode_dl_xpm[] = {
 "7 7 47 1",
 " 	c None",
 ".	c #E3DFE1",
diff -rupN dc-qt-0.1.2b/src/icons/minimode_full.xpm dc-qt-0.1.2/src/icons/minimode_full.xpm
--- dc-qt-0.1.2b/src/icons/minimode_full.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/minimode_full.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * minimode_full_xpm[] = {
+static const char * minimode_full_xpm[] = {
 "75 75 591 2",
 "  	c None",
 ". 	c #E5E3E3",
diff -rupN dc-qt-0.1.2b/src/icons/minimode_on.xpm dc-qt-0.1.2/src/icons/minimode_on.xpm
--- dc-qt-0.1.2b/src/icons/minimode_on.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/minimode_on.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * minimode_on_xpm[] = {
+static const char * minimode_on_xpm[] = {
 "20 20 188 2",
 "  	c None",
 ". 	c #D5D1D1",
diff -rupN dc-qt-0.1.2b/src/icons/minimode_ul.xpm dc-qt-0.1.2/src/icons/minimode_ul.xpm
--- dc-qt-0.1.2b/src/icons/minimode_ul.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/minimode_ul.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * minimode_ul_xpm[] = {
+static const char * minimode_ul_xpm[] = {
 "6 7 41 1",
 " 	c None",
 ".	c #DDD9DA",
diff -rupN dc-qt-0.1.2b/src/icons/mp3.xpm dc-qt-0.1.2/src/icons/mp3.xpm
--- dc-qt-0.1.2b/src/icons/mp3.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/mp3.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * mp3_xpm[] = {
+static const char * mp3_xpm[] = {
 "16 16 109 2",
 "  	c None",
 ". 	c #D5D5D5",
diff -rupN dc-qt-0.1.2b/src/icons/new.xpm dc-qt-0.1.2/src/icons/new.xpm
--- dc-qt-0.1.2b/src/icons/new.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/new.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * new_xpm[] = {
+static const char * new_xpm[] = {
 "22 22 93 2",
 "  	c None",
 ". 	c #B4BEDC",
diff -rupN dc-qt-0.1.2b/src/icons/next.xpm dc-qt-0.1.2/src/icons/next.xpm
--- dc-qt-0.1.2b/src/icons/next.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/next.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * next_xpm[] = {
+static const char * next_xpm[] = {
 "22 22 274 2",
 "  	c None",
 ". 	c #8D8DB2",
diff -rupN dc-qt-0.1.2b/src/icons/open.xpm dc-qt-0.1.2/src/icons/open.xpm
--- dc-qt-0.1.2b/src/icons/open.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/open.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * open_xpm[] = {
+static const char * open_xpm[] = {
 "22 22 75 1",
 " 	c None",
 ".	c #7F7FA3",
diff -rupN dc-qt-0.1.2b/src/icons/pause.xpm dc-qt-0.1.2/src/icons/pause.xpm
--- dc-qt-0.1.2b/src/icons/pause.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/pause.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * pause_xpm[] = {
+static const char * pause_xpm[] = {
 "22 22 190 2",
 "  	c None",
 ". 	c #9191AE",
diff -rupN dc-qt-0.1.2b/src/icons/play.xpm dc-qt-0.1.2/src/icons/play.xpm
--- dc-qt-0.1.2b/src/icons/play.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/play.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * play_xpm[] = {
+static const char * play_xpm[] = {
 "22 22 153 2",
 "  	c None",
 ". 	c #8080B8",
diff -rupN dc-qt-0.1.2b/src/icons/prev.xpm dc-qt-0.1.2/src/icons/prev.xpm
--- dc-qt-0.1.2b/src/icons/prev.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/prev.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * prev_xpm[] = {
+static const char * prev_xpm[] = {
 "22 22 274 2",
 "  	c None",
 ". 	c #63639A",
diff -rupN dc-qt-0.1.2b/src/icons/preview.xpm dc-qt-0.1.2/src/icons/preview.xpm
--- dc-qt-0.1.2b/src/icons/preview.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/preview.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * preview_xpm[] = {
+static const char * preview_xpm[] = {
 "22 22 236 2",
 "  	c None",
 ". 	c #01004C",
diff -rupN dc-qt-0.1.2b/src/icons/reshare.xpm dc-qt-0.1.2/src/icons/reshare.xpm
--- dc-qt-0.1.2b/src/icons/reshare.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/reshare.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * reshare_xpm[] = {
+static const char * reshare_xpm[] = {
 "22 22 106 2",
 "  	c None",
 ". 	c #0DA900",
diff -rupN dc-qt-0.1.2b/src/icons/rwnd.xpm dc-qt-0.1.2/src/icons/rwnd.xpm
--- dc-qt-0.1.2b/src/icons/rwnd.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/rwnd.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * rwnd_xpm[] = {
+static const char * rwnd_xpm[] = {
 "22 22 198 2",
 "  	c None",
 ". 	c #C4C4DC",
diff -rupN dc-qt-0.1.2b/src/icons/search.xpm dc-qt-0.1.2/src/icons/search.xpm
--- dc-qt-0.1.2b/src/icons/search.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/search.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * search_xpm[] = {
+static const char * search_xpm[] = {
 "22 22 191 2",
 "  	c None",
 ". 	c #A9A9B9",
diff -rupN dc-qt-0.1.2b/src/icons/stop.xpm dc-qt-0.1.2/src/icons/stop.xpm
--- dc-qt-0.1.2b/src/icons/stop.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/stop.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * stop_xpm[] = {
+static const char * stop_xpm[] = {
 "22 22 149 2",
 "  	c None",
 ". 	c #8282A6",
diff -rupN dc-qt-0.1.2b/src/icons/tar.xpm dc-qt-0.1.2/src/icons/tar.xpm
--- dc-qt-0.1.2b/src/icons/tar.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/tar.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * tar_xpm[] = {
+static const char * tar_xpm[] = {
 "16 16 62 1",
 " 	c None",
 ".	c #BCBCBC",
diff -rupN dc-qt-0.1.2b/src/icons/unknown.xpm dc-qt-0.1.2/src/icons/unknown.xpm
--- dc-qt-0.1.2b/src/icons/unknown.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/unknown.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * unknown_xpm[] = {
+static const char * unknown_xpm[] = {
 "16 16 88 1",
 " 	c None",
 ".	c #BCBCBC",
diff -rupN dc-qt-0.1.2b/src/icons/user.xpm dc-qt-0.1.2/src/icons/user.xpm
--- dc-qt-0.1.2b/src/icons/user.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/user.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * user_xpm[] = {
+static const char * user_xpm[] = {
 "16 16 87 1",
 " 	c None",
 ".	c #212121",
diff -rupN dc-qt-0.1.2b/src/icons/user_active.xpm dc-qt-0.1.2/src/icons/user_active.xpm
--- dc-qt-0.1.2b/src/icons/user_active.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/user_active.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * user_active_xpm[] = {
+static const char * user_active_xpm[] = {
 "16 16 58 1",
 " 	c None",
 ".	c #45F445",
diff -rupN dc-qt-0.1.2b/src/icons/user_op.xpm dc-qt-0.1.2/src/icons/user_op.xpm
--- dc-qt-0.1.2b/src/icons/user_op.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/user_op.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * user_op_xpm[] = {
+static const char * user_op_xpm[] = {
 "16 16 3 1",
 " 	c None",
 ".	c #008EFF",
diff -rupN dc-qt-0.1.2b/src/icons/user_passive.xpm dc-qt-0.1.2/src/icons/user_passive.xpm
--- dc-qt-0.1.2b/src/icons/user_passive.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/user_passive.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * user_passive_xpm[] = {
+static const char * user_passive_xpm[] = {
 "16 16 59 1",
 " 	c None",
 ".	c #D07749",
diff -rupN dc-qt-0.1.2b/src/icons/user_prof.xpm dc-qt-0.1.2/src/icons/user_prof.xpm
--- dc-qt-0.1.2b/src/icons/user_prof.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/user_prof.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * user_prof_xpm[] = {
+static const char * user_prof_xpm[] = {
 "32 32 247 2",
 "  	c None",
 ". 	c #C7C7C7",
diff -rupN dc-qt-0.1.2b/src/icons/video.xpm dc-qt-0.1.2/src/icons/video.xpm
--- dc-qt-0.1.2b/src/icons/video.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/video.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * video_xpm[] = {
+static const char * video_xpm[] = {
 "16 16 91 1",
 " 	c None",
 ".	c #BCBCBC",
diff -rupN dc-qt-0.1.2b/src/icons/volume.xpm dc-qt-0.1.2/src/icons/volume.xpm
--- dc-qt-0.1.2b/src/icons/volume.xpm	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/icons/volume.xpm	2009-01-26 02:39:23.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * volume_xpm[] = {
+static const char * volume_xpm[] = {
 "22 22 102 2",
 "  	c None",
 ". 	c #D6D6DD",
diff -rupN dc-qt-0.1.2b/src/mainwdgt.cc dc-qt-0.1.2/src/mainwdgt.cc
--- dc-qt-0.1.2b/src/mainwdgt.cc	2009-01-26 02:34:50.000000000 -0400
+++ dc-qt-0.1.2/src/mainwdgt.cc	2009-01-26 02:36:25.000000000 -0400
@@ -43,6 +43,8 @@
 #include <unistd.h>
 #include <qprogressdialog.h>
 
+#include <cstdlib>
+
 #include "dc_settings.h"
 #include "preferences_dlg.h"
 #include "searchwdgt.h"
--- dc-qt-0.1.2b/src/transfer_storage.cc	2009-01-26 02:49:37.000000000 -0400
+++ dc-qt-0.1.2/src/transfer_storage.cc	2009-01-26 02:49:57.000000000 -0400
@@ -1,4 +1,5 @@
 #include <iostream>
+#include <algorithm>
 
 #include "transfer_storage.h"
 
