diff -urN krename-4.0.0.orig/CMakeLists.txt krename-4.0.0/CMakeLists.txt
--- krename-4.0.0.orig/CMakeLists.txt	2009-09-24 16:35:53.000000000 +0200
+++ krename-4.0.0/CMakeLists.txt	2009-09-24 16:42:17.000000000 +0200
@@ -22,6 +22,7 @@
 macro_log_feature( TAGLIB_FOUND "taglib" "Read and write tags in audio files" "http://www.freshmeat.net/projects/taglib" TRUE "1.5.0" "")
 
 IF(TAGLIB_FOUND)
+  ADD_DEFINITIONS( -DWITH_TAGLIB )
   MESSAGE("-- The tablib library was found. KRename will be built with tablib support.")
 ELSE(TAGLIB_FOUND)
   MESSAGE("-- The tablib library was not found. KRename will not be built with tablib support.")
diff -urN krename-4.0.0.orig/src/CMakeLists.txt krename-4.0.0/src/CMakeLists.txt
--- krename-4.0.0.orig/src/CMakeLists.txt	2009-09-24 16:35:53.000000000 +0200
+++ krename-4.0.0/src/CMakeLists.txt	2009-09-24 16:37:30.000000000 +0200
@@ -47,11 +47,17 @@
   replacedialog.cpp
   scriptplugin.cpp
   systemplugin.cpp
-  taglibplugin.cpp
   tokenhelpdialog.cpp
   translitplugin.cpp
   )
 
+IF ( TAGLIB_FOUND )
+	SET(krename_SRCS 
+		${krename_SRCS}
+		taglibplugin.cpp
+	)
+ENDIF ( TAGLIB_FOUND )
+
 kde4_add_ui_files( krename_SRCS 
   customfilenamedlg.ui		
   datetimepluginwidget.ui
diff -urN krename-4.0.0.orig/src/pluginloader.cpp krename-4.0.0/src/pluginloader.cpp
--- krename-4.0.0.orig/src/pluginloader.cpp	2009-09-24 16:35:53.000000000 +0200
+++ krename-4.0.0/src/pluginloader.cpp	2009-09-24 16:41:19.000000000 +0200
@@ -27,7 +27,9 @@
 #include "permissionsplugin.h"
 #include "scriptplugin.h"
 #include "systemplugin.h"
+#ifdef WITH_TAGLIB 
 #include "taglibplugin.h"
+#endif
 #include "translitplugin.h"
 
 #include "../config-krename.h"
@@ -132,7 +134,9 @@
 #endif // _WIN32
     m_plugins.append( new ScriptPlugin( this ) );
     m_plugins.append( new SystemPlugin( this ) );
+#ifdef WITH_TAGLIB
     m_plugins.append( new TagLibPlugin( this ) );
+#endif
     m_plugins.append( new TranslitPlugin( this ) );
     //this->loadFilePlugins();
 
