diff -urN krename-4.0.1.old/CMakeLists.txt krename-4.0.1/CMakeLists.txt
--- krename-4.0.1.old/CMakeLists.txt	2009-10-20 18:51:45.000000000 +0200
+++ krename-4.0.1/CMakeLists.txt	2009-10-20 18:59:12.000000000 +0200
@@ -34,6 +34,7 @@
 macro_log_feature(EXIV2_FOUND "Exiv2" "A library to access image metadata" "http://www.exiv2.org" FALSE "")
 
 IF(EXIV2_FOUND)
+  ADD_DEFINITIONS( -DWITH_EXIV2 )
   MESSAGE("-- The exiv2 library was found. KRename will be built with exiv2 support.")
 ELSE(EXIV2_FOUND)
   MESSAGE("-- The exiv2 library was not found. KRename will not be built with exiv2 support.")
diff -urN krename-4.0.1.old/src/CMakeLists.txt krename-4.0.1/src/CMakeLists.txt
--- krename-4.0.1.old/src/CMakeLists.txt	2009-10-20 18:51:45.000000000 +0200
+++ krename-4.0.1/src/CMakeLists.txt	2009-10-20 18:55:35.000000000 +0200
@@ -25,7 +25,6 @@
   batchrenamer.cpp
   customdialog.cpp
   datetimeplugin.cpp
-  exiv2plugin.cpp
   exthistorycombo.cpp
   filedialogextwidget.cpp
   fileplugin.cpp
@@ -51,6 +50,13 @@
   translitplugin.cpp
   )
 
+IF ( EXIV2_FOUND )
+	SET(krename_SRCS
+		${krename_SRCS}
+		exiv2plugin.cpp
+	)
+ENDIF ( EXIV2_FOUND )
+
 IF ( TAGLIB_FOUND )
 	SET(krename_SRCS 
 		${krename_SRCS}
diff -urN krename-4.0.1.old/src/pluginloader.cpp krename-4.0.1/src/pluginloader.cpp
--- krename-4.0.1.old/src/pluginloader.cpp	2009-10-20 18:51:45.000000000 +0200
+++ krename-4.0.1/src/pluginloader.cpp	2009-10-20 18:58:44.000000000 +0200
@@ -21,7 +21,9 @@
 #include "plugin.h"
 
 #include "datetimeplugin.h"
+#ifdef WITH_EXIV2
 #include "exiv2plugin.h"
+#endif
 #include "fileplugin.h"
 #include "increasecounterplugin.h"
 #include "permissionsplugin.h"
@@ -125,7 +127,7 @@
 #ifndef _WIN32
     m_plugins.append( new DateTimePlugin( this ) );
 #endif // _WIN32
-#ifdef HAVE_EXIV2
+#ifdef WITH_EXIV2
     m_plugins.append( new Exiv2Plugin( this ) );
 #endif // HAVE_EXIV2
 	m_plugins.append( new IncreaseCounterPlugin( this ) );
