--- notify.c
+++ notify.c
@@ -58,7 +58,12 @@
     } else if(G_IS_FILE_ICON(icon)){
         icon_name = g_icon_to_string(icon);
     }
-    NotifyNotification *n = notify_notification_new(summary, body, icon_name, NULL);
+    NotifyNotification *n = notify_notification_new(summary, body, icon_name
+#ifdef HAVE_LIBNOTIFY_07
+    );
+#else
+    , NULL);
+#endif
     GError *err = NULL;
     if(!notify_notification_show(n, &err)){
         warn(DEBUG_ERROR, "Failed to show notification \"%s\" \"%s\": %s", summary, body, err?err->message:"<unknown error>");
