diff -up amanith/plugins/jpeg/gjpegimpexp.cpp.BAD amanith/plugins/jpeg/gjpegimpexp.cpp
--- amanith/plugins/jpeg/gjpegimpexp.cpp.BAD	2007-11-14 22:06:33.000000000 -0500
+++ amanith/plugins/jpeg/gjpegimpexp.cpp	2007-11-14 22:07:15.000000000 -0500
@@ -36,7 +36,7 @@
 #include <csetjmp>
 
 G_EXTERN_C {
-	#include "../../3rdpart/libjpeg/jpeglib.h"
+	#include <jpeglib.h>
 };
 
 /****************************************************************************
