diff -up amanith/include/amanith/support/gutilities.h.BAD amanith/include/amanith/support/gutilities.h
--- amanith/include/amanith/support/gutilities.h.BAD	2008-02-19 11:00:42.000000000 -0500
+++ amanith/include/amanith/support/gutilities.h	2008-02-19 11:00:49.000000000 -0500
@@ -34,6 +34,7 @@
 #include "amanith/geometry/gvect.h"
 #include "amanith/geometry/gquat.h"
 #include <algorithm>
+#include <cstring>
 
 /*!
 	\file gutilities.h
diff -up amanith/include/amanith/rendering/gopenglboard.h.BAD amanith/include/amanith/rendering/gopenglboard.h
--- amanith/include/amanith/rendering/gopenglboard.h.BAD	2008-02-19 10:59:06.000000000 -0500
+++ amanith/include/amanith/rendering/gopenglboard.h	2008-02-19 10:59:13.000000000 -0500
@@ -32,6 +32,7 @@
 #include "amanith/rendering/gdrawboard.h"
 #include "amanith/2d/gtesselator2d.h"
 #include "amanith/gopenglext.h"
+#include <cstring>
 #include <utility>
 
 /*!
