diff -up amanith/build/build.pro.BAD amanith/build/build.pro
--- amanith/build/build.pro.BAD	2007-11-14 17:41:00.000000000 -0500
+++ amanith/build/build.pro	2007-11-14 17:41:41.000000000 -0500
@@ -69,8 +69,9 @@ SOURCES = gelement.cpp \
 #
 #*********************************************************
 contains(DEFINES, _OPENGLEXT_PLUGIN) {
-    SOURCES += support/glew.c gopenglext.cpp
+    SOURCES += gopenglext.cpp
     CONFIG += opengl
+    LIBS += -lGLEW
 
     # Qt4 syntax
     contains(DEFINES, USE_QT4) {
diff -up amanith/include/amanith/gopenglext.h.BAD amanith/include/amanith/gopenglext.h
--- amanith/include/amanith/gopenglext.h.BAD	2007-11-14 17:42:28.000000000 -0500
+++ amanith/include/amanith/gopenglext.h	2007-11-14 17:42:52.000000000 -0500
@@ -29,7 +29,7 @@
 #ifndef GOPENGLEXT_H
 #define GOPENGLEXT_H
 
-#include "GL/glew.h"
+#include <GL/glew.h>
 // this inclusion to make glew linking working with some platforms (ex: Mac)
 #include "amanith/gelement.h"
 #ifdef G_OS_WIN
