--- src/profileengine/editor/Makefile.am 2007-12-01 00:34:47.000000000 +0200
+++ src/profileengine/editor/Makefile.am 2009-05-02 20:26:30.000000000 +0300
@@ -6,7 +6,7 @@
 libprofileeditor_la_LDFLAGS = $(all_libraries)
 libprofileeditor_la_LIBADD = $(top_builddir)/src/profileengine/lib/libprofileengine.la
 noinst_LTLIBRARIES = libprofileeditor.la
-libprofileeditor_COMPILE_FIRST = profileeditorbase.h
+libprofileeditor_COMPILE_FIRST = profileeditorbase.h addprofilewidget.h
 libprofileeditor_la_SOURCES = profileeditorbase.ui profileeditor.cpp \
 	addprofilewidget.ui
 noinst_HEADERS = profileeditor.h
