Index: lib/cppparser/lexer.cpp
===================================================================
--- lib/cppparser/lexer.cpp	(revision 926528)
+++ lib/cppparser/lexer.cpp	(working copy)
@@ -241,7 +241,7 @@
         m_startLine = false;
         int ppe = preprocessorEnabled();
 	setPreprocessorEnabled( false );
-	while( currentChar() && currentChar() != '\n' ){
+	while( !currentChar().isNull() && currentChar() != '\n' ){
             Token tok(m_source);
             nextToken( tok, true );
         }

