--- configure	2008-04-16 17:08:39.000000000 +0300
+++ /tmp/roadsend/configure	2008-06-19 01:03:00.000000000 +0300
@@ -3121,7 +3121,7 @@
 
 # can't figure out how to get this into AC_CHECK_PROG_VER below?
 # have to change it there too
-want_bigloo="3.0c, 3.0d"
+want_bigloo="3.0c, 3.0d 3.1a"
 want_curl=7.15.1
 
 # defaults
@@ -3191,7 +3191,7 @@
   ac_prog_version=`$HAVE_BIGLOO -revision 2>&1 | sed -n 's/^.*\([2-9].[0-9][a-z]\).*$/\1/p'`
   case $ac_prog_version in
     '') ac_prog_version_m="v. ?.??, bad"; ac_verc_fail=yes;;
-    3.0c | 3.0d)
+    3.0c | 3.0d | 3.1a)
        ac_prog_version_m="$ac_prog_version, ok"; ac_verc_fail=no;;
     *) ac_prog_version_m="$ac_prog_version, bad"; ac_verc_fail=yes;;
 
--- configure.in	2008-04-16 17:08:39.000000000 +0300
+++ /tmp/roadsend/configure.in	2008-06-19 01:03:16.000000000 +0300
@@ -26,7 +26,7 @@
 
 # can't figure out how to get this into AC_CHECK_PROG_VER below?
 # have to change it there too
-want_bigloo="3.0c, 3.0d"
+want_bigloo="3.0c, 3.0d, 3.1a"
 want_curl=7.15.1
 
 # defaults
@@ -49,7 +49,7 @@
 		  bigloo, 
 		  -revision,
 		  [\([2-9].[0-9][a-z]\)],
-		  [3.0c | 3.0d],
+		  [3.0c | 3.0d | 3.1a],
 		  AC_MSG_FAILURE([Bigloo version $want_bigloo required in the system path]))
 
 BIGLOO_VERSION=$ac_prog_version
