--- simgear/misc/strutils.cxx.old	2008-09-18 23:38:03.000000000 +0200
+++ simgear/misc/strutils.cxx	2008-09-18 23:38:50.000000000 +0200
@@ -22,6 +22,7 @@
 
 #include <ctype.h>
 #include "strutils.hxx"
+#include <string.h>
 
 namespace simgear {
     namespace strutils {
--- simgear/misc/tabbed_values.cxx.old	2008-09-18 23:43:06.000000000 +0200
+++ simgear/misc/tabbed_values.cxx	2008-09-18 23:43:36.000000000 +0200
@@ -23,6 +23,7 @@
 #include "tabbed_values.hxx"
 
 #include "assert.h"
+#include <stdlib.h>
 
 SGTabbedValues::SGTabbedValues(const char *line) :
 	_line(line)
--- simgear/io/sg_file.cxx.old	2008-09-18 23:45:26.000000000 +0200
+++ simgear/io/sg_file.cxx	2008-09-18 23:45:47.000000000 +0200
@@ -22,6 +22,7 @@
 
 
 #include <simgear/compiler.h>
+#include <string.h>
 
 #include STL_STRING
 
--- simgear/io/sg_serial.cxx.old	2008-09-18 23:46:32.000000000 +0200
+++ simgear/io/sg_serial.cxx	2008-09-18 23:47:17.000000000 +0200
@@ -29,6 +29,8 @@
 #include <simgear/serial/serial.hxx>
 
 #include "sg_serial.hxx"
+#include <string.h>
+#include <stdlib.h>
 
 SG_USING_STD(string);
 
--- simgear/screen/RenderTexture.cpp.old	2008-09-18 23:51:48.000000000 +0200
+++ simgear/screen/RenderTexture.cpp	2008-09-18 23:52:15.000000000 +0200
@@ -64,6 +64,7 @@
 #include <stdlib.h>
 #include <assert.h>
 #include <stdarg.h>
+#include <string.h>
 
 #ifdef _WIN32
 #pragma comment(lib, "gdi32.lib") // required for GetPixelFormat()
--- simgear/screen/shader.cpp.old	2008-09-18 23:54:02.000000000 +0200
+++ simgear/screen/shader.cpp	2008-09-18 23:54:45.000000000 +0200
@@ -26,6 +26,8 @@
 #include "shader.h"
 #include <stdio.h>
 #include <stdarg.h>
+#include <string.h>
+#include <stdlib.h>
 
 
 glVertexAttrib1dProc glVertexAttrib1dPtr = NULL;
--- simgear/screen/TestRenderTexture.cpp.old	2008-09-18 23:55:46.000000000 +0200
+++ simgear/screen/TestRenderTexture.cpp	2008-09-18 23:56:28.000000000 +0200
@@ -16,6 +16,7 @@
 
 #include <assert.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 void Reshape(int w, int h);
 
