diff -ur SimGear-0.3.9-orig/simgear/scene/model/placement.hxx SimGear-0.3.9/simgear/scene/model/placement.hxx
--- SimGear-0.3.9-orig/simgear/scene/model/placement.hxx	2006-03-31 20:29:37.000000000 -0500
+++ SimGear-0.3.9/simgear/scene/model/placement.hxx	2006-03-31 20:30:03.000000000 -0500
@@ -46,7 +46,7 @@
   SGModelPlacement ();
   virtual ~SGModelPlacement ();
 
-  virtual void SGModelPlacement::init( ssgBranch * model );
+  virtual void init( ssgBranch * model );
 
   virtual void update();
 

