diff -NaurwB audiorecorder-0.1.0-pre6-1.4.5/audioreceiver.c audiorecorder-0.1.0-pre6-1.5.0/audioreceiver.c
--- audiorecorder-0.1.0-pre6-1.4.5/audioreceiver.c	2006-08-31 19:20:10.000000000 +0200
+++ audiorecorder-0.1.0-pre6-1.5.0/audioreceiver.c	2007-01-19 17:43:51.000000000 +0100
@@ -9,7 +9,11 @@
 /* --- cAudioReceiver ------------------------------------------------------- */
 
 cAudioReceiver::cAudioReceiver(const cChannel *_channel) 
+#if VDRVERSNUM >= 10500
+:cReceiver(_channel->GetChannelID(), -2, _channel->Apid(0)), cThread()
+#else
 :cReceiver(0, -2, _channel->Apid(0)), cThread()
+#endif
 {
         channel = _channel;
         
