Nur in audiorecorder-0.1.0-pre6: audioreceiver.o.
diff -ru audiorecorder-0.1.0-pre6-orig/audiorecorder.c audiorecorder-0.1.0-pre6/audiorecorder.c
--- audiorecorder-0.1.0-pre6-orig/audiorecorder.c	2008-04-21 12:48:24.858682169 +0200
+++ audiorecorder-0.1.0-pre6/audiorecorder.c	2008-04-21 12:49:40.198680861 +0200
@@ -21,7 +21,7 @@
 #include <iostream>
 
 extern "C" {
-        #include <ffmpeg/avcodec.h>
+        #include <libavcodec/avcodec.h>
 }
 
 
Nur in audiorecorder-0.1.0-pre6: audiorecorder.o.
Nur in audiorecorder-0.1.0-pre6: browse-item.o.
Nur in audiorecorder-0.1.0-pre6: browse.o.
Nur in audiorecorder-0.1.0-pre6: cache.o.
Nur in audiorecorder-0.1.0-pre6: column.o.
diff -ru audiorecorder-0.1.0-pre6-orig/convert.h audiorecorder-0.1.0-pre6/convert.h
--- audiorecorder-0.1.0-pre6-orig/convert.h	2008-04-21 12:48:24.858682169 +0200
+++ audiorecorder-0.1.0-pre6/convert.h	2008-04-21 12:49:22.928686427 +0200
@@ -10,7 +10,7 @@
 #include "a-tools.h"
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 }
 
 
Nur in audiorecorder-0.1.0-pre6: convert.o.
Nur in audiorecorder-0.1.0-pre6: dispatcher.o.
Nur in audiorecorder-0.1.0-pre6: libvdr-audiorecorder.so.
Nur in audiorecorder-0.1.0-pre6: libvdr-audiorecorder.so.1.6.0.
Nur in audiorecorder-0.1.0-pre6: mainmenu.o.
Nur in audiorecorder-0.1.0-pre6: mpa-frame.o.
Nur in audiorecorder-0.1.0-pre6: postdata.o.
Nur in audiorecorder-0.1.0-pre6: postproc.o.
Nur in audiorecorder-0.1.0-pre6: rds.o.
Nur in audiorecorder-0.1.0-pre6: setup.o.
Nur in audiorecorder-0.1.0-pre6/tinyxml: tinyxmlerror.o.
Nur in audiorecorder-0.1.0-pre6/tinyxml: tinyxml.o.
Nur in audiorecorder-0.1.0-pre6/tinyxml: tinyxmlparser.o.
Nur in audiorecorder-0.1.0-pre6: trackinfo.o.
Nur in audiorecorder-0.1.0-pre6: xml-base.o.
Nur in audiorecorder-0.1.0-pre6: xml-cache.o.
