diff -ruN -X /home/marka/diff-ignore.txt soundtracker-0.6.7.orig/app/midi-09x.c soundtracker-0.6.7/app/midi-09x.c
--- soundtracker-0.6.7.orig/app/midi-09x.c	2003-05-17 07:33:13.000000000 -0400
+++ soundtracker-0.6.7/app/midi-09x.c	2005-09-04 10:49:38.000000000 -0400
@@ -307,7 +307,6 @@
 
       rc = snd_seq_subscribe_port( midi_handle, port_sub);
       if (rc < 0) {
-	snd_seq_port_subscribe_free(port_sub);
 	close_handle( midi_handle);
 	midi_handle = NULL;
 	g_warning( "error subscribing sequencer port (%s)\n",  snd_strerror(rc));
