--- build.xml	2002-11-06 00:10:24.000000000 -0500
+++ build.xml.patched	2004-05-14 16:22:54.196814648 -0400
@@ -1,15 +1,23 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
 <project default="jar" name="commons-cli" basedir=".">
-  
+
+  <property file="${basedir}/build.properties"/>
+
+  <path id="compile.classpath">
+  	<pathelement location="${commons-logging.jar}"/>
+  	<pathelement location="${commons-lang.jar}"/>
+  	<pathelement location="${junit.jar}"/>
+  </path>
+
   <property name="defaulttargetdir" value="target"></property> 
   <property name="classesdir" value="target/classes"></property>
   <property name="testclassesdir" value="target/test-classes"></property>
   <property name="testreportdir" value="target/test-reports"></property>
   <property name="distdir" value="dist"></property>
   <property name="javadocdir" value="target/docs/apidocs"></property>
-  <property name="final.name" value="commons-cli-1.0-beta-2-dev"></property>
-  
+  <property name="final.name" value="commons-cli-1.0"></property>
+
   <target name="init" description="o Initializes some properties">
 
     <condition property="noget">
@@ -17,21 +25,13 @@
     </condition>
   </target>
     
-  <target name="jar" description="o Create the jar" depends="get-deps">
+  <target name="jar" description="o Create the jar">
 
     <mkdir dir="${classesdir}"></mkdir>
 
-    <javac destdir="${classesdir}" deprecation="true" debug="true" optimize="false" excludes="**/package.html">
-      <src>
-        <pathelement location="src/java"></pathelement>
-      </src>
-      <classpath>
-        <fileset dir="lib">
-          <include name="*.jar"></include>
-        </fileset>
-      </classpath>
+    <javac srcdir="src/java" destdir="${classesdir}" deprecation="true" debug="true" optimize="false" excludes="**/package.html">
+    	<classpath refid="compile.classpath"/>
     </javac>
-
     
     <jar jarfile="target/${final.name}.jar" excludes="**/package.html" basedir="${classesdir}"></jar>
 
@@ -55,10 +55,8 @@
       <sysproperty key="basedir" value="src/test"></sysproperty>
       <formatter type="xml"></formatter>
       <formatter usefile="true" type="plain"></formatter>
+      <classpath refid="compile.classpath"/>
       <classpath>
-        <fileset dir="lib">
-          <include name="*.jar"></include>
-        </fileset>
         <pathelement location="target/${final.name}.jar"></pathelement>
         <pathelement path="${testclassesdir}"></pathelement>
       </classpath>
@@ -79,10 +77,8 @@
       <src>
         <pathelement location="src/test"></pathelement>
       </src>
+      <classpath refid="compile.classpath"/>
       <classpath>
-        <fileset dir="lib">
-          <include name="*.jar"></include>
-        </fileset>
         <pathelement location="target/${final.name}.jar"></pathelement>
       </classpath>
     </javac>
@@ -115,39 +111,17 @@
     <property name="title" value="CLI 1.0-beta-2-dev API"></property>
  
     <javadoc use="true" private="true" destdir="${javadocdir}" author="true" version="true" sourcepath="src/java" packagenames="org.apache.commons.cli.*">
+      
+      <classpath refid="compile.classpath"/>
       <classpath>
-        <fileset dir="lib">
-          <include name="*.jar"></include>
-        </fileset>
         <pathelement location="target/${final.name}.jar"></pathelement>
       </classpath>
     </javadoc>
  
   </target>
 
-  <target name="get-deps" unless="noget" depends="init">
-  
-    <mkdir dir="lib"></mkdir>
-    
-    <get dest="lib/commons-logging-1.0.jar" usetimestamp="true" ignoreerrors="true" src="http://www.ibiblio.org/maven/commons-logging/jars/commons-logging-1.0.jar"></get>
-    <get dest="lib/commons-lang-SNAPSHOT.jar" usetimestamp="true" ignoreerrors="true" src="http://www.ibiblio.org/maven/commons-lang/jars/commons-lang-SNAPSHOT.jar"></get>
-    <get dest="lib/junit-3.7.jar" usetimestamp="true" ignoreerrors="true" src="http://www.ibiblio.org/maven/junit/jars/junit-3.7.jar"></get>
-  
-  </target>
-
-  
-  
-  
-  
-  
-  
-  
-  
-
   <target name="install-maven">
 
-    
-
     <get dest="${user.home}/maven-install-latest.jar" usetimestamp="true" src="${maven.repo.remote}/maven/maven-install-latest.jar"></get>
     
     <unjar dest="${maven.home}" src="${user.home}/maven-install-latest.jar"></unjar>
@@ -155,4 +129,4 @@
   </target>
 
 </project>
-    
\ No newline at end of file
+    
