--- tools/source/solar/solar.c	2003-04-15 17:55:43.000000000 +0000
+++ tools/source/solar/solar.c	2003-07-31 23:53:28.151632856 +0000
@@ -577,7 +577,7 @@
 
   {
 	char* p = NULL;
-	InfoMemoryAccess( p );
+	/*InfoMemoryAccess( p );*/
 	p = (char*)&p;
 	InfoMemoryAccess( p );
 	InfoMemoryTypeAccess( t_short );
--- tools/workben/solar.c	2003-04-15 17:56:37.000000000 +0000
+++ tools/workben/solar.c	2003-08-24 14:04:39.380218904 +0000
@@ -439,7 +439,7 @@
 
   {
 	char* p = NULL;
-	CheckCharAccess( p );
+	/*CheckCharAccess( p );*/
 	p = (char*)&p;
 	CheckCharAccess( p );
   }
