diff -Nru dansguardian-2.9.8.2.orig/configure.ac dansguardian-2.9.8.2/configure.ac
--- dansguardian-2.9.8.2.orig/configure.ac	2006-12-20 11:59:42.000000000 +0200
+++ dansguardian-2.9.8.2/configure.ac	2007-02-02 00:54:36.000000000 +0200
@@ -104,6 +104,7 @@
 AC_CHECK_FUNCS([dup2 gettimeofday memset select])
 AC_CHECK_FUNCS([strerror strstr strtol])
 AC_CHECK_FUNCS([setuid setgid umask seteuid setreuid setlocale strdup])
+AC_SEARCH_LIBS([floor], [m])
 AC_SEARCH_LIBS([gethostbyname], [nsl])
 AC_SEARCH_LIBS([socket], [socket], [], [
 	AC_CHECK_LIB(
diff -Nru dansguardian-2.9.8.2.orig/src/ConnectionHandler.cpp dansguardian-2.9.8.2/src/ConnectionHandler.cpp
--- dansguardian-2.9.8.2.orig/src/ConnectionHandler.cpp	2007-01-19 11:59:04.000000000 +0200
+++ dansguardian-2.9.8.2/src/ConnectionHandler.cpp	2007-02-02 00:54:37.000000000 +0200
@@ -32,6 +32,7 @@
 #include <syslog.h>
 #include <cerrno>
 #include <cstdio>
+#include <ctime>
 #include <algorithm>
 #include <netdb.h>
 #include <cstdlib>
diff -Nru dansguardian-2.9.8.2.orig/src/dansguardian.cpp dansguardian-2.9.8.2/src/dansguardian.cpp
--- dansguardian-2.9.8.2.orig/src/dansguardian.cpp	2006-12-19 12:47:44.000000000 +0200
+++ dansguardian-2.9.8.2/src/dansguardian.cpp	2007-02-02 00:54:37.000000000 +0200
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <cstdio>
+#include <ctime>
 #include <unistd.h>
 #include <cerrno>
 #include <syslog.h>
diff -Nru dansguardian-2.9.8.2.orig/src/DynamicURLList.cpp dansguardian-2.9.8.2/src/DynamicURLList.cpp
--- dansguardian-2.9.8.2.orig/src/DynamicURLList.cpp	2006-12-19 12:26:47.000000000 +0200
+++ dansguardian-2.9.8.2/src/DynamicURLList.cpp	2007-02-02 00:54:37.000000000 +0200
@@ -25,6 +25,7 @@
 
 #include <syslog.h>
 #include <algorithm>
+#include <ctime>
 #include <sys/stat.h>
 #include <sys/time.h>
 
diff -Nru dansguardian-2.9.8.2.orig/src/FatController.cpp dansguardian-2.9.8.2/src/FatController.cpp
--- dansguardian-2.9.8.2.orig/src/FatController.cpp	2007-01-15 16:06:50.000000000 +0200
+++ dansguardian-2.9.8.2/src/FatController.cpp	2007-02-02 00:54:37.000000000 +0200
@@ -33,6 +33,7 @@
 
 #include <syslog.h>
 #include <csignal>
+#include <ctime>
 #include <sys/stat.h>
 #include <pwd.h>
 #include <cerrno>
diff -Nru dansguardian-2.9.8.2.orig/src/ListContainer.cpp dansguardian-2.9.8.2/src/ListContainer.cpp
--- dansguardian-2.9.8.2.orig/src/ListContainer.cpp	2007-01-15 13:44:31.000000000 +0200
+++ dansguardian-2.9.8.2/src/ListContainer.cpp	2007-02-02 00:54:37.000000000 +0200
@@ -29,6 +29,7 @@
 #include "RegExp.hpp"
 #include <cstdlib>
 #include <cstdio>
+#include <ctime>
 #include <unistd.h>
 #include "String.hpp"
 #include <iostream>
diff -Nru dansguardian-2.9.8.2.orig/src/ListManager.cpp dansguardian-2.9.8.2/src/ListManager.cpp
--- dansguardian-2.9.8.2.orig/src/ListManager.cpp	2006-01-04 12:19:59.000000000 +0200
+++ dansguardian-2.9.8.2/src/ListManager.cpp	2007-02-02 00:54:37.000000000 +0200
@@ -25,6 +25,7 @@
 
 #include <syslog.h>
 #include <sys/stat.h>
+#include <ctime>
 
 
 // GLOBALS
