diff -Nru dansguardian-2.9.8.5.orig/configure.ac dansguardian-2.9.8.5/configure.ac
--- dansguardian-2.9.8.5.orig/configure.ac	2007-03-19 19:07:55.000000000 +0200
+++ dansguardian-2.9.8.5/configure.ac	2007-04-04 15:11:45.000000000 +0300
@@ -104,6 +104,7 @@
 AC_CHECK_FUNCS([dup2 gettimeofday memset select])
 AC_CHECK_FUNCS([strerror strstr strtol])
 AC_CHECK_FUNCS([setuid setgid umask seteuid setreuid setlocale strdup])
+AC_SEARCH_LIBS([floor], [m])
 AC_SEARCH_LIBS([gethostbyname], [nsl])
 AC_SEARCH_LIBS([socket], [socket], [], [
 	AC_CHECK_LIB(
diff -Nru dansguardian-2.9.8.5.orig/src/ConnectionHandler.cpp dansguardian-2.9.8.5/src/ConnectionHandler.cpp
--- dansguardian-2.9.8.5.orig/src/ConnectionHandler.cpp	2007-03-15 17:56:44.000000000 +0200
+++ dansguardian-2.9.8.5/src/ConnectionHandler.cpp	2007-04-04 15:11:45.000000000 +0300
@@ -32,6 +32,7 @@
 #include <syslog.h>
 #include <cerrno>
 #include <cstdio>
+#include <ctime>
 #include <algorithm>
 #include <netdb.h>
 #include <cstdlib>
diff -Nru dansguardian-2.9.8.5.orig/src/dansguardian.cpp dansguardian-2.9.8.5/src/dansguardian.cpp
--- dansguardian-2.9.8.5.orig/src/dansguardian.cpp	2007-02-02 12:16:16.000000000 +0200
+++ dansguardian-2.9.8.5/src/dansguardian.cpp	2007-04-04 15:11:45.000000000 +0300
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <cstdio>
+#include <ctime>
 #include <unistd.h>
 #include <cerrno>
 #include <syslog.h>
diff -Nru dansguardian-2.9.8.5.orig/src/DynamicURLList.cpp dansguardian-2.9.8.5/src/DynamicURLList.cpp
--- dansguardian-2.9.8.5.orig/src/DynamicURLList.cpp	2006-12-19 12:26:47.000000000 +0200
+++ dansguardian-2.9.8.5/src/DynamicURLList.cpp	2007-04-04 15:11:45.000000000 +0300
@@ -25,6 +25,7 @@
 
 #include <syslog.h>
 #include <algorithm>
+#include <ctime>
 #include <sys/stat.h>
 #include <sys/time.h>
 
diff -Nru dansguardian-2.9.8.5.orig/src/FatController.cpp dansguardian-2.9.8.5/src/FatController.cpp
--- dansguardian-2.9.8.5.orig/src/FatController.cpp	2007-02-02 12:16:16.000000000 +0200
+++ dansguardian-2.9.8.5/src/FatController.cpp	2007-04-04 15:11:45.000000000 +0300
@@ -33,6 +33,7 @@
 
 #include <syslog.h>
 #include <csignal>
+#include <ctime>
 #include <sys/stat.h>
 #include <pwd.h>
 #include <cerrno>
diff -Nru dansguardian-2.9.8.5.orig/src/ListContainer.cpp dansguardian-2.9.8.5/src/ListContainer.cpp
--- dansguardian-2.9.8.5.orig/src/ListContainer.cpp	2007-02-06 18:16:47.000000000 +0200
+++ dansguardian-2.9.8.5/src/ListContainer.cpp	2007-04-04 15:11:45.000000000 +0300
@@ -29,6 +29,7 @@
 #include "RegExp.hpp"
 #include <cstdlib>
 #include <cstdio>
+#include <ctime>
 #include <unistd.h>
 #include "String.hpp"
 #include <iostream>
diff -Nru dansguardian-2.9.8.5.orig/src/ListManager.cpp dansguardian-2.9.8.5/src/ListManager.cpp
--- dansguardian-2.9.8.5.orig/src/ListManager.cpp	2007-02-02 12:16:16.000000000 +0200
+++ dansguardian-2.9.8.5/src/ListManager.cpp	2007-04-04 15:11:45.000000000 +0300
@@ -25,6 +25,7 @@
 
 #include <syslog.h>
 #include <sys/stat.h>
+#include <ctime>
 
 
 // GLOBALS
