diff -Nru dansguardian-2.9.8.1.orig/configure.ac dansguardian-2.9.8.1/configure.ac
--- dansguardian-2.9.8.1.orig/configure.ac	2006-11-17 13:59:00.000000000 +0200
+++ dansguardian-2.9.8.1/configure.ac	2006-12-19 19:59:56.000000000 +0200
@@ -104,6 +104,7 @@
 AC_CHECK_FUNCS([dup2 gettimeofday memset select])
 AC_CHECK_FUNCS([strerror strstr strtol])
 AC_CHECK_FUNCS([setuid setgid umask seteuid setreuid setlocale strdup])
+AC_SEARCH_LIBS([floor], [m])
 AC_SEARCH_LIBS([gethostbyname], [nsl])
 AC_SEARCH_LIBS([socket], [socket], [], [
 	AC_CHECK_LIB(
diff -Nru dansguardian-2.9.8.1.orig/src/ConnectionHandler.cpp dansguardian-2.9.8.1/src/ConnectionHandler.cpp
--- dansguardian-2.9.8.1.orig/src/ConnectionHandler.cpp	2006-11-21 18:52:31.000000000 +0200
+++ dansguardian-2.9.8.1/src/ConnectionHandler.cpp	2006-12-19 19:59:56.000000000 +0200
@@ -32,6 +32,7 @@
 #include <syslog.h>
 #include <cerrno>
 #include <cstdio>
+#include <ctime>
 #include <algorithm>
 #include <netdb.h>
 #include <cstdlib>
diff -Nru dansguardian-2.9.8.1.orig/src/dansguardian.cpp dansguardian-2.9.8.1/src/dansguardian.cpp
--- dansguardian-2.9.8.1.orig/src/dansguardian.cpp	2006-11-29 17:03:55.000000000 +0200
+++ dansguardian-2.9.8.1/src/dansguardian.cpp	2006-12-19 19:59:56.000000000 +0200
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <cstdio>
+#include <ctime>
 #include <unistd.h>
 #include <cerrno>
 #include <syslog.h>
diff -Nru dansguardian-2.9.8.1.orig/src/DynamicURLList.cpp dansguardian-2.9.8.1/src/DynamicURLList.cpp
--- dansguardian-2.9.8.1.orig/src/DynamicURLList.cpp	2006-08-03 13:57:46.000000000 +0300
+++ dansguardian-2.9.8.1/src/DynamicURLList.cpp	2006-12-19 19:59:56.000000000 +0200
@@ -25,6 +25,7 @@
 
 #include <syslog.h>
 #include <algorithm>
+#include <ctime>
 #include <sys/stat.h>
 #include <sys/time.h>
 
diff -Nru dansguardian-2.9.8.1.orig/src/FatController.cpp dansguardian-2.9.8.1/src/FatController.cpp
--- dansguardian-2.9.8.1.orig/src/FatController.cpp	2006-11-27 14:08:55.000000000 +0200
+++ dansguardian-2.9.8.1/src/FatController.cpp	2006-12-19 19:59:56.000000000 +0200
@@ -33,6 +33,7 @@
 
 #include <syslog.h>
 #include <csignal>
+#include <ctime>
 #include <sys/stat.h>
 #include <pwd.h>
 #include <cerrno>
diff -Nru dansguardian-2.9.8.1.orig/src/ListContainer.cpp dansguardian-2.9.8.1/src/ListContainer.cpp
--- dansguardian-2.9.8.1.orig/src/ListContainer.cpp	2006-11-29 18:01:24.000000000 +0200
+++ dansguardian-2.9.8.1/src/ListContainer.cpp	2006-12-19 19:59:56.000000000 +0200
@@ -29,6 +29,7 @@
 #include "RegExp.hpp"
 #include <cstdlib>
 #include <cstdio>
+#include <ctime>
 #include <unistd.h>
 #include "String.hpp"
 #include <iostream>
diff -Nru dansguardian-2.9.8.1.orig/src/ListManager.cpp dansguardian-2.9.8.1/src/ListManager.cpp
--- dansguardian-2.9.8.1.orig/src/ListManager.cpp	2006-01-04 12:19:59.000000000 +0200
+++ dansguardian-2.9.8.1/src/ListManager.cpp	2006-12-19 19:59:56.000000000 +0200
@@ -25,6 +25,7 @@
 
 #include <syslog.h>
 #include <sys/stat.h>
+#include <ctime>
 
 
 // GLOBALS
