--- libtwnotification/main.cpp
+++ libtwnotification/main.cpp
@@ -2,6 +2,10 @@
 #include <iostream>
 #include <libnotify/notify.h>
 
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 using namespace std;
 
 int main(int argv, char* argc[])
@@ -9,7 +13,12 @@
     cout << "HOLA" << endl;
     notify_init("Twittaré");
 
-    NotifyNotification* nots = notify_notification_new("New twitt", "Tabaré Caorsi:\nEsto es un twit de prueba", "", 0);
+    NotifyNotification* nots = notify_notification_new("New twitt", "Tabaré Caorsi:\nEsto es un twit de prueba", ""
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+    );
+#else
+    , 0);
+#endif
     GError* error = 0;
     notify_notification_show(nots, &error);
 
--- libtwnotification/twnotification.cpp
+++ libtwnotification/twnotification.cpp
@@ -1,6 +1,10 @@
 #include "twnotification.h"
 #include <libnotify/notify.h>
 
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 TwNotification::TwNotification()
 {
         notify_init("Twittaré");
@@ -8,7 +12,12 @@
 
 void TwNotification::notify(const char * title, const char * text, const char * icon)
 {
-    NotifyNotification* nots = notify_notification_new(title, text, icon, 0);
+    NotifyNotification* nots = notify_notification_new(title, text, icon
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+    );
+#else
+    , 0);
+#endif
     GError* error = 0;
     notify_notification_show(nots, &error);
 }
