diff -ur psmisc-21.4.orig/src/fuser.c psmisc-21.4/src/fuser.c
--- psmisc-21.4.orig/src/fuser.c	2005-01-01 16:10:44.601397328 -0500
+++ psmisc-21.4/src/fuser.c	2005-01-01 16:11:22.884577400 -0500
@@ -31,3 +31,7 @@
+#ifdef USE_NLS
 #include <libintl.h>
 #include <locale.h>
 #define _(String) gettext (String)
+#else
+#define _(String) (String)
+#endif
@@ -1006,4 +1010,6 @@
+#ifdef USE_NLS
   /* Setup the i18n */
   setlocale(LC_ALL, "");
   bindtextdomain(PACKAGE, LOCALEDIR);
   textdomain(PACKAGE);
+#endif
diff -ur psmisc-21.4.orig/src/killall.c psmisc-21.4/src/killall.c
--- psmisc-21.4.orig/src/killall.c	2005-01-01 16:10:44.602397176 -0500
+++ psmisc-21.4/src/killall.c	2005-01-01 16:11:54.860716288 -0500
@@ -25,3 +25,7 @@
+#ifdef USE_NLS
 #include <libintl.h>
 #include <locale.h>
 #define _(String) gettext (String)
+#else
+#define _(String) (String)
+#endif
@@ -466,4 +470,6 @@
+#ifdef USE_NLS
   /* Setup the i18n */
   setlocale(LC_ALL, "");
   bindtextdomain(PACKAGE, LOCALEDIR);
   textdomain(PACKAGE);
+#endif
diff -ur psmisc-21.4.orig/src/pstree.c psmisc-21.4/src/pstree.c
--- psmisc-21.4.orig/src/pstree.c	2005-01-01 16:10:44.602397176 -0500
+++ psmisc-21.4/src/pstree.c	2005-01-01 16:12:14.659706384 -0500
@@ -22,2 +22,6 @@
+#ifdef USE_NLS
 #include <libintl.h>
 #define _(String) gettext (String)
+#else
+#define _(String) (String)
+#endif
