diff --git a/Build/lib/Makefile.am b/Build/lib/Makefile.am
index 3f5f0aa..d60dfcc 100644
--- a/Build/lib/Makefile.am
+++ b/Build/lib/Makefile.am
@@ -160,6 +160,7 @@ libapplications_la_CFLAGS = $(common_CFLAGS) @LIBXML_CFLAGS@
 
 COMMON_LDFLAGS = -export-dynamic -version-info @RFM_VERINFO@
 librfm_la_LDFLAGS = $(COMMON_LDFLAGS)
+librfm_la_LIBADD = @LIBSM_LIBS@
 librodent_la_LDFLAGS = $(COMMON_LDFLAGS)
 librfm_deskview_la_LDFLAGS = $(COMMON_LDFLAGS) 
 librfm_gridview_la_LDFLAGS = $(COMMON_LDFLAGS) 
