diff -ru centericq-4.8.7/configure centericq-4.8.7-patched/configure
--- centericq-4.8.7/configure	2002-11-01 16:39:11.000000000 +0100
+++ centericq-4.8.7-patched/configure	2002-11-05 23:09:37.000000000 +0100
@@ -1587,7 +1587,7 @@
   fi
 
       if test "x$no_sigc" = x ; then
-     CXXFLAGS="$CXXFLAGS $SIGC_CFLAGS" LIBS="$LIBS $SIGC_LIBS"     
+     CXXFLAGS="$CXXFLAGS $SIGC_CFLAGS"
   else
      SIGC_CFLAGS=""
      SIGC_LIBS=""
@@ -4468,6 +4468,7 @@
      GT_YES="#YES#"
    fi
    
+   LIBS="$LIBS $SIGC_LIBS"   
    
 
             MKINSTALLDIRS=
diff -ru centericq-4.8.7/configure.in centericq-4.8.7-patched/configure.in
--- centericq-4.8.7/configure.in	2002-11-01 16:38:40.000000000 +0100
+++ centericq-4.8.7-patched/configure.in	2002-11-05 23:24:42.000000000 +0100
@@ -24,7 +24,7 @@
 ###     sigc++ check
 
 AM_PATH_SIGC(1.0.0,
-    [CXXFLAGS="$CXXFLAGS $SIGC_CFLAGS" LIBS="$LIBS $SIGC_LIBS"],
+    [CXXFLAGS="$CXXFLAGS $SIGC_CFLAGS"],
     [AC_MSG_ERROR([You need to have the libsigc++ libraries installed - see the README for more information])])
 
 ###     use option --with-fribidi to compile with fribidi support
@@ -88,5 +88,6 @@
 
 ALL_LINGUAS="zh_TW.Big5 uk sv ru ro pt_BR pl nl ms it hu fr es de cs"
 AM_GNU_GETTEXT
+LIBS="$LIBS $SIGC_LIBS"
 AC_CONFIG_SUBDIRS(kkconsui-0.1 kkstrtext-0.1 kksystr-0.1 libicq2000-0.3.3 libmsn-0.1 firetalk-0.1 libyahoo2-0.1)
 AC_OUTPUT(Makefile contrib/Makefile contrib/spanish/Makefile misc/Makefile share/Makefile src/Makefile src/hooks/Makefile po/Makefile.in intl/Makefile)
