diff -rNu bluefish-0.13.orig/Makefile.in bluefish-0.13/Makefile.in
--- bluefish-0.13.orig/Makefile.in	 2004-04-09 12:30:01.000000000 +0200
+++ bluefish-0.13/Makefile.in		 2004-09-02 15:48:12.046968018 +0200
@@ -75,7 +75,8 @@
 	$(INSTALL_DIR) $(DESTDIR)$(pkgdatadir)
 	$(INSTALL_DATA) images/bluefish_splash.png $(DESTDIR)$(pkgdatadir)
 	@cd icons/ && $(MAKE) install
-	if [ -d $(iconpath) ]; then \
+	if [ -n $(iconpath) ]; then \
+		mkdir $(DESTDIR)$(iconpath) ;\
 		$(INSTALL_DATA) inline_images/bluefish_icon1.png $(DESTDIR)$(iconpath)/bluefish-icon.png ;\
 	fi
 	@cd src/ && $(MAKE) install
diff -rNu bluefish-0.13.orig/data/Makefile.in bluefish-0.13/data/Makefile.in
--- bluefish-0.13.orig/data/Makefile.in	2004-04-09 12:30:01.000000000 +0200
+++ bluefish-0.13/data/Makefile.in	2004-09-02 15:33:13.168328851 +0200
@@ -52,21 +52,26 @@
 	@for file in $(FUNCREFFILES) ; do \
 		$(INSTALL_DATA) $$file $(DESTDIR)$(pkgdatadir)/$$file ;\
 	done
-	if [ -d $(gnome2applications) ] ; then \
+	if [ -n $(gnome2applications) ] ; then \
+		mkdir $(DESTDIR)$(gnome2applications) ;\
 		$(INSTALL_DATA) bluefish.desktop $(DESTDIR)$(gnome2applications) ;\
 	fi
-	if [ -d $(gnome1menupath) ] ; then \
+	if [ -n $(gnome1menupath) ] ; then \
+		mkdir $(DESTDIR)$(gnome1menupath) ;\
 		$(INSTALL_DIR) $(DESTDIR)$(gnome1menupath)/Applications ;\
 		$(INSTALL_DATA) bluefish.desktop $(DESTDIR)$(gnome1menupath)/Applications ;\
 	fi
-	if [ -d $(gnome2appreg) ] ; then \
+	if [ -n $(gnome2appreg) ] ; then \
+		mkdir $(DESTDIR)$(gnome2appreg) ;\
 		$(INSTALL_DATA) bluefish.applications $(DESTDIR)$(gnome2appreg) ;\
 	fi
-	if [ -d $(gnome2mime-info) ] ; then \
+	if [ -n $(gnome2mime-info) ] ; then \
+		mkdir $(DESTDIR)$(gnome2mime-info) ;\
 		$(INSTALL_DATA) bluefish.mime $(DESTDIR)$(gnome2mime-info) ;\
 		$(INSTALL_DATA) bluefish.keys $(DESTDIR)$(gnome2mime-info) ;\
 	fi
-	if [ -d $(gnome2icons) ] ; then \
+	if [ -n $(gnome2icons) ] ; then \
+		mkdir $(DESTDIR)$(gnome2icons) ;\
 		$(INSTALL_DATA) gnome-application-bluefish.png $(DESTDIR)$(gnome2icons) ;\
 	fi
 
