diff -ur Desktop.old/Desktop.m Desktop/Desktop.m
--- Desktop.old/Desktop.m	2004-09-24 07:34:47.000000000 -0400
+++ Desktop/Desktop.m	2004-10-05 11:46:03.407363952 -0400
@@ -30,9 +30,9 @@
 #include "Dock.h"
 #include "Preferences/DesktopPrefs.h"
 #include "Dialogs/StartAppWin.h"
-#include "FSNode.h"
-#include "FSNodeRep.h"
-#include "FSNFunctions.h"
+#include "FSNode/FSNode.h"
+#include "FSNode/FSNodeRep.h"
+#include "FSNode/FSNFunctions.h"
 
 #define CHECKSIZE(sz) \
 if (sz.width < 0) sz.width = 0; \
diff -ur Desktop.old/DesktopView.h Desktop/DesktopView.h
--- Desktop.old/DesktopView.h	2004-09-02 07:31:38.000000000 -0400
+++ Desktop/DesktopView.h	2004-10-05 11:45:08.133766816 -0400
@@ -25,7 +25,7 @@
 #ifndef DESKTOP_VIEW_H
 #define DESKTOP_VIEW_H
 
-#include "FSNIconsView.h"
+#include "FSNode/FSNIconsView.h"
 
 @class NSImage;
 @class Desktop;
diff -ur Desktop.old/DesktopView.m Desktop/DesktopView.m
--- Desktop.old/DesktopView.m	2004-09-21 20:00:25.000000000 -0400
+++ Desktop/DesktopView.m	2004-10-05 11:46:54.995521360 -0400
@@ -26,9 +26,9 @@
 #include <AppKit/AppKit.h>
 #include <GNUstepGUI/GSDisplayServer.h>
 #include <math.h>
-#include "FSNodeRep.h"
-#include "FSNIcon.h"
-#include "FSNFunctions.h"
+#include "FSNode/FSNodeRep.h"
+#include "FSNode/FSNIcon.h"
+#include "FSNode/FSNFunctions.h"
 #include "DesktopView.h"
 #include "Desktop.h"
 #include "Dock.h"
diff -ur Desktop.old/Dialogs/StartAppWin.m Desktop/Dialogs/StartAppWin.m
--- Desktop.old/Dialogs/StartAppWin.m	2004-05-11 08:31:37.000000000 -0400
+++ Desktop/Dialogs/StartAppWin.m	2004-10-05 11:48:23.311095352 -0400
@@ -25,7 +25,7 @@
 #include <Foundation/Foundation.h>
 #include <AppKit/AppKit.h>
 #include "StartAppWin.h"
-#include "GNUstep.h"
+//#include "GNUstep.h"
 
 static NSString *nibName = @"StartAppWin";
 
diff -ur Desktop.old/Dock/Dock.h Desktop/Dock/Dock.h
--- Desktop.old/Dock/Dock.h	2004-09-02 07:31:38.000000000 -0400
+++ Desktop/Dock/Dock.h	2004-10-05 11:45:44.758199056 -0400
@@ -27,7 +27,7 @@
 
 #include <AppKit/NSView.h>
 #include "Desktop.h"
-#include "FSNodeRep.h"
+#include "FSNode/FSNodeRep.h"
 
 @class NSWindow;
 @class NSColor;
diff -ur Desktop.old/Dock/DockIcon.h Desktop/Dock/DockIcon.h
--- Desktop.old/Dock/DockIcon.h	2004-06-27 07:30:51.000000000 -0400
+++ Desktop/Dock/DockIcon.h	2004-10-05 11:47:10.600149096 -0400
@@ -26,7 +26,7 @@
 #define DOCK_ICON_H
 
 #include <AppKit/NSView.h>
-#include "FSNIcon.h"
+#include "FSNode/FSNIcon.h"
 
 @class NSColor;
 @class NSImage;
