diff -urN magyarispell-1.0/kr/Makefile magyarispell-1.0-fixed/kr/Makefile
--- magyarispell-1.0/kr/Makefile	2005-10-24 09:53:53.000000000 +0200
+++ magyarispell-1.0-fixed/kr/Makefile	2006-07-21 19:46:09.000000000 +0200
@@ -3,14 +3,14 @@
 
 WRKDIR=../tmp
 
-all: h test
+all: h
 
 test:
 	@hunmorph ../hu_HU_morph2.aff ../hu_HU_morph2.dic test.txt >../tmp/test.out
 	@#diff test.morph ../tmp/test.out && echo 'Elemzés ellenõrizve.' || echo 'Hiba!'
 
 h: hu
-	sed ':a s#<NOU\(.*\)>+*<NOUN#<NOU\1#;t a \
+	sed ":a s#<NOU\(.*\)>+*<NOUN#<NOU\1#;t a \
 	:b s#<VER\(.*\)>+*<VERB#<VER\1#;t b \
 	s#~##g; \
 	s#_[/(),A-Z]*_noun]#]NOUN{>}#g; \
@@ -23,8 +23,8 @@
 	s#_[/(),A-Z]*_(noun,adj)]#]NOUN{>}#g; \
 	s#_[/(),A-Z]*_num]#]NUM{>}#g; \
 	s#_[/(),A-Z]*_(adj,present_part)]#]ADJ{>}#g; \
-	' $(WRKDIR)/hu.aff >../hu_HU_morph2.aff
-	sed ':a s#<NOU\(.*\)>+*<NOUN#<NOU\1#;t a \
+	" $(WRKDIR)/hu.aff >../hu_HU_morph2.aff
+	sed ":a s#<NOU\(.*\)>+*<NOUN#<NOU\1#;t a \
 	:b s#<VER\(.*\)>+*<VERB#<VER\1#;t b \
 	s#~##g; \
 	s#_[/(),A-Z]*_noun]#]NOUN{>}#g; \
@@ -37,7 +37,7 @@
 	s#_[/(),A-Z]*_(noun,adj)]#]NOUN{>}#g; \
 	s#_[/(),A-Z]*_num]#]NUM{>}#g; \
 	s#_[/(),A-Z]*_(adj,present_part)]#]ADJ{>}#g; \
-	' $(WRKDIR)/hu.dic >../hu_HU_morph2.dic
+	" $(WRKDIR)/hu.dic >../hu_HU_morph2.dic
 
 hu: ../hu_HU_morph.aff ../hu_HU_morph.dic
 	sed -f csere.txt ../hu_HU_morph.aff >$(WRKDIR)/hu.aff
diff -urN magyarispell-1.0/Makefile magyarispell-1.0-fixed/Makefile
--- magyarispell-1.0/Makefile	2005-10-24 09:55:14.000000000 +0200
+++ magyarispell-1.0-fixed/Makefile	2006-07-21 19:46:04.000000000 +0200
@@ -122,10 +122,10 @@
 
 # egyéb kivételek: az m4 miatt nem megy külön héjprogramba
 	@echo "==> kivételek"
-	@#cat $(SZOTAR)/kivetelek/igekotos/*.1* \
-	@#	| grep -v ^# | $(M4) | tr "\t" "\n" \
-	@#	| grep -v ^$$ | sed -e 's/$$/\/X/' \
-	@#	>> $(DICT)
+	#@cat $(SZOTAR)/kivetelek/igekotos/*.1* \
+	#@	| grep -v ^# | $(M4) | tr "\t" "\n" \
+	#@	| grep -v ^$$ | sed -e 's/$$/\/X/' \
+	#@	>> $(DICT)
 	@echo $(EN) ."$(EC)"
 	@cat $(SZOTAR)/kivetelek/igekotos/*.2* \
 		| grep -v ^# | $(M4) | $(AWK) -f $(AWKDIR)/tovek.awk -v param="/X" \
@@ -220,7 +220,7 @@
 install_all: install_myspell install_ispell install_OO 
 
 # ispell opcióval kompilált hashtábla 
-#ispell: magyar4ispell.hash
+ispell: magyar4ispell.hash
 
 # hu_HU.aff myspell ragozási táblázat és a hu_HU.dic szótár elõállítása
 myspell: magyar4myspell.dict magyar.aff hu_HU.aff hu_HU.dic kr
