diff -Naur unionfs-1.0.13_orig/unionimap.c unionfs-1.0.13/unionimap.c
--- unionfs-1.0.13_orig/unionimap.c	2005-07-22 18:16:48.000000000 +0100
+++ unionfs-1.0.13/unionimap.c	2005-08-17 18:01:10.000000000 +0100
@@ -244,7 +244,7 @@
 			"Lower-Level");
 		while (read(file, (void *)&entry, sizeof(struct fmapent))) {
 			fprintf(stdout, "%-11lu %-8d %-22llu\n", inode++,
-				entry.fsnum, entry.inode);
+				entry.fsnum, (long long unsigned int) entry.inode);
 		}
 	}
       out:
@@ -293,7 +293,7 @@
 			unsigned long lowerlevel = 0;
 			if (inode) {
 				fprintf(stdout, "%-11lu %-22llu\n",
-					lowerlevel++, inode);
+					lowerlevel++, (long long unsigned int) inode);
 			}
 		}
 	}
