From 54fd2afbc1596cbb21a96adab89525ac65e06df4 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 16 Mar 2011 23:56:19 +0000
Subject: [PATCH] youtube: Only accept URLs with video IDs in them

---
 share/lua/website/youtube.lua |   23 +++++++++++++++++------
 1 files changed, 17 insertions(+), 6 deletions(-)

--- a/share/lua/website/youtube.lua
+++ b/share/lua/website/youtube.lua
@@ -45,17 +45,28 @@ local domains = {
     "youtu.be"
 }
 
--- Check whether the domain is handled
-function is_handled (page_url)
+-- Return the video ID
+function get_video_id (page_url)
     if page_url == nil then
-        return false
+        return nil
     end
     for k,domain in pairs(domains) do
         if page_url:find(domain) ~= nil then
-            return true
+            page_url = youtubify(page_url)
+            local _,_,s = page_url:find("v=([%w-_]+)")
+            return s
         end
     end
-    return false
+    return nil
+end
+
+-- Check whether the domain is handled
+function is_handled (page_url)
+    local id = get_video_id(page_url)
+    if id == nil then
+        return false
+    end
+    return true
 end
 
 -- Identify the script.
@@ -81,7 +92,7 @@ function parse (self)
     self.host_id  = "youtube"
     local page_url = youtubify(self.page_url)
 
-    local _,_,s = page_url:find("v=([%w-_]+)")
+    local s = get_video_id(page_url)
     self.id    = s or error ("no match: video id")
 
     local _,_,s = page_url:find('#t=(.+)')
-- 
1.7.4.1

