diff --git a/gentoo/splash-functions-bl1.sh b/gentoo/splash-functions-bl1.sh
index 4c07359..b1ba65d 100644
--- a/gentoo/splash-functions-bl1.sh
+++ b/gentoo/splash-functions-bl1.sh
@@ -22,6 +22,7 @@
 # is a size limit in KB, and it should probably be left with the
 # default value.
 spl_daemon="/sbin/fbsplashd.static"
+spl_fbcd="/sbin/fbcondecor_ctl"
 spl_bindir="/lib/splash/bin"
 spl_cachedir="/lib/splash/cache"
 spl_tmpdir="/lib/splash/tmp"
@@ -215,7 +216,7 @@ splash_start() {
 	rm -f ${spl_fifo} 2>/dev/null
 
 	if [[ ${SPLASH_MODE_REQ} == "verbose" ]]; then
-		/sbin/fbcondecor_ctl -c on 2>/dev/null
+		${spl_fbcd} -c on 2>/dev/null
 		return 0
 	elif [[ ${SPLASH_MODE_REQ} != "silent" ]]; then
 		return 0
@@ -263,7 +264,7 @@ splash_start() {
 	[[ ${SPLASH_KDMODE} == "GRAPHICS" ]] && options="--kdgraphics"
 
 	# Start the splash daemon
-	BOOT_MSG="$(splash_get_boot_message)" ${spl_daemon} -d --theme=${SPLASH_THEME} --pidfile=${spl_pidfile} ${options}
+	BOOT_MSG="$(splash_get_boot_message)" ${spl_daemon} --theme=${SPLASH_THEME} --pidfile=${spl_pidfile} ${options}
 
 	# Set the silent TTY and boot message
 	splash_comm_send "set tty silent ${SPLASH_TTY}"
@@ -271,12 +272,12 @@ splash_start() {
 	if [[ ${SPLASH_MODE_REQ} == "silent" ]] ; then
 		splash_comm_send "set mode silent"
 		splash_comm_send "repaint"
-		/sbin/fbcondecor_ctl -c on 2>/dev/null
+		${spl_fbcd} -c on 2>/dev/null
 	fi
 
 	# Set the input device if it exists. This will make it possible to use F2 to
 	# switch from verbose to silent.
-	local t=$(grep -Hsi keyboard /sys/class/input/input*/name | sed -e 's#.*input\\([0-9]*\\)/name.*#event\\1#')
+	local t=$(grep -Hsi keyboard /sys/class/input/input*/name | sed -e 's#.*input\([0-9]*\)/name.*#event\1#')
 	if [[ -z "${t}" ]]; then
 		t=$(grep -Hsi keyboard /sys/class/input/event*/device/driver/description | grep -o 'event[0-9]\+')
 		if [[ -z "${t}" ]]; then
@@ -351,7 +352,7 @@ splash_get_mode() {
 	if [[ ${ctty} == "${SPLASH_TTY}" ]]; then
 		echo "silent"
 	else
-		if [[ -z "$(/sbin/fbcondecor_ctl -c getstate --vc=$(($ctty-1)) 2>/dev/null | grep off)" ]]; then
+		if [[ -z "$(${spl_fbcd} -c getstate --vc=$(($ctty-1)) 2>/dev/null | grep off)" ]]; then
 			echo "verbose"
 		else
 			echo "off"
@@ -371,7 +372,7 @@ splash_verbose() {
 # Switches to silent mode.
 splash_silent() {
 	splash_comm_send "set mode silent"
-	/sbin/fbcondecor_ctl -c on 2>/dev/null
+	${spl_fbcd} -c on 2>/dev/null
 }
 
 splash_load_vars() {
