diff --git a/spec/aws/cloud_watch/alarm_history_item_collection_spec.rb b/spec/aws/cloud_watch/alarm_history_item_collection_spec.rb
index 912d057..5b65426 100644
--- a/spec/aws/cloud_watch/alarm_history_item_collection_spec.rb
+++ b/spec/aws/cloud_watch/alarm_history_item_collection_spec.rb
@@ -38,7 +38,7 @@ module AWS
         it 'returns a collection that applies the filter when enumerating' do
           client.should_receive(:describe_alarm_history).with(
             :abc => 'xyz',
-            :mno => 'hjk',
+            :mno => 'hjk'
           ).and_return(client.stub_for(:describe_alarm_history))
           items.filter('abc', 'xyz').filter('mno', 'hjk').to_a
         end
diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index c55235c..fea5527 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -14,12 +14,11 @@
 $: << File.join(File.dirname(File.dirname(__FILE__)), "lib")
 
 require 'rspec'
+require 'aws'
 
 # require all _examples.rb files in spec/shared/
 Dir.glob("#{File.dirname(__FILE__)}/shared/**/*_examples.rb").each do |file|
   require file
 end
 
-require 'aws'
-
 AWS.eager_autoload!
