--- imap/src/osdep/unix/env_unix.c.boguswarning	Mon Jun 25 23:10:04 2001
+++ imap/src/osdep/unix/env_unix.c	Thu Oct  4 05:12:36 2001
@@ -48,7 +48,7 @@
 				/* flock() emulator is a no-op */
 static short disableFcntlLock = NIL;
 				/* warning on EACCES errors on .lock files */
-static short lockEaccesError = T;
+static short lockEaccesError = NIL;
 static short hideDotFiles = NIL;/* hide files whose names start with . */
 				/* 1 = disable plaintext, 2 = if not SSL */
 static long disablePlaintext = NIL;
