--- db/Makefile.am.orig	2005-07-10 01:27:24.000000000 +0200
+++ db/Makefile.am	2005-07-10 01:27:58.000000000 +0200
@@ -137,6 +137,7 @@
 	xa_db.c \
 	xa_map.c
 
+libhtdb_la_CFLAGS = -fPIC -DPIC
 libhtdb_la_LDFLAGS = -release $(MAJOR_VERSION).$(MINOR_VERSION).$(MICRO_VERSION)
 
 pkginclude_HEADERS = \
--- htlib/Makefile.am.orig	2005-07-10 02:21:10.000000000 +0200
+++ htlib/Makefile.am	2005-07-10 02:21:29.000000000 +0200
@@ -20,6 +20,7 @@
 libht_la_LIBADD=@LTLIBOBJS@
 # this includes regex.c
 
+libht_la_CFLAGS = -fPIC -DPIC
 libht_la_LDFLAGS = -release $(HTDIG_MAJOR_VERSION).$(HTDIG_MINOR_VERSION).$(HTDIG_MICRO_VERSION) ${extra_ldflags}
 
 pkginclude_HEADERS = \
