--- gdgen.c-orig	2002-12-04 11:21:19.000000000 -0800
+++ gdgen.c	2002-12-04 11:22:52.000000000 -0800
@@ -754,8 +754,11 @@
 	else {
 	suffix = strrchr(imagefile,'.');
 	if (!suffix) suffix = imagefile; else suffix++;
+	#ifdef HAVE_LIBPNG
 	if (!strcasecmp(suffix,"png")) rv = gdImageCreateFromPng(in);
-	else if (!strcasecmp(suffix,"gif")) rv = gdImageCreateFromGif(in);
+	else //this split seems to be the easiest way to disable png correctly
+	#endif
+	  if (!strcasecmp(suffix,"gif")) rv = gdImageCreateFromGif(in);
 	else if (!strcasecmp(suffix,"jpeg")||!strcasecmp(suffix,"jpg")) rv = gdImageCreateFromJpeg(in);
 	else if (!strcasecmp(suffix,"wbmp")) rv = gdImageCreateFromWBMP(in);
 	else if (!strcasecmp(suffix,"xbm")) rv = gdImageCreateFromXbm(in);
