--- mailgraph.pl.orig	2005-06-27 01:34:55.000000000 +0200
+++ mailgraph.pl	2005-06-27 01:34:31.000000000 +0200
@@ -138,7 +138,7 @@
             (\d+):(\d+):(\d+) # time  -- 3, 4, 5
             (?:\s<\w+\.\w+>)? # FreeBSD's verbose-mode
             \s
-            ([-\w\.]+)        # host  -- 6
+            ([-\w\.\@]+)        # host  -- 6
             \s+
             (.*)              # text  -- 7
             $/x or do
