--- src/bacon-video-widget-xine.c.old	2003-12-08 15:21:56.000000000 +0100
+++ src/bacon-video-widget-xine.c	2003-12-08 15:21:18.000000000 +0100
@@ -581,10 +581,12 @@
 	 * we need to try loading the other ones, skipping dxr3 */
 
 	/* The types are hardcoded for now */
+	/*
 	vo_driver = xine_open_video_driver (bvw->priv->xine, "xvmc",
 			XINE_VISUAL_TYPE_X11, (void *) &vis);
 	if (vo_driver)
 		return vo_driver;
+*/
 
 	vo_driver = xine_open_video_driver (bvw->priv->xine, "xv",
 			XINE_VISUAL_TYPE_X11, (void *) &vis);
