--- totem-1.1.5/configure.in	2005-08-24 05:33:44.000000000 -0400
+++ totem-1.1.5.new/configure.in	2005-08-29 22:07:54.000000000 -0400
@@ -276,15 +276,8 @@
 dnl Search for the idl include directory
 if test "x$ENABLE_MOZILLA" = "xyes" ; then
 	dnl This only works on gecko 1.8
-	MOZILLA_IDLDIR="`$PKG_CONFIG --variable=idldir $MOZILLA-xpcom`"
-	dnl Fallback for older versions
-	if test "x$MOZILLA_IDLDIR" = "x"; then
-		MOZILLA_IDLDIR="`echo $MOZILLA_LIBDIR | sed -e s!lib!share/idl!`"
-	fi
-	dnl Some distributions (Gentoo) have it in unusual places
-	if test "x$MOZILLA_IDLDIR" = "x"; then
-		MOZILLA_IDLDIR="$MOZILLA_INCLUDE_ROOT/idl"
-	fi
+	dnl MOZILLA_IDLDIR="`$PKG_CONFIG --variable=idldir $MOZILLA-xpcom`"
+	MOZILLA_IDLDIR="`$PKG_CONFIG --variable=libdir $MOZILLA-xpcom`/include/idl"
 
 	AC_SUBST([MOZILLA_IDLDIR])
 fi
