--- browser-plugin/totemPlugin.cpp.orig	2006-08-15 23:29:30.000000000 -0500
+++ browser-plugin/totemPlugin.cpp	2006-08-15 23:29:43.000000000 -0500
@@ -42,8 +42,8 @@
 #include "npapi.h"
 #include "npupp.h"
 
-#include <nsIDOMWindow.h>
-#include <nsIURI.h>
+#include <dom/nsIDOMWindow.h>
+#include <necko/nsIURI.h>
 #include <nsEmbedString.h>
 #include <nsIInterfaceRequestor.h>
 #include <nsIInterfaceRequestorUtils.h>
