--- configure.in.new	2005-12-13 18:13:21.000000000 -0600
+++ configure.in	2005-12-13 18:14:54.000000000 -0600
@@ -211,7 +211,7 @@
 	AC_ARG_WITH([mozilla],
 		AS_HELP_STRING([--with-mozilla],[Which gecko engine to use (default: autodetect)]))
 
-	GECKOS="firefox mozilla-firefox seamonkey mozilla xulrunner"
+	GECKOS="gecko-sdk firefox mozilla-firefox seamonkey mozilla xulrunner"
 	gecko=$with_mozilla
 
 	if test "x$gecko" = "x"; then
@@ -241,6 +241,7 @@
 	AC_MSG_RESULT([$gecko])
 
 	case "$gecko" in
+      gecko-sdk) MOZILLA_VERSION_MIN=1.7 ;;
 		mozilla) MOZILLA_VERSION_MIN=1.7 ;;
 		seamonkey) MOZILLA_VERSION_MIN=1.0 ;;
 		*firefox) MOZILLA_VERSION_MIN=1.0 ;;
@@ -284,7 +285,8 @@
 	MOZILLA_XPCOM_CFLAGS="-I`$PKG_CONFIG --variable=includedir $MOZILLA-xpcom`"
 	MOZILLA_PLUGINDIR="${MOZILLA_PLUGINDIR:-$MOZILLA_LIBDIR}"
 
-	AC_PATH_PROG([MOZILLA_XPIDL], [xpidl], [no], [$MOZILLA_LIBDIR:$PATH])
+	AC_PATH_PROG([MOZILLA_XPIDL], [xpidl], [no],
+                [$MOZILLA_LIBDIR:$MOZILLA_LIBDIR/../bin:$PATH])
 	if test "x$MOZILLA_XPIDL" = "xno"; then
 		ENABLE_MOZILLA=no
 		AC_MSG_WARN([xpidl compiler not found])
