Index: totem-1.4.0/src/totem.c
===================================================================
--- totem-1.4.0.orig/src/totem.c
+++ totem-1.4.0/src/totem.c
@@ -1959,7 +1959,7 @@ on_about1_activate (GtkButton *button, T
 
 	totem->about = g_object_new (GTK_TYPE_ABOUT_DIALOG,
 			"name", _("Totem"),
-			"version", VERSION,
+			"version", PACKAGE_VERSION,
 			"copyright", _("Copyright \xc2\xa9 2002-2005 Bastien Nocera"),
 			"comments", description,
 			"authors", authors,
@@ -3512,7 +3512,7 @@ main (int argc, char **argv)
 #ifdef HAVE_GTK_ONLY
 	gtk_init (&argc, &argv);
 #else
-	gnome_program_init ("totem", VERSION,
+	gnome_program_init ("totem", PACKAGE_VERSION,
 			LIBGNOMEUI_MODULE,
 			argc, argv,
 			GNOME_PARAM_APP_DATADIR, DATADIR,
Index: totem-1.4.0/src/totem-mozilla-viewer.c
===================================================================
--- totem-1.4.0.orig/src/totem-mozilla-viewer.c
+++ totem-1.4.0/src/totem-mozilla-viewer.c
@@ -272,7 +272,7 @@ on_about1_activate (GtkButton *button, T
 
 	emb->about = g_object_new (GTK_TYPE_ABOUT_DIALOG,
 			"name", _("Totem Mozilla Plugin"),
-			"version", VERSION,
+			"version", PACKAGE_VERSION,
 			"copyright", _("Copyright \xc2\xa9 2002-2005 Bastien Nocera"),
 			"comments", description,
 			"authors", authors,
Index: totem-1.4.0/src/vanity.c
===================================================================
--- totem-1.4.0.orig/src/vanity.c
+++ totem-1.4.0/src/vanity.c
@@ -257,7 +257,7 @@ on_about1_activate (GtkButton *button, V
 
 	vanity->about = g_object_new (GTK_TYPE_ABOUT_DIALOG,
 			"name", _("Vanity"),
-			"version", VERSION,
+			"version", PACKAGE_VERSION,
 			"copyright", _("Copyright \xc2\xa9 2002-2005 Bastien Nocera"),
 			"comments", description,
 			"authors", authors,
@@ -654,7 +654,7 @@ main (int argc, char **argv)
 
 	gtk_init (&argc, &argv);
 #ifndef HAVE_GTK_ONLY
-	gnome_program_init ("vanity", VERSION,
+	gnome_program_init ("vanity", PACKAGE_VERSION,
 			LIBGNOMEUI_MODULE,
 			argc, argv,
 			GNOME_PARAM_APP_DATADIR, DATADIR,
