diff -ur src.orig/common.py src/common.py
--- src.orig/common.py	2007-05-03 14:42:01.000000000 -0700
+++ src/common.py	2007-06-09 19:22:00.000000000 -0700
@@ -860,7 +860,7 @@
         # windows running from exe, return as is
         return os.path.abspath(os.path.dirname(sys.executable))
     # running from src, up one
-    return os.path.split(os.path.abspath(os.path.dirname(sys.argv[0])))[0]
+    return os.path.abspath(os.path.dirname(sys.argv[0]))
 
 if sys.platform=='win32':
     # From Tim Golden's Win32 How Do I ...?
diff -ur src.orig/conversions.py src/conversions.py
--- src.orig/conversions.py	2007-05-03 14:42:01.000000000 -0700
+++ src/conversions.py	2007-06-09 19:22:00.000000000 -0700
@@ -20,11 +20,11 @@
 
 class ConversionFailed(Exception): pass
 
-helperdir=os.path.join(common.get_main_dir(), "helpers")
+helperdir="/usr/bin"
 
 osext={'win32': '.exe',
        'darwin': '.mbin',
-       'linux2': '.lbin'} \
+       'linux2': ''} \
        [sys.platform]
 
 # This shortname crap is needed because Windows programs (including ffmpeg)
@@ -128,14 +128,14 @@
     p=sys.path[0]
     if os.path.isfile(p):
         p=os.path.dirname(p)
-    helpersdirectory=os.path.abspath(os.path.join(p, 'helpers'))
+    helpersdirectory="/usr/bin"
     print "Helper Directory: "+helpersdirectory
     if sys.platform=='win32':
         osext=".exe"
     if sys.platform=='darwin':
         osext=".mbin"
     if sys.platform=='linux2':
-        osext=".lbin"
+        osext=""
         
     pngtopnmbin=gethelperbinary('pngtopnm')
     ppmquantbin=gethelperbinary('ppmquant')
diff -ur src.orig/importexport.py src/importexport.py
--- src.orig/importexport.py	2007-04-02 18:07:57.000000000 -0700
+++ src/importexport.py	2007-06-09 19:22:00.000000000 -0700
@@ -21,7 +21,7 @@
 import wx.html
 
 # Others
-from thirdparty import DSV
+import DSV
 
 # My modules
 import common
diff -ur src.orig/native/av/bmp2avi/Makefile src/native/av/bmp2avi/Makefile
--- src.orig/native/av/bmp2avi/Makefile	2006-05-05 00:10:08.000000000 -0700
+++ src/native/av/bmp2avi/Makefile	2007-06-09 19:22:00.000000000 -0700
@@ -13,6 +13,6 @@
 
 SRC=avi_file.cxx bmp_file.cxx b2a.cxx
 
-bmp2avi.lbin bmp2avi.mbin bmp2avi.exe: $(SRC)
+bmp2avi bmp2avi.mbin bmp2avi.exe: $(SRC)
 	$(CXX) $(EXTRADEFINES) -O2 -o $@ $(SRC)
 
diff -ur src.orig/native/usb/build.sh src/native/usb/build.sh
--- src.orig/native/usb/build.sh	2007-06-09 19:21:30.000000000 -0700
+++ src/native/usb/build.sh	2007-06-09 19:22:00.000000000 -0700
@@ -4,4 +4,4 @@
 swig -python -I/usr/include libusb.i
 
 gcc -fPIC -Wall -fno-strict-aliasing -O2 -g  -shared  -I $INCLUDEDIR -o _libusb.so libusb_wrap.c -lusb 
-strip _libusb.so
+#strip _libusb.so
diff -ur src.orig/phones/com_samsung.py src/phones/com_samsung.py
--- src.orig/phones/com_samsung.py	2007-01-09 14:28:37.000000000 -0800
+++ src/phones/com_samsung.py	2007-06-09 19:22:00.000000000 -0700
@@ -16,7 +16,7 @@
 import time
 
 # site-packages
-from thirdparty import DSV
+import DSV
 
 # BitPim modules
 import bpcalendar
