diff -uNr libpri-1.4.10.2.ORIG/Makefile libpri-1.4.10.2/Makefile
--- libpri-1.4.10.2.ORIG/Makefile	2009-12-08 22:55:12.729738494 +0000
+++ libpri-1.4.10.2/Makefile	2009-12-08 22:55:58.645775496 +0000
@@ -145,7 +145,7 @@
 	ranlib $(STATIC_LIBRARY)
 
 $(DYNAMIC_LIBRARY): $(DYNAMIC_OBJS)
-	$(CC) -shared $(SOFLAGS) -o $@ $(DYNAMIC_OBJS)
+	$(CC) -shared $(SOFLAGS) $(LDFLAGS) -o $@ $(DYNAMIC_OBJS)
 	$(LDCONFIG) $(LDCONFIG_FLAGS) .
 	ln -sf libpri.so.$(SONAME) libpri.so
 
