diff -Naur glimpse-4.17.4-old/libtemplate/include/util.h glimpse-4.17.4/libtemplate/include/util.h
--- glimpse-4.17.4-old/libtemplate/include/util.h	1999-11-03 16:40:57.000000000 -0500
+++ glimpse-4.17.4/libtemplate/include/util.h	2004-03-01 19:07:06.438665221 -0500
@@ -93,9 +93,14 @@
 void init_log3 _PARAMS((char *,FILE *,FILE *));	/* Initialize log routines */
 void log_errno _PARAMS((char *));		/* Same as perror(3) */
 void fatal_errno _PARAMS((char *));		/* Same as perror(3) & exit */
-#ifdef __STRICT_ANSI__
+#if defined(__STRICT_ANSI__) || (defined(__GNUC__) && __GNUC__ >= 3)
 #include <stdarg.h>
+# if defined(__GNUC__)
+#define log xlog
+void xlog _PARAMS((char *, ...));		/* Log a message */
+# else
 void log _PARAMS((char *, ...));		/* Log a message */
+# endif
 void errorlog _PARAMS((char *, ...));		/* Log an error message */
 void fatal _PARAMS((char *, ...));		/* Log error msg and exit */
 #else
diff -Naur glimpse-4.17.4-old/libtemplate/util/log.c glimpse-4.17.4/libtemplate/util/log.c
--- glimpse-4.17.4-old/libtemplate/util/log.c	1999-11-03 15:42:14.000000000 -0500
+++ glimpse-4.17.4/libtemplate/util/log.c	2004-03-01 19:07:13.934576926 -0500
@@ -55,7 +55,7 @@
 #include <errno.h>
 #include <time.h>
 #include <sys/file.h>
-#if defined(__STRICT_ANSI__)
+#if defined(__STRICT_ANSI__) || (defined(__GNUC__) && __GNUC__ >= 3)
 #include <stdarg.h>
 #else
 #include <varargs.h>
@@ -119,7 +119,7 @@
 /*
  *  log() - used like printf(3).  Prints message to stdout.
  */
-#if defined(__STRICT_ANSI__)
+#if defined(__STRICT_ANSI__) || (defined(__GNUC__) && __GNUC__ >= 3)
 void log(char *fmt,...)
 {
 	va_list ap;
@@ -155,7 +155,7 @@
 /*
  *  errorlog() - used like printf(3).  Prints error message to stderr.
  */
-#if defined(__STRICT_ANSI__)
+#if defined(__STRICT_ANSI__) || (defined(__GNUC__) && __GNUC__ >= 3)
 void errorlog(char *fmt,...)
 {
 	va_list ap;
@@ -193,7 +193,7 @@
 /*
  *  fatal() - used like printf(3).  Prints error message to stderr and exits
  */
-#if defined(__STRICT_ANSI__)
+#if defined(__STRICT_ANSI__) || (defined(__GNUC__) && __GNUC__ >= 3)
 void fatal(char *fmt,...)
 {
 	va_list ap;
