Index: minitube/src/MainWindow.h
===================================================================
--- minitube.orig/src/MainWindow.h
+++ minitube/src/MainWindow.h
@@ -4,7 +4,10 @@
 #include <QtGui>
 #include "faderwidget/FaderWidget.h"
 #include "searchlineedit.h"
-#include <phonon>
+#include <phonon/audiooutput.h>
+#include <phonon/volumeslider.h>
+#include <phonon/mediaobject.h>
+#include <phonon/seekslider.h>
 #include "View.h"
 #include "SearchView.h"
 #include "MediaView.h"
Index: minitube/src/MediaView.h
===================================================================
--- minitube.orig/src/MediaView.h
+++ minitube/src/MediaView.h
@@ -3,7 +3,8 @@
 
 #include <QtGui>
 #include <QtNetwork>
-#include <phonon>
+#include <phonon/mediaobject.h>
+#include <phonon/videowidget.h>
 #include "View.h"
 #include "ListModel.h"
 #include "thaction.h"
Index: minitube/minitube.pro
===================================================================
--- minitube.orig/minitube.pro
+++ minitube/minitube.pro
@@ -1,5 +1,5 @@
 # If Phonon cannot be found, uncomment the following (and set the correct path)
-# INCLUDEPATH += /usr/include/phonon
+INCLUDEPATH += /usr/include/phonon
 CONFIG += release
 TEMPLATE = app
 QT += network \
Index: minitube/src/videowidget.h
===================================================================
--- minitube.orig/src/videowidget.h
+++ minitube/src/videowidget.h
@@ -2,7 +2,7 @@
 #define VIDEOWIDGET_H
 
 #include <QtGui>
-#include <phonon>
+#include <phonon/videowidget.h>
 
 class VideoWidget : public Phonon::VideoWidget {
 
Index: minitube/src/main.cpp
===================================================================
--- minitube.orig/src/main.cpp
+++ minitube/src/main.cpp
@@ -22,8 +22,7 @@ int main(int argc, char **argv) {
     app.installTranslator(&qtTranslator);
 
     // translations
-    QString localeDir = QCoreApplication::applicationDirPath()
-                        + QDir::separator() + "locale";
+    QString localeDir = "/usr/share/minitube/locale";
     QTranslator translator;
     translator.load(locale, localeDir);
     app.installTranslator(&translator);
