diff -Nru vzctl-3.0.22.orig/src/lib/config.c vzctl-3.0.22/src/lib/config.c
--- vzctl-3.0.22.orig/src/lib/config.c	2008-06-18 20:12:15.000000000 -0400
+++ vzctl-3.0.22/src/lib/config.c	2008-06-18 20:12:58.000000000 -0400
@@ -29,6 +29,7 @@
 #include <linux/vzcalluser.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <limits.h>
 
 #include "logger.h"
 #include "list.h"
diff -Nru vzctl-3.0.22.orig/src/lib/util.c vzctl-3.0.22/src/lib/util.c
--- vzctl-3.0.22.orig/src/lib/util.c	2008-06-18 20:12:15.000000000 -0400
+++ vzctl-3.0.22/src/lib/util.c	2008-06-18 20:14:08.000000000 -0400
@@ -28,7 +28,7 @@
 #include <fcntl.h>
 #include <arpa/inet.h>
 #include <stdarg.h>
-#include <linux/limits.h>
+#include <limits.h>
 #include <dirent.h>
 
 #include "util.h"
