diff -NurpP vzctl-2.7.0-25/etc/ifup-venet vzctl-2.7.0-25-amd64/etc/ifup-venet
--- vzctl-2.7.0-25/etc/ifup-venet	2005-12-29 16:57:05.000000000 +0300
+++ vzctl-2.7.0-25-amd64/etc/ifup-venet	2006-01-12 16:54:15.145147672 +0300
@@ -21,7 +21,7 @@ vznet=$DEVICE
 modvenet="vznetdev vznet"
 
 . /etc/init.d/functions
-. /usr/lib/vzctl/scripts/vps-functions
+. /usr/lib64/vzctl/scripts/vps-functions
 . /etc/sysconfig/vz
 
 veinfo="/proc/vz/veinfo"
diff -NurpP vzctl-2.7.0-25/etc/vps-create vzctl-2.7.0-25-amd64/etc/vps-create
--- vzctl-2.7.0-25/etc/vps-create	2005-12-29 16:57:05.000000000 +0300
+++ vzctl-2.7.0-25-amd64/etc/vps-create	2006-01-12 16:54:09.680978352 +0300
@@ -16,7 +16,7 @@
 #   VE_PRVT		- path to root of VPS private areas
 #   PRIVATE_TEMPLATE	- path to private template used as a source for copying
 
-. /usr/lib/vzctl/scripts/vps-functions
+. /usr/lib64/vzctl/scripts/vps-functions
 
 vzcheckvar VE_PRVT PRIVATE_TEMPLATE
 
diff -NurpP vzctl-2.7.0-25/etc/vps-net_add vzctl-2.7.0-25-amd64/etc/vps-net_add
--- vzctl-2.7.0-25/etc/vps-net_add	2005-12-29 16:57:05.000000000 +0300
+++ vzctl-2.7.0-25-amd64/etc/vps-net_add	2006-01-12 16:54:09.681978200 +0300
@@ -18,7 +18,7 @@
 #                   (several addresses should be divided by space)
 #   VE_STATE      - state of VPS; could be one of:
 #                     starting | stopping | running | stopped
-. /usr/lib/vzctl/scripts/vps-functions
+. /usr/lib64/vzctl/scripts/vps-functions
 
 vzcheckvar VEID
 
diff -NurpP vzctl-2.7.0-25/etc/vps-net_del vzctl-2.7.0-25-amd64/etc/vps-net_del
--- vzctl-2.7.0-25/etc/vps-net_del	2005-12-29 16:57:05.000000000 +0300
+++ vzctl-2.7.0-25-amd64/etc/vps-net_del	2006-01-12 16:54:09.682978048 +0300
@@ -18,7 +18,7 @@
 #   VE_STATE      - state of VPS; could be one of:
 #                     starting | stopping | running | stopped
 
-. /usr/lib/vzctl/scripts/vps-functions
+. /usr/lib64/vzctl/scripts/vps-functions
 
 vzcheckvar IP_ADDR VEID
 
diff -NurpP vzctl-2.7.0-25/etc/vps-stop vzctl-2.7.0-25-amd64/etc/vps-stop
--- vzctl-2.7.0-25/etc/vps-stop	2005-12-29 16:57:05.000000000 +0300
+++ vzctl-2.7.0-25-amd64/etc/vps-stop	2006-01-12 16:54:09.682978048 +0300
@@ -15,7 +15,7 @@
 #   VEID    - VPS id
 #   IP_ADDR - VPS IP address(es) divided by spaces
 
-. /usr/lib/vzctl/scripts/vps-functions
+. /usr/lib64/vzctl/scripts/vps-functions
 
 # get list of network devices for vzarp
 vzgetnetdev
diff -NurpP vzctl-2.7.0-25/etc/vpsnetclean vzctl-2.7.0-25-amd64/etc/vpsnetclean
--- vzctl-2.7.0-25/etc/vpsnetclean	2005-12-29 16:57:06.000000000 +0300
+++ vzctl-2.7.0-25-amd64/etc/vpsnetclean	2006-01-12 16:54:09.683977896 +0300
@@ -9,7 +9,7 @@
 # WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
 #
 
-. /usr/lib/vzctl/scripts/vps-functions
+. /usr/lib64/vzctl/scripts/vps-functions
 
 VEINFO=/proc/vz/veinfo
 PROC_PROXY_TR=/proc/proxy/targets
diff -NurpP vzctl-2.7.0-25/include/modules.h vzctl-2.7.0-25-amd64/include/modules.h
--- vzctl-2.7.0-25/include/modules.h	2005-12-29 16:57:06.000000000 +0300
+++ vzctl-2.7.0-25-amd64/include/modules.h	2006-01-12 16:52:10.262132776 +0300
@@ -17,7 +17,7 @@
 #define	TYPE_CMD	0
 #define	TYPE_CONF	1
 
-#define MOD_DIR		"/usr/lib/vzctl/modules/"
+#define MOD_DIR		"/usr/lib64/vzctl/modules/"
 
 #define MOD_INFO_SYM    "vz_get_mod_info"
 
diff -NurpP vzctl-2.7.0-25/include/types.h vzctl-2.7.0-25-amd64/include/types.h
--- vzctl-2.7.0-25/include/types.h	2005-12-29 16:57:06.000000000 +0300
+++ vzctl-2.7.0-25-amd64/include/types.h	2006-01-12 16:52:23.307149632 +0300
@@ -13,8 +13,8 @@
 
 #define GLOBAL_CFG              "/etc/conf.d/vz"
 #define VPS_CONF_DIR            "/etc/vz/"
-#define LIB_DIR			"/usr/lib/vzctl/lib/"
-#define VPS_SCRIPTS_DIR         "/usr/lib/vzctl/scripts/"
+#define LIB_DIR			"/usr/lib64/vzctl/lib/"
+#define VPS_SCRIPTS_DIR         "/usr/lib64/vzctl/scripts/"
 
 #define VPS_STOP		VPS_SCRIPTS_DIR "vps-stop"
 #define VPS_NET_ADD		VPS_SCRIPTS_DIR "vps-net_add"
diff -NurpP vzctl-2.7.0-25/src/Makefile vzctl-2.7.0-25-amd64/src/Makefile
--- vzctl-2.7.0-25/src/Makefile	2005-12-29 16:57:06.000000000 +0300
+++ vzctl-2.7.0-25-amd64/src/Makefile	2006-01-12 16:51:56.383242688 +0300
@@ -11,7 +11,7 @@
      INSTALL = install
       PREFIX = /usr
      SBINDIR = $(PREFIX)/sbin
-      LIBDIR = $(PREFIX)/lib/vzctl
+      LIBDIR = $(PREFIX)/lib64/vzctl
     CACHEDIR = /vz/template/cache
    GLBCONFIG = /etc/sysconfig/vz
   VPSCONFDIR = /etc/sysconfig/vz-scripts
