--- lib/as_ares.c.orig	2005-02-09 20:46:07.000000000 +0100
+++ lib/as_ares.c	2005-04-27 21:23:20.083273544 +0200
@@ -61,7 +61,7 @@
 	/* id, name, type, data, callback, udata */
 	{ AS_LISTEN_PORT,                 "main/port",      AS_CONF_INT, {59049},
 	                                  port_change_cb,   NULL },
-	{ AS_USER_NAME,                   "main/username",  AS_CONF_STR, {"antares"},
+	{ AS_USER_NAME,                   "main/username",  AS_CONF_STR, {0},
 	                                  NULL,             NULL },
 	{ AS_DOWNLOAD_MAX_ACTIVE,         NULL,             AS_CONF_INT, {6},
 	                                  NULL,             NULL },
