--- eix-0.3.0/src/formatstring.cc.orig	2005-08-27 19:41:55.000000000 +0200
+++ eix-0.3.0/src/formatstring.cc	2005-08-27 19:42:03.000000000 +0200
@@ -283,6 +283,9 @@
 		p = keller.top();
 		keller.pop();
 	}
+	if(q == NULL) {
+		q = new Text("");
+	}
 	((ConditionBlock*)p)->if_true = q;
 	keller.push(p);
 	return START;
@@ -307,7 +310,12 @@
 		p = keller.top();
 		keller.pop();
 	}
-	((ConditionBlock*)p)->if_false = q;
+	if(((ConditionBlock*)p)->if_true == NULL) {
+		((ConditionBlock*)p)->if_true = q;
+	}
+	else {
+		((ConditionBlock*)p)->if_false = q;
+	}
 	((ConditionBlock*)p)->final = true; 
 	keller.push(p);
 	return START;
