--- eix-0.3.0/src/cascadingprofile.cc.orig	2005-08-26 21:10:43.000000000 +0200
+++ eix-0.3.0/src/cascadingprofile.cc	2005-09-19 14:39:13.000000000 +0200
@@ -59,8 +59,7 @@
  * if there is no parent profile. */
 bool CascadingProfile::getParentProfile(string &path_buffer)
 {
-	int _buflen = 128;
-	char _buf[_buflen];
+	string buf;
 
 	/* Open stream and check if it's open */
 	ifstream ifstr((path_buffer + "parent").c_str());
@@ -68,16 +67,14 @@
 		return false;
 
 	/* while there are lines in the file */
-	while(ifstr.getline(_buf, _buflen))
+	while(getline(ifstr, buf))
 	{
+		buf = trim(buf, "\t\n\r ");
 		/* If it's a comment or a empty line continue with the next line */
-		int p = strspn(_buf,"\t ");
-		if(_buf[p] == '#' || _buf[p] == '\0')
+		if(buf.size() == 0 || buf[0] == '#')
 			continue;
-		_buf[ strcspn(_buf + p, "\t ") + p] = 0;
 
-		/* Check if the line would fit into the buffer */
-		path_buffer.append(_buf + p);
+		path_buffer.append(buf);
 		path_buffer.append("/");
 		return true;
 	}
