--- usr/bin/nxloadconfig.orig  2006-11-02 08:08:36.000000000 +0000
+++ usr/bin/nxloadconfig       2006-11-05 01:18:16.000000000 +0000
@@ -157,8 +157,8 @@
 KILL_DEFAULT_X_WM="1"
 USER_X_STARTUP_SCRIPT=.Xclients
 DEFAULT_X_SESSION=/etc/X11/xdm/Xsession
-COMMAND_START_KDE="/usr/bin/ssh-agent /usr/bin/dbus-launch --exit-with-session /usr/bin/startkde"
-COMMAND_START_GNOME="/usr/bin/ssh-agent /usr/bin/dbus-launch --exit-with-session /usr/bin/gnome-session"
+COMMAND_START_KDE="startkde"
+COMMAND_START_GNOME="/etc/X11/Sessions/Gnome"
 COMMAND_START_CDE=cdwm
 COMMAND_XTERM=xterm
 COMMAND_XAUTH=/usr/X11R6/bin/xauth
