--- include/config.h.orig	2003-12-01 17:13:46.792453840 -0500
+++ include/config.h	2003-12-01 17:15:08.339056872 -0500
@@ -179,8 +179,8 @@
 
 #ifdef UNIX
 /* path and file name extension for compression program */
-#define COMPRESS "/usr/bin/compress"	/* Lempel-Ziv compression */
-#define COMPRESS_EXTENSION ".Z"		/* compress's extension */
+#define COMPRESS "/bin/bzip2"
+#define COMPRESS_EXTENSION ".bz2"
 /* An example of one alternative you might want to use: */
 /* #define COMPRESS "/usr/local/bin/gzip" */	/* FSF gzip compression */
 /* #define COMPRESS_EXTENSION ".gz" */		/* normal gzip extension */
@@ -217,7 +217,7 @@
 #  ifdef __APPLE__
 #    define HACKDIR "nethackdir"	/* nethack directory */
 #  else
-#    define HACKDIR "/usr/games/lib/nethackdir"
+#    define HACKDIR "GENTOO_HACKDIR"
 #  endif
 # endif
 
@@ -314,7 +314,7 @@
  * functions that have been macroized.
  */
 
-/* #define VISION_TABLES */ /* use vision tables generated at compile time */
+#define VISION_TABLES 1
 #ifndef VISION_TABLES
 # ifndef NO_MACRO_CPATH
 #  define MACRO_CPATH	/* use clear_path macros instead of functions */
