diff -urN gpscorrelate-1.6.1.old/Makefile gpscorrelate-1.6.1/Makefile
--- gpscorrelate-1.6.1.old/Makefile	2011-06-07 11:01:35.671339619 +0200
+++ gpscorrelate-1.6.1/Makefile	2011-06-07 11:09:31.705388371 +0200
@@ -4,11 +4,10 @@
 
 COBJS    = main-command.o unixtime.o gpx-read.o correlate.o exif-gps.o
 GOBJS    = main-gui.o gui.o unixtime.o gpx-read.o correlate.o exif-gps.o
-CFLAGS   = -Wall
-override CFLAGS += $(shell pkg-config --cflags libxml-2.0 gtk+-2.0) -I/usr/include/exiv2
-OFLAGS   = -Wall
-override OFLAGS += $(shell pkg-config --libs libxml-2.0 gtk+-2.0) -lm -lexiv2
-prefix   = /usr/local
+CFLAGS += $(shell pkg-config --cflags libxml-2.0 gtk+-2.0) -I/usr/include/exiv2
+CXXFLAGS += $(shell pkg-config --cflags libxml-2.0 gtk+-2.0) -I/usr/include/exiv2
+CLIBS += $(shell pkg-config --libs libxml-2.0 gtk+-2.0) -lm -lexiv2
+prefix   ?= /usr/local
 bindir   = $(prefix)/bin
 datadir  = $(prefix)/share
 mandir   = $(datadir)/man
@@ -18,16 +17,16 @@
 all:	gpscorrelate gpscorrelate-gui gpscorrelate.1
 
 gpscorrelate: $(COBJS)
-	g++ $(OFLAGS) -o $@ $(COBJS)
+	$(CXX) $(LDFLAGS) -o $@ $(COBJS) $(CLIBS)
 
 gpscorrelate-gui: $(GOBJS)
-	g++ $(OFLAGS) -o $@ $(GOBJS)
+	$(CXX) $(LDFLAGS) -o $@ $(GOBJS) $(CLIBS)
 
 .c.o:
-	gcc $(CFLAGS) -c -o $*.o $<
+	$(CC) $(CFLAGS) -c -o $*.o $<
 
 .cpp.o:
-	g++ $(CFLAGS) -c -o $*.o $<
+	$(CXX) $(CXXFLAGS) -c -o $*.o $<
 
 clean:
 	rm -f *.o gpscorrelate{,.exe} gpscorrelate-gui{,.exe}
