--- uwsgiconfig.py
+++ uwsgiconfig.py
@@ -68,9 +68,9 @@
 	p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
 
 	if p.wait() == 0:
-        	return p.stdout.read().rstrip().decode()
+		return p.stdout.read().rstrip().decode()
 	else:
-        	return None
+		return None
 
 def add_o(x):
 	if x == 'uwsgi':
@@ -149,11 +149,11 @@
 	kvm_list = ['FreeBSD', 'OpenBSD', 'NetBSD', 'DragonFly']
 
 	if uwsgi_os == 'SunOS':
-        	ldflags.append('-lsendfile')
+		ldflags.append('-lsendfile')
 		ldflags.remove('-rdynamic')
 
 	if uwsgi_os in kvm_list:
-        	ldflags.append('-lkvm')
+		ldflags.append('-lkvm')
 
 	if uwsgi_os == 'OpenBSD':
 		UGREEN = False
@@ -245,8 +245,8 @@
 		gcc_list.append('erlang')
 
 	if SCTP:
-        	ldflags.append("-lsctp")
-        	cflags.append("-DUWSGI_SCTP")
+		ldflags.append("-lsctp")
+		cflags.append("-DUWSGI_SCTP")
 
 	if SPOOLER:
 		depends_on("SPOOLER", ['EMBEDDED'])
@@ -294,9 +294,9 @@
 		sys.exit(1)
 
 	if cmd == '--cflags':
-        	print(' '.join(cflags))
+		print(' '.join(cflags))
 	if cmd == '--ldflags':
-        	print(' '.join(ldflags))
+		print(' '.join(ldflags))
 	elif cmd == '--build':
 		parse_vars()
 		build_uwsgi(UWSGI_BIN_NAME)
