diff --git a/uwsgiconfig.py b/uwsgiconfig.py
index 5025b51..d817ae1 100644
--- a/uwsgiconfig.py
+++ b/uwsgiconfig.py
@@ -75,17 +75,7 @@ def spcall(cmd):
 	else:
 		return None
 
-def spcall2(cmd):
-	p = subprocess.Popen(cmd, shell=True, stderr=subprocess.PIPE)
-
-	if p.wait() == 0:
-		if sys.version_info[0] > 2:
-			return p.stderr.read().rstrip().decode()
-		return p.stderr.read().rstrip()
-	else:
-		return None
-
-gcc_version = str(spcall2("%s -v" % GCC)).split('\n')[-1].split()[2]
+gcc_version = str(spcall("%s -dumpversion" % GCC))
 
 gcc_major = int(gcc_version.split('.')[0])
 gcc_minor = int(gcc_version.split('.')[1])
