diff -Nura dasher-4.7.0.old/Src/Makefile.am dasher-4.7.0/Src/Makefile.am
--- dasher-4.7.0.old/Src/Makefile.am	2008-03-29 22:11:18.000000000 +0100
+++ dasher-4.7.0/Src/Makefile.am	2008-03-29 22:12:27.000000000 +0100
@@ -24,17 +24,17 @@
 	-DPACKAGE_LOCALE_DIR=\"$(datadir)/locale\"
 
 dasher_LDADD = \
-	$(GTK2BUILD_LIBS) \
-	-lexpat \
-	$(POPT_LIBS) \
-	$(GPELIB) \
 	Common/libdashermisc.a \
 	Gtk2/libdashergtk.a \
 	Gtk2/libdashercontrol.a \
 	DasherCore/libdashercore.a \
 	DasherCore/libdasherprefs.a \
 	DasherCore/LanguageModelling/libdasherlm.a \
-	Common/libdashermisc.a
+	Common/libdashermisc.a \
+	$(GTK2BUILD_LIBS) \
+	-lexpat \
+	$(POPT_LIBS) \
+	$(GPELIB)
 #endif
 
 #if DOQTE
