--- handler.c.old	2007-05-08 22:24:43.000000000 +0200
+++ handler.c	2007-05-08 22:29:15.000000000 +0200
@@ -19,6 +19,7 @@
 
 #include "main.h"
 #include "handler.h"
+static char state_dir[] = "@GENTOO_STATEDIR@/";
 
 /* This is an internally used function to set notebook tab widgets. */
 /*
@@ -1483,7 +1484,7 @@
    strcpy(rank_record[insert_num-1].word, gtk_entry_get_text(GTK_ENTRY(dialog_word_t)) );
    rank_record[insert_num-1].grade = grade;
 
-   strcpy(string, data_dir);
+   strcpy(string, state_dir);
    strcat(string, RANK_FILE);
    fp = fopen(string, "w");
    fd = fileno(fp);
@@ -1527,7 +1528,7 @@
    gint i;
    gchar string[100];
 
-   strcpy(string, data_dir);
+   strcpy(string, state_dir);
    strcat(string, RANK_FILE);
    fp = fopen(string, "r");
    if (fp==NULL)
@@ -1561,7 +1561,7 @@
 
    for (i=0;i<RANK_NUM*sizeof(struct rank_struct);i++)
       buf[i] = 0;
-   strcpy(string, data_dir);
+   strcpy(string, state_dir);
    strcat(string, RANK_FILE);
    fp = fopen(string, "w");
    fwrite(buf, 1, RANK_NUM*sizeof(struct rank_struct), fp);
--- Makefile.old	2007-05-08 22:59:55.000000000 +0200
+++ Makefile	2007-05-08 23:17:45.000000000 +0200
@@ -18,7 +18,6 @@
 #           -lz -lgif  -lXi -lXext -lX11 -lm
 #LIBS=-L/usr/lib -L/usr/X11R6/lib -L/usr/local/lib `gtk-config --libs`
 LIBS=-lgdk_imlib -ljpeg -ltiff -lpng -lz -lgif  `gtk-config --libs` -lXi -lXext -lX11 -lm
-CFLAGS= -g -O2 -Wall 
 
 INSTALL = /usr/bin/install -c
 INSTALL_DIR= $(INSTALL) -d -m 755
