diff -urN unicon-3.0.4.orig/unicon/ImmModules/cxterm/utils/cit2tit.c unicon-3.0.4/unicon/ImmModules/cxterm/utils/cit2tit.c
--- unicon-3.0.4.orig/unicon/ImmModules/cxterm/utils/cit2tit.c	2004-12-15 14:09:18.836641936 +0800
+++ unicon-3.0.4/unicon/ImmModules/cxterm/utils/cit2tit.c	2004-12-15 14:10:01.207200632 +0800
@@ -29,7 +29,7 @@
 #include <ctype.h>
 #include <stdio.h>
 
-extern char *malloc(), *calloc(), *realloc();
+/*extern char *malloc(), *calloc(), *realloc();*/
 
 extern char *HZencodeName();	/* from HZutil.c */
 
diff -urN unicon-3.0.4.orig/unicon/ImmModules/cxterm/utils/tit2cit.c unicon-3.0.4/unicon/ImmModules/cxterm/utils/tit2cit.c
--- unicon-3.0.4.orig/unicon/ImmModules/cxterm/utils/tit2cit.c	2004-12-15 14:09:18.836641936 +0800
+++ unicon-3.0.4/unicon/ImmModules/cxterm/utils/tit2cit.c	2004-12-15 14:11:06.249312728 +0800
@@ -54,7 +54,7 @@
 #include <ctype.h>
 #include <stdio.h>
 
-extern char *malloc(), *calloc(), *realloc();
+/*extern char *malloc(), *calloc(), *realloc();*/
 
 extern int HZencode();		/* from HZutil.o */
 extern int HZgetprompt();	/* from HZutil.o */
@@ -744,7 +744,7 @@
 	    Error ("Run out of memory");
 	strcpy (tptr->son->hzptr, hzptr);
     } else {
-	tptr->son->hzptr = realloc (tptr->son->hzptr,
+	tptr->son->hzptr = (void *)realloc (tptr->son->hzptr,
 		strlen (tptr->son->hzptr) + strlen (hzptr) + 1);
 	if (tptr->son->hzptr == NULL)
 	    Error ("Run out of memory");
diff -urN unicon-3.0.4.orig/tools/uniconcfg.c unicon-3.0.4/tools/uniconcfg.c
--- unicon-3.0.4.orig/tools/uniconcfg.c	2004-12-14 16:57:42.000000000 +0800
+++ unicon-3.0.4/tools/uniconcfg.c	2004-12-14 16:59:50.000000000 +0800
@@ -214,6 +214,7 @@
         case XL_DB_GBK:
         case XL_DB_EUCJP:
         case XL_DB_SJIS:
+		break;
     }
     return 1;
 }
