diff -U 4 -H -d -r -N -- include-orig/factory.h include/factory.h
--- include-orig/factory.h	2004-04-27 20:52:59.116161000 +0200
+++ include/factory.h	2004-04-27 21:05:17.617892056 +0200
@@ -488,9 +488,9 @@
 void psqr ( const CanonicalForm & f, const CanonicalForm & g, CanonicalForm & q, CanonicalForm & r, const Variable & x );
 
 CanonicalForm bCommonDen ( const CanonicalForm & f );
 
-bool divides ( const CanonicalForm & f, const CanonicalForm & g );
+//bool divides ( const CanonicalForm & f, const CanonicalForm & g );
 
 CanonicalForm maxNorm ( const CanonicalForm & f );
 
 CanonicalForm euclideanNorm ( const CanonicalForm & f );
