diff -ur cl-mpd.orig/mpdclient.lisp cl-mpd/mpdclient.lisp
--- cl-mpd.orig/mpdclient.lisp	2006-06-26 20:58:46.000000000 -0500
+++ cl-mpd/mpdclient.lisp	2006-07-09 22:04:14.000000000 -0500
@@ -56,9 +56,6 @@
 (eval-when (:compile-toplevel :load-toplevel)
   (declaim (optimize (speed 0) (space 0) (safety 1) (debug 3) (compilation-speed 0))))
 
-(defpackage #:mpd
-  (:use #:cl))
-
 (in-package #:mpd)
 
 ;;; Find out when a REQUIRE is normally evaluated.
@@ -66,7 +63,11 @@
   #+sbcl (require :sb-bsd-sockets)
   #+lispworks (require "comm"))
 
-(defconstant +mpd-welcome-message+       "OK MPD ")
+(defconstant +mpd-welcome-message+
+  (if (boundp '+mpd-welcome-message+)	;http://www.sbcl.org/manual/Defining-Constants.html#Defining-Constants
+      (symbol-value '+mpd-welcome-message+)
+      "OK MPD "))
+
 ;;; These aren't used yet, but will be soon.
 (defconstant +mpd-error-not-list+        1)
 (defconstant +mpd-error-arg+             2)
diff -ur cl-mpd.orig/package.lisp cl-mpd/package.lisp
--- cl-mpd.orig/package.lisp	2006-06-26 20:58:46.000000000 -0500
+++ cl-mpd/package.lisp	2006-07-09 22:10:12.000000000 -0500
@@ -130,5 +130,4 @@
       #:mpd-send-command-list
       ))
 
-  (pushnew :mpd *features*)
   )
