diff -ur xfburn-0.4.2.orig/configure.in xfburn-0.4.2/configure.in
--- xfburn-0.4.2.orig/configure.in	2009-07-24 08:40:05.000000000 +0300
+++ xfburn-0.4.2/configure.in	2009-10-24 21:33:09.000000000 +0300
@@ -34,7 +34,8 @@
 XDT_CHECK_PACKAGE([GTHREAD], [gthread-2.0], [2.6.0])
 XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.10.0])
 XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.4.0])
-XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.0])
+XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.0], [],
+  [XDT_CHECK_PACKAGE([EXO], [exo-1], [0.5.0])])
 XDT_CHECK_OPTIONAL_PACKAGE([THUNAR_VFS], [thunar-vfs-1], [0.3.0], [thunar-vfs],
 	AC_HELP_STRING([--disable-thunar-vfs],
 			[Disable the use of Thunar's VFS layer to show mime type and icons in file browser]))
