--- ripperX-2.6.6/src/Makefile.in-orig	2006-06-26 22:53:42.000000000 -0500
+++ ripperX-2.6.6/src/Makefile.in	2006-06-26 22:54:38.000000000 -0500
@@ -11,7 +11,8 @@
 
 CC = @CC@
 CFLAGS = @CFLAGS@
-LDFLAGS = @LDFLAGS@ @LIBS@ -lutil
+LDFLAGS = @LDFLAGS@ 
+LIBS = @LIBS@ -lutil
 INSTALL = @INSTALL@
 
 CFILES = main.c\
@@ -35,7 +36,7 @@
 all: ripperX
 
 ripperX: $(OFILES)
-	$(CC) $(LDFLAGS) $(OFILES) -o ripperX
+	$(CC) $(LDFLAGS) $(OFILES) $(LIBS) -o ripperX
 
 .c.o:
 	$(CC) $(CFLAGS) -c $<
