diff -urN cproto-4.6-orig/cproto.c cproto-4.6/cproto.c
--- cproto-4.6-orig/cproto.c	1998-01-23 20:42:07.000000000 -0500
+++ cproto-4.6/cproto.c	2002-10-18 16:20:34.000000000 -0400
@@ -770,7 +770,7 @@
 		while (s != temp && s[-1] != '/')
 		    s--;
 		(void)strcpy(s, "XXXXXX.c");
-	    	mktemp(temp);
+	    	mkstemp(temp);
 	    	if (link(argv[i], temp) < 0)
 		    (void)strcpy(temp, argv[i]);
 	    }
@@ -790,7 +790,7 @@
 		 * the file that we're writing to.
 		 */
 		sprintf(cpp_cmd, cpp,
-			mktemp(strcpy(temp, "sys$scratch:XXXXXX.i")));
+			mkstemp(strcpy(temp, "sys$scratch:XXXXXX.i")));
 		sprintf(cpp_cmd + strlen(cpp_cmd), "%s %s", cpp_opt, FileName);
 		system(cpp_cmd);
 		inf = fopen(temp, "r");
diff -urN cproto-4.6-orig/lex_yy.c cproto-4.6/lex_yy.c
--- cproto-4.6-orig/lex_yy.c	1996-04-13 00:29:18.000000000 -0400
+++ cproto-4.6/lex_yy.c	2002-10-18 16:20:47.000000000 -0400
@@ -2859,7 +2859,7 @@
 	tmpdir = "/tmp";
     }
     sprintf(name, "%s/TfXXXXXX", tmpdir);
-    mktemp(name);
+    mkstemp(name);
 
     if ((f = fopen(name, "w+")) == (FILE *)0)
     {
