--- espmsg.c	2005-01-23 03:52:44.000000000 +0000
+++ espmsg.c	2010-06-01 14:06:06.983548480 +0000
@@ -270,14 +270,18 @@
 compile_messages(const char *filename)	/* I - Message filename */
 {
   int		i;			/* Looping var */
-  char		msgname[1024],		/* Message filename */
-		orig[1024],		/* Original message text */
-		text[1024],		/* Message text */
+  char 		*msgname,		/* Message filename */
+  		*orig, 			/* Original message text */
+		*text,			/* Message text */
 		*ptr;			/* Pointer into filename */
   int		num_msgs;		/* Number of messages */
   espmsg_t	*msgs;			/* Messages */
   message_t	*temp;			/* Current message */
 
+  msgname = (char *)malloc(4096);
+  orig = (char *)malloc(4096);
+  text = (char *)malloc(4096);
+
 
   num_msgs = 0;
   msgs     = NULL;
@@ -307,6 +311,10 @@
            strerror(errno));
   else
     printf("espmsg: Compiled %d messages into \"%s\".\n", num_msgs, msgname);
+
+  free(msgname);
+  free(orig);
+  free(text);
 }
 
 
