--- nessus-libraries/libnessus/scanners_utils.c~	2004-12-18 21:11:06.320320160 +0100
+++ nessus-libraries/libnessus/scanners_utils.c	2004-12-18 21:11:06.319320312 +0100
@@ -28,11 +28,11 @@
  */
 ExtFunc
 int 
-comm_send_status(globals, hostname, action, current,max)
+comm_send_status(globals, hostname, action, current_nessus,max)
   struct arglist * globals;
   char * hostname;
   char * action;
-  int current, max;
+  int current_nessus, max;
 {
  struct arglist * prefs = arg_get_value(globals,"preferences");
  char * pref = arg_get_value(prefs, "ntp_short_status");
@@ -61,16 +61,16 @@
     {
     if(short_status)
       {
-      snprintf(buffer, sizeof(buffer), "s:%c:%s:%d:%d\n", action[0], hostname, current, max);
+      snprintf(buffer, sizeof(buffer), "s:%c:%s:%d:%d\n", action[0], hostname, current_nessus, max);
       }
     else
      snprintf(buffer, sizeof(buffer),
 		"SERVER <|> STATUS <|> %s <|> %s <|> %d/%d <|> SERVER\n",
-		hostname, action, current, max);
+		hostname, action, current_nessus, max);
     }
   else
    snprintf(buffer, sizeof(buffer), "SERVER <|> STAT <|> %s <|> %d/%d <|> SERVER\n",
-	      hostname, current, max);
+	      hostname, current_nessus, max);
 	      
 	      
  internal_send(soc, buffer, INTERNAL_COMM_MSG_TYPE_DATA);
