diff -ur gtop-1.0.13/main.c gtop-1.0.13-gentoo/main.c
--- gtop-1.0.13/main.c	2003-09-21 12:04:02.000000000 +1200
+++ gtop-1.0.13-gentoo/main.c	2003-09-21 12:15:06.000000000 +1200
@@ -149,7 +149,7 @@
 	GNOMEUIINFO_END
 };
 
-#define ADD_PROPERTIES(x,y) gtop_property_object_list = g_list_append (gtop_property_object_list, gnome_property_object_new (& ## x ## Property_Descriptor, &gtop_properties. ## y ##))
+#define ADD_PROPERTIES(x,y) gtop_property_object_list = g_list_append (gtop_property_object_list, gnome_property_object_new (& x ## Property_Descriptor, &gtop_properties. y ))
 
 void
 gtop_quit (void)
diff -ur gtop-1.0.13/procview.c gtop-1.0.13-gentoo/procview.c
--- gtop-1.0.13/procview.c	2003-09-21 12:04:02.000000000 +1200
+++ gtop-1.0.13-gentoo/procview.c	2003-09-21 12:26:17.000000000 +1200
@@ -96,29 +96,29 @@
 \
 static int cmp_ ## NAME (const ProcProcData **P, const ProcProcData **Q) \
 { \
-	  return (*P)->d->sort_order * ((*P)-> ## NAME - (*Q)-> ## NAME); \
+	  return (*P)->d->sort_order * ((*P)-> NAME - (*Q)-> NAME); \
 }
 
 #define CMP_INT(NAME) \
 \
 static int cmp_ ## NAME (const ProcProcData **P, const ProcProcData **Q) \
 { \
-	  return (*P)->d->sort_order * ((*P)->p-> ## NAME - (*Q)->p-> ## NAME); \
+	  return (*P)->d->sort_order * ((*P)->p-> NAME - (*Q)->p-> NAME); \
 }
 
 #define CMP_INT2(NAME,V1,V2) \
 \
 static int cmp_ ## NAME (const ProcProcData **P, const ProcProcData **Q) \
 { \
-	  return (*P)->d->sort_order * ((*P)->p-> ## V1 + (*P)->p-> ## V2 - (\
-                               (*Q)->p-> ## V1 + (*Q)->p-> ## V2)); \
+	  return (*P)->d->sort_order * ((*P)->p-> V1 + (*P)->p-> V2 - (\
+                               (*Q)->p-> V1 + (*Q)->p-> V2)); \
 }
 
 #define CMP_STR(NAME) \
 \
 static int cmp_ ## NAME (const ProcProcData **P, const ProcProcData **Q) \
 { \
-	  return (*P)->d->sort_order * strcasecmp ((*P)->p-> ## NAME, (*Q)->p-> ## NAME); \
+	  return (*P)->d->sort_order * strcasecmp ((*P)->p-> NAME, (*Q)->p-> NAME); \
 }
 
 #define SORT_ASC 1
