--- src/Makefile.am	2006-12-21 09:50:16.000000000 -0600
+++ src/Makefile.am	2006-12-21 09:50:21.000000000 -0600
@@ -1,5 +1,5 @@
 
-CSC = mcs -debug
+CSC = gmcs -debug
 
 TARGET_NAME = tomboy-reminder
 TARGET = $(TARGET_NAME).dll
--- configure.in	2006-12-21 09:50:32.000000000 -0600
+++ configure.in	2006-12-21 09:51:22.000000000 -0600
@@ -5,11 +5,11 @@
 AC_PROG_INSTALL
 
 #
-# Find mcs
+# Find gmcs
 #
-AC_PATH_PROG(MCS, mcs)
-if test "x$MCS" = "x" ; then
-	AC_MSG_ERROR([Can't find "mcs", the mono compiler in your PATH])
+AC_PATH_PROG(GMCS, gmcs)
+if test "x$GMCS" = "x" ; then
+	AC_MSG_ERROR([Can't find "gmcs", the mono compiler in your PATH])
 fi
 
 #
