diff -Naur poco-1.3.2-data.orig/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h poco-1.3.2-data/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h
--- poco-1.3.2-data.orig/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h	2008-05-29 20:47:39.000000000 +0200
+++ poco-1.3.2-data/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h	2008-05-29 20:49:09.000000000 +0200
@@ -42,6 +42,7 @@
 
 #include "Poco/Data/ODBC/ODBC.h"
 #include <vector>
+#include <cstring>
 #ifdef POCO_OS_FAMILY_WINDOWS
 #include <windows.h>
 #endif
diff -Naur poco-1.3.2-data.orig/Foundation/include/Poco/FIFOStrategy.h poco-1.3.2-data/Foundation/include/Poco/FIFOStrategy.h
--- poco-1.3.2-data.orig/Foundation/include/Poco/FIFOStrategy.h	2008-05-29 20:47:40.000000000 +0200
+++ poco-1.3.2-data/Foundation/include/Poco/FIFOStrategy.h	2008-05-29 22:01:36.000000000 +0200
@@ -42,7 +42,7 @@
 #include "Poco/NotificationStrategy.h"
 #include <map>
 #include <list>
-
+#include <memory>
 
 namespace Poco {
 
diff -Naur poco-1.3.2-data.orig/Foundation/include/Poco/HashTable.h poco-1.3.2-data/Foundation/include/Poco/HashTable.h
--- poco-1.3.2-data.orig/Foundation/include/Poco/HashTable.h	2008-05-29 20:47:40.000000000 +0200
+++ poco-1.3.2-data/Foundation/include/Poco/HashTable.h	2008-05-29 22:08:33.000000000 +0200
@@ -47,7 +47,7 @@
 #include <vector>
 #include <map>
 #include <cstddef>
-
+#include <cstring>
 
 namespace Poco {
 
diff -Naur poco-1.3.2-data.orig/Foundation/src/BinaryWriter.cpp poco-1.3.2-data/Foundation/src/BinaryWriter.cpp
--- poco-1.3.2-data.orig/Foundation/src/BinaryWriter.cpp	2008-05-29 20:47:40.000000000 +0200
+++ poco-1.3.2-data/Foundation/src/BinaryWriter.cpp	2008-05-29 20:48:05.000000000 +0200
@@ -33,6 +33,7 @@
 // DEALINGS IN THE SOFTWARE.
 //
 
+#include <cstring>
 
 #include "Poco/BinaryWriter.h"
 #include "Poco/ByteOrder.h"
diff -Naur poco-1.3.2-data.orig/Foundation/src/File_UNIX.cpp poco-1.3.2-data/Foundation/src/File_UNIX.cpp
--- poco-1.3.2-data.orig/Foundation/src/File_UNIX.cpp	2008-05-29 20:47:40.000000000 +0200
+++ poco-1.3.2-data/Foundation/src/File_UNIX.cpp	2008-05-29 20:48:05.000000000 +0200
@@ -33,6 +33,7 @@
 // DEALINGS IN THE SOFTWARE.
 //
 
+#include <cstring>
 
 #include "Poco/File_UNIX.h"
 #include "Poco/Buffer.h"
