--- lipstik-2.1/style/config/lipstikconf.cpp	2005-12-09 19:55:51.000000000 +0300
+++ lipstik-2.1/style/config/lipstikconf.cpp	2006-01-02 20:35:06.000000000 +0300
@@ -81,6 +81,7 @@ LipstikStyleConfig::LipstikStyleConfig(Q
 //	newKickerButtons = new QCheckBox(i18n("Flat taskbar buttons"), generalGrp);
 	flatHeaders = new QCheckBox(i18n("Flatten Headers"), generalGrp);
 	tickCheckMark = new QCheckBox(i18n("Tick Checkmarks"), generalGrp);
+	lowerCaseText = new QCheckBox(i18n("Use lowercased text"), generalGrp);
 
 	tabLayout->addWidget(generalGrp,0,0,0);
 
@@ -338,6 +339,9 @@ LipstikStyleConfig::LipstikStyleConfig(Q
  	origTickCheckMark = s.readBoolEntry("/lipstikstyle/Settings/tickCheckMark", true);
 	tickCheckMark->setChecked(origTickCheckMark);
 
+	origLowerCaseText = s.readBoolEntry("/lipstikstyle/Settings/lowerCaseText", false);
+	lowerCaseText->setChecked(origLowerCaseText);
+
 	origDrawTriangularExpander = s.readBoolEntry("/lipstikstyle/Settings/drawTriangularExpander", false);
 	drawTriangularExpander->setChecked(origDrawTriangularExpander);
 
@@ -403,6 +407,7 @@ LipstikStyleConfig::LipstikStyleConfig(Q
  	connect(flatHeaders, SIGNAL( toggled(bool) ), SLOT( updateChanged() ) );
 
  	connect(tickCheckMark, SIGNAL( toggled(bool) ), SLOT( updateChanged() ) );
+	connect(lowerCaseText, SIGNAL( toggled(bool) ), SLOT( updateChanged() ) );
 
 	connect(drawTriangularExpander, SIGNAL( toggled(bool) ), SLOT( updateChanged() ) );
 	connect(inputFocusHighlight, SIGNAL( toggled(bool) ), SLOT( updateChanged() ) );
@@ -483,6 +488,7 @@ void LipstikStyleConfig::save()
 	s.writeEntry("/lipstikstyle/Settings/alterMenuGradients", alterMenuGradients->isChecked());
 	s.writeEntry("/lipstikstyle/Settings/menuLeftSideBar", menuLeftSideBar->isChecked());
 	s.writeEntry("/lipstikstyle/Settings/flatStripe", flatStripe->isChecked());
+	s.writeEntry("/lipstikstyle/Settings/lowerCaseText", lowerCaseText->isChecked());
 
 	s.writeEntry("/lipstikstyle/Settings/compressedMenus", compressedMenus->isChecked());
 
@@ -535,6 +541,7 @@ void LipstikStyleConfig::defaults()
 	alterMenuGradients->setChecked(false);
 	menuLeftSideBar->setChecked(true);
 	flatStripe->setChecked(false);
+	lowerCaseText->setChecked(false);
 
 	compressedMenus->setChecked(false);
 
@@ -649,6 +656,7 @@ void LipstikStyleConfig::updateChanged()
 	    (alterMenuGradients->isChecked() == origAlterMenuGradients) &&
 	    (menuLeftSideBar->isChecked() == origMenuLeftSideBar) &&
 	    (flatStripe->isChecked() == origFlatStripe) &&
+	    (lowerCaseText->isChecked() == origLowerCaseText) &&
 
 	    (compressedMenus->isChecked() == origCompressedMenus) &&
 
--- lipstik-2.1/style/config/lipstikconf.h	2005-12-09 19:49:42.000000000 +0300
+++ lipstik-2.1/style/config/lipstikconf.h	2006-01-02 20:21:20.000000000 +0300
@@ -83,6 +83,7 @@ protected:
 	QCheckBox* 	 kickerHighlight;
 	QCheckBox* 	 drawTabHighlight;
 	QCheckBox* 	 flatStripe;
+	QCheckBox* 	 lowerCaseText;
 	QCheckBox* 	 compressedMenus;
 
 	QCheckBox* 	 drawToolBarSeparator;
@@ -122,6 +123,7 @@ protected:
 	bool       origKickerHighlight;
 	bool       origDrawTabHighlight;
     	bool       origFlatStripe;
+    	bool       origLowerCaseText;
     	bool       origCompressedMenus;
 
 	bool       origDrawToolBarSeparator;
--- lipstik-2.1/style/lipstik.cpp	2005-12-10 15:24:24.000000000 +0300
+++ lipstik-2.1/style/lipstik.cpp	2006-01-02 20:29:03.000000000 +0300
@@ -192,6 +192,8 @@ LipstikStyle::LipstikStyle() : KStyle( A
     _customCheckMarkColor = settings.readBoolEntry("/customCheckMarkColor", false);
     _checkMarkColor.setNamedColor( settings.readEntry("/checkMarkColor", "black") );
 
+    _lowerCaseText = settings.readBoolEntry("/lowerCaseText", false);
+
     _drawStatusBarFrame = settings.readBoolEntry("/drawStatusBarFrame", false);
     _scrollBarStyle = settings.readEntry("/scrollBarStyle", "WindowsStyleScrollBar");
     _scrollBarWidth = settings.readEntry("/scrollBarWidth", "Normal");
@@ -2709,7 +2711,8 @@ void LipstikStyle::drawControl(ControlEl
 
             // Make the label indicate if the button is a default button or not
             drawItem( p, QRect(x, y, w, h), AlignCenter|ShowPrefix, button->colorGroup(),
-                        button->isEnabled(), button->pixmap(), button->text(), -1,
+                        button->isEnabled(), button->pixmap(),
+			_lowerCaseText ? button->text().lower() : button->text(), -1,
                         &button->colorGroup().buttonText() );
 
 
@@ -2745,7 +2748,7 @@ void LipstikStyle::drawControl(ControlEl
                             		_sharperMenuGradient?_contrast+30:_contrast+3, Draw_Top|Draw_Bottom|Is_Horizontal);
 		    }
     		    p->setPen(cg.highlightedText() );
-        	    p->drawText(r, text_flags, mi->text());
+        	    p->drawText(r, text_flags, _lowerCaseText ? mi->text().lower() : mi->text());
             	} else {
             	    if (down) {
                 	drawPrimitive(PE_ButtonTool, p, r, cg, flags|Style_Down, opt);
@@ -2753,11 +2756,11 @@ void LipstikStyle::drawControl(ControlEl
                 	drawPrimitive(PE_ButtonTool, p, r, cg, flags, opt);
             	    }
     		    p->setPen(cg.buttonText() );
-        	    p->drawText(r, text_flags, mi->text());
+        	    p->drawText(r, text_flags, _lowerCaseText ? mi->text().lower() : mi->text());
                 }
 	    } else {
     		    p->setPen(cg.buttonText() );
-        	    p->drawText(r, text_flags, mi->text());
+        	    p->drawText(r, text_flags, _lowerCaseText ? mi->text().lower() : mi->text());
 	    }
             break;
         }
@@ -2935,7 +2938,7 @@ void LipstikStyle::drawControl(ControlEl
             }
             else {
                 // The menu item doesn't draw it's own label
-                QString s = mi->text();
+                QString s = _lowerCaseText ? mi->text().lower() : mi->text();
                 // Does the menu item have a text label?
                 if ( !s.isNull() ) {
                     int t = s.find( '\t' );
--- lipstik-2.1-orig/style/lipstik.h	2005-12-09 19:45:48.000000000 +0300
+++ lipstik-2.1/style/lipstik.h	2006-01-02 20:21:20.000000000 +0300
@@ -290,6 +290,7 @@ private:
     bool _customOverHighlightColor;
     bool _customFocusHighlightColor;
     bool _customCheckMarkColor;
+    bool _lowerCaseText;
 
     bool _drawStatusBarFrame;
     bool _paintGroove;
