diff -ur pcmcia-cs-3.1.34.orig/cardmgr/cardmgr.c pcmcia-cs-3.1.34/cardmgr/cardmgr.c
--- pcmcia-cs-3.1.34.orig/cardmgr/cardmgr.c	Wed Jul  3 10:40:21 2002
+++ pcmcia-cs-3.1.34/cardmgr/cardmgr.c	Wed Jul  3 10:42:41 2002
@@ -1222,6 +1222,11 @@
     unlink(stabfile);
 }
 
+static void dummy(void)
+{
+
+}
+
 /*====================================================================*/
 
 /* most recent signal */
@@ -1473,7 +1478,7 @@
 	if (one_pass)
 	    exit(EXIT_SUCCESS);
 	if (delay_fork) {
-	    atexit(NULL);
+	    atexit(&dummy);
 	    fork_now();
 	    atexit(&done);
 	    write_pid();
