diff -ur pcmcia-cs-3.2.8/cardmgr/Makefile pcmcia-cs-3.2.8-ldflags/cardmgr/Makefile
--- pcmcia-cs-3.2.8/cardmgr/Makefile	2003-11-22 09:54:13.000000000 +0100
+++ pcmcia-cs-3.2.8-ldflags/cardmgr/Makefile	2005-04-13 01:36:37.000000000 +0200
@@ -73,21 +73,21 @@
 	@mkdir -p .depfiles ; mv $*.d .depfiles
 
 cardinfo: cardinfo.o
-	$(CC) $< -o $@ $(FLIBS)
+	$(CC) $< -o $@ $(LDFLAGS) $(FLIBS)
 
 xcardinfo.o: xcardinfo.c
 	$(CC) $(CFLAGS) -MD $(CPPFLAGS) $(XCPPFLAGS) -c $<
 	@mkdir -p .depfiles ; mv $*.d .depfiles
 
 xcardinfo: xcardinfo.o
-	$(CC) $< -o $@ -L/usr/X11R6/lib -L/usr/X11/lib -lX11 -lXaw
+	$(CC) $< -o $@ $(LDFLAGS) -L/usr/X11R6/lib -L/usr/X11/lib -lX11 -lXaw
 
 gpccard.o: gpccard.c
 	$(CC) -O -Wall -MD $(CPPFLAGS) $(GTK_CFLAGS) -c $<
 	@mkdir -p .depfiles ; mv $*.d .depfiles
 
 gpccard: gpccard.o
-	$(CC) $< -o $@ $(GTK_LIBS)
+	$(CC) $< -o $@ $(LDFLAGS) $(GTK_LIBS)
 
 install-cardinfo: cardinfo
 	@mkdir -p $(PREFIX)/usr/X11R6/bin
