diff -ur pcmcia-cs-3.2.8/Configure pcmcia-cs-3.2.8-includes/Configure
--- pcmcia-cs-3.2.8/Configure	2004-07-13 09:33:31.000000000 +0200
+++ pcmcia-cs-3.2.8-includes/Configure	2005-04-13 01:18:55.000000000 +0200
@@ -853,7 +853,7 @@
 
 if [ "$CONFIG_PCMCIA" = "y" ] ; then
     # Use our kernel config, then kernel headers, then our headers
-    CPPFLAGS="-I../include/static -I\$(LINUX)/include -I../include"
+    CPPFLAGS="-I../include/static -I../include -I\$(LINUX)/include"
 else
     # Use our kernel config and headers, then kernel headers
     CPPFLAGS="-I../include -I\$(LINUX)/include"
