--- configure.in
+++ configure.in
@@ -64,7 +64,8 @@
 dnl ***********************************
 dnl *** Check for required packages ***
 dnl ***********************************
-XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.1.3])
+XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.1.3], [],
+  [XDT_CHECK_PACKAGE([EXO], [exo-1], [0.5.0])])
 XDT_CHECK_PACKAGE([THUNARX], [thunarx-1], [0.2.3svn-r20526])
 XDT_CHECK_PACKAGE([THUNARVFS], [thunar-vfs-1], [0.2.3svn-r20526])
 XDT_CHECK_PACKAGE([TAGLIB], [taglib], [1.4])
