From 8df3e82b02a980d4e00f3dc9c90c11e96501cf6f Mon Sep 17 00:00:00 2001
From: Manuel Nickschas <sputnick@quassel-irc.org>
Date: Tue, 19 Jan 2010 21:51:41 +0100
Subject: [PATCH] Don't require X11 headers for building quasselcore

---
 cmake/modules/Qt4ConfigDependentSettings.cmake |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/cmake/modules/Qt4ConfigDependentSettings.cmake b/cmake/modules/Qt4ConfigDependentSettings.cmake
index 1956aa5..a6b6c3d 100644
--- a/cmake/modules/Qt4ConfigDependentSettings.cmake
+++ b/cmake/modules/Qt4ConfigDependentSettings.cmake
@@ -27,7 +27,7 @@
 IF(UNIX)
   # on OS X X11 may not be required
   IF (Q_WS_X11)
-    FIND_PACKAGE(X11 REQUIRED)
+    FIND_PACKAGE(X11)
   ENDIF (Q_WS_X11)
   FIND_PACKAGE(Threads)
   SET(QT_QTCORE_LIBRARY ${QT_QTCORE_LIBRARY} ${CMAKE_THREAD_LIBS_INIT})
-- 
1.6.5.1

