diff -Nur gtk+-2.4.9.orig/gtk/gtkmarshal.c gtk+-2.4.9/gtk/gtkmarshal.c
--- gtk+-2.4.9.orig/gtk/gtkmarshal.c	2004-12-26 02:20:42.371982784 +0900
+++ gtk+-2.4.9/gtk/gtkmarshal.c	2004-12-26 02:23:34.879984760 +0900
@@ -35,8 +35,8 @@
 #define g_marshal_value_peek_ulong(v)    (v)->data[0].v_ulong
 #define g_marshal_value_peek_int64(v)    (v)->data[0].v_int64
 #define g_marshal_value_peek_uint64(v)   (v)->data[0].v_uint64
-#define g_marshal_value_peek_enum(v)     (v)->data[0].v_long
-#define g_marshal_value_peek_flags(v)    (v)->data[0].v_ulong
+#define g_marshal_value_peek_enum(v)     (v)->data[0].v_int
+#define g_marshal_value_peek_flags(v)    (v)->data[0].v_uint
 #define g_marshal_value_peek_float(v)    (v)->data[0].v_float
 #define g_marshal_value_peek_double(v)   (v)->data[0].v_double
 #define g_marshal_value_peek_string(v)   (v)->data[0].v_pointer
diff -Nur gtk+-2.4.9.orig/gtk/gtkmarshalers.c gtk+-2.4.9/gtk/gtkmarshalers.c
--- gtk+-2.4.9.orig/gtk/gtkmarshalers.c	2004-12-26 02:20:42.372982632 +0900
+++ gtk+-2.4.9/gtk/gtkmarshalers.c	2004-12-26 02:25:57.154922896 +0900
@@ -35,8 +35,8 @@
 #define g_marshal_value_peek_ulong(v)    (v)->data[0].v_ulong
 #define g_marshal_value_peek_int64(v)    (v)->data[0].v_int64
 #define g_marshal_value_peek_uint64(v)   (v)->data[0].v_uint64
-#define g_marshal_value_peek_enum(v)     (v)->data[0].v_long
-#define g_marshal_value_peek_flags(v)    (v)->data[0].v_ulong
+#define g_marshal_value_peek_enum(v)     (v)->data[0].v_int
+#define g_marshal_value_peek_flags(v)    (v)->data[0].v_uint
 #define g_marshal_value_peek_float(v)    (v)->data[0].v_float
 #define g_marshal_value_peek_double(v)   (v)->data[0].v_double
 #define g_marshal_value_peek_string(v)   (v)->data[0].v_pointer
