diff -NurdB gtk+-2.8.0/configure.in gtk+-2.8.0-patched/configure.in
--- gtk+-2.8.0/configure.in	2005-08-13 07:51:54.000000000 -0500
+++ gtk+-2.8.0-patched/configure.in	2005-08-18 16:28:14.000000000 -0500
@@ -1023,8 +1023,8 @@
 GDK_PIXBUF_XLIB_PACKAGES=
 GDK_PIXBUF_XLIB_EXTRA_CFLAGS=
 GDK_PIXBUF_XLIB_EXTRA_LIBS=
-
-X_PACKAGES=
+	
+X_PACKAGES=fontconfig
 GDK_EXTRA_LIBS="$GDK_WLIBS"
 GDK_EXTRA_CFLAGS=
 	
@@ -1034,11 +1034,20 @@
 
 if test "x$gdktarget" = "xx11"; then
   #
+  # We use fontconfig very peripherally when decoding the default
+  # settings.
+  #
+  if $PKG_CONFIG --exists fontconfig; then : ; else
+    AC_MSG_ERROR([
+*** fontconfig (http://www.fontconfig.org) is required by the X11 backend.])
+  fi
+
+  #
   # Check for basic X packages; we use pkg-config if available
   #
   if $PKG_CONFIG --exists x11 xext; then
     have_base_x_pc=true
-    X_PACKAGES="x11 xext"
+    X_PACKAGES="$X_PACKAGES x11 xext"
     x_libs="`pkg-config --libs x11 xext`"
     
     # Strip out any .la files that pkg-config might give us (this happens
@@ -1285,7 +1294,7 @@
     X_PACKAGES="$X_PACKAGES xfixes"
   fi
 
-  if $have_base_pc ; then
+  if $have_base_x_pc ; then
     GDK_EXTRA_LIBS="$x_extra_libs"
   else
     GDK_EXTRA_LIBS="$X_LIBS $x_extra_libs -lX11 $GDK_EXTRA_LIBS"
diff -NurdB gtk+-2.8.0/gdk/x11/gdkdrawable-x11.c gtk+-2.8.0-patched/gdk/x11/gdkdrawable-x11.c
--- gtk+-2.8.0/gdk/x11/gdkdrawable-x11.c	2005-07-20 22:57:00.000000000 -0500
+++ gtk+-2.8.0-patched/gdk/x11/gdkdrawable-x11.c	2005-08-18 16:29:26.000000000 -0500
@@ -29,8 +29,6 @@
 #include "gdkx.h"
 #include "gdkregion-generic.h"
 
-#include <pango/pangoxft.h>
-
 #include <cairo-xlib.h>
 
 #include <stdlib.h>
