diff -Naur qtella-0.7.0.vanilla/src/QtellaSubConfig.cpp qtella-0.7.0/src/QtellaSubConfig.cpp
--- qtella-0.7.0.vanilla/src/QtellaSubConfig.cpp	2004-05-26 16:32:00.000000000 -0500
+++ qtella-0.7.0/src/QtellaSubConfig.cpp	2004-07-01 21:37:51.156045376 -0500
@@ -519,7 +519,7 @@
             for(i = 0; i < vString.size(); i++)
               _widget_config->ui_blocked_words->insertItem( vString[i].c_str() );
 		  }
-		  
+				#ifdef USEKDE
 				if( s1 == "YID" ) {
 					_widget_config->lineedit_yahooid->setText( s2.c_str() );
 				}
@@ -529,6 +529,7 @@
 				if( s1 == "YCONNECT" ) {
 					_widget_config->checkbox_yahooconnect->setChecked( ( s2 == "1" ) ? true : false );
 				}
+				#endif
 			}
 		}
 		f.close();
@@ -632,11 +633,12 @@
     << "GROUP=" << (const char*)_widget_config->ui_lineedit_userid->text() << std::endl
     << "RESERVE_IN=" << _widget_config->ui_checkbox_reserve_in->isChecked() << std::endl
     << "RESERVE_UP=" << _widget_config->ui_checkbox_reserve_up->isChecked() << std::endl;
-  
+  #ifdef USEKDE
 	QCString result = KCodecs::base64Encode( _widget_config->lineedit_yahoopasswd->text().ascii() );
 	f << "YID=" << (const char*) _widget_config->lineedit_yahooid->text() << std::endl
 		<< "YPWD=" << (const char*) result << std::endl
 		<< "YCONNECT=" << ( _widget_config->checkbox_yahooconnect->isChecked() ? "1" : "0" ) << std::endl;
+  #endif
 	f << "GROUPS=";
 	for( unsigned i = 0; i < _widget_config->ui_listbox_groups->count(); ++i ) {
 		f << (const char*)_widget_config->ui_listbox_groups->text( i ) << ";";
diff -Naur qtella-0.7.0.vanilla/src/main.cpp qtella-0.7.0/src/main.cpp
--- qtella-0.7.0.vanilla/src/main.cpp	2004-05-20 11:14:13.000000000 -0500
+++ qtella-0.7.0/src/main.cpp	2004-07-01 21:09:23.000000000 -0500
@@ -73,8 +73,10 @@
 
 	a.setMainWidget(&qtella);
 	qtella.show();
+	#ifdef USEKDE
 	KCmdLineArgs* args = KCmdLineArgs::parsedArgs();
 	if( args->isSet( "show-searches" ) )
 		qtella.setSearchLogging( true );
+	#endif
 	return a.exec();
 }
