diff -Naur qtella-0.7.0.vanilla/src/MessengerThread.cpp qtella-0.7.0/src/MessengerThread.cpp
--- qtella-0.7.0.vanilla/src/MessengerThread.cpp	2004-05-27 22:02:18.000000000 +0200
+++ qtella-0.7.0/src/MessengerThread.cpp	2005-01-06 11:52:37.000000000 +0100
@@ -10,6 +10,7 @@
 #include <yahoo2.h>
 #include <yahoo2_callbacks.h>
 #include <yahoo_list.h>
+#include <errno.h>
 
 MessengerContacts* MessengerThread::_messenger_contacts = NULL;
 
