diff -ru tinyerp-server-4.0.1~/setup.py tinyerp-server-4.0.1/setup.py
--- tinyerp-server-4.0.1~/setup.py	2006-12-23 17:42:08.000000000 +0100
+++ tinyerp-server-4.0.1/setup.py	2006-12-23 17:47:59.000000000 +0100
@@ -89,6 +89,10 @@
               ['man/terp_serverrc.5']),
              (opj('share','doc', 'tinyerp-server-%s' % version), 
               [f for f in glob.glob('doc/*') if os.path.isfile(f)]),
+             (opj('share','doc', 'tinyerp-server-%s' % version, 'migrate', '3.3.0-3.4.0'), 
+              [f for f in glob.glob('doc/migrate/3.3.0-3.4.0/*') if os.path.isfile(f)]),
+             (opj('share','doc', 'tinyerp-server-%s' % version, 'migrate', '3.4.0-4.0.0'), 
+              [f for f in glob.glob('doc/migrate/3.4.0-4.0.0/*') if os.path.isfile(f)]),
              (opj('lib','python%s' % py_short_version, 'site-packages', 'tinyerp-server', 'i18n'), 
               glob.glob('bin/i18n/*')),
              (opj('lib', 'python%s' % py_short_version, 'site-packages', 'tinyerp-server', 'addons', 'custom'),
