diff -ru tinyerp-server-4.0.3~/setup.py tinyerp-server-4.0.3/setup.py
--- tinyerp-server-4.0.3~/setup.py	2007-03-17 12:33:46.000000000 +0100
+++ tinyerp-server-4.0.3/setup.py	2007-03-17 12:34:02.000000000 +0100
@@ -159,7 +159,6 @@
                           'tinyerp-server.osv',
                           'tinyerp-server.ssl',
                           'tinyerp-server.service', 'tinyerp-server.tools',
-                          'tinyerp-server.pychart', 'tinyerp-server.pychart.afm',
                           'tinyerp-server.report',
                           'tinyerp-server.report.printscreen',
                           'tinyerp-server.report.render',
