diff -urN thelove@canonical.com---dists--bazaar--1.4/src/baz/libarch/pfs.c thelove@canonical.com---dists--bazaar--1.4-neon-0.26-fix/src/baz/libarch/pfs.c
--- thelove@canonical.com---dists--bazaar--1.4/src/baz/libarch/pfs.c	2005-06-20 23:59:37.000000000 +0200
+++ thelove@canonical.com---dists--bazaar--1.4-neon-0.26-fix/src/baz/libarch/pfs.c	2006-07-21 23:46:47.000000000 +0200
@@ -513,10 +513,10 @@
     char *at_pos = str_chr_index (parsed_uri->host, '@');
     if (!at_pos)
         return;
-    parsed_uri->authinfo = str_replace (parsed_uri->authinfo, 
-					str_alloc_cat (0, parsed_uri->authinfo, "@"));
-    parsed_uri->authinfo = str_replace (parsed_uri->authinfo, 
-					str_alloc_cat_n (0, parsed_uri->authinfo, parsed_uri->host, at_pos - parsed_uri->host));
+    parsed_uri->userinfo = str_replace (parsed_uri->userinfo, 
+					str_alloc_cat (0, parsed_uri->userinfo, "@"));
+    parsed_uri->userinfo = str_replace (parsed_uri->userinfo, 
+					str_alloc_cat_n (0, parsed_uri->userinfo, parsed_uri->host, at_pos - parsed_uri->host));
     parsed_uri->host = str_replace (parsed_uri->host, str_save (0, at_pos + 1));
 }
 
diff -urN thelove@canonical.com---dists--bazaar--1.4/src/baz/libarch/pfs-sftp.c thelove@canonical.com---dists--bazaar--1.4-neon-0.26-fix/src/baz/libarch/pfs-sftp.c
--- thelove@canonical.com---dists--bazaar--1.4/src/baz/libarch/pfs-sftp.c	2005-06-20 23:59:37.000000000 +0200
+++ thelove@canonical.com---dists--bazaar--1.4-neon-0.26-fix/src/baz/libarch/pfs-sftp.c	2006-07-21 23:37:22.000000000 +0200
@@ -1496,7 +1496,7 @@
    */
   arch_uri_heuristics (&parsed_uri);
 
-  *user = str_save (0, parsed_uri.authinfo);
+  *user = str_save (0, parsed_uri.userinfo);
   *hostname = str_save (0, parsed_uri.host);
   if (parsed_uri.port)
     {
diff -urN thelove@canonical.com---dists--bazaar--1.4/src/baz/libarch/tests/unit-sftp.c thelove@canonical.com---dists--bazaar--1.4-neon-0.26-fix/src/baz/libarch/tests/unit-sftp.c
--- thelove@canonical.com---dists--bazaar--1.4/src/baz/libarch/tests/unit-sftp.c	2005-06-20 23:59:37.000000000 +0200
+++ thelove@canonical.com---dists--bazaar--1.4-neon-0.26-fix/src/baz/libarch/tests/unit-sftp.c	2006-07-21 23:39:07.000000000 +0200
@@ -35,7 +35,7 @@
     char *host;
     unsigned int port;
     char *path;
-    char *authinfo;
+    char *userinfo;
 } ne_uri;
 */
   invariant_int_cmp (ne_uri_parse ("sftp://user@email.com@host.phwoar/root/path", &parsed_uri), 0);
@@ -44,7 +44,7 @@
   invariant_str_cmp (parsed_uri.host, "email.com@host.phwoar");
   arch_uri_heuristics (&parsed_uri);
   invariant_str_cmp (parsed_uri.host, "host.phwoar");
-  invariant_str_cmp (parsed_uri.authinfo, "user@email.com");
+  invariant_str_cmp (parsed_uri.userinfo, "user@email.com");
   invariant_int_cmp (parsed_uri.port, 0);
   ne_uri_free(&parsed_uri);
 
