diff --exclude-from=/home/dang/.diffrc -u -ruN gnomemeeting-1.2.2.orig/Makefile.am gnomemeeting-1.2.2/Makefile.am
--- gnomemeeting-1.2.2.orig/Makefile.am	2004-12-02 11:45:19.000000000 -0500
+++ gnomemeeting-1.2.2/Makefile.am	2005-09-13 15:54:09.000000000 -0400
@@ -37,7 +37,8 @@
 	gconftool-2 --shutdown
 
 install-services: gnomemeeting.service
-	cp gnomemeeting.service $(libdir)/dbus-1.0/services/
+	mkdir -p $(DESTDIR)$(libdir)/dbus-1.0/services/
+	cp gnomemeeting.service $(DESTDIR)$(libdir)/dbus-1.0/services/
 
 if GCONF_SCHEMAS_INSTALL
 install-data-local: install-schemas 
diff --exclude-from=/home/dang/.diffrc -u -ruN gnomemeeting-1.2.2.orig/configure.in gnomemeeting-1.2.2/configure.in
--- gnomemeeting-1.2.2.orig/configure.in	2005-09-05 06:35:30.000000000 -0400
+++ gnomemeeting-1.2.2/configure.in	2005-09-13 15:52:48.000000000 -0400
@@ -424,7 +424,7 @@
 fi
 
 if test x"${enable_dbus}" = xyes ; then
-   PKG_CHECK_MODULES(DBUS,  dbus-1 = 0.22 dbus-glib-1 = 0.22)
+   PKG_CHECK_MODULES(DBUS,  dbus-1 >= 0.36 dbus-glib-1 >= 0.36)
    GNOMEMEETING_CFLAGS="$GNOMEMEETING_CFLAGS $DBUS_CFLAGS -DHAS_DBUS"
    GNOMEMEETING_LIBS="$GNOMEMEETING_LIBS $DBUS_LIBS"
 fi
diff --exclude-from=/home/dang/.diffrc -u -ruN gnomemeeting-1.2.2.orig/src/dbus_component.cpp gnomemeeting-1.2.2/src/dbus_component.cpp
--- gnomemeeting-1.2.2.orig/src/dbus_component.cpp	2005-04-06 16:10:34.000000000 -0400
+++ gnomemeeting-1.2.2/src/dbus_component.cpp	2005-09-13 15:52:48.000000000 -0400
@@ -346,7 +346,7 @@
     
     if (self->owns_the_service == FALSE)
       self->owns_the_service 
-	= (dbus_bus_acquire_service (self->connection, 
+	= (dbus_bus_request_name (self->connection, 
 				     GM_DBUS_SERVICE, 0, NULL) >= 0);
     
  
@@ -465,7 +465,7 @@
 			DBUS_HANDLER_RESULT_NOT_YET_HANDLED);
 
   if (dbus_message_is_signal (message,
-                              DBUS_INTERFACE_ORG_FREEDESKTOP_LOCAL,
+                              DBUS_INTERFACE_LOCAL,
                               "Disconnected"))
     {
 
