diff -ur Recycler.old/Recycler.m Recycler/Recycler.m
--- Recycler.old/Recycler.m	2004-09-21 20:00:27.000000000 -0400
+++ Recycler/Recycler.m	2004-10-05 11:58:44.337684960 -0400
@@ -28,10 +28,10 @@
 #include "RecyclerView.h"
 #include "Preferences/RecyclerPrefs.h"
 #include "Dialogs/StartAppWin.h"
-#include "FSNode.h"
-#include "FSNodeRep.h"
-#include "FSNFunctions.h"
-#include "GNUstep.h"
+#include "FSNode/FSNode.h"
+#include "FSNode/FSNodeRep.h"
+#include "FSNode/FSNFunctions.h"
+//#include "GNUstep.h"
 
 
 static Recycler *recycler = nil;
diff -ur Recycler.old/RecyclerIcon.h Recycler/RecyclerIcon.h
--- Recycler.old/RecyclerIcon.h	2004-06-15 06:20:26.000000000 -0400
+++ Recycler/RecyclerIcon.h	2004-10-05 11:58:59.077444176 -0400
@@ -26,7 +26,7 @@
 #define RECYCLER_ICON_H
 
 #include <AppKit/NSView.h>
-#include "FSNIcon.h"
+#include "FSNode/FSNIcon.h"
 
 @class NSColor;
 @class NSImage;
diff -ur Recycler.old/RecyclerView.h Recycler/RecyclerView.h
--- Recycler.old/RecyclerView.h	2004-09-02 07:31:38.000000000 -0400
+++ Recycler/RecyclerView.h	2004-10-05 11:58:15.332094480 -0400
@@ -27,7 +27,7 @@
 
 #include <AppKit/NSView.h>
 #include "Recycler.h"
-#include "FSNodeRep.h"
+#include "FSNode/FSNodeRep.h"
 
 @class NSImage;
 @class RecyclerIcon;
diff -ur Recycler.old/RecyclerView.m Recycler/RecyclerView.m
--- Recycler.old/RecyclerView.m	2004-09-21 20:00:27.000000000 -0400
+++ Recycler/RecyclerView.m	2004-10-05 11:59:18.241530792 -0400
@@ -27,8 +27,8 @@
 #include <math.h>
 #include "RecyclerView.h"
 #include "RecyclerIcon.h"
-#include "FSNFunctions.h"
-#include "GNUstep.h"
+#include "FSNode/FSNFunctions.h"
+//#include "GNUstep.h"
 
 #define WIN_SIZE 64
 #define ICN_SIZE 48
