--- iproute-20010824.orig/tc/tc_qdisc.c
+++ iproute-20010824/tc/tc_qdisc.c
@@ -245,7 +245,7 @@
 	fprintf(fp, "\n");
 	if (show_stats) {
 		if (tb[TCA_STATS]) {
-			if (RTA_PAYLOAD(tb[TCA_STATS]) < sizeof(struct tc_stats))
+			if (RTA_PAYLOAD(tb[TCA_STATS]) < offsetof(struct tc_stats,backlog)+sizeof(((struct tc_stats*)0)->backlog))
 				fprintf(fp, "statistics truncated");
 			else {
 				struct tc_stats st;
--- iproute-20010824.orig/include-glibc/glibc-bugs.h
+++ iproute-20010824/include-glibc/glibc-bugs.h
@@ -12,9 +12,6 @@
 
 #include <linux/types.h>
 
-typedef __u16 in_port_t;
-typedef __u32 in_addr_t;
-
 #endif
 
 #endif
