diff -Naurp lilypond-2.12.2-orig/lily/relocate.cc lilypond-2.12.2/lily/relocate.cc
--- lilypond-2.12.2-orig/lily/relocate.cc	2009-01-18 18:04:57.000000000 -0600
+++ lilypond-2.12.2/lily/relocate.cc	2009-07-29 14:31:12.908541341 -0600
@@ -276,13 +276,13 @@ expand_environment_variables (string ori
   string out;
   while (ptr < start_ptr + len)
     {
-      char *dollar = strchr (ptr, '$');
+      const char *dollar = strchr (ptr, '$');
       
       if (dollar != NULL)
 	{
-	  char *start_var = dollar + 1;
-	  char *end_var = start_var;
-	  char *start_next = end_var;
+	  const char *start_var = dollar + 1;
+	  const char *end_var = start_var;
+	  const char *start_next = end_var;
 	  
 	  out += string (ptr, dollar - ptr);
 	  ptr = dollar;
