--- lilypond-2.12.2.orig/python/GNUmakefile	2009-01-18 22:04:57.000000000 -0200
+++ lilypond-2.12.2/python/GNUmakefile	2009-12-10 12:28:27.000000000 -0300
@@ -16,7 +16,7 @@
 INSTALLATION_OUT_DIR=$(local_lilypond_libdir)/python
 
 INSTALLATION_OUT_DIR1=$(local_lilypond_datadir)/python
-INSTALLATION_OUT_FILES1=$(OUT_PY_MODULES) $(OUT_PYC_MODULES)
+INSTALLATION_OUT_FILES1=$(OUT_PY_MODULES)
 
 $(outdir)/midi.lo: $(outdir)/config.hh
 
