--- spcplay/Makefile.in.bak	2003-03-06 10:13:09.000000000 +0100
+++ spcplay/Makefile.in	2003-03-06 10:13:33.000000000 +0100
@@ -108,7 +108,7 @@
 LIBS = @LIBS@
 spcplay_OBJECTS =  getopt.o getopt1.o main.o
 spcplay_DEPENDENCIES =  ../libspc/libspc.la
-spcplay_LDFLAGS = 
+spcplay_LDFLAGS = -lstdc++
 CFLAGS = @CFLAGS@
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 LTCOMPILE = $(LIBTOOL) --mode=compile $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
--- xmms-plugin/Makefile.in.bak	2003-03-06 10:17:06.000000000 +0100
+++ xmms-plugin/Makefile.in	2003-03-06 10:17:23.000000000 +0100
@@ -93,7 +93,7 @@
 
 lib_LTLIBRARIES = libspc.la
 libspc_la_LDFLAGS = -module -avoid-version
-libspc_la_LIBADD = @GTK_LIBS@ @PTHREAD_LIBS@ ../libspc/libspc.la
+libspc_la_LIBADD = @GTK_LIBS@ -lstdc++ @PTHREAD_LIBS@ ../libspc/libspc.la
 libspc_la_SOURCES = spcxmms.c spcxmms.h 	interface.c interface.h support.c support.h callbacks.h gui.c
 
