--- mlt++-20060601/configure.old	2006-11-11 17:16:13.000000000 +0100
+++ mlt++-20060601/configure	2006-11-11 17:16:30.000000000 +0100
@@ -37,13 +37,13 @@
 		echo LIBSUF=.dylib
         echo "CXXFLAGS+=-D__DARWIN__ -Wall -fPIC `mlt-config --cflags`"
 		echo "LIBFLAGS=-dynamiclib -single_module"
-        echo "LDFLAGS+=`mlt-config --libs` `mlt-config miracle --libs` `mlt-config valerie --libs`"
+        echo "OURLDFLAGS=`mlt-config --libs` `mlt-config miracle --libs` `mlt-config valerie --libs`"
 		;;
 	Linux)
 		echo LIBSUF=.so
         echo "CXXFLAGS+=-pthread -Wall -fPIC `mlt-config --cflags`"
 		echo "LIBFLAGS=-shared"
-        echo "LDFLAGS+=`mlt-config --libs` `mlt-config miracle --libs`"
+        echo "OURLDFLAGS=`mlt-config --libs` `mlt-config miracle --libs`"
 		;;
 esac >> config.mak
 
--- mlt++-20060601/src/Makefile.old	2006-11-11 17:22:39.000000000 +0100
+++ mlt++-20060601/src/Makefile	2006-11-11 17:23:10.000000000 +0100
@@ -39,7 +39,7 @@
 all:		$(TARGET)
 
 $(TARGET):	$(OBJS)
-	$(CXX) $(LIBFLAGS) -o $@ $(OBJS) $(LDFLAGS)
+	$(CXX) $(LIBFLAGS) $(LDFLAGS) -o $@ $(OBJS) $(OURLDFLAGS)
 	ln -sf $(TARGET) $(NAME)
 
 clean:
--- mlt++-20060601/test/Makefile.old	2006-11-11 17:24:47.000000000 +0100
+++ mlt++-20060601/test/Makefile	2006-11-11 17:25:30.000000000 +0100
@@ -1,16 +1,18 @@
 include ../config.mak
 
 CXXFLAGS+=-Wall -g `mlt-config --cflags` -I ../src
-LDFLAGS+=-L../src -lmlt++ -lmiracle -lvalerie `mlt-config --libs`
+OURLDFLAGS+=-L../src -lmlt++ -lmiracle -lvalerie `mlt-config --libs`
 CC=c++
 
 all:		play server
 
 play:		play.o
+	$(CC) $(LDFLAGS) -o $@ play.o $(OURLDFLAGS)
 
 play.o:		play.cpp
 
 server:		server.o
+	$(CC) $(LDFLAGS) -o $@ server.o $(OURLDFLAGS)
 
 server.o:	server.cpp
 
