diff -Naur gdl-0.9rc2/configure.in gdl-0.9rc2.new/configure.in
--- gdl-0.9rc2/configure.in	2009-01-22 17:31:48.000000000 -0500
+++ gdl-0.9rc2.new/configure.in	2009-08-25 18:38:10.000000000 -0400
@@ -755,7 +755,7 @@
 dnl perform program name transformation
 dnl AC_ARG_PROGRAM
 
-AC_CONFIG_FILES(Makefile src/Makefile src/antlr/Makefile)
+AC_CONFIG_FILES(Makefile src/Makefile)
 
 AC_OUTPUT
 
diff -Naur gdl-0.9rc2/src/Makefile.am gdl-0.9rc2.new/src/Makefile.am
--- gdl-0.9rc2/src/Makefile.am	2009-01-20 14:33:56.000000000 -0500
+++ gdl-0.9rc2.new/src/Makefile.am	2009-08-25 21:36:55.000000000 -0400
@@ -1,6 +1,6 @@
 # set the include path found by configure
 INCLUDES = $(EXT_INCLUDES) $(all_includes)
-SUBDIRS = antlr
+SUBDIRS = 
 noinst_HEADERS = gdlwidget.hpp prognode.hpp prognodeexpr.hpp
 
 cpp_hpp_files = accessdesc.hpp arrayindex.hpp assocdata.cpp assocdata.hpp \
@@ -53,7 +53,7 @@
 gdl_CXXFLAGS = $(AM_CXXFLAGS)
 # the library search path.
 gdl_LDFLAGS = $(all_libraries)
-gdl_LDADD = $(top_builddir)/src/antlr/libantlr.a -L/usr/X11R6/lib64 -L/usr/X11R6/lib -lX11
+gdl_LDADD = -lantlr -L/usr/X11R6/lib64 -L/usr/X11R6/lib -lX11
 endif
 
 # build as library (for python module)
@@ -69,6 +69,6 @@
 libgdl_la_LDFLAGS = $(all_libraries) -Wl,-z,muldefs
 endif 
 
-libgdl_la_LIBADD = $(top_builddir)/src/antlr/libantlr.la -L/usr/X11R6/lib64 -L/usr/X11R6/lib -lX11
+libgdl_la_LIBADD = -L/usr/X11R6/lib64 -L/usr/X11R6/lib -lX11
 endif
 
