--- chkrootkit-0.42b/chkrootkit	2003-09-20 23:54:11.000000000 +0800
+++ chkrootkit	2003-10-05 15:30:39.000000000 +0800
@@ -10,6 +10,14 @@
 # (C)1997-2003 Nelson Murilo, Pangeia Informatica, AMS Foundation and others.
 # All rights reserved
 
+# Gentoo specific : Could use `type <command> | cut -f 3 -d " "`
+IFPROMISC="/usr/sbin/ifpromisc"
+CHKLASTLOG="/usr/sbin/chklastlog"
+CHKPROC="/usr/sbin/chkproc"
+CHKWTMP="/usr/sbin/chkwtmp"
+CHECK_WTMPX="/usr/sbin/check_wtmpx"
+STRINGS="/usr/sbin/strings-static"
+
 ### workaround for some Bourne shell implementations
 unalias login > /dev/null 2>&1
 unalias ls > /dev/null 2>&1
@@ -116,7 +124,7 @@
 
     if [ "${EXPERT}" = "t" ]; then
         expertmode_output "${egrep} ^asp ${ROOTDIR}etc/inetd.conf"
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
@@ -132,7 +140,7 @@
         STATUS=${INFECTED}
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${ASP_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${ASP_LABEL}" >/dev/null 2>&1
        then
           echo "INFECTED"
           STATUS=${INFECTED}
@@ -151,15 +159,15 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "./ifpromisc"
+        expertmode_output "${IFPROMISC}"
         return 5
     fi
     if [ ! -f ${ROOTDIR}proc/net/packet ]; then
-       if [ ! -x ./ifpromisc ]; then
-         echo "not tested: can't exec ./ifpromisc"
+       if [ ! -x ${IFPROMISC} ]; then
+         echo "not tested: can't exec ${IFPROMISC}"
          return ${NOT_TESTED}
        fi
-       [ "${QUIET}" != "t" ] && ./ifpromisc || ./ifpromisc -q
+       [ "${QUIET}" != "t" ] && ${IFPROMISC} || ${IFPROMISC} -q
     else
        if [ `${egrep} -c "3    0003" ${ROOTDIR}proc/net/packet 2>/dev/null` -gt 0 ]; then
 	       set `${egrep} ":" $ROOTDIR/proc/net/dev | ${egrep} -v "lo:" | cut -f 1 -d:`
@@ -174,8 +182,8 @@
 }
 
 z2 () {
-    if [ ! -x ./chklastlog ]; then
-      echo "not tested: can't exec ./chklastlog"
+    if [ ! -x ${CHKLASTLOG} ]; then
+      echo "not tested: can't exec ${CHKLASTLOG}"
       return ${NOT_TESTED}
     fi
 
@@ -183,31 +191,31 @@
     LASTLOG=`loc lastlog lastlog "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "./chklastlog -f ${WTMP} -l ${LASTLOG}"
+        expertmode_output "${CHKLASTLOG} -f ${WTMP} -l ${LASTLOG}"
         return 5
     fi
 
-    if ./chklastlog -f ${WTMP} -l ${LASTLOG}
+    if ${CHKLASTLOG} -f ${WTMP} -l ${LASTLOG}
     then
       if [ "${QUIET}" != "t" ]; then echo "nothing deleted"; fi
     fi
 }
 
 wted () {
-    if [ ! -x ./chkwtmp ]; then
-      echo "not tested: can't exec ./chkwtmp"
+    if [ ! -x ${CHKWTMP} ]; then
+      echo "not tested: can't exec ${CHKWTMP}"
       return ${NOT_TESTED}
     fi
 
    if [ "$SYSTEM" = "SunOS" ]; then
-       if [ ! -x ./check_wtmpx ]; then
-          echo "not tested: can't exec ./check_wtmpx"
+       if [ ! -x ${CHECK_WTMPX} ]; then
+          echo "not tested: can't exec ${CHECK_WTMPX}"
        else
           if [ "${EXPERT}" = "t" ]; then
-             expertmode_output "./check_wtmpx"
+             expertmode_output "${CHECK_WTMPX}"
               return 5
           fi
-          if ./check_wtmpx
+          if ${CHECK_WTMPX}
              then
              if [ "${QUIET}" != "t" ]; then \
                 echo "nothing deleted in /var/adm/wtmpx"; fi
@@ -217,12 +225,12 @@
        WTMP=`loc wtmp wtmp "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
 
        if [ "${EXPERT}" = "t" ]; then
-          expertmode_output "./chkwtmp -f ${WTMP}"
+          expertmode_output "${CHKWTMP} -f ${WTMP}"
           return 5
        fi
     fi
 
-    if ./chkwtmp -f ${WTMP}
+    if ${CHKWTMP} -f ${WTMP}
     then
       if [ "${QUIET}" != "t" ]; then echo "nothing deleted"; fi
     fi
@@ -261,7 +269,7 @@
     prog=""
     if [  \( "${SYSTEM}" = "Linux"  -o \( "${SYSTEM}" = "FreeBSD" -a \
        ${V} -gt 43 \) \) -a "${ROOTDIR}" = "/" ]; then
-      [ ! -x ./chkproc ] && prog="./chkproc"
+      [ ! -x ${CHKPROC} ] && prog="${CHKPROC}"
       [ ! -x ./chkdirs ] && prog="$prog ./chkdirs"
       if [ "$prog" != "" ]; then
 #        echo "not tested: can't exec $prog"
@@ -271,7 +279,7 @@
       if [ "${EXPERT}" = "t" ]; then
          [ -r /proc/ksyms ] &&  ${egrep} -i "adore|sebek" < /proc/ksyms 2>/dev/null
          [ -d /proc/knark ] &&  ${ls} -la /proc/knark 2> /dev/null
-          expertmode_output "./chkproc -v -v"
+          expertmode_output "${CHKPROC} -v -v"
           return 5
       fi
 
@@ -292,7 +300,7 @@
          echo "Warning: Knark LKM installed"
       fi
 
-      if ./chkproc
+      if ${CHKPROC}
       then
            if [ "${QUIET}" != "t" ]; then echo "nothing detected"; fi
       else
@@ -464,7 +472,7 @@
       ${egrep} "\.hk" ${ROOTDIR}etc/rc.d/init.d/network 2>/dev/null
 
       ## Suckit rootkit
-      expertmode_output "${strings} ${ROOTDIR}sbin/init | ${egrep} HOME"
+      expertmode_output "${STRINGS} ${ROOTDIR}sbin/init | ${egrep} HOME"
 
       ## Volc  rootkit
       expertmode_output "${ls} ${ROOTDIR}usr/bin/volc"
@@ -875,7 +883,7 @@
    ### Suckit
    if [ -f ${ROOTDIR}sbin/init ]; then
       if [ "${QUIET}" != "t" ];then printn "Searching for Suckit rootkit ... "; fi
-      if [ ${SYSTEM} != "HP-UX" ] && ${strings} ${ROOTDIR}sbin/init | ${egrep} HOME >/dev/null 2>&1
+      if [ ${SYSTEM} != "HP-UX" ] && ${STRINGS} ${ROOTDIR}sbin/init | ${egrep} HOME >/dev/null 2>&1
         then
         echo "Warning: ${ROOTDIR}sbin/init INFECTED"
       else
@@ -1026,20 +1034,20 @@
     STATUS=${NOT_INFECTED}
     CMD=`loc chfn chfn $pth`
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
     case "${SYSTEM}" in
        Linux)
-          if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
+          if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
              >/dev/null 2>&1
           then
              STATUS=${INFECTED}
           fi;;
        FreeBSD)
           [ $V -ge 50 ] && n=1 || n=2 
-          if [ `${strings} -a ${CMD} | \
+          if [ `${STRINGS} -a ${CMD} | \
                 ${egrep} -c "${GENERIC_ROOTKIT_LABEL}"` -ne $n ]
           then
              STATUS=${INFECTED}
@@ -1054,16 +1062,16 @@
     REDHAT_PAM_LABEL="*NOT*"
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
     case "${SYSTEM}" in
        Linux)
-          if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
+          if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
           >/dev/null 2>&1
              then
-             if ${strings} -a ${CMD} | ${egrep} "${REDHAT_PAM_LABEL}" \
+             if ${STRINGS} -a ${CMD} | ${egrep} "${REDHAT_PAM_LABEL}" \
              >/dev/null 2>&1
                 then
                 :
@@ -1073,7 +1081,7 @@
           fi;;
        FreeBSD)
           [ $V -ge 50 ] && n=1 || n=2 
-          if [ `${strings} -a ${CMD} | ${egrep} -c "${GENERIC_ROOTKIT_LABEL}"` -ne $n ]
+          if [ `${STRINGS} -a ${CMD} | ${egrep} -c "${GENERIC_ROOTKIT_LABEL}"` -ne $n ]
              then
              STATUS=${INFECTED}
           fi;;
@@ -1086,13 +1094,13 @@
     CMD=`loc login login $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
     if [ "$SYSTEM" = "SunOS" ]; then
       TROJED_L_L="porcao|/bin/xstat"
-      if ${strings} -a ${CMD} | ${egrep} "${TROJED_L_L}" >/dev/null 2>&1 ]; then
+      if ${STRINGS} -a ${CMD} | ${egrep} "${TROJED_L_L}" >/dev/null 2>&1 ]; then
           return ${INFECTED}
        else
           return ${NOT_TESTED}
@@ -1100,7 +1108,7 @@
     fi
     GENERAL="^root$"
     TROJED_L_L="vejeta|xlogin|^@\(#\)klogin\.c|lets_log|sukasuka|/usr/lib/.ark?|SucKIT"
-    ret=`${strings} -a ${CMD} | ${egrep} -c "${GENERAL}"`
+    ret=`${STRINGS} -a ${CMD} | ${egrep} -c "${GENERAL}"`
     if [ ${ret} -gt 0 ]; then
         case ${ret} in
         1) [ "${SYSTEM}" = "OpenBSD" -a ${V} -le 27 -o ${V} -ge 30 ] && \
@@ -1111,7 +1119,7 @@
         *) STATUS=${INFECTED};;
         esac
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${TROJED_L_L}" 2>&1 >/dev/null
+    if ${STRINGS} -a ${CMD} | ${egrep} "${TROJED_L_L}" 2>&1 >/dev/null
        then
         STATUS=${INFECTED}
     fi
@@ -1127,14 +1135,14 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-       expertmode_output "${strings} -a ${CMD}"
+       expertmode_output "${STRINGS} -a ${CMD}"
     fi
 
     if [ "${SYSTEM}" = "OpenBSD" -o "${SYSTEM}" = "SunOS" ]
     then
        return ${NOT_TESTED}
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}|/lib/security" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}|/lib/security" \
     >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1152,11 +1160,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
     >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1175,11 +1183,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${SYSLOG_I_L}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${SYSLOG_I_L}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1196,11 +1204,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${HDPARM_INFECTED_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${HDPARM_INFECTED_LABEL}" \
        >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1218,11 +1226,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GPM_INFECTED_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GPM_INFECTED_LABEL}" \
        >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1240,11 +1248,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${MINGETTY_INFECTED_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${MINGETTY_INFECTED_LABEL}" \
        >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1262,11 +1270,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${SENDMAIL_INFECTED_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${SENDMAIL_INFECTED_LABEL}" \
        >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1280,11 +1288,11 @@
     CMD=`loc ls ls $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${LS_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${LS_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1297,11 +1305,11 @@
     CMD=`loc du du $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${DU_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${DU_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1321,11 +1329,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${NAMED_I_L}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${NAMED_I_L}" \
     >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1339,11 +1347,11 @@
     CMD=`loc netstat netstat $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${NETSTAT_I_L}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${NETSTAT_I_L}" \
     >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1358,11 +1366,11 @@
    CMD=`loc ps ps $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${PS_I_L}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${PS_I_L}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1380,11 +1388,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${PSTREE_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${PSTREE_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1402,11 +1410,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${TOP_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${TOP_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1424,11 +1432,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${TOP_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${TOP_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1446,11 +1454,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${TOP_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${TOP_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1463,18 +1471,18 @@
 
    if [ "${SYSTEM}" = "Linux" ]
    then
-      if [ ! -x ./strings ]; then
-        printn "can't exec ./strings-static, "
+      if [ ! -x ${STRINGS} ]; then
+        printn "can't exec ${STRINGS}, "
         return ${NOT_TESTED}
       fi
 
       if [ "${EXPERT}" = "t" ]; then
-          expertmode_output "./strings -a ${CMD}"
+          expertmode_output "${STRINGS} -a ${CMD}"
           return 5
       fi
 
       ### strings must be a statically linked binary.
-      if ./strings -a ${CMD} > /dev/null 2>&1
+      if ${STRINGS} -a ${CMD} > /dev/null 2>&1
       then
          STATUS=${INFECTED}
       fi
@@ -1489,11 +1497,11 @@
    CMD=`loc basename basename $pth`
 
    if [ "${EXPERT}" = "t" ]; then
-       expertmode_output "${strings} -a ${CMD}"
+       expertmode_output "${STRINGS} -a ${CMD}"
        expertmode_output "${ls} -l ${CMD}"
        return 5
    fi
-   if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+   if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
    then
        STATUS=${INFECTED}
    fi
@@ -1513,11 +1521,11 @@
     CMD=`loc dirname dirname $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1538,11 +1546,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1554,12 +1562,12 @@
     CMD=`loc rpcinfo rpcinfo $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1576,19 +1584,19 @@
     CMD=`loc date date $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
     [ "${SYSTEM}" = "FreeBSD" -a $V -ge 50 ] &&
     {
-       if [ `${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" | \
+       if [ `${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" | \
           ${egrep} -c "$S_L"` -ne 2 ]; then
           STATUS=${INFECTED}
        fi
     } ||
     {
-       if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" 2>&1
+       if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" 2>&1
           then
           STATUS=${INFECTED}
        fi
@@ -1605,12 +1613,12 @@
     CMD=`loc echo echo $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1626,12 +1634,12 @@
     CMD=`loc env env $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1653,11 +1661,11 @@
        fi
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1671,11 +1679,11 @@
        return ${NOT_FOUND}
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1690,11 +1698,11 @@
        return ${NOT_FOUND}
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${INIT_INFECTED_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${INIT_INFECTED_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1708,11 +1716,11 @@
        return ${NOT_FOUND}
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1726,11 +1734,11 @@
         return ${NOT_FOUND}
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1742,12 +1750,12 @@
     CMD=`loc write write $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" | grep -v locale > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" | grep -v locale > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1764,11 +1772,11 @@
     W_INFECTED_LABEL="uname -a"
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${W_INFECTED_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${W_INFECTED_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1784,11 +1792,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${VDIR_INFECTED_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${VDIR_INFECTED_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1820,7 +1828,7 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
     STATUS=${INFECTED}
@@ -1837,12 +1845,12 @@
     MAIL_INFECTED_LABEL="sh -i"
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${MAIL_INFECTED_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${MAIL_INFECTED_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1862,12 +1870,12 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1884,11 +1892,11 @@
     CMD=`loc egrep egrep $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${EGREP_INFECTED_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${EGREP_INFECTED_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1901,12 +1909,12 @@
     CMD=`loc grep grep $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         expertmode_output "${ls} -l ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GREP_INFECTED_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GREP_INFECTED_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -1928,11 +1936,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${TOP_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${TOP_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1950,10 +1958,10 @@
        fi
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${RLOGIN_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${RLOGIN_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1968,10 +1976,10 @@
          return ${NOT_FOUND}
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${LSOF_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${LSOF_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -1986,10 +1994,10 @@
          return ${NOT_FOUND}
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${AMD_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${AMD_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -2004,10 +2012,10 @@
          return ${NOT_FOUND}
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${SLOGIN_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${SLOGIN_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -2026,10 +2034,10 @@
         return ${NOT_FOUND}
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${CRON_INFECTED_LABEL}" >/dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${CRON_INFECTED_LABEL}" >/dev/null 2>&1
     then
        STATUS=${INFECTED}
     fi
@@ -2041,18 +2049,18 @@
     CMD="${ROOTDIR}sbin/ifconfig"
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
     IFCONFIG_NOT_INFECTED_LABEL="PROMISC"
     IFCONFIG_INFECTED_LABEL="/dev/tux|/session.null"
-    if ${strings} -a ${CMD} | ${egrep} "${IFCONFIG_NOT_INFECTED_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${IFCONFIG_NOT_INFECTED_LABEL}" \
     >/dev/null 2>&1
     then
        STATUS=${NOT_INFECTED}
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${IFCONFIG_INFECTED_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${IFCONFIG_INFECTED_LABEL}" \
     >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -2072,12 +2080,12 @@
        return ${NOT_FOUND}
     fi
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
     RSHD_INFECTED_LABEL="HISTFILE"
-    if ${strings} -a ${CMD} | ${egrep} "${RSHD_INFECTED_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${RSHD_INFECTED_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
         if ${egrep} "^#.*rshd" ${ROOTDIR}etc/inetd.conf >/dev/null 2>&1 -o \
@@ -2113,11 +2121,11 @@
     [ "tcpd" = "${CMD}" ] && return ${NOT_FOUND};
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${TCPD_INFECTED_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${TCPD_INFECTED_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -2134,11 +2142,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${SSHD2_INFECTED_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${SSHD2_INFECTED_LABEL}" \
        > /dev/null 2>&1
     then
         STATUS=${INFECTED}
@@ -2155,11 +2163,11 @@
     CMD=`loc su su $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${SU_INFECTED_LABEL}" > /dev/null 2>&1
+    if ${STRINGS} -a ${CMD} | ${egrep} "${SU_INFECTED_LABEL}" > /dev/null 2>&1
     then
         STATUS=${INFECTED}
     fi
@@ -2179,11 +2187,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${FINGER_INFECTED_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${FINGER_INFECTED_LABEL}" \
 > /dev/null 2>&1
     then
         STATUS=${INFECTED}
@@ -2231,11 +2239,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${TELNETD_INFECTED_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${TELNETD_INFECTED_LABEL}" \
        >/dev/null 2>&1
     then
         STATUS=${INFECTED}
