diff -U 4 -H -b -w -B -E -d -i -r kwalletmanager-3.5.4-orig/kwallet/kwalletmanager.cpp kwalletmanager-3.5.4/kwallet/kwalletmanager.cpp
--- kwalletmanager-3.5.4-orig/kwallet/kwalletmanager.cpp	2006-09-27 00:38:20.000000000 +0200
+++ kwalletmanager-3.5.4/kwallet/kwalletmanager.cpp	2006-09-27 00:39:40.000000000 +0200
@@ -42,8 +42,9 @@
 #include <qaccel.h>
 #include <qguardedptr.h>
 #include <qptrstack.h>
 #include <qregexp.h>
+#include <qtimer.h>
 #include <qtooltip.h>
 
 KWalletManager::KWalletManager(QWidget *parent, const char *name, WFlags f)
 : KMainWindow(parent, name, f), DCOPObject("KWalletManager") {
@@ -74,9 +75,9 @@
 		}
 		if (!isOpen && kapp->isRestored()) {
 			delete _tray;
 			_tray = 0L;
-			kapp->exit();
+			QTimer::singleShot( 0, kapp, SLOT( quit()));
 			return;
 		}
 	} else {
 		_tray = 0L;
