--- kid-0.9/setup.py	2005-12-02 07:46:12.000000000 +0000
+++ kid-0.9-patched/setup.py	2006-01-03 08:49:19.000000000 +0000
@@ -1,6 +1,4 @@
 # bootstrap setuptools if necessary
-from ez_setup import use_setuptools
-use_setuptools()
 
 import os

@@ -8,7 +6,7 @@
 
 doc_parts = long_description.strip().splitlines()

-from setuptools import setup, find_packages
+from distutils.core import setup
 setup(
     name="kid",
     version=version,
