--- GCC_XML/GXFront/gxConfiguration.cxx       2004-01-22 22:51:36.000000000 +0100
+++ GCC_XML/GXFront/gxConfiguration.cxx       2008-11-26 18:17:34.000000000 +0100
@@ -15,6 +15,7 @@

 =========================================================================*/
 #include "gxConfiguration.h"
+#include <cstring>

 //----------------------------------------------------------------------------
 const char* gxConfigurationVc6Registry =
--- GCC_XML/KWSys/SystemTools.cxx       2004-01-22 16:30:01.000000000 +0100
+++ GCC_XML/KWSys/SystemTools.cxx       2008-11-26 18:16:53.000000000 +0100
@@ -17,6 +17,7 @@

 #include KWSYS_HEADER(ios/iostream)
 #include KWSYS_HEADER(ios/fstream)
+#include <cstring>

 #ifdef _MSC_VER
 # pragma warning (disable: 4786)

