Index: vdr-1.4.7/Makefile
===================================================================
--- vdr-1.4.7.orig/Makefile
+++ vdr-1.4.7/Makefile
@@ -19,6 +19,7 @@ MANDIR   = /usr/local/man
 BINDIR   = /usr/local/bin
 LIBS     = -ljpeg -lpthread -ldl -lcap
 INCLUDES =
+HEADERDIR = /usr/local/include
 
 PLUGINDIR= ./PLUGINS
 PLUGINLIBDIR= $(PLUGINDIR)/lib
@@ -196,7 +197,7 @@ clean-plugins:
 
 # Install the files:
 
-install: install-bin install-conf install-doc install-plugins
+install: install-bin install-conf install-doc install-plugins install-header
 
 # VDR binary:
 
@@ -227,6 +228,13 @@ install-plugins: plugins
 	@mkdir -p $(DESTDIR)$(PLUGINLIBDIR)
 	@cp --remove-destination $(PLUGINDIR)/lib/lib*-*.so.$(APIVERSION) $(DESTDIR)$(PLUGINLIBDIR)
 
+# Header files:
+
+install-header:
+	@mkdir -p $(DESTDIR)$(HEADERDIR)/vdr $(DESTDIR)$(HEADERDIR)/vdr/libsi
+	@cp *.h Make.config $(DESTDIR)$(HEADERDIR)/vdr
+	@cp libsi/*.h $(DESTDIR)$(HEADERDIR)/vdr/libsi
+
 # Source documentation:
 
 srcdoc:
