--- src/init2.c.orig	2010-12-31 12:47:48.270293037 -0500
+++ src/init2.c	2010-12-31 12:50:20.368081626 -0500
@@ -2719,10 +2719,10 @@
 	if (sval < 0)
 		return PARSE_ERROR_UNRECOGNISED_SVAL;
 
-	for (i = 0; i <= MAX_START_ITEMS; i++)
+	for (i = 0; i < MAX_START_ITEMS; i++)
 		if (!c->start_items[i].min)
 			break;
-	if (i > MAX_START_ITEMS)
+	if (i >= MAX_START_ITEMS)
 		return PARSE_ERROR_TOO_MANY_ENTRIES;
 	c->start_items[i].kind = objkind_get(tval, sval);
 	c->start_items[i].min = parser_getuint(p, "min");
