diff -ur cl-prevalence.orig/cl-prevalence.asd cl-prevalence/cl-prevalence.asd
--- cl-prevalence.orig/cl-prevalence.asd	2004-06-28 06:57:26.000000000 -0500
+++ cl-prevalence/cl-prevalence.asd	2004-07-10 23:57:21.757852640 -0500
@@ -33,6 +33,6 @@
      (:file "sysdeps" :depends-on ("package"))
      (:file "master-slave" :depends-on ("prevalence" "sysdeps"))
      (:file "blob" :depends-on ("managed-prevalence")))))
-  :depends-on (:s-xml))
+  :depends-on (:s-xml #+sbcl :sb-bsd-sockets))
 
 ;;;; eof
