--- branches/0.6.5/kadu-core/main.cpp	2009/01/04 13:18:33	6032
+++ branches/0.6.5/kadu-core/main.cpp	2009/02/22 10:45:49	6261
@@ -153,6 +153,11 @@
 	}
 
 #endif
+
+	kdebugm(KDEBUG_INFO, "before creation of new KaduApplication\n");
+	new KaduApplication(argc, argv);
+	kdebugm(KDEBUG_INFO, "after creation of new KaduApplication\n");
+
 	debug_mask = -1;
 	qInstallMsgHandler(kaduQtMessageHandler);
 	xml_config_file = new XmlConfigFile();
@@ -199,7 +204,6 @@
 		if (freopen(path, "w+", stderr) == 0)
 			fprintf(stdout, "freopen: %s\n", strerror(errno));
 #endif
-
 	}
 
 #ifdef DEBUG_ENABLED
@@ -227,10 +231,6 @@
 		exit(10);
 	}
 
-	kdebugm(KDEBUG_INFO, "before creation of new KaduApplication\n");
-	new KaduApplication(argc, argv);
-	kdebugm(KDEBUG_INFO, "after creation of new KaduApplication\n");
-
 	defaultFont = new QFont(qApp->font());
 	defaultFontInfo = new QFontInfo(*defaultFont);
 	// loading translation
