diff -Naur modules.orig/nas_sound/nas_sound.h modules/nas_sound/nas_sound.h
--- modules.orig/nas_sound/nas_sound.h	2004-05-11 14:47:05.000000000 +0200
+++ modules/nas_sound/nas_sound.h	2004-07-04 16:42:49.000000000 +0200
@@ -14,8 +14,8 @@
 #ifndef INTERNAL_QT_SOUND_SUPPORT
 #include <qsocketnotifier.h>
 namespace audiolib {
-#include <audiolib.h>
-#include <soundlib.h>
+#include <audio/audiolib.h>
+#include <audio/soundlib.h>
 }
 #endif
 
diff -Naur modules.orig/nas_sound/spec modules/nas_sound/spec
--- modules.orig/nas_sound/spec	2004-04-26 14:59:35.000000000 +0200
+++ modules/nas_sound/spec	2004-07-04 16:44:14.000000000 +0200
@@ -1,6 +1,6 @@
 MODULE_SOURCES=nas_sound.cpp
 MODULE_MOC_HEADERS=nas_sound.h
-MODULE_INCLUDES="soundlib.h audiolib.h"
-MODULE_INCLUDES_PATH="/usr/include/audio /usr/X11R6/include/audio"
+MODULE_INCLUDES="audio/soundlib.h audio/audiolib.h"
+MODULE_INCLUDES_PATH="/usr/include /usr/X11R6/include"
 MODULE_LIBS=audio
 MODULE_LIBS_PATH="/usr/X11R6/lib"
