--- Makefile.in
+++ Makefile.in
@@ -33,22 +33,22 @@
 all:		@LIB_TARGETS@
 
 libsqliteodbc.la:	sqliteodbc.lo
-		$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o libsqliteodbc.la \
+		$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o libsqliteodbc.la \
 		    sqliteodbc.lo -rpath $(libdir) $(SQLITE_LIB) \
 		    $(ODBC_LIB) -release $(VER_INFO)
 
 libsqlite3odbc.la:	sqlite3odbc.lo $(SQLITE3_A10N_O)
-		$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o libsqlite3odbc.la \
+		$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o libsqlite3odbc.la \
 		    sqlite3odbc.lo $(SQLITE3_A10N_O) -rpath $(libdir) \
 		    $(SQLITE3_LIB) $(ODBC_LIB) -release $(VER_INFO)
 
 libsqlite3_mod_blobtoxy.la:	blobtoxy.lo
-		$(LIBTOOL) --mode=link $(CC) $(CFLAGS) \
+		$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) \
 		    -o libsqlite3_mod_blobtoxy.la \
 		    blobtoxy.lo -rpath $(libdir) -release $(VER_INFO)
 
 libsqlite3_mod_impexp.la:	impexp.lo
-		$(LIBTOOL) --mode=link $(CC) $(CFLAGS) \
+		$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) \
 		    -o libsqlite3_mod_impexp.la \
 		    impexp.lo -rpath $(libdir) -release $(VER_INFO)
 
