diff -ru sqliteodbc-0.64.orig/configure sqliteodbc-0.64/configure
--- sqliteodbc-0.64.orig/configure	2004-07-08 09:33:50.000000000 +0200
+++ sqliteodbc-0.64/configure	2005-03-27 13:30:44.759094368 +0200
@@ -5412,9 +5412,9 @@
 ODBC_FLAGS=UNKNOWN
 ODBC_LIB=UNKNOWN
 for i in $ODBC_DIR ; do
-   if test -r "$i/include/sql.h" ; then
-      if test -r "$i/include/iodbcinst.h" ; then
-         ODBC_FLAGS="-DHAVE_IODBC=1 -I$i/include"
+   if test -r "$i/include/iodbc/sql.h" -o -r "$i/include/sql.h" ; then
+      if test -r "$i/include/iodbc/iodbcinst.h" ; then
+         ODBC_FLAGS="-DHAVE_IODBC=1 -I$i/include/iodbc"
          if test -r "$i/lib/libiodbcinst.la" ; then
             ODBC_LIB="$i/lib/libiodbcinst.la"
          else
diff -ru sqliteodbc-0.64.orig/configure.in sqliteodbc-0.64/configure.in
--- sqliteodbc-0.64.orig/configure.in	2004-07-08 09:30:56.000000000 +0200
+++ sqliteodbc-0.64/configure.in	2005-03-27 13:30:29.765373760 +0200
@@ -170,9 +170,9 @@
 ODBC_FLAGS=UNKNOWN
 ODBC_LIB=UNKNOWN
 for i in $ODBC_DIR ; do
-   if test -r "$i/include/sql.h" ; then
-      if test -r "$i/include/iodbcinst.h" ; then
-         ODBC_FLAGS="-DHAVE_IODBC=1 -I$i/include"
+   if test -r "$i/include/iodbc/sql.h" -o -r "$i/include/sql.h" ; then
+      if test -r "$i/include/iodbc/iodbcinst.h" ; then
+         ODBC_FLAGS="-DHAVE_IODBC=1 -I$i/include/iodbc"
          if test -r "$i/lib/libiodbcinst.la" ; then
             ODBC_LIB="$i/lib/libiodbcinst.la"
          else
Only in sqliteodbc-0.64: .configure.in.swp
