diff -Nur kdeartwork-3.4.3.orig/kscreensaver/configure.in.in kdeartwork-3.4.3/kscreensaver/configure.in.in
--- kdeartwork-3.4.3.orig/kscreensaver/configure.in.in	2005-10-06 08:12:25.000000000 +0200
+++ kdeartwork-3.4.3/kscreensaver/configure.in.in	2005-10-08 10:02:25.000000000 +0200
@@ -37,32 +37,41 @@
 AM_CONDITIONAL(COMPILE_ARTS_KSAVERS, test x$build_arts = xyes)
 AM_CONDITIONAL(COMPILE_ARTS_GL_KSAVERS, test x$have_gl = xyes && test x$ac_cv_kde_qt_has_opengl = xyes && test x$build_arts = xyes)
 
-LIBART=
+AC_ARG_WITH(libart,
+  [AC_HELP_STRING(--with-libart,
+    [enable support for libart @<:@default=check@:>@])],
+  [], with_libart=check)
 
-KDE_FIND_PATH(libart2-config, LIBART_CONFIG, [${prefix}/bin ${exec_prefix}/bin /usr/local/bin /opt/local/bin], [
-  AC_MSG_WARN([Could not find libart anywhere, check http://www.levien.com/libart/])
-])
-
-if test -n "$LIBART_CONFIG"; then
-  vers=`$LIBART_CONFIG --version 2>/dev/null | awk 'BEGIN { FS = "."; } { printf "%d", ($1 * 1000 + $2) * 1000 + $3;}'`
-  if test -n "$vers" && test "$vers" -ge 2003008
-  then
-     LIBART_LIBS="`$LIBART_CONFIG --libs`"
-     LIBART_RPATH=
-     for args in $LIBART_LIBS; do
+LIBART=
+if test "x$with_libart" != xno; then
+  KDE_FIND_PATH(libart2-config, LIBART_CONFIG, [${prefix}/bin ${exec_prefix}/bin /usr/local/bin /opt/local/bin], [
+    AC_MSG_WARN([Could not find libart anywhere, check http://www.levien.com/libart/])
+  ])
+
+  if test -n "$LIBART_CONFIG"; then
+    vers=`$LIBART_CONFIG --version 2>/dev/null | awk 'BEGIN { FS = "."; } { printf "%d", ($1 * 1000 + $2) * 1000 + $3;}'`
+    if test -n "$vers" && test "$vers" -ge 2003008; then
+      LIBART_LIBS="`$LIBART_CONFIG --libs`"
+      LIBART_RPATH=
+      for args in $LIBART_LIBS; do
 	  case $args in
 	    -L*)
 	       LIBART_RPATH="$LIBART_RPATH $args"
 	       ;;
           esac
-     done
-     LIBART_RPATH=`echo $LIBART_RPATH | sed -e "s/-L/-R/g"`
-     LIBART_CFLAGS="`$LIBART_CONFIG --cflags`"
-     LIBART=yes
-
-     AC_DEFINE_UNQUOTED(HAVE_LIBART, 1, [Defines if your system has the libart library])
-  else
-     AC_MSG_WARN([You need at least libart 2.3.8])
+      done
+      LIBART_RPATH=`echo $LIBART_RPATH | sed -e "s/-L/-R/g"`
+      LIBART_CFLAGS="`$LIBART_CONFIG --cflags`"
+      LIBART=yes
+
+      AC_DEFINE_UNQUOTED(HAVE_LIBART, 1, [Defines if your system has the libart library])
+    else
+      AC_MSG_WARN([You need at least libart 2.3.8])
+    fi
+  fi
+
+  if test "x$with_libart" != xcheck && test -z "$LIBART"; then
+      AC_MSG_ERROR([--with-libart was given, but test for libart failed])
   fi
 fi
 
diff -Nur kdeartwork-3.4.3.orig/kscreensaver/kxsconfig/configure.in.bot kdeartwork-3.4.3/kscreensaver/kxsconfig/configure.in.bot
--- kdeartwork-3.4.3.orig/kscreensaver/kxsconfig/configure.in.bot	2005-10-05 15:11:48.000000000 +0200
+++ kdeartwork-3.4.3/kscreensaver/kxsconfig/configure.in.bot	2005-10-08 10:02:25.000000000 +0200
@@ -1,4 +1,4 @@
-if test "$XSCREENSAVER" = "NO"; then
+if test "x$with_xscreensaver" = xcheck && test "$XSCREENSAVER" = "NO"; then
   echo ""
   echo "xscreensaver could not be detected on your system."
   echo "Check www.jwz.org/xscreensaver if you're loving to"
diff -Nur kdeartwork-3.4.3.orig/kscreensaver/kxsconfig/configure.in.in kdeartwork-3.4.3/kscreensaver/kxsconfig/configure.in.in
--- kdeartwork-3.4.3.orig/kscreensaver/kxsconfig/configure.in.in	2005-10-05 15:11:48.000000000 +0200
+++ kdeartwork-3.4.3/kscreensaver/kxsconfig/configure.in.in	2005-10-08 10:03:21.000000000 +0200
@@ -22,25 +22,30 @@
 done
 IFS=$old_IFS
 
-KDE_FIND_PATH(lib$kdelibsuff/xscreensaver/flame, XSCREENSAVER, [ $xscreensaver_prefixes ],
-   [
-        KDE_FIND_PATH(lib/xscreensaver/flame, XSCREENSAVER, [ $xscreensaver_prefixes ],
-        [
-    	    KDE_FIND_PATH(bin/xscreensaver-hacks/flame, XSCREENSAVER, 
-  	      [ $xscreensaver_prefixes ],
-	      [
-    	          KDE_FIND_PATH(hacks/flame, XSCREENSAVER, 
-  	          [ $xscreensaver_prefixes ],
-	          [
-		    XSCREENSAVER=NO
-	          ])
-	      ])
-        ])
-   ])
+AC_ARG_WITH(xscreensaver,
+  [AC_HELP_STRING(--with-xscreensaver,
+    [enable support for xscreensaver @<:@default=check@:>@])],
+  [], with_xscreensaver=check)
+
+XSCREENSAVER=''
+echo "Searching in $xscreensaver_prefixes"
+if test "x$with_xscreensaver" != xno; then
+  for xsdir in lib$kdelibsuff/xscreensaver lib/xscreensaver \
+               lib$kdelibsuff/misc/xscreensaver lib/misc/xscreensaver \
+               libexec$kdelibsuff/xscreensaver libexec/xscreensaver \
+               bin/xscreensaver-hacks hacks; do
+    test -n "$XSCREENSAVER" && break
+    KDE_FIND_PATH("$xsdir/flame", XSCREENSAVER, [ $xscreensaver_prefixes ])
+  done
+
+  if test "x$with_xscreensaver" != xcheck && test "x$XSCREENSAVER" = x; then
+    AC_MSG_ERROR([--with-xscreensaver was given, but test for xscreensaver failed])
+  fi
+fi
 
 XSCREENSAVER_HACKS_DIR=
 
-if test "$XSCREENSAVER" != "NO"; then
+if test -n "$XSCREENSAVER"; then
      XSCREENSAVER_HACKS_DIR=`echo $XSCREENSAVER | sed -e 's,/flame$,,'`
      AC_DEFINE_UNQUOTED(XSCREENSAVER_HACKS_DIR, "$XSCREENSAVER_HACKS_DIR", [Defines where xscreensaver stores its graphic hacks])
 
@@ -48,7 +53,6 @@
      for subdir in lib/xscreensaver/config \
 		   lib/X11/xscreensaver/config \
 		   share/control-center/screensavers \
-		   share/control-center/screensavers \
 		   share/xscreensaver/config \
                    xscreensaver \
 		   config/control-center-2.0; do
@@ -62,6 +66,5 @@
 	[Defines where xscreensaver stores it's config files])
 fi
 
-AM_CONDITIONAL(include_XSCREENSAVER, test "$XSCREENSAVER" != "NO")
+AM_CONDITIONAL(include_XSCREENSAVER, test -n "$XSCREENSAVER")
 AC_SUBST(XSCREENSAVER_HACKS_DIR)
-
