--- config_office/configure.in 
+++ config_office/configure.in
@@ -2885,7 +2885,7 @@
    dnl Test whether we want to use the Mozilla or the OpenLDAP LDAP SDK
    dnl ===================================================================
    AC_MSG_CHECKING([which LDAP SDK to use])
-   if test -n "$enable_openldap"; then
+   if test -n "$enable_openldap" && test "$enable_openldap" != "no"; then
   	   AC_MSG_RESULT([OpenLDAP])
 	   WITH_OPENLDAP=YES
 	   AC_CHECK_HEADERS(ldap.h, [],
