--- config_office/configure.in.orig	2005-08-11 16:07:16.000000000 +0200
+++ config_office/configure.in	2005-08-13 12:39:17.000000000 +0200
@@ -2784,7 +2784,7 @@
 fi
 
 AC_MSG_CHECKING([whether to build LDAP configuration backend])
-if test "$enable_mozilla" = "no" && test "$enable_openldap" != "yes"; then
+if test "$enable_openldap" != "yes"; then
    AC_MSG_RESULT([no. Either Mozilla or OpenLDAP needed.])
    WITH_LDAP=NO
 elif test "$with_system_mozilla" = "yes" || test "$enable_openldap" = "yes"; then
--- extensions/source/config/ldap/makefile.mk.orig	2005-06-19 02:53:37.725788224 +0200
+++ extensions/source/config/ldap/makefile.mk	2005-06-19 02:54:17.068807176 +0200
@@ -75,7 +75,7 @@
 .INCLUDE : settings.mk
 DLLPRE =
 
-.IF "$(WITH_MOZILLA)" == "NO" && "$(WITH_OPENLDAP)" != "YES"
+.IF "$(WITH_OPENLDAP)" != "YES"
 @all:
 	@echo "Mozilla disabled and no OpenLDAP wanted -> LDAP disabled."
 .ENDIF
--- scp2/source/ooo/file_library_ooo.scp.orig	2005-08-18 14:53:05.000000000 +0200
+++ scp2/source/ooo/file_library_ooo.scp	2005-08-18 14:53:11.000000000 +0200
@@ -185,7 +185,7 @@
 End
 
 #ifndef WITHOUT_LDAP
-#if !defined WITHOUT_MOZILLA || defined WITH_OPENLDAP
+#if defined WITH_OPENLDAP
 File gid_File_Lib_Ldapbe2
     TXT_FILE_BODY;
     Styles = (PACKED,UNO_COMPONENT,PATCH);
