Index: src/session.cpp
===================================================================
--- src/session.cpp	(revision 223)
+++ src/session.cpp	(working copy)
@@ -653,18 +653,19 @@
 		case DTMF_AUTO:
 		case DTMF_INBAND:
 			get_line()->ci_set_dtmf_supported(true, true);
+			ui->cb_dtmf_supported(get_line()->get_line_number());
 			break;
 		case DTMF_RFC2833:
 			get_line()->ci_set_dtmf_supported(false);
+			ui->cb_dtmf_not_supported(get_line()->get_line_number());
 			break;
 		case DTMF_INFO:
 			get_line()->ci_set_dtmf_supported(true, false, true);
+			ui->cb_dtmf_supported(get_line()->get_line_number());
 			break;
 		default:
 			assert(false);
 		}
-		
-		ui->cb_line_state_changed();
 	}
 
 	audio_rtp_session->run();
@@ -675,10 +676,10 @@
 		MEMMAN_DELETE(audio_rtp_session);
 		delete audio_rtp_session;
 		audio_rtp_session = NULL;
+	
+		get_line()->ci_set_dtmf_supported(false);
+		ui->cb_line_state_changed();
 	}
-	
-	get_line()->ci_set_dtmf_supported(false);
-	ui->cb_line_state_changed();
 }
 
 t_audio_session *t_session::get_audio_session(void) const {
