--- /tmp/configure.in	2006-07-06 14:31:01.000000000 +1000
+++ configure.in	2006-07-06 14:42:18.000000000 +1000
@@ -14,6 +14,10 @@
 	AC_HELP_STRING([--without-ilbc], [do not compile iLBC]),
 	[ac_cv_ilbc=$withval], [ac_cv_ilbc=yes])
 
+AC_ARG_WITH(speex,
+	AC_HELP_STRING([--without-speex], [do not compile speex]),
+	[ac_cv_speex=$withval], [ac_cv_speex=yes])
+
 AM_INIT_AUTOMAKE(twinkle,0.8)
 AC_DEFINE(VERSION_DATE,["July 1 2006"],[Version release date])
 AC_PROG_CC
@@ -138,14 +142,19 @@
 	], [have_libasound="no"])
 	
 # Check if SPEEX is available
-AC_CHECK_LIB(speex, main, [
-	AC_CHECK_HEADER(speex/speex.h, [],
-		[AC_MSG_ERROR([speex header files missing (speex-devel package)])])
-	AC_DEFINE(HAVE_SPEEX, 1, [Define to 1 if you have the <speex> library.])
-	LIBS="-lspeex $LIBS"
-	echo "LIBS += -lspeex" >> $QT_INCL_PRO
-	have_speex="yes"
-	], [have_speex="no"])
+if test "x$ac_cv_speex" = "xyes"
+then
+	AC_CHECK_LIB(speex, main, [
+		AC_CHECK_HEADER(speex/speex.h, [],
+			[AC_MSG_ERROR([speex header files missing (speex-devel package)])])
+		AC_DEFINE(HAVE_SPEEX, 1, [Define to 1 if you have the <speex> library.])
+		LIBS="-lspeex $LIBS"
+		echo "LIBS += -lspeex" >> $QT_INCL_PRO
+		have_speex="yes"
+		], [have_speex="no"])
+else
+	have_speex="no"
+fi
 
 # iLBC
 if test "x$ac_cv_ilbc" = "xyes"
