diff -Naur aterm-1.00.beta2/src/pixmap.c aterm-1.00.beta2-bgfix/src/pixmap.c
--- aterm-1.00.beta2/src/pixmap.c	2004-12-14 22:10:55.000000000 +0100
+++ aterm-1.00.beta2-bgfix/src/pixmap.c	2005-05-02 13:02:59.000000000 +0200
@@ -833,8 +833,8 @@
 {
 	XGCValues       gcvalue;
 	GC              gc;
-	unsigned int    width = TermWin_TotalWidth();
-	unsigned int    height = TermWin_TotalHeight();
+	unsigned int    width = TermWin_TotalWidth() + TermWin.fwidth;;
+	unsigned int    height = TermWin_TotalHeight() + TermWin.fheight;
  	unsigned int    fin_width, fin_height ;
 	int (*oldXErrorHandler) (Display *, XErrorEvent *) ;
 
