diff -urN calligra-2.3.81.old/krita/plugins/formats/jpeg/CMakeLists.txt calligra-2.3.81/krita/plugins/formats/jpeg/CMakeLists.txt
--- calligra-2.3.81.old/krita/plugins/formats/jpeg/CMakeLists.txt	2011-09-15 13:13:23.521381587 +0200
+++ calligra-2.3.81/krita/plugins/formats/jpeg/CMakeLists.txt	2011-09-15 13:13:51.571381250 +0200
@@ -1,6 +1,6 @@
 add_subdirectory(tests)
 
-include_directories( ${KRITA_INCLUDES} ${LCMS_INCLUDE_DIR} ${CMAKE_SOURCE_DIR}/krita/image/metadata ${EXIV2_INCLUDE_DIR})
+include_directories( ${KRITA_INCLUDES} ${LCMS2_INCLUDE_DIR} ${CMAKE_SOURCE_DIR}/krita/image/metadata ${EXIV2_INCLUDE_DIR})
 
 set(libkritaconverter_LIB_SRCS
     kis_jpeg_converter.cc
@@ -16,7 +16,7 @@
 
 kde4_add_plugin(kritajpegimport ${kritajpegimport_PART_SRCS})
 
-target_link_libraries(kritajpegimport kritaui jpeg ${LCMS_LIBRARIES} ${EXIV2_LIBRARIES} )
+target_link_libraries(kritajpegimport kritaui jpeg ${LCMS2_LIBRARIES} ${EXIV2_LIBRARIES} )
 
 install(TARGETS kritajpegimport  DESTINATION ${PLUGIN_INSTALL_DIR})
 
@@ -31,7 +31,7 @@
 
 kde4_add_plugin(kritajpegexport ${kritajpegexport_PART_SRCS})
 
-target_link_libraries(kritajpegexport kritaui jpeg ${LCMS_LIBRARIES} ${EXIV2_LIBRARIES} )
+target_link_libraries(kritajpegexport kritaui jpeg ${LCMS2_LIBRARIES} ${EXIV2_LIBRARIES} )
 
 install(TARGETS kritajpegexport  DESTINATION ${PLUGIN_INSTALL_DIR})
 
