https://bugs.launchpad.net/zodb/+bug/459889

--- setup.py
+++ setup.py
@@ -189,6 +189,7 @@
       package_dir = {'': 'src'},
       ext_modules = exts,
       headers = ['src/persistent/cPersistence.h',
+                 'src/persistent/py24compat.h',
                  'src/persistent/ring.h'],
       license = "ZPL 2.1",
       platforms = ["any"],
--- src/persistent/py24compat.h
+++ src/persistent/py24compat.h
@@ -1,5 +1,5 @@
 /* Backport type definitions from Python 2.5's object.h */
-#ifndef PERSISTENT_PY24COMPATH_H
+#ifndef PERSISTENT_PY24COMPAT_H
 #define PERSISTENT_PY24COMPAT_H
 #if PY_VERSION_HEX < 0x02050000
 typedef int Py_ssize_t;
