Modified: branches/lighttpd-merge-1.4.x/configure.in
===================================================================
--- branches/lighttpd-merge-1.4.x/configure.in	2006-02-14 16:09:23 UTC (rev 993)
+++ branches/lighttpd-merge-1.4.x/configure.in	2006-02-14 16:10:14 UTC (rev 994)
@@ -439,9 +439,6 @@
         * ) ;;
 esac
 
-dnl posix_* needs _XOPEN_SOURCE=600
-CFLAGS="${CFLAGS} -D_XOPEN_SOURCE=600 -D_BSD_SOURCE"
-
 AC_CHECK_FUNCS([dup2 getcwd inet_ntoa inet_ntop memset mmap munmap strchr \
 		  strdup strerror strstr strtol sendfile  getopt socket \
 		  gethostbyname poll sigtimedwait epoll_ctl getrlimit chroot \
