Modified: branches/lighttpd-merge-1.4.x/src/network_writev.c
===================================================================
--- branches/lighttpd-merge-1.4.x/src/network_writev.c	2006-02-14 16:10:54 UTC (rev 995)
+++ branches/lighttpd-merge-1.4.x/src/network_writev.c	2006-02-14 23:49:38 UTC (rev 996)
@@ -255,14 +255,14 @@
 #ifdef LOCAL_BUFFERING
 				buffer_copy_string_len(c->mem, c->file.mmap.start, c->file.mmap.length);
 #else
-#ifdef HAVE_POSIX_MADVISE
+#ifdef HAVE_MADVISE
 				/* don't advise files < 64Kb */
 				if (c->file.mmap.length > (64 KByte)) {
 					/* darwin 7 is returning EINVAL all the time and I don't know how to 
 					 * detect this at runtime.i
 					 *
 					 * ignore the return value for now */
-					posix_madvise(c->file.mmap.start, c->file.mmap.length, POSIX_MADV_WILLNEED);
+					madvise(c->file.mmap.start, c->file.mmap.length, MADV_WILLNEED);
 				}
 #endif
 #endif
