Modified: branches/lighttpd-merge-1.4.x/src/mod_cgi.c
===================================================================
--- branches/lighttpd-merge-1.4.x/src/mod_cgi.c	2006-02-15 10:46:17 UTC (rev 998)
+++ branches/lighttpd-merge-1.4.x/src/mod_cgi.c	2006-02-15 15:19:10 UTC (rev 999)
@@ -1128,7 +1128,7 @@
 	plugin_data *p = p_d;
 	buffer *fn = con->physical.path;
 	
-	if (fn->used == 0) return HANDLER_ERROR;
+	if (fn->used == 0) return HANDLER_GO_ON;
 	
 	mod_cgi_patch_connection(srv, con, p);
 	
