diff -Naurp kweather-old/dockwidget.cpp kweather/dockwidget.cpp
--- kweather-old/dockwidget.cpp	2005-05-24 00:09:06.000000000 +1200
+++ kweather/dockwidget.cpp	2005-06-07 09:42:49.000000000 +1200
@@ -169,6 +169,7 @@ void dockwidget::resizeView( const QSize
             m_lblTemp->setGeometry(h+1, 0, w-h, h/3);
             m_lblWind->setGeometry(h+1, h/3, w-h, h/3);
             m_lblPres->setGeometry(h+1, 2*h/3, w-h, h/3);
+	    m_button->setGeometry(0, 0, h, h);
         }
         else if ( m_mode == ShowTempOnly)
         {
