Index: gutenprint-5.0.0-rc2/configure.ac
===================================================================
--- gutenprint-5.0.0-rc2.orig/configure.ac
+++ gutenprint-5.0.0-rc2/configure.ac
@@ -701,8 +701,10 @@ AC_CHECK_LIB(ltdl, lt_dlopen, [LTDL_LIBS
 AC_CHECK_LIB(dl, dlopen, [DLOPEN_LIBS="-ldl"])
 
 AC_CHECK_LIB(m,pow,
-             GUTENPRINT_LIBDEPS="${GUTENPRINT_LIBDEPS} -lm"
-             gutenprint_libdeps="${gutenprint_libdeps} -lm")
+             MATH_LIB="-lm")
+GUTENPRINT_LIBDEPS="${GUTENPRINT_LIBDEPS} ${MATH_LIB}"
+gutenprint_libdeps="${gutenprint_libdeps} ${MATH_LIB}"
+AC_SUBST(MATH_LIB)
 
 STP_CUPS_LIBS
 
Index: gutenprint-5.0.0-rc2/src/cups/Makefile.am
===================================================================
--- gutenprint-5.0.0-rc2.orig/src/cups/Makefile.am
+++ gutenprint-5.0.0-rc2/src/cups/Makefile.am
@@ -73,11 +73,11 @@ epson_SOURCES = epson.c
 epson_LDADD = $(CUPS_LIBS)
 
 cups_genppd_@GUTENPRINT_MAJOR_VERSION@_@GUTENPRINT_MINOR_VERSION@_SOURCES = genppd.c
-cups_genppd_@GUTENPRINT_MAJOR_VERSION@_@GUTENPRINT_MINOR_VERSION@_LDADD = $(CUPS_LIBS) $(GENPPD_LIBS) $(GUTENPRINT_LIBS)
+cups_genppd_@GUTENPRINT_MAJOR_VERSION@_@GUTENPRINT_MINOR_VERSION@_LDADD = $(CUPS_LIBS) $(GENPPD_LIBS) $(GUTENPRINT_LIBS) $(MATH_LIB)
 cups_genppd_@GUTENPRINT_MAJOR_VERSION@_@GUTENPRINT_MINOR_VERSION@_LDFLAGS = $(STATIC_LDOPTS)
 
 rastertogutenprint_@GUTENPRINT_MAJOR_VERSION@_@GUTENPRINT_MINOR_VERSION@_SOURCES = rastertoprinter.c
-rastertogutenprint_@GUTENPRINT_MAJOR_VERSION@_@GUTENPRINT_MINOR_VERSION@_LDADD = $(CUPS_LIBS) $(GUTENPRINT_LIBS)
+rastertogutenprint_@GUTENPRINT_MAJOR_VERSION@_@GUTENPRINT_MINOR_VERSION@_LDADD = $(CUPS_LIBS) $(GUTENPRINT_LIBS) $(MATH_LIB)
 rastertogutenprint_@GUTENPRINT_MAJOR_VERSION@_@GUTENPRINT_MINOR_VERSION@_LDFLAGS = $(STATIC_LDOPTS)
 
 cups-genppdconfig.@GUTENPRINT_MAJOR_VERSION@.@GUTENPRINT_MINOR_VERSION@: $(top_builddir)/config.status cups-genppdconfig.in
