--- A/finch/finch.c	2011-01-08 16:19:44.000000000 +0100
+++ A/finch/finch.c	2011-06-17 14:10:21.611309935 +0200
@@ -77,8 +77,8 @@ 
 		 * can register for your own client key at
 		 * http://developer.aim.com/manageKeys.jsp
 		 */
-		g_hash_table_insert(ui_info, "prpl-aim-clientkey", "ma19sqWV9ymU6UYc");
-		g_hash_table_insert(ui_info, "prpl-icq-clientkey", "ma19sqWV9ymU6UYc");
+		g_hash_table_insert(ui_info, "prpl-aim-clientkey", "ma1cSASNCKFtrdv9");
+		g_hash_table_insert(ui_info, "prpl-icq-clientkey", "ma1cSASNCKFtrdv9");
 
 		/*
 		 * This is the distid for Finch, given to us by AOL.  Please
