Index: model.cxx
===================================================================
--- model.cxx.orig
+++ model.cxx
@@ -24,8 +24,8 @@ static const char *SCALC    = "private:f
 static const char *SDRAW    = "private:factory/sdraw";
 static const char *SIMPRESS = "private:factory/simpress";
 
-static const char *PLUGIN   = "-plugin";
-static const char *QSTART   = "-quickstart";
+static const char *NOLOGO   = "-nologo";
+static const char *NODEFAULT = "-nodefault";
 
 
 /*=========================================================================
@@ -160,7 +160,7 @@ void Model::process () {
     // accidentally terminated.
     if (_enabled) {
         if (!running && _soffice != 0) {
-            _daemonPID = startProcess(BIN_SH, _soffice, PLUGIN, QSTART);
+            _daemonPID = startProcess(BIN_SH, _soffice, NOLOGO, NODEFAULT);
             _status    = MODEL_OFFICE_STARTING;
             _tick      = time(0);
         }
