diff -urN checkpolicy-1.2.orig/Makefile checkpolicy-1.2/Makefile
--- checkpolicy-1.2.orig/Makefile	2003-09-11 15:57:56.000000000 -0500
+++ checkpolicy-1.2/Makefile	2003-10-02 14:06:42.000000000 -0500
@@ -16,13 +16,15 @@
 OBJS =
 endif
 
-CFLAGS = -g $(OPTIONS) -Wall -O2 -pipe
+CFLAGS = $(OPTIONS) -Wall $(EXTRA_CFLAGS)
 override CFLAGS += -include global.h -I. -Iinclude
 
 OBJS += ebitmap.o hashtab.o symtab.o sidtab.o avtab.o policydb.o services.o y.tab.o lex.yy.o queue.o write.o checkpolicy.o
 
 LIBS=-lfl
 
+YACC=bison -y
+
 all:  checkpolicy
 
 checkpolicy: $(OBJS)
