diff -Naurp xfractint-20.04p07/unix/unixscr.c xfractint-20.04p07-fixed/unix/unixscr.c
--- xfractint-20.04p07/unix/unixscr.c	2005-11-22 23:51:24.000000000 +0100
+++ xfractint-20.04p07-fixed/unix/unixscr.c	2007-12-28 13:40:50.000000000 +0100
@@ -137,6 +137,7 @@ static int doredraw = 0; /* 1 if we have
 static Window FindRootWindow(void);
 static Window pr_dwmroot(Display *dpy, Window pwin);
 static int errhand(Display *dp, XErrorEvent *xe);
+static int ioerrhand(Display *dp);
 static int getachar(void);
 static int handleesc(void); 
 static int translatekey(int ch); 
@@ -345,6 +346,30 @@ XErrorEvent *xe;
         return(0);
 }
 
+/*
+ *----------------------------------------------------------------------
+ *
+ * ioerrhand --
+ *
+ *	Called on an X IO server error.
+ *
+ * Results:
+ *	None.
+ *
+ * Side effects:
+ *	Prints the error message.
+ *
+ *----------------------------------------------------------------------
+ */
+static int ioerrhand(dp)
+Display *dp;
+{
+		UnixDone();
+        fflush(stdout);
+		printf("Fatal X IO error on display %s\n", DisplayString(dp));
+        return(0);
+}
+
 #ifdef FPUERR
 /*
  *----------------------------------------------------------------------
@@ -552,6 +577,7 @@ initUnixWindow()
       XSynchronize(Xdp, True);
     }
     XSetErrorHandler(errhand);
+	XSetIOErrorHandler(ioerrhand);
     Xsc = ScreenOfDisplay(Xdp, Xdscreen);
     select_visual();
     if (fixcolors > 0) {
