diff -Nru penggy-0.2.1.orig/configure.ac penggy-0.2.1/configure.ac
--- penggy-0.2.1.orig/configure.ac	2007-04-15 12:23:48.000000000 +0300
+++ penggy-0.2.1/configure.ac	2007-04-15 12:26:04.000000000 +0300
@@ -153,9 +153,6 @@
 if test "$debug" = yes; then
   CFLAGS="$CFLAGS -g"
   AC_DEFINE([DEBUG], 1,[ Enable/disable debugging ])
-else
-  CFLAGS="$CFLAGS -O2"
-  LDFLAGS="$LDFLAGS -s"
 fi
 AM_CONDITIONAL(DEBUG, test "$debug" = yes)
 
