diff -ur agistudio-1.2.2.orig/src/logedit.cpp agistudio-1.2.2/src/logedit.cpp
--- agistudio-1.2.2.orig/src/logedit.cpp	2003-05-17 14:47:13.000000000 +0300
+++ agistudio-1.2.2/src/logedit.cpp	2009-08-09 22:14:15.000000000 +0300
@@ -431,7 +431,7 @@
     editor->setText( filecont );
     fclose(fptr);
     logic->OutputText=editor->text().latin1();
-    if((ptr=strrchr(filename.c_str(),'/')))ptr++;
+    if((ptr=const_cast<char*> (strrchr(filename.c_str(),'/'))))ptr++;
     else ptr=(char *)filename.c_str();
     if(LogicNum!=-1)
       sprintf(tmp,"logic.%d (file %s)",LogicNum,ptr);
@@ -526,7 +526,7 @@
   else if(filename != ""){
     save((char *)filename.c_str());
     char *ptr;
-    if((ptr=strrchr(filename.c_str(),'/')))ptr++;
+    if((ptr=const_cast<char*> (strrchr(filename.c_str(),'/'))))ptr++;
     else ptr=(char *)filename.c_str();
     sprintf(tmp,"File %s",ptr);
     setCaption(tmp);
@@ -644,7 +644,7 @@
           if(winlist[i].type==TEXT){
             filename = winlist[i].w.t->filename;
             char *ptr2;
-            if((ptr2=strrchr(filename.c_str(),'/')))ptr2++;
+            if((ptr2=const_cast<char*> (strrchr(filename.c_str(),'/'))))ptr2++;
             else ptr2=(char *)filename.c_str();            
             if(!strcmp(ptr2,name)){
               int num=atoi(ptr+5);
@@ -665,7 +665,7 @@
               winlist[i].w.t = new TextEdit(NULL,NULL,i);
               winlist[i].type=TEXT;
               winlist[i].w.t->open(fullname);
-              ptr=strstr(tmp1.c_str(),"Line ");
+              ptr=const_cast<char*> (strstr(tmp1.c_str(),"Line "));
               int num=atoi(ptr+5);
               winlist[i].w.t->editor->setCursorPosition(num,0,false);
               ptr1=strchr(ptr,'\n');
@@ -1202,7 +1202,7 @@
   fclose(fptr);
   changed=false;
   char *ptr;
-  if((ptr=strrchr(filename,'/')))ptr++;
+  if((ptr=const_cast<char*> (strrchr(filename,'/'))))ptr++;
   else ptr=(char *)filename;
   sprintf(tmp,"File %s",ptr);
   setCaption(tmp);
