--- configure.ac	2006-02-04 16:04:12.000000000 -0600
+++ configure.ac	2006-02-04 16:09:37.000000000 -0600
@@ -130,14 +130,17 @@
 dnl #
 dnl # Check for Check
 dnl #
-AM_PATH_CHECK(, [have_check="yes"],
-[
-	AC_MSG_WARN([Check not found; cannot run some unit tests])
-	have_check="no"
-])
-
-AM_CONDITIONAL(HAVE_CHECK, test "$have_check" = "yes")
+AC_ARG_ENABLE([tests],
+              AC_HELP_STRING([--disable-tests],[Disable unit tests]),,
+              [enable_tests=yes])
+
+if test "x$enable_tests" = "xyes" ; then
+   AM_PATH_CHECK(, [have_check="yes"], AC_MSG_ERROR([Check not found]))
+else
+   have_check="no"
+fi
 
+AM_CONDITIONAL(HAVE_CHECK, test "$have_check" = "yes")
 
 dnl # Use wall if we have GCC
 if test "x$GCC" = "xyes"; then
