diff --git a/Makefile b/Makefile
index 7892054..938a8c8 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@ PROG_FLGS   = -D BOOLPREDEFINED
 
 SRCS = cluster.cpp DisjointSets.cpp utility.cpp
 
-CFLAGS = $(OPT) $(DEBUG) $(PROG_FLGS)
+CXXFLAGS += $(OPT) $(DEBUG) $(PROG_FLGS)
 
 LFLAGS = 
 
@@ -12,7 +12,7 @@ OBJLIST = cluster.o DisjointSets.o utility.o
 CXX		= g++
 
 .cpp.o:
-	$(CXX) -c $*.cpp $(CFLAGS)
+	$(CXX) $(CXXFLAGS) -c $*.cpp
 
 OPT             = -O3
 DEBUG           = $(CXXDEBUGFLAGS)
@@ -20,10 +20,10 @@ DEBUG           = $(CXXDEBUGFLAGS)
 all:    cluster
 
 cluster:    $(OBJLIST)
-	$(CXX) -o $@ $(OBJLIST) $(LFLAGS)
+	$(CXX) $(LDFLAGS) -o $@ $(OBJLIST)
 
 depend:
-	makedepend -- $(CFLAGS) -- $(SRCS)
+	makedepend -- $(CXXFLAGS) -- $(SRCS)
 
 clean:
 	@rm -rf *.o *.ckp ii_files
