--- tsp-svn-helper/tsh-common.c.orig	2009-04-30 23:03:34.037702354 -0500
+++ tsp-svn-helper/tsh-common.c	2009-04-30 23:03:59.114952043 -0500
@@ -34,6 +34,7 @@
 #include <subversion-1/svn_pools.h>
 #include <subversion-1/svn_fs.h>
 #include <subversion-1/svn_time.h>
+#include <subversion-1/svn_props.h>
 
 #include "tsh-dialog-common.h"
 #include "tsh-login-dialog.h"
--- tsp-svn-helper/tsh-properties-dialog.c.orig        2009-05-07 22:13:48.000000000 +0200
+++ tsp-svn-helper/tsh-properties-dialog.c     2009-05-07 22:13:00.000000000 +0200
@@ -25,6 +25,7 @@
 #include <gtk/gtk.h>

 #include <subversion-1/svn_client.h>
+#include <subversion-1/svn_props.h>

 #include "tsh-common.h"
 #include "tsh-properties-dialog.h"
