--- mozilla-launcher-1.33.multilib	2005-06-28 19:25:05.000000000 -0400
+++ mozilla-launcher-1.33	2005-06-28 19:25:55.000000000 -0400
@@ -135,28 +135,28 @@
   # just drops through for binary-based browsers.
   case $zero in
     *fox)
-      : ${MOZILLA_FIVE_HOME:="/usr/lib/MozillaFirefox"}
+      : ${MOZILLA_FIVE_HOME:="/usr/lib64/MozillaFirefox"}
       remote=$MOZILLA_FIVE_HOME/mozilla-xremote-client
       mozbin=$MOZILLA_FIVE_HOME/firefox-bin
       grepfor=Firefox-bin
       ;;
 
     *mozilla)
-      : ${MOZILLA_FIVE_HOME:="/usr/lib/mozilla"}
+      : ${MOZILLA_FIVE_HOME:="/usr/lib64/mozilla"}
       remote=$MOZILLA_FIVE_HOME/mozilla-xremote-client
       mozbin=$MOZILLA_FIVE_HOME/mozilla-bin
       grepfor=Mozilla-bin
       ;;
 
     *sunbird)
-      : ${MOZILLA_FIVE_HOME:="/usr/lib/MozillaSunbird"}
+      : ${MOZILLA_FIVE_HOME:="/usr/lib64/MozillaSunbird"}
       remote=$MOZILLA_FIVE_HOME/mozilla-xremote-client
       mozbin=$MOZILLA_FIVE_HOME/sunbird-bin
       grepfor=Sunbird-bin
       ;;
 
     *thunderbird)
-      : ${MOZILLA_FIVE_HOME:="/usr/lib/MozillaThunderbird"}
+      : ${MOZILLA_FIVE_HOME:="/usr/lib64/MozillaThunderbird"}
       remote=$MOZILLA_FIVE_HOME/mozilla-xremote-client
       mozbin=$MOZILLA_FIVE_HOME/thunderbird-bin
       grepfor=Thunderbird-bin
