diff -urNp suspend2-userui-0.7.0.org/userui.h suspend2-userui-0.7.0/userui.h
--- suspend2-userui-0.7.0.org/userui.h	2007-01-17 00:16:46.000000000 +0200
+++ suspend2-userui-0.7.0/userui.h	2007-02-11 12:21:22.000000000 +0200
@@ -1,7 +1,7 @@
 #ifndef _USERUI_H_
 #define _USERUI_H_
 
-#include <linux/unistd.h>
+#include <sys/syscall.h>
 #include <sys/types.h>
 #include "suspend_userui.h"
 
@@ -124,11 +124,6 @@ static __inline__ int generic_fls(int x)
 /*
  * Work around random glibc bugs where getpid() caches an invalid pid.
  */
-#ifdef _syscall0
-#define __NR_xgetpid __NR_getpid
-static inline _syscall0(pid_t, xgetpid);
-#else
-static inline pid_t xgetpid(void) { return syscall(__NR_getpid); }
-#endif
+#define xgetpid() syscall(SYS_getpid)
 
 #endif /* _USERUI_H_ */
