--- pyparsing_py3.py
+++ pyparsing_py3.py
@@ -138,10 +138,10 @@
 
 # build list of single arg builtins, tolerant of Python version, that can be used as parse actions
 singleArgBuiltins = []
-import __builtin__
+import builtins
 for fname in "sum len enumerate sorted reversed list tuple set any all".split():
     try:
-        singleArgBuiltins.append(getattr(__builtin__,fname))
+        singleArgBuiltins.append(getattr(builtins,fname))
     except AttributeError:
         continue
 
--- setup.py
+++ setup.py
@@ -23,8 +23,8 @@
         return False
 
 def copyfile(fromname, toname):
-    outf = file(toname,'w')
-    outf.write(file(fromname).read())
+    outf = open(toname,'w')
+    outf.write(open(fromname).read())
     outf.close()
     
 if "MAKING_PYPARSING_RELEASE" not in os.environ and not fileexists("pyparsing.py"):
