--- src/koverartist.desktop.o	2009-02-17 02:35:51.000000000 +0100
+++ src/koverartist.desktop	2009-02-17 02:42:18.000000000 +0100
@@ -1,11 +1,10 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=KoverArtist
 Name[xx]=xxKoverArtistxx
-Exec=koverartist %i %m -caption "%c"
+Exec=koverartist %i -caption "%c"
 Icon=koverartist
 Type=Application
-DocPath=koverartist/koverartist.html
+X-DocPath=koverartist/koverartist.html
 Comment=A KDE KPart Application
 Comment[br]=Ur meziant Kpart evit KDE
 Comment[ca]=Una aplicació KPart per a KDE
@@ -41,3 +40,4 @@
 Comment[xx]=xxA KDE KPart Applicationxx
 Comment[zh_CN]=一个 KDE KPart 应用程序
 Terminal=false
+Categories=Qt;KDE;Graphics;DiscBurning;
--- src/Makefile.am.o	2009-02-17 02:32:07.000000000 +0100
+++ src/Makefile.am	2009-02-17 03:10:29.000000000 +0100
@@ -34,13 +34,10 @@
 
 EXTRA_DIST = $(cases_DATA)
 
-kdelnkdir   = $(kde_appsdir)/Multimedia
-kdelnk_DATA = koverartist.desktop
-
 rcdir = $(kde_datadir)/koverartist
 rc_DATA = koverartistui.rc
 
-appsdir = $(kde_appsdir)/Multimedia
+appsdir = $(xdg_appsdir)
 apps_DATA = koverartist.desktop
 
 mimedir = $(kde_mimedir)/application
