diff -uNp --exclude='*.orig' --exclude='*.rej' --exclude='*.dmn' -r NVIDIA-Linux-x86-1.0-6111-pkg1/usr/src/nv/nv.c NVIDIA-Linux-x86-1.0-6111-pkg1-new/usr/src/nv/nv.c
--- NVIDIA-Linux-x86-1.0-6111-pkg1/usr/src/nv/nv.c	2004-08-31 22:52:17.000000000 +0100
+++ NVIDIA-Linux-x86-1.0-6111-pkg1-new/usr/src/nv/nv.c	2004-09-20 02:49:42.399538232 +0100
@@ -19,6 +19,10 @@
 MODULE_ALIAS_CHARDEV_MAJOR(NV_MAJOR_DEVICE_NUMBER);
 #endif
 
+#ifndef __VMALLOC_RESERVE
+unsigned int __VMALLOC_RESERVE;
+#endif
+
 /*
  * our global state; one per device
  */
