diff -Naur apache_1.3.29/src/modules/standard/mod_auth_db.c apache_1.3.29_gentoo/src/modules/standard/mod_auth_db.c
--- apache_1.3.29/src/modules/standard/mod_auth_db.c	2003-02-03 12:13:27.000000000 -0500
+++ apache_1.3.29_gentoo/src/modules/standard/mod_auth_db.c	2004-04-01 08:47:58.191511608 -0500
@@ -172,7 +172,11 @@
 
 #if defined(DB3) || defined(DB4)
     if (   db_create(&f, NULL, 0) != 0 
+#if DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR > 0
+	|| f->open(f, NULL, auth_dbpwfile, NULL, DB_HASH, DB_RDONLY, 0664) != 0) {
+#else
         || f->open(f, auth_dbpwfile, NULL, DB_HASH, DB_RDONLY, 0664) != 0) {
+#endif
 #elif defined(DB2)
     if (db_open(auth_dbpwfile, DB_HASH, DB_RDONLY, 0664, NULL, NULL, &f) != 0) {
 #else
