From 3c847af770b51ef618e1e636e157e7beccaf046e Mon Sep 17 00:00:00 2001
From: Tomas Chvatal <scarabeus@gentoo.org>
Date: Sun, 3 Jan 2010 22:20:50 +0100
Subject: [PATCH] .la file is totaly useless for python bindings and could even make issues with crosscompilation, thus remove.

---
 src/Makefile.am |   11 -----------
 1 files changed, 0 insertions(+), 11 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index dc8fbae..2ba4c55 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -7,17 +7,6 @@ EXTENSION_XML = xproto.xml \
 
 pkgpythondir = $(pyexecdir)/xcb
 
-pkgpython_LTLIBRARIES = xcb.la
-
-xcb_la_LIBADD = $(LIBXCB_LIBS)
-xcb_la_CPPFLAGS = -I$(PYTHON_INCLUDE)
-xcb_la_CFLAGS = -g $(CWARNFLAGS)
-xcb_la_LDFLAGS = -module
-xcb_la_SOURCES = conn.c constant.c cookie.c error.c event.c except.c \
-		 ext.c extkey.c iter.c list.c module.c protobj.c \
-		 reply.c request.c response.c struct.c union.c void.c \
-		 py_client.py
-
 noinst_HEADERS = conn.h constant.h cookie.h error.h event.h except.h \
 		 ext.h extkey.h iter.h list.h module.h protobj.h \
 		 reply.h request.h response.h struct.h union.h void.h
-- 
1.6.6

