--- src/CMakeLists.txt.old	2007-12-23 03:23:48.000000000 +0100
+++ src/CMakeLists.txt	2007-12-23 03:25:36.000000000 +0100
@@ -73,15 +73,15 @@
     ADD_DEPENDENCIES(rosegardensequencer precompiled_headers)
 ENDIF(USE_PCH)
 
-TARGET_LINK_LIBRARIES(rosegardensequencer 
-    ${QT_AND_KDECORE_LIBS} 
-    ${ALSA_LIBS} 
+TARGET_LINK_LIBRARIES(rosegardensequencer
+    RosegardenCommon
+    ${QT_AND_KDECORE_LIBS}
+    ${ALSA_LIBS}
     ${JACK_LIBS}
     ${LIBLO_LIBS}
     ${LRDF_LIBS}
     ${FFTW3F_LIBS}
     kdeui
-    RosegardenCommon
 )
 
 # gui executable
@@ -135,7 +135,11 @@
 ENDIF(USE_PCH)
 
 TARGET_LINK_LIBRARIES(rosegarden
-    ${QT_AND_KDECORE_LIBS} 
+    RosegardenExtended
+    RosegardenSegmentCanvas
+    RosegardenCommands
+    RosegardenCommon
+    ${QT_AND_KDECORE_LIBS}
     ${LIBLO_LIBS}
     ${LRDF_LIBS}
     ${FFTW3F_LIBS}
@@ -145,10 +149,6 @@
     kio
     kdeui
     kdeprint
-    RosegardenExtended
-    RosegardenSegmentCanvas
-    RosegardenCommands
-    RosegardenCommon
 )
 
 # Install targets
