--- scons_admin/kde.py.copy	2006-02-20 23:17:47.000000000 +0100
+++ scons_admin/kde.py	2006-02-20 23:19:17.000000000 +0100
@@ -533,7 +533,7 @@
 	def KDEinstallas(lenv, restype, destfile, file):
 		if not env['_INSTALL']: return
 		dir = getInstDirForResType(lenv, restype)
-		install_list = lenv.InstallAs(lenv.join(dir, destfile), file)
+		install_list = lenv.InstallAs(lenv.join(env['DESTDIR'], dir, destfile), file)
                 env.Alias('install', install_list)
 		return install_list
 
