--- src/dlgProgress.cpp.orig	2006-10-03 16:35:35.000000000 +0300
+++ src/dlgProgress.cpp	2006-10-03 16:35:44.000000000 +0300
@@ -315,7 +315,7 @@
  */
 bool dlgProgress::Update(int value, const wxString& newmsg)
 {
-  wxASSERT_MSG(state != paused, wxT("paused state isn't handled correctly"));
+  wxASSERT_MSG(state != Paused, wxT("paused state isn't handled correctly"));
   wxASSERT_MSG(value == -1 || gauProgress, wxT("cannot update non existent dialog"));
 
   #ifdef __WXMSW__
