Только в fotoxx-11.01.orig/: .pc
diff -r -U2 fotoxx-11.01.orig//Makefile fotoxx-11.01//Makefile
--- fotoxx-11.01.orig//Makefile	2011-01-01 19:10:53.000000000 +0600
+++ fotoxx-11.01//Makefile	2011-01-04 20:56:03.000000000 +0600
@@ -6,14 +6,14 @@
 
 # defaults for parameters that may be pre-defined
-CXXFLAGS ?= -O3 -g -Wall
-LDFLAGS ?= -O3 -g -Wall -rdynamic
+CXXFLAGS += -Wall
+LDFLAGS += -rdynamic
 PREFIX ?= /usr/local
 
-CFLAGS = $(CXXFLAGS) -c `pkg-config --cflags gtk+-2.0`
-LFLAGS = $(LDFLAGS) `pkg-config --libs gtk+-2.0 gthread-2.0`
+CXXFLAGS += -c `pkg-config --cflags gtk+-2.0`
+LIBS = `pkg-config --libs gtk+-2.0 gthread-2.0`
 
 BINDIR = $(PREFIX)/bin
 DATADIR = $(PREFIX)/share/$(PROGRAM)
-DOCDIR = $(PREFIX)/share/doc/$(PROGRAM)
+DOCDIR = $(PREFIX)/share/doc/$(PF)
 MANDIR = $(PREFIX)/share/man/man1
 ICONDIR = $(DATADIR)/icons
@@ -21,15 +21,14 @@
 
 $(PROGRAM): $(PROGRAM).o zfuncs.o
-	$(CXX) $(LFLAGS) -ltiff          \
-	   -o $(PROGRAM) $(PROGRAM).o zfuncs.o
+	$(CXX) $(LDFLAGS) -o $(PROGRAM) $(PROGRAM).o zfuncs.o -ltiff $(LIBS)
 	@ ./dependencies.sh
 
 $(PROGRAM).o: $(SOURCE)
-	$(CXX) $(CFLAGS) -o $(PROGRAM).o $(SOURCE)   \
+	$(CXX) $(CXXFLAGS) -o $(PROGRAM).o $(SOURCE) \
 	   -D "DATADIR=\"$(DATADIR)\""               \
 	   -D "DOCDIR=\"$(DOCDIR)\""
 
 zfuncs.o: zfuncs.cc
-	$(CXX) $(CFLAGS) zfuncs.cc	         \
+	$(CXX) $(CXXFLAGS) zfuncs.cc        \
 	   -D "DATADIR=\"$(DATADIR)\""      \
 	   -D "DOCDIR=\"$(DOCDIR)\""        \
