--- /tmp/SConstruct	2006-11-24 13:55:00.000000000 +0100
+++ SConstruct	2006-11-24 13:58:22.000000000 +0100
@@ -11,8 +11,8 @@
 #######################################
 # ENVIRONMENT
 #######################################
-env = Environment( CCFLAGS = [ '-Wall', '-g' ],
-		   CXXFLAGS = [ '-Wall', '-g' ],
+env = Environment( CCFLAGS = [ os.environ.get('CFLAGS', '').split() ],
+		   CXXFLAGS = [ os.environ.get('CXXFLAGS', '').split() ],
 		   CPPPATH = [ '/usr/local/include', '/usr/X11R6/include' ],
 		   LIBPATH = [ '/usr/X11R6/lib', '/usr/local/lib' ],
 		   options = opts )
