--- a/gaia-0.1.2/programs/gaia/SConscript	2006-12-06 12:09:48.000000000 +0100
+++ b/gaia-0.1.2/programs/gaia/SConscript	2006-12-06 12:10:17.000000000 +0100
@@ -27,43 +27,6 @@
 	config = env.Configure(log_file = '/dev/null')
 	config_ok = True
 
-	if not config.CheckHeader('SDL.h'):
-		print "You need to install SDL development libraries"
-		config_ok = False
-
-	if not config.CheckHeader('GL/gl.h'):
-		print "You need to install GL development libraries"
-		config_ok = False
-
-	if not config.CheckHeader( [ 'stdio.h', 'jpeglib.h' ] ):
-		print "You need to install jpeg development libraries"
-		config_ok = False
-
-	if not config.CheckHeader('png.h'):
-		print "You need to install png development libraries"
-		config_ok = False
-
-	if int(env['gpsd']) and not config.CheckHeader('gps.h'):
-		print "You need to install libgps from gpsd package (http://gpsd.berlios.de/), or turn off gpsd support with `gpsd=0' option (you'll still be able to use GPS)"
-		config_ok = False
-
-
-	if not config.CheckLib('curl'):
-		print "You need to install libcurl development libraries"
-		config_ok = False
-
-	if not config.CheckLib('jpeg'):
-		print "You need to install jpeg development libraries"
-		config_ok = False
-
-	if not config.CheckLib('png'):
-		print "You need to install png development libraries"
-		config_ok = False
-
-	if int(env['gpsd']) and not config.CheckLib('gps'):
-		print "You need to install libgps from gpsd package (http://gpsd.berlios.de/), or turn off gpsd support with `gpsd=0' option (you'll still be able to use GPS)"
-		config_ok = False
-
 	if not config_ok:
 		Exit(1)
 
