--- config.mk.orig	Mon Aug 20 12:20:58 2001
+++ config.mk	Mon Aug 20 14:05:35 2001
@@ -1,17 +1,12 @@
-PREFIX=/usr/local
+PREFIX=/usr
 PLUGINDIR=$(HOME)/.skipstone/plugins
 
 MOZILLA_INCLUDES = \
-	-I/usr/include/mozilla \
-	-I/usr/local/include/mozilla \
-	-I/usr/include/nspr \
-	-I/usr/local/include/nspr \
-	-I/usr/include/mozilla/nspr \
-	-I/usr/local/include/mozilla/nspr
-
+	-I/opt/mozilla/include \
+	-I/opt/mozilla/include/nspr
+	
 MOZILLA_LIBS = \
-	-L/usr/lib/mozilla \
-	-L/usr/local/lib/mozilla \
+	-L/opt/mozilla \
 	-lgtkembedmoz -lplds4 -lplc4 -lnspr4 -lgtksuperwin -lxpcom
 
 GTK_LIBS = $(shell gtk-config --libs)
