diff -Naur fwbuilder-3.0.7.orig/src/ipt/RoutingCompiler_ipt_writers.cpp fwbuilder-3.0.7/src/ipt/RoutingCompiler_ipt_writers.cpp
--- fwbuilder-3.0.7.orig/src/ipt/RoutingCompiler_ipt_writers.cpp	2009-11-12 09:54:55.913414002 +0100
+++ fwbuilder-3.0.7/src/ipt/RoutingCompiler_ipt_writers.cpp	2009-11-12 10:06:38.086706330 +0100
@@ -147,9 +147,9 @@
             << "#\n#\n# ============== ROUTING RULES ============== \n#"
             << endl;
 
-        compiler->output << "TMPDIRNAME=\"/tmp/.fwbuilder.tempdir.$$\"" << endl;
+        compiler->output << "TMPDIRNAME=\"$(umask 077 && mktemp --tmpdir=/tmp -d .fwbuilder.tempdir.XXXXXXXXXX)\"" << endl;
+        compiler->output << "[ $? != 0 ] && exit 1" << endl;
         compiler->output << "TMPFILENAME=\"$TMPDIRNAME/.fwbuilder.out\"" << endl;
-        compiler->output << "(umask 077 && mkdir $TMPDIRNAME) || exit 1" << endl;
         compiler->output << endl;
 
         compiler->output << "# This function stops stdout redirection"
