diff -Naurp fwbuilder-4.0.1.orig//src/res/configlets/linux24/routing_functions fwbuilder-4.0.1//src/res/configlets/linux24/routing_functions
--- fwbuilder-4.0.1.orig//src/res/configlets/linux24/routing_functions	2009-12-17 23:12:01.000000000 -0500
+++ fwbuilder-4.0.1//src/res/configlets/linux24/routing_functions	2010-06-05 14:04:15.735061939 -0400
@@ -14,9 +14,9 @@
 
 # ============== ROUTING RULES ============== 
 
-TMPDIRNAME="/tmp/.fwbuilder.tempdir.$$"
+TMPDIRNAME="$(umask 077 && mktemp --tmpdir=/tmp -d .fwbuilder.tempdir.XXXXXXXXXX)"
+[ $? != 0 ] && exit 1
 TMPFILENAME="$TMPDIRNAME/.fwbuilder.out"
-(umask 077 && mkdir $TMPDIRNAME) || exit 1
 
 #
 # This function stops stdout redirection
