--- McCLIM/system.lisp.original	2004-03-04 14:54:12.750240760 -0600
+++ McCLIM/system.lisp	2004-03-04 14:58:50.797971080 -0600
@@ -25,15 +25,12 @@
 
 #+cmu
 (progn
-  (unless (fboundp 'ext:stream-read-char)
-    (unless (ignore-errors (ext:search-list "gray-streams:"))
-      (setf (ext:search-list "gray-streams:")
-	'("target:pcl/" "library:subsystems/")))
-    (load "gray-streams:gray-streams-library"))
-  #-clx
-  (require :clx)
-  #-mk-defsystem
-  (load "library:subsystems/defsystem"))
+  (require :cmucl-graystream)
+  (require :cmucl-clx))
+
+#+sbcl 
+(progn 
+  (require :clx))
 
 (pushnew :clim *features*)
 (pushnew :mcclim *features*)
