--- kmyfirewall/kmfwidgets/kmfsystemsettingslinux.cpp.orig	2008-05-02 10:50:24.000000000 +0200
+++ kmyfirewall/kmfwidgets/kmfsystemsettingslinux.cpp	2008-05-02 12:08:26.000000000 +0200
@@ -57,6 +57,12 @@
 		this, SLOT( slotSettingChanged() ) );
 	connect( m_cob_distribution, SIGNAL( activated( const QString& ) ),
 		this, SLOT( slotDistributionChanged() ) );
+  
+	m_url_ipt_path->setEnabled( false );
+	m_url_mod_path->setEnabled( false );
+	m_url_init_path->setEnabled( false );
+	m_url_runlevel_path->setEnabled( false );
+	m_cob_distribution->setEnabled( false );
 }
 
 KMFSystemSettingsLinux::~KMFSystemSettingsLinux()
@@ -77,6 +83,11 @@
 		kdDebug() << "KMFInterfaceWidget::updateView() - ERROR: m_target == 0 " << endl;
 		return;
 	}
+	m_url_ipt_path->setEnabled( true );
+	m_url_mod_path->setEnabled( true );
+	m_url_init_path->setEnabled( true );
+	m_url_runlevel_path->setEnabled( true );
+	m_cob_distribution->setEnabled( true ); 
 	KMFTargetConfig *conf = m_target->config();
 
 	kdDebug() << "KMFInterfaceWidget::updateView() - config: " << endl;
--- kmyfirewall/kmfwidgets/kmfinterfacewidget.cpp.orig	2008-05-02 10:26:05.000000000 +0200
+++ kmyfirewall/kmfwidgets/kmfinterfacewidget.cpp	2008-05-02 12:08:51.000000000 +0200
@@ -53,6 +53,10 @@
 		m_cb_int_name->insertItem( "tap" );
 		m_cb_int_name->insertItem( "gre" );
 		m_cb_int_name->insertItem( "lo" );
+		m_cb_int_name->setEnabled( false );
+		m_sb_int_num->setEnabled( false );
+		m_b_add_int->setEnabled( false );
+		m_b_del_int->setEnabled( false );
 }
 
 KMFInterfaceWidget::~KMFInterfaceWidget() {}
@@ -68,6 +72,10 @@
 	if ( ! m_target ) {
 		return;
 	}
+	m_cb_int_name->setEnabled( true );
+	m_sb_int_num->setEnabled( true );
+	m_b_add_int->setEnabled( true );
+	m_b_del_int->setEnabled( true );
 	KMFTargetConfig *conf = m_target->config();
 	kdDebug() << "Load Config:" << endl;
 	kdDebug() << conf->toString() << endl;
