--- ../resin-3.0.12/configure	2005-01-12 00:00:26.000000000 +0100
+++ configure	2005-04-29 16:18:15.000000000 +0200
@@ -8261,6 +8261,10 @@
  apache_exe="${apache_bin}/httpd"
 elif test -x "${apache_bin}/httpd2"; then
  apache_exe="${apache_bin}/httpd2"
+elif test -x "${apache_bin}/apache"; then
+ apache_exe="${apache_bin}/apache"
+elif test -x "${apache_bin}/apache2"; then
+ apache_exe="${apache_bin}/apache2"
 else
   { { echo "$as_me:$LINENO: error: Can't find Apache binary in directory ${apache_bin}" >&5
 echo "$as_me: error: Can't find Apache binary in directory ${apache_bin}" >&2;}
diff -uprN ../resin-3.0.12/src/c/resin_os/Makefile.in src/c/resin_os/Makefile.in
--- ../resin-3.0.12/src/c/resin_os/Makefile.in	2004-12-01 00:22:38.000000000 +0100
+++ src/c/resin_os/Makefile.in	2005-04-29 16:24:36.000000000 +0200
@@ -13,7 +13,7 @@ resin_libexec=@apache_libexec@
 resin_home=@resin_home@
 JNI_INCLUDE=@JNI_INCLUDE@
 INCLUDES = @INCLUDES@ $(JNI_INCLUDE) -I../common
-CFLAGS = @CFLAGS@ -DRESIN_HOME=\"$(resin_home)\" $(INCLUDES) -DCPU=\"$(CPU)\" -DOS=$(OS)
+CFLAGS = @CFLAGS@ -fPIC -DRESIN_HOME=\"$(resin_home)\" $(INCLUDES) -DCPU=\"$(CPU)\" -DOS=$(OS)
 OBJ= jni_os.o
 
 all	: libresin_os.$(SO)
