--- ./italc-1.0.9/acinclude.m4	2008-05-24 11:19:33.000000000 +0200
+++ ./italc-1.0.9/acinclude.m4	2009-01-31 21:24:54.000000000 +0100
@@ -127,9 +127,9 @@
 	fi
         ;;
     *)
-       	QT_LIBS=`ls $QTDIR/lib64/libQt*.so 2> /dev/null` 
+       	QT_LIBS=`ls $QTDIR/lib64/qt4/libQt*.so 2> /dev/null` 
         if test "x$QT_LIBS" = x;  then
-        	QT_LIBS=`ls $QTDIR/lib/libQt*.so 2> /dev/null` 
+        	QT_LIBS=`ls $QTDIR/lib/qt4/libQt*.so 2> /dev/null` 
             	if test "x$QT_LIBS" = x;  then
 			AC_MSG_ERROR([*** Couldn't find any Qt4 libraries])
             	fi
