diff -ur italc-1.0.9.orig/lib/include/types.h italc-1.0.9/lib/include/types.h
--- italc-1.0.9.orig/lib/include/types.h	2008-02-10 01:03:25.000000000 +0200
+++ italc-1.0.9/lib/include/types.h	2009-07-30 11:03:00.000000000 +0300
@@ -25,6 +25,8 @@
 #ifndef _TYPES_H
 #define _TYPES_H
 
+#include <stdint.h>
+
 typedef signed char             Q_INT8;         /* 8 bit signed */
 typedef unsigned char           Q_UINT8;        /* 8 bit unsigned */
 typedef short                   Q_INT16;        /* 16 bit signed */
diff -ur italc-1.0.9.orig/lib/src/fast_qimage.cpp italc-1.0.9/lib/src/fast_qimage.cpp
--- italc-1.0.9.orig/lib/src/fast_qimage.cpp	2008-03-17 10:02:33.000000000 +0200
+++ italc-1.0.9/lib/src/fast_qimage.cpp	2009-07-30 10:56:19.000000000 +0300
@@ -52,6 +52,7 @@
   pete@shinners.org
 */
 
+#include <stdint.h>
 
 #define ALIGN_SIZE 16
 
