diff -ur italc-1.0.3.orig/configure.in italc-1.0.3/configure.in
--- italc-1.0.3.orig/configure.in	2007-06-04 14:34:19.000000000 +0300
+++ italc-1.0.3/configure.in	2007-11-03 19:06:06.000000000 +0200
@@ -25,7 +25,7 @@
 AC_PROG_LIBTOOL
 
 
-EXTRAFLAGS="-g0 -O2 -Wall"
+EXTRAFLAGS="-fPIC -Wall"
 #-floop-optimize2 -fomit-frame-pointer"
 
 if test "x`$CC --version|head -1|cut -d\  -f3|cut -d. -f1`" = "x4" ; then
diff -ur italc-1.0.3.orig/setup/Makefile.am italc-1.0.3/setup/Makefile.am
--- italc-1.0.3.orig/setup/Makefile.am	2007-05-11 17:35:44.000000000 +0300
+++ italc-1.0.3/setup/Makefile.am	2007-11-03 19:05:45.000000000 +0200
@@ -62,7 +62,7 @@
 	$(WINDRES) -o $@ $<
 PLATFORM_LDADD = setup_win_resources.o -L../lib/ -litalc_core
 else
-PLATFORM_LDADD = ../lib/libitalc_core-static.a
+PLATFORM_LDADD = -L../lib/ -litalc_core
 endif
 
 setup_LDADD = $(QT_LDADD) $(QT_LIB_GUI) $(LIBSSL_LDADD) -ljpeg $(PLATFORM_LDADD)
