diff -urNp orig-Emulator_Src_3.4/BuildUnix/Makefile.am Emulator_Src_3.4/BuildUnix/Makefile.am
--- orig-Emulator_Src_3.4/BuildUnix/Makefile.am	Thu Jan 31 02:55:31 2002
+++ Emulator_Src_3.4/BuildUnix/Makefile.am	Mon Feb 18 10:48:21 2002
@@ -415,7 +415,7 @@ ResStrings.cpp: $(srcdir)/../SrcShared/S
 	perl -x $(srcdir)/../SrcShared/Strings.txt
 
 EmDlgFltkFactory.h EmDlgFltkFactory.cpp : $(srcdir)/../SrcUnix/EmDlgFltkFactory.fl
-	fluid -c $(srcdir)/../SrcUnix/EmDlgFltkFactory.fl
+	$(FLUID) -c $(srcdir)/../SrcUnix/EmDlgFltkFactory.fl
 
 EmDlgFltk.o : EmDlgFltkFactory.h
 
diff -urNp orig-Emulator_Src_3.4/BuildUnix/configure.in Emulator_Src_3.4/BuildUnix/configure.in
--- orig-Emulator_Src_3.4/BuildUnix/configure.in	Thu Jan 31 02:55:25 2002
+++ Emulator_Src_3.4/BuildUnix/configure.in	Mon Feb 18 10:48:21 2002
@@ -34,6 +34,9 @@ AC_ARG_WITH(fltk,
 	[  --with-fltk=DIR         use FLTK in DIR], [
 	CPPFLAGS="${CPPFLAGS} -I$withval/include"
 	LDFLAGS="${LDFLAGS} -L$withval/lib"
+	if test "${FLUID+set}" != set; then
+		test -r $withval/bin/fluid && FLUID=$withval/bin/fluid
+	fi
 ])
 
 
@@ -340,7 +343,8 @@ AC_SUBST(GLLIB)
 
 dnl ========================================================
 dnl Make sure FLTK is available.  It must be installed on
-dnl a standard path, or on $x_libraries.
+dnl a standard path, or on $x_libraries.  Also check that
+dnl fluid is available (in PATH, or via --with-fltk).
 dnl ========================================================
 
 AC_CACHE_SAVE
@@ -355,8 +359,11 @@ AC_CACHE_CHECK([for 'int fl_height (void
 		pose_cv_lib_fltk_found=yes, pose_cv_lib_fltk_found=no))
 
 LIBS="$save_LIBS"
+ 
+dnl $FLUID may have already been set by --with-fltk, above
+AC_CHECK_PROG(FLUID, fluid, fluid, no)
 
-if test $pose_cv_lib_fltk_found = yes; then
+if test $pose_cv_lib_fltk_found = yes && test $FLUID != no; then
 	AC_DEFINE(HAVE_LIBFLTK)
 else
 	AC_MSG_ERROR(*** FLTK *must* be installed before running configure. ***)
