diff -urNp orig-Emulator_Src_3.4/BuildUnix/Gzip/Makefile.am Emulator_Src_3.4/BuildUnix/Gzip/Makefile.am
--- orig-Emulator_Src_3.4/BuildUnix/Gzip/Makefile.am	Thu Jan 31 02:55:25 2002
+++ Emulator_Src_3.4/BuildUnix/Gzip/Makefile.am	Mon Feb 18 10:55:27 2002
@@ -8,7 +8,7 @@
 ##
 SUFFIXES =	.cpp
 
-VPATH =		$(srcdir)/../../SrcShared/Gzip:
+VPATH =		$(srcdir):$(srcdir)/../../SrcShared/Gzip:
 
 ## no subdir specific flags needed -- just go with globals
 LOCAL_CFLAGS =
diff -urNp orig-Emulator_Src_3.4/BuildUnix/Makefile.am Emulator_Src_3.4/BuildUnix/Makefile.am
--- orig-Emulator_Src_3.4/BuildUnix/Makefile.am	Thu Jan 31 02:55:31 2002
+++ Emulator_Src_3.4/BuildUnix/Makefile.am	Mon Feb 18 10:55:27 2002
@@ -8,7 +8,8 @@ SUBDIRS			=	Gzip jpeg espws-2.0
 
 SUFFIXES		=	.cpp
 
-VPATH			=	$(srcdir)/../SrcUnix:					\
+VPATH			=	$(srcdir):								\
+					$(srcdir)/../SrcUnix:					\
 					$(srcdir)/../SrcShared:					\
 					$(srcdir)/../SrcShared/Hardware:		\
 					$(srcdir)/../SrcShared/Hardware/TRG:	\
@@ -402,9 +403,9 @@ endif
 
 bin_PROGRAMS	=	pose
 
-pose_LDADD		=	$(srcdir)/Gzip/libposergzip.a
-pose_LDADD		+=	$(srcdir)/jpeg/libposerjpeg.a
-pose_LDADD		+=	$(srcdir)/espws-2.0/libposerespws.a
+pose_LDADD		=	Gzip/libposergzip.a
+pose_LDADD		+=	jpeg/libposerjpeg.a
+pose_LDADD		+=	espws-2.0/libposerespws.a
 pose_LDADD		+=	-lfltk
 pose_LDADD		+=	-L/usr/local/lib $(X_LIBS) -lXext -lX11 $(THREAD_LIBS) $(GLLIB) -lm
 
diff -urNp orig-Emulator_Src_3.4/BuildUnix/espws-2.0/Makefile.am Emulator_Src_3.4/BuildUnix/espws-2.0/Makefile.am
--- orig-Emulator_Src_3.4/BuildUnix/espws-2.0/Makefile.am	Thu Jan 31 02:55:25 2002
+++ Emulator_Src_3.4/BuildUnix/espws-2.0/Makefile.am	Mon Feb 18 10:55:27 2002
@@ -9,7 +9,7 @@
 ##
 SUFFIXES		=	.cxx
 
-VPATH			=	$(srcdir)/../../SrcUnix/espws-2.0:
+VPATH			=	$(srcdir):$(srcdir)/../../SrcUnix/espws-2.0:
 
 CFLAGS			=   $(POSER_CFLAGS) $(LOCAL_CFLAGS) $(X_CFLAGS)
 CXXFLAGS		= 	$(POSER_CXXFLAGS) $(LOCAL_CFLAGS) $(X_CFLAGS)
diff -urNp orig-Emulator_Src_3.4/BuildUnix/jpeg/Makefile.am Emulator_Src_3.4/BuildUnix/jpeg/Makefile.am
--- orig-Emulator_Src_3.4/BuildUnix/jpeg/Makefile.am	Thu Jan 31 02:55:25 2002
+++ Emulator_Src_3.4/BuildUnix/jpeg/Makefile.am	Mon Feb 18 10:55:27 2002
@@ -9,7 +9,7 @@
 ##
 SUFFIXES		=	.cpp
 
-VPATH			=	$(srcdir)/../../SrcShared/jpeg:
+VPATH			=	$(srcdir):$(srcdir)/../../SrcShared/jpeg:
 
 ## no subdir specific flags needed -- just go with globals
 LOCAL_CFLAGS	=
