--- opera-10.10-4566.gcc4-qt4.i386/etc/operaprefs_default.ini.orig	2009-08-21 13:26:52.000000000 +0200
+++ opera-10.10-4566.gcc4-qt4.i386/etc/operaprefs_default.ini	2009-08-29 19:05:58.000000000 +0200
@@ -1,7 +1,7 @@
 ; Put any default settings here that are overridable by users
 
 [User Prefs]
-Language Files Directory=/usr/share/opera/locale
+Language Files Directory=/opt/opera/share/opera/locale
 Force DPI=96
 
 [Auto Update]
