Index: fdi/90defaultpolicy/storage-policy.fdi
===================================================================
RCS file: /cvs/hal/hal/fdi/90defaultpolicy/storage-policy.fdi,v
retrieving revision 1.4
diff -u -r1.4 storage-policy.fdi
--- fdi/90defaultpolicy/storage-policy.fdi	14 Oct 2004 21:58:55 -0000	1.4
+++ fdi/90defaultpolicy/storage-policy.fdi	15 Oct 2004 20:43:16 -0000
@@ -83,13 +83,10 @@
 	  <merge key="volume.policy.should_mount" type="bool">true</merge>
 	  <merge key="volume.policy.mount_filesystem" type="copy_property">volume.fstype</merge>
 	  
-	  <!-- Fallback is '<storage.drive_type>', e.g. disk, compact_flash etc. -->
-	  <merge key="volume.policy.desired_mount_point" type="copy_property">@block.storage_device:storage.drive_type</merge>
-	  
-	  <!-- Better: if available use UUID -->
-          <match key="volume.uuid" empty="false">
-            <merge key="volume.policy.desired_mount_point" type="copy_property">volume.uuid</merge>
-          </match>
+	  <!-- Fallback is '<storage.bus>', appended with 'disk', e.g. usbdisk,
+	       idedisk, scsidisk etc. -->
+	  <merge key="volume.policy.desired_mount_point" type="copy_property">@block.storage_device:storage.bus</merge>
+	  <append key="volume.policy.desired_mount_point" type="string">disk</append>
 	  
           <!-- Best: If available use filesystem label -->
           <match key="volume.label" empty="false">
