--- tools/hal-system-power-suspend	2006-02-19 00:00:55.000000000 +0100
+++ tools/hal-system-power-suspend2	2006-09-23 11:22:42.000000000 +0200
@@ -62,6 +62,9 @@
 	elif [ -x "/usr/sbin/pmi" ] ; then
 	    /usr/sbin/pmi action suspend force
 	    RET=$?
+	elif [ -x "/usr/sbin/hibernate-ram" ] ; then
+	    /usr/sbin/hibernate-ram
+	    RET=$?
 	elif [ -w "/sys/power/state" ] ; then
 	    # Use the raw kernel sysfs interface
 	    echo "mem" > /sys/power/state
