--- src/freepops.c.orig	2007-05-11 07:33:04.000000000 +1000
+++ src/freepops.c	2007-05-11 07:35:02.000000000 +1000
@@ -452,8 +452,12 @@
 
 
 #if CRYPTO_IMPLEMENTATION == 1
+	if (!gcry_check_version("1.2.2")) {
+		fprintf(stderr, "gcrypt version older than 1.2.2\n");
+		exit(1);
+	}
 	gcry_control(GCRYCTL_SET_THREAD_CBS,&gcry_threads_pthread);
-	gcry_check_version("1.2.2");
+	gcry_control (GCRYCTL_DISABLE_SECMEM, 0); 
 #endif
 	
 	curl_global_init(CURL_GLOBAL_ALL);
