--- lib/Makefile.am
+++ lib/Makefile.am
@@ -1,7 +1,7 @@
 
 INCLUDES = -I$(top_srcdir)/lib
 
-AM_CXXFLAGS =-O3 -Wall -fno-rtti -fno-exceptions -g3
+AM_CXXFLAGS = -Wall -fno-rtti -fno-exceptions
 
 flash_LTLIBRARIES = \
 	libflash.la
--- plugin/Makefile.am
+++ plugin/Makefile.am
@@ -10,21 +10,11 @@
 #		api gets control.
 #          
 
-
 PLUGIN_DEFINES= -DXP_UNIX -DPLUGIN_TRACE
-OPTIMIZER=$(OPTFLAGS)
-SHARED=-shared
 
 libnpflash_la_SOURCES = plugin.c npunix.c
-libnpflash_la_CFLAGS = -fpic -fPIC -DPIC $(OPTIMIZER) $(PLUGIN_DEFINES) -I../lib $(X_CFLAGS)
-libnpflash_la_LDFLAGS = -L../lib/.libs  -lflash
-
-install-pluginLTLIBRARIES: $(plugin_LIBRARIES)
-	mkdir -p "$(DESTDIR)$(PLUGIN_DIR)"
-	cp $(srcdir)/.libs/libnpflash.so.0.0.0 "$(DESTDIR)$(PLUGIN_DIR)/libnpflash.so"
-
-uninstall-pluginLTLIBRARIES: $(plugin_LIBRARIES)
-	rm -f "$(DESTDIR)$(PLUGIN_DIR)/libnpflash.so"
+libnpflash_la_CFLAGS = $(PLUGIN_DEFINES) -I$(top_builddir)/lib $(X_CFLAGS)
+libnpflash_la_LIBADD = $(top_builddir)/lib/libflash.la
 
 #	$(LIBTOOL) --mode=install -c .libs/libnpflash.so $(PLUGIN_DIR)
 
