diff -uNr lua-5.0.1.orig/Makefile lua-5.0.1/Makefile
--- lua-5.0.1.orig/Makefile	2003-12-19 23:41:25.394200808 +0000
+++ lua-5.0.1/Makefile	2003-12-19 23:42:39.829884864 +0000
@@ -38,8 +38,8 @@
 
 # shared libraries (for Linux)
 so:
-	ld -o lib/liblua.so.$V -shared src/*.o
-	ld -o lib/liblualib.so.$V -shared src/lib/*.o
+	gcc -o lib/liblua.so.$V -shared src/*.o
+	gcc -o lib/liblualib.so.$V -shared src/lib/*.o
 	cd lib; ln -fs liblua.so.$V liblua.so; ln -fs liblualib.so.$V liblualib.so
 
 # binaries using shared libraries
diff -uNr lua-5.0.1.orig/src/Makefile lua-5.0.1/src/Makefile
--- lua-5.0.1.orig/src/Makefile	2003-12-19 23:41:25.374203848 +0000
+++ lua-5.0.1/src/Makefile	2003-12-19 23:42:07.303829576 +0000
@@ -4,6 +4,8 @@
 
 include $(LUA)/config
 
+CFLAGS= $(MYCFLAGS) $(PICFLAGS) $(WARN) $(INCS) $(DEFS)
+
 OBJS=	\
 	lapi.o \
 	lcode.o \
diff -uNr lua-5.0.1.orig/src/lib/Makefile lua-5.0.1/src/lib/Makefile
--- lua-5.0.1.orig/src/lib/Makefile	2003-12-19 23:41:25.368204760 +0000
+++ lua-5.0.1/src/lib/Makefile	2003-12-19 23:42:22.414532400 +0000
@@ -4,6 +4,8 @@
 
 include $(LUA)/config
 
+CFLAGS= $(MYCFLAGS) $(PICFLAGS) $(WARN) $(INCS) $(DEFS)
+
 EXTRA_DEFS= $(POPEN) $(TMPNAM) $(DEGREES) $(LOADLIB)
 
 OBJS= lauxlib.o lbaselib.o ldblib.o liolib.o lmathlib.o ltablib.o lstrlib.o loadlib.o
