--- Makefile.old	2006-10-15 19:11:53.684244053 +0200
+++ Makefile	2006-10-15 19:13:19.637126651 +0200
@@ -30,11 +30,11 @@
 
 # official installation
 install: all strip
-	mkdir -p $(INSTALL_BIN) $(INSTALL_INC) $(INSTALL_LIB) $(INSTALL_MAN)
-	$(INSTALL_EXEC) bin/* $(INSTALL_BIN)
-	$(INSTALL_DATA) include/*.h $(INSTALL_INC)
-	$(INSTALL_DATA) lib/*.a $(INSTALL_LIB)
-	$(INSTALL_DATA) doc/*.1 $(INSTALL_MAN)
+	mkdir -p $(DESTDIR)$(INSTALL_BIN) $(DESTDIR)$(INSTALL_INC) $(DESTDIR)$(INSTALL_LIB) $(DESTDIR)$(INSTALL_MAN)
+	$(INSTALL_EXEC) bin/* $(DESTDIR)$(INSTALL_BIN)
+	$(INSTALL_DATA) include/*.h $(DESTDIR)$(INSTALL_INC)
+	$(INSTALL_DATA) lib/*.a $(DESTDIR)$(INSTALL_LIB)
+	$(INSTALL_DATA) doc/*.1 $(DESTDIR)$(INSTALL_MAN)
 
 # shared libraries (for Linux)
 so:
@@ -50,8 +50,8 @@
 
 # install shared libraries
 soinstall:
-	$(INSTALL_EXEC) lib/*.so.* $(INSTALL_LIB)
-	cd $(INSTALL_LIB); ln -fs liblua.so.$V liblua.so; ln -fs liblualib.so.$V liblualib.so
+	$(INSTALL_EXEC) lib/*.so.* $(DESTDIR)$(INSTALL_LIB)
+	cd $(DESTDIR)$(INSTALL_LIB); ln -fs liblua.so.$V liblua.so; ln -fs liblualib.so.$V liblualib.so
 
 # clean shared libraries
 soclean:
