diff -Naur gnome-print-0.37.origin/installer/gf-pfb.c gnome-print-0.37/installer/gf-pfb.c
--- gnome-print-0.37.origin/installer/gf-pfb.c	2002-01-19 01:23:00.000000000 +0100
+++ gnome-print-0.37/installer/gf-pfb.c	2004-03-08 19:40:35.861812512 +0100
@@ -12,7 +12,8 @@
 #include <fcntl.h>
 #include <string.h>
 #include <ctype.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include "gf-pfb.h"
 
 #define noVERBOSE
diff -Naur gnome-print-0.37.origin/installer/gf-ttf.c gnome-print-0.37/installer/gf-ttf.c
--- gnome-print-0.37.origin/installer/gf-ttf.c	2001-11-05 15:07:28.000000000 +0100
+++ gnome-print-0.37/installer/gf-ttf.c	2004-03-08 19:42:51.483194928 +0100
@@ -32,7 +32,8 @@
 #include <fcntl.h>
 #include <string.h>
 #include <ctype.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 
 #include "gf-ttf.h"
 
diff -Naur gnome-print-0.37.origin/installer/gnome-font-install.c gnome-print-0.37/installer/gnome-font-install.c
--- gnome-print-0.37.origin/installer/gnome-font-install.c	2002-01-22 22:44:55.000000000 +0100
+++ gnome-print-0.37/installer/gnome-font-install.c	2004-03-08 19:43:31.858057016 +0100
@@ -26,7 +26,8 @@
 #include <parser.h>
 #include <xmlmemory.h>
 /* End of ugly thing */
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <libgnome/gnome-defs.h>
 #include <libgnome/gnome-util.h>
 #include <libgnome/gnome-i18n.h>
diff -Naur gnome-print-0.37.origin/libgnomeprint/gnome-font-face.c gnome-print-0.37/libgnomeprint/gnome-font-face.c
--- gnome-print-0.37.origin/libgnomeprint/gnome-font-face.c	2002-09-15 22:49:12.000000000 +0200
+++ gnome-print-0.37/libgnomeprint/gnome-font-face.c	2004-03-08 19:46:11.832737168 +0100
@@ -10,7 +10,8 @@
 #include <fcntl.h>
 #include <stdio.h>
 #include <ctype.h>
-#include <freetype/ftoutln.h>
+#include <ft2build.h>
+#include FT_OUTLINE_H
 #include <locale.h>
 
 #include <libgnomeprint/gnome-print-i18n.h>
diff -Naur gnome-print-0.37.origin/libgnomeprint/gnome-font-private.h gnome-print-0.37/libgnomeprint/gnome-font-private.h
--- gnome-print-0.37.origin/libgnomeprint/gnome-font-private.h	2001-12-05 23:07:58.000000000 +0100
+++ gnome-print-0.37/libgnomeprint/gnome-font-private.h	2004-03-08 19:46:47.087377648 +0100
@@ -22,7 +22,8 @@
 
 BEGIN_GNOME_DECLS
 
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <libgnomeprint/gp-character-block.h>
 #include <libgnomeprint/gp-fontmap.h>
 #include <libgnomeprint/gnome-font.h>
diff -Naur gnome-print-0.37.origin/libgnomeprint/parseTT.c gnome-print-0.37/libgnomeprint/parseTT.c
--- gnome-print-0.37.origin/libgnomeprint/parseTT.c	2002-08-27 20:46:20.000000000 +0200
+++ gnome-print-0.37/libgnomeprint/parseTT.c	2004-03-08 19:45:02.876220152 +0100
@@ -107,9 +107,10 @@
 #include <config.h>
 
 #include <parseTT.h>
-#include <freetype/freetype.h>
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_GLYPH_H
+#include FT_OUTLINE_H
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
diff -Naur gnome-print-0.37.origin/libgnomeprint/parseTT.h gnome-print-0.37/libgnomeprint/parseTT.h
--- gnome-print-0.37.origin/libgnomeprint/parseTT.h	2001-11-05 20:27:09.000000000 +0100
+++ gnome-print-0.37/libgnomeprint/parseTT.h	2004-03-08 19:45:32.614699216 +0100
@@ -22,7 +22,8 @@
 BEGIN_GNOME_DECLS
 
 #include <glib.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 
 /*
  * Convert loaded Freetype TTF face to Type1
