diff -ruN OpenCV-2.2.0.orig/CMakeLists.txt OpenCV-2.2.0/CMakeLists.txt
--- OpenCV-2.2.0.orig/CMakeLists.txt	2011-07-13 21:49:54.474595691 +0200
+++ OpenCV-2.2.0/CMakeLists.txt	2011-07-13 21:51:01.917595716 +0200
@@ -196,10 +196,30 @@
 # ----------------------------------------------------------------------------
 #       CHECK FOR SYSTEM LIBRARIES, OPTIONS, ETC..
 # ----------------------------------------------------------------------------
+find_package(PkgConfig REQUIRED)
+
+if(BLAS_LIBRARIES)
+  include(CheckFortranFunctionExists)
+  set(CMAKE_REQUIRED_LIBRARIES ${BLAS_LIBRARIES})
+  CHECK_FORTRAN_FUNCTION_EXISTS("dgemm" BLAS_FOUND)
+  unset( CMAKE_REQUIRED_LIBRARIES )
+  if(BLAS_FOUND)
+    message(STATUS "BLAS supplied by user is WORKING, will use ${BLAS_LIBRARIES}.")
+  else(BLAS_FOUND)
+    message(FATAL_ERROR "BLAS supplied by user is not WORKING, CANNOT USE ${BLAS_LIBRARIES}.")
+  endif(BLAS_FOUND)
+else(BLAS_LIBRARIES)
+   # User did not provide a BLAS Library
+   pkg_check_modules(BLAS blas)
+endif (BLAS_LIBRARIES)
+
+if(NOT BLAS_FOUND)
+    message(FATAL_ERROR "BLAS libraries not found.")
+endif(NOT BLAS_FOUND)
+
 find_package(LAPACK REQUIRED)
-find_package(BLAS REQUIRED)
 find_package(ZLIB REQUIRED)
-find_package(PkgConfig REQUIRED)
+
 pkg_check_modules(FLANN REQUIRED flann)
 
 # Build/install (or not) some apps:
