diff -ur OpenCV-2.1.0.orig/3rdparty/flann/CMakeLists.txt OpenCV-2.1.0/3rdparty/flann/CMakeLists.txt
--- OpenCV-2.1.0.orig/3rdparty/flann/CMakeLists.txt	2010-04-06 04:24:34.000000000 +0300
+++ OpenCV-2.1.0/3rdparty/flann/CMakeLists.txt	2010-07-27 21:41:18.000000000 +0300
@@ -89,8 +89,8 @@
 INSTALL (
     TARGETS flann  # flann_s
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION "lib${LIB_SUFFIX}"
+    ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
 )
 
 # INSTALL (
diff -ur OpenCV-2.1.0.orig/CMakeLists.txt OpenCV-2.1.0/CMakeLists.txt
--- OpenCV-2.1.0.orig/CMakeLists.txt	2010-04-06 04:24:37.000000000 +0300
+++ OpenCV-2.1.0/CMakeLists.txt	2010-07-27 21:56:58.000000000 +0300
@@ -467,8 +467,8 @@
 
     string(REGEX MATCH "[0-9].[0-9]" PYTHON_VERSION_MAJOR_MINOR "${PYTHON_VERSION_FULL}")
     if(UNIX)
-        set(PYTHON_PLUGIN_INSTALL_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages/opencv)
-        set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages)
+        set(PYTHON_PLUGIN_INSTALL_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages/opencv)
+        set(PYTHON_PACKAGES_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages)
     endif()
     if(WIN32)
         get_filename_component(PYTHON_PATH "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\${PYTHON_VERSION_MAJOR_MINOR}\\InstallPath]" ABSOLUTE CACHE)
diff -ur OpenCV-2.1.0.orig/interfaces/ffopencv/CMakeLists.txt OpenCV-2.1.0/interfaces/ffopencv/CMakeLists.txt
--- OpenCV-2.1.0.orig/interfaces/ffopencv/CMakeLists.txt	2010-04-06 04:24:40.000000000 +0300
+++ OpenCV-2.1.0/interfaces/ffopencv/CMakeLists.txt	2010-07-27 21:42:00.000000000 +0300
@@ -40,6 +40,6 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
+	ARCHIVE DESTINATION "lib${LIB_SUFFIX}" COMPONENT main
+	LIBRARY DESTINATION "lib${LIB_SUFFIX}" COMPONENT main
     )
diff -ur OpenCV-2.1.0.orig/src/cv/CMakeLists.txt OpenCV-2.1.0/src/cv/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/cv/CMakeLists.txt	2010-04-06 04:24:42.000000000 +0300
+++ OpenCV-2.1.0/src/cv/CMakeLists.txt	2010-07-27 21:44:32.000000000 +0300
@@ -71,8 +71,8 @@
 
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-    LIBRARY DESTINATION lib COMPONENT main
-    ARCHIVE DESTINATION lib COMPONENT main)
+    LIBRARY DESTINATION "lib${LIB_SUFFIX}" COMPONENT main
+    ARCHIVE DESTINATION "lib${LIB_SUFFIX}" COMPONENT main)
 
 install(FILES ${lib_hdrs}
     DESTINATION include/opencv
diff -ur OpenCV-2.1.0.orig/src/cvaux/CMakeLists.txt OpenCV-2.1.0/src/cvaux/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/cvaux/CMakeLists.txt	2010-04-06 04:24:43.000000000 +0300
+++ OpenCV-2.1.0/src/cvaux/CMakeLists.txt	2010-07-27 21:43:22.000000000 +0300
@@ -76,8 +76,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION "lib${LIB_SUFFIX}" COMPONENT main
+	ARCHIVE DESTINATION "lib${LIB_SUFFIX}" COMPONENT main)
 
 install(FILES ${aux_hdrs_ext}
         DESTINATION include/opencv
diff -ur OpenCV-2.1.0.orig/src/cxcore/CMakeLists.txt OpenCV-2.1.0/src/cxcore/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/cxcore/CMakeLists.txt	2010-04-06 04:24:44.000000000 +0300
+++ OpenCV-2.1.0/src/cxcore/CMakeLists.txt	2010-07-27 21:44:51.000000000 +0300
@@ -71,8 +71,8 @@
 
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-    LIBRARY DESTINATION lib COMPONENT main
-    ARCHIVE DESTINATION lib COMPONENT main)
+    LIBRARY DESTINATION "lib${LIB_SUFFIX}" COMPONENT main
+    ARCHIVE DESTINATION "lib${LIB_SUFFIX}" COMPONENT main)
 
 # install each module headers separately
 install(FILES ${lib_hdrs}
diff -ur OpenCV-2.1.0.orig/src/highgui/CMakeLists.txt OpenCV-2.1.0/src/highgui/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/highgui/CMakeLists.txt	2010-04-06 04:24:44.000000000 +0300
+++ OpenCV-2.1.0/src/highgui/CMakeLists.txt	2010-07-27 21:43:00.000000000 +0300
@@ -282,8 +282,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION "lib${LIB_SUFFIX}" COMPONENT main
+	ARCHIVE DESTINATION "lib${LIB_SUFFIX}" COMPONENT main)
 
 install(FILES ${highgui_ext_hdrs}
         DESTINATION include/opencv
diff -ur OpenCV-2.1.0.orig/src/ml/CMakeLists.txt OpenCV-2.1.0/src/ml/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/ml/CMakeLists.txt	2010-04-06 04:24:45.000000000 +0300
+++ OpenCV-2.1.0/src/ml/CMakeLists.txt	2010-07-27 21:43:42.000000000 +0300
@@ -61,8 +61,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION "lib${LIB_SUFFIX}" COMPONENT main
+	ARCHIVE DESTINATION "lib${LIB_SUFFIX}" COMPONENT main)
 
 install(FILES ${lib_hdrs}
         DESTINATION include/opencv
