diff --git a/gs/base/Makefile.in b/gs/base/Makefile.in
index 8cc3b63..b014051 100644
--- a/gs/base/Makefile.in
+++ b/gs/base/Makefile.in
@@ -121,6 +121,11 @@ GENOPT=
 
 # Choose capability options.
 
+# -DHAVE_BSWAP32
+#       use bswap32 intrinsic
+# -DHAVE_BYTESWAP_H
+#       use byteswap.h functions
+#
 # -DHAVE_MKSTEMP
 #	uses mkstemp instead of mktemp
 #		This uses the more secure temporary file creation call
@@ -142,7 +147,7 @@ GENOPT=
 # -DHAVE_SSE2
 #       use sse2 intrinsics
 
-CAPOPT= @HAVE_MKSTEMP@ @HAVE_HYPOT@ @HAVE_FILE64@ @HAVE_MKSTEMP64@ @HAVE_FONTCONFIG@ @HAVE_LIBIDN@ @HAVE_SETLOCALE@ @HAVE_SSE2@ @HAVE_DBUS@
+CAPOPT= @HAVE_MKSTEMP@ @HAVE_HYPOT@ @HAVE_FILE64@ @HAVE_MKSTEMP64@ @HAVE_FONTCONFIG@ @HAVE_LIBIDN@ @HAVE_SETLOCALE@ @HAVE_SSE2@ @HAVE_DBUS@ @HAVE_BSWAP32@ @HAVE_BYTESWAP_H@
 
 # Define the name of the executable file.
 
diff --git a/gs/base/configure.ac b/gs/base/configure.ac
index 2385b18..346a7ae 100644
--- a/gs/base/configure.ac
+++ b/gs/base/configure.ac
@@ -1554,6 +1554,49 @@ AC_ARG_ENABLE([sse2], AC_HELP_STRING([--disable-sse2],
 AC_SUBST(HAVE_SSE2)
 
 dnl --------------------------------------------------
+dnl check for byte swap intrinsics
+dnl --------------------------------------------------
+
+HAVE_BSWAP32=""
+AC_COMPILE_IFELSE([
+int main (int argc, char *argv[])
+{
+  int a = __builtin_bswap32(argc);
+  return(0);
+}
+],HAVE_BSWAP32="-DHAVE_BSWAP32", HAVE_BSWAP32="")
+
+AC_ARG_ENABLE([bswap32], AC_HELP_STRING([--disable-bswap32],
+       [Do not use bswap32 instrinsic]), [
+             if test "x$enable_bswap32" = xno; then
+                HAVE_BSWAP32=""
+             fi])
+ 
+AC_SUBST(HAVE_BSWAP32)
+
+dnl --------------------------------------------------
+dnl check for byte swap header
+dnl --------------------------------------------------
+
+HAVE_BYTESWAP_H=""
+AC_COMPILE_IFELSE([
+#include "byteswap.h"
+int main (int argc, char *argv[])
+{
+  int a = bswap_32(argc);
+  return(0);
+}
+],HAVE_BYTESWAP_H="-DHAVE_BYTESWAP_H", HAVE_BYTESWAP_H="")
+
+AC_ARG_ENABLE([byteswap-h], AC_HELP_STRING([--disable-byteswap-h],
+       [Do not use byteswap.h functions]), [
+             if test "x$enable_byteswap-h" = xno; then
+                HAVE_BYTESWAP_H=""
+             fi])
+ 
+AC_SUBST(HAVE_BYTESWAP_H)
+
+dnl --------------------------------------------------
 dnl Do substitutions
 dnl --------------------------------------------------
 
diff --git a/gs/base/gsropt.h b/gs/base/gsropt.h
index 220ba84..3bbd81a 100644
--- a/gs/base/gsropt.h
+++ b/gs/base/gsropt.h
@@ -377,18 +377,15 @@ void rop_release_run_op(rop_run_op *op);
 #  define ENDIAN_SWAP_INT _byteswap_ulong
 #endif
 
-#elif defined(__GNUC__) /* Are we using GCC? */
 
-#ifdef __i386__ /* Are we on an x86? */
-#define GCC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
-#if GCC_VERSION >= 40300 /* Modern enough to have byteswap intrinsics? */
+#elif defined(HAVE_BSWAP32)
 
-#include <byteswap.h>
+#define ENDIAN_SWAP_INT __builtin_bswap32
 
-#define ENDIAN_SWAP_INT bswap_32
+#elif defined(HAVE_BYTESWAP_H)
 
-#endif /* GCC >= 4.3 */
-#endif /* x86 or later */
+#include <byteswap.h>
+#define ENDIAN_SWAP_INT bswap_32
 
 #endif
 
