diff -uNr gossip-0.13/protocols/jabber/gossip-sha.c gossip-0.13.patched/protocols/jabber/gossip-sha.c
--- gossip-0.13/protocols/jabber/gossip-sha.c	2006-07-21 13:49:32.000000000 +0200
+++ gossip-0.13.patched/protocols/jabber/gossip-sha.c	2006-08-03 17:55:03.000000000 +0200
@@ -67,9 +67,9 @@
 extern "C" {
 #endif
 
-void SHA1Init (SHA1Context *sc);
-void SHA1Update (SHA1Context *sc, const void *udata, guint32 len);
-void SHA1Final (SHA1Context *sc, guint8 hash[SHA1_HASH_SIZE]);
+static void SHA1Init (SHA1Context *sc);
+static void SHA1Update (SHA1Context *sc, const void *udata, guint32 len);
+static void SHA1Final (SHA1Context *sc, guint8 hash[SHA1_HASH_SIZE]);
 
 #ifdef __cplusplus
 }
@@ -77,7 +77,7 @@
 
 #ifndef lint
 static const char rcsid[] =
-	"$Id: gossip-0.13-ppc_conn_fix.patch,v 1.1 2006/08/03 16:37:04 foser Exp $";
+	"$Id: gossip-0.13-ppc_conn_fix.patch,v 1.1 2006/08/03 16:37:04 foser Exp $";
 #endif /* !lint */
 
 #define ROTL(x, n) (((x) << (n)) | ((x) >> (32 - (n))))
@@ -182,7 +182,7 @@
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
 };
 
-void
+static void
 SHA1Init (SHA1Context *sc)
 {
 #ifdef RUNTIME_ENDIAN
@@ -490,7 +490,7 @@
   sc->hash[4] += e;
 }
 
-void
+static void
 SHA1Update (SHA1Context *sc, const void *udata, guint32 len)
 {
   guint32 bufferBytesLeft;
@@ -566,7 +566,7 @@
     burnStack (sizeof (guint32[86]) + sizeof (guint32 *[5]) + sizeof (int));
 }
 
-void
+static void
 SHA1Final (SHA1Context *sc, guint8 hash[SHA1_HASH_SIZE])
 {
   guint32 bytesToPad;
diff -uNr gossip-0.13/protocols/jabber/Makefile.am gossip-0.13.patched/protocols/jabber/Makefile.am
--- gossip-0.13/protocols/jabber/Makefile.am	2006-07-30 14:19:41.000000000 +0200
+++ gossip-0.13.patched/protocols/jabber/Makefile.am	2006-08-03 17:54:48.000000000 +0200
@@ -2,6 +2,7 @@
 	-I.                                     \
 	-I$(top_srcdir)                         \
 	-DPROTOCOLSDIR=\"$(datadir)/gossip/protocols\" 	\
+	-DRUNTIME_ENDIAN			\
 	$(GOSSIP_JABBER_CFLAGS) $(WARN_CFLAGS)
 
 noinst_LTLIBRARIES = libgossip-jabber.la
