diff -ur gossip-0.6/src/gossip-app.c new-gossip-0.6/src/gossip-app.c
--- gossip-0.6/src/gossip-app.c	2003-11-14 15:35:22.000000000 +0100
+++ new-gossip-0.6/src/gossip-app.c	2003-12-07 17:16:12.000000000 +0100
@@ -1678,6 +1678,20 @@
 
 	priv = app->priv;
 
+	/* This was already handled the first time,
+	   when it was a single click.
+	   Return true, so gtk stops emitting the
+	   signal.
+	*/
+	if (
+	    (event -> type == GDK_2BUTTON_PRESS ||
+	     event->type==GDK_3BUTTON_PRESS) && 
+	    (event -> button == 1 ||
+	     event -> button == 3)
+	    ) {
+		return TRUE;
+	}
+
 	switch (event->button) {
 	case 1:
 		if (app_tray_pop_message (NULL)) {
