--- w3m.el.bak	2005-02-26 20:13:58.000000000 +0800
+++ w3m.el	2005-02-26 20:16:06.000000000 +0800
@@ -86,9 +86,10 @@
   (cond
    ((featurep 'xemacs)
     (require 'w3m-xmas))
-   ((>= emacs-major-version 22)
-    (require 'w3m-e22))
-   ((= emacs-major-version 21)
+   ((and (>= emacs-major-version 23)
+	 (fboundp 'define-coding-system))
+    (require 'w3m-e23))
+   ((>= emacs-major-version 21)
     (require 'w3m-e21))
    ((= emacs-major-version 20)
     (require 'w3m-e20))
--- w3mhack.el.bak	2005-02-26 20:16:27.000000000 +0800
+++ w3mhack.el	2005-02-26 20:28:09.000000000 +0800
@@ -275,9 +275,10 @@
 				   "w3m-xmas.el")
 				  ((boundp 'MULE)
 				   "w3m-om.el")
-				  ((>= emacs-major-version 22)
-				   '("w3m-e22.el" "w3m-fsf.el"))
-				  ((= emacs-major-version 21)
+				  ((and (>= emacs-major-version 23)
+					(fboundp 'define-coding-system))
+				   '("w3m-e23.el" "w3m-fsf.el"))
+				  ((>= emacs-major-version 21)
 				   '("w3m-e21.el" "w3m-fsf.el"))
 				  ((= emacs-major-version 20)
 				   '("w3m-e20.el" "w3m-fsf.el"))
