--- src/tools/port/shell_unix/mksecret.sh	2009-10-13 10:59:58.000000000 +0200
+++ src/tools/port/shell_unix/mksecret.sh	2009-10-13 11:00:23.000000000 +0200
@@ -1679,19 +1679,10 @@
 [ -r /usr/include/sys/file.h ] &&
 	echo "# define xCL_007_FILE_H_EXISTS"
 
-if [ "$vers" = "int_lnx" ] || [ "$vers" = "ibm_lnx" ] || \
-   [ "$vers" = "axp_lnx" ] || [ "$vers" = "i64_lnx" ] || \
-   [ "$vers" = "a64_lnx" ] || [ "$vers" = "int_rpl" ] || \
-   [ "$vers" = "ppc_lnx" ]
-then
-    grep fd_set /usr/include/linux/types.h > /dev/null &&
-	echo "# define xCL_014_FD_SET_TYPE_EXISTS"
-else
     grep fd_set /usr/include/sys/types.h > /dev/null || \
         grep fd_set /usr/include/sys/select.h > /dev/null || \
         grep fd_set /usr/include/sys/time.h > /dev/null &&
 	echo "# define xCL_014_FD_SET_TYPE_EXISTS"
-fi
 
 if [ "$vers" != "ds3_ulx" ]
 then
