--- src/front/st/specials_unix_vms/sharelib.ccpp	2008-10-01 15:06:20.000000000 +0200
+++ src/front/st/specials_unix_vms/sharelib.ccpp	2008-10-30 11:26:05.000000000 +0100
@@ -136,10 +136,9 @@
 # elif defined(OSX)
       EXECUTABLE libxerces-c.$XERCVERS.$SLSFX
 # else
-      EXECUTABLE libxerces-c.$SLSFX.$XERCVERS.0
+      EXECUTABLE $XERCESCLIB
 # endif /* rs4_us5 */
-      LINK libxerces-c.$SLSFX.$XERCVERS
-      LINK libxerces-c.$SLSFX
+      LINK $XERCESCLIB
 # endif
       PERMISSION s:re,o:rwed,g:re,w:re
       }
