Index: util-vserver-0.30.212/scripts/vupdateworld
===================================================================
--- util-vserver-0.30.212/scripts/vupdateworld (revision 2452)
+++ util-vserver-0.30.212/scripts/vupdateworld (revision 2486)
@@ -65,4 +65,14 @@
 done
 
+declare -a vserver_names
+
+while [ $# -gt 0 ]; do
+    case "$1" in
+	(--)		shift; break;;
+	(*)	   	vserver_names=( "${vserver_names[@]}" "$1" );;
+    esac
+    shift
+done
+
 VSOMETHING_TITLE=vemerge
 VSOMETHING_PKGMGMT=1
@@ -71,3 +81,3 @@
 
 test ${#vsomething_opts[@]} -eq 0 || vsomething_opts=( "${vsomething_opts[@]}" -- )
-exec $_VSOMETHING "${send_through[@]}" emerge "${vsomething_opts[@]}" --update --deep --newuse "$@"
+exec $_VSOMETHING "${send_through[@]}" emerge "${vsomething_opts[@]}" "${vserver_names[@]}" -- world --update --deep --newuse "$@"
Index: util-vserver-0.30.212/scripts/vdispatch-conf
===================================================================
--- util-vserver-0.30.212/scripts/vdispatch-conf (revision 2363)
+++ util-vserver-0.30.212/scripts/vdispatch-conf (revision 2486)
@@ -70,3 +70,3 @@
 export VSOMETHING_TITLE VSOMETHING_PKGMGMT
 
-exec $_VSOMETHING "${send_through[@]}" dispatch-conf "${vsomething_opts[@]}"
+exec $_VSOMETHING "${send_through[@]}" dispatch-conf "${vsomething_opts[@]}" "$@"
Index: util-vserver-0.30.212/scripts/vesync
===================================================================
--- util-vserver-0.30.212/scripts/vesync (revision 2363)
+++ util-vserver-0.30.212/scripts/vesync (revision 2486)
@@ -76,15 +76,13 @@
 export VSOMETHING_TITLE VSOMETHING_PKGMGMT
 
-test ${#vsomething_opts[@]} -eq 0 || vsomething_opts=( "${vsomething_opts[@]}" -- )
-
 ret=0
 
 test $overlay_only -eq 0 && \
-	$_VSOMETHING "${send_through[@]}" emerge "${vsomething_opts[@]}" --metadata
+	$_VSOMETHING "${send_through[@]}" emerge "${vsomething_opts[@]}" "$@" -- --metadata
 
 ret=$?
 
 test -n "$overlay" -a -n "$overlay_host" && \
-	$_VSOMETHING "${send_through[@]}" rsync "${vsomething_opts[@]}" \
+	$_VSOMETHING "${send_through[@]}" rsync "${vsomething_opts[@]}" "$@" -- \
 	-rtW --progress --delete --delete-after "${overlay_host}/" "${overlay}/"
 
