--- ns-2.28.orig/sctp/sctp.h	2006-03-28 18:53:00.496028739 +0200
+++ ns-2.28/sctp/sctp.h	2006-03-28 18:52:41.114371797 +0200
@@ -627,7 +627,7 @@
 
   /* debugging functions
    */
-  void SctpAgent::DumpSendBuffer();
+  void DumpSendBuffer();
 
   /* sctp association state variable
    */
--- ns-2.28.orig/mobile/god.h	2006-03-28 18:59:00.835568499 +0200
+++ ns-2.28/mobile/god.h	2006-03-28 18:59:29.789995089 +0200
@@ -85,20 +85,20 @@
 		return sqrt(X*X + Y*Y + Z*Z);
 	}
 
-	inline void vector::operator=(const vector a) {
+	inline void operator=(const vector a) {
 		X = a.X;
 		Y = a.Y;
 		Z = a.Z;
 	}
-	inline void vector::operator+=(const vector a) {
+	inline void operator+=(const vector a) {
 		X += a.X;
 		Y += a.Y;
 		Z += a.Z;
 	}
-	inline int vector::operator==(const vector a) {
+	inline int operator==(const vector a) {
 		return (X == a.X && Y == a.Y && Z == a.Z);
 	}
-	inline int vector::operator!=(const vector a) {
+	inline int operator!=(const vector a) {
 		return (X != a.X || Y != a.Y || Z != a.Z);
 	}
 	inline vector operator-(const vector a) {
