--- protector.dif	2003-09-22 11:39:22.000000000 +0200
+++ /tmp/protector.dif	2003-10-03 21:21:54.661231248 +0200
@@ -272,9 +272,9 @@
 *************** Software Foundation, 59 Temple Place - S
 *** 45,50 ****
 --- 45,51 ----
-  #include "langhooks.h"
   #include "intl.h"
   #include "tm_p.h"
+  #include "input.h"
 + #include "protector.h"
   
   /* Decide whether a function's arguments should be processed
