diff -Nru pptpd-1.3.1.orig/Makefile.am pptpd-1.3.1/Makefile.am
--- pptpd-1.3.1.orig/Makefile.am	2006-02-14 02:25:29.000000000 +0200
+++ pptpd-1.3.1/Makefile.am	2006-03-26 12:07:02.000000000 +0300
@@ -11,7 +11,7 @@
 ##  warning with -Wmissing-prototypes).
 ## -Wmissing-prototypes removed (eg, Linux 2.2.6 headers
 ##  aren't up to it).
-CFLAGS = -O2 -fno-builtin -Wall -DSBINDIR='"$(sbindir)"'
+CFLAGS += -fno-builtin -Wall -DSBINDIR='"$(sbindir)"'
 #CFLAGS = -O2 -fno-builtin -Wall -ansi -DSBINDIR='"$(sbindir)"'
 #CFLAGS = -O2 -fno-builtin -Wall -ansi -pedantic -Wmissing-prototypes -Werror -DSBINDIR='"$(sbindir)"'
 
diff -Nru pptpd-1.3.1.orig/plugins/Makefile pptpd-1.3.1/plugins/Makefile
--- pptpd-1.3.1.orig/plugins/Makefile	2004-05-21 12:26:37.000000000 +0300
+++ pptpd-1.3.1/plugins/Makefile	2006-03-26 12:07:02.000000000 +0300
@@ -1,6 +1,5 @@
 CC	= gcc
-COPTS	= -O2 -g
-CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
+CFLAGS	+= -g -I.. -I../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install -o root
 prefix  = /usr/local
diff -Nru pptpd-1.3.1.orig/pptpgre.c pptpd-1.3.1/pptpgre.c
--- pptpd-1.3.1.orig/pptpgre.c	2005-12-29 02:10:32.000000000 +0200
+++ pptpd-1.3.1/pptpgre.c	2006-03-26 12:08:10.000000000 +0300
@@ -399,7 +399,7 @@
 		}
 		/* check for out-of-order sequence number */
 		if (seq_greater(seq, gre.seq_recv)) {
-			syslog(LOG_DEBUG, "GRE: accepting packet #%d", seq);
+			/* syslog(LOG_DEBUG, "GRE: accepting packet #%d", seq); */
 			stats.rx_accepted++;
 			gre.seq_recv = seq;
 			return cb(cl, buffer + ip_len + headersize, payload_len);
diff -Nru pptpd-1.3.1.orig/pqueue.c pptpd-1.3.1/pqueue.c
--- pptpd-1.3.1.orig/pqueue.c	2005-08-03 11:53:22.000000000 +0300
+++ pptpd-1.3.1/pqueue.c	2006-03-26 12:07:04.000000000 +0300
@@ -12,7 +12,8 @@
 #define DEBUG_ON 0
 #endif
 
-#define DEBUG_CMD(_a) if (DEBUG_ON) { _a }
+/* #define DEBUG_CMD(_a) if (DEBUG_ON) { _a } */
+#define DEBUG_CMD(_a)
 
 #define MIN_CAPACITY 128 /* min allocated buffer for a packet */
 
