http://bugzilla.xfce.org/show_bug.cgi?id=7204

--- src/reminder.c
+++ src/reminder.c
@@ -557,12 +557,19 @@
         g_strlcat(heading, l_alarm->action_time, 90);
         g_strlcat(heading, "<\b>", 10);
     }
-    n = notify_notification_new(heading, l_alarm->description, NULL, NULL);
+    n = notify_notification_new(heading, l_alarm->description, NULL
+#ifdef HAVE_LIBNOTIFY_07
+    );
+#else
+    , NULL);
+#endif
     l_alarm->active_alarm->active_notify = n;
+#ifndef HAVE_LIBNOTIFY_07
     if (g_par.trayIcon 
     && gtk_status_icon_is_embedded((GtkStatusIcon *)g_par.trayIcon))
         notify_notification_attach_to_status_icon(n
                 , (GtkStatusIcon *)g_par.trayIcon);
+#endif
 
     if (l_alarm->notify_timeout == -1)
         notify_notification_set_timeout(n, NOTIFY_EXPIRES_NEVER);
