diff -ur orage-4.7.5.16.orig/globaltime/Makefile.am orage-4.7.5.16/globaltime/Makefile.am
--- orage-4.7.5.16.orig/globaltime/Makefile.am	2010-03-04 09:34:55.000000000 +0200
+++ orage-4.7.5.16/globaltime/Makefile.am	2010-04-19 18:35:33.000000000 +0300
@@ -26,6 +26,7 @@
 
 globaltime_CFLAGS =							\
 	$(LIBXFCEGUI4_CFLAGS)					\
+	$(LIBXFCE4PANEL_CFLAGS)					\
 	-DPACKAGE_DATA_DIR=\""$(datadir)"\"		\
 	-DPACKAGE_LOCALE_DIR=\""$(localedir)"\"
 
diff -ur orage-4.7.5.16.orig/panel-plugin/Makefile.am orage-4.7.5.16/panel-plugin/Makefile.am
--- orage-4.7.5.16.orig/panel-plugin/Makefile.am	2010-03-04 09:34:55.000000000 +0200
+++ orage-4.7.5.16/panel-plugin/Makefile.am	2010-04-19 18:32:05.000000000 +0300
@@ -9,6 +9,7 @@
 
 xfce4_orageclock_plugin_LDFLAGS =						\
 	@LIBXFCE4PANEL_LIBS@                            \
+	@LIBXFCEGUI4_LIBS@				\
 	-lX11
 
 xfce4_orageclock_plugin_SOURCES = 						\
