--- setup.py.orig	2010-05-19 22:51:26.000000000 +0200
+++ setup.py	2010-05-19 22:54:31.000000000 +0200
@@ -33,8 +33,7 @@
                     ('share/doc/yagtd-0.3.2/rest', ['doc/yagtd.rest', 'doc/yagtd.png',
                                               'doc/yagtd.html', 'doc/default.css']),
                     ('share/man/man1', ['doc/yagtd.1']),
-                    ('share/doc/yagtd-0.3.2/tools', ['tools/yagtd-mode.el',
-                                               'tools/yagtd-s60.py'])],
+                    ('share/doc/yagtd-0.3.2/tools', ['tools/yagtd-mode.el'])],
       scripts=['src/yagtd.py'],
       classifiers = ['Development Status :: 4 - Beta',
                      'Intended Audience :: End Users/Desktop',
