diff -ur basket-0.5.0.orig/src/item.cpp basket-0.5.0/src/item.cpp
--- basket-0.5.0.orig/src/item.cpp	2006-05-12 12:05:34.000000000 +0200
+++ basket-0.5.0/src/item.cpp	2006-05-12 12:05:55.000000000 +0200
@@ -422,8 +422,8 @@
 	if (selected == m_isSelected)
 		return;
 
-	DEBUG_WIN << QString(selected ? "Add" : "Remove") + " selected item " + QString::number((int)this) +
-	             "; b.lastIn=" + QString::number((int)(parentBasket()->lastInsertedItem()));
+	DEBUG_WIN << QString(selected ? "Add" : "Remove") + " selected item " + QString::number((long int)this) +
+	             "; b.lastIn=" + QString::number((long int)(parentBasket()->lastInsertedItem()));
 
 	m_isSelected = selected;
 	if (selected) {
