--- src/cluster_bg.c.orig	2009-10-03 09:14:28.000000000 +0200
+++ src/cluster_bg.c	2009-10-02 08:25:40.000000000 +0200
@@ -463,7 +463,10 @@
 for (j=15; j < 23; j++)
 	mvprintw(j,4, "                           ");
 
-if (cluster == SPOTS) linepos = i - 8;
+if (cluster == SPOTS){
+	linepos = i - 8;
+	if (linepos < 0) linepos = 0;
+}
 else linepos = 0;
 
 
--- src/clear_display.c.orig	2007-12-18 16:08:05.000000000 +0100
+++ src/clear_display.c	2009-10-06 07:25:30.000000000 +0200
@@ -100,10 +100,7 @@
  else
 	strftime(time_buf, 60, "DIG %d-%b-%y %H:%M ",  time_ptr);
 
-strncat (month, time_buf + 2, 2);      /* month for muf calc */
-m = atoi ( month);
-if (m == 0)
-	m++;
+	m = time_ptr->tm_mon;		/* month for muf calc */
 	
 mvprintw(12, 3,time_buf);
 
--- src/displayit.c.orig	2009-10-06 16:27:22.000000000 +0200
+++ src/displayit.c	2009-10-06 16:45:31.000000000 +0200
@@ -39,7 +39,7 @@
 		strncat(term2buf, termbuf,  strlen(termbuf) - 1  );
 		strncat(term2buf, backgrnd_str, 81 - strlen(termbuf));   /* fill with blanks */
 
-		strcat(term2buf, "\n");
+		term2buf[80] = '\0';
 		strcpy(terminal1, terminal2);
 		strcpy(terminal2, terminal3);
 		strcpy(terminal3, terminal4);
