diff -ru scim-qtimm-0.9.1.20050704.orig/src/plugin.cpp scim-qtimm-0.9.1.20050704/src/plugin.cpp
--- scim-qtimm-0.9.1.20050704.orig/src/plugin.cpp	2005-07-04 13:01:42.000000000 +0200
+++ scim-qtimm-0.9.1.20050704/src/plugin.cpp	2005-07-04 13:12:24.000000000 +0200
@@ -6,6 +6,8 @@
 
 #include <qinputcontextplugin.h>
 
+#define qDebug if (false) qDebug
+
 class ScimInputContextPlugin : public QInputContextPlugin 
 {
 public:
