Index: work/scim-qtimm-0.9.4/configure.in.in
===================================================================
--- work.orig/scim-qtimm-0.9.4/configure.in.in
+++ work/scim-qtimm-0.9.4/configure.in.in
@@ -131,12 +131,12 @@ INPUTMETHODSDIR=$QTDIR_LOCAL/plugins/inp
 AC_SUBST(INPUTMETHODSDIR)
 
 if test "$kde_qtver" = "3"; then
-if test ! -e $qt_incdir/qinputcontext.h -o ! -e $qt_incdir/qinputcontextplugin.h; then
+KDE_CHECK_HEADERS([qinputcontext.h qinputcontextplugin.h], , [
   AC_MSG_ERROR([
     The Qt library was not compiled with the qt-immodule patch
     applied please download it from
     http://immodule-qt.freedesktop.org/Software/ImmoduleQtDownload and recompile Qt.])
-fi
+])
 fi
 
 AC_MSG_RESULT([
