Index: nbsmtp.c
===================================================================
RCS file: /var/cvs/nbsmtp/nbsmtp.c,v
retrieving revision 1.139
diff -u -B -r1.139 nbsmtp.c
--- nbsmtp.c	12 Mar 2005 19:50:12 -0000	1.139
+++ nbsmtp.c	15 Jun 2005 09:13:56 -0000
@@ -492,8 +492,11 @@
 		}
 
 		strncpy(greeting,smtp_last_message(),sizeof(greeting));
-
-		from64tobits(challenge,greeting);
+		
+		i = from64tobits(challenge,greeting);
+		
+		/* Make sure challenge is '\0' ended, since from64tobits doesn't do it itself */
+		challenge[i] = '\0';
 
 		hmac_md5((unsigned char*)challenge, strlen(challenge),
 				(unsigned char*)serverinfo->auth_pass, strlen(serverinfo->auth_pass),
