--- mod_auth_ldap.c.orig	2003-01-18 19:28:40.000000000 -0500
+++ mod_auth_ldap.c	2003-01-18 19:30:01.000000000 -0500
@@ -168,6 +168,10 @@
         *dummy;
 } ldap_auth_config_rec;
 
+void ldap_module_init(void)
+{
+    ap_add_version_component("mod_auth_ldap/" MOD_AUTH_LDAP_VERSION_S);
+}
 
 static void *create_ldap_auth_dir_config(pool *p,char *d)
 {
@@ -1230,7 +1234,7 @@
 module ldap_auth_module =
 {
     STANDARD_MODULE_STUFF,
-    NULL,                         /* module initializer */
+    ldap_module_init,             /* module initializer */
     create_ldap_auth_dir_config,  /* per-directory config creator */
     NULL,                         /* dir config merger */
     NULL,                         /* server config creator */
