Index: configure.ac
===================================================================
--- configure.ac	(revision 6742)
+++ configure.ac	(revision 6836)
@@ -39,9 +39,9 @@
 AC_LIBTOOL_WIN32_DLL
 AC_LIBLTDL_INSTALLABLE
 AC_PROG_LIBTOOL
-AC_CONFIG_SUBDIRS(libltdl)
 AC_CANONICAL_HOST
 AC_LIB_LTDL
+m4_ifdef([_AC_SEEN_TAG(libltdl)], [], [AC_CONFIG_SUBDIRS(libltdl)])
 AC_SUBST(LTDLINCL)
 AC_SUBST(LIBLTDL)
 AC_SUBST(MKDIR_P)
