diff -ur -x '*~' share/mk/bsd.libnames.mk share-flex/mk/bsd.libnames.mk
--- share/mk/bsd.libnames.mk	2004-11-28 15:10:18.000000000 +0100
+++ share-flex/mk/bsd.libnames.mk	2005-07-10 01:45:38.009094320 +0200
@@ -39,7 +39,7 @@
 LIBDNS?=	${DESTDIR}${LIBDIR}/libdns.a
 LIBEDIT?=	${DESTDIR}${LIBDIR}/libedit.a
 LIBFETCH?=	${DESTDIR}${LIBDIR}/libfetch.a
-LIBFL?=		"don't use LIBFL, use LIBL"
+LIBFL?=		${DESTDIR}${LIBDOR}/libfl.a
 LIBFORM?=	${DESTDIR}${LIBDIR}/libform.a
 LIBFTPIO?=	${DESTDIR}${LIBDIR}/libftpio.a
 LIBG2C?=	${DESTDIR}${LIBDIR}/libg2c.a
@@ -64,7 +64,7 @@
 LIBKICONV?=	${DESTDIR}${LIBDIR}/libkiconv.a
 LIBKRB5?=	${DESTDIR}${LIBDIR}/libkrb5.a
 LIBKVM?=	${DESTDIR}${LIBDIR}/libkvm.a
-LIBL?=		${DESTDIR}${LIBDIR}/libl.a
+LIBL?=		${DESTDIR}${LIBDOR}/libfl.a
 LIBLN?=		"don't use LIBLN, use LIBL"
 .if !defined(NO_BIND)
 LIBLWRES?=	${DESTDIR}${LIBDIR}/liblwres.a
