diff -urN cl-tclink-3.3.1.orig/tclink.asd cl-tclink-3.3.1/tclink.asd
--- cl-tclink-3.3.1.orig/tclink.asd	2005-02-07 15:30:07.000000000 -0600
+++ cl-tclink-3.3.1/tclink.asd	2005-02-07 15:30:51.000000000 -0600
@@ -4,10 +4,6 @@
 ;;; Distributed under the terms of the LLGPL.
 ;;; See LICENSE file for more details.
 
-(eval-when (:compile-toplevel :load-toplevel :execute)
-  (require :uffi)
-  (require :split-sequence))
-
 (defpackage #:CL-TCLINK-SYSTEM
   (:use #:COMMON-LISP #:ASDF))
 (in-package #:CL-TCLINK-SYSTEM)
@@ -93,7 +89,7 @@
 		 (:static-file "LICENSE" :properties ((:licence-file . t)))
 		 (:static-file "LLGPL" :properties ((:installable . t)))
 		 (:static-file "Makefile" :properties ((:installable . t))))
-
+    :depends-on (uffi split-sequence)
     :maintainer "Matthew Danish <mrd@debian.org>"
     :description "Common Lisp bindings to the TrustCommerce transaction system")
 
diff -urN cl-tclink-3.3.1.orig/tclink.lisp cl-tclink-3.3.1/tclink.lisp
--- cl-tclink-3.3.1.orig/tclink.lisp	2005-02-07 15:30:07.000000000 -0600
+++ cl-tclink-3.3.1/tclink.lisp	2005-02-08 00:54:48.611661739 -0600
@@ -9,11 +9,7 @@
 (eval-when (:compile-toplevel :load-toplevel :execute)
 
   (defparameter *library-locations-list*
-    (let ((lib-list
-	   (list "/usr/lib/cl-tclink/"
-                 (translate-logical-pathname
-		  (logical-pathname "cl-library:cl-tclink;"))
-                 "/lib/" "/usr/lib/" "/usr/local/lib/")))
+    (let ((lib-list '("/usr/lib/cl-tclink/")))
       (when *load-truename*
 	(push (directory-namestring *load-truename*)
 	      lib-list))
