--- bbpager-0.3.0/LinkedList.hh	Wed Sep 27 22:35:33 2000
+++ bbpager-0.3.0-hack/LinkedList.hh	Mon Apr  8 01:59:31 2002
@@ -50,7 +50,7 @@
   __llist *list;
   __llist_node *node;
 
-  friend __llist;
+  friend class __llist;
 
 
 protected:
@@ -73,7 +73,7 @@
   __llist_node *_first, *_last;
   __llist *iterators;
 
-  friend __llist_iterator;
+  friend  class __llist_iterator;
 
 
 protected:
--- bbpager-0.3.0/Timer.hh	Tue Dec  7 20:44:23 1999
+++ bbpager-0.3.0-hack/Timer.hh	Mon Apr  8 01:59:45 2002
@@ -47,7 +47,7 @@
 
 
 class BTimer {
-  friend BaseDisplay;
+  friend class BaseDisplay;
 private:
   BaseDisplay *display;
   TimeoutHandler *handler;
