--- tools-20040406.old/uml_net/slip.c	2002-04-28 15:47:35.000000000 -0400
+++ tools-20040406.new/uml_net/slip.c	2005-03-07 10:40:37.000000000 -0500
@@ -111,7 +111,7 @@
 void slip_v4(int argc, char **argv)
 {
   struct output output = INIT_OUTPUT;
-  char *op;
+  char *op, dev[sizeof("slnnnnn\0")];
 
   if(setreuid(0, 0) < 0){
     output_errno(&output, "slip - setreuid failed");
@@ -135,7 +135,8 @@
     slip_up(0, argv[1], NULL, NULL, &output);
   }
   else if(!strcmp(op, "down")){
-    slip_down(argv[1], NULL, NULL, &output);
+    slip_name(0, dev, &output);
+    slip_down(dev, NULL, NULL, &output);
   }
   else {
     printf("slip - Unknown op '%s'\n", op);
