Index: nted-1.6.0/Makefile.am
===================================================================
--- nted-1.6.0.orig/Makefile.am
+++ nted-1.6.0/Makefile.am
@@ -33,8 +33,6 @@ AM_CPPFLAGS += -DLOCALEDIR=\"${localedir
 AM_CPPFLAGS += -I$(top_srcdir)/dialogs -I$(top_srcdir)/commands -I$(top_srcdir)/chords -I$(top_srcdir)/idiotseditor
 AM_CPPFLAGS += $(FT2_CFLAGS) $(CAIRO_CFLAGS) $(GTK_CFLAGS)
 
-AM_LDFLAGS = $(CAIRO_LIBS) $(PANGO_LIBS) $(GTK_LIBS) -lasound
-
 bin_PROGRAMS = nted
 nted_SOURCES = mainwindow.cpp resource.cpp page.cpp system.cpp staff.cpp \
 	voice.cpp chordorrest.cpp note.cpp musicxmlimport.cpp \
@@ -53,7 +51,8 @@ nted_SOURCES += acceleration.h beam.h be
 	temposign.h tuplet.h voice.h volumesign.h positionarray.h dynarray.h COPYING.GFDL COPYING.FONT.TXT \
 	freechord.h freespacer.h pangocairotext.h freechordname.h midiimporter2.h
 
-nted_LDADD = commands/libcommands.a dialogs/libdialogs.a chords/libchords.a idiotseditor/libidiotseditor.a
+nted_LDADD = commands/libcommands.a dialogs/libdialogs.a chords/libchords.a idiotseditor/libidiotseditor.a $(CAIRO_LIBS) $(PANGO_LIBS) $(GTK_LIBS) -lasound
+
 
 cleanup: maintainer-clean cleanup-local
 
