diff -ur vegastrike/src/boost129/boost/python/operators.hpp vegastrike-gentoo/src/boost129/boost/python/operators.hpp
--- vegastrike/src/boost129/boost/python/operators.hpp	2003-09-22 18:21:04.000000000 +1200
+++ vegastrike-gentoo/src/boost129/boost/python/operators.hpp	2003-09-22 19:15:13.000000000 +1200
@@ -175,7 +175,7 @@
 {                                                                       \
   template <class L, class R>                                           \
   inline detail::operator_<detail::op_##id,L,R>                         \
-  operator##op(L const&, R const&)                                      \
+  operator op(L const&, R const&)                                      \
   {                                                                     \
       return detail::operator_<detail::op_##id,L,R>();                  \
   }                                                                     \
@@ -262,7 +262,7 @@
 {                                                               \
   template <class R>                                            \
   inline detail::operator_<detail::op_##id,self_t,R>            \
-  operator##op(self_t const&, R const&)                         \
+  operator op(self_t const&, R const&)                         \
   {                                                             \
       return detail::operator_<detail::op_##id,self_t,R>();     \
   }                                                             \
diff -ur vegastrike/src/gui/text_area.h vegastrike-gentoo/src/gui/text_area.h
--- vegastrike/src/gui/text_area.h	2003-09-22 18:21:01.000000000 +1200
+++ vegastrike-gentoo/src/gui/text_area.h	2003-09-22 19:13:11.000000000 +1200
@@ -49,20 +49,20 @@
 #define DIR_TEXT "textures/gui/"
 
 // These defines are used by the image loader
-#define TEXT_AREA_00 DIR_TEXT ## "button_down.png"
-#define TEXT_AREA_01 DIR_TEXT ## "button_up.png"
-#define TEXT_AREA_02 DIR_TEXT ## "corner_top_left.png"
-#define TEXT_AREA_03 DIR_TEXT ## "corner_top_right.png"
-#define TEXT_AREA_04 DIR_TEXT ## "corner_bottom_left.png"
-#define TEXT_AREA_05 DIR_TEXT ## "corner_bottom_right.png"
-#define TEXT_AREA_06 DIR_TEXT ## "highlight_button_up.png"
-#define TEXT_AREA_07 DIR_TEXT ## "highlight_button_down.png"
-#define TEXT_AREA_08 DIR_TEXT ## "horizontal.png"
-#define TEXT_AREA_09 DIR_TEXT ## "left_side.png"
-#define TEXT_AREA_10 DIR_TEXT ## "right_side.png"
-#define TEXT_AREA_11 DIR_TEXT ## "scrollbar.png"
-#define TEXT_AREA_12 DIR_TEXT ## "highlight_scrollbar.png"
-#define TEXT_AREA_13 DIR_TEXT ## "horizontal_end.png"
+#define TEXT_AREA_00 DIR_TEXT "button_down.png"
+#define TEXT_AREA_01 DIR_TEXT "button_up.png"
+#define TEXT_AREA_02 DIR_TEXT "corner_top_left.png"
+#define TEXT_AREA_03 DIR_TEXT "corner_top_right.png"
+#define TEXT_AREA_04 DIR_TEXT "corner_bottom_left.png"
+#define TEXT_AREA_05 DIR_TEXT "corner_bottom_right.png"
+#define TEXT_AREA_06 DIR_TEXT "highlight_button_up.png"
+#define TEXT_AREA_07 DIR_TEXT "highlight_button_down.png"
+#define TEXT_AREA_08 DIR_TEXT "horizontal.png"
+#define TEXT_AREA_09 DIR_TEXT "left_side.png"
+#define TEXT_AREA_10 DIR_TEXT "right_side.png"
+#define TEXT_AREA_11 DIR_TEXT "scrollbar.png"
+#define TEXT_AREA_12 DIR_TEXT "highlight_scrollbar.png"
+#define TEXT_AREA_13 DIR_TEXT "horizontal_end.png"
 
 // These defines are used by the class functions to locate the texture in the array (above)
 #define IMG_BUTTON_DOWN 0
