--- ./configure.ac.orig	2007-05-12 15:16:22.000000000 +0400
+++ ./configure.ac	2007-05-12 15:16:38.000000000 +0400
@@ -518,7 +518,7 @@
 		  	  together with rrdtool in $prefix/lib/ruby. You have to
                           add $prefix/lib/ruby/$ruby_version/$sitearch to you $: variable
                           for ruby to find the RRD.so file.],
-[RUBY_MAKE_OPTIONS=],[RUBY_MAKE_OPTIONS="sitedir=$prefix/lib/ruby"])
+[RUBY_MAKE_OPTIONS=],[RUBY_MAKE_OPTIONS="sitedir="'$(DESTDIR)'"$prefix/lib/ruby"])
 
     
 AC_ARG_WITH(ruby-options,
