diff -urN ./rrdtool-1.2.19/bindings/python/Makefile.am ./rrdtool-1.2.18/bindings/python/Makefile.am
--- ./rrdtool-1.2.19-orig/bindings/python/Makefile.am	2007-02-01 06:51:38.000000000 +0100
+++ ./rrdtool-1.2.19/bindings/python/Makefile.am	2007-01-23 21:54:50.000000000 +0100
@@ -2,10 +2,11 @@
 
 pyexec_PROGRAMS = rrdtoolmodule.so
 
+rrdtoolmodule_so_LDFLAGS = -module -shared
 rrdtoolmodule_so_LDADD = $(top_builddir)/src/librrd.la
-rrdtoolmodule_so_SOURCES = rrdtoolmodule.c
+rrdtoolmodule_so_SOURCES =     rrdtoolmodule.c
 
-noinst_HEADERS =           rrd_extra.h
+noinst_HEADERS =                rrd_extra.h
 
 clean-local:
 	rm -rf build
