diff -urN ./rrdtool-1.2.19/bindings/tcl/tclrrd.c ./rrdtool-1.2.19-new/bindings/tcl/tclrrd.c
--- ./rrdtool-1.2.19-orig/bindings/tcl/tclrrd.c	2007-02-01 06:51:38.000000000 +0100
+++ ./rrdtool-1.2.19/bindings/tcl/tclrrd.c	2007-02-01 14:40:33.000000000 +0100
@@ -18,8 +18,8 @@
 #include <time.h>
 #include <unistd.h>
 #include <tcl.h>
-#include <rrd_tool.h>
-#include <rrd_format.h>
+#include "../../src/rrd_tool.h"
+#include "../../src/rrd_format.h"
 
 /* support pre-8.4 tcl */
 
diff -urN ./rrdtool-1.2.19-orig/configure.ac ./rrdtool-1.2.19/configure.ac
--- ./rrdtool-1.2.19-orig/configure.ac	2007-02-01 06:51:40.000000000 +0100
+++ ./rrdtool-1.2.19/configure.ac	2007-02-01 14:45:55.000000000 +0100
@@ -554,7 +554,7 @@
         AC_MSG_WARN([tclConfig.sh not found - Tcl interface won't be built])
   else
 	. $tcl_config
-	TCL_PACKAGE_DIR="$TCL_PACKAGE_PATH/tclrrd$VERSION"
+	TCL_PACKAGE_DIR="${TCL_PACKAGE_PATH% *}/tclrrd$VERSION"
   fi
   AC_ARG_ENABLE(tcl,[  --enable-tcl-site        install the tcl extension in the tcl tree],
   [],[enable_tcl_site=yes])
