Index: extragear/multimedia/k3b/src/option/base_k3bdivxoptiontab.ui
===================================================================
--- extragear/multimedia/k3b/src/option/base_k3bdivxoptiontab.ui	(revision 460630)
+++ extragear/multimedia/k3b/src/option/base_k3bdivxoptiontab.ui	(revision 460631)
@@ -326,7 +326,6 @@
 </widget>
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
-    <includehint>kdoublenuminput.h</includehint>
     <includehint>knuminput.h</includehint>
     <includehint>knuminput.h</includehint>
     <includehint>knuminput.h</includehint>
