--- src/audio/input/mp3/k3bmp3module.cpp	2003-05-27 03:43:45.000000000 -0500
+++ src/audio/input/mp3/k3bmp3module_1.cpp	2003-08-31 06:46:57.000000000 -0500
@@ -425,7 +425,7 @@ unsigned int K3bMp3Module::resampleBlock
 
     while (step < MAD_F_ONE) {
        *target++ = step ?
- 	last + mad_f_mul(*source - last, step) 
+ 	last + __extension__ mad_f_mul(*source - last, step) 
  	: last;
 
        step += m_madResampledRatio;
@@ -442,7 +442,7 @@ unsigned int K3bMp3Module::resampleBlock
     step = mad_f_fracpart(step);
 
     *target++ = step ?
-      *source + mad_f_mul(source[1] - source[0], step) 
+      *source + __extension__ mad_f_mul(source[1] - source[0], step) 
       : *source;
 
     step += m_madResampledRatio;
