diff -Naurp webmin-1.300.orig/postfix/postfix-lib.pl webmin-1.300/postfix/postfix-lib.pl
--- webmin-1.300.orig/postfix/postfix-lib.pl	2006-09-14 15:21:43.000000000 -0700
+++ webmin-1.300/postfix/postfix-lib.pl	2006-10-16 11:55:45.000000000 -0700
@@ -682,15 +682,15 @@ sub create_mapping
 {
 &get_maps($_[0], $_[2]);	# force cache init
 my @maps_files = $_[2] ? @{$_[2]} : &get_maps_files(&get_real_value($_[0]));
-local $lref = &read_file_lines($maps_files[0]);
+local $lref = &read_file_lines($maps_files[$#maps_files]);
 $_[1]->{'line'} = scalar(@$lref);
 push(@$lref, &make_table_comment($_[1]->{'cmt'}));
 push(@$lref, "$_[1]->{'name'}\t$_[1]->{'value'}");
 $_[1]->{'eline'} = scalar(@$lref)-1;
 &flush_file_lines();
 
-$_[1]->{'map_file'} = $maps_files[0];
-$_[1]->{'file'} = $maps_files[0];
+$_[1]->{'map_file'} = $maps_files[$#maps_files];
+$_[1]->{'file'} = $maps_files[$#maps_files];
 $_[1]->{'number'} = scalar(@{$maps_cache{$_[0]}});
 push(@{$maps_cache{$_[0]}}, $_[1]);
 }
