diff -ur cl-rsm-bitcomp-1.1.orig/bitcomp.lisp cl-rsm-bitcomp-1.1/bitcomp.lisp
--- cl-rsm-bitcomp-1.1.orig/bitcomp.lisp	2005-02-05 15:14:10.293213280 -0600
+++ cl-rsm-bitcomp-1.1/bitcomp.lisp	2005-02-05 15:15:16.191873179 -0600
@@ -22,7 +22,7 @@
     (:report (lambda (condition stream)
                (format stream "Warning: ~a~%" (message condition))))))
 
-(defconstant +null-rep+ (make-condition 'null-compressed))
+(cl+:defconst +null-rep+ (make-condition 'null-compressed))
 
 ;;;;
 ;;;; Representation Layer.
diff -ur cl-rsm-bitcomp-1.1.orig/rsm-bitcomp.asd cl-rsm-bitcomp-1.1/rsm-bitcomp.asd
--- cl-rsm-bitcomp-1.1.orig/rsm-bitcomp.asd	2005-02-05 15:14:10.293213280 -0600
+++ cl-rsm-bitcomp-1.1/rsm-bitcomp.asd	2005-02-05 15:15:47.245885072 -0600
@@ -25,7 +25,7 @@
   :licence "BSD-style"
   :description "Bit Compression."
 
-  :depends-on (rsm-queue)
+  :depends-on (rsm-queue cl-plus)
   
   :components
   ((:file "package")
