--- Chromium-0.9/src-setup/HelpPage.cpp.qt3	Thu Feb  8 06:18:59 2001
+++ Chromium-0.9/src-setup/HelpPage.cpp	Thu Apr 11 10:46:22 2002
@@ -22,8 +22,6 @@
 	cgroup.setColor(QColorGroup::Base, QColor(70, 60, 100));
 	cgroup.setColor(QColorGroup::Text, QColor(220, 220, 220));
 	QFont	font("Helvetica", 12, false);
-	page->setPaperColorGroup(cgroup);
-	page->setLinkColor(QColor(192, 192, 255));
 	page->setFont(font);
 
 	backBut = new QPushButton(this);
--- Chromium-0.9/src-setup/Makefile.qt3	Thu Apr 11 10:49:17 2002
+++ Chromium-0.9/src-setup/Makefile	Thu Apr 11 10:49:27 2002
@@ -15,7 +15,7 @@
 INCPATH	=	-I$(QTDIR)/include
 LINK	=	g++
 LFLAGS	=	
-LIBS	=	$(SUBLIBS) -L$(QTDIR)/lib -L/usr/X11R6/lib  $(AL_LIBS) $(SDL_LIBS) $(SMPEG_LIBS) $(VORBIS_LIBS) -lm -pthread -lqt -lXext -lX11 -lm
+LIBS	=	$(SUBLIBS) -L$(QTDIR)/lib -L/usr/X11R6/lib  $(AL_LIBS) $(SDL_LIBS) $(SMPEG_LIBS) $(VORBIS_LIBS) -lm -pthread -lqt-mt -lXext -lX11 -lm
 MOC	=	$(QTDIR)/bin/moc
 UIC	=	$(QTDIR)/bin/uic
 
