--- src/pcfx/soundbox.cpp.old	2010-05-24 11:15:20.000000000 +0200
+++ src/pcfx/soundbox.cpp	2010-05-24 11:15:36.000000000 +0200
@@ -416,7 +416,7 @@
   int rolloff = (int)((double)0.90 * 21477272 * 2 / 1365 / (1 << SampleFreq) ) / 2;
   for(int ch = 0; ch < 2; ch++)
    for(int lr = 0; lr < 2; lr++)
-    ADPCMSynth[ch][lr].treble_eq(blip_eq_t::blip_eq_t(-1000, rolloff, FSettings.SndRate));
+    ADPCMSynth[ch][lr].treble_eq(blip_eq_t(-1000, rolloff, FSettings.SndRate));
   LSampleFreq = SampleFreq;
  }
 }
