--- ./src/defines.h	2005-02-06 06:56:02.000000000 +0100
+++ ./src/defines.h	2007-02-04 00:37:53.000000000 +0100
@@ -22,8 +22,8 @@
 
 #define BUILD_STRING	"1.1.6 -- 2005-02-06 12:56:02"
 #define BUILD_VERSIONS	"2.6.1 / 2.6.1 / 2.9.1 / 2.6.17"
-#define SCHEMESDIR	"/usr/local/share/gtranslator/colorschemes"
-#define SCRIPTSDIR	"/usr/local/share/gtranslator/scripts"
-#define WINDOW_ICON	"/usr/local/share/pixmaps/gtranslator.png"
+#define SCHEMESDIR	"/usr/share/gtranslator/colorschemes"
+#define SCRIPTSDIR	"/usr/share/gtranslator/scripts"
+#define WINDOW_ICON	"/usr/share/pixmaps/gtranslator.png"
 
 #endif
