--- configure	2008-05-27 20:44:46.000000000 +0200
+++ configure	2008-05-27 20:45:29.000000000 +0200
@@ -7127,7 +7127,7 @@
 if test "z$sc_present" = "zyes" ; then
 	{ echo "$as_me:$LINENO: checking for scrollkeeper" >&5
 echo $ECHO_N "checking for scrollkeeper... $ECHO_C" >&6; }
-	_VERSION=`scrollkeeper-config --version|sed -e 's/\.//g'`
+	_VERSION=`scrollkeeper-config --version|sed -e 's/^\([0-9\.]\+\).*/\1/g;s/\.//g'`
 	if test $_VERSION -ge 0 -o $_VERSION -ge 02; then
 		{ echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
--- configure.in	2008-05-27 20:44:33.000000000 +0200
+++ configure.in	2008-05-27 20:45:10.000000000 +0200
@@ -117,7 +117,7 @@
 AC_CHECK_PROG(sc_present, scrollkeeper-config, yes, no)
 if test "z$sc_present" = "zyes" ; then
 	AC_MSG_CHECKING([for scrollkeeper])
-	_VERSION=`scrollkeeper-config --version|sed -e 's/\.//g'`
+	_VERSION=`scrollkeeper-config --version|sed -e 's/^\([0-9\.]\+\).*/\1/g;s/\.//g'`
 	if test $_VERSION -ge 0 -o $_VERSION -ge 02; then
 		AC_MSG_RESULT(yes)
 		SC_STATEDIR=`scrollkeeper-config --pkglocalstatedir`
