diff -uNr linux-2.6.0-test11/arch/i386/kernel/cpu/intel.c linux-2.6.0-test11-mm1/arch/i386/kernel/cpu/intel.c
--- linux-2.6.0-test11/arch/i386/kernel/cpu/intel.c	2003-12-17 14:21:29.060115753 +0100
+++ linux-2.6.0-test11-mm1/arch/i386/kernel/cpu/intel.c	2003-12-17 14:10:39.886919748 +0100
@@ -9,9 +9,11 @@
 #include <asm/msr.h>
 #include <asm/uaccess.h>
 #include <asm/desc.h>
+#include <asm/mpspec.h>
+#include <asm/apic.h>
 
 #include "cpu.h"
-#include "mach_apic.h"
+#include <mach_apic.h>
 
 #ifdef CONFIG_X86_INTEL_USERCOPY
 /*