diff -urN linux-2.6.0-test5-orig/mm/slab.c linux-2.6.0-test5/mm/slab.c
--- linux-2.6.0-test5-orig/mm/slab.c	2003-09-10 07:31:56.000000000 -0700
+++ linux-2.6.0-test5/mm/slab.c	2003-09-10 07:37:26.000000000 -0700
@@ -2794,11 +2794,13 @@
 		} else {
 			kernel_map_pages(virt_to_page(objp), c->objsize/PAGE_SIZE, 1);
 
+#if DEBUG
 			if (c->flags & SLAB_RED_ZONE)
 				printk("redzone: 0x%lx/0x%lx.\n", *dbg_redzone1(c, objp), *dbg_redzone2(c, objp));
 
 			if (c->flags & SLAB_STORE_USER)
 				printk("Last user: %p.\n", *dbg_userword(c, objp));
+#endif
 		}
 		spin_unlock_irqrestore(&c->spinlock, flags);
