--- ./work/serlook/src/serlook.desktop.orig	2009-02-11 00:20:13.000000000 +0100
+++ ./work/serlook/src/serlook.desktop	2009-02-11 00:20:13.000000000 +0100
@@ -1,12 +1,11 @@
 # KDE Config File  
 [Desktop Entry]
 Type=Application
-Exec=serlook -caption "%c" %i %m  
-Icon=serlook.png
-MiniIcon=serlook.png
-DocPath=serlook/index.html
+Exec=serlook -caption "%c" %i  
+Icon=serlook
+X-DocPath=serlook/index.html
 Comment=
 Comment[de]=
-Terminal=0
+Terminal=false
 Name=SerLooK
 Name[de]=SerLooK
