diff -rupN serlookb/src/khexedit/hexbuffer.h serlook/src/khexedit/hexbuffer.h
--- serlookb/src/khexedit/hexbuffer.h	2009-01-23 03:26:33.000000000 -0400
+++ serlook/src/khexedit/hexbuffer.h	2009-01-23 03:28:12.000000000 -0400
@@ -26,7 +26,7 @@
 #include <config.h>
 #endif 
 
-#include <iostream.h>
+#include <iostream>
 #include <time.h>
 
 #include <qbitarray.h> 
@@ -433,8 +433,8 @@ struct SCursorSpec
 
   void dump( void )
   {
-    cout << "offset: " << offset << " y: " << y << " x1: " << x1 << endl;
-    cout << "x2: " << x2 << " cell: " << cell << endl;
+    std::cout << "offset: " << offset << " y: " << y << " x1: " << x1 << std::endl;
+    std::cout << "x2: " << x2 << " cell: " << cell << std::endl;
   }
 
   bool inside( uint min, uint max )
diff -rupN serlookb/src/sldumpview.cpp serlook/src/sldumpview.cpp
--- serlookb/src/sldumpview.cpp	2009-01-23 03:26:34.000000000 -0400
+++ serlook/src/sldumpview.cpp	2009-01-23 03:29:17.000000000 -0400
@@ -24,7 +24,7 @@
 
 static char* cvs_id = "$Id: serlook-0.3.3-gcc43.diff,v 1.1 2009/02/10 23:33:41 carlo Exp $";
 
-#include <iostream.h>
+#include <iostream>
 
 #include "slsession.h"
 #include "slview.h"
