diff -urN gst-plugins-0.8.2/ext/alsa/gstalsasink.c gst-plugins-0.8.2-alsasink_seek_fix/ext/alsa/gstalsasink.c
--- gst-plugins-0.8.2/ext/alsa/gstalsasink.c	2004-06-23 06:19:48.000000000 -0700
+++ gst-plugins-0.8.2-alsasink_seek_fix/ext/alsa/gstalsasink.c	2004-07-06 02:33:35.000000000 -0700
@@ -220,13 +220,13 @@
             GST_SECOND * this->transmitted / this->format->rate -
             gst_alsa_sink_get_time (this);
         if (gst_event_discont_get_value (event, GST_FORMAT_TIME, &value)) {
-          gst_element_set_time_delay (GST_ELEMENT (this), MIN (value, delay), delay);
+          gst_element_set_time_delay (GST_ELEMENT (this), value, MIN (value, delay));
         } else if (this->format &&
             (gst_event_discont_get_value (event, GST_FORMAT_DEFAULT, &value) ||
                 gst_event_discont_get_value (event, GST_FORMAT_BYTES,
                     &value))) {
           value = gst_alsa_samples_to_timestamp (this, value);
-          gst_element_set_time_delay (GST_ELEMENT (this), MIN (value, delay), delay);
+          gst_element_set_time_delay (GST_ELEMENT (this), value, MIN (value, delay));
         } else {
           GST_WARNING_OBJECT (this,
               "couldn't extract time from discont event. Bad things might happen!");
