diff -ur realtime-lsm-0.8.5/realcap.c realtime-lsm-0.8.5-allcaps/realcap.c
--- realtime-lsm-0.8.5/realcap.c	2004-11-24 13:38:30.000000000 -0500
+++ realtime-lsm-0.8.5-allcaps/realcap.c	2005-06-23 23:44:03.000000000 -0400
@@ -43,6 +43,13 @@
 module_param_named(mlock, rt_mlock, int, 0644);
 MODULE_PARM_DESC(mlock, " enable memory locking privileges.");
 
+/* enable all capabilities */
+static int rt_allcaps = 0;
+module_param_named(allcaps, rt_allcaps, int, 0644);
+MODULE_PARM_DESC(allcaps, " enable all capabilities, including CAP_SETPCAP.");
+
+static kernel_cap_t cap_bset_save;      /* place to save cap-bound */
+
 /* helper function for testing group membership */
 static inline int gid_ok(int gid)
 {
@@ -112,6 +119,12 @@
 		secondary = 1;
 	}
 
+	cap_bset_save = cap_bset;       /* save cap-bound */
+	if (rt_allcaps) {
+		cap_bset = to_cap_t(~0);
+		printk(KERN_INFO "Realtime LSM enabling all capabilities\n");
+	}
+	
 	if (rt_any)
 		printk(KERN_INFO RT_LSM
 		       "initialized (all groups, mlock=%d)\n", rt_mlock);
@@ -127,6 +140,8 @@
 
 static void __exit realtime_exit(void)
 {
+	cap_bset = cap_bset_save;       /* restore cap-bound */
+	
 	/* remove ourselves from the security framework */
 	if (secondary) {
 		if (mod_unreg_security(MY_NAME, &capability_ops))
