--- openh323u.mak.in.orig	2003-11-03 04:12:31.000000000 +0100
+++ openh323u.mak.in	2003-11-03 04:16:07.000000000 +0100
@@ -121,11 +121,26 @@
 #
 
 PWLIBDIR	= @PWLIBDIR@
-STDCCFLAGS	+= @STDCCFLAGS@
 LDFLAGS		+= @LDFLAGS@
 ENDLDLIBS	:= @ENDLDLIBS@ $(ENDLDLIBS)
 H323_AVCODEC	= @H323_AVCODEC@
 
+#
+# stkn@gentoo.org
+# moved stuff around, this should fix problems with include order
+# blame me if this breaks things...
+#
+OH323_SRCDIR = $(OPENH323DIR)/src
+ifdef PREFIX
+OH323_INCDIR = $(PREFIX)/include/openh323
+else
+OH323_INCDIR = $(OPENH323DIR)/include
+endif
+OH323_LIBDIR = $(OPENH323DIR)/lib
+
+STDCCFLAGS   += -I$(OH323_INCDIR) @STDCCFLAGS@
+
+
 ifdef LIBRARY_MAKEFILE
 include $(PWLIBDIR)/make/unix.mak
 else
@@ -140,16 +155,6 @@
 
 #OH323_SUPPRESS_H235	= 1
 
-
-OH323_SRCDIR = $(OPENH323DIR)/src
-ifdef PREFIX
-OH323_INCDIR = $(PREFIX)/include/openh323
-else
-OH323_INCDIR = $(OPENH323DIR)/include
-endif
-OH323_LIBDIR = $(OPENH323DIR)/lib
-
-
 ifdef NOTRACE
 STDCCFLAGS += -DPASN_NOPRINTON -DPASN_LEANANDMEAN
 OH323_SUFFIX = n
@@ -168,15 +173,12 @@
 endif
 
 
-
 OH323_BASE  = h323_$(PLATFORM_TYPE)_$(OH323_SUFFIX)$(LIB_TYPE)
 OH323_FILE  = lib$(OH323_BASE).$(LIB_SUFFIX)
 
 LDFLAGS	    += -L$(OH323_LIBDIR)
 LDLIBS	    := -l$(OH323_BASE) $(LDLIBS)
 
-STDCCFLAGS  += -I$(OH323_INCDIR)
-
 
 ifneq ($(OS),Nucleus)
 
