--- glibc-2.2.5/sysdeps/unix/sysv/sysv4/solaris2/sparc/sysdep.h.old	Fri Jul  6 06:56:22 2001
+++ glibc-2.2.5/sysdeps/unix/sysv/sysv4/solaris2/sparc/sysdep.h	Wed Jul 24 14:36:31 2002
@@ -51,6 +51,6 @@
 #define	r1		%o1
 #define	MOVE(x,y)	mov x, y
 
-#define LOC(name)	.##L##name
+#define LOC(name)	.L##name
 
 #endif	/* __ASSEMBLER__ */
--- glibc-2.2.5/sysdeps/unix/sysv/linux/sparc/sparc32/sysdep.h.old	Fri Jul  6 06:56:21 2001
+++ glibc-2.2.5/sysdeps/unix/sysv/linux/sparc/sparc32/sysdep.h	Wed Jul 24 14:35:08 2002
@@ -48,7 +48,7 @@
 #define END(name) \
 	.size name, . - name
 
-#define LOC(name)  . ## L ## name
+#define LOC(name)  .L##name
 
 #ifdef PIC
 #define SYSCALL_ERROR_HANDLER					\
