--- 2.0.4.5.orig/CLHEP/Random/test/testBug58950.cc	2009-12-07 18:49:51.000000000 +0000
+++ 2.0.4.5/CLHEP/Random/test/testBug58950.cc	2010-01-19 17:25:07.493351325 +0000
@@ -9,6 +9,9 @@
 #include <iostream> 
 #include <stdexcept>
 #include <cmath>
+#if __GNUC__ >=4 && __GNUC_MINOR__ >=3
+#include <cstdlib>
+#endif
 #include "CLHEP/Random/RanecuEngine.h"
 #include "CLHEP/Random/Random.h"
 
