--- ksms/messagestore.cpp.orig	2002-11-23 01:35:30.000000000 +0100
+++ ksms/messagestore.cpp	2003-08-17 21:10:36.000000000 +0200
@@ -150,20 +150,25 @@
     messages_.clear();
 
     QFile xmlFile( messagesFile_ );
-    MessageStore::ReaderProgress progress( this );
-    XmlInputSourceWithProgress source( &xmlFile, &progress );
-    QXmlSimpleReader reader;
-    MessagesParser handler( this );
-    reader.setContentHandler( &handler );
-    reader.setDTDHandler( &handler );
-    source.reset();
-    reader.parse( source );
-
-    reading_ = false;
+    if (xmlFile.exists()) {
+        MessageStore::ReaderProgress progress( this );
+        XmlInputSourceWithProgress source( &xmlFile, &progress );
+        QXmlSimpleReader reader;
+        MessagesParser handler( this );
+        reader.setContentHandler( &handler );
+        reader.setDTDHandler( &handler );
+        source.reset();
+        reader.parse( source );
+    
+        reading_ = false;
 
-    emit storeChanged();
+        emit storeChanged();
     
-    modified_ = false;
+        modified_ = false;
+    } else {
+	reading_ = false;
+	modified_ = false;
+    }
 }
 
 
