--- Rules.linux	Sun Mar 31 03:06:22 2002
+++ Rules.make	Tue May 21 15:42:31 2002
@@ -12,14 +12,13 @@
 XLIB_DIR = -L/usr/X11R6/lib
 XINC_DIR = -I/usr/X11R6/include
 #--- Python
-PYTHON_EXE = $(PYMOL_PATH)/ext/bin/python
+PYTHON_EXE = /usr/bin/python
 PYTHON_LIB =
 PYTHON_LIB_DIR = 
-PYTHON_INC_DIR = -I$(PYMOL_PATH)/ext/include/python2.1 \
-	-I$(PYMOL_PATH)/ext/include/python2.1/Numeric
+PYTHON_INC_DIR = -I/usr/include/python2.2  -I/usr/include/python2.2/Numeric
 #--- Other external dependencies
-EXT_INC_DIR = -I$(PYMOL_PATH)/ext/include
-EXT_LIB_DIR = -L$(PYMOL_PATH)/ext/lib
+EXT_INC_DIR = 
+EXT_LIB_DIR = 
 #---------------------------------------------------------------------
 #
 #- Build for LINUX as an importable module ---------------------------
@@ -48,7 +47,7 @@
 #
 #- Choose One --------------------------------------------------------
 #--- Workaround for XFree86/DRI linux dll problem for module build
-BUGS = -D_DRI_WORKAROUND
+DEFS += -D_DRI_WORKAROUND
 #---
 #BUGS =
 #---------------------------------------------------------------------
@@ -58,7 +57,7 @@
 #PNG = -D_HAVE_LIBPNG 
 #ZLIB = 
 #--- Libpng2 available but needs zlib 
-PNG = -D_HAVE_LIBPNG 
+DEFS += -D_HAVE_LIBPNG 
 ZLIB = -lz
 #--- Libpng2 not available
 #PNG = 
@@ -77,12 +76,7 @@
 PYMOL_INC_DIRS = -I../layer0 -I../layer1 -I../layer2 \
    -I../layer3 -I../layer4 -I../layer5 
 
-C_FLAGS = $(CCOPT1) $(CCOPT2) $(EXT_INC_DIR) $(PYTHON_INC_DIR) \
-   $(XINC_DIR) $(PNG) $(DEFS) $(BUGS)
-
-CFLAGS = $(C_FLAGS)
-
-
+C_FLAGS := $(CFLAGS) $(PYTHON_INC_DIR) $(XINC_DIR) $(DEFS)
 
 
 
