--- mpeglib/lib/input/cdromAccess.cpp.orig	2004-12-17 19:48:52.348984280 +0000
+++ mpeglib/lib/input/cdromAccess.cpp	2004-12-17 19:49:01.656900680 +0000
@@ -26,9 +26,14 @@
   #include <linux/version.h>
   #include <sys/types.h>
   #ifndef __u64
-  #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,70)
-  typedef unsigned long long __u64;
-  #endif
+    #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,70)
+      #include <bits/wordsize.h>
+      #if __WORDSIZE == 64
+        typedef unsigned long __u64;
+      #else
+        typedef unsigned long long __u64;
+      #endif
+    #endif
   #endif
   #include "cdromAccess_Linux.cpp"
 #endif
