diff -U 3 -H -d -r -N -- java-gnome-2.6.0-orig/doc-core/Makefile.in java-gnome-2.6.0-neu/doc-core/Makefile.in
--- java-gnome-2.6.0-orig/doc-core/Makefile.in	2004-04-13 14:39:17.000000000 +0200
+++ java-gnome-2.6.0-neu/doc-core/Makefile.in	2004-04-13 22:29:38.000000000 +0200
@@ -96,7 +96,7 @@
 # Install targets
 install: faq api
 	-$(INSTALL) -d -m755 $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/core/
-	$(INSTALL) -m644 build-faq/t1.html $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/core/faq.html
+	$(INSTALL) -m644 build-faq/index.html $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/core/faq.html
 	$(INSTALL) -m644 tutorial/*.java $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/core/
 	# TODO: create a makefile in the following place, linking to the jar files,
 	# setting ld-library-path, etc. as determined by the configure script.
diff -U 3 -H -d -r -N -- java-gnome-2.6.0-orig/doc-core/Makefile.in.in java-gnome-2.6.0-neu/doc-core/Makefile.in.in
--- java-gnome-2.6.0-orig/doc-core/Makefile.in.in	2004-04-13 14:39:17.000000000 +0200
+++ java-gnome-2.6.0-neu/doc-core/Makefile.in.in	2004-04-13 22:29:45.000000000 +0200
@@ -43,7 +43,7 @@
 # Install targets
 install: faq api
 	-$(INSTALL) -d -m755 $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/core/
-	$(INSTALL) -m644 build-faq/t1.html $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/core/faq.html
+	$(INSTALL) -m644 build-faq/index.html $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/core/faq.html
 	$(INSTALL) -m644 tutorial/*.java $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/core/
 	# TODO: create a makefile in the following place, linking to the jar files,
 	# setting ld-library-path, etc. as determined by the configure script.
diff -U 3 -H -d -r -N -- java-gnome-2.6.0-orig/doc-gnome/Makefile.in java-gnome-2.6.0-neu/doc-gnome/Makefile.in
--- java-gnome-2.6.0-orig/doc-gnome/Makefile.in	2004-04-13 14:39:17.000000000 +0200
+++ java-gnome-2.6.0-neu/doc-gnome/Makefile.in	2004-04-13 22:41:21.000000000 +0200
@@ -87,7 +87,7 @@
 # Install targets
 install: tutorial api
 	-$(INSTALL) -d -m755 $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/gnome/
-	$(INSTALL) -m644 build-tutorial/book1.html $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/gnome/tutorial.html
+	$(INSTALL) -m644 build-tutorial/*.html $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/gnome/
 	# TODO: create a makefile in the following place, linking to the jar files,
 	# setting ld-library-path, etc. as determined by the configure script.
 	-$(INSTALL) -d -m755 $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/api/gnome/org/gnu/gnome/class-use/
diff -U 3 -H -d -r -N -- java-gnome-2.6.0-orig/doc-gnome/Makefile.in.in java-gnome-2.6.0-neu/doc-gnome/Makefile.in.in
--- java-gnome-2.6.0-orig/doc-gnome/Makefile.in.in	2004-04-13 14:39:17.000000000 +0200
+++ java-gnome-2.6.0-neu/doc-gnome/Makefile.in.in	2004-04-13 22:41:32.000000000 +0200
@@ -34,7 +34,7 @@
 # Install targets
 install: tutorial api
 	-$(INSTALL) -d -m755 $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/gnome/
-	$(INSTALL) -m644 build-tutorial/book1.html $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/gnome/tutorial.html
+	$(INSTALL) -m644 build-tutorial/*.html $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/doc/gnome/
 	# TODO: create a makefile in the following place, linking to the jar files,
 	# setting ld-library-path, etc. as determined by the configure script.
 	-$(INSTALL) -d -m755 $(DESTDIR)$(datadir)/doc/java-gnome-$(apiversion)/api/gnome/org/gnu/gnome/class-use/
