--- galeon-1.2.7/src/mozilla_prefs.c.orig	2002-12-26 04:05:23.000000000 +0200
+++ galeon-1.2.7/src/mozilla_prefs.c	2002-12-26 04:06:39.000000000 +0200
@@ -54,7 +54,7 @@
 					FALSE);
 
 	/* set default search engine */
-	mozilla_preference_set ("keyword.URL",_("http://www.google.com/search?q="));
+	mozilla_preference_set ("keyword.URL",_("http://www.google.com/search?ie=UTF-8&oe=UTF-8&q="));
 	mozilla_preference_set_boolean ("keyword.enabled", TRUE);
 	mozilla_preference_set_boolean ("security.checkloaduri", FALSE); 
 
--- galeon-1.2.7/src/newuser_callbacks.c.orig	2002-12-26 04:05:31.000000000 +0200
+++ galeon-1.2.7/src/newuser_callbacks.c	2002-12-26 04:08:04.000000000 +0200
@@ -230,14 +230,14 @@
 			 * the google homepage I think -- MattA) */
 			add_smart (toolbar, "Google", "google",
 				   "http://www.google.com",
-				   _("http://www.google.com/search?q=%s"),
+				   _("http://www.google.com/search?ie=UTF-8&oe=UTF-8&q=%s"),
 				   SHARE_DIR "/google.png", NULL);
 		}
 		if (google_groups_checked)
 		{
 			add_smart (toolbar, "Usenet", "news",
 				   "http://groups.google.com",
-				   "http://groups.google.com/groups?q=%s"
+				   "http://groups.google.com/groups?ie=UTF-8&oe=UTF-8&q=%s"
 				   "&meta=site%3Dgroups", 
 				   SHARE_DIR "/google_groups.png", NULL);
 		}
@@ -245,7 +245,7 @@
 		{
 			add_smart (toolbar, "Images", "img",
 				   "http://images.google.com",
-				   "http://images.google.com/images?q=%s",
+				   "http://images.google.com/images?ie=UTF-8&oe=UTF-8&q=%s",
 				   SHARE_DIR "/google_images.png", NULL);
 		}
 		if (dictionary_checked)
