--- src/org/eclipse/jsch/internal/ui/preference/PreferencePage.java.orig	2007-12-16 14:36:48.000000000 +1300
+++ src/org/eclipse/jsch/internal/ui/preference/PreferencePage.java	2007-12-16 15:05:08.000000000 +1300
@@ -886,10 +886,10 @@
 	      channel.connect();
 	      ChannelSftp c=(ChannelSftp)channel;
 
-	      /* String pwd= */c.pwd(); // Read off the channel
 	      SftpATTRS attr=null;
 	
 	      try{
+	        c.pwd(); // Read off the channel
 	        attr=c.stat(".ssh");} //$NON-NLS-1$
 	      catch(SftpException ee){
           // Ignore
