--- ATITVOExtensions.c.orig	2003-11-11 17:32:14.000000000 +0000
+++ ATITVOExtensions.c	2003-12-29 19:09:46.790701616 +0000
@@ -33,68 +33,15 @@
 #include <stdlib.h>
 
 #include "ATITVOExtensions.h"
-#include "Radeon_extension_tvout.h"
+#include "TVOExt.h"
 
 static XExtensionInfo _atitvo_info_data;
-static XExtensionInfo *atitvo_info = &_atitvo_info_data;
-static char *atitvo_extension_name = ATITVOUT_EXTENSION_NAME;
-
 
 //////////////////////////////////////////////////////////////////////////////
 // private functions
 
 XExtDisplayInfo *info;
 
-// X shutdown hook function that will cleanup the opened extension
-static XEXT_GENERATE_CLOSE_DISPLAY (close_display_atitvo, 
-                                    atitvo_info)
-
-// table with X hooks
-static /* const */ XExtensionHooks atitvo_extension_hooks = {
-    NULL,                   /* create_gc */
-    NULL,                   /* copy_gc */
-    NULL,                   /* flush_gc */
-    NULL,                   /* free_gc */
-    NULL,                   /* create_font */
-    NULL,                   /* free_font */
-    close_display_atitvo,   /* close_display */
-    NULL,                   /* wire_to_event */
-    NULL,                   /* event_to_wire */
-    NULL,                   /* error */
-    NULL,                   /* error_string */
-};
-
-// find_display_atitvo - get the display info block for atitvo extension
-static XEXT_GENERATE_FIND_DISPLAY (find_display_atitvo, 
-                                   atitvo_info,
-                                   atitvo_extension_name, 
-                                   &atitvo_extension_hooks, 
-                                   ATITVOUT_NUMBER_EVENTS, 
-                                   NULL)
-
-
-
-static int 
-answerisyes(char *s)
-{
-    if (s[0] == '\'')   /* For fools that type the ' literally. */
-        return tolower(s[1]) == 'y';
-    return tolower(s[0]) == 'y';
-}
-
-
-static void 
-getstring(char *s)
-{
-    char *cp;
-    fgets(s, 80, stdin);
-    cp = strchr(s, '\n');
-    if (cp)
-        *cp=0;
-}
-
-
-
 //////////////////////////////////////////////////////////////////////////////
 // type defines
 
