diff -ruN build_mod.orig/firegl_public.c build_mod/firegl_public.c
--- build_mod.orig/firegl_public.c	2003-09-14 19:58:43.586456656 +0200
+++ build_mod/firegl_public.c	2003-09-14 20:04:47.975061168 +0200
@@ -419,7 +419,9 @@
 
 static int firegl_stub_open(struct inode *inode, struct file *filp)
 {
-#if LINUX_VERSION_CODE > 0x020500
+#ifdef LINUX_VERSION_CODE >= 0x020600
+       int minor = iminor(inode);
+#elif LINUX_VERSION_CODE > 0x020500
 	int                    minor = minor(inode->i_rdev);
 #else
 	int                    minor = MINOR(inode->i_rdev);
@@ -845,7 +847,9 @@
 
 int __ke_inode_rdev_minor(struct inode* inode)
 {
-#if LINUX_VERSION_CODE > 0x020500
+#ifdef LINUX_VERSION_CODE >= 0x020600
+    return iminor(inode);
+#elif LINUX_VERSION_CODE > 0x020500
     return minor(inode->i_rdev);
 #else
     return MINOR(inode->i_rdev);
