--- help/C/Makefile.old	2004-02-22 11:07:30.758697592 +1100
+++ help/C/Makefile	2004-02-22 11:07:43.102821000 +1100
@@ -368,22 +368,6 @@
 	done
 	-rmdir $(DESTDIR)$(docdir)
 
-omf_timestamp: $(omffile)
-	-for file in $(omffile); do \
-		scrollkeeper-preinstall $(docdir)/$(docname).xml $(srcdir)/$$file $$file.out; \
-	done
-	-for file in $(extraomf); do \
-		scrollkeeper-preinstall $(docdir)/w`echo $$file | cut -f1 -d -`.xml $(srcdir)/$$file $$file.out; \
-	done
-	touch omf_timestamp
-
-install-data-hook-omf:
-	$(mkinstalldirs) $(DESTDIR)$(omf_dest_dir)
-	for file in $(extraomf); do \
-		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
-	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
-
 dist-hook: app-dist-hook
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
