
--- pyqtorrent/torrentwindow.py.orig	2005-02-22 21:55:59.000000000 +0100
+++ pyqtorrent/torrentwindow.py	2005-02-22 21:55:59.000000000 +0100
@@ -52,7 +52,7 @@
         if not name:
             self.setName("TorrentWindow")
 
-        self.setSizePolicy(QSizePolicy(3,3,0,0,self.sizePolicy().hasHeightForWidth()))
+        self.setSizePolicy(QSizePolicy(QSizePolicy.MinimumExpanding,QSizePolicy.MinimumExpanding,0,0,self.sizePolicy().hasHeightForWidth()))
 
         self.setCentralWidget(QWidget(self,"qt_central_widget"))
         TorrentWindowLayout = QVBoxLayout(self.centralWidget(),11,6,"TorrentWindowLayout")
@@ -77,7 +77,7 @@
         WStatsLayout.addWidget(self.textLabel1,2,0)
 
         self.WActive = QLabel(self.WStats,"WActive")
-        self.WActive.setSizePolicy(QSizePolicy(7,5,0,0,self.WActive.sizePolicy().hasHeightForWidth()))
+        self.WActive.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.WActive.sizePolicy().hasHeightForWidth()))
         self.WActive.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         WStatsLayout.addWidget(self.WActive,1,1)
@@ -88,25 +88,25 @@
         WStatsLayout.addWidget(self.WPaused,2,1)
 
         self.WDownRate = QLabel(self.WStats,"WDownRate")
-        self.WDownRate.setSizePolicy(QSizePolicy(7,5,0,0,self.WDownRate.sizePolicy().hasHeightForWidth()))
+        self.WDownRate.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.WDownRate.sizePolicy().hasHeightForWidth()))
         self.WDownRate.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         WStatsLayout.addWidget(self.WDownRate,6,1)
 
         self.WUpRate = QLabel(self.WStats,"WUpRate")
-        self.WUpRate.setSizePolicy(QSizePolicy(7,5,0,0,self.WUpRate.sizePolicy().hasHeightForWidth()))
+        self.WUpRate.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.WUpRate.sizePolicy().hasHeightForWidth()))
         self.WUpRate.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         WStatsLayout.addWidget(self.WUpRate,7,1)
 
         self.WDownTotal = QLabel(self.WStats,"WDownTotal")
-        self.WDownTotal.setSizePolicy(QSizePolicy(7,5,0,0,self.WDownTotal.sizePolicy().hasHeightForWidth()))
+        self.WDownTotal.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.WDownTotal.sizePolicy().hasHeightForWidth()))
         self.WDownTotal.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         WStatsLayout.addWidget(self.WDownTotal,9,1)
 
         self.WUpTotal = QLabel(self.WStats,"WUpTotal")
-        self.WUpTotal.setSizePolicy(QSizePolicy(7,5,0,0,self.WUpTotal.sizePolicy().hasHeightForWidth()))
+        self.WUpTotal.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.WUpTotal.sizePolicy().hasHeightForWidth()))
         self.WUpTotal.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         WStatsLayout.addWidget(self.WUpTotal,10,1)
@@ -124,13 +124,13 @@
         WStatsLayout.addWidget(self.textLabel38,3,0)
 
         self.WFinished = QLabel(self.WStats,"WFinished")
-        self.WFinished.setSizePolicy(QSizePolicy(2,5,0,0,self.WFinished.sizePolicy().hasHeightForWidth()))
+        self.WFinished.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WFinished.sizePolicy().hasHeightForWidth()))
         self.WFinished.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         WStatsLayout.addWidget(self.WFinished,3,1)
 
         self.WFailed = QLabel(self.WStats,"WFailed")
-        self.WFailed.setSizePolicy(QSizePolicy(7,5,0,0,self.WFailed.sizePolicy().hasHeightForWidth()))
+        self.WFailed.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.WFailed.sizePolicy().hasHeightForWidth()))
         self.WFailed.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         WStatsLayout.addWidget(self.WFailed,4,1)
@@ -170,7 +170,7 @@
         self.WTorrents.header().setClickEnabled(0,self.WTorrents.header().count() - 1)
         self.WTorrents.addColumn(self.__tr("Last error"))
         self.WTorrents.header().setClickEnabled(0,self.WTorrents.header().count() - 1)
-        self.WTorrents.setSizePolicy(QSizePolicy(2,2,0,0,self.WTorrents.sizePolicy().hasHeightForWidth()))
+        self.WTorrents.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Ignored,0,0,self.WTorrents.sizePolicy().hasHeightForWidth()))
         self.WTorrents.setSelectionMode(QListView.Extended)
         self.WTorrents.setAllColumnsShowFocus(1)
         self.WTorrents.setItemMargin(5)
@@ -184,7 +184,7 @@
         layout7 = QGridLayout(None,1,1,0,6,"layout7")
 
         self.WFilesize = QLabel(self.WTorrentPage,"WFilesize")
-        self.WFilesize.setSizePolicy(QSizePolicy(2,5,0,0,self.WFilesize.sizePolicy().hasHeightForWidth()))
+        self.WFilesize.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WFilesize.sizePolicy().hasHeightForWidth()))
 
         layout7.addWidget(self.WFilesize,1,1)
 
@@ -205,7 +205,7 @@
         layout7.addWidget(self.textLabel16,3,0)
 
         self.WFilename = QLabel(self.WTorrentPage,"WFilename")
-        self.WFilename.setSizePolicy(QSizePolicy(2,5,0,0,self.WFilename.sizePolicy().hasHeightForWidth()))
+        self.WFilename.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WFilename.sizePolicy().hasHeightForWidth()))
 
         layout7.addWidget(self.WFilename,0,1)
 
@@ -214,7 +214,7 @@
         layout7.addWidget(self.WProgress,3,1)
 
         self.WStatus = QLabel(self.WTorrentPage,"WStatus")
-        self.WStatus.setSizePolicy(QSizePolicy(2,5,0,0,self.WStatus.sizePolicy().hasHeightForWidth()))
+        self.WStatus.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WStatus.sizePolicy().hasHeightForWidth()))
 
         layout7.addWidget(self.WStatus,2,1)
         WTorrentPageLayout.addLayout(layout7)
@@ -224,13 +224,13 @@
         layout6 = QGridLayout(None,1,1,0,6,"layout6")
 
         self.WUpped = QLabel(self.WTorrentPage,"WUpped")
-        self.WUpped.setSizePolicy(QSizePolicy(2,5,0,0,self.WUpped.sizePolicy().hasHeightForWidth()))
+        self.WUpped.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WUpped.sizePolicy().hasHeightForWidth()))
         self.WUpped.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WUpped,1,4)
 
         self.WUpTop = QLabel(self.WTorrentPage,"WUpTop")
-        self.WUpTop.setSizePolicy(QSizePolicy(2,5,0,0,self.WUpTop.sizePolicy().hasHeightForWidth()))
+        self.WUpTop.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WUpTop.sizePolicy().hasHeightForWidth()))
         self.WUpTop.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WUpTop,1,2)
@@ -250,25 +250,25 @@
         layout6.addWidget(self.textLabel7_2,1,0)
 
         self.WUpCurrent = QLabel(self.WTorrentPage,"WUpCurrent")
-        self.WUpCurrent.setSizePolicy(QSizePolicy(2,5,0,0,self.WUpCurrent.sizePolicy().hasHeightForWidth()))
+        self.WUpCurrent.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WUpCurrent.sizePolicy().hasHeightForWidth()))
         self.WUpCurrent.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WUpCurrent,1,1)
 
         self.WDownCurrent = QLabel(self.WTorrentPage,"WDownCurrent")
-        self.WDownCurrent.setSizePolicy(QSizePolicy(2,5,0,0,self.WDownCurrent.sizePolicy().hasHeightForWidth()))
+        self.WDownCurrent.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WDownCurrent.sizePolicy().hasHeightForWidth()))
         self.WDownCurrent.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WDownCurrent,0,1)
 
         self.WUpRatio = QProgressBar(self.WTorrentPage,"WUpRatio")
-        self.WUpRatio.setSizePolicy(QSizePolicy(0,0,0,0,self.WUpRatio.sizePolicy().hasHeightForWidth()))
+        self.WUpRatio.setSizePolicy(QSizePolicy(QSizePolicy.Fixed,QSizePolicy.Fixed,0,0,self.WUpRatio.sizePolicy().hasHeightForWidth()))
         self.WUpRatio.setMaximumSize(QSize(50,32767))
 
         layout6.addWidget(self.WUpRatio,1,3)
 
         self.WDowned = QLabel(self.WTorrentPage,"WDowned")
-        self.WDowned.setSizePolicy(QSizePolicy(2,5,0,0,self.WDowned.sizePolicy().hasHeightForWidth()))
+        self.WDowned.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WDowned.sizePolicy().hasHeightForWidth()))
         self.WDowned.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WDowned,0,4)
@@ -278,13 +278,13 @@
         layout6.addWidget(self.textLabel7,0,0)
 
         self.WDownTop = QLabel(self.WTorrentPage,"WDownTop")
-        self.WDownTop.setSizePolicy(QSizePolicy(2,5,0,0,self.WDownTop.sizePolicy().hasHeightForWidth()))
+        self.WDownTop.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WDownTop.sizePolicy().hasHeightForWidth()))
         self.WDownTop.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WDownTop,0,2)
 
         self.WDownRatio = QProgressBar(self.WTorrentPage,"WDownRatio")
-        self.WDownRatio.setSizePolicy(QSizePolicy(0,0,0,0,self.WDownRatio.sizePolicy().hasHeightForWidth()))
+        self.WDownRatio.setSizePolicy(QSizePolicy(QSizePolicy.Fixed,QSizePolicy.Fixed,0,0,self.WDownRatio.sizePolicy().hasHeightForWidth()))
         self.WDownRatio.setMaximumSize(QSize(50,32767))
 
         layout6.addWidget(self.WDownRatio,0,3)
@@ -295,7 +295,7 @@
         layout14 = QGridLayout(None,1,1,0,6,"layout14")
 
         self.line1 = QFrame(self.WTorrentPage,"line1")
-        self.line1.setSizePolicy(QSizePolicy(0,1,0,0,self.line1.sizePolicy().hasHeightForWidth()))
+        self.line1.setSizePolicy(QSizePolicy(QSizePolicy.Fixed,QSizePolicy.Minimum,0,0,self.line1.sizePolicy().hasHeightForWidth()))
         self.line1.setFrameShape(QFrame.VLine)
         self.line1.setFrameShadow(QFrame.Sunken)
         self.line1.setFrameShape(QFrame.VLine)
@@ -303,7 +303,7 @@
         layout14.addMultiCellWidget(self.line1,0,2,3,3)
 
         self.textLabel40 = QLabel(self.WTorrentPage,"textLabel40")
-        self.textLabel40.setSizePolicy(QSizePolicy(5,5,0,0,self.textLabel40.sizePolicy().hasHeightForWidth()))
+        self.textLabel40.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Preferred,0,0,self.textLabel40.sizePolicy().hasHeightForWidth()))
 
         layout14.addWidget(self.textLabel40,0,0)
 
@@ -338,7 +338,7 @@
         layout13 = QHBoxLayout(None,0,6,"layout13")
 
         self.WListenPort = QLabel(self.WTorrentPage,"WListenPort")
-        self.WListenPort.setSizePolicy(QSizePolicy(7,5,0,0,self.WListenPort.sizePolicy().hasHeightForWidth()))
+        self.WListenPort.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.WListenPort.sizePolicy().hasHeightForWidth()))
         layout13.addWidget(self.WListenPort)
 
         self.line2 = QFrame(self.WTorrentPage,"line2")
@@ -353,18 +353,18 @@
         layout14.addLayout(layout13,0,5)
 
         self.textLabel41 = QLabel(self.WTorrentPage,"textLabel41")
-        self.textLabel41.setSizePolicy(QSizePolicy(5,5,0,0,self.textLabel41.sizePolicy().hasHeightForWidth()))
+        self.textLabel41.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Preferred,0,0,self.textLabel41.sizePolicy().hasHeightForWidth()))
 
         layout14.addWidget(self.textLabel41,1,0)
 
         layout8 = QHBoxLayout(None,0,6,"layout8")
 
         self.WSeeds = QLabel(self.WTorrentPage,"WSeeds")
-        self.WSeeds.setSizePolicy(QSizePolicy(5,5,0,0,self.WSeeds.sizePolicy().hasHeightForWidth()))
+        self.WSeeds.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Preferred,0,0,self.WSeeds.sizePolicy().hasHeightForWidth()))
         layout8.addWidget(self.WSeeds)
 
         self.WCopies = QLabel(self.WTorrentPage,"WCopies")
-        self.WCopies.setSizePolicy(QSizePolicy(2,5,0,0,self.WCopies.sizePolicy().hasHeightForWidth()))
+        self.WCopies.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WCopies.sizePolicy().hasHeightForWidth()))
         self.WCopies.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
         layout8.addWidget(self.WCopies)
 
@@ -377,11 +377,11 @@
         layout9 = QHBoxLayout(None,0,6,"layout9")
 
         self.WClients = QLabel(self.WTorrentPage,"WClients")
-        self.WClients.setSizePolicy(QSizePolicy(5,5,0,0,self.WClients.sizePolicy().hasHeightForWidth()))
+        self.WClients.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Preferred,0,0,self.WClients.sizePolicy().hasHeightForWidth()))
         layout9.addWidget(self.WClients)
 
         self.WClientStats = QLabel(self.WTorrentPage,"WClientStats")
-        self.WClientStats.setSizePolicy(QSizePolicy(2,5,0,0,self.WClientStats.sizePolicy().hasHeightForWidth()))
+        self.WClientStats.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WClientStats.sizePolicy().hasHeightForWidth()))
         self.WClientStats.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
         layout9.addWidget(self.WClientStats)
 
@@ -407,7 +407,7 @@
         layout26 = QHBoxLayout(None,0,6,"layout26")
 
         self.WDropper = TorrentDrop(self.centralWidget(),"WDropper")
-        self.WDropper.setSizePolicy(QSizePolicy(1,0,0,0,self.WDropper.sizePolicy().hasHeightForWidth()))
+        self.WDropper.setSizePolicy(QSizePolicy(QSizePolicy.Minimum,QSizePolicy.Fixed,0,0,self.WDropper.sizePolicy().hasHeightForWidth()))
         layout26.addWidget(self.WDropper)
 
         self.line5 = QFrame(self.centralWidget(),"line5")
@@ -429,11 +429,11 @@
         layout26.addWidget(self.line6)
 
         self.WPrevButton = QPushButton(self.centralWidget(),"WPrevButton")
-        self.WPrevButton.setSizePolicy(QSizePolicy(0,0,0,0,self.WPrevButton.sizePolicy().hasHeightForWidth()))
+        self.WPrevButton.setSizePolicy(QSizePolicy(QSizePolicy.Fixed,QSizePolicy.Fixed,0,0,self.WPrevButton.sizePolicy().hasHeightForWidth()))
         layout26.addWidget(self.WPrevButton)
 
         self.WNextButton = QPushButton(self.centralWidget(),"WNextButton")
-        self.WNextButton.setSizePolicy(QSizePolicy(0,0,0,0,self.WNextButton.sizePolicy().hasHeightForWidth()))
+        self.WNextButton.setSizePolicy(QSizePolicy(QSizePolicy.Fixed,QSizePolicy.Fixed,0,0,self.WNextButton.sizePolicy().hasHeightForWidth()))
         layout26.addWidget(self.WNextButton)
 
         self.line7 = QFrame(self.centralWidget(),"line7")
--- pyqtorrent/torrentsettings.py.orig	2005-02-22 21:55:59.000000000 +0100
+++ pyqtorrent/torrentsettings.py	2005-02-22 21:55:59.000000000 +0100
@@ -34,12 +34,12 @@
         TorrentSettingsLayout.addMultiCellLayout(layout56,3,3,0,1)
 
         self.listBox1 = QListBox(self,"listBox1")
-        self.listBox1.setSizePolicy(QSizePolicy(1,7,0,0,self.listBox1.sizePolicy().hasHeightForWidth()))
+        self.listBox1.setSizePolicy(QSizePolicy(QSizePolicy.Minimum,QSizePolicy.Expanding,0,0,self.listBox1.sizePolicy().hasHeightForWidth()))
 
         TorrentSettingsLayout.addMultiCellWidget(self.listBox1,0,2,0,0)
 
         self.widgetStack1 = QWidgetStack(self,"widgetStack1")
-        self.widgetStack1.setSizePolicy(QSizePolicy(7,5,0,0,self.widgetStack1.sizePolicy().hasHeightForWidth()))
+        self.widgetStack1.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.widgetStack1.sizePolicy().hasHeightForWidth()))
 
         self.WStackPage = QWidget(self.widgetStack1,"WStackPage")
         WStackPageLayout = QGridLayout(self.WStackPage,1,1,11,6,"WStackPageLayout")
@@ -53,12 +53,12 @@
         WStackPageLayout.addWidget(self.textLabel1,0,0)
 
         self.WBind = QLineEdit(self.WStackPage,"WBind")
-        self.WBind.setSizePolicy(QSizePolicy(5,0,0,0,self.WBind.sizePolicy().hasHeightForWidth()))
+        self.WBind.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Fixed,0,0,self.WBind.sizePolicy().hasHeightForWidth()))
 
         WStackPageLayout.addWidget(self.WBind,0,1)
 
         self.WIP = QLineEdit(self.WStackPage,"WIP")
-        self.WIP.setSizePolicy(QSizePolicy(5,0,0,0,self.WIP.sizePolicy().hasHeightForWidth()))
+        self.WIP.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Fixed,0,0,self.WIP.sizePolicy().hasHeightForWidth()))
 
         WStackPageLayout.addWidget(self.WIP,1,1)
 
@@ -79,14 +79,14 @@
         WStackPageLayout.addWidget(self.textLabel4,3,0)
 
         self.WMinPort = QSpinBox(self.WStackPage,"WMinPort")
-        self.WMinPort.setSizePolicy(QSizePolicy(5,0,0,0,self.WMinPort.sizePolicy().hasHeightForWidth()))
+        self.WMinPort.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Fixed,0,0,self.WMinPort.sizePolicy().hasHeightForWidth()))
         self.WMinPort.setMaxValue(65535)
         self.WMinPort.setMinValue(1)
 
         WStackPageLayout.addWidget(self.WMinPort,2,1)
 
         self.WMaxPort = QSpinBox(self.WStackPage,"WMaxPort")
-        self.WMaxPort.setSizePolicy(QSizePolicy(5,0,0,0,self.WMaxPort.sizePolicy().hasHeightForWidth()))
+        self.WMaxPort.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Fixed,0,0,self.WMaxPort.sizePolicy().hasHeightForWidth()))
         self.WMaxPort.setMaxValue(65535)
         self.WMaxPort.setMinValue(1)
 
--- pyqtorrent/torrentwidget.py.orig	2005-02-22 21:55:59.000000000 +0100
+++ pyqtorrent/torrentwidget.py	2005-02-22 21:55:59.000000000 +0100
@@ -18,14 +18,14 @@
         if not name:
             self.setName("TorrentWidget")
 
-        self.setSizePolicy(QSizePolicy(7,7,0,0,self.sizePolicy().hasHeightForWidth()))
+        self.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Expanding,0,0,self.sizePolicy().hasHeightForWidth()))
 
         TorrentWidgetLayout = QVBoxLayout(self,0,6,"TorrentWidgetLayout")
 
         layout7 = QGridLayout(None,1,1,0,6,"layout7")
 
         self.WFilesize = QLabel(self,"WFilesize")
-        self.WFilesize.setSizePolicy(QSizePolicy(2,5,0,0,self.WFilesize.sizePolicy().hasHeightForWidth()))
+        self.WFilesize.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WFilesize.sizePolicy().hasHeightForWidth()))
 
         layout7.addWidget(self.WFilesize,1,1)
 
@@ -46,7 +46,7 @@
         layout7.addWidget(self.textLabel16,3,0)
 
         self.WFilename = QLabel(self,"WFilename")
-        self.WFilename.setSizePolicy(QSizePolicy(2,5,0,0,self.WFilename.sizePolicy().hasHeightForWidth()))
+        self.WFilename.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WFilename.sizePolicy().hasHeightForWidth()))
 
         layout7.addWidget(self.WFilename,0,1)
 
@@ -55,7 +55,7 @@
         layout7.addWidget(self.WProgress,3,1)
 
         self.WStatus = QLabel(self,"WStatus")
-        self.WStatus.setSizePolicy(QSizePolicy(2,5,0,0,self.WStatus.sizePolicy().hasHeightForWidth()))
+        self.WStatus.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WStatus.sizePolicy().hasHeightForWidth()))
 
         layout7.addWidget(self.WStatus,2,1)
         TorrentWidgetLayout.addLayout(layout7)
@@ -65,13 +65,13 @@
         layout6 = QGridLayout(None,1,1,0,6,"layout6")
 
         self.WUpTotal = QLabel(self,"WUpTotal")
-        self.WUpTotal.setSizePolicy(QSizePolicy(2,5,0,0,self.WUpTotal.sizePolicy().hasHeightForWidth()))
+        self.WUpTotal.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WUpTotal.sizePolicy().hasHeightForWidth()))
         self.WUpTotal.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WUpTotal,1,4)
 
         self.WUpTop = QLabel(self,"WUpTop")
-        self.WUpTop.setSizePolicy(QSizePolicy(2,5,0,0,self.WUpTop.sizePolicy().hasHeightForWidth()))
+        self.WUpTop.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WUpTop.sizePolicy().hasHeightForWidth()))
         self.WUpTop.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WUpTop,1,2)
@@ -91,26 +91,26 @@
         layout6.addWidget(self.textLabel7_2,1,0)
 
         self.WUpCurrent = QLabel(self,"WUpCurrent")
-        self.WUpCurrent.setSizePolicy(QSizePolicy(2,5,0,0,self.WUpCurrent.sizePolicy().hasHeightForWidth()))
+        self.WUpCurrent.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WUpCurrent.sizePolicy().hasHeightForWidth()))
         self.WUpCurrent.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WUpCurrent,1,1)
 
         self.WDownCurrent = QLabel(self,"WDownCurrent")
-        self.WDownCurrent.setSizePolicy(QSizePolicy(2,5,0,0,self.WDownCurrent.sizePolicy().hasHeightForWidth()))
+        self.WDownCurrent.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WDownCurrent.sizePolicy().hasHeightForWidth()))
         self.WDownCurrent.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WDownCurrent,0,1)
 
         self.WUpRatio = QProgressBar(self,"WUpRatio")
-        self.WUpRatio.setSizePolicy(QSizePolicy(0,0,0,0,self.WUpRatio.sizePolicy().hasHeightForWidth()))
+        self.WUpRatio.setSizePolicy(QSizePolicy(QSizePolicy.Fixed,QSizePolicy.Fixed,0,0,self.WUpRatio.sizePolicy().hasHeightForWidth()))
         self.WUpRatio.setMaximumSize(QSize(50,32767))
         self.WUpRatio.setPercentageVisible(0)
 
         layout6.addWidget(self.WUpRatio,1,3)
 
         self.WDownTotal = QLabel(self,"WDownTotal")
-        self.WDownTotal.setSizePolicy(QSizePolicy(2,5,0,0,self.WDownTotal.sizePolicy().hasHeightForWidth()))
+        self.WDownTotal.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WDownTotal.sizePolicy().hasHeightForWidth()))
         self.WDownTotal.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WDownTotal,0,4)
@@ -120,13 +120,13 @@
         layout6.addWidget(self.textLabel7,0,0)
 
         self.WDownTop = QLabel(self,"WDownTop")
-        self.WDownTop.setSizePolicy(QSizePolicy(2,5,0,0,self.WDownTop.sizePolicy().hasHeightForWidth()))
+        self.WDownTop.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WDownTop.sizePolicy().hasHeightForWidth()))
         self.WDownTop.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
 
         layout6.addWidget(self.WDownTop,0,2)
 
         self.WDownRatio = QProgressBar(self,"WDownRatio")
-        self.WDownRatio.setSizePolicy(QSizePolicy(0,0,0,0,self.WDownRatio.sizePolicy().hasHeightForWidth()))
+        self.WDownRatio.setSizePolicy(QSizePolicy(QSizePolicy.Fixed,QSizePolicy.Fixed,0,0,self.WDownRatio.sizePolicy().hasHeightForWidth()))
         self.WDownRatio.setMaximumSize(QSize(50,32767))
         self.WDownRatio.setPercentageVisible(0)
 
@@ -138,7 +138,7 @@
         layout14 = QGridLayout(None,1,1,0,6,"layout14")
 
         self.line1 = QFrame(self,"line1")
-        self.line1.setSizePolicy(QSizePolicy(0,1,0,0,self.line1.sizePolicy().hasHeightForWidth()))
+        self.line1.setSizePolicy(QSizePolicy(QSizePolicy.Fixed,QSizePolicy.Minimum,0,0,self.line1.sizePolicy().hasHeightForWidth()))
         self.line1.setFrameShape(QFrame.VLine)
         self.line1.setFrameShadow(QFrame.Sunken)
         self.line1.setFrameShape(QFrame.VLine)
@@ -146,7 +146,7 @@
         layout14.addMultiCellWidget(self.line1,0,2,3,3)
 
         self.textLabel40 = QLabel(self,"textLabel40")
-        self.textLabel40.setSizePolicy(QSizePolicy(5,5,0,0,self.textLabel40.sizePolicy().hasHeightForWidth()))
+        self.textLabel40.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Preferred,0,0,self.textLabel40.sizePolicy().hasHeightForWidth()))
 
         layout14.addWidget(self.textLabel40,0,0)
 
@@ -183,7 +183,7 @@
         layout13 = QHBoxLayout(None,0,6,"layout13")
 
         self.WListenPort = QLabel(self,"WListenPort")
-        self.WListenPort.setSizePolicy(QSizePolicy(7,5,0,0,self.WListenPort.sizePolicy().hasHeightForWidth()))
+        self.WListenPort.setSizePolicy(QSizePolicy(QSizePolicy.Expanding,QSizePolicy.Preferred,0,0,self.WListenPort.sizePolicy().hasHeightForWidth()))
         layout13.addWidget(self.WListenPort)
 
         self.line2 = QFrame(self,"line2")
@@ -199,18 +199,18 @@
         layout14.addLayout(layout13,0,5)
 
         self.textLabel41 = QLabel(self,"textLabel41")
-        self.textLabel41.setSizePolicy(QSizePolicy(5,5,0,0,self.textLabel41.sizePolicy().hasHeightForWidth()))
+        self.textLabel41.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Preferred,0,0,self.textLabel41.sizePolicy().hasHeightForWidth()))
 
         layout14.addWidget(self.textLabel41,1,0)
 
         layout8 = QHBoxLayout(None,0,6,"layout8")
 
         self.WSeeds = QLabel(self,"WSeeds")
-        self.WSeeds.setSizePolicy(QSizePolicy(5,5,0,0,self.WSeeds.sizePolicy().hasHeightForWidth()))
+        self.WSeeds.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Preferred,0,0,self.WSeeds.sizePolicy().hasHeightForWidth()))
         layout8.addWidget(self.WSeeds)
 
         self.WCopies = QLabel(self,"WCopies")
-        self.WCopies.setSizePolicy(QSizePolicy(2,5,0,0,self.WCopies.sizePolicy().hasHeightForWidth()))
+        self.WCopies.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WCopies.sizePolicy().hasHeightForWidth()))
         self.WCopies.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
         layout8.addWidget(self.WCopies)
 
@@ -223,11 +223,11 @@
         layout9 = QHBoxLayout(None,0,6,"layout9")
 
         self.WClients = QLabel(self,"WClients")
-        self.WClients.setSizePolicy(QSizePolicy(5,5,0,0,self.WClients.sizePolicy().hasHeightForWidth()))
+        self.WClients.setSizePolicy(QSizePolicy(QSizePolicy.Preferred,QSizePolicy.Preferred,0,0,self.WClients.sizePolicy().hasHeightForWidth()))
         layout9.addWidget(self.WClients)
 
         self.WClientStats = QLabel(self,"WClientStats")
-        self.WClientStats.setSizePolicy(QSizePolicy(2,5,0,0,self.WClientStats.sizePolicy().hasHeightForWidth()))
+        self.WClientStats.setSizePolicy(QSizePolicy(QSizePolicy.Ignored,QSizePolicy.Preferred,0,0,self.WClientStats.sizePolicy().hasHeightForWidth()))
         self.WClientStats.setAlignment(QLabel.AlignVCenter | QLabel.AlignRight)
         layout9.addWidget(self.WClientStats)
 
