diff -urN scli-0.2.12.orig/scli/atm.c scli-0.2.12/scli/atm.c
--- scli-0.2.12.orig/scli/atm.c	2005-01-28 01:19:14.000000000 -0500
+++ scli-0.2.12/scli/atm.c	2005-01-28 02:44:55.501555168 -0500
@@ -25,7 +25,7 @@
 #include "atm-mib.h"
 #include "if-mib.h"
 
-
+int interface_match(regex_t *regex, if_mib_ifEntry_t *ifEntry);
 
 static void
 fmt_atm_interface_info(GString *s,
diff -urN scli-0.2.12.orig/scli/isdn.c scli-0.2.12/scli/isdn.c
--- scli-0.2.12.orig/scli/isdn.c	2005-01-28 01:19:14.000000000 -0500
+++ scli-0.2.12/scli/isdn.c	2005-01-28 02:45:24.388163736 -0500
@@ -25,6 +25,7 @@
 #include "if-mib.h"
 #include "isdn-mib.h"
 
+int interface_match(regex_t *regex, if_mib_ifEntry_t *ifEntry);
 
 static void
 fmt_isdn_bri(GString *s, isdn_mib_isdnBasicRateEntry_t *briEntry,
@@ -227,7 +228,7 @@
 
     if (endpointEntry->isdnEndpointIfIndex) {
 	g_string_sprintfa(s, "%-*s %d\n", indent, "Interface:",
-			  endpointEntry->isdnEndpointIfIndex);
+			  *(endpointEntry->isdnEndpointIfIndex));
     }
 
 #if 0
diff -urN scli-0.2.12.orig/scli/sonet.c scli-0.2.12/scli/sonet.c
--- scli-0.2.12.orig/scli/sonet.c	2005-01-28 01:19:14.000000000 -0500
+++ scli-0.2.12/scli/sonet.c	2005-01-28 02:45:15.788471088 -0500
@@ -31,6 +31,7 @@
 #include "if-mib.h"
 #include "sonet-mib.h"
 
+int interface_match(regex_t *regex, if_mib_ifEntry_t *ifEntry);
 
 static GSnmpEnum const sonetMediumType[] = {
     { SONET_MIB_SONETMEDIUMTYPE_SONET, "SONET" },
diff -urN scli-0.2.12.orig/snmp/g_session.c scli-0.2.12/snmp/g_session.c
--- scli-0.2.12.orig/snmp/g_session.c	2005-01-28 01:19:14.000000000 -0500
+++ scli-0.2.12/snmp/g_session.c	2005-01-28 02:43:13.875004752 -0500
@@ -21,13 +21,12 @@
  * $Id: scli-0.2.12-gentoo.diff,v 1.1 2005/01/28 11:49:51 ka0ttic Exp $
  */
 
-#include "g_snmp.h"
-
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/socket.h>
 #include <sys/time.h>
 #include <time.h>
+#include "g_snmp.h"
 
 static GSList  *request_queue = NULL;   /* queue of active requests */
 
diff -urN scli-0.2.12.orig/snmp/g_snmp.h scli-0.2.12/snmp/g_snmp.h
--- scli-0.2.12.orig/snmp/g_snmp.h	2005-01-28 01:19:14.000000000 -0500
+++ scli-0.2.12/snmp/g_snmp.h	2005-01-28 02:41:37.152708776 -0500
@@ -28,15 +28,15 @@
 #include "config.h"
 #endif
 
-#ifdef HAVE_DMALLOC_H
-#include <dmalloc.h>
-#endif
-
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <string.h>
 #include <glib.h>
 
+#ifdef HAVE_DMALLOC_H
+#include <dmalloc.h>
+#endif
+
 /* Our Enterprise Handle */
 
 #define ENTERPRISE_GNOME 3319
