--- lsp/Makefile.pamphlet~	2006-01-13 18:16:07.000000000 +0000
+++ lsp/Makefile.pamphlet	2006-01-13 18:15:33.000000000 +0000
@@ -16,6 +16,14 @@
 
 Patches are applied to the distribution before we do the configure.
 \section{Gnu Common Lisp 2.6.7}
+\subsubsection{fix-configure.in-gentoo.patch}
+Fix a strange configure peculiarity; Gentoo bug #118633. Patch backported
+from the gcl fix by Matthew Kennedy.
+<<gcl-2.6.7.fix-configure.in-gentoo.patch>>=
+	@(cd ${GCLVERSION} ; \
+	  echo 30 applying fix-configure.in-gentoo.patch ; \
+	  ${PATCH} <${SPD}/zips/${GCLVERSION}.fix-configure.in-gentoo.patch )
+@
 \subsubsection{fix-out-of-build-root-compile.patch}
 GCL's compiler system includes a -I<gclbuildroot> flag in the gcc args;
 which causes issues if <gclbuildroot> does not exist at a later date and
@@ -1031,6 +1039,7 @@
 <<gcl-2.6.7.tail-recursive.patch>>
 <<gcl-2.6.7.collectfn.fix>>
 <<gcl-2.6.7.fix-out-of-build-root-compile.patch>>
+<<gcl-2.6.7.fix-configure.in-gentoo.patch>>
 <<gclConfigureMake>>
 	@echo 13 finished system build on `date` | tee >gcldir
 
