--- kdirstat/Makefile.am.orig	2008-09-15 00:43:37.000000000 +0200
+++ kdirstat/Makefile.am	2008-09-15 00:44:16.000000000 +0200
@@ -71,7 +71,7 @@
 
 KDE_ICON = kdirstat
 
-applnkdir = $(kde_appsdir)/Utilities
+applnkdir = $(xdg_appsdir)
 applnk_DATA = kdirstat.desktop
 
 
--- kdirstat/kdirstat.desktop.orig	2008-09-15 01:01:06.000000000 +0200
+++ kdirstat/kdirstat.desktop	2008-09-15 01:02:18.000000000 +0200
@@ -1,20 +1,14 @@
-# KDE Config File
 [Desktop Entry]
+MimeType=inode/directory;
+Exec=kdirstat -caption "%c" %i
+Icon=kdirstat
 Type=Application
-MimeType=inode/directory
-Exec=kdirstat -caption "%c" %i %m
-Icon=kdirstat.png
-MiniIcon=kdirstat.png
-DocPath=kdirstat/index.html
-Encoding=UTF-8
-Terminal=0
+X-DocPath=kdirstat/index.html
 Name=KDirStat
+Categories=Qt;KDE;Utility;X-KDE-Utilities-File;
 GenericName=Directory Statistics
 GenericName[de]=Verzeichnisstatistik
 GenericName[hu]=könyvtárstatisztika
 Comment=Directory statistics and disk usage
 Comment[de]=Verzeichnisstatistik und Platzverbrauch
 Comment[hu]=Könyvtárstatisztikák és szabad hely
-
-
-
