--- usr/include/GL/gl.g.orig	2004-07-17 19:56:59.789410584 +1000
+++ usr/include/GL/gl.h	2004-07-17 19:59:08.844791184 +1000
@@ -66,6 +66,11 @@
 typedef double GLclampd;
 typedef void GLvoid;
 
+/* Patching for some better defines in the global system */
+#define GL_VERSION_1_2
+#define GL_VERSION_1_3
+#define GL_VERSION_1_4
+#include <GL/glext.h>
 
 /*************************************************************/
 
