diff -Naur skycat-3.0.2/tclutil/generic/HTTP.C skycat-3.0.2.new/tclutil/generic/HTTP.C
--- skycat-3.0.2/tclutil/generic/HTTP.C	2006-01-12 11:40:58.000000000 -0500
+++ skycat-3.0.2.new/tclutil/generic/HTTP.C	2009-09-16 00:13:30.000000000 -0400
@@ -1108,7 +1108,7 @@
             //  host. Note need a copy of variable as strtok modifies it.
             char *ptr = getenv( "http_noproxy" );
             if ( ptr != NULL ) {
-                char *hostdomain = strchr(host, '.');
+                char *hostdomain = strchr(const_cast<char*>(host), '.');
                 if (hostdomain != NULL) {
                     hostdomain++;
 		    // make a copy of the http_noproxy string for strtok
