--- astrotcl/generic/ImageIO.h.orig	2010-09-24 06:31:03.000000000 +0100
+++ astrotcl/generic/ImageIO.h	2010-09-24 06:32:06.000000000 +0100
@@ -28,6 +28,7 @@
 
 #include <iostream>
 #include <cmath>
+#include <cstdlib>
 #include "WCSRep.h"
 #include "Mem.h"
 
--- ./rtd/generic/histzero.c.orig	2010-09-24 06:36:08.000000000 +0100
+++ ./rtd/generic/histzero.c	2010-09-24 06:37:48.000000000 +0100
@@ -15,6 +15,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include "histeq.h"		/* define SubrangeLink */
 
 static void merge_links();
--- ./rtd/generic/histmap.c.orig	2010-09-24 06:35:59.000000000 +0100
+++ ./rtd/generic/histmap.c	2010-09-24 06:37:26.000000000 +0100
@@ -17,6 +17,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include "histeq.h"		/* define SubrangeLink, List */
 
 
