--- trunk/KDE/kdebase/workspace/kwin/effects/logout/logout.cpp	2009/10/05 09:47:13	1031490
+++ trunk/KDE/kdebase/workspace/kwin/effects/logout/logout.cpp	2009/10/07 15:10:15	1032334
@@ -110,11 +110,11 @@
             frameDelay = 2;
             }
         }
-#endif
 
     if( frameDelay )
         --frameDelay;
     else
+#endif
         {
         if( logoutWindow != NULL && !logoutWindowClosed )
             progress = qMin( 1.0, progress + time / animationTime( 2000.0 ));
@@ -234,14 +234,16 @@
 
 void LogoutEffect::postPaintScreen()
     {
+#ifdef KWIN_HAVE_OPENGL_COMPOSITING
     if(( progress != 0.0 && progress != 1.0 ) || frameDelay )
-        {
         effects->addRepaintFull();
-        }
+#else
+    if( progress != 0.0 && progress != 1.0 )
+        effects->addRepaintFull();
+#endif
+
     if( progress > 0.0 )
-        {
         logoutWindowPassed = false;
-        }
     effects->postPaintScreen();
     }
 
