--- configure.in.orig	2006-07-29 12:27:29.000000000 +0200
+++ configure.in	2006-07-29 12:28:20.000000000 +0200
@@ -73,12 +73,6 @@
              AC_MSG_ERROR([*** SDL-Mixer not found!]))
 AC_CHECK_HEADER(SDL/SDL_mixer.h, , AC_MSG_ERROR([*** SDL-mixer devel files not found!]))
 
-AC_ARG_VAR(WINDRES, [Windows Resource compiler tool path])
-AC_PATH_TOOL(WINDRES,windres,
-	[AC_MSG_WARN(Could not find a windres tool in your PATH.)]
-)
-AC_SUBST(WINDRES)
-
 # Be paranoid
 CFLAGS="$CFLAGS -Wall"
 dnl Checks for header files
