diff -urpN tclxml-2.4.orig/expat/Makefile.in tclxml-2.4/expat/Makefile.in
--- tclxml-2.4.orig/expat/Makefile.in	2009-08-08 03:56:41.000000000 +0200
+++ tclxml-2.4/expat/Makefile.in	2009-08-08 03:58:13.000000000 +0200
@@ -31,14 +31,10 @@ FILEMAP	=	@FILEMAP@
 
 Tclexpat_SOURCES	=	\
 		tclexpat.c	\
-		xmltok.c	\
-		xmlrole.c	\
-		xmlwf.c		\
 		xmlfile.c	\
 		codepage.c	\
 		hashtable.c	\
 		$(FILEMAP).c	\
-		xmlparse.c	\
 		@EXTRA_SOURCES@
 
 WIN_SOURCES	= 
@@ -143,8 +139,8 @@ OBJEXT		= @OBJEXT@
 RANLIB		= @RANLIB@
 SHLIB_CFLAGS	= @SHLIB_CFLAGS@
 SHLIB_LD	= @SHLIB_LD@
-SHLIB_LDFLAGS	= @SHLIB_LDFLAGS@
-SHLIB_LD_LIBS	= @SHLIB_LD_LIBS@
+SHLIB_LDFLAGS	= @SHLIB_LDFLAGS@ $(LDFLAGS)
+SHLIB_LD_LIBS	= @SHLIB_LD_LIBS@ -lexpat
 STLIB_LD	= @STLIB_LD@
 TCL_DEFS	= @TCL_DEFS@
 TCL_BIN_DIR	= @TCL_BIN_DIR@
@@ -230,7 +226,7 @@ install-binaries: binaries install-lib-b
 install-libraries: libraries
 	@mkdir -p $(DESTDIR)$(includedir)
 	@echo "Installing header files in $(DESTDIR)$(includedir)"
-	@for i in $(GENERIC_HDRS) ; do \
+	@for i in ; do \
 	    echo "Installing $$i" ; \
 	    $(INSTALL_DATA) $$i $(DESTDIR)$(includedir) ; \
 	done;
diff -urpN tclxml-2.4.orig/expat/xmlwf/xmlfile.c tclxml-2.4/expat/xmlwf/xmlfile.c
--- tclxml-2.4.orig/expat/xmlwf/xmlfile.c	2009-08-08 03:56:41.000000000 +0200
+++ tclxml-2.4/expat/xmlwf/xmlfile.c	2009-08-08 03:58:56.000000000 +0200
@@ -33,7 +33,7 @@ your version of this file under either t
 #include <stddef.h>
 #include <string.h>
 #include <fcntl.h>
-#include "xmlparse.h"
+#include <expat.h>
 #include "xmlfile.h"
 #include "xmltchar.h"
 #include "filemap.h"
diff -urpN tclxml-2.4.orig/expat/xmlwf/xmlwf.c tclxml-2.4/expat/xmlwf/xmlwf.c
--- tclxml-2.4.orig/expat/xmlwf/xmlwf.c	2009-08-08 03:56:41.000000000 +0200
+++ tclxml-2.4/expat/xmlwf/xmlwf.c	2009-08-08 03:59:13.000000000 +0200
@@ -33,7 +33,7 @@ your version of this file under either t
 #include <stddef.h>
 #include <string.h>
 
-#include "xmlparse.h"
+#include <expat.h>
 #include "codepage.h"
 #include "xmlfile.h"
 #include "xmltchar.h"
diff -urpN tclxml-2.4.orig/tclexpat.c tclxml-2.4/tclexpat.c
--- tclxml-2.4.orig/tclexpat.c	2009-08-08 03:56:41.000000000 +0200
+++ tclxml-2.4/tclexpat.c	2009-08-08 04:02:20.000000000 +0200
@@ -20,7 +20,7 @@
 
 #include <tcl.h>
 #include "tclxml.h"
-#include <xmlparse.h>
+#include <expat.h>
 
 #undef TCL_STORAGE_CLASS
 #define TCL_STORAGE_CLASS DLLEXPORT
@@ -467,7 +467,16 @@ TclExpatConfigure (clientData, objc, obj
 	  return TCL_ERROR;
 	}
 	
+        /*
+         * This method is not exported, so kludge around it by calling one of the
+         * SetDefaultHandler* functions that also sets the variable to the desired
+         * value.
         XML_SetDefaultExpandInternalEntities(expat->parser, bool);
+         */
+        if (bool)
+          XML_SetDefaultHandlerExpand(expat->parser, TclExpatDefaultHandler);
+        else
+          XML_SetDefaultHandler(expat->parser, TclExpatDefaultHandler);
 
 	break;
 
