--- timidity.cfg.old	1999-12-14 10:13:04.000000000 -0500
+++ timidity.cfg	2002-05-19 15:25:51.000000000 -0400
@@ -27,7 +27,7 @@
 # If you've moved things around since compiling, uncomment and edit this to
 # point to the library directory: 
 #
-dir c:\timidity
+dir /usr/share/timidity
 #
 
 ##############################################################################
@@ -58,7 +58,7 @@
 # If you have the GUS General MIDI patch set, uncomment the following lines 
 # and edit the "dir" statement to point to your GUS patch directory. 
 #
-dir c:\eawpats
+dir /usr/share/timidity/eawpatches
 source gravis.cfg
 source gsdrums.cfg
 source gssfx.cfg
