
Index: linux-2.6.9-rsbac-v1.2.3/include/rsbac/fs.h
===================================================================
--- linux-2.6.9-rsbac-v1.2.3/include/rsbac/fs.h	(revision 46)
+++ linux-2.6.9-rsbac-v1.2.3/include/rsbac/fs.h	(working copy)
@@ -33,7 +33,9 @@
 
 struct super_block * rsbac_get_super_block(kdev_t kdev);
 
+#ifndef __fput
 extern void __fput(struct file *);
+#endif
 
 #ifndef SHM_FS_MAGIC
 #define SHM_FS_MAGIC 0x02011994
Index: linux-2.6.9-rsbac-v1.2.3/rsbac/adf/reg/kproc_hide.c
===================================================================
--- linux-2.6.9-rsbac-v1.2.3/rsbac/adf/reg/kproc_hide.c	(revision 46)
+++ linux-2.6.9-rsbac-v1.2.3/rsbac/adf/reg/kproc_hide.c	(working copy)
@@ -10,6 +10,8 @@
 #include <linux/kernel.h>
 #include <linux/string.h>
 #include <linux/fs.h>
+#include <linux/sched.h>
+#include <linux/file.h>
 #include <rsbac/types.h>
 #include <rsbac/reg.h>
 #include <rsbac/adf.h>
@@ -17,8 +19,6 @@
 #include <rsbac/getname.h>
 #include <rsbac/error.h>
 #include <rsbac/proc_fs.h>
-#include <linux/sched.h>
-#include <linux/file.h>
 
 MODULE_AUTHOR("Michal Purzynski");
 MODULE_DESCRIPTION("RSBAC REG kproc_hide decision module");
