--- startkde.orig	2005-02-23 13:24:24.000000000 +0200
+++ startkde	2005-02-27 21:08:37.483412112 +0200
@@ -14,14 +14,42 @@
 	exit 1
 fi
 
-# Set the background to plain grey.
+# Set the background to plain cyan.
 # The standard X background is nasty, causing moire effects and exploding
 # people's heads. We use colours from the standard KDE palette for those with
 # palettised displays.
 if test -z "$XDM_MANAGED" || echo "$XDM_MANAGED" | grep ",auto" > /dev/null; then
-  xsetroot -solid "#C0C0C0"
+	xsetroot -solid "#5477A0"
 fi
 
+# Gentoo part begins
+
+export PATH="_KDEDIR_/bin:${PATH}"
+
+# Handle ~/.kde* dirs - separate ones for separate KDEs.
+# We've backtracked on the scheme where .kdeX.Y directories are temporarily
+# moved to .kde.
+cd ~
+if [ -e .kde ]; then
+	if [ ! -L .kde ]; then
+		/bin/mv -f .kde .kde.backup
+	else
+		rm -f .kde
+	fi
+fi
+if [ ! -e .kde3.4 ]; then
+	if [ -e .kde3.3 ]; then
+		/bin/cp -r .kde3.3 .kde3.4
+	elif [ -e .kde3.2 ]; then
+		/bin/cp -r .kde3.2 .kde3.4
+	else
+		/bin/mkdir .kde3.4
+	fi
+fi
+/bin/ln -sf .kde3.4 .kde
+
+# Gentoo part ends
+
 # we have to unset this for Darwin since it will screw up KDE's dynamic-loading
 unset DYLD_FORCE_FLAT_NAMESPACE
 
