diff --git a/gst/parole-gst.c b/gst/parole-gst.c
index 2a51a35..6676675 100644
--- a/gst/parole-gst.c
+++ b/gst/parole-gst.c
@@ -655,7 +655,7 @@ parole_gst_query_duration (ParoleGst *gst)
 	duration =  absolute_duration / GST_SECOND;
 	live = ( absolute_duration == 0 );
 	
-	TRACE ("Duration %lld is_live=%d", duration, live);
+	TRACE ("Duration %" G_GINT64_FORMAT "is_live=%d", duration, live);
 	
 	g_object_set (G_OBJECT (gst->priv->stream),
 		      "absolute-duration", absolute_duration,
@@ -2169,7 +2169,7 @@ gint parole_gst_get_current_cdda_track (ParoleGst *gst)
     
     if ( gst_element_query_position (gst->priv->playbin, &format, &pos) )
     {
-	TRACE ("Pos %lld", pos);
+	TRACE ("Pos %" G_GINT64_FORMAT, pos);
 	ret_val = (gint) pos;
     }
 	
