--- src/xitk/config_wrapper.c	2004/01/25 03:19:14	1.16
+++ src/xitk/config_wrapper.c	2004/03/16 21:13:27	1.17
@@ -140,9 +140,18 @@
  * Handle 'cfg:/' mrl style
  */
 void config_mrl(char *mrl) {
+  xine_cfg_entry_t entry;
   char *key;
   char *config;
   char *_mrl;
+  
+  if (!xine_config_lookup_entry(gGui->xine, "misc.implicit_config", &entry) ||
+      entry.type != XINE_CONFIG_TYPE_BOOL || !entry.num_value) {
+    xine_info(_("You tried to change the configuration with a cfg: MRL.\n"
+		"This is not allowed unless you enable the 'misc.implicit_config' setting "
+		"after reading and understanding its help text."));
+    return;
+  }
 
   xine_strdupa(_mrl, mrl);
   config = strchr(_mrl, '/');
@@ -157,7 +166,6 @@
       *str_value++ = '\0';
 
     if(str_value && strlen(str_value)) {
-      xine_cfg_entry_t entry;
       
       if(xine_config_lookup_entry(gGui->xine, key, &entry)) {
 
--- src/fb/config_wrapper.c	2003/11/08 10:19:48	1.2
+++ src/fb/config_wrapper.c	2004/03/18 14:40:44	1.4
@@ -136,10 +136,19 @@
  * Handle 'cfg:/' mrl style
  */
 void config_mrl(const char *mrl) {
+  xine_cfg_entry_t entry;
   char *key;
   char *config;
   char *_mrl;
 
+  if (!xine_config_lookup_entry(fbxine.xine, "misc.implicit_config", &entry) ||
+      entry.type != XINE_CONFIG_TYPE_BOOL || !entry.num_value) {
+    fprintf(stderr, "You tried to change the configuration with a cfg: MRL.\n"
+	    "This is not allowed unless you enable the 'misc.implicit_config' setting "
+	    "after reading and understanding its help text.");
+    return;
+  }
+
   xine_strdupa(_mrl, mrl);
   config = strchr(_mrl, '/');
   
@@ -153,7 +162,6 @@
       *str_value++ = '\0';
 
     if(str_value && strlen(str_value)) {
-      xine_cfg_entry_t entry;
       
       if(xine_config_lookup_entry(fbxine.xine, key, &entry)) {
 
