--- /trunk/util/fluxbox-generate_menu.in	2005/01/24 16:27:06	3859
+++ trunk/util/fluxbox-generate_menu.in	2005/01/24 16:32:57	3860
@@ -949,8 +949,8 @@
     if [ -f ${MENUFILENAME}.firstbak ]; then
         cp ${MENUFILENAME} ${MENUFILENAME}.firstbak
     fi
-    if [ -s "${MENUFILE}" ]; then
-       mv ${MENUFILE} ${MENUFILE}.bak
+    if [ -s "${MENUFILENAME}" ]; then
+       mv ${MENUFILENAME} ${MENUFILENAME}.bak
     fi
 fi
 
