diff -Naur gts-0.7.6/examples/Makefile.in gts-0.7.6.new/examples/Makefile.in
--- gts-0.7.6/examples/Makefile.in	2006-03-09 19:31:05.000000000 -0500
+++ gts-0.7.6.new/examples/Makefile.in	2008-02-20 09:08:47.000000000 -0500
@@ -139,7 +139,7 @@
 volume_OBJECTS = volume.$(OBJEXT)
 volume_LDADD = $(LDADD)
 volume_DEPENDENCIES = $(top_builddir)/src/libgts.la
-DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
+DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir) -I/usr/include/netpbm
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 @AMDEP_TRUE@DEP_FILES = ./$(DEPDIR)/cartesian.Po \
diff -Naur gts-0.7.6/src/Makefile.in gts-0.7.6.new/src/Makefile.in
--- gts-0.7.6/src/Makefile.in	2006-03-09 19:31:05.000000000 -0500
+++ gts-0.7.6.new/src/Makefile.in	2008-02-20 09:08:17.000000000 -0500
@@ -354,7 +354,7 @@
 	  rm -f "$${dir}/so_locations"; \
 	done
 libgts.la: $(libgts_la_OBJECTS) $(libgts_la_DEPENDENCIES) 
-	$(LINK) -rpath $(libdir) $(libgts_la_LDFLAGS) $(libgts_la_OBJECTS) $(libgts_la_LIBADD) $(LIBS)
+	$(LINK) -rpath $(libdir) $(libgts_la_LDFLAGS) $(libgts_la_OBJECTS) $(libgts_la_LIBADD) $(LIBS) -lm
 install-binSCRIPTS: $(bin_SCRIPTS)
 	@$(NORMAL_INSTALL)
 	test -z "$(bindir)" || $(mkdir_p) "$(DESTDIR)$(bindir)"
