--- examples/Makefile.am.orig	2010-01-15 07:32:31.000000000 +0000
+++ examples/Makefile.am	2010-01-15 07:34:11.000000000 +0000
@@ -8,17 +8,15 @@
 EXTRA_PROGRAMS = happrox
 
 happrox_LDADD = $(LDADD) -lnetpbm
-
+happrox_CPPFLAGS = -I/usr/include/netpbm
 if HAS_NETPBM
 NETPBM_EXTRA = happrox
 endif
 
-bin_PROGRAMS = \
+check_PROGRAMS = \
 	transform \
 	delaunay \
-	$(NETPBM_EXTRA)
-
-noinst_PROGRAMS = \
+	$(NETPBM_EXTRA) \
 	set \
 	volume \
 	cleanup \
