diff -bB -r -u html2text-1.3.1/Area.C html2text-1.3.1-Patch_C/Area.C
--- html2text-1.3.1/Area.C	Tue Nov  2 18:57:49 1999
+++ html2text-1.3.1-Patch_C/Area.C	Wed Jul  2 21:58:32 2003
@@ -1,40 +1,32 @@
 
-/* ------------------------------------------------------------------------- */
+ /***************************************************************************/
 
 /*
- * Copyright (c) 1999
- *      GMRS Software GmbH, Innsbrucker Ring 159, 81669 Munich, Germany.
- *      http://www.gmrs.de
+ * Portions Copyright (c) 1999 GMRS Software GmbH
+ * Carl-von-Linde-Str. 38, D-85716 Unterschleissheim, http://www.gmrs.de
  *      All rights reserved.
- *      Author: Arno Unkrig (arno.unkrig@gmrs.de)
  *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions
- * are met:
- * 1. Redistributions of source code must retain the above copyright
- *    notice, this list of conditions and the following disclaimer.
- * 2. Redistributions in binary form must reproduce the above copyright
- *    notice, this list of conditions and the following disclaimer in the
- *    documentation and/or other materials provided with the distribution.
- * 3. All advertising materials mentioning features or use of this software
- *    must display the following acknowledgement:
- *      This product includes software developed by GMRS Software GmbH.
- * 4. The name of GMRS Software GmbH may not be used to endorse or promote
- *    products derived from this software without specific prior written
- *    permission.
+ * Author: Arno Unkrig <arno@unkrig.de>
+ */
+ 
+/* This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
  *
- * THIS SOFTWARE IS PROVIDED BY GMRS SOFTWARE GMBH ``AS IS'' AND ANY
- * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
- * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL GMRS SOFTWARE GMBH BE LIABLE
- * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
- * THE POSSIBILITY OF SUCH DAMAGE.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License in the file COPYING for more details.
  */
 
-/* ------------------------------------------------------------------------- */
+ /***************************************************************************/
+
+/*
+ * Changes to version 1.2.2 were made by Martin Bayer <mbayer@zedat.fu-berlin.de>
+ * Dates and reasons of modifications:
+ * Wed Jul  2 21:56:45 CEST 2003: ported to g++ 3.3
+ */
+
+ /***************************************************************************/
 
@@ -42,6 +34,6 @@ 
 #include <stdlib.h>
 #include <string.h>
-#include <iostream.h>
+#include <iostream>
 
 #include "Area.h"
 #include "string.h"
@@ -477,7 +468,7 @@
 	}
       }
     }
-    os << endl;
+    os << std::endl;
   }
 
   return os;
diff -bB -r -u html2text-1.3.1/Area.h html2text-1.3.1-Patch_C/Area.h
--- html2text-1.3.1/Area.h	Thu Oct  4 22:18:09 2001
+++ html2text-1.3.1-Patch_C/Area.h	Wed Jul  2 22:00:00 2003
@@ -7,13 +7,6 @@
  * All rights reserved.
  *
  * Author: Arno Unkrig <arno@unkrig.de>
- *
- * All advertising materials mentioning features or use of this software
- * must display the following acknowledgement:
- * "This product includes software developed by GMRS Software GmbH."
- * The name of GMRS Software GmbH may not be used to endorse or promote
- * products derived from this software without specific prior written
- * permission.
  */
  
 /* This program is free software; you can redistribute it and/or modify
@@ -33,6 +26,7 @@
  * Changes to version 1.2.2 were made by Martin Bayer <mbayer@zedat.fu-berlin.de>
  * Dates and reasons of modifications:
  * Thu Oct  4 21:21:10 CEST 2001: ported to g++ 3.0
+ * Wed Jul  2 21:59:41 CEST 2003: ported to g++ 3.3
  */
   
  /***************************************************************************/
@@ -46,7 +40,7 @@
 
 #include <sys/types.h>
 #include <string>
-#include <istream.h>
+#include <istream>
 
 #ifdef BOOL_DEFINITION
 BOOL_DEFINITION
diff -bB -r -u html2text-1.3.1/CREDITS html2text-1.3.1-Patch_C/CREDITS
--- html2text-1.3.1/CREDITS	Mon Sep  2 11:51:10 2002
+++ html2text-1.3.1-Patch_C/CREDITS	Wed Jul  2 22:13:37 2003
@@ -1,4 +1,4 @@
-## CREDITS - Thanks to...                        Mon Sep  2 11:50:47 CEST 2002 
+## CREDITS - Thanks to...                        Wed Jul  2 22:12:38 CEST 2003
 ## ===========================================================================
 #
 # Since september 2000, these people have contributed to the development of
@@ -13,6 +13,9 @@
 
 Arno Unkrig <arno@unkrig.de>
 	+ almost all bugfixes in 1.3.1. Thanks!
+
+Nicolas Boullis <nboullis@debian.org>
+	+ ported to g++-3.3 (this change is not backward-compatible)
 
 
 ## ---------------------------------------------------------------------------
diff -bB -r -u html2text-1.3.1/HTMLControl.C html2text-1.3.1-Patch_C/HTMLControl.C
--- html2text-1.3.1/HTMLControl.C	Sun Apr  7 16:30:27 2002
+++ html2text-1.3.1-Patch_C/HTMLControl.C	Wed Jul  2 22:00:43 2003
@@ -27,12 +27,13 @@
  * Dates and reasons of modifications:
  * Fre Jun  8 17:20:38 CEST 2001: fixed coredump
  * Sun Apr  7 11:31:10 CEST 2002: fixed parser errors
+ * Wed Jul  2 22:00:21 CEST 2003: ported to g++ 3.3
  */
   
  /***************************************************************************/
 
 
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
@@ -423,15 +424,15 @@
         if (c != '>') return SCAN_ERROR;
 
         if (debug_scanner) {
-          cerr << "Scanned tag \"<" << (is_end_tag ? "/" : "") << tag_name;
+          std::cerr << "Scanned tag \"<" << (is_end_tag ? "/" : "") << tag_name;
           if (!is_end_tag && tag_attributes.get()) {
 	    const list<TagAttribute>           &ta(*tag_attributes);
             list<TagAttribute>::const_iterator j;
             for (j = ta.begin(); j != ta.end(); ++j) {
-	      cerr << " " << (*j).first << "=\"" << (*j).second << "\"";
+	      std::cerr << " " << (*j).first << "=\"" << (*j).second << "\"";
             }
           }
-          cerr << ">\"" << endl;
+          std::cerr << ">\"" << std::endl;
         }
 
         /*
@@ -445,7 +446,7 @@
         );
         if (tag == NULL) { /* EXTENSION: Swallow unknown tags. */
           if (debug_scanner) {
-            cerr << "Tag unknown -- swallowed." << endl;
+            std::cerr << "Tag unknown -- swallowed." << std::endl;
           }
           continue;
         }
@@ -456,7 +457,7 @@
         if (is_end_tag) {
           if (!tag->end_tag_code) {
             if (debug_scanner) {
-              cerr << "Non-container end tag scanned." << endl;
+              std::cerr << "Non-container end tag scanned." << std::endl;
             }
             continue;
           }
@@ -506,7 +507,7 @@
        */
       if (s->empty()) { delete s; continue; }
 
-      if (debug_scanner) cerr << "Scanned PCDATA \"" << *s << "\"" << endl;
+      if (debug_scanner) std::cerr << "Scanned PCDATA \"" << *s << "\"" << std::endl;
 
       return PCDATA;
     }
diff -bB -r -u html2text-1.3.1/HTMLControl.h html2text-1.3.1-Patch_C/HTMLControl.h
--- html2text-1.3.1/HTMLControl.h	Thu Oct  4 22:06:54 2001
+++ html2text-1.3.1-Patch_C/HTMLControl.h	Wed Jul  2 22:01:29 2003
@@ -7,13 +7,6 @@
  * All rights reserved.
  *
  * Author: Arno Unkrig <arno@unkrig.de>
- *
- * All advertising materials mentioning features or use of this software
- * must display the following acknowledgement:
- * "This product includes software developed by GMRS Software GmbH."
- * The name of GMRS Software GmbH may not be used to endorse or promote
- * products derived from this software without specific prior written
- * permission.
  */
  
 /* This program is free software; you can redistribute it and/or modify
@@ -33,6 +26,7 @@
  * Changes to version 1.2.2 were made by Martin Bayer <mbayer@zedat.fu-berlin.de>
  * Dates and reasons of modifications:
  * Thu Oct  4 21:25:07 CEST 2001: ported to g++ 3.0
+ * Wed Jul  2 22:01:12 CEST 2003: ported to g++ 3.3
  */
   
  /***************************************************************************/
@@ -45,7 +39,7 @@
 
 #include "HTMLParser.h"
 #include "urlistream.h"
-#include <istream.h>
+#include <istream>
 
 using std::istream;
 
diff -bB -r -u html2text-1.3.1/Makefile.in html2text-1.3.1-Patch_C/Makefile.in
--- html2text-1.3.1/Makefile.in	Mon Sep  2 12:06:49 2002
+++ html2text-1.3.1-Patch_C/Makefile.in	Wed Jul  2 22:09:32 2003
@@ -23,7 +23,7 @@
 #
 # -----------------------------------------------------------------------------
 
-VERSION=1.3.1
+VERSION=1.3.1-Patch_C
 
 BISONXX  = bison++
 YFLAGS   =
diff -bB -r -u html2text-1.3.1/Properties.C html2text-1.3.1-Patch_C/Properties.C
--- html2text-1.3.1/Properties.C	Fri Jun  8 17:51:09 2001
+++ html2text-1.3.1-Patch_C/Properties.C	Wed Jul  2 22:03:09 2003
@@ -7,52 +7,33 @@
  * All rights reserved.
  *
  * Author: Arno Unkrig <arno@unkrig.de>
+ */
+ 
+/* This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
  *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions
- * are met:
- * 1. Redistributions of source code must retain the above copyright
- *    notice, this list of conditions and the following disclaimer.
- * 2. Redistributions in binary form must reproduce the above copyright
- *    notice, this list of conditions and the following disclaimer in the
- *    documentation and/or other materials provided with the distribution.
- * 3. All advertising materials mentioning features or use of this software
- *    must display the following acknowledgement:
- *    This product includes software developed by GMRS Software GmbH.
- * 4. The name of GMRS Software GmbH may not be used to endorse or promote
- *    products derived from this software without specific prior written
- *    permission.
- *
- * THIS SOFTWARE IS PROVIDED BY GMRS SOFTWARE GMBH ``AS IS'' AND ANY
- * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
- * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL GMRS SOFTWARE GMBH BE LIABLE
- * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
- * THE POSSIBILITY OF SUCH DAMAGE.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License in the file COPYING for more details.
  */
 
  /***************************************************************************/
  
 /*
  * Changes to version 1.2.2 were made by Martin Bayer <mbayer@zedat.fu-berlin.de>
- * Date and reason of last modification: Fre Jun  8 17:24:35 CEST 2001 - new method
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the file
- * COPYING for more details.
+ * Dates and reasons of modifications:
+ * Fre Jun  8 17:24:35 CEST 2001: new method
+ * Wed Jul  2 22:02:51 CEST 2003: ported to g++ 3.3
  */
 
  /***************************************************************************/
 
 
 #include <ctype.h>
-#include <iostream.h>
+#include <iostream>
 
 #include "Properties.h"
 
diff -bB -r -u html2text-1.3.1/Properties.h html2text-1.3.1-Patch_C/Properties.h
--- html2text-1.3.1/Properties.h	Thu Oct  4 22:39:10 2001
+++ html2text-1.3.1-Patch_C/Properties.h	Wed Jul  2 22:03:45 2003
@@ -7,13 +7,6 @@
  * All rights reserved.
  *
  * Author: Arno Unkrig <arno@unkrig.de>
- *
- * All advertising materials mentioning features or use of this software
- * must display the following acknowledgement:
- * "This product includes software developed by GMRS Software GmbH."
- * The name of GMRS Software GmbH may not be used to endorse or promote
- * products derived from this software without specific prior written
- * permission.
  */
  
 /* This program is free software; you can redistribute it and/or modify
@@ -34,6 +27,7 @@
  * Dates and reasons of modifications:
  * Fre Jun  8 17:29:50 CEST 2001
  * Thu Oct  4 22:38:59 CEST 2001: ported to g++ 3.0
+ * Wed Jul  2 22:03:29 CEST 2003: ported to g++ 3.3
  */
   
  /***************************************************************************/
@@ -51,7 +45,7 @@
 
 #include <string>
 #include <map>
-#include <istream.h>
+#include <istream>
 
 using std::string;
 using std::map;
diff -bB -r -u html2text-1.3.1/format.C html2text-1.3.1-Patch_C/format.C
--- html2text-1.3.1/format.C	Mon Jul 29 13:12:13 2002
+++ html2text-1.3.1-Patch_C/format.C	Wed Jul  2 22:04:27 2003
@@ -29,12 +29,13 @@
  * Fre Jun  8 17:37:12 CEST 2001: new image handling
  * Thu Oct  4 21:34:26 CEST 2001: ported to g++ 3.0
  * Mon Jul 29 13:09:26 CEST 2002: fixed runtime increment
+ * Wed Jul  2 22:04:08 CEST 2003: ported to g++ 3.3
  */
   
  /***************************************************************************/
 
 
-#include <strstream.h>
+#include <sstream>
 
 #include <stdlib.h>
 #include <ctype.h>
@@ -374,10 +375,10 @@
   case CUSTOM3:   bullet = custom3_bullet; break;
   case ARABIC_NUMBERS:
     {
-      ostrstream oss;
-      oss << number << '.' << std::ends;
+      std::ostringstream oss;
+      oss << number << '.'; // << std::ends;
       bullet = oss.str();
-      oss.rdbuf()->freeze(0);
+      // oss.rdbuf()->freeze(0);
     }
     break;
   case LOWER_ALPHA:
diff -bB -r -u html2text-1.3.1/format.h html2text-1.3.1-Patch_C/format.h
--- html2text-1.3.1/format.h	Thu Oct  4 22:04:15 2001
+++ html2text-1.3.1-Patch_C/format.h	Wed Jul  2 22:06:16 2003
@@ -7,13 +7,6 @@
  * All rights reserved.
  *
  * Author: Arno Unkrig <arno@unkrig.de>
- *
- * All advertising materials mentioning features or use of this software
- * must display the following acknowledgement:
- * "This product includes software developed by GMRS Software GmbH."
- * The name of GMRS Software GmbH may not be used to endorse or promote
- * products derived from this software without specific prior written
- * permission.
  */
  
 /* This program is free software; you can redistribute it and/or modify
@@ -34,6 +27,7 @@
  * Dates and reasons of modifications:
  * Fre Jun  8 17:49:39 CEST 2001: new method
  * Thu Oct  4 21:36:51 CEST 2001: ported to g++ 3.0
+ * Wed Jul  2 22:05:32 CEST 2003: ported to g++ 3.3
  */
   
  /***************************************************************************/
@@ -46,7 +40,7 @@
 
 #include <string>
 #include <vector>
-#include <istream.h>
+#include <istream>
 
 using std::string;
 using std::vector;
diff -bB -r -u html2text-1.3.1/html.C html2text-1.3.1-Patch_C/html.C
--- html2text-1.3.1/html.C	Fri Jun  8 17:43:53 2001
+++ html2text-1.3.1-Patch_C/html.C	Wed Jul  2 22:07:34 2003
@@ -7,45 +7,26 @@
  * All rights reserved.
  *
  * Author: Arno Unkrig <arno@unkrig.de>
+ */
+ 
+/* This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
  *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions
- * are met:
- * 1. Redistributions of source code must retain the above copyright
- *    notice, this list of conditions and the following disclaimer.
- * 2. Redistributions in binary form must reproduce the above copyright
- *    notice, this list of conditions and the following disclaimer in the
- *    documentation and/or other materials provided with the distribution.
- * 3. All advertising materials mentioning features or use of this software
- *    must display the following acknowledgement:
- *    This product includes software developed by GMRS Software GmbH.
- * 4. The name of GMRS Software GmbH may not be used to endorse or promote
- *    products derived from this software without specific prior written
- *    permission.
- *
- * THIS SOFTWARE IS PROVIDED BY GMRS SOFTWARE GMBH ``AS IS'' AND ANY
- * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
- * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL GMRS SOFTWARE GMBH BE LIABLE
- * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
- * THE POSSIBILITY OF SUCH DAMAGE.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License in the file COPYING for more details.
  */
 
  /***************************************************************************/
  
 /*
  * Changes to version 1.2.2 were made by Martin Bayer <mbayer@zedat.fu-berlin.de>
- * Date and reason of last modification: Fre Jun  8 17:43:02 CEST 2001 - new method
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the file
- * COPYING for more details.
+ * Dates and reasons of modifications:
+ * Fre Jun  8 17:43:02 CEST 2001: new method
+ * Wed Jul  2 22:07:12 CEST 2003: ported to g++ 3.3
  */
 
  /***************************************************************************/
@@ -53,7 +34,7 @@
 
 #include <stdlib.h>
 #include <stdarg.h>
-#include <iostream.h>
+#include <iostream>
 
 #include "html.h"
 #include "HTMLParser.h"
@@ -145,13 +126,13 @@
     os << "</TITLE>" << separator;
   }
   if (isindex_attributes.get()) {
-    os << "<ISINDEX"<< isindex_attributes << ">" << endl;
+    os << "<ISINDEX"<< isindex_attributes << ">" << std::endl;
   }
-  if (base_attributes.get()) os << "<BASE" << base_attributes << ">" << endl;
+  if (base_attributes.get()) os << "<BASE" << base_attributes << ">" << std::endl;
   foreach(scripts, os, separator);
   foreach(styles, os, separator);
-  if (meta_attributes.get()) os << "<META" << meta_attributes << ">" << endl;
-  if (link_attributes.get()) os << "<LINK" << link_attributes << ">" << endl;
+  if (meta_attributes.get()) os << "<META" << meta_attributes << ">" << std::endl;
+  if (link_attributes.get()) os << "<LINK" << link_attributes << ">" << std::endl;
   os << "</HEAD>" << separator;
 }
 
@@ -306,7 +287,7 @@
 {
   os << "<OPTION" << attributes << ">" << separator;
   if (pcdata.get()) pcdata->unparse(os, separator);
-  os << "</OPTION>" << endl;
+  os << "</OPTION>" << std::endl;
 }
 
 void
@@ -314,7 +295,7 @@
 {
   os << "<SELECT" << attributes << ">" << separator;
   if (content.get()) foreach(*content, os, separator);
-  os << "</SELECT>" << endl;
+  os << "</SELECT>" << std::endl;
 }
 
 void
@@ -322,7 +303,7 @@
 {
   os << "<TEXTAREA" << attributes << ">" << separator;
   if (pcdata.get()) pcdata->unparse(os, separator);
-  os << "</TEXTAREA>" << endl;
+  os << "</TEXTAREA>" << std::endl;
 }
 
 /* ------------------------------------------------------------------------- */
diff -bB -r -u html2text-1.3.1/html2text.C html2text-1.3.1-Patch_C/html2text.C
--- html2text-1.3.1/html2text.C	Mon Jul 22 13:49:53 2002
+++ html2text-1.3.1-Patch_C/html2text.C	Wed Jul  2 22:09:04 2003
@@ -28,12 +28,13 @@
  * Fre Jun  8 19:00:26 CEST 2001: new image handling
  * Thu Oct  4 21:42:24 CEST 2001: ported to g++ 3.0, bugfix for '-' as synonym for STDIN
  * Mon Jul 22 13:48:26 CEST 2002: Made finaly reading from STDIN work.
+ * Wed Jul  2 22:08:45 CEST 2003: ported to g++ 3.3
  */
   
  /***************************************************************************/
 
 
-#include <iostream.h>
+#include <iostream>
 #include <string.h>
 #include <stdlib.h>
 
@@ -87,7 +88,7 @@
    */
   if (mode != SYNTAX_CHECK && !strcmp(p, "parse error")) return;
 
-  cerr
+  std::cerr
     << "File \""
     << file_name
     << "\", line "
@@ -96,7 +97,7 @@
     << current_column
     << ": "
     << p
-    << endl;
+    << std::endl;
 }
 
 /*virtual*/ void
@@ -109,14 +110,14 @@
     break;
 
   case UNPARSE:
-    document.unparse(os, endl);
+    document.unparse(os, std::endl);
     break;
 
   case SYNTAX_CHECK:
     break;
 
   default:
-    cerr << "??? Invalid mode " << mode << " ??? " << endl;
+    std::cerr << "??? Invalid mode " << mode << " ??? " << std::endl;
     exit(1);
     break;
   }
@@ -151,23 +152,23 @@
 main(int argc, char **argv)
 {
   if (argc == 2 && !strcmp(argv[1], "-help")) {
-    cout
-      << "This is html2text, version " stringify(VERSION) << endl
-      << endl
+    std::cout
+      << "This is html2text, version " stringify(VERSION) << std::endl
+      << std::endl
       << usage;
     exit(0);
   }
 
   if (argc == 2 && !strcmp(argv[1], "-version")) {
-    cout
-      << "This is html2text, version " stringify(VERSION) << endl
-      << endl
-      << "The latest version can be found at http://userpage.fu-berlin.de/~mbayer/tools/" << endl
-      << endl
-      << "This program is distributed in the hope that it will be useful, but WITHOUT" << endl
-      << "ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS" << endl
-      << "FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details." << endl
-      << endl;
+    std::cout
+      << "This is html2text, version " stringify(VERSION) << std::endl
+      << std::endl
+      << "The latest version can be found at http://userpage.fu-berlin.de/~mbayer/tools/" << std::endl
+      << std::endl
+      << "This program is distributed in the hope that it will be useful, but WITHOUT" << std::endl
+      << "ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS" << std::endl
+      << "FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details." << std::endl
+      << std::endl;
     exit(0);
   }
 
@@ -195,20 +196,20 @@
     if (!strcmp(arg, "-o"            )) { output_file_name = argv[++i];  } else
     if (!strcmp(arg, "-nobs"         )) { use_backspaces = false;        } else
     {
-      cerr
+      std::cerr
 	<< "Unrecognized command line option \""
 	<< arg
 	<< "\", try \"-help\"."
-	<< endl;
+	<< std::endl;
       exit(1);
     }
   }
   if (i > argc) {
-    cerr
+    std::cerr
       << "Error: Required parameter after \""
       << argv[argc - 1]
       << "\" missing."
-      << endl;
+      << std::endl;
     exit(1);
   }
 
@@ -299,16 +300,16 @@
       Formatting::setProperty(p->key, p->value);
     }
   } else {
-    cerr
+    std::cerr
       << "Unknown style \""
       << style
       << "\" specified -- try \"-help\"."
-      << endl;
+      << std::endl;
     ::exit(1);
   }
 
   {
-    ifstream ifs(rcfile.c_str());
+    std::ifstream ifs(rcfile.c_str());
     if (!ifs.rdbuf()->is_open()) ifs.open("/etc/html2textrc");
     if (ifs.rdbuf()->is_open()) {
       Formatting::loadProperties(ifs);
@@ -321,18 +322,18 @@
   Area::use_backspaces = use_backspaces;
 
   ostream  *osp;
-  ofstream ofs;
+  std::ofstream ofs;
 
   if (!strcmp(output_file_name, "-")) {
-    osp = &cout;
+    osp = &std::cout;
   } else {
-    ofs.open(output_file_name, ios::out);
+    ofs.open(output_file_name, std::ios::out);
     if (!ofs) {
-      cerr
+      std::cerr
         << "Could not open output file \""
         << output_file_name
         << "\"."
-        << endl;
+        << std::endl;
     exit(1);
     }
     osp = &ofs;
@@ -342,7 +343,7 @@
     const char *input_url = input_urls[i];
 
     if (number_of_input_urls != 1) {
-      *osp << "###### " << input_url << " ######" << endl;
+      *osp << "###### " << input_url << " ######" << std::endl;
     }
 
     istream    *isp;
@@ -350,12 +351,12 @@
 
     uis.open(input_url);
     if (!uis.is_open()) {
-      cerr
+      std::cerr
         << "Opening input URL \""
 	<< input_url
         << "\": "
         << uis.open_error()
-        << endl;
+        << std::endl;
       exit(1);
     }
 
diff -bB -r -u html2text-1.3.1/urlistream.h html2text-1.3.1-Patch_C/urlistream.h
--- html2text-1.3.1/urlistream.h	Sun Apr  7 12:01:21 2002
+++ html2text-1.3.1-Patch_C/urlistream.h	Wed Jul  2 22:08:32 2003
@@ -27,6 +27,7 @@
  * Dates and reasons of modifications:
  * Thu Oct  4 21:50:51 CEST 2001: ported to g++ 3.0
  * Sun Apr  7 12:00:50 CEST 2002: Handle URLs with missing node
+ * Wed Jul  2 22:08:05 CEST 2003: ported to g++ 3.3
  */
   
  /***************************************************************************/
@@ -37,7 +38,7 @@
 
 /* ------------------------------------------------------------------------- */
 
-#include <fstream.h>
+#include <fstream>
 #include <string>
 #include <unistd.h>
 
