diff -urN Linux-PAM-0.75/modules/pam_console/config.y Linux-PAM-0.75.bison/modules/pam_console/config.y
--- Linux-PAM-0.75/modules/pam_console/config.y	2001-10-02 22:29:10.000000000 +0200
+++ Linux-PAM-0.75.bison/modules/pam_console/config.y	2002-11-21 23:40:36.000000000 +0200
@@ -117,11 +117,11 @@
 	|	stringlist string {$$ = g_slist_append($1, $2);}
 	;
 
-optstring:	string	{$$=$1}
-	|	/* empty */ {$$=NULL}
+optstring:	string	{$$=$1;}
+	|	/* empty */ {$$=NULL;}
 	;
 
-string:		STRING {$$=$1}
+string:		STRING {$$=$1;} ;
 
 %%
 
