diff -ru menuorg-0.4.3-orig/src/MenuConfigurationRepository.cpp menuorg-0.4.3/src/MenuConfigurationRepository.cpp
--- menuorg-0.4.3-orig/src/MenuConfigurationRepository.cpp	2008-06-20 15:42:26.363703821 +0200
+++ menuorg-0.4.3/src/MenuConfigurationRepository.cpp	2008-06-20 15:42:31.180893833 +0200
@@ -24,6 +24,7 @@
 #include "MenuConfigurationRepository.h"
 #include <exception>
 #include <iostream>
+#include <algorithm>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <libxml++/libxml++.h>
