--- src/nautilus-main.c.old	2003-10-05 18:47:28.336065464 +0800
+++ src/nautilus-main.c	2003-10-05 18:48:37.015624584 +0800
@@ -252,6 +252,8 @@
 	
 	bonobo_activate (); /* do now since we need it before main loop */
 
+	application = NULL;
+
 	/* Do either the self-check or the real work. */
 	if (perform_self_check) {
 #ifndef NAUTILUS_OMIT_SELF_CHECK
@@ -279,12 +281,17 @@
 		if (is_event_loop_needed ()) {
 			gtk_main ();
 		}
-		bonobo_object_unref (application);
+
 	}
 
 	poptFreeContext (popt_context);
 
 	gnome_vfs_shutdown ();
+
+	if( application != NULL ) {
+		bonobo_object_unref( application );
+	}
+	
 	eel_debug_shut_down ();
 	bonobo_ui_debug_shutdown ();
 
