diff -ur nautilus-2.2.0.2/src/file-manager/fm-directory-view.c nautilus-2.2.0.2.patched/src/file-manager/fm-directory-view.c
--- nautilus-2.2.0.2/src/file-manager/fm-directory-view.c	2003-01-20 18:29:21.000000000 +0100
+++ nautilus-2.2.0.2.patched/src/file-manager/fm-directory-view.c	2003-02-19 19:08:36.000000000 +0100
@@ -3727,6 +3727,29 @@
 
 }
 
+static gboolean
+no_locale_at_end (const char *str)
+{
+	int len;
+	len = strlen (str);
+	if (len > 3 &&
+		str[len-3] == '-' &&
+		g_ascii_isalpha (str[len-2]) &&
+		g_ascii_isalpha (str[len-1])) {
+			return FALSE;
+		}
+	if (len > 6 &&
+		str[len-6] == '-' &&
+		g_ascii_isalpha (str[len-5]) &&
+		g_ascii_isalpha (str[len-4]) &&
+		str[len-3] == '_' &&
+		g_ascii_isalpha (str[len-2]) &&
+		g_ascii_isalpha (str[len-1])) {
+			return FALSE;
+		}
+	return TRUE;
+}
+
 static GList *
 get_bonobo_menu_verb_names (Bonobo_ServerInfo *info)
 {
@@ -3742,7 +3765,8 @@
 		/* look for properties that start with "nautilusverb:".  The
 		 * part following the colon is the verb name
 		 */
-		if (strstr (info->props._buffer[i].name, "nautilusverb:")) {
+		if (strstr (info->props._buffer[i].name, "nautilusverb:") &&
+			no_locale_at_end (info->props._buffer[i].name)) {           
 			l = g_list_prepend (l,
 			      g_strdup (&info->props._buffer[i].name[offset]));	
 		}
