--- nautilus-1.0.6/components/mozilla/nautilus-mozilla-embed-extensions.cpp.original	Sat Jun  1 15:16:41 2002
+++ nautilus-1.0.6/components/mozilla/nautilus-mozilla-embed-extensions.cpp	Sat Jun  1 15:18:39 2002
@@ -51,8 +51,13 @@
  * by the "string" directory in the mozilla include directory. So we do the
  * following as a workaround.
  */
+#if __GNUC__ == 3 && __GNUC_MINOR__ == 1
+#include <string>
+using namespace std;
+#else
 #include <std/bastring.h>
 typedef basic_string<char> string;
+#endif
 
 struct Encoding
 {
