--- ./kxsldbg/kxsldbgpart/libxsldbg/file_cmds.cpp.orig	2005-12-04 11:02:02.000000000 +0100
+++ ./kxsldbg/kxsldbgpart/libxsldbg/file_cmds.cpp	2005-12-04 11:04:00.000000000 +0100
@@ -175,7 +175,7 @@
         } else {
 	    xsldbgGenericErrorFunc(i18n("PublicID \"%1\" was not found in current catalog.\n").arg(xsldbgText(arg)));
         }
-        xsltGenericError(xsltGenericErrorContext, buffer);
+        xsltGenericError(xsltGenericErrorContext, "%s", buffer);
     }
     return result;
 }
