diff -ur xplanet-1.2.0.orig/src/buildPlanetMap.cpp xplanet-1.2.0/src/buildPlanetMap.cpp
--- xplanet-1.2.0.orig/src/buildPlanetMap.cpp	2004-04-20 23:47:21.000000000 +0200
+++ xplanet-1.2.0/src/buildPlanetMap.cpp	2008-03-20 15:48:24.000000000 +0100
@@ -1,5 +1,6 @@
 #include <cmath>
 #include <cstdio>
+#include <cstdlib>
 #include <map>
 using namespace std;
 
diff -ur xplanet-1.2.0.orig/src/libannotate/addArcs.cpp xplanet-1.2.0/src/libannotate/addArcs.cpp
--- xplanet-1.2.0.orig/src/libannotate/addArcs.cpp	2005-03-15 19:27:13.000000000 +0100
+++ xplanet-1.2.0/src/libannotate/addArcs.cpp	2008-03-20 15:44:58.000000000 +0100
@@ -3,7 +3,7 @@
 #include <fstream>
 #include <sstream>
 #include <map>
-#include <string>
+#include <cstring>
 #include <vector>
 using namespace std;
 
diff -ur xplanet-1.2.0.orig/src/libannotate/addMarkers.cpp xplanet-1.2.0/src/libannotate/addMarkers.cpp
--- xplanet-1.2.0.orig/src/libannotate/addMarkers.cpp	2004-08-29 22:41:58.000000000 +0200
+++ xplanet-1.2.0/src/libannotate/addMarkers.cpp	2008-03-20 15:45:11.000000000 +0100
@@ -5,7 +5,7 @@
 #include <fstream>
 #include <map>
 #include <sstream>
-#include <string>
+#include <cstring>
 #include <vector>
 using namespace std;
 
diff -ur xplanet-1.2.0.orig/src/libannotate/addSatellites.cpp xplanet-1.2.0/src/libannotate/addSatellites.cpp
--- xplanet-1.2.0.orig/src/libannotate/addSatellites.cpp	2004-07-23 02:46:20.000000000 +0200
+++ xplanet-1.2.0/src/libannotate/addSatellites.cpp	2008-03-20 15:45:23.000000000 +0100
@@ -4,7 +4,7 @@
 #include <fstream>
 #include <map>
 #include <sstream>
-#include <string>
+#include <cstring>
 #include <vector>
 using namespace std;
 
diff -ur xplanet-1.2.0.orig/src/libannotate/Icon.cpp xplanet-1.2.0/src/libannotate/Icon.cpp
--- xplanet-1.2.0.orig/src/libannotate/Icon.cpp	2004-02-16 19:36:42.000000000 +0100
+++ xplanet-1.2.0/src/libannotate/Icon.cpp	2008-03-20 15:45:58.000000000 +0100
@@ -1,4 +1,5 @@
 #include <sstream>
+#include <cstring>
 using namespace std;
 
 #include "findFile.h"
diff -ur xplanet-1.2.0.orig/src/libplanet/Planet.cpp xplanet-1.2.0/src/libplanet/Planet.cpp
--- xplanet-1.2.0.orig/src/libplanet/Planet.cpp	2005-06-14 20:04:05.000000000 +0200
+++ xplanet-1.2.0/src/libplanet/Planet.cpp	2008-03-20 15:46:53.000000000 +0100
@@ -2,7 +2,7 @@
 #include <cstdio>
 #include <cctype>
 #include <sstream>
-#include <string>
+#include <cstring>
 using namespace std;
 
 #include "findFile.h"
diff -ur xplanet-1.2.0.orig/src/libprojection/getProjection.cpp xplanet-1.2.0/src/libprojection/getProjection.cpp
--- xplanet-1.2.0.orig/src/libprojection/getProjection.cpp	2005-06-13 17:13:11.000000000 +0200
+++ xplanet-1.2.0/src/libprojection/getProjection.cpp	2008-03-20 15:47:34.000000000 +0100
@@ -1,5 +1,6 @@
 #include <cctype>
 #include <cstring>
+#include <cstdlib>
 #include <sstream>
 #include <vector>
 using namespace std;
diff -ur xplanet-1.2.0.orig/src/Map.cpp xplanet-1.2.0/src/Map.cpp
--- xplanet-1.2.0.orig/src/Map.cpp	2005-06-15 01:40:46.000000000 +0200
+++ xplanet-1.2.0/src/Map.cpp	2008-03-20 15:48:01.000000000 +0100
@@ -1,5 +1,6 @@
 #include <map>
 #include <sstream>
+#include <cstdlib>
 #include <string>
 using namespace std;
 
diff -ur xplanet-1.2.0.orig/src/parseColor.cpp xplanet-1.2.0/src/parseColor.cpp
--- xplanet-1.2.0.orig/src/parseColor.cpp	2004-11-23 18:49:48.000000000 +0100
+++ xplanet-1.2.0/src/parseColor.cpp	2008-03-20 15:49:23.000000000 +0100
@@ -2,7 +2,8 @@
 #include <fstream>
 #include <map>
 #include <sstream>
-#include <string>
+#include <cstring>
+#include <cstdlib>
 using namespace std;
 
 #include "findFile.h"
diff -ur xplanet-1.2.0.orig/src/PlanetProperties.h xplanet-1.2.0/src/PlanetProperties.h
--- xplanet-1.2.0.orig/src/PlanetProperties.h	2005-04-02 02:43:06.000000000 +0200
+++ xplanet-1.2.0/src/PlanetProperties.h	2008-03-20 15:46:16.000000000 +0100
@@ -1,7 +1,7 @@
 #ifndef PLANETPROPERTIES_H
 #define PLANETPROPERTIES_H
 
-#include <string>
+#include <cstring>
 #include <vector>
 
 #include "body.h"
diff -ur xplanet-1.2.0.orig/src/xplanet.cpp xplanet-1.2.0/src/xplanet.cpp
--- xplanet-1.2.0.orig/src/xplanet.cpp	2004-09-09 17:46:05.000000000 +0200
+++ xplanet-1.2.0/src/xplanet.cpp	2008-03-20 15:50:21.000000000 +0100
@@ -1,5 +1,6 @@
 #include <clocale>
 #include <cstdio>
+#include <cstdlib>
 #include <iostream>
 #include <map>
 #include <sstream>
diff -ur xplanet-1.2.0.orig/src/xpUtil.cpp xplanet-1.2.0/src/xpUtil.cpp
--- xplanet-1.2.0.orig/src/xpUtil.cpp	2005-04-02 02:22:51.000000000 +0200
+++ xplanet-1.2.0/src/xpUtil.cpp	2008-03-20 15:50:06.000000000 +0100
@@ -5,7 +5,8 @@
 #include <ctime>
 #include <iostream>
 #include <sstream>
-#include <string>
+#include <cstring>
+#include <cstdlib>
 using namespace std;
 
 #include <unistd.h>
