--- src/wrapper.c.orig	2003-10-18 22:20:38.154897496 -0400
+++ src/wrapper.c	2003-10-18 22:21:11.628808688 -0400
@@ -84,28 +84,28 @@
 char * GetBitmapDirectory(void)
 {
 #if defined( __WIN_32)  || defined(__LINUX)
-	return("./bitmaps/");
+	return(GENTOO_DATADIR "/bitmaps/");
 #endif
 }
 
 char * GetIconDirectory(void)
 {
 #if defined( __WIN_32)  || defined(__LINUX)
-    return("./icons/");
+    return(GENTOO_DATADIR "/icons/");
 #endif
 }
 
 char * GetSfxDirectory(void)
 {
 #if defined( __WIN_32)  || defined(__LINUX)
-	return("./sfx/");
+	return(GENTOO_DATADIR "/sfx/");
 #endif
 }
 
 char * GetCacheDirectory(void)
 {
 #if defined( __WIN_32)  || defined(__LINUX)
-	return("./cache/");
+	return(GENTOO_DATADIR "/cache/");
 #endif
 }
 
@@ -113,7 +113,7 @@
 char * GetMediaDirectory(void)
 {
 #if defined( __WIN_32)  || defined(__LINUX)	
-    return("./media/");
+    return(GENTOO_DATADIR "/media/");
 #endif
 }
 
--- src/menu.c.orig	2003-10-18 22:55:47.836177192 -0400
+++ src/menu.c	2003-10-18 22:56:29.063909624 -0400
@@ -884,7 +885,7 @@
     FILE *stream;
     char line[255], name[255], d1[255], d2[255], d3[255], d4[255], icon[128];
     
-    if( (stream = fopen("./spells.dat", "r" )) != NULL )
+    if( (stream = fopen(GENTOO_DATADIR "/spells.dat", "r" )) != NULL )
     {
         for(i=0;;i++)
         {
@@ -949,7 +950,7 @@
     FILE *stream;
     char line[255], name[255], d1[255], d2[255], d3[255], d4[255], icon[128];
     
-    if( (stream = fopen("./skills.dat", "r" )) != NULL )
+    if( (stream = fopen(GENTOO_DATADIR "/skills.dat", "r" )) != NULL )
     {
         for(i=0;;i++)
         {
--- src/include/config.h.orig	2003-10-18 22:49:31.166439696 -0400
+++ src/include/config.h	2003-10-18 22:55:37.103808760 -0400
@@ -29,11 +29,13 @@
 #define VERSION_CS 991013
 #define VERSION_SC 991013
 
-#define KEYBIND_FILE "./keys.dat"
-#define OPTION_FILE  "./options.dat"
-#define ARCHDEF_FILE "./archdef.dat"
+#include "userfile.h"
 
-#define LOG_FILE     "./client.log"
+#define KEYBIND_FILE user_file("keys.dat")
+#define OPTION_FILE  user_file("options.dat")
+#define ARCHDEF_FILE GENTOO_DATADIR "/archdef.dat"
+
+#define LOG_FILE     user_file("client.log")
 
 #define CLIENT_ICON_NAME "icon.png"
  
--- src/include/userfile.h	1969-12-31 19:00:00.000000000 -0500
+++ src/include/userfile.h	2003-10-18 22:55:26.081484408 -0400
@@ -0,0 +1,17 @@
+#include <stdio.h> /* snprintf */
+#include <stdlib.h> /* getenv */
+#include <sys/stat.h> /* mkdir */
+#include <sys/types.h> /* mkdir */
+#include <unistd.h> /* lstat */
+static char *user_file(char *filename)
+{
+	#define MAX_PATH 1024
+	static char pathname[MAX_PATH];
+	static struct stat mystat;
+	snprintf(pathname, MAX_PATH, "%s/.daimonin", getenv("HOME"));
+	mkdir(pathname, 0700);
+	snprintf(pathname, MAX_PATH, "%s/.daimonin/%s", getenv("HOME"), filename);
+	if (lstat(pathname, &mystat))
+		sprintf(pathname, "%s/%s", GENTOO_DATADIR, filename);
+	return pathname;
+}
