--- Makefile.am.old	2010-03-26 16:00:53.000000000 +0100
+++ Makefile.am	2010-03-26 16:01:39.000000000 +0100
@@ -2,21 +2,16 @@
 daimonin_PROGRAMS = daimonin daimonin-updater
 
 daimonin_SOURCES =  \
-	../../src/adler32.c \
 	../../src/book.c \
 	../../src/buddy.c \
 	../../src/chatfilter.c \
 	../../src/client.c \
 	../../src/commands.c \
-	../../src/crc32.c \
 	../../src/dialog.c \
 	../../src/event.c \
 	../../src/filewrap.c \
 	../../src/group.c \
 	../../src/ignore.c \
-	../../src/inffast.c \
-	../../src/inflate.c \
-	../../src/inftrees.c \
 	../../src/interface.c \
 	../../src/inventory.c \
 	../../src/item.c \
@@ -31,11 +26,9 @@
 	../../src/sound.c \
 	../../src/sprite.c \
 	../../src/textwin.c \
-	../../src/uncompr.c \
 	../../src/wrapper.c \
-	../../src/widget.c \
-	../../src/zutil.c
-daimonin_LDADD = -lSDL_mixer -lSDL_image @SDL_LIBS@ @EXTRA_LIBS@ @PHYSFS_LIBS@
+	../../src/widget.c
+daimonin_LDADD = -lSDL_mixer -lSDL_image @SDL_LIBS@ @EXTRA_LIBS@ @PHYSFS_LIBS@ -lz
 daimonin_OBJDIR = objs
 
 daimonin_updater_SOURCES = \
