--- Makefile	Thu Aug  3 14:31:22 2000
+++ Makefile.new	Thu Aug  3 14:55:17 2000
@@ -4,16 +4,17 @@
 	tex jadetex.ins
 
 jadetex.fmt: basic
-	tex -ini "&hugelatex" -progname=jadetex jadetex.ini
+	tex -ini "&latex" -progname=jadetex jadetex.ini
 
 pdfjadetex.fmt: basic
 	pdftex -ini "&pdflatex" -progname=pdfjadetex pdfjadetex.ini
 
 install: jadetex.fmt pdfjadetex.fmt
-	(TT=`kpsewhich -expand-var '$$TEXMFMAIN'`;cp pdfjadetex.fmt $$TT/web2c)
-	(TT=`kpsewhich -expand-var '$$TEXMFMAIN'`;cp jadetex.fmt $$TT/web2c)
-	-(TT=`kpsewhich -expand-var '$$TEXMFMAIN'`;\
-	mkdir -p $$TT/tex/jadetex ; \
+	(TT=$$DESTDIR/`kpsewhich -expand-var '$$TEXMFMAIN'`; \
+	mkdir -p $$TT/web2c; \
+	cp pdfjadetex.fmt $$TT/web2c; \
+	cp jadetex.fmt $$TT/web2c; \
+	mkdir -p $$TT/tex/jadetex; \
 	cp dsssl.def jadetex.ltx $$TT/tex/jadetex)
 
 realclean: clean
