diff -ur a/libapol/apol_tcl_fc.c b/libapol/apol_tcl_fc.c
--- a/libapol/apol_tcl_fc.c	2006-04-11 17:27:18.000000000 -0300
+++ b/libapol/apol_tcl_fc.c	2006-05-12 17:11:07.000000000 -0300
@@ -178,16 +178,16 @@
  */
 static int Apol_Search_FC_Index_DB(ClientData clientData, Tcl_Interp *interp, int argc, CONST char *argv[])
 {	
+#ifndef LIBSEFS
+	Tcl_SetResult(interp, "You need to build apol with libsefs to use this feature!", TCL_STATIC);
+	return TCL_ERROR;
+#else		
 	sefs_search_keys_t search_keys;
 	CONST char **object_classes = NULL, **types = NULL, **users = NULL,
 	    **ranges, **paths = NULL;
 	int retval = TCL_ERROR;
 	Tcl_Obj *result_obj = Tcl_NewListObj(0, NULL);
 
-#ifndef LIBSEFS
-	Tcl_SetResult(interp, "You need to build apol with libsefs to use this feature!", TCL_STATIC);
-	return TCL_ERROR;
-#else		
 	memset(&search_keys, 0, sizeof(search_keys));
 	if (argc != 10) {
 		Tcl_SetResult(interp, "Need a list of users, list of types, list of object classes, list of ranges, list of paths, user_regex, type_regex, range_regex, and path_regex", TCL_STATIC);
