Index: seaudit/Makefile
===================================================================
--- seaudit/Makefile	(revision 3303)
+++ seaudit/Makefile	(revision 3304)
@@ -7,8 +7,6 @@
 
 CFLAGS	          += -DDEFAULT_LOG='"$(DEFAULT_LOG_FILE)"' 
 CFLAGS            += -DINSTALL_LIBDIR='"$(INSTALL_LIBDIR)"'
-CFLAGS            += -DSEAUDIT_GUI_VERSION_STRING='"$(shell cat SEAUDIT_GUI_VERSION)"'
-CFLAGS            += -DSEREPORT_VERSION_NUM='"$(shell cat SEAUDIT_REPORT_VERSION)"'
 
 SEAUDIT_REPORT_BACKEND  = report.o
 
@@ -57,8 +55,14 @@
 	$(MAKE) -C ../libapol libapolso
 
 %.o:  %.c 
-	$(CC) $(CFLAGS) $(GTK_CFLAGS) $(LIBXML_FLAGS) $(INCLUDE) -c $<
+	$(CC) $(CFLAGS) $(LIBXML_FLAGS) $(INCLUDE) -c $<
 
+$(OBJS): CFLAGS += $(GTK_CFLAGS)
+
+seaudit.o: CFLAGS += -DSEAUDIT_GUI_VERSION_STRING='"$(shell cat SEAUDIT_GUI_VERSION)"'
+
+seaudit-report.o: CFLAGS += -DSEREPORT_VERSION_NUM='"$(shell cat SEAUDIT_REPORT_VERSION)"'
+
 install: seaudit seaudit-report
 	install -m 755 seaudit $(BINDIR)
 	install -m 755 seaudit-report $(BINDIR)
Index: sediff/sediff.c
===================================================================
--- sediff/sediff.c	(revision 3303)
+++ sediff/sediff.c	(revision 3304)
@@ -10,7 +10,6 @@
  
 /* libapol */
 #include "util.h"
-#include "sediff_rename_types.h"
 #include <policy.h>
 #include <policy-io.h>
 #include <policy-query.h>
Index: sediff/Makefile
===================================================================
--- sediff/Makefile	(revision 3303)
+++ sediff/Makefile	(revision 3304)
@@ -7,11 +7,6 @@
 GTK_LIBS        = `pkg-config --libs libglade-2.0`
 GLADE_FILES     = sediff.glade
 
-CFLAGS  	+= -DSEDIFF_VERSION_NUM='"$(shell cat SEDIFF_VERSION)"'
-CFLAGS  	+= -DSEDIFF_GUI_VERSION_NUM='"$(shell cat SEDIFF_GUI_VERSION)"'
-CFLAGS  	+= -DSEDIFF_GUI_VERSION_NUM='"$(shell cat SEDIFF_GUI_VERSION)"'
-
-
 all: libapol sediff sediffx
 
 sediff: sediff.o $(LIBAPOL) 
@@ -29,8 +24,13 @@
 	install -m 644 sediff_help.txt $(INSTALL_LIBDIR)
 
 %.o:  %.c
-	$(CC) $(CFLAGS) $(GTK_CFLAGS) $(INCLUDE) -c $<
+	$(CC) $(CFLAGS) $(INCLUDE) -c $<
 
+$(GUI-OBJS): CFLAGS += $(GTK_CFLAGS)
+sediff.o: CFLAGS += -DSEDIFF_VERSION_NUM='"$(shell cat SEDIFF_VERSION)"'
+sediff_gui.o: CFLAGS += -DSEDIFF_VERSION_NUM='"$(shell cat SEDIFF_VERSION)"'
+sediff_gui.o: CFLAGS += -DSEDIFF_GUI_VERSION_NUM='"$(shell cat SEDIFF_GUI_VERSION)"'
+
 libapol:
 	$(MAKE) -C .. libapol
 
Index: secmds/Makefile
===================================================================
--- secmds/Makefile	(revision 3303)
+++ secmds/Makefile	(revision 3304)
@@ -22,13 +22,6 @@
 SE_CMDS += $(SELINUX_TOOLS)
 endif
 
-CFLAGS  	+= -DSEINFO_VERSION_NUM='"$(shell cat SEINFO_VERSION)"'
-CFLAGS  	+= -DSESEARCH_VERSION_NUM='"$(shell cat SESEARCH_VERSION)"'
-CFLAGS          += -DREPLCON_VERSION_NUM='"$(shell cat REPLCON_VERSION)"'
-CFLAGS          += -DFINDCON_VERSION_NUM='"$(shell cat FINDCON_VERSION)"'
-CFLAGS          += -DINDEXCON_VERSION_NUM='"$(shell cat INDEXCON_VERSION)"'
-CFLAGS          += -DSEARCHCON_VERSION_NUM='"$(shell cat SEARCHCON_VERSION)"'
-CFLAGS          += -DAPOL_DEFAULT_PERM_MAP="\"/usr/share/setools/apol_perm_mapping\""
 
 SHELL = /bin/sh
 
@@ -79,6 +72,14 @@
 %.o:  %.c
 	$(CC) $(CFLAGS) $(INCLUDE) -c $<
 
+seinfo.o: CFLAGS += -DSEINFO_VERSION_NUM='"$(shell cat SEINFO_VERSION)"'
+sesearch.o: CFLAGS += -DSESEARCH_VERSION_NUM='"$(shell cat SESEARCH_VERSION)"'
+replcon.o: CFLAGS += -DREPLCON_VERSION_NUM='"$(shell cat REPLCON_VERSION)"'
+findcon.o: CFLAGS += -DFINDCON_VERSION_NUM='"$(shell cat FINDCON_VERSION)"'
+indexcon.o: CFLAGS += -DINDEXCON_VERSION_NUM='"$(shell cat INDEXCON_VERSION)"'
+searchcon.o: CFLAGS += -DSEARCHCON_VERSION_NUM='"$(shell cat SEARCHCON_VERSION)"'
+flowtool.o: CFLAGS += -DAPOL_DEFAULT_PERM_MAP="\"/usr/share/setools/apol_perm_mapping\""
+
 libapol:
 	$(MAKE) -C .. libapol
 
