--- ./cpp/src/IceSSL/Instance.cpp.orig	2010-04-03 16:00:51.000000000 -0400
+++ ./cpp/src/IceSSL/Instance.cpp	2010-04-03 16:03:45.000000000 -0400
@@ -1058,7 +1058,7 @@
 {
     Trace out(_logger, _securityTraceCategory);
     out << "SSL summary for " << (incoming ? "incoming" : "outgoing") << " connection\n";
-    SSL_CIPHER* cipher = SSL_get_current_cipher(ssl);
+    const SSL_CIPHER* cipher = SSL_get_current_cipher(ssl);
     if(!cipher)
     {
         out << "unknown cipher\n";
