diff -U 4 -H -b -w -B -E -d -i -r digikam-0.9.0-beta1-orig/digikam/digikam/digikamapp.cpp digikam-0.9.0-beta1/digikam/digikam/digikamapp.cpp
--- digikam-0.9.0-beta1-orig/digikam/digikam/digikamapp.cpp	2006-09-02 13:30:02.000000000 +0200
+++ digikam-0.9.0-beta1/digikam/digikam/digikamapp.cpp	2006-09-02 13:32:15.000000000 +0200
@@ -678,24 +678,24 @@
                                    "gamma_adjustment");
 
     // -- Rating actions ---------------------------------------------------------------
 
-    m_0Star = new KAction(i18n("No Star"), CTRL+Key_0,
+    m_0Star = new KAction(i18n("Assign Rating \"No Star\""), CTRL+Key_0,
                           mView, SLOT(slotAssignRatingNoStar()),
                           actionCollection(), "imageview_ratenostar");
-    m_1Star = new KAction(i18n("One Star"), CTRL+Key_1,
+    m_1Star = new KAction(i18n("Assign Rating \"One Star\""), CTRL+Key_1,
                           mView, SLOT(slotAssignRatingOneStar()),
                           actionCollection(), "imageview_rateonestar");
-    m_2Star = new KAction(i18n("Two Star"), CTRL+Key_2, 
+    m_2Star = new KAction(i18n("Assign Rating \"Two Stars\""), CTRL+Key_2, 
                           mView, SLOT(slotAssignRatingTwoStar()),
                           actionCollection(), "imageview_ratetwostar");
-    m_3Star = new KAction(i18n("Three Star"), CTRL+Key_3, 
+    m_3Star = new KAction(i18n("Assign Rating \"Three Stars\""), CTRL+Key_3, 
                           mView, SLOT(slotAssignRatingThreeStar()),
                           actionCollection(), "imageview_ratethreestar");
-    m_4Star = new KAction(i18n("Four Star"), CTRL+Key_4, 
+    m_4Star = new KAction(i18n("Assign Rating \"Four Stars\""), CTRL+Key_4, 
                           mView, SLOT(slotAssignRatingFourStar()),
                           actionCollection(), "imageview_ratefourstar");
-    m_5Star = new KAction(i18n("Five Star"), CTRL+Key_5, 
+    m_5Star = new KAction(i18n("Assign Rating \"Five Stars\""), CTRL+Key_5, 
                           mView, SLOT(slotAssignRatingFiveStar()),
                           actionCollection(), "imageview_ratefivestar");
 
     // -----------------------------------------------------------
Only in digikam-0.9.0-beta1/digikam/digikam: digikamapp.cpp~
diff -U 4 -H -b -w -B -E -d -i -r digikam-0.9.0-beta1-orig/digikam/digikam/digikamview.cpp digikam-0.9.0-beta1/digikam/digikam/digikamview.cpp
--- digikam-0.9.0-beta1-orig/digikam/digikam/digikamview.cpp	2006-09-02 13:30:02.000000000 +0200
+++ digikam-0.9.0-beta1/digikam/digikam/digikamview.cpp	2006-09-02 13:34:06.000000000 +0200
@@ -882,39 +882,39 @@
 }
 
 void DigikamView::slotAssignRating(int rating)
 {
-    d->rightSideBar->slotAssignRating(rating);
+    d->iconView->slotAssignRating(rating);
 }
 
 void DigikamView::slotAssignRatingNoStar()
 {
-    d->rightSideBar->slotAssignRating(0);
+    d->iconView->slotAssignRating(0);
 }
 
 void DigikamView::slotAssignRatingOneStar()
 {
-    d->rightSideBar->slotAssignRating(1);
+    d->iconView->slotAssignRating(1);
 }
 
 void DigikamView::slotAssignRatingTwoStar()
 {
-    d->rightSideBar->slotAssignRating(2);
+    d->iconView->slotAssignRating(2);
 }
 
 void DigikamView::slotAssignRatingThreeStar()
 {
-    d->rightSideBar->slotAssignRating(3);
+    d->iconView->slotAssignRating(3);
 }
 
 void DigikamView::slotAssignRatingFourStar()
 {
-    d->rightSideBar->slotAssignRating(4);
+    d->iconView->slotAssignRating(4);
 }
 
 void DigikamView::slotAssignRatingFiveStar()
 {
-    d->rightSideBar->slotAssignRating(5);
+    d->iconView->slotAssignRating(5);
 }
 
 }  // namespace Digikam
 
Only in digikam-0.9.0-beta1/digikam/digikam: digikamview.cpp~
diff -U 4 -H -b -w -B -E -d -i -r digikam-0.9.0-beta1-orig/digikam/utilities/imageeditor/editor/imagewindow.cpp digikam-0.9.0-beta1/digikam/utilities/imageeditor/editor/imagewindow.cpp
--- digikam-0.9.0-beta1-orig/digikam/utilities/imageeditor/editor/imagewindow.cpp	2006-09-02 13:30:02.000000000 +0200
+++ digikam-0.9.0-beta1/digikam/utilities/imageeditor/editor/imagewindow.cpp	2006-09-02 13:36:25.000000000 +0200
@@ -223,24 +223,24 @@
     setupStandardActions();
 
     // -- Rating actions ---------------------------------------------------------------
 
-    m_0Star = new KAction(i18n("No Star"), CTRL+Key_0,
+    m_0Star = new KAction(i18n("Assign Rating \"No Star\""), CTRL+Key_0,
                           m_rightSidebar, SLOT(slotAssignRatingNoStar()),
                           actionCollection(), "imageview_ratenostar");
-    m_1Star = new KAction(i18n("One Star"), CTRL+Key_1,
+    m_1Star = new KAction(i18n("Assign Rating \"One Star\""), CTRL+Key_1,
                           m_rightSidebar, SLOT(slotAssignRatingOneStar()),
                           actionCollection(), "imageview_rateonestar");
-    m_2Star = new KAction(i18n("Two Star"), CTRL+Key_2, 
+    m_2Star = new KAction(i18n("Assign Rating \"Two Stars\""), CTRL+Key_2,
                           m_rightSidebar, SLOT(slotAssignRatingTwoStar()),
                           actionCollection(), "imageview_ratetwostar");
-    m_3Star = new KAction(i18n("Three Star"), CTRL+Key_3, 
+    m_3Star = new KAction(i18n("Assign Rating \"Three Stars\""), CTRL+Key_3,
                           m_rightSidebar, SLOT(slotAssignRatingThreeStar()),
                           actionCollection(), "imageview_ratethreestar");
-    m_4Star = new KAction(i18n("Four Star"), CTRL+Key_4, 
+    m_4Star = new KAction(i18n("Assign Rating \"Four Stars\""), CTRL+Key_4,
                           m_rightSidebar, SLOT(slotAssignRatingFourStar()),
                           actionCollection(), "imageview_ratefourstar");
-    m_5Star = new KAction(i18n("Five Star"), CTRL+Key_5, 
+    m_5Star = new KAction(i18n("Assign Rating \"Five Stars\""), CTRL+Key_5,
                           m_rightSidebar, SLOT(slotAssignRatingFiveStar()),
                           actionCollection(), "imageview_ratefivestar");
 
     // ---------------------------------------------------------------------------------
Only in digikam-0.9.0-beta1/digikam/utilities/imageeditor/editor: imagewindow.cpp~
