--- tests/Makefile.in~	2006-10-11 20:08:32.000000000 +0100
+++ tests/Makefile.in	2008-09-29 13:19:05.000000000 +0100
@@ -95,7 +95,9 @@
 MODULES_timesqrtmodp = timesqrtmodp
 MODULES_main = main
 
-all : exam${EXEEXT} tests${EXEEXT}
+all :
+
+build : exam${EXEEXT} tests${EXEEXT}
 
 %.s : %.c
 	${CC} ${CFLAGS} ${CPPFLAGS} ${TARGET_ARCH} -S $< -o $@
@@ -171,7 +173,7 @@
 
 uninstall :
 
-check : all
+check : build
 	./exam${EXEEXT}
 	./tests${EXEEXT}
 
--- examples/Makefile.in~	2006-09-06 21:09:08.000000000 +0100
+++ examples/Makefile.in	2008-09-29 13:17:24.000000000 +0100
@@ -62,7 +62,8 @@
 MODULES_perfnum = perfnum
 MODULES_pi = pi
 
-all : $(addsuffix ${EXEEXT}, contfrac e fibonacci legendre lucaslehmer nextprime perfnum pi)
+all:
+build : $(addsuffix ${EXEEXT}, contfrac e fibonacci legendre lucaslehmer nextprime perfnum pi)
 
 %.s : %.c
 	${CC} ${CFLAGS} ${CPPFLAGS} ${TARGET_ARCH} -S $< -o $@
