Index: nspr-4.6.3/mozilla/nsprpub/pr/src/misc/prnetdb.c
===================================================================
--- nspr-4.6.3.orig/mozilla/nsprpub/pr/src/misc/prnetdb.c
+++ nspr-4.6.3/mozilla/nsprpub/pr/src/misc/prnetdb.c
@@ -108,6 +108,14 @@ PRLock *_pr_dnsLock = NULL;
 #define _PR_HAVE_GETPROTO_R_INT
 #endif
 
+#if defined(__FreeBSD__)
+#include <sys/param.h>
+#if __FreeBSD_version >= 602000
+#define _PR_HAVE_GETPROTO_R
+#define _PR_HAVE_5_ARG_GETPROTO_R
+#endif
+#endif
+
 #if (defined(LINUX) && defined(__GLIBC__) && __GLIBC__ >= 2)
 #define _PR_HAVE_GETPROTO_R
 #define _PR_HAVE_5_ARG_GETPROTO_R
