--- contrib/cube/cubescan.l.orig	2006-07-05 06:56:04.000000000 +0000
+++ contrib/cube/cubescan.l	2006-07-05 06:56:12.000000000 +0000
@@ -9,6 +9,10 @@
 
 
 /* flex screws a couple symbols when used with the -P option; fix those */
+#ifndef YY_PROTO
+#define YY_PROTO(proto) proto
+#endif
+
 #define YY_DECL int cube_yylex YY_PROTO(( void )); \
 int cube_yylex YY_PROTO(( void ))
 #define yylval cube_yylval
--- contrib/seg/segscan.l.orig	2006-07-05 07:08:26.000000000 +0000
+++ contrib/seg/segscan.l	2006-07-05 07:08:51.000000000 +0000
@@ -9,6 +9,10 @@
 
 
 /* flex screws a couple symbols when used with the -P option; fix those */
+#ifndef YY_PROTO
+#define YY_PROTO(proto) proto
+#endif
+
 #define YY_DECL int seg_yylex YY_PROTO(( void )); \
 int seg_yylex YY_PROTO(( void ))
 #define yylval seg_yylval
--- contrib/tsearch/parser.l.orig	2006-07-05 07:22:36.000000000 +0000
+++ contrib/tsearch/parser.l	2006-07-05 07:23:33.000000000 +0000
@@ -18,6 +18,7 @@
 char *s     = NULL;  /* to return WHOLE hyphenated-word */
 
 YY_BUFFER_STATE buf = NULL; /* buffer to parse; it need for parse from string */
+static YY_BUFFER_STATE yy_current_buffer = 0;
 
 int lrlimit = -1;	/* for limiting read from filehandle ( -1 - unlimited read ) */
 int bytestoread = 0;	/* for limiting read from filehandle */
