Index: kscreensaver/kxsconfig/configure.in.bot
===================================================================
--- kscreensaver/kxsconfig/configure.in.bot	(revision 428301)
+++ kscreensaver/kxsconfig/configure.in.bot	(working copy)
@@ -1,4 +1,4 @@
-if test "$XSCREENSAVER" = "NO"; then
+if test "x$with_xscreensaver" = xcheck && test "$XSCREENSAVER" = "NO"; then
   echo ""
   echo "xscreensaver could not be detected on your system."
   echo "Check www.jwz.org/xscreensaver if you're loving to"
Index: kscreensaver/kxsconfig/configure.in.in
===================================================================
--- kscreensaver/kxsconfig/configure.in.in	(revision 428301)
+++ kscreensaver/kxsconfig/configure.in.in	(working copy)
@@ -22,7 +22,14 @@
 done
 IFS=$old_IFS
 
-KDE_FIND_PATH(lib$kdelibsuff/xscreensaver/flame, XSCREENSAVER, [ $xscreensaver_prefixes ],
+AC_ARG_WITH(xscreensaver,
+  [AS_HELP_STRING(--with-xscreensaver,
+    [enable support for xscreensaver @<:@default=check@:>@])],
+  [], with_xscreensaver=check)
+
+XSCREENSAVER=NO
+if test "x$with_xscreensaver" != xno; then
+  KDE_FIND_PATH(lib$kdelibsuff/xscreensaver/flame, XSCREENSAVER, [ $xscreensaver_prefixes ],
    [
         KDE_FIND_PATH(lib/xscreensaver/flame, XSCREENSAVER, [ $xscreensaver_prefixes ],
         [
@@ -38,6 +45,11 @@
         ])
    ])
 
+  if test "x$with_xscreensaver" != xcheck && test "x$XSCREENSAVER" = xNO; then
+    AC_MSG_FAILURE([--with-xscreensaver was given, but test for xscreensaver failed])
+  fi
+fi
+
 XSCREENSAVER_HACKS_DIR=
 
 if test "$XSCREENSAVER" != "NO"; then
