--- configure.in
+++ configure.in
@@ -149,10 +149,8 @@
           USE_DEBUG="$enableval", USE_DEBUG="no")
 if test "x$USE_DEBUG" == "xyes" ; then
     AC_DEFINE_UNQUOTED(DEBUG, 1, "Enable debug messages.")
-    CFLAGS="$CFLAGS -Wall -Werror -g"
-else
-    CFLAGS="$CFLAGS -Wall -Werror"
 fi
+CFLAGS="$CFLAGS -Wall"
 
 dnl WIN32 Specifics
 AC_MSG_CHECKING([for native Win32])
@@ -219,25 +217,25 @@
 dnl
 
 dnl Set PACKAGE_LOCALE_DIR in config.h.
-AC_DEFINE_UNQUOTED(PACKAGE_LOCALE_DIR, "${myprefix}/${DATADIRNAME}/locale", [GCompris locale directory])
+AC_DEFINE_UNQUOTED(PACKAGE_LOCALE_DIR, "/usr/share/locale", [GCompris locale directory])
 
 dnl Set PACKAGE_DATA_DIR
 PACKAGE_DATA_DIR="boards"
 AC_SUBST(PACKAGE_DATA_DIR)
-AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${myprefix}/${DATADIRNAME}", [GCompris data directory])
+AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${datadir}", [GCompris data directory])
 
 dnl Set PACKAGE_HELP_DIR
-PACKAGE_HELP_DIR="${myprefix}/${DATADIRNAME}/${PACKAGE}/help"
+PACKAGE_HELP_DIR="${datadir}/${PACKAGE}/help"
 AC_SUBST(PACKAGE_HELP_DIR)
 
 dnl Plugin Directory
 my_libdir=`eval echo "${libdir}"`
-AC_DEFINE_UNQUOTED(PACKAGE_CLIB_DIR, "${my_libdir}", [Gcompris plugins directory])
+AC_DEFINE_UNQUOTED(PACKAGE_CLIB_DIR, "${libdir}", [Gcompris plugins directory])
 plugindir=$libdir/gcompris
 AC_SUBST(plugindir)
 
 dnl Python plugin Directory
-PYTHON_PLUGIN_DIR="${myprefix}/${DATADIRNAME}/${PACKAGE}/python"
+PYTHON_PLUGIN_DIR="${datadir}/${PACKAGE}/python"
 AC_SUBST(PYTHON_PLUGIN_DIR)
 
 dnl Test for gnuchess
--- docs/C/Makefile.am
+++ docs/C/Makefile.am
@@ -10,7 +10,7 @@
 	     overview.dia \
 	     $(man_MANS)
 
-gcompris_helpdir = $(datadir)/gnome/help/$(PACKAGE)/C
+gcompris_helpdir = /usr/share/gnome/help/$(PACKAGE)/C
 
 all: $(doc)
 
--- docs/eu/Makefile.am
+++ docs/eu/Makefile.am
@@ -8,7 +8,7 @@
 	     overview.dia \
 	     gcompris.texi
 
-gcompris_helpdir = $(datadir)/gnome/help/$(PACKAGE)/eu
+gcompris_helpdir = /usr/share/gnome/help/$(PACKAGE)/eu
 
 all: $(doc)
 
--- docs/fr/Makefile.am
+++ docs/fr/Makefile.am
@@ -8,7 +8,7 @@
 	     overview.dia \
 	     gcompris.texi
 
-gcompris_helpdir = $(datadir)/gnome/help/$(PACKAGE)/fr
+gcompris_helpdir = /usr/share/gnome/help/$(PACKAGE)/fr
 
 all: $(doc)
 
--- Makefile.am
+++ Makefile.am
@@ -8,15 +8,13 @@
 
 SUBDIRS = po src boards $(DOCS_SUBDIR) autopackage
 
-icondir=$(datadir)/pixmaps
+icondir=/usr/share/pixmaps
 icon_DATA = gcompris.png gcompris-edit.png
 
-Applicationsdir = $(datadir)/applications/
+Applicationsdir = /usr/share/applications
 Applications_in_files = gcompris.desktop.in gcompris-edit.desktop.in
 Applications_DATA = gcompris.desktop gcompris-edit.desktop
 
-Menudir=$(prefix)/lib/menu
-Menu_DATA=gcompris
 
 @INTLTOOL_DESKTOP_RULE@
 
--- po/Makefile.in.in
+++ po/Makefile.in.in
@@ -34,7 +34,7 @@
 datarootdir = @datarootdir@
 libdir = @libdir@
 DATADIRNAME = @DATADIRNAME@
-itlocaledir = $(prefix)/$(DATADIRNAME)/locale
+itlocaledir = /usr/share/locale
 subdir = po
 install_sh = @install_sh@
 # Automake >= 1.8 provides @mkdir_p@.
--- src/gcompris/gcompris.c
+++ src/gcompris/gcompris.c
@@ -1117,7 +1117,7 @@
       properties->package_locale_dir = gbr_find_locale_dir(PACKAGE_LOCALE_DIR);
       properties->package_plugin_dir = g_strconcat(pkg_clib_dir, "/gcompris", NULL);
       properties->package_python_plugin_dir = g_strconcat(pkg_data_dir, "/gcompris/python", NULL);
-      properties->system_icon_dir = g_strconcat(pkg_data_dir, "/pixmaps", NULL);
+      properties->system_icon_dir = g_strconcat("/usr/share", "/pixmaps", NULL);
       g_free(pkg_data_dir);
       g_free(pkg_clib_dir);
     }
--- src/gcompris/Makefile.am
+++ src/gcompris/Makefile.am
@@ -69,7 +69,7 @@
 
 libgcomprisincludedir = $(includedir)/libgcompris-1.0/libgcompris
 
-libgcomprisinclude_HEADERS =			\
+noinst_HEADERS =			\
 	about.h \
 	anim.h \
 	binreloc.h \
@@ -97,7 +97,7 @@
 	$(cairo_h)
 
 libgcompris_1_la_SOURCES =			\
-	$(libgcomprisinclude_HEADERS)	\
+	$(noinst_HEADERS)	\
 	about.c \
 	anim.c \
 	bar.c \
