diff -Ndur gcompris-8.3.3.orig/configure.in gcompris-8.3.3.patched/configure.in
--- gcompris-8.3.3.orig/configure.in	2007-08-16 01:09:34.000000000 +0200
+++ gcompris-8.3.3.patched/configure.in	2007-08-22 07:51:31.000000000 +0200
@@ -258,10 +258,8 @@
           USE_DEBUG="$enableval", USE_DEBUG="no")
 if test "x$USE_DEBUG" == "xyes" ; then
     AC_DEFINE_UNQUOTED(DEBUG, 1, "Enable debug messages.")
-    CFLAGS="$CFLAGS $OSX_CFLAGS -Wall -Werror -g"
-else
-    CFLAGS="$CFLAGS $OSX_CFLAGS -Wall -Werror"
 fi
+CFLAGS="$CFLAGS $OSX_CFLAGS -Wall"
 
 dnl WIN32 Specifics
 AC_MSG_CHECKING([for native Win32])
@@ -330,25 +328,25 @@
 if test "x$platform_win32" = "xno" ; then
 
   dnl Set PACKAGE_LOCALE_DIR in config.h.
-  AC_DEFINE_UNQUOTED(PACKAGE_LOCALE_DIR, "${myprefix}/${DATADIRNAME}/locale", [GCompris locale directory])
+  AC_DEFINE_UNQUOTED(PACKAGE_LOCALE_DIR, "/usr/share/locale", [GCompris locale directory])
 
   dnl Set PACKAGE_DATA_DIR
   PACKAGE_DATA_DIR="boards"
   AC_SUBST(PACKAGE_DATA_DIR)
-  AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${myprefix}/${DATADIRNAME}", [GCompris data directory])
+  AC_DEFINE_UNQUOTED(PACKAGE_DATA_DIR, "${datadir}", [GCompris data directory])
 
   dnl Set PACKAGE_HELP_DIR
-  PACKAGE_HELP_DIR="${myprefix}/${DATADIRNAME}/${PACKAGE}/help"
+  PACKAGE_HELP_DIR="${datadir}/${PACKAGE}/help"
   AC_SUBST(PACKAGE_HELP_DIR)
 
   dnl Plugin Directory
   my_libdir=`eval echo "${libdir}"`
-  AC_DEFINE_UNQUOTED(PACKAGE_CLIB_DIR, "${my_libdir}", [Gcompris plugins directory])
+  AC_DEFINE_UNQUOTED(PACKAGE_CLIB_DIR, "${libdir}", [Gcompris plugins directory])
   plugindir=$libdir/gcompris
   AC_SUBST(plugindir)
 
   dnl Python plugin Directory
-  PYTHON_PLUGIN_DIR="${myprefix}/${DATADIRNAME}/${PACKAGE}/python"
+  PYTHON_PLUGIN_DIR="${datadir}/${PACKAGE}/python"
   AC_SUBST(PYTHON_PLUGIN_DIR)
 
   dnl System GCompris config directory
diff -Ndur gcompris-8.3.3.orig/docs/C/Makefile.am gcompris-8.3.3.patched/docs/C/Makefile.am
--- gcompris-8.3.3.orig/docs/C/Makefile.am	2007-06-17 21:23:54.000000000 +0200
+++ gcompris-8.3.3.patched/docs/C/Makefile.am	2007-08-22 07:35:21.000000000 +0200
@@ -10,7 +10,7 @@
 	     overview.dia \
 	     $(man_MANS)
 
-gcompris_helpdir = $(datadir)/gnome/help/$(PACKAGE)/C
+gcompris_helpdir = /usr/share/gnome/help/$(PACKAGE)/C
 
 all: $(doc)
 
diff -Ndur gcompris-8.3.3.orig/docs/eu/Makefile.am gcompris-8.3.3.patched/docs/eu/Makefile.am
--- gcompris-8.3.3.orig/docs/eu/Makefile.am	2007-06-17 21:23:55.000000000 +0200
+++ gcompris-8.3.3.patched/docs/eu/Makefile.am	2007-08-22 07:35:21.000000000 +0200
@@ -8,7 +8,7 @@
 	     overview.dia \
 	     gcompris.texi
 
-gcompris_helpdir = $(datadir)/gnome/help/$(PACKAGE)/eu
+gcompris_helpdir = /usr/share/gnome/help/$(PACKAGE)/eu
 
 all: $(doc)
 
diff -Ndur gcompris-8.3.3.orig/docs/fr/Makefile.am gcompris-8.3.3.patched/docs/fr/Makefile.am
--- gcompris-8.3.3.orig/docs/fr/Makefile.am	2007-06-17 21:23:55.000000000 +0200
+++ gcompris-8.3.3.patched/docs/fr/Makefile.am	2007-08-22 07:35:21.000000000 +0200
@@ -8,7 +8,7 @@
 	     overview.dia \
 	     gcompris.texi
 
-gcompris_helpdir = $(datadir)/gnome/help/$(PACKAGE)/fr
+gcompris_helpdir = /usr/share/gnome/help/$(PACKAGE)/fr
 
 all: $(doc)
 
diff -Ndur gcompris-8.3.3.orig/Makefile.am gcompris-8.3.3.patched/Makefile.am
--- gcompris-8.3.3.orig/Makefile.am	2007-06-17 21:27:17.000000000 +0200
+++ gcompris-8.3.3.patched/Makefile.am	2007-08-22 07:59:08.000000000 +0200
@@ -17,7 +17,7 @@
 endif
 
 
-icondir=$(datadir)/pixmaps
+icondir=/usr/share/pixmaps
 icon_DATA = gcompris.png gcompris-edit.png
 
 if BUILD_MINGW32
@@ -29,12 +29,10 @@
 Menu_DATA=
 
 else
-Applicationsdir = $(datadir)/applications/
+Applicationsdir = /usr/share/applications
 Applications_in_files = gcompris.desktop.in gcompris-edit.desktop.in
 Applications_DATA = gcompris.desktop gcompris-edit.desktop
 
-Menudir=$(prefix)/lib/menu
-Menu_DATA=gcompris
 endif
 
 @INTLTOOL_DESKTOP_RULE@
diff -Ndur gcompris-8.3.3.orig/po/Makefile.in.in gcompris-8.3.3.patched/po/Makefile.in.in
--- gcompris-8.3.3.orig/po/Makefile.in.in	2007-03-05 11:52:19.000000000 +0100
+++ gcompris-8.3.3.patched/po/Makefile.in.in	2007-08-22 07:35:21.000000000 +0200
@@ -34,7 +34,7 @@
 datarootdir = @datarootdir@
 libdir = @libdir@
 DATADIRNAME = @DATADIRNAME@
-itlocaledir = $(prefix)/$(DATADIRNAME)/locale
+itlocaledir = /usr/share/locale
 subdir = po
 install_sh = @install_sh@
 # Automake >= 1.8 provides @mkdir_p@.
diff -Ndur gcompris-8.3.3.orig/src/gcompris/gcompris.c gcompris-8.3.3.patched/src/gcompris/gcompris.c
--- gcompris-8.3.3.orig/src/gcompris/gcompris.c	2007-06-19 00:19:49.000000000 +0200
+++ gcompris-8.3.3.patched/src/gcompris/gcompris.c	2007-08-22 07:35:21.000000000 +0200
@@ -1120,7 +1120,7 @@
       properties->package_locale_dir = gbr_find_locale_dir(PACKAGE_LOCALE_DIR);
       properties->package_plugin_dir = g_strconcat(pkg_clib_dir, "/gcompris", NULL);
       properties->package_python_plugin_dir = g_strconcat(pkg_data_dir, "/gcompris/python", NULL);
-      properties->system_icon_dir = g_strconcat(pkg_data_dir, "/pixmaps", NULL);
+      properties->system_icon_dir = g_strconcat("/usr/share", "/pixmaps", NULL);
       g_free(pkg_data_dir);
       g_free(pkg_clib_dir);
     }
