--- genmenu.orig	2002-10-14 10:38:51.000000000 -0700
+++ genmenu	2002-10-27 00:08:47.000000000 -0700
@@ -796,7 +796,7 @@
 {
 	local thiswm
 
-	echo "\"Other Window Managers\""
+	echo '"Other Window Managers"'
 
 	for thiswm in $WINDOWMANAGERS; do
 		if [ "$thiswm" != "enlightenment" ]; then
@@ -913,6 +913,7 @@
 	fi
 
 	echo -e "\nWriting Fluxbox menu\n"
+	mv $fb_menu $fb_menu.orig
 	fb_writemenu >$fb_menu
 	echo
 
@@ -965,6 +966,7 @@
 	fi
 
 	echo -e "\nWriting Blackbox menu\n"
+	mv $bb_menu $bb_menu.orig
 	bb_writemenu >$bb_menu
 	echo
 
@@ -992,6 +994,7 @@
 	fi
 
 	echo -e "\nWriting Window Maker menu\n"
+	mv $menu $menu.orig
 	wm_writemenu >$menu
  echo
 }
@@ -1022,6 +1025,8 @@
 
 	programs
 
+	mv $menu $menu.orig
+	
 	( echo "\"User Menus\""
 
 	if [ -f "$ECONFDIR/genmenu.menu" ]; then
