--- Makefile
+++ Makefile
@@ -10,15 +10,16 @@
 MAN        = tcpsound.1
 MANGZ      = $(MAN:.1=.1.gz)
 MKTOOL     = ./mktool
-CFLAGS     = -c -v -g -W1 -I$(includedir) -L$(libdir) -lSDL -lmba -lpthread
+CFLAGS     = -c -v -g -W1 -I$(includedir) 
+LIBS	    = -L$(libdir) -lSDL -lmba -lpthread
 CC         = $(MKTOOL)
 
 .SUFFIXES: .1 .1.gz
 
 tcpsound: mktool src/tcpsound.c src/sound.h src/sound.c src/parse.h src/parse.c
-	@$(CC) $(CFLAGS) src/sound.c src/parse.c src/tcpsound.c -o tcpsound
+	@$(CC) $(CFLAGS) src/sound.c src/parse.c src/tcpsound.c -o tcpsound $(LIBS)
 play: mktool src/play.c
-	@$(CC) $(CFLAGS) src/play.c -o play
+	@$(CC) $(CFLAGS) src/play.c -o play $(LIBS)
 
 mktool:
 	cc -g -o mktool mktool.c
