Index: sooperlooper-1.1.0/src/Makefile.in
===================================================================
--- sooperlooper-1.1.0.orig/src/Makefile.in
+++ sooperlooper-1.1.0/src/Makefile.in
@@ -277,7 +277,7 @@ libsldrivers_a_SOURCES = \
 sooperlooper_SOURCES = \
 	sooperlooper.cpp
 
-sooperlooper_LDADD = libsldrivers.a libslcore.a @JACK_LIBS@ @LOSC_LIBS@ @SIGCPP_LIBS@ @SNDFILE_LIBS@ @SAMPLERATE_LIBS@ @AUDIO_LIBS@ @XML_LIBS@
+sooperlooper_LDADD = libsldrivers.a libslcore.a @BASE_LIBS@ @JACK_LIBS@ @LOSC_LIBS@ @SIGCPP_LIBS@ @SNDFILE_LIBS@ @SAMPLERATE_LIBS@ @AUDIO_LIBS@ @XML_LIBS@
 sooperlooper_LDFLAGS = 
 slconsole_SOURCES = slconsole.cpp
 slconsole_LDADD = @LOSC_LIBS@ @NCURSES_LIBS@ -lpthread
