--- qe.c.orig	2009-03-07 21:00:39.000000000 +0100
+++ qe.c	2009-03-07 21:10:50.000000000 +0100
@@ -7505,15 +7505,15 @@
     /* compute resources path */
     qs->res_path[0] = '\0';
 
-    /* put source directory first if qe invoked as ./qe */
+    /* put source directory first if qe invoked as ./qemacs */
     // should use actual directory
-    if (stristart(qs->argv[0], "./qe", NULL)) {
+    if (stristart(qs->argv[0], "./qemacs", NULL)) {
         pstrcat(qs->res_path, sizeof(qs->res_path), ".:");
     }
 
     /* put user directory before standard list */
     if (user) {
-        /* use ~USER/.qe instead of ~/.qe */
+        /* use ~USER/.qemacs instead of ~/.qemacs */
         /* CG: should get user homedir */
         snprintf(path, sizeof(path), "/home/%s", user);
         home_path = path;
@@ -7522,14 +7522,14 @@
     }
     if (home_path) {
         pstrcat(qs->res_path, sizeof(qs->res_path), home_path);
-        pstrcat(qs->res_path, sizeof(qs->res_path), "/.qe:");
+        pstrcat(qs->res_path, sizeof(qs->res_path), "/.qemacs:");
     }
 
     pstrcat(qs->res_path, sizeof(qs->res_path),
-            CONFIG_QE_PREFIX "/share/qe:"
-            CONFIG_QE_PREFIX "/lib/qe:"
-            "/usr/share/qe:"
-            "/usr/lib/qe");
+            CONFIG_QE_PREFIX "/share/qemacs:"
+            CONFIG_QE_PREFIX "/lib/qemacs:"
+            "/usr/share/qemacs:"
+            "/usr/lib/qemacs");
 }
 
 void set_tty_charset(const char *name)
@@ -7545,7 +7545,7 @@
       { .int_ptr = &no_init_file }},
     { "ttycharset", "c", "CHARSET", CMD_OPT_ARG, "specify tty charset",
       { .func_arg = set_tty_charset }},
-    { "user", "u", "USER", CMD_OPT_ARG, "load ~USER/.qe/config instead of your own",
+    { "user", "u", "USER", CMD_OPT_ARG, "load ~USER/.qemacs/config instead of your own",
       { .func_arg = set_user_option }},
     { "version", "V", NULL, 0, "display version information and exit",
       { .func_noarg = show_version }},
--- qe-doc.texi.orig	2009-03-07 21:00:56.000000000 +0100
+++ qe-doc.texi	2009-03-07 21:08:49.000000000 +0100
@@ -274,11 +274,11 @@
 @section Resource path
 
 All resources and configuration files are looked in the following paths:
-@file{/usr/share/qe:/usr/local/share/qe:/usr/lib/qe:/usr/local/lib/qe:~/.qe}
+@file{/usr/share/qemacs:/usr/local/share/qemacs:/usr/lib/qemacs:/usr/local/lib/qemacs:~/.qemacs}
 
 @section Configuration file
 
-QEmacs tries to load a configuration file in @file{~/.qe/config}. Each
+QEmacs tries to load a configuration file in @file{~/.qemacs/config}. Each
 line of the configuration file is a QEmacs command with a C like syntax
 ('-' in command name can be replaced by '_').
 
--- qe-doc.html.orig	2009-03-07 21:01:02.000000000 +0100
+++ qe-doc.html	2009-03-07 21:08:49.000000000 +0100
@@ -619,7 +619,7 @@
 <h2 class="section"> 4.1 Resource path </h2>
 
 <p>All resources and configuration files are looked in the following paths:
-&lsquo;<tt>/usr/share/qe:/usr/local/share/qe:/usr/lib/qe:/usr/local/lib/qe:~/.qe</tt>&rsquo;
+&lsquo;<tt>/usr/share/qemacs:/usr/local/share/qemacs:/usr/lib/qemacs:/usr/local/lib/qemacs:~/.qemacs</tt>&rsquo;
 </p>
 <hr size="6">
 <a name="SEC19"></a>
@@ -641,7 +641,7 @@
 </tr></table>
 <h2 class="section"> 4.2 Configuration file </h2>
 
-<p>QEmacs tries to load a configuration file in &lsquo;<tt>~/.qe/config</tt>&rsquo;. Each
+<p>QEmacs tries to load a configuration file in &lsquo;<tt>~/.qemacs/config</tt>&rsquo;. Each
 line of the configuration file is a QEmacs command with a C like syntax
 ('-' in command name can be replaced by '_').
 </p>
--- config.eg.orig	2009-03-07 21:00:21.000000000 +0100
+++ config.eg	2009-03-07 21:08:49.000000000 +0100
@@ -1,6 +1,6 @@
 // Sample qemacs config file 
 //
-// copy it in ~/.qe/config
+// copy it in ~/.qemacs/config
 //
 // You can call any Qemacs command with a C like syntax.
 //
