--- qemacs-0.3.1/Makefile	2005-02-21 10:04:00.000000000 +0000
+++ qemacs-0.3.1/Makefile	2005-02-21 10:04:00.000000000 +0000
@@ -29,14 +29,6 @@ ifdef TARGET_GPROF
 CFLAGS+= -p
 LDFLAGS+= -p
 endif
-ifdef TARGET_ARCH_X86
-#CFLAGS+=-fomit-frame-pointer
-ifeq ($(GCC_MAJOR),2)
-CFLAGS+=-m386 -malign-functions=0
-else
-CFLAGS+=-march=i386 -falign-functions=0
-endif
-endif
 DEFINES=-DHAVE_QE_CONFIG_H
 
 ########################################################
@@ -190,15 +182,17 @@ distclean: clean
 	rm -f config.h config.mak
 
 install: qe qe.1 kmaps ligatures html2png
-	install -m 755 qe $(prefix)/bin/qemacs
-	ln -sf qemacs $(prefix)/bin/qe
+	mkdir -p $(DESTDIR)$(prefix)/bin
+	install -m 755 qe $(DESTDIR)$(prefix)/bin/qemacs
+	install -m 755 libqhtml/csstoqe $(DESTDIR)$(prefix)/bin/csstoqe
 ifdef CONFIG_FFMPEG
-	ln -sf qemacs $(prefix)/bin/ffplay
+	ln -sf qemacs $(DESTDIR)$(prefix)/bin/qemacs-ffplay
 endif
-	mkdir -p $(prefix)/share/qe
-	install kmaps ligatures $(prefix)/share/qe
-	install qe.1 $(prefix)/man/man1
-	install -m 755 -s html2png $(prefix)/bin
+	mkdir -p $(DESTDIR)$(prefix)/share/qemacs
+	install -m644 kmaps ligatures $(DESTDIR)$(prefix)/share/qe
+	mkdir -p $(DESTDIR)$(prefix)/share/man/man1
+	install -m644 qe.1 $(DESTDIR)$(prefix)/share/man/man1/qemacs.1
+	install -m 755 -s html2png $(DESTDIR)$(prefix)/bin
 
 TAGS: force
 	etags *.[ch]
--- qemacs-0.3.1/plugin-example/Makefile	2005-02-21 03:24:37.000000000 +0000
+++ qemacs-0.3.1/plugin-example/Makefile	2005-02-21 03:24:37.000000000 +0000
@@ -6,7 +6,7 @@
 #
 
 # modify to get the path of the qemacs sources
-QEMACS_PATH=../
+QEMACS_PATH=/usr/include/qemacs
 # define the name of your plugin
 LIB=my_plugin.so
 # define the object files it contains
@@ -20,7 +20,7 @@ CFLAGS=-O2 -Wall -g -fPIC $(INCLUDES) $(
 
 # by default, the plugin is copied in ~/.qe/ some that qemacs can load
 # it automatically.
-all: $(DIST_LIB)
+all: $(LIB)
 
 $(DIST_LIB): $(LIB)
 	install -s -m 644 $< $@
--- qemacs-0.3.1/libqhtml/Makefile	2005-02-21 03:24:37.000000000 +0000
+++ qemacs-0.3.1/libqhtml/Makefile	2005-02-21 03:24:37.000000000 +0000
@@ -5,14 +5,6 @@ ifdef TARGET_GPROF
 CFLAGS+= -p
 LDFLAGS+= -p
 endif
-ifdef TARGET_ARCH_X86
-#CFLAGS+=-fomit-frame-pointer
-ifeq ($(GCC_MAJOR),2)
-CFLAGS+=-m386 -malign-functions=0
-else
-CFLAGS+=-march=i386 -falign-functions=0
-endif
-endif
 CFLAGS+=-I..
 
 OBJS=css.o xmlparse.o cssparse.o html_style.o docbook_style.o
