diff -urN gstreamer-0.8.10.orig/gst/cothreads.c gstreamer-0.8.10/gst/cothreads.c
--- gstreamer-0.8.10.orig/gst/cothreads.c	2004-11-02 16:02:11.000000000 +0100
+++ gstreamer-0.8.10/gst/cothreads.c	2005-08-07 15:19:11.000000000 +0200
@@ -651,7 +651,7 @@
     makecontext (&ucp, cothread_stub, 0);
     setcontext (&ucp);
 #else
-    GST_ARCH_SETUP_STACK ((char *) cothread->sp);
+    GST_ARCH_SETUP_STACK (cothread->sp);
     GST_ARCH_SET_SP (cothread->sp);
     /* start it */
     GST_ARCH_CALL (cothread_stub);
diff -urN gstreamer-0.8.10.orig/gst/gstarch.h gstreamer-0.8.10/gst/gstarch.h
--- gstreamer-0.8.10.orig/gst/gstarch.h	2004-03-15 15:43:21.000000000 +0100
+++ gstreamer-0.8.10/gst/gstarch.h	2005-08-07 15:19:01.000000000 +0200
@@ -38,7 +38,7 @@
     __asm__("call *%0" : : "r"(target) );
 
 /* assuming the stackframe is 16 bytes */
-#define GST_ARCH_SETUP_STACK(sp) sp -= 4
+#define GST_ARCH_SETUP_STACK(sp) (sp) = (char *)(sp) - 4
 
 
 
