--- gstreamer-0.5.0/docs/xsl/html.xsl.orig	2002-12-27 00:24:49.000000000 +0200
+++ gstreamer-0.5.0/docs/xsl/html.xsl	2002-12-27 00:35:51.000000000 +0200
@@ -7,7 +7,7 @@
 <!-- This file is for customizing the default XSL stylesheets. -->
 <!-- We include them here (this one is for HTML output):       -->
 <xsl:import
- href="http://docbook.sourceforge.net/release/xsl/1.50.0/xhtml/chunk.xsl"/>
+ href="http://docbook.sourceforge.net/release/xsl/1.57.0/xhtml/chunk.xsl"/>
 
 <!-- .. and customize them here:                               -->
 <xsl:include href="fileext.xsl"/>
