--- gst/registries/gstxmlregistry.c	2004/05/27 01:04:12	1.1
+++ gst/registries/gstxmlregistry.c	2004/05/27 01:05:58
@@ -1318,9 +1318,9 @@ gst_xml_registry_rebuild_recurse (GstXML
   GList *ret = NULL;
   gint dr_len, sf_len;
 
-  dir = g_dir_open (directory, 0, NULL);
-
-  if (dir) {
+  if (g_file_test (directory, G_FILE_TEST_IS_DIR)) {
+   dir = g_dir_open (directory, 0, NULL);
+   if (dir) {
     const gchar *dirent;
 
     while ((dirent = g_dir_read_name (dir))) {
@@ -1338,6 +1338,7 @@ gst_xml_registry_rebuild_recurse (GstXML
       g_free (dirname);
     }
     g_dir_close (dir);
+   }
   } else {
     dr_len = strlen (directory);
     sf_len = strlen (G_MODULE_SUFFIX);
