--- common/nx-X11/programs/Xserver/hw/nxagent/Display.c.orig	2005-07-29 09:01:11.000000000 -0700
+++ common/nx-X11/programs/Xserver/hw/nxagent/Display.c	2005-07-30 07:16:06.000000000 -0700
@@ -1531,11 +1531,12 @@
   reconnectDisplayState = NOTHING;
 }
 
+#define WARNING
 static Bool reloadVisuals(int flexibility)
 {
   int i, nitems;
   XVisualInfo *newVisuals, *vi;
-  unsigned long mask = VisualAllMask & ~VisualIDMask;
+  unsigned long mask = VisualAllMask & ~VisualIDMask & ~VisualBitsPerRGBMask;
 
   newVisuals = malloc(sizeof(XVisualInfo) * nxagentNumVisuals);
 
@@ -1577,6 +1578,10 @@
     }
   }
 
+        #ifdef WARNING
+        fprintf(stderr, "matched for %d visuals and found %d matching\n", nxagentNumVisuals,i);
+        #endif
+
   if (i == nxagentNumVisuals)
   {
     nxagentVisuals = newVisuals;
--- common/nx-X11/programs/Xserver/hw/nxagent/Visual.c.orig	2005-07-30 07:17:40.000000000 -0700
+++ common/nx-X11/programs/Xserver/hw/nxagent/Visual.c	2005-07-30 07:17:44.000000000 -0700
@@ -48,7 +48,6 @@
   for (i = 0; i < nxagentNumVisuals; i++)
   {
     if (pVisual->class == nxagentVisuals[i].class &&
-        pVisual->bitsPerRGBValue == nxagentVisuals[i].bits_per_rgb &&
         pVisual->ColormapEntries == nxagentVisuals[i].colormap_size &&
         pVisual->nplanes == nxagentVisuals[i].depth &&
         pVisual->redMask == nxagentVisuals[i].red_mask &&
