diff -ru nx-X11-orig/lib/X11/XlcDL.c nx-X11/lib/X11/XlcDL.c
--- common/nx-X11-orig/lib/X11/XlcDL.c	2006-10-04 16:51:54.000000000 -0600
+++ common/nx-X11/lib/X11/XlcDL.c	2006-10-04 16:52:55.000000000 -0600
@@ -58,10 +58,8 @@
 #ifdef _LP64
 # if defined(__sparcv9)
 #  define	_MACH64_NAME		"sparcv9"
-# elif defined(__ia64__) 
-#  undef MACH64_NAME
 # else
-#  error "Unknown architecture"
+#  undef MACH64_NAME
 # endif /* defined(__sparcv9) */
 # ifdef _MACH64_NAME
 #  define	_MACH64_NAME_LEN	(sizeof (_MACH64_NAME) - 1)
