--- common/nx-X11/programs/Xserver/hw/nxagent/GCOps.c.orig	2006-11-22 21:59:13.000000000 +0000
+++ common/nx-X11/programs/Xserver/hw/nxagent/GCOps.c	2006-11-22 22:03:12.000000000 +0000
@@ -2399,7 +2399,7 @@
   {
     if ((pDrawable)->type == DRAWABLE_PIXMAP)
     {
-      miPolySegment(nxagentVirtualDrawable(pDrawable), pGC, nSegments, pSegments);
+      fbPolySegment(nxagentVirtualDrawable(pDrawable), pGC, nSegments, pSegments);
     }
 
     return;
@@ -2435,7 +2435,7 @@
       XDrawSegments(nxagentDisplay, nxagentDrawable(pDrawable), nxagentGC(pGC),
                     (XSegment *)pSegments, nSegments);
     }
-    miPolySegment(nxagentVirtualDrawable(pDrawable), pGC, nSegments, pSegments);
+    fbPolySegment(nxagentVirtualDrawable(pDrawable), pGC, nSegments, pSegments);
     return;
   }
   else
@@ -2454,7 +2454,7 @@
   {
     if ((pDrawable)->type == DRAWABLE_PIXMAP)
     {
-      miPolyRectangle(nxagentVirtualDrawable(pDrawable), pGC, nRectangles, pRectangles);
+      fbPolyRectangle(nxagentVirtualDrawable(pDrawable), pGC, nRectangles, pRectangles);
     }
 
     return;
