diff -Naur scidavis-0.2.3/scidavis/scidavis.pro scidavis-0.2.3.new/scidavis/scidavis.pro
--- scidavis-0.2.3/scidavis/scidavis.pro	2009-06-28 05:35:12.000000000 -0400
+++ scidavis-0.2.3.new/scidavis/scidavis.pro	2009-08-07 17:53:39.000000000 -0400
@@ -51,7 +51,7 @@
 ### scidavisrc.py and scidavisUtil.py, respectively. Alternatively you can also put scidavisrc.py 
 ### (or ".scidavis.py") into the users home directory. scidavisUtil.py must be either in the 
 ### directory specified here or somewhere else in the python path (sys.path) where "import" can find it
-unix: pythonconfig.path = /etc						# where scidavisrc.py is installed
+unix: pythonconfig.path = $$INSTALLBASE/share/scidavis						# where scidavisrc.py is installed
 win32: pythonconfig.path = $$INSTALLBASE    
 unix: pythonutils.path = $$INSTALLBASE/share/scidavis        # where the scidavisUtil python modules is installed
 win32: pythonutils.path = $$INSTALLBASE        
@@ -70,7 +70,8 @@
 ### Note to packagers: If you want to use systemwide installed liborigin
 ### instead of the one provided in "3rdparty", uncomment the following 2 lines:
 # CONFIG += dynamic_liborigin
-# LIBS += -lorigin
+#LIBS += -lorigin2
+#INCLUDEPATH  += /usr/include/liborigin2
 ### Unfortunately, due to liborigin being in alpha stage, we cannot promise
 ### that SciDAVis works with any other version that the one in "3rdparty".
 ################### end of liborigin block 
@@ -82,18 +83,17 @@
 ### are compiled against Qt4), dynamically against everything else.
 #############################################################################
 
-unix:INCLUDEPATH  += ../3rdparty/qwtplot3d/include
-unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
+unix:INCLUDEPATH  += /usr/include/qwtplot3d
+unix:LIBS         += -lqwtplot3d
 
-unix:INCLUDEPATH  += ../3rdparty/qwt/src
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+unix:INCLUDEPATH  += /usr/include/qwt5
+unix:LIBS         += -lqwt
 
-unix:LIBS         += -L/usr/lib$${libsuff}
-unix:LIBS         += -lgsl -lgslcblas -lz
-### muparser 1.30 does not compile as a shared lib on Linux
-unix:LIBS         += -L/usr/local/lib$${libsuff}
-unix:LIBS         += /usr/local/lib/libmuparser.a
-unix:INCLUDEPATH  += /usr/local/include
+unix:INCLUDEPATH  += $$system(pkg-config --cflags gsl)
+unix:LIBS         += $$system(pkg-config --libs gsl)
+unix:LIBS         += $$system(pkg-config --libs muparser)
+unix:INCLUDEPATH  += $$system(pkg-config --cflags muparser)
+unix:LIBS         += -lz
 
 #############################################################################
 ### Link everything dynamically
diff -Naur scidavis-0.2.3/scidavis/sourcefiles.pri scidavis-0.2.3.new/scidavis/sourcefiles.pri
--- scidavis-0.2.3/scidavis/sourcefiles.pri	2009-03-16 17:08:31.000000000 -0400
+++ scidavis-0.2.3.new/scidavis/sourcefiles.pri	2009-08-07 17:53:39.000000000 -0400
@@ -44,12 +44,8 @@
 ###################### DOCUMENTATION ########################################
 
 documentation.files += ../manual/html \
-                       ../INSTALL.html \
                        ../scidavis.css \
-                       ../scidavis-logo.png \
-                       ../README \
-                       ../CHANGES \
-                       ../gpl.txt 
+                       ../scidavis-logo.png
 
 ###################### DESKTOP INTEGRATION ##################################
 
