Index: vorbis-tools-1.0.1/configure.in
===================================================================
--- vorbis-tools-1.0.1.orig/configure.in
+++ vorbis-tools-1.0.1/configure.in
@@ -27,6 +27,8 @@ AM_PROG_LIBTOOL
 ALL_LINGUAS="be cs da es fr hr hu nl ro ru sv"
 AM_GNU_GETTEXT
 
+AC_SYS_LARGEFILE
+
 dnl --------------------------------------------------
 dnl Set build flags based on environment
 dnl --------------------------------------------------
