Index: configure.in
===================================================================
--- configure.in	(revision 2433)
+++ configure.in	(working copy)
@@ -284,16 +284,12 @@
 		AC_MSG_RESULT(yes)
 		CFLAGS="$CFLAGS `$PKG_CONFIG --cflags gtkspell-2.0`"
 		LIBS="$LIBS `$PKG_CONFIG --libs gtkspell-2.0`"
-		if $PKG_CONFIG --atleast-version 2.0.13 gtkspell-2.0 ; then
-			AC_DEFINE(USE_ENCHANT, 1, Use Enchant with GtkSpell)
+		AC_DEFINE(USE_ENCHANT, 1, Use Enchant with GtkSpell)
 
-			dnl Workaroud for zemberek crash
-			if $PKG_CONFIG --atleast-version 1.4.2 enchant ; then
-				CFLAGS="$CFLAGS `$PKG_CONFIG --cflags dbus-glib-1`"
-				LIBS="$LIBS `$PKG_CONFIG --libs dbus-glib-1`"
-			fi
-		elif $PKG_CONFIG --atleast-version 2.0.12 gtkspell-2.0 ; then
-			LIBS="$LIBS -laspell"
+		dnl Workaroud for zemberek crash
+		if $PKG_CONFIG --atleast-version 1.4.2 enchant ; then
+			CFLAGS="$CFLAGS `$PKG_CONFIG --cflags dbus-glib-1`"
+			LIBS="$LIBS `$PKG_CONFIG --libs dbus-glib-1`"
 		fi
 		AC_DEFINE(USE_GTKSPELL, 1, Use GtkSpell in editor)
 	else
