diff -ur jabberd-2.0s8.orig/tools/jabberd.in jabberd-2.0s8/tools/jabberd.in
--- jabberd-2.0s8.orig/tools/jabberd.in	2004-12-07 18:08:38.000000000 +0100
+++ jabberd-2.0s8/tools/jabberd.in	2005-06-14 00:01:29.000000000 +0200
@@ -14,7 +14,7 @@
 
 use strict;
 use Getopt::Std;
-use FindBin qw($Bin);
+#use FindBin qw($Bin);
 use FileHandle;
 use IPC::Open3;
 use IO::Select;
@@ -25,6 +25,7 @@
 #-----------------------------------------------------------------------------
 # Define some initial variables and default them as needed.
 #-----------------------------------------------------------------------------
+my $Bin = "@bindir@";
 my $VERSION = "@VERSION@";
 my $config_dir = "@sysconfdir@";
 my $config = $config_dir."/jabberd.cfg";
diff -ur jabberd-2.0s8.orig/tools/Makefile.am jabberd-2.0s8/tools/Makefile.am
--- jabberd-2.0s8.orig/tools/Makefile.am	2005-04-14 21:43:52.000000000 +0200
+++ jabberd-2.0s8/tools/Makefile.am	2005-06-14 00:01:29.000000000 +0200
@@ -3,7 +3,8 @@
 
 edit = sed \
 	-e 's,@sysconfdir\@,$(sysconfdir),g' \
-	-e 's,@VERSION\@,$(VERSION),g'
+	-e 's,@VERSION\@,$(VERSION),g' \
+	-e 's,@bindir\@,$(bindir),g'
 
 $(bin_SCRIPTS): $(EXTRA_DIST)
 	@echo "generating $@ from $@.in"; \
diff -ur jabberd-2.0s8.orig/tools/Makefile.in jabberd-2.0s8/tools/Makefile.in
--- jabberd-2.0s8.orig/tools/Makefile.in	2005-04-20 09:37:33.000000000 +0200
+++ jabberd-2.0s8/tools/Makefile.in	2005-06-14 00:02:27.000000000 +0200
@@ -152,7 +152,8 @@
 EXTRA_DIST = db-setup.mysql db-setup.pgsql jabberd.in jabberd.rc pipe-auth.pl migrate.pl db-update.mysql db-setup.oracle
 edit = sed \
 	-e 's,@sysconfdir\@,$(sysconfdir),g' \
-	-e 's,@VERSION\@,$(VERSION),g'
+	-e 's,@VERSION\@,$(VERSION),g' \
+	-e 's,@bindir\@,$(bindir),g'
 
 all: all-am
 
