diff -uNr hping2-rc2.orig/getlhs.c hping2-rc2/getlhs.c
--- hping2-rc2.orig/getlhs.c	2003-11-27 00:10:32.000000000 +0100
+++ hping2-rc2/getlhs.c	2003-11-27 00:11:37.000000000 +0100
@@ -88,6 +88,9 @@
 	} else if ( strstr(ifname, "eth") ) {
 		linkhdr_size = ETHHDR_SIZE;
 		return 0;
+	} else if ( strstr(ifname, "wlan") ) {
+		linkhdr_size = WLANHDR_SIZE;
+		return 0;
 	} else if (strstr(ifname, "ets")) {
 		linkhdr_size = 0;
 		return 0;
diff -uNr hping2-rc2.orig/hping2.h hping2-rc2/hping2.h
--- hping2-rc2.orig/hping2.h	2003-11-27 00:10:32.000000000 +0100
+++ hping2-rc2/hping2.h	2003-11-27 00:11:44.000000000 +0100
@@ -97,6 +97,7 @@
 #define PPPHDR_SIZE_NETBSD	4
 #define ETHHDR_SIZE		14
 #define LOHDR_SIZE		14
+#define WLANHDR_SIZE		14
 
 /* packet size (physical header size + ip header + tcp header + 0 data bytes) */
 #ifndef IP_MAX_SIZE
