--- elogv.py.orig	2008-02-04 10:14:58.000000000 +0100
+++ elogv.py	2008-02-04 10:15:48.000000000 +0100
@@ -129,12 +129,18 @@
             
             elif c == ord("a"):
                 self.sort_method[0] = "alphabetical"
+                self.init_screen()
+                self.show_log()
                 
             elif c == ord("t"):
                 self.sort_method[0] = "date"
+                self.init_screen()
+                self.show_log()
                 
             elif c == ord("r"):
                 self.sort_method[1] = not self.sort_method[1]
+                self.init_screen()
+                self.show_log()
                 
             elif c == ord(" "):
                 # Now is used only for scrolling the text
