--- cheesetracker-0.9.15.3.orig/common/os/path_data.cpp
+++ cheesetracker-0.9.15.3/common/os/path_data.cpp
@@ -27,6 +27,7 @@
 //
 #include "path_data.h"
 #include <stdio.h>
+#include <stdlib.h>
 
 string Path_Data::get_home_dir() {
 
--- cheesetracker-0.9.15.3.orig/common/drivers/posix/mutex_lock_pthreads.h
+++ cheesetracker-0.9.15.3/common/drivers/posix/mutex_lock_pthreads.h
@@ -39,6 +39,7 @@
 #include <pthread.h>
 #include <errno.h>
 #include <string.h>
+#include <stdlib.h>
 /**
   *@author Juan Linietsky
   */
--- cheesetracker-0.9.15.3.orig/common/drivers/posix/sound_driver_oss.cpp
+++ cheesetracker-0.9.15.3/common/drivers/posix/sound_driver_oss.cpp
@@ -38,6 +38,7 @@
 #include "sound_driver_oss.h"
 #include "math.h"
 #include "components/audio/sample_conversion.h"
+#include <stdlib.h>
 
 
 
--- cheesetracker-0.9.15.3.orig/common/components/file/file_access.cpp
+++ cheesetracker-0.9.15.3/common/components/file/file_access.cpp
@@ -14,6 +14,7 @@
  *                                                                         *
  ***************************************************************************/
 #include "file_access.h"
+#include <string.h>
 
 File_Manager::File_Manager() {
 
--- cheesetracker-0.9.15.3.orig/common/components/data/property_bridges.cpp
+++ cheesetracker-0.9.15.3/common/components/data/property_bridges.cpp
@@ -31,6 +31,7 @@
  ***************************************************************************/
 
 #include "property_bridges.h"
+#include <string.h>
 
 
 void Property_Bridge::set_display_preference(string p_pref) {
--- cheesetracker-0.9.15.3.orig/common/components/data/dds_packer.cpp
+++ cheesetracker-0.9.15.3/common/components/data/dds_packer.cpp
@@ -1,4 +1,5 @@
 #include "dds_packer.h"
+#include <string.h>
 
 
 void DDSPacker::pack(DDS* p_struct, data& p_data, Uint32 p_offset) {
--- cheesetracker-0.9.15.3.orig/common/components/data/dds.cpp
+++ cheesetracker-0.9.15.3/common/components/data/dds.cpp
@@ -1,4 +1,5 @@
 #include "dds.h"
+#include <string.h>
 
 using std::vector;
 
--- cheesetracker-0.9.15.3.orig/common/components/audio/mixer.cpp
+++ cheesetracker-0.9.15.3/common/components/audio/mixer.cpp
@@ -35,6 +35,7 @@
 
 #include "cycles.h"
 #include "components/midi/midi_controls.h"
+#include <string.h>
 
 #define CYCMSEC(m_parm) (float)( ((double)m_parm/950000000.0) * 1000.0 )
 
--- cheesetracker-0.9.15.3.orig/common/property_server/bridge_converter.cpp
+++ cheesetracker-0.9.15.3/common/property_server/bridge_converter.cpp
@@ -1,4 +1,5 @@
 #include "bridge_converter.h"
+#include <string.h>
 
 int BridgeConverter::get_short_string(BufferControl& p_buffer, string& p_string) {
 
--- cheesetracker-0.9.15.3.orig/common/plugins/effects/custom/effect_pshift.cpp
+++ cheesetracker-0.9.15.3/common/plugins/effects/custom/effect_pshift.cpp
@@ -11,6 +11,7 @@
 //
 #include "effect_pshift.h"
 #include "math.h"
+#include <string.h>
 
 
 void Effect_PShift::reset() {
--- cheesetracker-0.9.15.3.orig/common/plugins/effects/custom/effect_echo.cpp
+++ cheesetracker-0.9.15.3/common/plugins/effects/custom/effect_echo.cpp
@@ -10,6 +10,7 @@
 //
 //
 #include "effect_echo.h"
+#include <string.h>
 
 
 void Effect_Echo::reset() {
--- cheesetracker-0.9.15.3.orig/common/plugins/effects/custom/chorus_effect.cpp
+++ cheesetracker-0.9.15.3/common/plugins/effects/custom/chorus_effect.cpp
@@ -10,7 +10,7 @@
 //
 //
 #include "chorus_effect.h"
-
+#include <string.h>
 
 #define MIX_CALCULATE_INCREMENT_INTERVAL 8
 
--- cheesetracker-0.9.15.3.orig/common/plugins/effects/custom/effect_stereo_fx.cpp
+++ cheesetracker-0.9.15.3/common/plugins/effects/custom/effect_stereo_fx.cpp
@@ -10,6 +10,7 @@
 //
 //
 #include "effect_stereo_fx.h"
+#include <string.h>
 
 
 inline static double get_samples_from_msecs(double p_msecs,double p_mixfreq) {
--- cheesetracker-0.9.15.3.orig/common/plugins/effects/ladspa/ladspa_effect.cpp
+++ cheesetracker-0.9.15.3/common/plugins/effects/ladspa/ladspa_effect.cpp
@@ -17,6 +17,7 @@
 #include <sys/types.h>
 #include <dirent.h>
 #include <dlfcn.h>
+#include <string.h>
 
 
 
--- cheesetracker-0.9.15.3.orig/common/plugins/effects/ladspa/ladspa_effect.h
+++ cheesetracker-0.9.15.3/common/plugins/effects/ladspa/ladspa_effect.h
@@ -18,8 +18,9 @@
 
 #include "common/plugins/effects/ladspa/ladspa.h"
 #include "components/audio/effect.h"
+#include <string.h>
 #include <vector>
-	using std::vector;
+using std::vector;
 
 /**
 @author Juan Linietsky
--- cheesetracker-0.9.15.3.orig/cheesetracker/trackercore/file_writer.cpp
+++ cheesetracker-0.9.15.3/cheesetracker/trackercore/file_writer.cpp
@@ -31,6 +31,7 @@
  ***************************************************************************/
 
 #include "file_writer.h"
+#include <string.h>
 
 File_Writer::File_Writer(){
 
--- cheesetracker-0.9.15.3.orig/cheesetracker/trackercore/instrument.cpp
+++ cheesetracker-0.9.15.3/cheesetracker/trackercore/instrument.cpp
@@ -31,6 +31,7 @@
  ***************************************************************************/
 
 #include <cassert>
+#include <string.h>
 #include "instrument.h"
 
 void
--- cheesetracker-0.9.15.3.orig/cheesetracker/trackercore/voice_control.cpp
+++ cheesetracker-0.9.15.3/cheesetracker/trackercore/voice_control.cpp
@@ -16,6 +16,7 @@
 
 #include "voice_control.h"
 #include "freq_conversion.h"
+#include <stdlib.h>
 
 
 #define PAN_SURROUND 512
