diff -Narup gcc-3.3.orig/gcc/version.c gcc-3.3/gcc/version.c
--- gcc-3.3.orig/gcc/version.c	2003-05-13 23:26:03.000000000 +0200
+++ gcc-3.3/gcc/version.c	2003-05-15 17:51:13.000000000 +0200
@@ -6,7 +6,7 @@
    please modify this string to indicate that, e.g. by putting your
    organization's name in parentheses at the end of the string.  */
 
-const char version_string[] = "3.3 (Gentoo Linux 1.4 @PV@)";
+const char version_string[] = "3.3 (Gentoo Linux 1.4 @PV@, propolice)";
 
 /* This is the location of the online document giving instructions for
    reporting bugs.  If you distribute a modified version of GCC,
