diff -Naurp src.orig/runtime/objs/mk.x86-linux src/runtime/objs/mk.x86-linux
--- src.orig/runtime/objs/mk.x86-linux	2004-06-09 16:48:42.214121992 +0200
+++ src/runtime/objs/mk.x86-linux	2004-06-09 16:49:05.758542696 +0200
@@ -4,9 +4,9 @@
 SHELL =		/bin/sh
 
 MAKE =		make
-CC =		gcc -ansi
+CC =		gcc
 CFLAGS =	-O2
-CPP =		gcc -x c -E -P -ansi
+CPP =		gcc -x assembler-with-cpp -E -P
 #XOBJS =	xmonitor.o
 #LD_LIBS =	-lX11
 #BASE_DEFS =	-DHEAP_MONITOR
