--- ipw2200-1.0.8.orig/ipw2200.c	2005-11-16 22:53:19.000000000 +0800
+++ ipw2200-1.0.8/ipw2200.c	2005-11-16 23:20:09.000000000 +0800
@@ -1109,8 +1109,7 @@ static struct ipw_fw_error *ipw_alloc_er
 	error->elem_len = elem_len;
 	error->log_len = log_len;
 	error->elem = (struct ipw_error_elem *)error->payload;
-	error->log = (struct ipw_event *)(error->elem +
-					  (sizeof(*error->elem) * elem_len));
+	error->log = (struct ipw_event *)(error->elem + elem_len);
 
 	ipw_capture_event_log(priv, log_len, error->log);
 
