diff -urp ipw2200-1.0.6/ipw2200.c ipw2200-1.0.6-suspend2/ipw2200.c
--- ipw2200-1.0.6/ipw2200.c	2005-07-13 21:00:28.000000000 +0200
+++ ipw2200-1.0.6-suspend2/ipw2200.c	2005-08-29 18:18:05.000000000 +0200
@@ -10146,11 +10146,7 @@ static int ipw_setup_deferred_work(struc
 {
 	int ret = 0;
 
-#ifdef PF_SYNCTHREAD
-	priv->workqueue = create_workqueue(DRV_NAME, 0);
-#else
 	priv->workqueue = create_workqueue(DRV_NAME);
-#endif
 	init_waitqueue_head(&priv->wait_command_queue);
 	init_waitqueue_head(&priv->wait_state);
 
