diff -urp ipw2200-1.0.8/ipw2200.c ipw2200-1.0.8-txbusy/ipw2200.c
--- ipw2200-1.0.8/ipw2200.c	2005-10-21 05:35:24.000000000 +0800
+++ ipw2200-1.0.8-txbusy/ipw2200.c	2005-10-25 13:22:38.000000000 +0800
@@ -10222,11 +10222,6 @@ static inline int ipw_tx_skb(struct ipw_
 	u16 remaining_bytes;
 	int fc;
 
-	/* If there isn't room in the queue, we return busy and let the
-	 * network stack requeue the packet for us */
-	if (ipw_queue_space(q) < q->high_mark)
-		return NETDEV_TX_BUSY;
-
 	switch (priv->ieee->iw_mode) {
 	case IW_MODE_ADHOC:
 		hdr_len = IEEE80211_3ADDR_LEN;
@@ -10444,7 +10439,7 @@ static int ipw_net_hard_start_xmit(struc
 
       fail_unlock:
 	spin_unlock_irqrestore(&priv->lock, flags);
-	return 1;
+	return -1;
 }
 
 static struct net_device_stats *ipw_net_get_stats(struct net_device *dev)
