diff -urp ipw2200-1.0.6/ipw2200.c ipw2200-1.0.6-broadcast/ipw2200.c
--- ipw2200-1.0.6/ipw2200.c	2005-07-13 21:00:28.000000000 +0200
+++ ipw2200-1.0.6-broadcast/ipw2200.c	2005-09-02 19:39:01.000000000 +0200
@@ -7669,7 +7669,8 @@ static inline int is_network_packet(stru
 			return 0;
 
 		/* multicast packets to our IBSS go through */
-		if (is_multicast_ether_addr(header->addr1))
+		if (is_multicast_ether_addr(header->addr1) ||
+                    is_broadcast_ether_addr(header->addr1))
 			return !memcmp(header->addr3, priv->bssid, ETH_ALEN);
 
 		/* packets to our adapter go through */
@@ -7682,7 +7683,8 @@ static inline int is_network_packet(stru
 			return 0;
 
 		/* {broad,multi}cast packets to our IBSS go through */
-		if (is_multicast_ether_addr(header->addr1))
+		if (is_multicast_ether_addr(header->addr1) ||
+                    is_broadcast_ether_addr(header->addr1))
 			return !memcmp(header->addr2, priv->bssid, ETH_ALEN);
 
 		/* packets to our adapter go through */
