From ea8c4b0db31855cb144cd5b863a045423d4afbfc Mon Sep 17 00:00:00 2001
From: Jannis Pohlmann <jannis@xfce.org>
Date: Wed, 08 Sep 2010 10:04:05 +0000
Subject: Drop ck-launch-session as CK is handled in xfce4-session (bug #6684).

---
diff --git a/scripts/xinitrc.in.in b/scripts/xinitrc.in.in
index 2aaf1df..2bd43ec 100755
--- a/scripts/xinitrc.in.in
+++ b/scripts/xinitrc.in.in
@@ -196,11 +196,7 @@ fi
 
 # Run xfce4-session if installed
 if which xfce4-session >/dev/null 2>&1; then
-    if which ck-launch-session >/dev/null 2>&1; then
-      ck-launch-session xfce4-session
-    else
-      xfce4-session
-    fi
+    xfce4-session
 
     if test "$ssh_agent_kill_cmd"; then
         echo "running '$ssh_agent_kill_cmd'"
--
cgit v0.8.2.1
