diff -Naur poedit-1.3.9-orig/src/catalog.cpp poedit-1.3.9/src/catalog.cpp
--- poedit-1.3.9-orig/src/catalog.cpp	2007-11-09 04:35:59.000000000 -0600
+++ poedit-1.3.9/src/catalog.cpp	2007-12-23 12:11:46.000000000 -0600
@@ -1497,7 +1497,7 @@
                               wxArrayString& snew, wxArrayString& sobsolete)
 {
     wxASSERT( snew.empty() );
-    wxASSERT( sbsolete.empty() );
+    wxASSERT( sobsolete.empty() );
 
     std::set<wxString> strsThis, strsRef;
 
