diff -ru remote-0.3.8-orig/remote.c remote-0.3.8/remote.c
--- remote-0.3.8-orig/remote.c	2006-12-03 16:24:36.438027599 +0100
+++ remote-0.3.8/remote.c	2006-12-03 16:26:19.222559030 +0100
@@ -56,7 +56,7 @@
 
 
 // ---------------------------------------------------------------------------
-bool cRemoteGeneric::Put(uint64 Code, bool Repeat, bool Release)
+bool cRemoteGeneric::Put(uint64_t Code, bool Repeat, bool Release)
 // ---------------------------------------------------------------------------
 {
     return cRemote::Put(Code, Repeat, Release);
@@ -72,7 +72,7 @@
 #else
     cTimeMs first, rate, timeout;
 #endif
-    uint64 code, lastcode = INVALID_KEY;
+    uint64_t code, lastcode = INVALID_KEY;
     bool repeat = false;
 
     for (;;)
@@ -412,12 +412,12 @@
 
 
 // ---------------------------------------------------------------------------
-uint64 cRemoteDevInput::getKey(void)
+uint64_t cRemoteDevInput::getKey(void)
 // ---------------------------------------------------------------------------
 {
     struct input_event ev;
     int n;
-    uint64 code;
+    uint64_t code;
 
     do
         n = read(fh, &ev, sizeof ev);
@@ -427,7 +427,7 @@
     {
         if (ev.value)
             ev.value = 1;
-        code = ((uint64)ev.value << 32) | ((uint64)ev.type << 16) | (uint64)ev.code;
+        code = ((uint64_t)ev.value << 32) | ((uint64_t)ev.type << 16) | (uint64_t)ev.code;
     }
     else
         code = INVALID_KEY;
@@ -443,7 +443,7 @@
 
 
 // ---------------------------------------------------------------------------
-bool cRemoteDevInput::keyPressed(uint64 code)
+bool cRemoteDevInput::keyPressed(uint64_t code)
 // ---------------------------------------------------------------------------
 {
     return (code & 0xFFFF00000000ULL);
@@ -455,7 +455,7 @@
 
 #ifdef REMOTE_FEATURE_LIRCOLD
 // ---------------------------------------------------------------------------
-uint64 cRemoteDevLirc::getKey(void)
+uint64_t cRemoteDevLirc::getKey(void)
 // ---------------------------------------------------------------------------
 {
     unsigned long code;
@@ -465,12 +465,12 @@
     if (n != sizeof code)
         return INVALID_KEY;
     else
-        return (uint64)code;
+        return (uint64_t)code;
 }
 
 
 // ---------------------------------------------------------------------------
-bool cRemoteDevLirc::keyPressed(uint64 code)
+bool cRemoteDevLirc::keyPressed(uint64_t code)
 // ---------------------------------------------------------------------------
 {
     return (code & 0x80);
@@ -510,11 +510,11 @@
 
 
 // ---------------------------------------------------------------------------
-uint64 cRemoteDevTty::getKey(void)
+uint64_t cRemoteDevTty::getKey(void)
 // ---------------------------------------------------------------------------
 {
     int n;
-    uint64 code = 0;
+    uint64_t code = 0;
 
     n = read(fh, &code, sizeof code);
     return (n > 0) ? code : INVALID_KEY;
@@ -522,7 +522,7 @@
 
 
 // ---------------------------------------------------------------------------
-bool cRemoteDevTty::keyPressed(uint64 code)
+bool cRemoteDevTty::keyPressed(uint64_t code)
 // ---------------------------------------------------------------------------
 {
     return true;
@@ -530,7 +530,7 @@
 
 
 // ---------------------------------------------------------------------------
-bool cRemoteDevTty::Put(uint64 Code, bool Repeat, bool Release)
+bool cRemoteDevTty::Put(uint64_t Code, bool Repeat, bool Release)
 // ---------------------------------------------------------------------------
 {
     bool rc = cRemote::Put(Code, Repeat, Release);
diff -ru remote-0.3.8-orig/remote.h remote-0.3.8/remote.h
--- remote-0.3.8-orig/remote.h	2006-12-03 16:24:36.438027599 +0100
+++ remote-0.3.8/remote.h	2006-12-03 16:27:08.510333521 +0100
@@ -43,7 +43,7 @@
 /*****************************************************************************/
 {
 protected:
-  static const uint64 INVALID_KEY = (uint64) -1;
+  static const uint64_t INVALID_KEY = (uint64_t) -1;
   int fh;
   char *device;
   int polldelay;
@@ -52,9 +52,9 @@
   int repeattimeout;
   cRemoteGeneric(const char *name, int f, char *d);
   virtual ~cRemoteGeneric();
-  virtual uint64 getKey(void) = 0;
-  virtual bool keyPressed(uint64 code) = 0;
-  virtual bool Put(uint64 Code, bool Repeat = false, bool Release = false);
+  virtual uint64_t getKey(void) = 0;
+  virtual bool keyPressed(uint64_t code) = 0;
+  virtual bool Put(uint64_t Code, bool Repeat = false, bool Release = false);
   virtual void Action(void);
 };
 
@@ -66,11 +66,11 @@
 {
 private:
   bool testMode;
-  uint64 testKey;
+  uint64_t testKey;
   bool loadKeymap(const char *devname, uint32_t options);
 protected:
-  virtual uint64 getKey(void);
-  virtual bool keyPressed(uint64 code);
+  virtual uint64_t getKey(void);
+  virtual bool keyPressed(uint64_t code);
 public:
   cRemoteDevInput(const char *name, int f, char *d);
   virtual bool Initialize(void);
@@ -84,8 +84,8 @@
 /*****************************************************************************/
 {
 protected:
-  virtual uint64 getKey(void);
-  virtual bool keyPressed(uint64 code);
+  virtual uint64_t getKey(void);
+  virtual bool keyPressed(uint64_t code);
 public:
   cRemoteDevLirc(const char *name, int f, char *d)
     :cRemoteGeneric(name, f, d) { Start(); }
@@ -101,9 +101,9 @@
 private:
   struct termios tm;
 protected:
-  virtual uint64 getKey(void);
-  virtual bool keyPressed(uint64 code);
-  virtual bool Put(uint64 Code, bool Repeat = false, bool Release = false);
+  virtual uint64_t getKey(void);
+  virtual bool keyPressed(uint64_t code);
+  virtual bool Put(uint64_t Code, bool Repeat = false, bool Release = false);
 public:
   cRemoteDevTty(const char *name, int f, char *d);
   virtual ~cRemoteDevTty(); 
diff -ru remote-0.3.8-orig/remotetcp.c remote-0.3.8/remotetcp.c
--- remote-0.3.8-orig/remotetcp.c	2006-12-03 16:24:36.438027599 +0100
+++ remote-0.3.8/remotetcp.c	2006-12-03 16:25:09.357882797 +0100
@@ -36,7 +36,7 @@
 }		  
 
 
-uint64 cTcpRemote::getKey(void)
+uint64_t cTcpRemote::getKey(void)
 {
     if (csock == NULL)
     {
@@ -80,7 +80,7 @@
     }
 
 
-    uint64 key = cRemoteDevTty::getKey();
+    uint64_t key = cRemoteDevTty::getKey();
 
     if (key == INVALID_KEY)
     {
diff -ru remote-0.3.8-orig/remotetcp.h remote-0.3.8/remotetcp.h
--- remote-0.3.8-orig/remotetcp.h	2006-12-03 16:24:36.438027599 +0100
+++ remote-0.3.8/remotetcp.h	2006-12-03 16:24:52.237797789 +0100
@@ -25,7 +25,7 @@
   cSocket *csock;
 
 protected:
-  virtual uint64 getKey(void);
+  virtual uint64_t getKey(void);
 
 public:
   cTcpRemote(const char *name, int f, char *d);
