diff -ru src-old/graphics/objects.h src/graphics/objects.h
--- src-old/graphics/objects.h	2005-01-04 16:21:49.000000000 +0100
+++ src/graphics/objects.h	2005-01-04 12:26:57.000000000 +0100
@@ -177,7 +177,7 @@
 {
 	protected:
 	
-	friend obj_loc_data * dummy_object::GetLocDataRW(void) const;
+	friend class dummy_object;
 	
 	obj_loc_data * data;
 	
diff -ru src-old/target3/t3views.cpp src/target3/t3views.cpp
--- src-old/target3/t3views.cpp	2002-10-23 11:01:09.000000000 +0200
+++ src/target3/t3views.cpp	2005-01-04 16:22:26.000000000 +0100
@@ -7,6 +7,7 @@
 
 /*################################################################################################*/
 
+#include <libintl.h>
 #include "t3views.h"
 
 #include "t3main.h"
diff -ru src-old/target3/t3main.cpp src/target3/t3main.cpp
--- src-old/target3/t3main.cpp	2002-11-20 16:49:39.000000000 +0100
+++ src/target3/t3main.cpp	2005-01-04 16:46:44.762693881 +0100
@@ -7,6 +7,7 @@
 
 /*################################################################################################*/
 
+#include <libintl.h>
 #include "t3main.h"
 
 #include "t3errutil.h"
