diff -urN gnomemm-1.2.4.orig/src/gnome--/app-helper.h gnomemm-1.2.4/src/gnome--/app-helper.h
--- gnomemm-1.2.4.orig/src/gnome--/app-helper.h	2006-06-03 08:55:29.000000000 +0200
+++ gnomemm-1.2.4/src/gnome--/app-helper.h	2006-06-03 08:56:00.000000000 +0200
@@ -76,7 +76,7 @@
   ~Info();
   Info& operator =(const Info&);
 
-  Gtk::Widget* Info::get_widget() const;
+  Gtk::Widget* get_widget() const;
 
   Type type() const
     { return Type(GnomeUIInfo::type); }
diff -urN gnomemm-1.2.4.orig/src/pixmapmenuitem.gen_h gnomemm-1.2.4/src/pixmapmenuitem.gen_h
--- gnomemm-1.2.4.orig/src/pixmapmenuitem.gen_h	2006-06-03 08:55:29.000000000 +0200
+++ gnomemm-1.2.4/src/pixmapmenuitem.gen_h	2006-06-03 09:16:16.000000000 +0200
@@ -136,7 +136,7 @@
   gtk_pixmap_menu_item_set_pixmap(gtkobj(), GTK_WIDGET(pixmap.gtkobj()));
 }
 
-namespace Gnome {
+namespace Gtk {
 
 namespace Menu_Helpers
 {
