diff -rN -U3 l7-filter-userspace-0.8.old/l7-conntrack.cpp l7-filter-userspace-0.8/l7-conntrack.cpp
--- l7-filter-userspace-0.8.old/l7-conntrack.cpp	2008-10-26 13:47:53.000000000 +0100
+++ l7-filter-userspace-0.8/l7-conntrack.cpp	2008-10-26 13:48:43.000000000 +0100
@@ -24,6 +24,7 @@
 #include <cstring>
 
 extern "C" {
+#include <netinet/in.h>
 #include <linux/netfilter.h>
 #include <libnetfilter_conntrack/libnetfilter_conntrack.h>
 }
diff -rN -U3 l7-filter-userspace-0.8.old/l7-filter.cpp l7-filter-userspace-0.8/l7-filter.cpp
--- l7-filter-userspace-0.8.old/l7-filter.cpp	2008-10-26 13:47:53.000000000 +0100
+++ l7-filter-userspace-0.8/l7-filter.cpp	2008-10-26 13:48:49.000000000 +0100
@@ -39,6 +39,7 @@
 #include "config.h"
 
 extern "C" {
+#include <netinet/in.h>
 #include <linux/netfilter.h>
 #include <libnetfilter_conntrack/libnetfilter_conntrack.h>
 #include <libnetfilter_queue/libnetfilter_queue.h>
