--- txt2man-1.4.8.orig/txt2man
+++ txt2man-1.4.8/txt2man
@@ -181,6 +181,7 @@
 { 
 	sub(/\.\.\./, "\\.\\.\\.")	# to avoid some side effects in regexp
 	sub(/^ +$/,"") 			# remove spaces in empty lines
+	sub(/-/,"\\-")                  # backslash-escape hyphens
 }
 /^[[:upper:][:space:]]+$/ {		# Section header
 	if ((in_bd + 0) == 1) {
--- txt2man-1.4.8.orig/Makefile
+++ txt2man-1.4.8/Makefile
@@ -6,7 +6,7 @@
 	cp -p txt2man.1 $(prefix)/man/man1
 
 txt2man.1 txt2man.man: txt2man
-	./txt2man -h 2>&1 | ./txt2man -t txt2man >$@
+	./txt2man -h 2>&1 | ./txt2man -s 1 -t txt2man >$@
 
 txt2man.ps: txt2man.man
 	groff -man txt2man.man > txt2man.ps
