--- ./kwave/kwave.desktop.orig	2008-07-14 21:53:59.000000000 +0200
+++ ./kwave/kwave.desktop	2008-07-14 22:03:32.000000000 +0200
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Encoding=UTF-8
 Version=1.0
 Type=Application
 Name=Kwave Sound Editor
@@ -10,6 +9,6 @@
 Comment[fr]=Un éditeur de sons pour KDE
 TryExec=kwave
 Exec=kwave %F
-Icon=kwave.png
-MimeType=audio/wav;audio/x-wav;audio/basic;audio/x-mp1;audio/x-mp2;audio/x-mp3;audio/x-ogg;application/ogg;audio/x-aifc;audio/x-aiff;audio/x-ircam
-Terminal=0
+Icon=kwave
+MimeType=audio/wav;audio/x-wav;audio/basic;audio/x-mp1;audio/x-mp2;audio/x-mp3;audio/x-ogg;application/ogg;audio/x-aifc;audio/x-aiff;audio/x-ircam;
+Terminal=false
