--- session.c.dist	2003-08-02 10:36:27.000000000 -0400
+++ session.c	2003-08-02 10:34:10.000000000 -0400
@@ -36,14 +36,13 @@
 #include <gnome.h>
 #include <stdio.h>
 #include <fcntl.h>
+#include <errno.h>
 #include "gnewspost.h"
 #include "prefs.h"
 #include "app.h"
 #include "profile.h"
 #include "session.h"
 
-extern int errno;
-
 static void      _ok_button_clicked_cb(GtkWidget *widget, gpointer data);
 static gboolean  _session_dlg_close_cb(GtkWidget *widget, gpointer data);
 static void      _zero_file_tb_clicked_cb(GtkWidget *widget, gpointer data);
@@ -437,7 +436,7 @@
 static gchar  *
 _make_zero_file(void)
 {
-	gchar *fname, *txt;
+	gchar *fname = "tmpgnewspostXXXXXX", *txt;
 	int  fd;
 
 	if (!gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(w_zero_file_enabled_tb)))
@@ -447,8 +446,8 @@
 	if (!strcmp(txt, ""))
 		return NULL;
 	
-	fname = g_strdup(tmpnam(NULL));
-	if ((fd = open(fname, O_WRONLY|O_CREAT)) == -1) {
+	fd = mkstemp(fname);
+	if( fd == -1 ) {
 		g_warning("Can't make zero-file: %s", strerror(errno));
 		g_free(fname);
 		return NULL;
