diff -Naurp mon-0.99.2.orig/mon mon-0.99.2/mon
--- mon-0.99.2.orig/mon	2006-08-12 08:16:46.000000000 -0400
+++ mon-0.99.2/mon	2006-08-12 08:18:08.000000000 -0400
@@ -2879,12 +2879,13 @@ debug (1, "PID $p ($runningpid{$p}) exit
 	    $sref->{"_last_detail"} = $detail;
 
 	    #
-	    # reset the alertevery timer
+	    # reset the alertevery timer, and alertafter interval timer
 	    #
 	    foreach my $period (keys %{$sref->{"periods"}})
 	    {
 		$sref->{"periods"}->{$period}->{"_last_alert"} = 0;
 		$sref->{"periods"}->{$period}->{"_alert_sent"} = 0;
+        $sref->{"periods"}->{$period}->{"_1stfailtime"} = 0;
 	    }
 
 	    #
