diff -ur lisp-cgi-utils-0.7.orig/doc/Makefile lisp-cgi-utils-0.7/doc/Makefile
--- lisp-cgi-utils-0.7.orig/doc/Makefile	2005-05-17 16:38:20.000000000 -0500
+++ lisp-cgi-utils-0.7/doc/Makefile	2005-08-05 10:16:39.000000000 -0500
@@ -12,7 +12,7 @@
 examples: examples.dvi examples.ps examples.pdf examples.html examples.txt
 
 examples.ps: examples.dvi
-	dvips examples.dvi
+	dvips -o examples.ps examples.dvi
 
 # LaTeX runs are usually done twice for crossreferences
 
@@ -34,7 +34,7 @@
 cookies: cookies.dvi cookies.ps cookies.pdf cookies.html cookies.txt
 
 cookies.ps: cookies.dvi
-	dvips cookies.dvi
+	dvips -o cookies.ps cookies.dvi
 
 # LaTeX runs are usually done twice for crossreferences
 
Only in lisp-cgi-utils-0.7/doc: Makefile~
