diff -ur --exclude doc --exclude '*.fasl' --exclude '*~' lisp-cgi-utils-0.7.orig/html.lisp lisp-cgi-utils-0.7/html.lisp
--- lisp-cgi-utils-0.7.orig/html.lisp	2005-05-17 16:38:20.000000000 -0500
+++ lisp-cgi-utils-0.7/html.lisp	2005-08-05 10:28:04.000000000 -0500
@@ -60,7 +60,7 @@
 (proclaim '(optimize (safety 3) (speed 3))) ;
 
 
-(defconstant +html-header+ 
+(cl+:defconst +html-header+ 
   "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\">
 <html>"
   "default HTML header")
diff -ur --exclude doc --exclude '*.fasl' --exclude '*~' lisp-cgi-utils-0.7.orig/http.lisp lisp-cgi-utils-0.7/http.lisp
--- lisp-cgi-utils-0.7.orig/http.lisp	2005-05-17 16:38:20.000000000 -0500
+++ lisp-cgi-utils-0.7/http.lisp	2005-08-05 10:28:31.000000000 -0500
@@ -113,10 +113,10 @@
 (defvar *cookie-name-xlator* (make-hash-table :test #'equal)
   "translator for case insensitive cookie lookup (according to RFC2109)")
 
-(defconstant  +hex-char-digits+ "0123456789ABCDEF" "valid digits of 
+(cl+:defconst  +hex-char-digits+ "0123456789ABCDEF" "valid digits of 
 hexadecimal numbers, inorder")
 
-(defconstant +url-encode-ok-chars+ '( #\. #\/) 
+(cl+:defconst +url-encode-ok-chars+ '( #\. #\/) 
   "Characters that are ok in URL-encoding and need not be encoded.
    Note: alphanumerics are automagically assumed to not need encoding.")
 
diff -ur --exclude doc --exclude '*.fasl' --exclude '*~' lisp-cgi-utils-0.7.orig/lisp-cgi-utils.asd lisp-cgi-utils-0.7/lisp-cgi-utils.asd
--- lisp-cgi-utils-0.7.orig/lisp-cgi-utils.asd	2005-05-17 16:38:20.000000000 -0500
+++ lisp-cgi-utils-0.7/lisp-cgi-utils.asd	2005-08-05 10:28:55.000000000 -0500
@@ -25,6 +25,7 @@
     :long-description "THis package contains code to support writing CGI applications in Common lisp."
     
     :components ((:file "http")
-                 (:file "html")))
+                 (:file "html"))
+    :depends-on (#:cl-plus))
 
 
