--- configure.orig	2004-10-16 12:23:11.798429288 -0500
+++ configure	2004-10-16 12:27:04.419065576 -0500
@@ -1590,11 +1590,24 @@
 	fi
     done
 }
-CheckForLibrary strftime -lc || {
+
+CheckForLibrary_strftime()
+{
+    f=$1; shift
+    libs="$@";
+    cat>t.c<<EOF
+int t() { $f(0, 0, 0, 0); return 0; }
+int main(){ t(); return 0; }
+EOF
+    capture cat t.c
+    runMake t "t:; \${CCF} t.c $libs"
+}
+
+CheckForLibrary_strftime strftime -lc || {
     #
     # SCO has strftime in -lintl.
     #
-    if CheckForLibrary strftime -lintl; then
+    if CheckForLibrary_strftime strftime -lintl; then
 	Note "Looks like -lintl is needed for strftime."
 	MACHDEPLIBS="$MACHDEPLIBS -lintl"
     else
