--- hylafax-4.2.0.orig/configure	Wed Aug 18 08:22:54 2004
+++ hylafax-4.2.0/configure	Wed Aug 25 07:41:00 2004
@@ -2355,7 +2355,7 @@
 	if [ ${header_ver} -ge 19960307 ]; then
 	    if [ "${lib_ver}" = "3.4" ]; then
 		tiff_runlen_t="uint16"
-	    elif [ "${lib_ver}" = "3.5" -o "${lib_ver}" = "3.6" ]; then
+	    elif [ "${lib_ver}" = "3.5" -o "${lib_ver}" = "3.6" -o "${lib_ver}" = "3.7" ]; then
 		tiff_runlen_t="uint32"
 	    fi
 	fi
@@ -2392,9 +2392,9 @@
 
 Incompatible TIFF Library.
 
-HylaFAX ${VERSION} requires TIFF software distribution ver 3.4, 3.5, or
-3.6.  If you do not have up to date TIFF software on your system then
-you can retrieve it from the location where you obtained this software.
+HylaFAX ${VERSION} requires TIFF software distribution ver 3.4, 3.5, 
+3.6, or 3.7.  If you do not have up to date TIFF software on your system
+then you can retrieve it from the location where you obtained this software.
 The Home Page for version 3.5 and later is http://www.libtiff.org
 EOF
 	boom

