--- faxmail/faxmail.c++.orig	2005-08-09 16:40:21.000000000 +0200
+++ faxmail/faxmail.c++	2005-08-09 16:40:41.000000000 +0200
@@ -392,8 +392,9 @@
 	if (verbose)
 	    mime.trace(stderr);
 	// XXX anything but us-ascii is treated as ISO-8859-1
-	setISO8859(mime.getCharset() != CS_USASCII);
-
+	// setISO8859(mime.getCharset() != CS_USASCII);
+	setISO8859(true);
+	
 	/*
 	 * Check first for any external script/command to
 	 * use in converting the body part to PostScript.
