diff -ru hylafax-4.2.1/util/faxcron.sh.in hylafax-4.2.1.new/util/faxcron.sh.in
--- hylafax-4.2.1/util/faxcron.sh.in	2004-12-22 14:44:06.000000000 +0100
+++ hylafax-4.2.1.new/util/faxcron.sh.in	2005-08-06 12:24:16.000000000 +0200
@@ -69,12 +69,18 @@
 UPDATE="date +'%D %H:%M' >$LAST"
 
 # security
-TMPDIR=/tmp/.faxcron.sh$$
-rm -rf $TMPDIR
-mkdir $TMPDIR || exit 1
+if test -n "`type -p mktemp`" ; then
+        TMPFDIR=`mktemp -d -t xferfax.XXXXXX`  || { echo "$0: Cannot create temporary dir!" >&2 ; exit 1; }
+else
+	TMPFDIR=${TMPDIR-/tmp}/.faxcron.sh$$
+	rm -rf $TMPFDIR
+	mkdir $TMPFDIR || { echo "$0: Cannot create temporary dir! Aborting." ; exit 1; }
+fi
+
+JUNK=$TMPFDIR/faxjunk$$         # temp file used multiple times
+AWKTMP=$TMPFDIR/faxawk$$                # temp file for awk program
 
-JUNK=$TMPDIR/faxjunk$$         # temp file used multiple times
-AWKTMP=$TMPDIR/faxawk$$                # temp file for awk program
+trap "$RM \$AWKTMP \$JUNK; [ -d $TMPFDIR ] && $RM -rf $TMPFDIR" 0 1 2 15
 
 while [ x"$1" != x"" ] ; do
     case $1 in
@@ -90,7 +96,6 @@
     shift
 done
 
-trap "$RM \$AWKTMP \$JUNK; $RM -rf $TMPDIR; exit 1" 0 1 2 15
 
 
 test -z "$LASTRUN" && LASTRUN=`$CAT $LAST 2>/dev/null`
@@ -285,7 +290,7 @@
 EOF
 $AWK -f $AWKTMP -v LASTRUN="$LASTRUN" TRANSCRIPT="\
     LOGFILE=log/%s;\
-    TMP=$TMPDIR/faxlog\$\$;\
+    TMP=$TMPFDIR/faxlog\$\$;\
     if [ -f \$LOGFILE ]; then\
 	$SED -n -e '/%s %s %s.*SESSION BEGIN/,/SESSION END/p' \$LOGFILE |\
 	$SED -e '/start.*timer/d'\
diff -ru hylafax-4.2.1/util/recvstats.sh.in hylafax-4.2.1.new/util/recvstats.sh.in
--- hylafax-4.2.1/util/recvstats.sh.in	2004-06-18 06:10:29.000000000 +0200
+++ hylafax-4.2.1.new/util/recvstats.sh.in	2005-08-06 12:25:17.000000000 +0200
@@ -153,15 +153,16 @@
 
 #
 # Generate an awk program to process the statistics file.
-#
-tmpAwk=/tmp/xferfax$$
-trap "rm -f $tmpAwk; exit 1" 0 1 2 15
-
-# security
-rm -rf $tmpAwk
-${NOCLOBBER_ON}
-> $tmpAwk || exit 1
-${NOCLOBBER_OFF}
+if test -n "`type -p mktemp`" ; then
+        tmpAwk=`mktemp -t xferfax.XXXXXX`  || { echo "$0: Cannot create temporary file!" >&2 ; exit 1; }
+else
+        tmpAwk=${TMPDIR-/tmp}/xferfax$$
+        rm -f $tmpAwk
+        ${NOCLOBBER_ON}
+        > $tmpAwk || { echo "$0: Cannot create temporary file! Aborting." ; exit 1; }
+        ${NOCLOBBER_OFF}
+fi
+trap "[ -f $tmpAwk ] && rm -f $tmpAwk" 0 1 2 15
 
 ($CAT<<'EOF'
 #
diff -ru hylafax-4.2.1/util/xferfaxstats.sh.in hylafax-4.2.1.new/util/xferfaxstats.sh.in
--- hylafax-4.2.1/util/xferfaxstats.sh.in	2004-06-18 06:10:29.000000000 +0200
+++ hylafax-4.2.1.new/util/xferfaxstats.sh.in	2005-08-06 12:25:28.000000000 +0200
@@ -26,6 +26,9 @@
 # LIABILITY, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE 
 # OF THIS SOFTWARE.
 #
+# security
+NOCLOBBER_ON=@NOCLOBBER_ON@
+NOCLOBBER_OFF=@NOCLOBBER_OFF@
 
 #
 # Print Statistics about Transmitted Facsimile.
@@ -169,8 +172,17 @@
 #
 # Generate an awk program to process the statistics file.
 #
-tmpAwk=/tmp/xferfax$$
-trap "rm -f $tmpAwk; exit 1" 0 1 2 15
+if test -n "`type -p mktemp`" ; then
+        tmpAwk=`mktemp -t xferfax.XXXXXX`  || { echo "$0: Cannot create temporary file!" >&2 ; exit 1; }
+else
+	tmpAwk=${TMPDIR-/tmp}/xferfax$$
+	rm -f $tmpAwk
+	${NOCLOBBER_ON}
+	> $tmpAwk || { echo "$0: Cannot create temporary file! Aborting." ; exit 1; }
+	${NOCLOBBER_OFF}
+fi
+trap "[ -f $tmpAwk ] && rm -f $tmpAwk" 0 1 2 15
+
 
 ($CAT<<'EOF'
 #
