--- configure.ac.orig	2010-04-20 20:55:13.775539738 +0200
+++ configure.ac	2010-04-20 20:57:56.872533578 +0200
@@ -973,7 +973,12 @@
 	if test "x$libetpan_result" = "xyes"; then
 	   LIBETPAN_CPPFLAGS="`$libetpanconfig --cflags`"
 	   LIBETPAN_LIBS="`$libetpanconfig --libs`"
+	   LIBETPAN_STABLE=`$libetpanconfig --version | grep -v ^0`
 	   LIBETPAN_VERSION=`$libetpanconfig --version | sed "s/\.//g" | sed "s/-.*$//"`
+	   if test x"$LIBETPAN_STABLE" != "x"; then
+              LIBETPAN_VERSION="100"
+           fi
+
 	   if test "$LIBETPAN_VERSION" -lt "057"; then
 		AC_MSG_RESULT([*** Claws Mail requires libetpan 0.57 or newer. See http://www.etpan.org/])
 		AC_MSG_RESULT([*** You can use --disable-libetpan if you don't need IMAP4 and/or NNTP support.])
