--- configure.in.orig	2005-07-18 18:01:26.000000000 +0100
+++ configure.in	2005-07-18 18:03:31.000000000 +0100
@@ -13,7 +13,7 @@
 AC_PROG_MAKE_SET
 
 ETCSEARCHPATH=${prefix}/etc:/usr/etc:/usr/local/etc
-LIBSEARCHPATH=${prefix}/lib:/usr/lib:/usr/local/lib:${prefix}/libexec:/usr/libexec:/usr/local/libexec
+LIBSEARCHPATH=${libdir}:${prefix}/lib:/usr/lib:/usr/local/lib:${prefix}/libexec:/usr/libexec:/usr/local/libexec
 
 dnl Get the operating system and version number...
 uname=`uname`
@@ -38,19 +38,19 @@
 AC_PATH_DIR(PRINTCAP,printcap,/etc/printcap,$ETCSEARCHPATH)
 
 dnl Check for /usr/lib/cups
-AC_PATH_DIR(CUPS,cups,/usr/lib/cups,$LIBSEARCHPATH)
+AC_PATH_DIR(CUPS,cups,${libdir}/cups,$LIBSEARCHPATH)
 
 dnl Check for /usr/lib/cups/filter
-AC_PATH_DIR(CUPS_FILTERS,cups/filter,/usr/lib/cups/filter,$LIBSEARCHPATH)
+AC_PATH_DIR(CUPS_FILTERS,cups/filter,${libdir}/cups/filter,$LIBSEARCHPATH)
 
 dnl Check for /usr/lib/ppr
-AC_PATH_DIR(PPR,ppr,/usr/lib/ppr,$LIBSEARCHPATH)
+AC_PATH_DIR(PPR,ppr,${libdir}/ppr,$LIBSEARCHPATH)
 
 dnl Check for /usr/lib/ppr/interfaces
-AC_PATH_DIR(PPR_INTERFACES,ppr/interfaces,/usr/lib/ppr/interfaces,$LIBSEARCHPATH)
+AC_PATH_DIR(PPR_INTERFACES,ppr/interfaces,${libdir}/ppr/interfaces,$LIBSEARCHPATH)
 
 dnl Check for /usr/lib/ppr/lib
-AC_PATH_DIR(PPR_LIB,ppr/lib,/usr/lib/ppr/lib,$LIBSEARCHPATH)
+AC_PATH_DIR(PPR_LIB,ppr/lib,${libdir}/ppr/lib,$LIBSEARCHPATH)
 
 AC_ARG_ENABLE(file-converter-check, [  --enable-file-converter-check  check whether a file converter is installed [default=yes]],
 [if eval "test x${enableval} = xno"; then
