fixes from upstream cvs

http://bugs.gentoo.org/162789

--- strace/mem.c	2007/07/24 01:52:58	1.40
+++ strace/mem.c	2007/09/12 01:26:29	1.41
@@ -544,7 +544,7 @@
 	return 0;
 }
 
-#if defined(ALPHA) || defined(FREEBSD) || defined(IA64) || defined(SUNOS4) || defined(SVR4)
+#if defined(ALPHA) || defined(FREEBSD) || defined(IA64) || defined(SUNOS4) || defined(SVR4) || defined(SPARC) || defined(SPARC64)
 int
 sys_getpagesize(tcp)
 struct tcb *tcp;
--- strace/linux/sparc/syscall.h	2006/01/12 22:34:50	1.11
+++ strace/linux/sparc/syscall.h	2007/09/12 01:26:24	1.12
@@ -107,6 +107,7 @@
 int sys_rt_sigtimedwait(), sys_prctl(), sys_poll();
 int sys_sendfile(), sys_query_module(), sys_capget(), sys_capset();
 int sys_create_module(), sys_init_module();
+int sys_setgroups32(), sys_getgroups32();
 
 int	sys_umask();		/* XXX */
 
@@ -203,4 +204,14 @@
 int	sys_clock_getres(), sys_clock_nanosleep();
 int	sys_timer_create(), sys_timer_settime(), sys_timer_gettime();
 
+int	sys_io_setup(), sys_io_destroy(), sys_io_submit(), sys_io_cancel(), sys_io_getevents();
+int	sys_mq_open(), sys_mq_unlink(), sys_mq_timedsend(), sys_mq_timedreceive(), sys_mq_notify(), sys_mq_getsetattr();
+int	sys_waitid();
+int	sys_openat(), sys_mkdirat(), sys_mknodat(), sys_fchownat(), sys_futimesat(), sys_newfstatat(), sys_unlinkat(), sys_renameat(), sys_linkat(), sys_symlinkat(), sys_readlinkat(), sys_fchmodat(),	sys_faccessat();
+int	sys_pselect6(), sys_ppoll();
+int	sys_unshare();
+int	sys_move_pages(), sys_getcpu();
+int	sys_epoll_pwait();
+int	sys_signalfd(), sys_timerfd(), sys_eventfd();
+
 #include "syscall1.h"
