--- GNUmakefile.in.orig	Tue Dec 18 19:36:17 2001
+++ GNUmakefile.in	Tue Dec 18 19:40:09 2001
@@ -455,21 +455,21 @@
 		-e 'chmod(oct($(PERM_RW))), $$m{$$_} or warn "chmod $(PERM_RW) $$m{$$_}: $$!\n";}' \
 		$(perldir)/AbiWord.pm $(perldir)/blib/man3/AbiWord.3pm
 	@perl -Iblib/lib -I/usr/lib/perl5/5.6.0 -MExtUtils::Install -e "install({@ARGV},'1',0,'0');" \
-		read /usr/lib/perl5/site_perl/5.6.0/i386-linux/auto/AbiWord/.packlist \
-		write /usr/lib/perl5/site_perl/5.6.0/i386-linux/auto/AbiWord/.packlist \
-		blib/lib /usr/lib/perl5/site_perl/5.6.0 \
-		blib/arch /usr/lib/perl5/site_perl/5.6.0/i386-linux \
-		blib/bin /usr/bin \
-		blib/script /usr/bin \
-		blib/man1 /usr/share/man/man1 \
-		blib/man3 /usr/share/man/man3
+		read $(D)/usr/lib/perl5/site_perl/5.6.0/i386-linux/auto/AbiWord/.packlist \
+		write $(D)/usr/lib/perl5/site_perl/5.6.0/i386-linux/auto/AbiWord/.packlist \
+		blib/lib $(D)/usr/lib/perl5/site_perl/5.6.0 \
+		blib/arch $(D)/usr/lib/perl5/site_perl/5.6.0/i386-linux \
+		blib/bin $(D)/usr/bin \
+		blib/script $(D)/usr/bin \
+		blib/man1 $(D)/usr/share/man/man1 \
+		blib/man3 $(D)/usr/share/man/man3
 	@perl -we 'exit unless -f $$ARGV[0];' \
 		-e 'print "WARNING: I have found an old package in\n";' \
 		-e 'print "\t$$ARGV[0].\n";' \
 		-e 'print "Please make sure the two installations are not conflicting\n";' \
-		/usr/lib/perl5/5.6.0/i386-linux/auto/AbiWord
+		$(D)/usr/lib/perl5/5.6.0/i386-linux/auto/AbiWord
 	@perl -I/usr/lib/perl5/5.6.0/i386-linux -I/usr/lib/perl5/5.6.0 -MExtUtils::Command \
-		-e mkpath /usr/lib/perl5/5.6.0/i386-linux
+		-e mkpath $(D)/usr/lib/perl5/5.6.0/i386-linux
 	@perl -e '$$\="\n\n";' -e 'print "=head2 ", scalar(localtime), ": C<", shift, ">", " L<", shift, ">";' \
 		-e 'print "=over 4";' \
 		-e 'while (defined($$key = shift) and defined($$val = shift)){print "=item *";print "C<$$key: $$val>";}' \
@@ -479,7 +479,7 @@
 		LINKTYPE "dynamic" \
 		VERSION "0.01" \
 		EXE_FILES "" \
-		>> /usr/lib/perl5/5.6.0/i386-linux/perllocal.pod
+		>> $(D)/usr/lib/perl5/5.6.0/i386-linux/perllocal.pod.new
 	@echo Appending installation info to /usr/lib/perl5/5.6.0/i386-linux/perllocal.pod
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- GNUmakefile.am.orig	Tue Dec 18 20:10:48 2001
+++ GNUmakefile.am	Tue Dec 18 20:10:38 2001
@@ -43,21 +43,21 @@
 		-e 'chmod(oct($(PERM_RW))), $$m{$$_} or warn "chmod $(PERM_RW) $$m{$$_}: $$!\n";}' \
 		$(perldir)/AbiWord.pm $(perldir)/blib/man3/AbiWord.3pm
 	@perl -Iblib/lib -I/usr/lib/perl5/5.6.0 -MExtUtils::Install -e "install({@ARGV},'1',0,'0');" \
-		read /usr/lib/perl5/site_perl/5.6.0/i386-linux/auto/AbiWord/.packlist \
-		write /usr/lib/perl5/site_perl/5.6.0/i386-linux/auto/AbiWord/.packlist \
-		blib/lib /usr/lib/perl5/site_perl/5.6.0 \
-		blib/arch /usr/lib/perl5/site_perl/5.6.0/i386-linux \
-		blib/bin /usr/bin \
-		blib/script /usr/bin \
-		blib/man1 /usr/share/man/man1 \
-		blib/man3 /usr/share/man/man3
+		read $(D)/usr/lib/perl5/site_perl/5.6.0/i386-linux/auto/AbiWord/.packlist \
+		write $(D)/usr/lib/perl5/site_perl/5.6.0/i386-linux/auto/AbiWord/.packlist \
+		blib/lib $(D)/usr/lib/perl5/site_perl/5.6.0 \
+		blib/arch $(D)/usr/lib/perl5/site_perl/5.6.0/i386-linux \
+		blib/bin $(D)/usr/bin \
+		blib/script $(D)/usr/bin \
+		blib/man1 $(D)/usr/share/man/man1 \
+		blib/man3 $(D)/usr/share/man/man3
 	@perl -we 'exit unless -f $$ARGV[0];' \
 		-e 'print "WARNING: I have found an old package in\n";' \
 		-e 'print "\t$$ARGV[0].\n";' \
 		-e 'print "Please make sure the two installations are not conflicting\n";' \
-		/usr/lib/perl5/5.6.0/i386-linux/auto/AbiWord
+		$(D)/usr/lib/perl5/5.6.0/i386-linux/auto/AbiWord
 	@perl -I/usr/lib/perl5/5.6.0/i386-linux -I/usr/lib/perl5/5.6.0 -MExtUtils::Command \
-		-e mkpath /usr/lib/perl5/5.6.0/i386-linux
+		-e mkpath $(D)/usr/lib/perl5/5.6.0/i386-linux
 	@perl -e '$$\="\n\n";' -e 'print "=head2 ", scalar(localtime), ": C<", shift, ">", " L<", shift, ">";' \
 		-e 'print "=over 4";' \
 		-e 'while (defined($$key = shift) and defined($$val = shift)){print "=item *";print "C<$$key: $$val>";}' \
@@ -67,5 +67,5 @@
 		LINKTYPE "dynamic" \
 		VERSION "0.01" \
 		EXE_FILES "" \
-		>> /usr/lib/perl5/5.6.0/i386-linux/perllocal.pod
+		>> $(D)/usr/lib/perl5/5.6.0/i386-linux/perllocal.pod.new
 	@echo Appending installation info to /usr/lib/perl5/5.6.0/i386-linux/perllocal.pod
