diff -ur scribus-1.3.5.1.orig/scribus/plugins/tools/2geomtools/lib2geom/CMakeLists.txt scribus-1.3.5.1/scribus/plugins/tools/2geomtools/lib2geom/CMakeLists.txt
--- scribus-1.3.5.1.orig/scribus/plugins/tools/2geomtools/lib2geom/CMakeLists.txt	2008-06-02 01:38:43.000000000 +0300
+++ scribus-1.3.5.1/scribus/plugins/tools/2geomtools/lib2geom/CMakeLists.txt	2010-01-06 20:31:13.000000000 +0200
@@ -16,6 +16,10 @@
 #CB SET(CMAKE_MAKE_PROGRAM "${CMAKE_MAKE_PROGRAM}")
 # We need to pass -fPIC only to lib2geom on amd64 - avoid it elsewhere see: http://www.gentoo.org/proj/en/base/amd64/howtos/index.xml?part=1&chap=3
 
+IF (ARCH_X86_64 EQUAL 1)
+  SET(CMAKE_CXX_FLAGS "$(CMAKE_CXX_FLAGS) -fPIC")
+ENDIF (ARCH_X86_64 EQUAL 1)
+
 OPTION(2GEOM_BUILD_SHARED
   "Build lib2geom and libtoy as shared libraries."
   OFF)
@@ -112,4 +116,4 @@
 #CONFIGURE_FILE( ${CMAKE_SOURCE_DIR}/2geom.pc.in
 #                ${CMAKE_BINARY_DIR}/2geom.pc @ONLY IMMEDIATE )
 #INSTALL(FILES "${CMAKE_BINARY_DIR}/2geom.pc" DESTINATION lib/pkgconfig)
-#
\ No newline at end of file
+#
