--- libwpd-0.8.1-orig/src/lib/WP3FixedLengthGroup.h.old	2005-11-23 00:39:46.000000000 +0100
+++ libwpd-0.8.1/src/lib/WP3FixedLengthGroup.h	2005-11-23 00:40:07.000000000 +0100
@@ -30,8 +30,8 @@
 class WP3FixedLengthGroup : public WP3Part
 {
  public:
-	WP3FixedLengthGroup::WP3FixedLengthGroup(int groupID);
-	static WP3FixedLengthGroup * WP3FixedLengthGroup::constructFixedLengthGroup(WPXInputStream *input, uint8_t groupID);
+	WP3FixedLengthGroup(int groupID);
+	static WP3FixedLengthGroup * constructFixedLengthGroup(WPXInputStream *input, uint8_t groupID);
 
 	const uint8_t getGroup() const { return m_group; } 
  
--- libwpd-0.8.1-orig/src/lib/WP3SingleByteFunction.h.old	2005-11-23 00:41:25.000000000 +0100
+++ libwpd-0.8.1/src/lib/WP3SingleByteFunction.h	2005-11-23 00:41:41.000000000 +0100
@@ -32,7 +32,7 @@
 class WP3SingleByteFunction : public WP3Part
 {
  public:
-	static WP3SingleByteFunction * WP3SingleByteFunction::constructSingleByteFunction(WPXInputStream *input, uint8_t groupID);
+	static WP3SingleByteFunction * constructSingleByteFunction(WPXInputStream *input, uint8_t groupID);
 };
 
 /*class WP3SpaceFunction : public WP3SingleByteFunction
--- libwpd-0.8.1-orig/src/lib/WP5SingleByteFunction.h.old	2005-11-23 00:43:46.000000000 +0100
+++ libwpd-0.8.1/src/lib/WP5SingleByteFunction.h	2005-11-23 00:44:03.000000000 +0100
@@ -32,7 +32,7 @@
 class WP5SingleByteFunction : public WP5Part
 {
  public:
-	static WP5SingleByteFunction * WP5SingleByteFunction::constructSingleByteFunction(WPXInputStream *input, uint8_t groupID);
+	static WP5SingleByteFunction * constructSingleByteFunction(WPXInputStream *input, uint8_t groupID);
 };
 
 class WP5EOLFunction : public WP5SingleByteFunction
--- libwpd-0.8.1-orig/src/lib/WP6FixedLengthGroup.h.old	2005-11-23 00:44:34.000000000 +0100
+++ libwpd-0.8.1/src/lib/WP6FixedLengthGroup.h	2005-11-23 00:44:52.000000000 +0100
@@ -32,7 +32,7 @@
 {
  public:
 	WP6FixedLengthGroup(uint8_t groupID);
-	static WP6FixedLengthGroup * WP6FixedLengthGroup::constructFixedLengthGroup(WPXInputStream *input, uint8_t groupID);
+	static WP6FixedLengthGroup * constructFixedLengthGroup(WPXInputStream *input, uint8_t groupID);
 
  	const uint8_t getGroup() const { return m_group; }
  
--- libwpd-0.8.1-orig/src/lib/WP6SingleByteFunction.h.old	2005-11-23 00:46:34.000000000 +0100
+++ libwpd-0.8.1/src/lib/WP6SingleByteFunction.h	2005-11-23 00:46:47.000000000 +0100
@@ -31,7 +31,7 @@
 class WP6SingleByteFunction : public WP6Part
 {
  public:
-	static WP6SingleByteFunction * WP6SingleByteFunction::constructSingleByteFunction(WPXInputStream *input, uint8_t groupID);
+	static WP6SingleByteFunction * constructSingleByteFunction(WPXInputStream *input, uint8_t groupID);
 };
 
 class WP6SpaceFunction : public WP6SingleByteFunction
--- libwpd-0.8.1-orig/src/lib/WP6VariableLengthGroup.h.old	2005-11-23 00:45:14.000000000 +0100
+++ libwpd-0.8.1/src/lib/WP6VariableLengthGroup.h	2005-11-23 00:45:29.000000000 +0100
@@ -41,7 +41,7 @@
 	WP6VariableLengthGroup(); // WP6VariableLengthGroup should _never_ be constructed, only its inherited classes
 	virtual ~WP6VariableLengthGroup();
 
-	static WP6VariableLengthGroup * WP6VariableLengthGroup::constructVariableLengthGroup(WPXInputStream *input, uint8_t groupID);
+	static WP6VariableLengthGroup * constructVariableLengthGroup(WPXInputStream *input, uint8_t groupID);
 
  protected:
 	void _read(WPXInputStream *input);
--- libwpd-0.8.1-orig/src/lib/WPXHLListener.h.old	2005-11-23 00:40:32.000000000 +0100
+++ libwpd-0.8.1/src/lib/WPXHLListener.h	2005-11-23 00:40:48.000000000 +0100
@@ -157,7 +157,7 @@
 {
 public:
 	WPXHLListener(std::vector<WPXPageSpan *> *pageList, WPXHLListenerImpl *listenerImpl);
-	WPXHLListener::~WPXHLListener();
+	~WPXHLListener();
 
 	void startDocument();
 	void handleSubDocument(uint16_t textPID, const bool isHeaderFooter, WPXTableList tableList, int nextTableIndice);
