--- pisg
+++ pisg	2004-01-11 00:51:16.076387512 +0000
@@ -79,7 +79,7 @@
 
 my $usage = <<END_USAGE;
 Usage: pisg [-ch channel] [-l logfile] [-o outputfile] [-ma maintainer]
-[-f format] [-n network] [-d logdir] [-mo moduledir] [-s] [-v] [-h]
+[-f format] [-ne network] [-d logdir] [-mo moduledir] [-s] [-v] [-h]
 
 -ch --channel=xxx      : Set channel name
 -cc --cchannels=xxx    : Only do this channel from cfg file, give multiple
@@ -89,7 +89,7 @@
 -t  --tag=xxx          : Replace \%t in --outfile by xxx
 -ma --maintainer=xxx   : Channel/statistics maintainer
 -f  --format=xxx       : Logfile format [see FORMATS file]
--n  --network=xxx      : IRC network for the channel
+-ne --network=xxx      : IRC network for the channel
 -d  --dir=xxx          : Analyze all files in this dir. Ignores logfile
 -nf --nfiles=xxx       : Analyze the last xxx files if used with --dir
 -p  --prefix=xxx       : Analyse only files prefixed by xxx in dir
@@ -103,7 +103,7 @@
 
 Example:
 
- \$ pisg -n IRCnet -f xchat -o suid.html -ch \\#channel -l logfile.log
+ \$ pisg -ne IRCnet -f xchat -o suid.html -ch \\#channel -l logfile.log
 
 All options may also be defined by editing the configuration file and
 calling pisg without arguments.
