diff -uNr remoteosd-0.0.1.orig/menu.h remoteosd-0.0.1/menu.h
--- remoteosd-0.0.1.orig/menu.h	2006-05-29 01:02:35.000000000 +0200
+++ remoteosd-0.0.1/menu.h	2006-07-25 19:45:32.325726750 +0200
@@ -10,7 +10,7 @@
 #include <vdr/tools.h>
 #include <vdr/osdbase.h>
 #include <vdr/plugin.h>
-#include "../svdrpservice/svdrpservice.h"
+#include "svdrpservice/svdrpservice.h"
 
 class cRemoteOsdMenu: public cOsdMenu {
 	private:
