--- evolution-1.0.2/mail/mail-display.c.fileurl	Mon Dec 10 20:55:20 2001
+++ evolution-1.0.2/mail/mail-display.c	Mon Feb 25 19:18:49 2002
@@ -311,7 +311,7 @@
 	}
 	
 	command = g_strdup_printf ("%s %s%s &", app->command,
-				   app->expects_uris == GNOME_VFS_MIME_APPLICATION_ARGUMENT_TYPE_URIS ? "file:" : "",
+				   app->expects_uris == GNOME_VFS_MIME_APPLICATION_ARGUMENT_TYPE_URIS ? "file://" : "",
 				   filename);
 	system (command);
 	g_free (command);
