--- evolution-1.1.90/mail/mail-ops.c.subversion	2002-09-24 19:45:37.000000000 +0200
+++ evolution-1.1.90/mail/mail-ops.c	2002-11-07 13:17:20.000000000 +0100
@@ -467,7 +467,7 @@
 	int i;
 	
 	camel_medium_set_header (CAMEL_MEDIUM (message), "X-Mailer",
-				 "Ximian Evolution " VERSION SUB_VERSION " " VERSION_COMMENT);
+				 "Ximian Evolution " VERSION "-" SUB_VERSION " " VERSION_COMMENT);
 	
 	camel_mime_message_set_date (message, CAMEL_MESSAGE_DATE_CURRENT, 0);
 	
