--- evolution-1.1.90/help/C/evolution-C.omf.scrollkeeper	2002-09-24 19:45:25.000000000 +0200
+++ evolution-1.1.90/help/C/evolution-C.omf	2002-11-07 13:31:58.000000000 +0100
@@ -1,4 +1,5 @@
-<?xml version="1.0" standalone="no"?>
+<?xml version="1.0" encoding="ISO-8859-1" standalone="no"?>
+<!DOCTYPE omf PUBLIC "-//OMF//DTD Scrollkeeper OMF Variant V1.0//EN" "http://scrollkeeper.sourceforge.net/dtds/scrollkeeper-omf-1.0/scrollkeeper-omf.dtd">
 <omf>
   <resource>
    <creator>
@@ -12,5 +13,6 @@
     <format mime="text/html"/>
     <identifier url="index.html"/>
     <language code="C"/>
+    <relation seriesid="6a2207d8-8aa4-11d6-8cb4-f0bb5a765891"/>
   </resource>
 </omf>
--- evolution-1.1.90/help/no/evolution-no.omf.scrollkeeper	2002-09-24 19:45:27.000000000 +0200
+++ evolution-1.1.90/help/no/evolution-no.omf	2002-11-07 13:33:03.000000000 +0100
@@ -1,4 +1,5 @@
-<?xml version="1.0" standalone="no"?>
+<?xml version="1.0" encoding="ISO-8859-1" standalone="no"?>
+<!DOCTYPE omf PUBLIC "-//OMF//DTD Scrollkeeper OMF Variant V1.0//EN" "http://scrollkeeper.sourceforge.net/dtds/scrollkeeper-omf-1.0/scrollkeeper-omf.dtd">
 <omf>
   <resource>
    <creator>
@@ -12,5 +13,6 @@
     <format mime="text/sgml"/>
     <identifier url="index.html"/>
     <language code="no"/>
+    <relation seriesid="6a2207d8-8aa4-11d6-8cb4-f0bb5a765891"/>
   </resource>
 </omf>
