--- evolution-1.1.90/shell/e-shell-settings-dialog.c.orig	2002-10-26 08:47:09.000000000 +0200
+++ evolution-1.1.90/shell/e-shell-settings-dialog.c	2002-11-07 15:24:49.000000000 +0100
@@ -32,6 +32,7 @@
 #include "e-util/e-lang-utils.h"
 
 #include <gal/util/e-util.h>
+#include <gal/util/e-unicode-i18n.h>
 
 #include <bonobo/bonobo-widget.h>
 #include <bonobo/bonobo-exception.h>
@@ -93,8 +94,8 @@
 		gdk_pixbuf_ref (icon);
 
 	page = g_new (Page, 1);
-	page->title       = g_strdup (_(title));
-	page->description = g_strdup (_(description));
+	page->title       = g_strdup (U_(title));
+	page->description = g_strdup (U_(description));
 	page->icon        = icon;
 	page->type        = type;
 	page->priority    = priority;
