--- evolution-1.0.2/configure.in.orig	Mon Feb 25 18:26:05 2002
+++ evolution-1.0.2/configure.in	Mon Feb 25 19:01:52 2002
@@ -990,14 +990,9 @@
 if test -z "$with_kde_applnk_path"; then
 	with_kde_applnk_path="$prefix/share/applnk"
 fi
-if test -d "$with_kde_applnk_path" ; then
-	AC_MSG_RESULT("$with_kde_applnk_path" found)
-	AC_DEFINE(HAVE_KDE_APPLNK)
-	KDE_APPLNK_DIR="$with_kde_applnk_path"
-else
-	AC_MSG_RESULT(not found)
-	KDE_APPLNK_DIR=""
-fi
+AC_MSG_RESULT("$with_kde_applnk_path" found)
+AC_DEFINE(HAVE_KDE_APPLNK)
+KDE_APPLNK_DIR="$with_kde_applnk_path"
 AM_CONDITIONAL(HAVE_KDE_APPLNK, test x"$KDE_APPLNK_DIR" != x)
 AC_SUBST(KDE_APPLNK_DIR)
 
