diff -Naur kdemultimedia-3.5.1/akode_artsplugin/akodePlayObject_impl.cpp kdemultimedia-3.5.1-2/akode_artsplugin/akodePlayObject_impl.cpp
--- kdemultimedia-3.5.1/akode_artsplugin/akodePlayObject_impl.cpp	2005-09-10 18:20:04.000000000 +1000
+++ kdemultimedia-3.5.1-2/akode_artsplugin/akodePlayObject_impl.cpp	2006-03-01 10:54:32.000000000 +1000
@@ -83,6 +83,8 @@
 akodePlayObject_impl::~akodePlayObject_impl()
 {
     unload();
+
+    delete m_packetQueue;
 }
 
 bool akodePlayObject_impl::loadPlugin(const string &plugin)
@@ -219,10 +221,9 @@
         delete bufferedDecoder;
         bufferedDecoder = 0;
     }
-#else
-    delete frameDecoder
-    frameDecoder = 0;
 #endif
+    delete frameDecoder;
+    frameDecoder = 0;
     decoder = 0;
     if (buffer != inBuffer)
         delete inBuffer;
