diff --exclude='*~' -Naur tardy-1.16.orig/common/filenamelist/filter/progress.cc tardy-1.16/common/filenamelist/filter/progress.cc
--- tardy-1.16.orig/common/filenamelist/filter/progress.cc	2008-04-28 22:54:43.000000000 +0000
+++ tardy-1.16/common/filenamelist/filter/progress.cc	2008-06-22 15:14:33.000000000 +0000
@@ -23,6 +23,7 @@
 #include <ac/unistd.h>
 #include <ac/sys/types.h>
 #include <ac/sys/stat.h>
+#include <string.h>
 
 #include <filenamelist/filter/progress.h>
 
