diff -uNr obexfs-0.8/fuse/Makefile.am obexfs-0.8.new/fuse/Makefile.am
--- obexfs-0.8/fuse/Makefile.am	2005-12-09 18:58:52.000000000 +0000
+++ obexfs-0.8.new/fuse/Makefile.am	2006-04-02 16:14:30.000000000 +0100
@@ -1,9 +1,9 @@
 ## Process this file with automake to produce Makefile.in
 
-LDADD =				-lfuse -lpthread \
+LDADD =				-lfuse -lpthread -lobexftp \
 				@OPENOBEX_LIBS@ \
 				-lbluetooth \
-				-lobexftp -lmulticobex -lbfb
+				-lmulticobex -lbfb
 
 obexfs_SOURCES =		obexfs.c
 
