--- ./src/Makefile.am.orig	2010-01-15 18:33:26.000000000 +0000
+++ ./src/Makefile.am	2010-01-15 18:42:10.000000000 +0000
@@ -2,10 +2,8 @@
 
 AUTOMAKE_OPTIONS = 1.6
 
-AM_CPPFLAGS = -DGFS_MODULES_DIR=\"$(libdir)/gerris\"
-
-INCLUDES = -I$(top_srcdir) -I$(includedir) \
-           -DG_LOG_DOMAIN=\"Gfs\" $(GTS_CFLAGS)
+AM_CPPFLAGS = -DGFS_MODULES_DIR=\"$(libdir)/gerris\" \
+        -DG_LOG_DOMAIN=\"Gfs\" $(GTS_CFLAGS)
 
 bin_SCRIPTS=gfs-config
 
@@ -116,18 +114,21 @@
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)\
 	-release $(LT_RELEASE) -export-dynamic
 libgfs3D_la_SOURCES = $(SRC)
+libgfs3D_la_LIBADD = $(GTS_LIBS)
 
 libgfs2D_la_LDFLAGS = $(NO_UNDEFINED)\
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)\
 	-release $(LT_RELEASE) -export-dynamic
 libgfs2D_la_SOURCES = $(SRC)
-libgfs2D_la_CFLAGS = $(AM_CFLAGS) -DFTT_2D=1
+libgfs2D_la_CPPFLAGS = $(AM_CPPFLAGS) -DFTT_2D=1
+libgfs2D_la_LIBADD = $(GTS_LIBS)
 
 libgfs2D3_la_LDFLAGS = $(NO_UNDEFINED)\
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)\
 	-release $(LT_RELEASE) -export-dynamic
 libgfs2D3_la_SOURCES = $(SRC)
-libgfs2D3_la_CFLAGS = $(AM_CFLAGS) -DFTT_2D3=1
+libgfs2D3_la_CPPFLAGS = $(AM_CPPFLAGS) -DFTT_2D3=1
+libgfs2D3_la_LIBADD = $(GTS_LIBS)
 
 CLEANFILES = $(BUILT_SOURCES)
 
@@ -139,15 +140,15 @@
 bin_PROGRAMS = gerris2D gerris3D gerris2D3
 
 gerris2D_SOURCES = gerris.c
-gerris2D_CFLAGS = $(AM_CFLAGS) -DFTT_2D=1
-gerris2D_LDADD = $(GFS2D_LIBS)
+gerris2D_CPPFLAGS = $(AM_CPPFLAGS) -DFTT_2D=1
+gerris2D_LDADD = libgfs2D.la
 
 gerris3D_SOURCES = gerris.c
-gerris3D_LDADD = $(GFS3D_LIBS)
+gerris3D_LDADD = libgfs3D.la
 
 gerris2D3_SOURCES = gerris.c
-gerris2D3_CFLAGS = $(AM_CFLAGS) -DFTT_2D3=1
-gerris2D3_LDADD = $(GFS2D3_LIBS)
+gerris2D3_CPPFLAGS = $(AM_CPPFLAGS) -DFTT_2D3=1
+gerris2D3_LDADD = libgfs2D3.la
 
 if DARCS_CONTROLLED
 DARCS_VERSION = darcsversion
