diff -rup ssmtp-2.60/Makefile.in ssmtp-2.60-gentoo/Makefile.in
--- ssmtp-2.60/Makefile.in	2003-08-21 13:27:23.000000000 +1200
+++ ssmtp-2.60-gentoo/Makefile.in	2003-08-21 13:30:41.000000000 +1200
@@ -23,7 +23,7 @@ INSTALLED_REVALIASES_FILE=$(REVALIASES_F
 # Programs
 GEN_CONFIG=$(srcdir)/generate_config
 
-SRCS=ssmtp.c arpadate.c base64.c
+SRCS=ssmtp.c arpadate.c base64.c @SRCS@
 
 OBJS=$(SRCS:.c=.o)
 
diff -rup ssmtp-2.60/configure ssmtp-2.60-gentoo/configure
--- ssmtp-2.60/configure	2003-08-21 13:27:23.000000000 +1200
+++ ssmtp-2.60-gentoo/configure	2003-08-21 13:31:54.000000000 +1200
@@ -1503,6 +1503,7 @@ s%@FFLAGS@%$FFLAGS%g
 s%@DEFS@%$DEFS%g
 s%@LDFLAGS@%$LDFLAGS%g
 s%@LIBS@%$LIBS%g
+s%@SRCS@%$SRCS%g
 s%@exec_prefix@%$exec_prefix%g
 s%@prefix@%$prefix%g
 s%@program_transform_name@%$program_transform_name%g
diff -rup ssmtp-2.60/ssmtp.c ssmtp-2.60-gentoo/ssmtp.c
--- ssmtp-2.60/ssmtp.c	2003-08-21 13:27:23.000000000 +1200
+++ ssmtp-2.60-gentoo/ssmtp.c	2003-08-21 13:31:20.000000000 +1200
@@ -1257,7 +1257,7 @@ int ssmtp(char *argv[])
 	/* Try to log in if username was supplied */
 	if(auth_user) {
 #ifdef MD5AUTH
-		if(auth_pass == (char *)NULL)
+		if(auth_pass == (char *)NULL) {
 			auth_pass = strdup("");
 		}
 
@@ -1271,7 +1271,7 @@ int ssmtp(char *argv[])
 			strncpy(challenge, strchr(buf,' ') + 1, sizeof(challenge));
 
 			memset(buf, 0, sizeof(buf));
-			crammd5(challenge, authUsername, authPassword, buf);
+			crammd5(challenge, auth_user, auth_pass, buf);
 		}
 		else {
 #endif
