--- src/ksmoothdock.h.orig	2008-06-13 22:05:51.000000000 +0000
+++ src/ksmoothdock.h	2008-06-13 22:06:06.000000000 +0000
@@ -27,6 +27,7 @@
 #endif
 
 #include <vector>
+#include <memory>
 
 #include <qwidget.h>
 

--- src/ksmoothdock.cpp.orig	2008-06-13 22:08:26.000000000 +0000
+++ src/ksmoothdock.cpp	2008-06-13 22:08:47.000000000 +0000
@@ -24,6 +24,8 @@
 
 #include <iostream>
 #include <iterator>
+#include <typeinfo>
+
 
 #include <qdir.h>
 

--- src/buttonisedstate.h.orig	2008-06-13 22:10:48.000000000 +0000
+++ src/buttonisedstate.h	2008-06-13 22:11:03.000000000 +0000
@@ -20,6 +20,8 @@
 #ifndef BUTTONISEDSTATE_H
 #define BUTTONISEDSTATE_H
 
+#include <memory>
+#include <typeinfo>
 #include <qobject.h>
 #include "dockstate.h"
 

--- src/normalzoomstate.h.orig	2008-06-13 22:13:10.000000000 +0000
+++ src/normalzoomstate.h	2008-06-13 22:13:23.000000000 +0000
@@ -20,6 +20,8 @@
 #ifndef NORMALZOOMSTATE_H
 #define NORMALZOOMSTATE_H
 
+#include <memory>
+
 #include <qobject.h>
 #include "dockstate.h"
 

--- src/paraboliczoomstate.h.orig	2008-06-13 22:15:19.000000000 +0000
+++ src/paraboliczoomstate.h	2008-06-13 22:15:32.000000000 +0000
@@ -20,6 +20,8 @@
 #ifndef PARABOLICZOOMSTATE_H
 #define PARABOLICZOOMSTATE_H
 
+#include <memory>
+#include <typeinfo>
 #include <qobject.h>
 #include "dockstate.h"
 
--- src/normalzoomstate.cpp.orig	2008-06-13 22:22:16.000000000 +0000
+++ src/normalzoomstate.cpp	2008-06-13 22:22:28.000000000 +0000
@@ -19,6 +19,8 @@
  ***************************************************************************/
 #include "normalzoomstate.h"
 
+
+#include <typeinfo>
 #include <iostream>
 #include <unistd.h>
 

--- src/paraboliczoomstate.cpp.orig	2008-06-13 22:25:09.000000000 +0000
+++ src/paraboliczoomstate.cpp	2008-06-13 22:25:33.000000000 +0000
@@ -19,6 +19,8 @@
  ***************************************************************************/
 #include "paraboliczoomstate.h"
 
+
+#include <cstdlib>
 #include <iostream>
 #include <unistd.h>
 

