diff -urpN gdb-5.3.90_20030710.orig/ltcf-c.sh gdb-5.3.90_20030710/ltcf-c.sh
--- ./ltcf-c.sh	2003-09-20 18:15:02.280018176 +0200
+++ ./ltcf-c.sh	2003-09-20 18:15:18.688523704 +0200
@@ -153,7 +153,7 @@ EOF
     # If the export-symbols file already is a .def file (1st line
     # is EXPORTS), use it as is.
     # If DATA tags from a recent dlltool are present, honour them!
-    archive_expsym_cmds='if test "x`head -1 $export_symbols`" = xEXPORTS; then
+    archive_expsym_cmds='if test "x`head -n 1 $export_symbols`" = xEXPORTS; then
         cp $export_symbols $output_objdir/$soname-def;
       else
         echo EXPORTS > $output_objdir/$soname-def;
diff -urpN gdb-5.3.90_20030710.orig/ltcf-gcj.sh gdb-5.3.90_20030710/ltcf-gcj.sh
--- ./ltcf-gcj.sh	2003-09-20 18:15:02.249022888 +0200
+++ ./ltcf-gcj.sh	2003-09-20 18:15:18.660527960 +0200
@@ -156,7 +156,7 @@ EOF
     # If the export-symbols file already is a .def file (1st line
     # is EXPORTS), use it as is.
     # If DATA tags from a recent dlltool are present, honour them!
-    archive_expsym_cmds='if test "x`head -1 $export_symbols`" = xEXPORTS; then
+    archive_expsym_cmds='if test "x`head -n 1 $export_symbols`" = xEXPORTS; then
         cp $export_symbols $output_objdir/$soname-def;
       else
         echo EXPORTS > $output_objdir/$soname-def;
diff -urpN gdb-5.3.90_20030710.orig/src-release gdb-5.3.90_20030710/src-release
--- ./src-release	2003-09-20 18:15:02.282017872 +0200
+++ ./src-release	2003-09-20 18:15:18.704521272 +0200
@@ -70,7 +70,7 @@ VER = `	if grep 'AM_INIT_AUTOMAKE.*BFD_V
 	elif grep AM_INIT_AUTOMAKE $(TOOL)/configure.in >/dev/null 2>&1; then \
 	  sed < $(TOOL)/configure.in -n 's/AM_INIT_AUTOMAKE[^,]*, *\([^)]*\))/\1/p'; \
 	elif test -f $(TOOL)/version.in; then \
-	  head -1 $(TOOL)/version.in; \
+	  head -n 1 $(TOOL)/version.in; \
 	elif grep VERSION $(TOOL)/Makefile.in > /dev/null 2>&1; then \
 	  sed < $(TOOL)/Makefile.in -n 's/^VERSION *= *//p'; \
 	else \

