diff -urN gdb-5.2.1/gdb/gdbserver/Makefile.in gdb-5.2.1-modified/gdb/gdbserver/Makefile.in
--- gdb-5.2.1/gdb/gdbserver/Makefile.in	Fri Mar 29 13:03:33 2002
+++ gdb-5.2.1-modified/gdb/gdbserver/Makefile.in	Tue Aug 20 16:43:58 2002
@@ -139,7 +139,7 @@
 	n=`echo gdbserver | sed '$(program_transform_name)'`; \
 	if [ x$$n = x ]; then n=gdbserver; else true; fi; \
 	$(INSTALL_PROGRAM) gdbserver $(bindir)/$$n; \
-	$(INSTALL_DATA) $(srcdir)/gdbserver.1 $(man1dir)/$$n.1
+	$(INSTALL_DATA) $(srcdir)/gdbserver.1 ${D}/$(man1dir)/$$n.1
 
 uninstall: force
 	n=`echo gdbserver | sed '$(program_transform_name)'`; \
