diff -urp mpatrol-old/src/config.h mpatrol/src/config.h
--- mpatrol-old/src/config.h	2002-01-08 21:13:59.000000000 +0100
+++ mpatrol/src/config.h	2005-12-22 17:14:25.000000000 +0100
@@ -796,7 +796,7 @@
 
 #ifndef MP_INIT_SUPPORT
 #if SYSTEM == SYSTEM_DGUX || SYSTEM == SYSTEM_DRSNX || \
-    SYSTEM == SYSTEM_DYNIX || SYSTEM == SYSTEM_LINUX || \
+    SYSTEM == SYSTEM_DYNIX || \
     SYSTEM == SYSTEM_SOLARIS || SYSTEM == SYSTEM_UNIXWARE
 #define MP_INIT_SUPPORT 1
 #elif SYSTEM == SYSTEM_FREEBSD || SYSTEM == SYSTEM_NETBSD || \
