From cfd3641501b23ae68beb9656489e8c3077c3dad5 Mon Sep 17 00:00:00 2001
From: Edward Sheldrake <ejsheldrake@gmail.com>
Date: Tue, 05 Oct 2010 10:03:42 +0000
Subject: Fix building with gobject-introspection 0.9.10

---
diff --git a/libupower-glib/Makefile.am b/libupower-glib/Makefile.am
index 83322c4..652f2de 100644
--- a/libupower-glib/Makefile.am
+++ b/libupower-glib/Makefile.am
@@ -60,14 +60,15 @@ EXTRA_DIST =							\
 CLEANFILES = $(BUILT_SOURCES)
 
 if HAVE_INTROSPECTION
-introspection_sources = $(libupower_glib_la_SOURCES)
+introspection_sources = $(libupower_glib_include_HEADERS) $(libupower_glib_la_SOURCES)
 
 UPowerGlib-1.0.gir: libupower-glib.la
 UPowerGlib_1_0_gir_INCLUDES = GObject-2.0 Gio-2.0
 UPowerGlib_1_0_gir_CFLAGS = $(INCLUDES)
-PackageKitGlib_1_0_gir_SCANNERFLAGS = --identifier-prefix=Up --warn-all --add-include-path=$(srcdir)
+UPowerGlib_1_0_gir_SCANNERFLAGS = --identifier-prefix=Up --symbol-prefix=up_ --warn-all --add-include-path=$(srcdir)
 UPowerGlib_1_0_gir_LIBS = libupower-glib.la
 UPowerGlib_1_0_gir_FILES = $(introspection_sources)
+UPowerGlib_1_0_gir_NAMESPACE = UPowerGlib
 INTROSPECTION_GIRS += UPowerGlib-1.0.gir
 
 girdir = $(datadir)/gir-1.0
--
cgit v0.8.3-6-g21f6
