--- freeswan-1.97/libdes/Makefile	2002-04-01 16:48:03.000000000 -0500
+++ freeswan-gentoo/libdes/Makefile	2002-05-23 15:49:15.000000000 -0400
@@ -63,10 +63,10 @@
 #DES_ENC=asm/dx86-sol.o	asm/yx86-sol.o	# solaris format x86 
 #DES_ENC=asm/dx86bsdi.o	asm/yx86basi.o	# bsdi format x86 
 
-LIBDIR=${DESTDIR}/usr/local/lib
-BINDIR=${DESTDIR}/usr/local/bin
-INCDIR=${DESTDIR}/usr/local/include
-MANDIR=${DESTDIR}/usr/local/man
+LIBDIR=${DESTDIR}/usr/lib
+BINDIR=${DESTDIR}/usr/bin
+INCDIR=${DESTDIR}/usr/include
+MANDIR=${DESTDIR}/usr/share/man
 MAN1=1
 MAN3=3
 SHELL=/bin/sh
