diff -ru freeswan-2.01/Makefile.inc freeswan-2.01-gentoo/Makefile.inc
--- freeswan-2.01/Makefile.inc	2003-06-05 11:40:42.000000000 -0500
+++ freeswan-2.01-gentoo/Makefile.inc	2003-08-17 11:40:43.000000000 -0500
@@ -59,7 +59,7 @@
 DESTDIR?=
 
 # "local" part of tree, used in building other pathnames
-INC_USRLOCAL=/usr/local
+INC_USRLOCAL=/usr
 
 # PUBDIR is where the "ipsec" command goes; beware, many things define PATH
 # settings which are assumed to include it (or at least, to include *some*
@@ -93,7 +93,7 @@
 MANPLACES=man3 man5 man8
 
 # where configuration files go
-FINALCONFFILE?=/etc/ipsec.conf
+FINALCONFFILE?=/etc/ipsec/ipsec.conf
 CONFFILE=$(DESTDIR)$(FINALCONFFILE)
 
 FINALCONFDIR?=/etc
@@ -104,10 +104,10 @@
 
 # sample configuration files go into
 INC_DOCDIR?=share/doc
-FINALEXAMPLECONFDIR=${INC_USRLOCAL}/${INC_DOCDIR}/freeswan
+FINALEXAMPLECONFDIR=${INC_USRLOCAL}/${INC_DOCDIR}/freeswan-2.02
 EXAMPLECONFDIR=${DESTDIR}${FINALEXAMPLECONFDIR}
 
-FINALDOCDIR?=${INC_USRLOCAL}/${INC_DOCDIR}/freeswan
+FINALDOCDIR?=${INC_USRLOCAL}/${INC_DOCDIR}/freeswan-2.02
 DOCDIR=${DESTDIR}${FINALDOCDIR}
 
 
@@ -231,7 +231,7 @@
 # installed one in RH 7.2, won't work - you wind up depending upon
 # openssl.
 
-BIND9STATICLIBDIR?=/usr/local/lib
+BIND9STATICLIBDIR?=/usr/lib
 
 # set this to space where a linked/configured tree can be created by 
 # preprhkern4module. Only needed if you are going to be created RPMs
diff -ru freeswan-2.01/lib/libdes/Makefile freeswan-2.01-gentoo/lib/libdes/Makefile
--- freeswan-2.01/lib/libdes/Makefile	2003-02-20 19:32:12.000000000 -0600
+++ freeswan-2.01-gentoo/lib/libdes/Makefile	2003-08-17 11:16:26.000000000 -0500
@@ -60,7 +60,7 @@
 # normally overridden by FreeS/WAN Makefiles anyway
 CFLAG= -O3 -fomit-frame-pointer -I${KLIPSD}/include -I${SRCDIR}
 
-CFLAGS=$(OPTS) $(CFLAG)
+CFLAGS=$(OPTS) $(CFLAG) $(USERCOMPILE)
 CPP=$(CC) -E
 
 # Assember version of des_encrypt*().
diff -ru freeswan-2.01/linux/crypto/ciphers/des/asm/crypt586.pl freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/crypt586.pl
--- freeswan-2.01/linux/crypto/ciphers/des/asm/crypt586.pl	2002-04-24 02:36:37.000000000 -0500
+++ freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/crypt586.pl	2003-08-17 11:14:45.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 #
 # The inner loop instruction sequence and the IP/FP modifications are from
 # Svend Olaf Mikkelsen <svolaf@inet.uni-c.dk>
diff -ru freeswan-2.01/linux/crypto/ciphers/des/asm/des-586.pl freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/des-586.pl
--- freeswan-2.01/linux/crypto/ciphers/des/asm/des-586.pl	2002-04-24 02:36:37.000000000 -0500
+++ freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/des-586.pl	2003-08-17 11:14:45.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 #
 # The inner loop instruction sequence and the IP/FP modifications are from
 # Svend Olaf Mikkelsen <svolaf@inet.uni-c.dk>
diff -ru freeswan-2.01/linux/crypto/ciphers/des/asm/des686.pl freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/des686.pl
--- freeswan-2.01/linux/crypto/ciphers/des/asm/des686.pl	2002-04-24 02:36:37.000000000 -0500
+++ freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/des686.pl	2003-08-17 11:14:45.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 $prog="des686.pl";
 
diff -ru freeswan-2.01/linux/crypto/ciphers/des/asm/desboth.pl freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/desboth.pl
--- freeswan-2.01/linux/crypto/ciphers/des/asm/desboth.pl	2002-04-24 02:36:37.000000000 -0500
+++ freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/desboth.pl	2003-08-17 11:14:45.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 $L="edi";
 $R="esi";
diff -ru freeswan-2.01/linux/crypto/ciphers/des/asm/perlasm/cbc.pl freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/perlasm/cbc.pl
--- freeswan-2.01/linux/crypto/ciphers/des/asm/perlasm/cbc.pl	2002-04-24 02:36:37.000000000 -0500
+++ freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/perlasm/cbc.pl	2003-08-17 11:14:45.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 # void des_ncbc_encrypt(input, output, length, schedule, ivec, enc)
 # des_cblock (*input);
diff -ru freeswan-2.01/linux/crypto/ciphers/des/asm/perlasm/x86asm.pl freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/perlasm/x86asm.pl
--- freeswan-2.01/linux/crypto/ciphers/des/asm/perlasm/x86asm.pl	2002-04-24 02:36:37.000000000 -0500
+++ freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/perlasm/x86asm.pl	2003-08-17 11:14:45.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 # require 'x86asm.pl';
 # &asm_init("cpp","des-586.pl");
diff -ru freeswan-2.01/linux/crypto/ciphers/des/asm/perlasm/x86ms.pl freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/perlasm/x86ms.pl
--- freeswan-2.01/linux/crypto/ciphers/des/asm/perlasm/x86ms.pl	2002-04-24 02:36:37.000000000 -0500
+++ freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/perlasm/x86ms.pl	2003-08-17 11:14:45.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 package x86ms;
 
diff -ru freeswan-2.01/linux/crypto/ciphers/des/asm/perlasm/x86unix.pl freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/perlasm/x86unix.pl
--- freeswan-2.01/linux/crypto/ciphers/des/asm/perlasm/x86unix.pl	2002-04-24 02:36:37.000000000 -0500
+++ freeswan-2.01-gentoo/linux/crypto/ciphers/des/asm/perlasm/x86unix.pl	2003-08-17 11:14:45.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 package x86unix;
 
diff -ru freeswan-2.01/programs/_include/_include.in freeswan-2.01-gentoo/programs/_include/_include.in
--- freeswan-2.01/programs/_include/_include.in	2003-01-06 15:44:04.000000000 -0600
+++ freeswan-2.01-gentoo/programs/_include/_include.in	2003-08-17 11:23:15.000000000 -0500
@@ -47,10 +47,10 @@
 do
 	if test ! -r "$f"
 	then
-		if test ! "$f" = "/etc/ipsec.conf"
+		if test ! "$f" = "/etc/ipsec/ipsec.conf"
 		then
 			echo "#:cannot open configuration file \'$f\'"
-			if test "$f" = "/etc/ipsec.secrets"
+			if test "$f" = "/etc/ipsec/ipsec.secrets"
 			then
 				echo "#:Your secrets file will be created when you start FreeS/WAN for the first time."
 			fi
diff -ru freeswan-2.01/programs/eroute/eroute.5 freeswan-2.01-gentoo/programs/eroute/eroute.5
--- freeswan-2.01/programs/eroute/eroute.5	2002-04-24 02:35:38.000000000 -0500
+++ freeswan-2.01-gentoo/programs/eroute/eroute.5	2003-08-17 11:14:45.000000000 -0500
@@ -203,7 +203,7 @@
 in hexadecimal using Authentication Header protocol (51,
 IPPROTO_AH) with no identies defined for either end.
 .SH FILES
-/proc/net/ipsec_eroute, /usr/local/bin/ipsec
+/proc/net/ipsec_eroute, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_tncfg(5), ipsec_spi(5),
 ipsec_spigrp(5), ipsec_klipsdebug(5), ipsec_eroute(8), ipsec_version(5),
diff -ru freeswan-2.01/programs/eroute/eroute.8 freeswan-2.01-gentoo/programs/eroute/eroute.8
--- freeswan-2.01/programs/eroute/eroute.8	2002-04-24 02:35:38.000000000 -0500
+++ freeswan-2.01-gentoo/programs/eroute/eroute.8	2003-08-17 11:14:45.000000000 -0500
@@ -225,7 +225,7 @@
 .BR www.ietf.org
 to pass in the clear, unprocessed.
 .SH FILES
-/proc/net/ipsec_eroute, /usr/local/bin/ipsec
+/proc/net/ipsec_eroute, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_tncfg(8), ipsec_spi(8),
 ipsec_spigrp(8), ipsec_klipsdebug(8), ipsec_eroute(5)
diff -ru freeswan-2.01/programs/ipsec/ipsec.8 freeswan-2.01-gentoo/programs/ipsec/ipsec.8
--- freeswan-2.01/programs/ipsec/ipsec.8	2003-02-27 10:51:54.000000000 -0600
+++ freeswan-2.01-gentoo/programs/ipsec/ipsec.8	2003-08-17 11:14:45.000000000 -0500
@@ -81,7 +81,7 @@
 .I ipsec
 thinks the IPsec configuration files are stored.
 .SH FILES
-/usr/local/lib/ipsec	usual utilities directory
+/usr/lib/ipsec	usual utilities directory
 .SH ENVIRONMENT
 .PP
 The following environment variables control where FreeS/WAN finds its
diff -ru freeswan-2.01/programs/klipsdebug/klipsdebug.5 freeswan-2.01-gentoo/programs/klipsdebug/klipsdebug.5
--- freeswan-2.01/programs/klipsdebug/klipsdebug.5	2002-04-24 02:35:38.000000000 -0500
+++ freeswan-2.01-gentoo/programs/klipsdebug/klipsdebug.5	2003-08-17 11:14:45.000000000 -0500
@@ -103,7 +103,7 @@
 sockets debugging has been set and everything else is not set.
 .LP
 .SH FILES
-/proc/net/ipsec_klipsdebug, /usr/local/bin/ipsec
+/proc/net/ipsec_klipsdebug, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_tncfg(8), ipsec_eroute(8),
 ipsec_spi(8), ipsec_spigrp(8), ipsec_klipsdebug(5), ipsec_version(5),
diff -ru freeswan-2.01/programs/klipsdebug/klipsdebug.8 freeswan-2.01-gentoo/programs/klipsdebug/klipsdebug.8
--- freeswan-2.01/programs/klipsdebug/klipsdebug.8	2002-04-24 02:35:39.000000000 -0500
+++ freeswan-2.01-gentoo/programs/klipsdebug/klipsdebug.8	2003-08-17 11:14:45.000000000 -0500
@@ -117,7 +117,7 @@
 debugging messages.
 .LP
 .SH FILES
-/proc/net/ipsec_klipsdebug, /usr/local/bin/ipsec
+/proc/net/ipsec_klipsdebug, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_tncfg(8), ipsec_eroute(8),
 ipsec_spi(8), ipsec_spigrp(8), ipsec_klipsdebug(5)
diff -ru freeswan-2.01/programs/lwdnsq/lwdnsq.xml.in freeswan-2.01-gentoo/programs/lwdnsq/lwdnsq.xml.in
--- freeswan-2.01/programs/lwdnsq/lwdnsq.xml.in	2003-02-07 22:03:06.000000000 -0600
+++ freeswan-2.01-gentoo/programs/lwdnsq/lwdnsq.xml.in	2003-08-17 11:24:01.000000000 -0500
@@ -430,7 +430,7 @@
 <refsect1><title>Special IPSECKEY processing</title>
 
 <programlisting>
-/etc/ipsec.d/lwdnsq.conf
+/etc/ipsec/ipsec.d/lwdnsq.conf
 </programlisting>
 
 </refsect1>
diff -ru freeswan-2.01/programs/mailkey/mailkey.in freeswan-2.01-gentoo/programs/mailkey/mailkey.in
--- freeswan-2.01/programs/mailkey/mailkey.in	2003-06-07 14:16:33.000000000 -0500
+++ freeswan-2.01-gentoo/programs/mailkey/mailkey.in	2003-08-17 11:24:35.000000000 -0500
@@ -60,7 +60,7 @@
 
 "$test1st"
 
-Common concerns: This account must be able to read /etc/ipsec.secrets. 
+Common concerns: This account must be able to read /etc/ipsec/ipsec.secrets. 
 If you haven't generated your key yet, please run 'ipsec newhostkey'." 
 exit 0
 }
diff -ru freeswan-2.01/programs/pluto/Makefile freeswan-2.01-gentoo/programs/pluto/Makefile
--- freeswan-2.01/programs/pluto/Makefile	2003-06-19 13:08:30.000000000 -0500
+++ freeswan-2.01-gentoo/programs/pluto/Makefile	2003-08-17 11:17:32.000000000 -0500
@@ -94,7 +94,7 @@
 	-DSHARED_SECRETS_FILE=\"${FINALCONFDIR}/ipsec.secrets\" \
 	-DPOLICYGROUPSDIR=\"${FINALCONFDDIR}/policies\"
 
-ALLFLAGS = $(CPPFLAGS) $(CFLAGS)
+ALLFLAGS = $(CPPFLAGS) $(CFLAGS) $(USERCOMPILE)
 
 # libefence is a free memory allocation debugger
 # Solaris 2 needs -lsocket -lnsl
Only in freeswan-2.01-gentoo/programs/setup: #Makefile#
diff -ru freeswan-2.01/programs/setup/Makefile freeswan-2.01-gentoo/programs/setup/Makefile
--- freeswan-2.01/programs/setup/Makefile	2002-08-10 14:47:06.000000000 -0500
+++ freeswan-2.01-gentoo/programs/setup/Makefile	2003-08-17 11:31:56.000000000 -0500
@@ -33,25 +33,10 @@
 	@rm -f $(BINDIR)/setup
 	@$(INSTALL) $(INSTBINFLAGS) setup $(RCDIR)/ipsec
 	@ln -s $(FINALRCDIR)/ipsec $(BINDIR)/setup
-	-@for i in 0 1 2 3 4 5 6; do mkdir -p $(RCDIR)/../rc$$i.d; done
-	-@cd $(RCDIR)/../rc0.d && ln -f -s ../init.d/ipsec K68ipsec
-	-@cd $(RCDIR)/../rc1.d && ln -f -s ../init.d/ipsec K68ipsec
-	-@cd $(RCDIR)/../rc2.d && ln -f -s ../init.d/ipsec S47ipsec
-	-@cd $(RCDIR)/../rc3.d && ln -f -s ../init.d/ipsec S47ipsec
-	-@cd $(RCDIR)/../rc4.d && ln -f -s ../init.d/ipsec S47ipsec
-	-@cd $(RCDIR)/../rc5.d && ln -f -s ../init.d/ipsec S47ipsec
-	-@cd $(RCDIR)/../rc6.d && ln -f -s ../init.d/ipsec K68ipsec
 
 install_file_list::
 	@echo $(RCDIR)/ipsec
 	@echo $(BINDIR)/setup
-	@echo $(RCDIR)/../rc0.d/K68ipsec
-	@echo $(RCDIR)/../rc1.d/K68ipsec
-	@echo $(RCDIR)/../rc2.d/S47ipsec
-	@echo $(RCDIR)/../rc3.d/S47ipsec
-	@echo $(RCDIR)/../rc4.d/S47ipsec
-	@echo $(RCDIR)/../rc5.d/S47ipsec
-	@echo $(RCDIR)/../rc6.d/K68ipsec
 
 clean::
 	@rm -f setup
diff -ru freeswan-2.01/programs/showhostkey/showhostkey.in freeswan-2.01-gentoo/programs/showhostkey/showhostkey.in
--- freeswan-2.01/programs/showhostkey/showhostkey.in	2003-06-07 14:15:25.000000000 -0500
+++ freeswan-2.01-gentoo/programs/showhostkey/showhostkey.in	2003-08-17 11:18:33.000000000 -0500
@@ -17,7 +17,7 @@
 me="ipsec showhostkey"
 usage="Usage: $me [--file secrets] [--left] [--right] [--txt gateway] [--id id]"
 
-file=/etc/ipsec.secrets
+file=/etc/ipsec/ipsec.secrets
 fmt=""
 gw=
 id=
diff -ru freeswan-2.01/programs/spi/spi.5 freeswan-2.01-gentoo/programs/spi/spi.5
--- freeswan-2.01/programs/spi/spi.5	2002-04-24 02:35:39.000000000 -0500
+++ freeswan-2.01-gentoo/programs/spi/spi.5	2003-08-17 11:14:45.000000000 -0500
@@ -162,7 +162,7 @@
 3858 seconds ago and has been idle for 23 seconds.
 .LP
 .SH FILES
-/proc/net/ipsec_spi, /usr/local/bin/ipsec
+/proc/net/ipsec_spi, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_tncfg(5), ipsec_eroute(5),
 ipsec_spigrp(5), ipsec_klipsdebug(5), ipsec_spi(8), ipsec_version(5),
diff -ru freeswan-2.01/programs/spi/spi.8 freeswan-2.01-gentoo/programs/spi/spi.8
--- freeswan-2.01/programs/spi/spi.8	2002-04-24 02:35:40.000000000 -0500
+++ freeswan-2.01-gentoo/programs/spi/spi.8	2003-08-17 11:14:45.000000000 -0500
@@ -461,7 +461,7 @@
 (4).
 .LP
 .SH FILES
-/proc/net/ipsec_spi, /usr/local/bin/ipsec
+/proc/net/ipsec_spi, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_tncfg(8), ipsec_eroute(8),
 ipsec_spigrp(8), ipsec_klipsdebug(8), ipsec_spi(5)
diff -ru freeswan-2.01/programs/spigrp/spigrp.5 freeswan-2.01-gentoo/programs/spigrp/spigrp.5
--- freeswan-2.01/programs/spigrp/spigrp.5	2002-04-24 02:35:40.000000000 -0500
+++ freeswan-2.01-gentoo/programs/spigrp/spigrp.5	2003-08-17 11:14:45.000000000 -0500
@@ -77,7 +77,7 @@
 machine.
 .LP
 .SH FILES
-/proc/net/ipsec_spigrp, /usr/local/bin/ipsec
+/proc/net/ipsec_spigrp, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_tncfg(5), ipsec_eroute(5),
 ipsec_spi(5), ipsec_klipsdebug(5), ipsec_spigrp(8), ipsec_version(5),
diff -ru freeswan-2.01/programs/spigrp/spigrp.8 freeswan-2.01-gentoo/programs/spigrp/spigrp.8
--- freeswan-2.01/programs/spigrp/spigrp.8	2002-04-24 02:35:41.000000000 -0500
+++ freeswan-2.01-gentoo/programs/spigrp/spigrp.8	2003-08-17 11:14:45.000000000 -0500
@@ -128,7 +128,7 @@
 .BR 0x236 .
 .LP
 .SH FILES
-/proc/net/ipsec_spigrp, /usr/local/bin/ipsec
+/proc/net/ipsec_spigrp, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_tncfg(8), ipsec_eroute(8),
 ipsec_spi(8), ipsec_klipsdebug(8), ipsec_spigrp(5)
diff -ru freeswan-2.01/programs/tncfg/tncfg.5 freeswan-2.01-gentoo/programs/tncfg/tncfg.5
--- freeswan-2.01/programs/tncfg/tncfg.5	2002-04-24 02:35:41.000000000 -0500
+++ freeswan-2.01-gentoo/programs/tncfg/tncfg.5	2003-08-17 11:14:45.000000000 -0500
@@ -75,7 +75,7 @@
 is not connected to any physical device.
 .LP
 .SH "FILES"
-/proc/net/ipsec_tncfg, /usr/local/bin/ipsec
+/proc/net/ipsec_tncfg, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_eroute(5), ipsec_spi(5),
 ipsec_spigrp(5), ipsec_klipsdebug(5), ipsec_tncfg(8), ipsec_version(5),
diff -ru freeswan-2.01/programs/tncfg/tncfg.8 freeswan-2.01-gentoo/programs/tncfg/tncfg.8
--- freeswan-2.01/programs/tncfg/tncfg.8	2002-04-24 02:35:41.000000000 -0500
+++ freeswan-2.01-gentoo/programs/tncfg/tncfg.8	2003-08-17 11:14:45.000000000 -0500
@@ -76,7 +76,7 @@
 physical device.
 .LP
 .SH "FILES"
-/proc/net/ipsec_tncfg, /usr/local/bin/ipsec
+/proc/net/ipsec_tncfg, /usr/bin/ipsec
 .SH "SEE ALSO"
 ipsec(8), ipsec_manual(8), ipsec_eroute(8), ipsec_spi(8),
 ipsec_spigrp(8), ipsec_klipsdebug(8), ipsec_tncfg(5)
diff -ru freeswan-2.01/programs/barf/barf.in freeswan-2.01-gentoo/programs/barf/barf.in
--- freeswan-2.01/programs/barf/barf.in	2003-02-03 22:59:22.000000000 -0600
+++ freeswan-2.01-gentoo/programs/barf/barf.in	2003-08-17 11:23:35.000000000 -0500
@@ -17,7 +17,7 @@
 KERNSRC=${KERNSRC-/usr/src/linux}
 LOGS=${LOGS-/var/log}
 CONFS=${IPSEC_CONFS-/etc}
-CONFDDIR=${IPSEC_CONFDDIR-/etc/ipsec.d}
+CONFDDIR=${IPSEC_CONFDDIR-/etc/ipsec/ipsec.d}
 me="ipsec barf"
 
 # kludge to produce no barf output mentioning policygroups if none are present.
@@ -205,13 +205,13 @@
 	done
 fi
 _________________________ ipsec/ls-libdir
-ls -l ${IPSEC_LIBDIR-/usr/local/lib/ipsec}
+ls -l ${IPSEC_LIBDIR-/usr/lib/ipsec}
 _________________________ ipsec/ls-execdir
-ls -l ${IPSEC_EXECDIR-/usr/local/libexec/ipsec}
+ls -l ${IPSEC_EXECDIR-/usr/libexec/ipsec}
 _________________________ ipsec/updowns
-for f in `ls ${IPSEC_EXECDIR-/usr/local/libexec/ipsec} | egrep updown`
+for f in `ls ${IPSEC_EXECDIR-/usr/libexec/ipsec} | egrep updown`
 do
-	cat ${IPSEC_EXECDIR-/usr/local/libexec/ipsec}/$f
+	cat ${IPSEC_EXECDIR-/usr/libexec/ipsec}/$f
 done
 _________________________ proc/net/dev
 cat /proc/net/dev
