--- qc-usb-messenger-1.7.orig/qc-driver.c	2007-09-02 20:51:16.000000000 +0200
+++ qc-usb-messenger-1.7/qc-driver.c	2007-12-21 03:13:50.000000000 +0100
@@ -845,7 +845,7 @@
 			 usb_rcvintpipe(qc->dev, qc->bEndpointAddress),
 			 id->buffer,
 			 id->rawbufsize,
-			 qc_int_handler,
+			 (usb_complete_t)qc_int_handler,
 			 qc,
 			 16);
 #else
@@ -2206,7 +2206,7 @@
 		urb->context                = qc;
 		urb->pipe                   = usb_rcvisocpipe(qc->dev, QUICKCAM_ISOPIPE);
 		urb->transfer_flags         = URB_ISO_ASAP;
-		urb->complete               = qc_isoc_handler;
+		urb->complete               = (usb_complete_t)qc_isoc_handler;
 		urb->number_of_packets      = ISOC_PACKETS;
 		urb->transfer_buffer        = id->buffer;
 		urb->transfer_buffer_length = ISOC_URBS * ISOC_PACKETS * ISOC_PACKET_SIZE;
@@ -2774,10 +2774,14 @@
 }
 /* }}} */
 /* {{{ [fold] qc_v4l_read(struct video_device *dev, char *buf, unsigned long count, int noblock) */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
-static int qc_v4l_read(struct file *file, char *buf, size_t count, loff_t *ppos)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24)
+static ssize_t qc_v4l_read(struct file *file, char *buf, size_t count, loff_t *ppos)
 #else
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
+static int qc_v4l_read(struct file *file, char *buf, size_t count, loff_t *ppos)
+	#else
 static long qc_v4l_read(struct video_device *dev, char *buf, unsigned long count, int noblock)
+	#endif
 #endif
 {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
@@ -3481,7 +3485,9 @@
 static struct video_device qc_v4l_template = {
 	name:		"QuickCam USB",
 	type:		VID_TYPE_CAPTURE, // | VID_TYPE_SUBCAPTURE,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24)
 	hardware:	VID_HARDWARE_QCAM_USB,
+#endif
 	minor:		-1,
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
 	release:	qc_v4l_release,
@@ -3658,7 +3664,7 @@
 	qc->input_ptr->name = "Quickcam snapshot button";
 	qc->input_ptr->private = qc;
 	qc->input_ptr->evbit[0] = BIT(EV_KEY);
-	qc->input_ptr->keybit[LONG(BTN_0)] = BIT(BTN_0);
+	qc->input_ptr->keybit[(BTN_0)/BITS_PER_LONG] = BIT(BTN_0);
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
 	qc->input_ptr->id.bustype = BUS_USB;


--- qc-usb-messenger-1.7.orig/qcset.c	2007-09-02 20:54:56.000000000 +0200
+++ qc-usb-messenger-1.7/qcset.c	2007-12-21 03:14:57.000000000 +0100
@@ -376,7 +376,7 @@
 }
 /* }}} */
 /* {{{ [fold] void pnm_open(unsigned char *name, unsigned int *width, unsigned int *height) */
-void pnm_open(unsigned char *name, unsigned int *width, unsigned int *height) {
+void pnm_open(char *name, unsigned int *width, unsigned int *height) {
 	char c1,c2,c3;
 	unsigned int maxval;
 
