diff -ru src/python/python_config.c src.new/python/python_config.c
--- src/python/python_config.c	2001-11-30 18:03:23.000000000 +0100
+++ src.new/python/python_config.c	2004-11-30 15:33:59.000000000 +0100
@@ -235,7 +235,7 @@
            if (func) func(GTK_WIDGET(widget),&my_var_info[i]);
          }
          break;
-      default:
+      default: ;
      }
   } 
 #ifdef WITH_GNOME
@@ -609,7 +609,7 @@
 
     break;
 
-    default:
+    default:;
     }
   }
 
