=== modified file 'configure.in'
--- configure.in	2008-11-21 18:47:07 +0000
+++ configure.in	2008-12-28 19:57:19 +0000
@@ -88,8 +88,10 @@
 	AS_HELP_STRING([--enable-geoip],
 	               [Enable GeoIP support (default: disabled)]),
     [
-        GEOIP="$enableval"
-        GEOIP_LIB="-lGeoIP"
+	if test "$enableval" != "no"; then
+        	GEOIP="$enableval"
+        	GEOIP_LIB="-lGeoIP"
+	fi
     ],
     [GEOIP="no"])
 

