Index: x11vnc/sslhelper.c
===================================================================
RCS file: /cvsroot/libvncserver/libvncserver/x11vnc/sslhelper.c,v
retrieving revision 1.31
retrieving revision 1.32
diff --unified -r1.31 -r1.32
--- x11vnc/sslhelper.c	16 May 2007 21:29:44 -0000	1.31
+++ x11vnc/sslhelper.c	15 Jun 2007 03:19:06 -0000	1.32
@@ -34,6 +34,11 @@
 pid_t openssl_last_helper_pid = 0;
 char *openssl_last_ip = NULL;
 
+static char *certret = NULL;
+static int certret_fd = -1;
+static mode_t omode;
+char *certret_str = NULL;
+
 void raw_xfer(int csock, int s_in, int s_out);
 
 #if !LIBVNCSERVER_HAVE_LIBSSL
@@ -1315,11 +1320,6 @@
 	return 1;
 }
 
-static char *certret = NULL;
-static int certret_fd = -1;
-static mode_t omode;
-char *certret_str = NULL;
-
 void accept_openssl(int mode, int presock) {
 	int sock = -1, listen = -1, cport, csock, vsock;	
 	int peerport = 0;
